---
title: "starterCultureDiversiy_metagenome"
author: "Vincent Somerville"
date: "18/12/2019"
output: html_document
editor_options: 
  chunk_output_type: console
---
##Setup

```{r setup ,echo=FALSE}
knitr::opts_chunk$set(message=FALSE, echo=FALSE, eval=FALSE)
```

#Wetlab
#### colony pick genotypes

```{r}
library(readr)
library(plyr)
library(tidyverse)
library(ggplot2)
strain_counts_of_plates <- read_delim("~/Desktop/Projects/2020_strainDelineation/01_logs/GENOTYPE_count.csv",  "\t", escape_double = FALSE, trim_ws = TRUE)

# strain_counts_of_plates$sample <-
strain_counts_of_plates$`starter culture` <-  as.character(strain_counts_of_plates$`starter culture`)
strain_counts_of_plates$genotypes_extended <- paste(strain_counts_of_plates$`starter culture`,strain_counts_of_plates$species,strain_counts_of_plates$genotype,sep="_")
# ggplot(strain_counts_of_plates,aes(x=`starter culture`,color=genotype,fill=genotype))+geom_bar(aes(y=(..counts..)/sum(..counts..)),stat="identity")+facet_wrap(~species,scales = "free")
genotypePlot <- ggplot(strain_counts_of_plates,aes(x=`starter culture`,color=genotype,fill=genotype,text =paste("genotype:", genotypes_extended)))+geom_bar(aes(y=count),stat="identity")+labs(x="",y="genotype counts")+facet_wrap(~species,scales = "free")+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))
genotypePlot

table(interaction(strain_counts_of_plates$genotype,strain_counts_of_plates$species))
grep("NA",strain_counts_of_plates$genotypes_extended,invert = TRUE) %>% length()
grep("NA",strain_counts_of_plates$genotypes_extended,invert = TRUE) %>% length()
strain_counts_of_plates[grep("NA",strain_counts_of_plates$genotypes_extended,invert = TRUE),] %>% filter(species=="S_thermophilus") %>% nrow()
strain_counts_of_plates[grep("NA",strain_counts_of_plates$genotypes_extended,invert = TRUE),] %>% filter(species=="L_delbrueckii") %>% nrow()

 svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/genotypeAbundance.svg",width=6,height=3)


genotypePlot+theme(legend.position = "none")

dev.off()


library(plotly)
ggp <- ggplotly(genotypePlot)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/genotypeAbundance.html")

unique(strain_counts_of_plates$genotypes_extended)
strain_counts_of_plates$genotypes_extended <- revalue(strain_counts_of_plates$genotypes_extended,c("RMK101_L_delbrueckii_A"="dominant","RMK105_L_delbrueckii_B"="dominant","RMK115_L_delbrueckii_B"="dominant","RMK124_L_delbrueckii_J"="dominant","RMK150_L_delbrueckii_J"="dominant","RMK190_L_delbrueckii_B" ="dominant", "RMK202_L_delbrueckii_T"="dominant", "RMK203_L_delbrueckii_J"="dominant", "RMK280_L_delbrueckii_R"="dominant","RMK302_L_delbrueckii_B"="dominant","RMK305_L_delbrueckii_H"="dominant", "RMK101_S_thermophilus_a"="dominant","RMK105_S_thermophilus_c"="dominant","RMK115_S_thermophilus_c"="dominant","RMK124_S_thermophilus_e"="dominant","RMK150_S_thermophilus_a"="dominant", "RMK190_S_thermophilus_I"="dominant", "RMK202_S_thermophilus_p"="dominant","RMK203_S_thermophilus_l"="dominant","RMK280_S_thermophilus_m"="dominant", "RMK302_S_thermophilus_s"="dominant","RMK305_S_thermophilus_e"="dominant")) 
strain_counts_of_plates$genotypes_extended <- ifelse(strain_counts_of_plates$genotypes_extended=="dominant","dominant","subdominant")
strain_counts_of_plates$genotype_dominant <- NA
for (i in 1:nrow(strain_counts_of_plates)) {
  strain_counts_of_plates[i,"genotype_dominant"] <- ifelse(strain_counts_of_plates[i,"genotypes_extended"]=="dominant",as.character(strain_counts_of_plates[i,"genotype"]),NA)
  
}

strain_counts_of_plates$names <- as.character(gsub("RMK","",strain_counts_of_plates$`starter culture`))
genotypePlot <- ggplot(strain_counts_of_plates,aes(x=names,fill=genotype_dominant,color=genotype,text =paste("genotype:", genotypes_extended)))+geom_bar(aes(y=count),stat="identity", colour="#FF9999")+labs(x="",y="colony counts")+facet_wrap(~species,scales = "free")+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+scale_fill_discrete(na.value="grey81")
genotypePlot

 # png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/genotypeAbundance_onlyDominant.png", width = 2400, height = 1200,res=300)
svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/genotypeAbundance_onlyDominant.svg",width=6,height=3)


genotypePlot

dev.off()

###---------------------
##for mid-thesis

genotypePlot <- ggplot(strain_counts_of_plates,aes(x=names,fill=genotype_dominant,color=genotype,text =paste("genotype:", genotypes_extended)))+geom_bar(aes(y=count),stat="identity", colour="#FF9999")+labs(x="",y="colony counts")+facet_wrap(~species,scales = "free",nrow = 2)+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+scale_fill_discrete(na.value="grey81")+scale_y_continuous(breaks =c(0,25,50,75,96),labels=c(0,25,50,75,96))
genotypePlot

 # png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/genotypeAbundance_onlyDominant.png", width = 2400, height = 1200,res=300)
svg("~/Desktop/mid_thesis/report/figures/20200430/chapter3/genotypeAbundance_onlyDominant.svg",width=2.5,height=5)


genotypePlot

dev.off()


```

## d/l lactate
The ration d- to l-lactat gives an idea of L.delbreuckii to S. thermophilus
Here, I plot the ratios measured in 2015 of all Starter cultures in use.

```{r}
library(readr)
library(ggplot2)
library(tidyverse)
d_l_lactat_2015 <- read_delim("~/Desktop/Projects/2020_strainDelineation/06_RMKStrainStock/d_l_lactat_2015.csv",  "\t", escape_double = FALSE, trim_ws = TRUE)
d_l_lactat_2015$sample <- paste("RMK",as.character(d_l_lactat_2015$sample))

colors <- c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#80e5ff","#008181","#00dcdc")


plotl_d <- ggplot(d_l_lactat_2015,aes(x=sample,y=L_LACTAT,group=sample,fill=sample))+geom_boxplot()+theme_classic()+ylim(0,100)+theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1),legend.position = "none")+labs(x="",y= "l-lactat [%]")+scale_fill_manual(values = colors)

plotl_d

# svg("~/Desktop/Projects/2020_strainDelineation/06_RMKStrainStock/plot_d_l_lactat.svg",width=4,height=4)
png("~/Desktop/Projects/2020_strainDelineation/06_RMKStrainStock/plot_d_l_lactat.png", width = 1800, height = 1600,res=300)
plotl_d
dev.off()

```




##Cell counts
Here, I first download the cell counts sheet from google sheets. 
This sheet also includes the columns abount the generation time, doubling time and survival rate. 
From this information I plot the following parameters

```{r}
##----------------------
##import data
##----------------------
library(readr)
library(ggplot2)
library(plyr)
library(tidyverse)
library(patchwork)
CellCount <- read_csv("~/Desktop/Projects/2020_StarterEvolutionExperiment/01_cellCount/rmk202_test_cellCounts.csv", skip = 2)

CellCount$step <- revalue(CellCount$step,c("first_passage"="first passage","second_passage"="second passage","freeze_dry"="freeze dry &\nstorage"))



##----------------------
##generation
##----------------------
colnames(CellCount)
table(CellCount$step)
CellCount_generations <- CellCount %>% filter(step %in% c("first passage","second passage")) %>% filter(!is.na(generations_all)) %>% select(c(sample,treatment,step,generations_all,generations_sterm,generations_ldel)) %>% gather(., sample, generations, c(generations_all,generations_sterm,generations_ldel), factor_key=TRUE,na.rm = TRUE) 
colors <- c("grey","#EB4D4D","#10B552")
CellCount_generations$sample <- revalue(CellCount_generations$sample,c("generations_all"="All","generations_sterm"="S. thermophilus","generations_ldel"="L. delbrueckii"))

generationsPlot <- ggplot(CellCount_generations,aes(x=step,y=generations,fill=sample))+geom_boxplot()+theme_classic()+theme(legend.title = element_blank(),legend.position = "top",axis.title.x = element_blank())+scale_fill_manual(values=colors)

##----------------------
##survival rate
##----------------------

colnames(CellCount)
table(CellCount$step)
CellCount_survival <- CellCount %>% filter(step %in% c("freeze dry &\nstorage")) %>% filter(!is.na(generations_all)) %>% select(c(sample,treatment,step,survival_rate_all,survival_rate_sterm,survival_rate_ldel)) %>% gather(., sample, "survival rate", c(survival_rate_all,survival_rate_sterm,survival_rate_ldel), factor_key=TRUE,na.rm = TRUE) %>% filter(sample!="survival_rate_ldel")
colors <- c("grey","#EB4D4D","#10B552")
CellCount_survival$`survival rate`
CellCount_survival$sample <- revalue(CellCount_survival$sample,c("survival_rate_all"="All","survival_rate_sterm"="S. thermophilus","survival_rate_ldel"="L. delbrueckii"))

SurvivalPlot <- ggplot(CellCount_survival,aes(x=step,y=`survival rate`,fill=sample))+geom_boxplot()+theme_classic()+theme(legend.title = element_blank(),legend.position = "top",axis.title.x = element_blank())+scale_fill_manual(values=colors)
SurvivalPlot
##----------------------
##survival rate
##----------------------



svg("~/Desktop/mid_thesis/report/figures/20200430/chapter3/generations_survivalRate.svg",width=4,height=4)
# png("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.png", width = 1800, height = 1600,res=300)
generationsPlot+SurvivalPlot+ plot_layout(ncol=2,widths = c(2,0.8))
dev.off()
```


#Cellcounts again
This is from the experiment on the 20200901 were I grew all dominant ldel and Sterm in isoaltion and in co-culture aswell as all complex RMK mixes. 
Here I analyse the coressponding CFU counts

```{r}
library(readr)
library(ggsignif)
library(Rmisc)
dominantStrains_RMK_CFU <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/13_CFU/20200904_dominantStrains_RMK_CFU .csv", skip = 1) %>% dplyr::select(c(-Comments,-dilution)) %>% filter(experiment=="normalGrowth")


dominantStrains_RMK_CFU_long <- gather(dominantStrains_RMK_CFU, plate, CFU, "SR9.3":"BM_1", factor_key=TRUE,na.rm = TRUE) %>% filter(!plate %in% c("BM","BM_1"))

# dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("SR9.3_1"="SR9.3","MR_1"="MR","BM_1"="BM"))
dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("SR9.3_1"="Sterm","MR_1"="Ldel","SR9.3"="Sterm","MR"="Ldel"))
# dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("SR9.3_1"="SR9.3","MR_1"="MR","BM_1"="BM"))

dominantStrains_RMK_CFU_long$CFU <- abs(dominantStrains_RMK_CFU_long$CFU)


dominantStrains_RMK_CFU_long$category <- revalue(dominantStrains_RMK_CFU_long$category,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus","pairwise"="pairwise\nco-culture","RMK"="complex\nco-culture"))
dominantStrains_RMK_CFU_long$category = factor(dominantStrains_RMK_CFU_long$category, levels=c("S.thermophilus",  "L.delbrueckii",  "pairwise\nco-culture" , "complex\nco-culture"))
dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus"))


tgc <- summarySE(dominantStrains_RMK_CFU_long, measurevar="CFU", groupvars=c("category","plate"))
tgc

# tgc$category

# tgc$category <- revalue(tgc$category,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus","pairwise"="pairwise\nco-culture","RMK"="complex\nco-culture"))
tgc$category = factor(tgc$category, levels=c("S.thermophilus",  "L.delbrueckii",  "pairwise\nco-culture" , "complex\nco-culture"))
# tgc$plate <- revalue(tgc$plate,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus"))

annot_1 <- t.test(dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "S.thermophilus" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"], dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "pairwise\nco-culture" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"])$p.value

annot_2 <- t.test(dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "S.thermophilus" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"], dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "complex\nco-culture" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"])$p.value

# all_colours <- (c("#EB4D4D","#10B552") )

all_colours <- (c("#ff6f69","#88d8b0") )


#
require(scales)

CFU_plot <- ggplot(tgc, aes(x=category, y=CFU, fill=plate,group=plate)) + theme_classic()+
    # lims(y=c(0,10^10))+
   scale_y_continuous(trans="log10",breaks = trans_breaks("log10", function(x) 10^x),
    labels = trans_format("log10", math_format(10^.x)))+
theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 10, hjust = 1),legend.title = element_blank())+  labs(x="",y="CFU/ml")+
    geom_bar(position=position_dodge(), stat="identity") +
  scale_fill_manual(values=all_colours)+
    # geom_signif(annotation=formatC(annot_1, digits=1),)+
    geom_errorbar(aes(ymin=CFU-ci, ymax=CFU+ci),
                  width=.2,                    # Width of the error bars
                  position=position_dodge(.9))


CFU_plot

svg("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/CFU_pairwise_rmkstocks.svg",width=4.5,height=3.5)
# png("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/HPLC_pairwise_rmkstocks.png", width = 1800, height = 1600,res=300)
CFU_plot
dev.off()

####----------------------------------------------------
#again
####----------------------------------------------------

library(readr)
library(ggsignif)
library(Rmisc)
dominantStrains_RMK_CFU <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/13_CFU/dominantStrains_RMK_CFU_new.csv", skip = 1) %>% dplyr::select(c(-Comments,-dilution)) 


dominantStrains_RMK_CFU_long <- gather(dominantStrains_RMK_CFU, plate, CFU, "SR9.3":"BM_1", factor_key=TRUE,na.rm = TRUE) %>% filter(!plate %in% c("BM","BM_1"))

# dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("SR9.3_1"="SR9.3","MR_1"="MR","BM_1"="BM"))
dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("SR9.3_1"="Sterm","MR_1"="Ldel","SR9.3"="Sterm","MR"="Ldel"))
# dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("SR9.3_1"="SR9.3","MR_1"="MR","BM_1"="BM"))

dominantStrains_RMK_CFU_long$CFU <- abs(dominantStrains_RMK_CFU_long$CFU)

unique(dominantStrains_RMK_CFU_long$category )

dominantStrains_RMK_CFU_long$category <- revalue(dominantStrains_RMK_CFU_long$category,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus","pairwise"="pairwise\nco-culture","RMK"="complex\nco-culture","Strepto_IN_Lacto"="S.thermophilus in\n L.delbrueckii supernatant","Strepto_IN_Strepto"="S.thermophilus in\n S.thermophilus supernatant","Lacto_IN_Lacto"="L.delbrueckii in\n L.delbrueckii supernatant","Lacto_IN_Strepto"="L.delbrueckii in\n S.thermophilus supernatant","Ldel_phControlled"="L.delbrueckii \n pH-controlled","Sterm_phControlled"="S.thermophilus \n pH-controlled"))


dominantStrains_RMK_CFU_long$category = factor(dominantStrains_RMK_CFU_long$category, levels=c( "L.delbrueckii",  "S.thermophilus","L.delbrueckii in\n L.delbrueckii supernatant", "L.delbrueckii in\n S.thermophilus supernatant","L.delbrueckii \n pH-controlled","S.thermophilus in\n L.delbrueckii supernatant","S.thermophilus in\n S.thermophilus supernatant","S.thermophilus \n pH-controlled","pairwise\nco-culture" , "complex\nco-culture"))
dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus"))


tgc <- summarySE(dominantStrains_RMK_CFU_long, measurevar="CFU", groupvars=c("category","plate"))
tgc

# tgc$category

# tgc$category <- revalue(tgc$category,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus","pairwise"="pairwise\nco-culture","RMK"="complex\nco-culture"))
tgc$category = factor(tgc$category, levels=c( "L.delbrueckii",  "S.thermophilus","L.delbrueckii in\n L.delbrueckii supernatant", "L.delbrueckii in\n S.thermophilus supernatant","L.delbrueckii \n pH-controlled","S.thermophilus in\n L.delbrueckii supernatant","S.thermophilus in\n S.thermophilus supernatant","S.thermophilus \n pH-controlled","pairwise\nco-culture" , "complex\nco-culture"))
# tgc$plate <- revalue(tgc$plate,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus"))

# annot_1 <- t.test(dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "S.thermophilus" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"], dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "pairwise\nco-culture" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"])$p.value

# annot_2 <- t.test(dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "S.thermophilus" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"], dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "complex\nco-culture" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"])$p.value

# all_colours <- (c("#EB4D4D","#10B552") )

all_colours <- (c("#ff6f69","#88d8b0") )


#
require(scales)

tgc$ymin <- tgc$CFU-tgc$ci
tgc$ymax <- tgc$CFU+tgc$ci


CFU_plot <- ggplot(tgc, aes(x=category, y=CFU, fill=plate,group=plate)) + theme_classic()+
    # lims(y=c(0,10^10))+
   scale_y_continuous(trans="log10",breaks = trans_breaks("log10", function(x) 10^x),
    labels = trans_format("log10", math_format(10^.x)))+
theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 7, hjust = 1),legend.title = element_blank(),legend.position = "none")+  labs(x="",y="CFU/ml")+
    geom_bar(position=position_dodge(), stat="identity") +
  scale_fill_manual(values=all_colours)+
    # geom_signif(annotation=formatC(annot_1, digits=1),)+
    geom_errorbar(aes(ymin=CFU-se, ymax=CFU+se),
                  width=.2,                    # Width of the error bars
                  position=position_dodge(.9))




CFU_plot

svg("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/CFU_pairwise_rmkstocks_new.svg",width=4.5,height=3.5)
# png("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/HPLC_pairwise_rmkstocks.png", width = 1800, height = 1600,res=300)
CFU_plot
dev.off()


svg("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/CFU_and_Aciditiy_pairwise_rmkstocks_new.svg",width=5.5,height=7)
# png("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/HPLC_pairwise_rmkstocks.png", width = 1800, height = 1600,res=300)
k0plot+theme(axis.text.x=element_blank())+CFU_plot+ plot_layout(nrow=2)
dev.off()

```


##HPLC

```{r}
library(readr)
data_vincent_hplc <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/13_HPLC/data_vincent_hplc.csv","\t", escape_double = FALSE, trim_ws = TRUE)%>% filter(`Lactose [g/L]`>1)%>% filter(!SAMPLE_NAME %in% c("r+0","0+c","r+c"))

# data_vincent_hplc$category

data_vincent_hplc$category <- revalue(data_vincent_hplc$category,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus","pairwise"="pairwise\nco-culture","RMK"="complex\nco-culture","prts_pos"="prts +\nS.thermophilus"))
data_vincent_hplc$category = factor(data_vincent_hplc$category, levels=c("S.thermophilus",  "L.delbrueckii",  "pairwise\nco-culture" , "complex\nco-culture","prts +\nS.thermophilus"))

##--------------------------
#lactose
##--------------------------
data_vincent_hplc$`Lactose [g/L]` <- data_vincent_hplc$`Lactose [g/L]`*3
lactosePlot <- ggplot(data_vincent_hplc,aes(x=category,y=`Lactose [g/L]`,fill=category))+geom_boxplot()+theme_classic()+theme(axis.text.x = element_text(angle = 45, vjust = 1, size = 10, hjust = 1),legend.title = element_blank(),legend.position = "none")+labs(x="")


##--------------------------
#lactate
##--------------------------
data_vincent_hplc$`Lactate [g/L]` <- data_vincent_hplc$`Lactate [g/L]`*3
lacPlot <- ggplot(data_vincent_hplc,aes(x=category,y=`Lactate [g/L]`,fill=category))+geom_boxplot()+theme_classic()+theme(axis.text.x = element_text(angle = 45, vjust = 1, size = 10, hjust = 1),legend.title = element_blank(),legend.position = "none")+labs(x="")



##--------------------------
#glucose
##--------------------------
data_vincent_hplc$`Glucose [g/L]` <- data_vincent_hplc$`Glucose [g/L]`*3
gluPlot <- ggplot(data_vincent_hplc,aes(x=category,y=`Glucose [g/L]`,fill=category))+geom_boxplot()+theme_classic()+theme(axis.text.x = element_text(angle = 45, vjust = 1, size = 10, hjust = 1),legend.title = element_blank(),legend.position = "none")+labs(x="")



##--------------------------
#galactose
##--------------------------
data_vincent_hplc$`Galactose [g/L]` <- data_vincent_hplc$`Galactose [g/L]`*3
galPlot <- ggplot(data_vincent_hplc,aes(x=category,y=`Galactose [g/L]`,fill=category))+geom_boxplot()+theme_classic()+theme(axis.text.x = element_text(angle = 45, vjust = 1, size = 10, hjust = 1),legend.title = element_blank(),legend.position = "none")+labs(x="")

(lactosePlot+lacPlot)/
(galPlot+gluPlot)

svg("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/HPLC_pairwise_rmkstocks.svg",width=6,height=7)
# png("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/HPLC_pairwise_rmkstocks.png", width = 1800, height = 1600,res=300)
(lactosePlot+lacPlot)/
(galPlot+gluPlot)
dev.off()


##--------------------------
#overall sugar in system
##--------------------------

data_vincent_hplc$totalSugar <- (data_vincent_hplc$`Lactose [g/L]`+data_vincent_hplc$`Glucose [g/L]`+data_vincent_hplc$`Galactose [g/L]`+data_vincent_hplc$`Lactate [g/L]`)

totalPlot <- ggplot(data_vincent_hplc,aes(x=category,y=totalSugar,fill=category))+geom_boxplot()+theme_classic()+theme(axis.text.x = element_text(angle = 45, vjust = 1, size = 12, hjust = 1),legend.title = element_blank(),legend.position = "none")+labs(x="")

totalPlot

```

##Proteolytic activity

```{r}
library(readr)
Vincent_Prot_Akt_again <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/12_proteas_activity/Vincent_Prot_Akt.csv","\t", escape_double = FALSE, trim_ws = TRUE)

Vincent_Prot_Akt_again_long <- gather(Vincent_Prot_Akt_again, replicate, gylcerine,c("replicate1","replicate2"), factor_key=TRUE,na.rm = TRUE) 

ggplot(Vincent_Prot_Akt_again_long,aes(x=group,y=gylcerine,fill=group))+geom_boxplot()+theme_classic()
Vincent_Prot_Akt_again_long$group <- revalue(Vincent_Prot_Akt_again_long$group,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus","pairwise"="pairwise\nco-culture","RMK"="complex\nco-culture"))



tgc <- summarySE(Vincent_Prot_Akt_again_long, measurevar="gylcerine", groupvars="group")
# tgc


tgc$group = factor(tgc$group, levels=c("S.thermophilus",  "L.delbrueckii",  "pairwise\nco-culture" , "complex\nco-culture","Milk","Prts+ S. thermophilus","L.helveticus"))


all_colours <- (c("#d7301f",
"#238b45",
"#4eb3d3",
"#0868ac",
"#d9d9d9",
"#bdbdbd",
"#969696") )


#
require(scales)

proteolytic_plot <- ggplot(tgc, aes(x=group, y=gylcerine, fill=group,group=group)) + theme_classic()+
    # lims(y=c(0,10^10))+
   # scale_y_continuous(trans="log10",breaks = trans_breaks("log10", function(x) 10^x),
    # labels = trans_format("log10", math_format(10^.x)))+
theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 10, hjust = 1),legend.title = element_blank(),legend.position = "none")+  labs(x="",y="Glycine-concentration\n[mmol/l]")+
    geom_bar(position=position_dodge(), stat="identity") +
  scale_fill_manual(values=all_colours)+
    # geom_signif(annotation=formatC(annot_1, digits=1),)+
    geom_errorbar(aes(ymin=gylcerine-ci, ymax=gylcerine+ci),
                  width=.2)

proteolytic_plot


svg("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/proteolyticAcivity_pairwise_rmkstocks.svg",width=4,height=5.5)
# png("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/HPLC_pairwise_rmkstocks.png", width = 1800, height = 1600,res=300)
proteolytic_plot
dev.off()
```


##interaction prediction
what does interaction look like

```{r}

library(readr)
library(tidyverse)
library(reshape2)
starter_culture_interaction_matrix <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/starter_culture_interaction_matrix.csv", "\t", escape_double = FALSE, trim_ws = TRUE) 

starter_culture_interaction_matrix_new <- starter_culture_interaction_matrix %>% column_to_rownames(var="X1") %>% as.matrix %>% t()

meltR = melt(starter_culture_interaction_matrix_new, id = "Portion")
meltR$species1 <-  str_split_fixed(meltR$Var1, fixed(":"), 2)[,1]
meltR$starterculture1 <-  str_split_fixed(meltR$Var1, fixed(":"), 2)[,2]
meltR$species2 <-  str_split_fixed(meltR$Var2, fixed(":"), 2)[,1]
meltR$starterculture2 <-  str_split_fixed(meltR$Var2, fixed(":"), 2)[,2]

# creat Heatmap
ggheatmap <- ggplot(meltR, aes(starterculture1, starterculture2, fill = value))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "blue", high = "red", mid = "white",
   midpoint = 0.75, limit = c(0,1), space = "Lab",
   name="fitness") +
  theme_minimal()+ # minimal theme
  labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)

##save heatmap
# svg("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.svg",width=2.5,height=2.5)
png("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.png", width = 1800, height = 1600,res=300)
print(ggheatmap)
dev.off()

```



##Data preperation
####Download Metagenomes

```{bash}

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/99_log

vim /data/Project/2020_StarterCultureDiversity/99_log/20200111_download.txt

vim /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change.txt

##--------------download special case

cd /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/

wget -i /data/Project/2020_StarterCultureDiversity/99_log/20200111_download_new.txt

http://gtf-owncloud.unil.ch/index.php/s/wWUzyjKJJOpe9y2/download?path=%2F&files=V42_S41_L002_R1_001.fastq.gz
cp download?path=%2F&files=V42_S41_L002_R1_001.fastq.gz Lyo_105_76_5_13-4-R1.fastq.gz
Lyo_105_76_5_13

##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================


mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/99_log

vim /data/Project/2020_StarterCultureDiversity/99_log/20200625_download.txt

vim /data/Project/2020_StarterCultureDiversity/99_log/20200625_sampleNames2change.txt

##--------------download special case

cd /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/

wget -i /data/Project/2020_StarterCultureDiversity/99_log/20200625_download.txt



```


####name change

create a file with the labels and the names of the file in each a column

```{bash}
name=$(cut -d '/' -f 5 /data/Project/2020_StarterCultureDiversity/99_log/20200111_download.txt|head -1)
name=$(cut -d '/' -f 5 /data/Project/2020_StarterCultureDiversity/99_log/20200111_download.txt|head -2|tail -1)
#for name in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz |grep "^V")
for name in $(cut -d '/' -f 5 /data/Project/2020_StarterCultureDiversity/99_log/20200111_download.txt)
do

  name_01=$(echo $name |cut -d '_' -f 1)
  Lane=$(echo $name |cut -d '_' -f 2|sed 's/L//g')
  Read=$(echo $name |cut -d '_' -f 3)  
  
  Lane_new=$((Lane+2))

  newName=$(awk -F "\t" -v name="$name_01" '{if($2==name) {print $1}}' /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change.txt)
  
  echo ${newName}"-L"${Lane_new}"-"${Read}".fastq.gz"
    
  mv /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${name} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${newName}-${Lane_new}-${Read}.fastq.gz
    
done
##-------qualtiy control

ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/|cut -d '-' -f 1|sort|uniq -c |wc -l
ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/|cut -d '-' -f 1|sort|uniq -c 

##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================
for name in $(cut -d '/' -f 5 /data/Project/2020_StarterCultureDiversity/99_log/20200625_download.txt)
do

  name_01=$(echo $name |cut -d '_' -f 1)
  Lane=$(echo $name |cut -d '_' -f 2|sed 's/L//g')
  Read=$(echo $name |cut -d '_' -f 3)  
  
  Lane_new=$((Lane+2))

  newName=$(awk -F "\t" -v name="$name_01" '{if($2==name) {print $1}}' /data/Project/2020_StarterCultureDiversity/99_log/20200625_sampleNames2change.txt)
  
  echo ${newName}"-L"${Lane_new}"-"${Read}".fastq.gz"
    
  mv /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/${name} /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/${newName}-${Lane_new}-${Read}.fastq.gz
    
done
##-------qualtiy control

ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/|cut -d '-' -f 1|sort|uniq -c |wc -l
ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/|cut -d '-' -f 1|sort|uniq -c 


```



concatenenat the fastq.gz files

```{bash}
rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/02_concatenate/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/02_concatenate/gz

for name in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change.txt)
  do
  names=$(echo ${name}"-")
   echo ${names}
  
  forward_01=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep "$names"  |grep "\-L1\-R1.fastq.gz")
  forward_02=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep "$names"  |grep "\-L2\-R1.fastq.gz")
  forward_03=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep "$names"  |grep "\-3\-R1.fastq.gz")
  forward_04=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep "$names"  |grep "\-4\-R1.fastq.gz")

  
  
  cat /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${forward_01} \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${forward_02}  \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${forward_03} \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${forward_04} > \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/02_concatenate/gz/${name}-R1.fastq.gz
  
  reverse_01=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep "$names"  |grep "\-L1\-R2.fastq.gz")
  reverse_02=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep "$names"  |grep "\-L2\-R2.fastq.gz")
  reverse_03=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep "$names"  |grep "\-3\-R2.fastq.gz")
  reverse_04=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep "$names"  |grep "\-4\-R2.fastq.gz")

  cat /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${reverse_01} \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${reverse_02}  \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${reverse_03}  \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${reverse_04}  > \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/02_concatenate/gz/${name}-R2.fastq.gz
  
  done

```



####Adapter removal


In the following we clip adaptors with [trim-galore](https://github.com/FelixKrueger/TrimGalore)

```{bash trimGalore}
threads=38
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz
      mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz
  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/{afterTrimgalore,logs}
num=1
for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change.txt |head -25)
  do
  echo ${num}"/46  :" ${names}
  num=$((num+1))
  
  
  

      trim_galore -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz/ \
        --fastqc_args "--outdir /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/afterTrimgalore/" \
        /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/02_concatenate/gz/${names}-R1.fastq.gz   \
        /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/02_concatenate/gz/${names}-R2.fastq.gz
        
        
        
  done 
  rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/afterTrimgalore//Multiqc
  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/afterTrimgalore//Multiqc
    /home/vincent/.local/bin/multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/afterTrimgalore/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/afterTrimgalore//Multiqc
    
    
 
##------------------------------------
##===============================
##add 20200624
##------------------------------------
##===============================   
    threads=38
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/gz
      mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/gz
  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/99_QCs/{afterTrimgalore,logs}
num=1
for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20200625_sampleNames2change.txt )
  do
  echo ${num}"/41  :" ${names}
  num=$((num+1))
  
  
  #    trim_galore -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/gz/ \
   #     --fastqc_args "--outdir /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/afterTrimgalore/" \
    #    /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/${names}-9-R1.fastq.gz   \
     #   /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/${names}-9-R2.fastq.gz
      
     rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq
echo "unzipping..."

#gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/gz/${names}-9-R1_val_1.fq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R1_val_1.fq &
#gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/gz/${names}-9-R2_val_2.fq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R2_val_2.fq
echo "unzipped"

echo "removing cow reads"



kneaddata --max-memory 110000m --no-discordant -t 37 -p 37 --input /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R1_val_1.fq \
  --input /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R2_val_2.fq \
  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  \
  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
  --output /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/
  
rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}

gzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/${names}*neaddata_paired_1.fastq > /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz

gzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/${names}*neaddata_paired_2.fastq > /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz

  rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/
        
  done 
  rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/99_QCs/afterTrimgalore//Multiqc
  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/99_QCs/afterTrimgalore//Multiqc
    /home/vincent/.local/bin/multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/99_QCs/afterTrimgalore/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/99_QCs/afterTrimgalore//Multiqc
    
```


```{bash trimGalore}
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/afterTrimgalore//Multiqc/multiqc_report.html /home/vincent/Desktop/Projects/2019_Nano_Meta/Illumina_analysis/

```


####KneadData

Here, we filter out the cow genome contamination reads

```{bash}

##===================
##remove reads
##===================

##--------------
##experiment
##--------------
names=ws_101_19
threads=38
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/fq/
num=1
for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change.txt)
  do
  

  echo ${num}"/37  :" ${names}
  num=$((num+1))
#rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/${names}/
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/${names}/
echo "unzipping..."

#gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz/${names}-R1_val_1.fq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/fq/${names}-R1_val_1.fq
#gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz/${names}-R2_val_2.fq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/fq/${names}-R2_val_2.fq
echo "unzipped"
echo "removing cow reads"



#kneaddata --max-memory 110000m --no-discordant -t 37 -p 37 --input /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/fq/${names}-R1_val_1.fq \
#  --input /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/fq/${names}-R2_val_2.fq \
#  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  \
#  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
#  --output /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/${names}/
rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}
gzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/${names}/${names}*neaddata_paired_1.fastq > /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz

gzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/${names}/${names}*neaddata_paired_2.fastq > /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz

done


##------------------------------------
##===============================
##add 20200624
##------------------------------------
##===============================   

names=Vers_280_20
threads=38
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/
num=1
for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20200625_sampleNames2change.txt)
  do
  

  echo ${num}"/37  :" ${names}
  num=$((num+1))
#rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/
rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/
echo "unzipping..."

gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/gz/${names}-R1_val_1.fq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R1_val_1.fq
gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/gz/${names}-R2_val_2.fq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R2_val_2.fq
echo "unzipped"
echo "removing cow reads"



kneaddata --max-memory 110000m --no-discordant -t 37 -p 37 --input /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R1_val_1.fq \
  --input /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R2_val_2.fq \
  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  \
  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
  --output /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/
  
rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}

gzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/${names}*neaddata_paired_1.fastq > /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz

gzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/${names}*neaddata_paired_2.fastq > /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz

done


##-------------move------------
cp /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change.txt /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with_New.txt

num=1
for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20200625_sampleNames2change.txt)
  do
  echo ${names}
  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/
  
  cat /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz

  cat /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz


grep "${names}" /data/Project/2020_StarterCultureDiversity/99_log/20200625_sampleNames2change.txt >> /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with_New.txt
  
  done

##===============================   
###------------------------------------
add rmk202 samples
###------------------------------------
##===============================   

cp /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with_New.txt /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt

 for names in $(awk -F "\t" '{print $1}' /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_newNames.txt)
 do

 NewName=$(grep "${names}" /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_newNames.txt |awk -F "\t" '{print $2}')
echo "transfer "$names" to " $NewName
#rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${NewName}/
# mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${NewName}/
#gzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq > \
#   /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${NewName}/${NewName}-R1_kneaddata.fq.gz
#gzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq  > \
#/data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${NewName}/${NewName}-R2_kneaddata.fq.gz


grep "${names}" /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_newNames.txt |awk -F "\t" '{OFS="\t"} {print $2,$1}' >> /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt
done

```

## Analysis

###mOTUs

```{bash}


num=1
for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt)
  do
  echo ${num}"/95  :" ${names}
  num=$((num+1))
  rm -r /data/Project/2020_StarterCultureDiversity/02_mOTU/${names}/

  mkdir -p /data/Project/2020_StarterCultureDiversity/02_mOTU/${names}/
 
/home/vincent/apps/mOTUs_v2/motus profile -f /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz \
  -r /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz -o /data/Project/2020_StarterCultureDiversity/02_mOTU/${names}/ -t 37 -n ${names}_mOTU

done


###-------------------------------------
##reduce data
###-------------------------------------

rm /data/Project/2020_StarterCultureDiversity/02_mOTU/all_mOTUs_prepforR.txt
for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt)
  do
#  echo "-----------------------------------------"
#  echo ${names}

grep "^#" -v /data/Project/2020_StarterCultureDiversity/02_mOTU/${names}/* |grep -v "0.0000000000" |grep -v "sp." |awk -F "\t" -v sampless="${names}" 'BEGIN{OFS="\t"} {print $1, $2, sampless}' |sed 's/\[.*\]//g' |sed 's/-1/NA NA/g'|awk -F "\t" 'BEGIN{OFS="\t"} {print $1,$2,$3}' >> /data/Project/2020_StarterCultureDiversity/02_mOTU/all_mOTUs_prepforR.txt

done

scp -r vincent@130.223.49.145:/data/Project/2020_StarterCultureDiversity/02_mOTU/all_mOTUs_prepforR.txt /home/vincent/Desktop/Projects/2019_Nano_Meta/Illumina_analysis/


```
naming
```{r}
###----------------------------
##make a names file
###----------------------------
library(readr)
library(plyr)
library(tidyverse)
motus_abundance <- read_delim("~/Desktop/Projects/2019_Nano_Meta/Illumina_analysis/all_mOTUs_prepforR.txt", "\t", escape_double = FALSE, col_names = c("species","abundance","sample"),trim_ws = TRUE) #%>% filter(!sample %in% c("C_202_18_02","C_202_18_01"))
# motus_abundance$sample
sort(unique(motus_abundance$sample))

unique_samples <- motus_abundance$sample %>% unique()
unique_samples_type <- str_split_fixed(unique_samples, "_", 5)[,1] #%>%  mutate(Valence = ifelse(Participant == 1:41, rep(0:2), 0))
unique_samples_culture <- str_split_fixed(unique_samples, "_", 5)[,2]
unique_samples_date <- str_split_fixed(unique_samples, "_", 3)[,3]


# dput(as.character(unique_samples_date))
# unique_samples_dateNew <- c("2002 Jul",    "1977" ,     "1976 Mar 31" ,"1997 Apr 7" , "1975 Apr" ,   "2011 Mar 11" ,"1976 Apr 6" , "2019"   ,   "2017"  ,    "2011 Mar"  ,  "2019"  ,    "2019"   ,   "2016"    ,  "2013" ,     "2017"   ,   "2010 Feb" ,   "1976 Mar 12" ,"2019"    ,  "2019"  ,    "1975" ,     "2019"  , "2019"  ,    "2019"   ,   "2019"   ,   "2016"   ,   "1996 Jan" ,   "2016"   ,   "2008"  ,    "2018"  ,    "2014 Oct"  , "1995 Apr"  ,  "2009"   , "2019","1975" ,     "1976 Mai 11", "1976 Nov",   "1996 Jan","1976 Feb"  ,  "1976 Apr 2",  "1977","1976 Mai 13", "2016 Apr", "1976 Apr 1" , "1976"    ,  "1976 Apr" , "1976 Jan", "2020", "2020", "2020", "2020", "2020", "2020", "2020",
# "2020", "2020", "2020", "2020", "2020", "2020", "2020", "2020", "2020", "2020", "2020", 
# "2020", "2020", "2020", "2020", "2020", "2020", "2020", "2020", "2020", "2020", "2020", 
# "2020", "2020", "2020", "2020", "2020", "2020", "2020", "2020", "2020", "2020", "2020", 
# "2020", "2018", "2012", "2014", "2014", "2018", "1996", "2018 Jan", "2018 Feb" )

unique_samples_dateNew <- revalue(unique_samples_date,c("02_7"="2002 Jul",    "77"="1977" ,     "76_3_31"="1976 Mar 31" ,"76_4_7"="1976 Apr 7" ,"75_4" = "1975 Apr" ,   "11_3_11"="2011 Mar 11", "76_4_6"="1976 Apr 6" , "19"="2019"     ,   "11_5"= "2011 Mar"  ,   "16" ="2016"    , "13"= "2013" ,     "17"= "2017"   ,  "10_2"  = "2010 Feb" ,   "76_5_12"="1976 Mar 12"  ,   "75"= "1975"    ,   "96_1"   ="1996 Jan" ,     "08"= "2008"  ,   "14_10" =  "2014 Oct"  ,  "95_3" ="1995 Apr"  , "09" = "2009"   ,"76_5_11" = "1976 Mai 11","76_11" = "1976 Nov","76_2"="1976 Feb"  ,  "76_4_2" = "1976 Apr 2", "76_5_13"="1976 Mai 13", "16_4"= "2016 Apr",  "76_4_1"= "1976 Apr 1" ,  "76"="1976"    ,  "76_4"= "1976 Apr" ,"76_1" = "1976 Jan", "18"   ="2018"  ,"14" ="2014","96"="1996","18_01"="2018 Jan","18_02"="2018 Feb","12"="2012","20"="2020"))



naming_DF <- data.frame(namesOLD=unique_samples,sampleType=unique_samples_type,CultureName=unique_samples_culture,SampleDateOld=unique_samples_date,SampleDateNew=unique_samples_dateNew)
dput(as.character(sort(unique(naming_DF$namesOLD))))

naming_DF$simpleNames <- revalue(naming_DF$namesOLD,c("FSK_101_75"="starter 1","Lyo_101_76_4_7"="starter 2","Lyo_101_77"="starter 3","Lyo_101_02_7"="starter 4","Lyo_101_10_2"="starter 5","Lyo_101_16"="starter 6","ws_101_19"="starter 7","FSK_105_75"="starter 1","Lyo_105_76_4_1"="starter 2","Lyo_105_76_4_2"="starter 3","FSK_105_76_4_6"="starter 4","Lyo_105_76_2"="starter 5","Lyo_105_76_1"="starter 6","Lyo_105_76_5_11"="starter 7","Lyo_105_76_5_13"="starter 8","Lyo_105_76_5_12"="starter 9","Lyo_105_76_3_31"="starter 10","Lyo_105_11_5"="starter 11","Ws_105_19"="starter 12","Lyo_115_09"="starter 1","Ws_115_19"="starter 2","FSK_124_75_4"="starter 1","Lyo_124_08"="starter 2", "Ws_124_19"="starter 3","Lyo_150_76_4"="starter 1","Lyo_150_76_11"="starter 2","Lyo_150_11_3_11"="starter 3","Lyo_150_14_10"="starter 4","Lyo_150_18"="starter 5","Ws_150_19"="starter 6","Lyo_190_95_3"="starter 1","Lyo_190_17"="starter 2","Ws_190_19"="starter 3","Lyo_202_96"="starter 1","Lyo_202_12"="starter 2","Lyo_202_14"="starter 3","S_202_14"="starter 4","Ws_202_18"="starter 5","S_202_18"="starter 6", "C_202_18_02"="starter 7","C_202_18_01"="starter 8","Lyo_203_77"="starter 1","Lyo_203_96_1"="starter 2","Lyo_203_16"="starter 3","Ws_203_19"="starter 4","Lyo_280_16"="starter 1","Lyo_280_16_4"="starter 2", "Ws_280_19"="starter 3","Lyo_302_13"="starter 1", "Ws_302_19"="starter 2","FSK_305_76"="starter 1", "Lyo_305_96_1"="starter 2","Lyo_305_17"="starter 3","Ws_305_19"="starter 4",
"Vers_101_20"="starter 8", "Vers_105_20"="starter 13", 
"Vers_115_20"="starter 3", "Vers_124_20"="starter 4", "Vers_150_20"="starter 7", "Vers_190_20"="starter 4", "Vers_203_20"="starter 5", 
"Vers_280_20"="starter 4", "Vers_302_20"="starter 3", "Vers_305_20"="starter 5",
"Therm_101_20"="starter 9", "Therm_105_20"="starter 14", "Therm_115_20"="starter 4", "Therm_124_20"="starter 5", 
"Therm_150_20"="starter 8", "Therm_190_20"="starter 5", "Therm_203_20"="starter 6", "Therm_280_20"="starter 5", 
"Therm_302_20"="starter 4", "Therm_305_20"="starter 6",
"24h_101_20"="starter 10", "24h_105_20"="starter 15", "24h_115_20"="starter 5", "24h_124_20"="starter 6", "24h_150_20"="starter 9", 
"24h_190_20"="starter 6", "24h_203_20"="starter 7", "24h_280_20"="starter 6", "24h_302_20"="starter 5", "24h_305_20"="starter 7",
"M17x_101_20"="starter 11", "M17X_105_20"="starter 16", "M17X_115_20"="starter 6", 
"M17X_124_20"="starter 7", "M17x_150_20"="starter 10", "M17x_190_20"="starter 7", "M17x_202_20"="starter 9", "M17x_203_20"="starter 8", 
"M17x_280_20"="starter 7", "M17x_302_20"="starter 6", "M17x_305_20"="starter 8"))

table(naming_DF$simpleNames)

naming_DF$sampleType <- revalue(naming_DF$sampleType,c("ws"="Ws"))
naming_DF$namesNew <- paste0(naming_DF$sampleType," RMK",naming_DF$CultureName,"\n",naming_DF$SampleDateNew)
7

naming_DF <- naming_DF[order(unique_samples_culture, unique_samples_dateNew),]
naming_DF$order <- 1:nrow(naming_DF)

  write.csv(naming_DF,"~/Desktop/Projects/2020_StarterCultureDiversity/98_script/naming_file_new.csv")

```


```{r}
###----------------------------
##script
###----------------------------

library(readr)
library(ggplot2)

library(plyr)
library(tidyverse)
# naming_DF <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/98_script/naming_file.csv")
naming_DF <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/98_script/naming_file_new.csv")

motus_abundance <- read_delim("/home/vincent/Desktop/Projects/2019_Nano_Meta/Illumina_analysis/all_mOTUs_prepforR.txt", "\t", escape_double = FALSE, col_names = c("species","abundance","sample"),trim_ws = TRUE)
unique(motus_abundance$simpleNames)

# motus_abundance %>%  filter(is.na(simpleNames))

motus_abundance <- merge(motus_abundance,naming_DF,by.y = "namesOLD",by.x="sample",all.x = TRUE)
motus_abundance$simpleNames = factor(motus_abundance$simpleNames, levels=c("starter 1",  "starter 2",  "starter 3" , "starter 4" , "starter 5",  "starter 6",  "starter 7" , "starter 8" , "starter 9" ,"starter 10" ,"starter 11"  ,"starter 12", "starter 13" ,"starter 14" ,"starter 15"  ,"starter 16"))

# motus_abundance <- motus_abundance[order(unique_samples_culture, unique_samples_dateNew),]
# 
# newNames <- setNames(naming_DF$namesNew,naming_DF$namesOLD)
# motus_abundance$sample <- revalue(motus_abundance$sample, newNames)
# motus_abundance$sample = factor(motus_abundance$sample, levels=c(as.character(naming_DF$namesNew)))
# newNames <- setNames(as.character(naming_DF$CultureName),naming_DF$namesNew)
# motus_abundance$culture <- revalue(as.character(motus_abundance$sample),newNames)
# levels(motus_abundance$sample)
motus_abundance$abundance <- 100*motus_abundance$abundance
# motus_abundance$species <- 100*motus_abundance$abundance
# dim(table(motus_abundance$species))


motus_abundance$species = factor(motus_abundance$species, levels=c("Lactobacillus delbrueckii","Streptococcus thermophilus","Lactobacillus helveticus/gallinarum","Lactococcus raffinolactis"    , "Thermus thermophilus/parvatiensis"  ,"Cutibacterium acnes","Lactococcus lactis","Rothia mucilaginosa","Burkholderia gladioli","Streptococcus agalactiae", "Streptococcus parauberis" ,  "Lactobacillus fermentum/oris","Wohlfahrtiimonas chitiniclastica","Leuconostoc mesenteroides/suionicum","NA NA"))

motus_abundance$species_new<- revalue(motus_abundance$species,c("Lactobacillus helveticus/gallinarum"="Lactobacillus helveticus","Lactococcus raffinolactis"="subdominant"    , "Thermus thermophilus/parvatiensis"="subdominant"   ,"Cutibacterium acnes"="subdominant" ,"Lactococcus lactis"="subdominant" ,"Rothia mucilaginosa"="subdominant" ,"Burkholderia gladioli"="subdominant" ,"Streptococcus agalactiae"="subdominant" , "Streptococcus parauberis"="subdominant"  ,  "Lactobacillus fermentum/oris"="subdominant" ,"Wohlfahrtiimonas chitiniclastica"="subdominant" ,"Leuconostoc mesenteroides/suionicum"="subdominant" ,"NA"="subdominant","NA NA"="subdominant"))

motus_abundance[is.na(motus_abundance$species_new),"species_new"] <- "subdominant"


all_colours <- (c("#fec3b7ff","#0081a7ff","grey70","grey88") ) 

# all_colours <- (c("#10B552","#EB4D4D","grey70","grey") ) 
#fec3b7ff
#0081a7ff
  library(viridis)
colorRest <- viridis_pal(option = "D")(12) 

# all_colours <- (c("#EB4D4D","#10B552", "grey51",colorRest) ) 



##--------------------
##ploting

motus_abundance
pMOTUs <-  ggplot( data = motus_abundance,aes(y = abundance, x = simpleNames, group=species_new,fill = species_new))+ geom_bar( stat="identity")+
  facet_grid(~CultureName,scales = "free",space = "free")+
    labs("",
         x="",
         y="relative Abundance")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
  scale_fill_manual(values=all_colours)+
  scale_color_manual(values=all_colours)+
  scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous( expand = c(0, 0)) +
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=3),
      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  pMOTUs
  
  
  
    svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/AbundanceSpecies_all.svg",width=10,height=3)
   #png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/AbundanceSpecies_new.png", width = 1800, height = 1500,res=300)


pMOTUs

dev.off()
  

  
##=========================
##split cheese
##=========================

  ##------------
  ##emmentaler
  ##------------ 
  
 motus_abundance_ementaler <- motus_abundance %>% filter(CultureName %in% c("101","105","115","124","150","190")) 
pMOTUs_emmentaler <-  ggplot( data = motus_abundance_ementaler,aes(y = abundance, x = simpleNames, group=species_new,fill = species_new))+ geom_bar( stat="identity")+
  facet_wrap(~CultureName,scales = "free",ncol = 6)+
    # facet_wrap(~CultureName,scales = "free", labeller = label_wrap_gen(multi_line=FALSE))+
    labs("",
         x="",
         y="")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
  scale_fill_manual(values=all_colours)+
  scale_color_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=3),
      #axis.text.x = element_blank(),
          legend.position="none",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  pMOTUs_emmentaler
  
  ##------------
  ##Gruyere
  ##------------ 
  
 motus_abundance_gruyere <- motus_abundance %>% filter(CultureName %in% c("202","203","280")) 
pMOTUs_gruyere <-  ggplot( data = motus_abundance_gruyere,aes(y = abundance, x = simpleNames, group=species_new,fill = species_new))+ geom_bar( stat="identity")+
  facet_wrap(~CultureName,scales = "free",ncol = 6)+
    # facet_wrap(~CultureName,scales = "free", labeller = label_wrap_gen(multi_line=FALSE))+
    labs("",
         x="",
         y="relative Abundance")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
  scale_fill_manual(values=all_colours)+
  scale_color_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=3),
      #axis.text.x = element_blank(),
          legend.position="none",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  pMOTUs_gruyere
  
   ##------------
  ##Gruyere
  ##------------ 
  
 motus_abundance_Sbrinz <- motus_abundance %>% filter(CultureName %in% c("302","305")) 
pMOTUs_brinz <-  ggplot( data = motus_abundance_Sbrinz,aes(y = abundance, x = simpleNames, group=species_new,fill = species_new))+ geom_bar( stat="identity")+
  facet_wrap(~CultureName,scales = "free",ncol = 5)+
    # facet_wrap(~CultureName,scales = "free", labeller = label_wrap_gen(multi_line=FALSE))+
    labs("",
         x="",
         y="")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
  scale_fill_manual(values=all_colours)+
  scale_color_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=3),
      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
      
          )

  pMOTUs_brinz
  source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R") ##for plotting only legends
legend <- g_legend(pMOTUs_brinz)

##--------------------
##merge
##--------------------
library(patchwork)  
  pMOTUs_emmentaler / pMOTUs_gruyere /pMOTUs_brinz
  
  
  p1 <-  pMOTUs_emmentaler /
     (pMOTUs_gruyere +  plot_spacer()+ plot_layout(ncol=2,widths = c(2.9, 3)))/
    (pMOTUs_brinz +theme( legend.position="none") + plot_spacer()+ legend+ plot_spacer()+ plot_layout(ncol=4,widths = c(1.9,1,1, 2)))
  p1
  
   svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/AbundanceSpecies_new_again.svg",width=9,height=7)
   #png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/AbundanceSpecies_new.png", width = 1800, height = 1500,res=300)


p1

dev.off()


##----------------------------------
#only current cheese starter cultures
##----------------------------------



 motus_abundance_newest <- motus_abundance %>% filter(namesNew %in% c("Ws RMK101\n2019","Ws RMK105\n2019","Ws RMK115\n2019","Ws RMK124\n2019","Ws RMK150\n2019","Ws RMK190\n2019","Ws RMK202\n2018","Ws RMK203\n2019","Ws RMK280\n2019","Ws RMK302\n2019","Ws RMK305\n2019"))
motus_abundance_newest$namesNew
motus_abundance_newest$CultureName <- as.character(motus_abundance_newest$CultureName)

pMOTUs_newest <-  ggplot( data = motus_abundance_newest,aes(y = abundance, x = CultureName, group=species_new,fill = species_new))+ geom_bar( stat="identity")+
    # facet_wrap(~CultureName,scales = "free", labeller = label_wrap_gen(multi_line=FALSE))+
    labs("",
         x="",
         y="relative abundance")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
  scale_fill_manual(values=all_colours)+
  scale_color_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
      
          )

  pMOTUs_newest
  
    svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/AbundanceSpecies_onlyCurrentSample.svg",width=4,height=3)
   #png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/AbundanceSpecies_new.png", width = 1800, height = 1500,res=300)


pMOTUs_newest

dev.off()
```

###mapping to RMK202

```{bash}

scp /home/vincent/Downloads/sequence.fasta vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/Ldelv_mag.fasta

```


```{bash}
#add L.helveticus

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta \
  /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/Ldelv_mag.fasta > \
  /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv.fasta
  
###================
##description file
###================
#cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202.txt > /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames.txt

#for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt)


threads=37
samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt ##the file with all sample names
logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref
Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv.fasta


names=Ws_150_19
#/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 index ${Assembly}
#/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 mem -t 16 ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz/${namezzz}-R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz/${namezzz}-R2_val_2.fq.gz |head
##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}
rm ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
num=1

for names in $(cut -f 1 ${samplesss})
  do
  echo ${num}"/95  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 #/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 index ${Assembly} 
  
#  /home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 mem -t ${threads} ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
bwa mem -t ${threads} ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${ynames}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   

tail -95 ${BaseLocation}/log/multiqc_logfile_finalGenome.txt > ${BaseLocation}/log/multiqc_logfile_finalGenome_onlyMETA.txt
   
    rm -r ${BaseLocation}/MultiQC_onlyMETA
 
  mkdir -p ${BaseLocation}/MultiQC_onlyMETA
  
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome_onlyMETA.txt \
    -o ${BaseLocation}/MultiQC_onlyMETA

#/home/vincent/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/MultiQC/multiqc_report.html

scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/MultiQC/multiqc_report.html /Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/

##-------------------------------------------------  
##-----check read coverage
##-------------------------------------------------  
name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
threads=37
names=Konserve_202


num=1
rm -r ${BaseLocation}/log/mappings.txt
rm -r ${BaseLocation}/log/mappings_species.txt


for names in $(cut -f 1 ${samplesss})
  do
mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')

mapped_sterm=$(grep "CP046134" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)
mapped_ldel=$(grep "CP046131" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)

mkdir -p ${BaseLocation}/log
echo -e ${names}"\tonlyMeta\t"${mapped} >> ${BaseLocation}/log/mappings.txt

echo -e ${names}"\tLactobacillus delbrueckii\t"$mapped_ldel >> ${BaseLocation}/log/mappings_species.txt
echo -e ${names}"\tStreptococcus thermophilusi\t"$mapped_sterm >> ${BaseLocation}/log/mappings_species.txt

done

##-------------------------------------------------  
##move local
##-------------------------------------------------  

##bring together 

cat /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings.txt \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202/log/mappings.txt > \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings_final.txt


scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/log/mappings_species.txt Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/
```

####mapping strains to genome

```{bash}

threads=37
samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames.txt ##the file with all sample names
logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref
Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv.fasta

##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1

for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/)
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)
names=$(echo ${samplezzKurz})
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 # gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq
  
  
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} ${Assembly} \
  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   
   
    rm -r ${BaseLocation}/MultiQC
 
  mkdir -p ${BaseLocation}/MultiQC
  
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome.txt \
    -o ${BaseLocation}/MultiQC






##-------------------------------------------------  
##-----check read coverage
##-------------------------------------------------  
name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
threads=37
names=Konserve_202


num=1
rm -r ${BaseLocation}/log/mappings.txt
rm -r ${BaseLocation}/log/mappings_species.txt


for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/)
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)
names=$(echo ${samplezzKurz})
mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')

#mapped_sterm=$(grep "CP046134" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)
#mapped_ldel=$(grep "CP046131" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)

mkdir -p ${BaseLocation}/log
echo -e ${names}"\tonlyMeta\t"${mapped} >> ${BaseLocation}/log/mappings.txt

echo -e ${names}"\tLactobacillus delbrueckii\t"$mapped_ldel >> ${BaseLocation}/log/mappings_species.txt
echo -e ${names}"\tStreptococcus thermophilusi\t"$mapped_sterm >> ${BaseLocation}/log/mappings_species.txt

done

grep "0." ${BaseLocation}/log/mappings_species.txt | sort -k 4

grep -v "0.0" ${BaseLocation}/log/mappings_species.txt|awk '{ sum += $4; n++ } END { if (n > 0) print sum / n; }'

```



```{r}
naming_DF <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/98_script/naming_file.csv")
###----------------------------
##script
###----------------------------
library(readr)
library(ggplot2)
#library(plyr)

species_coverage <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/mappings_species.txt", "\t", escape_double = FALSE, col_names = c("sample","species","coverage"),trim_ws = TRUE)

species_coverage$sample

newNames <- setNames(naming_DF$namesNew,naming_DF$namesOLD)
species_coverage$sample <- revalue(species_coverage$sample, newNames)
species_coverage$sample = factor(species_coverage$sample, levels=c(as.character(naming_DF$namesNew)))


newNames <- setNames(as.character(naming_DF$CultureName),naming_DF$namesNew)
species_coverage$culture <- revalue(as.character(species_coverage$sample),newNames)
levels(species_coverage$culture)

species_coverage$coverageLog <- log2(species_coverage$coverage)

all_colours <- (c("#EB4D4D","#10B552") ) 



##--------------------
##ploting


pMOTUs <-  ggplot( data = species_coverage,aes(y = coverage, x = sample, group=species,fill = culture,color=culture))+geom_point(size=2)+geom_hline(yintercept=250)+ # geom_bar( stat="identity")+
  facet_wrap(~species,scales = "free_x")+
   coord_trans(y="log10")+
    labs("",
         x="",
         y="Coverage")+
    theme_classic()+
   scale_color_viridis(discrete=TRUE)+
      scale_y_continuous(breaks =c(0,50,100,250,500,1000),labels=c(0,50,100,250,500,1000))+
  scale_fill_manual(values=all_colours)+
  scale_color_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=4),
      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          # legend.title = element_blank()
          )

  pMOTUs

  
  
  svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/CoverageSpecies.svg",width=5,height=5)
   # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)

pMOTUs

dev.off()

```



#### SNP calling freebayes (K1)

This is a good (blog)[https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2017/Day2/vcf-intro.nb.html] on how to use freebayes and how to parse the output. 
Finally I will do freebayes on all samples simultaniously. 

Interesting output from column 11 on:

GT	Genotype v :1
DP	Read Depth  :1040
DPR	Number of observation for each allele  : 1,1037
RO	Reference allele observation count :1
QR	Sum of quality of the reference observations :10
AO	Alternate allele observation count :1037
QA	Sum of quality of the alternate observations :36917
GL	Genotype Likelihood, log10-scaled likelihoods of the data given the called genotype for each possible genotype generated from the reference and alternate alleles given the sample ploidy

Very nice [blog](http://www.ddocent.com/filtering/) illustrating good filtering practices.

filtering of samples:
1. take away all Lhelv SNVs. 


```{bash,eval=FALSE,echo=FALSE}
ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/ |cut -d '_' -f 1 > /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_withStrains.txt

cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt >> /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_withStrains.txt
###================
##description file
###================
  
  #cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202.txt > /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames.txt
  threads=30
  samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_withStrains.txt ##the file with all sample names
  logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
  BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref
  Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv.fasta
  
  


###--------special
FREEBAYES_out=freebayesOuput

###================
##script
###================

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------change read group
##--------------------------------------------------------------------------------------------------------------------------------------

#name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
names=Lyo_202_2014
###------------------adding readgroup to bam
rm ${BaseLocation}/log/multiqc_logfile_bams.txt
mkdir -p  ${BaseLocation}/log/
  #for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
  for names in $(cat ${samplesss})
do
echo ${names}

java -jar /home/vincent/apps/picard/picard.jar AddOrReplaceReadGroups \
       I=${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam \
       O=${BaseLocation}/${names}/bwaMapping2DB/${names}_mapping2ref.bam \
       RGID=${names} \
       RGLB=lib1 \
       RGPL=illumina \
       RGPU=unit1 \
       RGSM=${names}

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam

echo ${BaseLocation}"/"${names}"/bwaMapping2DB/"${names}"_mapping2ref.bam " >> ${BaseLocation}/log/multiqc_logfile_bams.txt

done

rm ${BaseLocation}/log/multiqc_logfile_bams_new.txt
  for names in $(cut -f 1 ${samplesss})
do

echo ${BaseLocation}"/"${names}"/bwaMapping2DB/"${names}"_mapping2ref.bam " >> ${BaseLocation}/log/multiqc_logfile_bams_new.txt

done


##remove trailing whitespaces

 sed -i 's/[ \t]*$//' ${BaseLocation}/log/multiqc_logfile_bams_new.txt

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------check bam mapping
##--------------------------------------------------------------------------------------------------------------------------------------

 for map in $(cat ${BaseLocation}/log/multiqc_logfile_bams_new.txt)
    do
    echo "----------------------------------------------------"
    echo $map
    samtools flagstat $map |grep "mapped ("
    
    done

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------indexing file
##--------------------------------------------------------------------------------------------------------------------------------------
##need before freebayes parallel

  for names in $(cat ${BaseLocation}/log/multiqc_logfile_bams_new.txt)
do

samtools index ${names}

done

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------SNP calling
##--------------------------------------------------------------------------------------------------------------------------------------

#rm -r ${BaseLocation}/freebayesOuput_WithONT_notParallel_default
#mkdir -p ${BaseLocation}/freebayesOuput_WithONT_notParallel_default


  #  freebayes -F 0.05 -C 3 --min-coverage 30 --pooled-discrete \
  #    --bam-list ${BaseLocation}/log/multiqc_logfile_bams.txt -f \
  #    ${Assembly} > \
  #   ${BaseLocation}/freebayesOuput_WithONT_notParallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf
  
##--------------------------------------------------------------------------------------------------------------------------------------
##-----------SNP calling parallel
##--------------------------------------------------------------------------------------------------------------------------------------
samtools faidx ${Assembly}
bwa index ${Assembly}
FREEBAYES_out=freebayesOuput_all
  rm -r ${BaseLocation}/${FREEBAYES_out}
  mkdir -p ${BaseLocation}/${FREEBAYES_out}
 
  
    freebayes-parallel <(/home/vincent/apps/freebayes/scripts/fasta_generate_regions.py $Assembly 100000) 20 -f ${Assembly} -C 5 --pooled-continuous --min-alternate-fraction 0.01 --min-coverage 10 --bam-list ${BaseLocation}/log/multiqc_logfile_bams_new.txt > ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf
  
  #  freebayes-parallel <(/home/vincent/apps/freebayes/scripts/fasta_generate_regions.py $Assembly 100000) 30 -f ${Assembly} -C 5 --pooled-continuous --min-alternate-fraction 0.01 --min-coverage 10 --bam-list ${BaseLocation}/log/multiqc_logfile_bams.txt > ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf
    
  
  grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf |cut -f 1 |sort|uniq -c
  
  grep -v "^#"  ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf|cut -f 1|sort|uniq -c

  
 wc -l /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf
##-------------zipping

#bgzip -c /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains/freebayesOuput_WithONT_02/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains/freebayesOuput_WithONT_02/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf.gz
#tabix -p  vcf /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final/freebayesOuput_WithONT/variantCallsfreebayes_all.vcf.gz


##--------------------------------------------------------------------------------------------------------------------------------------
##------------vcftools filtering
##--------------------------------------------------------------------------------------------------------------------------------------


###==============================1. Qualty filtering QC>30

vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf --minQ 30 --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC  

  grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC.recode.vcf   |cut -f 1 |sort|uniq -c


vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf  --counts2 --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC  

###==============================4. hist-indel-len


vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC.recode.vcf --hist-indel-len --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC

###==============================2. remove indel

vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC.recode.vcf --mac 1 --remove-indels --minQ 30 --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL

  grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf   |cut -f 1 |sort|uniq -c

###==============================3. INFO

vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --missing-indv --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC  
###==============================4. hist-indel-len


vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --hist-indel-len --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL

grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf |cut -f 1 |sort|uniq -c

###==============================5. filter samples
##1. only sites that actualy have a single count (--non-ref-ac-any 1)
##2. only sites that are not in all different from the reference (HAS TO BE CHANGED AGAIN WITH THE RIGHT REF!!!!!) ##mutations that actually occur in the metagenome
##3. only biallelic sites (--min-alleles 2 --max-alleles 2)

vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --indv C_202_18_02 --min-alleles 2 --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK202 

rm -r ${BaseLocation}/${FREEBAYES_out}/prepedR/
mkdir -p ${BaseLocation}/${FREEBAYES_out}/prepedR/
#for culture in $(cut -d "_" -f 2 $samplesss|sort|uniq )
for culture in $(grep "FAM" -v $samplesss | cut -d "_" -f 2 |sort|uniq )
do
grep "\_$culture\_" $samplesss >  ${BaseLocation}/${FREEBAYES_out}/tmp.samples
#total=$(wc -l ${BaseLocation}/${FREEBAYES_out}/tmp.samples|cut -d ' ' -f 1)
vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --keep ${BaseLocation}/${FREEBAYES_out}/tmp.samples --non-ref-ac-any 1 --max-non-ref-af 0.99 --min-alleles 2 --max-alleles 2 --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK${culture}

grep "^##" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK${culture}.recode.vcf |grep "CP031023.1" -v  >  ${BaseLocation}/${FREEBAYES_out}/prepedR/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK${culture}_prepR.vcf


#--non-ref-af and --max-non-ref-af
done

##---------------------------------
##-FAM strains
##---------------------------------
#for culture in $(grep "FAM" $samplesss |sort|uniq )
#do
grep "^FAM" $samplesss >  ${BaseLocation}/${FREEBAYES_out}/tmp.samples
#total=$(wc -l ${BaseLocation}/${FREEBAYES_out}/tmp.samples|cut -d ' ' -f 1)
vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --keep ${BaseLocation}/${FREEBAYES_out}/tmp.samples --non-ref-ac-any 1 --max-non-ref-af 0.99 --min-alleles 2 --max-alleles 2 --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMKFAMstrains

grep "^##" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMKFAMstrains.recode.vcf |grep "CP031023.1" -v >  ${BaseLocation}/${FREEBAYES_out}/prepedR/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMKFAMstrains_prepR.vcf

  grep "^#" -v /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf |cut -f 1 |sort|uniq -c

  grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/prepedR/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMKFAMstrains_prepR.vcf |cut -f 1 |sort|uniq -c





#rm ${BaseLocation}/${FREEBAYES_out}/prepedR_merged/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK_allSamples_prepR.vcf
#mkdir -p ${BaseLocation}/${FREEBAYES_out}/prepedR_merged/
#for culture in $(cut -d "_" -f 2 $samplesss|sort|uniq )
#do

#cat ${BaseLocation}/${FREEBAYES_out}/prepedR/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK${culture}_prepR.vcf >> ${BaseLocation}/${FREEBAYES_out}/prepedR_merged/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK${culture}_prepR.vcf

#done

#bcftools query -f '%CHROM %POS [%AO/%DP\t]\n'  ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK${culture}.recode.vcf | head -3
##--------------------------------------------------------------------------------------------------------------------------------------
##------------prep for R
##--------------------------------------------------------------------------------------------------------------------------------------


#grep "^##" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf >  ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_prepR.vcf


##--------------------------------------------------------------------------------------------------------------------------------------
##------------bring local
##--------------------------------------------------------------------------------------------------------------------------------------
mkdir -p /home/vincent/Desktop/Projects/2020_strainDelineation/05_SNP_called/raw_snps
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/prepedR/ /home/vincent/Desktop/Projects/2020_strainDelineation/05_SNP_called/raw_snps/


```


##Add Information

#####snpEff

Here we do [snpEff](http://snpeff.sourceforge.net/SnpEff_manual.html#buildAddConfig).


```{bash}


###================
##description file
###================
threads=37
FREEBAYES_out=freebayesOuput
sampleShort=PGAP_meta_all ##For st_thermophilus

 threads=37
  samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_withStrains.txt ##the file with all sample names
  logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
  BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref
  Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv.fasta
  



#${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf


##=======================================================================================================================
##-----------------------------------------------all together----------------------------------------------
##=======================================================================================================================
##==================
##Building Database
##==================
#/archiv/Projects/2018_Culturomics/genomes/Lactobacillus_delbrueckii_RMK202.fna
#/archiv/Projects/2018_Culturomics/genomes/Streptococcus_thermophilus_RMK202.fna

##-------------
##01_add genome name to
##-------------

vim /home/vincent/miniconda2/share/snpeff-4.3.1t-2/snpEff.config
#S_thermophilus_mag_rmk202.genome : S_thermophilus_mag_rmk202
#L_delbrueckii_mag_rmk202.genome : L_delbrueckii_mag_rmk202
#Streptococcus_phage_rmk202.genome : Streptococcus_phage_rmk202
PGAP_meta_all.genome : PGAP_meta_all
#PGAP_meta_all_StarterCultureDiversity.genome : PGAP_meta_all_StarterCultureDiversity
##-------------
#02_put genome fasta and gff into right directory:
##-------------

#sampleShort=S_thermophilus_mag_rmk202
#sampleShort=CP046134 ##For st_thermophilus

#/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta 

##-------------
#03_put genome fasta 
##-------------
rm -r /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}
mkdir -p /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}

cat ${Assembly} > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}/sequences.fa



##-------------
#04 genes gff 
##-------------

#grep "^#" -v /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.gff|grep -e "${sampleShort}" |cut -f 1|sort|uniq -c

#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.gff | grep -e "^${sampleShort}" |awk -F "\t" '{OFS"\t"}{if($3=="CDS") print $0}' > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff


##-------------
#04 all genes gff 
##-------------

##already preped in RAST chunk
 cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/merged/all_genes_rmk202.gff |sed 's/gene/CDS/g' > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff

##-------------
##05_build database
##-------------

cd /home/vincent/miniconda2/share/snpeff-4.3.1t-2/
java -jar snpEff.jar build -gff3 -v ${sampleShort}


##==================
##running snpEff
##==================

##-------------
##06_preperation of SNV file
##-------------

#grep -e "${sampleShort}" ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf > ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_all.vcf

##-------------
##07 SNveff calling
##-------------

cd /home/vincent/miniconda2/share/snpeff-4.3.1t-2/
#rm -r ${sampleShort} ${BaseLocation}/${FREEBAYES_out}/SnpEff/${sampleShort}/
mkdir -p ${sampleShort} ${BaseLocation}/${FREEBAYES_out}/SnpEff/${sampleShort}/

grep "CP031023.1" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf > ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_withoutLhelv.recode.vcf


java -Xmx64g -jar snpEff.jar ${sampleShort} ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_withoutLhelv.recode.vcf > ${BaseLocation}/${FREEBAYES_out}/SnpEff/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_snpeff_woLhevl.vcf


 # grep "^#" -v /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf |cut -f 1 |sort|uniq -c


##-------------
##08_prep for R
##-------------

grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/SnpEff/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_snpeff_woLhevl.vcf  | \
  awk -F "\t" 'BEGIN{OFS="\t"} {print $1,$2,$3,$4,$5,$6,$7,$8}' > ${BaseLocation}/${FREEBAYES_out}/SnpEff/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_snpeff_prepforR.vcf


cut -f 1 ${BaseLocation}/${FREEBAYES_out}/SnpEff/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_snpeff_prepforR.vcf|sort|uniq -c

##==================
##transfer local 
##==================

##-------------
###snpEff
##-------------
mkdir -p ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new_startercultureDiversity/
scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/SnpEff/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_snpeff_prepforR.vcf ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new_startercultureDiversity/

##-------------
###eggnog
##-------------

#scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/eggnog/${sampleShort}_pgap_query_seqs.fa.emapper.annotations  ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_${sampleShort}_eggnog.vcf 



```

here we merge all annotation knowledge we have of the SNPs:

1. SNPeffect
2. eggnog annotation
3. repeat annotation
4. core genes

Thereafter we have a long list of information for every SNP. 

#####repeat genes

We could potentially have a problem of alignement in genes highly similiar. Therefore I look if which genes are cluster together. 

```{bash}



###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished.fasta



BaseLocation_Ldel=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/
BaseFILNAME=L_delbrueckii_RMK202


BaseLocation_Sterm=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/
BaseFILNAME_sterm=S_thermophilus_RMK202


###================
##script
###================


##==================
##merge all fna files
##==================


rm -r ${BaseLocation}/Repeats/
mkdir -p ${BaseLocation}/Repeats/

cat ${BaseLocation_Ldel}/${BaseFILNAME}_genes.ffn \
  ${BaseLocation_Sterm}/${BaseFILNAME_sterm}_genes.ffn > \
 ${BaseLocation}/Repeats/both_genes.ffn

cat ${BaseLocation_Ldel}/${BaseFILNAME}.current.gff \
  ${BaseLocation_Sterm}/${BaseFILNAME_sterm}.current.gff > \
 ${BaseLocation}/Repeats/both_genes.ffn
 
##==================
##cd-hit
##==================

cd-hit-est -i ${BaseLocation}/Repeats/both_genes.ffn \
    -o ${BaseLocation}/Repeats/both__cdsClustering.txt -c 0.90 -M 30000 -T 37 -sc 1 -sf 1

  # cd-hit -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_cds.fna \
  #  -o /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/sterm_cdsClustering.txt -c 0.85 -M 30000 -T 37 -sc 1 -sf 1
  






  ###multifasta2fasta
  mkdir -p ${BaseLocation}/Repeats/splitClusters
  cd ${BaseLocation}/Repeats/splitClusters
  awk 'NR%1==0{ print > "Cluster_"++i".txt" }' RS='>Cl' ${BaseLocation}/Repeats/both__cdsClustering.txt.clstr

  ##prepare log file
  #cluster_2.txt till cluster_24.txt
  
  ls ${BaseLocation}/Repeats/splitClusters |sort -V |head -84 |tail -83 > ${BaseLocation}/Repeats/splitClusters/files_of_INterest.txt
     
   ##get gene names
   num=2
   rm ${BaseLocation}/Repeats/bigCluster/all_repeatingGenes.txt
   mkdir -p ${BaseLocation}/Repeats/bigCluster/
   i=Cluster_2.txt
   for i in $(cat ${BaseLocation}/Repeats/splitClusters/files_of_INterest.txt)
    do
   grep -v "^uster" ${BaseLocation}/Repeats/splitClusters/${i} |awk -F ">" '{print $2}' |awk -F "... at " 'BEGIN{OFS="\t"}{print $1,$2}' |sed 's/... \*//g' |sed '$ d' > \
    ${BaseLocation}/Repeats/bigCluster/names_${i}
    
   grep -v "^uster" ${BaseLocation}/Repeats/splitClusters/${i} |awk -F ">" '{print $2}'  |awk -F "... at " -v count="$num" 'BEGIN{OFS="\t"}{print $1,$2,count}' |sed 's/... \*//g' |sed '$ d' >> \
   ${BaseLocation}/Repeats/bigCluster/all_repeatingGenes.txt
      
    num=$((num+1))
    
    
    done
 

    ##get gff
    
  # genes=pgaptmp_001292
    mkdir -p  ${BaseLocation}/Repeats//bigCluster_genes/
    for i in $(cat ${BaseLocation}/Repeats/splitClusters/files_of_INterest.txt)
    do
    rm  ${BaseLocation}/Repeats//bigCluster_genes//genes_${i}
    for genes in $(cut -f 1 ${BaseLocation}/Repeats/bigCluster//names_${i})
      do
        
        grep "${genes}.p" ${BaseLocation}/Repeats/both_genes.ffn >> \
          ${BaseLocation}/Repeats//bigCluster_genes/genes_${i}
       
        
      done # $genes   
    done   # $i
    
    
    
  mkdir -p ~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes


scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/Repeats/bigCluster/all_repeatingGenes.txt ~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes/

```


#####Core genes prep

```{bash}

rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt

##=======================================================================================================================
##-----------------------------------------------Streptococcus thermophilus----------------------------------------------
##=======================================================================================================================
species=Sterm

##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/OG_names_${species}.txt
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups.txt | sed 's/gnl|extdb|//g'| awk -F " " 'BEGIN{OFS="\t"}{print $1,"S_thermophilus_RMK202",$2}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01/Orthogroups/OG_names_${species}.txt
#grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups.txt | sed 's/gnl|extdb|//g'| awk -F " " 'BEGIN{OFS="\t"}{print $1,$2}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01/Orthogroups/OG_names_${species}.txt


done #OG

cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/OG_names_${species}.txt >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt


##=======================================================================================================================
##-----------------------------------------------Lactobacillus delbrueckii----------------------------------------------
##=======================================================================================================================
species=Ldel

##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/OG_names_${species}.txt
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

#grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups.txt| sed 's/gnl|extdb|//g' |awk -F " " 'BEGIN{OFS="\t"}{print $1,"	L_delbrueckii_RMK202",$11}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/OG_names_${species}.txt
grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups.txt| sed 's/gnl|extdb|//g' |awk -F " " 'BEGIN{OFS="\t"}{print $1,"	L_delbrueckii_RMK202",$2}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/OG_names_${species}.txt

#grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups.txt| sed 's/gnl|extdb|//g' |awk -F " " 'BEGIN{OFS="\t"}{print $1,$2}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/OG_names_${species}.txt


done #OG

cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/OG_names_${species}.txt >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt



##=======================================================================================================================
##-----------------------------------------------Lactobacillus delbrueckii----------------------------------------------
##=======================================================================================================================

mkdir -p ~/Desktop/Projects/2019_Pilotplan/09_annoation/coreGenes

scp vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt  ~/Desktop/Projects/2019_Pilotplan/09_annoation/coreGenes/coreGenes_rmk202_both.txt


```

#####merge all infomration

```{r}
library(readr)
library(stringr)

##==================
##01-SNPeff
##==================
RMK202_both_snpEff <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new_startercultureDiversity/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_snpeff_prepforR.vcf",  "\t", escape_double = FALSE, col_names = c("X.CHROM","POS","unknown","REF","ALT","quality","uknown2","snpeff"), trim_ws = TRUE)
table(RMK202_both_snpEff$X.CHROM)
RMK202_both_snpEff_02 <- subset(RMK202_both_snpEff, select=-c(unknown,uknown2))
RMK202_both_snpEff_02$site <- paste(RMK202_both_snpEff_02$X.CHROM,RMK202_both_snpEff_02$POS,RMK202_both_snpEff_02$REF,RMK202_both_snpEff_02$ALT,sep="_")

tmp_snpeff <-  data.frame(str_split_fixed(RMK202_both_snpEff_02$snpeff, fixed("|"), 136)[,2:8])
colnames(tmp_snpeff) <- c("effect","significance","geneName","PGAP_name","type","typeName","typeGene")

RMK202_both_snpEff_02 <- subset(RMK202_both_snpEff_02, select=-snpeff)
RMK202_both_SNPeff_final <-  cbind(RMK202_both_snpEff_02,tmp_snpeff)
# table(S_thermophilus_snpEff_03$typeGene)
# 
# S_thermophilus_snpEff_03[which(S_thermophilus_snpEff_03$typeGene=="pseudogene"),]
# table(S_thermophilus_snpEff_03[which(S_thermophilus_snpEff_03$typeGene=="protein_coding"),]$type)
# S_thermophilus_snpEff_03[which(S_thermophilus_snpEff_03$typeGene=="rRNA"),]

RMK202_both_SNPeff_final$finalName <- RMK202_both_SNPeff_final$typeName %>% gsub("rna-","",.) %>%  gsub("TRANSCRIPT_gene-","",.) %>%  gsub("gene-","",.) 
RMK202_both_SNPeff_final$finalName <- paste(RMK202_both_SNPeff_final$X.CHROM,RMK202_both_SNPeff_final$finalName,sep = "_")
# length(grep("pgaptmp_",S_thermophilus_snpEff_03$finalName,invert = TRUE))
table(RMK202_both_SNPeff_final$X.CHROM)
#table(RMK202_both_SNPeff_final$significance)
##==================
##02 eggnog
##==================
RMK202_both_eggnog <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/merged/merged_all_eggnog.annotations",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description"))
# RMK202_both_eggnog[RMK202_both_eggnog$query_name %in% SNPeff_final$finalName,]

##==================
##03 repeats
##==================

RMK202_both_repeats <- read_delim("~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes/all_repeatingGenes.txt",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("geneNAME","Repeat_identity","Repeat_cluster")) #%>% slice( 1:(n()-3))
# nrow(RMK202_both_repeats)

##==================
##04_core genes
##==================
library(tidyverse)
RMK202_both_coreGenes <- read_delim("~/Desktop/Projects/2019_Pilotplan/09_annoation/coreGenes/coreGenes_rmk202_both.txt",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("orthogroup","species","geneName")) %>% add_column(.,core="core")
table(RMK202_both_coreGenes$species)
table(RMK202_both_coreGenes$core, RMK202_both_coreGenes$species)
tmp <- RMK202_both_coreGenes %>% filter(species=="L_delbrueckii_RMK202")
RMK202_both_coreGenes %>% filter(species=="S_thermophilus_RMK202")

##==================
##05_merge
##==================
nrow(RMK202_both_SNPeff_final)

RMK202_snpeff_eggnog <- merge(RMK202_both_SNPeff_final, RMK202_both_eggnog, by.x = "geneName", by.y = "query_name",all.x = TRUE)
RMK202_snpeff_eggnog_repeats <- merge(RMK202_snpeff_eggnog, RMK202_both_repeats, by.x = "geneName", by.y = "geneNAME",all.x = TRUE)
RMK202_snpeff_eggnog_repeats$geneName <- as.character(RMK202_snpeff_eggnog_repeats$geneName)
RMK202_snpeff_eggnog_repeats_core <- merge(RMK202_snpeff_eggnog_repeats, RMK202_both_coreGenes, by.x = "geneName", by.y = "geneName",all.x = TRUE)
# RMK202_both_coreGenes$geneName
# RMK202_snpeff_eggnog_repeats$geneName

nrow(RMK202_snpeff_eggnog_repeats_core)

table(RMK202_snpeff_eggnog_repeats_core$core)
table(RMK202_snpeff_eggnog_repeats_core$species)
RMK202_snpeff_eggnog_repeats_core$species <- revalue(RMK202_snpeff_eggnog_repeats_core$X.CHROM, c("CP046131"="L. delbrueckii","CP046134"="S. thermophilus"))
table(RMK202_snpeff_eggnog_repeats_core$species)

# write.csv(RMK202_snpeff_eggnog_repeats_core,"~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new_startercultureDiversity//RMK202_snpeff_eggnog_repeats_core.txt")
write.csv(RMK202_snpeff_eggnog_repeats_core,"~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new_startercultureDiversity//RMK202_snpeff_eggnog_repeats_core_02.txt",row.names = FALSE)


RMK202_snpeff_eggnog_repeats_core[grep("S_thermophilus",RMK202_snpeff_eggnog_repeats_core$finalName),]
RMK202_snpeff_eggnog_repeats_core[grep("CP046131",RMK202_snpeff_eggnog_repeats_core$finalName),]

nrow(RMK202_snpeff_eggnog_repeats_core)

RMK202_snpeff_eggnog_repeats_core_ldel <- RMK202_snpeff_eggnog_repeats_core %>% filter(X.CHROM=="CP046131")
table(RMK202_snpeff_eggnog_repeats_core_ldel$core)
```



##first analysis

```{r}

library(tidyverse)
library(vcfR)
library(readr)
library(plyr)
library(dplyr)
library(ggplot2)
library(xlsx)

##==================
##01_import data
##==================


###====================================================
###====================================================
##specifically for metagenomes
###====================================================
###====================================================
culturesss <- FAMstrains
culturesss <- 101
snps_long <- NA
for (culturesss in c("101","105","115","124","150","190","202","203","280","302","305")) {
  

snps_freebayes <- read_delim(paste0("/home/vincent/Desktop/Projects/2020_strainDelineation/05_SNP_called/raw_snps/prepedR/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK",culturesss,"_prepR.vcf"), "\t", escape_double = FALSE, trim_ws = TRUE) %>%  select(-c(QUAL,FILTER,FORMAT))


##==================
##01_filter SNVs MQM larger than 30 
##INFO=<ID=MQM,Number=A,Type=Float,Description="Mean mapping quality of observed alternate alleles">
##==================
snps_freebayes$MQM <- str_split_fixed(snps_freebayes$INFO, ";", 130)[,16] %>% gsub("MQM=","",.) 
snps_freebayes$MQM <- str_split_fixed(snps_freebayes$MQM, ",", 3)[,1] %>%  as.numeric()
snps_freebayes <- snps_freebayes[which(snps_freebayes$MQM>30),] %>% select(-MQM,-INFO)

##==================
##02_long list and frequency calc
##==================

snps_freebayes_long <- gather(snps_freebayes, sample, Snps, colnames(snps_freebayes)[6]:colnames(snps_freebayes)[ncol(snps_freebayes)], factor_key=TRUE,na.rm = TRUE) 

###------------------------calculate allel frequency   

sample_relative_temp <-snps_freebayes_long %>% filter(Snps!=".:.:.:.:.:.:.:.")%>% separate("Snps", into=c("GT","DP","AD","RO","QR","AO","QA","GL"), sep = ":") %>% transform(.,allel_frequency=as.numeric(AO)/as.numeric(DP),QualityScore=as.numeric(QA)/as.numeric(DP),Sequencing_depth=as.numeric(DP))

##==================
##03_filter QA/DP > 30
##INFO=<ID=QA,Number=A,Type=Integer,Description="Alternate allele quality sum in phred">
##INFO=<ID=DP,Number=1,Type=Integer,Description="Total read depth at the locus"
##==================
sample_relative_temp_filter <- sample_relative_temp[which(!is.na(sample_relative_temp$QualityScore)),]
sample_relative_temp_withONT <-sample_relative_temp_filter[which(sample_relative_temp_filter$QualityScore>5),]
table(sample_relative_temp_withONT$sample)
sample_relative_temp_withONT$site <- paste(sample_relative_temp_withONT$X.CHROM,sample_relative_temp_withONT$POS,sample_relative_temp_withONT$REF,sample_relative_temp_withONT$ALT,sep="_")
sample_relative_temp <- sample_relative_temp_withONT


# sample_relative_temp_withONT <-snps_freebayes_long %>% separate("Snps", into=c("GT","DP","AD","RO","QR","AO","QA","GL"), sep = ":") %>% transform(.,allel_frequency=as.numeric(AO)/as.numeric(DP))

# sample_relative_temp_withONT <- sample_relative_temp
sample_relative_temp_withONT$site <- paste(sample_relative_temp_withONT$X.CHROM,sample_relative_temp_withONT$POS,sample_relative_temp_withONT$REF,sample_relative_temp_withONT$ALT,sep="_")
sample_relative_safe_final_prep <- sample_relative_temp_withONT %>% select(site,X.CHROM,sample,allel_frequency)

sample_relative_safe_final_tsne <- spread(sample_relative_safe_final_prep, sample, allel_frequency)

sample_relative_wide <- sample_relative_safe_final_tsne[rowSums(sample_relative_safe_final_tsne[,3:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)>0,]
sample_relative_wide$species <- revalue(sample_relative_wide$X.CHROM, c("CP046131"="L. delbrueckii","CP046134"="S. thermophilus"))
sample_relative_wide$culture <- culturesss



##==================
##11_make long
##==================
colnames(sample_relative_wide)
sample_relative_long <- gather(sample_relative_wide, sample, Snps, colnames(sample_relative_wide)[3]:colnames(sample_relative_wide)[ncol(sample_relative_wide)-2], factor_key=TRUE,na.rm = TRUE) 
# nrow(sample_relative_long)
table(sample_relative_long$X.CHROM)

snps_long <- rbind(snps_long,sample_relative_long)
}

###====================================================
###====================================================
##specifically for strains
###====================================================
###====================================================
culturesss <- "FAMstrains"
#snps_long <- NA
for (culturesss in c("FAMstrains")) {
  

snps_freebayes <- read_delim(paste0("/home/vincent/Desktop/Projects/2020_strainDelineation/05_SNP_called/raw_snps/prepedR/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK",culturesss,"_prepR.vcf"), "\t", escape_double = FALSE, trim_ws = TRUE) %>%  select(-c(QUAL,FILTER,FORMAT))


##==================
##01_filter SNVs MQM larger than 30 
##INFO=<ID=MQM,Number=A,Type=Float,Description="Mean mapping quality of observed alternate alleles">
##==================
snps_freebayes$MQM <- str_split_fixed(snps_freebayes$INFO, ";", 130)[,16] %>% gsub("MQM=","",.) 
snps_freebayes$MQM <- str_split_fixed(snps_freebayes$MQM, ",", 3)[,1] %>%  as.numeric()
snps_freebayes <- snps_freebayes[which(snps_freebayes$MQM>30),] %>% select(-MQM,-INFO)

##==================
##02_long list and frequency calc
##==================

snps_freebayes_long <- gather(snps_freebayes, sample, Snps, colnames(snps_freebayes)[6]:colnames(snps_freebayes)[ncol(snps_freebayes)], factor_key=TRUE,na.rm = TRUE) 

###------------------------calculate allel frequency   
###!!!!!!!filter allele frequency >0.8
###!!!!!!!only genomic SNVs


sample_relative_temp <-snps_freebayes_long %>% filter(Snps!=".:.:.:.:.:.:.:.") %>% separate("Snps", into=c("GT","DP","AD","RO","QR","AO","QA","GL"), sep = ":") %>% transform(.,allel_frequency=as.numeric(AO)/as.numeric(DP),QualityScore=as.numeric(QA)/as.numeric(DP),Sequencing_depth=as.numeric(DP)) %>% filter(allel_frequency>0.8) %>% filter(X.CHROM %in% c("CP046131","CP046134"))

ggplot(sample_relative_temp,aes(x=allel_frequency))+geom_density()+facet_wrap(~X.CHROM,scales = "free")

##==================
##03_filter QA/DP > 30
##INFO=<ID=QA,Number=A,Type=Integer,Description="Alternate allele quality sum in phred">
##INFO=<ID=DP,Number=1,Type=Integer,Description="Total read depth at the locus"
##==================
sample_relative_temp_filter <- sample_relative_temp[which(!is.na(sample_relative_temp$QualityScore)),]
#table(sample_relative_temp_withONT$sample)
sample_relative_temp_filter$site <- paste(sample_relative_temp_filter$X.CHROM,sample_relative_temp_filter$POS,sample_relative_temp_filter$REF,sample_relative_temp_filter$ALT,sep="_")
sample_relative_temp <- sample_relative_temp_filter


# sample_relative_temp_withONT <-snps_freebayes_long %>% separate("Snps", into=c("GT","DP","AD","RO","QR","AO","QA","GL"), sep = ":") %>% transform(.,allel_frequency=as.numeric(AO)/as.numeric(DP))

sample_relative_temp_withONT <- sample_relative_temp
sample_relative_safe_final_prep <- sample_relative_temp_withONT %>% select(site,X.CHROM,sample,allel_frequency)

sample_relative_safe_final_tsne <- spread(sample_relative_safe_final_prep, sample, allel_frequency)

sample_relative_wide <- sample_relative_safe_final_tsne[rowSums(sample_relative_safe_final_tsne[,3:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)>0,]
sample_relative_wide$species <- revalue(sample_relative_wide$X.CHROM, c("CP046131"="L. delbrueckii","CP046134"="S. thermophilus"))
sample_relative_wide$culture <- culturesss



##==================
##11_make long
##==================
colnames(sample_relative_wide)
sample_relative_long <- gather(sample_relative_wide, sample, Snps, colnames(sample_relative_wide)[3]:colnames(sample_relative_wide)[ncol(sample_relative_wide)-2], factor_key=TRUE,na.rm = TRUE) 
# nrow(sample_relative_long)
table(sample_relative_long$X.CHROM)

snps_long <- rbind(snps_long,sample_relative_long)
}


# sample_relative_temp  %>% group_by(X.CHROM) %>% summarize(meansample=mean(Sequencing_depth))
##==================
##remove samples
##==================

##the cheese samples from rmk202 are weird because they were passaged on plate before.
table(snps_long$sample) %>% length()
snps_long <- snps_long %>% filter(!sample %in% c("C_202_18_02","C_202_18_01"))
table(snps_long$sample)%>% length()

##==================
##11_make long
##==================
snps_long <- snps_long[-1,]

table(snps_long$sample)
#snps_long$samplesNew <- revalue(snps_long$sample,c("FSK_101_75"="starter 1","Lyo_101_76_4_7"="starter 2","Lyo_101_77"="starter 3","Lyo_101_02_7"="starter 4","Lyo_101_10_2"="starter 5","Lyo_101_16"="starter 6","ws_101_19"="starter 7","FSK_105_75"="starter 1","Lyo_105_76_4_1"="starter 2","Lyo_105_76_4_2"="starter 3","FSK_105_76_4_6"="starter 4","Lyo_105_76_2"="starter 5","Lyo_105_76_1"="starter 6","Lyo_105_76_5_11"="starter 7","Lyo_105_76_5_13"="starter 8","Lyo_105_76_5_12"="starter 9","Lyo_105_76_3_31"="starter 10","Lyo_105_11_5"="starter 11","Ws_105_19"="starter 12","Lyo_115_09"="starter 1","Ws_115_19"="starter 2","FSK_124_75_4"="starter 1","Lyo_124_08"="starter 2", "Ws_124_19"="starter 3","Lyo_150_76_4"="starter 1","Lyo_150_76_11"="starter 2","Lyo_150_11_3_11"="starter 3","Lyo_150_14_10"="starter 4","Lyo_150_18"="starter 5","Ws_150_19"="starter 6","Lyo_190_95_3"="starter 1","Lyo_190_17"="starter 2","Ws_190_19"="starter 3","Lyo_202_96"="starter 1","Lyo_202_12"="starter 2","Lyo_202_14"="starter 3","S_202_14"="starter 4","Ws_202_18"="starter 5","S_202_18"="starter 6", "C_202_18_02"="starter 7","C_202_18_01"="starter 8","Lyo_203_77"="starter 1","Lyo_203_96_1"="starter 2","Lyo_203_16"="starter 3","Ws_203_19"="starter 4","Lyo_280_16"="starter 1","Lyo_280_16_4"="starter 2", "Ws_280_19"="starter 3","Lyo_302_13"="starter 1", "Ws_302_19"="starter 2","FSK_305_76"="starter 1", "Lyo_305_96_1"="starter 2","Lyo_305_17"="starter 3","Ws_305_19"="starter 4"))


snps_long$samplesNew <- as.character(revalue(snps_long$sample,c("FSK_101_75"="starter 1","Lyo_101_76_4_7"="starter 2","Lyo_101_77"="starter 3","Lyo_101_02_7"="starter 4","Lyo_101_10_2"="starter 5","Lyo_101_16"="starter 6","ws_101_19"="starter 7","FSK_105_75"="starter 1","Lyo_105_76_4_1"="starter 2","Lyo_105_76_4_2"="starter 3","FSK_105_76_4_6"="starter 4","Lyo_105_76_2"="starter 5","Lyo_105_76_1"="starter 6","Lyo_105_76_5_11"="starter 7","Lyo_105_76_5_13"="starter 8","Lyo_105_76_5_12"="starter 9","Lyo_105_76_3_31"="starter 10","Lyo_105_11_5"="starter 11","Ws_105_19"="starter 12","Lyo_115_09"="starter 1","Ws_115_19"="starter 2","FSK_124_75_4"="starter 1","Lyo_124_08"="starter 2", "Ws_124_19"="starter 3","Lyo_150_76_4"="starter 1","Lyo_150_76_11"="starter 2","Lyo_150_11_3_11"="starter 3","Lyo_150_14_10"="starter 4","Lyo_150_18"="starter 5","Ws_150_19"="starter 6","Lyo_190_95_3"="starter 1","Lyo_190_17"="starter 2","Ws_190_19"="starter 3","Lyo_202_96"="starter 1","Lyo_202_12"="starter 2","Lyo_202_14"="starter 3","S_202_14"="starter 4","Ws_202_18"="starter 5","S_202_18"="starter 6", "C_202_18_02"="starter 7","C_202_18_01"="starter 8","Lyo_203_77"="starter 1","Lyo_203_96_1"="starter 2","Lyo_203_16"="starter 3","Ws_203_19"="starter 4","Lyo_280_16"="starter 1","Lyo_280_16_4"="starter 2", "Ws_280_19"="starter 3","Lyo_302_13"="starter 1", "Ws_302_19"="starter 2","FSK_305_76"="starter 1", "Lyo_305_96_1"="starter 2","Lyo_305_17"="starter 3","Ws_305_19"="starter 4",
"Vers_101_20"="starter 8", "Vers_105_20"="starter 13", 
"Vers_115_20"="starter 3", "Vers_124_20"="starter 4", "Vers_150_20"="starter 7", "Vers_190_20"="starter 4", "Vers_203_20"="starter 5", 
"Vers_280_20"="starter 4", "Vers_302_20"="starter 3", "Vers_305_20"="starter 5",
"Therm_101_20"="starter 9", "Therm_105_20"="starter 14", "Therm_115_20"="starter 4", "Therm_124_20"="starter 5", 
"Therm_150_20"="starter 8", "Therm_190_20"="starter 5", "Therm_203_20"="starter 6", "Therm_280_20"="starter 5", 
"Therm_302_20"="starter 4", "Therm_305_20"="starter 6",
"24h_101_20"="starter 10", "24h_105_20"="starter 15", "24h_115_20"="starter 5", "24h_124_20"="starter 6", "24h_150_20"="starter 9", 
"24h_190_20"="starter 6", "24h_203_20"="starter 7", "24h_280_20"="starter 6", "24h_302_20"="starter 5", "24h_305_20"="starter 7",
"M17x_101_20"="starter 11", "M17X_105_20"="starter 16", "M17X_115_20"="starter 6", 
"M17X_124_20"="starter 7", "M17x_150_20"="starter 10", "M17x_190_20"="starter 7", "M17x_202_20"="starter 9", "M17x_203_20"="starter 8", 
"M17x_280_20"="starter 7", "M17x_302_20"="starter 6", "M17x_305_20"="starter 8")))


table(snps_long$samplesNew)
unique(snps_long$samplesNew)
sum(is.na(snps_long$samplesNew))

dput(unique(snps_long$samplesNew))

snps_long$samplesNew = factor(snps_long$samplesNew, levels=c("starter 10", "starter 5", "starter 9", "starter 6", "starter 4", 
"starter 3", "starter 2", "starter 7", "starter 1", "starter 11", 
"starter 8", "starter 13", "starter 12", "starter 14", "starter 15", 
"starter 16", "FAM24791", "FAM24784", "FAM24790", "FAM24777", 
"FAM24787", "FAM13491", "FAM24776", "FAM24775", "FAM24772", "FAM24769", 
"FAM24768", "FAM24767", "FAM24782", "FAM24786", "FAM24774", "FAM24759", 
"FAM24746", "FAM24758", "FAM24757", "FAM24742", "FAM24756", "FAM24755", 
"FAM24792", "FAM24741", "FAM24749", "FAM24770", "FAM24726", "FAM24752", 
"FAM24734", "FAM24739", "FAM24785", "FAM13496", "FAM24725", "FAM24853", 
"FAM24735", "FAM12107", "FAM24724", "FAM24753", "FAM24799", "FAM24743", 
"FAM24728", "FAM24736", "FAM24764", "FAM24731", "FAM13498", "FAM24727", 
"FAM24783", "FAM13495", "FAM24729", "FAM24760", "FAM24738", "FAM24740", 
"FAM24732", "FAM24744", "FAM13500", "FAM24745", "FAM24754", "FAM24730", 
"FAM24794", "FAM24773", "FAM24747", "FAM24781", "FAM24795", "FAM24797", 
"FAM24854", "FAM24855", "FAM24778", "FAM24748", "FAM24761", "FAM24779", 
"FAM13492", "FAM24762", "FAM13499", "FAM24751", "FAM13497", "FAM24765", 
"FAM24733", "FAM24780", "FAM24771", "FAM24737", "FAM24763", "FAM24793", 
"FAM24750", "FAM13494", "FAM13493", "FAM24766", "FAM24798"))
snps_long$samplesNew <- droplevels(snps_long$samplesNew)
# snps_long %>% filter(is.na(samplesNew)) %>% select(sample) %>% table()

sum(is.na(snps_long$samplesNew))

snps_long_final <- snps_long %>% select(site,species,samplesNew,culture,Snps)
table(snps_long$culture)

snps_long$samplesFINAL <- paste0(snps_long$culture,"-",snps_long$samplesNew)
table(snps_long$samplesFINAL )

# snps_long %>% filter(samplesFINAL=="FAMstrains-NA")

write.csv(snps_long_final,"~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all.csv",row.names = FALSE)
  # write.csv(naming_DF,"~/Desktop/Projects/2020_StarterCultureDiversity/98_script/naming_file.csv")


###----------------------------------------
##add SNPeffect INFO
###----------------------------------------


sample_relative_temp_cleaned <- snps_long_final

###spread the dataframe again

nrow(sample_relative_temp_cleaned)
table(sample_relative_temp_cleaned$culture)
sample_relative_temp_cleaned$samplesFINAL <- paste0(sample_relative_temp_cleaned$culture,"-",sample_relative_temp_cleaned$samplesNew)

table(sample_relative_temp_cleaned$samplesFINAL)

sample_relative_safe_final_prep <- sample_relative_temp_cleaned %>% select(site,species,samplesFINAL,Snps)

# table(sample_relative_safe_final_prep$samplesNew)
# table(sample_relative_safe_final_prep$culture)
table(sample_relative_safe_final_prep$samplesFINAL)

#nrow(sample_relative_safe_final_prep[grep("S_thermophilus_RMK202",sample_relative_safe_final_prep$site),])
sample_relative_safe_final_tsne <- spread(sample_relative_safe_final_prep, samplesFINAL, Snps)  %>%replace(is.na(.), 0)
nrow(sample_relative_safe_final_tsne)
# table(sample_relative_safe_final_prep$sample)
##==================
##05_filter rowsums!=0
##==================

##remove row sums ==0
nrow(sample_relative_safe_final_tsne)
sum(rowSums(sample_relative_safe_final_tsne[,3:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)==0)
# nrow(sample_relative_safe_final_tsne)


sample_relative_wide <- sample_relative_safe_final_tsne[rowSums(sample_relative_safe_final_tsne[,4:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)>0,]
colnames(sample_relative_wide)

# sample_relative_wide_phagesONly <- sample_relative_wide[grep("phage",sample_relative_wide$X.CHROM),]
# table(sample_relative_wide_phagesONly$X.CHROM)
# nrow(sample_relative_wide)
# sample_relative_wide <- subset(sample_relative_wide, select=-POS)
# sample_relative_wide$species <- sample_relative_wide$X.CHROM
# sample_relative_wide$species <- revalue(sample_relative_wide$species, c("CP046131"="L. delbrueckii","CP046134"="S. thermophilus"))
# sample_relative_wide$culture <- "RMK202"
# nrow(sample_relative_wide)
##==================
##06_add gene information
##==================

# RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new_startercultureDiversity/RMK202_snpeff_eggnog_repeats_core.txt")

# RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new_startercultureDiversity//RMK202_snpeff_eggnog_repeats_core_02.txt")
RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new_startercultureDiversity/RMK202_snpeff_eggnog_repeats_core_02.txt", col_types = cols(core = col_character(),geneName = col_character(),effect = col_character(),PGAP_name = col_character(),type = col_character(),typeName = col_character(),typeGene = col_character(),Preferred_name = col_character()))
table(RMK202_snpeff_eggnog_repeats_core$core)

# nrow(RMK202_snpeff_eggnog_repeats_core)
# table(RMK202_snpeff_eggnog_repeats_core$species)
# table(RMK202_snpeff_eggnog_repeats_core$core)
table(RMK202_snpeff_eggnog_repeats_core$core,RMK202_snpeff_eggnog_repeats_core$species)
# table(RMK202_snpeff_eggnog_repeats_core$core,RMK202_snpeff_eggnog_repeats_core$species)
# nrow(sample_relative_wide)
#RMK202_snpeff_eggnog_repeats_core_subset <- subset(RMK202_snpeff_eggnog_repeats_core, select=c(X.CHROM,site,finalName,effect,significance,geneName,GOs,EC,KEGG_ko,COG_Functional_Category,Repeat_identity,core))
dim(sample_relative_wide)
sample_relative_wide_description <- merge(sample_relative_wide,RMK202_snpeff_eggnog_repeats_core, by = "site",all.x = TRUE)
dim(sample_relative_wide_description)
table(sample_relative_wide_description$species)
# nrow(sample_relative_wide_description)
table(sample_relative_wide_description$core)
table(sample_relative_wide_description$significance)
#---------------------------------subset for interesting columns


# sample_relative_wide_description_interest <- subset(sample_relative_wide_description, select=c(species.x,site,culture,effect,significance,geneName,COG_Functional_Category,Repeat_cluster,core,culture,"starter 1","starter 2","starter 3","starter 4","starter 5","starter 6","starter 7","starter 8","starter 9","starter 10","starter 11","starter 12"))
dput(colnames(sample_relative_wide_description))
# site,species.x,effect,significance,geneName,COG_Functional_Category,Repeat_cluster,core
# sample_relative_wide_description_interest <- sample_relative_wide_description %>% select(species.x,site,culture,effect,significance,geneName,COG_Functional_Category,Repeat_cluster,core,culture,"starter 1","starter 2","starter 3","starter 4","starter 5","starter 6","starter 7","starter 8","starter 9","starter 10","starter 11","starter 12")
sample_relative_wide_description_interest <- sample_relative_wide_description %>% select(site,species.x,effect,significance,geneName,COG_Functional_Category,Repeat_cluster,core,"101-starter 1", "101-starter 10", "101-starter 11", 
"101-starter 2", "101-starter 3", "101-starter 4", "101-starter 5", 
"101-starter 6", "101-starter 7", "101-starter 8", "101-starter 9", 
"105-starter 1", "105-starter 10", "105-starter 11", "105-starter 12", 
"105-starter 13", "105-starter 14", "105-starter 15", "105-starter 16", 
"105-starter 2", "105-starter 3", "105-starter 4", "105-starter 5", 
"105-starter 6", "105-starter 7", "105-starter 8", "105-starter 9", 
"115-starter 1", "115-starter 2", "115-starter 3", "115-starter 4", 
"115-starter 5", "115-starter 6", "124-starter 1", "124-starter 2", 
"124-starter 3", "124-starter 4", "124-starter 5", "124-starter 6", 
"124-starter 7", "150-starter 1", "150-starter 10", "150-starter 2", 
"150-starter 3", "150-starter 4", "150-starter 5", "150-starter 6", 
"150-starter 7", "150-starter 8", "150-starter 9", "190-starter 1", 
"190-starter 2", "190-starter 3", "190-starter 4", "190-starter 5", 
"190-starter 6", "190-starter 7", "202-starter 1", "202-starter 2", 
"202-starter 3", "202-starter 4", "202-starter 5", "202-starter 6", 
"202-starter 9", "203-starter 1", "203-starter 2", "203-starter 3", 
"203-starter 4", "203-starter 5", "203-starter 6", "203-starter 7", 
"203-starter 8", "280-starter 1", "280-starter 2", "280-starter 3", 
"280-starter 4", "280-starter 5", "280-starter 6", "280-starter 7", 
"302-starter 1", "302-starter 2", "302-starter 3", "302-starter 4", 
"302-starter 5", "302-starter 6", "305-starter 1", "305-starter 2", 
"305-starter 3", "305-starter 4", "305-starter 5", "305-starter 6", 
"305-starter 7", "305-starter 8", "FAMstrains-FAM12107", "FAMstrains-FAM13491", 
"FAMstrains-FAM13492", "FAMstrains-FAM13493", "FAMstrains-FAM13494", 
"FAMstrains-FAM13495", "FAMstrains-FAM13496", "FAMstrains-FAM13497", 
"FAMstrains-FAM13498", "FAMstrains-FAM13499", "FAMstrains-FAM13500", 
"FAMstrains-FAM24724", "FAMstrains-FAM24725", "FAMstrains-FAM24726", 
"FAMstrains-FAM24727", "FAMstrains-FAM24728", "FAMstrains-FAM24729", 
"FAMstrains-FAM24730", "FAMstrains-FAM24731", "FAMstrains-FAM24732", 
"FAMstrains-FAM24733", "FAMstrains-FAM24734", "FAMstrains-FAM24735", 
"FAMstrains-FAM24736", "FAMstrains-FAM24737", "FAMstrains-FAM24738", 
"FAMstrains-FAM24739", "FAMstrains-FAM24740", "FAMstrains-FAM24741", 
"FAMstrains-FAM24742", "FAMstrains-FAM24743", "FAMstrains-FAM24744", 
"FAMstrains-FAM24745", "FAMstrains-FAM24746", "FAMstrains-FAM24747", 
"FAMstrains-FAM24748", "FAMstrains-FAM24749", "FAMstrains-FAM24750", 
"FAMstrains-FAM24751", "FAMstrains-FAM24752", "FAMstrains-FAM24753", 
"FAMstrains-FAM24754", "FAMstrains-FAM24755", "FAMstrains-FAM24756", 
"FAMstrains-FAM24757", "FAMstrains-FAM24758", "FAMstrains-FAM24759", 
"FAMstrains-FAM24760", "FAMstrains-FAM24761", "FAMstrains-FAM24762", 
"FAMstrains-FAM24763", "FAMstrains-FAM24764", "FAMstrains-FAM24765", 
"FAMstrains-FAM24766", "FAMstrains-FAM24767", "FAMstrains-FAM24768", 
"FAMstrains-FAM24769", "FAMstrains-FAM24770", "FAMstrains-FAM24771", 
"FAMstrains-FAM24772", "FAMstrains-FAM24773", "FAMstrains-FAM24774", 
"FAMstrains-FAM24775", "FAMstrains-FAM24776", "FAMstrains-FAM24777", 
"FAMstrains-FAM24778", "FAMstrains-FAM24779", "FAMstrains-FAM24780", 
"FAMstrains-FAM24781", "FAMstrains-FAM24782", "FAMstrains-FAM24783", 
"FAMstrains-FAM24784", "FAMstrains-FAM24785", "FAMstrains-FAM24786", 
"FAMstrains-FAM24787", "FAMstrains-FAM24790", "FAMstrains-FAM24791", 
"FAMstrains-FAM24792", "FAMstrains-FAM24793", "FAMstrains-FAM24794", 
"FAMstrains-FAM24795", "FAMstrains-FAM24797", "FAMstrains-FAM24798", 
"FAMstrains-FAM24799", "FAMstrains-FAM24853", "FAMstrains-FAM24854", 
"FAMstrains-FAM24855")


# dim(sample_relative_wide_description)
# dim(sample_relative_wide_description_interest)
# table(sample_relative_wide_description_interest$core)

# sample_relative_wide_description_interest <- sample_relative_wide_description

##==================
##08_remove non-bacterial SNPs
##==================
table(sample_relative_wide_description_interest$species.x)
nrow(sample_relative_wide_description_interest)
# sample_relative_wide_description_interest_subset <- sample_relative_wide_description_interest[which(sample_relative_wide_description_interest$X.CHROM.x %in% c("L_delbrueckii_RMK202","S_thermophilus_RMK202")),]
sample_relative_wide_description_interest_subset <- sample_relative_wide_description_interest[which(sample_relative_wide_description_interest$species.x %in% c("L. delbrueckii","S. thermophilus")),]
nrow(sample_relative_wide_description_interest_subset)
table(sample_relative_wide_description_interest_subset$species.x)
sample_relative_wide <- sample_relative_wide_description_interest_subset
table(sample_relative_wide_description_interest_subset$core)
table(sample_relative_wide_description_interest_subset$core,sample_relative_wide_description_interest_subset$species.x)


##==================
##09_order variables properly
##==================


sample_relative_wide$significance <- factor(sample_relative_wide$significance, levels=c("MODIFIER","LOW","MODERATE","HIGH"))


##-------------------------------wirte to file

# write.xlsx(dN_dS_ratios_final_02,file="Users//Desktop/presenations/my_presentations/20191007_groupmeeting/figures/dN_dS_ratios_genes_rmk202.xls", sheetName = "Sheet1", 
  # col.names = TRUE, row.names = FALSE, append = FALSE)

##==================
##11_make long
##==================
colnames(sample_relative_wide)
# sample_relative_long <- gather(sample_relative_wide, sample, Snps, "starter 1":"starter 12", factor_key=TRUE,na.rm = TRUE) 
sample_relative_long <- gather(sample_relative_wide, sample, Snps, "101-starter 1":"FAMstrains-FAM24855", factor_key=TRUE,na.rm = TRUE) %>% filter(Snps>0)%>% filter(!is.na(Snps))
nrow(sample_relative_long)
table(sample_relative_long$species.x)
table(sample_relative_long$core)

table(sample_relative_long$culture)
sample_relative_long$samplesss < paste0(sample_relative_long$culture,"_",sample_relative_long$sample)
# sample_relative_long_again <- spread(sample_relative_long, site, Snps)%>%replace(is.na(.), 0) %>% column_to_rownames(var = "sampleName")


##==================
##12_make a unique gene infomoration
##==================

# RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/RMK202_snpeff_eggnog_repeats_core.txt")
# RMK202_snpeff_eggnog_repeats_core_uniuqes <- RMK202_snpeff_eggnog_repeats_core %>% select(-c("X1","X.CHROM","POS","REF","ALT","quality","site","effect","significance","best_tax_level")) %>% distinct()
# table(RMK202_snpeff_eggnog_repeats_core_uniuqes$core)
# nrow(RMK202_snpeff_eggnog_repeats_core_uniuqes)
# RMK202_snpeff_eggnog_repeats_core_uniuqes_final <- merge(RMK202_snpeff_eggnog_repeats_core_uniuqes,dN_dS_ratios_prep,by.y="gene",by.x="finalName",all.x = TRUE)  

# write.csv(RMK202_snpeff_eggnog_repeats_core_uniuqes_final,file="~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/RMK202_snpeff_eggnog_repeats_core_dNdS_uniqueGenes.txt")

# colnames(RMK202_snpeff_eggnog_repeats_core_uniuqes_final)
# RMK202_snpeff_eggnog_repeats_core_uniuqes_final_reduce <- RMK202_snpeff_eggnog_repeats_core %>% select(-c("finalName","Preferred_name","EC","KEGG_ko","KEGG_Reaction","CAZy","BiGG_Reaction","COG_Functional_Category","eggNOG free text description")) %>% distinct()

##==================
##output infomration
##==================
# sample_relative_wide
write.csv(sample_relative_wide,"~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all_withINfomration_wide.csv",row.names = FALSE)
# table(sample_relative_wide$core)
# unique(sample_relative_wide$core)
dim(sample_relative_wide)
sample_relative_wide[which(sample_relative_wide$core=="core"),]

snps_long_final <-  gather(sample_relative_wide, sample, Snps, "101-starter 1":"FAMstrains-FAM24855", factor_key=TRUE,na.rm = TRUE)
snps_long_final[which(snps_long_final$core=="core"),]

snps_long_final$species <- snps_long_final$species.x
snps_long_final$culture <- str_split_fixed(snps_long_final$sample,"-",3)[,1]
snps_long_final <- snps_long_final %>% select(-"species.x")
dim(snps_long_final$core)
```


## strian phasing

```{bash}

library(robustbase)
library(VennDiagram)
##==================================================Streptococcus thermophilus===================================================================
colnames(sample_relative_wide)
colnames(sample_relative_wide) <- colnames(sample_relative_wide) %>% gsub("FAMstrains-FAM","",.)

dput(colnames(sample_relative_wide))


sample_relative_wide_snpsUsed <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")    %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 | "experiment_A">0.05| "experiment_B">0.05| "experiment_C">0.05 | "experiment_D">0.05 | "experiment_E">0.05)



"101-starter 1", 
"101-starter 10", "101-starter 11", "101-starter 2", "101-starter 3", 
"101-starter 4", "101-starter 5", "101-starter 6", "101-starter 7", 
"101-starter 8", "101-starter 9", "105-starter 1", "105-starter 10", 
"105-starter 11", "105-starter 12", "105-starter 13", "105-starter 14", 
"105-starter 15", "105-starter 16", "105-starter 2", "105-starter 3", 
"105-starter 4", "105-starter 5", "105-starter 6", "105-starter 7", 
"105-starter 8", "105-starter 9", "115-starter 1", "115-starter 2", 
"115-starter 3", "115-starter 4", "115-starter 5", "115-starter 6", 
"124-starter 1", "124-starter 2", "124-starter 3", "124-starter 4", 
"124-starter 5", "124-starter 6", "124-starter 7", "150-starter 1", 
"150-starter 10", "150-starter 2", "150-starter 3", "150-starter 4", 
"150-starter 5", "150-starter 6", "150-starter 7", "150-starter 8", 
"150-starter 9", "190-starter 1", "190-starter 2", "190-starter 3", 
"190-starter 4", "190-starter 5", "190-starter 6", "190-starter 7", 
"202-starter 1", "202-starter 2", "202-starter 3", "202-starter 4", 
"202-starter 5", "202-starter 6", "202-starter 9", "203-starter 1", 
"203-starter 2", "203-starter 3", "203-starter 4", "203-starter 5", 
"203-starter 6", "203-starter 7", "203-starter 8", "280-starter 1", 
"280-starter 2", "280-starter 3", "280-starter 4", "280-starter 5", 
"280-starter 6", "280-starter 7", "302-starter 1", "302-starter 2", 
"302-starter 3", "302-starter 4", "302-starter 5", "302-starter 6", 
"305-starter 1", "305-starter 2", "305-starter 3", "305-starter 4", 
"305-starter 5", "305-starter 6", "305-starter 7", "305-starter 8", 
"12107", "13491", "13492", "13493", "13494", "13495", "13496", 
"13497", "13498", "13499", "13500", "24724", "24725", "24726", 
"24727", "24728", "24729", "24730", "24731", "24732", "24733", 
"24734", "24735", "24736", "24737", "24738", "24739", "24740", 
"24741", "24742", "24743", "24744", "24745", "24746", "24747", 
"24748", "24749", "24750", "24751", "24752", "24753", "24754", 
"24755", "24756", "24757", "24758", "24759", "24760", "24761", 
"24762", "24763", "24764", "24765", "24766", "24767", "24768", 
"24769", "24770", "24771", "24772", "24773", "24774", "24775", 
"24776", "24777", "24778", "24779", "24780", "24781", "24782", 
"24783", "24784", "24785", "24786", "24787", "24790", "24791", 
"24792", "24793", "24794", "24795", "24797", "24798", "24799", 
"24853", "24854", "24855"


colnames(sample_relative_wide_snpsUsed)
sample_relative_wide_snpsUsed_lin1 <- sample_relative_wide_snpsUsed %>% filter(`24853`>0.8| `24798`>0.8| `13493`>0.8| `13500`>0.8 | `24737`>0.8 | `24854`>0.8| `13491`>0.8 | `13492`>0.8 ) %>% add_column(explained="lin1")
sample_relative_wide_snpsUsed_lin2 <- sample_relative_wide_snpsUsed %>% filter(`S72`>0.8| `24855`>0.8| `13494`>0.8)%>% add_column(explained="lin2")
sample_relative_wide_snpsUsed_lin3 <- sample_relative_wide_snpsUsed %>% filter(`S50`>0.8| `24740`>0.8| `24738`>0.8| `13499`>0.8 )%>% add_column(explained="lin3")
sample_relative_wide_snpsUsed_lin4 <- sample_relative_wide_snpsUsed %>% filter(`24739`>0.8| `13497`>0.8| `13496`>0.8| `13495`>0.8 )%>% add_column(explained="lin4")
allsites <- c(sample_relative_wide_snpsUsed_lin1$site, sample_relative_wide_snpsUsed_lin2$site, sample_relative_wide_snpsUsed_lin3$site,sample_relative_wide_snpsUsed_lin4$site)

##----------------------
# Chart venn diagramm
##----------------------
temp <-venn.diagram(
  x = list(sample_relative_wide_snpsUsed_lin1$site, sample_relative_wide_snpsUsed_lin2$site, sample_relative_wide_snpsUsed_lin3$site,sample_relative_wide_snpsUsed_lin4$site),
  category.names = c("lin 1" , "lin 2 " , "lin 3", "lin 4"),
  filename = NULL
)

plot.new() 

grid.draw(temp)
# 
# pdf("testpdf", width = 14, height = 7)
# 
# grid.draw(temp)
# 
# dev.off()

grid.draw(temp)

allsites[duplicated(allsites)]
sum(duplicated(allsites))
sum(!duplicated(allsites))

sum(table(allsites)>1)
sum(table(allsites)==1)

duplictednames <- names(table(allsites)[(table(allsites)>1)])

##----------------------
# Chart venn diagramm
##----------------------

sample_relative_wide_snpsUsed_new_01 <- rbind(sample_relative_wide_snpsUsed_lin1,sample_relative_wide_snpsUsed_lin2,sample_relative_wide_snpsUsed_lin3,sample_relative_wide_snpsUsed_lin4)
sample_relative_wide_snpsUsed_new_02 <- sample_relative_wide_snpsUsed_new_01 %>% filter(!site %in%duplictednames) #lineage specific snps
table(sample_relative_wide_snpsUsed_new_02$explained)
sample_relative_wide_snpsUsed_new_temp <- sample_relative_wide_snpsUsed %>% filter(site %in%duplictednames)%>% add_column(explained="multiple") ##duplicated sites
sample_relative_wide_snpsUsed_new_03 <- rbind(sample_relative_wide_snpsUsed_new_02,sample_relative_wide_snpsUsed_new_temp) ##all explained sites
sample_relative_wide_snpsUsed_tmp <-  sample_relative_wide_snpsUsed %>% filter(!site %in%sample_relative_wide_snpsUsed_new_03$site) %>% add_column(explained="not explained")
sample_relative_wide_snpsUsed_final <- rbind(sample_relative_wide_snpsUsed_new_03,sample_relative_wide_snpsUsed_tmp) ##all explained sites


##EXKURSION LINEAGE SPECIFIC DUPLICATIONS
TMP <- sample_relative_wide_snpsUsed %>% filter(site %in%duplictednames)
TMP$e

short_explained <- sample_relative_wide_snpsUsed_new_01 %>%filter(site %in%duplictednames) %>%  select(c("site","species","explained")) %>% mutate(abundance = explained) %>% spread(., explained, abundance)
short_explained$explained <- paste(short_explained$lin1,short_explained$lin2,short_explained$lin3,short_explained$lin4,sep="_")
short_explained_multiple <- short_explained%>% select("site","explained")
sample_relative_wide_snpsUsed_new_temp <- sample_relative_wide_snpsUsed %>% filter(site %in%duplictednames) ##duplicated sites
sample_relative_wide_snpsUsed_new_temp_02 <- merge(sample_relative_wide_snpsUsed_new_temp,short_explained_multiple,by="site")
sample_relative_wide_snpsUsed_new_03 <- rbind(sample_relative_wide_snpsUsed_new_02,sample_relative_wide_snpsUsed_new_temp_02) ##all explained sites
sample_relative_wide_snpsUsed_tmp <-  sample_relative_wide_snpsUsed %>% filter(!site %in%sample_relative_wide_snpsUsed_new_03$site) %>% add_column(explained="not explained")
sample_relative_wide_snpsUsed_final <- rbind(sample_relative_wide_snpsUsed_new_03,sample_relative_wide_snpsUsed_tmp) ##all explained sites


totalSNPS <- nrow(sample_relative_wide_snpsUsed_final)
100*(table(sample_relative_wide_snpsUsed_final$explained)/totalSNPS)
table(sample_relative_wide_snpsUsed_final$explained)
##----------------------
# boxplot
##----------------------
# sample_relative_wide_snpsUsed_final_long <- sample_relative_wide_snpsUsed_final %>% select( "site","Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E","explained")%>% gather(.,sample,Median,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E") 

sample_relative_wide_snpsUsed_final <- sample_relative_wide_snpsUsed_final %>%  dplyr::rename("Reference 2"="cheesemaking\nday2","Reference 1"="cheesemaking\nday1" )

sample_relative_wide_snpsUsed_final_long <- sample_relative_wide_snpsUsed_final %>% select( "site","Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E","Reference 1","Reference 2","explained")%>% gather(.,sample,Median,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E","Reference 1","Reference 2") 

# ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median))+geom_boxplot()+theme_classic()+facet_wrap(~explained)

#remove zeroes
sample_relative_wide_snpsUsed_final_long_woZeros <- sample_relative_wide_snpsUsed_final_long %>% filter(Median>0.1)
ggplot(sample_relative_wide_snpsUsed_final_long_woZeros,aes(x=sample,y=Median))+geom_boxplot()+theme_classic()+facet_wrap(~explained)


##----------------------
# lineplot
##----------------------
sample_relative_wide_snpsUsed_final_long_woZeros$sample = factor(sample_relative_wide_snpsUsed_final_long_woZeros$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E","Reference 1","Reference 2"))

sample_relative_wide_snpsUsed_final_long_woZeros$explained = factor(sample_relative_wide_snpsUsed_final_long_woZeros$explained, levels=c("lin1","lin2","lin3","lin4","not explained","lin1_lin2_NA_NA","NA_NA_lin3_lin4","lin1_lin2_lin3_lin4","lin1_lin2_NA_lin4","lin1_NA_lin3_lin4","lin1_NA_lin3_NA","lin1_NA_NA_lin4","NA_lin2_lin3_NA","NA_lin2_NA_lin4","NA_lin2_lin3_lin4"))


# ggplot(sample_relative_wide_snpsUsed_final_long_woZeros,aes(x=sample,y=Median,group=site))+geom_line(color="red",size=0.2, alpha=.1)+theme_classic()
sample_relative_wide_snpsUsed_final_long_woZeros <- sample_relative_wide_snpsUsed_final_long_woZeros %>% filter(explained!="not explained")

pExplainted <- ggplot(sample_relative_wide_snpsUsed_final_long_woZeros,aes(x=sample,y=Median,group=site,color=explained,fill=explained))+geom_line(alpha=.5,size=0.5)+theme_classic()+facet_wrap(~explained)+theme(legend.position = "none",axis.text.x = element_text(angle = 75, hjust = 1,size=9))

table(sample_relative_wide_snpsUsed_final_long_woZeros$explained)
pExplainted


  svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_explained_by_isolates.svg",width=7,height=4.5)
# png("~/Desktop/Projects/2019_RMK202_analysis/plot/supp_altNucFre_lineages.png", width = 3400, height = 2600,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted
dev.off()
###--------------------
##subset for ppx
table(sample_relative_wide_snpsUsed_final_long_woZeros$explained)
# sample_relative_wide_snpsUsed_final_long_tmp <-  sample_relative_wide_snpsUsed_final_long_woZeros %>% filter(explained %in% c("NA_NA_lin3_lin4","NA_lin2_lin3_lin4","NA_lin2_NA_lin4","lin3"))

sample_relative_wide_snpsUsed_final_long_tmp <-  sample_relative_wide_snpsUsed_final_long_woZeros %>% filter(explained %in% c("lin1_NA_NA_lin4","NA_lin2_lin3_NA","NA_lin2_NA_lin4","lin1_NA_lin3_NA"))


  sample_relative_wide_snpsUsed_final_long_tmp$explained = factor(sample_relative_wide_snpsUsed_final_long_tmp$explained, levels=c("lin1_NA_NA_lin4","NA_lin2_lin3_NA","NA_lin2_NA_lin4","lin1_NA_lin3_NA"))

# sample_relative_wide_snpsUsed_final_long$explained_02 <- revalue(sample_relative_wide_snpsUsed_final_long$explained, c("lin1"="explained by isolates", "lin2"="explained by isolates","lin3"="explained by isolates","lin4"="explained by isolates","multiple"="explained by isolates","not explained"="not explained by isolates"))

pExplainted <- ggplot(sample_relative_wide_snpsUsed_final_long_tmp,aes(x=sample,y=Median,group=site,color=explained,fill=explained))+geom_line(alpha=.5,size=0.5)+theme_classic()+facet_wrap(~explained,ncol=4)+theme(legend.position = "none",axis.text.x = element_text(angle = 75, hjust = 1,size=6))

pExplainted

png("~/Desktop/Projects/2019_RMK202_analysis/plot/supp_altNucFre_lineages_03.png", width = 3400, height = 1000,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted
dev.off()

###--------------------

sample_relative_wide_snpsUsed_final_long$explained_02 <- revalue(sample_relative_wide_snpsUsed_final_long$explained, c("lin1"="explained by isolates", "lin2"="explained by isolates","lin3"="explained by isolates","lin4"="explained by isolates","multiple"="explained by isolates","not explained"="not explained by isolates"))

pExplainted <- ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median,group=site,color=explained_02,fill=explained_02))+geom_line(size=0.5,alpha=0.5)+theme_classic()+facet_wrap(~explained_02)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+
    labs("",
         x="",
         y="Alternative allele frequency")
pExplainted

pExplainted <- ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median,group=site,color=explained_02,fill=explained_02))+geom_line(size=0.5,alpha=0.5)+theme_classic()+facet_wrap(~explained_02)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+
    labs("",
         x="",
         y="Alternative allele frequency")
pExplainted


  svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_explained_by_isolates..svg",width=7,height=4.5)
#    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted

dev.off()



##----------------------
# not explained
##----------------------

sample_relative_wide_snpsUsed_final_long$explained_03 <- ifelse(sample_relative_wide_snpsUsed_final_long$explained_02=="not explained by isolates","not explained by isolates","explained")
  
  sample_relative_wide_snpsUsed_final_long$sample = factor(sample_relative_wide_snpsUsed_final_long$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E"))

  
pExplainted_rough <- ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median,group=site,color=explained_03,fill=explained_03))+geom_line(size=0.5,alpha=0.5)+theme_classic()+facet_wrap(~explained_03)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+
    labs("",
         x="",
         y="Alternative allele frequency")
pExplainted_rough

  svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_notExplained.svg",width=7,height=4.5)
#    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted_rough

dev.off()

##----------------------
# where do these funky linkes (~recombinatnts) locate on the genome
##----------------------

sample_relative_wide_snpsUsed_final_long_recombinants <- sample_relative_wide_snpsUsed_final_long %>% filter(explained %in% c("lin1_lin2_NA_lin4","lin1_NA_lin3_lin4","lin1_NA_lin3_NA","lin1_NA_NA_lin4","NA_lin2_lin3_NA","NA_lin2_NA_lin4","NA_lin2_lin3_lin4")) %>% select("site","explained") %>% unique() 

sample_relative_wide_snpsUsed_final_long_recombinants <- sample_relative_wide_snpsUsed_final_long %>% select("site","explained") %>% unique() 


sample_relative_wide_snpsUsed_final_long_recombinants$location <- as.numeric(str_split_fixed(sample_relative_wide_snpsUsed_final_long_recombinants$site, "_", 4)[,2])
sample_relative_wide_snpsUsed_final_long_recombinants$explained = factor(sample_relative_wide_snpsUsed_final_long_recombinants$explained, levels=c("lin1","lin2","lin3","lin4","not explained","lin1_lin2_NA_NA","NA_NA_lin3_lin4","lin1_lin2_lin3_lin4","NA_lin2_lin3_lin4","lin1_lin2_NA_lin4","lin1_NA_lin3_lin4","lin1_NA_lin3_NA","lin1_NA_NA_lin4","NA_lin2_lin3_NA","NA_lin2_NA_lin4"))


sample_relative_wide_snpsUsed_final_long_recombinants$explained_2 <- revalue(sample_relative_wide_snpsUsed_final_long_recombinants$explained, c("lin1_lin2_NA_NA"="lin1_lin2_NA_NA=~evolved_after_split","NA_NA_lin3_lin4"="NA_NA_lin3_lin4=~evolved_after_split","lin1_lin2_lin3_lin4"="lin1_lin2_lin3_lin4=polishingERRORs","NA_lin2_lin3_lin4"="NA_lin2_lin3_lin4=lost_in_lin1","lin1_NA_lin3_lin4"="lin1_NA_lin3_lin4=lost_in_lin2","lin1_lin2_NA_lin4"="lin1_lin2_NA_lin4=lost_in_lin3","lin1_NA_lin3_NA"="lin1_NA_lin3_NA=recombination","lin1_NA_NA_lin4"="lin1_NA_NA_lin4=recombination","NA_lin2_lin3_NA"="NA_lin2_lin3_NA=recombination","NA_lin2_NA_lin4"="NA_lin2_NA_lin4=recombination"))

# ggplot(sample_relative_wide_snpsUsed_final_long_recombinants,aes(x=location,color=explained,fill=explained))+geom_density()+facet_wrap(~explained,scales = "free_y")+theme_classic()
plocation <- ggplot(sample_relative_wide_snpsUsed_final_long_recombinants,aes(x=location,color=explained,fill=explained))+geom_histogram()+facet_wrap(~explained_2,scales = "free_y")+theme_classic()+labs(title="location of SNVs coming from different ",x="genomic location")+theme(legend.position = "none")

plocation


  # svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_explained_by_isolates..svg",width=7,height=4.5)
png("~/Desktop/supp_altNucFre_lineages_location.png", width = 3400, height = 2600,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
plocation
dev.off()

##----------------------
# not explained
##----------------------
table(sample_relative_wide_snpsUsed_final_long$explained)
sample_relative_wide_snpsUsed_final_long_notExplained <- sample_relative_wide_snpsUsed_final_long %>% filter(explained=="not explained")

ggplot(sample_relative_wide_snpsUsed_final_long_notExplained,aes(x=Median))+geom_density()+facet_wrap(~sample)+theme_classic()


###nomany multiallelic sites
sitesss <- str_split_fixed(sample_relative_wide_snpsUsed_final_long_notExplained$site, "_", 4)[,2]
length(sitesss)/6
unique(sitesss) %>% length()

sitesss <- str_split_fixed(sample_relative_wide_snpsUsed_final_long$site, "_", 4)[,2]
length(sitesss) /6
unique(sitesss) %>% length()

##----------------------
# meansss and medianss
##----------------------
library(patchwork)
# mediannssss <-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long[,c("explained","sample","Median")], median, na.rm=TRUE)%>% filter(explained!="multiple")
# meanssss <-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long[,c("explained","sample","Median")], median, na.rm=TRUE) %>% filter(explained!="multiple")
# 
# mediansplot <- ggplot(mediannssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat="identity")+theme_classic()+theme(legend.position = "none")
# meansssplot <- ggplot(meanssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat="identity")+theme_classic()
# 
# mediansplot+meansssplot

##---without zeros
# table(sample_relative_wide_snpsUsed_final_long_woZeros$sample)
mediannssss <-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long_woZeros[,c("explained","sample","Median")], median, na.rm=TRUE)%>% filter(explained!="multiple")
meanssss <-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long_woZeros[,c("explained","sample","Median")], median, na.rm=TRUE) %>% filter(explained!="multiple")

mediansplot <- ggplot(mediannssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat="identity")+theme_classic()+theme(legend.position = "none")
meansssplot <- ggplot(meanssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat="identity")+theme_classic()

mediansplot+meansssplot

##---------------------------------
##lineage abundance
##---------------------------------
ratio_larger <- mediannssss %>% filter(explained=="NA_NA_lin3_lin4") %>% select("sample","Median")  %>% dplyr::rename(Overall_ratio = Median) 
ratio_larger_2 <- mediannssss %>% filter(explained=="lin2") %>% select("sample","Median")  %>% dplyr::rename(rel_lin2 = Median) 
ratio_larger_3 <- mediannssss %>% filter(explained=="lin3") %>% select("sample","Median")  %>% dplyr::rename(rel_lin3 = Median) 
# ratio_larger_4 <- mediannssss %>% filter(explained=="lin4") %>% select("sample","Median")  %>% dplyr::rename(rel_lin4 = Median) 

ratio_together_1 <- merge(ratio_larger,ratio_larger_2,by="sample",all.x = TRUE)
ratio_together_2 <- merge(ratio_together_1,ratio_larger_3,by="sample",all.x = TRUE)
ratio_together_2$lin1 <- (1-ratio_together_2$Overall_ratio)*(1-ratio_together_2$rel_lin2)
ratio_together_2$lin2 <- (1-ratio_together_2$Overall_ratio)*(ratio_together_2$rel_lin2)
ratio_together_2$lin3 <- (ratio_together_2$Overall_ratio)*(ratio_together_2$rel_lin3)
ratio_together_2$lin4 <- (ratio_together_2$Overall_ratio)*(1-ratio_together_2$rel_lin3)


ratio_larger_4 <- mediannssss %>% filter(explained=="lin3") %>% select("sample","Median")  %>% dplyr::rename(rel_lin3 = Median) 


##ratios in Reference sequences
#ther is no lin3 or lin4 specific hits that is why we cannot calculate it like before. 
#lin1 is the reference everything that is not lin1 is lin2. 

REffs2 <- mediannssss %>% filter(sample=="Reference 1") %>% filter(explained=="lin2") %>% select(Median) %>% unlist() %>% as.numeric()
ratio_together_final_referenceSamples <- data.frame(sample="Reference 1",lin4=0,lin3=0,lin2=REffs2,lin1=1-REffs2)
REffs2 <- mediannssss %>% filter(sample=="Reference 2") %>% filter(explained=="lin2") %>% select(Median) %>% unlist() %>% as.numeric()
ratio_together_final_referenceSamples2 <- data.frame(sample="Reference 2",lin4=0,lin3=0,lin2=REffs2,lin1=1-REffs2)

# ratio_together_final_referenceSamples <- 

# ratio_together_2$Unknown <- 1-(ratio_together_2$lin1+ratio_together_2$lin2+ratio_together_2$lin3+ratio_together_2$lin4)

# ratio_together_final <- ratio_together_2 %>% select(sample,lin1,lin2,lin3,lin4) %>% gather(.,lineage,"Relative abundance","lin1","lin2","lin3","lin4")
ratio_together_final_tmp1 <- ratio_together_2 %>% select(sample,lin4,lin3,lin2,lin1) 

ratio_together_final <- rbind(ratio_together_final_tmp1,ratio_together_final_referenceSamples,ratio_together_final_referenceSamples2) %>% gather(.,lineage,"Relative abundance","lin4","lin3","lin2","lin1")

table(ratio_together_final$sample)



ratio_together_final <- ratio_together_final %>% filter(!sample %in% c("Reference 1","Reference 2"))
ratio_together_final$sample = factor(ratio_together_final$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E","Reference 1","Reference 2"))

ratio_together_final$lineage = factor(ratio_together_final$lineage, levels=(c("lin4","lin3","lin2","lin1")))


# colorsss <- c("#0000FF","#6699FF","#99CCFF","#00FFFF")
colorsss <- c("#99CCFF","#00FFFF","#0000FF","#6699FF")
colorsss <- c("#99CCFF","#00d5d5ff","#0000FF","#6699FF")


plot_strain_abundance <- ggplot(ratio_together_final,aes(x=sample,y=`Relative abundance`,color=lineage,fill=lineage))+geom_bar(stat="identity")+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+
    labs(x="")+scale_color_manual(values=colorsss)+scale_fill_manual(values=colorsss)+theme(legend.position = "none")
plot_strain_abundance
# png("~/Desktop/supp_strain_Abundance.png", width = 2800, height = 2200,res=300)
svg("~/Desktop/Projects/2019_RMK202_analysis/plot/strain_rel_abundance.svg",width=6,height=3.5)

# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
plot_strain_abundance
dev.off()


##area plot

plot_strain_abreasss <- ggplot(ratio_together_final,aes(x=sample,y=`Relative abundance`,group=lineage,color=lineage,fill=lineage))+geom_area(alpha=0.5)+geom_point(position = "stack",size=2)+geom_line(position = "stack",size=1.25)+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+
    labs(x="")+scale_color_manual(values=colorsss)+scale_fill_manual(values=colorsss)+theme(legend.position = "none")+scale_x_discrete( expand = c(0.01,0.01)) +scale_y_discrete( expand = c(0.01,0.01)) 

plot_strain_abreasss


svg("~/Desktop/Projects/2019_RMK202_analysis/plot/strain_rel_abundance_02.svg",width=2.5,height=3.5)

# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
plot_strain_abreasss
dev.off()
```


##not explained 

Here, we test which SNVs are not explained with the isolated strains.
```{r}



colnames(sample_relative_wide)
sample_relative_wide_explained <- sample_relative_wide
sample_relative_wide_explained$metagenomSNP_onlyNewest_sample <- rowSums(sample_relative_wide[,c(15,27,29,32,38,41,47,51,54,56,60)])
sample_relative_wide_explained$metagenomSNP <- rowSums(sample_relative_wide[,9:60])
sample_relative_wide_explained$StrainSNP <- rowSums(sample_relative_wide[,61:131])
sample_relative_wide_explained_extended <- sample_relative_wide_explained %>% select(site,species.x,metagenomSNP,StrainSNP,metagenomSNP_onlyNewest_sample,effect,significance,geneName,COG_Functional_Category,Repeat_cluster,core)







##----------------------
##snps explained in current working stock
##----------------------

notExplaine <- sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP<0.8) %>% filter(metagenomSNP_onlyNewest_sample>0.1) %>% nrow()
explaine <- sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP>=0.8) %>% filter(metagenomSNP_onlyNewest_sample>0.1)%>% nrow()

100*(notExplaine/explaine)

ggplot(notExplained,aes(metagenomSNP))+geom_density()+facet_wrap(~species.x,scales = "free")



sample_relative_wide_explained$explained <- ifelse(sample_relative_wide_explained_extended$StrainSNP<0.8,"not explained","explained")
colnames(sample_relative_wide_explained)
dput(colnames(sample_relative_wide_explained))

sample_relative_wide_explained_long_onlyNewest <- sample_relative_wide_explained%>% filter(core=="core") %>% filter(metagenomSNP_onlyNewest_sample>0.1)%>% select("site", "species.x","explained", "101-starter 1", 
"101-starter 10", "101-starter 11", "101-starter 2", "101-starter 3", 
"101-starter 4", "101-starter 5", "101-starter 6", "101-starter 7", 
"101-starter 8", "101-starter 9", "105-starter 1", "105-starter 10", 
"105-starter 11", "105-starter 12", "105-starter 13", "105-starter 14", 
"105-starter 15", "105-starter 16", "105-starter 2", "105-starter 3", 
"105-starter 4", "105-starter 5", "105-starter 6", "105-starter 7", 
"105-starter 8", "105-starter 9", "115-starter 1", "115-starter 2", 
"115-starter 3", "115-starter 4", "115-starter 5", "115-starter 6", 
"124-starter 1", "124-starter 2", "124-starter 3", "124-starter 4", 
"124-starter 5", "124-starter 6", "124-starter 7", "150-starter 1", 
"150-starter 10", "150-starter 2", "150-starter 3", "150-starter 4", 
"150-starter 5", "150-starter 6", "150-starter 7", "150-starter 8", 
"150-starter 9", "190-starter 1", "190-starter 2", "190-starter 3", 
"190-starter 4", "190-starter 5", "190-starter 6", "190-starter 7", 
"202-starter 1", "202-starter 2", "202-starter 3", "202-starter 4", 
"202-starter 5", "202-starter 6", "202-starter 9", "203-starter 1", 
"203-starter 2", "203-starter 3", "203-starter 4", "203-starter 5", 
"203-starter 6", "203-starter 7", "203-starter 8", "280-starter 1", 
"280-starter 2", "280-starter 3", "280-starter 4", "280-starter 5", 
"280-starter 6", "280-starter 7", "302-starter 1", "302-starter 2", 
"302-starter 3", "302-starter 4", "302-starter 5", "302-starter 6", 
"305-starter 1", "305-starter 2", "305-starter 3", "305-starter 4", 
"305-starter 5", "305-starter 6", "305-starter 7", "305-starter 8") %>% gather(., sample, Snps, "101-starter 1":"305-starter 4", factor_key=TRUE,na.rm = TRUE) %>% filter(Snps>0.05)

ggplot(sample_relative_wide_explained_long_onlyNewest,aes(x=explained,y=Snps))+geom_boxplot()+theme_classic()

ggplot(sample_relative_wide_explained_long_onlyNewest,aes(x=Snps,group=explained,fill=explained))+geom_histogram(alpha=0.2, position="identity",bins = 100)+theme_classic()
table(sample_relative_wide_explained_long_onlyNewest$sample)

    t.test(Snps ~ explained, data = sample_relative_wide_explained_long_onlyNewest)



##----------------------
##snps explained from all samples
##----------------------

sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP<0.8) %>% filter(metagenomSNP>0.1) %>% nrow()
sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP>=0.8) %>% filter(metagenomSNP>0.1)%>% nrow()



notExplained <- sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP<0.8) %>% filter(metagenomSNP>1)

ggplot(notExplained,aes(metagenomSNP))+geom_density()+facet_wrap(~species.x,scales = "free")
table(notExplained$species.x)
sample_relative_wide_explained_extended %>% filter(core=="core") %>%  nrow()

sum(sample_relative_wide_explained_extended$StrainSNP<0.8)
sum(sample_relative_wide_explained_extended$StrainSNP>0.8)

sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP<0.8) %>% filter(metagenomSNP>0.1) %>% nrow()
sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP>=0.8) %>% filter(metagenomSNP>0.1)%>% nrow()

sample_relative_wide_explained$explained <- ifelse(sample_relative_wide_explained_extended$StrainSNP<0.8,"not explained","explained")
colnames(sample_relative_wide_explained)
dput(colnames(sample_relative_wide_explained))

sample_relative_wide_explained_long <- sample_relative_wide_explained%>% filter(core=="core") %>% select("site", "species","explained", "101-starter 1", 
"101-starter 10", "101-starter 11", "101-starter 2", "101-starter 3", 
"101-starter 4", "101-starter 5", "101-starter 6", "101-starter 7", 
"101-starter 8", "101-starter 9", "105-starter 1", "105-starter 10", 
"105-starter 11", "105-starter 12", "105-starter 13", "105-starter 14", 
"105-starter 15", "105-starter 16", "105-starter 2", "105-starter 3", 
"105-starter 4", "105-starter 5", "105-starter 6", "105-starter 7", 
"105-starter 8", "105-starter 9", "115-starter 1", "115-starter 2", 
"115-starter 3", "115-starter 4", "115-starter 5", "115-starter 6", 
"124-starter 1", "124-starter 2", "124-starter 3", "124-starter 4", 
"124-starter 5", "124-starter 6", "124-starter 7", "150-starter 1", 
"150-starter 10", "150-starter 2", "150-starter 3", "150-starter 4", 
"150-starter 5", "150-starter 6", "150-starter 7", "150-starter 8", 
"150-starter 9", "190-starter 1", "190-starter 2", "190-starter 3", 
"190-starter 4", "190-starter 5", "190-starter 6", "190-starter 7", 
"202-starter 1", "202-starter 2", "202-starter 3", "202-starter 4", 
"202-starter 5", "202-starter 6", "202-starter 9", "203-starter 1", 
"203-starter 2", "203-starter 3", "203-starter 4", "203-starter 5", 
"203-starter 6", "203-starter 7", "203-starter 8", "280-starter 1", 
"280-starter 2", "280-starter 3", "280-starter 4", "280-starter 5", 
"280-starter 6", "280-starter 7", "302-starter 1", "302-starter 2", 
"302-starter 3", "302-starter 4", "302-starter 5", "302-starter 6", 
"305-starter 1", "305-starter 2", "305-starter 3", "305-starter 4", 
"305-starter 5", "305-starter 6", "305-starter 7", "305-starter 8")  %>% gather(., sample, Snps, "101-starter 1":"305-starter 4", factor_key=TRUE,na.rm = TRUE) %>% filter(Snps>0.05)

ggplot(sample_relative_wide_explained_long,aes(x=explained,y=Snps))+geom_boxplot()+theme_classic()

ggplot(sample_relative_wide_explained_long,aes(x=Snps,group=explained,fill=explained))+geom_histogram(alpha=0.2, position="identity",bins = 100)+theme_classic()
table(sample_relative_wide_explained_long$sample)

##----------------------------------
###--------------------lineplot
##----------------------------------
#Sterm

  
#Ldel

sample_relative_wide_explained_long$culture <- str_split_fixed(sample_relative_wide_explained_long$sample, "_", 3)[,1]

sample_relative_wide_explained_long_ldel <- sample_relative_wide_explained_long %>% filter(species.x!="S. thermophilus")

all_colours <- rev(c("grey30","#10B552") )


 p3 <- ggplot(sample_relative_wide_explained_long_ldel,aes(x=sample,y=Snps,group=site,color=explained,fill=explained))+ geom_line(size=0.3, alpha=.1)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = "free_x", space = "free")+
            facet_grid(explained~culture,scales = "free_x", space = "free")+
            # ggforce::facet_row(vars(culture)~species,scales = "free_x", space = "free")+
            # ggforce::facet_col(vars(species),scales = "free_x", space = "free")+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
          panel.spacing = unit(1, "lines")
  )
  
  p3

  
###merge plots

  library(patchwork)  
p3 + labs(subtitle="A")+ p2+ labs(subtitle="B")+plot_layout(ncol=1)

 # png("~/Desktop/mid_thesis/report/figures/20200430/chapter3/explainedSNVs.png", width = 6000, height = 3500,res=300)
svg("~/Desktop/mid_thesis/report/figures/20200430/chapter2/snpscalled_sterm_wide_01.svg", width = 12, height = 5)

 p3 + labs(subtitle="A")+ p2+ labs(subtitle="B") +plot_layout(nrow =2)
dev.off()


svg("~/Der/ordner/wo/das/file/abgespeichert_werden_soll.svg", width = 12, height = 5) #mit de width und height mues biz spiele bises passt
 dinPlot
dev.off()


##----------------------------------
###--------------------of only the unexplained SNVs
##----------------------------------
sample_relative_wide_explained_long$culture <- str_split_fixed(sample_relative_wide_explained_long$sample, "-", 3)[,1]

sample_relative_wide_explained_long_notExplained <- sample_relative_wide_explained_long %>% filter(explained=="not explained")
all_colours <- rev(c("#EB4D4D","#10B552") )
sample_relative_wide_explained_long_notExplained$site <- as.factor(sample_relative_wide_explained_long_notExplained$site)
p4 <- ggplot(sample_relative_wide_explained_long_notExplained,aes(x=sample,y=Snps,group=site,color=species.x,fill=species.x))+ geom_line(size=0.3, alpha=.1)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = "free_x", space = "free")+
            facet_grid(species.x~culture,scales = "free_x", space = "free")+
            # ggforce::facet_row(vars(culture)~species,scales = "free_x", space = "free")+
            # ggforce::facet_col(vars(species),scales = "free_x", space = "free")+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
          panel.spacing = unit(1.2, "lines")
  )
  
  p4
  
  # png("~/Desktop/mid_thesis/report/figures/20200430/chapter3/explainedSNVs_onlyUnexplained.png", width = 4500, height = 2000,res=300)
svg("~/Desktop/mid_thesis/report/figures/20200430/chapter2/snpscalled_sterm_wide_unexplained.svg", width = 12, height = 6)

p4
dev.off()


```

## tsne snp

maybe I have to polish the snps first. I think without cleaning away super low abudnant and noisy snps it does not detect the underlining pattern

```{r}

library(forcats) 
library(Rtsne)
library(fpc)
library(ggrepel)
library( ggforce)
###--------------------------------------------------------------------
##colors
###--------------------------------------------------------------------

colors <- c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#80e5ff","#008181","#00dcdc")

###--------------------------------------------------------------------
##prep for tSNE
###--------------------------------------------------------------------
table(snps_long_final$core)
# snps_long_final_forTSNE <- snps_long_final 
snps_long_final_forTSNE <- snps_long_final  %>% filter(core=="core")
table(snps_long_final$core,snps_long_final$species)
snps_long_final_forTSNE$sampleName <- paste0(snps_long_final_forTSNE$culture,"_",snps_long_final_forTSNE$sample)
colnames(snps_long_final_forTSNE)
dim(snps_long_final_forTSNE)
snps_long_final_forTSNE <-  snps_long_final_forTSNE %>% select(species,site,sampleName,Snps)%>% filter(.,Snps>0.1) 
dim(snps_long_final_forTSNE)
table(snps_long_final_forTSNE$species)

###go wide
snps_long_final_forTSNE_all <- snps_long_final_forTSNE
snps_long_final_forTSNE <- snps_long_final_forTSNE %>% select(-species)
# snps_long_final_forTSNE <- snps_long_final_forTSNE %>% filter(species.x=="L. delbrueckii") %>% select(-species.x)

mst1_metagenomicMutations_tsne <- spread(snps_long_final_forTSNE, site, Snps)%>%replace(is.na(.), 0) %>% column_to_rownames(var = "sampleName")
dim(mst1_metagenomicMutations_tsne)



fortsne <- mst1_metagenomicMutations_tsne[!duplicated(mst1_metagenomicMutations_tsne), ]

# fortsne[duplicated(fortsne), ] %>% nrow()
dim(fortsne)
set.seed(1234567)
# set.seed(12314567)
set.seed(007)

tsne <- Rtsne(fortsne, dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- rownames(fortsne)
tsne_plot$sample <- str_split_fixed(tsne_plot$site, "_", 2)[,1]
tsne_plot$number <- str_split_fixed(tsne_plot$site, "_", 2)[,2]
tsne_plot$cheese <- revalue(tsne_plot$sample,c("101"="emmentaler","105"="emmentaler","115"="emmentaler","124"="emmentaler","150"="emmentaler","190"="emmentaler","202"="gruyere","203"="gruyere","280"="gruyere","302"="sbrinz","305"="sbrinz"))

# colors <- c("#aa0000","#d40000","#ff0000","#ff2a2a","#ff5555","#ff8080","#009cf0","#34d0fc","#b4ecfc","#008181","#00dcdc")
colors <- c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#b4ecfc","#008181","#00dcdc","grey")

tsne_plotss <- ggplot(tsne_plot, aes(x = V1, y = V2,color=sample,fill=sample,shape=cheese)) +labs(title=paste0("all SNV (n=",ncol(fortsne),")"))+
    geom_mark_ellipse(aes(fill = sample),alpha=0.1) +geom_point(alpha = 0.8,size=5) + theme_classic() +theme(legend.position = "none")+scale_color_manual(values = colors)+scale_fill_manual(values = colors)+geom_text(aes(label=site),hjust=0, vjust=0)

tsne_plotss

###------------------------------
##only Ldel
###------------------------------

snps_long_final_forTSNE_ldel <- snps_long_final_forTSNE_all %>% filter(species.x=="L. delbrueckii") %>% select(-species.x)

mst1_metagenomicMutations_tsne_ldel <- spread(snps_long_final_forTSNE_ldel, site, Snps)%>%replace(is.na(.), 0) %>% column_to_rownames(var = "sampleName")
dim(mst1_metagenomicMutations_tsne_ldel)



fortsne <- mst1_metagenomicMutations_tsne_ldel[!duplicated(mst1_metagenomicMutations_tsne_ldel), ]

# fortsne[duplicated(fortsne), ] %>% nrow()
dim(fortsne)
set.seed(1234567)
# set.seed(12314567)
set.seed(007)

tsne_ldel <- Rtsne(fortsne, dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne_ldel$Y)
tsne_plot$site <- rownames(fortsne)
tsne_plot$sample <- str_split_fixed(tsne_plot$site, "_", 2)[,1]
tsne_plot$number <- str_split_fixed(tsne_plot$site, "_", 2)[,2]
tsne_plot$cheese <- revalue(tsne_plot$sample,c("101"="emmentaler","105"="emmentaler","115"="emmentaler","124"="emmentaler","150"="emmentaler","190"="emmentaler","202"="gruyere","203"="gruyere","280"="gruyere","302"="sbrinz","305"="sbrinz"))

# colors <- c("#aa0000","#d40000","#ff0000","#ff2a2a","#ff5555","#ff8080","#009cf0","#34d0fc","#b4ecfc","#008181","#00dcdc")
# colors <- c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#b4ecfc","#008181","#00dcdc")

tsne_plotss_ldel <- ggplot(tsne_plot, aes(x = V1, y = V2,color=sample,fill=sample,shape=cheese)) +labs(title=paste0("L. delbrueckii SNV (n=",ncol(fortsne),")"))+
    geom_mark_ellipse(aes(fill = sample),alpha=0.1) +geom_point(alpha = 0.8,size=5) + theme_classic() +scale_color_manual(values = colors)+scale_fill_manual(values = colors)+geom_text(aes(label=site),hjust=0, vjust=0)

tsne_plotss_ldel


###------------------------------
##only Sterm
###------------------------------

snps_long_final_forTSNE_sterm <- snps_long_final_forTSNE_all %>% filter(species.x=="S. thermophilus") %>% select(-species.x)

mst1_metagenomicMutations_tsne_sterm <- spread(snps_long_final_forTSNE_sterm, site, Snps)%>%replace(is.na(.), 0) %>% column_to_rownames(var = "sampleName")
dim(mst1_metagenomicMutations_tsne_sterm)



fortsne <- mst1_metagenomicMutations_tsne_sterm[!duplicated(mst1_metagenomicMutations_tsne_sterm), ]

# fortsne[duplicated(fortsne), ] %>% nrow()
dim(fortsne)
set.seed(1234567)
# set.seed(12314567)
set.seed(007)

tsne_sterm <- Rtsne(fortsne, dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne_sterm$Y)
tsne_plot$site <- rownames(fortsne)
tsne_plot$sample <- str_split_fixed(tsne_plot$site, "_", 2)[,1]
tsne_plot$number <- str_split_fixed(tsne_plot$site, "_", 2)[,2]
tsne_plot$cheese <- revalue(tsne_plot$sample,c("101"="emmentaler","105"="emmentaler","115"="emmentaler","124"="emmentaler","150"="emmentaler","190"="emmentaler","202"="gruyere","203"="gruyere","280"="gruyere","302"="sbrinz","305"="sbrinz"))

# colors <- c("#aa0000","#d40000","#ff0000","#ff2a2a","#ff5555","#ff8080","#009cf0","#34d0fc","#b4ecfc","#008181","#00dcdc")
# colors <- c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#80e5ff","#008181","#00dcdc")

tsne_plotss_sterm <- ggplot(tsne_plot, aes(x = V1, y = V2,color=sample,fill=sample,shape=cheese)) +labs(title=paste0("S. thermophilus SNV (n=",ncol(fortsne),")"))+
    geom_mark_ellipse(aes(fill = sample),alpha=0.1) +geom_point(alpha = 0.8,size=5) + theme_classic() +theme(legend.position = "none")+scale_color_manual(values = colors)+scale_fill_manual(values = colors)+geom_text(aes(label=site),hjust=0, vjust=0)

tsne_plotss_sterm

###------------------------------
##all togehter
###------------------------------

library(patchwork)

tsne_plotss+tsne_plotss_sterm+tsne_plotss_ldel


svg("~/Desktop/mid_thesis/report/figures/20200430/chapter2/Tsne_diversity",width=18,height=8)
# png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_sterm.png", width = 1800, height = 4800,res=300)
tsne_plotss+tsne_plotss_sterm+tsne_plotss_ldel
dev.off()
```


#plot snps

###### Sterm
```{r}
library(readr)
dim(snps_wide_final)
# snps_long_final <- gather(sample_relative_wide, sample, Snps, "starter 1":"starter 12", factor_key=TRUE,na.rm = TRUE) 
snps_long_final <-  sample_relative_wide %>%  select(site,species.x,effect,significance,geneName,COG_Functional_Category,Repeat_cluster,core, "101-starter 1", 
"101-starter 10", "101-starter 11", "101-starter 2", "101-starter 3", 
"101-starter 4", "101-starter 5", "101-starter 6", "101-starter 7", 
"101-starter 8", "101-starter 9", "105-starter 1", "105-starter 10", 
"105-starter 11", "105-starter 12", "105-starter 13", "105-starter 14", 
"105-starter 15", "105-starter 16", "105-starter 2", "105-starter 3", 
"105-starter 4", "105-starter 5", "105-starter 6", "105-starter 7", 
"105-starter 8", "105-starter 9", "115-starter 1", "115-starter 2", 
"115-starter 3", "115-starter 4", "115-starter 5", "115-starter 6", 
"124-starter 1", "124-starter 2", "124-starter 3", "124-starter 4", 
"124-starter 5", "124-starter 6", "124-starter 7", "150-starter 1", 
"150-starter 10", "150-starter 2", "150-starter 3", "150-starter 4", 
"150-starter 5", "150-starter 6", "150-starter 7", "150-starter 8", 
"150-starter 9", "190-starter 1", "190-starter 2", "190-starter 3", 
"190-starter 4", "190-starter 5", "190-starter 6", "190-starter 7", 
"202-starter 1", "202-starter 2", "202-starter 3", "202-starter 4", 
"202-starter 5", "202-starter 6", "202-starter 9", "203-starter 1", 
"203-starter 2", "203-starter 3", "203-starter 4", "203-starter 5", 
"203-starter 6", "203-starter 7", "203-starter 8", "280-starter 1", 
"280-starter 2", "280-starter 3", "280-starter 4", "280-starter 5", 
"280-starter 6", "280-starter 7", "302-starter 1", "302-starter 2", 
"302-starter 3", "302-starter 4", "302-starter 5", "302-starter 6", 
"305-starter 1", "305-starter 2", "305-starter 3", "305-starter 4", 
"305-starter 5", "305-starter 6", "305-starter 7", "305-starter 8") %>% gather(., sample, Snps, "101-starter 1":"305-starter 8", factor_key=TRUE,na.rm = TRUE)

##----------------
##analysis
##----------------

table(snps_long_final$core)
nrow(snps_long_final)
sample_relative_safe_woSTRAINS_sub01 <- snps_long_final
sample_relative_safe_woSTRAINS_sub01$culture <- str_split_fixed(sample_relative_safe_woSTRAINS_sub01$sample, "-", 3)[,1]
sample_relative_safe_woSTRAINS_sub01$species.x <- sample_relative_safe_woSTRAINS_sub01$species

table(sample_relative_safe_woSTRAINS_sub01$species)
table(sample_relative_safe_woSTRAINS_sub01$culture)
table(sample_relative_safe_woSTRAINS_sub01$sample)

table(sample_relative_safe_woSTRAINS_sub01$culture,sample_relative_safe_woSTRAINS_sub01$sample)


# sample_relative_safe_woSTRAINS_sub01 <- sample_relative_safe_woSTRAINS[which(sample_relative_safe_woSTRAINS$significance!="MODIFIER"),]
nrow(sample_relative_safe_woSTRAINS_sub01)
sample_relative_safe_woSTRAINS_sub02 <- sample_relative_safe_woSTRAINS_sub01
sample_relative_safe_woSTRAINS_sub02 <- sample_relative_safe_woSTRAINS_sub01[which(sample_relative_safe_woSTRAINS_sub01$core=="core"),]
nrow(sample_relative_safe_woSTRAINS_sub02)

sample_relative_safe_woSTRAINS_sub03 <- sample_relative_safe_woSTRAINS_sub02[which(sample_relative_safe_woSTRAINS_sub02$Snps>0.03),]
# sample_relative_safe_woSTRAINS_sub05 <- sample_relative_safe_woSTRAINS_sub03
sample_relative_safe_woSTRAINS_sub04 <- sample_relative_safe_woSTRAINS_sub02[which(sample_relative_safe_woSTRAINS_sub02$Snps<=0.03),]
sample_relative_safe_woSTRAINS_sub04$Snps <- 0
nrow(sample_relative_safe_woSTRAINS_sub04)
sample_relative_safe_woSTRAINS_sub05 <- rbind(sample_relative_safe_woSTRAINS_sub03,sample_relative_safe_woSTRAINS_sub04)
sample_relative_safe_woSTRAINS_sub06 <- sample_relative_safe_woSTRAINS_sub05[which(sample_relative_safe_woSTRAINS_sub05$significance %in% c("MODERATE","HIGH")),]
nrow(sample_relative_safe_woSTRAINS_sub05)
table(sample_relative_safe_woSTRAINS_sub05$species)
# sample_relative_safe_Sterm_final <- sample_relative_safe_woSTRAINS_sub06[which(sample_relative_safe_woSTRAINS_sub06$species %in %"S. thermophilus"),]

sample_relative_safe_Sterm_final <- sample_relative_safe_woSTRAINS_sub06[which(sample_relative_safe_woSTRAINS_sub06$species %in% c("L. delbrueckii","S. thermophilus")),]
sample_relative_safe_Sterm_final_wide <- spread(sample_relative_safe_Sterm_final, sample, Snps)  %>%replace(is.na(.), 0)
sample_relative_safe_Sterm_final_wide
table(sample_relative_safe_Sterm_final$sampleComplete)
sample_relative_safe_Sterm_final_new <- sample_relative_safe_Sterm_final %>% select(c(-"species.x",-"effect",-"significance",-"geneName",-"COG_Functional_Category",-"Repeat_cluster",-"core"))

sample_relative_safe_Sterm_final_new$sampleComplete <- sample_relative_safe_Sterm_final_new$sample
sample_relative_safe_Sterm_final_new <- sample_relative_safe_Sterm_final_new %>% select(c(-"culture",-"sample"))
sample_relative_safe_Sterm_final_new <- sample_relative_safe_Sterm_final_new %>% select(c(-"sample"))

sample_relative_safe_Sterm_final_wide <- spread(sample_relative_safe_Sterm_final_new, sampleComplete, Snps)  %>%replace(is.na(.), 0)
# sample_relative_safe_Sterm_final_wide[] %>% select()
colnames(sample_relative_safe_Sterm_final_wide)
dim(sample_relative_safe_Sterm_final_wide)

sample_relative_safe_Sterm_final_wide_filtered <- sample_relative_safe_Sterm_final_wide[,c(1,which(colSums(sample_relative_safe_Sterm_final_wide[,-1])>0)+1)]
dim(sample_relative_safe_Sterm_final_wide_filtered)
colnames(sample_relative_safe_Sterm_final_wide_filtered)

sample_relative_safe_Sterm_final_wide_filtered_long <-  gather(sample_relative_safe_Sterm_final_wide_filtered, sampleComplete, Snps, "101-starter 1":"305-starter 8", factor_key=TRUE,na.rm = TRUE) 

sample_relative_safe_Sterm_final_wide_filtered_long$culture <- str_split_fixed(sample_relative_safe_Sterm_final_wide_filtered_long$sampleComplete, "-", 2)[,1]

sample_relative_safe_Sterm_final_wide_filtered_long$sample <- str_split_fixed(sample_relative_safe_Sterm_final_wide_filtered_long$sampleComplete, "-", 2)[,2]
sample_relative_safe_Sterm_final_wide_filtered_long$species <- str_split_fixed(sample_relative_safe_Sterm_final_wide_filtered_long$site, "_", 4)[,1]
table(sample_relative_safe_Sterm_final_wide_filtered_long$sample)
  # sample_relative_safe_Sterm_final_longAll <-
sample_relative_safe_woSTRAINS <- sample_relative_safe_Sterm_final_wide_filtered_long
sample_relative_safe_Sterm_final_wide_filtered_long$sample = factor(sample_relative_safe_Sterm_final_wide_filtered_long$sample, levels=c("starter 1","starter 2","starter 3","starter 4","starter 5","starter 6","starter 7","starter 8","starter 9","starter 10","starter 11","starter 12","starter 13","starter 14","starter 15","starter 16"))
table(sample_relative_safe_Sterm_final_wide_filtered_long$culture,sample_relative_safe_Sterm_final_wide_filtered_long$sample)
  ##============
### plot Streptococcus thermophius
   ##============

sample_relative_safe_woSTRAINS$sample <- droplevels(sample_relative_safe_woSTRAINS$sample)
levels(sample_relative_safe_Sterm_final_wide_filtered_long$sample)
sample_relative_safe_Sterm_final_wide_filtered_long_woZero <- sample_relative_safe_Sterm_final_wide_filtered_long %>% filter(Snps>0)
  ##============
### plot 
   ##============

all_colours <- rev(c("#EB4D4D","#10B552") )
all_colours <- rev(c("#EB4D4D","#10B552") )

# all_colours <- rev(c("#EB4D4D") ) 

# sample_relative_safe_Sterm_final <- sample_relative_safe_woSTRAINS_sub06[which(sample_relative_safe_woSTRAINS_sub06$species %in% c("L. delbrueckii","S. thermophilus")),]
sample_relative_safe_Sterm_final_wide_filtered_long_woZero$species <- revalue(sample_relative_safe_Sterm_final_wide_filtered_long_woZero$species,c("CP046131"="L. delbrueckii","CP046134"="S. thermophilus"))
# 
# table(sample_relative_safe_Sterm_final_wide_filtered_long_woZero$species)

# colnames(sample_relative_safe_woSTRAINS)
table(sample_relative_safe_Sterm_final_wide_filtered_long_woZero$culture,sample_relative_safe_Sterm_final_wide_filtered_long_woZero$sample)
##plot
  p2 <- ggplot(sample_relative_safe_Sterm_final_wide_filtered_long_woZero,aes(x=sample,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.3, alpha=.1)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = "free_x", space = "free")+
            facet_grid(species~culture,scales = "free_x", space = "free")+
            # ggforce::facet_row(vars(culture)~species,scales = "free_x", space = "free")+
            # ggforce::facet_col(vars(species),scales = "free_x", space = "free")+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
          panel.spacing = unit(1.2, "lines")
  )
  
  p2
  

# svg("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.svg",width=2.5,height=2.5)
# png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_sterm.png", width = 1800, height = 4800,res=300)
# png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_sterm_wide.png", width = 6000, height = 1800,res=300)
# png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_sterm_wide.png", width = 6000, height = 1800,res=300)
svg("~/Desktop/mid_thesis/report/figures/20200430/chapter2/snpscalled_sterm_wide.svg", width = 12, height = 6)


p2
dev.off()

```

####all samples together
Here, I put all cultures together and see if I can phase them with tSNE

```{r}

# sample_relative_safe_Sterm_final$sampleNew <- paste0(sample_relative_safe_Sterm_final$culture,"_",sample_relative_safe_Sterm_final$sample)
# table(sample_relative_safe_Sterm_final$sampleNew)
# colnames(sample_relative_safe_Sterm_final_wide)
sample_relative_safe_Sterm_final_longAll <- gather(sample_relative_safe_Sterm_final_wide, sample, Snps, "starter 1":"starter 12", factor_key=TRUE,na.rm = TRUE) 

sample_relative_safe_Sterm_final_longAll$sampleNew <- paste0(sample_relative_safe_Sterm_final_longAll$culture,"_",sample_relative_safe_Sterm_final_longAll$sample)
dim(sample_relative_safe_Sterm_final_longAll)
# snps_freebayes_long <- gather(snps_freebayes, sample, Snps, colnames(snps_freebayes)[6]:colnames(snps_freebayes)[ncol(snps_freebayes)], factor_key=TRUE,na.rm = TRUE) 

sample_relative_safe_Sterm_final_longAll_filtered <- sample_relative_safe_Sterm_final_longAll %>% filter(sampleNew %in% c("101_starter 1","101_starter 2","101_starter 3","101_starter 4","101_starter 5","101_starter 6","101_starter 7","105_starter 1","105_starter 2","105_starter 3","105_starter 4","105_starter 5","105_starter 6","105_starter 7","105_starter 8","105_starter 9","105_starter 10","105_starter 11","105_starter 12","115_starter 1","115_starter 2","124_starter 1","124_starter 2", "124_starter 3","150_starter 1","150_starter 2","150_starter 3","150_starter 4","150_starter 5","150_starter 6","190_starter 1","190_starter 2","190_starter 3","202_starter 1","202_starter 2","202_starter 3","202_starter 4","202_starter 5","202_starter 6", "202_starter 7","202_starter 8","203_starter 1","203_starter 2","203_starter 3","203_starter 4","280_starter 1","280_starter 2", "280_starter 3","302_starter 1", "302_starter 2","305_starter 1", "305_starter 2","305_starter 3","305_starter 4"))

sample_relative_safe_Sterm_final_longAll_filtered_prep <- sample_relative_safe_Sterm_final_longAll_filtered %>% select(-sample)

sample_relative_safe_Sterm_final_wide_new <- spread(sample_relative_safe_Sterm_final_longAll_filtered_prep, sampleNew, Snps)  %>%replace(is.na(.), 0)
colnames(sample_relative_safe_Sterm_final_wide_new)
sample_relative_safe_Sterm_final_longAll_final <- gather(sample_relative_safe_Sterm_final_wide_new, sampleNew, Snps, "101_starter 1":"305_starter 4", factor_key=TRUE,na.rm = TRUE) 

# 
# snps_long$samplesNew <- revalue(snps_long$sample,c("101_starter 1","101_starter 2","101_starter 3","101_starter 4","101_starter 5","101_starter 6","101_starter 7","105_starter 1","105_starter 2","105_starter 3","105_starter 4","105_starter 5","105_starter 6","105_starter 7","105_starter 8","105_starter 9","105_starter 10","105_starter 11","105_starter 12","115_starter 1","115_starter 2","124_starter 1","124_starter 2", "124_starter 3","150_starter 1","150_starter 2","150_starter 3","150_starter 4","150_starter 5","150_starter 6","190_starter 1","190_starter 2","190_starter 3","202_starter 1","202_starter 2","202_starter 3","202_starter 4","202_starter 5","202_starter 6", "202_starter 7","202_starter 8","203_starter 1","203_starter 2","203_starter 3","203_starter 4","280_starter 1","280_starter 2", "280_starter 3","302_starter 1", "302_starter 2","305_starter 1", "305_starter 2","305_starter 3","305_starter 4"))
# 



p3 <- ggplot(sample_relative_safe_Sterm_final_longAll_final,aes(x=sampleNew,y=Snps,group=site,color=species.x,fill=species.x))+ geom_line(size=0.2, alpha=.1)+ 
    # facet_wrap(culture~.,nrow = 11)+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none"
  )
  
  # p3
  

# svg("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.svg",width=2.5,height=2.5)
png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_sterm_wide.png", width = 4800, height = 1800,res=300)
p3
dev.off()  

```

####tSNE

```{r}

library(forcats) 
library(Rtsne)
library(fpc)
library(ggrepel)
###--------------------------------------------------------------------
##colors
###--------------------------------------------------------------------

# colorsTSNE <- c("#DD8D29","#E2D200","#46ACC8","black","#B40F20")

# colorsTSNE <- c("#E69F00", "#56B4E9", "#009E73",
          # "#F0E442", "#0072B2", "#D55E00", "#CC79A7","grey","#000000")

# colorsTSNE <- rev(c("#FF9E79","#FB6D4C","#C23B22","#8A0000","#580000","#D55E00"))
###--------------------------------------------------------------------
##subset
###--------------------------------------------------------------------

# mst1_metagenomicMutations <- sample_relative_wide %>% filter(.,species=="S. thermophilus") %>% filter(.,significance %in% c("MODERATE","HIGH")) %>% filter(.,mst1>0.9 & mst2 <=0.9) %>% add_column(group="mst1") %>%replace(is.na(.), 0)
# mst1_metagenomicMutations <- sample_relative_safe_Sterm_final_wide %>% filter(.,mst1>0.9 & mst2 <=0.9)%>% add_column(group="mst1") 


# mst1_metagenomicMutations_long <-  mst1_metagenomicMutations %>% gather(., sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2", factor_key=TRUE,na.rm = TRUE)# %>% filter(.,Snps>0.005)

# mst1_metagenomicMutations_tsne <- sample_relative_safe_Sterm_final_wide_new


###--------------------------------------------------------------------
##prep for tSNE
###--------------------------------------------------------------------



fortsne <- sample_relative_safe_Sterm_final_wide_new %>% select(.,"site","101_starter 1","101_starter 2","101_starter 3","101_starter 4","101_starter 5","101_starter 6","101_starter 7","105_starter 1","105_starter 2","105_starter 3","105_starter 4","105_starter 5","105_starter 6","105_starter 7","105_starter 8","105_starter 9","105_starter 10","105_starter 11","105_starter 12","115_starter 1","115_starter 2","124_starter 1","124_starter 2", "124_starter 3","150_starter 1","150_starter 2","150_starter 3","150_starter 4","150_starter 5","150_starter 6","190_starter 1","190_starter 2","190_starter 3","202_starter 1","202_starter 2","202_starter 3","202_starter 4","202_starter 5","202_starter 6", "202_starter 7","202_starter 8","203_starter 1","203_starter 2","203_starter 3","203_starter 4","280_starter 1","280_starter 2", "280_starter 3","302_starter 1", "302_starter 2","305_starter 1", "305_starter 2","305_starter 3","305_starter 4")
fortsne <- fortsne[!duplicated(fortsne[,-1]), ]
# fortsne[duplicated(fortsne), ] %>% nrow()

set.seed(1234567)
# set.seed(12314567)

tsne <- Rtsne(fortsne[,-1], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- fortsne[,1]
tsne_plot$species <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_") %>%  gsub("_",". ",.)



##------------plot blank
# ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()

##------------clusters

# set.seed(123456)

set.seed(1234567)
ds.real = dbscan(tsne_plot[,c(1,2)], 15)
# decreasing_clusternames <- (names(sort(table(ds.real$cluster),decreasing = TRUE)))
# change_name <- setNames(1:length(decreasing_clusternames),decreasing_clusternames)
# clusterNames <- revalue(as.factor(ds.real$cluster),change_name)
# clusterNames<- revalue(clusterNames,c("4"="1","6"="1","7"="1","10"="1","11"="1","5"="4","8"="5","9"="6"))##add certain clusters to others
tsne_plot$density = factor(ds.real$cluster)
# levels(tsne_plot$density) <- c(levels(tsne_plot$density), "5")

# special_cases <- c("CP046134_494890_A_G","CP046134_482906_A_C")
# tsne_plot[which(tsne_plot$site %in% special_cases),"density"] <- "5"
ds.cent = tsne_plot %>% group_by(density) %>% select(V1,
    V2) %>% summarize_all(mean)
ggplot(tsne_plot, aes(x = V1, y = V2, colour = density)) +
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density),
    data = ds.cent) + guides(colour = FALSE)


```



###### Ldel
```{r}




sample_relative_safe_Sterm_final <- sample_relative_safe_woSTRAINS_sub06[which(sample_relative_safe_woSTRAINS_sub06$species=="L. delbrueckii"),]
sample_relative_safe_Sterm_final_wide <- spread(sample_relative_safe_Sterm_final, sample, Snps)  %>%replace(is.na(.), 0)
dim(sample_relative_safe_Sterm_final_wide)

  ##============
### plot Streptococcus thermophius
   ##============

sample_relative_safe_woSTRAINS$sample <- droplevels(sample_relative_safe_woSTRAINS$sample)
levels(sample_relative_safe_Sterm_final$sample)

  ##============
### plot 
   ##============

# all_colours <- rev(c("#EB4D4D","#10B552") ) 
all_colours <- rev(c("#EB4D4D") ) 

# colnames(sample_relative_safe_woSTRAINS)

##plot
  p2 <- ggplot(sample_relative_safe_Sterm_final,aes(x=sample,y=Snps,group=site,color=species.x,fill=species.x))+ geom_line(size=0.2, alpha=.1)+ 
    facet_wrap(culture~.,nrow = 11)+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none"
  )
  
  p2
  

# svg("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.svg",width=2.5,height=2.5)
png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_sterm.png", width = 1800, height = 4800,res=300)
p2
dev.off()


##------------------------------------------------------
##old
##------------------------------------------------------





library(readr)
snps_long_final <- read_csv("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all.csv")

##----------------
##01_remove strains
##----------------
# sample_relative_safe_woSTRAINS <- snps_long_final[which(snps_long_final$species %in% c("L. delbrueckii","S. thermophilus")),] 
sample_relative_safe_woSTRAINS <- snps_long_final[which(snps_long_final$species %in% c("L. delbrueckii")),] 


sample_relative_safe_woSTRAINS$samplesNew = factor(sample_relative_safe_woSTRAINS$samplesNew, levels=c("starter 1","starter 2","starter 3","starter 4","starter 5","starter 6","starter 7","starter 8","starter 9","starter 10","starter 11","starter 12"))
sample_relative_safe_woSTRAINS$samplesNew <- droplevels(sample_relative_safe_woSTRAINS$samplesNew)

  ##============
### plot 
   ##============

# all_colours <- rev(c("#EB4D4D","#10B552") ) 
all_colours <- rev(c("#10B552") ) 

# colnames(sample_relative_safe_woSTRAINS)

##plot
  p3 <- ggplot(sample_relative_safe_woSTRAINS,aes(x=samplesNew,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.2, alpha=.1)+ 
    facet_wrap(culture~.,nrow = 11)+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none"
  )
  
  p3
  

# svg("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.svg",width=2.5,height=2.5)
png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_ldel.png", width = 1800, height = 4800,res=300)
p3
dev.off()
library(patchwork)
png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled.png", width = 3000, height = 4800,res=300)
p2+p3
dev.off()

```


#nucleotide diversity
```{r}
library(readr)
library(plyr)
library(tidyverse)

# write.csv(snps_long_final,"~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all.csv",row.names = FALSE)

# snps_long_final_02 <- read_csv("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all.csv")
# snps_long_final <- read_csv("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all.csv", col_types = cols(culture = col_character()))
# ~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all.csv
# table(snps_long_final$sample)
# table(snps_long_final$culture)

sample_relative_safe_woSTRAINS <- snps_long_final %>% filter(species %in% c("L. delbrueckii","S. thermophilus")) %>%  filter(Snps>0.05) %>% filter(culture!="FAMstrains")
table(sample_relative_safe_woSTRAINS$culture)
snp_count <- as.data.frame(table(sample_relative_safe_woSTRAINS$culture,sample_relative_safe_woSTRAINS$species,sample_relative_safe_woSTRAINS$sample)) %>% dplyr::rename(culture=Var1,species=Var2,sample=Var3)

snp_count <- snp_count %>% filter(Freq>0)
snp_count$sample = factor(snp_count$sample, levels=c("starter 1","starter 2","starter 3","starter 4","starter 5","starter 6","starter 7","starter 8","starter 9","starter 10","starter 11","starter 12"))
snp_count$sample <- droplevels(snp_count$sample)

snp_count_ldel <- snp_count %>% filter(species=="L. delbrueckii")

   pldel <- ggplot()+ #stat_summary(fun.y=sum, geom="line")+
     geom_point(data = snp_count_ldel,aes(y = Freq, x = sample),fill="#10B552",color="#10B552")+
     geom_line(data = snp_count_ldel,aes(y = Freq, x = sample,group=species),fill="#10B552",color="#10B552")+
     facet_wrap(culture~.,nrow = 11)+
   labs("",
         x="",
         y="SNPs",
          title="")+
  theme_classic()+
     scale_x_discrete( expand = c(0, 0)) +
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"))
  pldel

  
  ##------sterm
  
  snp_count_sterm <- snp_count %>% filter(species=="S. thermophilus")

   psterm <- ggplot()+ #stat_summary(fun.y=sum, geom="line")+
     geom_point(data = snp_count_sterm,aes(y = Freq, x = sample),fill="#EB4D4D",color="#EB4D4D")+
     geom_line(data = snp_count_sterm,aes(y = Freq, x = sample,group=species),fill="#EB4D4D",color="#EB4D4D")+
     facet_wrap(culture~.,nrow = 11)+
   labs("",
         x="",
         y="SNPs",
          title="")+
  theme_classic()+
     scale_x_discrete( expand = c(0, 0)) +
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"))
  psterm
  
library(patchwork)
png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_nucleotide_diversity.png", width = 3000, height = 4800,res=300)
psterm+pldel
dev.off()


###----------------------------
##box plot
###----------------------------

snp_count_ldel_extended <- snp_count_ldel%>% mutate(nucleotideDiversity=Freq/22000)
snp_count_sterm_extended <- snp_count_sterm%>% mutate(nucleotideDiversity=Freq/18000)
all_nucleotideDiv <- rbind(snp_count_ldel_extended,snp_count_sterm_extended)
# all_colours <- rev(c("#EB4D4D","#10B552") )
colors_rmks <-  c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#80e5ff","#008181","#00dcdc")

boxplot <- ggplot(all_nucleotideDiv,aes(x=culture,y=nucleotideDiversity,group=culture,fill=culture))+geom_boxplot()+theme_classic()+facet_wrap(species~.,scales="free")+scale_fill_manual(values =colors_rmks )+
  labs(y="Nucleotide diversity [%]",
       x="")+
    theme(legend.position = "none",axis.title.x = element_blank(),axis.text.x = element_text(angle = 75, hjust = 1,size=9))

  # theme(legend.position = "none",axis.text.x = element_blank())
boxplot
png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_nucleotide_diversity_boxplot.png", width = 2000, height = 1000,res=300)
boxplot
dev.off()

###----------------------------
##violin plot
###----------------------------

sample_relative_safe_woSTRAINS_violin <- snps_long_final %>% filter(species %in% c("L. delbrueckii","S. thermophilus")) 

sample_relative_safe_woSTRAINS_violin$culture <- as.factor(sample_relative_safe_woSTRAINS_violin$culture)
colors_rmks <-  c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#80e5ff","#008181","#00dcdc")

violinPlota <- ggplot(sample_relative_safe_woSTRAINS_violin,aes(x=culture,y=Snps,group=culture,fill=culture))+labs(y="alternative allele frequency")+geom_violin()+theme_classic()+facet_wrap(species~.,scales="free")+scale_fill_manual(values=colors_rmks)+theme(legend.title = element_blank(),axis.title.x = element_blank(),axis.text.x = element_text(angle = 75, hjust = 1,size=9))
violinPlota
png("~/Desktop/mid_thesis/report/figures/20200430/chapter2//snpsCalled_nucleotide_diversity_boxplot_alternativeAlleleFreq.png", width = 3500, height = 2500,res=300)
# png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_nucleotide_diversity_boxplot_alternativeAlleleFreq.png", width = 1500, height = 1500,res=300)

boxplot + labs(subtitle="A")+ violinPlota+ labs(subtitle="B")+plot_layout(ncol=1)
dev.off()

boxplot / violinPlota
```



#single strain analysis

Here, I analyse the different reference genome sequences

## ONT

```{bash}

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x1_T43_Seq1_VS_fq
cp /home/vincent/Downloads/fastq_pass.tar.gz.zip /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x1_T43_Seq1_VS_fq
cd /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x1_T43_Seq1_VS_fq
openssl enc -aes-256-cbc -md md5 -d -in fastq_pass.tar.gz.enc > fastq_pass.tar.gz
tar xvzf fastq_pass.tar.gz

##------------------------2
rm  /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq
cp /home/vincent/Downloads/'fastq_pass.tar.gz(1).enc' /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq/
cd /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq/
openssl enc -aes-256-cbc -md md5 -d -in fastq_pass.tar.gz.enc > fastq_pass.tar.gz
tar xvzf fastq_pass.tar.gz

scp /home/vincent/Downloads/fastq_pass.tar.gz.enc vincent@130.223.49.145:/data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq/

wget https://campuscloud.unibe.ch/ssf/s/readFile/folderEntry/64630263/02dc805872bb70a00172d615fd105f8f/1592660049000/last/fastq_pass.tar.gz.enc
##------------------------3
rm  /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x3_ES43_EXP02_Seq3_VS_fq

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x3_ES43_EXP02_Seq3_VS_fq
cp /home/vincent/Downloads/fastq_pass.tar.gz.enc /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x3_ES43_EXP02_Seq3_VS_fq/
cd /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x3_ES43_EXP02_Seq3_VS_fq/
#unzip fastq_pass.tar.gz.zip
openssl enc -aes-256-cbc -md md5 -d -in fastq_pass.tar.gz.enc > fastq_pass.tar.gz
tar xvzf fastq_pass.tar.gz

##------------------------2

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x4_ES43_EXP02_Seq4_VS
cp /home/vincent/Downloads/fastq_pass.tar.gz.enc /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x4_ES43_EXP02_Seq4_VS/
cd /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x4_ES43_EXP02_Seq4_VS/
#unzip fastq_pass.tar.gz.zip
openssl enc -aes-256-cbc -md md5 -d -in fastq_pass.tar.gz.enc > fastq_pass.tar.gz
tar xvzf fastq_pass.tar.gz

```

```{bash}

scp /home/vincent/Desktop/Projects/2020_strainDelineation/ONT_sequencing/20200619_Seq_run_ONT_ifik.csv vincent@130.223.49.145:/data/Project/2020_StarterCultureDiversity/99_log/

```


####rename and merge
```{bash}

for runnsss in $(echo "2")
do

longRunssss=$(echo -e "_Seq"${runnsss}"_VS")
directory=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/ |grep ${longRunssss})
echo "============================="
echo ${longRunssss}
for samplezzz in $(grep "${longRunssss}" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |cut -f 3)
do

rm -r  /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${samplezzz}/
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${samplezzz}/
barcode=$(grep "${samplezzz}" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |cut -f 4|sed 's/RB //g')
echo "----------------------"
echo -e "barcode :" ${barcode} " & sample: " ${samplezzz}
cat /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/${directory}/fastq_pass/barcode${barcode}/*fastq > /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${samplezzz}/${samplezzz}.fastq

done #samplezzz

done #runsss



for runnsss in $(echo "2")
do

longRunssss=$(echo -e "_Seq"${runnsss}"_VS")
directory=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/ |grep ${longRunssss})
echo "============================="
echo ${longRunssss}
for samplezzz in $(grep "${longRunssss}" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |cut -f 3)
do

barcode=$(grep "${samplezzz}" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |cut -f 4|sed 's/RB //g')
echo "----------------------"
echo -e "barcode :" ${barcode} " & sample: " ${samplezzz}
grep "^@" -c /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${samplezzz}/${samplezzz}.fastq

done #samplezzz

done #runsss


```


##flye

```{bash}
sample=24740
##===========
##Assembly plasmid
##===========
for sample in $(grep "20200527_x2_T43EXP01_Seq2_VS" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3|sed '1d' )
  do
  
  sample=24740

    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo "=============="
  echo ${sample}
  echo "=============="

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
  done
  
  
  ##===========
##length
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#for sample in $(grep "20200527_x2_T43EXP01_Seq2_VS" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3 )
  do

  echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
  #==============================================================================================================
  ##----------------------------
  ###assemblies with problems
  ##----------------------------
    #==============================================================================================================
    
    ##-------------------
    ##count reads
    ##-------------------
    for sample in $(echo "24855 24773 24765 24742 24740 24738 24736 24726")
  do
   echo "=============="
  echo ${sample}
  
  grep "^@" -c /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq
done


    ##-------------------
    ##too few reads
    ##-------------------
      for sample in $(echo "24773 24765")
    do
    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo "=============="
  echo ${sample}
  echo "=============="

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
    
    done
  ##-------------------
    ##assembly length
    ##-------------------
  for sample in $(echo "24855 24742 24740 24738 24736 24726")
  do
   echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
  
  
    ##-------------------
    ##too many reads
    ###----->>>>here I filter for only the longest reads
    ##-------------------
    
    
    
        for sample in $(echo "24855 24742 24740 24738 24736 24726")
        do
        
        rm /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz
         
    
/home/vincent/apps/Filtlong/bin/filtlong --min_length 6000 --keep_percent 90 --target_bases 500000000 --mean_q_weight 10 /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq | gzip > /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz

#mv /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq.gz

        
        done
    
    
    ##-------------------
    ##too many reads
    ##assembly
    ##-------------------
    
    
      for sample in $(echo "24855 24742 24740 24738 24736 24726")
        do
        
       
    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo "=============="
  echo ${sample}
  echo "=============="

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
        
        done
    
    
      ##-------------------
    ##check
    ##-------------------
    
  for sample in $(echo "24855 24773 24765 24742 24740 24738 24736 24726")
  do
   echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
    
    
    
    
  for sample in $(echo "24855 24773 24765 24742 24740 24738 24736 24726")
  do
   echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
  sample=24740

    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo "=============="
  echo ${sample}
  echo "=============="

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
  done
  
    #==============================================================================================================
  ##----------------------------
  ###assemblies partially problematic
  ##optimise!!!
  ##change Minimum overlap 
  ##----------------------------
    #==============================================================================================================
    
      for sample in $(echo "24751 24731 24743 24758 24729 24795 24790 24777 24762 24759 24749 24731")
  do
   echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
    
   ##-------------------
    ##count reads
    ##-------------------
    for sample in $(echo "24751 24731 24743 24758 24729 24795 24790 24777 24762 24759 24749 24731")
  do
   echo "=============="
  echo ${sample}
  
  grep "^@" -c /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq
done

 ##-------------------
    ##filtlong
    ##-------------------
    for sample in $(echo "24751 24731 24743 24758 24729 24795 24790 24777 24762 24759 24749 24731")
  do
   echo "=============="
  echo ${sample}
   rm /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz
         
    
/home/vincent/apps/Filtlong/bin/filtlong --min_length 6000 --keep_percent 90 --target_bases 150000000 --mean_q_weight 10 /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq | gzip > /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz

done


##-------------------
    ##too many reads
    ##assembly
    ##-------------------
    
    
    for sample in $(echo "24751 24731 24743 24758 24729 24795 24790 24777 24762 24759 24749 24731")
        do
        
       
    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo "=============="
  echo ${sample}
  echo "=============="

   python /home/vincent/Flye/bin/flye --min-overlap 4000 --threads 5  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
        
        done
        
##-------------------
    ##final samples
    ##------------------   

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765")
#for sample in $(grep "20200527_x2_T43EXP01_Seq2_VS" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3 )
  do

  echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done



```

check circlatrity

```{bash}
rm /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt
echo -e "sample\tcircular_contigs\tlargestContigSize\tCoverage\tCircularity" > /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
  do
  echo "====================================================="
  echo $sample
  circularsss=$(grep "#" -v /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/30-contigger/contigs_stats.txt |awk -F "\t" '{OFS="\t"}{if($4=="Y")print $0}'| wc -l )
grep "#" -v /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/30-contigger/contigs_stats.txt |sort -t$'\t' -k2 -nr |head -1 |awk -F "\t" -v cics="$circularsss" -v samss="$sample" '{OFS="\t"}{print samss,cics,$2,$3,$4}' >> /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt
done
```

move local
```{bash}

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/ONT_assembly_stats.txt

```


##circlator

```{bash}


##===========
##circlator
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
  do
  

  echo "=============="
  echo ${sample}

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/circlator/${sample}
#mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/circlator/${sample}

mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/

#cat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R1_val_1.fq.gz \
#/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R2_val_2.fq.gz > \
#/data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/${sample}.fq.gz


circlator all /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta /data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/${sample}.fq.gz /data/Project/2020_StarterCultureDiversity/09_flye/circlator/${sample}


  done
  

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/
```
##polishing

```{bash}


###===========================
###-----------------
##cleanup
###-----------------
###===========================    
     
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv|
grep -v "24724"|
grep -v "24726"|
grep -v "24728"|
grep -v "24730"|
grep -v "24731"|
grep -v "24734"|
grep -v "24736"|
grep -v "24737"|
grep -v "24738"|
grep -v "24739"|
grep -v "24740")
  do

name=$(echo $sample)
echo ${name}
done
#sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}
reference_fasta=/data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta
ont_reads=/data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq
#reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
###===========================
###-----------------
##graphmap mapping
###-----------------
###===========================      

name=first
num=1

for name in first second third fourth
do 
#rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/
#mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc


#rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam
rm -r $Overall_output_directory
#rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2Final*

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 



done

done #sample


###===========================
###-----------------
##samples
###-----------------
###===========================    
     
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765")
#for sample in $(grep "20200527_x2_T43EXP01_Seq2_VS" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3 )
  do

  echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done

###===========================
###-----------------
##Racon polishing with graphmap
###-----------------
###===========================    
     
     
     24724  24726  24728  24730  24731  24734  24736  24737  24738  24739  24740
##------------------define variables------------------

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765"|
grep -v "24724"|
grep -v "24726"|
grep -v "24728"|
grep -v "24730"|
grep -v "24731"|
grep -v "24734"|
grep -v "24736"|
grep -v "24737"|
grep -v "24738"|
grep -v "24739"|
grep -v "24740")

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765")
  do

name=$(echo $sample)
#sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}
reference_fasta=/data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta
ont_reads=/data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq
#reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
###===========================
###-----------------
##graphmap mapping
###-----------------
###===========================      

name=first
num=1

for name in first second third fourth
do 
rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/
mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc

#name=first

graphmap align -t ${threads} \
  -r $reference_fasta \
  -d $ont_reads \
  -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam 

#  -d $ont_reads | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam - 

  
###===========================
###-----------------
##racon polishing
###-----------------
###===========================   

#rm -r$Overall_output_directory/0${num}_${name}Racon_graphmap/assembly

mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly

samtools view -bS $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam -

#samtools sort -@${threads} -O SAM $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam -

#samtools view -S -b $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam > $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam 

/home/vincent/apps/racon/new/racon/build/bin/racon \
 -t ${threads} \
  $ont_reads \
  $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam \
  $reference_fasta > \
  $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta
  


samtools index $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam 

qualimap bamqc -bam $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam \
  -outdir $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc --java-mem-size=40G 

#rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam
rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2Final*

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 



done

done #sample
##--------------------cleanup


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz
#reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
name=first
num=1

for name in first second third fourth
do 



#rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 

done
#done # loop for 1AB and 2AB samples


##--------------------test PROKKA
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
name=fourth
num=4

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes   $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta
  
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon/PROKKA_*.gff
  
  echo "=======================================================ALT==============================================================="
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all/PROKKA_*.gff


  


###===========================
###-----------------
##quast
###-----------------
###===========================  

#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz


echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
    rm -r $Overall_output_directory/quastwithRaconONT/
    mkdir -p $Overall_output_directory/quastwithRaconONT
    quast.py \
      -t ${threads} -l "FlyeAssembly,firstONTracon,secondONTracon,thirdONTracon" \
      -R $Overall_output_directory/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta \
      $reference_fasta \
      $Overall_output_directory/01_firstRacon_graphmap/assembly/firstRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/02_secondRacon_graphmap/assembly/secondRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/03_thirdRacon_graphmap/assembly/thirdRacon_NWC_2Assembly.fasta \
      -o $Overall_output_directory/quastwithRaconONT
 
 
 ##-------------------making MultiQC output
 rm $Overall_output_directory/logs_MultiQC_raconONT.txt
 
 echo $Overall_output_directory/quastwithRaconONT/report.tsv >> $Overall_output_directory/logs_MultiQC_raconONT.txt
 
 num=1

for name in first second third fourth
do 

echo $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc >> $Overall_output_directory/logs_MultiQC_raconONT.txt
num=$((num+1)) 

done ##loop for name
 
#done #loop for sample
###===========================
###-----------------
##MultiQC
###-----------------
###===========================  


#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do
name=di_K2_6h
sample=di_K2_6h
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${sample}/
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/assembly.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz

echo "========================================="
echo $sample
echo "========================================="



#done #loop for sample
 
rm -r $Overall_output_directory/MultiQCRaconONT/
mkdir -p $Overall_output_directory/MultiQCRaconONT/
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list   $Overall_output_directory/logs_MultiQC_raconONT.txt \
    -o $Overall_output_directory/MultiQCRaconONT/



###--------------on local

name=di_K2_6h
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/quastwithRaconONT/report.html ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_quast_report.html

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/MultiQCRaconONT/multiqc_report.html ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report.html



```


#### Illumina polishing freebayes 

Now we polish with the high quality Illumina reads

```{bash}


###-----------------
##Illuminamapping rawReads to circlator genome
###-----------------
sample=24758
   for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765"|tail -28)
do
##------------------define variables------------------
#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do

name=$(echo $sample)
#sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}
reference_fasta=/data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta
ont_reads=/data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq

num=5


for run in first second third fourth
   do 

#name=run
##--------------create directories----------------
rm -r $Overall_output_directory/0${num}_${run}Freebayes/
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/{rawIlluminaMapped,freebayesOuput}
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc

#mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq

# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R1_val_1.fq
# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R2_val_2.fq


#/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 index $reference_fasta

 # /home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 mem -t ${threads} $reference_fasta /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R2_val_2.fq.gz | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 

bwa index $reference_fasta

 bwa mem -t ${threads} $reference_fasta /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R2_val_2.fq.gz | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 



#bwa mem -t ${threads}  $reference_fasta /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq |samtools sort -@8 -O BAM \
#-o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 
     
#/data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/


qualimap bamqc -bam $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam \
        -outdir $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc --java-mem-size=40G
    
samtools index $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam

###-----------------
##freebayesAfterCirclator
###-----------------
echo freebayes

    SECONDS=0

    freebayes -F 0.5 --min-coverage 4 -p 1 -f \
      $reference_fasta \
      $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam > \
      $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf
      
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`

##-------   
##bcftools variant integration of Illumina data into Pacbio polished genome
##-------

    SECONDS=0
    echo "bgzip"
bgzip -c  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf > \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      echo "tabix"
tabix -p vcf  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
    echo "bcftools"
bcftools consensus -f $reference_fasta \
      -o $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`
  
  #bwa index $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta 
  
  rm $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam &
  
   ##-------------------new variable name for next polishing steps-----------
  reference_fasta=$Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta  
  num=$((num+1)) 
done
  done #sample  


##--------------indexing


#bwa index $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta  

##---------------
##PostAssemblyAnalysis
##---------------
for samplezzz in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765")
do

#samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

  echo "##=============="
  echo ${samplezzKurz}
  echo "##=============="
    #tail -3 $mainPath/03_metaphlan2/${id}_trimmed/metaphlan_profile/profiled_metagenome_${id}_trimmed.txt
    PostAssemblyAnalysis.R -i Y -a /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${samplezzz}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta \
      -F /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${samplezzz}/FAM${samplezzz}_R1_val_1.fq.gz \
      -R /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${samplezzz}/FAM${samplezzz}_R1_val_1.fq.gz \
      -o /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/PostaAssembly/FAM${samplezzz}/ \
      -t 35
   
      
    done




###--------------on local



 name=di_K2_6h
 scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/quastwithRacon_FreebayesONT/ ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/


scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/MultiQCRaconFreebayesONT/multiqc_report.html
~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report_final.html



```


##circlator

Start align with [fasta_shift](https://github.com/b-brankovics/fasta_tools).
start align only single contig genomes

```{bash}


##===========
##PROKKA annotate for fasta_shift
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765")
do
#for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#  do
  

  echo "=============="
  echo ${sample}


rm -r /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/
#mkdir -p /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${culture}_${strainzzz}

/usr/bin/perl /usr/bin/prokka --cpus 5 --outdir /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/ --locustag ${sample} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta

#prokka --outdir /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${culture}_${strainzzz} --force --usegenus --genus ${speciesss} --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes/${genomeszzz}

  done


##===========
##single conitg genome
##===========

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765" |
grep -v "24731"|
grep -v "24745"|
grep -v "24749"|
grep -v "24759"|
grep -v "24777")
do
#for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#  do
  

  echo "=============="
  echo ${sample}

seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*.fna
done

##===========
##complete but not circular genomes
##===========
grep -v "24731"
grep -v "24745"
grep -v "24749"
grep -v "24759"
grep -v "24777"

##===========
##identify dnaA
##===========

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765"|
grep -v "24731"|
grep -v "24745"|
grep -v "24749"|
grep -v "24759"|
grep -v "24777")
do
#for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#  do
  

  echo "=============="
  echo ${sample}

grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}'
done



##===========
##start align
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
  grep -v "24743"|
  grep -v "24729"|
  grep -v "24795"|
  grep -v "24794"|
  grep -v "24762"|
  grep -v "24751"|
  grep -v "24773"|
  grep -v "24765"|
  grep -v "24731"|
  grep -v "24745"|
  grep -v "24749"|
  grep -v "24759"|
  grep -v "24777"|
  grep -v "24758"); do      echo "=============="
 echo ${sample};  grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}';  orientation=$(grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}' |cut -f 7); contigName=$(grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}' |cut -f 1);  samtools faidx /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta ${contigName} > /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/bac_genome.fna;  rm -r /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned; mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned;  if [ $orientation == "+" ]
 then              echo "forward oriented";            startAlign=$(grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}' |awk -F "\t" '{print $4-5}')
 echo -e "start align at:  "${startAlign}
 /home/vincent/apps/fasta_tools/bin/fasta_shift -i /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/bac_genome.fna -p ${startAlign} > /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta
 else              echo "reverse oriented"
 startAlign=$(grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}' |awk -F "\t" '{print $5+5}')
 echo -e "start align at:  "${startAlign}
 /home/vincent/apps/fasta_tools/bin/fasta_shift -i /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/bac_genome.fna -p ${startAlign} > /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/tmp.fasta;  revseq -sequence /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/tmp.fasta -outseq /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta -notag;              fi;  for non_bac in $(grep ">" /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta | grep -v "${contigName}"|awk -F " " '{print $1}' |sed 's/>//g'); do samtools faidx /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta ${non_bac} >> /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta
 done #non_bac;  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta;  done
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765"|
grep -v "24731"|
grep -v "24745"|
grep -v "24749"|
grep -v "24759"|
grep -v "24777"|
grep -v "24758")
do      echo "=============="
echo ${sample}
rm -r /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter n/perl /usr/bin/prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter --locustag ${sample} --proteins s.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta
done

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765"|
grep -v "24731"|
grep -v "24745"|
grep -v "24749"|
grep -v "24759"|
grep -v "24777"|
grep -v "24758")
do      echo "=============="
echo ${sample}
grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter/*gff |awk -F "\t" '{if($3=="gene")print $0}'
done

```




```{bash}
##===========
##circlator
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765")
do
#for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#  do
  

  echo "=============="
  echo ${sample}

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/${sample}
#mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/circlator/${sample}

mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/tmp/

cat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R1_val_1.fq.gz \
/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R2_val_2.fq.gz > \
/data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/tmp//${sample}.fq.gz


circlator all /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta /data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/tmp/${sample}.fq.gz /data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/${sample}


  done
  

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/
```


## Illumina
Illumina only based assemblies. 



```{bash}
scp -r /media/vincent/Elements/seqData/20200515_UniBe vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/

```

download genomes for polishing
```{bash}
##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================
#http://uhts-lgtf.vital-it.ch/symlink/S1_L7_R1_001_JWc1BQSlYm0n.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S1_L7_R2_001_azQERS6e6Mf3.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S2_L7_R1_001_tAyGggUA76Sv.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S2_L7_R2_001_Z7AkETEcWnrG.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S3_L7_R1_001_QG9hjiVdib1X.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S3_L7_R2_001_hhue2W6w2qqw.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S4_L7_R1_001_sCvHPBHZTe80.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S4_L7_R2_001_mlaWMoWEB7ur.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S5_L7_R1_001_jcQlllEuwmpJ.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S5_L7_R2_001_WsUA1vmuMQds.fastq.gz
rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/99_log

vim /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_download.txt

vim /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_sampleNames2change.txt

#FAM24737	S1
#FAM24738	S2
#FAM24739	S3
#FAM24740	S4
#FAM24777	S5


##--------------download special case

cd /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/

wget -i /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_download.txt

```




###strain information

get strain information

```{bash}

scp /home/vincent/Desktop/Projects/2020_strainDelineation/01_logs/isolatedStrainsRMK.csv vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/99_log/

```


### renaming

```{bash}


for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

cat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_L*_R1_*.fastq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R1.fastq.gz

cat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_L*_R2_*.fastq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R2.fastq.gz


done #samplezzz


##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================

for name in $(cut -d '/' -f 5 /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_download.txt)
do

  name_01=$(echo $name |cut -d '_' -f 1)
  Lane=$(echo $name |cut -d '_' -f 2|sed 's/L//g')
  Read=$(echo $name |cut -d '_' -f 3)  
  
  Lane_new=$((Lane+2))

  newName=$(awk -F "\t" -v name="$name_01" '{if($2==name) {print $1}}' /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_sampleNames2change.txt)
  
  echo ${newName}"_"${Read}".fastq.gz"
    
  mv /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${name} /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${newName}_${Read}.fastq.gz
    
done
##-------qualtiy control

ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/|cut -d '_' -f 1|sort|uniq -c |wc -l
ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/|cut -d '_' -f 1|sort|uniq -c 


```



###FastQC
In a first step we creat [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) reports for all the libraries. 
These are then visualized with [MultiQC](https://multiqc.info/)

```{bash fastqc}
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh
threads=8

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)


    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R1.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R2.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
    
done #samplezzz

  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/MultiQC

  
  multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/MultiQC

##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================
threads=8

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/ |grep ".fastq")
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)


    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${samplezzKurz}_R1.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${samplezzKurz}_R2.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
    
done #samplezzz

  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/MultiQC

  
  multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/MultiQC


```

### Trim-galore
In the following we clip adaptors with [trim-galore]()

```{bash trimGalore}
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh
threads=30
id=RMK202_withOld

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

  echo "##=============="
  echo ${samplezzKurz}
  echo "##=============="
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      trim_galore -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz} \
        /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R1.fastq.gz  \
        /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R2.fastq.gz 
        
  echo "##=============="
  echo ${id} "fastQC of trimmed"
  echo "##=============="
      
    mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/
      
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R1.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/
    
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R2.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/
    
        
  done 
  
python3 /home/vincent/miniconda2/bin/multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore//Multiqc




mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do


echo "/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/bamqc" >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log/multiqc_mapping_logfile.txt
   

done


  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore//Multiqc

  multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore//Multiqc


echo "/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/bamqc" >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log/multiqc_mapping_logfile.txt
   

samtools index /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam &


done

###--------------------------------multiQC

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/


  multiqc --config /archiv/logs/multiQC_config.txt --file-list /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log/multiqc_mapping_logfile.txt \
  -o archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/
##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================
threads=35
#id=RMK202_withOld

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz |grep ".fastq" )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

  echo "##=============="
  echo ${samplezzKurz}
  echo "##=============="
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      trim_galore --fastqc -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/${samplezzKurz} \
        /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${samplezzKurz}_R1.fastq.gz  \
        /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${samplezzKurz}_R2.fastq.gz 
        
  echo "##=============="
  echo ${id} "fastQC of trimmed"
  echo "##=============="
      

  done 
  
##--------------------------------------
##bring together
##--------------------------------------


for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz |grep ".fastq" )
do
samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

cp -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/ /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/
  
done

##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================

##special case

cat /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_1A3P4_S293_R1_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S301_R1_001.fastq.gz > /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S3012_R1_001.fastq.gz

cat /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_1A3P4_S293_R2_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S301_R2_001.fastq.gz > /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S3012_R2_001.fastq.gz

mkdir -p /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns
mv /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_1A3P4_S293_R1_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns/

mv /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S301_R1_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns/

mv /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_1A3P4_S293_R2_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns/

mv /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S301_R2_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns/

##redo

threads=35
#id=RMK202_withOld

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
for samplezzz in $(ls /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/ |grep "^1" )
do
samplezzz=13499

samplezzKurz=$(echo -e "FAM"${samplezzz} )

  echo "##=============="
  echo ${samplezzKurz}
  echo "##=============="
  
  

  
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      trim_galore --fastqc -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz} \
        /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/${samplezzz}/${samplezzz}*_R1*.fastq.gz  \
        /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/${samplezzz}/${samplezzz}*_R2*.fastq.gz
        
  echo "##=============="
  echo ${id} "fastQC of trimmed"
  echo "##=============="
      

  done 



##--------------------------------------
##bring together
##--------------------------------------


for samplezzz in $(ls /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/ |grep "^1" )
do
samplezzKurz=$(echo -e "FAM"${samplezzz} )
rm  -r /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}

cp /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzz}*_R1*val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz

cp /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzz}*_R2*_val_2.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz

done
  
```

```{bash}


scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore//Multiqc /home/vincent/Desktop/Projects/2020_strainDelineation/01_logs

```
###metaphlan
I do metaphlan to check for contamination

```{bash}


rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/115_i_24728.fasta

 rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes/202_i_24753.fasta
 rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes/202_i_24754.fasta
 rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes/190_i_24736.fasta

rm /data/Project/2020_StarterCultureDiversity/04_metaphlan/allStrains_metaphlan.txt
#for speciesss in $(echo "L.delbrueckii S.thermophilus")
for speciesss in $(echo "S.thermophilus")
do
rm -r /data/Project/2020_StarterCultureDiversity/04_metaphlan/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/04_metaphlan/${speciesss}
echo "-----------------"
echo ${speciesss}
#for strainzzz in $(grep "^${speciesss}" /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv |cut -f 11 |head -1)
for strainzzz in $(echo "24853 24854 24855")
do
culture=$(awk -F "\t" -v strainsss="$strainzzz" '{OFS="\t"}{if($11==strainsss)print $4}' /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed 's/RMK//g')

echo -e ${strainzzz} " from " ${culture}


mkdir -p /data/Project/2020_StarterCultureDiversity/04_metaphlan/${speciesss}/${culture}_i_${strainzzz}/{bowtie2ouput,metaphlan_profile}
 
   metaphlan2 --nproc 30 --input_type fastq <(zcat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${strainzzz}/FAM${strainzzz}_R1_val_1.fq.gz  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${strainzzz}/FAM${strainzzz}_R2_val_2.fq.gz ) \
   --bowtie2out  /data/Project/2020_StarterCultureDiversity/04_metaphlan/${speciesss}/${culture}_i_${strainzzz}//bowtie2ouput/metagenome_${strainzzz}_trimmed.bowtie2.bz2 --nproc 37 > \
   /data/Project/2020_StarterCultureDiversity/04_metaphlan/${speciesss}/${culture}_i_${strainzzz}//metaphlan_profile/profiled_metagenome_${names}_trimmed.txt
   
   grep "s__" /data/Project/2020_StarterCultureDiversity/04_metaphlan/${speciesss}/${culture}_i_${strainzzz}//metaphlan_profile/profiled_metagenome_${names}_trimmed.txt |grep "t__" -v|awk -F "[\t|]" -v culturess="$culture" -v strainzzzss="$strainzzz" -v specisss="$speciesss" '{OFS="\t"}{print culturess,strainzzzss,specisss,$7,$8}' >>  /data/Project/2020_StarterCultureDiversity/04_metaphlan/allStrains_metaphlan.txt
   
done #strainzzz
done #species




```



###spades

In this step we assemble the raw reads of the isolate with [spades](http://bioinf.spbau.ru/en/spades3.7).

```{bash spades}
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh

threads=35
num=1
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do
samplezzKurz=FAM24777
samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)
  echo "##=============="
  echo ${samplezzKurz}
  echo $num
  echo "##=============="
       mkdir -p /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly/
 
 
     /home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --pe1-1 /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz \
       --pe1-2 /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz \
       -o /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly//assembly_Spades_20200516 -t ${threads} -m 100

   num=$((num+1))


    done

#---------------
##short info
##---------------
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)


 count=$(grep ">" -c /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly//assembly_Spades_20200516/contigs.fasta)
   
     #echo "##=============="
  echo -e ${samplezzKurz} " : " ${count}
#  echo "##=============="
 
    done
##---------------
##PostAssemblyAnalysis
##---------------
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

  echo "##=============="
  echo ${samplezzKurz}
  echo "##=============="
    #tail -3 $mainPath/03_metaphlan2/${id}_trimmed/metaphlan_profile/profiled_metagenome_${id}_trimmed.txt
    PostAssemblyAnalysis.R -a /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly//assembly_Spades_20200516/scaffolds.fasta \
      -F /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz \
      -R /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz \
      -o /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly/PostaAssembly \
      -t 35
    
    done




```

###metabat

The sample 24728 has a s__Pediococcus_pentosaceus contaminatino. this is most likely from sequencing as we had samples of pediococcus on the same lane. I am going to take them out by binning. 

```{bash}
##-------
##mapping
##-------
samplezzKurz=FAM24728
threads=30
rm - /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/
mkdir -p /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/${samplezzKurz}/{binning,mapping}

gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq

gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq

bwa mem -t ${threads} /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly//assembly_Spades_20200516/contigs.fasta \
  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq | samtools sort -@${threads} -O BAM -o /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/${samplezzKurz}/mapping/${samplezzKurz}_rawIllumina_bwamem_sorted.bam


##-------
##cleaning contigs
##-------

metabat2 -t ${threads} \
  -i /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly//assembly_Spades_20200516/contigs.fasta \
  /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/${samplezzKurz}/mapping/${samplezzKurz}_rawIllumina_bwamem_sorted.bam \
  -o /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/${samplezzKurz}/binning
  
  
##-------
##BLAST QC
##-------

for genome in $(ls /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/)
  do
  
  genome_renamed=$(echo $genome |sed 's/.fa//g'  )
  
  #mkdir -p /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/${genome_renamed}_metaphlan
  
  
   infoseq /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/${samplezzKurz}/binning.1.fa
   infoseq /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/${samplezzKurz}/binning.2.fa
  
 
  done

```



##Dialact strains



####download

Here, I download the spades genomes of the strains isolated out of the cheese starter cultures

```{bash}

##=============
##Download Genomes
##=============
date=20201010
downloadFormat=fna

##-----------------
##Dialact
##-----------------

   species=$(echo "Streptococcus_salivarius")
 #   species=$(echo "Lactobacillus_delbrueckii")
   echo ==========================
   echo ${species}
   echo ==========================
   
   DirectoryName=/data/Project/2020_StarterCultureDiversity//07_DialactStrains/${date}/Genomes/Dialact_02/
   #species=Streptococcus_thermophilus
   rm -r $DirectoryName/$species
   mkdir -p $DirectoryName/$species
 
 python3 /home/vincent/apps/Dialact_access.py -u vincent_somerville \
  -t $species \
  -f $downloadFormat \
  -s $DirectoryName/$species
 

```

Get strains that have been isolated from the cheese starter culture and have been sequenced. 
Noam added the list with additional infomormation. 
Here, I transfer it to the big machine. 


```{bash}

scp /home/vincent/Desktop/Projects/2020_strainDelineation/01_logs/20200113_Staemme_RMK_onlySequenced.csv vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/

```

isolate the strains from all strains
```{bash}
rm -r /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/
mkdir -p /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/{Sterm,Ldel}
for sampleszzz in $(sed '1d' /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/20200113_Staemme_RMK_onlySequenced.csv| cut -f 1 )
do
culture=$(grep "${sampleszzz}" /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/20200113_Staemme_RMK_onlySequenced.csv| cut -f 2|sed 's/ //g'|sed 's/RMK//g')
echo -e ${sampleszzz} " from : "${culture}

cat /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/Streptococcus_salivarius/FAM${sampleszzz}_*.fna > \
  /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/Sterm/${culture}_d_${sampleszzz}.fna
cat /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/Lactobacillus_delbrueckii/FAM${sampleszzz}_*.fna > \
  /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/Ldel/${culture}_d_${sampleszzz}.fna
done

find /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/Ldel/ -type f -empty -delete
find /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/Sterm/ -type f -empty -delete


```


#Strain collection

###complete NCBI Refs
```{bash}
##=====================================================================================
##------------------------all complete streptos
##=====================================================================================
species=$(echo "Streptococcus thermophilus")
species_short=Sterm_references
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
    #pecies_short=Sterm_references

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_thermophilus_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm_names.txt

##--------------------------
##prepare list
##--------------------------
species=$(echo "Streptococcus thermophilus")
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_00.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/S-thermophilus-/S-/g'| sed 's/-STH-CIRM-/-I-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_01.txt
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt
while read -r line
do
#echo $line
echo -e ${line} |head -c 10 |head -1 |sed 's/^$//g' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 
echo -e "\n" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 

done < /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_01.txt
sed -i '/^[[:space:]]*$/d' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 

paste -d '\t' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_00.txt > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt

#-------------------- add outgroup
###CHANGED OUTGRPOUP ON 28.1.2021 TO S. VESTIBULARIS

outgroupNAME=NCTC8618

outgroupNAME=$(echo "Streptococcus vestibularis")


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >>    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
    species_short=Sterm_references

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_salivarius_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'>> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm_names.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print "Out-sali",$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print ,$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/S-thermophilus-/S-/g'| sed 's/-STH-CIRM-/-I-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt


cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
gunzip *.fna.gz


##------------change name
sed -i 's/ //g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt

for genomesss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt)
do
shortnamesss=$(grep "${genomesss}" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt | cut -f 1)
echo -e ${genomesss} " to " ${shortnamesss}

mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${genomesss}*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna

done
mv GCF_900636445.1_41965_G01_genomic.fna Out-vest.fna

##=====================================================================================
##------------------------all complete lactos
##=====================================================================================
species_short=Ldel_references
species=$(echo "Lactobacillus delbrueckii subsp. lactis")

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
#curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" )print $0}' |cut -f 20| sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_ldel.txt
    #pecies_short=Sterm_references
    

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}"|grep "strain=" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "L_delbrueckii_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_ldel_names.txt


##--------------------------
##prepare list
##--------------------------
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1,$2,$3,"L_delbrueckii_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_00.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1,$2,$3,"L_delbrueckii_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/L-delbrueckii-/L-/g'| sed 's/-STH-CIRM-/-I-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_01.txt
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt
while read -r line
do
#echo $line
echo -e ${line} |head -c 10 |head -1 |sed 's/^$//g' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 
echo -e "\n" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 

done < /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_01.txt
sed -i '/^[[:space:]]*$/d' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 

paste -d '\t' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_00.txt > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt


##-------------------------
##outgroup
##type strain l. delbrueckii spp. bulgaricus


outgroupNAME=$(echo "ATCC 11842")

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >>    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_ldel.txt

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_salivarius_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'>> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm_names.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print "Out-bulg",$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print ,$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/S-thermophilus-/S-/g'| sed 's/-STH-CIRM-/-I-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt


cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_ldel.txt
gunzip *.fna.gz




##------------change name
sed -i 's/ //g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt

for genomesss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt)
do
shortnamesss=$(grep "${genomesss}" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt | cut -f 1)
echo -e ${genomesss} " to " ${shortnamesss}

mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${genomesss}*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna

done

##=====================================================================================
##------------------------PROKKA annotate (for panaroo)
##=====================================================================================$

##-------------------------Sterm
shortnamesss=$(echo "Out-vest")
species_short=Sterm_references
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKK*
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_GFF
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN
for shortnamesss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt)
#for genome in $(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FNA |grep "fna")
do

#shortnamesss=$(grep "${genome}" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt | cut -f 2)
#echo -e ${genome} " to " ${shortnamesss}

locustagss=$(echo ${shortnamesss}|cut -d '_' -f 3,4,5,6|sed 's/L-delbrueckii-/L-/g' |sed 's/S-thermophilus-/S-/g'|sed 's/STH-//g'|sed 's/S-salivarius-/REF-/g'|cut -d '.' -f 1)
echo $locustagss
#rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/PROKKA/${shortnamesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}

/usr/bin/perl /usr/bin/prokka --cpus 37 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss} --force --usegenus --genus streptococcus --locustag ${locustagss} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.gff > \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_GFF/${shortnamesss}.gff

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.ffn > \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN/${shortnamesss}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.faa > \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA/${shortnamesss}.faa


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.ffn > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN/${strainzzz}.ffn

done
##--------------------------------------Ldel
species_short=Ldel_references
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKK*
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_GFF
rm -r  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN

for shortnamesss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt)
#for genome in $(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FNA |grep "fna")
do

#shortnamesss=$(grep "${genome}" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt | cut -f 2)
#echo -e ${genome} " to " ${shortnamesss}

locustagss=$(echo ${shortnamesss}|cut -d '_' -f 3,4,5,6|sed 's/L-delbrueckii-/L-/g' |sed 's/S-thermophilus-/S-/g'|sed 's/STH-//g'|sed 's/S-salivarius-/REF-/g'|cut -d '.' -f 1)
echo -e ${shortnamesss} "and locusTag " $locustagss

#done
#rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/PROKKA/${shortnamesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}

/usr/bin/perl /usr/bin/prokka --cpus 5 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss} --force --usegenus --genus lactobacillus --locustag ${locustagss} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.gff > \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_GFF/${shortnamesss}.gff

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.ffn > \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN/${shortnamesss}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.faa > \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA/${shortnamesss}.faa


#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.ffn > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN/${strainzzz}.ffn

done

```


move local
```{bash}
mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/


scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm.txt

```

###gather own genomes

first I need to gather all necessary genomes. 

```{bash}
scp /home/vincent/Downloads/RMK_strain_stock\ -\ Sheet1\ \(1\).tsv vincent@130.223.49.145:/data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv

##------------------------------------
##ONT + Illumina genomes 
##------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/

awk -F "\t" '{if($7=="Illumina"||$7=="ONT")print $0}' /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |grep -v  "species" > /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765"|
grep -v "24731"|
grep -v "24745"|
grep -v "24749"|
grep -v "24759"|
grep -v "24777"|
grep -v "24758")
do      echo "=============="
echo ${sample}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/{all,Sterm,Ldel}


speciesss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 2|head -c 1)
culturesss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 4)
methodsss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 7|head -c 1)


cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${speciesss}_${methodsss}_${culturesss}_${sample}.fasta

grep -v "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv > /data/Project/2020_StarterCultureDiversity/99_log/tmp

mv /data/Project/2020_StarterCultureDiversity/99_log/tmp /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv

done
wc -l /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv
wc -l /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv
cut -f 2 /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv|sort|uniq -c
##------------------------------------
##Illumina genomes Sterm
##------------------------------------

for sample in $(grep "thermophilus" /data/Project/2020_StarterCultureDiversity/99_log/tmp /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv |cut -f 1|cut -d ':' -f 2|grep -v "1708"|grep -v "24052"|grep -v "24054")
do      echo "=============="
echo ${sample}


speciesss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 2|head -c 1)
culturesss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 4)
methodsss=$(echo "I")


cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*_${sample}*fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${speciesss}_${methodsss}_${culturesss}_${sample}.fasta


done

##------------------------------------
##Illumina genomes Ldel
##------------------------------------

for sample in $(grep "delbrueckii" /data/Project/2020_StarterCultureDiversity/99_log/tmp /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv |cut -f 1|cut -d ':' -f 2)
do      echo "=============="
echo ${sample}
#sample=21745

speciesss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 2|head -c 1)
culturesss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 4)
methodsss=$(echo "I")


#cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/*_${sample}*fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${speciesss}_${methodsss}_${culturesss}_${sample}.fasta
cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/*_${sample}*fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${sample}.fasta



done


cat /data/Project/2020_StarterCultureDiversity/06_Spades/FAM24777/assembly/assembly_Spades_20200516/contigs.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/L_I_202_24777.fasta 


grep "21745" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv
cat /archiv/Dialact/ReferenceDatabase/20190322/Annotation/Prokka/Lactobacillus_delbrueckii/FAM21745c1_11092016.fna/PROKKA_03272019.fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/L_I_101_21745.fasta 



ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |cut -d _ -f 1|sort|uniq -c

##-------------------------------------================================================================
#rename contigs
##-------------------------------------================================================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |sed 's/.fasta//g')
do
echo "============================================"
num=1
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed/${genomesss}.fasta
for contigs in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${genomesss}.fasta |sed 's/>//g')
  do
  echo ${contigs} " to "${genomesss}"_c"$num
  
  sed -i "s/$contigs/${genomesss}_c$num/g"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed/${genomesss}.fasta
  
  #sed "s/$contigs/S_term_phage_01_contig$num/g" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta |grep ">"
  
  num=$((num+1))
  done


done #genomesss

##-------------------------------------================================================================
#remove small contigs
##-------------------------------------================================================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |sed 's/.fasta//g')
do
  echo ==========================
   echo ${genomesss}
   echo ==========================
 

  seqkit seq -m 500 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed/${genomesss}.fasta -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved/${genomesss}.fasta
 
    done


##-------------------------------------================================================================
#Final and split species
##-------------------------------------================================================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/{Sterm,Ldel}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |grep "^S" |sed 's/.fasta//g')
do
  echo ==========================
   echo ${genomesss}
   echo ==========================
 

  cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved/${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/${genomesss}.fasta
 
    done
    
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |grep "^L" |sed 's/.fasta//g')
do
  echo ==========================
   echo ${genomesss}
   echo ==========================
 

  cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved/${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Ldel/${genomesss}.fasta
 
    done
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Ldel/ |wc -l
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/ |wc -l


##=================================================================
##DO A NEW COLLECTION 20210201
##=================================================================
##------------------------------------------
##STERM
##------------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm

mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/{Sterm,Ldel}



##--------01 Ilumina genomes


for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/ |cut -d '_' -f 3|sed 's/.fasta//g')
do
echo "=========================================================================="
echo ${genomesss}
#seq_length.py /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*${genomesss}.fasta |head -1
num=1
cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

for contigs in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta |sed 's/>//g')
  do
  echo ${contigs} " to "${genomesss}"_c"$num
  
  sed -i "s/$contigs/${genomesss}_contig_$num/g"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta
  
  #sed "s/$contigs/S_term_phage_01_contig$num/g" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta |grep ">"
  
  num=$((num+1))
  done
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta |head -1
done

 ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm |wc -l

##--------02 flye genomes

for genomesss in $(echo "24750
24739
13496
24738
13499
24730
24726
24728
13498
24734
24737
24853
24854
24855
24742
13494
24758
24735
24736
24724
24748
24740")
do
echo "--------------------------------------------"
#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta



seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

done



for genomesss in $(echo "24745

")
do
echo "--------------------------------------------"
#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

shortsss=$(echo ${genomesss} |cut -d '_' -f 1)

sed "s/>contig/>${shortsss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${shortsss}.fasta



seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${shortsss}.fasta

done

###---------------------rmk genome
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/*.fna
cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_O_202_SMAG.fna  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/202SMAG.fasta

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_I_202_S50.fna  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/S50.fasta

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_I_202_S72.fna  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/S72.fasta

##------------------------------------------
##Ldel
##------------------------------------------


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel

mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/{Sterm,Ldel}



##--------01 Ilumina genomes


for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/ |cut -d '_' -f 3|sed 's/.fasta//g')
do
echo "=========================================================================="
echo ${genomesss}
#seq_length.py /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*${genomesss}.fasta |head -1
num=1
cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/*${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

for contigs in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta |sed 's/>//g')
  do
  echo ${contigs} " to "${genomesss}"_c"$num
  
  sed -i "s/${contigs}/${genomesss}_contig_${num}/g"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta
  
  #sed "s/$contigs/S_term_phage_01_contig$num/g" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta |grep ">"
  
  num=$((num+1))
  done
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta |head -1
done

 ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel |wc -l


##--------02 flye genomes
genomesss=24758
#24759
#24777

for genomesss in $(echo "24756
24786
24790")
do
echo "--------------------------------------------"
echo ${genomesss}
#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

num=1
#seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta
for contigs in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta |sed 's/>//g')
  do

 sed -i "s/${contigs}/${genomesss}_contig_${num}/g"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

num=$((num+1))

done
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

done




```

### align contigs

Here, I use Rattag to align the illumina (draft) genome according to the synteny of the ONT genomes. 

```{bash}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/
ragtag.py correct ref.fasta query.fasta

mkdir -p aligned

###==========================================================
##start align Illumina assemblies
###==========================================================
###lin1

ragtag.py correct 24750.fasta 24732.fasta -u -t 35  --inter -o ./aligned/24732_corrected
ragtag.py correct 24750.fasta 24731.fasta -u -t 35  --inter -o ./aligned/24731_corrected
ragtag.py correct 24750.fasta 24733.fasta -u -t 35  --inter -o ./aligned/24733_corrected
ragtag.py correct 24750.fasta 24748.fasta -u -t 35  --inter -o ./aligned/24748_corrected

###lin2
ragtag.py correct 24739.fasta 13497.fasta -u -t 35  --inter -o ./aligned/13497_corrected
ragtag.py correct 24739.fasta 13495.fasta -u -t 35  --inter -o ./aligned/13495_corrected

###lin3
ragtag.py correct 24738.fasta 24744.fasta -u -t 35  --inter -o ./aligned/24744_corrected
ragtag.py correct 24738.fasta 24745.fasta -u -t 35  --inter -o ./aligned/24745_corrected
ragtag.py correct 24738.fasta S50.fasta -u -t 35  --inter -o ./aligned/S50_corrected

###lin4
ragtag.py correct 24730.fasta 24729.fasta -u -t 35  --inter -o ./aligned/24729_corrected

###lin5
ragtag.py correct 24730.fasta 24749.fasta -u -t 35  --inter -o ./aligned/24749_corrected
ragtag.py correct 24730.fasta 24747.fasta -u -t 35  --inter -o ./aligned/24747_corrected


###lin6

ragtag.py correct 24726.fasta 24727.fasta -u -t 35  --inter -o ./aligned/24727_corrected
###lin7
ragtag.py correct 24734.fasta 24724.fasta -u -t 35  --inter -o ./aligned/24724_corrected
ragtag.py correct 24734.fasta 24725.fasta -u -t 35  --inter -o ./aligned/24725_corrected

###lin8
ragtag.py correct 24737.fasta 13492.fasta -u -t 35  --inter -o ./aligned/13492_corrected
ragtag.py correct 24737.fasta 13500.fasta -u -t 35  --inter -o ./aligned/13500_corrected
ragtag.py correct 24737.fasta 13493.fasta -u -t 35  --inter -o ./aligned/13493_corrected
ragtag.py correct 24742.fasta 24741.fasta -u -t 35  --inter -o ./aligned/24741_corrected
ragtag.py correct 24742.fasta 13491.fasta -u -t 35  --inter -o ./aligned/13491_corrected

###lin9
ragtag.py correct 24742.fasta 24743.fasta -u -t 35  --inter -o ./aligned/24743_corrected


###lin10
ragtag.py correct 24742.fasta 24746.fasta -u -t 35  --inter -o ./aligned/24746_corrected
###lin11
ragtag.py correct 24736.fasta 24758.fasta -u -t 35  --inter -o ./aligned/24758_corrected
ragtag.py correct 24736.fasta 12273.fasta -u -t 35  --inter -o ./aligned/12273_corrected
#ragtag.py correct 24736.fasta 12273.fasta -u -t 35 -o ./aligned/12273_corrected
#ragtag.py correct 24736.fasta 12273.fasta -u -t 35 -b 0 -o ./aligned/12273_corrected

###linx
ragtag.py correct 24855.fasta S72.fasta -u -t 35  --inter -o ./aligned/S72_corrected


###==========================================================
##bring all genomes together
###==========================================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210204_final_startAligned
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210204_final_startAligned

genomesss=12273
##--------------------------Illuminas
  for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/ |grep ".fasta$"|cut -d '.' -f 1)
  do
  echo "--------------------------------------------"
  #sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta
  
 cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm//aligned/${genomesss}_corrected/*.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210204_final_startAligned/${genomesss}.fasta
  
num=1
for contigss in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm//aligned/${genomesss}_corrected/*.fasta|sed 's/>//g')
do

  sed -i "s/>${contigss}$/>${genomesss}_contig_${num}/g" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210204_final_startAligned/${genomesss}.fasta
  
  

#sed "s/>${contigss}/>${genomesss}_contig_${num}/g" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210204_final_startAligned/${genomesss}.fasta

num=$((num+1))
done
  

  
  #seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta
  
  done


##--------------------------ONT
genomesss=24855
for genomesss in $(echo "24750
24739
13496
24738
13499
24730
24726
24728
13498
24734
24737
24853
24854
24855
24742
13494
24735
24736
24724
24748
24740
202SMAG")
do
echo "--------------------------------------------"
#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210204_final_startAligned/${genomesss}.fasta

num=1
for contigss in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta|sed 's/>//g')
do



sed -i "s/>${contigss}$/>${genomesss}_contig_${num}/g" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210204_final_startAligned/${genomesss}.fasta

num=$((num+1))
done

#seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

done


ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/ |grep ".fasta$" |wc -l
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210204_final_startAligned |grep ".fasta$" |wc -l


#============================================================================
##Ldel
#============================================================================

##aligne all Illumina genomes


for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/|grep ".fasta$" |sed 's/.fasta//g'|grep "24756" -v |grep "24786" -v |grep "24790" -v )
do
echo "=========================================================================="
echo ${genomesss}
#seq_length.py /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*${genomesss}.fasta |head -1
num=1

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/
rm -r ./aligned/${genomesss}_corrected
mkdir -p ./aligned/
ragtag.py correct 24790.fasta ${genomesss}.fasta -u -t 35  --inter -o ./aligned/${genomesss}_corrected

done



##---------------------------------
##bring together
##---------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210204_final_startAligned
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210204_final_startAligned

##--------------------------Illuminas
  for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/ |grep ".fasta$"|cut -d '.' -f 1)
  do
  echo "--------------------------------------------"
  #sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta
  
 cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel//aligned/${genomesss}_corrected/*.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210204_final_startAligned/${genomesss}.fasta
  
num=1
for contigss in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel//aligned/${genomesss}_corrected/*.fasta|sed 's/>//g')
do

  sed -i "s/>${contigss}$/>${genomesss}_contig_${num}/g" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210204_final_startAligned/${genomesss}.fasta
  
#sed "s/>${contigss}/>${genomesss}_contig_${num}/g" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210204_final_startAligned/${genomesss}.fasta

num=$((num+1))
done
    #seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta
    done

###ONT


for genomesss in $(echo "24756
24786
24790")
do
echo "--------------------------------------------"
echo ${genomesss}
#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210204_final_startAligned/${genomesss}.fasta

num=1
#seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta
for contigs in $(grep ">"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210204_final_startAligned/${genomesss}.fasta |sed 's/>//g')
  do

 sed -i "s/${contigs}/${genomesss}_contig_${num}/g"   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210204_final_startAligned/${genomesss}.fasta

num=$((num+1))

done
seq_length.py  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210204_final_startAligned/${genomesss}.fasta

done

####---------------------------------
#contaminatino removal
####---------------------------------



for genomesss in $(echo "21748
21753
21780")
do
echo "--------------------------------------------"
echo ${genomesss}
#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

rm  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210204_final_startAligned/${genomesss}.fasta

done


```

there are still some undiscussed issues:

- The pipeline does not rearrange the contigs if they are in the wrong orientation. 
- the dnaA often is in the middle of the contig. It should therefore be cut and the contigs devided


###PROKKA of strains

```{bash}

conda activate PROKKA
species=Sterm
strainzzz=24855
export PATH=/home/vincent/apps/ncbi-blast-2.10.0+/bin/:$PATH
speciesss=$(echo "S.thermophilus")
genus=Streptococcus
echo ${speciesss}
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/
#rm -r /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}
for strainzzz in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210204_final_startAligned/ |grep "fasta$" |sed 's/.fasta//g')
do
#culture=$(awk -F "\t" -v strainsss="$strainzzz" '{OFS="\t"}{if($11==strainsss)print $4}' /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed 's/RMK//g')
#ONLYsTRAIN=$(echo ${strainzzz} |cut -d _ -f 4)
echo -e ${strainzzz} 

#genomeszzz=$(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes/ |grep  "_${strainzzz}" )
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${strainzzz}*
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/

#prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${strainzzz}/ --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210204_final_startAligned/${strainzzz}.fasta

prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}// --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210204_final_startAligned/${strainzzz}.fasta


done #strainzzz
##-----------------------
##l.delbrueckii
##-----------------------

conda activate PROKKA

strainzzz=24855
export PATH=/home/vincent/apps/ncbi-blast-2.10.0+/bin/:$PATH
speciesss=$(echo "L.delbrueckii")
species=Ldel
genus=Streptococcus
echo ${speciesss}
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/
#rm -r /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}
for strainzzz in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210204_final_startAligned/ |grep "fasta$" |sed 's/.fasta//g')
do
#culture=$(awk -F "\t" -v strainsss="$strainzzz" '{OFS="\t"}{if($11==strainsss)print $4}' /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed 's/RMK//g')
#ONLYsTRAIN=$(echo ${strainzzz} |cut -d _ -f 4)
echo -e ${strainzzz} 

#genomeszzz=$(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes/ |grep  "_${strainzzz}" )
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${strainzzz}*
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/

#prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${strainzzz}/ --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210204_final_startAligned/${strainzzz}.fasta

prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}// --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210204_final_startAligned/${strainzzz}.fasta

done #strainzzz


##-----------------------
##l.delbrueckii
##-----------------------

#export PATH=/home/vincent/apps/ncbi-blast-2.10.0+/bin/:$PATH
#speciesss=$(echo "L.delbrueckii")
#genus=Lactobacillus
#echo ${speciesss}

#for strainzzz in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Ldel/|grep "fasta$" |sed 's/.fasta//g')
#do
#culture=$(awk -F "\t" -v strainsss="$strainzzz" '{OFS="\t"}{if($11==strainsss)print $4}' /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed 's/RMK//g')
#ONLYsTRAIN=$(echo ${strainzzz} |cut -d _ -f 4)
#echo -e ${strainzzz} 
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${strainzzz}
#/usr/bin/perl /usr/bin/prokka --cpus 30 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${strainzzz}/ --force --usegenus --genus ${speciesss} --locustag ${ONLYsTRAIN} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Ldel/${strainzzz}.fasta

#done #strainzzz

###================================================
##create FAA and FNA folders
###================================================

for species in $(echo "Ldel Sterm")
#for species in $(echo "Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all*
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/{01_all_FAA,01_all_FNA,01_all_GFF,01_all_FFN}

for strainzzz in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection//${species}/210204_final_startAligned/ |grep ".fasta$"|sed 's/.fasta//g')
do

#culture=$(awk -F "\t" -v strainsss="$strainzzz" '{OFS="\t"}{if($11==strainsss)print $4}' /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed 's/RMK//g')
ONLYsTRAIN=$(echo ${strainzzz} |cut -d _ -f 3)
echo -e ${strainzzz} 

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/${strainzzz}.fna

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.ffn > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/${strainzzz}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.faa > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/${strainzzz}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.gff > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/${strainzzz}.gff

done #strainzzz
done #species

##--------------------
##check DNAa
##--------------------
species=Sterm

for strainzzz in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection//${species}/210204_final_startAligned/ |grep ".fasta$"|sed 's/.fasta//g')
do
#culture=$(awk -F "\t" -v strainsss="$strainzzz" '{OFS="\t"}{if($11==strainsss)print $4}' /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed 's/RMK//g')
#ONLYsTRAIN=$(echo ${strainzzz} |cut -d _ -f 3)
#echo -e ${strainzzz} 

#seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${strainzzz}/PROKKA*.fna |head -1

grep "dnaA" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.gff |head -1 |cut -d ':' -f 2
#grep "dnaA" i /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${strainzzz}//PROKKA*.gff |head -1

done #strainzzz

```

###Virsorter

```{bash}

conda activate vs2

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter

for species in $(echo "Ldel Sterm" )
do
for sample in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ |grep ".fna$"|sed 's/.fna//g')
do
echo "========================================="
echo $sample
echo "========================================="

  rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${species}/${sample}
  mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${species}/${sample}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${species}/${sample}

virsorter run -w /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${species}/${sample}/ -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/${sample}.fna -j 35 all

#grep -v "^#" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${sample}/VIRSorter_global-phage-signal.csv 
 
 
done #samples
done #species



###============================================
#old
###============================================

for species in $(echo "Ldel Sterm" )
do
for sample in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/ |grep ".fasta$"|sed 's/.fasta//g')
do
#sample=1
threads=10
Overall_output_directory=/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${sample}


echo "========================================="
echo $sample
echo "========================================="



  rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${sample}
  mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${sample}
 
#source activate /home/vincent/miniconda3/envs/virsorter

#source activate /home/vincent/miniconda3/envs/virsorter_new
conda activate virsorter_new

wrapper_phage_contigs_sorter_iPlant.pl \
    -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/${sample}.fasta \
    --db 1 \
    --no_c \
    --wdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${sample} --ncpu ${threads} --data-dir /archiv/VirSorter/virsorter-data



grep -v "^#" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${sample}/VIRSorter_global-phage-signal.csv 
 
 
done #samples
done #species


###============================================
#show data
###============================================

for species in $(echo "Ldel Sterm")
do
for sample in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/ |sed 's/.fasta//g')
do
#sample=1
threads=10
Overall_output_directory=/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${sample}


echo "========================================="
echo $sample
echo "========================================="


grep -v "^#" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${sample}/VIRSorter_global-phage-signal.csv 
 
 
 done #samples
done #species

```

### BUSCO

```{bash}



##-----------------------------------
##script
##-----------------------------------
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis.txt

for speciesss in $(echo "Sterm Ldel")
do
echo "##------------------------"
echo ${speciesss}
echo "##------------------------"

for genemess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${speciesss}/ |sed 's/.fasta//g')
  do
  #genemess=L_delbrueckii_RMK202_mag.faa
  #genomesssss=$(echo ${genemess}|cut -d '.' -f 1)
  echo $genemess
  
  rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/${genemess}

  mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/${genemess}
  cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/
   export BUSCO_CONFIG_FILE=/home/vincent/apps/busco_v4/busco/config/myconfig.ini
  python3.6 /home/vincent/apps/busco_v4/busco/bin/busco -f -m prot -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FAA/${genemess}.faa -o ${genemess} -l lactobacillales -c 10
  #-i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FAA/${genemess}.faa -o ${genemess} -l /home/vincent/apps/busco/lactobacillales_odb9 -m prot

  #grep -v "^#" run_${genomesssss}/short_summary_${genomesssss}.txt| grep "C:" -A 7 
  all_INFO=$(grep -v "^#" ${genemess}/run_lactobacillales_odb10/short_summary.txt| grep "C:" | sed "s/^[ \t]*//"  )
  complet=$(echo ${all_INFO} | awk -F "[,[]" '{print $1}'|sed 's/%//g'|sed 's/C://g' )
  single=$(echo ${all_INFO} | awk -F "[,[]" '{print $2}'|sed 's/%//g'|sed 's/S://g')
  dupli=$(echo ${all_INFO} | awk -F "[,[]" '{print $3}'|sed 's/%]//g'|sed 's/D://g')
  fragm=$(echo ${all_INFO} | awk -F "[,[]" '{print $4}'|sed 's/%//g'|sed 's/F://g')
  Missi=$(echo ${all_INFO} | awk -F "[,[]" '{print $5}'|sed 's/%//g'|sed 's/M://g')
  numbi=$(echo ${all_INFO} | awk -F "[,[]" '{print $6}'|sed 's/n://g')
  
 echo -e ${speciesss}"\t"${genemess}"\t"${complet}"\t"${single}"\t"${dupli}"\t"${fragm}"\t"${Missi}"\t"${numbi}  \
 >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis.txt

  done
done ## species

#sed -i  's/Streptococcus_thermophilus_RMK202_pgap/S_thermophilus_mag_rmk202/g' /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt
#sed -i  's/Lactobacillus_delbrueckii_RMK202_pgap/L_delbrueckii_mag_rmk202/g' /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt

```


#### genome stats

```{bash}

##---------------
##genome
##---------------

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/all_assembly_analysis.txt
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats
for species in $(echo "Sterm Ldel")
do
for id in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/ |sed 's/.fasta//g')
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
# /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/
 
   contiNumber=$(grep ">" -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/${id}.fasta) 
   GenomeLength=$(grep -v "^>" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/${id}.fasta  |wc -c)
   
   # contiNumber=$(grep ">" -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta) 
   #GenomeLength=$(grep -v "^>" /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta  |wc -c)
    
 echo -e ${id}"\t"${contiNumber}"\t"${GenomeLength}  >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/all_assembly_analysis.txt

#"\t"${single}"\t"${dupli}"\t"${fragm}"\t"${Missi}"\t"${numbi}  \
    done
done #species


```

#### annotation stats

```{bash}


rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_analysis.txt
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_transposases_analysis.txt
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats
for species in $(echo "Sterm Ldel")
do
for id in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/ |sed 's/.fasta//g')
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
 
#   contiNumber=$(grep ">" -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/2*/*19.gff) 
#   GenomeLength=$(grep -v "^>" /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/2*/*19.gff  |wc -c)
  
  gensss=$(grep "gene: " /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${id}/PROKKA_*.txt |sed 's/gene: //g')
  tRNAsss=$(grep "tRNA: "  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${id}/PROKKA_*.txt |sed 's/tRNA: //g')
  rRNAsss=$(grep "rRNA: "  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${id}/PROKKA_*.txt |sed 's/rRNA: //g')
  
  
  
  transpononssss=$(grep "ISfinder:" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${id}/PROKKA_*.gff |wc -l)

grep "ISfinder:" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${id}/PROKKA_*.gff |grep ";product=" |awk -F ";product=" '{OFS="\t"}{print $2}'|awk -F " family transposase " -v namzzz="$id" '{OFS="\t"}{print namzzz,$1,$2}' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_transposases_analysis.txt

 echo -e ${id}"\t"${gensss}"\t"${tRNAsss}"\t"${rRNAsss}"\t"${transpononssss}  >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_analysis.txt

    done
done #species


```

####mapping stats

```{bash}

 
for species in $(echo "Sterm Ldel")
do
for id in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/|grep ".fasta$" |sed 's/.fasta//g')
  do

samplezzKurz=$(echo $id |cut -d '_' -f 4)
  echo ==========================
   echo ${id}
   echo ==========================


 rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}
  
  
  bwa index /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/${id}.fasta
  
  bwa mem -t 32 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/${id}.fasta \
  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${samplezzKurz}/FAM${samplezzKurz}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${samplezzKurz}/FAM${samplezzKurz}_R2_val_2.fq.gz | samtools sort -@32 -O BAM -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}/${id}_rawIllumina_bwamem_sorted.bam - 

##-----Qualimap

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc

qualimap bamqc -bam /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}/${id}_rawIllumina_bwamem_sorted.bam -outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc --java-mem-size=80G

    
    done
done #species

##---------------------
##get data
##---------------------

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_genome_analysis.txt
for species in $(echo "Sterm Ldel")
do
for id in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/|grep ".fasta$" |sed 's/.fasta//g')
  do

#GCss=$(grep "GC percentage" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt| sed 's/GC percentage = //g'| sed "s/^[ \t]*//"|sed 's/%//g'  )

coveragesss=$(grep "mean coverageData = " /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt| sed 's/mean coverageData = //g'| sed "s/^[ \t]*//"|sed 's/X//g'  )

#readsss=$(grep "number of reads = " /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt |awk -F "number of reads = " '{print $2}'|sed 's/,//g')

#mappedReadsss=$(grep "number of mapped reads = " /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt |awk -F "number of mapped reads = " '{print $2}'|awk -F " " '{print $1}'|sed 's/,//g')

percentmappedReadsss=$(grep "number of mapped reads = " /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt |awk -F "number of mapped reads = " '{print $2}'|awk -F " " '{print $2}'|sed 's/,//g'|sed 's/(//g'|sed 's/%)//g')

GCss=$(infoseq /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/${id}.fasta -noheading -nodatabase -nolength -nousa -nohead -noacc -notype -nodesc -delimiter '_' | awk -F " " '{sum+=$2} END {print sum/NR}')

#grep -A 10000 ">>>>>>> Coverage per contig" /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly//bamqc/genome_results.txt| grep -v ">>>>>>> Coverage per contig"| sed "s/^[ \t]*//" |sed -r '/^\s*$/d' 


echo -e ${id}"\t"${GCss}"\t"${coveragesss}"\t"${percentmappedReadsss} |sed 's/,//g' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_genome_analysis.txt

done
done #species


```

#### bring together
```{bash}

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal
cp all_assembly_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
cp all_genome_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
cp all_annotation_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
cp all_annotation_transposases_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/


mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
scp -r vincent@130.223.49.145:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/ ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/

```

```{r}
library(readr)
library(plyr)
library(tidyverse)
ONT_assembly_stats <- read_delim("Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/ONT_assembly_stats.txt", "\t", escape_double = FALSE, trim_ws = TRUE) %>% filter(largestContigSize>1700000)
genome_stats <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal//all_assembly_analysis.txt", "\t", escape_double = FALSE, col_names = c("strain","contigNumber","genomeSize"),  trim_ws = TRUE)
Genes_stats <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal//all_annotation_analysis.txt", "\t", escape_double = FALSE, col_names = c("strain","genes","tRNA","rRNA","transposons"),  trim_ws = TRUE)
BUSCO_stats <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal//all_analysis.txt", "\t", escape_double = FALSE, col_names = c("species","strain","completness","singleCopy","duplicated","fragmentation","Missing","numberGenesTested"),  trim_ws = TRUE)
mapping_stats <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal//all_genome_analysis.txt", "\t", escape_double = FALSE, col_names =c("strain","GC","IlluminaCoverage","percentMapped"),  trim_ws = TRUE)



tmp <- merge(BUSCO_stats,genome_stats,by="strain")
tmp <- merge(tmp,Genes_stats,by="strain")
GenomeAssemblyStats_tmp <- merge(tmp,mapping_stats,by="strain")

GenomeAssemblyStats_tmp$strain_short <- str_split_fixed(GenomeAssemblyStats_tmp$strain,"_",4)[,4]

GenomeAssemblyStats <- merge(GenomeAssemblyStats_tmp,ONT_assembly_stats,by.x="strain_short",by.y="sample",all.x=TRUE)
GenomeAssemblyStats$CIRCnUM <- revalue(GenomeAssemblyStats$Circularity, c("Y"="1","N"="0")) %>% as.numeric()
GenomeAssemblyStats$Sequencing <- ifelse(is.na(GenomeAssemblyStats$Circularity),"Illumina",ifelse(GenomeAssemblyStats$Circularity=="Y","ONT","Illumina"))

GenomeAssemblyStats$additional_circular_contigs <- GenomeAssemblyStats$circular_contigs-GenomeAssemblyStats$CIRCnUM
GenomeAssemblyStats <- GenomeAssemblyStats %>%  select(-CIRCnUM) %>%  select(-circular_contigs)
GenomeAssemblyStats$starterCulture <- str_split_fixed(GenomeAssemblyStats$strain,"_",4)[,3]
GenomeAssemblyStats <- GenomeAssemblyStats %>%  select(-strain)%>% dplyr::rename(strain="strain_short")


GenomeAssemblyStats$Cheese <- revalue(GenomeAssemblyStats$starterCulture,c("101"="emmentaler","105"="emmentaler","115"="emmentaler","124"="emmentaler","150"="emmentaler","190"="emmentaler","202"="gruyere","203"="gruyere","280"="gruyere","302"="sbrinz","305"="sbrinz"))

##------------------------
##remove 
##------------------------

 GenomeAssemblyStats %>% filter(completness<95|duplicated>5)
GenomeAssemblyStats_cleaned <- GenomeAssemblyStats %>% filter(completness>95&duplicated<5)
GenomeAssemblyStats_cleaned %>% filter(completness<95|duplicated>5)
dput(colnames(GenomeAssemblyStats_cleaned))


GenomeAssemblyStats_cleaned <- GenomeAssemblyStats_cleaned %>% select(c("strain","Cheese" ,"starterCulture","species", 
"Sequencing", "completness","contigNumber", 
"genomeSize", "genes", "tRNA", "rRNA", "transposons", "GC",  "Circularity", "additional_circular_contigs"))

GenomeAssemblyStats <- GenomeAssemblyStats_cleaned

##----------------------cleaning--------------------------------
table(GenomeAssemblyStats$strain)
GenomeAssemblyStats$species <- revalue(GenomeAssemblyStats$species, c("Sterm"="S.thermophilus","Ldel"="L.delbrueckii"))

# GenomeAssemblyStats$strain <- revalue(GenomeAssemblyStats$strain, c("S50"="mst1","S72"="mst2","L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4","L_delbrueckii_mag_rmk202"="Ldel_mag_202","S_thermophilus_mag_rmk202"="Sterm_mag_202"))

# GenomeAssemblyStats <- GenomeAssemblyStats[-which(GenomeAssemblyStats$strain=="L39"),]
colnames(GenomeAssemblyStats)

# GenomeAssemblyStats_final <- GenomeAssemblyStats %>% select(species,strain,contigNumber,genomeSize,GC,IlluminaCoverage,percentMapped,genes,tRNA,rRNA,transposons,completness)
GenomeAssemblyStats_final <- GenomeAssemblyStats 

##----------------------write table--------------------------------
  GenomeAssemblyStats_final %>% filter(species=="S.thermophilus")
write.csv(GenomeAssemblyStats_final,file="~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal///genomeAssemblyStats.csv",quote = FALSE,row.names = FALSE)
# 
# write.xlsx(GenomeAssemblyStats_final,file="Users//Desktop/presenations/my_presentations/20191007_groupmeeting/figures/genomeAssemblyStats.xls", sheetName = "Sheet1",
#   col.names = TRUE, row.names = FALSE, append = FALSE)

```


#Genome analysis

###Restriction modification

```{bash}
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/


grep "restriction" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff

grep "restriction" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff |grep "hsdR" -v |grep "sau3AIR" -v 

hsdR #Type-1 restriction 
sau3AIR # type II
hsdM # type I
mboIR  # type II

grep "HsdM" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff
grep "Hsds" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff

grep "HsdM" -A 5 -B 5 -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff

```

## EPS

maybe I should make genus specific or hmm databases (https://github.com/tseemann/prokka).

https://www.nature.com/articles/srep04974/figures/4

```{bash}

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/ |grep ".fna$" |head -1)
do
grep "=eps" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/S_I_190_24758.fna |awk -F ";gene=" '{print $2}' |cut -d ';' -f 1 |sort|uniq -c


done
```


## RGP

## Bacteriocins

## quorum sensing (ComABCDE)

## naturally competence system

## autolysis

## CRISPR-system

## plasmid

## prophage

with Phaster. I use the [API tool](https://github.com/boulund/phaster_scripts).

```{bash}

genomesss=L_I_101_10973
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/ |grep ".fna$"|sed 's/.fna//g' |head -1)
do
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER

#grep "=eps" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/S_I_190_24758.fna |awk -F ";gene=" '{print $2}' |cut -d ';' -f 1 |sort|uniq -c
echo $genomesss
#/home/vincent/apps/phaster_scripts/phaster.py  --contigs --fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/${genomesss}.fna --database /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt

wget --post-file="/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/${genomesss}.fna" "http://phaster.ca/phaster_api?contigs=1" -O /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt


wget "http://phaster.ca/phaster_api?acc=ZZ_0863a29868" -O /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt


done


wget "http://phaster.ca/phaster_api?acc=ZZ_0863a29868" -O /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt

mkdir -p  ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES


#scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/S_O_124_24730/PROKKA_07282020.gbf ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/PROKKA_07282020.gbk


```

In the following I will run the phaster online. therefore I have to transfer all fna files first to local

```{bash}

```


## codon usage (avoidance between species)

## transporters

## amino acid synthesis

#Phylogeny

Here, we do the analysis of the strain collection. 
I am doing the following analysis:
1. parSNP phylogeny. 
2. ...

### Poppunk

Here, I use [poppunk](https://poppunk.readthedocs.io/en/latest/model_fitting.html) to identify genome clusters

```{bash}

###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Sterm

for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/FNA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/
done

##--------------------------
##create genome list with path to genome assembly
##--------------------------
species=Sterm
rm  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
mkdir -p  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/ | grep ".fna$"|sed 's/.fna//g'|grep "Out" -v)
do

genomesss_short=$(echo $genomesss |sed 's/-//g'|sed 's/_//g')

#echo -e ${genomesss}"\t/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/"${species}"/"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}"\t"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
echo -e "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/"${species}"/"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log

done


##--------------------------
##create DB
##--------------------------
mkdir -p /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm
cd /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/

poppunk --create-db --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database --r-files /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log --k-step 2 --min-k 17 --max-k 31 --plot-fit 5


##--------------------------
##fit model
##--------------------------

poppunk --fit-model lineage --ref-db /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database \
    --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_lineageFIT --overwrite


poppunk --ref-db /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/ --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_lineageFIT  --fit-model 



##--------------------------
#easy run 
##--------------------------

poppunk --easy-run --r-files /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun --threads 4 --plot-fit 5 --min-k 13 --full-db

poppunk --fit-model --distances /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/20210128_easyrun.dists --ref-db /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/ --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/ --full-db --K 3


##====================================================================
##only RMKs
##====================================================================



##--------------------------
##create genome list with path to genome assembly
##--------------------------
species=Sterm
rm  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db_onlyRMK.log
mkdir -p  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA | grep ".fna$"|grep -v "^S-" | sed 's/.fna//g'|grep "Out" -v)
do

genomesss_short=$(echo $genomesss |sed 's/-//g'|sed 's/_//g')

#echo -e ${genomesss}"\t/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/"${species}"/"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}"\t"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
echo -e "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/"${species}"/01_all_FNA/"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db_onlyRMK.log

done


##--------------------------
#easy run 
##--------------------------
rm -r /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun
poppunk --easy-run --r-files /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db_onlyRMK.log --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun --threads 4 --plot-fit 5 --min-k 13 --full-db

poppunk --fit-model --distances /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun.dists --ref-db /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/ --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/ --full-db --K 3


sed 's;\/data\/Project\/2020_StarterCultureDiversity\/10_REFERENCE_GEN0MES\/annotation\/PROKKA\/Sterm\/01_all_FNA\/;;g' /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters.csv |sed 's/.fna//g' |cut -d '_' -f 4 >  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv
```


transfer local

```{bash}

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/fit*.pdf /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/*.png /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/



mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/*.png /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv

```


```{bash}
#==========================================================
#lDEL
#==========================================================
###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Ldel

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/

leterzzz=$(echo $species |head -c 1)

#cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/FNA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/${leterzzz}*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/

##HERE I remove the genomes with completness <95 or duplication >10
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/L_I_101_21748.fna
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/L_I_124_21753.fna
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/Ldel/L_I_101_21780.fna

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/L_I_101_21748.fna
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/L_I_124_21753.fna
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/Ldel/L_I_101_21780.fna


##--------------------------
##create genome list with path to genome assembly
##--------------------------

species=Ldel
rm  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/${species}_poppunk_db.log
mkdir -p  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/ | grep ".fna$"|sed 's/.fna//g'|grep "Out" -v)
do

genomesss_short=$(echo $genomesss |sed 's/-//g'|sed 's/_//g')

#echo -e ${genomesss}"\t/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/"${species}"/"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}"\t"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
echo -e "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/"${species}"/"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/${species}_poppunk_db.log

done


##--------------------------
##create DB
##--------------------------
#mkdir -p /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}
#cd /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/

#poppunk --create-db --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_database --r-files /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/${species}_poppunk_db.log --k-step 2 --min-k 17 --max-k 31 --plot-fit 5


##--------------------------
##fit model
##--------------------------

#poppunk --fit-model lineage --ref-db /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_database \
#    --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_lineageFIT --overwrite


#poppunk --ref-db /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_database/ --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_lineageFIT  --fit-model 



##--------------------------
#easy run 
##--------------------------
rm -r /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}
cd /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/

poppunk --easy-run --r-files /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/${species}_poppunk_db.log --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun --threads 4 --plot-fit 5 --min-k 13 --full-db

poppunk --fit-model --distances /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/20210128_easyrun.dists --ref-db /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/ --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/ --full-db --K 2

##====================================================================
##only RMKs
##====================================================================



##--------------------------
##create genome list with path to genome assembly
##--------------------------
species=Sterm
rm  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db_onlyRMK.log
mkdir -p  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA | grep ".fna$"|grep "^S_" | sed 's/.fna//g'|grep "Out" -v)
do

genomesss_short=$(echo $genomesss |sed 's/-//g'|sed 's/_//g')

#echo -e ${genomesss}"\t/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/"${species}"/"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}"\t"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
echo -e "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db_onlyRMK.log

done


##--------------------------
#easy run 
##--------------------------

poppunk --easy-run --r-files /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db_onlyRMK.log --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun --threads 4 --plot-fit 5 --min-k 13 --full-db

poppunk --fit-model --distances /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun.dists --ref-db /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/ --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/ --full-db --K 6


```


transfer local

```{bash}

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/fit*.pdf /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/*.png /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/



mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/*.png /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/*.csv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/

```



### fastANI

```{bash}
for speciesss in $(echo "L.delbrueckii S.thermophilus")
do
rm -r /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/{log,analysis}
echo "-----------------"
echo ${speciesss}

mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/{log,analysis}
for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes)
do
echo -e "/data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/"${speciesss}"/01_genomes/"${genomessss} >> /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt

done
done #speciesss


##------------new PREP
species=Sterm




rm -r /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/{log,analysis}
echo "-----------------"
echo ${speciesss}

mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/{log,analysis}
for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/)
do
echo -e "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/"${species}"/01_all_FNA/"${genomessss} >> /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt

done

###---------------------------
##fastaANI
###---------------------------
#for speciesss in $(echo "L.delbrueckii S.thermophilus")
for speciesss in $(echo "Sterm")
do
fastANI --fragLen 1000 -t 6 --ql /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt \
  --rl /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt \
  -o /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out
done #speciesss
###----------------------Sterm


cut -f 3 /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out |sort|uniq -c |head
grep "74\." /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out

##====================================
##all against all (independent of species)
##====================================
rm -r /data/Project/2020_StarterCultureDiversity/05_fastANI/all
mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/all/{log,analysis}
for speciesss in $(echo "L.delbrueckii S.thermophilus")
do

echo "-----------------"
echo ${speciesss}

mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/all/{log,analysis}
for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes)
do
echo -e "/data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/"${speciesss}"/01_genomes/"${genomessss} >> /data/Project/2020_StarterCultureDiversity/05_fastANI/all/log/genomes.txt

done
done #speciesss
###---------------------------
##fastaANI
###---------------------------

fastANI --fragLen 1000 -t 37 --ql /data/Project/2020_StarterCultureDiversity/05_fastANI/all/log/genomes.txt \
  --rl /data/Project/2020_StarterCultureDiversity/05_fastANI/all/log/genomes.txt \
  -o /data/Project/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis.out



cut -f 3 /data/Project/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis.out |sort|uniq -c |head

```

```{bash}
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis.out /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/05_fastANI/Sterm/analysis/analysis.out /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis_sterm.out

```

```{r}

library(readr)
analysis_FASTANI <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis_sterm.out",  "\t", escape_double = FALSE, col_names = c("genome1","genome2","ANI","comparabedWindows","totalWindows"), trim_ws = TRUE)

analysis_FASTANI$species1 <- str_split_fixed(analysis_FASTANI$genome1,"/",11)[,8]
analysis_FASTANI$species2 <- str_split_fixed(analysis_FASTANI$genome2,"/",11)[,8]

analysis_FASTANI$GENOME1 <- str_split_fixed(analysis_FASTANI$genome1,"/",10)[,10] %>% gsub(".fna","",.)
analysis_FASTANI$GENOME2 <- str_split_fixed(analysis_FASTANI$genome2,"/",10)[,10] %>% gsub(".fna","",.)


analysis_FASTANI$comparision <- ifelse(analysis_FASTANI$species1==analysis_FASTANI$species2,"same species","different species")
table(analysis_FASTANI$comparision)

analysis_FASTANI %>%group_by(comparision)  %>% select(ANI) %>%summarize(meansample=mean(ANI))

sample_relative_temp  %>% group_by(X.CHROM) %>% summarize(meansample=mean(Sequencing_depth))


analysis_FASTANI_final <- analysis_FASTANI %>% select(!c(genome1,genome2))
###----------------------
##whithin an between clusters
###----------------------

analysis_clusters <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv",  ",")

analysis_FASTANI_final02 <- merge(analysis_FASTANI_final,analysis_clusters,by.x="GENOME1",by.y="Taxon")
analysis_FASTANI_final03 <- merge(analysis_FASTANI_final02,analysis_clusters,by.x="GENOME2",by.y="Taxon")

analysis_FASTANI_final03$clustercomparison <- ifelse(analysis_FASTANI_final03$Cluster.x==analysis_FASTANI_final03$Cluster.y,"same cluster","different cluster")
analysis_FASTANI_final03$cov <- 100*(analysis_FASTANI_final03$comparabedWindows/analysis_FASTANI_final03$totalWindows)
ggplot(analysis_FASTANI_final03,aes(x=clustercomparison,y=ANI))+geom_boxplot()+theme_classic()
ggplot(analysis_FASTANI_final03,aes(x=cov,y=ANI,color=clustercomparison,fill=clustercomparison))+geom_point()+theme_classic()


```


##core genome nucleotide tree

Here, I creat a core genome nucleotide tree (inspired by Kirsten).
I also include the complete Reference genomes from NCBI. 

The following steps are incorporated:
1. pandaroo: find core genome
2. Extract gene sequences  corresponding to core gene famiilies into multi-fasta files:  2 files per gene-family, corresponding to nucleotide and protein sequences respectively. 
3. Align sequences for each family at protein level (resulting in one alignment file per core gene family) (MAFFT)
4. Back-translate to nucleotide alignments (respecting codon alignment) (Kirsten-script)
5. prune alignments (removing columns represented by less than 50% of the sequences) (Kirsten-script)
6. rename sequence-ids to genome-id only (Kirsten-script)
7. Concatenate pruned alignments (Kirsten-script)
8. infer the phylogeny (RAxML)

```{bash}
#get kirstens scripts
scp -r  /home/vincent/Downloads/aln_aa_to_dna vincent@130.223.49.145:/data/Project/2020_StarterCultureDiversity/99_log/


```


##### 1.Pandaroo

2. find core genomes
```{bash}
##==========================================
###-----------------bring all gffs together
##==========================================
species=Sterm

for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_GFF/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/
done

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/Out-sali.gff
##==========================================
###-----------------panaroo
##==========================================

##-----sterm

species=Sterm

conda activate panaroo

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species}_references/01_Panarooo/
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo

panaroo -t 37 -i //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//*.gff -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species}_references/01_Panarooo/ --clean-mode strict


##==========================================
###-----------------analysis panaroo
##==========================================


ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/GFF/ |grep ".gff" -c

cut -d ',' -f 4 /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|sort|uniq -c
awk -F "," '{if($4=="65" && $5=="65") print $0}' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|wc -l

##==========================================
###-----------------loop through single copy core genes and gather single copy core faa and ffn
##==========================================


roaryOUT=/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv

awk -F "," '{if($4=="65" && $5=="65") print $0}' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|wc -l

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG

for SCCG in $(awk -F "," '{if($4=="65" && $5=="65") print $1}' ${roaryOUT})
do
echo "================================================="
echo ${SCCG}
colzz=$(head -1 ${roaryOUT}  |tr ',' '\n' |wc -l)
for genomesss in $(seq 15 1 $colzz)
do

genesss=$(grep "^${SCCG}" ${roaryOUT} |cut -d ',' -f ${genomesss})
echo ${genesss}
samtools faidx /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/complete_Sterm_aminoAcid_gene.fasta ${genesss} >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG/${SCCG}.faa

samtools faidx /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/complete_Sterm_nucleotide_gene.fasta ${genesss} >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG/${SCCG}.ffn

done #genomesss
done #SCCG


```

```{bash}



mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/01_Panarooo//Sterm_references/

scp -r vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/01_Panarooo//Sterm_references/gene_presence_absence_roary.csv

scp -r vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/pan_genome_reference.fa /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/01_Panarooo//Sterm_references/



```

###### lineage specific genes

```{r}
library(tidyverse)

gene_presence_absence_roary <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/01_Panarooo/Sterm_references/gene_presence_absence_roary.csv")
analysis_clusters <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv",  ",", col_types = cols(Cluster = col_character(),  Taxon = col_character()))

dim(gene_presence_absence_roary)
gene_presence_absence_roary_long <- gene_presence_absence_roary[,c(1,15:ncol(gene_presence_absence_roary))] %>%gather(., sample, presence, colnames(gene_presence_absence_roary)[15]:colnames(gene_presence_absence_roary)[ncol(gene_presence_absence_roary)], factor_key=TRUE,na.rm = TRUE) %>% filter(!is.na(presence))
table(gene_presence_absence_roary_long$sample) %>% length()
gene_presence_absence_roary_long$sample <- as.character(gene_presence_absence_roary_long$sample)
lineageAbundance <- data.frame()
for (linss in unique(analysis_clusters$Cluster)) {
  
  print(linss)
  tmp <- analysis_clusters %>% filter(Cluster==linss) %>%  select(Taxon) %>% unlist() %>% as.character()
  tmp2 <- gene_presence_absence_roary_long %>% filter(sample %in% tmp) %>% select(Gene) %>% table() %>% data.frame()
  tmp3 <- data.frame(geneName=tmp2[,1],GenomesNum=tmp2[,2],lineage=as.character(linss))
  lineageAbundance <- rbind(lineageAbundance,tmp3)
  
  
}

lineageAbundance_wide <- spread(lineageAbundance, lineage, GenomesNum)  %>%replace(is.na(.), 0)

lineageAbundance$Presence_lin <- ifelse(lineageAbundance$GenomesNum>0,1,0)
lineageAbundance_presence_wide <- lineageAbundance %>% select(c(geneName,Presence_lin,lineage)) %>% spread(., lineage, Presence_lin)  %>%replace(is.na(.), 0)

lineageAbundance_presence_wide$NumberLineages <- rowSums(lineageAbundance_presence_wide[,-1])
table(lineageAbundance_presence_wide$NumberLineages)

lineageAbundance_presence_wide %>% filter(NumberLineages==1) %>%gather(., lineage, Presence_Abundance, colnames(lineageAbundance_presence_wide)[2]:colnames(lineageAbundance_presence_wide)[13], factor_key=TRUE,na.rm = TRUE) %>% filter(Presence_Abundance>0) %>% select(lineage) %>% table()

lineageAbundance_presence_unique <- lineageAbundance_presence_wide %>% filter(NumberLineages==1) %>%gather(., lineage, Presence_Abundance, colnames(lineageAbundance_presence_wide)[2]:colnames(lineageAbundance_presence_wide)[13], factor_key=TRUE,na.rm = TRUE) %>% filter(Presence_Abundance>0)

##how many cluster specific genes are there?
ggplot(lineageAbundance_presence_unique,aes(x=lineage))+geom_bar()+theme_classic()+labs(x="",y="number of cluster specific\ngenes")


##how many genome within this cluster contain the gene?
lineageAbundance_subset <- lineageAbundance %>% select(geneName,GenomesNum)
lineageAbundance_presence_count <- merge(lineageAbundance_presence_unique,lineageAbundance_subset,by="geneName")
table(lineageAbundance_presence_count$GenomesNum)



##---------------------------
#add eggnog information about the cluster-specific genes. 
##---------------------------

sterm_all_unique <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/01_Panarooo/Sterm_references/20210201_query_seqs.fa.emapper.annotations_all_pangenome",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description"),skip=4)
# RMK202_both_eggnog[RMK202_both_eggnog$query_name %in% SNPeff_final$finalName,]


COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA") %>% select(-divided)

unique(sterm_all_unique$COG_Functional_Category)[which(!unique(sterm_all_unique$COG_Functional_Category) %in% COG_functional_categories$short )]

# table(sterm_all_unique$COG_Functional_Category)

sterm_all_unique_extended <- merge(sterm_all_unique,COG_functional_categories,by.x = "COG_Functional_Category",by.y="short")


CogPlotSterm <- ggplot(sterm_all_unique_extended,aes(x=long))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10),
          legend.position = c(0.02, 0.98),
          legend.justification = c("left", "top")
          )
CogPlotSterm
#    svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/cogPlot.svg",width=7,height=7)
# CogPlotSterm 
#  dev.off()


##-------------------------
#add info
#with unnested
##-------------------------

lineageAbundance_presence_count_eggnog <- merge(lineageAbundance_presence_count,sterm_all_unique,by.x="geneName",by.y="query_name",all.x=TRUE)

ALL_sterm <- lineageAbundance_presence_count_eggnog
nrow(ALL_sterm)

ALL_sterm <- ALL_sterm %>% 
    mutate(COG_Functional_Category = strsplit(as.character(COG_Functional_Category), "")) %>% 
    unnest(COG_Functional_Category)


lineageAbundance_presence_count_eggnog_COG <- merge(ALL_sterm,COG_functional_categories,by.x = "COG_Functional_Category",by.y="short")

ggplot(lineageAbundance_presence_count_eggnog_COG,aes(x=lineage,color=long,fill=long))+geom_bar()+theme_classic()+labs(x="",y="number of cluster specific\ngenes")


table(lineageAbundance_presence_count_eggnog_COG$long) %>% as.data.frame() %>% arrange(desc(Freq))
table(lineageAbundance_presence_count_eggnog_COG$tax_scope)  %>% as.data.frame() %>% arrange(desc(Freq))

tmp <- lineageAbundance_presence_count_eggnog_COG %>% filter(long=="Cell wall and membrane biogenesis")

tmp <- lineageAbundance_presence_count_eggnog_COG %>% filter(long=="Function Unknown")
tmp <- lineageAbundance_presence_count_eggnog_COG %>% filter(long=="Replication and repair")


write.table(lineageAbundance_presence_count_eggnog_COG,"/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/01_Panarooo/Sterm_references//CLUSTERspecificGENES_eggnog",sep="\t",quote = FALSE,row.names = FALSE)

###---------------------------------------
#group the genes appriopiately
##following catagroeies
#1. phage derived (do phaster and see which genes are associated to phage)
#2. all EPS/Rhamnos genes (group into phage receptor modification)
#3. add plasmid to phages
#4. bacertiocin
#interesting genes (mutL, mutS)
#new phage defense mechanisms
# conjugative elements, is-elements, transposase


###---------------------------------------

####---------01.glycosyl EPS related
lineageAbundance_presence_count_eggnog_COG%>% filter(grepl('glycosyl',`eggNOG free text description`,ignore.case = TRUE)|grepl('rham',`eggNOG free text description`,ignore.case = TRUE)|grepl('sugar transfera',`eggNOG free text description`,ignore.case = TRUE)|grepl('sugar transfera',`eggNOG free text description`,ignore.case = TRUE)|grepl('epimerase',`eggNOG free text description`,ignore.case = TRUE)) 

####---------02. plasmids, tranposons, is elements
lineageAbundance_presence_count_eggnog_COG%>% filter(grepl('plasmid',`eggNOG free text description`,ignore.case = TRUE)|grepl('Transposase',`eggNOG free text description`,ignore.case = TRUE)|grepl('family IS',`eggNOG free text description`,ignore.case = TRUE)|grepl('mobilisation',`eggNOG free text description`,ignore.case = TRUE))

####---------03. phage
lineageAbundance_presence_count_eggnog_COG%>% filter(grepl('phage',`eggNOG free text description`,ignore.case = TRUE)|grepl('integra',`eggNOG free text description`,ignore.case = TRUE)|grepl('viruses',`tax_scope`,ignore.case = TRUE)|grepl('reverse transcriptase',`eggNOG free text description`,ignore.case = TRUE))

####---------04. phage-defense
lineageAbundance_presence_count_eggnog_COG%>% filter(grepl('restriction-modification',`eggNOG free text description`,ignore.case = TRUE)|grepl('crispr',`eggNOG free text description`,ignore.case = TRUE)|grepl('endonuclease',`eggNOG free text description`,ignore.case = TRUE))

####---------05. bacteria-bacteria interaction
lineageAbundance_presence_count_eggnog_COG%>% filter(grepl('Bacteriocin',`eggNOG free text description`,ignore.case = TRUE)|grepl('antitoxin',`eggNOG free text description`,ignore.case = TRUE))

# lineageAbundance_presence_count_eggnog_COG %>% filter(grep("Glycosyl",ignore.case = TRUE))

##!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
##maybe we can show that many of the cluster-specific genes are associated to a plasmids, phages, conjugative elements as vectore
##!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!


lineageAbundance_presence_count_genes_location <- merge(lineageAbundance_presence_count,gene_presence_absence_roary_long,by.x = "geneName",by.y ="Gene",all.x=TRUE)


lineageAbundance_presence_count_genes_location <- lineageAbundance_presence_count_genes_location %>% 
    mutate(presence = strsplit(as.character(presence), ";")) %>% 
    unnest(presence) %>%  mutate(presence_location = strsplit(as.character(presence), "_")%>% map_chr(., 2))

lineageAbundance_presence_count_genes_location$presence_location <- as.numeric(lineageAbundance_presence_count_genes_location$presence_location)

lineageAbundance_presence_count_genes_location <- lineageAbundance_presence_count_genes_location %>% filter(!is.na(presence_location))

max(lineageAbundance_presence_count_genes_location$presence_location)


ggplot(lineageAbundance_presence_count_genes_location,aes(x=presence_location,group=sample,color=sample,fill=sample))+geom_density(alpha=0.4)+facet_wrap(~lineage,scales="free_y")+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))


##In ONT genomes

lineageAbundance_presence_count_genes_location %>% select(geneName) %>% unique() %>% nrow()
lineageAbundance_presence_count_genes_location  %>% filter(sample %in% c("24750","24739","13496","24738","13499","24730","24726","24728","13498","24734","24737","24853","24854","24855","24742","13494","24735","24736","24724","24748","24748","24740")) %>% select(geneName) %>% unique() %>% nrow()
# lineageAbundance_presence_count_genes_location  %>% filter(!sample %in% c("24750","24739","13496","24738","13499","24730","24726","24728","13498","24734","24737","24853","24854","24855","24742","13494","24735","24736","24724","24748","24748","24740")) %>% select(geneName) %>% unique() %>% nrow()

# lineageAbundance_presence_count_genes_location %>% filter(sample %in% c("24750","24739","13496","24738","13499","24730","24726","24728","13498","24734","24737","24853","24854","24855","24742","13494","24735","24736","24724","24748","24748","24740")) %>% select(GenomesNum) %>% table()
# lineageAbundance_presence_count_genes_location %>% filter(!sample %in% c("24750","24739","13496","24738","13499","24730","24726","24728","13498","24734","24737","24853","24854","24855","24742","13494","24735","24736","24724","24748","24748","24740")) %>% select(GenomesNum) %>% table()


lineageAbundance_presence_count_genes_location_ONT <- lineageAbundance_presence_count_genes_location %>% filter(sample %in% c("24750","24739","13496","24738","13499","24730","24726","24728","13498","24734","24737","24853","24854","24855","24742","13494","24735","24736","24724","24748","24748","24740"))

ggplot(lineageAbundance_presence_count_genes_location_ONT,aes(x=presence_location,group=sample,color=sample,fill=sample))+geom_density(alpha=0.4)+facet_wrap(~lineage,scales="free_y")+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))

lineageAbundance_presence_count_genes_location_ONT$lineageNUM <- as.numeric(lineageAbundance_presence_count_genes_location_ONT$lineage)


pointsPlot <- ggplot(lineageAbundance_presence_count_genes_location_ONT,aes(x=presence_location,y=lineageNUM,group=sample,color=sample,fill=sample))+geom_point()+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))
pointsPlot

###--------------------add information (e.g. COg or description to identify flanking transposons)

 lineageAbundance_presence_count_eggnog_COG_sub <- lineageAbundance_presence_count_eggnog_COG %>% select(c(geneName,`eggNOG free text description`, long,tax_scope))

lineageAbundance_presence_count_genes_location_ONT_EGGNOG <- merge( lineageAbundance_presence_count_genes_location_ONT,lineageAbundance_presence_count_eggnog_COG_sub,by="geneName")



pointsPlot <- ggplot(lineageAbundance_presence_count_genes_location_ONT_EGGNOG,aes(x=presence_location,y=lineageNUM,group=sample,color=sample,fill=sample,text =paste("COG:", long,"\neggog:",`eggNOG free text description`)))+geom_point()+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))
pointsPlot

 library(plotly)
ggp <- ggplotly(pointsPlot)
 ggp
 
 
 
 
pointsPlot <- ggplot(lineageAbundance_presence_count_genes_location_ONT_EGGNOG,aes(x=presence_location,y=sample,group=sample,color=sample,fill=sample,text =paste("COG:", long,"\neggog:",`eggNOG free text description`)))+facet_wrap(~lineageNUM,scales="free_y")+geom_point()+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))
pointsPlot

 library(plotly)
ggp <- ggplotly(pointsPlot)
 ggp
 
 ###--------------------------------------
 ##see potential phage, plasmid or transposon associated genes
  ###--------------------------------------
 
 
 
lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended <- lineageAbundance_presence_count_genes_location_ONT_EGGNOG%>% mutate(associated= ifelse(grepl('plasmid',`eggNOG free text description`,ignore.case = TRUE)|grepl('Transposase',`eggNOG free text description`,ignore.case = TRUE)|grepl('family IS',`eggNOG free text description`,ignore.case = TRUE)|grepl('mobilisation',`eggNOG free text description`,ignore.case = TRUE)|grepl('phage',`eggNOG free text description`,ignore.case = TRUE)|grepl('integra',`eggNOG free text description`,ignore.case = TRUE)|grepl('viruses',`tax_scope`,ignore.case = TRUE)|grepl('reverse transcriptase',`eggNOG free text description`,ignore.case = TRUE),"mechanisms known","unknown"))

 
pointsPlot <- ggplot(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended,aes(x=presence_location,y=sample,group=sample,color=associated,fill=associated,text =paste("COG:", long,"\neggog:",`eggNOG free text description`)))+facet_wrap(~lineageNUM,scales="free_y")+geom_jitter(width = 0, height = 0.3,alpha=0.8,)+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))
pointsPlot

 library(plotly)
ggp <- ggplotly(pointsPlot)
 ggp

  ###--------------------------------------
 ##identify gene clusters
 #so consecutive genes
  ###--------------------------------------
 CutOFFnumGENESS=15
 lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added <- data.frame()
 
 for (samplesss in unique(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended$sample)){
   
   tmp_01 <- lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended %>% filter(sample==samplesss) %>% arrange(presence_location)
   ClusterNUmber <- as.numeric(1)
 
   # for (genessss in unique(tmp_01$geneName)){
   for (genessss in 1:nrow(tmp_01)){

   tmp_02 <- tmp_01[genessss,] %>% arrange(presence_location)

   
  Gene_loc <-  tmp_01[genessss,"presence_location"]
  NextGene_loc <-  tmp_01[genessss+1,"presence_location"]
  Gene_div <-  NextGene_loc-Gene_loc
  if (!is.na(Gene_div)&&Gene_div<=CutOFFnumGENESS) {
    tmp3 <- cbind(tmp_02,Cluster=ClusterNUmber,NextGene=Gene_div)
    lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added <- rbind(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added,tmp3)
  } else if(!is.na(Gene_div)&&Gene_div>CutOFFnumGENESS) {
     tmp3 <- cbind(tmp_02,Cluster=ClusterNUmber,NextGene=Gene_div)
    lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added <- rbind(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added,tmp3)
    ClusterNUmber=as.numeric(ClusterNUmber+1)
   } else if(is.na(Gene_div)){
     tmp3 <- cbind(tmp_02,Cluster=ClusterNUmber,NextGene=Gene_div)
    lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added <- rbind(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added,tmp3)
     
   }
   
 }
 }
 
 table(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$NextGene)
  table(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$sample,lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Cluster)
lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Cluster <- as.character(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Cluster)
##color according to cluster
  

pointsPlot <- ggplot(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added,aes(x=presence_location,y=sample,group=Cluster,color=Cluster,fill=Cluster,text =paste("COG:", long,"\neggog:",`eggNOG free text description`)))+facet_wrap(~lineageNUM,scales="free_y")+geom_jitter(width = 0, height = 0.3,alpha=1,size=1.5)+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))
pointsPlot

 library(plotly)
ggp <- ggplotly(pointsPlot)
 ggp  
 

 lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Clustername <- paste0(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$sample,"_cluster_",lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Cluster)
  
 analysis_clusters <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv",  ",",  col_types = cols(Taxon = col_character(),Cluster = col_character()))

 
 lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added_cluster <- table(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Clustername) %>% as.data.frame() %>%  mutate(sample = strsplit(as.character(Var1), "_")%>% map_chr(., 1))
 
 lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added_cluster <- merge(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added_cluster,analysis_clusters,by.x="sample",by.y="Taxon")
 
 ggplot(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added_cluster,aes(x=Var1,y=Freq))+geom_bar(stat="identity")+theme_classic()+facet_wrap(~Cluster,scales="free_x")
 
##--------------------------------------
### extract cluster extend
##--------------------------------------
extension=2
 ClusterDescription <- data.frame()
  for (clustersss in unique(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Clustername)) {
    
    tmp1 <- lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added %>% filter(Clustername==clustersss)
    tmp2 <- data.frame(clusterName=unique(tmp1$Clustername),Genome=unique(tmp1$sample),Lineage=unique(tmp1$lineage),NumGenesLineage=length(unique(tmp1$presence)),NumGenesTotal=max(tmp1$presence_location)-min(tmp1$presence_location)+1,startGene=min(tmp1$presence_location),endGene=max(tmp1$presence_location),startExt=min(tmp1$presence_location)-extension,endExt=max(tmp1$presence_location)+extension)
    ClusterDescription <- rbind(ClusterDescription,tmp2)
  }
 
 ##reformat the gene-names
 
ClusterDescription$startGene <- paste0(ClusterDescription$Genome,"_",formatC(ClusterDescription$startGene, width = 5, format = "d", flag = "0"))
ClusterDescription$endGene <- paste0(ClusterDescription$Genome,"_",formatC(ClusterDescription$endGene, width = 5, format = "d", flag = "0"))
ClusterDescription$startExt <- paste0(ClusterDescription$Genome,"_",formatC(ClusterDescription$startExt, width = 5, format = "d", flag = "0"))
ClusterDescription$endExt <- paste0(ClusterDescription$Genome,"_",formatC(ClusterDescription$endExt, width = 5, format = "d", flag = "0"))

write.table(ClusterDescription,"/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/cluster_lineageSpecific.csv",sep="\t",quote = FALSE,row.names = FALSE)


```

Here, I have identified clusters of lineage specific genes. Some of these clusters have phage related genes in them. 
We further identified and extracted cluster regions (+/- 2genes=maybe flanking transposons). 

Next I want to:


1. see which clusters are shared amongst strains (with cdHit)
2. analyse potential mechanisms of transposition (identify: prophage, plasmid, transposon)


```{bash}
scp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/cluster_lineageSpecific.csv vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/


```

```{bash}
##==================================================
##1. see which clusters are shared amongst strains (with cdHit)
##==================================================
clustersss=24855_cluster_1

###First extract fasta of clusters
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.bed
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.fasta
for clustersss in $(cut -f 1 /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/cluster_lineageSpecific.csv |sed '1d')
do
echo ${clustersss}

Genome=$(echo ${clustersss} |cut -d '_' -f 1)
startGEne=$(grep "${clustersss}$(printf '\t')" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/cluster_lineageSpecific.csv |cut -f 6)
EndGene=$(grep "${clustersss}$(printf '\t')" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/cluster_lineageSpecific.csv |cut -f 7)




contigName=$(grep "ID=${startGEne}_gene" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//${Genome}.gff |cut -f 1)
startName=$(grep "ID=${startGEne}_gene" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//${Genome}.gff |cut -f 4)
endName=$(grep "ID=${EndGene}_gene" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//${Genome}.gff |cut -f 5)

echo -e ${contigName}"\t"${startName}"\t"${endName}"\t"${clustersss} > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.bed
cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.bed >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.bed


##---------------get fasta

bedtools getfasta -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/${Genome}.fna \
  -bed /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.bed -name \
  -fo /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.fasta

cat  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.fasta >>  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.fasta

rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.fasta
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.bed
done

sed -i "s/\:.*$//g"  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.fasta
seq_length.py /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.fasta


### cluster with cd-hit-est


cd-hit-est -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.fasta \
    -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_cd_hit -c 0.98 -M 30000 -T 37 -s 0.8 -aS 0.8 -sc 1 -sf 1
    
grep ">Cluster" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_cd_hit.clstr |sed 's/ /_/g' >  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_cd_hit.log
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecific_clustering.txt
for clustersss in $(cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_cd_hit.log)
do
#aryName=$(grep "${clustersss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log |cut -f 2)
ClusterNumber=$(echo $clustersss|awk -F "_" '{print $2}')
EndCluster=$((ClusterNumber+1))

sed -n "/>Cluster $ClusterNumber$/,/>Cluster $EndCluster$/p" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_cd_hit.clstr | grep -v "^>Cluster" |awk -F "[>...]" -v numbersss="$EndCluster" '{OFS="\t"}{print "lineage cluster "numbersss,$2}' >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecific_clustering.txt

#for samplesss in $(cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt)
num=1

done

```

```{bash}
scp  vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecific_clustering.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/

```

```{r}

lineageSpecific_clustering <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/lineageSpecific_clustering.txt",  "\t", escape_double = FALSE, col_names = c("ClusterName_overall","clusterName"),  trim_ws = TRUE)

ClusterDescription_sub <- ClusterDescription %>% select(-c(startGene,endGene,startExt,endExt))

ClusterDescription_wide <- merge(ClusterDescription_sub,lineageSpecific_clustering,by="clusterName")
repeated_clusters <- data.frame()
for (variablesss in unique(ClusterDescription_wide$ClusterName_overall)) {
  
  tmp1 <- ClusterDescription_wide %>% filter(ClusterName_overall==variablesss)
  tmp2 <- ifelse(nrow(tmp1)!=length(unique(tmp1$Genome)),"repeated","not repeated")
  tmp4 <- max(tmp1$NumGenesLineage)
  tmp3 <- data.frame(ClusterName_overall=variablesss,repeatStructure=tmp2,maxNumGenesCluster=tmp4)
  repeated_clusters <- rbind(repeated_clusters,tmp3)
}


cutoffNumberGENES<-3

ClusterDescription_wider <- merge(ClusterDescription_wide,repeated_clusters,by="ClusterName_overall") %>%  mutate(Lineage =as.character(Lineage))
nums <- length(unique(ClusterDescription_wider[ClusterDescription_wider$maxNumGenesCluster<=cutoffNumberGENES,"ClusterName_overall"]))

ClusterDescription_wider[ClusterDescription_wider$maxNumGenesCluster<=cutoffNumberGENES,"ClusterName_overall"] <- paste0("small clusters (n=",nums,")")

ClusterDescription_wider[ClusterDescription_wider$maxNumGenesCluster<=cutoffNumberGENES,"Lineage"] <- paste0("small clusters (n=",nums,")")

# ggplot(ClusterDescription_wider_sub,aes(x=ClusterName_overall,y=NumGenesLineage,fill=Lineage))+geom_boxplot()+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=10))


ClusterDescription_summs <- ClusterDescription_wider  %>% group_by(ClusterName_overall,Lineage) %>%    dplyr::summarize(mean = mean(NumGenesLineage),sd=sd(NumGenesLineage))


ggplot(ClusterDescription_summs, aes(x=reorder(ClusterName_overall, -mean), y=mean, fill=Lineage)) +   
  geom_bar(stat="identity", color="black",  position=position_dodge()) +theme_classic() +
  # geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.2,position=position_dodge(.9)) +
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=10))+
  labs(x="",y="number of genes in\nlinegae specific cluster")
  

ClusterDescription_wider %>% filter(ClusterName_overall!="small clusters (n=37)")

# lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added

ClusterDescription_wider_ext <- merge(ClusterDescription_wide,repeated_clusters,by="ClusterName_overall") %>%  mutate(Lineage =as.character(Lineage))

nums <- length(unique(ClusterDescription_wider_ext[ClusterDescription_wider_ext$maxNumGenesCluster>cutoffNumberGENES,"ClusterName_overall"]))
ClusterDescription_wider_ext[ClusterDescription_wider_ext$maxNumGenesCluster>cutoffNumberGENES,"ClusterName_overall"] <- paste0("large clusters (n=",nums,")")
nums <- length(unique(ClusterDescription_wider_ext[ClusterDescription_wider_ext$maxNumGenesCluster<=cutoffNumberGENES,"ClusterName_overall"]))
ClusterDescription_wider_ext[ClusterDescription_wider_ext$maxNumGenesCluster<=cutoffNumberGENES,"ClusterName_overall"] <- paste0("small clusters (n=",nums,")")




tot_obs <- sum(ClusterDescription_wider_ext$NumGenesLineage)
ClusterDescription_wider_ext %>%   mutate(perc = n / nrow(tips)) 

ggplot(ClusterDescription_wider_ext, aes(fill=ClusterName_overall, y=NumGenesLineage, x=ClusterName_overall)) + 
    geom_bar(position="stack", stat="identity")+theme_classic()+labs(x="",y="number of lineage specif genes")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=10),legend.position = "none")+
  scale_y_continuous(sec.axis = sec_axis(trans = ~./tot_obs, labels = percent, name = "percent")) 


ClusterDescription_wider_ext <- merge(ClusterDescription_wide,repeated_clusters,by="ClusterName_overall") %>%  mutate(Lineage =as.character(Lineage))

numslarge <- length(unique(ClusterDescription_wider_ext[ClusterDescription_wider_ext$maxNumGenesCluster>cutoffNumberGENES,"ClusterName_overall"]))
numssmall <- length(unique(ClusterDescription_wider_ext[ClusterDescription_wider_ext$maxNumGenesCluster<=cutoffNumberGENES,"ClusterName_overall"]))

ClusterDescription_wider_ext$grouping <- ifelse(ClusterDescription_wider_ext$maxNumGenesCluster<=cutoffNumberGENES,paste0("small clusters (n=",numssmall,")"),paste0("large clusters (n=",numslarge,")"))

##---------------------
###-------------------------what genes are contained within small clusters
##---------------------

ClusterDescription_wider_ext_EGGNOG <- merge(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added,ClusterDescription_wider_ext,by.x="Clustername",by.y="clusterName") %>% filter(!is.na(long))

table(ClusterDescription_wider_ext_EGGNOG$grouping)
ClusterDescription_wider_ext_EGGNOG_small <- ClusterDescription_wider_ext_EGGNOG %>% filter(grouping=="small clusters (n=37)")


ggplot(ClusterDescription_wider_ext_EGGNOG_small,aes(x=forcats::fct_infreq(long)))+geom_histogram(stat = "count")+theme_classic()+
  labs(x="",y="gene count")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=10))

write.table(ClusterDescription_wider_ext,"~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun//clusters_lineage_grouping.txt",sep="\t",quote = FALSE,row.names = FALSE)


```
keine klare Gene ersichtlich 

schauen was für functionen die restlichen clusters haben könnten. 


```{bash}
scp ~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun//clusters_lineage_grouping.txt vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/

```

2. analyse potential mechanisms of transposition (identify: prophage, plasmid, transposon)
herefore I focus only on the larger clusters (more the 3 lineage specific genes). I take two up and downstream genes and look for the presence of Is-elements or phage genes. 

```{bash}

##==================================================
##1. extract only larger clusters
##==================================================

clustersss=24735_cluster_12
species=Sterm
rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/
for clustersss in $(awk -F "\t" '{OFS="\t"}{if($9~"large clusters") print $2 }' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/clusters_lineage_grouping.txt)
do
echo $clustersss

Genome=$(echo ${clustersss} |cut -d '_' -f 1)
startGEne=$(awk -F "\t" -v clusterzzz="$clustersss" '{OFS="\t"}{if($1==clusterzzz) print $8 }' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/cluster_lineageSpecific.csv)
EndGene=$(awk -F "\t" -v clusterzzz="$clustersss" '{OFS="\t"}{if($1==clusterzzz) print $9 }' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/cluster_lineageSpecific.csv)

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//${Genome}.gff |grep -A 1000 "ID=${startGEne};" |grep -B 1000 "ID=${EndGene};" > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/${clustersss}.gff

#awk -F "\t" '{if($3=="CDS")print $0}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//${Genome}.gff |grep -A 1000 "ID=${startGEne};" |grep -B 1000 "ID=${EndGene};" > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/${clustersss}.gff



done


grep -i "ISfinder" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/*.gff
grep -i "ISfinder" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/*.gff|cut -d ':' -f 1 |sort|uniq -c
grep -i "ISfinder" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/*.gff|cut -d ':' -f 1 |sort|uniq -c |wc -l

grep "integr" -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/*.gff
grep "phage" -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/*.gff

```

###### Gene rarefaction

```{r}

library(tidyverse)

gene_presence_absence_roary <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/01_Panarooo/Sterm_references/gene_presence_absence_roary.csv")
analysis_clusters <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv",  ",", col_types = cols(Cluster = col_character(),  Taxon = col_character()))


###----------------------------------
##over all
###----------------------------------

library(micropan)
typeof(xmpl.panmat)
##creat a clustering matrix
gene_presence_absence_roary_preped <- gene_presence_absence_roary[,-c(2:14)] %>% remove_rownames()%>% column_to_rownames(., var = "Gene") %>% t() 
# dim(gene_presence_absence_roary_preped)
# gene_presence_absence_roary_preped_na <- is.na(gene_presence_absence_roary_preped)
# sum(!is.na(gene_presence_absence_roary_preped))

gene_presence_absence_roary_prep_final <- as.data.frame(!is.na(gene_presence_absence_roary_preped))
# dim(gene_presence_absence_roary_prep_final)
# gene_presence_absence_roary_prep_final[1,1]
# gene_presence_absence_roary_prep_final[2,2]
# str(gene_presence_absence_roary_prep_final)
gene_presence_absence_roary_prep_final <- as.data.frame(lapply(gene_presence_absence_roary_prep_final,as.numeric))
# str(gene_presence_absence_roary_prep_final)
# 
# gene_presence_absence_roary_preped[1,1]
# gene_presence_absence_roary_prep_final[1,1]
# gene_presence_absence_roary_prep_final[2,2]
rar.tbl <- rarefaction(gene_presence_absence_roary_prep_final, n.perm = 300)
data_genome_all <- rar.tbl %>%   gather(key = "Permutation", value = "Clusters", -Genome) %>% add_column(lineage="all_genomes")

# ggplot(data_genome,aes(x = Genome, y = Clusters))+
     # stat_smooth(method = "loess", formula = y ~ x, size = 1,se=TRUE)
# data_genome_all_cleaned <- data_genome_all %>% filter(!Genome %in% c(1,2,3,4,5))


# m<-nls(data_genome_all_cleaned$Clusters~a*data_genome_all_cleaned$Genome/(b+data_genome_all_cleaned$Genome))
# m

Rarefaction_all <- ggplot(data_genome_all,aes(x = Genome, y = Clusters))+
     geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c( F = 10,s=2000 )), se=FALSE)+theme_classic()+
    labs(x="Genomes",y="Genes")
Rarefaction_all

###----------------------------------
##lineages
###----------------------------------


rarefaction_lineages <- data.frame()
for (linss in unique(analysis_clusters$Cluster)) {
  
  print(linss)
  if (analysis_clusters[which(analysis_clusters$Cluster==linss),"Taxon"] %>% nrow()>1 ) {
    
    
  filtering <- analysis_clusters[which(analysis_clusters$Cluster==linss),"Taxon"] %>% unlist() %>% as.character()
length(filtering)
gene_presence_absence_roary_preped_lin1 <- as.data.frame(gene_presence_absence_roary_preped[rownames(gene_presence_absence_roary_preped)%in% filtering,])
dim(gene_presence_absence_roary_preped_lin1)

gene_presence_absence_roary_preped_lin1_final <- as.data.frame(!is.na(gene_presence_absence_roary_preped_lin1))
dim(gene_presence_absence_roary_preped_lin1_final)
gene_presence_absence_roary_preped_lin1_final <- as.data.frame(lapply(gene_presence_absence_roary_preped_lin1_final,as.numeric))
dim(gene_presence_absence_roary_preped_lin1_final)
rar.tbl_lin <- rarefaction(gene_presence_absence_roary_preped_lin1_final, n.perm = 50)
data_genome <- rar.tbl_lin %>%   gather(key = "Permutation", value = "Clusters", -Genome) %>% add_column(lineage=as.character(linss))

rarefaction_lineages <- rbind(rarefaction_lineages,data_genome)
    
  } else { }
  
}

g0<-ggplot(data=rarefaction_lineages,aes(x=Genome,y=Clusters,group=lineage))
# g1<-g0+geom_point()
g2<-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE)+theme_classic()+ labs(x="Genomes",y="Genes")+ coord_cartesian(ylim=c(1500, 2100))
g2


g2<-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE)+lims(y=c(1500,2200))
g2


g2<-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE)+coord_trans(x="log2")
g2
###----------------------------------
##all together
###----------------------------------


rarefaction_alltogether <- rbind(rarefaction_lineages,data_genome_all)
g0<-ggplot(data=rarefaction_alltogether,aes(x=Genome,y=Clusters,group=lineage))
# g1<-g0+geom_point()
g2<-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE)+theme_classic()+ labs(x="Genomes",y="Genes")
g2

 library(plotly)
ggp <- ggplotly(g2)
 ggp
 
```



##### 1.orthofinder

```{bash}
###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Sterm

for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

#leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
done

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/Out-sali.faa
###-----------------------------------------
##orthorfinder
###-----------------------------------------
#for species in $(echo "Ldel Sterm"|tail -1)
for species in $(echo "Sterm"|tail -1)
do

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/


    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}// \
      -t 35 \
      -o /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ \
      -a 35

done # species

```

##### 1.Kirstin's approach


```{bash}
date=Results_Feb01
species=Sterm
#/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/Results_Jul28/
for species in $(echo "Ldel Sterm"|tail -1)
do
echo "==========================================="
echo -e ${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/get_singlecp_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt

wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt
wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt

done
```

Here, I make a directory containing the fna and faa files of all genomes with the file name corresponding to the locus tag. 

```{bash}


for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA/
done

###--------------------
for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/ |grep ".faa$"|sed 's/.faa//g'|grep -v "Out-sali")
do
locusTagsss=$(head -1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/${genomesss}.faa|cut -d '_' -f 1|sed 's/>//g')
echo -e "This  genome : " ${genomesss} " has the following locus Tag : "${locusTagsss}

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/${genomesss}.faa > \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/${genomesss}.ffn > \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.ffn


done
done

##------=================================================================
#extract orthogroups
##------=================================================================

for species in $(echo "Ldel Sterm"|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/extract_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt --folder ${species}

done
```

align protein families

```{bash}


for species in $(echo "Ldel Sterm"|tail -1)
do

cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep ".faa$" |sed 's/.faa//g'|grep -v "Out-sali" )
do
rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.aln.fasta
mafft --thread 37 --maxiterate 1000  --auto /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.faa > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}_aln.fasta

done 

done #species
```
here I back translate the faa alignment to dna.

```{bash}

for species in $(echo "Ldel Sterm"|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/aln_aa_to_dna.pl

done

```

With perl: prune alignments (removing columns represented by less than 50% of the sequences)


```{bash}


for species in $(echo "Ldel Sterm"|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/prune_aln.pl

done

##================================================================
#remove all infomration (except genome info) from multifasta header
##================================================================

for species in $(echo "Ldel Sterm"|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i "s/_0.*//" ${genesss}_aln_prune.fasta 

done
done

##================================================================
#shorten header
##================================================================
species=Ldel
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/L-delbrueckii-//g' ${genesss}_aln_prune.fasta 

done



species=Sterm
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/-CIRM-//g' ${genesss}_aln_prune.fasta 

done

species=Sterm
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/-BM-A0/-BM-A/g' ${genesss}_aln_prune.fasta 

done


```
With perl: Concatenate pruned alignments
```{bash}
  
#  for species in $(echo "Ldel Sterm"|tail -1)
  for species in $(echo "Sterm"|tail -1)
  do
  echo -e ${species}
  cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
  perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/cat_align.pl > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}.phylip
  
  done
```
With RAxML: infer the phylogeny
```{bash}
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/

#  for species in $(echo "Ldel Sterm"|tail -1)
   for species in $(echo "Sterm"|tail -1)
  do
  echo -e ${species}
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/

rm -r *${species}_all*

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}.phylip -n ${species}_all -T 37
done
```

move local

```{bash}

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new/

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new//
```

### ggtree

preperation

```{bash}
###-------------------------
#get RMK genomes and REF genomes
###-------------------------
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/

species=Sterm

leterzzz=$(echo $species |head -c 1)

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/ |grep "Out" -v | sed 's/.faa//g' |awk -F "\t" '{OFS="\t"}{print $1, "NCBI_refs"}' >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FAA/ |grep "^${leterzzz}" |grep ".faa$"|cut -d '_' -f 4|sed 's/.faa//g' |awk -F "\t" '{OFS="\t"}{print $1, "RMK_genomes"}' >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/ |grep "Out-ves" | sed 's/.faa//g' |awk -F "\t" '{OFS="\t"}{print $1, "outgroup"}' >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt


sed -i 's/-BM-A0/-BM-A/g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt
sed -i 's/S-KLDS-3.1/S-KLDS-3/g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt

grep "S-KLDS-3" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt
```

move local
```{bash}
mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new//logs/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new//logs/
```



do in R
```{r}
library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated", tree.names = NULL, force.multi = FALSE)


# x <- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
ggtree(Sterm_tree_raw, layout="circular") +
  geom_tiplab(aes(angle=angle),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  
##--------------------------------
##color labels of of only rmk phages
##--------------------------------
Sterm_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/logs/Sterm_genome_origin.txt",  "\t", escape_double = FALSE, col_names = c("label","origin"),  trim_ws = TRUE) %>% as.data.frame()

# tmp <- full_join(as_tibble(Sterm_tree_raw), Sterm_genome_origin, by='label')
# table(tmp$origin)
tree <- full_join(as_tibble(Sterm_tree_raw), Sterm_genome_origin, by='label') %>% as.treedata()


p <- ggtree(tree, layout="circular") +
  geom_tiplab(aes(angle=angle,color=origin),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) + #NO ROOT  
  scale_color_manual(values=c("grey66", 'grey55', 'black')) 
p
##--------------------------------
##all info at once (geography and origin)
##--------------------------------
all_sterm_strain_info <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited.csv", "\t", escape_double = FALSE, trim_ws = TRUE) %>% select(c(shortName, strain, origin, location_continent))

all_sterm_strain_info$strain <- gsub("S-thermophilus-","",all_sterm_strain_info$strain)

##-------------
#change tiplabel and color according to rmk or non-rmk sample
##-------------

tmp = as.character(all_sterm_strain_info$strain)
names(tmp) <- all_sterm_strain_info$shortName

# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated", tree.names = NULL, force.multi = FALSE)



Sterm_tree_raw_tible <- as_tibble(tree) 
Sterm_tree_raw_tible$label

Sterm_tree_raw_tible$label <- plyr::revalue(Sterm_tree_raw_tible$label,tmp)
tree_02 <- Sterm_tree_raw_tible%>% as.treedata()




p2 <- ggtree(tree_02, layout="circular") +
  geom_tiplab(aes(angle=angle,color=origin),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) + #NO ROOT  
  scale_color_manual(values=c("grey66", 'grey55', 'black')) 
p2

##-------------
#add geograph and origin
##-------------
all_sterm_strain_info$shortName <- as.character(all_sterm_strain_info$shortName)
all_sterm_strain_info_ready <- all_sterm_strain_info %>% select(-shortName) %>% remove_rownames()%>% column_to_rownames(., var = "strain")%>%replace(is.na(.), "Unknown") 
table(all_sterm_strain_info_ready$location_continent)
 all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c("Europe",  "Asia", "USA", "Australia" , "Unknown","Outgroup"))
# colors_area <- c("#464D77","#36827F","#F9DB6D","grey88")
# colors_area <- c("#F9DB6D","#36827F","#464D77","grey88")

# location_dataset$area

# tree_02 <- full_join(as_tibble(tree), location_dataset, by='label') %>% as.treedata()

colnames(all_sterm_strain_info_ready) <- c("A","B")
rownames(all_sterm_strain_info_ready)

p3 <- gheatmap(p2, all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
p3



  # svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/CoverageSpecies.svg",width=5,height=5)
   png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.png", width = 3000, height = 3000,res=300)

p2

dev.off()

##---------------------
#prune tree
##---------------------

all_sterm_strain_info_keepers <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited.csv", "\t", escape_double = FALSE, trim_ws = TRUE) %>% select(c( strain, keep)) %>% filter( keep=="yes") %>% select(strain) %>% unlist() %>% as.vector

all_sterm_strain_info_keepers <- gsub("S-thermophilus-","",all_sterm_strain_info_keepers)

#But get the tips to keep
# keeptips <- fulltree$tip.label[!fulltree$tip.label %in% prunetips]

#Group the tips to keep
# prunetree <- groupOTU(tree_02, focus=all_sterm_strain_info_keepers)

#And plot
ggtree(prunetree, layout="fan", aes(color=group))+
scale_color_manual(values=c("lightgrey","black"))+
geom_tiplab()

# prunetree <- drop.tip(tree_02,node=5)


```
### make high resolution plot

FIRST orthofinder of the necessary genomes
```{bash}
###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Sterm

outgroup=$(echo "S-PNGR-Z-K")

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/${outgroup}.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/

###-----------------------------------------
##orthorfinder
###-----------------------------------------
#for species in $(echo "Ldel Sterm"|tail -1)


rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/


    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}// \
      -t 35 \
      -o /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/ \
      -a 35

```



```{bash}
date=Results_Feb03
species=Sterm
#/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/Results_Jul28/

echo "==========================================="
echo -e ${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/get_singlecp_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/${date}/Orthogroups/Orthogroups.txt > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt

wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt
wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/${date}/Orthogroups/Orthogroups.txt

```

Here, I make a directory containing the fna and faa files of all genomes with the file name corresponding to the locus tag. 

```{bash}


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FFN/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FFN/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FFN/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FFN/${species}/

#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA/

###--------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/ |grep ".faa$"|sed 's/.faa//g'|grep -v "Out-sali")
do
locusTagsss=$(head -1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/${genomesss}.faa|cut -d '_' -f 1|sed 's/>//g')
echo -e "This  genome : " ${genomesss} " has the following locus Tag : "${locusTagsss}

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/${genomesss}.faa > \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FFN/${species}/${genomesss}.ffn > \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.ffn


done


##------=================================================================
#extract orthogroups
##------=================================================================

cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/extract_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt --folder ${species}

```

align protein families

```{bash}


cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep ".faa$" |sed 's/.faa//g'|grep -v "Out-sali" )
do
rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.aln.fasta
mafft --thread 37 --maxiterate 1000  --auto /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.faa > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}_aln.fasta

done 

```
here I back translate the faa alignment to dna.

```{bash}

cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/aln_aa_to_dna.pl


```

With perl: prune alignments (removing columns represented by less than 50% of the sequences)


```{bash}


cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/prune_aln.pl


##================================================================
#remove all infomration (except genome info) from multifasta header
##================================================================


cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i "s/_0.*//" ${genesss}_aln_prune.fasta 

done

##================================================================
#shorten header
##================================================================

species=Sterm
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/-CIRM-//g' ${genesss}_aln_prune.fasta 

done

species=Sterm
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/-BM-A0/-BM-A/g' ${genesss}_aln_prune.fasta 

done


```
With perl: Concatenate pruned alignments
```{bash}
  

  echo -e ${species}
  cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}
  perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/cat_align.pl > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/${species}.phylip
  
```
With RAxML: infer the phylogeny
```{bash}
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_ncbi_own_ZOOMED/

  echo -e ${species}
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_ncbi_own_ZOOMED/

rm -r *${species}_all*

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/${species}.phylip -n ${species}_all -T 37
```

move local

```{bash}

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ZOOMED/

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_ncbi_own_ZOOMED/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ZOOMED//
```

do in R
```{r}
library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Sterm_tree_raw_zoomed <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new_ZOOMED/raxml_all_combined_ncbi_own_ZOOMED/RAxML_bipartitions.Sterm_all_cleaned", tree.names = NULL, force.multi = FALSE)


# x <- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
p <-ggtree(Sterm_tree_raw_zoomed) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)+  #NO ROOT  
  geom_treescale(x=0, y=35)

p

##--------------------------------
##color labels of of only rmk phages
##--------------------------------
Sterm_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/logs/Sterm_genome_origin.txt",  "\t", escape_double = FALSE, col_names = c("label","origin"),  trim_ws = TRUE) %>% as.data.frame()

# tmp <- full_join(as_tibble(Sterm_tree_raw), Sterm_genome_origin, by='label')
# table(tmp$origin)
tree <- left_join(as_tibble(Sterm_tree_raw_zoomed), Sterm_genome_origin, by='label') %>% as.treedata()



tmp <- as.tibble(tree)
tmp$label <- revalue(tmp$label,c("S-PNGR-Z-K"="PNGR-Z_K3"))
tree <- tmp %>% as.treedata()


p <- ggtree(tree) +
  geom_tiplab(aes(color=origin),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) + #NO ROOT  
  scale_color_manual(values=c("grey66",  'black')) 
p
##--------------------------------
##all info at once (geography and origin)
##--------------------------------
all_sterm_strain_info_ZOOMED <- read_csv("Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) %>% filter(species=="S.thermophilus") %>% select(c(strain,Cheese,starterCulture,Sequencing)) %>% remove_rownames()%>% column_to_rownames(., var = "strain")%>%replace(is.na(.), "Unknown") 

p3 <- gheatmap(p, all_sterm_strain_info_ZOOMED, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0)#+  scale_fill_manual(values=colors_area)
p3


##----------------CHANGE COLORS
COLORSSS <- c("#d45400ff","#de592bff","#e36b42ff","#e67d59ff","#eb8f70ff","#eea38aff","#009cf0ff","#33cffaff","#7fe3ffff","#007f7fff","#00dcdcff","#ff0000ff","#0000ffff","grey","#fde623ff","#00ff00ff")
p4 <- gheatmap(p, all_sterm_strain_info_ZOOMED, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0,)+  scale_fill_manual(values=COLORSSS)
p4

##--------------------------------
##ad poppunk clustering
##--------------------------------
Sterm_clusters_RMKS <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv", col_names = c("label","cluster"), col_types = cols(cluster = col_character(),  label = col_character()), skip = 1)%>% remove_rownames()%>% column_to_rownames(., var = "label")%>%replace(is.na(.), "Unknown") 



p5 <- gheatmap(p, Sterm_clusters_RMKS, offset=0.001, width=.05,colnames_angle=75, colnames_offset_y = 0)  #+  scale_fill_manual(values=colors_area)
p5

library(patchwork)



  # svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/CoverageSpecies.svg",width=5,height=5)
   png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_zoomed", width = 4500, height = 4500,res=300)

p4+p5

dev.off()


###--------------------------------
#distance 
library(ape)

## get the distance between every two nodes
tree_phylog <- tree %>% as.phylo()
distance_matrix <-cophenetic.phylo(tree_phylog)


```

## Gal's HGT appraoch

Here, I apply Gals paper appraoch. 
Github can be found [here](https://github.com/ghoresh11/twilight).

I want to do this analysis for both species together. 
therefore I gather all genomes (gffs) 

```{bash}
--------------------------------------
##create protein database
###---------------------------------------

##==========================================
###-----------------bring all gffs together
##==========================================
species=Sterm

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//GFFs/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//FAAs/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//GFFs/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//FAAs/


species=Ldel

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//GFFs/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//FAAs/

##==========================================
###-----------------Panaroo
##==========================================

conda activate panaroo

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/01_Panarooo/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/01_Panarooo/

panaroo -t 37 -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//GFFs/*.gff -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/01_Panarooo// --clean-mode strict

##==========================================
###-----------------classify_genes.R
##==========================================


/home/vincent/apps/twilight/classify_genes.R -p


```



## protein clustering 

Here, I cluster all protein sequences with mmseqs2

```{bash}

###---------------------------------------
##create protein database
###---------------------------------------

##==========================================
###-----------------bring all gffs together
##==========================================
species=Sterm

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/For_MMseqs2/all_ldels_sterms/GFFs/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/For_MMseqs2/all_ldels_sterms/Faas/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/For_MMseqs2/all_ldels_sterms/GFFs/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/For_MMseqs2/all_ldels_sterms/Faas/


species=Ldel

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/For_MMseqs2/all_ldels_sterms/GFFs/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/For_MMseqs2/all_ldels_sterms/Faas/


for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_GFF/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/
done

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/Out-sali.gff
##==========================================
###-----------------panaroo
##==========================================

##-----sterm

species=Sterm

conda activate panaroo

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species}_references/01_Panarooo/
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo

panaroo -t 37 -i //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//*.gff -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species}_references/01_Panarooo/ --clean-mode strict





###---------------------------------------
##cluster
###---------------------------------------
mmseqs easy-cluster examples/DB.fasta clusterRes tmp --min-seq-id 0.5 -c 0.8 --cov-mode 1

```

##Gene/Species tree

In order to compare the gene-2-species tree I use [genevax](https://github.com/BenoitMorel/GeneRax)

```{bash}

```


## CRISPR spacers

#### pilerCR
Here, I run pilerCR to extract all spacers to finally check if any of the strain do not have spacers against the RMK202 phage. 

I identify the CRISPR spacers with [PilerCR](https://www.drive5.com/pilercr/).

```{bash}

##=============
##pilerCR
##=============
for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     species_short=$(echo ${species}|head -c1)
     echo ==========================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/${species}
for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/ |grep "^${species_short}"|grep ".fna$" |sed 's/.fna//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR//${species}/${genomes}


  ~/apps/PilerCR/pilercr1.06/pilercr -in /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA//${genomes}.fna \
    -out /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR//${species}/${genomes}/${genomes}.pilarCR_out -noinfo\
    -seq /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR//${species}/${genomes}/${genomes}.pilarCR_out.fasta

  
  perl ~/apps/PilerCR/PilerCR_parser_new_vincent /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR//${species}/${genomes}/${genomes}.pilarCR_out > \
   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR//${species}/${genomes}/${genomes}.pilarCR_out_final
  done #genomes
  #  grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_final_new | sed 's/>//g' |awk -F "__" '{OFS="\t"} {print $4,$2,$3,$1}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_spacer.bed

done #species

###--------------do it from DIALACT sterms

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm
for genomes in $(ls /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20201010/Genomes/Dialact_02/Streptococcus_salivarius/|grep ".fna$" |sed 's/.fna//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}


  ~/apps/PilerCR/pilercr1.06/pilercr -in /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20201010/Genomes/Dialact_02/Streptococcus_salivarius//${genomes}.fna \
    -out  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}/${genomes}.pilarCR_out -noinfo\
    -seq  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}/${genomes}.pilarCR_out.fasta

  
  perl ~/apps/PilerCR/PilerCR_parser_new_vincent  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}/${genomes}.pilarCR_out > \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}/${genomes}.pilarCR_out_final
  done #genomes
```
all CRISPR arrays are in the same orientation.

#### blast CRISPRspacers
Here, I blast the CRISPR spacers against different databases. 
First I want to check if the spacers from the different genomes map against the RMK202 phages. 


```{bash}
###================
##description file
###================
threads=37
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/
species=Sterm
MetaspacerFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse//all_unique_spacers.fasta
mappingFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt

##--------------------------blast to local phages 
  

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     species_short=$(echo ${species}|head -c1)
     echo ==========================
     rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast
for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/ |grep "^${species_short}"|grep ".fna$" |sed 's/.fna//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast/${genomes}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast/${genomes}


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast \
  -query /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR//${species}/${genomes}/${genomes}.pilarCR_out_final \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sacc sblastnames sscinames" \
  -db ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome \
  -out /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast/${genomes}/${genomes}_CRISPRblast_rmk202phages.txt \
  -evalue 0.01 -word_size 16



  done #genomes
done #species



for species in $(echo "Ldel Sterm")
do

for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/ |grep "^${species_short}"|grep ".fna$" |sed 's/.fna//g')
do
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast/${genomes}/${genomes}_CRISPRblast_rmk202phages.txt


  done #genomes
done #species

##--------------------------DIALACT STERM


     rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact
for genomes in $(ls /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20201010/Genomes/Dialact_02/Streptococcus_salivarius/|grep ".fna$" |sed 's/.fna//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/${genomes}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/${genomes}


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 2 -task megablast \
  -query /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}/${genomes}.pilarCR_out_final \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sacc sblastnames sscinames" \
  -db ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome \
  -out /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/${genomes}/${genomes}_CRISPRblast_rmk202phages.txt \
  -evalue 0.01 -word_size 16



  done #genomes


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS.txt
for genomes in $(ls /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20201010/Genomes/Dialact_02/Streptococcus_salivarius/|grep ".fna$" |sed 's/.fna//g')
do
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/${genomes}/${genomes}_CRISPRblast_rmk202phages.txt   >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS.txt


  done #genomes
awk -F " " '{if($1<4)print $0}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS.txt


for samplesss in $(awk -F " " '{if($1<4)print $2}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS.txt|)
do
echo "==============================================="
cat ${samplesss}
done


awk -F " " '{if($1<4)print $0}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS.txt | grep -v "FAM1708_" |grep "FAM20897_" -v  |cut -f 2| cut -d '/' -f 9 >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS_lowHITsamples.txt

for genomes in $(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS_lowHITsamples.txt)
do
echo "==============================================="

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}/${genomes}.pilarCR_out_final
done

cat 
```




## Gene-sharing network

##### download all reference NCBI strains

Here, I have a script to download all genomes of S.thermophilus also if they are not complete and circular.

```{bash}

##=====================================================================================
##------------------------all complete streptos
##=====================================================================================
species=$(echo "Streptococcus thermophilus")
species_short=Sterm_references
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA

##currently taking all genomes not selecting for highest quality

cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt |grep -e "${species}" cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log//download_sterm.txt
    #pecies_short=Sterm_references

cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {print $1, "S_thermophilus_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log///download_sterm_names.txt

##--------------------------
##prepare list
##--------------------------
species=$(echo "Streptococcus thermophilus")
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {print $1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log_shortNames_00.txt

cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {print $1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/S-thermophilus-/S-/g'| sed 's/-STH-CIRM-/-I-/g' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log_shortNames_01.txt
rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log//20201216_NCBI_log_shortNames_02.txt
while read -r line
do
#echo $line
echo -e ${line} |head -c 10 |head -1 |sed 's/^$//g' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log//20201216_NCBI_log_shortNames_02.txt
echo -e "\n" >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log//20201216_NCBI_log_shortNames_02.txt

done < /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log_shortNames_01.txt
sed -i '/^[[:space:]]*$/d' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log//20201216_NCBI_log_shortNames_02.txt

paste -d '\t' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log//20201216_NCBI_log_shortNames_02.txt /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log_shortNames_00.txt > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log_shortNames_final.txt

#-------------------- add outgroup

outgroupNAME=NCTC8618

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >>    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
    species_short=Sterm_references

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_salivarius_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'>> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm_names.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print "Out-sali",$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print ,$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/S-thermophilus-/S-/g'| sed 's/-STH-CIRM-/-I-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt


cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
gunzip *.fna.gz

#-------------------- add outgroup
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
outgroupNAME=$(echo "Streptococcus vestibularis")

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >>    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
    species_short=Sterm_references

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_salivarius_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'>> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm_names.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print "Out-sali",$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print ,$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/S-thermophilus-/S-/g'| sed 's/-STH-CIRM-/-I-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt


cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
gunzip *.fna.gz

mv GCF_900636445.1_41965_G01_genomic.fna Out-vest.fna

rm Out-sali.fna
```

to start of with I use the already generated collection of complete and ciruclare genomes create on 20200728 to do the analysis. 

test outgroup with FASTani

```{bash}

echo "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/FNA/Out-sali.fna" > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/fastaANI_out_test.txt
echo "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/FNA/Out-vest.fna" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/fastaANI_out_test.txt
echo "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/FNA/S-LMD-9.fna" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/fastaANI_out_test.txt

###---------------------------
##fastaANI
###---------------------------

fastANI --fragLen 1000 -t 37 --ql /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/fastaANI_out_test.txt \
  --rl /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/fastaANI_out_test.txt \
  -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/fastaANI_out_test.fastani
```


####Orthofinder REFTREE

Here, I use orthofinder to create a nice reference tree 

```{bash}
###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Sterm

for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FAA/${leterzzz}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
done


###-----------------------------------------
##orthorfinder
###-----------------------------------------
#for species in $(echo "Ldel Sterm"|tail -1)
for species in $(echo "Sterm"|tail -1)
do

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/


    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}// \
      -t 35 \
      -o /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ \
      -a 35

done # species

```

move local

```{bash}
##--------------local

mkdir -p /home/vincent/Desktop/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm/Results_Aug05/Species_Tree/

scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm/Results_Aug05/Species_Tree/SpeciesTree_rooted* /home/vincent/Desktop/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm/Results_Aug05/Species_Tree/
```

preperation for itol

```{bash}
cut -f 5,8,9 /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm_edited.csv|sed '1d' > /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm_edited_forITOL.csv
```

#### POPCOGent


```{bash}
#conda activate PopCOGenT

#adjust the /home/vincent/apps/PopCOGenT/src/PopCOGenT/config.sh file and then run /home/vincent/apps/PopCOGenT/src/PopCOGenT/PopCOGenT.sh

###-----------------------------------------
##combine data
###-----------------------------------------

species=Sterm

for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/FNA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/${leterzzz}*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/
done


##-------------
#gather Sterm genomes
##-------------
rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test/
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test//{genomes,output}


cp -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species} \
/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test/


##-------------
#run script
##-------------

#problem with networkx version
#pip install --upgrade networkx==1.11

#export MUGSY_INSTALL=/home/vincent/apps/mugsy_x86-64-v1r2.3
#source /home/vincent/apps/mugsy_x86-64-v1r2.3/mugsyenv.sh

##!!!!!!!!!!!!!!!!!!!!!!!CHANGE CONFIG.SH file

cd /home/vincent/apps/PopCOGenT/src/PopCOGenT
bash /home/vincent/apps/PopCOGenT/src/PopCOGenT/PopCOGenT.sh
 
 ##===============
 #sweeps
 ##===============
 
cd /home/vincent/apps/PopCOGenT/src/core_gene_sweeps/
source activate PopCOGenT
python phybreak1.generate_maf.py
python phybreak2.maf_to_fasta.py


python phybreak3.MSAsubset_runPhyML.py

```

```{bash}

mkdir -p ~/Desktop/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test/output/
scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test/output/* ~/Desktop/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test/output/


```


read the graphML in r

```{r}
library(igraph)
 library(ggnetwork)
 library(ITNr)
 library(intergraph)
library(tidyverse)

net1<-read.graph("~/Desktop/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test/output/all_ncbi_rmk_Sterm_0.000355362.txt.unclust.graphml", format = "graphml")
# names(net1)
 

# net1[]
vcount(net1)
ecount(net1)
graph.density(net1)
diameter(net1)
transitivity(net1)
# hist(degree(net1), col="firebrick", xlab="Vertex Degree", ylab="Frequency", main="")


net<-asNetwork(net1) #ggnetwork requires a network object
 n<-ggnetwork(net)
 hist(n$weight)
 
 nExtended <- n# %>% filter(weight>300)
 networkplot <- ggplot(nExtended, aes(x = x, y = y, xend = xend, yend = yend,text =paste("ID:", id))) + geom_edges(color = "lightgray") +
   geom_nodes()+ theme_blank()
 networkplot
 
 library(plotly)
ggp <- ggplotly(networkplot)
 ggp
 
 
 length(table(n$id))
 n$weight
 ggplot(n,aes(x=weight))+geom_density()+facet_wrap(~id,scales = "free")+coord_trans(x="log10")+geom_vline(xintercept = 300)
 
 
 nExtended$catWeight <- ifelse(nExtended$weight<300,"low",ifelse(nExtended$weight<1000,"medium","high"))
 
  networkplot <- ggplot(data=nExtended) + geom_edges( aes(x = x, y = y, xend = xend, yend = yend,size=weight,alpha=catWeight)) +
   geom_nodes(aes(x = x, y = y, xend = xend, yend = yend,size=4,text =paste("ID:", id)))+ theme_blank()+ scale_alpha(range = c(0.6,0))
 networkplot
 ggp <- ggplotly(networkplot)
 ggp
 
 ###---------------
 
 ga<-read.graph("~/Desktop/Project/2021_Recominbation/PopCOGenT/201209_test/output/RMK_Sterm_0.000355362.txt.unclust.graphml", format = "graphml")

 clu <- cluster_louvain(ga)

#membership vector
mem <- membership(clu)
 com <- communities(clu)
com[[1]]
```




## Recombinants


move local

```{bash}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/S_O_202_24854.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/S_O_202_24854.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/S_O_202_24855.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/
progressiveMauve --output=my_seqs.xmfa \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/S_O_202_24855.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/S_O_202_24854.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/S_O_202_24854.fasta
```


```{bash}
##--------------local

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/ /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/
```

# phage assembly (from metagenome)

Here, I want to assemble all non-mapping reads

## mapping to all genomes


### create a mapping DB

First I create a reference DB containing all genomes from that cheese starter culture. 

```{bash}

for starterCultures in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/|cut -d '_' -f 3 |sort|uniq)
do
echo -e "starter culture :" ${starterCultures}

mkdir -p /data/Project/2020_StarterCultureDiversity/phage4meta/01_bacterialREFdb/
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/*_${starterCultures}_*.fasta > /data/Project/2020_StarterCultureDiversity/phage4meta/01_bacterialREFdb/${starterCultures}.fasta
 
 cat /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/Ldelv_mag.fasta >> \
  /data/Project/2020_StarterCultureDiversity/phage4meta/01_bacterialREFdb/${starterCultures}.fasta
 
done

```


```{bash}


names=Ws_150_19
#/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 index ${Assembly}
#/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 mem -t 16 ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz/${namezzz}-R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz/${namezzz}-R2_val_2.fq.gz |head
##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##script
###================
##============
##mapping to reference
##============
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

#mkdir -p ${logFilelocation}
num=1
samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt ##the file with all sample names
logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
threads=37
BaseLocation=/data/Project/2020_StarterCultureDiversity/phage4meta/01_bacterialREFdb/02_mapping2REFS
#rm ${BaseLocation}/log/multiqc_logfile_finalGenome.txt

for names in $(cut -f 1 ${samplesss} |grep "Lyo_105_76_1" -A 100)
  do

  starterCultures=$(echo ${names} |cut -d '_' -f 2)

Assembly=/data/Project/2020_StarterCultureDiversity/phage4meta/01_bacterialREFdb/${starterCultures}.fasta

  echo ${num}"/95  :" ${names} "from " ${starterCultures}
  num=$((num+1))
  
 bwa index $Assembly

  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 #/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 index ${Assembly} 
  
#  /home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 mem -t ${threads} ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
bwa mem -t ${threads} ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${ynames}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   



tail -95 ${BaseLocation}/log/multiqc_logfile_finalGenome.txt > ${BaseLocation}/log/multiqc_logfile_finalGenome_onlyMETA.txt
   
    rm -r ${BaseLocation}/MultiQC_onlyMETA
 
  mkdir -p ${BaseLocation}/MultiQC_onlyMETA
  
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome_onlyMETA.txt \
    -o ${BaseLocation}/MultiQC_onlyMETA

#/home/vincent/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/MultiQC/multiqc_report.html

scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/MultiQC/multiqc_report.html /Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/

##-------------------------------------------------  
##-----check read coverage
##-------------------------------------------------  


##-------------------------------------------------  
#name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
threads=37
names=Konserve_202


num=1
rm -r ${BaseLocation}/log/mappings.txt
#rm -r ${BaseLocation}/log/mappings_species.txt


for names in $(cut -f 1 ${samplesss})
  do
mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')

#mapped_sterm=$(grep "CP046134" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)
#mapped_ldel=$(grep "CP046131" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)
rm ${BaseLocation}/${names}/bwaMapping2DB//bamqc
mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc
qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G

#mkdir -p ${BaseLocation}/log
echo -e ${names}"\tonlyMeta\t"${mapped} >> ${BaseLocation}/log/mappings.txt

#echo -e ${names}"\tLactobacillus delbrueckii\t"$mapped_ldel >> ${BaseLocation}/log/mappings_species.txt
#echo -e ${names}"\tStreptococcus thermophilusi\t"$mapped_sterm >> ${BaseLocation}/log/mappings_species.txt

done

##-------------------------------------------------  
##move local
##-------------------------------------------------  

##bring together 

cat /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings.txt \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202/log/mappings.txt > \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings_final.txt


scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/log/mappings_species.txt Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/
```

####mapping strains to genome

```{bash}

threads=37
samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames.txt ##the file with all sample names
logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref
Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv.fasta

##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1

for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/)
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)
names=$(echo ${samplezzKurz})
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 # gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq
  
  
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} ${Assembly} \
  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   
   
    rm -r ${BaseLocation}/MultiQC
 
  mkdir -p ${BaseLocation}/MultiQC
  
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome.txt \
    -o ${BaseLocation}/MultiQC

##-------------------------------------------------  
##-----check read coverage
##-------------------------------------------------  


##-------------------------------------------------  
#name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
threads=37
names=Konserve_202


num=1
rm -r ${BaseLocation}/log/mappings.txt
#rm -r ${BaseLocation}/log/mappings_species.txt


for names in $(cut -f 1 ${samplesss})
  do
mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')

#mapped_sterm=$(grep "CP046134" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)
#mapped_ldel=$(grep "CP046131" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)

#mkdir -p ${BaseLocation}/log
echo -e ${names}"\tonlyMeta\t"${mapped} >> ${BaseLocation}/log/mappings.txt

#echo -e ${names}"\tLactobacillus delbrueckii\t"$mapped_ldel >> ${BaseLocation}/log/mappings_species.txt
#echo -e ${names}"\tStreptococcus thermophilusi\t"$mapped_sterm >> ${BaseLocation}/log/mappings_species.txt

done

grep "0." ${BaseLocation}/log/mappings_species.txt | sort -k 4

grep -v "0.0" ${BaseLocation}/log/mappings_species.txt|awk '{ sum += $4; n++ } END { if (n > 0) print sum / n; }'

```



```{r}
naming_DF <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/98_script/naming_file.csv")
###----------------------------
##script
###----------------------------
library(readr)
library(ggplot2)
#library(plyr)

species_coverage <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/mappings_species.txt", "\t", escape_double = FALSE, col_names = c("sample","species","coverage"),trim_ws = TRUE)

species_coverage$sample

newNames <- setNames(naming_DF$namesNew,naming_DF$namesOLD)
species_coverage$sample <- revalue(species_coverage$sample, newNames)
species_coverage$sample = factor(species_coverage$sample, levels=c(as.character(naming_DF$namesNew)))


newNames <- setNames(as.character(naming_DF$CultureName),naming_DF$namesNew)
species_coverage$culture <- revalue(as.character(species_coverage$sample),newNames)
levels(species_coverage$culture)

species_coverage$coverageLog <- log2(species_coverage$coverage)

all_colours <- (c("#EB4D4D","#10B552") ) 



##--------------------
##ploting


pMOTUs <-  ggplot( data = species_coverage,aes(y = coverage, x = sample, group=species,fill = culture,color=culture))+geom_point(size=2)+geom_hline(yintercept=250)+ # geom_bar( stat="identity")+
  facet_wrap(~species,scales = "free_x")+
   coord_trans(y="log10")+
    labs("",
         x="",
         y="Coverage")+
    theme_classic()+
   scale_color_viridis(discrete=TRUE)+
      scale_y_continuous(breaks =c(0,50,100,250,500,1000),labels=c(0,50,100,250,500,1000))+
  scale_fill_manual(values=all_colours)+
  scale_color_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=4),
      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          # legend.title = element_blank()
          )

  pMOTUs

  
  
  svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/CoverageSpecies.svg",width=5,height=5)
   # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)

pMOTUs

dev.off()

```




#appendix
#####pcoa

Here, I make a pcoa of the jaccard distance (presence/absence) matrix from the snps.
Additionally, I want to try the bray curtis distance (count data).

```{r}


library(readr)
snps_analysed_all <- read_csv("Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all.csv")
snps_analysed_all$samplecomplete <- paste(snps_analysed_all$culture,"_",snps_analysed_all$samplesNew,sep="")
snps_analysed_all_pcoa <- spread(snps_analysed_all, samplecomplete, Snps)  %>%replace(is.na(.), 0) %>% select(c(-species,-samplesNew,-culture)) 
snps_analysed_all_pcoa_final <- snps_analysed_all_pcoa[!duplicated(snps_analysed_all_pcoa[,1]), ]%>% column_to_rownames(var = "site") %>% t()
snps_analysed_all_pcoa_final <- snps_analysed_all_pcoa_final[,which(colSums(snps_analysed_all_pcoa_final)!=0)]
rownames(snps_analysed_all_pcoa_final[which(rowSums(snps_analysed_all_pcoa_final)==0),])
snps_analysed_all_pcoa_final <- snps_analysed_all_pcoa_final[which(rowSums(snps_analysed_all_pcoa_final)!=0),]

dim(snps_analysed_all_pcoa_final)

##----------------------------
##pcoa


library(ggplot2)
library(plyr)
library(tidyverse)
library(vegan)
# Here we use Bray-Curtis distance metric
# dist <- vegdist(snps_analysed_all_pcoa_final,  method = "bray")
library(ape)
dist <- vegdist(snps_analysed_all_pcoa_final,  method = "jaccard")
PCOA <- pcoa(dist)
barplot(PCOA$values$Relative_eig[1:10])
pcoa <- ggplot(PCOA_final,aes(x=Axis.1,y=Axis.2,color=sample,fill=sample,shape=cheese))+labs(title = "jaccard distance and PCOA")+geom_point(size=6,alpha=0.5)+theme_classic()

# PCoA is not included in vegan. 
# We will use the ape package instead
PCOA <- pcoa(dist)
barplot(PCOA$values$Relative_eig[1:10])

# Some distance measures may result in negative eigenvalues. In that case, add a correction:
# PCOA <- pcoa(dist, correction = "cailliez")

# Plot your results
biplot.pcoa(PCOA)

biplot.pcoa(PCOA, snps_analysed_all_pcoa_final)
  
PCOA_final <- PCOA$vectors %>% data.frame()%>% rownames_to_column("samplecomplete")
PCOA_final$sample <- str_split_fixed(PCOA_final$samplecomplete, "_", 2)[,1]
PCOA_final$number <- str_split_fixed(PCOA_final$samplecomplete, "_", 2)[,2]

PCOA_final$cheese <- revalue(PCOA_final$sample,c("101"="emmentaler","105"="emmentaler","115"="emmentaler","124"="emmentaler","150"="emmentaler","190"="emmentaler","202"="gruyere","203"="gruyere","280"="gruyere","302"="sbrinz","305"="sbrinz"))

pcoa <- ggplot(PCOA_final,aes(x=Axis.1,y=Axis.2,color=sample,fill=sample,shape=cheese))+labs(title = "jaccard distance and PCOA")+geom_point(size=6,alpha=0.5)+theme_classic()


###-----------------------
##maybe with bray curtis



library(vegan)
# Here we use Bray-Curtis distance metric
# dist <- vegdist(snps_analysed_all_pcoa_final,  method = "bray")
dist <- vegdist(snps_analysed_all_pcoa_final,  method = "bray")

# PCoA is not included in vegan. 
# We will use the ape package instead
library(ape)
PCOA <- pcoa(dist)
barplot(PCOA$values$Relative_eig[1:10])

# Some distance measures may result in negative eigenvalues. In that case, add a correction:
# PCOA <- pcoa(dist, correction = "cailliez")
PCOA$values$Relative_eig
values$Relative_eig

# Plot your results
biplot.pcoa(PCOA)

biplot.pcoa(PCOA, snps_analysed_all_pcoa_final)
  
PCOA_final <- PCOA$vectors %>% data.frame()%>% rownames_to_column("samplecomplete")
PCOA_final$sample <- str_split_fixed(PCOA_final$samplecomplete, "_", 2)[,1]
PCOA_final$number <- str_split_fixed(PCOA_final$samplecomplete, "_", 2)[,2]

PCOA_final$cheese <- revalue(PCOA_final$sample,c("101"="emmentaler","105"="emmentaler","115"="emmentaler","124"="emmentaler","150"="emmentaler","190"="emmentaler","202"="gruyere","203"="gruyere","280"="gruyere","302"="sbrinz","305"="sbrinz"))

ggplot(PCOA_final,aes(x=Axis.1,y=Axis.2,color=sample,fill=sample,shape=cheese))+geom_point(size=6,alpha=0.5)+theme_classic()

```



### ParSNP
here I make a snp tree
```{bash}
##==========================================
###-----------------preperation
##==========================================


##-------------------------------
##prep for parSNP isolated Strains
##-------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/

for speciesss in $(echo "L.delbrueckii S.thermophilus")
do
mkdir -p /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes
echo "-----------------"
echo ${speciesss}
for strainzzz in $(grep "^${speciesss}" /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv |cut -f 11)
do
culture=$(awk -F "\t" -v strainsss="$strainzzz" '{OFS="\t"}{if($11==strainsss)print $4}' /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed 's/RMK//g')

echo -e ${strainzzz} " from " ${culture}

tmp=$(ls /data/Project/2020_StarterCultureDiversity//06_Spades/FAM24731/assembly/)
cat /data/Project/2020_StarterCultureDiversity//06_Spades/FAM${strainzzz}/assembly/${tmp}/contigs.fasta > /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes/${culture}_i_${strainzzz}.fna



done #strainzzz
done #species

##-------------------------------
##prep for parSNP isolated Dialact
##-------------------------------

cp /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/Sterm/* /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/

cp /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/Ldel/* /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/

find /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/ -type f -empty -delete
find /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/ -type f -empty -delete

##-------------------------------
##change fna to fasta
##-------------------------------

###I have to make all genomes to fasta files and not fna

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/ |grep ".fna"|sed 's/.fna//g')
do

mv /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/${genomesss}.fna /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/${genomesss}.fasta 

done

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/ |grep ".fna"|sed 's/.fna//g')
do

mv /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/${genomesss}.fna /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/${genomesss}.fasta 

done

##-------------------------------
###still sample seems odd
##-------------------------------

rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/115_i_24728.fasta

# rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes/202_i_24753.fasta
# rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes/202_i_24754.fasta
#  rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes/202_i_24755.fasta

cat /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/FAM24728//binning.1.fa > /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/115_i_24728.fasta

##==========================================
###-----------------run parsnp
##==========================================
species=S.thermophilus
genomess=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/GCF_010120595.1_ASM1012059v1_genomic.fna

parsnp -r ${genomess} -v -c  -d /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes/ -p 5 -o  /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis/

sed -i 's/.fasta//g' /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis//parsnp.tree
sed -i 's/GCF_010120595.1_ASM1012059v1_genomic.fna/S. thermophilus type strain ATCC 19258/g' /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis//parsnp.tree

#parsnp -r ${genomess} -v -c  -d /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/Sterm/genome/ -p 5 -o  /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis_test/

harvesttools -i /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis//parsnp.ggr -V  /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis//parsnp.vcf

grep -v -c "^#" /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis//parsnp.vcf


##------------Ldel

species=L.delbrueckii
genomess=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/GCF_002278095.1_ASM227809v1_genomic.fna

parsnp -r ${genomess} -v -c  -d /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes/ -p 5 -o  /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis/

sed -i 's/.fasta//g' /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis//parsnp.tree
sed -i 's/GCF_002278095.1_ASM227809v1_genomic.fna/L. delbrueckii type strain ATCC 12315/g' /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis//parsnp.tree

harvesttools -i /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis//parsnp.ggr -V  /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis//parsnp.vcf

grep -v -c "^#" /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis//parsnp.vcf

###without type strain
##------------Ldel

species=L.delbrueckii
genomess=/data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes/101_d_21748.fasta
mkdir -p /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes_withoutTypStrain
cp /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes/* /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes_withoutTypStrain/

rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes_withoutTypStrain/101_d_21748.fasta

rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes_withoutTypStrain/302_i_24793.fasta
parsnp -r ${genomess} -v -c  -d /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes_withoutTypStrain/ -p 5 -o  /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis_withoutTypeStrain/

sed -i 's/.fasta//g' /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis_withoutTypeStrain//parsnp.tree
#sed -i 's/GCF_002278095.1_ASM227809v1_genomic.fna/L. delbrueckii type strain ATCC 12315/g' /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis//parsnp.tree

harvesttools -i /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis_withoutTypeStrain//parsnp.ggr -V  /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis_withoutTypeStrain//parsnp.vcf

grep -v -c "^#" /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/analysis_withoutTypeStrain//parsnp.vcf

```

move local

```{bash}
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/analysis/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/analysis//parsnp.tree /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/analysis//parsnp.tree

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/analysis/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/analysis//parsnp.tree /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/analysis//parsnp.tree

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/analysis_withoutTypeStrain/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/analysis_withoutTypeStrain//parsnp.tree /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/analysis_withoutTypeStrain//parsnp.tree

```

SNP tree from parSNP data


some good chapters 
[here](https://yulab-smu.github.io/treedata-book/chapter7.html)
[here](https://4va.github.io/biodatasci/r-ggtree.html)
```{r}
library(tidyverse)
library(ggtree)
library(ape)
library(phytools)
library(plyr)
library(tidyverse)
library(aplot)
library(patchwork)
tree <- read.tree("~/Desktop/Projects/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/analysis/parsnp.tree_rerooted")
isolatedStrainsRMK <- read_delim("Desktop/Projects/2020_strainDelineation/01_logs/isolatedStrainsRMK.csv","\t", escape_double = FALSE, trim_ws = TRUE) 


# isolatedStrainsRMK$ONT_cleaned <-  ifelse(isolatedStrainsRMK$ONT=="X","ONT","Illumina")
isolatedStrainsRMK$ONT_cleaned <-  ifelse(is.na(isolatedStrainsRMK$ONT),"Illumina","ONT")
# isolatedStrainsRMK$experiment_cleaned <-  ifelse(is.na(isolatedStrainsRMK$ONT),"Experiment","not used")

isolatedStrainsRMKexperiment <- isolatedStrainsRMK %>% select(FAM_number,ONT,Experiment,colony_genotype) 


isolatedStrainsRMKONT <- isolatedStrainsRMK %>% select(FAM_number,ONT_cleaned) 


isolatedStrainsRMK_sub <- isolatedStrainsRMK %>% select(FAM_number,colony_genotype)

##--------------------------
# change labels
##--------------------------
longName <- tree$tip.label

tree_plotting <- tree
tree_backup <- tree
tree_plotting$tip.label <- str_split_fixed(tree_plotting$tip.label,"_",3)[,3]
tree_plotting$tip.label[35] <- "type Strain"

finalPlotTree <- ggtree(tree_plotting)+ geom_tiplab(align=TRUE, linesize=.5,geom="text")
finalPlotTree
tree <- tree_plotting
tree$tip.label[35] <- "type Strain"

##--------------------------
# basic treee
##--------------------------
p12 <- ggtree(tree)+ geom_tiplab(align=TRUE, linesize=.5,geom="text")
# p12
# tree$edge




##--------------------------
# ordering
##--------------------------

tmp1 <- ggplot_build(p12)$data[[4]][,c("y","label")]
properOrder <- tmp1[order(-tmp1$y),] 
properOrder
##--------------------------
# additional information
##--------------------------
# cheeeeese <- str_split_fixed(tree$tip.label,"_",3)[,1]
# cheeeeese[grep("^3",str_split_fixed(tree$tip.label,"_",3)[,1])] <- "Sbrinz"
# cheeeeese[grep("^1",str_split_fixed(tree$tip.label,"_",3)[,1])] <- "Emmentaler"
# cheeeeese[grep("^2",str_split_fixed(tree$tip.label,"_",3)[,1])] <- "Gruyere"
# 
# cheeeeese[74] <- "type Strain"
# 
# FAMnumber <- c(tree$tip.label[1:73],"type Strain")
# starterCulturezzz <- c(as.character(str_split_fixed(longName,"_",3)[,1])[1:73],"type Strain")
# sourcesss <- c(as.character(str_split_fixed(longName,"_",3)[,2])[1:73],"type Strain") %>% revalue(.,c("i"="isolated","d"="dialact"))
# FAMss <- c(as.character(str_split_fixed(longName,"_",3)[,3])[1:73],"type Strain")
# trait.three <- data.frame("starter culture"=starterCulturezzz,"cheese"=cheeeeese,"source"=sourcesss)
# rownames(trait.three) <- tree$tip.label 


cheeeeese <- tree_backup$tip.label
cheeeeese[grep("^3",tree_backup$tip.label)] <- "Sbrinz"
cheeeeese[grep("^1",tree_backup$tip.label)] <- "Emmentaler"
cheeeeese[grep("^2",tree_backup$tip.label)] <- "Gruyere"
cheeeeese[35] <- "type Strain"

FAMnumber <- c(tree$tip.label[1:34],"type Strain")
starterCulturezzz <-c(as.character(str_split_fixed(longName,"_",3)[,1])[1:34],"type Strain")
sourcesss <- c(as.character(str_split_fixed(longName,"_",3)[,2])[1:34],"type Strain") %>% revalue(.,c("i"="isolated","d"="dialact"))
FAMss <- c(as.character(str_split_fixed(longName,"_",3)[,3])[1:34],"type Strain")
trait.three <- data.frame("starter culture"=starterCulturezzz,"cheese"=cheeeeese,"source"=sourcesss)
rownames(trait.three) <- tree$tip.label 


###--------------prep table
tabelSterm <- data.frame("long.name"=as.character(tree_backup$tip.label),"FAM"=as.character(str_split_fixed(tree_backup$tip.label,"_",3)[,3]),"starter culture"=as.character(str_split_fixed(tree_backup$tip.label,"_",3)[,1]),"cheese"=cheeeeese,"source"=c(as.character(str_split_fixed(tree_backup$tip.label,"_",3)[,2])[1:34],"type Strain"))

table_sterm_complete <- merge(tabelSterm,isolatedStrainsRMKexperiment,by.x = "FAM",by.y="FAM_number",all.x = TRUE)
table_sterm_complete$ONT <-  ifelse(is.na(table_sterm_complete$ONT),"Illumina","ONT")
table_sterm_complete$Experiment <-  ifelse(is.na(table_sterm_complete$Experiment),"not used","Experiment")

table_sterm_complete$source <- table_sterm_complete$source %>% revalue(.,c("i"="isolated","d"="dialact"))
table_sterm_complete <- table_sterm_complete %>% filter(source!="type Strain")

write.table(table_sterm_complete,"~/Desktop/Projects/2020_strainDelineation/01_logs/all_RMK_curated_Sterm.csv",sep="\t",quote = FALSE,row.names = FALSE)

# isolatedStrainsRMKexperiment

##--------------------------
# make combined
##--------------------------

trait.four <- data.frame("FAM"=FAMss)
trait.four$long <- tree$tip.label
trait.four <- merge(trait.four,isolatedStrainsRMK_sub,by.y = "FAM_number",by.x="FAM",all.x = TRUE) %>% column_to_rownames(var="long") %>% select(-FAM)

# trait.five <- data.frame("FAM"=FAMss)
# trait.five$long <- tree$tip.label
# trait.five <- merge(trait.five,isolatedStrainsRMKONT,by.y = "FAM_number",by.x="FAM",all.x = TRUE) %>% column_to_rownames(var="long") %>% select(-FAM)
trait.five <- table_sterm_complete %>% select(FAM,ONT,Experiment) %>% column_to_rownames(var="FAM") 
# trait.five <- table_sterm_complete %>% select(FAM,ONT) %>% column_to_rownames(var="FAM") 

# 
# 
# trait.four <- data.frame("FAM"=rev(FAMss))
# trait.four$names <- rev(tree$tip.label)
# trait.four <- merge(trait.four,isolatedStrainsRMK_sub,by.y = "FAM_number",by.x="FAM",all.x = TRUE) %>% select(-FAM)
# trait.four$second <- 1
# trait.four$colony_genotype <- as.character(trait.four$colony_genotype)
# 
# trait.four <- trait.four[match(properOrder$label, trait.four$names),]

# ggplot(trait.four, aes(x=second,y=names)) +geom_tile(aes(fill=colony_genotype))



colors_rmks <-  c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#80e5ff","#008181","#00dcdc")
# colors_cheese <- c("blue","red","green")
# colors_soure <- c("pink","green")
colors_remaining <- c("pink","red","blue","orange","green","white","gray89")
colorsss <- c(colors_rmks,colors_remaining)

# trait.three = factor(trait.three, levels=c("101","105","115","124","150","190","202","203","280","302","305","Emmentaler","Gruyere","Sbrinz","isolated","dialact","type Strain"))


p13 <- gheatmap(p12, trait.three, offset=.12, width=.1, colnames=TRUE,colnames_position="top", colnames_angle=45, colnames_offset_y = 2, colnames_offset_x = 0.02)+ scale_fill_manual(values = colorsss)
p13



p14 <- gheatmap(p12, trait.four, offset=.12, width=.033, colnames=TRUE,colnames_position="top", colnames_angle=45, colnames_offset_y = 2, colnames_offset_x = 0.02)+ scale_fill_viridis_d(option="D", name="discrete\nvalue")
p14

p15 <- gheatmap(p12, trait.five, offset=.12, width=.033, colnames=TRUE,colnames_position="top", colnames_angle=45, colnames_offset_y = 2, colnames_offset_x = 0.02)+ scale_fill_viridis_d(option="D", name="discrete\nvalue")
p15

# p14 <- gheatmap(p13, trait.four, offset=.1, width=.1, colnames=TRUE,colnames_position="top", colnames_angle=45, colnames_offset_y = 2, colnames_offset_x = 0.02)+ scale_fill_manual(values = colorsss)
# p14

 # png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/S_thermophilus_strains.png", width = 3200, height = 2000,res=300)
svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/S_thermophilus_strains.svg",width=27,height=18)


p13+p14+p15

dev.off()



###gather information

# tree_plotting$tip.label
# tree_backup$tip.label
```


```{r}
##-------------
##ldel
##-------------

tree <- read.tree("~/Desktop/Projects/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/analysis/parsnp_rerooted.tree")
isolatedStrainsRMK <- read_delim("Desktop/Projects/2020_strainDelineation/01_logs/isolatedStrainsRMK.csv","\t", escape_double = FALSE, trim_ws = TRUE) 
isolatedStrainsRMK$ONT_cleaned <-  ifelse(is.na(isolatedStrainsRMK$ONT),"Illumina","ONT")
# isolatedStrainsRMK$experiment_cleaned <-  ifelse(is.na(isolatedStrainsRMK$ONT),"Experiment","not used")

isolatedStrainsRMKexperiment <- isolatedStrainsRMK %>% select(FAM_number,ONT,Experiment,colony_genotype) 


isolatedStrainsRMK_sub <- isolatedStrainsRMK %>% select(FAM_number,colony_genotype)

##--------------------------
# change labels
##--------------------------
longName <- tree$tip.label
§
tree_plotting <- tree
tree_backup <- tree
tree_plotting$tip.label <- str_split_fixed(tree_plotting$tip.label,"_",3)[,3]
tree_plotting$tip.label[74] <- "type Strain"

finalPlotTree <- ggtree(tree_plotting)+ geom_tiplab(align=TRUE, linesize=.5,geom="text")
finalPlotTree
tree <- tree_plotting
tree$tip.label[74] <- "type Strain"

# tree_plotting <- tree
# tree_plotting$tip.label <- str_split_fixed(tree_plotting$tip.label,"_",3)[,3]
# tree_plotting$tip.label[74] <- "type Strain"
# 
# finalPlotTree <- ggtree(tree_plotting)+ geom_tiplab(align=TRUE, linesize=.5,geom="text")
# finalPlotTree
# tree <- tree_plotting
# tree$tip.label[74] <- "type Strain"


# tree$tip.label[74] <- "type Strain"

##--------------------------
# basic treee
##--------------------------
p12 <- ggtree(tree)+ geom_tiplab(align=TRUE, linesize=.5,geom="text")
# p12
tree$edge

##--------------------------
# ordering
##--------------------------

tmp1 <- ggplot_build(p12)$data[[4]][,c("y","label")]
properOrder <- tmp1[order(-tmp1$y),] 
properOrder
##--------------------------
# additional information
##--------------------------

# cheeeeese <- str_split_fixed(tree$tip.label,"_",3)[,1]
# cheeeeese[grep("^3",str_split_fixed(tree$tip.label,"_",3)[,1])] <- "Sbrinz"
# cheeeeese[grep("^1",str_split_fixed(tree$tip.label,"_",3)[,1])] <- "Emmentaler"
# cheeeeese[grep("^2",str_split_fixed(tree$tip.label,"_",3)[,1])] <- "Gruyere"
# 
# FAMnumber <- c(str_split_fixed(tree$tip.label,"_",3)[,3][1:73],"type Strain")
# starterCulturezzz <- as.character(str_split_fixed(tree$tip.label,"_",3)[,1])
# sourcesss <- c(as.character(str_split_fixed(tree$tip.label,"_",3)[,2])[1:73],"type Strain") %>% revalue(.,c("i"="isolated","d"="dialact"))
# FAMss <- c(as.character(str_split_fixed(tree$tip.label,"_",3)[,3])[1:73],"type Strain")
# trait.three <- data.frame("starter culture"=starterCulturezzz,"cheese"=cheeeeese,"source"=sourcesss)
# rownames(trait.three) <- tree$tip.label 

# trait.three <- data.frame("starter culture"=as.character(str_split_fixed(tree$tip.label,"_",3)[,1]),"cheese"=cheeeeese,"source"=c(as.character(str_split_fixed(tree$tip.label,"_",3)[,2])[1:34],"type Strain"))


cheeeeese <- tree_backup$tip.label
cheeeeese[grep("^3",tree_backup$tip.label)] <- "Sbrinz"
cheeeeese[grep("^1",tree_backup$tip.label)] <- "Emmentaler"
cheeeeese[grep("^2",tree_backup$tip.label)] <- "Gruyere"
cheeeeese[74] <- "type Strain"

FAMnumber <- c(tree$tip.label[1:73],"type Strain")
starterCulturezzz <- c(as.character(str_split_fixed(longName,"_",3)[,1])[1:73],"type Strain")
sourcesss <- c(as.character(str_split_fixed(longName,"_",3)[,2])[1:73],"type Strain") %>% revalue(.,c("i"="isolated","d"="dialact"))
FAMss <- c(as.character(str_split_fixed(longName,"_",3)[,3])[1:73],"type Strain")
trait.three <- data.frame("starter culture"=starterCulturezzz,"cheese"=cheeeeese,"source"=sourcesss)
rownames(trait.three) <- tree$tip.label 


###--------------prep table
tabelSterm <- data.frame("long.name"=as.character(tree_backup$tip.label),"FAM"=as.character(str_split_fixed(tree_backup$tip.label,"_",3)[,3]),"starter culture"=as.character(str_split_fixed(tree_backup$tip.label,"_",3)[,1]),"cheese"=cheeeeese,"source"=c(as.character(str_split_fixed(tree_backup$tip.label,"_",3)[,2])[1:73],"type Strain"))
isolatedStrainsRMKexperiment %>% filter(Experiment=="used")
table_sterm_complete <- merge(tabelSterm,isolatedStrainsRMKexperiment,by.x = "FAM",by.y="FAM_number",all.x = TRUE)
table_sterm_complete$ONT <-  ifelse(is.na(table_sterm_complete$ONT),"Illumina","ONT")
table_sterm_complete$Experiment <-  ifelse(is.na(table_sterm_complete$Experiment),"not used","Experiment")

table_sterm_complete$source <- table_sterm_complete$source %>% revalue(.,c("i"="isolated","d"="dialact"))
table_sterm_complete <- table_sterm_complete %>% filter(source!="type Strain")

write.table(table_sterm_complete,"~/Desktop/Projects/2020_strainDelineation/01_logs/all_RMK_curated_Ldel.csv",sep="\t",quote = FALSE,row.names = FALSE)

table_sterm_complete %>% filter(Experiment=="Experiment")

##--------------------------
# make combined
##--------------------------
# 
# trait.four <- data.frame("FAM"=FAMss)
# trait.four$long <- tree$tip.label
# trait.four <- merge(trait.four,isolatedStrainsRMK_sub,by.y = "FAM_number",by.x="FAM",all.x = TRUE) %>% column_to_rownames(var="long") %>% select(-FAM)

trait.four <- table_sterm_complete %>% select(FAM,colony_genotype) %>% column_to_rownames(var="FAM") 
trait.five <- table_sterm_complete %>% select(FAM,ONT,Experiment) %>% column_to_rownames(var="FAM") 

# 
# 
# trait.four <- data.frame("FAM"=rev(FAMss))
# trait.four$names <- rev(tree$tip.label)
# trait.four <- merge(trait.four,isolatedStrainsRMK_sub,by.y = "FAM_number",by.x="FAM",all.x = TRUE) %>% select(-FAM)
# trait.four$second <- 1
# trait.four$colony_genotype <- as.character(trait.four$colony_genotype)
# 
# trait.four <- trait.four[match(properOrder$label, trait.four$names),]

# ggplot(trait.four, aes(x=second,y=names)) +geom_tile(aes(fill=colony_genotype))



colors_rmks <-  c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#80e5ff","#008181","#00dcdc")
# colors_cheese <- c("blue","red","green")
# colors_soure <- c("pink","green")
colors_remaining <- c("pink","red","blue","orange","green","white","gray89")
colorsss <- c(colors_rmks,colors_remaining)

# trait.three = factor(trait.three, levels=c("101","105","115","124","150","190","202","203","280","302","305","Emmentaler","Gruyere","Sbrinz","isolated","dialact","type Strain"))


p13 <- gheatmap(p12, trait.three, offset=.12, width=.1, colnames=TRUE,colnames_position="top", colnames_angle=45, colnames_offset_y = 2, colnames_offset_x = 0.02)+ scale_fill_manual(values = colorsss)
p13



p14 <- gheatmap(p12, trait.four, offset=.12, width=.033, colnames=TRUE,colnames_position="top", colnames_angle=45, colnames_offset_y = 2, colnames_offset_x = 0.02)+ scale_fill_viridis_d(option="D", name="discrete\nvalue")
p14

p15 <- gheatmap(p12, trait.five, offset=.12, width=.033, colnames=TRUE,colnames_position="top", colnames_angle=45, colnames_offset_y = 2, colnames_offset_x = 0.02)+ scale_fill_viridis_d(option="D", name="discrete\nvalue")
p15
# p14 <- gheatmap(p13, trait.four, offset=.1, width=.1, colnames=TRUE,colnames_position="top", colnames_angle=45, colnames_offset_y = 2, colnames_offset_x = 0.02)+ scale_fill_manual(values = colorsss)
# p14

 # png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/S_thermophilus_strains.png", width = 3200, height = 2000,res=300)
svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Ldel_strains.svg",width=27,height=18)


p13+p14+p15

dev.off()
```


```{r}
##-------------
##ldel
##-------------

tree <- read.tree("~/Desktop/Projects/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/analysis_withoutTypeStrain/parsnp_rerooted.tree")
isolatedStrainsRMK <- read_delim("Desktop/Projects/2020_strainDelineation/01_logs/isolatedStrainsRMK.csv","\t", escape_double = FALSE, trim_ws = TRUE) 
isolatedStrainsRMK$ONT_cleaned <-  ifelse(is.na(isolatedStrainsRMK$ONT),"Illumina","ONT")
# isolatedStrainsRMK$experiment_cleaned <-  ifelse(is.na(isolatedStrainsRMK$ONT),"Experiment","not used")

isolatedStrainsRMKexperiment <- isolatedStrainsRMK %>% select(FAM_number,ONT,Experiment,colony_genotype) 


isolatedStrainsRMK_sub <- isolatedStrainsRMK %>% select(FAM_number,colony_genotype)

##--------------------------
# change labels
##--------------------------
longName <- tree$tip.label

tree_plotting <- tree
tree_backup <- tree
tree_plotting$tip.label <- str_split_fixed(tree_plotting$tip.label,"_",3)[,3]
# tree_plotting$tip.label[74] <- "type Strain"

finalPlotTree <- ggtree(tree_plotting)+ geom_tiplab(align=TRUE, linesize=.5,geom="text")
finalPlotTree
tree <- tree_plotting
# tree$tip.label[74] <- "type Strain"

# tree_plotting <- tree
# tree_plotting$tip.label <- str_split_fixed(tree_plotting$tip.label,"_",3)[,3]
# tree_plotting$tip.label[74] <- "type Strain"
# 
# finalPlotTree <- ggtree(tree_plotting)+ geom_tiplab(align=TRUE, linesize=.5,geom="text")
# finalPlotTree
# tree <- tree_plotting
# tree$tip.label[74] <- "type Strain"


# tree$tip.label[74] <- "type Strain"

##--------------------------
# basic treee
##--------------------------
p12 <- ggtree(tree)+ geom_tiplab(align=TRUE, linesize=.5,geom="text")
# p12
tree$edge

##--------------------------
# ordering
##--------------------------

tmp1 <- ggplot_build(p12)$data[[4]][,c("y","label")]
properOrder <- tmp1[order(-tmp1$y),] 
properOrder
##--------------------------
# additional information
##--------------------------

# cheeeeese <- str_split_fixed(tree$tip.label,"_",3)[,1]
# cheeeeese[grep("^3",str_split_fixed(tree$tip.label,"_",3)[,1])] <- "Sbrinz"
# cheeeeese[grep("^1",str_split_fixed(tree$tip.label,"_",3)[,1])] <- "Emmentaler"
# cheeeeese[grep("^2",str_split_fixed(tree$tip.label,"_",3)[,1])] <- "Gruyere"
# 
# FAMnumber <- c(str_split_fixed(tree$tip.label,"_",3)[,3][1:73],"type Strain")
# starterCulturezzz <- as.character(str_split_fixed(tree$tip.label,"_",3)[,1])
# sourcesss <- c(as.character(str_split_fixed(tree$tip.label,"_",3)[,2])[1:73],"type Strain") %>% revalue(.,c("i"="isolated","d"="dialact"))
# FAMss <- c(as.character(str_split_fixed(tree$tip.label,"_",3)[,3])[1:73],"type Strain")
# trait.three <- data.frame("starter culture"=starterCulturezzz,"cheese"=cheeeeese,"source"=sourcesss)
# rownames(trait.three) <- tree$tip.label 

# trait.three <- data.frame("starter culture"=as.character(str_split_fixed(tree$tip.label,"_",3)[,1]),"cheese"=cheeeeese,"source"=c(as.character(str_split_fixed(tree$tip.label,"_",3)[,2])[1:34],"type Strain"))


cheeeeese <- tree_backup$tip.label
cheeeeese[grep("^3",tree_backup$tip.label)] <- "Sbrinz"
cheeeeese[grep("^1",tree_backup$tip.label)] <- "Emmentaler"
cheeeeese[grep("^2",tree_backup$tip.label)] <- "Gruyere"
# cheeeeese[74] <- "type Strain"

FAMnumber <- c(tree$tip.label)
starterCulturezzz <- c(as.character(str_split_fixed(longName,"_",3)[,1]))
sourcesss <- c(as.character(str_split_fixed(longName,"_",3)[,2])) %>% revalue(.,c("i"="isolated","d"="dialact"))
FAMss <- c(as.character(str_split_fixed(longName,"_",3)[,3]))
trait.three <- data.frame("starter culture"=starterCulturezzz,"cheese"=cheeeeese,"source"=sourcesss)
rownames(trait.three) <- tree$tip.label 


###--------------prep table
tabelSterm <- data.frame("long.name"=as.character(tree_backup$tip.label),"FAM"=as.character(str_split_fixed(tree_backup$tip.label,"_",3)[,3]),"starter culture"=as.character(str_split_fixed(tree_backup$tip.label,"_",3)[,1]),"cheese"=cheeeeese,"source"=c(as.character(str_split_fixed(tree_backup$tip.label,"_",3)[,2])))
isolatedStrainsRMKexperiment %>% filter(Experiment=="used")
table_sterm_complete <- merge(tabelSterm,isolatedStrainsRMKexperiment,by.x = "FAM",by.y="FAM_number",all.x = TRUE)
table_sterm_complete$ONT <-  ifelse(is.na(table_sterm_complete$ONT),"Illumina","ONT")
table_sterm_complete$Experiment <-  ifelse(is.na(table_sterm_complete$Experiment),"not used","Experiment")

table_sterm_complete$source <- table_sterm_complete$source %>% revalue(.,c("i"="isolated","d"="dialact"))
table_sterm_complete <- table_sterm_complete %>% filter(source!="type Strain")

# write.table(table_sterm_complete,"~/Desktop/Projects/2020_strainDelineation/01_logs/all_RMK_curated_Ldel.csv",sep="\t",quote = FALSE,row.names = FALSE)

table_sterm_complete %>% filter(Experiment=="Experiment")

##--------------------------
# make combined
##--------------------------
# 
# trait.four <- data.frame("FAM"=FAMss)
# trait.four$long <- tree$tip.label
# trait.four <- merge(trait.four,isolatedStrainsRMK_sub,by.y = "FAM_number",by.x="FAM",all.x = TRUE) %>% column_to_rownames(var="long") %>% select(-FAM)

trait.four <- table_sterm_complete %>% select(FAM,colony_genotype) %>% column_to_rownames(var="FAM") 
trait.five <- table_sterm_complete %>% select(FAM,ONT,Experiment) %>% column_to_rownames(var="FAM") 

# 
# 
# trait.four <- data.frame("FAM"=rev(FAMss))
# trait.four$names <- rev(tree$tip.label)
# trait.four <- merge(trait.four,isolatedStrainsRMK_sub,by.y = "FAM_number",by.x="FAM",all.x = TRUE) %>% select(-FAM)
# trait.four$second <- 1
# trait.four$colony_genotype <- as.character(trait.four$colony_genotype)
# 
# trait.four <- trait.four[match(properOrder$label, trait.four$names),]

# ggplot(trait.four, aes(x=second,y=names)) +geom_tile(aes(fill=colony_genotype))



colors_rmks <-  c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#80e5ff","#008181","#00dcdc")
# colors_cheese <- c("blue","red","green")
# colors_soure <- c("pink","green")
colors_remaining <- c("pink","red","blue","orange","green","white","gray89")
colorsss <- c(colors_rmks,colors_remaining)

# trait.three = factor(trait.three, levels=c("101","105","115","124","150","190","202","203","280","302","305","Emmentaler","Gruyere","Sbrinz","isolated","dialact","type Strain"))


p13 <- gheatmap(p12, trait.three, offset=.12, width=.1, colnames=TRUE,colnames_position="top", colnames_angle=45, colnames_offset_y = 2, colnames_offset_x = 0.02)+ scale_fill_manual(values = colorsss)
p13



p14 <- gheatmap(p12, trait.four, offset=.12, width=.033, colnames=TRUE,colnames_position="top", colnames_angle=45, colnames_offset_y = 2, colnames_offset_x = 0.02)+ scale_fill_viridis_d(option="D", name="discrete\nvalue")
p14

p15 <- gheatmap(p12, trait.five, offset=.12, width=.033, colnames=TRUE,colnames_position="top", colnames_angle=45, colnames_offset_y = 2, colnames_offset_x = 0.02)+ scale_fill_viridis_d(option="D", name="discrete\nvalue")
p15
# p14 <- gheatmap(p13, trait.four, offset=.1, width=.1, colnames=TRUE,colnames_position="top", colnames_angle=45, colnames_offset_y = 2, colnames_offset_x = 0.02)+ scale_fill_manual(values = colorsss)
# p14

 # png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/S_thermophilus_strains.png", width = 3200, height = 2000,res=300)
svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Ldel_strains_zoomIN.svg",width=27,height=18)


p13+p14+p15

dev.off()
```
