---
title: "Thesis chapter 1"
author: "Vincent Somerville"
date: "17/05/2022"
output: html_document
editor_options: 
  chunk_output_type: console
---


# Code of the Thesis Chapter 1

This is the code that has been generated to produce the thesis chapter 1. 
As the project is currently still Ongoing it has not been polished or cleaned to remove unnecessary code chunks. 

Thank you for your interest :)


## Setup

```{r setup ,echo=FALSE}
knitr::opts_chunk$set(message=FALSE, echo=FALSE, eval=FALSE)
```

## starter culture historic data temperol


## starter culture historic data


```{r}
library(readr)
library(tsibble)
library(tidyverse)
# frmAuswert_Flüssigkulturen <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/starterCultureProduction/frmAuswert_Flüssigkulturen.csv",  "\t", escape_double = FALSE, col_types = cols(qCharge = col_date(format = "%d/%m/%y")),  trim_ws = TRUE)

# ph_workingStock_rmk202 <- read_delim("Desktop/Projects/2019_RMK202_analysis/workingStock_pH/ph_workingStock_rmk202.csv", "\t", escape_double = FALSE, trim_ws = TRUE, col_types = cols(mixDate = col_date(format = "%W/%y")))

ph_workingStock_rmk202 <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/workingStock_pH/ph_workingStock_rmk202.csv", "\t", escape_double = FALSE, trim_ws = TRUE)

# Path$New.Week <-
  # strftime(ph_workingStock_rmk202$mixDate, format = "%W/%Y")

  # yearweek(ph_workingStock_rmk202$yearweekcol)
ph_workingStock_rmk202$week
 ph_workingStock_rmk202$date <-  MMWRweek::MMWRweek2Date(MMWRyear = ph_workingStock_rmk202$year,
                        MMWRweek = ph_workingStock_rmk202$week,
                        MMWRday = 3)
  
  ph_workingStock_rmk202$Versand
acidicationPlot <- ggplot(ph_workingStock_rmk202,aes(x=date,y=Versand))+geom_point()+theme_classic()+scale_y_continuous(limits = c(0,150))+labs(y="acid-base titration value of starter culture",x="time [year]")+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9))+geom_hline(yintercept = 110)
 
 acidicationPlot
 is.numeric(ph_workingStock_rmk202$`D(-)%`)
 ph_workingStock_rmk202 %>% filter(!is.na("D(-)%"))
 
# ph_workingStock_rmk202_sub <-   ph_workingStock_rmk202 %>% filter(!is.na(`D(-)%`))

ph_workingStock_rmk202_sub <-   ph_workingStock_rmk202%>% filter(!is.na(`D(-)%`)) %>% filter(as.numeric(`D(-)%`)<100)
ph_workingStock_rmk202_sub$`D(-)%` <-  as.numeric(ph_workingStock_rmk202_sub$`D(-)%`)

lactatePlot <- ggplot(ph_workingStock_rmk202_sub,aes(x=mixDate,y=`D(-)%`))+geom_point()+theme_classic()+scale_y_continuous(limits = c(0,100))+labs(y="Percent of D-lactate",x="time")+theme(axis.text.x = element_blank())
 
 lactatePlot
 
png("~/Desktop/mid_thesis/report/figures/20200430/chapter1/lactatePlot.png", width = 2100, height = 1500,res=300)
 # svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
lactatePlot
dev.off()
 
##---------------------------
###----------------over time
##---------------------------

 
# colnames(frmAuswert_Flüssigkulturen)
# table(frmAuswert_Flüssigkulturen$tKulturBezeichnung)


# subsetAcidification <- frmAuswert_Flüssigkulturen %>% filter(tKulturBezeichnung=="RMK 202")
# subsetAcidification$`BK38Th 18 h` <- as.Date(subsetAcidification$`BK38Th 18 h`,format = "%d/%m/%y")
# str(subsetAcidification)
# subsetAcidification$qCharge <- as.Date(subsetAcidification$qCharge)
# subsetAcidification$qWoche
# acidicationPlot <- ggplot(subsetAcidification,aes(x=qCharge,`BK38Th 18 h`))+geom_point()+theme_classic()+scale_y_continuous(limits = c(0,150))+labs(y="acid-base titration value after 18 h",x="time [month-year]")+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9))+scale_x_date(date_labels = "%m-%y")
# ss <- subset(subsetAcidification, qWoche > as.Date("2006-1-1"))
# ggplot(subsetAcidification,aes(x=qWoche,`BK38Th 18 h`))+geom_point()+theme_classic()+scale_y_continuous(limits = c(0,150))+labs(y="acid-base titration value after 18 h",x="")+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9))

png("~/Desktop/mid_thesis/report/figures/20200430/chapter1/acidificationPlot.png", width = 2100, height = 1500,res=300)
 # svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
acidicationPlot
dev.off()

##--------------------------------------------------
#new
##--------------------------------------------------


library(readr)
library(tsibble)
library(tidyverse)


colors <- c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#80e5ff","#008181","#00dcdc")



# plotl_d

# ph_workingStock_rmk202 <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/workingStock_pH/ph_workingStock_rmk202.csv", "\t", escape_double = FALSE, trim_ws = TRUE)
# Säuregrad_RMK_final <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/historic_samples/Säuregrad_RMK_final_reduced.csv",   delim = ",", escape_double = FALSE,    trim_ws = TRUE)

Säuregrad_RMK_final <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/historic_samples/Säuregrad_RMK_final_reduced.csv",   delim = ",", escape_double = FALSE,    col_types = cols(RMK = col_character(),`BK38°C` = col_character(),`KZ` = col_character()),  trim_ws = TRUE) %>% filter(week<54) %>% filter(`BK38°C`!="k.R.") %>% filter(!KZ %in% "Nur Listerien geprüft (wegen Krankheit)")

table(Säuregrad_RMK_final$KZ)
Säuregrad_RMK_final$KZ <- as.numeric(Säuregrad_RMK_final$KZ)
Säuregrad_RMK_final$`BK38°C` <- as.numeric(Säuregrad_RMK_final$`BK38°C`)
Säuregrad_RMK_final$`D(-)%` <-  Säuregrad_RMK_final$`D(-)%` %>% gsub(",",".",.) %>% as.numeric()


  # yearweek(Säuregrad_RMK_final$yearweekcol)
Säuregrad_RMK_final$week
 Säuregrad_RMK_final$date <-  MMWRweek::MMWRweek2Date(MMWRyear = Säuregrad_RMK_final$year,
                        MMWRweek = Säuregrad_RMK_final$week,
                        MMWRday = 3)
  
  Säuregrad_RMK_final$Versand
acidicationPlot <- ggplot(Säuregrad_RMK_final,aes(x=date,y=`BK38°C`,fill=RMK,color=RMK))+geom_point(alpha=0.2)+theme_classic()+scale_y_continuous(limits = c(0,150))+labs(y="acid-base titration value of starter culture",x="time [year]")+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9))+geom_hline(yintercept = 110)+scale_fill_manual(values = colors)+scale_color_manual(values = colors)+geom_smooth()
 
 acidicationPlot
 # is.num(Säuregrad_RMK_final$`D(-)%`)
 Säuregrad_RMK_final %>% filter(!is.na("D(-)%"))
 
# ph_workingStock_rmk202_sub <-   Säuregrad_RMK_final %>% filter(!is.na(`D(-)%`))

ph_workingStock_rmk202_sub <-   Säuregrad_RMK_final%>% filter(!is.na(`D(-)%`))  %>% filter(`D(-)%` < 100) #%>% filter(as.numeric(`D(-)%`)<100)
table(ph_workingStock_rmk202_sub$year)

lactatePlot <- ggplot(ph_workingStock_rmk202_sub,aes(x=date,y=`D(-)%`,fill=RMK,color=RMK))+geom_point(alpha=0.2)+theme_classic()+scale_y_continuous(limits = c(0,100))+labs(y="Percent of D-lactate",x="time")+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9))+scale_fill_manual(values = colors)+scale_color_manual(values = colors)+geom_smooth()
 
 lactatePlot
 
 library(patchwork)
 


 png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement/historic_data_starterCultures.png", width = 2400, height = 2000,res=300)
# svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement/historic_data_starterCultures.png",width=6,height=6)


acidicationPlot+lactatePlot+ plot_layout(ncol=1)

dev.off()




```

##FINAL: starter culture historic data

```{r}
library(tsibble)
library(tidyverse)

colors <- c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#80e5ff","#008181","#00dcdc")

Säuregrad_RMK_final <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/historic_samples/Säuregrad_RMK_final_comasGONE.csv", 
    delim = "\t", escape_double = FALSE, 
    col_types = cols(RMK = col_character(), 
        `GMS [g/kg]` = col_double()), trim_ws = TRUE) %>% filter(week<54) %>% filter(!KZ %in% "Nur Listerien geprüft (wegen Krankheit)") %>% add_column(group="RMK") #%>% filter(`BK38°C`!="k.R.") 

table(Säuregrad_RMK_final$`GMS [g/kg]`)

Säuregrad_RMK_final$RMK  = factor(Säuregrad_RMK_final$RMK , levels=c("101","105","115","124","150","190","202","203","280","302","305") )
table(Säuregrad_RMK_final$RMK)


table(Säuregrad_RMK_final$RMK)

table(Säuregrad_RMK_final$KZ)
Säuregrad_RMK_final$KZ <- as.numeric(Säuregrad_RMK_final$KZ)
Säuregrad_RMK_final$`BK38°C` <- as.numeric(Säuregrad_RMK_final$`BK38°C`)
Säuregrad_RMK_final$`D(-)%` <-  Säuregrad_RMK_final$`D(-)%` %>% gsub(",",".",.) %>% as.numeric()


  # yearweek(Säuregrad_RMK_final$yearweekcol)
Säuregrad_RMK_final$week
 Säuregrad_RMK_final$date <-  MMWRweek::MMWRweek2Date(MMWRyear = Säuregrad_RMK_final$year,
                        MMWRweek = Säuregrad_RMK_final$week,
                        MMWRday = 3)
  
  Säuregrad_RMK_final$Versand
    Säuregrad_RMK_final$date
    
    Säuregrad_RMK_final_prep <- Säuregrad_RMK_final %>% filter(year>1996) %>% filter(year<2016) %>%  filter(!is.na(`BK38°C`)) 

#   
acidicationPlot <- ggplot(Säuregrad_RMK_final_prep,aes(x=date,y=`BK38°C`,fill=RMK,color=RMK))+geom_point(alpha=0.05)+
  theme_classic()+scale_y_continuous(limits = c(90,150))+labs(y="acid-base titration value of starter culture",x="time [year]")+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9))+geom_hline(yintercept = 110)+scale_fill_manual(values = colors)+scale_color_manual(values = colors)+geom_smooth(se = TRUE)

 acidicationPlot
 
 ##---add area
 
 Säuregrad_RMK_final_prep %>%  filter(!is.na(`BK38°C`)) %>% pull(`BK38°C`) %>% max()
maxndata <-  Säuregrad_RMK_final_prep %>%  filter(!is.na(date)) %>% pull(date) %>% max() 
mindata <- Säuregrad_RMK_final_prep %>%  filter(!is.na(date)) %>% pull(date) %>% min() 
 

acidicationPlot <- ggplot()+geom_rect(data=NULL,aes(xmin=mindata,xmax=maxndata,ymin=110,ymax=161),fill="green",alpha=0.1)+geom_point(data=Säuregrad_RMK_final_prep,aes(x=date,y=`BK38°C`,fill=RMK,color=RMK),alpha=0.05)+ theme_classic()+scale_y_continuous(limits = c(90,161),expand = c(0,0))+scale_x_continuous(expand = c(0,0))+labs(y="acid-base titration",x="time [year]")+theme(axis.text.x = element_blank(),axis.ticks.x=element_blank(),axis.title.x = element_blank(),legend.position = "none")+geom_hline(yintercept = 110)+scale_fill_manual(values = colors)+scale_color_manual(values = colors)+geom_smooth(data=Säuregrad_RMK_final_prep,aes(x=date,y=`BK38°C`,fill=RMK,color=RMK),se = TRUE)
 
acidicationPlot
 
 
 # max(as.numeric(Säuregrad_RMK_final_prep$`BK38°C`))
 
 
 
 

 ###-------------------------lactate plot




# GMS_02 <- ggplot()+geom_point(data=Säuregrad_RMK_final_subGSM,aes(x=date,y=`GMS [g/kg]`,fill=RMK,color=RMK),alpha=0.05)+ theme_classic()+scale_y_continuous(expand = c(0,0))+scale_x_continuous(expand = c(0,0))+labs(y="total lactate\n[g/kg]",x="time [year]")+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9))+geom_hline(yintercept = 110)+scale_fill_manual(values = colors)+scale_color_manual(values = colors)+geom_smooth(data=Säuregrad_RMK_final_subGSM,aes(x=date,y=`GMS [g/kg]`,fill=RMK,color=RMK),se = TRUE)
#  
# GMS_02

Säuregrad_RMK_final_subGSM <- Säuregrad_RMK_final_prep

Säuregrad_RMK_final_subGSM_prep <- Säuregrad_RMK_final_subGSM %>% filter(year>1996) %>% filter(year<2016) 

# Säuregrad_RMK_final_subGSM

GMS_02 <- ggplot()+geom_point(data=Säuregrad_RMK_final_subGSM,aes(x=date,y=`GMS [g/kg]`,fill=RMK,color=RMK),alpha=0.1)+theme_classic()+scale_y_continuous(limits = c(3,14), expand = c(0, 0))+scale_x_continuous(expand = c(0,0))+labs(y="total lactate\n[g/kg]",x="")+theme( axis.text.x = element_blank(),axis.ticks.x=element_blank(),legend.position = "none",axis.title.x = element_blank())+scale_fill_manual(values = colors)+scale_color_manual(values = colors)+geom_smooth(data=Säuregrad_RMK_final_subGSM,aes(x=date,y=`GMS [g/kg]`,fill=RMK,color=RMK),se = TRUE)
 #
 GMS_02




 
###-------------------------d-l lactate plot

# Säuregrad_RMK_final_subGSM$`GMS [g/kg]`

 Säuregrad_RMK_final_prep_02 <- Säuregrad_RMK_final_prep %>% filter(!is.na(`D(-)%`))%>% filter(`D(-)%`>0.1)%>% filter(`D(-)%`<99)
 table(Säuregrad_RMK_final_prep_02$RMK)
 
   Säuregrad_RMK_final_prep_02$date <-  MMWRweek::MMWRweek2Date(MMWRyear = Säuregrad_RMK_final_prep_02$year,
                        MMWRweek = Säuregrad_RMK_final_prep_02$week,
                        MMWRday = 3)
  
  # Säuregrad_RMK_final_prep_02$Versand
    Säuregrad_RMK_final_prep_02$date

 
d_l_02 <- ggplot()+geom_point(data=Säuregrad_RMK_final_prep_02,aes(x=date,y=`D(-)%`,fill=RMK,color=RMK),alpha=0.1)+theme_classic()+scale_y_continuous(limits = c(0,65), expand = c(0, 0))+scale_x_date(expand = c(0,0))+labs(y="D-lactate\n[%]",x="Date [year]")+theme(legend.position = "none",axis.text.x = element_text(angle = 75, hjust = 1,size=7))+scale_fill_manual(values = colors)+scale_color_manual(values = colors)+geom_smooth(data=Säuregrad_RMK_final_prep_02,aes(x=date,y=`D(-)%`,fill=RMK,color=RMK),se = TRUE) #+geom_smooth()+geom_hline(yintercept = 110) axis.text.x = element_blank(),axis.ticks.x=element_blank()
 
 d_l_02

 
 
 
 ###----------------------------------------------------
 ##all together
  ###----------------------------------------------------
 
library(patchwork)
acidicationPlot+GMS_02+d_l_02+ plot_layout(ncol=1)


pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/long_term_analysis_02.pdf",width=4.8,height=4.3)

acidicationPlot+GMS_02+d_l_02+ plot_layout(ncol=1)

dev.off()


```

### Do stats
```{r}
library(lme4)
library(lmerTest)

###-------------------lactactate ratio
# Fit the linear mixed effects model


# Säuregrad_RMK_final_prep_02,aes(x=date,y=`D(-)%`,fill=RMK,color=RMK
model <- lmer(`D(-)%` ~ date + (1 | RMK), data = Säuregrad_RMK_final_prep_02)

# Check the summary of the model
summary(model)
summary(model)$coefficients

###-------------------lactat
# Fit the linear mixed effects model


# Säuregrad_RMK_final_prep_02,aes(x=date,y=`D(-)%`,fill=RMK,color=RMK
model <- lmer(`GMS [g/kg]` ~ date + (1 | RMK), data = Säuregrad_RMK_final_prep_02)

# Check the summary of the model
summary(model)
summary(model)$coefficients

multivariate_lm <- lm(cbind(date, RMK) ~ `GMS [g/kg]`, data = Säuregrad_RMK_final_prep_02)
summary(multivariate_lm)



manova_result <- manova(cbind(date, RMK) ~ `GMS [g/kg]`, data = Säuregrad_RMK_final_prep_02)
summary(manova_result)


time_model <- lm(`GMS [g/kg]` ~ date, data = Säuregrad_RMK_final_prep_02)
summary(time_model)


time_model <- lm(`GMS [g/kg]` ~ RMK, data = Säuregrad_RMK_final_prep_02)
summary(time_model)

# multiple_model <- lm(`GMS [g/kg]` ~ RMK, data = Säuregrad_RMK_final_prep_02)
multiple_model <- lm(`GMS [g/kg]` ~ date + as.factor(RMK), data = Säuregrad_RMK_final_prep_02)

# multiple_model <- lm(acidification ~ time + as.factor(facility_id), data = cheese_data)
summary(multiple_model)

```


# NCBI upload


## Raw data
```{bash}
ll /data/Project/2020_StarterCultureDiversity/01_Data/

```


## SRA: metagenomic rawREADS

```{bash}
ll /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/


ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/ |wc -l

ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/ |wc -l


###----------other media #41

ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/ |cut -d '-' -f 1|sort|uniq|wc -l

###----------old samples  #46

ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/ |cut -d '-' -f 1|sort|uniq|wc -l
 
###-----------------------how many files

ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/ |cut -d '-' -f 1|sort|uniq -c

ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/ |cut -d '-' -f 1|sort|uniq -c


##-------202 #3
ll /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq
ll /archiv/Projects/2018_Culturomics/01_rawData/fastq/gz/ |grep "202"|wc -l
ls /archiv/Projects/2018_Culturomics/01_rawData/fastq/gz/ |grep "202" |grep "202_R" |cut -d '_' -f 1 |sort|uniq  |wc -l

##-------pilotplant  #4

ll /archiv/Projects/2019_pilotplant/01_rawData/
ll /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/ |grep "Lyo_" 

ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/ |grep "Lyo_"  |cut -d '-' -f 1 |sort|uniq |wc -l

###==============================================
#prep final 
###==============================================


mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/Prep_for_NCBI_submission/gz/

ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/ |wc -l


for namesss in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/)
do

echo ${namesss}
ll /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${namesss}
echo "-----------------------------"


cp /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${namesss}/${namesss}-R1_kneaddata.fq.gz   /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/Prep_for_NCBI_submission/gz/${namesss}-R1.fq.gz

cp /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${namesss}/${namesss}-R2_kneaddata.fq.gz   /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/Prep_for_NCBI_submission/gz/${namesss}-R2.fq.gz
done


ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/Prep_for_NCBI_submission/gz/ |cut -d '_' -f 1 |sort|uniq |wc -l

###==============================================
#upload files via aspera
###==============================================


##----get keyFILE
scp aspera.openssh vincent@130.223.110.142:/data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/Prep_for_NCBI_submission/ 



/home/vincent/.aspera/connect/bin/ascp -h


/home/vincent/.aspera/connect/bin/ascp -i /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/Prep_for_NCBI_submission/aspera.openssh -QT -l100m -k1 -d /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/Prep_for_NCBI_submission/gz/ subasp@upload.ncbi.nlm.nih.gov:uploads/vincent.somerville_agroscope.admin.ch_SuKFA98K

###==============================================
#which are missing
###==============================================


ll /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/Prep_for_NCBI_submission/gz/

ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/ |cut -d '_' -f 2 |sort|uniq -c

```



# Wetlab
### hydroplate

With the hydroplate I want to see if any of the Evolution experiment strains grows better/faster. 
I am only focusing on the Streptococcus thermophilus strains that where used in the evolution experiment:
First I create a list to see which strains come from which lineage

```{r }
library(plyr)
library(tidyverse)
genomeAssemblyStats <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",   col_types = cols(lineage = col_character(),starterCulture = col_character())) %>% select(strain,starterCulture,lineage)
Poppunk_out <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/Poppunk_out.all","\t", escape_double = FALSE, col_names = c("genome","group"), trim_ws = TRUE)%>% mutate(group=as.character(group))

Poppunk_out$group <- revalue(Poppunk_out$group,c("1"="8","2"="6","3"="4","4"="5","5"="1","6" ="3", "7"="7", "8"="11", "9"="10","10"="2","11"="12", "12"="9")) 

Poppunk_out$genome_02 <- gsub("S","",Poppunk_out$genome)

experiment_genomes <- c("24724","24726","24728","24730","24734","24736","24739","24742","24745","24748","24750")

Poppunk_out_experiment <- Poppunk_out %>% filter(genome_02 %in% experiment_genomes)

genomeAssemblyStats$genome <- gsub("S","",genomeAssemblyStats$strain)
genomeAssemblyStats_experiment <- genomeAssemblyStats %>% filter(genome %in% experiment_genomes)

genomeAssemblyStats_experiment$genome

tmp <- genomeAssemblyStats_experiment %>% select(strain,starterCulture)


Poppunk_out_experiment_02 <- Poppunk_out_experiment %>%  dplyr::rename("strain"="genome","genome"="genome_02","lineage"="group")
colnames(genomeAssemblyStats_experiment)

genomeAssemblyStats_experiment <- merge(Poppunk_out_experiment_02,tmp,by="strain") %>% select("strain"     ,    "starterCulture" ,"lineage"    ,    "genome" )

##-----------------
#for evolution experiment script
##-----------------
tmp_popunk_out <- genomeAssemblyStats_experiment %>% select(-genome)%>%  dplyr::rename("genome"="strain","group"="lineage") %>% select(genome,group)

write.table(tmp_popunk_out,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/Poppunk_sub_forSNVS.all_evolution",sep="\t",quote = FALSE,row.names = FALSE)

```

for the streptococcus strain I only need to look at the plate 1 and plate 2 samples. 

```{r}

library(readr)
library(tidyverse)
library(ggplot2)
library(patchwork)
library(ggmap)
library(ggsignif)
library(ggpubr)
#plate1
##leave r02 out... it is very strange as it first goes up then donw
##leave r01 out... first plate did not work

# replicate_01 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200701_hydroplates_PLATTE1/200701_plate_01_rmk_r01_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
# replicate_02 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200703_hydroplates_PLATTE1_r2/200703_plate_01_rmk_r02_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
replicate_03 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200706_hydroplates_PLATTE1_r3/200706_plate_01_rmk_r03_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
replicate_04 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200805_hydroplate_plate1_r04/200805_plate_01_rmk_r04_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
replicate_05 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200807_hydroplate_plate1_r05/200807_plate_01_rmk_r05_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)

#plate2
PLATE02_replicate_01 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200709_hydroplates_plate2_r01/200709_plate02_r01.txt_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
PLATE02_replicate_02 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200727_hydroplates_plate2_r03/200727_plate02_r02.txt_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)

#plate3
PLATE03_replicate_01 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200714_hydroplates_plate3_r01/200714_plate03_r01.txt_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
PLATE03_replicate_02 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200714_hydroplates_plate3_r02/200714_plate03_r02_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
# PLATE03_replicate_03 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200717_hydroplates_plate3_r03/200717_plate03_r03_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)

#plate4
PLATE04_replicate_01 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200724_hydroplates_plate4_r01/200724_plate04_r01_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
PLATE04_replicate_02 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200727_hydroplates_plate4_r02/200725_plate04_r02_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
# PLATE04_replicate_03 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200813_hydroplates_plate4_r03/200813_plate04_r03.txt_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)

# all_samples <- rbind(replicate_01,replicate_02,replicate_03)
# all_samples <- rbind(replicate_02,replicate_03,PLATE02_replicate_01,PLATE02_replicate_02,PLATE03_replicate_01,PLATE03_replicate_02,PLATE03_replicate_03,PLATE04_replicate_01,PLATE04_replicate_02)
# all_samples <- rbind(replicate_03,replicate_04,replicate_05,PLATE02_replicate_01,PLATE02_replicate_02,PLATE03_replicate_01,PLATE03_replicate_02,PLATE03_replicate_03,PLATE04_replicate_01,PLATE04_replicate_02,PLATE04_replicate_03)

all_samples <- rbind(replicate_03,replicate_04,replicate_05,PLATE02_replicate_01,PLATE02_replicate_02,PLATE03_replicate_01,PLATE03_replicate_02,PLATE04_replicate_01,PLATE04_replicate_02)

all_samples$lagPhase <- all_samples$h0/all_samples$mumax

##=================================
###only look at the streptococci. 
##=================================

all_samples_sub <- all_samples %>% filter(no_growth=="isolate")
all_samples_sub$newNames <- gsub("none\\+","",all_samples_sub$name)
all_samples_sub_isolates <- all_samples_sub %>% filter(newNames %in% genomeAssemblyStats_experiment$genome)

all_samples_sub_isolates_final <- merge(all_samples_sub_isolates,genomeAssemblyStats_experiment,by.x="newNames",by.y="genome")
table(all_samples_sub_isolates_final$newNames) 
table(all_samples_sub_isolates_final$lineage) 
table(genomeAssemblyStats_experiment$lineage)


samplesss <- all_samples %>% dplyr::select("name","no_growth") %>% unique()

mumax_plot <- ggplot(all_samples_sub_isolates_final,aes(x=lineage,y=mumax,fill=newNames))+geom_boxplot()+theme_classic()+labs(x="",y="maximum pH decrease")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7))+facet_wrap(~no_growth,scales = "free_x",nrow=1)#+facet_wrap(~plate,nrow=4)
mumax_plot
# ggplot(many_baranyi2_res_preped,aes(x=name,y=y0))+geom_boxplot()+theme_classic()+labs(x="","intitial pH")
k0plot <- ggplot(all_samples_sub_isolates_final,aes(x=lineage,y=K,fill=newNames))+geom_boxplot()+theme_classic()+labs(x="",y="lowest pH")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7))
# ggplot(many_baranyi2_res_preped,aes(x=name,y=h0))+geom_boxplot()+theme_classic()
k0plot

lagPhaseplot <- ggplot(all_samples_sub_isolates_final,aes(x=lineage,y=lagPhase,fill=newNames))+geom_boxplot()+theme_classic()+labs(x="",y="lagPhase")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7))
# ggplot(many_baranyi2_res_preped,aes(x=name,y=h0))+geom_boxplot()+theme_classic()
lagPhaseplot

mumax_plot+k0plot+lagPhaseplot+ plot_layout(ncol=1)


##=================================
###look at pairwise growth samples because Ldel might help lineage 4 disproportionally well. 
##=================================

all_samples_sub <- all_samples %>% filter(no_growth!="isolate")
all_samples_sub$newNames <- gsub("none\\+","",all_samples_sub$name)
all_samples_sub$newNames <-  str_split_fixed(all_samples_sub$name, fixed("+"), 2)[,2]
all_samples_sub_pairwise <- all_samples_sub %>% filter(newNames %in% genomeAssemblyStats_experiment$genome)



all_samples_sub_pairwise_final <- merge(all_samples_sub_pairwise,genomeAssemblyStats_experiment,by.x="newNames",by.y="genome")
table(all_samples_sub_pairwise_final$newNames) 
table(all_samples_sub_pairwise_final$lineage) 
table(genomeAssemblyStats_experiment$lineage)


samplesss <- all_samples %>% dplyr::select("name","no_growth") %>% unique()

mumax_plot_pair <- ggplot(all_samples_sub_pairwise_final,aes(x=lineage,y=mumax,fill=newNames))+geom_boxplot()+theme_classic()+labs(x="",y="maximum pH decrease")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7))+facet_wrap(~no_growth,scales = "free_x",nrow=1)#+facet_wrap(~plate,nrow=4)
mumax_plot_pair
# ggplot(many_baranyi2_res_preped,aes(x=name,y=y0))+geom_boxplot()+theme_classic()+labs(x="","intitial pH")
k0plot_pair <- ggplot(all_samples_sub_pairwise_final,aes(x=lineage,y=K,fill=newNames))+geom_boxplot()+theme_classic()+labs(x="",y="lowest pH")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7))
# ggplot(many_baranyi2_res_preped,aes(x=name,y=h0))+geom_boxplot()+theme_classic()
k0plot_pair

lagPhaseplot_pair <- ggplot(all_samples_sub_pairwise_final,aes(x=lineage,y=lagPhase,fill=newNames))+geom_boxplot()+theme_classic()+labs(x="",y="lagPhase")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7))
# ggplot(many_baranyi2_res_preped,aes(x=name,y=h0))+geom_boxplot()+theme_classic()
lagPhaseplot_pair

mumax_plot_pair+k0plot_pair+lagPhaseplot_pair+ plot_layout(ncol=1)


##=================================
###together
##=================================

mumax_plot_pair+mumax_plot+k0plot_pair+k0plot+lagPhaseplot_pair+lagPhaseplot+  plot_layout(ncol=2)

##=================================
###check labels
##=================================
all_samples_sub_pairwise_final %>% select(c(newNames,lineage,starterCulture)) %>% unique()


```

##### make plot

This needs to be prepared in the hydroplates script.

Then I import the raw data of the select samples.

```{r}

library(readr)
library(tidyverse)
library(ggplot2)
library(patchwork)
library(ggmap)
library(ggsignif)
library(ggpubr)
#plate1
##leave r02 out... it is very strange as it first goes up then donw
##leave r01 out... first plate did not work

# replicate_01 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200701_hydroplates_PLATTE1/200701_plate_01_rmk_r01_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
# replicate_02 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200703_hydroplates_PLATTE1_r2/200703_plate_01_rmk_r02_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
replicate_03_d <- read_rds("~/Desktop/Projects/plateReader/rawRuns/20200706_hydroplates_PLATTE1_r3/200706_plate_01_rmk_r03_all_data_experiment.rds")
replicate_04_d <- read_rds("~/Desktop/Projects/plateReader/rawRuns/20200805_hydroplate_plate1_r04/200805_plate_01_rmk_r04_all_data_experiment.rds")
replicate_05_d <- read_rds("~/Desktop/Projects/plateReader/rawRuns/20200807_hydroplate_plate1_r05/200807_plate_01_rmk_r05_all_data_experiment.rds")
# replicate_03_d <- read_rds("~/Desktop/Projects/plateReader/rawRuns/20200706_hydroplates_PLATTE1_r3/200706_plate_01_rmk_r03_all_data_experiment.rds")
# str(replicate_03_d)
# read_rds(paste0(location,sampleName,"_all_data_experiment.rds"))


#plate2
PLATE02_replicate_01_d <- read_rds("~/Desktop/Projects/plateReader/rawRuns/20200709_hydroplates_plate2_r01/200709_plate02_r01.txt_all_data_experiment.rds")
PLATE02_replicate_02_d <- read_rds("~/Desktop/Projects/plateReader/rawRuns/20200727_hydroplates_plate2_r03/200727_plate02_r02.txt_all_data_experiment.rds")

#plate3
# PLATE03_replicate_01_d <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200714_hydroplates_plate3_r01/200714_plate03_r01.txt_all_data_experiment.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
# PLATE03_replicate_02_d <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200714_hydroplates_plate3_r02/200714_plate03_r02_all_data_experiment.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
# PLATE03_replicate_03 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200717_hydroplates_plate3_r03/200717_plate03_r03_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)

#plate4
# PLATE04_replicate_01_d <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200724_hydroplates_plate4_r01/200724_plate04_r01_all_data_experiment.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
# PLATE04_replicate_02_d <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200727_hydroplates_plate4_r02/200725_plate04_r02_all_data_experiment.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)


# PLATE04_replicate_03 <- read_delim("~/Desktop/Projects/plateReader/rawRuns/20200813_hydroplates_plate4_r03/200813_plate04_r03.txt_model.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)

# all_samples <- rbind(replicate_01,replicate_02,replicate_03)
# all_samples <- rbind(replicate_02,replicate_03,PLATE02_replicate_01,PLATE02_replicate_02,PLATE03_replicate_01,PLATE03_replicate_02,PLATE03_replicate_03,PLATE04_replicate_01,PLATE04_replicate_02)
# all_samples <- rbind(replicate_03,replicate_04,replicate_05,PLATE02_replicate_01,PLATE02_replicate_02,PLATE03_replicate_01,PLATE03_replicate_02,PLATE03_replicate_03,PLATE04_replicate_01,PLATE04_replicate_02,PLATE04_replicate_03)

# all_samples <- rbind(replicate_03_d,replicate_04_d,replicate_05_d,PLATE02_replicate_01_d,PLATE02_replicate_02_d,PLATE03_replicate_01_d,PLATE03_replicate_02_d,PLATE04_replicate_01_d,PLATE04_replicate_02_d)
all_samples <- rbind(replicate_04_d,replicate_05_d,PLATE02_replicate_01_d,PLATE02_replicate_02_d) %>% filter(identfier!="24724_G4_1_5")
table(all_samples$Sterm)
# str(all_samples)
# 
# all_samples$TIMEfinal_02 <- as.
# 
# is.duration(all_samples$TIMEfinal)

# mutate(
#     days = duration(days, 'day'),
#     hourOld = duration(hourOld, 'hour'),
#     min = duration(min, 'minute'),
#     # sec = duration(sec, 'second'),
#     TIMEfinal = hourOld + days + min
#   )


# all_samples %>% 
#   mutate(
#     days = duration(days, 'day'),
#     hourOld = duration(hourOld, 'hour'),
#     min = duration(min, 'minute'),
#     # sec = duration(sec, 'second'),
#     TIMEfinal_02 = hourOld + days + min
#   )


htmlPrep <- ggplot(all_samples,aes(x=TIMEfinal,y=intensity_calibrated,group=identfier,fill=identfier,color=identfier))+geom_point()+theme_classic()+facet_wrap(~name)+labs(y="pH",x="")+lims(x=c(12000,86400))
ggp <- ggplotly(htmlPrep)
ggp


boxPrep <- ggplot(all_samples,aes(x=TIMEfinal,y=intensity_calibrated))+geom_boxplot()+theme_classic()+facet_wrap(~name)+labs(y="pH",x="")+lims(x=c(12000,86400))
ggp <- ggplotly(boxPrep)
ggp

htmlPrep <- ggplot(all_samples,aes(x=TIMEfinal,y=intensity_calibrated,group=identfier,fill=identfier,color=identfier))+geom_line()+theme_classic()+facet_wrap(~name)+labs(y="pH",x="")+lims(x=c(12000,86400))
ggp <- ggplotly(htmlPrep)
ggp

# str(hydroplate_wide_ph_calibration_05)


###-------------------------------------
##make a subsetting and averaging
###-------------------------------------

Poppunk_out_sub <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/Poppunk_sub_forSNVS.all_evolution", 
    "\t", escape_double = FALSE, trim_ws = TRUE) %>% mutate(genome=gsub("S","",genome))%>% mutate(group=paste0("lineage_",group))

# all_samples <- merge(all_samples,Poppunk_out_sub,by.x="Sterm",by.y="genome")

require(zoo)
# TS <- zoo(c(4, 5, 7, 3, 9, 8))
# rollapply(TS, width = 3, by = 2, FUN = mean, align = "left")
final_subsetting <- data_frame()
for (groupsss  in unique(all_samples$Sterm)) {
  # hydroplate_wide_ph_calibration_06$
  tmp <- all_samples %>% filter(Sterm==groupsss)%>% arrange(TIMEfinal)
  
  min <- rollapply(tmp$intensity_calibrated, width = 10, by = 5, FUN = min, align = "left")
  mean <- rollapply(tmp$intensity_calibrated, width = 10, by = 5, FUN = mean, align = "left")
  max <- rollapply(tmp$intensity_calibrated, width = 10, by = 5, FUN = max, align = "left")
 namesss <- tmp %>% filter(row_number() %% 5 == 1) 
 namesssss <- namesss[1:length(max),"TIMEfinal"]
  length(max)

 tmp02 <- data_frame(time=namesssss,grouping=groupsss,minum=min,maxum=max,median=mean)
  final_subsetting <- rbind(final_subsetting,tmp02)
}

table(final_subsetting$grouping)


 final_subsetting_subset <- final_subsetting %>% filter(time<86400)

final_subsetting_subset$timeFinal <- final_subsetting_subset$time/3600

final_subsetting_subset_final <- final_subsetting_subset %>% filter(grouping!="complex")

minTime <- min(final_subsetting_subset_final$timeFinal)

# colorsss <- c("#0081a7ff","#fec3b7ff","#9f9f92ff","#6d6466ff","#4e3d42ff")
# colorss_02 <- c("#fec3b7ff","#9f9f92ff","#6d6466ff","#0081a7ff")


low_phased_ribbon <- ggplot()+
    geom_ribbon(aes(x=timeFinal,ymin = minum , ymax = maxum,group=grouping,fill=grouping),data = final_subsetting_subset_final, alpha=.4)+
       # scale_x_discrete( expand = c(0, 0)) +
  # facet_wrap(~grouping,scales = "free")+
    labs(y="pH",x="incubation time")+
  # scale_fill_manual(values = colorss_02)+
    scale_x_continuous(breaks =c(0,6,12,18),labels=c(0,6,12,18))+
  theme_classic()+
 theme(rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )


  low_phased_ribbon
  

###-------------------------------------
##make a subsetting and averaging
  #ON LINEAGE
###-------------------------------------

Poppunk_out_sub <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/Poppunk_sub_forSNVS.all_evolution", 
    "\t", escape_double = FALSE, trim_ws = TRUE) %>% mutate(genome=gsub("S","",genome))%>% mutate(group=paste0("lineage_",group))

all_samples_new <- merge(all_samples,Poppunk_out_sub,by.x="Sterm",by.y="genome") %>% filter(TIME<86400)
table(all_samples_new$Sterm)

all_samples_new %>% as_tibble()%>% select(Sterm,group) %>% unique()

 # final_subsetting_subset <- final_subsetting %>% filter(time<86400)

require(zoo)
# TS <- zoo(c(4, 5, 7, 3, 9, 8))
# rollapply(TS, width = 3, by = 2, FUN = mean, align = "left")
final_subsetting <- data_frame()
for (groupsss  in unique(all_samples_new$group)) {
  # hydroplate_wide_ph_calibration_06$
  tmp <- all_samples_new %>% filter(group==groupsss)%>% arrange(TIME)
  sampleNumber <- round(nrow(tmp)/25,digits = 0)
  
  min <- rollapply(tmp$intensity_calibrated, width = sampleNumber, by = round(sampleNumber/2,digits = 0), FUN = min, align = "left")
  mean <- rollapply(tmp$intensity_calibrated, width = sampleNumber, by = round(sampleNumber/2,digits = 0), FUN = mean, align = "left")
  max <- rollapply(tmp$intensity_calibrated, width = sampleNumber, by = round(sampleNumber/2,digits = 0), FUN = max, align = "left")
 namesss <- tmp %>% filter(row_number() %% round(sampleNumber/2,digits = 0) == 1) 
 namesssss <- namesss[1:length(max),"TIME"]
  length(max)

 tmp02 <- data_frame(time=namesssss,grouping=groupsss,minum=min,maxum=max,median=mean)
  final_subsetting <- rbind(final_subsetting,tmp02)
}

table(final_subsetting$grouping)


 final_subsetting_subset <- final_subsetting %>% filter(time<86400)
table(final_subsetting_subset$grouping)

final_subsetting_subset$timeFinal <- final_subsetting_subset$time/3600

final_subsetting_subset_final <- final_subsetting_subset %>% filter(grouping!="complex")

minTime <- min(final_subsetting_subset_final$timeFinal)

# colorsss <- c("#0081a7ff","#fec3b7ff","#9f9f92ff","#6d6466ff","#4e3d42ff")
# colorss_02 <- c("#fec3b7ff","#9f9f92ff","#6d6466ff","#0081a7ff")


low_phased_ribbon <- ggplot()+
    # geom_ribbon(aes(x=timeFinal,ymin = minum , ymax = maxum,group=grouping,fill=grouping),data = final_subsetting_subset_final, alpha=.1)+
  geom_line(aes(x=timeFinal,y=median,group=grouping,color=grouping),data = final_subsetting_subset_final)+
       # scale_x_discrete( expand = c(0, 0)) +
  # facet_wrap(~grouping,scales = "free")+
    labs(y="pH",x="phased incubation time")+
  # lims(x=c(3,18))+
  # scale_fill_manual(values = colorss_02)+
    scale_x_continuous(limits = c(2,18),breaks =c(6,12,18),labels=c(6,12,18))+
  theme_classic()+
 theme(rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )


  low_phased_ribbon

ggp <- ggplotly(low_phased_ribbon)
ggp

##-----------------------------------------
##optimise
##-----------------------------------------

final_subsetting_subset_final$grouping  = factor(final_subsetting_subset_final$grouping , levels=c("lineage_1","lineage_3","lineage_4","lineage_5","lineage_6","lineage_7","lineage_8","lineage_11"))
table(final_subsetting_subset_final$grouping)

levels(final_subsetting_subset_final$grouping)
  cols <- rainbow(14)[c(1,3,4,5,6,7,8,11)] #I am not sure this is correct -- did not find the correct color labels anymore
lowss <- 0.4
highss <- 1
  alps <- c(lowss,highss,lowss,lowss,lowss,lowss,lowss,lowss)
     
smallss <- "dotted"
linesss <- c(smallss,"solid",smallss,smallss,smallss,smallss,smallss,smallss)

smallss <- 1.2

sizesss <- c(smallss,2,smallss,smallss,smallss,smallss,smallss,smallss)

  # ggplot(sample_medians_normalised_subset,aes(x=replicate,y=normalised_ratio,fill=groupAssignment))+geom_bar(position="stack", stat="identity",alpha=0.5)+theme_classic()+facet_grid(~timepoint,scales = "free_x", space = "free")+scale_fill_manual(values = cols)+scale_color_manual(values = cols)+labs(y="relative abundance")


low_phased_ribbon <- ggplot()+
    # geom_ribbon(aes(x=timeFinal,ymin = minum , ymax = maxum,group=grouping,fill=grouping),data = final_subsetting_subset_final, alpha=.1)+
  geom_line(aes(x=timeFinal,y=median,group=grouping,color=grouping,alpha=grouping,linetype=grouping,size=grouping),data = final_subsetting_subset_final)+
  scale_fill_manual(values = cols)+scale_color_manual(values = cols)+scale_alpha_manual(values = alps)+scale_linetype_manual(values = linesss)+scale_size_manual(values = sizesss)+
       # scale_x_discrete( expand = c(0, 0)) +
  # facet_wrap(~grouping,scales = "free")+
    labs(y="pH",x="incubation time")+
  # lims(x=c(3,18))+
  # scale_fill_manual(values = colorss_02)+
    scale_x_continuous(limits = c(2,18),breaks =c(6,12,18),labels=c(6,12,18))+
  theme_classic()+
 theme(rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )


  low_phased_ribbon
  
     pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/acidification_evolution.pdf",width=6,height=4)

  low_phased_ribbon

dev.off()
  

```


## Survival freez drying

Here, I want to look at the survival rate after freeze drying. 

#### colony pick genotypes

```{r}
library(readr)
library(plyr)
library(tidyverse)
library(ggplot2)
strain_counts_of_plates <- read_delim("~/Desktop/Projects/2020_strainDelineation/01_logs/GENOTYPE_count.csv",  "\t", escape_double = FALSE, trim_ws = TRUE)
strain_counts_of_plates <- read_delim("~/Desktop/Projects/2020_strainDelineation/01_logs/GENOTYPE_count.csv",  ",", escape_double = FALSE, trim_ws = TRUE)

# strain_counts_of_plates$sample <-
strain_counts_of_plates$`starter culture` <-  as.character(strain_counts_of_plates$`starter culture`)
strain_counts_of_plates$genotypes_extended <- paste(strain_counts_of_plates$`starter culture`,strain_counts_of_plates$species,strain_counts_of_plates$genotype,sep="_")
# ggplot(strain_counts_of_plates,aes(x=`starter culture`,color=genotype,fill=genotype))+geom_bar(aes(y=(..counts..)/sum(..counts..)),stat="identity")+facet_wrap(~species,scales = "free")
genotypePlot <- ggplot(strain_counts_of_plates,aes(x=`starter culture`,color=genotype,fill=genotype,text =paste("genotype:", genotypes_extended)))+geom_bar(aes(y=count),stat="identity")+labs(x="",y="genotype counts")+facet_wrap(~species,scales = "free")+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))
genotypePlot

table(interaction(strain_counts_of_plates$genotype,strain_counts_of_plates$species))
grep("NA",strain_counts_of_plates$genotypes_extended,invert = TRUE) %>% length()
grep("NA",strain_counts_of_plates$genotypes_extended,invert = TRUE) %>% length()
strain_counts_of_plates[grep("NA",strain_counts_of_plates$genotypes_extended,invert = TRUE),] %>% filter(species=="S_thermophilus") %>% nrow()
strain_counts_of_plates[grep("NA",strain_counts_of_plates$genotypes_extended,invert = TRUE),] %>% filter(species=="L_delbrueckii") %>% nrow()

 svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/genotypeAbundance.svg",width=6,height=3)


genotypePlot+theme(legend.position = "none")

dev.off()


library(plotly)
ggp <- ggplotly(genotypePlot)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/genotypeAbundance.html")

unique(strain_counts_of_plates$genotypes_extended)
strain_counts_of_plates$genotypes_extended <- revalue(strain_counts_of_plates$genotypes_extended,c("RMK101_L_delbrueckii_A"="dominant","RMK105_L_delbrueckii_B"="dominant","RMK115_L_delbrueckii_B"="dominant","RMK124_L_delbrueckii_J"="dominant","RMK150_L_delbrueckii_J"="dominant","RMK190_L_delbrueckii_B" ="dominant", "RMK202_L_delbrueckii_T"="dominant", "RMK203_L_delbrueckii_J"="dominant", "RMK280_L_delbrueckii_R"="dominant","RMK302_L_delbrueckii_B"="dominant","RMK305_L_delbrueckii_H"="dominant", "RMK101_S_thermophilus_a"="dominant","RMK105_S_thermophilus_c"="dominant","RMK115_S_thermophilus_c"="dominant","RMK124_S_thermophilus_e"="dominant","RMK150_S_thermophilus_a"="dominant", "RMK190_S_thermophilus_I"="dominant", "RMK202_S_thermophilus_p"="dominant","RMK203_S_thermophilus_l"="dominant","RMK280_S_thermophilus_m"="dominant", "RMK302_S_thermophilus_s"="dominant","RMK305_S_thermophilus_e"="dominant")) 
strain_counts_of_plates$genotypes_extended <- ifelse(strain_counts_of_plates$genotypes_extended=="dominant","dominant","subdominant")
strain_counts_of_plates$genotype_dominant <- NA
for (i in 1:nrow(strain_counts_of_plates)) {
  strain_counts_of_plates[i,"genotype_dominant"] <- ifelse(strain_counts_of_plates[i,"genotypes_extended"]=="dominant",as.character(strain_counts_of_plates[i,"genotype"]),NA)
  
}

strain_counts_of_plates$names <- as.character(gsub("RMK","",strain_counts_of_plates$`starter culture`))
genotypePlot <- ggplot(strain_counts_of_plates,aes(x=names,fill=genotype_dominant,color=genotype,text =paste("genotype:", genotypes_extended)))+geom_bar(aes(y=count),stat="identity", colour="#FF9999")+labs(x="",y="colony counts")+facet_wrap(~species,scales = "free")+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+scale_fill_discrete(na.value="grey81")
genotypePlot

 # png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/genotypeAbundance_onlyDominant.png", width = 2400, height = 1200,res=300)
svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/genotypeAbundance_onlyDominant.svg",width=6,height=3)


genotypePlot

dev.off()

###---------------------
##for mid-thesis

genotypePlot <- ggplot(strain_counts_of_plates,aes(x=names,fill=genotype_dominant,color=genotype,text =paste("genotype:", genotypes_extended)))+geom_bar(aes(y=count),stat="identity", colour="#FF9999")+labs(x="",y="colony counts")+facet_wrap(~species,scales = "free",nrow = 2)+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+scale_fill_discrete(na.value="grey81")+scale_y_continuous(breaks =c(0,25,50,75,96),labels=c(0,25,50,75,96))
genotypePlot

 # png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/genotypeAbundance_onlyDominant.png", width = 2400, height = 1200,res=300)
svg("~/Desktop/mid_thesis/report/figures/20200430/chapter3/genotypeAbundance_onlyDominant.svg",width=2.5,height=5)


genotypePlot

dev.off()


```

## d/l lactate
The ration d- to l-lactat gives an idea of L.delbreuckii to S. thermophilus
Here, I plot the ratios measured in 2015 of all Starter cultures in use.

```{r}
library(readr)
library(ggplot2)
library(tidyverse)
d_l_lactat_2015 <- read_delim("~/Desktop/Projects/2020_strainDelineation/06_RMKStrainStock/d_l_lactat_2015.csv",  "\t", escape_double = FALSE, trim_ws = TRUE)
d_l_lactat_2015$sample <- paste("RMK",as.character(d_l_lactat_2015$sample))

colors <- c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#80e5ff","#008181","#00dcdc")


plotl_d <- ggplot(d_l_lactat_2015,aes(x=sample,y=L_LACTAT,group=sample,fill=sample))+geom_boxplot()+theme_classic()+ylim(0,100)+theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1),legend.position = "none")+labs(x="",y= "l-lactat [%]")+scale_fill_manual(values = colors)

plotl_d

# svg("~/Desktop/Projects/2020_strainDelineation/06_RMKStrainStock/plot_d_l_lactat.svg",width=4,height=4)
png("~/Desktop/Projects/2020_strainDelineation/06_RMKStrainStock/plot_d_l_lactat.png", width = 1800, height = 1600,res=300)
plotl_d
dev.off()

```




##Cell counts
Here, I first download the cell counts sheet from google sheets. 
This sheet also includes the columns abount the generation time, doubling time and survival rate. 
From this information I plot the following parameters

```{r}
##----------------------
##import data
##----------------------
library(readr)
library(ggplot2)
library(plyr)
library(tidyverse)
library(patchwork)
CellCount <- read_csv("~/Desktop/Projects/2020_StarterEvolutionExperiment/01_cellCount/rmk202_test_cellCounts.csv", skip = 2)

CellCount$step <- revalue(CellCount$step,c("first_passage"="first passage","second_passage"="second passage","freeze_dry"="freeze dry &\nstorage"))



##----------------------
##generation
##----------------------
colnames(CellCount)
table(CellCount$step)
CellCount_generations <- CellCount %>% filter(step %in% c("first passage","second passage")) %>% filter(!is.na(generations_all)) %>% select(c(sample,treatment,step,generations_all,generations_sterm,generations_ldel)) %>% gather(., sample, generations, c(generations_all,generations_sterm,generations_ldel), factor_key=TRUE,na.rm = TRUE) 
colors <- c("grey","#EB4D4D","#10B552")
CellCount_generations$sample <- revalue(CellCount_generations$sample,c("generations_all"="All","generations_sterm"="S. thermophilus","generations_ldel"="L. delbrueckii"))

generationsPlot <- ggplot(CellCount_generations,aes(x=step,y=generations,fill=sample))+geom_boxplot()+theme_classic()+theme(legend.title = element_blank(),legend.position = "top",axis.title.x = element_blank())+scale_fill_manual(values=colors)

##----------------------
##survival rate
##----------------------

colnames(CellCount)
table(CellCount$step)
CellCount_survival <- CellCount %>% filter(step %in% c("freeze dry &\nstorage")) %>% filter(!is.na(generations_all)) %>% select(c(sample,treatment,step,survival_rate_all,survival_rate_sterm,survival_rate_ldel)) %>% gather(., sample, "survival rate", c(survival_rate_all,survival_rate_sterm,survival_rate_ldel), factor_key=TRUE,na.rm = TRUE) %>% filter(sample!="survival_rate_ldel")
colors <- c("grey","#EB4D4D","#10B552")
CellCount_survival$`survival rate`
CellCount_survival$sample <- revalue(CellCount_survival$sample,c("survival_rate_all"="All","survival_rate_sterm"="S. thermophilus","survival_rate_ldel"="L. delbrueckii"))

SurvivalPlot <- ggplot(CellCount_survival,aes(x=step,y=`survival rate`,fill=sample))+geom_boxplot()+theme_classic()+theme(legend.title = element_blank(),legend.position = "top",axis.title.x = element_blank())+scale_fill_manual(values=colors)
SurvivalPlot
##----------------------
##survival rate
##----------------------

# 
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/survival_rate",width=6,height=8)
# SurvivalPlot
# dev.off()

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/survival_rate.pdf",width=4,height=4)
# svg("~/Desktop/mid_thesis/report/figures/20200430/chapter3/generations_survivalRate.svg",width=4,height=4)
# png("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.png", width = 1800, height = 1600,res=300)
generationsPlot+SurvivalPlot+ plot_layout(ncol=2,widths = c(2,0.8))
dev.off()


##----------------------
##over time 
#in theroy it should increase over time
##----------------------




```

#metabolomics
Here, I analyse the metabolomics data I received from the GC/MS of pascal an yihelen on 13.8.2020.
```{r}
library(openxlsx)
library(readr)
library(ggbiplot)
library(tidyverse)
library(patchwork)
library(plotly)
X20200810_Vincent_cultures_profinder_untargeted <- read.xlsx("~/Desktop/Projects/2020_StarterCultureDiversity/GCMS/210610_vincent_yogurt_expt2_wc.xlsx")
# pca_res <- prcomp(df, scale. = TRUE)

###----------------------if we filter metabolites based on size

# library(readr)
# Metabolites_information_reduced <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/Metabolites_information_reduced.csv",  "\t", escape_double = FALSE, trim_ws = TRUE)
# 
# nrow(Metabolites_information_reduced)
# par(mfrow=c(1,1))
# hist(Metabolites_information_reduced$`RT (avg)`)
# Metabolites_information_reduced_goodOnes <- Metabolites_information_reduced %>% filter(`RT (avg)`<46.6)
# nrow(Metabolites_information_reduced_goodOnes)
# # colnames(X20200810_Vincent_cultures_profinder_untargeted)[1:4]
# X20200810_Vincent_cultures_profinder_untargeted <-   X20200810_Vincent_cultures_profinder_untargeted %>% dplyr::select(c(colnames(X20200810_Vincent_cultures_profinder_untargeted)[1:4],Metabolites_information_reduced_goodOnes$compounds))
# # Metabolites_information_reduced_goodOnes

###----------------------if we DONT filter metabolites based on size

# colnames(X20200810_Vincent_cultures_profinder_untargeted)[1:3,1:10]
# X20200810_Vincent_cultures_profinder_untargeted[1:3,1:10]
##------------------remove zero columns
X20200810_Vincent_cultures_profinder_untargeted_new <- X20200810_Vincent_cultures_profinder_untargeted[,!(colnames(X20200810_Vincent_cultures_profinder_untargeted)%in% names(which(colSums(X20200810_Vincent_cultures_profinder_untargeted[,5:ncol(X20200810_Vincent_cultures_profinder_untargeted)])==0)))]

dim(X20200810_Vincent_cultures_profinder_untargeted)
dim(X20200810_Vincent_cultures_profinder_untargeted_new)

# which(colSums(X20200810_Vincent_cultures_profinder_untargeted_new[,5:ncol(X20200810_Vincent_cultures_profinder_untargeted_new)])==0)
# ploting_data

####exclude analytic replicate 1
tmp <- ploting_data %>% filter(analytical_duplicate=="1")
samples_to_exclude <-  X20200810_Vincent_cultures_profinder_untargeted$sample[as.numeric(rownames(tmp))]

X20200810_Vincent_cultures_profinder_untargeted_new <- X20200810_Vincent_cultures_profinder_untargeted_new %>% filter(!sample %in% samples_to_exclude)

metabolo.pca <- prcomp(X20200810_Vincent_cultures_profinder_untargeted_new[,5:ncol(X20200810_Vincent_cultures_profinder_untargeted_new)], center = TRUE,scale. = TRUE)
metabolo.pca <- prcomp(X20200810_Vincent_cultures_profinder_untargeted_new[,5:ncol(X20200810_Vincent_cultures_profinder_untargeted_new)], center = FALSE,scale. = FALSE)

metabolo.sum <- summary(metabolo.pca)
metabolo.sum
metabolo.sum <- metabolo.sum$importance %>% as.data.frame()



ploting_data <- metabolo.pca$x %>% as.data.frame()
ploting_data$sample <- X20200810_Vincent_cultures_profinder_untargeted_new$sample %>% gsub("^1","",.) %>% gsub("^2","",.)%>% gsub("^3","",.)%>% gsub("a$","",.)%>% gsub("b$","",.)

# gsub("[^[:digit:].[:letter:].[:letter:]., ]", "", classification_SNVs_lineage$details) %>% as.numeric()


ploting_data$biological_duplicate<- X20200810_Vincent_cultures_profinder_untargeted_new$sample %>% gsub("^[[:digit:]][[:upper:]][[:digit:]]", "",.) 


ploting_data$analytical_duplicate<- X20200810_Vincent_cultures_profinder_untargeted_new$sample %>% gsub("[[:upper:]][[:digit:]][[:lower:]]$", "",.) 

X200701_names <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/GCMS/200701_names.csv")

ploting_data$sample

plotting_data_tmp <- merge(ploting_data,X200701_names,by.x="sample",by.y="well")
plotting_data_tmp$no_growth

# ploting_data$biological_duplicate <- X20200810_Vincent_cultures_profinder_untargeted$biological_duplicate
# ploting_data$analytical_duplicate <- X20200810_Vincent_cultures_profinder_untargeted$analytical_duplicate
ploting_data$grouping <- plotting_data_tmp$no_growth
ploting_data$grouping_long <- plotting_data_tmp$no_growth
table(ploting_data$grouping_long)
# ploting_data$grouping <- substring(plotting_data_tmp$no_growth, 1, 1)
# table(ploting_data$grouping)
# ploting_data$grouping_long  <- plyr::revalue(ploting_data$grouping, c("A"="All strains combined","L"="L. delbrueckii only","S"="S. thermophilus only","R"="original RMK202 starter culture","M"="pairwise strain mix"))

# pca1 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1,text =paste("sample: ",ploting_data$sample,"\nbio. dupli.: ",ploting_data$biological_duplicate, "\ntec.dupli.: ",ploting_data$analytical_duplicate))+theme_classic()
# 
# ggp <- ggplotly(pca1)
# ggp
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca.html")


ploting_data$analytical_duplicate
pca0 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$biological_duplicate,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by biological replicates")+theme(legend.position = "right")
pca1 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$analytical_duplicate,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by analytic replicates")+theme(legend.position = "right")
pca2 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by treatment")+theme(legend.position = "right")
pca3 <- ggbiplot(metabolo.pca,var.axes=TRUE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by treatment")+theme(legend.position = "right")

pca0+pca1+pca2+plot_layout(nrow = 3)


png("~/Desktop/Projects/2020_StarterCultureDiversity/GCMS//20200810_Vincent_cultures_pca_01.png", width = 1500, height = 2900,res=300)
 # svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
pca0+pca1+pca2+plot_layout(nrow = 3)
dev.off()


##-----------------------
##----------------------subset
##-----------------------

metabolo.pca
table(X200701_names$no_growth)

keppers <- X200701_names %>% filter(no_growth %in% c("blank","Ldel","Sterm","original"))

table(X20200810_Vincent_cultures_profinder_untargeted_new$sample)

X20200810_Vincent_cultures_profinder_untargeted_sub <- X20200810_Vincent_cultures_profinder_untargeted_new %>% filter(sample %in% keppers$)


##-----------------------
##----------------------NON-ADJUSTED SCALES
##-----------------------
metabolo.pca <- prcomp(X20200810_Vincent_cultures_profinder_untargeted_new[,5:ncol(X20200810_Vincent_cultures_profinder_untargeted_new)], center = FALSE,scale. = FALSE)
metabolo.sum <- summary(metabolo.pca)
metabolo.sum
metabolo.sum <- metabolo.sum$importance %>% as.data.frame()


# pca1 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1,text =paste("sample: ",ploting_data$sample,"\nbio. dupli.: ",ploting_data$biological_duplicate, "\ntec.dupli.: ",ploting_data$analytical_duplicate))+theme_classic()
# 
# ggp <- ggplotly(pca1)
# ggp
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca.html")


pca4 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$biological_duplicate,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by biological replicates")
pca5 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by treatment")
pca6 <- ggbiplot(metabolo.pca,var.axes=TRUE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by treatment")



pca7 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$sample,ellipse=FALSE,obs.scale = 1)+theme_classic()+labs(title="grouping by sample")
pca7
 ggp <- ggplotly(pca7)
ggp

pca7_2 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$analytical_duplicate,ellipse=FALSE,obs.scale = 1)+theme_classic()+labs(title="grouping by analytic samples")
pca7_2
 ggp <- ggplotly(pca7_2)
ggp

 pca1+pca2+plot_layout(nrow = 2)


png("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca_02.png", width = 1500, height = 2100,res=300)
 # svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
pca1+pca2+plot_layout(nrow = 2)
dev.off()



png("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca_explained.png", width = 2500, height = 1800,res=300)
 # svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
pca6
dev.off()

pca1+pca4+pca2+pca5+pca3+pca6+plot_layout(nrow = 3)

png("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca_04.png", width = 3000, height = 4200,res=300)
 # svg("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca_04.svg",width=15,height=21)
pca1+pca4+pca2+pca5+pca3+pca6+plot_layout(nrow = 3)
dev.off()



colorsss <- c("0081a7ff","fec3b7ff","9f9f92ff","6d6466ff","4e3d42ff")

unique(ploting_data$grouping_long)

colorsss <- c("#4e3d42ff","#fec3b7ff","#9f9f92ff","#6d6466ff","#0081a7ff")


pcaFINAL <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1,size=2)+theme_classic()+scale_color_manual(values=colorsss)+scale_fill_manual(values=colorsss)
pcaFINAL

svg("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca_FINAL.svg",width=8,height=5)
pcaFINAL
dev.off()


###------------------------------------
##remove all isolates mixed
###------------------------------------









##-----------------------
##----------------------unnormalised metabolo
##-----------------------
unnormalized_metabolo <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_profinder_untargeted_paraldehyde-normalised.csv", ";", escape_double = FALSE, trim_ws = TRUE)

unnormalized_metabolo_new <- unnormalized_metabolo[,!(colnames(unnormalized_metabolo)%in% names(which(colSums(unnormalized_metabolo[,5:ncol(unnormalized_metabolo)])==0)))]

dim(unnormalized_metabolo)
dim(unnormalized_metabolo_new)

# which(colSums(X20200810_Vincent_cultures_profinder_untargeted_new[,5:ncol(X20200810_Vincent_cultures_profinder_untargeted_new)])==0)


metabolo.pca <- prcomp(unnormalized_metabolo_new[,5:ncol(unnormalized_metabolo_new)], center = TRUE,scale. = TRUE)
metabolo.sum <- summary(metabolo.pca)
metabolo.sum
metabolo.sum <- metabolo.sum$importance %>% as.data.frame()

pca1 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1,text =paste("sample: ",ploting_data$sample,"\nbio. dupli.: ",ploting_data$biological_duplicate, "\ntec.dupli.: ",ploting_data$analytical_duplicate))+theme_classic()
# 
# ggp <- ggplotly(pca1)
# ggp
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca.html")
# 

pca1 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$biological_duplicate,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by biological replicates")
pca2 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by treatment")
pca3 <- ggbiplot(metabolo.pca,choices=c(2,3),var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by treatment")

pca1+pca2+plot_layout(nrow = 2)


png("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca_03.png", width = 1500, height = 2100,res=300)
 # svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
pca1+pca2+plot_layout(nrow = 2)
dev.off()


# ggp <- ggplotly(pca1)
# ggp
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca.html")
# 
# pca1Prop <- metabolo.sum["Proportion of Variance","PC1"]*100 
# pca1Prop <- round(pca1Prop,digits =1 )
# pca2Prop <- metabolo.sum["Proportion of Variance","PC2"]*100
# pca2Prop <- round(pca2Prop,digits =1 )
# 

# autoplot(metabolo.pca)
# 
# 
# pca1 <- ggplot(ploting_data,aes(x=PC1,y=PC2,color=grouping_long,fill=grouping_long,text =paste("sample: ",ploting_data$sample,"\nbio. dupli.: ",ploting_data$biological_duplicate, "\ntec.dupli.: ",ploting_data$analytical_duplicate)))+geom_point(size=2)+theme_classic()+labs(x=paste0("PC1 [explained=",pca1Prop,"%]"),y=paste0("PC1 [explained=",pca2Prop,"%]"))
# 
# ggp <- ggplotly(pca1)
# ggp
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca.html")
# 
# 
# ploting_data$PC18
# plot.new()
# par(mfrow=c(1,1))
# plot(metabolo.pca$x[,1],metabolo.pca$x[,2],)
##========================================================================================================
##------------------------------------------------
##z-Score calculation
##------------------------------------------------

tmpss <- t(X20200810_Vincent_cultures_profinder_untargeted_new[5:3085])

dim(X20200810_Vincent_cultures_profinder_untargeted_new)
for (metaolsss in colnames(X20200810_Vincent_cultures_profinder_untargeted_new)[5:3085]) {
  tmp <- X20200810_Vincent_cultures_profinder_untargeted_new %>% mutate(zscore = (metaolsss - mean(metaolsss))/sd(metaolsss)) %>% select(zscore)
}


dat %>% mutate(zscore = (BMI - mean(BMI))/sd(BMI))


##--------------------------------------------------------------------
##test
##--------------------------------------------------------------------

##multiple T-test function
ttestRat <- function(df, grp1, grp2) {
  x = df[grp1]
  y = df[grp2]
  x = as.numeric(x)
  y = as.numeric(y)  
  results = t.test(x, y)
  results$p.value
}


ttestmaxMEAN <- function(df, grp1, grp2) {
  x = df[grp1]
  y = df[grp2]
  x = as.numeric(x)
  y = as.numeric(y)  
  results = t.test(x, y)
  max(results$estimate)
}

##prepare for function
X20200810_Vincent_cultures_profinder_untargeted_new[,2]

X20200810_Vincent_cultures_profinder_untargeted_new[,2][c(1:6,13:34,51:56,63:84)]
X20200810_Vincent_cultures_profinder_untargeted_new[,2][c(7:12,35:50,57:62,85:100)]

# metabolite=10
# results2 <- t.test(X20200810_Vincent_cultures_profinder_untargeted_new[ c(1:6,13:34,51:56,63:84),metabolite],X20200810_Vincent_cultures_profinder_untargeted_new[ c(7:12,35:50,57:62,85:100),metabolite])
# max(results2$estimate)
metagData <- t(X20200810_Vincent_cultures_profinder_untargeted_new[,5:ncol(X20200810_Vincent_cultures_profinder_untargeted_new)])
dim(metagData)
###aply function
rawpvalue = apply(metagData, 1, ttestRat, grp1 = c(1:6,13:34,51:56,63:84), grp2 =c(7:12,35:50,57:62,85:100))
hist(rawpvalue)

###----fold change
##transform our data into log2 base.
metagData_log = log2(metagData)

#calculate the mean of each gene per control group
control = apply(metagData_log[,c(1:6,13:34,51:56,63:84)], 1, mean)

#calcuate the mean of each gene per test group
test = apply(metagData_log[, c(7:12,35:50,57:62,85:100)], 1, mean) 

#confirming that we have a vector of numbers
class(control) 
foldchange <- control - test 

hist(foldchange, xlab = "log2 Fold Change (Control vs Test)")


###maxMEANvalue

rawpmaxMean = apply(metagData, 1, ttestmaxMEAN, grp1 = c(1:6,13:34,51:56,63:84), grp2 =c(7:12,35:50,57:62,85:100))
hist(log10(rawpmaxMean))

##----volcano

results = cbind(foldchange, rawpvalue,MeanExpression=log10(rawpmaxMean))
results = as.data.frame(results)
results$probename <- rownames(results)

results$fill <- ifelse(results$foldchange>2&-log10(results$rawpvalue)>10,"significant","NS")
table(results$fill)
colorsss <- c("grey","red")

library(ggplot2)
volcano = ggplot(data = results, aes(x = foldchange, y = -1*log10(rawpvalue),color=fill,size=MeanExpression))+theme_classic()+geom_point()+theme(legend.position = "none")+scale_color_manual(values=colorsss)
volcano

namessSignificant <- results %>% filter(fill=="significant") %>% rownames()


significant_samples <- X20200810_Vincent_cultures_profinder_untargeted_new %>%  dplyr::select(namessSignificant)
significant_samples$groups <- substr(X20200810_Vincent_cultures_profinder_untargeted_new$sample, 1, 1)
significant_samples$sample <- X20200810_Vincent_cultures_profinder_untargeted_new$sample
# gather(significant_samples,)

significant_samples_long <- gather(significant_samples, feature, intensity, colnames(significant_samples)[1:10], factor_key=TRUE,na.rm = TRUE) 
ggplot(significant_samples_long,aes(x=groups,intensity,fill=groups))+theme_classic()+geom_boxplot()+facet_wrap(~feature,scales="free")

# str_split_fixed(X20200810_Vincent_cultures_profinder_untargeted_new$sample, fixed("*"), 3)[,1]

png("~/Desktop/Projects/2019_RMK202_analysis/plot/GC_MS_ttest_coculture-vs-isolates.png", width = 2500, height = 2500,res=300)
 # svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
volcano
dev.off()
```



#Cellcounts again
This is from the experiment on the 20200901 were I grew all dominant ldel and Sterm in isoaltion and in co-culture aswell as all complex RMK mixes. 
Here I analyse the coressponding CFU counts

```{r}
library(readr)
library(ggsignif)
library(Rmisc)
dominantStrains_RMK_CFU <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/13_CFU/20200904_dominantStrains_RMK_CFU .csv", skip = 1) %>% dplyr::select(c(-Comments,-dilution)) %>% filter(experiment=="normalGrowth")
# dominantStrains_RMK_CFU <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/13_CFU/dominantStrains_RMK_CFU_new.csv", skip = 1) %>% dplyr::select(c(-Comments,-dilution)) %>% filter(experiment=="normalGrowth")


dominantStrains_RMK_CFU_long <- gather(dominantStrains_RMK_CFU, plate, CFU, "SR9.3":"BM_1", factor_key=TRUE,na.rm = TRUE) %>% filter(!plate %in% c("BM","BM_1"))

# dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("SR9.3_1"="SR9.3","MR_1"="MR","BM_1"="BM"))
dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("SR9.3_1"="Sterm","MR_1"="Ldel","SR9.3"="Sterm","MR"="Ldel"))
# dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("SR9.3_1"="SR9.3","MR_1"="MR","BM_1"="BM"))

dominantStrains_RMK_CFU_long$CFU <- abs(dominantStrains_RMK_CFU_long$CFU)


dominantStrains_RMK_CFU_long$category <- revalue(dominantStrains_RMK_CFU_long$category,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus","pairwise"="pairwise\nco-culture","RMK"="complex\nco-culture"))
dominantStrains_RMK_CFU_long$category = factor(dominantStrains_RMK_CFU_long$category, levels=c("S.thermophilus",  "L.delbrueckii",  "pairwise\nco-culture" , "complex\nco-culture"))
dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus"))


tgc <- summarySE(dominantStrains_RMK_CFU_long, measurevar="CFU", groupvars=c("category","plate"))
tgc

# tgc$category

# tgc$category <- revalue(tgc$category,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus","pairwise"="pairwise\nco-culture","RMK"="complex\nco-culture"))
tgc$category = factor(tgc$category, levels=c("S.thermophilus",  "L.delbrueckii",  "pairwise\nco-culture" , "complex\nco-culture"))
# tgc$plate <- revalue(tgc$plate,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus"))

annot_1 <- t.test(dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "S.thermophilus" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"], dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "pairwise\nco-culture" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"])$p.value

annot_2 <- t.test(dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "S.thermophilus" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"], dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "complex\nco-culture" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"])$p.value

# all_colours <- (c("#EB4D4D","#10B552") )

all_colours <- (c("#ff6f69","#88d8b0") )


#
require(scales)

CFU_plot <- ggplot(tgc, aes(x=category, y=CFU, fill=plate,group=plate)) + theme_classic()+
    # lims(y=c(0,10^10))+
   scale_y_continuous(trans="log10",breaks = trans_breaks("log10", function(x) 10^x),
    labels = trans_format("log10", math_format(10^.x)))+
theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 10, hjust = 1),legend.title = element_blank())+  labs(x="",y="CFU/ml")+
    geom_bar(position=position_dodge(), stat="identity") +
  scale_fill_manual(values=all_colours)+
    # geom_signif(annotation=formatC(annot_1, digits=1),)+
    geom_errorbar(aes(ymin=CFU-ci, ymax=CFU+ci),
                  width=.2,                    # Width of the error bars
                  position=position_dodge(.9))


CFU_plot

svg("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/CFU_pairwise_rmkstocks.svg",width=4.5,height=3.5)
# png("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/HPLC_pairwise_rmkstocks.png", width = 1800, height = 1600,res=300)
CFU_plot
dev.off()

####----------------------------------------------------
#again
####----------------------------------------------------

library(readr)
library(ggsignif)
library(Rmisc)
library(tidyverse)
dominantStrains_RMK_CFU <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/13_CFU/dominantStrains_RMK_CFU_new.csv", skip = 1) %>% dplyr::select(c(-Comments,-dilution)) 


dominantStrains_RMK_CFU_long <- gather(dominantStrains_RMK_CFU, plate, CFU, "SR9.3...5":"BM...10", factor_key=TRUE,na.rm = TRUE) %>% mutate(plate=gsub("\\.\\..*","",plate)) %>% filter(!plate %in% c("BM","BM_1"))

# dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("SR9.3_1"="SR9.3","MR_1"="MR","BM_1"="BM"))
dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("SR9.3"="Sterm","MR"="Ldel","SR9.3"="Sterm","MR"="Ldel"))
# dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("SR9.3_1"="SR9.3","MR_1"="MR","BM_1"="BM"))

dominantStrains_RMK_CFU_long$CFU <- abs(dominantStrains_RMK_CFU_long$CFU)

unique(dominantStrains_RMK_CFU_long$category )

dominantStrains_RMK_CFU_long$category <- revalue(dominantStrains_RMK_CFU_long$category,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus","pairwise"="pairwise\nco-culture","RMK"="complex\nco-culture","Strepto_IN_Lacto"="S.thermophilus in\n L.delbrueckii supernatant","Strepto_IN_Strepto"="S.thermophilus in\n S.thermophilus supernatant","Lacto_IN_Lacto"="L.delbrueckii in\n L.delbrueckii supernatant","Lacto_IN_Strepto"="L.delbrueckii in\n S.thermophilus supernatant","Ldel_phControlled"="L.delbrueckii \n pH-controlled","Sterm_phControlled"="S.thermophilus \n pH-controlled"))


dominantStrains_RMK_CFU_long$category = factor(dominantStrains_RMK_CFU_long$category, levels=c( "L.delbrueckii",  "S.thermophilus","L.delbrueckii in\n L.delbrueckii supernatant", "L.delbrueckii in\n S.thermophilus supernatant","L.delbrueckii \n pH-controlled","S.thermophilus in\n L.delbrueckii supernatant","S.thermophilus in\n S.thermophilus supernatant","S.thermophilus \n pH-controlled","pairwise\nco-culture" , "complex\nco-culture"))
dominantStrains_RMK_CFU_long$plate <- revalue(dominantStrains_RMK_CFU_long$plate,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus"))


tgc <- summarySE(dominantStrains_RMK_CFU_long, measurevar="CFU", groupvars=c("category","plate"))
tgc

# tgc$category

# tgc$category <- revalue(tgc$category,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus","pairwise"="pairwise\nco-culture","RMK"="complex\nco-culture"))
tgc$category = factor(tgc$category, levels=c( "L.delbrueckii",  "S.thermophilus","L.delbrueckii in\n L.delbrueckii supernatant", "L.delbrueckii in\n S.thermophilus supernatant","L.delbrueckii \n pH-controlled","S.thermophilus in\n L.delbrueckii supernatant","S.thermophilus in\n S.thermophilus supernatant","S.thermophilus \n pH-controlled","pairwise\nco-culture" , "complex\nco-culture"))
# tgc$plate <- revalue(tgc$plate,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus"))

# annot_1 <- t.test(dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "S.thermophilus" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"], dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "pairwise\nco-culture" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"])$p.value

# annot_2 <- t.test(dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "S.thermophilus" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"], dominantStrains_RMK_CFU_long[dominantStrains_RMK_CFU_long$category == "complex\nco-culture" & dominantStrains_RMK_CFU_long$plate == "S.thermophilus", "CFU"])$p.value

# all_colours <- (c("#EB4D4D","#10B552") )
# colorsss <- c("#99CCFF","#00d5d5ff","#0000FF","#6699FF")
# colorsss <- rev(c("#007fa6ff","#fdc2b5ff"))
# 
# all_colours <- (c("#ff6f69","#88d8b0") )

all_colours <- rev(c("#007fa6ff","#fdc2b5ff") )

#
require(scales)

tgc$ymin <- tgc$CFU-tgc$ci
tgc$ymax <- tgc$CFU+tgc$ci


CFU_plot <- ggplot(tgc, aes(x=category, y=CFU, fill=plate,group=plate)) + theme_classic()+
    # lims(y=c(0,10^10))+
   scale_y_continuous(trans="log10",breaks = trans_breaks("log10", function(x) 10^x),
    labels = trans_format("log10", math_format(10^.x)))+
theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 7, hjust = 1),legend.title = element_blank(),legend.position = "none")+  labs(x="",y="CFU/ml")+
    geom_bar(position=position_dodge(), stat="identity") +
  scale_fill_manual(values=all_colours)+
    # geom_signif(annotation=formatC(annot_1, digits=1),)+
    geom_errorbar(aes(ymin=CFU-se, ymax=CFU+se),
                  width=.2,                    # Width of the error bars
                  position=position_dodge(.9))




CFU_plot

svg("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/CFU_pairwise_rmkstocks_new.svg",width=4.5,height=3.5)
# png("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/HPLC_pairwise_rmkstocks.png", width = 1800, height = 1600,res=300)
CFU_plot
dev.off()


svg("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/CFU_and_Aciditiy_pairwise_rmkstocks_new.svg",width=5.5,height=7)
# png("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/HPLC_pairwise_rmkstocks.png", width = 1800, height = 1600,res=300)
k0plot+theme(axis.text.x=element_blank())+CFU_plot+ plot_layout(nrow=2)
dev.off()

##--------------------------make boxplots

ggplot(tgc, aes(x=category, y=CFU, fill=plate)) + theme_classic()+geom_boxplot()


CFU_plot <- ggplot(tgc, aes(x=category, y=CFU, fill=plate,group=category)) + theme_classic()+
    # lims(y=c(0,10^10))+
   scale_y_continuous(trans="log10",breaks = trans_breaks("log10", function(x) 10^x),
    labels = trans_format("log10", math_format(10^.x)))+
theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 7, hjust = 1),legend.title = element_blank(),legend.position = "none")+  labs(x="",y="CFU/ml")+
    geom_boxplot() +
  geom_jitter()+
      # geom_bar(position=position_dodge(), stat="identity") +
  scale_fill_manual(values=all_colours)+
    # geom_signif(annotation=formatC(annot_1, digits=1),)+
    geom_errorbar(aes(ymin=CFU-se, ymax=CFU+se),
                  width=.2,                    # Width of the error bars
                  position=position_dodge(.9))




CFU_plot

##--------------------------make boxplots ALL DATA

# ggplot(tgc, aes(x=category, y=CFU, fill=plate)) + theme_classic()+geom_boxplot()

dominantStrains_RMK_CFU_long_sub <- dominantStrains_RMK_CFU_long %>% filter(category %in% c( "L.delbrueckii",  "S.thermophilus","L.delbrueckii \n pH-controlled","S.thermophilus \n pH-controlled","pairwise\nco-culture" , "complex\nco-culture"))


# tgc$category = factor(tgc$category, levels=c( "L.delbrueckii",  "S.thermophilus","L.delbrueckii in\n L.delbrueckii supernatant", "L.delbrueckii in\n S.thermophilus supernatant","L.delbrueckii \n pH-controlled","S.thermophilus in\n L.delbrueckii supernatant","S.thermophilus in\n S.thermophilus supernatant","S.thermophilus \n pH-controlled","pairwise\nco-culture" , "complex\nco-culture"))


 # color=plate,

CFU_plot_02 <- ggplot(dominantStrains_RMK_CFU_long_sub, aes(x=category, y=CFU, fill=plate,group=interaction(category,plate))) + theme_classic()+
    # lims(y=c(0,10^10))+
   scale_y_continuous(trans="log10",breaks = trans_breaks("log10", function(x) 10^x),
    labels = trans_format("log10", math_format(10^.x)))+
theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 7, hjust = 1),legend.title = element_blank(),legend.position = "none")+  labs(x="",y="CFU/ml")+
    geom_boxplot() +
  geom_jitter(alpha=0.5)+
      # geom_bar(position=position_dodge(), stat="identity") +
  scale_fill_manual(values=all_colours)
    # geom_signif(annotation=formatC(annot_1, digits=1),)+
    # geom_errorbar(aes(ymin=CFU-se, ymax=CFU+se),
    #               width=.2,                    # Width of the error bars
    #               position=position_dodge(.9))
    # 

CFU_plot_02

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/CFU_species.pdf",width=4.6,height=4)

# svg("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/CFU_and_Aciditiy_pairwise_rmkstocks_new.svg",width=5.5,height=7)
# png("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/HPLC_pairwise_rmkstocks.png", width = 1800, height = 1600,res=300)
CFU_plot_02

dev.off()


##---------------------------------
#make subset plot
##---------------------------------

dominantStrains_RMK_CFU_long_sub_sub <- dominantStrains_RMK_CFU_long_sub %>% filter(category %in% c("S.thermophilus","L.delbrueckii","complex\nco-culture","S.thermophilus \n pH-controlled","L.delbrueckii \n pH-controlled"))
table(dominantStrains_RMK_CFU_long_sub_sub$category)


dominantStrains_RMK_CFU_long_sub_sub$category <- plyr::revalue(dominantStrains_RMK_CFU_long_sub_sub$category , c("complex\nco-culture"="Starter culture","L.delbrueckii \n pH-controlled"="Control","S.thermophilus \n pH-controlled"="Control"))
table(dominantStrains_RMK_CFU_long_sub_sub$category)

dominantStrains_RMK_CFU_long_sub_sub$category = factor(dominantStrains_RMK_CFU_long_sub_sub$category, levels=c(   "S.thermophilus","L.delbrueckii","Starter culture","Control"))


all_colours <- (c("#007fa6ff",
"#fdc2b5ff",
"#404E5C",
"#d9d9d9",
"#bdbdbd",
"#969696") )



CFU_plot_03 <- ggplot() + theme_classic()+
    # lims(y=c(0,10^10))+
   scale_y_continuous(trans="log10",breaks = trans_breaks("log10", function(x) 10^x),
    labels = trans_format("log10", math_format(10^.x)))+
theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 7, hjust = 1),legend.title = element_blank(),legend.position = "none")+  labs(x="",y="CFU/ml")+
    geom_boxplot(data = dominantStrains_RMK_CFU_long_sub_sub, aes(x=category, y=CFU, fill=plate,group=interaction(category,plate)),outlier.alpha = 0) +
  geom_jitter(data=dominantStrains_RMK_CFU_long_sub_sub, aes(x=category, y=CFU, fill=plate,color=plate,group=interaction(category,plate)),alpha=0.7,width = 0.1)+
      # geom_bar(position=position_dodge(), stat="identity") +
  scale_fill_manual(values=all_colours)+
  scale_color_manual(values=all_colours)+
  theme(axis.text.x = element_blank())
    # geom_signif(annotation=formatC(annot_1, digits=1),)+
    # geom_errorbar(aes(ymin=CFU-se, ymax=CFU+se),
    #               width=.2,                    # Width of the error bars
    #               position=position_dodge(.9))
    # 

CFU_plot_03


```

##Biolog
prep biolog plate names
```{bash}
cd Desktop/Projects/2020_StarterCultureDiversity/Biolog/

log_pm1.csv

rm log_pm1_final.csv
for locationsss in $(seq 1 2 191)
do
namessss=$((locationsss+1))
   echo "location is at $locationsss and names at $namessss"
   
final1=$(sed -n "${locationsss}p" log_pm1.csv)
final2=$(sed -n "${namessss}p" log_pm1.csv)

echo -e ${final1}"\t"${final2} >> log_pm1_final.csv

done

```

PREP file because the labels where filled into the wrong column

```{bash}
sed 's/Sample Number/tmp/g' ~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/biolog_all_strains/20210511__allHours_kinetic.csv |sed 's/Strain Name  /Sample Number/g' |sed 's/tmp/Strain Name  /g' > ~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/biolog_all_strains/20210511__allHours_kinetic_cleaned.csv

sed 's/Sample Number/tmp/g' ~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/biolog_all_strains/20210511__allHours_kinetic.csv |sed 's/Strain Name  /Sample Number/g' |sed 's/tmp/Strain Name  /g' | grep -v "Sample Number" |sed 's/Strain Name/Sample Number,\nStrain Name/g' > ~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/biolog_all_strains/20210511__allHours_kinetic_cleaned.csv


sed 's/-1/NA NA/g'

Sample Number
```



with opm package


```{r}
library(opm)
library(tidyverse)
# library(plyr)


  # files <- opm_files("~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/biolog_all_strains/20210511__allHours_kinetic.csv")
  
example.opm <- read_opm(names = "~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/biolog_all_strains/20210511__allHours_kinetic_cleaned.csv", convert = "try")

replicatess <- "1"

flattened_opm <- opm::flatten(example.opm)

meta_data <- csv_data(example.opm) %>% as.data.frame()

meta_data$platename <- paste0("Plate ",1:nrow(meta_data))
  

meta_data_sub <- meta_data %>% select("platename","Strain Name") %>% rename(., "Strain_Name"="Strain Name" )
cleaned_biolog_file <- merge(flattened_opm,meta_data_sub,by.x="Plate",by.y="platename") %>% add_column(replicate=replicatess)
unique(cleaned_biolog_file$Well)
cleaned_biolog_file$Well_cleaned <-  gsub(".*\\(","",cleaned_biolog_file$Well) %>% gsub("\\)","",.)
cleaned_biolog_file <- as_tibble(cleaned_biolog_file)
###--------------------kinetics name


p1 <- ggplot(cleaned_biolog_file,aes(x=Time,y=Value,group=Well_cleaned,color=Well_cleaned))+geom_line()+facet_wrap(~Strain_Name)+theme_classic()
p1


unique(cleaned_biolog_file$Well_cleaned)

cleaned_biolog_file_test <- cleaned_biolog_file %>% filter(Well_cleaned=="a-D-Lactose")

p1 <- ggplot(cleaned_biolog_file_test,aes(x=Time,y=Value,group=Strain_Name,color=Strain_Name))+geom_line()+theme_classic()
p1


cleaned_biolog_file_test <- cleaned_biolog_file %>% filter(Well_cleaned=="D-Xylose")

p1 <- ggplot(cleaned_biolog_file_test,aes(x=Time,y=Value,group=Strain_Name,color=Strain_Name))+geom_line()+theme_classic()
p1



unique(cleaned_biolog_file$Well_cleaned)

cleaned_biolog_file_test <- cleaned_biolog_file %>% filter(Well_cleaned== "Negative Control" )

p1 <- ggplot(cleaned_biolog_file_test,aes(x=Time,y=Value,group=Strain_Name,color=Strain_Name))+geom_line()+theme_classic()
p1



  library(plotly)
ggp <- ggplotly(p1)
 ggp

 
 


 
 
###--------------------make binary


Biolog_diff_new <- aggregate(. ~`Strain_Name`+Well_cleaned, data=cleaned_biolog_file[,c("Strain_Name","Well_cleaned","Value")], FUN = function(i)max(i) - min(i))
hist(Biolog_diff_new$Value,breaks = 20)
pDensity <- ggplot(Biolog_diff_new,aes(x=Value,group=Strain_Name,color=`Strain_Name`,fill=Strain_Name))+geom_density()+theme_classic()+facet_wrap(Strain_Name~.)+geom_vline(xintercept =90)+labs(x="Intensity difference between maximal and minimal value")+theme(legend.position = "none")
pDensity


###--------------------check nagative control

cleaned_biolog_file_Neg_con <- cleaned_biolog_file %>% filter(Well_cleaned== "Negative Control" ) %>% group_by(Plate,Strain_Name,replicate) %>% summarise(maxs=max(Value),medianss=median(Value),sds=sd(Value)) %>% select(Strain_Name,medianss)

Biolog_diff_new_norm <- Biolog_diff_new %>% left_join(.,cleaned_biolog_file_Neg_con) %>% as_tibble() %>% mutate(Value=Value-medianss) %>% select(-medianss) %>% mutate(Value=ifelse(Value<0,0,Value))

Biolog_diff_new_norm %>% filter(Well_cleaned %in% "a-D-Lactose")

# Biolog_diff_new_norm <- Biolog_diff_new_norm %>% mutate(Value=ifelse(Value>20,1,0))

# Biolog_diff_new %>% as_tibble()
 

ggheatmap <- ggplot(Biolog_diff_new_norm, aes(y=Well_cleaned, x=Strain_Name, fill = Value))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "white", high = "red", mid = "white",
   midpoint = 100, limit = c(0,259), space = "Lab",
   name="max growth") +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = "top") +
  # labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)


#----discrete
Biolog_diff_new_norm_discrete <- Biolog_diff_new_norm %>% mutate(Value=ifelse(Value>50,1,0))



ggheatmap <- ggplot(Biolog_diff_new_norm_discrete, aes(y=Well_cleaned, x=Strain_Name, fill = Value))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "white", high = "red", mid = "white",
   midpoint = 0.5, limit = c(0,1), space = "Lab",
   name="max growth") +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = "top") +
  # labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)
###--------------------change educt order


Biolog_diff_new_high <- Biolog_diff_new
Biolog_diff_new_high$Well_cleaned <- droplevels(as.factor(Biolog_diff_new_high$Well_cleaned))
Biolog_diff_new_high$Strain_Name <- droplevels(as.factor(Biolog_diff_new_high$Strain_Name))
table(Biolog_diff_new_high$Well_cleaned,Biolog_diff_new_high$Strain_Name)

# Biolog_diff %>%  filter(well=="C09")

# Biolog_diff_high_ext <- Biolog_diff %>% filter(well %in% c(as.character(unique(Biolog_diff_high$well)),"A06")) #subset dataset
# Biolog_diff_high_ext <- Biolog_diff  #%>% filter(well %in% c(as.character(unique(Biolog_diff_high$well)),"A06"))
# Biolog_diff_high_ext <- merge(Biolog_diff,log_pm1_final,by="well") %>% mutate(well=metabolite) %>% select(-metabolite)  #%>% filter(well %in% c(as.character(unique(Biolog_diff_high$well)),"A06"))
table(Biolog_diff_new_high$Well_cleaned)
table(Biolog_diff_new_high$well)
# unique(Biolog_diff_high_ext$`Field 2`)
# Biolog_diff_high_ext$`Field 2` <- revalue(Biolog_diff_high_ext$`Field 2`, c( "Stamm 37"="S24737", "Stamm 38"="S24738", "Stamm 39"="S24739", "Stamm 40"="S24740" ,"Stamm new 1"="24854", "Stamm new 2"="24855"))#subset dataset





# Biolog_diff_high_widess <- spread(Biolog_diff_new, Strain_Name, Value) %>% column_to_rownames(var="Well_cleaned")



sorting_vector <- Biolog_diff_new %>% group_by(Well_cleaned) %>% summarize(final=sum(Value)) %>% arrange(final) %>% dplyr::pull(., Well_cleaned)

Biolog_diff_new$Well_cleaned = factor(Biolog_diff_new$Well_cleaned, levels=sorting_vector)


library(gplots)
library(RColorBrewer)


    dev.off()
range(Biolog_diff_new$Value)
ggheatmap <- ggplot(Biolog_diff_new, aes(y=Well_cleaned, x=Strain_Name, fill = Value))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "white", high = "red", mid = "white",
   midpoint = 100, limit = c(0,259), space = "Lab",
   name="max growth") +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = "top") +
  # labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)
  
  


#    pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf",width=6,height=8)
# 
#   print(ggheatmap)
# 
# dev.off()


##-------------------------------
#only Streptococcus
##-------------------------------


Biolog_diff_new_trepto <- Biolog_diff_new %>% filter(Strain_Name %in% c("24736","24735","24734","24750","24739","24745","24728","24742","24746","24749","24730","24743","S_sal_48"))


Biolog_diff_new_trepto$Strain_Name <- plyr::revalue(Biolog_diff_new_trepto$Strain_Name, c("24736"="lineage 1","24735"="lineage 2","24734"="lineage 3","24750"="lineage 4","24739"="lineage 5","24745"="lineage 6","24728"="lineage 7","24742"="lineage 8","24746"="lineage 9","24749"="lineage 10","24730"="lineage 11","24743"="lineage 12","S_sal_48"="S. salivarius"))


Biolog_diff_new_trepto$Strain_Name  = factor(Biolog_diff_new_trepto$Strain_Name , levels=c("lineage 1","lineage 2","lineage 3","lineage 4","lineage 5","lineage 6","lineage 7","lineage 8","lineage 9","lineage 10","lineage 11","lineage 12","S. salivarius"))



    dev.off()
range(Biolog_diff_new_trepto$Value)
ggheatmap <- ggplot(Biolog_diff_new_trepto, aes(y=Well_cleaned, x=Strain_Name, fill = Value))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "white", high = "red", mid = "white",
   midpoint = 100, limit = c(0,224), space = "Lab",
   name="max growth") +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = "top") +
  # labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)


png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms_new.png", width = 2500, height = 3000,res=300)

   # pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf",width=6,height=8)

  print(ggheatmap)

dev.off()

##-------------------------------------------
#growthcurver
##-------------------------------------------
library(growthcurver)

cleaned_biolog_file_prep <- cleaned_biolog_file %>% select(Time,Well_cleaned,Strain_Name,Value)

cleaned_biolog_file_prep$Well_cleaned <- plyr::revalue(cleaned_biolog_file_prep$Well_cleaned, c("Negative Control"="blank"))


growthcurves_strains <- data.frame()
unique(cleaned_biolog_file_prep$Strain_Name)
strainss <- "24742"
for (strainss in unique(cleaned_biolog_file_prep$Strain_Name)) {
  
  tmps <- cleaned_biolog_file_prep %>% filter(Strain_Name==strainss)
  
  tmps_wide <- spread(tmps, Well_cleaned, Value) %>% select(-Strain_Name)

  # gc_out <- SummarizeGrowthByPlate(tmps_wide)

  gc_out <- SummarizeGrowthByPlate(tmps_wide)
  # gc_out <- SummarizeGrowthByPlate(tmps_wide, bg_correct = "blank")

  
  gc_out_filtered <- gc_out %>% filter(note == "")  %>% filter(sample!= "") %>% add_column(Strain_Name=strainss) %>% add_column(replicate=replicatess)
  # hist(gc_out_filtered$sigma, main = "Histogram of sigma values", xlab = "sigma")
growthcurves_strains <- rbind(growthcurves_strains,gc_out_filtered)
  
}



table(growthcurves_strains$Strain_Name)

# growthcurves_strains_sub <- growthcurves_strains 

growthcurves_strains_sub <- growthcurves_strains %>% filter(!Strain_Name %in% c("24746","24750","24743","24749"))

    dev.off()
range(growthcurves_strains$r)
ggheatmap <- ggplot(growthcurves_strains_sub, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "white", high = "red", mid = "white", space = "Lab",
   name="max growth") +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = "top") +
  # labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)
##-------------------------------
#only Streptococcus
##-------------------------------


growthcurves_strains_sub_strepto <- growthcurves_strains_sub %>% filter(Strain_Name %in% c("24736","24735","24734","24750","24739","24745","24728","24742","24746","24749","24730","24743","S_sal_48"))


growthcurves_strains_sub_strepto$Strain_Name <- plyr::revalue(growthcurves_strains_sub_strepto$Strain_Name, c("24736"="lineage 1","24735"="lineage 2","24734"="lineage 3","24750"="lineage 4","24739"="lineage 5","24745"="lineage 6","24728"="lineage 7","24742"="lineage 8","24746"="lineage 9","24749"="lineage 10","24730"="lineage 11","24743"="lineage 12","S_sal_48"="S. salivarius"))


growthcurves_strains_sub_strepto$Strain_Name  = factor(growthcurves_strains_sub_strepto$Strain_Name , levels=c("lineage 1","lineage 2","lineage 3","lineage 4","lineage 5","lineage 6","lineage 7","lineage 8","lineage 9","lineage 10","lineage 11","lineage 12","S. salivarius"))


    dev.off()
range(growthcurves_strains$r)
ggheatmap <- ggplot(growthcurves_strains_sub_strepto, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "white", high = "red", mid = "white", space = "Lab",
   name="max growth") +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = "top") +
  # labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)

##------------------------------order correctly


growthcurves_strains_sub_strepto


sorting_vector <- growthcurves_strains_sub_strepto %>% group_by(sample) %>% summarize(final=sum(r)) %>% arrange(final) %>% dplyr::pull(., sample)

growthcurves_strains_sub_strepto$sample = factor(growthcurves_strains_sub_strepto$sample, levels=sorting_vector)




    dev.off()
range(growthcurves_strains$r)
ggheatmap <- ggplot(growthcurves_strains_sub_strepto, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "white", high = "red", mid = "white", 
                     # midpoint = 0.0, limit = c(0,4),
                     space = "Lab",

   name="max growth") +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = "top") +
  # labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)


png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms_new_01.png", width = 1800, height = 3000,res=300)

   # pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf",width=6,height=8)

  print(ggheatmap)

dev.off()


###--------------sub

ggplot(growthcurves_strains_sub_strepto,aes(x=r))+geom_density()+geom_vline(xintercept = 0.1)

growthcurves_strains_sub_strepto_sub <- growthcurves_strains_sub_strepto %>% filter(r>0.1)

  dev.off()

ggheatmap_strepto <- ggplot(growthcurves_strains_sub_strepto_sub, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "white", high = "red", mid = "white", 
                     limit = c(0,1.75),
                     space = "Lab",

   name="max growth") +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = "top") +
  # labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap_strepto)


png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms_new_01.png", width = 1800, height = 3000,res=300)

   # pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf",width=6,height=8)

  print(ggheatmap_strepto)

dev.off()


growthcurves_strains_sub_strepto_sub %>% group_by(Strain_Name) %>% summarise(counts=n()) %>% filter(Strain_Name !="S. salivarius") %>% summarise(mean=mean(counts))
# growthcurves_strains_sub_strepto

growthcurves_strains_sub_strepto_sub %>% group_by(Strain_Name) %>% summarise(counts=n())

##-------------------------------
#only lactobacillus
##-------------------------------


growthcurves_strains_sub_lacto <- growthcurves_strains_sub %>% filter(!Strain_Name %in% c("24736","24735","24734","24750","24739","24745","24728","24742","24746","24749","24730","24743","S_sal_48"))


growthcurves_strains_sub_lacto$Strain_Name <- plyr::revalue(growthcurves_strains_sub_lacto$Strain_Name, c("24790"="lineage 13","24793"="lineage 14"))


growthcurves_strains_sub_lacto$Strain_Name  = factor(growthcurves_strains_sub_lacto$Strain_Name , levels=c("lineage 13","lineage 14","L_bul_50","l_sunkii_45","L_indus_23","L_equi_23349"))


    dev.off()
range(growthcurves_strains$r)
ggheatmap <- ggplot(growthcurves_strains_sub_lacto, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "white", high = "red", mid = "white", space = "Lab",
   name="max growth") +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = "top") +
  # labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)

##------------------------------order correctly


growthcurves_strains_sub_lacto


sorting_vector <- growthcurves_strains_sub_lacto %>% group_by(sample) %>% summarize(final=sum(r)) %>% arrange(final) %>% dplyr::pull(., sample)

growthcurves_strains_sub_lacto$sample = factor(growthcurves_strains_sub_lacto$sample, levels=sorting_vector)




    dev.off()
range(growthcurves_strains$r)
ggheatmap <- ggplot(growthcurves_strains_sub_lacto, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "white", high = "red", mid = "white", 
                     # midpoint = 0.0, limit = c(0,4),
                     space = "Lab",

   name="max growth") +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = "top") +
  # labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)


png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_lacto_new_01.png", width = 1800, height = 3000,res=300)

   # pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf",width=6,height=8)

  print(ggheatmap)

dev.off()


###--------------sub

ggplot(growthcurves_strains_sub_lacto,aes(x=r))+geom_density()+geom_vline(xintercept = 0.1)

growthcurves_strains_sub_lacto_sub <- growthcurves_strains_sub_lacto %>% filter(r>0.1) %>% filter(Strain_Name %in% c("lineage 13","L_equi_23349"))


# growthcurves_strains_sub_lacto <- growthcurves_strains_sub %>% filter(!Strain_Name %in% c("24736","24735","24734","24750","24739","24745","24728","24742","24746","24749","24730","24743","S_sal_48"))



  dev.off()

ggheatmap_lacto <- ggplot(growthcurves_strains_sub_lacto_sub, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "white", high = "red", mid = "white", 
                     limit = c(0,1.75),
                     space = "Lab",

   name="max growth") +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = "top") +
  # labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap_lacto)


png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_lacto_new_01.png", width = 1800, height = 3000,res=300)

   # pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf",width=6,height=8)

  print(ggheatmap_lacto)

dev.off()

growthcurves_strains_sub_lacto_sub %>% group_by(Strain_Name) %>% summarise(counts=n())

61-9

```


old way


Here, I analyse the Biolog data of the different strains.

```{r}
# library(readr)
# library(ggplot2)
# library(plyr)
library(tidyverse)
library(opm)
library(patchwork)
Biolog_all_plates <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/Biolog/strains/Export_All_Plates_Single_File.csv", ";", escape_double = FALSE, trim_ws = TRUE)
Biolog_all_plates$`Field 2`


log_pm1_final <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/log_pm1_final.csv", "\t", escape_double = FALSE, col_names = c("well","metabolite"),   trim_ws = TRUE)

ggplot(Biolog_all_plates,aes(x=Hour,y=A06,group=`Field 2`,color=`Field 2`))+geom_line()

       
Biolog_long <- gather(Biolog_all_plates, well, intensity, A01:H12, factor_key=TRUE,na.rm = TRUE) 
Biolog_long$intensity <- as.double(Biolog_long$intensity)
Biolog_long$Hour <- as.double(Biolog_long$Hour)

str(Biolog_long)
p1 <- ggplot(Biolog_long,aes(x=Hour,y=intensity,group=well,color=well))+geom_line()+facet_wrap(`Field 2`~.)+theme_classic()
p1
png("~/Desktop/Projects/2019_RMK202_analysis/biolog/plots/Biolog_sterm_rmk202_strains_01.png", width = 4000, height = 4000,res=300)
 # svg("~/Desktop/Projects/2019_RMK202_analysis/10_assembly/heatmap_all_Sterm_strains.svg",width=14,height=14)
p1
dev.off()

##---------------------------------------------
##growth (binary)
##calculate different of highest and lowest intensity value
##---------------------------------------------
Biolog_diff <- aggregate(. ~`Field 2`+well, data=Biolog_long[,c("Field 2","well","intensity")], FUN = function(i)max(i) - min(i))
hist(Biolog_diff$intensity,breaks = 20)
pDensity <- ggplot(Biolog_diff,aes(x=intensity,group=`Field 2`,color=`Field 2`,fill=`Field 2`))+geom_density()+theme_classic()+facet_wrap(`Field 2`~.)+geom_vline(xintercept =90)+labs(x="Intensity difference between maximal and minimal value")+theme(legend.position = "none")

# Biolog_diff_high <- Biolog_diff %>% filter(intensity>90)  #subset dataset
# table(Biolog_diff$well)
Biolog_diff_high <- Biolog_diff
Biolog_diff_high$well <- droplevels(Biolog_diff_high$well)
Biolog_diff_high$`Field 2` <- droplevels(Biolog_diff_high$`Field 2`)
table(Biolog_diff_high$well,Biolog_diff_high$`Field 2`)

Biolog_diff %>%  filter(well=="C09")

# Biolog_diff_high_ext <- Biolog_diff %>% filter(well %in% c(as.character(unique(Biolog_diff_high$well)),"A06")) #subset dataset
# Biolog_diff_high_ext <- Biolog_diff  #%>% filter(well %in% c(as.character(unique(Biolog_diff_high$well)),"A06"))
Biolog_diff_high_ext <- merge(Biolog_diff,log_pm1_final,by="well") %>% mutate(well=metabolite) %>% select(-metabolite)  #%>% filter(well %in% c(as.character(unique(Biolog_diff_high$well)),"A06"))
table(Biolog_diff$well)
table(Biolog_diff_high_ext$well)
unique(Biolog_diff_high_ext$`Field 2`)
Biolog_diff_high_ext$`Field 2` <- revalue(Biolog_diff_high_ext$`Field 2`, c( "Stamm 37"="S24737", "Stamm 38"="S24738", "Stamm 39"="S24739", "Stamm 40"="S24740" ,"Stamm new 1"="24854", "Stamm new 2"="24855"))#subset dataset




sorting_vector <- Biolog_diff_high_ext %>% group_by(well) %>% summarize(final=sum(intensity)) %>% arrange(final) %>% dplyr::pull(., well)

Biolog_diff_high_ext$well = factor(Biolog_diff_high_ext$well, levels=sorting_vector)



##---------------------------------------------
##make heatmap of strain growth 
##only use the carbohydrates that have at least one strain that grows
##---------------------------------------------


# Biolog_diff_wide <- spread(Biolog_diff, `Field 2`, intensity) %>% column_to_rownames(var="well")
Biolog_diff_high_wide <- spread(Biolog_diff_high, `Field 2`, intensity) %>% column_to_rownames(var="well")
Biolog_diff_high_wide_ext <- spread(Biolog_diff_high_ext, `Field 2`, intensity) %>% column_to_rownames(var="well")

# rownames(Biolog_diff_high_wide_ext) <- revalue(rownames(Biolog_diff_high_wide_ext), c( "A02"="L-Arabinose", "B08"="D-Xylose", "B11"="D-Mannitol", "C04"="D-Ribose" ,"C07"="D-Fructose", "C09"="alpha-D-Glucose", "D09"="alpha-D-Lactose" ,"D10"="Lactulose", "D11"="Sucrose", "G08"="N-Acetyl-Beta-D-Mannosamine" ,"H06"="L-Lyxose","A06"="D-Galactose"))#subset dataset




# rownames(Biolog_diff_high_wide_ext) <- revalue(rownames(Biolog_diff_high_wide_ext), c( "A02"="L-Arabinose", "B08"="D-Xylose", "B11"="D-Mannitol", "C04"="D-Ribose" ,"C07"="D-Fructose", "C09"="alpha-D-Glucose", "D09"="alpha-D-Lactose" ,"D10"="Lactulose", "D11"="Sucrose", "G08"="N-Acetyl-Beta-D-Mannosamine" ,"H06"="L-Lyxose","A06"="D-Galactose","A01"="Negative control","A03"="N-acetyl-d-glucosamine","A04"="D-saccharic acid","A05"="Succinic acid","A07"="L-aspartic acid","A08"="L-proline","A09"="D-alanine","A10"="D-trehalose","A11"="D-mannose","A12"="Dulcitol","B01"="D-serine","B02"="D-sorbitol","B03"="Glycerol","B04"="L-fucose","B05"="D-Glucuronic acid","B06"="D-gluconic acid","B07"="D,L-alpha-glycerol-phosphate","B08"="D-xylose","B09"="D-xylose","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY","XX"="YYYYYYY"))


# rownames(Biolog_diff_high_wide_ext)

if (!require("gplots")) {
   install.packages("gplots", dependencies = TRUE)
   library(gplots)
   }
if (!require("RColorBrewer")) {
   install.packages("RColorBrewer", dependencies = TRUE)
   library(RColorBrewer)
   }


# mat_data <- as.matrix(Biolog_diff_wide) # transform column 2-5 into a matrix
# mat_data <- as.matrix(Biolog_diff_high_wide) # transform column 2-5 into a matrix
# mat_data <- as.matrix(Biolog_diff_high_wide_ext) # transform column 2-5 into a matrix
# 
# my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 119)
# # col_breaks = c(seq(0,max(mat_data),length=300))
# # length(col_breaks)
# col_breaks = c(seq(min(mat_data),60,length=40),  # for red
#   seq(61,90,length=40),           # for yellow
#   seq(91,max(mat_data),length=40))
# 
# str(mat_data)
# dev.off()
#     # png("~/Desktop/Projects/2019_RMK202<_analysis/10_assembly/heatmap_all_Sterm_strains.png", width = 4000, height = 4000,res=300)
#  # svg("~/Desktop/Projects/2019_RMK202_analysis/10_assembly/heatmap_all_Sterm_strains.svg",width=14,height=14)
# pHeatmap <- heatmap.2(mat_data,
#   # cellnote = mat_data,  # same data set for cell labels
#   main = "", # heat map title
#   notecol="black",      # change font color of cell labels to black
#   density.info="none",
#   trace="none",         # turns off trace lines inside the heat map
#  margins = c(16, 16),     # widens margins around plot
#   col=my_palette,   
#   breaks=col_breaks,    # enable color transition at specified limits
#   sepcolor="black",
#   sepwidth=c(0.5,0.5),
#   dendrogram="none",
#  Colv=FALSE)
# 
# svg("~/Desktop/Projects/2019_RMK202_analysis/biolog/plots/DenstiyPlot_biolog_02.svg",width=8,height=8)
# heatmap.2(mat_data,
#   # cellnote = mat_data,  # same data set for cell labels
#   main = "", # heat map title
#   notecol="black",      # change font color of cell labels to black
#   density.info="none",
#   trace="none",         # turns off trace lines inside the heat map
#  margins = c(16, 16),     # widens margins around plot
#   col=my_palette,   
#   breaks=col_breaks,    # enable color transition at specified limits
#   sepcolor="black",
#   sepwidth=c(0.5,0.5),
#   dendrogram="none",
#  Colv=FALSE)
# dev.off()
# # dev.off()
# 
#   
#   svg("~/Desktop/Projects/2019_RMK202_analysis/biolog/plots/DenstiyPlot_biolog_01.svg",width=5,height=5)
#    pDensity
#   dev.off()

  
  ###----------------------------------
  ##all data
  ###----------------------------------
  

    dev.off()
range(Biolog_diff_high_ext$intensity)
ggheatmap <- ggplot(Biolog_diff_high_ext, aes(y=well, x=`Field 2`, fill = intensity))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "white", high = "red", mid = "white",
   midpoint = 100, limit = c(0,232), space = "Lab",
   name="max growth") +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = "top") +
  # labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)
  
  


   pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf",width=6,height=8)

  print(ggheatmap)

dev.off()
```




##HPLC

```{r}
library(readr)
data_vincent_hplc <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/13_HPLC/data_vincent_hplc.csv","\t", escape_double = FALSE, trim_ws = TRUE)%>% filter(`Lactose [g/L]`>1)%>% filter(!SAMPLE_NAME %in% c("r+0","0+c","r+c"))

# data_vincent_hplc$category

data_vincent_hplc$category <- revalue(data_vincent_hplc$category,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus","pairwise"="pairwise\nco-culture","RMK"="complex\nco-culture","prts_pos"="prts +\nS.thermophilus"))
data_vincent_hplc$category = factor(data_vincent_hplc$category, levels=c("S.thermophilus",  "L.delbrueckii",  "pairwise\nco-culture" , "complex\nco-culture","prts +\nS.thermophilus"))

##--------------------------
#lactose
##--------------------------
data_vincent_hplc$`Lactose [g/L]` <- data_vincent_hplc$`Lactose [g/L]`*3
lactosePlot <- ggplot(data_vincent_hplc,aes(x=category,y=`Lactose [g/L]`,fill=category))+geom_boxplot()+theme_classic()+theme(axis.text.x = element_text(angle = 45, vjust = 1, size = 10, hjust = 1),legend.title = element_blank(),legend.position = "none")+labs(x="")


##--------------------------
#lactate
##--------------------------
data_vincent_hplc$`Lactate [g/L]` <- data_vincent_hplc$`Lactate [g/L]`*3
lacPlot <- ggplot(data_vincent_hplc,aes(x=category,y=`Lactate [g/L]`,fill=category))+geom_boxplot()+theme_classic()+theme(axis.text.x = element_text(angle = 45, vjust = 1, size = 10, hjust = 1),legend.title = element_blank(),legend.position = "none")+labs(x="")



##--------------------------
#glucose
##--------------------------
data_vincent_hplc$`Glucose [g/L]` <- data_vincent_hplc$`Glucose [g/L]`*3
gluPlot <- ggplot(data_vincent_hplc,aes(x=category,y=`Glucose [g/L]`,fill=category))+geom_boxplot()+theme_classic()+theme(axis.text.x = element_text(angle = 45, vjust = 1, size = 10, hjust = 1),legend.title = element_blank(),legend.position = "none")+labs(x="")



##--------------------------
#galactose
##--------------------------
data_vincent_hplc$`Galactose [g/L]` <- data_vincent_hplc$`Galactose [g/L]`*3
galPlot <- ggplot(data_vincent_hplc,aes(x=category,y=`Galactose [g/L]`,fill=category))+geom_boxplot()+theme_classic()+theme(axis.text.x = element_text(angle = 45, vjust = 1, size = 10, hjust = 1),legend.title = element_blank(),legend.position = "none")+labs(x="")

(lactosePlot+lacPlot)/
(galPlot+gluPlot)

svg("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/HPLC_pairwise_rmkstocks.svg",width=6,height=7)
# png("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/HPLC_pairwise_rmkstocks.png", width = 1800, height = 1600,res=300)
(lactosePlot+lacPlot)/
(galPlot+gluPlot)
dev.off()


##--------------------------
#overall sugar in system
##--------------------------

data_vincent_hplc$totalSugar <- (data_vincent_hplc$`Lactose [g/L]`+data_vincent_hplc$`Glucose [g/L]`+data_vincent_hplc$`Galactose [g/L]`+data_vincent_hplc$`Lactate [g/L]`)

totalPlot <- ggplot(data_vincent_hplc,aes(x=category,y=totalSugar,fill=category))+geom_boxplot()+theme_classic()+theme(axis.text.x = element_text(angle = 45, vjust = 1, size = 12, hjust = 1),legend.title = element_blank(),legend.position = "none")+labs(x="")

totalPlot

```

##Proteolytic activity

```{r}
library(readr)
library(Rmisc)
Vincent_Prot_Akt_again <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/12_proteas_activity/Vincent_Prot_Akt.csv","\t", escape_double = FALSE, trim_ws = TRUE)

Vincent_Prot_Akt_again_long <- gather(Vincent_Prot_Akt_again, replicate, gylcerine,c("replicate1","replicate2"), factor_key=TRUE,na.rm = TRUE) 

ggplot(Vincent_Prot_Akt_again_long,aes(x=group,y=gylcerine,fill=group))+geom_boxplot()+theme_classic()
Vincent_Prot_Akt_again_long$group <- plyr::revalue(Vincent_Prot_Akt_again_long$group,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus","pairwise"="pairwise\nco-culture","RMK"="complex\nco-culture"))



tgc <- summarySE(Vincent_Prot_Akt_again_long, measurevar="gylcerine", groupvars="group")
# tgc


tgc$group = factor(tgc$group, levels=c("S.thermophilus",  "L.delbrueckii",  "pairwise\nco-culture" , "complex\nco-culture","Milk","Prts+ S. thermophilus","L.helveticus"))


all_colours <- (c("#d7301f",
"#238b45",
"#4eb3d3",
"#0868ac",
"#d9d9d9",
"#bdbdbd",
"#969696") )


#
require(scales)

proteolytic_plot <- ggplot(tgc, aes(x=group, y=gylcerine, fill=group,group=group)) + theme_classic()+
    # lims(y=c(0,10^10))+
   # scale_y_continuous(trans="log10",breaks = trans_breaks("log10", function(x) 10^x),
    # labels = trans_format("log10", math_format(10^.x)))+
theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 10, hjust = 1),legend.title = element_blank(),legend.position = "none")+  labs(x="",y="Glycine-concentration\n[mmol/l]")+
    geom_bar(position=position_dodge(), stat="identity") +
  scale_fill_manual(values=all_colours)+
    # geom_signif(annotation=formatC(annot_1, digits=1),)+
    geom_errorbar(aes(ymin=gylcerine-ci, ymax=gylcerine+ci),
                  width=.2)

proteolytic_plot


# svg("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/proteolyticAcivity_pairwise_rmkstocks.svg",width=4,height=5.5)
# png("/home/vincent/Desktop/presenations/my_presentations/20200909_committee_agroscope/HPLC_pairwise_rmkstocks.png", width = 1800, height = 1600,res=300)
proteolytic_plot
dev.off()

##--------------------------------------
#final figure
##--------------------------------------

tgc_sub <- tgc %>% filter(!group %in% c("Prts+ S. thermophilus","L.helveticus"))
all_colours <- (c("#007fa6ff",
"#fdc2b5ff",
"#4F6272",
"#404E5C",
"#d9d9d9",
"#bdbdbd",
"#969696") )

# all_colours <- rev(c("#EB4D4D","#10B552") )

proteolytic_plot_final <- ggplot(tgc_sub, aes(x=group, y=gylcerine, fill=group,group=group)) + theme_classic()+
    # lims(y=c(0,10^10))+
   # scale_y_continuous(trans="log10",breaks = trans_breaks("log10", function(x) 10^x),
    # labels = trans_format("log10", math_format(10^.x)))+
theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 10, hjust = 1),legend.title = element_blank(),legend.position = "none")+  labs(x="",y="Glycine-concentration\n[mmol/l]")+
    geom_bar(position=position_dodge(), stat="identity") +
  scale_fill_manual(values=all_colours)+
    # geom_signif(annotation=formatC(annot_1, digits=1),)+
    geom_errorbar(aes(ymin=gylcerine-ci, ymax=gylcerine+ci),
                  width=.2)

proteolytic_plot_final


  pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/proteolyticActivity.pdf",width=5,height=6)
 proteolytic_plot_final
  dev.off()



##--------------------------------------
#reduced final figure
##--------------------------------------
  
 Vincent_Prot_Akt_again_long_sub <-  Vincent_Prot_Akt_again_long %>% filter(group %in% c("S.thermophilus","L.delbrueckii","complex\nco-culture","Milk"))
  
Vincent_Prot_Akt_again_long_sub$group <- plyr::revalue(Vincent_Prot_Akt_again_long_sub$group , c("complex\nco-culture"="Starter culture","Milk"="Control"))


  
tgc_sub_sub <- tgc_sub %>% filter(group!="pairwise\nco-culture")  
  
  tgc_sub_sub$group %>% unique()
tgc_sub_sub$group <- plyr::revalue(tgc_sub_sub$group , c("complex\nco-culture"="Starter culture","Milk"="Control"))

all_colours <- (c("#007fa6ff",
"#fdc2b5ff",
"#404E5C",
"#d9d9d9",
"#bdbdbd",
"#969696") )

proteolytic_plot_final_sub <- ggplot() + theme_classic()+
    # lims(y=c(0,10^10))+
   # scale_y_continuous(trans="log10",breaks = trans_breaks("log10", function(x) 10^x),
    # labels = trans_format("log10", math_format(10^.x)))+
theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 10, hjust = 1),legend.title = element_blank(),legend.position = "none")+  labs(x="",y="Glycine-concentration\n[mmol/l]")+
    geom_bar(data=tgc_sub_sub, aes(x=group, y=gylcerine, fill=group,group=group),position=position_dodge(), stat="identity") +
  scale_fill_manual(values=all_colours)+
    # geom_signif(annotation=formatC(annot_1, digits=1),)+
    geom_errorbar(data=tgc_sub_sub,aes(x=group,ymin=gylcerine-ci, ymax=gylcerine+ci),
                  width=.2)+
  geom_jitter(data=Vincent_Prot_Akt_again_long_sub, aes(x=group, y=gylcerine, fill=group,group=group),alpha=0.4,width = 0.2)

proteolytic_plot_final_sub
  

##--------------------------------------
#put figure together
##--------------------------------------

CFU_plot_03+proteolytic_plot_final_sub+plot_layout(ncol=1)


  pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/proteolyticActivity_CFU.pdf",width=4,height=5)
 CFU_plot_03+proteolytic_plot_final_sub+plot_layout(ncol=1)

  dev.off()
```


##interaction prediction
what does interaction look like

```{r}

library(readr)
library(tidyverse)
library(reshape2)
starter_culture_interaction_matrix <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/starter_culture_interaction_matrix.csv", "\t", escape_double = FALSE, trim_ws = TRUE) 

starter_culture_interaction_matrix_new <- starter_culture_interaction_matrix %>% column_to_rownames(var="X1") %>% as.matrix %>% t()

meltR = melt(starter_culture_interaction_matrix_new, id = "Portion")
meltR$species1 <-  str_split_fixed(meltR$Var1, fixed(":"), 2)[,1]
meltR$starterculture1 <-  str_split_fixed(meltR$Var1, fixed(":"), 2)[,2]
meltR$species2 <-  str_split_fixed(meltR$Var2, fixed(":"), 2)[,1]
meltR$starterculture2 <-  str_split_fixed(meltR$Var2, fixed(":"), 2)[,2]

# creat Heatmap
ggheatmap <- ggplot(meltR, aes(starterculture1, starterculture2, fill = value))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "blue", high = "red", mid = "white",
   midpoint = 0.75, limit = c(0,1), space = "Lab",
   name="fitness") +
  theme_minimal()+ # minimal theme
  labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)

##save heatmap
# svg("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.svg",width=2.5,height=2.5)
png("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.png", width = 1800, height = 1600,res=300)
print(ggheatmap)
dev.off()

```



##Data preperation
many of these steps have beeen removed into the 2020_scrip! look there

###mapping to RMK202

```{bash}

scp /home/vincent/Downloads/sequence.fasta vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/Ldelv_mag.fasta

```


```{bash}
#add L.helveticus

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta \
  /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/Ldelv_mag.fasta > \
  /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv.fasta
  
###================
##description file
###================
#cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202.txt > /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames.txt

#for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt)


threads=37
samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt ##the file with all sample names
logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref
Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv.fasta


names=Ws_150_19
#/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 index ${Assembly}
#/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 mem -t 16 ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz/${namezzz}-R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz/${namezzz}-R2_val_2.fq.gz |head
##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}
rm ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
num=1

for names in $(cut -f 1 ${samplesss})
  do
  echo ${num}"/95  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 #/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 index ${Assembly} 
  
#  /home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 mem -t ${threads} ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
bwa mem -t ${threads} ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${ynames}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   

tail -95 ${BaseLocation}/log/multiqc_logfile_finalGenome.txt > ${BaseLocation}/log/multiqc_logfile_finalGenome_onlyMETA.txt
   
    rm -r ${BaseLocation}/MultiQC_onlyMETA
 
  mkdir -p ${BaseLocation}/MultiQC_onlyMETA
  
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome_onlyMETA.txt \
    -o ${BaseLocation}/MultiQC_onlyMETA

#/home/vincent/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/MultiQC/multiqc_report.html

scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/MultiQC/multiqc_report.html /Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/

##-------------------------------------------------  
##-----check read coverage
##-------------------------------------------------  
name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
threads=37
names=Konserve_202


num=1
rm -r ${BaseLocation}/log/mappings.txt
rm -r ${BaseLocation}/log/mappings_species.txt


for names in $(cut -f 1 ${samplesss})
  do
mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')

mapped_sterm=$(grep "CP046134" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)
mapped_ldel=$(grep "CP046131" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)

mkdir -p ${BaseLocation}/log
echo -e ${names}"\tonlyMeta\t"${mapped} >> ${BaseLocation}/log/mappings.txt

echo -e ${names}"\tLactobacillus delbrueckii\t"$mapped_ldel >> ${BaseLocation}/log/mappings_species.txt
echo -e ${names}"\tStreptococcus thermophilusi\t"$mapped_sterm >> ${BaseLocation}/log/mappings_species.txt

done

##-------------------------------------------------  
##move local
##-------------------------------------------------  

##bring together 

cat /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings.txt \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202/log/mappings.txt > \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings_final.txt


scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/log/mappings_species.txt Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/
```

####mapping strains to genome

```{bash}

threads=37
samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames.txt ##the file with all sample names
logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref
Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv.fasta

##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1

for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/)
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)
names=$(echo ${samplezzKurz})
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 # gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq
  
  
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} ${Assembly} \
  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   
   
    rm -r ${BaseLocation}/MultiQC
 
  mkdir -p ${BaseLocation}/MultiQC
  
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome.txt \
    -o ${BaseLocation}/MultiQC






##-------------------------------------------------  
##-----check read coverage
##-------------------------------------------------  
name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
threads=37
names=Konserve_202


num=1
rm -r ${BaseLocation}/log/mappings.txt
rm -r ${BaseLocation}/log/mappings_species.txt


for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/)
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)
names=$(echo ${samplezzKurz})
mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')

#mapped_sterm=$(grep "CP046134" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)
#mapped_ldel=$(grep "CP046131" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)

mkdir -p ${BaseLocation}/log
echo -e ${names}"\tonlyMeta\t"${mapped} >> ${BaseLocation}/log/mappings.txt

echo -e ${names}"\tLactobacillus delbrueckii\t"$mapped_ldel >> ${BaseLocation}/log/mappings_species.txt
echo -e ${names}"\tStreptococcus thermophilusi\t"$mapped_sterm >> ${BaseLocation}/log/mappings_species.txt

done

grep "0." ${BaseLocation}/log/mappings_species.txt | sort -k 4

grep -v "0.0" ${BaseLocation}/log/mappings_species.txt|awk '{ sum += $4; n++ } END { if (n > 0) print sum / n; }'

```



```{r}
naming_DF <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/98_script/naming_file.csv")
###----------------------------
##script
###----------------------------
library(readr)
library(ggplot2)
library(plyr)
library(tidyverse)
#library(plyr)

species_coverage <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/mappings_species.txt", "\t", escape_double = FALSE, col_names = c("sample","species","coverage"),trim_ws = TRUE)

species_coverage$sample

newNames <- setNames(naming_DF$namesNew,naming_DF$namesOLD)
species_coverage$sample <- revalue(species_coverage$sample, newNames)
species_coverage$sample = factor(species_coverage$sample, levels=c(as.character(naming_DF$namesNew)))


newNames <- setNames(as.character(naming_DF$CultureName),naming_DF$namesNew)
species_coverage$culture <- revalue(as.character(species_coverage$sample),newNames)
levels(species_coverage$culture)

species_coverage$coverageLog <- log2(species_coverage$coverage)

all_colours <- (c("#EB4D4D","#10B552") ) 



##--------------------
##ploting


pMOTUs <-  ggplot( data = species_coverage,aes(y = coverage, x = sample, group=species,fill = culture,color=culture))+geom_point(size=2)+geom_hline(yintercept=250)+ # geom_bar( stat="identity")+
  facet_wrap(~species,scales = "free_x")+
   coord_trans(y="log10")+
    labs("",
         x="",
         y="Coverage")+
    theme_classic()+
   scale_color_viridis(discrete=TRUE)+
      scale_y_continuous(breaks =c(0,50,100,250,500,1000),labels=c(0,50,100,250,500,1000))+
  scale_fill_manual(values=all_colours)+
  scale_color_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=4),
      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          # legend.title = element_blank()
          )

  pMOTUs

  
  
  svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/CoverageSpecies.svg",width=5,height=5)
   # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)

pMOTUs

dev.off()

```

check which raw read files are missing

```{bash}
mkdir -p /home/vincent/Desktop/Projects/2020_strainDelineation/05_SNP_called/raw_snps
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/log_genomes.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/


#ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/ > /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/log_genomes.txt

cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Poppunk_out.all |grep "S50" -v|grep "S72" -v|grep "SMAG" -v|sed 's/L//g' |sed 's/S//g' >  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/log_genomes.txt


for genomess in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/ |sed 's/FAM//g')
do

sed -i "s/${genomess}//g" /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/log_genomes.txt
#sed 's/${genomess}//g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/.tmp > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/.tmp

done
sort /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/log_genomes.txt |uniq 

```


####mapping to unknown REF

INclude the following strains:

- S-PNGR-Z-K.fna (Streptococcus thermophilus)
- L-KCTC-303.fna (Lactobacillus delbrueckii)
- NZ_LR698986.1 (Lactobacillus helveticus)


```{bash}

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/FNA/S-PNGR-Z-K.fna \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/FNA/L-KCTC-3034.fna \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/FNA/Out-helv.fna > \
  /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv_unknownREF.fasta

seq_length.py /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv_unknownREF.fasta
###----------------------------------
##mapping
###----------------------------------

threads=37
samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_withStrains.txt ##the file with all sample names
#samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames.txt ##the file with all sample names
logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref_final
Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv_unknownREF.fasta

##!!!!!!!!!!!!!change names of Forward and Reverse reads


#for samplezzKurz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/ |grep "fastqc" -i -v )


##============
##mapping REFERENCES
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1
#for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/ |grep "fastqc" -i -v )
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/)
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)
names=$(echo ${samplezzKurz})
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 # gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq
  
  
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} ${Assembly} \
  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   
   

##============
##mapping METAGENOMES
##============


threads=37
samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt ##the file with all sample names
#BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref


names=Ws_150_19

#bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}
rm ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
num=1

for names in $(cut -f 1 ${samplesss} |grep -v "FAM")
  do
  echo ${num}"/95  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 #/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 index ${Assembly} 
  
#  /home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 mem -t ${threads} ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
bwa mem -t ${threads} ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
#samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

#bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

#rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${ynames}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   




##-------------------------------------------------  
##-----check read coverage
##-------------------------------------------------  
name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
threads=37
names=Konserve_202


num=1
rm -r ${BaseLocation}/log/mappings.txt
rm -r ${BaseLocation}/log/mappings_species.txt


for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/)
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)
names=$(echo ${samplezzKurz})
mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')

#mapped_sterm=$(grep "CP046134" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)
#mapped_ldel=$(grep "CP046131" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)

mkdir -p ${BaseLocation}/log
echo -e ${names}"\tonlyMeta\t"${mapped} >> ${BaseLocation}/log/mappings.txt

echo -e ${names}"\tLactobacillus delbrueckii\t"$mapped_ldel >> ${BaseLocation}/log/mappings_species.txt
echo -e ${names}"\tStreptococcus thermophilusi\t"$mapped_sterm >> ${BaseLocation}/log/mappings_species.txt

done

grep "0." ${BaseLocation}/log/mappings_species.txt | sort -k 4

grep -v "0.0" ${BaseLocation}/log/mappings_species.txt|awk '{ sum += $4; n++ } END { if (n > 0) print sum / n; }'



```


#### SNP calling freebayes (K1)

This is a good (blog)[https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2017/Day2/vcf-intro.nb.html] on how to use freebayes and how to parse the output. 
Finally I will do freebayes on all samples simultaniously. 

Interesting output from column 11 on:

GT	Genotype v :1
DP	Read Depth  :1040
DPR	Number of observation for each allele  : 1,1037
RO	Reference allele observation count :1
QR	Sum of quality of the reference observations :10
AO	Alternate allele observation count :1037
QA	Sum of quality of the alternate observations :36917
GL	Genotype Likelihood, log10-scaled likelihoods of the data given the called genotype for each possible genotype generated from the reference and alternate alleles given the sample ploidy

Very nice [blog](http://www.ddocent.com/filtering/) illustrating good filtering practices.

filtering of samples:
1. take away all Lhelv SNVs. 


```{bash,eval=FALSE,echo=FALSE}

###================
##merge sample names (REF+meta)
###================

ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/ |cut -d '_' -f 1 > /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_withStrains.txt

cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt >> /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_withStrains.txt

#ls /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/ |grep "fastqc" -i -v  >> /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_withStrains.txt

#sort /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_withStrains.txt|uniq > /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_withStrains_uniques.txt

###================
##description file
###================
  
  #cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202.txt > /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames.txt
  threads=30
  samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_withStrains.txt ##the file with all sample names
  logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
  BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref_final
#  BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref

#Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv.fasta
  Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv_unknownREF.fasta



###--------special
#FREEBAYES_out=freebayesOuput #this is the old one
FREEBAYES_out=freebayesOuput_all
###================
##script
###================

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------change read group
##--------------------------------------------------------------------------------------------------------------------------------------

#name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
names=Lyo_202_2014
###------------------adding readgroup to bam
#!!!!!!!!!!!!!!!!!!!!!!!!!rm ${BaseLocation}/log/multiqc_logfile_bams.txt
mkdir -p  ${BaseLocation}/log/
  #for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
  #for names in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/ |grep "fastqc" -i -v )
  for names in $(cat ${samplesss})
do
echo ${names}

java -jar /home/vincent/apps/picard/picard.jar AddOrReplaceReadGroups \
       I=${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam \
       O=${BaseLocation}/${names}/bwaMapping2DB/${names}_mapping2ref.bam \
       RGID=${names} \
       RGLB=lib1 \
       RGPL=illumina \
       RGPU=unit1 \
       RGSM=${names}

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam

echo ${BaseLocation}"/"${names}"/bwaMapping2DB/"${names}"_mapping2ref.bam " >> ${BaseLocation}/log/multiqc_logfile_bams.txt

done

rm ${BaseLocation}/log/multiqc_logfile_bams_new.txt
  for names in $(cut -f 1 ${samplesss})
do
echo "============================================="
echo $names

samtools view -h -b -q 30 ${BaseLocation}/${names}/bwaMapping2DB/${names}_mapping2ref.bam -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_mapping2ref_mapq30.bam

echo ${BaseLocation}"/"${names}"/bwaMapping2DB/"${names}"_mapping2ref_mapq30.bam" >> ${BaseLocation}/log/multiqc_logfile_bams_new.txt
#samtools view ${BaseLocation}/${names}/bwaMapping2DB/${names}_mapping2ref.bam |head -1
done


##remove trailing whitespaces

 sed -i 's/[ \t]*$//' ${BaseLocation}/log/multiqc_logfile_bams_new.txt

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------check bam mapping
##--------------------------------------------------------------------------------------------------------------------------------------

 for names in $(cut -f 1 ${samplesss})
    do
    echo "----------------------------------------------------"
    echo $names
    samtools flagstat ${BaseLocation}/${names}/bwaMapping2DB/${names}_mapping2ref.bam |grep "mapped ("
    
    done

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------indexing file
##--------------------------------------------------------------------------------------------------------------------------------------
##need before freebayes parallel

  for names in $(cat ${BaseLocation}/log/multiqc_logfile_bams_new.txt)
do

samtools index ${names}

done

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------SNP calling
##--------------------------------------------------------------------------------------------------------------------------------------

#rm -r ${BaseLocation}/freebayesOuput_WithONT_notParallel_default
#mkdir -p ${BaseLocation}/freebayesOuput_WithONT_notParallel_default

  #  freebayes -F 0.05 -C 3 --min-coverage 30 --pooled-discrete \
  #    --bam-list ${BaseLocation}/log/multiqc_logfile_bams.txt -f \
  #    ${Assembly} > \
  #   ${BaseLocation}/freebayesOuput_WithONT_notParallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf
  
##--------------------------------------------------------------------------------------------------------------------------------------
##-----------SNP calling parallel
##--------------------------------------------------------------------------------------------------------------------------------------
samtools faidx ${Assembly}
bwa index ${Assembly}
FREEBAYES_out=freebayesOuput_all
  rm -r ${BaseLocation}/${FREEBAYES_out}
  mkdir -p ${BaseLocation}/${FREEBAYES_out}
 
 
 ##!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 ##it is crucial to play around with the threads because the chunks eat up alot of memory if you have many samples. However changing chunk size does not have such a large effect. 
 
 
    freebayes-parallel <(/home/vincent/apps/freebayes/scripts/fasta_generate_regions.py $Assembly 100000) 8 -f ${Assembly} -C 5 --pooled-continuous --min-alternate-fraction 0.01 --min-coverage 10 --bam-list ${BaseLocation}/log/multiqc_logfile_bams_new.txt > ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf
  
#ll /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all
  #  freebayes-parallel <(/home/vincent/apps/freebayes/scripts/fasta_generate_regions.py $Assembly 100000) 30 -f ${Assembly} -C 5 --pooled-continuous --min-alternate-fraction 0.01 --min-coverage 10 --bam-list ${BaseLocation}/log/multiqc_logfile_bams.txt > ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf
    
  
  grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf |cut -f 1 |sort|uniq -c
  
  grep -v "^#"  ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf|cut -f 1|sort|uniq -c

  
 wc -l /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf
##-------------zipping

#bgzip -c /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains/freebayesOuput_WithONT_02/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains/freebayesOuput_WithONT_02/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf.gz
#tabix -p  vcf /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final/freebayesOuput_WithONT/variantCallsfreebayes_all.vcf.gz


##--------------------------------------------------------------------------------------------------------------------------------------
##------------vcftools filtering
##--------------------------------------------------------------------------------------------------------------------------------------
###============================== remove all fixed mutations
#--non-ref-af and --max-non-ref-af

vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf  --min-alleles 2 --max-alleles 2 --remove-indels --minQ 30  --max-non-ref-af 0.95 --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_w_o_ones

##filtering
#1. only biallelic (--min-alleles 2 --max-alleles 2)
#2. no indels (--remove-indels)
#3. only good Quality (--minQ 30)
#4. no fixed mutations=REF mutations  ( --max-non-ref-af 0.95)


#vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf --non-ref-af 0.95 --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_only_ones


###==============================1. Qualty filtering QC>30

vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf --minQ 30 --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC  

  grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC.recode.vcf   |cut -f 1 |sort|uniq -c


vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf  --counts2 --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC  

###==============================4. hist-indel-len


vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC.recode.vcf --hist-indel-len --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC

###==============================2. remove indel

vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC.recode.vcf --mac 1 --remove-indels --minQ 30 --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL

  grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf   |cut -f 1 |sort|uniq -c

###==============================3. INFO

vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --missing-indv --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC  
###==============================4. hist-indel-len


vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --hist-indel-len --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL

grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf |cut -f 1 |sort|uniq -c

###==============================5. filter samples
##1. only sites that actualy have a single count (--non-ref-ac-any 1)
##2. only sites that are not in all different from the reference (HAS TO BE CHANGED AGAIN WITH THE RIGHT REF!!!!!) ##mutations that actually occur in the metagenome
##3. only biallelic sites (--min-alleles 2 --max-alleles 2)

vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --indv C_202_18_02 --min-alleles 2 --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK202 

rm -r ${BaseLocation}/${FREEBAYES_out}/prepedR/
mkdir -p ${BaseLocation}/${FREEBAYES_out}/prepedR/
#for culture in $(cut -d "_" -f 2 $samplesss|sort|uniq )
for culture in $(grep "FAM" -v $samplesss | cut -d "_" -f 2 |sort|uniq )
do
grep "\_$culture\_" $samplesss >  ${BaseLocation}/${FREEBAYES_out}/tmp.samples
#total=$(wc -l ${BaseLocation}/${FREEBAYES_out}/tmp.samples|cut -d ' ' -f 1)
vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --keep ${BaseLocation}/${FREEBAYES_out}/tmp.samples --non-ref-ac-any 1 --max-non-ref-af 0.99 --min-alleles 2 --max-alleles 2 --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK${culture}

grep "^##" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK${culture}.recode.vcf |grep "CP031023.1" -v  >  ${BaseLocation}/${FREEBAYES_out}/prepedR/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK${culture}_prepR.vcf


#--non-ref-af and --max-non-ref-af
done
  
##---------------------------------
##-FAM strains
##---------------------------------
#for culture in $(grep "FAM" $samplesss |sort|uniq )
#do
grep "^FAM" $samplesss >  ${BaseLocation}/${FREEBAYES_out}/tmp.samples
#total=$(wc -l ${BaseLocation}/${FREEBAYES_out}/tmp.samples|cut -d ' ' -f 1)
vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --keep ${BaseLocation}/${FREEBAYES_out}/tmp.samples --non-ref-ac-any 1 --max-non-ref-af 0.99 --min-alleles 2 --max-alleles 2 --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMKFAMstrains

grep "^##" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMKFAMstrains.recode.vcf |grep "CP031023.1" -v >  ${BaseLocation}/${FREEBAYES_out}/prepedR/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMKFAMstrains_prepR.vcf

  #grep "^#" -v /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf |cut -f 1 |sort|uniq -c

  grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/prepedR/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMKFAMstrains_prepR.vcf |cut -f 1 |sort|uniq -c




#rm ${BaseLocation}/${FREEBAYES_out}/prepedR_merged/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK_allSamples_prepR.vcf
#mkdir -p ${BaseLocation}/${FREEBAYES_out}/prepedR_merged/
#for culture in $(cut -d "_" -f 2 $samplesss|sort|uniq )
#do

#cat ${BaseLocation}/${FREEBAYES_out}/prepedR/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK${culture}_prepR.vcf >> ${BaseLocation}/${FREEBAYES_out}/prepedR_merged/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK${culture}_prepR.vcf

#done

#bcftools query -f '%CHROM %POS [%AO/%DP\t]\n'  ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK${culture}.recode.vcf | head -3
##--------------------------------------------------------------------------------------------------------------------------------------
##------------prep for R
##--------------------------------------------------------------------------------------------------------------------------------------


#grep "^##" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf >  ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_prepR.vcf


##--------------------------------------------------------------------------------------------------------------------------------------
##------------bring local
##--------------------------------------------------------------------------------------------------------------------------------------
mkdir -p /home/vincent/Desktop/Projects/2020_strainDelineation/05_SNP_called/raw_snps
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/prepedR/ /home/vincent/Desktop/Projects/2020_strainDelineation/05_SNP_called/raw_snps/



mkdir -p /home/vincent/Desktop/Projects/2020_strainDelineation/05_SNP_called/raw_snps_onlyforSTERM
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/prepedR/ /home/vincent/Desktop/Projects/2020_strainDelineation/05_SNP_called/raw_snps_onlyforSTERM/


```


##Add Information

#####snpEff

Here we do [snpEff](http://snpeff.sourceforge.net/SnpEff_manual.html#buildAddConfig).


```{bash}


###================
##description file
###================
threads=37
FREEBAYES_out=freebayesOuput_all
sampleShort=PROKKA_out_both ##For st_thermophilus

 threads=37
  samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_withStrains.txt ##the file with all sample names
  logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
  BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref
  Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv_unknownREF.fasta
  

#${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf

###The two genomes are from

#!!!Sterm
#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/FNA/S-PNGR-Z-K.fna \
#!!!Ldel
#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/FNA/L-KCTC-3034.fna \


##=======================================================================================================================
##-----------------------------------------------all together----------------------------------------------
##=======================================================================================================================
##==================
##Building Database
##==================
#/archiv/Projects/2018_Culturomics/genomes/Lactobacillus_delbrueckii_RMK202.fna
#/archiv/Projects/2018_Culturomics/genomes/Streptococcus_thermophilus_RMK202.fna

##-------------
##01_add genome name to
##-------------

vim /home/vincent/miniconda2/share/snpeff-4.3.1t-2/snpEff.config
#S_thermophilus_mag_rmk202.genome : S_thermophilus_mag_rmk202
#L_delbrueckii_mag_rmk202.genome : L_delbrueckii_mag_rmk202
#Streptococcus_phage_rmk202.genome : Streptococcus_phage_rmk202
#PGAP_meta_all.genome : PGAP_meta_all
PROKKA_out_both.genome : PROKKA_out_both
#PGAP_meta_all_StarterCultureDiversity.genome : PGAP_meta_all_StarterCultureDiversity
##-------------
#02_put genome fasta and gff into right directory:
##-------------

#sampleShort=S_thermophilus_mag_rmk202
#sampleShort=CP046134 ##For st_thermophilus

#/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta 

##-------------
#03_put genome fasta 
##-------------
rm -r /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}
mkdir -p /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}

cat ${Assembly} > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}/sequences.fa


##-------------
#04 genes gff 
##-------------

#grep "^#" -v /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.gff|grep -e "${sampleShort}" |cut -f 1|sort|uniq -c

#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.gff | grep -e "^${sampleShort}" |awk -F "\t" '{OFS"\t"}{if($3=="CDS") print $0}' > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff


##-------------
#04 all genes gff 
##-------------

##already preped in RAST chunk
# cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/merged/all_genes_rmk202.gff |sed 's/gene/CDS/g' > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff

grep "CDS" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/PROKKA_GFF/S-PNGR-Z-K.gff > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff
#!!!Ldel
#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/FNA/L-KCTC-3034.fna \
grep "CDS" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/PROKKA_GFF/L-KCTC-3034.gff >> /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff



##-------------
##05_build database
##-------------

cd /home/vincent/miniconda2/share/snpeff-4.3.1t-2/
java -jar snpEff.jar build -gff3 -v ${sampleShort}

##==================
##running snpEff
##==================

##-------------
##06_preperation of SNV file
##-------------

#grep -e "${sampleShort}" ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf > ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_all.vcf

##-------------
##07 SNveff calling
##-------------

cd /home/vincent/miniconda2/share/snpeff-4.3.1t-2/
#rm -r ${sampleShort} ${BaseLocation}/${FREEBAYES_out}/SnpEff/${sampleShort}/
mkdir -p ${sampleShort} ${BaseLocation}/${FREEBAYES_out}/SnpEff/${sampleShort}/


grep "NZ_CP048750.1" /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf > /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv.vcf

#grep "NZ_LR698986.1" -v /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf > /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv.vcf


grep "^#" -v /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf|cut -f 1  |sort|uniq -c
grep "^#" -v  /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv.vcf|cut -f 1  |sort|uniq -c

java -Xmx64g -jar snpEff.jar ${sampleShort} /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv.vcf > /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect.vcf

grep "^#" -v  /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect.vcf|cut -f 1  |sort|uniq -c

 # grep "^#" -v /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf |cut -f 1 |sort|uniq -c


##-------------
##08_prep for R
##-------------

grep "^#" -v /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect.vcf  | \
  awk -F "\t" 'BEGIN{OFS="\t"} {print $1,$2,$3,$4,$5,$6,$7,$8}' > /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect_prepforR.vcf


cut -f 1 /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect_prepforR.vcf|sort|uniq -c

##==================
##transfer local 
##==================

##-------------
###snpEff
##-------------
mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling
scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect_prepforR.vcf ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/

```

here we merge all annotation knowledge we have of the SNPs:

1. SNPeffect
2. eggnog annotation
3. repeat annotation
4. core genes
5. seq_length

Thereafter we have a long list of information for every SNP. 

#####repeat genes

We could potentially have a problem of alignement in genes highly similiar. Therefore I look if which genes are cluster together. 

```{bash}


###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished.fasta



BaseLocation_Ldel=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/
BaseFILNAME=L_delbrueckii_RMK202


BaseLocation_Sterm=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/
BaseFILNAME_sterm=S_thermophilus_RMK202


###================
##script
###================


##==================
##merge all fna files
##==================


rm -r ${BaseLocation}/Repeats/
mkdir -p ${BaseLocation}/Repeats/

cat ${BaseLocation_Ldel}/${BaseFILNAME}_genes.ffn \
  ${BaseLocation_Sterm}/${BaseFILNAME_sterm}_genes.ffn > \
 ${BaseLocation}/Repeats/both_genes.ffn

cat ${BaseLocation_Ldel}/${BaseFILNAME}.current.gff \
  ${BaseLocation_Sterm}/${BaseFILNAME_sterm}.current.gff > \
 ${BaseLocation}/Repeats/both_genes.ffn
 
##==================
##cd-hit
##==================

cd-hit-est -i ${BaseLocation}/Repeats/both_genes.ffn \
    -o ${BaseLocation}/Repeats/both__cdsClustering.txt -c 0.90 -M 30000 -T 37 -sc 1 -sf 1

  # cd-hit -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_cds.fna \
  #  -o /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/sterm_cdsClustering.txt -c 0.85 -M 30000 -T 37 -sc 1 -sf 1
  



  ###multifasta2fasta
  mkdir -p ${BaseLocation}/Repeats/splitClusters
  cd ${BaseLocation}/Repeats/splitClusters
  awk 'NR%1==0{ print > "Cluster_"++i".txt" }' RS='>Cl' ${BaseLocation}/Repeats/both__cdsClustering.txt.clstr

  ##prepare log file
  #cluster_2.txt till cluster_24.txt
  
  ls ${BaseLocation}/Repeats/splitClusters |sort -V |head -84 |tail -83 > ${BaseLocation}/Repeats/splitClusters/files_of_INterest.txt
     
   ##get gene names
   num=2
   rm ${BaseLocation}/Repeats/bigCluster/all_repeatingGenes.txt
   mkdir -p ${BaseLocation}/Repeats/bigCluster/
   i=Cluster_2.txt
   for i in $(cat ${BaseLocation}/Repeats/splitClusters/files_of_INterest.txt)
    do
   grep -v "^uster" ${BaseLocation}/Repeats/splitClusters/${i} |awk -F ">" '{print $2}' |awk -F "... at " 'BEGIN{OFS="\t"}{print $1,$2}' |sed 's/... \*//g' |sed '$ d' > \
    ${BaseLocation}/Repeats/bigCluster/names_${i}
    
   grep -v "^uster" ${BaseLocation}/Repeats/splitClusters/${i} |awk -F ">" '{print $2}'  |awk -F "... at " -v count="$num" 'BEGIN{OFS="\t"}{print $1,$2,count}' |sed 's/... \*//g' |sed '$ d' >> \
   ${BaseLocation}/Repeats/bigCluster/all_repeatingGenes.txt
      
    num=$((num+1))
    
    
    done
 

    ##get gff
    
  # genes=pgaptmp_001292
    mkdir -p  ${BaseLocation}/Repeats//bigCluster_genes/
    for i in $(cat ${BaseLocation}/Repeats/splitClusters/files_of_INterest.txt)
    do
    rm  ${BaseLocation}/Repeats//bigCluster_genes//genes_${i}
    for genes in $(cut -f 1 ${BaseLocation}/Repeats/bigCluster//names_${i})
      do
        
        grep "${genes}.p" ${BaseLocation}/Repeats/both_genes.ffn >> \
          ${BaseLocation}/Repeats//bigCluster_genes/genes_${i}
       
        
      done # $genes   
    done   # $i
    
    
    
  mkdir -p ~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes


scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/Repeats/bigCluster/all_repeatingGenes.txt ~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes/

```


#####Core genes prep

```{bash}
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt

##=======================================================================================================================
##-----------------------------------------------Streptococcus thermophilus----------------------------------------------
##=======================================================================================================================
species=Sterm
date=$(ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ |grep "Resul")
##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups//OG_names_${species}.txt
for OG in $(cat /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

#grep "$OG" /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt | sed 's/gnl|extdb|//g'| awk -F " " 'BEGIN{OFS="\t"}{print $1,"S_thermophilus_RMK202",$2}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01/Orthogroups/OG_names_${species}.txt
#grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups.txt | sed 's/gnl|extdb|//g'| awk -F " " 'BEGIN{OFS="\t"}{print $1,$2}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01/Orthogroups/OG_names_${species}.txt
grep "$OG" /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.tsv |awk -F "\t" '{OFS="\t"}{print $1,"S_thermophilus__RMK202_extern", $105}'  >> /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups//OG_names_${species}.txt

done #OG
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling
cat /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups//OG_names_${species}.txt >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/coreGenes_rmk_202_extern_both.txt


##=======================================================================================================================
##-----------------------------------------------Lactobacillus delbrueckii----------------------------------------------
##=======================================================================================================================
species=Ldel
date=$(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder |grep "Resul")


/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder/Results_Feb11/

##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder/${date}/Orthogroups//OG_names_${species}.txt
for OG in $(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder/${date}/Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

#grep "$OG" /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt | sed 's/gnl|extdb|//g'| awk -F " " 'BEGIN{OFS="\t"}{print $1,"S_thermophilus_RMK202",$2}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01/Orthogroups/OG_names_${species}.txt
#grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups.txt | sed 's/gnl|extdb|//g'| awk -F " " 'BEGIN{OFS="\t"}{print $1,$2}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01/Orthogroups/OG_names_${species}.txt
grep "$OG" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder/${date}/Orthogroups/Orthogroups.tsv |awk -F "\t" '{OFS="\t"}{print $1,"L_delbrueckii_extern", $14}'  >>/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder/${date}/Orthogroups//OG_names_${species}.txt

done #OG

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder/${date}/Orthogroups//OG_names_${species}.txt >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/coreGenes_rmk_202_extern_both.txt
```


```{bash}
##=======================================================================================================================
##----------------------------------------------transfer local----------------------------------------------
##=======================================================================================================================

mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling

scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/coreGenes_rmk_202_extern_both.txt  ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/coreGenes_rmk_202_extern_both.txt


```

##### for Eggnog

```{bash}


mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/EGGNOG_ref/

scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/PROKKA_FAA/S-PNGR-Z-K.faa  ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/EGGNOG_ref/

scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/PROKKA_FAA/L-KCTC-3034.faa  ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/EGGNOG_ref/

```
#####gene length

```{bash}

seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/PROKKA_FAA/S-PNGR-Z-K.faa |awk -F "\t" '{OFS="\t"}{print $1,3*$3}' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/geneLength.txt
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/PROKKA_FAA/L-KCTC-3034.faa |awk -F "\t" '{OFS="\t"}{print $1,3*$3}' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/geneLength.txt

```

```{bash}


scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/geneLength.txt  ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/


```


#####merge all infomration

```{r}
library(readr)
library(stringr)

##==================
##01-SNPeff
##==================
#RMK202_both_snpEff <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new_startercultureDiversity/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_snpeff_prepforR.vcf",  "\t", escape_double = FALSE, col_names = c("X.CHROM","POS","unknown","REF","ALT","quality","uknown2","snpeff"), trim_ws = TRUE)
RMK202_both_snpEff <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect_prepforR.vcf",  "\t", escape_double = FALSE, col_names = c("X.CHROM","POS","unknown","REF","ALT","quality","uknown2","snpeff"), trim_ws = TRUE)


table(RMK202_both_snpEff$X.CHROM)
RMK202_both_snpEff_02 <- subset(RMK202_both_snpEff, select=-c(unknown,uknown2))
RMK202_both_snpEff_02$site <- paste(RMK202_both_snpEff_02$X.CHROM,RMK202_both_snpEff_02$POS,RMK202_both_snpEff_02$REF,RMK202_both_snpEff_02$ALT,sep="_")

tmp_snpeff <-  data.frame(str_split_fixed(RMK202_both_snpEff_02$snpeff, fixed("|"), 136)[,2:8])
colnames(tmp_snpeff) <- c("effect","significance","geneName","PGAP_name","type","typeName","typeGene")

RMK202_both_snpEff_02 <- subset(RMK202_both_snpEff_02, select=-snpeff)
RMK202_both_SNPeff_final <-  cbind(RMK202_both_snpEff_02,tmp_snpeff)
# table(S_thermophilus_snpEff_03$typeGene)
# 
# S_thermophilus_snpEff_03[which(S_thermophilus_snpEff_03$typeGene=="pseudogene"),]
# table(S_thermophilus_snpEff_03[which(S_thermophilus_snpEff_03$typeGene=="protein_coding"),]$type)
# S_thermophilus_snpEff_03[which(S_thermophilus_snpEff_03$typeGene=="rRNA"),]

# RMK202_both_SNPeff_final$finalName <- RMK202_both_SNPeff_final$typeName %>% gsub("rna-","",.) %>%  gsub("TRANSCRIPT_gene-","",.) %>%  gsub("gene-","",.) 
# RMK202_both_SNPeff_final$finalName <- paste(RMK202_both_SNPeff_final$X.CHROM,RMK202_both_SNPeff_final$typeName,sep = "_")
RMK202_both_SNPeff_final$finalName <- RMK202_both_SNPeff_final$typeName

# length(grep("pgaptmp_",S_thermophilus_snpEff_03$finalName,invert = TRUE))
table(RMK202_both_SNPeff_final$X.CHROM)
#table(RMK202_both_SNPeff_final$significance)
##==================
##02 eggnog
##==================
RMK202_Ldel_eggnog <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/Ldel_eggnog.annotations",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description"),skip=4)
# RMK202_both_eggnog[RMK202_both_eggnog$query_name %in% SNPeff_final$finalName,]

RMK202_Sterm_eggnog <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/Sterm_eggnog.annotations",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description"),skip=4)
RMK202_both_eggnog <- rbind(RMK202_Sterm_eggnog,RMK202_Ldel_eggnog)

##==================
##03 repeats
##==================

# RMK202_both_repeats <- read_delim("~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes/all_repeatingGenes.txt",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("geneNAME","Repeat_identity","Repeat_cluster")) #%>% slice( 1:(n()-3))
# nrow(RMK202_both_repeats)

##==================
##04_core genes
##==================
library(tidyverse)
RMK202_both_coreGenes <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/coreGenes_rmk_202_extern_both.txt",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("orthogroup","species","geneName")) %>% add_column(.,core="core")
table(RMK202_both_coreGenes$species)
table(RMK202_both_coreGenes$core, RMK202_both_coreGenes$species)
# tmp <- RMK202_both_coreGenes %>% filter(species=="L_delbrueckii_RMK202")
# RMK202_both_coreGenes %>% filter(species=="S_thermophilus_RMK202")

##==================
##gene_length
##==================

RMK202_both_length <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/geneLength.txt",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("gene","length")) 


##==================
##05_merge
##==================
nrow(RMK202_both_SNPeff_final)

RMK202_both_SNPeff_final <- RMK202_both_SNPeff_final %>% select(X.CHROM,quality, site, effect, significance, geneName)

RMK202_snpeff_eggnog <- merge(RMK202_both_SNPeff_final, RMK202_both_eggnog, by.x = "geneName", by.y = "query_name",all.x = TRUE)
# RMK202_snpeff_eggnog_repeats <- merge(RMK202_snpeff_eggnog, RMK202_both_repeats, by.x = "geneName", by.y = "geneNAME",all.x = TRUE)
# RMK202_snpeff_eggnog_repeats$geneName <- as.character(RMK202_snpeff_eggnog_repeats$geneName)
RMK202_snpeff_eggnog_core <- merge(RMK202_snpeff_eggnog, RMK202_both_coreGenes, by.x = "geneName", by.y = "geneName",all.x = TRUE)
RMK202_snpeff_eggnog_core_length <- merge(RMK202_snpeff_eggnog_core, RMK202_both_length, by.x = "geneName", by.y = "gene",all.x = TRUE)
dim(RMK202_snpeff_eggnog_core_length)
# RMK202_both_coreGenes$geneName
# RMK202_snpeff_eggnog_repeats$geneName

nrow(RMK202_snpeff_eggnog_core_length)

table(RMK202_snpeff_eggnog_core_length$core)
table(RMK202_snpeff_eggnog_core_length$species)
table(RMK202_snpeff_eggnog_core_length$X.CHROM)

RMK202_snpeff_eggnog_core_length$species <- plyr::revalue(RMK202_snpeff_eggnog_core_length$X.CHROM, c("NZ_CP048750.12"="L. delbrueckii","NZ_CP048750.1"="S. thermophilus"))
table(RMK202_snpeff_eggnog_core_length$species)

# write.csv(RMK202_snpeff_eggnog_repeats_core,"~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new_startercultureDiversity//RMK202_snpeff_eggnog_repeats_core.txt")
write.csv(RMK202_snpeff_eggnog_core_length,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling//RMK202_snpeff_eggnog_length_core_extern.txt",row.names = FALSE)


# RMK202_snpeff_eggnog_repeats_core[grep("S_thermophilus",RMK202_snpeff_eggnog_repeats_core$finalName),]
# RMK202_snpeff_eggnog_repeats_core[grep("CP046131",RMK202_snpeff_eggnog_repeats_core$finalName),]
# 
# nrow(RMK202_snpeff_eggnog_repeats_core)
# 
# RMK202_snpeff_eggnog_repeats_core_ldel <- RMK202_snpeff_eggnog_repeats_core %>% filter(X.CHROM=="CP046131")
# table(RMK202_snpeff_eggnog_repeats_core_ldel$core)
```
### POGENOM
This is the analysis done with [POGENOM](https://pogenom.readthedocs.io/en/latest/POGENOM_Usage.html)
We do this to get population genomic insights such as dN/dS ratios.
Important to note is that the gff file should not contain any fasta entries. 

```{bash}


scp /home/vincent/Downloads/*eggnog.annotations vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/ 

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/S_I_202_SMAG/PROKKA_08282020.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/


scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/L_I_202_LMAG/*.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/


/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA_new/all_interaction_functioning_Genes_cleaned.txt

```


```{bash}
VCFFILE=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

mkdir -p  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/

#grep "^202-LMAG-1" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/202-LMAG/PROKKA_04012020.gff |sed 's/^202-LMAG-1/CP046131/g' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped.gff

#grep "^202-SMAG-1" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA_04012020.gff |sed 's/^202-SMAG-1/CP046134/g' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped.gff

grep "^#" -v /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/202-LMAG/PROKKA_04012020.gff |sed 's/^202-LMAG-1/CP046131/g' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped.gff

grep "^#" -v /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA_04012020.gff |sed 's/^202-SMAG-1/CP046134/g' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped.gff


awk -F "\t" '{OFS="\t"}{if($3=="CDS")print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped.gff > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped_ready.gff

#GFF_file=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/MAG_annotation_onlyBAC_mod.gff 


###--------------------------------
#RMK202
#Konserve_202
#Versand_202
#Lyo_202_2012
#lyo202_96
#Lyo_202_2014




#vcftools --vcf /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --minQ 30 --remove-indels --recode --recode-INFO-all \
#  --keep /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/SamplesMeta.txt --out \
#  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta

sed -e 's/\r/\n/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/MAG_annotation_onlyBAC_mod.gff  >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/MAG_annotation_onlyBAC_mod_cleaned.gff

#GFF_file=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/MAG_annotation_onlyBAC_mod_cleaned.gff
GFF_file=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped_ready.gff


Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

VCFFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta.recode.vcf

grep "^#" -v ${GFF_file} |cut -f 1|sort|uniq -c
#tail ${GFF_file}
grep ">" ${Assembly}
grep "^#" -v ${VCFFILE} |cut -f 1|sort|uniq -c
##========================================
##Run POGENOME
##========================================


###-----------------
##Sterm
###-----------------
sed 's/^202-SMAG-1/CP046134/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/S_I_202_SMAG/PROKKA_08282020.gff|sed 's/>202-SMAG-1/CP046134/g' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped_sterm_cleaned.gff
GFF_file_03=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped_sterm_cleaned.gff

#genomes=S_M_202_SMAG
#echo "##gff-version 3" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}_forPOGENOM.gff
#echo "##sequence-region 202-SMAG-1 1 1865439" >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}_forPOGENOM.gff

#grep "^CP046134" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene")print $0}' |sed 's/ID=.*locus_tag=/ID=/g' |sed 's/gene/CDS/g'|sed 's/;.*$//g'| awk '{print $0"_gene"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}_forPOGENOM.gff
#GFF_file_03=/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}_forPOGENOM.gff
#echo "##FASTA" >> ${GFF_file_03}
#cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/S_I_202_SMAG/PROKKA_08282020.fna |sed 's/>202-SMAG-1/CP046134/g' >> ${GFF_file_03}
#samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}.fasta CP046134 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}_forPOGENOM.fasta


mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/

vcftools --vcf /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --chr CP046134 --minQ 30 --remove-indels --recode --recode-INFO-all \
  --keep /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt --out \
  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta_onlySterm

VCFFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta_onlySterm.recode.vcf


rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/sterm/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/sterm/


perl /home/vincent/apps/POGENOM/POGENOM-0.8.1/pogenom.pl --vcf_file ${VCFFILE} --out /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/sterm/RunPogenome_all_rmk202_new --gff_file ${GFF_file_03}   --genetic_code_file /home/vincent/apps/POGENOM/POGENOM-0.8.1/bacterial_genetic_code_table11_ncbi.txt

#--fasta_file /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}_forPOGENOM.fasta 
#--genome_size 1865459
###-----------------
##Ldel
###-----------------
sed 's/^202-LMAG-1/CP046131/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/L_I_202_LMAG/PROKKA_10022020.gff|sed 's/>202-LMAG-1/CP046131/g' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped_ldel_cleaned.gff
GFF_file_03=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped_ldel_cleaned.gff


#genomes=L_M_202_LMAG



#grep "^CP046131" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene")print $0}' |sed 's/ID=.*locus_tag=/ID=/g' |sed 's/gene/CDS/g'|sed 's/;.*$//g'| awk '{print $0"_gene"}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}_forPOGENOM.gff
#GFF_file_03=/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}_forPOGENOM.gff
#echo "##FASTA" >> ${GFF_file_03}
#samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}.fasta CP046131 |sed 's/>//g' >> ${GFF_file_03}


mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/


vcftools --vcf /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --chr CP046131 --minQ 30 --remove-indels --recode --recode-INFO-all \
  --keep /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt --out \
  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta_onlyLdel

VCFFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta_onlyLdel.recode.vcf


rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/Ldel/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/Ldel/


perl /home/vincent/apps/POGENOM/POGENOM-0.8.1/pogenom.pl --vcf_file ${VCFFILE} --out /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/Ldel/RunPogenome_all_rmk202_new --gff_file ${GFF_file_03}   --genetic_code_file /home/vincent/apps/POGENOM/POGENOM-0.8.1/bacterial_genetic_code_table11_ncbi.txt

#--genome_size 2166765
```




move local for Anders
```{bash}

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/POGENOM_test/

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta.recode.vcf ~/Desktop/Projects/2019_RMK202_analysis/POGENOM_test/longitudinalSamples_snv.vcf

scp vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta ~/Desktop/Projects/2019_RMK202_analysis/POGENOM_test/MAG_assembly.fasta

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped.gff ~/Desktop/Projects/2019_RMK202_analysis/POGENOM_test/MAG_annotation_onlyBAC.gff

##-------------------------final

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/Ldel/RunPogenome_all_rmk202_new.pNpS-per-gene.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/RunPogenome_Ldel.pNpS-per-gene.txt

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/sterm/RunPogenome_all_rmk202_new.pNpS-per-gene.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/RunPogenome_Sterm.pNpS-per-gene.txt

sed -i 's/ID=//g' ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/RunPogenome_Ldel.pNpS-per-gene.txt
sed -i 's/ID=//g' ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/RunPogenome_Sterm.pNpS-per-gene.txt

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA_new/all_interaction_functioning_Genes_cleaned.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/




```

```{r}
library(readr)
library(ggplot2)
library(lubridate)
library(tidyverse)
Genes_of_Interest <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/all_interaction_functioning_Genes.txt",  "\t", escape_double = FALSE, col_names = c("species","gene","OG","numberGenomes"),  trim_ws = TRUE) %>% select(-numberGenomes)


OGs_Ldel <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/OGs_Ldel.txt", "\t", escape_double = FALSE, col_names = c("OG","Gene"), trim_ws = TRUE)
OGs_Sterm <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/OGs_Sterm.txt", "\t", escape_double = FALSE, col_names = c("OG","Gene"), trim_ws = TRUE)
Genes_of_Interest_Ldel <- Genes_of_Interest %>% filter(species=="Ldel")
Genes_of_Interest_Sterm <- Genes_of_Interest %>% filter(species=="Sterm")

GENES_Ldel <- merge(OGs_Ldel,Genes_of_Interest_Ldel,by="OG",all = TRUE)
# table(GENES_Ldel$gene)
GENES_Sterm <- merge(OGs_Sterm,Genes_of_Interest_Sterm,by="OG",all = TRUE)
# table(GENES_Sterm$gene)

table(GENES_Ldel_extended$gene)

RunPogenome_Ldel_pNpS_per_gene <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/RunPogenome_Ldel.pNpS-per-gene.txt",   "\t", escape_double = FALSE, trim_ws = TRUE) %>% select(-c(`RMK202 pNpS`,`Konserve_202 pNpS`,`Versand_202 pNpS`,`Lyo_202_2012 pNpS`,`lyo202_96 pNpS`,`Lyo_202_2014 pNpS`))
RunPogenome_Sterm_pNpS_per_gene <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/RunPogenome_Sterm.pNpS-per-gene.txt",   "\t", escape_double = FALSE, trim_ws = TRUE) %>% select(-c(`RMK202 pNpS`,`Konserve_202 pNpS`,`Versand_202 pNpS`,`Lyo_202_2012 pNpS`,`lyo202_96 pNpS`,`Lyo_202_2014 pNpS`))

ggplot(RunPogenome_Sterm_pNpS_per_gene,aes(x=`All_samples_combined pNpS`))+geom_density()
ggplot(RunPogenome_Ldel_pNpS_per_gene,aes(x=`All_samples_combined pNpS`))+geom_density()

all_interaction_functioning_Genes_cleaned <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/all_interaction_functioning_Genes_cleaned.txt",   "\t", escape_double = FALSE, col_names = c("geness","Name"),  trim_ws = TRUE)





all_interaction_functioning_Genes_cleaned$gene <-  str_split_fixed(all_interaction_functioning_Genes_cleaned$geness, fixed("_"), 2)[,2]
all_interaction_functioning_Genes_cleaned$species <-  str_split_fixed(all_interaction_functioning_Genes_cleaned$geness, fixed("_"), 2)[,1]

all_interaction_functioning_Genes_cleaned <- all_interaction_functioning_Genes_cleaned %>% select(-geness)


GENES_Sterm_extended <- merge(RunPogenome_Sterm_pNpS_per_gene,all_interaction_functioning_Genes_cleaned,by.x="Gene",by.y="Name",all.x = TRUE)
GENES_Ldel_extended <- merge(RunPogenome_Ldel_pNpS_per_gene,all_interaction_functioning_Genes_cleaned,by.x="Gene",by.y="Name",all.x = TRUE)


# GENES_Ldel_extended <- merge(GENES_Ldel,RunPogenome_Ldel_pNpS_per_gene,by="Gene",all = TRUE)
# table(GENES_Ldel$gene)
# GENES_Sterm_extended  <- merge(GENES_Sterm,RunPogenome_Sterm_pNpS_per_gene,by="Gene",all = TRUE)
# table(GENES_Sterm$gene)


GENES_Sterm_reduced <- GENES_Sterm_extended %>% filter(!is.na(gene)) %>% filter(!is.na(Num_loci)) 
GENES_Ldel_reduced <- GENES_Ldel_extended %>% filter(!is.na(gene)) %>% filter(!is.na(Num_loci))
GENES_Ldel_reduced$`All_samples_combined pNpS`[is.na(GENES_Ldel_reduced$`All_samples_combined pNpS`)] <- 0
GENES_Sterm_reduced$`All_samples_combined pNpS`[is.na(GENES_Sterm_reduced$`All_samples_combined pNpS`)] <- 0

GENES_Sterm_reduced_prep <- GENES_Sterm_reduced %>% select(c(gene,`All_samples_combined pNpS`,species,Num_loci))
GENES_ldel_reduced_prep <- GENES_Ldel_reduced %>% select(c(gene,`All_samples_combined pNpS`,species,Num_loci))




GENES_long <- rbind(GENES_Sterm_reduced_prep,GENES_ldel_reduced_prep) #%>% add_column(functioning="protocooperation\nrelated")

GENES_long$functioning <- ifelse(grepl("pep",GENES_long$gene),"peptidase","protocooperation\nrelated")


GENES_long <- GENES_long[
  order( GENES_long[,5] ),
]



ggheatmap_CRISPR <- ggplot(GENES_long, aes(species, gene,color = `All_samples_combined pNpS`))+
 # geom_tile(color = "white",size=1.1,shape="circle")+
 geom_point(size=8,shape="circle")+
  geom_point(shape = 1,size = 8,colour = "black")+
  # theme_classic()+
 # scale_fill_gradient2(low = "grey", high = "red",
  # midpoint = 90, limit = c(80,100), space = "Lab",
  # name="protein id",na.value = 'white',colour="black",pch=21) +
   scale_color_gradient2(low = "red", high = "grey",
  midpoint = 1, limit = c(0,2), space = "Lab",
  name="pN/pS",na.value = 'white') +
  # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_viridis(alpha=0.8)+
  labs(legend="CRISPR ID",x="",y="")+
  # scale_fill_distiller(name = "SNPs", palette = "Viridis", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap_CRISPR)


##===================================================
##plot
##=================================================== 

  svg("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder//HEATMAP_interactionGenes_pN_pS.svg",width=3,height=9)

print(ggheatmap_CRISPR)

 dev.off()
 
##===================================================
##distribution
##=================================================== 

 
GENES_Sterm_reverse <- GENES_Sterm_extended %>% filter(is.na(gene)) %>% filter(!is.na(Num_loci)) 
GENES_Ldel_reverse <- GENES_Ldel_extended %>% filter(is.na(gene)) %>% filter(!is.na(Num_loci))
GENES_Ldel_reverse$`All_samples_combined pNpS`[is.na(GENES_Ldel_reverse$`All_samples_combined pNpS`)] <- 0
GENES_Sterm_reverse$`All_samples_combined pNpS`[is.na(GENES_Sterm_reverse$`All_samples_combined pNpS`)] <- 0

GENES_Sterm_reverse_prep <- GENES_Sterm_reverse %>% select(c(gene,`All_samples_combined pNpS`,species,Num_loci))
GENES_ldel_reverse_prep <- GENES_Ldel_reverse %>% select(c(gene,`All_samples_combined pNpS`,species,Num_loci))


GENES_long_reverse <- rbind(GENES_Sterm_reverse_prep,GENES_ldel_reverse_prep) %>% add_column(functioning="other function")

GENES_complete <- rbind(GENES_long_reverse,GENES_long)

ggplot(GENES_complete,aes(x=`All_samples_combined pNpS`))+geom_density()+facet_wrap(~functioning)
 
ggplot(GENES_complete,aes(x=`All_samples_combined pNpS`,y=Num_loci))+geom_point()+facet_wrap(~functioning)


  GENES_complete$functioning <- factor(GENES_complete$functioning, levels=c("protocooperation\nrelated","peptidase","other function"))


colorssss <- c("red","orange","grey88")
dnDSplot <- ggplot(GENES_complete,aes(x=`All_samples_combined pNpS`,y=Num_loci,fill=functioning,color=functioning))+geom_point(size=2)+theme_classic()+labs(x="pN/pS",y="Number of mutations")+scale_fill_manual(values = colorssss)+scale_color_manual(values = colorssss)+theme(legend.title = element_blank())
dnDSplot


  svg("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder//dot_plot_pN_pS.svg",width=4.5,height=4)

dnDSplot

 dev.off()
 

```




##first analysis

```{r}

library(tidyverse)
library(vcfR)
library(readr)
library(plyr)
library(dplyr)
library(ggplot2)
library(xlsx)

##==================
##01_import data
##==================


###====================================================
###====================================================
##specifically for metagenomes
###====================================================
###====================================================
culturesss <- FAMstrains
culturesss <- 101
snps_long <- NA
for (culturesss in c("101","105","115","124","150","190","202","203","280","302","305")) {
  

snps_freebayes <- read_delim(paste0("/home/vincent/Desktop/Projects/2020_strainDelineation/05_SNP_called/raw_snps_onlyforSTERM/prepedR//variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK",culturesss,"_prepR.vcf"), "\t", escape_double = FALSE, trim_ws = TRUE) %>%  select(-c(QUAL,FILTER,FORMAT))


##==================
##01_filter SNVs MQM larger than 30 
##INFO=<ID=MQM,Number=A,Type=Float,Description="Mean mapping quality of observed alternate alleles">
##==================
snps_freebayes$MQM <- str_split_fixed(snps_freebayes$INFO, ";", 130)[,16] %>% gsub("MQM=","",.) 
snps_freebayes$MQM <- str_split_fixed(snps_freebayes$MQM, ",", 3)[,1] %>%  as.numeric()
snps_freebayes <- snps_freebayes[which(snps_freebayes$MQM>30),] %>% select(-MQM,-INFO)

##==================
##02_long list and frequency calc
##==================

snps_freebayes_long <- gather(snps_freebayes, sample, Snps, colnames(snps_freebayes)[6]:colnames(snps_freebayes)[ncol(snps_freebayes)], factor_key=TRUE,na.rm = TRUE) 

###------------------------calculate allel frequency   

sample_relative_temp <-snps_freebayes_long %>% filter(Snps!=".:.:.:.:.:.:.:.")%>% separate("Snps", into=c("GT","DP","AD","RO","QR","AO","QA","GL"), sep = ":") %>% transform(.,allel_frequency=as.numeric(AO)/as.numeric(DP),QualityScore=as.numeric(QA)/as.numeric(DP),Sequencing_depth=as.numeric(DP))

##==================
##03_filter QA/DP > 30
##INFO=<ID=QA,Number=A,Type=Integer,Description="Alternate allele quality sum in phred">
##INFO=<ID=DP,Number=1,Type=Integer,Description="Total read depth at the locus"
##==================
sample_relative_temp_filter <- sample_relative_temp[which(!is.na(sample_relative_temp$QualityScore)),]
sample_relative_temp_withONT <-sample_relative_temp_filter[which(sample_relative_temp_filter$QualityScore>5),]
table(sample_relative_temp_withONT$sample)
sample_relative_temp_withONT$site <- paste(sample_relative_temp_withONT$X.CHROM,sample_relative_temp_withONT$POS,sample_relative_temp_withONT$REF,sample_relative_temp_withONT$ALT,sep="_")
sample_relative_temp <- sample_relative_temp_withONT


# sample_relative_temp_withONT <-snps_freebayes_long %>% separate("Snps", into=c("GT","DP","AD","RO","QR","AO","QA","GL"), sep = ":") %>% transform(.,allel_frequency=as.numeric(AO)/as.numeric(DP))

# sample_relative_temp_withONT <- sample_relative_temp
sample_relative_temp_withONT$site <- paste(sample_relative_temp_withONT$X.CHROM,sample_relative_temp_withONT$POS,sample_relative_temp_withONT$REF,sample_relative_temp_withONT$ALT,sep="_")
sample_relative_safe_final_prep <- sample_relative_temp_withONT %>% select(site,X.CHROM,sample,allel_frequency)

sample_relative_safe_final_tsne <- spread(sample_relative_safe_final_prep, sample, allel_frequency)

sample_relative_wide <- sample_relative_safe_final_tsne[rowSums(sample_relative_safe_final_tsne[,3:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)>0,]
sample_relative_wide$species <- revalue(sample_relative_wide$X.CHROM, c("NZ_CP023139.1"="L. delbrueckii","NZ_CP048750.1"="S. thermophilus"))
sample_relative_wide$culture <- culturesss




##==================
##11_make long
##==================
colnames(sample_relative_wide)
sample_relative_long <- gather(sample_relative_wide, sample, Snps, colnames(sample_relative_wide)[3]:colnames(sample_relative_wide)[ncol(sample_relative_wide)-2], factor_key=TRUE,na.rm = TRUE) 
# nrow(sample_relative_long)
table(sample_relative_long$X.CHROM)

snps_long <- rbind(snps_long,sample_relative_long)
}

###====================================================
###====================================================
##specifically for strains
###====================================================
###====================================================
culturesss <- "FAMstrains"
#snps_long <- NA
for (culturesss in c("FAMstrains")) {
  

snps_freebayes <- read_delim(paste0("/home/vincent/Desktop/Projects/2020_strainDelineation/05_SNP_called/raw_snps_onlyforSTERM//prepedR/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK",culturesss,"_prepR.vcf"), "\t", escape_double = FALSE, trim_ws = TRUE) %>%  select(-c(QUAL,FILTER,FORMAT))


##==================
##01_filter SNVs MQM larger than 30 
##INFO=<ID=MQM,Number=A,Type=Float,Description="Mean mapping quality of observed alternate alleles">
##==================
snps_freebayes$MQM <- str_split_fixed(snps_freebayes$INFO, ";", 130)[,16] %>% gsub("MQM=","",.) 
snps_freebayes$MQM <- str_split_fixed(snps_freebayes$MQM, ",", 3)[,1] %>%  as.numeric()
snps_freebayes <- snps_freebayes[which(snps_freebayes$MQM>30),] %>% select(-MQM,-INFO)

##==================
##02_long list and frequency calc
##==================

snps_freebayes_long <- gather(snps_freebayes, sample, Snps, colnames(snps_freebayes)[6]:colnames(snps_freebayes)[ncol(snps_freebayes)], factor_key=TRUE,na.rm = TRUE) 

###------------------------calculate allel frequency   
###!!!!!!!filter allele frequency >0.8
###!!!!!!!only genomic SNVs


sample_relative_temp <-snps_freebayes_long %>% filter(Snps!=".:.:.:.:.:.:.:.") %>% separate("Snps", into=c("GT","DP","AD","RO","QR","AO","QA","GL"), sep = ":") %>% transform(.,allel_frequency=as.numeric(AO)/as.numeric(DP),QualityScore=as.numeric(QA)/as.numeric(DP),Sequencing_depth=as.numeric(DP)) %>% filter(allel_frequency>0.8) %>% filter(X.CHROM %in% c("NZ_CP023139.1","NZ_CP048750.1"))

# ggplot(sample_relative_temp,aes(x=allel_frequency))+geom_density()+facet_wrap(~X.CHROM,scales = "free")

##==================
##03_filter QA/DP > 30
##INFO=<ID=QA,Number=A,Type=Integer,Description="Alternate allele quality sum in phred">
##INFO=<ID=DP,Number=1,Type=Integer,Description="Total read depth at the locus"
##==================
sample_relative_temp_filter <- sample_relative_temp[which(!is.na(sample_relative_temp$QualityScore)),]
#table(sample_relative_temp_withONT$sample)
sample_relative_temp_filter$site <- paste(sample_relative_temp_filter$X.CHROM,sample_relative_temp_filter$POS,sample_relative_temp_filter$REF,sample_relative_temp_filter$ALT,sep="_")
sample_relative_temp <- sample_relative_temp_filter


# sample_relative_temp_withONT <-snps_freebayes_long %>% separate("Snps", into=c("GT","DP","AD","RO","QR","AO","QA","GL"), sep = ":") %>% transform(.,allel_frequency=as.numeric(AO)/as.numeric(DP))

sample_relative_temp_withONT <- sample_relative_temp
sample_relative_safe_final_prep <- sample_relative_temp_withONT %>% select(site,X.CHROM,sample,allel_frequency)

sample_relative_safe_final_tsne <- spread(sample_relative_safe_final_prep, sample, allel_frequency)

sample_relative_wide <- sample_relative_safe_final_tsne[rowSums(sample_relative_safe_final_tsne[,3:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)>0,]
sample_relative_wide$species <- revalue(sample_relative_wide$X.CHROM,  c("NZ_CP023139.1"="L. delbrueckii","NZ_CP048750.1"="S. thermophilus"))
sample_relative_wide$culture <- culturesss



##==================
##11_make long
##==================
colnames(sample_relative_wide)
sample_relative_long <- gather(sample_relative_wide, sample, Snps, colnames(sample_relative_wide)[3]:colnames(sample_relative_wide)[ncol(sample_relative_wide)-2], factor_key=TRUE,na.rm = TRUE) 
# nrow(sample_relative_long)
table(sample_relative_long$X.CHROM)

snps_long <- rbind(snps_long,sample_relative_long)
}


# sample_relative_temp  %>% group_by(X.CHROM) %>% summarize(meansample=mean(Sequencing_depth))
##==================
##remove samples
##==================

##the cheese samples from rmk202 are weird because they were passaged on plate before.
table(snps_long$sample) %>% length()
snps_long <- snps_long %>% filter(!sample %in% c("C_202_18_02","C_202_18_01"))
table(snps_long$sample)%>% length()

##==================
##11_make long
##==================
snps_long <- snps_long[-1,]

table(snps_long$sample)
#snps_long$samplesNew <- revalue(snps_long$sample,c("FSK_101_75"="starter 1","Lyo_101_76_4_7"="starter 2","Lyo_101_77"="starter 3","Lyo_101_02_7"="starter 4","Lyo_101_10_2"="starter 5","Lyo_101_16"="starter 6","ws_101_19"="starter 7","FSK_105_75"="starter 1","Lyo_105_76_4_1"="starter 2","Lyo_105_76_4_2"="starter 3","FSK_105_76_4_6"="starter 4","Lyo_105_76_2"="starter 5","Lyo_105_76_1"="starter 6","Lyo_105_76_5_11"="starter 7","Lyo_105_76_5_13"="starter 8","Lyo_105_76_5_12"="starter 9","Lyo_105_76_3_31"="starter 10","Lyo_105_11_5"="starter 11","Ws_105_19"="starter 12","Lyo_115_09"="starter 1","Ws_115_19"="starter 2","FSK_124_75_4"="starter 1","Lyo_124_08"="starter 2", "Ws_124_19"="starter 3","Lyo_150_76_4"="starter 1","Lyo_150_76_11"="starter 2","Lyo_150_11_3_11"="starter 3","Lyo_150_14_10"="starter 4","Lyo_150_18"="starter 5","Ws_150_19"="starter 6","Lyo_190_95_3"="starter 1","Lyo_190_17"="starter 2","Ws_190_19"="starter 3","Lyo_202_96"="starter 1","Lyo_202_12"="starter 2","Lyo_202_14"="starter 3","S_202_14"="starter 4","Ws_202_18"="starter 5","S_202_18"="starter 6", "C_202_18_02"="starter 7","C_202_18_01"="starter 8","Lyo_203_77"="starter 1","Lyo_203_96_1"="starter 2","Lyo_203_16"="starter 3","Ws_203_19"="starter 4","Lyo_280_16"="starter 1","Lyo_280_16_4"="starter 2", "Ws_280_19"="starter 3","Lyo_302_13"="starter 1", "Ws_302_19"="starter 2","FSK_305_76"="starter 1", "Lyo_305_96_1"="starter 2","Lyo_305_17"="starter 3","Ws_305_19"="starter 4"))


snps_long$samplesNew <- as.character(revalue(snps_long$sample,c("FSK_101_75"="starter 1","Lyo_101_76_4_7"="starter 2","Lyo_101_77"="starter 3","Lyo_101_02_7"="starter 4","Lyo_101_10_2"="starter 5","Lyo_101_16"="starter 6","ws_101_19"="starter 7","FSK_105_75"="starter 1","Lyo_105_76_4_1"="starter 2","Lyo_105_76_4_2"="starter 3","FSK_105_76_4_6"="starter 4","Lyo_105_76_2"="starter 5","Lyo_105_76_1"="starter 6","Lyo_105_76_5_11"="starter 7","Lyo_105_76_5_13"="starter 8","Lyo_105_76_5_12"="starter 9","Lyo_105_76_3_31"="starter 10","Lyo_105_11_5"="starter 11","Ws_105_19"="starter 12","Lyo_115_09"="starter 1","Ws_115_19"="starter 2","FSK_124_75_4"="starter 1","Lyo_124_08"="starter 2", "Ws_124_19"="starter 3","Lyo_150_76_4"="starter 1","Lyo_150_76_11"="starter 2","Lyo_150_11_3_11"="starter 3","Lyo_150_14_10"="starter 4","Lyo_150_18"="starter 5","Ws_150_19"="starter 6","Lyo_190_95_3"="starter 1","Lyo_190_17"="starter 2","Ws_190_19"="starter 3","Lyo_202_96"="starter 1","Lyo_202_12"="starter 2","Lyo_202_14"="starter 3","S_202_14"="starter 4","Ws_202_18"="starter 5","S_202_18"="starter 6", "C_202_18_02"="starter 7","C_202_18_01"="starter 8","Lyo_203_77"="starter 1","Lyo_203_96_1"="starter 2","Lyo_203_16"="starter 3","Ws_203_19"="starter 4","Lyo_280_16"="starter 1","Lyo_280_16_4"="starter 2", "Ws_280_19"="starter 3","Lyo_302_13"="starter 1", "Ws_302_19"="starter 2","FSK_305_76"="starter 1", "Lyo_305_96_1"="starter 2","Lyo_305_17"="starter 3","Ws_305_19"="starter 4",
"Vers_101_20"="starter 8", "Vers_105_20"="starter 13", 
"Vers_115_20"="starter 3", "Vers_124_20"="starter 4", "Vers_150_20"="starter 7", "Vers_190_20"="starter 4", "Vers_203_20"="starter 5", 
"Vers_280_20"="starter 4", "Vers_302_20"="starter 3", "Vers_305_20"="starter 5",
"Therm_101_20"="starter 9", "Therm_105_20"="starter 14", "Therm_115_20"="starter 4", "Therm_124_20"="starter 5", 
"Therm_150_20"="starter 8", "Therm_190_20"="starter 5", "Therm_203_20"="starter 6", "Therm_280_20"="starter 5", 
"Therm_302_20"="starter 4", "Therm_305_20"="starter 6",
"24h_101_20"="starter 10", "24h_105_20"="starter 15", "24h_115_20"="starter 5", "24h_124_20"="starter 6", "24h_150_20"="starter 9", 
"24h_190_20"="starter 6", "24h_203_20"="starter 7", "24h_280_20"="starter 6", "24h_302_20"="starter 5", "24h_305_20"="starter 7",
"M17x_101_20"="starter 11", "M17X_105_20"="starter 16", "M17X_115_20"="starter 6", 
"M17X_124_20"="starter 7", "M17x_150_20"="starter 10", "M17x_190_20"="starter 7", "M17x_202_20"="starter 9", "M17x_203_20"="starter 8", 
"M17x_280_20"="starter 7", "M17x_302_20"="starter 6", "M17x_305_20"="starter 8")))

sum(grepl("FAM",unique(snps_long$samplesNew)))

table(snps_long$samplesNew)
unique(snps_long$samplesNew)
sum(is.na(snps_long$samplesNew))

dput(unique(snps_long$samplesNew))

snps_long$samplesNew = factor(snps_long$samplesNew, levels=c("starter 10", "starter 5", "starter 9", "starter 6", "starter 4", 
"starter 3", "starter 2", "starter 7", "starter 1", "starter 11", 
"starter 8", "starter 13", "starter 12", "starter 14", "starter 15", 
"starter 16",  "FAM24786", "FAM24790", "FAM24759", "FAM24746", 
"FAM11071", "FAM24758", "FAM24757", "FAM24755", "FAM24776", "FAM24741", 
"FAM24749", "FAM24775", "FAM24754", "FAM24744", "FAM24743", "FAM24740", 
"FAM24738", "FAM24760", "FAM24729", "FAM24764", "FAM24731", "FAM24730", 
"FAM24727", "FAM24726", "FAM13496", "FAM11141", "FAM13497", "FAM24724", 
"FAM24753", "FAM24799", "FAM21747", "FAM13495", "FAM24739", "FAM24734", 
"FAM24752", "FAM11024", "FAM24736", "FAM24728", "FAM10980", "FAM13498", 
"FAM10973", "FAM12107", "FAM11009", "FAM11008", "FAM24745", "FAM21769", 
"FAM12273", "FAM12105", "FAM12104", "FAM13491", "FAM11049", "FAM24787", 
"FAM24756", "FAM24742", "FAM24774", "FAM11142", "FAM12080", "FAM12109", 
"FAM10996", "FAM24732", "FAM11063", "FAM24784", "FAM24791", "FAM13500", 
"FAM24725", "FAM24853", "FAM24770", "FAM21781", "FAM24750", "FAM13494", 
"FAM13493", "FAM24766", "FAM24767", "FAM24782", "FAM24768", "FAM24769", 
"FAM24772", "FAM24777", "FAM24748", "FAM24761", "FAM24778", "FAM24779", 
"FAM24780", "FAM24747", "FAM24773", "FAM24794", "FAM24781", "FAM11021", 
"FAM24783", "FAM24785", "FAM24735", "FAM24792", "FAM24763", "FAM24793", 
"FAM24795", "FAM24797", "FAM24798", "FAM24854", "FAM24855", "FAM11075", 
"FAM13492", "FAM11001", "FAM24762", "FAM13499", "FAM11143", "FAM12062", 
"FAM24751", "FAM21277", "FAM24765", "FAM24733", "FAM24771", "FAM24737"))
snps_long$samplesNew <- droplevels(snps_long$samplesNew)
# snps_long %>% filter(is.na(samplesNew)) %>% select(sample) %>% table()

sum(is.na(snps_long$samplesNew))

snps_long_final <- snps_long %>% select(site,species,samplesNew,culture,Snps)
table(snps_long$culture)

snps_long$samplesFINAL <- paste0(snps_long$culture,"-",snps_long$samplesNew)
table(snps_long$samplesFINAL )

# snps_long %>% filter(samplesFINAL=="FAMstrains-NA")

write.csv(snps_long_final,"~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all.csv",row.names = FALSE)
  # write.csv(naming_DF,"~/Desktop/Projects/2020_StarterCultureDiversity/98_script/naming_file.csv")


###----------------------------------------
##add SNPeffect INFO
###----------------------------------------


sample_relative_temp_cleaned <- snps_long_final

###spread the dataframe again

nrow(sample_relative_temp_cleaned)
table(sample_relative_temp_cleaned$culture)
sample_relative_temp_cleaned$samplesFINAL <- paste0(sample_relative_temp_cleaned$culture,"-",sample_relative_temp_cleaned$samplesNew)

table(sample_relative_temp_cleaned$samplesFINAL)

sample_relative_safe_final_prep <- sample_relative_temp_cleaned %>% select(site,species,samplesFINAL,Snps)

# table(sample_relative_safe_final_prep$samplesNew)
# table(sample_relative_safe_final_prep$culture)
table(sample_relative_safe_final_prep$samplesFINAL)

#nrow(sample_relative_safe_final_prep[grep("S_thermophilus_RMK202",sample_relative_safe_final_prep$site),])
sample_relative_safe_final_tsne <- spread(sample_relative_safe_final_prep, samplesFINAL, Snps)  %>%replace(is.na(.), 0)
nrow(sample_relative_safe_final_tsne)
# table(sample_relative_safe_final_prep$sample)
##==================
##05_filter rowsums!=0
##==================

##remove row sums ==0
nrow(sample_relative_safe_final_tsne)
sum(rowSums(sample_relative_safe_final_tsne[,3:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)==0)
# nrow(sample_relative_safe_final_tsne)


sample_relative_wide <- sample_relative_safe_final_tsne[rowSums(sample_relative_safe_final_tsne[,4:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)>0,]
colnames(sample_relative_wide)

# sample_relative_wide_phagesONly <- sample_relative_wide[grep("phage",sample_relative_wide$X.CHROM),]
# table(sample_relative_wide_phagesONly$X.CHROM)
# nrow(sample_relative_wide)
# sample_relative_wide <- subset(sample_relative_wide, select=-POS)
# sample_relative_wide$species <- sample_relative_wide$X.CHROM
# sample_relative_wide$species <- revalue(sample_relative_wide$species, c("CP046131"="L. delbrueckii","CP046134"="S. thermophilus"))
# sample_relative_wide$culture <- "RMK202"
# nrow(sample_relative_wide)
##==================
##06_add gene information
##==================

# RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new_startercultureDiversity/RMK202_snpeff_eggnog_repeats_core.txt")

# RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new_startercultureDiversity//RMK202_snpeff_eggnog_repeats_core_02.txt")
RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling//RMK202_snpeff_eggnog_length_core_extern.txt", col_types = cols(core = col_character(),geneName = col_character(),effect = col_character(),PGAP_name = col_character(),type = col_character(),typeName = col_character(),typeGene = col_character(),Preferred_name = col_character())) %>% select(-species)
table(RMK202_snpeff_eggnog_repeats_core$core)

# nrow(RMK202_snpeff_eggnog_repeats_core)
# table(RMK202_snpeff_eggnog_repeats_core$species)
# table(RMK202_snpeff_eggnog_repeats_core$core)
table(RMK202_snpeff_eggnog_repeats_core$core,RMK202_snpeff_eggnog_repeats_core$species)
# table(RMK202_snpeff_eggnog_repeats_core$core,RMK202_snpeff_eggnog_repeats_core$species)
# nrow(sample_relative_wide)
#RMK202_snpeff_eggnog_repeats_core_subset <- subset(RMK202_snpeff_eggnog_repeats_core, select=c(X.CHROM,site,finalName,effect,significance,geneName,GOs,EC,KEGG_ko,COG_Functional_Category,Repeat_identity,core))
dim(sample_relative_wide)
sample_relative_wide_description <- merge(sample_relative_wide,RMK202_snpeff_eggnog_repeats_core, by = "site",all.x = TRUE)
dim(sample_relative_wide_description)
table(sample_relative_wide_description$species)
# nrow(sample_relative_wide_description)
table(sample_relative_wide_description$core)
table(sample_relative_wide_description$significance)
#---------------------------------subset for interesting columns


# sample_relative_wide_description_interest <- subset(sample_relative_wide_description, select=c(species.x,site,culture,effect,significance,geneName,COG_Functional_Category,Repeat_cluster,core,culture,"starter 1","starter 2","starter 3","starter 4","starter 5","starter 6","starter 7","starter 8","starter 9","starter 10","starter 11","starter 12"))
dput(colnames(sample_relative_wide_description))
# site,species.x,effect,significance,geneName,COG_Functional_Category,Repeat_cluster,core
# sample_relative_wide_description_interest <- sample_relative_wide_description %>% select(species.x,site,culture,effect,significance,geneName,COG_Functional_Category,Repeat_cluster,core,culture,"starter 1","starter 2","starter 3","starter 4","starter 5","starter 6","starter 7","starter 8","starter 9","starter 10","starter 11","starter 12")
sample_relative_wide_description_interest <- sample_relative_wide_description %>% select(site,species,effect,significance,geneName,COG_Functional_Category,length,core,"101-starter 1", "101-starter 10", "101-starter 11", 
"101-starter 2", "101-starter 3", "101-starter 4", "101-starter 5", 
"101-starter 6", "101-starter 7", "101-starter 8", "101-starter 9", 
"105-starter 1", "105-starter 10", "105-starter 11", "105-starter 12", 
"105-starter 13", "105-starter 14", "105-starter 15", "105-starter 16", 
"105-starter 2", "105-starter 3", "105-starter 4", "105-starter 5", 
"105-starter 6", "105-starter 7", "105-starter 8", "105-starter 9", 
"115-starter 1", "115-starter 2", "115-starter 3", "115-starter 4", 
"115-starter 5", "115-starter 6", "124-starter 1", "124-starter 2", 
"124-starter 3", "124-starter 4", "124-starter 5", "124-starter 6", 
"124-starter 7", "150-starter 1", "150-starter 10", "150-starter 2", 
"150-starter 3", "150-starter 4", "150-starter 5", "150-starter 6", 
"150-starter 7", "150-starter 8", "150-starter 9", "190-starter 1", 
"190-starter 2", "190-starter 3", "190-starter 4", "190-starter 5", 
"190-starter 6", "190-starter 7", "202-starter 1", "202-starter 2", 
"202-starter 3", "202-starter 4", "202-starter 5", "202-starter 6", 
"202-starter 9", "203-starter 1", "203-starter 2", "203-starter 3", 
"203-starter 4", "203-starter 5", "203-starter 6", "203-starter 7", 
"203-starter 8", "280-starter 1", "280-starter 2", "280-starter 3", 
"280-starter 4", "280-starter 5", "280-starter 6", "280-starter 7", 
"302-starter 1", "302-starter 2", "302-starter 3", "302-starter 4", 
"302-starter 5", "302-starter 6", "305-starter 1", "305-starter 2", 
"305-starter 3", "305-starter 4", "305-starter 5", "305-starter 6", 
"305-starter 7", "305-starter 8", "FAMstrains-FAM10973", "FAMstrains-FAM10980", 
"FAMstrains-FAM10996", "FAMstrains-FAM11001", "FAMstrains-FAM11008", 
"FAMstrains-FAM11009", "FAMstrains-FAM11021", "FAMstrains-FAM11024", 
"FAMstrains-FAM11049", "FAMstrains-FAM11063", "FAMstrains-FAM11071", 
"FAMstrains-FAM11075", "FAMstrains-FAM11141", "FAMstrains-FAM11142", 
"FAMstrains-FAM11143", "FAMstrains-FAM12062", "FAMstrains-FAM12080", 
"FAMstrains-FAM12104", "FAMstrains-FAM12105", "FAMstrains-FAM12107", 
"FAMstrains-FAM12109", "FAMstrains-FAM12273", "FAMstrains-FAM13491", 
"FAMstrains-FAM13492", "FAMstrains-FAM13493", "FAMstrains-FAM13494", 
"FAMstrains-FAM13495", "FAMstrains-FAM13496", "FAMstrains-FAM13497", 
"FAMstrains-FAM13498", "FAMstrains-FAM13499", "FAMstrains-FAM13500", 
"FAMstrains-FAM21277", "FAMstrains-FAM21747", "FAMstrains-FAM21769", 
"FAMstrains-FAM21781", "FAMstrains-FAM24724", "FAMstrains-FAM24725", 
"FAMstrains-FAM24726", "FAMstrains-FAM24727", "FAMstrains-FAM24728", 
"FAMstrains-FAM24729", "FAMstrains-FAM24730", "FAMstrains-FAM24731", 
"FAMstrains-FAM24732", "FAMstrains-FAM24733", "FAMstrains-FAM24734", 
"FAMstrains-FAM24735", "FAMstrains-FAM24736", "FAMstrains-FAM24737", 
"FAMstrains-FAM24738", "FAMstrains-FAM24739", "FAMstrains-FAM24740", 
"FAMstrains-FAM24741", "FAMstrains-FAM24742", "FAMstrains-FAM24743", 
"FAMstrains-FAM24744", "FAMstrains-FAM24745", "FAMstrains-FAM24746", 
"FAMstrains-FAM24747", "FAMstrains-FAM24748", "FAMstrains-FAM24749", 
"FAMstrains-FAM24750", "FAMstrains-FAM24751", "FAMstrains-FAM24752", 
"FAMstrains-FAM24753", "FAMstrains-FAM24754", "FAMstrains-FAM24755", 
"FAMstrains-FAM24756", "FAMstrains-FAM24757", "FAMstrains-FAM24758", 
"FAMstrains-FAM24759", "FAMstrains-FAM24760", "FAMstrains-FAM24761", 
"FAMstrains-FAM24762", "FAMstrains-FAM24763", "FAMstrains-FAM24764", 
"FAMstrains-FAM24765", "FAMstrains-FAM24766", "FAMstrains-FAM24767", 
"FAMstrains-FAM24768", "FAMstrains-FAM24769", "FAMstrains-FAM24770", 
"FAMstrains-FAM24771", "FAMstrains-FAM24772", "FAMstrains-FAM24773", 
"FAMstrains-FAM24774", "FAMstrains-FAM24775", "FAMstrains-FAM24776", 
"FAMstrains-FAM24777", "FAMstrains-FAM24778", "FAMstrains-FAM24779", 
"FAMstrains-FAM24780", "FAMstrains-FAM24781", "FAMstrains-FAM24782", 
"FAMstrains-FAM24783", "FAMstrains-FAM24784", "FAMstrains-FAM24785", 
"FAMstrains-FAM24786", "FAMstrains-FAM24787", "FAMstrains-FAM24790", 
"FAMstrains-FAM24791", "FAMstrains-FAM24792", "FAMstrains-FAM24793", 
"FAMstrains-FAM24794", "FAMstrains-FAM24795", "FAMstrains-FAM24797", 
"FAMstrains-FAM24798", "FAMstrains-FAM24799", "FAMstrains-FAM24853", 
"FAMstrains-FAM24854", "FAMstrains-FAM24855")


# dim(sample_relative_wide_description)
# dim(sample_relative_wide_description_interest)
# table(sample_relative_wide_description_interest$core)

# sample_relative_wide_description_interest <- sample_relative_wide_description

##==================
##08_remove non-bacterial SNPs
##==================
table(sample_relative_wide_description_interest$species)
nrow(sample_relative_wide_description_interest)
# sample_relative_wide_description_interest_subset <- sample_relative_wide_description_interest[which(sample_relative_wide_description_interest$X.CHROM.x %in% c("L_delbrueckii_RMK202","S_thermophilus_RMK202")),]
sample_relative_wide_description_interest_subset <- sample_relative_wide_description_interest[which(sample_relative_wide_description_interest$species %in% c("L. delbrueckii","S. thermophilus")),]
nrow(sample_relative_wide_description_interest_subset)
table(sample_relative_wide_description_interest_subset$species)
sample_relative_wide <- sample_relative_wide_description_interest_subset
table(sample_relative_wide_description_interest_subset$core)
table(sample_relative_wide_description_interest_subset$core,sample_relative_wide_description_interest_subset$species)


##==================
##09_order variables properly
##==================


sample_relative_wide$significance <- factor(sample_relative_wide$significance, levels=c("MODIFIER","LOW","MODERATE","HIGH"))


##-------------------------------wirte to file

# write.xlsx(dN_dS_ratios_final_02,file="Users//Desktop/presenations/my_presentations/20191007_groupmeeting/figures/dN_dS_ratios_genes_rmk202.xls", sheetName = "Sheet1", 
  # col.names = TRUE, row.names = FALSE, append = FALSE)

##==================
##11_make long
##==================
colnames(sample_relative_wide)
# sample_relative_long <- gather(sample_relative_wide, sample, Snps, "starter 1":"starter 12", factor_key=TRUE,na.rm = TRUE) 
sample_relative_long <- gather(sample_relative_wide, sample, Snps, "101-starter 1":"FAMstrains-FAM24855", factor_key=TRUE,na.rm = TRUE) %>% filter(Snps>0)%>% filter(!is.na(Snps))
nrow(sample_relative_long)
table(sample_relative_long$species)
table(sample_relative_long$core)

# table(sample_relative_long$culture)
# sample_relative_long$samplesss < paste0(sample_relative_long$culture,"_",sample_relative_long$sample)
# sample_relative_long_again <- spread(sample_relative_long, site, Snps)%>%replace(is.na(.), 0) %>% column_to_rownames(var = "sampleName")


##==================
##12_make a unique gene infomoration
##==================

# RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/RMK202_snpeff_eggnog_repeats_core.txt")
# RMK202_snpeff_eggnog_repeats_core_uniuqes <- RMK202_snpeff_eggnog_repeats_core %>% select(-c("X1","X.CHROM","POS","REF","ALT","quality","site","effect","significance","best_tax_level")) %>% distinct()
# table(RMK202_snpeff_eggnog_repeats_core_uniuqes$core)
# nrow(RMK202_snpeff_eggnog_repeats_core_uniuqes)
# RMK202_snpeff_eggnog_repeats_core_uniuqes_final <- merge(RMK202_snpeff_eggnog_repeats_core_uniuqes,dN_dS_ratios_prep,by.y="gene",by.x="finalName",all.x = TRUE)  

# write.csv(RMK202_snpeff_eggnog_repeats_core_uniuqes_final,file="~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/RMK202_snpeff_eggnog_repeats_core_dNdS_uniqueGenes.txt")

# colnames(RMK202_snpeff_eggnog_repeats_core_uniuqes_final)
# RMK202_snpeff_eggnog_repeats_core_uniuqes_final_reduce <- RMK202_snpeff_eggnog_repeats_core %>% select(-c("finalName","Preferred_name","EC","KEGG_ko","KEGG_Reaction","CAZy","BiGG_Reaction","COG_Functional_Category","eggNOG free text description")) %>% distinct()

##==================
##output infomration
##==================
# sample_relative_wide
write.csv(sample_relative_wide,"~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all_withINfomration_wide.csv",row.names = FALSE)
# table(sample_relative_wide$core)
# unique(sample_relative_wide$core)
dim(sample_relative_wide)
sample_relative_wide[which(sample_relative_wide$core=="core"),]

snps_long_final <-  gather(sample_relative_wide, sample, Snps, "101-starter 1":"FAMstrains-FAM24855", factor_key=TRUE,na.rm = TRUE)
snps_long_final[which(snps_long_final$core=="core"),]

snps_long_final$species <- snps_long_final$species
snps_long_final$culture <- str_split_fixed(snps_long_final$sample,"-",3)[,1]
# snps_long_final <- snps_long_final %>% select(-"species.x")
table(snps_long_final$core)
```

#plot snps

###### Sterm strains
```{r}
library(tidyverse)

sample_relative_wide <- read_csv("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all_withINfomration_wide.csv")

snps_wide_final <- read_csv("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all_withINfomration_wide.csv")
# write.csv(sample_relative_wide,"~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all_withINfomration_wide.csv",row.names = FALSE)


dim(snps_wide_final)
# snps_long_final <- gather(sample_relative_wide, sample, Snps, "starter 1":"starter 12", factor_key=TRUE,na.rm = TRUE) 
# snps_long_final <-  sample_relative_wide %>%  select(site,species,effect,significance,geneName,COG_Functional_Category,core, "101-starter 1", 
snps_long_final <-  sample_relative_wide %>%  select(site,species,"101-starter 1", 
"101-starter 10", "101-starter 11", "101-starter 2", "101-starter 3", 
"101-starter 4", "101-starter 5", "101-starter 6", "101-starter 7", 
"101-starter 8", "101-starter 9", "105-starter 1", "105-starter 10", 
"105-starter 11", "105-starter 12", "105-starter 13", "105-starter 14", 
"105-starter 15", "105-starter 16", "105-starter 2", "105-starter 3", 
"105-starter 4", "105-starter 5", "105-starter 6", "105-starter 7", 
"105-starter 8", "105-starter 9", "115-starter 1", "115-starter 2", 
"115-starter 3", "115-starter 4", "115-starter 5", "115-starter 6", 
"124-starter 1", "124-starter 2", "124-starter 3", "124-starter 4", 
"124-starter 5", "124-starter 6", "124-starter 7", "150-starter 1", 
"150-starter 10", "150-starter 2", "150-starter 3", "150-starter 4", 
"150-starter 5", "150-starter 6", "150-starter 7", "150-starter 8", 
"150-starter 9", "190-starter 1", "190-starter 2", "190-starter 3", 
"190-starter 4", "190-starter 5", "190-starter 6", "190-starter 7", 
"202-starter 1", "202-starter 2", "202-starter 3", "202-starter 4", 
"202-starter 5", "202-starter 6", "202-starter 9", "203-starter 1", 
"203-starter 2", "203-starter 3", "203-starter 4", "203-starter 5", 
"203-starter 6", "203-starter 7", "203-starter 8", "280-starter 1", 
"280-starter 2", "280-starter 3", "280-starter 4", "280-starter 5", 
"280-starter 6", "280-starter 7", "302-starter 1", "302-starter 2", 
"302-starter 3", "302-starter 4", "302-starter 5", "302-starter 6", 
"305-starter 1", "305-starter 2", "305-starter 3", "305-starter 4", 
"305-starter 5", "305-starter 6", "305-starter 7", "305-starter 8") %>% gather(., sample, Snps, "101-starter 1":"305-starter 8", factor_key=TRUE,na.rm = TRUE)



##----------------
##analysis
##----------------

table(snps_long_final$core)
nrow(snps_long_final)
sample_relative_safe_woSTRAINS_sub01 <- snps_long_final
sample_relative_safe_woSTRAINS_sub01$culture <- str_split_fixed(sample_relative_safe_woSTRAINS_sub01$sample, "-", 3)[,1]
sample_relative_safe_woSTRAINS_sub01$species.x <- sample_relative_safe_woSTRAINS_sub01$species

table(sample_relative_safe_woSTRAINS_sub01$species)
table(sample_relative_safe_woSTRAINS_sub01$culture)
table(sample_relative_safe_woSTRAINS_sub01$sample)

table(sample_relative_safe_woSTRAINS_sub01$culture,sample_relative_safe_woSTRAINS_sub01$sample)


# sample_relative_safe_woSTRAINS_sub01 <- sample_relative_safe_woSTRAINS[which(sample_relative_safe_woSTRAINS$significance!="MODIFIER"),]
nrow(sample_relative_safe_woSTRAINS_sub01)
sample_relative_safe_woSTRAINS_sub02 <- sample_relative_safe_woSTRAINS_sub01
# sample_relative_safe_woSTRAINS_sub02 <- sample_relative_safe_woSTRAINS_sub01[which(sample_relative_safe_woSTRAINS_sub01$core=="core"),]
nrow(sample_relative_safe_woSTRAINS_sub02)

sample_relative_safe_woSTRAINS_sub03 <- sample_relative_safe_woSTRAINS_sub02[which(sample_relative_safe_woSTRAINS_sub02$Snps>0.03),]
# sample_relative_safe_woSTRAINS_sub05 <- sample_relative_safe_woSTRAINS_sub03
sample_relative_safe_woSTRAINS_sub04 <- sample_relative_safe_woSTRAINS_sub02[which(sample_relative_safe_woSTRAINS_sub02$Snps<=0.03),]
# sample_relative_safe_woSTRAINS_sub04 <- sample_relative_safe_woSTRAINS_sub02[which(sample_relative_safe_woSTRAINS_sub02$Snps>0.03),]

sample_relative_safe_woSTRAINS_sub04$Snps <- 0
nrow(sample_relative_safe_woSTRAINS_sub04)
sample_relative_safe_woSTRAINS_sub05 <- rbind(sample_relative_safe_woSTRAINS_sub03,sample_relative_safe_woSTRAINS_sub04)
sample_relative_safe_woSTRAINS_sub06 <- sample_relative_safe_woSTRAINS_sub05[which(sample_relative_safe_woSTRAINS_sub05$significance %in% c("MODERATE","HIGH")),]
sample_relative_safe_woSTRAINS_sub06 <- sample_relative_safe_woSTRAINS_sub05

nrow(sample_relative_safe_woSTRAINS_sub05)
table(sample_relative_safe_woSTRAINS_sub05$species)
# sample_relative_safe_Sterm_final <- sample_relative_safe_woSTRAINS_sub06[which(sample_relative_safe_woSTRAINS_sub06$species %in %"S. thermophilus"),]

sample_relative_safe_Sterm_final <- sample_relative_safe_woSTRAINS_sub06[which(sample_relative_safe_woSTRAINS_sub06$species %in% c("L. delbrueckii","S. thermophilus")),] %>% select(-species.x) 
# sample_relative_safe_Sterm_final <- sample_relative_safe_Sterm_final[!duplicated(sample_relative_safe_Sterm_final), ]
sample_relative_safe_Sterm_final <- unique(sample_relative_safe_Sterm_final)

sample_relative_safe_Sterm_final[4092,]
sample_relative_safe_Sterm_final[47853,]


sample_relative_safe_Sterm_final_wide <- spread(sample_relative_safe_Sterm_final, sample, Snps)  %>%replace(is.na(.), 0) %>% as_tibble()
sample_relative_safe_Sterm_final_wide
table(sample_relative_safe_Sterm_final$sampleComplete)
# sample_relative_safe_Sterm_final_new <- sample_relative_safe_Sterm_final %>% select(c(-"species.x",-"effect",-"significance",-"geneName",-"COG_Functional_Category",-"core"))

sample_relative_safe_Sterm_final_new <- sample_relative_safe_Sterm_final

sample_relative_safe_Sterm_final_new$sampleComplete <- sample_relative_safe_Sterm_final_new$sample
sample_relative_safe_Sterm_final_new <- sample_relative_safe_Sterm_final_new %>% select(c(-"culture",-"sample"))
# sample_relative_safe_Sterm_final_new <- sample_relative_safe_Sterm_final_new %>% select(c(-"sample"))

sample_relative_safe_Sterm_final_wide <- spread(sample_relative_safe_Sterm_final_new, sampleComplete, Snps)  %>%replace(is.na(.), 0)
# sample_relative_safe_Sterm_final_wide[] %>% select()
colnames(sample_relative_safe_Sterm_final_wide)
dim(sample_relative_safe_Sterm_final_wide)

sample_relative_safe_Sterm_final_wide_filtered <- sample_relative_safe_Sterm_final_wide[,c(1,2,which(colSums(sample_relative_safe_Sterm_final_wide[,c(-1,-2)])>0)+2)]
dim(sample_relative_safe_Sterm_final_wide_filtered)
colnames(sample_relative_safe_Sterm_final_wide_filtered)

sample_relative_safe_Sterm_final_wide_filtered_long <-  gather(sample_relative_safe_Sterm_final_wide_filtered, sampleComplete, Snps, "101-starter 1":"305-starter 8", factor_key=TRUE,na.rm = TRUE) 

sample_relative_safe_Sterm_final_wide_filtered_long$culture <- str_split_fixed(sample_relative_safe_Sterm_final_wide_filtered_long$sampleComplete, "-", 2)[,1]

sample_relative_safe_Sterm_final_wide_filtered_long$sample <- str_split_fixed(sample_relative_safe_Sterm_final_wide_filtered_long$sampleComplete, "-", 2)[,2]
sample_relative_safe_Sterm_final_wide_filtered_long$species <- str_split_fixed(sample_relative_safe_Sterm_final_wide_filtered_long$site, "_", 4)[,2]
table(sample_relative_safe_Sterm_final_wide_filtered_long$sample)
  # sample_relative_safe_Sterm_final_longAll <-
sample_relative_safe_woSTRAINS <- sample_relative_safe_Sterm_final_wide_filtered_long
sample_relative_safe_Sterm_final_wide_filtered_long$sample = factor(sample_relative_safe_Sterm_final_wide_filtered_long$sample, levels=c("starter 1","starter 2","starter 3","starter 4","starter 5","starter 6","starter 7","starter 8","starter 9","starter 10","starter 11","starter 12","starter 13","starter 14","starter 15","starter 16"))
table(sample_relative_safe_Sterm_final_wide_filtered_long$culture,sample_relative_safe_Sterm_final_wide_filtered_long$sample)
  ##============
### plot Streptococcus thermophius
   ##============
# droplevels
sample_relative_safe_woSTRAINS$sample <- base::droplevels(factor(sample_relative_safe_woSTRAINS$sample))
levels(sample_relative_safe_Sterm_final_wide_filtered_long$sample)
sample_relative_safe_Sterm_final_wide_filtered_long_woZero <- sample_relative_safe_Sterm_final_wide_filtered_long %>% filter(Snps>0)
  ##============
### plot 
   ##============

all_colours <- rev(c("#EB4D4D","#10B552") )
all_colours <- rev(c("#EB4D4D","#10B552") )

# all_colours <- rev(c("#EB4D4D") ) 

# sample_relative_safe_Sterm_final <- sample_relative_safe_woSTRAINS_sub06[which(sample_relative_safe_woSTRAINS_sub06$species %in% c("L. delbrueckii","S. thermophilus")),]
table(sample_relative_safe_Sterm_final_wide_filtered_long_woZero$species)
sample_relative_safe_Sterm_final_wide_filtered_long_woZero$species <- plyr::revalue(sample_relative_safe_Sterm_final_wide_filtered_long_woZero$species,c("xy"="L. delbrueckii","CP048750.1"="S. thermophilus"))
# 
# table(sample_relative_safe_Sterm_final_wide_filtered_long_woZero$species)

# colnames(sample_relative_safe_woSTRAINS)
table(sample_relative_safe_Sterm_final_wide_filtered_long_woZero$culture,sample_relative_safe_Sterm_final_wide_filtered_long_woZero$sample)
##plot


  p2 <- ggplot(sample_relative_safe_Sterm_final_wide_filtered_long_woZero,aes(x=sample,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.3, alpha=.1)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = "free_x", space = "free")+
            facet_grid(species~culture,scales = "free_x", space = "free")+
            # ggforce::facet_row(vars(culture)~species,scales = "free_x", space = "free")+
            # ggforce::facet_col(vars(species),scales = "free_x", space = "free")+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
          panel.spacing = unit(1.2, "lines")
  )
  
  p2
  

# svg("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.svg",width=2.5,height=2.5)
# png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_sterm.png", width = 1800, height = 4800,res=300)
# png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_sterm_wide.png", width = 6000, height = 1800,res=300)
# png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_sterm_wide.png", width = 6000, height = 1800,res=300)
svg("~/Desktop/mid_thesis/report/figures/20200430/chapter2/snpscalled_sterm_wide.svg", width = 12, height = 6)


p2
dev.off()

##-------------------------------------
#only Sterm

all_colours <- rev(c("#007fa6ff") )

sample_relative_safe_Sterm_final_wide_filtered_long_woZero$species


sample_relative_safe_Sterm_final_wide_filtered_long_woZero_sterm <- sample_relative_safe_Sterm_final_wide_filtered_long_woZero%>% filter(species=="S. thermophilus")
 p3 <- ggplot(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_sterm,aes(x=sample,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.2, alpha=.05)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = "free_x", space = "free")+
            # facet_grid(species~culture,scales = "free_x", space = "free")+
               facet_grid(~culture,scales = "free_x", space = "free")+

            # ggforce::facet_row(vars(culture)~species,scales = "free_x", space = "free")+
            # ggforce::facet_col(vars(species),scales = "free_x", space = "free")+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
   ylim(c(0,1))+
    # theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
   theme(axis.text.x = element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
          panel.spacing = unit(1, "lines")
  )
  
  # p3
  
    pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/AbundanceStrain_SNPs_all.pdf",width=12,height=8)
 p3
  dev.off()

```


## strain phasing

```{r}
##===============================
#old
##===============================
colnames(sample_relative_wide) <- colnames(sample_relative_wide) %>% gsub("FAMstrains-FAM","",.)

dput(colnames(sample_relative_wide))

sample_relative_wide_snpsUsed <- sample_relative_wide    %>% filter(core=="core")  %>%  select("site",
"10973", "10980", "10996", 
"11001", "11008", "11009", "11021", "11024", "11049", "11063", 
"11071", "11075", "11141", "11142", "11143", "12062", "12080", 
"12104", "12105", "12107", "12109", "12273", "13491", "13492", 
"13493", "13494", "13495", "13496", "13497", "13498", "13499", 
"13500", "21277", "21747", "21769", "21781", "24724", "24725", 
"24726", "24727", "24728", "24729", "24730", "24731", "24732", 
"24733", "24734", "24735", "24736", "24737", "24738", "24739", 
"24740", "24741", "24742", "24743", "24744", "24745", "24746", 
"24747", "24748", "24749", "24750", "24751", "24752", "24753", 
"24754", "24755", "24756", "24757", "24758", "24759", "24760", 
"24761", "24762", "24763", "24764", "24765", "24766", "24767", 
"24768", "24769", "24770", "24771", "24772", "24773", "24774", 
"24775", "24776", "24777", "24778", "24779", "24780", "24781", 
"24782", "24783", "24784", "24785", "24786", "24787", "24790", 
"24791", "24792", "24793", "24794", "24795", "24797", "24798", 
"24799", "24853", "24854", "24855")%>%replace(is.na(.), 0)
sample_relative_wide_snpsUsed <- sample_relative_wide_snpsUsed[rowSums(sample_relative_wide_snpsUsed[-1])>0.5,]
dim(sample_relative_wide_snpsUsed)
is.na(colnames(sample_relative_wide_snpsUsed))

tmp <- data.frame(sample_relative_wide_snpsUsed[-1]>0.5)*1
colnames(tmp) <- gsub("X","",colnames(tmp))
length(colnames(tmp) )

SNVs_strains <- cbind(group=sample_relative_wide_snpsUsed[,1],tmp)
dim(SNVs_strains)


Poppunk_out <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/Poppunk_out.all","\t", escape_double = FALSE, col_names = c("genome","group"), trim_ws = TRUE)
dim(Poppunk_out)
# colnames(SNVs_strains) %IN%
Poppunk_out <- rbind(Poppunk_out,data.frame(genome="24777",group="13"))

Poppunk_out %>% filter(genome==24777)

for (genomesss in colnames(SNVs_strains)[-1]) {
  colnames(SNVs_strains)[which(colnames(SNVs_strains)==genomesss)] <- Poppunk_out[grep(genomesss,Poppunk_out$genome),"genome"]
}

Poppunk_out_sub <- Poppunk_out %>% filter(genome %in%  colnames(SNVs_strains)[-1])

dim(Poppunk_out_sub)

write.table(Poppunk_out_sub,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/Poppunk_sub_forSNVS.all",sep="\t",quote = FALSE,row.names = FALSE)

write.table(SNVs_strains,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/gene_presence_absence.Rtab",sep="\t",quote = FALSE,row.names = FALSE)
```

```{bash}


scp ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/gene_presence_absence.Rtab vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/SNVprofiling/20210208/

scp ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/Poppunk_sub_forSNVS.all vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/SNVprofiling/20210208/

```


Here, I apply Gals paper appraoch. 
Github can be found [here](https://github.com/ghoresh11/twilight).


```{bash}

rm -r  /data/Project/2020_StarterCultureDiversity/SNVprofiling/20210208//Gals_analysis/
mkdir -p /data/Project/2020_StarterCultureDiversity/SNVprofiling/20210208//Gals_analysis/
cd /data/Project/2020_StarterCultureDiversity/SNVprofiling/20210208//Gals_analysis/

Rscript /home/vincent/apps/twilight/classify_genes_VS.R --min_size 1 -p /data/Project/2020_StarterCultureDiversity/SNVprofiling/20210208/gene_presence_absence.Rtab -g /data/Project/2020_StarterCultureDiversity/SNVprofiling/20210208/Poppunk_sub_forSNVS.all


grep -i "lineage specific" out/genes_per_isolate.tab 
grep -i "lineage specific"  out/classification.tab


```

```{bash}



scp  -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/SNVprofiling/20210208/Gals_analysis/out/ ~/Desktop/Projects/2020_StarterCultureDiversity/SNVprofiling/20210208/
```

first have to run plot SNPs sterm chunk. 

```{r}

library(tidyverse)

classification_SNVs <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/SNVprofiling/20210208/out/classification.tab", "\t", escape_double = FALSE, trim_ws = TRUE)

classification_SNVs_lineage <- classification_SNVs %>% filter(grepl("Lineage specific ",specific_class))

classification_SNVs_lineage$lineage <- gsub("[^[:digit:]., ]", "", classification_SNVs_lineage$details) %>% as.numeric()
classification_SNVs_lineage_cleaned <- classification_SNVs_lineage %>% select(gene_name,lineage,specific_class)

lineage_relabel <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/lineage_relabel.csv")
classification_SNVs_lineage_cleaned <- merge(classification_SNVs_lineage_cleaned,lineage_relabel,by.x="lineage",by.y="poppunk",all.x=TRUE)
##------------------------------------------
##plotting
##------------------------------------------
sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended <- merge(sample_relative_safe_Sterm_final_wide_filtered_long_woZero,classification_SNVs_lineage_cleaned,by.x="site",by.y="gene_name") %>% filter(groupAssignment!="13")

sum(is.na(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended$specific_class))

dim(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended)

sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended  %>% dim()
sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended %>% filter((culture %in% c(101,105,115,124,150,190) & bacsfrom %in% c("emmentaler","mix")) | (culture %in% c(202,203,280) & bacsfrom %in% c("gruyere"))| (culture %in% c(302,305) & bacsfrom %in% c("sbrinz","mix")) ) %>% dim()
sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended <- sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended %>% filter((culture %in% c(101,105,115,124,150,190) & bacsfrom %in% c("emmentaler","mix")) | (culture %in% c(202,203,280) & bacsfrom %in% c("gruyere"))| (culture %in% c(302,305) & bacsfrom %in% c("sbrinz","mix")) )


# sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended %>% filter((culture %in% c(202,203,280) & bacsfrom =="gruyere") ) %>% dim()
# table(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended$bacsfrom)

 p3 <- ggplot(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended,aes(x=sample,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.3, alpha=.8)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = "free_x", space = "free")+
            facet_grid(groupAssignment~culture,scales = "free_x", space = "free")+
            # ggforce::facet_row(vars(culture)~species,scales = "free_x", space = "free")+
            # ggforce::facet_col(vars(species),scales = "free_x", space = "free")+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
          panel.spacing = unit(1.2, "lines")
  )
  
  p3
  
  ###------------------------
  ##relative abundance of SNVs and strains
  ###------------------------
  
#  classification_SNVs_multi <- classification_SNVs %>% filter(total==11)
#  classification_SNVs_multi_withoutONE <- classification_SNVs_multi[(gsub(" Inter:.*","",classification_SNVs_multi$details ) %>% gsub("Core: " ,"",.)) =="10+11+12+2+3+4+5+6+7+8+9",]
#  
# sample_relative_safe_Sterm_final_wide_filtered_long_woZero %>% filter(site %in% classification_SNVs_multi_withoutONE$gene_name) %>% group_by(sampleComplete) %>% dplyr::summarize(medianss=median(Snps))

  
  sample_medians <- sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended %>% filter(!groupAssignment %in% c(13,14)) %>% group_by(sampleComplete,groupAssignment) %>% dplyr::summarize(medianss=median(Snps),Errors=sd(Snps),nsss=length(Snps))
  sample_medians$groupAssignment <- as.character(sample_medians$groupAssignment)
  sample_medians$sample <- str_split_fixed(sample_medians$sampleComplete, "-", 2)[,2]
      sample_medians$culture <- as.character(str_split_fixed(sample_medians$sampleComplete, "-", 2)[,1])

  sample_medians$groupAssignment = factor(sample_medians$groupAssignment, levels=as.character(1:12))

    cols <- rainbow(14)[1:12] #I am not sure this is correct -- did not find the correct color labels anymore

  ggplot(sample_medians,aes(x=sample,y=medianss,fill=groupAssignment))+geom_bar(position="stack", stat="identity",alpha=0.5)+facet_grid(~culture,scales = "free_x", space = "free")+theme_classic()+scale_fill_manual(values = cols)+scale_color_manual(values = cols)
  
  
  ###--------------------------------
  ##normalise counts
  ###--------------------------------
  
  
    sample_medians_max <- sample_medians %>% group_by(sampleComplete) %>% dplyr::summarize(factorsss=sum(medianss))

 sample_medians_normalised <-  merge(sample_medians,sample_medians_max,by="sampleComplete")
  
sample_medians_normalised$normalised_ratio <- sample_medians_normalised$medianss/sample_medians_normalised$factorsss


  sample_medians_normalised$groupAssignment = factor(sample_medians_normalised$groupAssignment, levels=as.character(1:12))


  ggplot(sample_medians_normalised,aes(x=sample,y=normalised_ratio,fill=groupAssignment))+geom_bar(position="stack", stat="identity",alpha=0.5)+facet_grid(~culture,scales = "free_x", space = "free")+theme_classic()+scale_fill_manual(values = cols)+scale_color_manual(values = cols)

  
  sample_medians_normalised_reduced <- sample_medians_normalised %>% select(sampleComplete,groupAssignment,sample, culture , normalised_ratio)
  
  
  ####-----------------------------------
  #how many strains are lost
  ####-----------------------------------
  
  sample_medians_normalised_reduced
  
  
  
  ###--------------------------------
  ##add missing samples
  ##some metagenomes do not have any strain infomration
  #here, I find out which ones and then I add the strain which we isolated from
  ###--------------------------------
  
  
  
  unique(sample_medians_normalised$sampleComplete) %>% length()
  sample_medians_normalised %>% select(sampleComplete,sample,culture) %>% unique() %>% dim()

  strained_samples <- unique(sample_medians_normalised$sampleComplete) %>% as.character()
  
  colnames(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended)
  table(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended$sampleComplete) %>% length()
  
  all_samples <- unique(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended$sampleComplete) %>% levels()
  
  # strained_samples %in%  all_samples
  
   all_samples[ which(!all_samples %in%  strained_samples)]
   
   ##!!!!!!this has to be changed if rerun
   
   all_sterm_strain_info_ZOOMED <- read_csv("Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) %>% filter(species=="S.thermophilus") %>% select(c(strain,Cheese,starterCulture,Sequencing)) %>% remove_rownames()%>% column_to_rownames(., var = "strain")%>%replace(is.na(.), "Unknown") 
   
   ##-------the two 302 have lineage 4 strains
     all_sterm_strain_info_ZOOMED %>% filter(starterCulture=="302")

     addon302 <- data.frame(sampleComplete=c("302-starter 4","302-starter 5"),
               groupAssignment="4",
               sample=c("starter 4","starter 5"),
               culture="302",
               normalised_ratio=1)
     
   ##-------the five 115 metagenomes have lineage 7

  all_sterm_strain_info_ZOOMED %>% filter(starterCulture=="115")
  
  addon115 <- data.frame(sampleComplete=c( "115-starter 1" ,"115-starter 2" ,"115-starter 3", "115-starter 4" ,"115-starter 5", "115-starter 6"),
               groupAssignment="7",
               sample=c("starter 1","starter 2","starter 3","starter 4","starter 5","starter 6"),
               culture="115",
               normalised_ratio=1)
  
  ##---------bring together
  
  
 sample_medians_normalised_final <-  rbind(sample_medians_normalised_reduced,addon115,addon302)
 
 
 sample_medians_normalised_final$sample = factor(sample_medians_normalised_final$sample, levels=c("starter 1","starter 2","starter 3","starter 4","starter 5","starter 6","starter 7","starter 8","starter 9","starter 10","starter 11","starter 12","starter 13","starter 14","starter 15","starter 16"))
 
   final_strain_abundance_plot <- ggplot(sample_medians_normalised_final,aes(x=sample,y=normalised_ratio,fill=groupAssignment))+geom_bar(position="stack", stat="identity",alpha=0.5)+facet_grid(~culture,scales = "free_x", space = "free")+theme_classic()+scale_fill_manual(values = cols)+scale_color_manual(values = cols)+labs(x="",y="Relative abundance")+theme(legend.title = element_blank(),axis.text.x = element_text(angle = 75, hjust = 1,size=6))+ scale_y_continuous(expand = c(0,0))
   
   final_strain_abundance_plot
   
   
   pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/AbundanceStrain_all.pdf",width=10,height=3)

    # svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/AbundanceStrain_all.svg",width=10,height=3)
   #png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/AbundanceSpecies_new.png", width = 1800, height = 1500,res=300)


   final_strain_abundance_plot

dev.off()
  ###--------------------------------
  ##can assign the write strain based on the picked isolates
   #becuase some metagenomes have similiar abundant strains 
  ###--------------------------------
   
     all_sterm_strain_info_ZOOMED %>% filter(starterCulture=="101")
     all_sterm_strain_info_ZOOMED %>% filter(starterCulture=="105")
     all_sterm_strain_info_ZOOMED %>% filter(starterCulture=="124")

     ##doesn't work
  ###--------------------------------
  ##number of strains over time
  ###-------------------------------- 
     
     sample_medians_normalised_final <- sample_medians_normalised_final %>% as_tibble()
     table(sample_medians_normalised_final$culture)
     sample_medians_normalised_final_numberSTRAINs <- sample_medians_normalised_final %>%  group_by(culture,sample) %>% summarise(strains=n())
     
     
     final_strain_abundance_plot_diveristy <- ggplot(sample_medians_normalised_final_numberSTRAINs,aes(x=sample,y=strains,group=1))+theme_classic()+facet_grid(~culture,scales = "free_x", space = "free")+geom_point(stat='summary', fun.y=sum) +stat_summary(fun.y=sum, geom="line")+labs(x="",y="Number of strains")+theme(legend.title = element_blank(),axis.text.x = element_text(angle = 75, hjust = 1,size=6))
   
   final_strain_abundance_plot_diveristy
     
   
     # pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement/Strain_count_samples.pdf",width=10,height=3)
   png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement/Strain_count_samples.png", width = 1800, height = 1500,res=300)


   final_strain_abundance_plot_diveristy

dev.off()
```

###### Sterm nucleotide diversity

```{r}
snps_long_final <-  gather(sample_relative_wide, sample, Snps, "101-starter 1":"FAMstrains-FAM24855", factor_key=TRUE,na.rm = TRUE)

table(sample_relative_wide$species)



coregenomeSizeSterm=1300000
colnames(snps_long_final)
table(snps_long_final$species)
# sample_relative_wide %>% filter(core=="core") %>% dim()
snps_long_final_cleaned <- snps_long_final%>% filter(core=="core") %>% filter(Snps>0.05) %>% filter(species=="S. thermophilus") %>% group_by(sample) %>% dplyr::summarize(counts=n()) %>% mutate(nuclotideDiv=100*(counts/coregenomeSizeSterm))%>% separate(sample, c("culture", "sample"),sep = "-" )

COLORSSS <- c("#d45400ff","#de592bff","#e36b42ff","#e67d59ff","#eb8f70ff","#eea38aff","#009cf0ff","#33cffaff","#7fe3ffff","#007f7fff","#00dcdcff","#ff0000ff","#0000ffff","grey","#fde623ff","#00ff00ff")


nucleotide_div_plot <- ggplot(snps_long_final_cleaned,aes(x=culture,y=nuclotideDiv,fill=culture))+geom_boxplot()+theme_classic()+scale_fill_manual(values=COLORSSS)+theme(legend.position ="none",axis.title.x=element_blank(),axis.text.x = element_text(angle = 75, hjust = 1))+labs(y="Nucleotide diversity [%]")
nucleotide_div_plot
pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/nucleotide_div.pdf",width=3.25,height=2.5)

nucleotide_div_plot

dev.off()

###----------------------------------------------
#nucleotide diversity both species

coregenomeSizeSterm=1300000
colnames(snps_long_final)
table(snps_long_final$species)
# sample_relative_wide %>% filter(core=="core") %>% dim()
snps_long_final_cleaned <- snps_long_final %>% filter(Snps>0.05)  %>% group_by(species,sample) %>% dplyr::summarize(counts=n()) %>% mutate(nuclotideDiv=100*(counts/coregenomeSizeSterm)) %>% separate(sample, c("culture", "sample"),sep = "-" ) %>% filter(culture!="FAMstrains")

COLORSSS <- c("#d45400ff","#de592bff","#e36b42ff","#e67d59ff","#eb8f70ff","#eea38aff","#009cf0ff","#33cffaff","#7fe3ffff","#007f7fff","#00dcdcff","#ff0000ff","#0000ffff","grey","#fde623ff","#00ff00ff")


nucleotide_div_plot <- ggplot(snps_long_final_cleaned,aes(x=culture,y=nuclotideDiv,fill=culture))+geom_boxplot()+theme_classic()+scale_fill_manual(values=COLORSSS)+theme(legend.position ="none",axis.title.x=element_blank(),axis.text.x = element_text(angle = 75, hjust = 1))+labs(y="Nucleotide diversity [%]")+facet_wrap(~species,scales = "free",ncol = 1)
nucleotide_div_plot

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement/nucleotide_div.pdf",width=3.25,height=2.5)
png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement/nucleotide_div.png", width = 3000, height = 3000,res=300)

nucleotide_div_plot

dev.off()


snps_long_final_cleaned %>% group_by(species) %>% summarise(means=mean(nuclotideDiv),medians=median(nuclotideDiv),sdss=sd(nuclotideDiv))

#---------------------------------
##


unique(snps_long_final_cleaned$sample) %>% sort()
snps_long_final_cleaned$sample = factor(snps_long_final_cleaned$sample, levels=c("starter 1", "starter 2",  "starter 3" , "starter 4" , "starter 5" , "starter 6" , "starter 7" , "starter 8" , "starter 9" ,"starter 10" ,"starter 11" ,"starter 12" ,"starter 13","starter 14" ,"starter 15" ,"starter 16" ))



nucleotide_Div_time <- ggplot(snps_long_final_cleaned,aes(x=sample,nuclotideDiv,color=species,fill=species,group=species))+geom_point()+geom_line()+facet_wrap(~culture,scales="free")+theme_classic()+theme(legend.position = "none",axis.text.x = element_text(angle = 75, hjust = 1,size=9))

nucleotide_Div_time

png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/nucleotide_diversity_over_time.png", width = 3000, height = 3000,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
nucleotide_Div_time
dev.off()
###------------------------------------
##-------finalise figure similar to species diversity
###------------------------------------


coregenomeSizeSterm=1300000
colnames(snps_long_final)
table(snps_long_final$species)
# sample_relative_wide %>% filter(core=="core") %>% dim()
snps_long_final_cleaned <- snps_long_final %>% filter(Snps>0.05) %>% filter(Snps>0.2) %>% filter(Snps<0.8) %>% group_by(species,sample) %>% dplyr::summarize(counts=n()) %>% mutate(nuclotideDiv=100*(counts/coregenomeSizeSterm)) %>% separate(sample, c("culture", "sample"),sep = "-" ) %>% filter(culture!="FAMstrains")

# snps_long_final_cleaned <- snps_long_final %>% filter(Snps>0.05) %>% filter(Snps>0.2) %>% filter(Snps<0.8) %>% group_by(species,sample) %>% dplyr::summarize(counts=n()) %>% mutate(nuclotideDiv=100*(counts/coregenomeSizeSterm)) %>% separate(sample, c("culture", "sample"),sep = "-" ) %>% filter(culture!="FAMstrains") 




nucleotide_Div_time <- ggplot(snps_long_final_cleaned,aes(x=sample,nuclotideDiv,color=species,fill=species,group=species))+geom_point()+geom_line()+facet_grid(~culture,scales = "free_x", space = "free")+theme_classic()+theme(legend.position = "none",axis.text.x =element_blank(),axis.title.x = element_blank(),axis.ticks.x=element_blank())+scale_y_continuous(trans = "log10",breaks=c(0.0001,0.001,0.01,0.1,0.4))+labs(y="Nucleotide diversity [%]")+scale_fill_manual(values=c("blue","pink"))+scale_color_manual(values=c("#fdc2b5ff","#007fa6ff"))

nucleotide_Div_time


pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/nucleotid_diversity.pdf", width = 10, height = 3)

nucleotide_Div_time


dev.off()


# +facet_grid(~culture,scales = "free_x", space = "free")



  final_strain_abundance_plot <- ggplot(sample_medians_normalised_final,aes(x=sample,y=normalised_ratio,fill=groupAssignment))+geom_bar(position="stack", stat="identity",alpha=0.5)+facet_grid(~culture,scales = "free_x", space = "free")+theme_classic()+scale_fill_manual(values = cols)+scale_color_manual(values = cols)+labs(x="",y="Relative abundance")+theme(legend.title = element_blank(),axis.text.x = element_text(angle = 75, hjust = 1,size=6))+ scale_y_continuous(expand = c(0,0))
   
   final_strain_abundance_plot


##-------------------------------------------------
#do it by date

sample_names <- c("FSK_101_75"="starter 1","Lyo_101_76_4_7"="starter 2","Lyo_101_77"="starter 3","Lyo_101_02_7"="starter 4","Lyo_101_10_2"="starter 5","Lyo_101_16"="starter 6","ws_101_19"="starter 7","FSK_105_75"="starter 1","Lyo_105_76_4_1"="starter 2","Lyo_105_76_4_2"="starter 3","FSK_105_76_4_6"="starter 4","Lyo_105_76_2"="starter 5","Lyo_105_76_1"="starter 6","Lyo_105_76_5_11"="starter 7","Lyo_105_76_5_13"="starter 8","Lyo_105_76_5_12"="starter 9","Lyo_105_76_3_31"="starter 10","Lyo_105_11_5"="starter 11","Ws_105_19"="starter 12","Lyo_115_09"="starter 1","Ws_115_19"="starter 2","FSK_124_75_4"="starter 1","Lyo_124_08"="starter 2", "Ws_124_19"="starter 3","Lyo_150_76_4"="starter 1","Lyo_150_76_11"="starter 2","Lyo_150_11_3_11"="starter 3","Lyo_150_14_10"="starter 4","Lyo_150_18"="starter 5","Ws_150_19"="starter 6","Lyo_190_95_3"="starter 1","Lyo_190_17"="starter 2","Ws_190_19"="starter 3","Lyo_202_96"="starter 1","Lyo_202_12"="starter 2","Lyo_202_14"="starter 3","S_202_14"="starter 4","Ws_202_18"="starter 5","S_202_18"="starter 6", "C_202_18_02"="starter 7","C_202_18_01"="starter 8","Lyo_203_77"="starter 1","Lyo_203_96_1"="starter 2","Lyo_203_16"="starter 3","Ws_203_19"="starter 4","Lyo_280_16"="starter 1","Lyo_280_16_4"="starter 2", "Ws_280_19"="starter 3","Lyo_302_13"="starter 1", "Ws_302_19"="starter 2","FSK_305_76"="starter 1", "Lyo_305_96_1"="starter 2","Lyo_305_17"="starter 3","Ws_305_19"="starter 4",
"Vers_101_20"="starter 8", "Vers_105_20"="starter 13", 
"Vers_115_20"="starter 3", "Vers_124_20"="starter 4", "Vers_150_20"="starter 7", "Vers_190_20"="starter 4", "Vers_203_20"="starter 5", 
"Vers_280_20"="starter 4", "Vers_302_20"="starter 3", "Vers_305_20"="starter 5",
"Therm_101_20"="starter 9", "Therm_105_20"="starter 14", "Therm_115_20"="starter 4", "Therm_124_20"="starter 5", 
"Therm_150_20"="starter 8", "Therm_190_20"="starter 5", "Therm_203_20"="starter 6", "Therm_280_20"="starter 5", 
"Therm_302_20"="starter 4", "Therm_305_20"="starter 6",
"24h_101_20"="starter 10", "24h_105_20"="starter 15", "24h_115_20"="starter 5", "24h_124_20"="starter 6", "24h_150_20"="starter 9", 
"24h_190_20"="starter 6", "24h_203_20"="starter 7", "24h_280_20"="starter 6", "24h_302_20"="starter 5", "24h_305_20"="starter 7",
"M17x_101_20"="starter 11", "M17X_105_20"="starter 16", "M17X_115_20"="starter 6", 
"M17X_124_20"="starter 7", "M17x_150_20"="starter 10", "M17x_190_20"="starter 7", "M17x_202_20"="starter 9", "M17x_203_20"="starter 8", 
"M17x_280_20"="starter 7", "M17x_302_20"="starter 6", "M17x_305_20"="starter 8") %>% as.data.frame() %>% rownames_to_column(.,var="sample_name") %>% as_tibble()


colnames(sample_names)[2] <- "starter_num"


sample_names$sample <- as.character(str_split_fixed(sample_names$sample_name, "_", 5)[,2])
sample_names$year <- as.character(str_split_fixed(sample_names$sample_name, "_", 5)[,3])

merge(snps_long_final_cleaned,sample_names,by.x = c("culture","sample"),by.y = c("sample","starter_num")) %>% as_tibble()


##------------------------------
##plot




```



```{r}
##===============================
#old
##===============================
library(robustbase)
library(VennDiagram)
##==================================================Streptococcus thermophilus===================================================================
colnames(sample_relative_wide)
colnames(sample_relative_wide) <- colnames(sample_relative_wide) %>% gsub("FAMstrains-FAM","",.)

dput(colnames(sample_relative_wide))

 # %>% filter(species=="S. thermophilus") 
sample_relative_wide_snpsUsed <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core")   %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 | "experiment_A">0.05| "experiment_B">0.05| "experiment_C">0.05 | "experiment_D">0.05 | "experiment_E">0.05)


colnames(sample_relative_wide_snpsUsed)
sample_relative_wide_snpsUsed_lin1 <- sample_relative_wide_snpsUsed %>% filter(`24853`>0.8| `24798`>0.8| `13493`>0.8| `13500`>0.8 | `24737`>0.8 | `24854`>0.8| `13491`>0.8 | `13492`>0.8 ) %>% add_column(explained="lin1")
sample_relative_wide_snpsUsed_lin2 <- sample_relative_wide_snpsUsed %>% filter(`S72`>0.8| `24855`>0.8| `13494`>0.8)%>% add_column(explained="lin2")
sample_relative_wide_snpsUsed_lin3 <- sample_relative_wide_snpsUsed %>% filter(`S50`>0.8| `24740`>0.8| `24738`>0.8| `13499`>0.8 )%>% add_column(explained="lin3")
sample_relative_wide_snpsUsed_lin4 <- sample_relative_wide_snpsUsed %>% filter(`24739`>0.8| `13497`>0.8| `13496`>0.8| `13495`>0.8 )%>% add_column(explained="lin4")
allsites <- c(sample_relative_wide_snpsUsed_lin1$site, sample_relative_wide_snpsUsed_lin2$site, sample_relative_wide_snpsUsed_lin3$site,sample_relative_wide_snpsUsed_lin4$site)

##----------------------
# Chart venn diagramm
##----------------------
temp <-venn.diagram(
  x = list(sample_relative_wide_snpsUsed_lin1$site, sample_relative_wide_snpsUsed_lin2$site, sample_relative_wide_snpsUsed_lin3$site,sample_relative_wide_snpsUsed_lin4$site),
  category.names = c("lin 1" , "lin 2 " , "lin 3", "lin 4"),
  filename = NULL
)

plot.new() 

grid.draw(temp)
# 
# pdf("testpdf", width = 14, height = 7)
# 
# grid.draw(temp)
# 
# dev.off()

grid.draw(temp)

allsites[duplicated(allsites)]
sum(duplicated(allsites))
sum(!duplicated(allsites))

sum(table(allsites)>1)
sum(table(allsites)==1)

duplictednames <- names(table(allsites)[(table(allsites)>1)])

##----------------------
# Chart venn diagramm
##----------------------

sample_relative_wide_snpsUsed_new_01 <- rbind(sample_relative_wide_snpsUsed_lin1,sample_relative_wide_snpsUsed_lin2,sample_relative_wide_snpsUsed_lin3,sample_relative_wide_snpsUsed_lin4)
sample_relative_wide_snpsUsed_new_02 <- sample_relative_wide_snpsUsed_new_01 %>% filter(!site %in%duplictednames) #lineage specific snps
table(sample_relative_wide_snpsUsed_new_02$explained)
sample_relative_wide_snpsUsed_new_temp <- sample_relative_wide_snpsUsed %>% filter(site %in%duplictednames)%>% add_column(explained="multiple") ##duplicated sites
sample_relative_wide_snpsUsed_new_03 <- rbind(sample_relative_wide_snpsUsed_new_02,sample_relative_wide_snpsUsed_new_temp) ##all explained sites
sample_relative_wide_snpsUsed_tmp <-  sample_relative_wide_snpsUsed %>% filter(!site %in%sample_relative_wide_snpsUsed_new_03$site) %>% add_column(explained="not explained")
sample_relative_wide_snpsUsed_final <- rbind(sample_relative_wide_snpsUsed_new_03,sample_relative_wide_snpsUsed_tmp) ##all explained sites


##EXKURSION LINEAGE SPECIFIC DUPLICATIONS
TMP <- sample_relative_wide_snpsUsed %>% filter(site %in%duplictednames)
TMP$e

short_explained <- sample_relative_wide_snpsUsed_new_01 %>%filter(site %in%duplictednames) %>%  select(c("site","species","explained")) %>% mutate(abundance = explained) %>% spread(., explained, abundance)
short_explained$explained <- paste(short_explained$lin1,short_explained$lin2,short_explained$lin3,short_explained$lin4,sep="_")
short_explained_multiple <- short_explained%>% select("site","explained")
sample_relative_wide_snpsUsed_new_temp <- sample_relative_wide_snpsUsed %>% filter(site %in%duplictednames) ##duplicated sites
sample_relative_wide_snpsUsed_new_temp_02 <- merge(sample_relative_wide_snpsUsed_new_temp,short_explained_multiple,by="site")
sample_relative_wide_snpsUsed_new_03 <- rbind(sample_relative_wide_snpsUsed_new_02,sample_relative_wide_snpsUsed_new_temp_02) ##all explained sites
sample_relative_wide_snpsUsed_tmp <-  sample_relative_wide_snpsUsed %>% filter(!site %in%sample_relative_wide_snpsUsed_new_03$site) %>% add_column(explained="not explained")
sample_relative_wide_snpsUsed_final <- rbind(sample_relative_wide_snpsUsed_new_03,sample_relative_wide_snpsUsed_tmp) ##all explained sites


totalSNPS <- nrow(sample_relative_wide_snpsUsed_final)
100*(table(sample_relative_wide_snpsUsed_final$explained)/totalSNPS)
table(sample_relative_wide_snpsUsed_final$explained)
##----------------------
# boxplot
##----------------------
# sample_relative_wide_snpsUsed_final_long <- sample_relative_wide_snpsUsed_final %>% select( "site","Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E","explained")%>% gather(.,sample,Median,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E") 

sample_relative_wide_snpsUsed_final <- sample_relative_wide_snpsUsed_final %>%  dplyr::rename("Reference 2"="cheesemaking\nday2","Reference 1"="cheesemaking\nday1" )

sample_relative_wide_snpsUsed_final_long <- sample_relative_wide_snpsUsed_final %>% select( "site","Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E","Reference 1","Reference 2","explained")%>% gather(.,sample,Median,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E","Reference 1","Reference 2") 

# ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median))+geom_boxplot()+theme_classic()+facet_wrap(~explained)

#remove zeroes
sample_relative_wide_snpsUsed_final_long_woZeros <- sample_relative_wide_snpsUsed_final_long %>% filter(Median>0.1)
ggplot(sample_relative_wide_snpsUsed_final_long_woZeros,aes(x=sample,y=Median))+geom_boxplot()+theme_classic()+facet_wrap(~explained)


##----------------------
# lineplot
##----------------------
sample_relative_wide_snpsUsed_final_long_woZeros$sample = factor(sample_relative_wide_snpsUsed_final_long_woZeros$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E","Reference 1","Reference 2"))

sample_relative_wide_snpsUsed_final_long_woZeros$explained = factor(sample_relative_wide_snpsUsed_final_long_woZeros$explained, levels=c("lin1","lin2","lin3","lin4","not explained","lin1_lin2_NA_NA","NA_NA_lin3_lin4","lin1_lin2_lin3_lin4","lin1_lin2_NA_lin4","lin1_NA_lin3_lin4","lin1_NA_lin3_NA","lin1_NA_NA_lin4","NA_lin2_lin3_NA","NA_lin2_NA_lin4","NA_lin2_lin3_lin4"))


# ggplot(sample_relative_wide_snpsUsed_final_long_woZeros,aes(x=sample,y=Median,group=site))+geom_line(color="red",size=0.2, alpha=.1)+theme_classic()
sample_relative_wide_snpsUsed_final_long_woZeros <- sample_relative_wide_snpsUsed_final_long_woZeros %>% filter(explained!="not explained")

pExplainted <- ggplot(sample_relative_wide_snpsUsed_final_long_woZeros,aes(x=sample,y=Median,group=site,color=explained,fill=explained))+geom_line(alpha=.5,size=0.5)+theme_classic()+facet_wrap(~explained)+theme(legend.position = "none",axis.text.x = element_text(angle = 75, hjust = 1,size=9))

table(sample_relative_wide_snpsUsed_final_long_woZeros$explained)
pExplainted


  svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_explained_by_isolates.svg",width=7,height=4.5)
# png("~/Desktop/Projects/2019_RMK202_analysis/plot/supp_altNucFre_lineages.png", width = 3400, height = 2600,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted
dev.off()
###--------------------
##subset for ppx
table(sample_relative_wide_snpsUsed_final_long_woZeros$explained)
# sample_relative_wide_snpsUsed_final_long_tmp <-  sample_relative_wide_snpsUsed_final_long_woZeros %>% filter(explained %in% c("NA_NA_lin3_lin4","NA_lin2_lin3_lin4","NA_lin2_NA_lin4","lin3"))

sample_relative_wide_snpsUsed_final_long_tmp <-  sample_relative_wide_snpsUsed_final_long_woZeros %>% filter(explained %in% c("lin1_NA_NA_lin4","NA_lin2_lin3_NA","NA_lin2_NA_lin4","lin1_NA_lin3_NA"))


  sample_relative_wide_snpsUsed_final_long_tmp$explained = factor(sample_relative_wide_snpsUsed_final_long_tmp$explained, levels=c("lin1_NA_NA_lin4","NA_lin2_lin3_NA","NA_lin2_NA_lin4","lin1_NA_lin3_NA"))

# sample_relative_wide_snpsUsed_final_long$explained_02 <- revalue(sample_relative_wide_snpsUsed_final_long$explained, c("lin1"="explained by isolates", "lin2"="explained by isolates","lin3"="explained by isolates","lin4"="explained by isolates","multiple"="explained by isolates","not explained"="not explained by isolates"))

pExplainted <- ggplot(sample_relative_wide_snpsUsed_final_long_tmp,aes(x=sample,y=Median,group=site,color=explained,fill=explained))+geom_line(alpha=.5,size=0.5)+theme_classic()+facet_wrap(~explained,ncol=4)+theme(legend.position = "none",axis.text.x = element_text(angle = 75, hjust = 1,size=6))

pExplainted

png("~/Desktop/Projects/2019_RMK202_analysis/plot/supp_altNucFre_lineages_03.png", width = 3400, height = 1000,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted
dev.off()

###--------------------

sample_relative_wide_snpsUsed_final_long$explained_02 <- revalue(sample_relative_wide_snpsUsed_final_long$explained, c("lin1"="explained by isolates", "lin2"="explained by isolates","lin3"="explained by isolates","lin4"="explained by isolates","multiple"="explained by isolates","not explained"="not explained by isolates"))

pExplainted <- ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median,group=site,color=explained_02,fill=explained_02))+geom_line(size=0.5,alpha=0.5)+theme_classic()+facet_wrap(~explained_02)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+
    labs("",
         x="",
         y="Alternative allele frequency")
pExplainted

pExplainted <- ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median,group=site,color=explained_02,fill=explained_02))+geom_line(size=0.5,alpha=0.5)+theme_classic()+facet_wrap(~explained_02)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+
    labs("",
         x="",
         y="Alternative allele frequency")
pExplainted


  svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_explained_by_isolates..svg",width=7,height=4.5)
#    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted

dev.off()



##----------------------
# not explained
##----------------------

sample_relative_wide_snpsUsed_final_long$explained_03 <- ifelse(sample_relative_wide_snpsUsed_final_long$explained_02=="not explained by isolates","not explained by isolates","explained")
  
  sample_relative_wide_snpsUsed_final_long$sample = factor(sample_relative_wide_snpsUsed_final_long$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E"))

  
pExplainted_rough <- ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median,group=site,color=explained_03,fill=explained_03))+geom_line(size=0.5,alpha=0.5)+theme_classic()+facet_wrap(~explained_03)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+
    labs("",
         x="",
         y="Alternative allele frequency")
pExplainted_rough

  svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_notExplained.svg",width=7,height=4.5)
#    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted_rough

dev.off()

##----------------------
# where do these funky linkes (~recombinatnts) locate on the genome
##----------------------

sample_relative_wide_snpsUsed_final_long_recombinants <- sample_relative_wide_snpsUsed_final_long %>% filter(explained %in% c("lin1_lin2_NA_lin4","lin1_NA_lin3_lin4","lin1_NA_lin3_NA","lin1_NA_NA_lin4","NA_lin2_lin3_NA","NA_lin2_NA_lin4","NA_lin2_lin3_lin4")) %>% select("site","explained") %>% unique() 

sample_relative_wide_snpsUsed_final_long_recombinants <- sample_relative_wide_snpsUsed_final_long %>% select("site","explained") %>% unique() 


sample_relative_wide_snpsUsed_final_long_recombinants$location <- as.numeric(str_split_fixed(sample_relative_wide_snpsUsed_final_long_recombinants$site, "_", 4)[,2])
sample_relative_wide_snpsUsed_final_long_recombinants$explained = factor(sample_relative_wide_snpsUsed_final_long_recombinants$explained, levels=c("lin1","lin2","lin3","lin4","not explained","lin1_lin2_NA_NA","NA_NA_lin3_lin4","lin1_lin2_lin3_lin4","NA_lin2_lin3_lin4","lin1_lin2_NA_lin4","lin1_NA_lin3_lin4","lin1_NA_lin3_NA","lin1_NA_NA_lin4","NA_lin2_lin3_NA","NA_lin2_NA_lin4"))


sample_relative_wide_snpsUsed_final_long_recombinants$explained_2 <- revalue(sample_relative_wide_snpsUsed_final_long_recombinants$explained, c("lin1_lin2_NA_NA"="lin1_lin2_NA_NA=~evolved_after_split","NA_NA_lin3_lin4"="NA_NA_lin3_lin4=~evolved_after_split","lin1_lin2_lin3_lin4"="lin1_lin2_lin3_lin4=polishingERRORs","NA_lin2_lin3_lin4"="NA_lin2_lin3_lin4=lost_in_lin1","lin1_NA_lin3_lin4"="lin1_NA_lin3_lin4=lost_in_lin2","lin1_lin2_NA_lin4"="lin1_lin2_NA_lin4=lost_in_lin3","lin1_NA_lin3_NA"="lin1_NA_lin3_NA=recombination","lin1_NA_NA_lin4"="lin1_NA_NA_lin4=recombination","NA_lin2_lin3_NA"="NA_lin2_lin3_NA=recombination","NA_lin2_NA_lin4"="NA_lin2_NA_lin4=recombination"))

# ggplot(sample_relative_wide_snpsUsed_final_long_recombinants,aes(x=location,color=explained,fill=explained))+geom_density()+facet_wrap(~explained,scales = "free_y")+theme_classic()
plocation <- ggplot(sample_relative_wide_snpsUsed_final_long_recombinants,aes(x=location,color=explained,fill=explained))+geom_histogram()+facet_wrap(~explained_2,scales = "free_y")+theme_classic()+labs(title="location of SNVs coming from different ",x="genomic location")+theme(legend.position = "none")

plocation


  # svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_explained_by_isolates..svg",width=7,height=4.5)
png("~/Desktop/supp_altNucFre_lineages_location.png", width = 3400, height = 2600,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
plocation
dev.off()

##----------------------
# not explained
##----------------------
table(sample_relative_wide_snpsUsed_final_long$explained)
sample_relative_wide_snpsUsed_final_long_notExplained <- sample_relative_wide_snpsUsed_final_long %>% filter(explained=="not explained")

ggplot(sample_relative_wide_snpsUsed_final_long_notExplained,aes(x=Median))+geom_density()+facet_wrap(~sample)+theme_classic()


###nomany multiallelic sites
sitesss <- str_split_fixed(sample_relative_wide_snpsUsed_final_long_notExplained$site, "_", 4)[,2]
length(sitesss)/6
unique(sitesss) %>% length()

sitesss <- str_split_fixed(sample_relative_wide_snpsUsed_final_long$site, "_", 4)[,2]
length(sitesss) /6
unique(sitesss) %>% length()

##----------------------
# meansss and medianss
##----------------------
library(patchwork)
# mediannssss <-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long[,c("explained","sample","Median")], median, na.rm=TRUE)%>% filter(explained!="multiple")
# meanssss <-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long[,c("explained","sample","Median")], median, na.rm=TRUE) %>% filter(explained!="multiple")
# 
# mediansplot <- ggplot(mediannssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat="identity")+theme_classic()+theme(legend.position = "none")
# meansssplot <- ggplot(meanssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat="identity")+theme_classic()
# 
# mediansplot+meansssplot

##---without zeros
# table(sample_relative_wide_snpsUsed_final_long_woZeros$sample)
mediannssss <-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long_woZeros[,c("explained","sample","Median")], median, na.rm=TRUE)%>% filter(explained!="multiple")
meanssss <-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long_woZeros[,c("explained","sample","Median")], median, na.rm=TRUE) %>% filter(explained!="multiple")

mediansplot <- ggplot(mediannssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat="identity")+theme_classic()+theme(legend.position = "none")
meansssplot <- ggplot(meanssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat="identity")+theme_classic()

mediansplot+meansssplot

##---------------------------------
##lineage abundance
##---------------------------------
ratio_larger <- mediannssss %>% filter(explained=="NA_NA_lin3_lin4") %>% select("sample","Median")  %>% dplyr::rename(Overall_ratio = Median) 
ratio_larger_2 <- mediannssss %>% filter(explained=="lin2") %>% select("sample","Median")  %>% dplyr::rename(rel_lin2 = Median) 
ratio_larger_3 <- mediannssss %>% filter(explained=="lin3") %>% select("sample","Median")  %>% dplyr::rename(rel_lin3 = Median) 
# ratio_larger_4 <- mediannssss %>% filter(explained=="lin4") %>% select("sample","Median")  %>% dplyr::rename(rel_lin4 = Median) 

ratio_together_1 <- merge(ratio_larger,ratio_larger_2,by="sample",all.x = TRUE)
ratio_together_2 <- merge(ratio_together_1,ratio_larger_3,by="sample",all.x = TRUE)
ratio_together_2$lin1 <- (1-ratio_together_2$Overall_ratio)*(1-ratio_together_2$rel_lin2)
ratio_together_2$lin2 <- (1-ratio_together_2$Overall_ratio)*(ratio_together_2$rel_lin2)
ratio_together_2$lin3 <- (ratio_together_2$Overall_ratio)*(ratio_together_2$rel_lin3)
ratio_together_2$lin4 <- (ratio_together_2$Overall_ratio)*(1-ratio_together_2$rel_lin3)


ratio_larger_4 <- mediannssss %>% filter(explained=="lin3") %>% select("sample","Median")  %>% dplyr::rename(rel_lin3 = Median) 


##ratios in Reference sequences
#ther is no lin3 or lin4 specific hits that is why we cannot calculate it like before. 
#lin1 is the reference everything that is not lin1 is lin2. 

REffs2 <- mediannssss %>% filter(sample=="Reference 1") %>% filter(explained=="lin2") %>% select(Median) %>% unlist() %>% as.numeric()
ratio_together_final_referenceSamples <- data.frame(sample="Reference 1",lin4=0,lin3=0,lin2=REffs2,lin1=1-REffs2)
REffs2 <- mediannssss %>% filter(sample=="Reference 2") %>% filter(explained=="lin2") %>% select(Median) %>% unlist() %>% as.numeric()
ratio_together_final_referenceSamples2 <- data.frame(sample="Reference 2",lin4=0,lin3=0,lin2=REffs2,lin1=1-REffs2)

# ratio_together_final_referenceSamples <- 

# ratio_together_2$Unknown <- 1-(ratio_together_2$lin1+ratio_together_2$lin2+ratio_together_2$lin3+ratio_together_2$lin4)

# ratio_together_final <- ratio_together_2 %>% select(sample,lin1,lin2,lin3,lin4) %>% gather(.,lineage,"Relative abundance","lin1","lin2","lin3","lin4")
ratio_together_final_tmp1 <- ratio_together_2 %>% select(sample,lin4,lin3,lin2,lin1) 

ratio_together_final <- rbind(ratio_together_final_tmp1,ratio_together_final_referenceSamples,ratio_together_final_referenceSamples2) %>% gather(.,lineage,"Relative abundance","lin4","lin3","lin2","lin1")

table(ratio_together_final$sample)



ratio_together_final <- ratio_together_final %>% filter(!sample %in% c("Reference 1","Reference 2"))
ratio_together_final$sample = factor(ratio_together_final$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E","Reference 1","Reference 2"))

ratio_together_final$lineage = factor(ratio_together_final$lineage, levels=(c("lin4","lin3","lin2","lin1")))


# colorsss <- c("#0000FF","#6699FF","#99CCFF","#00FFFF")
colorsss <- c("#99CCFF","#00FFFF","#0000FF","#6699FF")
colorsss <- c("#99CCFF","#00d5d5ff","#0000FF","#6699FF")


plot_strain_abundance <- ggplot(ratio_together_final,aes(x=sample,y=`Relative abundance`,color=lineage,fill=lineage))+geom_bar(stat="identity")+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+
    labs(x="")+scale_color_manual(values=colorsss)+scale_fill_manual(values=colorsss)+theme(legend.position = "none")
plot_strain_abundance
# png("~/Desktop/supp_strain_Abundance.png", width = 2800, height = 2200,res=300)
svg("~/Desktop/Projects/2019_RMK202_analysis/plot/strain_rel_abundance.svg",width=6,height=3.5)

# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
plot_strain_abundance
dev.off()


##area plot

plot_strain_abreasss <- ggplot(ratio_together_final,aes(x=sample,y=`Relative abundance`,group=lineage,color=lineage,fill=lineage))+geom_area(alpha=0.5)+geom_point(position = "stack",size=2)+geom_line(position = "stack",size=1.25)+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+
    labs(x="")+scale_color_manual(values=colorsss)+scale_fill_manual(values=colorsss)+theme(legend.position = "none")+scale_x_discrete( expand = c(0.01,0.01)) +scale_y_discrete( expand = c(0.01,0.01)) 

plot_strain_abreasss


svg("~/Desktop/Projects/2019_RMK202_analysis/plot/strain_rel_abundance_02.svg",width=2.5,height=3.5)

# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
plot_strain_abreasss
dev.off()
```

##number of sample specific SNPs

Here, I want to see for every starter culture if we have sample specific SNPs. 

In order for this chunk to work, you have to first run the Sterm strains 

```{r}

table(sample_relative_safe_Sterm_final$species)
range(sample_relative_safe_Sterm_final$Snps)
table(sample_relative_safe_Sterm_final$site) %>% sort()


cutoffSNPS <- 0.05


StarterUniqueSNPs <- data.frame()
for (Startersss in unique(sample_relative_safe_Sterm_final$culture)) {
  
  tmp1 <- sample_relative_safe_Sterm_final %>% filter(culture==Startersss) %>% mutate(SNPspresent=ifelse(Snps>cutoffSNPS,1,0))
  
  tmp2 <- tmp1 %>% ungroup() %>% group_by(site,species) %>% summarise(numberOFsample=sum(SNPspresent))
  
  
  nsamples <- length(unique(tmp1$sample))
  
  
  tmp2 %>% group_by(numberOFsample,species) %>% summarise(counts=n()) %>% as.data.frame() %>% as_tibble() %>% filter(numberOFsample %in% c(1,nsamples-1))%>% group_by(species) %>% summarise(totalNumberOfsingleSNPs=sum(counts)) 
  
  
  namesSites_01 <- tmp2 %>% filter(numberOFsample %in% c(1)) %>% pull(site)
  namesSites_allbut1 <- tmp2 %>% filter(numberOFsample %in% c(nsamples-1)) %>% pull(site)

  tmp3 <- tmp1 %>% dplyr::filter(site %in% namesSites_01) %>% filter(SNPspresent==1)%>% mutate(group="StarterUnique")
  tmp4 <- tmp1 %>% dplyr::filter(site %in% namesSites_allbut1) %>% filter(SNPspresent==0) %>% mutate(group="StarterUnique")
  
  tmp6 <- data.frame()
  for (samplesss in unique(tmp1$sample)) {
    tmp5 <- tmp1 %>% filter(sample==samplesss) %>% head(1)%>% mutate(group="dummy")%>% mutate(site="dummy") %>% mutate(site="dummy") %>% filter(SNPspresent==1)
    tmp6 <- rbind(tmp6,tmp5)
  }
  
  
  
 StarterUniqueSNPs <-  rbind(StarterUniqueSNPs,tmp3,tmp4,tmp6)
  
}


###------------------------plot

coregenomesize=130000


StarterUniqueSNPs_summary <- StarterUniqueSNPs %>% group_by(species,sample,culture) %>% summarise(UNIQUEsnps=n()) %>% mutate(relativeSNPs=UNIQUEsnps/coregenomesize) %>% mutate(UNIQUEsnps=ifelse(UNIQUEsnps==1,1,UNIQUEsnps-1))


# 20/coregenomesize
tmp1 <- StarterUniqueSNPs %>% filter(sample=="305-starter 1")

# ggplot()+geom_density(data=tmp1,aes(x=Snps))


# plotbox <- ggplot()+geom_boxplot(data=StarterUniqueSNPs_summary,aes(x=culture,group=culture,color=species,fill=species,y=UNIQUEsnps))+scale_y_continuous(trans = "log2",breaks=c(1,10, 100,1000,10000),name="Unique SNPs", sec.axis = sec_axis(~./coregenomesize, name="Relative number of sites [%]",breaks=c(1e-5,1e-4,1e-3,1e-2)))+facet_grid(~species)




 ggplot(StarterUniqueSNPs_summary,aes(x=sample,color=species,fill=species,group=species))+geom_point(aes(y=UNIQUEsnps))+geom_line(aes(y=UNIQUEsnps))+facet_grid(~culture,scales = "free_x", space = "free")+theme_classic()+theme(legend.position = "none",axis.text.x =element_blank(),axis.title.x = element_blank(),axis.ticks.x=element_blank())+
   scale_fill_manual(values=c("blue","pink"))+scale_color_manual(values=c("#fdc2b5ff","#007fa6ff")) +scale_y_continuous(trans = "log2",breaks=c(1,10, 100,1000,10000),name="Unique SNPs", sec.axis = sec_axis(~./coregenomesize, name="Relative number of sites [%]",breaks=c(1e-5,1e-4,1e-3,1e-2)))

 
 
 ###-----------------------with quartile info
 
 tmps <- StarterUniqueSNPs_summary %>% group_by(culture,species) %>% summarise(meanSNPs=mean(UNIQUEsnps),sdSNPs=sd(UNIQUEsnps),medianSNPs=median(UNIQUEsnps),first=quantile(UNIQUEsnps, 0.25),third=quantile(UNIQUEsnps, 0.75))

 
 StarterUniqueSNPs_range <- data.frame()
for (Startersss in unique(sample_relative_safe_Sterm_final$culture)) {
  
  tmp1 <- sample_relative_safe_Sterm_final %>% filter(culture==Startersss) %>% pull(sample) %>% as.character() %>% unique() 
  
  tmp2 <- tmp1 %>% head(1)
  tmp3 <- tmp1 %>% tail(1)
  
  tmp5 <- data.frame()
  for (speciesss in rev(unique(sample_relative_safe_Sterm_final$species))) {
    
    tmp4 <- tmps %>% filter(culture==Startersss) %>% filter(species==speciesss) %>% mutate(sampleFirst=tmp2)%>% mutate(samplelast=tmp3)
    
    tmp5 <- rbind(tmp5,tmp4)
    
  }

  StarterUniqueSNPs_range <- rbind(StarterUniqueSNPs_range, tmp5)
  tmp5
  
}
 
 
 
 
 # ggplot()+geom_point(data=StarterUniqueSNPs_summary,aes(x=sample,color=species,fill=species,group=species,y=UNIQUEsnps))+geom_line(data=StarterUniqueSNPs_summary,aes(x=sample,color=species,fill=species,group=species,y=UNIQUEsnps))+
 #   + geom_rect(data=StarterUniqueSNPs_range, (aes(xmin=sampleFirst, xmax=samplelast, 
 #                   ymin=first,ymax=third,color=species,fill=species,group=species)))+
 #   facet_grid(~culture,scales = "free_x", space = "free")+
 #   theme_classic()+theme(legend.position = "none",axis.text.x =element_blank(),axis.title.x = element_blank(),axis.ticks.x=element_blank())+
 #   scale_fill_manual(values=c("blue","pink"))+scale_color_manual(values=c("#fdc2b5ff","#007fa6ff")) +scale_y_continuous(trans = "log2",breaks=c(1,10, 100,1000,10000),name="Unique SNPs", sec.axis = sec_axis(~./coregenomesize, name="Relative number of sites [%]",breaks=c(1e-5,1e-4,1e-3,1e-2)))
library(ggnewscale)
 
 values=c("#fdc2b5ff","#007fa6ff")
alpha(c("#fdc2b5ff","#007fa6ff"), alpha = 0.2)

 levels(StarterUniqueSNPs_range$species) <- c(levels(StarterUniqueSNPs_range$species), rev(c("S. thermophilus", "L. delbrueckii")))

 lighterversionOFblue <- c("#79dfff","#26ccff","#5ad8ff")
 # "#007fa6ff","#fdc2b5ff"

NumberOFsnpsPlot <-  ggplot()+geom_point(data=StarterUniqueSNPs_summary,aes(x=sample,color=species,group=species,y=UNIQUEsnps))+geom_line(data=StarterUniqueSNPs_summary,aes(x=sample,color=species,group=species,y=UNIQUEsnps))+
   scale_color_manual(values=c("#fdc2b5ff","#007fa6ff"))+
   # new_scale_fill()+
   geom_rect(aes(xmin = sampleFirst, xmax = samplelast, ymin = first, ymax = third, fill = species),
            alpha=0.3,
            data = StarterUniqueSNPs_range) +
   scale_fill_manual(values=c("#fdc2b5ff","#5ad8ff"))+
   facet_grid(~culture,scales = "free_x", space = "free")+
   theme_classic()+theme(legend.position = "none",axis.text.x =element_blank(),axis.title.x = element_blank(),axis.ticks.x=element_blank())+ scale_y_continuous(trans = "log2",breaks=c(1,10, 100,1000,10000),name="Sample-specific SNPs", sec.axis = sec_axis(~./coregenomesize, name="Relative sample-specific SNPs [%]",breaks=c(1e-5,1e-4,1e-3,1e-2)))

NumberOFsnpsPlot




pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Unique_SNPs_01.pdf", width = 10, height = 3)

NumberOFsnpsPlot


dev.off()

StarterUniqueSNPs_range %>% group_by(species) %>% summarise(median=median(medianSNPs),sd=sd(medianSNPs)) %>% mutate(relativemedian=coregenomesize/median)%>% mutate(relativesd=coregenomesize/sd)
median(StarterUniqueSNPs_range$medianSNPs)

```




##not explained 2

```{r}

FAMnames <- colnames(sample_relative_wide)[colnames(sample_relative_wide) %>% grepl("^FAM",.)]

FAMstrainsexist <- sample_relative_wide %>% select(FAMnames) %>% rowSums()#%>% mutate(sumss=rowSums())

starternames <- colnames(sample_relative_wide)[colnames(sample_relative_wide) %>% grepl("-starter",.)]
startersexist <- sample_relative_wide %>% select(starternames) %>% rowSums()#%>% mutate(sumss=rowSums())


tmp1 <- sample_relative_wide %>% select("site" ,  "species"  , "effect" , "significance","geneName" , "COG_Functional_Category", "length"   ,"core" )


sample_relative_wide_explained <- cbind(tmp1,startersexist,FAMstrainsexist) %>% mutate(category=ifelse(startersexist>0&FAMstrainsexist>0,"both",ifelse(startersexist>0&FAMstrainsexist==0,"only metaG","onlyFAM"))) %>% as_tibble()

table(sample_relative_wide_explained$category)

sample_relative_wide_explained %>% group_by(species,core,category) %>% summarise(counts=n())


sample_relative_wide_explained_small <- sample_relative_wide_explained %>% select(site,category)


###------------------------plot SNPs nonexplained 


sample_relative_safe_Sterm_final_wide_filtered_long_woZero_completed <- left_join(sample_relative_safe_Sterm_final_wide_filtered_long_woZero,sample_relative_wide_explained_small)



###------------------------plot SNPs nonexplained 

sample_relative_safe_Sterm_final_wide_filtered_long_woZero_onlyMETAG <- sample_relative_safe_Sterm_final_wide_filtered_long_woZero_completed %>% filter(category=="only metaG")


  ponlyMetaG <- ggplot(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_onlyMETAG,aes(x=sample,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.3, alpha=.1)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = "free_x", space = "free")+
            facet_grid(species~culture,scales = "free_x", space = "free")+
            # ggforce::facet_row(vars(culture)~species,scales = "free_x", space = "free")+
            # ggforce::facet_col(vars(species),scales = "free_x", space = "free")+
     labs(title = "not Explained",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
          panel.spacing = unit(1.2, "lines")
  )
  
  ponlyMetaG
  
  

###------------------------plot SNPs nonexplained 

  sample_relative_safe_Sterm_final_wide_filtered_long_woZero_both <- sample_relative_safe_Sterm_final_wide_filtered_long_woZero_completed %>% filter(category=="both")

  ponlyBTOH <- ggplot(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_both,aes(x=sample,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.3, alpha=.1)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = "free_x", space = "free")+
            facet_grid(species~culture,scales = "free_x", space = "free")+
            # ggforce::facet_row(vars(culture)~species,scales = "free_x", space = "free")+
            # ggforce::facet_col(vars(species),scales = "free_x", space = "free")+
     labs(title = "Explained by isolates",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
          panel.spacing = unit(1.2, "lines")
  )
  
  ponlyBTOH
  
 library(patchwork)
 ponlyMetaG+ponlyBTOH+plot_layout(ncol=1)
   # png("~/Desktop/mid_thesis/report/figures/20200430/chapter3/explainedSNVs.png", width = 6000, height = 3500,res=300)
# svg("~/Desktop/mid_thesis/report/figures/20200430/chapter2/snpscalled_sterm_wide_01.svg", width = 12, height = 5)
   png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images//20210126/supplement//SNPs_plot_suppl.png", width = 4000, height = 3500,res=400)

 ponlyMetaG+ponlyBTOH+plot_layout(ncol=1)
dev.off()
  
##----------------QUNATIFY
  
tmp1 <- sample_relative_safe_Sterm_final_wide_filtered_long_woZero_completed %>% group_by(species) %>% summarise(total=n())

 sample_relative_safe_Sterm_final_wide_filtered_long_woZero_completed %>% group_by(species,category) %>% summarise(counts=n()) %>% left_join(.,tmp1) %>% mutate(percent=100*(counts/total))
  
 
 sample_relative_safe_Sterm_final_wide_filtered_long_woZero_completed %>% 

```



##not explained 

Here, we test which SNVs are not explained with the isolated strains.
```{r}



colnames(sample_relative_wide)
sample_relative_wide_explained <- sample_relative_wide
sample_relative_wide_explained$metagenomSNP_onlyNewest_sample <- rowSums(sample_relative_wide[,c(15,27,29,32,38,41,47,51,54,56,60)])
sample_relative_wide_explained$metagenomSNP <- rowSums(sample_relative_wide[,9:60])
sample_relative_wide_explained$StrainSNP <- rowSums(sample_relative_wide[,61:131])
sample_relative_wide_explained$species.x <- sample_relative_wide_explained$species
# sample_relative_wide_explained_extended <- sample_relative_wide_explained %>% select(site,species.x,metagenomSNP,StrainSNP,metagenomSNP_onlyNewest_sample,effect,significance,geneName,COG_Functional_Category,Repeat_cluster,core)
sample_relative_wide_explained_extended <- sample_relative_wide_explained %>% select(site,species.x,metagenomSNP,StrainSNP,metagenomSNP_onlyNewest_sample,effect,significance,geneName,COG_Functional_Category,core)







##----------------------
##snps explained in current working stock
##----------------------

notExplaine <- sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP<0.8) %>% filter(metagenomSNP_onlyNewest_sample>0.1) %>% nrow()
explaine <- sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP>=0.8) %>% filter(metagenomSNP_onlyNewest_sample>0.1)%>% nrow()

100*(notExplaine/explaine)

notExplained <- sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP<0.8) %>% filter(metagenomSNP_onlyNewest_sample>0.1) 


ggplot(notExplained,aes(metagenomSNP))+geom_density()+facet_wrap(~species.x,scales = "free")



sample_relative_wide_explained$explained <- ifelse(sample_relative_wide_explained_extended$StrainSNP<0.8,"not explained","explained")
colnames(sample_relative_wide_explained)
dput(colnames(sample_relative_wide_explained))

sample_relative_wide_explained_long_onlyNewest <- sample_relative_wide_explained%>% filter(core=="core") %>% filter(metagenomSNP_onlyNewest_sample>0.1)%>% select("site", "species.x","explained", "101-starter 1", 
"101-starter 10", "101-starter 11", "101-starter 2", "101-starter 3", 
"101-starter 4", "101-starter 5", "101-starter 6", "101-starter 7", 
"101-starter 8", "101-starter 9", "105-starter 1", "105-starter 10", 
"105-starter 11", "105-starter 12", "105-starter 13", "105-starter 14", 
"105-starter 15", "105-starter 16", "105-starter 2", "105-starter 3", 
"105-starter 4", "105-starter 5", "105-starter 6", "105-starter 7", 
"105-starter 8", "105-starter 9", "115-starter 1", "115-starter 2", 
"115-starter 3", "115-starter 4", "115-starter 5", "115-starter 6", 
"124-starter 1", "124-starter 2", "124-starter 3", "124-starter 4", 
"124-starter 5", "124-starter 6", "124-starter 7", "150-starter 1", 
"150-starter 10", "150-starter 2", "150-starter 3", "150-starter 4", 
"150-starter 5", "150-starter 6", "150-starter 7", "150-starter 8", 
"150-starter 9", "190-starter 1", "190-starter 2", "190-starter 3", 
"190-starter 4", "190-starter 5", "190-starter 6", "190-starter 7", 
"202-starter 1", "202-starter 2", "202-starter 3", "202-starter 4", 
"202-starter 5", "202-starter 6", "202-starter 9", "203-starter 1", 
"203-starter 2", "203-starter 3", "203-starter 4", "203-starter 5", 
"203-starter 6", "203-starter 7", "203-starter 8", "280-starter 1", 
"280-starter 2", "280-starter 3", "280-starter 4", "280-starter 5", 
"280-starter 6", "280-starter 7", "302-starter 1", "302-starter 2", 
"302-starter 3", "302-starter 4", "302-starter 5", "302-starter 6", 
"305-starter 1", "305-starter 2", "305-starter 3", "305-starter 4", 
"305-starter 5", "305-starter 6", "305-starter 7", "305-starter 8") %>% gather(., sample, Snps, "101-starter 1":"305-starter 4", factor_key=TRUE,na.rm = TRUE) %>% filter(Snps>0.05)

ggplot(sample_relative_wide_explained_long_onlyNewest,aes(x=explained,y=Snps))+geom_boxplot()+theme_classic()

ggplot(sample_relative_wide_explained_long_onlyNewest,aes(x=Snps,group=explained,fill=explained))+geom_histogram(alpha=0.2, position="identity",bins = 100)+theme_classic()
table(sample_relative_wide_explained_long_onlyNewest$sample)

    t.test(Snps ~ explained, data = sample_relative_wide_explained_long_onlyNewest)



##----------------------
##snps explained from all samples
##----------------------

sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP<0.8) %>% filter(metagenomSNP>0.1) %>% nrow()
sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP>=0.8) %>% filter(metagenomSNP>0.1)%>% nrow()



notExplained <- sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP<0.8) %>% filter(metagenomSNP>1)

ggplot(notExplained,aes(metagenomSNP))+geom_density()+facet_wrap(~species.x,scales = "free")
table(notExplained$species.x)
sample_relative_wide_explained_extended %>% filter(core=="core") %>%  nrow()

sum(sample_relative_wide_explained_extended$StrainSNP<0.8)
sum(sample_relative_wide_explained_extended$StrainSNP>0.8)

sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP<0.8) %>% filter(metagenomSNP>0.1) %>% nrow()
sample_relative_wide_explained_extended %>% filter(core=="core") %>% filter(StrainSNP>=0.8) %>% filter(metagenomSNP>0.1)%>% nrow()

sample_relative_wide_explained$explained <- ifelse(sample_relative_wide_explained_extended$StrainSNP<0.8,"not explained","explained")
colnames(sample_relative_wide_explained)
dput(colnames(sample_relative_wide_explained))

sample_relative_wide_explained_long <- sample_relative_wide_explained%>% filter(core=="core") %>% select("site", "species","explained", "101-starter 1", 
"101-starter 10", "101-starter 11", "101-starter 2", "101-starter 3", 
"101-starter 4", "101-starter 5", "101-starter 6", "101-starter 7", 
"101-starter 8", "101-starter 9", "105-starter 1", "105-starter 10", 
"105-starter 11", "105-starter 12", "105-starter 13", "105-starter 14", 
"105-starter 15", "105-starter 16", "105-starter 2", "105-starter 3", 
"105-starter 4", "105-starter 5", "105-starter 6", "105-starter 7", 
"105-starter 8", "105-starter 9", "115-starter 1", "115-starter 2", 
"115-starter 3", "115-starter 4", "115-starter 5", "115-starter 6", 
"124-starter 1", "124-starter 2", "124-starter 3", "124-starter 4", 
"124-starter 5", "124-starter 6", "124-starter 7", "150-starter 1", 
"150-starter 10", "150-starter 2", "150-starter 3", "150-starter 4", 
"150-starter 5", "150-starter 6", "150-starter 7", "150-starter 8", 
"150-starter 9", "190-starter 1", "190-starter 2", "190-starter 3", 
"190-starter 4", "190-starter 5", "190-starter 6", "190-starter 7", 
"202-starter 1", "202-starter 2", "202-starter 3", "202-starter 4", 
"202-starter 5", "202-starter 6", "202-starter 9", "203-starter 1", 
"203-starter 2", "203-starter 3", "203-starter 4", "203-starter 5", 
"203-starter 6", "203-starter 7", "203-starter 8", "280-starter 1", 
"280-starter 2", "280-starter 3", "280-starter 4", "280-starter 5", 
"280-starter 6", "280-starter 7", "302-starter 1", "302-starter 2", 
"302-starter 3", "302-starter 4", "302-starter 5", "302-starter 6", 
"305-starter 1", "305-starter 2", "305-starter 3", "305-starter 4", 
"305-starter 5", "305-starter 6", "305-starter 7", "305-starter 8")  %>% gather(., sample, Snps, "101-starter 1":"305-starter 8", factor_key=TRUE,na.rm = TRUE) %>% filter(Snps>0.15)

ggplot(sample_relative_wide_explained_long,aes(x=explained,y=Snps))+geom_boxplot()+theme_classic()

table(sample_relative_wide_explained_long$explained)

ggplot(sample_relative_wide_explained_long,aes(x=Snps,group=explained,fill=explained))+geom_histogram(alpha=0.2, position="identity",bins = 100)+theme_classic()
table(sample_relative_wide_explained_long$sample)

##----------------------------------
###--------------------lineplot
##----------------------------------
#Sterm

  
#Ldel

sample_relative_wide_explained_long$culture <- str_split_fixed(sample_relative_wide_explained_long$sample, "_", 3)[,1]

sample_relative_wide_explained_long_ldel <- sample_relative_wide_explained_long %>% filter(species!="S. thermophilus")
sample_relative_wide_explained_long_ldel <- sample_relative_wide_explained_long %>% filter(species.x!="S. thermophilus")

all_colours <- rev(c("grey30","#10B552") )


 p3 <- ggplot(sample_relative_wide_explained_long_ldel,aes(x=sample,y=Snps,group=site,color=explained,fill=explained))+ geom_line(size=0.3, alpha=.1)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = "free_x", space = "free")+
            facet_grid(explained~culture,scales = "free_x", space = "free")+
            # ggforce::facet_row(vars(culture)~species,scales = "free_x", space = "free")+
            # ggforce::facet_col(vars(species),scales = "free_x", space = "free")+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
          panel.spacing = unit(1, "lines")
  )
  
  p3

  
###merge plots

  library(patchwork)  
p3 + labs(subtitle="A")+ p2+ labs(subtitle="B")+plot_layout(ncol=1)

 # png("~/Desktop/mid_thesis/report/figures/20200430/chapter3/explainedSNVs.png", width = 6000, height = 3500,res=300)
svg("~/Desktop/mid_thesis/report/figures/20200430/chapter2/snpscalled_sterm_wide_01.svg", width = 12, height = 5)

 p3 + labs(subtitle="A")+ p2+ labs(subtitle="B") +plot_layout(nrow =2)
dev.off()


svg("~/Der/ordner/wo/das/file/abgespeichert_werden_soll.svg", width = 12, height = 5) #mit de width und height mues biz spiele bises passt
 dinPlot
dev.off()


##----------------------------------
###--------------------of only the unexplained SNVs
##----------------------------------
sample_relative_wide_explained_long$culture <- str_split_fixed(sample_relative_wide_explained_long$sample, "-", 3)[,1]

sample_relative_wide_explained_long_notExplained <- sample_relative_wide_explained_long %>% filter(explained=="not explained")

sample_relative_wide_explained_long_notExplained$

all_colours <- rev(c("#EB4D4D","#10B552") )
sample_relative_wide_explained_long_notExplained$site <- as.factor(sample_relative_wide_explained_long_notExplained$site)


p4 <- ggplot(sample_relative_wide_explained_long_notExplained,aes(x=sample,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.3, alpha=.1)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = "free_x", space = "free")+
            facet_grid(~culture,scales = "free_x", space = "free")+
            # ggforce::facet_row(vars(culture)~species,scales = "free_x", space = "free")+
            # ggforce::facet_col(vars(species),scales = "free_x", space = "free")+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
          panel.spacing = unit(1.2, "lines")
  )
  
  p4
  
  # png("~/Desktop/mid_thesis/report/figures/20200430/chapter3/explainedSNVs_onlyUnexplained.png", width = 4500, height = 2000,res=300)
svg("~/Desktop/mid_thesis/report/figures/20200430/chapter2/snpscalled_sterm_wide_unexplained.svg", width = 12, height = 6)

p4
dev.off()

##-------------------------------------
#only Sterm

all_colours <- rev(c("#007fa6ff") )



 p5 <- ggplot(sample_relative_wide_explained_long_notExplained,aes(x=sample,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.2, alpha=.05)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = "free_x", space = "free")+
            # facet_grid(species~culture,scales = "free_x", space = "free")+
               facet_grid(~culture,scales = "free_x", space = "free")+

            # ggforce::facet_row(vars(culture)~species,scales = "free_x", space = "free")+
            # ggforce::facet_col(vars(species),scales = "free_x", space = "free")+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
   ylim(c(0,1))+
    # theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
   theme(axis.text.x = element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
          panel.spacing = unit(1, "lines")
  )
  
  p5
  
    pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/AbundanceStrain_SNPs_all_notExplained.pdf",width=12,height=8)
 p5
  dev.off()

```

## tsne snp

maybe I have to polish the snps first. I think without cleaning away super low abudnant and noisy snps it does not detect the underlining pattern

```{r}

library(forcats) 
library(Rtsne)
library(fpc)
library(ggrepel)
library( ggforce)
###--------------------------------------------------------------------
##colors
###--------------------------------------------------------------------

colors <- c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#80e5ff","#008181","#00dcdc")

###--------------------------------------------------------------------
##prep for tSNE
###--------------------------------------------------------------------


# table(snps_long_final$core)
# # snps_long_final_forTSNE <- snps_long_final 
# snps_long_final_forTSNE <- snps_long_final  %>% filter(core=="core")
# table(snps_long_final$core,snps_long_final$species)
# snps_long_final_forTSNE$sampleName <- paste0(snps_long_final_forTSNE$culture,"_",snps_long_final_forTSNE$sample)
# colnames(snps_long_final_forTSNE)
# dim(snps_long_final_forTSNE)
# snps_long_final_forTSNE <-  snps_long_final_forTSNE %>% select(species,site,sampleName,Snps)%>% filter(.,Snps>0.1) 
# dim(snps_long_final_forTSNE)
# table(snps_long_final_forTSNE$species)
# table(snps_long_final_forTSNE$sampleName)


snps_long_final_forTSNE <-   snps_long_final %>%  mutate(sampleName=sample) %>% select(species,site,sampleName,Snps) %>% filter(.,Snps>0.2) 
# snps_long_final_forTSNE <- snps_long_final_forTSNE %>% filter(sampleName=)

snps_long_final_forTSNE <- filter(snps_long_final_forTSNE, !grepl('FAM', sampleName))


snps_long_final_forTSNE$sampleName <- gsub("_","",snps_long_final_forTSNE$sampleName) %>% gsub("-","_",.)
###go wide
snps_long_final_forTSNE_all <- snps_long_final_forTSNE
snps_long_final_forTSNE <- snps_long_final_forTSNE %>% select(-species)
# snps_long_final_forTSNE <- snps_long_final_forTSNE %>% filter(species.x=="L. delbrueckii") %>% select(-species.x)

mst1_metagenomicMutations_tsne <- spread(snps_long_final_forTSNE, site, Snps)%>%replace(is.na(.), 0) %>% column_to_rownames(var = "sampleName")
dim(mst1_metagenomicMutations_tsne)


# 
# fortsne <- mst1_metagenomicMutations_tsne[!duplicated(mst1_metagenomicMutations_tsne), ]
# 
# # fortsne[duplicated(fortsne), ] %>% nrow()
# dim(fortsne)
# set.seed(1234567)
# # set.seed(12314567)
# set.seed(007)
# 
# tsne <- Rtsne(fortsne, dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# # tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne_plot <- as.data.frame(tsne$Y)
# tsne_plot$site <- rownames(fortsne)
# tsne_plot$sample <- str_split_fixed(tsne_plot$site, "_", 2)[,1]
# tsne_plot$number <- str_split_fixed(tsne_plot$site, "_", 2)[,2]
# tsne_plot$cheese <- plyr::revalue(tsne_plot$sample,c("101"="emmentaler","105"="emmentaler","115"="emmentaler","124"="emmentaler","150"="emmentaler","190"="emmentaler","202"="gruyere","203"="gruyere","280"="gruyere","302"="sbrinz","305"="sbrinz"))
# 
# # colors <- c("#aa0000","#d40000","#ff0000","#ff2a2a","#ff5555","#ff8080","#009cf0","#34d0fc","#b4ecfc","#008181","#00dcdc")
# colors <- c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#b4ecfc","#008181","#00dcdc","grey")
# 
# tsne_plotss <- ggplot(tsne_plot, aes(x = V1, y = V2,color=sample,fill=sample,shape=cheese)) +labs(title=paste0("all SNV (n=",ncol(fortsne),")"))+
#     geom_mark_ellipse(aes(fill = sample),alpha=0.1) +geom_point(alpha = 0.8,size=5) + theme_classic() +theme(legend.position = "none")+scale_color_manual(values = colors)+scale_fill_manual(values = colors)+geom_text(aes(label=site),hjust=0, vjust=0)
# 
# tsne_plotss

###------------------------------
##only Ldel
###------------------------------

snps_long_final_forTSNE_ldel <- snps_long_final_forTSNE_all %>% filter(species=="L. delbrueckii") %>% select(-species)

mst1_metagenomicMutations_tsne_ldel <- spread(snps_long_final_forTSNE_ldel, site, Snps)%>%replace(is.na(.), 0) %>% column_to_rownames(var = "sampleName")
dim(mst1_metagenomicMutations_tsne_ldel)



fortsne <- mst1_metagenomicMutations_tsne_ldel[!duplicated(mst1_metagenomicMutations_tsne_ldel), ]

# fortsne[duplicated(fortsne), ] %>% nrow()
dim(fortsne)
set.seed(1234567)
# set.seed(12314567)
set.seed(007)

tsne_ldel <- Rtsne(fortsne, dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne_ldel$Y)
tsne_plot$site <- rownames(fortsne)
tsne_plot$sample <- str_split_fixed(tsne_plot$site, "_", 2)[,1]
tsne_plot$number <- str_split_fixed(tsne_plot$site, "_", 2)[,2]
tsne_plot$cheese <- plyr::revalue(tsne_plot$sample,c("101"="emmentaler","105"="emmentaler","115"="emmentaler","124"="emmentaler","150"="emmentaler","190"="emmentaler","202"="gruyere","203"="gruyere","280"="gruyere","302"="sbrinz","305"="sbrinz"))

# colors <- c("#aa0000","#d40000","#ff0000","#ff2a2a","#ff5555","#ff8080","#009cf0","#34d0fc","#b4ecfc","#008181","#00dcdc")
# colors <- c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#b4ecfc","#008181","#00dcdc")

tsne_plotss_ldel <- ggplot(tsne_plot, aes(x = V1, y = V2,color=sample,fill=sample,shape=cheese)) +labs(title=paste0("L. delbrueckii SNV (n=",ncol(fortsne),")"))+
    geom_mark_ellipse(aes(fill = sample),alpha=0.1) +geom_point(alpha = 0.8,size=5) + theme_classic() +scale_color_manual(values = colors) +theme(legend.position = "none")+scale_fill_manual(values = colors)+geom_text(aes(label=site),hjust=0, vjust=0)

tsne_plotss_ldel


###------------------------------
##only Sterm
###------------------------------

snps_long_final_forTSNE_sterm <- snps_long_final_forTSNE_all %>% filter(species=="S. thermophilus") %>% select(-species)

mst1_metagenomicMutations_tsne_sterm <- spread(snps_long_final_forTSNE_sterm, site, Snps)%>%replace(is.na(.), 0) %>% column_to_rownames(var = "sampleName")
dim(mst1_metagenomicMutations_tsne_sterm)



fortsne <- mst1_metagenomicMutations_tsne_sterm[!duplicated(mst1_metagenomicMutations_tsne_sterm), ]

# fortsne[duplicated(fortsne), ] %>% nrow()
dim(fortsne)
set.seed(1234567)
# set.seed(12314567)
set.seed(007)

tsne_sterm <- Rtsne(fortsne, dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne_sterm$Y)
tsne_plot$site <- rownames(fortsne)
tsne_plot$sample <- str_split_fixed(tsne_plot$site, "_", 2)[,1]
tsne_plot$number <- str_split_fixed(tsne_plot$site, "_", 2)[,2]
tsne_plot$cheese <- plyr::revalue(tsne_plot$sample,c("101"="emmentaler","105"="emmentaler","115"="emmentaler","124"="emmentaler","150"="emmentaler","190"="emmentaler","202"="gruyere","203"="gruyere","280"="gruyere","302"="sbrinz","305"="sbrinz"))

# colors <- c("#aa0000","#d40000","#ff0000","#ff2a2a","#ff5555","#ff8080","#009cf0","#34d0fc","#b4ecfc","#008181","#00dcdc")
# colors <- c("#d45500","#e05a2c","#e46d43","#e77f5b","#eb9172","#eea48a","#009cf0","#34d0fc","#80e5ff","#008181","#00dcdc")

# tsne_plotss_sterm <- ggplot(tsne_plot, aes(x = V1, y = V2,color=sample,fill=sample,shape=cheese)) +
#     geom_mark_ellipse(aes(fill = sample),alpha=0.1) +geom_point(alpha = 0.8,size=5) + theme_classic() +theme(legend.position = "right")+scale_color_manual(values = colors)+scale_fill_manual(values = colors)+geom_text(aes(label=site),hjust=0, vjust=0) #+labs(title=paste0("S. thermophilus SNV (n=",ncol(fortsne),")")



tsne_plotss_sterm <- ggplot(tsne_plot, aes(x = V1, y = V2,color=sample,fill=sample,shape=cheese)) +
    geom_point(alpha = 0.8,size=5) + theme_classic() +theme(legend.position = "none")+scale_color_manual(values = colors)+scale_fill_manual(values = colors)+labs(x="tsne1",y="tsne2") + geom_mark_ellipse(aes(fill = sample),alpha=0.1)+geom_text(aes(label=site),hjust=0, vjust=0)
#+labs(x="tsne1",y="tsne2") +labs(title=paste0("S. thermophilus SNV (n=",ncol(fortsne),")")) #

tsne_plotss_sterm

###------------------------------
##all togehter
###------------------------------

library(patchwork)

# tsne_plotss+tsne_plotss_sterm+tsne_plotss_ldel
tsne_plotss_sterm+tsne_plotss_ldel


svg("~/Desktop/mid_thesis/report/figures/20200430/chapter2/Tsne_diversity",width=18,height=8)
# png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_sterm.png", width = 1800, height = 4800,res=300)
tsne_plotss+tsne_plotss_sterm+tsne_plotss_ldel
dev.off()


pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/tsne_SNVs.pdf",width=4.5,height=3)

tsne_plotss_sterm

dev.off()

```


### ordination plot

```{r }

sample_relative_wide  <- read_csv("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all_withINfomration_wide.csv")
# table(sample_relative_wide$core)
# unique(sample_relative_wide$core)
dim(sample_relative_wide)
# sample_relative_wide[which(sample_relative_wide$core=="core"),]
# 
# snps_long_final <-  gather(sample_relative_wide, sample, Snps, "101-starter 1":"FAMstrains-FAM24855", factor_key=TRUE,na.rm = TRUE)
# snps_long_final[which(snps_long_final$core=="core"),]
# 
# 
# 
# snps_long_final$species <- snps_long_final$species
# snps_long_final$culture <- str_split_fixed(snps_long_final$sample,"-",3)[,1]
# # snps_long_final <- snps_long_final %>% select(-"species.x")
# table(snps_long_final$core)

colnames(sample_relative_wide)

sample_relative_wide_prep <- sample_relative_wide %>% filter(!is.na(core)) %>% select(site,colnames(sample_relative_wide)[grepl("-starter",colnames(sample_relative_wide))]) %>% column_to_rownames(.,var="site") %>%replace(is.na(.), 0)%>% as.matrix
library(vegan)
dist <- vegdist(sample_relative_wide_prep,  method = "bray")
head(dist)
dim(dist)
length(dist)

```


####all samples together
Here, I put all cultures together and see if I can phase them with tSNE

```{r}

# sample_relative_safe_Sterm_final$sampleNew <- paste0(sample_relative_safe_Sterm_final$culture,"_",sample_relative_safe_Sterm_final$sample)
# table(sample_relative_safe_Sterm_final$sampleNew)
# colnames(sample_relative_safe_Sterm_final_wide)
sample_relative_safe_Sterm_final_longAll <- gather(sample_relative_safe_Sterm_final_wide, sample, Snps, "starter 1":"starter 12", factor_key=TRUE,na.rm = TRUE) 

sample_relative_safe_Sterm_final_longAll$sampleNew <- paste0(sample_relative_safe_Sterm_final_longAll$culture,"_",sample_relative_safe_Sterm_final_longAll$sample)
dim(sample_relative_safe_Sterm_final_longAll)
# snps_freebayes_long <- gather(snps_freebayes, sample, Snps, colnames(snps_freebayes)[6]:colnames(snps_freebayes)[ncol(snps_freebayes)], factor_key=TRUE,na.rm = TRUE) 

sample_relative_safe_Sterm_final_longAll_filtered <- sample_relative_safe_Sterm_final_longAll %>% filter(sampleNew %in% c("101_starter 1","101_starter 2","101_starter 3","101_starter 4","101_starter 5","101_starter 6","101_starter 7","105_starter 1","105_starter 2","105_starter 3","105_starter 4","105_starter 5","105_starter 6","105_starter 7","105_starter 8","105_starter 9","105_starter 10","105_starter 11","105_starter 12","115_starter 1","115_starter 2","124_starter 1","124_starter 2", "124_starter 3","150_starter 1","150_starter 2","150_starter 3","150_starter 4","150_starter 5","150_starter 6","190_starter 1","190_starter 2","190_starter 3","202_starter 1","202_starter 2","202_starter 3","202_starter 4","202_starter 5","202_starter 6", "202_starter 7","202_starter 8","203_starter 1","203_starter 2","203_starter 3","203_starter 4","280_starter 1","280_starter 2", "280_starter 3","302_starter 1", "302_starter 2","305_starter 1", "305_starter 2","305_starter 3","305_starter 4"))

sample_relative_safe_Sterm_final_longAll_filtered_prep <- sample_relative_safe_Sterm_final_longAll_filtered %>% select(-sample)

sample_relative_safe_Sterm_final_wide_new <- spread(sample_relative_safe_Sterm_final_longAll_filtered_prep, sampleNew, Snps)  %>%replace(is.na(.), 0)
colnames(sample_relative_safe_Sterm_final_wide_new)
sample_relative_safe_Sterm_final_longAll_final <- gather(sample_relative_safe_Sterm_final_wide_new, sampleNew, Snps, "101_starter 1":"305_starter 4", factor_key=TRUE,na.rm = TRUE) 

# 
# snps_long$samplesNew <- revalue(snps_long$sample,c("101_starter 1","101_starter 2","101_starter 3","101_starter 4","101_starter 5","101_starter 6","101_starter 7","105_starter 1","105_starter 2","105_starter 3","105_starter 4","105_starter 5","105_starter 6","105_starter 7","105_starter 8","105_starter 9","105_starter 10","105_starter 11","105_starter 12","115_starter 1","115_starter 2","124_starter 1","124_starter 2", "124_starter 3","150_starter 1","150_starter 2","150_starter 3","150_starter 4","150_starter 5","150_starter 6","190_starter 1","190_starter 2","190_starter 3","202_starter 1","202_starter 2","202_starter 3","202_starter 4","202_starter 5","202_starter 6", "202_starter 7","202_starter 8","203_starter 1","203_starter 2","203_starter 3","203_starter 4","280_starter 1","280_starter 2", "280_starter 3","302_starter 1", "302_starter 2","305_starter 1", "305_starter 2","305_starter 3","305_starter 4"))
# 



p3 <- ggplot(sample_relative_safe_Sterm_final_longAll_final,aes(x=sampleNew,y=Snps,group=site,color=species.x,fill=species.x))+ geom_line(size=0.2, alpha=.1)+ 
    # facet_wrap(culture~.,nrow = 11)+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none"
  )
  
  # p3
  

# svg("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.svg",width=2.5,height=2.5)
png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_sterm_wide.png", width = 4800, height = 1800,res=300)
p3
dev.off()  

```

####tSNE

```{r}

library(forcats) 
library(Rtsne)
library(fpc)
library(ggrepel)
###--------------------------------------------------------------------
##colors
###--------------------------------------------------------------------

# colorsTSNE <- c("#DD8D29","#E2D200","#46ACC8","black","#B40F20")

# colorsTSNE <- c("#E69F00", "#56B4E9", "#009E73",
          # "#F0E442", "#0072B2", "#D55E00", "#CC79A7","grey","#000000")

# colorsTSNE <- rev(c("#FF9E79","#FB6D4C","#C23B22","#8A0000","#580000","#D55E00"))
###--------------------------------------------------------------------
##subset
###--------------------------------------------------------------------

# mst1_metagenomicMutations <- sample_relative_wide %>% filter(.,species=="S. thermophilus") %>% filter(.,significance %in% c("MODERATE","HIGH")) %>% filter(.,mst1>0.9 & mst2 <=0.9) %>% add_column(group="mst1") %>%replace(is.na(.), 0)
# mst1_metagenomicMutations <- sample_relative_safe_Sterm_final_wide %>% filter(.,mst1>0.9 & mst2 <=0.9)%>% add_column(group="mst1") 


# mst1_metagenomicMutations_long <-  mst1_metagenomicMutations %>% gather(., sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2", factor_key=TRUE,na.rm = TRUE)# %>% filter(.,Snps>0.005)

# mst1_metagenomicMutations_tsne <- sample_relative_safe_Sterm_final_wide_new


###--------------------------------------------------------------------
##prep for tSNE
###--------------------------------------------------------------------



fortsne <- sample_relative_safe_Sterm_final_wide_new %>% select(.,"site","101_starter 1","101_starter 2","101_starter 3","101_starter 4","101_starter 5","101_starter 6","101_starter 7","105_starter 1","105_starter 2","105_starter 3","105_starter 4","105_starter 5","105_starter 6","105_starter 7","105_starter 8","105_starter 9","105_starter 10","105_starter 11","105_starter 12","115_starter 1","115_starter 2","124_starter 1","124_starter 2", "124_starter 3","150_starter 1","150_starter 2","150_starter 3","150_starter 4","150_starter 5","150_starter 6","190_starter 1","190_starter 2","190_starter 3","202_starter 1","202_starter 2","202_starter 3","202_starter 4","202_starter 5","202_starter 6", "202_starter 7","202_starter 8","203_starter 1","203_starter 2","203_starter 3","203_starter 4","280_starter 1","280_starter 2", "280_starter 3","302_starter 1", "302_starter 2","305_starter 1", "305_starter 2","305_starter 3","305_starter 4")
fortsne <- fortsne[!duplicated(fortsne[,-1]), ]
# fortsne[duplicated(fortsne), ] %>% nrow()

set.seed(1234567)
# set.seed(12314567)

tsne <- Rtsne(fortsne[,-1], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- fortsne[,1]
tsne_plot$species <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_") %>%  gsub("_",". ",.)



##------------plot blank
# ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()

##------------clusters

# set.seed(123456)

set.seed(1234567)
ds.real = dbscan(tsne_plot[,c(1,2)], 15)
# decreasing_clusternames <- (names(sort(table(ds.real$cluster),decreasing = TRUE)))
# change_name <- setNames(1:length(decreasing_clusternames),decreasing_clusternames)
# clusterNames <- revalue(as.factor(ds.real$cluster),change_name)
# clusterNames<- revalue(clusterNames,c("4"="1","6"="1","7"="1","10"="1","11"="1","5"="4","8"="5","9"="6"))##add certain clusters to others
tsne_plot$density = factor(ds.real$cluster)
# levels(tsne_plot$density) <- c(levels(tsne_plot$density), "5")

# special_cases <- c("CP046134_494890_A_G","CP046134_482906_A_C")
# tsne_plot[which(tsne_plot$site %in% special_cases),"density"] <- "5"
ds.cent = tsne_plot %>% group_by(density) %>% select(V1,
    V2) %>% summarize_all(mean)
ggplot(tsne_plot, aes(x = V1, y = V2, colour = density)) +
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density),
    data = ds.cent) + guides(colour = FALSE)


```



###### Ldel
```{r}




sample_relative_safe_Sterm_final <- sample_relative_safe_woSTRAINS_sub06[which(sample_relative_safe_woSTRAINS_sub06$species=="L. delbrueckii"),]
sample_relative_safe_Sterm_final_wide <- spread(sample_relative_safe_Sterm_final, sample, Snps)  %>%replace(is.na(.), 0)
dim(sample_relative_safe_Sterm_final_wide)

  ##============
### plot Streptococcus thermophius
   ##============

sample_relative_safe_woSTRAINS$sample <- droplevels(sample_relative_safe_woSTRAINS$sample)
levels(sample_relative_safe_Sterm_final$sample)

  ##============
### plot 
   ##============

# all_colours <- rev(c("#EB4D4D","#10B552") ) 
all_colours <- rev(c("#EB4D4D") ) 

# colnames(sample_relative_safe_woSTRAINS)

##plot
  p2 <- ggplot(sample_relative_safe_Sterm_final,aes(x=sample,y=Snps,group=site,color=species.x,fill=species.x))+ geom_line(size=0.2, alpha=.1)+ 
    facet_wrap(culture~.,nrow = 11)+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none"
  )
  
  p2
  

# svg("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.svg",width=2.5,height=2.5)
png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_sterm.png", width = 1800, height = 4800,res=300)
p2
dev.off()


##------------------------------------------------------
##old
##------------------------------------------------------





library(readr)
snps_long_final <- read_csv("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all.csv")

##----------------
##01_remove strains
##----------------
# sample_relative_safe_woSTRAINS <- snps_long_final[which(snps_long_final$species %in% c("L. delbrueckii","S. thermophilus")),] 
sample_relative_safe_woSTRAINS <- snps_long_final[which(snps_long_final$species %in% c("L. delbrueckii")),] 


sample_relative_safe_woSTRAINS$samplesNew = factor(sample_relative_safe_woSTRAINS$samplesNew, levels=c("starter 1","starter 2","starter 3","starter 4","starter 5","starter 6","starter 7","starter 8","starter 9","starter 10","starter 11","starter 12"))
sample_relative_safe_woSTRAINS$samplesNew <- droplevels(sample_relative_safe_woSTRAINS$samplesNew)

  ##============
### plot 
   ##============

# all_colours <- rev(c("#EB4D4D","#10B552") ) 
all_colours <- rev(c("#10B552") ) 

# colnames(sample_relative_safe_woSTRAINS)

##plot
  p3 <- ggplot(sample_relative_safe_woSTRAINS,aes(x=samplesNew,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.2, alpha=.1)+ 
    facet_wrap(culture~.,nrow = 11)+
     labs("",
         x=" ",
         y="Alternative allele frequency")+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none"
  )
  
  p3
  

# svg("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/fitnesMatrix.svg",width=2.5,height=2.5)
png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_ldel.png", width = 1800, height = 4800,res=300)
p3
dev.off()
library(patchwork)
png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled.png", width = 3000, height = 4800,res=300)
p2+p3
dev.off()

```


#nucleotide diversity
```{r}
library(readr)
library(plyr)
library(tidyverse)

# write.csv(snps_long_final,"~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all.csv",row.names = FALSE)

# snps_long_final_02 <- read_csv("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all.csv")
# snps_long_final <- read_csv("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all.csv", col_types = cols(culture = col_character()))
# ~/Desktop/Projects/2020_strainDelineation/05_SNP_called/snps_analysed_all.csv
# table(snps_long_final$sample)
# table(snps_long_final$culture)

sample_relative_safe_woSTRAINS <- snps_long_final %>% filter(species %in% c("L. delbrueckii","S. thermophilus")) %>%  filter(Snps>0.05) %>% filter(culture!="FAMstrains")
table(sample_relative_safe_woSTRAINS$culture)
snp_count <- as.data.frame(table(sample_relative_safe_woSTRAINS$culture,sample_relative_safe_woSTRAINS$species,sample_relative_safe_woSTRAINS$sample)) %>% dplyr::rename(culture=Var1,species=Var2,sample=Var3)

snp_count <- snp_count %>% filter(Freq>0)
snp_count$sample = factor(snp_count$sample, levels=c("starter 1","starter 2","starter 3","starter 4","starter 5","starter 6","starter 7","starter 8","starter 9","starter 10","starter 11","starter 12"))
snp_count$sample <- droplevels(snp_count$sample)

snp_count_ldel <- snp_count %>% filter(species=="L. delbrueckii")

   pldel <- ggplot()+ #stat_summary(fun.y=sum, geom="line")+
     geom_point(data = snp_count_ldel,aes(y = Freq, x = sample),fill="#10B552",color="#10B552")+
     geom_line(data = snp_count_ldel,aes(y = Freq, x = sample,group=species),fill="#10B552",color="#10B552")+
     facet_wrap(culture~.,nrow = 11)+
   labs("",
         x="",
         y="SNPs",
          title="")+
  theme_classic()+
     scale_x_discrete( expand = c(0, 0)) +
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"))
  pldel

  ##------sterm
  
  snp_count_sterm <- snp_count %>% filter(species=="S. thermophilus")

   psterm <- ggplot()+ #stat_summary(fun.y=sum, geom="line")+
     geom_point(data = snp_count_sterm,aes(y = Freq, x = sample),fill="#EB4D4D",color="#EB4D4D")+
     geom_line(data = snp_count_sterm,aes(y = Freq, x = sample,group=species),fill="#EB4D4D",color="#EB4D4D")+
     facet_wrap(culture~.,nrow = 11)+
   labs("",
         x="",
         y="SNPs",
          title="")+
  theme_classic()+
     scale_x_discrete( expand = c(0, 0)) +
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"))
  psterm
  
library(patchwork)
png("~/Desktop/Projects/2020_strainDelineation/05_SNP_called/plots/snpsCalled_nucleotide_diversity_evolution.png", width = 3000, height = 4800,res=300)
psterm+pldel
dev.off()

```



#single strain analysis

Here, I analyse the different reference genome sequences

## ONT


```{bash}

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x1_T43_Seq1_VS_fq
cp /home/vincent/Downloads/fastq_pass.tar.gz.zip /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x1_T43_Seq1_VS_fq
cd /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x1_T43_Seq1_VS_fq
openssl enc -aes-256-cbc -md md5 -d -in fastq_pass.tar.gz.enc > fastq_pass.tar.gz
tar xvzf fastq_pass.tar.gz

##------------------------2
rm  /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq
cp /home/vincent/Downloads/'fastq_pass.tar.gz(1).enc' /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq/
cd /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq/
openssl enc -aes-256-cbc -md md5 -d -in fastq_pass.tar.gz.enc > fastq_pass.tar.gz
tar xvzf fastq_pass.tar.gz

scp /home/vincent/Downloads/fastq_pass.tar.gz.enc vincent@130.223.49.145:/data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq/

wget https://campuscloud.unibe.ch/ssf/s/readFile/folderEntry/64630263/02dc805872bb70a00172d615fd105f8f/1592660049000/last/fastq_pass.tar.gz.enc
##------------------------3
rm  /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x3_ES43_EXP02_Seq3_VS_fq

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x3_ES43_EXP02_Seq3_VS_fq
cp /home/vincent/Downloads/fastq_pass.tar.gz.enc /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x3_ES43_EXP02_Seq3_VS_fq/
cd /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x3_ES43_EXP02_Seq3_VS_fq/
#unzip fastq_pass.tar.gz.zip
openssl enc -aes-256-cbc -md md5 -d -in fastq_pass.tar.gz.enc > fastq_pass.tar.gz
tar xvzf fastq_pass.tar.gz

##------------------------2

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x4_ES43_EXP02_Seq4_VS
cp /home/vincent/Downloads/fastq_pass.tar.gz.enc /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x4_ES43_EXP02_Seq4_VS/
cd /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x4_ES43_EXP02_Seq4_VS/
#unzip fastq_pass.tar.gz.zip
openssl enc -aes-256-cbc -md md5 -d -in fastq_pass.tar.gz.enc > fastq_pass.tar.gz
tar xvzf fastq_pass.tar.gz

```

```{bash}

scp /home/vincent/Desktop/Projects/2020_strainDelineation/ONT_sequencing/20200619_Seq_run_ONT_ifik.csv vincent@130.223.49.145:/data/Project/2020_StarterCultureDiversity/99_log/

```


####rename and merge
```{bash}

for runnsss in $(echo "2")
do

longRunssss=$(echo -e "_Seq"${runnsss}"_VS")
directory=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/ |grep ${longRunssss})
echo "============================="
echo ${longRunssss}
for samplezzz in $(grep "${longRunssss}" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |cut -f 3)
do

rm -r  /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${samplezzz}/
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${samplezzz}/
barcode=$(grep "${samplezzz}" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |cut -f 4|sed 's/RB //g')
echo "----------------------"
echo -e "barcode :" ${barcode} " & sample: " ${samplezzz}
cat /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/${directory}/fastq_pass/barcode${barcode}/*fastq > /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${samplezzz}/${samplezzz}.fastq

done #samplezzz

done #runsss



for runnsss in $(echo "2")
do

longRunssss=$(echo -e "_Seq"${runnsss}"_VS")
directory=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/ |grep ${longRunssss})
echo "============================="
echo ${longRunssss}
for samplezzz in $(grep "${longRunssss}" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |cut -f 3)
do

barcode=$(grep "${samplezzz}" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |cut -f 4|sed 's/RB //g')
echo "----------------------"
echo -e "barcode :" ${barcode} " & sample: " ${samplezzz}
grep "^@" -c /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${samplezzz}/${samplezzz}.fastq

done #samplezzz

done #runsss

###---------------------------------------------
#20210310
###---------------------------------------------



for samplessss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/20210309_20210222_ES59_FASTQ/20210222_ES59_Somerville_FASTQ/sequencing_summary/sample_summary.txt) ;do

samplezzz=$(awk -F "\t" -v sams="$samplessss" '{OFS="\t"}{if($2==sams)print $1}' /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/20210309_20210222_ES59_FASTQ/20210222_ES59_Somerville_FASTQ/sequencing_summary/sample_summary.txt)
echo -e "make barcode"${samplessss} " into "${samplezzz}
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/

cat /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/20210309_20210222_ES59_FASTQ/20210222_ES59_Somerville_FASTQ/fastq_pass/barcode${samplessss}/*fastq > /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/${samplezzz}.fastq

done

###---------------------------------------------
#special case
###---------------------------------------------

sample=24749
cat /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/20210309_20210222_ES59_FASTQ/20210222_ES59_Somerville_FASTQ/fastq_pass/barcode06/*fastq /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/24749/24749.fastq > /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/${sample}.fastq


```


##flye

```{bash}

##===========
##Assembly plasmid
#samples 20210310
##===========

    rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz/
    mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz/
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep ".fastq$"|sed 's/.fastq//g' |sed '1d')
  do
  
#  ll /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/
  

  echo "=============="
  echo ${sample}
  echo "=============="

/home/vincent/apps/Filtlong/bin/filtlong --min_length 6000 --keep_percent 90 --target_bases 500000000 --mean_q_weight 10 /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/${sample}.fastq |gzip > /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz//${sample}_filtered.fastq.gz


  done

###---------------------------------------------
#special case
###---------------------------------------------
sample=24732

sample=24749

/home/vincent/apps/Filtlong/bin/filtlong --min_length 40000 --keep_percent 90 --target_bases 100000000 --mean_q_weight 10 /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/${sample}.fastq |gzip > /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz//${sample}_filtered.fastq.gz



##===========
##Assembly plasmid
#samples 20210310
##===========
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep ".fastq$"|sed 's/.fastq//g' |sed '1d' )
  do
  
#  ll /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/
  

    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/
  echo "=============="
  echo ${sample}
  echo "=============="

   #python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
  #    /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/${sample}.fastq \
  #    --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/
      

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 1800000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz//${sample}_filtered.fastq.gz \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/
      
  done
  
  
  
#mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/09_flye/assembly/24749_02/30-contigger/
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/09_flye/assembly/24749_02/30-contigger/graph_final.gfa ~/Desktop/Projects/2020_StarterCultureDiversity/09_flye/assembly/24749_02/30-contigger/
  

  ##===========
##length
##===========
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep ".fastq$"|sed 's/.fastq//g' )
#for sample in $(grep "20200527_x2_T43EXP01_Seq2_VS" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3 )
  do

  echo "=============="
  echo ${sample}

 # seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/assembly.fasta
cat /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/30-contigger/*stats*

  done

sample=24740
##===========
##Assembly old samples
##===========
for sample in $(grep "20200527_x2_T43EXP01_Seq2_VS" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3|sed '1d' )
  do
  
  sample=24740

    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo "=============="
  echo ${sample}
  echo "=============="

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
  done
  
  
  ##===========
##length
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#for sample in $(grep "20200527_x2_T43EXP01_Seq2_VS" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3 )
  do

  echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
  #==============================================================================================================
  ##----------------------------
  ###assemblies with problems
  ##----------------------------
    #==============================================================================================================
    
    ##-------------------
    ##count reads
    ##-------------------
    for sample in $(echo "24855 24773 24765 24742 24740 24738 24736 24726")
  do
   echo "=============="
  echo ${sample}
  
  grep "^@" -c /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq
done


    ##-------------------
    ##too few reads
    ##-------------------
      for sample in $(echo "24773 24765")
    do
    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo "=============="
  echo ${sample}
  echo "=============="

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
    
    done
  ##-------------------
    ##assembly length
    ##-------------------
  for sample in $(echo "24855 24742 24740 24738 24736 24726")
  do
   echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
  
  
    ##-------------------
    ##too many reads
    ###----->>>>here I filter for only the longest reads
    ##-------------------
    
    
    
        for sample in $(echo "24855 24742 24740 24738 24736 24726")
        do
        
        rm /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz
         
    
/home/vincent/apps/Filtlong/bin/filtlong --min_length 6000 --keep_percent 90 --target_bases 500000000 --mean_q_weight 10 /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq | gzip > /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz

#mv /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq.gz

        
        done
    
    
    ##-------------------
    ##too many reads
    ##assembly
    ##-------------------
    
    
      for sample in $(echo "24855 24742 24740 24738 24736 24726")
        do
        
       
    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo "=============="
  echo ${sample}
  echo "=============="

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
        
        done
    
    
      ##-------------------
    ##check
    ##-------------------
    
  for sample in $(echo "24855 24773 24765 24742 24740 24738 24736 24726")
  do
   echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
    
    
    
    
  for sample in $(echo "24855 24773 24765 24742 24740 24738 24736 24726")
  do
   echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
  sample=24740

    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo "=============="
  echo ${sample}
  echo "=============="

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
  done
  
    #==============================================================================================================
  ##----------------------------
  ###assemblies partially problematic
  ##optimise!!!
  ##change Minimum overlap 
  ##----------------------------
    #==============================================================================================================
    
      for sample in $(echo "24751 24731 24743 24758 24729 24795 24790 24777 24762 24759 24749 24731")
  do
   echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
    
   ##-------------------
    ##count reads
    ##-------------------
    for sample in $(echo "24751 24731 24743 24758 24729 24795 24790 24777 24762 24759 24749 24731")
  do
   echo "=============="
  echo ${sample}
  
  grep "^@" -c /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq
done

 ##-------------------
    ##filtlong
    ##-------------------
    for sample in $(echo "24751 24731 24743 24758 24729 24795 24790 24777 24762 24759 24749 24731")
  do
   echo "=============="
  echo ${sample}
   rm /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz
         
    
/home/vincent/apps/Filtlong/bin/filtlong --min_length 6000 --keep_percent 90 --target_bases 150000000 --mean_q_weight 10 /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq | gzip > /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz

done


##-------------------
    ##too many reads
    ##assembly
    ##-------------------
    
    
    for sample in $(echo "24751 24731 24743 24758 24729 24795 24790 24777 24762 24759 24749 24731")
        do
        
       
    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo "=============="
  echo ${sample}
  echo "=============="

   python /home/vincent/Flye/bin/flye --min-overlap 4000 --threads 5  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
        
        done
        
##-------------------
    ##final samples
    ##------------------   

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765")
#for sample in $(grep "20200527_x2_T43EXP01_Seq2_VS" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3 )
  do

  echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done



```

check circlatrity

```{bash}
rm /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt
echo -e "sample\tcircular_contigs\tlargestContigSize\tCoverage\tCircularity" > /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
  do
  echo "====================================================="
  echo $sample
  circularsss=$(grep "#" -v /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/30-contigger/contigs_stats.txt |awk -F "\t" '{OFS="\t"}{if($4=="Y")print $0}'| wc -l )
grep "#" -v /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/30-contigger/contigs_stats.txt |sort -t$'\t' -k2 -nr |head -1 |awk -F "\t" -v cics="$circularsss" -v samss="$sample" '{OFS="\t"}{print samss,cics,$2,$3,$4}' >> /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt
done


##-------------------------------------------
#new
##-------------------------------------------
rm /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt
echo -e "sample\tcircular_contigs\tlargestContigSize\tCoverage\tCircularity" > /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt

for sample in $( echo "S24750
S24739
S13496
S24738
S13499
S24730
S24726
S24728
S13498
S24734
S24737
S24853
S24854
S24855
S24742
S13494
S24758
S24735
S24736
S24724
S24748
S24740
S24745
L24756
L24786
L24790
S24732
S24733
S24743
S24746
S24747
S24749
L24754
L24798" |sort|uniq)
do
genomesss=$(echo ${sample}| sed 's/S//g'| sed 's/L//g')
filesss=$(ls /data/Project/2020_StarterCultureDiversity/09_flye/assembly/| grep "$genomesss")

  circularsss=$(grep "#" -v /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${filesss}/30-contigger/contigs_stats.txt |awk -F "\t" '{OFS="\t"}{if($4=="Y")print $0}'| wc -l )
grep "#" -v /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${filesss}/30-contigger/contigs_stats.txt |sort -t$'\t' -k2 -nr |head -1 |awk -F "\t" -v cics="$circularsss" -v samss="$sample" '{OFS="\t"}{print samss,cics,$2,$3,$4}' >> /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt


done

```

move local
```{bash}

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/ONT_assembly_stats.txt

```


##circlator

```{bash}


##===========
##circlator
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
  do
  

  echo "=============="
  echo ${sample}

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/circlator/${sample}
#mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/circlator/${sample}

mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/

#cat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R1_val_1.fq.gz \
#/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R2_val_2.fq.gz > \
#/data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/${sample}.fq.gz


circlator all /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta /data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/${sample}.fq.gz /data/Project/2020_StarterCultureDiversity/09_flye/circlator/${sample}


  done
  

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/
```
##polishing

```{bash}


###===========================
###-----------------
##cleanup
###-----------------
###===========================    
     
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv|
grep -v "24724"|
grep -v "24726"|
grep -v "24728"|
grep -v "24730"|
grep -v "24731"|
grep -v "24734"|
grep -v "24736"|
grep -v "24737"|
grep -v "24738"|
grep -v "24739"|
grep -v "24740")

for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep ".fastq$"|sed 's/.fastq//g' )
  do

name=$(echo $sample)
echo ${name}
done
#sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}
reference_fasta=/data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/assembly.fasta
#ont_reads=/data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq
ont_reads=/data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz//${sample}_filtered.fastq.gz
#reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
###===========================
###-----------------
##graphmap mapping
###-----------------
###===========================      

name=first
num=1

for name in first second third fourth
do 
#rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/
#mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc


#rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam
rm -r $Overall_output_directory
#rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2Final*

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 



done

done #sample


###===========================
###-----------------
##samples
###-----------------
###===========================    
     
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765")
#for sample in $(grep "20200527_x2_T43EXP01_Seq2_VS" /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3 )
  do

  echo "=============="
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done

###===========================
###-----------------
##Racon polishing with graphmap
###-----------------
###===========================    
     
     
     24724  24726  24728  24730  24731  24734  24736  24737  24738  24739  24740
##------------------define variables------------------

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765"|
grep -v "24724"|
grep -v "24726"|
grep -v "24728"|
grep -v "24730"|
grep -v "24731"|
grep -v "24734"|
grep -v "24736"|
grep -v "24737"|
grep -v "24738"|
grep -v "24739"|
grep -v "24740")

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765")

for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep ".fastq$"|sed 's/.fastq//g' )
  do

name=$(echo $sample)
#sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}
reference_fasta=/data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/assembly.fasta
#ont_reads=/data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq
ont_reads=/data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz//${sample}_filtered.fastq.gz
#reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
###===========================
###-----------------
##graphmap mapping
###-----------------
###===========================      

name=first
num=1

for name in first second third fourth
do 
rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/
mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc

#name=first

graphmap align -t ${threads} \
  -r $reference_fasta \
  -d $ont_reads \
  -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam 

#  -d $ont_reads | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam - 

  
###===========================
###-----------------
##racon polishing
###-----------------
###===========================   

#rm -r$Overall_output_directory/0${num}_${name}Racon_graphmap/assembly

mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly

samtools view -bS $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam -

#samtools sort -@${threads} -O SAM $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam -

#samtools view -S -b $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam > $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam 

/home/vincent/apps/racon/new/racon/build/bin/racon \
 -t ${threads} \
  $ont_reads \
  $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam \
  $reference_fasta > \
  $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta
  


samtools index $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam 

qualimap bamqc -bam $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam \
  -outdir $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc --java-mem-size=40G 

#rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam
rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2Final*

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 



done

done #sample
##--------------------cleanup


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz
#reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
name=first
num=1

for name in first second third fourth
do 



#rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 

done
#done # loop for 1AB and 2AB samples


##--------------------test PROKKA
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
name=fourth
num=4

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes   $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta
  
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon/PROKKA_*.gff
  
  echo "=======================================================ALT==============================================================="
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all/PROKKA_*.gff


  


###===========================
###-----------------
##quast
###-----------------
###===========================  

#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz


echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
    rm -r $Overall_output_directory/quastwithRaconONT/
    mkdir -p $Overall_output_directory/quastwithRaconONT
    quast.py \
      -t ${threads} -l "FlyeAssembly,firstONTracon,secondONTracon,thirdONTracon" \
      -R $Overall_output_directory/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta \
      $reference_fasta \
      $Overall_output_directory/01_firstRacon_graphmap/assembly/firstRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/02_secondRacon_graphmap/assembly/secondRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/03_thirdRacon_graphmap/assembly/thirdRacon_NWC_2Assembly.fasta \
      -o $Overall_output_directory/quastwithRaconONT
 
 
 ##-------------------making MultiQC output
 rm $Overall_output_directory/logs_MultiQC_raconONT.txt
 
 echo $Overall_output_directory/quastwithRaconONT/report.tsv >> $Overall_output_directory/logs_MultiQC_raconONT.txt
 
 num=1

for name in first second third fourth
do 

echo $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc >> $Overall_output_directory/logs_MultiQC_raconONT.txt
num=$((num+1)) 

done ##loop for name
 
#done #loop for sample
###===========================
###-----------------
##MultiQC
###-----------------
###===========================  


#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do
name=di_K2_6h
sample=di_K2_6h
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${sample}/
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/assembly.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz

echo "========================================="
echo $sample
echo "========================================="



#done #loop for sample
 
rm -r $Overall_output_directory/MultiQCRaconONT/
mkdir -p $Overall_output_directory/MultiQCRaconONT/
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list   $Overall_output_directory/logs_MultiQC_raconONT.txt \
    -o $Overall_output_directory/MultiQCRaconONT/



###--------------on local

name=di_K2_6h
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/quastwithRaconONT/report.html ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_quast_report.html

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/MultiQCRaconONT/multiqc_report.html ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report.html



```


#### Illumina polishing freebayes 

Now we polish with the high quality Illumina reads

```{bash}


###-----------------
##Illuminamapping rawReads to circlator genome
###-----------------
sample=24758
   for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765"|tail -28)
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep ".fastq$"|sed 's/.fastq//g' )
do
##------------------define variables------------------
#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do

name=$(echo $sample)
#sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}
reference_fasta=/data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta
ont_reads=/data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq

num=5


for run in first second third fourth
   do 

#name=run
##--------------create directories----------------
rm -r $Overall_output_directory/0${num}_${run}Freebayes/
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/{rawIlluminaMapped,freebayesOuput}
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc

#mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq

# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R1_val_1.fq
# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R2_val_2.fq


#/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 index $reference_fasta

 # /home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 mem -t ${threads} $reference_fasta /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R2_val_2.fq.gz | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 

bwa index $reference_fasta

 bwa mem -t ${threads} $reference_fasta /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R2_val_2.fq.gz | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 



#bwa mem -t ${threads}  $reference_fasta /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq |samtools sort -@8 -O BAM \
#-o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 
     
#/data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/


qualimap bamqc -bam $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam \
        -outdir $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc --java-mem-size=40G
    
samtools index $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam

###-----------------
##freebayesAfterCirclator
###-----------------
echo freebayes

    SECONDS=0

    freebayes -F 0.5 --min-coverage 4 -p 1 -f \
      $reference_fasta \
      $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam > \
      $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf
      
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`

##-------   
##bcftools variant integration of Illumina data into Pacbio polished genome
##-------

    SECONDS=0
    echo "bgzip"
bgzip -c  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf > \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      echo "tabix"
tabix -p vcf  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
    echo "bcftools"
bcftools consensus -f $reference_fasta \
      -o $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`
  
  #bwa index $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta 
  
  rm $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam &
  
   ##-------------------new variable name for next polishing steps-----------
  reference_fasta=$Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta  
  num=$((num+1)) 
done
  done #sample  


##--------------indexing


#bwa index $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta  

##---------------
##PostAssemblyAnalysis
##---------------
for samplezzz in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765")
do

#samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

  echo "##=============="
  echo ${samplezzKurz}
  echo "##=============="
    #tail -3 $mainPath/03_metaphlan2/${id}_trimmed/metaphlan_profile/profiled_metagenome_${id}_trimmed.txt
    PostAssemblyAnalysis.R -i Y -a /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${samplezzz}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta \
      -F /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${samplezzz}/FAM${samplezzz}_R1_val_1.fq.gz \
      -R /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${samplezzz}/FAM${samplezzz}_R1_val_1.fq.gz \
      -o /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/PostaAssembly/FAM${samplezzz}/ \
      -t 35
   
      
    done




###--------------on local



 name=di_K2_6h
 scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/quastwithRacon_FreebayesONT/ ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/


scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/MultiQCRaconFreebayesONT/multiqc_report.html
~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report_final.html



```


##circlator

Start align with [fasta_shift](https://github.com/b-brankovics/fasta_tools).
start align only single contig genomes

```{bash}


##===========
##PROKKA annotate for fasta_shift
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765")
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep ".fastq$"|sed 's/.fastq//g' )
do
#for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#  do
  

  echo "=============="
  echo ${sample}


rm -r /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/
#mkdir -p /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${culture}_${strainzzz}

/usr/bin/perl /usr/bin/prokka --cpus 5 --outdir /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/ --locustag ${sample} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta

#prokka --outdir /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${culture}_${strainzzz} --force --usegenus --genus ${speciesss} --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes/${genomeszzz}

  done


##===========
##single conitg genome
##===========

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765" |
grep -v "24731"|
grep -v "24745"|
grep -v "24749"|
grep -v "24759"|
grep -v "24777")
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep ".fastq$"|sed 's/.fastq//g' )
do
#for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#  do
  

  echo "=============="
  echo ${sample}

seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*.fna
done

##===========
##complete but not circular genomes
##===========
grep -v "24731"
grep -v "24745"
grep -v "24749"
grep -v "24759"
grep -v "24777"

##===========
##identify dnaA
##===========

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765"|
grep -v "24731"|
grep -v "24745"|
grep -v "24749"|
grep -v "24759"|
grep -v "24777")
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep ".fastq$"|sed 's/.fastq//g' )
do
#for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#  do
  

  echo "=============="
  echo ${sample}

grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}'
done



##===========
##start align
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
  grep -v "24743"|
  grep -v "24729"|
  grep -v "24795"|
  grep -v "24794"|
  grep -v "24762"|
  grep -v "24751"|
  grep -v "24773"|
  grep -v "24765"|
  grep -v "24731"|
  grep -v "24745"|
  grep -v "24749"|
  grep -v "24759"|
  grep -v "24777"|
  grep -v "24758")
  
  for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep ".fastq$"|sed 's/.fastq//g' )
  do     
  echo "=============="
 echo ${sample}
 grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}'
 orientation=$(grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}' |cut -f 7)
 contigName=$(grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}' |cut -f 1)
 samtools faidx /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta ${contigName} > /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/bac_genome.fna
 
 rm -r /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned
 mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned
 if [ $orientation == "+" ]
 then              echo "forward oriented"
 startAlign=$(grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}' |awk -F "\t" '{print $4-5}')
 echo -e "start align at:  "${startAlign}
 
 /home/vincent/apps/fasta_tools/bin/fasta_shift -i /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/bac_genome.fna -p ${startAlign} > /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta
 
 else              
 echo "reverse oriented"
 startAlign=$(grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}' |awk -F "\t" '{print $5+5}')
 echo -e "start align at:  "${startAlign}
 /home/vincent/apps/fasta_tools/bin/fasta_shift -i /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/bac_genome.fna -p ${startAlign} > /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/tmp.fasta
 revseq -sequence /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/tmp.fasta -outseq /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta -notag
 fi
 for non_bac in $(grep ">" /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta | grep -v "${contigName}"|awk -F " " '{print $1}' |sed 's/>//g')
 do
 samtools faidx /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta ${non_bac} >> /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta
 done #non_bac
 seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta
 done
 
##=============================
#check
##=============================

 

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765"|
grep -v "24731"|
grep -v "24745"|
grep -v "24749"|
grep -v "24759"|
grep -v "24777"|
grep -v "24758")
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep ".fastq$"|sed 's/.fastq//g' )
do      echo "=============="
echo ${sample}
rm -r /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter 
perl /usr/bin/prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter --locustag ${sample} --proteins s.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta

done

##-----------------------------------------
#check
##-----------------------------------------

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765"|
grep -v "24731"|
grep -v "24745"|
grep -v "24749"|
grep -v "24759"|
grep -v "24777"|
grep -v "24758")
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep ".fastq$"|sed 's/.fastq//g' )
do      echo "=============="
echo ${sample}
grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter/*gff |awk -F "\t" '{if($3=="gene")print $0}'
seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter/*.fna

done

```




```{bash}
##===========
##circlator
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765")
do
#for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#  do
  

  echo "=============="
  echo ${sample}

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/${sample}
#mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/circlator/${sample}

mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/tmp/

cat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R1_val_1.fq.gz \
/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R2_val_2.fq.gz > \
/data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/tmp//${sample}.fq.gz


circlator all /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta /data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/tmp/${sample}.fq.gz /data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/${sample}


  done
  

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/
```


## Illumina
Illumina only based assemblies. 



```{bash}
scp -r /media/vincent/Elements/seqData/20200515_UniBe vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/

```

download genomes for polishing
```{bash}
##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================
#http://uhts-lgtf.vital-it.ch/symlink/S1_L7_R1_001_JWc1BQSlYm0n.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S1_L7_R2_001_azQERS6e6Mf3.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S2_L7_R1_001_tAyGggUA76Sv.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S2_L7_R2_001_Z7AkETEcWnrG.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S3_L7_R1_001_QG9hjiVdib1X.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S3_L7_R2_001_hhue2W6w2qqw.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S4_L7_R1_001_sCvHPBHZTe80.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S4_L7_R2_001_mlaWMoWEB7ur.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S5_L7_R1_001_jcQlllEuwmpJ.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S5_L7_R2_001_WsUA1vmuMQds.fastq.gz
rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/99_log

vim /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_download.txt

vim /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_sampleNames2change.txt

#FAM24737	S1
#FAM24738	S2
#FAM24739	S3
#FAM24740	S4
#FAM24777	S5


##--------------download special case

cd /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/

wget -i /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_download.txt

```




###strain information

get strain information

```{bash}

scp /home/vincent/Desktop/Projects/2020_strainDelineation/01_logs/isolatedStrainsRMK.csv vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/99_log/


scp -r /media/vincent/Elements/rawdata/* vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/
scp -r /media/vincent/Elements/rawdata/FAM12273* vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/




```


### renaming

```{bash}


for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

cat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_L*_R1_*.fastq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R1.fastq.gz

cat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_L*_R2_*.fastq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R2.fastq.gz


done #samplezzz


##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================

for name in $(cut -d '/' -f 5 /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_download.txt)
do

  name_01=$(echo $name |cut -d '_' -f 1)
  Lane=$(echo $name |cut -d '_' -f 2|sed 's/L//g')
  Read=$(echo $name |cut -d '_' -f 3)  
  
  Lane_new=$((Lane+2))

  newName=$(awk -F "\t" -v name="$name_01" '{if($2==name) {print $1}}' /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_sampleNames2change.txt)
  
  echo ${newName}"_"${Read}".fastq.gz"
    
  mv /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${name} /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${newName}_${Read}.fastq.gz
    
done
##-------qualtiy control

ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/|cut -d '_' -f 1|sort|uniq -c |wc -l
ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/|cut -d '_' -f 1|sort|uniq -c 



##------------------------------------
##===============================
##download old samples downloaded 20210209
##------------------------------------
##===============================


for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/|grep "_R1")
do
genomeshort=$(echo $genomesss |cut -d '_' -f 1)
echo $genomeshort
#mv /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${genomesss} /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${genomeshort}

mv /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${genomeshort}_R1.fastq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${genomeshort}/${genomeshort}_R1.fastq.gz

mv /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${genomeshort}_R2.fastq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${genomeshort}/${genomeshort}_R2.fastq.gz

done

```



###FastQC
In a first step we creat [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) reports for all the libraries. 
These are then visualized with [MultiQC](https://multiqc.info/)

```{bash fastqc}
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh
threads=8

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)


    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R1.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R2.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
    
done #samplezzz

  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/MultiQC

  
  multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/MultiQC

##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================
threads=8

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/ |grep ".fastq")
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)


    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${samplezzKurz}_R1.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${samplezzKurz}_R2.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
    
done #samplezzz

  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/MultiQC

  
  multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/MultiQC

##------------------------------------
##===============================
##download old samples downloaded 20210209
##------------------------------------
##===============================

threads=37


mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz )
do

#samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)


    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzz}/${samplezzz}_R1.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/fastQC/
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzz}/${samplezzz}_R2.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/fastQC//
    
done #samplezzz
```

### Trim-galore
In the following we clip adaptors with [trim-galore]()

```{bash trimGalore}
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh
threads=30
id=RMK202_withOld

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

  echo "##=============="
  echo ${samplezzKurz}
  echo "##=============="
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      trim_galore -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz} \
        /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R1.fastq.gz  \
        /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R2.fastq.gz 
        
  echo "##=============="
  echo ${id} "fastQC of trimmed"
  echo "##=============="
      
    mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/
      
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R1.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/
    
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R2.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/
    
        
  done 
  
python3 /home/vincent/miniconda2/bin/multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore//Multiqc




mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do


echo "/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/bamqc" >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log/multiqc_mapping_logfile.txt
   

done


  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore//Multiqc

  multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore//Multiqc


echo "/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/bamqc" >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log/multiqc_mapping_logfile.txt
   

samtools index /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam &


done

###--------------------------------multiQC

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/


  multiqc --config /archiv/logs/multiQC_config.txt --file-list /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log/multiqc_mapping_logfile.txt \
  -o archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/
##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================
threads=35
#id=RMK202_withOld

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz |grep ".fastq" )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

  echo "##=============="
  echo ${samplezzKurz}
  echo "##=============="
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      trim_galore --fastqc -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/${samplezzKurz} \
        /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${samplezzKurz}_R1.fastq.gz  \
        /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${samplezzKurz}_R2.fastq.gz 
        
  echo "##=============="
  echo ${id} "fastQC of trimmed"
  echo "##=============="
      

  done 
  
##--------------------------------------
##bring together
##--------------------------------------


for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz |grep ".fastq" )
do
samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

cp -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/ /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/
  
done

##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================

##special case

cat /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_1A3P4_S293_R1_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S301_R1_001.fastq.gz > /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S3012_R1_001.fastq.gz

cat /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_1A3P4_S293_R2_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S301_R2_001.fastq.gz > /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S3012_R2_001.fastq.gz

mkdir -p /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns
mv /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_1A3P4_S293_R1_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns/

mv /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S301_R1_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns/

mv /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_1A3P4_S293_R2_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns/

mv /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S301_R2_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns/

##redo

threads=35
#id=RMK202_withOld

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
for samplezzz in $(ls /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/ |grep "^1" )
do
samplezzz=13499

samplezzKurz=$(echo -e "FAM"${samplezzz} )

  echo "##=============="
  echo ${samplezzKurz}
  echo "##=============="
  
  

  
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      trim_galore --fastqc -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz} \
        /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/${samplezzz}/${samplezzz}*_R1*.fastq.gz  \
        /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/${samplezzz}/${samplezzz}*_R2*.fastq.gz
        
  echo "##=============="
  echo ${id} "fastQC of trimmed"
  echo "##=============="
      

  done 



##--------------------------------------
##bring together
##--------------------------------------


for samplezzz in $(ls /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/ |grep "^1" )
do
samplezzKurz=$(echo -e "FAM"${samplezzz} )
rm  -r /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}

cp /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzz}*_R1*val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz

cp /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzz}*_R2*_val_2.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz

done


##------------------------------------
##===============================
##download old samples downloaded 20210209
##------------------------------------
##===============================

threads=37

/data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzz}/${samplezzz}_R1.fastq.gz

threads=35
#id=RMK202_withOld

#mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
for samplezzKurz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/ |grep "fastqc" -i -v )
do
#samplezzz=13499

#samplezzKurz=$(echo -e "FAM"${samplezzz} )

  echo "##=============="
  echo ${samplezzKurz}
  echo "##=============="
  
  

  
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/
      mkdir -p  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/
      

       cp /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzKurz}/${samplezzKurz}_R1.fastq.gz   /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz
       
       cp /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzKurz}/${samplezzKurz}_R2.fastq.gz   /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz
       
      #trim_galore --fastqc -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/ /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzKurz}/${samplezzKurz}_R1.fastq.gz   /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzKurz}/${samplezzKurz}_R2.fastq.gz
        


  done 



```

```{bash}


scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore//Multiqc /home/vincent/Desktop/Projects/2020_strainDelineation/01_logs

```
###metaphlan
I do metaphlan to check for contamination

```{bash}


rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/115_i_24728.fasta

 rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes/202_i_24753.fasta
 rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes/202_i_24754.fasta
 rm /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${species}/01_genomes/190_i_24736.fasta

rm /data/Project/2020_StarterCultureDiversity/04_metaphlan/allStrains_metaphlan.txt
#for speciesss in $(echo "L.delbrueckii S.thermophilus")
for speciesss in $(echo "S.thermophilus")
do
rm -r /data/Project/2020_StarterCultureDiversity/04_metaphlan/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/04_metaphlan/${speciesss}
echo "-----------------"
echo ${speciesss}
#for strainzzz in $(grep "^${speciesss}" /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv |cut -f 11 |head -1)
for strainzzz in $(echo "24853 24854 24855")
do
culture=$(awk -F "\t" -v strainsss="$strainzzz" '{OFS="\t"}{if($11==strainsss)print $4}' /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed 's/RMK//g')

echo -e ${strainzzz} " from " ${culture}


mkdir -p /data/Project/2020_StarterCultureDiversity/04_metaphlan/${speciesss}/${culture}_i_${strainzzz}/{bowtie2ouput,metaphlan_profile}
 
   metaphlan2 --nproc 30 --input_type fastq <(zcat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${strainzzz}/FAM${strainzzz}_R1_val_1.fq.gz  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${strainzzz}/FAM${strainzzz}_R2_val_2.fq.gz ) \
   --bowtie2out  /data/Project/2020_StarterCultureDiversity/04_metaphlan/${speciesss}/${culture}_i_${strainzzz}//bowtie2ouput/metagenome_${strainzzz}_trimmed.bowtie2.bz2 --nproc 37 > \
   /data/Project/2020_StarterCultureDiversity/04_metaphlan/${speciesss}/${culture}_i_${strainzzz}//metaphlan_profile/profiled_metagenome_${names}_trimmed.txt
   
   grep "s__" /data/Project/2020_StarterCultureDiversity/04_metaphlan/${speciesss}/${culture}_i_${strainzzz}//metaphlan_profile/profiled_metagenome_${names}_trimmed.txt |grep "t__" -v|awk -F "[\t|]" -v culturess="$culture" -v strainzzzss="$strainzzz" -v specisss="$speciesss" '{OFS="\t"}{print culturess,strainzzzss,specisss,$7,$8}' >>  /data/Project/2020_StarterCultureDiversity/04_metaphlan/allStrains_metaphlan.txt
   
done #strainzzz
done #species




```



###spades

In this step we assemble the raw reads of the isolate with [spades](http://bioinf.spbau.ru/en/spades3.7).

```{bash spades}
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh

threads=35
num=1
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do
samplezzKurz=FAM24777
samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)
  echo "##=============="
  echo ${samplezzKurz}
  echo $num
  echo "##=============="
       mkdir -p /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly/
 
 
     /home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --pe1-1 /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz \
       --pe1-2 /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz \
       -o /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly//assembly_Spades_20200516 -t ${threads} -m 100

   num=$((num+1))


    done

#---------------
##short info
##---------------
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)


 count=$(grep ">" -c /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly//assembly_Spades_20200516/contigs.fasta)
   
     #echo "##=============="
  echo -e ${samplezzKurz} " : " ${count}
#  echo "##=============="
 
    done
##---------------
##PostAssemblyAnalysis
##---------------
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

  echo "##=============="
  echo ${samplezzKurz}
  echo "##=============="
    #tail -3 $mainPath/03_metaphlan2/${id}_trimmed/metaphlan_profile/profiled_metagenome_${id}_trimmed.txt
    PostAssemblyAnalysis.R -a /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly//assembly_Spades_20200516/scaffolds.fasta \
      -F /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz \
      -R /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz \
      -o /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly/PostaAssembly \
      -t 35
    
    done




```

###metabat

The sample 24728 has a s__Pediococcus_pentosaceus contaminatino. this is most likely from sequencing as we had samples of pediococcus on the same lane. I am going to take them out by binning. 

```{bash}
##-------
##mapping
##-------
samplezzKurz=FAM24728
threads=30
rm - /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/
mkdir -p /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/${samplezzKurz}/{binning,mapping}

gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq

gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz > /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq

bwa mem -t ${threads} /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly//assembly_Spades_20200516/contigs.fasta \
  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq | samtools sort -@${threads} -O BAM -o /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/${samplezzKurz}/mapping/${samplezzKurz}_rawIllumina_bwamem_sorted.bam


##-------
##cleaning contigs
##-------

metabat2 -t ${threads} \
  -i /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly//assembly_Spades_20200516/contigs.fasta \
  /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/${samplezzKurz}/mapping/${samplezzKurz}_rawIllumina_bwamem_sorted.bam \
  -o /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/${samplezzKurz}/binning
  
  
##-------
##BLAST QC
##-------

for genome in $(ls /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/)
  do
  
  genome_renamed=$(echo $genome |sed 's/.fa//g'  )
  
  #mkdir -p /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/${genome_renamed}_metaphlan
  
  
   infoseq /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/${samplezzKurz}/binning.1.fa
   infoseq /data/Project/2020_StarterCultureDiversity//06_Spades/metabat_contamination_Clean/${samplezzKurz}/binning.2.fa
  
 
  done

```



##Dialact strains



####download

Here, I download the spades genomes of the strains isolated out of the cheese starter cultures

```{bash}

##=============
##Download Genomes
##=============
date=20201010
downloadFormat=fna

##-----------------
##Dialact
##-----------------

   species=$(echo "Streptococcus_salivarius")
 #   species=$(echo "Lactobacillus_delbrueckii")
   echo ==========================
   echo ${species}
   echo ==========================
   
   DirectoryName=/data/Project/2020_StarterCultureDiversity//07_DialactStrains/${date}/Genomes/Dialact_02/
   #species=Streptococcus_thermophilus
   rm -r $DirectoryName/$species
   mkdir -p $DirectoryName/$species
 
 python3 /home/vincent/apps/Dialact_access.py -u vincent_somerville \
  -t $species \
  -f $downloadFormat \
  -s $DirectoryName/$species
 

```

Get strains that have been isolated from the cheese starter culture and have been sequenced. 
Noam added the list with additional infomormation. 
Here, I transfer it to the big machine. 


```{bash}

scp /home/vincent/Desktop/Projects/2020_strainDelineation/01_logs/20200113_Staemme_RMK_onlySequenced.csv vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/

```

isolate the strains from all strains
```{bash}
rm -r /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/
mkdir -p /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/{Sterm,Ldel}
for sampleszzz in $(sed '1d' /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/20200113_Staemme_RMK_onlySequenced.csv| cut -f 1 )
do
culture=$(grep "${sampleszzz}" /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/20200113_Staemme_RMK_onlySequenced.csv| cut -f 2|sed 's/ //g'|sed 's/RMK//g')
echo -e ${sampleszzz} " from : "${culture}

cat /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/Streptococcus_salivarius/FAM${sampleszzz}_*.fna > \
  /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/Sterm/${culture}_d_${sampleszzz}.fna
cat /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/Lactobacillus_delbrueckii/FAM${sampleszzz}_*.fna > \
  /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/Ldel/${culture}_d_${sampleszzz}.fna
done

find /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/Ldel/ -type f -empty -delete
find /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20200516/Genomes/Dialact_02/used/Sterm/ -type f -empty -delete


```


#Strain collection

###complete NCBI Refs
```{bash}
##=====================================================================================
##------------------------all complete streptos
##=====================================================================================
species=$(echo "Streptococcus thermophilus")
species_short=Sterm_references
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
    #pecies_short=Sterm_references

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_thermophilus_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm_names.txt

##--------------------------
##prepare list
##--------------------------
species=$(echo "Streptococcus thermophilus")
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_00.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/S-thermophilus-/S-/g'| sed 's/-STH-CIRM-/-I-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_01.txt
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt
while read -r line
do
#echo $line
echo -e ${line} |head -c 10 |head -1 |sed 's/^$//g' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 
echo -e "\n" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 

done < /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_01.txt
sed -i '/^[[:space:]]*$/d' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 

paste -d '\t' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_00.txt > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt

#-------------------- add outgroup
###CHANGED OUTGRPOUP ON 28.1.2021 TO S. VESTIBULARIS

outgroupNAME=NCTC8618

outgroupNAME=$(echo "Streptococcus vestibularis")


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >>    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
    species_short=Sterm_references

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_salivarius_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'>> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm_names.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print "Out-sali",$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print ,$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/S-thermophilus-/S-/g'| sed 's/-STH-CIRM-/-I-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt


cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
gunzip *.fna.gz


##------------change name
sed -i 's/ //g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt

for genomesss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt)
do
shortnamesss=$(grep "${genomesss}" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt | cut -f 1)
echo -e ${genomesss} " to " ${shortnamesss}

mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${genomesss}*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna

done
mv GCF_900636445.1_41965_G01_genomic.fna Out-vest.fna

##=====================================================================================
##------------------------all complete lactos
##=====================================================================================
species_short=Ldel_references
species=$(echo "Lactobacillus delbrueckii")

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
#curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "Lactobacillus delbrueckii" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" )print $0}' |cut -f 20| sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_ldel.txt
    #pecies_short=Sterm_references
    


#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}"|grep "strain=" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "L_delbrueckii_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_ldel_names.txt


##--------------------------
##prepare list
##--------------------------
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1,$2,$3,"L_delbrueckii_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_00.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1,$2,$3,"L_delbrueckii_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/L-delbrueckii-/L-/g'| sed 's/L-Lactobacillus-bulgaricus-/L-/g' | sed 's/L-Lactobacillus-delbrueckii-subsp.-/L-/g'  > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_01.txt
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt
while read -r line
do
#echo $line
echo -e ${line}  |head -1 |sed 's/^$//g' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 
echo -e "\n" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 

done < /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_01.txt
sed -i '/^[[:space:]]*$/d' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 

paste -d '\t' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_00.txt > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt
cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt

##-------------------------
##outgroup
##type strain l. delbrueckii spp. bulgaricus


outgroupNAME=$(echo "Lh12")

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "Lactobacillus helveticus"|tail -1 |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >>    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_ldel.txt

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_salivarius_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'>> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm_names.txt


#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print "Out-bulg",$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print ,$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/S-thermophilus-/S-/g'| sed 's/-STH-CIRM-/-I-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt


cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_ldel.txt
gunzip *.fna.gz




##------------change name
sed -i 's/ //g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt

for genomesss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt)
do
shortnamesss=$(grep "${genomesss}" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt | cut -f 1)
echo -e ${genomesss} " to " ${shortnamesss}

mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${genomesss}*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna

done

mv GCF_902386585.1_UHGG_MGYG-HGUT-02384_genomic.fna  Out-helv.fna

##=====================================================================================
##------------------------PROKKA annotate (for panaroo)
##=====================================================================================$

##-------------------------Sterm
shortnamesss=$(echo "Out-vest")
species_short=Sterm_references
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKK*
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_GFF
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN
for shortnamesss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt)
#for genome in $(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FNA |grep "fna")
do

#shortnamesss=$(grep "${genome}" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt | cut -f 2)
#echo -e ${genome} " to " ${shortnamesss}

locustagss=$(echo ${shortnamesss}|cut -d '_' -f 3,4,5,6|sed 's/L-delbrueckii-/L-/g' |sed 's/S-thermophilus-/S-/g'|sed 's/STH-//g'|sed 's/S-salivarius-/REF-/g'|cut -d '.' -f 1)
echo $locustagss
#rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/PROKKA/${shortnamesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}

/usr/bin/perl /usr/bin/prokka --cpus 37 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss} --force --usegenus --genus streptococcus --locustag ${locustagss} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.gff > \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_GFF/${shortnamesss}.gff

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.ffn > \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN/${shortnamesss}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.faa > \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA/${shortnamesss}.faa


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.ffn > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN/${strainzzz}.ffn

done
##--------------------------------------Ldel
species_short=Ldel_references
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKK*
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_GFF
rm -r  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN
mkdir -p ${species_short}/PROKKA_FAA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN

conda activate PROKKA

shortnamesss=Out-helv
for shortnamesss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt )
#for genome in $(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FNA |grep "fna")
do

#shortnamesss=$(grep "${genome}" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt | cut -f 2)
#echo -e ${genome} " to " ${shortnamesss}

locustagss=$(echo ${shortnamesss}|cut -d '_' -f 3,4,5,6|sed 's/L-delbrueckii-/L-/g' |sed 's/S-thermophilus-/S-/g'|sed 's/STH-//g'|sed 's/S-salivarius-/REF-/g'|cut -d '.' -f 1)
echo -e ${shortnamesss} "and locusTag " $locustagss

#done
#rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/PROKKA/${shortnamesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}

#/usr/bin/perl /usr/bin/prokka --cpus 5 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss} --force --usegenus --genus lactobacillus --locustag ${locustagss} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna

prokka --cpus 20 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss} --force  --locustag ${shortnamesss} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.gff > \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_GFF/${shortnamesss}.gff

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.ffn > \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN/${shortnamesss}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.faa > \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA/${shortnamesss}.faa


#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.ffn > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN/${strainzzz}.ffn

done

```


move local
```{bash}
mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/


scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm.txt

```

###gather own genomes

first I need to gather all necessary genomes. 

```{bash}
scp /home/vincent/Downloads/RMK_strain_stock\ -\ Sheet1\ \(1\).tsv vincent@130.223.49.145:/data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv

##------------------------------------
##ONT + Illumina genomes 
##------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/

awk -F "\t" '{if($7=="Illumina"||$7=="ONT")print $0}' /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |grep -v  "species" > /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v "24746"|
grep -v "24743"|
grep -v "24729"|
grep -v "24795"|
grep -v "24794"|
grep -v "24762"|
grep -v "24751"|
grep -v "24773"|
grep -v "24765"|
grep -v "24731"|
grep -v "24745"|
grep -v "24749"|
grep -v "24759"|
grep -v "24777"|
grep -v "24758")
do      echo "=============="
echo ${sample}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/{all,Sterm,Ldel}


speciesss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 2|head -c 1)
culturesss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 4)
methodsss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 7|head -c 1)


cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${speciesss}_${methodsss}_${culturesss}_${sample}.fasta

grep -v "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv > /data/Project/2020_StarterCultureDiversity/99_log/tmp

mv /data/Project/2020_StarterCultureDiversity/99_log/tmp /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv

done
wc -l /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv
wc -l /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv
cut -f 2 /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv|sort|uniq -c
##------------------------------------
##Illumina genomes Sterm
##------------------------------------

for sample in $(grep "thermophilus" /data/Project/2020_StarterCultureDiversity/99_log/tmp /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv |cut -f 1|cut -d ':' -f 2|grep -v "1708"|grep -v "24052"|grep -v "24054")
do      echo "=============="
echo ${sample}


speciesss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 2|head -c 1)
culturesss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 4)
methodsss=$(echo "I")


cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*_${sample}*fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${speciesss}_${methodsss}_${culturesss}_${sample}.fasta


done

##------------------------------------
##Illumina genomes Ldel
##------------------------------------

for sample in $(grep "delbrueckii" /data/Project/2020_StarterCultureDiversity/99_log/tmp /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv |cut -f 1|cut -d ':' -f 2)
do      echo "=============="
echo ${sample}
#sample=21745

speciesss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 2|head -c 1)
culturesss=$(grep "${sample}" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 4)
methodsss=$(echo "I")


#cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/*_${sample}*fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${speciesss}_${methodsss}_${culturesss}_${sample}.fasta
cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/*_${sample}*fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${sample}.fasta



done


cat /data/Project/2020_StarterCultureDiversity/06_Spades/FAM24777/assembly/assembly_Spades_20200516/contigs.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/L_I_202_24777.fasta 


grep "21745" /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv
cat /archiv/Dialact/ReferenceDatabase/20190322/Annotation/Prokka/Lactobacillus_delbrueckii/FAM21745c1_11092016.fna/PROKKA_03272019.fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/L_I_101_21745.fasta 



ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |cut -d _ -f 1|sort|uniq -c

##-------------------------------------================================================================
#rename contigs
##-------------------------------------================================================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |sed 's/.fasta//g')
do
echo "============================================"
num=1
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed/${genomesss}.fasta
for contigs in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${genomesss}.fasta |sed 's/>//g')
  do
  echo ${contigs} " to "${genomesss}"_c"$num
  
  sed -i "s/$contigs/${genomesss}_c$num/g"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed/${genomesss}.fasta
  
  #sed "s/$contigs/S_term_phage_01_contig$num/g" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta |grep ">"
  
  num=$((num+1))
  done


done #genomesss

##-------------------------------------================================================================
#remove small contigs
##-------------------------------------================================================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |sed 's/.fasta//g')
do
  echo ==========================
   echo ${genomesss}
   echo ==========================
 

  seqkit seq -m 500 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed/${genomesss}.fasta -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved/${genomesss}.fasta
 
    done


##-------------------------------------================================================================
#Final and split species
##-------------------------------------================================================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/{Sterm,Ldel}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |grep "^S" |sed 's/.fasta//g')
do
  echo ==========================
   echo ${genomesss}
   echo ==========================
 

  cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved/${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/${genomesss}.fasta
 
    done
    
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |grep "^L" |sed 's/.fasta//g')
do
  echo ==========================
   echo ${genomesss}
   echo ==========================
 

  cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved/${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Ldel/${genomesss}.fasta
 
    done
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Ldel/ |wc -l
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/ |wc -l


##=================================================================
##DO A NEW COLLECTION 20210201
##=================================================================
##------------------------------------------
##STERM
##------------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm

mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/{Sterm,Ldel}



##--------01 Ilumina genomes


for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/ |cut -d '_' -f 3|sed 's/.fasta//g')
do
echo "=========================================================================="
echo ${genomesss}
#seq_length.py /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*${genomesss}.fasta |head -1
num=1
cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

for contigs in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta |sed 's/>//g')
  do
  echo ${contigs} " to "${genomesss}"_c"$num
  
  sed -i "s/$contigs/${genomesss}_contig_$num/g"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta
  
  #sed "s/$contigs/S_term_phage_01_contig$num/g" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta |grep ">"
  
  num=$((num+1))
  done
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta |head -1
done

 ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm |wc -l

##--------02 flye genomes

for genomesss in $(echo "24750
24739
13496
24738
13499
24730
24726
24728
13498
24734
24737
24853
24854
24855
24742
13494
24758
24735
24736
24724
24748
24740")
do
echo "--------------------------------------------"
#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta



seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

done



for genomesss in $(echo "24745

")
do
echo "--------------------------------------------"
#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

shortsss=$(echo ${genomesss} |cut -d '_' -f 1)

sed "s/>contig/>${shortsss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${shortsss}.fasta



seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${shortsss}.fasta

done

###---------------------rmk genome
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/*.fna
cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_O_202_SMAG.fna  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/202SMAG.fasta

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_I_202_S50.fna  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/S50.fasta

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_I_202_S72.fna  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/S72.fasta

##------------------------------------------
##Ldel
##------------------------------------------


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel

mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/{Sterm,Ldel}



##--------01 Ilumina genomes


for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/ |cut -d '_' -f 3|sed 's/.fasta//g')
do
echo "=========================================================================="
echo ${genomesss}
#seq_length.py /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*${genomesss}.fasta |head -1
num=1
cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/*${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

for contigs in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta |sed 's/>//g')
  do
  echo ${contigs} " to "${genomesss}"_c"$num
  
  sed -i "s/${contigs}/${genomesss}_contig_${num}/g"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta
  
  #sed "s/$contigs/S_term_phage_01_contig$num/g" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta |grep ">"
  
  num=$((num+1))
  done
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta |head -1
done

 ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel |wc -l


##--------02 flye genomes
genomesss=24758
#24759
#24777

for genomesss in $(echo "24756
24786
24790")
do
echo "--------------------------------------------"
echo ${genomesss}
#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

num=1
#seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta
for contigs in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta |sed 's/>//g')
  do

 sed -i "s/${contigs}/${genomesss}_contig_${num}/g"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

num=$((num+1))

done
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

done


##--------03 new flye genomes 20210310

##-------LDEL

for genomesss in $(echo "24754 24798")
do
echo "--------------------------------------------"
echo ${genomesss}
#seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta


#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

num=1
#seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta
for contigs in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta |sed 's/>//g')
  do

 sed -i "s/${contigs}/${genomesss}_contig_${num}/g"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

num=$((num+1))

done
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

done

##-------Sterm

for genomesss in $(echo "24732 24733 24743 24746 24747 24749")
do
echo "--------------------------------------------"
echo ${genomesss}
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta |head
done

#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

num=1
#seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta
for contigs in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta |sed 's/>//g')
  do

 sed -i "s/${contigs}/${genomesss}_contig_${num}/g"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

num=$((num+1))

done
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

done


```

somehow 24855_contig_2 is redundant remove this.

```{bash}
samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855.fasta
samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855.fasta 24855_contig_2 > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855_new.fasta
samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855.fasta 24855_contig_3 >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855_new.fasta
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855_new.fasta

sed 's/24855_contig_2/24855_contig_1/g'   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855_new.fasta |sed 's/24855_contig_3/24855_contig_2/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855.fasta
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855.fasta
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855_new.fasta
```


### align contigs

Here, I use Rattag to align the illumina (draft) genome according to the synteny of the ONT genomes. 

```{bash}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/
ragtag.py correct ref.fasta query.fasta

mkdir -p aligned

###==========================================================
##start align Illumina assemblies
###==========================================================
###lin1
(echo "24732 24733 24743 24746 24747 24749")

#ragtag.py correct 24750.fasta 24732.fasta -u -t 35  --inter -o ./aligned/24732_corrected
ragtag.py correct 24750.fasta 24731.fasta -u -t 35  --inter -o ./aligned/24731_corrected
#ragtag.py correct 24750.fasta 24733.fasta -u -t 35  --inter -o ./aligned/24733_corrected
ragtag.py correct 24750.fasta 24748.fasta -u -t 35  --inter -o ./aligned/24748_corrected

###lin2
ragtag.py correct 24739.fasta 13497.fasta -u -t 35  --inter -o ./aligned/13497_corrected
ragtag.py correct 24739.fasta 13495.fasta -u -t 35  --inter -o ./aligned/13495_corrected

###lin3
ragtag.py correct 24738.fasta 24744.fasta -u -t 35  --inter -o ./aligned/24744_corrected
ragtag.py correct 24738.fasta 24745.fasta -u -t 35  --inter -o ./aligned/24745_corrected
ragtag.py correct 24738.fasta S50.fasta -u -t 35  --inter -o ./aligned/S50_corrected

###lin4
ragtag.py correct 24730.fasta 24729.fasta -u -t 35  --inter -o ./aligned/24729_corrected

###lin5
#ragtag.py correct 24730.fasta 24749.fasta -u -t 35  --inter -o ./aligned/24749_corrected
#ragtag.py correct 24730.fasta 24747.fasta -u -t 35  --inter -o ./aligned/24747_corrected


###lin6

ragtag.py correct 24726.fasta 24727.fasta -u -t 35  --inter -o ./aligned/24727_corrected
###lin7
ragtag.py correct 24734.fasta 24724.fasta -u -t 35  --inter -o ./aligned/24724_corrected
ragtag.py correct 24734.fasta 24725.fasta -u -t 35  --inter -o ./aligned/24725_corrected

###lin8
ragtag.py correct 24737.fasta 13492.fasta -u -t 35  --inter -o ./aligned/13492_corrected
ragtag.py correct 24737.fasta 13500.fasta -u -t 35  --inter -o ./aligned/13500_corrected
ragtag.py correct 24737.fasta 13493.fasta -u -t 35  --inter -o ./aligned/13493_corrected
ragtag.py correct 24742.fasta 24741.fasta -u -t 35  --inter -o ./aligned/24741_corrected
ragtag.py correct 24742.fasta 13491.fasta -u -t 35  --inter -o ./aligned/13491_corrected

###lin9
#ragtag.py correct 24742.fasta 24743.fasta -u -t 35  --inter -o ./aligned/24743_corrected


###lin10
#ragtag.py correct 24742.fasta 24746.fasta -u -t 35  --inter -o ./aligned/24746_corrected
###lin11
ragtag.py correct 24736.fasta 24758.fasta -u -t 35  --inter -o ./aligned/24758_corrected
ragtag.py correct 24736.fasta 12273.fasta -u -t 35  --inter -o ./aligned/12273_corrected
#ragtag.py correct 24736.fasta 12273.fasta -u -t 35 -o ./aligned/12273_corrected
#ragtag.py correct 24736.fasta 12273.fasta -u -t 35 -b 0 -o ./aligned/12273_corrected

###linx
ragtag.py correct 24855.fasta S72.fasta -u -t 35  --inter -o ./aligned/S72_corrected


###==========================================================
##bring all genomes together
###==========================================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned

genomesss=12273
##--------------------------Illuminas
  for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/ |grep ".fasta$"|cut -d '.' -f 1)
  do
  echo "--------------------------------------------"
  #sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta
  
 cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm//aligned/${genomesss}_corrected/*.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${genomesss}.fasta
  
num=1
for contigss in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm//aligned/${genomesss}_corrected/*.fasta|sed 's/>//g')
do

  sed -i "s/>${contigss}$/>${genomesss}_contig_${num}/g" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${genomesss}.fasta
  
  

#sed "s/>${contigss}/>${genomesss}_contig_${num}/g" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${genomesss}.fasta

num=$((num+1))
done
  

  
  #seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta
  
  done


ll /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/
find /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/ -size  0 -print -delete
ll /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/
##--------------------------ONT
genomesss=24855
for genomesss in $(echo "24750
24739
13496
24738
13499
24730
24726
24728
13498
24734
24737
24853
24854
24855
24742
13494
24735
24736
24724
24748
24740
202SMAG 24732 24733 24743 24746 24747 24749")
do
echo "--------------------------------------------"
#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${genomesss}.fasta

num=1
for contigss in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta|sed 's/>//g')
do



sed -i "s/>${contigss}$/>${genomesss}_contig_${num}/g" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${genomesss}.fasta

num=$((num+1))
done

#seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

done


ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/ |grep ".fasta$" |wc -l
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned |grep ".fasta$" |wc -l


#============================================================================
##Ldel
#============================================================================

##aligne all Illumina genomes


for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/|grep ".fasta$" |sed 's/.fasta//g'|grep "24756" -v |grep "24786" -v |grep "24790" -v )
do
echo "=========================================================================="
echo ${genomesss}
#seq_length.py /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*${genomesss}.fasta |head -1
num=1

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/
rm -r ./aligned/${genomesss}_corrected
mkdir -p ./aligned/
ragtag.py correct 24790.fasta ${genomesss}.fasta -u -t 35  --inter -o ./aligned/${genomesss}_corrected

done



##---------------------------------
##bring together
##---------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned

##--------------------------Illuminas
  for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/ |grep ".fasta$"|cut -d '.' -f 1)
  do
  echo "--------------------------------------------"
  #sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta
  
 cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel//aligned/${genomesss}_corrected/*.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta
  
num=1
for contigss in $(grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel//aligned/${genomesss}_corrected/*.fasta|sed 's/>//g')
do

  sed -i "s/>${contigss}$/>${genomesss}_contig_${num}/g" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta
  
#sed "s/>${contigss}/>${genomesss}_contig_${num}/g" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta

num=$((num+1))
done
    #seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta
    done

###ONT


for genomesss in $(echo "24756
24786
24790 24754 24798")
do
echo "--------------------------------------------"
echo ${genomesss}
#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta

num=1
#seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta
for contigs in $(grep ">"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta |sed 's/>//g')
  do

 sed -i "s/${contigs}/${genomesss}_contig_${num}/g"   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta

num=$((num+1))

done
seq_length.py  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta

done

####---------------------------------
#contaminatino removal
####---------------------------------



for genomesss in $(echo "21748
21753
21780
11077")
do
echo "--------------------------------------------"
echo ${genomesss}
#sed "s/>contig/>${genomesss}_contig/g" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

rm  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta

done

###--------------------------------
##change names with letters
###--------------------------------
species=Ldel
 for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/ |grep ".fasta$"|cut -d '.' -f 1|grep "^L" -v)
  do
  echo "--------------------------------------------"
mv  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/L${genomesss}.fasta
  
    done
species=Sterm
 for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/ |grep ".fasta$"|cut -d '.' -f 1|grep "^S" -v)
  do
  echo "--------------------------------------------"
mv  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${genomesss}.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S${genomesss}.fasta
  
    done


###--------------------------------
##count
###--------------------------------

 for species in $(echo "Sterm Ldel")
  do
  echo "--------------------------------------------"
  echo ${species}
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/ |grep ".fasta$" -c
  
    done

###--------------------------------
##write names
###--------------------------------
rm   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/names_fam_strains.txt
 for species in $(echo "Sterm Ldel")
  do
  echo "--------------------------------------------"
  echo ${species}
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/ |grep ".fasta$" |grep -v "S50"|grep -v "S72"|grep -v "MAG" |sed 's/^S/FAM/g' |sed 's/^L/FAM/g' |sed 's/.fasta//g' >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/names_fam_strains.txt
  
    done
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/names_fam_strains.txt
```


bring local

```{bash}
mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/names_fam_strains.txt ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/

```


there are still some undiscussed issues:

- The pipeline does not rearrange the contigs if they are in the wrong orientation. 
- the dnaA often is in the middle of the contig. It should therefore be cut and the contigs devided

## NCBI annotation

Here, I transfer the final genome assemblies locally to after annotate and submit them to NCBI. 

```{bash}
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Sterm/

scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm/01_all_FNA/* /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Sterm/

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Ldel/

scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel/01_all_FNA/* /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Ldel/

```

subset and remove all RMK202 genome because they have already been submitted. 

```{bash}
###---------------------------------
#split complete and non-complete genome assemblies
###---------------------------------

###ONTs

rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/{ONT,Illumina}
for geneomesss in $(sed '1d' ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal///genomeAssemblyStats.csv |awk -F "," '{if($3!="202"&& $16=="Y")print $1}')
do
echo "==================================================="
echo ${geneomesss}
cp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Ldel/${geneomesss}.fna /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/

cp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Sterm/${geneomesss}.fna /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/


done

##Ilumina

for geneomesss in $(sed '1d' ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal///genomeAssemblyStats.csv |awk -F "," '{if($3!="202"&& $16!="Y")print $1}')
do
echo "==================================================="
echo ${geneomesss}
cp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Ldel/${geneomesss}.fna /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/

cp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Sterm/${geneomesss}.fna /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/


done


ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/ |wc -l 
ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/ |wc -l 

sed '1d' ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal///genomeAssemblyStats.csv |awk -F "," '{if($3!="202")print $1}' |wc -l
```

add necessary information to contigs

```{bash}
###---------------------------------
#change chromosome names
###---------------------------------

for geneomesss in $(ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/)
do
echo "==================================================="
echo ${geneomesss}

grep ">" /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss}

seqlength.py /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss}

sed 's/_contig_1/_contig_1 [location=chromosome] [topology=circular] [completeness=complete]/g'  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss} | sed 's/_contig_2/_contig_2 [plasmid-name=unnamed1]/g' | sed 's/_contig_3/_contig_3 [plasmid-name=unnamed2]/g'  > /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss}_02 
echo "----"


mv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss}_02  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss}

seqlength.py /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss}

done

rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/*_02


##-----------------------------------
#Illumina remove short contigs
##-----------------------------------
conda activate seqkits
genomesss=24753
rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/*_02.fna
for genomesss in $(ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/ |grep "_02.fna" -v |sed 's/.fna//g')
do
  echo ==========================
   echo ${genomesss}
 #  echo ==========================
 seqlength.py /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina//${genomesss}.fna  |tail -5

  seqkit seq -m 500 /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina//${genomesss}.fna -o /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina//${genomesss}_02.fna
 
 
 mv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina//${genomesss}_02.fna /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina//${genomesss}.fna
# seqlength.py /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina//${genomesss}_02.fna  |tail -5
 
 
 
    done

    
```

after NCBI annotation

download all genomes that are annotate and labeled by NCBI. 

First you make a dataframe that assigns the donwload name to the accession and the sample name. 


```{bash}

##------------------------------------------
#create

rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_rmk202_isolates.txt
for genomesss in $(ls /home/vincent/Downloads/genome_assemblies_genome_gb/ncbi-genomes-2021-03-25 |grep "gbff" |sed  's/_genomic.gbff.gz//g' )
do

acesssionss=$(zcat /home/vincent/Downloads/genome_assemblies_genome_gb/ncbi-genomes-2021-03-25/${genomesss}* |head -1| sed 's/ \+/\t/g' |cut -f 2)

acesssionss_short=$(zcat /home/vincent/Downloads/genome_assemblies_genome_gb/ncbi-genomes-2021-03-25/${genomesss}* |head -1| sed 's/ \+/\t/g' |cut -f 2|sed 's/0100.*//g')

strain_name_short=$(grep "${acesssionss_short}" /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/genome_info_rmk202_isolates.csv |cut -f 4)

if [ $strain_name_short == "24777" ] 
then     

 strain_name_long=$(echo "L24777")
 
 elif [ $strain_name_short == "13499c1" ] 
then     

 strain_name_long=$(echo "S13499c1")
  elif [ $strain_name_short == "13499c2" ] 
then     

 strain_name_long=$(echo "S13499c2")
 else
 
 strain_name_long=$(ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Sterm/ /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Ldel/ |grep ".fna"|sed 's/.fna//g'|grep "${strain_name_short}")
 
fi

echo -e ${genomesss}"\t"${acesssionss}"\t"${strain_name_short}"\t"${strain_name_long} >> /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_rmk202_isolates.txt

done

cat /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_rmk202_isolates.txt


###--------------------
#fix 1
###--------------------

samtools faidx /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775.fna

grep ">" /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775.fna |sed 's/>//g' |grep -v "contig_30$" |grep -v "contig_52$" |grep -v "contig_94$" >  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/fix_01.txt
rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775_02.fna
for genomesss in $(cat /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/fix_01.txt )
do

samtools faidx /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775.fna ${genomesss} >> /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775_02.fna

done

mv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775_02.fna /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775.fna
```

Next I relabel and arrange the different files. 

```{bash}
###===================================================================
#RMK202 isoaltes
###===================================================================
batch=RMK202
tablesss=/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_rmk202_isolates.txt
##----------------------------GBK--------------------------------------
output=gbf
input_location=/home/vincent/Downloads/genome_assemblies_genome_gb/ncbi-genomes-2021-03-25/

rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/

for genomess in $(cat ${tablesss} |cut -f 1)
do
sampleName=$(cat ${tablesss}|grep ${genomess}|cut -f 4)

zcat ${input_location}/${genomess}* > /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/${sampleName}.${output}

done

##----------------------------FNA--------------------------------------

output=fna
input_location=/home/vincent/Downloads/genome_assemblies_genome_fasta/ncbi-genomes-2021-03-25

rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/

for genomess in $(cat ${tablesss} |cut -f 1)
do
sampleName=$(cat ${tablesss}|grep ${genomess}|cut -f 4)

zcat ${input_location}/${genomess}* > /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/${sampleName}.${output}

done

##----------------------------GFF--------------------------------------

output=gff
input_location=/home/vincent/Downloads/genome_assemblies_genome_gff/ncbi-genomes-2021-03-25

rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/

for genomess in $(cat ${tablesss} |cut -f 1)
do
sampleName=$(cat ${tablesss}|grep ${genomess}|cut -f 4)

zcat ${input_location}/${genomess}* > /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/${sampleName}.${output}

done

##----------------------------FFN--------------------------------------

output=ffn
input_location=/home/vincent/Downloads/genome_assemblies_cds_fasta/ncbi-genomes-2021-03-25

rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/

for genomess in $(cat ${tablesss} |cut -f 1)
do
sampleName=$(cat ${tablesss}|grep ${genomess}|cut -f 4)

zcat ${input_location}/${genomess}* > /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/${sampleName}.${output}

done

##----------------------------FAA--------------------------------------

output=faa
input_location=/home/vincent/Downloads/genome_assemblies_prot_fasta_(1)/ncbi-genomes-2021-03-25

rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/

for genomess in $(cat ${tablesss} |cut -f 1)
do
sampleName=$(cat ${tablesss}|grep ${genomess}|cut -f 4)

zcat ${input_location}/${genomess}* > /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/${sampleName}.${output}

done


```


Here, I download the genome annotations that are in the submission portal. 

```{bash}
##-------------------------------------------------
##ONT genomes
##-------------------------------------------------
###------------creat accession table

tablesss=/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_ONT_isolates.txt


#vim /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_ONT_downloading
###-------------download
batch=ONT
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/


for genomesss in $(sed '1d' ${tablesss} |cut -f 2 |sed 's/S/s/g'|sed 's/L/l/g')
do
downloadsss=$(grep "$genomesss" /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_ONT_downloading |grep ".gb/?format=attachment" )

wget --content-disposition https: $downloadsss
done

##---------------rename

for genomesss in $(sed '1d' ${tablesss} |cut -f 2 )
do

cp ${genomesss}_*.gb ${genomesss}.gbf 

done
rm *.gb

##--------------------genacnk 2 gff
## gff without fasta
## and fasta only file
batch=ONT
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/

for genomesss in $(sed '1d' ${tablesss} |cut -f 2  )
do
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/


bp_genbank2gff3 --split --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/  ${genomesss}.gbf 

rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/${genomesss}.fna

rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff

for contigsss in $(ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/  |cut -d '.' -f 1|sort|uniq )
do
#contig_names=$(echo $contigsss |sed 's/.fa//g')
##--------fna
samtools faidx /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/${contigsss}.fa
samtools faidx /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/${contigsss}.fa ${contigsss} >> /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/${genomesss}.fna

##--------gff without fasta
cat /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/${contigsss}.gff >> /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff

done #contigssss

#bp_genbank2gff3 --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/  ${genomesss}.gbf 

done
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/


##--------------------genacnk 2 gff (including fasta)

batch=ONT
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/

for genomesss in $(sed '1d' ${tablesss} |cut -f 2  )
do
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/


bp_genbank2gff3 --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/  ${genomesss}.gbf 

mv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/${genomesss}.gbf.gff /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/${genomesss}.gff 

done
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/
```


Next up are the Illumina assemblies

```{bash}
##-------------------------------------------------
##Illumina genomes
##-------------------------------------------------


tablesss=/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_Illumina_isolates.txt

###-------------download
batch=Illumina
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/






for genomesss in $(sed '1d' ${tablesss} |cut -f 2 |sed 's/S/s/g'|sed 's/L/l/g')
do
downloadsss=$(grep "$genomesss" /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_Illumina_downloading |grep ".gb/?format=attachment" )

wget --content-disposition https: $downloadsss
done

##---------------rename & include accession

for genomesss in $(sed '1d' ${tablesss} |cut -f 2 )
do

#cp ${genomesss}.*.gb ${genomesss}.gbf 
sed "s/ACCESSION/ACCESSION   ${genomesss}/g" ${genomesss}.*.gb > ${genomesss}.gbf 
done
rm *.gb


##--------------------genacnk 2 gff
batch=Illumina
tablesss=/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_Illumina_isolates.txt


rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/

for genomesss in $(sed '1d' ${tablesss} |cut -f 2 |head -1 )
do
echo $genomesss
java -jar /home/vincent/apps/jvarkit/dist/gb2gff.jar  ~/Downloads/test_gb

done

wget -O - -q  "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&id=AF338247.1,D38149.1&rettype=gbwithparts&retmode=xml&api_key=62b713e0cd85e6ac79699ecdfa72e85af009"  |java -jar /home/vincent/apps/jvarkit/dist/gb2gff.jar  
##--------------------genacnk 2 gff
## gff without fasta
## and fasta only file
batch=Illumina
tablesss=/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_Illumina_isolates.txt


rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/

for genomesss in $(sed '1d' ${tablesss} |cut -f 2  )
do
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/


bp_genbank2gff3 --split --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/  ${genomesss}.gbf 


#bp_genbank2gff3 --nolump --typesource LOCUS --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/  ${genomesss}.gbf 


rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/${genomesss}.fna

rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff

for contigsss in $(ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/  |cut -d '.' -f 1|sort|uniq )
do
#contig_names=$(echo $contigsss |sed 's/.fa//g')
##--------fna
samtools faidx /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/${contigsss}.fa
samtools faidx /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/${contigsss}.fa ${contigsss} >> /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/${genomesss}.fna

##--------gff without fasta
cat /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/${contigsss}.gff >> /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff

done #contigssss

#bp_genbank2gff3 --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/  ${genomesss}.gbf 

done
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/


##--------------------genacnk 2 gff (including fasta)

batch=Illumina
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/

for genomesss in $(sed '1d' ${tablesss} |cut -f 2  )
do
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/


bp_genbank2gff3 --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/  ${genomesss}.gbf

mv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/${genomesss}.gbf.gff /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/${genomesss}.gff 

done
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/

##---------------------gff without fasta

batch=Illumina
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
for genomesss in $(sed '1d' ${tablesss} |cut -f 2  )
do
grep "##FASTA" -B 1000000 /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/${genomesss}.gff > /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff 
rowss=$(grep "##FASTA" -n  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff |cut -d ":" -f 1)
grep "##FASTA" -n  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff
sed -i "${rowss}d" /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff 
grep "##FASTA" -n  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff

done
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/

##---------------------download acc file to rename contig names



##---------------------fasta



scp -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/


```


## Annotations

###PROKKA of strains

```{bash}

conda activate PROKKA
species=Sterm
strainzzz=S24855
export PATH=/home/vincent/apps/ncbi-blast-2.10.0+/bin/:$PATH
speciesss=$(echo "S.thermophilus")
genus=Streptococcus
echo ${speciesss}
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/
#rm -r /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/
for strainzzz in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/ |grep "fasta$" |sed 's/.fasta//g')
do
#culture=$(awk -F "\t" -v strainsss="$strainzzz" '{OFS="\t"}{if($11==strainsss)print $4}' /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed 's/RMK//g')
#ONLYsTRAIN=$(echo ${strainzzz} |cut -d _ -f 4)
echo -e ${strainzzz} 

#genomeszzz=$(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes/ |grep  "_${strainzzz}" )
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${strainzzz}*
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/

#prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${strainzzz}/ --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${strainzzz}.fasta
conda activate PROKKA
prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}// --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/${strainzzz}.fasta

done #strainzzz
##-----------------------
##l.delbrueckii
##-----------------------

conda activate PROKKA

strainzzz=24855
export PATH=/home/vincent/apps/ncbi-blast-2.10.0+/bin/:$PATH
speciesss=$(echo "L.delbrueckii")
species=Ldel
genus=Streptococcus
echo ${speciesss}
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/
#rm -r /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/

for strainzzz in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/ |grep "fasta$" |sed 's/.fasta//g')
do
#culture=$(awk -F "\t" -v strainsss="$strainzzz" '{OFS="\t"}{if($11==strainsss)print $4}' /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed 's/RMK//g')
#ONLYsTRAIN=$(echo ${strainzzz} |cut -d _ -f 4)
echo -e ${strainzzz} 

#genomeszzz=$(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes/ |grep  "_${strainzzz}" )
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${strainzzz}*
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/

#prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${strainzzz}/ --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${strainzzz}.fasta

prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}// --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/${strainzzz}.fasta

done #strainzzz

##-------------------clean up
species=Ldel
for strainzzz in $(echo "L21748
L21753
L21780
L11077
L11063")
do
echo -e ${strainzzz} 
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}// 
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection//${species}/210315_final_startAligned/${strainzzz}.fasta
#head /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/${strainzzz}

done #strainzzz


###================================================
##create FAA and FNA folders
###================================================

for species in $(echo "Ldel Sterm")
#for species in $(echo "Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all*
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/{01_all_FAA,01_all_FNA,01_all_GFF,01_all_FFN}

for strainzzz in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection//${species}/210315_final_startAligned/ |grep ".fasta$"|sed 's/.fasta//g')
do

#culture=$(awk -F "\t" -v strainsss="$strainzzz" '{OFS="\t"}{if($11==strainsss)print $4}' /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed 's/RMK//g')
ONLYsTRAIN=$(echo ${strainzzz} |cut -d _ -f 3)
echo -e ${strainzzz} 

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/${strainzzz}.fna

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.ffn > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/${strainzzz}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.faa > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/${strainzzz}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.gff > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/${strainzzz}.gff

done #strainzzz
done #species

##--------------------
##check DNAa
##--------------------
species=Sterm
species=Ldel
for strainzzz in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection//${species}/210315_final_startAligned/ |grep ".fasta$"|sed 's/.fasta//g')
do
#culture=$(awk -F "\t" -v strainsss="$strainzzz" '{OFS="\t"}{if($11==strainsss)print $4}' /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed 's/RMK//g')
#ONLYsTRAIN=$(echo ${strainzzz} |cut -d _ -f 3)
#echo -e ${strainzzz} 

#seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${strainzzz}/PROKKA*.fna |head -1

grep "dnaA" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.gff |head -1 |cut -d ':' -f 2
#grep "dnaA" i /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${strainzzz}//PROKKA*.gff |head -1

done #strainzzz



```

###Virsorter

```{bash}

conda activate vs2

#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter

for species in $(echo "Ldel Sterm" )
do
for sample in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ |grep ".fna$"|sed 's/.fna//g'|grep "24756" -A 100)
do
echo "========================================="
echo $sample
echo "========================================="

leterzzz=$(echo $species |head -c 1)
sampsss=$(echo ${sample}|sed "s/${leterzzz}//g")


  rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${species}/${sampsss}
  mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${species}/${sampsss}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${species}/${sampsss}

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/${sample}.fna . 

virsorter run -w /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${species}/${sampsss}/ -i ${sample}.fna -j 5 all

#grep -v "^#" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${sample}/VIRSorter_global-phage-signal.csv 
 
 
done #samples
done #species




```

### BUSCO

```{bash}

speciesss=Sterm
genemess=S24855

grep "${genemess}" -v /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis.txt > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis_tmp.txt
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis_tmp.txt > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis.txt
##-----------------------------------
##script
##-----------------------------------
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis.txt

for speciesss in $(echo "Sterm Ldel")
do
echo "##------------------------"
echo ${speciesss}
echo "##------------------------"

for genemess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}/01_all_FAA/ |sed 's/.faa//g')
  do
  #genemess=L_delbrueckii_RMK202_mag.faa
  #genomesssss=$(echo ${genemess}|cut -d '.' -f 1)
  echo $genemess
  
  rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/${genemess}

  mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/${genemess}
  cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/
   export BUSCO_CONFIG_FILE=/home/vincent/apps/busco_v4/busco/config/myconfig.ini
  python3.6 /home/vincent/apps/busco_v4/busco/bin/busco -f -m prot -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}/01_all_FAA/${genemess}.faa -o ${genemess} -l lactobacillales -c 10
  #-i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FAA/${genemess}.faa -o ${genemess} -l /home/vincent/apps/busco/lactobacillales_odb9 -m prot

  #grep -v "^#" run_${genomesssss}/short_summary_${genomesssss}.txt| grep "C:" -A 7 
  all_INFO=$(grep -v "^#" ${genemess}/run_lactobacillales_odb10/short_summary.txt| grep "C:" | sed "s/^[ \t]*//"  )
  complet=$(echo ${all_INFO} | awk -F "[,[]" '{print $1}'|sed 's/%//g'|sed 's/C://g' )
  single=$(echo ${all_INFO} | awk -F "[,[]" '{print $2}'|sed 's/%//g'|sed 's/S://g')
  dupli=$(echo ${all_INFO} | awk -F "[,[]" '{print $3}'|sed 's/%]//g'|sed 's/D://g')
  fragm=$(echo ${all_INFO} | awk -F "[,[]" '{print $4}'|sed 's/%//g'|sed 's/F://g')
  Missi=$(echo ${all_INFO} | awk -F "[,[]" '{print $5}'|sed 's/%//g'|sed 's/M://g')
  numbi=$(echo ${all_INFO} | awk -F "[,[]" '{print $6}'|sed 's/n://g')
  
 echo -e ${speciesss}"\t"${genemess}"\t"${complet}"\t"${single}"\t"${dupli}"\t"${fragm}"\t"${Missi}"\t"${numbi}  \
 >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis.txt

  done
done ## species

#sed -i  's/Streptococcus_thermophilus_RMK202_pgap/S_thermophilus_mag_rmk202/g' /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt
#sed -i  's/Lactobacillus_delbrueckii_RMK202_pgap/L_delbrueckii_mag_rmk202/g' /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt

```


#### genome stats

```{bash}

##---------------
##genome
##---------------

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/all_assembly_analysis.txt
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats
for species in $(echo "Sterm Ldel")
do
for id in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ |sed 's/.fna//g')
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
# /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/
 
   contiNumber=$(grep ">" -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA//${id}.fna) 
   GenomeLength=$(grep -v "^>" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA//${id}.fna  |wc -c)
   
   # contiNumber=$(grep ">" -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta) 
   #GenomeLength=$(grep -v "^>" /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta  |wc -c)
    
 echo -e ${id}"\t"${contiNumber}"\t"${GenomeLength}  >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/all_assembly_analysis.txt

#"\t"${single}"\t"${dupli}"\t"${fragm}"\t"${Missi}"\t"${numbi}  \
    done
done #species


```


####mapping stats

```{bash}

 
for species in $(echo "Sterm Ldel")
do
for id in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ |grep ".fna$" |sed 's/.fna//g'|grep "MAG" -v|grep "S50" -v|grep "S72" -v)
  do

#samplezzKurz=$(echo $id |cut -d '_' -f 4)
samplezzKurz=$(echo $id |sed 's/^L//g'|sed 's/^S//g')
  echo ==========================
   echo ${id}
   echo ==========================


 rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}
  
  
  bwa index /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA//${id}.fna
  
  bwa mem -t 32 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA//${id}.fna \
  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${samplezzKurz}/FAM${samplezzKurz}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${samplezzKurz}/FAM${samplezzKurz}_R2_val_2.fq.gz | samtools sort -@32 -O BAM -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}/${id}_rawIllumina_bwamem_sorted.bam - 

##-----Qualimap

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc

qualimap bamqc -bam /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}/${id}_rawIllumina_bwamem_sorted.bam -outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc --java-mem-size=80G

    
    done
done #species

##---------------------
##get data
##---------------------

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_genome_analysis.txt
for species in $(echo "Sterm Ldel")
do
for id in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ |grep ".fna$" |sed 's/.fna//g')
  do

#GCss=$(grep "GC percentage" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt| sed 's/GC percentage = //g'| sed "s/^[ \t]*//"|sed 's/%//g'  )

coveragesss=$(grep "mean coverageData = " /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt| sed 's/mean coverageData = //g'| sed "s/^[ \t]*//"|sed 's/X//g'  )

#readsss=$(grep "number of reads = " /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt |awk -F "number of reads = " '{print $2}'|sed 's/,//g')

#mappedReadsss=$(grep "number of mapped reads = " /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt |awk -F "number of mapped reads = " '{print $2}'|awk -F " " '{print $1}'|sed 's/,//g')

percentmappedReadsss=$(grep "number of mapped reads = " /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt |awk -F "number of mapped reads = " '{print $2}'|awk -F " " '{print $2}'|sed 's/,//g'|sed 's/(//g'|sed 's/%)//g')

GCss=$(infoseq /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA//${id}.fna -noheading -nodatabase -nolength -nousa -nohead -noacc -notype -nodesc -delimiter '_' | awk -F " " '{sum+=$2} END {print sum/NR}')

#grep -A 10000 ">>>>>>> Coverage per contig" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt| grep -v ">>>>>>> Coverage per contig"| sed "s/^[ \t]*//" |sed -r '/^\s*$/d'|cut -f 1,4,5 |awk -F "\t" -v SPECIES="$names" '{OFS="\t"}{print SPECIES,$0}' |head
 
 
done
done #species


```

#### bring together
```{bash}

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal
cp all_assembly_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
cp all_genome_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
cp all_annotation_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
cp all_annotation_transposases_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/


mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
scp  -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/ ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/

```

```{r}

# all_sterm_strain_info_ZOOMED <- read_csv("Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) %>% select(c(strain,Cheese,starterCulture,species))
# 
# all_sterm_strain_info_ZOOMED$short <- plyr::revalue(all_sterm_strain_info_ZOOMED$species, c("L.delbrueckii"="L","S.thermophilus"="S"))
# all_sterm_strain_info_ZOOMED$name <- paste0(all_sterm_strain_info_ZOOMED$short,all_sterm_strain_info_ZOOMED$strain)
# all_sterm_strain_info_ZOOMED_final <- all_sterm_strain_info_ZOOMED %>% select(name,Cheese,starterCulture)
# 
# write.csv(all_sterm_strain_info_ZOOMED_final,file="~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/origin_genomes_all.csv",quote = FALSE,row.names = FALSE)



####--------------------------------------
library(readr)
library(plyr)
library(tidyverse)
origin_genomes_all <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/origin_genomes_all.csv",  col_types = cols(starterCulture = col_character()))

ONT_assembly_stats <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/ONT_assembly_stats.txt", "\t", escape_double = FALSE, trim_ws = TRUE) %>% filter(largestContigSize>1700000)
genome_stats <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal//all_assembly_analysis.txt", "\t", escape_double = FALSE, col_names = c("strain","contigNumber","genomeSize"),  trim_ws = TRUE)
Genes_stats <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal//all_annotation_analysis.txt", "\t", escape_double = FALSE, col_names = c("strain","genes","tRNA","rRNA","transposons"),  trim_ws = TRUE)
BUSCO_stats <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal//all_analysis.txt", "\t", escape_double = FALSE, col_names = c("species","strain","completness","singleCopy","duplicated","fragmentation","Missing","numberGenesTested"),  trim_ws = TRUE) %>% filter(strain!="L11063")
mapping_stats <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal//all_genome_analysis.txt", "\t", escape_double = FALSE, col_names =c("strain","GC","IlluminaCoverage","percentMapped"),  trim_ws = TRUE)



tmp <- merge(BUSCO_stats,genome_stats,by="strain",all.x=TRUE) %>% as.tibble()
table(tmp$species)
tmp <- merge(tmp,Genes_stats,by="strain",all.x=TRUE) %>% as.tibble()
table(tmp$species)
tmp <- merge(tmp,origin_genomes_all,by.x="strain",by.y = "name",all.x=TRUE) %>% as.tibble()
table(tmp$species)

GenomeAssemblyStats_tmp <- merge(tmp,mapping_stats,by="strain")

# GenomeAssemblyStats_tmp$strain_short <- str_split_fixed(GenomeAssemblyStats_tmp$strain,"_",4)[,4]

GenomeAssemblyStats <- merge(GenomeAssemblyStats_tmp,ONT_assembly_stats,by.x="strain",by.y="sample",all.x=TRUE)
GenomeAssemblyStats$CIRCnUM <- revalue(GenomeAssemblyStats$Circularity, c("Y"="1","N"="0")) %>% as.numeric()
# GenomeAssemblyStats$Sequencing <- ifelse(is.na(GenomeAssemblyStats$Circularity),"Illumina",ifelse(GenomeAssemblyStats$Circularity=="Y","ONT","Illumina"))
GenomeAssemblyStats$Sequencing <- ifelse(is.na(GenomeAssemblyStats$Circularity),"Illumina","ONT")

GenomeAssemblyStats$additional_circular_contigs <- GenomeAssemblyStats$circular_contigs-GenomeAssemblyStats$CIRCnUM
GenomeAssemblyStats <- GenomeAssemblyStats %>%  select(-CIRCnUM) %>%  select(-circular_contigs)
# GenomeAssemblyStats$starterCulture <- str_split_fixed(GenomeAssemblyStats$strain,"_",4)[,3]
# GenomeAssemblyStats <- GenomeAssemblyStats %>%  select(-strain)%>% dplyr::rename(strain="strain_short")


GenomeAssemblyStats$Cheese <- revalue(GenomeAssemblyStats$starterCulture,c("101"="emmentaler","105"="emmentaler","115"="emmentaler","124"="emmentaler","150"="emmentaler","190"="emmentaler","202"="gruyere","203"="gruyere","280"="gruyere","302"="sbrinz","305"="sbrinz"))


## add lineage information

##--------------------------------
##ad poppunk clustering
##--------------------------------
Ldel_clusters_RMKS <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_ldel.csv", col_names = c("strain","lineage"), col_types = cols(lineage = col_character(),  strain = col_character()), skip = 1) %>%  replace(is.na(.), "Unknown") %>% mutate(lineage=as.character(as.numeric(lineage)+12))
Sterm_clusters_RMKS <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv", col_names = c("strain","lineage"), col_types = cols(lineage = col_character(),  strain = col_character()), skip = 1)%>%replace(is.na(.), "Unknown") 
both_clusters_RMKS <- rbind(Sterm_clusters_RMKS,Ldel_clusters_RMKS)
table(Ldel_clusters_RMKS$lineage)



dim(GenomeAssemblyStats)
GenomeAssemblyStats_tmp <- merge(GenomeAssemblyStats,both_clusters_RMKS,by="strain")
dim(GenomeAssemblyStats_tmp)

GenomeAssemblyStats_tmp$Ref <- ifelse(GenomeAssemblyStats_tmp$strain %in% c("S24736","S24735","S24734","S24750","S24739","S24745","S24728","S24742","S24746","S24749","S24730","S24743","L24790","L24793"),"Ref","no")
table(GenomeAssemblyStats_tmp$Ref)

GenomeAssemblyStats <- GenomeAssemblyStats_tmp
##------------------------
##remove 
##------------------------

 GenomeAssemblyStats %>% filter(completness<95|duplicated>5)
GenomeAssemblyStats_cleaned <- GenomeAssemblyStats %>% filter(completness>95&duplicated<5)
GenomeAssemblyStats_cleaned %>% filter(completness<95|duplicated>5)
dput(colnames(GenomeAssemblyStats_cleaned))

write.csv(GenomeAssemblyStats_cleaned,file="~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal///genomeAssemblyStats_long.csv",quote = FALSE,row.names = FALSE)


GenomeAssemblyStats_cleaned <- GenomeAssemblyStats_cleaned %>% select(c("strain","Cheese" ,"starterCulture","species","lineage","Ref", 
"Sequencing", "completness","contigNumber", 
"genomeSize", "genes", "tRNA", "rRNA", "transposons", "GC",  "Circularity", "additional_circular_contigs"))

GenomeAssemblyStats <- GenomeAssemblyStats_cleaned

GenomeAssemblyStats %>% filter(Ref=="Ref")

##----------------------cleaning--------------------------------
table(GenomeAssemblyStats$strain)
# GenomeAssemblyStats$species <- revalue(GenomeAssemblyStats$species, c("Sterm"="S.thermophilus","Ldel"="L.delbrueckii"))

# GenomeAssemblyStats$strain <- revalue(GenomeAssemblyStats$strain, c("S50"="mst1","S72"="mst2","L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4","L_delbrueckii_mag_rmk202"="Ldel_mag_202","S_thermophilus_mag_rmk202"="Sterm_mag_202"))

# GenomeAssemblyStats <- GenomeAssemblyStats[-which(GenomeAssemblyStats$strain=="L39"),]
colnames(GenomeAssemblyStats)

# GenomeAssemblyStats_final <- GenomeAssemblyStats %>% select(species,strain,contigNumber,genomeSize,GC,IlluminaCoverage,percentMapped,genes,tRNA,rRNA,transposons,completness)
GenomeAssemblyStats_final <- GenomeAssemblyStats 

##----------------------write table--------------------------------
  GenomeAssemblyStats_final %>% filter(species=="S.thermophilus")
write.csv(GenomeAssemblyStats_final,file="~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal///genomeAssemblyStats.csv",quote = FALSE,row.names = FALSE)
# 
# write.xlsx(GenomeAssemblyStats_final,file="Users//Desktop/presenations/my_presentations/20191007_groupmeeting/figures/genomeAssemblyStats.xls", sheetName = "Sheet1",
#   col.names = TRUE, row.names = FALSE, append = FALSE)

```


####look at circlar plasmids

The following genomes are circlar
S13494
S13496
S24736
S24737
S24739
S24745
S24854
S24855


```{bash}
mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/cirlar_plasmids
cd  /data/Project/2020_StarterCultureDiversity/09_flye

##--------------------S13494
cat assembly/13494/30-contigger/contigs_stats.txt > cirlar_plasmids/13494_contigs_stats.txt
cat assembly/13494/assembly_graph.gfa > cirlar_plasmids/13494_assembly_graph.gfa

##--------------------S13496
cat assembly/13496//30-contigger/contigs_stats.txt > cirlar_plasmids/13496_contigs_stats.txt
cat assembly/13496/assembly_graph.gfa > cirlar_plasmids/13496_assembly_graph.gfa

##--------------------S24736

cat assembly/24736//30-contigger/contigs_stats.txt > cirlar_plasmids/24736_contigs_stats.txt
cat assembly/24736/assembly_graph.gfa > cirlar_plasmids/24736_assembly_graph.gfa

##--------------------S24737

cat assembly/24737//30-contigger/contigs_stats.txt > cirlar_plasmids/24737_contigs_stats.txt
cat assembly/24737/assembly_graph.gfa > cirlar_plasmids/24737_assembly_graph.gfa

##--------------------S24739

cat assembly/24739//30-contigger/contigs_stats.txt > cirlar_plasmids/24739_contigs_stats.txt
cat assembly/24739/assembly_graph.gfa > cirlar_plasmids/24739_assembly_graph.gfa

##--------------------S24745

cat assembly/24745//30-contigger/contigs_stats.txt > cirlar_plasmids/24745_contigs_stats.txt
cat assembly/24745/assembly_graph.gfa > cirlar_plasmids/24745_assembly_graph.gfa
##!!!!!!no plasmid
rm cirlar_plasmids/24745* 
##--------------------S24854

cat assembly/24854//30-contigger/contigs_stats.txt > cirlar_plasmids/24854_contigs_stats.txt
cat assembly/24854/assembly_graph.gfa > cirlar_plasmids/24854_assembly_graph.gfa

##--------------------S24855

cat assembly/24855//30-contigger/contigs_stats.txt > cirlar_plasmids/24855_contigs_stats.txt
cat assembly/24855/assembly_graph.gfa > cirlar_plasmids/24855_assembly_graph.gfa



```
move local to look at with bandage

```{bash}
scp  -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/09_flye/cirlar_plasmids ~/Desktop/Projects/2020_StarterCultureDiversity/09_flye/

```


#Genome analysis

###Restriction modification

```{bash}
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/


grep "restriction" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff

grep "restriction" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff |grep "hsdR" -v |grep "sau3AIR" -v 

hsdR #Type-1 restriction 
sau3AIR # type II
hsdM # type I
mboIR  # type II

grep "HsdM" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff
grep "Hsds" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff

grep "HsdM" -A 5 -B 5 -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff

```
## special annotations

### EPS

maybe I should make genus specific or hmm databases (https://github.com/tseemann/prokka).

https://www.nature.com/articles/srep04974/figures/4

```{bash}

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/ |grep ".fna$" |head -1)
do
grep "=eps" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/S_I_190_24758.fna |awk -F ";gene=" '{print $2}' |cut -d ';' -f 1 |sort|uniq -c


done
```


### RGP

### Bacteriocins

### quorum sensing (ComABCDE)

### naturally competence system

### autolysis

### CRISPR-system

### plasmid

### prophage

with Phaster. I use the [API tool](https://github.com/boulund/phaster_scripts).

```{bash}

genomesss=L_I_101_10973
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/ |grep ".fna$"|sed 's/.fna//g' |head -1)
do
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER

#grep "=eps" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/S_I_190_24758.fna |awk -F ";gene=" '{print $2}' |cut -d ';' -f 1 |sort|uniq -c
echo $genomesss
#/home/vincent/apps/phaster_scripts/phaster.py  --contigs --fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/${genomesss}.fna --database /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt

wget --post-file="/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/${genomesss}.fna" "http://phaster.ca/phaster_api?contigs=1" -O /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt


wget "http://phaster.ca/phaster_api?acc=ZZ_0863a29868" -O /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt


done


wget "http://phaster.ca/phaster_api?acc=ZZ_0863a29868" -O /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt

mkdir -p  ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES


#scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/S_O_124_24730/PROKKA_07282020.gbf ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/PROKKA_07282020.gbk


```

In the following I will run the phaster online. therefore I have to transfer all fna files first to local

```{bash}

```


### codon usage (avoidance between species)

check this out
```{bash}

###------------------------------------
##calculate codon usage
###------------------------------------

for species in $(echo "Ldel Sterm")
do

#for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|cut -d '_' -f 1|sed 's/.fna//g')
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/ |grep ".ffn$" |sed 's/.ffn//g')
do

sed 's/ .*//g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.ffn |sed 's/ //g' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/tmp.ffn

grep "^>" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.ffn|  awk -F " " '{OFS=" "}{print $2,$3,$4,$5,$6,$7}' |sed 's/[[:space:]]*$//'  > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/geneNamesfromFFN.txt

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/${species}/${genomess}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/${species}/${genomess}

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/${species}/${genomess}
fascodon  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/tmp.ffn |grep ">" |  sed 's/>//g' |sed 's/  / /g' |sed 's/ /\t/g'|sed 's/freq_.\{1\}_.\{3\}://g' |awk -F "\t" -v SPECIES="$species" -v GENOMESS="$genomess" '{OFS="\t"}{print SPECIES,GENOMESS,$0}' > ${genomess}_codonUsage.txt

paste /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/geneNamesfromFFN.txt ${genomess}_codonUsage.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/codonUsageAllstrains.txt

rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/tmp.ffn &
done #genomes
done # species
```



### transporters

### amino acid synthesis

### pseudogenes

In order to identify the pseudogene count we first need to download the pgap annotations from the opengenomebrowser (OGB). 

```{bash}
##=====================================
#get list of FAM genomes
##====================================

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log_annoation.txt
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/
for species in $(echo "Ldel Sterm")
do

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/ |grep ".faa$"|grep "MAG" -v |grep "S50" -v |grep "S72" -v  |sed 's/.faa//g' |sed 's/^S//g'|sed 's/^L//g'|sed "s/^/\\'FAM/g" |sed "s/$/\\'/g" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log_annoation.txt

done


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log_annoation.txt| tr '\n' ','

```


```{bash}

scp /home/vincent/test/download_ogb_gbk.py vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/

```

donwload the genome after filing in the FAM names. 

```{bash}
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK

sed  "s/FAM24855/FAM24855','FAM12273/g"  ../download_ogb_gbk.py  > ../download_ogb_gbk_new.py

#,'FAM24728'

python ../download_ogb_gbk_new.py


##--------------
------------------
##move into correct location and rename
##--------------------------------

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK
for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/ |grep ".faa$"|grep "MAG" -v |grep "S50" -v |grep "S72" -v  |sed 's/.faa//g' |sed 's/^S//g'|sed 's/^L//g')
do

echo ${genomesss}
firstCharacter=${species::1}

mv FAM${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/${firstCharacter}${genomesss}.gbk


done #genomesss

done


###-----Lhelv

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK


sed  "s/FAM10973.*FAM24855/FAM11051','FAM11961','FAM21456','FAM21462','FAM21463/g"  ../download_ogb_gbk.py > ../download_ogb_gbk_lhelv.py

python ../download_ogb_gbk_lhelv.py
species=Lhelv
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}
for genomesss in $(echo "FAM11051 FAM11961 FAM21456 FAM21462 FAM21463")
do

echo ${genomesss}
neuNAme=$(echo ${genomesss}|sed 's/FAM/Lh/g')
mv ${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/${neuNAme}.gbk
done 

#==============================================
#GFF
#==============================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF

#python ../download_ogb_gff.py

sed  "s/FAM24855/FAM24855','FAM12273/g"  ../download_ogb_gff.py  > ../download_ogb_gff_new.py
#cp  ../download_ogb_gbk.py   ../download_ogb_gff.py

#,'FAM24728'

python ../download_ogb_gff_new.py



for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${species}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/ |grep ".faa$"|grep "MAG" -v |grep "S50" -v |grep "S72" -v  |sed 's/.faa//g' |sed 's/^S//g'|sed 's/^L//g')
do

echo ${genomesss}
firstCharacter=${species::1}

mv FAM${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${species}/${firstCharacter}${genomesss}.gff




done #genomesss

done

###-----Lhelv

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF

#cp ../download_ogb_gbk.py ../download_ogb_gff.py

sed  "s/FAM10973.*FAM24855/FAM11051','FAM11961','FAM21456','FAM21462','FAM21463/g"  ../download_ogb_gff.py > ../download_ogb_gff_lhelv.py

python ../download_ogb_gff_lhelv.py
species=Lhelv
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${species}
for genomesss in $(echo "FAM11051 FAM11961 FAM21456 FAM21462 FAM21463")
do

echo ${genomesss}
neuNAme=$(echo ${genomesss}|sed 's/FAM/Lh/g')
mv ${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${species}/${neuNAme}.gff
done 
#==============================================
#ffn
#==============================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN

sed  "s/FAM24855/FAM24855','FAM12273/g"  ../download_ogb_ffn.py  > ../download_ogb_ffn_new.py

python ../download_ogb_ffn_new.py


for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${species}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/ |grep ".faa$"|grep "MAG" -v |grep "S50" -v |grep "S72" -v  |sed 's/.faa//g' |sed 's/^S//g'|sed 's/^L//g')
do

echo ${genomesss}

firstCharacter=${species::1}

mv FAM${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${species}/${firstCharacter}${genomesss}.ffn



done #genomesss

done


###-----Lhelv

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN


sed  "s/FAM10973.*FAM24855/FAM11051','FAM11961','FAM21456','FAM21462','FAM21463/g"  ../download_ogb_ffn.py > ../download_ogb_ffn_lhelv.py

python ../download_ogb_ffn_lhelv.py
species=Lhelv
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${species}
for genomesss in $(echo "FAM11051 FAM11961 FAM21456 FAM21462 FAM21463")
do

echo ${genomesss}
neuNAme=$(echo ${genomesss}|sed 's/FAM/Lh/g')
mv ${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${species}/${neuNAme}.ffn
done 

#==============================================
#fna
#==============================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA

#python ../download_ogb_fna.py

sed  "s/FAM24855/FAM24855','FAM12273/g"  ../download_ogb_fna.py  > ../download_ogb_fna_new.py

python ../download_ogb_fna_new.py



for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/ |grep ".faa$"|grep "MAG" -v |grep "S50" -v |grep "S72" -v  |sed 's/.faa//g' |sed 's/^S//g'|sed 's/^L//g')
do

echo ${genomesss}

firstCharacter=${species::1}

mv FAM${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}/${firstCharacter}${genomesss}.fna


sed -i 's/>lcl|/>/g'  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}/${firstCharacter}${genomesss}.fna

done #genomesss

done


###-----Lhelv

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA


sed  "s/FAM10973.*FAM24855/FAM11051','FAM11961','FAM21456','FAM21462','FAM21463/g"  ../download_ogb_fna.py > ../download_ogb_fna_lhelv.py

python ../download_ogb_fna_lhelv.py
species=Lhelv
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}
for genomesss in $(echo "FAM11051 FAM11961 FAM21456 FAM21462 FAM21463")
do

echo ${genomesss}
neuNAme=$(echo ${genomesss}|sed 's/FAM/Lh/g')
mv ${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}/${neuNAme}.fna
sed -i 's/>lcl|/>/g'  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}/${neuNAme}.fna

done 

for species in $(echo "Ldel Sterm Lhelv")
do

ls ${species}/ |wc -l

done


#missings
#FAM12273
#FAM24728
#FAM24737
#FAM24738
#FAM24739
#FAM24740

for genomesss in $(cut -d ',' -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/all_genomes_dialact.txt|grep "^S")
do
echo ${genomesss}
head -1  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/Sterm/${genomesss}.g*
echo "-----------------------------"

done
```


extract the following information:

- coding genes
- pseudogenes


```{bash}
###========================================================
#all NCBI genomes download
###========================================================
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log 
curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt


###-----------------------
#Streptococcus suis
###-----------------------
species=$(echo "Streptococcus suis")
shortnamess=Ssuis

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
#gunzip *.gbk.gz

###-----------------------
#Lactobacillus plantarum
###-----------------------
species=$(echo "Lactobacillus plantarum")
shortnamess=Lplan

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

species=$(echo "Lactiplantibacillus plantarum")

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>   \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

###-----------------------
#Baumannia cicadellinicola
###-----------------------
species=$(echo "Baumannia")
shortnamess=Bcica

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}

##no hits

###-----------------------
#Rickettsia
###-----------------------
species=$(echo "Rickettsia")
shortnamess=Rick

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}"|grep "prowazekii" -v|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

###-----------------------
#Portiera
###-----------------------
species=$(echo "Portiera")
shortnamess=Port

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}"|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt



###-----------------------
#Sodalis glossinidius
###-----------------------
species=$(echo "Sodalis")
shortnamess=Sglos

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt


###-----------------------
#Wigglesworthia
###-----------------------
species=$(echo "Wigglesworthia")
shortnamess=Wigg

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}"|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt


###-----------------------
#Buchneria
###-----------------------
species=$(echo "Buchnera")
shortnamess=Buch

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

###-----------------------
#Buchneria
###-----------------------
species=$(echo "leprae")
shortnamess=Mleprae

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt


###-----------------------
#Staphylococcus gallinarum
###-----------------------
species=$(echo "caprae")
shortnamess=Scap

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "Staphylococcus" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

###-----------------------
#lactobacillus equicursoris
###-----------------------
species=$(echo "equicursoris")
shortnamess=Lequi

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}"|grep "actobacillus" -i |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

###-----------------------
#streptococcus vestibularis
###-----------------------
species=$(echo "vestibularis")
shortnamess=Svest

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}"|grep "streptococcus" -i |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

###-----------------------
#Wolbachia
###-----------------------
species=$(echo "Wolbachia")
shortnamess=Wol

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt



###========================================================
#extract counts
###========================================================

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt
for species in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/)
do
echo $species

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${species} |grep ".gz$" )
do

genezzz=$(zcat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${species}/${genomesss}|grep "Genes (coding)" |cut -d ':' -f 3|sed 's/ //g'|sed 's/,//g'|head -1)
peudosss=$(zcat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${species}/${genomesss}|grep "Pseudo Genes (total)"  |cut -d ':' -f 3|sed 's/ //g'|sed 's/,//g'|head -1)
transpos=$(zgrep -i "transpos"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${species}/${genomesss} |grep "product=" -c)


done #genomesss

done

```

```{bash}
###-----------------------
#Sterm and Ldel
###-----------------------
for species in $(echo "Sterm Ldel")
do
echo $species

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/ |grep ".gbk$" |sed 's/.gbk//g')
do

genezzz=$(grep "Genes (coding)" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/${genomesss}.gbk |cut -d ':' -f 3|sed 's/ //g'|sed 's/,//g'|head -1)
peudosss=$(grep "Pseudo Genes (total)" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/${genomesss}.gbk |cut -d ':' -f 3|sed 's/ //g'|sed 's/,//g'|head -1)

transpos=$(grep "transposase" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/${genomesss}.gbk |grep "product=" -c)


echo -e ${species}"\t"${genomesss}"\t"${genezzz}"\t"${peudosss}"\t"${transpos} >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt
done #genomesss

done
```


transfer local

```{bash}

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt
```

do analysis 

```{r}
library(readr)
library(tidyverse)
pseudogeness_count <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt",  "\t", escape_double = FALSE, col_names = c("species","genome","genes_coding","pseudogenes"), trim_ws = TRUE) %>% filter(pseudogenes<1000) %>% filter(genes_coding<4000)%>% mutate(pseudogenes_percent = 100*(pseudogenes/genes_coding)) %>% filter(species!="Wigg")%>% filter(species!="Scap")


plotpoints <- ggplot(pseudogeness_count,aes(x=genes_coding,y=pseudogenes_percent,fill=species,color=species))+geom_point()+labs(x="Gene number",y="Pseudogenes [%]")+theme_classic()+theme(legend.title = element_blank())
plotpoints
  
  library(plotly)
ggp <- ggplotly(plotpoints)
 ggp
# pseudogeness_count %>% group_by(species)%>%ize(meanPseudo=mean(pseudogeness_count),meangenes=mean(genes_coding),stdevPseudo=sd(pseudogeness_count),stdevgenes=stderr(genes_coding))
# pseudogeness_count %>% group_by(species) %>% summarise(meanPseudo=mean(pseudogeness_count))

pseudogeness_sumary <- pseudogeness_count %>% group_by(species) %>%  dplyr::summarize(meanPseudo=100*(mean(pseudogenes)/mean(genes_coding)),meangenes=mean(genes_coding),stdevPseudo=100*(sd(pseudogenes)/mean(genes_coding)),stdevgene=sd(genes_coding))



 # dplyr::summarize(mean = mean(NumGenesLineage),sd=sd(NumGenesLineage))

##-----
#chnage names
##-----

  pseudogeness_sumary$species_new <- plyr::revalue(pseudogeness_sumary$species,c("Buch"="Buchnera spp.","Sglos"="S. glossinidius","Ldel"="L. delbrueckii","Sterm"="S. thermophilus","Mleprae"="M. leprae","Rick"="Rickettsia spp.","Lplan"="L. plantarum","Ssuis"="S. suis","Lequi"="L. equicursoris","Svest"="S. vestibularis","Wol"="Wolbachia spp."))
#,"Wigg"="Wigglesworthia"
table(pseudogeness_sumary$species)
table(pseudogeness_sumary$species_new)  
  
# species=$(echo "vestibularis")

 pseudogeness_sumary$species_new  = factor(pseudogeness_sumary$species_new , levels=c("L. delbrueckii","S. thermophilus","L. plantarum","L. equicursoris","S. suis","S. vestibularis","M. leprae","Rickettsia spp.","Buchnera spp.","S. glossinidius","Wolbachia spp."))

 
colorsss <- c("#fdc2b5ff","#007fa6ff","#ebbc51ff","#ebbc51ff","#9fcfffff","#9fcfffff","#51eb85ff","#51eb85ff","#d58656ff","#d58656ff","#d58656ff") 
colorsss <- c("#fdc2b5ff","#007fa6ff","grey77","grey77","grey77","grey77","#51eb85ff","#51eb85ff","#d58656ff","#d58656ff","#d58656ff")

##-----
#plot
##-----
plotpoints <-ggplot( pseudogeness_sumary, aes(x = meangenes, y = meanPseudo, xmin = meangenes-stdevgene, xmax = meangenes+stdevgene,ymin=meanPseudo-stdevPseudo,ymax=meanPseudo+stdevPseudo,fill=species_new,color=species_new)) +
  geom_point(size=2) +
  geom_errorbarh(alpha=.5)+
  geom_errorbar(alpha=.5)+
    labs(x="Gene number",y="Pseudogenes [%]")+theme_classic()+theme(legend.title = element_blank())+
  #geom_hline(yintercept =7.5, linetype="dotted")+geom_segment(aes(x = 1500, y = -1, xend = 1500, yend = 7.5), linetype = "dotted")+
  scale_color_manual(values = colorsss)+scale_fill_manual(values = colorsss)+
  annotate("text", x = 800, y = 6, label = "endosymbionts",col="#d58656ff")+
  annotate("text", x = 800, y = 25, label = "transition",col="#51eb85ff")+
  annotate("text", x = 2000, y = 1, label = "free living",col="grey77")

#+lims(x=c(250,4100),y=c(-1.5,40))
plotpoints


pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenes.pdf",width=6,height=4)

plotpoints

dev.off()


# plotpoints_wo <- plotpoints+theme(legend.position = "none")
# plotpoints_wo
# plotpoints+stat_ellipse(data=pseudogeness_count,aes(x=genes_coding,y=pseudogenes_percent,fill=species,color=species,group=species),type = "euclid", level = 3) 

  
  library(plotly)
ggp <- ggplotly(plotpoints)
 ggp
 
 ##------------------------
 ##eclipse
 #to do this we include all std points and mean points into one datafram
 ##------------------------
 
#  pseudogeness_sumary
#  
#   pseudogeness_sumary$groups <- plyr::revalue(pseudogeness_sumary$species,c("Buch"="endosmymbiont","Sglos"="endosmymbiont","Ldel"="starter cultures","Sterm"="starter cultures","Wigg"="endosmymbiont","Mleprae"="transition","Rick"="transition","Lplan"="closely related free-living","Ssuis"="closely related free-living"))
# 
#  pseudogeness_esclipse <-  rbind(data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes,y=pseudogeness_sumary$meanPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes-pseudogeness_sumary$stdevgene,y=pseudogeness_sumary$meanPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes+pseudogeness_sumary$stdevgene,y=pseudogeness_sumary$meanPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes,y=pseudogeness_sumary$meanPseudo-pseudogeness_sumary$stdevPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes,y=pseudogeness_sumary$meanPseudo+pseudogeness_sumary$stdevPseudo))
#  
#  
#  
#  plotpoints_esclipse <- ggplot(pseudogeness_esclipse,aes(x=x,y=y,fill=groups,color=groups),group=groups)+labs(x="Gene number",y="Pseudogenes [%]")+theme_classic()+theme(legend.title = element_blank())+stat_ellipse(type = "t", linetype = 2)+theme(legend.position = "none")+lims(x=c(250,4100),y=c(-1.5,40))
# plotpoints_esclipse
#  
# 
# library(patchwork)
# 
# plotpoints+plotpoints_wo+plotpoints_esclipse

```

#### pseudogene clustering

```{bash}
###-----------------------
#get pseudogenes ffn file
###-----------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/


rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.gff
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff
for speciesss in $(echo "Sterm Ldel")
do

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${speciesss}|grep ".fna$" |sed 's/.fna//g')
do

echo $genomesss

#grep "pseudogene" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff |awk -F "[\t;]" '{OFS="\t"}{print $1,$4,$5,$9}' |sed 's/ID=gene-FAM/ps-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/tmp.download

#bedtools getfasta  -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${speciesss}/${genomesss}.fna \
#  -bed /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/tmp.download -nameOnly \
#  -fo /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}/${genomesss}.ffn

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}/${genomesss}.ffn >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn


#grep "pseudogene" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff |awk -F "[\t;]" '{OFS="\t"}{print $1,$4,$5,$9,$10,$11,$12,$13,$14}' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.gff

grep ";pseudo=true;" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff  >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff

done #genomesss
done #species



##-----------pseudo


wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff

grep "=is" -i  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff |awk -F "[\t;]" '{OFS="\t"}{print $1,$4,$5,$9,$10,$11,$12,$13,$14}' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase.gff
grep "transposase" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff |awk -F "[\t;]" '{OFS="\t"}{print $1,$4,$5,$9,$10,$11,$12,$13,$14}' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase.gff


cut -f 4 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase.gff |sort|uniq |sed 's/ID=cds-FAM/ps-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned.gff

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff

for genesss in $(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned.gff)
do

echo $genesss

grep "${genesss}" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff

done

cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff |sort|uniq> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss_derep.gff

cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt|sort|uniq |wc -l 
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss_derep.gff
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff 


###-----------------------
#cluster
###-----------------------

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering

#mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/${genomesss}.ffn ${genomesss} tmp --min-seq-id 0.8 -c 0.9 --cov-mode 1
mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn pseudo tmp 

###-----------------------
#bring local
###-----------------------


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/pseudo_cluster.tsv  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/

```

add NCBI strains


```{bash}
###-----------------------
#get pseudogenes ffn file
###-----------------------

/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/


rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.gff
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff
for speciesss in $(echo "Sterm Ldel")
do

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${speciesss}|grep ".fna$" |sed 's/.fna//g')
do

echo $genomesss

#grep "pseudogene" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff |awk -F "[\t;]" '{OFS="\t"}{print $1,$4,$5,$9}' |sed 's/ID=gene-FAM/ps-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/tmp.download

#bedtools getfasta  -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${speciesss}/${genomesss}.fna \
#  -bed /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/tmp.download -nameOnly \
#  -fo /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}/${genomesss}.ffn

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}/${genomesss}.ffn >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn


#grep "pseudogene" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff |awk -F "[\t;]" '{OFS="\t"}{print $1,$4,$5,$9,$10,$11,$12,$13,$14}' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.gff

grep ";pseudo=true;" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff  >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff

done #genomesss
done #species



##-----------pseudo


wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff

grep "=is" -i  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff |awk -F "[\t;]" '{OFS="\t"}{print $1,$4,$5,$9,$10,$11,$12,$13,$14}' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase.gff
grep "transposase" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff |awk -F "[\t;]" '{OFS="\t"}{print $1,$4,$5,$9,$10,$11,$12,$13,$14}' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase.gff


cut -f 4 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase.gff |sort|uniq |sed 's/ID=cds-FAM/ps-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned.gff

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff

for genesss in $(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned.gff)
do

echo $genesss

grep "${genesss}" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff

done

cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff |sort|uniq> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss_derep.gff

cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt|sort|uniq |wc -l 
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss_derep.gff
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff 


###-----------------------
#cluster
###-----------------------

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering

#mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/${genomesss}.ffn ${genomesss} tmp --min-seq-id 0.8 -c 0.9 --cov-mode 1
mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn pseudo tmp 

###-----------------------
#bring local
###-----------------------


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/pseudo_cluster.tsv  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/

```


try to identify glycoside hydrolases

```{r}

##------------------------------------
#prep tree
##------------------------------------

library(ape)
Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_curated.txt", tree.names = NULL, force.multi = FALSE)

tree_phylo <- Sterm_tree_raw %>% as.phylo()

Sterm_tree_raw$tip.label
## get the distance between every two nodes
# tree_phylog <- tree %>% as.phylo()
distance_matrix <-cophenetic.phylo(tree_phylo)
 library(reshape2)
 phylo_distance <- melt(distance_matrix) %>% rename(genome_1=Var1)%>% rename(genome_2=Var2) %>% rename(phylo_distance=value)

#---------------------------
library(readr)
library(plyr)
library(tidyverse)
pseudo_cluster <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/pseudo_cluster.tsv", "\t", escape_double = FALSE, col_names = c("cluster","pseudogene"),  trim_ws = TRUE)

pseudo_cluster_counts <- table(pseudo_cluster$cluster) %>% data.frame()
ggplot(pseudo_cluster_counts,aes(x=Freq))+geom_bar()+theme_classic()

range(pseudo_cluster_counts$Freq)
pseudo_distance_df <- data.frame()
for (clusterzz in unique(pseudo_cluster$cluster)) {
  print(clusterzz)
  
  tmp <- pseudo_cluster %>% filter(cluster==clusterzz)
        counts <- 0
  for (rowwsss in tmp$pseudogene) {
     counts <- counts+1
    for (columns in tmp$pseudogene[counts:nrow(tmp)]) {
      
      genomesss_1 <- str_split_fixed(rowwsss,"-",4)[,2]
      genomesss_2 <- str_split_fixed(columns,"-",4)[,2]

      
      genomesss_1_name <- Sterm_tree_raw$tip.label[grepl(genomesss_1,Sterm_tree_raw$tip.label)]
      genomesss_2_name <- Sterm_tree_raw$tip.label[grepl(genomesss_2,Sterm_tree_raw$tip.label)]


      distanzzz <- phylo_distance %>% filter(genome_1==genomesss_1_name&genome_2==genomesss_2_name) %>% pull(.,phylo_distance)
      
    tmp_2 <- data.frame(pseudo_cluster=clusterzz,genome_1=genomesss_1_name,genome_2=genomesss_2_name,phylo_distanc=distanzzz)  
   pseudo_distance_df <-  rbind(pseudo_distance_df,tmp_2)
    
    }
           

  }
  
  
}

pseudo_distance_df

# write.table(pseudo_distance_df,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/pseudo_cluster_distance_raw.tsv",sep="\t",quote = FALSE,row.names = FALSE)
pseudo_cluster_distance_raw_import <- read_delim("Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/pseudo_cluster_distance_raw.tsv", "\t", escape_double = FALSE, trim_ws = TRUE)
##------------------------------
#plot
##------------------------------
# unique(pseudo_distance_df$pseudo_cluster)


table(pseudo_cluster_distance_raw_import$pseudo_cluster) %>% sqrt() %>% range()

pseudo_distance_df_summary <- pseudo_cluster_distance_raw_import %>% filter(genome_1!=genome_2)%>% group_by(pseudo_cluster) %>% summarise(median_distance=median(phylo_distanc),counts=n(),max_distance=max(phylo_distanc),sd_distance=sd(phylo_distanc))


# write.table(pseudo_distance_df_summary,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/pseudo_cluster_distance_summary.tsv",sep="\t",quote = FALSE,row.names = FALSE)

ggplot(pseudo_distance_df_summary,aes(x=sqrt(counts),y=max_distance))+geom_point()+theme_classic()+labs(x="number genomes",y="phylogentic distance")


```

next I want to ADD all genes to the list of pseudogenes and redo the cluster&analysis. The reason I do this is to see if certain genomes have completely lost or retained the gene. 

```{bash}

/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn


###-----------------------
#add normal genes to speudogenes
###-----------------------

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes.ffn
for speciesss in $(echo "Sterm Ldel")
do

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${speciesss}|sed 's/.ffn//g')
do

sed 's/>FAM/>re-/g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${speciesss}/${genomesss}.ffn >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes.ffn

done #genomesss
done #species


###--------------------------------------------
##remove pseudogenes from all genes
###--------------------------------------------
##---------------------with faidx

sed "s/ .*//" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes.ffn   > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_new.ffn

samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_new.ffn

##get names
grep ">"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn |sed 's/>ps-/re-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/names_pseudogenes.ffn 

##invert list
remove_ids=($(awk '{print $1}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_new.ffn.fai | grep -v -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/names_pseudogenes.ffn ))

awk '{print $1}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_new.ffn.fai | grep -v -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/names_pseudogenes.ffn >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/names_pseudogenes_wo.ffn 


samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_new.ffn "${remove_ids[@]}" > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_final.ffn

xargs --arg-file=/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/names_pseudogenes_wo.ffn  samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_new.ffn > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_final.ffn

grep ">" -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_final.ffn 
grep ">" -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes.ffn 
grep ">" -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn 


##put together
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_final.ffn  > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_final.ffn

###-----------------------
#cluster
###-----------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all

#mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/${genomesss}.ffn ${genomesss} tmp --min-seq-id 0.8 -c 0.9 --cov-mode 1
mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_final.ffn  pseudo_genes tmp 

#less pseudo_genes_cluster.tsv
grep "ps-" pseudo_genes_cluster.tsv |head
awk '{if($2~"ps-")print $1}' pseudo_genes_cluster.tsv |uniq > clusters_names_with_pg.txt
rm cluster_ps.txt
for clusersss in $(cat clusters_names_with_pg.txt)
do
echo $clusersss
#awk -v clusterzzz="$clusersss" '{if($1==clusterzzz)print 0}' pseudo_genes_cluster.tsv >> cluster_ps.txt

grep "${clusersss}" pseudo_genes_cluster.tsv >> cluster_ps.txt


done
wc -l clusters_names_with_pg.txt
cut -f 1 cluster_ps.txt |uniq |wc -l
cut -f 1 pseudo_genes_cluster.tsv |uniq |wc -l

```

```{bash}


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt

```


get the most common recent ancestor and all tips associated to this
look [here](https://rpubs.com/tskam/ternary_plot) for the ternary plot description. 

```{r}

library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)

Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_curated.txt", tree.names = NULL, force.multi = FALSE)
tree_phylo <- Sterm_tree_raw %>% as.phylo()
tree_phylo$tip.label

getMRCA(tree_phylo, c("S24736","S24758"))

numbers <- Descendants(tree_phylo, 156, type = "tips") %>% unlist()
numbers
tree_phylo$tip.label[numbers]

##------------------------------
#get data clusters
##------------------------------
cluster_ps <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt",  "\t", escape_double = FALSE, col_names = c("cluster","gene"),  trim_ws = TRUE)

cluster_ps$genome <- str_split_fixed(cluster_ps$gene,"-",4)[,2]

analysis_clusters_ldel <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_ldel.csv",  ",") %>% add_column(species="Ldel") %>% select(-Cluster)
analysis_clusters_sterm <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv",  ",")%>% add_column(species="Sterm") %>% select(-Cluster)

analysis_clusters_all <- rbind(analysis_clusters_sterm,analysis_clusters_ldel)

analysis_clusters_all$Taxon <- gsub("S","",analysis_clusters_all$Taxon) %>% gsub("L","",.)

# C("ps-10980-i1-1.1_000998" ,"ps-24763-i1-1.1_001759")



cluster_ps_species <- merge(cluster_ps,analysis_clusters_all,by.x="genome",by.y="Taxon") %>% as_tibble()

tmp <- cluster_ps_species %>% select(cluster,species) %>% unique() 
# table(tmp$cluster) %>% sort(decreasing = TRUE)


genome_names_corrected <- data.frame()

for (genomesss in unique(cluster_ps$genome)){
        tmp <- tree_phylo$tip.label[grepl(genomesss,tree_phylo$tip.label)]
tmp_2 <- data.frame(genome=genomesss,genome_name=tmp)
genome_names_corrected <- rbind(genome_names_corrected,tmp_2)
}

cluster_ps_final$cluster %>% unique()

cluster_ps_final <- merge(cluster_ps,genome_names_corrected,by="genome")
cluster_ps_final$gene_type <- str_split_fixed(cluster_ps_final$gene,"-",4)[,1]

cluster_ps_final <- cluster_ps_final %>% select(-c(genome,gene)) %>% unique()
dim(cluster_ps_final)
# dim(unique(cluster_ps_final))
cluster_ps_final %>% select(-gene_type) %>% unique() %>% dim()
##a few genes are both ps and re

cluster_ps_final_cleaned <- cluster_ps_final[!duplicated(cluster_ps_final[,-3]),]
dim(cluster_ps_final_cleaned)
cluster_ps_final_cleaned  %>% select(-gene_type) %>% unique() %>% dim()

# cluster_ps_final_cleaned %>% group_by(cluster) %>% summarise(counts=table(gene_type))
cluster_ps_final_cleaned_count <- cluster_ps_final_cleaned %>% group_by(cluster) %>% mutate(n=n(gene_type))
cluster_ps_final_cleaned_count %>% filter(cluster=="ps-10973-i1-1.1_000242")

cluster_ps_final_cleaned_count_wide <- spread(cluster_ps_final_cleaned_count, gene_type, n)  %>%replace(is.na(.), 0)
cluster_ps_final_cleaned_count_wide


###find number of genes per cluster
total_genomes <- data.frame()
for (clusterzzz in unique(cluster_ps_final_cleaned$cluster)) {
  print(clusterzzz)
  # tmp <- cluster_ps_final_cleaned %>% filter(cluster==clusterzzz) %>% pull(.,genome_name) %>% droplevels() %>% as.character()
    tmp <- cluster_ps_final_cleaned %>% filter(cluster==clusterzzz) %>% pull(.,genome_name) %>% as.character()

  nodess <- getMRCA(tree_phylo, tmp)
# nodess
numbers <- Descendants(tree_phylo, nodess, type = "tips") %>% unlist()
# numbers
genomesss <- tree_phylo$tip.label[numbers]

genomesss_final <- ifelse(length(genomesss)==0,1,length(genomesss))

tmp_2 <- data.frame(cluster=clusterzzz,total_genomes=genomesss_final)
total_genomes <- rbind(total_genomes,tmp_2 )
}
total_genomes
##------------merged with others

cluster_ps_final_cleaned_count_wide_final <- merge(cluster_ps_final_cleaned_count_wide,total_genomes,by="cluster") %>% as.tibble() %>% mutate(rm=total_genomes-ps-re)


cluster_ps_final_cleaned_count_wide_final <- cluster_ps_final_cleaned_count_wide_final %>% mutate(pseudogenes=100*(ps/total_genomes))%>% mutate(complete=100*(re/total_genomes))%>% mutate(lost=100*(rm/total_genomes))%>% mutate(summed=lost+pseudogenes+complete)

# cluster_ps_final_cleaned_count_wide_final %>% filter(summed>100.5)

##------------------------------
#make ternary plot
##------------------------------

library(ggtern)

# ggtern(data=cluster_ps_final_cleaned_count_wide_final,aes(x=complete,y=lost, z=pseudogenes,fill=total_genomes,color=total_genomes)) +
  # geom_point()
dim(cluster_ps_final_cleaned_count_wide_final)
pseudogenization_plot <- ggtern(data=cluster_ps_final_cleaned_count_wide_final,aes(x=complete,y=lost, z=pseudogenes,color=total_genomes,size=total_genomes)) +
  geom_point(alpha=0.5) +#scale_size(trans = 'reverse')+
  # labs(title="Population structure, 2015") +
  theme_rgbw()

pseudogenization_plot
pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization.pdf",width=11,height=9)

pseudogenization_plot

dev.off()
##------------------------------
#box plot
##------------------------------
cluster_ps_final_cleaned_count_wide_final_smaller <- cluster_ps_final_cleaned_count_wide_final %>% select(cluster,pseudogenes,complete,lost)%>% gather(.,grouping,percents,c(pseudogenes,complete,lost))
 cluster_ps_final_cleaned_count_wide_final_smaller$grouping = factor(cluster_ps_final_cleaned_count_wide_final_smaller$grouping, levels=c("lost","pseudogenes","complete"))

pseudogenization_boxplot <- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()
pseudogenization_boxplot





pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_violin.pdf",width=11,height=9)

pseudogenization_boxplot

dev.off()

##-------------------------------
#Update figure
##-------------------------------

analysis_clusters_all$Taxon <- gsub("S","",analysis_clusters_all$Taxon) %>% gsub("L","",.)


species_cluster <- cluster_ps_species %>% select(cluster,species) %>% unique() 


tmp <- cluster_ps_final_cleaned_count_wide_final %>% filter(!cluster %in% c("ps-10980-i1-1.1_000998" ,"ps-24763-i1-1.1_001759"))

cluster_ps_final_cleaned_count_wide_final_species <- merge(tmp,species_cluster, by.x="cluster",by.y="cluster") %>%  as_tibble()

colorsss <- rev(c("#007fa6ff","#fdc2b5ff"))

pseudogenization_plot_species <- ggtern(data=cluster_ps_final_cleaned_count_wide_final_species,aes(x=complete,y=lost, z=pseudogenes,color=species)) +
  geom_point(alpha=0.5,size=3.5) + scale_color_manual(values =colorsss )+#scale_size(trans = 'reverse')+
  # labs(title="Population structure, 2015") +
  theme_rgbw()

table(cluster_ps_final_cleaned_count_wide_final_species$species)
dim(cluster_ps_final_cleaned_count_wide_final_species)

pseudogenization_plot_species
pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species.pdf",width=11,height=9)

pseudogenization_plot_species

dev.off()


##----------------------
#prep 
##----------------------


analysis_clusters_ldel <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_ldel.csv",  ",") %>% add_column(species="Ldel") %>% select(-Cluster)
analysis_clusters_sterm <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv",  ",")%>% add_column(species="Sterm") %>% select(-Cluster)

analysis_clusters_all_prep <- rbind(analysis_clusters_sterm,analysis_clusters_ldel)


cluster_ps_final_prep <- merge(cluster_ps_final,analysis_clusters_all_prep,by.x="genome_name",by.y="Taxon")

# cluster_ps_final$genome_name %>% 
  
  write.csv(cluster_ps_final_prep,"~/Desktop/Projects/2020_StarterCultureDiversity/data/cluster_ps_final_prep.csv",quote = FALSE,row.names = FALSE)


##----------------------
#grouping
##----------------------

cluster_ps_final_cleaned_count_wide_final_species <- ifelse(lost>60&complete<40&pseudogenes<40,"old",ifelse(lost<40&complete>60&pseudogenes<40,"recent",ifelse(lost<40&complete<40&pseudogenes>60,"maintained","intermediate")))


cluster_ps_final_cleaned_count_wide_final_species_extended <- cluster_ps_final_cleaned_count_wide_final_species %>% mutate(category=ifelse(lost>60&complete<40&pseudogenes<40,"old",ifelse(lost<40&complete>60&pseudogenes<40,"recent",ifelse(lost<40&complete<40&pseudogenes>60,"maintained","intermediate"))))

table(cluster_ps_final_cleaned_count_wide_final_species_extended$category)


# ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=species, group=category,fill=category))+geom_bar(position = "dodge")+theme_classic()


cluster_ps_final_cleaned_count_wide_final_species_extended$species <- plyr::revalue(cluster_ps_final_cleaned_count_wide_final_species_extended$species,c( "Ldel"="L. delbrueckii",
 "Sterm"="S. thermophilus"))

# ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=interaction(species,category), group=category,fill=category))+geom_bar(position = "dodge")+theme_classic()
genes_time_pseudogenization <- ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=species, group=category,fill=category))+geom_bar(position = "dodge")+theme_classic()+facet_wrap(~category,nrow = 1)+theme(axis.title.x = element_blank(),legend.position = "none",axis.text.x = element_text(angle=75,hjust=1))+labs(y="pseudogene gene family")

genes_time_pseudogenization

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing.pdf",width=7,height=3)

genes_time_pseudogenization

dev.off()


##----------------------------------
#fraction of genomes containing this pseudogene
cluster_ps_final_cleaned_count_wide_final_species$species <- plyr::revalue(cluster_ps_final_cleaned_count_wide_final_species$species,c( "Ldel"="L. delbrueckii",
 "Sterm"="S. thermophilus"))
distribution_pseudogenes <- ggplot(cluster_ps_final_cleaned_count_wide_final_species,aes(x=pseudogenes,fill=species))+geom_histogram()+theme_classic()+facet_wrap(~species,nrow=2,scales="free_y")+labs(x="fraction of genomes containing the pseudogene",y="number of pseudogene containing gene families")+theme(legend.position="none")+ scale_fill_manual(values =colorsss )

distribution_pseudogenes

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing_02.pdf",width=7,height=6)

distribution_pseudogenes

dev.off()


```
## download complete FAM strains from NCBI 

I have downloaded them previousl and converted here, I just have to extract the gene and pseudogenes of these genomes
```{bash}



#mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/FAA/

#seqret -sformat genbank -sequence  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GBK/S24724.gbf -outseq /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/FAA/S24724.faa
speciesss=Sterm
genomesss=S24724

#rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG
for speciesss in $(echo "Sterm Ldel")
do
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FNA/
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_pseudo_faa/${speciesss}//FNA/


firstCharacter=${speciesss:0:1}

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT//pseudo/genes/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT//pseudo/genes/${speciesss}
#rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo/${speciesss}/all_pseudogenes_complete.gff
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF/  |grep "^${firstCharacter}"|grep ".gff" |sed 's/.gff//g')
do

echo $genomesss


#grep "pseudogene" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |awk -F "[\t;]" '{OFS="\t"}{print $1,$4,$5,$9}' |sed 's/ID=gene-FAM/ps-/g' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/tmp.download


#grep "CDS" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep "pseudo=true" |wc -l
#grep "CDS" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep "pseudo=true" -v |grep "ncRNA" -v |wc -l
#less /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.*

#grep "Pseudo Genes (total)" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1
#grep "CDSs (with protein)" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1

locusTagNCBI=$(grep ".pseudogene" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF//${genomesss}.gff |awk -F "[\t;]" '{OFS="\t"}{print $9}'|head -1 |sed 's/ID=//g'|cut -d '_' -f 1)
#locusTagNCBI=$(grep "pseudogene" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF//${genomesss}.gff |grep "pseudogene"|awk -F "[\t;]" '{OFS="\t"}{print $9}'|head -1 |sed 's/ID=cds-//g'|cut -d '_' -f 1)

grep ".pseudogene" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF//${genomesss}.gff |awk -F "[\t;]" '{OFS="\t"}{print $1,$2,$3,$4,$5,$6,$7,$8,$9}' |sed "s/ID=${locusTagNCBI}/ps:${genomesss}/g" |sed 's/\.pseu.*//g' |awk -F "[\t]" '{OFS="\t"}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}'  > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.pseudo.download


awk -F "[\t]" '{OFS="\t"}{if($3=="mRNA")print $0}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF//${genomesss}.gff    |awk -F "[\t;]" '{OFS="\t"}{print  $1,$2,$3,$4,$5,$6,$7,$8,$9}' |sed  "s/ID=${locusTagNCBI}/re:${genomesss}/g"|sed 's/.t.*//g' |awk -F "[\t]" '{OFS="\t"}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.complete.download


#grep "CDS" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF//${genomesss}.gff  |awk -F "[\t;]" '{OFS="\t"}{print $1,$2,$3,$4,$5,$6,$7,$8,$10}' |sed  "s/Parent=${locusTagNCBI}/re:${genomesss}/g" |sed 's/.t.*//g' |awk -F "[\t]" '{OFS="\t"}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.complete.download

#grep "${genomesss}$(printf '\t')" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |awk  -F "[\t]" -v loczzz="$locusTagNCBI" '{OFS="\t"}{print loczzz,$0}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG


bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/FNA//${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.pseudo.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.pseudo.ffn

bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/FNA//${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.complete.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.complete.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.pseudo.ffn >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/all_pseudogenes.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.complete.ffn >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/all_complete.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes//${speciesss}/${genomesss}.pseudo.ffn > /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes//${speciesss}/${genomesss}.complete.ffn >> /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes//${speciesss}/${genomesss}.pseudo.ffn > /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_pseudo_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes//${speciesss}/${genomesss}.complete.ffn > /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FNA/${genomesss}.all.ffn



#grep "pseudogene" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  |awk -F "[\t;]" -v genomessszz="$genomesss" '{OFS="\t"}{print $1,$4,$5,$9,$10,$11,$12,$13,$14,genomessszz}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/all_pseudogenes.gff

#grep ";pseudo=true;" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//${speciesss}/all_pseudogenes_complete.gff

done #genomesss
done #species

```



## download NCBI final

```{bash}
##---------------------------
#rename NCBI strains
##---------------------------


###-------------------------------------------------
#final list
###-------------------------------------------------


#missings
#FAM24728
#FAM24737
#FAM24738
#FAM24739
#FAM24740
rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log

for genomesss in $(echo "strain=24728
strain=24737
strain=24738
strain=24739
strain=24740")
do

grep ${genomesss}  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log.txt


genomesss_short=$(echo ${genomesss}|sed 's/strain=/S/g')
grep ${genomesss}  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log.txt |awk -F "\t" -v biosamplezzz="$genomesss_short"  '{OFS="\t"}{print biosamplezzz,$1,$2,$3,$8,$9}'  >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.txt


grep ${genomesss}  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log.txt |cut -f 20 | sed -r 's|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.downloadONLY


echo "------------------------"
#awk -F "\t" -v biosamplezzz="$biosampless" -v samplezzz="$sampleesss"  -v speciessssss="$speciesss"  -v ageezz="$agee"  -v GCFSssszz="$GCFSsss_new"  '{OFS="\t"}{if($3==biosamplezzz) print samplezzz,speciessssss,ageezz,GCFSssszz,$1,$2,$3,$8,$9}' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log.txt

done

paste -d '\t' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.txt /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.downloadONLY >  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL

##===================================================================================================
###---------------------------------------------------------------
#download genomes
###---------------------------------------------------------------
##===================================================================================================

whichFILE=fna

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA/
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA/

cut -f 7 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file

rm *

wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file


###-------------------------------------
##change name
###-------------------------------------

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA/

for genomessss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL )
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |cut -f 1 )
echo "-----------------------"
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz

#mv /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA//GCF_900322585.1_Lactobacillus_delbrueckii_subsp._lactis_protein.fna /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA/Lactobacillus_delbrueckii_subsp_lactis-7.fna


##===================================================================================================
###---------------------------------------------------------------
#download faa
###---------------------------------------------------------------
##===================================================================================================
whichFILE=faa

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FAA/
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FAA/

cut -f 7 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |sed 's/_genomic.fna./_protein.faa./g'  > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file

rm *

wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file


###-------------------------------------
##change name
###-------------------------------------

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FAA/

for genomessss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL )
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |cut -f 1 )
echo "-----------------------"
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz


##===================================================================================================
###---------------------------------------------------------------
#download gff
###---------------------------------------------------------------
##===================================================================================================
whichFILE=gff

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GFF/
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GFF/

cut -f 7 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |sed 's/_genomic.fna./_genomic.gff./g'  > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file

rm *

wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file


###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL )
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |cut -f 1 )
echo "-----------------------"
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz


##===================================================================================================
###---------------------------------------------------------------
#download gbff
###---------------------------------------------------------------
##===================================================================================================
whichFILE=gbff

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GBFF/
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GBFF/

cut -f 7 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |sed 's/_genomic.fna./_genomic.gbff./g'  > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file

rm *

wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file


###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL )
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |cut -f 1 )
echo "-----------------------"
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz



```


## pseudogene cluster with NCBI

```{bash}

###-----------------------
#get pseudogenes ffn file
###-----------------------


rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG
for speciesss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL|sort|uniq)
do


specizzzz=$(grep ${speciesss} /data/Project/2020_StarterCultureDiversity/log.species|cut -f 1)
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_pseudo_faa/${specizzzz}//FNA/

mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${specizzzz}//FNA/


rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo/genes/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo/genes/${speciesss}
#rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo/${speciesss}/all_pseudogenes_complete.gff
for genomesss in $(grep ${speciesss} /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL|cut -f 1)
do

echo $genomesss

#grep "pseudogene" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |awk -F "[\t;]" '{OFS="\t"}{print $1,$4,$5,$9}' |sed 's/ID=gene-FAM/ps-/g' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/tmp.download


#grep "CDS" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep "pseudo=true" |wc -l
#grep "CDS" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep "pseudo=true" -v |grep "ncRNA" -v |wc -l
#less /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.*

#grep "Pseudo Genes (total)" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1
#grep "CDSs (with protein)" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1

locusTagNCBI=$(grep "CDS" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep "pseudo=true"|awk -F "[\t;]" '{OFS="\t"}{print $9}'|head -1 |sed 's/ID=cds-//g'|cut -d '_' -f 1)

grep "CDS" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep "pseudo=true"|awk -F "[\t;]" '{OFS="\t"}{print $1,$2,$3,$4,$5,$6,$7,$8,$9}' |sed "s/ID=cds-${locusTagNCBI}/ps:${genomesss}/g" |awk -F "[\t]" '{OFS="\t"}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}'  > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.pseudo.download



awk -F "[\t]" '{OFS="\t"}{if($3=="gene")print $0}' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  |grep "rna" -i -v |awk -F "[\t;]" '{OFS="\t"}{print  $1,$2,$3,$4,$5,$6,$7,$8,$9}' |sed  "s/ID=gene-${locusTagNCBI}/re:${genomesss}/g" |awk -F "[\t]" '{OFS="\t"}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.complete.download

grep "${genomesss}$(printf '\t')" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |awk  -F "[\t]" -v loczzz="$locusTagNCBI" '{OFS="\t"}{print loczzz,$0}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG


bedtools getfasta -s  -fi /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/FNA//${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.pseudo.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.pseudo.ffn

bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/FNA//${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.complete.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.complete.ffn

cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.pseudo.ffn >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/all_pseudogenes.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.complete.ffn >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/all_complete.ffn


cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.pseudo.ffn  > /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${specizzzz}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.complete.ffn >> /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${specizzzz}//FNA/${genomesss}.all.ffn


cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.pseudo.ffn   > /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_pseudo_faa/${specizzzz}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.complete.ffn  > /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${specizzzz}//FNA/${genomesss}.all.ffn


#grep "pseudogene" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  |awk -F "[\t;]" -v genomessszz="$genomesss" '{OFS="\t"}{print $1,$4,$5,$9,$10,$11,$12,$13,$14,genomessszz}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/all_pseudogenes.gff

#grep ";pseudo=true;" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//${speciesss}/all_pseudogenes_complete.gff

done #genomesss
done #species
###------------------------------------------------------
#FAM strains
###------------------------------------------------------

#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK
for speciesss in $(echo "Sterm Ldel Lhelv")
do

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo${speciesss}/all_pseudogenes_complete.gff
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${speciesss}|sed 's/.gbk//g' )
do

echo $genomesss

#grep "pseudogene" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |awk -F "[\t;]" '{OFS="\t"}{print $1,$4,$5,$9}' |sed 's/ID=gene-FAM/ps-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/tmp.download


#grep "CDS" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep "pseudo=true" |wc -l
#grep "CDS" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep "pseudo=true" -v |grep "ncRNA" -v |wc -l
#less /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.*

#grep "Pseudo Genes (total)" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1
#grep "CDSs (with protein)" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1

locusTagNCBI=$(grep "CDS" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff |grep "pseudo=true"|awk -F "[\t;]" '{OFS="\t"}{print $9}'|head -1 |sed 's/ID=cds-//g'|cut -d '_' -f 1)

 grep "CDS" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff |grep "pseudo=true"|awk -F "[\t;]" '{OFS="\t"}{print  $1,$2,$3,$4,$5,$6,$7,$8,$9}' |sed "s/ID=cds-${locusTagNCBI}/ps:${genomesss}/g"  |awk -F "[\t]" '{OFS="\t"}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}'  > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.pseudo.download
 #grep "CDS" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff  |grep -v  "pseudo=true"|awk -F "[\t;]" '{OFS="\t"}{print $1,$2,$3,$4,$5,$6,$7,$8,$9}' |sed  "s/ID=cds-${locusTagNCBI}/re:${genomesss}/g"  |awk -F "[\t]" '{OFS="\t"}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}'  > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.complete.download

awk -F "[\t]" '{OFS="\t"}{if($3=="gene")print $0}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff   |grep "rna" -i -v |awk -F "[\t;]" '{OFS="\t"}{print  $1,$2,$3,$4,$5,$6,$7,$8,$9}' |sed  "s/ID=gene-${locusTagNCBI}/re:${genomesss}/g" |awk -F "[\t]" '{OFS="\t"}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.complete.download



#awk -F "[\t]" '{OFS="\t"}{if($3=="mRNA")print $0}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF//${genomesss}.gff    |awk -F "[\t;]" '{OFS="\t"}{print  $1,$2,$3,$4,$5,$6,$7,$8,$9}' |sed  "s/ID=${locusTagNCBI}/re:${genomesss}/g"|sed 's/.t.*//g' |awk -F "[\t]" '{OFS="\t"}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.complete.download



#grep "${genomesss}$(printf '\t')" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |awk  -F "[\t]" -v loczzz="$locusTagNCBI" '{OFS="\t"}{print loczzz,$0}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG


bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${speciesss}/${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.pseudo.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn

bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${speciesss}/${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.complete.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.complete.ffn





#grep "pseudogene" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  |awk -F "[\t;]" -v genomessszz="$genomesss" '{OFS="\t"}{print $1,$4,$5,$9,$10,$11,$12,$13,$14,genomessszz}' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_pseudogenes.gff

#grep ";pseudo=true;" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}/all_pseudogenes_complete.gff

done #genomesss
done #species

###-----------------------
#for some dailact genomes we have both ONT and Illumina assemblies---in thes cases I have to delete the Illumina only genomes
###-----------------------




#rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG
for speciesss in $(echo "Sterm Ldel")
do

firstCharacter=${speciesss:0:1}


#rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo/${speciesss}/all_pseudogenes_complete.gff
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF/  |grep "^${firstCharacter}"|grep ".gff" |sed 's/.gff//g')
do

echo $genomesss
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.*

done

done


#---------------------------------merge the genomes

for speciesss in $(echo "Sterm Ldel Lhelv")
do

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_pseudogenes.ffn
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_complete.ffn
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo${speciesss}/all_pseudogenes_complete.gff
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/ |grep ".pseudo.ffn"| sed 's/.pseudo.ffn//g'|sort|uniq )
do

echo $genomesss

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_pseudogenes.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.complete.ffn >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_complete.ffn


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes//${speciesss}/${genomesss}.pseudo.ffn    > /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes//${speciesss}/${genomesss}.complete.ffn >> /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes//${speciesss}/${genomesss}.pseudo.ffn   > /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_pseudo_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes//${speciesss}/${genomesss}.complete.ffn  > /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FNA/${genomesss}.all.ffn


done
done





###------------------------------------------------------
#FAM strains FROM NCBI
###------------------------------------------------------
#/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA/
#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK
for speciesss in $(echo "Sterm")
do

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}
#rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo${speciesss}/all_pseudogenes_complete.gff
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GBFF/|sed 's/.gbff//g' )
do

echo $genomesss

#grep "pseudogene" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |awk -F "[\t;]" '{OFS="\t"}{print $1,$4,$5,$9}' |sed 's/ID=gene-FAM/ps-/g' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/tmp.download


#grep "CDS" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep "pseudo=true" |wc -l
#grep "CDS" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep "pseudo=true" -v |grep "ncRNA" -v |wc -l
#less /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.*

#grep "Pseudo Genes (total)" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1
#grep "CDSs (with protein)" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1

locusTagNCBI=$(grep "CDS" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GFF/${genomesss}.gff |grep "pseudo=true"|awk -F "[\t;]" '{OFS="\t"}{print $9}'|head -1 |sed 's/ID=cds-//g'|cut -d '_' -f 1)

 grep "CDS" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GFF//${genomesss}.gff |grep "pseudo=true"|awk -F "[\t;]" '{OFS="\t"}{print $1,$2,$3,$4,$5,$6,$7,$8,$9}' |sed "s/ID=cds-${locusTagNCBI}/ps:${genomesss}/g"  |awk -F "[\t]" '{OFS="\t"}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}'  > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.download
# grep "CDS" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GFF//${genomesss}.gff  |grep -v  "pseudo=true"|awk -F "[\t;]" '{OFS="\t"}{print$1,$2,$3,$4,$5,$6,$7,$8,$10}' |sed  "s/Parent=gene-${locusTagNCBI}/re:${genomesss}/g"  |awk -F "[\t]" '{OFS="\t"}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.complete.download



awk -F "[\t]" '{OFS="\t"}{if($3=="gene")print $0}' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GFF//${genomesss}.gff   |grep "rna" -i -v |awk -F "[\t;]" '{OFS="\t"}{print  $1,$2,$3,$4,$5,$6,$7,$8,$9}' |sed  "s/ID=gene-${locusTagNCBI}/re:${genomesss}/g" |awk -F "[\t]" '{OFS="\t"}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.complete.download

#grep "${genomesss}$(printf '\t')" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |awk  -F "[\t]" -v loczzz="$locusTagNCBI" '{OFS="\t"}{print loczzz,$0}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG


bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA//${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn

bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA//${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.complete.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.complete.ffn

cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/all_pseudogenes.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.complete.ffn >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/all_complete.ffn



cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn > /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes//${speciesss}/${genomesss}.complete.ffn >> /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn

cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn    > /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes//${speciesss}/${genomesss}.complete.ffn >> /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn



cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn    > /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_pseudo_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes//${speciesss}/${genomesss}.complete.ffn  > /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FNA/${genomesss}.all.ffn


#grep "pseudogene" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  |awk -F "[\t;]" -v genomessszz="$genomesss" '{OFS="\t"}{print $1,$4,$5,$9,$10,$11,$12,$13,$14,genomessszz}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/all_pseudogenes.gff

#grep ";pseudo=true;" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/${speciesss}/all_pseudogenes_complete.gff

done #genomesss
ll /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA
done #species

```
add also non-pseudogenized genes


```{bash}


###-----------------------
#combine genes
###-----------------------

rm /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes
cd /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes
for speciesss in $(echo "Sterm Ldel Lhelv")
do

specizzzz=$(grep ${speciesss} /data/Project/2020_StarterCultureDiversity/log.species|cut -f 2)

##dialact but from NCBI
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/all_pseudogenes.ffn > ${speciesss}.all.genes.fasta
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/all_complete.ffn >>  ${speciesss}.all.genes.fasta

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/all_pseudogenes.ffn >> ${speciesss}.all.genes.fasta
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/all_complete.ffn >>  ${speciesss}.all.genes.fasta

##dialact 
cat  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_pseudogenes.ffn >>  ${speciesss}.all.genes.fasta
cat  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_complete.ffn >>  ${speciesss}.all.genes.fasta

##NCBI
cat   /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${specizzzz}//all_pseudogenes.ffn >>  ${speciesss}.all.genes.fasta
cat   /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${specizzzz}//all_complete.ffn >>  ${speciesss}.all.genes.fasta

echo ${speciesss}
grep ">" ${speciesss}.all.genes.fasta |cut -d ':' -f 2|cut -d '_' -f 1|sort|uniq -c
#grep ">" ${speciesss}.all.genes.fasta |cut -d '_' -f 1|sort|uniq -c

done #species


###-----------------------
#NEW combine genes
###-----------------------

rm /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis_02/
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis_02/genes
cd /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis_02/genes

 


for speciesss in $(echo "Sterm Ldel Lhelv")
do

specizzzz=$(grep ${speciesss} /data/Project/2020_StarterCultureDiversity/log.species|cut -f 2)

##dialact but from NCBI
cat /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/*.all.ffn > ${speciesss}_ALL.GENES.fna
grep ">" ${speciesss}_ALL.GENES.fna |cut -d ':' -f 2|cut -d '_' -f 1|sort|uniq -c
#grep ">" ${speciesss}.all.genes.fasta |cut -d '_' -f 1|sort|uniq -c

done #species


###-----------------------
#cluster
###-----------------------

rm /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/mmseqs2_clustering_all_restricted
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/mmseqs2_clustering_all_restricted
cd /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/mmseqs2_clustering_all_restricted
for speciesss in $(echo "Sterm Ldel Lhelv")
do

#mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis_02/genes/${speciesss}_ALL.GENES.fna  ${speciesss}_pseudo_genes tmp 
mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis_02/genes/${speciesss}_ALL.GENES.fna  --min-seq-id 0.8 -c 0.9  ${speciesss}_pseudo_genes tmp 

 
done


##-----------------prep
rm  ALL_pseudo_genes_cluster_ps.txt
for speciesss in $(echo "Sterm Ldel Lhelv")
do

#grep "ps:" ${speciesss}_pseudo_genes
awk '{if($2~"ps:")print $1}' ${speciesss}_pseudo_genes_cluster.tsv |uniq > ${speciesss}_pseudo_genes_clusters_names_with_pg.txt
rm ${speciesss}_pseudo_genes_cluster_ps.txt
for clusersss in $(cat ${speciesss}_pseudo_genes_clusters_names_with_pg.txt)
do
echo $clusersss
#awk -v clusterzzz="$clusersss" '{if($1==clusterzzz)print 0}' pseudo_genes_cluster.tsv >> cluster_ps.txt

grep "${clusersss}" ${speciesss}_pseudo_genes_cluster.tsv >> ${speciesss}_pseudo_genes_cluster_ps.txt

done
wc -l ${speciesss}_pseudo_genes_cluster_ps.txt
cut -f 1 ${speciesss}_pseudo_genes_cluster_ps.txt|uniq |wc -l
cut -f 1 ${speciesss}_pseudo_genes_clusters_names_with_pg.txt |uniq |wc -l

awk -F "\t" -v speciezzz="$speciesss" '{OFS="\t"}{print speciezzz,$0}' ${speciesss}_pseudo_genes_cluster_ps.txt >> ALL_pseudo_genes_cluster_ps.txt

done
wc -l ALL_pseudo_genes_cluster_ps.txt

##-----------------analysis

for speciesss in $(echo "Sterm Ldel Lhelv")
do
echo -e "--------------------------------"${speciesss}
#wc -l ${speciesss}_pseudo_genes_cluster_ps.txt
cut -f 1 ${speciesss}_pseudo_genes_cluster_ps.txt|uniq |wc -l
cut -f 1 ${speciesss}_pseudo_genes_cluster.tsv|uniq |wc -l
#cut -f 1 ${speciesss}_pseudo_genes_clusters_names_with_pg.txt |uniq |wc -l
#cut -f 2 ${speciesss}_pseudo_genes_cluster.tsv|sort|uniq |wc -l
done

#STerm  2163/6648
#Ldel   1965/7400
#Lhelv   1511/5680

1511



```


```{bash}


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/
scp  vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/mmseqs2_clustering_all_restricted/ALL_pseudo_genes_cluster_ps.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/

```


get the most common recent ancestor and all tips associated to this
look [here](https://rpubs.com/tskam/ternary_plot) for the ternary plot description. 

```{r}

library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)

##------------------------------
#get data clusters
##------------------------------
cluster_ps <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/ALL_pseudo_genes_cluster_ps.txt",  "\t", escape_double = FALSE, col_names = c("species","cluster","gene"),  trim_ws = TRUE)

cluster_ps$genome <- str_split_fixed(cluster_ps$gene,"_",2)[,1] %>% gsub(".*:","",.)

genome_count <- cluster_ps %>% select(species, genome) %>% unique() %>% group_by(species) %>% summarise(num_genome=n())

cluster_ps$gene_type <- gsub(":.*","",cluster_ps$gene)

cluster_ps_counts <- cluster_ps %>% group_by(species,cluster,genome,gene_type) %>% summarise(countss=n())
table(cluster_ps_counts$countss)

cluster_ps_counts_wide <- cluster_ps_counts %>% spread(., gene_type, countss)  %>%replace(is.na(.), 0) %>% mutate(gene_presence=ifelse(re>0,"re","ps"))


cluster_ps_counts_gene_counts <- cluster_ps_counts_wide %>% group_by(species,cluster,gene_presence) %>% summarise(countz=n()) %>% spread(., gene_presence, countz)  %>%replace(is.na(.), 0) %>% left_join(.,genome_count,by="species") %>% mutate(rm=num_genome-ps-re) %>% mutate(complete=100*(re/num_genome))%>% mutate(lost=100*(rm/num_genome))%>% mutate(pseudogenes=100*(ps/num_genome))

# write_csv(cluster_ps_counts_wide,"/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/genomes_ps.clustered.txt")

library(ggtern)

# ggtern(data=cluster_ps_final_cleaned_count_wide_final,aes(x=complete,y=lost, z=pseudogenes,fill=total_genomes,color=total_genomes)) +
  # geom_point()
dim(cluster_ps_counts_gene_counts)
pseudogenization_plot <- ggtern(data=cluster_ps_counts_gene_counts,aes(x=complete,y=lost, z=pseudogenes,color=species)) +
  geom_point(alpha=0.5) +#scale_size(trans = 'reverse')+
  # labs(title="Population structure, 2015") +
  theme_rgbw()

pseudogenization_plot
pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_all.pdf",width=11,height=9)

pseudogenization_plot

dev.off()
##------------------------------
#box plot
##------------------------------
cluster_ps_final_cleaned_count_wide_final_smaller <- cluster_ps_counts_gene_counts %>% select(cluster,pseudogenes,complete,lost)%>% gather(.,grouping,percents,c(pseudogenes,complete,lost))
 cluster_ps_final_cleaned_count_wide_final_smaller$grouping = factor(cluster_ps_final_cleaned_count_wide_final_smaller$grouping, levels=c("lost","pseudogenes","complete"))

pseudogenization_boxplot <- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()+facet_wrap(~species)
pseudogenization_boxplot

pseudogenization_boxplot_all <- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()
pseudogenization_boxplot_all



pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_violin_all.pdf",width=11,height=9)

pseudogenization_boxplot_all

dev.off()

##----------------------
#grouping
##----------------------



cluster_ps_counts_gene_counts_extended <- cluster_ps_counts_gene_counts %>% mutate(category=ifelse(lost>60&complete<40&pseudogenes<40,"old",ifelse(lost<40&complete>60&pseudogenes<40,"recent",ifelse(lost<40&complete<40&pseudogenes>60,"maintained","intermediate"))))

table(cluster_ps_counts_gene_counts_extended$category)


# ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=species, group=category,fill=category))+geom_bar(position = "dodge")+theme_classic()


# cluster_ps_counts_gene_counts_extended$species <- plyr::revalue(cluster_ps_final_cleaned_count_wide_final_species_extended$species,c( "Ldel"="L. delbrueckii",
 # "Sterm"="S. thermophilus"))

# ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=interaction(species,category), group=category,fill=category))+geom_bar(position = "dodge")+theme_classic()
genes_time_pseudogenization <- ggplot(cluster_ps_counts_gene_counts_extended,aes(x=species, group=category,fill=category))+geom_bar(position = "dodge")+theme_classic()+facet_wrap(~category,nrow = 1)+theme(axis.title.x = element_blank(),legend.position = "none",axis.text.x = element_text(angle=75,hjust=1))+labs(y="pseudogene gene family")

genes_time_pseudogenization

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing_all.pdf",width=7,height=3)

genes_time_pseudogenization

dev.off()


##----------------------------------
#fraction of genomes containing this pseudogene
# cluster_ps_final_cleaned_count_wide_final_species$species <- plyr::revalue(cluster_ps_final_cleaned_count_wide_final_species$species,c( "Ldel"="L. delbrueckii",
#  "Sterm"="S. thermophilus"))
# distribution_pseudogenes <- ggplot(cluster_ps_final_cleaned_count_wide_final_species,aes(x=pseudogenes,fill=species))+geom_histogram()+theme_classic()+facet_wrap(~species,nrow=2,scales="free_y")+labs(x="fraction of genomes containing the pseudogene",y="number of pseudogene containing gene families")+theme(legend.position="none")+ scale_fill_manual(values =colorsss )
# 
# distribution_pseudogenes
# 
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing_02.pdf",width=7,height=6)
# 
# distribution_pseudogenes
# 
# dev.off()


```

## pseudogenecluster with orthofinder

```{bash}


###-----------------------
#translate
###-----------------------

for speciesss in $(echo "Sterm Ldel Lhelv")
do

rm -r /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}/FAA
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}/FAA

rm /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/all.all.ffn

ll /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/
ls /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/ |wc -l


#specizzzz=$(grep ${speciesss} /data/Project/2020_StarterCultureDiversity/log.species|cut -f 2)
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/ |grep ".all.ffn" |sed 's/.all.ffn//g' |grep -v "all")
do

 transeq -sformat pearson -sequence /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn -outseq /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FAA/${genomesss}.all.faa

sed -i 's/_1$//g' /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FAA/${genomesss}.all.faa

  done


ll /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FAA/
done #species




###-----------------------
#cluster
###-----------------------


for speciesss in $(echo "Lhelv Sterm Ldel ")
do

rm -r /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/

    orthofinder -f /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FAA/ \
      -t 35 \
      -o /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/${speciesss}/ \
      -a 35

done #species



#--------------------------------------
#check duplicate gene names
#--------------------------------------


for speciesss in $(echo "Sterm Ldel Lhelv")
do

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FNA/ |grep ".all.ffn" |sed 's/.all.ffn//g' |grep -v "all")
do

#grep ">" /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FAA/${genomesss}.all.faa | cut -d '|' -f 2 |sort|uniq -c |sort |sed 's/^[[:blank:]]*//;s/[[:blank:]]*$//' |grep "^2" -c


#grep ">" /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FAA/${genomesss}.all.faa |sort|uniq -c |sort |sed 's/^[[:blank:]]*//;s/[[:blank:]]*$//' |grep "^2" -c
grep ">" /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FAA/${genomesss}.all.faa |sort|uniq -c |sort |sed 's/^[[:blank:]]*//;s/[[:blank:]]*$//' |grep "^1" -v 
  done
done


##---------------------
#extract the files
##---------------------
/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Lhelv/Results_Apr04/Orthogroups/

/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Lhelv/Results_Apr04/Orthogroups/Orthogroups.GeneCount.tsv




  mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/
  scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Lhelv/Results_Apr04/Orthogroups/Orthogroups.tsv /home/vincent/Desktop/Pr
  
  
  
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps
scp  vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Lhelv/Results_Apr04/Orthogroups/Orthogroups.tsv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Orthogroups_Lhelv.tsv

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps
scp  vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Sterm/Results_Apr04/Orthogroups/Orthogroups.tsv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Orthogroups_Sterm.tsv

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps
scp  vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Ldel/Results_Apr04/Orthogroups/Orthogroups.tsv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Orthogroups_Ldel.tsv



```



```{bash}
##========================================
#get OG sequences for eggnog
##========================================

/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Lhelv

#rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm_OG_genes.faa
for species in $(echo "Lhelv Ldel Sterm")
do
rm /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}_OG_genes.faa

echo $species

firstletter=${species:0:2}

#done
date=$(ls //data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}/ |grep "Results")

for genesss in $(ls //data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}/${date}/Orthogroup_Sequences|grep ".fa$" |sed 's/.fa//g')
do
echo $genesss

#perl -00 -ne '/(>[^>]+)/ && print $1' /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroup_Sequences/${genesss}.fa |sed "s/>.*$/>S_${genesss}/g" >> /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm_OG_genes.faa

awk '/^>/{if(N)exit;++N;} {print;}' /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}/${date}/Orthogroup_Sequences/${genesss}.fa |sed "s/>.*$/>${firstletter}_${genesss}/g" >> /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}_OG_genes.faa


done #genesss
ls /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}/${date}/Orthogroup_Sequences/ |grep ".fa$" -c
grep ">" -c  /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}_OG_genes.faa

done #species

grep ">"  //data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Sterm_OG_genes.faa |head 
grep ">"  /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//Ldel_OG_genes.faa |head 
grep ">"  /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//Lhelv_OG_genes.faa |head 

##========================================
#bring local for eggnog
##========================================

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/forEGGNOG_pangenome
scp -r vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/*_genes.faa /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/

```





```{r}
#----------------------------------------------
#Lhelv
#----------------------------------------------

library(tidyverse)
Orthogroups_orthofinder <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Orthogroups_Lhelv.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE) %>% rename("Orthogroup"="...1")

sample_column <- 2

Orthogroups_orthofinder_stats <- data.frame()
for (sample_column in 2:ncol(Orthogroups_orthofinder)) {
  
  sampleNAME <- colnames(Orthogroups_orthofinder)[sample_column]
  print(sampleNAME)
  for (Orthogroupsss in 1:nrow(Orthogroups_orthofinder)) {
    
    
      OrthoNAME <- (Orthogroups_orthofinder)[Orthogroupsss,"Orthogroup"] %>% as.character()
tmp1 <- Orthogroups_orthofinder[which(Orthogroups_orthofinder$Orthogroup==OrthoNAME),sampleNAME] %>% rename("focal"=sampleNAME)%>%   mutate(focal = strsplit(as.character(focal), ", ")) %>% 
    unnest(focal)

count_re <- tmp1 %>% filter(grepl("^re",focal)) %>% nrow() %>% ifelse(.>0,.,"0")
count_ps <- tmp1 %>% filter(grepl("^ps",focal))%>% nrow() %>% ifelse(.>0,.,"0")


tmp2 <- data.frame(Orthogroup=OrthoNAME,sample=sampleNAME,ps=count_ps,re=count_re)

Orthogroups_orthofinder_stats <- rbind(Orthogroups_orthofinder_stats,tmp2)
  }
  
}

dim(Orthogroups_orthofinder_stats)
table(Orthogroups_orthofinder_stats$sample)

Orthogroups_orthofinder_stats_test <- Orthogroups_orthofinder_stats %>% as_tibble() %>% mutate(ps=as.numeric(ps))%>% mutate(re=as.numeric(re))

write_csv(Orthogroups_orthofinder_stats_test,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered_counts.txt")
Orthogroups_orthofinder_stats_test <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered_counts.txt")


##----------duplicated orthogroups

Orthogroups_orthofinder_stats_cum <- Orthogroups_orthofinder_stats_test %>% mutate(all_gene_counts=ps+re) %>% group_by(Orthogroup) %>% summarise(numGenes=max(all_gene_counts))
tmp_for_plotting <- table(Orthogroups_orthofinder_stats_cum$numGenes) %>% as.data.frame() %>% as_tibble()

# Orthogroups_orthofinder_stats_cum <- Orthogroups_orthofinder_stats_test %>% filter(ps>0)%>% mutate(all_gene_counts=ps+re) %>% group_by(Orthogroup) %>% summarise(numGenes=max(all_gene_counts))
# tmp_for_plotting <- table(Orthogroups_orthofinder_stats_cum$numGenes) %>% as.data.frame() %>% as_tibble()
numGenomes <- Orthogroups_orthofinder_stats_test %>% select(sample) %>% unique() %>% nrow()

# %>% mutate(num_genome=)
# %>% mutate(all_gene_counts=ps+re) 
OGs_to_keep <- Orthogroups_orthofinder_stats_cum %>% filter(numGenes==1) %>% pull(Orthogroup)
Orthogroups_orthofinder_stats_test_keepers_01 <- Orthogroups_orthofinder_stats_test %>% filter(Orthogroup %in% OGs_to_keep)  %>% gather(.,gene_type,percents,c(ps,re)) %>% filter(percents>0)  %>% mutate(genome=gsub(".all","",sample)) %>% select(-percents,-sample) %>% spread(., genome, gene_type)  %>%replace(is.na(.), "rm")
cluster_ps_counts_wide_imported_wide_final <-  Orthogroups_orthofinder_stats_test_keepers_01 %>%  gather(., genome,grouping , colnames(Orthogroups_orthofinder_stats_test_keepers_01)[2:ncol(Orthogroups_orthofinder_stats_test_keepers_01)], factor_key=TRUE,na.rm = TRUE)  %>% mutate(species="Lhelv") #%>% filter(grouping!="re")
write_csv(cluster_ps_counts_wide_imported_wide_final,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_cluster_ps_counts_wide_imported_wide_final")
table(cluster_ps_counts_wide_imported_wide_final$grouping)


Orthogroups_orthofinder_stats_test_keepers <- Orthogroups_orthofinder_stats_test %>% filter(Orthogroup %in% OGs_to_keep) %>% gather(.,gene_type,percents,c(ps,re)) %>% filter(percents>0) %>% select(-percents) %>% group_by(Orthogroup,gene_type) %>% summarise(counts=n())%>% spread(., gene_type, counts)%>%replace(is.na(.), 0) %>% mutate(num_genome=numGenomes) %>% mutate(rm=(num_genome-ps-re)) %>% mutate(complete=100*(re/num_genome)) %>% mutate(lost=100*(rm/num_genome)) %>% mutate(pseudogenes=100*(ps/num_genome)) %>% mutate(species="Lhelv") %>% filter(ps>0)
  
write_csv(Orthogroups_orthofinder_stats_test_keepers,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered.txt")

Orthogroups_orthofinder_stats_test_keepers_Ldel <- Orthogroups_orthofinder_stats_test_keepers
# Orthogroups_orthofinder_stats_test_keepers$Orthogroup %>% unique()
  

#----------------------------------------------
#Sterm
#----------------------------------------------


library(tidyverse)
Orthogroups_orthofinder <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Orthogroups_Sterm.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE) %>% rename("Orthogroup"="...1")

sample_column <- 2

Orthogroups_orthofinder_stats <- data.frame()
for (sample_column in 2:ncol(Orthogroups_orthofinder)) {
  
  sampleNAME <- colnames(Orthogroups_orthofinder)[sample_column]
  print(sampleNAME)
  for (Orthogroupsss in 1:nrow(Orthogroups_orthofinder)) {
    
    
      OrthoNAME <- (Orthogroups_orthofinder)[Orthogroupsss,"Orthogroup"] %>% as.character()
tmp1 <- Orthogroups_orthofinder[which(Orthogroups_orthofinder$Orthogroup==OrthoNAME),sampleNAME] %>% rename("focal"=sampleNAME)%>%   mutate(focal = strsplit(as.character(focal), ", ")) %>% 
    unnest(focal)

count_re <- tmp1 %>% filter(grepl("^re",focal)) %>% nrow() %>% ifelse(.>0,.,"0")
count_ps <- tmp1 %>% filter(grepl("^ps",focal))%>% nrow() %>% ifelse(.>0,.,"0")


tmp2 <- data.frame(Orthogroup=OrthoNAME,sample=sampleNAME,ps=count_ps,re=count_re)

Orthogroups_orthofinder_stats <- rbind(Orthogroups_orthofinder_stats,tmp2)
  }
  
}

dim(Orthogroups_orthofinder_stats)
table(Orthogroups_orthofinder_stats$sample)

Orthogroups_orthofinder_stats_test <- Orthogroups_orthofinder_stats %>% as_tibble() %>% mutate(ps=as.numeric(ps))%>% mutate(re=as.numeric(re))

write_csv(Orthogroups_orthofinder_stats_test,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered_counts.txt")
Orthogroups_orthofinder_stats_test <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered_counts.txt")


##----------duplicated orthogroups

Orthogroups_orthofinder_stats_cum <- Orthogroups_orthofinder_stats_test %>% mutate(all_gene_counts=ps+re) %>% group_by(Orthogroup) %>% summarise(numGenes=max(all_gene_counts))
tmp_for_plotting <- table(Orthogroups_orthofinder_stats_cum$numGenes) %>% as.data.frame() %>% as_tibble()

# Orthogroups_orthofinder_stats_cum <- Orthogroups_orthofinder_stats_test %>% filter(ps>0)%>% mutate(all_gene_counts=ps+re) %>% group_by(Orthogroup) %>% summarise(numGenes=max(all_gene_counts))
# tmp_for_plotting <- table(Orthogroups_orthofinder_stats_cum$numGenes) %>% as.data.frame() %>% as_tibble()
numGenomes <- Orthogroups_orthofinder_stats_test %>% select(sample) %>% unique() %>% nrow()

# %>% mutate(num_genome=)
# %>% mutate(all_gene_counts=ps+re) 
OGs_to_keep <- Orthogroups_orthofinder_stats_cum %>% filter(numGenes==1) %>% pull(Orthogroup)

Orthogroups_orthofinder_stats_test_keepers_01 <- Orthogroups_orthofinder_stats_test %>% filter(Orthogroup %in% OGs_to_keep)  %>% gather(.,gene_type,percents,c(ps,re)) %>% filter(percents>0)  %>% mutate(genome=gsub(".all","",sample)) %>% select(-percents,-sample) %>% spread(., genome, gene_type)  %>%replace(is.na(.), "rm")
cluster_ps_counts_wide_imported_wide_final <-  Orthogroups_orthofinder_stats_test_keepers_01 %>%  gather(., genome,grouping , colnames(Orthogroups_orthofinder_stats_test_keepers_01)[2:ncol(Orthogroups_orthofinder_stats_test_keepers_01)], factor_key=TRUE,na.rm = TRUE)  %>% mutate(species="Sterm") #%>% filter(grouping!="re")
write_csv(cluster_ps_counts_wide_imported_wide_final,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_cluster_ps_counts_wide_imported_wide_final")

table(cluster_ps_counts_wide_imported_wide_final$grouping)

Orthogroups_orthofinder_stats_test_keepers <- Orthogroups_orthofinder_stats_test %>% filter(Orthogroup %in% OGs_to_keep) %>% gather(.,gene_type,percents,c(ps,re)) %>% filter(percents>0) %>% select(-percents) %>% group_by(Orthogroup,gene_type) %>% summarise(counts=n())%>% spread(., gene_type, counts)%>%replace(is.na(.), 0) %>% mutate(num_genome=numGenomes) %>% mutate(rm=(num_genome-ps-re)) %>% mutate(complete=100*(re/num_genome)) %>% mutate(lost=100*(rm/num_genome)) %>% mutate(pseudogenes=100*(ps/num_genome)) %>% mutate(species="Sterm") %>% filter(ps>0)
  
write_csv(Orthogroups_orthofinder_stats_test_keepers,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered.txt")

Orthogroups_orthofinder_stats_test_keepers$Orthogroup %>% unique()
  


#----------------------------------------------
#Ldel
#----------------------------------------------


library(tidyverse)
Orthogroups_orthofinder <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Orthogroups_Ldel.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE) %>% rename("Orthogroup"="...1")

sample_column <- 2

Orthogroups_orthofinder_stats <- data.frame()
for (sample_column in 2:ncol(Orthogroups_orthofinder)) {
  
  sampleNAME <- colnames(Orthogroups_orthofinder)[sample_column]
  print(sampleNAME)
  for (Orthogroupsss in 1:nrow(Orthogroups_orthofinder)) {
    
    
      OrthoNAME <- (Orthogroups_orthofinder)[Orthogroupsss,"Orthogroup"] %>% as.character()
tmp1 <- Orthogroups_orthofinder[which(Orthogroups_orthofinder$Orthogroup==OrthoNAME),sampleNAME] %>% rename("focal"=sampleNAME)%>%   mutate(focal = strsplit(as.character(focal), ", ")) %>% 
    unnest(focal)

count_re <- tmp1 %>% filter(grepl("^re",focal)) %>% nrow() %>% ifelse(.>0,.,"0")
count_ps <- tmp1 %>% filter(grepl("^ps",focal))%>% nrow() %>% ifelse(.>0,.,"0")


tmp2 <- data.frame(Orthogroup=OrthoNAME,sample=sampleNAME,ps=count_ps,re=count_re)

Orthogroups_orthofinder_stats <- rbind(Orthogroups_orthofinder_stats,tmp2)
  }
  
}

dim(Orthogroups_orthofinder_stats)
table(Orthogroups_orthofinder_stats$sample)

Orthogroups_orthofinder_stats_test <- Orthogroups_orthofinder_stats %>% as_tibble() %>% mutate(ps=as.numeric(ps))%>% mutate(re=as.numeric(re))

write_csv(Orthogroups_orthofinder_stats_test,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered_counts.txt")
Orthogroups_orthofinder_stats_test <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered_counts.txt")


##----------duplicated orthogroups

Orthogroups_orthofinder_stats_cum <- Orthogroups_orthofinder_stats_test %>% mutate(all_gene_counts=ps+re) %>% group_by(Orthogroup) %>% summarise(numGenes=max(all_gene_counts))
tmp_for_plotting <- table(Orthogroups_orthofinder_stats_cum$numGenes) %>% as.data.frame() %>% as_tibble()

# Orthogroups_orthofinder_stats_cum <- Orthogroups_orthofinder_stats_test %>% filter(ps>0)%>% mutate(all_gene_counts=ps+re) %>% group_by(Orthogroup) %>% summarise(numGenes=max(all_gene_counts))
# tmp_for_plotting <- table(Orthogroups_orthofinder_stats_cum$numGenes) %>% as.data.frame() %>% as_tibble()
numGenomes <- Orthogroups_orthofinder_stats_test %>% select(sample) %>% unique() %>% nrow()

# %>% mutate(num_genome=)
# %>% mutate(all_gene_counts=ps+re) 
OGs_to_keep <- Orthogroups_orthofinder_stats_cum %>% filter(numGenes==1) %>% pull(Orthogroup)
Orthogroups_orthofinder_stats_test_keepers_01 <- Orthogroups_orthofinder_stats_test %>% filter(Orthogroup %in% OGs_to_keep)  %>% gather(.,gene_type,percents,c(ps,re)) %>% filter(percents>0)  %>% mutate(genome=gsub(".all","",sample)) %>% select(-percents,-sample) %>% spread(., genome, gene_type)  %>%replace(is.na(.), "rm")
cluster_ps_counts_wide_imported_wide_final <-  Orthogroups_orthofinder_stats_test_keepers_01 %>%  gather(., genome,grouping , colnames(Orthogroups_orthofinder_stats_test_keepers_01)[2:ncol(Orthogroups_orthofinder_stats_test_keepers_01)], factor_key=TRUE,na.rm = TRUE)  %>% mutate(species="Ldel") #%>% filter(grouping!="re")

# write_csv(cluster_ps_counts_wide_imported_wide_final,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_cluster_ps_counts_wide_imported_wide_final")

write_csv(cluster_ps_counts_wide_imported_wide_final,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_cluster_ps_counts_wide_imported_wide_final")

table(cluster_ps_counts_wide_imported_wide_final$grouping)


Orthogroups_orthofinder_stats_test_keepers <- Orthogroups_orthofinder_stats_test %>% filter(Orthogroup %in% OGs_to_keep) %>% gather(.,gene_type,percents,c(ps,re)) %>% filter(percents>0) %>% select(-percents) %>% group_by(Orthogroup,gene_type) %>% summarise(counts=n())%>% spread(., gene_type, counts)%>%replace(is.na(.), 0) %>% mutate(num_genome=numGenomes) %>% mutate(rm=(num_genome-ps-re)) %>% mutate(complete=100*(re/num_genome)) %>% mutate(lost=100*(rm/num_genome)) %>% mutate(pseudogenes=100*(ps/num_genome)) %>% mutate(species="Ldel") %>% filter(ps>0)
  
write_csv(Orthogroups_orthofinder_stats_test_keepers,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt")

Orthogroups_orthofinder_stats_test_keepers$Orthogroup %>% unique()
```


```{r}
library(tidyverse)

Ldel_genomes_ps.clustered.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt")
table(Ldel_genomes_ps.clustered.txt$complete)
Sterm_genomes_ps.clustered.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered.txt")
Lhelv_genomes_ps.clustered.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered.txt")

sum(Sterm_genomes_ps.clustered.txt$ps)
sum(Sterm_genomes_ps.clustered.txt$re)
sum(Sterm_genomes_ps.clustered.txt$rm)
table(Sterm_genomes_ps.clustered.txt$pseudogenes)

Orthogroups_orthofinder_stats_test_keepers <- rbind(Ldel_genomes_ps.clustered.txt, Sterm_genomes_ps.clustered.txt,Lhelv_genomes_ps.clustered.txt)

Orthogroups_orthofinder_stats_test_keepers

cluster_ps_counts_gene_counts <- cluster_ps_counts_wide %>% group_by(species,cluster,gene_presence) %>% summarise(countz=n()) %>% spread(., gene_presence, countz)  %>%replace(is.na(.), 0) %>% left_join(.,genome_count,by="species") %>% mutate(rm=num_genome-ps-re) %>% mutate(complete=100*(re/num_genome))%>% mutate(lost=100*(rm/num_genome))%>% mutate(pseudogenes=100*(ps/num_genome))

# write_csv(cluster_ps_counts_wide,"/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/genomes_ps.clustered.txt")


##for pseudogene
Ldel_genomes_ps.clustered_counts.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered_counts.txt") %>% mutate(species="Ldel")
Orthogroups_orthofinder_stats_cum <- Ldel_genomes_ps.clustered_counts.txt %>% mutate(all_gene_counts=ps+re) %>% group_by(species,Orthogroup) %>% summarise(numGenes=max(all_gene_counts))
tmp_for_plotting <- table(Orthogroups_orthofinder_stats_cum$numGenes) %>% as.data.frame() %>% as_tibble()
OGs_to_keep <- Orthogroups_orthofinder_stats_cum %>% filter(numGenes==1) %>% pull(Orthogroup)
Ldel_genomes_ps.clustered_counts.txt_filtered <- Ldel_genomes_ps.clustered_counts.txt %>% filter(Orthogroup %in% unique(Ldel_genomes_ps.clustered.txt$Orthogroup))

Sterm_genomes_ps.clustered_counts.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered_counts.txt") %>% mutate(species="Sterm")
Orthogroups_orthofinder_stats_cum <- Sterm_genomes_ps.clustered_counts.txt %>% mutate(all_gene_counts=ps+re) %>% group_by(species,Orthogroup) %>% summarise(numGenes=max(all_gene_counts))
tmp_for_plotting <- table(Orthogroups_orthofinder_stats_cum$numGenes) %>% as.data.frame() %>% as_tibble()
OGs_to_keep <- Orthogroups_orthofinder_stats_cum %>% filter(numGenes==1) %>% pull(Orthogroup)
Sterm_genomes_ps.clustered_counts.txt_filtered <- Sterm_genomes_ps.clustered_counts.txt %>% filter(Orthogroup %in% unique(Sterm_genomes_ps.clustered.txt$Orthogroup))

Lhelv_genomes_ps.clustered_counts.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered_counts.txt") %>% mutate(species="Lhelv")
Orthogroups_orthofinder_stats_cum <- Lhelv_genomes_ps.clustered_counts.txt %>% mutate(all_gene_counts=ps+re) %>% group_by(species,Orthogroup) %>% summarise(numGenes=max(all_gene_counts))
tmp_for_plotting <- table(Orthogroups_orthofinder_stats_cum$numGenes) %>% as.data.frame() %>% as_tibble()
OGs_to_keep <- Orthogroups_orthofinder_stats_cum %>% filter(numGenes==1) %>% pull(Orthogroup)
Lhelv_genomes_ps.clustered_counts.txt_filtered <- Lhelv_genomes_ps.clustered_counts.txt %>% filter(Orthogroup %in% unique(Lhelv_genomes_ps.clustered.txt$Orthogroup))

Orthogroups_orthofinder_stats_test_keepers_counts <- rbind(Ldel_genomes_ps.clustered_counts.txt_filtered, Sterm_genomes_ps.clustered_counts.txt_filtered,Lhelv_genomes_ps.clustered_counts.txt_filtered) %>% mutate(genome=gsub(".all","",sample)) %>% mutate(cluster=Orthogroup)%>% mutate(gene_presence=ifelse(re>0,"re","ps")) %>% select(species, cluster ,  genome  , ps,re, gene_presence)

write_csv(Orthogroups_orthofinder_stats_test_keepers_counts,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//genomes_ps.clustered.txt")

##=================================================
#Plots
##=================================================

library(ggtern)

# ggtern(data=cluster_ps_final_cleaned_count_wide_final,aes(x=complete,y=lost, z=pseudogenes,fill=total_genomes,color=total_genomes)) +
  # geom_point()
dim(Orthogroups_orthofinder_stats_test_keepers)
pseudogenization_plot <- ggtern(data=Orthogroups_orthofinder_stats_test_keepers,aes(x=complete,y=lost, z=pseudogenes,color=species)) +
  geom_point(alpha=0.5) +#scale_size(trans = 'reverse')+
  # labs(title="Population structure, 2015") +
  scale_color_manual(values=c("#fdc2b5ff","#b3b3b3ff","#007fa6ff"))+
  theme_rgbw()

pseudogenization_plot
pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_all.pdf",width=8,height=6)

pseudogenization_plot

dev.off()
 
##------------------------------
#box plot
##------------------------------

# cluster_ps_final_cleaned_count_wide_final_smaller <- cluster_ps_counts_gene_counts %>% select(cluster,pseudogenes,complete,lost)%>% gather(.,grouping,percents,c(pseudogenes,complete,lost))

cluster_ps_final_cleaned_count_wide_final_smaller <- Orthogroups_orthofinder_stats_test_keepers %>% select(Orthogroup,pseudogenes,complete,lost,species)%>% gather(.,grouping,percents,c(pseudogenes,complete,lost))
 cluster_ps_final_cleaned_count_wide_final_smaller$grouping = factor(cluster_ps_final_cleaned_count_wide_final_smaller$grouping, levels=c("lost","pseudogenes","complete"))

pseudogenization_boxplot <- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()+facet_wrap(~species)
pseudogenization_boxplot

pseudogenization_boxplot_all <- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()
pseudogenization_boxplot_all



pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_violin_all.pdf",width=8,height=7)

pseudogenization_boxplot_all

dev.off()

##----------------------
#grouping
##----------------------


cluster_ps_counts_gene_counts_extended <- Orthogroups_orthofinder_stats_test_keepers %>% mutate(category=ifelse(lost>60&complete<40&pseudogenes<40,"old",ifelse(lost<40&complete>60&pseudogenes<40,"recent",ifelse(lost<40&complete<40&pseudogenes>60,"maintained","intermediate"))))

table(cluster_ps_counts_gene_counts_extended$category)


genes_time_pseudogenization <- ggplot(cluster_ps_counts_gene_counts_extended,aes(x=species, group=category,fill=category))+geom_bar(position = "dodge")+theme_classic()+facet_wrap(~category,nrow = 1)+theme(axis.title.x = element_blank(),legend.position = "none",axis.text.x = element_text(angle=75,hjust=1))+labs(y="pseudogene gene family")

genes_time_pseudogenization

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing_all.pdf",width=7,height=3)

genes_time_pseudogenization

dev.off()
 

# write_csv(Orthogroups_orthofinder_stats_test_keepers,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt")


```


```{r}

library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)

##------------------------------
#get data clusters
##------------------------------
cluster_ps <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/ALL_pseudo_genes_cluster_ps.txt",  "\t", escape_double = FALSE, col_names = c("species","cluster","gene"),  trim_ws = TRUE)

cluster_ps$genome <- str_split_fixed(cluster_ps$gene,"_",2)[,1] %>% gsub(".*:","",.)

genome_count <- cluster_ps %>% select(species, genome) %>% unique() %>% group_by(species) %>% summarise(num_genome=n())

cluster_ps$gene_type <- gsub(":.*","",cluster_ps$gene)

cluster_ps_counts <- cluster_ps %>% group_by(species,cluster,genome,gene_type) %>% summarise(countss=n())
table(cluster_ps_counts$countss)

cluster_ps_counts_wide <- cluster_ps_counts %>% spread(., gene_type, countss)  %>%replace(is.na(.), 0) %>% mutate(gene_presence=ifelse(re>0,"re","ps"))


cluster_ps_counts_gene_counts <- cluster_ps_counts_wide %>% group_by(species,cluster,gene_presence) %>% summarise(countz=n()) %>% spread(., gene_presence, countz)  %>%replace(is.na(.), 0) %>% left_join(.,genome_count,by="species") %>% mutate(rm=num_genome-ps-re) %>% mutate(complete=100*(re/num_genome))%>% mutate(lost=100*(rm/num_genome))%>% mutate(pseudogenes=100*(ps/num_genome))

write_csv(cluster_ps_counts_wide,"/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/genomes_ps.clustered.txt")

library(ggtern)

# ggtern(data=cluster_ps_final_cleaned_count_wide_final,aes(x=complete,y=lost, z=pseudogenes,fill=total_genomes,color=total_genomes)) +
  # geom_point()
dim(cluster_ps_counts_gene_counts)
pseudogenization_plot <- ggtern(data=cluster_ps_counts_gene_counts,aes(x=complete,y=lost, z=pseudogenes,color=species)) +
  geom_point(alpha=0.5) +#scale_size(trans = 'reverse')+
  # labs(title="Population structure, 2015") +
  theme_rgbw()

pseudogenization_plot
pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_all.pdf",width=11,height=9)

pseudogenization_plot

dev.off()
##------------------------------
#box plot
##------------------------------
cluster_ps_final_cleaned_count_wide_final_smaller <- cluster_ps_counts_gene_counts %>% select(cluster,pseudogenes,complete,lost)%>% gather(.,grouping,percents,c(pseudogenes,complete,lost))
 cluster_ps_final_cleaned_count_wide_final_smaller$grouping = factor(cluster_ps_final_cleaned_count_wide_final_smaller$grouping, levels=c("lost","pseudogenes","complete"))

pseudogenization_boxplot <- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()+facet_wrap(~species)
pseudogenization_boxplot

pseudogenization_boxplot_all <- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()
pseudogenization_boxplot_all



pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_violin_all.pdf",width=11,height=9)

pseudogenization_boxplot_all

dev.off()

##----------------------
#grouping
##----------------------



cluster_ps_counts_gene_counts_extended <- cluster_ps_counts_gene_counts %>% mutate(category=ifelse(lost>60&complete<40&pseudogenes<40,"old",ifelse(lost<40&complete>60&pseudogenes<40,"recent",ifelse(lost<40&complete<40&pseudogenes>60,"maintained","intermediate"))))

table(cluster_ps_counts_gene_counts_extended$category)


# ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=species, group=category,fill=category))+geom_bar(position = "dodge")+theme_classic()


# cluster_ps_counts_gene_counts_extended$species <- plyr::revalue(cluster_ps_final_cleaned_count_wide_final_species_extended$species,c( "Ldel"="L. delbrueckii",
 # "Sterm"="S. thermophilus"))

# ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=interaction(species,category), group=category,fill=category))+geom_bar(position = "dodge")+theme_classic()
genes_time_pseudogenization <- ggplot(cluster_ps_counts_gene_counts_extended,aes(x=species, group=category,fill=category))+geom_bar(position = "dodge")+theme_classic()+facet_wrap(~category,nrow = 1)+theme(axis.title.x = element_blank(),legend.position = "none",axis.text.x = element_text(angle=75,hjust=1))+labs(y="pseudogene gene family")

genes_time_pseudogenization

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing_all.pdf",width=7,height=3)

genes_time_pseudogenization

dev.off()


##----------------------------------
#fraction of genomes containing this pseudogene
# cluster_ps_final_cleaned_count_wide_final_species$species <- plyr::revalue(cluster_ps_final_cleaned_count_wide_final_species$species,c( "Ldel"="L. delbrueckii",
#  "Sterm"="S. thermophilus"))
# distribution_pseudogenes <- ggplot(cluster_ps_final_cleaned_count_wide_final_species,aes(x=pseudogenes,fill=species))+geom_histogram()+theme_classic()+facet_wrap(~species,nrow=2,scales="free_y")+labs(x="fraction of genomes containing the pseudogene",y="number of pseudogene containing gene families")+theme(legend.position="none")+ scale_fill_manual(values =colorsss )
# 
# distribution_pseudogenes
# 
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing_02.pdf",width=7,height=6)
# 
# distribution_pseudogenes
# 
# dev.off()


```

# eggnog on ps

```{r}
library(corrplot)
library(tidyverse)
library(RColorBrewer)

Lhelv_Eggnog <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv.Eggnog.annotations", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE, skip = 4)%>% filter(grepl("^Lh",`#query`))%>% mutate(species="Lhelv")

Ldel_Eggnog <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel.Eggnog.annotations", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE, skip = 4)%>% filter(grepl("^Ld",`#query`))%>% mutate(species="Ldel")

Sterm_Eggnog <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm.Eggnog.annotations", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE, skip = 4) %>% filter(grepl("^St",`#query`)) %>% mutate(species="Sterm") 

tail(Lhelv_Eggnog)
all_eggnog <- rbind(Sterm_Eggnog,Ldel_Eggnog,Lhelv_Eggnog)


COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA") %>% select(-divided)

# unique(eggnog_annotation$COG_Functional_Category)[which(!unique(eggnog_annotation$COG_Functional_Category) %in% COG_functional_categories$short )]

ALL_sterm <- eggnog_annotation
nrow(ALL_sterm)

all_eggnog_long <- all_eggnog %>% 
    mutate(COG_category = strsplit(as.character(COG_category), "")) %>% 
    unnest(COG_category)


all_eggnog_long_extended <- merge(all_eggnog_long,COG_functional_categories,by.x = "COG_category",by.y="short")
table(all_eggnog_long_extended$long)


Ldel_genomes_ps.clustered.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt") %>% mutate(Orthogroup=paste0("Ld_",Orthogroup)) %>% select(Orthogroup,complete,lost,pseudogenes  )
# table(Ldel_genomes_ps.clustered.txt$complete)
Sterm_genomes_ps.clustered.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered.txt")%>% mutate(Orthogroup=paste0("St_",Orthogroup))%>% select(Orthogroup,complete,lost,pseudogenes  )
Lhelv_genomes_ps.clustered.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered.txt")%>% mutate(Orthogroup=paste0("Lh_",Orthogroup))%>% select(Orthogroup,complete,lost,pseudogenes  )

all_duplicated_OGs <- rbind(Ldel_genomes_ps.clustered.txt,Sterm_genomes_ps.clustered.txt,Lhelv_genomes_ps.clustered.txt)

all_info <- merge(all_eggnog_long_extended,all_duplicated_OGs,by.x="#query",by.y="Orthogroup",all.x = TRUE) %>% as_tibble()
is.na(all_info$long) %>% sum()

all_categories <- all_info %>% group_by(long,species) %>% summarise(counts=n()) %>% mutate(cogs="all")
only_ps_categories <- all_info %>% filter(pseudogenes>0)%>% group_by(long,species) %>% summarise(counts=n())%>% mutate(cogs="pseudo")
complete_cogs_ps <- rbind(all_categories,only_ps_categories)

###----------------------------------
#extract names to check manually
###----------------------------------


only_ps_categories_for_Philipp <- all_info %>% filter(pseudogenes>0)
only_ps_categories_for_Philipp$species <- str_split_fixed(only_ps_categories_for_Philipp$`#query`,"_",2)[,1] 
colnames(only_ps_categories_for_Philipp)
only_ps_categories_for_Philipp_final <- only_ps_categories_for_Philipp %>% select("#query" ,"complete"   ,    "lost"      ,     "pseudogenes" ,"species" ,"Preferred_name","COG_category","long" ,"Description","GOs" ,"EC" , "KEGG_ko" ,  "KEGG_Pathway" ,"KEGG_Module","KEGG_Reaction","KEGG_rclass" ,"BRITE" , "KEGG_TC","CAZy" ,  "BiGG_Reaction" , "PFAMs" ,"eggNOG_OGs" , "max_annot_lvl"         ) %>% filter(complete<90)

table(only_ps_categories_for_Philipp_final$species)
                 

write.table(only_ps_categories_for_Philipp_final,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/All_pseudogenes_annotation.csv",quote = FALSE,row.names = FALSE,sep = "\t")

###----------------------------------
#completeness of OG (core,cloud info)
###----------------------------------
Lhelv_OG_abundance <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_cluster_ps_counts_wide_imported_wide_final") %>% mutate(Orthogroup=paste0("Lh_",Orthogroup))
Lhelv_OG_abundancesums <- Lhelv_OG_abundance %>% group_by(Orthogroup,grouping,species)  %>% summarise(counts=n())%>% spread(., grouping, counts)%>%replace(is.na(.), 0)
total_gene <- rowSums(Lhelv_OG_abundancesums[,c("ps","re","rm")]) %>% head(1) %>% as.numeric()
Lhelv_OG_abundancesums$coreness <- 100*(Lhelv_OG_abundancesums$re/total_gene)
lhelv_coreness <- Lhelv_OG_abundancesums %>% select(Orthogroup,coreness)

###----------------------------------
#extract names of complete and pseudogenized genes for KEGG map
###----------------------------------

colnames(all_info)
all_info_KEGG <- all_info %>% select("#query","KEGG_ko", "complete"   ,"lost" ,"pseudogenes") %>% mutate(KEGG_Reaction=gsub("ko:","",KEGG_ko)) %>% select(-KEGG_ko)
all_info_KEGG$species <- str_split_fixed(all_info_KEGG$`#query`,"_",2)[,1] 
all_info_KEGG$category <- ifelse(is.na(all_info_KEGG$complete),"green",ifelse(all_info_KEGG$complete>90,"orange","red"))
all_info_KEGG_yes <- all_info_KEGG %>% filter(KEGG_Reaction!="-")

all_info_KEGG_yes_cleaned <- all_info_KEGG_yes %>% 
    mutate(KEGG_Reaction = strsplit(as.character(KEGG_Reaction), ",")) %>% 
    unnest(KEGG_Reaction)


#----------Ldel

all_info_KEGG_yes_cleaned_Ldel <- all_info_KEGG_yes_cleaned %>% filter(species=="Ld") %>% select(KEGG_Reaction,category)
all_info_KEGG_yes_cleaned_Ldel[duplicated(all_info_KEGG_yes_cleaned_Ldel$KEGG_Reaction),]
final_KEGG <- data.frame()

for (genesss in unique(all_info_KEGG_yes_cleaned_Ldel$KEGG_Reaction)) {
  
  tmp1 <- all_info_KEGG_yes_cleaned_Ldel %>% filter(KEGG_Reaction==genesss)
  greenezzz <- tmp1 %>% filter(tmp1$category=="green") %>% nrow
  orangezzz <- tmp1 %>% filter(tmp1$category=="orange") %>% nrow
  redzzz <- tmp1 %>% filter(tmp1$category=="red") %>% nrow
  
tmp2 <-   data.frame(KEGG_Reaction=genesss,complete=greenezzz,ps=redzzz,partial_ps=orangezzz)
final_KEGG <- rbind(final_KEGG,tmp2)
}



spread(all_info_KEGG_yes_cleaned_Ldel, category, KEGG_Reaction)  %>%replace(is.na(.), 0)


write.table(all_info_KEGG_yes_cleaned_Ldel,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/All_GENES_KEGGmap_ldel.tsv",quote = FALSE,row.names = FALSE,col.names = FALSE,sep = "\t")



lhelv_coreness

###----------------------------------
#continue
###----------------------------------
Cogs_duplicated_ext <- ggplot(complete_cogs_ps,aes(x=long,y=sort(counts,decreasing = TRUE),fill=cogs))+geom_bar(stat="identity")+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank())+facet_wrap(~species,ncol = 1)
Cogs_duplicated_ext

# cogs_comple_long <- merge(all_cogs,duplicated_cogs,by="long") %>%  mutate(Complete_genes=ALL_counts-Duplicated) %>% dplyr::select(-ALL_counts) %>% gather(.,group,count,c(Complete_genes,Duplicated))

colorsss=rev(c("grey44","grey90"))

all_categories_02 <- all_info %>% group_by(long,species) %>% summarise(all_genes=n())
only_ps_categories_02 <- all_info %>% filter(pseudogenes>0)%>% group_by(long,species) %>% summarise(Pseudo=n())
 all_combined_wide <- all_categories_02 %>% left_join(.,only_ps_categories_02,by=c("long","species")) %>% mutate(Complete_genes=all_genes-Pseudo) %>% select(-all_genes)%>% arrange(desc(Pseudo+Complete_genes))

all_combined_long <- all_combined_wide %>% mutate(Complete_genes=Complete_genes-Pseudo)%>% gather(.,group,count,c(Pseudo,Complete_genes))

all_combined_long$species <- plyr::revalue(all_combined_long$species,c("Ldel"="L. delbrueckii","Sterm"="S. thermophilus","Lhelv"="L. helveticus"))


Cogs_duplicated_ext <- ggplot(all_combined_long,aes(x=reorder(long,-count),y=count,fill=group))+geom_bar(stat="identity")+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1,size=7),axis.title.x = element_blank(),legend.title = element_blank(),legend.position = "none")+scale_fill_manual(values = colorsss)+facet_wrap(~species,ncol = 1,scales = "free_y")
Cogs_duplicated_ext


pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Genes_pseudo_rep.pdf",width=6,height=6)
Cogs_duplicated_ext
dev.off()





for (speciesss in c("Sterm","Ldel","Lhelv")) {
  all_combined_wide_species <- all_combined_wide %>% filter(species==speciesss) %>% select(-species)%>% replace(is.na(.), 0) %>% column_to_rownames(.,var="long")
chisq <- chisq.test(all_combined_wide_species)
dev.off()
plot.new()
col3 <- colorRampPalette(c("red", "white", "blue")) 
corsssplot <- corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = "color")
corsssplot
dev.off()
corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = "color")
}


  ###---------------Sterm
  speciesss <- "Sterm"
 
    all_combined_wide_species <- all_combined_wide %>% filter(species==speciesss) %>% select(-species)%>% replace(is.na(.), 0) %>% column_to_rownames(.,var="long")
chisq <- chisq.test(all_combined_wide_species)
dev.off()
plot.new()
col3 <- colorRampPalette(c("red", "white", "blue")) 
corsssplot <- corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = "color")
corsssplot_Sterm <- corsssplot$corrPos %>% as_tibble() %>% mutate(species="Sterm")

corsssplot
dev.off()
corrplot(chisq$resiStermduals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = "color", title = speciesss)
  

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_pseudo_rep.pdf",width=6,height=10)
corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = "color", title = speciesss)
dev.off()


###---------------Ldel
  speciesss <- "Ldel"
 
    all_combined_wide_species <- all_combined_wide %>% filter(species==speciesss) %>% select(-species)%>% replace(is.na(.), 0) %>% column_to_rownames(.,var="long")
chisq <- chisq.test(all_combined_wide_species)
dev.off()
plot.new()
col3 <- colorRampPalette(c("red", "white", "blue")) 
corsssplot <- corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = "color")
corsssplot_Ldel <- corsssplot$corrPos %>% as_tibble() %>% mutate(species="Ldel")

corsssplot
dev.off()
corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = "color", title = speciesss)
  

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Ldel_pseudo_rep.pdf",width=6,height=10)
corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = "color", title = speciesss)
dev.off()

###---------------Lhelv
  speciesss <- "Lhelv"
 
    all_combined_wide_species <- all_combined_wide %>% filter(species==speciesss) %>% select(-species)%>% replace(is.na(.), 0) %>% column_to_rownames(.,var="long")
chisq <- chisq.test(all_combined_wide_species)
dev.off()
plot.new()
col3 <- colorRampPalette(c("red", "white", "blue")) 
corsssplot <- corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = "color")
corsssplot_Lhelv <- corsssplot$corrPos %>% as_tibble() %>% mutate(species="Lhelv")
dev.off()
corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = "color", title = speciesss)
  

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Lhelv_pseudo_rep.pdf",width=6,height=10)
corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = "color", title = speciesss)
dev.off()

library(patchwork)

# Cogs_duplicated_ext+countsPlot_errorbar+ plot_layout(ncol=2,widths = c(1,3))

##----------------------------
###only illustrate the most and least conserved COG families 
##----------------------------

corsssplot_all <- rbind(corsssplot_Lhelv,corsssplot_Ldel,corsssplot_Sterm) %>% filter(xName=="Pseudo")%>% filter(yName %in% c("Carbohydrate metabolism and transport","Amino acid metabolism and transport","Inorganic ion transport and metabolism","Replication and repair","Translation"))


    # dev.off()
# range(Biolog_diff_new$Value)
ggheatmap <- ggplot(corsssplot_all, aes(y=yName, x=species, fill = corr))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "blue", high = "red", mid = "white",
   midpoint = 0, limit = c(-1,1), space = "Lab",
   name="representation") +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = "top") +
  # labs(x=expression(italic("S. thermophilus")),y=expression(italic("L. delbrueckii")))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())

print(ggheatmap)
  
  


   pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/all_COG_representative_pseudogenes_colors.pdf",width=6,height=8)

  print(ggheatmap)

dev.off()

##--------------------


table(all_combined_long$long)

all_combined_long_sub <- all_combined_long %>% filter(long %in% c("Carbohydrate metabolism and transport","Amino acid metabolism and transport","Inorganic ion transport and metabolism","Replication and repair","Translation")) %>% mutate(grouping=ifelse(long %in% c("Replication and repair","Translation"),"conserved","removed"))

all_combined_long_sub

all_combined_long_sub$long <- plyr::revalue(all_combined_long_sub$long,c("Carbohydrate metabolism and transport"="Carbohydrate\nmetabolism\nand\ntransport",
 "Amino acid metabolism and transport"="Amino acid\nmetabolism\nand\ntransport",
 "Inorganic ion transport and metabolism"="Inorganic ion\ntransport\nand\nmetabolism",
 "Replication and repair"="Replication\nand\nrepair"))
 
# all_combined_long_sub
colorsss=rev(c("grey44","grey95"))


Cogs_duplicated_sub <- ggplot(all_combined_long_sub,aes(x=species,y=count,fill=group))+geom_bar(stat="identity")+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1,size=7),axis.title.x = element_blank(),legend.title = element_blank(),legend.position = "none")+scale_fill_manual(values = colorsss)+facet_wrap(~long,ncol = 5)+labs(y="# Orthogroups")
Cogs_duplicated_sub+ggheatmap
Cogs_duplicated_sub

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/all_COG_representative_pseudogenes.pdf",width=7,height=3.5)
Cogs_duplicated_sub
dev.off()



```

## secretation proteins

```{r}
##-----------------------GO annotation
library(tidyverse)
 All_pseudogenes_annotation_again <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/All_pseudogenes_annotation.csv", 
     delim = "\t", escape_double = FALSE, 
     trim_ws = TRUE)
 
table(All_pseudogenes_annotation_again$GOs) %>% dim()
 
table(is.na(All_pseudogenes_annotation_again$GOs))
table(All_pseudogenes_annotation_again$GOs=="-")


##-----------------------all ps OG

Ldel_genomes_ps.clustered.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt") %>% mutate(Orthogroup=paste0("Ld_",Orthogroup)) %>% select(Orthogroup,complete,lost,pseudogenes  )
# table(Ldel_genomes_ps.clustered.txt$complete)
Sterm_genomes_ps.clustered.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered.txt")%>% mutate(Orthogroup=paste0("St_",Orthogroup))%>% select(Orthogroup,complete,lost,pseudogenes  )
Lhelv_genomes_ps.clustered.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered.txt")%>% mutate(Orthogroup=paste0("Lh_",Orthogroup))%>% select(Orthogroup,complete,lost,pseudogenes  )

all_duplicated_OGs <- rbind(Ldel_genomes_ps.clustered.txt,Sterm_genomes_ps.clustered.txt,Lhelv_genomes_ps.clustered.txt)

##-----------------------EGGNOG

Lhelv_Eggnog <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv.Eggnog.annotations", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE, skip = 4)%>% filter(grepl("^Lh",`#query`))%>% mutate(species="Lhelv")

Ldel_Eggnog <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel.Eggnog.annotations", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE, skip = 4)%>% filter(grepl("^Ld",`#query`))%>% mutate(species="Ldel")

Sterm_Eggnog <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm.Eggnog.annotations", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE, skip = 4) %>% filter(grepl("^St",`#query`)) %>% mutate(species="Sterm") 

tail(Lhelv_Eggnog)
all_eggnog <- rbind(Sterm_Eggnog,Ldel_Eggnog,Lhelv_Eggnog) %>% rename(.,"Orthogroup"="#query")

all_eggnog_peps <- all_eggnog %>% filter(grepl("pep",Preferred_name))

##-----------------------Signal P annotation
Sterm_all_genes_SignalP <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/cell_localization/Sterm_all_genes_SignalP.csv", 
    delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, skip = 1) %>% filter(X3=="signal_peptide") %>% rename(.,"Orthogroup"="X1")%>% left_join(.,all_duplicated_OGs) %>% left_join(.,all_eggnog)

Ldel_all_genes_SignalP <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/cell_localization/Ldel_all_genes_SignalP.csv", 
    delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, skip = 1) %>% filter(X3=="signal_peptide") %>% rename(.,"Orthogroup"="X1")%>% left_join(.,all_duplicated_OGs) %>% left_join(.,all_eggnog)


Lhelv_all_genes_SignalP <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/cell_localization/Lhelv_all_genes_SignalP.csv", 
    delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, skip = 1) %>% filter(X3=="signal_peptide") %>% rename(.,"Orthogroup"="X1")%>% left_join(.,all_duplicated_OGs) %>% left_join(.,all_eggnog)


 
```


## check peptidase

```{r}
library(tidyverse)
# library(plyr)

##-----------------------all ps OG

Ldel_genomes_ps.clustered.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt") %>% mutate(Orthogroup=paste0("Ld_",Orthogroup)) %>% select(Orthogroup,complete,lost,pseudogenes  )
# table(Ldel_genomes_ps.clustered.txt$complete)
Sterm_genomes_ps.clustered.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered.txt")%>% mutate(Orthogroup=paste0("St_",Orthogroup))%>% select(Orthogroup,complete,lost,pseudogenes  )
Lhelv_genomes_ps.clustered.txt <-  read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered.txt")%>% mutate(Orthogroup=paste0("Lh_",Orthogroup))%>% select(Orthogroup,complete,lost,pseudogenes  )

all_duplicated_OGs <- rbind(Ldel_genomes_ps.clustered.txt,Sterm_genomes_ps.clustered.txt,Lhelv_genomes_ps.clustered.txt)

##-----------------------Signal P annotation
Sterm_all_genes_SignalP <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/cell_localization/Sterm_all_genes_SignalP.csv", 
    delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, skip = 1) %>% rename(.,"Orthogroup"="X1") %>% rename(.,"CellularLoco"="X3") #%>% filter(X3=="signal_peptide") %>% left_join(.,all_duplicated_OGs) %>% left_join(.,all_eggnog)

Ldel_all_genes_SignalP <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/cell_localization/Ldel_all_genes_SignalP.csv", 
    delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, skip = 1) %>% rename(.,"Orthogroup"="X1") %>% rename(.,"CellularLoco"="X3") #%>% filter(X3=="signal_peptide") %>% rename(.,"Orthogroup"="X1")%>% left_join(.,all_duplicated_OGs) %>% left_join(.,all_eggnog)


Lhelv_all_genes_SignalP <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/cell_localization/Lhelv_all_genes_SignalP.csv", 
    delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, skip = 1) %>% rename(.,"Orthogroup"="X1") %>% rename(.,"CellularLoco"="X3") #%>% filter(X3=="signal_peptide") %>% rename(.,"Orthogroup"="X1")%>% left_join(.,all_duplicated_OGs) %>% left_join(.,all_eggnog)

all_SignalP<- rbind(Sterm_all_genes_SignalP,Ldel_all_genes_SignalP,Lhelv_all_genes_SignalP) %>% select(Orthogroup,CellularLoco)

##-----------------------EGGNOG

Lhelv_Eggnog <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv.Eggnog.annotations", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE, skip = 4)%>% filter(grepl("^Lh",`#query`))%>% mutate(species="Lhelv")

Ldel_Eggnog <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel.Eggnog.annotations", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE, skip = 4)%>% filter(grepl("^Ld",`#query`))%>% mutate(species="Ldel")

Sterm_Eggnog <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm.Eggnog.annotations", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE, skip = 4) %>% filter(grepl("^St",`#query`)) %>% mutate(species="Sterm") 

tail(Lhelv_Eggnog)
all_eggnog <- rbind(Sterm_Eggnog,Ldel_Eggnog,Lhelv_Eggnog) %>% rename(.,"Orthogroup"="#query")

all_eggnog_peps <- all_eggnog %>% filter(grepl("pep",Preferred_name)) %>% left_join(.,all_duplicated_OGs)%>% left_join(.,all_SignalP)
all_eggnog_prt <- all_eggnog %>% filter(grepl("prt",Preferred_name)) %>% left_join(.,all_duplicated_OGs)%>% left_join(.,all_SignalP)
all_eggnog_prt <- all_eggnog %>% filter(grepl("peptidase",Description)) %>% left_join(.,all_duplicated_OGs)%>% left_join(.,all_SignalP)

all_eggnog_prt <- all_eggnog %>% filter(grepl("protease",Description)) %>% left_join(.,all_duplicated_OGs)%>% left_join(.,all_SignalP)


all_eggnog_peps_presence <- all_eggnog_peps %>% group_by(Preferred_name,PFAMs,species,pseudogenes) %>% summarise(counts=n()) %>% mutate(pseudogenes=ifelse(pseudogenes>0,"yes","no"))%>%replace(is.na(.), "no")%>% select(-counts) %>% unique() %>% mutate(present="yes") %>% spread(., pseudogenes, present)  %>%replace(is.na(.), "noPresent") %>% mutate(final_psuedo=ifelse(yes=="yes","pseudo","complete")) %>% filter(species!="Lhelv")%>% select(-no,-yes)%>% spread(., species, final_psuedo) %>%replace(is.na(.), "missing")

all_eggnog_peps_presence_long <- all_eggnog_peps_presence %>% gather(., species,presence , c("Ldel","Sterm"), factor_key=TRUE,na.rm = TRUE) 

colorss <- c("red","grey88","grey77")

plotPeptidases <- ggplot(all_eggnog_peps_presence_long,aes(x=species,y=Preferred_name,fill=presence,color=presence))+geom_tile()+theme_classic()+scale_fill_manual(values = colorss)+scale_color_manual(values = colorss)+labs(x="",y="Orthogroups Peptidases")+theme(legend.title = element_blank())
png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images//20210126/supplement//Peptidases_presence.png", width = , height = 3000,res=400)

# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement//plot_genome_size.pdf",width=6,height=5)

plotPeptidases

dev.off()
all_eggnog_peps_presence_long %>% group_by(species,presence) %>% summarise(counts=n())

```


## check KEGG pathways

```{r}

# write.table(only_ps_categories_for_Philipp_final,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/All_pseudogenes_annotation.csv",quote = FALSE,row.names = FALSE,sep = "\t")
library(tidyverse)
 All_pseudogenes_annotation_again <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/All_pseudogenes_annotation.csv", 
     delim = "\t", escape_double = FALSE, 
     trim_ws = TRUE)
 
 
carbohydrate_modules_info <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/KEGG_info/carbohydrate_modules_info.csv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE) %>% mutate(KEGG_Module=Module)%>% mutate(long_KEGG_Module=Definition) %>% select(KEGG_Module,long_KEGG_Module)
  
carbohydrate_modules_info_02 <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/KEGG_info/carbohydrate_modules_info_02.csv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE) %>% select(KEGG_Module,KeggModuleName,CategoryGlobal)#%>% mutate(KEGG_Module=Module) %>% mutate(long_KEGG_Module=Definition) 
  

  All_pseudogenes_annotation_again_carbo <- All_pseudogenes_annotation_again %>% filter(COG_category=="G")
 
  
  ###----------------------------
  ##KEGG PATHWAYS
  ###----------------------------

  carbohydrate_pathways_info <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/KEGG_info/carbohydrate_pathways_info.csv", 
    col_names = c("KEGG_Pathway","long_KEGG_Pathway")) %>% mutate(KEGG_Pathway=gsub("^","ko",KEGG_Pathway))
  
# ALL_sterm <- ALL_sterm %>% 
#     mutate(COG_Functional_Category = strsplit(as.character(COG_Functional_Category), "")) %>% 
#     unnest(COG_Functional_Category)
 
 all_info_KEGG_pathway <- All_pseudogenes_annotation_again_carbo %>% select("#query","KEGG_Pathway","Preferred_name","COG_category", "complete"   ,"lost" ,"pseudogenes") %>% mutate(KEGG_Pathway = strsplit(as.character(KEGG_Pathway), ",")) %>%   unnest(KEGG_Pathway)
 
tmp <- all_info_KEGG_pathway %>% filter(grepl("ko",KEGG_Pathway)) %>% left_join(.,carbohydrate_pathways_info) 
all_info_KEGG_pathway %>% filter(grepl("ko",KEGG_Pathway)) %>% left_join(.,carbohydrate_pathways_info) %>% group_by(`#query`) %>% summarise(count=n())%>% arrange(-count) 


all_info_KEGG_pathway_counts <- all_info_KEGG_pathway %>% filter(grepl("ko",KEGG_Pathway)) %>% left_join(.,carbohydrate_pathways_info) %>% group_by(KEGG_Pathway) %>% summarise(count=n())%>% arrange(-count) 

all_info_KEGG_pathway_counts_sub <- all_info_KEGG_pathway_counts %>% filter(!is.na(long_KEGG_Pathway))




###----------------------summary of modules
 all_info_KEGG_pathway_02 <- All_pseudogenes_annotation_again_carbo %>% select("#query","KEGG_Pathway","KEGG_Module","Preferred_name","COG_category", "complete"   ,"lost" ,"pseudogenes") %>% mutate(KEGG_Pathway = strsplit(as.character(KEGG_Pathway), ",")) %>%   unnest(KEGG_Pathway) %>% filter(KEGG_Pathway %in% all_info_KEGG_pathway_counts_sub$KEGG_Pathway)%>% mutate(KEGG_Module = strsplit(as.character(KEGG_Module), ",")) %>%   unnest(KEGG_Module)%>% left_join(.,carbohydrate_modules_info_02) %>% left_join(.,carbohydrate_pathways_info)
 
 table(all_info_KEGG_pathway_02$KEGG_Pathway)

 all_info_KEGG_pathway_02_summary <- all_info_KEGG_pathway_02 %>% group_by(long_KEGG_Pathway,KeggModuleName) %>% summarise(counts=n())%>% arrange(-counts)
 
  all_info_KEGG_pathway_02_summary <- all_info_KEGG_pathway_02 %>% group_by(long_KEGG_Pathway,KeggModuleName) %>% summarise(counts=n())%>% arrange(-counts) %>% filter(!is.na(KeggModuleName))

   all_info_KEGG_pathway_02_summary$KeggModuleName  = factor(all_info_KEGG_pathway_02_summary$KeggModuleName , levels=all_info_KEGG_pathway_02_summary$KeggModuleName)

  
plot_KEGG <- ggplot(all_info_KEGG_pathway_02_summary,aes(x=reorder(KeggModuleName, -counts),y=counts,fill=long_KEGG_Pathway,color=long_KEGG_Pathway))+geom_bar(stat = "identity")+theme_classic()+theme(axis.text.x = element_text(angle=85,hjust=1),axis.title.x = element_blank())+labs(y="Pseudogenized OGs")
  
plot_KEGG

png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images//20210126/supplement//KEGG_Pathways_pseudo.png", width = 4000, height = 3000,res=400)

# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement//plot_genome_size.pdf",width=6,height=5)

plot_KEGG

dev.off()


  ###----------------------------
  ##KEGG MODULES
  ###----------------------------

# ALL_sterm <- ALL_sterm %>% 
#     mutate(COG_Functional_Category = strsplit(as.character(COG_Functional_Category), "")) %>% 
#     unnest(COG_Functional_Category)
 
 all_info_KEGG_MODULES <- All_pseudogenes_annotation_again_carbo %>% select("#query","KEGG_Module","Preferred_name","COG_category", "complete"   ,"lost" ,"pseudogenes") %>% mutate(KEGG_Module = strsplit(as.character(KEGG_Module), ",")) %>%   unnest(KEGG_Module)
 

all_info_KEGG_MODULES_counts <- all_info_KEGG_MODULES %>% group_by(KEGG_Module) %>% summarise(count=n())%>% arrange(-count) %>% left_join(.,carbohydrate_modules_info) %>% left_join(.,carbohydrate_modules_info_02)  #%>% filter(grepl("ko",KEGG_Module))

```


## draw pseudo gene KEGGmap

```{r}

library(pathview)

##------------example
load(url("http://genomedata.org/gen-viz-workshop/pathway_visualization/pathview_Data.RData"))
pathview(gene.data=tumor_v_normal_DE.fc, species="hsa", pathway.id="hsa03460", kegg.native=FALSE)

#-------data

###----------------------------------
#completeness of OG (core,cloud info)
###----------------------------------
# Lhelv_OG_abundance <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_cluster_ps_counts_wide_imported_wide_final") %>% mutate(Orthogroup=paste0("Lh_",Orthogroup))
# Lhelv_OG_abundancesums <- Lhelv_OG_abundance %>% group_by(Orthogroup,grouping,species)  %>% summarise(counts=n())%>% spread(., grouping, counts)%>%replace(is.na(.), 0)
# total_gene <- rowSums(Lhelv_OG_abundancesums[,c("ps","re","rm")]) %>% head(1) %>% as.numeric()
# Lhelv_OG_abundancesums$coreness <- 100*(Lhelv_OG_abundancesums$re/total_gene)
# lhelv_coreness <- Lhelv_OG_abundancesums %>% select(Orthogroup,coreness) %>% as_tibble()
# 
# merge(lhelv_coreness,all_info_KEGG_yes_cleaned,by.x="Orthogroup",by.y="#query",all=TRUE)%>% filter(species=="Lh") %>%replace(is.na(.), 0) %>% as_tibble()
# # all_info_KEGG_yes_cleaned
# 


Orthogroups_orthofinder_stats_test <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered_counts.txt")%>% mutate(Orthogroup=paste0("Lh_",Orthogroup))


Orthogroups_orthofinder_stats_test_keepers_01 <- Orthogroups_orthofinder_stats_test %>% mutate(gene_type=ifelse(re>0,"re",ifelse(ps>0,"ps","rm")))%>% mutate(genome=gsub(".all","",sample)) %>% select(-sample,-ps,-re) %>% spread(., genome, gene_type)  %>%replace(is.na(.), "rm") 

cluster_ps_counts_wide_imported_wide_final <-  Orthogroups_orthofinder_stats_test_keepers_01 %>%  gather(., genome,grouping , colnames(Orthogroups_orthofinder_stats_test_keepers_01)[2:ncol(Orthogroups_orthofinder_stats_test_keepers_01)], factor_key=TRUE,na.rm = TRUE)  %>% mutate(species="Lhelv")


Lhelv_OG_abundancesums <- cluster_ps_counts_wide_imported_wide_final %>% group_by(Orthogroup,grouping,species)  %>% summarise(counts=n())%>% spread(., grouping, counts)%>%replace(is.na(.), 0)
total_gene <- rowSums(Lhelv_OG_abundancesums[,c("ps","re","rm")]) %>% head(1) %>% as.numeric()
Lhelv_OG_abundancesums$coreness <- 100*(Lhelv_OG_abundancesums$re/total_gene)
lhelv_coreness <- Lhelv_OG_abundancesums %>% select(Orthogroup,coreness) %>% as_tibble()

Lhelv_all_info <- merge(lhelv_coreness,all_info_KEGG_yes_cleaned,by.x="Orthogroup",by.y="#query",all=TRUE)%>% filter(species=="Lh") %>%replace(is.na(.), 0) %>% as_tibble() %>% mutate(fraction_core=coreness/100)

Lhelv_all_info_KEGGmap <- setNames(as.character(Lhelv_all_info$KEGG_Reaction), Lhelv_all_info$fraction_core)

Lhelv_all_info_forONLINE <- Lhelv_all_info %>% select(KEGG_Reaction,category)
write.table(Lhelv_all_info_forONLINE,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/All_pseudogenes_annotation_02.csv",quote = FALSE,row.names = FALSE,sep = "\t")

Lhelv_all_info %>% filter(pseudogenes>0)
unique(Lhelv_all_info$KEGG_Reaction)

##---------------------maps

pathview(gene.data=Lhelv_all_info_KEGGmap,pathway.id="01200",species="ko", kegg.native=FALSE)
pathview(gene.data=Lhelv_all_info_KEGGmap,pathway.id="00052",species="ko", kegg.native=FALSE)

map01200
00052 
```



###extension pseudo genes
extenstion with new

```{bash}
##===============================
##Lactobacillus
##===============================
#mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
#curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt

mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/


rm /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
genus=Lactobacillus

rm  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=iners

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=rodentium

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=hominis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
 
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=gasseri

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep -v "paragasseri"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep -v "paragasseri"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=johnsonii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=taiwanensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=amylolyticus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names


###--------------------------
species=psittaci

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' |head -1 >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}'|head -1 >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=jensenii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=equicursoris

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=pasteurii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=gigeriorum

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=acetotolerans

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=helsingborgensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=melliventris

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kimbladii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kullabergensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "representative genome" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "representative genome" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=bombicola

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=panisapium

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=apis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'  |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=intestinalis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=hamsteri

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kalixensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=helveticus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "Lactobacillus"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "Lactobacillus"|cut -f 20   |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=gallinarum

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20   |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kefiranofaciens

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=crispatus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=acidophilus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
    
    
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20   |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=ultunensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kitasatonis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=amylovorus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20   |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=delbrueckii
subsp=jakobsenii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}" |cut -f 20  |awk -v namezzin="$species" -v subss="$subsp" '{OFS="\t"}{print $1,namezzin"_"subss}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=delbrueckii
subsp=indicus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}" |cut -f 20  |awk -v namezzin="$species" -v subss="$subsp" '{OFS="\t"}{print $1,namezzin"_"subss}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=delbrueckii
subsp=lactis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}" |cut -f 20  |awk -v namezzin="$species" -v subss="$subsp" '{OFS="\t"}{print $1,namezzin"_"subss}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=delbrueckii
subsp=sunkii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}"|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}" |cut -f 20 |awk -v namezzin="$species" -v subss="$subsp" '{OFS="\t"}{print $1,namezzin"_"subss}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=delbrueckii
subsp=bulgaricus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}" |cut -f 20 |awk -v namezzin="$species" -v subss="$subsp" '{OFS="\t"}{print $1,namezzin"_"subss}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###-------------------------------------
##download
###-------------------------------------

rm -r /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF
mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF

cd /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF

    wget -i /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
   
#mv GCF_000769515.1_ASM76951v1_genomic.gbff.gz Out_Bacsub.faa.gz

###========================================================
#extract counts
###========================================================

#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt
rm /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/pseudogeness_count_new.txt



#for genomesss in $(ls /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF |grep ".gz$" )

for genomesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt |cut -d '/' -f 11)
do
genomesss_short=$(echo ${genomesss} |sed 's/_genomic.gbff.gz//g' )
species=$(grep "${genomesss_short}" /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |cut -f 2)
genezzz=$(zcat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF/${genomesss}|grep "Genes (coding)" |cut -d ':' -f 3|sed 's/ //g'|sed 's/,//g'|head -1)
peudosss=$(zcat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF/${genomesss}|grep "Pseudo Genes (total)"  |cut -d ':' -f 3|sed 's/ //g'|sed 's/,//g'|head -1)
transpos=$(zgrep "transposase" /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF/${genomesss} |grep "product=" -c)


echo -e ${species}"\t"${genomesss}"\t"${genezzz}"\t"${peudosss}"\t"${transpos}  >>  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/pseudogeness_count_new.txt
done #genomesss


#cut -f 2 /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |sort|uniq| awk -F "\t" '{OFS="="}{print "\"Lactobacillus\"","\""$1"\""}'

cut -f 2 /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |sort|uniq| awk -F "\t" '{OFS="="}{print "\""$1"\"","\"Lactobacillus\""}'

```

###Strepto

```{bash}
##-----------------------------
##Streptococcus
##-----------------------------

#mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
#curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt

mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log


rm /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt
genus=Streptococcus

rm  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

###--------------------------
species=pneumoniae

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' |head -100 >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' |head -100 >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

###--------------------------
species=mitis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


###--------------------------
species=parasanguinis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=sanguinis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "para" -v|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "para" -v|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=suis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=anginosus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=dysgalactiae
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=pyogenes
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=uberis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=gallolyticus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names
##--------------------------
species=equinus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=vestibularis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=thermophilus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=salivarius
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "K12" |awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "K12" |awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=downei
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=criceti
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=mutans
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=macacae
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >>    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}' |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


###-------------------------------------
##download
###-------------------------------------

rm -r /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/genomes/GBF
mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/GBF

cd /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/GBF

    wget -i /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

#mv GCF_000769515.1_ASM76951v1_genomic.gbff.gz Out_Bacsub.faa.gz

###========================================================
#extract counts
###========================================================

#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt
#rm /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/pseudogeness_count_new.txt


for genomesss in $(ls /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/GBF |grep ".gz$" )
do
genomesss_short=$(echo ${genomesss} |sed 's/_genomic.gbff.gz//g' )
species=$(grep "${genomesss_short}" /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2)
genezzz=$(zcat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/GBF/${genomesss}|grep "Genes (coding)" |cut -d ':' -f 3|sed 's/ //g'|sed 's/,//g'|head -1)
peudosss=$(zcat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/GBF/${genomesss}|grep "Pseudo Genes (total)"  |cut -d ':' -f 3|sed 's/ //g'|sed 's/,//g'|head -1)
transpos=$(zgrep "transposase" /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/GBF/${genomesss} |grep "product=" -c)


echo -e ${species}"\t"${genomesss}"\t"${genezzz}"\t"${peudosss}"\t"${transpos}  >>  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/pseudogeness_count_new.txt
done #genomesss

###--------------------------
#rename
cut -f 2 /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |sort|uniq| awk -F "\t" '{OFS="="}{print "\""$1"\"","\"Streptococcus\""}'


```

transfer local

```{bash}
  
  mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/
  scp  vincent@130.223.51.66:/work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/pseudogeness_count_new.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count_new.txt

cat /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt \
/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count_new.txt > \
/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count_together.txt
```
#### together
do analysis 

```{r}
library(readr)
library(tidyverse)
pseudogeness_count <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count_together.txt",  "\t", escape_double = FALSE, col_names = c("species","genome","genes_coding","pseudogenes","tranposons"), trim_ws = TRUE) %>% filter(pseudogenes<1000) %>% filter(genes_coding<4000)%>% mutate(pseudogenes_percent = 100*(pseudogenes/genes_coding)) %>% mutate(tranposons_percent = 100*(tranposons/genes_coding))%>% filter(species!="Wigg")%>% filter(species!="Scap")%>% filter(species!="Lplan")


table(pseudogeness_count$species)

###--------------------
##filter
# remove species with less then 5 genomes
###--------------------

table(pseudogeness_count$species)[table(pseudogeness_count$species)>5] 
table(pseudogeness_count$species)[table(pseudogeness_count$species)<4]

keeps <- table(pseudogeness_count$species)[table(pseudogeness_count$species)>3] %>% names()

pseudogeness_count <- pseudogeness_count %>% filter(species %in% keeps)
###--------------------
##point plot
###--------------------
plotpoints <- ggplot(pseudogeness_count,aes(x=genes_coding,y=pseudogenes_percent,fill=species,color=species))+geom_point()+labs(x="Gene number",y="Pseudogenes [%]")+theme_classic()+theme(legend.title = element_blank())
plotpoints
  
  library(plotly)
ggp <- ggplotly(plotpoints)
 ggp
# pseudogeness_count %>% group_by(species)%>%ize(meanPseudo=mean(pseudogeness_count),meangenes=mean(genes_coding),stdevPseudo=sd(pseudogeness_count),stdevgenes=stderr(genes_coding))
# pseudogeness_count %>% group_by(species) %>% summarise(meanPseudo=mean(pseudogeness_count))

 
 
pseudogeness_count$species <- plyr::revalue(pseudogeness_count$species,c("delbrueckii_lactis"="L. delbrueckii subsp. lactis",
 "Ldel"="L. delbrueckii subsp. lactis",
 "thermophilus"="S. thermophilus",
 "Sterm"="S. thermophilus",
 "Ssuis"="suis",
 "Lequi"="equicursoris",
 "Svest"="vestibularis"))
 
pseudogeness_sumary <- pseudogeness_count %>% group_by(species) %>%  dplyr::summarize(meanPseudo=100*(mean(pseudogenes)/mean(genes_coding)),meangenes=mean(genes_coding),stdevPseudo=100*(sd(pseudogenes)/mean(genes_coding)),stdevgene=sd(genes_coding),meantransp=100*(mean(tranposons)/mean(genes_coding)),stdevtransp=100*(sd(tranposons)/mean(genes_coding)),meanboth=100*(mean(pseudogenes+tranposons)/mean(genes_coding)),stdevboth=100*(sd(pseudogenes+tranposons)/mean(genes_coding)),counts=n())



 # dplyr::summarize(mean = mean(NumGenesLineage),sd=sd(NumGenesLineage))

##-----
#chnage names
##-----
  pseudogeness_sumary$species_new <- pseudogeness_sumary$species

 
  # pseudogeness_sumary$species_new <- plyr::revalue(pseudogeness_sumary$species,c("Buch"="Buchnera spp.","Sglos"="S. glossinidius","Ldel"="L. delbrueckii","Sterm"="S. thermophilus","Mleprae"="M. leprae","Rick"="Rickettsia spp.","Lplan"="L. plantarum","Ssuis"="S. suis","Lequi"="L. equicursoris","Svest"="S. vestibularis","Wol"="Wolbachia spp."))
#,"Wigg"="Wigglesworthia"
table(pseudogeness_sumary$species)
table(pseudogeness_sumary$species_new)  
  


 # pseudogeness_sumary$species_new  = factor(pseudogeness_sumary$species_new , levels=c("L. delbrueckii","S. thermophilus","L. plantarum","L. equicursoris","S. suis","S. vestibularis","M. leprae","Rickettsia spp.","Buchnera spp.","S. glossinidius","Wolbachia spp."))

 
# colorsss <- c("#fdc2b5ff","#007fa6ff","#ebbc51ff","#ebbc51ff","#9fcfffff","#9fcfffff","#51eb85ff","#51eb85ff","#d58656ff","#d58656ff","#d58656ff") 
# colorsss <- c("#fdc2b5ff","#007fa6ff","grey77","grey77","grey77","grey77","#51eb85ff","#51eb85ff","#d58656ff","#d58656ff","#d58656ff")

##-----
#plot size vs. pseudogenes
##-----
plotpoints <-ggplot( pseudogeness_sumary, aes(x = meangenes, y = meanPseudo, xmin = meangenes-stdevgene, xmax = meangenes+stdevgene,ymin=meanPseudo-stdevPseudo,ymax=meanPseudo+stdevPseudo,fill=species_new,color=species_new)) +
  geom_point(size=2) +
  geom_errorbarh(alpha=.5)+
  geom_errorbar(alpha=.5)+
    labs(x="Gene number",y="Pseudogenes [%]")+theme_classic()+theme(legend.title = element_blank())+
  #geom_hline(yintercept =7.5, linetype="dotted")+geom_segment(aes(x = 1500, y = -1, xend = 1500, yend = 7.5), linetype = "dotted")+
  # scale_color_manual(values = colorsss)+scale_fill_manual(values = colorsss)+
  annotate("text", x = 800, y = 6, label = "endosymbionts",col="#d58656ff")+
  annotate("text", x = 800, y = 25, label = "transition",col="#51eb85ff")+
  annotate("text", x = 2000, y = 1, label = "free living",col="grey77")

#+lims(x=c(250,4100),y=c(-1.5,40))
plotpoints

  library(plotly)
ggp <- ggplotly(plotpoints)
 ggp

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenes.pdf",width=6,height=4)

plotpoints

dev.off()
##-----
#plot transposase vs. pseudogenes
##-----
plotpoints_withtranps <- ggplot(pseudogeness_sumary, aes(x = meantransp, y = meanPseudo, xmin = meantransp-stdevtransp, xmax = meantransp+stdevtransp,ymin=meanPseudo-stdevPseudo,ymax=meanPseudo+stdevPseudo,fill=species_new,color=species_new)) +
  geom_point(size=2) +
  geom_errorbarh(alpha=.5)+
  geom_errorbar(alpha=.5)+
    labs(x="transposase",y="Pseudogenes [%]")+theme_classic()+theme(legend.title = element_blank())
  #geom_hline(yintercept =7.5, linetype="dotted")+geom_segment(aes(x = 1500, y = -1, xend = 1500, yend = 7.5), linetype = "dotted")+
  # scale_color_manual(values = colorsss)+scale_fill_manual(values = colorsss)+
  # annotate("text", x = 800, y = 6, label = "endosymbionts",col="#d58656ff")+
  # annotate("text", x = 800, y = 25, label = "transition",col="#51eb85ff")+
  # annotate("text", x = 2000, y = 1, label = "free living",col="grey77")

#+lims(x=c(250,4100),y=c(-1.5,40))
plotpoints_withtranps

  library(plotly)
ggp <- ggplotly(plotpoints_withtranps)
 ggp
##-----
#plot transposase +pseudogenes vs. genome size
##-----
 
 
plotpoints_both <- ggplot( pseudogeness_sumary, aes(x = meangenes, y = meanboth, xmin = meangenes-stdevgene, xmax = meangenes+stdevgene,ymin=meanboth-stdevboth,ymax=meanboth+stdevboth,fill=species_new,color=species_new,text=paste("counts=",counts))) +
  geom_point(size=2) +
  geom_errorbarh(alpha=.2)+
  geom_errorbar(alpha=.2)+
    labs(x="Gene number",y="Pseudogenes+transposons [%]")+theme_classic()+theme(legend.title = element_blank())
  #geom_hline(yintercept =7.5, linetype="dotted")+geom_segment(aes(x = 1500, y = -1, xend = 1500, yend = 7.5), linetype = "dotted")+
  # scale_color_manual(values = colorsss)+scale_fill_manual(values = colorsss)+
  # annotate("text", x = 800, y = 6, label = "endosymbionts",col="#d58656ff")+
  # annotate("text", x = 800, y = 25, label = "transition",col="#51eb85ff")+
  # annotate("text", x = 2000, y = 1, label = "free living",col="grey77")

#+lims(x=c(250,4100),y=c(-1.5,40))
plotpoints_both

  library(plotly)
ggp <- ggplotly(plotpoints_both)
 ggp


# plotpoints_wo <- plotpoints+theme(legend.position = "none")
# plotpoints_wo
# plotpoints+stat_ellipse(data=pseudogeness_count,aes(x=genes_coding,y=pseudogenes_percent,fill=species,color=species,group=species),type = "euclid", level = 3) 

  
 ##----------------------------------------
 ##change coloring
 ##----------------------------------------
 
 dput(unique(pseudogeness_sumary$species))
 
 pseudogeness_sumary$species_new_02 <- plyr::revalue(pseudogeness_sumary$species,c("anginosus"="Streptococcus",
"criceti"="Streptococcus",
"downei"="Streptococcus",
"dysgalactiae"="Streptococcus",
"equinus"="Streptococcus",
"gallolyticus"="Streptococcus",
"macacae"="Streptococcus",
"mitis"="Streptococcus",
"mutans"="Streptococcus",
"parasanguinis"="Streptococcus",
"pneumoniae"="Streptococcus",
"pyogenes"="Streptococcus",
"salivarius"="Streptococcus",
"sanguinis"="Streptococcus",
"suis"="Streptococcus",
# "thermophilus"="Streptococcus",
"uberis"="Streptococcus",
"vestibularis"="Streptococcus",
"helveticus"="L.helveticus",
"acetotolerans"="Lactobacillus",
"acidophilus"="Lactobacillus",
"amylolyticus"="Lactobacillus",
"amylovorus"="Lactobacillus",
"apis"="Lactobacillus",
"bombicola"="Lactobacillus",
"_bulgaricus"="Lactobacillus",
"crispatus"="Lactobacillus",
# "delbrueckii_indicus"="Lactobacillus",
# "delbrueckii_lactis"="Lactobacillus",
# "delbrueckii_sunkii"="Lactobacillus",
"equicursoris"="Lactobacillus",
"gallinarum"="Lactobacillus",
"gasseri"="Lactobacillus",
"gigeriorum"="Lactobacillus",
"hamsteri"="Lactobacillus",
"helsingborgensis"="Lactobacillus",
"hominis"="Lactobacillus",
"iners"="Lactobacillus",
"intestinalis"="Lactobacillus",
"jensenii"="Lactobacillus",
"johnsonii"="Lactobacillus",
"kalixensis"="Lactobacillus",
"kefiranofaciens"="Lactobacillus",
"kimbladii"="Lactobacillus",
"kitasatonis"="Lactobacillus",
"kullabergensis"="Lactobacillus",
"melliventris"="Lactobacillus",
"panisapium"="Lactobacillus",
"pasteurii"="Lactobacillus",
"psittaci"="Lactobacillus",
"rodentium"="Lactobacillus",
"taiwanensis"="Lactobacillus",
"ultunensis"="Lactobacillus",
 #  "Buch"="Buchnera spp.",
 # "Sglos"="S. glossinidius",
 # "Mleprae"="M. leprae",
 # "Rick"="Rickettsia spp.",
 # "Wol"="Wolbachia spp."
 "Buch"="Endosymbiont",
 # "Sglos"="S. glossinidius",
 # "Mleprae"="M. leprae",
 "Rick"="Transition",
 "Wol"="Transition",
"Port"="Endosymbiont"))
  table(pseudogeness_sumary$species_new_02)
 pseudogeness_sumary$species_new_02 = factor(pseudogeness_sumary$species_new_02, levels=c("Endosymbiont",  "Transition", "S. thermophilus", "Streptococcus" , "L. delbrueckii subsp. lactis","Lactobacillus", "L.helveticus"))

 colorsss <- c("red","orange","#007fa6ff","#0080a332","#fdc2b5ff","#ffc3b433","#b3b3b3ff")
 # colorsss <- c("#fdc2b5ff","#")

  
plotpoints_both_colored <- ggplot( pseudogeness_sumary, aes(x = meangenes, y = meanboth, xmin = meangenes-stdevgene, xmax = meangenes+stdevgene,ymin=meanboth-stdevboth,ymax=meanboth+stdevboth,fill=species_new_02,color=species_new_02,text=paste("counts=",counts,"\nspecies=",species_new))) +
  geom_point(size=3) +
  geom_errorbarh(alpha=.2)+
  geom_errorbar(alpha=.2)+
    labs(x="Gene number",y="Pseudogenes+transposons [%]")+theme_classic()+theme(legend.title = element_blank())+
  geom_hline(yintercept =15, linetype="dotted")+geom_segment(aes(x = 1000, y = 1, xend = 1000, yend = 15), linetype = "dotted")+
  scale_color_manual(values = colorsss)+scale_fill_manual(values = colorsss)+
  annotate("text", x = 800, y = 6, label = "endosymbionts",col="red")+
  annotate("text", x = 800, y = 25, label = "transition",col="orange")+
  annotate("text", x = 2000, y = 1, label = "free living",col="grey77")

plotpoints_both_colored
#+lims(x=c(250,4100),y=c(-1.5,40))
# plotpoints_both_colored

  library(plotly)
ggp <- ggplotly(plotpoints_both_colored)
 ggp

 

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenes_colored.pdf",width=6,height=4)

plotpoints_both_colored

dev.off()

## only size--------------------------

plotpoints <-ggplot( pseudogeness_sumary, aes(x = meangenes, y = meanPseudo, xmin = meangenes-stdevgene, xmax = meangenes+stdevgene,ymin=meanPseudo-stdevPseudo,ymax=meanPseudo+stdevPseudo,fill=species_new_02,color=species_new_02,text=paste("counts=",counts,"\nspecies=",species_new))) +
  geom_point(aes(size=meantransp)) +
  geom_errorbarh(alpha=.5)+
  geom_errorbar(alpha=.5)+
    labs(x="Gene number",y="Pseudogenes [%]")+theme_classic()+theme(legend.title = element_blank())+
  geom_hline(yintercept =10, linetype="dotted")+geom_segment(aes(x = 1000, y = -1, xend = 1000, yend = 10), linetype = "dotted")+
  scale_color_manual(values = colorsss)+scale_fill_manual(values = colorsss)+
  annotate("text", x = 800, y = 6, label = "endosymbionts",col="red")+
  annotate("text", x = 800, y = 25, label = "transition",col="orange")+
  annotate("text", x = 2000, y = 1, label = "free living",col="grey77")

#+lims(x=c(250,4100),y=c(-1.5,40))
plotpoints

  library(plotly)
ggp <- ggplotly(plotpoints)
 ggp

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenes_new.pdf",width=6,height=4)

plotpoints

dev.off()
 
 ##------------------------
 ##eclipse
 #to do this we include all std points and mean points into one datafram
 ##------------------------
 
#  pseudogeness_sumary
#  
#   pseudogeness_sumary$groups <- plyr::revalue(pseudogeness_sumary$species,c("Buch"="endosmymbiont","Sglos"="endosmymbiont","Ldel"="starter cultures","Sterm"="starter cultures","Wigg"="endosmymbiont","Mleprae"="transition","Rick"="transition","Lplan"="closely related free-living","Ssuis"="closely related free-living"))
# 
#  pseudogeness_esclipse <-  rbind(data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes,y=pseudogeness_sumary$meanPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes-pseudogeness_sumary$stdevgene,y=pseudogeness_sumary$meanPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes+pseudogeness_sumary$stdevgene,y=pseudogeness_sumary$meanPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes,y=pseudogeness_sumary$meanPseudo-pseudogeness_sumary$stdevPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes,y=pseudogeness_sumary$meanPseudo+pseudogeness_sumary$stdevPseudo))
#  
#  
#  
#  plotpoints_esclipse <- ggplot(pseudogeness_esclipse,aes(x=x,y=y,fill=groups,color=groups),group=groups)+labs(x="Gene number",y="Pseudogenes [%]")+theme_classic()+theme(legend.title = element_blank())+stat_ellipse(type = "t", linetype = 2)+theme(legend.position = "none")+lims(x=c(250,4100),y=c(-1.5,40))
# plotpoints_esclipse
#  
# 
# library(patchwork)
# 
# plotpoints+plotpoints_wo+plotpoints_esclipse

###-------------------------------------only size and significanc  to closely relaed species
pseudogeness_sumary$species_new


pseudo_sub <- pseudogeness_sumary %>% filter(species_new %in% c("helveticus","S. thermophilus","L. delbrueckii subsp. lactis","vestibularis","crispatus","equicursoris"))


# ggplot(pseudo_sub,aes(x=species,y=meangenes))

branch_age_plot <- ggplot() +
  geom_pointrange(aes(x=species_new, y=meangenes, ymin = meangenes-stdevgene, ymax = meangenes+stdevgene,fill=species_new,color=species_new),data = pseudo_sub)+theme_classic()


# +facet_wrap(group~.,nrow = 3,scales = "free_y", strip.position = "right")+
#   coord_flip()+scale_fill_manual(values = colorsss)+scale_color_manual(values = colorsss)+
#   theme(axis.text.y = element_blank(),axis.title.y = element_blank(),axis.ticks.y = element_blank(),legend.position = "none",strip.text = element_text(size = 7))+labs(y="Branch age")+scale_y_continuous(breaks=c(-10000,-7500,-5000,-2500,0,2020))

branch_age_plot

 # S.vestibularis, L. equicursoris and L. acidophilus

pseudogeness_count$species <- plyr::revalue(pseudogeness_count$species,c("delbrueckii_lactis"="L. delbrueckii subsp. lactis",
 "Ldel"="L. delbrueckii subsp. lactis",
 "thermophilus"="S. thermophilus",
 "Sterm"="S. thermophilus",
 "Ssuis"="suis",
 "Lequi"="equicursoris",
 "Svest"="vestibularis"))

pseudo_sub_detail <- pseudogeness_count %>% filter(species %in% c("helveticus","S. thermophilus","L. delbrueckii subsp. lactis","vestibularis","crispatus","equicursoris"))

pseudo_sub_detail$species_group <- plyr::revalue(pseudo_sub_detail$species,c( "L. delbrueckii subsp. lactis"="Ldel",
 "S. thermophilus"="Sterm",
 "crispatus"="Lhelv",
 "helveticus"="Lhelv",
 "equicursoris"="Ldel",
 "vestibularis"="Sterm"))
library(ggpubr)
library(ggsignif)
ggplot(pseudo_sub_detail,aes(x=species,y=genes_coding,fill=species))+geom_boxplot()+theme_classic()+facet_wrap(species_group~.,nrow = 1,scales = "free")+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank())+ stat_compare_means(comparisons = list(data.frame(c("S. thermophilus", "helveticus"),c("vestibularis", "S. crispatus"))),method = "t.test")+ stat_compare_means(comparisons = list(c("vestibularis", "S. thermophilus")),method = "t.test")

list(data.frame(c("S. thermophilus", "helveticus"),c("vestibularis", "S. crispatus")))

# annotation table with adjusted pvals and y-position of the labels
anno_df <- compare_means(genes_coding ~ species, group.by = "species_group", data = pseudo_sub_detail) %>%
 mutate(y_pos = 2800, p.adj = format.pval(p.adj, digits = 2))



plot_genome_size <- ggplot()+geom_boxplot(data=pseudo_sub_detail,aes(x=species,y=genes_coding,fill=species))+theme_classic()+facet_wrap(species_group~.,nrow = 1,scales = "free")+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank()) + 
  geom_signif(
    data=anno_df, 
    aes(xmin = group1, xmax = group2, annotations = p.adj, y_position = y_pos), 
    manual= TRUE
  )

plot_genome_size

   png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images//20210126/supplement//plot_genome_size.png", width = 2600, height = 1900,res=400)

# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement//plot_genome_size.pdf",width=6,height=5)

plot_genome_size

dev.off()


##---------pseudo signficance
anno_df <- compare_means(pseudogenes ~ species, group.by = "species_group", data = pseudo_sub_detail) %>%
 mutate(y_pos = 400, p.adj = format.pval(p.adj, digits = 2))

plot_pseudo_sifn <- ggplot()+geom_boxplot(data=pseudo_sub_detail,aes(x=species,y=pseudogenes,fill=species))+theme_classic()+facet_wrap(species_group~.,nrow = 1,scales = "free")+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank()) + 
  geom_signif(
    data=anno_df, 
    aes(xmin = group1, xmax = group2, annotations = p.adj, y_position = y_pos), 
    manual= TRUE
  )

plot_pseudo_sifn

   png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images//20210126/supplement//plot_pseudo_sifn.png", width = 2600, height = 1900,res=400)

# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement//plot_genome_size.pdf",width=6,height=5)

plot_pseudo_sifn

dev.off()


```

### Extract 16sV4

1. download all chromosomes.fasta files of species
2. run [barrnap](https://github.com/tseemann/barrnap) to extract 16s
3. run [V-Xtractor](https://github.com/carden24/V-Xtractor) to extract 16s V4

```{bash}
##------------------------------------------
##1. download all chromosomes.fasta files of species
##------------------------------------------
sed 's/_genomic.gbff.gz/_genomic.fna.gz/g' /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt > /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes_contig.txt

sed 's/_genomic.gbff.gz/_genomic.fna.gz/g' /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt > /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes_contig.txt


###---
##download
###---

rm -r /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA
mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA
cd /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA
wget -i /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes_contig.txt


rm -r /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA
mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA
cd /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA
wget -i /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes_contig.txt


##------------------------------------------
##2. run barrnap to extract 16s
## AND
##3. run V-Xtractor to extract 16s V4
##------------------------------------------
###---
##Sterm
###---

rm -r /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/
mkdir -p /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/
rm -r /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/
mkdir -p /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/

for genomess in $(ls /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA |grep ".fna.gz$" |sed 's/_genomic.fna.gz//g')
do
echo ${genomess}
zcat  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/${genomess}_*.fna.gz  |barrnap --threads 37 -o /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}_rrna.fa 

speciess=$(grep "${genomess}" /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2)

samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}_rrna.fa 
num=1

rm  /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna
for f16sss in $(grep "16S" -i /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}_rrna.fa |sed 's/>//g')
do
echo ${f16sss}
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}_rrna.fa ${f16sss} > /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/tmp.fna

sed "s/^>.*$/>16S-${speciess}-${genomess}-${num}/g" /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/tmp.fna >> /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna
num=$((num+1))
  
done #16s
###---------------vxtractor
cd /home/vincent/apps/V-Xtractor

perl vxtractor.pl -a -r V4 -h HMMs/HMMs/SSU/bacteria/ -o /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}_V4.fa /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna
cat /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}_V4.fa >>  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/final_V4.fna
echo "================================================================="
rm /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}*

done #genomess

###---
##Ldel
###---
rm -r /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/
mkdir -p /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/

rm -r /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/
mkdir -p /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/
for genomess in $(ls /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA |grep ".fna.gz$" |sed 's/_genomic.fna.gz//g')
do
echo ${genomess}
zcat  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/${genomess}_*.fna.gz  |barrnap --threads 37 -o /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}_rrna.fa 

speciess=$(grep "${genomess}" /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |cut -f 2)

samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}_rrna.fa 
num=1
rm  /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/final.fna

for f16sss in $(grep "16S" -i /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}_rrna.fa |sed 's/>//g')
do
echo ${f16sss}
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}_rrna.fa ${f16sss} > /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/tmp.fna

sed "s/^>.*$/>16S-${speciess}-${genomess}-${num}/g" /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/tmp.fna >> /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/final.fna
num=$((num+1))
  
done #16s
###---------------vxtractor
cd /home/vincent/apps/V-Xtractor

perl vxtractor.pl -a -r V4 -h HMMs/HMMs/SSU/bacteria/ -o /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}_V4.fa /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/final.fna
cat /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}_V4.fa >>  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/final_V4.fna
echo "================================================================="
rm /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}*

done #genomess
##============================================================
##------------------------------------------
##4. devide into indiviual files per species
##------------------------------------------
##============================================================
##---------------------------------------
#Sterm
##---------------------------------------
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/final_V4.fna
for speciesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo ${speciesss}
grep "${speciesss}" -c /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/final_V4.fna
grep "${speciesss}" /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/final_V4.fna |sed 's/>//g' > /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/tmp.txt

xargs --arg-file=/work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/tmp.txt  samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/final_V4.fna > /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/${speciesss}_V4.fasta

grep "${speciesss}" -c /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/${speciesss}_V4.fasta

done


##---------------------------------------
#Ldel
##---------------------------------------
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/final_V4.fna
for speciesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo ${speciesss}
grep "${speciesss}" -c /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/final_V4.fna
grep "${speciesss}" /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/final_V4.fna |sed 's/>//g' > /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/tmp.txt

xargs --arg-file=/work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/tmp.txt  samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/final_V4.fna > /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/${speciesss}_V4.fasta


grep "${speciesss}" -c /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/${speciesss}_V4.fasta

done

##============================================================
##------------------------------------------
##5. DEREPLICATE WITH CD-HIT
##------------------------------------------
##============================================================
##---------------------------------------
#Sterm
##---------------------------------------
rm -r /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/
mkdir -p /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/
for speciesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo ${speciesss}


grep "${speciesss}" -c /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/${speciesss}_V4.fasta

cd-hit-est -i /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/${speciesss}_V4.fasta \
    -o /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss} -c 1 -M 30000 -T 37 

#cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.cl*
done

##----------------------- analyse

rm /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/Final_V4_derplicated.fasta
genus=Streptococcus
for speciesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo ${speciesss}


#cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.cl*

    
grep ">Cluster" /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.cl*|sed 's/ /_/g' >  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/tmp.txt

num=1
rm  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.fasta
for clustersss in $(cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/tmp.txt)
do
#aryName=$(grep "${clustersss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log |cut -f 2)
ClusterNumber=$(echo $clustersss|awk -F "_" '{print $2}')
EndCluster=$((ClusterNumber+1))

countss=$(sed -n "/>Cluster $ClusterNumber$/,/>Cluster $EndCluster$/p" /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.cl* | grep -v "^>Cluster" |wc -l)

namesss=$(grep ">" /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss} | sed 's/>//g' |sed -n ${num}p) 
echo ${namesss}
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss} 
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss} ${namesss} |sed "s/>${namesss}/>16SV4_${genus}_${speciesss}_n${countss}/g" >>  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.fasta

  num=$((num+1))


done

cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.fasta >>  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/Final_V4_derplicated.fasta

done
##---------------------------------------
#Ldel
##---------------------------------------
rm -r /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/
mkdir -p /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/
for speciesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo ${speciesss}


grep "${speciesss}" -c /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/${speciesss}_V4.fasta

cd-hit-est -i /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/${speciesss}_V4.fasta \
    -o /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss} -c 1 -M 30000 -T 37 

#cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.cl*
done

##----------------------- analyse

rm /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/Final_V4_derplicated.fasta
genus=Lactobacillus
for speciesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo ${speciesss}


#cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.cl*

    
grep ">Cluster" /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.cl*|sed 's/ /_/g' >  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/tmp.txt

num=1
rm  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.fasta
for clustersss in $(cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/tmp.txt)
do
#aryName=$(grep "${clustersss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log |cut -f 2)
ClusterNumber=$(echo $clustersss|awk -F "_" '{print $2}')
EndCluster=$((ClusterNumber+1))

countss=$(sed -n "/>Cluster $ClusterNumber$/,/>Cluster $EndCluster$/p" /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.cl* | grep -v "^>Cluster" |wc -l)

namesss=$(grep ">" /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss} | sed 's/>//g' |sed -n ${num}p) 
echo ${namesss}
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss} 
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss} ${namesss} |sed "s/>${namesss}/>16SV4_${genus}_${speciesss}_n${countss}/g" >>  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.fasta

  num=$((num+1))


done

cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.fasta >>  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/Final_V4_derplicated.fasta

done

##============================================================
##------------------------------------------
##6. Bring all together
##------------------------------------------
##============================================================


cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/Final_V4_derplicated.fasta \
/work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/Final_V4_derplicated.fasta > \
/work/Project/2020_StarterCultureDiversity/16S_search/V4/all_V4_Strepto_Lacto_species.fasta


```

```{bash}
mkdir -p //home/vincent/Desktop/Projects/2020_StarterCultureDiversity/16S_search/V4/
scp   vincent@130.223.51.116:/work/Project/2020_StarterCultureDiversity/16S_search/V4/all_V4_Strepto_Lacto_species.fasta //home/vincent/Desktop/Projects/2020_StarterCultureDiversity/16S_search/V4/all_V4_Strepto_Lacto_species.fasta



```


### Alternative: Extract 16sV4

Alternative to the extraction from the downloaded genomes, we can also get the 16s sequences from GTDB and extract the V4 from the species of interest. 
This is most likely much more extensive then the previous approach.

```{bash}
#mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication
scp  //home/vincent/Desktop/Projects/2020_StarterCultureDiversity//GTDB_RS89_20190703_1300_HIT_Strains_IDTAXA.fasta vincent@130.223.51.116:/work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/


```

```{bash}
for speciesss in $(cat  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo "================================================"
echo ${speciesss}
grep ${speciesss} /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/GTDB_RS89_20190703_1300_HIT_Strains_IDTAXA.fasta


done


###-----------------------------------
#ldel
###-----------------------------------


for speciesss in $(cat  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo "================================================"
echo ${speciesss}
count=$(grep ${speciesss} /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/GTDB_RS89_20190703_1300_HIT_Strains_IDTAXA.fasta|grep "Lactobacillus" -c)
grep ${speciesss} /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/GTDB_RS89_20190703_1300_HIT_Strains_IDTAXA.fasta|grep "Lactobacillus"
if [ $count -gt 0 ]
 then              
 
echo -e "has " ${count} "16s sequences in GTDB"

 else  
 echo -e "has no 16s sequences in GTDB"

 fi

done
```


### gene duplication

Here, I look for duplicated genes. I do this by clustering all genes of with mmseqs2 and looking which clusters are larger than 1. 
Further I look if among the duplicate genes we have a clustering of a genome location (E.g. duplicated operon structure). 

```{bash}
species=Sterm
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/Geneduplication.final 
for species in $(echo "Ldel Sterm")
do
date=$(ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ |grep "Results")
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/${species}/
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/ |grep ".ffn$" |sed 's/.ffn//g' )
do

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/${species}/${genomesss}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/${species}/${genomesss}

#mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/${genomesss}.ffn ${genomesss} tmp --min-seq-id 0.8 -c 0.9 --cov-mode 1
mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/${genomesss}.faa ${genomesss} tmp 

grep ">" -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/${genomesss}.faa
#grep ">" -c *_rep_seq.fasta
#grep ">" -c *_all_seqs.fasta
cut -f 1 ${genomesss}_cluster.tsv |sort|uniq -c  |wc -l
#cut -f 1 S12273_cluster.tsv |sort|uniq -c | sed -e 's/^[[:space:]]*//' |awk -F " " '{OFS="\t"}{if($1>1)print $0}'

for genesss in $(cut -f 1 ${genomesss}_cluster.tsv |sort|uniq -c | sed -e 's/^[[:space:]]*//' |awk -F " " '{OFS="\t"}{if($1>1)print $2}')
do
genesss_short=$(echo $genesss|sed 's/S//g'|sed 's/L//g')

descriptionsss=$(grep "${genesss}"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/${genomesss}.faa |awk -F " " '{OFS=" "}{print $2,$3,$4,$5}')
OGsss=$(grep "${genesss_short}" /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt |awk -F ":" '{print $1}')
numbergenes=$(cut -f 1 ${genomesss}_cluster.tsv |sort|uniq -c | sed -e 's/^[[:space:]]*//' |awk -F " " -v genezzz="$genesss" '{OFS="\t"}{if($2==genezzz)print $1}' )

for genesssAlll in $(awk -F "\t" -v genezzz="$genesss" '{if($1==genezzz) print $2}' ${genomesss}_cluster.tsv)
do


echo -e ${species}"\t"${genomesss}"\t"${genesss}"\t"${OGsss}"\t"${numbergenes}"\t"${descriptionsss}"\t"${genesssAlll} >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/Geneduplication.final 

done #genesssAlll
done #genesss

done #genomesss


done #species

###----------------------------------------------
#number of genes
###----------------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/gene_count.txt
for species in $(echo "Ldel Sterm")
do
#date=$(ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ |grep "Results")

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/ |grep ".gff$" |sed 's/.gff//g' )
do

countsss=$(grep "CDS" -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/${genomesss}.gff)

echo -e ${species}"\t"${genomesss}"\t"${countsss} >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/gene_count.txt


done #genomesss

done #species
###----------------------------------------------
#number of genes NEW 
#corrected for non-duplicated (do not count duplicated multiple times)
###----------------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/gene_count.txt
for species in $(echo "Ldel Sterm")
do
date=$(ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ |grep "Results")

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/ |grep ".ffn$" |sed 's/.ffn//g' )
do


countsss=$(sed '1d' /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt |grep "${genomesss}" -c)

echo -e ${species}"\t"${genomesss}"\t"${countsss} >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/gene_count.txt


done #genomesss


done #species
```

local
```{bash}
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/gene_count.txt  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/Geneduplication.final  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication

cut -f 2 /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/Geneduplication.final |sort|uniq -c
```


Now I analyse the data in R. I also combine the eggnog information of the OGs. 

Of special interest are if genes are:
- tandem repeat (if two genes)
- operon repeat (left and right flanking genes are also repeated?)
- what genes? (COG?)

```{r}
library(tidyverse)
Geneduplication <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/Geneduplication.final", "\t", escape_double = FALSE, col_names = c("species","genome","cluster","Orthogroup","copies","annotation","gene"),   trim_ws = TRUE)


Geneduplication_cleaned <- Geneduplication %>% mutate(Orthogroup_new = ifelse(species=="Ldel",paste0("L_",Orthogroup),paste0("S_",Orthogroup) ))

##---------------------------
#add eggnog information about the cluster-specific genes. 
##---------------------------

ldel_all_unique <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel_orthofinder_prokka.annotations",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description"),skip=4)
# RMK202_both_eggnog[RMK202_both_eggnog$query_name %in% SNPeff_final$finalName,]
Sterm_all_unique <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm_orthofinder_prokka.annotations",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description"),skip=4)

eggnog_annotation <- rbind(Sterm_all_unique,ldel_all_unique)

COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA") %>% select(-divided)

# unique(eggnog_annotation$COG_Functional_Category)[which(!unique(eggnog_annotation$COG_Functional_Category) %in% COG_functional_categories$short )]

ALL_sterm <- eggnog_annotation
nrow(ALL_sterm)

ALL_sterm <- ALL_sterm %>% 
    mutate(COG_Functional_Category = strsplit(as.character(COG_Functional_Category), "")) %>% 
    unnest(COG_Functional_Category)


lineageAbundance_presence_count_eggnog_COG <- merge(ALL_sterm,COG_functional_categories,by.x = "COG_Functional_Category",by.y="short")
table(lineageAbundance_presence_count_eggnog_COG$long)

##------------------------------
#merge to duplicated genes
##------------------------------
colnames(lineageAbundance_presence_count_eggnog_COG)
lineageAbundance_presence_count_eggnog_COG$query_name %>% length()
duplicate_with_eggnog <- merge(Geneduplication_cleaned,lineageAbundance_presence_count_eggnog_COG,by.x = "Orthogroup_new",by.y="query_name")

table(duplicate_with_eggnog$long) %>% sort(decreasing = TRUE)

##------------------------------
#find look for tandem repeats of genes
##------------------------------


for (clusterzzz in unique(duplicate_with_eggnog$cluster)[1]) {
  
  tmp <- duplicate_with_eggnog %>% filter(cluster==clusterzzz)
  genomesss <- duplicate_with_eggnog %>% filter(cluster==clusterzzz) %>% pull(genome) %>% unique()
  
  duplicate_with_eggnog %>% filter(genome==genomesss)%>% pull(gene) %>% sort()
  
  tmp2 <- duplicate_with_eggnog %>% filter(genome==genomesss)
  
}


##------------------------------
#cogs
##------------------------------

unique(duplicate_with_eggnog$Orthogroup) %>% length()

unique(duplicate_with_eggnog$cluster) %>% length()

duplicated_contigs <- duplicate_with_eggnog %>% select(c(Orthogroup_new,long)) %>% unique() 
library(forcats) 

Cogs_duplicated <- ggplot(duplicated_contigs,aes(x=forcats::fct_infreq(long)))+geom_bar()+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank())

Cogs_dall <- ggplot(lineageAbundance_presence_count_eggnog_COG,aes(x=forcats::fct_infreq(long)))+geom_bar()+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank())

Cogs_duplicated+Cogs_dall+ plot_layout(ncol=1,widths = c(1,1))


all_cogs <- lineageAbundance_presence_count_eggnog_COG %>% group_by(long) %>% summarise(counts=n()) %>% add_column(cogs="all")
duplicated_cogs <- duplicated_contigs %>% group_by(long) %>% summarise(counts=n())%>% add_column(cogs="duplicated")
complete_cogs <- rbind(all_cogs,duplicated_cogs)


Cogs_duplicated_ext <- ggplot(complete_cogs,aes(x=long,y=sort(counts,decreasing = TRUE),fill=cogs))+geom_bar(stat="identity")+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank())
Cogs_duplicated_ext


all_cogs <- lineageAbundance_presence_count_eggnog_COG %>% group_by(long) %>% summarise(ALL_counts=n()) 
duplicated_cogs <- duplicated_contigs %>% group_by(long) %>% summarise(Duplicated=n())

cogs_comple_long <- merge(all_cogs,duplicated_cogs,by="long") %>%  mutate(Complete_genes=ALL_counts-Duplicated) %>% dplyr::select(-ALL_counts) %>% gather(.,group,count,c(Complete_genes,Duplicated))



colorsss=rev(c("grey44","grey90"))


Cogs_duplicated_ext <- ggplot(cogs_comple_long,aes(x=reorder(long,-count),y=count,fill=group))+geom_bar(stat="identity")+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank(),legend.title = element_blank() )+scale_fill_manual(values = colorsss)
Cogs_duplicated_ext

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/COgs_duplicated.pdf",width=6,height=5)

Cogs_duplicated_ext

dev.off()

##------check with chi-square test

matchlist = c("Unknown")

cogs_comple_wide <- merge(all_cogs,duplicated_cogs,by="long") %>%  mutate(Complete_genes=ALL_counts-Duplicated) %>% dplyr::select(-ALL_counts) %>% mutate(long=replace_na(long,"Unknown")) %>% arrange(desc(Complete_genes+Duplicated))%>% arrange(long %in% matchlist) %>% column_to_rownames(.,var="long")

# cogs_comple_wide <- cogs_comple_wide %>% arrange(desc(Complete_genes+Duplicated))


# mydata %>% arrange(id %in% match_list)

chisq <- chisq.test(cogs_comple_wide)
# chisq

chisq$observed

round(chisq$expected,2)
round(chisq$residuals, 3)
library(corrplot)
dev.off()
plot.new()
# corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.5)

library(RColorBrewer)
# corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(brewer.pal(n = 8, name = "RdYlBu")), method = "color")

col3 <- colorRampPalette(c("red", "white", "blue")) 
corsssplot <- corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = "color")
corsssplot
dev.off()

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/corrplot_duplicatedGenes.pdf",width=6,height=10)

 corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = "color")
 dev.off()

##put residual into plot --> amino acid pathway is more common than expected by chance. 
##------------------------------
#Orthogroups without copies
##------------------------------
 
 # duplicate_with_eggnog %>% select(genome,cluster) %>% dim()
 #  duplicate_with_eggnog %>% select(genome,cluster) %>% unique() %>% select(genome) %>% 
 #  

##------------------------------
#percent of duplicated genes
##------------------------------
 ###
 
# library(ggpol)

genomes_tmps <- duplicate_with_eggnog %>% select(species,genome) %>% unique()

duplicate_genes <- duplicate_with_eggnog %>% group_by(genome) %>% select(cluster) %>% unique()%>% summarize(counts=n()) ## here I count the number of duplicated clusters
# duplicate_genes <- duplicate_with_eggnog %>% group_by(genome) %>% select(Orthogroup_new) %>% unique()%>% summarize(counts=n()) ## here I count the number of duplicated clusters

#duplicate_with_eggnog %>% group_by(genome) %>% select(gene) %>% unique()%>% summarize(counts=n()) ## I used to count the number of duplicated gene
duplicate_genes_ext <- merge(duplicate_genes,genomes_tmps,by="genome")

ggplot(duplicate_genes_ext,aes(x=species,y=counts,fill=species))+geom_boxplot()+theme_classic()+geom_jitter()+theme(legend.position = "none")

ggplot(duplicate_genes_ext,aes(x=species,y=counts,fill=species))+geom_boxplot(outlier.size = 0) + geom_point(pch = 21, position = position_jitterdodge())+theme_classic()+theme(legend.position = "none")

##get percent of multicopy genes by including number of genes in genomes

gene_count <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/gene_count.txt","\t", escape_double = FALSE, col_names = c("species","genome","geneCount"),   trim_ws = TRUE) %>% select(-species)
duplicate_genes_ext_02 <- merge(duplicate_genes_ext,gene_count,by="genome") %>% mutate(duplicatedPercent=100*(counts/geneCount))



duplicate_genes_ext_02$species <- plyr::revalue(duplicate_genes_ext_02$species,c("Ldel"="L. delbrueckii","Sterm"="S. thermophilus"))

##add seq-technology
genomeAssemblyStats <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",   col_types = cols(lineage = col_character())) %>% select(strain,lineage, Ref, Sequencing) %>%replace(is.na(.), "Illumina")

duplicate_genes_ext_02_tmp <- merge(duplicate_genes_ext_02,genomeAssemblyStats,by.y="strain",by.x="genome",all.x = TRUE )
duplicate_genes_ext_02 <- duplicate_genes_ext_02_tmp %>% filter(genome!="L11063") %>% filter(genome!="S50")
is.na(duplicate_genes_ext_02$Sequencing)

colorsss <- c("#fdc2b5ff","#007fa6ff")

# genesDuplicatedPlot_01 <- ggplot(duplicate_genes_ext_02)+geom_boxplot(aes(x=species,y=duplicatedPercent,fill=species),outlier.size = 0)+scale_fill_manual(values = colorsss) + geom_point(aes(x=Sequencing,y=duplicatedPercent,fill=species),pch = 21, position = position_jitterdodge())+theme_classic()+theme(legend.title = element_blank(),legend.position = "right",axis.title.x = element_blank(),axis.text.x =element_blank())+labs(y="duplicated genes [%]")
# genesDuplicatedPlot_01

genesDuplicatedPlot_01 <- ggplot(duplicate_genes_ext_02,aes(x=Sequencing,y=duplicatedPercent,fill=species))+geom_boxplot(outlier.shape=NA)+scale_fill_manual(values = colorsss) + geom_point(pch = 21, position = position_jitterdodge())+theme_classic()+theme(legend.title = element_blank(),legend.position = "right",axis.title.x = element_blank(),strip.background = element_blank(),   strip.text.x = element_blank())+labs(y="duplicated genes [%]")+facet_wrap(~species)
genesDuplicatedPlot_01

#  library(plotly)
# ggp <- ggplotly(genesDuplicatedPlot_01)
#  ggp

duplicate_genes_ext_02 %>% group_by(species) %>% summarise(median_count=median(duplicatedPercent),sd_count=sd(duplicatedPercent))


###-----------------------------------
#number of copies
###-----------------------------------
# duplicate_only_copies <- duplicate_with_eggnog %>% select(c(cluster,Orthogroup_new,copies)) %>% unique() 


duplicate_with_eggnog_02 <- merge(duplicate_with_eggnog,genomeAssemblyStats,by.x="genome",by.y="strain") %>% filter(Ref=="Ref")

duplicate_with_copies <- duplicate_with_eggnog_02 %>% select(c(cluster,Orthogroup_new,copies,species)) %>% unique() %>% group_by(Orthogroup_new,species) %>% summarise(counts=n(),medianCopies=mean(copies),stdvCopies=sd(copies))

100*(table(duplicate_only_copies$copies)/length(duplicate_only_copies$copies))

duplicate_with_copies_with_eggnog <- merge(duplicate_with_copies,lineageAbundance_presence_count_eggnog_COG,by.x = "Orthogroup_new",by.y="query_name") %>% select(c(species,Orthogroup_new,counts,medianCopies,stdvCopies,long))

ggplot(duplicate_only_copies,aes(x=copies))+geom_bar()+theme_classic()

ggplot(duplicate_with_copies_with_eggnog,aes(x=medianCopies,y=counts,color=long))+geom_point()+theme_classic()+coord_trans(x="log2",y="log2")

##add percent of genomes and sd bars

duplicate_with_eggnog %>% select(c(species,genome)) %>% unique() %>% group_by(species) %>% summarise(n())
number_genomes <- data.frame(species=c("Sterm","Ldel"),numGenome=c(12,2))

# duplicate_with_copies_with_eggnog_ext %>% filter(rel_counts>100)

# duplicate_with_copies_with_eggnog_ext <- merge(duplicate_with_copies_with_eggnog,number_genomes,by="species") %>% mutate(rel_counts=100*(counts/numGenome)) %>% mutate(rel_counts=ifelse(rel_counts>100,100,rel_counts)) %>% mutate(xmins=medianCopies-stdvCopies,xmaxs=medianCopies+stdvCopies) %>%  mutate(xmaxs=replace_na(xmaxs, 0.01))%>%  mutate(xmins=replace_na(xmins, 0.01))

duplicate_with_copies_with_eggnog_ext <- merge(duplicate_with_copies_with_eggnog,number_genomes,by="species") %>% mutate(rel_counts=100*(counts/numGenome)) %>% mutate(rel_counts=ifelse(rel_counts>100,100,rel_counts)) %>% mutate(xmins=medianCopies-stdvCopies,xmaxs=medianCopies+stdvCopies) %>%  mutate(xmins=ifelse(xmins<=2,2,xmins)) %>%  mutate(xmaxs=ifelse(xmaxs<=2,2,xmaxs))%>%  mutate(xmaxs=replace_na(xmaxs, 2))%>%  mutate(xmins=replace_na(xmins, 2))
range(duplicate_with_copies_with_eggnog_ext$xmins)

countsPlot <- ggplot(duplicate_with_copies_with_eggnog_ext,aes(x=medianCopies,y=rel_counts,color=long))+geom_point()+theme_classic()+coord_trans(x="log2",y="log2")+theme(legend.position = "none")
countsPlot

countsPlot_errorbar <- ggplot(duplicate_with_copies_with_eggnog_ext,aes(x=medianCopies,y=rel_counts,color=species,xmin=xmins,xmax=xmaxs))+geom_point()+theme_classic()+theme(legend.position = "none")+geom_errorbar(alpha=0.3,width=0)+scale_color_manual(values = colorsss)+labs(y="percent of lineage [%]",x="duplication copies")+ scale_y_continuous(trans="log2",breaks =c(8,20,50,100))+scale_x_continuous(trans="log2",breaks =c(2,5,10,20,30))
countsPlot_errorbar

# +coord_trans(x="log2",y="log2")
x_axis_plot <- ggplot(duplicate_with_copies_with_eggnog_ext,aes(x=medianCopies,color=species))+geom_density(adjust = 3)+theme_classic()+coord_trans(x="log2")+theme(legend.position = "none")+scale_color_manual(values = colorsss)+theme(axis.title.x = element_blank(),axis.text.x = element_blank(),axis.ticks.x=element_blank()) #+labs(y="number of genomes [%]",x="duplication copies")

y_axis_plot <-ggplot(duplicate_with_copies_with_eggnog_ext,aes(x=rel_counts,color=species))+geom_density(adjust = 3)+theme_classic()+coord_trans(x="log2")+theme(legend.position = "none")+scale_color_manual(values = colorsss)+coord_flip()+theme(axis.title.y = element_blank(),axis.text.y = element_blank(),axis.ticks.y=element_blank())  #+labs(y="number of genomes [%]",x="duplication copies")

# library(patchwork)

# x_axis_plotcountsPlot_errorbar
library(grid)


 blank <- grid.rect(gp=gpar(col="white"))

 library(gridExtra)
 
 grid.arrange(x_axis_plot,blank,countsPlot_errorbar,y_axis_plot, ncol=2, heights=c(1,3), widths=c(3,1))

 
 pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/duplicatedGenes_copies.pdf",width=6,height=6)

grid.arrange(x_axis_plot,blank,countsPlot_errorbar,y_axis_plot, ncol=2, heights=c(1,3), widths=c(3,1))

dev.off()


###-----------------------------------
#cog ctagroies
###-----------------------------------
 
ggplot(duplicate_with_copies_with_eggnog,aes(x=medianCopies,color=long))+geom_density()+theme_classic()+coord_trans(x="log2")
ggplot(duplicate_with_copies_with_eggnog,aes(x=medianCopies,color=long))+geom_histogram()+theme_classic()+coord_trans(x="log2")


###-----------------------------------
#final 
###-----------------------------------

library(patchwork)

genesDuplicatedPlot_01+countsPlot_errorbar+ plot_layout(ncol=2,widths = c(1,3))


pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/duplicatedGenes.pdf",width=6,height=4)

genesDuplicatedPlot_01+countsPlot_errorbar+ plot_layout(ncol=2,widths = c(1.5,3))

dev.off()


pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/duplicatedGenes.pdf",width=3,height=6)

genesDuplicatedPlot_01+theme(legend.position = "none")

dev.off()

```


Here, I use [nucmer](http://mummer.sourceforge.net/) to do the analysis of the largest repeats

```{bash}
  
###==========================
##nucmer
###==========================
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all
for species in $(echo "Ldel Sterm")
do
echo $species

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ | grep ".fna$" |sed 's/.fna//g')
do
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/${species}/${genomesss}
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/${species}/${genomesss}
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/${species}/${genomesss}


nucmer -b 100 -c 50 -maxmatch -nosimplify /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/${genomesss}.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/${genomesss}.fna

delta-filter -i 85 -l 2500   out.delta > ${genomesss}_nucmer.txt
##split files
split -t ">" -l 1 ${genomesss}_nucmer.txt
rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/${genomesss}_final.txt
for subsetsss in $(ls |grep "^x" |grep -v "xaa")
do

genomii1=$(grep "[[:space:]]" ${subsetsss} |head -1 |cut -d ' ' -f 1)
genomii2=$(grep "[[:space:]]" ${subsetsss} |head -1 |cut -d ' ' -f 2)

for linesss in $(seq 1 1 $(grep "[[:space:]]" ${subsetsss} |sed '1d' |wc -l))
do
start1=$(grep "[[:space:]]" ${subsetsss} |sed '1d' |sed -n "${linesss}p"  |cut -d ' ' -f 1)
end1=$(grep "[[:space:]]" ${subsetsss} |sed '1d'  |sed -n "${linesss}p"|cut -d ' ' -f 2)
start2=$(grep "[[:space:]]" ${subsetsss} |sed '1d' |sed -n "${linesss}p" |cut -d ' ' -f 3)
end2=$(grep "[[:space:]]" ${subsetsss} |sed '1d'  |sed -n "${linesss}p"|cut -d ' ' -f 4)

#echo -e ${genomii1}"\t"${start1}"\t"${end1}"\t"${genomii2}"\t"${start2}"\t"${end2} | awk -F "\t" '{OFS="\t"}{if($1!=$4 6) print $0}' >> /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/${genomesss}_final.txt


echo -e ${genomii1}"\t"${start1}"\t"${end1}"\t"${genomii2}"\t"${start2}"\t"${end2} | awk -F "\t" '{OFS="\t"}{if($1!=$4 || $2!=$5 || $3!=$6 ) print $0}' >> /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/${genomesss}_final.txt

done #linesss

done #subsetsss

 awk -F "\t" -v spezz="$species" -v genomezzz="$genomesss" '{OFS="\t"}{print spezz,genomezzz,$0,$3-$2,$6-$5}' /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/${genomesss}_final.txt >> /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/all_final.txt

done #genomesss

done #species

#---tests

#genomesss=L11071
#subsetsss=xas
#linesss=1


```

look for repeated operons

```{bash}

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/
scp  vincent@130.223.51.116:/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/all_final.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/

```

with nucmer I look for repeated regions
```{r}
all_final_operons_duplicated <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/all_final.txt", "\t", escape_double = FALSE, col_names = c("species","genome","contig1","start1","end1","contig2","start2","end2","diff1","diff3"),  trim_ws = TRUE)


ggplot(all_final_operons_duplicated,aes(x=diff1))+geom_histogram()+theme_classic()+facet_wrap(~species)+coord_trans(x="log2")+geom_vline(xintercept = 5000)

all_final_operons_duplicated_long <- all_final_operons_duplicated %>% filter(diff1>6000)

##------------------------------
#find overlaps
##------------------------------



GenomicRanges::findOverlaps

```

#### Enrichment analysis

Here, I aim to do the enrichment analysis. 

```{r}


library(tidyverse)
library(topGO)
library(ALL)
data(ALL)
data(geneList)

affyLib <- paste(annotation(ALL), "db", sep = ".")
library(package = affyLib, character.only = TRUE)

 sum(topDiffGenes(geneList))
 
 sampleGOdata <- new("topGOdata",
                     description = "Simple session", ontology = "BP",
                     allGenes = geneList, geneSel = topDiffGenes,
                     nodeSize = 10,
                     annot = annFUN.db, affyLib = affyLib)
 
 
```






#Phylogeny

## Tree of life

Here, I download the tree of life and check who has the lac operon annotated. 

```{bash}
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/genome/GFF

scp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree/LAB_renaming_list.csv vincent@130.223.51.66:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/LAB_renaming_list.csv

scp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree/download.list vincent@130.223.51.66:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/

scp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree/data vincent@130.223.51.66:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/

scp -r  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree/TOL_rrna/ vincent@130.223.51.66:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/


scp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree/name_TOL.txt vincent@130.223.51.66:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/

sed 's/.fna.gz/.gff.gz/g' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download.list > /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download_gff.list

##------------------------------------------------------------
###----------------------------------------------------download gffs
##------------------------------------------------------------
mkdir -p  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/genome_LAB/FNA
cd /work/Projects/2020_StarterCultureDiversity/11_treeoflife/genome_LAB/FNA

rm /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download_lab.list
for labss in $(awk -F "\t" '{OFS="\t"}{if($25=="Lactobacillales") print $3}' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data )
do

grep "${labss}" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download.list >>  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download_lab.list

done

awk -F "\t" '{OFS="\t"}{if($25=="Lactobacillales") print $3}' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data| wc -l
wc -l /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download_lab.list

wget -i /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download_gff.list
###----------------------------------------------------download gffs

cd /work/Projects/2020_StarterCultureDiversity/11_treeoflife/genome/GFF

wget -i /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download_gff.list


##-------------------------------------change nomenclature
#cp /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data_oldNAMES

cp /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data_oldNAMES /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data


sed 's/ /-/g' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/LAB_renaming_list.csv |sed 's/--/-/g'  > /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/LAB_renaming_list_noWhiteSpaces.csv

for NEwsss in $(awk -F "\t" '{OFS="\t"}{print $15}' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/LAB_renaming_list_noWhiteSpaces.csv|sort|uniq|awk -F "-ssp" '{OFS="\t"}{print $1}')
do
oldsss=$( grep "$NEwsss" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/LAB_renaming_list_noWhiteSpaces.csv|awk -F "\t" '{OFS="\t"}{print $3}'|sort|uniq|awk -F "-ssp" '{OFS="\t"}{print $1}'|head -1)
echo -e ${oldsss}" to "${NEwsss}


NewFinalss=$(echo $NEwsss |sed 's/-/ /g')
OLDFinalss=$(echo $oldsss |sed 's/-/ /g')

sed -i "s/${OLDFinalss}/${NewFinalss}/g" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data


done




##---------------------------------------------------------
## check for order lactobacillales (n=493)
##---------------------------------------------------------


awk -F "\t" '{OFS="\t"}{if($25=="Lactobacillales") print $0}' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |wc -l

awk -F "\t" '{OFS="\t"}{if($25=="Lactobacillales") print $0}' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |cut -f 27|sort|uniq -c

awk -F "\t" '{OFS="\t"}{if($25=="Lactobacillales") print $0}' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |cut -f 28|sort|uniq -c



##------------------------------------------------------------
###----------------------------------------------------get longest rRNA lab files
##------------------------------------------------------------
rm -r /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species
rm /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/rRNAs_lab.log
cd /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna

#samtools faidx 23s.fa
#samtools faidx 5s.fa
#samtools faidx 16s.fa
labss=$(grep "Bacillus pseudofirmus" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data|grep "Complete" -i|grep "repres"|cut -f 1)
labss=$(grep "Fusobacterium nucleatum" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data|grep "Complete" -i|grep "repres"|cut -f 1)


for labss in $(awk -F "\t" '{OFS="\t"}{if($25=="Lactobacillales") print $1}' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data )
do

speciesss=$(grep "^${labss}" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data|cut -f 28|sed 's/ /-/g')


echo $labss
for rrnasss in $(echo "23 16 5")
do

grep "^${labss}" ${rrnasss}s.tsv > tmp.tsv
countss=$(wc -l tmp.tsv|cut -d ' ' -f 1)

head -1 tmp.tsv|awk -F "\t" -v countzz="$countss" -v rrnazzz="$rrnasss" '{OFS="\t"}{print $0,$6-$5,rrnazzz"s",countzz}' >>  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/rRNAs_lab.log


#grep "^>${labss_1}" ${rrnasss}s.fa |tail -1 >> LAB/${rrnasss}s_LAB.fa

done #rrnasss


echo ">${speciesss}_${labss}" > LAB/species/${speciesss}_${labss}_rRNAoperon.fasta

grep "^>${labss}_1" -A 1 16s.fa |tail -1 >> LAB/species/${speciesss}_${labss}_rRNAoperon.fasta
echo "NNNNNNNNNNNN" >> LAB/species/${speciesss}_${labss}_rRNAoperon.fasta
grep "^>${labss}_1" -A 1  23s.fa |tail -1 >> LAB/species/${speciesss}_${labss}_rRNAoperon.fasta
echo "NNNNNNNNNNNN" >> LAB/species/${speciesss}_${labss}_rRNAoperon.fasta
grep "^>${labss}_1" -A 1  5s.fa |tail -1 >> LAB/species/${speciesss}_${labss}_rRNAoperon.fasta


##make the assembly fit together nicely
awk 'BEGIN{RS=">"}NR>1{sub("\n","\t"); gsub("\n",""); print RS$0}' LAB/species/${speciesss}_${labss}_rRNAoperon.fasta |sed 's/\t/\n/g' |awk -v FS= '/^>/{print;next}{for (i=0;i<=NF/60;i++) {for (j=1;j<=60;j++) printf "%s", $(i*60 +j); print ""}}' > LAB/species/${speciesss}_${labss}_rRNAoperon_prep.fasta

rm LAB/species/${speciesss}_${labss}_rRNAoperon.fasta
cat LAB/species/${speciesss}_${labss}_rRNAoperon_prep.fasta >> LAB/all_rRNAoperon_prep.fasta

done

##------QC
grep -c ">" LAB/all_rRNAoperon_prep.fasta
awk -F "\t" '{OFS="\t"}{if($25=="Lactobacillales") print $3}' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data| wc -l
seq_length.py LAB/all_rRNAoperon_prep.fasta |sort -n -k 3

cut -f 1 /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/rRNAs_lab.log|sort|uniq -c |sort -n |sed 's/      //g' |cut -d ' ' -f 1|sort|uniq -c

grep "5s" -v /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/rRNAs_lab.log|cut -f 1 |sort|uniq -c |sort -n |sed 's/      //g' |cut -d ' ' -f 1|sort|uniq -c

##-----------------remove genomes that are missing 16s or 23s

rm  removed_because_missing_genes.txt
for genomesss in $(grep "5s" -v /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/rRNAs_lab.log|cut -f 1 |sort|uniq -c |sort -n |sed 's/      //g' |awk -F " " '{OFS=" "}{if($1==1)print $2}')
do

grep "^${genomesss}"  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data >> removed_because_missing_genes.txt

done

cut -f 28  removed_because_missing_genes.txt


###----------extract all genomes if at least 16s and 23s and add Lactobacillus equicursoris
genomesss=$(grep "Bacillus pseudofirmus" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data|grep "Complete" -i|grep "repres"|cut -f 1)
genomesss=$(grep "Fusobacterium nucleatum" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data|grep "Complete" -i|grep "repres"|cut -f 1)

rm keeped_genomes.txt
for genomesss in $(grep "5s" -v /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/rRNAs_lab.log|cut -f 1 |sort|uniq -c |sort -n |sed 's/      //g' |awk -F " " '{OFS=" "}{if($1==2)print $2}')
do

speciesss=$(grep "^${genomesss}"  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |awk -F "\t" '{OFS="\t"}{print $28}'|sed 's/ /-/g')
genuss=$(grep "^${genomesss}"  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |awk -F "\t" '{OFS="\t"}{print $28}'|sed 's/ /-/g'|cut -d '-' -f 1)


grep "^${genomesss}"  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |awk -F "\t" -v speciezz="$speciesss" -v genuzzz="$genuss" '{OFS="\t"}{print $0,genuzzz,speciezz,speciezz"_"$1}' >> keeped_genomes.txt

done


genomesss$(grep " equicursoris" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |cut -f 1)

speciesss=$(grep "^${genomesss}"  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |awk -F "\t" '{OFS="\t"}{print $28}'|sed 's/ /-/g')
genuss=$(grep "^${genomesss}"  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |awk -F "\t" '{OFS="\t"}{print $28}'|sed 's/ /-/g'|cut -d '-' -f 1)

grep "^${genomesss}"  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |awk -F "\t" -v speciezz="$speciesss" -v genuzzz="$genuss" '{OFS="\t"}{print $0,genuzzz,speciezz,speciezz"_"$1}' >> keeped_genomes.txt


wc -l keeped_genomes.txt
grep "Streptococcus thermophilus" keeped_genomes.txt
grep "Lactobacillus delbrueckii" keeped_genomes.txt
grep "Lactobacillus helveticus" keeped_genomes.txt
grep "parafarraginis" keeped_genomes.txt

#Lentilactobacillus-parafarraginis_G000238835

###-----------------------
#add different lactobacillus subsp. 
###-----------------------
rm -r  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta
for labsss in $(grep "delbrueckii" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta |sed 's/>//g')
do

speciesss=$(echo $labsss |sed 's/rRNAoperon-//g' |sed 's/_/-/g')

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta ${labsss} |sed "s/${labsss}/${speciesss}_own/g" > /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${speciesss}_rRNAoperon_prep.fasta

done




###-----------------------
#add different Streptococcus vestibualris
###-----------------------
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta
for labsss in $(grep "vestibul" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta |sed 's/>//g')
do

speciesss=$(echo $labsss |sed 's/rRNAoperon-//g' |sed 's/_/-/g')

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta ${labsss} |sed "s/${labsss}/${speciesss}_own/g" > /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${speciesss}_rRNAoperon_prep.fasta

done

###-----------------------
##----------------------------add missing speciess!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
###-----------------------

#"L. amylovorus"   "L. bombicola"    "L. equicursoris" "L. gallinarum"   "L. johnsonii"    "L. kimbladii"    "L. kitasatonis"  "L. panisapium"   "L. rodentium"    "L. taiwanensis"  "L. ultunensis"   "S. downei"       "S. equinus"      "S. pneumoniae"

rm -r /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs
#mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/download


seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta
seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta

for genomesss in $(echo "amylovorus bombicola equicursoris gallinarum johnsonii kimbladii kitasatonis panisapium taiwanensis ultunensis")
do

namesss=$(grep "${genomesss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta |sed 's/>//g')

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta ${namesss} |sed "s/>.*/>Lactobacillus-${genomesss}_oldAnalysis/g" > /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs/Lactobacillus-${genomesss}_oldAnalysis_rRNAoperon.fasta


done
ll /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs


for genomesss in $(echo "downei equinus pneumoniae")
do

namesss=$(grep "${genomesss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta |sed 's/>//g')

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta ${namesss} |sed "s/>.*/>Streptococcus-${genomesss}_oldAnalysis/g" > /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs/Streptococcus-${genomesss}_oldAnalysis_rRNAoperon.fasta


done
ll /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs

for genomess in $(ls /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs)
do
echo ${genomess}
cp /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs/${genomess} /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${genomess}
done
###-----------------------
###add good genes
###-----------------------

speciesss=$(cut -f 56 /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes.txt |cut -d '_' -f 1|sort|uniq |grep -v "acidilactici" |head -1)
rm /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt
for speciesss in $(cut -f 56 /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes.txt |cut -d '_' -f 1|sort|uniq |grep -v "acidilactici" )
do


#samplesss=$(ls /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species/ |grep "${speciesss}_" |head -1)
samplesss=$(grep "${speciesss}" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes.txt |cut -f 56|head -1)
newNAME=$(echo $samplesss)
cp /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species/${samplesss}_rRNAoperon_prep.fasta /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${newNAME}_rRNAoperon.fasta

grep "${newNAME}" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes.txt >> /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt

#grep "${newNAME}" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes.txt|cut -f 56

done

wc -l /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt

ls /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/ |wc -l 

cat /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/*.fasta >  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa

grep ">" -c /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa

seq_length.py  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa

###---------------------------add outgroup

###---------------------------remove 
rm /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa

for genomess in $(grep ">"  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa |grep -v "sp.-X13SY08"|grep -v "sp.-HMSC24D01"|grep -v "sp.-wkB10"| grep -v "i_G000056065" |sed 's/>//g')
  do  
samtools faidx  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa  ${genomess} >> /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa


done

grep ">" -c /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa
grep ">" -c /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa


 for genomess in $(ls /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs)
do
echo ${genomess}
cat /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs/${genomess} >> /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa
done

grep ">" -c /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa

grep ">"  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa


```

##### extract 16s of only strepto/lacto

I am doing this for Flo

```{bash}
#...............................
#count lactose 
#...............................


##-------------------------------vestibularis 


cat /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/Streptococcus-vestibularis_rRNAoperon_prep.fasta >>  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/Streptococcus-vestibularis_own_rRNAoperon.fasta

##-------------------------------Sterptos 

rm /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa


for genomess in $(grep ">Streptococcus-"  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa |sed 's/>//g')
  do  
  #grep -A 3 "${genomess}" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa
#samtools faidx  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa  ${genomess} >> /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa

cat  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${genomess}_rRNAoperon.fasta >> /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa

done

##-------------------------------subsp. delbrueckii 

for genomess in $(ls /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/ |grep "delbrueckii-" | sed 's/_rRNAoperon_prep.fasta//g')
do


cat /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${genomess}_rRNAoperon_prep.fasta >  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${genomess}_own_rRNAoperon.fasta
head /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${genomess}_own_rRNAoperon.fasta

done

##-------------------------------Lactops 

for genomess in $(grep ">Lactobacillus-"  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa |sed 's/>//g'|grep -v "delbrueckii_G000056065")
  do  
  
#samtools faidx  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa  ${genomess} >> /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa

cat  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${genomess}_rRNAoperon.fasta >> /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa

done

grep ">" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa

#sed -i 's/delbrueckii_G000056065/delbrueckii-bulgaricus_G000056065/g' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa
grep ">" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa

sed -i 's/_.*$//g' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa
grep ">" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa 
grep ">" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa  |wc -l
grep ">" /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa |sort|uniq |wc -l


##move local
scp vincent@130.223.110.206:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/Earthmicrobiome_data/all_rRNAoperon_new_ForFLO.fa


```

##### check for Lac operon

Check the closely related lactobacillus and strepotococcus genomes for the presence of the lac operon

```{bash}


##££££££££££££££££££££££££££££££££ do this for all strains in the species
cd /work/Projects/2020_StarterCultureDiversity/11_treeoflife/genome/GFF


#genomess=GCF_001808965.1
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/lacOPERON/
rm /work/Projects/2020_StarterCultureDiversity/11_treeoflife/lacOPERON/lactose_streps_lactos.txt
#cut -f 56 /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt
for genomess in $(awk -F "\t" '{OFS="\t"}{if($56~"Lactobacillus" || $56~"Streptococcus")print $3}' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt)
do
echo ${genomess}
#zgrep lactose -c -i ${genomess}*.gff.gz |cut -d ':' -f 2
#zgrep "=lactose" -c -i ${genomess}*.gff.gz |cut -d ':' -f 2
#zgrep "3.2.1.23" -i ${genomess}*.gff.gz
#zgrep "lacZ" -i ${genomess}*.gff.gz
#zgrep "galactosidase" -i ${genomess}*.gff.gz

#countss=$(zgrep "beta-galactosidase" -A 2 -B 2 -i ${genomess}*.gff.gz|grep "PTS" -i |wc -l)
countss=$(zgrep "galactosidase" -A 2 -B 2 -i ${genomess}*.gff.gz)

namess=$(awk -F "\t" -v genomzzz="$genomess" '{OFS="\t"}{if($3==genomzzz)print $56}' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt)

#countss=$(zgrep "beta-galactosidase" -c -i ${genomess}*.gff.gz |cut -d ':' -f 2)

#echo -e ${namess}"\t"${countss} >> /work/Projects/2020_StarterCultureDiversity/11_treeoflife/lacOPERON/lactose_streps_lactos.txt
echo -e ${namess}"\t"${countss} #>> /work/Projects/2020_StarterCultureDiversity/11_treeoflife/lacOPERON/lactose_streps_lactos.txt


#zgrep "galactohydrolase" -i ${genomess}*.gff.gz

#zgrep "lacS" -i ${genomess}*.gff.gz
#zgrep "lacZ" -i ${genomess}*.gff.gz

done

cut -f 2 /work/Projects/2020_StarterCultureDiversity/11_treeoflife/lacOPERON/lactose_streps_lactos.txt|sort|uniq -c



genomess=GCF_000422165.1 #lhelv
genomess=GCF_000056065.1 #Ldel
genomess=GCF_000253395.1 #Sterm

##-------------------------
#transfer local
##-------------------------



scp vincent@130.223.51.66:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/lacOPERON/lactose_streps_lactos.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/lactose_streps_lactos.txt


```


new
```{bash}


rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/lactose_utilziation/presence_lacOPERON.txt

for alltoegether in $(grep ">" /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/Earthmicrobiome_data/all_rRNAoperon_new_ForFLO.fa |sed 's/>//g' )
do
genusss=$(echo $alltoegether|sed 's/-delbrueckii//g'  |cut -d '-' -f 1 )
species=$(echo $alltoegether |sed 's/-delbrueckii//g' |cut -d '-' -f 2 )

lacZ=$(grep "${genusss}" /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/lactose_utilziation/tax_report_LACz.txt |grep -c "${species}"  )
lacG=$(grep "${genusss}" /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/lactose_utilziation/tax_report_LACg.txt |grep -c "${species}"  )

echo -e ${alltoegether}"\t"${lacZ}"\t"${lacG}
echo -e ${alltoegether}"\t"${lacZ}"\t"${lacG} >>  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/lactose_utilziation/presence_lacOPERON.txt

done


```

##### check glycosidase hydrolases

```{bash}

##-----------------------------------
#01-create DBcan2 Db
##-----------------------------------
cd /work/Databases/20211209_DBcan2_v10

wget https://bcb.unl.edu/dbCAN2/download/dbCAN-HMMdb-V10.txt

##-----------------------------------
#HMM the faa files
##-----------------------------------
#mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FAA/GCF_900322585.1_Lactobacillus_delbrueckii_subsp._lactis_protein.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FAA/Lactobacillus_delbrueckii_subsp_lactis-7.faa

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/anlayis/HMM_GH/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/anlayis/HMM_GH/
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FAA/ |sed 's/.faa//g' )
do

echo "=================================="
echo ${genomesss}


hmmscan --cpu 5 --domtblout anlayis/HMM_GH/$genomesss.faa.out.dm /work/Databases/20211209_DBcan2_v10/dbCAN-HMMdb-V10.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FAA//$genomesss.faa >  anlayis/HMM_GH/$genomesss.faa.out


done

##-----------------------------------------------------------------------
#parse
##-----------------------------------------------------------------------

rm anlayis/HMM_GH/hmm_analysis.txt
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FAA/ | grep ".faa" |sed 's/.faa//g' )
do

echo "=================================="
echo ${genomesss}


##---filter
sh /work/Databases/20211209_DBcan2_v10/hmmscan-parser_03.sh  anlayis/HMM_GH/$genomesss.faa.out.dm > anlayis/HMM_GH/$genomesss.faa.out_parsed

countss=$(wc -l anlayis/HMM_GH/$genomesss.faa.out_parsed |cut -d ' ' -f 1)

echo -e  ${genomesss}"\t"${countss} >> anlayis/HMM_GH/hmm_analysis.txt
done


##-----------------------------------------------------------------------
#move local
##-----------------------------------------------------------------------
mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/anlayis/HMM_GH

scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/anlayis/HMM_GH/Streptococcus_oralis-16.faa.out.dm /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/tmp/


scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/anlayis/HMM_GH/hmm_analysis.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/anlayis/HMM_GH/hmm_analysis.txt


```

rAnalysis

```{r}

library(tidyverse)

hmm_analysis <- read_table2("~/Desktop/Projects/2020_StarterCultureDiversity/anlayis/HMM_GH/hmm_analysis.txt",  col_names = c("strain","GH")) %>% mutate(species=gsub("-.*$","",strain))

table(hmm_analysis$species)

hmm_analysis_median <-  hmm_analysis %>% group_by(species) %>% summarise(genome_count=n(),median_GH=median(GH))
hmm_analysis_median
write.csv(hmm_analysis_median,"~/Desktop/Projects/2020_StarterCultureDiversity/anlayis/HMM_GH/hmm_analysis_prepped.txt",quote = FALSE,row.names = FALSE)


```





##### check number of complete genomes NCBI

```{bash}



mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
#curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt


mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log

for genomess in $(awk -F "\t" '{OFS="\t"}{if($56~"Lactobacillus" || $56~"Streptococcus")print $56}' /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt |sed "s/_.*$//g"|grep "sp." -v |grep "delbrueckii" -v)
do
genomes_correct=$(echo ${genomess} |sed 's/-/ /g')
echo ${genomes_correct} 

#grep -c ${genomes_correct} /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt
countsss=$(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e "${genomes_correct}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'| awk 'BEGIN{FS="\t";OFS="\t"}  {print $20}'| sed -r 's|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' |wc -l ) #>    \
   # /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt



cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e "${genomes_correct}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'| awk 'BEGIN{FS="\t";OFS="\t"}  {print $20}'| sed -r 's|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|'  >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e "${genomes_correct}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | awk -v namezzin="$genomes_correct" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names

echo -e ${genomes_correct}"\t"${countsss} >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt



echo "================================================================="
done


###-------------------------------------
#vestibularis
###-------------------------------------
genomes_correct=$(echo "Streptococcus vestibularis")

#grep -c ${genomes_correct} /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt
countsss=$(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e "${genomes_correct}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'| awk 'BEGIN{FS="\t";OFS="\t"}  {print $20}'| sed -r 's|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' |wc -l ) #>    \
   # /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt



cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e "${genomes_correct}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'| awk 'BEGIN{FS="\t";OFS="\t"}  {print $20}'| sed -r 's|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|'  >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e "${genomes_correct}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | awk -v namezzin="$genomes_correct" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names

echo -e ${genomes_correct}"\t"${countsss} >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt

###-------------------------------------
#lactobacillus delbrueckiis
###-------------------------------------
genomes_correct=$(echo "Lactobacillus delbrueckii")


for subsss in $(echo "
bulgaricus
lactis
indicus
delbrueckii
sunkii
jakobsenii"
)
do
genomes_correct=$(echo "${subsss}" |awk '{OFS=" "}{print "Lactobacillus delbrueckii subsp.",$0}')

echo ${genomes_correct} 

#grep -c ${genomes_correct} /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt

#grep -c ${genomes_correct} /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt

#grep -c ${genomes_correct} /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt
countsss=$(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e "${genomes_correct}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'| awk 'BEGIN{FS="\t";OFS="\t"}  {print $20}'| sed -r 's|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' |wc -l ) #>    \
   # /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt



cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e "${genomes_correct}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'| awk 'BEGIN{FS="\t";OFS="\t"}  {print $20}'| sed -r 's|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|'  >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e "${genomes_correct}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | awk -v namezzin="$genomes_correct" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names

echo -e ${genomes_correct}"\t"${countsss} >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt

done



wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log

awk 'BEGIN {FS = "\t"} ; {sum+=$2} END {print sum}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt

awk '{FS = "\t"}  {if($2==0) print $0}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt


###---------------------------------------------------------------
#prepare names
###---------------------------------------------------------------
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned
for speciesss_old in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names|sed 's/ /_/g' |sort|uniq)
do
echo "================================================"
echo $speciesss_old
num=1
speciesss=$(echo $speciesss_old |sed 's/_/ /g')

for genomesss in $(grep "${speciesss}" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names |cut -f 1)
do

genomess_name_new=$(echo ${speciesss}|sed 's/ /_/g' |awk -v countss="$num" '{print $1"-"countss}')

grep "${genomesss}" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names |awk -v countss="$genomess_name_new" '{OFS="\t"}{print $0,countss}' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned


num=$((num+1))

done

done


sed -i 's/subsp._/subsp_/g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned

##===================================================================================================
###---------------------------------------------------------------
#download genomes
###---------------------------------------------------------------
##===================================================================================================
sed -r 's|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|'
whichFILE=fna

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/

rm *

wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log


###-------------------------------------
##change name
###-------------------------------------

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned| cut -d '/' -f 10)
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned|cut -f 3 )
echo "-----------------------"
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz

#mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA//GCF_900322585.1_Lactobacillus_delbrueckii_subsp._lactis_protein.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/Lactobacillus_delbrueckii_subsp_lactis-7.fna


##===================================================================================================
###---------------------------------------------------------------
#download faa
###---------------------------------------------------------------
##===================================================================================================
whichFILE=faa

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FAA/
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FAA/

sed 's/_genomic.fna./_protein.faa./g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/${whichFILE}_download.log

rm *

wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/${whichFILE}_download.log


###-------------------------------------
##change name
###-------------------------------------

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FAA/

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned| cut -d '/' -f 10)
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned|cut -f 3 )
echo "-----------------------"
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz

##===================================================================================================
###---------------------------------------------------------------
#download gff
###---------------------------------------------------------------
##===================================================================================================
whichFILE=gff

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/GFF/
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/GFF/

sed 's/_genomic.fna./_genomic.gff./g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/${whichFILE}_download.log

rm *

wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/${whichFILE}_download.log


###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned| cut -d '/' -f 10)
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned|cut -f 3 )
echo "-----------------------"
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz

##===================================================================================================
###---------------------------------------------------------------
#download gbff
###---------------------------------------------------------------
##===================================================================================================
whichFILE=gbff

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/GBFF/
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/GBFF/

sed 's/_genomic.fna./_genomic.gbff./g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/${whichFILE}_download.log

rm *

wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/${whichFILE}_download.log


###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned| cut -d '/' -f 10)
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned|cut -f 3 )
echo "-----------------------"
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz



##------------------------------------------------------------



##------------------------------------------------------
#move local
##------------------------------------------------------

scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/


```



##### make tree 

```{bash}
##------------------------------------------------------

##------------------------------------------------------
#make alignement
##------------------------------------------------------
rm -r /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/
cd /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/

#mafft-linsi --thread 37 --maxiterate 1000    /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa >  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/all_rRNAoperon.fa_aln

mafft-linsi --thread 37 --maxiterate 1000    /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa >  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/all_rRNAoperon.fa_aln

##------------------------------------------------------
#pyhlogeny
##------------------------------------------------------

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/all_rRNAoperon.fa_aln -n lactos -T 37
```


```{bash}
##------------------------------------------------------
#move local
##------------------------------------------------------

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03/
scp -r vincent@130.223.110.206:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/RA* /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03/

##------------------------------------------------------
#chnage names
##------------------------------------------------------

 #sed "s/_.*://g" /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large/RAxML_bipartitions.lactos > /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large/RAxML_bipartitions.lactos_02
 

##------------------------------------------------------
#start align with figtree
##------------------------------------------------------

java -jar /home/vincent/bin/apps/FigTree.v1.4.4/FigTree\ v1.4.4/lib/figtree.jar

```




in r after rerooting

```{r}
# 
# library(ape)
# library(ggtree)
# library(tidyverse)
# library(tidytree)
# library(treeio)
# library(TreeTools)
# 
# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large/RAxML_bipartitions.lactos_curated", tree.names = NULL, force.multi = FALSE)
# 
# 
# # x <- as_tibble(Sterm_tree_raw)
# 
# ##plot tree
# # ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
# tree_01 <- ggtree(Sterm_tree_raw, layout="circular") +
#   geom_tiplab(aes(angle=angle),align = TRUE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0)  #NO ROOT  
#   
# tree_01 <- ggtree(Sterm_tree_raw) +
#   geom_tiplab(size=0.5,align = FALSE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0)  #NO ROOT 
# 
#    png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png", width = 2200, height = 1500,res=400)
# # pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)
# 
# tree_02
# 
# 
# dev.off()
# 
# 
# 408=Oneococcus
# 409=Leuconostoc
# 398=Weissela
# #154=Furfurilactobacillus #is a tip
# 381=Limoslactibacillus
# 392=Paucilactobacillus
# 374=Lactiplantibacillus
# 353=Lentilactobacillus
# 348=Fructilactobacillus
# 351=Apilactobacillus
# 362=Secundilactobacillus
# 368=Levilactobacillus
# 337=Pediococcous
# 422=Liquorilactobacillus
# 
# 455=Lactobacillus
# 
# 479=Companilactobacillus
# 488=Bombilactobacillus
# 448=Lapidilactobacillus
# 446=Schleiferilactobacillus
# #177=Agrilactobacillus #tip
# 491=Lacticaseibacillus
# 500=Latilactobacillus
# #223=Paralactobacillus #tip
# 441=Loigolactobacillus
# 512=mix
# 504=Carnobacterium
# 
# 289=Streptococcus
# 
# 324=Lactococcus
# 
# ##enterococcus is polyphyletic
# 
# 
#  library(plotly)
# ggp <- ggplotly(tree_02)
#  ggp
#  
# tree_02  <- scaleClade(tree_01, 284, 2) %>%  scaleClade(., 452, 2)%>% collapse(409, 'max', fill="darkgreen")  %>% collapse(408, 'max', fill="darkgreen")   %>% collapse(398, 'max', fill="darkgreen") %>% collapse(381, 'max', fill="darkgreen")  %>% collapse(392, 'max', fill="darkgreen")  %>% collapse(374, 'max', fill="darkgreen")  %>% collapse(353, 'max', fill="darkgreen")   %>% collapse(348, 'max', fill="darkgreen")   %>% collapse(351, 'max', fill="darkgreen")   %>% collapse(362, 'max', fill="darkgreen")   %>% collapse(368, 'max', fill="darkgreen")   %>% collapse(337, 'max', fill="darkgreen")  %>% collapse(422, 'max', fill="darkgreen")  %>% 
# collapse(479, 'max', fill="darkgreen")    %>% collapse(488, 'max', fill="darkgreen")  %>% collapse(448, 'max', fill="darkgreen")  %>% collapse(446, 'max', fill="darkgreen")  %>% collapse(491, 'max', fill="darkgreen")  %>% collapse(500, 'max', fill="darkgreen")  %>% collapse(441, 'max', fill="darkgreen")  %>% collapse(512, 'max', fill="darkgreen")  %>% collapse(504, 'max', fill="darkgreen")  %>% collapse(324, 'max', fill="darkgreen") 
# 
# tree_02
#  
# 
# 
# ##-------------
# #change tiplabel and color according to rmk or non-rmk sample
# ##-------------
# 
# tmp = as.character(Sterm_tree_raw$strain)
# names(tmp) <- all_sterm_strain_info$shortName
# 
# # Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated", tree.names = NULL, force.multi = FALSE)
# Sterm_tree_raw$tip.label
# # Sterm_tree_raw$label
# 
# 
# 
# Sterm_tree_raw_tible <- as_tibble(Sterm_tree_raw) 
# Sterm_tree_raw_tible$label
# Sterm_tree_raw_tible$label <- gsub("_.*","",Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label <- gsub("Streptococcus-","S. ",Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label <- gsub("Lactobacillus-","L. ",Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label <- gsub("Enterococcus-","E. ",Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label <- gsub("Tetragenococcus-","T. ",Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label <- gsub("Holzapfelia-","H. ",Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label <- gsub("Amylolactobacillus-","A. ",Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label <- gsub("L. delbrueckii$","L. deblrueckii-delbrueckii",Sterm_tree_raw_tible$label)
# 
# 
# tree_02_tmp <- Sterm_tree_raw_tible %>% as.treedata()
# 
# # Sterm_tree_raw_tible$label <- plyr::revalue(Sterm_tree_raw_tible$label,tmp)
# tree_02 <- Sterm_tree_raw_tible%>% as.treedata()
# 
# 
# ##----------------------------------
# #after branch collaps
# ##----------------------------------
# 
# tree_03 <- ggtree(tree_02, layout="circular") +
#   geom_tiplab(size=3.5,align = FALSE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0)  
#    tree_04 <- scaleClade(tree_03, 284, 2.5) %>%  scaleClade(., 452, 2.5)%>% collapse(409, 'max', fill="darkgreen")  %>% collapse(408, 'max', fill="darkgreen")   %>% collapse(398, 'max', fill="darkgreen") %>% collapse(381, 'max', fill="darkgreen")  %>% collapse(392, 'max', fill="darkgreen")  %>% collapse(374, 'max', fill="darkgreen")  %>% collapse(353, 'max', fill="darkgreen")   %>% collapse(348, 'max', fill="darkgreen")   %>% collapse(351, 'max', fill="darkgreen")   %>% collapse(362, 'max', fill="darkgreen")   %>% collapse(368, 'max', fill="darkgreen")   %>% collapse(337, 'max', fill="darkgreen")  %>% collapse(422, 'max', fill="darkgreen")  %>% 
# collapse(479, 'max', fill="darkgreen")    %>% collapse(488, 'max', fill="darkgreen")  %>% collapse(448, 'max', fill="darkgreen")  %>% collapse(446, 'max', fill="darkgreen")  %>% collapse(491, 'max', fill="darkgreen")  %>% collapse(500, 'max', fill="darkgreen")  %>% collapse(441, 'max', fill="darkgreen")  %>% collapse(512, 'max', fill="darkgreen")  %>% collapse(504, 'max', fill="darkgreen")  %>% collapse(324, 'max', fill="darkgreen") 
# tree_04
# 
# 
# 
# 
# p2 <- ggtree(tree_02, layout="circular") +
#   geom_tiplab(aes(angle=angle,color=origin,size=origin),align = TRUE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0) + #NO ROOT  
#   scale_color_manual(values=c("grey66", 'grey55', 'black')) +
#   scale_size_manual(values=c(2.5,2.5,4)) 
# p2
# 
# ##-------------
# #add geograph and origin
# ##-------------
# all_sterm_strain_info$shortName <- as.character(all_sterm_strain_info$shortName)
# all_sterm_strain_info_ready <- all_sterm_strain_info %>% select(-shortName) %>% remove_rownames()%>% column_to_rownames(., var = "strain")%>%replace(is.na(.), "Unknown") 
# table(all_sterm_strain_info_ready$location_continent)
#  all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c("Europe",  "Asia", "USA", "Australia" , "Unknown","Outgroup"))
# # colors_area <- c("#464D77","#36827F","#F9DB6D","grey88")
# # colors_area <- c("#F9DB6D","#36827F","#464D77","grey88")
# 
# # location_dataset$area
# 
# # tree_02 <- full_join(as_tibble(tree), location_dataset, by='label') %>% as.treedata()
# 
# colnames(all_sterm_strain_info_ready) <- c("A","B")
# rownames(all_sterm_strain_info_ready)
# 
# p3 <- gheatmap(p2, all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
# p3
# 
# 
# 
# ##--------------------------------
# ##color labels of of only rmk phages
# ##--------------------------------
# 
# Sterm_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/logs/Sterm_genome_origin.txt",  "\t", escape_double = FALSE, col_names = c("label","origin"),  trim_ws = TRUE) %>% as.data.frame()
# 
# # tmp <- full_join(as_tibble(Sterm_tree_raw), Sterm_genome_origin, by='label')
# # table(tmp$origin)
# tree <- full_join(as_tibble(Sterm_tree_raw), Sterm_genome_origin, by='label') %>% as.treedata()
# 
# 
# p <- ggtree(tree, layout="circular") +
#   geom_tiplab(aes(angle=angle,color=origin,size=origin),align = TRUE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0) + #NO ROOT  
#   scale_color_manual(values=c("grey66", 'grey55', 'black')) +
#   scale_size_manual(values=c(4,4,7)) +geom_treescale()
# 
# p
# 
# ##--------------------------------
# #include scaleClade(p, node=17, scale=.1)  
# #include collapse(p2, node, 'mixed') %>% expand(node)
# #scaleClade(p, 23, .2) %>% collapse(23, 'min', fill="darkgreen")  
##--------------------------------

##==========================================================================================
#again 
#after adding S.vestibularis
##==========================================================================================


library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02//RAxML_bipartitions.lactos_curated", tree.names = NULL, force.multi = FALSE)


# x <- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
tree_01 <- ggtree(Sterm_tree_raw, layout="circular") +
  geom_tiplab(aes(angle=angle),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  tree_01
tree_01 <- ggtree(Sterm_tree_raw) +
  geom_tiplab(size=0.5,align = FALSE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT 
tree_01
   png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png", width = 2200, height = 1500,res=400)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

tree_01


dev.off()

 library(plotly)
ggp <- ggplotly(tree_01)
 ggp

# 
# 415=Oneococcus
# 416=Leuconostoc
# 429=Weissela
# #154=Furfurilactobacillus #is a tip
# 443=Limoslactibacillus
# 439=Paucilactobacillus
# 454=Lactiplantibacillus
# 467=Lentilactobacillus
# 462=Fructilactobacillus
# 461=Apilactobacillus
# 476=Secundilactobacillus
# 482=Levilactobacillus
# 402=Pediococcous
# 487=Liquorilactobacillus
# 
# 348=Lactobacillus
# 
# 376=Companilactobacillus
# 375=Bombilactobacillus
# 385=Lapidilactobacillus
# 343=Schleiferilactobacillus
# #177=Agrilactobacillus #tip
# 388=Lacticaseibacillus
# 397=Latilactobacillus
# #223=Paralactobacillus #tip
# 338=Loigolactobacillus
# 519=mix
# 506=Carnobacterium
# 
# 285=Streptococcus
# 
# 282=Lactococcus

##enterococcus is polyphyletic


 
tree_02  <- scaleClade(tree_01, 285, 2) %>%  scaleClade(., 348, 2)%>% collapse(416, 'max', fill="darkgreen")  %>% collapse(415, 'max', fill="darkgreen")   %>% collapse(429, 'max', fill="darkgreen") %>% collapse(443, 'max', fill="darkgreen")  %>% collapse(439, 'max', fill="darkgreen")  %>% collapse(454, 'max', fill="darkgreen")  %>% collapse(467, 'max', fill="darkgreen")   %>% collapse(462, 'max', fill="darkgreen")   %>% collapse(461, 'max', fill="darkgreen")   %>% collapse(476, 'max', fill="darkgreen")   %>% collapse(482, 'max', fill="darkgreen")   %>% collapse(402, 'max', fill="darkgreen")  %>% collapse(487, 'max', fill="darkgreen")  %>% 
collapse(376, 'max', fill="darkgreen")    %>% collapse(375, 'max', fill="darkgreen")  %>% collapse(385, 'max', fill="darkgreen")  %>% collapse(343, 'max', fill="darkgreen")  %>% collapse(388, 'max', fill="darkgreen")  %>% collapse(397, 'max', fill="darkgreen")  %>% collapse(338, 'max', fill="darkgreen")  %>% collapse(519, 'max', fill="darkgreen")  %>% collapse(506, 'max', fill="darkgreen")  %>% collapse(282, 'max', fill="darkgreen") 

tree_02

 library(plotly)
ggp <- ggplotly(tree_02)
 ggp


   png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png", width = 2200, height = 1500,res=400)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

tree_02


dev.off()


##-------------
#change tiplabel and color according to rmk or non-rmk sample
##-------------

tmp = as.character(Sterm_tree_raw$strain)
# names(tmp) <- all_sterm_strain_info$shortName

# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated", tree.names = NULL, force.multi = FALSE)
Sterm_tree_raw$tip.label
# Sterm_tree_raw$label



Sterm_tree_raw_tible <- as_tibble(Sterm_tree_raw) 
Sterm_tree_raw_tible$label
Sterm_tree_raw_tible$label <- gsub("_.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Streptococcus-","S. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Lactobacillus-","L. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Enterococcus-","E. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Tetragenococcus-","T. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Holzapfelia-","H. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Amylolactobacillus-","A. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("L. delbrueckii$","L. deblrueckii-delbrueckii",Sterm_tree_raw_tible$label)


tree_02_tmp <- Sterm_tree_raw_tible %>% as.treedata()

# Sterm_tree_raw_tible$label <- plyr::revalue(Sterm_tree_raw_tible$label,tmp)
tree_02 <- Sterm_tree_raw_tible%>% as.treedata()


##----------------------------------
#after branch collaps
##----------------------------------

tree_03 <- ggtree(tree_02, layout="circular") +
  geom_tiplab(size=3.5,align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  
   tree_04 <-  scaleClade(tree_03, 285, 2) %>%  scaleClade(., 348, 2)%>% collapse(416, 'max', fill="darkgreen")  %>% collapse(415, 'max', fill="darkgreen")   %>% collapse(429, 'max', fill="darkgreen") %>% collapse(443, 'max', fill="darkgreen")  %>% collapse(439, 'max', fill="darkgreen")  %>% collapse(454, 'max', fill="darkgreen")  %>% collapse(467, 'max', fill="darkgreen")   %>% collapse(462, 'max', fill="darkgreen")   %>% collapse(461, 'max', fill="darkgreen")   %>% collapse(476, 'max', fill="darkgreen")   %>% collapse(482, 'max', fill="darkgreen")   %>% collapse(402, 'max', fill="darkgreen")  %>% collapse(487, 'max', fill="darkgreen")  %>% 
collapse(376, 'max', fill="darkgreen")    %>% collapse(375, 'max', fill="darkgreen")  %>% collapse(385, 'max', fill="darkgreen")  %>% collapse(343, 'max', fill="darkgreen")  %>% collapse(388, 'max', fill="darkgreen")  %>% collapse(397, 'max', fill="darkgreen")  %>% collapse(338, 'max', fill="darkgreen")  %>% collapse(519, 'max', fill="darkgreen")  %>% collapse(506, 'max', fill="darkgreen")  %>% collapse(282, 'max', fill="darkgreen") 

tree_04
###-------------------------
#ADD GENOME COUNTS
###-------------------------
 
counts_streps_lactos_ <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/genome_counts_lactos_streptos.txt",   delim = "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("genome","count"))

counts_streps_lactos_$genome <- gsub("Streptococcus ","S. ",counts_streps_lactos_$genome)
counts_streps_lactos_$genome <- gsub("Lactobacillus ","L. ",counts_streps_lactos_$genome)
counts_streps_lactos_$genome <- gsub(" subsp. ","-",counts_streps_lactos_$genome)

Sterm_tree_raw_tible$label <- gsub("L. delbrueckii$","L. delbrueckii-delbrueckii",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("L. deblrueckii-delbrueckii","L. delbrueckii-delbrueckii",Sterm_tree_raw_tible$label)


all_sterm_strain_info_ready_01 <- counts_streps_lactos_  %>% remove_rownames()%>% column_to_rownames(., var = "genome")

# counts_streps_lactos_$genome %in% Sterm_tree_raw_tible$label
# Sterm_tree_raw_tible$label %in% counts_streps_lactos_$genome


p3 <- gheatmap(tree_04, all_sterm_strain_info_ready_01, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
p3
# "L. deblrueckii-delbrueckii"
###-------------------------
#ADD LAC operon
###-------------------------

lactose_streps_lactos <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/lactose_streps_lactos.txt",   delim = "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("genome","LAC"))
lactose_streps_lactos$genome <- gsub("_.*","",lactose_streps_lactos$genome)
lactose_streps_lactos$genome <- gsub("Streptococcus-","S. ",lactose_streps_lactos$genome)
lactose_streps_lactos$genome <- gsub("Lactobacillus-","L. ",lactose_streps_lactos$genome)

# lactose_streps_lactos$shortName <- as.character(lactose_streps_lactos$shortName)
all_sterm_strain_info_ready <- lactose_streps_lactos  %>% remove_rownames()%>% column_to_rownames(., var = "genome")
# table(all_sterm_strain_info_ready$location_continent)
 # all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c("Europe",  "Asia", "USA", "Australia" , "Unknown","Outgroup"))
# colors_area <- c("#464D77","#36827F","#F9DB6D","grey88")
# colors_area <- c("#F9DB6D","#36827F","#464D77","grey88")

# location_dataset$area

# tree_02 <- full_join(as_tibble(tree), location_dataset, by='label') %>% as.treedata()

# colnames(all_sterm_strain_info_ready) <- c("A","B")
# rownames(all_sterm_strain_info_ready)

p3 <- gheatmap(tree_04, all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
p3

###-------------------------
#clade label
###-------------------------


p4 <- p3+geom_cladelab(align=TRUE,node=415, label="Oneococcus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=416, label="Leuconostoc",geom="text",angle='auto',offset=-0.2,offset=-0.2)+geom_cladelab(align=TRUE,node=429, label="Weissela",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=443, label="Limoslactibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=439, label="Paucilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=454, label="Lactiplantibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=467, label="Lentilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=461, label="Apilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=476, label="Secundilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=482, label="Levilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=402, label="Pediococcous",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=487, label="Liquorilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=376, label="Companilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=375, label="Bombilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=385, label="Lapidilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=343, label="Schleiferilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=388, label="Lacticaseibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=397, label="Latilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=338, label="Loigolactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=519, label="mix",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=506, label="Carnobacterium",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=282, label="Lactococcus",geom="text",angle='auto',offset=-0.2)
  
p4

 png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_03.png", width = 3000, height = 3000,res=400)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

p4


dev.off()

# +geom_cladelab(align=TRUE,node=462, label="Fructilactobacillus",geom="text",angle='auto',offset=-0.2)

##-------------------------------------------------
#only tree
##-------------------------------------------------


p4 <- tree_04+geom_cladelab(align=TRUE,node=415, label="Oneococcus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=416, label="Leuconostoc",geom="text",angle='auto',offset=-0.2,offset=-0.2)+geom_cladelab(align=TRUE,node=429, label="Weissela",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=443, label="Limoslactibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=439, label="Paucilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=454, label="Lactiplantibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=467, label="Lentilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=461, label="Apilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=476, label="Secundilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=482, label="Levilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=402, label="Pediococcous",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=487, label="Liquorilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=376, label="Companilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=375, label="Bombilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=385, label="Lapidilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=343, label="Schleiferilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=388, label="Lacticaseibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=397, label="Latilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=338, label="Loigolactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=519, label="mix",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=506, label="Carnobacterium",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=282, label="Lactococcus",geom="text",angle='auto',offset=-0.2)
  
p4

# png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_03.png", width = 3000, height = 3000,res=400)
 pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total_onlyTREE.pdf",width=10,height=10)

p4


dev.off()


##-------------------------------------------------
#add GH
##-------------------------------------------------


hmm_analysis_median <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/anlayis/HMM_GH/hmm_analysis_prepped.txt") %>% mutate(genome=gsub("Lactobacillus_","L. ",species)) %>% mutate(genome=gsub("Streptococcus","S. ",genome))


hmm_analysis_median
hmm_analysis_median

lactose_streps_lactos <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/lactose_streps_lactos.txt",   delim = "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("genome","LAC"))
lactose_streps_lactos$genome <- gsub("_.*","",lactose_streps_lactos$genome)
lactose_streps_lactos$genome <- gsub("Streptococcus-","S. ",lactose_streps_lactos$genome)
lactose_streps_lactos$genome <- gsub("Lactobacillus-","L. ",lactose_streps_lactos$genome)

# lactose_streps_lactos$shortName <- as.character(lactose_streps_lactos$shortName)
all_sterm_strain_info_ready <- lactose_streps_lactos  %>% remove_rownames()%>% column_to_rownames(., var = "genome")
# table(all_sterm_strain_info_ready$location_continent)
 # all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c("Europe",  "Asia", "USA", "Australia" , "Unknown","Outgroup"))
# colors_area <- c("#464D77","#36827F","#F9DB6D","grey88")
# colors_area <- c("#F9DB6D","#36827F","#464D77","grey88")

# location_dataset$area

# tree_02 <- full_join(as_tibble(tree), location_dataset, by='label') %>% as.treedata()

# colnames(all_sterm_strain_info_ready) <- c("A","B")
# rownames(all_sterm_strain_info_ready)

p3 <- gheatmap(tree_04, all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
p3

```


```{r}
##==========================================================================================
#again 
#after adding S.vestibularis
##==========================================================================================


library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02//RAxML_bipartitions.lactos_curated", tree.names = NULL, force.multi = FALSE)

# Sterm_tree_raw <-    ape::drop.tip(Sterm_tree_raw,265) #%>%  ape::drop.tip(.,142) %>%  ape::drop.tip(.,415)

# x <- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
tree_01 <- ggtree(Sterm_tree_raw, layout="circular") +
  geom_tiplab(aes(angle=angle),align = TRUE,offset = -0.2) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  tree_01
tree_01 <- ggtree(Sterm_tree_raw) +
  geom_tiplab(size=0.5,align = FALSE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT
tree_01
# tmp <- tree_01$data
#    tree_01$data[tree_01$data$node %in% 415, "x"] <- 1.8
#    tree_01$data[tree_01$data$node %in% 415, "branch.length"] <- 0.1
# 

   
   # Sterm_tree_raw$data[Sterm_tree_raw$data$node %in% c(141,142), "x"] <- mean(Sterm_tree_raw$data$x)

   png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png", width = 2200, height = 1500,res=400)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)


   
tree_01


dev.off()

 library(plotly)
ggp <- ggplotly(tree_01)
 ggp

# 
# 415=Oneococcus
# 416=Leuconostoc
# 429=Weissela
# #154=Furfurilactobacillus #is a tip
# 443=Limoslactibacillus
# 439=Paucilactobacillus
# 454=Lactiplantibacillus
# 467=Lentilactobacillus
# 462=Fructilactobacillus
# 461=Apilactobacillus
# 476=Secundilactobacillus
# 482=Levilactobacillus
# 402=Pediococcous
# 487=Liquorilactobacillus
# 
# 348=Lactobacillus
# 
# 376=Companilactobacillus
# 375=Bombilactobacillus
# 385=Lapidilactobacillus
# 343=Schleiferilactobacillus
# #177=Agrilactobacillus #tip
# 388=Lacticaseibacillus
# 397=Latilactobacillus
# #223=Paralactobacillus #tip
# 338=Loigolactobacillus
# 519=mix
# 506=Carnobacterium
# 
# 285=Streptococcus
# 
# 282=Lactococcus

##enterococcus is polyphyletic


 
tree_02  <- scaleClade(tree_01, 285, 2) %>%  scaleClade(., 348, 2)%>% collapse(416, 'max', fill="darkgreen")  %>% collapse(415, 'max', fill="darkgreen")   %>% collapse(429, 'max', fill="darkgreen") %>% collapse(443, 'max', fill="darkgreen")  %>% collapse(439, 'max', fill="darkgreen")  %>% collapse(454, 'max', fill="darkgreen")  %>% collapse(467, 'max', fill="darkgreen")   %>% collapse(462, 'max', fill="darkgreen")   %>% collapse(461, 'max', fill="darkgreen")   %>% collapse(476, 'max', fill="darkgreen")   %>% collapse(482, 'max', fill="darkgreen")   %>% collapse(402, 'max', fill="darkgreen")  %>% collapse(487, 'max', fill="darkgreen")  %>% 
collapse(376, 'max', fill="darkgreen")    %>% collapse(375, 'max', fill="darkgreen")  %>% collapse(385, 'max', fill="darkgreen")  %>% collapse(343, 'max', fill="darkgreen")  %>% collapse(388, 'max', fill="darkgreen")  %>% collapse(397, 'max', fill="darkgreen")  %>% collapse(338, 'max', fill="darkgreen")  %>% collapse(519, 'max', fill="darkgreen")  %>% collapse(506, 'max', fill="darkgreen")  %>% collapse(282, 'max', fill="darkgreen") 

tree_02

 library(plotly)
ggp <- ggplotly(tree_02)
 ggp


   png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png", width = 2200, height = 1500,res=400)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

tree_02


dev.off()


##-------------
#change tiplabel and color according to rmk or non-rmk sample
##-------------

tmp = as.character(Sterm_tree_raw$strain)
# names(tmp) <- all_sterm_strain_info$shortName

# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated", tree.names = NULL, force.multi = FALSE)
Sterm_tree_raw$tip.label
# Sterm_tree_raw$label



Sterm_tree_raw_tible <- as_tibble(Sterm_tree_raw) 
Sterm_tree_raw_tible$label
Sterm_tree_raw_tible$label <- gsub("_.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Streptococcus-","S. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Lactobacillus-","L. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Enterococcus-","E. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Tetragenococcus-","T. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Holzapfelia-","H. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Amylolactobacillus-","A. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("L. delbrueckii$","L. deblrueckii-delbrueckii",Sterm_tree_raw_tible$label)


tree_02_tmp <- Sterm_tree_raw_tible %>% as.treedata()

# Sterm_tree_raw_tible$label <- plyr::revalue(Sterm_tree_raw_tible$label,tmp)
# tree_02 <- Sterm_tree_raw_tible%>% as.treedata() %>% ape::drop.tip(.,265)

# Sterm_tree_raw <-    ape::drop.tip(Sterm_tree_raw,265) #%>%  ape::drop.tip(.,142) %>%  ape::drop.tip(.,415)

##----------------------------------
#after branch collaps
##----------------------------------

tree_03 <- ggtree(tree_02, layout="circular") +
  # geom_tiplab(size=3.5,align = TRUE,offset = -0.16) + # tiplabls aligned
    geom_tiplab(size=3.5,align = TRUE,offset = -0.22) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  
   tree_04 <-  tree_03 %>% scaleClade(., 348, 3)%>% scaleClade(., 285, 3)  %>%  scaleClade(., 416, 0.2) %>% scaleClade(., 415, 0.2) %>% scaleClade(., 429, 0.2) %>% scaleClade(., 443, 0.2) %>% scaleClade(., 439, 0.2) %>% scaleClade(., 454, 0.2) %>% scaleClade(., 467, 0.2) %>% scaleClade(., 462, 0.2) %>% scaleClade(., 461, 0.2) %>% scaleClade(., 476, 0.2) %>% scaleClade(., 482, 0.2) %>% scaleClade(., 402, 0.2) %>% scaleClade(., 487, 0.2) %>% scaleClade(., 376, 0.2) %>% scaleClade(., 375, 0.2) %>% 
   scaleClade(., 385, 0.2) %>% scaleClade(., 397, 0.2) %>% scaleClade(., 338, 0.2) %>% scaleClade(., 519, 0.2) %>% scaleClade(., 506, 0.2) %>% scaleClade(., 282, 0.2) %>% 
    collapse(416, 'max', fill="darkgreen")  %>% collapse(415, 'max', fill="darkgreen")   %>% collapse(429, 'max', fill="darkgreen") %>% collapse(443, 'max', fill="darkgreen")  %>% collapse(439, 'max', fill="darkgreen")  %>% collapse(454, 'max', fill="darkgreen")  %>% collapse(467, 'max', fill="darkgreen")   %>% collapse(462, 'max', fill="darkgreen")   %>% collapse(461, 'max', fill="darkgreen")   %>% collapse(476, 'max', fill="darkgreen")   %>% collapse(482, 'max', fill="darkgreen")   %>% collapse(402, 'max', fill="darkgreen")  %>% collapse(487, 'max', fill="darkgreen")  %>% 
collapse(376, 'max', fill="darkgreen")    %>% collapse(375, 'max', fill="darkgreen")  %>% collapse(385, 'max', fill="darkgreen")  %>% collapse(343, 'max', fill="darkgreen")  %>% collapse(388, 'max', fill="darkgreen")  %>% collapse(397, 'max', fill="darkgreen")  %>% collapse(338, 'max', fill="darkgreen")  %>% collapse(519, 'max', fill="darkgreen")  %>% collapse(506, 'max', fill="darkgreen")  %>% collapse(282, 'max', fill="darkgreen")
    
   
tree_04+ geom_rootedge(265)

   tree_04$data[tree_04$data$node %in% 265, "x"] <- 0.5


# + geom_rootedge(5)
#  scaleClade(., 348, 4) %>%  scaleClade(., 290, 1) %>%


   tree_04 <-  scaleClade(tree_03, 285, 4) %>%  scaleClade(., 348, 4)%>% collapse(416, 'max', fill="darkgreen")  %>% collapse(415, 'max', fill="darkgreen")   %>% collapse(429, 'max', fill="darkgreen") %>% collapse(443, 'max', fill="darkgreen")  %>% collapse(439, 'max', fill="darkgreen")  %>% collapse(454, 'max', fill="darkgreen")  %>% collapse(467, 'max', fill="darkgreen")   %>% collapse(462, 'max', fill="darkgreen")   %>% collapse(461, 'max', fill="darkgreen")   %>% collapse(476, 'max', fill="darkgreen")   %>% collapse(482, 'max', fill="darkgreen")   %>% collapse(402, 'max', fill="darkgreen")  %>% collapse(487, 'max', fill="darkgreen")  %>% 
collapse(376, 'max', fill="darkgreen")    %>% collapse(375, 'max', fill="darkgreen")  %>% collapse(385, 'max', fill="darkgreen")  %>% collapse(343, 'max', fill="darkgreen")  %>% collapse(388, 'max', fill="darkgreen")  %>% collapse(397, 'max', fill="darkgreen")  %>% collapse(338, 'max', fill="darkgreen")  %>% collapse(519, 'max', fill="darkgreen")  %>% collapse(506, 'max', fill="darkgreen")  %>% collapse(282, 'max', fill="darkgreen") 
tree_04

# 415
# groupClade(tree_03, 415)

# tmp <- tree_03$data

# tree_03$data[tree_03$data$node %in% c(141,142), "x"] <- mean(tree_03$data$x)
# tree_03$data[tree_03$data$node %in% c(141,142), "y"] <- mean(tree_03$data$y)

# tree_03$data[tree_03$data$node %in% c(141,142), "branch"] <- 0.5
# tree_03$data[tree_03$data$node %in% 415, "branch"] <- 0.5
# tree_03$data[tree_03$data$node %in% 415, "x"] <- 22.427440



# ggp <- ggplotly(tree_03)
#  ggp
# + geom_hilight(285, fill="darkgreen")
###-------------------------
#ADD GENOME COUNTS
###-------------------------
 
counts_streps_lactos_ <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/genome_counts_lactos_streptos.txt",   delim = "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("genome","count"))

counts_streps_lactos_$genome <- gsub("Streptococcus ","S. ",counts_streps_lactos_$genome)
counts_streps_lactos_$genome <- gsub("Lactobacillus ","L. ",counts_streps_lactos_$genome)
counts_streps_lactos_$genome <- gsub(" subsp. ","-",counts_streps_lactos_$genome)

Sterm_tree_raw_tible$label <- gsub("L. delbrueckii$","L. delbrueckii-delbrueckii",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("L. deblrueckii-delbrueckii","L. delbrueckii-delbrueckii",Sterm_tree_raw_tible$label)


all_sterm_strain_info_ready_01 <- counts_streps_lactos_  %>% remove_rownames()%>% column_to_rownames(., var = "genome")

# counts_streps_lactos_$genome %in% Sterm_tree_raw_tible$label
# Sterm_tree_raw_tible$label %in% counts_streps_lactos_$genome


p3 <- gheatmap(tree_04, all_sterm_strain_info_ready_01, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
p3
# "L. deblrueckii-delbrueckii"
###-------------------------
#ADD LAC operon
###-------------------------

lactose_streps_lactos <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/lactose_streps_lactos.txt",   delim = "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("genome","LAC"))
lactose_streps_lactos$genome <- gsub("_.*","",lactose_streps_lactos$genome)
lactose_streps_lactos$genome <- gsub("Streptococcus-","S. ",lactose_streps_lactos$genome)
lactose_streps_lactos$genome <- gsub("Lactobacillus-","L. ",lactose_streps_lactos$genome)

# lactose_streps_lactos$shortName <- as.character(lactose_streps_lactos$shortName)
all_sterm_strain_info_ready <- lactose_streps_lactos  %>% remove_rownames()%>% column_to_rownames(., var = "genome")
# table(all_sterm_strain_info_ready$location_continent)
 # all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c("Europe",  "Asia", "USA", "Australia" , "Unknown","Outgroup"))
# colors_area <- c("#464D77","#36827F","#F9DB6D","grey88")
# colors_area <- c("#F9DB6D","#36827F","#464D77","grey88")

# location_dataset$area

# tree_02 <- full_join(as_tibble(tree), location_dataset, by='label') %>% as.treedata()

# colnames(all_sterm_strain_info_ready) <- c("A","B")
# rownames(all_sterm_strain_info_ready)

p3 <- gheatmap(tree_04, all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
p3

###-------------------------
#clade label
###-------------------------


p4 <- p3+geom_cladelab(align=TRUE,node=415, label="Oneococcus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=416, label="Leuconostoc",geom="text",angle='auto',offset=-0.2,offset=-0.2)+geom_cladelab(align=TRUE,node=429, label="Weissela",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=443, label="Limoslactibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=439, label="Paucilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=454, label="Lactiplantibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=467, label="Lentilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=461, label="Apilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=476, label="Secundilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=482, label="Levilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=402, label="Pediococcous",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=487, label="Liquorilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=376, label="Companilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=375, label="Bombilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=385, label="Lapidilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=343, label="Schleiferilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=388, label="Lacticaseibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=397, label="Latilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=338, label="Loigolactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=519, label="mix",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=506, label="Carnobacterium",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=282, label="Lactococcus",geom="text",angle='auto',offset=-0.2)
  

p4
#  png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_03.png", width = 3000, height = 3000,res=400)
# # pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)
# 
# p4
# 
# 
# dev.off()

# +geom_cladelab(align=TRUE,node=462, label="Fructilactobacillus",geom="text",angle='auto',offset=-0.2)
###-------------------------
#clade label
###-------------------------


p5 <-tree_04+geom_cladelab(align=TRUE,node=415, label="Oneococcus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=416, label="Leuconostoc",geom="text",angle='auto',offset=-0.2,offset=-0.2)+geom_cladelab(align=TRUE,node=429, label="Weissela",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=443, label="Limoslactibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=439, label="Paucilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=454, label="Lactiplantibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=467, label="Lentilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=461, label="Apilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=476, label="Secundilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=482, label="Levilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=402, label="Pediococcous",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=487, label="Liquorilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=376, label="Companilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=375, label="Bombilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=385, label="Lapidilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=343, label="Schleiferilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=388, label="Lacticaseibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=397, label="Latilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=338, label="Loigolactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=519, label="mix",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=506, label="Carnobacterium",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=282, label="Lactococcus",geom="text",angle='auto',offset=-0.2)
  
p5


###-------------------------
#zoom into strepto and lacto taxons
###-------------------------

p6 <- p5 + geom_hilight(285, fill="darkgreen")
p6



p6 <- p5 %>% scaleClade(285, scale=0.3)
p6

# 285=Streptococcus
# # 348=Lactobacillus

##enlarge center
# + geom_rootedge(5)



```

function needed in next chunkg
```{r}

offspring.tbl_tree_item <- function(.data, .node, ...) {
    valid.tbl_tree(.data)

    if (is.character(.node)) {
        .node <- .data$node[.data$label == .node]
    }

    .data[.data$parent == .node & .data$parent != .data$node,]
}

##' @method offspring tbl_tree
##' @export
##' @rdname offspring
##' @examples
##' library(ape)
##' tree <- rtree(4)
##' x <- as_tibble(tree)
##' offspring(x, 4)
offspring.tbl_tree <- function(.data, .node, tiponly = FALSE, self_include = FALSE, ...) {
    if (missing(.node) || is.null(.node)) {
        stop(".node is required")
    }
    if (length(.node) == 1) {
        res <- .offspring.tbl_tree_item(.data = .data, .node = .node,
                                       tiponly = tiponly, self_include = self_include, ...)
    } else {
        res <- lapply(.node, function(node) {
            .offspring.tbl_tree_item(.data = .data, .node = node,
                                    tiponly = tiponly, self_include = self_include, ...)
        })
        names(res) <- .node
    }
    return(res)
}

#' @noRd
#' @keywords internal
.offspring.tbl_tree_item <- function(.data, .node, tiponly = FALSE, self_include = FALSE, ...) {
    x <- child.tbl_tree(.data, .node)

    ## https://github.com/GuangchuangYu/ggtree/issues/239
    rn <- rootnode.tbl_tree(.data)$node
    x <- x[x$node != rn, ]

    if (nrow(x) == 0) {
        if (self_include) {
            x <- .data[.data$node == .node, ]
        } 

        return(x)
    }

    ## id <- x$node
    ## i <- 1
    ## while(i <= length(id)) {
    ##     id <- c(id, child(.data, id[i])$node)
    ##     i <- i + 1
    ## }
    ## filter_(.data, ~ node %in% id)

    parent <- .data$parent
    children <- .data$node
    ## n <- length(parent)
    n <- max(parent)

    kids <- vector("list", n)
    for (i in seq_along(parent)) {
        kids[[parent[i]]] <-c(kids[[parent[i]]], children[i])
    }

    id <- x$node
    i <- 1
    while(i <= length(id)) {
        id <- c(id, kids[[id[i]]])
        i <- i + 1
    }

    if (self_include) {
        id <- c(.node, id)
    }

    sp <- .data[children %in% id,]
    if (tiponly) {
        return(sp[sp$node < rn,])
    }
    return(sp)
}

##' @method child phylo
##' @export
child.phylo <- function(.data, .node, type = 'children', ...) {
    res <- offspring(.data=.data, .node = .node, type = type)
    return(res)
}

##' @method child treedata
##' @export
child.treedata <- function(.data, .node, type = 'children', ...) {
    child.phylo(as.phylo(.data), .node, type = type, ...)
}

.internal.child <- function(data, node, type = 'children'){
    if (!is_numeric(node)){
        all.labs <- c(data$tip.label, data$node.label)
        names(all.labs) <- seq_len(length(all.labs))
        node <- names(all.labs[all.labs %in% node])
    }
    edge <- data$edge
    res <- edge[edge[,1] == node, 2]
    if (type != 'children'){
        alltips <- edge[,2][! edge[,2] %in% edge[,1]]
        w <- which(res >= length(alltips))
        if(length(w)>0){
            for(i in 1:length(w)){
                res <- c(res,
                         .internal.child(
                           data = data,
                           node = res[w[i]],
                           type = type
                         )
                       )
            }
        }
        if (type %in% c('tips', 'external')){
            res <- res[res %in% alltips]
        }else if (type == "internal") {
            res <- res[!res %in% alltips]
        }
    }
    return(unname(res))
}

##' @method offspring phylo
##' @export
offspring.phylo <- function(.data, .node, tiponly = FALSE, self_include = FALSE, type = 'all', ...){
    type <- match.arg(type, c("children", 'tips', 'internal', 'external', 'all'))

    if (tiponly){
        message('The "tiponly = TRUE" can be replaced by type="tips".')
        type = 'tips'
    }

    res <- lapply(.node, .internal.child, data = .data, type = type)
    if (length(res) <= 1){
        res <- unlist(res)
        if (self_include){
            res <- c(.node, res)
        }
    }else{
        if (self_include){
            res <- mapply(append, .node, res, SIMPLIFY=FALSE)
        }
        names(res) <- .node
    }
    return (res)
    #if (self_include) {
    #    sp <- .node
    #} else {
    #    sp <- child(.data, .node)
    #}

    #sp <- sp[sp != 0]
    #if (length(sp) == 0) {
    #    return(sp)
    #    ## stop("input node is a tip...")
    #}
    #i <- 1
    #while (i <= length(sp)) {
    #    sp <- c(sp, child(.data, sp[i]))
    #    sp <- sp[sp != 0]
    #    i <- i + 1
    #}
    #if (tiponly) {
    #    return(sp[sp <= Ntip(.data)])
    #}
    #return(sp)
}


##' @method offspring treedata
##' @export
offspring.treedata <- function(.data, .node, tiponly = FALSE, self_include = FALSE, type = 'all', ...) {
    offspring.phylo(as.phylo(.data), .node,
                    tiponly = tiponly, self_include = self_include,
                    type = type,
                    ...)
}


valid.tbl_tree <- function(object, cols = c("parent", "node", "label")) {
    cc <- cols[!cols %in% colnames(object)]
    if (length(cc) > 0) {
        msg <- paste0("invalid tbl_tree object.\n  missing column:\n    ", paste(cc, collapse=","), ".")
    }
}
```


### final make tree

remove outgroup


```{r}
##==========================================================================================
#again 
#after adding S.vestibularis
##==========================================================================================


library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02//RAxML_bipartitions.lactos_curated", tree.names = NULL, force.multi = FALSE)

Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03//RAxML_bipartitions.lactos_curated", tree.names = NULL, force.multi = FALSE)

##------------------------------remove root! Fusobacterium-nucleatum_G000158275


Sterm_tree_raw %>% as_tibble() %>% filter(label=="Fusobacterium-nucleatum_G000158275") 
Sterm_tree_raw <-    ape::drop.tip(Sterm_tree_raw,274) #%>%  ape::drop.tip(.,142) %>%  ape::drop.tip(.,415)



# x <- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
tree_01 <- ggtree(Sterm_tree_raw, layout="fan") +
  geom_tiplab(aes(angle=angle),align = TRUE,offset = -0.2) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  tree_01
tree_01 <- ggtree(Sterm_tree_raw) +
  geom_tiplab(size=0.5,align = FALSE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT
tree_01
# tmp <- tree_01$data
#    tree_01$data[tree_01$data$node %in% 415, "x"] <- 1.8
#    tree_01$data[tree_01$data$node %in% 415, "branch.length"] <- 0.1
# 

   
   # Sterm_tree_raw$data[Sterm_tree_raw$data$node %in% c(141,142), "x"] <- mean(Sterm_tree_raw$data$x)

   png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png", width = 2200, height = 1500,res=400)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)


   
tree_01


dev.off()

 library(plotly)
ggp <- ggplotly(tree_01)
 ggp

# 
# 484=Oneococcus
# 486=Leuconostoc
# 474=Weissela
# ----------------------------------------#196=Furfurilactobacillus #is a tip
# 458=Limoslactibacillus
# 468=Paucilactobacillus
# 498=Lactiplantibacillus #!
# 447=Lentilactobacillus
# 444=Fructilactobacillus
# 442=Apilactobacillus
# 474=Secundilactobacillus
# 480=Levilactobacillus
# 400=Pediococcous
# 485=Liquorilactobacillus
# 
# 345=Lactobacillus
# 
# 374=Companilactobacillus
# 373=Bombilactobacillus
# 383=Lapidilactobacillus
# 341=Schleiferilactobacillus
# #177=Agrilactobacillus #tip
# 386=Lacticaseibacillus
# 395=Latilactobacillus
# #223=Paralactobacillus #tip
# 336=Loigolactobacillus
# 517=mix
## 509=Carnobacterium
 # 504=Carnobacterium

# 
# 288=Streptococcus
# 
# 280=Lactococcus

 
 ###---------------------------------find nodes automatically.
 ------------------------------------#196=Furfurilactobacillus #is a tip
#---------------------------=Agrilactobacillus #tip
#------------------------------   =Paralactobacillus #tip
#!!!!!mix
 # 
 # nodes_genus <- data.frame()
 # nodes_genus_new <- list()
for (genusss in c("Oenococcus","Leuconostoc","Weissella","Limosilactobacillus","Paucilactobacillus","Lactiplantibacillus","Lentilactobacillus","Fructilactobacillus","Apilactobacillus","Secundilactobacillus","Levilactobacillus","Pediococcus","Liquorilactobacillus","Lactobacillus","Companilactobacillus","Bombilactobacillus","Lapidilactobacillus","Schleiferilactobacillus","Lacticaseibacillus","Latilactobacillus","Loigolactobacillus","Carnobacterium","Carnobacterium","Streptococcus","Lactococcus","Ligilactobacillus")) {
   print(paste0("node_",genusss))
     tmp <- Sterm_tree_raw$tip.label[grepl(genusss,Sterm_tree_raw$tip.label)]
     tmp1 <- getMRCA(Sterm_tree_raw,tmp)

    # tmp2 <- data.frame(genus=genusss,node=tmp1)   
    # nodes_genus <- rbind(nodes_genus,tmp2)
    
    # nodes_genus_new[genusss] <- tmp1
    
      assign(paste0("node_",genusss), tmp1)

    # return(c(paste0("node_",genusss), tmp1))
    # paste0("node_",genusss) <- tmp1
 }
 
 node_Lactobacillus <- 356
 node_mix=535
  node_Carnobacterium=522

 
 library(tidytree)
Sterm_tree_raw$tip.label[(grepl("Li",Sterm_tree_raw$tip.label))]

##enterococcus is polyphyletic


tree_02  <- scaleClade(tree_01, node_Streptococcus, 2) %>%  scaleClade(., node_Lactobacillus, 2)%>% collapse(node_Oenococcus, 'max', fill="darkgreen")  %>% collapse(node_Leuconostoc, 'max', fill="darkgreen")   %>% collapse(node_Weissella, 'max', fill="darkgreen") %>% collapse(node_Limosilactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Paucilactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Lactiplantibacillus, 'max', fill="darkgreen")  %>% collapse(node_Lentilactobacillus, 'max', fill="darkgreen")   %>% collapse(node_Fructilactobacillus, 'max', fill="darkgreen")   %>% collapse(node_Apilactobacillus, 'max', fill="darkgreen")   %>% collapse(node_Secundilactobacillus, 'max', fill="darkgreen")   %>% collapse(node_Levilactobacillus, 'max', fill="darkgreen")   %>% collapse(node_Pediococcus, 'max', fill="darkgreen")  %>% collapse(node_Liquorilactobacillus, 'max', fill="darkgreen")  %>% 
collapse(node_Companilactobacillus, 'max', fill="darkgreen")    %>% collapse(node_Bombilactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Lapidilactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Schleiferilactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Lacticaseibacillus, 'max', fill="darkgreen")  %>% collapse(node_Latilactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Loigolactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Carnobacterium, 'max', fill="darkgreen")  %>% collapse(node_Lactococcus, 'max', fill="darkgreen")  %>% collapse(node_mix, 'max', fill="darkgreen")  %>% collapse(node_Ligilactobacillus, 'max', fill="darkgreen") 

tree_02

 
# tree_02  <- scaleClade(tree_01, 288, 2) %>%  scaleClade(., 345, 2)%>% collapse(412, 'max', fill="darkgreen")  %>% collapse(415, 'max', fill="darkgreen")   %>% collapse(427, 'max', fill="darkgreen") %>% collapse(441, 'max', fill="darkgreen")  %>% collapse(436, 'max', fill="darkgreen")  %>% collapse(452, 'max', fill="darkgreen")  %>% collapse(465, 'max', fill="darkgreen")   %>% collapse(460, 'max', fill="darkgreen")   %>% collapse(463, 'max', fill="darkgreen")   %>% collapse(474, 'max', fill="darkgreen")   %>% collapse(480, 'max', fill="darkgreen")   %>% collapse(400, 'max', fill="darkgreen")  %>% collapse(485, 'max', fill="darkgreen")  %>% 
# collapse(374, 'max', fill="darkgreen")    %>% collapse(373, 'max', fill="darkgreen")  %>% collapse(383, 'max', fill="darkgreen")  %>% collapse(341, 'max', fill="darkgreen")  %>% collapse(386, 'max', fill="darkgreen")  %>% collapse(395, 'max', fill="darkgreen")  %>% collapse(336, 'max', fill="darkgreen")  %>% collapse(517, 'max', fill="darkgreen")  %>% collapse(504, 'max', fill="darkgreen")  %>% collapse(280, 'max', fill="darkgreen") 


 library(plotly)
ggp <- ggplotly(tree_02)
 ggp


   png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png", width = 2200, height = 1500,res=400)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

tree_02


dev.off()


##-------------
#change tiplabel and color according to rmk or non-rmk sample
##-------------

tmp = as.character(Sterm_tree_raw$strain)
# names(tmp) <- all_sterm_strain_info$shortName

# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated", tree.names = NULL, force.multi = FALSE)
Sterm_tree_raw$tip.label
# Sterm_tree_raw$label



Sterm_tree_raw_tible <- as_tibble(Sterm_tree_raw) 
Sterm_tree_raw_tible$label
Sterm_tree_raw_tible$label <- gsub("_.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Streptococcus-","S. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Lactobacillus-","L. ",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Enterococcus-.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Tetragenococcus-.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Holzapfelia-.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Amylolactobacillus-.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("L. delbrueckii$","L. delbrueckii-delbrueckii",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Amylolactobacillus-.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Meliss.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Vagoc.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Catel.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Paralacto.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Agrilacto.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("L. delbrueckii-","L. del-",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("Furfuri.*","",Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label <- gsub("^'An.*","",Sterm_tree_raw_tible$label)


# Sterm_tree_raw_tible$label <- paste0("italic(",Sterm_tree_raw_tible$label,")")


# tree_05_as_tibble <- tree_03 %>% as_tibble()
# Sterm_tree_raw_tible$label <- ifelse(is.na(Sterm_tree_raw_tible$label),"NA",paste0("italic(",Sterm_tree_raw_tible$label,")"))
# Sterm_tree_raw_tible$label <- gsub("italic()","",Sterm_tree_raw_tible$label)

tree_02_tmp <- Sterm_tree_raw_tible %>% as.treedata()


# tree_02_tmp <- Sterm_tree_raw_tible %>% as.treedata()


tree_02 <- Sterm_tree_raw_tible%>% as.treedata()
# Sterm_tree_raw_tible$label <- plyr::revalue(Sterm_tree_raw_tible$label,tmp)
# tree_02 <- Sterm_tree_raw_tible%>% as.treedata() %>% ape::drop.tip(.,265)

# Sterm_tree_raw <-    ape::drop.tip(Sterm_tree_raw,265) #%>%  ape::drop.tip(.,142) %>%  ape::drop.tip(.,415)

##----------------------------------
#after branch collaps
##----------------------------------



# tree_03 <- ggtree(tree_02, layout="circular",open.angle = 90) +
  tree_03 <- ggtree(tree_02, layout="fan",open.angle = 50) +geom_rootedge(rootedge = 0)+geom_tiplab(size=3.5,align = TRUE,offset = 0.06,linesize = 0,fontface = 'italic') 
            

  tree_03
  # geom_tiplab(size=3.5,align = TRUE,offset = 0.06,linesize = 0,aes(label=paste0('italic(', label, ')')) )




  # geom_tiplab(size=3.5,align = TRUE,offset = -0.16) + # tiplabls aligned
    # geom_tiplab(size=3.5,align = TRUE,offset = -0.06,linesize = 0) + # tiplabls aligned
      # + # tiplabls aligned

   #+open_tree( 270)
tree_03


# tree_04  <- scaleClade(tree_03, 279, 3.5) %>%  scaleClade(., 345,3.5) %>%  scaleClade(., 412, 0.2)  %>%  scaleClade(., 415, 0.2)   %>%  scaleClade(., 427, 0.2) %>%  scaleClade(., 441, 0.2)  %>%  scaleClade(., 436, 0.2)  %>%  scaleClade(., 452, 0.2)  %>%  scaleClade(., 465, 0.2)   %>%  scaleClade(., 460, 0.2)   %>%  scaleClade(., 463, 0.2)   %>%  scaleClade(., 474, 0.2)   %>%  scaleClade(., 480, 0.2)   %>%  scaleClade(., 400, 0.2)  %>%  scaleClade(., 485, 0.2)  %>% 
#  scaleClade(., 374, 0.2)    %>%  scaleClade(., 373, 0.2)  %>%  scaleClade(., 383, 0.2)  %>%  scaleClade(., 341, 0.2)  %>%  scaleClade(., 386, 0.2)  %>%  scaleClade(., 395, 0.2)  %>%  scaleClade(., 336, 0.2)  %>%  scaleClade(., 517, 0.2)  %>%  scaleClade(., 504, 0.2)  %>%  scaleClade(., 280, 0.2) %>% collapse(412, 'max', fill="grey78")  %>% collapse(415, 'max', fill="grey78")   %>% collapse(427, 'max', fill="grey78") %>% collapse(441, 'max', fill="grey78")  %>% collapse(436, 'max', fill="grey78")  %>% collapse(452, 'max', fill="grey78")  %>% collapse(465, 'max', fill="grey78")   %>% collapse(460, 'max', fill="grey78")   %>% collapse(463, 'max', fill="grey78")   %>% collapse(474, 'max', fill="grey78")   %>% collapse(480, 'max', fill="grey78")   %>% collapse(400, 'max', fill="grey78")  %>% collapse(485, 'max', fill="grey78")  %>% 
# collapse(374, 'max', fill="grey78")    %>% collapse(373, 'max', fill="grey78")  %>% collapse(383, 'max', fill="grey78")  %>% collapse(341, 'max', fill="grey78")  %>% collapse(386, 'max', fill="grey78")  %>% collapse(395, 'max', fill="grey78")  %>% collapse(336, 'max', fill="grey78")  %>% collapse(517, 'max', fill="grey78")  %>% collapse(504, 'max', fill="grey78")  %>% collapse(280, 'max', fill="grey78") 



tree_04  <- scaleClade(tree_03, node_Streptococcus, 2.8) %>%  scaleClade(., node_Lactobacillus,2.8) %>%  scaleClade(., node_Oenococcus, 0.2)  %>%  scaleClade(., node_Leuconostoc, 0.2)   %>%  scaleClade(., node_Weissella, 0.2) %>%  scaleClade(., node_Limosilactobacillus, 0.2)  %>%  scaleClade(., node_Paucilactobacillus, 0.2)  %>%  scaleClade(., node_Lactiplantibacillus, 0.2)  %>%  scaleClade(., node_Lentilactobacillus, 0.2)   %>%  scaleClade(., node_Fructilactobacillus, 0.2)   %>%  scaleClade(., 463, 0.2)   %>%  scaleClade(., 474, 0.2)   %>%  scaleClade(., node_Apilactobacillus, 0.2)   %>%  scaleClade(., node_Secundilactobacillus, 0.2)  %>%  scaleClade(., node_Levilactobacillus, 0.2)  %>% 
 scaleClade(., node_Pediococcus, 0.2)    %>%  scaleClade(., node_Liquorilactobacillus, 0.2)  %>%  scaleClade(., node_Companilactobacillus, 0.2)  %>%  scaleClade(., node_Bombilactobacillus, 0.2)  %>%  scaleClade(., node_Lapidilactobacillus, 0.2)  %>%  scaleClade(., node_Schleiferilactobacillus, 0.2)  %>%  scaleClade(., node_Lacticaseibacillus, 0.2)  %>%  scaleClade(., node_Latilactobacillus, 0.2)  %>%  scaleClade(., node_Loigolactobacillus, 0.2)%>%  scaleClade(., node_Carnobacterium, 0.2)  %>%  scaleClade(., node_Lactococcus, 0.2) %>%  scaleClade(., node_mix, 0.2)  %>%  scaleClade(., node_Ligilactobacillus, 0.2) %>%
  collapse(node_Oenococcus, 'max', fill="darkgreen")  %>% collapse(node_Leuconostoc, 'max', fill="darkgreen")   %>% collapse(node_Weissella, 'max', fill="darkgreen") %>% collapse(node_Limosilactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Paucilactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Lactiplantibacillus, 'max', fill="darkgreen")  %>% collapse(node_Lentilactobacillus, 'max', fill="darkgreen")   %>% collapse(node_Fructilactobacillus, 'max', fill="darkgreen")   %>% collapse(node_Apilactobacillus, 'max', fill="darkgreen")   %>% collapse(node_Secundilactobacillus, 'max', fill="darkgreen")   %>% collapse(node_Levilactobacillus, 'max', fill="darkgreen")   %>% collapse(node_Pediococcus, 'max', fill="darkgreen")  %>% collapse(node_Liquorilactobacillus, 'max', fill="darkgreen")  %>% 
collapse(node_Companilactobacillus, 'max', fill="darkgreen")    %>% collapse(node_Bombilactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Lapidilactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Schleiferilactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Lacticaseibacillus, 'max', fill="darkgreen")  %>% collapse(node_Latilactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Loigolactobacillus, 'max', fill="darkgreen")  %>% collapse(node_Carnobacterium, 'max', fill="darkgreen")  %>% collapse(node_Lactococcus, 'max', fill="darkgreen")  %>% collapse(node_mix, 'max', fill="darkgreen")  %>% collapse(node_Ligilactobacillus, 'max', fill="darkgreen") %>% rotate_tree(., 54)
tree_04

# 
# # ggp <- ggplotly(tree_03)
# #  ggp
# # + geom_hilight(285, fill="darkgreen")
# ###-------------------------
# #ADD GENOME COUNTS
# ###-------------------------
#  
# counts_streps_lactos_ <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/genome_counts_lactos_streptos.txt",   delim = "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("genome","count"))
# 
# counts_streps_lactos_$genome <- gsub("Streptococcus ","S. ",counts_streps_lactos_$genome)
# counts_streps_lactos_$genome <- gsub("Lactobacillus ","L. ",counts_streps_lactos_$genome)
# counts_streps_lactos_$genome <- gsub(" subsp. ","-",counts_streps_lactos_$genome)
# 
# Sterm_tree_raw_tible$label <- gsub("L. delbrueckii$","L. delbrueckii-delbrueckii",Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label <- gsub("L. deblrueckii-delbrueckii","L. delbrueckii-delbrueckii",Sterm_tree_raw_tible$label)
# 
# 
# all_sterm_strain_info_ready_01 <- counts_streps_lactos_  %>% remove_rownames()%>% column_to_rownames(., var = "genome")
# 
# # counts_streps_lactos_$genome %in% Sterm_tree_raw_tible$label
# # Sterm_tree_raw_tible$label %in% counts_streps_lactos_$genome
# 
# 
# p3 <- gheatmap(tree_04, all_sterm_strain_info_ready_01, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
# p3
# "L. deblrueckii-delbrueckii"
###-------------------------
#ADD LAC operon
###-------------------------
# 
# lactose_streps_lactos <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/lactose_streps_lactos.txt",   delim = "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("genome","LAC"))
# lactose_streps_lactos$genome <- gsub("_.*","",lactose_streps_lactos$genome)
# lactose_streps_lactos$genome <- gsub("Streptococcus-","S. ",lactose_streps_lactos$genome)
# lactose_streps_lactos$genome <- gsub("Lactobacillus-","L. ",lactose_streps_lactos$genome)
# 
# # lactose_streps_lactos$shortName <- as.character(lactose_streps_lactos$shortName)
# all_sterm_strain_info_ready <- lactose_streps_lactos  %>% remove_rownames()%>% column_to_rownames(., var = "genome")
# # table(all_sterm_strain_info_ready$location_continent)
#  # all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c("Europe",  "Asia", "USA", "Australia" , "Unknown","Outgroup"))
# # colors_area <- c("#464D77","#36827F","#F9DB6D","grey88")
# # colors_area <- c("#F9DB6D","#36827F","#464D77","grey88")
# 
# # location_dataset$area
# 
# # tree_02 <- full_join(as_tibble(tree), location_dataset, by='label') %>% as.treedata()
# 
# # colnames(all_sterm_strain_info_ready) <- c("A","B")
# # rownames(all_sterm_strain_info_ready)
# 
# p3 <- gheatmap(tree_04, all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
# p3

###-------------------------
#clade label
###-------------------------

# 
# p4 <- p3+geom_cladelab(align=TRUE,node=415, label="Oneococcus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=416, label="Leuconostoc",geom="text",angle='auto',offset=-0.2,offset=-0.2)+geom_cladelab(align=TRUE,node=429, label="Weissela",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=443, label="Limoslactibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=439, label="Paucilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=454, label="Lactiplantibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=467, label="Lentilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=461, label="Apilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=476, label="Secundilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=482, label="Levilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=402, label="Pediococcous",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=487, label="Liquorilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=376, label="Companilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=375, label="Bombilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=385, label="Lapidilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=343, label="Schleiferilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=388, label="Lacticaseibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=397, label="Latilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=338, label="Loigolactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=519, label="mix",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=506, label="Carnobacterium",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=282, label="Lactococcus",geom="text",angle='auto',offset=-0.2)
#   
# 
# p4
#  png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_03.png", width = 3000, height = 3000,res=400)
# # pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)
# 
# p4
# 
# 
# dev.off()

# +geom_cladelab(align=TRUE,node=462, label="Fructilactobacillus",geom="text",angle='auto',offset=-0.2)
###-------------------------
#clade label
###-------------------------
# 
# 
# p5 <-tree_04+geom_cladelab(align=TRUE,node=415, label="Oneococcus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=416, label="Leuconostoc",geom="text",angle='auto',offset=-0.2,offset=-0.2)+geom_cladelab(align=TRUE,node=429, label="Weissela",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=443, label="Limoslactibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=439, label="Paucilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=454, label="Lactiplantibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=467, label="Lentilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=461, label="Apilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=476, label="Secundilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=482, label="Levilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=402, label="Pediococcous",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=487, label="Liquorilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=376, label="Companilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=375, label="Bombilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=385, label="Lapidilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=343, label="Schleiferilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=388, label="Lacticaseibacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=397, label="Latilactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=338, label="Loigolactobacillus",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=519, label="mix",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=506, label="Carnobacterium",geom="text",angle='auto',offset=-0.2)+geom_cladelab(align=TRUE,node=282, label="Lactococcus",geom="text",angle='auto',offset=-0.2)
#   
# p5


###-------------------------
#add EMA data
###-------------------------

Species_distribution <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/Earthmicrobiome_data/Final/Final/VinceDB_Species_distribution.csv")

Species_distribution <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/Earthmicrobiome_data/Final_03/Final/FinalVinceDB_Species_distribution.csv") %>% mutate(Species_name=qseqid)


table(Species_distribution$Species_name)

unique(Species_distribution$Species_name)
unique(Sterm_tree_raw_tible$label)


Species_distribution$Species_name <- gsub("Streptococcus-","S. ",Species_distribution$Species_name)
Species_distribution$Species_name <- gsub("Lactobacillus-","L. ",Species_distribution$Species_name)
Species_distribution$Species_name <- gsub("Lactobacillus-","L. ",Species_distribution$Species_name)
Species_distribution$Species_name <- gsub("L. delbrueckii-","L. del-",Species_distribution$Species_name)
Species_distribution$Species_name <- gsub("L. -bulgaricus","L. del-bulgaricus",Species_distribution$Species_name)

unique(Species_distribution$Species_name)[!unique(Species_distribution$Species_name) %in% unique(Sterm_tree_raw_tible$label)]


unique(Species_distribution$Habitat)

Species_distribution$Habitat_broad <- Species_distribution$Habitat %>% gsub("Animal.*","Animal sampling",.)%>% gsub("Plant.*","Plant sampling",.)%>% gsub("Sediment.*","Soil sampling",.)%>% gsub("Soil.*","Soil sampling",.) %>% gsub("Surface.*","Environemental sampling",.)%>% gsub("Water.*","Water sampling",.)

table(Species_distribution$Habitat_broad)

Species_distribution_sums <- Species_distribution %>% group_by(Species_name) %>% summarise(total=sum(Standardized_MeanCount_per_sample))

Species_distribution_preped <- Species_distribution %>% select(Species_name,Habitat_broad,Standardized_MeanCount_per_sample) 

Species_distribution_preped <- Species_distribution_preped %>% group_by(Species_name,Habitat_broad) %>% summarise(Standardized_MeanCount_per_sample=sum(Standardized_MeanCount_per_sample)) %>% filter(Standardized_MeanCount_per_sample>0)

 tmp <- Species_distribution_preped %>% group_by(Species_name) %>% summarise(total=sum(Standardized_MeanCount_per_sample))

 
 table(Species_distribution_preped$Species_name)
 
 
 write.csv(Species_distribution_preped, "~/Desktop/Projects/2020_StarterCultureDiversity/Earthmicrobiome_data/Final/Final/VinceDB_Species_distribution_prepeed.csv")
 
# is.na(Species_distribution_preped$Habitat_broad) %>%
# counts_streps_lactos_ <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/genome_counts_lactos_streptos.txt",   delim = "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("genome","count"))
# 
# counts_streps_lactos_$genome <- gsub("Streptococcus ","S. ",counts_streps_lactos_$genome)
# counts_streps_lactos_$genome <- gsub("Lactobacillus ","L. ",counts_streps_lactos_$genome)
# counts_streps_lactos_$genome <- gsub(" subsp. ","-",counts_streps_lactos_$genome)
# 
# all_sterm_strain_info_ready_01 <- counts_streps_lactos_  %>% remove_rownames()%>% column_to_rownames(., var = "genome")
# 
# p3 <- gheatmap(tree_04, all_sterm_strain_info_ready_01, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
# p3
  
table(Species_distribution_preped$Species_name)

library(ggstance)
# library(ggtree)
library(ggtreeExtra)
tree_05 <- tree_04+ geom_fruit(data=Species_distribution_preped, geom=geom_bar,
                    mapping=aes(y=Species_name, x=Standardized_MeanCount_per_sample,fill=Habitat_broad),
                    pwidth=0.09, position=position_stackx(),
                    # width=1,
                    offset=-0.07,
                    orientation="y", 
                    stat="identity",
         )+theme(legend.position=c(0.1, 0.2),legend.background=element_rect(fill = alpha("white", 0.1)),legend.title = element_blank(),legend.text = element_text(size = 7))

tree_05

    png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/LAB_pyhlo.png", width = 2200, height = 2200,res=300)
pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/LAB_pyhlo.pdf",width=10,height=10)

tree_05


dev.off()


###-------------------------
#add lac-operon
###-------------------------

presence_lacOPERON <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/lactose_utilziation/presence_lacOPERON.txt", 
    delim = "\t", escape_double = FALSE, 
    col_names = c("species","lacZ","lacG"), trim_ws = TRUE) 


presence_lacOPERON$lac_Operon <- ifelse(presence_lacOPERON$lacZ>0&presence_lacOPERON$lacG>0,"both",ifelse(presence_lacOPERON$lacZ>0,"lacZ",ifelse(presence_lacOPERON$lacG>0,"lacG","NA")))
table(presence_lacOPERON$lac_Operon)


presence_lacOPERON$species <- gsub("Streptococcus-","S. ",presence_lacOPERON$species)
presence_lacOPERON$species <- gsub("Lactobacillus-","L. ",presence_lacOPERON$species)
presence_lacOPERON$species <- gsub("L. delbrueckii-","L. del-",presence_lacOPERON$species)

unique(presence_lacOPERON$species)[!unique(presence_lacOPERON$species) %in% unique(Sterm_tree_raw_tible$label)]

    
library(ggnewscale)
     



     tree_06 <- tree_05+ new_scale_fill() +geom_fruit(data=presence_lacOPERON, geom=geom_tile,
                    mapping=aes(y=species,  fill=lac_Operon),
                    # pwidth=0.09, 
                    width=0.02,
                    offset=0.04,
                    orientation="y", 
                    stat="identity",
         )+     scale_fill_manual(values=c("grey66","grey77","grey88","white"))+geom_treescale()
tree_06

  # png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/LAB_pyhlo.png", width = 2200, height = 2200,res=300)
pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/LAB_pyhlo_02.pdf",width=10,height=10)

tree_06


dev.off()



```

### FINAL: make Ldel, Sterm and Lhelv trees

```{r}


library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02//RAxML_bipartitions.lactos_curated", tree.names = NULL, force.multi = FALSE)

Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03//RAxML_bipartitions.lactos_curated", tree.names = NULL, force.multi = FALSE)


tree_01 <- ggtree(Sterm_tree_raw) +
  geom_tiplab(size=0.5,align = FALSE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT
tree_01

library(castor)


##=======================================
##------Sterm
##=======================================


Sterm_tree_raw %>% as_tibble() %>% filter(!is.na(label)) %>% filter(grepl("treptococcus",label)) %>% pull(label) 

# 
tmp <- as.phylo(Sterm_tree_raw)
# typeof(tree_01)

Sterm_tree_raw_sub <- get_subtree_with_tips(tmp, 
                      only_tips               =c("Streptococcus-thermophilus_G000253395","Streptococcus-vestibularis_own","Streptococcus-salivarius_G000785515")) 

Sterm_tree_raw_sub_02 <- Sterm_tree_raw_sub$subtree

Sterm_tree_raw_sub_02_Sterm <- Sterm_tree_raw_sub_02 %>% as_tibble() %>% mutate(branch.length=branch.length*100000000) %>% mutate(label=gsub("_.*","",label))  %>% mutate(label=gsub("S.*-","S. ",label)) %>% as.treedata()





tree_01_sterm <- ggtree(Sterm_tree_raw_sub_02_Sterm) +
  geom_tiplab(size=3,align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) +geom_treescale(x=0.1, y=3.5, offset=.1,width=250000)  + annotate('text', x=120000, y=3.25, label="years\nsince divergence")  + ggplot2::xlim(0, 1000000)#NO ROOT
tree_01_sterm



# tree_01 <- ggtree(Sterm_tree_raw_sub_02) +
#   geom_tiplab(size=3,align = FALSE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0) + geom_text(aes(label=node)) #NO ROOT
# tree_01

# write.nexus(Sterm_tree_raw_sub_02,"~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03//RAxML_bipartitions.lactos_curated_sterm")
# write.tree(Sterm_tree_raw_sub_02,"~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03//RAxML_bipartitions.lactos_curated_sterm")
# 
# ##----------rotat
# tree_01 <- ggtree(Sterm_tree_raw_sub_02) +
#   geom_tiplab(size=3,align = FALSE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0) + geom_text(aes(label=node))%>% rotate(5)  #NO ROOT
# tree_01
# 
# library(gridExtra)
# library(tidytree)
# library(treeio)
# library(ape)
# flip(tree_01, 'Streptococcus-vestibularis_own', 'Streptococcus-thermophilus_G000253395')
# 
# 
# rotate(tree_01, c('Streptococcus-vestibularis_own', 'Streptococcus-thermophilus_G000253395'))
# 
# y <- ape::rotateConstr(tree_01, c('Streptococcus-salivarius_G000785515', 'Streptococcus-vestibularis_own', 'Streptococcus-thermophilus_G000253395'))
# y
# 
# # "Streptococcus-thermophilus_G000253395","Streptococcus-vestibularis_own","Streptococcus-salivarius_G000785515"
# # %>% rotate(33) 
# 
# gridExtra::grid.arrange(tree_01, flip(tree_01, 3, 2), ncol=2)
# 
# gridExtra::grid.arrange(Sterm_tree_raw_sub_02, flip(p, 5, 1), ncol=2)
# 
# ##----------reroot
# 
# Sterm_tree_raw_sub_03 <- root(Sterm_tree_raw_sub_02, outgroup = "Streptococcus-thermophilus_G000253395", edgelabel = TRUE)
# tree_01 <- ggtree(Sterm_tree_raw_sub_03) +
#   geom_tiplab(size=3,align = FALSE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0)   + geom_text(aes(label=node))#NO ROOT
# tree_01
# 
# 
# ggtree(tr) + geom_text(aes(label=node))
# 
# ##----------reimport
# 
# 
# Sterm_tree_raw_sterm <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03//RAxML_bipartitions.lactos_curated_sterm_curated", tree.names = NULL, force.multi = FALSE)
# 
# tree_01 <- ggtree(Sterm_tree_raw_sterm) +
#   geom_tiplab(size=3,align = TRUE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0)  #NO ROOT
# tree_01


##=======================================
##------Lhelv
##=======================================

Sterm_tree_raw %>% as_tibble() %>% filter(!is.na(label)) %>% filter(grepl("actobacillus",label)) %>% pull(label) 
Sterm_tree_raw %>% as_tibble() %>% filter(!is.na(label)) %>% filter(grepl("helveticus",label)) %>% pull(label) 
Sterm_tree_raw %>% as_tibble() %>% filter(!is.na(label)) %>% filter(grepl("gallinarum",label)) %>% pull(label) 
Sterm_tree_raw %>% as_tibble() %>% filter(!is.na(label)) %>% filter(grepl("crisp",label)) %>% pull(label) 


Sterm_tree_raw_sub <- get_subtree_with_tips(tmp, 
                      only_tips               =c("Lactobacillus-helveticus_G000422165","Lactobacillus-gallinarum_oldAnalysis","Lactobacillus-crispatus_G000091765")) 

Sterm_tree_raw_sub_Lhelv <- Sterm_tree_raw_sub$subtree

Sterm_tree_raw_sub_Lhelv_02 <- Sterm_tree_raw_sub_Lhelv %>% as_tibble() %>% mutate(branch.length=branch.length*100000000) %>% mutate(label=gsub("_.*","",label))  %>% mutate(label=gsub("L.*-","L. ",label)) %>% as.treedata()

tree_01_lhelv <- ggtree(Sterm_tree_raw_sub_Lhelv_02) +
  geom_tiplab(size=3,align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) +geom_treescale(x=0.1, y=3, offset=.1,width=250000)  + annotate('text', x=120000, y=2.8, label="years\nsince divergence")  + ggplot2::xlim(0, 4000000)#NO ROOT
tree_01_lhelv

##=======================================
##------Ldel
##=======================================

Sterm_tree_raw %>% as_tibble() %>% filter(!is.na(label)) %>% filter(grepl("delbrueckii",label)) %>% pull(label) 
Sterm_tree_raw %>% as_tibble() %>% filter(!is.na(label)) %>% filter(grepl("equicursoris",label)) %>% pull(label)
# Sterm_tree_raw %>% as_tibble() %>% filter(!is.na(label)) %>% filter(grepl("gallinarum",label)) %>% pull(label) 
# Sterm_tree_raw %>% as_tibble() %>% filter(!is.na(label)) %>% filter(grepl("crisp",label)) %>% pull(label) 


Sterm_tree_raw_sub <- get_subtree_with_tips(tmp, 
                      only_tips               =c("Lactobacillus-delbrueckii-bulgaricus_own","Lactobacillus-delbrueckii-indicus_own","Lactobacillus-delbrueckii-sunkii_own","Lactobacillus-delbrueckii-lactis_own","Lactobacillus-delbrueckii-jakobsenii_own","Lactobacillus-equicursoris_oldAnalysis")) 

Sterm_tree_raw_sub_Ldel <- Sterm_tree_raw_sub$subtree

Sterm_tree_raw_sub_Ldel_02 <- Sterm_tree_raw_sub_Ldel %>% as_tibble() %>% mutate(branch.length=branch.length*100000000)%>% mutate(label=gsub("_.*","",label))  %>% mutate(label=gsub("L.*-","L. ",label)) %>% as.treedata()

 

tree_01_ldel <- ggtree(Sterm_tree_raw_sub_Ldel_02) +
  geom_tiplab(size=3,align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) +geom_treescale(x=1, y=6, offset=.1,width=250000)  + annotate('text', x=120000, y=5.7, label="years\nsince divergence") + ggplot2::xlim(0, 6000000)#NO ROOT#NO ROOT
tree_01_ldel

###==================================
#strain age
###==================================


Sterm_tree_raw_species_focal <-  Sterm_tree_raw %>% as_tibble() %>%  filter(grepl("Streptococcus-thermophilus|Lactobacillus-delbrueckii-lacti|Lactobacillus-helv",label)) %>% mutate(age_of_species=branch.length*100000000)

mean(Sterm_tree_raw_species_focal$age_of_species)





###==================================
#bring trees together
###==================================

library(patchwork)
tree_01_sterm+tree_01_ldel+tree_01_lhelv


pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Zoom_species_rrna_phylo_01.pdf",width=9,height=3)

tree_01_sterm+tree_01_ldel+tree_01_lhelv


dev.off()

```



## Earthmicrobiome data

```{r}
 Species_distribution_preped <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/Earthmicrobiome_data/Final/Final/VinceDB_Species_distribution_prepeed.csv")


Species_distribution_preped_sub <- Species_distribution_preped %>% filter(Species_name %in% c("L. helveticus","S. thermophilus","L. del-lactis","S. vestibularis","L. crispatus","L. equicursoris"))

Species_distribution_preped_sub %>% filter(Habitat_broad=="Animal sampling")

pseudo_sub <- pseudogeness_sumary %>% filter(species_new %in% c("helveticus","S. thermophilus","L. delbrueckii subsp. lactis","vestibularis","crispatus","equicursoris"))

```

## data of tree

```{r}
library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(philr)
# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02//RAxML_bipartitions.lactos_curated", tree.names = NULL, force.multi = FALSE)


library(nodiv)
Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03//RAxML_bipartitions.lactos_curated", tree.names = NULL, force.multi = FALSE)


##-----------------------------------------
#make phylo
##-----------------------------------------

Sterm_tree_raw_phylo <- Sterm_tree_raw %>% as.phylo() 

distance <- mean_dist_to_tips(Sterm_tree_raw_phylo) %>% as.data.frame()  %>% as_tibble() %>% mutate(branchAge_calculated_02 =.*5000) %>% select(-.) #%>% left_join(.,info)

distance$node <- nodenumbers(Sterm_tree_raw_phylo)

range(distance$branchAge_calculated_02)

# write.csv(distance,"~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/age_LAB_phylo.csv")

##-----------------------------------------
#identify the age of the clades
##-----------------------------------------

Sterm_tree_raw_phylo_tips <- Sterm_tree_raw_phylo$tip.label %>% as.data.frame()  %>% as_tibble() %>% mutate(genus=gsub("-.*","",.))%>% mutate(genus=gsub("'","",genus))
table(Sterm_tree_raw_phylo_tips$genus) %>% length()

colnames(Sterm_tree_raw_phylo_tips)[1] <- "tip.name"
age_genus_LAB <- data.frame()
for (genusss in unique(Sterm_tree_raw_phylo_tips$genus)) {
    tmp <- Sterm_tree_raw_phylo_tips %>% filter(genus==genusss)

  if (nrow(tmp)>1) {
      tmp1 <- getMRCA(Sterm_tree_raw_phylo,tmp$tip.name)
     tmp2 <-  data.frame(genus=genusss,node=tmp1)
  
     
     age_genus_LAB <- rbind(age_genus_LAB,tmp2)
  }

}

age_genus_LAB_extension <- age_genus_LAB %>% left_join(.,distance) %>% as_tibble()

age_genus_LAB_extension_prep <- age_genus_LAB_extension %>% mutate(taxon=genus)%>% mutate(group="Genus") %>% select(taxon,group,branchAge_calculated_02)%>% mutate(speciess="rest")

age_genus_LAB_extension_prep_02 <- age_genus_LAB_extension_prep %>% filter(taxon %in% c("Streptococcus","Lactobacillus"))%>% mutate(speciess=taxon)



##-----------------------------------------
#identify the age of the clades species
##-----------------------------------------



Sterm_tree_raw_species <- Sterm_tree_raw %>% as_tibble() %>% mutate(Stem_to_species=parent)%>% mutate(node=parent) %>% select(Stem_to_species,label,node) %>% filter(!is.na(label))%>% left_join(.,distance) %>% as_tibble() %>% select(-node)


Sterm_tree_raw_species_focal_prep <- Sterm_tree_raw_species %>% filter(grepl('Streptococcus|Lactobacillus', label)) %>% mutate(group="Species") %>% mutate(taxon=label)%>% select(taxon,group,branchAge_calculated_02) %>% mutate(speciess="rest")

Sterm_tree_raw_species_focal_prep_02 <- Sterm_tree_raw_species_focal_prep %>% filter(grepl("Streptococcus-thermophilus|Lactobacillus-delbrueckii-lacti|Lactobacillus-helv",taxon)) %>% mutate(speciess=gsub("_.*","",taxon))

mean(Sterm_tree_raw_species_focal_prep_02$branchAge_calculated_02)


total_age <- rbind(age_genus_LAB_extension_prep,age_genus_LAB_extension_prep_02,Sterm_tree_raw_species_focal_prep,Sterm_tree_raw_species_focal_prep_02)


ggplot(total_age,aes(x=speciess,y=branchAge_calculated_02,color=speciess,fill=speciess,shape=group))+geom_boxplot()+theme_classic()+facet_wrap(~group,scales = "free_x")


total_age_summed <- total_age %>% group_by(speciess,group) %>% summarise(Age=mean(branchAge_calculated_02),variation=sd(branchAge_calculated_02))


ggplot(total_age_summed, aes(speciess, Age,fill=speciess,color=speciess)) +
  geom_point() + 
  geom_pointrange(aes(ymin = Age-variation, ymax = Age+variation),data = total_age_summed)+theme_classic()+facet_wrap(~group,scales = "free_y",nrow = 2)+
  coord_flip() +theme_classic()+coord_trans(y="log2")


total_age_summed$speciess = factor( total_age_summed$speciess, levels=c("rest",  "Streptococcus", "Streptococcus-thermophilus", "Lactobacillus" , "Lactobacillus-delbrueckii-lactis","Lactobacillus-helveticus"))


colorsss <- rev(c("#007fa6ff","#fdc2b5ff","#b3b3b3ff",))

 colorsss <- rev(c("#b3b3b3ff","#007fa6ff","#007fa6ff","#fdc2b5ff","#fdc2b5ff","grey88"))


plot_AGE <- ggplot(total_age_summed, aes(y=speciess, x=Age,fill=speciess,color=speciess)) +
  geom_point(position=position_dodge(width=7.0)) + 
  geom_pointrange(position=position_dodge(width=7.0),aes(xmin = Age-variation, xmax = Age+variation),data = total_age_summed)+theme_classic()+facet_wrap(~group,scales = "free_y",nrow = 2, strip.position = "right")+
  theme_classic()+scale_x_continuous(trans="log2",breaks =c(10,50,100,500,1000))+scale_fill_manual(values = colorsss)+scale_color_manual(values = colorsss)+
  theme(axis.text.y = element_blank(),axis.title.y = element_blank(),axis.ticks.y = element_blank(),legend.position = "none",strip.text = element_text(size = 9))+labs(x="Branch age (mio. years)")+geom_vline(xintercept = 250,color="grey")


plot_AGE

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/_Sterm_age_plot.pdf",width=3,height=3)

plot_AGE


dev.off()




cols_1 <- rainbow(14) 
cols_2 <- rainbow(16) 


branch_age_plot <- ggplot() +
  geom_pointrange(aes(x=grouping_extended, y=branchAge_calculated, ymin = branchAge_calculated-branchAge_variation, ymax = branchAge_calculated+branchAge_variation,fill=grouping_extended,color=grouping_extended),data = myTree_all_beast_tibble_extended_all_02)+theme_classic()+facet_wrap(group~.,nrow = 3,scales = "free_y", strip.position = "right")+
  coord_flip()+scale_fill_manual(values = colorsss)+scale_color_manual(values = colorsss)+
  theme(axis.text.y = element_blank(),axis.title.y = element_blank(),axis.ticks.y = element_blank(),legend.position = "none",strip.text = element_text(size = 7))+labs(y="Branch age")

branch_age_plot



# coord_flip() +
# facet_wrap(~group,nrow = 2)+

# getMRCA(tree_phylo, c("rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24736","rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24739","rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24743"))
# is.na(Sterm_tree_raw_phylo_tips$genus) %>% sum()


```




###PGAP

###### pseudogene origin

```{bash}

##-------------------------------------------------
#testing
##-------------------------------------------------

### pgap gbff 2 gff

gbff2gff.sh

genomesss_short=GCF_900322585.1_Lactobacillus_delbrueckii_subsp._lactis_genomic

mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/

zgrep "/pseudo" /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF/${genomesss_short}.gbff.gz |wc -l

#gbff2gff.sh /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF/${genomesss_short}.gbff.gz /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff
rm /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff

/home/vincent/anaconda3/envs/PROKKA/bin/bp_genbank2gff3.pl  --quiet  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF/${genomesss_short}.gbff.gz -out stdout > /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff



#of interes
pseudogenic_exon

grep "^NZ_LS99" /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff |cut -f 3 |sort|uniq -c
awk '{if($3=="pseudogene" || $3=="pseudogenic_exon") print $0}' /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff

awk '{if($3=="CDS" || $3=="pseudogenic_exon") print $0}' /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff |grep "pseudogenic_exon" -A 2 -B 2

# /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff |head -20

grep "transpos" /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff |head -20





##---------------------------------------------------
#only RMK strains
##---------------------------------------------------


##--------------------
#make  gff file
##--------------------

for species in $(echo "Ldel Sterm")
do
echo "==============================="
echo ${species}
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}

for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/ |grep ".gbk" |sed 's/.gbk//g')
do
echo ${genomessss}


/home/vincent/anaconda3/envs/PROKKA/bin/bp_genbank2gff3.pl  --quiet  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/${genomessss}.gbk -out stdout > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}/${genomessss}.gff


done #${genomessss}


done #species




##--------------------
#identify pseudo gene origin
##--------------------

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log/pseudo_origin.txt


#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/Sterm/24730.gbk
#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF

for species in $(echo "Ldel Sterm")
do


for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/ |grep ".gbk" |sed 's/.gbk//g')
do

#awk '{if($3=="CDS" || $3=="pseudogenic_exon") print $0}' /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/${species}/genomes/GFF/${genomesss_short}.gff |grep "pseudogenic_exon" -A 2 -B 2 |

for pseudognesss in $(awk '{if($3=="pseudogenic_exon") print $9}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}/${genomessss}.gff |cut -d ';' -f 1 )
do

#echo $pseudognesss

reason=$(awk '{if($3=="CDS" || $3=="pseudogenic_exon") print $0}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}/${genomessss}.gff |grep "${pseudognesss};"  |sed 's/.*;Note=//g' |sed 's/%.*//g')

count1=$(awk '{if($3=="CDS" || $3=="pseudogenic_exon") print $0}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}/${genomessss}.gff |grep "${pseudognesss};" -A 2 -B 2 |grep "transpos" -i |wc -l)
count2=$(awk '{if($3=="CDS" || $3=="pseudogenic_exon") print $0}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}/${genomessss}.gff |grep "${pseudognesss};" -A 2 -B 2 |grep "=IS" -i  |wc -l)
countFinal=$((count1+count2))
echo -e ${species}"\t"${genomessss}"\t"${pseudognesss}"\t"${reason}"\t"${countFinal} 
echo -e ${species}"\t"${genomessss}"\t"${pseudognesss}"\t"${reason}"\t"${countFinal} >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log/pseudo_origin.txt

done #pseudognesss


done #genomessss

done #species

##---------------------------------------
#move local
##---------------------------------------


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log/pseudo_origin.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/

```

plotting

```{r}
library(tidyverse)
pseudo_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/pseudo_origin.txt", 
    delim = "\t", escape_double = FALSE, 
    col_names = c("species","genome","gene","reason","transoposon"), trim_ws = TRUE, col_types = cols(genome = col_character())) %>% mutate(pseudoTrans=ifelse(transoposon==0,"NO","YES"))

pseudo_origin$species  <- plyr::revalue(pseudo_origin$species,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus"))



# table(pseudo_origin$pseudoTrans)

pseudogene_origin_summary <- pseudo_origin %>% group_by(species,genome,reason,pseudoTrans) %>% summarise(countss=n()) %>% filter(reason %in% c("frameshifted","incomplete","internal stop")) %>% group_by(species,reason,pseudoTrans) %>% summarise(means=mean(countss),medians=median(countss),sdss=sd(countss))



ggplot(pseudogene_origin_summary,aes(x=reason,y=medians,color=pseudoTrans,fill=pseudoTrans))+geom_col()+theme_classic()+facet_wrap(~species)
ggplot(pseudogene_origin_summary,aes(x=reason,y=medians,fill=pseudoTrans))+geom_bar(stat = "identity")+theme_classic()+facet_wrap(~species)+geom_errorbar(aes(ymin=medians-sdss, ymax=medians+sdss), width=.2) 



##with errorbars

pseudogene_origin_summary %>% spread(., reason, medians) %>% mutate(y=error=ifelse(pseudoTrans=="YES",medians,))

###Overall


pseudogene_origin_summary_wo <- pseudo_origin %>% group_by(species,genome,reason) %>% summarise(countss=n()) %>% filter(reason %in% c("frameshifted","incomplete","internal stop")) %>% group_by(species,reason) %>% summarise(means=mean(countss),medians=median(countss),sdss=sd(countss))

# ggplot(pseudogene_origin_summary_wo,aes(x=reason,y=medians))+geom_bar(stat = "identity")+theme_classic()+facet_wrap(~species)+geom_errorbar(aes(ymin=medians-sdss, ymax=medians+sdss), width=.2) 
plotOverall <- ggplot()+geom_bar(data=pseudogene_origin_summary_wo,aes(x=reason,y=medians),stat = "identity",fill="#254E70")+theme_classic()+facet_wrap(~species)+geom_errorbar(data=pseudogene_origin_summary_wo,aes(x=reason,y=medians,ymin=medians-sdss, ymax=medians+sdss), width=.2) +labs(x="",y="Pseudogenes")
plotOverall



  png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/plot_pseudoCount.png", width = 2200, height = 1500,res=300)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

plotOverall


dev.off()

###transposons


pseudogene_origin_summary_transp <- pseudogene_origin_summary %>% filter(pseudoTrans=="YES")

plottranpso <-ggplot()+geom_bar(data=pseudogene_origin_summary_wo,aes(x=reason,y=medians),stat = "identity",fill="#254E70")+theme_classic()+facet_wrap(~species)+geom_errorbar(data=pseudogene_origin_summary_wo,aes(x=reason,y=medians,ymin=medians-sdss, ymax=medians+sdss), width=.2) +
      geom_bar(data=pseudogene_origin_summary_transp,aes(x=reason,y=medians),stat = "identity",fill="#C33C54")+geom_errorbar(data=pseudogene_origin_summary_transp,aes(x=reason,y=medians,ymin=medians-sdss, ymax=medians+sdss), width=.2)+labs(x="",y="Pseudogenes")
plottranpso

  png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement/plot_pseudoCount_colored.png", width = 2200, height = 1500,res=300)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

plottranpso

dev.off()

#---------------------------------counts

pseudo_origin_strain <- pseudo_origin %>% group_by(species,genome,pseudoTrans) %>% summarise(counts=n()) %>% spread(., pseudoTrans, counts) %>% mutate(total=NO+YES) %>% mutate(percent=100*(YES/total))

pseudo_origin_strain %>% group_by(species) %>% summarise(median=median(percent),sdss=sd(percent))
pseudo_origin_strain %>% mutate(all="all")%>% group_by(all) %>% summarise(median=median(percent),sdss=sd(percent))

```




### Prokka


Here, I prokka annotate all genomes

```{bash}
##---------------------------------------
#merge NCBI and own genomes
##---------------------------------------

##Sterm

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Sterm/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/Streptococcus_thermophilus-*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Sterm/


cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm/01_all_FNA/*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Sterm/

##Lhelv

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Lhelv/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/Lactobacillus_helveticus-*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Lhelv/

##Ldel

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Ldel/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/Lactobacillus_delbrueckii_*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Ldel/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel/01_all_FNA/*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Ldel/


##---------------------------------------
#run Prokka
##---------------------------------------
conda activate PROKKA


for speciess in $(echo "Ldel Sterm Lhelv")
do

for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/${speciess}/ |grep ".fna" |sed 's/.fna//g')
do
echo "==========================================="
echo ${genomes}
rm -r  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/

prokka --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/ --force --proteins proteins.faa --evalue 0.001 --addgenes   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/${speciess}/${genomes}.fna --cpus 5

done
done

#===========================================================================
#ALL SPECIES
#===========================================================================



##------------------------------------------------------------------
#PREP
##------------------------------------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel/01_all_FNA|sed 's/.fna//g')
do

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel/01_all_FNA/${genomesss}.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/Lactobacillus_delbrueckii_subsp_lactis-${genomesss}.fna

done

##----------Sterm

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm/01_all_FNA|sed 's/.fna//g')
do

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm/01_all_FNA/${genomesss}.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/Streptococcus_thermophilus-${genomesss}.fna

done

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/

ls  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/ |wc -l

##------------------------------------------------------------------
#PROKKA
##------------------------------------------------------------------

conda activate PROKKA


for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all// |grep ".fna" |sed 's/.fna//g')
do
echo "==========================================="
echo ${genomes}

rm -r  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${genomes}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/${genomes}/

prokka --outdir //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/${genomes}/ --force --proteins proteins.faa --evalue 0.001 --addgenes   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/${genomes}.fna --cpus 25

done


##------------------------------------------------------------------
##create FAA and FNA folders
##------------------------------------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all*
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/{01_all_FAA,01_all_FNA,01_all_GFF,01_all_FFN}

for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all// |grep ".fna" |sed 's/.fna//g')
do

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/${genomes}//PROKKA*.fna > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FNA/${genomes}.fna

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/${genomes}//PROKKA*.ffn > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FFN/${genomes}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/${genomes}//PROKKA*.faa > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FAA/${genomes}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/${genomes}//PROKKA*.gff > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_GFF/${genomes}.gff

done #genomes


```

###### transposase diversity



```{bash}


rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_analysis.txt
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_transposases_analysis.txt
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats
for speciesss in $(echo "Sterm Ldel")
do
for id in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}/01_all_GFF/ |sed 's/.gff//g')
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
 
#   contiNumber=$(grep ">" -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/2*/*19.gff) 
#   GenomeLength=$(grep -v "^>" /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/2*/*19.gff  |wc -c)
  
  gensss=$(grep "gene: " /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}//${id}/PROKKA_*.txt |sed 's/gene: //g')
  tRNAsss=$(grep "tRNA: "  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}//${id}/PROKKA_*.txt |sed 's/tRNA: //g')
  rRNAsss=$(grep "rRNA: "  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}//${id}/PROKKA_*.txt |sed 's/rRNA: //g')
  
  
  
  transpononssss=$(grep "ISfinder:" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}//${id}/PROKKA_*.gff |wc -l)

grep "ISfinder:" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}//${id}/PROKKA_*.gff |grep ";product=" |awk -F ";product=" '{OFS="\t"}{print $2}'|awk -F " family transposase " -v namzzz="$id" '{OFS="\t"}{print namzzz,$1,$2}' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_transposases_analysis.txt

 echo -e ${id}"\t"${gensss}"\t"${tRNAsss}"\t"${rRNAsss}"\t"${transpononssss}  >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_analysis.txt

    done
done #species


##---------------------------------------
#move local
##---------------------------------------


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_transposases_analysis.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/

```

```{r}
library(tidyverse)
all_annotation_transposases_analysis <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/all_annotation_transposases_analysis.txt", 
    delim = "\t", escape_double = FALSE, 
    col_names = c("genome","Isfamily","ISsubfamily"), trim_ws = TRUE) %>% mutate(species=substr(genome, 1, 1))

all_annotation_transposases_analysis$species  <- plyr::revalue(all_annotation_transposases_analysis$species,c("L"="L.delbrueckii","S"="S.thermophilus"))


table(all_annotation_transposases_analysis$species)


all_annotation_transposases_analysis_summary <- all_annotation_transposases_analysis %>% group_by(species,genome,Isfamily) %>% summarise(countss=n()) %>% group_by(species,Isfamily) %>% summarise(medians=median(countss),means=mean(countss),sdss=sd(countss))%>%replace(is.na(.), 0) 


plottranpso <- ggplot()+geom_bar(data=all_annotation_transposases_analysis_summary,aes(x=Isfamily,y=means),stat = "identity",fill="#254E70")+theme_classic()+facet_wrap(~species, scales="free_x")+geom_errorbar(data=all_annotation_transposases_analysis_summary,aes(x=Isfamily,y=means,ymin=means-sdss, ymax=means+sdss), width=.2) +labs(x="Insertation Sequence Families",y="Transposons")+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),legend.position = "none")
plottranpso


  png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement/plot_transposons_families.png", width = 3000, height = 1500,res=300)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

plottranpso

dev.off()

all_annotation_transposases_analysis_summary$Isfamily %>% unique() %>% length()
```




### pandaroo

2. find core genomes
```{bash}
##==========================================
###-----------------bring all gffs together
##==========================================
species=Sterm

for speciess in $(echo "Ldel Sterm Lhelv")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_GFFs/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_GFFs/
echo $speciess

for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/${speciess}/ |grep ".fna" |sed 's/.fna//g')
do
echo $genomes

#leterzzz=$(echo $genomes |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_GFFs/${genomes}.gff

done
ll  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_GFFs/
done

##==========================================
###-----------------panaroo
##==========================================

##-----sterm

species=Sterm
conda deactivate
conda activate panaroo

#for speciess in $(echo "Ldel Sterm Lhelv")
for speciess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/|grep "^01" -v |sed "s/-.*$//g" |sort|uniq)

do
echo ${speciess}
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/ |grep "${speciess}" -c

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/

panaroo -t 20 -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_GFF/${speciess}-*.gff -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/ --clean-mode strict

done


##==========================================
###-----------------panaroo all species 
##==========================================

##-----sterm

species=Sterm
conda deactivate

conda activate panaroo

for speciess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/ |sed "s/-.*$//g" |sort|uniq)
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/

panaroo -t 25 -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_GFFs/*.gff -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/ --clean-mode strict

done



####-------------------------------------------------
##identify transposase genes
####-------------------------------------------------




```

### pan-core ratio
```{bash}
##==========================================
###-----------------calculate pan-core ratio
##==========================================


rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/All_pan_core.txt
#for speciess in $(echo "Ldel Sterm Lhelv")
for speciess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/|grep "^01" -v |sed "s/-.*$//g" |sort|uniq)

do
echo ${speciess}
countsss=$(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/ |grep "${speciess}" -c)


# cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/summary_statistics.txt
 
pan_count=$(grep "Total genes" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/summary_statistics.txt |awk -F "%)" '{print $2}'| sed 's/^ *//g')
 
core_count=$(grep "Core genes" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/summary_statistics.txt |awk -F "%)" '{print $2}'| sed 's/^ *//g')
 
echo -e ${speciess}"\t"${pan_count}"\t"${core_count}"\t"${countsss} >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/All_pan_core.txt
done

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/All_pan_core.txt

##----------------------------------------------
#bring local
##----------------------------------------------

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/All_pan_core.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/



```
### species infos

```{r}

library(tidyverse)
All_pan_core <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/All_pan_core.txt", delim = "\t", escape_double = FALSE,  col_names = c("species","pan","core","genomes"), trim_ws = TRUE) %>% mutate(pan_core_ratio=pan/core)
All_pan_core$catagory <- ifelse(All_pan_core$species %in% c("Lactobacillus_delbrueckii_subsp_lactis","Lactobacillus_helveticus","Streptococcus_thermophilus"),"focal","not")

# All_pan_core$species  <- plyr::revalueAll_pan_core$species 1,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus"))
# 

All_pan_core$species

ggplot(All_pan_core,aes(x=species,y=core,fill=catagory))+geom_bar(stat = "identity")+theme_classic()+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),legend.position = "none")


ggplot(All_pan_core,aes(x=species,y=pan_core_ratio,fill=catagory))+geom_bar(stat = "identity")+theme_classic()+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),legend.position = "none")


plot_pan_core <- ggplot(All_pan_core,aes(x=core,y=pan_core_ratio,fill=catagory,color=catagory,size=genomes))+geom_point(stat = "identity")+theme_classic()+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),legend.position = "none")+labs(x="core genome size",y="pan-core Ratio")
  
plot_pan_core
png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/plot_pan_core.png", width = 2200, height = 1500,res=300)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

plot_pan_core


dev.off()


##with genome size


pseudogeness_count <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt",  "\t", escape_double = FALSE, col_names = c("species","genome","genes_coding","pseudogenes"), trim_ws = TRUE) %>% filter(pseudogenes<1000) %>% filter(genes_coding<4000)%>% mutate(pseudogenes_percent = 100*(pseudogenes/genes_coding)) %>% filter(species!="Wigg")%>% filter(species!="Scap")


pseudogeness_sumary <- pseudogeness_count %>% group_by(species) %>%  dplyr::summarize(meanPseudo=100*(mean(pseudogenes)/mean(genes_coding)),meangenes=mean(genes_coding),stdevPseudo=100*(sd(pseudogenes)/mean(genes_coding)),stdevgene=sd(genes_coding))



```



### identify duplicated genes

```{bash}
speciess=Sterm
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/
/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv

/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Ldel/gene_presence_absence_roary.csv
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Sterm/gene_presence_absence_roary.csv
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Lhelv/gene_presence_absence_roary.csv


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Ldel/gene_presence_absence_roary.csv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Ldel_gene_presence_absence_roary.csv

scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Sterm/gene_presence_absence_roary.csv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Sterm_gene_presence_absence_roary.csv

scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Lhelv/gene_presence_absence_roary.csv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Lhelv_gene_presence_absence_roary.csv
```

```{r}
library(tidyverse)

##===========================================================
#sterm
#maybe include outgroup!
##===========================================================
speciess <- "Sterm"
Sterm_pandaroo <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Sterm_gene_presence_absence_roary.csv")


##-------------------script-------------

Sterm_pandaroo_short <- Sterm_pandaroo[,c(1,15:ncol(Sterm_pandaroo))]

Sterm_pandaroo_long <- gather(Sterm_pandaroo_short, genome, gene_name, colnames(Sterm_pandaroo_short)[2:ncol(Sterm_pandaroo_short)], factor_key=TRUE,na.rm = TRUE)  %>% mutate(gene_name = strsplit(as.character(gene_name), ";")) %>% unnest(gene_name)


Sterm_pandaroo_count <- Sterm_pandaroo_long %>% group_by(Gene,genome) %>% summarise(gene_count=n())
Sterm_pandaroo_max_gene <- Sterm_pandaroo_count %>% group_by(Gene) %>% summarise(max_gene=max(gene_count))


duplicated_gene_count <-Sterm_pandaroo_max_gene %>% filter(max_gene>1) %>% nrow()
total_gene_count <- Sterm_pandaroo_max_gene$Gene %>% unique %>% length()

##-------------------duplication genes-------------

table(Sterm_pandaroo_max_gene$max_gene)
print(paste0("There are ",total_gene_count," in ",speciess," of which ",duplicated_gene_count," are duplicated"))

##-------------------duplication distribution-------------

duplicated_count <- Sterm_pandaroo_max_gene %>% filter(max_gene>1)

Sterm_pandaroo_count_duplic <- Sterm_pandaroo_count %>% filter(Gene %in% unique(duplicated_count$Gene))
genome_count <- unique(Sterm_pandaroo_long$genome) %>% length()

Sterm_pandaroo_count_duplic$duplicate <- ifelse(Sterm_pandaroo_count_duplic$gene_count>1,"duplicate","single")

Sterm_pandaroo_presence <- Sterm_pandaroo_count_duplic %>% count(Gene, duplicate) %>% mutate(percent_n=100*(n/genome_count)) %>%  select(-n)%>% spread(.,duplicate,percent_n) %>% mutate(missing=100-duplicate-single)

Sterm_pandaroo_presence_long <- gather(Sterm_pandaroo_presence, grouping, percent_group, colnames(Sterm_pandaroo_presence)[2:4], factor_key=TRUE,na.rm = TRUE) 
ggplot(Sterm_pandaroo_presence_long,aes(x=percent_group,fill=grouping))+theme_classic()+geom_histogram()+facet_wrap(~grouping)

##-------------------last common ancestor of all genomes that share a duplicated gene. 

library(ape)
library(ggtree)
library(tidytree)
library(treeio)
library(TreeTools)

Sterm_species_tree <- read.nexus("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Sterm/SpeciesTree_rooted_node_labels_curated.txt", tree.names = NULL, force.multi = FALSE)

gtree_01 <- ggtree(Sterm_species_tree) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  


##-------------------number of ds between duplicate genes within a genome. 
#identify the genes to align
#afterwards use 
#library(PopGenome)

Sterm_pandaroo_count_duplic_sub <- Sterm_pandaroo_count_duplic %>% filter(duplicate=="duplicate")


Sterm_pandaroo_short <- Sterm_pandaroo[,c(1,15:ncol(Sterm_pandaroo))]
Sterm_pandaroo_short$uniqNumber <- 1:nrow(Sterm_pandaroo_short)


Sterm_pandaroo_long <- gather(Sterm_pandaroo_short, genome, gene_name, colnames(Sterm_pandaroo_short)[2:(ncol(Sterm_pandaroo_short)-1)], factor_key=TRUE,na.rm = TRUE)  %>% mutate(gene_name = strsplit(as.character(gene_name), ";")) %>% unnest(gene_name)

extended_gene_dup <- merge(Sterm_pandaroo_long,Sterm_pandaroo_count_duplic_sub,by=c("Gene","genome")) %>% mutate(uniqName=paste0(genome,".",uniqNumber))

write.table(extended_gene_dup,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/extended_gene_dup.csv",sep="\t",quote = FALSE,row.names = FALSE)


##-----------------------------------
#check which are transpoases
##-----------------------------------


Sterm_pandaroo$Gene %>% length()
Sterm_pandaroo$Gene %>% unique() %>% length()


TRANSPOSASE_GENES_01 <- Sterm_pandaroo[grepl("transposas",ignore.case = TRUE,Sterm_pandaroo$Annotation),]
TRANSPOSASE_GENES_02 <- Sterm_pandaroo[grepl("^IS",Sterm_pandaroo$Annotation),]
TRANSPOSASE_GENES_03 <- Sterm_pandaroo[grepl(" insertion sequence",Sterm_pandaroo$Annotation),]

TRANSPOSASE_GENES_total <- rbind(TRANSPOSASE_GENES_01,TRANSPOSASE_GENES_02,TRANSPOSASE_GENES_03)

Sterm_pandaroo_transposase <- Sterm_pandaroo %>% filter(Gene %in% unique(TRANSPOSASE_GENES_total$Gene))


extended_gene_dup_transposase <- extended_gene_dup %>% filter(Gene %in% unique(TRANSPOSASE_GENES_total$Gene))

TRANSPOSASE_GENES_total_sub <- TRANSPOSASE_GENES_total %>% select(Gene,Annotation) %>% unique()

extended_gene_dup_transposase_ext <- merge(extended_gene_dup_transposase,TRANSPOSASE_GENES_total_sub, by="Gene")

write.table(extended_gene_dup_transposase_ext,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/extended_gene_dup_transposase.csv",sep="\t",quote = FALSE,row.names = FALSE)
```

##### calculate duplication substitution

```{bash}
##-----------------------------------------
#transfer to bigmachin
##-----------------------------------------

scp ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/extended_gene_dup.csv  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/



```

```{bash}

speciess=Sterm
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements

maxnum=$(wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/extended_gene_dup.csv |cut -d ' '  -f 1)

counts=2
for counts in {2..20579}
do

#echo -e ${counts}
genomes=$(sed -n "${counts}p" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/extended_gene_dup.csv|cut -f 2)

geness=$(sed -n "${counts}p" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/extended_gene_dup.csv|cut -f 4)

uniqNAME=$(sed -n "${counts}p" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/extended_gene_dup.csv|cut -f 7)


echo -e "Numb: "${counts}" von "${maxnum} ": "${uniqNAME}
samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/*.ffn

samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/*.ffn ${geness} >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/${uniqNAME}.fasta

done

#------------------------------------------------------------------------------------------------
#align the duplicated genes
#------------------------------------------------------------------------------------------------

for uniqNAME in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/ |grep ".fasta" |sed 's/.fasta//g' )
do
echo ${uniqNAME}

#mafft --maxiterate 1000 --globalpair --thread 10  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/${uniqNAME}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/${uniqNAME}_gl.aln

mafft --maxiterate 1000 --genafpair --thread 10  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/${uniqNAME}.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/${uniqNAME}_genefpair.aln

done

#------------------------------------------------------------------------------------------------
#concat genes
#------------------------------------------------------------------------------------------------

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/*.aln |head -100


#seqkit concat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/*.aln > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/all.msa


#------------------------------------------------------------------------------------------------
#bring to local
#------------------------------------------------------------------------------------------------

mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/*.aln ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/


mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm_gl/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/*_gl.aln ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm_gl/

mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm_genefpair/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/*_genefpair.aln ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm_genefpair/

#----------------------------------------

#------------------------------------------------------------------------------------------------
#change names
#------------------------------------------------------------------------------------------------
#cp  ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/S24736.*  ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test/
#cd ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test/
#for filesss in $(ls ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test// |grep ".aln" |sed 's/.aln//g')
#do
#newName=$(echo $filesss|sed 's/\./_/g')
#echo -e $filesss " old 2 new  "$newName
#mv ${filesss}.aln  ${newName}.aln 


done 

```

```{r}

##-------------------number of ds between duplicate genes within a genome. 
#identify the genes to align
#afterwards use 
library(PopGenome)
##-----------------------------------------------------------------
library(tidyverse)
library(turner)

##-----------------------------------------------------------------prepare
# tot_num_file <- list.files("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/", full.names = TRUE, recursive = TRUE) %>% length()
tot_num_file <- list.files("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm_genefpair/", full.names = TRUE, recursive = TRUE) %>% length()


added_final <- data.frame()

for (start_file in seq(1,tot_num_file,1)) {
  end_file <- start_file
  print(paste0("the start is: ",start_file," and the end ",end_file))
  
  dir.create("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test_r/")

 files_2_move <- list.files("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm_genefpair/",  full.names = TRUE, recursive = TRUE)[start_file:end_file]
 file.copy(files_2_move, "~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test_r/")

 try(GENOME.class <- readData("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test_r//"))

 final_data <- data.frame(get.sum.data(GENOME.class)) %>% rownames_to_column(., var = "alignment") %>% as_tibble() %>%
  cbind(.,synonymous=sapply(GENOME.class@region.data@synonymous, sum))%>%
  cbind(.,transitions=sapply(GENOME.class@region.data@transitions, sum))

final_data

added_final <- rbind(added_final,final_data)
 
 unlink("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test_r/", recursive = TRUE)

}

added_final
#write.csv(added_final,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/final_analysis_genefpair_Alignement.txt",quote = FALSE,row.names = FALSE)

##-----------------------------
#read the files
##-----------------------------

analysis_genefpair <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/final_analysis_genefpair_Alignement.txt") %>% mutate(alignment=gsub("_genefpair.aln","",alignment)) %>% mutate(type="genefpair")

analysis_global <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/final_analysis_global_Alignement.txt") %>% mutate(alignment=gsub("_gl.aln","",alignment)) %>% mutate(type="global")

analysis_local <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/final_analysis_local_Alignement.txt") %>% mutate(alignment=gsub(".aln","",alignment)) %>% mutate(type="local")


analysis_all <- rbind(analysis_genefpair,analysis_global,analysis_local) %>% mutate(fac_vallied=100*(n.valid.sites/n.sites))

ggplot(analysis_all,aes(x=fac_vallied,color=type,fill=type))+geom_density(alpha=0.5)
ggplot(analysis_all,aes(x=fac_vallied,y=n.sites,color=type,fill=type))+geom_point()+theme_classic()+facet_wrap(~type)

###---------------------------
#take only > 75% alginment
###---------------------------


analysis_filtered <- analysis_global %>% mutate(fac_vallied=100*(n.valid.sites/n.sites)) %>% filter(fac_vallied>75) %>% mutate(fraction_syn=synonymous/n.valid.sites)


plotSynonomous <- ggplot(analysis_filtered,aes(x=fraction_syn,color=type,fill=type))+geom_density(alpha=0.5)+theme_classic()+theme(legend.position = "none")+labs(x="dS, synonymous substitution rate")
plotSynonomous

  png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/plot_synomous_transposons.png", width = 2200, height = 1500,res=300)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

plotSynonomous


dev.off()

##--------------------------------------
#only transposase
##--------------------------------------

extended_gene_dup_transposase <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/extended_gene_dup_transposase.csv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)

analysis_filtered_tranpos <- analysis_filtered %>% mutate(grouping=ifelse(alignment %in% unique(extended_gene_dup_transposase$uniqName),"transposase","other"))


analysis_filtered_tranpos_ext <- merge(analysis_filtered_tranpos,extended_gene_dup_transposase,by.x="alignment",by.y="uniqName")


analysis_filtered_tranpos_ext_02 <- analysis_filtered_tranpos_ext %>% mutate(grouping_2=ifelse(grepl("IS3 ",Annotation),"transposase","other"))

plotSynonomous <- ggplot(analysis_filtered_tranpos_ext_02,aes(x=fraction_syn,color=type,fill=grouping_2))+geom_density(alpha=0.5)+theme_classic()+theme(legend.position = "right")+labs(x="dS, synonymous substitution rate")
plotSynonomous

```



### species phylogeny: orthofinder

```{bash}

##==========================================
###-----------------bring all faas together
##==========================================
species=Sterm

for speciess in $(echo "Ldel Sterm Lhelv")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_FAAs_with_OUT/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_FAAs_with_OUT/
echo $speciess

for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/${speciess}/ |grep ".fna" |sed 's/.fna//g')
do
echo $genomes

#leterzzz=$(echo $genomes |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_FAAs_with_OUT/${genomes}.faa

done
ll  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_FAAs_with_OUT/
done

##----------------------add outgroup

cp /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA/acidophilus.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/Lhelv/01_all_FAAs_with_OUT/

cp /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA/equicursoris.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/Ldel/01_all_FAAs_with_OUT/

cp /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA/vestibularis.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/Sterm/01_all_FAAs_with_OUT/

##----------------------orthofinder
for speciess in $(echo "Ldel Sterm Lhelv")
do
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/${speciess}/
rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/${speciess}/

    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_FAAs_with_OUT/ \
      -t 10 \
      -o /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/${speciess}/ \
      -a 10
done


##----------------------move loca

mkdir -p //home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Ldel/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Ldel/Results_Jan07/Species_Tree/SpeciesTree_rooted_node_labels.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Ldel/

mkdir -p //home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Sterm/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Sterm/Results_Jan07/Species_Tree/SpeciesTree_rooted_node_labels.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Sterm/

mkdir -p //home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Lhelv/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Lhelv/Results_Jan08/Species_Tree/SpeciesTree_rooted_node_labels.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Lhelv/


```



## large/broad Phylogeny
Here, I do a phylogeny of the entire Lactobacillus delbrueckii complex and the larger Streptococcus complex and include were they are mainly found. 
I do this in order to see if closely related species are also used in food fermentation backgrounds. 
For the lactobacillus part I mainly focus on the [Lactobacillus genus complex describtion ](https://msystems.asm.org/content/4/5/e00264-19) and a [recent comparatative genomics paper](https://aem.asm.org/content/84/17/e00993-18) and the [lifesytles in transition paper](https://academic.oup.com/femsre/article/41/Supp_1/S27/3902999).
For the Streptococcus I look at...

What I do is:

1. Download 1 genome per species (ideally complete and type strain)
2. identify on NCBI were it is extracted from (otherwise look in literature.)
3. do a orthofinder proteome tree

#### Lactobacillus genus complex

```{bash}

#mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
#curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/


rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
genus=Lactobacillus

rm  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=iners

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names


###--------------------------
species=gasseri

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep -v "paragasseri"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep -v "paragasseri"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=johnsonii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=taiwanensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=amylolyticus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names



###--------------------------
species=jensenii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=equicursoris

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "=66c"|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "=66c"|cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=pasteurii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=gigeriorum

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|head -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=acetotolerans

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|head -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=helsingborgensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=melliventris

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kimbladii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=bombicola

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=panisapium

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=apis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=intestinalis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=hamsteri

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kalixensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=helveticus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|tail -1 |cut -f 20   |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=gallinarum

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|tail -1 |cut -f 20   |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kefiranofaciens

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|head -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=crispatus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|head -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=acidophilus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
    
    
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|head -1 |cut -f 20   |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=ultunensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|head -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kitasatonis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|head -1 |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=amylovorus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "DSM"|head -1 |cut -f 20   |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=delbrueckii
subsp=jakobsenii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}"|grep "DSM"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}"|grep "DSM"|head -1 |cut -f 20  |awk -v namezzin="$species" -v subss="$subsp" '{OFS="\t"}{print $1,namezzin"_"subss}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=delbrueckii
subsp=indicus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|head -1 |cut -f 20  |awk -v namezzin="$species" -v subss="$subsp" '{OFS="\t"}{print $1,namezzin"_"subss}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=delbrueckii
subsp=lactis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}"|grep "DSM"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}"|grep "DSM"|head -1 |cut -f 20  |awk -v namezzin="$species" -v subss="$subsp" '{OFS="\t"}{print $1,namezzin"_"subss}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=delbrueckii
subsp=sunkii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}"|head -1 |cut -f 20 |awk -v namezzin="$species" -v subss="$subsp" '{OFS="\t"}{print $1,namezzin"_"subss}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=delbrueckii
subsp=bulgaricus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "${subsp}"|head -1 |cut -f 20 |awk -v namezzin="$species" -v subss="$subsp" '{OFS="\t"}{print $1,namezzin"_"subss}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
##!!!outgroup

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "Holzapfelia" |grep -i "floricola"|grep "DSM"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "Holzapfelia" |grep -i "floricola"|grep "DSM"|head -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,"out_flor"}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA

wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
#_genomic.gff.gz   
#mv GCF_000769515.1_ASM76951v1_protein.faa.gz Out_Bacsub.faa.gz

###-------------------------------------
##change name
###-------------------------------------
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names| cut -d '/' -f 10)
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names|cut -f 2)
echo "-----------------------"
echo $newwsss
mv ${genomessss}*.faa.gz ${newwsss}.faa.gz

done

gunzip *.faa.gz


###-------------------------------------
##count number of genes
###-------------------------------------
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA

for newwsss in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA/ |sed 's/.faa//g')
do
echo "-----------------------"

countsss=$(grep ">" -c  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA/${newwsss}.faa)
echo -e $newwsss"\t"${countsss}

done


##-------------------------16s length
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes
genomes=delbrueckii_lactis.fna
genomes=equicursoris.faa
for genomes in $(ls FAA/)
do

echo "-------------------------------------------"
echo $genomes
grep "16s" -i  -c FAA/${genomes}

grep "16s" -i  FAA/${genomes} |awk -F "location=" '{OFS="\t"}{print $2}'|sed 's/complement(//g'|sed 's/)//g'|awk -F "]" '{OFS="\t"}{print $1}'|awk -F "." '{OFS="\t"}{print $3-$1}'


done
```


excluded because incomplete

```{bash}
###--------------------------
species=kullabergensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=psittaci

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=rodentium

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}'|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} { print $0}'|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=hominis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
 
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|tail -1 |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

```


orthofinder of the faa for the species tree

```{bash}

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/orthofinder

    orthofinder -f /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA/ \
      -t 20 \
      -o /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/orthofinder \
      -a 20

```

also download the genomes (.fna) to do fastANI analysis

```{bash}

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/

sed  's/_protein.faa./_genomic.fna./g' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes_fna.txt

###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA

    wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes_fna.txt

###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names| cut -d '/' -f 10)
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names|cut -f 2)
echo "-----------------------"
echo $newwsss
mv ${genomessss}*.fna.gz ${newwsss}.fna.gz

done
ll
gunzip *.fna.gz

###-------------------------------------
##fastani
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/
mkdir -p  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI//{log,analysis}

for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA |grep ".fna$"|grep -v "lactis")
do
#echo -e "/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/"${genomessss} >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes.txt
echo -e ${genomessss} >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes.txt

done

#echo "/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/delbrueckii_lactis.fna" > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes_lactis.txt

echo "delbrueckii_lactis.fna" > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes_lactis.txt
###---------------------------
##fastaANI
###---------------------------
#for speciesss in $(echo "L.delbrueckii S.thermophilus")

cd  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/
fastANI --fragLen 1000 -t 6 --ql /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes_lactis.txt \
  --rl /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes.txt\
  -o /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/analysis/analysis.out

sed -i 's/.fna//g' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/analysis/analysis.out

```

move local
```{bash}


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/analysis/analysis.out /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/orthofinder/Results_Apr09/Species_Tree/SpeciesTree_rooted_node_labels.txt  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/


```
also download the nucleotide fasta files (.ffn) for mmseqs and generax

```{bash}
##===========================================================
#Ldel
##===========================================================

sed 's/_protein.faa.gz/_cds_from_genomic.fna.gz/g' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes_ffn.txt


#GCF_000494795.1_ASM49479v1_cds_from_genomic.fna.gz
###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN

    wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes_ffn.txt

###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names| cut -d '/' -f 10)
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names|cut -f 2)
echo "-----------------------"
echo $newwsss
mv ${genomessss}*_cds_from_genomic.fna.gz ${newwsss}.ffn.gz

done

gunzip *.ffn.gz


##-------------------------16s length
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes
genomes=delbrueckii_lactis.ffn
genomes=equicursoris.ffn
for genomes in $(ls FFN/)
do

echo "-------------------------------------------"
echo $genomes
grep "16s" -i  FFN/${genomes}
grep "23s" -i  FFN/${genomes}

grep "16s" -i  FFN/${genomes} |awk -F "location=" '{OFS="\t"}{print $2}'|sed 's/complement(//g'|sed 's/)//g'|awk -F "]" '{OFS="\t"}{print $1}'|awk -F "." '{OFS="\t"}{print $3-$1}'
grep "23s" -i  FFN/${genomes} |awk -F "location=" '{OFS="\t"}{print $2}'|sed 's/complement(//g'|sed 's/)//g'|awk -F "]" '{OFS="\t"}{print $1}'|awk -F "." '{OFS="\t"}{print $3-$1}'

done

###-------------?????????????????????????????

sed -i 's/_protein.ffn./_cds_from_genomic.fna./g' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN

    wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt| cut -d '/' -f 10)
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt|cut -f 2)
echo "-----------------------"
echo $newwsss
mv ${genomessss}*.fna.gz ${newwsss}.fna.gz

done
ll
gunzip *.fna.gz

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names| cut -d '/' -f 10)
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names|cut -f 2)
sed -i 's/_protein.ffn./_cds_from_genomic.fna./g' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN

    wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

echo "-----------------------"
echo $newwsss
mv ${genomessss}*.fna ${newwsss}.fna

done
```

```{r}

library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Ldel_broad_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/SpeciesTree_rooted_node_labels_rerooted.txt", tree.names = NULL, force.multi = FALSE)


gtree_01 <- ggtree(Ldel_broad_tree_raw) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  

##--------------------------------
##all info at once 
##--------------------------------
all_sterm_strain_info <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/broad_phylogeny_Ldel_description.csv", ",", escape_double = FALSE, trim_ws = TRUE) %>% select(label,category)

# all_sterm_strain_info$strain <- gsub("S-thermophilus-","",all_sterm_strain_info$strain)

as_tibble(Ldel_broad_tree_raw)$label %in% all_sterm_strain_info$label
 all_sterm_strain_info[all_sterm_strain_info$label %in% as_tibble(Ldel_broad_tree_raw)$label ,]
 all_sterm_strain_info[!all_sterm_strain_info$label %in% as_tibble(Ldel_broad_tree_raw)$label ,]

tmp <- as_tibble(Ldel_broad_tree_raw) %>% filter(!is.na(label))

tmp[!tmp$label %in% all_sterm_strain_info$label,]


tree <- full_join(as_tibble(Ldel_broad_tree_raw), all_sterm_strain_info, by='label') %>% as.treedata()

# gtree_01 <- ggtree(tree) +
#   geom_tiplab(align = TRUE,aes(color=category)) + # tiplabls aligned
#   geom_rootedge(rootedge = 0) +geom_treescale()  #NO ROOT  
#   gtree_01

###--------------------------
#add ANI heatmap
###--------------------------
library(RColorBrewer)
library(ggpubr)
library(ggsignif)
analysis_FASTANI <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/analysis.out",  "\t", escape_double = FALSE, col_names = c("genome1","genome2","ANI","comparabedWindows","totalWindows"), trim_ws = TRUE) %>% select(genome2,ANI)%>% remove_rownames()%>% column_to_rownames(., var = "genome2")%>%replace(is.na(.), "Unknown") 



# p3 <- gheatmap(gtree_01, analysis_FASTANI, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0, color=colorRampPalette(brewer.pal(6,name="RdYlBu"))(37))#+  scale_fill_manual(values=colors_area)
# p3

p3 <- gheatmap(gtree_01, analysis_FASTANI, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0) + scale_fill_viridis_c(option="viridis", name="ANI",na.value = "grey95")
p3


###--------------------------
#add category
###--------------------------
library(RColorBrewer)
library(ggpubr)
library(ggsignif)
# analysis_FASTANI <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/analysis.out",  "\t", escape_double = FALSE, col_names = c("genome1","genome2","ANI","comparabedWindows","totalWindows"), trim_ws = TRUE) %>% select(genome2,ANI)%>% remove_rownames()%>% column_to_rownames(., var = "genome2")%>%replace(is.na(.), "Unknown") 

all_sterm_strain_info_prepped <- all_sterm_strain_info%>% remove_rownames()%>% column_to_rownames(., var = "label")%>%replace(is.na(.), "Unknown") 

# p3 <- gheatmap(gtree_01, analysis_FASTANI, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0, color=colorRampPalette(brewer.pal(6,name="RdYlBu"))(37))#+  scale_fill_manual(values=colors_area)
# p3

p4 <- gheatmap(gtree_01, all_sterm_strain_info_prepped, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0)#+ scale_fill_viridis_d()
p4

###--------------------------
#correct name
###--------------------------

tree <- full_join(as_tibble(Ldel_broad_tree_raw), all_sterm_strain_info, by='label') %>% as.treedata()

tree_02 <- as_tibble(Ldel_broad_tree_raw) %>% mutate(label = ifelse(is.na(label), NA, paste0("L. ",label)))%>% as.treedata()

gtree_02 <- ggtree(tree_02) +
  geom_tiplab(align = TRUE,size=5) + # tiplabls aligned
  geom_rootedge(rootedge = 0)+geom_treescale()  #NO ROOT  
  
gtree_02
###--------------------------
#stitch together
###--------------------------
library(patchwork)

p3+p4+gtree_02

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/phylogeny_Ldel.pdf",width=20,height=10)

p4+gtree_02+p3

dev.off()

```



#### Streptococcus




 add NCBI genomes
```{bash}

#mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
#curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/


rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt
genus=Streptococcus

rm  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

###--------------------------
species=pneumoniae

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20 |awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

###--------------------------
species=mitis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i " ${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i " ${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


###--------------------------
species=parasanguinis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i " ${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i " ${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=sanguinis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i " ${species}"|grep "para" -v|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i " ${species}"|grep "para" -v|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=suis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|'|tail -400 >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

species=suis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}'|tail -400 >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=anginosus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=dysgalactiae
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=pyogenes
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' |tail -400 >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' |tail -400>> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=uberis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=gallolyticus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names
##--------------------------
species=equinus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=vestibularis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=thermophilus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=salivarius
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i " ${species}"|grep -v "thermophilus" |grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >   \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep -v "thermophilus"  |grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=downei
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=criceti
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=mutans
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=macacae
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "representative genome"|tail -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep "representative genome"|tail -1 |cut -f 20|awk -v namezzin="$species" '{OFS="\t"}{print $1,namezzin}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


###--------------------------
##!!!outgroup

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -i "lactococcus lactis subsp. lactis"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -i "lactococcus lactis subsp. lactis"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|head -1  |cut -f 20  |awk -v namezzin="$species" '{OFS="\t"}{print $1,"out_flor"}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA

    wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

#mv GCF_000769515.1_ASM76951v1_protein.faa.gz Out_Bacsub.faa.gz

###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names| cut -d '/' -f 10)
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names|cut -f 2)
echo "-----------------------"
echo $newwsss
mv ${genomessss}*.faa.gz ${newwsss}.faa.gz

done

gunzip *.faa.gz


###-------------------------------------
##count number of genes
###-------------------------------------
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA

for newwsss in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA/ |sed 's/.faa//g')
do
echo "-----------------------"

countsss=$(grep ">" -c  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA/${newwsss}.faa)
echo -e $newwsss"\t"${countsss}

done
###-------------------------------------
##-------------------------16s length
###-------------------------------------

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes
genomes=delbrueckii_lactis.fna
for genomes in $(ls FFN/)
do

echo "-------------------------------------------"
echo $genomes
grep "16s" -i  -c FFN/${genomes}

grep "16s" -i  FFN/${genomes} |awk -F "location=" '{OFS="\t"}{print $2}'|sed 's/complement(//g'|sed 's/)//g'|awk -F "]" '{OFS="\t"}{print $1}'|awk -F "." '{OFS="\t"}{print $3-$1}'


done
```


orthofinder of the faa for the species tree

```{bash}

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/orthofinder

    orthofinder -f /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA/ \
      -t 20 \
      -o /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/orthofinder \
      -a 20

```

also download the genomes (.fna) to do fastANI analysis

```{bash}

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/

sed -i 's/_protein.faa./_genomic.fna./g' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA

    wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names| cut -d '/' -f 10)
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names|cut -f 2)
echo "-----------------------"
echo $newwsss
mv ${genomessss}*.fna.gz ${newwsss}.fna.gz

done
ll
gunzip *.fna.gz

###-------------------------------------
##fastani
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/
mkdir -p  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI//{log,analysis}

for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA |grep ".fna$"|grep -v "thermophilus")
do
#echo -e "/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/"${genomessss} >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes.txt
echo -e ${genomessss} >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/log/genomes.txt

done

#echo "/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/delbrueckii_lactis.fna" > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes_lactis.txt

echo "thermophilus.fna" > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/log/genomes_thermophilus.txt
###---------------------------
##fastaANI
###---------------------------
#for speciesss in $(echo "L.delbrueckii S.thermophilus")

cd  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/
fastANI --fragLen 1000 -t 6 --ql /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/log/genomes_thermophilus.txt \
  --rl /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/log/genomes.txt\
  -o /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/analysis/analysis.out

sed -i 's/.fna//g' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/analysis/analysis.out

```

move local
```{bash}


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/analysis/analysis.out /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/orthofinder/Results_Apr09/Species_Tree/SpeciesTree_rooted_node_labels.txt  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/


```


```{r}

library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Ldel_broad_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/SpeciesTree_rooted_node_labels_rerooted.txt", tree.names = NULL, force.multi = FALSE)


gtree_01 <- ggtree(Ldel_broad_tree_raw) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  
gtree_01
##--------------------------------
##all info at once 
##--------------------------------
all_sterm_strain_info <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/broad_phylogeny_Sterm_description.csv", ",", escape_double = FALSE, trim_ws = TRUE) %>% select(label,category)

# all_sterm_strain_info$strain <- gsub("S-thermophilus-","",all_sterm_strain_info$strain)

# as_tibble(Ldel_broad_tree_raw)$label %in% all_sterm_strain_info$label
#  all_sterm_strain_info[all_sterm_strain_info$label %in% as_tibble(Ldel_broad_tree_raw)$label ,]
#  all_sterm_strain_info[!all_sterm_strain_info$label %in% as_tibble(Ldel_broad_tree_raw)$label ,]
# 
# tmp <- as_tibble(Ldel_broad_tree_raw) %>% filter(!is.na(label))
# 
# tmp[!tmp$label %in% all_sterm_strain_info$label,]
# 
# 
# tree <- full_join(as_tibble(Ldel_broad_tree_raw), all_sterm_strain_info, by='label') %>% as.treedata()

# gtree_01 <- ggtree(tree) +
#   geom_tiplab(align = TRUE,aes(color=category)) + # tiplabls aligned
#   geom_rootedge(rootedge = 0) +geom_treescale()  #NO ROOT  
#   gtree_01

###--------------------------
#add ANI heatmap
###--------------------------
library(RColorBrewer)
library(ggpubr)
library(ggsignif)
analysis_FASTANI <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/analysis.out",  "\t", escape_double = FALSE, col_names = c("genome1","genome2","ANI","comparabedWindows","totalWindows"), trim_ws = TRUE) %>% select(genome2,ANI)%>% remove_rownames()%>% column_to_rownames(., var = "genome2")%>%replace(is.na(.), "Unknown") 



# p3 <- gheatmap(gtree_01, analysis_FASTANI, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0, color=colorRampPalette(brewer.pal(6,name="RdYlBu"))(37))#+  scale_fill_manual(values=colors_area)
# p3

p3 <- gheatmap(gtree_01, analysis_FASTANI, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0) + scale_fill_viridis_c(option="viridis", name="ANI",na.value = "grey95")
p3


###--------------------------
#add category
###--------------------------
library(RColorBrewer)
library(ggpubr)
library(ggsignif)
# analysis_FASTANI <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/analysis.out",  "\t", escape_double = FALSE, col_names = c("genome1","genome2","ANI","comparabedWindows","totalWindows"), trim_ws = TRUE) %>% select(genome2,ANI)%>% remove_rownames()%>% column_to_rownames(., var = "genome2")%>%replace(is.na(.), "Unknown") 

all_sterm_strain_info_prepped <- all_sterm_strain_info%>% remove_rownames()%>% column_to_rownames(., var = "label")%>%replace(is.na(.), "Unknown") 

# p3 <- gheatmap(gtree_01, analysis_FASTANI, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0, color=colorRampPalette(brewer.pal(6,name="RdYlBu"))(37))#+  scale_fill_manual(values=colors_area)
# p3

p4 <- gheatmap(gtree_01, all_sterm_strain_info_prepped, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0)#+ scale_fill_viridis_d()
p4

###--------------------------
#correct name
###--------------------------

tree <- full_join(as_tibble(Ldel_broad_tree_raw), all_sterm_strain_info, by='label') %>% as.treedata()

tree_02 <- as_tibble(Ldel_broad_tree_raw) %>% mutate(label = ifelse(is.na(label), NA, paste0("S. ",label)))%>% as.treedata()

gtree_02 <- ggtree(tree_02) +
  geom_tiplab(align = TRUE,size=5) + # tiplabls aligned
  geom_rootedge(rootedge = 0)+geom_treescale()  #NO ROOT  
  
gtree_02
###--------------------------
#stitch together
###--------------------------
library(patchwork)

p3+p4+gtree_02

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/phylogeny_Sterm.pdf",width=15,height=7)

p4+gtree_02+p3

dev.off()

```


### Lactococcus lactis

```{bash}
Lactococcus_lactis_lactis
Lactococcus_lactis_cremoris
L. lactis subsp. hordniae
L. lactis subsp. tructae

Lactococcus_piscium
Lactococcus_plantarum
Lactococcus_raffinolactis
Lactococcus_chungangensis
Lactococcus_fujiensis
Lactococcus_garvieae
Lactococcus_hircilactis
Lactococcus_petauri
Lactococcus_laudensis
Lactococcus_fujiensis


```
### lactobacillus helvectius

first check which dialact Lactobacillus helvectus strains come from the RMKs

```{bash}

for strainsss in $(cut -f 1 /home/vincent/Desktop/Projects/2020_strainDelineation/01_logs/20200113_Staemme_RMK_onlySequenced.csv |sed '1d')
do
#echo "============================================"
#echo ${strainsss}

grep "${strainsss}"  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/lhelv_staemme_dialact.txt
done

###---use the following strains in the study. 

22155
11051
11961
21456
21462
21463

#---------------------------------------change names


cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/Lhelv_REFs/3bc412fec911891640e38564f3efe5b385f24eb788f380fc60540275_assembly_fasta


for strainnssss in $(echo "22155 11051 11961 21456 21462 21463" )
do
echo -e  ${strainnssss}
mv *${strainnssss}*.fna ${strainnssss}.fna


done

#---------------------------------------move to big machine
scp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/Lhelv_REFs/3bc412fec911891640e38564f3efe5b385f24eb788f380fc60540275_assembly_fasta/* vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/Lhelv/fna/

#---------------------------------------


```


PROKKA annotate of Lhelv

```{bash}
conda activate PROKKA

species=Lhelv
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/

for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/Lhelv/fna/ |sed 's/.fna//g')
do

echo ${genomess}

prokka --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${genomess}/ --force --usegenus --genus Lactobacillus --locustag Lh${genomess} --proteins proteins.faa --evalue 0.001 --addgenes   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/Lhelv/fna/${genomess}.fna --cpus 35


done  

##------------------------------------------------------------------
##create FAA and FNA folders
##------------------------------------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//01_all*
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//{01_all_FAA,01_all_FNA,01_all_GFF,01_all_FFN}

for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/Lhelv/fna/ |sed 's/.fna//g' |grep -v "22155")
do

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${genomess}//PROKKA*.fna >/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//01_all_FNA/${genomess}.fna

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${genomess}//PROKKA*.ffn > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//01_all_FFN/${genomess}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${genomess}//PROKKA*.faa > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//01_all_FAA/${genomess}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${genomess}//PROKKA*.gff > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//01_all_GFF/${genomess}.gff

done #genomes
##------------------------------------------------------------------
##get Lhelv REFs
##------------------------------------------------------------------

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss.txt

for genomessss in $(grep "Lactobacillus_helveticus-" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned |cut -f 3)
do

genomessNEW=$(grep "${genomessss}$" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned  |cut -f 1|sed 's/.*\///g' |cut -d '_' -f 1,2)

echo -e ${genomessss} " new name " ${genomessNEW}

strainsss=$(grep "${genomessNEW}"   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |cut -f 9 |sed 's/strain=//g' |sed 's/ /-/g')
 
grep "${genomessss}$" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned| awk -v strainzz="$strainsss" '{OFS="\t"}{print $0,strainzz}' >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss.txt

done

#--------------------------add biosampl

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss_extended.txt

for genomessss in $(grep "Lactobacillus_helveticus-" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned |cut -f 3)
do

genomessNEW=$(grep "${genomessss}$" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned  |cut -f 1|sed 's/.*\///g' |cut -d '_' -f 1,2)

echo -e ${genomessss} " new name " ${genomessNEW}

biosample=$(grep "${genomessNEW}"   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |cut -f 3 |sed 's/strain=//g' |sed 's/ /-/g')
 
grep "${genomessss}$(printf '\t')" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss.txt | awk -v strainzz="$strainsss" -v biosamplezz="$biosample" '{OFS="\t"}{print $0,biosamplezz}' >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss_extended.txt

done

##------------move local


scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss_extended.txt ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/ #Lhelv_Refsss_extended_cleaned.txt
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss_extended_02.txt ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/ #Lhelv_Refsss_extended_cleaned.txt


##------------------------------------------------------------------
##get Lhelv REFs
##------------------------------------------------------------------

species=Lhelv




#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//01_all*
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_*

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}/{01_all_FAA,01_all_FNA,01_all_GFFs,01_all_FFN}

for genomess in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss.txt)
do

strainsss=$(grep "${genomess}$(printf '\t')"  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss.txt |cut -f 4)

cat  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${species}/${genomess}//PROKKA*.fna >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_FNA/Lh-${strainsss}.fna

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${species}/${genomess}//PROKKA*.ffn >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_FFN/Lh-${strainsss}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${species}/${genomess}//PROKKA*.faa >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_FAA/Lh-${strainsss}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${species}/${genomess}//PROKKA*.gff >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_GFFs/Lh-${strainsss}.gff

done #genomes

###----------------------------------
#add locustag info to pyhlo
###----------------------------------
head -1 ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned.txt  | awk  '{OFS="\t"}{print $0,"locusTAG"}' > ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt

for genomessss in $(cut -f 1 ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info//Lhelv_Refsss_extended_02.txt |sed 's/Lh-//g')
do
echo ${genomessss}
locusTAG=$(grep "${genomessss}$(printf '\t')"  ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info//Lhelv_Refsss_extended_02.txt |cut -f 2)
grep "${genomessss}$(printf '\t')" ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned.txt | awk -v strainzz="$locusTAG" '{OFS="\t"}{print $0,strainzz}' >>  ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt
grep "${genomessss}$(printf '\t')" ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned.txt | awk -v strainzz="$locusTAG" '{OFS="\t"}{print $0,strainzz}'

done




```


## 16s broad phylogeny

#### extract 16s

```{bash}
##------------------------------------------
##2. run barrnap to extract 16s
## AND
##3. run V-Xtractor to extract 16s V4
##------------------------------------------
###---
##Sterm
###---


for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/ |grep ".fna$" |sed 's/.fna//g')
do
echo ${genomess}

seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/${genomess}.fna

echo "================================"
done

#/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/anginosus.fna



rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s
#cd 
speciess=Sterm
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/ |grep ".fna$" |sed 's/.fna//g')
do
echo ${genomess}
cat  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/${genomess}.fna |barrnap --threads 37 -o //data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa 

#speciess=$(grep "${genomess}" /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2)

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa 


#rm  /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna
f16sss=$(grep "16S" -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  |sed 's/>//g'|head -1)


#echo ${f16sss}
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  ${f16sss} > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_16s.fasta

#sed "s/^>.*$/>16S-${speciess}-${genomess}-${num}/g" /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/tmp.fna >> /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna

f23sss=$(grep "23S" -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  |sed 's/>//g'|head -1)
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  ${f23sss} > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_23s.fasta
sed "s/^>.*$/>23S-${speciess}-${genomess}/g" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_23s.fasta >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_23s.ffn


f5sss=$(grep "5S" -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  |sed 's/>//g'|head -1)
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  ${f5sss} > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_5s.fasta
sed "s/^>.*$/>5S-${speciess}-${genomess}/g" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_5s.fasta >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_5s.ffn

  
done #16s


speciess=Streptococcus
rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_16s.ffn
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/ |grep ".fna$" |sed 's/.fna//g')
do
echo ${genomess}

seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa

#grep "16S" -c /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa 

#seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_16s.fasta

sed "s/^>.*$/>16S-${speciess}-${genomess}/g" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_16s.fasta >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_16s.ffn


echo "================================"
done

grep ">" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_16s.ffn

###-----------------------------------------------------------
#merge the different contigs
###-----------------------------------------------------------
speciess=Streptococcus
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/
rm all_rRNAoperon.fasta
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/ |grep ".fna$" |sed 's/.fna//g')
do

echo ">rRNAoperon-${speciess}-${genomess}" >> all_rRNAoperon.fasta
grep ">" -v  ${genomess}_16s.fasta >> all_rRNAoperon.fasta
echo "NNNNNNNNNNNN" >> all_rRNAoperon.fasta
grep ">" -v  ${genomess}_23s.fasta >> all_rRNAoperon.fasta
echo "NNNNNNNNNNNN" >> all_rRNAoperon.fasta
grep ">" -v ${genomess}_5s.fasta >> all_rRNAoperon.fasta


##make the assembly fit together nicely
awk 'BEGIN{RS=">"}NR>1{sub("\n","\t"); gsub("\n",""); print RS$0}' all_rRNAoperon.fasta |sed 's/\t/\n/g' |awk -v FS= '/^>/{print;next}{for (i=0;i<=NF/60;i++) {for (j=1;j<=60;j++) printf "%s", $(i*60 +j); print ""}}' > all_rRNAoperon_prep.fasta


done

seq_length.py all_rRNAoperon_prep.fasta
```

##### lactobacillus

```{bash}
##------------------------------------------
##2. run barrnap to extract 16s
## AND
##3. run V-Xtractor to extract 16s V4
##------------------------------------------
###---
##Ldel
###---



for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/ |grep ".fna$" |sed 's/.fna//g')
do
echo ${genomess}

seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/${genomess}.fna

echo "================================"
done

#/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/anginosus.fna



rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s
genomess=equicursoris

speciess=Ldel
cd //data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/ |grep ".fna$" |sed 's/.fna//g')
do
echo ${genomess}
cat  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/${genomess}.fna |barrnap --threads 37 -o //data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa 

#speciess=$(grep "${genomess}" /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/LdelDownloadgenomes.names |cut -f 2)

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa 


#rm  /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna
f16sss=$(grep "16S" -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa  |sed 's/>//g'|head -1)


#echo ${f16sss}
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa  ${f16sss} > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_16s.fasta

#sed "s/^>.*$/>16S-${speciess}-${genomess}-${num}/g" /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/tmp.fna >> /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna

f23sss=$(grep "23S" -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa  |sed 's/>//g'|head -1)
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa  ${f23sss} > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_23s.fasta
sed "s/^>.*$/>23S-${speciess}-${genomess}/g" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_23s.fasta >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_23s.ffn


f5sss=$(grep "5S" -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa  |sed 's/>//g'|head -1)
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa  ${f5sss} > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_5s.fasta
sed "s/^>.*$/>5S-${speciess}-${genomess}/g" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_5s.fasta >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_5s.ffn

  
done #16s


speciess=Lactobacillus
rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_16s.ffn
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/ |grep ".fna$" |sed 's/.fna//g')
do
echo ${genomess}

#seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/${genomess}.fna

#grep "16S" -c /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa 

seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_16s.fasta
seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_23s.fasta
seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_5s.fasta

sed "s/^>.*$/>16S-${speciess}-${genomess}/g" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_16s.fasta >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_16s.ffn


echo "================================"
done

grep ">" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_16s.ffn

###-----------------------------------------------------------
#merge the different contigs
###-----------------------------------------------------------
speciess=Lactobacillus
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/
rm all_rRNAoperon.fasta
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/ |grep ".fna$" |sed 's/.fna//g')
do

echo ">rRNAoperon-${speciess}-${genomess}" >> all_rRNAoperon.fasta
grep ">" -v  ${genomess}_16s.fasta >> all_rRNAoperon.fasta
echo "NNNNNNNNNNNN" >> all_rRNAoperon.fasta
grep ">" -v  ${genomess}_23s.fasta >> all_rRNAoperon.fasta
echo "NNNNNNNNNNNN" >> all_rRNAoperon.fasta
grep ">" -v ${genomess}_5s.fasta >> all_rRNAoperon.fasta


##make the assembly fit together nicely
awk 'BEGIN{RS=">"}NR>1{sub("\n","\t"); gsub("\n",""); print RS$0}' all_rRNAoperon.fasta |sed 's/\t/\n/g' |awk -v FS= '/^>/{print;next}{for (i=0;i<=NF/60;i++) {for (j=1;j<=60;j++) printf "%s", $(i*60 +j); print ""}}' > all_rRNAoperon_prep.fasta


done

seq_length.py all_rRNAoperon_prep.fasta

```


### phylogeny

```{bash}
##------------------------------------------------------
#make alignement
##------------------------------------------------------

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/

rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta.aln
mafft-linsi --thread 37 --maxiterate 1000   /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta.aln

##------------------------------------------------------
#concatenate
##------------------------------------------------------
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/

  perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/cat_align.pl > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all.phylip
  


##------------------------------------------------------
#pyhlogeny
##------------------------------------------------------


mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/raxml_tree
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/raxml_tree

rm -r *

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta.aln -n Streptococcus_all -T 37


##------------------------------------------------------
#move local
##------------------------------------------------------

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_16s_strepto/


scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/raxml_tree/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_16s_strepto//
```
##### lactobacillus

```{bash}
##------------------------------------------------------
#make alignement
##------------------------------------------------------

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/

rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta.aln
mafft-linsi --thread 37 --maxiterate 1000   /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta.aln

##------------------------------------------------------
#concatenate
##------------------------------------------------------
#cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/

 # perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/cat_align.pl > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all.phylip
  


##------------------------------------------------------
#pyhlogeny
##------------------------------------------------------


mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/raxml_tree
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/raxml_tree

rm -r *

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta.aln -n L_all -T 37


##------------------------------------------------------
#move local
##------------------------------------------------------

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_16s_lacto/


scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/raxml_tree/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_16s_lacto//
```

## include more sal/ves

```{bash}

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log

genus=Streptococcus
##--------------------------
species=vestibularis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"  |awk -F "\t" -v namezzin="$species" '{OFS="\t"}{print $20,namezzin,$2,$3}'  >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log/StermDownloadgenomes.names

##--------------------------
species=salivarius
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i " ${species}"|grep -v "thermophilus" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>  \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i "${species}"|grep -v "thermophilus" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |awk -F "\t" -v namezzin="$species" '{OFS="\t"}{print $20,namezzin,$2,$3}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log/StermDownloadgenomes.names

sed 's/^.*\///g' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log/StermDownloadgenomes.names


##--------------------------
species=equinus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i " ${species}"|grep "representative genome"|grep -v "thermophilus" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>  \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${genus}" |grep -i " ${species}"|grep "representative genome"|grep -v "thermophilus" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|awk -F "\t" -v namezzin="$species" '{OFS="\t"}{print $20,namezzin,$2,$3}' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log/StermDownloadgenomes.names

sed 's/^.*\///g' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log/StermDownloadgenomes.names

###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/Sterm/genomes/FAA
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/Sterm/genomes/FAA

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/Sterm/genomes/FAA

    wget -i  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log/StermDownloadgenomes.txt

#mv GCF_000769515.1_ASM76951v1_protein.faa.gz Out_Bacsub.faa.gz

###-------------------------------------
##change name
###-------------------------------------


scp ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited_extended.csv vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log/

###-------------------------------------
##change name
###-------------------------------------
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/Sterm/genomes/FAA

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log/StermDownloadgenomes.names| cut -d '/' -f 10)
do
newwsss=$(grep "${genomessss}" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen_but_narrower/log/20200728_NCBI_log_shortNames_final_sterm_edited_extended.csv|cut -d ',' -f 1)
echo "-----------------------"
echo $genomessss
echo $newwsss
mv ${genomessss}*.faa.gz ${newwsss}.faa.gz

done

gunzip *.faa.gz



```


## narrow species Phylogeny

Here, we do the analysis of the strain collection. 
I am doing the following analysis:
1. parSNP phylogeny. 
2. ...


### 16S


```{bash}



#for speciess in $(echo "Ldel Sterm Lhelv")
for speciess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/|grep "^01" -v |sed "s/-.*$//g" |sort|uniq)

do
echo ${speciess}
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/ |grep "${speciess}" -c

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/

panaroo -t 20 -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_GFF/${speciess}-*.gff -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/ --clean-mode strict

done




rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/
mkdir -p //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/


#cd 
speciess=Sterm
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FNA/ |grep ".fna$" |sed 's/.fna//g' |grep "Streptococcus_thermophilus-")
do
echo ${genomess}
cat  //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FNA//${genomess}.fna |barrnap --threads 37 -o //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa 

#speciess=$(grep "${genomess}" /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2)

samtools faidx //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa 


#rm  /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna
f16sss=$(grep "16S" -i //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa   |sed 's/>//g'|head -1)


#echo ${f16sss}
samtools faidx //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  ${f16sss} > //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_16s.fasta

#sed "s/^>.*$/>16S-${speciess}-${genomess}-${num}/g" /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/tmp.fna >> /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna

f23sss=$(grep "23S" -i //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_rrna.fa  |sed 's/>//g'|head -1)
samtools faidx //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_rrna.fa  ${f23sss} > //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_23s.fasta
sed "s/^>.*$/>23S-${speciess}-${genomess}/g" //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_23s.fasta >> //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//all_23s.ffn


f5sss=$(grep "5S" -i //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_rrna.fa  |sed 's/>//g'|head -1)
samtools faidx //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_rrna.fa  ${f5sss} > //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_5s.fasta
sed "s/^>.*$/>5S-${speciess}-${genomess}/g" //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_5s.fasta >> //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//all_5s.ffn

  
done #16s


speciess=Streptococcus
rm //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_16s.ffn
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FNA/ |grep ".fna$" |sed 's/.fna//g' |grep "Streptococcus_thermophilus-")
do
echo ${genomess}

seq_length.py //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa

#grep "16S" -c /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa 

#seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_16s.fasta

sed "s/^>.*$/>16S-${speciess}-${genomess}/g" //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_16s.fasta >> //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_16s.ffn


echo "================================"
done

grep ">" //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_16s.ffn

###-----------------------------------------------------------
#merge the different contigs
###-----------------------------------------------------------
speciess=Streptococcus
cd //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s
rm all_rRNAoperon.fasta
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FNA/ |grep ".fna$" |sed 's/.fna//g' |grep "Streptococcus_thermophilus-")
do

echo ">rRNAoperon-${speciess}-${genomess}" >> all_rRNAoperon.fasta
grep ">" -v  ${genomess}_16s.fasta >> all_rRNAoperon.fasta
echo "NNNNNNNNNNNN" >> all_rRNAoperon.fasta
grep ">" -v  ${genomess}_23s.fasta >> all_rRNAoperon.fasta
echo "NNNNNNNNNNNN" >> all_rRNAoperon.fasta
grep ">" -v ${genomess}_5s.fasta >> all_rRNAoperon.fasta


##make the assembly fit together nicely
awk 'BEGIN{RS=">"}NR>1{sub("\n","\t"); gsub("\n",""); print RS$0}' all_rRNAoperon.fasta |sed 's/\t/\n/g' |awk -v FS= '/^>/{print;next}{for (i=0;i<=NF/60;i++) {for (j=1;j<=60;j++) printf "%s", $(i*60 +j); print ""}}' > all_rRNAoperon_prep.fasta


done

seq_length.py all_rRNAoperon_prep.fasta

##-------------------------------------
#add vestibularis
##-------------------------------------

cp all_rRNAoperon_prep.fasta all_rRNAoperon_prep_withVestibularis.fasta

grep ">" /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta


seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta


samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta


samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta rRNAoperon-Streptococcus-vestibularis >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis.fasta

seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis.fasta


###-------------------------------
#extract only complete RNA operons
###-------------------------------

seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/complete_rnas.names | awk '{if($3>4000)print $1}'  > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/complete_rnas.names


samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis.fasta

xargs --arg-file=/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/complete_rnas.names  samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis.fasta >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta


seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta

###---------------------------------------------------
#small test
###---------------------------------------------------


grep "24743" -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta


rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24736
rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24743

samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24743 >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_test.fasta


samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24736 >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_test.fasta

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta rRNAoperon-Streptococcus-vestibularis >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_test.fasta

seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_test.fasta

```


```{bash}
##------------------------------------------------------
#make alignement
##------------------------------------------------------

cd //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s

rm //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta.aln
mafft-linsi --thread 37 --maxiterate 1000   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta.aln




##------------------------------------------------------
#pyhlogeny
##------------------------------------------------------


mkdir -p //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/raxml_tree
cd //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/raxml_tree

rm -r *

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta.aln -n Streptococcus_all -T 37


##------------------------------------------------------
#move local
##------------------------------------------------------

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_16s_strepto_narrow


scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/raxml_tree/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_16s_strepto_narrow//

```


```{r}

##---------------------------------
#identify  most recent common ancestor (MRCA)
##---------------------------------

library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)

Sterm_tree_raw <- read.nexus("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_16s_strepto_narrow_test//raxml_tree_test/RAxML_bipartitions.Streptococcus_all_curated", tree.names = NULL, force.multi = FALSE)
tree_phylo <- Sterm_tree_raw %>% as.phylo()
tree_phylo$tip.label

getMRCA(tree_phylo, c("rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24736","rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24739","rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24743"))




numbers <- Descendants(tree_phylo, 9, type = "tips") %>% unlist()
numbers
tree_phylo$tip.label[numbers]




p <- ggtree(tree) + geom_tiplab()
viewClade(p, MRCA(p, "I", "L"))


getMRCA(tree_phylo, c("S24736","S24758"))

numbers <- Descendants(tree_phylo, 156, type = "tips") %>% unlist()
numbers
tree_phylo$tip.label[numbers]




```


### Poppunk

Here, I use [poppunk](https://poppunk.readthedocs.io/en/latest/model_fitting.html) to identify genome clusters

```{bash}

###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Sterm

#for species in $(echo "Ldel Sterm")
#do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/FNA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/
#done


##====================================================================
##only RMKs
##====================================================================



##--------------------------
##create genome list with path to genome assembly
##--------------------------
species=Sterm
rm  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db_onlyRMK.log
mkdir -p  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA | grep ".fna$"|grep -v "^S-" | sed 's/.fna//g'|grep "Out" -v)
do

genomesss_short=$(echo $genomesss |sed 's/-//g'|sed 's/_//g')

#echo -e ${genomesss}"\t/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/"${species}"/"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}"\t"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
echo -e "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/"${species}"/01_all_FNA/"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db_onlyRMK.log

done


##--------------------------
#easy run 
##--------------------------
rm -r /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun
poppunk --easy-run --r-files /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db_onlyRMK.log --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun --threads 4 --plot-fit 5 --min-k 13 --full-db

poppunk --fit-model --distances /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun.dists --ref-db /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/ --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/ --full-db --K 3


sed 's;\/data\/Project\/2020_StarterCultureDiversity\/10_REFERENCE_GEN0MES\/annotation\/PROKKA\/Sterm\/01_all_FNA\/;;g' /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters.csv |sed 's/.fna//g' |cut -d '_' -f 4 >  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv
```


transfer local

```{bash}

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/fit*.pdf /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/*.png /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/



mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/*.png /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv

```


```{bash}
#==========================================================
#lDEL
#==========================================================
###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Ldel

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/

leterzzz=$(echo $species |head -c 1)

#cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/FNA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/

##HERE I remove the genomes with completness <95 or duplication >10
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/L_I_101_21748.fna
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/L_I_124_21753.fna
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/Ldel/L_I_101_21780.fna

#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/L_I_101_21748.fna
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/L_I_124_21753.fna
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/Ldel/L_I_101_21780.fna
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/Ldel/11077.fna



##--------------------------
##create genome list with path to genome assembly
##--------------------------

species=Ldel
rm  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/${species}_poppunk_db.log
mkdir -p  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/ | grep ".fna$"|sed 's/.fna//g'|grep "Out" -v)
do

genomesss_short=$(echo $genomesss |sed 's/-//g'|sed 's/_//g')

#echo -e ${genomesss}"\t/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/"${species}"/"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}"\t"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
echo -e "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/"${species}"/"${genomesss}".fna" >> /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/${species}_poppunk_db.log

done



##--------------------------
#easy run 
##--------------------------
species=Ldel

rm -r /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}
cd /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/

poppunk --easy-run --ignore-length --r-files /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/${species}_poppunk_db.log --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun --threads 4 --plot-fit 5 --min-k 21 --full-db

poppunk --fit-model --distances /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/20210128_easyrun.dists --ref-db /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/ --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/ --full-db --K 2
#PopPUNK/${species}/20210128_easyrun/ --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/ --full-db --K 6



sed 's;\/data\/Project\/2020_StarterCultureDiversity\/10_REFERENCE_GEN0MES\/genomes\/combined_ncbi_own_popPunk\/FNA\/Ldel\/;;g' /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/20210128_easyrun_clusters.csv |sed 's/.fna//g' >  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/20210128_easyrun_clusters_RMKS_ldel.csv


```


transfer local

```{bash}

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/fit*.pdf /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/*.png /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/



mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/*.png /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Ldel/20210128_easyrun/*.csv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/

```



### fastANI

```{bash}

##------------new PREP
speciesss=Sterm
##-------------------------------------------------



rm -r /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/{log,analysis}
echo "-----------------"
echo ${speciesss}

mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/{log,analysis}
for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ |grep ".fna$")
do
echo -e "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/"${speciesss}"/01_all_FNA/"${genomessss} >> /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt

done

###---------------------------
##fastaANI
###---------------------------
#for speciesss in $(echo "L.delbrueckii S.thermophilus")
for speciesss in $(echo "Sterm")
do
fastANI --fragLen 1000 -t 6 --ql /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt \
  --rl /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt \
  -o /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out
done #speciesss
###----------------------Sterm


cut -f 3 /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out |sort|uniq -c |head
grep "74\." /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out


##------------new PREP
speciesss=Ldel
##-------------------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/{log,analysis}
echo "-----------------"
echo ${speciesss}

mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/{log,analysis}
for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ |grep ".fna$")
do
echo -e "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/"${species}"/01_all_FNA/"${genomessss} >> /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt

done

###---------------------------
##fastaANI
###---------------------------
#for speciesss in $(echo "L.delbrueckii S.thermophilus")
for speciesss in $(echo "Ldel")
do
fastANI --fragLen 1000 -t 6 --ql /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt \
  --rl /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt \
  -o /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out
done #speciesss
###----------------------Sterm


cut -f 3 /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out |sort|uniq -c |head
grep "74\." /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out

##====================================
##all against all (independent of species)
##====================================
rm -r /data/Project/2020_StarterCultureDiversity/05_fastANI/all
mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/all/{log,analysis}
for speciesss in $(echo "L.delbrueckii S.thermophilus")
do

echo "-----------------"
echo ${speciesss}

mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/all/{log,analysis}
for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes)
do
echo -e "/data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/"${speciesss}"/01_genomes/"${genomessss} >> /data/Project/2020_StarterCultureDiversity/05_fastANI/all/log/genomes.txt

done
done #speciesss
###---------------------------
##fastaANI
###---------------------------

fastANI --fragLen 1000 -t 37 --ql /data/Project/2020_StarterCultureDiversity/05_fastANI/all/log/genomes.txt \
  --rl /data/Project/2020_StarterCultureDiversity/05_fastANI/all/log/genomes.txt \
  -o /data/Project/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis.out



cut -f 3 /data/Project/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis.out |sort|uniq -c |head

```

```{bash}
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis.out /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/05_fastANI/Sterm/analysis/analysis.out /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis_sterm.out
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/05_fastANI/Ldel/analysis/analysis.out /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis_ldel.out
```

```{r}

library(readr)
library(tidyverse)
library(ggpubr)
library(ggsignif)
analysis_FASTANI <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis_sterm.out",  "\t", escape_double = FALSE, col_names = c("genome1","genome2","ANI","comparabedWindows","totalWindows"), trim_ws = TRUE)

analysis_FASTANI$species1 <- str_split_fixed(analysis_FASTANI$genome1,"/",11)[,8]
analysis_FASTANI$species2 <- str_split_fixed(analysis_FASTANI$genome2,"/",11)[,8]

analysis_FASTANI$GENOME1 <- str_split_fixed(analysis_FASTANI$genome1,"/",10)[,10] %>% gsub(".fna","",.)
analysis_FASTANI$GENOME2 <- str_split_fixed(analysis_FASTANI$genome2,"/",10)[,10] %>% gsub(".fna","",.)


analysis_FASTANI$comparision <- ifelse(analysis_FASTANI$species1==analysis_FASTANI$species2,"same species","different species")
table(analysis_FASTANI$comparision)

analysis_FASTANI %>%group_by(comparision)  %>% select(ANI) %>%summarize(meansample=mean(ANI))

# sample_relative_temp  %>% group_by(X.CHROM) %>% summarize(meansample=mean(Sequencing_depth))


analysis_FASTANI_final_sterm <- analysis_FASTANI %>% select(!c(genome1,genome2))
###----------------------
##whithin an between clusters
###----------------------

analysis_clusters <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv",  ",")

analysis_FASTANI_final02 <- merge(analysis_FASTANI_final_sterm,analysis_clusters,by.x="GENOME1",by.y="Taxon")
analysis_FASTANI_final03 <- merge(analysis_FASTANI_final02,analysis_clusters,by.x="GENOME2",by.y="Taxon")

analysis_FASTANI_final03$clustercomparison <- ifelse(analysis_FASTANI_final03$Cluster.x==analysis_FASTANI_final03$Cluster.y,"same cluster","different cluster")
analysis_FASTANI_final03$cov <- 100*(analysis_FASTANI_final03$comparabedWindows/analysis_FASTANI_final03$totalWindows)
analysis_FASTANI_sterm <- analysis_FASTANI_final03
plotANI <- ggplot(analysis_FASTANI_final03,aes(x=clustercomparison,y=ANI))+geom_boxplot()+theme_classic()+lims(y=c(98.5,100))+stat_compare_means(label = "p.signif")+labs(x="",y="pairwise ANI")
# ggplot(analysis_FASTANI_final03,aes(x=cov,y=ANI,color=clustercomparison,fill=clustercomparison))+geom_point()+theme_classic()
plotANI 

ggplot(analysis_FASTANI_final03,aes(x=ANI))+geom_density()+theme_classic()+lims(x=c(98.5,100))



# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/ANI_genomes_boxplot.pdf",width=3,height=6)
# 
# plotANI
# 
# dev.off()

##---------------------------------------------------------
#LDEL
##---------------------------------------------------------

library(readr)
library(tidyverse)
library(ggpubr)
library(ggsignif)
analysis_FASTANI <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis_ldel.out",  "\t", escape_double = FALSE, col_names = c("genome1","genome2","ANI","comparabedWindows","totalWindows"), trim_ws = TRUE)

analysis_FASTANI$species1 <- str_split_fixed(analysis_FASTANI$genome1,"/",11)[,8]
analysis_FASTANI$species2 <- str_split_fixed(analysis_FASTANI$genome2,"/",11)[,8]

analysis_FASTANI$GENOME1 <- str_split_fixed(analysis_FASTANI$genome1,"/",10)[,10] %>% gsub(".fna","",.)
analysis_FASTANI$GENOME2 <- str_split_fixed(analysis_FASTANI$genome2,"/",10)[,10] %>% gsub(".fna","",.)


analysis_FASTANI$comparision <- ifelse(analysis_FASTANI$species1==analysis_FASTANI$species2,"same species","different species")
table(analysis_FASTANI$comparision)

analysis_FASTANI %>%group_by(comparision)  %>% select(ANI) %>%summarize(meansample=mean(ANI))

# sample_relative_temp  %>% group_by(X.CHROM) %>% summarize(meansample=mean(Sequencing_depth))


analysis_FASTANI_final_ldel <- analysis_FASTANI %>% select(!c(genome1,genome2))
###----------------------
##whithin an between clusters
###----------------------

analysis_clusters <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_ldel.csv",  ",")

analysis_FASTANI_final02 <- merge(analysis_FASTANI_final_ldel,analysis_clusters,by.x="GENOME1",by.y="Taxon")
analysis_FASTANI_final03 <- merge(analysis_FASTANI_final02,analysis_clusters,by.x="GENOME2",by.y="Taxon")

analysis_FASTANI_final03$clustercomparison <- ifelse(analysis_FASTANI_final03$Cluster.x==analysis_FASTANI_final03$Cluster.y,"same cluster","different cluster")
analysis_FASTANI_final03$cov <- 100*(analysis_FASTANI_final03$comparabedWindows/analysis_FASTANI_final03$totalWindows)

analysis_FASTANI_ldel <- analysis_FASTANI_final03
plotANI <- ggplot(analysis_FASTANI_final03,aes(x=clustercomparison,y=ANI))+geom_boxplot()+theme_classic()+lims(y=c(98.5,100))+stat_compare_means(label = "p.signif")+labs(x="",y="pairwise ANI")
# ggplot(analysis_FASTANI_final03,aes(x=cov,y=ANI,color=clustercomparison,fill=clustercomparison))+geom_point()+theme_classic()
plotANI 



ggplot(analysis_FASTANI_final03,aes(x=ANI))+geom_density()+theme_classic()+lims(x=c(98.5,100))


analysis_FASTANI_together %>% dplyr::group_by(comparision,species1) %>% dplyr::summarise(min=min(ANI))


##---------------------------------------------------------
#toether
##---------------------------------------------------------

analysis_FASTANI_together <- rbind(analysis_FASTANI_sterm,analysis_FASTANI_ldel)


analysis_FASTANI_together$species1  <- plyr::revalue(analysis_FASTANI_together$species1,c("Ldel"="L.delbrueckii","Sterm"="S.thermophilus"))
all_colours <- (c("#fec3b7ff","#0081a7ff","grey70","grey88") ) 
densityPlot <- ggplot(analysis_FASTANI_together,aes(x=ANI,color=species1,fill=species1))+geom_density(alpha=0.4)+theme_classic()+lims(x=c(98.5,100))+geom_vline(xintercept = 99.75)+scale_fill_manual(values=all_colours)+scale_color_manual(values=all_colours)+theme(legend.position = "none")+labs(x="ANI",y="density")
densityPlot
colorsss <- c("grey77","orange")
plotANI <- ggplot(analysis_FASTANI_together,aes(x=clustercomparison,y=ANI,fill=clustercomparison))+geom_boxplot()+theme_classic()+lims(y=c(98.5,100))+stat_compare_means(label = "p.signif")+labs(x="",y="pairwise ANI")+facet_wrap(~species1,ncol = 1, strip.position="right")+scale_fill_manual(values = colorsss)+theme(legend.title = element_blank(),legend.position = "top",axis.title = element_blank(),axis.text = element_blank())+ coord_flip()
plotANI 

library(patchwork)
plotANI +densityPlot+ plot_layout(nrow=2,heights = c(1,2))
pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/ANI_genomes_boxplot.pdf",width=7,height=7)

plotANI +densityPlot+ plot_layout(nrow=2,heights = c(1,2))

dev.off()

```



##core genome nucleotide tree

Here, I creat a core genome nucleotide tree (inspired by Kirsten).
I also include the complete Reference genomes from NCBI. 

The following steps are incorporated:
1. pandaroo: find core genome
2. Extract gene sequences  corresponding to core gene famiilies into multi-fasta files:  2 files per gene-family, corresponding to nucleotide and protein sequences respectively. 
3. Align sequences for each family at protein level (resulting in one alignment file per core gene family) (MAFFT)
4. Back-translate to nucleotide alignments (respecting codon alignment) (Kirsten-script)
5. prune alignments (removing columns represented by less than 50% of the sequences) (Kirsten-script)
6. rename sequence-ids to genome-id only (Kirsten-script)
7. Concatenate pruned alignments (Kirsten-script)
8. infer the phylogeny (RAxML)

```{bash}
#get kirstens scripts
scp -r  /home/vincent/Downloads/aln_aa_to_dna vincent@130.223.49.145:/data/Project/2020_StarterCultureDiversity/99_log/


```


##### 1.Pandaroo

2. find core genomes
```{bash}
##==========================================
###-----------------bring all gffs together
##==========================================
species=Sterm

for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_GFF/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/
done

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/Sterm/Out-sali.gff
##==========================================
###-----------------panaroo
##==========================================

##-----sterm

species=Sterm

conda activate panaroo

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species}_references/01_Panarooo/
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo

panaroo -t 37 -i //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//*.gff -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species}_references/01_Panarooo/ --clean-mode strict

##-----ldel

species=Ldel

conda activate panaroo

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species}_references/01_Panarooo/
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species}_references/01_Panarooo

panaroo -t 37 -i //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//*.gff -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species}_references/01_Panarooo/ --clean-mode strict



##==========================================
###-----------------analysis panaroo
##==========================================


ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/GFF/ |grep ".gff" -c

cut -d ',' -f 4 /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|sort|uniq -c
awk -F "," '{if($4=="65" && $5=="65") print $0}' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|wc -l

##==========================================
###-----------------loop through single copy core genes and gather single copy core faa and ffn
##==========================================


roaryOUT=/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv

awk -F "," '{if($4=="65" && $5=="65") print $0}' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|wc -l

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG

for SCCG in $(awk -F "," '{if($4=="65" && $5=="65") print $1}' ${roaryOUT})
do
echo "================================================="
echo ${SCCG}
colzz=$(head -1 ${roaryOUT}  |tr ',' '\n' |wc -l)
for genomesss in $(seq 15 1 $colzz)
do

genesss=$(grep "^${SCCG}" ${roaryOUT} |cut -d ',' -f ${genomesss})
echo ${genesss}
samtools faidx /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/complete_Sterm_aminoAcid_gene.fasta ${genesss} >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG/${SCCG}.faa

samtools faidx /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/complete_Sterm_nucleotide_gene.fasta ${genesss} >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG/${SCCG}.ffn

done #genomesss
done #SCCG


```

```{bash}



mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/01_Panarooo//Sterm_references/

scp -r vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/01_Panarooo//Sterm_references/gene_presence_absence_roary.csv
mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/01_Panarooo//Ldel_references/
scp -r vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Ldel_references/01_Panarooo/gene_presence_absence_roary.csv /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/01_Panarooo//Ldel_references/gene_presence_absence_roary.csv

scp -r vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/pan_genome_reference.fa /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/01_Panarooo//Sterm_references/



```

###### lineage specific genes

```{r}
library(tidyverse)

##==================================
##Sterm
##==================================

library(tidyverse)

gene_presence_absence_roary <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/01_Panarooo/Sterm_references/gene_presence_absence_roary.csv")
analysis_clusters <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv",  ",", col_types = cols(Cluster = col_character(),  Taxon = col_character()))

dim(gene_presence_absence_roary)
gene_presence_absence_roary_long <- gene_presence_absence_roary[,c(1,15:ncol(gene_presence_absence_roary))] %>%gather(., sample, presence, colnames(gene_presence_absence_roary)[15]:colnames(gene_presence_absence_roary)[ncol(gene_presence_absence_roary)], factor_key=TRUE,na.rm = TRUE) %>% filter(!is.na(presence))
table(gene_presence_absence_roary_long$sample) %>% length()
gene_presence_absence_roary_long$sample <- as.character(gene_presence_absence_roary_long$sample)
lineageAbundance <- data.frame()
for (linss in unique(analysis_clusters$Cluster)) {
  
  print(linss)
  tmp <- analysis_clusters %>% filter(Cluster==linss) %>%  select(Taxon) %>% unlist() %>% as.character()
  tmp2 <- gene_presence_absence_roary_long %>% filter(sample %in% tmp) %>% select(Gene) %>% table() %>% data.frame()
  tmp3 <- data.frame(geneName=tmp2[,1],GenomesNum=tmp2[,2],lineage=as.character(linss))
  lineageAbundance <- rbind(lineageAbundance,tmp3)
  
}

lineageAbundance_wide <- spread(lineageAbundance, lineage, GenomesNum)  %>%replace(is.na(.), 0)

lineageAbundance$Presence_lin <- ifelse(lineageAbundance$GenomesNum>0,1,0)
lineageAbundance_presence_wide <- lineageAbundance %>% select(c(geneName,Presence_lin,lineage)) %>% spread(., lineage, Presence_lin)  %>%replace(is.na(.), 0)

lineageAbundance_presence_wide$NumberLineages <- rowSums(lineageAbundance_presence_wide[,-1])
table(lineageAbundance_presence_wide$NumberLineages)

lineageAbundance_presence_wide %>% filter(NumberLineages==1) %>%gather(., lineage, Presence_Abundance, colnames(lineageAbundance_presence_wide)[2]:colnames(lineageAbundance_presence_wide)[13], factor_key=TRUE,na.rm = TRUE) %>% filter(Presence_Abundance>0) %>% select(lineage) %>% table()

lineageAbundance_presence_unique_sterm <- lineageAbundance_presence_wide %>% filter(NumberLineages==1) %>%gather(., lineage, Presence_Abundance, colnames(lineageAbundance_presence_wide)[2]:colnames(lineageAbundance_presence_wide)[13], factor_key=TRUE,na.rm = TRUE) %>% filter(Presence_Abundance>0)

##how many cluster specific genes are there?
ggplot(lineageAbundance_presence_unique_sterm,aes(x=lineage))+geom_bar()+theme_classic()+labs(x="",y="number of cluster specific\ngenes")


##how many genome within this cluster contain the gene?
lineageAbundance_subset <- lineageAbundance %>% select(geneName,GenomesNum)
lineageAbundance_presence_count <- merge(lineageAbundance_presence_unique,lineageAbundance_subset,by="geneName")
table(lineageAbundance_presence_count$GenomesNum)


##---------------------------
#add eggnog information about the cluster-specific genes. 
##---------------------------

sterm_all_unique <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/01_Panarooo/Sterm_references/20210201_query_seqs.fa.emapper.annotations_all_pangenome",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description"),skip=4)
# RMK202_both_eggnog[RMK202_both_eggnog$query_name %in% SNPeff_final$finalName,]


COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA") %>% select(-divided)

unique(sterm_all_unique$COG_Functional_Category)[which(!unique(sterm_all_unique$COG_Functional_Category) %in% COG_functional_categories$short )]

# table(sterm_all_unique$COG_Functional_Category)

sterm_all_unique_extended <- merge(sterm_all_unique,COG_functional_categories,by.x = "COG_Functional_Category",by.y="short")


CogPlotSterm <- ggplot(sterm_all_unique_extended,aes(x=long))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10),
          legend.position = c(0.02, 0.98),
          legend.justification = c("left", "top")
          )
CogPlotSterm
#    svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/cogPlot.svg",width=7,height=7)
# CogPlotSterm 
#  dev.off()


##-------------------------
#add info
#with unnested
##-------------------------

lineageAbundance_presence_unique_sterm

lineageAbundance_presence_count_eggnog <- merge(lineageAbundance_presence_unique_sterm,sterm_all_unique,by.x="geneName",by.y="query_name",all.x=TRUE)

ALL_sterm <- lineageAbundance_presence_count_eggnog
nrow(ALL_sterm)

ALL_sterm <- ALL_sterm %>% 
    mutate(COG_Functional_Category = strsplit(as.character(COG_Functional_Category), "")) %>% 
    unnest(COG_Functional_Category)


lineageAbundance_presence_count_eggnog_COG <- merge(ALL_sterm,COG_functional_categories,by.x = "COG_Functional_Category",by.y="short")

ggplot(lineageAbundance_presence_count_eggnog_COG,aes(x=lineage,color=long,fill=long))+geom_bar()+theme_classic()+labs(x="",y="number of cluster specific\ngenes")


table(lineageAbundance_presence_count_eggnog_COG$long) %>% as.data.frame() %>% arrange(desc(Freq))
table(lineageAbundance_presence_count_eggnog_COG$tax_scope)  %>% as.data.frame() %>% arrange(desc(Freq))

tmp <- lineageAbundance_presence_count_eggnog_COG %>% filter(long=="Cell wall and membrane biogenesis")

tmp <- lineageAbundance_presence_count_eggnog_COG %>% filter(long=="Function Unknown")
tmp <- lineageAbundance_presence_count_eggnog_COG %>% filter(long=="Replication and repair")


lineageAbundance_presence_count_eggnog_COG



CogPlotSterm <- ggplot(lineageAbundance_presence_count_eggnog_COG,aes(x=forcats::fct_infreq(long)))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10),
          legend.position = c(0.02, 0.98),
          legend.justification = c("left", "top")
          )
CogPlotSterm
   png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/lineage_specific_genes.png", width = 2200, height = 1500,res=300)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

CogPlotSterm


dev.off()


write.table(lineageAbundance_presence_count_eggnog_COG,"/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/01_Panarooo/Sterm_references//CLUSTERspecificGENES_eggnog",sep="\t",quote = FALSE,row.names = FALSE)

###---------------------------------------
#group the genes appriopiately
##following catagroeies
#1. phage derived (do phaster and see which genes are associated to phage)
#2. all EPS/Rhamnos genes (group into phage receptor modification)
#3. add plasmid to phages
#4. bacertiocin
#interesting genes (mutL, mutS)
#new phage defense mechanisms
# conjugative elements, is-elements, transposase


###---------------------------------------

####---------01.glycosyl EPS related
lineageAbundance_presence_count_eggnog_COG%>% filter(grepl('glycosyl',`eggNOG free text description`,ignore.case = TRUE)|grepl('rham',`eggNOG free text description`,ignore.case = TRUE)|grepl('sugar transfera',`eggNOG free text description`,ignore.case = TRUE)|grepl('sugar transfera',`eggNOG free text description`,ignore.case = TRUE)|grepl('epimerase',`eggNOG free text description`,ignore.case = TRUE)) 

####---------02. plasmids, tranposons, is elements
lineageAbundance_presence_count_eggnog_COG%>% filter(grepl('plasmid',`eggNOG free text description`,ignore.case = TRUE)|grepl('Transposase',`eggNOG free text description`,ignore.case = TRUE)|grepl('family IS',`eggNOG free text description`,ignore.case = TRUE)|grepl('mobilisation',`eggNOG free text description`,ignore.case = TRUE))

####---------03. phage
lineageAbundance_presence_count_eggnog_COG%>% filter(grepl('phage',`eggNOG free text description`,ignore.case = TRUE)|grepl('integra',`eggNOG free text description`,ignore.case = TRUE)|grepl('viruses',`tax_scope`,ignore.case = TRUE)|grepl('reverse transcriptase',`eggNOG free text description`,ignore.case = TRUE))

####---------04. phage-defense
lineageAbundance_presence_count_eggnog_COG%>% filter(grepl('restriction-modification',`eggNOG free text description`,ignore.case = TRUE)|grepl('crispr',`eggNOG free text description`,ignore.case = TRUE)|grepl('endonuclease',`eggNOG free text description`,ignore.case = TRUE))

####---------05. bacteria-bacteria interaction
lineageAbundance_presence_count_eggnog_COG%>% filter(grepl('Bacteriocin',`eggNOG free text description`,ignore.case = TRUE)|grepl('antitoxin',`eggNOG free text description`,ignore.case = TRUE))

# lineageAbundance_presence_count_eggnog_COG %>% filter(grep("Glycosyl",ignore.case = TRUE))

##!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
##maybe we can show that many of the cluster-specific genes are associated to a plasmids, phages, conjugative elements as vectore
##!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!


lineageAbundance_presence_count_genes_location <- merge(lineageAbundance_presence_count,gene_presence_absence_roary_long,by.x = "geneName",by.y ="Gene",all.x=TRUE)


lineageAbundance_presence_count_genes_location <- lineageAbundance_presence_count_genes_location %>% 
    mutate(presence = strsplit(as.character(presence), ";")) %>% 
    unnest(presence) %>%  mutate(presence_location = strsplit(as.character(presence), "_")%>% map_chr(., 2))

lineageAbundance_presence_count_genes_location$presence_location <- as.numeric(lineageAbundance_presence_count_genes_location$presence_location)

lineageAbundance_presence_count_genes_location <- lineageAbundance_presence_count_genes_location %>% filter(!is.na(presence_location))

max(lineageAbundance_presence_count_genes_location$presence_location)


ggplot(lineageAbundance_presence_count_genes_location,aes(x=presence_location,group=sample,color=sample,fill=sample))+geom_density(alpha=0.4)+facet_wrap(~lineage,scales="free_y")+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))


##In ONT genomes

lineageAbundance_presence_count_genes_location %>% select(geneName) %>% unique() %>% nrow()
lineageAbundance_presence_count_genes_location  %>% filter(sample %in% c("24750","24739","13496","24738","13499","24730","24726","24728","13498","24734","24737","24853","24854","24855","24742","13494","24735","24736","24724","24748","24748","24740")) %>% select(geneName) %>% unique() %>% nrow()
# lineageAbundance_presence_count_genes_location  %>% filter(!sample %in% c("24750","24739","13496","24738","13499","24730","24726","24728","13498","24734","24737","24853","24854","24855","24742","13494","24735","24736","24724","24748","24748","24740")) %>% select(geneName) %>% unique() %>% nrow()

# lineageAbundance_presence_count_genes_location %>% filter(sample %in% c("24750","24739","13496","24738","13499","24730","24726","24728","13498","24734","24737","24853","24854","24855","24742","13494","24735","24736","24724","24748","24748","24740")) %>% select(GenomesNum) %>% table()
# lineageAbundance_presence_count_genes_location %>% filter(!sample %in% c("24750","24739","13496","24738","13499","24730","24726","24728","13498","24734","24737","24853","24854","24855","24742","13494","24735","24736","24724","24748","24748","24740")) %>% select(GenomesNum) %>% table()


lineageAbundance_presence_count_genes_location_ONT <- lineageAbundance_presence_count_genes_location %>% filter(sample %in% c("24750","24739","13496","24738","13499","24730","24726","24728","13498","24734","24737","24853","24854","24855","24742","13494","24735","24736","24724","24748","24748","24740"))

ggplot(lineageAbundance_presence_count_genes_location_ONT,aes(x=presence_location,group=sample,color=sample,fill=sample))+geom_density(alpha=0.4)+facet_wrap(~lineage,scales="free_y")+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))

lineageAbundance_presence_count_genes_location_ONT$lineageNUM <- as.numeric(lineageAbundance_presence_count_genes_location_ONT$lineage)


pointsPlot <- ggplot(lineageAbundance_presence_count_genes_location_ONT,aes(x=presence_location,y=lineageNUM,group=sample,color=sample,fill=sample))+geom_point()+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))
pointsPlot

###--------------------add information (e.g. COg or description to identify flanking transposons)

 lineageAbundance_presence_count_eggnog_COG_sub <- lineageAbundance_presence_count_eggnog_COG %>% select(c(geneName,`eggNOG free text description`, long,tax_scope))

lineageAbundance_presence_count_genes_location_ONT_EGGNOG <- merge( lineageAbundance_presence_count_genes_location_ONT,lineageAbundance_presence_count_eggnog_COG_sub,by="geneName")



pointsPlot <- ggplot(lineageAbundance_presence_count_genes_location_ONT_EGGNOG,aes(x=presence_location,y=lineageNUM,group=sample,color=sample,fill=sample,text =paste("COG:", long,"\neggog:",`eggNOG free text description`)))+geom_point()+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))
pointsPlot

 library(plotly)
ggp <- ggplotly(pointsPlot)
 ggp
 
 
 
 
pointsPlot <- ggplot(lineageAbundance_presence_count_genes_location_ONT_EGGNOG,aes(x=presence_location,y=sample,group=sample,color=sample,fill=sample,text =paste("COG:", long,"\neggog:",`eggNOG free text description`)))+facet_wrap(~lineageNUM,scales="free_y")+geom_point()+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))
pointsPlot

 library(plotly)
ggp <- ggplotly(pointsPlot)
 ggp
 
 ###--------------------------------------
 ##see potential phage, plasmid or transposon associated genes
  ###--------------------------------------
 
 
 
lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended <- lineageAbundance_presence_count_genes_location_ONT_EGGNOG%>% mutate(associated= ifelse(grepl('plasmid',`eggNOG free text description`,ignore.case = TRUE)|grepl('Transposase',`eggNOG free text description`,ignore.case = TRUE)|grepl('family IS',`eggNOG free text description`,ignore.case = TRUE)|grepl('mobilisation',`eggNOG free text description`,ignore.case = TRUE)|grepl('phage',`eggNOG free text description`,ignore.case = TRUE)|grepl('integra',`eggNOG free text description`,ignore.case = TRUE)|grepl('viruses',`tax_scope`,ignore.case = TRUE)|grepl('reverse transcriptase',`eggNOG free text description`,ignore.case = TRUE),"mechanisms known","unknown"))

 
pointsPlot <- ggplot(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended,aes(x=presence_location,y=sample,group=sample,color=associated,fill=associated,text =paste("COG:", long,"\neggog:",`eggNOG free text description`)))+facet_wrap(~lineageNUM,scales="free_y")+geom_jitter(width = 0, height = 0.3,alpha=0.8,)+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))
pointsPlot

 library(plotly)
ggp <- ggplotly(pointsPlot)
 ggp

  ###--------------------------------------
 ##identify gene clusters
 #so consecutive genes
  ###--------------------------------------
 CutOFFnumGENESS=15
 lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added <- data.frame()
 
 for (samplesss in unique(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended$sample)){
   
   tmp_01 <- lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended %>% filter(sample==samplesss) %>% arrange(presence_location)
   ClusterNUmber <- as.numeric(1)
 
   # for (genessss in unique(tmp_01$geneName)){
   for (genessss in 1:nrow(tmp_01)){

   tmp_02 <- tmp_01[genessss,] %>% arrange(presence_location)

   
  Gene_loc <-  tmp_01[genessss,"presence_location"]
  NextGene_loc <-  tmp_01[genessss+1,"presence_location"]
  Gene_div <-  NextGene_loc-Gene_loc
  if (!is.na(Gene_div)&&Gene_div<=CutOFFnumGENESS) {
    tmp3 <- cbind(tmp_02,Cluster=ClusterNUmber,NextGene=Gene_div)
    lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added <- rbind(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added,tmp3)
  } else if(!is.na(Gene_div)&&Gene_div>CutOFFnumGENESS) {
     tmp3 <- cbind(tmp_02,Cluster=ClusterNUmber,NextGene=Gene_div)
    lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added <- rbind(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added,tmp3)
    ClusterNUmber=as.numeric(ClusterNUmber+1)
   } else if(is.na(Gene_div)){
     tmp3 <- cbind(tmp_02,Cluster=ClusterNUmber,NextGene=Gene_div)
    lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added <- rbind(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added,tmp3)
     
   }
   
 }
 }
 
 table(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$NextGene)
  table(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$sample,lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Cluster)
lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Cluster <- as.character(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Cluster)
##color according to cluster
  

pointsPlot <- ggplot(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added,aes(x=presence_location,y=sample,group=Cluster,color=Cluster,fill=Cluster,text =paste("COG:", long,"\neggog:",`eggNOG free text description`)))+facet_wrap(~lineageNUM,scales="free_y")+geom_jitter(width = 0, height = 0.3,alpha=1,size=1.5)+lims(x=c(0,max(lineageAbundance_presence_count_genes_location$presence_location)))
pointsPlot

 library(plotly)
ggp <- ggplotly(pointsPlot)
 ggp  
 

 lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Clustername <- paste0(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$sample,"_cluster_",lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Cluster)
  
 analysis_clusters <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv",  ",",  col_types = cols(Taxon = col_character(),Cluster = col_character()))

 
 lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added_cluster <- table(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Clustername) %>% as.data.frame() %>%  mutate(sample = strsplit(as.character(Var1), "_")%>% map_chr(., 1))
 
 lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added_cluster <- merge(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added_cluster,analysis_clusters,by.x="sample",by.y="Taxon")
 
 ggplot(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added_cluster,aes(x=Var1,y=Freq))+geom_bar(stat="identity")+theme_classic()+facet_wrap(~Cluster,scales="free_x")
 
##--------------------------------------
### extract cluster extend
##--------------------------------------
extension=2
 ClusterDescription <- data.frame()
  for (clustersss in unique(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added$Clustername)) {
    
    tmp1 <- lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added %>% filter(Clustername==clustersss)
    tmp2 <- data.frame(clusterName=unique(tmp1$Clustername),Genome=unique(tmp1$sample),Lineage=unique(tmp1$lineage),NumGenesLineage=length(unique(tmp1$presence)),NumGenesTotal=max(tmp1$presence_location)-min(tmp1$presence_location)+1,startGene=min(tmp1$presence_location),endGene=max(tmp1$presence_location),startExt=min(tmp1$presence_location)-extension,endExt=max(tmp1$presence_location)+extension)
    ClusterDescription <- rbind(ClusterDescription,tmp2)
  }
 
 ##reformat the gene-names
 
ClusterDescription$startGene <- paste0(ClusterDescription$Genome,"_",formatC(ClusterDescription$startGene, width = 5, format = "d", flag = "0"))
ClusterDescription$endGene <- paste0(ClusterDescription$Genome,"_",formatC(ClusterDescription$endGene, width = 5, format = "d", flag = "0"))
ClusterDescription$startExt <- paste0(ClusterDescription$Genome,"_",formatC(ClusterDescription$startExt, width = 5, format = "d", flag = "0"))
ClusterDescription$endExt <- paste0(ClusterDescription$Genome,"_",formatC(ClusterDescription$endExt, width = 5, format = "d", flag = "0"))

write.table(ClusterDescription,"/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/cluster_lineageSpecific.csv",sep="\t",quote = FALSE,row.names = FALSE)


```

Here, I have identified clusters of lineage specific genes. Some of these clusters have phage related genes in them. 
We further identified and extracted cluster regions (+/- 2genes=maybe flanking transposons). 

Next I want to:

1. see which clusters are shared amongst strains (with cdHit)
2. analyse potential mechanisms of transposition (identify: prophage, plasmid, transposon)


```{bash}
scp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/cluster_lineageSpecific.csv vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/


```

```{bash}
##==================================================
##1. see which clusters are shared amongst strains (with cdHit)
##==================================================
clustersss=24855_cluster_1

###First extract fasta of clusters
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.bed
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.fasta
for clustersss in $(cut -f 1 /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/cluster_lineageSpecific.csv |sed '1d')
do
echo ${clustersss}

Genome=$(echo ${clustersss} |cut -d '_' -f 1)
startGEne=$(grep "${clustersss}$(printf '\t')" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/cluster_lineageSpecific.csv |cut -f 6)
EndGene=$(grep "${clustersss}$(printf '\t')" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/cluster_lineageSpecific.csv |cut -f 7)




contigName=$(grep "ID=${startGEne}_gene" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//${Genome}.gff |cut -f 1)
startName=$(grep "ID=${startGEne}_gene" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//${Genome}.gff |cut -f 4)
endName=$(grep "ID=${EndGene}_gene" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//${Genome}.gff |cut -f 5)

echo -e ${contigName}"\t"${startName}"\t"${endName}"\t"${clustersss} > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.bed
cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.bed >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.bed


##---------------get fasta

bedtools getfasta -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/${Genome}.fna \
  -bed /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.bed -name \
  -fo /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.fasta

cat  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.fasta >>  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.fasta

rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.fasta
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_tmp.bed
done

sed -i "s/\:.*$//g"  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.fasta
seq_length.py /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.fasta


### cluster with cd-hit-est


cd-hit-est -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters.fasta \
    -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_cd_hit -c 0.98 -M 30000 -T 37 -s 0.8 -aS 0.8 -sc 1 -sf 1
    
grep ">Cluster" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_cd_hit.clstr |sed 's/ /_/g' >  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_cd_hit.log
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecific_clustering.txt
for clustersss in $(cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_cd_hit.log)
do
#aryName=$(grep "${clustersss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log |cut -f 2)
ClusterNumber=$(echo $clustersss|awk -F "_" '{print $2}')
EndCluster=$((ClusterNumber+1))

sed -n "/>Cluster $ClusterNumber$/,/>Cluster $EndCluster$/p" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecificClusters_cd_hit.clstr | grep -v "^>Cluster" |awk -F "[>...]" -v numbersss="$EndCluster" '{OFS="\t"}{print "lineage cluster "numbersss,$2}' >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecific_clustering.txt

#for samplesss in $(cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt)
num=1

done

```

```{bash}
scp  vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/lineageSpecific_clustering.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/

```

```{r}

lineageSpecific_clustering <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/lineageSpecific_clustering.txt",  "\t", escape_double = FALSE, col_names = c("ClusterName_overall","clusterName"),  trim_ws = TRUE)

ClusterDescription_sub <- ClusterDescription %>% select(-c(startGene,endGene,startExt,endExt))

ClusterDescription_wide <- merge(ClusterDescription_sub,lineageSpecific_clustering,by="clusterName")
repeated_clusters <- data.frame()
for (variablesss in unique(ClusterDescription_wide$ClusterName_overall)) {
  
  tmp1 <- ClusterDescription_wide %>% filter(ClusterName_overall==variablesss)
  tmp2 <- ifelse(nrow(tmp1)!=length(unique(tmp1$Genome)),"repeated","not repeated")
  tmp4 <- max(tmp1$NumGenesLineage)
  tmp3 <- data.frame(ClusterName_overall=variablesss,repeatStructure=tmp2,maxNumGenesCluster=tmp4)
  repeated_clusters <- rbind(repeated_clusters,tmp3)
}


cutoffNumberGENES<-3

ClusterDescription_wider <- merge(ClusterDescription_wide,repeated_clusters,by="ClusterName_overall") %>%  mutate(Lineage =as.character(Lineage))
nums <- length(unique(ClusterDescription_wider[ClusterDescription_wider$maxNumGenesCluster<=cutoffNumberGENES,"ClusterName_overall"]))

ClusterDescription_wider[ClusterDescription_wider$maxNumGenesCluster<=cutoffNumberGENES,"ClusterName_overall"] <- paste0("small clusters (n=",nums,")")

ClusterDescription_wider[ClusterDescription_wider$maxNumGenesCluster<=cutoffNumberGENES,"Lineage"] <- paste0("small clusters (n=",nums,")")

# ggplot(ClusterDescription_wider_sub,aes(x=ClusterName_overall,y=NumGenesLineage,fill=Lineage))+geom_boxplot()+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=10))


ClusterDescription_summs <- ClusterDescription_wider  %>% group_by(ClusterName_overall,Lineage) %>%    dplyr::summarize(mean = mean(NumGenesLineage),sd=sd(NumGenesLineage))


ggplot(ClusterDescription_summs, aes(x=reorder(ClusterName_overall, -mean), y=mean, fill=Lineage)) +   
  geom_bar(stat="identity", color="black",  position=position_dodge()) +theme_classic() +
  # geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.2,position=position_dodge(.9)) +
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=10))+
  labs(x="",y="number of genes in\nlinegae specific cluster")
  

ClusterDescription_wider %>% filter(ClusterName_overall!="small clusters (n=37)")

# lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added

ClusterDescription_wider_ext <- merge(ClusterDescription_wide,repeated_clusters,by="ClusterName_overall") %>%  mutate(Lineage =as.character(Lineage))

nums <- length(unique(ClusterDescription_wider_ext[ClusterDescription_wider_ext$maxNumGenesCluster>cutoffNumberGENES,"ClusterName_overall"]))
ClusterDescription_wider_ext[ClusterDescription_wider_ext$maxNumGenesCluster>cutoffNumberGENES,"ClusterName_overall"] <- paste0("large clusters (n=",nums,")")
nums <- length(unique(ClusterDescription_wider_ext[ClusterDescription_wider_ext$maxNumGenesCluster<=cutoffNumberGENES,"ClusterName_overall"]))
ClusterDescription_wider_ext[ClusterDescription_wider_ext$maxNumGenesCluster<=cutoffNumberGENES,"ClusterName_overall"] <- paste0("small clusters (n=",nums,")")




tot_obs <- sum(ClusterDescription_wider_ext$NumGenesLineage)
ClusterDescription_wider_ext %>%   mutate(perc = n / nrow(tips)) 

ggplot(ClusterDescription_wider_ext, aes(fill=ClusterName_overall, y=NumGenesLineage, x=ClusterName_overall)) + 
    geom_bar(position="stack", stat="identity")+theme_classic()+labs(x="",y="number of lineage specif genes")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=10),legend.position = "none")+
  scale_y_continuous(sec.axis = sec_axis(trans = ~./tot_obs, labels = percent, name = "percent")) 


ClusterDescription_wider_ext <- merge(ClusterDescription_wide,repeated_clusters,by="ClusterName_overall") %>%  mutate(Lineage =as.character(Lineage))

numslarge <- length(unique(ClusterDescription_wider_ext[ClusterDescription_wider_ext$maxNumGenesCluster>cutoffNumberGENES,"ClusterName_overall"]))
numssmall <- length(unique(ClusterDescription_wider_ext[ClusterDescription_wider_ext$maxNumGenesCluster<=cutoffNumberGENES,"ClusterName_overall"]))

ClusterDescription_wider_ext$grouping <- ifelse(ClusterDescription_wider_ext$maxNumGenesCluster<=cutoffNumberGENES,paste0("small clusters (n=",numssmall,")"),paste0("large clusters (n=",numslarge,")"))

##---------------------
###-------------------------what genes are contained within small clusters
##---------------------

ClusterDescription_wider_ext_EGGNOG <- merge(lineageAbundance_presence_count_genes_location_ONT_EGGNOG_extended_added,ClusterDescription_wider_ext,by.x="Clustername",by.y="clusterName") %>% filter(!is.na(long))

table(ClusterDescription_wider_ext_EGGNOG$grouping)
ClusterDescription_wider_ext_EGGNOG_small <- ClusterDescription_wider_ext_EGGNOG %>% filter(grouping=="small clusters (n=37)")


ggplot(ClusterDescription_wider_ext_EGGNOG_small,aes(x=forcats::fct_infreq(long)))+geom_histogram(stat = "count")+theme_classic()+
  labs(x="",y="gene count")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=10))

write.table(ClusterDescription_wider_ext,"~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun//clusters_lineage_grouping.txt",sep="\t",quote = FALSE,row.names = FALSE)


```
keine klare Gene ersichtlich 

schauen was für functionen die restlichen clusters haben könnten. 


```{bash}
scp ~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun//clusters_lineage_grouping.txt vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/

```

2. analyse potential mechanisms of transposition (identify: prophage, plasmid, transposon)
herefore I focus only on the larger clusters (more the 3 lineage specific genes). I take two up and downstream genes and look for the presence of Is-elements or phage genes. 

```{bash}

##==================================================
##1. extract only larger clusters
##==================================================

clustersss=24735_cluster_12
species=Sterm
rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/
for clustersss in $(awk -F "\t" '{OFS="\t"}{if($9~"large clusters") print $2 }' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/clusters_lineage_grouping.txt)
do
echo $clustersss

Genome=$(echo ${clustersss} |cut -d '_' -f 1)
startGEne=$(awk -F "\t" -v clusterzzz="$clustersss" '{OFS="\t"}{if($1==clusterzzz) print $8 }' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/cluster_lineageSpecific.csv)
EndGene=$(awk -F "\t" -v clusterzzz="$clustersss" '{OFS="\t"}{if($1==clusterzzz) print $9 }' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/cluster_lineageSpecific.csv)

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//${Genome}.gff |grep -A 1000 "ID=${startGEne};" |grep -B 1000 "ID=${EndGene};" > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/${clustersss}.gff

#awk -F "\t" '{if($3=="CDS")print $0}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//${Genome}.gff |grep -A 1000 "ID=${startGEne};" |grep -B 1000 "ID=${EndGene};" > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/${clustersss}.gff



done


grep -i "ISfinder" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/*.gff
grep -i "ISfinder" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/*.gff|cut -d ':' -f 1 |sort|uniq -c
grep -i "ISfinder" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/*.gff|cut -d ':' -f 1 |sort|uniq -c |wc -l

grep "integr" -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/*.gff
grep "phage" -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/ClusterGenes/*.gff

```

###### Gene rarefaction

```{r}

library(tidyverse)

gene_presence_absence_roary <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/01_Panarooo/Sterm_references/gene_presence_absence_roary.csv")
analysis_clusters <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv",  ",", col_types = cols(Cluster = col_character(),  Taxon = col_character()))


###----------------------------------
##over all
###----------------------------------

library(micropan)
typeof(xmpl.panmat)
##creat a clustering matrix
gene_presence_absence_roary_preped <- gene_presence_absence_roary[,-c(2:14)] %>% remove_rownames()%>% column_to_rownames(., var = "Gene") %>% t() 
# dim(gene_presence_absence_roary_preped)
# gene_presence_absence_roary_preped_na <- is.na(gene_presence_absence_roary_preped)
# sum(!is.na(gene_presence_absence_roary_preped))

gene_presence_absence_roary_prep_final <- as.data.frame(!is.na(gene_presence_absence_roary_preped))
# dim(gene_presence_absence_roary_prep_final)
# gene_presence_absence_roary_prep_final[1,1]
# gene_presence_absence_roary_prep_final[2,2]
# str(gene_presence_absence_roary_prep_final)
gene_presence_absence_roary_prep_final <- as.data.frame(lapply(gene_presence_absence_roary_prep_final,as.numeric))
# str(gene_presence_absence_roary_prep_final)
# 
# gene_presence_absence_roary_preped[1,1]
# gene_presence_absence_roary_prep_final[1,1]
# gene_presence_absence_roary_prep_final[2,2]
rar.tbl <- rarefaction(gene_presence_absence_roary_prep_final, n.perm = 300)
data_genome_all <- rar.tbl %>%   gather(key = "Permutation", value = "Clusters", -Genome) %>% add_column(lineage="all_genomes")

# ggplot(data_genome,aes(x = Genome, y = Clusters))+
     # stat_smooth(method = "loess", formula = y ~ x, size = 1,se=TRUE)
# data_genome_all_cleaned <- data_genome_all %>% filter(!Genome %in% c(1,2,3,4,5))


# m<-nls(data_genome_all_cleaned$Clusters~a*data_genome_all_cleaned$Genome/(b+data_genome_all_cleaned$Genome))
# m

Rarefaction_all <- ggplot(data_genome_all,aes(x = Genome, y = Clusters))+
     geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c( F = 10,s=2000 )), se=FALSE)+theme_classic()+
    labs(x="Genomes",y="Genes")
Rarefaction_all

###----------------------------------
##lineages
###----------------------------------


rarefaction_lineages <- data.frame()
for (linss in unique(analysis_clusters$Cluster)) {
  
  print(linss)
  if (analysis_clusters[which(analysis_clusters$Cluster==linss),"Taxon"] %>% nrow()>1 ) {
    
    
  filtering <- analysis_clusters[which(analysis_clusters$Cluster==linss),"Taxon"] %>% unlist() %>% as.character()
length(filtering)
gene_presence_absence_roary_preped_lin1 <- as.data.frame(gene_presence_absence_roary_preped[rownames(gene_presence_absence_roary_preped)%in% filtering,])
dim(gene_presence_absence_roary_preped_lin1)

gene_presence_absence_roary_preped_lin1_final <- as.data.frame(!is.na(gene_presence_absence_roary_preped_lin1))
dim(gene_presence_absence_roary_preped_lin1_final)
gene_presence_absence_roary_preped_lin1_final <- as.data.frame(lapply(gene_presence_absence_roary_preped_lin1_final,as.numeric))
dim(gene_presence_absence_roary_preped_lin1_final)
rar.tbl_lin <- rarefaction(gene_presence_absence_roary_preped_lin1_final, n.perm = 50)
data_genome <- rar.tbl_lin %>%   gather(key = "Permutation", value = "Clusters", -Genome) %>% add_column(lineage=as.character(linss))

rarefaction_lineages <- rbind(rarefaction_lineages,data_genome)
    
  } else { }
  
}

g0<-ggplot(data=rarefaction_lineages,aes(x=Genome,y=Clusters,group=lineage))
# g1<-g0+geom_point()
g2<-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE)+theme_classic()+ labs(x="Genomes",y="Genes")+ coord_cartesian(ylim=c(1600, 2100))
g2
# ggp <- ggplotly(g2)
#  ggp

 maxss <- rarefaction_lineages %>% group_by(lineage) %>% summarise(Genomess=max(Genome),genesss=max(Clusters))
 maxss$lineage <- paste0("lin ",maxss$lineage)
 
rarefaction_lineages_plot <-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE,color="grey44")+theme_classic()+ labs(x="Genomes",y="Genes")+ coord_cartesian(ylim=c(1600, 2100))+geom_text(aes(x = Genomess, y = genesss, label =lineage), data = maxss)
rarefaction_lineages_plot

rarefaction_lineages_sterm <- rarefaction_lineages
# g2<-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE)+lims(y=c(1500,2200))
# g2

# g2<-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE)+coord_cartesian(ylim=c(1500,2200))
# g2
# g2<-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE)+coord_trans(x="log2")
# g2
###----------------------------------
##all together
###----------------------------------


rarefaction_alltogether <- rbind(rarefaction_lineages,data_genome_all)
g0<-ggplot(data=rarefaction_alltogether,aes(x=Genome,y=Clusters,group=lineage))
# g1<-g0+geom_point()
g2<-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE)+theme_classic()+ labs(x="Genomes",y="Genes") +coord_cartesian(ylim=c(1600,2550))
g2

# g2<-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE)+theme_classic()+ labs(x="Genomes",y="Genes") +coord_cartesian(ylim=c(1600,2550),xlim=c(0,15))
# g2

library(gg.gap)
gg.gap(plot=g2,segments=c(2100,2400),ylim = c(1600,2550))

  library(plotly)
ggp <- ggplotly(g2)
 ggp
 
#=================================== 
##-------------------------------------------------
#Ldel
##-------------------------------------------------
#=================================== 
 
gene_presence_absence_roary <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/01_Panarooo/Ldel_references/gene_presence_absence_roary.csv")
analysis_clusters <- read_delim("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_ldel.csv",  ",", col_types = cols(Cluster = col_character(),  Taxon = col_character()))


###----------------------------------
##over all
###----------------------------------

library(micropan)
typeof(xmpl.panmat)
##creat a clustering matrix
gene_presence_absence_roary_preped <- gene_presence_absence_roary[,-c(2:14)] %>% remove_rownames()%>% column_to_rownames(., var = "Gene") %>% t() 
# dim(gene_presence_absence_roary_preped)
# gene_presence_absence_roary_preped_na <- is.na(gene_presence_absence_roary_preped)
# sum(!is.na(gene_presence_absence_roary_preped))

gene_presence_absence_roary_prep_final <- as.data.frame(!is.na(gene_presence_absence_roary_preped))
# dim(gene_presence_absence_roary_prep_final)
# gene_presence_absence_roary_prep_final[1,1]
# gene_presence_absence_roary_prep_final[2,2]
# str(gene_presence_absence_roary_prep_final)
gene_presence_absence_roary_prep_final <- as.data.frame(lapply(gene_presence_absence_roary_prep_final,as.numeric))
# str(gene_presence_absence_roary_prep_final)
# 
# gene_presence_absence_roary_preped[1,1]
# gene_presence_absence_roary_prep_final[1,1]
# gene_presence_absence_roary_prep_final[2,2]
rar.tbl <- rarefaction(gene_presence_absence_roary_prep_final, n.perm = 300)
data_genome_all <- rar.tbl %>%   gather(key = "Permutation", value = "Clusters", -Genome) %>% add_column(lineage="all_genomes")

# ggplot(data_genome,aes(x = Genome, y = Clusters))+
     # stat_smooth(method = "loess", formula = y ~ x, size = 1,se=TRUE)
# data_genome_all_cleaned <- data_genome_all %>% filter(!Genome %in% c(1,2,3,4,5))


# m<-nls(data_genome_all_cleaned$Clusters~a*data_genome_all_cleaned$Genome/(b+data_genome_all_cleaned$Genome))
# m

Rarefaction_all <- ggplot(data_genome_all,aes(x = Genome, y = Clusters))+
     geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c( F = 10,s=2000 )), se=FALSE)+theme_classic()+
    labs(x="Genomes",y="Genes")
Rarefaction_all

###----------------------------------
##lineages
###----------------------------------


rarefaction_lineages <- data.frame()
for (linss in unique(analysis_clusters$Cluster)) {
  
  print(linss)
  if (analysis_clusters[which(analysis_clusters$Cluster==linss),"Taxon"] %>% nrow()>1 ) {
    
    
  filtering <- analysis_clusters[which(analysis_clusters$Cluster==linss),"Taxon"] %>% unlist() %>% as.character()
length(filtering)
gene_presence_absence_roary_preped_lin1 <- as.data.frame(gene_presence_absence_roary_preped[rownames(gene_presence_absence_roary_preped)%in% filtering,])
dim(gene_presence_absence_roary_preped_lin1)

gene_presence_absence_roary_preped_lin1_final <- as.data.frame(!is.na(gene_presence_absence_roary_preped_lin1))
dim(gene_presence_absence_roary_preped_lin1_final)
gene_presence_absence_roary_preped_lin1_final <- as.data.frame(lapply(gene_presence_absence_roary_preped_lin1_final,as.numeric))
dim(gene_presence_absence_roary_preped_lin1_final)
rar.tbl_lin <- rarefaction(gene_presence_absence_roary_preped_lin1_final, n.perm = 50)
data_genome <- rar.tbl_lin %>%   gather(key = "Permutation", value = "Clusters", -Genome) %>% add_column(lineage=as.character(linss))

rarefaction_lineages <- rbind(rarefaction_lineages,data_genome)
    
  } else { }
  
}
rarefaction_lineages$lineage <- as.character(14)
g0<-ggplot(data=rarefaction_lineages,aes(x=Genome,y=Clusters,group=lineage))
# g1<-g0+geom_point()
g2<-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE)+theme_classic()+ labs(x="Genomes",y="Genes")+ coord_cartesian(ylim=c(1600, 2100))
g2
# ggp <- ggplotly(g2)
#  ggp

 maxss <- rarefaction_lineages %>% group_by(lineage) %>% summarise(Genomess=max(Genome),genesss=max(Clusters))
 maxss$lineage <- paste0("lin ",maxss$lineage)
 
rarefaction_lineages_plot <-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE,color="grey44")+theme_classic()+ labs(x="Genomes",y="Genes")+ coord_cartesian(ylim=c(1600, 2200))+geom_text(aes(x = Genomess, y = genesss, label =lineage), data = maxss)
rarefaction_lineages_plot
 

rarefaction_lineages_ldel <- rarefaction_lineages
 #=================================== 
##-------------------------------------------------
#both species together
##-------------------------------------------------
#=================================== 
rarefaction_lineages_together <-  rbind(rarefaction_lineages_sterm,rarefaction_lineages_ldel)
table(rarefaction_lineages_together$lineage)

g0<-ggplot(data=rarefaction_lineages_together,aes(x=Genome,y=Clusters,group=lineage))
# g1<-g0+geom_point()
g2<-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE)+theme_classic()+ labs(x="Genomes",y="Genes")+ coord_cartesian(ylim=c(1600, 2100))
g2
# ggp <- ggplotly(g2)
#  ggp

 maxss <- rarefaction_lineages_together %>% group_by(lineage) %>% summarise(Genomess=max(Genome),genesss=max(Clusters))
 maxss$lineage <- paste0("lin ",maxss$lineage)
 
rarefaction_lineages_plot <-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE,color="grey44")+theme_classic()+ labs(x="Genomes",y="Genes")+ coord_cartesian(ylim=c(1750, 2200))+geom_text(aes(x = Genomess, y = genesss, label =lineage), data = maxss)+geom_vline(xintercept=14, colour="grey55", linetype="dotted")+geom_hline(yintercept=2100, colour="grey55", linetype="dotted")
rarefaction_lineages_plot

##--------------------------zoom
g0<-ggplot(data=rarefaction_lineages_sterm,aes(x=Genome,y=Clusters,group=lineage))
# g1<-g0+geom_point()
g2<-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE)+theme_classic()+ labs(x="Genomes",y="Genes")+ coord_cartesian(ylim=c(1600, 2100))
g2
# ggp <- ggplotly(g2)
#  ggp

 maxss <- rarefaction_lineages_sterm %>% group_by(lineage) %>% summarise(Genomess=max(Genome),genesss=max(Clusters))
 maxss$lineage <- paste0("lin ",maxss$lineage)
rarefaction_lineages_plot_zoom <-g0+geom_smooth(method="nls",formula=y~s*x/(F+x),method.args=list(start = c(  F = 1,s=2000 )), se=FALSE,color="grey44")+theme_classic()+ labs(x="Genomes",y="Genes")+ coord_cartesian(ylim=c(1750, 2100),xlim=c(0,14))+geom_text(aes(x = Genomess, y = genesss, label =lineage), data = maxss)
rarefaction_lineages_plot_zoom
pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/rarafeaction_genomes.pdf",width=14,height=7)

rarefaction_lineages_plot+rarefaction_lineages_plot_zoom

dev.off()

```



##### 1.orthofinder



```{bash}
###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Sterm

for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

#leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
done

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/Out-sali.faa



###-----------------------------------------
##orthorfinder
###-----------------------------------------
#for species in $(echo "Ldel Sterm"|tail -1)
species=Ldel
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/
#cp -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/Ldel/OrthoFinder/Results_Feb11/ /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/

for species in $(echo "Sterm Ldel")
do

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/


    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}// \
      -t 20 \
      -o /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ \
      -a 20

done # species

###-----------------------------------------
##orthorfinder
###-----------------------------------------
#for species in $(echo "Ldel Sterm"|tail -1)
species=Ldel
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/
cp -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/Ldel/OrthoFinder/Results_Feb11/ /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/

for species in $(echo "Sterm Ldel")
do

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/


    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA// \
      -t 37 \
      -o /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ \
      -a 37

done # species


##===============================================
##all genomes together 
##for GeneRax
##===============================================

###-------------------add Bacillus subtilis as outgroup

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/
species=$(echo "Bacillus subtilis")

#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
#mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
#cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_BACSUB.txt
    #pecies_short=Sterm_references

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_BACSUB.txt

mv GCF_000769515.1_ASM76951v1_protein.faa.gz Out_Bacsub.faa.gz

gunzip Out_Bacsub.faa.gz

###-------------------make collection of all rmk strains
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/

###-------------------add all species close to Ldel and Sterm from figure 1D/E

cp  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/
cp /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/thermophilus.faa
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/delbrueckii_*


###-------------------add all strains close to Ldel and Sterm from figure 1F/G

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/PROKKA_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/
cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/PROKKA_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/


    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/ \
      -t 25 \
      -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/  \
      -a 25


```

```{bash}

mkdir -p  /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Apr12/Species_Tree/SpeciesTree_rooted_node_labels.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/
#sent back to bigmachine
scp /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/SpeciesTree_rooted_node_labels_rerooted.txt vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Apr12/Species_Tree/ 



##---Lhelv


scp -r vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Lhelv/Results_Mar16/Species_Tree/SpeciesTree_rooted_node_labels.txt ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_SpeciesTree_rooted_node_labels.txt


```


```{bash}
##========================================
#get OG sequences for eggnog
##========================================
#rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm_OG_genes.faa
for species in $(echo "Ldel")
do
rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}_OG_genes.faa

echo $species
#done
date=$(ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ |grep "Results")

for genesss in $(ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroup_Sequences|grep ".fa$" |sed 's/.fa//g')
do
echo $genesss

#perl -00 -ne '/(>[^>]+)/ && print $1' /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroup_Sequences/${genesss}.fa |sed "s/>.*$/>S_${genesss}/g" >> /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm_OG_genes.faa

awk '/^>/{if(N)exit;++N;} {print;}' /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroup_Sequences/${genesss}.fa |sed "s/>.*$/>L_${genesss}/g" >> /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}_OG_genes.faa


done #genesss
ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroup_Sequences/ |grep ".fa$" -c
grep ">" -c  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}_OG_genes.faa

done #species

grep ">"  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm_OG_genes.faa |head 
grep ">"  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Ldel_OG_genes.faa |head 

##========================================
#bring local for eggnog
##========================================

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/forEGGNOG_pangenome
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Ldel_OG_genes.faa /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/forEGGNOG_pangenome

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm_OG_genes.faa /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/forEGGNOG_pangenome/

###----------------------------------------
#extract only pseudogne OG
###----------------------------------------


#------------------------Lhelv
mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Lhelv

cut -d ',' -f 1 ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered.txt |sed '1d'  |sed 's/^O/Lh_O/g' > ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Lhelv/OG_ps.nmaes


cd ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder
xargs --arg-file=Lhelv/OG_ps.nmaes  samtools faidx ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_OG_genes.faa > ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Lhelv/OG_ps.faa


grep -c ">" ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_OG_genes.faa
grep -c ">" ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Lhelv/OG_ps.faa


#------------------------Sterm
mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Sterm

cut -d ',' -f 1 ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered.txt |sed '1d'  |sed 's/^O/St_O/g' > ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Sterm/OG_ps.nmaes


cd ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder
xargs --arg-file=Sterm/OG_ps.nmaes  samtools faidx ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_OG_genes.faa > ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Sterm/OG_ps.faa


grep -c ">" ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_OG_genes.faa
grep -c ">" ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Sterm/OG_ps.faa



#------------------------Ldel
mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Ldel

cut -d ',' -f 1 ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt |sed '1d'  |sed 's/^O/Ld_O/g' > ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Ldel/OG_ps.nmaes


cd ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder
xargs --arg-file=Ldel/OG_ps.nmaes  samtools faidx ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_OG_genes.faa > ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Ldel/OG_ps.faa


grep -c ">" ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_OG_genes.faa
grep -c ">" ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Ldel/OG_ps.faa



```


```{bash}

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_Ldel/
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/Ldel//OrthoFinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_node_labels.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_Ldel//SpeciesTree_rooted_node_labels_orthofinder_ldel_complete.txt

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_node_labels.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_Ldel//SpeciesTree_generax.txt

```

###### add MAG sterm

```{bash}

/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_and_MAG/Sterm/Results_Nov24/Species_Tree/SpeciesTree_rooted.txt

mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/bothspecies_MAG_out/Sterm/genomes

scp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/Sterm_Ldel_refs/X398_Eurasian_Reed_Warbler__metabat2_low_PE.003.contigs.fa vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/bothspecies_MAG_out/Sterm/genomes/

##-------------------------------PROKKA


grep ">" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/bothspecies_MAG_out/Sterm/genomes//X398_Eurasian_Reed_Warbler__metabat2_low_PE.003.contigs.fa
sed 's/X398_Eurasian_Reed_Warbler__coassemble/S_MAG_RW/g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/bothspecies_MAG_out/Sterm/genomes//X398_Eurasian_Reed_Warbler__metabat2_low_PE.003.contigs.fa >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/bothspecies_MAG_out/Sterm/genomes//X398_Eurasian_Reed_Warbler__metabat2_low_PE.003.contigs_shortName.fa 

conda activate PROKKA

prokka --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/bothspecies_MAG_out/Sterm/Prokka/Sterm_MAG_reed_warbler --force --usegenus --genus Streptococcus --locustag S_MAG_RW --proteins proteins.faa --evalue 0.001 --addgenes   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/bothspecies_MAG_out/Sterm/genomes//X398_Eurasian_Reed_Warbler__metabat2_low_PE.003.contigs_shortName.fa --cpus 35
  
##-----------------------------------


###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Sterm

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_and_MAG/FAA/${species}/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_and_MAG/FAA/${species}/

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/bothspecies_MAG_out/Sterm/Prokka/Sterm_MAG_reed_warbler/PROKKA_11242021.faa > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_and_MAG/FAA/${species}/S-MAG-RW.faa

###-----------------------------------------
##orthorfinder
###-----------------------------------------

species=Sterm

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_and_MAG/${species}
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_and_MAG/${species}


    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_and_MAG/FAA/${species}/ \
      -t 35 \
      -o /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_and_MAG/${species}/ \
      -a 35


```



```{bash}

#mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_Ldel/
#scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/Ldel//OrthoFinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_node_labels.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_Ldel//SpeciesTree_rooted_node_labels_orthofinder_ldel_complete.txt

scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_and_MAG/Sterm/Results_Nov24/Species_Tree/SpeciesTree_rooted_node_labels.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_Ldel//SpeciesTree_generax_02.txt

```

##### 1.Kirstin's approach


```{bash}
date=Results_Feb01
species=Sterm
#/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/Results_Jul28/
for species in $(echo "Ldel Sterm"|tail -1)
do
echo "==========================================="
echo -e ${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/get_singlecp_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt

wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt
wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt

done
```


Here, I make a directory containing the fna and faa files of all genomes with the file name corresponding to the locus tag. 

```{bash}


for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA/
done

###--------------------
for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/ |grep ".faa$"|sed 's/.faa//g'|grep -v "Out-sali")
do
locusTagsss=$(head -1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/${genomesss}.faa|cut -d '_' -f 1|sed 's/>//g')
echo -e "This  genome : " ${genomesss} " has the following locus Tag : "${locusTagsss}

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/${genomesss}.faa > \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/${genomesss}.ffn > \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.ffn


done
done

##------=================================================================
#extract orthogroups
##------=================================================================

for species in $(echo "Ldel Sterm"|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/extract_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt --folder ${species}

done
```

align protein families

```{bash}


for species in $(echo "Ldel Sterm"|tail -1)
do

cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep ".faa$" |sed 's/.faa//g'|grep -v "Out-sali" )
do
rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.aln.fasta
mafft --thread 37 --maxiterate 1000  --auto /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.faa > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}_aln.fasta

done 

done #species
```
here I back translate the faa alignment to dna.

```{bash}

for species in $(echo "Ldel Sterm"|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/aln_aa_to_dna.pl

done

```   

With perl: prune alignments (removing columns represented by less than 50% of the sequences)


```{bash}


for species in $(echo "Ldel Sterm"|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/prune_aln.pl

done

##================================================================
#remove all infomration (except genome info) from multifasta header
##================================================================

for species in $(echo "Ldel Sterm"|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i "s/_0.*//" ${genesss}_aln_prune.fasta 

done
done

##================================================================
#shorten header
##================================================================
species=Ldel
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/L-delbrueckii-//g' ${genesss}_aln_prune.fasta 

done



species=Sterm
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/-CIRM-//g' ${genesss}_aln_prune.fasta 

done

species=Sterm
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/-BM-A0/-BM-A/g' ${genesss}_aln_prune.fasta 

done


```
With perl: Concatenate pruned alignments
```{bash}
  
#  for species in $(echo "Ldel Sterm"|tail -1)
  for species in $(echo "Sterm"|tail -1)
  do
  echo -e ${species}
  cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
  perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/cat_align.pl > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}.phylip
  
  done
```
With RAxML: infer the phylogeny
```{bash}
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/

#  for species in $(echo "Ldel Sterm"|tail -1)
   for species in $(echo "Sterm"|tail -1)
  do
  echo -e ${species}
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/

rm -r *${species}_all*

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}.phylip -n ${species}_all -T 37
done
```

move local

```{bash}

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new/


scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new//
```

### ggtree

preperation

```{bash}
###-------------------------
#get RMK genomes and REF genomes
###-------------------------
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/

species=Sterm

leterzzz=$(echo $species |head -c 1)

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/ |grep "Out" -v | sed 's/.faa//g' |awk -F "\t" '{OFS="\t"}{print $1, "NCBI_refs"}' >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FAA/ |grep "^${leterzzz}" |grep ".faa$"|cut -d '_' -f 4|sed 's/.faa//g' |awk -F "\t" '{OFS="\t"}{print $1, "RMK_genomes"}' >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/ |grep "Out-ves" | sed 's/.faa//g' |awk -F "\t" '{OFS="\t"}{print $1, "outgroup"}' >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt


sed -i 's/-BM-A0/-BM-A/g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt
sed -i 's/S-KLDS-3.1/S-KLDS-3/g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt

grep "S-KLDS-3" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt

###=============================================
##Ldel
###=============================================

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/

species=Ldel

leterzzz=$(echo $species |head -c 1)

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/ |grep "Out" -v | sed 's/.faa//g' |awk -F "\t" '{OFS="\t"}{print $1, "NCBI_refs"}' >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Ldel_genome_origin.txt

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel/01_all_FAA/ |grep "^${leterzzz}" |grep ".faa$"|cut -d '_' -f 4|sed 's/.faa//g' |awk -F "\t" '{OFS="\t"}{print $1, "RMK_genomes"}' >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Ldel_genome_origin.txt

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/ |grep "Out-" | sed 's/.faa//g' |awk -F "\t" '{OFS="\t"}{print $1, "outgroup"}' >>  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Ldel_genome_origin.txt


sed -i 's/-BM-A0/-BM-A/g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Ldel_genome_origin.txt
sed -i 's/S-KLDS-3.1/S-KLDS-3/g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Ldel_genome_origin.txt

grep "S-KLDS-3" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Ldel_genome_origin.txt

```

move local
```{bash}
mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new//logs/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Sterm_genome_origin.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new//logs/



mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_Ldel//logs/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/log/Ldel_genome_origin.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_Ldel//logs/
```



do in R

first Sterm
```{r}
library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated", tree.names = NULL, force.multi = FALSE)


# x <- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
ggtree(Sterm_tree_raw, layout="circular") +
  geom_tiplab(aes(angle=angle),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  
##--------------------------------
##color labels of of only rmk phages
##--------------------------------
Sterm_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/logs/Sterm_genome_origin.txt",  "\t", escape_double = FALSE, col_names = c("label","origin"),  trim_ws = TRUE) %>% as.data.frame()



tmp <- full_join(as_tibble(Sterm_tree_raw), Sterm_genome_origin, by='label')
# table(tmp$origin)
tree <- tmp %>% as.treedata()


p <- ggtree(tree, layout="circular") +
  geom_tiplab(aes(angle=angle,color=origin,size=origin),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) + #NO ROOT  
  scale_color_manual(values=c("grey66", 'grey55', 'black')) +
  scale_size_manual(values=c(4,4,7))# +geom_treescale()
p
##--------------------------------
##all info at once (geography and origin)
##--------------------------------
all_sterm_strain_info <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited.csv", "\t", escape_double = FALSE, trim_ws = TRUE) %>% select(c(shortName, strain, origin, location_continent))

table(all_sterm_strain_info$origin)

all_sterm_strain_info$strain <- gsub("S-thermophilus-","",all_sterm_strain_info$strain)

##-------------
#change tiplabel and color according to rmk or non-rmk sample
##-------------

tmp = as.character(all_sterm_strain_info$strain)
names(tmp) <- all_sterm_strain_info$shortName

# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated", tree.names = NULL, force.multi = FALSE)



Sterm_tree_raw_tible <- as_tibble(tree) 
Sterm_tree_raw_tible$label

Sterm_tree_raw_tible$label <- plyr::revalue(Sterm_tree_raw_tible$label,tmp)
tree_02 <- Sterm_tree_raw_tible%>% as.treedata()




p2 <- ggtree(tree_02, layout="circular") +
  geom_tiplab(aes(angle=angle,color=origin,size=origin),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) + #NO ROOT  
  scale_color_manual(values=c("grey66", 'grey55', 'black')) +
  scale_size_manual(values=c(2.5,2.5,4)) 
p2

##-------------
#add geograph and origin
##-------------
all_sterm_strain_info$shortName <- as.character(all_sterm_strain_info$shortName)
all_sterm_strain_info_ready <- all_sterm_strain_info %>% select(-shortName) %>% remove_rownames()%>% column_to_rownames(., var = "strain")%>%replace(is.na(.), "Unknown") 
table(all_sterm_strain_info_ready$location_continent)
 all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c("Europe",  "Asia", "USA", "Australia" , "Unknown","Outgroup"))
# colors_area <- c("#464D77","#36827F","#F9DB6D","grey88")
# colors_area <- c("#F9DB6D","#36827F","#464D77","grey88")

# location_dataset$area

# tree_02 <- full_join(as_tibble(tree), location_dataset, by='label') %>% as.treedata()

colnames(all_sterm_strain_info_ready) <- c("A","B")
rownames(all_sterm_strain_info_ready)

p3 <- gheatmap(p2, all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
p3

library(svglite)
# svglite("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.svg",width=10,height=10)
  # svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.svg",width=9,height=9)
   # png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.png", width = 3000, height = 3000,res=300)
pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

p3

dev.off()

##---------------------
#prune tree
##---------------------

# all_sterm_strain_info_keepers <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited.csv", "\t", escape_double = FALSE, trim_ws = TRUE) %>% select(c( strain, keep)) %>% filter( keep=="yes") %>% select(strain) %>% unlist() %>% as.vector
# 
# all_sterm_strain_info_keepers <- gsub("S-thermophilus-","",all_sterm_strain_info_keepers)

#But get the tips to keep
# keeptips <- fulltree$tip.label[!fulltree$tip.label %in% prunetips]

#Group the tips to keep
# prunetree <- groupOTU(tree_02, focus=all_sterm_strain_info_keepers)

#And plot
# ggtree(prunetree, layout="fan", aes(color=group))+
# scale_color_manual(values=c("lightgrey","black"))+
# geom_tiplab()

# prunetree <- drop.tip(tree_02,node=5)

##--------------------------------------------
##make the main figure tree circular with highlights therefore identify the highlitable nodes with identify()
#geom_hilight() and identify()
##--------------------------------------------
cols <- rainbow(14) #I am not sure this is correct -- did not find the correct color labels anymore
pz <- ggtree(tree) +
  geom_tiplab(aes(color=origin,size=origin)) # tiplabls aligned
  # geom_rootedge(rootedge = 0) + #NO ROOT  
  # scale_color_manual(values=c("grey66", 'grey55', 'black')) +
  # scale_size_manual(values=c(4,4,8)) 
pz

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total_linear.pdf",width=10,height=10)

pz

dev.off()

library(plotly)
plotly::ggplotly(pz)


p4 <- p3+geom_hilight(c(53),fill=cols[1],extendto=0.14)+geom_hilight(c(155),fill=cols[2],extendto=0.14)+geom_hilight(c(190),fill=cols[3],extendto=0.14)+
  geom_hilight(c(206),fill=cols[4],extendto=0.14)+
  geom_hilight(c(203),fill=cols[5],extendto=0.14)+
  geom_hilight(c(198),fill=cols[6],extendto=0.14)+
  geom_hilight(c(210),fill=cols[7],extendto=0.14)+
  geom_hilight(c(216),fill=cols[8],extendto=0.14)+
  geom_hilight(c(100),fill=cols[9],extendto=0.14)+
  geom_hilight(c(129),fill=cols[10],extendto=0.14)+
  geom_hilight(c(128),fill=cols[11],extendto=0.14)+
  geom_hilight(c(11),fill=cols[12],extendto=0.14)+geom_treescale()
p4

##--------------------------------------------
#do a new circular tree with the cheese starer culture info
##--------------------------------------------

# 
# Sterm_clusters_RMKS <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv", col_names = c("label","cluster"), col_types = cols(cluster = col_character(),  label = col_character()), skip = 1)%>% remove_rownames()%>% column_to_rownames(., var = "label")%>%replace(is.na(.), "Unknown") 
# 
# 
# 
# p5_sterm <- gheatmap(p, Sterm_clusters_RMKS, offset=0.001, width=.05,colnames_angle=75, colnames_offset_y = 0)  #+  scale_fill_manual(values=colors_area)
# p5_sterm

all_sterm_strain_info_ZOOMED <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) %>% filter(species=="S.thermophilus") %>% select(c(strain,Cheese,starterCulture,Sequencing)) %>% remove_rownames()%>% column_to_rownames(., var = "strain")%>%replace(is.na(.), "Unknown") 

p3 <- gheatmap(p, all_sterm_strain_info_ZOOMED, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0)+ geom_treescale()#+  scale_fill_manual(values=colors_area)
p3


##----------------CHANGE COLORS
COLORSSS <- c("#d45400ff","#de592bff","#e36b42ff","#e67d59ff","#eb8f70ff","#eea38aff","#009cf0ff","#33cffaff","#7fe3ffff","#007f7fff","#00dcdcff","#ff0000ff","#0000ffff","grey","#fde623ff","#00ff00ff")
p4 <- gheatmap(p, all_sterm_strain_info_ZOOMED, offset=0.025, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)+  scale_fill_manual(values=COLORSSS)+ geom_treescale()
p4

p5 <- p4+geom_hilight(c(53),fill=cols[1],extendto=0.145)+geom_hilight(c(155),fill=cols[2],extendto=0.145)+geom_hilight(c(190),fill=cols[3],extendto=0.145)+
  geom_hilight(c(206),fill=cols[4],extendto=0.145)+
  geom_hilight(c(203),fill=cols[5],extendto=0.145)+
  geom_hilight(c(198),fill=cols[6],extendto=0.145)+
  geom_hilight(c(210),fill=cols[7],extendto=0.145)+
  geom_hilight(c(216),fill=cols[8],extendto=0.145)+
  geom_hilight(c(100),fill=cols[9],extendto=0.145)+
  geom_hilight(c(129),fill=cols[10],extendto=0.145)+
  geom_hilight(c(128),fill=cols[11],extendto=0.145)+
  geom_hilight(c(11),fill=cols[12],extendto=0.145)

# p5
#  library(plotly)
# ggp <- ggplotly(p5)
#  ggp

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total_circlar.pdf",width=15,height=15)

p5

dev.off()

###--------------------------------------
#remove certain tips
###--------------------------------------


cols <- rainbow(14) #I am not sure this is correct -- did not find the correct color labels anymore
pz <- ggtree(tree) +
  geom_tiplab(aes(color=origin,size=origin)) # tiplabls aligned
  # geom_rootedge(rootedge = 0) + #NO ROOT  
  # scale_color_manual(values=c("grey66", 'grey55', 'black')) +
  # scale_size_manual(values=c(4,4,8)) 
pz

tree <- full_join(as_tibble(tree), Sterm_genome_origin, by='label') %>% as.treedata()

tree$tip

tiplabels(as.phylo(tree))

drop.tip(tree, tip, trim.internal = TRUE, subtree = FALSE,
         root.edge = 0, rooted = is.rooted(phy), collapse.singles = TRUE,
         interactive = FALSE)


tree_sub <- drop.tip(tree, c("24736","24735"))
pz <- ggtree(tree_sub) +
  geom_tiplab(aes(color=origin,size=origin)) # tiplabls aligned
  # geom_rootedge(rootedge = 0) + #NO ROOT  
  # scale_color_manual(values=c("grey66", 'grey55', 'black')) +
  # scale_size_manual(values=c(4,4,8)) 
pz


tree_labels <- as_tibble(tree)$label[which(!is.na(as_tibble(tree)$label))]

tree_labels_remove <- tree_labels[!tree_labels %in% c("24736","24735","24734","24750","24739","24745","24728","24742","24746","24749","24730","24743")]

tree_sub <- drop.tip(tree,tree_labels_remove)
pz <- ggtree(tree_sub) +
  geom_tiplab(aes(color=origin,size=origin)) # tiplabls aligned
  # geom_rootedge(rootedge = 0) + #NO ROOT  
  # scale_color_manual(values=c("grey66", 'grey55', 'black')) +
  # scale_size_manual(values=c(4,4,8)) 
pz

##----------------reorder tree

 as_tibble(tree_sub)$label

 y <- ape::rotateConstr(as.phylo(tree_sub), c("24736","24735","24734","24750","24739","24745","24728","24742","24746","24749","24730","24743"))
 
pz <-ggtree(y, ladderize=F)+ geom_tiplab()

pz

##----------------no brnach length

pz <-ggtree(y, ladderize=F,branch.length="none")+ geom_tiplab()

pz

###------change tip labels


y_df <- as_tibble(y)

y_df$label <- plyr::revalue(y_df$label, c("24736"="lineage 1","24735"="lineage 2","24734"="lineage 3","24750"="lineage 4","24739"="lineage 5","24745"="lineage 6","24728"="lineage 7","24742"="lineage 8","24746"="lineage 9","24749"="lineage 10","24730"="lineage 11","24743"="lineage 12"))

 y_new <- as.treedata(y_df)
  y <- ape::rotateConstr(as.phylo(y_new), c("lineage 1","lineage 2","lineage 3","lineage 4","lineage 5","lineage 6","lineage 7","lineage 8","lineage 9","lineage 10","lineage 11","lineage 12"))

 
 pz <-ggtree(y, ladderize=F,branch.length="none")+ geom_tiplab()

pz

# tree <- full_join(as_tibble(y), Sterm_genome_origin, by='label') %>% as.treedata()


###------add rmk info

all_sterm_strain_info_ZOOMED <- read_csv("Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) %>% filter(species=="Sterm") %>% select(c(strain,Cheese)) %>% mutate(strain=gsub("S","",strain))%>% remove_rownames() #%>% column_to_rownames(., var = "strain")%>%replace(is.na(.), "Unknown") 

all_sterm_strain_info_ZOOMED$strain <- plyr::revalue(all_sterm_strain_info_ZOOMED$strain, c("24736"="lineage 1","24735"="lineage 2","24734"="lineage 3","24750"="lineage 4","24739"="lineage 5","24745"="lineage 6","24728"="lineage 7","24742"="lineage 8","24746"="lineage 9","24749"="lineage 10","24730"="lineage 11","24743"="lineage 12"))

all_sterm_strain_info_ZOOMED <- all_sterm_strain_info_ZOOMED%>% column_to_rownames(., var = "strain")%>%replace(is.na(.), "Unknown") 

COLORSSS <- c("#ff0000ff","#0000ffff","#808001ff")


p3 <- gheatmap(pz, all_sterm_strain_info_ZOOMED, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0)+ geom_treescale()#+  scale_fill_manual(values=colors_area)
p3


pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/schematic_sterm_lineages.pdf",width=8,height=8)

p3

dev.off()

```

Ldel

```{bash}
sed 's/KLDS1./KLDS1-/g' ~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_Ldel/SpeciesTree_rooted_node_labels_orthofinder_ldel_complete_rerooted.txt | sed "s/'//g" > ~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_Ldel/SpeciesTree_rooted_node_labels_orthofinder_ldel_complete_rerooted_cleaned.txt

sed 's/KLDS1./KLDS1-/g' ~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_Ldel/logs/Ldel_genome_origin.txt > ~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_Ldel/logs/Ldel_genome_origin_cleaned.txt

sed 's/KLDS1./KLDS1-/g' ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel_edited.csv | sed 's/^Lactobacillus-bulgaricus-/L-/g' | sed 's/^Lactobacillus-delbrueckii-subsp.-/L-/g'  > ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel_edited_cleaned.csv



```



```{r}
library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Ldel_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_Ldel/SpeciesTree_rooted_node_labels_orthofinder_ldel_complete_rerooted_cleaned.txt", tree.names = NULL, force.multi = FALSE)


# x <- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
ggtree(Ldel_tree_raw, layout="circular") +
  geom_tiplab(aes(angle=angle),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  
##--------------------------------
##color labels of of only rmk phages
##--------------------------------
Ldel_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_Ldel/logs/Ldel_genome_origin_cleaned.txt",  "\t", escape_double = FALSE, col_names = c("label","origin"),  trim_ws = TRUE) %>% as.data.frame()

# tmp <- full_join(as_tibble(Ldel_tree_raw), Ldel_genome_origin, by='label') 
# table(tmp$origin)
tree <- full_join(as_tibble(Ldel_tree_raw), Ldel_genome_origin, by='label') %>% as.treedata()


p <- ggtree(tree, layout="circular") +
  geom_tiplab(aes(angle=angle,color=origin,size=origin),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) + #NO ROOT  
  scale_color_manual(values=c("grey66", 'grey55', 'black')) +
  scale_size_manual(values=c(2.5,2.5,5)) 
p
##--------------------------------
##all info at once (geography and origin)
##--------------------------------
all_sterm_strain_info <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel_edited_cleaned.csv", "\t", escape_double = FALSE, trim_ws = TRUE) %>% select(c(shortName, strain, origin, location_continent,subspecies,keep))

table(all_sterm_strain_info$origin)

# all_sterm_strain_info$strain <- gsub("L-delbrueckii-","",all_sterm_strain_info$strain)

##-------------
#change tiplabel and color according to rmk or non-rmk sample
##-------------

tmp = as.character(all_sterm_strain_info$strain)
names(tmp) <- all_sterm_strain_info$shortName

# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated", tree.names = NULL, force.multi = FALSE)



Sterm_tree_raw_tible <- as_tibble(tree) 
Sterm_tree_raw_tible$label

# Sterm_tree_raw_tible$label <- plyr::revalue(Sterm_tree_raw_tible$label,tmp)
tree_02 <- Sterm_tree_raw_tible%>% as.treedata()




p2 <- ggtree(tree_02, layout="circular") +
  geom_tiplab(aes(angle=angle,color=origin,size=origin),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) + #NO ROOT  
  scale_color_manual(values=c("grey66", 'grey55', 'black')) +
  scale_size_manual(values=c(2.5,2.5,4)) 
p2



##-------------
#add geograph and origin
##-------------
all_sterm_strain_info$shortName <- as.character(all_sterm_strain_info$shortName)
all_sterm_strain_info_ready <- all_sterm_strain_info %>% select(-strain)%>% select(-subspecies) %>% remove_rownames()%>% column_to_rownames(., var = "shortName")%>%replace(is.na(.), "Unknown") 
table(all_sterm_strain_info_ready$location_continent)
all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c("Europe",  "Asia", "Africa", "Unknown"))
# colors_area <- c("#464D77","#36827F","#F9DB6D","grey88")
# colors_area <- c("#F9DB6D","#36827F","#464D77","grey88")

# location_dataset$area

# tree_02 <- full_join(as_tibble(tree), location_dataset, by='label') %>% as.treedata()

colnames(all_sterm_strain_info_ready) <- c("A","B","C")
rownames(all_sterm_strain_info_ready)

##-------------------
##color==subspecies

all_sterm_strain_info_subspecies <- all_sterm_strain_info %>% select(c(shortName,subspecies)) %>% remove_rownames()%>%replace(is.na(.), "Unknown") 
colnames(all_sterm_strain_info_subspecies)[1] <- "label"
tree_03 <- full_join(as_tibble(tree_02), all_sterm_strain_info_subspecies, by='label') %>% tidyr::replace_na(., list(subspecies="REFs"))  %>% as.treedata()

tmp <- full_join(as_tibble(tree_02), all_sterm_strain_info_subspecies, by='label') %>% tidyr::replace_na(., list(subspecies="REFs"))
table(tmp$subspecies)
colorsss <- c("grey80","grey44","grey66","grey70","purple","black","grey55")
# colorsss <- c("grey80","blue","grey66","grey70","green","black","grey55")

p2 <- ggtree(tree_03) +
  geom_tiplab(aes(color=subspecies,size=origin),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) + #NO ROOT  
  scale_color_manual(values=colorsss) +
  scale_size_manual(values=c(2.5,2.5,4)) +
  geom_treescale()
p2

library(plotly)
plotly::ggplotly(p2)


all_sterm_strain_info_ZOOMED <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) %>% filter(species!="S.thermophilus") %>% select(c(strain,Cheese,starterCulture,Sequencing)) %>% remove_rownames() %>%  mutate(strain =paste0("L",strain))%>% column_to_rownames(., var = "strain")%>%replace(is.na(.), "Unknown") 

colsss <- cols[13:14]

COLORSSS <- c("#d45400ff","#de592bff","#e36b42ff","#e67d59ff","#eb8f70ff","#eea38aff","#009cf0ff","#33cffaff","#7fe3ffff","#007f7fff","#00dcdcff","#ff0000ff","#0000ffff","grey","#fde623ff","#00ff00ff")

p3 <- gheatmap(p, all_sterm_strain_info_ZOOMED, offset=0.074, width=.05,colnames_angle=75, colnames_offset_y = 0)+ geom_treescale()+geom_hilight(c(125),fill=cols[13],extendto=0.362)+geom_hilight(c(23),fill=cols[14],extendto=0.362)+  scale_fill_manual(values=COLORSSS)+
  geom_cladelabel(node=118,label="L.delbrueckii subsp. lactis", offset=0.11, offset.text=.1,    barsiz=3, color='grey55', hjust='center')+
  geom_cladelabel(node=100,label="L.delbrueckii subsp. bulgaricus", offset=0.13, offset.text=.1,    barsiz=3, color='grey55', hjust='center')#+
  # geom_cladelabel(node=117,label="L.delbrueckii subsp. delbrueckii", offset=0.12, offset.text=.1,    barsiz=3, color='grey55', hjust='center') #+
  # geom_cladelabel(node=13,label="L.delbrueckii subsp. indicus", offset=0.12, offset.text=.1,    barsiz=3, color='grey55', hjust='center')  #+  scale_fill_manual(values=colors_area)
p3
# , offset=0.001, offset.text=.5,    barsiz=3, color='darkgreen', angle=90, hjust='center'
# 
# 
# p3 <- gheatmap(p2, all_sterm_strain_info_ZOOMED, offset=0.1, width=.1,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
# p3

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Ldel_phylogney_total_circlar.pdf",width=15,height=15)

p3

dev.off()



library(svglite)
# svglite("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.svg",width=10,height=10)
  # svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.svg",width=9,height=9)
   # png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.png", width = 3000, height = 3000,res=300)
pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Ldel_phylogney_total.pdf",width=10,height=10)

p3

dev.off()


# + geom_cladelabel(118, 'a selected clade', offset=6.5, offset.text=.5,    barsiz=3, color='darkgreen', angle=90, hjust='center')

118


# COLORSSS <- c("#d45400ff","#de592bff","#e36b42ff","#e67d59ff","#eb8f70ff","#eea38aff","#009cf0ff","#33cffaff","#7fe3ffff","#007f7fff","#00dcdcff","#ff0000ff","#0000ffff","grey","#fde623ff","#00ff00ff")
p4 <- gheatmap(p, all_sterm_strain_info_ZOOMED, offset=0.025, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)+  scale_fill_manual(values=COLORSSS)+ geom_treescale()
p4

p5 <- p4+geom_hilight(c(53),fill=cols[1],extendto=0.145)




# +geom_hilight(c(155),fill=cols[2],extendto=0.145)+geom_hilight(c(190),fill=cols[3],extendto=0.145)+
#   geom_hilight(c(206),fill=cols[4],extendto=0.145)+
#   geom_hilight(c(203),fill=cols[5],extendto=0.145)+
#   geom_hilight(c(198),fill=cols[6],extendto=0.145)+
#   geom_hilight(c(210),fill=cols[7],extendto=0.145)+
#   geom_hilight(c(216),fill=cols[8],extendto=0.145)+
#   geom_hilight(c(100),fill=cols[9],extendto=0.145)+
#   geom_hilight(c(129),fill=cols[10],extendto=0.145)+
#   geom_hilight(c(128),fill=cols[11],extendto=0.145)+
#   geom_hilight(c(11),fill=cols[12],extendto=0.145)

# p5
#  library(plotly)
# ggp <- ggplotly(p5)
#  ggp

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total_circlar.pdf",width=15,height=15)

p5

dev.off()


```

### identify nodes

here, I want to create a list containing a ll parents and children from the nodes in order to then assign the horizontally transfered genes

get local
```{bash}

mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted.txt ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/


```

```{r}

library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)

Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_curated.txt", tree.names = NULL, force.multi = FALSE)

##--------------------------------
##color labels of of only rmk phages
##--------------------------------

tree_phylo <- Sterm_tree_raw %>% as.phylo()

# phangorn::Children(tree_phylo, 2)
# Descendants(tree_phylo, 53, type = c("tips", "children", "all"))
# Descendants(tree_phylo, 53, type = "tips")


# 
# p <- ggtree(tree, layout="circular") +
#   geom_tiplab(aes(angle=angle,color=origin,size=origin),align = TRUE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0) + #NO ROOT  
#   scale_color_manual(values=c("grey66", 'grey55', 'black')) +
#   scale_size_manual(values=c(2.5,2.5,5)) 


plot(tree_phylo, show.tip.label = FALSE)
# tiplabels()

nodelabels()
p

# Children(tree_phylo,114)
Descendants(tree_phylo, 122, "tips")
Descendants(tree_phylo, 155, "tips")

tree_phylo$edge.length %>% length()

Descendants(tree_phylo, 236, "tips")
Descendants(tree_phylo, 238, "tips")
Descendants(tree_phylo, 1, "tips")
Descendants(tree_phylo, 2, "tips")
Descendants(tree_phylo, 2, "tips")


tree_phylo$tip.label[11]
tree_phylo$tip.label[34]
tree_phylo$tip.label[35]

###-------------------------
#create the list
###-------------------------


nodes_2_tips <- data.frame()
for (nodessss in (1:(tree_phylo$edge.length %>% length()+1))) {
  print(nodessss)
  for (tipssss in (Descendants(tree_phylo, nodessss, "tips")[[1]])) {
      print(tipssss)
      tipsnamesss <- tree_phylo$tip.label[tipssss]
      tmp <- data.frame(node=nodessss,tip=tipsnamesss)
      nodes_2_tips <- rbind(nodes_2_tips,tmp)
  }

}
dim(nodes_2_tips)
table(nodes_2_tips$node)

nodes_2_tips$node <- paste0("N",nodes_2_tips$node)

write.table(nodes_2_tips,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree//nodes_2_tips.txt",sep="\t",quote = FALSE,row.names = FALSE)

```


### make high resolution plot

local names
```{bash}
scp ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel_edited_cleaned.csv vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel_edited_cleaned.csv

```


FIRST orthofinder of the necessary genomes
```{bash}
###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Sterm

outgroup=$(echo "S-PNGR-Z-K")

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/

leterzzz=$(echo $species |head -c 1)
outgroup=$(echo "S-PNGR-Z-K")

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/${outgroup}.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/

###-----------------------------------------
##orthorfinder
###-----------------------------------------
#for species in $(echo "Ldel Sterm"|tail -1)


rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/


    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}// \
      -t 35 \
      -o /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/ \
      -a 35
      
##===========================================
#Ldel
##===========================================

Ldel=Ldel

outgroup=$(echo "S-PNGR-Z-K")

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED_Ldel/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED_Ldel/FAA/${species}/

##---lactis
for genomess in $(awk -F "\t" '{if($8=="lactis")print $1}'  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel_edited_cleaned.csv)
do



cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/${genomess}.faa  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED_Ldel/FAA/${species}/

done


##---outgroup

leterzzz=$(echo $species |head -c 1)
outgroup=$(echo "L-ATCC-11842")


cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/${outgroup}.faa  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED_Ldel/FAA/${species}/


##---REFs

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED_Ldel/FAA/${species}/




###-----------------------------------------
##orthorfinder
###-----------------------------------------
#for species in $(echo "Ldel Sterm"|tail -1)


rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/


    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED_Ldel/FAA/${species}/ \
      -t 20 \
      -o /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/ \
      -a 20
      
```

move local only temporary solution for Ldel

```{bash}

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ZOOMED/Ldel

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/Ldel/Results_Feb12/Species_Tree/SpeciesTree_rooted_node_labels.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ZOOMED/Ldel/
```




```{bash}
date=Results_Feb12
species=Ldel
#/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/Results_Jul28/

echo "==========================================="
echo -e ${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/get_singlecp_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/${date}/Orthogroups/Orthogroups.txt > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt

wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt
wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/${date}/Orthogroups/Orthogroups.txt

```

Here, I make a directory containing the fna and faa files of all genomes with the file name corresponding to the locus tag. 

```{bash}


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FFN/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FFN/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FFN/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FFN/${species}/

#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA/

###--------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/ |grep ".faa$"|sed 's/.faa//g'|grep -v "Out-sali")
do
locusTagsss=$(head -1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/${genomesss}.faa|cut -d '_' -f 1|sed 's/>//g')
echo -e "This  genome : " ${genomesss} " has the following locus Tag : "${locusTagsss}

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FAA/${species}/${genomesss}.faa > \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_ZOOMED/FFN/${species}/${genomesss}.ffn > \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.ffn


done


##------=================================================================
#extract orthogroups
##------=================================================================

cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/extract_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own_ZOOMED/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt --folder ${species}

```

align protein families

```{bash}


cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep ".faa$" |sed 's/.faa//g'|grep -v "Out-sali" )
do
rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.aln.fasta
mafft --thread 37 --maxiterate 1000  --auto /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.faa > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}_aln.fasta

done 

```
here I back translate the faa alignment to dna.

```{bash}

cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/aln_aa_to_dna.pl


```

With perl: prune alignments (removing columns represented by less than 50% of the sequences)


```{bash}


cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/prune_aln.pl


##================================================================
#remove all infomration (except genome info) from multifasta header
##================================================================


cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i "s/_0.*//" ${genesss}_aln_prune.fasta 

done

##================================================================
#shorten header
##================================================================
species=Ldel
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/-KCTC-/-/g' ${genesss}_aln_prune.fasta 

done




species=Sterm
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/-CIRM-//g' ${genesss}_aln_prune.fasta 

done

species=Sterm
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/-BM-A0/-BM-A/g' ${genesss}_aln_prune.fasta 

done


```
With perl: Concatenate pruned alignments
```{bash}
  

  echo -e ${species}
  cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}
  perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/cat_align.pl > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/${species}.phylip
  
```
With RAxML: infer the phylogeny
```{bash}
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_ncbi_own_ZOOMED/

  echo -e ${species}
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_ncbi_own_ZOOMED/

rm -r *${species}_all*

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/${species}.phylip -n ${species}_all -T 20


```

move local

```{bash}

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ZOOMED/Ldel

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_ncbi_own_ZOOMED/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ZOOMED//Ldel/
```

do in R
```{r}
library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Sterm_tree_raw_zoomed <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new_ZOOMED/raxml_all_combined_ncbi_own_ZOOMED/RAxML_bipartitions.Sterm_all_cleaned", tree.names = NULL, force.multi = FALSE)


# x <- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
p <-ggtree(Sterm_tree_raw_zoomed) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)+  #NO ROOT  
  geom_treescale(x=0, y=35)

p

##--------------------------------
##color labels of of only rmk phages
##--------------------------------
Sterm_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/logs/Sterm_genome_origin.txt",  "\t", escape_double = FALSE, col_names = c("label","origin"),  trim_ws = TRUE) %>% as.data.frame()

# tmp <- full_join(as_tibble(Sterm_tree_raw), Sterm_genome_origin, by='label')
# table(tmp$origin)
tree <- left_join(as_tibble(Sterm_tree_raw_zoomed), Sterm_genome_origin, by='label') %>% as.treedata()



tmp <- as.tibble(tree)
tmp$label <- revalue(tmp$label,c("S-PNGR-Z-K"="PNGR-Z_K3"))
tree <- tmp %>% as.treedata()


p <- ggtree(tree) +
  geom_tiplab(aes(color=origin),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) + #NO ROOT  
  scale_color_manual(values=c("grey66",  'black')) 
p
##--------------------------------
##all info at once (geography and origin)
##--------------------------------
all_sterm_strain_info_ZOOMED <- read_csv("Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) %>% filter(species=="S.thermophilus") %>% select(c(strain,Cheese,starterCulture,Sequencing)) %>% remove_rownames()%>% column_to_rownames(., var = "strain")%>%replace(is.na(.), "Unknown") 

p3 <- gheatmap(p, all_sterm_strain_info_ZOOMED, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0)+ geom_treescale()#+  scale_fill_manual(values=colors_area)
p3


##----------------CHANGE COLORS
COLORSSS <- c("#d45400ff","#de592bff","#e36b42ff","#e67d59ff","#eb8f70ff","#eea38aff","#009cf0ff","#33cffaff","#7fe3ffff","#007f7fff","#00dcdcff","#ff0000ff","#0000ffff","grey","#fde623ff","#00ff00ff")
p4 <- gheatmap(p, all_sterm_strain_info_ZOOMED, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0,)+  scale_fill_manual(values=COLORSSS)+ geom_treescale()
p4

##--------------------------------
##ad poppunk clustering
##--------------------------------
Sterm_clusters_RMKS <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv", col_names = c("label","cluster"), col_types = cols(cluster = col_character(),  label = col_character()), skip = 1)%>% remove_rownames()%>% column_to_rownames(., var = "label")%>%replace(is.na(.), "Unknown") 



p5 <- gheatmap(p, Sterm_clusters_RMKS, offset=0.001, width=.05,colnames_angle=75, colnames_offset_y = 0)  #+  scale_fill_manual(values=colors_area)
p5

library(patchwork)



  pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_zoomed.pdf",width=15,height=12)
   # png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_zoomed", width = 4500, height = 4500,res=300)

p4+p5

dev.off()


###--------------------------------
#distance 
library(ape)

## get the distance between every two nodes
tree_phylog <- tree %>% as.phylo()
distance_matrix <-cophenetic.phylo(tree_phylog)


```

 Ldel


```{r}
library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Sterm_tree_raw_zoomed <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new_ZOOMED/Ldel/raxml_all_combined_ncbi_own_ZOOMED/RAxML_bipartitions.Ldel_all_curated", tree.names = NULL, force.multi = FALSE)


# x <- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
p <-ggtree(Sterm_tree_raw_zoomed) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)+  #NO ROOT  
  geom_treescale(x=0, y=35)

p

##--------------------------------
##color labels of of only rmk phages
##--------------------------------
# Sterm_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/logs/",  "\t", escape_double = FALSE, col_names = c("label","origin"),  trim_ws = TRUE) %>% as.data.frame()

Sterm_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_Ldel/logs/Ldel_genome_origin_cleaned.txt",  "\t", escape_double = FALSE, col_names = c("label","origin"),  trim_ws = TRUE) %>% as.data.frame()

tmp <- as_tibble(Sterm_tree_raw_zoomed)

table(tmp$label)
# tmp$label2 <- "na"
#change label
variable <- 1
for (variable in 1:78) {
  
  tmp1 <- tmp[variable,"label"] %>% gsub("L-","",.) %>% unlist() %>% as.character()
  
  tmp[variable,"label"] <- Sterm_genome_origin[grepl(tmp1,Sterm_genome_origin$label),"label"][1]
  
}


Sterm_tree_raw_zoomed <- tmp %>% as.treedata()

tree <- left_join(as_tibble(Sterm_tree_raw_zoomed), Sterm_genome_origin, by='label') %>% as.treedata()



tmp <- as.tibble(tree)
tmp$label <- revalue(tmp$label,c("S-PNGR-Z-K"="PNGR-Z_K3"))
tree <- tmp %>% as.treedata()


p <- ggtree(tree) +
  geom_tiplab(aes(color=origin),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) + #NO ROOT  
  scale_color_manual(values=c("grey66",  'black')) 
p
##--------------------------------
##all info at once (geography and origin)
##--------------------------------
all_sterm_strain_info_ZOOMED <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) %>% filter(species!="S.thermophilus") %>% select(c(strain,Cheese,starterCulture,Sequencing)) %>% remove_rownames() %>%  mutate(strain =paste0("L",strain))%>% column_to_rownames(., var = "strain")%>%replace(is.na(.), "Unknown") 




p3 <- gheatmap(p, all_sterm_strain_info_ZOOMED, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0)+ geom_treescale()#+  scale_fill_manual(values=colors_area)
p3


##----------------CHANGE COLORS
COLORSSS <- c("#d45400ff","#de592bff","#e36b42ff","#e67d59ff","#eb8f70ff","#eea38aff","#009cf0ff","#33cffaff","#7fe3ffff","#007f7fff","#00dcdcff","#ff0000ff","#0000ffff","grey","#fde623ff","#00ff00ff")
p4 <- gheatmap(p, all_sterm_strain_info_ZOOMED, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0) +  scale_fill_manual(values=COLORSSS)+ geom_treescale()
p4

##--------------------------------
##ad poppunk clustering
##--------------------------------
# Sterm_clusters_RMKS <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv", col_names = c("label","cluster"), col_types = cols(cluster = col_character(),  label = col_character()), skip = 1)%>% remove_rownames()%>% column_to_rownames(., var = "label")%>%replace(is.na(.), "Unknown") 
# 
# 
# 
# p5 <- gheatmap(p, Sterm_clusters_RMKS, offset=0.001, width=.05,colnames_angle=75, colnames_offset_y = 0)  #+  scale_fill_manual(values=colors_area)
# p5

library(patchwork)



  pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Ldel_phylogney_zoomed.pdf",width=15,height=12)
   # png("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_zoomed", width = 4500, height = 4500,res=300)

p4+p4

dev.off()


###--------------------------------
#distance 
library(ape)

## get the distance between every two nodes
tree_phylog <- tree %>% as.phylo()
distance_matrix <-cophenetic.phylo(tree_phylog)


```

### trees for different geographies

###### Sterm


```{r}
library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(RColorBrewer)
library(ggnewscale)

##==================================================================
##==Sterm
##==================================================================




Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated", tree.names = NULL, force.multi = FALSE)


# x <- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
p <- ggtree(Sterm_tree_raw) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  
p 
# Sterm_tree_raw$tip.label
##--------------------------------
##all info at once (geography and origin)
##--------------------------------
# all_sterm_strain_info <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited.csv", "\t", escape_double = FALSE, trim_ws = TRUE) %>% select(c(shortName, strain, origin, location_continent))
# 
# all_sterm_strain_info$strain <- gsub("S-thermophilus-","",all_sterm_strain_info$strain)
# 
# all_sterm_strain_info$shortName <- as.character(all_sterm_strain_info$shortName)
# 
# all_sterm_strain_info$shortName %in% Sterm_tree_raw$tip.label
# Sterm_tree_raw$tip.label %in% all_sterm_strain_info$shortName 
# 
# all_sterm_strain_info_ready <- all_sterm_strain_info %>% select(-strain) %>% remove_rownames()%>% column_to_rownames(., var = "shortName")%>%replace(is.na(.), "Unknown") 
# table(all_sterm_strain_info_ready$location_continent)
#  all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c("Europe",  "Asia", "USA", "Australia" , "Unknown","Outgroup"))
# 
#  
#  colnames(all_sterm_strain_info_ready) <- c("source","geography")
# rownames(all_sterm_strain_info_ready)
# 
# p3 <- gheatmap(p , all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
# p3
#  
##-----------------------------------------------
###import metaData
##-----------------------------------------------

meta_sterm <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited_extended_02.csv")
# colnames(meta_sterm)
meta_sterm$shortName_new <- meta_sterm$shortName %>% gsub("^","S",.) %>% gsub("^SS","S",.)
# names_strains$strain


##-----------------------------------------------
###add source
##-----------------------------------------------


meta_sterm_ready <- meta_sterm %>% select(shortName,origin) %>% unique() %>% filter(shortName %in%Sterm_tree_raw$tip.label) %>% remove_rownames()%>% column_to_rownames(., var = "shortName") #%>%replace(is.na(.), "Unknown")

table(meta_sterm_ready$origin)
meta_sterm_ready$origin <- droplevels(as.factor(meta_sterm_ready$origin))

meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c("Cheese","milk","yogurt","cow","human","host","Unknown"))
meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c("Cheese","milk","yogurt","host"))

colorsss <- c("#cdb4db","#e63946","#ffc8dd","#ffafcc","white")

meta_sterm_ready_origin <- meta_sterm_ready %>% rownames_to_column() %>% as.tibble()

p2 <- gheatmap(p , meta_sterm_ready, offset=0.04, width=.05) + scale_fill_manual(values=colorsss)#+  scale_fill_manual(values=colors_area)
p2

##-----------------------------------------------
###add geography
##-----------------------------------------------


meta_sterm_ready <- meta_sterm %>% select(shortName,location_continent) %>% unique()%>% remove_rownames()%>% column_to_rownames(., var = "shortName")

table(meta_sterm_ready$location_continent)
# meta_sterm_ready$origin <- droplevels(as.factor(meta_sterm_ready$origin))

# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c("Cheese","milk","yogurt","cow","human","host","Unknown"))
# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c("Cheese","milk","yogurt","host"))

colorsss <- c("#2a9d8f","#e9c46a","#f4a261","#e76f51","white")
p2 <- p2 + new_scale_fill()

p3 <- gheatmap(p2 , meta_sterm_ready, offset=0.08, width=.05 ) + scale_fill_manual(values=colorsss)#+  scale_fill_manual(values=colors_area)
p3

##-----------------------------------------------
###add time
##-----------------------------------------------

meta_sterm_ready <- meta_sterm %>% select(shortName,location_continent,origin,extraction_time) %>% unique()%>% remove_rownames()%>% column_to_rownames(., var = "shortName") %>% select(-location_continent,-origin) %>% mutate(extraction_time=as.numeric(extraction_time))

p3 <- p3 + new_scale_fill()
# p3 <- gheatmap(p2, df, offset=i*offset, width=.08,colnames_angle=0, colnames_offset_y = .05)  

p4 <- gheatmap(p3 , meta_sterm_ready, offset=0.12, width=.05) #+ scale_color_brewer(type="div")

p4

###----------------------------------------------
# add ancestral niche reconstruction
###----------------------------------------------

library(ape)
# ape::ace


###------------test

### For discrete characters:
data(bird.orders)
x <- factor(c(rep(0, 5), rep(1, 18)))
ans <- ace(x, bird.orders, type = "d")
#### Showing the likelihoods on each node:
plot(bird.orders, type = "c", FALSE, label.offset = 1)
co <- c("blue", "yellow")
tiplabels(pch = 22, bg = co[as.numeric(x)], cex = 2, adj = 1)
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.75)


###------------real data

meta_sterm_ready_origin
colorsss <- c("#cdb4db","#e63946","#ffc8dd","#ffafcc","white")

Sterm_tree_raw %>% tip.label()
Sterm_tree_raw$tip.label

# is_tip <- Sterm_tree_raw$edge[,2] <= length(Sterm_tree_raw$tip.label)


# x <- get_taxa_name(p2) %>% as.data.frame() %>% rename("rowname"=".") %>% as.tibble() %>% left_join(.,meta_sterm_ready_origin) %>% pull(origin)
bird.orders <- drop.tip(Sterm_tree_raw,"Out-vest")

is_tip <- bird.orders$edge[,2] <= length(bird.orders$tip.label)
ordered_tips <- bird.orders$edge[is_tip, 2]

x <- bird.orders$tip.label[ordered_tips] %>% as.data.frame() %>% rename("rowname"=".") %>% as.tibble() %>% left_join(.,meta_sterm_ready_origin) %>% pull(origin)  %>% as.character()

table(x)

ans <- ace(x, bird.orders, type = "d",method = "ML")
#### Showing the likelihoods on each node:
plot(bird.orders, type = "c", FALSE, label.offset = 1, cex = 0.5)
# co <- c("blue", "yellow")
co <- c("#cdb4db","#e63946","#ffc8dd","grey","#ffafcc","white")
# co[as.numeric(x)]


legend("bottomleft", legend=levels(as.factor(x)),
       col=c("#cdb4db","#e63946","#ffc8dd","grey"), fill=c("#cdb4db","#e63946","#ffc8dd","grey"), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8

# legend("bottomleft", inset=.02, title="Number of Cylinders",
   # c("4","6","8"), fill=topo.colors(3), horiz=TRUE, cex=0.8)
tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )

# p2

ans$lik.anc
ans$rates
ans$se
ans$index.matrix


###---------------


 pdf("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/phylog_supp_ACE.pdf",width=12,height=9)

 
plot(bird.orders, type = "c", FALSE, label.offset = 1, cex = 0.5)

legend("bottomleft", legend=levels(as.factor(x)),
       col=c("#cdb4db","#e63946","#ffc8dd","grey"), fill=c("#cdb4db","#e63946","#ffc8dd","grey"), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8

tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )


dev.off()

####---------dairy and host-associated


meta_sterm_ready_origin
colorsss <- c("#cdb4db","#e63946","#ffc8dd","#ffafcc","white")
table(meta_sterm_ready_origin$origin)
meta_sterm_ready_origin$origin <- plyr::revalue(meta_sterm_ready_origin$origin, c("yogurt"="Dairy","Cheese"="Dairy","milk"="Dairy"))



Sterm_tree_raw %>% tip.label()
Sterm_tree_raw$tip.label

# is_tip <- Sterm_tree_raw$edge[,2] <= length(Sterm_tree_raw$tip.label)


# x <- get_taxa_name(p2) %>% as.data.frame() %>% rename("rowname"=".") %>% as.tibble() %>% left_join(.,meta_sterm_ready_origin) %>% pull(origin)
bird.orders <- drop.tip(Sterm_tree_raw,"Out-vest")

is_tip <- bird.orders$edge[,2] <= length(bird.orders$tip.label)
ordered_tips <- bird.orders$edge[is_tip, 2]

x <- bird.orders$tip.label[ordered_tips] %>% as.data.frame() %>% rename("rowname"=".") %>% as.tibble() %>% left_join(.,meta_sterm_ready_origin) %>% pull(origin)  %>% as.character()

table(x)

ans <- ace(x, bird.orders, type = "d",method = "ML")
#### Showing the likelihoods on each node:
plot(bird.orders, type = "c", FALSE, label.offset = 1, cex = 0.5)
# co <- c("blue", "yellow")
co <- c("blue","red","white")
# co[as.numeric(x)]


legend("bottomleft", legend=levels(as.factor(x)),
       col=c("blue","red","white"), fill=c("blue","red","white"), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8

# legend("bottomleft", inset=.02, title="Number of Cylinders",
   # c("4","6","8"), fill=topo.colors(3), horiz=TRUE, cex=0.8)
tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )


 pdf("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/phylog_supp_ACE_dariy.pdf",width=12,height=9)

 plot(bird.orders, type = "c", FALSE, label.offset = 1, cex = 0.5)
# co <- c("blue", "yellow")
co <- c("blue","red","white")
# co[as.numeric(x)]


legend("bottomleft", legend=levels(as.factor(x)),
       col=c("blue","red","white"), fill=c("blue","red","white"), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8

# legend("bottomleft", inset=.02, title="Number of Cylinders",
   # c("4","6","8"), fill=topo.colors(3), horiz=TRUE, cex=0.8)
tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )


dev.off()

```


###### Ldel
```{r}
library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(RColorBrewer)
library(ggnewscale)

##==================================================================
##==Ldel
##==================================================================


Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_Ldel/SpeciesTree_rooted_node_labels_orthofinder_ldel_complete_rerooted_cleaned.txt", tree.names = NULL, force.multi = FALSE)

# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Ldel_all", tree.names = NULL, force.multi = FALSE)


# x <- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
p <- ggtree(Sterm_tree_raw) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  
p 

##-----------------------------------------------
###import metaData
##-----------------------------------------------
# meta_sterm <-  read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Ldel_phylogeny_info.tsv", 
    # delim = "\t", escape_double = FALSE, 
    # trim_ws = TRUE)

meta_ldel <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_ldel_edited_02.csv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)

colnames(meta_ldel)
meta_ldel$shortName_new <- meta_ldel$shortName %>% gsub("-DSM-","-",.) %>% gsub("-KCTC-","-",.)



# names_strains$strain


# meta_sterm <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_ldel_edited.csv")
# # colnames(meta_sterm)
# meta_sterm$shortName_new <- meta_sterm$shortName %>% gsub("^","S",.) %>% gsub("^SS","S",.)
# # names_strains$strain


##-----------------------------------------------
###add source
##-----------------------------------------------


meta_sterm_ready <- meta_ldel %>% select(shortName,origin) %>% unique() %>% filter(shortName %in%Sterm_tree_raw$tip.label) %>% remove_rownames()%>% column_to_rownames(., var = "shortName") #%>%replace(is.na(.), "Unknown")

table(meta_sterm_ready$origin)
meta_sterm_ready$origin <- droplevels(as.factor(meta_sterm_ready$origin))

# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c("Cheese","milk","yogurt","cow","human","host","Unknown"))
meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c("Cheese","milk","yogurt","host"))

colorsss <- c("#cdb4db","#e63946","#ffc8dd","#ffafcc","white")


meta_sterm_ready_origin <- meta_sterm_ready %>% rownames_to_column() %>% as.tibble()


p2 <- gheatmap(p , meta_sterm_ready, offset=0.04, width=.05) + scale_fill_manual(values=colorsss)#+  scale_fill_manual(values=colors_area)
p2

##-----------------------------------------------
###add geography
##-----------------------------------------------


meta_sterm_ready <- meta_ldel %>% select(shortName,location_continent) %>% unique()%>% remove_rownames()%>% column_to_rownames(., var = "shortName")

table(meta_sterm_ready$location_continent)
# meta_sterm_ready$origin <- droplevels(as.factor(meta_sterm_ready$origin))

# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c("Cheese","milk","yogurt","cow","human","host","Unknown"))
# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c("Cheese","milk","yogurt","host"))

colorsss <- c("#2a9d8f","#e9c46a","#f4a261","#e76f51","white")
p2 <- p2 + new_scale_fill()

p3 <- gheatmap(p2 , meta_sterm_ready, offset=0.08, width=.05 ) + scale_fill_manual(values=colorsss)#+  scale_fill_manual(values=colors_area)
p3

##-----------------------------------------------
###add time
##-----------------------------------------------

meta_sterm_ready <- meta_ldel %>% select(shortName,location_continent,origin,extraction_time) %>% unique()%>% remove_rownames()%>% column_to_rownames(., var = "shortName") %>% select(-location_continent,-origin) %>% mutate(extraction_time=as.numeric(extraction_time))

p3 <- p3 + new_scale_fill()
# p3 <- gheatmap(p2, df, offset=i*offset, width=.08,colnames_angle=0, colnames_offset_y = .05)  

p4_ldel <- gheatmap(p3 , meta_sterm_ready, offset=0.12, width=.05) #+ scale_color_brewer(type="div")

p4_ldel

###----------------------------------------------
# add ancestral niche reconstruction
###----------------------------------------------

library(ape)
# ape::ace


###------------test

### For discrete characters:
data(bird.orders)
x <- factor(c(rep(0, 5), rep(1, 18)))
ans <- ace(x, bird.orders, type = "d")
#### Showing the likelihoods on each node:
plot(bird.orders, type = "c", FALSE, label.offset = 1)
co <- c("blue", "yellow")
tiplabels(pch = 22, bg = co[as.numeric(x)], cex = 2, adj = 1)
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.75)


###------------real data

meta_sterm_ready_origin
colorsss <- c("#cdb4db","#e63946","#ffc8dd","#ffafcc","white")

Sterm_tree_raw %>% tip.label()
Sterm_tree_raw$tip.label

# is_tip <- Sterm_tree_raw$edge[,2] <= length(Sterm_tree_raw$tip.label)


# x <- get_taxa_name(p2) %>% as.data.frame() %>% rename("rowname"=".") %>% as.tibble() %>% left_join(.,meta_sterm_ready_origin) %>% pull(origin)
bird.orders <- drop.tip(Sterm_tree_raw,"Out-helv")

is_tip <- bird.orders$edge[,2] <= length(bird.orders$tip.label)
ordered_tips <- bird.orders$edge[is_tip, 2]

x <- bird.orders$tip.label[ordered_tips] %>% as.data.frame() %>% rename("rowname"=".") %>% as.tibble() %>% left_join(.,meta_sterm_ready_origin) %>% pull(origin)  %>% as.character()

table(x)

ans <- ace(x, bird.orders, type = "d",method = "ML")
#### Showing the likelihoods on each node:
plot(bird.orders, type = "c", FALSE, label.offset = 1, cex = 0.5)
# co <- c("blue", "yellow")
co <- c("#cdb4db","#e63946","#ffc8dd","grey","#ffafcc","white")
# co[as.numeric(x)]


legend("bottomleft", legend=levels(as.factor(x)),
       col=c("#cdb4db","#e63946","#ffc8dd","grey"), fill=c("#cdb4db","#e63946","#ffc8dd","grey"), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8

# legend("bottomleft", inset=.02, title="Number of Cylinders",
   # c("4","6","8"), fill=topo.colors(3), horiz=TRUE, cex=0.8)
tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )

p2

ans$lik.anc
ans$rates
ans$se
ans$index.matrix


###---------------

# 
#  pdf("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/phylog_supp_ACE.pdf",width=12,height=9)
# 
#  
# plot(bird.orders, type = "c", FALSE, label.offset = 1, cex = 0.5)
# 
# legend("bottomleft", legend=levels(as.factor(x)),
#        col=c("#cdb4db","#e63946","#ffc8dd","grey"), fill=c("#cdb4db","#e63946","#ffc8dd","grey"), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8
# 
# tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
# nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )
# 
# 
# dev.off()

```

###### Lhelv


```{r}
library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(RColorBrewer)
library(ggnewscale)

##==================================================================
##==Lhelv
##==================================================================


Sterm_tree_raw <- read.nexus("/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/lhelv/new", tree.names = NULL, force.multi = FALSE)


# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Ldel_all", tree.names = NULL, force.multi = FALSE)


# x <- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
p <- ggtree(Sterm_tree_raw) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  
p 

##-----------------------------------------------
###import metaData
##-----------------------------------------------
# meta_sterm <-  read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Ldel_phylogeny_info.tsv", 
    # delim = "\t", escape_double = FALSE, 
    # trim_ws = TRUE)

# meta_ldel <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_03.csv", 
#     delim = "\t", escape_double = FALSE, 
#     trim_ws = TRUE)
meta_ldel <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_03.csv") %>% mutate(shortName=strain_name)

meta_ldel$shortName <- gsub("^","Lh-",meta_ldel$strain_name) %>% gsub("^Lh-Lh","",.)


colnames(meta_ldel)
meta_ldel$shortName_new <- meta_ldel$shortName %>% gsub("-DSM-","-",.) %>% gsub("-KCTC-","-",.)



# names_strains$strain


# meta_sterm <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_ldel_edited.csv")
# # colnames(meta_sterm)
# meta_sterm$shortName_new <- meta_sterm$shortName %>% gsub("^","S",.) %>% gsub("^SS","S",.)
# # names_strains$strain


##-----------------------------------------------
###add source
##-----------------------------------------------


meta_sterm_ready <- meta_ldel %>% select(shortName,origin) %>% unique() %>% filter(shortName %in%Sterm_tree_raw$tip.label) %>% remove_rownames()%>% column_to_rownames(., var = "shortName") #%>%replace(is.na(.), "Unknown")

table(meta_sterm_ready$origin)
meta_sterm_ready$origin <- droplevels(as.factor(meta_sterm_ready$origin))

# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c("Cheese","milk","yogurt","cow","human","host","Unknown"))
meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c("Cheese","milk","yogurt","host"))

colorsss <- c("#cdb4db","#e63946","#ffc8dd","#ffafcc","white")

meta_sterm_ready_origin <- meta_sterm_ready %>% rownames_to_column() %>% as.tibble()

p2 <- gheatmap(p , meta_sterm_ready, offset=0.04, width=.05) + scale_fill_manual(values=colorsss)#+  scale_fill_manual(values=colors_area)
p2

##-----------------------------------------------
###add geography
##-----------------------------------------------


meta_sterm_ready <- meta_ldel %>% select(shortName,location_continent) %>% unique()%>% remove_rownames()%>% column_to_rownames(., var = "shortName")

table(meta_sterm_ready$location_continent)
# meta_sterm_ready$origin <- droplevels(as.factor(meta_sterm_ready$origin))

# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c("Cheese","milk","yogurt","cow","human","host","Unknown"))
# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c("Cheese","milk","yogurt","host"))

colorsss <- c("#2a9d8f","#e9c46a","#f4a261","#e76f51","white")





p2 <- p2 + new_scale_fill()

p3 <- gheatmap(p2 , meta_sterm_ready, offset=0.08, width=.05 ) + scale_fill_manual(values=colorsss)#+  scale_fill_manual(values=colors_area)
p3

##-----------------------------------------------
###add time
##-----------------------------------------------

meta_sterm_ready <- meta_ldel %>% select(shortName,location_continent,origin,extraction_time) %>% unique()%>% remove_rownames()%>% column_to_rownames(., var = "shortName") %>% select(-location_continent,-origin) %>% mutate(extraction_time=as.numeric(extraction_time))

p3 <- p3 + new_scale_fill()
# p3 <- gheatmap(p2, df, offset=i*offset, width=.08,colnames_angle=0, colnames_offset_y = .05)  

p4_lhelv <- gheatmap(p3 , meta_sterm_ready, offset=0.12, width=.05) #+ scale_color_brewer(type="div")

p4_lhelv

###----------------------------------------------
# add ancestral niche reconstruction
###----------------------------------------------

library(ape)
# ape::ace


###------------test

### For discrete characters:
data(bird.orders)
x <- factor(c(rep(0, 5), rep(1, 18)))
ans <- ace(x, bird.orders, type = "d")
#### Showing the likelihoods on each node:
plot(bird.orders, type = "c", FALSE, label.offset = 1)
co <- c("blue", "yellow")
tiplabels(pch = 22, bg = co[as.numeric(x)], cex = 2, adj = 1)
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.75)


###------------real data

meta_sterm_ready_origin
colorsss <- c("#cdb4db","#e63946","#ffc8dd","#ffafcc","white")

Sterm_tree_raw %>% tip.label()
Sterm_tree_raw$tip.label

# is_tip <- Sterm_tree_raw$edge[,2] <= length(Sterm_tree_raw$tip.label)


# x <- get_taxa_name(p2) %>% as.data.frame() %>% rename("rowname"=".") %>% as.tibble() %>% left_join(.,meta_sterm_ready_origin) %>% pull(origin)
# bird.orders <- drop.tip(Sterm_tree_raw,"Out-helv")
bird.orders <- Sterm_tree_raw

is_tip <- bird.orders$edge[,2] <= length(bird.orders$tip.label)
ordered_tips <- bird.orders$edge[is_tip, 2]

x <- bird.orders$tip.label[ordered_tips] %>% as.data.frame() %>% rename("rowname"=".") %>% as.tibble() %>% left_join(.,meta_sterm_ready_origin) %>% pull(origin)  %>% as.character()

table(x)

ans <- ace(x, bird.orders, type = "d",method = "ML")
#### Showing the likelihoods on each node:
plot(bird.orders, type = "c", FALSE, label.offset = 1, cex = 0.5)
# co <- c("blue", "yellow")
co <- c("#cdb4db","#e63946","#ffc8dd","grey","#ffafcc","white")
# co[as.numeric(x)]


legend("bottomleft", legend=levels(as.factor(x)),
       col=c("#cdb4db","#e63946","#ffc8dd","grey"), fill=c("#cdb4db","#e63946","#ffc8dd","grey"), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8

# legend("bottomleft", inset=.02, title="Number of Cylinders",
   # c("4","6","8"), fill=topo.colors(3), horiz=TRUE, cex=0.8)
tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )

p2

ans$lik.anc
ans$rates
ans$se
ans$index.matrix


###---------------

# 
#  pdf("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/phylog_supp_ACE.pdf",width=12,height=9)
# 
#  
# plot(bird.orders, type = "c", FALSE, label.offset = 1, cex = 0.5)
# 
# legend("bottomleft", legend=levels(as.factor(x)),
#        col=c("#cdb4db","#e63946","#ffc8dd","grey"), fill=c("#cdb4db","#e63946","#ffc8dd","grey"), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8
# 
# tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
# nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )
# 
# 
# dev.off()

```


```{r}
###----------------------------merge

library(patchwork)

p4+p4_ldel+p4_lhelv


 # png("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/phylog_supp.png", width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
 pdf("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/phylog_supp.pdf",width=12,height=9)

 
p4+p4_ldel+p4_lhelv

dev.off()
```


## species tree for HGT

I have gathered all the genomes explained in the fig. 1. these include all our sequenced Ldel and Sterm+ all ncbi Ldel and Sterm + many Streptococcus and lactobacillus. 

1. first I need to prepare a list with the metadata. 

```{bash}


cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/
echo -e "Out_Bacsub\tout\tout" > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/genome_description.list
###-------------------rmk genomes

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Poppunk_out.all |grep "^S" |awk -F "\t" '{OFS="\t"}{print $1,"Streptococcus",$2}' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/genome_description.list
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Poppunk_out.all |grep "^L" |awk -F "\t" '{OFS="\t"}{print $1,"Lactobacillus",$2}' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/genome_description.list

###-------------------add all species close to Ldel and Sterm from figure 1D/E

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA/|grep ".faa$" | sed 's/.faa//g'|grep -v "thermophilus")
do
echo -e ${genomesss}"\tStreptococcus\tNCBI_genus" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/genome_description.list

done


for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA/|grep ".faa$" | sed 's/.faa//g'|grep -v "delbrueckii")
do
echo -e ${genomesss}"\tLactobacillus\tNCBI_genus" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/genome_description.list

done

###-------------------add all strains close to Ldel and Sterm from figure 1F/G

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/PROKKA_FAA/|grep ".faa$" | sed 's/.faa//g'|grep -v "Out")
do
echo -e ${genomesss}"\tStreptococcus\tNCBI_species" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/genome_description.list

done

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/PROKKA_FAA//|grep ".faa$" | sed 's/.faa//g'|grep -v "Out")
do
echo -e ${genomesss}"\tLactobacillus\tNCBI_species" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/genome_description.list

done
###-------------------add all strains close to Ldel and Sterm from figure 1F/G out groups


for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/PROKKA_FAA/|grep ".faa$" | sed 's/.faa//g'|grep  "Out")
do
echo -e ${genomesss}"\tStreptococcus\tNCBI_genus" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/genome_description.list

done

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/PROKKA_FAA//|grep ".faa$" | sed 's/.faa//g'|grep "Out")
do
echo -e ${genomesss}"\tLactobacillus\tNCBI_genus" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/genome_description.list

done
```

move local 
```{bash}
scp  -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/genome_description.list /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/

```



```{r}

library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Sterm_tree_raw <- read.nexus("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/SpeciesTree_rooted_node_labels_rerooted.txt", tree.names = NULL, force.multi = FALSE)


# x <- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
ggtree(Sterm_tree_raw, layout="circular") +
  #geom_tiplab(aes(angle=angle),align = FALSE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)+
  geom_treescale() + geom_tippoint( size=1)

full_join(as_tibble(Sterm_tree_raw), genome_description, by='label') %>% select(label) %>% table %>% sort()

# tree <- as_tibble(Sterm_tree_raw)
# tree %>% select(label) %>% filter(!is.na(label)) %>% unlist() %>% as.vector()

library(readr)
genome_description <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/genome_description.list",  "\t", escape_double = FALSE, col_names = c("label","genus","clustering"),  trim_ws = TRUE) %>% filter(!(label=="out_flor"&genus=="Streptococcus")) %>% filter(label!="L11063")# %>%  recode(label, S-KLDS-3.1= 'S-KLDS-3.1')

tmp1 <- as_tibble(Sterm_tree_raw) #%>% filter(!is.na(label))

tmp1$label <- gsub("\'S-KLDS-3.1\'","S-KLDS-3.1",tmp1$label)
tmp1$label <- gsub("\'L-KLDS1.0207\'","L-KLDS1.0207",tmp1$label)
tmp1$label <- gsub("\'L-KLDS1.1011\'","L-KLDS1.1011",tmp1$label)

tmp1[!tmp1$label %in% genome_description$label,]
genome_description[!genome_description$label %in% tmp1$label,]


# tree <- tmp1  %>% as.treedata()
tmp2 <- full_join(tmp1, genome_description, by='label')
tmp2$clustering = factor(tmp2$clustering, levels=c(as.character(1:14),"NCBI_genus","NCBI_species","out"))

tree <- tmp2 %>% as.treedata()

cols <- c(rainbow(14),"grey77","grey55","white") #I am not sure this is correct -- did not find the correct color labels anymore


plotTreee <- ggtree(tree, layout="circular") +
  #geom_tiplab(aes(angle=angle),align = FALSE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)+
  geom_treescale() + geom_tippoint(aes(color=clustering,shape=genus), size=2,alpha=0.8)+scale_color_manual(values = cols)
plotTreee
pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/generax_phylogney_total.pdf",width=8,height=8)

plotTreee

dev.off()


```


## Gal's HGT appraoch

Here, I apply Gals paper appraoch. 
Github can be found [here](https://github.com/ghoresh11/twilight).

I want to do this analysis for both species together. 
therefore I gather all genomes (gffs) 

```{bash}
--------------------------------------
##create protein database
###---------------------------------------

##==========================================
###-----------------bring all gffs together
##==========================================
species=Sterm
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//GFFs/
rm -r  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//FAAs/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//GFFs/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//FAAs/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//GFFs/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//FAAs/


species=Ldel

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//GFFs/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//FAAs/


rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Poppunk_out.all

grep "Taxon" -v /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv | awk -F "," '{OFS="\t"}{print $1,$2}' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Poppunk_out.all
grep "Taxon" -v /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Ldel/20210128_easyrun/20210128_easyrun_clusters_RMKS.csv | awk -F "," '{OFS="\t"}{print $1,12+$2}' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Poppunk_out.all


##---------------------check

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Poppunk_out.all |wc -l

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//GFFs/ |wc -l
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//GFFs/ |grep ".gff" |sed 's/.gff//g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/.tmp

for genomess in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Poppunk_out.all )
do

sed -i "s/${genomess}//g" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/.tmp 
#sed 's/${genomess}//g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/.tmp > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/.tmp

done

sort /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/.tmp  |uniq


##==========================================
###-----------------Panaroo
##==========================================
  
  conda activate panaroo
  
  rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/01_Panarooo/
  mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/01_Panarooo/
  
  panaroo -t 37 -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/genomes//GFFs/*.gff -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/01_Panarooo// --clean-mode strict

##==========================================
###-----------------classify_genes.R
##==========================================
rm -r  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/

Rscript /home/vincent/apps/twilight/classify_genes_VS.R --min_size 1 -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/01_Panarooo/gene_presence_absence.Rtab -g /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Poppunk_out.all


#Rscript /home/vincent/apps/twilight/classify_genes.R --min_size 1 -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/01_Panarooo/gene_presence_absence.Rtab -g /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Poppunk_out.all

grep "colSums(gene_class_presen"  /home/vincent/apps/twilight/classify_genes.R 

##==========================================
###-----------------lineage specific core
##==========================================


grep -i "lineage specific core" out/genes_per_isolate.tab 

```


```{bash}


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/

#scp  -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/out/plots/ /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/

scp  -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Poppunk_out.all /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/


scp  -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/out/ /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/
```

#### lineage specific genes

```{r}


classification_genes <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/out/classification.tab", "\t", escape_double = FALSE, trim_ws = TRUE)

classification_genes_lineage <- classification_genes %>% filter(grepl("Lineage specific ",specific_class))

classification_genes_lineage$lineage <- gsub("[^[:digit:]., ]", "", classification_genes_lineage$details) %>% as.numeric() %>% as.character()
# classification_SNVs_lineage_cleaned <- classification_SNVs_lineage %>% select(gene_name,lineage,specific_class)

table(classification_genes_lineage$lineage,classification_genes_lineage$specific_class)

species_tmp <- data.frame(lineage=as.character(1:14),species=c(rep("S.thermophilus",12),rep("L.delbrueckii",2)))

classification_genes_lineage_final <-  merge(classification_genes_lineage,species_tmp,by="lineage") %>% select(gene_name,lineage,specific_class,species)

classification_genes_lineage_final$lineage = factor(classification_genes_lineage_final$lineage, levels=unique(as.character(1:14)))
classification_genes_lineage_final$species = factor(classification_genes_lineage_final$species, levels=c("S.thermophilus","L.delbrueckii"))
classification_genes_lineage_final$specific_class = factor(classification_genes_lineage_final$specific_class, levels=c("Lineage specific rare","Lineage specific intermediate","Lineage specific core"))
# table(classification_genes_lineage_final$specific_class)

ggplot(classification_genes_lineage_final,aes(x=lineage,fill=specific_class))+geom_histogram(stat="count")+theme_classic()+labs(x="lineage",y="gene count")+facet_wrap(~species,scales="free_x")


##-----make species inot plot and sam size bars  

# genes_per_isolate_lineage_final_count <- table(genes_per_isolate_lineage_final$class,genes_per_isolate_lineage_final$count) %>% as.data.frame()%>% dplyr::rename(class = Var1) 

genes_per_isolate_lineage_final_count <- table(classification_genes_lineage_final$specific_class,classification_genes_lineage_final$lineage) %>% as.data.frame()%>% dplyr::rename(specific_class = Var1,lineage = Var2) 
  genes_per_isolate_lineage_final_count <- merge(genes_per_isolate_lineage_final_count,species_tmp,by="lineage")
genes_per_isolate_lineage_final_count$lineage = factor(genes_per_isolate_lineage_final_count$lineage, levels=as.character(1:14))

genes_per_isolate_lineage_final_count$species = factor(genes_per_isolate_lineage_final_count$species, levels=c("S.thermophilus","L.delbrueckii"))
genes_per_isolate_lineage_final_count$specific_class = factor(genes_per_isolate_lineage_final_count$specific_class, levels=c("Lineage specific rare","Lineage specific intermediate","Lineage specific core"))
colorsss <- rev(c("grey44","grey60","grey77"))
lineage_specific_genes_plot <- ggplot(genes_per_isolate_lineage_final_count,aes(x=lineage,y=Freq,fill=specific_class))+geom_col()+theme_classic()+labs(x="lineage",y="lineage specific genes")+facet_grid(~species, scales = "free", space = "free_x")+scale_fill_manual(values = colorsss)+theme(legend.title = element_blank())
lineage_specific_genes_plot


pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/lineage_specificGenes.pdf",width=6,height=3.5)

lineage_specific_genes_plot

dev.off()


##----larger plot

library(patchwork)
 pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/lineage_specificGenes_rarefaction.pdf",width=8,height=3.5)

rarefaction_lineages_plot+lineage_specific_genes_plot

 dev.off()

```

### core genes

```{r}


classification_genes <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/out/classification.tab", "\t", escape_double = FALSE, trim_ws = TRUE)

# classification_genes_lineage <- classification_genes %>% filter(grepl("Lineage specific ",specific_class))

 classification_genes$details

classification_genes$lineage <- gsub(" Inter:.*", "", classification_genes$details) %>% gsub("Core: ", "", .)
 
classification_genes$lineage <- gsub("[^[:digit:]., ]", "", classification_genes$details) %>% as.numeric() %>% as.character()


classification_genes_core <- classification_genes %>% 
    mutate(lineage_split = strsplit(lineage, "\\+")) %>% 
    unnest(lineage_split)# %>% filter(lineage_split!="+")

classification_genes_core_count <- table(classification_genes_core$lineage_split) %>% as.data.frame() %>% rename(., lineage = Var1) %>% add_column(specific_class="Core")


species_tmp <- data.frame(lineage=as.character(1:14),species=c(rep("S.thermophilus",12),rep("L.delbrueckii",2)))
classification_genes_core_count_final <-  merge(classification_genes_core_count,species_tmp,by="lineage") %>% select(lineage,specific_class,Freq,species)

classification_genes_core_count_final$lineage = factor(classification_genes_core_count_final$lineage, levels=unique(as.character(1:14)))

classification_genes_core_count_final$species = factor(classification_genes_core_count_final$species, levels=c("S.thermophilus","L.delbrueckii"))

ggplot(classification_genes_core_count_final,aes(x=lineage,y=Freq))+geom_bar( stat="identity")+theme_classic()+ coord_cartesian(ylim=c(min(classification_genes_core_count$Freq),max(classification_genes_core_count$Freq)))+facet_grid(~species, scales = "free", space = "free_x")

##---------------------------
#add lineage specific genes 

classification_genes_all <- rbind(classification_genes_core_count_final,genes_per_isolate_lineage_final_count)

colorsss <- rev(c("grey44","grey60","grey77","grey22"))
classification_genes_all$specific_class = factor(classification_genes_all$specific_class, levels=c("Core","Lineage specific rare","Lineage specific intermediate","Lineage specific core"))

classification_genes_all$Class <- plyr::revalue(as.character(classification_genes_all$specific_class), c("Core"="Core","Lineage specific"="Lineage specific rare","Lineage specific"="Lineage specific intermediate","Lineage specific"="Lineage specific core"))
classification_genes_all$Class <- plyr::revalue(as.character(classification_genes_all$specific_class), c("Core"="Core","Lineage specific rare"="Lineage specific","Lineage specific intermediate"="Lineage specific","Lineage specific core"="Lineage specific"))

# ggplot(classification_genes_all,aes(x=lineage,y=Freq,fill=specific_class))+geom_bar( stat="identity", position = position_dodge(width = 0), width = 2.5)+theme_classic()+facet_grid(Class~species, scales = "free", space = "free")+theme(legend.title = element_blank())+scale_fill_manual(values = colorsss)


ggplot(classification_genes_all,aes(x=lineage,y=Freq,fill=specific_class))+geom_bar( stat="identity", position = position_dodge(width = 0), width = 3.5)+theme_classic()+facet_grid(~species, scales = "free", space = "free")+theme(legend.title = element_blank())+scale_fill_manual(values = colorsss)

##---------------make breaks
  #Function to transform data to y positions
  trans <- function(x){pmin(x,200) + 0.9*pmax(x-1650,0)}
  
  yticks <- c(0, 50, 100, 150, 200, 1750, 1850,1950)
  
  #Transform the data onto the display scale
  classification_genes_all$Freq_t <- trans(classification_genes_all$Freq)
  # dat$sd_up_t <- trans(dat$mean + dat$sd)
  # dat$sd_low_t <- pmax(trans(dat$mean - dat$sd),1) #
  # 
  # ggplot(data=dat, aes(x=AAP, y=mean_t, group=Sex,fill=Sex)) +
  #   # geom_errorbar(aes(ymin=sd_low_t, ymax=sd_up_t),position="dodge") + 
  #   geom_col(position="dodge") +
  #   geom_rect(aes(xmin=0, xmax=6, ymin=42, ymax=48), fill="white") +
  #   scale_y_continuous(limits=c(0,NA), breaks=trans(yticks), labels=yticks) +
  #   labs(y="Relative titer of CLas")
  
  
 gene_plot <-  ggplot(classification_genes_all,aes(x=lineage,y=Freq_t,fill=specific_class))+geom_bar( stat="identity", position = position_dodge(width = 0), width = 3.5)+theme_classic()+facet_grid(~species, scales = "free", space = "free_x")+theme(legend.title = element_blank(),axis.title.x = element_blank())+scale_fill_manual(values = colorsss)+geom_rect(aes(xmin=-5, xmax=15, ymin=205, ymax=220), fill="white") +scale_y_continuous(limits=c(0,NA), breaks=trans(yticks), labels=yticks) +labs(y="gene count")

# ggplot(classification_genes_all,aes(x=lineage,y=Freq_t,fill=specific_class))+geom_bar( stat="identity", position = position_dodge(width = 0), width = 3.5)+theme_classic()+facet_wrap(~species, scales = "free_x")+theme(legend.title = element_blank())+scale_fill_manual(values = colorsss)+geom_rect(aes(xmin=-5, xmax=15, ymin=205, ymax=220), fill="white") +scale_y_continuous(limits=c(0,NA), breaks=trans(yticks), labels=yticks) 
#   
# ggplot(classification_genes_all,aes(x=lineage,y=Freq_t,fill=specific_class))+geom_col( stat="identity", position = position_dodge(width = 0), width = 3.5)+theme_classic()+facet_grid(~species, scales = "free", space = "free")+theme(legend.title = element_blank())+scale_fill_manual(values = colorsss)+geom_rect(aes(xmin=-1, xmax=15, ymin=205, ymax=220), fill="white") +scale_y_continuous(limits=c(0,NA), breaks=trans(yticks), labels=yticks) 
 gene_plot
  pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/gene_count_plot.pdf",width=16,height=6)

gene_plot

dev.off()

##--------------------------STATS


classification_genes_all %>% group_by(Class) %>% summarise(mean=mean(Freq),std=sd(Freq))


```



## protein clustering 

Here, I cluster all protein sequences with mmseqs2

```{bash}

###---------------------------------------
##create protein database
###---------------------------------------

##==========================================
###-----------------bring all gffs together
##==========================================
species=Sterm

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/For_MMseqs2/all_ldels_sterms/GFFs/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/For_MMseqs2/all_ldels_sterms/Faas/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/For_MMseqs2/all_ldels_sterms/GFFs/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/For_MMseqs2/all_ldels_sterms/Faas/


species=Ldel

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/For_MMseqs2/all_ldels_sterms/GFFs/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/For_MMseqs2/all_ldels_sterms/Faas/


for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_GFF/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/
done

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/GFF/${species}/Out-sali.gff
##==========================================
###-----------------panaroo
##==========================================

##-----sterm

species=Sterm

conda activate panaroo

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species}_references/01_Panarooo/
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo

panaroo -t 37 -i //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF//*.gff -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species}_references/01_Panarooo/ --clean-mode strict





###---------------------------------------
##cluster
###---------------------------------------
mmseqs easy-cluster examples/DB.fasta clusterRes tmp --min-seq-id 0.5 -c 0.8 --cov-mode 1

```

##Gene/Species tree

#### Xavier Didelots approach

example from [staph aureus](https://xavierdidelot.github.io/BactDating/articles/Staph.html#initialisation-1).
```{r}
##-----------------------------
#test
##-----------------------------

library(BactDating)
library(ape)
set.seed(0)
data(staph)
plot(staph$tree, show.tip.label = F)
axisPhylo(backward = F)


### dates in the same order as tip.label  
staph$tree$tip.label
staph$dates
staph$tree


res=roottotip(staph$tree,staph$dates)

##----------run Dactdate
res=bactdate(unroot(staph$tree),staph$dates,nbIts=1000)
plot(res,'treeCI',show.tip.label = F)

plot(res,'trace')


plot(res,'treeRoot',show.tip.label=F)


```

[This tutorial ](https://xavierdidelot.github.io/BactDating/articles/yourData.html) will enable us to do the analysis on our data

##### 0. Meta-data

For Sterm I already have a metadata file. However for Ldel (or Lhelv) not yet.
In the end I do it manually!!!!!!!!!

```{r}
library(rentrez)
taxize_summ <- entrez_summary(db="pubmed", id=24555091)
taxize_summ <- entrez_summary(db="biosample",  id=2442656)

taxize_summ$sampledata


# entrez_summary()


entrez_dbs()


```


```{bash}
##information gathered earlier

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt

##---------------------fetch info

head -1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt |cut -f 4 > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.tmp


efetch -db biosample -id SAMN02598530 -format xml | xtract -pattern BioSampleSet -division BioSample -group Attributes -element Attribute |grep "collection"
efetch -db biosample -id SAMN02598530 -format xml | xtract -pattern BioSampleSet -division BioSample -group Attributes -element Attribute 
efetch -db biosample -id SAMN02442656 -format xml | xtract -pattern BioSampleSet -division BioSample -group Attributes -element collection_date 


collection date



efetch -db biosample -id SAMN02598530 -format xml | xtract -pattern BioSampleSet -division BioSample -group Attributes -element Attribute |grep "collection"


efetch -db biosample -input /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.tmp -format xml


epost -db biosample -input /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.tmp -format acc |elink -target biosample| 

epost -db biosample -input /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.tmp -format acc | \
elink -target sra| \
efetch -db sra -format runinfo -mode xml | \
xtract -pattern Row -def "NA" -element Run spots bases spots_with_mates avgLength \
size_MB download_path Experiment LibraryStrategy LibrarySelection LibrarySource \
LibraryLayout InsertSize InsertDev Platform Model SRAStudy BioProject ProjectID \
Sample BioSample SampleType TaxID ScientificName SampleName CenterName \
Submission Consent > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp_03


 wget -q -O - "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=biosample&id=SAMN04383980" |\
   xmllint --xpath "/BioSampleSet/BioSample/Attributes/Attribute[@attribute_name='source_name']/text()"  -
   
   
 wget -q -O - "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=biosample&id=SAMEA6957293" |\
   xmllint --xpath "/BioSampleSet/BioSample/Attributes/Attribute[@attribute_name='source_name']/text()"  -


 wget -q -O - "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=biosample&id=SAMEA418087" |\
   xmllint --xpath "/BioSampleSet/BioSample/Attributes/Attribute[@attribute_name='source_name']/text()"  -



 wget -q -O - "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=biosample&id=SAMN02598530" |\
   xmllint --xpath "/BioSampleSet/BioSample/Attributes/Attribute[@attribute_name='source_name']/text()"  -


```



##### 1. create a starting tree


1.orthofinder

```{bash}
###-----------------------------------------
##Lhelv
###-----------------------------------------

species=Lhelv


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

#leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/


rm -r  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/
cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Lhelv

for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

#leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
done

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/Out-sali.faa

###-----------------------------------------
##remove outgroups and meta
###-----------------------------------------

species=Sterm
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/
mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/Out-vest.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/
mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/S202SMAG.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/  
mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/S50.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/  
mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/S72.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/  
  
species=Ldel
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/
mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/Out-helv.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/
mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/S202SMAG.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/   
  


###-----------------------------------------
##orthorfinder
###-----------------------------------------
#for species in $(echo "Ldel Sterm"|tail -1)
species=Lhelv
mkdir -p    
#cp -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/Ldel/OrthoFinder/Results_Feb11/ /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/

for species in $(echo "Sterm Ldel")
do

cd  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/


    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/ \
      -t 32 \
      -o    /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own//${species}/ \
      -a 32

done # species


```

1.Kirstin's approach


```{bash}
date=Results_Mar11
species=Sterm
date=Results_Mar16
species=Lhelv
#/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/Results_Jul28/
for species in $(echo "Ldel Sterm"|tail -1)
do
echo "==========================================="
echo -e ${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/get_singlecp_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt

wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt
wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt

done
```

Here, I make a directory containing the fna and faa files of all genomes with the file name corresponding to the locus tag. 

```{bash}


for species in $(echo "Ldel Sterm")
do
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

leterzzz=$(echo $species |head -c 1)

#cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

#cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA/
done
  
###--------------------
rm  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss_extended_02.txt
for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/ |grep ".faa$"|sed 's/.faa//g'|grep -v "Out-sali")
do
locusTagsss=$(head -1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/${genomesss}.faa|cut -d '_' -f 1|sed 's/>//g')
echo -e "This  genome : " ${genomesss} " has the following locus Tag : "${locusTagsss}
echo -e ${genomesss}"\t"${locusTagsss} >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss_extended_02.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/${genomesss}.faa > \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/${genomesss}.ffn > \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.ffn


done
done

##------=================================================================
#extract orthogroups
##------=================================================================

for species in $(echo "Ldel Sterm"|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/extract_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt --folder ${species}

done
```

align protein families

```{bash}


for species in $(echo "Ldel Sterm"|tail -1)
do

cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep ".faa$" |sed 's/.faa//g'|grep -v "Out-sali" )
do
rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.aln.fasta
mafft --thread 37 --maxiterate 1000  --auto /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.faa > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}_aln.fasta

done 

done #species
```
here I back translate the faa alignment to dna.

```{bash}

for species in $(echo "Ldel Sterm"|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/aln_aa_to_dna.pl

done

```

With perl: prune alignments (removing columns represented by less than 50% of the sequences)


```{bash}


for species in $(echo "Ldel Sterm"|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/prune_aln.pl

done

##================================================================
#remove all infomration (except genome info) from multifasta header
##================================================================

for species in $(echo "Ldel Sterm"|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i "s/_0.*//" ${genesss}_aln_prune.fasta 

done
done


##-------------ldel names too long --> shorten

species=Ldel
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/-DSM-/-/g' ${genesss}_aln_prune.fasta 
sed -i 's/-KCTC-/-/g' ${genesss}_aln_prune.fasta 
done


```
With perl: Concatenate pruned alignments
```{bash}
  
#  for species in $(echo "Ldel Sterm"|tail -1)
  for species in $(echo "Sterm Ldel"|tail -1)
  do
  echo -e ${species}
  cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
  perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/cat_align.pl > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}.phylip
  
  done
  
    for species in $(echo "Sterm Ldel"|tail -1)
  do
  echo -e ${species}
  cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
  perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/cat_align_2fasta.pl > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta
  
  done
  
```
With RAxML: infer the phylogeny
```{bash}
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/

  for species in $(echo "Ldel Sterm"|tail -1)
    for species in $(echo "Ldel"|tail -1)

#   for species in $(echo "Sterm"|tail -1)
  do
  echo -e ${species}
  mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/${species}
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/${species}
sed 's/\/.*//g' /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta

rm -r *${species}_all*

#/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta -n ${species}_all -T 37
/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta -n ${species}_all -T 37

done
```

prepare tip.names file


```{bash}
rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/names.strains.txt
for species in $(echo "Sterm Ldel"|tail -1)
do
grep ">" /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta| sed 's/\/.*//g'| sed 's/>//g'  |awk -v namezzz="$species" '{OFS="\t"}{print namezzz,$1}' >> /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/names.strains.txt
seq_length.py /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta
done
```



calculate kappa============================


like [here](https://github.com/xavierdidelot/ClonalFrameML/issues/95)


Sterm-----------------------

alpha: 0.020000
Tree-Length: 0.268312
rate A <-> C: 1.264669
rate A <-> G: 3.668788
rate A <-> T: 0.752548
rate C <-> G: 0.588894
rate C <-> T: 5.357598
rate G <-> T: 1.000000

freq pi(A): 0.292688
freq pi(C): 0.189221
freq pi(G): 0.220427
freq pi(T): 0.297664


```{r}
(3.668788+5.357598)/(1.264669+0.752548+0.588894+1.000000)
```
Ldel-----------------------
Tree-Length: 0.104705
rate A <-> C: 1.197734
rate A <-> G: 3.026860
rate A <-> T: 0.483401
rate C <-> G: 0.491727
rate C <-> T: 5.401485
rate G <-> T: 1.000000

freq pi(A): 0.253359
freq pi(C): 0.264227
freq pi(G): 0.269964
freq pi(T): 0.212450
```{r}
(3.026860+5.401485)/(1.197734+0.483401+0.491727+1.000000)
```

Lhelv-----------------------
alpha: 0.020000
Tree-Length: 0.044759
rate A <-> C: 1.446084
rate A <-> G: 3.449104
rate A <-> T: 0.727126
rate C <-> G: 0.814041
rate C <-> T: 4.536195
rate G <-> T: 1.000000

freq pi(A): 0.320335
freq pi(C): 0.168171
freq pi(G): 0.209049
freq pi(T): 0.302445

```{r}
(3.449104+4.536195)/(1.446084+0.727126+0.814041+1.000000)
```



##### 2. create a ClonalFrameML phylogeny

```{bash}


conda create --name ClonalFrameML

conda install -c conda-forge -c bioconda -c defaults clonalframeml

conda activate ClonalFrameML

ClonalFrameML newick_file seq_file output_prefix [OPTIONS]

##-----------------Sterm-----------------------------

phylip2fasta.sh in=/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}.phylip out=/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}.fasta


conda activate ClonalFrameML

date=Results_Mar11
species=Sterm

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML

sed 's/\/.*//g' /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta

seq_length.py /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta


ClonalFrameML /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/${species}/RAxML_bestTree.${species}_all \
       /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta  \
      /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML/${species}_clonal_tree -emsim 100  -kappa 2.50308 > \
      /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML/${species}_clonal_tree.log.txt


##-----------------ldel-----------------------------


conda activate ClonalFrameML

date=Results_Mar11
species=Ldel

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML

sed 's/\/.*//g' /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta

seq_length.py /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta


ClonalFrameML /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/${species}/RAxML_bestTree.${species}_all \
       /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta  \
      /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML/${species}_clonal_tree -emsim 100  -kappa 2.50308 > \
      /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML/${species}_clonal_tree.log.txt


##-----------------Lhelv-----------------------------


conda activate ClonalFrameML

date=Results_Mar16
species=Lhelv

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML

sed 's/\/.*//g' /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta > /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta

seq_length.py /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta


ClonalFrameML /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/${species}/RAxML_bestTree.${species}_all \
       /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta  \
      /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML/${species}_clonal_tree -emsim 100  -kappa 2.002708 > \
      /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML/${species}_clonal_tree.log.txt


```

move local

```{bash}



mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/


scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/names.strains.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//

scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm/ClonalFrameML/Sterm_clonal_tree* /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//

scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Ldel/ClonalFrameML/Ldel_clonal_tree* /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//

scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Lhelv/ClonalFrameML/Lhelv_clonal_tree* /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//

```


##### 3. BactDating

```{r}
##-------------------------------Sterm
library(ape)
library(BactDating)
library(ggtree)
library(phytools)
library(treeio)
library(tidyverse)
library(coda)

t=loadCFML("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_clonal_tree")

# plot(t, show.tip.label = T)
# axisPhylo(backward = F)

##------------import names

# names_strains <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/names.strains.txt", 
#     delim = "\t", escape_double = FALSE, 
#     col_names = c("Species","strain"), trim_ws = TRUE)

##------------import metaData
meta_sterm <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited_extended_02.csv")
# colnames(meta_sterm)
meta_sterm$shortName_new <- meta_sterm$shortName %>% gsub("^","S",.) %>% gsub("^SS","S",.)
# names_strains$strain

# t$tip.label[!t$tip.label %in% meta_sterm$shortName_new]

meta_sterm_onlySterm <- meta_sterm[which(meta_sterm$shortName_new %in% t$tip.label),]

meta_sterm_onlySterm <-meta_sterm_onlySterm[match(t$tip.label,meta_sterm_onlySterm$shortName_new),]

d <- meta_sterm_onlySterm$extraction_time
names(d) <- meta_sterm_onlySterm$shortName_new

##------------make dated tree

rooted=initRoot(t,d)
r=roottotip(rooted,d)
# r=roottotip(t,d)

##------------running calculation
res=bactdate(unroot(t),d,nbIts=100000,showProgress = TRUE)

##-----------------------plots
 png("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_sterm_roottotip.png", width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
r=roottotip(rooted,d)

dev.off()

 png("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_sterm_trace.png", width = 4000, height = 3000,res=300)
plot(res,'trace')
dev.off()


 png("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_sterm_treeCI.png", width = 4000, height = 3000,res=300)
plot(res,'treeCI',show.tip.label = F)
dev.off()


l=as.treedata.resBactDating(res)
obj=methods::new('treedata',phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

ggtree(obj, right=TRUE, mrsd="2020-04-02") + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)


write.beast(obj,'/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_phylo_beast.txt')


mcmc=as.mcmc.resBactDating(res)
coda::effectiveSize(mcmc)
# mcmc_02 <- mcmc
save(mcmc, file= "/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_model.rda")

 load('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_model.rda')
 coda::effectiveSize(mcmc)


```

move files to bigmachine

```{bash}



mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm//{input,script,output}

scp /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_clonal_tree* vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm/input/

scp ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited_extended_02.csv vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm//input/

```


```{r}
###------------------------------------------
#running on bigmachine
##saved in /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm/script/Sterm_script_01.rscript
# cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm/script/
#run: R CMD BATCH /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm/script/Sterm_script_01.rscript

###------------------------------------------

set_rounds=100000000
speciesss="Sterm"
trial="v1"
##-------------------------------Sterm
library(ape)
library(BactDating)
library(ggtree)
library(phytools)
library(treeio)
library(tidyverse)
library(coda)

t=loadCFML("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"//input//Sterm_clonal_tree")

##------------import metaData
meta_sterm <- read_csv("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"//input//20200728_NCBI_log_shortNames_final_sterm_edited_extended_02.csv")

meta_sterm$shortName_new <- meta_sterm$shortName %>% gsub("^","S",.) %>% gsub("^SS","S",.)

meta_sterm_onlySterm <- meta_sterm[which(meta_sterm$shortName_new %in% t$tip.label),]

meta_sterm_onlySterm <-meta_sterm_onlySterm[match(t$tip.label,meta_sterm_onlySterm$shortName_new),]

d <- meta_sterm_onlySterm$extraction_time
names(d) <- meta_sterm_onlySterm$shortName_new

##------------make dated tree

rooted=initRoot(t,d)
# r=roottotip(rooted,d)
# r=roottotip(t,d)

##------------running calculation
res=bactdate(unroot(t),d,nbIts=set_rounds,showProgress = TRUE)

##-----------------------plots


 png(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_roottotip.png"), width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
r=roottotip(rooted,d)

dev.off()

 png(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_trace.png"), width = 4000, height = 3000,res=300)
plot(res,'trace')
dev.off()


 png(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_treeCI.png"), width = 4000, height = 3000,res=300)
plot(res,'treeCI',show.tip.label = F)
dev.off()




mcmc=as.mcmc.resBactDating(res)
coda::effectiveSize(mcmc)
# mcmc_02 <- mcmc

paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_model.rda")

save(mcmc, file= paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_model.rda"))

 load(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_model.rda"))
text <-  coda::effectiveSize(mcmc)


l=as.treedata.resBactDating(res)
obj=methods::new('treedata',phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

tree_plot <- ggtree(obj, right=TRUE, mrsd="2020-04-02") + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = paste("mu=",round(text[1],digits=4),", sigma=",round(text[2],digits=4),", alpha=",round(text[3],digits=4)))


write.beast(obj,paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_phylo_beast.txt"))

 png(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_final_tree.png"), width = 4000, height = 3000,res=300)
tree_plot
dev.off()
 
```

transfer local

```{bash}

scp -r vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm/output/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm/

```

Ldel

```{r}
##-------------------------------Ldel
library(ape)

library(BactDating)

t=loadCFML("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Ldel_clonal_tree")

t$tip.label


plot(t, show.tip.label = T)
axisPhylo(backward = F)

##------------import names

library(tidyverse)
# names_strains <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/names.strains.txt", 
#     delim = "\t", escape_double = FALSE, 
#     col_names = c("Species","strain"), trim_ws = TRUE)

##------------import metaData
meta_sterm <-  read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Ldel_phylogeny_info.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)
colnames(meta_sterm)
meta_sterm$shortName_new <- meta_sterm$shortName %>% gsub("-DSM-","-",.) %>% gsub("-KCTC-","-",.)
# names_strains$strain


# sed -i 's/-DSM-/-/g' ${genesss}_aln_prune.fasta 
# sed -i 's/-KCTC-/-/g' ${genesss}_aln_prune.fasta 

t$tip.label[!t$tip.label %in% meta_sterm$shortName_new]

meta_sterm_onlySterm <- meta_sterm[which(meta_sterm$shortName_new %in% t$tip.label),]

meta_sterm_onlySterm <-meta_sterm_onlySterm[match(t$tip.label,meta_sterm_onlySterm$shortName_new),]


d <- meta_sterm_onlySterm$extraction_time
names(d) <- meta_sterm_onlySterm$shortName_new

##------------make dated tree

rooted=initRoot(t,d)
r=roottotip(t,d)

r=roottotip(rooted,d)

plot(rooted, show.tip.label = T)
axisPhylo(backward = F)

##------------plot
res=bactdate(unroot(t),d,nbIts=10000000,showProgress = TRUE)

# res=bactdate(obsphy,dates,updateRoot=F,model='mixedgamma')


# plot(res,'treeCI',show.tip.label = F)
# 
# plot(res,'trace')
# 
# library(mcmcr)
# mcmc=as.mcmc(res)
# library(coda)
# 
# coda::save(mcmc, file= "/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_tree_100000_01")
# model = load("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_tree_100000_01")
# 
# coda
##-----------------------plots
 png("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_ldel_roottotip.png", width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
r=roottotip(rooted,d)

dev.off()

 png("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_ldel_trace.png", width = 4000, height = 3000,res=300)
plot(res,'trace')
dev.off()


 png("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_ldel_treeCI.png", width = 4000, height = 3000,res=300)
plot(res,'treeCI',show.tip.label = F)
dev.off()


plot(res,'treeCI',show.tip.label = F)


mcmc=as.mcmc(res)


library(ggtree)
library(phytools)

library(treeio)

l=as.treedata.resBactDating(res)
obj=methods::new('treedata',phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

ggtree(obj, right=TRUE, mrsd="2020-04-02") + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)

write.beast(obj,'/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel_phylo_beast.txt')


myModel.coda <- read.bugs(myModel)

coda::write.model(mcmc,'/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ldel_model.txt')

save(mcmc, file= "/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ldel_model.txt")
mcmc_02 = load("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ldel_model.txt")


mcmc=as.mcmc.resBactDating(res)
coda::effectiveSize(mcmc)
effectiveSize(mcmc)
mcmc_02 <- mcmc
save(mcmc_02, file= "/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ldel_model.rda")

 load('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ldel_model.rda')
coda::effectiveSize(mcmc_02)



l=as.treedata.resBactDating(res)
obj=methods::new('treedata',phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

ggtree(obj, right=TRUE, mrsd="2020-04-02") + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)


print(res$pstrict)


# modelcompare()

```

       mu     sigma     alpha 
10.366933 56.111430  4.878684 


move files to bigmachine

```{bash}

mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel//{input,script,output}

scp /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Ldel_clonal_tree* vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel/input/

scp ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Ldel_phylogeny_info.tsv vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel//input/

```


```{r}
###------------------------------------------
#running on bigmachine
##saved in /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel/script/Ldel_script_01.rscript
# cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel/script/
#run: R CMD BATCH /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel/script/Ldel_script_01.rscript

###------------------------------------------

set_rounds=1000000000
speciesss="Ldel"
trial="v1"
##-------------------------------Sterm
library(ape)
library(BactDating)
library(ggtree)
library(phytools)
library(treeio)
library(tidyverse)
library(coda)

t=loadCFML(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"//input//Ldel_clonal_tree"))

##------------import metaData

meta_sterm <-  read_delim(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"//input//Ldel_phylogeny_info.tsv"),  delim = "\t", escape_double = FALSE,  trim_ws = TRUE)

meta_sterm$shortName_new <- meta_sterm$shortName %>% gsub("-DSM-","-",.) %>% gsub("-KCTC-","-",.)

meta_sterm_onlySterm <- meta_sterm[which(meta_sterm$shortName_new %in% t$tip.label),]

meta_sterm_onlySterm <-meta_sterm_onlySterm[match(t$tip.label,meta_sterm_onlySterm$shortName_new),]

d <- meta_sterm_onlySterm$extraction_time
names(d) <- meta_sterm_onlySterm$shortName_new

##------------make dated tree

rooted=initRoot(t,d)
# r=roottotip(rooted,d)
# r=roottotip(t,d)

##------------running calculation
res=bactdate(unroot(t),d,nbIts=set_rounds,showProgress = TRUE)

##-----------------------plots


 png(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_roottotip.png"), width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
r=roottotip(rooted,d)

dev.off()

 png(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_trace.png"), width = 4000, height = 3000,res=300)
plot(res,'trace')
dev.off()


 png(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_treeCI.png"), width = 4000, height = 3000,res=300)
plot(res,'treeCI',show.tip.label = F)
dev.off()




mcmc=as.mcmc.resBactDating(res)
coda::effectiveSize(mcmc)
# mcmc_02 <- mcmc

paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_model.rda")

save(mcmc, file= paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_model.rda"))

 load(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_model.rda"))
text <-  coda::effectiveSize(mcmc)


l=as.treedata.resBactDating(res)
obj=methods::new('treedata',phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

tree_plot <- ggtree(obj, right=TRUE, mrsd="2020-04-02") + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = paste("mu=",round(text[1],digits=4),", sigma=",round(text[2],digits=4),", alpha=",round(text[3],digits=4)))


write.beast(obj,paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_phylo_beast.txt"))

 png(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_final_tree.png"), width = 4000, height = 3000,res=300)
tree_plot
dev.off()
 
```

transfer local

```{bash}
mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel/
scp -r vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel/output/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel/

```


Lhelv

```{r}
##-------------------------------Lhelv
library(ape)
library(tidyverse)
library(BactDating)
library(mcmcr)



t=loadCFML("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Lhelv_clonal_tree")

t$tip.label
t$edge.length
# 
# L=896286
# t$edge.length/L
# t$unrec
# 
# r$rate/L

plot(t, show.tip.label = T)
axisPhylo(backward = F)

##------------import names

library(tidyverse)
meta_sterm <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt",  delim = "\t", escape_double = FALSE,) %>% mutate(shortName_new=locusTAG)
meta_sterm_names <- meta_sterm %>% select(strain_name,shortName_new)

meta_sterm <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned.txt")  %>% left_join(.,meta_sterm_names, by="strain_name")

##------------import metaData
# meta_sterm <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/")
# colnames(meta_sterm)
# meta_sterm$shortName_new <- meta_sterm$shortName %>% gsub("^","S",.) %>% gsub("^SS","S",.)
# names_strains$strain

t$tip.label[!t$tip.label %in% meta_sterm$shortName_new]

meta_sterm_onlySterm <- meta_sterm[which(meta_sterm$shortName_new %in% t$tip.label),]

meta_sterm_onlySterm <-meta_sterm_onlySterm[match(t$tip.label,meta_sterm_onlySterm$shortName_new),]


d <- meta_sterm_onlySterm$isolation_date
names(d) <- meta_sterm_onlySterm$shortName_new

##------------make dated tree

# res=bactdate(t,d)

# res=bactdate(t,d,nbIts=1e5) #increase number of cycles


rooted=initRoot(t,d)
r=roottotip(rooted,d)
r=roottotip(t,d)

##------------plot
res=bactdate(unroot(t),d,nbIts=10000000,showProgress = TRUE)
# plot(res,'treeCI',show.tip.label = F)
# 
# plot(res,'trace')
# 
# library(mcmcr)
# mcmc=as.mcmc(res)
# library(coda)
# 
# coda::save(mcmc, file= "/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_tree_100000_01")
# model = load("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_tree_100000_01")
# 
# coda
##-----------------------plots
 png("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_Lhelv_roottotip.png", width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
r=roottotip(rooted,d)

dev.off()

 png("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_Lhelv_trace.png", width = 4000, height = 3000,res=300)
plot(res,'trace',show.axis = TRUE)
dev.off()
# plot(res,'treeRoot',)


 png("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_Lhelv_treeCI.png", width = 4000, height = 3000,res=300)
plot(res,'treeCI',show.tip.label = F,)
dev.off()


# plotDualScale(res)

# plot
# plot(res,'treeCI',show.tip.label = TRUE,show.axis = TRUE)

# unlist(res)
res$mu
mcmc=as.mcmc(res)
mcmc$model
mcmc$record

as.mcmc.resBactDating(res)
as.phylo(res)

tressss <- mcmc$tree %>% as.phylo()

library(ggtree)
library(phytools)

library(treeio)

l=as.treedata.resBactDating(res)
obj=methods::new('treedata',phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

ggtree(obj, right=TRUE, mrsd="2020-04-02") + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)

write.beast(obj,'/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_phylo_beast.txt')

library(coda)
# mcmc = coda::as.mcmc(res)
# coda::effectiveSize(mcmc)

##----------output mcmc model 

myModel.coda <- read.bugs(myModel)

coda::write.model(mcmc,'/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_model.txt')

save(mcmc, file= "/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_model.txt")
mcmc_02 = load("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_model.txt")


mcmc=as.mcmc.resBactDating(res)
coda::effectiveSize(mcmc)
effectiveSize(mcmc)
mcmc_02 <- mcmc
save(mcmc_02, file= "/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_model.rda")

 load('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_model.rda')
 effectiveSize(mcmc_02)

# 
# # 
# summary(res)
# tressss_02 <- as.treedata.resBactDating(res)


# library(treedata)
# library(ggtree)
# res=bactdate(...)
# typeof(tressss_02)
# 
# 
# typeof(tressss)
# 
# tressss$tip.label
# meta_sterm_onlySterm$shortName_new
# 
# tressss$tip.label <- meta_sterm_onlySterm$strain_name
# 
# ggtree(tressss)+ theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom="text")
# 
# ape::write.tree(tressss, file='/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_phylo.txt')




# p12 <- ggtree(tressss)+ geom_tiplab(align=TRUE, linesize=.5,geom="text")
# p12
# 
# # beast_file <- system.file("examples/MCC_FluA_H3.tree", 
#                           # package="ggtree")
# beast_tree <- read.beast(mcmc)
# ggtree(beast_tree, mrsd="2013-01-01") + theme_tree2()
# 

```


These things should be checked:
-  Effective Sample Size (ESS)  > 100 (https://github.com/xavierdidelot/BactDating/issues/53) with the following function *effectiveSize(mcmc)*
- output mcmc model with bugs 

ESS for Lhelv:
     mu     sigma     alpha 
 15.15673 259.35243  40.90496
 
        mu     sigma     alpha 
  6.71606 501.00000  11.15303 
 

move files to bigmachine

```{bash}




mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv//{input,script,output}

scp /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Lhelv_clonal_tree* vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv/input/

scp ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended* vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv//input/

```


```{r}
###------------------------------------------
#running on bigmachine
##saved in /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv/script/Lhelv_script_01.rscript
# cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv/script/
#run: R CMD BATCH /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv/script/Lhelv_script_01.rscript

###------------------------------------------

set_rounds=10000000
speciesss="Lhelv"
trial="v1"
##-------------------------------Sterm
library(ape)
library(BactDating)
library(ggtree)
library(phytools)
library(treeio)
library(tidyverse)
library(coda)

t=loadCFML(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"//input//Lhelv_clonal_tree"))

##------------import metaData
library(tidyverse)



meta_sterm <- read_delim(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"//input//Lhelv_Refsss_extended_cleaned_02.txt"),  delim = "\t", escape_double = FALSE,) %>% mutate(shortName_new=locusTAG)
meta_sterm_names <- meta_sterm %>% select(strain_name,shortName_new)

meta_sterm <- read_csv(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"//input//Lhelv_Refsss_extended_cleaned.txt")) %>% left_join(.,meta_sterm_names, by="strain_name")

#meta_sterm$shortName_new <- meta_sterm$shortName  #%>% gsub("^","S",.) %>% gsub("^SS","S",.)

meta_sterm_onlySterm <- meta_sterm[which(meta_sterm$shortName_new %in% t$tip.label),]

meta_sterm_onlySterm <-meta_sterm_onlySterm[match(t$tip.label,meta_sterm_onlySterm$shortName_new),]

d <- meta_sterm_onlySterm$isolation_date
names(d) <- meta_sterm_onlySterm$shortName_new

##------------make dated tree

rooted=initRoot(t,d)
# r=roottotip(rooted,d)
# r=roottotip(t,d)

##------------running calculation
res=bactdate(unroot(t),d,nbIts=set_rounds,showProgress = TRUE)

##-----------------------plots


 png(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_roottotip.png"), width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
r=roottotip(rooted,d)

dev.off()

 png(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_trace.png"), width = 4000, height = 3000,res=300)
plot(res,'trace')
dev.off()


 png(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_treeCI.png"), width = 4000, height = 3000,res=300)
plot(res,'treeCI',show.tip.label = F)
dev.off()


mcmc=as.mcmc.resBactDating(res)
coda::effectiveSize(mcmc)
# mcmc_02 <- mcmc

paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_model.rda")

save(mcmc, file= paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_model.rda"))

 load(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_model.rda"))
text <-  coda::effectiveSize(mcmc)


l=as.treedata.resBactDating(res)
obj=methods::new('treedata',phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

tree_plot <- ggtree(obj, right=TRUE, mrsd="2020-04-02") + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = paste("mu=",round(text[1],digits=4),", sigma=",round(text[2],digits=4),", alpha=",round(text[3],digits=4)))

write.beast(obj,paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_phylo_beast.txt"))

 png(paste0("/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/",speciesss,"/output/PLOTS_",speciesss,"_rounds",set_rounds,"_",trial,"_final_tree.png"), width = 4000, height = 3000,res=300)
tree_plot
dev.off()
 
```

transfer local

```{bash}
mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv/
scp -r vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv/output/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv/

```

##### 4. Plotting


```{r}
library(ape)
library(BactDating)
library(ggtree)
library(phytools)
library(treeio)
library(tidyverse)
library(coda)
library(philr)
##----------------Sterm

# myTree_Sterm_beast <- read.beast('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_phylo_beast.txt')

# myTree_Sterm_beast <- read.beast('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm/output/PLOTS_Sterm_rou')

myTree_Sterm_beast <- read.beast('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_phylo_beast.txt')
tree_myTree_Sterm_beast <- ggtree(myTree_Sterm_beast, right=TRUE, mrsd="2020-04-02") + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "S.thermophilus")+ coord_cartesian(clip = 'off')
tree_myTree_Sterm_beast

##----------------Ldel
# myTree_ldel_beast <- read.beast('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel_phylo_beast.txt')


myTree_ldel_beast <- read.beast('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel/output/PLOTS_Ldel_rounds1e+07_v1_phylo_beast.txt')
tree_myTree_ldel_beast <- ggtree(myTree_ldel_beast, right=TRUE, mrsd="2020-04-02") + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "L.delbrueckii",x="Dating (year)")+ coord_cartesian(clip = 'off')



##----------------Lhelv
myTree_Lhelv_beast <- read.beast('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_phylo_beast.txt')

# myTree_Lhelv_beast <- read.beast('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv/output/PLOTS_Lhelv_rounds1e+05_v1_phylo_beast.txt')
gtree_myTree_Lhelv_beast <- ggtree(myTree_Lhelv_beast, right=TRUE, mrsd="2020-04-02")+ theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "L.helveticus")+ coord_cartesian(clip = 'off')
gtree_myTree_Lhelv_beast
##====================combine

library(patchwork)


tree_myTree_Sterm_beast+tree_myTree_ldel_beast+gtree_myTree_Lhelv_beast

 
 pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/dated_phylogenies.pdf",width=10,height=6)

tree_myTree_Sterm_beast+tree_myTree_ldel_beast+gtree_myTree_Lhelv_beast


dev.off()


###-------------------------------------------------
##calculation of rootage and varation
###-------------------------------------------------

#----------Sterm
myTree_Sterm_beast_phylo <- myTree_Sterm_beast %>% as.phylo()
myTree_Sterm_beast_tibble <- myTree_Sterm_beast %>% as_tibble() 

myTree_Sterm_beast_tibble$branchAge_low <- NA
myTree_Sterm_beast_tibble$branchAge_high <- NA

for (listElement in 1:length(myTree_Sterm_beast_tibble$height_0.95_HPD)) {
  
myTree_Sterm_beast_tibble[listElement,"branchAge_low"] <-    2020-as.numeric(myTree_Sterm_beast_tibble$height_0.95_HPD[[listElement]][1])
myTree_Sterm_beast_tibble[listElement,"branchAge_high"] <-   2020-as.numeric(myTree_Sterm_beast_tibble$height_0.95_HPD[[listElement]][2])

}

myTree_Sterm_beast_tibble$branchAge_calculated <- rowMeans(myTree_Sterm_beast_tibble[,c('branchAge_high', 'branchAge_low')], na.rm=TRUE)
myTree_Sterm_beast_tibble$branchAge_variation <-  myTree_Sterm_beast_tibble$branchAge_calculated - myTree_Sterm_beast_tibble$branchAge_low

info <- as_tibble(myTree_Sterm_beast_phylo)
distance <- mean_dist_to_tips(myTree_Sterm_beast_phylo)

distance <- mean_dist_to_tips(myTree_Sterm_beast_phylo) %>% as.data.frame()%>% rownames_to_column("label") %>% as_tibble() %>% mutate(branchAge_calculated_02 = 2020-.)  %>% left_join(.,info)%>% select(node,branchAge_calculated_02)

myTree_Sterm_beast_tibble_extended <- left_join(myTree_Sterm_beast_tibble,distance,by="node") %>% mutate(species="S. thermophilus")%>% select(-height_0.95_HPD,-length_0.95_HPD)
write.csv(myTree_Sterm_beast_tibble_extended,"/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_tree_age.csv")

#----------Ldel
myTree_Sterm_beast_phylo <- myTree_ldel_beast %>% as.phylo()
myTree_Sterm_beast_tibble <- myTree_ldel_beast %>% as_tibble() 

myTree_Sterm_beast_tibble$branchAge_low <- NA
myTree_Sterm_beast_tibble$branchAge_high <- NA

for (listElement in 1:length(myTree_Sterm_beast_tibble$height_0.95_HPD)) {
  
myTree_Sterm_beast_tibble[listElement,"branchAge_low"] <-    2020-as.numeric(myTree_Sterm_beast_tibble$height_0.95_HPD[[listElement]][1])
myTree_Sterm_beast_tibble[listElement,"branchAge_high"] <-   2020-as.numeric(myTree_Sterm_beast_tibble$height_0.95_HPD[[listElement]][2])

}

myTree_Sterm_beast_tibble$branchAge_calculated <- rowMeans(myTree_Sterm_beast_tibble[,c('branchAge_high', 'branchAge_low')], na.rm=TRUE)
myTree_Sterm_beast_tibble$branchAge_variation <-  myTree_Sterm_beast_tibble$branchAge_calculated - myTree_Sterm_beast_tibble$branchAge_low

info <- as_tibble(myTree_Sterm_beast_phylo)
distance <- mean_dist_to_tips(myTree_Sterm_beast_phylo)

distance <- mean_dist_to_tips(myTree_Sterm_beast_phylo) %>% as.data.frame()%>% rownames_to_column("label") %>% as_tibble() %>% mutate(branchAge_calculated_02 = 2020-.)  %>% left_join(.,info)%>% select(node,branchAge_calculated_02)

myTree_Ldel_beast_tibble_extended <- left_join(myTree_Sterm_beast_tibble,distance,by="node") %>% mutate(species="L. delbrueckii")%>% select(-height_0.95_HPD,-length_0.95_HPD)
write.csv(myTree_Ldel_beast_tibble_extended,"/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel_tree_age.csv")
#----------Lhelv
myTree_Sterm_beast_phylo <- myTree_Lhelv_beast %>% as.phylo()
myTree_Sterm_beast_tibble <- myTree_Lhelv_beast %>% as_tibble() 

myTree_Sterm_beast_tibble$branchAge_low <- NA
myTree_Sterm_beast_tibble$branchAge_high <- NA

for (listElement in 1:length(myTree_Sterm_beast_tibble$height_0.95_HPD)) {
  
myTree_Sterm_beast_tibble[listElement,"branchAge_low"] <-    2020-as.numeric(myTree_Sterm_beast_tibble$height_0.95_HPD[[listElement]][1])
myTree_Sterm_beast_tibble[listElement,"branchAge_high"] <-   2020-as.numeric(myTree_Sterm_beast_tibble$height_0.95_HPD[[listElement]][2])

}

myTree_Sterm_beast_tibble$branchAge_calculated <- rowMeans(myTree_Sterm_beast_tibble[,c('branchAge_high', 'branchAge_low')], na.rm=TRUE)
myTree_Sterm_beast_tibble$branchAge_variation <-  myTree_Sterm_beast_tibble$branchAge_calculated - myTree_Sterm_beast_tibble$branchAge_low

info <- as_tibble(myTree_Sterm_beast_phylo)
distance <- mean_dist_to_tips(myTree_Sterm_beast_phylo)

distance <- mean_dist_to_tips(myTree_Sterm_beast_phylo) %>% as.data.frame()%>% rownames_to_column("label") %>% as_tibble() %>% mutate(branchAge_calculated_02 = 2020-.)  %>% left_join(.,info)%>% select(node,branchAge_calculated_02)

Lhelv_genome_origin <- read.csv("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt") %>% mutate(label=locusTAG) %>% select(label,strain_name)


myTree_Lhelv_beast_tibble_extended <- left_join(myTree_Sterm_beast_tibble,distance,by="node") %>% mutate(species="L. helveticus")%>% select(-height_0.95_HPD,-length_0.95_HPD)%>% left_join(., Lhelv_genome_origin, by='label') %>% mutate(label=ifelse(is.na(strain_name),label,strain_name)) %>% select(-strain_name)


write.csv(myTree_Lhelv_beast_tibble_extended,"/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_tree_age.csv")

myTree_all_beast_tibble_extended <- rbind(myTree_Sterm_beast_tibble_extended,myTree_Ldel_beast_tibble_extended,myTree_Lhelv_beast_tibble_extended)
write.csv(myTree_all_beast_tibble_extended,"/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ALL_tree_age.csv")

```

###### Rootage info
```{r}
###-------------------------------------------------
##correlation std error vs. rootage
###-------------------------------------------------
library(tidyverse)
myTree_all_beast_tibble_extended <- read_csv("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ALL_tree_age.csv")


plot_variation <- ggplot(myTree_all_beast_tibble_extended,aes(x=branchAge_calculated,y=branchAge_variation))+geom_point()+facet_wrap(~species,scales = "free")+theme_classic()+geom_smooth()+labs(x="Branch age",y="Branch age variation")
plot_variation

 png("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/BranchAgeVariation.png", width = 2800, height = 2500,res=300)
 plot_variation
 dev.off()

```

###### Plot phyloeny 1

remove all but root variation

```{r}

library(ape)
library(BactDating)
library(ggtree)
library(phytools)
library(treeio)
library(tidyverse)
library(coda)
library(philr)
##----------------Sterm
myTree_Sterm_beast <- read.beast('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_phylo_beast.txt')

# myTree_Sterm_beast_data <- myTree_Sterm_beast %>% as_tibble() %>% select(-parent,-branch.length)%>% mutate(height_0.95_HPD=ifelse(label=="Root",height_0.95_HPD,NA)) %>% mutate(length_0.95_HPD=ifelse(label=="Root",length_0.95_HPD,NA))%>% select(-label)
myTree_Sterm_beast_data <- myTree_Sterm_beast %>% as_tibble() %>% select(-parent,-branch.length)%>% mutate(height_0.95_HPD=ifelse(label=="nonAtAll",height_0.95_HPD,NA)) %>% mutate(length_0.95_HPD=ifelse(label=="nonAtAll",length_0.95_HPD,NA))%>% select(-label)
myTree_Sterm_beast_phylo <- myTree_Sterm_beast %>% as.phylo()

myTree_Sterm_beast_cleaned <- methods::new('treedata',phylo=myTree_Sterm_beast_phylo,data=dplyr::tbl_df(myTree_Sterm_beast_data))
tree_myTree_Sterm_beast <- ggtree(myTree_Sterm_beast_cleaned, right=TRUE, mrsd="2020-04-02") + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "S.thermophilus")+ coord_cartesian(clip = 'off')
tree_myTree_Sterm_beast
##----------------Ldel
myTree_ldel_beast <- read.beast('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel/output/PLOTS_Ldel_rounds1e+07_v1_phylo_beast.txt')
myTree_ldel_beast_data <- myTree_ldel_beast %>% as_tibble() %>% select(-parent,-branch.length)%>% mutate(height_0.95_HPD=ifelse(label=="Root",height_0.95_HPD,NA)) %>% mutate(length_0.95_HPD=ifelse(label=="Root",length_0.95_HPD,NA))%>% select(-label)

myTree_ldel_beast_data <- myTree_ldel_beast %>% as_tibble() %>% select(-parent,-branch.length)%>% mutate(height_0.95_HPD=ifelse(label=="nonAtAll",height_0.95_HPD,NA)) %>% mutate(length_0.95_HPD=ifelse(label=="nonAtAll",length_0.95_HPD,NA))%>% select(-label)
myTree_ldel_beast_phylo <- myTree_ldel_beast %>% as.phylo()

myTree_ldel_beast_cleaned <- methods::new('treedata',phylo=myTree_ldel_beast_phylo,data=dplyr::tbl_df(myTree_ldel_beast_data))
tree_myTree_ldel_beast <- ggtree(myTree_ldel_beast_cleaned, right=TRUE, mrsd="2020-04-02") + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "L.delbrueckii",x="Dating (year)")+ coord_cartesian(clip = 'off')
tree_myTree_ldel_beast

##----------------Lhelv
myTree_Lhelv_beast <- read.beast('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_phylo_beast.txt')
myTree_Lhelv_beast_data <- myTree_Lhelv_beast %>% as_tibble() %>% select(-parent,-branch.length)%>% mutate(height_0.95_HPD=ifelse(label=="Root",height_0.95_HPD,NA)) %>% mutate(length_0.95_HPD=ifelse(label=="Root",length_0.95_HPD,NA))%>% select(-label)



# Lhelv_genome_origin <- read.csv("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt") %>% mutate(label=locusTAG) %>% select(label,strain_name,origin)
# 
# 
# myTree_Lhelv_beast_cleaned_extended <- left_join(as_tibble(myTree_Lhelv_beast_cleaned), Lhelv_genome_origin, by='label') %>% mutate(label=ifelse(is.na(strain_name),label,strain_name)) %>% select(-strain_name)  %>% as.treedata()
# myTree_Lhelv_beast_cleaned_extended



myTree_Lhelv_beast_data <- myTree_Lhelv_beast %>% as_tibble() %>% select(-parent,-branch.length)%>% mutate(height_0.95_HPD=ifelse(label=="nonAtAll",height_0.95_HPD,NA)) %>% mutate(length_0.95_HPD=ifelse(label=="nonAtAll",length_0.95_HPD,NA))%>% select(-label)
myTree_Lhelv_beast_phylo <- myTree_Lhelv_beast %>% as.phylo()



myTree_Lhelv_beast_cleaned <- methods::new('treedata',phylo=myTree_Lhelv_beast_phylo,data=dplyr::tbl_df(myTree_Lhelv_beast_data))
tree_myTree_ldel_beast <- ggtree(myTree_Lhelv_beast_cleaned, right=TRUE, mrsd="2020-04-02") + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "L.delbrueckii",x="Dating (year)")+ coord_cartesian(clip = 'off')
tree_myTree_ldel_beast

# meta_Lhelv <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt",  delim = "\t", escape_double = FALSE,)
# meta_Lhelv_names <- meta_Lhelv %>% select(strain_name,locusTAG)

```


```{r}
library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)
##===========================================================================
##----------------------------------------------------------------------------
##Sterm
##----------------------------------------------------------------------------
##===========================================================================


tree_myTree_Sterm_beast <- ggtree(myTree_Sterm_beast_cleaned, right=TRUE, mrsd="2020-04-02") + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "S.thermophilus")+ coord_cartesian(clip = 'off')
tree_myTree_Sterm_beast
  
##--------------------------------
##color labels of of only rmk phages
##--------------------------------
Sterm_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/logs/Sterm_genome_origin.txt",  "\t", escape_double = FALSE, col_names = c("label","origin"),  trim_ws = TRUE) %>% as.data.frame() %>% mutate(label=paste0("S",label))%>% mutate(label=gsub("^SS","S",label))

tmp <- left_join(as_tibble(myTree_Sterm_beast_cleaned), Sterm_genome_origin, by='label')
tmp[is.na(tmp$origin),]
table(tmp$origin)


myTree_Sterm_beast_cleaned_extended <- left_join(as_tibble(myTree_Sterm_beast_cleaned), Sterm_genome_origin, by='label') %>% as.treedata()
myTree_Sterm_beast_cleaned_extended

tree_myTree_Sterm_beast_02 <- ggtree(myTree_Sterm_beast_cleaned_extended, right=TRUE, mrsd="2020-04-02") +
        theme_tree2(plot.margin=margin(3, 60, 3, 3) )+
        geom_tiplab(align=TRUE, linesize=.5,geom="text",aes(color=origin,size=origin)) +
        geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "S.thermophilus")+ 
        coord_cartesian(clip = 'off')+
        scale_color_manual(values=c("grey66",  'black')) +
        scale_size_manual(values=c(1,1.5)) 
tree_myTree_Sterm_beast_02

##--------------------------------
##add popunk lineages
##--------------------------------
# library(plotly)
# plotly::ggplotly(tree_myTree_Sterm_beast)

all_sterm_strain_info_ZOOMED <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) 
all_sterm_strain_info_ZOOMED_necessary <- all_sterm_strain_info_ZOOMED %>% mutate(label=strain) %>% select(label,starterCulture,Cheese,lineage)

myTree_Sterm_beast_cleaned_extended_02_phylo <- myTree_Sterm_beast_cleaned_extended %>% as.phylo()
lineage_node_df <- data.frame()
for (lineagesss in unique(all_sterm_strain_info_ZOOMED_necessary$lineage)) {
  
  tmp_names <- all_sterm_strain_info_ZOOMED_necessary %>% filter(lineage==lineagesss) %>% filter(label %in% myTree_Sterm_beast_cleaned_extended_02_phylo$tip.label) %>% pull(label)
  if (length(tmp_names)>1) {
    
    node_tmp <- getMRCA(myTree_Sterm_beast_cleaned_extended_02_phylo, tmp_names)

  lineage_node_df <- rbind(lineage_node_df,data.frame(lineage=as.character(lineagesss),node=node_tmp))
    
  }
  if (length(tmp_names)==1) {
    
    node_tmp <- as_tibble(myTree_Sterm_beast_cleaned_extended_02_phylo) %>% filter(label==tmp_names) %>% pull(node)

  lineage_node_df <- rbind(lineage_node_df,data.frame(lineage=as.character(lineagesss),node=node_tmp))
    
  }
  
}
lineage_node_df_Sterm <- lineage_node_df %>% mutate(species="Sterm")

dim(lineage_node_df)
cols <- rainbow(14) #I am not sure this is correct -- did not find the correct color labels anymore

tree_myTree_Sterm_beast_03 <- tree_myTree_Sterm_beast_02+geom_hilight(node=lineage_node_df[1,2],fill=cols[1],extendto=2700)+
                          geom_hilight(node=lineage_node_df[2,2],fill=cols[2],extendto=2700)+
                          geom_hilight(node=lineage_node_df[3,2],fill=cols[3],extendto=2700)+
                            geom_hilight(node=lineage_node_df[4,2],fill=cols[4],extendto=2700)+
                            geom_hilight(node=lineage_node_df[5,2],fill=cols[5],extendto=2700)+
                            geom_hilight(node=lineage_node_df[6,2],fill=cols[6],extendto=2700)+
                            geom_hilight(node=lineage_node_df[7,2],fill=cols[7],extendto=2700)+
                            geom_hilight(node=lineage_node_df[8,2],fill=cols[8],extendto=2700)+
                            geom_hilight(node=lineage_node_df[9,2],fill=cols[9],extendto=2700)+
                            geom_hilight(node=lineage_node_df[10,2],fill=cols[10],extendto=2700)+
                            geom_hilight(node=lineage_node_df[11,2],fill=cols[11],extendto=2700)+
                            geom_hilight(node=lineage_node_df[12,2],fill=cols[12],extendto=2700)+
                            theme(legend.position=c(0.2, 0.4),legend.background=element_rect(fill = alpha("white", 0.1)),legend.title = element_blank(),legend.text = element_text(size = 7))

tree_myTree_Sterm_beast_03

##--------------------------------
##add RMK origin
##--------------------------------

# table_ldel_complete_test <- read_delim("~/Desktop/Projects/2020_strainDelineation/01_logs/all_RMK_curated_Ldel.csv")
# table_sterm_complete_test02 <- read_delim("~/Desktop/Projects/2020_strainDelineation/01_logs/all_RMK_curated_Sterm.csv") %>% mutate(label=paste0("S",FAM)) %>% select(label,starter.culture,cheese)%>% mutate(starter.culture=as.character(starter.culture))


# myTree_Sterm_beast_cleaned_extended_02 <- left_join(as_tibble(myTree_Sterm_beast_cleaned_extended), table_sterm_complete_test02, by='label')
# tmp[is.na(myTree_Sterm_beast_cleaned_extended_02$origin),]
# table(myTree_Sterm_beast_cleaned_extended_02$starter.culture)
# table(myTree_Sterm_beast_cleaned_extended_02$cheese)


myTree_Sterm_beast_cleaned_extended_02 <- left_join(as_tibble(myTree_Sterm_beast_cleaned_extended), all_sterm_strain_info_ZOOMED_necessary, by='label') %>% as.treedata()
myTree_Sterm_beast_cleaned_extended_02

library(ggnewscale)
COLORSSS <- c("#d45400ff","#de592bff","#e36b42ff","#e67d59ff","#eb8f70ff","#eea38aff","#009cf0ff","#33cffaff","#7fe3ffff","#007f7fff","#00dcdcff","#ff0000ff","#0000ffff","#fde623ff","#00ff00ff","grey")


tree_myTree_Sterm_beast_03 <- ggtree(myTree_Sterm_beast_cleaned_extended_02, right=TRUE, mrsd="2020-04-02") +
        theme_tree2(plot.margin=margin(3, 60, 3, 3) )+
        geom_tiplab(align=TRUE, linesize=.5,geom="text",aes(color=origin,size=origin),hjust =-0.5) +
        geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "S.thermophilus")+ 
        coord_cartesian(clip = 'off')+
        scale_color_manual(values=c("grey66",  'black')) +
        scale_size_manual(values=c(1,1.5))  +
                            geom_hilight(node=lineage_node_df[1,2],fill=cols[1],extendto=2700)+
                            geom_hilight(node=lineage_node_df[2,2],fill=cols[2],extendto=2700)+
                            geom_hilight(node=lineage_node_df[3,2],fill=cols[3],extendto=2700)+
                            geom_hilight(node=lineage_node_df[4,2],fill=cols[4],extendto=2700)+
                            geom_hilight(node=lineage_node_df[5,2],fill=cols[5],extendto=2700)+
                            geom_hilight(node=lineage_node_df[6,2],fill=cols[6],extendto=2700)+
                            geom_hilight(node=lineage_node_df[7,2],fill=cols[7],extendto=2700)+
                            geom_hilight(node=lineage_node_df[8,2],fill=cols[8],extendto=2700)+
                            geom_hilight(node=lineage_node_df[9,2],fill=cols[9],extendto=2700)+
                            geom_hilight(node=lineage_node_df[10,2],fill=cols[10],extendto=2700)+
                            geom_hilight(node=lineage_node_df[11,2],fill=cols[11],extendto=2700)+
                            geom_hilight(node=lineage_node_df[12,2],fill=cols[12],extendto=2700)+
                            theme(legend.position="none") +
                                  new_scale_fill() + new_scale_color()  + geom_tippoint(aes(shape=Cheese,fill=starterCulture,color=starterCulture), size=2)+scale_fill_manual(values = COLORSSS)+scale_color_manual(values = COLORSSS)


 # theme(legend.position=c(0.2, 0.4),legend.background=element_rect(fill = alpha("white", 0.1)),legend.title = element_blank(),legend.text = element_text(size = 7)) +
tree_myTree_Sterm_beast_03

write.beast(myTree_Sterm_beast_cleaned_extended_02,'/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_phylo_beast_final.txt')


##===========================================================================
##----------------------------------------------------------------------------
##Ldel
##----------------------------------------------------------------------------
##===========================================================================


tree_myTree_ldel_beast <- ggtree(myTree_ldel_beast_cleaned, right=TRUE, mrsd="2020-04-02") + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "L.delbrueckii",x="Dating (year)")+ coord_cartesian(clip = 'off')
tree_myTree_ldel_beast
  

##--------------------------------
##color labels of of only rmk phages
##--------------------------------
Ldel_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Ldel_phylogeny_info.tsv",  delim = "\t", escape_double = FALSE,) %>% mutate(label=shortName) %>% select(label,origin)

# Ldel_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_Ldel/logs/Ldel_genome_origin_cleaned_02.txt",  "\t", escape_double = FALSE, col_names = c("label","origin"),  trim_ws = TRUE) %>% as.data.frame()

tmp <- left_join(as_tibble(myTree_ldel_beast_cleaned), Ldel_genome_origin, by='label')
tmp[is.na(tmp$origin),]
table(tmp$origin)

myTree_ldel_beast_cleaned_extended <- left_join(as_tibble(myTree_ldel_beast_cleaned), Ldel_genome_origin, by='label') %>% as.treedata()
myTree_ldel_beast_cleaned_extended

tree_myTree_Ldel_beast_02 <- ggtree(myTree_ldel_beast_cleaned_extended, right=TRUE, mrsd="2020-04-02") +
        theme_tree2(plot.margin=margin(3, 60, 3, 3) )+
        geom_tiplab(align=TRUE, linesize=.5,geom="text",aes(color=origin,size=origin)) +
        geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "L. delbrueckii")+ 
        coord_cartesian(clip = 'off')+
        scale_color_manual(values=c("grey66",  'black')) +
        scale_size_manual(values=c(1,1.5)) 
tree_myTree_Ldel_beast_02

##--------------------------------
##add popunk lineages
##--------------------------------
# library(plotly)
# plotly::ggplotly(tree_myTree_Sterm_beast)

all_sterm_strain_info_ZOOMED <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) 
all_sterm_strain_info_ZOOMED_necessary <- all_sterm_strain_info_ZOOMED %>% mutate(label=strain) %>% select(label,starterCulture,Cheese,lineage)

myTree_ldel_beast_cleaned_extended_02_phylo <- myTree_ldel_beast_cleaned_extended %>% as.phylo()
lineage_node_df <- data.frame()
for (lineagesss in unique(all_sterm_strain_info_ZOOMED_necessary$lineage)) {
  
  tmp_names <- all_sterm_strain_info_ZOOMED_necessary %>% filter(lineage==lineagesss) %>% filter(label %in% myTree_ldel_beast_cleaned_extended_02_phylo$tip.label) %>% pull(label)
  if (length(tmp_names)>1) {
    
    node_tmp <- getMRCA(myTree_ldel_beast_cleaned_extended_02_phylo, tmp_names)

  lineage_node_df <- rbind(lineage_node_df,data.frame(lineage=as.character(lineagesss),node=node_tmp))
    
  }
  if (length(tmp_names)==1) {
    
    node_tmp <- as_tibble(myTree_ldel_beast_cleaned_extended_02_phylo) %>% filter(label==tmp_names) %>% pull(node)

  lineage_node_df <- rbind(lineage_node_df,data.frame(lineage=as.character(lineagesss),node=node_tmp))
    
  }
  
}

dim(lineage_node_df)
lineage_node_df_Ldel <- lineage_node_df %>% mutate(species="Ldel")

cols <- rainbow(14) #I am not sure this is correct -- did not find the correct color labels anymore

tree_myTree_Ldel_beast_03 <- tree_myTree_Ldel_beast_02+geom_hilight(node=lineage_node_df[1,2],fill=cols[13],extendto=2700)+
                          geom_hilight(node=lineage_node_df[2,2],fill=cols[14],extendto=2700)+
                            theme(legend.position=c(0.2, 0.4),legend.background=element_rect(fill = alpha("white", 0.1)),legend.title = element_blank(),legend.text = element_text(size = 7))

tree_myTree_Ldel_beast_03

##--------------------------------
##add RMK origin
##--------------------------------


myTree_ldel_beast_cleaned_extended_02 <- left_join(as_tibble(myTree_ldel_beast_cleaned_extended), all_sterm_strain_info_ZOOMED_necessary, by='label') %>% as.treedata()
myTree_ldel_beast_cleaned_extended_02

library(ggnewscale)
COLORSSS <- c("#d45400ff","#de592bff","#e36b42ff","#e67d59ff","#eb8f70ff","#eea38aff","#009cf0ff","#33cffaff","#7fe3ffff","#007f7fff","#00dcdcff","#ff0000ff","#0000ffff","#fde623ff","#00ff00ff","grey")


tree_myTree_Ldel_beast_03 <- ggtree(myTree_ldel_beast_cleaned_extended_02, right=TRUE, mrsd="2020-04-02") +
        theme_tree2(plot.margin=margin(3, 60, 3, 3) )+
        geom_tiplab(align=TRUE, linesize=.5,geom="text",aes(color=origin,size=origin),hjust =-0.5) +
        geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "L. delbrueckii")+ 
        coord_cartesian(clip = 'off')+
        scale_color_manual(values=c("grey66",  'black')) +
        scale_size_manual(values=c(1,1.5)) +
                          geom_hilight(node=lineage_node_df[1,2],fill=cols[13],extendto=2700)+
                          geom_hilight(node=lineage_node_df[2,2],fill=cols[14],extendto=2700)+
                            theme(legend.position="none") +
                                  new_scale_fill() + new_scale_color()  + geom_tippoint(aes(shape=Cheese,fill=starterCulture,color=starterCulture), size=2)+scale_fill_manual(values = COLORSSS)+scale_color_manual(values = COLORSSS)

tree_myTree_Ldel_beast_03


write.beast(myTree_ldel_beast_cleaned_extended_02,'/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel_phylo_beast_final.txt')

##===========================================================================
##----------------------------------------------------------------------------
##Lhelv
##----------------------------------------------------------------------------
##===========================================================================


tree_myTree_Lhelv_beast <- ggtree(myTree_Lhelv_beast_cleaned, right=TRUE, mrsd="2020-04-02") + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+ geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "L. helveticus",x="Dating (year)")+ coord_cartesian(clip = 'off')
tree_myTree_Lhelv_beast
  
##--------------------------------
##change name
# and
##color labels of of only rmk phages
##--------------------------------
Lhelv_genome_origin <- read.csv("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt") %>% mutate(label=locusTAG) %>% select(label,strain_name,origin)
# Lhelv_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt",  delim = "\t", escape_double = FALSE,) %>% mutate(label=locusTAG) %>% select(label,strain_name,origin)


# tmp <- left_join(as_tibble(myTree_Lhelv_beast_cleaned), Lhelv_genome_origin, by='label')
# tmp[is.na(tmp$origin),]
# table(tmp$origin)

myTree_Lhelv_beast_cleaned_extended <- left_join(as_tibble(myTree_Lhelv_beast_cleaned), Lhelv_genome_origin, by='label') %>% mutate(label=ifelse(is.na(strain_name),label,strain_name)) %>% select(-strain_name)  %>% as.treedata()
myTree_Lhelv_beast_cleaned_extended
# myTree_Lhelv_beast_cleaned_extended <- left_join(as_tibble(myTree_Lhelv_beast_cleaned), Lhelv_genome_origin, by='label') %>% as.treedata()
# myTree_Lhelv_beast_cleaned_extended

tree_myTree_Lhelv_beast_02 <- ggtree(myTree_Lhelv_beast_cleaned_extended, right=TRUE, mrsd="2020-04-02") +
        theme_tree2(plot.margin=margin(3, 60, 3, 3) )+
        geom_tiplab(align=TRUE, linesize=.5,geom="text",aes(color=origin,size=origin),hjust =-0.5) +
        geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "L. helveticus")+ 
        coord_cartesian(clip = 'off')+
        scale_color_manual(values=c("grey66",  'black')) +
        scale_size_manual(values=c(1,1.5)) 
tree_myTree_Lhelv_beast_02

##--------------------------------
##add popunk lineages
##--------------------------------
# library(plotly)
# plotly::ggplotly(tree_myTree_Sterm_beast)

all_sterm_strain_info_ZOOMED <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats_Lhelv.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) 
all_sterm_strain_info_ZOOMED_necessary <- all_sterm_strain_info_ZOOMED %>% mutate(label=strain) %>% select(label,starterCulture,Cheese,lineage)

myTree_Lhelv_beast_cleaned_extended_02_phylo <- myTree_Lhelv_beast_cleaned_extended %>% as.phylo()
lineage_node_df <- data.frame()
for (lineagesss in unique(all_sterm_strain_info_ZOOMED_necessary$lineage)) {
  
  tmp_names <- all_sterm_strain_info_ZOOMED_necessary %>% filter(lineage==lineagesss) %>% filter(label %in% myTree_Lhelv_beast_cleaned_extended_02_phylo$tip.label) %>% pull(label)
  if (length(tmp_names)>1) {
    
    node_tmp <- getMRCA(myTree_Lhelv_beast_cleaned_extended_02_phylo, tmp_names)

  lineage_node_df <- rbind(lineage_node_df,data.frame(lineage=as.character(lineagesss),node=node_tmp))
    
  }
  if (length(tmp_names)==1) {
    
    node_tmp <- as_tibble(myTree_Lhelv_beast_cleaned_extended_02_phylo) %>% filter(label==tmp_names) %>% pull(node)

  lineage_node_df <- rbind(lineage_node_df,data.frame(lineage=as.character(lineagesss),node=node_tmp))
    
  }
  
}

dim(lineage_node_df)

lineage_node_df_lhelv <- lineage_node_df %>% mutate(species="Lhelv")

cols <- rainbow(16) #I am not sure this is correct -- did not find the correct color labels anymore

tree_myTree_Lhelv_beast_03 <- tree_myTree_Lhelv_beast_02+geom_hilight(node=lineage_node_df[1,2],fill=cols[15],extendto=2700)+
                                  geom_hilight(node=lineage_node_df[2,2],fill=cols[16],extendto=2700)+
                                  theme(legend.position=c(0.2, 0.4),legend.background=element_rect(fill = alpha("white", 0.1)),legend.title = element_blank(),legend.text = element_text(size = 7))

tree_myTree_Lhelv_beast_03

##--------------------------------
##add RMK origin
##--------------------------------


myTree_Lhelv_beast_cleaned_extended_02 <- left_join(as_tibble(myTree_Lhelv_beast_cleaned_extended), all_sterm_strain_info_ZOOMED_necessary, by='label') %>% as.treedata()
myTree_Lhelv_beast_cleaned_extended_02

library(ggnewscale)
# COLORSSS <- c("#d45400ff","#de592bff","#e36b42ff","#e67d59ff","#eb8f70ff","#eea38aff","#009cf0ff","#33cffaff","#7fe3ffff","#007f7fff","#00dcdcff","#ff0000ff","#0000ffff","#fde623ff","#00ff00ff","grey")
COLORSSS <- c("#de592bff","grey")


tree_myTree_Lhelv_beast_03 <- ggtree(myTree_Lhelv_beast_cleaned_extended_02, right=TRUE, mrsd="2020-04-02") +
        theme_tree2(plot.margin=margin(3, 60, 3, 3) )+
        geom_tiplab(align=TRUE, linesize=.5,geom="text",aes(color=origin,size=origin),hjust =-0.5) +
        geom_range(range='length_0.95_HPD', color='red', alpha=.2, size=2)+labs(title = "L. helveticus")+ 
        coord_cartesian(clip = 'off')+
        scale_color_manual(values=c("grey66",  'black')) +
        scale_size_manual(values=c(1,1.5)) +
                          geom_hilight(node=lineage_node_df[1,2],fill=cols[15],extendto=2700)+
                          geom_hilight(node=lineage_node_df[2,2],fill=cols[16],extendto=2700)+
                            theme(legend.position="none") +
                                  new_scale_fill() + new_scale_color()  + geom_tippoint(aes(shape=Cheese,fill=starterCulture,color=starterCulture), size=2)+scale_fill_manual(values = COLORSSS)+scale_color_manual(values = COLORSSS)

tree_myTree_Lhelv_beast_03

write.beast(myTree_Lhelv_beast_cleaned_extended_02,'/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_phylo_beast_final.txt')


###==================================================================
#merge plots
###==================================================================
library(patchwork)


tree_myTree_Sterm_beast_03+tree_myTree_Ldel_beast_03+tree_myTree_Lhelv_beast_03

 
 pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/dated_phylogenies_colored.pdf",width=22,height=16)

tree_myTree_Sterm_beast_03+tree_myTree_Ldel_beast_03+tree_myTree_Lhelv_beast_03

dev.off()

###==================================================================
#output info
###==================================================================
# table(Sterm_genome_origin$species)
Sterm_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited_extended_02.csv", ",", escape_double = FALSE, trim_ws = TRUE) %>% filter(species %in% c("streptococcus thermophilus","thermophilus"))%>% filter(!is.na(biosample))%>% select(c(shortName, biosample, extraction_time))%>% mutate(species="S.thermophilus")



Ldel_genome_origin <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Ldel_phylogeny_info.tsv",  delim = "\t", escape_double = FALSE,) %>% filter(origin=="NCBI_refs") %>% select(c(shortName, biosample, extraction_time)) %>% mutate(species="L.delbrueckii")
Lhelv_genome_origin <- read.csv("~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned.txt") %>% mutate(extraction_time=isolation_date)%>% mutate(shortName=strain_name) %>% filter(species=="helveticus") %>% select(c(shortName, biosample, extraction_time))%>% mutate(species="L.helveticus")


all_genome_origin <-   rbind(Sterm_genome_origin,Ldel_genome_origin,Lhelv_genome_origin) %>% as_tibble()

write_csv(all_genome_origin,"~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/all_genome_origin_final.txt") 





```

```{bash}


scp ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/all_genome_origin_final.txt vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/

```


###### Age of the strains


```{r}
###-------------------------------------------
#Species age
#including subsp. age
###-------------------------------------------
library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)


###-------------------------------------------
#Crown/Root age
#including subsp. age
###-------------------------------------------

myTree_all_beast_tibble_extended <- read.csv("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ALL_tree_age.csv")
myTree_all_beast_tibble_extended_crownAge <- myTree_all_beast_tibble_extended %>% filter(label=="Root") %>% select(species,branchAge_calculated,branchAge_variation,node,label) %>% mutate(group="Crown")

# Combine with jitter points
ggplot(myTree_all_beast_tibble_extended_crownAge, aes(species, branchAge_calculated,fill=species,color=species)) +
  geom_point() + 
  geom_pointrange(aes(ymin = branchAge_calculated-branchAge_variation, ymax = branchAge_calculated+branchAge_variation),data = myTree_all_beast_tibble_extended_crownAge)+theme_classic()+facet_wrap(~group)+
  coord_flip()

#RMK strain age
###-------------------------------------------

all_sterm_strain_info_ZOOMED <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats_Lhelv.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) 
all_sterm_strain_info_ZOOMED_necessary <- all_sterm_strain_info_ZOOMED %>% mutate(label=strain) %>% select(label,starterCulture,Cheese,lineage)
table(all_sterm_strain_info_ZOOMED$species)

# tmp <- all_sterm_strain_info_ZOOMED %>% filter(species=="Sterm")

table(all_sterm_strain_info_ZOOMED$species)
table(myTree_all_beast_tibble_extended$species)


strains_tmp <- all_sterm_strain_info_ZOOMED %>% filter(species=="Ldel") %>% filter(strain %in% myTree_ldel_beast_cleaned_extended_02_phylo$tip.label)%>% pull(strain)
Ldel_MRCA <- getMRCA(myTree_ldel_beast_cleaned_extended_02_phylo, strains_tmp)
myTree_all_beast_tibble_extended_Age_strains_Ldel <- myTree_all_beast_tibble_extended %>% filter(species=="L. delbrueckii") %>% filter(node==Ldel_MRCA) %>% select(species,branchAge_calculated,branchAge_variation,node,label) %>% mutate(group="RMK strains")

strains_tmp <- all_sterm_strain_info_ZOOMED %>% filter(species=="Sterm") %>% filter(strain %in% myTree_Sterm_beast_cleaned_extended_02_phylo$tip.label)%>% pull(strain)
Sterm_MRCA <- getMRCA(myTree_Sterm_beast_cleaned_extended_02_phylo, strains_tmp)
myTree_all_beast_tibble_extended_Age_strains_Sterm <- myTree_all_beast_tibble_extended %>% filter(species=="S. thermophilus") %>% filter(node==Sterm_MRCA) %>% select(species,branchAge_calculated,branchAge_variation,node,label) %>% mutate(group="RMK strains")

strains_tmp <- all_sterm_strain_info_ZOOMED %>% filter(species=="Lhelv") %>% filter(strain %in% myTree_Lhelv_beast_cleaned_extended_02_phylo$tip.label)%>% pull(strain)
Lhelv_MRCA <- getMRCA(myTree_Lhelv_beast_cleaned_extended_02_phylo, strains_tmp)
myTree_all_beast_tibble_extended_Age_strains_Lhelv <- myTree_all_beast_tibble_extended %>% filter(species=="L. helveticus") %>% filter(node==Lhelv_MRCA) %>% select(species,branchAge_calculated,branchAge_variation,node,label) %>% mutate(group="RMK strains")


myTree_all_beast_tibble_extended_all_01 <- rbind(myTree_all_beast_tibble_extended_crownAge,myTree_all_beast_tibble_extended_Age_strains_Ldel,myTree_all_beast_tibble_extended_Age_strains_Sterm,myTree_all_beast_tibble_extended_Age_strains_Lhelv) %>% mutate(grouping_extended=species)

ggplot() +
  # geom_point() + 
  geom_pointrange(aes(x=species, y=branchAge_calculated, ymin = branchAge_calculated-branchAge_variation, ymax = branchAge_calculated+branchAge_variation,fill=species,color=species),data = myTree_all_beast_tibble_extended_all_01)+theme_classic()+facet_wrap(~group,nrow = 2)+
  coord_flip()
###-------------------------------------------
#lineage age
###-------------------------------------------

lineage_node_df_Sterm$species <- plyr::revalue(lineage_node_df_Sterm$species,c("Lhelv"="L. helveticus","Sterm"="S. thermophilus","Ldel"="L. delbrueckii"))
lineage_node_df_Sterm 

lineage_node_df_Ldel$species <- plyr::revalue(lineage_node_df_Ldel$species,c("Lhelv"="L. helveticus","Sterm"="S. thermophilus","Ldel"="L. delbrueckii"))
lineage_node_df_Ldel

lineage_node_df_lhelv$species <- plyr::revalue(lineage_node_df_lhelv$species,c("Lhelv"="L. helveticus","Sterm"="S. thermophilus","Ldel"="L. delbrueckii"))
lineage_node_df_lhelv

lineage_dating <- rbind(lineage_node_df_Sterm,lineage_node_df_Ldel,lineage_node_df_lhelv) %>% mutate(grouping_extended=paste0(species,"_",lineage)) %>% left_join(.,myTree_all_beast_tibble_extended,by=c("species","node")) %>% filter(!is.na(branchAge_calculated_02))%>% mutate(group="Lineage") %>% mutate(branchAge_calculated=branchAge_calculated_02)%>% select(species,branchAge_calculated,branchAge_variation,node,label,group,grouping_extended)


# myTree_all_beast_tibble_extended_all_01 <- rbind(myTree_all_beast_tibble_extended_crownAge,myTree_all_beast_tibble_extended_Age_strains_Ldel,myTree_all_beast_tibble_extended_Age_strains_Sterm,myTree_all_beast_tibble_extended_Age_strains_Lhelv)
myTree_all_beast_tibble_extended_all_02 <-  rbind(myTree_all_beast_tibble_extended_all_01,lineage_dating)

myTree_all_beast_tibble_extended_all_02$group = factor(myTree_all_beast_tibble_extended_all_02$group, levels=rev(c("Crown","RMK strains","Lineage")))

unique(myTree_all_beast_tibble_extended_all_02$grouping_extended)
myTree_all_beast_tibble_extended_all_02$grouping_extended = factor(myTree_all_beast_tibble_extended_all_02$grouping_extended, levels=rev(c("S. thermophilus"  , "L. delbrueckii" ,   "L. helveticus" ,"S. thermophilus_1","S. thermophilus_2" ,  "S. thermophilus_3","S. thermophilus_4" ,"S. thermophilus_5" ,"S. thermophilus_6" ,"S. thermophilus_7" ,"S. thermophilus_8", "S. thermophilus_9", "L. delbrueckii_13" ,"L. helveticus_15" )))

cols_1 <- rainbow(14) 
cols_2 <- rainbow(16) 

colorsss <- rev(c("#007fa6ff","#fdc2b5ff","#b3b3b3ff",cols_1[c(1,2,3,4,5,6,7,8,9,13)],cols_2[c(15)]))

branch_age_plot <- ggplot() +
  geom_pointrange(aes(x=grouping_extended, y=branchAge_calculated, ymin = branchAge_calculated-branchAge_variation, ymax = branchAge_calculated+branchAge_variation,fill=grouping_extended,color=grouping_extended),data = myTree_all_beast_tibble_extended_all_02)+theme_classic()+facet_wrap(group~.,nrow = 3,scales = "free_y", strip.position = "right")+
  coord_flip()+scale_fill_manual(values = colorsss)+scale_color_manual(values = colorsss)+
  theme(axis.text.y = element_blank(),axis.title.y = element_blank(),axis.ticks.y = element_blank(),legend.position = "none",strip.text = element_text(size = 7))+labs(y="Branch age")+scale_y_continuous(breaks=c(-10000,-7500,-5000,-2500,0,2020))

branch_age_plot

###==================================================================
#merge plots
###==================================================================
library(patchwork)


tree_myTree_Sterm_beast_03+tree_myTree_Ldel_beast_03+(tree_myTree_Lhelv_beast_03 + branch_age_plot+plot_layout(ncol=1,heights = c(2,1.5)))

 
 pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/dated_phylogenies_colored_extended.pdf",width=10,height=6)

tree_myTree_Sterm_beast_03+tree_myTree_Ldel_beast_03+(tree_myTree_Lhelv_beast_03 + branch_age_plot+plot_layout(ncol=1,heights = c(2,1.5)))

dev.off()
###==================================================================
#output info
###==================================================================

tmp1 <- myTree_Lhelv_beast_cleaned_extended_02 %>% as.tibble %>% filter(origin!="NCBI_refs")%>% pull(label) %>% as_tibble %>% mutate(species="Lhelv")
tmp2 <- myTree_Sterm_beast_cleaned_extended_02 %>% as.tibble %>% filter(origin!="NCBI_refs")%>% pull(label) %>% as_tibble %>% mutate(species="Sterm")
tmp3 <- myTree_ldel_beast_cleaned_extended_02 %>% as.tibble %>% filter(origin!="NCBI_refs")%>% pull(label) %>% as_tibble %>% mutate(species="Ldel")



tmp1 <- myTree_Lhelv_beast_cleaned_extended_02 %>% as.tibble %>% filter(origin!="NCBI_refs")%>% pull(label) %>% as_tibble %>% mutate(species="Lhelv")
tmp2 <- myTree_Sterm_beast_cleaned_extended_02 %>% as.tibble %>% filter(origin!="NCBI_refs")%>% pull(label) %>% as_tibble %>% mutate(species="Sterm")
tmp3 <- myTree_ldel_beast_cleaned_extended_02 %>% as.tibble %>% filter(origin!="NCBI_refs")%>% pull(label) %>% as_tibble %>% mutate(species="Ldel")

all_genomes_toether <- rbind(tmp1,tmp2,tmp3)

table(all_genomes_toether$species)

write_csv(all_genomes_toether,"~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/all_genomes_dialact.txt") 

#scp ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/all_genomes_dialact.txt vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/all_genomes_dialact.txt

```




##### 5. Pseudogene 


```{r}

library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)


##----------------------------------------------------
#Sterm
##----------------------------------------------------
# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_curated.txt", tree.names = NULL, force.multi = FALSE)


myTree_Sterm_beast <- read.beast('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_phylo_beast_final.txt')
tree_myTree_Sterm_beast <- ggtree(myTree_Sterm_beast, right=TRUE, mrsd="2021-04-02") + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+labs(title = "S.thermophilus")
tree_myTree_Sterm_beast

# write_csv(Orthogroups_orthofinder_stats_test_keepers_counts,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//genomes_ps.clustered.txt")


cluster_ps_counts_wide_imported <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//Sterm_cluster_ps_counts_wide_imported_wide_final")

# write_csv(cluster_ps_counts_wide_imported_wide_final,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_cluster_ps_counts_wide_imported_wide_final")

myTree_all_beast_tibble_extended_imported <- read_csv("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ALL_tree_age.csv")

cluster_ps_counts_wide_imported$species <-  plyr::revalue(cluster_ps_counts_wide_imported$species,c("Ldel"="L. delbrueckii","Sterm"="S. thermophilus","Lhelv"="L. helveticus"))

myTree_Sterm_beast_phylo <- as.phylo(myTree_Sterm_beast)
myTree_Sterm_beast_phylo$tip.label
library(philr)

age_cluster_info <- data.frame()




# for (speciesss in unique(cluster_ps_counts_wide_imported$species)) {
  speciesss <- "S. thermophilus"
  
#  cluster_ps_counts_wide_imported_wide <-  cluster_ps_counts_wide_imported %>% filter(species==speciesss) %>% select(-ps,-re,-species) %>% spread(., genome, gene_presence)  %>%replace(is.na(.), "rm")
#   
# cluster_ps_counts_wide_imported_wide_final <-  cluster_ps_counts_wide_imported_wide %>%  gather(., genome,grouping , colnames(cluster_ps_counts_wide_imported_wide)[2:ncol(cluster_ps_counts_wide_imported_wide)], factor_key=TRUE,na.rm = TRUE) %>% filter(grouping!="re")
 cluster_ps_counts_wide_imported_wide_final <- cluster_ps_counts_wide_imported %>% filter(species==speciesss) %>% filter(grouping!="re") %>% filter(grouping!="rm") %>% mutate(cluster=Orthogroup)
 table(cluster_ps_counts_wide_imported_wide_final$grouping)
 
 
# cluster_ps_counts_wide_imported_species <- cluster_ps_counts_wide_imported %>% filter(species==speciesss)
print(speciesss)

for (clusterssss in unique(cluster_ps_counts_wide_imported_wide_final$cluster)) {
  print(clusterssss)

  cluster_ps_final_prep_tmp <- cluster_ps_counts_wide_imported_wide_final %>% filter(cluster==clusterssss)

  if (nrow(cluster_ps_final_prep_tmp)>1) {
    
MRCA_of_tmp <- getMRCA(as.phylo(myTree_Sterm_beast), cluster_ps_final_prep_tmp$genome)

branchAGE <- myTree_all_beast_tibble_extended_imported %>% filter(species==speciesss) %>% filter(node==MRCA_of_tmp) %>% pull(branchAge_calculated_02)

branchVariation <- myTree_all_beast_tibble_extended_imported %>% filter(species==speciesss) %>% filter(node==MRCA_of_tmp) %>% pull(branchAge_variation)
  

tmp_02 <- data.frame(species=speciesss,cluster=clusterssss,node=MRCA_of_tmp,AGE=branchAGE,variation=branchVariation)

age_cluster_info <- rbind(age_cluster_info,tmp_02)

  }

}
# }


myTree_all_beast_tibble_extended_imported %>% filter(label=="Root")

age_Sterm_ps <- ggplot(age_cluster_info,aes(x=AGE))+geom_density()+theme_classic()+geom_vline(xintercept =-1843 )+lims(x=c(-5000,2020))

age_Sterm_ps


pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/age_Sterm_ps.pdf",width=7,height=3)

age_Sterm_ps

dev.off()

##----------------------------------------------------
#Ldel
##----------------------------------------------------
# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_curated.txt", tree.names = NULL, force.multi = FALSE)


myTree_Ldel_beast <- read.beast('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel_phylo_beast_final.txt')
tree_myTree_Sterm_beast <- ggtree(myTree_Ldel_beast, right=TRUE, mrsd="2021-04-02") + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+labs(title = "S.thermophilus")
tree_myTree_Sterm_beast

# cluster_ps_counts_wide_imported <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//genomes_ps.clustered.txt")
 # cluster_ps_counts_wide_imported <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//Ldel_cluster_ps_counts_wide_imported_wide_final")
cluster_ps_counts_wide_imported <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//Ldel_cluster_ps_counts_wide_imported_wide_final")
# write_csv(cluster_ps_counts_wide_imported_wide_final,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_cluster_ps_counts_wide_imported_wide_final")

myTree_all_beast_tibble_extended_imported <- read_csv("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ALL_tree_age.csv")

cluster_ps_counts_wide_imported$species <-  plyr::revalue(cluster_ps_counts_wide_imported$species,c("Ldel"="L. delbrueckii","Sterm"="S. thermophilus","Lhelv"="L. helveticus"))

myTree_Ldel_beast_phylo <- as.phylo(myTree_Ldel_beast)
myTree_Ldel_beast_phylo$tip.label
library(philr)

age_cluster_info_Ldel <- data.frame()

# for (speciesss in unique(cluster_ps_counts_wide_imported$species)) {
  speciesss <- "L. delbrueckii"
   # cluster_ps_counts_wide_imported_wide <-  cluster_ps_counts_wide_imported %>% filter(species==speciesss) %>% select(-ps,-re,-species) %>% spread(., genome, gene_presence)  %>%replace(is.na(.), "rm")
  
# cluster_ps_counts_wide_imported_wide_final <-  cluster_ps_counts_wide_imported_wide %>%  gather(., genome,grouping , colnames(cluster_ps_counts_wide_imported_wide)[2:ncol(cluster_ps_counts_wide_imported_wide)], factor_key=TRUE,na.rm = TRUE) %>% filter(grouping!="re")
   cluster_ps_counts_wide_imported_wide_final <- cluster_ps_counts_wide_imported %>% filter(species==speciesss) %>% filter(grouping!="re") %>% filter(grouping!="rm")%>% mutate(cluster=Orthogroup)
 table(cluster_ps_counts_wide_imported_wide_final$grouping)

 table(cluster_ps_counts_wide_imported_wide_final$grouping)

# cluster_ps_counts_wide_imported_species <- cluster_ps_counts_wide_imported %>% filter(species==speciesss)
print(speciesss)


unique(cluster_ps_counts_wide_imported_wide_final$genome) %in% myTree_Ldel_beast_phylo$tip.label
 myTree_Ldel_beast_phylo$tip.label %in% unique(cluster_ps_counts_wide_imported_wide_final$genome)


for (clusterssss in unique(cluster_ps_counts_wide_imported_wide_final$cluster)) {
  print(clusterssss)

  cluster_ps_final_prep_tmp <- cluster_ps_counts_wide_imported_wide_final %>% filter(cluster==clusterssss)

  if (nrow(cluster_ps_final_prep_tmp)>1) {
    
MRCA_of_tmp <- getMRCA(as.phylo(myTree_Ldel_beast_phylo), cluster_ps_final_prep_tmp$genome)

branchAGE <- myTree_all_beast_tibble_extended_imported %>% filter(species==speciesss) %>% filter(node==MRCA_of_tmp) %>% pull(branchAge_calculated_02)

branchVariation <- myTree_all_beast_tibble_extended_imported %>% filter(species==speciesss) %>% filter(node==MRCA_of_tmp) %>% pull(branchAge_variation)
  

tmp_02 <- data.frame(species=speciesss,cluster=clusterssss,node=MRCA_of_tmp,AGE=branchAGE,variation=branchVariation)

age_cluster_info_Ldel <- rbind(age_cluster_info_Ldel,tmp_02)

  }

}
# }


myTree_all_beast_tibble_extended_imported %>% filter(label=="Root")

age_Ldel_ps <- ggplot(age_cluster_info_Ldel,aes(x=AGE))+geom_density()+theme_classic()+geom_vline(xintercept =-4933 )+lims(x=c(-5000,2020))

age_Ldel_ps


##----------------------------------------------------
#Lhelv
##----------------------------------------------------
# Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_curated.txt", tree.names = NULL, force.multi = FALSE)


myTree_Lhelv_beast <- read.beast('/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_phylo_beast_final.txt')
tree_myTree_Sterm_beast <- ggtree(myTree_Lhelv_beast, right=TRUE, mrsd="2021-04-02") + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom="text")+labs(title = "S.thermophilus")
tree_myTree_Sterm_beast

# cluster_ps_counts_wide_imported <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//genomes_ps.clustered.txt")
cluster_ps_counts_wide_imported <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_cluster_ps_counts_wide_imported_wide_final")

myTree_all_beast_tibble_extended_imported <- read_csv("/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ALL_tree_age.csv")

cluster_ps_counts_wide_imported$species <-  plyr::revalue(cluster_ps_counts_wide_imported$species,c("Ldel"="L. delbrueckii","Sterm"="S. thermophilus","Lhelv"="L. helveticus"))

myTree_Lhelv_beast_phylo <- as.phylo(myTree_Lhelv_beast)
myTree_Lhelv_beast_phylo$tip.label
library(philr)

age_cluster_info_Lhelv <- data.frame()

# for (speciesss in unique(cluster_ps_counts_wide_imported$species)) {
  speciesss <- "L. helveticus"
   # cluster_ps_counts_wide_imported_wide <-  cluster_ps_counts_wide_imported %>% filter(species==speciesss) %>% select(-ps,-re,-species) %>% spread(., genome, gene_presence)  %>%replace(is.na(.), "rm")
  
# cluster_ps_counts_wide_imported_wide_final <-  cluster_ps_counts_wide_imported_wide %>%  gather(., genome,grouping , colnames(cluster_ps_counts_wide_imported_wide)[2:ncol(cluster_ps_counts_wide_imported_wide)], factor_key=TRUE,na.rm = TRUE) %>% filter(grouping!="re")
    cluster_ps_counts_wide_imported_wide_final <- cluster_ps_counts_wide_imported %>% filter(species==speciesss) %>% filter(grouping!="re") %>% filter(grouping!="rm")%>% mutate(cluster=Orthogroup)
 table(cluster_ps_counts_wide_imported_wide_final$grouping)

 table(cluster_ps_counts_wide_imported_wide_final$grouping)
 
# cluster_ps_counts_wide_imported_species <- cluster_ps_counts_wide_imported %>% filter(species==speciesss)
print(speciesss)

unique(cluster_ps_counts_wide_imported_wide_final$genome) %in% myTree_Lhelv_beast_phylo$tip.label
 myTree_Lhelv_beast_phylo$tip.label %in% unique(cluster_ps_counts_wide_imported_wide_final$genome)

for (clusterssss in unique(cluster_ps_counts_wide_imported_wide_final$cluster)) {
  print(clusterssss)

  cluster_ps_final_prep_tmp <- cluster_ps_counts_wide_imported_wide_final %>% filter(cluster==clusterssss)

  if (nrow(cluster_ps_final_prep_tmp)>1) {
    
MRCA_of_tmp <- getMRCA(as.phylo(myTree_Lhelv_beast_phylo), cluster_ps_final_prep_tmp$genome)

branchAGE <- myTree_all_beast_tibble_extended_imported %>% filter(species==speciesss) %>% filter(node==MRCA_of_tmp) %>% pull(branchAge_calculated_02)

branchVariation <- myTree_all_beast_tibble_extended_imported %>% filter(species==speciesss) %>% filter(node==MRCA_of_tmp) %>% pull(branchAge_variation)
  

tmp_02 <- data.frame(species=speciesss,cluster=clusterssss,node=MRCA_of_tmp,AGE=branchAGE,variation=branchVariation)

age_cluster_info_Lhelv <- rbind(age_cluster_info_Lhelv,tmp_02)

  }

}
# }


myTree_all_beast_tibble_extended_imported %>% filter(label=="Root") %>% pull(branchAge_calculated_02)
myTree_all_beast_tibble_extended_imported %>% filter(label=="Root") %>% select(species,branchAge_calculated_02)

age_Lhelv_ps <- ggplot(age_cluster_info_Lhelv,aes(x=AGE))+geom_density()+theme_classic()+geom_vline(xintercept =-2650 )+lims(x=c(-5000,2020))

age_Lhelv_ps

###==================================================
#age see if younger then root
###==================================================
age_cluster_info_Lhelv %>% mutate(root_or_not=ifelse(AGE>-2645,"new","root")) %>% group_by(species,root_or_not) %>% dplyr::summarise(countss=n()) %>%  mutate(percentss=100*(countss/399))

age_cluster_info%>% mutate(root_or_not=ifelse(AGE>-1201,"new","root")) %>% group_by(species,root_or_not) %>% dplyr::summarise(countss=n()) %>%  mutate(percentss=100*(countss/546))


age_cluster_info_Ldel%>% mutate(root_or_not=ifelse(AGE>-4930,"new","root")) %>% group_by(species,root_or_not) %>% dplyr::summarise(countss=n()) %>%  mutate(percentss=100*(countss/333))

###==================================================
#all toegther
###==================================================

age_cluster_info_ALL <- rbind(age_cluster_info_Lhelv,age_cluster_info,age_cluster_info_Ldel)

myTree_all_beast_tibble_extended_imported %>% filter(label=="Root") %>% pull(branchAge_calculated_02)
myTree_all_beast_tibble_extended_imported %>% filter(label=="Root") %>% select(species,branchAge_calculated_02)


age_cluster_info_ALL$species = factor(age_cluster_info_ALL$species, levels=c("S. thermophilus","L. delbrueckii","L. helveticus"))


All_plot <- ggplot(age_cluster_info_ALL,aes(x=AGE,color=species,fill=species))+geom_density(alpha=0.7)+theme_classic()+
            geom_vline(xintercept =-1201,color="#007fa6ff",size=1.5 )+
            geom_vline(xintercept =-4933,color="#fdc2b5ff",size=1.5 )+
            geom_vline(xintercept =-2650,color="grey77" ,size=1.5)+
            lims(x=c(-5000,2020))+
            scale_color_manual(values=c("#007fa6ff","#fdc2b5ff","grey77"))+
            scale_fill_manual(values=c("#007fa6ff","#fdc2b5ff","grey77"))+labs(x="Pseudogene age" )+theme(axis.text.y = element_blank(),axis.title.y = element_blank(),axis.ticks.y = element_blank(),legend.position = "none")

All_plot

pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/age_ALL_ps.pdf",width=7,height=3)

All_plot

dev.off()

```


#### MMseqs2

```{bash}
###-----------------------------------
#merge all faa
###-----------------------------------

#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/allGenes.faa
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm/01_all_FAA/*.faa >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/allGenes.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel/01_all_FAA/*.faa >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/allGenes.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/PROKKA_FAA/Out* >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/allGenes.faa
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/PROKKA_FAA/Out* >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/allGenes.faa


# /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FFN

##-----------ffn

#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/allGenes.ffn
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm/01_all_FFN/*.ffn >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/allGenes.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel/01_all_FFN/*.ffn >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/allGenes.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/PROKKA_FFN/Out* >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/allGenes.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/PROKKA_FFN/Out* >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/allGenes.ffn



grep ">" -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/allGenes.ffn 

###⨪---------------------NEW----------------------------------------

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/*.faa > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes.faa
##----ffn

###-------------------make collection of all rmk strains
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm/01_all_FFN/*.ffn >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel/01_all_FFN/*.ffn >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes.ffn

###-------------------add all species close to Ldel and Sterm from figure 1D/E

cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FFN/*.fna >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN/*.fna >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes.ffn

#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/thermophilus.faa
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/delbrueckii_*


###-------------------add all strains close to Ldel and Sterm from figure 1F/G

cat  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/PROKKA_FFN/*.ffn >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes.ffn
cat  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/PROKKA_FFN/*.ffn >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes.ffn


sed 's/lcl|//g' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes.ffn > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes_wo_whitespace.ffn

#grep "L-KLDS1.1011_01743" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes.ffn
#grep "L-KLDS1.1011_01743" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes_wo_whitespace.ffn

grep "NZ_CP013610.1_cds_WP_060598094.1_1247" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes.ffn
grep "NZ_CP013610.1_cds_WP_060598094.1_1247" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes_wo_whitespace.ffn

###-----------------------------------
#cluster
###-----------------------------------
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/tmp/

mmseqs easy-cluster  --threads 10 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes.ffn /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/results_NUC /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/tmp/

###-----------------------------------
#extract all cluster sequeneces
###-----------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/
cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/results_NUC_cluster.tsv |sort|uniq -c| sed 's/^[[:blank:]]*//g'|awk -F " " '{if($1>3) print $0 }' |wc -l


for clustersss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/results_NUC_cluster.tsv |sort|uniq -c| sed 's/^[[:blank:]]*//g'|awk -F " " '{if($1>2) print $2 }')
do
echo "========================================"
echo $clustersss
echo "========================================"
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/${clustersss}_homology.ffn
for geneessss in $(awk -F "\t" -v clusterzzz="$clustersss" '{if($1==clusterzzz)print $2}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/results_NUC_cluster.tsv)
do
echo $geneessss
samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/20210412_allGenes_wo_whitespace.ffn ${geneessss} >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/${clustersss}_homology.ffn

done
done #clustersss

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/ |grep "_aln.fasta" -c
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/ |grep "_aln.fasta" -v -c

cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/results_NUC_cluster.tsv |sort|uniq -c| sed 's/^[[:blank:]]*//g'|awk -F " " '{if($1>2) print $2 }'|wc -l
cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/results_NUC_cluster.tsv |sort|uniq -c| sed 's/^[[:blank:]]*//g'|awk -F " " '{if($1>2) print $2 }'|grep "S13495_01777" -n

```


align protein families

```{bash}


cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/
for genesss in $(ls |grep ".ffn$" |sed 's/.ffn//g')
do
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc//${genesss}_aln.fasta
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc//${genesss}_aln_nuc.fasta

mafft --thread 37 --maxiterate 1000  --auto /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc//${genesss}.ffn > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc//${genesss}_aln_nuc.fasta

done 

ll /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/ |grep "aln.fasta" -c
ll /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/ |grep "_aln_nuc.fasta" -c


rm *_aln_prune.fasta

for genesss in $(ls |grep "_aln_nuc.fasta$" |sed 's/_aln_nuc.fasta//g'|grep -v "-")
do
genesss_new=$(echo $genesss |sed 's/_/-/g')
echo -e "make" ${genesss}"_aln_nuc.fasta  into "${genesss_new}"_aln_nuc.fasta"
mv ${genesss}_aln_nuc.fasta ${genesss_new}_aln_nuc.fasta

done 

```


```{bash}

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/prune_aln.pl

```


make a raxml genetree

```{bash}
genesss=S72-01829-homology

#/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s OG0000000_aln_prune.fasta -n OG0000000_tree -T 20

#seqret -snucleotide1 -osformat2 phylip -sequence  ${genesss}_aln_prune.fasta -outseq ${genesss}_aln_prune.phylip
 #head ${genesss}_aln_prune.phylip
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/tree
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/tree

for genesss in $(ls ../ |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g'|grep -v "_")
do
echo ${genesss}

#seqret -snucleotide1 -osformat2 phylip -sequence  ../${genesss}_aln_prune.fasta -outseq ../${genesss}_aln_prune.phylip
# head ${genesss}_aln_prune.phylip
rm *${genesss}_tre*

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 10 -m GTRCAT -s ../${genesss}_aln_prune.fasta -n ${genesss}_tree -T 20

done 


```




#### generax
In order to compare the gene-2-species tree I use [genevax](https://github.com/BenoitMorel/GeneRax).
alternatively we could look at [HoMeR](https://compbio.engr.uconn.edu/software/homer/). 

In any case what we need is:

- a species tree
- gene trees or gene alignments
- the genes should be in particular format (speciesA_geneX)


first I need to root the sepceis tree

```{bash}
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Apr12/Species_Tree/SpeciesTree_rooted_node_labels.txt


mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/

scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Apr12/Species_Tree/SpeciesTree_rooted_node_labels.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/

scp /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/SpeciesTree_rooted_node_labels_rooted.txt vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Apr12/Species_Tree/
scp /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/SpeciesTree_rooted_node_labels_rooted.newick vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Apr12/Species_Tree/

```



```{bash}

##=======================================
#prep
##=======================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/prepedDir/included
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/prepedDir/included/

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/log
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/log/families_final.log

echo -e "[FAMILIES]" > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/log/families_final.log
for genesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/tree |grep "RAxML_info."|sed 's/RAxML_info.//g'|sed 's/_tree//g' )
do

#RAxML_bestTree.S50-00142-homology_tree

cp ${genesss}_aln_prune.fasta  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/prepedDir/included/${genesss}.fasta

cp  tree/RAxML_bestTree.${genesss}_tree  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/prepedDir/included/${genesss}.newick


echo -e "- "${genesss}"\nstarting_gene_tree = "${genesss}".newick\nalignment = "${genesss}".fasta\nsubst_model = GTR" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/log/families_final.log

#echo -e "[FAMILIES]\n- "${genesss}"\nstarting_gene_tree = RAxML_bestTree."${genesss}"_tree\nalignment = "${genesss}"_aln_prune.fasta\nsubst_model = GTR" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/log/families_final.log


done

##=======================================
#run
##=======================================


conda activate GeneRaxss
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/prepedDir/included/
#generax -s /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_node_labels.txt \
#  -f  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/log/families_final.log  -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final


#mpiexec -np 2 generax -s /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_node_labels.txt \
#  -f  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/log/families_final.log  -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final


#generax --rec-model UndatedDTL --max-spr-radius 3 -s /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Apr12/Species_Tree/SpeciesTree_rooted_node_labels_rerooted_02.txt \
#  -f  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/log/families_final.log  -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final



#generax --rec-model UndatedDTL --max-spr-radius 3 -s /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Apr12/Species_Tree/SpeciesTree_rooted.txt \
#  -f  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/log/families_final.log  -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final


generax --rec-model UndatedDTL --max-spr-radius 3 -s MiniNJ  -f  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/log/families_final.log  -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final

```

analysis of generax transfer

```{bash}

##------------------------------------
##without nodes
##------------------------------------
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_transfers_woNODES.txt
for genesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/tree |grep "RAxML_info."|sed 's/RAxML_info.//g'|sed 's/_tree//g' )
do

grep "N" -v /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/reconciliations/${genesss}_transfers.txt >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_transfers_woNODES.txt

done

#ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/tree |grep "RAxML_info."|sed 's/RAxML_info.//g'|sed 's/_tree//g'|cut -d '-' -f 1|sort|uniq > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_genome_names.txt
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/tree |grep "RAxML_info."|sed 's/RAxML_info.//g'|sed 's/_tree//g'|sort|uniq > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_genome_names.txt

grep "S-I-956" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_transfers_woNODES.txt

grep "S24735-01046-homology" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_transfers_woNODES.txt


##------------------------------------
##WITH  nodes
##------------------------------------
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_transfers_withNODES.txt
for genesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/MMseqs/HomologyGroups_nuc/tree |grep "RAxML_info."|sed 's/RAxML_info.//g'|sed 's/_tree//g' )
do

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/reconciliations/${genesss}_transfers.txt >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_transfers_withNODES.txt

done


###------------------------------
#transfer local


mkdir -p /home/vincent/Desktop/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/

scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_transfers_woNODES.txt /home/vincent/Desktop/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/

scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_genome_names.txt /home/vincent/Desktop/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/

scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_transfers_withNODES.txt /home/vincent/Desktop/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/


scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/perSpeciesCoverage.txt /home/vincent/Desktop/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/

scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/inferred_species_tree.newick /home/vincent/Desktop/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/



```


```{r}

library(readr)
library(plyr)
library(tidyverse)

##------------------all edges
all_transfers_woNODES <- read_table2("~/Desktop/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_transfers_woNODES.txt", col_names = c("genome_1","genome_2"))

# all_transfers_woNODES %>%  filter(genome_1=="L24760") %>%  filter(genome_2=="L24778")
# all_transfers_woNODES %>%  filter(genome_1=="L24778") %>%  filter(genome_2=="L24760")


unique(all_transfers_woNODES$genome_1)

#---------------all nodes
all_genomes <- read_table2("~/Desktop/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_genome_names.txt", col_names = c("genome_name")) %>% pull(genome_name)

all_genomes <- read_delim("~/Desktop/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/perSpeciesCoverage.txt", ":", escape_double = FALSE, trim_ws = TRUE)%>% pull(SPECIES)


#---------------count transfers
i <- 1
transferss <- data.frame()
for (rowsss in all_genomes) {
  
  
    for (columnnss in all_genomes[i:length(all_genomes)]) {
      # print(paste0("comparison of ",rowsss," with ",columnnss ))
      count1 <- all_transfers_woNODES %>% filter(genome_1==rowsss)%>% filter(genome_2==columnnss) %>% nrow()
      count2 <- all_transfers_woNODES %>% filter(genome_1==columnnss)%>% filter(genome_2==rowsss) %>% nrow()
      tmp1 <- data.frame(genome_1=rowsss,genome_2=columnnss,transfers=count1+count2)
      transferss <- rbind(transferss,tmp1)
    }
  
  i <- i+1
}

table(transferss$transfers)

transferss_reduced <- transferss %>%  filter(transfers>2)
 transferss %>%  filter(transfers!=0) %>% select(genome_1) %>% unique() %>% nrow()

ggplot(transferss_reduced,aes(x=transfers))+geom_histogram()+theme_classic()

transferss_final <- transferss %>%  filter(genome_1!=genome_2)
table(transferss$transfers)
table(transferss_final$transfers)

#-----------------------------------------------------------
### add infos
#-----------------------------------------------------------

transferss_reduced %>% filter(genome_1=="Out-helv")
transferss_reduced %>% filter(genome_1=="L24793")
 # table(all_sterm_strain_info_ZOOMED$species)
 
 all_strain_info_ZOOMED <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) %>% select(c(strain,Cheese,species,lineage,starterCulture,Ref)) %>% mutate(group="rmks")
table(all_strain_info_ZOOMED$Ref)
# transferss_reduced %>% left_join(transferss_reduced,all_strain_info_ZOOMED,by.x= "genome_1",by.y="strain")

tmp1 <- merge(transferss_reduced,all_strain_info_ZOOMED,by.x= "genome_1",by.y="strain",all.x = TRUE)
transferss_reduced_extended <- merge(tmp1,all_strain_info_ZOOMED,by.x= "genome_2",by.y="strain",all.x = TRUE) %>%
    mutate(lineage.x = if_else(is.na(lineage.x), 15, lineage.x)) %>%
    mutate(lineage.y = if_else(is.na(lineage.y), 15, lineage.y))  %>%
    mutate(Ref.x = if_else(is.na(Ref.x), "Ref", Ref.x)) %>%
    mutate(Ref.y = if_else(is.na(Ref.y), "Ref", Ref.y)) 

#%>%replace(is.na(lineage.x), 15) #%>%replace(is.na("lineage.y"), 15)




transferss_reduced_extended %>% filter(genome_1=="Out-helv")

transferss_reduced_extended_reduced <- transferss_reduced_extended %>% filter(lineage.x!=lineage.y)%>% filter(Ref.x=="Ref")%>% filter(Ref.y=="Ref")

unique(transferss_reduced_extended_reduced$genome_1)
transferss_reduced <- transferss_reduced_extended_reduced

write.table(transferss_reduced_extended_reduced,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree//transferss_reduced_extended_reduced.tsv",sep="\t",quote = FALSE,row.names = FALSE)


#-----------------------------------------------------------
#---------------gene sharing network
#-----------------------------------------------------------

library(igraph)

# Sterm_net <- graph_from_data_frame(Sterm_cluster_sub,direct=F)


##==================================
 ##new analysis
  ##==================================
 library(ggnetwork)
# install.packages("ITNr")
# install.packages("intergraph")

 library(ITNr)
 library(intergraph)
 Sterm_net <- graph_from_data_frame(transferss_reduced,direct=F)

 net<-asNetwork(Sterm_net) #ggnetwork requires a network object
 n<-ggnetwork(net)
##------------------
##color and size change
##------------------
 
###--------
##coloring
#---------------
# n[grep("RMK202_",n$vertex.names),"vertex.names"] %>% table()
 # n$color <- as.character("Viral DB")
 # n$color[grep("RMK202_",n$vertex.names)] <- "RMK202 phages"

 
 
 # n$color = factor(n$color, levels=c("RMK202 phages","Viral DB"))
 # colorzzz <- c("red","grey50")
 # size <- c("4","1.5")
 ##------------------
 ##extend
 ##------------------
 
 # nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE)
 # nrow(n)
 # nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE) 
 # nrow(nExtended)
 # nExtended <- n
 # ccolnames(nExtended)
# colnames(nExtended)
# colnames(nExtended) <- revalue(colnames(nExtended), c("Adj P-value"="Adj_P_value","Genera in VC"="Genera_in_VC","Topology Confidence Score"="Topology_Confidence_Score","Genus Confidence Score"="Genus_Confidence_Score"))

# nExtended$ad
 ##------------------
 ##plot
 ##------------------
 networkplot <- ggplot(n, aes(x = x, y = y, xend = xend, yend = yend,text =paste("genome:", vertex.names))) +
   geom_nodes() +
    # geom_edges(aes(alpha=transfers)) +
   geom_edges(alpha=0.2) +
   # scale_fill_manual(values="gray41")+
    # scale_color_manual(values=colorzzz)+
    # scale_size_manual(values=as.numeric(size))+
   # guides(color=FALSE,size=FALSE)+ #Remove legends
 theme_blank()+theme(legend.title = element_blank())
 networkplot
  
 library(plotly)
ggp <- ggplotly(networkplot)
 ggp
 
  # svg("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset_ldel_02.svg",width=7,height=7)
 # png("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset.png", width = 2800, height = 2800,res=300)
  # networkplot
   # dev.off()
 
 ##-----------------------
 #add node info
 
 # table(all_sterm_strain_info_ZOOMED$species)
 
 all_sterm_strain_info_ZOOMED <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) %>% filter(species=="Sterm") %>% select(c(strain,Cheese,starterCulture)) %>% mutate(strainNew=strain) #mutate(strainNew=paste0("S",strain)) 

 all_Ldel_strain_info_ZOOMED <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv",  col_types = cols(starterCulture = col_character(),  strain = col_character())) %>% filter(species!="Sterm") %>% select(c(strain,Cheese,starterCulture)) %>% mutate(strainNew=strain) #%>% mutate(strainNew=paste0("L",strain))

 
all_info <- rbind(all_sterm_strain_info_ZOOMED,all_Ldel_strain_info_ZOOMED) %>% select(-strain)
 
table(all_info$strainNew)

 nExtended <-  merge(n, all_info, by.y="strainNew",by.x="vertex.names",all.x = TRUE)
nExtended$starterCulture

nExtended$starterCulture = factor(nExtended$starterCulture, levels=c("101","105","115","124","150","190","202","203","280","302","305"))

 COLORSSS <- c("#d45400ff","#de592bff","#e36b42ff","#e67d59ff","#eb8f70ff","#eea38aff","#009cf0ff","#33cffaff","#7fe3ffff","#007f7fff","#00dcdcff","#ff0000ff","#0000ffff","grey","#fde623ff","#00ff00ff")

 networkplot <- ggplot(nExtended, aes(x = x, y = y, xend = xend, yend = yend,text =paste("genome:", vertex.names))) +
   geom_nodes(aes(color=starterCulture,fill=starterCulture),size=2) +
    # geom_edges(aes(alpha=transfers)) +
   geom_edges(aes(size=log(transfers)),alpha=0.2) +
   scale_fill_manual(values=COLORSSS)+
    scale_color_manual(values=COLORSSS)+
    # scale_size_manual(values=as.numeric(size))+
   # guides(color=FALSE,size=FALSE)+ #Remove legends
 theme_blank()+theme(legend.title = element_blank())
 networkplot
  
 library(plotly)
ggp <- ggplotly(networkplot)
 ggp
 

  networkplot <- ggplot(nExtended, aes(x = x, y = y, xend = xend, yend = yend,text =paste("genome:", vertex.names))) +
   geom_nodes(aes(color=starterCulture,fill=starterCulture),size=5) +
    # geom_edges(aes(alpha=transfers)) +
    # geom_edges(aes(alpha=transfers)) +
   geom_edges(alpha=0.2) +
   scale_fill_manual(values=COLORSSS)+
    scale_color_manual(values=COLORSSS)+
    # scale_size_manual(values=as.numeric(size))+
   # guides(color=FALSE,size=FALSE)+ #Remove legends
 theme_blank()+theme(legend.title = element_blank())
 networkplot
 
 
 pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/gene_sharing_network.pdf",width=6,height=6)

networkplot

dev.off()
 
unique(nExtended$vertex.names)

nExtended %>% filter(vertex.names %in% c("Out-helv" ,"Out-sali", "Out-vest"))



##--------------------
#different approach
##but samesamebutdfifferent
##--------------------

 library(network)

all_info_final <-  all_info %>% remove_rownames()%>% column_to_rownames(., var = "strainNew")
routes_network <- network(transferss_reduced,vertex.attr = all_info_final, matrix.type = "edgelist", ignore.eval = FALSE)
plot(routes_network, vertex.cex = 3)

###------------------------
#show data
###------------------------
transferss_final_02 <- merge(transferss_final,all_info,by.x="genome_1",by.y="strainNew",all.x=TRUE) %>% rename(., Cheese_1=Cheese)%>% rename(., starterCulture_1=starterCulture)
transferss_final_03 <- merge(transferss_final_02,all_info,by.x="genome_2",by.y="strainNew",all.x=TRUE) %>% rename(., Cheese_2=Cheese)%>% rename(., starterCulture_2=starterCulture)


transferss_final_04 <- transferss_final_03 %>% mutate(comparisionss=ifelse(Cheese_1==Cheese_2,"same cheese","different cheese")) %>% filter(!is.na(comparisionss))
# table(transferss_final_04$comparisionss)
# table(transferss_final_04$transfers)

ggplot(transferss_final_04,aes(x=comparisionss,y=transfers,group=comparisionss,fill=comparisionss))+geom_boxplot()+theme_classic()+ coord_cartesian(ylim=c(-1, 10))

##------------------------------------------
#include the knowledge of the which cluster the strains came from and exclude if they are from the same cluster. 
##------------------------------------------

Poppunk_out <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/Poppunk_out.all","\t", escape_double = FALSE, col_names = c("genome","group"), trim_ws = TRUE)
# dim(Poppunk_out)
# colnames(SNVs_strains) %IN%
Poppunk_out <- rbind(Poppunk_out,data.frame(genome="L24777",group="13"))
Poppunk_out$species <- ifelse(grepl("S",Poppunk_out$genome),"S.thermophilus","L.delbreuckii")
Poppunk_out %>% filter(genome=="L24777")


transferss_final_05 <- merge(transferss_final_04,Poppunk_out,by.x="genome_1",by.y="genome",all.x=TRUE) %>% rename(., group_1=group) %>% rename(., species_1=species)
transferss_final_06 <- merge(transferss_final_05,Poppunk_out,by.x="genome_2",by.y="genome",all.x=TRUE) %>% rename(., group_2=group) %>% rename(., species_2=species)

transferss_final_07 <- transferss_final_06 %>% mutate(popunk_grouping=ifelse(group_1==group_2,"same lineage","different lineage")) %>% filter(!is.na(popunk_grouping))%>% filter(popunk_grouping!="same lineage")%>% filter(group_1!="4")%>% filter(group_2!="4")%>% mutate(species_match=ifelse(species_2!=species_1,"different species",ifelse(species_2=="L.delbreuckii","L.delbreuckii","S.thermophilus")))


ggplot(transferss_final_07,aes(x=comparisionss,y=transfers,group=comparisionss,fill=comparisionss))+geom_boxplot()+theme_classic()+ coord_cartesian(ylim=c(-1, 20))+facet_wrap(~popunk_grouping)

##------------------------------------------
#see if between species or within species (+which species)
##------------------------------------------

ggplot(transferss_final_07,aes(x=comparisionss,y=transfers,group=comparisionss,fill=comparisionss))+geom_boxplot()+theme_classic()+ coord_cartesian(ylim=c(-1, 20))+facet_wrap(~species_match)


```


including also the node mappings


```{r}

library(readr)
library(plyr)
library(tidyverse)

##------------------all edges
all_transfers_withNODES <- read_table2("~/Desktop/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_transfers_withNODES.txt", col_names = c("genome_1","genome_2"))

nodes_2_tips <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/nodes_2_tips.txt",   "\t", escape_double = FALSE, trim_ws = TRUE)

write.table(nodes_2_tips,"~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree//nodes_2_tips.txt",sep="\t",quote = FALSE,row.names = FALSE)

all_transfers_withNODES_02 <- merge(all_transfers_withNODES,nodes_2_tips,by.x="genome_1",by.y="node",all.x=TRUE,all.y=TRUE)  %>% rename(., tip_1=tip) %>% filter(!is.na(tip_1)||!is.na(genome_1))
dim(all_transfers_withNODES_02)
all_transfers_withNODES_02 %>% filter(!is.na(tip_1))

all_transfers_withNODES_03 <- merge(all_transfers_withNODES_02,nodes_2_tips,by.x="genome_2",by.y="node",all.x=TRUE,all.y=TRUE)  %>% rename(., tip_2=tip) %>% filter(!is.na(tip_2)||!is.na(genome_2)) %>% mutate(genome_1=ifelse(is.na(tip_1),genome_1,tip_1))%>% mutate(genome_2=ifelse(is.na(tip_2),genome_2,tip_2))  %>% select(-c(tip_1,tip_2))

dim(all_transfers_withNODES)
dim(all_transfers_withNODES_03)

all_transfers_withNODES <- all_transfers_withNODES_03
# nodes_2_tips%>% filter(node=="N123")
# all_transfers_withNODES_03 %>% filter(genome_1=="N123")

# all_transfers_woNODES %>%  filter(genome_1=="L24760") %>%  filter(genome_2=="L24778")
# all_transfers_woNODES %>%  filter(genome_1=="L24778") %>%  filter(genome_2=="L24760")


#---------------all nodes
all_genomes <- read_table2("~/Desktop/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/ouput_generax_final/all_genome_names.txt", col_names = c("genome_name")) %>% filter(genome_name!="Out")%>% pull(genome_name) 

all_genomes <- c(all_genomes,"Out-helv","Out-sali","Out-vest")

# all_transfers_withNODES$genome_2 %>% unique() %>% table()

#---------------count transfers

all_transfers_woNODES %>% filter(grepl("Out",genome_1)&grepl("Out",genome_2))


i <- 1
transferss <- data.frame()
for (rowsss in all_genomes) {
  
  
    for (columnnss in all_genomes[i:length(all_genomes)]) {
      # print(paste0("comparison of ",rowsss," with ",columnnss ))
      count1 <- all_transfers_woNODES %>% filter(genome_1==rowsss)%>% filter(genome_2==columnnss) %>% nrow()
      count2 <- all_transfers_woNODES %>% filter(genome_1==columnnss)%>% filter(genome_2==rowsss) %>% nrow()
      tmp1 <- data.frame(genome_1=rowsss,genome_2=columnnss,transfers=count1+count2)
      transferss <- rbind(transferss,tmp1)
    }
  
  i <- i+1
}

table(transferss$transfers)

# transferss_reduced <- transferss %>%  filter(transfers>2)
 transferss %>%  filter(transfers>50)

ggplot(transferss,aes(x=transfers))+geom_histogram()+theme_classic()

transferss_final <- transferss %>%  filter(genome_1!=genome_2)
table(transferss$transfers)
table(transferss_final$transfers)


transferss_final %>% filter(grepl("Out",genome_1)|grepl("Out",genome_2))

table(transferss_final$genome_2)

###------------------------
#show data
###------------------------
transferss_final_02 <- merge(transferss_final,all_info,by.x="genome_1",by.y="strainNew",all.x=TRUE) %>% rename(., Cheese_1=Cheese)%>% rename(., starterCulture_1=starterCulture)
transferss_final_03 <- merge(transferss_final_02,all_info,by.x="genome_2",by.y="strainNew",all.x=TRUE) %>% rename(., Cheese_2=Cheese)%>% rename(., starterCulture_2=starterCulture)


transferss_final_04 <- transferss_final_03 %>% mutate(comparisionss=ifelse(Cheese_1==Cheese_2,"same cheese","different cheese")) %>% filter(!is.na(comparisionss))
# table(transferss_final_04$comparisionss)
# table(transferss_final_04$transfers)

ggplot(transferss_final_04,aes(x=comparisionss,y=transfers,group=comparisionss,fill=comparisionss))+geom_boxplot()+theme_classic()+ coord_cartesian(ylim=c(-1, 10))

##------------------------------------------
#include the knowledge of the which cluster the strains came from and exclude if they are from the same cluster. 
##------------------------------------------

Poppunk_out <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GALs_analysis/Gals_analysis/Poppunk_out.all","\t", escape_double = FALSE, col_names = c("genome","group"), trim_ws = TRUE)
# dim(Poppunk_out)
# colnames(SNVs_strains) %IN%
Poppunk_out <- rbind(Poppunk_out,data.frame(genome="L24777",group="13"))
Poppunk_out$species <- ifelse(grepl("S",Poppunk_out$genome),"S.thermophilus","L.delbreuckii")
Poppunk_out %>% filter(genome=="L24777")


transferss_final_05 <- merge(transferss_final_04,Poppunk_out,by.x="genome_1",by.y="genome",all.x=TRUE) %>% rename(., group_1=group) %>% rename(., species_1=species)
transferss_final_06 <- merge(transferss_final_05,Poppunk_out,by.x="genome_2",by.y="genome",all.x=TRUE) %>% rename(., group_2=group) %>% rename(., species_2=species)


transferss_final_06 %>% filter(genome_2=="S24733")
transferss_final_07 <- transferss_final_06 %>% mutate(popunk_grouping=ifelse(group_1==group_2,"same lineage","different lineage")) %>% filter(!is.na(popunk_grouping))%>% filter(group_1!="3")%>% filter(group_2!="3")%>% mutate(species_match=ifelse(species_2!=species_1,"different species",ifelse(species_2=="L.delbreuckii","L.delbreuckii","S.thermophilus"))) %>% filter(transfers>=0) #%>% filter(popunk_grouping!="same lineage")


ggplot(transferss_final_07,aes(x=comparisionss,y=transfers,group=comparisionss,fill=comparisionss))+geom_boxplot()+theme_classic()+ coord_cartesian(ylim=c(-1, 20))+facet_wrap(~popunk_grouping)

##------------------------------------------
#see if between species or within species (+which species)
##------------------------------------------

ggplot(transferss_final_07,aes(x=comparisionss,y=transfers,group=comparisionss,fill=comparisionss))+geom_boxplot()+theme_classic()+ coord_cartesian(ylim=c(-1, 10))+facet_wrap(~species_match)
ggplot(transferss_final_07,aes(x=transfers,group=comparisionss,fill=comparisionss))+geom_density()+theme_classic()+ xlim(c(-1, 10))+facet_wrap(~species_match+comparisionss,ncol=2)

# transferss_final_08 %>% filter(species_match=="S.thermophilus") %>% filter(transfers>3.5)

##------------------------------------------
#starter culture level
##------------------------------------------
# transferss_final_07$starterCulture_1
transferss_final_08 <- transferss_final_07 %>% mutate(comparisionss_starterculture=ifelse(starterCulture_1==starterCulture_2,"same starter culture","different starter culture")) %>% filter(!is.na(comparisionss_starterculture))

boxplot_01 <- ggplot(transferss_final_08,aes(x=comparisionss_starterculture,y=transfers,group=comparisionss_starterculture,fill=comparisionss_starterculture,text =paste("genome_1:", genome_1,"\ngenome_2:", genome_2)))+geom_boxplot()+theme_classic()+ coord_cartesian(ylim=c(-1, 10))+facet_wrap(~species_match)


 library(plotly)
ggp <- ggplotly(boxplot_01)
 ggp
 
 transferss_final_08 %>% select(genome_1) %>% unique() %>% table()
 
 ##------------------------------------------
#every starter culture
##------------------------------------------
 
 genomessss <- "S13495"
 
tmp_2 <- transferss_final_08 %>% filter(genome_1==genomessss|genome_2==genomessss )
 boxplot_02 <- ggplot(tmp_2,aes(x=comparisionss,y=transfers,group=comparisionss_starterculture,fill=comparisionss_starterculture))+geom_boxplot()+theme_classic()+ coord_cartesian(ylim=c(-1, 10))
boxplot_02
  

 ##------------------------------------------
#examples L.del
##------------------------------------------

 genomessss <- "L10973"
 
tmp_2 <- transferss_final_08 %>% filter(genome_1==genomessss|genome_2==genomessss )


 boxplot_02 <- ggplot(tmp_2,aes(x=comparisionss,y=transfers,group=comparisionss_starterculture,fill=comparisionss_starterculture))+geom_boxplot()+theme_classic()+ coord_cartesian(ylim=c(-1, 10))
boxplot_02
  
 ##------------------------------------------
#including phylonetic distance
#like in Gal's twilight zone paper
# phylogenetic distance between every two lineages was measured as the patristic distance using the function ‘cophenetic’ 
##------------------------------------------
library(ape)
# cophenetic()


Sterm_tree_raw <- read.nexus("~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_curated.txt", tree.names = NULL, force.multi = FALSE)

tree_phylo <- Sterm_tree_raw %>% as.phylo()


## get the distance between every two nodes
# tree_phylog <- tree %>% as.phylo()
distance_matrix <-cophenetic.phylo(tree_phylo)
 library(reshape2)
 phylo_distance <- melt(distance_matrix) %>% rename(genome_1=Var1)%>% rename(genome_2=Var2) %>% rename(phylo_distance=value)
 
ggplot(transferss_final_08,aes(x=transfers))+geom_histogram(bins = 200)+theme_classic()
table(transferss_final_08$transfers)

transferss_final_09 <- merge(transferss_final_08,phylo_distance,by=c("genome_1","genome_2"))

transferss_final_09 %>% filter(transfers>100&phylo_distance>0.007)

ggplot(transferss_final_09,aes(x=transfers+0.01,y=phylo_distance+0.01))+geom_point()+theme_classic()+coord_trans(x="log2",y="log2")+s

pointplot_01 <- ggplot(transferss_final_09,aes(x=transfers+0.01,y=phylo_distance+0.01))+geom_point()+theme_classic()+
  scale_y_continuous(trans="log10",breaks =c(0.01,0.1,0.2,0.5,0.8,1))+scale_x_continuous(trans="log10",breaks =c(0,1,10,50,100,200))+labs(x="transfered genes",y="phylogentic distance")    # scale_y_continuous(breaks =c(0,50,100,250,500,1000),labels=c(0,50,100,250,500,1000))+
pointplot_01
# plotly(pointplot_01)
transferss_final_09 %>% filter(transfers>1&phylo_distance>0.5)

 pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/gene_sharing_network_count_01.pdf",width=6,height=5)

pointplot_01

dev.off()

```

I think I need to carefully check if the old genomes are included too and an then systematically check if there are transfers from the old genomes to new ones and if they are within the starter cultures. 




## CRISPR spacers

#### pilerCR
Here, I run pilerCR to extract all spacers to finally check if any of the strain do not have spacers against the RMK202 phage. 

I identify the CRISPR spacers with [PilerCR](https://www.drive5.com/pilercr/).

```{bash}

##=============
##pilerCR
##=============
for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     species_short=$(echo ${species}|head -c1)
     echo ==========================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/${species}
for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/ |grep "^${species_short}"|grep ".fna$" |sed 's/.fna//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR//${species}/${genomes}


  ~/apps/PilerCR/pilercr1.06/pilercr -in /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA//${genomes}.fna \
    -out /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR//${species}/${genomes}/${genomes}.pilarCR_out -noinfo\
    -seq /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR//${species}/${genomes}/${genomes}.pilarCR_out.fasta

  
  perl ~/apps/PilerCR/PilerCR_parser_new_vincent /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR//${species}/${genomes}/${genomes}.pilarCR_out > \
   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR//${species}/${genomes}/${genomes}.pilarCR_out_final
  done #genomes
  #  grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_final_new | sed 's/>//g' |awk -F "__" '{OFS="\t"} {print $4,$2,$3,$1}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_spacer.bed

done #species

###--------------do it from DIALACT sterms

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm
for genomes in $(ls /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20201010/Genomes/Dialact_02/Streptococcus_salivarius/|grep ".fna$" |sed 's/.fna//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}


  ~/apps/PilerCR/pilercr1.06/pilercr -in /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20201010/Genomes/Dialact_02/Streptococcus_salivarius//${genomes}.fna \
    -out  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}/${genomes}.pilarCR_out -noinfo\
    -seq  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}/${genomes}.pilarCR_out.fasta

  
  perl ~/apps/PilerCR/PilerCR_parser_new_vincent  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}/${genomes}.pilarCR_out > \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}/${genomes}.pilarCR_out_final
  done #genomes
```
all CRISPR arrays are in the same orientation.

#### blast CRISPRspacers
Here, I blast the CRISPR spacers against different databases. 
First I want to check if the spacers from the different genomes map against the RMK202 phages. 


```{bash}
###================
##description file
###================
threads=37
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/
species=Sterm
MetaspacerFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse//all_unique_spacers.fasta
mappingFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt

##--------------------------blast to local phages 
  

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     species_short=$(echo ${species}|head -c1)
     echo ==========================
     rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast
for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/ |grep "^${species_short}"|grep ".fna$" |sed 's/.fna//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast/${genomes}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast/${genomes}


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast \
  -query /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR//${species}/${genomes}/${genomes}.pilarCR_out_final \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sacc sblastnames sscinames" \
  -db ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome \
  -out /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast/${genomes}/${genomes}_CRISPRblast_rmk202phages.txt \
  -evalue 0.01 -word_size 16



  done #genomes
done #species



for species in $(echo "Ldel Sterm")
do

for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/ |grep "^${species_short}"|grep ".fna$" |sed 's/.fna//g')
do
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast/${genomes}/${genomes}_CRISPRblast_rmk202phages.txt


  done #genomes
done #species

##--------------------------DIALACT STERM


     rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact
for genomes in $(ls /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20201010/Genomes/Dialact_02/Streptococcus_salivarius/|grep ".fna$" |sed 's/.fna//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/${genomes}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/${genomes}


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 2 -task megablast \
  -query /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}/${genomes}.pilarCR_out_final \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sacc sblastnames sscinames" \
  -db ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome \
  -out /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/${genomes}/${genomes}_CRISPRblast_rmk202phages.txt \
  -evalue 0.01 -word_size 16



  done #genomes


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS.txt
for genomes in $(ls /data/Project/2020_StarterCultureDiversity//07_DialactStrains/20201010/Genomes/Dialact_02/Streptococcus_salivarius/|grep ".fna$" |sed 's/.fna//g')
do
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/${genomes}/${genomes}_CRISPRblast_rmk202phages.txt   >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS.txt


  done #genomes
awk -F " " '{if($1<4)print $0}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS.txt


for samplesss in $(awk -F " " '{if($1<4)print $2}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS.txt|)
do
echo "==============================================="
cat ${samplesss}
done


awk -F " " '{if($1<4)print $0}' /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS.txt | grep -v "FAM1708_" |grep "FAM20897_" -v  |cut -f 2| cut -d '/' -f 9 >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS_lowHITsamples.txt

for genomes in $(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/CRISPRspacerblast_Dialact/num_HITS_lowHITsamples.txt)
do
echo "==============================================="

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/CRISPR/pilerCR/Dialact_sterm/${genomes}/${genomes}.pilarCR_out_final
done

cat 
```




## Gene-sharing network

##### download all reference NCBI strains

Here, I have a script to download all genomes of S.thermophilus also if they are not complete and circular.

```{bash}

##=====================================================================================
##------------------------all complete streptos
##=====================================================================================
species=$(echo "Streptococcus thermophilus")
species_short=Sterm_references
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA

##currently taking all genomes not selecting for highest quality

cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt |grep -e "${species}" cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    \
    /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log//download_sterm.txt
    #pecies_short=Sterm_references

cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {print $1, "S_thermophilus_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log///download_sterm_names.txt

##--------------------------
##prepare list
##--------------------------
species=$(echo "Streptococcus thermophilus")
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {print $1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log_shortNames_00.txt

cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {print $1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/S-thermophilus-/S-/g'| sed 's/-STH-CIRM-/-I-/g' > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log_shortNames_01.txt
rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log//20201216_NCBI_log_shortNames_02.txt
while read -r line
do
#echo $line
echo -e ${line} |head -c 10 |head -1 |sed 's/^$//g' >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log//20201216_NCBI_log_shortNames_02.txt
echo -e "\n" >> /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log//20201216_NCBI_log_shortNames_02.txt

done < /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log_shortNames_01.txt
sed -i '/^[[:space:]]*$/d' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log//20201216_NCBI_log_shortNames_02.txt

paste -d '\t' /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log//20201216_NCBI_log_shortNames_02.txt /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log_shortNames_00.txt > /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log_shortNames_final.txt

#-------------------- add outgroup

outgroupNAME=NCTC8618

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >>    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
    species_short=Sterm_references

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_salivarius_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'>> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm_names.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print "Out-sali",$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print ,$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/S-thermophilus-/S-/g'| sed 's/-STH-CIRM-/-I-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt


cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
gunzip *.fna.gz

#-------------------- add outgroup
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
outgroupNAME=$(echo "Streptococcus vestibularis")

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >>    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
    species_short=Sterm_references

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_salivarius_"$9$10}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g'>> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm_names.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${outgroupNAME}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print "Out-sali",$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'   >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print ,$1,$2,$3,"S_thermophilus_"$9$10,$15,$17}'|sed 's/strain=//g' | sed 's/_/-/g' |sed 's/GCF-/GCF_/g' |sed 's/ /-/g'|cut -f 4 |sed 's/S-thermophilus-/S-/g'| sed 's/-STH-CIRM-/-I-/g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt


cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
gunzip *.fna.gz

mv GCF_900636445.1_41965_G01_genomic.fna Out-vest.fna

rm Out-sali.fna
```

to start of with I use the already generated collection of complete and ciruclare genomes create on 20200728 to do the analysis. 

test outgroup with FASTani

```{bash}

echo "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/FNA/Out-sali.fna" > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/fastaANI_out_test.txt
echo "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/FNA/Out-vest.fna" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/fastaANI_out_test.txt
echo "/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/FNA/S-LMD-9.fna" >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/fastaANI_out_test.txt

###---------------------------
##fastaANI
###---------------------------

fastANI --fragLen 1000 -t 37 --ql /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/fastaANI_out_test.txt \
  --rl /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/fastaANI_out_test.txt \
  -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/fastaANI_out_test.fastani
```


####Orthofinder REFTREE

Here, I use orthofinder to create a nice reference tree 

```{bash}
###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Sterm

for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FAA/${leterzzz}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
done


###-----------------------------------------
##orthorfinder
###-----------------------------------------
#for species in $(echo "Ldel Sterm"|tail -1)
for species in $(echo "Sterm"|tail -1)
do

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/


    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}// \
      -t 35 \
      -o /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ \
      -a 35

done # species

```

move local

```{bash}
##--------------local

mkdir -p /home/vincent/Desktop/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm/Results_Aug05/Species_Tree/

scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm/Results_Aug05/Species_Tree/SpeciesTree_rooted* /home/vincent/Desktop/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm/Results_Aug05/Species_Tree/
```

preperation for itol

```{bash}
cut -f 5,8,9 /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm_edited.csv|sed '1d' > /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm_edited_forITOL.csv
```

#### POPCOGent


```{bash}
#conda activate PopCOGenT

#adjust the /home/vincent/apps/PopCOGenT/src/PopCOGenT/config.sh file and then run /home/vincent/apps/PopCOGenT/src/PopCOGenT/PopCOGenT.sh

###-----------------------------------------
##combine data
###-----------------------------------------

species=Sterm

for species in $(echo "Ldel Sterm")
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/FNA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/${leterzzz}*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/
done


##-------------
#gather Sterm genomes
##-------------
rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test/
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test//{genomes,output}


cp -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species} \
/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test/


##-------------
#run script
##-------------

#problem with networkx version
#pip install --upgrade networkx==1.11

#export MUGSY_INSTALL=/home/vincent/apps/mugsy_x86-64-v1r2.3
#source /home/vincent/apps/mugsy_x86-64-v1r2.3/mugsyenv.sh

##!!!!!!!!!!!!!!!!!!!!!!!CHANGE CONFIG.SH file

cd /home/vincent/apps/PopCOGenT/src/PopCOGenT
bash /home/vincent/apps/PopCOGenT/src/PopCOGenT/PopCOGenT.sh
 
 ##===============
 #sweeps
 ##===============
 
cd /home/vincent/apps/PopCOGenT/src/core_gene_sweeps/
source activate PopCOGenT
python phybreak1.generate_maf.py
python phybreak2.maf_to_fasta.py


python phybreak3.MSAsubset_runPhyML.py

```

```{bash}

mkdir -p ~/Desktop/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test/output/
scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test/output/* ~/Desktop/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test/output/


```


read the graphML in r

```{r}
library(igraph)
 library(ggnetwork)
 library(ITNr)
 library(intergraph)
library(tidyverse)

net1<-read.graph("~/Desktop/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/PopCOGenT/201216_test/output/all_ncbi_rmk_Sterm_0.000355362.txt.unclust.graphml", format = "graphml")
# names(net1)
 

# net1[]
vcount(net1)
ecount(net1)
graph.density(net1)
diameter(net1)
transitivity(net1)
# hist(degree(net1), col="firebrick", xlab="Vertex Degree", ylab="Frequency", main="")


net<-asNetwork(net1) #ggnetwork requires a network object
 n<-ggnetwork(net)
 hist(n$weight)
 
 nExtended <- n# %>% filter(weight>300)
 networkplot <- ggplot(nExtended, aes(x = x, y = y, xend = xend, yend = yend,text =paste("ID:", id))) + geom_edges(color = "lightgray") +
   geom_nodes()+ theme_blank()
 networkplot
 
 library(plotly)
ggp <- ggplotly(networkplot)
 ggp
 
 
 length(table(n$id))
 n$weight
 ggplot(n,aes(x=weight))+geom_density()+facet_wrap(~id,scales = "free")+coord_trans(x="log10")+geom_vline(xintercept = 300)
 
 
 nExtended$catWeight <- ifelse(nExtended$weight<300,"low",ifelse(nExtended$weight<1000,"medium","high"))
 
  networkplot <- ggplot(data=nExtended) + geom_edges( aes(x = x, y = y, xend = xend, yend = yend,size=weight,alpha=catWeight)) +
   geom_nodes(aes(x = x, y = y, xend = xend, yend = yend,size=4,text =paste("ID:", id)))+ theme_blank()+ scale_alpha(range = c(0.6,0))
 networkplot
 ggp <- ggplotly(networkplot)
 ggp
 
 ###---------------
 
 ga<-read.graph("~/Desktop/Project/2021_Recominbation/PopCOGenT/201209_test/output/RMK_Sterm_0.000355362.txt.unclust.graphml", format = "graphml")

 clu <- cluster_louvain(ga)

#membership vector
mem <- membership(clu)
 com <- communities(clu)
com[[1]]
```




## Recombinants


move local

```{bash}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/S_O_202_24854.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/S_O_202_24854.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/S_O_202_24855.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/
progressiveMauve --output=my_seqs.xmfa \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/S_O_202_24855.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/S_O_202_24854.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/S_O_202_24854.fasta
```


```{bash}
##--------------local

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/ /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/Recombinant_202/
```

# phage assembly (from metagenome)

Here, I want to assemble all non-mapping reads

## mapping to all genomes


### create a mapping DB

First I create a reference DB containing all genomes from that cheese starter culture. 

```{bash}

for starterCultures in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/|cut -d '_' -f 3 |sort|uniq)
do
echo -e "starter culture :" ${starterCultures}

mkdir -p /data/Project/2020_StarterCultureDiversity/phage4meta/01_bacterialREFdb/
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/*_${starterCultures}_*.fasta > /data/Project/2020_StarterCultureDiversity/phage4meta/01_bacterialREFdb/${starterCultures}.fasta
 
 cat /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/Ldelv_mag.fasta >> \
  /data/Project/2020_StarterCultureDiversity/phage4meta/01_bacterialREFdb/${starterCultures}.fasta
 
done

```


```{bash}


names=Ws_150_19
#/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 index ${Assembly}
#/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 mem -t 16 ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz/${namezzz}-R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz/${namezzz}-R2_val_2.fq.gz |head
##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##script
###================
##============
##mapping to reference
##============
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

#mkdir -p ${logFilelocation}
num=1
samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt ##the file with all sample names
logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
threads=37
BaseLocation=/data/Project/2020_StarterCultureDiversity/phage4meta/01_bacterialREFdb/02_mapping2REFS
#rm ${BaseLocation}/log/multiqc_logfile_finalGenome.txt

for names in $(cut -f 1 ${samplesss} |grep "Lyo_105_76_1" -A 100)
  do

  starterCultures=$(echo ${names} |cut -d '_' -f 2)

Assembly=/data/Project/2020_StarterCultureDiversity/phage4meta/01_bacterialREFdb/${starterCultures}.fasta

  echo ${num}"/95  :" ${names} "from " ${starterCultures}
  num=$((num+1))
  
 bwa index $Assembly

  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 #/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 index ${Assembly} 
  
#  /home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 mem -t ${threads} ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
bwa mem -t ${threads} ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${ynames}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   



tail -95 ${BaseLocation}/log/multiqc_logfile_finalGenome.txt > ${BaseLocation}/log/multiqc_logfile_finalGenome_onlyMETA.txt
   
    rm -r ${BaseLocation}/MultiQC_onlyMETA
 
  mkdir -p ${BaseLocation}/MultiQC_onlyMETA
  
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome_onlyMETA.txt \
    -o ${BaseLocation}/MultiQC_onlyMETA

#/home/vincent/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/MultiQC/multiqc_report.html

scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/MultiQC/multiqc_report.html /Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/

##-------------------------------------------------  
##-----check read coverage
##-------------------------------------------------  


##-------------------------------------------------  
#name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
threads=37
names=Konserve_202


num=1
rm -r ${BaseLocation}/log/mappings.txt
#rm -r ${BaseLocation}/log/mappings_species.txt


for names in $(cut -f 1 ${samplesss})
  do
mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')

#mapped_sterm=$(grep "CP046134" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)
#mapped_ldel=$(grep "CP046131" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)
rm ${BaseLocation}/${names}/bwaMapping2DB//bamqc
mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc
qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G

#mkdir -p ${BaseLocation}/log
echo -e ${names}"\tonlyMeta\t"${mapped} >> ${BaseLocation}/log/mappings.txt

#echo -e ${names}"\tLactobacillus delbrueckii\t"$mapped_ldel >> ${BaseLocation}/log/mappings_species.txt
#echo -e ${names}"\tStreptococcus thermophilusi\t"$mapped_sterm >> ${BaseLocation}/log/mappings_species.txt

done

##-------------------------------------------------  
##move local
##-------------------------------------------------  

##bring together 

cat /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings.txt \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202/log/mappings.txt > \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings_final.txt


scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/log/mappings_species.txt Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/
```

####mapping strains to genome

```{bash}

threads=37
samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames.txt ##the file with all sample names
logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref
Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv.fasta

##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1

for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/)
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)
names=$(echo ${samplezzKurz})
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 # gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq
  
  
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} ${Assembly} \
  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   
   
    rm -r ${BaseLocation}/MultiQC
 
  mkdir -p ${BaseLocation}/MultiQC
  
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome.txt \
    -o ${BaseLocation}/MultiQC

##-------------------------------------------------  
##-----check read coverage
##-------------------------------------------------  


##-------------------------------------------------  
#name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
threads=37
names=Konserve_202


num=1
rm -r ${BaseLocation}/log/mappings.txt
#rm -r ${BaseLocation}/log/mappings_species.txt


for names in $(cut -f 1 ${samplesss})
  do
mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')

#mapped_sterm=$(grep "CP046134" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)
#mapped_ldel=$(grep "CP046131" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt|sed 's/^[[:blank:]]*//g'|cut -f 4)

#mkdir -p ${BaseLocation}/log
echo -e ${names}"\tonlyMeta\t"${mapped} >> ${BaseLocation}/log/mappings.txt

#echo -e ${names}"\tLactobacillus delbrueckii\t"$mapped_ldel >> ${BaseLocation}/log/mappings_species.txt
#echo -e ${names}"\tStreptococcus thermophilusi\t"$mapped_sterm >> ${BaseLocation}/log/mappings_species.txt

done

grep "0." ${BaseLocation}/log/mappings_species.txt | sort -k 4

grep -v "0.0" ${BaseLocation}/log/mappings_species.txt|awk '{ sum += $4; n++ } END { if (n > 0) print sum / n; }'

```



```{r}
naming_DF <- read_csv("~/Desktop/Projects/2020_StarterCultureDiversity/98_script/naming_file.csv")
###----------------------------
##script
###----------------------------
library(readr)
library(ggplot2)
library(tidyverse)
#library(plyr)

species_coverage <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/mappings_species.txt", "\t", escape_double = FALSE, col_names = c("sample","species","coverage"),trim_ws = TRUE)

species_coverage$sample

newNames <- setNames(naming_DF$namesNew,naming_DF$namesOLD)
species_coverage$sample <- revalue(species_coverage$sample, newNames)
species_coverage$sample = factor(species_coverage$sample, levels=c(as.character(naming_DF$namesNew)))


newNames <- setNames(as.character(naming_DF$CultureName),naming_DF$namesNew)
species_coverage$culture <- revalue(as.character(species_coverage$sample),newNames)
levels(species_coverage$culture)

species_coverage$coverageLog <- log2(species_coverage$coverage)

all_colours <- (c("#EB4D4D","#10B552") ) 



##--------------------
##ploting


pMOTUs <-  ggplot( data = species_coverage,aes(y = coverage, x = sample, group=species,fill = culture,color=culture))+geom_point(size=2)+geom_hline(yintercept=250)+ # geom_bar( stat="identity")+
  facet_wrap(~species,scales = "free_x")+
   coord_trans(y="log10")+
    labs("",
         x="",
         y="Coverage")+
    theme_classic()+
   scale_color_viridis(discrete=TRUE)+
      scale_y_continuous(breaks =c(0,50,100,250,500,1000),labels=c(0,50,100,250,500,1000))+
  scale_fill_manual(values=all_colours)+
  scale_color_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=4),
      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          # legend.title = element_blank()
          )

  pMOTUs

  
  
  svg("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/CoverageSpecies.svg",width=5,height=5)
   # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)

pMOTUs

dev.off()

```


## assemble unmapped

```{bash}


  samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt ##the file with all sample names

for samples in $(cut -f 1 $samplesss )
do
echo "=============================="
echo ${samples}


rm -r //data/Project/2020_StarterCultureDiversity/MAGs/assembly/${samples}/
mkdir -p /data/Project/2020_StarterCultureDiversity/MAGs/assembly/${samples}/

spades --meta -t 37 -1 /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${samples}/${samples}-R1_kneaddata.fq.gz -2 /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${samples}/${samples}-R2_kneaddata.fq.gz -o /data/Project/2020_StarterCultureDiversity/MAGs/assembly/${samples}/

done

##------------------------------------
#only contigs binrg together
##------------------------------------

mkdir -p /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/perFile

for samples in $(cut -f 1 $samplesss )
do
echo "=============================="
echo ${samples}

cat /data/Project/2020_StarterCultureDiversity/MAGs/assembly/${samples}//contigs.fasta > /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/perFile/${samples}.fasta

done



##------------------------------------
#merge all files
##------------------------------------


rm /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/Alpage_unmappedAseemblies.fasta

for samples in $(cut -f 1 $samplesss )
do
echo "=============================="
echo ${samples}

sed 's/C_202/c_202/g' /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/perFile/${samples}.fasta | sed "s/>/>${samples}C/g"  >> /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/Alpage_unmappedAseemblies.fasta

#|sed "s/_length.*//g"
done

grep ">" /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/Alpage_unmappedAseemblies.fasta |head

grep ">"  /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/Alpage_unmappedAseemblies.fasta  |grep EH


```




### cd-hit assemblies

```{bash}

##-------------------------------------
#filter for only long sequences
##-------------------------------------

seqkit seq --min-len 1000 -M 80000 /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/Alpage_unmappedAseemblies.fasta > /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/Alpage_unmappedAseemblies_min1000_max80000.fasta

seqkit seq --min-len 2000 /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/Alpage_unmappedAseemblies.fasta > /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/Alpage_unmappedAseemblies_min2000.fasta

grep ">" -c  /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/Alpage_unmappedAseemblies.fasta
grep ">" -c  /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/Alpage_unmappedAseemblies_min2000.fasta

##-------------------------------------
#cd-hit the entire assembly
##-------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether//cdHIT/
mkdir -p /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether//cdHIT/

cd-hit-est -d 0 -r 1 -M 3298 -i  /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/Alpage_unmappedAseemblies_min2000.fasta  -o /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/cdHIT/Alpage_unmappedAseemblies_min2000.fasta -c 0.90 -T 35

##-------------------------------------
#quickly analyse
##-------------------------------------

grep "*" /media/vincent/Elements1/Projects/2021_FSK_Alpage/Mapping_to_REF/20220525_mapping/Assembled_unmapped/Alltogether/cdHIT/Alpage_unmappedAseemblies_min1000_cdHit.clstr |cut -f 2 |cut -d ',' -f 1 |sort -n


##-------------------------------------
#WTP running
##-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/MAGs/assembly/WTP/
mkdir -p /data/Project/2020_StarterCultureDiversity/MAGs/assembly/WTP/
cd /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/cdHIT/


cp Alpage_unmappedAseemblies_min2000.fasta Alpage_unmappedAseemblies_min2000.fa
cp Alpage_unmappedAseemblies_min2000.fasta Alpage_unmappedAseemblies_min2000.fna
cp  /data/Project/2020_StarterCultureDiversity/MAGs/assembly/Alltogether/cdHIT/Alpage_unmappedAseemblies_min2000.fasta /media/vincent/Elements1/Projects/2020_StarterCultureDiversity/MAGs/assembly/Alpage_unmappedAseemblies_min2000.fa


grep ">" /media/vincent/Elements1/Projects/2020_StarterCultureDiversity/MAGs/assembly/Alpage_unmappedAseemblies_min2000.fa |head

cd 

rm -r /media/vincent/Elements1/Projects/2020_StarterCultureDiversity/MAGs/assembly/WTP_2/
mkdir -p /media/vincent/Elements1/Projects/2020_StarterCultureDiversity/MAGs/assembly/WTP_2/

cd /media/vincent/Elements1/Projects/2020_StarterCultureDiversity/MAGs/assembly/WTP_2/


sudo nextflow run  replikation/What_the_Phage --fasta /media/vincent/Elements1/Projects/2020_StarterCultureDiversity/MAGs/assembly/Alpage_unmappedAseemblies_min2000.fa --cores 35 -profile local,docker  -r v1.2.0      --output /media/vincent/Elements1/Projects/2020_StarterCultureDiversity/MAGs/assembly/WTP_2/wtp --workdir /media/vincent/Elements1/Projects/2020_StarterCultureDiversity/MAGs/assembly/WTP_2/  --database /media/vincent/Elements1/Database/WTF/nextflow-autodownload-databases/


###recreate error

cd /media/vincent/Elements1/Projects/2020_StarterCultureDiversity/MAGs/assembly/WTP_2/
bash .command.run

##--------------------------------------------------
#bring local

mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/WTF/
scp vincent@130.223.110.142:/media/vincent/Elements1/Projects/2020_StarterCultureDiversity/MAGs/assembly/WTP_2/wtp/final_report.html ~/Desktop/Projects/2020_StarterCultureDiversity/WTF/


/media/vincent/Elements1/Projects/2020_StarterCultureDiversity/MAGs/assembly/WTP_2/wtp



```







