<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Vincent Somerville" />


<title>Starter culture domestication main script</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>
<script>/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.org/license */
!function(e,t){"use strict";"object"==typeof module&&"object"==typeof module.exports?module.exports=e.document?t(e,!0):function(e){if(!e.document)throw new Error("jQuery requires a window with a document");return t(e)}:t(e)}("undefined"!=typeof window?window:this,function(C,e){"use strict";var t=[],r=Object.getPrototypeOf,s=t.slice,g=t.flat?function(e){return t.flat.call(e)}:function(e){return t.concat.apply([],e)},u=t.push,i=t.indexOf,n={},o=n.toString,v=n.hasOwnProperty,a=v.toString,l=a.call(Object),y={},m=function(e){return"function"==typeof e&&"number"!=typeof e.nodeType&&"function"!=typeof e.item},x=function(e){return null!=e&&e===e.window},E=C.document,c={type:!0,src:!0,nonce:!0,noModule:!0};function b(e,t,n){var r,i,o=(n=n||E).createElement("script");if(o.text=e,t)for(r in c)(i=t[r]||t.getAttribute&&t.getAttribute(r))&&o.setAttribute(r,i);n.head.appendChild(o).parentNode.removeChild(o)}function w(e){return null==e?e+"":"object"==typeof e||"function"==typeof e?n[o.call(e)]||"object":typeof e}var f="3.6.0",S=function(e,t){return new S.fn.init(e,t)};function p(e){var t=!!e&&"length"in e&&e.length,n=w(e);return!m(e)&&!x(e)&&("array"===n||0===t||"number"==typeof t&&0<t&&t-1 in e)}S.fn=S.prototype={jquery:f,constructor:S,length:0,toArray:function(){return s.call(this)},get:function(e){return null==e?s.call(this):e<0?this[e+this.length]:this[e]},pushStack:function(e){var t=S.merge(this.constructor(),e);return t.prevObject=this,t},each:function(e){return S.each(this,e)},map:function(n){return this.pushStack(S.map(this,function(e,t){return n.call(e,t,e)}))},slice:function(){return this.pushStack(s.apply(this,arguments))},first:function(){return this.eq(0)},last:function(){return this.eq(-1)},even:function(){return this.pushStack(S.grep(this,function(e,t){return(t+1)%2}))},odd:function(){return this.pushStack(S.grep(this,function(e,t){return t%2}))},eq:function(e){var t=this.length,n=+e+(e<0?t:0);return this.pushStack(0<=n&&n<t?[this[n]]:[])},end:function(){return this.prevObject||this.constructor()},push:u,sort:t.sort,splice:t.splice},S.extend=S.fn.extend=function(){var e,t,n,r,i,o,a=arguments[0]||{},s=1,u=arguments.length,l=!1;for("boolean"==typeof a&&(l=a,a=arguments[s]||{},s++),"object"==typeof a||m(a)||(a={}),s===u&&(a=this,s--);s<u;s++)if(null!=(e=arguments[s]))for(t in e)r=e[t],"__proto__"!==t&&a!==r&&(l&&r&&(S.isPlainObject(r)||(i=Array.isArray(r)))?(n=a[t],o=i&&!Array.isArray(n)?[]:i||S.isPlainObject(n)?n:{},i=!1,a[t]=S.extend(l,o,r)):void 0!==r&&(a[t]=r));return a},S.extend({expando:"jQuery"+(f+Math.random()).replace(/\D/g,""),isReady:!0,error:function(e){throw new Error(e)},noop:function(){},isPlainObject:function(e){var t,n;return!(!e||"[object Object]"!==o.call(e))&&(!(t=r(e))||"function"==typeof(n=v.call(t,"constructor")&&t.constructor)&&a.call(n)===l)},isEmptyObject:function(e){var t;for(t in e)return!1;return!0},globalEval:function(e,t,n){b(e,{nonce:t&&t.nonce},n)},each:function(e,t){var n,r=0;if(p(e)){for(n=e.length;r<n;r++)if(!1===t.call(e[r],r,e[r]))break}else for(r in e)if(!1===t.call(e[r],r,e[r]))break;return e},makeArray:function(e,t){var n=t||[];return null!=e&&(p(Object(e))?S.merge(n,"string"==typeof e?[e]:e):u.call(n,e)),n},inArray:function(e,t,n){return null==t?-1:i.call(t,e,n)},merge:function(e,t){for(var n=+t.length,r=0,i=e.length;r<n;r++)e[i++]=t[r];return e.length=i,e},grep:function(e,t,n){for(var r=[],i=0,o=e.length,a=!n;i<o;i++)!t(e[i],i)!==a&&r.push(e[i]);return r},map:function(e,t,n){var r,i,o=0,a=[];if(p(e))for(r=e.length;o<r;o++)null!=(i=t(e[o],o,n))&&a.push(i);else for(o in e)null!=(i=t(e[o],o,n))&&a.push(i);return g(a)},guid:1,support:y}),"function"==typeof Symbol&&(S.fn[Symbol.iterator]=t[Symbol.iterator]),S.each("Boolean Number String Function Array Date RegExp Object Error Symbol".split(" "),function(e,t){n["[object "+t+"]"]=t.toLowerCase()});var d=function(n){var e,d,b,o,i,h,f,g,w,u,l,T,C,a,E,v,s,c,y,S="sizzle"+1*new Date,p=n.document,k=0,r=0,m=ue(),x=ue(),A=ue(),N=ue(),j=function(e,t){return e===t&&(l=!0),0},D={}.hasOwnProperty,t=[],q=t.pop,L=t.push,H=t.push,O=t.slice,P=function(e,t){for(var n=0,r=e.length;n<r;n++)if(e[n]===t)return n;return-1},R="checked|selected|async|autofocus|autoplay|controls|defer|disabled|hidden|ismap|loop|multiple|open|readonly|required|scoped",M="[\\x20\\t\\r\\n\\f]",I="(?:\\\\[\\da-fA-F]{1,6}"+M+"?|\\\\[^\\r\\n\\f]|[\\w-]|[^\0-\\x7f])+",W="\\["+M+"*("+I+")(?:"+M+"*([*^$|!~]?=)"+M+"*(?:'((?:\\\\.|[^\\\\'])*)'|\"((?:\\\\.|[^\\\\\"])*)\"|("+I+"))|)"+M+"*\\]",F=":("+I+")(?:\\((('((?:\\\\.|[^\\\\'])*)'|\"((?:\\\\.|[^\\\\\"])*)\")|((?:\\\\.|[^\\\\()[\\]]|"+W+")*)|.*)\\)|)",B=new RegExp(M+"+","g"),$=new RegExp("^"+M+"+|((?:^|[^\\\\])(?:\\\\.)*)"+M+"+$","g"),_=new RegExp("^"+M+"*,"+M+"*"),z=new RegExp("^"+M+"*([>+~]|"+M+")"+M+"*"),U=new RegExp(M+"|>"),X=new RegExp(F),V=new RegExp("^"+I+"$"),G={ID:new RegExp("^#("+I+")"),CLASS:new RegExp("^\\.("+I+")"),TAG:new RegExp("^("+I+"|[*])"),ATTR:new RegExp("^"+W),PSEUDO:new RegExp("^"+F),CHILD:new RegExp("^:(only|first|last|nth|nth-last)-(child|of-type)(?:\\("+M+"*(even|odd|(([+-]|)(\\d*)n|)"+M+"*(?:([+-]|)"+M+"*(\\d+)|))"+M+"*\\)|)","i"),bool:new RegExp("^(?:"+R+")$","i"),needsContext:new RegExp("^"+M+"*[>+~]|:(even|odd|eq|gt|lt|nth|first|last)(?:\\("+M+"*((?:-\\d)?\\d*)"+M+"*\\)|)(?=[^-]|$)","i")},Y=/HTML$/i,Q=/^(?:input|select|textarea|button)$/i,J=/^h\d$/i,K=/^[^{]+\{\s*\[native \w/,Z=/^(?:#([\w-]+)|(\w+)|\.([\w-]+))$/,ee=/[+~]/,te=new RegExp("\\\\[\\da-fA-F]{1,6}"+M+"?|\\\\([^\\r\\n\\f])","g"),ne=function(e,t){var n="0x"+e.slice(1)-65536;return t||(n<0?String.fromCharCode(n+65536):String.fromCharCode(n>>10|55296,1023&n|56320))},re=/([\0-\x1f\x7f]|^-?\d)|^-$|[^\0-\x1f\x7f-\uFFFF\w-]/g,ie=function(e,t){return t?"\0"===e?"\ufffd":e.slice(0,-1)+"\\"+e.charCodeAt(e.length-1).toString(16)+" ":"\\"+e},oe=function(){T()},ae=be(function(e){return!0===e.disabled&&"fieldset"===e.nodeName.toLowerCase()},{dir:"parentNode",next:"legend"});try{H.apply(t=O.call(p.childNodes),p.childNodes),t[p.childNodes.length].nodeType}catch(e){H={apply:t.length?function(e,t){L.apply(e,O.call(t))}:function(e,t){var n=e.length,r=0;while(e[n++]=t[r++]);e.length=n-1}}}function se(t,e,n,r){var i,o,a,s,u,l,c,f=e&&e.ownerDocument,p=e?e.nodeType:9;if(n=n||[],"string"!=typeof t||!t||1!==p&&9!==p&&11!==p)return n;if(!r&&(T(e),e=e||C,E)){if(11!==p&&(u=Z.exec(t)))if(i=u[1]){if(9===p){if(!(a=e.getElementById(i)))return n;if(a.id===i)return n.push(a),n}else if(f&&(a=f.getElementById(i))&&y(e,a)&&a.id===i)return n.push(a),n}else{if(u[2])return H.apply(n,e.getElementsByTagName(t)),n;if((i=u[3])&&d.getElementsByClassName&&e.getElementsByClassName)return H.apply(n,e.getElementsByClassName(i)),n}if(d.qsa&&!N[t+" "]&&(!v||!v.test(t))&&(1!==p||"object"!==e.nodeName.toLowerCase())){if(c=t,f=e,1===p&&(U.test(t)||z.test(t))){(f=ee.test(t)&&ye(e.parentNode)||e)===e&&d.scope||((s=e.getAttribute("id"))?s=s.replace(re,ie):e.setAttribute("id",s=S)),o=(l=h(t)).length;while(o--)l[o]=(s?"#"+s:":scope")+" "+xe(l[o]);c=l.join(",")}try{return H.apply(n,f.querySelectorAll(c)),n}catch(e){N(t,!0)}finally{s===S&&e.removeAttribute("id")}}}return g(t.replace($,"$1"),e,n,r)}function ue(){var r=[];return function e(t,n){return r.push(t+" ")>b.cacheLength&&delete e[r.shift()],e[t+" "]=n}}function le(e){return e[S]=!0,e}function ce(e){var t=C.createElement("fieldset");try{return!!e(t)}catch(e){return!1}finally{t.parentNode&&t.parentNode.removeChild(t),t=null}}function fe(e,t){var n=e.split("|"),r=n.length;while(r--)b.attrHandle[n[r]]=t}function pe(e,t){var n=t&&e,r=n&&1===e.nodeType&&1===t.nodeType&&e.sourceIndex-t.sourceIndex;if(r)return r;if(n)while(n=n.nextSibling)if(n===t)return-1;return e?1:-1}function de(t){return function(e){return"input"===e.nodeName.toLowerCase()&&e.type===t}}function he(n){return function(e){var t=e.nodeName.toLowerCase();return("input"===t||"button"===t)&&e.type===n}}function ge(t){return function(e){return"form"in e?e.parentNode&&!1===e.disabled?"label"in e?"label"in e.parentNode?e.parentNode.disabled===t:e.disabled===t:e.isDisabled===t||e.isDisabled!==!t&&ae(e)===t:e.disabled===t:"label"in e&&e.disabled===t}}function ve(a){return le(function(o){return o=+o,le(function(e,t){var n,r=a([],e.length,o),i=r.length;while(i--)e[n=r[i]]&&(e[n]=!(t[n]=e[n]))})})}function ye(e){return e&&"undefined"!=typeof e.getElementsByTagName&&e}for(e in d=se.support={},i=se.isXML=function(e){var t=e&&e.namespaceURI,n=e&&(e.ownerDocument||e).documentElement;return!Y.test(t||n&&n.nodeName||"HTML")},T=se.setDocument=function(e){var t,n,r=e?e.ownerDocument||e:p;return r!=C&&9===r.nodeType&&r.documentElement&&(a=(C=r).documentElement,E=!i(C),p!=C&&(n=C.defaultView)&&n.top!==n&&(n.addEventListener?n.addEventListener("unload",oe,!1):n.attachEvent&&n.attachEvent("onunload",oe)),d.scope=ce(function(e){return a.appendChild(e).appendChild(C.createElement("div")),"undefined"!=typeof e.querySelectorAll&&!e.querySelectorAll(":scope fieldset div").length}),d.attributes=ce(function(e){return e.className="i",!e.getAttribute("className")}),d.getElementsByTagName=ce(function(e){return e.appendChild(C.createComment("")),!e.getElementsByTagName("*").length}),d.getElementsByClassName=K.test(C.getElementsByClassName),d.getById=ce(function(e){return a.appendChild(e).id=S,!C.getElementsByName||!C.getElementsByName(S).length}),d.getById?(b.filter.ID=function(e){var t=e.replace(te,ne);return function(e){return e.getAttribute("id")===t}},b.find.ID=function(e,t){if("undefined"!=typeof t.getElementById&&E){var n=t.getElementById(e);return n?[n]:[]}}):(b.filter.ID=function(e){var n=e.replace(te,ne);return function(e){var t="undefined"!=typeof e.getAttributeNode&&e.getAttributeNode("id");return t&&t.value===n}},b.find.ID=function(e,t){if("undefined"!=typeof t.getElementById&&E){var n,r,i,o=t.getElementById(e);if(o){if((n=o.getAttributeNode("id"))&&n.value===e)return[o];i=t.getElementsByName(e),r=0;while(o=i[r++])if((n=o.getAttributeNode("id"))&&n.value===e)return[o]}return[]}}),b.find.TAG=d.getElementsByTagName?function(e,t){return"undefined"!=typeof t.getElementsByTagName?t.getElementsByTagName(e):d.qsa?t.querySelectorAll(e):void 0}:function(e,t){var n,r=[],i=0,o=t.getElementsByTagName(e);if("*"===e){while(n=o[i++])1===n.nodeType&&r.push(n);return r}return o},b.find.CLASS=d.getElementsByClassName&&function(e,t){if("undefined"!=typeof t.getElementsByClassName&&E)return t.getElementsByClassName(e)},s=[],v=[],(d.qsa=K.test(C.querySelectorAll))&&(ce(function(e){var t;a.appendChild(e).innerHTML="<a id='"+S+"'></a><select id='"+S+"-\r\\' msallowcapture=''><option selected=''></option></select>",e.querySelectorAll("[msallowcapture^='']").length&&v.push("[*^$]="+M+"*(?:''|\"\")"),e.querySelectorAll("[selected]").length||v.push("\\["+M+"*(?:value|"+R+")"),e.querySelectorAll("[id~="+S+"-]").length||v.push("~="),(t=C.createElement("input")).setAttribute("name",""),e.appendChild(t),e.querySelectorAll("[name='']").length||v.push("\\["+M+"*name"+M+"*="+M+"*(?:''|\"\")"),e.querySelectorAll(":checked").length||v.push(":checked"),e.querySelectorAll("a#"+S+"+*").length||v.push(".#.+[+~]"),e.querySelectorAll("\\\f"),v.push("[\\r\\n\\f]")}),ce(function(e){e.innerHTML="<a href='' disabled='disabled'></a><select disabled='disabled'><option/></select>";var t=C.createElement("input");t.setAttribute("type","hidden"),e.appendChild(t).setAttribute("name","D"),e.querySelectorAll("[name=d]").length&&v.push("name"+M+"*[*^$|!~]?="),2!==e.querySelectorAll(":enabled").length&&v.push(":enabled",":disabled"),a.appendChild(e).disabled=!0,2!==e.querySelectorAll(":disabled").length&&v.push(":enabled",":disabled"),e.querySelectorAll("*,:x"),v.push(",.*:")})),(d.matchesSelector=K.test(c=a.matches||a.webkitMatchesSelector||a.mozMatchesSelector||a.oMatchesSelector||a.msMatchesSelector))&&ce(function(e){d.disconnectedMatch=c.call(e,"*"),c.call(e,"[s!='']:x"),s.push("!=",F)}),v=v.length&&new RegExp(v.join("|")),s=s.length&&new RegExp(s.join("|")),t=K.test(a.compareDocumentPosition),y=t||K.test(a.contains)?function(e,t){var n=9===e.nodeType?e.documentElement:e,r=t&&t.parentNode;return e===r||!(!r||1!==r.nodeType||!(n.contains?n.contains(r):e.compareDocumentPosition&&16&e.compareDocumentPosition(r)))}:function(e,t){if(t)while(t=t.parentNode)if(t===e)return!0;return!1},j=t?function(e,t){if(e===t)return l=!0,0;var n=!e.compareDocumentPosition-!t.compareDocumentPosition;return n||(1&(n=(e.ownerDocument||e)==(t.ownerDocument||t)?e.compareDocumentPosition(t):1)||!d.sortDetached&&t.compareDocumentPosition(e)===n?e==C||e.ownerDocument==p&&y(p,e)?-1:t==C||t.ownerDocument==p&&y(p,t)?1:u?P(u,e)-P(u,t):0:4&n?-1:1)}:function(e,t){if(e===t)return l=!0,0;var n,r=0,i=e.parentNode,o=t.parentNode,a=[e],s=[t];if(!i||!o)return e==C?-1:t==C?1:i?-1:o?1:u?P(u,e)-P(u,t):0;if(i===o)return pe(e,t);n=e;while(n=n.parentNode)a.unshift(n);n=t;while(n=n.parentNode)s.unshift(n);while(a[r]===s[r])r++;return r?pe(a[r],s[r]):a[r]==p?-1:s[r]==p?1:0}),C},se.matches=function(e,t){return se(e,null,null,t)},se.matchesSelector=function(e,t){if(T(e),d.matchesSelector&&E&&!N[t+" "]&&(!s||!s.test(t))&&(!v||!v.test(t)))try{var n=c.call(e,t);if(n||d.disconnectedMatch||e.document&&11!==e.document.nodeType)return n}catch(e){N(t,!0)}return 0<se(t,C,null,[e]).length},se.contains=function(e,t){return(e.ownerDocument||e)!=C&&T(e),y(e,t)},se.attr=function(e,t){(e.ownerDocument||e)!=C&&T(e);var n=b.attrHandle[t.toLowerCase()],r=n&&D.call(b.attrHandle,t.toLowerCase())?n(e,t,!E):void 0;return void 0!==r?r:d.attributes||!E?e.getAttribute(t):(r=e.getAttributeNode(t))&&r.specified?r.value:null},se.escape=function(e){return(e+"").replace(re,ie)},se.error=function(e){throw new Error("Syntax error, unrecognized expression: "+e)},se.uniqueSort=function(e){var t,n=[],r=0,i=0;if(l=!d.detectDuplicates,u=!d.sortStable&&e.slice(0),e.sort(j),l){while(t=e[i++])t===e[i]&&(r=n.push(i));while(r--)e.splice(n[r],1)}return u=null,e},o=se.getText=function(e){var t,n="",r=0,i=e.nodeType;if(i){if(1===i||9===i||11===i){if("string"==typeof e.textContent)return e.textContent;for(e=e.firstChild;e;e=e.nextSibling)n+=o(e)}else if(3===i||4===i)return e.nodeValue}else while(t=e[r++])n+=o(t);return n},(b=se.selectors={cacheLength:50,createPseudo:le,match:G,attrHandle:{},find:{},relative:{">":{dir:"parentNode",first:!0}," ":{dir:"parentNode"},"+":{dir:"previousSibling",first:!0},"~":{dir:"previousSibling"}},preFilter:{ATTR:function(e){return e[1]=e[1].replace(te,ne),e[3]=(e[3]||e[4]||e[5]||"").replace(te,ne),"~="===e[2]&&(e[3]=" "+e[3]+" "),e.slice(0,4)},CHILD:function(e){return e[1]=e[1].toLowerCase(),"nth"===e[1].slice(0,3)?(e[3]||se.error(e[0]),e[4]=+(e[4]?e[5]+(e[6]||1):2*("even"===e[3]||"odd"===e[3])),e[5]=+(e[7]+e[8]||"odd"===e[3])):e[3]&&se.error(e[0]),e},PSEUDO:function(e){var t,n=!e[6]&&e[2];return G.CHILD.test(e[0])?null:(e[3]?e[2]=e[4]||e[5]||"":n&&X.test(n)&&(t=h(n,!0))&&(t=n.indexOf(")",n.length-t)-n.length)&&(e[0]=e[0].slice(0,t),e[2]=n.slice(0,t)),e.slice(0,3))}},filter:{TAG:function(e){var t=e.replace(te,ne).toLowerCase();return"*"===e?function(){return!0}:function(e){return e.nodeName&&e.nodeName.toLowerCase()===t}},CLASS:function(e){var t=m[e+" "];return t||(t=new RegExp("(^|"+M+")"+e+"("+M+"|$)"))&&m(e,function(e){return t.test("string"==typeof e.className&&e.className||"undefined"!=typeof e.getAttribute&&e.getAttribute("class")||"")})},ATTR:function(n,r,i){return function(e){var t=se.attr(e,n);return null==t?"!="===r:!r||(t+="","="===r?t===i:"!="===r?t!==i:"^="===r?i&&0===t.indexOf(i):"*="===r?i&&-1<t.indexOf(i):"$="===r?i&&t.slice(-i.length)===i:"~="===r?-1<(" "+t.replace(B," ")+" ").indexOf(i):"|="===r&&(t===i||t.slice(0,i.length+1)===i+"-"))}},CHILD:function(h,e,t,g,v){var y="nth"!==h.slice(0,3),m="last"!==h.slice(-4),x="of-type"===e;return 1===g&&0===v?function(e){return!!e.parentNode}:function(e,t,n){var r,i,o,a,s,u,l=y!==m?"nextSibling":"previousSibling",c=e.parentNode,f=x&&e.nodeName.toLowerCase(),p=!n&&!x,d=!1;if(c){if(y){while(l){a=e;while(a=a[l])if(x?a.nodeName.toLowerCase()===f:1===a.nodeType)return!1;u=l="only"===h&&!u&&"nextSibling"}return!0}if(u=[m?c.firstChild:c.lastChild],m&&p){d=(s=(r=(i=(o=(a=c)[S]||(a[S]={}))[a.uniqueID]||(o[a.uniqueID]={}))[h]||[])[0]===k&&r[1])&&r[2],a=s&&c.childNodes[s];while(a=++s&&a&&a[l]||(d=s=0)||u.pop())if(1===a.nodeType&&++d&&a===e){i[h]=[k,s,d];break}}else if(p&&(d=s=(r=(i=(o=(a=e)[S]||(a[S]={}))[a.uniqueID]||(o[a.uniqueID]={}))[h]||[])[0]===k&&r[1]),!1===d)while(a=++s&&a&&a[l]||(d=s=0)||u.pop())if((x?a.nodeName.toLowerCase()===f:1===a.nodeType)&&++d&&(p&&((i=(o=a[S]||(a[S]={}))[a.uniqueID]||(o[a.uniqueID]={}))[h]=[k,d]),a===e))break;return(d-=v)===g||d%g==0&&0<=d/g}}},PSEUDO:function(e,o){var t,a=b.pseudos[e]||b.setFilters[e.toLowerCase()]||se.error("unsupported pseudo: "+e);return a[S]?a(o):1<a.length?(t=[e,e,"",o],b.setFilters.hasOwnProperty(e.toLowerCase())?le(function(e,t){var n,r=a(e,o),i=r.length;while(i--)e[n=P(e,r[i])]=!(t[n]=r[i])}):function(e){return a(e,0,t)}):a}},pseudos:{not:le(function(e){var r=[],i=[],s=f(e.replace($,"$1"));return s[S]?le(function(e,t,n,r){var i,o=s(e,null,r,[]),a=e.length;while(a--)(i=o[a])&&(e[a]=!(t[a]=i))}):function(e,t,n){return r[0]=e,s(r,null,n,i),r[0]=null,!i.pop()}}),has:le(function(t){return function(e){return 0<se(t,e).length}}),contains:le(function(t){return t=t.replace(te,ne),function(e){return-1<(e.textContent||o(e)).indexOf(t)}}),lang:le(function(n){return V.test(n||"")||se.error("unsupported lang: "+n),n=n.replace(te,ne).toLowerCase(),function(e){var t;do{if(t=E?e.lang:e.getAttribute("xml:lang")||e.getAttribute("lang"))return(t=t.toLowerCase())===n||0===t.indexOf(n+"-")}while((e=e.parentNode)&&1===e.nodeType);return!1}}),target:function(e){var t=n.location&&n.location.hash;return t&&t.slice(1)===e.id},root:function(e){return e===a},focus:function(e){return e===C.activeElement&&(!C.hasFocus||C.hasFocus())&&!!(e.type||e.href||~e.tabIndex)},enabled:ge(!1),disabled:ge(!0),checked:function(e){var t=e.nodeName.toLowerCase();return"input"===t&&!!e.checked||"option"===t&&!!e.selected},selected:function(e){return e.parentNode&&e.parentNode.selectedIndex,!0===e.selected},empty:function(e){for(e=e.firstChild;e;e=e.nextSibling)if(e.nodeType<6)return!1;return!0},parent:function(e){return!b.pseudos.empty(e)},header:function(e){return J.test(e.nodeName)},input:function(e){return Q.test(e.nodeName)},button:function(e){var t=e.nodeName.toLowerCase();return"input"===t&&"button"===e.type||"button"===t},text:function(e){var t;return"input"===e.nodeName.toLowerCase()&&"text"===e.type&&(null==(t=e.getAttribute("type"))||"text"===t.toLowerCase())},first:ve(function(){return[0]}),last:ve(function(e,t){return[t-1]}),eq:ve(function(e,t,n){return[n<0?n+t:n]}),even:ve(function(e,t){for(var n=0;n<t;n+=2)e.push(n);return e}),odd:ve(function(e,t){for(var n=1;n<t;n+=2)e.push(n);return e}),lt:ve(function(e,t,n){for(var r=n<0?n+t:t<n?t:n;0<=--r;)e.push(r);return e}),gt:ve(function(e,t,n){for(var r=n<0?n+t:n;++r<t;)e.push(r);return e})}}).pseudos.nth=b.pseudos.eq,{radio:!0,checkbox:!0,file:!0,password:!0,image:!0})b.pseudos[e]=de(e);for(e in{submit:!0,reset:!0})b.pseudos[e]=he(e);function me(){}function xe(e){for(var t=0,n=e.length,r="";t<n;t++)r+=e[t].value;return r}function be(s,e,t){var u=e.dir,l=e.next,c=l||u,f=t&&"parentNode"===c,p=r++;return e.first?function(e,t,n){while(e=e[u])if(1===e.nodeType||f)return s(e,t,n);return!1}:function(e,t,n){var r,i,o,a=[k,p];if(n){while(e=e[u])if((1===e.nodeType||f)&&s(e,t,n))return!0}else while(e=e[u])if(1===e.nodeType||f)if(i=(o=e[S]||(e[S]={}))[e.uniqueID]||(o[e.uniqueID]={}),l&&l===e.nodeName.toLowerCase())e=e[u]||e;else{if((r=i[c])&&r[0]===k&&r[1]===p)return a[2]=r[2];if((i[c]=a)[2]=s(e,t,n))return!0}return!1}}function we(i){return 1<i.length?function(e,t,n){var r=i.length;while(r--)if(!i[r](e,t,n))return!1;return!0}:i[0]}function Te(e,t,n,r,i){for(var o,a=[],s=0,u=e.length,l=null!=t;s<u;s++)(o=e[s])&&(n&&!n(o,r,i)||(a.push(o),l&&t.push(s)));return a}function Ce(d,h,g,v,y,e){return v&&!v[S]&&(v=Ce(v)),y&&!y[S]&&(y=Ce(y,e)),le(function(e,t,n,r){var i,o,a,s=[],u=[],l=t.length,c=e||function(e,t,n){for(var r=0,i=t.length;r<i;r++)se(e,t[r],n);return n}(h||"*",n.nodeType?[n]:n,[]),f=!d||!e&&h?c:Te(c,s,d,n,r),p=g?y||(e?d:l||v)?[]:t:f;if(g&&g(f,p,n,r),v){i=Te(p,u),v(i,[],n,r),o=i.length;while(o--)(a=i[o])&&(p[u[o]]=!(f[u[o]]=a))}if(e){if(y||d){if(y){i=[],o=p.length;while(o--)(a=p[o])&&i.push(f[o]=a);y(null,p=[],i,r)}o=p.length;while(o--)(a=p[o])&&-1<(i=y?P(e,a):s[o])&&(e[i]=!(t[i]=a))}}else p=Te(p===t?p.splice(l,p.length):p),y?y(null,t,p,r):H.apply(t,p)})}function Ee(e){for(var i,t,n,r=e.length,o=b.relative[e[0].type],a=o||b.relative[" "],s=o?1:0,u=be(function(e){return e===i},a,!0),l=be(function(e){return-1<P(i,e)},a,!0),c=[function(e,t,n){var r=!o&&(n||t!==w)||((i=t).nodeType?u(e,t,n):l(e,t,n));return i=null,r}];s<r;s++)if(t=b.relative[e[s].type])c=[be(we(c),t)];else{if((t=b.filter[e[s].type].apply(null,e[s].matches))[S]){for(n=++s;n<r;n++)if(b.relative[e[n].type])break;return Ce(1<s&&we(c),1<s&&xe(e.slice(0,s-1).concat({value:" "===e[s-2].type?"*":""})).replace($,"$1"),t,s<n&&Ee(e.slice(s,n)),n<r&&Ee(e=e.slice(n)),n<r&&xe(e))}c.push(t)}return we(c)}return me.prototype=b.filters=b.pseudos,b.setFilters=new me,h=se.tokenize=function(e,t){var n,r,i,o,a,s,u,l=x[e+" "];if(l)return t?0:l.slice(0);a=e,s=[],u=b.preFilter;while(a){for(o in n&&!(r=_.exec(a))||(r&&(a=a.slice(r[0].length)||a),s.push(i=[])),n=!1,(r=z.exec(a))&&(n=r.shift(),i.push({value:n,type:r[0].replace($," ")}),a=a.slice(n.length)),b.filter)!(r=G[o].exec(a))||u[o]&&!(r=u[o](r))||(n=r.shift(),i.push({value:n,type:o,matches:r}),a=a.slice(n.length));if(!n)break}return t?a.length:a?se.error(e):x(e,s).slice(0)},f=se.compile=function(e,t){var n,v,y,m,x,r,i=[],o=[],a=A[e+" "];if(!a){t||(t=h(e)),n=t.length;while(n--)(a=Ee(t[n]))[S]?i.push(a):o.push(a);(a=A(e,(v=o,m=0<(y=i).length,x=0<v.length,r=function(e,t,n,r,i){var o,a,s,u=0,l="0",c=e&&[],f=[],p=w,d=e||x&&b.find.TAG("*",i),h=k+=null==p?1:Math.random()||.1,g=d.length;for(i&&(w=t==C||t||i);l!==g&&null!=(o=d[l]);l++){if(x&&o){a=0,t||o.ownerDocument==C||(T(o),n=!E);while(s=v[a++])if(s(o,t||C,n)){r.push(o);break}i&&(k=h)}m&&((o=!s&&o)&&u--,e&&c.push(o))}if(u+=l,m&&l!==u){a=0;while(s=y[a++])s(c,f,t,n);if(e){if(0<u)while(l--)c[l]||f[l]||(f[l]=q.call(r));f=Te(f)}H.apply(r,f),i&&!e&&0<f.length&&1<u+y.length&&se.uniqueSort(r)}return i&&(k=h,w=p),c},m?le(r):r))).selector=e}return a},g=se.select=function(e,t,n,r){var i,o,a,s,u,l="function"==typeof e&&e,c=!r&&h(e=l.selector||e);if(n=n||[],1===c.length){if(2<(o=c[0]=c[0].slice(0)).length&&"ID"===(a=o[0]).type&&9===t.nodeType&&E&&b.relative[o[1].type]){if(!(t=(b.find.ID(a.matches[0].replace(te,ne),t)||[])[0]))return n;l&&(t=t.parentNode),e=e.slice(o.shift().value.length)}i=G.needsContext.test(e)?0:o.length;while(i--){if(a=o[i],b.relative[s=a.type])break;if((u=b.find[s])&&(r=u(a.matches[0].replace(te,ne),ee.test(o[0].type)&&ye(t.parentNode)||t))){if(o.splice(i,1),!(e=r.length&&xe(o)))return H.apply(n,r),n;break}}}return(l||f(e,c))(r,t,!E,n,!t||ee.test(e)&&ye(t.parentNode)||t),n},d.sortStable=S.split("").sort(j).join("")===S,d.detectDuplicates=!!l,T(),d.sortDetached=ce(function(e){return 1&e.compareDocumentPosition(C.createElement("fieldset"))}),ce(function(e){return e.innerHTML="<a href='#'></a>","#"===e.firstChild.getAttribute("href")})||fe("type|href|height|width",function(e,t,n){if(!n)return e.getAttribute(t,"type"===t.toLowerCase()?1:2)}),d.attributes&&ce(function(e){return e.innerHTML="<input/>",e.firstChild.setAttribute("value",""),""===e.firstChild.getAttribute("value")})||fe("value",function(e,t,n){if(!n&&"input"===e.nodeName.toLowerCase())return e.defaultValue}),ce(function(e){return null==e.getAttribute("disabled")})||fe(R,function(e,t,n){var r;if(!n)return!0===e[t]?t.toLowerCase():(r=e.getAttributeNode(t))&&r.specified?r.value:null}),se}(C);S.find=d,S.expr=d.selectors,S.expr[":"]=S.expr.pseudos,S.uniqueSort=S.unique=d.uniqueSort,S.text=d.getText,S.isXMLDoc=d.isXML,S.contains=d.contains,S.escapeSelector=d.escape;var h=function(e,t,n){var r=[],i=void 0!==n;while((e=e[t])&&9!==e.nodeType)if(1===e.nodeType){if(i&&S(e).is(n))break;r.push(e)}return r},T=function(e,t){for(var n=[];e;e=e.nextSibling)1===e.nodeType&&e!==t&&n.push(e);return n},k=S.expr.match.needsContext;function A(e,t){return e.nodeName&&e.nodeName.toLowerCase()===t.toLowerCase()}var N=/^<([a-z][^\/\0>:\x20\t\r\n\f]*)[\x20\t\r\n\f]*\/?>(?:<\/\1>|)$/i;function j(e,n,r){return m(n)?S.grep(e,function(e,t){return!!n.call(e,t,e)!==r}):n.nodeType?S.grep(e,function(e){return e===n!==r}):"string"!=typeof n?S.grep(e,function(e){return-1<i.call(n,e)!==r}):S.filter(n,e,r)}S.filter=function(e,t,n){var r=t[0];return n&&(e=":not("+e+")"),1===t.length&&1===r.nodeType?S.find.matchesSelector(r,e)?[r]:[]:S.find.matches(e,S.grep(t,function(e){return 1===e.nodeType}))},S.fn.extend({find:function(e){var t,n,r=this.length,i=this;if("string"!=typeof e)return this.pushStack(S(e).filter(function(){for(t=0;t<r;t++)if(S.contains(i[t],this))return!0}));for(n=this.pushStack([]),t=0;t<r;t++)S.find(e,i[t],n);return 1<r?S.uniqueSort(n):n},filter:function(e){return this.pushStack(j(this,e||[],!1))},not:function(e){return this.pushStack(j(this,e||[],!0))},is:function(e){return!!j(this,"string"==typeof e&&k.test(e)?S(e):e||[],!1).length}});var D,q=/^(?:\s*(<[\w\W]+>)[^>]*|#([\w-]+))$/;(S.fn.init=function(e,t,n){var r,i;if(!e)return this;if(n=n||D,"string"==typeof e){if(!(r="<"===e[0]&&">"===e[e.length-1]&&3<=e.length?[null,e,null]:q.exec(e))||!r[1]&&t)return!t||t.jquery?(t||n).find(e):this.constructor(t).find(e);if(r[1]){if(t=t instanceof S?t[0]:t,S.merge(this,S.parseHTML(r[1],t&&t.nodeType?t.ownerDocument||t:E,!0)),N.test(r[1])&&S.isPlainObject(t))for(r in t)m(this[r])?this[r](t[r]):this.attr(r,t[r]);return this}return(i=E.getElementById(r[2]))&&(this[0]=i,this.length=1),this}return e.nodeType?(this[0]=e,this.length=1,this):m(e)?void 0!==n.ready?n.ready(e):e(S):S.makeArray(e,this)}).prototype=S.fn,D=S(E);var L=/^(?:parents|prev(?:Until|All))/,H={children:!0,contents:!0,next:!0,prev:!0};function O(e,t){while((e=e[t])&&1!==e.nodeType);return e}S.fn.extend({has:function(e){var t=S(e,this),n=t.length;return this.filter(function(){for(var e=0;e<n;e++)if(S.contains(this,t[e]))return!0})},closest:function(e,t){var n,r=0,i=this.length,o=[],a="string"!=typeof e&&S(e);if(!k.test(e))for(;r<i;r++)for(n=this[r];n&&n!==t;n=n.parentNode)if(n.nodeType<11&&(a?-1<a.index(n):1===n.nodeType&&S.find.matchesSelector(n,e))){o.push(n);break}return this.pushStack(1<o.length?S.uniqueSort(o):o)},index:function(e){return e?"string"==typeof e?i.call(S(e),this[0]):i.call(this,e.jquery?e[0]:e):this[0]&&this[0].parentNode?this.first().prevAll().length:-1},add:function(e,t){return this.pushStack(S.uniqueSort(S.merge(this.get(),S(e,t))))},addBack:function(e){return this.add(null==e?this.prevObject:this.prevObject.filter(e))}}),S.each({parent:function(e){var t=e.parentNode;return t&&11!==t.nodeType?t:null},parents:function(e){return h(e,"parentNode")},parentsUntil:function(e,t,n){return h(e,"parentNode",n)},next:function(e){return O(e,"nextSibling")},prev:function(e){return O(e,"previousSibling")},nextAll:function(e){return h(e,"nextSibling")},prevAll:function(e){return h(e,"previousSibling")},nextUntil:function(e,t,n){return h(e,"nextSibling",n)},prevUntil:function(e,t,n){return h(e,"previousSibling",n)},siblings:function(e){return T((e.parentNode||{}).firstChild,e)},children:function(e){return T(e.firstChild)},contents:function(e){return null!=e.contentDocument&&r(e.contentDocument)?e.contentDocument:(A(e,"template")&&(e=e.content||e),S.merge([],e.childNodes))}},function(r,i){S.fn[r]=function(e,t){var n=S.map(this,i,e);return"Until"!==r.slice(-5)&&(t=e),t&&"string"==typeof t&&(n=S.filter(t,n)),1<this.length&&(H[r]||S.uniqueSort(n),L.test(r)&&n.reverse()),this.pushStack(n)}});var P=/[^\x20\t\r\n\f]+/g;function R(e){return e}function M(e){throw e}function I(e,t,n,r){var i;try{e&&m(i=e.promise)?i.call(e).done(t).fail(n):e&&m(i=e.then)?i.call(e,t,n):t.apply(void 0,[e].slice(r))}catch(e){n.apply(void 0,[e])}}S.Callbacks=function(r){var e,n;r="string"==typeof r?(e=r,n={},S.each(e.match(P)||[],function(e,t){n[t]=!0}),n):S.extend({},r);var i,t,o,a,s=[],u=[],l=-1,c=function(){for(a=a||r.once,o=i=!0;u.length;l=-1){t=u.shift();while(++l<s.length)!1===s[l].apply(t[0],t[1])&&r.stopOnFalse&&(l=s.length,t=!1)}r.memory||(t=!1),i=!1,a&&(s=t?[]:"")},f={add:function(){return s&&(t&&!i&&(l=s.length-1,u.push(t)),function n(e){S.each(e,function(e,t){m(t)?r.unique&&f.has(t)||s.push(t):t&&t.length&&"string"!==w(t)&&n(t)})}(arguments),t&&!i&&c()),this},remove:function(){return S.each(arguments,function(e,t){var n;while(-1<(n=S.inArray(t,s,n)))s.splice(n,1),n<=l&&l--}),this},has:function(e){return e?-1<S.inArray(e,s):0<s.length},empty:function(){return s&&(s=[]),this},disable:function(){return a=u=[],s=t="",this},disabled:function(){return!s},lock:function(){return a=u=[],t||i||(s=t=""),this},locked:function(){return!!a},fireWith:function(e,t){return a||(t=[e,(t=t||[]).slice?t.slice():t],u.push(t),i||c()),this},fire:function(){return f.fireWith(this,arguments),this},fired:function(){return!!o}};return f},S.extend({Deferred:function(e){var o=[["notify","progress",S.Callbacks("memory"),S.Callbacks("memory"),2],["resolve","done",S.Callbacks("once memory"),S.Callbacks("once memory"),0,"resolved"],["reject","fail",S.Callbacks("once memory"),S.Callbacks("once memory"),1,"rejected"]],i="pending",a={state:function(){return i},always:function(){return s.done(arguments).fail(arguments),this},"catch":function(e){return a.then(null,e)},pipe:function(){var i=arguments;return S.Deferred(function(r){S.each(o,function(e,t){var n=m(i[t[4]])&&i[t[4]];s[t[1]](function(){var e=n&&n.apply(this,arguments);e&&m(e.promise)?e.promise().progress(r.notify).done(r.resolve).fail(r.reject):r[t[0]+"With"](this,n?[e]:arguments)})}),i=null}).promise()},then:function(t,n,r){var u=0;function l(i,o,a,s){return function(){var n=this,r=arguments,e=function(){var e,t;if(!(i<u)){if((e=a.apply(n,r))===o.promise())throw new TypeError("Thenable self-resolution");t=e&&("object"==typeof e||"function"==typeof e)&&e.then,m(t)?s?t.call(e,l(u,o,R,s),l(u,o,M,s)):(u++,t.call(e,l(u,o,R,s),l(u,o,M,s),l(u,o,R,o.notifyWith))):(a!==R&&(n=void 0,r=[e]),(s||o.resolveWith)(n,r))}},t=s?e:function(){try{e()}catch(e){S.Deferred.exceptionHook&&S.Deferred.exceptionHook(e,t.stackTrace),u<=i+1&&(a!==M&&(n=void 0,r=[e]),o.rejectWith(n,r))}};i?t():(S.Deferred.getStackHook&&(t.stackTrace=S.Deferred.getStackHook()),C.setTimeout(t))}}return S.Deferred(function(e){o[0][3].add(l(0,e,m(r)?r:R,e.notifyWith)),o[1][3].add(l(0,e,m(t)?t:R)),o[2][3].add(l(0,e,m(n)?n:M))}).promise()},promise:function(e){return null!=e?S.extend(e,a):a}},s={};return S.each(o,function(e,t){var n=t[2],r=t[5];a[t[1]]=n.add,r&&n.add(function(){i=r},o[3-e][2].disable,o[3-e][3].disable,o[0][2].lock,o[0][3].lock),n.add(t[3].fire),s[t[0]]=function(){return s[t[0]+"With"](this===s?void 0:this,arguments),this},s[t[0]+"With"]=n.fireWith}),a.promise(s),e&&e.call(s,s),s},when:function(e){var n=arguments.length,t=n,r=Array(t),i=s.call(arguments),o=S.Deferred(),a=function(t){return function(e){r[t]=this,i[t]=1<arguments.length?s.call(arguments):e,--n||o.resolveWith(r,i)}};if(n<=1&&(I(e,o.done(a(t)).resolve,o.reject,!n),"pending"===o.state()||m(i[t]&&i[t].then)))return o.then();while(t--)I(i[t],a(t),o.reject);return o.promise()}});var W=/^(Eval|Internal|Range|Reference|Syntax|Type|URI)Error$/;S.Deferred.exceptionHook=function(e,t){C.console&&C.console.warn&&e&&W.test(e.name)&&C.console.warn("jQuery.Deferred exception: "+e.message,e.stack,t)},S.readyException=function(e){C.setTimeout(function(){throw e})};var F=S.Deferred();function B(){E.removeEventListener("DOMContentLoaded",B),C.removeEventListener("load",B),S.ready()}S.fn.ready=function(e){return F.then(e)["catch"](function(e){S.readyException(e)}),this},S.extend({isReady:!1,readyWait:1,ready:function(e){(!0===e?--S.readyWait:S.isReady)||(S.isReady=!0)!==e&&0<--S.readyWait||F.resolveWith(E,[S])}}),S.ready.then=F.then,"complete"===E.readyState||"loading"!==E.readyState&&!E.documentElement.doScroll?C.setTimeout(S.ready):(E.addEventListener("DOMContentLoaded",B),C.addEventListener("load",B));var $=function(e,t,n,r,i,o,a){var s=0,u=e.length,l=null==n;if("object"===w(n))for(s in i=!0,n)$(e,t,s,n[s],!0,o,a);else if(void 0!==r&&(i=!0,m(r)||(a=!0),l&&(a?(t.call(e,r),t=null):(l=t,t=function(e,t,n){return l.call(S(e),n)})),t))for(;s<u;s++)t(e[s],n,a?r:r.call(e[s],s,t(e[s],n)));return i?e:l?t.call(e):u?t(e[0],n):o},_=/^-ms-/,z=/-([a-z])/g;function U(e,t){return t.toUpperCase()}function X(e){return e.replace(_,"ms-").replace(z,U)}var V=function(e){return 1===e.nodeType||9===e.nodeType||!+e.nodeType};function G(){this.expando=S.expando+G.uid++}G.uid=1,G.prototype={cache:function(e){var t=e[this.expando];return t||(t={},V(e)&&(e.nodeType?e[this.expando]=t:Object.defineProperty(e,this.expando,{value:t,configurable:!0}))),t},set:function(e,t,n){var r,i=this.cache(e);if("string"==typeof t)i[X(t)]=n;else for(r in t)i[X(r)]=t[r];return i},get:function(e,t){return void 0===t?this.cache(e):e[this.expando]&&e[this.expando][X(t)]},access:function(e,t,n){return void 0===t||t&&"string"==typeof t&&void 0===n?this.get(e,t):(this.set(e,t,n),void 0!==n?n:t)},remove:function(e,t){var n,r=e[this.expando];if(void 0!==r){if(void 0!==t){n=(t=Array.isArray(t)?t.map(X):(t=X(t))in r?[t]:t.match(P)||[]).length;while(n--)delete r[t[n]]}(void 0===t||S.isEmptyObject(r))&&(e.nodeType?e[this.expando]=void 0:delete e[this.expando])}},hasData:function(e){var t=e[this.expando];return void 0!==t&&!S.isEmptyObject(t)}};var Y=new G,Q=new G,J=/^(?:\{[\w\W]*\}|\[[\w\W]*\])$/,K=/[A-Z]/g;function Z(e,t,n){var r,i;if(void 0===n&&1===e.nodeType)if(r="data-"+t.replace(K,"-$&").toLowerCase(),"string"==typeof(n=e.getAttribute(r))){try{n="true"===(i=n)||"false"!==i&&("null"===i?null:i===+i+""?+i:J.test(i)?JSON.parse(i):i)}catch(e){}Q.set(e,t,n)}else n=void 0;return n}S.extend({hasData:function(e){return Q.hasData(e)||Y.hasData(e)},data:function(e,t,n){return Q.access(e,t,n)},removeData:function(e,t){Q.remove(e,t)},_data:function(e,t,n){return Y.access(e,t,n)},_removeData:function(e,t){Y.remove(e,t)}}),S.fn.extend({data:function(n,e){var t,r,i,o=this[0],a=o&&o.attributes;if(void 0===n){if(this.length&&(i=Q.get(o),1===o.nodeType&&!Y.get(o,"hasDataAttrs"))){t=a.length;while(t--)a[t]&&0===(r=a[t].name).indexOf("data-")&&(r=X(r.slice(5)),Z(o,r,i[r]));Y.set(o,"hasDataAttrs",!0)}return i}return"object"==typeof n?this.each(function(){Q.set(this,n)}):$(this,function(e){var t;if(o&&void 0===e)return void 0!==(t=Q.get(o,n))?t:void 0!==(t=Z(o,n))?t:void 0;this.each(function(){Q.set(this,n,e)})},null,e,1<arguments.length,null,!0)},removeData:function(e){return this.each(function(){Q.remove(this,e)})}}),S.extend({queue:function(e,t,n){var r;if(e)return t=(t||"fx")+"queue",r=Y.get(e,t),n&&(!r||Array.isArray(n)?r=Y.access(e,t,S.makeArray(n)):r.push(n)),r||[]},dequeue:function(e,t){t=t||"fx";var n=S.queue(e,t),r=n.length,i=n.shift(),o=S._queueHooks(e,t);"inprogress"===i&&(i=n.shift(),r--),i&&("fx"===t&&n.unshift("inprogress"),delete o.stop,i.call(e,function(){S.dequeue(e,t)},o)),!r&&o&&o.empty.fire()},_queueHooks:function(e,t){var n=t+"queueHooks";return Y.get(e,n)||Y.access(e,n,{empty:S.Callbacks("once memory").add(function(){Y.remove(e,[t+"queue",n])})})}}),S.fn.extend({queue:function(t,n){var e=2;return"string"!=typeof t&&(n=t,t="fx",e--),arguments.length<e?S.queue(this[0],t):void 0===n?this:this.each(function(){var e=S.queue(this,t,n);S._queueHooks(this,t),"fx"===t&&"inprogress"!==e[0]&&S.dequeue(this,t)})},dequeue:function(e){return this.each(function(){S.dequeue(this,e)})},clearQueue:function(e){return this.queue(e||"fx",[])},promise:function(e,t){var n,r=1,i=S.Deferred(),o=this,a=this.length,s=function(){--r||i.resolveWith(o,[o])};"string"!=typeof e&&(t=e,e=void 0),e=e||"fx";while(a--)(n=Y.get(o[a],e+"queueHooks"))&&n.empty&&(r++,n.empty.add(s));return s(),i.promise(t)}});var ee=/[+-]?(?:\d*\.|)\d+(?:[eE][+-]?\d+|)/.source,te=new RegExp("^(?:([+-])=|)("+ee+")([a-z%]*)$","i"),ne=["Top","Right","Bottom","Left"],re=E.documentElement,ie=function(e){return S.contains(e.ownerDocument,e)},oe={composed:!0};re.getRootNode&&(ie=function(e){return S.contains(e.ownerDocument,e)||e.getRootNode(oe)===e.ownerDocument});var ae=function(e,t){return"none"===(e=t||e).style.display||""===e.style.display&&ie(e)&&"none"===S.css(e,"display")};function se(e,t,n,r){var i,o,a=20,s=r?function(){return r.cur()}:function(){return S.css(e,t,"")},u=s(),l=n&&n[3]||(S.cssNumber[t]?"":"px"),c=e.nodeType&&(S.cssNumber[t]||"px"!==l&&+u)&&te.exec(S.css(e,t));if(c&&c[3]!==l){u/=2,l=l||c[3],c=+u||1;while(a--)S.style(e,t,c+l),(1-o)*(1-(o=s()/u||.5))<=0&&(a=0),c/=o;c*=2,S.style(e,t,c+l),n=n||[]}return n&&(c=+c||+u||0,i=n[1]?c+(n[1]+1)*n[2]:+n[2],r&&(r.unit=l,r.start=c,r.end=i)),i}var ue={};function le(e,t){for(var n,r,i,o,a,s,u,l=[],c=0,f=e.length;c<f;c++)(r=e[c]).style&&(n=r.style.display,t?("none"===n&&(l[c]=Y.get(r,"display")||null,l[c]||(r.style.display="")),""===r.style.display&&ae(r)&&(l[c]=(u=a=o=void 0,a=(i=r).ownerDocument,s=i.nodeName,(u=ue[s])||(o=a.body.appendChild(a.createElement(s)),u=S.css(o,"display"),o.parentNode.removeChild(o),"none"===u&&(u="block"),ue[s]=u)))):"none"!==n&&(l[c]="none",Y.set(r,"display",n)));for(c=0;c<f;c++)null!=l[c]&&(e[c].style.display=l[c]);return e}S.fn.extend({show:function(){return le(this,!0)},hide:function(){return le(this)},toggle:function(e){return"boolean"==typeof e?e?this.show():this.hide():this.each(function(){ae(this)?S(this).show():S(this).hide()})}});var ce,fe,pe=/^(?:checkbox|radio)$/i,de=/<([a-z][^\/\0>\x20\t\r\n\f]*)/i,he=/^$|^module$|\/(?:java|ecma)script/i;ce=E.createDocumentFragment().appendChild(E.createElement("div")),(fe=E.createElement("input")).setAttribute("type","radio"),fe.setAttribute("checked","checked"),fe.setAttribute("name","t"),ce.appendChild(fe),y.checkClone=ce.cloneNode(!0).cloneNode(!0).lastChild.checked,ce.innerHTML="<textarea>x</textarea>",y.noCloneChecked=!!ce.cloneNode(!0).lastChild.defaultValue,ce.innerHTML="<option></option>",y.option=!!ce.lastChild;var ge={thead:[1,"<table>","</table>"],col:[2,"<table><colgroup>","</colgroup></table>"],tr:[2,"<table><tbody>","</tbody></table>"],td:[3,"<table><tbody><tr>","</tr></tbody></table>"],_default:[0,"",""]};function ve(e,t){var n;return n="undefined"!=typeof e.getElementsByTagName?e.getElementsByTagName(t||"*"):"undefined"!=typeof e.querySelectorAll?e.querySelectorAll(t||"*"):[],void 0===t||t&&A(e,t)?S.merge([e],n):n}function ye(e,t){for(var n=0,r=e.length;n<r;n++)Y.set(e[n],"globalEval",!t||Y.get(t[n],"globalEval"))}ge.tbody=ge.tfoot=ge.colgroup=ge.caption=ge.thead,ge.th=ge.td,y.option||(ge.optgroup=ge.option=[1,"<select multiple='multiple'>","</select>"]);var me=/<|&#?\w+;/;function xe(e,t,n,r,i){for(var o,a,s,u,l,c,f=t.createDocumentFragment(),p=[],d=0,h=e.length;d<h;d++)if((o=e[d])||0===o)if("object"===w(o))S.merge(p,o.nodeType?[o]:o);else if(me.test(o)){a=a||f.appendChild(t.createElement("div")),s=(de.exec(o)||["",""])[1].toLowerCase(),u=ge[s]||ge._default,a.innerHTML=u[1]+S.htmlPrefilter(o)+u[2],c=u[0];while(c--)a=a.lastChild;S.merge(p,a.childNodes),(a=f.firstChild).textContent=""}else p.push(t.createTextNode(o));f.textContent="",d=0;while(o=p[d++])if(r&&-1<S.inArray(o,r))i&&i.push(o);else if(l=ie(o),a=ve(f.appendChild(o),"script"),l&&ye(a),n){c=0;while(o=a[c++])he.test(o.type||"")&&n.push(o)}return f}var be=/^([^.]*)(?:\.(.+)|)/;function we(){return!0}function Te(){return!1}function Ce(e,t){return e===function(){try{return E.activeElement}catch(e){}}()==("focus"===t)}function Ee(e,t,n,r,i,o){var a,s;if("object"==typeof t){for(s in"string"!=typeof n&&(r=r||n,n=void 0),t)Ee(e,s,n,r,t[s],o);return e}if(null==r&&null==i?(i=n,r=n=void 0):null==i&&("string"==typeof n?(i=r,r=void 0):(i=r,r=n,n=void 0)),!1===i)i=Te;else if(!i)return e;return 1===o&&(a=i,(i=function(e){return S().off(e),a.apply(this,arguments)}).guid=a.guid||(a.guid=S.guid++)),e.each(function(){S.event.add(this,t,i,r,n)})}function Se(e,i,o){o?(Y.set(e,i,!1),S.event.add(e,i,{namespace:!1,handler:function(e){var t,n,r=Y.get(this,i);if(1&e.isTrigger&&this[i]){if(r.length)(S.event.special[i]||{}).delegateType&&e.stopPropagation();else if(r=s.call(arguments),Y.set(this,i,r),t=o(this,i),this[i](),r!==(n=Y.get(this,i))||t?Y.set(this,i,!1):n={},r!==n)return e.stopImmediatePropagation(),e.preventDefault(),n&&n.value}else r.length&&(Y.set(this,i,{value:S.event.trigger(S.extend(r[0],S.Event.prototype),r.slice(1),this)}),e.stopImmediatePropagation())}})):void 0===Y.get(e,i)&&S.event.add(e,i,we)}S.event={global:{},add:function(t,e,n,r,i){var o,a,s,u,l,c,f,p,d,h,g,v=Y.get(t);if(V(t)){n.handler&&(n=(o=n).handler,i=o.selector),i&&S.find.matchesSelector(re,i),n.guid||(n.guid=S.guid++),(u=v.events)||(u=v.events=Object.create(null)),(a=v.handle)||(a=v.handle=function(e){return"undefined"!=typeof S&&S.event.triggered!==e.type?S.event.dispatch.apply(t,arguments):void 0}),l=(e=(e||"").match(P)||[""]).length;while(l--)d=g=(s=be.exec(e[l])||[])[1],h=(s[2]||"").split(".").sort(),d&&(f=S.event.special[d]||{},d=(i?f.delegateType:f.bindType)||d,f=S.event.special[d]||{},c=S.extend({type:d,origType:g,data:r,handler:n,guid:n.guid,selector:i,needsContext:i&&S.expr.match.needsContext.test(i),namespace:h.join(".")},o),(p=u[d])||((p=u[d]=[]).delegateCount=0,f.setup&&!1!==f.setup.call(t,r,h,a)||t.addEventListener&&t.addEventListener(d,a)),f.add&&(f.add.call(t,c),c.handler.guid||(c.handler.guid=n.guid)),i?p.splice(p.delegateCount++,0,c):p.push(c),S.event.global[d]=!0)}},remove:function(e,t,n,r,i){var o,a,s,u,l,c,f,p,d,h,g,v=Y.hasData(e)&&Y.get(e);if(v&&(u=v.events)){l=(t=(t||"").match(P)||[""]).length;while(l--)if(d=g=(s=be.exec(t[l])||[])[1],h=(s[2]||"").split(".").sort(),d){f=S.event.special[d]||{},p=u[d=(r?f.delegateType:f.bindType)||d]||[],s=s[2]&&new RegExp("(^|\\.)"+h.join("\\.(?:.*\\.|)")+"(\\.|$)"),a=o=p.length;while(o--)c=p[o],!i&&g!==c.origType||n&&n.guid!==c.guid||s&&!s.test(c.namespace)||r&&r!==c.selector&&("**"!==r||!c.selector)||(p.splice(o,1),c.selector&&p.delegateCount--,f.remove&&f.remove.call(e,c));a&&!p.length&&(f.teardown&&!1!==f.teardown.call(e,h,v.handle)||S.removeEvent(e,d,v.handle),delete u[d])}else for(d in u)S.event.remove(e,d+t[l],n,r,!0);S.isEmptyObject(u)&&Y.remove(e,"handle events")}},dispatch:function(e){var t,n,r,i,o,a,s=new Array(arguments.length),u=S.event.fix(e),l=(Y.get(this,"events")||Object.create(null))[u.type]||[],c=S.event.special[u.type]||{};for(s[0]=u,t=1;t<arguments.length;t++)s[t]=arguments[t];if(u.delegateTarget=this,!c.preDispatch||!1!==c.preDispatch.call(this,u)){a=S.event.handlers.call(this,u,l),t=0;while((i=a[t++])&&!u.isPropagationStopped()){u.currentTarget=i.elem,n=0;while((o=i.handlers[n++])&&!u.isImmediatePropagationStopped())u.rnamespace&&!1!==o.namespace&&!u.rnamespace.test(o.namespace)||(u.handleObj=o,u.data=o.data,void 0!==(r=((S.event.special[o.origType]||{}).handle||o.handler).apply(i.elem,s))&&!1===(u.result=r)&&(u.preventDefault(),u.stopPropagation()))}return c.postDispatch&&c.postDispatch.call(this,u),u.result}},handlers:function(e,t){var n,r,i,o,a,s=[],u=t.delegateCount,l=e.target;if(u&&l.nodeType&&!("click"===e.type&&1<=e.button))for(;l!==this;l=l.parentNode||this)if(1===l.nodeType&&("click"!==e.type||!0!==l.disabled)){for(o=[],a={},n=0;n<u;n++)void 0===a[i=(r=t[n]).selector+" "]&&(a[i]=r.needsContext?-1<S(i,this).index(l):S.find(i,this,null,[l]).length),a[i]&&o.push(r);o.length&&s.push({elem:l,handlers:o})}return l=this,u<t.length&&s.push({elem:l,handlers:t.slice(u)}),s},addProp:function(t,e){Object.defineProperty(S.Event.prototype,t,{enumerable:!0,configurable:!0,get:m(e)?function(){if(this.originalEvent)return e(this.originalEvent)}:function(){if(this.originalEvent)return this.originalEvent[t]},set:function(e){Object.defineProperty(this,t,{enumerable:!0,configurable:!0,writable:!0,value:e})}})},fix:function(e){return e[S.expando]?e:new S.Event(e)},special:{load:{noBubble:!0},click:{setup:function(e){var t=this||e;return pe.test(t.type)&&t.click&&A(t,"input")&&Se(t,"click",we),!1},trigger:function(e){var t=this||e;return pe.test(t.type)&&t.click&&A(t,"input")&&Se(t,"click"),!0},_default:function(e){var t=e.target;return pe.test(t.type)&&t.click&&A(t,"input")&&Y.get(t,"click")||A(t,"a")}},beforeunload:{postDispatch:function(e){void 0!==e.result&&e.originalEvent&&(e.originalEvent.returnValue=e.result)}}}},S.removeEvent=function(e,t,n){e.removeEventListener&&e.removeEventListener(t,n)},S.Event=function(e,t){if(!(this instanceof S.Event))return new S.Event(e,t);e&&e.type?(this.originalEvent=e,this.type=e.type,this.isDefaultPrevented=e.defaultPrevented||void 0===e.defaultPrevented&&!1===e.returnValue?we:Te,this.target=e.target&&3===e.target.nodeType?e.target.parentNode:e.target,this.currentTarget=e.currentTarget,this.relatedTarget=e.relatedTarget):this.type=e,t&&S.extend(this,t),this.timeStamp=e&&e.timeStamp||Date.now(),this[S.expando]=!0},S.Event.prototype={constructor:S.Event,isDefaultPrevented:Te,isPropagationStopped:Te,isImmediatePropagationStopped:Te,isSimulated:!1,preventDefault:function(){var e=this.originalEvent;this.isDefaultPrevented=we,e&&!this.isSimulated&&e.preventDefault()},stopPropagation:function(){var e=this.originalEvent;this.isPropagationStopped=we,e&&!this.isSimulated&&e.stopPropagation()},stopImmediatePropagation:function(){var e=this.originalEvent;this.isImmediatePropagationStopped=we,e&&!this.isSimulated&&e.stopImmediatePropagation(),this.stopPropagation()}},S.each({altKey:!0,bubbles:!0,cancelable:!0,changedTouches:!0,ctrlKey:!0,detail:!0,eventPhase:!0,metaKey:!0,pageX:!0,pageY:!0,shiftKey:!0,view:!0,"char":!0,code:!0,charCode:!0,key:!0,keyCode:!0,button:!0,buttons:!0,clientX:!0,clientY:!0,offsetX:!0,offsetY:!0,pointerId:!0,pointerType:!0,screenX:!0,screenY:!0,targetTouches:!0,toElement:!0,touches:!0,which:!0},S.event.addProp),S.each({focus:"focusin",blur:"focusout"},function(e,t){S.event.special[e]={setup:function(){return Se(this,e,Ce),!1},trigger:function(){return Se(this,e),!0},_default:function(){return!0},delegateType:t}}),S.each({mouseenter:"mouseover",mouseleave:"mouseout",pointerenter:"pointerover",pointerleave:"pointerout"},function(e,i){S.event.special[e]={delegateType:i,bindType:i,handle:function(e){var t,n=e.relatedTarget,r=e.handleObj;return n&&(n===this||S.contains(this,n))||(e.type=r.origType,t=r.handler.apply(this,arguments),e.type=i),t}}}),S.fn.extend({on:function(e,t,n,r){return Ee(this,e,t,n,r)},one:function(e,t,n,r){return Ee(this,e,t,n,r,1)},off:function(e,t,n){var r,i;if(e&&e.preventDefault&&e.handleObj)return r=e.handleObj,S(e.delegateTarget).off(r.namespace?r.origType+"."+r.namespace:r.origType,r.selector,r.handler),this;if("object"==typeof e){for(i in e)this.off(i,t,e[i]);return this}return!1!==t&&"function"!=typeof t||(n=t,t=void 0),!1===n&&(n=Te),this.each(function(){S.event.remove(this,e,n,t)})}});var ke=/<script|<style|<link/i,Ae=/checked\s*(?:[^=]|=\s*.checked.)/i,Ne=/^\s*<!(?:\[CDATA\[|--)|(?:\]\]|--)>\s*$/g;function je(e,t){return A(e,"table")&&A(11!==t.nodeType?t:t.firstChild,"tr")&&S(e).children("tbody")[0]||e}function De(e){return e.type=(null!==e.getAttribute("type"))+"/"+e.type,e}function qe(e){return"true/"===(e.type||"").slice(0,5)?e.type=e.type.slice(5):e.removeAttribute("type"),e}function Le(e,t){var n,r,i,o,a,s;if(1===t.nodeType){if(Y.hasData(e)&&(s=Y.get(e).events))for(i in Y.remove(t,"handle events"),s)for(n=0,r=s[i].length;n<r;n++)S.event.add(t,i,s[i][n]);Q.hasData(e)&&(o=Q.access(e),a=S.extend({},o),Q.set(t,a))}}function He(n,r,i,o){r=g(r);var e,t,a,s,u,l,c=0,f=n.length,p=f-1,d=r[0],h=m(d);if(h||1<f&&"string"==typeof d&&!y.checkClone&&Ae.test(d))return n.each(function(e){var t=n.eq(e);h&&(r[0]=d.call(this,e,t.html())),He(t,r,i,o)});if(f&&(t=(e=xe(r,n[0].ownerDocument,!1,n,o)).firstChild,1===e.childNodes.length&&(e=t),t||o)){for(s=(a=S.map(ve(e,"script"),De)).length;c<f;c++)u=e,c!==p&&(u=S.clone(u,!0,!0),s&&S.merge(a,ve(u,"script"))),i.call(n[c],u,c);if(s)for(l=a[a.length-1].ownerDocument,S.map(a,qe),c=0;c<s;c++)u=a[c],he.test(u.type||"")&&!Y.access(u,"globalEval")&&S.contains(l,u)&&(u.src&&"module"!==(u.type||"").toLowerCase()?S._evalUrl&&!u.noModule&&S._evalUrl(u.src,{nonce:u.nonce||u.getAttribute("nonce")},l):b(u.textContent.replace(Ne,""),u,l))}return n}function Oe(e,t,n){for(var r,i=t?S.filter(t,e):e,o=0;null!=(r=i[o]);o++)n||1!==r.nodeType||S.cleanData(ve(r)),r.parentNode&&(n&&ie(r)&&ye(ve(r,"script")),r.parentNode.removeChild(r));return e}S.extend({htmlPrefilter:function(e){return e},clone:function(e,t,n){var r,i,o,a,s,u,l,c=e.cloneNode(!0),f=ie(e);if(!(y.noCloneChecked||1!==e.nodeType&&11!==e.nodeType||S.isXMLDoc(e)))for(a=ve(c),r=0,i=(o=ve(e)).length;r<i;r++)s=o[r],u=a[r],void 0,"input"===(l=u.nodeName.toLowerCase())&&pe.test(s.type)?u.checked=s.checked:"input"!==l&&"textarea"!==l||(u.defaultValue=s.defaultValue);if(t)if(n)for(o=o||ve(e),a=a||ve(c),r=0,i=o.length;r<i;r++)Le(o[r],a[r]);else Le(e,c);return 0<(a=ve(c,"script")).length&&ye(a,!f&&ve(e,"script")),c},cleanData:function(e){for(var t,n,r,i=S.event.special,o=0;void 0!==(n=e[o]);o++)if(V(n)){if(t=n[Y.expando]){if(t.events)for(r in t.events)i[r]?S.event.remove(n,r):S.removeEvent(n,r,t.handle);n[Y.expando]=void 0}n[Q.expando]&&(n[Q.expando]=void 0)}}}),S.fn.extend({detach:function(e){return Oe(this,e,!0)},remove:function(e){return Oe(this,e)},text:function(e){return $(this,function(e){return void 0===e?S.text(this):this.empty().each(function(){1!==this.nodeType&&11!==this.nodeType&&9!==this.nodeType||(this.textContent=e)})},null,e,arguments.length)},append:function(){return He(this,arguments,function(e){1!==this.nodeType&&11!==this.nodeType&&9!==this.nodeType||je(this,e).appendChild(e)})},prepend:function(){return He(this,arguments,function(e){if(1===this.nodeType||11===this.nodeType||9===this.nodeType){var t=je(this,e);t.insertBefore(e,t.firstChild)}})},before:function(){return He(this,arguments,function(e){this.parentNode&&this.parentNode.insertBefore(e,this)})},after:function(){return He(this,arguments,function(e){this.parentNode&&this.parentNode.insertBefore(e,this.nextSibling)})},empty:function(){for(var e,t=0;null!=(e=this[t]);t++)1===e.nodeType&&(S.cleanData(ve(e,!1)),e.textContent="");return this},clone:function(e,t){return e=null!=e&&e,t=null==t?e:t,this.map(function(){return S.clone(this,e,t)})},html:function(e){return $(this,function(e){var t=this[0]||{},n=0,r=this.length;if(void 0===e&&1===t.nodeType)return t.innerHTML;if("string"==typeof e&&!ke.test(e)&&!ge[(de.exec(e)||["",""])[1].toLowerCase()]){e=S.htmlPrefilter(e);try{for(;n<r;n++)1===(t=this[n]||{}).nodeType&&(S.cleanData(ve(t,!1)),t.innerHTML=e);t=0}catch(e){}}t&&this.empty().append(e)},null,e,arguments.length)},replaceWith:function(){var n=[];return He(this,arguments,function(e){var t=this.parentNode;S.inArray(this,n)<0&&(S.cleanData(ve(this)),t&&t.replaceChild(e,this))},n)}}),S.each({appendTo:"append",prependTo:"prepend",insertBefore:"before",insertAfter:"after",replaceAll:"replaceWith"},function(e,a){S.fn[e]=function(e){for(var t,n=[],r=S(e),i=r.length-1,o=0;o<=i;o++)t=o===i?this:this.clone(!0),S(r[o])[a](t),u.apply(n,t.get());return this.pushStack(n)}});var Pe=new RegExp("^("+ee+")(?!px)[a-z%]+$","i"),Re=function(e){var t=e.ownerDocument.defaultView;return t&&t.opener||(t=C),t.getComputedStyle(e)},Me=function(e,t,n){var r,i,o={};for(i in t)o[i]=e.style[i],e.style[i]=t[i];for(i in r=n.call(e),t)e.style[i]=o[i];return r},Ie=new RegExp(ne.join("|"),"i");function We(e,t,n){var r,i,o,a,s=e.style;return(n=n||Re(e))&&(""!==(a=n.getPropertyValue(t)||n[t])||ie(e)||(a=S.style(e,t)),!y.pixelBoxStyles()&&Pe.test(a)&&Ie.test(t)&&(r=s.width,i=s.minWidth,o=s.maxWidth,s.minWidth=s.maxWidth=s.width=a,a=n.width,s.width=r,s.minWidth=i,s.maxWidth=o)),void 0!==a?a+"":a}function Fe(e,t){return{get:function(){if(!e())return(this.get=t).apply(this,arguments);delete this.get}}}!function(){function e(){if(l){u.style.cssText="position:absolute;left:-11111px;width:60px;margin-top:1px;padding:0;border:0",l.style.cssText="position:relative;display:block;box-sizing:border-box;overflow:scroll;margin:auto;border:1px;padding:1px;width:60%;top:1%",re.appendChild(u).appendChild(l);var e=C.getComputedStyle(l);n="1%"!==e.top,s=12===t(e.marginLeft),l.style.right="60%",o=36===t(e.right),r=36===t(e.width),l.style.position="absolute",i=12===t(l.offsetWidth/3),re.removeChild(u),l=null}}function t(e){return Math.round(parseFloat(e))}var n,r,i,o,a,s,u=E.createElement("div"),l=E.createElement("div");l.style&&(l.style.backgroundClip="content-box",l.cloneNode(!0).style.backgroundClip="",y.clearCloneStyle="content-box"===l.style.backgroundClip,S.extend(y,{boxSizingReliable:function(){return e(),r},pixelBoxStyles:function(){return e(),o},pixelPosition:function(){return e(),n},reliableMarginLeft:function(){return e(),s},scrollboxSize:function(){return e(),i},reliableTrDimensions:function(){var e,t,n,r;return null==a&&(e=E.createElement("table"),t=E.createElement("tr"),n=E.createElement("div"),e.style.cssText="position:absolute;left:-11111px;border-collapse:separate",t.style.cssText="border:1px solid",t.style.height="1px",n.style.height="9px",n.style.display="block",re.appendChild(e).appendChild(t).appendChild(n),r=C.getComputedStyle(t),a=parseInt(r.height,10)+parseInt(r.borderTopWidth,10)+parseInt(r.borderBottomWidth,10)===t.offsetHeight,re.removeChild(e)),a}}))}();var Be=["Webkit","Moz","ms"],$e=E.createElement("div").style,_e={};function ze(e){var t=S.cssProps[e]||_e[e];return t||(e in $e?e:_e[e]=function(e){var t=e[0].toUpperCase()+e.slice(1),n=Be.length;while(n--)if((e=Be[n]+t)in $e)return e}(e)||e)}var Ue=/^(none|table(?!-c[ea]).+)/,Xe=/^--/,Ve={position:"absolute",visibility:"hidden",display:"block"},Ge={letterSpacing:"0",fontWeight:"400"};function Ye(e,t,n){var r=te.exec(t);return r?Math.max(0,r[2]-(n||0))+(r[3]||"px"):t}function Qe(e,t,n,r,i,o){var a="width"===t?1:0,s=0,u=0;if(n===(r?"border":"content"))return 0;for(;a<4;a+=2)"margin"===n&&(u+=S.css(e,n+ne[a],!0,i)),r?("content"===n&&(u-=S.css(e,"padding"+ne[a],!0,i)),"margin"!==n&&(u-=S.css(e,"border"+ne[a]+"Width",!0,i))):(u+=S.css(e,"padding"+ne[a],!0,i),"padding"!==n?u+=S.css(e,"border"+ne[a]+"Width",!0,i):s+=S.css(e,"border"+ne[a]+"Width",!0,i));return!r&&0<=o&&(u+=Math.max(0,Math.ceil(e["offset"+t[0].toUpperCase()+t.slice(1)]-o-u-s-.5))||0),u}function Je(e,t,n){var r=Re(e),i=(!y.boxSizingReliable()||n)&&"border-box"===S.css(e,"boxSizing",!1,r),o=i,a=We(e,t,r),s="offset"+t[0].toUpperCase()+t.slice(1);if(Pe.test(a)){if(!n)return a;a="auto"}return(!y.boxSizingReliable()&&i||!y.reliableTrDimensions()&&A(e,"tr")||"auto"===a||!parseFloat(a)&&"inline"===S.css(e,"display",!1,r))&&e.getClientRects().length&&(i="border-box"===S.css(e,"boxSizing",!1,r),(o=s in e)&&(a=e[s])),(a=parseFloat(a)||0)+Qe(e,t,n||(i?"border":"content"),o,r,a)+"px"}function Ke(e,t,n,r,i){return new Ke.prototype.init(e,t,n,r,i)}S.extend({cssHooks:{opacity:{get:function(e,t){if(t){var n=We(e,"opacity");return""===n?"1":n}}}},cssNumber:{animationIterationCount:!0,columnCount:!0,fillOpacity:!0,flexGrow:!0,flexShrink:!0,fontWeight:!0,gridArea:!0,gridColumn:!0,gridColumnEnd:!0,gridColumnStart:!0,gridRow:!0,gridRowEnd:!0,gridRowStart:!0,lineHeight:!0,opacity:!0,order:!0,orphans:!0,widows:!0,zIndex:!0,zoom:!0},cssProps:{},style:function(e,t,n,r){if(e&&3!==e.nodeType&&8!==e.nodeType&&e.style){var i,o,a,s=X(t),u=Xe.test(t),l=e.style;if(u||(t=ze(s)),a=S.cssHooks[t]||S.cssHooks[s],void 0===n)return a&&"get"in a&&void 0!==(i=a.get(e,!1,r))?i:l[t];"string"===(o=typeof n)&&(i=te.exec(n))&&i[1]&&(n=se(e,t,i),o="number"),null!=n&&n==n&&("number"!==o||u||(n+=i&&i[3]||(S.cssNumber[s]?"":"px")),y.clearCloneStyle||""!==n||0!==t.indexOf("background")||(l[t]="inherit"),a&&"set"in a&&void 0===(n=a.set(e,n,r))||(u?l.setProperty(t,n):l[t]=n))}},css:function(e,t,n,r){var i,o,a,s=X(t);return Xe.test(t)||(t=ze(s)),(a=S.cssHooks[t]||S.cssHooks[s])&&"get"in a&&(i=a.get(e,!0,n)),void 0===i&&(i=We(e,t,r)),"normal"===i&&t in Ge&&(i=Ge[t]),""===n||n?(o=parseFloat(i),!0===n||isFinite(o)?o||0:i):i}}),S.each(["height","width"],function(e,u){S.cssHooks[u]={get:function(e,t,n){if(t)return!Ue.test(S.css(e,"display"))||e.getClientRects().length&&e.getBoundingClientRect().width?Je(e,u,n):Me(e,Ve,function(){return Je(e,u,n)})},set:function(e,t,n){var r,i=Re(e),o=!y.scrollboxSize()&&"absolute"===i.position,a=(o||n)&&"border-box"===S.css(e,"boxSizing",!1,i),s=n?Qe(e,u,n,a,i):0;return a&&o&&(s-=Math.ceil(e["offset"+u[0].toUpperCase()+u.slice(1)]-parseFloat(i[u])-Qe(e,u,"border",!1,i)-.5)),s&&(r=te.exec(t))&&"px"!==(r[3]||"px")&&(e.style[u]=t,t=S.css(e,u)),Ye(0,t,s)}}}),S.cssHooks.marginLeft=Fe(y.reliableMarginLeft,function(e,t){if(t)return(parseFloat(We(e,"marginLeft"))||e.getBoundingClientRect().left-Me(e,{marginLeft:0},function(){return e.getBoundingClientRect().left}))+"px"}),S.each({margin:"",padding:"",border:"Width"},function(i,o){S.cssHooks[i+o]={expand:function(e){for(var t=0,n={},r="string"==typeof e?e.split(" "):[e];t<4;t++)n[i+ne[t]+o]=r[t]||r[t-2]||r[0];return n}},"margin"!==i&&(S.cssHooks[i+o].set=Ye)}),S.fn.extend({css:function(e,t){return $(this,function(e,t,n){var r,i,o={},a=0;if(Array.isArray(t)){for(r=Re(e),i=t.length;a<i;a++)o[t[a]]=S.css(e,t[a],!1,r);return o}return void 0!==n?S.style(e,t,n):S.css(e,t)},e,t,1<arguments.length)}}),((S.Tween=Ke).prototype={constructor:Ke,init:function(e,t,n,r,i,o){this.elem=e,this.prop=n,this.easing=i||S.easing._default,this.options=t,this.start=this.now=this.cur(),this.end=r,this.unit=o||(S.cssNumber[n]?"":"px")},cur:function(){var e=Ke.propHooks[this.prop];return e&&e.get?e.get(this):Ke.propHooks._default.get(this)},run:function(e){var t,n=Ke.propHooks[this.prop];return this.options.duration?this.pos=t=S.easing[this.easing](e,this.options.duration*e,0,1,this.options.duration):this.pos=t=e,this.now=(this.end-this.start)*t+this.start,this.options.step&&this.options.step.call(this.elem,this.now,this),n&&n.set?n.set(this):Ke.propHooks._default.set(this),this}}).init.prototype=Ke.prototype,(Ke.propHooks={_default:{get:function(e){var t;return 1!==e.elem.nodeType||null!=e.elem[e.prop]&&null==e.elem.style[e.prop]?e.elem[e.prop]:(t=S.css(e.elem,e.prop,""))&&"auto"!==t?t:0},set:function(e){S.fx.step[e.prop]?S.fx.step[e.prop](e):1!==e.elem.nodeType||!S.cssHooks[e.prop]&&null==e.elem.style[ze(e.prop)]?e.elem[e.prop]=e.now:S.style(e.elem,e.prop,e.now+e.unit)}}}).scrollTop=Ke.propHooks.scrollLeft={set:function(e){e.elem.nodeType&&e.elem.parentNode&&(e.elem[e.prop]=e.now)}},S.easing={linear:function(e){return e},swing:function(e){return.5-Math.cos(e*Math.PI)/2},_default:"swing"},S.fx=Ke.prototype.init,S.fx.step={};var Ze,et,tt,nt,rt=/^(?:toggle|show|hide)$/,it=/queueHooks$/;function ot(){et&&(!1===E.hidden&&C.requestAnimationFrame?C.requestAnimationFrame(ot):C.setTimeout(ot,S.fx.interval),S.fx.tick())}function at(){return C.setTimeout(function(){Ze=void 0}),Ze=Date.now()}function st(e,t){var n,r=0,i={height:e};for(t=t?1:0;r<4;r+=2-t)i["margin"+(n=ne[r])]=i["padding"+n]=e;return t&&(i.opacity=i.width=e),i}function ut(e,t,n){for(var r,i=(lt.tweeners[t]||[]).concat(lt.tweeners["*"]),o=0,a=i.length;o<a;o++)if(r=i[o].call(n,t,e))return r}function lt(o,e,t){var n,a,r=0,i=lt.prefilters.length,s=S.Deferred().always(function(){delete u.elem}),u=function(){if(a)return!1;for(var e=Ze||at(),t=Math.max(0,l.startTime+l.duration-e),n=1-(t/l.duration||0),r=0,i=l.tweens.length;r<i;r++)l.tweens[r].run(n);return s.notifyWith(o,[l,n,t]),n<1&&i?t:(i||s.notifyWith(o,[l,1,0]),s.resolveWith(o,[l]),!1)},l=s.promise({elem:o,props:S.extend({},e),opts:S.extend(!0,{specialEasing:{},easing:S.easing._default},t),originalProperties:e,originalOptions:t,startTime:Ze||at(),duration:t.duration,tweens:[],createTween:function(e,t){var n=S.Tween(o,l.opts,e,t,l.opts.specialEasing[e]||l.opts.easing);return l.tweens.push(n),n},stop:function(e){var t=0,n=e?l.tweens.length:0;if(a)return this;for(a=!0;t<n;t++)l.tweens[t].run(1);return e?(s.notifyWith(o,[l,1,0]),s.resolveWith(o,[l,e])):s.rejectWith(o,[l,e]),this}}),c=l.props;for(!function(e,t){var n,r,i,o,a;for(n in e)if(i=t[r=X(n)],o=e[n],Array.isArray(o)&&(i=o[1],o=e[n]=o[0]),n!==r&&(e[r]=o,delete e[n]),(a=S.cssHooks[r])&&"expand"in a)for(n in o=a.expand(o),delete e[r],o)n in e||(e[n]=o[n],t[n]=i);else t[r]=i}(c,l.opts.specialEasing);r<i;r++)if(n=lt.prefilters[r].call(l,o,c,l.opts))return m(n.stop)&&(S._queueHooks(l.elem,l.opts.queue).stop=n.stop.bind(n)),n;return S.map(c,ut,l),m(l.opts.start)&&l.opts.start.call(o,l),l.progress(l.opts.progress).done(l.opts.done,l.opts.complete).fail(l.opts.fail).always(l.opts.always),S.fx.timer(S.extend(u,{elem:o,anim:l,queue:l.opts.queue})),l}S.Animation=S.extend(lt,{tweeners:{"*":[function(e,t){var n=this.createTween(e,t);return se(n.elem,e,te.exec(t),n),n}]},tweener:function(e,t){m(e)?(t=e,e=["*"]):e=e.match(P);for(var n,r=0,i=e.length;r<i;r++)n=e[r],lt.tweeners[n]=lt.tweeners[n]||[],lt.tweeners[n].unshift(t)},prefilters:[function(e,t,n){var r,i,o,a,s,u,l,c,f="width"in t||"height"in t,p=this,d={},h=e.style,g=e.nodeType&&ae(e),v=Y.get(e,"fxshow");for(r in n.queue||(null==(a=S._queueHooks(e,"fx")).unqueued&&(a.unqueued=0,s=a.empty.fire,a.empty.fire=function(){a.unqueued||s()}),a.unqueued++,p.always(function(){p.always(function(){a.unqueued--,S.queue(e,"fx").length||a.empty.fire()})})),t)if(i=t[r],rt.test(i)){if(delete t[r],o=o||"toggle"===i,i===(g?"hide":"show")){if("show"!==i||!v||void 0===v[r])continue;g=!0}d[r]=v&&v[r]||S.style(e,r)}if((u=!S.isEmptyObject(t))||!S.isEmptyObject(d))for(r in f&&1===e.nodeType&&(n.overflow=[h.overflow,h.overflowX,h.overflowY],null==(l=v&&v.display)&&(l=Y.get(e,"display")),"none"===(c=S.css(e,"display"))&&(l?c=l:(le([e],!0),l=e.style.display||l,c=S.css(e,"display"),le([e]))),("inline"===c||"inline-block"===c&&null!=l)&&"none"===S.css(e,"float")&&(u||(p.done(function(){h.display=l}),null==l&&(c=h.display,l="none"===c?"":c)),h.display="inline-block")),n.overflow&&(h.overflow="hidden",p.always(function(){h.overflow=n.overflow[0],h.overflowX=n.overflow[1],h.overflowY=n.overflow[2]})),u=!1,d)u||(v?"hidden"in v&&(g=v.hidden):v=Y.access(e,"fxshow",{display:l}),o&&(v.hidden=!g),g&&le([e],!0),p.done(function(){for(r in g||le([e]),Y.remove(e,"fxshow"),d)S.style(e,r,d[r])})),u=ut(g?v[r]:0,r,p),r in v||(v[r]=u.start,g&&(u.end=u.start,u.start=0))}],prefilter:function(e,t){t?lt.prefilters.unshift(e):lt.prefilters.push(e)}}),S.speed=function(e,t,n){var r=e&&"object"==typeof e?S.extend({},e):{complete:n||!n&&t||m(e)&&e,duration:e,easing:n&&t||t&&!m(t)&&t};return S.fx.off?r.duration=0:"number"!=typeof r.duration&&(r.duration in S.fx.speeds?r.duration=S.fx.speeds[r.duration]:r.duration=S.fx.speeds._default),null!=r.queue&&!0!==r.queue||(r.queue="fx"),r.old=r.complete,r.complete=function(){m(r.old)&&r.old.call(this),r.queue&&S.dequeue(this,r.queue)},r},S.fn.extend({fadeTo:function(e,t,n,r){return this.filter(ae).css("opacity",0).show().end().animate({opacity:t},e,n,r)},animate:function(t,e,n,r){var i=S.isEmptyObject(t),o=S.speed(e,n,r),a=function(){var e=lt(this,S.extend({},t),o);(i||Y.get(this,"finish"))&&e.stop(!0)};return a.finish=a,i||!1===o.queue?this.each(a):this.queue(o.queue,a)},stop:function(i,e,o){var a=function(e){var t=e.stop;delete e.stop,t(o)};return"string"!=typeof i&&(o=e,e=i,i=void 0),e&&this.queue(i||"fx",[]),this.each(function(){var e=!0,t=null!=i&&i+"queueHooks",n=S.timers,r=Y.get(this);if(t)r[t]&&r[t].stop&&a(r[t]);else for(t in r)r[t]&&r[t].stop&&it.test(t)&&a(r[t]);for(t=n.length;t--;)n[t].elem!==this||null!=i&&n[t].queue!==i||(n[t].anim.stop(o),e=!1,n.splice(t,1));!e&&o||S.dequeue(this,i)})},finish:function(a){return!1!==a&&(a=a||"fx"),this.each(function(){var e,t=Y.get(this),n=t[a+"queue"],r=t[a+"queueHooks"],i=S.timers,o=n?n.length:0;for(t.finish=!0,S.queue(this,a,[]),r&&r.stop&&r.stop.call(this,!0),e=i.length;e--;)i[e].elem===this&&i[e].queue===a&&(i[e].anim.stop(!0),i.splice(e,1));for(e=0;e<o;e++)n[e]&&n[e].finish&&n[e].finish.call(this);delete t.finish})}}),S.each(["toggle","show","hide"],function(e,r){var i=S.fn[r];S.fn[r]=function(e,t,n){return null==e||"boolean"==typeof e?i.apply(this,arguments):this.animate(st(r,!0),e,t,n)}}),S.each({slideDown:st("show"),slideUp:st("hide"),slideToggle:st("toggle"),fadeIn:{opacity:"show"},fadeOut:{opacity:"hide"},fadeToggle:{opacity:"toggle"}},function(e,r){S.fn[e]=function(e,t,n){return this.animate(r,e,t,n)}}),S.timers=[],S.fx.tick=function(){var e,t=0,n=S.timers;for(Ze=Date.now();t<n.length;t++)(e=n[t])()||n[t]!==e||n.splice(t--,1);n.length||S.fx.stop(),Ze=void 0},S.fx.timer=function(e){S.timers.push(e),S.fx.start()},S.fx.interval=13,S.fx.start=function(){et||(et=!0,ot())},S.fx.stop=function(){et=null},S.fx.speeds={slow:600,fast:200,_default:400},S.fn.delay=function(r,e){return r=S.fx&&S.fx.speeds[r]||r,e=e||"fx",this.queue(e,function(e,t){var n=C.setTimeout(e,r);t.stop=function(){C.clearTimeout(n)}})},tt=E.createElement("input"),nt=E.createElement("select").appendChild(E.createElement("option")),tt.type="checkbox",y.checkOn=""!==tt.value,y.optSelected=nt.selected,(tt=E.createElement("input")).value="t",tt.type="radio",y.radioValue="t"===tt.value;var ct,ft=S.expr.attrHandle;S.fn.extend({attr:function(e,t){return $(this,S.attr,e,t,1<arguments.length)},removeAttr:function(e){return this.each(function(){S.removeAttr(this,e)})}}),S.extend({attr:function(e,t,n){var r,i,o=e.nodeType;if(3!==o&&8!==o&&2!==o)return"undefined"==typeof e.getAttribute?S.prop(e,t,n):(1===o&&S.isXMLDoc(e)||(i=S.attrHooks[t.toLowerCase()]||(S.expr.match.bool.test(t)?ct:void 0)),void 0!==n?null===n?void S.removeAttr(e,t):i&&"set"in i&&void 0!==(r=i.set(e,n,t))?r:(e.setAttribute(t,n+""),n):i&&"get"in i&&null!==(r=i.get(e,t))?r:null==(r=S.find.attr(e,t))?void 0:r)},attrHooks:{type:{set:function(e,t){if(!y.radioValue&&"radio"===t&&A(e,"input")){var n=e.value;return e.setAttribute("type",t),n&&(e.value=n),t}}}},removeAttr:function(e,t){var n,r=0,i=t&&t.match(P);if(i&&1===e.nodeType)while(n=i[r++])e.removeAttribute(n)}}),ct={set:function(e,t,n){return!1===t?S.removeAttr(e,n):e.setAttribute(n,n),n}},S.each(S.expr.match.bool.source.match(/\w+/g),function(e,t){var a=ft[t]||S.find.attr;ft[t]=function(e,t,n){var r,i,o=t.toLowerCase();return n||(i=ft[o],ft[o]=r,r=null!=a(e,t,n)?o:null,ft[o]=i),r}});var pt=/^(?:input|select|textarea|button)$/i,dt=/^(?:a|area)$/i;function ht(e){return(e.match(P)||[]).join(" ")}function gt(e){return e.getAttribute&&e.getAttribute("class")||""}function vt(e){return Array.isArray(e)?e:"string"==typeof e&&e.match(P)||[]}S.fn.extend({prop:function(e,t){return $(this,S.prop,e,t,1<arguments.length)},removeProp:function(e){return this.each(function(){delete this[S.propFix[e]||e]})}}),S.extend({prop:function(e,t,n){var r,i,o=e.nodeType;if(3!==o&&8!==o&&2!==o)return 1===o&&S.isXMLDoc(e)||(t=S.propFix[t]||t,i=S.propHooks[t]),void 0!==n?i&&"set"in i&&void 0!==(r=i.set(e,n,t))?r:e[t]=n:i&&"get"in i&&null!==(r=i.get(e,t))?r:e[t]},propHooks:{tabIndex:{get:function(e){var t=S.find.attr(e,"tabindex");return t?parseInt(t,10):pt.test(e.nodeName)||dt.test(e.nodeName)&&e.href?0:-1}}},propFix:{"for":"htmlFor","class":"className"}}),y.optSelected||(S.propHooks.selected={get:function(e){var t=e.parentNode;return t&&t.parentNode&&t.parentNode.selectedIndex,null},set:function(e){var t=e.parentNode;t&&(t.selectedIndex,t.parentNode&&t.parentNode.selectedIndex)}}),S.each(["tabIndex","readOnly","maxLength","cellSpacing","cellPadding","rowSpan","colSpan","useMap","frameBorder","contentEditable"],function(){S.propFix[this.toLowerCase()]=this}),S.fn.extend({addClass:function(t){var e,n,r,i,o,a,s,u=0;if(m(t))return this.each(function(e){S(this).addClass(t.call(this,e,gt(this)))});if((e=vt(t)).length)while(n=this[u++])if(i=gt(n),r=1===n.nodeType&&" "+ht(i)+" "){a=0;while(o=e[a++])r.indexOf(" "+o+" ")<0&&(r+=o+" ");i!==(s=ht(r))&&n.setAttribute("class",s)}return this},removeClass:function(t){var e,n,r,i,o,a,s,u=0;if(m(t))return this.each(function(e){S(this).removeClass(t.call(this,e,gt(this)))});if(!arguments.length)return this.attr("class","");if((e=vt(t)).length)while(n=this[u++])if(i=gt(n),r=1===n.nodeType&&" "+ht(i)+" "){a=0;while(o=e[a++])while(-1<r.indexOf(" "+o+" "))r=r.replace(" "+o+" "," ");i!==(s=ht(r))&&n.setAttribute("class",s)}return this},toggleClass:function(i,t){var o=typeof i,a="string"===o||Array.isArray(i);return"boolean"==typeof t&&a?t?this.addClass(i):this.removeClass(i):m(i)?this.each(function(e){S(this).toggleClass(i.call(this,e,gt(this),t),t)}):this.each(function(){var e,t,n,r;if(a){t=0,n=S(this),r=vt(i);while(e=r[t++])n.hasClass(e)?n.removeClass(e):n.addClass(e)}else void 0!==i&&"boolean"!==o||((e=gt(this))&&Y.set(this,"__className__",e),this.setAttribute&&this.setAttribute("class",e||!1===i?"":Y.get(this,"__className__")||""))})},hasClass:function(e){var t,n,r=0;t=" "+e+" ";while(n=this[r++])if(1===n.nodeType&&-1<(" "+ht(gt(n))+" ").indexOf(t))return!0;return!1}});var yt=/\r/g;S.fn.extend({val:function(n){var r,e,i,t=this[0];return arguments.length?(i=m(n),this.each(function(e){var t;1===this.nodeType&&(null==(t=i?n.call(this,e,S(this).val()):n)?t="":"number"==typeof t?t+="":Array.isArray(t)&&(t=S.map(t,function(e){return null==e?"":e+""})),(r=S.valHooks[this.type]||S.valHooks[this.nodeName.toLowerCase()])&&"set"in r&&void 0!==r.set(this,t,"value")||(this.value=t))})):t?(r=S.valHooks[t.type]||S.valHooks[t.nodeName.toLowerCase()])&&"get"in r&&void 0!==(e=r.get(t,"value"))?e:"string"==typeof(e=t.value)?e.replace(yt,""):null==e?"":e:void 0}}),S.extend({valHooks:{option:{get:function(e){var t=S.find.attr(e,"value");return null!=t?t:ht(S.text(e))}},select:{get:function(e){var t,n,r,i=e.options,o=e.selectedIndex,a="select-one"===e.type,s=a?null:[],u=a?o+1:i.length;for(r=o<0?u:a?o:0;r<u;r++)if(((n=i[r]).selected||r===o)&&!n.disabled&&(!n.parentNode.disabled||!A(n.parentNode,"optgroup"))){if(t=S(n).val(),a)return t;s.push(t)}return s},set:function(e,t){var n,r,i=e.options,o=S.makeArray(t),a=i.length;while(a--)((r=i[a]).selected=-1<S.inArray(S.valHooks.option.get(r),o))&&(n=!0);return n||(e.selectedIndex=-1),o}}}}),S.each(["radio","checkbox"],function(){S.valHooks[this]={set:function(e,t){if(Array.isArray(t))return e.checked=-1<S.inArray(S(e).val(),t)}},y.checkOn||(S.valHooks[this].get=function(e){return null===e.getAttribute("value")?"on":e.value})}),y.focusin="onfocusin"in C;var mt=/^(?:focusinfocus|focusoutblur)$/,xt=function(e){e.stopPropagation()};S.extend(S.event,{trigger:function(e,t,n,r){var i,o,a,s,u,l,c,f,p=[n||E],d=v.call(e,"type")?e.type:e,h=v.call(e,"namespace")?e.namespace.split("."):[];if(o=f=a=n=n||E,3!==n.nodeType&&8!==n.nodeType&&!mt.test(d+S.event.triggered)&&(-1<d.indexOf(".")&&(d=(h=d.split(".")).shift(),h.sort()),u=d.indexOf(":")<0&&"on"+d,(e=e[S.expando]?e:new S.Event(d,"object"==typeof e&&e)).isTrigger=r?2:3,e.namespace=h.join("."),e.rnamespace=e.namespace?new RegExp("(^|\\.)"+h.join("\\.(?:.*\\.|)")+"(\\.|$)"):null,e.result=void 0,e.target||(e.target=n),t=null==t?[e]:S.makeArray(t,[e]),c=S.event.special[d]||{},r||!c.trigger||!1!==c.trigger.apply(n,t))){if(!r&&!c.noBubble&&!x(n)){for(s=c.delegateType||d,mt.test(s+d)||(o=o.parentNode);o;o=o.parentNode)p.push(o),a=o;a===(n.ownerDocument||E)&&p.push(a.defaultView||a.parentWindow||C)}i=0;while((o=p[i++])&&!e.isPropagationStopped())f=o,e.type=1<i?s:c.bindType||d,(l=(Y.get(o,"events")||Object.create(null))[e.type]&&Y.get(o,"handle"))&&l.apply(o,t),(l=u&&o[u])&&l.apply&&V(o)&&(e.result=l.apply(o,t),!1===e.result&&e.preventDefault());return e.type=d,r||e.isDefaultPrevented()||c._default&&!1!==c._default.apply(p.pop(),t)||!V(n)||u&&m(n[d])&&!x(n)&&((a=n[u])&&(n[u]=null),S.event.triggered=d,e.isPropagationStopped()&&f.addEventListener(d,xt),n[d](),e.isPropagationStopped()&&f.removeEventListener(d,xt),S.event.triggered=void 0,a&&(n[u]=a)),e.result}},simulate:function(e,t,n){var r=S.extend(new S.Event,n,{type:e,isSimulated:!0});S.event.trigger(r,null,t)}}),S.fn.extend({trigger:function(e,t){return this.each(function(){S.event.trigger(e,t,this)})},triggerHandler:function(e,t){var n=this[0];if(n)return S.event.trigger(e,t,n,!0)}}),y.focusin||S.each({focus:"focusin",blur:"focusout"},function(n,r){var i=function(e){S.event.simulate(r,e.target,S.event.fix(e))};S.event.special[r]={setup:function(){var e=this.ownerDocument||this.document||this,t=Y.access(e,r);t||e.addEventListener(n,i,!0),Y.access(e,r,(t||0)+1)},teardown:function(){var e=this.ownerDocument||this.document||this,t=Y.access(e,r)-1;t?Y.access(e,r,t):(e.removeEventListener(n,i,!0),Y.remove(e,r))}}});var bt=C.location,wt={guid:Date.now()},Tt=/\?/;S.parseXML=function(e){var t,n;if(!e||"string"!=typeof e)return null;try{t=(new C.DOMParser).parseFromString(e,"text/xml")}catch(e){}return n=t&&t.getElementsByTagName("parsererror")[0],t&&!n||S.error("Invalid XML: "+(n?S.map(n.childNodes,function(e){return e.textContent}).join("\n"):e)),t};var Ct=/\[\]$/,Et=/\r?\n/g,St=/^(?:submit|button|image|reset|file)$/i,kt=/^(?:input|select|textarea|keygen)/i;function At(n,e,r,i){var t;if(Array.isArray(e))S.each(e,function(e,t){r||Ct.test(n)?i(n,t):At(n+"["+("object"==typeof t&&null!=t?e:"")+"]",t,r,i)});else if(r||"object"!==w(e))i(n,e);else for(t in e)At(n+"["+t+"]",e[t],r,i)}S.param=function(e,t){var n,r=[],i=function(e,t){var n=m(t)?t():t;r[r.length]=encodeURIComponent(e)+"="+encodeURIComponent(null==n?"":n)};if(null==e)return"";if(Array.isArray(e)||e.jquery&&!S.isPlainObject(e))S.each(e,function(){i(this.name,this.value)});else for(n in e)At(n,e[n],t,i);return r.join("&")},S.fn.extend({serialize:function(){return S.param(this.serializeArray())},serializeArray:function(){return this.map(function(){var e=S.prop(this,"elements");return e?S.makeArray(e):this}).filter(function(){var e=this.type;return this.name&&!S(this).is(":disabled")&&kt.test(this.nodeName)&&!St.test(e)&&(this.checked||!pe.test(e))}).map(function(e,t){var n=S(this).val();return null==n?null:Array.isArray(n)?S.map(n,function(e){return{name:t.name,value:e.replace(Et,"\r\n")}}):{name:t.name,value:n.replace(Et,"\r\n")}}).get()}});var Nt=/%20/g,jt=/#.*$/,Dt=/([?&])_=[^&]*/,qt=/^(.*?):[ \t]*([^\r\n]*)$/gm,Lt=/^(?:GET|HEAD)$/,Ht=/^\/\//,Ot={},Pt={},Rt="*/".concat("*"),Mt=E.createElement("a");function It(o){return function(e,t){"string"!=typeof e&&(t=e,e="*");var n,r=0,i=e.toLowerCase().match(P)||[];if(m(t))while(n=i[r++])"+"===n[0]?(n=n.slice(1)||"*",(o[n]=o[n]||[]).unshift(t)):(o[n]=o[n]||[]).push(t)}}function Wt(t,i,o,a){var s={},u=t===Pt;function l(e){var r;return s[e]=!0,S.each(t[e]||[],function(e,t){var n=t(i,o,a);return"string"!=typeof n||u||s[n]?u?!(r=n):void 0:(i.dataTypes.unshift(n),l(n),!1)}),r}return l(i.dataTypes[0])||!s["*"]&&l("*")}function Ft(e,t){var n,r,i=S.ajaxSettings.flatOptions||{};for(n in t)void 0!==t[n]&&((i[n]?e:r||(r={}))[n]=t[n]);return r&&S.extend(!0,e,r),e}Mt.href=bt.href,S.extend({active:0,lastModified:{},etag:{},ajaxSettings:{url:bt.href,type:"GET",isLocal:/^(?:about|app|app-storage|.+-extension|file|res|widget):$/.test(bt.protocol),global:!0,processData:!0,async:!0,contentType:"application/x-www-form-urlencoded; charset=UTF-8",accepts:{"*":Rt,text:"text/plain",html:"text/html",xml:"application/xml, text/xml",json:"application/json, text/javascript"},contents:{xml:/\bxml\b/,html:/\bhtml/,json:/\bjson\b/},responseFields:{xml:"responseXML",text:"responseText",json:"responseJSON"},converters:{"* text":String,"text html":!0,"text json":JSON.parse,"text xml":S.parseXML},flatOptions:{url:!0,context:!0}},ajaxSetup:function(e,t){return t?Ft(Ft(e,S.ajaxSettings),t):Ft(S.ajaxSettings,e)},ajaxPrefilter:It(Ot),ajaxTransport:It(Pt),ajax:function(e,t){"object"==typeof e&&(t=e,e=void 0),t=t||{};var c,f,p,n,d,r,h,g,i,o,v=S.ajaxSetup({},t),y=v.context||v,m=v.context&&(y.nodeType||y.jquery)?S(y):S.event,x=S.Deferred(),b=S.Callbacks("once memory"),w=v.statusCode||{},a={},s={},u="canceled",T={readyState:0,getResponseHeader:function(e){var t;if(h){if(!n){n={};while(t=qt.exec(p))n[t[1].toLowerCase()+" "]=(n[t[1].toLowerCase()+" "]||[]).concat(t[2])}t=n[e.toLowerCase()+" "]}return null==t?null:t.join(", ")},getAllResponseHeaders:function(){return h?p:null},setRequestHeader:function(e,t){return null==h&&(e=s[e.toLowerCase()]=s[e.toLowerCase()]||e,a[e]=t),this},overrideMimeType:function(e){return null==h&&(v.mimeType=e),this},statusCode:function(e){var t;if(e)if(h)T.always(e[T.status]);else for(t in e)w[t]=[w[t],e[t]];return this},abort:function(e){var t=e||u;return c&&c.abort(t),l(0,t),this}};if(x.promise(T),v.url=((e||v.url||bt.href)+"").replace(Ht,bt.protocol+"//"),v.type=t.method||t.type||v.method||v.type,v.dataTypes=(v.dataType||"*").toLowerCase().match(P)||[""],null==v.crossDomain){r=E.createElement("a");try{r.href=v.url,r.href=r.href,v.crossDomain=Mt.protocol+"//"+Mt.host!=r.protocol+"//"+r.host}catch(e){v.crossDomain=!0}}if(v.data&&v.processData&&"string"!=typeof v.data&&(v.data=S.param(v.data,v.traditional)),Wt(Ot,v,t,T),h)return T;for(i in(g=S.event&&v.global)&&0==S.active++&&S.event.trigger("ajaxStart"),v.type=v.type.toUpperCase(),v.hasContent=!Lt.test(v.type),f=v.url.replace(jt,""),v.hasContent?v.data&&v.processData&&0===(v.contentType||"").indexOf("application/x-www-form-urlencoded")&&(v.data=v.data.replace(Nt,"+")):(o=v.url.slice(f.length),v.data&&(v.processData||"string"==typeof v.data)&&(f+=(Tt.test(f)?"&":"?")+v.data,delete v.data),!1===v.cache&&(f=f.replace(Dt,"$1"),o=(Tt.test(f)?"&":"?")+"_="+wt.guid+++o),v.url=f+o),v.ifModified&&(S.lastModified[f]&&T.setRequestHeader("If-Modified-Since",S.lastModified[f]),S.etag[f]&&T.setRequestHeader("If-None-Match",S.etag[f])),(v.data&&v.hasContent&&!1!==v.contentType||t.contentType)&&T.setRequestHeader("Content-Type",v.contentType),T.setRequestHeader("Accept",v.dataTypes[0]&&v.accepts[v.dataTypes[0]]?v.accepts[v.dataTypes[0]]+("*"!==v.dataTypes[0]?", "+Rt+"; q=0.01":""):v.accepts["*"]),v.headers)T.setRequestHeader(i,v.headers[i]);if(v.beforeSend&&(!1===v.beforeSend.call(y,T,v)||h))return T.abort();if(u="abort",b.add(v.complete),T.done(v.success),T.fail(v.error),c=Wt(Pt,v,t,T)){if(T.readyState=1,g&&m.trigger("ajaxSend",[T,v]),h)return T;v.async&&0<v.timeout&&(d=C.setTimeout(function(){T.abort("timeout")},v.timeout));try{h=!1,c.send(a,l)}catch(e){if(h)throw e;l(-1,e)}}else l(-1,"No Transport");function l(e,t,n,r){var i,o,a,s,u,l=t;h||(h=!0,d&&C.clearTimeout(d),c=void 0,p=r||"",T.readyState=0<e?4:0,i=200<=e&&e<300||304===e,n&&(s=function(e,t,n){var r,i,o,a,s=e.contents,u=e.dataTypes;while("*"===u[0])u.shift(),void 0===r&&(r=e.mimeType||t.getResponseHeader("Content-Type"));if(r)for(i in s)if(s[i]&&s[i].test(r)){u.unshift(i);break}if(u[0]in n)o=u[0];else{for(i in n){if(!u[0]||e.converters[i+" "+u[0]]){o=i;break}a||(a=i)}o=o||a}if(o)return o!==u[0]&&u.unshift(o),n[o]}(v,T,n)),!i&&-1<S.inArray("script",v.dataTypes)&&S.inArray("json",v.dataTypes)<0&&(v.converters["text script"]=function(){}),s=function(e,t,n,r){var i,o,a,s,u,l={},c=e.dataTypes.slice();if(c[1])for(a in e.converters)l[a.toLowerCase()]=e.converters[a];o=c.shift();while(o)if(e.responseFields[o]&&(n[e.responseFields[o]]=t),!u&&r&&e.dataFilter&&(t=e.dataFilter(t,e.dataType)),u=o,o=c.shift())if("*"===o)o=u;else if("*"!==u&&u!==o){if(!(a=l[u+" "+o]||l["* "+o]))for(i in l)if((s=i.split(" "))[1]===o&&(a=l[u+" "+s[0]]||l["* "+s[0]])){!0===a?a=l[i]:!0!==l[i]&&(o=s[0],c.unshift(s[1]));break}if(!0!==a)if(a&&e["throws"])t=a(t);else try{t=a(t)}catch(e){return{state:"parsererror",error:a?e:"No conversion from "+u+" to "+o}}}return{state:"success",data:t}}(v,s,T,i),i?(v.ifModified&&((u=T.getResponseHeader("Last-Modified"))&&(S.lastModified[f]=u),(u=T.getResponseHeader("etag"))&&(S.etag[f]=u)),204===e||"HEAD"===v.type?l="nocontent":304===e?l="notmodified":(l=s.state,o=s.data,i=!(a=s.error))):(a=l,!e&&l||(l="error",e<0&&(e=0))),T.status=e,T.statusText=(t||l)+"",i?x.resolveWith(y,[o,l,T]):x.rejectWith(y,[T,l,a]),T.statusCode(w),w=void 0,g&&m.trigger(i?"ajaxSuccess":"ajaxError",[T,v,i?o:a]),b.fireWith(y,[T,l]),g&&(m.trigger("ajaxComplete",[T,v]),--S.active||S.event.trigger("ajaxStop")))}return T},getJSON:function(e,t,n){return S.get(e,t,n,"json")},getScript:function(e,t){return S.get(e,void 0,t,"script")}}),S.each(["get","post"],function(e,i){S[i]=function(e,t,n,r){return m(t)&&(r=r||n,n=t,t=void 0),S.ajax(S.extend({url:e,type:i,dataType:r,data:t,success:n},S.isPlainObject(e)&&e))}}),S.ajaxPrefilter(function(e){var t;for(t in e.headers)"content-type"===t.toLowerCase()&&(e.contentType=e.headers[t]||"")}),S._evalUrl=function(e,t,n){return S.ajax({url:e,type:"GET",dataType:"script",cache:!0,async:!1,global:!1,converters:{"text script":function(){}},dataFilter:function(e){S.globalEval(e,t,n)}})},S.fn.extend({wrapAll:function(e){var t;return this[0]&&(m(e)&&(e=e.call(this[0])),t=S(e,this[0].ownerDocument).eq(0).clone(!0),this[0].parentNode&&t.insertBefore(this[0]),t.map(function(){var e=this;while(e.firstElementChild)e=e.firstElementChild;return e}).append(this)),this},wrapInner:function(n){return m(n)?this.each(function(e){S(this).wrapInner(n.call(this,e))}):this.each(function(){var e=S(this),t=e.contents();t.length?t.wrapAll(n):e.append(n)})},wrap:function(t){var n=m(t);return this.each(function(e){S(this).wrapAll(n?t.call(this,e):t)})},unwrap:function(e){return this.parent(e).not("body").each(function(){S(this).replaceWith(this.childNodes)}),this}}),S.expr.pseudos.hidden=function(e){return!S.expr.pseudos.visible(e)},S.expr.pseudos.visible=function(e){return!!(e.offsetWidth||e.offsetHeight||e.getClientRects().length)},S.ajaxSettings.xhr=function(){try{return new C.XMLHttpRequest}catch(e){}};var Bt={0:200,1223:204},$t=S.ajaxSettings.xhr();y.cors=!!$t&&"withCredentials"in $t,y.ajax=$t=!!$t,S.ajaxTransport(function(i){var o,a;if(y.cors||$t&&!i.crossDomain)return{send:function(e,t){var n,r=i.xhr();if(r.open(i.type,i.url,i.async,i.username,i.password),i.xhrFields)for(n in i.xhrFields)r[n]=i.xhrFields[n];for(n in i.mimeType&&r.overrideMimeType&&r.overrideMimeType(i.mimeType),i.crossDomain||e["X-Requested-With"]||(e["X-Requested-With"]="XMLHttpRequest"),e)r.setRequestHeader(n,e[n]);o=function(e){return function(){o&&(o=a=r.onload=r.onerror=r.onabort=r.ontimeout=r.onreadystatechange=null,"abort"===e?r.abort():"error"===e?"number"!=typeof r.status?t(0,"error"):t(r.status,r.statusText):t(Bt[r.status]||r.status,r.statusText,"text"!==(r.responseType||"text")||"string"!=typeof r.responseText?{binary:r.response}:{text:r.responseText},r.getAllResponseHeaders()))}},r.onload=o(),a=r.onerror=r.ontimeout=o("error"),void 0!==r.onabort?r.onabort=a:r.onreadystatechange=function(){4===r.readyState&&C.setTimeout(function(){o&&a()})},o=o("abort");try{r.send(i.hasContent&&i.data||null)}catch(e){if(o)throw e}},abort:function(){o&&o()}}}),S.ajaxPrefilter(function(e){e.crossDomain&&(e.contents.script=!1)}),S.ajaxSetup({accepts:{script:"text/javascript, application/javascript, application/ecmascript, application/x-ecmascript"},contents:{script:/\b(?:java|ecma)script\b/},converters:{"text script":function(e){return S.globalEval(e),e}}}),S.ajaxPrefilter("script",function(e){void 0===e.cache&&(e.cache=!1),e.crossDomain&&(e.type="GET")}),S.ajaxTransport("script",function(n){var r,i;if(n.crossDomain||n.scriptAttrs)return{send:function(e,t){r=S("<script>").attr(n.scriptAttrs||{}).prop({charset:n.scriptCharset,src:n.url}).on("load error",i=function(e){r.remove(),i=null,e&&t("error"===e.type?404:200,e.type)}),E.head.appendChild(r[0])},abort:function(){i&&i()}}});var _t,zt=[],Ut=/(=)\?(?=&|$)|\?\?/;S.ajaxSetup({jsonp:"callback",jsonpCallback:function(){var e=zt.pop()||S.expando+"_"+wt.guid++;return this[e]=!0,e}}),S.ajaxPrefilter("json jsonp",function(e,t,n){var r,i,o,a=!1!==e.jsonp&&(Ut.test(e.url)?"url":"string"==typeof e.data&&0===(e.contentType||"").indexOf("application/x-www-form-urlencoded")&&Ut.test(e.data)&&"data");if(a||"jsonp"===e.dataTypes[0])return r=e.jsonpCallback=m(e.jsonpCallback)?e.jsonpCallback():e.jsonpCallback,a?e[a]=e[a].replace(Ut,"$1"+r):!1!==e.jsonp&&(e.url+=(Tt.test(e.url)?"&":"?")+e.jsonp+"="+r),e.converters["script json"]=function(){return o||S.error(r+" was not called"),o[0]},e.dataTypes[0]="json",i=C[r],C[r]=function(){o=arguments},n.always(function(){void 0===i?S(C).removeProp(r):C[r]=i,e[r]&&(e.jsonpCallback=t.jsonpCallback,zt.push(r)),o&&m(i)&&i(o[0]),o=i=void 0}),"script"}),y.createHTMLDocument=((_t=E.implementation.createHTMLDocument("").body).innerHTML="<form></form><form></form>",2===_t.childNodes.length),S.parseHTML=function(e,t,n){return"string"!=typeof e?[]:("boolean"==typeof t&&(n=t,t=!1),t||(y.createHTMLDocument?((r=(t=E.implementation.createHTMLDocument("")).createElement("base")).href=E.location.href,t.head.appendChild(r)):t=E),o=!n&&[],(i=N.exec(e))?[t.createElement(i[1])]:(i=xe([e],t,o),o&&o.length&&S(o).remove(),S.merge([],i.childNodes)));var r,i,o},S.fn.load=function(e,t,n){var r,i,o,a=this,s=e.indexOf(" ");return-1<s&&(r=ht(e.slice(s)),e=e.slice(0,s)),m(t)?(n=t,t=void 0):t&&"object"==typeof t&&(i="POST"),0<a.length&&S.ajax({url:e,type:i||"GET",dataType:"html",data:t}).done(function(e){o=arguments,a.html(r?S("<div>").append(S.parseHTML(e)).find(r):e)}).always(n&&function(e,t){a.each(function(){n.apply(this,o||[e.responseText,t,e])})}),this},S.expr.pseudos.animated=function(t){return S.grep(S.timers,function(e){return t===e.elem}).length},S.offset={setOffset:function(e,t,n){var r,i,o,a,s,u,l=S.css(e,"position"),c=S(e),f={};"static"===l&&(e.style.position="relative"),s=c.offset(),o=S.css(e,"top"),u=S.css(e,"left"),("absolute"===l||"fixed"===l)&&-1<(o+u).indexOf("auto")?(a=(r=c.position()).top,i=r.left):(a=parseFloat(o)||0,i=parseFloat(u)||0),m(t)&&(t=t.call(e,n,S.extend({},s))),null!=t.top&&(f.top=t.top-s.top+a),null!=t.left&&(f.left=t.left-s.left+i),"using"in t?t.using.call(e,f):c.css(f)}},S.fn.extend({offset:function(t){if(arguments.length)return void 0===t?this:this.each(function(e){S.offset.setOffset(this,t,e)});var e,n,r=this[0];return r?r.getClientRects().length?(e=r.getBoundingClientRect(),n=r.ownerDocument.defaultView,{top:e.top+n.pageYOffset,left:e.left+n.pageXOffset}):{top:0,left:0}:void 0},position:function(){if(this[0]){var e,t,n,r=this[0],i={top:0,left:0};if("fixed"===S.css(r,"position"))t=r.getBoundingClientRect();else{t=this.offset(),n=r.ownerDocument,e=r.offsetParent||n.documentElement;while(e&&(e===n.body||e===n.documentElement)&&"static"===S.css(e,"position"))e=e.parentNode;e&&e!==r&&1===e.nodeType&&((i=S(e).offset()).top+=S.css(e,"borderTopWidth",!0),i.left+=S.css(e,"borderLeftWidth",!0))}return{top:t.top-i.top-S.css(r,"marginTop",!0),left:t.left-i.left-S.css(r,"marginLeft",!0)}}},offsetParent:function(){return this.map(function(){var e=this.offsetParent;while(e&&"static"===S.css(e,"position"))e=e.offsetParent;return e||re})}}),S.each({scrollLeft:"pageXOffset",scrollTop:"pageYOffset"},function(t,i){var o="pageYOffset"===i;S.fn[t]=function(e){return $(this,function(e,t,n){var r;if(x(e)?r=e:9===e.nodeType&&(r=e.defaultView),void 0===n)return r?r[i]:e[t];r?r.scrollTo(o?r.pageXOffset:n,o?n:r.pageYOffset):e[t]=n},t,e,arguments.length)}}),S.each(["top","left"],function(e,n){S.cssHooks[n]=Fe(y.pixelPosition,function(e,t){if(t)return t=We(e,n),Pe.test(t)?S(e).position()[n]+"px":t})}),S.each({Height:"height",Width:"width"},function(a,s){S.each({padding:"inner"+a,content:s,"":"outer"+a},function(r,o){S.fn[o]=function(e,t){var n=arguments.length&&(r||"boolean"!=typeof e),i=r||(!0===e||!0===t?"margin":"border");return $(this,function(e,t,n){var r;return x(e)?0===o.indexOf("outer")?e["inner"+a]:e.document.documentElement["client"+a]:9===e.nodeType?(r=e.documentElement,Math.max(e.body["scroll"+a],r["scroll"+a],e.body["offset"+a],r["offset"+a],r["client"+a])):void 0===n?S.css(e,t,i):S.style(e,t,n,i)},s,n?e:void 0,n)}})}),S.each(["ajaxStart","ajaxStop","ajaxComplete","ajaxError","ajaxSuccess","ajaxSend"],function(e,t){S.fn[t]=function(e){return this.on(t,e)}}),S.fn.extend({bind:function(e,t,n){return this.on(e,null,t,n)},unbind:function(e,t){return this.off(e,null,t)},delegate:function(e,t,n,r){return this.on(t,e,n,r)},undelegate:function(e,t,n){return 1===arguments.length?this.off(e,"**"):this.off(t,e||"**",n)},hover:function(e,t){return this.mouseenter(e).mouseleave(t||e)}}),S.each("blur focus focusin focusout resize scroll click dblclick mousedown mouseup mousemove mouseover mouseout mouseenter mouseleave change select submit keydown keypress keyup contextmenu".split(" "),function(e,n){S.fn[n]=function(e,t){return 0<arguments.length?this.on(n,null,e,t):this.trigger(n)}});var Xt=/^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g;S.proxy=function(e,t){var n,r,i;if("string"==typeof t&&(n=e[t],t=e,e=n),m(e))return r=s.call(arguments,2),(i=function(){return e.apply(t||this,r.concat(s.call(arguments)))}).guid=e.guid=e.guid||S.guid++,i},S.holdReady=function(e){e?S.readyWait++:S.ready(!0)},S.isArray=Array.isArray,S.parseJSON=JSON.parse,S.nodeName=A,S.isFunction=m,S.isWindow=x,S.camelCase=X,S.type=w,S.now=Date.now,S.isNumeric=function(e){var t=S.type(e);return("number"===t||"string"===t)&&!isNaN(e-parseFloat(e))},S.trim=function(e){return null==e?"":(e+"").replace(Xt,"")},"function"==typeof define&&define.amd&&define("jquery",[],function(){return S});var Vt=C.jQuery,Gt=C.$;return S.noConflict=function(e){return C.$===S&&(C.$=Gt),e&&C.jQuery===S&&(C.jQuery=Vt),S},"undefined"==typeof e&&(C.jQuery=C.$=S),S});
</script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<style type="text/css">html{font-family:sans-serif;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%}body{margin:0}article,aside,details,figcaption,figure,footer,header,hgroup,main,menu,nav,section,summary{display:block}audio,canvas,progress,video{display:inline-block;vertical-align:baseline}audio:not([controls]){display:none;height:0}[hidden],template{display:none}a{background-color:transparent}a:active,a:hover{outline:0}abbr[title]{border-bottom:1px dotted}b,strong{font-weight:700}dfn{font-style:italic}h1{margin:.67em 0;font-size:2em}mark{color:#000;background:#ff0}small{font-size:80%}sub,sup{position:relative;font-size:75%;line-height:0;vertical-align:baseline}sup{top:-.5em}sub{bottom:-.25em}img{border:0}svg:not(:root){overflow:hidden}figure{margin:1em 40px}hr{height:0;-webkit-box-sizing:content-box;-moz-box-sizing:content-box;box-sizing:content-box}pre{overflow:auto}code,kbd,pre,samp{font-family:monospace,monospace;font-size:1em}button,input,optgroup,select,textarea{margin:0;font:inherit;color:inherit}button{overflow:visible}button,select{text-transform:none}button,html input[type=button],input[type=reset],input[type=submit]{-webkit-appearance:button;cursor:pointer}button[disabled],html input[disabled]{cursor:default}button::-moz-focus-inner,input::-moz-focus-inner{padding:0;border:0}input{line-height:normal}input[type=checkbox],input[type=radio]{-webkit-box-sizing:border-box;-moz-box-sizing:border-box;box-sizing:border-box;padding:0}input[type=number]::-webkit-inner-spin-button,input[type=number]::-webkit-outer-spin-button{height:auto}input[type=search]{-webkit-box-sizing:content-box;-moz-box-sizing:content-box;box-sizing:content-box;-webkit-appearance:textfield}input[type=search]::-webkit-search-cancel-button,input[type=search]::-webkit-search-decoration{-webkit-appearance:none}fieldset{padding:.35em .625em .75em;margin:0 2px;border:1px solid silver}legend{padding:0;border:0}textarea{overflow:auto}optgroup{font-weight:700}table{border-spacing:0;border-collapse:collapse}td,th{padding:0}@media print{*,:after,:before{color:#000!important;text-shadow:none!important;background:0 0!important;-webkit-box-shadow:none!important;box-shadow:none!important}a,a:visited{text-decoration:underline}a[href]:after{content:" (" attr(href) ")"}abbr[title]:after{content:" (" attr(title) ")"}a[href^="javascript:"]:after,a[href^="#"]:after{content:""}blockquote,pre{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}img,tr{page-break-inside:avoid}img{max-width:100%!important}h2,h3,p{orphans:3;widows:3}h2,h3{page-break-after:avoid}.navbar{display:none}.btn>.caret,.dropup>.btn>.caret{border-top-color:#000!important}.label{border:1px solid #000}.table{border-collapse:collapse!important}.table td,.table th{background-color:#fff!important}.table-bordered td,.table-bordered th{border:1px solid #ddd!important}}@font-face{font-family:'Glyphicons Halflings';src:url(data:application/vnd.ms-fontobject;base64,n04AAEFNAAACAAIABAAAAAAABQAAAAAAAAABAJABAAAEAExQAAAAAAAAAAIAAAAAAAAAAAEAAAAAAAAAJxJ/LAAAAAAAAAAAAAAAAAAAAAAAACgARwBMAFkAUABIAEkAQwBPAE4AUwAgAEgAYQBsAGYAbABpAG4AZwBzAAAADgBSAGUAZwB1AGwAYQByAAAAeABWAGUAcgBzAGkAbwBuACAAMQAuADAAMAA5ADsAUABTACAAMAAwADEALgAwADAAOQA7AGgAbwB0AGMAbwBuAHYAIAAxAC4AMAAuADcAMAA7AG0AYQBrAGUAbwB0AGYALgBsAGkAYgAyAC4ANQAuADUAOAAzADIAOQAAADgARwBMAFkAUABIAEkAQwBPAE4AUwAgAEgAYQBsAGYAbABpAG4AZwBzACAAUgBlAGcAdQBsAGEAcgAAAAAAQlNHUAAAAAAAAAAAAAAAAAAAAAADAKncAE0TAE0ZAEbuFM3pjM/SEdmjKHUbyow8ATBE40IvWA3vTu8LiABDQ+pexwUMcm1SMnNryctQSiI1K5ZnbOlXKmnVV5YvRe6RnNMFNCOs1KNVpn6yZhCJkRtVRNzEufeIq7HgSrcx4S8h/v4vnrrKc6oCNxmSk2uKlZQHBii6iKFoH0746ThvkO1kJHlxjrkxs+LWORaDQBEtiYJIR5IB9Bi1UyL4Rmr0BNigNkMzlKQmnofBHviqVzUxwdMb3NdCn69hy+pRYVKGVS/1tnsqv4LL7wCCPZZAZPT4aCShHjHJVNuXbmMrY5LeQaGnvAkXlVrJgKRAUdFjrWEah9XebPeQMj7KS7DIBAFt8ycgC5PLGUOHSE3ErGZCiViNLL5ZARfywnCoZaKQCu6NuFX42AEeKtKUGnr/Cm2Cy8tpFhBPMW5Fxi4Qm4TkDWh4IWFDClhU2hRWosUWqcKLlgyXB+lSHaWaHiWlBAR8SeSgSPCQxdVQgzUixWKSTrIQEbU94viDctkvX+VSjJuUmV8L4CXShI11esnp0pjWNZIyxKHS4wVQ2ime1P4RnhvGw0aDN1OLAXGERsB7buFpFGGBAre4QEQR0HOIO5oYH305G+KspT/FupEGGafCCwxSe6ZUa+073rXHnNdVXE6eWvibUS27XtRzkH838mYLMBmYysZTM0EM3A1fbpCBYFccN1B/EnCYu/TgCGmr7bMh8GfYL+BfcLvB0gRagC09w9elfldaIy/hNCBLRgBgtCC7jAF63wLSMAfbfAlEggYU0bUA7ACCJmTDpEmJtI78w4/BO7dN7JR7J7ZvbYaUbaILSQsRBiF3HGk5fEg6p9unwLvn98r+vnsV+372uf1xBLq4qU/45fTuqaAP+pssmCCCTF0mhEow8ZXZOS8D7Q85JsxZ+Azok7B7O/f6J8AzYBySZQB/QHYUSA+EeQhEWiS6AIQzgcsDiER4MjgMBAWDV4AgQ3g1eBgIdweCQmCjJEMkJ+PKRWyFHHmg1Wi/6xzUgA0LREoKJChwnQa9B+5RQZRB3IlBlkAnxyQNaANwHMowzlYSMCBgnbpzvqpl0iTJNCQidDI9ZrSYNIRBhHtUa5YHMHxyGEik9hDE0AKj72AbTCaxtHPUaKZdAZSnQTyjGqGLsmBStCejApUhg4uBMU6mATujEl+KdDPbI6Ag4vLr+hjY6lbjBeoLKnZl0UZgRX8gTySOeynZVz1wOq7e1hFGYIq+MhrGxDLak0PrwYzSXtcuyhXEhwOYofiW+EcI/jw8P6IY6ed+etAbuqKp5QIapT77LnAe505lMuqL79a0ut4rWexzFttsOsLDy7zvtQzcq3U1qabe7tB0wHWVXji+zDbo8x8HyIRUbXnwUcklFv51fvTymiV+MXLSmGH9d9+aXpD5X6lao41anWGig7IwIdnoBY2ht/pO9mClLo4NdXHAsefqWUKlXJkbqPOFhMoR4aiA1BXqhRNbB2Xwi+7u/jpAoOpKJ0UX24EsrzMfHXViakCNcKjBxuQX8BO0ZqjJ3xXzf+61t2VXOSgJ8xu65QKgtN6FibPmPYsXbJRHHqbgATcSZxBqGiDiU4NNNsYBsKD0MIP/OfKnlk/Lkaid/O2NbKeuQrwOB2Gq3YHyr6ALgzym5wIBnsdC1ZkoBFZSQXChZvlesPqvK2c5oHHT3Q65jYpNxnQcGF0EHbvYqoFw60WNlXIHQF2HQB7zD6lWjZ9rVqUKBXUT6hrkZOle0RFYII0V5ZYGl1JAP0Ud1fZZMvSomBzJ710j4Me8mjQDwEre5Uv2wQfk1ifDwb5ksuJQQ3xt423lbuQjvoIQByQrNDh1JxGFkOdlJvu/gFtuW0wR4cgd+ZKesSV7QkNE2kw6AV4hoIuC02LGmTomyf8PiO6CZzOTLTPQ+HW06H+tx+bQ8LmDYg1pTFrp2oJXgkZTyeRJZM0C8aE2LpFrNVDuhARsN543/FV6klQ6Tv1OoZGXLv0igKrl/CmJxRmX7JJbJ998VSIPQRyDBICzl4JJlYHbdql30NvYcOuZ7a10uWRrgoieOdgIm4rlq6vNOQBuqESLbXG5lzdJGHw2m0sDYmODXbYGTfSTGRKpssTO95fothJCjUGQgEL4yKoGAF/0SrpUDNn8CBgBcSDQByAeNkCXp4S4Ro2Xh4OeaGRgR66PVOsU8bc6TR5/xTcn4IVMLOkXSWiXxkZQCbvKfmoAvQaKjO3EDKwkwqHChCDEM5loQRPd5ACBki1TjF772oaQhQbQ5C0lcWXPFOzrfsDGUXGrpxasbG4iab6eByaQkQfm0VFlP0ZsDkvvqCL6QXMUwCjdMx1ZOyKhTJ7a1GWAdOUcJ8RSejxNVyGs31OKMyRyBVoZFjqIkmKlLQ5eHMeEL4MkUf23cQ/1SgRCJ1dk4UdBT7OoyuNgLs0oCd8RnrEIb6QdMxT2QjD4zMrJkfgx5aDMcA4orsTtKCqWb/Veyceqa5OGSmB28YwH4rFbkQaLoUN8OQQYnD3w2eXpI4ScQfbCUZiJ4yMOIKLyyTc7BQ4uXUw6Ee6/xM+4Y67ngNBknxIPwuppgIhFcwJyr6EIj+LzNj/mfR2vhhRlx0BILZoAYruF0caWQ7YxO66UmeguDREAFHYuC7HJviRgVO6ruJH59h/C/PkgSle8xNzZJULLWq9JMDTE2fjGE146a1Us6PZDGYle6ldWRqn/pdpgHKNGrGIdkRK+KPETT9nKT6kLyDI8xd9A1FgWmXWRAIHwZ37WyZHOVyCadJEmMVz0MadMjDrPho+EIochkVC2xgGiwwsQ6DMv2P7UXqT4x7CdcYGId2BJQQa85EQKmCmwcRejQ9Bm4oATENFPkxPXILHpMPUyWTI5rjNOsIlmEeMbcOCEqInpXACYQ9DDxmFo9vcmsDblcMtg4tqBerNngkIKaFJmrQAPnq1dEzsMXcwjcHdfdCibcAxxA+q/j9m3LM/O7WJka4tSidVCjsvo2lQ/2ewyoYyXwAYyr2PlRoR5MpgVmSUIrM3PQxXPbgjBOaDQFIyFMJvx3Pc5RSYj12ySVF9fwFPQu2e2KWVoL9q3Ayv3IzpGHUdvdPdrNUdicjsTQ2ISy7QU3DrEytIjvbzJnAkmANXjAFERA0MUoPF3/5KFmW14bBNOhwircYgMqoDpUMcDtCmBE82QM2YtdjVLB4kBuKho/bcwQdeboqfQartuU3CsCf+cXkgYAqp/0Ee3RorAZt0AvvOCSI4JICIlGlsV0bsSid/NIEALAAzb6HAgyWHBps6xAOwkJIGcB82CxRQq4sJf3FzA70A+TRqcqjEMETCoez3mkPcpnoALs0ugJY8kQwrC+JE5ik3w9rzrvDRjAQnqgEVvdGrNwlanR0SOKWzxOJOvLJhcd8Cl4AshACUkv9czdMkJCVQSQhp6kp7StAlpVRpK0t0SW6LHeBJnE2QchB5Ccu8kxRghZXGIgZIiSj7gEKMJDClcnX6hgoqJMwiQDigIXg3ioFLCgDgjPtYHYpsF5EiA4kcnN18MZtOrY866dEQAb0FB34OGKHGZQjwW/WDHA60cYFaI/PjpzquUqdaYGcIq+mLez3WLFFCtNBN2QJcrlcoELgiPku5R5dSlJFaCEqEZle1AQzAKC+1SotMcBNyQUFuRHRF6OlimSBgjZeTBCwLyc6A+P/oFRchXTz5ADknYJHxzrJ5pGuIKRQISU6WyKTBBjD8WozmVYWIsto1AS5rxzKlvJu4E/vwOiKxRtCWsDM+eTHUrmwrCK5BIfMzGkD+0Fk5LzBs0jMYXktNDblB06LMNJ09U8pzSLmo14MS0OMjcdrZ31pyQqxJJpRImlSvfYAK8inkYU52QY2FPEVsjoWewpwhRp5yAuNpkqhdb7ku9Seefl2D0B8SMTFD90xi4CSOwwZy9IKkpMtI3FmFUg3/kFutpQGNc3pCR7gvC4sgwbupDu3DyEN+W6YGLNM21jpB49irxy9BSlHrVDlnihGKHwPrbVFtc+h1rVQKZduxIyojccZIIcOCmhEnC7UkY68WXKQgLi2JCDQkQWJRQuk60hZp0D3rtCTINSeY9Ej2kIKYfGxwOs4j9qMM7fYZiipzgcf7TamnehqdhsiMiCawXnz4xAbyCkLAx5EGbo3Ax1u3dUIKnTxIaxwQTHehPl3V491H0+bC5zgpGz7Io+mjdhKlPJ01EeMpM7UsRJMi1nGjmJg35i6bQBAAxjO/ENJubU2mg3ONySEoWklCwdABETcs7ck3jgiuU9pcKKpbgn+3YlzV1FzIkB6pmEDOSSyDfPPlQskznctFji0kpgZjW5RZe6x9kYT4KJcXg0bNiCyif+pZACCyRMmYsfiKmN9tSO65F0R2OO6ytlEhY5Sj6uRKfFxw0ijJaAx/k3QgnAFSq27/2i4GEBA+UvTJKK/9eISNvG46Em5RZfjTYLdeD8kdXHyrwId/DQZUaMCY4gGbke2C8vfjgV/Y9kkRQOJIn/xM9INZSpiBnqX0Q9GlQPpPKAyO5y+W5NMPSRdBCUlmuxl40ZfMCnf2Cp044uI9WLFtCi4YVxKjuRCOBWIb4XbIsGdbo4qtMQnNOQz4XDSui7W/N6l54qOynCqD3DpWQ+mpD7C40D8BZEWGJX3tlAaZBMj1yjvDYKwCJBa201u6nBKE5UE+7QSEhCwrXfbRZylAaAkplhBWX50dumrElePyNMRYUrC99UmcSSNgImhFhDI4BXjMtiqkgizUGCrZ8iwFxU6fQ8GEHCFdLewwxYWxgScAYMdMLmcZR6b7rZl95eQVDGVoUKcRMM1ixXQtXNkBETZkVVPg8LoSrdetHzkuM7DjZRHP02tCxA1fmkXKF3VzfN1pc1cv/8lbTIkkYpqKM9VOhp65ktYk+Q46myFWBapDfyWUCnsnI00QTBQmuFjMZTcd0V2NQ768Fhpby04k2IzNR1wKabuGJqYWwSly6ocMFGTeeI+ejsWDYgEvr66QgqdcIbFYDNgsm0x9UHY6SCd5+7tpsLpKdvhahIDyYmEJQCqMqtCF6UlrE5GXRmbu+vtm3BFSxI6ND6UxIE7GsGMgWqghXxSnaRJuGFveTcK5ZVSPJyjUxe1dKgI6kNF7EZhIZs8y8FVqwEfbM0Xk2ltORVDKZZM40SD3qQoQe0orJEKwPfZwm3YPqwixhUMOndis6MhbmfvLBKjC8sKKIZKbJk8L11oNkCQzCgvjhyyEiQSuJcgCQSG4Mocfgc0Hkwcjal1UNgP0CBPikYqBIk9tONv4kLtBswH07vUCjEaHiFGlLf8MgXKzSgjp2HolRRccAOh0ILHz9qlGgIFkwAnzHJRjWFhlA7ROwINyB5HFj59PRZHFor6voq7l23EPNRwdWhgawqbivLSjRA4htEYUFkjESu67icTg5S0aW1sOkCiIysfJ9UnIWevOOLGpepcBxy1wEhd2WI3AZg7sr9WBmHWyasxMcvY/iOmsLtHSWNUWEGk9hScMPShasUA1AcHOtRZlqMeQ0OzYS9vQvYUjOLrzP07BUAFikcJNMi7gIxEw4pL1G54TcmmmoAQ5s7TGWErJZ2Io4yQ0ljRYhL8H5e62oDtLF8aDpnIvZ5R3GWJyAugdiiJW9hQAVTsnCBHhwu7rkBlBX6r3b7ejEY0k5GGeyKv66v+6dg7mcJTrWHbtMywbedYqCQ0FPwoytmSWsL8WTtChZCKKzEF7vP6De4x2BJkkniMgSdWhbeBSLtJZR9CTHetK1xb34AYIJ37OegYIoPVbXgJ/qDQK+bfCtxQRVKQu77WzOoM6SGL7MaZwCGJVk46aImai9fmam+WpHG+0BtQPWUgZ7RIAlPq6lkECUhZQ2gqWkMYKcYMYaIc4gYCDFHYa2d1nzp3+J1eCBay8IYZ0wQRKGAqvCuZ/UgbQPyllosq+XtfKIZOzmeJqRazpmmoP/76YfkjzV2NlXTDSBYB04SVlNQsFTbGPk1t/I4Jktu0XSgifO2ozFOiwd/0SssJDn0dn4xqk4GDTTKX73/wQyBLdqgJ+Wx6AQaba3BA9CKEzjtQYIfAsiYamapq80LAamYjinlKXUkxdpIDk0puXUEYzSalfRibAeDAKpNiqQ0FTwoxuGYzRnisyTotdVTclis1LHRQCy/qqL8oUaQzWRxilq5Mi0IJGtMY02cGLD69vGjkj3p6pGePKI8bkBv5evq8SjjyU04vJR2cQXQwSJyoinDsUJHCQ50jrFTT7yRdbdYQMB3MYCb6uBzJ9ewhXYPAIZSXfeEQBZZ3GPN3Nbhh/wkvAJLXnQMdi5NYYZ5GHE400GS5rXkOZSQsdZgIbzRnF9ueLnsfQ47wHAsirITnTlkCcuWWIUhJSbpM3wWhXNHvt2xUsKKMpdBSbJnBMcihkoDqAd1Zml/R4yrzow1Q2A5G+kzo/RhRxQS2lCSDRV8LlYLBOOoo1bF4jwJAwKMK1tWLHlu9i0j4Ig8qVm6wE1DxXwAwQwsaBWUg2pOOol2dHxyt6npwJEdLDDVYyRc2D0HbcbLUJQj8gPevQBUBOUHXPrsAPBERICpnYESeu2OHotpXQxRGlCCtLdIsu23MhZVEoJg8Qumj/UMMc34IBqTKLDTp76WzL/dMjCxK7MjhiGjeYAC/kj/jY/Rde7hpSM1xChrog6yZ7OWTuD56xBJnGFE+pT2ElSyCnJcwVzCjkqeNLfMEJqKW0G7OFIp0G+9mh50I9o8k1tpCY0xYqFNIALgIfc2me4n1bmJnRZ89oepgLPT0NTMLNZsvSCZAc3TXaNB07vail36/dBySis4m9/DR8izaLJW6bWCkVgm5T+ius3ZXq4xI+GnbveLbdRwF2mNtsrE0JjYc1AXknCOrLSu7Te/r4dPYMCl5qtiHNTn+TPbh1jCBHH+dMJNhwNgs3nT+OhQoQ0vYif56BMG6WowAcHR3DjQolxLzyVekHj00PBAaW7IIAF1EF+uRIWyXjQMAs2chdpaKPNaB+kSezYt0+CA04sOg5vx8Fr7Ofa9sUv87h7SLAUFSzbetCCZ9pmyLt6l6/TzoA1/ZBG9bIUVHLAbi/kdBFgYGyGwRQGBpkqCEg2ah9UD6EedEcEL3j4y0BQQCiExEnocA3SZboh+epgd3YsOkHskZwPuQ5OoyA0fTA5AXrHcUOQF+zkJHIA7PwCDk1gGVmGUZSSoPhNf+Tklauz98QofOlCIQ/tCD4dosHYPqtPCXB3agggQQIqQJsSkB+qn0rkQ1toJjON/OtCIB9RYv3PqRA4C4U68ZMlZn6BdgEvi2ziU+TQ6NIw3ej+AtDwMGEZk7e2IjxUWKdAxyaw9OCwSmeADTPPleyk6UhGDNXQb++W6Uk4q6F7/rg6WVTo82IoCxSIsFDrav4EPHphD3u4hR53WKVvYZUwNCCeM4PMBWzK+EfIthZOkuAwPo5C5jgoZgn6dUdvx5rIDmd58cXXdKNfw3l+wM2UjgrDJeQHhbD7HW2QDoZMCujgIUkk5Fg8VCsdyjOtnGRx8wgKRPZN5dR0zPUyfGZFVihbFRniXZFOZGKPnEQzU3AnD1KfR6weHW2XS6KbPJxUkOTZsAB9vTVp3Le1F8q5l+DMcLiIq78jxAImD2pGFw0VHfRatScGlK6SMu8leTmhUSMy8Uhdd6xBiH3Gdman4tjQGLboJfqz6fL2WKHTmrfsKZRYX6BTDjDldKMosaSTLdQS7oDisJNqAUhw1PfTlnacCO8vl8706Km1FROgLDmudzxg+EWTiArtHgLsRrAXYWdB0NmToNCJdKm0KWycZQqb+Mw76Qy29iQ5up/X7oyw8QZ75kP5F6iJAJz6KCmqxz8fEa/xnsMYcIO/vEkGRuMckhr4rIeLrKaXnmIzlNLxbFspOphkcnJdnz/Chp/Vlpj2P7jJQmQRwGnltkTV5dbF9fE3/fxoSqTROgq9wFUlbuYzYcasE0ouzBo+dDCDzxKAfhbAZYxQiHrLzV2iVexnDX/QnT1fsT/xuhu1ui5qIytgbGmRoQkeQooO8eJNNZsf0iALur8QxZFH0nCMnjerYQqG1pIfjyVZWxhVRznmmfLG00BcBWJE6hzQWRyFknuJnXuk8A5FRDCulwrWASSNoBtR+CtGdkPwYN2o7DOw/VGlCZPusRBFXODQdUM5zeHDIVuAJBLqbO/f9Qua+pDqEPk230Sob9lEZ8BHiCorjVghuI0lI4JDgHGRDD/prQ84B1pVGkIpVUAHCG+iz3Bn3qm2AVrYcYWhock4jso5+J7HfHVj4WMIQdGctq3psBCVVzupQOEioBGA2Bk+UILT7+VoX5mdxxA5fS42gISQVi/HTzrgMxu0fY6hE1ocUwwbsbWcezrY2n6S8/6cxXkOH4prpmPuFoikTzY7T85C4T2XYlbxLglSv2uLCgFv8Quk/wdesUdWPeHYIH0R729JIisN9Apdd4eB10aqwXrPt+Su9mA8k8n1sjMwnfsfF2j3jMUzXepSHmZ/BfqXvzgUNQQWOXO8YEuFBh4QTYCkOAPxywpYu1VxiDyJmKVcmJPGWk/gc3Pov02StyYDahwmzw3E1gYC9wkupyWfDqDSUMpCTH5e5N8B//lHiMuIkTNw4USHrJU67bjXGqNav6PBuQSoqTxc8avHoGmvqNtXzIaoyMIQIiiUHIM64cXieouplhNYln7qgc4wBVAYR104kO+CvKqsg4yIUlFNThVUAKZxZt1XA34h3TCUUiXVkZ0w8Hh2R0Z5L0b4LZvPd/p1gi/07h8qfwHrByuSxglc9cI4QIg2oqvC/qm0i7tjPLTgDhoWTAKDO2ONW5oe+/eKB9vZB8K6C25yCZ9RFVMnb6NRdRjyVK57CHHSkJBfnM2/j4ODUwRkqrtBBCrDsDpt8jhZdXoy/1BCqw3sSGhgGGy0a5Jw6BP/TExoCmNFYjZl248A0osgPyGEmRA+fAsqPVaNAfytu0vuQJ7rk3J4kTDTR2AlCHJ5cls26opZM4w3jMULh2YXKpcqGBtuleAlOZnaZGbD6DHzMd6i2oFeJ8z9XYmalg1Szd/ocZDc1C7Y6vcALJz2lYnTXiWEr2wawtoR4g3jvWUU2Ngjd1cewtFzEvM1NiHZPeLlIXFbBPawxNgMwwAlyNSuGF3zizVeOoC9bag1qRAQKQE/EZBWC2J8mnXAN2aTBboZ7HewnObE8CwROudZHmUM5oZ/Ugd/JZQK8lvAm43uDRAbyW8gZ+ZGq0EVerVGUKUSm/Idn8AQHdR4m7bue88WBwft9mSCeMOt1ncBwziOmJYI2ZR7ewNMPiCugmSsE4EyQ+QATJG6qORMGd4snEzc6B4shPIo4G1T7PgSm8PY5eUkPdF8JZ0VBtadbHXoJgnEhZQaODPj2gpODKJY5Yp4DOsLBFxWbvXN755KWylJm+oOd4zEL9Hpubuy2gyyfxh8oEfFutnYWdfB8PdESLWYvSqbElP9qo3u6KTmkhoacDauMNNjj0oy40DFV7Ql0aZj77xfGl7TJNHnIwgqOkenruYYNo6h724+zUQ7+vkCpZB+pGA562hYQiDxHVWOq0oDQl/QsoiY+cuI7iWq/ZIBtHcXJ7kks+h2fCNUPA82BzjnqktNts+RLdk1VSu+tqEn7QZCCsvEqk6FkfiOYkrsw092J8jsfIuEKypNjLxrKA9kiA19mxBD2suxQKCzwXGws7kEJvlhUiV9tArLIdZW0IORcxEzdzKmjtFhsjKy/44XYXdI5noQoRcvjZ1RMPACRqYg2V1+OwOepcOknRLLFdYgTkT5UApt/JhLM3jeFYprZV+Zow2g8fP+U68hkKFWJj2yBbKqsrp25xkZX1DAjUw52IMYWaOhab8Kp05VrdNftqwRrymWF4OQSjbdfzmRZirK8FMJELEgER2PHjEAN9pGfLhCUiTJFbd5LBkOBMaxLr/A1SY9dXFz4RjzoU9ExfJCmx/I9FKEGT3n2cmzl2X42L3Jh+AbQq6sA+Ss1kitoa4TAYgKHaoybHUDJ51oETdeI/9ThSmjWGkyLi5QAGWhL0BG1UsTyRGRJOldKBrYJeB8ljLJHfATWTEQBXBDnQexOHTB+Un44zExFE4vLytcu5NwpWrUxO/0ZICUGM7hGABXym0V6ZvDST0E370St9MIWQOTWngeoQHUTdCJUP04spMBMS8LSker9cReVQkULFDIZDFPrhTzBl6sed9wcZQTbL+BDqMyaN3RJPh/anbx+Iv+qgQdAa3M9Z5JmvYlh4qop+Ho1F1W5gbOE9YKLgAnWytXElU4G8GtW47lhgFE6gaSs+gs37sFvi0PPVvA5dnCBgILTwoKd/+DoL9F6inlM7H4rOTzD79KJgKlZO/Zgt22UsKhrAaXU5ZcLrAglTVKJEmNJvORGN1vqrcfSMizfpsgbIe9zno+gBoKVXgIL/VI8dB1O5o/R3Suez/gD7M781ShjKpIIORM/nxG+jjhhgPwsn2IoXsPGPqYHXA63zJ07M2GPEykQwJBYLK808qYxuIew4frk52nhCsnCYmXiR6CuapvE1IwRB4/QftDbEn+AucIr1oxrLabRj9q4ae0+fXkHnteAJwXRbVkR0mctVSwEbqhJiMSZUp9DNbEDMmjX22m3ABpkrPQQTP3S1sib5pD2VRKRd+eNAjLYyT0hGrdjWJZy24OYXRoWQAIhGBZRxuBFMjjZQhpgrWo8SiFYbojcHO8V5DyscJpLTHyx9Fimassyo5U6WNtquUMYgccaHY5amgR3PQzq3ToNM5ABnoB9kuxsebqmYZm0R9qxJbFXCQ1UPyFIbxoUraTJFDpCk0Wk9GaYJKz/6oHwEP0Q14lMtlddQsOAU9zlYdMVHiT7RQP3XCmWYDcHCGbVRHGnHuwzScA0BaSBOGkz3lM8CArjrBsyEoV6Ys4qgDK3ykQQPZ3hCRGNXQTNNXbEb6tDiTDLKOyMzRhCFT+mAUmiYbV3YQVqFVp9dorv+TsLeCykS2b5yyu8AV7IS9cxcL8z4Kfwp+xJyYLv1OsxQCZwTB4a8BZ/5EdxTBJthApqyfd9u3ifr/WILTqq5VqgwMT9SOxbSGWLQJUUWCVi4k9tho9nEsbUh7U6NUsLmkYFXOhZ0kmamaJLRNJzSj/qn4Mso6zb6iLLBXoaZ6AqeWCjHQm2lztnejYYM2eubnpBdKVLORZhudH3JF1waBJKA9+W8EhMj3Kzf0L4vi4k6RoHh3Z5YgmSZmk6ns4fjScjAoL8GoOECgqgYEBYUGFVO4FUv4/YtowhEmTs0vrvlD/CrisnoBNDAcUi/teY7OctFlmARQzjOItrrlKuPO6E2Ox93L4O/4DcgV/dZ7qR3VBwVQxP1GCieA4RIpweYJ5FoYrHxqRBdJjnqbsikA2Ictbb8vE1GYIo9dacK0REgDX4smy6GAkxlH1yCGGsk+tgiDhNKuKu3yNrMdxafmKTF632F8Vx4BNK57GvlFisrkjN9WDAtjsWA0ENT2e2nETUb/n7qwhvGnrHuf5bX6Vh/n3xffU3PeHdR+FA92i6ufT3AlyAREoNDh6chiMWTvjKjHDeRhOa9YkOQRq1vQXEMppAQVwHCuIcV2g5rBn6GmZZpTR7vnSD6ZmhdSl176gqKTXu5E+YbfL0adwNtHP7dT7t7b46DVZIkzaRJOM+S6KcrzYVg+T3wSRFRQashjfU18NutrKa/7PXbtuJvpIjbgPeqd+pjmRw6YKpnANFSQcpzTZgpSNJ6J7uiagAbir/8tNXJ/OsOnRh6iuIexxrmkIneAgz8QoLmiaJ8sLQrELVK2yn3wOHp57BAZJhDZjTBzyoRAuuZ4eoxHruY1pSb7qq79cIeAdOwin4GdgMeIMHeG+FZWYaiUQQyC5b50zKjYw97dFjAeY2I4Bnl105Iku1y0lMA1ZHolLx19uZnRdILcXKlZGQx/GdEqSsMRU1BIrFqRcV1qQOOHyxOLXEGcbRtAEsuAC2V4K3p5mFJ22IDWaEkk9ttf5Izb2LkD1MnrSwztXmmD/Qi/EmVEFBfiKGmftsPwVaIoZanlKndMZsIBOskFYpDOq3QUs9aSbAAtL5Dbokus2G4/asthNMK5UQKCOhU97oaOYNGsTah+jfCKsZnTRn5TbhFX8ghg8CBYt/BjeYYYUrtUZ5jVij/op7V5SsbA4mYTOwZ46hqdpbB6Qvq3AS2HHNkC15pTDIcDNGsMPXaBidXYPHc6PJAkRh29Vx8KcgX46LoUQBhRM+3SW6Opll/wgxxsPgKJKzr5QCmwkUxNbeg6Wj34SUnEzOemSuvS2OetRCO8Tyy+QbSKVJcqkia+GvDefFwMOmgnD7h81TUtMn+mRpyJJ349HhAnoWFTejhpYTL9G8N2nVg1qkXBeoS9Nw2fB27t7trm7d/QK7Cr4uoCeOQ7/8JfKT77KiDzLImESHw/0wf73QeHu74hxv7uihi4fTX+XEwAyQG3264dwv17aJ5N335Vt9sdrAXhPOAv8JFvzqyYXwfx8WYJaef1gMl98JRFyl5Mv5Uo/oVH5ww5OzLFsiTPDns7fS6EURSSWd/92BxMYQ8sBaH+j+wthQPdVgDGpTfi+JQIWMD8xKqULliRH01rTeyF8x8q/GBEEEBrAJMPf25UQwi0b8tmqRXY7kIvNkzrkvRWLnxoGYEJsz8u4oOyMp8cHyaybb1HdMCaLApUE+/7xLIZGP6H9xuSEXp1zLIdjk5nBaMuV/yTDRRP8Y2ww5RO6d2D94o+6ucWIqUAvgHIHXhZsmDhjVLczmZ3ca0Cb3PpKwt2UtHVQ0BgFJsqqTsnzZPlKahRUkEu4qmkJt+kqdae76ViWe3STan69yaF9+fESD2lcQshLHWVu4ovItXxO69bqC5p1nZLvI8NdQB9s9UNaJGlQ5mG947ipdDA0eTIw/A1zEdjWquIsQXXGIVEH0thC5M+W9pZe7IhAVnPJkYCCXN5a32HjN6nsvokEqRS44tGIs7s2LVTvcrHAF+RVmI8L4HUYk4x+67AxSMJKqCg8zrGOgvK9kNMdDrNiUtSWuHFpC8/p5qIQrEo/H+1l/0cAwQ2nKmpWxKcMIuHY44Y6DlkpO48tRuUGBWT0FyHwSKO72Ud+tJUfdaZ4CWNijzZtlRa8+CkmO/EwHYfPZFU/hzjFWH7vnzHRMo+aF9u8qHSAiEkA2HjoNQPEwHsDKOt6hOoK3Ce/+/9boMWDa44I6FrQhdgS7OnNaSzwxWKZMcyHi6LN4WC6sSj0qm2PSOGBTvDs/GWJS6SwEN/ULwpb4LQo9fYjUfSXRwZkynUazlSpvX9e+G2zor8l+YaMxSEomDdLHGcD6YVQPegTaA74H8+V4WvJkFUrjMLGLlvSZQWvi8/QA7yzQ8GPno//5SJHRP/OqKObPCo81s/+6WgLqykYpGAgQZhVDEBPXWgU/WzFZjKUhSFInufPRiMAUULC6T11yL45ZrRoB4DzOyJShKXaAJIBS9wzLYIoCEcJKQW8GVCx4fihqJ6mshBUXSw3wWVj3grrHQlGNGhIDNNzsxQ3M+GWn6ASobIWC+LbYOC6UpahVO13Zs2zOzZC8z7FmA05JhUGyBsF4tsG0drcggIFzgg/kpf3+CnAXKiMgIE8Jk/Mhpkc8DUJEUzDSnWlQFme3d0sHZDrg7LavtsEX3cHwjCYA17pMTfx8Ajw9hHscN67hyo+RJQ4458RmPywXykkVcW688oVUrQhahpPRvTWPnuI0B+SkQu7dCyvLRyFYlC1LG1gRCIvn3rwQeINzZQC2KXq31FaR9UmVV2QeGVqBHjmE+VMd3b1fhCynD0pQNhCG6/WCDbKPyE7NRQzL3BzQAJ0g09aUzcQA6mUp9iZFK6Sbp/YbHjo++7/Wj8S4YNa+ZdqAw1hDrKWFXv9+zaXpf8ZTDSbiqsxnwN/CzK5tPkOr4tRh2kY3Bn9JtalbIOI4b3F7F1vPQMfoDcdxMS8CW9m/NCW/HILTUVWQIPiD0j1A6bo8vsv6P1hCESl2abrSJWDrq5sSzUpwoxaCU9FtJyYH4QFMxDBpkkBR6kn0LMPO+5EJ7Z6bCiRoPedRZ/P0SSdii7ZnPAtVwwHUidcdyspwncz5uq6vvm4IEDbJVLUFCn/LvIHfooUBTkFO130FC7CmmcrKdgDJcid9mvVzsDSibOoXtIf9k6ABle3PmIxejodc4aob0QKS432srrCMndbfD454q52V01G4q913mC5HOsTzWF4h2No1av1VbcUgWAqyoZl+11PoFYnNv2HwAODeNRkHj+8SF1fcvVBu6MrehHAZK1Gm69ICcTKizykHgGFx7QdowTVAsYEF2tVc0Z6wLryz2FI1sc5By2znJAAmINndoJiB4sfPdPrTC8RnkW7KRCwxC6YvXg5ahMlQuMpoCSXjOlBy0Kij+bsCYPbGp8BdCBiLmLSAkEQRaieWo1SYvZIKJGj9Ur/eWHjiB7SOVdqMAVmpBvfRiebsFjger7DC+8kRFGtNrTrnnGD2GAJb8rQCWkUPYHhwXsjNBSkE6lGWUj5QNhK0DMNM2l+kXRZ0KLZaGsFSIdQz/HXDxf3/TE30+DgBKWGWdxElyLccJfEpjsnszECNoDGZpdwdRgCixeg9L4EPhH+RptvRMVRaahu4cySjS3P5wxAUCPkmn+rhyASpmiTaiDeggaIxYBmtLZDDhiWIJaBgzfCsAGUF1Q1SFZYyXDt9skCaxJsxK2Ms65dmdp5WAZyxik/zbrTQk5KmgxCg/f45L0jywebOWUYFJQAJia7XzCV0x89rpp/f3AVWhSPyTanqmik2SkD8A3Ml4NhIGLAjBXtPShwKYfi2eXtrDuKLk4QlSyTw1ftXgwqA2jUuopDl+5tfUWZNwBpEPXghzbBggYCw/dhy0ntds2yeHCDKkF/YxQjNIL/F/37jLPHCKBO9ibwYCmuxImIo0ijV2Wbg3kSN2psoe8IsABv3RNFaF9uMyCtCYtqcD+qNOhwMlfARQUdJ2tUX+MNJqOwIciWalZsmEjt07tfa8ma4cji9sqz+Q9hWfmMoKEbIHPOQORbhQRHIsrTYlnVTNvcq1imqmmPDdVDkJgRcTgB8Sb6epCQVmFZe+jGDiNJQLWnfx+drTKYjm0G8yH0ZAGMWzEJhUEQ4Maimgf/bkvo8PLVBsZl152y5S8+HRDfZIMCbYZ1WDp4yrdchOJw8k6R+/2pHmydK4NIK2PHdFPHtoLmHxRDwLFb7eB+M4zNZcB9NrAgjVyzLM7xyYSY13ykWfIEEd2n5/iYp3ZdrCf7fL+en+sIJu2W7E30MrAgZBD1rAAbZHPgeAMtKCg3NpSpYQUDWJu9bT3V7tOKv+NRiJc8JAKqqgCA/PNRBR7ChpiEulyQApMK1AyqcWnpSOmYh6yLiWkGJ2mklCSPIqN7UypWj3dGi5MvsHQ87MrB4VFgypJaFriaHivwcHIpmyi5LhNqtem4q0n8awM19Qk8BOS0EsqGscuuydYsIGsbT5GHnERUiMpKJl4ON7qjB4fEqlGN/hCky89232UQCiaeWpDYCJINXjT6xl4Gc7DxRCtgV0i1ma4RgWLsNtnEBRQFqZggCLiuyEydmFd7WlogpkCw5G1x4ft2psm3KAREwVwr1Gzl6RT7FDAqpVal34ewVm3VH4qn5mjGj+bYL1NgfLNeXDwtmYSpwzbruDKpTjOdgiIHDVQSb5/zBgSMbHLkxWWgghIh9QTFSDILixVwg0Eg1puooBiHAt7DzwJ7m8i8/i+jHvKf0QDnnHVkVTIqMvIQImOrzCJwhSR7qYB5gSwL6aWL9hERHCZc4G2+JrpgHNB8eCCmcIWIQ6rSdyPCyftXkDlErUkHafHRlkOIjxGbAktz75bnh50dU7YHk+Mz7wwstg6RFZb+TZuSOx1qqP5C66c0mptQmzIC2dlpte7vZrauAMm/7RfBYkGtXWGiaWTtwvAQiq2oD4YixPLXE2khB2FRaNRDTk+9sZ6K74Ia9VntCpN4BhJGJMT4Z5c5FhSepRCRWmBXqx+whVZC4me4saDs2iNqXMuCl6iAZflH8fscC1sTsy4PHeC+XYuqMBMUun5YezKbRKmEPwuK+CLzijPEQgfhahQswBBLfg/GBgBiI4QwAqzJkkyYAWtjzSg2ILgMAgqxYfwERRo3zruBL9WOryUArSD8sQOcD7fvIODJxKFS615KFPsb68USBEPPj1orNzFY2xoTtNBVTyzBhPbhFH0PI5AtlJBl2aSgNPYzxYLw7XTDBDinmVoENwiGzmngrMo8OmnRP0Z0i0Zrln9DDFcnmOoBZjABaQIbPOJYZGqX+RCMlDDbElcjaROLDoualmUIQ88Kekk3iM4OQrADcxi3rJguS4MOIBIgKgXrjd1WkbCdqxJk/4efRIFsavZA7KvvJQqp3Iid5Z0NFc5aiMRzGN3vrpBzaMy4JYde3wr96PjN90AYOIbyp6T4zj8LoE66OGcX1Ef4Z3KoWLAUF4BTg7ug/AbkG5UNQXAMkQezujSHeir2uTThgd3gpyzDrbnEdDRH2W7U6PeRvBX1ZFMP5RM+Zu6UUZZD8hDPHldVWntTCNk7To8IeOW9yn2wx0gmurwqC60AOde4r3ETi5pVMSDK8wxhoGAoEX9NLWHIR33VbrbMveii2jAJlrxwytTHbWNu8Y4N8vCCyZjAX/pcsfwXbLze2+D+u33OGBoJyAAL3jn3RuEcdp5If8O+a4NKWvxOTyDltG0IWoHhwVGe7dKkCWFT++tm+haBCikRUUMrMhYKZJKYoVuv/bsJzO8DwfVIInQq3g3BYypiz8baogH3r3GwqCwFtZnz4xMjAVOYnyOi5HWbFA8n0qz1OjSpHWFzpQOpvkNETZBGpxN8ybhtqV/DMUxd9uFZmBfKXMCn/SqkWJyKPnT6lq+4zBZni6fYRByJn6OK+OgPBGRAJluwGSk4wxjOOzyce/PKODwRlsgrVkdcsEiYrqYdXo0Er2GXi2GQZd0tNJT6c9pK1EEJG1zgDJBoTVuCXGAU8BKTvCO/cEQ1Wjk3Zzuy90JX4m3O5IlxVFhYkSUwuQB2up7jhvkm+bddRQu5F9s0XftGEJ9JSuSk+ZachCbdU45fEqbugzTIUokwoAKvpUQF/CvLbWW5BNQFqFkJg2f30E/48StNe5QwBg8zz3YAJ82FZoXBxXSv4QDooDo79NixyglO9AembuBcx5Re3CwOKTHebOPhkmFC7wNaWtoBhFuV4AkEuJ0J+1pT0tLkvFVZaNzfhs/Kd3+A9YsImlO4XK4vpCo/elHQi/9gkFg07xxnuXLt21unCIpDV+bbRxb7FC6nWYTsMFF8+1LUg4JFjVt3vqbuhHmDKbgQ4e+RGizRiO8ky05LQGMdL2IKLSNar0kNG7lHJMaXr5mLdG3nykgj6vB/KVijd1ARWkFEf3yiUw1v/WaQivVUpIDdSNrrKbjO5NPnxz6qTTGgYg03HgPhDrCFyYZTi3XQw3HXCva39mpLNFtz8AiEhxAJHpWX13gCTAwgm9YTvMeiqetdNQv6IU0hH0G+ZManTqDLPjyrOse7WiiwOJCG+J0pZYULhN8NILulmYYvmVcV2MjAfA39sGKqGdjpiPo86fecg65UPyXDIAOyOkCx5NQsLeD4gGVjTVDwOHWkbbBW0GeNjDkcSOn2Nq4cEssP54t9D749A7M1AIOBl0Fi0sSO5v3P7LCBrM6ZwFY6kp2FX6AcbGUdybnfChHPyu6WlRZ2Fwv9YM0RMI7kISRgR8HpQSJJOyTfXj/6gQKuihPtiUtlCQVPohUgzfezTg8o1b3n9pNZeco1QucaoXe40Fa5JYhqdTspFmxGtW9h5ezLFZs3j/N46f+S2rjYNC2JySXrnSAFhvAkz9a5L3pza8eYKHNoPrvBRESpxYPJdKVUxBE39nJ1chrAFpy4MMkf0qKgYALctGg1DQI1kIymyeS2AJNT4X240d3IFQb/0jQbaHJ2YRK8A+ls6WMhWmpCXYG5jqapGs5/eOJErxi2/2KWVHiPellTgh/fNl/2KYPKb7DUcAg+mCOPQFCiU9Mq/WLcU1xxC8aLePFZZlE+PCLzf7ey46INWRw2kcXySR9FDgByXzfxiNKwDFbUSMMhALPFSedyjEVM5442GZ4hTrsAEvZxIieSHGSgkwFh/nFNdrrFD4tBH4Il7fW6ur4J8Xaz7RW9jgtuPEXQsYk7gcMs2neu3zJwTyUerHKSh1iTBkj2YJh1SSOZL5pLuQbFFAvyO4k1Hxg2h99MTC6cTUkbONQIAnEfGsGkNFWRbuRyyaEZInM5pij73EA9rPIUfU4XoqQpHT9THZkW+oKFLvpyvTBMM69tN1Ydwv1LIEhHsC+ueVG+w+kyCPsvV3erRikcscHjZCkccx6VrBkBRusTDDd8847GA7p2Ucy0y0HdSRN6YIBciYa4vuXcAZbQAuSEmzw+H/AuOx+aH+tBL88H57D0MsqyiZxhOEQkF/8DR1d2hSPMj/sNOa5rxcUnBgH8ictv2J+cb4BA4v3MCShdZ2vtK30vAwkobnEWh7rsSyhmos3WC93Gn9C4nnAd/PjMMtQfyDNZsOPd6XcAsnBE/mRHtHEyJMzJfZFLE9OvQa0i9kUmToJ0ZxknTgdl/XPV8xoh0K7wNHHsnBdvFH3sv52lU7UFteseLG/VanIvcwycVA7+BE1Ulyb20BvwUWZcMTKhaCcmY3ROpvonVMV4N7yBXTL7IDtHzQ4CCcqF66LjF3xUqgErKzolLyCG6Kb7irP/MVTCCwGRxfrPGpMMGvPLgJ881PHMNMIO09T5ig7AzZTX/5PLlwnJLDAPfuHynSGhV4tPqR3gJ4kg4c06c/F1AcjGytKm2Yb5jwMotF7vro4YDLWlnMIpmPg36NgAZsGA0W1spfLSue4xxat0Gdwd0lqDBOgIaMANykwwDKejt5YaNtJYIkrSgu0KjIg0pznY0SCd1qlC6R19g97UrWDoYJGlrvCE05J/5wkjpkre727p5PTRX5FGrSBIfJqhJE/IS876PaHFkx9pGTH3oaY3jJRvLX9Iy3Edoar7cFvJqyUlOhAEiOSAyYgVEGkzHdug+oRHIEOXAExMiTSKU9A6nmRC8mp8iYhwWdP2U/5EkFAdPrZw03YA3gSyNUtMZeh7dDCu8pF5x0VORCTgKp07ehy7NZqKTpIC4UJJ89lnboyAfy5OyXzXtuDRbtAFjZRSyGFTpFrXwkpjSLIQIG3N0Vj4BtzK3wdlkBJrO18MNsgseR4BysJilI0wI6ZahLhBFA0XBmV8d4LUzEcNVb0xbLjLTETYN8OEVqNxkt10W614dd1FlFFVTIgB7/BQQp1sWlNolpIu4ekxUTBV7NmxOFKEBmmN+nA7pvF78/RII5ZHA09OAiE/66MF6HQ+qVEJCHxwymukkNvzqHEh52dULPbVasfQMgTDyBZzx4007YiKdBuUauQOt27Gmy8ISclPmEUCIcuLbkb1mzQSqIa3iE0PJh7UMYQbkpe+hXjTJKdldyt2mVPwywoODGJtBV1lJTgMsuSQBlDMwhEKIfrvsxGQjHPCEfNfMAY2oxvyKcKPUbQySkKG6tj9AQyEW3Q5rpaDJ5Sns9ScLKeizPRbvWYAw4bXkrZdmB7CQopCH8NAmqbuciZChHN8lVGaDbCnmddnqO1PQ4ieMYfcSiBE5zzMz+JV/4eyzrzTEShvqSGzgWimkNxLvUj86iAwcZuIkqdB0VaIB7wncLRmzHkiUQpPBIXbDDLHBlq7vp9xwuC9AiNkIptAYlG7Biyuk8ILdynuUM1cHWJgeB+K3wBP/ineogxkvBNNQ4AkW0hvpBOQGFfeptF2YTR75MexYDUy7Q/9uocGsx41O4IZhViw/2FvAEuGO5g2kyXBUijAggWM08bRhXg5ijgMwDJy40QeY/cQpUDZiIzmvskQpO5G1zyGZA8WByjIQU4jRoFJt56behxtHUUE/om7Rj2psYXGmq3llVOCgGYKNMo4pzwntITtapDqjvQtqpjaJwjHmDzSVGLxMt12gEXAdLi/caHSM3FPRGRf7dB7YC+cD2ho6oL2zGDCkjlf/DFoQVl8GS/56wur3rdV6ggtzZW60MRB3g+U1W8o8cvqIpMkctiGVMzXUFI7FacFLrgtdz4mTEr4aRAaQ2AFQaNeG7GX0yOJgMRYFziXdJf24kg/gBQIZMG/YcPEllRTVNoDYR6oSJ8wQNLuihfw81UpiKPm714bZX1KYjcXJdfclCUOOpvTxr9AAJevTY4HK/G7F3mUc3GOAKqh60zM0v34v+ELyhJZqhkaMA8UMMOU90f8RKEJFj7EqepBVwsRiLbwMo1J2zrE2UYJnsgIAscDmjPjnzI8a719Wxp757wqmSJBjXowhc46QN4RwKIxqEE6E5218OeK7RfcpGjWG1jD7qND+/GTk6M56Ig4yMsU6LUW1EWE+fIYycVV1thldSlbP6ltdC01y3KUfkobkt2q01YYMmxpKRvh1Z48uNKzP/IoRIZ/F6buOymSnW8gICitpJjKWBscSb9JJKaWkvEkqinAJ2kowKoqkqZftRqfRQlLtKoqvTRDi2vg/RrPD/d3a09J8JhGZlEkOM6znTsoMCsuvTmywxTCDhw5dd0GJOHCMPbsj3QLkTE3MInsZsimDQ3HkvthT7U9VA4s6G07sID0FW4SHJmRGwCl+Mu4xf0ezqeXD2PtPDnwMPo86sbwDV+9PWcgFcARUVYm3hrFQrHcgMElFGbSM2A1zUYA3baWfheJp2AINmTJLuoyYD/OwA4a6V0ChBN97E8YtDBerUECv0u0TlxR5yhJCXvJxgyM73Bb6pyq0jTFJDZ4p1Am1SA6sh8nADd1hAcGBMfq4d/UfwnmBqe0Jun1n1LzrgKuZMAnxA3NtCN7Klf4BH+14B7ibBmgt0TGUafVzI4uKlpF7v8NmgNjg90D6QE3tbx8AjSAC+OA1YJvclyPKgT27QpIEgVYpbPYGBsnyCNrGz9XUsCHkW1QAHgL2STZk12QGqmvAB0NFteERkvBIH7INDsNW9KKaAYyDMdBEMzJiWaJHZALqDxQDWRntumSDPcplyFiI1oDpT8wbwe01AHhW6+vAUUBoGhY3CT2tgwehdPqU/4Q7ZLYvhRl/ogOvR9O2+wkkPKW5vCTjD2fHRYXONCoIl4Jh1bZY0ZE1O94mMGn/dFSWBWzQ/VYk+Gezi46RgiDv3EshoTmMSlioUK6MQEN8qeyK6FRninyX8ZPeUWjjbMJChn0n/yJvrq5bh5UcCAcBYSafTFg7p0jDgrXo2QWLb3WpSOET/Hh4oSadBTvyDo10IufLzxiMLAnbZ1vcUmj3w7BQuIXjEZXifwukVxrGa9j+DXfpi12m1RbzYLg9J2wFergEwOxFyD0/JstNK06ZN2XdZSGWxcJODpQHOq4iKqjqkJUmPu1VczL5xTGUfCgLEYyNBCCbMBFT/cUP6pE/mujnHsSDeWxMbhrNilS5MyYR0nJyzanWXBeVcEQrRIhQeJA6Xt4f2eQESNeLwmC10WJVHqwx8SSyrtAAjpGjidcj1E2FYN0LObUcFQhafUKTiGmHWRHGsFCB+HEXgrzJEB5bp0QiF8ZHh11nFX8AboTD0PS4O1LqF8XBks2MpjsQnwKHF6HgaKCVLJtcr0XjqFMRGfKv8tmmykhLRzu+vqQ02+KpJBjaLt9ye1Ab+BbEBhy4EVdIJDrL2naV0o4wU8YZ2Lq04FG1mWCKC+UwkXOoAjneU/xHplMQo2cXUlrVNqJYczgYlaOEczVCs/OCgkyvLmTmdaBJc1iBLuKwmr6qtRnhowngsDxhzKFAi02tf8bmET8BO27ovJKF1plJwm3b0JpMh38+xsrXXg7U74QUM8ZCIMOpXujHntKdaRtsgyEZl5MClMVMMMZkZLNxH9+b8fH6+b8Lev30A9TuEVj9CqAdmwAAHBPbfOBFEATAPZ2CS0OH1Pj/0Q7PFUcC8hDrxESWdfgFRm+7vvWbkEppHB4T/1ApWnlTIqQwjcPl0VgS1yHSmD0OdsCVST8CQVwuiew1Y+g3QGFjNMzwRB2DSsAk26cmA8lp2wIU4p93AUBiUHFGOxOajAqD7Gm6NezNDjYzwLOaSXRBYcWipTSONHjUDXCY4mMI8XoVCR/Rrs/JLKXgEx+qkmeDlFOD1/yTQNDClRuiUyKYCllfMiQiyFkmuTz2vLsBNyRW+xz+5FElFxWB28VjYIGZ0Yd+5wIjkcoMaggxswbT0pCmckRAErbRlIlcOGdBo4djTNO8FAgQ+lT6vPS60BwTRSUAM3ddkEAZiwtEyArrkiDRnS7LJ+2hwbzd2YDQagSgACpsovmjil5wfPuXq3GuH0CyE7FK3M4FgRaFoIkaodORrPx1+JpI9psyNYIFuJogZa0/1AhOWdlHQxdAgbwacsHqPZo8u/ngAH2GmaTdhYnBfSDbBfh8CHq6Bx5bttP2+RdM+MAaYaZ0Y/ADkbNCZuAyAVQa2OcXOeICmDn9Q/eFkDeFQg5MgHEDXq/tVjj+jtd26nhaaolWxs1ixSUgOBwrDhRIGOLyOVk2/Bc0UxvseQCO2pQ2i+Krfhu/WeBovNb5dJxQtJRUDv2mCwYVpNl2efQM9xQHnK0JwLYt/U0Wf+phiA4uw8G91slC832pmOTCAoZXohg1fewCZqLBhkOUBofBWpMPsqg7XEXgPfAlDo2U5WXjtFdS87PIqClCK5nW6adCeXPkUiTGx0emOIDQqw1yFYGHEVx20xKjJVYe0O8iLmnQr3FA9nSIQilUKtJ4ZAdcTm7+ExseJauyqo30hs+1qSW211A1SFAOUgDlCGq7eTIcMAeyZkV1SQJ4j/e1Smbq4HcjqgFbLAGLyKxlMDMgZavK5NAYH19Olz3la/QCTiVelFnU6O/GCvykqS/wZJDhKN9gBtSOp/1SP5VRgJcoVj+kmf2wBgv4gjrgARBWiURYx8xENV3bEVUAAWWD3dYDKAIWk5opaCFCMR5ZjJExiCAw7gYiSZ2rkyTce4eNMY3lfGn+8p6+vBckGlKEXnA6Eota69OxDO9oOsJoy28BXOR0UoXNRaJD5ceKdlWMJlOFzDdZNpc05tkMGQtqeNF2lttZqNco1VtwXgRstLSQ6tSPChgqtGV5h2DcDReIQadaNRR6AsAYKL5gSFsCJMgfsaZ7DpKh8mg8Wz8V7H+gDnLuMxaWEIUPevIbClgap4dqmVWSrPgVYCzAoZHIa5z2Ocx1D/GvDOEqMOKLrMefWIbSWHZ6jbgA8qVBhYNHpx0P+jAgN5TB3haSifDcApp6yymEi6Ij/GsEpDYUgcHATJUYDUAmC1SCkJ4cuZXSAP2DEpQsGUjQmKJfJOvlC2x/pChkOyLW7KEoMYc5FDC4v2FGqSoRWiLsbPCiyg1U5yiHZVm1XLkHMMZL11/yxyw0UnGig3MFdZklN5FI/qiT65T+jOXOdO7XbgWurOAZR6Cv9uu1cm5LjkXX4xi6mWn5r5NjBS0gTliHhMZI2WNqSiSphEtiCAwnafS11JhseDGHYQ5+bqWiAYiAv6Jsf79/VUs4cIl+n6+WOjcgB/2l5TreoAV2717JzZbQIR0W1cl/dEqCy5kJ3ZSIHuU0vBoHooEpiHeQWVkkkOqRX27eD1FWw4BfO9CJDdKoSogQi3hAAwsPRFrN5RbX7bqLdBJ9JYMohWrgJKHSjVl1sy2xAG0E3sNyO0oCbSGOxCNBRRXTXenYKuwAoDLfnDcQaCwehUOIDiHAu5m5hMpKeKM4sIo3vxACakIxKoH2YWF2QM84e6F5C5hJU4g8uxuFOlAYnqtwxmHyNEawLW/PhoawJDrGAP0JYWHgAVUByo/bGdiv2T2EMg8gsS14/rAdzlOYazFE7w4OzxeKiWdm3nSOnQRRKXSlVo8HEAbBfyJMKqoq+SCcTSx5NDtbFwNlh8VhjGGDu7JG5/TAGAvniQSSUog0pNzTim8Owc6QTuSKSTXlQqwV3eiEnklS3LeSXYPXGK2VgeZBqNcHG6tZHvA3vTINhV0ELuQdp3t1y9+ogD8Kk/W7QoRN1UWPqM4+xdygkFDPLoTaumKReKiLWoPHOfY54m3qPx4c+4pgY3MRKKbljG8w4wvz8pxk3AqKsy4GMAkAtmRjRMsCxbb4Q2Ds0Ia9ci8cMT6DmsJG00XaHCIS+o3F8YVVeikw13w+OEDaCYYhC0ZE54kA4jpjruBr5STWeqQG6M74HHL6TZ3lXrd99ZX++7LhNatQaZosuxEf5yRA15S9gPeHskBIq3Gcw81AGb9/O53DYi/5CsQ51EmEh8Rkg4vOciClpy4d04eYsfr6fyQkBmtD+P8sNh6e+XYHJXT/lkXxT4KXU5F2sGxYyzfniMMQkb9OjDN2C8tRRgTyL7GwozH14PrEUZc6oz05Emne3Ts5EG7WolDmU8OB1LDG3VrpQxp+pT0KYV5dGtknU64JhabdqcVQbGZiAxQAnvN1u70y1AnmvOSPgLI6uB4AuDGhmAu3ATkJSw7OtS/2ToPjqkaq62/7WFG8advGlRRqxB9diP07JrXowKR9tpRa+jGJ91zxNTT1h8I2PcSfoUPtd7NejVoH03EUcqSBuFZPkMZhegHyo2ZAITovmm3zAIdGFWxoNNORiMRShgwdYwFzkPw5PA4a5MIIQpmq+nsp3YMuXt/GkXxLx/P6+ZJS0lFyz4MunC3eWSGE8xlCQrKvhKUPXr0hjpAN9ZK4PfEDrPMfMbGNWcHDzjA7ngMxTPnT7GMHar+gMQQ3NwHCv4zH4BIMYvzsdiERi6gebRmerTsVwZJTRsL8dkZgxgRxmpbgRcud+YlCIRpPwHShlUSwuipZnx9QCsEWziVazdDeKSYU5CF7UVPAhLer3CgJOQXl/zh575R5rsrmRnKAzq4POFdgbYBuEviM4+LVC15ssLNFghbTtHWerS1hDt5s4qkLUha/qpZXhWh1C6lTQAqCNQnaDjS7UGFBC6wTu8yFnKJnExCnAs3Ok9yj5KpfZESQ4lTy5pTGTnkAUpxI+yjEldJfSo4y0QhG4i4IwkRFGcjWY8+EzgYYJUK7BXQksLxAww/YYWBMhJILB9e8ePEJ4OP7z+4/wOQDl64iOYDp26DaONPxpKtBxq/aTzRGarm3VkPYTLJKx6Z/Mw2YbBGseJhPMwhhNswrIkyvV2BYzrvZbxLpKwcWJhYmFtVZ+lPEq91FzVp1HlQY1bZVLqeNR9SAUn6n0E28k/UuGkNpP1DBI5ch/EehZfjUQ9aE41NhETExoPT2gGQz0IhWJbEOvTQ4wgcXCHHFBhewYUiFHuhRSAUVmEHeCRQHQkXGFwkAgyzREJCVN7TRnTon36Zw3tPhx4EALwNdwDv+J41YSP4B2CQqz0EFgARZ4ESgBHQgROwAVn9GTI+HYexTUevLUeta4/DqKrbMVS+Yqb8hUwYCrlgKtmAq1YCrFgKrd4qpXiqZcKn1oqdWipjYKpWwVPVYqW6xUpVipKqFR3QKjagVEtAqHpxUMTitsnFaJOKx2cVhswq35RVpyiq9lFVNIKnOQVMkgqtYxVNxiqQjFS7GKlSIVIsQqPIhUWwioigFQ++KkN8VHr49HDw9Ebo9EDo9DTo9Crg9BDg9/Wx7gWx7YWwlobYrOGxWPNisAaAHEyALpkAVDIAeWAArsABVXACYuAD5cAF6wAKFQAQqgAbVAAsoAAlQAUaYAfkwAvogBWQACOgAD9AAHSAAKT4GUdMiOvFngBTwCn2AZ7Dv6B6k/90B8+yRnkV144AIBoAMTQATGgAjNAA4YABgwABZgB/mQCwyAVlwCguASlwCEuAQFwB4uAMlwBYuAJlQAUVAAhUD2KgdpUDaJgaRMDFJgX5MC1JgWJEAokQCWRAHxEAWkQBMRADpEAMkQAYROAEecC484DRpwBDTnwNOdw05tjTmiNOYwtswhYFwLA7BYG4LA2BYGOLAwRYFuLAsxYFQJAohIEyJAMwkAwiQC0JAJgkAeiQBkJAFokAPCQA0JABwcD4Dgc4cDdDgaYcDIDgYgUC6CgWgUClCgUYUAVBQBOFAEYMALgwAgDA9QYAdIn8AZzeBB2L5EcWrenUT1KXienEsuJJ7x5U8XlTjc1NVzUyXFTGb1LlpUtWlTDIjqwE4LsagowoCi2gJLKAkpoBgJQNpAIhNqaEoneI6kiiqQ6Go/n6j0cS+a2gEU8gIHJ+BwfgZX4GL+Bd/gW34FZ+BS/gUH4FN6BTegTvoEv6BJegRnYEF2A79gOvYDl2BdEjCkqkGtwXp0LNToIskOTXzh/F062yJ7AAAAEDAWAAABWhJ+KPEIJgBFxMVP7w2QJBGHASQnOBKXKFIdUK4igKA9IEaYJg);src:url(data:application/vnd.ms-fontobject;base64,n04AAEFNAAACAAIABAAAAAAABQAAAAAAAAABAJABAAAEAExQAAAAAAAAAAIAAAAAAAAAAAEAAAAAAAAAJxJ/LAAAAAAAAAAAAAAAAAAAAAAAACgARwBMAFkAUABIAEkAQwBPAE4AUwAgAEgAYQBsAGYAbABpAG4AZwBzAAAADgBSAGUAZwB1AGwAYQByAAAAeABWAGUAcgBzAGkAbwBuACAAMQAuADAAMAA5ADsAUABTACAAMAAwADEALgAwADAAOQA7AGgAbwB0AGMAbwBuAHYAIAAxAC4AMAAuADcAMAA7AG0AYQBrAGUAbwB0AGYALgBsAGkAYgAyAC4ANQAuADUAOAAzADIAOQAAADgARwBMAFkAUABIAEkAQwBPAE4AUwAgAEgAYQBsAGYAbABpAG4AZwBzACAAUgBlAGcAdQBsAGEAcgAAAAAAQlNHUAAAAAAAAAAAAAAAAAAAAAADAKncAE0TAE0ZAEbuFM3pjM/SEdmjKHUbyow8ATBE40IvWA3vTu8LiABDQ+pexwUMcm1SMnNryctQSiI1K5ZnbOlXKmnVV5YvRe6RnNMFNCOs1KNVpn6yZhCJkRtVRNzEufeIq7HgSrcx4S8h/v4vnrrKc6oCNxmSk2uKlZQHBii6iKFoH0746ThvkO1kJHlxjrkxs+LWORaDQBEtiYJIR5IB9Bi1UyL4Rmr0BNigNkMzlKQmnofBHviqVzUxwdMb3NdCn69hy+pRYVKGVS/1tnsqv4LL7wCCPZZAZPT4aCShHjHJVNuXbmMrY5LeQaGnvAkXlVrJgKRAUdFjrWEah9XebPeQMj7KS7DIBAFt8ycgC5PLGUOHSE3ErGZCiViNLL5ZARfywnCoZaKQCu6NuFX42AEeKtKUGnr/Cm2Cy8tpFhBPMW5Fxi4Qm4TkDWh4IWFDClhU2hRWosUWqcKLlgyXB+lSHaWaHiWlBAR8SeSgSPCQxdVQgzUixWKSTrIQEbU94viDctkvX+VSjJuUmV8L4CXShI11esnp0pjWNZIyxKHS4wVQ2ime1P4RnhvGw0aDN1OLAXGERsB7buFpFGGBAre4QEQR0HOIO5oYH305G+KspT/FupEGGafCCwxSe6ZUa+073rXHnNdVXE6eWvibUS27XtRzkH838mYLMBmYysZTM0EM3A1fbpCBYFccN1B/EnCYu/TgCGmr7bMh8GfYL+BfcLvB0gRagC09w9elfldaIy/hNCBLRgBgtCC7jAF63wLSMAfbfAlEggYU0bUA7ACCJmTDpEmJtI78w4/BO7dN7JR7J7ZvbYaUbaILSQsRBiF3HGk5fEg6p9unwLvn98r+vnsV+372uf1xBLq4qU/45fTuqaAP+pssmCCCTF0mhEow8ZXZOS8D7Q85JsxZ+Azok7B7O/f6J8AzYBySZQB/QHYUSA+EeQhEWiS6AIQzgcsDiER4MjgMBAWDV4AgQ3g1eBgIdweCQmCjJEMkJ+PKRWyFHHmg1Wi/6xzUgA0LREoKJChwnQa9B+5RQZRB3IlBlkAnxyQNaANwHMowzlYSMCBgnbpzvqpl0iTJNCQidDI9ZrSYNIRBhHtUa5YHMHxyGEik9hDE0AKj72AbTCaxtHPUaKZdAZSnQTyjGqGLsmBStCejApUhg4uBMU6mATujEl+KdDPbI6Ag4vLr+hjY6lbjBeoLKnZl0UZgRX8gTySOeynZVz1wOq7e1hFGYIq+MhrGxDLak0PrwYzSXtcuyhXEhwOYofiW+EcI/jw8P6IY6ed+etAbuqKp5QIapT77LnAe505lMuqL79a0ut4rWexzFttsOsLDy7zvtQzcq3U1qabe7tB0wHWVXji+zDbo8x8HyIRUbXnwUcklFv51fvTymiV+MXLSmGH9d9+aXpD5X6lao41anWGig7IwIdnoBY2ht/pO9mClLo4NdXHAsefqWUKlXJkbqPOFhMoR4aiA1BXqhRNbB2Xwi+7u/jpAoOpKJ0UX24EsrzMfHXViakCNcKjBxuQX8BO0ZqjJ3xXzf+61t2VXOSgJ8xu65QKgtN6FibPmPYsXbJRHHqbgATcSZxBqGiDiU4NNNsYBsKD0MIP/OfKnlk/Lkaid/O2NbKeuQrwOB2Gq3YHyr6ALgzym5wIBnsdC1ZkoBFZSQXChZvlesPqvK2c5oHHT3Q65jYpNxnQcGF0EHbvYqoFw60WNlXIHQF2HQB7zD6lWjZ9rVqUKBXUT6hrkZOle0RFYII0V5ZYGl1JAP0Ud1fZZMvSomBzJ710j4Me8mjQDwEre5Uv2wQfk1ifDwb5ksuJQQ3xt423lbuQjvoIQByQrNDh1JxGFkOdlJvu/gFtuW0wR4cgd+ZKesSV7QkNE2kw6AV4hoIuC02LGmTomyf8PiO6CZzOTLTPQ+HW06H+tx+bQ8LmDYg1pTFrp2oJXgkZTyeRJZM0C8aE2LpFrNVDuhARsN543/FV6klQ6Tv1OoZGXLv0igKrl/CmJxRmX7JJbJ998VSIPQRyDBICzl4JJlYHbdql30NvYcOuZ7a10uWRrgoieOdgIm4rlq6vNOQBuqESLbXG5lzdJGHw2m0sDYmODXbYGTfSTGRKpssTO95fothJCjUGQgEL4yKoGAF/0SrpUDNn8CBgBcSDQByAeNkCXp4S4Ro2Xh4OeaGRgR66PVOsU8bc6TR5/xTcn4IVMLOkXSWiXxkZQCbvKfmoAvQaKjO3EDKwkwqHChCDEM5loQRPd5ACBki1TjF772oaQhQbQ5C0lcWXPFOzrfsDGUXGrpxasbG4iab6eByaQkQfm0VFlP0ZsDkvvqCL6QXMUwCjdMx1ZOyKhTJ7a1GWAdOUcJ8RSejxNVyGs31OKMyRyBVoZFjqIkmKlLQ5eHMeEL4MkUf23cQ/1SgRCJ1dk4UdBT7OoyuNgLs0oCd8RnrEIb6QdMxT2QjD4zMrJkfgx5aDMcA4orsTtKCqWb/Veyceqa5OGSmB28YwH4rFbkQaLoUN8OQQYnD3w2eXpI4ScQfbCUZiJ4yMOIKLyyTc7BQ4uXUw6Ee6/xM+4Y67ngNBknxIPwuppgIhFcwJyr6EIj+LzNj/mfR2vhhRlx0BILZoAYruF0caWQ7YxO66UmeguDREAFHYuC7HJviRgVO6ruJH59h/C/PkgSle8xNzZJULLWq9JMDTE2fjGE146a1Us6PZDGYle6ldWRqn/pdpgHKNGrGIdkRK+KPETT9nKT6kLyDI8xd9A1FgWmXWRAIHwZ37WyZHOVyCadJEmMVz0MadMjDrPho+EIochkVC2xgGiwwsQ6DMv2P7UXqT4x7CdcYGId2BJQQa85EQKmCmwcRejQ9Bm4oATENFPkxPXILHpMPUyWTI5rjNOsIlmEeMbcOCEqInpXACYQ9DDxmFo9vcmsDblcMtg4tqBerNngkIKaFJmrQAPnq1dEzsMXcwjcHdfdCibcAxxA+q/j9m3LM/O7WJka4tSidVCjsvo2lQ/2ewyoYyXwAYyr2PlRoR5MpgVmSUIrM3PQxXPbgjBOaDQFIyFMJvx3Pc5RSYj12ySVF9fwFPQu2e2KWVoL9q3Ayv3IzpGHUdvdPdrNUdicjsTQ2ISy7QU3DrEytIjvbzJnAkmANXjAFERA0MUoPF3/5KFmW14bBNOhwircYgMqoDpUMcDtCmBE82QM2YtdjVLB4kBuKho/bcwQdeboqfQartuU3CsCf+cXkgYAqp/0Ee3RorAZt0AvvOCSI4JICIlGlsV0bsSid/NIEALAAzb6HAgyWHBps6xAOwkJIGcB82CxRQq4sJf3FzA70A+TRqcqjEMETCoez3mkPcpnoALs0ugJY8kQwrC+JE5ik3w9rzrvDRjAQnqgEVvdGrNwlanR0SOKWzxOJOvLJhcd8Cl4AshACUkv9czdMkJCVQSQhp6kp7StAlpVRpK0t0SW6LHeBJnE2QchB5Ccu8kxRghZXGIgZIiSj7gEKMJDClcnX6hgoqJMwiQDigIXg3ioFLCgDgjPtYHYpsF5EiA4kcnN18MZtOrY866dEQAb0FB34OGKHGZQjwW/WDHA60cYFaI/PjpzquUqdaYGcIq+mLez3WLFFCtNBN2QJcrlcoELgiPku5R5dSlJFaCEqEZle1AQzAKC+1SotMcBNyQUFuRHRF6OlimSBgjZeTBCwLyc6A+P/oFRchXTz5ADknYJHxzrJ5pGuIKRQISU6WyKTBBjD8WozmVYWIsto1AS5rxzKlvJu4E/vwOiKxRtCWsDM+eTHUrmwrCK5BIfMzGkD+0Fk5LzBs0jMYXktNDblB06LMNJ09U8pzSLmo14MS0OMjcdrZ31pyQqxJJpRImlSvfYAK8inkYU52QY2FPEVsjoWewpwhRp5yAuNpkqhdb7ku9Seefl2D0B8SMTFD90xi4CSOwwZy9IKkpMtI3FmFUg3/kFutpQGNc3pCR7gvC4sgwbupDu3DyEN+W6YGLNM21jpB49irxy9BSlHrVDlnihGKHwPrbVFtc+h1rVQKZduxIyojccZIIcOCmhEnC7UkY68WXKQgLi2JCDQkQWJRQuk60hZp0D3rtCTINSeY9Ej2kIKYfGxwOs4j9qMM7fYZiipzgcf7TamnehqdhsiMiCawXnz4xAbyCkLAx5EGbo3Ax1u3dUIKnTxIaxwQTHehPl3V491H0+bC5zgpGz7Io+mjdhKlPJ01EeMpM7UsRJMi1nGjmJg35i6bQBAAxjO/ENJubU2mg3ONySEoWklCwdABETcs7ck3jgiuU9pcKKpbgn+3YlzV1FzIkB6pmEDOSSyDfPPlQskznctFji0kpgZjW5RZe6x9kYT4KJcXg0bNiCyif+pZACCyRMmYsfiKmN9tSO65F0R2OO6ytlEhY5Sj6uRKfFxw0ijJaAx/k3QgnAFSq27/2i4GEBA+UvTJKK/9eISNvG46Em5RZfjTYLdeD8kdXHyrwId/DQZUaMCY4gGbke2C8vfjgV/Y9kkRQOJIn/xM9INZSpiBnqX0Q9GlQPpPKAyO5y+W5NMPSRdBCUlmuxl40ZfMCnf2Cp044uI9WLFtCi4YVxKjuRCOBWIb4XbIsGdbo4qtMQnNOQz4XDSui7W/N6l54qOynCqD3DpWQ+mpD7C40D8BZEWGJX3tlAaZBMj1yjvDYKwCJBa201u6nBKE5UE+7QSEhCwrXfbRZylAaAkplhBWX50dumrElePyNMRYUrC99UmcSSNgImhFhDI4BXjMtiqkgizUGCrZ8iwFxU6fQ8GEHCFdLewwxYWxgScAYMdMLmcZR6b7rZl95eQVDGVoUKcRMM1ixXQtXNkBETZkVVPg8LoSrdetHzkuM7DjZRHP02tCxA1fmkXKF3VzfN1pc1cv/8lbTIkkYpqKM9VOhp65ktYk+Q46myFWBapDfyWUCnsnI00QTBQmuFjMZTcd0V2NQ768Fhpby04k2IzNR1wKabuGJqYWwSly6ocMFGTeeI+ejsWDYgEvr66QgqdcIbFYDNgsm0x9UHY6SCd5+7tpsLpKdvhahIDyYmEJQCqMqtCF6UlrE5GXRmbu+vtm3BFSxI6ND6UxIE7GsGMgWqghXxSnaRJuGFveTcK5ZVSPJyjUxe1dKgI6kNF7EZhIZs8y8FVqwEfbM0Xk2ltORVDKZZM40SD3qQoQe0orJEKwPfZwm3YPqwixhUMOndis6MhbmfvLBKjC8sKKIZKbJk8L11oNkCQzCgvjhyyEiQSuJcgCQSG4Mocfgc0Hkwcjal1UNgP0CBPikYqBIk9tONv4kLtBswH07vUCjEaHiFGlLf8MgXKzSgjp2HolRRccAOh0ILHz9qlGgIFkwAnzHJRjWFhlA7ROwINyB5HFj59PRZHFor6voq7l23EPNRwdWhgawqbivLSjRA4htEYUFkjESu67icTg5S0aW1sOkCiIysfJ9UnIWevOOLGpepcBxy1wEhd2WI3AZg7sr9WBmHWyasxMcvY/iOmsLtHSWNUWEGk9hScMPShasUA1AcHOtRZlqMeQ0OzYS9vQvYUjOLrzP07BUAFikcJNMi7gIxEw4pL1G54TcmmmoAQ5s7TGWErJZ2Io4yQ0ljRYhL8H5e62oDtLF8aDpnIvZ5R3GWJyAugdiiJW9hQAVTsnCBHhwu7rkBlBX6r3b7ejEY0k5GGeyKv66v+6dg7mcJTrWHbtMywbedYqCQ0FPwoytmSWsL8WTtChZCKKzEF7vP6De4x2BJkkniMgSdWhbeBSLtJZR9CTHetK1xb34AYIJ37OegYIoPVbXgJ/qDQK+bfCtxQRVKQu77WzOoM6SGL7MaZwCGJVk46aImai9fmam+WpHG+0BtQPWUgZ7RIAlPq6lkECUhZQ2gqWkMYKcYMYaIc4gYCDFHYa2d1nzp3+J1eCBay8IYZ0wQRKGAqvCuZ/UgbQPyllosq+XtfKIZOzmeJqRazpmmoP/76YfkjzV2NlXTDSBYB04SVlNQsFTbGPk1t/I4Jktu0XSgifO2ozFOiwd/0SssJDn0dn4xqk4GDTTKX73/wQyBLdqgJ+Wx6AQaba3BA9CKEzjtQYIfAsiYamapq80LAamYjinlKXUkxdpIDk0puXUEYzSalfRibAeDAKpNiqQ0FTwoxuGYzRnisyTotdVTclis1LHRQCy/qqL8oUaQzWRxilq5Mi0IJGtMY02cGLD69vGjkj3p6pGePKI8bkBv5evq8SjjyU04vJR2cQXQwSJyoinDsUJHCQ50jrFTT7yRdbdYQMB3MYCb6uBzJ9ewhXYPAIZSXfeEQBZZ3GPN3Nbhh/wkvAJLXnQMdi5NYYZ5GHE400GS5rXkOZSQsdZgIbzRnF9ueLnsfQ47wHAsirITnTlkCcuWWIUhJSbpM3wWhXNHvt2xUsKKMpdBSbJnBMcihkoDqAd1Zml/R4yrzow1Q2A5G+kzo/RhRxQS2lCSDRV8LlYLBOOoo1bF4jwJAwKMK1tWLHlu9i0j4Ig8qVm6wE1DxXwAwQwsaBWUg2pOOol2dHxyt6npwJEdLDDVYyRc2D0HbcbLUJQj8gPevQBUBOUHXPrsAPBERICpnYESeu2OHotpXQxRGlCCtLdIsu23MhZVEoJg8Qumj/UMMc34IBqTKLDTp76WzL/dMjCxK7MjhiGjeYAC/kj/jY/Rde7hpSM1xChrog6yZ7OWTuD56xBJnGFE+pT2ElSyCnJcwVzCjkqeNLfMEJqKW0G7OFIp0G+9mh50I9o8k1tpCY0xYqFNIALgIfc2me4n1bmJnRZ89oepgLPT0NTMLNZsvSCZAc3TXaNB07vail36/dBySis4m9/DR8izaLJW6bWCkVgm5T+ius3ZXq4xI+GnbveLbdRwF2mNtsrE0JjYc1AXknCOrLSu7Te/r4dPYMCl5qtiHNTn+TPbh1jCBHH+dMJNhwNgs3nT+OhQoQ0vYif56BMG6WowAcHR3DjQolxLzyVekHj00PBAaW7IIAF1EF+uRIWyXjQMAs2chdpaKPNaB+kSezYt0+CA04sOg5vx8Fr7Ofa9sUv87h7SLAUFSzbetCCZ9pmyLt6l6/TzoA1/ZBG9bIUVHLAbi/kdBFgYGyGwRQGBpkqCEg2ah9UD6EedEcEL3j4y0BQQCiExEnocA3SZboh+epgd3YsOkHskZwPuQ5OoyA0fTA5AXrHcUOQF+zkJHIA7PwCDk1gGVmGUZSSoPhNf+Tklauz98QofOlCIQ/tCD4dosHYPqtPCXB3agggQQIqQJsSkB+qn0rkQ1toJjON/OtCIB9RYv3PqRA4C4U68ZMlZn6BdgEvi2ziU+TQ6NIw3ej+AtDwMGEZk7e2IjxUWKdAxyaw9OCwSmeADTPPleyk6UhGDNXQb++W6Uk4q6F7/rg6WVTo82IoCxSIsFDrav4EPHphD3u4hR53WKVvYZUwNCCeM4PMBWzK+EfIthZOkuAwPo5C5jgoZgn6dUdvx5rIDmd58cXXdKNfw3l+wM2UjgrDJeQHhbD7HW2QDoZMCujgIUkk5Fg8VCsdyjOtnGRx8wgKRPZN5dR0zPUyfGZFVihbFRniXZFOZGKPnEQzU3AnD1KfR6weHW2XS6KbPJxUkOTZsAB9vTVp3Le1F8q5l+DMcLiIq78jxAImD2pGFw0VHfRatScGlK6SMu8leTmhUSMy8Uhdd6xBiH3Gdman4tjQGLboJfqz6fL2WKHTmrfsKZRYX6BTDjDldKMosaSTLdQS7oDisJNqAUhw1PfTlnacCO8vl8706Km1FROgLDmudzxg+EWTiArtHgLsRrAXYWdB0NmToNCJdKm0KWycZQqb+Mw76Qy29iQ5up/X7oyw8QZ75kP5F6iJAJz6KCmqxz8fEa/xnsMYcIO/vEkGRuMckhr4rIeLrKaXnmIzlNLxbFspOphkcnJdnz/Chp/Vlpj2P7jJQmQRwGnltkTV5dbF9fE3/fxoSqTROgq9wFUlbuYzYcasE0ouzBo+dDCDzxKAfhbAZYxQiHrLzV2iVexnDX/QnT1fsT/xuhu1ui5qIytgbGmRoQkeQooO8eJNNZsf0iALur8QxZFH0nCMnjerYQqG1pIfjyVZWxhVRznmmfLG00BcBWJE6hzQWRyFknuJnXuk8A5FRDCulwrWASSNoBtR+CtGdkPwYN2o7DOw/VGlCZPusRBFXODQdUM5zeHDIVuAJBLqbO/f9Qua+pDqEPk230Sob9lEZ8BHiCorjVghuI0lI4JDgHGRDD/prQ84B1pVGkIpVUAHCG+iz3Bn3qm2AVrYcYWhock4jso5+J7HfHVj4WMIQdGctq3psBCVVzupQOEioBGA2Bk+UILT7+VoX5mdxxA5fS42gISQVi/HTzrgMxu0fY6hE1ocUwwbsbWcezrY2n6S8/6cxXkOH4prpmPuFoikTzY7T85C4T2XYlbxLglSv2uLCgFv8Quk/wdesUdWPeHYIH0R729JIisN9Apdd4eB10aqwXrPt+Su9mA8k8n1sjMwnfsfF2j3jMUzXepSHmZ/BfqXvzgUNQQWOXO8YEuFBh4QTYCkOAPxywpYu1VxiDyJmKVcmJPGWk/gc3Pov02StyYDahwmzw3E1gYC9wkupyWfDqDSUMpCTH5e5N8B//lHiMuIkTNw4USHrJU67bjXGqNav6PBuQSoqTxc8avHoGmvqNtXzIaoyMIQIiiUHIM64cXieouplhNYln7qgc4wBVAYR104kO+CvKqsg4yIUlFNThVUAKZxZt1XA34h3TCUUiXVkZ0w8Hh2R0Z5L0b4LZvPd/p1gi/07h8qfwHrByuSxglc9cI4QIg2oqvC/qm0i7tjPLTgDhoWTAKDO2ONW5oe+/eKB9vZB8K6C25yCZ9RFVMnb6NRdRjyVK57CHHSkJBfnM2/j4ODUwRkqrtBBCrDsDpt8jhZdXoy/1BCqw3sSGhgGGy0a5Jw6BP/TExoCmNFYjZl248A0osgPyGEmRA+fAsqPVaNAfytu0vuQJ7rk3J4kTDTR2AlCHJ5cls26opZM4w3jMULh2YXKpcqGBtuleAlOZnaZGbD6DHzMd6i2oFeJ8z9XYmalg1Szd/ocZDc1C7Y6vcALJz2lYnTXiWEr2wawtoR4g3jvWUU2Ngjd1cewtFzEvM1NiHZPeLlIXFbBPawxNgMwwAlyNSuGF3zizVeOoC9bag1qRAQKQE/EZBWC2J8mnXAN2aTBboZ7HewnObE8CwROudZHmUM5oZ/Ugd/JZQK8lvAm43uDRAbyW8gZ+ZGq0EVerVGUKUSm/Idn8AQHdR4m7bue88WBwft9mSCeMOt1ncBwziOmJYI2ZR7ewNMPiCugmSsE4EyQ+QATJG6qORMGd4snEzc6B4shPIo4G1T7PgSm8PY5eUkPdF8JZ0VBtadbHXoJgnEhZQaODPj2gpODKJY5Yp4DOsLBFxWbvXN755KWylJm+oOd4zEL9Hpubuy2gyyfxh8oEfFutnYWdfB8PdESLWYvSqbElP9qo3u6KTmkhoacDauMNNjj0oy40DFV7Ql0aZj77xfGl7TJNHnIwgqOkenruYYNo6h724+zUQ7+vkCpZB+pGA562hYQiDxHVWOq0oDQl/QsoiY+cuI7iWq/ZIBtHcXJ7kks+h2fCNUPA82BzjnqktNts+RLdk1VSu+tqEn7QZCCsvEqk6FkfiOYkrsw092J8jsfIuEKypNjLxrKA9kiA19mxBD2suxQKCzwXGws7kEJvlhUiV9tArLIdZW0IORcxEzdzKmjtFhsjKy/44XYXdI5noQoRcvjZ1RMPACRqYg2V1+OwOepcOknRLLFdYgTkT5UApt/JhLM3jeFYprZV+Zow2g8fP+U68hkKFWJj2yBbKqsrp25xkZX1DAjUw52IMYWaOhab8Kp05VrdNftqwRrymWF4OQSjbdfzmRZirK8FMJELEgER2PHjEAN9pGfLhCUiTJFbd5LBkOBMaxLr/A1SY9dXFz4RjzoU9ExfJCmx/I9FKEGT3n2cmzl2X42L3Jh+AbQq6sA+Ss1kitoa4TAYgKHaoybHUDJ51oETdeI/9ThSmjWGkyLi5QAGWhL0BG1UsTyRGRJOldKBrYJeB8ljLJHfATWTEQBXBDnQexOHTB+Un44zExFE4vLytcu5NwpWrUxO/0ZICUGM7hGABXym0V6ZvDST0E370St9MIWQOTWngeoQHUTdCJUP04spMBMS8LSker9cReVQkULFDIZDFPrhTzBl6sed9wcZQTbL+BDqMyaN3RJPh/anbx+Iv+qgQdAa3M9Z5JmvYlh4qop+Ho1F1W5gbOE9YKLgAnWytXElU4G8GtW47lhgFE6gaSs+gs37sFvi0PPVvA5dnCBgILTwoKd/+DoL9F6inlM7H4rOTzD79KJgKlZO/Zgt22UsKhrAaXU5ZcLrAglTVKJEmNJvORGN1vqrcfSMizfpsgbIe9zno+gBoKVXgIL/VI8dB1O5o/R3Suez/gD7M781ShjKpIIORM/nxG+jjhhgPwsn2IoXsPGPqYHXA63zJ07M2GPEykQwJBYLK808qYxuIew4frk52nhCsnCYmXiR6CuapvE1IwRB4/QftDbEn+AucIr1oxrLabRj9q4ae0+fXkHnteAJwXRbVkR0mctVSwEbqhJiMSZUp9DNbEDMmjX22m3ABpkrPQQTP3S1sib5pD2VRKRd+eNAjLYyT0hGrdjWJZy24OYXRoWQAIhGBZRxuBFMjjZQhpgrWo8SiFYbojcHO8V5DyscJpLTHyx9Fimassyo5U6WNtquUMYgccaHY5amgR3PQzq3ToNM5ABnoB9kuxsebqmYZm0R9qxJbFXCQ1UPyFIbxoUraTJFDpCk0Wk9GaYJKz/6oHwEP0Q14lMtlddQsOAU9zlYdMVHiT7RQP3XCmWYDcHCGbVRHGnHuwzScA0BaSBOGkz3lM8CArjrBsyEoV6Ys4qgDK3ykQQPZ3hCRGNXQTNNXbEb6tDiTDLKOyMzRhCFT+mAUmiYbV3YQVqFVp9dorv+TsLeCykS2b5yyu8AV7IS9cxcL8z4Kfwp+xJyYLv1OsxQCZwTB4a8BZ/5EdxTBJthApqyfd9u3ifr/WILTqq5VqgwMT9SOxbSGWLQJUUWCVi4k9tho9nEsbUh7U6NUsLmkYFXOhZ0kmamaJLRNJzSj/qn4Mso6zb6iLLBXoaZ6AqeWCjHQm2lztnejYYM2eubnpBdKVLORZhudH3JF1waBJKA9+W8EhMj3Kzf0L4vi4k6RoHh3Z5YgmSZmk6ns4fjScjAoL8GoOECgqgYEBYUGFVO4FUv4/YtowhEmTs0vrvlD/CrisnoBNDAcUi/teY7OctFlmARQzjOItrrlKuPO6E2Ox93L4O/4DcgV/dZ7qR3VBwVQxP1GCieA4RIpweYJ5FoYrHxqRBdJjnqbsikA2Ictbb8vE1GYIo9dacK0REgDX4smy6GAkxlH1yCGGsk+tgiDhNKuKu3yNrMdxafmKTF632F8Vx4BNK57GvlFisrkjN9WDAtjsWA0ENT2e2nETUb/n7qwhvGnrHuf5bX6Vh/n3xffU3PeHdR+FA92i6ufT3AlyAREoNDh6chiMWTvjKjHDeRhOa9YkOQRq1vQXEMppAQVwHCuIcV2g5rBn6GmZZpTR7vnSD6ZmhdSl176gqKTXu5E+YbfL0adwNtHP7dT7t7b46DVZIkzaRJOM+S6KcrzYVg+T3wSRFRQashjfU18NutrKa/7PXbtuJvpIjbgPeqd+pjmRw6YKpnANFSQcpzTZgpSNJ6J7uiagAbir/8tNXJ/OsOnRh6iuIexxrmkIneAgz8QoLmiaJ8sLQrELVK2yn3wOHp57BAZJhDZjTBzyoRAuuZ4eoxHruY1pSb7qq79cIeAdOwin4GdgMeIMHeG+FZWYaiUQQyC5b50zKjYw97dFjAeY2I4Bnl105Iku1y0lMA1ZHolLx19uZnRdILcXKlZGQx/GdEqSsMRU1BIrFqRcV1qQOOHyxOLXEGcbRtAEsuAC2V4K3p5mFJ22IDWaEkk9ttf5Izb2LkD1MnrSwztXmmD/Qi/EmVEFBfiKGmftsPwVaIoZanlKndMZsIBOskFYpDOq3QUs9aSbAAtL5Dbokus2G4/asthNMK5UQKCOhU97oaOYNGsTah+jfCKsZnTRn5TbhFX8ghg8CBYt/BjeYYYUrtUZ5jVij/op7V5SsbA4mYTOwZ46hqdpbB6Qvq3AS2HHNkC15pTDIcDNGsMPXaBidXYPHc6PJAkRh29Vx8KcgX46LoUQBhRM+3SW6Opll/wgxxsPgKJKzr5QCmwkUxNbeg6Wj34SUnEzOemSuvS2OetRCO8Tyy+QbSKVJcqkia+GvDefFwMOmgnD7h81TUtMn+mRpyJJ349HhAnoWFTejhpYTL9G8N2nVg1qkXBeoS9Nw2fB27t7trm7d/QK7Cr4uoCeOQ7/8JfKT77KiDzLImESHw/0wf73QeHu74hxv7uihi4fTX+XEwAyQG3264dwv17aJ5N335Vt9sdrAXhPOAv8JFvzqyYXwfx8WYJaef1gMl98JRFyl5Mv5Uo/oVH5ww5OzLFsiTPDns7fS6EURSSWd/92BxMYQ8sBaH+j+wthQPdVgDGpTfi+JQIWMD8xKqULliRH01rTeyF8x8q/GBEEEBrAJMPf25UQwi0b8tmqRXY7kIvNkzrkvRWLnxoGYEJsz8u4oOyMp8cHyaybb1HdMCaLApUE+/7xLIZGP6H9xuSEXp1zLIdjk5nBaMuV/yTDRRP8Y2ww5RO6d2D94o+6ucWIqUAvgHIHXhZsmDhjVLczmZ3ca0Cb3PpKwt2UtHVQ0BgFJsqqTsnzZPlKahRUkEu4qmkJt+kqdae76ViWe3STan69yaF9+fESD2lcQshLHWVu4ovItXxO69bqC5p1nZLvI8NdQB9s9UNaJGlQ5mG947ipdDA0eTIw/A1zEdjWquIsQXXGIVEH0thC5M+W9pZe7IhAVnPJkYCCXN5a32HjN6nsvokEqRS44tGIs7s2LVTvcrHAF+RVmI8L4HUYk4x+67AxSMJKqCg8zrGOgvK9kNMdDrNiUtSWuHFpC8/p5qIQrEo/H+1l/0cAwQ2nKmpWxKcMIuHY44Y6DlkpO48tRuUGBWT0FyHwSKO72Ud+tJUfdaZ4CWNijzZtlRa8+CkmO/EwHYfPZFU/hzjFWH7vnzHRMo+aF9u8qHSAiEkA2HjoNQPEwHsDKOt6hOoK3Ce/+/9boMWDa44I6FrQhdgS7OnNaSzwxWKZMcyHi6LN4WC6sSj0qm2PSOGBTvDs/GWJS6SwEN/ULwpb4LQo9fYjUfSXRwZkynUazlSpvX9e+G2zor8l+YaMxSEomDdLHGcD6YVQPegTaA74H8+V4WvJkFUrjMLGLlvSZQWvi8/QA7yzQ8GPno//5SJHRP/OqKObPCo81s/+6WgLqykYpGAgQZhVDEBPXWgU/WzFZjKUhSFInufPRiMAUULC6T11yL45ZrRoB4DzOyJShKXaAJIBS9wzLYIoCEcJKQW8GVCx4fihqJ6mshBUXSw3wWVj3grrHQlGNGhIDNNzsxQ3M+GWn6ASobIWC+LbYOC6UpahVO13Zs2zOzZC8z7FmA05JhUGyBsF4tsG0drcggIFzgg/kpf3+CnAXKiMgIE8Jk/Mhpkc8DUJEUzDSnWlQFme3d0sHZDrg7LavtsEX3cHwjCYA17pMTfx8Ajw9hHscN67hyo+RJQ4458RmPywXykkVcW688oVUrQhahpPRvTWPnuI0B+SkQu7dCyvLRyFYlC1LG1gRCIvn3rwQeINzZQC2KXq31FaR9UmVV2QeGVqBHjmE+VMd3b1fhCynD0pQNhCG6/WCDbKPyE7NRQzL3BzQAJ0g09aUzcQA6mUp9iZFK6Sbp/YbHjo++7/Wj8S4YNa+ZdqAw1hDrKWFXv9+zaXpf8ZTDSbiqsxnwN/CzK5tPkOr4tRh2kY3Bn9JtalbIOI4b3F7F1vPQMfoDcdxMS8CW9m/NCW/HILTUVWQIPiD0j1A6bo8vsv6P1hCESl2abrSJWDrq5sSzUpwoxaCU9FtJyYH4QFMxDBpkkBR6kn0LMPO+5EJ7Z6bCiRoPedRZ/P0SSdii7ZnPAtVwwHUidcdyspwncz5uq6vvm4IEDbJVLUFCn/LvIHfooUBTkFO130FC7CmmcrKdgDJcid9mvVzsDSibOoXtIf9k6ABle3PmIxejodc4aob0QKS432srrCMndbfD454q52V01G4q913mC5HOsTzWF4h2No1av1VbcUgWAqyoZl+11PoFYnNv2HwAODeNRkHj+8SF1fcvVBu6MrehHAZK1Gm69ICcTKizykHgGFx7QdowTVAsYEF2tVc0Z6wLryz2FI1sc5By2znJAAmINndoJiB4sfPdPrTC8RnkW7KRCwxC6YvXg5ahMlQuMpoCSXjOlBy0Kij+bsCYPbGp8BdCBiLmLSAkEQRaieWo1SYvZIKJGj9Ur/eWHjiB7SOVdqMAVmpBvfRiebsFjger7DC+8kRFGtNrTrnnGD2GAJb8rQCWkUPYHhwXsjNBSkE6lGWUj5QNhK0DMNM2l+kXRZ0KLZaGsFSIdQz/HXDxf3/TE30+DgBKWGWdxElyLccJfEpjsnszECNoDGZpdwdRgCixeg9L4EPhH+RptvRMVRaahu4cySjS3P5wxAUCPkmn+rhyASpmiTaiDeggaIxYBmtLZDDhiWIJaBgzfCsAGUF1Q1SFZYyXDt9skCaxJsxK2Ms65dmdp5WAZyxik/zbrTQk5KmgxCg/f45L0jywebOWUYFJQAJia7XzCV0x89rpp/f3AVWhSPyTanqmik2SkD8A3Ml4NhIGLAjBXtPShwKYfi2eXtrDuKLk4QlSyTw1ftXgwqA2jUuopDl+5tfUWZNwBpEPXghzbBggYCw/dhy0ntds2yeHCDKkF/YxQjNIL/F/37jLPHCKBO9ibwYCmuxImIo0ijV2Wbg3kSN2psoe8IsABv3RNFaF9uMyCtCYtqcD+qNOhwMlfARQUdJ2tUX+MNJqOwIciWalZsmEjt07tfa8ma4cji9sqz+Q9hWfmMoKEbIHPOQORbhQRHIsrTYlnVTNvcq1imqmmPDdVDkJgRcTgB8Sb6epCQVmFZe+jGDiNJQLWnfx+drTKYjm0G8yH0ZAGMWzEJhUEQ4Maimgf/bkvo8PLVBsZl152y5S8+HRDfZIMCbYZ1WDp4yrdchOJw8k6R+/2pHmydK4NIK2PHdFPHtoLmHxRDwLFb7eB+M4zNZcB9NrAgjVyzLM7xyYSY13ykWfIEEd2n5/iYp3ZdrCf7fL+en+sIJu2W7E30MrAgZBD1rAAbZHPgeAMtKCg3NpSpYQUDWJu9bT3V7tOKv+NRiJc8JAKqqgCA/PNRBR7ChpiEulyQApMK1AyqcWnpSOmYh6yLiWkGJ2mklCSPIqN7UypWj3dGi5MvsHQ87MrB4VFgypJaFriaHivwcHIpmyi5LhNqtem4q0n8awM19Qk8BOS0EsqGscuuydYsIGsbT5GHnERUiMpKJl4ON7qjB4fEqlGN/hCky89232UQCiaeWpDYCJINXjT6xl4Gc7DxRCtgV0i1ma4RgWLsNtnEBRQFqZggCLiuyEydmFd7WlogpkCw5G1x4ft2psm3KAREwVwr1Gzl6RT7FDAqpVal34ewVm3VH4qn5mjGj+bYL1NgfLNeXDwtmYSpwzbruDKpTjOdgiIHDVQSb5/zBgSMbHLkxWWgghIh9QTFSDILixVwg0Eg1puooBiHAt7DzwJ7m8i8/i+jHvKf0QDnnHVkVTIqMvIQImOrzCJwhSR7qYB5gSwL6aWL9hERHCZc4G2+JrpgHNB8eCCmcIWIQ6rSdyPCyftXkDlErUkHafHRlkOIjxGbAktz75bnh50dU7YHk+Mz7wwstg6RFZb+TZuSOx1qqP5C66c0mptQmzIC2dlpte7vZrauAMm/7RfBYkGtXWGiaWTtwvAQiq2oD4YixPLXE2khB2FRaNRDTk+9sZ6K74Ia9VntCpN4BhJGJMT4Z5c5FhSepRCRWmBXqx+whVZC4me4saDs2iNqXMuCl6iAZflH8fscC1sTsy4PHeC+XYuqMBMUun5YezKbRKmEPwuK+CLzijPEQgfhahQswBBLfg/GBgBiI4QwAqzJkkyYAWtjzSg2ILgMAgqxYfwERRo3zruBL9WOryUArSD8sQOcD7fvIODJxKFS615KFPsb68USBEPPj1orNzFY2xoTtNBVTyzBhPbhFH0PI5AtlJBl2aSgNPYzxYLw7XTDBDinmVoENwiGzmngrMo8OmnRP0Z0i0Zrln9DDFcnmOoBZjABaQIbPOJYZGqX+RCMlDDbElcjaROLDoualmUIQ88Kekk3iM4OQrADcxi3rJguS4MOIBIgKgXrjd1WkbCdqxJk/4efRIFsavZA7KvvJQqp3Iid5Z0NFc5aiMRzGN3vrpBzaMy4JYde3wr96PjN90AYOIbyp6T4zj8LoE66OGcX1Ef4Z3KoWLAUF4BTg7ug/AbkG5UNQXAMkQezujSHeir2uTThgd3gpyzDrbnEdDRH2W7U6PeRvBX1ZFMP5RM+Zu6UUZZD8hDPHldVWntTCNk7To8IeOW9yn2wx0gmurwqC60AOde4r3ETi5pVMSDK8wxhoGAoEX9NLWHIR33VbrbMveii2jAJlrxwytTHbWNu8Y4N8vCCyZjAX/pcsfwXbLze2+D+u33OGBoJyAAL3jn3RuEcdp5If8O+a4NKWvxOTyDltG0IWoHhwVGe7dKkCWFT++tm+haBCikRUUMrMhYKZJKYoVuv/bsJzO8DwfVIInQq3g3BYypiz8baogH3r3GwqCwFtZnz4xMjAVOYnyOi5HWbFA8n0qz1OjSpHWFzpQOpvkNETZBGpxN8ybhtqV/DMUxd9uFZmBfKXMCn/SqkWJyKPnT6lq+4zBZni6fYRByJn6OK+OgPBGRAJluwGSk4wxjOOzyce/PKODwRlsgrVkdcsEiYrqYdXo0Er2GXi2GQZd0tNJT6c9pK1EEJG1zgDJBoTVuCXGAU8BKTvCO/cEQ1Wjk3Zzuy90JX4m3O5IlxVFhYkSUwuQB2up7jhvkm+bddRQu5F9s0XftGEJ9JSuSk+ZachCbdU45fEqbugzTIUokwoAKvpUQF/CvLbWW5BNQFqFkJg2f30E/48StNe5QwBg8zz3YAJ82FZoXBxXSv4QDooDo79NixyglO9AembuBcx5Re3CwOKTHebOPhkmFC7wNaWtoBhFuV4AkEuJ0J+1pT0tLkvFVZaNzfhs/Kd3+A9YsImlO4XK4vpCo/elHQi/9gkFg07xxnuXLt21unCIpDV+bbRxb7FC6nWYTsMFF8+1LUg4JFjVt3vqbuhHmDKbgQ4e+RGizRiO8ky05LQGMdL2IKLSNar0kNG7lHJMaXr5mLdG3nykgj6vB/KVijd1ARWkFEf3yiUw1v/WaQivVUpIDdSNrrKbjO5NPnxz6qTTGgYg03HgPhDrCFyYZTi3XQw3HXCva39mpLNFtz8AiEhxAJHpWX13gCTAwgm9YTvMeiqetdNQv6IU0hH0G+ZManTqDLPjyrOse7WiiwOJCG+J0pZYULhN8NILulmYYvmVcV2MjAfA39sGKqGdjpiPo86fecg65UPyXDIAOyOkCx5NQsLeD4gGVjTVDwOHWkbbBW0GeNjDkcSOn2Nq4cEssP54t9D749A7M1AIOBl0Fi0sSO5v3P7LCBrM6ZwFY6kp2FX6AcbGUdybnfChHPyu6WlRZ2Fwv9YM0RMI7kISRgR8HpQSJJOyTfXj/6gQKuihPtiUtlCQVPohUgzfezTg8o1b3n9pNZeco1QucaoXe40Fa5JYhqdTspFmxGtW9h5ezLFZs3j/N46f+S2rjYNC2JySXrnSAFhvAkz9a5L3pza8eYKHNoPrvBRESpxYPJdKVUxBE39nJ1chrAFpy4MMkf0qKgYALctGg1DQI1kIymyeS2AJNT4X240d3IFQb/0jQbaHJ2YRK8A+ls6WMhWmpCXYG5jqapGs5/eOJErxi2/2KWVHiPellTgh/fNl/2KYPKb7DUcAg+mCOPQFCiU9Mq/WLcU1xxC8aLePFZZlE+PCLzf7ey46INWRw2kcXySR9FDgByXzfxiNKwDFbUSMMhALPFSedyjEVM5442GZ4hTrsAEvZxIieSHGSgkwFh/nFNdrrFD4tBH4Il7fW6ur4J8Xaz7RW9jgtuPEXQsYk7gcMs2neu3zJwTyUerHKSh1iTBkj2YJh1SSOZL5pLuQbFFAvyO4k1Hxg2h99MTC6cTUkbONQIAnEfGsGkNFWRbuRyyaEZInM5pij73EA9rPIUfU4XoqQpHT9THZkW+oKFLvpyvTBMM69tN1Ydwv1LIEhHsC+ueVG+w+kyCPsvV3erRikcscHjZCkccx6VrBkBRusTDDd8847GA7p2Ucy0y0HdSRN6YIBciYa4vuXcAZbQAuSEmzw+H/AuOx+aH+tBL88H57D0MsqyiZxhOEQkF/8DR1d2hSPMj/sNOa5rxcUnBgH8ictv2J+cb4BA4v3MCShdZ2vtK30vAwkobnEWh7rsSyhmos3WC93Gn9C4nnAd/PjMMtQfyDNZsOPd6XcAsnBE/mRHtHEyJMzJfZFLE9OvQa0i9kUmToJ0ZxknTgdl/XPV8xoh0K7wNHHsnBdvFH3sv52lU7UFteseLG/VanIvcwycVA7+BE1Ulyb20BvwUWZcMTKhaCcmY3ROpvonVMV4N7yBXTL7IDtHzQ4CCcqF66LjF3xUqgErKzolLyCG6Kb7irP/MVTCCwGRxfrPGpMMGvPLgJ881PHMNMIO09T5ig7AzZTX/5PLlwnJLDAPfuHynSGhV4tPqR3gJ4kg4c06c/F1AcjGytKm2Yb5jwMotF7vro4YDLWlnMIpmPg36NgAZsGA0W1spfLSue4xxat0Gdwd0lqDBOgIaMANykwwDKejt5YaNtJYIkrSgu0KjIg0pznY0SCd1qlC6R19g97UrWDoYJGlrvCE05J/5wkjpkre727p5PTRX5FGrSBIfJqhJE/IS876PaHFkx9pGTH3oaY3jJRvLX9Iy3Edoar7cFvJqyUlOhAEiOSAyYgVEGkzHdug+oRHIEOXAExMiTSKU9A6nmRC8mp8iYhwWdP2U/5EkFAdPrZw03YA3gSyNUtMZeh7dDCu8pF5x0VORCTgKp07ehy7NZqKTpIC4UJJ89lnboyAfy5OyXzXtuDRbtAFjZRSyGFTpFrXwkpjSLIQIG3N0Vj4BtzK3wdlkBJrO18MNsgseR4BysJilI0wI6ZahLhBFA0XBmV8d4LUzEcNVb0xbLjLTETYN8OEVqNxkt10W614dd1FlFFVTIgB7/BQQp1sWlNolpIu4ekxUTBV7NmxOFKEBmmN+nA7pvF78/RII5ZHA09OAiE/66MF6HQ+qVEJCHxwymukkNvzqHEh52dULPbVasfQMgTDyBZzx4007YiKdBuUauQOt27Gmy8ISclPmEUCIcuLbkb1mzQSqIa3iE0PJh7UMYQbkpe+hXjTJKdldyt2mVPwywoODGJtBV1lJTgMsuSQBlDMwhEKIfrvsxGQjHPCEfNfMAY2oxvyKcKPUbQySkKG6tj9AQyEW3Q5rpaDJ5Sns9ScLKeizPRbvWYAw4bXkrZdmB7CQopCH8NAmqbuciZChHN8lVGaDbCnmddnqO1PQ4ieMYfcSiBE5zzMz+JV/4eyzrzTEShvqSGzgWimkNxLvUj86iAwcZuIkqdB0VaIB7wncLRmzHkiUQpPBIXbDDLHBlq7vp9xwuC9AiNkIptAYlG7Biyuk8ILdynuUM1cHWJgeB+K3wBP/ineogxkvBNNQ4AkW0hvpBOQGFfeptF2YTR75MexYDUy7Q/9uocGsx41O4IZhViw/2FvAEuGO5g2kyXBUijAggWM08bRhXg5ijgMwDJy40QeY/cQpUDZiIzmvskQpO5G1zyGZA8WByjIQU4jRoFJt56behxtHUUE/om7Rj2psYXGmq3llVOCgGYKNMo4pzwntITtapDqjvQtqpjaJwjHmDzSVGLxMt12gEXAdLi/caHSM3FPRGRf7dB7YC+cD2ho6oL2zGDCkjlf/DFoQVl8GS/56wur3rdV6ggtzZW60MRB3g+U1W8o8cvqIpMkctiGVMzXUFI7FacFLrgtdz4mTEr4aRAaQ2AFQaNeG7GX0yOJgMRYFziXdJf24kg/gBQIZMG/YcPEllRTVNoDYR6oSJ8wQNLuihfw81UpiKPm714bZX1KYjcXJdfclCUOOpvTxr9AAJevTY4HK/G7F3mUc3GOAKqh60zM0v34v+ELyhJZqhkaMA8UMMOU90f8RKEJFj7EqepBVwsRiLbwMo1J2zrE2UYJnsgIAscDmjPjnzI8a719Wxp757wqmSJBjXowhc46QN4RwKIxqEE6E5218OeK7RfcpGjWG1jD7qND+/GTk6M56Ig4yMsU6LUW1EWE+fIYycVV1thldSlbP6ltdC01y3KUfkobkt2q01YYMmxpKRvh1Z48uNKzP/IoRIZ/F6buOymSnW8gICitpJjKWBscSb9JJKaWkvEkqinAJ2kowKoqkqZftRqfRQlLtKoqvTRDi2vg/RrPD/d3a09J8JhGZlEkOM6znTsoMCsuvTmywxTCDhw5dd0GJOHCMPbsj3QLkTE3MInsZsimDQ3HkvthT7U9VA4s6G07sID0FW4SHJmRGwCl+Mu4xf0ezqeXD2PtPDnwMPo86sbwDV+9PWcgFcARUVYm3hrFQrHcgMElFGbSM2A1zUYA3baWfheJp2AINmTJLuoyYD/OwA4a6V0ChBN97E8YtDBerUECv0u0TlxR5yhJCXvJxgyM73Bb6pyq0jTFJDZ4p1Am1SA6sh8nADd1hAcGBMfq4d/UfwnmBqe0Jun1n1LzrgKuZMAnxA3NtCN7Klf4BH+14B7ibBmgt0TGUafVzI4uKlpF7v8NmgNjg90D6QE3tbx8AjSAC+OA1YJvclyPKgT27QpIEgVYpbPYGBsnyCNrGz9XUsCHkW1QAHgL2STZk12QGqmvAB0NFteERkvBIH7INDsNW9KKaAYyDMdBEMzJiWaJHZALqDxQDWRntumSDPcplyFiI1oDpT8wbwe01AHhW6+vAUUBoGhY3CT2tgwehdPqU/4Q7ZLYvhRl/ogOvR9O2+wkkPKW5vCTjD2fHRYXONCoIl4Jh1bZY0ZE1O94mMGn/dFSWBWzQ/VYk+Gezi46RgiDv3EshoTmMSlioUK6MQEN8qeyK6FRninyX8ZPeUWjjbMJChn0n/yJvrq5bh5UcCAcBYSafTFg7p0jDgrXo2QWLb3WpSOET/Hh4oSadBTvyDo10IufLzxiMLAnbZ1vcUmj3w7BQuIXjEZXifwukVxrGa9j+DXfpi12m1RbzYLg9J2wFergEwOxFyD0/JstNK06ZN2XdZSGWxcJODpQHOq4iKqjqkJUmPu1VczL5xTGUfCgLEYyNBCCbMBFT/cUP6pE/mujnHsSDeWxMbhrNilS5MyYR0nJyzanWXBeVcEQrRIhQeJA6Xt4f2eQESNeLwmC10WJVHqwx8SSyrtAAjpGjidcj1E2FYN0LObUcFQhafUKTiGmHWRHGsFCB+HEXgrzJEB5bp0QiF8ZHh11nFX8AboTD0PS4O1LqF8XBks2MpjsQnwKHF6HgaKCVLJtcr0XjqFMRGfKv8tmmykhLRzu+vqQ02+KpJBjaLt9ye1Ab+BbEBhy4EVdIJDrL2naV0o4wU8YZ2Lq04FG1mWCKC+UwkXOoAjneU/xHplMQo2cXUlrVNqJYczgYlaOEczVCs/OCgkyvLmTmdaBJc1iBLuKwmr6qtRnhowngsDxhzKFAi02tf8bmET8BO27ovJKF1plJwm3b0JpMh38+xsrXXg7U74QUM8ZCIMOpXujHntKdaRtsgyEZl5MClMVMMMZkZLNxH9+b8fH6+b8Lev30A9TuEVj9CqAdmwAAHBPbfOBFEATAPZ2CS0OH1Pj/0Q7PFUcC8hDrxESWdfgFRm+7vvWbkEppHB4T/1ApWnlTIqQwjcPl0VgS1yHSmD0OdsCVST8CQVwuiew1Y+g3QGFjNMzwRB2DSsAk26cmA8lp2wIU4p93AUBiUHFGOxOajAqD7Gm6NezNDjYzwLOaSXRBYcWipTSONHjUDXCY4mMI8XoVCR/Rrs/JLKXgEx+qkmeDlFOD1/yTQNDClRuiUyKYCllfMiQiyFkmuTz2vLsBNyRW+xz+5FElFxWB28VjYIGZ0Yd+5wIjkcoMaggxswbT0pCmckRAErbRlIlcOGdBo4djTNO8FAgQ+lT6vPS60BwTRSUAM3ddkEAZiwtEyArrkiDRnS7LJ+2hwbzd2YDQagSgACpsovmjil5wfPuXq3GuH0CyE7FK3M4FgRaFoIkaodORrPx1+JpI9psyNYIFuJogZa0/1AhOWdlHQxdAgbwacsHqPZo8u/ngAH2GmaTdhYnBfSDbBfh8CHq6Bx5bttP2+RdM+MAaYaZ0Y/ADkbNCZuAyAVQa2OcXOeICmDn9Q/eFkDeFQg5MgHEDXq/tVjj+jtd26nhaaolWxs1ixSUgOBwrDhRIGOLyOVk2/Bc0UxvseQCO2pQ2i+Krfhu/WeBovNb5dJxQtJRUDv2mCwYVpNl2efQM9xQHnK0JwLYt/U0Wf+phiA4uw8G91slC832pmOTCAoZXohg1fewCZqLBhkOUBofBWpMPsqg7XEXgPfAlDo2U5WXjtFdS87PIqClCK5nW6adCeXPkUiTGx0emOIDQqw1yFYGHEVx20xKjJVYe0O8iLmnQr3FA9nSIQilUKtJ4ZAdcTm7+ExseJauyqo30hs+1qSW211A1SFAOUgDlCGq7eTIcMAeyZkV1SQJ4j/e1Smbq4HcjqgFbLAGLyKxlMDMgZavK5NAYH19Olz3la/QCTiVelFnU6O/GCvykqS/wZJDhKN9gBtSOp/1SP5VRgJcoVj+kmf2wBgv4gjrgARBWiURYx8xENV3bEVUAAWWD3dYDKAIWk5opaCFCMR5ZjJExiCAw7gYiSZ2rkyTce4eNMY3lfGn+8p6+vBckGlKEXnA6Eota69OxDO9oOsJoy28BXOR0UoXNRaJD5ceKdlWMJlOFzDdZNpc05tkMGQtqeNF2lttZqNco1VtwXgRstLSQ6tSPChgqtGV5h2DcDReIQadaNRR6AsAYKL5gSFsCJMgfsaZ7DpKh8mg8Wz8V7H+gDnLuMxaWEIUPevIbClgap4dqmVWSrPgVYCzAoZHIa5z2Ocx1D/GvDOEqMOKLrMefWIbSWHZ6jbgA8qVBhYNHpx0P+jAgN5TB3haSifDcApp6yymEi6Ij/GsEpDYUgcHATJUYDUAmC1SCkJ4cuZXSAP2DEpQsGUjQmKJfJOvlC2x/pChkOyLW7KEoMYc5FDC4v2FGqSoRWiLsbPCiyg1U5yiHZVm1XLkHMMZL11/yxyw0UnGig3MFdZklN5FI/qiT65T+jOXOdO7XbgWurOAZR6Cv9uu1cm5LjkXX4xi6mWn5r5NjBS0gTliHhMZI2WNqSiSphEtiCAwnafS11JhseDGHYQ5+bqWiAYiAv6Jsf79/VUs4cIl+n6+WOjcgB/2l5TreoAV2717JzZbQIR0W1cl/dEqCy5kJ3ZSIHuU0vBoHooEpiHeQWVkkkOqRX27eD1FWw4BfO9CJDdKoSogQi3hAAwsPRFrN5RbX7bqLdBJ9JYMohWrgJKHSjVl1sy2xAG0E3sNyO0oCbSGOxCNBRRXTXenYKuwAoDLfnDcQaCwehUOIDiHAu5m5hMpKeKM4sIo3vxACakIxKoH2YWF2QM84e6F5C5hJU4g8uxuFOlAYnqtwxmHyNEawLW/PhoawJDrGAP0JYWHgAVUByo/bGdiv2T2EMg8gsS14/rAdzlOYazFE7w4OzxeKiWdm3nSOnQRRKXSlVo8HEAbBfyJMKqoq+SCcTSx5NDtbFwNlh8VhjGGDu7JG5/TAGAvniQSSUog0pNzTim8Owc6QTuSKSTXlQqwV3eiEnklS3LeSXYPXGK2VgeZBqNcHG6tZHvA3vTINhV0ELuQdp3t1y9+ogD8Kk/W7QoRN1UWPqM4+xdygkFDPLoTaumKReKiLWoPHOfY54m3qPx4c+4pgY3MRKKbljG8w4wvz8pxk3AqKsy4GMAkAtmRjRMsCxbb4Q2Ds0Ia9ci8cMT6DmsJG00XaHCIS+o3F8YVVeikw13w+OEDaCYYhC0ZE54kA4jpjruBr5STWeqQG6M74HHL6TZ3lXrd99ZX++7LhNatQaZosuxEf5yRA15S9gPeHskBIq3Gcw81AGb9/O53DYi/5CsQ51EmEh8Rkg4vOciClpy4d04eYsfr6fyQkBmtD+P8sNh6e+XYHJXT/lkXxT4KXU5F2sGxYyzfniMMQkb9OjDN2C8tRRgTyL7GwozH14PrEUZc6oz05Emne3Ts5EG7WolDmU8OB1LDG3VrpQxp+pT0KYV5dGtknU64JhabdqcVQbGZiAxQAnvN1u70y1AnmvOSPgLI6uB4AuDGhmAu3ATkJSw7OtS/2ToPjqkaq62/7WFG8advGlRRqxB9diP07JrXowKR9tpRa+jGJ91zxNTT1h8I2PcSfoUPtd7NejVoH03EUcqSBuFZPkMZhegHyo2ZAITovmm3zAIdGFWxoNNORiMRShgwdYwFzkPw5PA4a5MIIQpmq+nsp3YMuXt/GkXxLx/P6+ZJS0lFyz4MunC3eWSGE8xlCQrKvhKUPXr0hjpAN9ZK4PfEDrPMfMbGNWcHDzjA7ngMxTPnT7GMHar+gMQQ3NwHCv4zH4BIMYvzsdiERi6gebRmerTsVwZJTRsL8dkZgxgRxmpbgRcud+YlCIRpPwHShlUSwuipZnx9QCsEWziVazdDeKSYU5CF7UVPAhLer3CgJOQXl/zh575R5rsrmRnKAzq4POFdgbYBuEviM4+LVC15ssLNFghbTtHWerS1hDt5s4qkLUha/qpZXhWh1C6lTQAqCNQnaDjS7UGFBC6wTu8yFnKJnExCnAs3Ok9yj5KpfZESQ4lTy5pTGTnkAUpxI+yjEldJfSo4y0QhG4i4IwkRFGcjWY8+EzgYYJUK7BXQksLxAww/YYWBMhJILB9e8ePEJ4OP7z+4/wOQDl64iOYDp26DaONPxpKtBxq/aTzRGarm3VkPYTLJKx6Z/Mw2YbBGseJhPMwhhNswrIkyvV2BYzrvZbxLpKwcWJhYmFtVZ+lPEq91FzVp1HlQY1bZVLqeNR9SAUn6n0E28k/UuGkNpP1DBI5ch/EehZfjUQ9aE41NhETExoPT2gGQz0IhWJbEOvTQ4wgcXCHHFBhewYUiFHuhRSAUVmEHeCRQHQkXGFwkAgyzREJCVN7TRnTon36Zw3tPhx4EALwNdwDv+J41YSP4B2CQqz0EFgARZ4ESgBHQgROwAVn9GTI+HYexTUevLUeta4/DqKrbMVS+Yqb8hUwYCrlgKtmAq1YCrFgKrd4qpXiqZcKn1oqdWipjYKpWwVPVYqW6xUpVipKqFR3QKjagVEtAqHpxUMTitsnFaJOKx2cVhswq35RVpyiq9lFVNIKnOQVMkgqtYxVNxiqQjFS7GKlSIVIsQqPIhUWwioigFQ++KkN8VHr49HDw9Ebo9EDo9DTo9Crg9BDg9/Wx7gWx7YWwlobYrOGxWPNisAaAHEyALpkAVDIAeWAArsABVXACYuAD5cAF6wAKFQAQqgAbVAAsoAAlQAUaYAfkwAvogBWQACOgAD9AAHSAAKT4GUdMiOvFngBTwCn2AZ7Dv6B6k/90B8+yRnkV144AIBoAMTQATGgAjNAA4YABgwABZgB/mQCwyAVlwCguASlwCEuAQFwB4uAMlwBYuAJlQAUVAAhUD2KgdpUDaJgaRMDFJgX5MC1JgWJEAokQCWRAHxEAWkQBMRADpEAMkQAYROAEecC484DRpwBDTnwNOdw05tjTmiNOYwtswhYFwLA7BYG4LA2BYGOLAwRYFuLAsxYFQJAohIEyJAMwkAwiQC0JAJgkAeiQBkJAFokAPCQA0JABwcD4Dgc4cDdDgaYcDIDgYgUC6CgWgUClCgUYUAVBQBOFAEYMALgwAgDA9QYAdIn8AZzeBB2L5EcWrenUT1KXienEsuJJ7x5U8XlTjc1NVzUyXFTGb1LlpUtWlTDIjqwE4LsagowoCi2gJLKAkpoBgJQNpAIhNqaEoneI6kiiqQ6Go/n6j0cS+a2gEU8gIHJ+BwfgZX4GL+Bd/gW34FZ+BS/gUH4FN6BTegTvoEv6BJegRnYEF2A79gOvYDl2BdEjCkqkGtwXp0LNToIskOTXzh/F062yJ7AAAAEDAWAAABWhJ+KPEIJgBFxMVP7w2QJBGHASQnOBKXKFIdUK4igKA9IEaYJg) format('embedded-opentype'),url(data:application/font-woff;base64,d09GRgABAAAAAFuAAA8AAAAAsVwAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAABGRlRNAAABWAAAABwAAAAcbSqX3EdERUYAAAF0AAAAHwAAACABRAAET1MvMgAAAZQAAABFAAAAYGe5a4ljbWFwAAAB3AAAAsAAAAZy2q3jgWN2dCAAAAScAAAABAAAAAQAKAL4Z2FzcAAABKAAAAAIAAAACP//AANnbHlmAAAEqAAATRcAAJSkfV3Cb2hlYWQAAFHAAAAANAAAADYFTS/YaGhlYQAAUfQAAAAcAAAAJApEBBFobXR4AABSEAAAAU8AAAN00scgYGxvY2EAAFNgAAACJwAAAjBv+5XObWF4cAAAVYgAAAAgAAAAIAFqANhuYW1lAABVqAAAAZ4AAAOisyygm3Bvc3QAAFdIAAAELQAACtG6o+U1d2ViZgAAW3gAAAAGAAAABsMYVFAAAAABAAAAAMw9os8AAAAA0HaBdQAAAADQdnOXeNpjYGRgYOADYgkGEGBiYGRgZBQDkixgHgMABUgASgB42mNgZulmnMDAysDCzMN0gYGBIQpCMy5hMGLaAeQDpRCACYkd6h3ux+DAoPD/P/OB/wJAdSIM1UBhRiQlCgyMADGWCwwAAAB42u2UP2hTQRzHf5ekaVPExv6JjW3fvTQ0sa3QLA5xylBLgyBx0gzSWEUaXbIoBBQyCQGHLqXUqYNdtIIgIg5FHJxEtwqtpbnfaV1E1KFaSvX5vVwGEbW6OPngk8/vvXfv7pt3v4SImojIDw6BViKxRgIVBaZwVdSv+xvXA+Iuzqcog2cOkkvDNE8Lbqs74k64i+5Sf3u8Z2AnIRLbyVCyTflVSEXVoEqrrMqrgiqqsqqqWQ5xlAc5zWOc5TwXucxVnuE5HdQhHdFRHdNJndZZndeFLc/zsKJLQ/WV6BcrCdWkwspVKZVROaw0qUqqoqZZcJhdTnGGxznHBS5xhad5VhNWCuturBTXKZ3RObuS98pb9c57k6ql9rp2v1as5deb1r6s9q1GV2IrHSt73T631424YXzjgPwqt+Rn+VG+lRvyirwsS/KCPCfPytPypDwhj8mjctRZd9acF86y89x55jxxHjkPnXstXfbt/pNjj/nwXW+cHa6/SYvZ7yEwbDYazDcIgoUGzY3h2HtqgUcs1AFPWKgTXrRQF7xkoQhRf7uF9hPFeyzUTTSwY6EoUUJY6AC8bSGMS4Ys1Au3WaiPSGGsMtkdGH2rzJgYHAaYjxIwQqtB1CnYkEZ9BM6ALOpROAfyqI/DBQudgidBETXuqRIooz4DV0AV9UV4GsyivkTEyMMmw1UYGdhkuAYjA5sMGMvIwCbDDRgZeAz1TXgcmDy3YeRhk+cOjCxsMjyAkYFNhscwMrDJ8BQ2886gXoaRhedQvyTSkDZ7uA6HLLQBI5vGntAbGHugTc53cMxC7+E4SKL+ACOzNpk3YWTWJid+iRo5NXIKM3fBItAPW55FdJLY3FeHBDr90606JCIU9Jk+Ms3/Y/8L8jUq3y79bJ/0/+ROoP4v9v/4/mj+i7HBXUd0/elU6IHfHt8Aj9EPGAAoAvgAAAAB//8AAnjaxb0JfBvVtTA+dxaN1hltI1m2ZVuSJVneLVlSHCdy9oTEWchqtrBEJRAgCYEsQNhC2EsbWmpI2dqkQBoSYgKlpaQthVL0yusrpW77aEubfq/ly+ujvJampSTW5Dvnzmi1E+jr//3+Xmbu3Llz77nnbuece865DMu0MAy5jGtiOEZkOp8lTNeUwyLP/DH+rEH41ZTDHAtB5lkOowWMPiwayNiUwwTjE46AI5xwhFrINPXYn/7ENY0dbWHfZAiTZbL8ID/InAd5xz2NpIH4STpDGonHIJNE3OP1KG4ISaSNeBuITAyRLgIxoiEUhFAnmUpEiXSRSGqAQEw0kuyFUIb0k2gnGSApyBFi0il2SI5YLGb5MdFjXCey4mNHzQ7WwLGEdZiPPgYR64we8THZHAt+wnT84D/x8YTpGPgheKH4CMEDVF9xBOIeP3EbQgGH29BGgpGkIxCMTCW9qUTA0Zsir+QUP1mt+P2KusevwIO6Bx/Iaj8/OD5O0VNrZW2EsqZBWbO1skRiEKE0DdlKKaSVO5VAuRpqk8VQJAqY7ydxaK44YJvrO2EWjOoDBoFYzQbDNkON+UbiKoRkywMWWf1j4bEY2iIY1AeMgvmEz/kVo9v4FSc/aMZMrFbjl4zWLL0+Y5FlyzNlEVYDudJohg8gPUP7kcB/mn+G6cd+5PV4Q72dXCgocWJADBgUuDTwiXiGSyZo14HOEQ2lE6k0XDIEusexDzZOMXwt1Dutz+tqmxTvlskNWXXUQIbhaurum9GrePqm9Yaeabjkiqf+bUvzDOvb2Y1E+EX2DnemcTP/zLcuu7xjQXdAtjR0Lo5n4/Hs/GtntMlysHt+29NXbH6se//WbFcyu+r28H0MwzI30DYeYTLMXIA2EG8QlHpAsyS0EfEToR0a3utIxFPJ3kiIHCCrZ66b0e2xEmL1dM9YN/MwS5p01N5jMX/BLKt/1R83l0LyC29M6+iYxo/UNg/EF7c2WyyW5tYl8WnhWg2/hyySbD5UhnDyS7OcU0dnrFw+DfGdI7v4QfYIIzOMq9hFtY55gmvC7jZ2FK7sEdrn6IXBuucYhjsGdQ8z0yEbWkkczjjsE5hNAIZrPx2zOLZDmKNXcXtg7EMqidAEEWg+SJCBBNwxvxJfc/bZa+KKf+xoKZybnq5vaqpPTye7CiF+ZFjxZ8/7Qij0hfOG/cowPA1rT1l4ymWnrKmxxqfErTVrpgwPlz1kC+Oy8NMDz6c+IO38K/x0xkPnLW8Kx6qGAoQdL+TD9V9rb+/ctn//trxz8dUrZrD/zk/ferF0cNt1BzctmX2FZPXt/jnFCQNz4Ah/iKllGiCMs1w5Lkg0kiEwj6VTXCDKsX9rMpnvIj9pcDecXAIXMnqn2dTUbN6w0XQ9ue6FV/nnXCH7S3lPWGltVcLsH75ub3ab7A8M28caNrIeOr3o5Q0yFsYL80xaa0EY/UEczV7icUMY5pnelAkmUAXmHYjvFWFGxuqlSaow3OM+/iYY7/l/hVELF4EjRqNR/bvRbOY+DUGzGR/Oh3EqmE/ugIQQguGt/eMYz/+L0cimjeZfQDI3phXMbMQsqH+CjwVz/hf4idHovgVmB8gLvjbicDcC/NypP536E/9N/puMibExdohBmNwyiaZdJGoigos7GpF222xrfnZhML/7Z+ylaqP63Hr+m7bdUkQ6/2cXqdfmvwixY+s2ksXFeXcE+iX0Z+Iow76DBNgjJ7TOdUK18iPsPflfQD+DPsZG2Aj9VmKMMJ4fYRrhIaxhTDR0Elh2vA6h/AE6xUb29mj3sjmL72petXjejPy+oel60M99tFduCI59N3221xe7apOvxs6aHs7vab1IqY2tv7q2xsHeHGml/cV06u/8S/xTjJ+JYc0bWEX0ukW6YmIbGkJRMdjJ9mYIH5QIdJF4hvRGyK7cC7ctImQRcUET99fGXOoft35GYLMQu+g2smnkgZUrH8AL/9Si217IssJ916nv14ZrJrvdxLkQvrvtBcjgPC0NXOicO8Qf4mcxPqh3hgUw3DDfdvLJXngg7N3dN2zbPJSaed3OfZnMU7dvmznp3C3bruO+Nmue0LFsy7S+6265+fCKFYdvvuW6vmlblnUI8xCXp37CrOZv4B9gauDBlYp7adcUXB5DNCwYImlXOJJKkAdvExXxVvKEYnCo+3eIskP9qrrfIYs71CccBjfXRC52udTHHdaP1A1ui/VvH1otbrLrpNXBsGX5B89QghDyimlvNB2KfkxZ5C9/em3+d1+d//IfFp2+2Oxn/s+9n/79p39S3s8idN6g0yZObwJOgKUpNB3GyU0Ls0PbRzIRq4lcarLKOJBkLRzJQD4j2090XrbA7DW8K3jNF5hlGS5e4V2D17zgss4T20egOJte5iD0bReM9yjTxnQxCRj3c5kFzGJmGbNKmwGw39IJDJcXJZGMkaAB4jyJAKw0jt5IAuIE+A+U3cVAZZrq9zhDyBrU8oosuxcGNTzCKJfla7JjNVmuSb/+tuzN2H+X4vlB+PpdfMXXmuVsNiub1T34SFbjYw5itEvVi0K0Nt9pNJUMI7SLGRhf2xipfCYf8z5OdlGKayOucFeVPeS/dbo3lBrbSMmwUiQN5/ed7g0Ds1s17IuZC5kNzM3MZ6EWCa0DtekdJfAxz+R/OX28sND7yRMTBcf++s8mQCQWHya4qBv/ufeMoWyslPA9DtMxUknxkH/yfTnm2CMYzs+Cq3r7PxY/MXomrvTEsRpfEGHa+WN8E1AHjElb7d06ddA7oK/+5Mdsv9EtPms0jv0Z5kf1FqPxWdFtfFr0kHfgDX0Y+5PRSG7RUj0tQr7rmfX8DH4G5W28kKeJLtmQsQkuwMP1pk16EV4sl7vrMJATfyUWo/GwEco4rh4XFQgaiUX9qxZHrMQqKnz/c2d8b9TysYrAuXpP/Rf/Gr8b1qwwc5a+euLa6S6sneNXToG2XrEJi4R5SGs8Sq2S3d97bsfCRaTdaLwKClRHt37mkudvXbjwVrLhuYeGhh56bvfQkHpk2CwvwClqgWwuBfndC3c8dwmstj81KkagcUgbfPY8Zje0W/82VPWJHmSq6pP8hPWpotc/EexDOK3qU+wngPhOCiO9MJRm8TJefjelrzoKnG2Bn+1NCUmPE4gHFmBN9jrTigRIpsACrc9Gstg58ULkp9467+Gf/eFnD5/31lNrt2967dhrm7bzI+VT5m+fzKhvf2MzpICEm79Bopkn07lt1762adNr127LwVqQLdJ5+lpQDcvHPQtVY5knhYrK6q8/JsiP6EuhGZdFdaNszjvpqvc+PI0CdjN0AXsFOC3ZfALDJwr4q2Xq+GF+GNbsxUg5NLLIEXi8otcDQcUts0D8eQ1iVDRAMBTsYiNdRIxE09EIBJO9A2xqgERTaW86BUFn0OD2xFO97FAgFhF6OoQ7prYt4XwSeUgQHiJyDbeke9IdQntciLQ1FlJMaYcUNvZBg+FB1ubjlnRNvl3o6IEU2w7fdNPhm/hh+FLysUu6++DLHkOkrSHYEjH0tEPe7WdD3uyDgvAgK/m4szFFR7ch0toUgBTdWHr7EpaWru6+6dmbbnqWEbV2EtxAsXiZAPTtGPSbHsotI2leoM8TePEqgSQprs7AGFf8kuOkPdZPXGb55POAW1d/jLST9v5YflasP6v/CO7+GNAPC2BMZWmsOjp2NNbfHwMCJD+LPVL+D/OYlWEEI/9jpPddOFkB5d1GSuKZYggmCCd7JUxD7EXAzxyirYnNDLdDZoFdx14kivkvGc3579Jm36reTTvDgBnaO6vzyQ6chQmlsMoIkIQ2+bBDWBud1Va4pcCn8CPqxlh/fgtG8IPaPH8C5wk6/nZDv69jurV5QhtwE0x2iqOsj9Mx8B9/0EaUdiPfOYYDCi/q9jhWRuupMDEU0+CtX0sDFxv07T/K5niBPqN9+tQjgEc31NGCXFeMcCEuQBIc/BK4CO78u7EPYvl3yaEfK3vcb6qP1R2tI7vUjVDDUdKubsSrNjYKY1qBEa2P50SJoaXiksIoLiCwnxS6EBuBde87botNfdEWwYvF/R0/u5yCqhGeEOR2ynSeyXjt6ka7neyye8kryBSWE52y+RBgogrXPZ8E1yIHoHIFUM+AbJhE7lbMtt8ApL+xmZW7PwbjAO0fAVoXQOuiSP/ksIVdFZ0aulsamKUzwPZ/NYDMJRBPCxsBqLzqHyneXF6Ej9HlIFo7+pg+jUb3unRmGpstGkm6etOuDBGA5wCMefp1gTHcdZlvPBXlOslvYTp1cd8UjYLVd/J5awNrIOKLnIt9MD9qdrKrWCvA6ALm3QV9VrsPm60Q7+RHJHP+2hqfugo/MvI2H/mqr4b9tFnKSRY1Y5Ek80Nm/WIhr1ikKnxGz9TWXrokf9xwujfvcOTtNTWnxd0F37Y2W79tteBqZ4G5qLCuomw+nSr28QESCRVLTyYKILGJOPfcnaIFOsewhRdvv+rWa/Wih0vlbX6Zb75T5C0qNKVFvH1QL/vazSWgC2s6oWXXIuUxQelKiJbowuJDQViatLmLijg9CQBMg8WiPgiw3LEeYRmm5f+XdnvkDnxLLjMLxtvX74C3OlwPQqx4xwIdpPx38LrlDphiyWUWHWKAzzxurS/xTo+P5wGFak62ap1PVFFN4v/y+xuR39WnIO7lsWfwgVsK17wxrs9K8ltIKuhkw7f/6dhK6gQokFKhWX3urrjk/rnI0pgfpGMeuQIUaEM7+GF5q2iMkCaMQwxxOzcvU0eXbsnS9XknXvP7Gtw5dwPXlFu2ecvSHEZgNDsU6x/GdXBYXyOQjzZReSedeEPY6nEv9gJR4oBQJtFO6Kd0fwC6BO4LNHDeBujB6dSNcUQC9zIv2LnAzGk99bUDrdFY+9yGFQtEo0GQPNv6vS2drj4+1jHbv3aJSMUWP+QTZrmbNTjU8wyG/iXNNpskybLcJ3CiTF5Ir+JYzmJwE0mSVhlxbtbmvweB3ulB6Til5UuUZydpgiFVeobhU0WaBqpJ198d+/XeNRTZ9/1OPfG7+2hwzd5W3D+hmyjsRcUg/+Cavb++Vh2ls3L7zT/etOnHNxeerv313vzLVqPai4nJv+K1FC6040/4udw7sAb3laSg0XCkAAs0npBO6VJabS4Elk/U+D4gTXW+j0wnrMlqNamq4tMIYB87tE10i0FR3LZNhJsb7/R561btmes8YBCRkhYNByRtKd55mqTas9FYhJnbRGHuOh3M4QTdgQSqmgRxuzGdSvZGcbMxNQGk5C3ebLjoXIOFM4l+WKHmLTJwRv9E8GWJ6dYvf/FmEyEGr+gyrr1p5zrgkz0Cw2j94Hv8Jdx7dIVegBSNtgsqGsRQEYiIBoXwD0LNvQ5d7s5Z00QzwNhqZA0b+tMG1tQq5nd84uq8R0zPvX35G8uRaze4jcOHzz0w1+Q2BIRvf6J6Kgatnrbiem+CFvAxfkrndzD9MFPP1GWTUHclpASUkCNAQkpCCcCgDSUDAhDZ+CuEkgn8J7i9nMA7pA4lISappxILKfAeSAbIcSDuN2bJcfZILqeO5rLs0MnngSHYRdrHjmaz7JEsEPw51ZqDJDmUIOZIe34WaQeegNsJn1qz8AIpT3yCjyEih/xELkuJ0lEMYTLVCiWpo5oYMleMH6USyYJcD+uOe+kWKpn1Qns34iyYDjkSLvgnZXcgVQNeqINXr48m3iS7cjm8tedyY0f1QvTnHHdsrKby/+SSbPY8/NH6vpl/Esq3Ae4ZU1HC44KFiI9o7CEgab/RqHbj7s5KAg06s39ZP/zxI/mVuF/TbTSy+3Fb8If9/cv7+wt91yy8RfP1QXtW5RzQn7qIiZyuFM5QfJ5E9uVnqT85TanFx0lkP3ukBAMprvsRyi/C8NAJL1xbIIirSvnSj4O5netb4JxmNANHPssHAcHMHsFRgEug816gDBeMbdfiuRcghqYcm0+Xxx/5IAEtN3fqFF3LzAXqwoT0PN0OVTNqxo8sxMkd5Ig6k79Zk7VxxX6gMLOZFQgvpW2RrMW1D0BDihaXQ9wVRoBxPLfpknmkeMtoB/qM9cRc9IqmMD2XUmdZ7GSRKPUZvChf8BoykriM2MnKYbOHX8R7cLdNCxSFFVQqoYswnlWtlFS2mNkhswVpZiQW1J/UKFfipHGlUkM6UKBhMz1istELIHJLMSctu3ugzfaVSOjKvUgc/THK4Sdg2Wscz69leKIkkrwuuWiOe9yGYKQXRumkC3qbRcMwrvhjNXgdZk3RxAUEhuSPvn3nnd++U/3vlVOmrJzCD8JLxV1OHRjrZifbcFDOuRNTGqdgQm1tSNJ2OcQ04YiEXuxtII1ECSQRoQGYioEsgCfchB4ghAtw7FfJre4WZ9hkVi9MtjuWqtdNDlpMrfEG9fOT6q21okg+e4As38MfGquNt7oUws6Ysarj1/efE+yst86YUVNvDdts3Pv5c8m/aP0C+f8/Qb+IMnGq09BgwN01oIOAnAdagI8mBSrqk1gxTDUBOtk2ousEtBH2z4Ir2d3f6k8PXXVlt2qN9RODxRuoJT/v27wm09jRYVc/e++iyx2tyzJb/n3J0htXP87eSsQaf2Ly0s6Zmxela88REy1cf4273mI3iXNJ7KxrZibOm9xm6rl4fqy/t27smU8tOfdW2ucBzg2UfmOIVyLIl3kpYlwphDISTXJXsctmiDtN7fNV6zelgxwnWxsVr83Aj/S5ki1jL/a0GC6+2L6Um+aoddlNFuj+bJ8mH/iaLh8I0/U51NspIEfq0dohwyFXKgm4NggwQ4rRhCOUFtxxo8XnitT4cnGfT93IS8FaT85XE3H5LMY4zIEPL1hw443wz+1UmhTJyJGxZzw+wsKkKZgUiVtKOKMEb2AKHTv61FNc01PQFwKnvsZ/9pPA4RKTASWahmh+8MxwzHxKy74IRn5LGRjsPUUwTu64UYNY38caqd7HKucZ/tHnODtENw/2UfHRMaq1UUPDJQ0OKkWCeet5fYOhII1VRz8+/Elg5j4Gxur3J8o2PJ4rg+2d08T/fwEzSVbyZ9XPro95T477lRKqUSRXQnauHNsISAl27oWi6Fv9z48JMv8r/aMMj8onCP/DuDZOuN+GPPr/+p7bx+7JlbYdppcNhzKU/1Px5aiaGDn/s1iGMaBcleKUo/v9rcxkZj7DBEKOfrayytXNLYiUdBY+pleQXdnscKlQcpzuWluxsieeyuXIK6SdxozitWyGOV3vOHHjguyCQ6fpIYy2JwvrQEF/Qa9Pdf/QqOSqCiE/EE1/XIVKTc2tzWbHnimrEd+Vyz311Ml3P0GVTj7PD5aDnsvCvH36alEaPMePcMegXs7x8igTu4B9v7G9vTHvhCu/kzIdx+BxC0ay9zRSvoS0F2lIxI+X7klU63I40gLQ3w5ep5na+SFnba3z5D64zv+QtM4n4ffG3tq4aNHGRfxgrXPMim+5487abL7xhdseIRn1KDl+7aINixdv0OD+JSPwKf5+xoP6aiTeQIDVlIhMcL1H5R9PYXvprs3fv2bO7MOplCmweuiq2JRZ1zz+9a/v2PH1Hfz9236w+ZrPXvWfAxlj4NLLHpq3c/PQ3uvmvbrjG7fe+o2y/cLdtE6VUlXi0ASb1VLUBVSUWSU4HdvAraTyS8xzM8NxvxFkXV6pUVRiJwcgC5zEeht4rwcp7ki0k41G0qlQhG1Vzlq8alEmnFi58caB5Q9vn988MLhqyVlHvLEWjtQFeupdiocF/tkkOGPW2ibWaBTkeZ/dvPWazXfOnnvL6jkRXpi85sFzZt+55ZptW3bl1cCCHZPD06MhySha7UFzjcjbp8fOecFCirzAG/yVjBX6OFIaadSjQq1nNhyIe8tVbaaSdHlXIWKacMeuZA1uxS95zILhyrxAdsXTL6m7kNQlx2P9uZf2qhufePFFbpI6/OU0WcP99RrCsrwseVot5mtytpf6Y0gm9sdeyKnPQ7onyK4nXlR/rg7H95M1upzu89DH6pgUcikoiihJ6NJKmRxV1x+MJiOA3YwhDRQrWU0u/0rvq0VYXnyCwsLeTJYBq3dAtJDavuzyoVpzZ99Z0+a0uoiFH/xcqgDR7rUFeOrUn6Cywb8ZeNMbhLV5ugP9l0zv9UN5b5mFkjzxUcpPJCn3V402pRxtJd2GrnLdhtVk9ZSZh9W91fCSH5B7ofxPiWL+j3D/uwhBRdyAyozeZwvQzs79soi+BKSnafLviZCcfrpBpLyimfLfTyJtbyruIQKD01tUwJyKEo/ybaxkSNFUMdMkhQoJyRBQFhnUkDQSXhTM+3NmY0EDM7ffLIjqWEGt8lCO6mLia3PukFnghosJD5p5SIho/VDkzQfLE+IrYoJXkD19pdP7OwG/voIUtagiWiZ4PAFTHHlTVhRZ7dYmPar+NJ+8JhmR6DFK5DV1foHoLNO/pHrvZfmWZ15RQlwvoVDKhCWNK3CCch9lfFBuAqUgpFSShmNaPj+i5++WZfKeViJfW5HnUakVL4UCNVkA4+ETfIqx4B5xSaP2L1yn0zn2ltPn4+OqZGmwwEVCaCSqG53ldtL1oLGAhdMLd09MpCCF6tD6ZnAZBY9hDaYsP0jzZ0j5ZjKsF4i1UmLuhbJMCnYJPt5VwFNvmZawXjEvLJqIH8STonZjq7BZ8gKgR20C9MDFqJAX1H64QW2NEup6qgzLP8cvppL/NNTOBTCJABOHeWoXzLhw4Wuy7gaBtjKr9kgKq8ZlRYBS32Lpxc8vIhpNDTfyNXWybMJbn2RyQ5EmWc2QF9wmSZ0KYCE+cPuYO6b15Uotj2Kd4MItLS7gtFbkTdrFND6pvEZqv5Yv7jXAus7Pg7avo7KDot50NX3CPkP+Kps8J9/3mGQIteY/LGPC+L7872SPR2br5fy8MtKBMHedGuM28/MZmPJMrGgi3Gb1S+Si1/L/zrZwO9XH1ce/z7ZQ1WSoY/+pMb5FT4ua0Wm+Jf/298nFmChEQ+Ti71est4mq9VYI6RsymoRJKYidElT2FGnDTZvqtfhGAFTbeqEw68GqtfmbVa/1IFO1/jdWr/8BDRRtQh9XNjubEm4aWVpVonpTGR7PVGc+KJNoBIWF7kYi4gUV3r1U6723i6TxUl3n3/tM27aZfKb7THiHW9VzFSwHJ05VfK6Ar7kaB0XgPPE0BSkSFKsBUpaLihEWoA9wBt8qirh2VSOkZwXEwyrxZ5jyt2rJmSo9gX7cg6jsEUGJU9z9xJPOEM3uQQxKgkh35DNATnVyrmJ3mbCNyIB/yox4wH1bg2DwN7q9kov4pFqny8oSm3RQbGgJ1QQTs6ZMLilOVYJ9v6Wha3HcJ9jddsXp9YhGUXLXt/qMDnvLpPNTXfNa60z5/yjXQOMq+lNmwh5egpYrdfZQZV9rI47xlRkuyTjpzsmCBSWNkAXVoK8sgYWqQJWbo1RLo6QH0YW6pxqfCnRgkd+RiFjUQUQ7poIaYoakgXxwFd9BuuI38H1xBxXSFb/pBDIKQFn7YB3dB36l7sG1FLaKiBdp1KxLvfswap/30lnVESgNnvjbUoT6w9N+Xoio0qcYOIM+heg940YimsucQVvli9NEcft2UZwGQwLuilj1fFr1i3NP94X+PE7Hpvtj6lBJfJ4R6NvWiaL6MgzWHxiN66DExa+dAdAbMYX6HVF8A+7rjEZIXAVbDe7PVI9rmN69JOLV1DOSvRPxWNPZBZf/Nf+Ny65BhYxxxV+77XJ2wfQ389/IQPgajXbwMsuAz/0IaQcXJavKbRqR2IqyZruXjVC2+hdee/5vdnYOedpmVtR3NGXldxSzDSIiBVpkGb9by89UpEPKrSLZmyFDzMab/wXl2CNe7s/qCtTvWgG5kpBmCBlSzDS/r8N4uwBwohRW63JTS1y32f0TQsPfXVGEHQrV8/NCfiOUVirYcBbIeA2+iF68rQIo3B/S628vYESr79ehzS7Q9LEL9UXmik9XVHb1yBO3Ngvt5935+k1efkV51mzzrM0LL3/20avnwMeKuWyOUZg2TasSqZ+KcZQiOn1Iu2Vh497ALUVZiCKt/gh6IvTIj1ZLRjWAkpHKOKovNwp00eqPROiAbiNEKieXwMLcXhVJ1/uzmLP4tfxaHR59cBdJVG1kTAgl9ze9QKUEQ946Hkb+okJ5JRDyf54Axur1D+WS49cLr0tTPEu7UmXrxcSr3XNvumv4yXzInXKH4F7Tc7p17Zt+t/qW2+93k063X7VW6lALxTY7i1nBXMxcxmzQbabxz+tJo+wijYaIGMNS8AoSMgAPt84DdHOoMPfjXhF+kuH1tZvuFQrRCN07xGcXRX9MYxYchDe5BcHj+Z4i+42WyPc8Xofi7bbZJN5nJLJ5qr6IqRtzqNlM17SpFsnkEyTWoABEjz4JXOQvzWYuwdnV5LNGOwTM5v9r4RpQ8ZXsYodks3o31JBlzbYtNotisnm22MxiwGFXam5oN1n0TA/hRvshvTSDwHff4nNzRo9Dum6PaJbMXzDz+x+Fkj4L4bFNBb1asqsgH7Dyh4DvbkPtf5yMDKzEwyoaESMSNS9P9gJVA3/RTlwoMwZvxECFWxIPNw9gi01nOHjP32esZTtmXHnxvZd8ZtakqQ7ekajbXetpNa6ocTVxJtY+uSe69OLz77zh5bDR3xjZMzUz6fxrz1nqrZGcHQHfPVefN+fiK86LeXj+Sc5lPKy+k/vCUI/DaLFYCWHr6nbXuILTIsb5imNKY/rCm28fSMxPhkN1XbNMNZGuqwOBhtTSxWuTk6bw0ZaG86b1hKddePOKuBvmiguYBn4T/yOqOyGRBt7bKUI1GjioBC8aUKwF7Q319UgcmtFGIzCJGBqwQij0ynDsfdFGc3TS3BlNfJ25xmzniMkpXXTPvCaD3ZaZvyzjmZdudBostmhb0ORZNN2sJBeed1HXkrUsywueQH+L0eCPxmsa5ZpgRJSDZ11yDv+jmbd86vxZfc1WcZJ3UkMq1BOOOVtvu/+pB+en186d3GTwWAw2jheaJs09/+LNfZft37DALyrNj1wABMuUKbODyTVnT/KYbJ3Tpq8IrNh92dkxOj5P/YpZx4/ycyiVcDYdn4JbEoKdQi9054iBKsygLW46FRGxAb0NPNCm8BSNCPjoKcj6EAus4SuP3rB+cV99/eTF6294dA8+TK6v74MHVpYNRt/I30e8QGTOOdfGWzzxcy+87a7bLjw37rHw1nPzp0KyyRSeZO+QQhInt3dYgvycjrPOv+T8s1rptaP84VeywdWX2T4ysr0/7TLIs6+x9zib56ye1dM9e/XsZmePY3NDs9zlnNVt4+WgHJbbz3Livg4P9WWgviOMm4kCRT6I8vw0NbUUEnFvOuFKoxQW1gTsvFirsF5pb7qTUCx4i7VmtToveaDxvK9uOaedVvPRpVOnNz0Q6bry7uiSdQ8t7Vy4JQKVS+XPplV2ts4bvCwZu+KzgITtxepaPRzWdpv74muvv6RO0SorX6cu/dqKn/XWnrtp/Zragz13DUCl5myiFW2Ycvb0PtsXnU+tx8pvLFbUspLX68mdegwmOif/NPDONajTGoUh6tU56HBJCTBASVvNUB5VIiKpc9kd7kludodSFz7xQbiOmMk5dOYk56gzL6uaf7N8a6MQOHm0ae6snZpFDfuT3/jdYzjzwkXXIVHoXNuCfQslQZqBZjTsoHMqrkE4jaYdgkGz2ATOgB3cPkSukD01DnV3ttb1wx+6arPqbkcNAHoFPzKUUQ+qL0k97pjbZv1I/egC9zTFbrrlFpNdmea+gIgfWW3wqkcis8ky5FAcRd1If5nNZrl2FFpungc8wpoCl1BpQV/ScS+zjlASyUTVv/AJ46gkJI4bHX4lTnloctxPZE1ckS3+jG2fKIjkQFyzuo8jvYQG1OrGvJPSTu/nSp9PHNTl4z5hK/8gtXVKF6gEKiglgcKiRlCESsQCV5QIlKWKpr34lt/wkSx/JCmP5/cBKQfl/5gd+rOS/+p91/+YCg5CXK2W4M9fu+/6xxX+vnelVuldIDCG0VQTpU9Dw4pRfei+6zWx0MLie0gPbyrkmRU7OwT16JGeyXLHqOLqAfVN1GPlBzWtFNzj0TRTCjogtP1NjIvu5habN5Aoa1k66wGpqriVetJgiGdwDZtKhnN0y4n9sXYnsqGmZfDSR15+5NLBlhoDaedEm7sxmpqRija6ZEEg2EAnTiAC8IrmFbGz1q08P9PSkjl/5bqzYqT9hMmptEXDgTqP3Wiye+sD4Wir4jCeoHbbp5hRfpB7BakUIppIlPCD30dR1GtslDz8OsqbXmejFC/v8wu5X2myq7SJ8Avzv9DFUJySf5uNvq4+Ti7W9D/OZrLChdwxmPNiBRqVjnpK/aGxRCDspVYKAW9AN1JANoo8wP4BJUlGqdgw6m1qPQ2QW3+OfU5/ieLS/NuKpDU3uf8bcAXyBal5jMR2NEAbPAZt0K3hvxHBEDlUxfIGcD+N2gNSNx36nfqlAYow0puatNpRz0e4W2oahKzQHsjf2c16ad/3t2KTtPobnX6D8C8pd0MDP+Kx7wnXqGGlLQcvikMErm6TmfsuxJXbSAxqNjOogJLQBLiKEHAE+JGTS3JoEhTrz8/CB+5YlupJ58aOat8Kv4JvregxwcU5Cp8GFAFm1FyOfto6GS2m1NGTS6CPNKkbsTdCBlnN9onMho55BX8IJZtEQ35lk+htwN5A0V3RCPoD/yXAcv6pAtbZczRUA64JmcUf4q7Q89ZHLeJVZ5D1Ps/t+0iCT3AHVtZC7JDCXfR7OSb/Xja5H3zQbZL1B+ULX1BMTEk3AseSpmnKEK4T9ekMIidUCRQFfcbj7z8gNLvzF7mbhQN8h6ZbRset+nQWdS/ZX3k7WpS8P9sfo0iGS64wV516pOhjI6TZ2dApgI5+LhxywYoWxKUrykKJsIoDsR4mSrCTg0egMPnLW/3Q5Nn8BZEuzqEI7HK3n0+zFmuO3TtWQ5WJoG9YqCD6Gc32SxnbnVPfsxvrFXK2dILl7bLthDp6glhcsfp4bYvbSmj/mQ94uBTw0E73x2jbNRCvC6VL6GCFDwU7eWQDcC5FY5s0slieRDwtAbRsbLXbaXAuu14e2OJw1dc6jQ3ZdY8v7rv2/BWZLqvFWVvvcmwZkK9f5jS4muO9yR5res4kfkRxhV03L1RfPOiPtYi8pd7jNEsOpyTwxpaY/yCZu/Amd5Or9uS3DYaeqVOhH7gZN/8I/wi1fEuLXvyNivibjuKvN+1Nc01HF/3h+ef/sOhox8MPd5SFucPjorQwXT+ytA8EmA5mamHNFDVhBI5pjZbQpugBNkO8MvRub8KVDKST1Wag7D3xlin1ZF7LFP/79nbvCXFOY+PUjrT7/otsPXXZ4exdPzuhZuL5LUXVAn7k7PbhG89uz3b41X01gbjP1xwlu5rrvvf9+pbs6E/Vu7Nk642/PYRaAiUBdrmO6CDTBLPQFA1ur0uXoBR1INDMkypKpoTqnSMx5GiEdTEaSHLs0Alvu/19/5QW9Rv1U1ridT22i+53pzumbs+XFFXYC++CGsTj5JUT/GCgRt3n78i2n71FHG4/u6X++9+raya7os3ZbDmgWfXun44e+u2NZKuGZ0HiF8M4TlMPR+EU6rPKRJ8wOU2RFUFLex3egEsz3YqEAq0cqhAAW19dBZIlVzR61tuIdTnpXH7l+uXrbjPUyep+8cl6aXKWhPHpDcXl9KiTWDNr4mBQc8Tq+NzK/OKSbsfl79o9G20R+brBXYvUg0rLHhtrc4TN81TTOWSZ0gL1ZVlOYH2ery/7XVUjFMbzYpg7UswcqJPQwBd0LKLabJ8IaCr2otcjSkIrGwootKECaUd4XH1+SdazRrfddkBU98t1htvWrbjqSqjaCguxrffM/5zDCpBALUycmajhd+R6ww4SWafuZ5eU+tPid4lgd3gt+b/Y9rQoZNmiXYPXyRHbRs8zX/f4WIFjWZJtUdSD55AP3xtXH+ZipC0EqdBGDA4CoYEU6gRLGPU11QhkLTBiEYPiqOeQgwTCl9aok1Qr5pFf71qEeNxjy/8F0GoqYPv75Yh9j3x4DuJ+uEzHRpAq2lMqb+qfTdiq6kGtzfOWsv0c7lSeMXDHBDe1MT+LUgx0Pg/p87u2UicdIvqQi8DkxhcUwUXCedMpb4NQjwY3npTmgsURJavLwCRyEcN2HfWsDVGfv/u9ZUWUx+PYFueUKwaNvbtu+Xps3eVWbN1GcgVrdMnWJ7WmJz9SD66EBidag0NF1Ukep0t5A7sFCWdhzvYwHv6L/BehXuHqfaBwBEU7hfVLcXvS4VQv+T/vaSIl7cbeMc7ekv9i8S3e1L5xxpvMGcu1EYPbKyCiijjGXcDKckm43PqU2qNWlXusZMiqF82cuVzolUHN9NNR0HZPxFPV9V0wLtvq+k4DqOwVWDlzuQLVdqFiP08cRX7aRlBVfR8cb55bWe5LExnlcsDp1vAP8Q9BucPMk1Ulh4GnN0SAdxcNHv3q9ohx1Ati4S/tkWjIDe3hQdkUGrGRaFBiUdiTSkI41UkMuuQHP+EaSQYlPQTFWJF03BNPpTu5KFAdkWgDukzsZKMG0Q1TAQQglScOaP/dsZ8+fP75D/9Uu5Gs3FY/2SxPld0DHOciXI9gqjcEidXjE+3BLosy0OcX3T7O5g65ROGyzQ2BZs7WbZVnO5ydLe32hMwTQ4wnnKXW6XW5LAa7oaXOIHoUl0FgLQLH2by8wSTWeAx2Y5PDazK3BqZbeJZwXGPaYhX87ZNszoDdaRxotXO1nNlpdvAPFWHDm8PqEE0sZxDEqGzxisFNnuCWetPcGrObN0p23tTZwMuRVodSV8+LTrOV3eRvzjQZiSjaLYS1WEJe0kNsJlZu9LFun7++wW4gRDRbaxw2nrOGm+xOj9cmtbp9ZqeTM1m8UXfQQCSTVSQox6pvtjot/FpHvIUjJovFEoYvHYV9C5Y/xN9OfcalvII37UEhTbTg/AQIaPb4Vz6j5u8/aViycMod/fkDcpu8QZbZoeBi/vbzP3XPsZvOubMtaPHkD9jt6+U2O7vqU/9C9SMvgrXpQNG/E0oJxun+CiElUa0IKQSUwERxOntKSV7ekcuh9VBZBBo3VUcB58ofKBHCwLyf9qFosz9Ibf8dGqwaBMjRig4SGOZ2UkWI7UiO9OfUPdxOYFApUZyfpY7mgEc5rtNGGk2H1lPhAk1Hp/VAMqQEHEUfEYkkUQq1JMdzsX7kklRrTrUi1wMcDjmu1YYfATj7Y+pGpPEBXuoQIj8rR9mgCl4C9yqmF7xnVWxGVniNqtpVmXBvQ6iwni5YQ8a1jYrXtc2J13HvgkvqWxuva1sbr+P2S5ceKGyBwDv2DbrToe1u6BkAJV7xnVLUaq0sJB8pFqcUIPi3yuwxi4JuLr+P30f3OkPQ72aO0xYo3/EsmO3QO5qEF8S0qQH0UsKXv0brnl9+8M7jF174+DsfvPOl1au/RL5/9DsbNnwHL2pHR1NTRxMZhJtHktOOxLxErPF6YlLvpC9YP73x+4ofw+3xVdrHcDE0dQQCmCRgvt9b35xINDf1CDcRSfJ+pYl+Sf8YcurfmXP5F/kj6J82jNsrkWiEuhVlgFfyNkB3S5MUzLhoNiwSCYcxQ7Ui4J0Xh7fmqRbaPa1tzujxkBRlsEHy0/OM4pYLPb7g9O6BQJN6l9zQ0OGyCaZz0vMTbHOzXfQ7a2tsterTcqxeInODoemdktw+1SbVhKwtW9ffe8VKadK0OVuC3bWzyKm5LeddsWTeorWyY9IMtUFutdu5g+Rn533qkocdvLs2HmhU75br/MmWtD8zA3OP2t1ea636jEzqYxJZGAwFiDEd61oTsrRuW3/3pYNi3bS+Rd+GjOfVpAPNd6y64Gsz1GaZleWIPoYL/v9mTeQBENVEguiF1aC4YeXxFETw6QyPfn0m9g8IrMFAvKM1EI11DARnbqibHk/Iojy5rSdgCyZi06y8sS024PeuO4MfwQ5Y9yKRZCqyYaF30vzeHlmUprR21tR0t0yz8KZY66zWuGvxVQB/36kP+K38t2Hu6NQ9SFJfw0AdpqPEK2qTMpf2VCqJwqPoJezTL824b8akoL+x03nhh+oNo5e77psxg9Q5LzebIKD+fsY34f2MtB9fk9v5b8PT6tYrgv4kRPwd0q9z3gdJSJ0653KjCYPwCaR5aUY63eW48O/kdo33yxX9wCiMv2QTrk8eGSI6Ag6moG9t2P/F7GRNlDjl0gw7pJ5aOXXqyqn8SENnXBmbSwUYLyqJjv3UmY1nKr4t80no0faXsaIEiF/BRaIBnItSce4OUif7W6Vm9T9H1X9Vj71BEm+RdmIJQST/ZfVdudUvh9S/qqNvqT98g9SQ3lHibZY0mRVHooyDN/FHmTgzjdozKw28NwQ0hwN6BCoPKaEk3YtKwNhwRLXuk076CGoZNXDQcRwZvreTZY9EZi+d0s4+ztv8iei04JQl6ZbDD2eHV7X4uHuFVfPrOmcs6m6Kr7hssr+1VZFcEZ/PdJkn1hOs8SXS/NFFgqt94PIZzZ3tdaL6Q5vo6piSzdy737pwsX1VyxUrF15iJ4uNkq+rbyg1Z+O8VsNC1UmcvORPRfxtPrfRwL2p/oA1eZp6Z/aGffoewaXcA/xBlKlQLfhQL/oPgBGP3qsA7IQS8qDVNswHKRSheDUvA3Q7MZoRcJMxlEygujn1QdyzfPfq3dEp/bXh5e5YXW2Ngfvza0ZF6UgFL/E0fTq4LBlvTE2qb/KuuzYSXVnjTfM1osvqMHVbm9950quIZlbqaL6YP7jk3kUtA0GnX2nvq53f3WoSsvEdDRnULgo2fN7lNZJgI8/VWi33c3bBZnGY05+dm+3qc7fNmj4YGKLj2nfqFP+g7jdDlxEV5XsJQZP6hYrS1l0VQr4c69Xueixp90gnZPmE5OF22j+SYEWHlZ0K/Hgsh/Ztsbh6h2DNRlvv6jJh9XaJaHCZDiUDKNTMkvb8vsqCyf3ZNdSmO0fa0Y4baJTtpbKzuVzeeSI7fCKr2Z0WypapnXJ4gnoWy3PoUIlIQ1TXdqhQJIXp9Wx5fYdpeWh2TY5D+YVyKd0jw3iumwi/BC3cEy4o83QlZnW79MrCgCjbhWXBlRZVVZZv4rIKpXC01HFlHdHLoeWVl6UVc/J5uGm6CViW5mulYMk+HqNYr0AyUPivLg2oMs2MPqtuhHyRyiwvNJej1Br+fcLyoAyu8D9B7bgmzUqfFobF5nKnK4+t8MPJkI/xHUNWk117jugWF+xazTAALQn6+UE9lhoI5ApGA/iuJOsrlNP28SVVuBVajXmircLel46w2bJS1Q0Ft0KDuikDFL/3pYrid1Q4FvofwRIo4R9h2ftSwc6jHAMqLcCql8YPHtlzGoByNXYN6v8hXnRaOhUvx0sVLCexwupGDR4NOYC7PePa5keIPACnuAdD7dEadRuTIiS6Lb7uskb381My5yjzF8lGCjBRqdwrWJCagfB3yCy7XT1i92hbcZ5Ci1FJkgYMDf6n+jspIsHFjJrTOdzSMuOa9DbDcj/nH9N9bIoGVgzHPWIQuFuYtaMRaq8eCKI0gEF6lPOZjBz3EEvaaxwSUT9U/8JbJZPJJLBLolH1La/RbF9AbC8JJjv/mMnssKjLRBJyqj9QXxNko0Ux/X79epfiXkm6fmKwF/en1HLc6LxloXWKvGa5rVCVL83VuiPcDEX/K5pTXOxHfx6HHB0t2FI0qI2rCZFTrvPWU67zVuS/kTsLnc7IKhFg30e4FOkqNSfH5PtkmUy6Cpiv/36k2sbqCeCFNa+URpoY0sZoYmCgCr3qgZz6s8I0gP1bYiR+D79H56NOz0EVWCTy2/fffvSCCx59W7uRV9995eqrX8GLesOXNm360iZ+T/El3uZqL+FyzSZ8XxpTiI/G0nkT4zznFZ0t4ipMz5v4q9ssqbdKUZt6u82knPCrt6PZwsnn0XySVnyPR1ZXAn72yx48bWJsu7apnI3Hy8bygUK5Js32qcytapqgmn95uexccj205vGgJ+euOeG2SORmKZr/qKzcx9SFctMJdwMUFZDJITs7dnOp1EKZCxg304Cevyfya+vlKqv6aXK1qIj3imL+L6hL+yvUlFfE0VKZ7E8gBY3M/8VoJCFgizH1W6VyC76nH6b7jiibYVxUmVIEspry/LgZIlCeP11Z4zs/AwvVwtGFEut5S1JY4lfyT0N/evOLo+rUEgjcqc9IkGpQbv3iW7Co5b+KgjvpzYdH85PLcc4X21ouwEGl/S4qnUAvoSlXUUhR1eKr2VWFTB+GMl6FsiQsVD1R3urlAAIoSn7JQkmiVVCHSpCwDH/qPepXQ0Db77CJOAImohB+RPWr31ev5g/kE+zTa4lbvZo8xdWPffQu9yJTPCNB66s+zXoJt/0L6hSoCuBIoK8fnBGG87OoRckJpLqyWe4YbpGi50g0+3I3UD85Oa0fzubfoXxPLbW3FDWzigmyJeM0tQkax7PqTy80+UxfUHPlBZIRVNQ+v0xRm8REKPoLmNr0+Uo48v9GFbXPKylqQ2IKm00QddgyWGMROCTxdLB9nCY8P7j2DjlsV/+mfr0C0r/NkeXbbpPlOTBBwT0mVz1zx9S/wJecBF9Wgv3p032iP2v4VSgfgW2G+HUEdEXU6iq4CtpLJfIN9XQG8dwa1VoO8XC2SrPDDyCOQptXgbcPvlAgBfxBoGwftQKeKFrNTASPt3pGGqDt/QRasn2kri+H6L80MJRsmVYJrAKyDItpJUy3/15WYIJqcJ9Q5N/LFJ4c3dc1URpWl9hW6mu50MUIelg4ucTPf15zs5DFo1c0VSp1tKB9jkwIyuM45kb+IP8gHed+6jO3v0KbIknzLy636E8KPTdCuUpB0wLo9JKnAO6pv0vS31EtBha/fJemkgLVVnd8KCk4qBTpQ5m7FbifBKrPJcq0pZAFVG/XbOFz+Tcq2MLrcmV28Nmi/OHskh82bau0k8eWCaPijQPWQ5lUvslwVCfHkXBMIehqUgtDNLeauH1huvZTbYmw+luPjyWoNGEuxRLR7LK5fSyXFUyK7PURQv2v8D3XOt2NJ6liBbmPGOsakw1kbeOs+31Wm5qpH+iJWSzqdPr2O7zc2TmtnrzCig6bBd/vgQmzOlz0STWIlmZEQfupogOZFHUZ7EkUnMn0RrpIMqAgHRJAOjIJ3yGw1I/MAp9q9S3Q/clADNm1wEeO+xbwg5OIYHZLY3ehG5lJk2xhco+6JWybpEVz2wrR6hZyD0QXZbeDVB+onmlimpkWprdAs4WEZDSQppsDlcdCBJJESIYFuAtUnC4GIF2C3Uu2Kv7L1bdz6FxtqxpG4TqQOqOUNAJ2HLvPWA2GgDy4O4vaDrtyl6P+1fAll+SyFcQ28GHqh7fvvf37udylf0fNwhzgz87Y+cf5x9GnF6ygHu18sAbipWeF0YPBgp2GaKeQduxxdEr3SgbH1kvH7tvqSLhedomOvZyts2dw8acu3dY/f+ucuMtCuP/e4zC4XnH3OLZ8ZuxTWxy8dJfU5dhDeKPSlJy5pn/+7u3XrJhmr9C5CuleGflGQocKnlAUaRKp0BAHV0ZwUt9VCqk6zYOgRIuMfePJzdmBdpPJ7/6B23+f+sp9NMDZevovvfYHG5dGPISQq1DojqNckchVrCcCYz/Q0hI0m3NKDRfkgsrnamo+p0CAq1FyvC3a3Nak/s5VX282x9Ufy3E39VAx6o7LpCvO2wK+ch9jNqpJCutcIOooKnYWtDK8gTRVYygRQfwgzKM5+jP2jOZdx3r32Py7rQUPOzAnoRs95NvRAR0qLGU11Taqu1bUYSzMcWjMEir067JQQHfIrLBHsrgv00/Wavd8HRLMEEYFSW3HCSNQehnrHztKqHcDyo4VfZ6gPKCR+gufwA8GegxUEo4A+gd0BASHiH6jYMLIsUdQJTs/C641KN4oCHWolCMLlMfIdtWKScjx7SM5LD9HnfmhrGI0S139UWfUnxgOXdJFW+AMcGjKr6eHAttHF5sUoeArYKDcxMSYcKA/xUDhPiEOEAPafSIUFArN0r24ynI91EPARDXvIDYyvqZaWeroBOUABQA/E+DXC7PWafDLQY2oiwpUEyj4RQtVlUp1GrM7In2p2A7VuiOW6otMiGOo5Mrp05ejVuTy6dNX/k/7mybZQ0nUmfrbx3U4KueDnlHm5wdh8FFeKnoaKKh/TK18StOPhwG9Xo5mqXAxvw/79YQwwDR+nAKQQ4izVXioB84qcppWB7IqjU45z4CE17OvF1Dw+oTFqxtz8dxwtogBnF9MjIl/in+K8s3hM9laIn0TiCbTAXL0T798bPXqx36p3chrv0O+GC9Xaj48Ecv8U8UEeBvUEsDlTepiU5OvlpeNGvpnKF0RvUooWhIjnx6GeBapXCQYTw9DNg6/OC3gZjp76oNTj9Kz6Jqobxb9NDqc08vcKReOpcsQV2K8InXFaXW3aI6Ofr1k48rp7CX7rx+v1UKPsfvzQU0Kc83i2VdILmd2/yX55zT9luN2+Cu4nKfwPcK/CvDVU+pHh8+LaldIf1fA5h3ndT6Fln9/W/9Ce1vndfvJtnPVO2xhm3qbafHVCN1X363UXHq9xuVD8OSD29Z8pZ5cZrern9cAdGW/uib/ud+VK0L9a42r6C90kL8KzxwLQw9NkIQJL0ASU8M+VG0KsUdgdvpgP/6NqqP0/gHZFUfGEijZLHpiIgvV5/Bltrj8Qd7XQd5p4P+7tJo30NMO6VGBwahSPMYiaaBYoLY6uEnciyhhh1Z/vvacG/rjpsvnpzs0B1Id6fmX8119l88XnOxe/uGrzzHcdu7UtY3+2vmXN5zUyj3ZcPl8p1sZSs6/nGXtwrV7Ka0XZdz83fwjjINpZWYw85lL8BRK4nGyIir2RiOsEyipuEcIakpGjWgBjLiHWOgj0Yi34gW1kKPxHt2Na5q+lwg1RdRSpFDNzosb44YJXnAfoEOpZW//6u1lhYA6leevezbI26zNHO811M2dc5HFxpk4i1jPC0s21/BWW5DnPQbn2X1WK43/aM2n18DfSoybbNHijFpamzXI31eRibGUOxSu/lT96YZlq1Yt20DaSBuG6knw2eusHs5EPBfNmVvHKdaQzcDfz9ZsXmLDWGXy2U5OsYSsIn8CS12jQIyD12KKqZrLPy7mSPdICmd6WGHG8NDZkkHuE4h9TU8FpmUO/VjC/EinToFyoNDz2p9XD6g78WgQdPG7Z3R0T/Z5dTM9lsL8Ktek7szl2L+gQwGgwkZHc2g5Su7NvVqwGy2Ua4KSXUwt1X4PaM5paaEu6jQ5zVFyNabxvUksVt2T/4VeamYPlLtffdQsk+2sUTY/zDXl/05W53/Bz9UK3p7LjapZ2ZxOm+UlZXrL3HHGqO8+wVroDaCTTnTxitMxmiAAYQzVJQH+nj3oIHnPaN6Zq6sNSLjBl8tKgVr2mj/9CWi9dnKca8rBQBsd5R1tzVlgrl5pbnPw6kZclCr2CHxMnHohLz+3KRQokzALyeIKFU1TNCiayJdoHvDYe7K6mZLm8S3uJ9dojuaJ62/qN/tjQxnSnhnKPw+LNrLi8ZKyJ3x1YhiI1aNAtP6NzCGzYv3DmaGh/LvQZnt0evgIhTFV0kE/PYxAnOHhCQUZdCWY5JWJwMzlAGl1mpNbDU7yyGnhRMILsYhH3VRAijrPcBU8/Cj1Y9NY6cnGVW0CjTLaz7E3epvaT/LtTV72Rs+0WVVmd0dz/MGTI5F0OsIviaqDlbbO5X6xT3PeXbXHRtf/z+fdka+eKPr8KF7IF4vBsT9MFPuPJMBTBMq9hQxXelQ+bewnf18ap4Ib+mSMrtDU5zqlD8QANa5MBGh/OwOvSDfcV2d66mfEWsbGWmIz6nsyZDWQSmqmxDneYyvjHPmRXHZxeueyRGLZzvRioKnGto9nIPkibAJA16adcOZRQr1iAP3bUyBR7T4RgAWTKxhkCYFwshq+7iV9r0whk50cmRcTg4fy5x4OmmNkHndIA2+YuMbmE9dwGYB4KFTsvnDE6Ah47r/fE3AYI+oXADpkdlENcZ8OZEEf8FFGZNxMs6ZLpG3SUFLL7Q2kcFU/A/Jsw+vWDa/7emewLaoeibaF1B9qUNnuqWK3+UfXYVL1v/omD15xxeDkPnXTOKSVcCbDGtOu0YQNpGAP7U1HU58UrqGu8xIbHtkQ3LVhb7Dx46ET3Ffcm1q0YcOizNmf3bC3VjWfAcpSv3MyTlgJ23FHQgmgvk+gk8pL0mcCDOn08MDAQlf+/SlTZ1z12fnqntOhbOTL9/ZdevbAPN+yby1f/uUtC/ixm8ZBo59LTXEW060hGrTDplNprWd58fwB/b/E27BdS/s7U+rGVCeQ46nzaw9QccnmZerGZZs3Yw9aVHt+Kh6HN4ti6lxIhT/wahnZtWwzlY9QHQ2c79C+dxzvVDKy8GqKWQERO9YAKbpsDUTLdWV5dE8PVPjvj9pqw7ah/PFVtkit7aj6G5xY9mfJrCz1j1e0BcnPol4UjtrCdbahIVtd2HaURujnFJR8CuOuUUfhrGhgKKgjCYNSvCc1WKlEp8wHUaAYynFNyzZn+2MnYv36dbMDBTonl/T/ma5IKAyEGz+4eRnVtaX6tss2o34u8mWorFtuFgm4A6qK/yp/gLEBVat5WnPDdKA574ubuFJ/IUfZ/Y2Nt6mN+ZNNTSTaeI56gKwkXerTe9DDHUw8/H35FY3nNN7GGuBKWhrV9ep+0k1WjNWVaHkW1yA+QHWNu8rtBw2a5YXuE40rs7/GA+j09V3hA98yRnFPOGr8ltGlsFdD/7tRce3LH6Trcneuiy7K7J3khKu+3qUaXPWaX7T6/Kfj9BX2eZq2XAcZT79u1ClJzUtHUqfqSMWBcZS43Ena0cUGLgpkKxB1QM+0Fxz10wgg6r5rltnFpH05pepUq3Y2HfYqeKRntmUFNz+XmcOs1H31U6cC6RTVLfCg7RNBF1UF2/wBgu0fFQtPEU1sSg3VcNsR7dWq3af87tUFn1l3ltXpaJxpNvtcZkH2WmMst3JqRpxUH+WC0E1qOGtP66s1MYv+VLu8/XFXvV/ZbunYYBeVN64ls0ur6NzpV9xzlmQwB5qC4Tq70WC0tk8dWJXeHvkD0h9zJOM0vD86/1NJMaIAolctvlByferCsqOKDKceOfUu1PsmoFCamV5mCrMUOCi6V6FJosMF22AcrKJgQDVhfYh6tepp/lYgvnCEAbJQ1L0rOpajEmRcasMiPfxhgGoVo4rwreQpV6fUJHH2e8fa1s2c13Apl1b89a58ozdoap2sjgLN9uISl7P1DrulyeIkt0zr6JjWocoPOZsaXPb6jtqBblsgsaRre2xHi4nELm0MhG1+x1SXwLpFi53b+aHRYo/IrbZtuWAKu5cSEXfybnnmUCaXGTpQr0xK2O2WWY76f+nAjNVf7nCZHU5XqIkTnpt6VtvsFlPXg1031g/VRdpkkyVpD7jnmax88QwDvg/66NnMRdRXTcGTmQc3cuINwN5IQqi0yzb+YFVHuVqI5s4ADfg5oE4ybDLd28mFSFmYvRoomsWXEdLU2Wl3GJy93ZNb/d5gqmNaqJZSO1l6PVRy0nZIj/45EetjLguh1rLqR+SK0hO6NrsqcNX8zoUdjQYDJ7tb4os6+i+Y0qpY2AWlnLRDWdGFTfGY1gV0zNAtJ7pdo24se0D88AwLY/gZmE9iuP4V5v7CSR/RThaHLh+UeBkXwU6BC7lGOevK65udTv+tS/PfW7qj3ljTcj3b9OkbV85t8xsMj7Ddj7DGpthZKwKPvso/c/1K9aLE12fMWLV1y1D9ua8lyJdWXr/bG+noCFutf/mLILe39ITUV4igr3876fpX5g2zeB52sWnIL4fXHlgeUzOx5QfIvJQyrKQE9wHUqVq+PEaOrz0wVvNbJZVSfsuMzxN4l9PkedFzw9V5Dj+nzpgoT4ZxCxJfC5RWLc74YVHxKlExCYt0JAOMatREhHBSCAtSfod6x6Ls8HCWECLwXZ9nd5Dz1T24JUdWs6fU3++fcnT49Qe+kBs+wdsMZgPXMp3U5S958snPP/EE7bvkOPCuTUDTUQ/UzirLhML9yPahoe1D5Fj5jWsaoveyP00PehdUAHk/seDVWsvDWXXXsyn/4wfpXc2V3/Qxli3jl/5hj/83avSCfpTNxOEKLmTjxOEKuxgNlsQn0xgct724mhynupNW1Ph6o3RYS3/+2TJrzLlkFz+ip3qCHKf6eqW02QJLjBYuuj4sobhCWqa/YHGEHpcnumuWSOhxeaL7sOakNR6vvmo+YcfFA8UFXEPZf9UjyudIOyNwx/i90DdsujS/FX2UAwvWSVK4NxaMhAGw3oowp/uc8CTi7D2rBgZWwb/60faR7SPsEbjkXy4G0XaqhXPwe2cePjxjxuHD6ssQuR1fq6PF0E+o2t1nePTn8TUmxz/A3crMoCc7egESuoTHYc7mYdg6etORoOhR7BBGD+qJopELrl4S6cJNRtEAsLP/OdvnJq0Wo0GolY2Et9VFB2Kf+4bZvVyxfOMz3WdFfSIryj6DwWghre7aQbdiDrkTL3A3vNDuDpk93HqXwam+bWmUJZfNn5ozKV5Pmmq8PF/jVY+2Tlk2M2RzSXKjmbQ4RZcQavEYrN/9rlXwtIQqzxQNMzPPfHYLvuPoO9TbT8bpGw5CQPGd+SyX/Cyf0Vxjd2R9NmsunnXYa8xGHzn+sSfM5J0y0DZEXWWxkXjcR75KBLNLHi7XvX2G8VOrf4Ykg0AMdBESIpo7MgAfyakA6rkqpI6UjNs0px7cMV+D5BF49Tez1VGnYmq0WIijp985m4Sn2gJR9b07riPPFo97OYbUZbxJCpot7H/lpZBicglCPN7WOfJkcHqc3ElWqvvz/1E6bIQrG+tz6WkM1SM9FBTR7FSs8KyBBytSmNEoquJNFN5EQyTiCrnKDx1h58yxCepPHU5nxGoxEQeeOZi2m80DxNxncVhr6BmEfUarxejw+WSiHhWk19bSY7aKR5MsteblJpfTLtjimBouXsm3d3djjYM+wEW0El9dM/ueVRWIsXwe43R7SgbVZqrnqoJ1X/kuF7pcgf8duv4q6vayV5U9zMV91GxO59UUjW8rHV6u799WzKMT7umRCXbYUKM+foaCcwgaoqZUtmodV3p+X7akb4dnU9B9La38RPFUG2SCC90tVA4XwEFhyOpZZrUCsgWYHsczLFBBVGNtstoN1bw0Z+O4fYIbvZVt4EUcJEKOhHeincWqONw+q6w5Go+WGOSR7LhKV+KBqbBPpfUvOf9QqkpDyVhBeyyZQGMsdA5FBUqvFMtUyGq9vjnsAJU4UcrxldP1CCaofyDkSAifoP5QwWx+SyUGxp75BzGAvtG7uQ38LehlyEQMeh0TeE6Bm7tYdXqdkt0uOb3kfYlNwmOdDyacOq/qlFo1v+PTmTi3E/glC9W11b34A22zmLzvb231Q0L2Bgg60OTW4YdstO+YOJnO38TtpH7zy9ymokWyA79qlVSn38HtpFlImFnhu3b4boNWXklOXV0Iwo7lQ1hrZyPFcwtjwFP7iEKSHSSJw509kh8kj6pr+H1jR7km9vcvqN9657vffefkv+fKxge1X+7RdjYUPIESN7gTvRkB/RMYtEkaVkdHApmdBPpnKmz0n1xSWFOyVIuLrinZwpoCRe6kyiVZoHX088F+UX4+WKS4iBTP0IWxGtZgOdMaV4KTayqHQF/VihBwTbgDXTCmKoOBJeNhwJMzEVjtjIFLuU38fPR7hqNG1JS7g/qRCuy3vmQ3W9Vu8qbVbP+SzazGRJH83MzP90Ck2m31mMjP8TiLn5uwD2Ugr2PFvPQjB5BnSJvQxGQZZEB+LopqzGzDbMmbkAPkZVJjeO5FzOSBKCgJze2ZS4Gemc9twrwY6u9H61iUQTcRvtdT9RW3tRxAWwFs2tcuJRnI6xjmBdWjbgFNRHMHiF1uHYBfUR/ut5Ug2jXAaT96+9RH/FToRwIzGbKmVJ1AZQnoabSB1yyIg7ByAridHApPMjyw0OiV6RjSbCuzwLAvFizBliWJua1tsuAgvNPbmljYbpt8lkWam7b3XZiOiKJskMOtmfScnsbPW208knwjuXrXK4Q1iKIgNyYXXDVT9C2Ye/78GQ5BEEXfFdde2RwauOysdJNL5AzCy84ard/nGAVN8alecnFdgu5Gbd5DJTL+hHZK0vApVy3OfU8XTSJg1TlssivsPYUlIqvn66PzrVTymCc4wgF6SDNR0pDf+9Gp+VnsUH5WtpHYsuhOaey8zdwLN47V8MTbm78g687+P3cx6tcAeNpjYGRgYGBk8s0/zBIfz2/zlUGeZQNQhOFCWfF0GP0/8P8c1jusIkAuBwMTSBQAYwQM6HjaY2BkYGAV+d8KJgP/XWG9wwAUQQGLAYqPBl942n1TvUoDQRCe1VM8kWARjNrZGIurBAsRBIuA2vkAFsJiKTYW4guIjT5ARMgTxCLoA1hcb5OgDyGHrY7f7M65e8fpLF++2W/nZ2eTmGfaIJi5I0qGDlZZcD51QzTTJirZPAI9JIwVA+wT8L5nOdMaV0AuMJ+icRHq8of6LSD18fzq8ds7xjpwBnQiSI9V5QVl6NwPvgM15NXn/AtWZyj3W0HjEXitOc/dIdbetPdFTZ+P6t+X7xU0/k6GJtOe1/B3arN0/pmz1J4UZc+D6ExwjD7vioeGd5HvhvU+R+DZcGZ6YBPNfAi0G97iBPwFXqph2cW8+D7kjMfwtinHb6kLb6Wygk3cZytSEoptGrlScdHtLPeri1JKueACMZfU1ViJG1Sq5E43dIt7SZZFl1zuRhb/GOs44xFVDbrJzB5tYs35OmaXTrEmkv0DajnMWQB42mNgYNCCwk0MLxheMPrhgUuY2JiUmOqY2pjWMD1hdmPOY+5hPsLCwWLEksSyiOUOawzrLrYiti/sCuxJ7Kc45DiSOPZxmnG2cG7jvMelweXDNYXrEbcBdxf3KR4OngheLd443g18fHwZfFv4NfiX8T8TEBIIEZggsEpQS7BMcJsQl5CFUI3QAWEp4RLhCyJaIldEbURXiJ4RYxEzE0sQ2yD2TzxIfJkEk4SeRJbENIkNEg8k/klqSGZITpE8InlL8p2UmVSG1A6pb9Jx0ltkjGSmyDySlZF1kc2RnSK7R/aZnJ5cmdwB+ST5SwpuCvsUjRTLFHcoOShNU9qhzKespGyhXKV8SPmBCpOKgUqcyjSVR6omqgmqe9RE1OrUnqkHqO9R/6FholGgsUZzgeYZLTUtL60WbS7tKh0OnQydXTpvdGV0O3S/6Gnopekt0ruhz6fvpl+nv0n/h4GdQYvBJUMhwwTDdYYvjFSM4oxmGd0zVjK2M84w3mYiYZJgssLkkqmO6TzTF2Z2ZjVmd8ylzP3MJ5lfsRCwcLJoszhhyWXpZdlhecZKxirHapbVPesF1ndsJGwCbBbZ/LA1sn1jZ2XXY3fFXsM+z36V/S8HD4cGh2OOTI51ThJOK5zeOUs4OzmXOS9wPuUi4JLgss7lm2uU6zY3NrcSty1u39zN3Mvct7l/8xDzMPLw88jyaPM44ynkaeEZ59niucqLyUvPKwgAn3OqOQAAAQAAARcApwARAAAAAAACAAAAAQABAAAAQAAuAAAAAHjarZK9TgJBEMf/d6CRaAyRhMLqCgsbL4ciglTGRPEjSiSKlnLycXJ86CEniU/hM9jYWPgIFkYfwd6nsDD+d1mBIIUx3mZnfzs3MzszuwDCeIYG8UUwQxmAFgxxPeeuyxrmcaNYxzTuFAewi0fFQSTxqXgM11pC8TgS2oPiCUS1d8Uh8ofiSczpYcVT5LjiCPlY8Qui+ncOr7D02y6/BTCrP/m+b5bdTrPi2I26Z9qNGtbRQBMdXMJBGRW0YOCecxEWYoiTCvxrYBunqHPdoX2bLOyrMKlZg8thDETw5K7Itci1TXlGy0124QRZZLDFU/exhxztMozlosTpMH6ZPge0L+OKGnFKjJ4WRwppHPL0PP3SI2P9jLQwFOu3GRhDfkeyDo//G7IHgzllZQxLdquvrdCyBVvat3seJlYo06gxapUxhU2JWnFygR03sSxnEkvcpf5Y5eibGq315TDp7fKWm8zbUVl71Aqq/ZtNnlkWmLnQtno9ycvXYbA6W2pF3aKfCayyC0Ja7Fr/PW70/HO4YM0OKxFvzf0C1MyPjwAAeNpt1VWUU2cYRuHsgxenQt1d8/3JOUnqAyR1d/cCLQVKO22pu7tQd3d3d3d3d3cXmGzumrWy3pWLs/NdPDMpZaWu1783l1Lpf14MnfzO6FbqVupfGkD30iR60JNe9KYP09CXfvRnAAMZxGCGMG3pW6ZjemZgKDMyEzMzC7MyG7MzB3MyF3MzD/MyH/OzAAuyEAuzCIuyGIuzBGWCRIUqOQU16jRYkqVYmmVYluVYng6GMZwRNGmxAiuyEiuzCquyGquzBmuyFmuzDuuyHuuzARuyERuzCZuyGZuzBVuyFVuzDduyHdszklGMZgd2ZAw7MZZxjGdnJrALu9LJbuzOHkxkT/Zib/ZhX/Zjfw7gQA7iYA7hUA7jcI7gSI7iaI7hWI7jeE7gRE7iZE5hEqdyGqdzBmdyFmdzDudyHudzARdyERdzCZdyGZdzBVdyFVdzDddyHddzAzdyEzdzC7dyG7dzB3dyF3dzD/dyH/fzAA/yEA/zCI/yGI/zBE/yFE/zDM/yHM/zAi/yEi/zCq/yGq/zBm/yFm/zDu/yHu/zAR/yER/zCZ/yGZ/zBV/yFV/zDd/yHd/zAz/yEz/zC7/yG7/zB3/yF3/zD/9mpYwsy7pl3bMeWc+sV9Y765NNk/XN+mX9swHZwGxQNjgb0nPkmInjR0V7Uq/OsaPL5Y7ylE3l8tQNN7kVt+rmbuHW3LrbcDvam1rtzVvdm50TxrU/DBvRtZUY1rV5a3jXFn550Wo/XDNWK3dFmh7X9LimxzU9qulRTY9qelTTo5rlKLt2wk7YiaprL+yFvbAX9pK9ZC/ZS/aSvWQv2Uv2kr1kr2KvYq9ir2KvYq9ir2KvYq9ir2Kvaq9qr2qvaq9qr2qvaq9qr2qvai+3l9vL7eX2cnu5vdxebi+3l9sr7BV2CjuFncJOYaewU9gp7NTs1LyrZq9mr2avZq9mr2avZq9mr26vbq9ur26vbq9ur26vbq9ur26vYa9hr2GvYa9hr2GvYa/R7oXuQ/eh+2j/UU7e3C3cqc/V3fYdof/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6D92H7kP3ofvQfeg+dB+6D92H7kP3ofvQfRT29B/6D/2H/kP/of/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6D/2H/kP/of/Qf+g/9B/6j6nuG3Ya7U5q/0hN3nCTW3Grbu4Wrs/rP+k/6T/pP+k/6T/pP+k+6T7pPek86TzpPOk86TzpOuk66TrpOuk66TrpOlWmPu/36zrpOuk66TrpOuk66TrpOvl/Pek76TvpO+k76TvpO+k76TvpO+k76TvpO7V9t+qtVs/OaOURU6bo6PgPt6rZbwAAAAABVFDDFwAA) format('woff'),url(data:application/font-sfnt;base64,AAEAAAAPAIAAAwBwRkZUTW0ql9wAAAD8AAAAHEdERUYBRAAEAAABGAAAACBPUy8yZ7lriQAAATgAAABgY21hcNqt44EAAAGYAAAGcmN2dCAAKAL4AAAIDAAAAARnYXNw//8AAwAACBAAAAAIZ2x5Zn1dwm8AAAgYAACUpGhlYWQFTS/YAACcvAAAADZoaGVhCkQEEQAAnPQAAAAkaG10eNLHIGAAAJ0YAAADdGxvY2Fv+5XOAACgjAAAAjBtYXhwAWoA2AAAorwAAAAgbmFtZbMsoJsAAKLcAAADonBvc3S6o+U1AACmgAAACtF3ZWJmwxhUUAAAsVQAAAAGAAAAAQAAAADMPaLPAAAAANB2gXUAAAAA0HZzlwABAAAADgAAABgAAAAAAAIAAQABARYAAQAEAAAAAgAAAAMEiwGQAAUABAMMAtAAAABaAwwC0AAAAaQAMgK4AAAAAAUAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAFVLV04AQAAg//8DwP8QAAAFFAB7AAAAAQAAAAAAAAAAAAAAIAABAAAABQAAAAMAAAAsAAAACgAAAdwAAQAAAAAEaAADAAEAAAAsAAMACgAAAdwABAGwAAAAaABAAAUAKAAgACsAoAClIAogLyBfIKwgvSISIxsl/CYBJvonCScP4APgCeAZ4CngOeBJ4FngYOBp4HngieCX4QnhGeEp4TnhRuFJ4VnhaeF54YnhleGZ4gbiCeIW4hniIeIn4jniSeJZ4mD4////AAAAIAAqAKAApSAAIC8gXyCsIL0iEiMbJfwmASb6JwknD+AB4AXgEOAg4DDgQOBQ4GDgYuBw4IDgkOEB4RDhIOEw4UDhSOFQ4WDhcOGA4ZDhl+IA4gniEOIY4iHiI+Iw4kDiUOJg+P/////j/9r/Zv9i4Ajf5N+132nfWd4F3P3aHdoZ2SHZE9kOIB0gHCAWIBAgCiAEH/4f+B/3H/Ef6x/lH3wfdh9wH2ofZB9jH10fVx9RH0sfRR9EHt4e3B7WHtUezh7NHsUevx65HrMIFQABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAAAAAACjAAAAAAAAAA1AAAAIAAAACAAAAADAAAAKgAAACsAAAAEAAAAoAAAAKAAAAAGAAAApQAAAKUAAAAHAAAgAAAAIAoAAAAIAAAgLwAAIC8AAAATAAAgXwAAIF8AAAAUAAAgrAAAIKwAAAAVAAAgvQAAIL0AAAAWAAAiEgAAIhIAAAAXAAAjGwAAIxsAAAAYAAAl/AAAJfwAAAAZAAAmAQAAJgEAAAAaAAAm+gAAJvoAAAAbAAAnCQAAJwkAAAAcAAAnDwAAJw8AAAAdAADgAQAA4AMAAAAeAADgBQAA4AkAAAAhAADgEAAA4BkAAAAmAADgIAAA4CkAAAAwAADgMAAA4DkAAAA6AADgQAAA4EkAAABEAADgUAAA4FkAAABOAADgYAAA4GAAAABYAADgYgAA4GkAAABZAADgcAAA4HkAAABhAADggAAA4IkAAABrAADgkAAA4JcAAAB1AADhAQAA4QkAAAB9AADhEAAA4RkAAACGAADhIAAA4SkAAACQAADhMAAA4TkAAACaAADhQAAA4UYAAACkAADhSAAA4UkAAACrAADhUAAA4VkAAACtAADhYAAA4WkAAAC3AADhcAAA4XkAAADBAADhgAAA4YkAAADLAADhkAAA4ZUAAADVAADhlwAA4ZkAAADbAADiAAAA4gYAAADeAADiCQAA4gkAAADlAADiEAAA4hYAAADmAADiGAAA4hkAAADtAADiIQAA4iEAAADvAADiIwAA4icAAADwAADiMAAA4jkAAAD1AADiQAAA4kkAAAD/AADiUAAA4lkAAAEJAADiYAAA4mAAAAETAAD4/wAA+P8AAAEUAAH1EQAB9REAAAEVAAH2qgAB9qoAAAEWAAYCCgAAAAABAAABAAAAAAAAAAAAAAAAAAAAAQACAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAEAAUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAVAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKAL4AAAAAf//AAIAAgAoAAABaAMgAAMABwAusQEALzyyBwQA7TKxBgXcPLIDAgDtMgCxAwAvPLIFBADtMrIHBgH8PLIBAgDtMjMRIRElMxEjKAFA/ujw8AMg/OAoAtAAAQBkAGQETARMAFsAAAEyFh8BHgEdATc+AR8BFgYPATMyFhcWFRQGDwEOASsBFx4BDwEGJi8BFRQGBwYjIiYvAS4BPQEHDgEvASY2PwEjIiYnJjU0Nj8BPgE7AScuAT8BNhYfATU0Njc2AlgPJgsLCg+eBxYIagcCB57gChECBgMCAQIRCuCeBwIHaggWB54PCikiDyYLCwoPngcWCGoHAgee4AoRAgYDAgECEQrgngcCB2oIFgeeDwopBEwDAgECEQrgngcCB2oIFgeeDwopIg8mCwsKD54HFghqBwIHnuAKEQIGAwIBAhEK4J4HAgdqCBYHng8KKSIPJgsLCg+eBxYIagcCB57gChECBgAAAAABAAAAAARMBEwAIwAAATMyFhURITIWHQEUBiMhERQGKwEiJjURISImPQE0NjMhETQ2AcLIFR0BXhUdHRX+oh0VyBUd/qIVHR0VAV4dBEwdFf6iHRXIFR3+ohUdHRUBXh0VyBUdAV4VHQAAAAABAHAAAARABEwARQAAATMyFgcBBgchMhYPAQ4BKwEVITIWDwEOASsBFRQGKwEiJj0BISImPwE+ATsBNSEiJj8BPgE7ASYnASY2OwEyHwEWMj8BNgM5+goFCP6UBgUBDAoGBngGGAp9ARMKBgZ4BhgKfQ8LlAsP/u0KBgZ4BhgKff7tCgYGeAYYCnYFBv6UCAUK+hkSpAgUCKQSBEwKCP6UBgwMCKAIDGQMCKAIDK4LDw8LrgwIoAgMZAwIoAgMDAYBbAgKEqQICKQSAAABAGQABQSMBK4AOwAAATIXFhcjNC4DIyIOAwchByEGFSEHIR4EMzI+AzUzBgcGIyInLgEnIzczNjcjNzM+ATc2AujycDwGtSM0QDkXEys4MjAPAXtk/tQGAZZk/tQJMDlCNBUWOUA0I64eYmunznYkQgzZZHABBdpkhhQ+H3UErr1oaS1LMCEPCx4uTzJkMjJkSnRCKw8PIjBKK6trdZ4wqndkLzVkV4UljQAAAgB7AAAETASwAD4ARwAAASEyHgUVHAEVFA4FKwEHITIWDwEOASsBFRQGKwEiJj0BISImPwE+ATsBNSEiJj8BPgE7ARE0NhcRMzI2NTQmIwGsAV5DakIwFgwBAQwWMEJqQ7ICASAKBgZ4BhgKigsKlQoP/vUKBgZ4BhgKdf71CgYGeAYYCnUPtstALS1ABLAaJD8yTyokCwsLJCpQMkAlGmQMCKAIDK8LDg8KrwwIoAgMZAwIoAgMAdsKD8j+1EJWVEAAAAEAyAGQBEwCvAAPAAATITIWHQEUBiMhIiY9ATQ2+gMgFR0dFfzgFR0dArwdFcgVHR0VyBUdAAAAAgDIAAAD6ASwACUAQQAAARUUBisBFRQGBx4BHQEzMhYdASE1NDY7ATU0NjcuAT0BIyImPQEXFRQWFx4BFAYHDgEdASE1NCYnLgE0Njc+AT0BA+gdFTJjUVFjMhUd/OAdFTJjUVFjMhUdyEE3HCAgHDdBAZBBNxwgIBw3QQSwlhUdZFuVIyOVW5YdFZaWFR2WW5UjI5VbZB0VlshkPGMYDDI8MgwYYzyWljxjGAwyPDIMGGM8ZAAAAAEAAAAAAAAAAAAAAAAxAAAB//IBLATCBEEAFgAAATIWFzYzMhYVFAYjISImNTQ2NyY1NDYB9261LCwueKqqeP0ST3FVQgLYBEF3YQ6teHmtclBFaw4MGZnXAAAAAgAAAGQEsASvABoAHgAAAB4BDwEBMzIWHQEhNTQ2OwEBJyY+ARYfATc2AyEnAwL2IAkKiAHTHhQe+1AeFB4B1IcKCSAkCm9wCXoBebbDBLMTIxC7/RYlFSoqFSUC6rcQJBQJEJSWEPwecAIWAAAAAAQAAABkBLAETAALABcAIwA3AAATITIWBwEGIicBJjYXARYUBwEGJjURNDYJATYWFREUBicBJjQHARYGIyEiJjcBNjIfARYyPwE2MhkEfgoFCP3MCBQI/cwIBQMBCAgI/vgICgoDjAEICAoKCP74CFwBbAgFCvuCCgUIAWwIFAikCBQIpAgUBEwKCP3JCAgCNwgK2v74CBQI/vgIBQoCJgoF/vABCAgFCv3aCgUIAQgIFID+lAgKCggBbAgIpAgIpAgAAAAD//D/8AS6BLoACQANABAAAAAyHwEWFA8BJzcTAScJAQUTA+AmDpkNDWPWXyL9mdYCZv4f/rNuBLoNmQ4mDlzWYP50/ZrWAmb8anABTwAAAAEAAAAABLAEsAAPAAABETMyFh0BITU0NjsBEQEhArz6FR384B0V+v4MBLACiv3aHRUyMhUdAiYCJgAAAAEADgAIBEwEnAAfAAABJTYWFREUBgcGLgE2NzYXEQURFAYHBi4BNjc2FxE0NgFwAoUnMFNGT4gkV09IQv2oWEFPiCRXT0hCHQP5ow8eIvzBN1EXGSltchkYEAIJm/2iKmAVGilucRoYEQJ/JioAAAACAAn/+AS7BKcAHQApAAAAMh4CFQcXFAcBFgYPAQYiJwEGIycHIi4CND4BBCIOARQeATI+ATQmAZDItoNOAQFOARMXARY7GikT/u13jgUCZLaDTk6DAXKwlFZWlLCUVlYEp06DtmQCBY15/u4aJRg6FBQBEk0BAU6Dtsi2g1tWlLCUVlaUsJQAAQBkAFgErwREABkAAAE+Ah4CFRQOAwcuBDU0PgIeAQKJMHt4dVg2Q3mEqD4+p4V4Qzhadnh5A7VESAUtU3ZAOXmAf7JVVbJ/gHk5QHZTLQVIAAAAAf/TAF4EewSUABgAAAETNjIXEyEyFgcFExYGJyUFBiY3EyUmNjMBl4MHFQeBAaUVBhH+qoIHDxH+qf6qEQ8Hgv6lEQYUAyABYRMT/p8RDPn+bxQLDPb3DAsUAZD7DBEAAv/TAF4EewSUABgAIgAAARM2MhcTITIWBwUTFgYnJQUGJjcTJSY2MwUjFwc3Fyc3IycBl4MHFQeBAaUVBhH+qoIHDxH+qf6qEQ8Hgv6lEQYUAfPwxUrBw0rA6k4DIAFhExP+nxEM+f5vFAsM9vcMCxQBkPsMEWSO4ouM5YzTAAABAAAAAASwBLAAJgAAATIWHQEUBiMVFBYXBR4BHQEUBiMhIiY9ATQ2NyU+AT0BIiY9ATQ2Alh8sD4mDAkBZgkMDwr7ggoPDAkBZgkMJj6wBLCwfPouaEsKFwbmBRcKXQoPDwpdChcF5gYXCktoLvp8sAAAAA0AAAAABLAETAAPABMAIwAnACsALwAzADcARwBLAE8AUwBXAAATITIWFREUBiMhIiY1ETQ2FxUzNSkBIgYVERQWMyEyNjURNCYzFTM1BRUzNSEVMzUFFTM1IRUzNQchIgYVERQWMyEyNjURNCYFFTM1IRUzNQUVMzUhFTM1GQR+Cg8PCvuCCg8PVWQCo/3aCg8PCgImCg8Pc2T8GGQDIGT8GGQDIGTh/doKDw8KAiYKDw/872QDIGT8GGQDIGQETA8K++YKDw8KBBoKD2RkZA8K/qIKDw8KAV4KD2RkyGRkZGTIZGRkZGQPCv6iCg8PCgFeCg9kZGRkZMhkZGRkAAAEAAAAAARMBEwADwAfAC8APwAAEyEyFhURFAYjISImNRE0NikBMhYVERQGIyEiJjURNDYBITIWFREUBiMhIiY1ETQ2KQEyFhURFAYjISImNRE0NjIBkBUdHRX+cBUdHQJtAZAVHR0V/nAVHR39vQGQFR0dFf5wFR0dAm0BkBUdHRX+cBUdHQRMHRX+cBUdHRUBkBUdHRX+cBUdHRUBkBUd/agdFf5wFR0dFQGQFR0dFf5wFR0dFQGQFR0AAAkAAAAABEwETAAPAB8ALwA/AE8AXwBvAH8AjwAAEzMyFh0BFAYrASImPQE0NiEzMhYdARQGKwEiJj0BNDYhMzIWHQEUBisBIiY9ATQ2ATMyFh0BFAYrASImPQE0NiEzMhYdARQGKwEiJj0BNDYhMzIWHQEUBisBIiY9ATQ2ATMyFh0BFAYrASImPQE0NiEzMhYdARQGKwEiJj0BNDYhMzIWHQEUBisBIiY9ATQ2MsgVHR0VyBUdHQGlyBUdHRXIFR0dAaXIFR0dFcgVHR389cgVHR0VyBUdHQGlyBUdHRXIFR0dAaXIFR0dFcgVHR389cgVHR0VyBUdHQGlyBUdHRXIFR0dAaXIFR0dFcgVHR0ETB0VyBUdHRXIFR0dFcgVHR0VyBUdHRXIFR0dFcgVHf5wHRXIFR0dFcgVHR0VyBUdHRXIFR0dFcgVHR0VyBUd/nAdFcgVHR0VyBUdHRXIFR0dFcgVHR0VyBUdHRXIFR0ABgAAAAAEsARMAA8AHwAvAD8ATwBfAAATMzIWHQEUBisBIiY9ATQ2KQEyFh0BFAYjISImPQE0NgEzMhYdARQGKwEiJj0BNDYpATIWHQEUBiMhIiY9ATQ2ATMyFh0BFAYrASImPQE0NikBMhYdARQGIyEiJj0BNDYyyBUdHRXIFR0dAaUCvBUdHRX9RBUdHf6FyBUdHRXIFR0dAaUCvBUdHRX9RBUdHf6FyBUdHRXIFR0dAaUCvBUdHRX9RBUdHQRMHRXIFR0dFcgVHR0VyBUdHRXIFR3+cB0VyBUdHRXIFR0dFcgVHR0VyBUd/nAdFcgVHR0VyBUdHRXIFR0dFcgVHQAAAAABACYALAToBCAAFwAACQE2Mh8BFhQHAQYiJwEmND8BNjIfARYyAdECOwgUB7EICPzxBxUH/oAICLEHFAirBxYB3QI7CAixBxQI/PAICAGACBQHsQgIqwcAAQBuAG4EQgRCACMAAAEXFhQHCQEWFA8BBiInCQEGIi8BJjQ3CQEmND8BNjIXCQE2MgOIsggI/vUBCwgIsggVB/70/vQHFQiyCAgBC/71CAiyCBUHAQwBDAcVBDuzCBUH/vT+9AcVCLIICAEL/vUICLIIFQcBDAEMBxUIsggI/vUBDAcAAwAX/+sExQSZABkAJQBJAAAAMh4CFRQHARYUDwEGIicBBiMiLgI0PgEEIg4BFB4BMj4BNCYFMzIWHQEzMhYdARQGKwEVFAYrASImPQEjIiY9ATQ2OwE1NDYBmcSzgk1OASwICG0HFQj+1HeOYrSBTU2BAW+zmFhYmLOZWFj+vJYKD0sKDw8KSw8KlgoPSwoPDwpLDwSZTYKzYo15/tUIFQhsCAgBK01NgbTEs4JNWJmzmFhYmLOZIw8KSw8KlgoPSwoPDwpLDwqWCg9LCg8AAAMAF//rBMUEmQAZACUANQAAADIeAhUUBwEWFA8BBiInAQYjIi4CND4BBCIOARQeATI+ATQmBSEyFh0BFAYjISImPQE0NgGZxLOCTU4BLAgIbQcVCP7Ud45itIFNTYEBb7OYWFiYs5lYWP5YAV4KDw8K/qIKDw8EmU2Cs2KNef7VCBUIbAgIAStNTYG0xLOCTViZs5hYWJizmYcPCpYKDw8KlgoPAAAAAAIAFwAXBJkEsAAPAC0AAAEzMhYVERQGKwEiJjURNDYFNRYSFRQOAiIuAjU0EjcVDgEVFB4BMj4BNTQmAiZkFR0dFWQVHR0BD6fSW5vW6tabW9KnZ3xyxejFcnwEsB0V/nAVHR0VAZAVHeGmPv7ZuHXWm1tbm9Z1uAEnPqY3yHh0xXJyxXR4yAAEAGQAAASwBLAADwAfAC8APwAAATMyFhURFAYrASImNRE0NgEzMhYVERQGKwEiJjURNDYBMzIWFREUBisBIiY1ETQ2BTMyFh0BFAYrASImPQE0NgQBlgoPDwqWCg8P/t6WCg8PCpYKDw/+3pYKDw8KlgoPD/7elgoPDwqWCg8PBLAPCvuCCg8PCgR+Cg/+cA8K/RIKDw8KAu4KD/7UDwr+PgoPDwoBwgoPyA8K+goPDwr6Cg8AAAAAAgAaABsElgSWAEcATwAAATIfAhYfATcWFwcXFh8CFhUUDwIGDwEXBgcnBwYPAgYjIi8CJi8BByYnNycmLwImNTQ/AjY/ASc2Nxc3Nj8CNhIiBhQWMjY0AlghKSYFMS0Fhj0rUAMZDgGYBQWYAQ8YA1AwOIYFLDIFJisfISkmBTEtBYY8LFADGQ0ClwYGlwINGQNQLzqFBS0xBSYreLJ+frJ+BJYFmAEOGQJQMDmGBSwxBiYrHiIoJgYxLAWGPSxRAxkOApcFBZcCDhkDUTA5hgUtMAYmKiAhKCYGMC0Fhj0sUAIZDgGYBf6ZfrF+frEABwBkAAAEsAUUABMAFwAhACUAKQAtADEAAAEhMhYdASEyFh0BITU0NjMhNTQ2FxUhNQERFAYjISImNREXETMRMxEzETMRMxEzETMRAfQBLCk7ARMKD/u0DwoBEzspASwBLDsp/UQpO2RkZGRkZGRkBRQ7KWQPCktLCg9kKTtkZGT+1PzgKTs7KQMgZP1EArz9RAK8/UQCvP1EArwAAQAMAAAFCATRAB8AABMBNjIXARYGKwERFAYrASImNREhERQGKwEiJjURIyImEgJsCBUHAmAIBQqvDwr6Cg/+1A8K+goPrwoFAmoCYAcH/aAICv3BCg8PCgF3/okKDw8KAj8KAAIAZAAAA+gEsAARABcAAAERFBYzIREUBiMhIiY1ETQ2MwEjIiY9AQJYOykBLB0V/OAVHR0VA1L6FR0EsP5wKTv9dhUdHRUETBUd/nAdFfoAAwAXABcEmQSZAA8AGwAwAAAAMh4CFA4CIi4CND4BBCIOARQeATI+ATQmBTMyFhURMzIWHQEUBisBIiY1ETQ2AePq1ptbW5vW6tabW1ubAb/oxXJyxejFcnL+fDIKD68KDw8K+goPDwSZW5vW6tabW1ub1urWmztyxejFcnLF6MUNDwr+7Q8KMgoPDwoBXgoPAAAAAAL/nAAABRQEsAALAA8AACkBAyMDIQEzAzMDMwEDMwMFFP3mKfIp/eYBr9EVohTQ/p4b4BsBkP5wBLD+1AEs/nD+1AEsAAAAAAIAZAAABLAEsAAVAC8AAAEzMhYVETMyFgcBBiInASY2OwERNDYBMzIWFREUBiMhIiY1ETQ2OwEyFh0BITU0NgImyBUdvxQLDf65DSYN/rkNCxS/HQJUMgoPDwr75goPDwoyCg8DhA8EsB0V/j4XEP5wEBABkBAXAcIVHfzgDwr+ogoPDwoBXgoPDwqvrwoPAAMAFwAXBJkEmQAPABsAMQAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JgUzMhYVETMyFgcDBiInAyY2OwERNDYB4+rWm1tbm9bq1ptbW5sBv+jFcnLF6MVycv58lgoPiRUKDd8NJg3fDQoViQ8EmVub1urWm1tbm9bq1ps7csXoxXJyxejFDQ8K/u0XEP7tEBABExAXARMKDwAAAAMAFwAXBJkEmQAPABsAMQAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JiUTFgYrAREUBisBIiY1ESMiJjcTNjIB4+rWm1tbm9bq1ptbW5sBv+jFcnLF6MVycv7n3w0KFYkPCpYKD4kVCg3fDSYEmVub1urWm1tbm9bq1ps7csXoxXJyxejFAf7tEBf+7QoPDwoBExcQARMQAAAAAAIAAAAABLAEsAAZADkAABMhMhYXExYVERQGBwYjISImJyY1EzQ3Ez4BBSEiBgcDBhY7ATIWHwEeATsBMjY/AT4BOwEyNicDLgHhAu4KEwO6BwgFDBn7tAweAgYBB7kDEwKX/dQKEgJXAgwKlgoTAiYCEwr6ChMCJgITCpYKDAJXAhIEsA4K/XQYGf5XDB4CBggEDRkBqRkYAowKDsgOC/4+Cw4OCpgKDg4KmAoODgsBwgsOAAMAFwAXBJkEmQAPABsAJwAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JgUXFhQPAQYmNRE0NgHj6tabW1ub1urWm1tbmwG/6MVycsXoxXJy/ov9ERH9EBgYBJlbm9bq1ptbW5vW6tabO3LF6MVycsXoxV2+DCQMvgwLFQGQFQsAAQAXABcEmQSwACgAAAE3NhYVERQGIyEiJj8BJiMiDgEUHgEyPgE1MxQOAiIuAjQ+AjMyA7OHBwsPCv6WCwQHhW2BdMVycsXoxXKWW5vW6tabW1ub1nXABCSHBwQL/pYKDwsHhUxyxejFcnLFdHXWm1tbm9bq1ptbAAAAAAIAFwABBJkEsAAaADUAAAE3NhYVERQGIyEiJj8BJiMiDgEVIzQ+AjMyEzMUDgIjIicHBiY1ETQ2MyEyFg8BFjMyPgEDs4cHCw8L/pcLBAeGboF0xXKWW5vWdcDrllub1nXAnIYHCw8LAWgKBQiFboJ0xXIEJIcHBAv+lwsPCweGS3LFdHXWm1v9v3XWm1t2hggFCgFoCw8LB4VMcsUAAAAKAGQAAASwBLAADwAfAC8APwBPAF8AbwB/AI8AnwAAEyEyFhURFAYjISImNRE0NgUhIgYVERQWMyEyNjURNCYFMzIWHQEUBisBIiY9ATQ2MyEyFh0BFAYjISImPQE0NgczMhYdARQGKwEiJj0BNDYzITIWHQEUBiMhIiY9ATQ2BzMyFh0BFAYrASImPQE0NjMhMhYdARQGIyEiJj0BNDYHMzIWHQEUBisBIiY9ATQ2MyEyFh0BFAYjISImPQE0Nn0EGgoPDwr75goPDwPA/K4KDw8KA1IKDw/9CDIKDw8KMgoPD9IBwgoPDwr+PgoPD74yCg8PCjIKDw/SAcIKDw8K/j4KDw++MgoPDwoyCg8P0gHCCg8PCv4+Cg8PvjIKDw8KMgoPD9IBwgoPDwr+PgoPDwSwDwr7ggoPDwoEfgoPyA8K/K4KDw8KA1IKD2QPCjIKDw8KMgoPDwoyCg8PCjIKD8gPCjIKDw8KMgoPDwoyCg8PCjIKD8gPCjIKDw8KMgoPDwoyCg8PCjIKD8gPCjIKDw8KMgoPDwoyCg8PCjIKDwAAAAACAAAAAARMBLAAGQAjAAABNTQmIyEiBh0BIyIGFREUFjMhMjY1ETQmIyE1NDY7ATIWHQEDhHVT/tRSdmQpOzspA4QpOzsp/ageFMgUHgMgyFN1dlLIOyn9qCk7OykCWCk7lhUdHRWWAAIAZAAABEwETAAJADcAABMzMhYVESMRNDYFMhcWFREUBw4DIyIuAScuAiMiBwYjIicmNRE+ATc2HgMXHgIzMjc2fTIKD2QPA8AEBRADIUNAMRwaPyonKSxHHlVLBwgGBQ4WeDsXKC4TOQQpLUUdZ1AHBEwPCvvNBDMKDzACBhH+WwYGO1AkDQ0ODg8PDzkFAwcPAbY3VwMCAwsGFAEODg5XCAAAAwAAAAAEsASXACEAMQBBAAAAMh4CFREUBisBIiY1ETQuASAOARURFAYrASImNRE0PgEDMzIWFREUBisBIiY1ETQ2ITMyFhURFAYrASImNRE0NgHk6N6jYw8KMgoPjeT++uSNDwoyCg9joyqgCAwMCKAIDAwCYKAIDAwIoAgMDASXY6PedP7UCg8PCgEsf9FyctF//tQKDw8KASx03qP9wAwI/jQIDAwIAcwIDAwI/jQIDAwIAcwIDAAAAAACAAAA0wRHA90AFQA5AAABJTYWFREUBiclJisBIiY1ETQ2OwEyBTc2Mh8BFhQPARcWFA8BBiIvAQcGIi8BJjQ/AScmND8BNjIXAUEBAgkMDAn+/hUZ+goPDwr6GQJYeAcUByIHB3h4BwciBxQHeHgHFAciBwd3dwcHIgcUBwMurAYHCv0SCgcGrA4PCgFeCg+EeAcHIgcUB3h4BxQHIgcHd3cHByIHFAd4eAcUByIICAAAAAACAAAA0wNyA90AFQAvAAABJTYWFREUBiclJisBIiY1ETQ2OwEyJTMWFxYVFAcGDwEiLwEuATc2NTQnJjY/ATYBQQECCQwMCf7+FRn6Cg8PCvoZAdIECgZgWgYLAwkHHQcDBkhOBgMIHQcDLqwGBwr9EgoHBqwODwoBXgoPZAEJgaGafwkBAQYXBxMIZ36EaggUBxYFAAAAAAMAAADEBGID7AAbADEASwAAATMWFxYVFAYHBgcjIi8BLgE3NjU0JicmNj8BNgUlNhYVERQGJyUmKwEiJjURNDY7ATIlMxYXFhUUBwYPASIvAS4BNzY1NCcmNj8BNgPHAwsGh0RABwoDCQcqCAIGbzs3BgIJKgf9ggECCQwMCf7+FRn6Cg8PCvoZAdIECgZgWgYLAwkHHQcDBkhOBgMIHQcD7AEJs9lpy1QJAQYiBhQIlrJarEcJFAYhBb6sBgcK/RIKBwasDg8KAV4KD2QBCYGhmn8JAQEGFwcTCGd+hGoIFQYWBQAAAAANAAAAAASwBLAACQAVABkAHQAhACUALQA7AD8AQwBHAEsATwAAATMVIxUhFSMRIQEjFTMVIREjESM1IQURIREhESERBSM1MwUjNTMBMxEhETM1MwEzFSMVIzUjNTM1IzUhBREhEQcjNTMFIzUzASM1MwUhNSEB9GRk/nBkAfQCvMjI/tTIZAJY+7QBLAGQASz84GRkArxkZP1EyP4MyGQB9MhkyGRkyAEs/UQBLGRkZAOEZGT+DGRkAfT+1AEsA4RkZGQCWP4MZMgBLAEsyGT+1AEs/tQBLMhkZGT+DP4MAfRk/tRkZGRkyGTI/tQBLMhkZGT+1GRkZAAAAAAJAAAAAASwBLAAAwAHAAsADwATABcAGwAfACMAADcjETMTIxEzASMRMxMjETMBIxEzASE1IRcjNTMXIzUzBSM1M2RkZMhkZAGQyMjIZGQBLMjI/OD+1AEsyGRkyGRkASzIyMgD6PwYA+j8GAPo/BgD6PwYA+j7UGRkW1tbW1sAAAIAAAAKBKYEsAANABUAAAkBFhQHAQYiJwETNDYzBCYiBhQWMjYB9AKqCAj+MAgUCP1WAQ8KAUM7Uzs7UzsEsP1WCBQI/jAICAKqAdsKD807O1Q7OwAAAAADAAAACgXSBLAADQAZACEAAAkBFhQHAQYiJwETNDYzIQEWFAcBBiIvAQkBBCYiBhQWMjYB9AKqCAj+MAgUCP1WAQ8KAwYCqggI/jAIFAg4Aaj9RP7TO1M7O1M7BLD9VggUCP4wCAgCqgHbCg/9VggUCP4wCAg4AaoCvM07O1Q7OwAAAAABAGQAAASwBLAAJgAAASEyFREUDwEGJjURNCYjISIPAQYWMyEyFhURFAYjISImNRE0PwE2ASwDOUsSQAgKDwr9RBkSQAgFCgK8Cg8PCvyuCg8SixIEsEv8fBkSQAgFCgO2Cg8SQAgKDwr8SgoPDwoDzxkSixIAAAABAMj//wRMBLAACgAAEyEyFhURCQERNDb6AyAVHf4+/j4dBLAdFfuCAbz+QwR/FR0AAAAAAwAAAAAEsASwABUARQBVAAABISIGBwMGHwEeATMhMjY/ATYnAy4BASMiBg8BDgEjISImLwEuASsBIgYVERQWOwEyNj0BNDYzITIWHQEUFjsBMjY1ETQmASEiBg8BBhYzITI2LwEuAQM2/kQLEAFOBw45BhcKAcIKFwY+DgdTARABVpYKFgROBBYK/doKFgROBBYKlgoPDwqWCg8PCgLuCg8PCpYKDw/+sf4MChMCJgILCgJYCgsCJgITBLAPCv7TGBVsCQwMCWwVGAEtCg/+cA0JnAkNDQmcCQ0PCv12Cg8PCpYKDw8KlgoPDwoCigoP/agOCpgKDg4KmAoOAAAAAAQAAABkBLAETAAdACEAKQAxAAABMzIeAh8BMzIWFREUBiMhIiY1ETQ2OwE+BAEVMzUEIgYUFjI2NCQyFhQGIiY0AfTIOF00JAcGlik7Oyn8GCk7OymWAgknM10ByGT+z76Hh76H/u9WPDxWPARMKTs7FRQ7Kf2oKTs7KQJYKTsIG0U1K/7UZGRGh76Hh74IPFY8PFYAAAAAAgA1AAAEsASvACAAIwAACQEWFx4BHwEVITUyNi8BIQYHBh4CMxUhNTY3PgE/AQEDIQMCqQGBFCgSJQkK/l81LBFS/nk6IgsJKjIe/pM4HAwaBwcBj6wBVKIEr/waMioTFQECQkJXLd6RWSIuHAxCQhgcDCUNDQPu/VoByQAAAAADAGQAAAPwBLAAJwAyADsAAAEeBhUUDgMjITU+ATURNC4EJzUFMh4CFRQOAgclMzI2NTQuAisBETMyNjU0JisBAvEFEzUwOyodN1htbDD+DCk7AQYLFyEaAdc5dWM+Hy0tEP6Pi05pESpTPnbYUFJ9Xp8CgQEHGB0zOlIuQ3VONxpZBzMoAzsYFBwLEAkHRwEpSXNDM1s6KwkxYUopOzQb/K5lUFqBAAABAMgAAANvBLAAGQAAARcOAQcDBhYXFSE1NjcTNjQuBCcmJzUDbQJTQgeECSxK/gy6Dq0DAw8MHxUXDQYEsDkTNSj8uTEoBmFhEFIDQBEaExAJCwYHAwI5AAAAAAL/tQAABRQEsAAlAC8AAAEjNC4FKwERFBYfARUhNTI+AzURIyIOBRUjESEFIxEzByczESM3BRQyCAsZEyYYGcgyGRn+cAQOIhoWyBkYJhMZCwgyA+j7m0tLfX1LS30DhBUgFQ4IAwH8rhYZAQJkZAEFCRUOA1IBAwgOFSAVASzI/OCnpwMgpwACACH/tQSPBLAAJQAvAAABIzQuBSsBERQWHwEVITUyPgM1ESMiDgUVIxEhEwc1IRUnNxUhNQRMMggLGRMmGBnIMhkZ/nAEDiIaFsgZGCYTGQsIMgPoQ6f84KenAyADhBUgFQ4IAwH9dhYZAQJkZAEFCRUOAooBAwgOFSAVASz7gn1LS319S0sABAAAAAAEsARMAA8AHwAvAD8AABMhMhYdARQGIyEiJj0BNDYTITIWHQEUBiMhIiY9ATQ2EyEyFh0BFAYjISImPQE0NhMhMhYdARQGIyEiJj0BNDYyAlgVHR0V/agVHR0VA+gVHR0V/BgVHR0VAyAVHR0V/OAVHR0VBEwVHR0V+7QVHR0ETB0VZBUdHRVkFR3+1B0VZBUdHRVkFR3+1B0VZBUdHRVkFR3+1B0VZBUdHRVkFR0ABAAAAAAEsARMAA8AHwAvAD8AABMhMhYdARQGIyEiJj0BNDYDITIWHQEUBiMhIiY9ATQ2EyEyFh0BFAYjISImPQE0NgMhMhYdARQGIyEiJj0BNDb6ArwVHR0V/UQVHR2zBEwVHR0V+7QVHR3dArwVHR0V/UQVHR2zBEwVHR0V+7QVHR0ETB0VZBUdHRVkFR3+1B0VZBUdHRVkFR3+1B0VZBUdHRVkFR3+1B0VZBUdHRVkFR0ABAAAAAAEsARMAA8AHwAvAD8AAAE1NDYzITIWHQEUBiMhIiYBNTQ2MyEyFh0BFAYjISImEzU0NjMhMhYdARQGIyEiJgE1NDYzITIWHQEUBiMhIiYB9B0VAlgVHR0V/agVHf5wHRUD6BUdHRX8GBUdyB0VAyAVHR0V/OAVHf7UHRUETBUdHRX7tBUdA7ZkFR0dFWQVHR3+6WQVHR0VZBUdHf7pZBUdHRVkFR0d/ulkFR0dFWQVHR0AAAQAAAAABLAETAAPAB8ALwA/AAATITIWHQEUBiMhIiY9ATQ2EyEyFh0BFAYjISImPQE0NhMhMhYdARQGIyEiJj0BNDYTITIWHQEUBiMhIiY9ATQ2MgRMFR0dFfu0FR0dFQRMFR0dFfu0FR0dFQRMFR0dFfu0FR0dFQRMFR0dFfu0FR0dBEwdFWQVHR0VZBUd/tQdFWQVHR0VZBUd/tQdFWQVHR0VZBUd/tQdFWQVHR0VZBUdAAgAAAAABLAETAAPAB8ALwA/AE8AXwBvAH8AABMzMhYdARQGKwEiJj0BNDYpATIWHQEUBiMhIiY9ATQ2ATMyFh0BFAYrASImPQE0NikBMhYdARQGIyEiJj0BNDYBMzIWHQEUBisBIiY9ATQ2KQEyFh0BFAYjISImPQE0NgEzMhYdARQGKwEiJj0BNDYpATIWHQEUBiMhIiY9ATQ2MmQVHR0VZBUdHQFBAyAVHR0V/OAVHR3+6WQVHR0VZBUdHQFBAyAVHR0V/OAVHR3+6WQVHR0VZBUdHQFBAyAVHR0V/OAVHR3+6WQVHR0VZBUdHQFBAyAVHR0V/OAVHR0ETB0VZBUdHRVkFR0dFWQVHR0VZBUd/tQdFWQVHR0VZBUdHRVkFR0dFWQVHf7UHRVkFR0dFWQVHR0VZBUdHRVkFR3+1B0VZBUdHRVkFR0dFWQVHR0VZBUdAAAG/5wAAASwBEwAAwATACMAKgA6AEoAACEjETsCMhYdARQGKwEiJj0BNDYTITIWHQEUBiMhIiY9ATQ2BQc1IzUzNQUhMhYdARQGIyEiJj0BNDYTITIWHQEUBiMhIiY9ATQ2AZBkZJZkFR0dFWQVHR0VAfQVHR0V/gwVHR3++qfIyAHCASwVHR0V/tQVHR0VAlgVHR0V/agVHR0ETB0VZBUdHRVkFR3+1B0VZBUdHRVkFR36fUtkS68dFWQVHR0VZBUd/tQdFWQVHR0VZBUdAAAABgAAAAAFFARMAA8AEwAjACoAOgBKAAATMzIWHQEUBisBIiY9ATQ2ASMRMwEhMhYdARQGIyEiJj0BNDYFMxUjFSc3BSEyFh0BFAYjISImPQE0NhMhMhYdARQGIyEiJj0BNDYyZBUdHRVkFR0dA2dkZPyuAfQVHR0V/gwVHR0EL8jIp6f75gEsFR0dFf7UFR0dFQJYFR0dFf2oFR0dBEwdFWQVHR0VZBUd+7QETP7UHRVkFR0dFWQVHchkS319rx0VZBUdHRVkFR3+1B0VZBUdHRVkFR0AAAAAAgAAAMgEsAPoAA8AEgAAEyEyFhURFAYjISImNRE0NgkCSwLuHywsH/0SHywsBIT+1AEsA+gsH/12HywsHwKKHyz9RAEsASwAAwAAAAAEsARMAA8AFwAfAAATITIWFREUBiMhIiY1ETQ2FxE3BScBExEEMhYUBiImNCwEWBIaGhL7qBIaGkr3ASpKASXs/NJwTk5wTgRMGhL8DBIaGhID9BIaZP0ftoOcAT7+4AH0dE5vT09vAAAAAAIA2wAFBDYEkQAWAB4AAAEyHgEVFAcOAQ8BLgQnJjU0PgIWIgYUFjI2NAKIdcZzRkWyNjYJIV5YbSk8RHOft7eCgreCBJF4ynVzj23pPz4IIWZomEiEdVijeUjDgriBgbgAAAACABcAFwSZBJkADwAXAAAAMh4CFA4CIi4CND4BAREiDgEUHgEB4+rWm1tbm9bq1ptbW5sBS3TFcnLFBJlbm9bq1ptbW5vW6tab/G8DVnLF6MVyAAACAHUAAwPfBQ8AGgA1AAABHgYVFA4DBy4DNTQ+BQMOAhceBBcWNj8BNiYnLgInJjc2IyYCKhVJT1dOPiUzVnB9P1SbfEokP0xXUEm8FykoAwEbITEcExUWAgYCCQkFEikMGiACCAgFD0iPdXdzdYdFR4BeRiYEBTpjl1lFh3ZzeHaQ/f4hS4I6JUEnIw4IBwwQIgoYBwQQQSlZtgsBAAAAAwAAAAAEywRsAAwAKgAvAAABNz4CHgEXHgEPAiUhMhcHISIGFREUFjMhMjY9ATcRFAYjISImNRE0NgkBBzcBA+hsAgYUFR0OFgoFBmz9BQGQMje7/pApOzspAfQpO8i7o/5wpbm5Azj+lqE3AWMD9XMBAgIEDw4WKgsKc8gNuzsp/gwpOzsptsj+tKW5uaUBkKW5/tf+ljKqAWMAAgAAAAAEkwRMABsANgAAASEGByMiBhURFBYzITI2NTcVFAYjISImNRE0NgUBFhQHAQYmJzUmDgMHPgY3NT4BAV4BaaQ0wyk7OykB9Ck7yLml/nClubkCfwFTCAj+rAcLARo5ZFRYGgouOUlARioTAQsETJI2Oyn+DCk7OymZZ6W5uaUBkKW5G/7TBxUH/s4GBAnLAQINFjAhO2JBNB0UBwHSCgUAAAAAAgAAAAAEnQRMAB0ANQAAASEyFwchIgYVERQWMyEyNj0BNxUUBiMhIiY1ETQ2CQE2Mh8BFhQHAQYiLwEmND8BNjIfARYyAV4BXjxDsv6jKTs7KQH0KTvIuaX+cKW5uQHKAYsHFQdlBwf97QcVB/gHB2UHFQdvCBQETBexOyn+DCk7OylFyNulubmlAZCluf4zAYsHB2UHFQf97AcH+AcVB2UHB28HAAAAAQAKAAoEpgSmADsAAAkBNjIXARYGKwEVMzU0NhcBFhQHAQYmPQEjFTMyFgcBBiInASY2OwE1IxUUBicBJjQ3ATYWHQEzNSMiJgE+AQgIFAgBBAcFCqrICggBCAgI/vgICsiqCgUH/vwIFAj++AgFCq/ICgj++AgIAQgICsivCgUDlgEICAj++AgKyK0KBAf+/AcVB/73BwQKrcgKCP74CAgBCAgKyK0KBAcBCQcVBwEEBwQKrcgKAAEAyAAAA4QETAAZAAATMzIWFREBNhYVERQGJwERFAYrASImNRE0NvpkFR0B0A8VFQ/+MB0VZBUdHQRMHRX+SgHFDggV/BgVCA4Bxf5KFR0dFQPoFR0AAAABAAAAAASwBEwAIwAAEzMyFhURATYWFREBNhYVERQGJwERFAYnAREUBisBIiY1ETQ2MmQVHQHQDxUB0A8VFQ/+MBUP/jAdFWQVHR0ETB0V/koBxQ4IFf5KAcUOCBX8GBUIDgHF/koVCA4Bxf5KFR0dFQPoFR0AAAABAJ0AGQSwBDMAFQAAAREUBicBERQGJwEmNDcBNhYVEQE2FgSwFQ/+MBUP/hQPDwHsDxUB0A8VBBr8GBUIDgHF/koVCA4B4A4qDgHgDggV/koBxQ4IAAAAAQDIABYEMwQ2AAsAABMBFhQHAQYmNRE0NvMDLhIS/NISGRkEMv4OCx4L/g4LDhUD6BUOAAIAyABkA4QD6AAPAB8AABMzMhYVERQGKwEiJjURNDYhMzIWFREUBisBIiY1ETQ2+sgVHR0VyBUdHQGlyBUdHRXIFR0dA+gdFfzgFR0dFQMgFR0dFfzgFR0dFQMgFR0AAAEAyABkBEwD6AAPAAABERQGIyEiJjURNDYzITIWBEwdFfzgFR0dFQMgFR0DtvzgFR0dFQMgFR0dAAAAAAEAAAAZBBMEMwAVAAABETQ2FwEWFAcBBiY1EQEGJjURNDYXAfQVDwHsDw/+FA8V/jAPFRUPAmQBthUIDv4gDioO/iAOCBUBtv47DggVA+gVCA4AAAH//gACBLMETwAjAAABNzIWFRMUBiMHIiY1AwEGJjUDAQYmNQM0NhcBAzQ2FwEDNDYEGGQUHgUdFWQVHQL+MQ4VAv4yDxUFFQ8B0gIVDwHSAh0ETgEdFfwYFR0BHRUBtf46DwkVAbX+OQ4JFAPoFQkP/j4BthQJDv49AbYVHQAAAQEsAAAD6ARMABkAAAEzMhYVERQGKwEiJjURAQYmNRE0NhcBETQ2A1JkFR0dFWQVHf4wDxUVDwHQHQRMHRX8GBUdHRUBtv47DggVA+gVCA7+OwG2FR0AAAIAZADIBLAESAALABsAAAkBFgYjISImNwE2MgEhMhYdARQGIyEiJj0BNDYCrgH1DwkW++4WCQ8B9Q8q/fcD6BUdHRX8GBUdHQQ5/eQPFhYPAhwP/UgdFWQVHR0VZBUdAAEAiP/8A3UESgAFAAAJAgcJAQN1/qABYMX92AIoA4T+n/6fxgIoAiYAAAAAAQE7//wEKARKAAUAAAkBJwkBNwQo/dnGAWH+n8YCI/3ZxgFhAWHGAAIAFwAXBJkEmQAPADMAAAAyHgIUDgIiLgI0PgEFIyIGHQEjIgYdARQWOwEVFBY7ATI2PQEzMjY9ATQmKwE1NCYB4+rWm1tbm9bq1ptbW5sBfWQVHZYVHR0Vlh0VZBUdlhUdHRWWHQSZW5vW6tabW1ub1urWm7odFZYdFWQVHZYVHR0Vlh0VZBUdlhUdAAAAAAIAFwAXBJkEmQAPAB8AAAAyHgIUDgIiLgI0PgEBISIGHQEUFjMhMjY9ATQmAePq1ptbW5vW6tabW1ubAkX+DBUdHRUB9BUdHQSZW5vW6tabW1ub1urWm/5+HRVkFR0dFWQVHQACABcAFwSZBJkADwAzAAAAMh4CFA4CIi4CND4BBCIPAScmIg8BBhQfAQcGFB8BFjI/ARcWMj8BNjQvATc2NC8BAePq1ptbW5vW6tabW1ubAeUZCXh4CRkJjQkJeHgJCY0JGQl4eAkZCY0JCXh4CQmNBJlbm9bq1ptbW5vW6tabrQl4eAkJjQkZCXh4CRkJjQkJeHgJCY0JGQl4eAkZCY0AAgAXABcEmQSZAA8AJAAAADIeAhQOAiIuAjQ+AQEnJiIPAQYUHwEWMjcBNjQvASYiBwHj6tabW1ub1urWm1tbmwEVVAcVCIsHB/IHFQcBdwcHiwcVBwSZW5vW6tabW1ub1urWm/4xVQcHiwgUCPEICAF3BxUIiwcHAAAAAAMAFwAXBJkEmQAPADsASwAAADIeAhQOAiIuAjQ+AQUiDgMVFDsBFjc+ATMyFhUUBgciDgUHBhY7ATI+AzU0LgMTIyIGHQEUFjsBMjY9ATQmAePq1ptbW5vW6tabW1ubAT8dPEIyIRSDHgUGHR8UFw4TARkOGhITDAIBDQ6tBx4oIxgiM0Q8OpYKDw8KlgoPDwSZW5vW6tabW1ub1urWm5ELHi9PMhkFEBQQFRIXFgcIBw4UHCoZCBEQKDhcNi9IKhsJ/eMPCpYKDw8KlgoPAAADABcAFwSZBJkADwAfAD4AAAAyHgIUDgIiLgI0PgEFIyIGHQEUFjsBMjY9ATQmAyMiBh0BFBY7ARUjIgYdARQWMyEyNj0BNCYrARE0JgHj6tabW1ub1urWm1tbmwGWlgoPDwqWCg8PCvoKDw8KS0sKDw8KAV4KDw8KSw8EmVub1urWm1tbm9bq1ptWDwqWCg8PCpYKD/7UDwoyCg/IDwoyCg8PCjIKDwETCg8AAgAAAAAEsASwAC8AXwAAATMyFh0BHgEXMzIWHQEUBisBDgEHFRQGKwEiJj0BLgEnIyImPQE0NjsBPgE3NTQ2ExUUBisBIiY9AQ4BBzMyFh0BFAYrAR4BFzU0NjsBMhYdAT4BNyMiJj0BNDY7AS4BAg2WCg9nlxvCCg8PCsIbl2cPCpYKD2eXG8IKDw8KwhuXZw+5DwqWCg9EZheoCg8PCqgXZkQPCpYKD0RmF6gKDw8KqBdmBLAPCsIbl2cPCpYKD2eXG8IKDw8KwhuXZw8KlgoPZ5cbwgoP/s2oCg8PCqgXZkQPCpYKD0RmF6gKDw8KqBdmRA8KlgoPRGYAAwAXABcEmQSZAA8AGwA/AAAAMh4CFA4CIi4CND4BBCIOARQeATI+ATQmBxcWFA8BFxYUDwEGIi8BBwYiLwEmND8BJyY0PwE2Mh8BNzYyAePq1ptbW5vW6tabW1ubAb/oxXJyxejFcnKaQAcHfHwHB0AHFQd8fAcVB0AHB3x8BwdABxUHfHwHFQSZW5vW6tabW1ub1urWmztyxejFcnLF6MVaQAcVB3x8BxUHQAcHfHwHB0AHFQd8fAcVB0AHB3x8BwAAAAMAFwAXBJkEmQAPABsAMAAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JgcXFhQHAQYiLwEmND8BNjIfATc2MgHj6tabW1ub1urWm1tbmwG/6MVycsXoxXJyg2oHB/7ACBQIyggIagcVB0/FBxUEmVub1urWm1tbm9bq1ps7csXoxXJyxejFfWoHFQf+vwcHywcVB2oICE/FBwAAAAMAFwAXBJkEmQAPABgAIQAAADIeAhQOAiIuAjQ+AQUiDgEVFBcBJhcBFjMyPgE1NAHj6tabW1ub1urWm1tbmwFLdMVyQQJLafX9uGhzdMVyBJlbm9bq1ptbW5vW6tabO3LFdHhpAktB0P24PnLFdHMAAAAAAQAXAFMEsAP5ABUAABMBNhYVESEyFh0BFAYjIREUBicBJjQnAgoQFwImFR0dFf3aFxD99hACRgGrDQoV/t0dFcgVHf7dFQoNAasNJgAAAAABAAAAUwSZA/kAFQAACQEWFAcBBiY1ESEiJj0BNDYzIRE0NgJ/AgoQEP32EBf92hUdHRUCJhcD8f5VDSYN/lUNChUBIx0VyBUdASMVCgAAAAEAtwAABF0EmQAVAAAJARYGIyERFAYrASImNREhIiY3ATYyAqoBqw0KFf7dHRXIFR3+3RUKDQGrDSYEif32EBf92hUdHRUCJhcQAgoQAAAAAQC3ABcEXQSwABUAAAEzMhYVESEyFgcBBiInASY2MyERNDYCJsgVHQEjFQoN/lUNJg3+VQ0KFQEjHQSwHRX92hcQ/fYQEAIKEBcCJhUdAAABAAAAtwSZBF0AFwAACQEWFAcBBiY1EQ4DBz4ENxE0NgJ/AgoQEP32EBdesKWBJAUsW4fHfhcEVf5VDSYN/lUNChUBIwIkRHVNabGdcUYHAQYVCgACAAAAAASwBLAAFQArAAABITIWFREUBi8BBwYiLwEmND8BJyY2ASEiJjURNDYfATc2Mh8BFhQPARcWBgNSASwVHRUOXvkIFAhqBwf5Xg4I/iH+1BUdFQ5e+QgUCGoHB/leDggEsB0V/tQVCA5e+QcHaggUCPleDhX7UB0VASwVCA5e+QcHaggUCPleDhUAAAACAEkASQRnBGcAFQArAAABFxYUDwEXFgYjISImNRE0Nh8BNzYyASEyFhURFAYvAQcGIi8BJjQ/AScmNgP2agcH+V4OCBX+1BUdFQ5e+QgU/QwBLBUdFQ5e+QgUCGoHB/leDggEYGoIFAj5Xg4VHRUBLBUIDl75B/3xHRX+1BUIDl75BwdqCBQI+V4OFQAAAAADABcAFwSZBJkADwAfAC8AAAAyHgIUDgIiLgI0PgEFIyIGFxMeATsBMjY3EzYmAyMiBh0BFBY7ATI2PQE0JgHj6tabW1ub1urWm1tbmwGz0BQYBDoEIxQ2FCMEOgQYMZYKDw8KlgoPDwSZW5vW6tabW1ub1urWm7odFP7SFB0dFAEuFB3+DA8KlgoPDwqWCg8AAAAABQAAAAAEsASwAEkAVQBhAGgAbwAAATIWHwEWHwEWFxY3Nj8BNjc2MzIWHwEWHwIeATsBMhYdARQGKwEiBh0BIREjESE1NCYrASImPQE0NjsBMjY1ND8BNjc+BAUHBhY7ATI2LwEuAQUnJgYPAQYWOwEyNhMhIiY1ESkBERQGIyERAQQJFAUFFhbEFQ8dCAsmxBYXERUXMA0NDgQZCAEPCj0KDw8KMgoP/nDI/nAPCjIKDw8KPQsOCRkFDgIGFRYfAp2mBwQK2woKAzMDEP41sQgQAzMDCgrnCwMe/okKDwGQAlgPCv6JBLAEAgIKDXYNCxUJDRZ2DQoHIREQFRh7LAkLDwoyCg8PCq8BLP7UrwoPDwoyCg8GBQQwgBkUAwgWEQ55ogcKDgqVCgSqnQcECo8KDgr8cg8KAXf+iQoPAZAAAAAAAgAAAAwErwSmACsASQAAATYWFQYCDgQuAScmByYOAQ8BBiY1NDc+ATc+AScuAT4BNz4GFyYGBw4BDwEOBAcOARY2Nz4CNz4DNz4BBI0IGgItQmxhi2KORDg9EQQRMxuZGhYqCFUYEyADCQIQOjEnUmFch3vAJQgdHyaiPT44XHRZUhcYDhItIRmKcVtGYWtbKRYEBKYDEwiy/t3IlVgxEQgLCwwBAQIbG5kYEyJAJghKFRE8Hzdff4U/M0o1JSMbL0QJGCYvcSEhHjZST2c1ODwEJygeW0AxJUBff1UyFAABAF0AHgRyBM8ATwAAAQ4BHgQXLgc+ATceAwYHDgQHBicmNzY3PgQuAScWDgMmJy4BJyY+BDcGHgM3PgEuAicmPgMCjScfCic4R0IgBBsKGAoQAwEJEg5gikggBhANPkpTPhZINx8SBgsNJysiCRZOQQoVNU1bYC9QZwICBAUWITsoCAYdJzIYHw8YIiYHDyJJYlkEz0OAZVxEOSQMBzgXOB42IzElKRIqg5Gnl0o3Z0c6IAYWCwYNAwQFIDhHXGF1OWiqb0sdBxUknF0XNTQ8PEUiNWNROBYJDS5AQVUhVZloUSkAAAAAA//cAGoE1ARGABsAPwBRAAAAMh4FFA4FIi4FND4EBSYGFxYVFAYiJjU0NzYmBwYHDgEXHgQyPgM3NiYnJgUHDgEXFhcWNj8BNiYnJicuAQIGpJ17bk85HBw6T257naKde25POhwcOU9uewIPDwYIGbD4sBcIBw5GWg0ECxYyWl+DiINfWjIWCwQMWv3/Iw8JCSU4EC0OIw4DDywtCyIERi1JXGJcSSpJXGJcSS0tSVxiXEkqSVxiXEncDwYTOT58sLB8OzcTBg9FcxAxEiRGXkQxMEVeRSQSMRF1HiQPLxJEMA0EDyIPJQ8sSRIEAAAABP/cAAAE1ASwABQAJwA7AEwAACEjNy4ENTQ+BTMyFzczEzceARUUDgMHNz4BNzYmJyYlBgcOARceBBc3LgE1NDc2JhcHDgEXFhcWNj8CJyYnLgECUJQfW6l2WSwcOU9ue51SPUEglCYvbIknUGqYUi5NdiYLBAw2/VFGWg0ECxIqSExoNSlrjxcIB3wjDwkJJTgQLQ4MFgMsLQsieBRhdHpiGxVJXGJcSS0Pef5StVXWNBpacm5jGq0xiD8SMRFGckVzEDESHjxRQTkNmhKnbjs3EwZwJA8vEkQwDQQPC1YELEkSBAAAAAP/ngAABRIEqwALABgAKAAAJwE2FhcBFgYjISImJSE1NDY7ATIWHQEhAQczMhYPAQ4BKwEiJi8BJjZaAoIUOBQCghUbJfryJRsBCgFZDwqWCg8BWf5DaNAUGAQ6BCMUNhQjBDoEGGQEKh8FIfvgIEdEhEsKDw8KSwLT3x0U/BQdHRT8FB0AAAABAGQAFQSwBLAAKAAAADIWFREBHgEdARQGJyURFh0BFAYvAQcGJj0BNDcRBQYmPQE0NjcBETQCTHxYAWsPFhgR/plkGhPNzRMaZP6ZERgWDwFrBLBYPv6t/rsOMRQpFA0M+f75XRRAFRAJgIAJEBVAFF0BB/kMDRQpFDEOAUUBUz4AAAARAAAAAARMBLAAHQAnACsALwAzADcAOwA/AEMARwBLAE8AUwBXAFsAXwBjAAABMzIWHQEzMhYdASE1NDY7ATU0NjsBMhYdASE1NDYBERQGIyEiJjURFxUzNTMVMzUzFTM1MxUzNTMVMzUFFTM1MxUzNTMVMzUzFTM1MxUzNQUVMzUzFTM1MxUzNTMVMzUzFTM1A1JkFR0yFR37tB0VMh0VZBUdAfQdAQ8dFfwYFR1kZGRkZGRkZGRk/HxkZGRkZGRkZGT8fGRkZGRkZGRkZASwHRUyHRWWlhUdMhUdHRUyMhUd/nD9EhUdHRUC7shkZGRkZGRkZGRkyGRkZGRkZGRkZGTIZGRkZGRkZGRkZAAAAAMAAAAZBXcElwAZACUANwAAARcWFA8BBiY9ASMBISImPQE0NjsBATM1NDYBBycjIiY9ATQ2MyEBFxYUDwEGJj0BIyc3FzM1NDYEb/kPD/kOFZ/9qP7dFR0dFdECWPEV/amNetEVHR0VASMDGvkPD/kOFfG1jXqfFQSN5g4qDuYOCBWW/agdFWQVHQJYlhUI/piNeh0VZBUd/k3mDioO5g4IFZa1jXqWFQgAAAABAAAAAASwBEwAEgAAEyEyFhURFAYjIQERIyImNRE0NmQD6Ck7Oyn9rP7QZCk7OwRMOyn9qCk7/tQBLDspAlgpOwAAAAMAZAAABEwEsAAJABMAPwAAEzMyFh0BITU0NiEzMhYdASE1NDYBERQOBSIuBTURIRUUFRwBHgYyPgYmNTQ9AZbIFR3+1B0C0cgVHf7UHQEPBhgoTGacwJxmTCgYBgEsAwcNFB8nNkI2Jx8TDwUFAQSwHRX6+hUdHRX6+hUd/nD+1ClJalZcPigoPlxWakkpASz6CRIVKyclIRsWEAgJEBccISUnKhURCPoAAAAB//8A1ARMA8IABQAAAQcJAScBBEzG/p/+n8UCJwGbxwFh/p/HAicAAQAAAO4ETQPcAAUAAAkCNwkBBE392v3ZxgFhAWEDFf3ZAifH/p8BYQAAAAAC/1EAZAVfA+gAFAApAAABITIWFREzMhYPAQYiLwEmNjsBESElFxYGKwERIRchIiY1ESMiJj8BNjIBlALqFR2WFQgO5g4qDuYOCBWW/oP+HOYOCBWWAYHX/RIVHZYVCA7mDioD6B0V/dkVDvkPD/kOFQGRuPkOFf5wyB0VAiYVDvkPAAABAAYAAASeBLAAMAAAEzMyFh8BITIWBwMOASMhFyEyFhQGKwEVFAYiJj0BIRUUBiImPQEjIiYvAQMjIiY0NjheERwEJgOAGB4FZAUsIf2HMAIXFR0dFTIdKh3+1B0qHR8SHQYFyTYUHh4EsBYQoiUY/iUVK8gdKh0yFR0dFTIyFR0dFTIUCQoDwR0qHQAAAAACAAAAAASwBEwACwAPAAABFSE1MzQ2MyEyFhUFIREhBLD7UMg7KQEsKTv9RASw+1AD6GRkKTs7Kcj84AACAAAAAAXcBEwADAAQAAATAxEzNDYzITIWFSEVBQEhAcjIyDspASwqOgH0ASz+1PtQASwDIP5wAlgpOzspyGT9RAK8AAEBRQAAA2sErwAbAAABFxYGKwERMzIWDwEGIi8BJjY7AREjIiY/ATYyAnvmDggVlpYVCA7mDioO5g4IFZaWFQgO5g4qBKD5DhX9pxUO+Q8P+Q4VAlkVDvkPAAAAAQABAUQErwNrABsAAAEXFhQPAQYmPQEhFRQGLwEmND8BNhYdASE1NDYDqPkODvkPFf2oFQ/5Dg75DxUCWBUDYOUPKQ/lDwkUl5cUCQ/lDykP5Q8JFZWVFQkAAAAEAAAAAASwBLAACQAZAB0AIQAAAQMuASMhIgYHAwUhIgYdARQWMyEyNj0BNCYFNTMVMzUzFQSRrAUkFP1gFCQFrAQt/BgpOzspA+gpOzv+q2RkZAGQAtwXLSgV/R1kOylkKTs7KWQpO8hkZGRkAAAAA/+cAGQEsARMAAsAIwAxAAAAMhYVERQGIiY1ETQDJSMTFgYjIisBIiYnAj0BNDU0PgE7ASUBFSIuAz0BND4CNwRpKh0dKh1k/V0mLwMRFQUCVBQdBDcCCwzIAqP8GAQOIhoWFR0dCwRMHRX8rhUdHRUDUhX8mcj+7BAIHBUBUQ76AgQQDw36/tT6AQsTKRwyGigUDAEAAAACAEoAAARmBLAALAA1AAABMzIWDwEeARcTFzMyFhQGBw4EIyIuBC8BLgE0NjsBNxM+ATcnJjYDFjMyNw4BIiYCKV4UEgYSU3oPP3YRExwaEggeZGqfTzl0XFU+LwwLEhocExF2Pw96UxIGEyQyNDUxDDdGOASwFRMlE39N/rmtHSkoBwQLHBYSCg4REg4FBAgoKR2tAUdNfhQgExr7vgYGMT09AAEAFAAUBJwEnAAXAAABNwcXBxcHFycHJwcnBzcnNyc3Jxc3FzcDIOBO6rS06k7gLZubLeBO6rS06k7gLZubA7JO4C2bmy3gTuq0tOpO4C2bmy3gTuq0tAADAAAAZASwBLAAIQAtAD0AAAEzMhYdAQchMhYdARQHAw4BKwEiJi8BIyImNRE0PwI+ARcPAREzFzMTNSE3NQEzMhYVERQGKwEiJjURNDYCijIoPBwBSCg8He4QLBf6B0YfHz0tNxSRYA0xG2SWZIjW+v4+Mv12ZBUdHRVkFR0dBLBRLJZ9USxkLR3+qBghMhkZJCcBkCQbxMYcKGTU1f6JZAF3feGv/tQdFf4MFR0dFQH0FR0AAAAAAwAAAAAEsARMACAAMAA8AAABMzIWFxMWHQEUBiMhFh0BFAYrASImLwImNRE0NjsBNgUzMhYVERQGKwEiJjURNDYhByMRHwEzNSchNQMCWPoXLBDuHTwo/rgcPCgyGzENYJEUNy09fP3pZBUdHRVkFR0dAl+IZJZkMjIBwvoETCEY/qgdLWQsUXYHlixRKBzGxBskAZAnJGRkHRX+DBUdHRUB9BUdZP6J1dSv4X0BdwADAAAAZAUOBE8AGwA3AEcAAAElNh8BHgEPASEyFhQGKwEDDgEjISImNRE0NjcXERchEz4BOwEyNiYjISoDLgQnJj8BJwUzMhYVERQGKwEiJjURNDYBZAFrHxZuDQEMVAEuVGxuVGqDBhsP/qoHphwOOmQBJYMGGw/LFRMSFv44AgoCCQMHAwUDAQwRklb9T2QVHR0VZBUdHQNp5hAWcA0mD3lMkE7+rRUoog0CDRElCkj+CVkBUxUoMjIBAgIDBQIZFrdT5B0V/gwVHR0VAfQVHQAAAAP/nABkBLAETwAdADYARgAAAQUeBBURFAYjISImJwMjIiY0NjMhJyY2PwE2BxcWBw4FKgIjIRUzMhYXEyE3ESUFMzIWFREUBisBIiY1ETQ2AdsBbgIIFBANrAf+qg8bBoNqVW1sVAEuVQsBDW4WSpIRDAIDBQMHAwkDCgH+Jd0PHAaCASZq/qoCUGQVHR0VZBUdHQRP5gEFEBEXC/3zDaIoFQFTTpBMeQ8mDXAWrrcWGQIFAwICAWQoFf6tWQH37OQdFf4MFR0dFQH0FR0AAAADAGEAAARMBQ4AGwA3AEcAAAAyFh0BBR4BFREUBiMhIiYvAQMmPwE+AR8BETQXNTQmBhURHAMOBAcGLwEHEyE3ESUuAQMhMhYdARQGIyEiJj0BNDYB3pBOAVMVKKIN/fMRJQoJ5hAWcA0mD3nGMjIBAgIDBQIZFrdT7AH3Wf6tFSiWAfQVHR0V/gwVHR0FDm5UaoMGGw/+qgemHA4OAWsfFm4NAQxUAS5U1ssVExIW/jgCCgIJAwcDBQMBDBGSVv6tZAElgwYb/QsdFWQVHR0VZBUdAAP//QAGA+gFFAAPAC0ASQAAASEyNj0BNCYjISIGHQEUFgEVFAYiJjURBwYmLwEmNxM+BDMhMhYVERQGBwEDFzc2Fx4FHAIVERQWNj0BNDY3JREnAV4B9BUdHRX+DBUdHQEPTpBMeQ8mDXAWEOYBBRARFwsCDQ2iKBX9iexTtxYZAgUDAgIBMjIoFQFTWQRMHRVkFR0dFWQVHfzmalRubFQBLlQMAQ1uFh8BawIIEw8Mpgf+qg8bBgHP/q1WkhEMAQMFAwcDCQIKAv44FhITFcsPGwaDASVkAAIAFgAWBJoEmgAPACUAAAAyHgIUDgIiLgI0PgEBJSYGHQEhIgYdARQWMyEVFBY3JTY0AeLs1ptbW5vW7NabW1ubAob+7RAX/u0KDw8KARMXEAETEASaW5vW7NabW1ub1uzWm/453w0KFYkPCpYKD4kVCg3fDSYAAAIAFgAWBJoEmgAPACUAAAAyHgIUDgIiLgI0PgENAQYUFwUWNj0BITI2PQE0JiMhNTQmAeLs1ptbW5vW7NabW1ubASX+7RAQARMQFwETCg8PCv7tFwSaW5vW7NabW1ub1uzWm+jfDSYN3w0KFYkPCpYKD4kVCgAAAAIAFgAWBJoEmgAPACUAAAAyHgIUDgIiLgI0PgEBAyYiBwMGFjsBERQWOwEyNjURMzI2AeLs1ptbW5vW7NabW1ubAkvfDSYN3w0KFYkPCpYKD4kVCgSaW5vW7NabW1ub1uzWm/5AARMQEP7tEBf+7QoPDwoBExcAAAIAFgAWBJoEmgAPACUAAAAyHgIUDgIiLgI0PgEFIyIGFREjIgYXExYyNxM2JisBETQmAeLs1ptbW5vW7NabW1ubAZeWCg+JFQoN3w0mDd8NChWJDwSaW5vW7NabW1ub1uzWm7sPCv7tFxD+7RAQARMQFwETCg8AAAMAGAAYBJgEmAAPAJYApgAAADIeAhQOAiIuAjQ+ASUOAwcGJgcOAQcGFgcOAQcGFgcUFgcyHgEXHgIXHgI3Fg4BFx4CFxQGFBcWNz4CNy4BJy4BJyIOAgcGJyY2NS4BJzYuAQYHBicmNzY3HgIXHgMfAT4CJyY+ATc+AzcmNzIWMjY3LgMnND4CJiceAT8BNi4CJwYHFB4BFS4CJz4BNxYyPgEB5OjVm1xcm9Xo1ZtcXJsBZA8rHDoKDz0PFD8DAxMBAzEFCRwGIgEMFhkHECIvCxU/OR0HFBkDDRQjEwcFaHUeISQDDTAMD0UREi4oLBAzDwQBBikEAQMLGhIXExMLBhAGKBsGBxYVEwYFAgsFAwMNFwQGCQcYFgYQCCARFwkKKiFBCwQCAQMDHzcLDAUdLDgNEiEQEgg/KhADGgMKEgoRBJhcm9Xo1ZtcXJvV6NWbEQwRBwkCAwYFBycPCxcHInIWInYcCUcYChQECA4QBAkuHgQPJioRFRscBAcSCgwCch0kPiAIAQcHEAsBAgsLIxcBMQENCQIPHxkCFBkdHB4QBgEBBwoMGBENBAMMJSAQEhYXDQ4qFBkKEhIDCQsXJxQiBgEOCQwHAQ0DBAUcJAwSCwRnETIoAwEJCwsLJQcKDBEAAAAAAQAAAAIErwSFABYAAAE2FwUXNxYGBw4BJwEGIi8BJjQ3ASY2AvSkjv79kfsGUE08hjv9rA8rD28PDwJYIk8EhVxliuh+WYcrIgsW/awQEG4PKxACV2XJAAYAAABgBLAErAAPABMAIwAnADcAOwAAEyEyFh0BFAYjISImPQE0NgUjFTMFITIWHQEUBiMhIiY9ATQ2BSEVIQUhMhYdARQGIyEiJj0BNDYFIRUhZAPoKTs7KfwYKTs7BBHIyPwYA+gpOzsp/BgpOzsEEf4MAfT8GAPoKTs7KfwYKTs7BBH+1AEsBKw7KWQpOzspZCk7ZGTIOylkKTs7KWQpO2RkyDspZCk7OylkKTtkZAAAAAIAZAAABEwEsAALABEAABMhMhYUBiMhIiY0NgERBxEBIZYDhBUdHRX8fBUdHQI7yP6iA4QEsB0qHR0qHf1E/tTIAfQB9AAAAAMAAABkBLAEsAAXABsAJQAAATMyFh0BITIWFREhNSMVIRE0NjMhNTQ2FxUzNQEVFAYjISImPQEB9MgpOwEsKTv+DMj+DDspASw7KcgB9Dsp/BgpOwSwOylkOyn+cGRkAZApO2QpO2RkZP1EyCk7OynIAAAABAAAAAAEsASwABUAKwBBAFcAABMhMhYPARcWFA8BBiIvAQcGJjURNDYpATIWFREUBi8BBwYiLwEmND8BJyY2ARcWFA8BFxYGIyEiJjURNDYfATc2MgU3NhYVERQGIyEiJj8BJyY0PwE2MhcyASwVCA5exwcHaggUCMdeDhUdAzUBLBUdFQ5exwgUCGoHB8deDgj+L2oHB8deDggV/tQVHRUOXscIFALLXg4VHRX+1BUIDl7HBwdqCBQIBLAVDl7HCBQIagcHx14OCBUBLBUdHRX+1BUIDl7HBwdqCBQIx14OFf0maggUCMdeDhUdFQEsFQgOXscHzl4OCBX+1BUdFQ5exwgUCGoHBwAAAAYAAAAABKgEqAAPABsAIwA7AEMASwAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JiQyFhQGIiY0JDIWFAYjIicHFhUUBiImNTQ2PwImNTQEMhYUBiImNCQyFhQGIiY0Advy3Z9fX5/d8t2gXl6gAcbgv29vv+C/b2/+LS0gIC0gAUwtICAWDg83ETNIMykfegEJ/octICAtIAIdLSAgLSAEqF+f3fLdoF5eoN3y3Z9Xb7/gv29vv+C/BiAtISEtICAtIQqRFxwkMzMkIDEFfgEODhekIC0gIC0gIC0gIC0AAf/YAFoEuQS8AFsAACUBNjc2JicmIyIOAwcABw4EFx4BMzI3ATYnLgEjIgcGBwEOASY0NwA3PgEzMhceARcWBgcOBgcGIyImJyY2NwE2NzYzMhceARcWBgcBDgEnLgECIgHVWwgHdl8WGSJBMD8hIP6IDx4eLRMNBQlZN0ozAiQkEAcdEhoYDRr+qw8pHA4BRyIjQS4ODyw9DQ4YIwwod26La1YOOEBGdiIwGkQB/0coW2tQSE5nDxE4Qv4eDyoQEAOtAdZbZWKbEQQUGjIhH/6JDxsdNSg3HT5CMwIkJCcQFBcMGv6uDwEcKQ4BTSIjIQEINykvYyMLKnhuiWZMBxtAOU6+RAH/SBg3ISSGV121Qv4kDwIPDyYAAAACAGQAWASvBEQAGQBEAAABPgIeAhUUDgMHLgQ1ND4CHgEFIg4DIi4DIyIGFRQeAhcWFx4EMj4DNzY3PgQ1NCYCiTB7eHVYNkN5hKg+PqeFeEM4WnZ4eQEjIT8yLSohJyktPyJDbxtBMjMPBw86KzEhDSIzKUAMBAgrKT8dF2oDtURIBS1TdkA5eYB/slVVsn+AeTlAdlMtBUgtJjY1JiY1NiZvTRc4SjQxDwcOPCouGBgwKEALBAkpKkQqMhNPbQACADn/8gR3BL4AFwAuAAAAMh8BFhUUBg8BJi8BNycBFwcvASY0NwEDNxYfARYUBwEGIi8BJjQ/ARYfAQcXAQKru0KNQjgiHR8uEl/3/nvUaRONQkIBGxJpCgmNQkL+5UK6Qo1CQjcdLhJf9wGFBL5CjUJeKmsiHTUuEl/4/nvUahKNQrpCARv+RmkICY1CukL+5UJCjUK7Qjc3LxFf+AGFAAAAAAMAyAAAA+gEsAARABUAHQAAADIeAhURFAYjISImNRE0PgEHESERACIGFBYyNjQCBqqaZDo7Kf2oKTs8Zj4CWP7/Vj09Vj0EsB4uMhX8Ryk7OykDuRUzLar9RAK8/RY9Vj09VgABAAAAAASwBLAAFgAACQEWFAYiLwEBEScBBRMBJyEBJyY0NjIDhgEbDx0qDiT+6dT+zP7oywEz0gEsAQsjDx0qBKH+5g8qHQ8j/vX+1NL+zcsBGAE01AEXJA4qHQAAAAADAScAEQQJBOAAMgBAAEsAAAEVHgQXIy4DJxEXHgQVFAYHFSM1JicuASczHgEXEScuBDU0PgI3NRkBDgMVFB4DFxYXET4ENC4CArwmRVI8LAKfBA0dMydAIjxQNyiym2SWVygZA4sFV0obLkJOMCAyVWg6HSoqFQ4TJhkZCWgWKTEiGBkzNwTgTgUTLD9pQiQuLBsH/s0NBxMtPGQ+i6oMTU8QVyhrVk1iEAFPCA4ZLzlYNkZwSCoGTf4SARIEDh02Jh0rGRQIBgPQ/soCCRYgNEM0JRkAAAABAGQAZgOUBK0ASgAAATIeARUjNC4CIyIGBwYVFB4BFxYXMxUjFgYHBgc+ATM2FjMyNxcOAyMiLgEHDgEPASc+BTc+AScjNTMmJy4CPgE3NgIxVJlemSc8OxolVBQpGxoYBgPxxQgVFS02ImIWIIwiUzUyHzY4HCAXanQmJ1YYFzcEGAcTDBEJMAwk3aYXFQcKAg4tJGEErVCLTig/IhIdFSw5GkowKgkFZDKCHj4yCg8BIh6TExcIASIfBAMaDAuRAxAFDQsRCjePR2QvORQrREFMIVgAAAACABn//wSXBLAADwAfAAABMzIWDwEGIi8BJjY7AREzBRcWBisBESMRIyImPwE2MgGQlhUIDuYOKg7mDggVlsgCF+YOCBWWyJYVCA7mDioBLBYO+g8P+g4WA4QQ+Q4V/HwDhBUO+Q8AAAQAGf//A+gEsAAHABcAGwAlAAABIzUjFSMRIQEzMhYPAQYiLwEmNjsBETMFFTM1EwczFSE1NyM1IQPoZGRkASz9qJYVCA7mDioO5g4IFZbIAZFkY8jI/tTIyAEsArxkZAH0/HwWDvoPD/oOFgOEZMjI/RL6ZJb6ZAAAAAAEABn//wPoBLAADwAZACEAJQAAATMyFg8BBiIvASY2OwERMwUHMxUhNTcjNSERIzUjFSMRIQcVMzUBkJYVCA7mDioO5g4IFZbIAljIyP7UyMgBLGRkZAEsx2QBLBYO+g8P+g4WA4SW+mSW+mT7UGRkAfRkyMgAAAAEABn//wRMBLAADwAVABsAHwAAATMyFg8BBiIvASY2OwERMwEjESM1MxMjNSMRIQcVMzUBkJYVCA7mDioO5g4IFZbIAlhkZMhkZMgBLMdkASwWDvoPD/oOFgOE/gwBkGT7UGQBkGTIyAAAAAAEABn//wRMBLAADwAVABkAHwAAATMyFg8BBiIvASY2OwERMwEjNSMRIQcVMzUDIxEjNTMBkJYVCA7mDioO5g4IFZbIArxkyAEsx2QBZGTIASwWDvoPD/oOFgOE/gxkAZBkyMj7tAGQZAAAAAAFABn//wSwBLAADwATABcAGwAfAAABMzIWDwEGIi8BJjY7AREzBSM1MxMhNSETITUhEyE1IQGQlhUIDuYOKg7mDggVlsgB9MjIZP7UASxk/nABkGT+DAH0ASwWDvoPD/oOFgOEyMj+DMj+DMj+DMgABQAZ//8EsASwAA8AEwAXABsAHwAAATMyFg8BBiIvASY2OwERMwUhNSEDITUhAyE1IQMjNTMBkJYVCA7mDioO5g4IFZbIAyD+DAH0ZP5wAZBk/tQBLGTIyAEsFg76Dw/6DhYDhMjI/gzI/gzI/gzIAAIAAAAABEwETAAPAB8AAAEhMhYVERQGIyEiJjURNDYFISIGFREUFjMhMjY1ETQmAV4BkKK8u6P+cKW5uQJn/gwpOzspAfQpOzsETLuj/nClubmlAZClucg7Kf4MKTs7KQH0KTsAAAAAAwAAAAAETARMAA8AHwArAAABITIWFREUBiMhIiY1ETQ2BSEiBhURFBYzITI2NRE0JgUXFhQPAQYmNRE0NgFeAZClubml/nCju7wCZP4MKTs7KQH0KTs7/m/9ERH9EBgYBEy5pf5wpbm5pQGQo7vIOyn+DCk7OykB9Ck7gr4MJAy+DAsVAZAVCwAAAAADAAAAAARMBEwADwAfACsAAAEhMhYVERQGIyEiJjURNDYFISIGFREUFjMhMjY1ETQmBSEyFg8BBiIvASY2AV4BkKO7uaX+cKW5uQJn/gwpOzspAfQpOzv+FQGQFQsMvgwkDL4MCwRMvKL+cKW5uaUBkKO7yDsp/gwpOzspAfQpO8gYEP0REf0QGAAAAAMAAAAABEwETAAPAB8AKwAAASEyFhURFAYjISImNRE0NgUhIgYVERQWMyEyNjURNCYFFxYGIyEiJj8BNjIBXgGQpbm5pf5wo7u5Amf+DCk7OykB9Ck7O/77vgwLFf5wFQsMvgwkBEy5pf5wo7u8ogGQpbnIOyn+DCk7OykB9Ck7z/0QGBgQ/REAAAAAAgAAAAAFFARMAB8ANQAAASEyFhURFAYjISImPQE0NjMhMjY1ETQmIyEiJj0BNDYHARYUBwEGJj0BIyImPQE0NjsBNTQ2AiYBkKW5uaX+cBUdHRUBwik7Oyn+PhUdHb8BRBAQ/rwQFvoVHR0V+hYETLml/nCluR0VZBUdOykB9Ck7HRVkFR3p/uQOJg7+5A4KFZYdFcgVHZYVCgAAAQDZAAID1wSeACMAAAEXFgcGAgclMhYHIggBBwYrAScmNz4BPwEhIicmNzYANjc2MwMZCQgDA5gCASwYEQ4B/vf+8wQMDgkJCQUCUCcn/tIXCAoQSwENuwUJEASeCQoRC/5TBwEjEv7K/sUFDwgLFQnlbm4TFRRWAS/TBhAAAAACAAAAAAT+BEwAHwA1AAABITIWHQEUBiMhIgYVERQWMyEyFh0BFAYjISImNRE0NgUBFhQHAQYmPQEjIiY9ATQ2OwE1NDYBXgGQFR0dFf4+KTs7KQHCFR0dFf5wpbm5AvEBRBAQ/rwQFvoVHR0V+hYETB0VZBUdOyn+DCk7HRVkFR25pQGQpbnp/uQOJg7+5A4KFZYdFcgVHZYVCgACAAAAAASwBLAAFQAxAAABITIWFREUBi8BAQYiLwEmNDcBJyY2ASMiBhURFBYzITI2PQE3ERQGIyEiJjURNDYzIQLuAZAVHRUObf7IDykPjQ8PAThtDgj+75wpOzspAfQpO8i7o/5wpbm5pQEsBLAdFf5wFQgObf7IDw+NDykPAThtDhX+1Dsp/gwpOzsplMj+1qW5uaUBkKW5AAADAA4ADgSiBKIADwAbACMAAAAyHgIUDgIiLgI0PgEEIg4BFB4BMj4BNCYEMhYUBiImNAHh7tmdXV2d2e7ZnV1dnQHD5sJxccLmwnFx/nugcnKgcgSiXZ3Z7tmdXV2d2e7ZnUdxwubCcXHC5sJzcqBycqAAAAMAAAAABEwEsAAVAB8AIwAAATMyFhURMzIWBwEGIicBJjY7ARE0NgEhMhYdASE1NDYFFTM1AcLIFR31FAoO/oEOJw3+hQ0JFfod/oUD6BUd+7QdA2dkBLAdFf6iFg/+Vg8PAaoPFgFeFR38fB0V+voVHWQyMgAAAAMAAAAABEwErAAVAB8AIwAACQEWBisBFRQGKwEiJj0BIyImNwE+AQEhMhYdASE1NDYFFTM1AkcBeg4KFfQiFsgUGPoUCw4Bfw4n/fkD6BUd+7QdA2dkBJ7+TQ8g+hQeHRX6IQ8BrxAC/H8dFfr6FR1kMjIAAwAAAAAETARLABQAHgAiAAAJATYyHwEWFAcBBiInASY0PwE2MhcDITIWHQEhNTQ2BRUzNQGMAXEHFQeLBwf98wcVB/7cBweLCBUH1APoFR37tB0DZ2QC0wFxBweLCBUH/fMICAEjCBQIiwcH/dIdFfr6FR1kMjIABAAAAAAETASbAAkAGQAjACcAABM3NjIfAQcnJjQFNzYWFQMOASMFIiY/ASc3ASEyFh0BITU0NgUVMzWHjg4qDk3UTQ4CFtIOFQIBHRX9qxUIDtCa1P49A+gVHfu0HQNnZAP/jg4OTdRMDyqa0g4IFf2pFB4BFQ7Qm9T9Oh0V+voVHWQyMgAAAAQAAAAABEwEsAAPABkAIwAnAAABBR4BFRMUBi8BByc3JyY2EwcGIi8BJjQ/AQEhMhYdASE1NDYFFTM1AV4CVxQeARUO0JvUm9IOCMNMDyoOjg4OTf76A+gVHfu0HQNnZASwAgEdFf2rFQgO0JrUmtIOFf1QTQ4Ojg4qDk3+WB0V+voVHWQyMgACAAT/7ASwBK8ABQAIAAAlCQERIQkBFQEEsP4d/sb+cQSs/TMCq2cBFP5xAacDHPz55gO5AAAAAAIAAABkBEwEsAAVABkAAAERFAYrAREhESMiJjURNDY7AREhETMHIzUzBEwdFZb9RJYVHR0V+gH0ZMhkZAPo/K4VHQGQ/nAdFQPoFB7+1AEsyMgAAAMAAABFBN0EsAAWABoALwAAAQcBJyYiDwEhESMiJjURNDY7AREhETMHIzUzARcWFAcBBiIvASY0PwE2Mh8BATYyBEwC/tVfCRkJlf7IlhUdHRX6AfRkyGRkAbBqBwf+XAgUCMoICGoHFQdPASkHFQPolf7VXwkJk/5wHRUD6BQe/tQBLMjI/c5qBxUH/lsHB8sHFQdqCAhPASkHAAMAAAANBQcEsAAWABoAPgAAAREHJy4BBwEhESMiJjURNDY7AREhETMHIzUzARcWFA8BFxYUDwEGIi8BBwYiLwEmND8BJyY0PwE2Mh8BNzYyBExnhg8lEP72/reWFR0dFfoB9GTIZGQB9kYPD4ODDw9GDykPg4MPKQ9GDw+Dgw8PRg8pD4ODDykD6P7zZ4YPAw7+9v5wHRUD6BQe/tQBLMjI/YxGDykPg4MPKQ9GDw+Dgw8PRg8pD4ODDykPRg8Pg4MPAAADAAAAFQSXBLAAFQAZAC8AAAERISIGHQEhESMiJjURNDY7AREhETMHIzUzEzMyFh0BMzIWDwEGIi8BJjY7ATU0NgRM/qIVHf4MlhUdHRX6AfRkyGRklmQVHZYVCA7mDioO5g4IFZYdA+j+1B0Vlv5wHRUD6BQe/tQBLMjI/agdFfoVDuYODuYOFfoVHQAAAAADAAAAAASXBLAAFQAZAC8AAAERJyYiBwEhESMiJjURNDY7AREhETMHIzUzExcWBisBFRQGKwEiJj0BIyImPwE2MgRMpQ4qDv75/m6WFR0dFfoB9GTIZGTr5g4IFZYdFWQVHZYVCA7mDioD6P5wpQ8P/vf+cB0VA+gUHv7UASzIyP2F5Q8V+hQeHhT6FQ/lDwADAAAAyASwBEwACQATABcAABMhMhYdASE1NDYBERQGIyEiJjURExUhNTIETBUd+1AdBJMdFfu0FR1kAZAETB0VlpYVHf7U/doVHR0VAib+1MjIAAAGAAMAfQStBJcADwAZAB0ALQAxADsAAAEXFhQPAQYmPQEhNSE1NDYBIyImPQE0NjsBFyM1MwE3NhYdASEVIRUUBi8BJjQFIzU7AjIWHQEUBisBA6f4Dg74DhX+cAGQFf0vMhUdHRUyyGRk/oL3DhUBkP5wFQ73DwOBZGRkMxQdHRQzBI3mDioO5g4IFZbIlhUI/oUdFWQVHcjI/cvmDggVlsiWFQgO5g4qecgdFWQVHQAAAAACAGQAAASwBLAAFgBRAAABJTYWFREUBisBIiY1ES4ENRE0NiUyFh8BERQOAg8BERQGKwEiJjURLgQ1ETQ+AzMyFh8BETMRPAE+AjMyFh8BETMRND4DA14BFBklHRXIFR0EDiIaFiX+4RYZAgEVHR0LCh0VyBUdBA4iGhYBBwoTDRQZAgNkBQkVDxcZAQFkAQUJFQQxdBIUH/uuFR0dFQGNAQgbHzUeAWcfRJEZDA3+Phw/MSkLC/5BFR0dFQG/BA8uLkAcAcICBxENCxkMDf6iAV4CBxENCxkMDf6iAV4CBxENCwABAGQAAASwBEwAMwAAARUiDgMVERQWHwEVITUyNjURIREUFjMVITUyPgM1ETQmLwE1IRUiBhURIRE0JiM1BLAEDiIaFjIZGf5wSxn+DBlL/nAEDiIaFjIZGQGQSxkB9BlLBEw4AQUKFA78iBYZAQI4OA0lAYr+diUNODgBBQoUDgN4FhkBAjg4DSX+dgGKJQ04AAAABgAAAAAETARMAAwAHAAgACQAKAA0AAABITIWHQEjBTUnITchBSEyFhURFAYjISImNRE0NhcVITUBBTUlBRUhNQUVFAYjIQchJyE3MwKjAXcVHWn+2cj+cGQBd/4lASwpOzsp/tQpOzspASwCvP5wAZD8GAEsArwdFf6JZP6JZAGQyGkD6B0VlmJiyGTIOyn+DCk7OykB9Ck7ZMjI/veFo4XGyMhm+BUdZGTIAAEAEAAQBJ8EnwAmAAATNzYWHwEWBg8BHgEXNz4BHwEeAQ8BBiIuBicuBTcRohEuDosOBhF3ZvyNdxEzE8ATBxGjAw0uMUxPZWZ4O0p3RjITCwED76IRBhPCFDERdo78ZXYRBA6IDi8RogEECBUgNUNjO0qZfHNVQBAAAAACAAAAAASwBEwAIwBBAAAAMh4EHwEVFAYvAS4BPQEmIAcVFAYPAQYmPQE+BRIyHgIfARUBHgEdARQGIyEiJj0BNDY3ATU0PgIB/LimdWQ/LAkJHRTKFB2N/sKNHRTKFB0DDTE7ZnTKcFImFgEBAW0OFR0V+7QVHRUOAW0CFiYETBUhKCgiCgrIFRgDIgMiFZIYGJIVIgMiAxgVyAQNJyQrIP7kExwcCgoy/tEPMhTUFR0dFdQUMg8BLzIEDSEZAAADAAAAAASwBLAADQAdACcAAAEHIScRMxUzNTMVMzUzASEyFhQGKwEXITcjIiY0NgMhMhYdASE1NDYETMj9qMjIyMjIyPyuArwVHR0VDIn8SokMFR0dswRMFR37UB0CvMjIAfTIyMjI/OAdKh1kZB0qHf7UHRUyMhUdAAAAAwBkAAAEsARMAAkAEwAdAAABIyIGFREhETQmASMiBhURIRE0JgEhETQ2OwEyFhUCvGQpOwEsOwFnZCk7ASw7/Rv+1DspZCk7BEw7KfwYA+gpO/7UOyn9RAK8KTv84AGQKTs7KQAAAAAF/5wAAASwBEwADwATAB8AJQApAAATITIWFREUBiMhIiY1ETQ2FxEhEQUjFTMRITUzNSMRIQURByMRMwcRMxHIArx8sLB8/UR8sLAYA4T+DMjI/tTIyAEsAZBkyMhkZARMsHz+DHywsHwB9HywyP1EArzIZP7UZGQBLGT+1GQB9GT+1AEsAAAABf+cAAAEsARMAA8AEwAfACUAKQAAEyEyFhURFAYjISImNRE0NhcRIREBIzUjFSMRMxUzNTMFEQcjETMHETMRyAK8fLCwfP1EfLCwGAOE/gxkZGRkZGQBkGTIyGRkBEywfP4MfLCwfAH0fLDI/UQCvP2oyMgB9MjIZP7UZAH0ZP7UASwABP+cAAAEsARMAA8AEwAbACMAABMhMhYVERQGIyEiJjURNDYXESERBSMRMxUhESEFIxEzFSERIcgCvHywsHz9RHywsBgDhP4MyMj+1AEsAZDIyP7UASwETLB8/gx8sLB8AfR8sMj9RAK8yP7UZAH0ZP7UZAH0AAAABP+cAAAEsARMAA8AEwAWABkAABMhMhYVERQGIyEiJjURNDYXESERAS0BDQERyAK8fLCwfP1EfLCwGAOE/gz+1AEsAZD+1ARMsHz+DHywsHwB9HywyP1EArz+DJaWlpYBLAAAAAX/nAAABLAETAAPABMAFwAgACkAABMhMhYVERQGIyEiJjURNDYXESERAyERIQcjIgYVFBY7AQERMzI2NTQmI8gCvHywsHz9RHywsBgDhGT9RAK8ZIImOTYpgv4Mgik2OSYETLB8/gx8sLB8AfR8sMj9RAK8/agB9GRWQUFUASz+1FRBQVYAAAAF/5wAAASwBEwADwATAB8AJQApAAATITIWFREUBiMhIiY1ETQ2FxEhEQUjFTMRITUzNSMRIQEjESM1MwMjNTPIArx8sLB8/UR8sLAYA4T+DMjI/tTIyAEsAZBkZMjIZGQETLB8/gx8sLB8AfR8sMj9RAK8yGT+1GRkASz+DAGQZP4MZAAG/5wAAASwBEwADwATABkAHwAjACcAABMhMhYVERQGIyEiJjURNDYXESERBTMRIREzASMRIzUzBRUzNQEjNTPIArx8sLB8/UR8sLAYA4T9RMj+1GQCWGRkyP2oZAEsZGQETLB8/gx8sLB8AfR8sMj9RAK8yP5wAfT+DAGQZMjIyP7UZAAF/5wAAASwBEwADwATABwAIgAmAAATITIWFREUBiMhIiY1ETQ2FxEhEQEHIzU3NSM1IQEjESM1MwMjNTPIArx8sLB8/UR8sLAYA4T+DMdkx8gBLAGQZGTIx2RkBEywfP4MfLCwfAH0fLDI/UQCvP5wyDLIlmT+DAGQZP4MZAAAAAMACQAJBKcEpwAPABsAJQAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JgchFSEVISc1NyEB4PDbnl5entvw255eXp4BxeTCcXHC5MJxcWz+1AEs/tRkZAEsBKdentvw255eXp7b8NueTHHC5MJxccLkwtDIZGTIZAAAAAAEAAkACQSnBKcADwAbACcAKwAAADIeAhQOAiIuAjQ+AQQiDgEUHgEyPgE0JgcVBxcVIycjFSMRIQcVMzUB4PDbnl5entvw255eXp4BxeTCcXHC5MJxcWwyZGRklmQBLMjIBKdentvw255eXp7b8NueTHHC5MJxccLkwtBkMmQyZGQBkGRkZAAAAv/y/50EwgRBACAANgAAATIWFzYzMhYUBisBNTQmIyEiBh0BIyImNTQ2NyY1ND4BEzMyFhURMzIWDwEGIi8BJjY7ARE0NgH3brUsLC54qqp4gB0V/tQVHd5QcFZBAmKqepYKD4kVCg3fDSYN3w0KFYkPBEF3YQ6t8a36FR0dFfpzT0VrDhMSZKpi/bMPCv7tFxD0EBD0EBcBEwoPAAAAAAL/8v+cBMMEQQAcADMAAAEyFhc2MzIWFxQGBwEmIgcBIyImNTQ2NyY1ND4BExcWBisBERQGKwEiJjURIyImNzY3NjIB9m62LCsueaoBeFr+hg0lDf6DCU9xVkECYqnm3w0KFYkPCpYKD4kVCg3HGBMZBEF3YQ+teGOkHAFoEBD+k3NPRWsOExNkqWP9kuQQF/7tCg8PCgETFxDMGBMAAAABAGQAAARMBG0AGAAAJTUhATMBMwkBMwEzASEVIyIGHQEhNTQmIwK8AZD+8qr+8qr+1P7Uqv7yqv7yAZAyFR0BkB0VZGQBLAEsAU3+s/7U/tRkHRUyMhUdAAAAAAEAeQAABDcEmwAvAAABMhYXHgEVFAYHFhUUBiMiJxUyFh0BITU0NjM1BiMiJjU0Ny4BNTQ2MzIXNCY1NDYCWF6TGll7OzIJaUo3LRUd/tQdFS03SmkELzlpSgUSAqMEm3FZBoNaPWcfHRpKaR77HRUyMhUd+x5pShIUFVg1SmkCAhAFdKMAAAAGACcAFASJBJwAEQAqAEIASgBiAHsAAAEWEgIHDgEiJicmAhI3PgEyFgUiBw4BBwYWHwEWMzI3Njc2Nz4BLwEmJyYXIgcOAQcGFh8BFjMyNz4BNz4BLwEmJyYWJiIGFBYyNjciBw4BBw4BHwEWFxYzMjc+ATc2Ji8BJhciBwYHBgcOAR8BFhcWMzI3PgE3NiYvASYD8m9PT29T2dzZU29PT29T2dzZ/j0EBHmxIgQNDCQDBBcGG0dGYAsNAwkDCwccBAVQdRgEDA0iBAQWBhJROQwMAwkDCwf5Y4xjY4xjVhYGElE6CwwDCQMLBwgEBVB1GAQNDCIEjRcGG0dGYAsNAwkDCwcIBAR5sSIEDQwkAwPyb/7V/tVvU1dXU28BKwErb1NXVxwBIrF5DBYDCQEWYEZHGwMVDCMNBgSRAhh1UA0WAwkBFTpREgMVCyMMBwT6Y2OMY2MVFTpREQQVCyMMBwQCGHVQDRYDCQEkFmBGRxsDFQwjDQYEASKxeQwWAwkBAAAABQBkAAAD6ASwAAwADwAWABwAIgAAASERIzUhFSERNDYzIQEjNQMzByczNTMDISImNREFFRQGKwECvAEstP6s/oQPCgI/ASzIZKLU1KJktP51Cg8DhA8KwwMg/oTIyALzCg/+1Mj84NTUyP4MDwoBi8jDCg8AAAAABQBkAAAD6ASwAAkADAATABoAIQAAASERCQERNDYzIQEjNRMjFSM1IzcDISImPQEpARUUBisBNQK8ASz+ov3aDwoCPwEsyD6iZKLUqv6dCg8BfAIIDwqbAyD9+AFe/doERwoP/tTI/HzIyNT+ZA8KNzcKD1AAAAAAAwAAAAAEsAP0AAgAGQAfAAABIxUzFyERIzcFMzIeAhUhFSEDETM0PgIBMwMhASEEiqJkZP7UotT9EsgbGiEOASz9qMhkDiEaAnPw8PzgASwB9AMgyGQBLNTUBBErJGT+ogHCJCsRBP5w/nAB9AAAAAMAAAAABEwETAAZADIAOQAAATMyFh0BMzIWHQEUBiMhIiY9ATQ2OwE1NDYFNTIWFREUBiMhIic3ARE0NjMVFBYzITI2AQc1IzUzNQKKZBUdMhUdHRX+1BUdHRUyHQFzKTs7Kf2oARP2/ro7KVg+ASw+WP201MjIBEwdFTIdFWQVHR0VZBUdMhUd+pY7KfzgKTsE9gFGAUQpO5Y+WFj95tSiZKIAAwBkAAAEvARMABkANgA9AAABMzIWHQEzMhYdARQGIyEiJj0BNDY7ATU0NgU1MhYVESMRMxQOAiMhIiY1ETQ2MxUUFjMhMjYBBzUjNTM1AcJkFR0yFR0dFf7UFR0dFTIdAXMpO8jIDiEaG/2oKTs7KVg+ASw+WAGc1MjIBEwdFTIdFWQVHR0VZBUdMhUd+pY7Kf4M/tQkKxEEOykDICk7lj5YWP3m1KJkogAAAAP/ogAABRYE1AALABsAHwAACQEWBiMhIiY3ATYyEyMiBhcTHgE7ATI2NxM2JgMVMzUCkgJ9FyAs+wQsIBcCfRZARNAUGAQ6BCMUNhQjBDoEGODIBK37sCY3NyYEUCf+TB0U/tIUHR0UAS4UHf4MZGQAAAAACQAAAAAETARMAA8AHwAvAD8ATwBfAG8AfwCPAAABMzIWHQEUBisBIiY9ATQ2EzMyFh0BFAYrASImPQE0NiEzMhYdARQGKwEiJj0BNDYBMzIWHQEUBisBIiY9ATQ2ITMyFh0BFAYrASImPQE0NiEzMhYdARQGKwEiJj0BNDYBMzIWHQEUBisBIiY9ATQ2ITMyFh0BFAYrASImPQE0NiEzMhYdARQGKwEiJj0BNDYBqfoKDw8K+goPDwr6Cg8PCvoKDw8BmvoKDw8K+goPD/zq+goPDwr6Cg8PAZr6Cg8PCvoKDw8BmvoKDw8K+goPD/zq+goPDwr6Cg8PAZr6Cg8PCvoKDw8BmvoKDw8K+goPDwRMDwqWCg8PCpYKD/7UDwqWCg8PCpYKDw8KlgoPDwqWCg/+1A8KlgoPDwqWCg8PCpYKDw8KlgoPDwqWCg8PCpYKD/7UDwqWCg8PCpYKDw8KlgoPDwqWCg8PCpYKDw8KlgoPAAAAAwAAAAAEsAUUABkAKQAzAAABMxUjFSEyFg8BBgchJi8BJjYzITUjNTM1MwEhMhYUBisBFyE3IyImNDYDITIWHQEhNTQ2ArxkZAFePjEcQiko/PwoKUIcMT4BXmRkyP4+ArwVHR0VDIn8SooNFR0dswRMFR37UB0EsMhkTzeEUzMzU4Q3T2TIZPx8HSodZGQdKh3+1B0VMjIVHQAABAAAAAAEsAUUAAUAGQArADUAAAAyFhUjNAchFhUUByEyFg8BIScmNjMhJjU0AyEyFhQGKwEVBSElNSMiJjQ2AyEyFh0BITU0NgIwUDnCPAE6EgMBSCkHIq/9WrIiCikBSAOvArwVHR0VlgET/EoBE5YVHR2zBEwVHftQHQUUOykpjSUmCBEhFpGRFiERCCb+lR0qHcjIyMgdKh39qB0VMjIVHQAEAAAAAASwBJ0ABwAUACQALgAAADIWFAYiJjQTMzIWFRQXITY1NDYzASEyFhQGKwEXITcjIiY0NgMhMhYdASE1NDYCDZZqapZqty4iKyf+vCcrI/7NArwVHR0VDYr8SokMFR0dswRMFR37UB0EnWqWamqW/us5Okxra0w6Of5yHSodZGQdKh3+1B0VMjIVHQAEAAAAAASwBRQADwAcACwANgAAATIeARUUBiImNTQ3FzcnNhMzMhYVFBchNjU0NjMBITIWFAYrARchNyMiJjQ2AyEyFh0BITU0NgJYL1szb5xvIpBvoyIfLiIrJ/68Jysj/s0CvBUdHRUNivxKiQwVHR2zBEwVHftQHQUUa4s2Tm9vTj5Rj2+jGv4KOTpMa2tMOjn+ch0qHWRkHSod/tQdFTIyFR0AAAADAAAAAASwBRIAEgAiACwAAAEFFSEUHgMXIS4BNTQ+AjcBITIWFAYrARchNyMiJjQ2AyEyFh0BITU0NgJYASz+1CU/P00T/e48PUJtj0r+ogK8FR0dFQ2K/EqJDBUdHbMETBUd+1AdBLChizlmUT9IGVO9VFShdksE/H4dKh1kZB0qHf7UHRUyMhUdAAIAyAAAA+gFFAAPACkAAAAyFh0BHgEdASE1NDY3NTQDITIWFyMVMxUjFTMVIxUzFAYjISImNRE0NgIvUjsuNv5wNi5kAZA2XBqsyMjIyMh1U/5wU3V1BRQ7KU4aXDYyMjZcGk4p/kc2LmRkZGRkU3V1UwGQU3UAAAMAZP//BEwETAAPAC8AMwAAEyEyFhURFAYjISImNRE0NgMhMhYdARQGIyEXFhQGIi8BIQcGIiY0PwEhIiY9ATQ2BQchJ5YDhBUdHRX8fBUdHQQDtgoPDwr+5eANGiUNWP30Vw0mGg3g/t8KDw8BqmQBRGQETB0V/gwVHR0VAfQVHf1EDwoyCg/gDSUbDVhYDRslDeAPCjIKD2RkZAAAAAAEAAAAAASwBEwAGQAjAC0ANwAAEyEyFh0BIzQmKwEiBhUjNCYrASIGFSM1NDYDITIWFREhETQ2ExUUBisBIiY9ASEVFAYrASImPQHIAyBTdWQ7KfopO2Q7KfopO2R1EQPoKTv7UDvxHRVkFR0D6B0VZBUdBEx1U8gpOzspKTs7KchTdf4MOyn+1AEsKTv+DDIVHR0VMjIVHR0VMgADAAEAAASpBKwADQARABsAAAkBFhQPASEBJjQ3ATYyCQMDITIWHQEhNTQ2AeACqh8fg/4f/fsgIAEnH1n+rAFWAS/+q6IDIBUd/HwdBI39VR9ZH4MCBh9ZHwEoH/5u/qoBMAFV/BsdFTIyFR0AAAAAAgCPAAAEIQSwABcALwAAAQMuASMhIgYHAwYWMyEVFBYyNj0BMzI2AyE1NDY7ATU0NjsBETMRMzIWHQEzMhYVBCG9CCcV/nAVJwi9CBMVAnEdKh19FROo/a0dFTIdFTDILxUdMhUdAocB+hMcHBP+BhMclhUdHRWWHP2MMhUdMhUdASz+1B0VMh0VAAAEAAAAAASwBLAADQAQAB8AIgAAASERFAYjIREBNTQ2MyEBIzUBIREUBiMhIiY1ETQ2MyEBIzUDhAEsDwr+if7UDwoBdwEsyP2oASwPCv12Cg8PCgF3ASzIAyD9wQoPAk8BLFQKD/7UyP4M/cEKDw8KA7YKD/7UyAAC/5wAZAUUBEcARgBWAAABMzIeAhcWFxY2NzYnJjc+ARYXFgcOASsBDgEPAQ4BKwEiJj8BBisBIicHDgErASImPwEmLwEuAT0BNDY7ATY3JyY2OwE2BSMiBh0BFBY7ATI2PQE0JgHkw0uOakkMEhEfQwoKGRMKBQ8XDCkCA1Y9Pgc4HCcDIhVkFRgDDDEqwxgpCwMiFWQVGAMaVCyfExwdFXwLLW8QBxXLdAFF+goPDwr6Cg8PBEdBa4pJDgYKISAiJRsQCAYIDCw9P1c3fCbqFB0dFEYOCEAUHR0UnUplNQcmFTIVHVdPXw4TZV8PCjIKDw8KMgoPAAb/nP/mBRQEfgAJACQANAA8AFIAYgAAASU2Fh8BFgYPASUzMhYfASEyFh0BFAYHBQYmJyYjISImPQE0NhcjIgYdARQ7ATI2NTQmJyYEIgYUFjI2NAE3PgEeARceAT8BFxYGDwEGJi8BJjYlBwYfAR4BPwE2Jy4BJy4BAoEBpxMuDiAOAxCL/CtqQ0geZgM3FR0cE/0fFyIJKjr+1D5YWLlQExIqhhALIAsSAYBALS1ALf4PmBIgHhMQHC0aPzANITNQL3wpgigJASlmHyElDR0RPRMFAhQHCxADhPcICxAmDyoNeMgiNtQdFTIVJgeEBBQPQ1g+yD5YrBwVODMQEAtEERzJLUAtLUD+24ITChESEyMgAwWzPUkrRSgJL5cvfRxYGyYrDwkLNRAhFEgJDAQAAAAAAwBkAAAEOQSwAFEAYABvAAABMzIWHQEeARcWDgIPATIeBRUUDgUjFRQGKwEiJj0BIxUUBisBIiY9ASMiJj0BNDY7AREjIiY9ATQ2OwE1NDY7ATIWHQEzNTQ2AxUhMj4CNTc0LgMjARUhMj4CNTc0LgMjAnGWCg9PaAEBIC4uEBEGEjQwOiodFyI2LUAjGg8KlgoPZA8KlgoPrwoPDwpLSwoPDwqvDwqWCg9kD9cBBxwpEwsBAQsTKRz++QFrHCkTCwEBCxMpHASwDwptIW1KLk0tHwYGAw8UKDJOLTtdPCoVCwJLCg8PCktLCg8PCksPCpYKDwJYDwqWCg9LCg8PCktLCg/+1MgVHR0LCgQOIhoW/nDIFR0dCwoEDiIaFgAAAwAEAAIEsASuABcAKQAsAAATITIWFREUBg8BDgEjISImJy4CNRE0NgQiDgQPARchNy4FAyMT1AMMVnokEhIdgVL9xFKCHAgYKHoCIIx9VkcrHQYGnAIwnAIIIClJVSGdwwSuelb+YDO3QkJXd3ZYHFrFMwGgVnqZFyYtLSUMDPPzBQ8sKDEj/sIBBQACAMgAAAOEBRQADwAZAAABMzIWFREUBiMhIiY1ETQ2ARUUBisBIiY9AQHblmesVCn+PilUrAFINhWWFTYFFKxn/gwpVFQpAfRnrPwY4RU2NhXhAAACAMgAAAOEBRQADwAZAAABMxQWMxEUBiMhIiY1ETQ2ARUUBisBIiY9AQHbYLOWVCn+PilUrAFINhWWFTYFFJaz/kIpVFQpAfRnrPwY4RU2NhXhAAACAAAAFAUOBBoAFAAaAAAJASUHFRcVJwc1NzU0Jj4CPwEnCQEFJTUFJQUO/YL+hk5klpZkAQEBBQQvkwKCAVz+ov6iAV4BXgL//uWqPOCWx5SVyJb6BA0GCgYDKEEBG/1ipqaTpaUAAAMAZAH0BLADIAAHAA8AFwAAEjIWFAYiJjQkMhYUBiImNCQyFhQGIiY0vHxYWHxYAeh8WFh8WAHofFhYfFgDIFh8WFh8WFh8WFh8WFh8WFh8AAAAAAMBkAAAArwETAAHAA8AFwAAADIWFAYiJjQSMhYUBiImNBIyFhQGIiY0Aeh8WFh8WFh8WFh8WFh8WFh8WARMWHxYWHz+yFh8WFh8/shYfFhYfAAAAAMAZABkBEwETAAPAB8ALwAAEyEyFh0BFAYjISImPQE0NhMhMhYdARQGIyEiJj0BNDYTITIWHQEUBiMhIiY9ATQ2fQO2Cg8PCvxKCg8PCgO2Cg8PCvxKCg8PCgO2Cg8PCvxKCg8PBEwPCpYKDw8KlgoP/nAPCpYKDw8KlgoP/nAPCpYKDw8KlgoPAAAABAAAAAAEsASwAA8AHwAvADMAAAEhMhYVERQGIyEiJjURNDYFISIGFREUFjMhMjY1ETQmBSEyFhURFAYjISImNRE0NhcVITUBXgH0ory7o/4Mpbm5Asv9qCk7OykCWCk7O/2xAfQVHR0V/gwVHR1HAZAEsLuj/gylubmlAfSlucg7Kf2oKTs7KQJYKTtkHRX+1BUdHRUBLBUdZMjIAAAAAAEAZABkBLAETAA7AAATITIWFAYrARUzMhYUBisBFTMyFhQGKwEVMzIWFAYjISImNDY7ATUjIiY0NjsBNSMiJjQ2OwE1IyImNDaWA+gVHR0VMjIVHR0VMjIVHR0VMjIVHR0V/BgVHR0VMjIVHR0VMjIVHR0VMjIVHR0ETB0qHcgdKh3IHSodyB0qHR0qHcgdKh3IHSodyB0qHQAAAAYBLAAFA+gEowAHAA0AEwAZAB8AKgAAAR4BBgcuATYBMhYVIiYlFAYjNDYBMhYVIiYlFAYjNDYDFRQGIiY9ARYzMgKKVz8/V1c/P/75fLB8sAK8sHyw/cB8sHywArywfLCwHSodKAMRBKNDsrJCQrKy/sCwfLB8fLB8sP7UsHywfHywfLD+05AVHR0VjgQAAAH/tQDIBJQDgQBCAAABNzYXAR4BBw4BKwEyFRQOBCsBIhE0NyYiBxYVECsBIi4DNTQzIyImJyY2NwE2HwEeAQ4BLwEHIScHBi4BNgLpRRkUASoLCAYFGg8IAQQNGyc/KZK4ChRUFQu4jjBJJxkHAgcPGQYGCAsBKhQaTBQVCiMUM7YDe7YsFCMKFgNuEwYS/tkLHw8OEw0dNkY4MhwBIBgXBAQYF/7gKjxTQyMNEw4PHwoBKBIHEwUjKBYGDMHBDAUWKCMAAAAAAgAAAAAEsASwACUAQwAAASM0LgUrAREUFh8BFSE1Mj4DNREjIg4FFSMRIQEjNC4DKwERFBYXMxUjNTI1ESMiDgMVIzUhBLAyCAsZEyYYGcgyGRn+cAQOIhoWyBkYJhMZCwgyA+j9RBkIChgQEWQZDQzIMmQREBgKCBkB9AOEFSAVDggDAfyuFhkBAmRkAQUJFQ4DUgEDCA4VIBUBLP0SDxMKBQH+VwsNATIyGQGpAQUKEw+WAAAAAAMAAAAABEwErgAdACAAMAAAATUiJy4BLwEBIwEGBw4BDwEVITUiJj8BIRcWBiMVARsBARUUBiMhIiY9ATQ2MyEyFgPoGR4OFgUE/t9F/tQSFQkfCwsBETE7EkUBJT0NISf+7IZ5AbEdFfwYFR0dFQPoFR0BLDIgDiIKCwLr/Q4jFQkTBQUyMisusKYiQTIBhwFW/qr942QVHR0VZBUdHQADAAAAAASwBLAADwBHAEoAABMhMhYVERQGIyEiJjURNDYFIyIHAQYHBgcGHQEUFjMhMjY9ATQmIyInJj8BIRcWBwYjIgYdARQWMyEyNj0BNCYnIicmJyMBJhMjEzIETBUdHRX7tBUdHQJGRg0F/tUREhImDAsJAREIDAwINxAKCj8BCjkLEQwYCAwMCAE5CAwLCBEZGQ8B/uAFDsVnBLAdFfu0FR0dFQRMFR1SDP0PIBMSEAUNMggMDAgyCAwXDhmjmR8YEQwIMggMDAgyBwwBGRskAuwM/gUBCAAABAAAAAAEsASwAAMAEwAjACcAAAEhNSEFITIWFREUBiMhIiY1ETQ2KQEyFhURFAYjISImNRE0NhcRIREEsPtQBLD7ggGQFR0dFf5wFR0dAm0BkBUdHRX+cBUdHUcBLARMZMgdFfx8FR0dFQOEFR0dFf5wFR0dFQGQFR1k/tQBLAAEAAAAAASwBLAADwAfACMAJwAAEyEyFhURFAYjISImNRE0NgEhMhYVERQGIyEiJjURNDYXESEREyE1ITIBkBUdHRX+cBUdHQJtAZAVHR0V/nAVHR1HASzI+1AEsASwHRX8fBUdHRUDhBUd/gwdFf5wFR0dFQGQFR1k/tQBLP2oZAAAAAACAAAAZASwA+gAJwArAAATITIWFREzNTQ2MyEyFh0BMxUjFRQGIyEiJj0BIxEUBiMhIiY1ETQ2AREhETIBkBUdZB0VAZAVHWRkHRX+cBUdZB0V/nAVHR0CnwEsA+gdFf6ilhUdHRWWZJYVHR0Vlv6iFR0dFQMgFR3+1P7UASwAAAQAAAAABLAEsAADABMAFwAnAAAzIxEzFyEyFhURFAYjISImNRE0NhcRIREBITIWFREUBiMhIiY1ETQ2ZGRklgGQFR0dFf5wFR0dRwEs/qIDhBUdHRX8fBUdHQSwZB0V/nAVHR0VAZAVHWT+1AEs/gwdFf5wFR0dFQGQFR0AAAAAAgBkAAAETASwACcAKwAAATMyFhURFAYrARUhMhYVERQGIyEiJjURNDYzITUjIiY1ETQ2OwE1MwcRIRECWJYVHR0VlgHCFR0dFfx8FR0dFQFelhUdHRWWZMgBLARMHRX+cBUdZB0V/nAVHR0VAZAVHWQdFQGQFR1kyP7UASwAAAAEAAAAAASwBLAAAwATABcAJwAAISMRMwUhMhYVERQGIyEiJjURNDYXESERASEyFhURFAYjISImNRE0NgSwZGT9dgGQFR0dFf5wFR0dRwEs/K4DhBUdHRX8fBUdHQSwZB0V/nAVHR0VAZAVHWT+1AEs/gwdFf5wFR0dFQGQFR0AAAEBLAAwA28EgAAPAAAJAQYjIiY1ETQ2MzIXARYUA2H+EhcSDhAQDhIXAe4OAjX+EhcbGQPoGRsX/hIOKgAAAAABAUEAMgOEBH4ACwAACQE2FhURFAYnASY0AU8B7h0qKh3+Eg4CewHuHREp/BgpER0B7g4qAAAAAAEAMgFBBH4DhAALAAATITIWBwEGIicBJjZkA+gpER3+Eg4qDv4SHREDhCod/hIODgHuHSoAAAAAAQAyASwEfgNvAAsAAAkBFgYjISImNwE2MgJ7Ae4dESn8GCkRHQHuDioDYf4SHSoqHQHuDgAAAAACAAgAAASwBCgABgAKAAABFQE1LQE1ASE1IQK8/UwBnf5jBKj84AMgAuW2/r3dwcHd+9jIAAAAAAIAAABkBLAEsAALADEAAAEjFTMVIREzNSM1IQEzND4FOwERFAYPARUhNSIuAzURMzIeBRUzESEEsMjI/tTIyAEs+1AyCAsZEyYYGWQyGRkBkAQOIhoWZBkYJhMZCwgy/OADhGRkASxkZP4MFSAVDggDAf3aFhkBAmRkAQUJFQ4CJgEDCA4VIBUBLAAAAgAAAAAETAPoACUAMQAAASM0LgUrAREUFh8BFSE1Mj4DNREjIg4FFSMRIQEjFTMVIREzNSM1IQMgMggLGRMmGBlkMhkZ/nAEDiIaFmQZGCYTGQsIMgMgASzIyP7UyMgBLAK8FSAVDggDAf3aFhkCAWRkAQUJFQ4CJgEDCA4VIBUBLPzgZGQBLGRkAAABAMgAZgNyBEoAEgAAATMyFgcJARYGKwEiJwEmNDcBNgK9oBAKDP4wAdAMChCgDQr+KQcHAdcKBEoWDP4w/jAMFgkB1wgUCAHXCQAAAQE+AGYD6ARKABIAAAEzMhcBFhQHAQYrASImNwkBJjYBU6ANCgHXBwf+KQoNoBAKDAHQ/jAMCgRKCf4pCBQI/ikJFgwB0AHQDBYAAAEAZgDIBEoDcgASAAAAFh0BFAcBBiInASY9ATQ2FwkBBDQWCf4pCBQI/ikJFgwB0AHQA3cKEKANCv4pBwcB1woNoBAKDP4wAdAAAAABAGYBPgRKA+gAEgAACQEWHQEUBicJAQYmPQE0NwE2MgJqAdcJFgz+MP4wDBYJAdcIFAPh/ikKDaAQCgwB0P4wDAoQoA0KAdcHAAAAAgDZ//kEPQSwAAUAOgAAARQGIzQ2BTMyFh8BNjc+Ah4EBgcOBgcGIiYjIgYiJy4DLwEuAT4EHgEXJyY2A+iwfLD+VmQVJgdPBQsiKFAzRyorDwURAQQSFyozTSwNOkkLDkc3EDlfNyYHBw8GDyUqPjdGMR+TDA0EsHywfLDIHBPCAQIGBwcFDx81S21DBxlLR1xKQhEFBQcHGWt0bCQjP2hJNyATBwMGBcASGAAAAAACAMgAFQOEBLAAFgAaAAATITIWFREUBisBEQcGJjURIyImNRE0NhcVITX6AlgVHR0Vlv8TGpYVHR2rASwEsB0V/nAVHf4MsgkQFQKKHRUBkBUdZGRkAAAAAgDIABkETASwAA4AEgAAEyEyFhURBRElIREjETQ2ARU3NfoC7ic9/UQCWP1EZB8BDWQEsFEs/Ft1A7Z9/BgEARc0/V1kFGQAAQAAAAECTW/DBF9fDzz1AB8EsAAAAADQdnOXAAAAANB2c5f/Uf+cBdwFFAAAAAgAAgAAAAAAAAABAAAFFP+FAAAFFP9R/tQF3AABAAAAAAAAAAAAAAAAAAAAowG4ACgAAAAAAZAAAASwAAAEsABkBLAAAASwAAAEsABwAooAAAUUAAACigAABRQAAAGxAAABRQAAANgAAADYAAAAogAAAQQAAABIAAABBAAAAUUAAASwAGQEsAB7BLAAyASwAMgB9AAABLD/8gSwAAAEsAAABLD/8ASwAAAEsAAOBLAACQSwAGQEsP/TBLD/0wSwAAAEsAAABLAAAASwAAAEsAAABLAAJgSwAG4EsAAXBLAAFwSwABcEsABkBLAAGgSwAGQEsAAMBLAAZASwABcEsP+cBLAAZASwABcEsAAXBLAAAASwABcEsAAXBLAAFwSwAGQEsAAABLAAZASwAAAEsAAABLAAAASwAAAEsAAABLAAAASwAAAEsAAABLAAZASwAMgEsAAABLAAAASwADUEsABkBLAAyASw/7UEsAAhBLAAAASwAAAEsAAABLAAAASwAAAEsP+cBLAAAASwAAAEsAAABLAA2wSwABcEsAB1BLAAAASwAAAEsAAABLAACgSwAMgEsAAABLAAnQSwAMgEsADIBLAAyASwAAAEsP/+BLABLASwAGQEsACIBLABOwSwABcEsAAXBLAAFwSwABcEsAAXBLAAFwSwAAAEsAAXBLAAFwSwABcEsAAXBLAAAASwALcEsAC3BLAAAASwAAAEsABJBLAAFwSwAAAEsAAABLAAXQSw/9wEsP/cBLD/nwSwAGQEsAAABLAAAASwAAAEsABkBLD//wSwAAAEsP9RBLAABgSwAAAEsAAABLABRQSwAAEEsAAABLD/nASwAEoEsAAUBLAAAASwAAAEsAAABLD/nASwAGEEsP/9BLAAFgSwABYEsAAWBLAAFgSwABgEsAAABMQAAASwAGQAAAAAAAD/2ABkADkAyAAAAScAZAAZABkAGQAZABkAGQAZAAAAAAAAAAAAAADZAAAAAAAOAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAMAZABkAAAAEAAAAAAAZP+c/5z/nP+c/5z/nP+c/5wACQAJ//L/8gBkAHkAJwBkAGQAAAAAAGT/ogAAAAAAAAAAAAAAAADIAGQAAAABAI8AAP+c/5wAZAAEAMgAyAAAAGQBkABkAAAAZAEs/7UAAAAAAAAAAAAAAAAAAABkAAABLAFBADIAMgAIAAAAAADIAT4AZgBmANkAyADIAAAAKgAqACoAKgCyAOgA6AFOAU4BTgFOAU4BTgFOAU4BTgFOAU4BTgFOAU4BpAIGAiICfgKGAqwC5ANGA24DjAPEBAgEMgRiBKIE3AVcBboGcgb0ByAHYgfKCB4IYgi+CTYJhAm2Cd4KKApMCpQK4gswC4oLygwIDFgNKg1eDbAODg5oDrQPKA+mD+YQEhBUEJAQqhEqEXYRthIKEjgSfBLAExoTdBPQFCoU1BU8FagVzBYEFjYWYBawFv4XUhemGAIYLhhqGJYYsBjgGP4ZKBloGZQZxBnaGe4aNhpoGrga9hteG7QcMhyUHOIdHB1EHWwdlB28HeYeLh52HsAfYh/SIEYgviEyIXYhuCJAIpYiuCMOIyIjOCN6I8Ij4CQCJDAkXiSWJOIlNCVgJbwmFCZ+JuYnUCe8J/goNChwKKwpoCnMKiYqSiqEKworeiwILGgsuizsLRwtiC30LiguZi6iLtgvDi9GL34vsi/4MD4whDDSMRIxYDGuMegyJDJeMpoy3jMiMz4zaDO2NBg0YDSoNNI1LDWeNeg2PjZ8Ntw3GjdON5I31DgQOEI4hjjIOQo5SjmIOcw6HDpsOpo63jugO9w8GDxQPKI8+D0yPew+Oj6MPtQ/KD9uP6o/+kBIQIBAxkECQX5CGEKoQu5DGENCQ3ZDoEPKRBBEYESuRPZFWkW2RgZGdEa0RvZHNkd2R7ZH9kgWSDJITkhqSIZIzEkSSThJXkmESapKAkouSlIAAQAAARcApwARAAAAAAACAAAAAQABAAAAQAAuAAAAAAAAABAAxgABAAAAAAATABIAAAADAAEECQAAAGoAEgADAAEECQABACgAfAADAAEECQACAA4ApAADAAEECQADAEwAsgADAAEECQAEADgA/gADAAEECQAFAHgBNgADAAEECQAGADYBrgADAAEECQAIABYB5AADAAEECQAJABYB+gADAAEECQALACQCEAADAAEECQAMACQCNAADAAEECQATACQCWAADAAEECQDIABYCfAADAAEECQDJADACkgADAAEECdkDABoCwnd3dy5nbHlwaGljb25zLmNvbQBDAG8AcAB5AHIAaQBnAGgAdAAgAKkAIAAyADAAMQA0ACAAYgB5ACAASgBhAG4AIABLAG8AdgBhAHIAaQBrAC4AIABBAGwAbAAgAHIAaQBnAGgAdABzACAAcgBlAHMAZQByAHYAZQBkAC4ARwBMAFkAUABIAEkAQwBPAE4AUwAgAEgAYQBsAGYAbABpAG4AZwBzAFIAZQBnAHUAbABhAHIAMQAuADAAMAA5ADsAVQBLAFcATgA7AEcATABZAFAASABJAEMATwBOAFMASABhAGwAZgBsAGkAbgBnAHMALQBSAGUAZwB1AGwAYQByAEcATABZAFAASABJAEMATwBOAFMAIABIAGEAbABmAGwAaQBuAGcAcwAgAFIAZQBnAHUAbABhAHIAVgBlAHIAcwBpAG8AbgAgADEALgAwADAAOQA7AFAAUwAgADAAMAAxAC4AMAAwADkAOwBoAG8AdABjAG8AbgB2ACAAMQAuADAALgA3ADAAOwBtAGEAawBlAG8AdABmAC4AbABpAGIAMgAuADUALgA1ADgAMwAyADkARwBMAFkAUABIAEkAQwBPAE4AUwBIAGEAbABmAGwAaQBuAGcAcwAtAFIAZQBnAHUAbABhAHIASgBhAG4AIABLAG8AdgBhAHIAaQBrAEoAYQBuACAASwBvAHYAYQByAGkAawB3AHcAdwAuAGcAbAB5AHAAaABpAGMAbwBuAHMALgBjAG8AbQB3AHcAdwAuAGcAbAB5AHAAaABpAGMAbwBuAHMALgBjAG8AbQB3AHcAdwAuAGcAbAB5AHAAaABpAGMAbwBuAHMALgBjAG8AbQBXAGUAYgBmAG8AbgB0ACAAMQAuADAAVwBlAGQAIABPAGMAdAAgADIAOQAgADAANgA6ADMANgA6ADAANwAgADIAMAAxADQARgBvAG4AdAAgAFMAcQB1AGkAcgByAGUAbAAAAAIAAAAAAAD/tQAyAAAAAAAAAAAAAAAAAAAAAAAAAAABFwAAAQIBAwADAA0ADgEEAJYBBQEGAQcBCAEJAQoBCwEMAQ0BDgEPARABEQESARMA7wEUARUBFgEXARgBGQEaARsBHAEdAR4BHwEgASEBIgEjASQBJQEmAScBKAEpASoBKwEsAS0BLgEvATABMQEyATMBNAE1ATYBNwE4ATkBOgE7ATwBPQE+AT8BQAFBAUIBQwFEAUUBRgFHAUgBSQFKAUsBTAFNAU4BTwFQAVEBUgFTAVQBVQFWAVcBWAFZAVoBWwFcAV0BXgFfAWABYQFiAWMBZAFlAWYBZwFoAWkBagFrAWwBbQFuAW8BcAFxAXIBcwF0AXUBdgF3AXgBeQF6AXsBfAF9AX4BfwGAAYEBggGDAYQBhQGGAYcBiAGJAYoBiwGMAY0BjgGPAZABkQGSAZMBlAGVAZYBlwGYAZkBmgGbAZwBnQGeAZ8BoAGhAaIBowGkAaUBpgGnAagBqQGqAasBrAGtAa4BrwGwAbEBsgGzAbQBtQG2AbcBuAG5AboBuwG8Ab0BvgG/AcABwQHCAcMBxAHFAcYBxwHIAckBygHLAcwBzQHOAc8B0AHRAdIB0wHUAdUB1gHXAdgB2QHaAdsB3AHdAd4B3wHgAeEB4gHjAeQB5QHmAecB6AHpAeoB6wHsAe0B7gHvAfAB8QHyAfMB9AH1AfYB9wH4AfkB+gH7AfwB/QH+Af8CAAIBAgICAwIEAgUCBgIHAggCCQIKAgsCDAINAg4CDwIQAhECEgZnbHlwaDEGZ2x5cGgyB3VuaTAwQTAHdW5pMjAwMAd1bmkyMDAxB3VuaTIwMDIHdW5pMjAwMwd1bmkyMDA0B3VuaTIwMDUHdW5pMjAwNgd1bmkyMDA3B3VuaTIwMDgHdW5pMjAwOQd1bmkyMDBBB3VuaTIwMkYHdW5pMjA1RgRFdXJvB3VuaTIwQkQHdW5pMjMxQgd1bmkyNUZDB3VuaTI2MDEHdW5pMjZGQQd1bmkyNzA5B3VuaTI3MEYHdW5pRTAwMQd1bmlFMDAyB3VuaUUwMDMHdW5pRTAwNQd1bmlFMDA2B3VuaUUwMDcHdW5pRTAwOAd1bmlFMDA5B3VuaUUwMTAHdW5pRTAxMQd1bmlFMDEyB3VuaUUwMTMHdW5pRTAxNAd1bmlFMDE1B3VuaUUwMTYHdW5pRTAxNwd1bmlFMDE4B3VuaUUwMTkHdW5pRTAyMAd1bmlFMDIxB3VuaUUwMjIHdW5pRTAyMwd1bmlFMDI0B3VuaUUwMjUHdW5pRTAyNgd1bmlFMDI3B3VuaUUwMjgHdW5pRTAyOQd1bmlFMDMwB3VuaUUwMzEHdW5pRTAzMgd1bmlFMDMzB3VuaUUwMzQHdW5pRTAzNQd1bmlFMDM2B3VuaUUwMzcHdW5pRTAzOAd1bmlFMDM5B3VuaUUwNDAHdW5pRTA0MQd1bmlFMDQyB3VuaUUwNDMHdW5pRTA0NAd1bmlFMDQ1B3VuaUUwNDYHdW5pRTA0Nwd1bmlFMDQ4B3VuaUUwNDkHdW5pRTA1MAd1bmlFMDUxB3VuaUUwNTIHdW5pRTA1Mwd1bmlFMDU0B3VuaUUwNTUHdW5pRTA1Ngd1bmlFMDU3B3VuaUUwNTgHdW5pRTA1OQd1bmlFMDYwB3VuaUUwNjIHdW5pRTA2Mwd1bmlFMDY0B3VuaUUwNjUHdW5pRTA2Ngd1bmlFMDY3B3VuaUUwNjgHdW5pRTA2OQd1bmlFMDcwB3VuaUUwNzEHdW5pRTA3Mgd1bmlFMDczB3VuaUUwNzQHdW5pRTA3NQd1bmlFMDc2B3VuaUUwNzcHdW5pRTA3OAd1bmlFMDc5B3VuaUUwODAHdW5pRTA4MQd1bmlFMDgyB3VuaUUwODMHdW5pRTA4NAd1bmlFMDg1B3VuaUUwODYHdW5pRTA4Nwd1bmlFMDg4B3VuaUUwODkHdW5pRTA5MAd1bmlFMDkxB3VuaUUwOTIHdW5pRTA5Mwd1bmlFMDk0B3VuaUUwOTUHdW5pRTA5Ngd1bmlFMDk3B3VuaUUxMDEHdW5pRTEwMgd1bmlFMTAzB3VuaUUxMDQHdW5pRTEwNQd1bmlFMTA2B3VuaUUxMDcHdW5pRTEwOAd1bmlFMTA5B3VuaUUxMTAHdW5pRTExMQd1bmlFMTEyB3VuaUUxMTMHdW5pRTExNAd1bmlFMTE1B3VuaUUxMTYHdW5pRTExNwd1bmlFMTE4B3VuaUUxMTkHdW5pRTEyMAd1bmlFMTIxB3VuaUUxMjIHdW5pRTEyMwd1bmlFMTI0B3VuaUUxMjUHdW5pRTEyNgd1bmlFMTI3B3VuaUUxMjgHdW5pRTEyOQd1bmlFMTMwB3VuaUUxMzEHdW5pRTEzMgd1bmlFMTMzB3VuaUUxMzQHdW5pRTEzNQd1bmlFMTM2B3VuaUUxMzcHdW5pRTEzOAd1bmlFMTM5B3VuaUUxNDAHdW5pRTE0MQd1bmlFMTQyB3VuaUUxNDMHdW5pRTE0NAd1bmlFMTQ1B3VuaUUxNDYHdW5pRTE0OAd1bmlFMTQ5B3VuaUUxNTAHdW5pRTE1MQd1bmlFMTUyB3VuaUUxNTMHdW5pRTE1NAd1bmlFMTU1B3VuaUUxNTYHdW5pRTE1Nwd1bmlFMTU4B3VuaUUxNTkHdW5pRTE2MAd1bmlFMTYxB3VuaUUxNjIHdW5pRTE2Mwd1bmlFMTY0B3VuaUUxNjUHdW5pRTE2Ngd1bmlFMTY3B3VuaUUxNjgHdW5pRTE2OQd1bmlFMTcwB3VuaUUxNzEHdW5pRTE3Mgd1bmlFMTczB3VuaUUxNzQHdW5pRTE3NQd1bmlFMTc2B3VuaUUxNzcHdW5pRTE3OAd1bmlFMTc5B3VuaUUxODAHdW5pRTE4MQd1bmlFMTgyB3VuaUUxODMHdW5pRTE4NAd1bmlFMTg1B3VuaUUxODYHdW5pRTE4Nwd1bmlFMTg4B3VuaUUxODkHdW5pRTE5MAd1bmlFMTkxB3VuaUUxOTIHdW5pRTE5Mwd1bmlFMTk0B3VuaUUxOTUHdW5pRTE5Nwd1bmlFMTk4B3VuaUUxOTkHdW5pRTIwMAd1bmlFMjAxB3VuaUUyMDIHdW5pRTIwMwd1bmlFMjA0B3VuaUUyMDUHdW5pRTIwNgd1bmlFMjA5B3VuaUUyMTAHdW5pRTIxMQd1bmlFMjEyB3VuaUUyMTMHdW5pRTIxNAd1bmlFMjE1B3VuaUUyMTYHdW5pRTIxOAd1bmlFMjE5B3VuaUUyMjEHdW5pRTIyMwd1bmlFMjI0B3VuaUUyMjUHdW5pRTIyNgd1bmlFMjI3B3VuaUUyMzAHdW5pRTIzMQd1bmlFMjMyB3VuaUUyMzMHdW5pRTIzNAd1bmlFMjM1B3VuaUUyMzYHdW5pRTIzNwd1bmlFMjM4B3VuaUUyMzkHdW5pRTI0MAd1bmlFMjQxB3VuaUUyNDIHdW5pRTI0Mwd1bmlFMjQ0B3VuaUUyNDUHdW5pRTI0Ngd1bmlFMjQ3B3VuaUUyNDgHdW5pRTI0OQd1bmlFMjUwB3VuaUUyNTEHdW5pRTI1Mgd1bmlFMjUzB3VuaUUyNTQHdW5pRTI1NQd1bmlFMjU2B3VuaUUyNTcHdW5pRTI1OAd1bmlFMjU5B3VuaUUyNjAHdW5pRjhGRgZ1MUY1MTEGdTFGNkFBAAAAAAFUUMMXAAA=) format('truetype'),url(data:image/svg+xml;base64,<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd" >
<svg xmlns="http://www.w3.org/2000/svg">
<metadata></metadata>
<defs>
<font id="glyphicons_halflingsregular" horiz-adv-x="1200" >
<font-face units-per-em="1200" ascent="960" descent="-240" />
<missing-glyph horiz-adv-x="500" />
<glyph horiz-adv-x="0" />
<glyph horiz-adv-x="400" />
<glyph unicode=" " />
<glyph unicode="*" d="M600 1100q15 0 34 -1.5t30 -3.5l11 -1q10 -2 17.5 -10.5t7.5 -18.5v-224l158 158q7 7 18 8t19 -6l106 -106q7 -8 6 -19t-8 -18l-158 -158h224q10 0 18.5 -7.5t10.5 -17.5q6 -41 6 -75q0 -15 -1.5 -34t-3.5 -30l-1 -11q-2 -10 -10.5 -17.5t-18.5 -7.5h-224l158 -158 q7 -7 8 -18t-6 -19l-106 -106q-8 -7 -19 -6t-18 8l-158 158v-224q0 -10 -7.5 -18.5t-17.5 -10.5q-41 -6 -75 -6q-15 0 -34 1.5t-30 3.5l-11 1q-10 2 -17.5 10.5t-7.5 18.5v224l-158 -158q-7 -7 -18 -8t-19 6l-106 106q-7 8 -6 19t8 18l158 158h-224q-10 0 -18.5 7.5 t-10.5 17.5q-6 41 -6 75q0 15 1.5 34t3.5 30l1 11q2 10 10.5 17.5t18.5 7.5h224l-158 158q-7 7 -8 18t6 19l106 106q8 7 19 6t18 -8l158 -158v224q0 10 7.5 18.5t17.5 10.5q41 6 75 6z" />
<glyph unicode="+" d="M450 1100h200q21 0 35.5 -14.5t14.5 -35.5v-350h350q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-350v-350q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v350h-350q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5 h350v350q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xa0;" />
<glyph unicode="&#xa5;" d="M825 1100h250q10 0 12.5 -5t-5.5 -13l-364 -364q-6 -6 -11 -18h268q10 0 13 -6t-3 -14l-120 -160q-6 -8 -18 -14t-22 -6h-125v-100h275q10 0 13 -6t-3 -14l-120 -160q-6 -8 -18 -14t-22 -6h-125v-174q0 -11 -7.5 -18.5t-18.5 -7.5h-148q-11 0 -18.5 7.5t-7.5 18.5v174 h-275q-10 0 -13 6t3 14l120 160q6 8 18 14t22 6h125v100h-275q-10 0 -13 6t3 14l120 160q6 8 18 14t22 6h118q-5 12 -11 18l-364 364q-8 8 -5.5 13t12.5 5h250q25 0 43 -18l164 -164q8 -8 18 -8t18 8l164 164q18 18 43 18z" />
<glyph unicode="&#x2000;" horiz-adv-x="650" />
<glyph unicode="&#x2001;" horiz-adv-x="1300" />
<glyph unicode="&#x2002;" horiz-adv-x="650" />
<glyph unicode="&#x2003;" horiz-adv-x="1300" />
<glyph unicode="&#x2004;" horiz-adv-x="433" />
<glyph unicode="&#x2005;" horiz-adv-x="325" />
<glyph unicode="&#x2006;" horiz-adv-x="216" />
<glyph unicode="&#x2007;" horiz-adv-x="216" />
<glyph unicode="&#x2008;" horiz-adv-x="162" />
<glyph unicode="&#x2009;" horiz-adv-x="260" />
<glyph unicode="&#x200a;" horiz-adv-x="72" />
<glyph unicode="&#x202f;" horiz-adv-x="260" />
<glyph unicode="&#x205f;" horiz-adv-x="325" />
<glyph unicode="&#x20ac;" d="M744 1198q242 0 354 -189q60 -104 66 -209h-181q0 45 -17.5 82.5t-43.5 61.5t-58 40.5t-60.5 24t-51.5 7.5q-19 0 -40.5 -5.5t-49.5 -20.5t-53 -38t-49 -62.5t-39 -89.5h379l-100 -100h-300q-6 -50 -6 -100h406l-100 -100h-300q9 -74 33 -132t52.5 -91t61.5 -54.5t59 -29 t47 -7.5q22 0 50.5 7.5t60.5 24.5t58 41t43.5 61t17.5 80h174q-30 -171 -128 -278q-107 -117 -274 -117q-206 0 -324 158q-36 48 -69 133t-45 204h-217l100 100h112q1 47 6 100h-218l100 100h134q20 87 51 153.5t62 103.5q117 141 297 141z" />
<glyph unicode="&#x20bd;" d="M428 1200h350q67 0 120 -13t86 -31t57 -49.5t35 -56.5t17 -64.5t6.5 -60.5t0.5 -57v-16.5v-16.5q0 -36 -0.5 -57t-6.5 -61t-17 -65t-35 -57t-57 -50.5t-86 -31.5t-120 -13h-178l-2 -100h288q10 0 13 -6t-3 -14l-120 -160q-6 -8 -18 -14t-22 -6h-138v-175q0 -11 -5.5 -18 t-15.5 -7h-149q-10 0 -17.5 7.5t-7.5 17.5v175h-267q-10 0 -13 6t3 14l120 160q6 8 18 14t22 6h117v100h-267q-10 0 -13 6t3 14l120 160q6 8 18 14t22 6h117v475q0 10 7.5 17.5t17.5 7.5zM600 1000v-300h203q64 0 86.5 33t22.5 119q0 84 -22.5 116t-86.5 32h-203z" />
<glyph unicode="&#x2212;" d="M250 700h800q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#x231b;" d="M1000 1200v-150q0 -21 -14.5 -35.5t-35.5 -14.5h-50v-100q0 -91 -49.5 -165.5t-130.5 -109.5q81 -35 130.5 -109.5t49.5 -165.5v-150h50q21 0 35.5 -14.5t14.5 -35.5v-150h-800v150q0 21 14.5 35.5t35.5 14.5h50v150q0 91 49.5 165.5t130.5 109.5q-81 35 -130.5 109.5 t-49.5 165.5v100h-50q-21 0 -35.5 14.5t-14.5 35.5v150h800zM400 1000v-100q0 -60 32.5 -109.5t87.5 -73.5q28 -12 44 -37t16 -55t-16 -55t-44 -37q-55 -24 -87.5 -73.5t-32.5 -109.5v-150h400v150q0 60 -32.5 109.5t-87.5 73.5q-28 12 -44 37t-16 55t16 55t44 37 q55 24 87.5 73.5t32.5 109.5v100h-400z" />
<glyph unicode="&#x25fc;" horiz-adv-x="500" d="M0 0z" />
<glyph unicode="&#x2601;" d="M503 1089q110 0 200.5 -59.5t134.5 -156.5q44 14 90 14q120 0 205 -86.5t85 -206.5q0 -121 -85 -207.5t-205 -86.5h-750q-79 0 -135.5 57t-56.5 137q0 69 42.5 122.5t108.5 67.5q-2 12 -2 37q0 153 108 260.5t260 107.5z" />
<glyph unicode="&#x26fa;" d="M774 1193.5q16 -9.5 20.5 -27t-5.5 -33.5l-136 -187l467 -746h30q20 0 35 -18.5t15 -39.5v-42h-1200v42q0 21 15 39.5t35 18.5h30l468 746l-135 183q-10 16 -5.5 34t20.5 28t34 5.5t28 -20.5l111 -148l112 150q9 16 27 20.5t34 -5zM600 200h377l-182 112l-195 534v-646z " />
<glyph unicode="&#x2709;" d="M25 1100h1150q10 0 12.5 -5t-5.5 -13l-564 -567q-8 -8 -18 -8t-18 8l-564 567q-8 8 -5.5 13t12.5 5zM18 882l264 -264q8 -8 8 -18t-8 -18l-264 -264q-8 -8 -13 -5.5t-5 12.5v550q0 10 5 12.5t13 -5.5zM918 618l264 264q8 8 13 5.5t5 -12.5v-550q0 -10 -5 -12.5t-13 5.5 l-264 264q-8 8 -8 18t8 18zM818 482l364 -364q8 -8 5.5 -13t-12.5 -5h-1150q-10 0 -12.5 5t5.5 13l364 364q8 8 18 8t18 -8l164 -164q8 -8 18 -8t18 8l164 164q8 8 18 8t18 -8z" />
<glyph unicode="&#x270f;" d="M1011 1210q19 0 33 -13l153 -153q13 -14 13 -33t-13 -33l-99 -92l-214 214l95 96q13 14 32 14zM1013 800l-615 -614l-214 214l614 614zM317 96l-333 -112l110 335z" />
<glyph unicode="&#xe001;" d="M700 650v-550h250q21 0 35.5 -14.5t14.5 -35.5v-50h-800v50q0 21 14.5 35.5t35.5 14.5h250v550l-500 550h1200z" />
<glyph unicode="&#xe002;" d="M368 1017l645 163q39 15 63 0t24 -49v-831q0 -55 -41.5 -95.5t-111.5 -63.5q-79 -25 -147 -4.5t-86 75t25.5 111.5t122.5 82q72 24 138 8v521l-600 -155v-606q0 -42 -44 -90t-109 -69q-79 -26 -147 -5.5t-86 75.5t25.5 111.5t122.5 82.5q72 24 138 7v639q0 38 14.5 59 t53.5 34z" />
<glyph unicode="&#xe003;" d="M500 1191q100 0 191 -39t156.5 -104.5t104.5 -156.5t39 -191l-1 -2l1 -5q0 -141 -78 -262l275 -274q23 -26 22.5 -44.5t-22.5 -42.5l-59 -58q-26 -20 -46.5 -20t-39.5 20l-275 274q-119 -77 -261 -77l-5 1l-2 -1q-100 0 -191 39t-156.5 104.5t-104.5 156.5t-39 191 t39 191t104.5 156.5t156.5 104.5t191 39zM500 1022q-88 0 -162 -43t-117 -117t-43 -162t43 -162t117 -117t162 -43t162 43t117 117t43 162t-43 162t-117 117t-162 43z" />
<glyph unicode="&#xe005;" d="M649 949q48 68 109.5 104t121.5 38.5t118.5 -20t102.5 -64t71 -100.5t27 -123q0 -57 -33.5 -117.5t-94 -124.5t-126.5 -127.5t-150 -152.5t-146 -174q-62 85 -145.5 174t-150 152.5t-126.5 127.5t-93.5 124.5t-33.5 117.5q0 64 28 123t73 100.5t104 64t119 20 t120.5 -38.5t104.5 -104z" />
<glyph unicode="&#xe006;" d="M407 800l131 353q7 19 17.5 19t17.5 -19l129 -353h421q21 0 24 -8.5t-14 -20.5l-342 -249l130 -401q7 -20 -0.5 -25.5t-24.5 6.5l-343 246l-342 -247q-17 -12 -24.5 -6.5t-0.5 25.5l130 400l-347 251q-17 12 -14 20.5t23 8.5h429z" />
<glyph unicode="&#xe007;" d="M407 800l131 353q7 19 17.5 19t17.5 -19l129 -353h421q21 0 24 -8.5t-14 -20.5l-342 -249l130 -401q7 -20 -0.5 -25.5t-24.5 6.5l-343 246l-342 -247q-17 -12 -24.5 -6.5t-0.5 25.5l130 400l-347 251q-17 12 -14 20.5t23 8.5h429zM477 700h-240l197 -142l-74 -226 l193 139l195 -140l-74 229l192 140h-234l-78 211z" />
<glyph unicode="&#xe008;" d="M600 1200q124 0 212 -88t88 -212v-250q0 -46 -31 -98t-69 -52v-75q0 -10 6 -21.5t15 -17.5l358 -230q9 -5 15 -16.5t6 -21.5v-93q0 -10 -7.5 -17.5t-17.5 -7.5h-1150q-10 0 -17.5 7.5t-7.5 17.5v93q0 10 6 21.5t15 16.5l358 230q9 6 15 17.5t6 21.5v75q-38 0 -69 52 t-31 98v250q0 124 88 212t212 88z" />
<glyph unicode="&#xe009;" d="M25 1100h1150q10 0 17.5 -7.5t7.5 -17.5v-1050q0 -10 -7.5 -17.5t-17.5 -7.5h-1150q-10 0 -17.5 7.5t-7.5 17.5v1050q0 10 7.5 17.5t17.5 7.5zM100 1000v-100h100v100h-100zM875 1000h-550q-10 0 -17.5 -7.5t-7.5 -17.5v-350q0 -10 7.5 -17.5t17.5 -7.5h550 q10 0 17.5 7.5t7.5 17.5v350q0 10 -7.5 17.5t-17.5 7.5zM1000 1000v-100h100v100h-100zM100 800v-100h100v100h-100zM1000 800v-100h100v100h-100zM100 600v-100h100v100h-100zM1000 600v-100h100v100h-100zM875 500h-550q-10 0 -17.5 -7.5t-7.5 -17.5v-350q0 -10 7.5 -17.5 t17.5 -7.5h550q10 0 17.5 7.5t7.5 17.5v350q0 10 -7.5 17.5t-17.5 7.5zM100 400v-100h100v100h-100zM1000 400v-100h100v100h-100zM100 200v-100h100v100h-100zM1000 200v-100h100v100h-100z" />
<glyph unicode="&#xe010;" d="M50 1100h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM650 1100h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v400 q0 21 14.5 35.5t35.5 14.5zM50 500h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM650 500h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400 q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe011;" d="M50 1100h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM450 1100h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200 q0 21 14.5 35.5t35.5 14.5zM850 1100h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM50 700h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200 q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM450 700h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM850 700h200q21 0 35.5 -14.5t14.5 -35.5v-200 q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM50 300h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM450 300h200 q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM850 300h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5 t35.5 14.5z" />
<glyph unicode="&#xe012;" d="M50 1100h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM450 1100h700q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-700q-21 0 -35.5 14.5t-14.5 35.5v200 q0 21 14.5 35.5t35.5 14.5zM50 700h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM450 700h700q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-700 q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM50 300h200q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5zM450 300h700q21 0 35.5 -14.5t14.5 -35.5v-200 q0 -21 -14.5 -35.5t-35.5 -14.5h-700q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe013;" d="M465 477l571 571q8 8 18 8t17 -8l177 -177q8 -7 8 -17t-8 -18l-783 -784q-7 -8 -17.5 -8t-17.5 8l-384 384q-8 8 -8 18t8 17l177 177q7 8 17 8t18 -8l171 -171q7 -7 18 -7t18 7z" />
<glyph unicode="&#xe014;" d="M904 1083l178 -179q8 -8 8 -18.5t-8 -17.5l-267 -268l267 -268q8 -7 8 -17.5t-8 -18.5l-178 -178q-8 -8 -18.5 -8t-17.5 8l-268 267l-268 -267q-7 -8 -17.5 -8t-18.5 8l-178 178q-8 8 -8 18.5t8 17.5l267 268l-267 268q-8 7 -8 17.5t8 18.5l178 178q8 8 18.5 8t17.5 -8 l268 -267l268 268q7 7 17.5 7t18.5 -7z" />
<glyph unicode="&#xe015;" d="M507 1177q98 0 187.5 -38.5t154.5 -103.5t103.5 -154.5t38.5 -187.5q0 -141 -78 -262l300 -299q8 -8 8 -18.5t-8 -18.5l-109 -108q-7 -8 -17.5 -8t-18.5 8l-300 299q-119 -77 -261 -77q-98 0 -188 38.5t-154.5 103t-103 154.5t-38.5 188t38.5 187.5t103 154.5 t154.5 103.5t188 38.5zM506.5 1023q-89.5 0 -165.5 -44t-120 -120.5t-44 -166t44 -165.5t120 -120t165.5 -44t166 44t120.5 120t44 165.5t-44 166t-120.5 120.5t-166 44zM425 900h150q10 0 17.5 -7.5t7.5 -17.5v-75h75q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5 t-17.5 -7.5h-75v-75q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v75h-75q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5h75v75q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe016;" d="M507 1177q98 0 187.5 -38.5t154.5 -103.5t103.5 -154.5t38.5 -187.5q0 -141 -78 -262l300 -299q8 -8 8 -18.5t-8 -18.5l-109 -108q-7 -8 -17.5 -8t-18.5 8l-300 299q-119 -77 -261 -77q-98 0 -188 38.5t-154.5 103t-103 154.5t-38.5 188t38.5 187.5t103 154.5 t154.5 103.5t188 38.5zM506.5 1023q-89.5 0 -165.5 -44t-120 -120.5t-44 -166t44 -165.5t120 -120t165.5 -44t166 44t120.5 120t44 165.5t-44 166t-120.5 120.5t-166 44zM325 800h350q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-350q-10 0 -17.5 7.5 t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe017;" d="M550 1200h100q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM800 975v166q167 -62 272 -209.5t105 -331.5q0 -117 -45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5 t-184.5 123t-123 184.5t-45.5 224q0 184 105 331.5t272 209.5v-166q-103 -55 -165 -155t-62 -220q0 -116 57 -214.5t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5q0 120 -62 220t-165 155z" />
<glyph unicode="&#xe018;" d="M1025 1200h150q10 0 17.5 -7.5t7.5 -17.5v-1150q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v1150q0 10 7.5 17.5t17.5 7.5zM725 800h150q10 0 17.5 -7.5t7.5 -17.5v-750q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v750 q0 10 7.5 17.5t17.5 7.5zM425 500h150q10 0 17.5 -7.5t7.5 -17.5v-450q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v450q0 10 7.5 17.5t17.5 7.5zM125 300h150q10 0 17.5 -7.5t7.5 -17.5v-250q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5 v250q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe019;" d="M600 1174q33 0 74 -5l38 -152l5 -1q49 -14 94 -39l5 -2l134 80q61 -48 104 -105l-80 -134l3 -5q25 -44 39 -93l1 -6l152 -38q5 -43 5 -73q0 -34 -5 -74l-152 -38l-1 -6q-15 -49 -39 -93l-3 -5l80 -134q-48 -61 -104 -105l-134 81l-5 -3q-44 -25 -94 -39l-5 -2l-38 -151 q-43 -5 -74 -5q-33 0 -74 5l-38 151l-5 2q-49 14 -94 39l-5 3l-134 -81q-60 48 -104 105l80 134l-3 5q-25 45 -38 93l-2 6l-151 38q-6 42 -6 74q0 33 6 73l151 38l2 6q13 48 38 93l3 5l-80 134q47 61 105 105l133 -80l5 2q45 25 94 39l5 1l38 152q43 5 74 5zM600 815 q-89 0 -152 -63t-63 -151.5t63 -151.5t152 -63t152 63t63 151.5t-63 151.5t-152 63z" />
<glyph unicode="&#xe020;" d="M500 1300h300q41 0 70.5 -29.5t29.5 -70.5v-100h275q10 0 17.5 -7.5t7.5 -17.5v-75h-1100v75q0 10 7.5 17.5t17.5 7.5h275v100q0 41 29.5 70.5t70.5 29.5zM500 1200v-100h300v100h-300zM1100 900v-800q0 -41 -29.5 -70.5t-70.5 -29.5h-700q-41 0 -70.5 29.5t-29.5 70.5 v800h900zM300 800v-700h100v700h-100zM500 800v-700h100v700h-100zM700 800v-700h100v700h-100zM900 800v-700h100v700h-100z" />
<glyph unicode="&#xe021;" d="M18 618l620 608q8 7 18.5 7t17.5 -7l608 -608q8 -8 5.5 -13t-12.5 -5h-175v-575q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v375h-300v-375q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v575h-175q-10 0 -12.5 5t5.5 13z" />
<glyph unicode="&#xe022;" d="M600 1200v-400q0 -41 29.5 -70.5t70.5 -29.5h300v-650q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v1100q0 21 14.5 35.5t35.5 14.5h450zM1000 800h-250q-21 0 -35.5 14.5t-14.5 35.5v250z" />
<glyph unicode="&#xe023;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5t-57 214.5t-155.5 155.5t-214.5 57zM525 900h50q10 0 17.5 -7.5t7.5 -17.5v-275h175q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v350q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe024;" d="M1300 0h-538l-41 400h-242l-41 -400h-538l431 1200h209l-21 -300h162l-20 300h208zM515 800l-27 -300h224l-27 300h-170z" />
<glyph unicode="&#xe025;" d="M550 1200h200q21 0 35.5 -14.5t14.5 -35.5v-450h191q20 0 25.5 -11.5t-7.5 -27.5l-327 -400q-13 -16 -32 -16t-32 16l-327 400q-13 16 -7.5 27.5t25.5 11.5h191v450q0 21 14.5 35.5t35.5 14.5zM1125 400h50q10 0 17.5 -7.5t7.5 -17.5v-350q0 -10 -7.5 -17.5t-17.5 -7.5 h-1050q-10 0 -17.5 7.5t-7.5 17.5v350q0 10 7.5 17.5t17.5 7.5h50q10 0 17.5 -7.5t7.5 -17.5v-175h900v175q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe026;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5t-57 214.5t-155.5 155.5t-214.5 57zM525 900h150q10 0 17.5 -7.5t7.5 -17.5v-275h137q21 0 26 -11.5t-8 -27.5l-223 -275q-13 -16 -32 -16t-32 16l-223 275q-13 16 -8 27.5t26 11.5h137v275q0 10 7.5 17.5t17.5 7.5z " />
<glyph unicode="&#xe027;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5t-57 214.5t-155.5 155.5t-214.5 57zM632 914l223 -275q13 -16 8 -27.5t-26 -11.5h-137v-275q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v275h-137q-21 0 -26 11.5t8 27.5l223 275q13 16 32 16 t32 -16z" />
<glyph unicode="&#xe028;" d="M225 1200h750q10 0 19.5 -7t12.5 -17l186 -652q7 -24 7 -49v-425q0 -12 -4 -27t-9 -17q-12 -6 -37 -6h-1100q-12 0 -27 4t-17 8q-6 13 -6 38l1 425q0 25 7 49l185 652q3 10 12.5 17t19.5 7zM878 1000h-556q-10 0 -19 -7t-11 -18l-87 -450q-2 -11 4 -18t16 -7h150 q10 0 19.5 -7t11.5 -17l38 -152q2 -10 11.5 -17t19.5 -7h250q10 0 19.5 7t11.5 17l38 152q2 10 11.5 17t19.5 7h150q10 0 16 7t4 18l-87 450q-2 11 -11 18t-19 7z" />
<glyph unicode="&#xe029;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5t-57 214.5t-155.5 155.5t-214.5 57zM540 820l253 -190q17 -12 17 -30t-17 -30l-253 -190q-16 -12 -28 -6.5t-12 26.5v400q0 21 12 26.5t28 -6.5z" />
<glyph unicode="&#xe030;" d="M947 1060l135 135q7 7 12.5 5t5.5 -13v-362q0 -10 -7.5 -17.5t-17.5 -7.5h-362q-11 0 -13 5.5t5 12.5l133 133q-109 76 -238 76q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5h150q0 -117 -45.5 -224 t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5q192 0 347 -117z" />
<glyph unicode="&#xe031;" d="M947 1060l135 135q7 7 12.5 5t5.5 -13v-361q0 -11 -7.5 -18.5t-18.5 -7.5h-361q-11 0 -13 5.5t5 12.5l134 134q-110 75 -239 75q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5h-150q0 117 45.5 224t123 184.5t184.5 123t224 45.5q192 0 347 -117zM1027 600h150 q0 -117 -45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5q-192 0 -348 118l-134 -134q-7 -8 -12.5 -5.5t-5.5 12.5v360q0 11 7.5 18.5t18.5 7.5h360q10 0 12.5 -5.5t-5.5 -12.5l-133 -133q110 -76 240 -76q116 0 214.5 57t155.5 155.5t57 214.5z" />
<glyph unicode="&#xe032;" d="M125 1200h1050q10 0 17.5 -7.5t7.5 -17.5v-1150q0 -10 -7.5 -17.5t-17.5 -7.5h-1050q-10 0 -17.5 7.5t-7.5 17.5v1150q0 10 7.5 17.5t17.5 7.5zM1075 1000h-850q-10 0 -17.5 -7.5t-7.5 -17.5v-850q0 -10 7.5 -17.5t17.5 -7.5h850q10 0 17.5 7.5t7.5 17.5v850 q0 10 -7.5 17.5t-17.5 7.5zM325 900h50q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-50q-10 0 -17.5 7.5t-7.5 17.5v50q0 10 7.5 17.5t17.5 7.5zM525 900h450q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-450q-10 0 -17.5 7.5t-7.5 17.5v50 q0 10 7.5 17.5t17.5 7.5zM325 700h50q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-50q-10 0 -17.5 7.5t-7.5 17.5v50q0 10 7.5 17.5t17.5 7.5zM525 700h450q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-450q-10 0 -17.5 7.5t-7.5 17.5v50 q0 10 7.5 17.5t17.5 7.5zM325 500h50q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-50q-10 0 -17.5 7.5t-7.5 17.5v50q0 10 7.5 17.5t17.5 7.5zM525 500h450q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-450q-10 0 -17.5 7.5t-7.5 17.5v50 q0 10 7.5 17.5t17.5 7.5zM325 300h50q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-50q-10 0 -17.5 7.5t-7.5 17.5v50q0 10 7.5 17.5t17.5 7.5zM525 300h450q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-450q-10 0 -17.5 7.5t-7.5 17.5v50 q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe033;" d="M900 800v200q0 83 -58.5 141.5t-141.5 58.5h-300q-82 0 -141 -59t-59 -141v-200h-100q-41 0 -70.5 -29.5t-29.5 -70.5v-600q0 -41 29.5 -70.5t70.5 -29.5h900q41 0 70.5 29.5t29.5 70.5v600q0 41 -29.5 70.5t-70.5 29.5h-100zM400 800v150q0 21 15 35.5t35 14.5h200 q20 0 35 -14.5t15 -35.5v-150h-300z" />
<glyph unicode="&#xe034;" d="M125 1100h50q10 0 17.5 -7.5t7.5 -17.5v-1075h-100v1075q0 10 7.5 17.5t17.5 7.5zM1075 1052q4 0 9 -2q16 -6 16 -23v-421q0 -6 -3 -12q-33 -59 -66.5 -99t-65.5 -58t-56.5 -24.5t-52.5 -6.5q-26 0 -57.5 6.5t-52.5 13.5t-60 21q-41 15 -63 22.5t-57.5 15t-65.5 7.5 q-85 0 -160 -57q-7 -5 -15 -5q-6 0 -11 3q-14 7 -14 22v438q22 55 82 98.5t119 46.5q23 2 43 0.5t43 -7t32.5 -8.5t38 -13t32.5 -11q41 -14 63.5 -21t57 -14t63.5 -7q103 0 183 87q7 8 18 8z" />
<glyph unicode="&#xe035;" d="M600 1175q116 0 227 -49.5t192.5 -131t131 -192.5t49.5 -227v-300q0 -10 -7.5 -17.5t-17.5 -7.5h-50q-10 0 -17.5 7.5t-7.5 17.5v300q0 127 -70.5 231.5t-184.5 161.5t-245 57t-245 -57t-184.5 -161.5t-70.5 -231.5v-300q0 -10 -7.5 -17.5t-17.5 -7.5h-50 q-10 0 -17.5 7.5t-7.5 17.5v300q0 116 49.5 227t131 192.5t192.5 131t227 49.5zM220 500h160q8 0 14 -6t6 -14v-460q0 -8 -6 -14t-14 -6h-160q-8 0 -14 6t-6 14v460q0 8 6 14t14 6zM820 500h160q8 0 14 -6t6 -14v-460q0 -8 -6 -14t-14 -6h-160q-8 0 -14 6t-6 14v460 q0 8 6 14t14 6z" />
<glyph unicode="&#xe036;" d="M321 814l258 172q9 6 15 2.5t6 -13.5v-750q0 -10 -6 -13.5t-15 2.5l-258 172q-21 14 -46 14h-250q-10 0 -17.5 7.5t-7.5 17.5v350q0 10 7.5 17.5t17.5 7.5h250q25 0 46 14zM900 668l120 120q7 7 17 7t17 -7l34 -34q7 -7 7 -17t-7 -17l-120 -120l120 -120q7 -7 7 -17 t-7 -17l-34 -34q-7 -7 -17 -7t-17 7l-120 119l-120 -119q-7 -7 -17 -7t-17 7l-34 34q-7 7 -7 17t7 17l119 120l-119 120q-7 7 -7 17t7 17l34 34q7 8 17 8t17 -8z" />
<glyph unicode="&#xe037;" d="M321 814l258 172q9 6 15 2.5t6 -13.5v-750q0 -10 -6 -13.5t-15 2.5l-258 172q-21 14 -46 14h-250q-10 0 -17.5 7.5t-7.5 17.5v350q0 10 7.5 17.5t17.5 7.5h250q25 0 46 14zM766 900h4q10 -1 16 -10q96 -129 96 -290q0 -154 -90 -281q-6 -9 -17 -10l-3 -1q-9 0 -16 6 l-29 23q-7 7 -8.5 16.5t4.5 17.5q72 103 72 229q0 132 -78 238q-6 8 -4.5 18t9.5 17l29 22q7 5 15 5z" />
<glyph unicode="&#xe038;" d="M967 1004h3q11 -1 17 -10q135 -179 135 -396q0 -105 -34 -206.5t-98 -185.5q-7 -9 -17 -10h-3q-9 0 -16 6l-42 34q-8 6 -9 16t5 18q111 150 111 328q0 90 -29.5 176t-84.5 157q-6 9 -5 19t10 16l42 33q7 5 15 5zM321 814l258 172q9 6 15 2.5t6 -13.5v-750q0 -10 -6 -13.5 t-15 2.5l-258 172q-21 14 -46 14h-250q-10 0 -17.5 7.5t-7.5 17.5v350q0 10 7.5 17.5t17.5 7.5h250q25 0 46 14zM766 900h4q10 -1 16 -10q96 -129 96 -290q0 -154 -90 -281q-6 -9 -17 -10l-3 -1q-9 0 -16 6l-29 23q-7 7 -8.5 16.5t4.5 17.5q72 103 72 229q0 132 -78 238 q-6 8 -4.5 18.5t9.5 16.5l29 22q7 5 15 5z" />
<glyph unicode="&#xe039;" d="M500 900h100v-100h-100v-100h-400v-100h-100v600h500v-300zM1200 700h-200v-100h200v-200h-300v300h-200v300h-100v200h600v-500zM100 1100v-300h300v300h-300zM800 1100v-300h300v300h-300zM300 900h-100v100h100v-100zM1000 900h-100v100h100v-100zM300 500h200v-500 h-500v500h200v100h100v-100zM800 300h200v-100h-100v-100h-200v100h-100v100h100v200h-200v100h300v-300zM100 400v-300h300v300h-300zM300 200h-100v100h100v-100zM1200 200h-100v100h100v-100zM700 0h-100v100h100v-100zM1200 0h-300v100h300v-100z" />
<glyph unicode="&#xe040;" d="M100 200h-100v1000h100v-1000zM300 200h-100v1000h100v-1000zM700 200h-200v1000h200v-1000zM900 200h-100v1000h100v-1000zM1200 200h-200v1000h200v-1000zM400 0h-300v100h300v-100zM600 0h-100v91h100v-91zM800 0h-100v91h100v-91zM1100 0h-200v91h200v-91z" />
<glyph unicode="&#xe041;" d="M500 1200l682 -682q8 -8 8 -18t-8 -18l-464 -464q-8 -8 -18 -8t-18 8l-682 682l1 475q0 10 7.5 17.5t17.5 7.5h474zM319.5 1024.5q-29.5 29.5 -71 29.5t-71 -29.5t-29.5 -71.5t29.5 -71.5t71 -29.5t71 29.5t29.5 71.5t-29.5 71.5z" />
<glyph unicode="&#xe042;" d="M500 1200l682 -682q8 -8 8 -18t-8 -18l-464 -464q-8 -8 -18 -8t-18 8l-682 682l1 475q0 10 7.5 17.5t17.5 7.5h474zM800 1200l682 -682q8 -8 8 -18t-8 -18l-464 -464q-8 -8 -18 -8t-18 8l-56 56l424 426l-700 700h150zM319.5 1024.5q-29.5 29.5 -71 29.5t-71 -29.5 t-29.5 -71.5t29.5 -71.5t71 -29.5t71 29.5t29.5 71.5t-29.5 71.5z" />
<glyph unicode="&#xe043;" d="M300 1200h825q75 0 75 -75v-900q0 -25 -18 -43l-64 -64q-8 -8 -13 -5.5t-5 12.5v950q0 10 -7.5 17.5t-17.5 7.5h-700q-25 0 -43 -18l-64 -64q-8 -8 -5.5 -13t12.5 -5h700q10 0 17.5 -7.5t7.5 -17.5v-950q0 -10 -7.5 -17.5t-17.5 -7.5h-850q-10 0 -17.5 7.5t-7.5 17.5v975 q0 25 18 43l139 139q18 18 43 18z" />
<glyph unicode="&#xe044;" d="M250 1200h800q21 0 35.5 -14.5t14.5 -35.5v-1150l-450 444l-450 -445v1151q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe045;" d="M822 1200h-444q-11 0 -19 -7.5t-9 -17.5l-78 -301q-7 -24 7 -45l57 -108q6 -9 17.5 -15t21.5 -6h450q10 0 21.5 6t17.5 15l62 108q14 21 7 45l-83 301q-1 10 -9 17.5t-19 7.5zM1175 800h-150q-10 0 -21 -6.5t-15 -15.5l-78 -156q-4 -9 -15 -15.5t-21 -6.5h-550 q-10 0 -21 6.5t-15 15.5l-78 156q-4 9 -15 15.5t-21 6.5h-150q-10 0 -17.5 -7.5t-7.5 -17.5v-650q0 -10 7.5 -17.5t17.5 -7.5h150q10 0 17.5 7.5t7.5 17.5v150q0 10 7.5 17.5t17.5 7.5h750q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 7.5 -17.5t17.5 -7.5h150q10 0 17.5 7.5 t7.5 17.5v650q0 10 -7.5 17.5t-17.5 7.5zM850 200h-500q-10 0 -19.5 -7t-11.5 -17l-38 -152q-2 -10 3.5 -17t15.5 -7h600q10 0 15.5 7t3.5 17l-38 152q-2 10 -11.5 17t-19.5 7z" />
<glyph unicode="&#xe046;" d="M500 1100h200q56 0 102.5 -20.5t72.5 -50t44 -59t25 -50.5l6 -20h150q41 0 70.5 -29.5t29.5 -70.5v-600q0 -41 -29.5 -70.5t-70.5 -29.5h-1000q-41 0 -70.5 29.5t-29.5 70.5v600q0 41 29.5 70.5t70.5 29.5h150q2 8 6.5 21.5t24 48t45 61t72 48t102.5 21.5zM900 800v-100 h100v100h-100zM600 730q-95 0 -162.5 -67.5t-67.5 -162.5t67.5 -162.5t162.5 -67.5t162.5 67.5t67.5 162.5t-67.5 162.5t-162.5 67.5zM600 603q43 0 73 -30t30 -73t-30 -73t-73 -30t-73 30t-30 73t30 73t73 30z" />
<glyph unicode="&#xe047;" d="M681 1199l385 -998q20 -50 60 -92q18 -19 36.5 -29.5t27.5 -11.5l10 -2v-66h-417v66q53 0 75 43.5t5 88.5l-82 222h-391q-58 -145 -92 -234q-11 -34 -6.5 -57t25.5 -37t46 -20t55 -6v-66h-365v66q56 24 84 52q12 12 25 30.5t20 31.5l7 13l399 1006h93zM416 521h340 l-162 457z" />
<glyph unicode="&#xe048;" d="M753 641q5 -1 14.5 -4.5t36 -15.5t50.5 -26.5t53.5 -40t50.5 -54.5t35.5 -70t14.5 -87q0 -67 -27.5 -125.5t-71.5 -97.5t-98.5 -66.5t-108.5 -40.5t-102 -13h-500v89q41 7 70.5 32.5t29.5 65.5v827q0 24 -0.5 34t-3.5 24t-8.5 19.5t-17 13.5t-28 12.5t-42.5 11.5v71 l471 -1q57 0 115.5 -20.5t108 -57t80.5 -94t31 -124.5q0 -51 -15.5 -96.5t-38 -74.5t-45 -50.5t-38.5 -30.5zM400 700h139q78 0 130.5 48.5t52.5 122.5q0 41 -8.5 70.5t-29.5 55.5t-62.5 39.5t-103.5 13.5h-118v-350zM400 200h216q80 0 121 50.5t41 130.5q0 90 -62.5 154.5 t-156.5 64.5h-159v-400z" />
<glyph unicode="&#xe049;" d="M877 1200l2 -57q-83 -19 -116 -45.5t-40 -66.5l-132 -839q-9 -49 13 -69t96 -26v-97h-500v97q186 16 200 98l173 832q3 17 3 30t-1.5 22.5t-9 17.5t-13.5 12.5t-21.5 10t-26 8.5t-33.5 10q-13 3 -19 5v57h425z" />
<glyph unicode="&#xe050;" d="M1300 900h-50q0 21 -4 37t-9.5 26.5t-18 17.5t-22 11t-28.5 5.5t-31 2t-37 0.5h-200v-850q0 -22 25 -34.5t50 -13.5l25 -2v-100h-400v100q4 0 11 0.5t24 3t30 7t24 15t11 24.5v850h-200q-25 0 -37 -0.5t-31 -2t-28.5 -5.5t-22 -11t-18 -17.5t-9.5 -26.5t-4 -37h-50v300 h1000v-300zM175 1000h-75v-800h75l-125 -167l-125 167h75v800h-75l125 167z" />
<glyph unicode="&#xe051;" d="M1100 900h-50q0 21 -4 37t-9.5 26.5t-18 17.5t-22 11t-28.5 5.5t-31 2t-37 0.5h-200v-650q0 -22 25 -34.5t50 -13.5l25 -2v-100h-400v100q4 0 11 0.5t24 3t30 7t24 15t11 24.5v650h-200q-25 0 -37 -0.5t-31 -2t-28.5 -5.5t-22 -11t-18 -17.5t-9.5 -26.5t-4 -37h-50v300 h1000v-300zM1167 50l-167 -125v75h-800v-75l-167 125l167 125v-75h800v75z" />
<glyph unicode="&#xe052;" d="M50 1100h600q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-600q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 800h1000q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1000q-21 0 -35.5 14.5t-14.5 35.5v100 q0 21 14.5 35.5t35.5 14.5zM50 500h800q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 200h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe053;" d="M250 1100h700q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-700q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 800h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v100 q0 21 14.5 35.5t35.5 14.5zM250 500h700q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-700q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 200h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe054;" d="M500 950v100q0 21 14.5 35.5t35.5 14.5h600q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-600q-21 0 -35.5 14.5t-14.5 35.5zM100 650v100q0 21 14.5 35.5t35.5 14.5h1000q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1000 q-21 0 -35.5 14.5t-14.5 35.5zM300 350v100q0 21 14.5 35.5t35.5 14.5h800q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5zM0 50v100q0 21 14.5 35.5t35.5 14.5h1100q21 0 35.5 -14.5t14.5 -35.5v-100 q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5z" />
<glyph unicode="&#xe055;" d="M50 1100h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 800h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v100 q0 21 14.5 35.5t35.5 14.5zM50 500h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 200h1100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe056;" d="M50 1100h100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM350 1100h800q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v100 q0 21 14.5 35.5t35.5 14.5zM50 800h100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM350 800h800q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-800 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 500h100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM350 500h800q21 0 35.5 -14.5t14.5 -35.5v-100 q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 200h100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM350 200h800 q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe057;" d="M400 0h-100v1100h100v-1100zM550 1100h100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM550 800h500q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-500 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM267 550l-167 -125v75h-200v100h200v75zM550 500h300q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-300q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM550 200h600 q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-600q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe058;" d="M50 1100h100q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM900 0h-100v1100h100v-1100zM50 800h500q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-500 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM1100 600h200v-100h-200v-75l-167 125l167 125v-75zM50 500h300q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-300q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5zM50 200h600 q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-600q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe059;" d="M75 1000h750q31 0 53 -22t22 -53v-650q0 -31 -22 -53t-53 -22h-750q-31 0 -53 22t-22 53v650q0 31 22 53t53 22zM1200 300l-300 300l300 300v-600z" />
<glyph unicode="&#xe060;" d="M44 1100h1112q18 0 31 -13t13 -31v-1012q0 -18 -13 -31t-31 -13h-1112q-18 0 -31 13t-13 31v1012q0 18 13 31t31 13zM100 1000v-737l247 182l298 -131l-74 156l293 318l236 -288v500h-1000zM342 884q56 0 95 -39t39 -94.5t-39 -95t-95 -39.5t-95 39.5t-39 95t39 94.5 t95 39z" />
<glyph unicode="&#xe062;" d="M648 1169q117 0 216 -60t156.5 -161t57.5 -218q0 -115 -70 -258q-69 -109 -158 -225.5t-143 -179.5l-54 -62q-9 8 -25.5 24.5t-63.5 67.5t-91 103t-98.5 128t-95.5 148q-60 132 -60 249q0 88 34 169.5t91.5 142t137 96.5t166.5 36zM652.5 974q-91.5 0 -156.5 -65 t-65 -157t65 -156.5t156.5 -64.5t156.5 64.5t65 156.5t-65 157t-156.5 65z" />
<glyph unicode="&#xe063;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 173v854q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57z" />
<glyph unicode="&#xe064;" d="M554 1295q21 -72 57.5 -143.5t76 -130t83 -118t82.5 -117t70 -116t49.5 -126t18.5 -136.5q0 -71 -25.5 -135t-68.5 -111t-99 -82t-118.5 -54t-125.5 -23q-84 5 -161.5 34t-139.5 78.5t-99 125t-37 164.5q0 69 18 136.5t49.5 126.5t69.5 116.5t81.5 117.5t83.5 119 t76.5 131t58.5 143zM344 710q-23 -33 -43.5 -70.5t-40.5 -102.5t-17 -123q1 -37 14.5 -69.5t30 -52t41 -37t38.5 -24.5t33 -15q21 -7 32 -1t13 22l6 34q2 10 -2.5 22t-13.5 19q-5 4 -14 12t-29.5 40.5t-32.5 73.5q-26 89 6 271q2 11 -6 11q-8 1 -15 -10z" />
<glyph unicode="&#xe065;" d="M1000 1013l108 115q2 1 5 2t13 2t20.5 -1t25 -9.5t28.5 -21.5q22 -22 27 -43t0 -32l-6 -10l-108 -115zM350 1100h400q50 0 105 -13l-187 -187h-368q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5v182l200 200v-332 q0 -165 -93.5 -257.5t-256.5 -92.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400q0 165 92.5 257.5t257.5 92.5zM1009 803l-362 -362l-161 -50l55 170l355 355z" />
<glyph unicode="&#xe066;" d="M350 1100h361q-164 -146 -216 -200h-195q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5l200 153v-103q0 -165 -92.5 -257.5t-257.5 -92.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400q0 165 92.5 257.5t257.5 92.5z M824 1073l339 -301q8 -7 8 -17.5t-8 -17.5l-340 -306q-7 -6 -12.5 -4t-6.5 11v203q-26 1 -54.5 0t-78.5 -7.5t-92 -17.5t-86 -35t-70 -57q10 59 33 108t51.5 81.5t65 58.5t68.5 40.5t67 24.5t56 13.5t40 4.5v210q1 10 6.5 12.5t13.5 -4.5z" />
<glyph unicode="&#xe067;" d="M350 1100h350q60 0 127 -23l-178 -177h-349q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5v69l200 200v-219q0 -165 -92.5 -257.5t-257.5 -92.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400q0 165 92.5 257.5t257.5 92.5z M643 639l395 395q7 7 17.5 7t17.5 -7l101 -101q7 -7 7 -17.5t-7 -17.5l-531 -532q-7 -7 -17.5 -7t-17.5 7l-248 248q-7 7 -7 17.5t7 17.5l101 101q7 7 17.5 7t17.5 -7l111 -111q8 -7 18 -7t18 7z" />
<glyph unicode="&#xe068;" d="M318 918l264 264q8 8 18 8t18 -8l260 -264q7 -8 4.5 -13t-12.5 -5h-170v-200h200v173q0 10 5 12t13 -5l264 -260q8 -7 8 -17.5t-8 -17.5l-264 -265q-8 -7 -13 -5t-5 12v173h-200v-200h170q10 0 12.5 -5t-4.5 -13l-260 -264q-8 -8 -18 -8t-18 8l-264 264q-8 8 -5.5 13 t12.5 5h175v200h-200v-173q0 -10 -5 -12t-13 5l-264 265q-8 7 -8 17.5t8 17.5l264 260q8 7 13 5t5 -12v-173h200v200h-175q-10 0 -12.5 5t5.5 13z" />
<glyph unicode="&#xe069;" d="M250 1100h100q21 0 35.5 -14.5t14.5 -35.5v-438l464 453q15 14 25.5 10t10.5 -25v-1000q0 -21 -10.5 -25t-25.5 10l-464 453v-438q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v1000q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe070;" d="M50 1100h100q21 0 35.5 -14.5t14.5 -35.5v-438l464 453q15 14 25.5 10t10.5 -25v-438l464 453q15 14 25.5 10t10.5 -25v-1000q0 -21 -10.5 -25t-25.5 10l-464 453v-438q0 -21 -10.5 -25t-25.5 10l-464 453v-438q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5 t-14.5 35.5v1000q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe071;" d="M1200 1050v-1000q0 -21 -10.5 -25t-25.5 10l-464 453v-438q0 -21 -10.5 -25t-25.5 10l-492 480q-15 14 -15 35t15 35l492 480q15 14 25.5 10t10.5 -25v-438l464 453q15 14 25.5 10t10.5 -25z" />
<glyph unicode="&#xe072;" d="M243 1074l814 -498q18 -11 18 -26t-18 -26l-814 -498q-18 -11 -30.5 -4t-12.5 28v1000q0 21 12.5 28t30.5 -4z" />
<glyph unicode="&#xe073;" d="M250 1000h200q21 0 35.5 -14.5t14.5 -35.5v-800q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v800q0 21 14.5 35.5t35.5 14.5zM650 1000h200q21 0 35.5 -14.5t14.5 -35.5v-800q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v800 q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe074;" d="M1100 950v-800q0 -21 -14.5 -35.5t-35.5 -14.5h-800q-21 0 -35.5 14.5t-14.5 35.5v800q0 21 14.5 35.5t35.5 14.5h800q21 0 35.5 -14.5t14.5 -35.5z" />
<glyph unicode="&#xe075;" d="M500 612v438q0 21 10.5 25t25.5 -10l492 -480q15 -14 15 -35t-15 -35l-492 -480q-15 -14 -25.5 -10t-10.5 25v438l-464 -453q-15 -14 -25.5 -10t-10.5 25v1000q0 21 10.5 25t25.5 -10z" />
<glyph unicode="&#xe076;" d="M1048 1102l100 1q20 0 35 -14.5t15 -35.5l5 -1000q0 -21 -14.5 -35.5t-35.5 -14.5l-100 -1q-21 0 -35.5 14.5t-14.5 35.5l-2 437l-463 -454q-14 -15 -24.5 -10.5t-10.5 25.5l-2 437l-462 -455q-15 -14 -25.5 -9.5t-10.5 24.5l-5 1000q0 21 10.5 25.5t25.5 -10.5l466 -450 l-2 438q0 20 10.5 24.5t25.5 -9.5l466 -451l-2 438q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe077;" d="M850 1100h100q21 0 35.5 -14.5t14.5 -35.5v-1000q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v438l-464 -453q-15 -14 -25.5 -10t-10.5 25v1000q0 21 10.5 25t25.5 -10l464 -453v438q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe078;" d="M686 1081l501 -540q15 -15 10.5 -26t-26.5 -11h-1042q-22 0 -26.5 11t10.5 26l501 540q15 15 36 15t36 -15zM150 400h1000q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1000q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe079;" d="M885 900l-352 -353l352 -353l-197 -198l-552 552l552 550z" />
<glyph unicode="&#xe080;" d="M1064 547l-551 -551l-198 198l353 353l-353 353l198 198z" />
<glyph unicode="&#xe081;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM650 900h-100q-21 0 -35.5 -14.5t-14.5 -35.5v-150h-150 q-21 0 -35.5 -14.5t-14.5 -35.5v-100q0 -21 14.5 -35.5t35.5 -14.5h150v-150q0 -21 14.5 -35.5t35.5 -14.5h100q21 0 35.5 14.5t14.5 35.5v150h150q21 0 35.5 14.5t14.5 35.5v100q0 21 -14.5 35.5t-35.5 14.5h-150v150q0 21 -14.5 35.5t-35.5 14.5z" />
<glyph unicode="&#xe082;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM850 700h-500q-21 0 -35.5 -14.5t-14.5 -35.5v-100q0 -21 14.5 -35.5 t35.5 -14.5h500q21 0 35.5 14.5t14.5 35.5v100q0 21 -14.5 35.5t-35.5 14.5z" />
<glyph unicode="&#xe083;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM741.5 913q-12.5 0 -21.5 -9l-120 -120l-120 120q-9 9 -21.5 9 t-21.5 -9l-141 -141q-9 -9 -9 -21.5t9 -21.5l120 -120l-120 -120q-9 -9 -9 -21.5t9 -21.5l141 -141q9 -9 21.5 -9t21.5 9l120 120l120 -120q9 -9 21.5 -9t21.5 9l141 141q9 9 9 21.5t-9 21.5l-120 120l120 120q9 9 9 21.5t-9 21.5l-141 141q-9 9 -21.5 9z" />
<glyph unicode="&#xe084;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM546 623l-84 85q-7 7 -17.5 7t-18.5 -7l-139 -139q-7 -8 -7 -18t7 -18 l242 -241q7 -8 17.5 -8t17.5 8l375 375q7 7 7 17.5t-7 18.5l-139 139q-7 7 -17.5 7t-17.5 -7z" />
<glyph unicode="&#xe085;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM588 941q-29 0 -59 -5.5t-63 -20.5t-58 -38.5t-41.5 -63t-16.5 -89.5 q0 -25 20 -25h131q30 -5 35 11q6 20 20.5 28t45.5 8q20 0 31.5 -10.5t11.5 -28.5q0 -23 -7 -34t-26 -18q-1 0 -13.5 -4t-19.5 -7.5t-20 -10.5t-22 -17t-18.5 -24t-15.5 -35t-8 -46q-1 -8 5.5 -16.5t20.5 -8.5h173q7 0 22 8t35 28t37.5 48t29.5 74t12 100q0 47 -17 83 t-42.5 57t-59.5 34.5t-64 18t-59 4.5zM675 400h-150q-10 0 -17.5 -7.5t-7.5 -17.5v-150q0 -10 7.5 -17.5t17.5 -7.5h150q10 0 17.5 7.5t7.5 17.5v150q0 10 -7.5 17.5t-17.5 7.5z" />
<glyph unicode="&#xe086;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM675 1000h-150q-10 0 -17.5 -7.5t-7.5 -17.5v-150q0 -10 7.5 -17.5 t17.5 -7.5h150q10 0 17.5 7.5t7.5 17.5v150q0 10 -7.5 17.5t-17.5 7.5zM675 700h-250q-10 0 -17.5 -7.5t-7.5 -17.5v-50q0 -10 7.5 -17.5t17.5 -7.5h75v-200h-75q-10 0 -17.5 -7.5t-7.5 -17.5v-50q0 -10 7.5 -17.5t17.5 -7.5h350q10 0 17.5 7.5t7.5 17.5v50q0 10 -7.5 17.5 t-17.5 7.5h-75v275q0 10 -7.5 17.5t-17.5 7.5z" />
<glyph unicode="&#xe087;" d="M525 1200h150q10 0 17.5 -7.5t7.5 -17.5v-194q103 -27 178.5 -102.5t102.5 -178.5h194q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-194q-27 -103 -102.5 -178.5t-178.5 -102.5v-194q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v194 q-103 27 -178.5 102.5t-102.5 178.5h-194q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5h194q27 103 102.5 178.5t178.5 102.5v194q0 10 7.5 17.5t17.5 7.5zM700 893v-168q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v168q-68 -23 -119 -74 t-74 -119h168q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-168q23 -68 74 -119t119 -74v168q0 10 7.5 17.5t17.5 7.5h150q10 0 17.5 -7.5t7.5 -17.5v-168q68 23 119 74t74 119h-168q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5h168 q-23 68 -74 119t-119 74z" />
<glyph unicode="&#xe088;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5t-57 214.5t-155.5 155.5t-214.5 57zM759 823l64 -64q7 -7 7 -17.5t-7 -17.5l-124 -124l124 -124q7 -7 7 -17.5t-7 -17.5l-64 -64q-7 -7 -17.5 -7t-17.5 7l-124 124l-124 -124q-7 -7 -17.5 -7t-17.5 7l-64 64 q-7 7 -7 17.5t7 17.5l124 124l-124 124q-7 7 -7 17.5t7 17.5l64 64q7 7 17.5 7t17.5 -7l124 -124l124 124q7 7 17.5 7t17.5 -7z" />
<glyph unicode="&#xe089;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5t57 -214.5 t155.5 -155.5t214.5 -57t214.5 57t155.5 155.5t57 214.5t-57 214.5t-155.5 155.5t-214.5 57zM782 788l106 -106q7 -7 7 -17.5t-7 -17.5l-320 -321q-8 -7 -18 -7t-18 7l-202 203q-8 7 -8 17.5t8 17.5l106 106q7 8 17.5 8t17.5 -8l79 -79l197 197q7 7 17.5 7t17.5 -7z" />
<glyph unicode="&#xe090;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM600 1027q-116 0 -214.5 -57t-155.5 -155.5t-57 -214.5q0 -120 65 -225 l587 587q-105 65 -225 65zM965 819l-584 -584q104 -62 219 -62q116 0 214.5 57t155.5 155.5t57 214.5q0 115 -62 219z" />
<glyph unicode="&#xe091;" d="M39 582l522 427q16 13 27.5 8t11.5 -26v-291h550q21 0 35.5 -14.5t14.5 -35.5v-200q0 -21 -14.5 -35.5t-35.5 -14.5h-550v-291q0 -21 -11.5 -26t-27.5 8l-522 427q-16 13 -16 32t16 32z" />
<glyph unicode="&#xe092;" d="M639 1009l522 -427q16 -13 16 -32t-16 -32l-522 -427q-16 -13 -27.5 -8t-11.5 26v291h-550q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5h550v291q0 21 11.5 26t27.5 -8z" />
<glyph unicode="&#xe093;" d="M682 1161l427 -522q13 -16 8 -27.5t-26 -11.5h-291v-550q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v550h-291q-21 0 -26 11.5t8 27.5l427 522q13 16 32 16t32 -16z" />
<glyph unicode="&#xe094;" d="M550 1200h200q21 0 35.5 -14.5t14.5 -35.5v-550h291q21 0 26 -11.5t-8 -27.5l-427 -522q-13 -16 -32 -16t-32 16l-427 522q-13 16 -8 27.5t26 11.5h291v550q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe095;" d="M639 1109l522 -427q16 -13 16 -32t-16 -32l-522 -427q-16 -13 -27.5 -8t-11.5 26v291q-94 -2 -182 -20t-170.5 -52t-147 -92.5t-100.5 -135.5q5 105 27 193.5t67.5 167t113 135t167 91.5t225.5 42v262q0 21 11.5 26t27.5 -8z" />
<glyph unicode="&#xe096;" d="M850 1200h300q21 0 35.5 -14.5t14.5 -35.5v-300q0 -21 -10.5 -25t-24.5 10l-94 94l-249 -249q-8 -7 -18 -7t-18 7l-106 106q-7 8 -7 18t7 18l249 249l-94 94q-14 14 -10 24.5t25 10.5zM350 0h-300q-21 0 -35.5 14.5t-14.5 35.5v300q0 21 10.5 25t24.5 -10l94 -94l249 249 q8 7 18 7t18 -7l106 -106q7 -8 7 -18t-7 -18l-249 -249l94 -94q14 -14 10 -24.5t-25 -10.5z" />
<glyph unicode="&#xe097;" d="M1014 1120l106 -106q7 -8 7 -18t-7 -18l-249 -249l94 -94q14 -14 10 -24.5t-25 -10.5h-300q-21 0 -35.5 14.5t-14.5 35.5v300q0 21 10.5 25t24.5 -10l94 -94l249 249q8 7 18 7t18 -7zM250 600h300q21 0 35.5 -14.5t14.5 -35.5v-300q0 -21 -10.5 -25t-24.5 10l-94 94 l-249 -249q-8 -7 -18 -7t-18 7l-106 106q-7 8 -7 18t7 18l249 249l-94 94q-14 14 -10 24.5t25 10.5z" />
<glyph unicode="&#xe101;" d="M600 1177q117 0 224 -45.5t184.5 -123t123 -184.5t45.5 -224t-45.5 -224t-123 -184.5t-184.5 -123t-224 -45.5t-224 45.5t-184.5 123t-123 184.5t-45.5 224t45.5 224t123 184.5t184.5 123t224 45.5zM704 900h-208q-20 0 -32 -14.5t-8 -34.5l58 -302q4 -20 21.5 -34.5 t37.5 -14.5h54q20 0 37.5 14.5t21.5 34.5l58 302q4 20 -8 34.5t-32 14.5zM675 400h-150q-10 0 -17.5 -7.5t-7.5 -17.5v-150q0 -10 7.5 -17.5t17.5 -7.5h150q10 0 17.5 7.5t7.5 17.5v150q0 10 -7.5 17.5t-17.5 7.5z" />
<glyph unicode="&#xe102;" d="M260 1200q9 0 19 -2t15 -4l5 -2q22 -10 44 -23l196 -118q21 -13 36 -24q29 -21 37 -12q11 13 49 35l196 118q22 13 45 23q17 7 38 7q23 0 47 -16.5t37 -33.5l13 -16q14 -21 18 -45l25 -123l8 -44q1 -9 8.5 -14.5t17.5 -5.5h61q10 0 17.5 -7.5t7.5 -17.5v-50 q0 -10 -7.5 -17.5t-17.5 -7.5h-50q-10 0 -17.5 -7.5t-7.5 -17.5v-175h-400v300h-200v-300h-400v175q0 10 -7.5 17.5t-17.5 7.5h-50q-10 0 -17.5 7.5t-7.5 17.5v50q0 10 7.5 17.5t17.5 7.5h61q11 0 18 3t7 8q0 4 9 52l25 128q5 25 19 45q2 3 5 7t13.5 15t21.5 19.5t26.5 15.5 t29.5 7zM915 1079l-166 -162q-7 -7 -5 -12t12 -5h219q10 0 15 7t2 17l-51 149q-3 10 -11 12t-15 -6zM463 917l-177 157q-8 7 -16 5t-11 -12l-51 -143q-3 -10 2 -17t15 -7h231q11 0 12.5 5t-5.5 12zM500 0h-375q-10 0 -17.5 7.5t-7.5 17.5v375h400v-400zM1100 400v-375 q0 -10 -7.5 -17.5t-17.5 -7.5h-375v400h400z" />
<glyph unicode="&#xe103;" d="M1165 1190q8 3 21 -6.5t13 -17.5q-2 -178 -24.5 -323.5t-55.5 -245.5t-87 -174.5t-102.5 -118.5t-118 -68.5t-118.5 -33t-120 -4.5t-105 9.5t-90 16.5q-61 12 -78 11q-4 1 -12.5 0t-34 -14.5t-52.5 -40.5l-153 -153q-26 -24 -37 -14.5t-11 43.5q0 64 42 102q8 8 50.5 45 t66.5 58q19 17 35 47t13 61q-9 55 -10 102.5t7 111t37 130t78 129.5q39 51 80 88t89.5 63.5t94.5 45t113.5 36t129 31t157.5 37t182 47.5zM1116 1098q-8 9 -22.5 -3t-45.5 -50q-38 -47 -119 -103.5t-142 -89.5l-62 -33q-56 -30 -102 -57t-104 -68t-102.5 -80.5t-85.5 -91 t-64 -104.5q-24 -56 -31 -86t2 -32t31.5 17.5t55.5 59.5q25 30 94 75.5t125.5 77.5t147.5 81q70 37 118.5 69t102 79.5t99 111t86.5 148.5q22 50 24 60t-6 19z" />
<glyph unicode="&#xe104;" d="M653 1231q-39 -67 -54.5 -131t-10.5 -114.5t24.5 -96.5t47.5 -80t63.5 -62.5t68.5 -46.5t65 -30q-4 7 -17.5 35t-18.5 39.5t-17 39.5t-17 43t-13 42t-9.5 44.5t-2 42t4 43t13.5 39t23 38.5q96 -42 165 -107.5t105 -138t52 -156t13 -159t-19 -149.5q-13 -55 -44 -106.5 t-68 -87t-78.5 -64.5t-72.5 -45t-53 -22q-72 -22 -127 -11q-31 6 -13 19q6 3 17 7q13 5 32.5 21t41 44t38.5 63.5t21.5 81.5t-6.5 94.5t-50 107t-104 115.5q10 -104 -0.5 -189t-37 -140.5t-65 -93t-84 -52t-93.5 -11t-95 24.5q-80 36 -131.5 114t-53.5 171q-2 23 0 49.5 t4.5 52.5t13.5 56t27.5 60t46 64.5t69.5 68.5q-8 -53 -5 -102.5t17.5 -90t34 -68.5t44.5 -39t49 -2q31 13 38.5 36t-4.5 55t-29 64.5t-36 75t-26 75.5q-15 85 2 161.5t53.5 128.5t85.5 92.5t93.5 61t81.5 25.5z" />
<glyph unicode="&#xe105;" d="M600 1094q82 0 160.5 -22.5t140 -59t116.5 -82.5t94.5 -95t68 -95t42.5 -82.5t14 -57.5t-14 -57.5t-43 -82.5t-68.5 -95t-94.5 -95t-116.5 -82.5t-140 -59t-159.5 -22.5t-159.5 22.5t-140 59t-116.5 82.5t-94.5 95t-68.5 95t-43 82.5t-14 57.5t14 57.5t42.5 82.5t68 95 t94.5 95t116.5 82.5t140 59t160.5 22.5zM888 829q-15 15 -18 12t5 -22q25 -57 25 -119q0 -124 -88 -212t-212 -88t-212 88t-88 212q0 59 23 114q8 19 4.5 22t-17.5 -12q-70 -69 -160 -184q-13 -16 -15 -40.5t9 -42.5q22 -36 47 -71t70 -82t92.5 -81t113 -58.5t133.5 -24.5 t133.5 24t113 58.5t92.5 81.5t70 81.5t47 70.5q11 18 9 42.5t-14 41.5q-90 117 -163 189zM448 727l-35 -36q-15 -15 -19.5 -38.5t4.5 -41.5q37 -68 93 -116q16 -13 38.5 -11t36.5 17l35 34q14 15 12.5 33.5t-16.5 33.5q-44 44 -89 117q-11 18 -28 20t-32 -12z" />
<glyph unicode="&#xe106;" d="M592 0h-148l31 120q-91 20 -175.5 68.5t-143.5 106.5t-103.5 119t-66.5 110t-22 76q0 21 14 57.5t42.5 82.5t68 95t94.5 95t116.5 82.5t140 59t160.5 22.5q61 0 126 -15l32 121h148zM944 770l47 181q108 -85 176.5 -192t68.5 -159q0 -26 -19.5 -71t-59.5 -102t-93 -112 t-129 -104.5t-158 -75.5l46 173q77 49 136 117t97 131q11 18 9 42.5t-14 41.5q-54 70 -107 130zM310 824q-70 -69 -160 -184q-13 -16 -15 -40.5t9 -42.5q18 -30 39 -60t57 -70.5t74 -73t90 -61t105 -41.5l41 154q-107 18 -178.5 101.5t-71.5 193.5q0 59 23 114q8 19 4.5 22 t-17.5 -12zM448 727l-35 -36q-15 -15 -19.5 -38.5t4.5 -41.5q37 -68 93 -116q16 -13 38.5 -11t36.5 17l12 11l22 86l-3 4q-44 44 -89 117q-11 18 -28 20t-32 -12z" />
<glyph unicode="&#xe107;" d="M-90 100l642 1066q20 31 48 28.5t48 -35.5l642 -1056q21 -32 7.5 -67.5t-50.5 -35.5h-1294q-37 0 -50.5 34t7.5 66zM155 200h345v75q0 10 7.5 17.5t17.5 7.5h150q10 0 17.5 -7.5t7.5 -17.5v-75h345l-445 723zM496 700h208q20 0 32 -14.5t8 -34.5l-58 -252 q-4 -20 -21.5 -34.5t-37.5 -14.5h-54q-20 0 -37.5 14.5t-21.5 34.5l-58 252q-4 20 8 34.5t32 14.5z" />
<glyph unicode="&#xe108;" d="M650 1200q62 0 106 -44t44 -106v-339l363 -325q15 -14 26 -38.5t11 -44.5v-41q0 -20 -12 -26.5t-29 5.5l-359 249v-263q100 -93 100 -113v-64q0 -21 -13 -29t-32 1l-205 128l-205 -128q-19 -9 -32 -1t-13 29v64q0 20 100 113v263l-359 -249q-17 -12 -29 -5.5t-12 26.5v41 q0 20 11 44.5t26 38.5l363 325v339q0 62 44 106t106 44z" />
<glyph unicode="&#xe109;" d="M850 1200h100q21 0 35.5 -14.5t14.5 -35.5v-50h50q21 0 35.5 -14.5t14.5 -35.5v-150h-1100v150q0 21 14.5 35.5t35.5 14.5h50v50q0 21 14.5 35.5t35.5 14.5h100q21 0 35.5 -14.5t14.5 -35.5v-50h500v50q0 21 14.5 35.5t35.5 14.5zM1100 800v-750q0 -21 -14.5 -35.5 t-35.5 -14.5h-1000q-21 0 -35.5 14.5t-14.5 35.5v750h1100zM100 600v-100h100v100h-100zM300 600v-100h100v100h-100zM500 600v-100h100v100h-100zM700 600v-100h100v100h-100zM900 600v-100h100v100h-100zM100 400v-100h100v100h-100zM300 400v-100h100v100h-100zM500 400 v-100h100v100h-100zM700 400v-100h100v100h-100zM900 400v-100h100v100h-100zM100 200v-100h100v100h-100zM300 200v-100h100v100h-100zM500 200v-100h100v100h-100zM700 200v-100h100v100h-100zM900 200v-100h100v100h-100z" />
<glyph unicode="&#xe110;" d="M1135 1165l249 -230q15 -14 15 -35t-15 -35l-249 -230q-14 -14 -24.5 -10t-10.5 25v150h-159l-600 -600h-291q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h209l600 600h241v150q0 21 10.5 25t24.5 -10zM522 819l-141 -141l-122 122h-209q-21 0 -35.5 14.5 t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h291zM1135 565l249 -230q15 -14 15 -35t-15 -35l-249 -230q-14 -14 -24.5 -10t-10.5 25v150h-241l-181 181l141 141l122 -122h159v150q0 21 10.5 25t24.5 -10z" />
<glyph unicode="&#xe111;" d="M100 1100h1000q41 0 70.5 -29.5t29.5 -70.5v-600q0 -41 -29.5 -70.5t-70.5 -29.5h-596l-304 -300v300h-100q-41 0 -70.5 29.5t-29.5 70.5v600q0 41 29.5 70.5t70.5 29.5z" />
<glyph unicode="&#xe112;" d="M150 1200h200q21 0 35.5 -14.5t14.5 -35.5v-250h-300v250q0 21 14.5 35.5t35.5 14.5zM850 1200h200q21 0 35.5 -14.5t14.5 -35.5v-250h-300v250q0 21 14.5 35.5t35.5 14.5zM1100 800v-300q0 -41 -3 -77.5t-15 -89.5t-32 -96t-58 -89t-89 -77t-129 -51t-174 -20t-174 20 t-129 51t-89 77t-58 89t-32 96t-15 89.5t-3 77.5v300h300v-250v-27v-42.5t1.5 -41t5 -38t10 -35t16.5 -30t25.5 -24.5t35 -19t46.5 -12t60 -4t60 4.5t46.5 12.5t35 19.5t25 25.5t17 30.5t10 35t5 38t2 40.5t-0.5 42v25v250h300z" />
<glyph unicode="&#xe113;" d="M1100 411l-198 -199l-353 353l-353 -353l-197 199l551 551z" />
<glyph unicode="&#xe114;" d="M1101 789l-550 -551l-551 551l198 199l353 -353l353 353z" />
<glyph unicode="&#xe115;" d="M404 1000h746q21 0 35.5 -14.5t14.5 -35.5v-551h150q21 0 25 -10.5t-10 -24.5l-230 -249q-14 -15 -35 -15t-35 15l-230 249q-14 14 -10 24.5t25 10.5h150v401h-381zM135 984l230 -249q14 -14 10 -24.5t-25 -10.5h-150v-400h385l215 -200h-750q-21 0 -35.5 14.5 t-14.5 35.5v550h-150q-21 0 -25 10.5t10 24.5l230 249q14 15 35 15t35 -15z" />
<glyph unicode="&#xe116;" d="M56 1200h94q17 0 31 -11t18 -27l38 -162h896q24 0 39 -18.5t10 -42.5l-100 -475q-5 -21 -27 -42.5t-55 -21.5h-633l48 -200h535q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-50v-50q0 -21 -14.5 -35.5t-35.5 -14.5t-35.5 14.5t-14.5 35.5v50h-300v-50 q0 -21 -14.5 -35.5t-35.5 -14.5t-35.5 14.5t-14.5 35.5v50h-31q-18 0 -32.5 10t-20.5 19l-5 10l-201 961h-54q-20 0 -35 14.5t-15 35.5t15 35.5t35 14.5z" />
<glyph unicode="&#xe117;" d="M1200 1000v-100h-1200v100h200q0 41 29.5 70.5t70.5 29.5h300q41 0 70.5 -29.5t29.5 -70.5h500zM0 800h1200v-800h-1200v800z" />
<glyph unicode="&#xe118;" d="M200 800l-200 -400v600h200q0 41 29.5 70.5t70.5 29.5h300q42 0 71 -29.5t29 -70.5h500v-200h-1000zM1500 700l-300 -700h-1200l300 700h1200z" />
<glyph unicode="&#xe119;" d="M635 1184l230 -249q14 -14 10 -24.5t-25 -10.5h-150v-601h150q21 0 25 -10.5t-10 -24.5l-230 -249q-14 -15 -35 -15t-35 15l-230 249q-14 14 -10 24.5t25 10.5h150v601h-150q-21 0 -25 10.5t10 24.5l230 249q14 15 35 15t35 -15z" />
<glyph unicode="&#xe120;" d="M936 864l249 -229q14 -15 14 -35.5t-14 -35.5l-249 -229q-15 -15 -25.5 -10.5t-10.5 24.5v151h-600v-151q0 -20 -10.5 -24.5t-25.5 10.5l-249 229q-14 15 -14 35.5t14 35.5l249 229q15 15 25.5 10.5t10.5 -25.5v-149h600v149q0 21 10.5 25.5t25.5 -10.5z" />
<glyph unicode="&#xe121;" d="M1169 400l-172 732q-5 23 -23 45.5t-38 22.5h-672q-20 0 -38 -20t-23 -41l-172 -739h1138zM1100 300h-1000q-41 0 -70.5 -29.5t-29.5 -70.5v-100q0 -41 29.5 -70.5t70.5 -29.5h1000q41 0 70.5 29.5t29.5 70.5v100q0 41 -29.5 70.5t-70.5 29.5zM800 100v100h100v-100h-100 zM1000 100v100h100v-100h-100z" />
<glyph unicode="&#xe122;" d="M1150 1100q21 0 35.5 -14.5t14.5 -35.5v-850q0 -21 -14.5 -35.5t-35.5 -14.5t-35.5 14.5t-14.5 35.5v850q0 21 14.5 35.5t35.5 14.5zM1000 200l-675 200h-38l47 -276q3 -16 -5.5 -20t-29.5 -4h-7h-84q-20 0 -34.5 14t-18.5 35q-55 337 -55 351v250v6q0 16 1 23.5t6.5 14 t17.5 6.5h200l675 250v-850zM0 750v-250q-4 0 -11 0.5t-24 6t-30 15t-24 30t-11 48.5v50q0 26 10.5 46t25 30t29 16t25.5 7z" />
<glyph unicode="&#xe123;" d="M553 1200h94q20 0 29 -10.5t3 -29.5l-18 -37q83 -19 144 -82.5t76 -140.5l63 -327l118 -173h17q19 0 33 -14.5t14 -35t-13 -40.5t-31 -27q-8 -4 -23 -9.5t-65 -19.5t-103 -25t-132.5 -20t-158.5 -9q-57 0 -115 5t-104 12t-88.5 15.5t-73.5 17.5t-54.5 16t-35.5 12l-11 4 q-18 8 -31 28t-13 40.5t14 35t33 14.5h17l118 173l63 327q15 77 76 140t144 83l-18 32q-6 19 3.5 32t28.5 13zM498 110q50 -6 102 -6q53 0 102 6q-12 -49 -39.5 -79.5t-62.5 -30.5t-63 30.5t-39 79.5z" />
<glyph unicode="&#xe124;" d="M800 946l224 78l-78 -224l234 -45l-180 -155l180 -155l-234 -45l78 -224l-224 78l-45 -234l-155 180l-155 -180l-45 234l-224 -78l78 224l-234 45l180 155l-180 155l234 45l-78 224l224 -78l45 234l155 -180l155 180z" />
<glyph unicode="&#xe125;" d="M650 1200h50q40 0 70 -40.5t30 -84.5v-150l-28 -125h328q40 0 70 -40.5t30 -84.5v-100q0 -45 -29 -74l-238 -344q-16 -24 -38 -40.5t-45 -16.5h-250q-7 0 -42 25t-66 50l-31 25h-61q-45 0 -72.5 18t-27.5 57v400q0 36 20 63l145 196l96 198q13 28 37.5 48t51.5 20z M650 1100l-100 -212l-150 -213v-375h100l136 -100h214l250 375v125h-450l50 225v175h-50zM50 800h100q21 0 35.5 -14.5t14.5 -35.5v-500q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v500q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe126;" d="M600 1100h250q23 0 45 -16.5t38 -40.5l238 -344q29 -29 29 -74v-100q0 -44 -30 -84.5t-70 -40.5h-328q28 -118 28 -125v-150q0 -44 -30 -84.5t-70 -40.5h-50q-27 0 -51.5 20t-37.5 48l-96 198l-145 196q-20 27 -20 63v400q0 39 27.5 57t72.5 18h61q124 100 139 100z M50 1000h100q21 0 35.5 -14.5t14.5 -35.5v-500q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v500q0 21 14.5 35.5t35.5 14.5zM636 1000l-136 -100h-100v-375l150 -213l100 -212h50v175l-50 225h450v125l-250 375h-214z" />
<glyph unicode="&#xe127;" d="M356 873l363 230q31 16 53 -6l110 -112q13 -13 13.5 -32t-11.5 -34l-84 -121h302q84 0 138 -38t54 -110t-55 -111t-139 -39h-106l-131 -339q-6 -21 -19.5 -41t-28.5 -20h-342q-7 0 -90 81t-83 94v525q0 17 14 35.5t28 28.5zM400 792v-503l100 -89h293l131 339 q6 21 19.5 41t28.5 20h203q21 0 30.5 25t0.5 50t-31 25h-456h-7h-6h-5.5t-6 0.5t-5 1.5t-5 2t-4 2.5t-4 4t-2.5 4.5q-12 25 5 47l146 183l-86 83zM50 800h100q21 0 35.5 -14.5t14.5 -35.5v-500q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v500 q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe128;" d="M475 1103l366 -230q2 -1 6 -3.5t14 -10.5t18 -16.5t14.5 -20t6.5 -22.5v-525q0 -13 -86 -94t-93 -81h-342q-15 0 -28.5 20t-19.5 41l-131 339h-106q-85 0 -139.5 39t-54.5 111t54 110t138 38h302l-85 121q-11 15 -10.5 34t13.5 32l110 112q22 22 53 6zM370 945l146 -183 q17 -22 5 -47q-2 -2 -3.5 -4.5t-4 -4t-4 -2.5t-5 -2t-5 -1.5t-6 -0.5h-6h-6.5h-6h-475v-100h221q15 0 29 -20t20 -41l130 -339h294l106 89v503l-342 236zM1050 800h100q21 0 35.5 -14.5t14.5 -35.5v-500q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5 v500q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe129;" d="M550 1294q72 0 111 -55t39 -139v-106l339 -131q21 -6 41 -19.5t20 -28.5v-342q0 -7 -81 -90t-94 -83h-525q-17 0 -35.5 14t-28.5 28l-9 14l-230 363q-16 31 6 53l112 110q13 13 32 13.5t34 -11.5l121 -84v302q0 84 38 138t110 54zM600 972v203q0 21 -25 30.5t-50 0.5 t-25 -31v-456v-7v-6v-5.5t-0.5 -6t-1.5 -5t-2 -5t-2.5 -4t-4 -4t-4.5 -2.5q-25 -12 -47 5l-183 146l-83 -86l236 -339h503l89 100v293l-339 131q-21 6 -41 19.5t-20 28.5zM450 200h500q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-500 q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe130;" d="M350 1100h500q21 0 35.5 14.5t14.5 35.5v100q0 21 -14.5 35.5t-35.5 14.5h-500q-21 0 -35.5 -14.5t-14.5 -35.5v-100q0 -21 14.5 -35.5t35.5 -14.5zM600 306v-106q0 -84 -39 -139t-111 -55t-110 54t-38 138v302l-121 -84q-15 -12 -34 -11.5t-32 13.5l-112 110 q-22 22 -6 53l230 363q1 2 3.5 6t10.5 13.5t16.5 17t20 13.5t22.5 6h525q13 0 94 -83t81 -90v-342q0 -15 -20 -28.5t-41 -19.5zM308 900l-236 -339l83 -86l183 146q22 17 47 5q2 -1 4.5 -2.5t4 -4t2.5 -4t2 -5t1.5 -5t0.5 -6v-5.5v-6v-7v-456q0 -22 25 -31t50 0.5t25 30.5 v203q0 15 20 28.5t41 19.5l339 131v293l-89 100h-503z" />
<glyph unicode="&#xe131;" d="M600 1178q118 0 225 -45.5t184.5 -123t123 -184.5t45.5 -225t-45.5 -225t-123 -184.5t-184.5 -123t-225 -45.5t-225 45.5t-184.5 123t-123 184.5t-45.5 225t45.5 225t123 184.5t184.5 123t225 45.5zM914 632l-275 223q-16 13 -27.5 8t-11.5 -26v-137h-275 q-10 0 -17.5 -7.5t-7.5 -17.5v-150q0 -10 7.5 -17.5t17.5 -7.5h275v-137q0 -21 11.5 -26t27.5 8l275 223q16 13 16 32t-16 32z" />
<glyph unicode="&#xe132;" d="M600 1178q118 0 225 -45.5t184.5 -123t123 -184.5t45.5 -225t-45.5 -225t-123 -184.5t-184.5 -123t-225 -45.5t-225 45.5t-184.5 123t-123 184.5t-45.5 225t45.5 225t123 184.5t184.5 123t225 45.5zM561 855l-275 -223q-16 -13 -16 -32t16 -32l275 -223q16 -13 27.5 -8 t11.5 26v137h275q10 0 17.5 7.5t7.5 17.5v150q0 10 -7.5 17.5t-17.5 7.5h-275v137q0 21 -11.5 26t-27.5 -8z" />
<glyph unicode="&#xe133;" d="M600 1178q118 0 225 -45.5t184.5 -123t123 -184.5t45.5 -225t-45.5 -225t-123 -184.5t-184.5 -123t-225 -45.5t-225 45.5t-184.5 123t-123 184.5t-45.5 225t45.5 225t123 184.5t184.5 123t225 45.5zM855 639l-223 275q-13 16 -32 16t-32 -16l-223 -275q-13 -16 -8 -27.5 t26 -11.5h137v-275q0 -10 7.5 -17.5t17.5 -7.5h150q10 0 17.5 7.5t7.5 17.5v275h137q21 0 26 11.5t-8 27.5z" />
<glyph unicode="&#xe134;" d="M600 1178q118 0 225 -45.5t184.5 -123t123 -184.5t45.5 -225t-45.5 -225t-123 -184.5t-184.5 -123t-225 -45.5t-225 45.5t-184.5 123t-123 184.5t-45.5 225t45.5 225t123 184.5t184.5 123t225 45.5zM675 900h-150q-10 0 -17.5 -7.5t-7.5 -17.5v-275h-137q-21 0 -26 -11.5 t8 -27.5l223 -275q13 -16 32 -16t32 16l223 275q13 16 8 27.5t-26 11.5h-137v275q0 10 -7.5 17.5t-17.5 7.5z" />
<glyph unicode="&#xe135;" d="M600 1176q116 0 222.5 -46t184 -123.5t123.5 -184t46 -222.5t-46 -222.5t-123.5 -184t-184 -123.5t-222.5 -46t-222.5 46t-184 123.5t-123.5 184t-46 222.5t46 222.5t123.5 184t184 123.5t222.5 46zM627 1101q-15 -12 -36.5 -20.5t-35.5 -12t-43 -8t-39 -6.5 q-15 -3 -45.5 0t-45.5 -2q-20 -7 -51.5 -26.5t-34.5 -34.5q-3 -11 6.5 -22.5t8.5 -18.5q-3 -34 -27.5 -91t-29.5 -79q-9 -34 5 -93t8 -87q0 -9 17 -44.5t16 -59.5q12 0 23 -5t23.5 -15t19.5 -14q16 -8 33 -15t40.5 -15t34.5 -12q21 -9 52.5 -32t60 -38t57.5 -11 q7 -15 -3 -34t-22.5 -40t-9.5 -38q13 -21 23 -34.5t27.5 -27.5t36.5 -18q0 -7 -3.5 -16t-3.5 -14t5 -17q104 -2 221 112q30 29 46.5 47t34.5 49t21 63q-13 8 -37 8.5t-36 7.5q-15 7 -49.5 15t-51.5 19q-18 0 -41 -0.5t-43 -1.5t-42 -6.5t-38 -16.5q-51 -35 -66 -12 q-4 1 -3.5 25.5t0.5 25.5q-6 13 -26.5 17.5t-24.5 6.5q1 15 -0.5 30.5t-7 28t-18.5 11.5t-31 -21q-23 -25 -42 4q-19 28 -8 58q6 16 22 22q6 -1 26 -1.5t33.5 -4t19.5 -13.5q7 -12 18 -24t21.5 -20.5t20 -15t15.5 -10.5l5 -3q2 12 7.5 30.5t8 34.5t-0.5 32q-3 18 3.5 29 t18 22.5t15.5 24.5q6 14 10.5 35t8 31t15.5 22.5t34 22.5q-6 18 10 36q8 0 24 -1.5t24.5 -1.5t20 4.5t20.5 15.5q-10 23 -31 42.5t-37.5 29.5t-49 27t-43.5 23q0 1 2 8t3 11.5t1.5 10.5t-1 9.5t-4.5 4.5q31 -13 58.5 -14.5t38.5 2.5l12 5q5 28 -9.5 46t-36.5 24t-50 15 t-41 20q-18 -4 -37 0zM613 994q0 -17 8 -42t17 -45t9 -23q-8 1 -39.5 5.5t-52.5 10t-37 16.5q3 11 16 29.5t16 25.5q10 -10 19 -10t14 6t13.5 14.5t16.5 12.5z" />
<glyph unicode="&#xe136;" d="M756 1157q164 92 306 -9l-259 -138l145 -232l251 126q6 -89 -34 -156.5t-117 -110.5q-60 -34 -127 -39.5t-126 16.5l-596 -596q-15 -16 -36.5 -16t-36.5 16l-111 110q-15 15 -15 36.5t15 37.5l600 599q-34 101 5.5 201.5t135.5 154.5z" />
<glyph unicode="&#xe137;" horiz-adv-x="1220" d="M100 1196h1000q41 0 70.5 -29.5t29.5 -70.5v-100q0 -41 -29.5 -70.5t-70.5 -29.5h-1000q-41 0 -70.5 29.5t-29.5 70.5v100q0 41 29.5 70.5t70.5 29.5zM1100 1096h-200v-100h200v100zM100 796h1000q41 0 70.5 -29.5t29.5 -70.5v-100q0 -41 -29.5 -70.5t-70.5 -29.5h-1000 q-41 0 -70.5 29.5t-29.5 70.5v100q0 41 29.5 70.5t70.5 29.5zM1100 696h-500v-100h500v100zM100 396h1000q41 0 70.5 -29.5t29.5 -70.5v-100q0 -41 -29.5 -70.5t-70.5 -29.5h-1000q-41 0 -70.5 29.5t-29.5 70.5v100q0 41 29.5 70.5t70.5 29.5zM1100 296h-300v-100h300v100z " />
<glyph unicode="&#xe138;" d="M150 1200h900q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-900q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM700 500v-300l-200 -200v500l-350 500h900z" />
<glyph unicode="&#xe139;" d="M500 1200h200q41 0 70.5 -29.5t29.5 -70.5v-100h300q41 0 70.5 -29.5t29.5 -70.5v-400h-500v100h-200v-100h-500v400q0 41 29.5 70.5t70.5 29.5h300v100q0 41 29.5 70.5t70.5 29.5zM500 1100v-100h200v100h-200zM1200 400v-200q0 -41 -29.5 -70.5t-70.5 -29.5h-1000 q-41 0 -70.5 29.5t-29.5 70.5v200h1200z" />
<glyph unicode="&#xe140;" d="M50 1200h300q21 0 25 -10.5t-10 -24.5l-94 -94l199 -199q7 -8 7 -18t-7 -18l-106 -106q-8 -7 -18 -7t-18 7l-199 199l-94 -94q-14 -14 -24.5 -10t-10.5 25v300q0 21 14.5 35.5t35.5 14.5zM850 1200h300q21 0 35.5 -14.5t14.5 -35.5v-300q0 -21 -10.5 -25t-24.5 10l-94 94 l-199 -199q-8 -7 -18 -7t-18 7l-106 106q-7 8 -7 18t7 18l199 199l-94 94q-14 14 -10 24.5t25 10.5zM364 470l106 -106q7 -8 7 -18t-7 -18l-199 -199l94 -94q14 -14 10 -24.5t-25 -10.5h-300q-21 0 -35.5 14.5t-14.5 35.5v300q0 21 10.5 25t24.5 -10l94 -94l199 199 q8 7 18 7t18 -7zM1071 271l94 94q14 14 24.5 10t10.5 -25v-300q0 -21 -14.5 -35.5t-35.5 -14.5h-300q-21 0 -25 10.5t10 24.5l94 94l-199 199q-7 8 -7 18t7 18l106 106q8 7 18 7t18 -7z" />
<glyph unicode="&#xe141;" d="M596 1192q121 0 231.5 -47.5t190 -127t127 -190t47.5 -231.5t-47.5 -231.5t-127 -190.5t-190 -127t-231.5 -47t-231.5 47t-190.5 127t-127 190.5t-47 231.5t47 231.5t127 190t190.5 127t231.5 47.5zM596 1010q-112 0 -207.5 -55.5t-151 -151t-55.5 -207.5t55.5 -207.5 t151 -151t207.5 -55.5t207.5 55.5t151 151t55.5 207.5t-55.5 207.5t-151 151t-207.5 55.5zM454.5 905q22.5 0 38.5 -16t16 -38.5t-16 -39t-38.5 -16.5t-38.5 16.5t-16 39t16 38.5t38.5 16zM754.5 905q22.5 0 38.5 -16t16 -38.5t-16 -39t-38 -16.5q-14 0 -29 10l-55 -145 q17 -23 17 -51q0 -36 -25.5 -61.5t-61.5 -25.5t-61.5 25.5t-25.5 61.5q0 32 20.5 56.5t51.5 29.5l122 126l1 1q-9 14 -9 28q0 23 16 39t38.5 16zM345.5 709q22.5 0 38.5 -16t16 -38.5t-16 -38.5t-38.5 -16t-38.5 16t-16 38.5t16 38.5t38.5 16zM854.5 709q22.5 0 38.5 -16 t16 -38.5t-16 -38.5t-38.5 -16t-38.5 16t-16 38.5t16 38.5t38.5 16z" />
<glyph unicode="&#xe142;" d="M546 173l469 470q91 91 99 192q7 98 -52 175.5t-154 94.5q-22 4 -47 4q-34 0 -66.5 -10t-56.5 -23t-55.5 -38t-48 -41.5t-48.5 -47.5q-376 -375 -391 -390q-30 -27 -45 -41.5t-37.5 -41t-32 -46.5t-16 -47.5t-1.5 -56.5q9 -62 53.5 -95t99.5 -33q74 0 125 51l548 548 q36 36 20 75q-7 16 -21.5 26t-32.5 10q-26 0 -50 -23q-13 -12 -39 -38l-341 -338q-15 -15 -35.5 -15.5t-34.5 13.5t-14 34.5t14 34.5q327 333 361 367q35 35 67.5 51.5t78.5 16.5q14 0 29 -1q44 -8 74.5 -35.5t43.5 -68.5q14 -47 2 -96.5t-47 -84.5q-12 -11 -32 -32 t-79.5 -81t-114.5 -115t-124.5 -123.5t-123 -119.5t-96.5 -89t-57 -45q-56 -27 -120 -27q-70 0 -129 32t-93 89q-48 78 -35 173t81 163l511 511q71 72 111 96q91 55 198 55q80 0 152 -33q78 -36 129.5 -103t66.5 -154q17 -93 -11 -183.5t-94 -156.5l-482 -476 q-15 -15 -36 -16t-37 14t-17.5 34t14.5 35z" />
<glyph unicode="&#xe143;" d="M649 949q48 68 109.5 104t121.5 38.5t118.5 -20t102.5 -64t71 -100.5t27 -123q0 -57 -33.5 -117.5t-94 -124.5t-126.5 -127.5t-150 -152.5t-146 -174q-62 85 -145.5 174t-150 152.5t-126.5 127.5t-93.5 124.5t-33.5 117.5q0 64 28 123t73 100.5t104 64t119 20 t120.5 -38.5t104.5 -104zM896 972q-33 0 -64.5 -19t-56.5 -46t-47.5 -53.5t-43.5 -45.5t-37.5 -19t-36 19t-40 45.5t-43 53.5t-54 46t-65.5 19q-67 0 -122.5 -55.5t-55.5 -132.5q0 -23 13.5 -51t46 -65t57.5 -63t76 -75l22 -22q15 -14 44 -44t50.5 -51t46 -44t41 -35t23 -12 t23.5 12t42.5 36t46 44t52.5 52t44 43q4 4 12 13q43 41 63.5 62t52 55t46 55t26 46t11.5 44q0 79 -53 133.5t-120 54.5z" />
<glyph unicode="&#xe144;" d="M776.5 1214q93.5 0 159.5 -66l141 -141q66 -66 66 -160q0 -42 -28 -95.5t-62 -87.5l-29 -29q-31 53 -77 99l-18 18l95 95l-247 248l-389 -389l212 -212l-105 -106l-19 18l-141 141q-66 66 -66 159t66 159l283 283q65 66 158.5 66zM600 706l105 105q10 -8 19 -17l141 -141 q66 -66 66 -159t-66 -159l-283 -283q-66 -66 -159 -66t-159 66l-141 141q-66 66 -66 159.5t66 159.5l55 55q29 -55 75 -102l18 -17l-95 -95l247 -248l389 389z" />
<glyph unicode="&#xe145;" d="M603 1200q85 0 162 -15t127 -38t79 -48t29 -46v-953q0 -41 -29.5 -70.5t-70.5 -29.5h-600q-41 0 -70.5 29.5t-29.5 70.5v953q0 21 30 46.5t81 48t129 37.5t163 15zM300 1000v-700h600v700h-600zM600 254q-43 0 -73.5 -30.5t-30.5 -73.5t30.5 -73.5t73.5 -30.5t73.5 30.5 t30.5 73.5t-30.5 73.5t-73.5 30.5z" />
<glyph unicode="&#xe146;" d="M902 1185l283 -282q15 -15 15 -36t-14.5 -35.5t-35.5 -14.5t-35 15l-36 35l-279 -267v-300l-212 210l-308 -307l-280 -203l203 280l307 308l-210 212h300l267 279l-35 36q-15 14 -15 35t14.5 35.5t35.5 14.5t35 -15z" />
<glyph unicode="&#xe148;" d="M700 1248v-78q38 -5 72.5 -14.5t75.5 -31.5t71 -53.5t52 -84t24 -118.5h-159q-4 36 -10.5 59t-21 45t-40 35.5t-64.5 20.5v-307l64 -13q34 -7 64 -16.5t70 -32t67.5 -52.5t47.5 -80t20 -112q0 -139 -89 -224t-244 -97v-77h-100v79q-150 16 -237 103q-40 40 -52.5 93.5 t-15.5 139.5h139q5 -77 48.5 -126t117.5 -65v335l-27 8q-46 14 -79 26.5t-72 36t-63 52t-40 72.5t-16 98q0 70 25 126t67.5 92t94.5 57t110 27v77h100zM600 754v274q-29 -4 -50 -11t-42 -21.5t-31.5 -41.5t-10.5 -65q0 -29 7 -50.5t16.5 -34t28.5 -22.5t31.5 -14t37.5 -10 q9 -3 13 -4zM700 547v-310q22 2 42.5 6.5t45 15.5t41.5 27t29 42t12 59.5t-12.5 59.5t-38 44.5t-53 31t-66.5 24.5z" />
<glyph unicode="&#xe149;" d="M561 1197q84 0 160.5 -40t123.5 -109.5t47 -147.5h-153q0 40 -19.5 71.5t-49.5 48.5t-59.5 26t-55.5 9q-37 0 -79 -14.5t-62 -35.5q-41 -44 -41 -101q0 -26 13.5 -63t26.5 -61t37 -66q6 -9 9 -14h241v-100h-197q8 -50 -2.5 -115t-31.5 -95q-45 -62 -99 -112 q34 10 83 17.5t71 7.5q32 1 102 -16t104 -17q83 0 136 30l50 -147q-31 -19 -58 -30.5t-55 -15.5t-42 -4.5t-46 -0.5q-23 0 -76 17t-111 32.5t-96 11.5q-39 -3 -82 -16t-67 -25l-23 -11l-55 145q4 3 16 11t15.5 10.5t13 9t15.5 12t14.5 14t17.5 18.5q48 55 54 126.5 t-30 142.5h-221v100h166q-23 47 -44 104q-7 20 -12 41.5t-6 55.5t6 66.5t29.5 70.5t58.5 71q97 88 263 88z" />
<glyph unicode="&#xe150;" d="M400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM935 1184l230 -249q14 -14 10 -24.5t-25 -10.5h-150v-900h-200v900h-150q-21 0 -25 10.5t10 24.5l230 249q14 15 35 15t35 -15z" />
<glyph unicode="&#xe151;" d="M1000 700h-100v100h-100v-100h-100v500h300v-500zM400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM801 1100v-200h100v200h-100zM1000 350l-200 -250h200v-100h-300v150l200 250h-200v100h300v-150z " />
<glyph unicode="&#xe152;" d="M400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM1000 1050l-200 -250h200v-100h-300v150l200 250h-200v100h300v-150zM1000 0h-100v100h-100v-100h-100v500h300v-500zM801 400v-200h100v200h-100z " />
<glyph unicode="&#xe153;" d="M400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM1000 700h-100v400h-100v100h200v-500zM1100 0h-100v100h-200v400h300v-500zM901 400v-200h100v200h-100z" />
<glyph unicode="&#xe154;" d="M400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM1100 700h-100v100h-200v400h300v-500zM901 1100v-200h100v200h-100zM1000 0h-100v400h-100v100h200v-500z" />
<glyph unicode="&#xe155;" d="M400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM900 1000h-200v200h200v-200zM1000 700h-300v200h300v-200zM1100 400h-400v200h400v-200zM1200 100h-500v200h500v-200z" />
<glyph unicode="&#xe156;" d="M400 300h150q21 0 25 -11t-10 -25l-230 -250q-14 -15 -35 -15t-35 15l-230 250q-14 14 -10 25t25 11h150v900h200v-900zM1200 1000h-500v200h500v-200zM1100 700h-400v200h400v-200zM1000 400h-300v200h300v-200zM900 100h-200v200h200v-200z" />
<glyph unicode="&#xe157;" d="M350 1100h400q162 0 256 -93.5t94 -256.5v-400q0 -165 -93.5 -257.5t-256.5 -92.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400q0 165 92.5 257.5t257.5 92.5zM800 900h-500q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5 v500q0 41 -29.5 70.5t-70.5 29.5z" />
<glyph unicode="&#xe158;" d="M350 1100h400q165 0 257.5 -92.5t92.5 -257.5v-400q0 -165 -92.5 -257.5t-257.5 -92.5h-400q-163 0 -256.5 92.5t-93.5 257.5v400q0 163 94 256.5t256 93.5zM800 900h-500q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5 v500q0 41 -29.5 70.5t-70.5 29.5zM440 770l253 -190q17 -12 17 -30t-17 -30l-253 -190q-16 -12 -28 -6.5t-12 26.5v400q0 21 12 26.5t28 -6.5z" />
<glyph unicode="&#xe159;" d="M350 1100h400q163 0 256.5 -94t93.5 -256v-400q0 -165 -92.5 -257.5t-257.5 -92.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400q0 163 92.5 256.5t257.5 93.5zM800 900h-500q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5 v500q0 41 -29.5 70.5t-70.5 29.5zM350 700h400q21 0 26.5 -12t-6.5 -28l-190 -253q-12 -17 -30 -17t-30 17l-190 253q-12 16 -6.5 28t26.5 12z" />
<glyph unicode="&#xe160;" d="M350 1100h400q165 0 257.5 -92.5t92.5 -257.5v-400q0 -163 -92.5 -256.5t-257.5 -93.5h-400q-163 0 -256.5 94t-93.5 256v400q0 165 92.5 257.5t257.5 92.5zM800 900h-500q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5 v500q0 41 -29.5 70.5t-70.5 29.5zM580 693l190 -253q12 -16 6.5 -28t-26.5 -12h-400q-21 0 -26.5 12t6.5 28l190 253q12 17 30 17t30 -17z" />
<glyph unicode="&#xe161;" d="M550 1100h400q165 0 257.5 -92.5t92.5 -257.5v-400q0 -165 -92.5 -257.5t-257.5 -92.5h-400q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h450q41 0 70.5 29.5t29.5 70.5v500q0 41 -29.5 70.5t-70.5 29.5h-450q-21 0 -35.5 14.5t-14.5 35.5v100 q0 21 14.5 35.5t35.5 14.5zM338 867l324 -284q16 -14 16 -33t-16 -33l-324 -284q-16 -14 -27 -9t-11 26v150h-250q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5h250v150q0 21 11 26t27 -9z" />
<glyph unicode="&#xe162;" d="M793 1182l9 -9q8 -10 5 -27q-3 -11 -79 -225.5t-78 -221.5l300 1q24 0 32.5 -17.5t-5.5 -35.5q-1 0 -133.5 -155t-267 -312.5t-138.5 -162.5q-12 -15 -26 -15h-9l-9 8q-9 11 -4 32q2 9 42 123.5t79 224.5l39 110h-302q-23 0 -31 19q-10 21 6 41q75 86 209.5 237.5 t228 257t98.5 111.5q9 16 25 16h9z" />
<glyph unicode="&#xe163;" d="M350 1100h400q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-450q-41 0 -70.5 -29.5t-29.5 -70.5v-500q0 -41 29.5 -70.5t70.5 -29.5h450q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400 q0 165 92.5 257.5t257.5 92.5zM938 867l324 -284q16 -14 16 -33t-16 -33l-324 -284q-16 -14 -27 -9t-11 26v150h-250q-21 0 -35.5 14.5t-14.5 35.5v200q0 21 14.5 35.5t35.5 14.5h250v150q0 21 11 26t27 -9z" />
<glyph unicode="&#xe164;" d="M750 1200h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -10.5 -25t-24.5 10l-109 109l-312 -312q-15 -15 -35.5 -15t-35.5 15l-141 141q-15 15 -15 35.5t15 35.5l312 312l-109 109q-14 14 -10 24.5t25 10.5zM456 900h-156q-41 0 -70.5 -29.5t-29.5 -70.5v-500 q0 -41 29.5 -70.5t70.5 -29.5h500q41 0 70.5 29.5t29.5 70.5v148l200 200v-298q0 -165 -93.5 -257.5t-256.5 -92.5h-400q-165 0 -257.5 92.5t-92.5 257.5v400q0 165 92.5 257.5t257.5 92.5h300z" />
<glyph unicode="&#xe165;" d="M600 1186q119 0 227.5 -46.5t187 -125t125 -187t46.5 -227.5t-46.5 -227.5t-125 -187t-187 -125t-227.5 -46.5t-227.5 46.5t-187 125t-125 187t-46.5 227.5t46.5 227.5t125 187t187 125t227.5 46.5zM600 1022q-115 0 -212 -56.5t-153.5 -153.5t-56.5 -212t56.5 -212 t153.5 -153.5t212 -56.5t212 56.5t153.5 153.5t56.5 212t-56.5 212t-153.5 153.5t-212 56.5zM600 794q80 0 137 -57t57 -137t-57 -137t-137 -57t-137 57t-57 137t57 137t137 57z" />
<glyph unicode="&#xe166;" d="M450 1200h200q21 0 35.5 -14.5t14.5 -35.5v-350h245q20 0 25 -11t-9 -26l-383 -426q-14 -15 -33.5 -15t-32.5 15l-379 426q-13 15 -8.5 26t25.5 11h250v350q0 21 14.5 35.5t35.5 14.5zM50 300h1000q21 0 35.5 -14.5t14.5 -35.5v-250h-1100v250q0 21 14.5 35.5t35.5 14.5z M900 200v-50h100v50h-100z" />
<glyph unicode="&#xe167;" d="M583 1182l378 -435q14 -15 9 -31t-26 -16h-244v-250q0 -20 -17 -35t-39 -15h-200q-20 0 -32 14.5t-12 35.5v250h-250q-20 0 -25.5 16.5t8.5 31.5l383 431q14 16 33.5 17t33.5 -14zM50 300h1000q21 0 35.5 -14.5t14.5 -35.5v-250h-1100v250q0 21 14.5 35.5t35.5 14.5z M900 200v-50h100v50h-100z" />
<glyph unicode="&#xe168;" d="M396 723l369 369q7 7 17.5 7t17.5 -7l139 -139q7 -8 7 -18.5t-7 -17.5l-525 -525q-7 -8 -17.5 -8t-17.5 8l-292 291q-7 8 -7 18t7 18l139 139q8 7 18.5 7t17.5 -7zM50 300h1000q21 0 35.5 -14.5t14.5 -35.5v-250h-1100v250q0 21 14.5 35.5t35.5 14.5zM900 200v-50h100v50 h-100z" />
<glyph unicode="&#xe169;" d="M135 1023l142 142q14 14 35 14t35 -14l77 -77l-212 -212l-77 76q-14 15 -14 36t14 35zM655 855l210 210q14 14 24.5 10t10.5 -25l-2 -599q-1 -20 -15.5 -35t-35.5 -15l-597 -1q-21 0 -25 10.5t10 24.5l208 208l-154 155l212 212zM50 300h1000q21 0 35.5 -14.5t14.5 -35.5 v-250h-1100v250q0 21 14.5 35.5t35.5 14.5zM900 200v-50h100v50h-100z" />
<glyph unicode="&#xe170;" d="M350 1200l599 -2q20 -1 35 -15.5t15 -35.5l1 -597q0 -21 -10.5 -25t-24.5 10l-208 208l-155 -154l-212 212l155 154l-210 210q-14 14 -10 24.5t25 10.5zM524 512l-76 -77q-15 -14 -36 -14t-35 14l-142 142q-14 14 -14 35t14 35l77 77zM50 300h1000q21 0 35.5 -14.5 t14.5 -35.5v-250h-1100v250q0 21 14.5 35.5t35.5 14.5zM900 200v-50h100v50h-100z" />
<glyph unicode="&#xe171;" d="M1200 103l-483 276l-314 -399v423h-399l1196 796v-1096zM483 424v-230l683 953z" />
<glyph unicode="&#xe172;" d="M1100 1000v-850q0 -21 -14.5 -35.5t-35.5 -14.5h-150v400h-700v-400h-150q-21 0 -35.5 14.5t-14.5 35.5v1000q0 20 14.5 35t35.5 15h250v-300h500v300h100zM700 1000h-100v200h100v-200z" />
<glyph unicode="&#xe173;" d="M1100 1000l-2 -149l-299 -299l-95 95q-9 9 -21.5 9t-21.5 -9l-149 -147h-312v-400h-150q-21 0 -35.5 14.5t-14.5 35.5v1000q0 20 14.5 35t35.5 15h250v-300h500v300h100zM700 1000h-100v200h100v-200zM1132 638l106 -106q7 -7 7 -17.5t-7 -17.5l-420 -421q-8 -7 -18 -7 t-18 7l-202 203q-8 7 -8 17.5t8 17.5l106 106q7 8 17.5 8t17.5 -8l79 -79l297 297q7 7 17.5 7t17.5 -7z" />
<glyph unicode="&#xe174;" d="M1100 1000v-269l-103 -103l-134 134q-15 15 -33.5 16.5t-34.5 -12.5l-266 -266h-329v-400h-150q-21 0 -35.5 14.5t-14.5 35.5v1000q0 20 14.5 35t35.5 15h250v-300h500v300h100zM700 1000h-100v200h100v-200zM1202 572l70 -70q15 -15 15 -35.5t-15 -35.5l-131 -131 l131 -131q15 -15 15 -35.5t-15 -35.5l-70 -70q-15 -15 -35.5 -15t-35.5 15l-131 131l-131 -131q-15 -15 -35.5 -15t-35.5 15l-70 70q-15 15 -15 35.5t15 35.5l131 131l-131 131q-15 15 -15 35.5t15 35.5l70 70q15 15 35.5 15t35.5 -15l131 -131l131 131q15 15 35.5 15 t35.5 -15z" />
<glyph unicode="&#xe175;" d="M1100 1000v-300h-350q-21 0 -35.5 -14.5t-14.5 -35.5v-150h-500v-400h-150q-21 0 -35.5 14.5t-14.5 35.5v1000q0 20 14.5 35t35.5 15h250v-300h500v300h100zM700 1000h-100v200h100v-200zM850 600h100q21 0 35.5 -14.5t14.5 -35.5v-250h150q21 0 25 -10.5t-10 -24.5 l-230 -230q-14 -14 -35 -14t-35 14l-230 230q-14 14 -10 24.5t25 10.5h150v250q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe176;" d="M1100 1000v-400l-165 165q-14 15 -35 15t-35 -15l-263 -265h-402v-400h-150q-21 0 -35.5 14.5t-14.5 35.5v1000q0 20 14.5 35t35.5 15h250v-300h500v300h100zM700 1000h-100v200h100v-200zM935 565l230 -229q14 -15 10 -25.5t-25 -10.5h-150v-250q0 -20 -14.5 -35 t-35.5 -15h-100q-21 0 -35.5 15t-14.5 35v250h-150q-21 0 -25 10.5t10 25.5l230 229q14 15 35 15t35 -15z" />
<glyph unicode="&#xe177;" d="M50 1100h1100q21 0 35.5 -14.5t14.5 -35.5v-150h-1200v150q0 21 14.5 35.5t35.5 14.5zM1200 800v-550q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v550h1200zM100 500v-200h400v200h-400z" />
<glyph unicode="&#xe178;" d="M935 1165l248 -230q14 -14 14 -35t-14 -35l-248 -230q-14 -14 -24.5 -10t-10.5 25v150h-400v200h400v150q0 21 10.5 25t24.5 -10zM200 800h-50q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h50v-200zM400 800h-100v200h100v-200zM18 435l247 230 q14 14 24.5 10t10.5 -25v-150h400v-200h-400v-150q0 -21 -10.5 -25t-24.5 10l-247 230q-15 14 -15 35t15 35zM900 300h-100v200h100v-200zM1000 500h51q20 0 34.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-34.5 -14.5h-51v200z" />
<glyph unicode="&#xe179;" d="M862 1073l276 116q25 18 43.5 8t18.5 -41v-1106q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v397q-4 1 -11 5t-24 17.5t-30 29t-24 42t-11 56.5v359q0 31 18.5 65t43.5 52zM550 1200q22 0 34.5 -12.5t14.5 -24.5l1 -13v-450q0 -28 -10.5 -59.5 t-25 -56t-29 -45t-25.5 -31.5l-10 -11v-447q0 -21 -14.5 -35.5t-35.5 -14.5h-200q-21 0 -35.5 14.5t-14.5 35.5v447q-4 4 -11 11.5t-24 30.5t-30 46t-24 55t-11 60v450q0 2 0.5 5.5t4 12t8.5 15t14.5 12t22.5 5.5q20 0 32.5 -12.5t14.5 -24.5l3 -13v-350h100v350v5.5t2.5 12 t7 15t15 12t25.5 5.5q23 0 35.5 -12.5t13.5 -24.5l1 -13v-350h100v350q0 2 0.5 5.5t3 12t7 15t15 12t24.5 5.5z" />
<glyph unicode="&#xe180;" d="M1200 1100v-56q-4 0 -11 -0.5t-24 -3t-30 -7.5t-24 -15t-11 -24v-888q0 -22 25 -34.5t50 -13.5l25 -2v-56h-400v56q75 0 87.5 6.5t12.5 43.5v394h-500v-394q0 -37 12.5 -43.5t87.5 -6.5v-56h-400v56q4 0 11 0.5t24 3t30 7.5t24 15t11 24v888q0 22 -25 34.5t-50 13.5 l-25 2v56h400v-56q-75 0 -87.5 -6.5t-12.5 -43.5v-394h500v394q0 37 -12.5 43.5t-87.5 6.5v56h400z" />
<glyph unicode="&#xe181;" d="M675 1000h375q21 0 35.5 -14.5t14.5 -35.5v-150h-105l-295 -98v98l-200 200h-400l100 100h375zM100 900h300q41 0 70.5 -29.5t29.5 -70.5v-500q0 -41 -29.5 -70.5t-70.5 -29.5h-300q-41 0 -70.5 29.5t-29.5 70.5v500q0 41 29.5 70.5t70.5 29.5zM100 800v-200h300v200 h-300zM1100 535l-400 -133v163l400 133v-163zM100 500v-200h300v200h-300zM1100 398v-248q0 -21 -14.5 -35.5t-35.5 -14.5h-375l-100 -100h-375l-100 100h400l200 200h105z" />
<glyph unicode="&#xe182;" d="M17 1007l162 162q17 17 40 14t37 -22l139 -194q14 -20 11 -44.5t-20 -41.5l-119 -118q102 -142 228 -268t267 -227l119 118q17 17 42.5 19t44.5 -12l192 -136q19 -14 22.5 -37.5t-13.5 -40.5l-163 -162q-3 -1 -9.5 -1t-29.5 2t-47.5 6t-62.5 14.5t-77.5 26.5t-90 42.5 t-101.5 60t-111 83t-119 108.5q-74 74 -133.5 150.5t-94.5 138.5t-60 119.5t-34.5 100t-15 74.5t-4.5 48z" />
<glyph unicode="&#xe183;" d="M600 1100q92 0 175 -10.5t141.5 -27t108.5 -36.5t81.5 -40t53.5 -37t31 -27l9 -10v-200q0 -21 -14.5 -33t-34.5 -9l-202 34q-20 3 -34.5 20t-14.5 38v146q-141 24 -300 24t-300 -24v-146q0 -21 -14.5 -38t-34.5 -20l-202 -34q-20 -3 -34.5 9t-14.5 33v200q3 4 9.5 10.5 t31 26t54 37.5t80.5 39.5t109 37.5t141 26.5t175 10.5zM600 795q56 0 97 -9.5t60 -23.5t30 -28t12 -24l1 -10v-50l365 -303q14 -15 24.5 -40t10.5 -45v-212q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v212q0 20 10.5 45t24.5 40l365 303v50 q0 4 1 10.5t12 23t30 29t60 22.5t97 10z" />
<glyph unicode="&#xe184;" d="M1100 700l-200 -200h-600l-200 200v500h200v-200h200v200h200v-200h200v200h200v-500zM250 400h700q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-12l137 -100h-950l137 100h-12q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM50 100h1100q21 0 35.5 -14.5 t14.5 -35.5v-50h-1200v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe185;" d="M700 1100h-100q-41 0 -70.5 -29.5t-29.5 -70.5v-1000h300v1000q0 41 -29.5 70.5t-70.5 29.5zM1100 800h-100q-41 0 -70.5 -29.5t-29.5 -70.5v-700h300v700q0 41 -29.5 70.5t-70.5 29.5zM400 0h-300v400q0 41 29.5 70.5t70.5 29.5h100q41 0 70.5 -29.5t29.5 -70.5v-400z " />
<glyph unicode="&#xe186;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM500 700h-200v-100h200v-300h-300v100h200v100h-200v300h300v-100zM900 700v-300l-100 -100h-200v500h200z M700 700v-300h100v300h-100z" />
<glyph unicode="&#xe187;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM500 300h-100v200h-100v-200h-100v500h100v-200h100v200h100v-500zM900 700v-300l-100 -100h-200v500h200z M700 700v-300h100v300h-100z" />
<glyph unicode="&#xe188;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM500 700h-200v-300h200v-100h-300v500h300v-100zM900 700h-200v-300h200v-100h-300v500h300v-100z" />
<glyph unicode="&#xe189;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM500 400l-300 150l300 150v-300zM900 550l-300 -150v300z" />
<glyph unicode="&#xe190;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM900 300h-700v500h700v-500zM800 700h-130q-38 0 -66.5 -43t-28.5 -108t27 -107t68 -42h130v300zM300 700v-300 h130q41 0 68 42t27 107t-28.5 108t-66.5 43h-130z" />
<glyph unicode="&#xe191;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM500 700h-200v-100h200v-300h-300v100h200v100h-200v300h300v-100zM900 300h-100v400h-100v100h200v-500z M700 300h-100v100h100v-100z" />
<glyph unicode="&#xe192;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM300 700h200v-400h-300v500h100v-100zM900 300h-100v400h-100v100h200v-500zM300 600v-200h100v200h-100z M700 300h-100v100h100v-100z" />
<glyph unicode="&#xe193;" d="M200 1100h700q124 0 212 -88t88 -212v-500q0 -124 -88 -212t-212 -88h-700q-124 0 -212 88t-88 212v500q0 124 88 212t212 88zM100 900v-700h900v700h-900zM500 500l-199 -200h-100v50l199 200v150h-200v100h300v-300zM900 300h-100v400h-100v100h200v-500zM701 300h-100 v100h100v-100z" />
<glyph unicode="&#xe194;" d="M600 1191q120 0 229.5 -47t188.5 -126t126 -188.5t47 -229.5t-47 -229.5t-126 -188.5t-188.5 -126t-229.5 -47t-229.5 47t-188.5 126t-126 188.5t-47 229.5t47 229.5t126 188.5t188.5 126t229.5 47zM600 1021q-114 0 -211 -56.5t-153.5 -153.5t-56.5 -211t56.5 -211 t153.5 -153.5t211 -56.5t211 56.5t153.5 153.5t56.5 211t-56.5 211t-153.5 153.5t-211 56.5zM800 700h-300v-200h300v-100h-300l-100 100v200l100 100h300v-100z" />
<glyph unicode="&#xe195;" d="M600 1191q120 0 229.5 -47t188.5 -126t126 -188.5t47 -229.5t-47 -229.5t-126 -188.5t-188.5 -126t-229.5 -47t-229.5 47t-188.5 126t-126 188.5t-47 229.5t47 229.5t126 188.5t188.5 126t229.5 47zM600 1021q-114 0 -211 -56.5t-153.5 -153.5t-56.5 -211t56.5 -211 t153.5 -153.5t211 -56.5t211 56.5t153.5 153.5t56.5 211t-56.5 211t-153.5 153.5t-211 56.5zM800 700v-100l-50 -50l100 -100v-50h-100l-100 100h-150v-100h-100v400h300zM500 700v-100h200v100h-200z" />
<glyph unicode="&#xe197;" d="M503 1089q110 0 200.5 -59.5t134.5 -156.5q44 14 90 14q120 0 205 -86.5t85 -207t-85 -207t-205 -86.5h-128v250q0 21 -14.5 35.5t-35.5 14.5h-300q-21 0 -35.5 -14.5t-14.5 -35.5v-250h-222q-80 0 -136 57.5t-56 136.5q0 69 43 122.5t108 67.5q-2 19 -2 37q0 100 49 185 t134 134t185 49zM525 500h150q10 0 17.5 -7.5t7.5 -17.5v-275h137q21 0 26 -11.5t-8 -27.5l-223 -244q-13 -16 -32 -16t-32 16l-223 244q-13 16 -8 27.5t26 11.5h137v275q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe198;" d="M502 1089q110 0 201 -59.5t135 -156.5q43 15 89 15q121 0 206 -86.5t86 -206.5q0 -99 -60 -181t-150 -110l-378 360q-13 16 -31.5 16t-31.5 -16l-381 -365h-9q-79 0 -135.5 57.5t-56.5 136.5q0 69 43 122.5t108 67.5q-2 19 -2 38q0 100 49 184.5t133.5 134t184.5 49.5z M632 467l223 -228q13 -16 8 -27.5t-26 -11.5h-137v-275q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v275h-137q-21 0 -26 11.5t8 27.5q199 204 223 228q19 19 31.5 19t32.5 -19z" />
<glyph unicode="&#xe199;" d="M700 100v100h400l-270 300h170l-270 300h170l-300 333l-300 -333h170l-270 -300h170l-270 -300h400v-100h-50q-21 0 -35.5 -14.5t-14.5 -35.5v-50h400v50q0 21 -14.5 35.5t-35.5 14.5h-50z" />
<glyph unicode="&#xe200;" d="M600 1179q94 0 167.5 -56.5t99.5 -145.5q89 -6 150.5 -71.5t61.5 -155.5q0 -61 -29.5 -112.5t-79.5 -82.5q9 -29 9 -55q0 -74 -52.5 -126.5t-126.5 -52.5q-55 0 -100 30v-251q21 0 35.5 -14.5t14.5 -35.5v-50h-300v50q0 21 14.5 35.5t35.5 14.5v251q-45 -30 -100 -30 q-74 0 -126.5 52.5t-52.5 126.5q0 18 4 38q-47 21 -75.5 65t-28.5 97q0 74 52.5 126.5t126.5 52.5q5 0 23 -2q0 2 -1 10t-1 13q0 116 81.5 197.5t197.5 81.5z" />
<glyph unicode="&#xe201;" d="M1010 1010q111 -111 150.5 -260.5t0 -299t-150.5 -260.5q-83 -83 -191.5 -126.5t-218.5 -43.5t-218.5 43.5t-191.5 126.5q-111 111 -150.5 260.5t0 299t150.5 260.5q83 83 191.5 126.5t218.5 43.5t218.5 -43.5t191.5 -126.5zM476 1065q-4 0 -8 -1q-121 -34 -209.5 -122.5 t-122.5 -209.5q-4 -12 2.5 -23t18.5 -14l36 -9q3 -1 7 -1q23 0 29 22q27 96 98 166q70 71 166 98q11 3 17.5 13.5t3.5 22.5l-9 35q-3 13 -14 19q-7 4 -15 4zM512 920q-4 0 -9 -2q-80 -24 -138.5 -82.5t-82.5 -138.5q-4 -13 2 -24t19 -14l34 -9q4 -1 8 -1q22 0 28 21 q18 58 58.5 98.5t97.5 58.5q12 3 18 13.5t3 21.5l-9 35q-3 12 -14 19q-7 4 -15 4zM719.5 719.5q-49.5 49.5 -119.5 49.5t-119.5 -49.5t-49.5 -119.5t49.5 -119.5t119.5 -49.5t119.5 49.5t49.5 119.5t-49.5 119.5zM855 551q-22 0 -28 -21q-18 -58 -58.5 -98.5t-98.5 -57.5 q-11 -4 -17 -14.5t-3 -21.5l9 -35q3 -12 14 -19q7 -4 15 -4q4 0 9 2q80 24 138.5 82.5t82.5 138.5q4 13 -2.5 24t-18.5 14l-34 9q-4 1 -8 1zM1000 515q-23 0 -29 -22q-27 -96 -98 -166q-70 -71 -166 -98q-11 -3 -17.5 -13.5t-3.5 -22.5l9 -35q3 -13 14 -19q7 -4 15 -4 q4 0 8 1q121 34 209.5 122.5t122.5 209.5q4 12 -2.5 23t-18.5 14l-36 9q-3 1 -7 1z" />
<glyph unicode="&#xe202;" d="M700 800h300v-380h-180v200h-340v-200h-380v755q0 10 7.5 17.5t17.5 7.5h575v-400zM1000 900h-200v200zM700 300h162l-212 -212l-212 212h162v200h100v-200zM520 0h-395q-10 0 -17.5 7.5t-7.5 17.5v395zM1000 220v-195q0 -10 -7.5 -17.5t-17.5 -7.5h-195z" />
<glyph unicode="&#xe203;" d="M700 800h300v-520l-350 350l-550 -550v1095q0 10 7.5 17.5t17.5 7.5h575v-400zM1000 900h-200v200zM862 200h-162v-200h-100v200h-162l212 212zM480 0h-355q-10 0 -17.5 7.5t-7.5 17.5v55h380v-80zM1000 80v-55q0 -10 -7.5 -17.5t-17.5 -7.5h-155v80h180z" />
<glyph unicode="&#xe204;" d="M1162 800h-162v-200h100l100 -100h-300v300h-162l212 212zM200 800h200q27 0 40 -2t29.5 -10.5t23.5 -30t7 -57.5h300v-100h-600l-200 -350v450h100q0 36 7 57.5t23.5 30t29.5 10.5t40 2zM800 400h240l-240 -400h-800l300 500h500v-100z" />
<glyph unicode="&#xe205;" d="M650 1100h100q21 0 35.5 -14.5t14.5 -35.5v-50h50q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-300q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h50v50q0 21 14.5 35.5t35.5 14.5zM1000 850v150q41 0 70.5 -29.5t29.5 -70.5v-800 q0 -41 -29.5 -70.5t-70.5 -29.5h-600q-1 0 -20 4l246 246l-326 326v324q0 41 29.5 70.5t70.5 29.5v-150q0 -62 44 -106t106 -44h300q62 0 106 44t44 106zM412 250l-212 -212v162h-200v100h200v162z" />
<glyph unicode="&#xe206;" d="M450 1100h100q21 0 35.5 -14.5t14.5 -35.5v-50h50q21 0 35.5 -14.5t14.5 -35.5v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-300q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h50v50q0 21 14.5 35.5t35.5 14.5zM800 850v150q41 0 70.5 -29.5t29.5 -70.5v-500 h-200v-300h200q0 -36 -7 -57.5t-23.5 -30t-29.5 -10.5t-40 -2h-600q-41 0 -70.5 29.5t-29.5 70.5v800q0 41 29.5 70.5t70.5 29.5v-150q0 -62 44 -106t106 -44h300q62 0 106 44t44 106zM1212 250l-212 -212v162h-200v100h200v162z" />
<glyph unicode="&#xe209;" d="M658 1197l637 -1104q23 -38 7 -65.5t-60 -27.5h-1276q-44 0 -60 27.5t7 65.5l637 1104q22 39 54 39t54 -39zM704 800h-208q-20 0 -32 -14.5t-8 -34.5l58 -302q4 -20 21.5 -34.5t37.5 -14.5h54q20 0 37.5 14.5t21.5 34.5l58 302q4 20 -8 34.5t-32 14.5zM500 300v-100h200 v100h-200z" />
<glyph unicode="&#xe210;" d="M425 1100h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5zM425 800h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5 t17.5 7.5zM825 800h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5zM25 500h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150 q0 10 7.5 17.5t17.5 7.5zM425 500h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5zM825 500h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5 v150q0 10 7.5 17.5t17.5 7.5zM25 200h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5zM425 200h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5 t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5zM825 200h250q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-250q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe211;" d="M700 1200h100v-200h-100v-100h350q62 0 86.5 -39.5t-3.5 -94.5l-66 -132q-41 -83 -81 -134h-772q-40 51 -81 134l-66 132q-28 55 -3.5 94.5t86.5 39.5h350v100h-100v200h100v100h200v-100zM250 400h700q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-12l137 -100 h-950l138 100h-13q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM50 100h1100q21 0 35.5 -14.5t14.5 -35.5v-50h-1200v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe212;" d="M600 1300q40 0 68.5 -29.5t28.5 -70.5h-194q0 41 28.5 70.5t68.5 29.5zM443 1100h314q18 -37 18 -75q0 -8 -3 -25h328q41 0 44.5 -16.5t-30.5 -38.5l-175 -145h-678l-178 145q-34 22 -29 38.5t46 16.5h328q-3 17 -3 25q0 38 18 75zM250 700h700q21 0 35.5 -14.5 t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-150v-200l275 -200h-950l275 200v200h-150q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM50 100h1100q21 0 35.5 -14.5t14.5 -35.5v-50h-1200v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe213;" d="M600 1181q75 0 128 -53t53 -128t-53 -128t-128 -53t-128 53t-53 128t53 128t128 53zM602 798h46q34 0 55.5 -28.5t21.5 -86.5q0 -76 39 -183h-324q39 107 39 183q0 58 21.5 86.5t56.5 28.5h45zM250 400h700q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-13 l138 -100h-950l137 100h-12q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM50 100h1100q21 0 35.5 -14.5t14.5 -35.5v-50h-1200v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe214;" d="M600 1300q47 0 92.5 -53.5t71 -123t25.5 -123.5q0 -78 -55.5 -133.5t-133.5 -55.5t-133.5 55.5t-55.5 133.5q0 62 34 143l144 -143l111 111l-163 163q34 26 63 26zM602 798h46q34 0 55.5 -28.5t21.5 -86.5q0 -76 39 -183h-324q39 107 39 183q0 58 21.5 86.5t56.5 28.5h45 zM250 400h700q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-13l138 -100h-950l137 100h-12q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM50 100h1100q21 0 35.5 -14.5t14.5 -35.5v-50h-1200v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe215;" d="M600 1200l300 -161v-139h-300q0 -57 18.5 -108t50 -91.5t63 -72t70 -67.5t57.5 -61h-530q-60 83 -90.5 177.5t-30.5 178.5t33 164.5t87.5 139.5t126 96.5t145.5 41.5v-98zM250 400h700q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-13l138 -100h-950l137 100 h-12q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5zM50 100h1100q21 0 35.5 -14.5t14.5 -35.5v-50h-1200v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe216;" d="M600 1300q41 0 70.5 -29.5t29.5 -70.5v-78q46 -26 73 -72t27 -100v-50h-400v50q0 54 27 100t73 72v78q0 41 29.5 70.5t70.5 29.5zM400 800h400q54 0 100 -27t72 -73h-172v-100h200v-100h-200v-100h200v-100h-200v-100h200q0 -83 -58.5 -141.5t-141.5 -58.5h-400 q-83 0 -141.5 58.5t-58.5 141.5v400q0 83 58.5 141.5t141.5 58.5z" />
<glyph unicode="&#xe218;" d="M150 1100h900q21 0 35.5 -14.5t14.5 -35.5v-500q0 -21 -14.5 -35.5t-35.5 -14.5h-900q-21 0 -35.5 14.5t-14.5 35.5v500q0 21 14.5 35.5t35.5 14.5zM125 400h950q10 0 17.5 -7.5t7.5 -17.5v-50q0 -10 -7.5 -17.5t-17.5 -7.5h-283l224 -224q13 -13 13 -31.5t-13 -32 t-31.5 -13.5t-31.5 13l-88 88h-524l-87 -88q-13 -13 -32 -13t-32 13.5t-13 32t13 31.5l224 224h-289q-10 0 -17.5 7.5t-7.5 17.5v50q0 10 7.5 17.5t17.5 7.5zM541 300l-100 -100h324l-100 100h-124z" />
<glyph unicode="&#xe219;" d="M200 1100h800q83 0 141.5 -58.5t58.5 -141.5v-200h-100q0 41 -29.5 70.5t-70.5 29.5h-250q-41 0 -70.5 -29.5t-29.5 -70.5h-100q0 41 -29.5 70.5t-70.5 29.5h-250q-41 0 -70.5 -29.5t-29.5 -70.5h-100v200q0 83 58.5 141.5t141.5 58.5zM100 600h1000q41 0 70.5 -29.5 t29.5 -70.5v-300h-1200v300q0 41 29.5 70.5t70.5 29.5zM300 100v-50q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v50h200zM1100 100v-50q0 -21 -14.5 -35.5t-35.5 -14.5h-100q-21 0 -35.5 14.5t-14.5 35.5v50h200z" />
<glyph unicode="&#xe221;" d="M480 1165l682 -683q31 -31 31 -75.5t-31 -75.5l-131 -131h-481l-517 518q-32 31 -32 75.5t32 75.5l295 296q31 31 75.5 31t76.5 -31zM108 794l342 -342l303 304l-341 341zM250 100h800q21 0 35.5 -14.5t14.5 -35.5v-50h-900v50q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe223;" d="M1057 647l-189 506q-8 19 -27.5 33t-40.5 14h-400q-21 0 -40.5 -14t-27.5 -33l-189 -506q-8 -19 1.5 -33t30.5 -14h625v-150q0 -21 14.5 -35.5t35.5 -14.5t35.5 14.5t14.5 35.5v150h125q21 0 30.5 14t1.5 33zM897 0h-595v50q0 21 14.5 35.5t35.5 14.5h50v50 q0 21 14.5 35.5t35.5 14.5h48v300h200v-300h47q21 0 35.5 -14.5t14.5 -35.5v-50h50q21 0 35.5 -14.5t14.5 -35.5v-50z" />
<glyph unicode="&#xe224;" d="M900 800h300v-575q0 -10 -7.5 -17.5t-17.5 -7.5h-375v591l-300 300v84q0 10 7.5 17.5t17.5 7.5h375v-400zM1200 900h-200v200zM400 600h300v-575q0 -10 -7.5 -17.5t-17.5 -7.5h-650q-10 0 -17.5 7.5t-7.5 17.5v950q0 10 7.5 17.5t17.5 7.5h375v-400zM700 700h-200v200z " />
<glyph unicode="&#xe225;" d="M484 1095h195q75 0 146 -32.5t124 -86t89.5 -122.5t48.5 -142q18 -14 35 -20q31 -10 64.5 6.5t43.5 48.5q10 34 -15 71q-19 27 -9 43q5 8 12.5 11t19 -1t23.5 -16q41 -44 39 -105q-3 -63 -46 -106.5t-104 -43.5h-62q-7 -55 -35 -117t-56 -100l-39 -234q-3 -20 -20 -34.5 t-38 -14.5h-100q-21 0 -33 14.5t-9 34.5l12 70q-49 -14 -91 -14h-195q-24 0 -65 8l-11 -64q-3 -20 -20 -34.5t-38 -14.5h-100q-21 0 -33 14.5t-9 34.5l26 157q-84 74 -128 175l-159 53q-19 7 -33 26t-14 40v50q0 21 14.5 35.5t35.5 14.5h124q11 87 56 166l-111 95 q-16 14 -12.5 23.5t24.5 9.5h203q116 101 250 101zM675 1000h-250q-10 0 -17.5 -7.5t-7.5 -17.5v-50q0 -10 7.5 -17.5t17.5 -7.5h250q10 0 17.5 7.5t7.5 17.5v50q0 10 -7.5 17.5t-17.5 7.5z" />
<glyph unicode="&#xe226;" d="M641 900l423 247q19 8 42 2.5t37 -21.5l32 -38q14 -15 12.5 -36t-17.5 -34l-139 -120h-390zM50 1100h106q67 0 103 -17t66 -71l102 -212h823q21 0 35.5 -14.5t14.5 -35.5v-50q0 -21 -14 -40t-33 -26l-737 -132q-23 -4 -40 6t-26 25q-42 67 -100 67h-300q-62 0 -106 44 t-44 106v200q0 62 44 106t106 44zM173 928h-80q-19 0 -28 -14t-9 -35v-56q0 -51 42 -51h134q16 0 21.5 8t5.5 24q0 11 -16 45t-27 51q-18 28 -43 28zM550 727q-32 0 -54.5 -22.5t-22.5 -54.5t22.5 -54.5t54.5 -22.5t54.5 22.5t22.5 54.5t-22.5 54.5t-54.5 22.5zM130 389 l152 130q18 19 34 24t31 -3.5t24.5 -17.5t25.5 -28q28 -35 50.5 -51t48.5 -13l63 5l48 -179q13 -61 -3.5 -97.5t-67.5 -79.5l-80 -69q-47 -40 -109 -35.5t-103 51.5l-130 151q-40 47 -35.5 109.5t51.5 102.5zM380 377l-102 -88q-31 -27 2 -65l37 -43q13 -15 27.5 -19.5 t31.5 6.5l61 53q19 16 14 49q-2 20 -12 56t-17 45q-11 12 -19 14t-23 -8z" />
<glyph unicode="&#xe227;" d="M625 1200h150q10 0 17.5 -7.5t7.5 -17.5v-109q79 -33 131 -87.5t53 -128.5q1 -46 -15 -84.5t-39 -61t-46 -38t-39 -21.5l-17 -6q6 0 15 -1.5t35 -9t50 -17.5t53 -30t50 -45t35.5 -64t14.5 -84q0 -59 -11.5 -105.5t-28.5 -76.5t-44 -51t-49.5 -31.5t-54.5 -16t-49.5 -6.5 t-43.5 -1v-75q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v75h-100v-75q0 -10 -7.5 -17.5t-17.5 -7.5h-150q-10 0 -17.5 7.5t-7.5 17.5v75h-175q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5h75v600h-75q-10 0 -17.5 7.5t-7.5 17.5v150 q0 10 7.5 17.5t17.5 7.5h175v75q0 10 7.5 17.5t17.5 7.5h150q10 0 17.5 -7.5t7.5 -17.5v-75h100v75q0 10 7.5 17.5t17.5 7.5zM400 900v-200h263q28 0 48.5 10.5t30 25t15 29t5.5 25.5l1 10q0 4 -0.5 11t-6 24t-15 30t-30 24t-48.5 11h-263zM400 500v-200h363q28 0 48.5 10.5 t30 25t15 29t5.5 25.5l1 10q0 4 -0.5 11t-6 24t-15 30t-30 24t-48.5 11h-363z" />
<glyph unicode="&#xe230;" d="M212 1198h780q86 0 147 -61t61 -147v-416q0 -51 -18 -142.5t-36 -157.5l-18 -66q-29 -87 -93.5 -146.5t-146.5 -59.5h-572q-82 0 -147 59t-93 147q-8 28 -20 73t-32 143.5t-20 149.5v416q0 86 61 147t147 61zM600 1045q-70 0 -132.5 -11.5t-105.5 -30.5t-78.5 -41.5 t-57 -45t-36 -41t-20.5 -30.5l-6 -12l156 -243h560l156 243q-2 5 -6 12.5t-20 29.5t-36.5 42t-57 44.5t-79 42t-105 29.5t-132.5 12zM762 703h-157l195 261z" />
<glyph unicode="&#xe231;" d="M475 1300h150q103 0 189 -86t86 -189v-500q0 -41 -42 -83t-83 -42h-450q-41 0 -83 42t-42 83v500q0 103 86 189t189 86zM700 300v-225q0 -21 -27 -48t-48 -27h-150q-21 0 -48 27t-27 48v225h300z" />
<glyph unicode="&#xe232;" d="M475 1300h96q0 -150 89.5 -239.5t239.5 -89.5v-446q0 -41 -42 -83t-83 -42h-450q-41 0 -83 42t-42 83v500q0 103 86 189t189 86zM700 300v-225q0 -21 -27 -48t-48 -27h-150q-21 0 -48 27t-27 48v225h300z" />
<glyph unicode="&#xe233;" d="M1294 767l-638 -283l-378 170l-78 -60v-224l100 -150v-199l-150 148l-150 -149v200l100 150v250q0 4 -0.5 10.5t0 9.5t1 8t3 8t6.5 6l47 40l-147 65l642 283zM1000 380l-350 -166l-350 166v147l350 -165l350 165v-147z" />
<glyph unicode="&#xe234;" d="M250 800q62 0 106 -44t44 -106t-44 -106t-106 -44t-106 44t-44 106t44 106t106 44zM650 800q62 0 106 -44t44 -106t-44 -106t-106 -44t-106 44t-44 106t44 106t106 44zM1050 800q62 0 106 -44t44 -106t-44 -106t-106 -44t-106 44t-44 106t44 106t106 44z" />
<glyph unicode="&#xe235;" d="M550 1100q62 0 106 -44t44 -106t-44 -106t-106 -44t-106 44t-44 106t44 106t106 44zM550 700q62 0 106 -44t44 -106t-44 -106t-106 -44t-106 44t-44 106t44 106t106 44zM550 300q62 0 106 -44t44 -106t-44 -106t-106 -44t-106 44t-44 106t44 106t106 44z" />
<glyph unicode="&#xe236;" d="M125 1100h950q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-950q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5zM125 700h950q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-950q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5 t17.5 7.5zM125 300h950q10 0 17.5 -7.5t7.5 -17.5v-150q0 -10 -7.5 -17.5t-17.5 -7.5h-950q-10 0 -17.5 7.5t-7.5 17.5v150q0 10 7.5 17.5t17.5 7.5z" />
<glyph unicode="&#xe237;" d="M350 1200h500q162 0 256 -93.5t94 -256.5v-500q0 -165 -93.5 -257.5t-256.5 -92.5h-500q-165 0 -257.5 92.5t-92.5 257.5v500q0 165 92.5 257.5t257.5 92.5zM900 1000h-600q-41 0 -70.5 -29.5t-29.5 -70.5v-600q0 -41 29.5 -70.5t70.5 -29.5h600q41 0 70.5 29.5 t29.5 70.5v600q0 41 -29.5 70.5t-70.5 29.5zM350 900h500q21 0 35.5 -14.5t14.5 -35.5v-300q0 -21 -14.5 -35.5t-35.5 -14.5h-500q-21 0 -35.5 14.5t-14.5 35.5v300q0 21 14.5 35.5t35.5 14.5zM400 800v-200h400v200h-400z" />
<glyph unicode="&#xe238;" d="M150 1100h1000q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-50v-200h50q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-50v-200h50q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5t-35.5 -14.5h-50v-200h50q21 0 35.5 -14.5t14.5 -35.5t-14.5 -35.5 t-35.5 -14.5h-1000q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5h50v200h-50q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5h50v200h-50q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5h50v200h-50q-21 0 -35.5 14.5t-14.5 35.5t14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe239;" d="M650 1187q87 -67 118.5 -156t0 -178t-118.5 -155q-87 66 -118.5 155t0 178t118.5 156zM300 800q124 0 212 -88t88 -212q-124 0 -212 88t-88 212zM1000 800q0 -124 -88 -212t-212 -88q0 124 88 212t212 88zM300 500q124 0 212 -88t88 -212q-124 0 -212 88t-88 212z M1000 500q0 -124 -88 -212t-212 -88q0 124 88 212t212 88zM700 199v-144q0 -21 -14.5 -35.5t-35.5 -14.5t-35.5 14.5t-14.5 35.5v142q40 -4 43 -4q17 0 57 6z" />
<glyph unicode="&#xe240;" d="M745 878l69 19q25 6 45 -12l298 -295q11 -11 15 -26.5t-2 -30.5q-5 -14 -18 -23.5t-28 -9.5h-8q1 0 1 -13q0 -29 -2 -56t-8.5 -62t-20 -63t-33 -53t-51 -39t-72.5 -14h-146q-184 0 -184 288q0 24 10 47q-20 4 -62 4t-63 -4q11 -24 11 -47q0 -288 -184 -288h-142 q-48 0 -84.5 21t-56 51t-32 71.5t-16 75t-3.5 68.5q0 13 2 13h-7q-15 0 -27.5 9.5t-18.5 23.5q-6 15 -2 30.5t15 25.5l298 296q20 18 46 11l76 -19q20 -5 30.5 -22.5t5.5 -37.5t-22.5 -31t-37.5 -5l-51 12l-182 -193h891l-182 193l-44 -12q-20 -5 -37.5 6t-22.5 31t6 37.5 t31 22.5z" />
<glyph unicode="&#xe241;" d="M1200 900h-50q0 21 -4 37t-9.5 26.5t-18 17.5t-22 11t-28.5 5.5t-31 2t-37 0.5h-200v-850q0 -22 25 -34.5t50 -13.5l25 -2v-100h-400v100q4 0 11 0.5t24 3t30 7t24 15t11 24.5v850h-200q-25 0 -37 -0.5t-31 -2t-28.5 -5.5t-22 -11t-18 -17.5t-9.5 -26.5t-4 -37h-50v300 h1000v-300zM500 450h-25q0 15 -4 24.5t-9 14.5t-17 7.5t-20 3t-25 0.5h-100v-425q0 -11 12.5 -17.5t25.5 -7.5h12v-50h-200v50q50 0 50 25v425h-100q-17 0 -25 -0.5t-20 -3t-17 -7.5t-9 -14.5t-4 -24.5h-25v150h500v-150z" />
<glyph unicode="&#xe242;" d="M1000 300v50q-25 0 -55 32q-14 14 -25 31t-16 27l-4 11l-289 747h-69l-300 -754q-18 -35 -39 -56q-9 -9 -24.5 -18.5t-26.5 -14.5l-11 -5v-50h273v50q-49 0 -78.5 21.5t-11.5 67.5l69 176h293l61 -166q13 -34 -3.5 -66.5t-55.5 -32.5v-50h312zM412 691l134 342l121 -342 h-255zM1100 150v-100q0 -21 -14.5 -35.5t-35.5 -14.5h-1000q-21 0 -35.5 14.5t-14.5 35.5v100q0 21 14.5 35.5t35.5 14.5h1000q21 0 35.5 -14.5t14.5 -35.5z" />
<glyph unicode="&#xe243;" d="M50 1200h1100q21 0 35.5 -14.5t14.5 -35.5v-1100q0 -21 -14.5 -35.5t-35.5 -14.5h-1100q-21 0 -35.5 14.5t-14.5 35.5v1100q0 21 14.5 35.5t35.5 14.5zM611 1118h-70q-13 0 -18 -12l-299 -753q-17 -32 -35 -51q-18 -18 -56 -34q-12 -5 -12 -18v-50q0 -8 5.5 -14t14.5 -6 h273q8 0 14 6t6 14v50q0 8 -6 14t-14 6q-55 0 -71 23q-10 14 0 39l63 163h266l57 -153q11 -31 -6 -55q-12 -17 -36 -17q-8 0 -14 -6t-6 -14v-50q0 -8 6 -14t14 -6h313q8 0 14 6t6 14v50q0 7 -5.5 13t-13.5 7q-17 0 -42 25q-25 27 -40 63h-1l-288 748q-5 12 -19 12zM639 611 h-197l103 264z" />
<glyph unicode="&#xe244;" d="M1200 1100h-1200v100h1200v-100zM50 1000h400q21 0 35.5 -14.5t14.5 -35.5v-900q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v900q0 21 14.5 35.5t35.5 14.5zM650 1000h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400 q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM700 900v-300h300v300h-300z" />
<glyph unicode="&#xe245;" d="M50 1200h400q21 0 35.5 -14.5t14.5 -35.5v-900q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v900q0 21 14.5 35.5t35.5 14.5zM650 700h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v400 q0 21 14.5 35.5t35.5 14.5zM700 600v-300h300v300h-300zM1200 0h-1200v100h1200v-100z" />
<glyph unicode="&#xe246;" d="M50 1000h400q21 0 35.5 -14.5t14.5 -35.5v-350h100v150q0 21 14.5 35.5t35.5 14.5h400q21 0 35.5 -14.5t14.5 -35.5v-150h100v-100h-100v-150q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v150h-100v-350q0 -21 -14.5 -35.5t-35.5 -14.5h-400 q-21 0 -35.5 14.5t-14.5 35.5v800q0 21 14.5 35.5t35.5 14.5zM700 700v-300h300v300h-300z" />
<glyph unicode="&#xe247;" d="M100 0h-100v1200h100v-1200zM250 1100h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM300 1000v-300h300v300h-300zM250 500h900q21 0 35.5 -14.5t14.5 -35.5v-400 q0 -21 -14.5 -35.5t-35.5 -14.5h-900q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe248;" d="M600 1100h150q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-150v-100h450q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-900q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5h350v100h-150q-21 0 -35.5 14.5 t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5h150v100h100v-100zM400 1000v-300h300v300h-300z" />
<glyph unicode="&#xe249;" d="M1200 0h-100v1200h100v-1200zM550 1100h400q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-400q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM600 1000v-300h300v300h-300zM50 500h900q21 0 35.5 -14.5t14.5 -35.5v-400 q0 -21 -14.5 -35.5t-35.5 -14.5h-900q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5z" />
<glyph unicode="&#xe250;" d="M865 565l-494 -494q-23 -23 -41 -23q-14 0 -22 13.5t-8 38.5v1000q0 25 8 38.5t22 13.5q18 0 41 -23l494 -494q14 -14 14 -35t-14 -35z" />
<glyph unicode="&#xe251;" d="M335 635l494 494q29 29 50 20.5t21 -49.5v-1000q0 -41 -21 -49.5t-50 20.5l-494 494q-14 14 -14 35t14 35z" />
<glyph unicode="&#xe252;" d="M100 900h1000q41 0 49.5 -21t-20.5 -50l-494 -494q-14 -14 -35 -14t-35 14l-494 494q-29 29 -20.5 50t49.5 21z" />
<glyph unicode="&#xe253;" d="M635 865l494 -494q29 -29 20.5 -50t-49.5 -21h-1000q-41 0 -49.5 21t20.5 50l494 494q14 14 35 14t35 -14z" />
<glyph unicode="&#xe254;" d="M700 741v-182l-692 -323v221l413 193l-413 193v221zM1200 0h-800v200h800v-200z" />
<glyph unicode="&#xe255;" d="M1200 900h-200v-100h200v-100h-300v300h200v100h-200v100h300v-300zM0 700h50q0 21 4 37t9.5 26.5t18 17.5t22 11t28.5 5.5t31 2t37 0.5h100v-550q0 -22 -25 -34.5t-50 -13.5l-25 -2v-100h400v100q-4 0 -11 0.5t-24 3t-30 7t-24 15t-11 24.5v550h100q25 0 37 -0.5t31 -2 t28.5 -5.5t22 -11t18 -17.5t9.5 -26.5t4 -37h50v300h-800v-300z" />
<glyph unicode="&#xe256;" d="M800 700h-50q0 21 -4 37t-9.5 26.5t-18 17.5t-22 11t-28.5 5.5t-31 2t-37 0.5h-100v-550q0 -22 25 -34.5t50 -14.5l25 -1v-100h-400v100q4 0 11 0.5t24 3t30 7t24 15t11 24.5v550h-100q-25 0 -37 -0.5t-31 -2t-28.5 -5.5t-22 -11t-18 -17.5t-9.5 -26.5t-4 -37h-50v300 h800v-300zM1100 200h-200v-100h200v-100h-300v300h200v100h-200v100h300v-300z" />
<glyph unicode="&#xe257;" d="M701 1098h160q16 0 21 -11t-7 -23l-464 -464l464 -464q12 -12 7 -23t-21 -11h-160q-13 0 -23 9l-471 471q-7 8 -7 18t7 18l471 471q10 9 23 9z" />
<glyph unicode="&#xe258;" d="M339 1098h160q13 0 23 -9l471 -471q7 -8 7 -18t-7 -18l-471 -471q-10 -9 -23 -9h-160q-16 0 -21 11t7 23l464 464l-464 464q-12 12 -7 23t21 11z" />
<glyph unicode="&#xe259;" d="M1087 882q11 -5 11 -21v-160q0 -13 -9 -23l-471 -471q-8 -7 -18 -7t-18 7l-471 471q-9 10 -9 23v160q0 16 11 21t23 -7l464 -464l464 464q12 12 23 7z" />
<glyph unicode="&#xe260;" d="M618 993l471 -471q9 -10 9 -23v-160q0 -16 -11 -21t-23 7l-464 464l-464 -464q-12 -12 -23 -7t-11 21v160q0 13 9 23l471 471q8 7 18 7t18 -7z" />
<glyph unicode="&#xf8ff;" d="M1000 1200q0 -124 -88 -212t-212 -88q0 124 88 212t212 88zM450 1000h100q21 0 40 -14t26 -33l79 -194q5 1 16 3q34 6 54 9.5t60 7t65.5 1t61 -10t56.5 -23t42.5 -42t29 -64t5 -92t-19.5 -121.5q-1 -7 -3 -19.5t-11 -50t-20.5 -73t-32.5 -81.5t-46.5 -83t-64 -70 t-82.5 -50q-13 -5 -42 -5t-65.5 2.5t-47.5 2.5q-14 0 -49.5 -3.5t-63 -3.5t-43.5 7q-57 25 -104.5 78.5t-75 111.5t-46.5 112t-26 90l-7 35q-15 63 -18 115t4.5 88.5t26 64t39.5 43.5t52 25.5t58.5 13t62.5 2t59.5 -4.5t55.5 -8l-147 192q-12 18 -5.5 30t27.5 12z" />
<glyph unicode="&#x1f511;" d="M250 1200h600q21 0 35.5 -14.5t14.5 -35.5v-400q0 -21 -14.5 -35.5t-35.5 -14.5h-150v-500l-255 -178q-19 -9 -32 -1t-13 29v650h-150q-21 0 -35.5 14.5t-14.5 35.5v400q0 21 14.5 35.5t35.5 14.5zM400 1100v-100h300v100h-300z" />
<glyph unicode="&#x1f6aa;" d="M250 1200h750q39 0 69.5 -40.5t30.5 -84.5v-933l-700 -117v950l600 125h-700v-1000h-100v1025q0 23 15.5 49t34.5 26zM500 525v-100l100 20v100z" />
</font>
</defs></svg> ) format('svg')}.glyphicon{position:relative;top:1px;display:inline-block;font-family:'Glyphicons Halflings';font-style:normal;font-weight:400;line-height:1;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale}.glyphicon-asterisk:before{content:"\2a"}.glyphicon-plus:before{content:"\2b"}.glyphicon-eur:before,.glyphicon-euro:before{content:"\20ac"}.glyphicon-minus:before{content:"\2212"}.glyphicon-cloud:before{content:"\2601"}.glyphicon-envelope:before{content:"\2709"}.glyphicon-pencil:before{content:"\270f"}.glyphicon-glass:before{content:"\e001"}.glyphicon-music:before{content:"\e002"}.glyphicon-search:before{content:"\e003"}.glyphicon-heart:before{content:"\e005"}.glyphicon-star:before{content:"\e006"}.glyphicon-star-empty:before{content:"\e007"}.glyphicon-user:before{content:"\e008"}.glyphicon-film:before{content:"\e009"}.glyphicon-th-large:before{content:"\e010"}.glyphicon-th:before{content:"\e011"}.glyphicon-th-list:before{content:"\e012"}.glyphicon-ok:before{content:"\e013"}.glyphicon-remove:before{content:"\e014"}.glyphicon-zoom-in:before{content:"\e015"}.glyphicon-zoom-out:before{content:"\e016"}.glyphicon-off:before{content:"\e017"}.glyphicon-signal:before{content:"\e018"}.glyphicon-cog:before{content:"\e019"}.glyphicon-trash:before{content:"\e020"}.glyphicon-home:before{content:"\e021"}.glyphicon-file:before{content:"\e022"}.glyphicon-time:before{content:"\e023"}.glyphicon-road:before{content:"\e024"}.glyphicon-download-alt:before{content:"\e025"}.glyphicon-download:before{content:"\e026"}.glyphicon-upload:before{content:"\e027"}.glyphicon-inbox:before{content:"\e028"}.glyphicon-play-circle:before{content:"\e029"}.glyphicon-repeat:before{content:"\e030"}.glyphicon-refresh:before{content:"\e031"}.glyphicon-list-alt:before{content:"\e032"}.glyphicon-lock:before{content:"\e033"}.glyphicon-flag:before{content:"\e034"}.glyphicon-headphones:before{content:"\e035"}.glyphicon-volume-off:before{content:"\e036"}.glyphicon-volume-down:before{content:"\e037"}.glyphicon-volume-up:before{content:"\e038"}.glyphicon-qrcode:before{content:"\e039"}.glyphicon-barcode:before{content:"\e040"}.glyphicon-tag:before{content:"\e041"}.glyphicon-tags:before{content:"\e042"}.glyphicon-book:before{content:"\e043"}.glyphicon-bookmark:before{content:"\e044"}.glyphicon-print:before{content:"\e045"}.glyphicon-camera:before{content:"\e046"}.glyphicon-font:before{content:"\e047"}.glyphicon-bold:before{content:"\e048"}.glyphicon-italic:before{content:"\e049"}.glyphicon-text-height:before{content:"\e050"}.glyphicon-text-width:before{content:"\e051"}.glyphicon-align-left:before{content:"\e052"}.glyphicon-align-center:before{content:"\e053"}.glyphicon-align-right:before{content:"\e054"}.glyphicon-align-justify:before{content:"\e055"}.glyphicon-list:before{content:"\e056"}.glyphicon-indent-left:before{content:"\e057"}.glyphicon-indent-right:before{content:"\e058"}.glyphicon-facetime-video:before{content:"\e059"}.glyphicon-picture:before{content:"\e060"}.glyphicon-map-marker:before{content:"\e062"}.glyphicon-adjust:before{content:"\e063"}.glyphicon-tint:before{content:"\e064"}.glyphicon-edit:before{content:"\e065"}.glyphicon-share:before{content:"\e066"}.glyphicon-check:before{content:"\e067"}.glyphicon-move:before{content:"\e068"}.glyphicon-step-backward:before{content:"\e069"}.glyphicon-fast-backward:before{content:"\e070"}.glyphicon-backward:before{content:"\e071"}.glyphicon-play:before{content:"\e072"}.glyphicon-pause:before{content:"\e073"}.glyphicon-stop:before{content:"\e074"}.glyphicon-forward:before{content:"\e075"}.glyphicon-fast-forward:before{content:"\e076"}.glyphicon-step-forward:before{content:"\e077"}.glyphicon-eject:before{content:"\e078"}.glyphicon-chevron-left:before{content:"\e079"}.glyphicon-chevron-right:before{content:"\e080"}.glyphicon-plus-sign:before{content:"\e081"}.glyphicon-minus-sign:before{content:"\e082"}.glyphicon-remove-sign:before{content:"\e083"}.glyphicon-ok-sign:before{content:"\e084"}.glyphicon-question-sign:before{content:"\e085"}.glyphicon-info-sign:before{content:"\e086"}.glyphicon-screenshot:before{content:"\e087"}.glyphicon-remove-circle:before{content:"\e088"}.glyphicon-ok-circle:before{content:"\e089"}.glyphicon-ban-circle:before{content:"\e090"}.glyphicon-arrow-left:before{content:"\e091"}.glyphicon-arrow-right:before{content:"\e092"}.glyphicon-arrow-up:before{content:"\e093"}.glyphicon-arrow-down:before{content:"\e094"}.glyphicon-share-alt:before{content:"\e095"}.glyphicon-resize-full:before{content:"\e096"}.glyphicon-resize-small:before{content:"\e097"}.glyphicon-exclamation-sign:before{content:"\e101"}.glyphicon-gift:before{content:"\e102"}.glyphicon-leaf:before{content:"\e103"}.glyphicon-fire:before{content:"\e104"}.glyphicon-eye-open:before{content:"\e105"}.glyphicon-eye-close:before{content:"\e106"}.glyphicon-warning-sign:before{content:"\e107"}.glyphicon-plane:before{content:"\e108"}.glyphicon-calendar:before{content:"\e109"}.glyphicon-random:before{content:"\e110"}.glyphicon-comment:before{content:"\e111"}.glyphicon-magnet:before{content:"\e112"}.glyphicon-chevron-up:before{content:"\e113"}.glyphicon-chevron-down:before{content:"\e114"}.glyphicon-retweet:before{content:"\e115"}.glyphicon-shopping-cart:before{content:"\e116"}.glyphicon-folder-close:before{content:"\e117"}.glyphicon-folder-open:before{content:"\e118"}.glyphicon-resize-vertical:before{content:"\e119"}.glyphicon-resize-horizontal:before{content:"\e120"}.glyphicon-hdd:before{content:"\e121"}.glyphicon-bullhorn:before{content:"\e122"}.glyphicon-bell:before{content:"\e123"}.glyphicon-certificate:before{content:"\e124"}.glyphicon-thumbs-up:before{content:"\e125"}.glyphicon-thumbs-down:before{content:"\e126"}.glyphicon-hand-right:before{content:"\e127"}.glyphicon-hand-left:before{content:"\e128"}.glyphicon-hand-up:before{content:"\e129"}.glyphicon-hand-down:before{content:"\e130"}.glyphicon-circle-arrow-right:before{content:"\e131"}.glyphicon-circle-arrow-left:before{content:"\e132"}.glyphicon-circle-arrow-up:before{content:"\e133"}.glyphicon-circle-arrow-down:before{content:"\e134"}.glyphicon-globe:before{content:"\e135"}.glyphicon-wrench:before{content:"\e136"}.glyphicon-tasks:before{content:"\e137"}.glyphicon-filter:before{content:"\e138"}.glyphicon-briefcase:before{content:"\e139"}.glyphicon-fullscreen:before{content:"\e140"}.glyphicon-dashboard:before{content:"\e141"}.glyphicon-paperclip:before{content:"\e142"}.glyphicon-heart-empty:before{content:"\e143"}.glyphicon-link:before{content:"\e144"}.glyphicon-phone:before{content:"\e145"}.glyphicon-pushpin:before{content:"\e146"}.glyphicon-usd:before{content:"\e148"}.glyphicon-gbp:before{content:"\e149"}.glyphicon-sort:before{content:"\e150"}.glyphicon-sort-by-alphabet:before{content:"\e151"}.glyphicon-sort-by-alphabet-alt:before{content:"\e152"}.glyphicon-sort-by-order:before{content:"\e153"}.glyphicon-sort-by-order-alt:before{content:"\e154"}.glyphicon-sort-by-attributes:before{content:"\e155"}.glyphicon-sort-by-attributes-alt:before{content:"\e156"}.glyphicon-unchecked:before{content:"\e157"}.glyphicon-expand:before{content:"\e158"}.glyphicon-collapse-down:before{content:"\e159"}.glyphicon-collapse-up:before{content:"\e160"}.glyphicon-log-in:before{content:"\e161"}.glyphicon-flash:before{content:"\e162"}.glyphicon-log-out:before{content:"\e163"}.glyphicon-new-window:before{content:"\e164"}.glyphicon-record:before{content:"\e165"}.glyphicon-save:before{content:"\e166"}.glyphicon-open:before{content:"\e167"}.glyphicon-saved:before{content:"\e168"}.glyphicon-import:before{content:"\e169"}.glyphicon-export:before{content:"\e170"}.glyphicon-send:before{content:"\e171"}.glyphicon-floppy-disk:before{content:"\e172"}.glyphicon-floppy-saved:before{content:"\e173"}.glyphicon-floppy-remove:before{content:"\e174"}.glyphicon-floppy-save:before{content:"\e175"}.glyphicon-floppy-open:before{content:"\e176"}.glyphicon-credit-card:before{content:"\e177"}.glyphicon-transfer:before{content:"\e178"}.glyphicon-cutlery:before{content:"\e179"}.glyphicon-header:before{content:"\e180"}.glyphicon-compressed:before{content:"\e181"}.glyphicon-earphone:before{content:"\e182"}.glyphicon-phone-alt:before{content:"\e183"}.glyphicon-tower:before{content:"\e184"}.glyphicon-stats:before{content:"\e185"}.glyphicon-sd-video:before{content:"\e186"}.glyphicon-hd-video:before{content:"\e187"}.glyphicon-subtitles:before{content:"\e188"}.glyphicon-sound-stereo:before{content:"\e189"}.glyphicon-sound-dolby:before{content:"\e190"}.glyphicon-sound-5-1:before{content:"\e191"}.glyphicon-sound-6-1:before{content:"\e192"}.glyphicon-sound-7-1:before{content:"\e193"}.glyphicon-copyright-mark:before{content:"\e194"}.glyphicon-registration-mark:before{content:"\e195"}.glyphicon-cloud-download:before{content:"\e197"}.glyphicon-cloud-upload:before{content:"\e198"}.glyphicon-tree-conifer:before{content:"\e199"}.glyphicon-tree-deciduous:before{content:"\e200"}.glyphicon-cd:before{content:"\e201"}.glyphicon-save-file:before{content:"\e202"}.glyphicon-open-file:before{content:"\e203"}.glyphicon-level-up:before{content:"\e204"}.glyphicon-copy:before{content:"\e205"}.glyphicon-paste:before{content:"\e206"}.glyphicon-alert:before{content:"\e209"}.glyphicon-equalizer:before{content:"\e210"}.glyphicon-king:before{content:"\e211"}.glyphicon-queen:before{content:"\e212"}.glyphicon-pawn:before{content:"\e213"}.glyphicon-bishop:before{content:"\e214"}.glyphicon-knight:before{content:"\e215"}.glyphicon-baby-formula:before{content:"\e216"}.glyphicon-tent:before{content:"\26fa"}.glyphicon-blackboard:before{content:"\e218"}.glyphicon-bed:before{content:"\e219"}.glyphicon-apple:before{content:"\f8ff"}.glyphicon-erase:before{content:"\e221"}.glyphicon-hourglass:before{content:"\231b"}.glyphicon-lamp:before{content:"\e223"}.glyphicon-duplicate:before{content:"\e224"}.glyphicon-piggy-bank:before{content:"\e225"}.glyphicon-scissors:before{content:"\e226"}.glyphicon-bitcoin:before{content:"\e227"}.glyphicon-btc:before{content:"\e227"}.glyphicon-xbt:before{content:"\e227"}.glyphicon-yen:before{content:"\00a5"}.glyphicon-jpy:before{content:"\00a5"}.glyphicon-ruble:before{content:"\20bd"}.glyphicon-rub:before{content:"\20bd"}.glyphicon-scale:before{content:"\e230"}.glyphicon-ice-lolly:before{content:"\e231"}.glyphicon-ice-lolly-tasted:before{content:"\e232"}.glyphicon-education:before{content:"\e233"}.glyphicon-option-horizontal:before{content:"\e234"}.glyphicon-option-vertical:before{content:"\e235"}.glyphicon-menu-hamburger:before{content:"\e236"}.glyphicon-modal-window:before{content:"\e237"}.glyphicon-oil:before{content:"\e238"}.glyphicon-grain:before{content:"\e239"}.glyphicon-sunglasses:before{content:"\e240"}.glyphicon-text-size:before{content:"\e241"}.glyphicon-text-color:before{content:"\e242"}.glyphicon-text-background:before{content:"\e243"}.glyphicon-object-align-top:before{content:"\e244"}.glyphicon-object-align-bottom:before{content:"\e245"}.glyphicon-object-align-horizontal:before{content:"\e246"}.glyphicon-object-align-left:before{content:"\e247"}.glyphicon-object-align-vertical:before{content:"\e248"}.glyphicon-object-align-right:before{content:"\e249"}.glyphicon-triangle-right:before{content:"\e250"}.glyphicon-triangle-left:before{content:"\e251"}.glyphicon-triangle-bottom:before{content:"\e252"}.glyphicon-triangle-top:before{content:"\e253"}.glyphicon-console:before{content:"\e254"}.glyphicon-superscript:before{content:"\e255"}.glyphicon-subscript:before{content:"\e256"}.glyphicon-menu-left:before{content:"\e257"}.glyphicon-menu-right:before{content:"\e258"}.glyphicon-menu-down:before{content:"\e259"}.glyphicon-menu-up:before{content:"\e260"}*{-webkit-box-sizing:border-box;-moz-box-sizing:border-box;box-sizing:border-box}:after,:before{-webkit-box-sizing:border-box;-moz-box-sizing:border-box;box-sizing:border-box}html{font-size:10px;-webkit-tap-highlight-color:rgba(0,0,0,0)}body{font-family:"Helvetica Neue",Helvetica,Arial,sans-serif;font-size:14px;line-height:1.42857143;color:#333;background-color:#fff}button,input,select,textarea{font-family:inherit;font-size:inherit;line-height:inherit}a{color:#337ab7;text-decoration:none}a:focus,a:hover{color:#23527c;text-decoration:underline}a:focus{outline:thin dotted;outline:5px auto -webkit-focus-ring-color;outline-offset:-2px}figure{margin:0}img{vertical-align:middle}.carousel-inner>.item>a>img,.carousel-inner>.item>img,.img-responsive,.thumbnail a>img,.thumbnail>img{display:block;max-width:100%;height:auto}.img-rounded{border-radius:6px}.img-thumbnail{display:inline-block;max-width:100%;height:auto;padding:4px;line-height:1.42857143;background-color:#fff;border:1px solid #ddd;border-radius:4px;-webkit-transition:all .2s ease-in-out;-o-transition:all .2s ease-in-out;transition:all .2s ease-in-out}.img-circle{border-radius:50%}hr{margin-top:20px;margin-bottom:20px;border:0;border-top:1px solid #eee}.sr-only{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;clip:rect(0,0,0,0);border:0}.sr-only-focusable:active,.sr-only-focusable:focus{position:static;width:auto;height:auto;margin:0;overflow:visible;clip:auto}[role=button]{cursor:pointer}.h1,.h2,.h3,.h4,.h5,.h6,h1,h2,h3,h4,h5,h6{font-family:inherit;font-weight:500;line-height:1.1;color:inherit}.h1 .small,.h1 small,.h2 .small,.h2 small,.h3 .small,.h3 small,.h4 .small,.h4 small,.h5 .small,.h5 small,.h6 .small,.h6 small,h1 .small,h1 small,h2 .small,h2 small,h3 .small,h3 small,h4 .small,h4 small,h5 .small,h5 small,h6 .small,h6 small{font-weight:400;line-height:1;color:#777}.h1,.h2,.h3,h1,h2,h3{margin-top:20px;margin-bottom:10px}.h1 .small,.h1 small,.h2 .small,.h2 small,.h3 .small,.h3 small,h1 .small,h1 small,h2 .small,h2 small,h3 .small,h3 small{font-size:65%}.h4,.h5,.h6,h4,h5,h6{margin-top:10px;margin-bottom:10px}.h4 .small,.h4 small,.h5 .small,.h5 small,.h6 .small,.h6 small,h4 .small,h4 small,h5 .small,h5 small,h6 .small,h6 small{font-size:75%}.h1,h1{font-size:36px}.h2,h2{font-size:30px}.h3,h3{font-size:24px}.h4,h4{font-size:18px}.h5,h5{font-size:14px}.h6,h6{font-size:12px}p{margin:0 0 10px}.lead{margin-bottom:20px;font-size:16px;font-weight:300;line-height:1.4}@media (min-width:768px){.lead{font-size:21px}}.small,small{font-size:85%}.mark,mark{padding:.2em;background-color:#fcf8e3}.text-left{text-align:left}.text-right{text-align:right}.text-center{text-align:center}.text-justify{text-align:justify}.text-nowrap{white-space:nowrap}.text-lowercase{text-transform:lowercase}.text-uppercase{text-transform:uppercase}.text-capitalize{text-transform:capitalize}.text-muted{color:#777}.text-primary{color:#337ab7}a.text-primary:focus,a.text-primary:hover{color:#286090}.text-success{color:#3c763d}a.text-success:focus,a.text-success:hover{color:#2b542c}.text-info{color:#31708f}a.text-info:focus,a.text-info:hover{color:#245269}.text-warning{color:#8a6d3b}a.text-warning:focus,a.text-warning:hover{color:#66512c}.text-danger{color:#a94442}a.text-danger:focus,a.text-danger:hover{color:#843534}.bg-primary{color:#fff;background-color:#337ab7}a.bg-primary:focus,a.bg-primary:hover{background-color:#286090}.bg-success{background-color:#dff0d8}a.bg-success:focus,a.bg-success:hover{background-color:#c1e2b3}.bg-info{background-color:#d9edf7}a.bg-info:focus,a.bg-info:hover{background-color:#afd9ee}.bg-warning{background-color:#fcf8e3}a.bg-warning:focus,a.bg-warning:hover{background-color:#f7ecb5}.bg-danger{background-color:#f2dede}a.bg-danger:focus,a.bg-danger:hover{background-color:#e4b9b9}.page-header{padding-bottom:9px;margin:40px 0 20px;border-bottom:1px solid #eee}ol,ul{margin-top:0;margin-bottom:10px}ol ol,ol ul,ul ol,ul ul{margin-bottom:0}.list-unstyled{padding-left:0;list-style:none}.list-inline{padding-left:0;margin-left:-5px;list-style:none}.list-inline>li{display:inline-block;padding-right:5px;padding-left:5px}dl{margin-top:0;margin-bottom:20px}dd,dt{line-height:1.42857143}dt{font-weight:700}dd{margin-left:0}@media (min-width:768px){.dl-horizontal dt{float:left;width:160px;overflow:hidden;clear:left;text-align:right;text-overflow:ellipsis;white-space:nowrap}.dl-horizontal dd{margin-left:180px}}abbr[data-original-title],abbr[title]{cursor:help;border-bottom:1px dotted #777}.initialism{font-size:90%;text-transform:uppercase}blockquote{padding:10px 20px;margin:0 0 20px;font-size:17.5px;border-left:5px solid #eee}blockquote ol:last-child,blockquote p:last-child,blockquote ul:last-child{margin-bottom:0}blockquote .small,blockquote footer,blockquote small{display:block;font-size:80%;line-height:1.42857143;color:#777}blockquote .small:before,blockquote footer:before,blockquote small:before{content:'\2014 \00A0'}.blockquote-reverse,blockquote.pull-right{padding-right:15px;padding-left:0;text-align:right;border-right:5px solid #eee;border-left:0}.blockquote-reverse .small:before,.blockquote-reverse footer:before,.blockquote-reverse small:before,blockquote.pull-right .small:before,blockquote.pull-right footer:before,blockquote.pull-right small:before{content:''}.blockquote-reverse .small:after,.blockquote-reverse footer:after,.blockquote-reverse small:after,blockquote.pull-right .small:after,blockquote.pull-right footer:after,blockquote.pull-right small:after{content:'\00A0 \2014'}address{margin-bottom:20px;font-style:normal;line-height:1.42857143}code,kbd,pre,samp{font-family:monospace}code{padding:2px 4px;font-size:90%;color:#c7254e;background-color:#f9f2f4;border-radius:4px}kbd{padding:2px 4px;font-size:90%;color:#fff;background-color:#333;border-radius:3px;-webkit-box-shadow:inset 0 -1px 0 rgba(0,0,0,.25);box-shadow:inset 0 -1px 0 rgba(0,0,0,.25)}kbd kbd{padding:0;font-size:100%;font-weight:700;-webkit-box-shadow:none;box-shadow:none}pre{display:block;padding:9.5px;margin:0 0 10px;font-size:13px;line-height:1.42857143;color:#333;word-break:break-all;word-wrap:break-word;background-color:#f5f5f5;border:1px solid #ccc;border-radius:4px}pre code{padding:0;font-size:inherit;color:inherit;white-space:pre-wrap;background-color:transparent;border-radius:0}.pre-scrollable{max-height:340px;overflow-y:scroll}.container{padding-right:15px;padding-left:15px;margin-right:auto;margin-left:auto}@media (min-width:768px){.container{width:750px}}@media (min-width:992px){.container{width:970px}}@media (min-width:1200px){.container{width:1170px}}.container-fluid{padding-right:15px;padding-left:15px;margin-right:auto;margin-left:auto}.row{margin-right:-15px;margin-left:-15px}.col-lg-1,.col-lg-10,.col-lg-11,.col-lg-12,.col-lg-2,.col-lg-3,.col-lg-4,.col-lg-5,.col-lg-6,.col-lg-7,.col-lg-8,.col-lg-9,.col-md-1,.col-md-10,.col-md-11,.col-md-12,.col-md-2,.col-md-3,.col-md-4,.col-md-5,.col-md-6,.col-md-7,.col-md-8,.col-md-9,.col-sm-1,.col-sm-10,.col-sm-11,.col-sm-12,.col-sm-2,.col-sm-3,.col-sm-4,.col-sm-5,.col-sm-6,.col-sm-7,.col-sm-8,.col-sm-9,.col-xs-1,.col-xs-10,.col-xs-11,.col-xs-12,.col-xs-2,.col-xs-3,.col-xs-4,.col-xs-5,.col-xs-6,.col-xs-7,.col-xs-8,.col-xs-9{position:relative;min-height:1px;padding-right:15px;padding-left:15px}.col-xs-1,.col-xs-10,.col-xs-11,.col-xs-12,.col-xs-2,.col-xs-3,.col-xs-4,.col-xs-5,.col-xs-6,.col-xs-7,.col-xs-8,.col-xs-9{float:left}.col-xs-12{width:100%}.col-xs-11{width:91.66666667%}.col-xs-10{width:83.33333333%}.col-xs-9{width:75%}.col-xs-8{width:66.66666667%}.col-xs-7{width:58.33333333%}.col-xs-6{width:50%}.col-xs-5{width:41.66666667%}.col-xs-4{width:33.33333333%}.col-xs-3{width:25%}.col-xs-2{width:16.66666667%}.col-xs-1{width:8.33333333%}.col-xs-pull-12{right:100%}.col-xs-pull-11{right:91.66666667%}.col-xs-pull-10{right:83.33333333%}.col-xs-pull-9{right:75%}.col-xs-pull-8{right:66.66666667%}.col-xs-pull-7{right:58.33333333%}.col-xs-pull-6{right:50%}.col-xs-pull-5{right:41.66666667%}.col-xs-pull-4{right:33.33333333%}.col-xs-pull-3{right:25%}.col-xs-pull-2{right:16.66666667%}.col-xs-pull-1{right:8.33333333%}.col-xs-pull-0{right:auto}.col-xs-push-12{left:100%}.col-xs-push-11{left:91.66666667%}.col-xs-push-10{left:83.33333333%}.col-xs-push-9{left:75%}.col-xs-push-8{left:66.66666667%}.col-xs-push-7{left:58.33333333%}.col-xs-push-6{left:50%}.col-xs-push-5{left:41.66666667%}.col-xs-push-4{left:33.33333333%}.col-xs-push-3{left:25%}.col-xs-push-2{left:16.66666667%}.col-xs-push-1{left:8.33333333%}.col-xs-push-0{left:auto}.col-xs-offset-12{margin-left:100%}.col-xs-offset-11{margin-left:91.66666667%}.col-xs-offset-10{margin-left:83.33333333%}.col-xs-offset-9{margin-left:75%}.col-xs-offset-8{margin-left:66.66666667%}.col-xs-offset-7{margin-left:58.33333333%}.col-xs-offset-6{margin-left:50%}.col-xs-offset-5{margin-left:41.66666667%}.col-xs-offset-4{margin-left:33.33333333%}.col-xs-offset-3{margin-left:25%}.col-xs-offset-2{margin-left:16.66666667%}.col-xs-offset-1{margin-left:8.33333333%}.col-xs-offset-0{margin-left:0}@media (min-width:768px){.col-sm-1,.col-sm-10,.col-sm-11,.col-sm-12,.col-sm-2,.col-sm-3,.col-sm-4,.col-sm-5,.col-sm-6,.col-sm-7,.col-sm-8,.col-sm-9{float:left}.col-sm-12{width:100%}.col-sm-11{width:91.66666667%}.col-sm-10{width:83.33333333%}.col-sm-9{width:75%}.col-sm-8{width:66.66666667%}.col-sm-7{width:58.33333333%}.col-sm-6{width:50%}.col-sm-5{width:41.66666667%}.col-sm-4{width:33.33333333%}.col-sm-3{width:25%}.col-sm-2{width:16.66666667%}.col-sm-1{width:8.33333333%}.col-sm-pull-12{right:100%}.col-sm-pull-11{right:91.66666667%}.col-sm-pull-10{right:83.33333333%}.col-sm-pull-9{right:75%}.col-sm-pull-8{right:66.66666667%}.col-sm-pull-7{right:58.33333333%}.col-sm-pull-6{right:50%}.col-sm-pull-5{right:41.66666667%}.col-sm-pull-4{right:33.33333333%}.col-sm-pull-3{right:25%}.col-sm-pull-2{right:16.66666667%}.col-sm-pull-1{right:8.33333333%}.col-sm-pull-0{right:auto}.col-sm-push-12{left:100%}.col-sm-push-11{left:91.66666667%}.col-sm-push-10{left:83.33333333%}.col-sm-push-9{left:75%}.col-sm-push-8{left:66.66666667%}.col-sm-push-7{left:58.33333333%}.col-sm-push-6{left:50%}.col-sm-push-5{left:41.66666667%}.col-sm-push-4{left:33.33333333%}.col-sm-push-3{left:25%}.col-sm-push-2{left:16.66666667%}.col-sm-push-1{left:8.33333333%}.col-sm-push-0{left:auto}.col-sm-offset-12{margin-left:100%}.col-sm-offset-11{margin-left:91.66666667%}.col-sm-offset-10{margin-left:83.33333333%}.col-sm-offset-9{margin-left:75%}.col-sm-offset-8{margin-left:66.66666667%}.col-sm-offset-7{margin-left:58.33333333%}.col-sm-offset-6{margin-left:50%}.col-sm-offset-5{margin-left:41.66666667%}.col-sm-offset-4{margin-left:33.33333333%}.col-sm-offset-3{margin-left:25%}.col-sm-offset-2{margin-left:16.66666667%}.col-sm-offset-1{margin-left:8.33333333%}.col-sm-offset-0{margin-left:0}}@media (min-width:992px){.col-md-1,.col-md-10,.col-md-11,.col-md-12,.col-md-2,.col-md-3,.col-md-4,.col-md-5,.col-md-6,.col-md-7,.col-md-8,.col-md-9{float:left}.col-md-12{width:100%}.col-md-11{width:91.66666667%}.col-md-10{width:83.33333333%}.col-md-9{width:75%}.col-md-8{width:66.66666667%}.col-md-7{width:58.33333333%}.col-md-6{width:50%}.col-md-5{width:41.66666667%}.col-md-4{width:33.33333333%}.col-md-3{width:25%}.col-md-2{width:16.66666667%}.col-md-1{width:8.33333333%}.col-md-pull-12{right:100%}.col-md-pull-11{right:91.66666667%}.col-md-pull-10{right:83.33333333%}.col-md-pull-9{right:75%}.col-md-pull-8{right:66.66666667%}.col-md-pull-7{right:58.33333333%}.col-md-pull-6{right:50%}.col-md-pull-5{right:41.66666667%}.col-md-pull-4{right:33.33333333%}.col-md-pull-3{right:25%}.col-md-pull-2{right:16.66666667%}.col-md-pull-1{right:8.33333333%}.col-md-pull-0{right:auto}.col-md-push-12{left:100%}.col-md-push-11{left:91.66666667%}.col-md-push-10{left:83.33333333%}.col-md-push-9{left:75%}.col-md-push-8{left:66.66666667%}.col-md-push-7{left:58.33333333%}.col-md-push-6{left:50%}.col-md-push-5{left:41.66666667%}.col-md-push-4{left:33.33333333%}.col-md-push-3{left:25%}.col-md-push-2{left:16.66666667%}.col-md-push-1{left:8.33333333%}.col-md-push-0{left:auto}.col-md-offset-12{margin-left:100%}.col-md-offset-11{margin-left:91.66666667%}.col-md-offset-10{margin-left:83.33333333%}.col-md-offset-9{margin-left:75%}.col-md-offset-8{margin-left:66.66666667%}.col-md-offset-7{margin-left:58.33333333%}.col-md-offset-6{margin-left:50%}.col-md-offset-5{margin-left:41.66666667%}.col-md-offset-4{margin-left:33.33333333%}.col-md-offset-3{margin-left:25%}.col-md-offset-2{margin-left:16.66666667%}.col-md-offset-1{margin-left:8.33333333%}.col-md-offset-0{margin-left:0}}@media (min-width:1200px){.col-lg-1,.col-lg-10,.col-lg-11,.col-lg-12,.col-lg-2,.col-lg-3,.col-lg-4,.col-lg-5,.col-lg-6,.col-lg-7,.col-lg-8,.col-lg-9{float:left}.col-lg-12{width:100%}.col-lg-11{width:91.66666667%}.col-lg-10{width:83.33333333%}.col-lg-9{width:75%}.col-lg-8{width:66.66666667%}.col-lg-7{width:58.33333333%}.col-lg-6{width:50%}.col-lg-5{width:41.66666667%}.col-lg-4{width:33.33333333%}.col-lg-3{width:25%}.col-lg-2{width:16.66666667%}.col-lg-1{width:8.33333333%}.col-lg-pull-12{right:100%}.col-lg-pull-11{right:91.66666667%}.col-lg-pull-10{right:83.33333333%}.col-lg-pull-9{right:75%}.col-lg-pull-8{right:66.66666667%}.col-lg-pull-7{right:58.33333333%}.col-lg-pull-6{right:50%}.col-lg-pull-5{right:41.66666667%}.col-lg-pull-4{right:33.33333333%}.col-lg-pull-3{right:25%}.col-lg-pull-2{right:16.66666667%}.col-lg-pull-1{right:8.33333333%}.col-lg-pull-0{right:auto}.col-lg-push-12{left:100%}.col-lg-push-11{left:91.66666667%}.col-lg-push-10{left:83.33333333%}.col-lg-push-9{left:75%}.col-lg-push-8{left:66.66666667%}.col-lg-push-7{left:58.33333333%}.col-lg-push-6{left:50%}.col-lg-push-5{left:41.66666667%}.col-lg-push-4{left:33.33333333%}.col-lg-push-3{left:25%}.col-lg-push-2{left:16.66666667%}.col-lg-push-1{left:8.33333333%}.col-lg-push-0{left:auto}.col-lg-offset-12{margin-left:100%}.col-lg-offset-11{margin-left:91.66666667%}.col-lg-offset-10{margin-left:83.33333333%}.col-lg-offset-9{margin-left:75%}.col-lg-offset-8{margin-left:66.66666667%}.col-lg-offset-7{margin-left:58.33333333%}.col-lg-offset-6{margin-left:50%}.col-lg-offset-5{margin-left:41.66666667%}.col-lg-offset-4{margin-left:33.33333333%}.col-lg-offset-3{margin-left:25%}.col-lg-offset-2{margin-left:16.66666667%}.col-lg-offset-1{margin-left:8.33333333%}.col-lg-offset-0{margin-left:0}}table{background-color:transparent}caption{padding-top:8px;padding-bottom:8px;color:#777;text-align:left}th{}.table{width:100%;max-width:100%;margin-bottom:20px}.table>tbody>tr>td,.table>tbody>tr>th,.table>tfoot>tr>td,.table>tfoot>tr>th,.table>thead>tr>td,.table>thead>tr>th{padding:8px;line-height:1.42857143;vertical-align:top;border-top:1px solid #ddd}.table>thead>tr>th{vertical-align:bottom;border-bottom:2px solid #ddd}.table>caption+thead>tr:first-child>td,.table>caption+thead>tr:first-child>th,.table>colgroup+thead>tr:first-child>td,.table>colgroup+thead>tr:first-child>th,.table>thead:first-child>tr:first-child>td,.table>thead:first-child>tr:first-child>th{border-top:0}.table>tbody+tbody{border-top:2px solid #ddd}.table .table{background-color:#fff}.table-condensed>tbody>tr>td,.table-condensed>tbody>tr>th,.table-condensed>tfoot>tr>td,.table-condensed>tfoot>tr>th,.table-condensed>thead>tr>td,.table-condensed>thead>tr>th{padding:5px}.table-bordered{border:1px solid #ddd}.table-bordered>tbody>tr>td,.table-bordered>tbody>tr>th,.table-bordered>tfoot>tr>td,.table-bordered>tfoot>tr>th,.table-bordered>thead>tr>td,.table-bordered>thead>tr>th{border:1px solid #ddd}.table-bordered>thead>tr>td,.table-bordered>thead>tr>th{border-bottom-width:2px}.table-striped>tbody>tr:nth-of-type(odd){background-color:#f9f9f9}.table-hover>tbody>tr:hover{background-color:#f5f5f5}table col[class*=col-]{position:static;display:table-column;float:none}table td[class*=col-],table th[class*=col-]{position:static;display:table-cell;float:none}.table>tbody>tr.active>td,.table>tbody>tr.active>th,.table>tbody>tr>td.active,.table>tbody>tr>th.active,.table>tfoot>tr.active>td,.table>tfoot>tr.active>th,.table>tfoot>tr>td.active,.table>tfoot>tr>th.active,.table>thead>tr.active>td,.table>thead>tr.active>th,.table>thead>tr>td.active,.table>thead>tr>th.active{background-color:#f5f5f5}.table-hover>tbody>tr.active:hover>td,.table-hover>tbody>tr.active:hover>th,.table-hover>tbody>tr:hover>.active,.table-hover>tbody>tr>td.active:hover,.table-hover>tbody>tr>th.active:hover{background-color:#e8e8e8}.table>tbody>tr.success>td,.table>tbody>tr.success>th,.table>tbody>tr>td.success,.table>tbody>tr>th.success,.table>tfoot>tr.success>td,.table>tfoot>tr.success>th,.table>tfoot>tr>td.success,.table>tfoot>tr>th.success,.table>thead>tr.success>td,.table>thead>tr.success>th,.table>thead>tr>td.success,.table>thead>tr>th.success{background-color:#dff0d8}.table-hover>tbody>tr.success:hover>td,.table-hover>tbody>tr.success:hover>th,.table-hover>tbody>tr:hover>.success,.table-hover>tbody>tr>td.success:hover,.table-hover>tbody>tr>th.success:hover{background-color:#d0e9c6}.table>tbody>tr.info>td,.table>tbody>tr.info>th,.table>tbody>tr>td.info,.table>tbody>tr>th.info,.table>tfoot>tr.info>td,.table>tfoot>tr.info>th,.table>tfoot>tr>td.info,.table>tfoot>tr>th.info,.table>thead>tr.info>td,.table>thead>tr.info>th,.table>thead>tr>td.info,.table>thead>tr>th.info{background-color:#d9edf7}.table-hover>tbody>tr.info:hover>td,.table-hover>tbody>tr.info:hover>th,.table-hover>tbody>tr:hover>.info,.table-hover>tbody>tr>td.info:hover,.table-hover>tbody>tr>th.info:hover{background-color:#c4e3f3}.table>tbody>tr.warning>td,.table>tbody>tr.warning>th,.table>tbody>tr>td.warning,.table>tbody>tr>th.warning,.table>tfoot>tr.warning>td,.table>tfoot>tr.warning>th,.table>tfoot>tr>td.warning,.table>tfoot>tr>th.warning,.table>thead>tr.warning>td,.table>thead>tr.warning>th,.table>thead>tr>td.warning,.table>thead>tr>th.warning{background-color:#fcf8e3}.table-hover>tbody>tr.warning:hover>td,.table-hover>tbody>tr.warning:hover>th,.table-hover>tbody>tr:hover>.warning,.table-hover>tbody>tr>td.warning:hover,.table-hover>tbody>tr>th.warning:hover{background-color:#faf2cc}.table>tbody>tr.danger>td,.table>tbody>tr.danger>th,.table>tbody>tr>td.danger,.table>tbody>tr>th.danger,.table>tfoot>tr.danger>td,.table>tfoot>tr.danger>th,.table>tfoot>tr>td.danger,.table>tfoot>tr>th.danger,.table>thead>tr.danger>td,.table>thead>tr.danger>th,.table>thead>tr>td.danger,.table>thead>tr>th.danger{background-color:#f2dede}.table-hover>tbody>tr.danger:hover>td,.table-hover>tbody>tr.danger:hover>th,.table-hover>tbody>tr:hover>.danger,.table-hover>tbody>tr>td.danger:hover,.table-hover>tbody>tr>th.danger:hover{background-color:#ebcccc}.table-responsive{min-height:.01%;overflow-x:auto}@media screen and (max-width:767px){.table-responsive{width:100%;margin-bottom:15px;overflow-y:hidden;-ms-overflow-style:-ms-autohiding-scrollbar;border:1px solid #ddd}.table-responsive>.table{margin-bottom:0}.table-responsive>.table>tbody>tr>td,.table-responsive>.table>tbody>tr>th,.table-responsive>.table>tfoot>tr>td,.table-responsive>.table>tfoot>tr>th,.table-responsive>.table>thead>tr>td,.table-responsive>.table>thead>tr>th{white-space:nowrap}.table-responsive>.table-bordered{border:0}.table-responsive>.table-bordered>tbody>tr>td:first-child,.table-responsive>.table-bordered>tbody>tr>th:first-child,.table-responsive>.table-bordered>tfoot>tr>td:first-child,.table-responsive>.table-bordered>tfoot>tr>th:first-child,.table-responsive>.table-bordered>thead>tr>td:first-child,.table-responsive>.table-bordered>thead>tr>th:first-child{border-left:0}.table-responsive>.table-bordered>tbody>tr>td:last-child,.table-responsive>.table-bordered>tbody>tr>th:last-child,.table-responsive>.table-bordered>tfoot>tr>td:last-child,.table-responsive>.table-bordered>tfoot>tr>th:last-child,.table-responsive>.table-bordered>thead>tr>td:last-child,.table-responsive>.table-bordered>thead>tr>th:last-child{border-right:0}.table-responsive>.table-bordered>tbody>tr:last-child>td,.table-responsive>.table-bordered>tbody>tr:last-child>th,.table-responsive>.table-bordered>tfoot>tr:last-child>td,.table-responsive>.table-bordered>tfoot>tr:last-child>th{border-bottom:0}}fieldset{min-width:0;padding:0;margin:0;border:0}legend{display:block;width:100%;padding:0;margin-bottom:20px;font-size:21px;line-height:inherit;color:#333;border:0;border-bottom:1px solid #e5e5e5}label{display:inline-block;max-width:100%;margin-bottom:5px;font-weight:700}input[type=search]{-webkit-box-sizing:border-box;-moz-box-sizing:border-box;box-sizing:border-box}input[type=checkbox],input[type=radio]{margin:4px 0 0;margin-top:1px\9;line-height:normal}input[type=file]{display:block}input[type=range]{display:block;width:100%}select[multiple],select[size]{height:auto}input[type=file]:focus,input[type=checkbox]:focus,input[type=radio]:focus{outline:thin dotted;outline:5px auto -webkit-focus-ring-color;outline-offset:-2px}output{display:block;padding-top:7px;font-size:14px;line-height:1.42857143;color:#555}.form-control{display:block;width:100%;height:34px;padding:6px 12px;font-size:14px;line-height:1.42857143;color:#555;background-color:#fff;background-image:none;border:1px solid #ccc;border-radius:4px;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075);box-shadow:inset 0 1px 1px rgba(0,0,0,.075);-webkit-transition:border-color ease-in-out .15s,-webkit-box-shadow ease-in-out .15s;-o-transition:border-color ease-in-out .15s,box-shadow ease-in-out .15s;transition:border-color ease-in-out .15s,box-shadow ease-in-out .15s}.form-control:focus{border-color:#66afe9;outline:0;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 8px rgba(102,175,233,.6);box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 8px rgba(102,175,233,.6)}.form-control::-moz-placeholder{color:#999;opacity:1}.form-control:-ms-input-placeholder{color:#999}.form-control::-webkit-input-placeholder{color:#999}.form-control[disabled],.form-control[readonly],fieldset[disabled] .form-control{background-color:#eee;opacity:1}.form-control[disabled],fieldset[disabled] .form-control{cursor:not-allowed}textarea.form-control{height:auto}input[type=search]{-webkit-appearance:none}@media screen and (-webkit-min-device-pixel-ratio:0){input[type=date].form-control,input[type=time].form-control,input[type=datetime-local].form-control,input[type=month].form-control{line-height:34px}.input-group-sm input[type=date],.input-group-sm input[type=time],.input-group-sm input[type=datetime-local],.input-group-sm input[type=month],input[type=date].input-sm,input[type=time].input-sm,input[type=datetime-local].input-sm,input[type=month].input-sm{line-height:30px}.input-group-lg input[type=date],.input-group-lg input[type=time],.input-group-lg input[type=datetime-local],.input-group-lg input[type=month],input[type=date].input-lg,input[type=time].input-lg,input[type=datetime-local].input-lg,input[type=month].input-lg{line-height:46px}}.form-group{margin-bottom:15px}.checkbox,.radio{position:relative;display:block;margin-top:10px;margin-bottom:10px}.checkbox label,.radio label{min-height:20px;padding-left:20px;margin-bottom:0;font-weight:400;cursor:pointer}.checkbox input[type=checkbox],.checkbox-inline input[type=checkbox],.radio input[type=radio],.radio-inline input[type=radio]{position:absolute;margin-top:4px\9;margin-left:-20px}.checkbox+.checkbox,.radio+.radio{margin-top:-5px}.checkbox-inline,.radio-inline{position:relative;display:inline-block;padding-left:20px;margin-bottom:0;font-weight:400;vertical-align:middle;cursor:pointer}.checkbox-inline+.checkbox-inline,.radio-inline+.radio-inline{margin-top:0;margin-left:10px}fieldset[disabled] input[type=checkbox],fieldset[disabled] input[type=radio],input[type=checkbox].disabled,input[type=checkbox][disabled],input[type=radio].disabled,input[type=radio][disabled]{cursor:not-allowed}.checkbox-inline.disabled,.radio-inline.disabled,fieldset[disabled] .checkbox-inline,fieldset[disabled] .radio-inline{cursor:not-allowed}.checkbox.disabled label,.radio.disabled label,fieldset[disabled] .checkbox label,fieldset[disabled] .radio label{cursor:not-allowed}.form-control-static{min-height:34px;padding-top:7px;padding-bottom:7px;margin-bottom:0}.form-control-static.input-lg,.form-control-static.input-sm{padding-right:0;padding-left:0}.input-sm{height:30px;padding:5px 10px;font-size:12px;line-height:1.5;border-radius:3px}select.input-sm{height:30px;line-height:30px}select[multiple].input-sm,textarea.input-sm{height:auto}.form-group-sm .form-control{height:30px;padding:5px 10px;font-size:12px;line-height:1.5;border-radius:3px}.form-group-sm select.form-control{height:30px;line-height:30px}.form-group-sm select[multiple].form-control,.form-group-sm textarea.form-control{height:auto}.form-group-sm .form-control-static{height:30px;min-height:32px;padding:6px 10px;font-size:12px;line-height:1.5}.input-lg{height:46px;padding:10px 16px;font-size:18px;line-height:1.3333333;border-radius:6px}select.input-lg{height:46px;line-height:46px}select[multiple].input-lg,textarea.input-lg{height:auto}.form-group-lg .form-control{height:46px;padding:10px 16px;font-size:18px;line-height:1.3333333;border-radius:6px}.form-group-lg select.form-control{height:46px;line-height:46px}.form-group-lg select[multiple].form-control,.form-group-lg textarea.form-control{height:auto}.form-group-lg .form-control-static{height:46px;min-height:38px;padding:11px 16px;font-size:18px;line-height:1.3333333}.has-feedback{position:relative}.has-feedback .form-control{padding-right:42.5px}.form-control-feedback{position:absolute;top:0;right:0;z-index:2;display:block;width:34px;height:34px;line-height:34px;text-align:center;pointer-events:none}.form-group-lg .form-control+.form-control-feedback,.input-group-lg+.form-control-feedback,.input-lg+.form-control-feedback{width:46px;height:46px;line-height:46px}.form-group-sm .form-control+.form-control-feedback,.input-group-sm+.form-control-feedback,.input-sm+.form-control-feedback{width:30px;height:30px;line-height:30px}.has-success .checkbox,.has-success .checkbox-inline,.has-success .control-label,.has-success .help-block,.has-success .radio,.has-success .radio-inline,.has-success.checkbox label,.has-success.checkbox-inline label,.has-success.radio label,.has-success.radio-inline label{color:#3c763d}.has-success .form-control{border-color:#3c763d;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075);box-shadow:inset 0 1px 1px rgba(0,0,0,.075)}.has-success .form-control:focus{border-color:#2b542c;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 6px #67b168;box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 6px #67b168}.has-success .input-group-addon{color:#3c763d;background-color:#dff0d8;border-color:#3c763d}.has-success .form-control-feedback{color:#3c763d}.has-warning .checkbox,.has-warning .checkbox-inline,.has-warning .control-label,.has-warning .help-block,.has-warning .radio,.has-warning .radio-inline,.has-warning.checkbox label,.has-warning.checkbox-inline label,.has-warning.radio label,.has-warning.radio-inline label{color:#8a6d3b}.has-warning .form-control{border-color:#8a6d3b;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075);box-shadow:inset 0 1px 1px rgba(0,0,0,.075)}.has-warning .form-control:focus{border-color:#66512c;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 6px #c0a16b;box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 6px #c0a16b}.has-warning .input-group-addon{color:#8a6d3b;background-color:#fcf8e3;border-color:#8a6d3b}.has-warning .form-control-feedback{color:#8a6d3b}.has-error .checkbox,.has-error .checkbox-inline,.has-error .control-label,.has-error .help-block,.has-error .radio,.has-error .radio-inline,.has-error.checkbox label,.has-error.checkbox-inline label,.has-error.radio label,.has-error.radio-inline label{color:#a94442}.has-error .form-control{border-color:#a94442;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075);box-shadow:inset 0 1px 1px rgba(0,0,0,.075)}.has-error .form-control:focus{border-color:#843534;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 6px #ce8483;box-shadow:inset 0 1px 1px rgba(0,0,0,.075),0 0 6px #ce8483}.has-error .input-group-addon{color:#a94442;background-color:#f2dede;border-color:#a94442}.has-error .form-control-feedback{color:#a94442}.has-feedback label~.form-control-feedback{top:25px}.has-feedback label.sr-only~.form-control-feedback{top:0}.help-block{display:block;margin-top:5px;margin-bottom:10px;color:#737373}@media (min-width:768px){.form-inline .form-group{display:inline-block;margin-bottom:0;vertical-align:middle}.form-inline .form-control{display:inline-block;width:auto;vertical-align:middle}.form-inline .form-control-static{display:inline-block}.form-inline .input-group{display:inline-table;vertical-align:middle}.form-inline .input-group .form-control,.form-inline .input-group .input-group-addon,.form-inline .input-group .input-group-btn{width:auto}.form-inline .input-group>.form-control{width:100%}.form-inline .control-label{margin-bottom:0;vertical-align:middle}.form-inline .checkbox,.form-inline .radio{display:inline-block;margin-top:0;margin-bottom:0;vertical-align:middle}.form-inline .checkbox label,.form-inline .radio label{padding-left:0}.form-inline .checkbox input[type=checkbox],.form-inline .radio input[type=radio]{position:relative;margin-left:0}.form-inline .has-feedback .form-control-feedback{top:0}}.form-horizontal .checkbox,.form-horizontal .checkbox-inline,.form-horizontal .radio,.form-horizontal .radio-inline{padding-top:7px;margin-top:0;margin-bottom:0}.form-horizontal .checkbox,.form-horizontal .radio{min-height:27px}.form-horizontal .form-group{margin-right:-15px;margin-left:-15px}@media (min-width:768px){.form-horizontal .control-label{padding-top:7px;margin-bottom:0;text-align:right}}.form-horizontal .has-feedback .form-control-feedback{right:15px}@media (min-width:768px){.form-horizontal .form-group-lg .control-label{padding-top:14.33px;font-size:18px}}@media (min-width:768px){.form-horizontal .form-group-sm .control-label{padding-top:6px;font-size:12px}}.btn{display:inline-block;padding:6px 12px;margin-bottom:0;font-size:14px;font-weight:400;line-height:1.42857143;text-align:center;white-space:nowrap;vertical-align:middle;-ms-touch-action:manipulation;touch-action:manipulation;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-image:none;border:1px solid transparent;border-radius:4px}.btn.active.focus,.btn.active:focus,.btn.focus,.btn:active.focus,.btn:active:focus,.btn:focus{outline:thin dotted;outline:5px auto -webkit-focus-ring-color;outline-offset:-2px}.btn.focus,.btn:focus,.btn:hover{color:#333;text-decoration:none}.btn.active,.btn:active{background-image:none;outline:0;-webkit-box-shadow:inset 0 3px 5px rgba(0,0,0,.125);box-shadow:inset 0 3px 5px rgba(0,0,0,.125)}.btn.disabled,.btn[disabled],fieldset[disabled] .btn{cursor:not-allowed;filter:alpha(opacity=65);-webkit-box-shadow:none;box-shadow:none;opacity:.65}a.btn.disabled,fieldset[disabled] a.btn{pointer-events:none}.btn-default{color:#333;background-color:#fff;border-color:#ccc}.btn-default.focus,.btn-default:focus{color:#333;background-color:#e6e6e6;border-color:#8c8c8c}.btn-default:hover{color:#333;background-color:#e6e6e6;border-color:#adadad}.btn-default.active,.btn-default:active,.open>.dropdown-toggle.btn-default{color:#333;background-color:#e6e6e6;border-color:#adadad}.btn-default.active.focus,.btn-default.active:focus,.btn-default.active:hover,.btn-default:active.focus,.btn-default:active:focus,.btn-default:active:hover,.open>.dropdown-toggle.btn-default.focus,.open>.dropdown-toggle.btn-default:focus,.open>.dropdown-toggle.btn-default:hover{color:#333;background-color:#d4d4d4;border-color:#8c8c8c}.btn-default.active,.btn-default:active,.open>.dropdown-toggle.btn-default{background-image:none}.btn-default.disabled,.btn-default.disabled.active,.btn-default.disabled.focus,.btn-default.disabled:active,.btn-default.disabled:focus,.btn-default.disabled:hover,.btn-default[disabled],.btn-default[disabled].active,.btn-default[disabled].focus,.btn-default[disabled]:active,.btn-default[disabled]:focus,.btn-default[disabled]:hover,fieldset[disabled] .btn-default,fieldset[disabled] .btn-default.active,fieldset[disabled] .btn-default.focus,fieldset[disabled] .btn-default:active,fieldset[disabled] .btn-default:focus,fieldset[disabled] .btn-default:hover{background-color:#fff;border-color:#ccc}.btn-default .badge{color:#fff;background-color:#333}.btn-primary{color:#fff;background-color:#337ab7;border-color:#2e6da4}.btn-primary.focus,.btn-primary:focus{color:#fff;background-color:#286090;border-color:#122b40}.btn-primary:hover{color:#fff;background-color:#286090;border-color:#204d74}.btn-primary.active,.btn-primary:active,.open>.dropdown-toggle.btn-primary{color:#fff;background-color:#286090;border-color:#204d74}.btn-primary.active.focus,.btn-primary.active:focus,.btn-primary.active:hover,.btn-primary:active.focus,.btn-primary:active:focus,.btn-primary:active:hover,.open>.dropdown-toggle.btn-primary.focus,.open>.dropdown-toggle.btn-primary:focus,.open>.dropdown-toggle.btn-primary:hover{color:#fff;background-color:#204d74;border-color:#122b40}.btn-primary.active,.btn-primary:active,.open>.dropdown-toggle.btn-primary{background-image:none}.btn-primary.disabled,.btn-primary.disabled.active,.btn-primary.disabled.focus,.btn-primary.disabled:active,.btn-primary.disabled:focus,.btn-primary.disabled:hover,.btn-primary[disabled],.btn-primary[disabled].active,.btn-primary[disabled].focus,.btn-primary[disabled]:active,.btn-primary[disabled]:focus,.btn-primary[disabled]:hover,fieldset[disabled] .btn-primary,fieldset[disabled] .btn-primary.active,fieldset[disabled] .btn-primary.focus,fieldset[disabled] .btn-primary:active,fieldset[disabled] .btn-primary:focus,fieldset[disabled] .btn-primary:hover{background-color:#337ab7;border-color:#2e6da4}.btn-primary .badge{color:#337ab7;background-color:#fff}.btn-success{color:#fff;background-color:#5cb85c;border-color:#4cae4c}.btn-success.focus,.btn-success:focus{color:#fff;background-color:#449d44;border-color:#255625}.btn-success:hover{color:#fff;background-color:#449d44;border-color:#398439}.btn-success.active,.btn-success:active,.open>.dropdown-toggle.btn-success{color:#fff;background-color:#449d44;border-color:#398439}.btn-success.active.focus,.btn-success.active:focus,.btn-success.active:hover,.btn-success:active.focus,.btn-success:active:focus,.btn-success:active:hover,.open>.dropdown-toggle.btn-success.focus,.open>.dropdown-toggle.btn-success:focus,.open>.dropdown-toggle.btn-success:hover{color:#fff;background-color:#398439;border-color:#255625}.btn-success.active,.btn-success:active,.open>.dropdown-toggle.btn-success{background-image:none}.btn-success.disabled,.btn-success.disabled.active,.btn-success.disabled.focus,.btn-success.disabled:active,.btn-success.disabled:focus,.btn-success.disabled:hover,.btn-success[disabled],.btn-success[disabled].active,.btn-success[disabled].focus,.btn-success[disabled]:active,.btn-success[disabled]:focus,.btn-success[disabled]:hover,fieldset[disabled] .btn-success,fieldset[disabled] .btn-success.active,fieldset[disabled] .btn-success.focus,fieldset[disabled] .btn-success:active,fieldset[disabled] .btn-success:focus,fieldset[disabled] .btn-success:hover{background-color:#5cb85c;border-color:#4cae4c}.btn-success .badge{color:#5cb85c;background-color:#fff}.btn-info{color:#fff;background-color:#5bc0de;border-color:#46b8da}.btn-info.focus,.btn-info:focus{color:#fff;background-color:#31b0d5;border-color:#1b6d85}.btn-info:hover{color:#fff;background-color:#31b0d5;border-color:#269abc}.btn-info.active,.btn-info:active,.open>.dropdown-toggle.btn-info{color:#fff;background-color:#31b0d5;border-color:#269abc}.btn-info.active.focus,.btn-info.active:focus,.btn-info.active:hover,.btn-info:active.focus,.btn-info:active:focus,.btn-info:active:hover,.open>.dropdown-toggle.btn-info.focus,.open>.dropdown-toggle.btn-info:focus,.open>.dropdown-toggle.btn-info:hover{color:#fff;background-color:#269abc;border-color:#1b6d85}.btn-info.active,.btn-info:active,.open>.dropdown-toggle.btn-info{background-image:none}.btn-info.disabled,.btn-info.disabled.active,.btn-info.disabled.focus,.btn-info.disabled:active,.btn-info.disabled:focus,.btn-info.disabled:hover,.btn-info[disabled],.btn-info[disabled].active,.btn-info[disabled].focus,.btn-info[disabled]:active,.btn-info[disabled]:focus,.btn-info[disabled]:hover,fieldset[disabled] .btn-info,fieldset[disabled] .btn-info.active,fieldset[disabled] .btn-info.focus,fieldset[disabled] .btn-info:active,fieldset[disabled] .btn-info:focus,fieldset[disabled] .btn-info:hover{background-color:#5bc0de;border-color:#46b8da}.btn-info .badge{color:#5bc0de;background-color:#fff}.btn-warning{color:#fff;background-color:#f0ad4e;border-color:#eea236}.btn-warning.focus,.btn-warning:focus{color:#fff;background-color:#ec971f;border-color:#985f0d}.btn-warning:hover{color:#fff;background-color:#ec971f;border-color:#d58512}.btn-warning.active,.btn-warning:active,.open>.dropdown-toggle.btn-warning{color:#fff;background-color:#ec971f;border-color:#d58512}.btn-warning.active.focus,.btn-warning.active:focus,.btn-warning.active:hover,.btn-warning:active.focus,.btn-warning:active:focus,.btn-warning:active:hover,.open>.dropdown-toggle.btn-warning.focus,.open>.dropdown-toggle.btn-warning:focus,.open>.dropdown-toggle.btn-warning:hover{color:#fff;background-color:#d58512;border-color:#985f0d}.btn-warning.active,.btn-warning:active,.open>.dropdown-toggle.btn-warning{background-image:none}.btn-warning.disabled,.btn-warning.disabled.active,.btn-warning.disabled.focus,.btn-warning.disabled:active,.btn-warning.disabled:focus,.btn-warning.disabled:hover,.btn-warning[disabled],.btn-warning[disabled].active,.btn-warning[disabled].focus,.btn-warning[disabled]:active,.btn-warning[disabled]:focus,.btn-warning[disabled]:hover,fieldset[disabled] .btn-warning,fieldset[disabled] .btn-warning.active,fieldset[disabled] .btn-warning.focus,fieldset[disabled] .btn-warning:active,fieldset[disabled] .btn-warning:focus,fieldset[disabled] .btn-warning:hover{background-color:#f0ad4e;border-color:#eea236}.btn-warning .badge{color:#f0ad4e;background-color:#fff}.btn-danger{color:#fff;background-color:#d9534f;border-color:#d43f3a}.btn-danger.focus,.btn-danger:focus{color:#fff;background-color:#c9302c;border-color:#761c19}.btn-danger:hover{color:#fff;background-color:#c9302c;border-color:#ac2925}.btn-danger.active,.btn-danger:active,.open>.dropdown-toggle.btn-danger{color:#fff;background-color:#c9302c;border-color:#ac2925}.btn-danger.active.focus,.btn-danger.active:focus,.btn-danger.active:hover,.btn-danger:active.focus,.btn-danger:active:focus,.btn-danger:active:hover,.open>.dropdown-toggle.btn-danger.focus,.open>.dropdown-toggle.btn-danger:focus,.open>.dropdown-toggle.btn-danger:hover{color:#fff;background-color:#ac2925;border-color:#761c19}.btn-danger.active,.btn-danger:active,.open>.dropdown-toggle.btn-danger{background-image:none}.btn-danger.disabled,.btn-danger.disabled.active,.btn-danger.disabled.focus,.btn-danger.disabled:active,.btn-danger.disabled:focus,.btn-danger.disabled:hover,.btn-danger[disabled],.btn-danger[disabled].active,.btn-danger[disabled].focus,.btn-danger[disabled]:active,.btn-danger[disabled]:focus,.btn-danger[disabled]:hover,fieldset[disabled] .btn-danger,fieldset[disabled] .btn-danger.active,fieldset[disabled] .btn-danger.focus,fieldset[disabled] .btn-danger:active,fieldset[disabled] .btn-danger:focus,fieldset[disabled] .btn-danger:hover{background-color:#d9534f;border-color:#d43f3a}.btn-danger .badge{color:#d9534f;background-color:#fff}.btn-link{font-weight:400;color:#337ab7;border-radius:0}.btn-link,.btn-link.active,.btn-link:active,.btn-link[disabled],fieldset[disabled] .btn-link{background-color:transparent;-webkit-box-shadow:none;box-shadow:none}.btn-link,.btn-link:active,.btn-link:focus,.btn-link:hover{border-color:transparent}.btn-link:focus,.btn-link:hover{color:#23527c;text-decoration:underline;background-color:transparent}.btn-link[disabled]:focus,.btn-link[disabled]:hover,fieldset[disabled] .btn-link:focus,fieldset[disabled] .btn-link:hover{color:#777;text-decoration:none}.btn-group-lg>.btn,.btn-lg{padding:10px 16px;font-size:18px;line-height:1.3333333;border-radius:6px}.btn-group-sm>.btn,.btn-sm{padding:5px 10px;font-size:12px;line-height:1.5;border-radius:3px}.btn-group-xs>.btn,.btn-xs{padding:1px 5px;font-size:12px;line-height:1.5;border-radius:3px}.btn-block{display:block;width:100%}.btn-block+.btn-block{margin-top:5px}input[type=button].btn-block,input[type=reset].btn-block,input[type=submit].btn-block{width:100%}.fade{opacity:0;-webkit-transition:opacity .15s linear;-o-transition:opacity .15s linear;transition:opacity .15s linear}.fade.in{opacity:1}.collapse{display:none}.collapse.in{display:block}tr.collapse.in{display:table-row}tbody.collapse.in{display:table-row-group}.collapsing{position:relative;height:0;overflow:hidden;-webkit-transition-timing-function:ease;-o-transition-timing-function:ease;transition-timing-function:ease;-webkit-transition-duration:.35s;-o-transition-duration:.35s;transition-duration:.35s;-webkit-transition-property:height,visibility;-o-transition-property:height,visibility;transition-property:height,visibility}.caret{display:inline-block;width:0;height:0;margin-left:2px;vertical-align:middle;border-top:4px dashed;border-top:4px solid\9;border-right:4px solid transparent;border-left:4px solid transparent}.dropdown,.dropup{position:relative}.dropdown-toggle:focus{outline:0}.dropdown-menu{position:absolute;top:100%;left:0;z-index:1000;display:none;float:left;min-width:160px;padding:5px 0;margin:2px 0 0;font-size:14px;text-align:left;list-style:none;background-color:#fff;-webkit-background-clip:padding-box;background-clip:padding-box;border:1px solid #ccc;border:1px solid rgba(0,0,0,.15);border-radius:4px;-webkit-box-shadow:0 6px 12px rgba(0,0,0,.175);box-shadow:0 6px 12px rgba(0,0,0,.175)}.dropdown-menu.pull-right{right:0;left:auto}.dropdown-menu .divider{height:1px;margin:9px 0;overflow:hidden;background-color:#e5e5e5}.dropdown-menu>li>a{display:block;padding:3px 20px;clear:both;font-weight:400;line-height:1.42857143;color:#333;white-space:nowrap}.dropdown-menu>li>a:focus,.dropdown-menu>li>a:hover{color:#262626;text-decoration:none;background-color:#f5f5f5}.dropdown-menu>.active>a,.dropdown-menu>.active>a:focus,.dropdown-menu>.active>a:hover{color:#fff;text-decoration:none;background-color:#337ab7;outline:0}.dropdown-menu>.disabled>a,.dropdown-menu>.disabled>a:focus,.dropdown-menu>.disabled>a:hover{color:#777}.dropdown-menu>.disabled>a:focus,.dropdown-menu>.disabled>a:hover{text-decoration:none;cursor:not-allowed;background-color:transparent;background-image:none;filter:progid:DXImageTransform.Microsoft.gradient(enabled=false)}.open>.dropdown-menu{display:block}.open>a{outline:0}.dropdown-menu-right{right:0;left:auto}.dropdown-menu-left{right:auto;left:0}.dropdown-header{display:block;padding:3px 20px;font-size:12px;line-height:1.42857143;color:#777;white-space:nowrap}.dropdown-backdrop{position:fixed;top:0;right:0;bottom:0;left:0;z-index:990}.pull-right>.dropdown-menu{right:0;left:auto}.dropup .caret,.navbar-fixed-bottom .dropdown .caret{content:"";border-top:0;border-bottom:4px dashed;border-bottom:4px solid\9}.dropup .dropdown-menu,.navbar-fixed-bottom .dropdown .dropdown-menu{top:auto;bottom:100%;margin-bottom:2px}@media (min-width:768px){.navbar-right .dropdown-menu{right:0;left:auto}.navbar-right .dropdown-menu-left{right:auto;left:0}}.btn-group,.btn-group-vertical{position:relative;display:inline-block;vertical-align:middle}.btn-group-vertical>.btn,.btn-group>.btn{position:relative;float:left}.btn-group-vertical>.btn.active,.btn-group-vertical>.btn:active,.btn-group-vertical>.btn:focus,.btn-group-vertical>.btn:hover,.btn-group>.btn.active,.btn-group>.btn:active,.btn-group>.btn:focus,.btn-group>.btn:hover{z-index:2}.btn-group .btn+.btn,.btn-group .btn+.btn-group,.btn-group .btn-group+.btn,.btn-group .btn-group+.btn-group{margin-left:-1px}.btn-toolbar{margin-left:-5px}.btn-toolbar .btn,.btn-toolbar .btn-group,.btn-toolbar .input-group{float:left}.btn-toolbar>.btn,.btn-toolbar>.btn-group,.btn-toolbar>.input-group{margin-left:5px}.btn-group>.btn:not(:first-child):not(:last-child):not(.dropdown-toggle){border-radius:0}.btn-group>.btn:first-child{margin-left:0}.btn-group>.btn:first-child:not(:last-child):not(.dropdown-toggle){border-top-right-radius:0;border-bottom-right-radius:0}.btn-group>.btn:last-child:not(:first-child),.btn-group>.dropdown-toggle:not(:first-child){border-top-left-radius:0;border-bottom-left-radius:0}.btn-group>.btn-group{float:left}.btn-group>.btn-group:not(:first-child):not(:last-child)>.btn{border-radius:0}.btn-group>.btn-group:first-child:not(:last-child)>.btn:last-child,.btn-group>.btn-group:first-child:not(:last-child)>.dropdown-toggle{border-top-right-radius:0;border-bottom-right-radius:0}.btn-group>.btn-group:last-child:not(:first-child)>.btn:first-child{border-top-left-radius:0;border-bottom-left-radius:0}.btn-group .dropdown-toggle:active,.btn-group.open .dropdown-toggle{outline:0}.btn-group>.btn+.dropdown-toggle{padding-right:8px;padding-left:8px}.btn-group>.btn-lg+.dropdown-toggle{padding-right:12px;padding-left:12px}.btn-group.open .dropdown-toggle{-webkit-box-shadow:inset 0 3px 5px rgba(0,0,0,.125);box-shadow:inset 0 3px 5px rgba(0,0,0,.125)}.btn-group.open .dropdown-toggle.btn-link{-webkit-box-shadow:none;box-shadow:none}.btn .caret{margin-left:0}.btn-lg .caret{border-width:5px 5px 0;border-bottom-width:0}.dropup .btn-lg .caret{border-width:0 5px 5px}.btn-group-vertical>.btn,.btn-group-vertical>.btn-group,.btn-group-vertical>.btn-group>.btn{display:block;float:none;width:100%;max-width:100%}.btn-group-vertical>.btn-group>.btn{float:none}.btn-group-vertical>.btn+.btn,.btn-group-vertical>.btn+.btn-group,.btn-group-vertical>.btn-group+.btn,.btn-group-vertical>.btn-group+.btn-group{margin-top:-1px;margin-left:0}.btn-group-vertical>.btn:not(:first-child):not(:last-child){border-radius:0}.btn-group-vertical>.btn:first-child:not(:last-child){border-top-right-radius:4px;border-bottom-right-radius:0;border-bottom-left-radius:0}.btn-group-vertical>.btn:last-child:not(:first-child){border-top-left-radius:0;border-top-right-radius:0;border-bottom-left-radius:4px}.btn-group-vertical>.btn-group:not(:first-child):not(:last-child)>.btn{border-radius:0}.btn-group-vertical>.btn-group:first-child:not(:last-child)>.btn:last-child,.btn-group-vertical>.btn-group:first-child:not(:last-child)>.dropdown-toggle{border-bottom-right-radius:0;border-bottom-left-radius:0}.btn-group-vertical>.btn-group:last-child:not(:first-child)>.btn:first-child{border-top-left-radius:0;border-top-right-radius:0}.btn-group-justified{display:table;width:100%;table-layout:fixed;border-collapse:separate}.btn-group-justified>.btn,.btn-group-justified>.btn-group{display:table-cell;float:none;width:1%}.btn-group-justified>.btn-group .btn{width:100%}.btn-group-justified>.btn-group .dropdown-menu{left:auto}[data-toggle=buttons]>.btn input[type=checkbox],[data-toggle=buttons]>.btn input[type=radio],[data-toggle=buttons]>.btn-group>.btn input[type=checkbox],[data-toggle=buttons]>.btn-group>.btn input[type=radio]{position:absolute;clip:rect(0,0,0,0);pointer-events:none}.input-group{position:relative;display:table;border-collapse:separate}.input-group[class*=col-]{float:none;padding-right:0;padding-left:0}.input-group .form-control{position:relative;z-index:2;float:left;width:100%;margin-bottom:0}.input-group-lg>.form-control,.input-group-lg>.input-group-addon,.input-group-lg>.input-group-btn>.btn{height:46px;padding:10px 16px;font-size:18px;line-height:1.3333333;border-radius:6px}select.input-group-lg>.form-control,select.input-group-lg>.input-group-addon,select.input-group-lg>.input-group-btn>.btn{height:46px;line-height:46px}select[multiple].input-group-lg>.form-control,select[multiple].input-group-lg>.input-group-addon,select[multiple].input-group-lg>.input-group-btn>.btn,textarea.input-group-lg>.form-control,textarea.input-group-lg>.input-group-addon,textarea.input-group-lg>.input-group-btn>.btn{height:auto}.input-group-sm>.form-control,.input-group-sm>.input-group-addon,.input-group-sm>.input-group-btn>.btn{height:30px;padding:5px 10px;font-size:12px;line-height:1.5;border-radius:3px}select.input-group-sm>.form-control,select.input-group-sm>.input-group-addon,select.input-group-sm>.input-group-btn>.btn{height:30px;line-height:30px}select[multiple].input-group-sm>.form-control,select[multiple].input-group-sm>.input-group-addon,select[multiple].input-group-sm>.input-group-btn>.btn,textarea.input-group-sm>.form-control,textarea.input-group-sm>.input-group-addon,textarea.input-group-sm>.input-group-btn>.btn{height:auto}.input-group .form-control,.input-group-addon,.input-group-btn{display:table-cell}.input-group .form-control:not(:first-child):not(:last-child),.input-group-addon:not(:first-child):not(:last-child),.input-group-btn:not(:first-child):not(:last-child){border-radius:0}.input-group-addon,.input-group-btn{width:1%;white-space:nowrap;vertical-align:middle}.input-group-addon{padding:6px 12px;font-size:14px;font-weight:400;line-height:1;color:#555;text-align:center;background-color:#eee;border:1px solid #ccc;border-radius:4px}.input-group-addon.input-sm{padding:5px 10px;font-size:12px;border-radius:3px}.input-group-addon.input-lg{padding:10px 16px;font-size:18px;border-radius:6px}.input-group-addon input[type=checkbox],.input-group-addon input[type=radio]{margin-top:0}.input-group .form-control:first-child,.input-group-addon:first-child,.input-group-btn:first-child>.btn,.input-group-btn:first-child>.btn-group>.btn,.input-group-btn:first-child>.dropdown-toggle,.input-group-btn:last-child>.btn-group:not(:last-child)>.btn,.input-group-btn:last-child>.btn:not(:last-child):not(.dropdown-toggle){border-top-right-radius:0;border-bottom-right-radius:0}.input-group-addon:first-child{border-right:0}.input-group .form-control:last-child,.input-group-addon:last-child,.input-group-btn:first-child>.btn-group:not(:first-child)>.btn,.input-group-btn:first-child>.btn:not(:first-child),.input-group-btn:last-child>.btn,.input-group-btn:last-child>.btn-group>.btn,.input-group-btn:last-child>.dropdown-toggle{border-top-left-radius:0;border-bottom-left-radius:0}.input-group-addon:last-child{border-left:0}.input-group-btn{position:relative;font-size:0;white-space:nowrap}.input-group-btn>.btn{position:relative}.input-group-btn>.btn+.btn{margin-left:-1px}.input-group-btn>.btn:active,.input-group-btn>.btn:focus,.input-group-btn>.btn:hover{z-index:2}.input-group-btn:first-child>.btn,.input-group-btn:first-child>.btn-group{margin-right:-1px}.input-group-btn:last-child>.btn,.input-group-btn:last-child>.btn-group{z-index:2;margin-left:-1px}.nav{padding-left:0;margin-bottom:0;list-style:none}.nav>li{position:relative;display:block}.nav>li>a{position:relative;display:block;padding:10px 15px}.nav>li>a:focus,.nav>li>a:hover{text-decoration:none;background-color:#eee}.nav>li.disabled>a{color:#777}.nav>li.disabled>a:focus,.nav>li.disabled>a:hover{color:#777;text-decoration:none;cursor:not-allowed;background-color:transparent}.nav .open>a,.nav .open>a:focus,.nav .open>a:hover{background-color:#eee;border-color:#337ab7}.nav .nav-divider{height:1px;margin:9px 0;overflow:hidden;background-color:#e5e5e5}.nav>li>a>img{max-width:none}.nav-tabs{border-bottom:1px solid #ddd}.nav-tabs>li{float:left;margin-bottom:-1px}.nav-tabs>li>a{margin-right:2px;line-height:1.42857143;border:1px solid transparent;border-radius:4px 4px 0 0}.nav-tabs>li>a:hover{border-color:#eee #eee #ddd}.nav-tabs>li.active>a,.nav-tabs>li.active>a:focus,.nav-tabs>li.active>a:hover{color:#555;cursor:default;background-color:#fff;border:1px solid #ddd;border-bottom-color:transparent}.nav-tabs.nav-justified{width:100%;border-bottom:0}.nav-tabs.nav-justified>li{float:none}.nav-tabs.nav-justified>li>a{margin-bottom:5px;text-align:center}.nav-tabs.nav-justified>.dropdown .dropdown-menu{top:auto;left:auto}@media (min-width:768px){.nav-tabs.nav-justified>li{display:table-cell;width:1%}.nav-tabs.nav-justified>li>a{margin-bottom:0}}.nav-tabs.nav-justified>li>a{margin-right:0;border-radius:4px}.nav-tabs.nav-justified>.active>a,.nav-tabs.nav-justified>.active>a:focus,.nav-tabs.nav-justified>.active>a:hover{border:1px solid #ddd}@media (min-width:768px){.nav-tabs.nav-justified>li>a{border-bottom:1px solid #ddd;border-radius:4px 4px 0 0}.nav-tabs.nav-justified>.active>a,.nav-tabs.nav-justified>.active>a:focus,.nav-tabs.nav-justified>.active>a:hover{border-bottom-color:#fff}}.nav-pills>li{float:left}.nav-pills>li>a{border-radius:4px}.nav-pills>li+li{margin-left:2px}.nav-pills>li.active>a,.nav-pills>li.active>a:focus,.nav-pills>li.active>a:hover{color:#fff;background-color:#337ab7}.nav-stacked>li{float:none}.nav-stacked>li+li{margin-top:2px;margin-left:0}.nav-justified{width:100%}.nav-justified>li{float:none}.nav-justified>li>a{margin-bottom:5px;text-align:center}.nav-justified>.dropdown .dropdown-menu{top:auto;left:auto}@media (min-width:768px){.nav-justified>li{display:table-cell;width:1%}.nav-justified>li>a{margin-bottom:0}}.nav-tabs-justified{border-bottom:0}.nav-tabs-justified>li>a{margin-right:0;border-radius:4px}.nav-tabs-justified>.active>a,.nav-tabs-justified>.active>a:focus,.nav-tabs-justified>.active>a:hover{border:1px solid #ddd}@media (min-width:768px){.nav-tabs-justified>li>a{border-bottom:1px solid #ddd;border-radius:4px 4px 0 0}.nav-tabs-justified>.active>a,.nav-tabs-justified>.active>a:focus,.nav-tabs-justified>.active>a:hover{border-bottom-color:#fff}}.tab-content>.tab-pane{display:none}.tab-content>.active{display:block}.nav-tabs .dropdown-menu{margin-top:-1px;border-top-left-radius:0;border-top-right-radius:0}.navbar{position:relative;min-height:50px;margin-bottom:20px;border:1px solid transparent}@media (min-width:768px){.navbar{border-radius:4px}}@media (min-width:768px){.navbar-header{float:left}}.navbar-collapse{padding-right:15px;padding-left:15px;overflow-x:visible;-webkit-overflow-scrolling:touch;border-top:1px solid transparent;-webkit-box-shadow:inset 0 1px 0 rgba(255,255,255,.1);box-shadow:inset 0 1px 0 rgba(255,255,255,.1)}.navbar-collapse.in{overflow-y:auto}@media (min-width:768px){.navbar-collapse{width:auto;border-top:0;-webkit-box-shadow:none;box-shadow:none}.navbar-collapse.collapse{display:block!important;height:auto!important;padding-bottom:0;overflow:visible!important}.navbar-collapse.in{overflow-y:visible}.navbar-fixed-bottom .navbar-collapse,.navbar-fixed-top .navbar-collapse,.navbar-static-top .navbar-collapse{padding-right:0;padding-left:0}}.navbar-fixed-bottom .navbar-collapse,.navbar-fixed-top .navbar-collapse{max-height:340px}@media (max-device-width:480px) and (orientation:landscape){.navbar-fixed-bottom .navbar-collapse,.navbar-fixed-top .navbar-collapse{max-height:200px}}.container-fluid>.navbar-collapse,.container-fluid>.navbar-header,.container>.navbar-collapse,.container>.navbar-header{margin-right:-15px;margin-left:-15px}@media (min-width:768px){.container-fluid>.navbar-collapse,.container-fluid>.navbar-header,.container>.navbar-collapse,.container>.navbar-header{margin-right:0;margin-left:0}}.navbar-static-top{z-index:1000;border-width:0 0 1px}@media (min-width:768px){.navbar-static-top{border-radius:0}}.navbar-fixed-bottom,.navbar-fixed-top{position:fixed;right:0;left:0;z-index:1030}@media (min-width:768px){.navbar-fixed-bottom,.navbar-fixed-top{border-radius:0}}.navbar-fixed-top{top:0;border-width:0 0 1px}.navbar-fixed-bottom{bottom:0;margin-bottom:0;border-width:1px 0 0}.navbar-brand{float:left;height:50px;padding:15px 15px;font-size:18px;line-height:20px}.navbar-brand:focus,.navbar-brand:hover{text-decoration:none}.navbar-brand>img{display:block}@media (min-width:768px){.navbar>.container .navbar-brand,.navbar>.container-fluid .navbar-brand{margin-left:-15px}}.navbar-toggle{position:relative;float:right;padding:9px 10px;margin-top:8px;margin-right:15px;margin-bottom:8px;background-color:transparent;background-image:none;border:1px solid transparent;border-radius:4px}.navbar-toggle:focus{outline:0}.navbar-toggle .icon-bar{display:block;width:22px;height:2px;border-radius:1px}.navbar-toggle .icon-bar+.icon-bar{margin-top:4px}@media (min-width:768px){.navbar-toggle{display:none}}.navbar-nav{margin:7.5px -15px}.navbar-nav>li>a{padding-top:10px;padding-bottom:10px;line-height:20px}@media (max-width:767px){.navbar-nav .open .dropdown-menu{position:static;float:none;width:auto;margin-top:0;background-color:transparent;border:0;-webkit-box-shadow:none;box-shadow:none}.navbar-nav .open .dropdown-menu .dropdown-header,.navbar-nav .open .dropdown-menu>li>a{padding:5px 15px 5px 25px}.navbar-nav .open .dropdown-menu>li>a{line-height:20px}.navbar-nav .open .dropdown-menu>li>a:focus,.navbar-nav .open .dropdown-menu>li>a:hover{background-image:none}}@media (min-width:768px){.navbar-nav{float:left;margin:0}.navbar-nav>li{float:left}.navbar-nav>li>a{padding-top:15px;padding-bottom:15px}}.navbar-form{padding:10px 15px;margin-top:8px;margin-right:-15px;margin-bottom:8px;margin-left:-15px;border-top:1px solid transparent;border-bottom:1px solid transparent;-webkit-box-shadow:inset 0 1px 0 rgba(255,255,255,.1),0 1px 0 rgba(255,255,255,.1);box-shadow:inset 0 1px 0 rgba(255,255,255,.1),0 1px 0 rgba(255,255,255,.1)}@media (min-width:768px){.navbar-form .form-group{display:inline-block;margin-bottom:0;vertical-align:middle}.navbar-form .form-control{display:inline-block;width:auto;vertical-align:middle}.navbar-form .form-control-static{display:inline-block}.navbar-form .input-group{display:inline-table;vertical-align:middle}.navbar-form .input-group .form-control,.navbar-form .input-group .input-group-addon,.navbar-form .input-group .input-group-btn{width:auto}.navbar-form .input-group>.form-control{width:100%}.navbar-form .control-label{margin-bottom:0;vertical-align:middle}.navbar-form .checkbox,.navbar-form .radio{display:inline-block;margin-top:0;margin-bottom:0;vertical-align:middle}.navbar-form .checkbox label,.navbar-form .radio label{padding-left:0}.navbar-form .checkbox input[type=checkbox],.navbar-form .radio input[type=radio]{position:relative;margin-left:0}.navbar-form .has-feedback .form-control-feedback{top:0}}@media (max-width:767px){.navbar-form .form-group{margin-bottom:5px}.navbar-form .form-group:last-child{margin-bottom:0}}@media (min-width:768px){.navbar-form{width:auto;padding-top:0;padding-bottom:0;margin-right:0;margin-left:0;border:0;-webkit-box-shadow:none;box-shadow:none}}.navbar-nav>li>.dropdown-menu{margin-top:0;border-top-left-radius:0;border-top-right-radius:0}.navbar-fixed-bottom .navbar-nav>li>.dropdown-menu{margin-bottom:0;border-top-left-radius:4px;border-top-right-radius:4px;border-bottom-right-radius:0;border-bottom-left-radius:0}.navbar-btn{margin-top:8px;margin-bottom:8px}.navbar-btn.btn-sm{margin-top:10px;margin-bottom:10px}.navbar-btn.btn-xs{margin-top:14px;margin-bottom:14px}.navbar-text{margin-top:15px;margin-bottom:15px}@media (min-width:768px){.navbar-text{float:left;margin-right:15px;margin-left:15px}}@media (min-width:768px){.navbar-left{float:left!important}.navbar-right{float:right!important;margin-right:-15px}.navbar-right~.navbar-right{margin-right:0}}.navbar-default{background-color:#f8f8f8;border-color:#e7e7e7}.navbar-default .navbar-brand{color:#777}.navbar-default .navbar-brand:focus,.navbar-default .navbar-brand:hover{color:#5e5e5e;background-color:transparent}.navbar-default .navbar-text{color:#777}.navbar-default .navbar-nav>li>a{color:#777}.navbar-default .navbar-nav>li>a:focus,.navbar-default .navbar-nav>li>a:hover{color:#333;background-color:transparent}.navbar-default .navbar-nav>.active>a,.navbar-default .navbar-nav>.active>a:focus,.navbar-default .navbar-nav>.active>a:hover{color:#555;background-color:#e7e7e7}.navbar-default .navbar-nav>.disabled>a,.navbar-default .navbar-nav>.disabled>a:focus,.navbar-default .navbar-nav>.disabled>a:hover{color:#ccc;background-color:transparent}.navbar-default .navbar-toggle{border-color:#ddd}.navbar-default .navbar-toggle:focus,.navbar-default .navbar-toggle:hover{background-color:#ddd}.navbar-default .navbar-toggle .icon-bar{background-color:#888}.navbar-default .navbar-collapse,.navbar-default .navbar-form{border-color:#e7e7e7}.navbar-default .navbar-nav>.open>a,.navbar-default .navbar-nav>.open>a:focus,.navbar-default .navbar-nav>.open>a:hover{color:#555;background-color:#e7e7e7}@media (max-width:767px){.navbar-default .navbar-nav .open .dropdown-menu>li>a{color:#777}.navbar-default .navbar-nav .open .dropdown-menu>li>a:focus,.navbar-default .navbar-nav .open .dropdown-menu>li>a:hover{color:#333;background-color:transparent}.navbar-default .navbar-nav .open .dropdown-menu>.active>a,.navbar-default .navbar-nav .open .dropdown-menu>.active>a:focus,.navbar-default .navbar-nav .open .dropdown-menu>.active>a:hover{color:#555;background-color:#e7e7e7}.navbar-default .navbar-nav .open .dropdown-menu>.disabled>a,.navbar-default .navbar-nav .open .dropdown-menu>.disabled>a:focus,.navbar-default .navbar-nav .open .dropdown-menu>.disabled>a:hover{color:#ccc;background-color:transparent}}.navbar-default .navbar-link{color:#777}.navbar-default .navbar-link:hover{color:#333}.navbar-default .btn-link{color:#777}.navbar-default .btn-link:focus,.navbar-default .btn-link:hover{color:#333}.navbar-default .btn-link[disabled]:focus,.navbar-default .btn-link[disabled]:hover,fieldset[disabled] .navbar-default .btn-link:focus,fieldset[disabled] .navbar-default .btn-link:hover{color:#ccc}.navbar-inverse{background-color:#222;border-color:#080808}.navbar-inverse .navbar-brand{color:#9d9d9d}.navbar-inverse .navbar-brand:focus,.navbar-inverse .navbar-brand:hover{color:#fff;background-color:transparent}.navbar-inverse .navbar-text{color:#9d9d9d}.navbar-inverse .navbar-nav>li>a{color:#9d9d9d}.navbar-inverse .navbar-nav>li>a:focus,.navbar-inverse .navbar-nav>li>a:hover{color:#fff;background-color:transparent}.navbar-inverse .navbar-nav>.active>a,.navbar-inverse .navbar-nav>.active>a:focus,.navbar-inverse .navbar-nav>.active>a:hover{color:#fff;background-color:#080808}.navbar-inverse .navbar-nav>.disabled>a,.navbar-inverse .navbar-nav>.disabled>a:focus,.navbar-inverse .navbar-nav>.disabled>a:hover{color:#444;background-color:transparent}.navbar-inverse .navbar-toggle{border-color:#333}.navbar-inverse .navbar-toggle:focus,.navbar-inverse .navbar-toggle:hover{background-color:#333}.navbar-inverse .navbar-toggle .icon-bar{background-color:#fff}.navbar-inverse .navbar-collapse,.navbar-inverse .navbar-form{border-color:#101010}.navbar-inverse .navbar-nav>.open>a,.navbar-inverse .navbar-nav>.open>a:focus,.navbar-inverse .navbar-nav>.open>a:hover{color:#fff;background-color:#080808}@media (max-width:767px){.navbar-inverse .navbar-nav .open .dropdown-menu>.dropdown-header{border-color:#080808}.navbar-inverse .navbar-nav .open .dropdown-menu .divider{background-color:#080808}.navbar-inverse .navbar-nav .open .dropdown-menu>li>a{color:#9d9d9d}.navbar-inverse .navbar-nav .open .dropdown-menu>li>a:focus,.navbar-inverse .navbar-nav .open .dropdown-menu>li>a:hover{color:#fff;background-color:transparent}.navbar-inverse .navbar-nav .open .dropdown-menu>.active>a,.navbar-inverse .navbar-nav .open .dropdown-menu>.active>a:focus,.navbar-inverse .navbar-nav .open .dropdown-menu>.active>a:hover{color:#fff;background-color:#080808}.navbar-inverse .navbar-nav .open .dropdown-menu>.disabled>a,.navbar-inverse .navbar-nav .open .dropdown-menu>.disabled>a:focus,.navbar-inverse .navbar-nav .open .dropdown-menu>.disabled>a:hover{color:#444;background-color:transparent}}.navbar-inverse .navbar-link{color:#9d9d9d}.navbar-inverse .navbar-link:hover{color:#fff}.navbar-inverse .btn-link{color:#9d9d9d}.navbar-inverse .btn-link:focus,.navbar-inverse .btn-link:hover{color:#fff}.navbar-inverse .btn-link[disabled]:focus,.navbar-inverse .btn-link[disabled]:hover,fieldset[disabled] .navbar-inverse .btn-link:focus,fieldset[disabled] .navbar-inverse .btn-link:hover{color:#444}.breadcrumb{padding:8px 15px;margin-bottom:20px;list-style:none;background-color:#f5f5f5;border-radius:4px}.breadcrumb>li{display:inline-block}.breadcrumb>li+li:before{padding:0 5px;color:#ccc;content:"/\00a0"}.breadcrumb>.active{color:#777}.pagination{display:inline-block;padding-left:0;margin:20px 0;border-radius:4px}.pagination>li{display:inline}.pagination>li>a,.pagination>li>span{position:relative;float:left;padding:6px 12px;margin-left:-1px;line-height:1.42857143;color:#337ab7;text-decoration:none;background-color:#fff;border:1px solid #ddd}.pagination>li:first-child>a,.pagination>li:first-child>span{margin-left:0;border-top-left-radius:4px;border-bottom-left-radius:4px}.pagination>li:last-child>a,.pagination>li:last-child>span{border-top-right-radius:4px;border-bottom-right-radius:4px}.pagination>li>a:focus,.pagination>li>a:hover,.pagination>li>span:focus,.pagination>li>span:hover{z-index:3;color:#23527c;background-color:#eee;border-color:#ddd}.pagination>.active>a,.pagination>.active>a:focus,.pagination>.active>a:hover,.pagination>.active>span,.pagination>.active>span:focus,.pagination>.active>span:hover{z-index:2;color:#fff;cursor:default;background-color:#337ab7;border-color:#337ab7}.pagination>.disabled>a,.pagination>.disabled>a:focus,.pagination>.disabled>a:hover,.pagination>.disabled>span,.pagination>.disabled>span:focus,.pagination>.disabled>span:hover{color:#777;cursor:not-allowed;background-color:#fff;border-color:#ddd}.pagination-lg>li>a,.pagination-lg>li>span{padding:10px 16px;font-size:18px;line-height:1.3333333}.pagination-lg>li:first-child>a,.pagination-lg>li:first-child>span{border-top-left-radius:6px;border-bottom-left-radius:6px}.pagination-lg>li:last-child>a,.pagination-lg>li:last-child>span{border-top-right-radius:6px;border-bottom-right-radius:6px}.pagination-sm>li>a,.pagination-sm>li>span{padding:5px 10px;font-size:12px;line-height:1.5}.pagination-sm>li:first-child>a,.pagination-sm>li:first-child>span{border-top-left-radius:3px;border-bottom-left-radius:3px}.pagination-sm>li:last-child>a,.pagination-sm>li:last-child>span{border-top-right-radius:3px;border-bottom-right-radius:3px}.pager{padding-left:0;margin:20px 0;text-align:center;list-style:none}.pager li{display:inline}.pager li>a,.pager li>span{display:inline-block;padding:5px 14px;background-color:#fff;border:1px solid #ddd;border-radius:15px}.pager li>a:focus,.pager li>a:hover{text-decoration:none;background-color:#eee}.pager .next>a,.pager .next>span{float:right}.pager .previous>a,.pager .previous>span{float:left}.pager .disabled>a,.pager .disabled>a:focus,.pager .disabled>a:hover,.pager .disabled>span{color:#777;cursor:not-allowed;background-color:#fff}.label{display:inline;padding:.2em .6em .3em;font-size:75%;font-weight:700;line-height:1;color:#fff;text-align:center;white-space:nowrap;vertical-align:baseline;border-radius:.25em}a.label:focus,a.label:hover{color:#fff;text-decoration:none;cursor:pointer}.label:empty{display:none}.btn .label{position:relative;top:-1px}.label-default{background-color:#777}.label-default[href]:focus,.label-default[href]:hover{background-color:#5e5e5e}.label-primary{background-color:#337ab7}.label-primary[href]:focus,.label-primary[href]:hover{background-color:#286090}.label-success{background-color:#5cb85c}.label-success[href]:focus,.label-success[href]:hover{background-color:#449d44}.label-info{background-color:#5bc0de}.label-info[href]:focus,.label-info[href]:hover{background-color:#31b0d5}.label-warning{background-color:#f0ad4e}.label-warning[href]:focus,.label-warning[href]:hover{background-color:#ec971f}.label-danger{background-color:#d9534f}.label-danger[href]:focus,.label-danger[href]:hover{background-color:#c9302c}.badge{display:inline-block;min-width:10px;padding:3px 7px;font-size:12px;font-weight:700;line-height:1;color:#fff;text-align:center;white-space:nowrap;vertical-align:middle;background-color:#777;border-radius:10px}.badge:empty{display:none}.btn .badge{position:relative;top:-1px}.btn-group-xs>.btn .badge,.btn-xs .badge{top:0;padding:1px 5px}a.badge:focus,a.badge:hover{color:#fff;text-decoration:none;cursor:pointer}.list-group-item.active>.badge,.nav-pills>.active>a>.badge{color:#337ab7;background-color:#fff}.list-group-item>.badge{float:right}.list-group-item>.badge+.badge{margin-right:5px}.nav-pills>li>a>.badge{margin-left:3px}.jumbotron{padding-top:30px;padding-bottom:30px;margin-bottom:30px;color:inherit;background-color:#eee}.jumbotron .h1,.jumbotron h1{color:inherit}.jumbotron p{margin-bottom:15px;font-size:21px;font-weight:200}.jumbotron>hr{border-top-color:#d5d5d5}.container .jumbotron,.container-fluid .jumbotron{border-radius:6px}.jumbotron .container{max-width:100%}@media screen and (min-width:768px){.jumbotron{padding-top:48px;padding-bottom:48px}.container .jumbotron,.container-fluid .jumbotron{padding-right:60px;padding-left:60px}.jumbotron .h1,.jumbotron h1{font-size:63px}}.thumbnail{display:block;padding:4px;margin-bottom:20px;line-height:1.42857143;background-color:#fff;border:1px solid #ddd;border-radius:4px;-webkit-transition:border .2s ease-in-out;-o-transition:border .2s ease-in-out;transition:border .2s ease-in-out}.thumbnail a>img,.thumbnail>img{margin-right:auto;margin-left:auto}a.thumbnail.active,a.thumbnail:focus,a.thumbnail:hover{border-color:#337ab7}.thumbnail .caption{padding:9px;color:#333}.alert{padding:15px;margin-bottom:20px;border:1px solid transparent;border-radius:4px}.alert h4{margin-top:0;color:inherit}.alert .alert-link{font-weight:700}.alert>p,.alert>ul{margin-bottom:0}.alert>p+p{margin-top:5px}.alert-dismissable,.alert-dismissible{padding-right:35px}.alert-dismissable .close,.alert-dismissible .close{position:relative;top:-2px;right:-21px;color:inherit}.alert-success{color:#3c763d;background-color:#dff0d8;border-color:#d6e9c6}.alert-success hr{border-top-color:#c9e2b3}.alert-success .alert-link{color:#2b542c}.alert-info{color:#31708f;background-color:#d9edf7;border-color:#bce8f1}.alert-info hr{border-top-color:#a6e1ec}.alert-info .alert-link{color:#245269}.alert-warning{color:#8a6d3b;background-color:#fcf8e3;border-color:#faebcc}.alert-warning hr{border-top-color:#f7e1b5}.alert-warning .alert-link{color:#66512c}.alert-danger{color:#a94442;background-color:#f2dede;border-color:#ebccd1}.alert-danger hr{border-top-color:#e4b9c0}.alert-danger .alert-link{color:#843534}@-webkit-keyframes progress-bar-stripes{from{background-position:40px 0}to{background-position:0 0}}@-o-keyframes progress-bar-stripes{from{background-position:40px 0}to{background-position:0 0}}@keyframes progress-bar-stripes{from{background-position:40px 0}to{background-position:0 0}}.progress{height:20px;margin-bottom:20px;overflow:hidden;background-color:#f5f5f5;border-radius:4px;-webkit-box-shadow:inset 0 1px 2px rgba(0,0,0,.1);box-shadow:inset 0 1px 2px rgba(0,0,0,.1)}.progress-bar{float:left;width:0;height:100%;font-size:12px;line-height:20px;color:#fff;text-align:center;background-color:#337ab7;-webkit-box-shadow:inset 0 -1px 0 rgba(0,0,0,.15);box-shadow:inset 0 -1px 0 rgba(0,0,0,.15);-webkit-transition:width .6s ease;-o-transition:width .6s ease;transition:width .6s ease}.progress-bar-striped,.progress-striped .progress-bar{background-image:-webkit-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:-o-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);-webkit-background-size:40px 40px;background-size:40px 40px}.progress-bar.active,.progress.active .progress-bar{-webkit-animation:progress-bar-stripes 2s linear infinite;-o-animation:progress-bar-stripes 2s linear infinite;animation:progress-bar-stripes 2s linear infinite}.progress-bar-success{background-color:#5cb85c}.progress-striped .progress-bar-success{background-image:-webkit-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:-o-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent)}.progress-bar-info{background-color:#5bc0de}.progress-striped .progress-bar-info{background-image:-webkit-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:-o-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent)}.progress-bar-warning{background-color:#f0ad4e}.progress-striped .progress-bar-warning{background-image:-webkit-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:-o-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent)}.progress-bar-danger{background-color:#d9534f}.progress-striped .progress-bar-danger{background-image:-webkit-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:-o-linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent);background-image:linear-gradient(45deg,rgba(255,255,255,.15) 25%,transparent 25%,transparent 50%,rgba(255,255,255,.15) 50%,rgba(255,255,255,.15) 75%,transparent 75%,transparent)}.media{margin-top:15px}.media:first-child{margin-top:0}.media,.media-body{overflow:hidden;zoom:1}.media-body{width:10000px}.media-object{display:block}.media-object.img-thumbnail{max-width:none}.media-right,.media>.pull-right{padding-left:10px}.media-left,.media>.pull-left{padding-right:10px}.media-body,.media-left,.media-right{display:table-cell;vertical-align:top}.media-middle{vertical-align:middle}.media-bottom{vertical-align:bottom}.media-heading{margin-top:0;margin-bottom:5px}.media-list{padding-left:0;list-style:none}.list-group{padding-left:0;margin-bottom:20px}.list-group-item{position:relative;display:block;padding:10px 15px;margin-bottom:-1px;background-color:#fff;border:1px solid #ddd}.list-group-item:first-child{border-top-left-radius:4px;border-top-right-radius:4px}.list-group-item:last-child{margin-bottom:0;border-bottom-right-radius:4px;border-bottom-left-radius:4px}a.list-group-item,button.list-group-item{color:#555}a.list-group-item .list-group-item-heading,button.list-group-item .list-group-item-heading{color:#333}a.list-group-item:focus,a.list-group-item:hover,button.list-group-item:focus,button.list-group-item:hover{color:#555;text-decoration:none;background-color:#f5f5f5}button.list-group-item{width:100%;text-align:left}.list-group-item.disabled,.list-group-item.disabled:focus,.list-group-item.disabled:hover{color:#777;cursor:not-allowed;background-color:#eee}.list-group-item.disabled .list-group-item-heading,.list-group-item.disabled:focus .list-group-item-heading,.list-group-item.disabled:hover .list-group-item-heading{color:inherit}.list-group-item.disabled .list-group-item-text,.list-group-item.disabled:focus .list-group-item-text,.list-group-item.disabled:hover .list-group-item-text{color:#777}.list-group-item.active,.list-group-item.active:focus,.list-group-item.active:hover{z-index:2;color:#fff;background-color:#337ab7;border-color:#337ab7}.list-group-item.active .list-group-item-heading,.list-group-item.active .list-group-item-heading>.small,.list-group-item.active .list-group-item-heading>small,.list-group-item.active:focus .list-group-item-heading,.list-group-item.active:focus .list-group-item-heading>.small,.list-group-item.active:focus .list-group-item-heading>small,.list-group-item.active:hover .list-group-item-heading,.list-group-item.active:hover .list-group-item-heading>.small,.list-group-item.active:hover .list-group-item-heading>small{color:inherit}.list-group-item.active .list-group-item-text,.list-group-item.active:focus .list-group-item-text,.list-group-item.active:hover .list-group-item-text{color:#c7ddef}.list-group-item-success{color:#3c763d;background-color:#dff0d8}a.list-group-item-success,button.list-group-item-success{color:#3c763d}a.list-group-item-success .list-group-item-heading,button.list-group-item-success .list-group-item-heading{color:inherit}a.list-group-item-success:focus,a.list-group-item-success:hover,button.list-group-item-success:focus,button.list-group-item-success:hover{color:#3c763d;background-color:#d0e9c6}a.list-group-item-success.active,a.list-group-item-success.active:focus,a.list-group-item-success.active:hover,button.list-group-item-success.active,button.list-group-item-success.active:focus,button.list-group-item-success.active:hover{color:#fff;background-color:#3c763d;border-color:#3c763d}.list-group-item-info{color:#31708f;background-color:#d9edf7}a.list-group-item-info,button.list-group-item-info{color:#31708f}a.list-group-item-info .list-group-item-heading,button.list-group-item-info .list-group-item-heading{color:inherit}a.list-group-item-info:focus,a.list-group-item-info:hover,button.list-group-item-info:focus,button.list-group-item-info:hover{color:#31708f;background-color:#c4e3f3}a.list-group-item-info.active,a.list-group-item-info.active:focus,a.list-group-item-info.active:hover,button.list-group-item-info.active,button.list-group-item-info.active:focus,button.list-group-item-info.active:hover{color:#fff;background-color:#31708f;border-color:#31708f}.list-group-item-warning{color:#8a6d3b;background-color:#fcf8e3}a.list-group-item-warning,button.list-group-item-warning{color:#8a6d3b}a.list-group-item-warning .list-group-item-heading,button.list-group-item-warning .list-group-item-heading{color:inherit}a.list-group-item-warning:focus,a.list-group-item-warning:hover,button.list-group-item-warning:focus,button.list-group-item-warning:hover{color:#8a6d3b;background-color:#faf2cc}a.list-group-item-warning.active,a.list-group-item-warning.active:focus,a.list-group-item-warning.active:hover,button.list-group-item-warning.active,button.list-group-item-warning.active:focus,button.list-group-item-warning.active:hover{color:#fff;background-color:#8a6d3b;border-color:#8a6d3b}.list-group-item-danger{color:#a94442;background-color:#f2dede}a.list-group-item-danger,button.list-group-item-danger{color:#a94442}a.list-group-item-danger .list-group-item-heading,button.list-group-item-danger .list-group-item-heading{color:inherit}a.list-group-item-danger:focus,a.list-group-item-danger:hover,button.list-group-item-danger:focus,button.list-group-item-danger:hover{color:#a94442;background-color:#ebcccc}a.list-group-item-danger.active,a.list-group-item-danger.active:focus,a.list-group-item-danger.active:hover,button.list-group-item-danger.active,button.list-group-item-danger.active:focus,button.list-group-item-danger.active:hover{color:#fff;background-color:#a94442;border-color:#a94442}.list-group-item-heading{margin-top:0;margin-bottom:5px}.list-group-item-text{margin-bottom:0;line-height:1.3}.panel{margin-bottom:20px;background-color:#fff;border:1px solid transparent;border-radius:4px;-webkit-box-shadow:0 1px 1px rgba(0,0,0,.05);box-shadow:0 1px 1px rgba(0,0,0,.05)}.panel-body{padding:15px}.panel-heading{padding:10px 15px;border-bottom:1px solid transparent;border-top-left-radius:3px;border-top-right-radius:3px}.panel-heading>.dropdown .dropdown-toggle{color:inherit}.panel-title{margin-top:0;margin-bottom:0;font-size:16px;color:inherit}.panel-title>.small,.panel-title>.small>a,.panel-title>a,.panel-title>small,.panel-title>small>a{color:inherit}.panel-footer{padding:10px 15px;background-color:#f5f5f5;border-top:1px solid #ddd;border-bottom-right-radius:3px;border-bottom-left-radius:3px}.panel>.list-group,.panel>.panel-collapse>.list-group{margin-bottom:0}.panel>.list-group .list-group-item,.panel>.panel-collapse>.list-group .list-group-item{border-width:1px 0;border-radius:0}.panel>.list-group:first-child .list-group-item:first-child,.panel>.panel-collapse>.list-group:first-child .list-group-item:first-child{border-top:0;border-top-left-radius:3px;border-top-right-radius:3px}.panel>.list-group:last-child .list-group-item:last-child,.panel>.panel-collapse>.list-group:last-child .list-group-item:last-child{border-bottom:0;border-bottom-right-radius:3px;border-bottom-left-radius:3px}.panel>.panel-heading+.panel-collapse>.list-group .list-group-item:first-child{border-top-left-radius:0;border-top-right-radius:0}.panel-heading+.list-group .list-group-item:first-child{border-top-width:0}.list-group+.panel-footer{border-top-width:0}.panel>.panel-collapse>.table,.panel>.table,.panel>.table-responsive>.table{margin-bottom:0}.panel>.panel-collapse>.table caption,.panel>.table caption,.panel>.table-responsive>.table caption{padding-right:15px;padding-left:15px}.panel>.table-responsive:first-child>.table:first-child,.panel>.table:first-child{border-top-left-radius:3px;border-top-right-radius:3px}.panel>.table-responsive:first-child>.table:first-child>tbody:first-child>tr:first-child,.panel>.table-responsive:first-child>.table:first-child>thead:first-child>tr:first-child,.panel>.table:first-child>tbody:first-child>tr:first-child,.panel>.table:first-child>thead:first-child>tr:first-child{border-top-left-radius:3px;border-top-right-radius:3px}.panel>.table-responsive:first-child>.table:first-child>tbody:first-child>tr:first-child td:first-child,.panel>.table-responsive:first-child>.table:first-child>tbody:first-child>tr:first-child th:first-child,.panel>.table-responsive:first-child>.table:first-child>thead:first-child>tr:first-child td:first-child,.panel>.table-responsive:first-child>.table:first-child>thead:first-child>tr:first-child th:first-child,.panel>.table:first-child>tbody:first-child>tr:first-child td:first-child,.panel>.table:first-child>tbody:first-child>tr:first-child th:first-child,.panel>.table:first-child>thead:first-child>tr:first-child td:first-child,.panel>.table:first-child>thead:first-child>tr:first-child th:first-child{border-top-left-radius:3px}.panel>.table-responsive:first-child>.table:first-child>tbody:first-child>tr:first-child td:last-child,.panel>.table-responsive:first-child>.table:first-child>tbody:first-child>tr:first-child th:last-child,.panel>.table-responsive:first-child>.table:first-child>thead:first-child>tr:first-child td:last-child,.panel>.table-responsive:first-child>.table:first-child>thead:first-child>tr:first-child th:last-child,.panel>.table:first-child>tbody:first-child>tr:first-child td:last-child,.panel>.table:first-child>tbody:first-child>tr:first-child th:last-child,.panel>.table:first-child>thead:first-child>tr:first-child td:last-child,.panel>.table:first-child>thead:first-child>tr:first-child th:last-child{border-top-right-radius:3px}.panel>.table-responsive:last-child>.table:last-child,.panel>.table:last-child{border-bottom-right-radius:3px;border-bottom-left-radius:3px}.panel>.table-responsive:last-child>.table:last-child>tbody:last-child>tr:last-child,.panel>.table-responsive:last-child>.table:last-child>tfoot:last-child>tr:last-child,.panel>.table:last-child>tbody:last-child>tr:last-child,.panel>.table:last-child>tfoot:last-child>tr:last-child{border-bottom-right-radius:3px;border-bottom-left-radius:3px}.panel>.table-responsive:last-child>.table:last-child>tbody:last-child>tr:last-child td:first-child,.panel>.table-responsive:last-child>.table:last-child>tbody:last-child>tr:last-child th:first-child,.panel>.table-responsive:last-child>.table:last-child>tfoot:last-child>tr:last-child td:first-child,.panel>.table-responsive:last-child>.table:last-child>tfoot:last-child>tr:last-child th:first-child,.panel>.table:last-child>tbody:last-child>tr:last-child td:first-child,.panel>.table:last-child>tbody:last-child>tr:last-child th:first-child,.panel>.table:last-child>tfoot:last-child>tr:last-child td:first-child,.panel>.table:last-child>tfoot:last-child>tr:last-child th:first-child{border-bottom-left-radius:3px}.panel>.table-responsive:last-child>.table:last-child>tbody:last-child>tr:last-child td:last-child,.panel>.table-responsive:last-child>.table:last-child>tbody:last-child>tr:last-child th:last-child,.panel>.table-responsive:last-child>.table:last-child>tfoot:last-child>tr:last-child td:last-child,.panel>.table-responsive:last-child>.table:last-child>tfoot:last-child>tr:last-child th:last-child,.panel>.table:last-child>tbody:last-child>tr:last-child td:last-child,.panel>.table:last-child>tbody:last-child>tr:last-child th:last-child,.panel>.table:last-child>tfoot:last-child>tr:last-child td:last-child,.panel>.table:last-child>tfoot:last-child>tr:last-child th:last-child{border-bottom-right-radius:3px}.panel>.panel-body+.table,.panel>.panel-body+.table-responsive,.panel>.table+.panel-body,.panel>.table-responsive+.panel-body{border-top:1px solid #ddd}.panel>.table>tbody:first-child>tr:first-child td,.panel>.table>tbody:first-child>tr:first-child th{border-top:0}.panel>.table-bordered,.panel>.table-responsive>.table-bordered{border:0}.panel>.table-bordered>tbody>tr>td:first-child,.panel>.table-bordered>tbody>tr>th:first-child,.panel>.table-bordered>tfoot>tr>td:first-child,.panel>.table-bordered>tfoot>tr>th:first-child,.panel>.table-bordered>thead>tr>td:first-child,.panel>.table-bordered>thead>tr>th:first-child,.panel>.table-responsive>.table-bordered>tbody>tr>td:first-child,.panel>.table-responsive>.table-bordered>tbody>tr>th:first-child,.panel>.table-responsive>.table-bordered>tfoot>tr>td:first-child,.panel>.table-responsive>.table-bordered>tfoot>tr>th:first-child,.panel>.table-responsive>.table-bordered>thead>tr>td:first-child,.panel>.table-responsive>.table-bordered>thead>tr>th:first-child{border-left:0}.panel>.table-bordered>tbody>tr>td:last-child,.panel>.table-bordered>tbody>tr>th:last-child,.panel>.table-bordered>tfoot>tr>td:last-child,.panel>.table-bordered>tfoot>tr>th:last-child,.panel>.table-bordered>thead>tr>td:last-child,.panel>.table-bordered>thead>tr>th:last-child,.panel>.table-responsive>.table-bordered>tbody>tr>td:last-child,.panel>.table-responsive>.table-bordered>tbody>tr>th:last-child,.panel>.table-responsive>.table-bordered>tfoot>tr>td:last-child,.panel>.table-responsive>.table-bordered>tfoot>tr>th:last-child,.panel>.table-responsive>.table-bordered>thead>tr>td:last-child,.panel>.table-responsive>.table-bordered>thead>tr>th:last-child{border-right:0}.panel>.table-bordered>tbody>tr:first-child>td,.panel>.table-bordered>tbody>tr:first-child>th,.panel>.table-bordered>thead>tr:first-child>td,.panel>.table-bordered>thead>tr:first-child>th,.panel>.table-responsive>.table-bordered>tbody>tr:first-child>td,.panel>.table-responsive>.table-bordered>tbody>tr:first-child>th,.panel>.table-responsive>.table-bordered>thead>tr:first-child>td,.panel>.table-responsive>.table-bordered>thead>tr:first-child>th{border-bottom:0}.panel>.table-bordered>tbody>tr:last-child>td,.panel>.table-bordered>tbody>tr:last-child>th,.panel>.table-bordered>tfoot>tr:last-child>td,.panel>.table-bordered>tfoot>tr:last-child>th,.panel>.table-responsive>.table-bordered>tbody>tr:last-child>td,.panel>.table-responsive>.table-bordered>tbody>tr:last-child>th,.panel>.table-responsive>.table-bordered>tfoot>tr:last-child>td,.panel>.table-responsive>.table-bordered>tfoot>tr:last-child>th{border-bottom:0}.panel>.table-responsive{margin-bottom:0;border:0}.panel-group{margin-bottom:20px}.panel-group .panel{margin-bottom:0;border-radius:4px}.panel-group .panel+.panel{margin-top:5px}.panel-group .panel-heading{border-bottom:0}.panel-group .panel-heading+.panel-collapse>.list-group,.panel-group .panel-heading+.panel-collapse>.panel-body{border-top:1px solid #ddd}.panel-group .panel-footer{border-top:0}.panel-group .panel-footer+.panel-collapse .panel-body{border-bottom:1px solid #ddd}.panel-default{border-color:#ddd}.panel-default>.panel-heading{color:#333;background-color:#f5f5f5;border-color:#ddd}.panel-default>.panel-heading+.panel-collapse>.panel-body{border-top-color:#ddd}.panel-default>.panel-heading .badge{color:#f5f5f5;background-color:#333}.panel-default>.panel-footer+.panel-collapse>.panel-body{border-bottom-color:#ddd}.panel-primary{border-color:#337ab7}.panel-primary>.panel-heading{color:#fff;background-color:#337ab7;border-color:#337ab7}.panel-primary>.panel-heading+.panel-collapse>.panel-body{border-top-color:#337ab7}.panel-primary>.panel-heading .badge{color:#337ab7;background-color:#fff}.panel-primary>.panel-footer+.panel-collapse>.panel-body{border-bottom-color:#337ab7}.panel-success{border-color:#d6e9c6}.panel-success>.panel-heading{color:#3c763d;background-color:#dff0d8;border-color:#d6e9c6}.panel-success>.panel-heading+.panel-collapse>.panel-body{border-top-color:#d6e9c6}.panel-success>.panel-heading .badge{color:#dff0d8;background-color:#3c763d}.panel-success>.panel-footer+.panel-collapse>.panel-body{border-bottom-color:#d6e9c6}.panel-info{border-color:#bce8f1}.panel-info>.panel-heading{color:#31708f;background-color:#d9edf7;border-color:#bce8f1}.panel-info>.panel-heading+.panel-collapse>.panel-body{border-top-color:#bce8f1}.panel-info>.panel-heading .badge{color:#d9edf7;background-color:#31708f}.panel-info>.panel-footer+.panel-collapse>.panel-body{border-bottom-color:#bce8f1}.panel-warning{border-color:#faebcc}.panel-warning>.panel-heading{color:#8a6d3b;background-color:#fcf8e3;border-color:#faebcc}.panel-warning>.panel-heading+.panel-collapse>.panel-body{border-top-color:#faebcc}.panel-warning>.panel-heading .badge{color:#fcf8e3;background-color:#8a6d3b}.panel-warning>.panel-footer+.panel-collapse>.panel-body{border-bottom-color:#faebcc}.panel-danger{border-color:#ebccd1}.panel-danger>.panel-heading{color:#a94442;background-color:#f2dede;border-color:#ebccd1}.panel-danger>.panel-heading+.panel-collapse>.panel-body{border-top-color:#ebccd1}.panel-danger>.panel-heading .badge{color:#f2dede;background-color:#a94442}.panel-danger>.panel-footer+.panel-collapse>.panel-body{border-bottom-color:#ebccd1}.embed-responsive{position:relative;display:block;height:0;padding:0;overflow:hidden}.embed-responsive .embed-responsive-item,.embed-responsive embed,.embed-responsive iframe,.embed-responsive object,.embed-responsive video{position:absolute;top:0;bottom:0;left:0;width:100%;height:100%;border:0}.embed-responsive-16by9{padding-bottom:56.25%}.embed-responsive-4by3{padding-bottom:75%}.well{min-height:20px;padding:19px;margin-bottom:20px;background-color:#f5f5f5;border:1px solid #e3e3e3;border-radius:4px;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.05);box-shadow:inset 0 1px 1px rgba(0,0,0,.05)}.well blockquote{border-color:#ddd;border-color:rgba(0,0,0,.15)}.well-lg{padding:24px;border-radius:6px}.well-sm{padding:9px;border-radius:3px}.close{float:right;font-size:21px;font-weight:700;line-height:1;color:#000;text-shadow:0 1px 0 #fff;filter:alpha(opacity=20);opacity:.2}.close:focus,.close:hover{color:#000;text-decoration:none;cursor:pointer;filter:alpha(opacity=50);opacity:.5}button.close{-webkit-appearance:none;padding:0;cursor:pointer;background:0 0;border:0}.modal-open{overflow:hidden}.modal{position:fixed;top:0;right:0;bottom:0;left:0;z-index:1050;display:none;overflow:hidden;-webkit-overflow-scrolling:touch;outline:0}.modal.fade .modal-dialog{-webkit-transition:-webkit-transform .3s ease-out;-o-transition:-o-transform .3s ease-out;transition:transform .3s ease-out;-webkit-transform:translate(0,-25%);-ms-transform:translate(0,-25%);-o-transform:translate(0,-25%);transform:translate(0,-25%)}.modal.in .modal-dialog{-webkit-transform:translate(0,0);-ms-transform:translate(0,0);-o-transform:translate(0,0);transform:translate(0,0)}.modal-open .modal{overflow-x:hidden;overflow-y:auto}.modal-dialog{position:relative;width:auto;margin:10px}.modal-content{position:relative;background-color:#fff;-webkit-background-clip:padding-box;background-clip:padding-box;border:1px solid #999;border:1px solid rgba(0,0,0,.2);border-radius:6px;outline:0;-webkit-box-shadow:0 3px 9px rgba(0,0,0,.5);box-shadow:0 3px 9px rgba(0,0,0,.5)}.modal-backdrop{position:fixed;top:0;right:0;bottom:0;left:0;z-index:1040;background-color:#000}.modal-backdrop.fade{filter:alpha(opacity=0);opacity:0}.modal-backdrop.in{filter:alpha(opacity=50);opacity:.5}.modal-header{min-height:16.43px;padding:15px;border-bottom:1px solid #e5e5e5}.modal-header .close{margin-top:-2px}.modal-title{margin:0;line-height:1.42857143}.modal-body{position:relative;padding:15px}.modal-footer{padding:15px;text-align:right;border-top:1px solid #e5e5e5}.modal-footer .btn+.btn{margin-bottom:0;margin-left:5px}.modal-footer .btn-group .btn+.btn{margin-left:-1px}.modal-footer .btn-block+.btn-block{margin-left:0}.modal-scrollbar-measure{position:absolute;top:-9999px;width:50px;height:50px;overflow:scroll}@media (min-width:768px){.modal-dialog{width:600px;margin:30px auto}.modal-content{-webkit-box-shadow:0 5px 15px rgba(0,0,0,.5);box-shadow:0 5px 15px rgba(0,0,0,.5)}.modal-sm{width:300px}}@media (min-width:992px){.modal-lg{width:900px}}.tooltip{position:absolute;z-index:1070;display:block;font-family:"Helvetica Neue",Helvetica,Arial,sans-serif;font-size:12px;font-style:normal;font-weight:400;line-height:1.42857143;text-align:left;text-align:start;text-decoration:none;text-shadow:none;text-transform:none;letter-spacing:normal;word-break:normal;word-spacing:normal;word-wrap:normal;white-space:normal;filter:alpha(opacity=0);opacity:0;line-break:auto}.tooltip.in{filter:alpha(opacity=90);opacity:.9}.tooltip.top{padding:5px 0;margin-top:-3px}.tooltip.right{padding:0 5px;margin-left:3px}.tooltip.bottom{padding:5px 0;margin-top:3px}.tooltip.left{padding:0 5px;margin-left:-3px}.tooltip-inner{max-width:200px;padding:3px 8px;color:#fff;text-align:center;background-color:#000;border-radius:4px}.tooltip-arrow{position:absolute;width:0;height:0;border-color:transparent;border-style:solid}.tooltip.top .tooltip-arrow{bottom:0;left:50%;margin-left:-5px;border-width:5px 5px 0;border-top-color:#000}.tooltip.top-left .tooltip-arrow{right:5px;bottom:0;margin-bottom:-5px;border-width:5px 5px 0;border-top-color:#000}.tooltip.top-right .tooltip-arrow{bottom:0;left:5px;margin-bottom:-5px;border-width:5px 5px 0;border-top-color:#000}.tooltip.right .tooltip-arrow{top:50%;left:0;margin-top:-5px;border-width:5px 5px 5px 0;border-right-color:#000}.tooltip.left .tooltip-arrow{top:50%;right:0;margin-top:-5px;border-width:5px 0 5px 5px;border-left-color:#000}.tooltip.bottom .tooltip-arrow{top:0;left:50%;margin-left:-5px;border-width:0 5px 5px;border-bottom-color:#000}.tooltip.bottom-left .tooltip-arrow{top:0;right:5px;margin-top:-5px;border-width:0 5px 5px;border-bottom-color:#000}.tooltip.bottom-right .tooltip-arrow{top:0;left:5px;margin-top:-5px;border-width:0 5px 5px;border-bottom-color:#000}.popover{position:absolute;top:0;left:0;z-index:1060;display:none;max-width:276px;padding:1px;font-family:"Helvetica Neue",Helvetica,Arial,sans-serif;font-size:14px;font-style:normal;font-weight:400;line-height:1.42857143;text-align:left;text-align:start;text-decoration:none;text-shadow:none;text-transform:none;letter-spacing:normal;word-break:normal;word-spacing:normal;word-wrap:normal;white-space:normal;background-color:#fff;-webkit-background-clip:padding-box;background-clip:padding-box;border:1px solid #ccc;border:1px solid rgba(0,0,0,.2);border-radius:6px;-webkit-box-shadow:0 5px 10px rgba(0,0,0,.2);box-shadow:0 5px 10px rgba(0,0,0,.2);line-break:auto}.popover.top{margin-top:-10px}.popover.right{margin-left:10px}.popover.bottom{margin-top:10px}.popover.left{margin-left:-10px}.popover-title{padding:8px 14px;margin:0;font-size:14px;background-color:#f7f7f7;border-bottom:1px solid #ebebeb;border-radius:5px 5px 0 0}.popover-content{padding:9px 14px}.popover>.arrow,.popover>.arrow:after{position:absolute;display:block;width:0;height:0;border-color:transparent;border-style:solid}.popover>.arrow{border-width:11px}.popover>.arrow:after{content:"";border-width:10px}.popover.top>.arrow{bottom:-11px;left:50%;margin-left:-11px;border-top-color:#999;border-top-color:rgba(0,0,0,.25);border-bottom-width:0}.popover.top>.arrow:after{bottom:1px;margin-left:-10px;content:" ";border-top-color:#fff;border-bottom-width:0}.popover.right>.arrow{top:50%;left:-11px;margin-top:-11px;border-right-color:#999;border-right-color:rgba(0,0,0,.25);border-left-width:0}.popover.right>.arrow:after{bottom:-10px;left:1px;content:" ";border-right-color:#fff;border-left-width:0}.popover.bottom>.arrow{top:-11px;left:50%;margin-left:-11px;border-top-width:0;border-bottom-color:#999;border-bottom-color:rgba(0,0,0,.25)}.popover.bottom>.arrow:after{top:1px;margin-left:-10px;content:" ";border-top-width:0;border-bottom-color:#fff}.popover.left>.arrow{top:50%;right:-11px;margin-top:-11px;border-right-width:0;border-left-color:#999;border-left-color:rgba(0,0,0,.25)}.popover.left>.arrow:after{right:1px;bottom:-10px;content:" ";border-right-width:0;border-left-color:#fff}.carousel{position:relative}.carousel-inner{position:relative;width:100%;overflow:hidden}.carousel-inner>.item{position:relative;display:none;-webkit-transition:.6s ease-in-out left;-o-transition:.6s ease-in-out left;transition:.6s ease-in-out left}.carousel-inner>.item>a>img,.carousel-inner>.item>img{line-height:1}@media all and (transform-3d),(-webkit-transform-3d){.carousel-inner>.item{-webkit-transition:-webkit-transform .6s ease-in-out;-o-transition:-o-transform .6s ease-in-out;transition:transform .6s ease-in-out;-webkit-backface-visibility:hidden;backface-visibility:hidden;-webkit-perspective:1000px;perspective:1000px}.carousel-inner>.item.active.right,.carousel-inner>.item.next{left:0;-webkit-transform:translate3d(100%,0,0);transform:translate3d(100%,0,0)}.carousel-inner>.item.active.left,.carousel-inner>.item.prev{left:0;-webkit-transform:translate3d(-100%,0,0);transform:translate3d(-100%,0,0)}.carousel-inner>.item.active,.carousel-inner>.item.next.left,.carousel-inner>.item.prev.right{left:0;-webkit-transform:translate3d(0,0,0);transform:translate3d(0,0,0)}}.carousel-inner>.active,.carousel-inner>.next,.carousel-inner>.prev{display:block}.carousel-inner>.active{left:0}.carousel-inner>.next,.carousel-inner>.prev{position:absolute;top:0;width:100%}.carousel-inner>.next{left:100%}.carousel-inner>.prev{left:-100%}.carousel-inner>.next.left,.carousel-inner>.prev.right{left:0}.carousel-inner>.active.left{left:-100%}.carousel-inner>.active.right{left:100%}.carousel-control{position:absolute;top:0;bottom:0;left:0;width:15%;font-size:20px;color:#fff;text-align:center;text-shadow:0 1px 2px rgba(0,0,0,.6);filter:alpha(opacity=50);opacity:.5}.carousel-control.left{background-image:-webkit-linear-gradient(left,rgba(0,0,0,.5) 0,rgba(0,0,0,.0001) 100%);background-image:-o-linear-gradient(left,rgba(0,0,0,.5) 0,rgba(0,0,0,.0001) 100%);background-image:-webkit-gradient(linear,left top,right top,from(rgba(0,0,0,.5)),to(rgba(0,0,0,.0001)));background-image:linear-gradient(to right,rgba(0,0,0,.5) 0,rgba(0,0,0,.0001) 100%);filter:progid:DXImageTransform.Microsoft.gradient(startColorstr='#80000000', endColorstr='#00000000', GradientType=1);background-repeat:repeat-x}.carousel-control.right{right:0;left:auto;background-image:-webkit-linear-gradient(left,rgba(0,0,0,.0001) 0,rgba(0,0,0,.5) 100%);background-image:-o-linear-gradient(left,rgba(0,0,0,.0001) 0,rgba(0,0,0,.5) 100%);background-image:-webkit-gradient(linear,left top,right top,from(rgba(0,0,0,.0001)),to(rgba(0,0,0,.5)));background-image:linear-gradient(to right,rgba(0,0,0,.0001) 0,rgba(0,0,0,.5) 100%);filter:progid:DXImageTransform.Microsoft.gradient(startColorstr='#00000000', endColorstr='#80000000', GradientType=1);background-repeat:repeat-x}.carousel-control:focus,.carousel-control:hover{color:#fff;text-decoration:none;filter:alpha(opacity=90);outline:0;opacity:.9}.carousel-control .glyphicon-chevron-left,.carousel-control .glyphicon-chevron-right,.carousel-control .icon-next,.carousel-control .icon-prev{position:absolute;top:50%;z-index:5;display:inline-block;margin-top:-10px}.carousel-control .glyphicon-chevron-left,.carousel-control .icon-prev{left:50%;margin-left:-10px}.carousel-control .glyphicon-chevron-right,.carousel-control .icon-next{right:50%;margin-right:-10px}.carousel-control .icon-next,.carousel-control .icon-prev{width:20px;height:20px;font-family:serif;line-height:1}.carousel-control .icon-prev:before{content:'\2039'}.carousel-control .icon-next:before{content:'\203a'}.carousel-indicators{position:absolute;bottom:10px;left:50%;z-index:15;width:60%;padding-left:0;margin-left:-30%;text-align:center;list-style:none}.carousel-indicators li{display:inline-block;width:10px;height:10px;margin:1px;text-indent:-999px;cursor:pointer;background-color:#000\9;background-color:rgba(0,0,0,0);border:1px solid #fff;border-radius:10px}.carousel-indicators .active{width:12px;height:12px;margin:0;background-color:#fff}.carousel-caption{position:absolute;right:15%;bottom:20px;left:15%;z-index:10;padding-top:20px;padding-bottom:20px;color:#fff;text-align:center;text-shadow:0 1px 2px rgba(0,0,0,.6)}.carousel-caption .btn{text-shadow:none}@media screen and (min-width:768px){.carousel-control .glyphicon-chevron-left,.carousel-control .glyphicon-chevron-right,.carousel-control .icon-next,.carousel-control .icon-prev{width:30px;height:30px;margin-top:-15px;font-size:30px}.carousel-control .glyphicon-chevron-left,.carousel-control .icon-prev{margin-left:-15px}.carousel-control .glyphicon-chevron-right,.carousel-control .icon-next{margin-right:-15px}.carousel-caption{right:20%;left:20%;padding-bottom:30px}.carousel-indicators{bottom:20px}}.btn-group-vertical>.btn-group:after,.btn-group-vertical>.btn-group:before,.btn-toolbar:after,.btn-toolbar:before,.clearfix:after,.clearfix:before,.container-fluid:after,.container-fluid:before,.container:after,.container:before,.dl-horizontal dd:after,.dl-horizontal dd:before,.form-horizontal .form-group:after,.form-horizontal .form-group:before,.modal-footer:after,.modal-footer:before,.nav:after,.nav:before,.navbar-collapse:after,.navbar-collapse:before,.navbar-header:after,.navbar-header:before,.navbar:after,.navbar:before,.pager:after,.pager:before,.panel-body:after,.panel-body:before,.row:after,.row:before{display:table;content:" "}.btn-group-vertical>.btn-group:after,.btn-toolbar:after,.clearfix:after,.container-fluid:after,.container:after,.dl-horizontal dd:after,.form-horizontal .form-group:after,.modal-footer:after,.nav:after,.navbar-collapse:after,.navbar-header:after,.navbar:after,.pager:after,.panel-body:after,.row:after{clear:both}.center-block{display:block;margin-right:auto;margin-left:auto}.pull-right{float:right!important}.pull-left{float:left!important}.hide{display:none!important}.show{display:block!important}.invisible{visibility:hidden}.text-hide{font:0/0 a;color:transparent;text-shadow:none;background-color:transparent;border:0}.hidden{display:none!important}.affix{position:fixed}@-ms-viewport{width:device-width}.visible-lg,.visible-md,.visible-sm,.visible-xs{display:none!important}.visible-lg-block,.visible-lg-inline,.visible-lg-inline-block,.visible-md-block,.visible-md-inline,.visible-md-inline-block,.visible-sm-block,.visible-sm-inline,.visible-sm-inline-block,.visible-xs-block,.visible-xs-inline,.visible-xs-inline-block{display:none!important}@media (max-width:767px){.visible-xs{display:block!important}table.visible-xs{display:table!important}tr.visible-xs{display:table-row!important}td.visible-xs,th.visible-xs{display:table-cell!important}}@media (max-width:767px){.visible-xs-block{display:block!important}}@media (max-width:767px){.visible-xs-inline{display:inline!important}}@media (max-width:767px){.visible-xs-inline-block{display:inline-block!important}}@media (min-width:768px) and (max-width:991px){.visible-sm{display:block!important}table.visible-sm{display:table!important}tr.visible-sm{display:table-row!important}td.visible-sm,th.visible-sm{display:table-cell!important}}@media (min-width:768px) and (max-width:991px){.visible-sm-block{display:block!important}}@media (min-width:768px) and (max-width:991px){.visible-sm-inline{display:inline!important}}@media (min-width:768px) and (max-width:991px){.visible-sm-inline-block{display:inline-block!important}}@media (min-width:992px) and (max-width:1199px){.visible-md{display:block!important}table.visible-md{display:table!important}tr.visible-md{display:table-row!important}td.visible-md,th.visible-md{display:table-cell!important}}@media (min-width:992px) and (max-width:1199px){.visible-md-block{display:block!important}}@media (min-width:992px) and (max-width:1199px){.visible-md-inline{display:inline!important}}@media (min-width:992px) and (max-width:1199px){.visible-md-inline-block{display:inline-block!important}}@media (min-width:1200px){.visible-lg{display:block!important}table.visible-lg{display:table!important}tr.visible-lg{display:table-row!important}td.visible-lg,th.visible-lg{display:table-cell!important}}@media (min-width:1200px){.visible-lg-block{display:block!important}}@media (min-width:1200px){.visible-lg-inline{display:inline!important}}@media (min-width:1200px){.visible-lg-inline-block{display:inline-block!important}}@media (max-width:767px){.hidden-xs{display:none!important}}@media (min-width:768px) and (max-width:991px){.hidden-sm{display:none!important}}@media (min-width:992px) and (max-width:1199px){.hidden-md{display:none!important}}@media (min-width:1200px){.hidden-lg{display:none!important}}.visible-print{display:none!important}@media print{.visible-print{display:block!important}table.visible-print{display:table!important}tr.visible-print{display:table-row!important}td.visible-print,th.visible-print{display:table-cell!important}}.visible-print-block{display:none!important}@media print{.visible-print-block{display:block!important}}.visible-print-inline{display:none!important}@media print{.visible-print-inline{display:inline!important}}.visible-print-inline-block{display:none!important}@media print{.visible-print-inline-block{display:inline-block!important}}@media print{.hidden-print{display:none!important}}
</style>
<script>/*!
 * Bootstrap v3.3.5 (http://getbootstrap.com)
 * Copyright 2011-2015 Twitter, Inc.
 * Licensed under the MIT license
 */
if("undefined"==typeof jQuery)throw new Error("Bootstrap's JavaScript requires jQuery");+function(a){"use strict";var b=a.fn.jquery.split(" ")[0].split(".");if(b[0]<2&&b[1]<9||1==b[0]&&9==b[1]&&b[2]<1)throw new Error("Bootstrap's JavaScript requires jQuery version 1.9.1 or higher")}(jQuery),+function(a){"use strict";function b(){var a=document.createElement("bootstrap"),b={WebkitTransition:"webkitTransitionEnd",MozTransition:"transitionend",OTransition:"oTransitionEnd otransitionend",transition:"transitionend"};for(var c in b)if(void 0!==a.style[c])return{end:b[c]};return!1}a.fn.emulateTransitionEnd=function(b){var c=!1,d=this;a(this).one("bsTransitionEnd",function(){c=!0});var e=function(){c||a(d).trigger(a.support.transition.end)};return setTimeout(e,b),this},a(function(){a.support.transition=b(),a.support.transition&&(a.event.special.bsTransitionEnd={bindType:a.support.transition.end,delegateType:a.support.transition.end,handle:function(b){return a(b.target).is(this)?b.handleObj.handler.apply(this,arguments):void 0}})})}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var c=a(this),e=c.data("bs.alert");e||c.data("bs.alert",e=new d(this)),"string"==typeof b&&e[b].call(c)})}var c='[data-dismiss="alert"]',d=function(b){a(b).on("click",c,this.close)};d.VERSION="3.3.5",d.TRANSITION_DURATION=150,d.prototype.close=function(b){function c(){g.detach().trigger("closed.bs.alert").remove()}var e=a(this),f=e.attr("data-target");f||(f=e.attr("href"),f=f&&f.replace(/.*(?=#[^\s]*$)/,""));var g=a(f);b&&b.preventDefault(),g.length||(g=e.closest(".alert")),g.trigger(b=a.Event("close.bs.alert")),b.isDefaultPrevented()||(g.removeClass("in"),a.support.transition&&g.hasClass("fade")?g.one("bsTransitionEnd",c).emulateTransitionEnd(d.TRANSITION_DURATION):c())};var e=a.fn.alert;a.fn.alert=b,a.fn.alert.Constructor=d,a.fn.alert.noConflict=function(){return a.fn.alert=e,this},a(document).on("click.bs.alert.data-api",c,d.prototype.close)}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var d=a(this),e=d.data("bs.button"),f="object"==typeof b&&b;e||d.data("bs.button",e=new c(this,f)),"toggle"==b?e.toggle():b&&e.setState(b)})}var c=function(b,d){this.$element=a(b),this.options=a.extend({},c.DEFAULTS,d),this.isLoading=!1};c.VERSION="3.3.5",c.DEFAULTS={loadingText:"loading..."},c.prototype.setState=function(b){var c="disabled",d=this.$element,e=d.is("input")?"val":"html",f=d.data();b+="Text",null==f.resetText&&d.data("resetText",d[e]()),setTimeout(a.proxy(function(){d[e](null==f[b]?this.options[b]:f[b]),"loadingText"==b?(this.isLoading=!0,d.addClass(c).attr(c,c)):this.isLoading&&(this.isLoading=!1,d.removeClass(c).removeAttr(c))},this),0)},c.prototype.toggle=function(){var a=!0,b=this.$element.closest('[data-toggle="buttons"]');if(b.length){var c=this.$element.find("input");"radio"==c.prop("type")?(c.prop("checked")&&(a=!1),b.find(".active").removeClass("active"),this.$element.addClass("active")):"checkbox"==c.prop("type")&&(c.prop("checked")!==this.$element.hasClass("active")&&(a=!1),this.$element.toggleClass("active")),c.prop("checked",this.$element.hasClass("active")),a&&c.trigger("change")}else this.$element.attr("aria-pressed",!this.$element.hasClass("active")),this.$element.toggleClass("active")};var d=a.fn.button;a.fn.button=b,a.fn.button.Constructor=c,a.fn.button.noConflict=function(){return a.fn.button=d,this},a(document).on("click.bs.button.data-api",'[data-toggle^="button"]',function(c){var d=a(c.target);d.hasClass("btn")||(d=d.closest(".btn")),b.call(d,"toggle"),a(c.target).is('input[type="radio"]')||a(c.target).is('input[type="checkbox"]')||c.preventDefault()}).on("focus.bs.button.data-api blur.bs.button.data-api",'[data-toggle^="button"]',function(b){a(b.target).closest(".btn").toggleClass("focus",/^focus(in)?$/.test(b.type))})}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var d=a(this),e=d.data("bs.carousel"),f=a.extend({},c.DEFAULTS,d.data(),"object"==typeof b&&b),g="string"==typeof b?b:f.slide;e||d.data("bs.carousel",e=new c(this,f)),"number"==typeof b?e.to(b):g?e[g]():f.interval&&e.pause().cycle()})}var c=function(b,c){this.$element=a(b),this.$indicators=this.$element.find(".carousel-indicators"),this.options=c,this.paused=null,this.sliding=null,this.interval=null,this.$active=null,this.$items=null,this.options.keyboard&&this.$element.on("keydown.bs.carousel",a.proxy(this.keydown,this)),"hover"==this.options.pause&&!("ontouchstart"in document.documentElement)&&this.$element.on("mouseenter.bs.carousel",a.proxy(this.pause,this)).on("mouseleave.bs.carousel",a.proxy(this.cycle,this))};c.VERSION="3.3.5",c.TRANSITION_DURATION=600,c.DEFAULTS={interval:5e3,pause:"hover",wrap:!0,keyboard:!0},c.prototype.keydown=function(a){if(!/input|textarea/i.test(a.target.tagName)){switch(a.which){case 37:this.prev();break;case 39:this.next();break;default:return}a.preventDefault()}},c.prototype.cycle=function(b){return b||(this.paused=!1),this.interval&&clearInterval(this.interval),this.options.interval&&!this.paused&&(this.interval=setInterval(a.proxy(this.next,this),this.options.interval)),this},c.prototype.getItemIndex=function(a){return this.$items=a.parent().children(".item"),this.$items.index(a||this.$active)},c.prototype.getItemForDirection=function(a,b){var c=this.getItemIndex(b),d="prev"==a&&0===c||"next"==a&&c==this.$items.length-1;if(d&&!this.options.wrap)return b;var e="prev"==a?-1:1,f=(c+e)%this.$items.length;return this.$items.eq(f)},c.prototype.to=function(a){var b=this,c=this.getItemIndex(this.$active=this.$element.find(".item.active"));return a>this.$items.length-1||0>a?void 0:this.sliding?this.$element.one("slid.bs.carousel",function(){b.to(a)}):c==a?this.pause().cycle():this.slide(a>c?"next":"prev",this.$items.eq(a))},c.prototype.pause=function(b){return b||(this.paused=!0),this.$element.find(".next, .prev").length&&a.support.transition&&(this.$element.trigger(a.support.transition.end),this.cycle(!0)),this.interval=clearInterval(this.interval),this},c.prototype.next=function(){return this.sliding?void 0:this.slide("next")},c.prototype.prev=function(){return this.sliding?void 0:this.slide("prev")},c.prototype.slide=function(b,d){var e=this.$element.find(".item.active"),f=d||this.getItemForDirection(b,e),g=this.interval,h="next"==b?"left":"right",i=this;if(f.hasClass("active"))return this.sliding=!1;var j=f[0],k=a.Event("slide.bs.carousel",{relatedTarget:j,direction:h});if(this.$element.trigger(k),!k.isDefaultPrevented()){if(this.sliding=!0,g&&this.pause(),this.$indicators.length){this.$indicators.find(".active").removeClass("active");var l=a(this.$indicators.children()[this.getItemIndex(f)]);l&&l.addClass("active")}var m=a.Event("slid.bs.carousel",{relatedTarget:j,direction:h});return a.support.transition&&this.$element.hasClass("slide")?(f.addClass(b),f[0].offsetWidth,e.addClass(h),f.addClass(h),e.one("bsTransitionEnd",function(){f.removeClass([b,h].join(" ")).addClass("active"),e.removeClass(["active",h].join(" ")),i.sliding=!1,setTimeout(function(){i.$element.trigger(m)},0)}).emulateTransitionEnd(c.TRANSITION_DURATION)):(e.removeClass("active"),f.addClass("active"),this.sliding=!1,this.$element.trigger(m)),g&&this.cycle(),this}};var d=a.fn.carousel;a.fn.carousel=b,a.fn.carousel.Constructor=c,a.fn.carousel.noConflict=function(){return a.fn.carousel=d,this};var e=function(c){var d,e=a(this),f=a(e.attr("data-target")||(d=e.attr("href"))&&d.replace(/.*(?=#[^\s]+$)/,""));if(f.hasClass("carousel")){var g=a.extend({},f.data(),e.data()),h=e.attr("data-slide-to");h&&(g.interval=!1),b.call(f,g),h&&f.data("bs.carousel").to(h),c.preventDefault()}};a(document).on("click.bs.carousel.data-api","[data-slide]",e).on("click.bs.carousel.data-api","[data-slide-to]",e),a(window).on("load",function(){a('[data-ride="carousel"]').each(function(){var c=a(this);b.call(c,c.data())})})}(jQuery),+function(a){"use strict";function b(b){var c,d=b.attr("data-target")||(c=b.attr("href"))&&c.replace(/.*(?=#[^\s]+$)/,"");return a(d)}function c(b){return this.each(function(){var c=a(this),e=c.data("bs.collapse"),f=a.extend({},d.DEFAULTS,c.data(),"object"==typeof b&&b);!e&&f.toggle&&/show|hide/.test(b)&&(f.toggle=!1),e||c.data("bs.collapse",e=new d(this,f)),"string"==typeof b&&e[b]()})}var d=function(b,c){this.$element=a(b),this.options=a.extend({},d.DEFAULTS,c),this.$trigger=a('[data-toggle="collapse"][href="#'+b.id+'"],[data-toggle="collapse"][data-target="#'+b.id+'"]'),this.transitioning=null,this.options.parent?this.$parent=this.getParent():this.addAriaAndCollapsedClass(this.$element,this.$trigger),this.options.toggle&&this.toggle()};d.VERSION="3.3.5",d.TRANSITION_DURATION=350,d.DEFAULTS={toggle:!0},d.prototype.dimension=function(){var a=this.$element.hasClass("width");return a?"width":"height"},d.prototype.show=function(){if(!this.transitioning&&!this.$element.hasClass("in")){var b,e=this.$parent&&this.$parent.children(".panel").children(".in, .collapsing");if(!(e&&e.length&&(b=e.data("bs.collapse"),b&&b.transitioning))){var f=a.Event("show.bs.collapse");if(this.$element.trigger(f),!f.isDefaultPrevented()){e&&e.length&&(c.call(e,"hide"),b||e.data("bs.collapse",null));var g=this.dimension();this.$element.removeClass("collapse").addClass("collapsing")[g](0).attr("aria-expanded",!0),this.$trigger.removeClass("collapsed").attr("aria-expanded",!0),this.transitioning=1;var h=function(){this.$element.removeClass("collapsing").addClass("collapse in")[g](""),this.transitioning=0,this.$element.trigger("shown.bs.collapse")};if(!a.support.transition)return h.call(this);var i=a.camelCase(["scroll",g].join("-"));this.$element.one("bsTransitionEnd",a.proxy(h,this)).emulateTransitionEnd(d.TRANSITION_DURATION)[g](this.$element[0][i])}}}},d.prototype.hide=function(){if(!this.transitioning&&this.$element.hasClass("in")){var b=a.Event("hide.bs.collapse");if(this.$element.trigger(b),!b.isDefaultPrevented()){var c=this.dimension();this.$element[c](this.$element[c]())[0].offsetHeight,this.$element.addClass("collapsing").removeClass("collapse in").attr("aria-expanded",!1),this.$trigger.addClass("collapsed").attr("aria-expanded",!1),this.transitioning=1;var e=function(){this.transitioning=0,this.$element.removeClass("collapsing").addClass("collapse").trigger("hidden.bs.collapse")};return a.support.transition?void this.$element[c](0).one("bsTransitionEnd",a.proxy(e,this)).emulateTransitionEnd(d.TRANSITION_DURATION):e.call(this)}}},d.prototype.toggle=function(){this[this.$element.hasClass("in")?"hide":"show"]()},d.prototype.getParent=function(){return a(this.options.parent).find('[data-toggle="collapse"][data-parent="'+this.options.parent+'"]').each(a.proxy(function(c,d){var e=a(d);this.addAriaAndCollapsedClass(b(e),e)},this)).end()},d.prototype.addAriaAndCollapsedClass=function(a,b){var c=a.hasClass("in");a.attr("aria-expanded",c),b.toggleClass("collapsed",!c).attr("aria-expanded",c)};var e=a.fn.collapse;a.fn.collapse=c,a.fn.collapse.Constructor=d,a.fn.collapse.noConflict=function(){return a.fn.collapse=e,this},a(document).on("click.bs.collapse.data-api",'[data-toggle="collapse"]',function(d){var e=a(this);e.attr("data-target")||d.preventDefault();var f=b(e),g=f.data("bs.collapse"),h=g?"toggle":e.data();c.call(f,h)})}(jQuery),+function(a){"use strict";function b(b){var c=b.attr("data-target");c||(c=b.attr("href"),c=c&&/#[A-Za-z]/.test(c)&&c.replace(/.*(?=#[^\s]*$)/,""));var d=c&&a(c);return d&&d.length?d:b.parent()}function c(c){c&&3===c.which||(a(e).remove(),a(f).each(function(){var d=a(this),e=b(d),f={relatedTarget:this};e.hasClass("open")&&(c&&"click"==c.type&&/input|textarea/i.test(c.target.tagName)&&a.contains(e[0],c.target)||(e.trigger(c=a.Event("hide.bs.dropdown",f)),c.isDefaultPrevented()||(d.attr("aria-expanded","false"),e.removeClass("open").trigger("hidden.bs.dropdown",f))))}))}function d(b){return this.each(function(){var c=a(this),d=c.data("bs.dropdown");d||c.data("bs.dropdown",d=new g(this)),"string"==typeof b&&d[b].call(c)})}var e=".dropdown-backdrop",f='[data-toggle="dropdown"]',g=function(b){a(b).on("click.bs.dropdown",this.toggle)};g.VERSION="3.3.5",g.prototype.toggle=function(d){var e=a(this);if(!e.is(".disabled, :disabled")){var f=b(e),g=f.hasClass("open");if(c(),!g){"ontouchstart"in document.documentElement&&!f.closest(".navbar-nav").length&&a(document.createElement("div")).addClass("dropdown-backdrop").insertAfter(a(this)).on("click",c);var h={relatedTarget:this};if(f.trigger(d=a.Event("show.bs.dropdown",h)),d.isDefaultPrevented())return;e.trigger("focus").attr("aria-expanded","true"),f.toggleClass("open").trigger("shown.bs.dropdown",h)}return!1}},g.prototype.keydown=function(c){if(/(38|40|27|32)/.test(c.which)&&!/input|textarea/i.test(c.target.tagName)){var d=a(this);if(c.preventDefault(),c.stopPropagation(),!d.is(".disabled, :disabled")){var e=b(d),g=e.hasClass("open");if(!g&&27!=c.which||g&&27==c.which)return 27==c.which&&e.find(f).trigger("focus"),d.trigger("click");var h=" li:not(.disabled):visible a",i=e.find(".dropdown-menu"+h);if(i.length){var j=i.index(c.target);38==c.which&&j>0&&j--,40==c.which&&j<i.length-1&&j++,~j||(j=0),i.eq(j).trigger("focus")}}}};var h=a.fn.dropdown;a.fn.dropdown=d,a.fn.dropdown.Constructor=g,a.fn.dropdown.noConflict=function(){return a.fn.dropdown=h,this},a(document).on("click.bs.dropdown.data-api",c).on("click.bs.dropdown.data-api",".dropdown form",function(a){a.stopPropagation()}).on("click.bs.dropdown.data-api",f,g.prototype.toggle).on("keydown.bs.dropdown.data-api",f,g.prototype.keydown).on("keydown.bs.dropdown.data-api",".dropdown-menu",g.prototype.keydown)}(jQuery),+function(a){"use strict";function b(b,d){return this.each(function(){var e=a(this),f=e.data("bs.modal"),g=a.extend({},c.DEFAULTS,e.data(),"object"==typeof b&&b);f||e.data("bs.modal",f=new c(this,g)),"string"==typeof b?f[b](d):g.show&&f.show(d)})}var c=function(b,c){this.options=c,this.$body=a(document.body),this.$element=a(b),this.$dialog=this.$element.find(".modal-dialog"),this.$backdrop=null,this.isShown=null,this.originalBodyPad=null,this.scrollbarWidth=0,this.ignoreBackdropClick=!1,this.options.remote&&this.$element.find(".modal-content").load(this.options.remote,a.proxy(function(){this.$element.trigger("loaded.bs.modal")},this))};c.VERSION="3.3.5",c.TRANSITION_DURATION=300,c.BACKDROP_TRANSITION_DURATION=150,c.DEFAULTS={backdrop:!0,keyboard:!0,show:!0},c.prototype.toggle=function(a){return this.isShown?this.hide():this.show(a)},c.prototype.show=function(b){var d=this,e=a.Event("show.bs.modal",{relatedTarget:b});this.$element.trigger(e),this.isShown||e.isDefaultPrevented()||(this.isShown=!0,this.checkScrollbar(),this.setScrollbar(),this.$body.addClass("modal-open"),this.escape(),this.resize(),this.$element.on("click.dismiss.bs.modal",'[data-dismiss="modal"]',a.proxy(this.hide,this)),this.$dialog.on("mousedown.dismiss.bs.modal",function(){d.$element.one("mouseup.dismiss.bs.modal",function(b){a(b.target).is(d.$element)&&(d.ignoreBackdropClick=!0)})}),this.backdrop(function(){var e=a.support.transition&&d.$element.hasClass("fade");d.$element.parent().length||d.$element.appendTo(d.$body),d.$element.show().scrollTop(0),d.adjustDialog(),e&&d.$element[0].offsetWidth,d.$element.addClass("in"),d.enforceFocus();var f=a.Event("shown.bs.modal",{relatedTarget:b});e?d.$dialog.one("bsTransitionEnd",function(){d.$element.trigger("focus").trigger(f)}).emulateTransitionEnd(c.TRANSITION_DURATION):d.$element.trigger("focus").trigger(f)}))},c.prototype.hide=function(b){b&&b.preventDefault(),b=a.Event("hide.bs.modal"),this.$element.trigger(b),this.isShown&&!b.isDefaultPrevented()&&(this.isShown=!1,this.escape(),this.resize(),a(document).off("focusin.bs.modal"),this.$element.removeClass("in").off("click.dismiss.bs.modal").off("mouseup.dismiss.bs.modal"),this.$dialog.off("mousedown.dismiss.bs.modal"),a.support.transition&&this.$element.hasClass("fade")?this.$element.one("bsTransitionEnd",a.proxy(this.hideModal,this)).emulateTransitionEnd(c.TRANSITION_DURATION):this.hideModal())},c.prototype.enforceFocus=function(){a(document).off("focusin.bs.modal").on("focusin.bs.modal",a.proxy(function(a){this.$element[0]===a.target||this.$element.has(a.target).length||this.$element.trigger("focus")},this))},c.prototype.escape=function(){this.isShown&&this.options.keyboard?this.$element.on("keydown.dismiss.bs.modal",a.proxy(function(a){27==a.which&&this.hide()},this)):this.isShown||this.$element.off("keydown.dismiss.bs.modal")},c.prototype.resize=function(){this.isShown?a(window).on("resize.bs.modal",a.proxy(this.handleUpdate,this)):a(window).off("resize.bs.modal")},c.prototype.hideModal=function(){var a=this;this.$element.hide(),this.backdrop(function(){a.$body.removeClass("modal-open"),a.resetAdjustments(),a.resetScrollbar(),a.$element.trigger("hidden.bs.modal")})},c.prototype.removeBackdrop=function(){this.$backdrop&&this.$backdrop.remove(),this.$backdrop=null},c.prototype.backdrop=function(b){var d=this,e=this.$element.hasClass("fade")?"fade":"";if(this.isShown&&this.options.backdrop){var f=a.support.transition&&e;if(this.$backdrop=a(document.createElement("div")).addClass("modal-backdrop "+e).appendTo(this.$body),this.$element.on("click.dismiss.bs.modal",a.proxy(function(a){return this.ignoreBackdropClick?void(this.ignoreBackdropClick=!1):void(a.target===a.currentTarget&&("static"==this.options.backdrop?this.$element[0].focus():this.hide()))},this)),f&&this.$backdrop[0].offsetWidth,this.$backdrop.addClass("in"),!b)return;f?this.$backdrop.one("bsTransitionEnd",b).emulateTransitionEnd(c.BACKDROP_TRANSITION_DURATION):b()}else if(!this.isShown&&this.$backdrop){this.$backdrop.removeClass("in");var g=function(){d.removeBackdrop(),b&&b()};a.support.transition&&this.$element.hasClass("fade")?this.$backdrop.one("bsTransitionEnd",g).emulateTransitionEnd(c.BACKDROP_TRANSITION_DURATION):g()}else b&&b()},c.prototype.handleUpdate=function(){this.adjustDialog()},c.prototype.adjustDialog=function(){var a=this.$element[0].scrollHeight>document.documentElement.clientHeight;this.$element.css({paddingLeft:!this.bodyIsOverflowing&&a?this.scrollbarWidth:"",paddingRight:this.bodyIsOverflowing&&!a?this.scrollbarWidth:""})},c.prototype.resetAdjustments=function(){this.$element.css({paddingLeft:"",paddingRight:""})},c.prototype.checkScrollbar=function(){var a=window.innerWidth;if(!a){var b=document.documentElement.getBoundingClientRect();a=b.right-Math.abs(b.left)}this.bodyIsOverflowing=document.body.clientWidth<a,this.scrollbarWidth=this.measureScrollbar()},c.prototype.setScrollbar=function(){var a=parseInt(this.$body.css("padding-right")||0,10);this.originalBodyPad=document.body.style.paddingRight||"",this.bodyIsOverflowing&&this.$body.css("padding-right",a+this.scrollbarWidth)},c.prototype.resetScrollbar=function(){this.$body.css("padding-right",this.originalBodyPad)},c.prototype.measureScrollbar=function(){var a=document.createElement("div");a.className="modal-scrollbar-measure",this.$body.append(a);var b=a.offsetWidth-a.clientWidth;return this.$body[0].removeChild(a),b};var d=a.fn.modal;a.fn.modal=b,a.fn.modal.Constructor=c,a.fn.modal.noConflict=function(){return a.fn.modal=d,this},a(document).on("click.bs.modal.data-api",'[data-toggle="modal"]',function(c){var d=a(this),e=d.attr("href"),f=a(d.attr("data-target")||e&&e.replace(/.*(?=#[^\s]+$)/,"")),g=f.data("bs.modal")?"toggle":a.extend({remote:!/#/.test(e)&&e},f.data(),d.data());d.is("a")&&c.preventDefault(),f.one("show.bs.modal",function(a){a.isDefaultPrevented()||f.one("hidden.bs.modal",function(){d.is(":visible")&&d.trigger("focus")})}),b.call(f,g,this)})}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var d=a(this),e=d.data("bs.tooltip"),f="object"==typeof b&&b;(e||!/destroy|hide/.test(b))&&(e||d.data("bs.tooltip",e=new c(this,f)),"string"==typeof b&&e[b]())})}var c=function(a,b){this.type=null,this.options=null,this.enabled=null,this.timeout=null,this.hoverState=null,this.$element=null,this.inState=null,this.init("tooltip",a,b)};c.VERSION="3.3.5",c.TRANSITION_DURATION=150,c.DEFAULTS={animation:!0,placement:"top",selector:!1,template:'<div class="tooltip" role="tooltip"><div class="tooltip-arrow"></div><div class="tooltip-inner"></div></div>',trigger:"hover focus",title:"",delay:0,html:!1,container:!1,viewport:{selector:"body",padding:0}},c.prototype.init=function(b,c,d){if(this.enabled=!0,this.type=b,this.$element=a(c),this.options=this.getOptions(d),this.$viewport=this.options.viewport&&a(a.isFunction(this.options.viewport)?this.options.viewport.call(this,this.$element):this.options.viewport.selector||this.options.viewport),this.inState={click:!1,hover:!1,focus:!1},this.$element[0]instanceof document.constructor&&!this.options.selector)throw new Error("`selector` option must be specified when initializing "+this.type+" on the window.document object!");for(var e=this.options.trigger.split(" "),f=e.length;f--;){var g=e[f];if("click"==g)this.$element.on("click."+this.type,this.options.selector,a.proxy(this.toggle,this));else if("manual"!=g){var h="hover"==g?"mouseenter":"focusin",i="hover"==g?"mouseleave":"focusout";this.$element.on(h+"."+this.type,this.options.selector,a.proxy(this.enter,this)),this.$element.on(i+"."+this.type,this.options.selector,a.proxy(this.leave,this))}}this.options.selector?this._options=a.extend({},this.options,{trigger:"manual",selector:""}):this.fixTitle()},c.prototype.getDefaults=function(){return c.DEFAULTS},c.prototype.getOptions=function(b){return b=a.extend({},this.getDefaults(),this.$element.data(),b),b.delay&&"number"==typeof b.delay&&(b.delay={show:b.delay,hide:b.delay}),b},c.prototype.getDelegateOptions=function(){var b={},c=this.getDefaults();return this._options&&a.each(this._options,function(a,d){c[a]!=d&&(b[a]=d)}),b},c.prototype.enter=function(b){var c=b instanceof this.constructor?b:a(b.currentTarget).data("bs."+this.type);return c||(c=new this.constructor(b.currentTarget,this.getDelegateOptions()),a(b.currentTarget).data("bs."+this.type,c)),b instanceof a.Event&&(c.inState["focusin"==b.type?"focus":"hover"]=!0),c.tip().hasClass("in")||"in"==c.hoverState?void(c.hoverState="in"):(clearTimeout(c.timeout),c.hoverState="in",c.options.delay&&c.options.delay.show?void(c.timeout=setTimeout(function(){"in"==c.hoverState&&c.show()},c.options.delay.show)):c.show())},c.prototype.isInStateTrue=function(){for(var a in this.inState)if(this.inState[a])return!0;return!1},c.prototype.leave=function(b){var c=b instanceof this.constructor?b:a(b.currentTarget).data("bs."+this.type);return c||(c=new this.constructor(b.currentTarget,this.getDelegateOptions()),a(b.currentTarget).data("bs."+this.type,c)),b instanceof a.Event&&(c.inState["focusout"==b.type?"focus":"hover"]=!1),c.isInStateTrue()?void 0:(clearTimeout(c.timeout),c.hoverState="out",c.options.delay&&c.options.delay.hide?void(c.timeout=setTimeout(function(){"out"==c.hoverState&&c.hide()},c.options.delay.hide)):c.hide())},c.prototype.show=function(){var b=a.Event("show.bs."+this.type);if(this.hasContent()&&this.enabled){this.$element.trigger(b);var d=a.contains(this.$element[0].ownerDocument.documentElement,this.$element[0]);if(b.isDefaultPrevented()||!d)return;var e=this,f=this.tip(),g=this.getUID(this.type);this.setContent(),f.attr("id",g),this.$element.attr("aria-describedby",g),this.options.animation&&f.addClass("fade");var h="function"==typeof this.options.placement?this.options.placement.call(this,f[0],this.$element[0]):this.options.placement,i=/\s?auto?\s?/i,j=i.test(h);j&&(h=h.replace(i,"")||"top"),f.detach().css({top:0,left:0,display:"block"}).addClass(h).data("bs."+this.type,this),this.options.container?f.appendTo(this.options.container):f.insertAfter(this.$element),this.$element.trigger("inserted.bs."+this.type);var k=this.getPosition(),l=f[0].offsetWidth,m=f[0].offsetHeight;if(j){var n=h,o=this.getPosition(this.$viewport);h="bottom"==h&&k.bottom+m>o.bottom?"top":"top"==h&&k.top-m<o.top?"bottom":"right"==h&&k.right+l>o.width?"left":"left"==h&&k.left-l<o.left?"right":h,f.removeClass(n).addClass(h)}var p=this.getCalculatedOffset(h,k,l,m);this.applyPlacement(p,h);var q=function(){var a=e.hoverState;e.$element.trigger("shown.bs."+e.type),e.hoverState=null,"out"==a&&e.leave(e)};a.support.transition&&this.$tip.hasClass("fade")?f.one("bsTransitionEnd",q).emulateTransitionEnd(c.TRANSITION_DURATION):q()}},c.prototype.applyPlacement=function(b,c){var d=this.tip(),e=d[0].offsetWidth,f=d[0].offsetHeight,g=parseInt(d.css("margin-top"),10),h=parseInt(d.css("margin-left"),10);isNaN(g)&&(g=0),isNaN(h)&&(h=0),b.top+=g,b.left+=h,a.offset.setOffset(d[0],a.extend({using:function(a){d.css({top:Math.round(a.top),left:Math.round(a.left)})}},b),0),d.addClass("in");var i=d[0].offsetWidth,j=d[0].offsetHeight;"top"==c&&j!=f&&(b.top=b.top+f-j);var k=this.getViewportAdjustedDelta(c,b,i,j);k.left?b.left+=k.left:b.top+=k.top;var l=/top|bottom/.test(c),m=l?2*k.left-e+i:2*k.top-f+j,n=l?"offsetWidth":"offsetHeight";d.offset(b),this.replaceArrow(m,d[0][n],l)},c.prototype.replaceArrow=function(a,b,c){this.arrow().css(c?"left":"top",50*(1-a/b)+"%").css(c?"top":"left","")},c.prototype.setContent=function(){var a=this.tip(),b=this.getTitle();a.find(".tooltip-inner")[this.options.html?"html":"text"](b),a.removeClass("fade in top bottom left right")},c.prototype.hide=function(b){function d(){"in"!=e.hoverState&&f.detach(),e.$element.removeAttr("aria-describedby").trigger("hidden.bs."+e.type),b&&b()}var e=this,f=a(this.$tip),g=a.Event("hide.bs."+this.type);return this.$element.trigger(g),g.isDefaultPrevented()?void 0:(f.removeClass("in"),a.support.transition&&f.hasClass("fade")?f.one("bsTransitionEnd",d).emulateTransitionEnd(c.TRANSITION_DURATION):d(),this.hoverState=null,this)},c.prototype.fixTitle=function(){var a=this.$element;(a.attr("title")||"string"!=typeof a.attr("data-original-title"))&&a.attr("data-original-title",a.attr("title")||"").attr("title","")},c.prototype.hasContent=function(){return this.getTitle()},c.prototype.getPosition=function(b){b=b||this.$element;var c=b[0],d="BODY"==c.tagName,e=c.getBoundingClientRect();null==e.width&&(e=a.extend({},e,{width:e.right-e.left,height:e.bottom-e.top}));var f=d?{top:0,left:0}:b.offset(),g={scroll:d?document.documentElement.scrollTop||document.body.scrollTop:b.scrollTop()},h=d?{width:a(window).width(),height:a(window).height()}:null;return a.extend({},e,g,h,f)},c.prototype.getCalculatedOffset=function(a,b,c,d){return"bottom"==a?{top:b.top+b.height,left:b.left+b.width/2-c/2}:"top"==a?{top:b.top-d,left:b.left+b.width/2-c/2}:"left"==a?{top:b.top+b.height/2-d/2,left:b.left-c}:{top:b.top+b.height/2-d/2,left:b.left+b.width}},c.prototype.getViewportAdjustedDelta=function(a,b,c,d){var e={top:0,left:0};if(!this.$viewport)return e;var f=this.options.viewport&&this.options.viewport.padding||0,g=this.getPosition(this.$viewport);if(/right|left/.test(a)){var h=b.top-f-g.scroll,i=b.top+f-g.scroll+d;h<g.top?e.top=g.top-h:i>g.top+g.height&&(e.top=g.top+g.height-i)}else{var j=b.left-f,k=b.left+f+c;j<g.left?e.left=g.left-j:k>g.right&&(e.left=g.left+g.width-k)}return e},c.prototype.getTitle=function(){var a,b=this.$element,c=this.options;return a=b.attr("data-original-title")||("function"==typeof c.title?c.title.call(b[0]):c.title)},c.prototype.getUID=function(a){do a+=~~(1e6*Math.random());while(document.getElementById(a));return a},c.prototype.tip=function(){if(!this.$tip&&(this.$tip=a(this.options.template),1!=this.$tip.length))throw new Error(this.type+" `template` option must consist of exactly 1 top-level element!");return this.$tip},c.prototype.arrow=function(){return this.$arrow=this.$arrow||this.tip().find(".tooltip-arrow")},c.prototype.enable=function(){this.enabled=!0},c.prototype.disable=function(){this.enabled=!1},c.prototype.toggleEnabled=function(){this.enabled=!this.enabled},c.prototype.toggle=function(b){var c=this;b&&(c=a(b.currentTarget).data("bs."+this.type),c||(c=new this.constructor(b.currentTarget,this.getDelegateOptions()),a(b.currentTarget).data("bs."+this.type,c))),b?(c.inState.click=!c.inState.click,c.isInStateTrue()?c.enter(c):c.leave(c)):c.tip().hasClass("in")?c.leave(c):c.enter(c)},c.prototype.destroy=function(){var a=this;clearTimeout(this.timeout),this.hide(function(){a.$element.off("."+a.type).removeData("bs."+a.type),a.$tip&&a.$tip.detach(),a.$tip=null,a.$arrow=null,a.$viewport=null})};var d=a.fn.tooltip;a.fn.tooltip=b,a.fn.tooltip.Constructor=c,a.fn.tooltip.noConflict=function(){return a.fn.tooltip=d,this}}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var d=a(this),e=d.data("bs.popover"),f="object"==typeof b&&b;(e||!/destroy|hide/.test(b))&&(e||d.data("bs.popover",e=new c(this,f)),"string"==typeof b&&e[b]())})}var c=function(a,b){this.init("popover",a,b)};if(!a.fn.tooltip)throw new Error("Popover requires tooltip.js");c.VERSION="3.3.5",c.DEFAULTS=a.extend({},a.fn.tooltip.Constructor.DEFAULTS,{placement:"right",trigger:"click",content:"",template:'<div class="popover" role="tooltip"><div class="arrow"></div><h3 class="popover-title"></h3><div class="popover-content"></div></div>'}),c.prototype=a.extend({},a.fn.tooltip.Constructor.prototype),c.prototype.constructor=c,c.prototype.getDefaults=function(){return c.DEFAULTS},c.prototype.setContent=function(){var a=this.tip(),b=this.getTitle(),c=this.getContent();a.find(".popover-title")[this.options.html?"html":"text"](b),a.find(".popover-content").children().detach().end()[this.options.html?"string"==typeof c?"html":"append":"text"](c),a.removeClass("fade top bottom left right in"),a.find(".popover-title").html()||a.find(".popover-title").hide()},c.prototype.hasContent=function(){return this.getTitle()||this.getContent()},c.prototype.getContent=function(){var a=this.$element,b=this.options;return a.attr("data-content")||("function"==typeof b.content?b.content.call(a[0]):b.content)},c.prototype.arrow=function(){return this.$arrow=this.$arrow||this.tip().find(".arrow")};var d=a.fn.popover;a.fn.popover=b,a.fn.popover.Constructor=c,a.fn.popover.noConflict=function(){return a.fn.popover=d,this}}(jQuery),+function(a){"use strict";function b(c,d){this.$body=a(document.body),this.$scrollElement=a(a(c).is(document.body)?window:c),this.options=a.extend({},b.DEFAULTS,d),this.selector=(this.options.target||"")+" .nav li > a",this.offsets=[],this.targets=[],this.activeTarget=null,this.scrollHeight=0,this.$scrollElement.on("scroll.bs.scrollspy",a.proxy(this.process,this)),this.refresh(),this.process()}function c(c){return this.each(function(){var d=a(this),e=d.data("bs.scrollspy"),f="object"==typeof c&&c;e||d.data("bs.scrollspy",e=new b(this,f)),"string"==typeof c&&e[c]()})}b.VERSION="3.3.5",b.DEFAULTS={offset:10},b.prototype.getScrollHeight=function(){return this.$scrollElement[0].scrollHeight||Math.max(this.$body[0].scrollHeight,document.documentElement.scrollHeight)},b.prototype.refresh=function(){var b=this,c="offset",d=0;this.offsets=[],this.targets=[],this.scrollHeight=this.getScrollHeight(),a.isWindow(this.$scrollElement[0])||(c="position",d=this.$scrollElement.scrollTop()),this.$body.find(this.selector).map(function(){var b=a(this),e=b.data("target")||b.attr("href"),f=/^#./.test(e)&&a(e);return f&&f.length&&f.is(":visible")&&[[f[c]().top+d,e]]||null}).sort(function(a,b){return a[0]-b[0]}).each(function(){b.offsets.push(this[0]),b.targets.push(this[1])})},b.prototype.process=function(){var a,b=this.$scrollElement.scrollTop()+this.options.offset,c=this.getScrollHeight(),d=this.options.offset+c-this.$scrollElement.height(),e=this.offsets,f=this.targets,g=this.activeTarget;if(this.scrollHeight!=c&&this.refresh(),b>=d)return g!=(a=f[f.length-1])&&this.activate(a);if(g&&b<e[0])return this.activeTarget=null,this.clear();for(a=e.length;a--;)g!=f[a]&&b>=e[a]&&(void 0===e[a+1]||b<e[a+1])&&this.activate(f[a])},b.prototype.activate=function(b){this.activeTarget=b,this.clear();var c=this.selector+'[data-target="'+b+'"],'+this.selector+'[href="'+b+'"]',d=a(c).parents("li").addClass("active");d.parent(".dropdown-menu").length&&(d=d.closest("li.dropdown").addClass("active")),
d.trigger("activate.bs.scrollspy")},b.prototype.clear=function(){a(this.selector).parentsUntil(this.options.target,".active").removeClass("active")};var d=a.fn.scrollspy;a.fn.scrollspy=c,a.fn.scrollspy.Constructor=b,a.fn.scrollspy.noConflict=function(){return a.fn.scrollspy=d,this},a(window).on("load.bs.scrollspy.data-api",function(){a('[data-spy="scroll"]').each(function(){var b=a(this);c.call(b,b.data())})})}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var d=a(this),e=d.data("bs.tab");e||d.data("bs.tab",e=new c(this)),"string"==typeof b&&e[b]()})}var c=function(b){this.element=a(b)};c.VERSION="3.3.5",c.TRANSITION_DURATION=150,c.prototype.show=function(){var b=this.element,c=b.closest("ul:not(.dropdown-menu)"),d=b.data("target");if(d||(d=b.attr("href"),d=d&&d.replace(/.*(?=#[^\s]*$)/,"")),!b.parent("li").hasClass("active")){var e=c.find(".active:last a"),f=a.Event("hide.bs.tab",{relatedTarget:b[0]}),g=a.Event("show.bs.tab",{relatedTarget:e[0]});if(e.trigger(f),b.trigger(g),!g.isDefaultPrevented()&&!f.isDefaultPrevented()){var h=a(d);this.activate(b.closest("li"),c),this.activate(h,h.parent(),function(){e.trigger({type:"hidden.bs.tab",relatedTarget:b[0]}),b.trigger({type:"shown.bs.tab",relatedTarget:e[0]})})}}},c.prototype.activate=function(b,d,e){function f(){g.removeClass("active").find("> .dropdown-menu > .active").removeClass("active").end().find('[data-toggle="tab"]').attr("aria-expanded",!1),b.addClass("active").find('[data-toggle="tab"]').attr("aria-expanded",!0),h?(b[0].offsetWidth,b.addClass("in")):b.removeClass("fade"),b.parent(".dropdown-menu").length&&b.closest("li.dropdown").addClass("active").end().find('[data-toggle="tab"]').attr("aria-expanded",!0),e&&e()}var g=d.find("> .active"),h=e&&a.support.transition&&(g.length&&g.hasClass("fade")||!!d.find("> .fade").length);g.length&&h?g.one("bsTransitionEnd",f).emulateTransitionEnd(c.TRANSITION_DURATION):f(),g.removeClass("in")};var d=a.fn.tab;a.fn.tab=b,a.fn.tab.Constructor=c,a.fn.tab.noConflict=function(){return a.fn.tab=d,this};var e=function(c){c.preventDefault(),b.call(a(this),"show")};a(document).on("click.bs.tab.data-api",'[data-toggle="tab"]',e).on("click.bs.tab.data-api",'[data-toggle="pill"]',e)}(jQuery),+function(a){"use strict";function b(b){return this.each(function(){var d=a(this),e=d.data("bs.affix"),f="object"==typeof b&&b;e||d.data("bs.affix",e=new c(this,f)),"string"==typeof b&&e[b]()})}var c=function(b,d){this.options=a.extend({},c.DEFAULTS,d),this.$target=a(this.options.target).on("scroll.bs.affix.data-api",a.proxy(this.checkPosition,this)).on("click.bs.affix.data-api",a.proxy(this.checkPositionWithEventLoop,this)),this.$element=a(b),this.affixed=null,this.unpin=null,this.pinnedOffset=null,this.checkPosition()};c.VERSION="3.3.5",c.RESET="affix affix-top affix-bottom",c.DEFAULTS={offset:0,target:window},c.prototype.getState=function(a,b,c,d){var e=this.$target.scrollTop(),f=this.$element.offset(),g=this.$target.height();if(null!=c&&"top"==this.affixed)return c>e?"top":!1;if("bottom"==this.affixed)return null!=c?e+this.unpin<=f.top?!1:"bottom":a-d>=e+g?!1:"bottom";var h=null==this.affixed,i=h?e:f.top,j=h?g:b;return null!=c&&c>=e?"top":null!=d&&i+j>=a-d?"bottom":!1},c.prototype.getPinnedOffset=function(){if(this.pinnedOffset)return this.pinnedOffset;this.$element.removeClass(c.RESET).addClass("affix");var a=this.$target.scrollTop(),b=this.$element.offset();return this.pinnedOffset=b.top-a},c.prototype.checkPositionWithEventLoop=function(){setTimeout(a.proxy(this.checkPosition,this),1)},c.prototype.checkPosition=function(){if(this.$element.is(":visible")){var b=this.$element.height(),d=this.options.offset,e=d.top,f=d.bottom,g=Math.max(a(document).height(),a(document.body).height());"object"!=typeof d&&(f=e=d),"function"==typeof e&&(e=d.top(this.$element)),"function"==typeof f&&(f=d.bottom(this.$element));var h=this.getState(g,b,e,f);if(this.affixed!=h){null!=this.unpin&&this.$element.css("top","");var i="affix"+(h?"-"+h:""),j=a.Event(i+".bs.affix");if(this.$element.trigger(j),j.isDefaultPrevented())return;this.affixed=h,this.unpin="bottom"==h?this.getPinnedOffset():null,this.$element.removeClass(c.RESET).addClass(i).trigger(i.replace("affix","affixed")+".bs.affix")}"bottom"==h&&this.$element.offset({top:g-b-f})}};var d=a.fn.affix;a.fn.affix=b,a.fn.affix.Constructor=c,a.fn.affix.noConflict=function(){return a.fn.affix=d,this},a(window).on("load",function(){a('[data-spy="affix"]').each(function(){var c=a(this),d=c.data();d.offset=d.offset||{},null!=d.offsetBottom&&(d.offset.bottom=d.offsetBottom),null!=d.offsetTop&&(d.offset.top=d.offsetTop),b.call(c,d)})})}(jQuery);</script>
<script>/**
* @preserve HTML5 Shiv 3.7.2 | @afarkas @jdalton @jon_neal @rem | MIT/GPL2 Licensed
*/
// Only run this code in IE 8
if (!!window.navigator.userAgent.match("MSIE 8")) {
!function(a,b){function c(a,b){var c=a.createElement("p"),d=a.getElementsByTagName("head")[0]||a.documentElement;return c.innerHTML="x<style>"+b+"</style>",d.insertBefore(c.lastChild,d.firstChild)}function d(){var a=t.elements;return"string"==typeof a?a.split(" "):a}function e(a,b){var c=t.elements;"string"!=typeof c&&(c=c.join(" ")),"string"!=typeof a&&(a=a.join(" ")),t.elements=c+" "+a,j(b)}function f(a){var b=s[a[q]];return b||(b={},r++,a[q]=r,s[r]=b),b}function g(a,c,d){if(c||(c=b),l)return c.createElement(a);d||(d=f(c));var e;return e=d.cache[a]?d.cache[a].cloneNode():p.test(a)?(d.cache[a]=d.createElem(a)).cloneNode():d.createElem(a),!e.canHaveChildren||o.test(a)||e.tagUrn?e:d.frag.appendChild(e)}function h(a,c){if(a||(a=b),l)return a.createDocumentFragment();c=c||f(a);for(var e=c.frag.cloneNode(),g=0,h=d(),i=h.length;i>g;g++)e.createElement(h[g]);return e}function i(a,b){b.cache||(b.cache={},b.createElem=a.createElement,b.createFrag=a.createDocumentFragment,b.frag=b.createFrag()),a.createElement=function(c){return t.shivMethods?g(c,a,b):b.createElem(c)},a.createDocumentFragment=Function("h,f","return function(){var n=f.cloneNode(),c=n.createElement;h.shivMethods&&("+d().join().replace(/[\w\-:]+/g,function(a){return b.createElem(a),b.frag.createElement(a),'c("'+a+'")'})+");return n}")(t,b.frag)}function j(a){a||(a=b);var d=f(a);return!t.shivCSS||k||d.hasCSS||(d.hasCSS=!!c(a,"article,aside,dialog,figcaption,figure,footer,header,hgroup,main,nav,section{display:block}mark{background:#FF0;color:#000}template{display:none}")),l||i(a,d),a}var k,l,m="3.7.2",n=a.html5||{},o=/^<|^(?:button|map|select|textarea|object|iframe|option|optgroup)$/i,p=/^(?:a|b|code|div|fieldset|h1|h2|h3|h4|h5|h6|i|label|li|ol|p|q|span|strong|style|table|tbody|td|th|tr|ul)$/i,q="_html5shiv",r=0,s={};!function(){try{var a=b.createElement("a");a.innerHTML="<xyz></xyz>",k="hidden"in a,l=1==a.childNodes.length||function(){b.createElement("a");var a=b.createDocumentFragment();return"undefined"==typeof a.cloneNode||"undefined"==typeof a.createDocumentFragment||"undefined"==typeof a.createElement}()}catch(c){k=!0,l=!0}}();var t={elements:n.elements||"abbr article aside audio bdi canvas data datalist details dialog figcaption figure footer header hgroup main mark meter nav output picture progress section summary template time video",version:m,shivCSS:n.shivCSS!==!1,supportsUnknownElements:l,shivMethods:n.shivMethods!==!1,type:"default",shivDocument:j,createElement:g,createDocumentFragment:h,addElements:e};a.html5=t,j(b)}(this,document);
};
</script>
<script>/*! Respond.js v1.4.2: min/max-width media query polyfill * Copyright 2013 Scott Jehl
 * Licensed under https://github.com/scottjehl/Respond/blob/master/LICENSE-MIT
 *  */

// Only run this code in IE 8
if (!!window.navigator.userAgent.match("MSIE 8")) {
!function(a){"use strict";a.matchMedia=a.matchMedia||function(a){var b,c=a.documentElement,d=c.firstElementChild||c.firstChild,e=a.createElement("body"),f=a.createElement("div");return f.id="mq-test-1",f.style.cssText="position:absolute;top:-100em",e.style.background="none",e.appendChild(f),function(a){return f.innerHTML='&shy;<style media="'+a+'"> #mq-test-1 { width: 42px; }</style>',c.insertBefore(e,d),b=42===f.offsetWidth,c.removeChild(e),{matches:b,media:a}}}(a.document)}(this),function(a){"use strict";function b(){u(!0)}var c={};a.respond=c,c.update=function(){};var d=[],e=function(){var b=!1;try{b=new a.XMLHttpRequest}catch(c){b=new a.ActiveXObject("Microsoft.XMLHTTP")}return function(){return b}}(),f=function(a,b){var c=e();c&&(c.open("GET",a,!0),c.onreadystatechange=function(){4!==c.readyState||200!==c.status&&304!==c.status||b(c.responseText)},4!==c.readyState&&c.send(null))};if(c.ajax=f,c.queue=d,c.regex={media:/@media[^\{]+\{([^\{\}]*\{[^\}\{]*\})+/gi,keyframes:/@(?:\-(?:o|moz|webkit)\-)?keyframes[^\{]+\{(?:[^\{\}]*\{[^\}\{]*\})+[^\}]*\}/gi,urls:/(url\()['"]?([^\/\)'"][^:\)'"]+)['"]?(\))/g,findStyles:/@media *([^\{]+)\{([\S\s]+?)$/,only:/(only\s+)?([a-zA-Z]+)\s?/,minw:/\([\s]*min\-width\s*:[\s]*([\s]*[0-9\.]+)(px|em)[\s]*\)/,maxw:/\([\s]*max\-width\s*:[\s]*([\s]*[0-9\.]+)(px|em)[\s]*\)/},c.mediaQueriesSupported=a.matchMedia&&null!==a.matchMedia("only all")&&a.matchMedia("only all").matches,!c.mediaQueriesSupported){var g,h,i,j=a.document,k=j.documentElement,l=[],m=[],n=[],o={},p=30,q=j.getElementsByTagName("head")[0]||k,r=j.getElementsByTagName("base")[0],s=q.getElementsByTagName("link"),t=function(){var a,b=j.createElement("div"),c=j.body,d=k.style.fontSize,e=c&&c.style.fontSize,f=!1;return b.style.cssText="position:absolute;font-size:1em;width:1em",c||(c=f=j.createElement("body"),c.style.background="none"),k.style.fontSize="100%",c.style.fontSize="100%",c.appendChild(b),f&&k.insertBefore(c,k.firstChild),a=b.offsetWidth,f?k.removeChild(c):c.removeChild(b),k.style.fontSize=d,e&&(c.style.fontSize=e),a=i=parseFloat(a)},u=function(b){var c="clientWidth",d=k[c],e="CSS1Compat"===j.compatMode&&d||j.body[c]||d,f={},o=s[s.length-1],r=(new Date).getTime();if(b&&g&&p>r-g)return a.clearTimeout(h),h=a.setTimeout(u,p),void 0;g=r;for(var v in l)if(l.hasOwnProperty(v)){var w=l[v],x=w.minw,y=w.maxw,z=null===x,A=null===y,B="em";x&&(x=parseFloat(x)*(x.indexOf(B)>-1?i||t():1)),y&&(y=parseFloat(y)*(y.indexOf(B)>-1?i||t():1)),w.hasquery&&(z&&A||!(z||e>=x)||!(A||y>=e))||(f[w.media]||(f[w.media]=[]),f[w.media].push(m[w.rules]))}for(var C in n)n.hasOwnProperty(C)&&n[C]&&n[C].parentNode===q&&q.removeChild(n[C]);n.length=0;for(var D in f)if(f.hasOwnProperty(D)){var E=j.createElement("style"),F=f[D].join("\n");E.type="text/css",E.media=D,q.insertBefore(E,o.nextSibling),E.styleSheet?E.styleSheet.cssText=F:E.appendChild(j.createTextNode(F)),n.push(E)}},v=function(a,b,d){var e=a.replace(c.regex.keyframes,"").match(c.regex.media),f=e&&e.length||0;b=b.substring(0,b.lastIndexOf("/"));var g=function(a){return a.replace(c.regex.urls,"$1"+b+"$2$3")},h=!f&&d;b.length&&(b+="/"),h&&(f=1);for(var i=0;f>i;i++){var j,k,n,o;h?(j=d,m.push(g(a))):(j=e[i].match(c.regex.findStyles)&&RegExp.$1,m.push(RegExp.$2&&g(RegExp.$2))),n=j.split(","),o=n.length;for(var p=0;o>p;p++)k=n[p],l.push({media:k.split("(")[0].match(c.regex.only)&&RegExp.$2||"all",rules:m.length-1,hasquery:k.indexOf("(")>-1,minw:k.match(c.regex.minw)&&parseFloat(RegExp.$1)+(RegExp.$2||""),maxw:k.match(c.regex.maxw)&&parseFloat(RegExp.$1)+(RegExp.$2||"")})}u()},w=function(){if(d.length){var b=d.shift();f(b.href,function(c){v(c,b.href,b.media),o[b.href]=!0,a.setTimeout(function(){w()},0)})}},x=function(){for(var b=0;b<s.length;b++){var c=s[b],e=c.href,f=c.media,g=c.rel&&"stylesheet"===c.rel.toLowerCase();e&&g&&!o[e]&&(c.styleSheet&&c.styleSheet.rawCssText?(v(c.styleSheet.rawCssText,e,f),o[e]=!0):(!/^([a-zA-Z:]*\/\/)/.test(e)&&!r||e.replace(RegExp.$1,"").split("/")[0]===a.location.host)&&("//"===e.substring(0,2)&&(e=a.location.protocol+e),d.push({href:e,media:f})))}w()};x(),c.update=x,c.getEmValue=t,a.addEventListener?a.addEventListener("resize",b,!1):a.attachEvent&&a.attachEvent("onresize",b)}}(this);
};
</script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script>

/**
 * jQuery Plugin: Sticky Tabs
 *
 * @author Aidan Lister <aidan@php.net>
 * adapted by Ruben Arslan to activate parent tabs too
 * http://www.aidanlister.com/2014/03/persisting-the-tab-state-in-bootstrap/
 */
(function($) {
  "use strict";
  $.fn.rmarkdownStickyTabs = function() {
    var context = this;
    // Show the tab corresponding with the hash in the URL, or the first tab
    var showStuffFromHash = function() {
      var hash = window.location.hash;
      var selector = hash ? 'a[href="' + hash + '"]' : 'li.active > a';
      var $selector = $(selector, context);
      if($selector.data('toggle') === "tab") {
        $selector.tab('show');
        // walk up the ancestors of this element, show any hidden tabs
        $selector.parents('.section.tabset').each(function(i, elm) {
          var link = $('a[href="#' + $(elm).attr('id') + '"]');
          if(link.data('toggle') === "tab") {
            link.tab("show");
          }
        });
      }
    };


    // Set the correct tab when the page loads
    showStuffFromHash(context);

    // Set the correct tab when a user uses their back/forward button
    $(window).on('hashchange', function() {
      showStuffFromHash(context);
    });

    // Change the URL when tabs are clicked
    $('a', context).on('click', function(e) {
      history.pushState(null, null, this.href);
      showStuffFromHash(context);
    });

    return this;
  };
}(jQuery));

window.buildTabsets = function(tocID) {

  // build a tabset from a section div with the .tabset class
  function buildTabset(tabset) {

    // check for fade and pills options
    var fade = tabset.hasClass("tabset-fade");
    var pills = tabset.hasClass("tabset-pills");
    var navClass = pills ? "nav-pills" : "nav-tabs";

    // determine the heading level of the tabset and tabs
    var match = tabset.attr('class').match(/level(\d) /);
    if (match === null)
      return;
    var tabsetLevel = Number(match[1]);
    var tabLevel = tabsetLevel + 1;

    // find all subheadings immediately below
    var tabs = tabset.find("div.section.level" + tabLevel);
    if (!tabs.length)
      return;

    // create tablist and tab-content elements
    var tabList = $('<ul class="nav ' + navClass + '" role="tablist"></ul>');
    $(tabs[0]).before(tabList);
    var tabContent = $('<div class="tab-content"></div>');
    $(tabs[0]).before(tabContent);

    // build the tabset
    var activeTab = 0;
    tabs.each(function(i) {

      // get the tab div
      var tab = $(tabs[i]);

      // get the id then sanitize it for use with bootstrap tabs
      var id = tab.attr('id');

      // see if this is marked as the active tab
      if (tab.hasClass('active'))
        activeTab = i;

      // remove any table of contents entries associated with
      // this ID (since we'll be removing the heading element)
      $("div#" + tocID + " li a[href='#" + id + "']").parent().remove();

      // sanitize the id for use with bootstrap tabs
      id = id.replace(/[.\/?&!#<>]/g, '').replace(/\s/g, '_');
      tab.attr('id', id);

      // get the heading element within it, grab it's text, then remove it
      var heading = tab.find('h' + tabLevel + ':first');
      var headingText = heading.html();
      heading.remove();

      // build and append the tab list item
      var a = $('<a role="tab" data-toggle="tab">' + headingText + '</a>');
      a.attr('href', '#' + id);
      a.attr('aria-controls', id);
      var li = $('<li role="presentation"></li>');
      li.append(a);
      tabList.append(li);

      // set it's attributes
      tab.attr('role', 'tabpanel');
      tab.addClass('tab-pane');
      tab.addClass('tabbed-pane');
      if (fade)
        tab.addClass('fade');

      // move it into the tab content div
      tab.detach().appendTo(tabContent);
    });

    // set active tab
    $(tabList.children('li')[activeTab]).addClass('active');
    var active = $(tabContent.children('div.section')[activeTab]);
    active.addClass('active');
    if (fade)
      active.addClass('in');

    if (tabset.hasClass("tabset-sticky"))
      tabset.rmarkdownStickyTabs();
  }

  // convert section divs with the .tabset class to tabsets
  var tabsets = $("div.section.tabset");
  tabsets.each(function(i) {
    buildTabset($(tabsets[i]));
  });
};

</script>
<style type="text/css">.hljs-literal {
color: #990073;
}
.hljs-number {
color: #099;
}
.hljs-comment {
color: #998;
font-style: italic;
}
.hljs-keyword {
color: #900;
font-weight: bold;
}
.hljs-string {
color: #d14;
}
</style>
<script src="data:application/javascript;base64,/*! highlight.js v9.12.0 | BSD3 License | git.io/hljslicense */
!function(e){var n="object"==typeof window&&window||"object"==typeof self&&self;"undefined"!=typeof exports?e(exports):n&&(n.hljs=e({}),"function"==typeof define&&define.amd&&define([],function(){return n.hljs}))}(function(e){function n(e){return e.replace(/&/g,"&amp;").replace(/</g,"&lt;").replace(/>/g,"&gt;")}function t(e){return e.nodeName.toLowerCase()}function r(e,n){var t=e&&e.exec(n);return t&&0===t.index}function a(e){return k.test(e)}function i(e){var n,t,r,i,o=e.className+" ";if(o+=e.parentNode?e.parentNode.className:"",t=B.exec(o))return w(t[1])?t[1]:"no-highlight";for(o=o.split(/\s+/),n=0,r=o.length;r>n;n++)if(i=o[n],a(i)||w(i))return i}function o(e){var n,t={},r=Array.prototype.slice.call(arguments,1);for(n in e)t[n]=e[n];return r.forEach(function(e){for(n in e)t[n]=e[n]}),t}function u(e){var n=[];return function r(e,a){for(var i=e.firstChild;i;i=i.nextSibling)3===i.nodeType?a+=i.nodeValue.length:1===i.nodeType&&(n.push({event:"start",offset:a,node:i}),a=r(i,a),t(i).match(/br|hr|img|input/)||n.push({event:"stop",offset:a,node:i}));return a}(e,0),n}function c(e,r,a){function i(){return e.length&&r.length?e[0].offset!==r[0].offset?e[0].offset<r[0].offset?e:r:"start"===r[0].event?e:r:e.length?e:r}function o(e){function r(e){return" "+e.nodeName+'="'+n(e.value).replace('"',"&quot;")+'"'}s+="<"+t(e)+E.map.call(e.attributes,r).join("")+">"}function u(e){s+="</"+t(e)+">"}function c(e){("start"===e.event?o:u)(e.node)}for(var l=0,s="",f=[];e.length||r.length;){var g=i();if(s+=n(a.substring(l,g[0].offset)),l=g[0].offset,g===e){f.reverse().forEach(u);do c(g.splice(0,1)[0]),g=i();while(g===e&&g.length&&g[0].offset===l);f.reverse().forEach(o)}else"start"===g[0].event?f.push(g[0].node):f.pop(),c(g.splice(0,1)[0])}return s+n(a.substr(l))}function l(e){return e.v&&!e.cached_variants&&(e.cached_variants=e.v.map(function(n){return o(e,{v:null},n)})),e.cached_variants||e.eW&&[o(e)]||[e]}function s(e){function n(e){return e&&e.source||e}function t(t,r){return new RegExp(n(t),"m"+(e.cI?"i":"")+(r?"g":""))}function r(a,i){if(!a.compiled){if(a.compiled=!0,a.k=a.k||a.bK,a.k){var o={},u=function(n,t){e.cI&&(t=t.toLowerCase()),t.split(" ").forEach(function(e){var t=e.split("|");o[t[0]]=[n,t[1]?Number(t[1]):1]})};"string"==typeof a.k?u("keyword",a.k):x(a.k).forEach(function(e){u(e,a.k[e])}),a.k=o}a.lR=t(a.l||/\w+/,!0),i&&(a.bK&&(a.b="\\b("+a.bK.split(" ").join("|")+")\\b"),a.b||(a.b=/\B|\b/),a.bR=t(a.b),a.e||a.eW||(a.e=/\B|\b/),a.e&&(a.eR=t(a.e)),a.tE=n(a.e)||"",a.eW&&i.tE&&(a.tE+=(a.e?"|":"")+i.tE)),a.i&&(a.iR=t(a.i)),null==a.r&&(a.r=1),a.c||(a.c=[]),a.c=Array.prototype.concat.apply([],a.c.map(function(e){return l("self"===e?a:e)})),a.c.forEach(function(e){r(e,a)}),a.starts&&r(a.starts,i);var c=a.c.map(function(e){return e.bK?"\\.?("+e.b+")\\.?":e.b}).concat([a.tE,a.i]).map(n).filter(Boolean);a.t=c.length?t(c.join("|"),!0):{exec:function(){return null}}}}r(e)}function f(e,t,a,i){function o(e,n){var t,a;for(t=0,a=n.c.length;a>t;t++)if(r(n.c[t].bR,e))return n.c[t]}function u(e,n){if(r(e.eR,n)){for(;e.endsParent&&e.parent;)e=e.parent;return e}return e.eW?u(e.parent,n):void 0}function c(e,n){return!a&&r(n.iR,e)}function l(e,n){var t=N.cI?n[0].toLowerCase():n[0];return e.k.hasOwnProperty(t)&&e.k[t]}function p(e,n,t,r){var a=r?"":I.classPrefix,i='<span class="'+a,o=t?"":C;return i+=e+'">',i+n+o}function h(){var e,t,r,a;if(!E.k)return n(k);for(a="",t=0,E.lR.lastIndex=0,r=E.lR.exec(k);r;)a+=n(k.substring(t,r.index)),e=l(E,r),e?(B+=e[1],a+=p(e[0],n(r[0]))):a+=n(r[0]),t=E.lR.lastIndex,r=E.lR.exec(k);return a+n(k.substr(t))}function d(){var e="string"==typeof E.sL;if(e&&!y[E.sL])return n(k);var t=e?f(E.sL,k,!0,x[E.sL]):g(k,E.sL.length?E.sL:void 0);return E.r>0&&(B+=t.r),e&&(x[E.sL]=t.top),p(t.language,t.value,!1,!0)}function b(){L+=null!=E.sL?d():h(),k=""}function v(e){L+=e.cN?p(e.cN,"",!0):"",E=Object.create(e,{parent:{value:E}})}function m(e,n){if(k+=e,null==n)return b(),0;var t=o(n,E);if(t)return t.skip?k+=n:(t.eB&&(k+=n),b(),t.rB||t.eB||(k=n)),v(t,n),t.rB?0:n.length;var r=u(E,n);if(r){var a=E;a.skip?k+=n:(a.rE||a.eE||(k+=n),b(),a.eE&&(k=n));do E.cN&&(L+=C),E.skip||(B+=E.r),E=E.parent;while(E!==r.parent);return r.starts&&v(r.starts,""),a.rE?0:n.length}if(c(n,E))throw new Error('Illegal lexeme "'+n+'" for mode "'+(E.cN||"<unnamed>")+'"');return k+=n,n.length||1}var N=w(e);if(!N)throw new Error('Unknown language: "'+e+'"');s(N);var R,E=i||N,x={},L="";for(R=E;R!==N;R=R.parent)R.cN&&(L=p(R.cN,"",!0)+L);var k="",B=0;try{for(var M,j,O=0;;){if(E.t.lastIndex=O,M=E.t.exec(t),!M)break;j=m(t.substring(O,M.index),M[0]),O=M.index+j}for(m(t.substr(O)),R=E;R.parent;R=R.parent)R.cN&&(L+=C);return{r:B,value:L,language:e,top:E}}catch(T){if(T.message&&-1!==T.message.indexOf("Illegal"))return{r:0,value:n(t)};throw T}}function g(e,t){t=t||I.languages||x(y);var r={r:0,value:n(e)},a=r;return t.filter(w).forEach(function(n){var t=f(n,e,!1);t.language=n,t.r>a.r&&(a=t),t.r>r.r&&(a=r,r=t)}),a.language&&(r.second_best=a),r}function p(e){return I.tabReplace||I.useBR?e.replace(M,function(e,n){return I.useBR&&"\n"===e?"<br>":I.tabReplace?n.replace(/\t/g,I.tabReplace):""}):e}function h(e,n,t){var r=n?L[n]:t,a=[e.trim()];return e.match(/\bhljs\b/)||a.push("hljs"),-1===e.indexOf(r)&&a.push(r),a.join(" ").trim()}function d(e){var n,t,r,o,l,s=i(e);a(s)||(I.useBR?(n=document.createElementNS("http://www.w3.org/1999/xhtml","div"),n.innerHTML=e.innerHTML.replace(/\n/g,"").replace(/<br[ \/]*>/g,"\n")):n=e,l=n.textContent,r=s?f(s,l,!0):g(l),t=u(n),t.length&&(o=document.createElementNS("http://www.w3.org/1999/xhtml","div"),o.innerHTML=r.value,r.value=c(t,u(o),l)),r.value=p(r.value),e.innerHTML=r.value,e.className=h(e.className,s,r.language),e.result={language:r.language,re:r.r},r.second_best&&(e.second_best={language:r.second_best.language,re:r.second_best.r}))}function b(e){I=o(I,e)}function v(){if(!v.called){v.called=!0;var e=document.querySelectorAll("pre code");E.forEach.call(e,d)}}function m(){addEventListener("DOMContentLoaded",v,!1),addEventListener("load",v,!1)}function N(n,t){var r=y[n]=t(e);r.aliases&&r.aliases.forEach(function(e){L[e]=n})}function R(){return x(y)}function w(e){return e=(e||"").toLowerCase(),y[e]||y[L[e]]}var E=[],x=Object.keys,y={},L={},k=/^(no-?highlight|plain|text)$/i,B=/\blang(?:uage)?-([\w-]+)\b/i,M=/((^(<[^>]+>|\t|)+|(?:\n)))/gm,C="</span>",I={classPrefix:"hljs-",tabReplace:null,useBR:!1,languages:void 0};return e.highlight=f,e.highlightAuto=g,e.fixMarkup=p,e.highlightBlock=d,e.configure=b,e.initHighlighting=v,e.initHighlightingOnLoad=m,e.registerLanguage=N,e.listLanguages=R,e.getLanguage=w,e.inherit=o,e.IR="[a-zA-Z]\\w*",e.UIR="[a-zA-Z_]\\w*",e.NR="\\b\\d+(\\.\\d+)?",e.CNR="(-?)(\\b0[xX][a-fA-F0-9]+|(\\b\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)",e.BNR="\\b(0b[01]+)",e.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|-|-=|/=|/|:|;|<<|<<=|<=|<|===|==|=|>>>=|>>=|>=|>>>|>>|>|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~",e.BE={b:"\\\\[\\s\\S]",r:0},e.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[e.BE]},e.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[e.BE]},e.PWM={b:/\b(a|an|the|are|I'm|isn't|don't|doesn't|won't|but|just|should|pretty|simply|enough|gonna|going|wtf|so|such|will|you|your|they|like|more)\b/},e.C=function(n,t,r){var a=e.inherit({cN:"comment",b:n,e:t,c:[]},r||{});return a.c.push(e.PWM),a.c.push({cN:"doctag",b:"(?:TODO|FIXME|NOTE|BUG|XXX):",r:0}),a},e.CLCM=e.C("//","$"),e.CBCM=e.C("/\\*","\\*/"),e.HCM=e.C("#","$"),e.NM={cN:"number",b:e.NR,r:0},e.CNM={cN:"number",b:e.CNR,r:0},e.BNM={cN:"number",b:e.BNR,r:0},e.CSSNM={cN:"number",b:e.NR+"(%|em|ex|ch|rem|vw|vh|vmin|vmax|cm|mm|in|pt|pc|px|deg|grad|rad|turn|s|ms|Hz|kHz|dpi|dpcm|dppx)?",r:0},e.RM={cN:"regexp",b:/\//,e:/\/[gimuy]*/,i:/\n/,c:[e.BE,{b:/\[/,e:/\]/,r:0,c:[e.BE]}]},e.TM={cN:"title",b:e.IR,r:0},e.UTM={cN:"title",b:e.UIR,r:0},e.METHOD_GUARD={b:"\\.\\s*"+e.UIR,r:0},e});hljs.registerLanguage("sql",function(e){var t=e.C("--","$");return{cI:!0,i:/[<>{}*#]/,c:[{bK:"begin end start commit rollback savepoint lock alter create drop rename call delete do handler insert load replace select truncate update set show pragma grant merge describe use explain help declare prepare execute deallocate release unlock purge reset change stop analyze cache flush optimize repair kill install uninstall checksum restore check backup revoke comment",e:/;/,eW:!0,l:/[\w\.]+/,k:{keyword:"abort abs absolute acc acce accep accept access accessed accessible account acos action activate add addtime admin administer advanced advise aes_decrypt aes_encrypt after agent aggregate ali alia alias allocate allow alter always analyze ancillary and any anydata anydataset anyschema anytype apply archive archived archivelog are as asc ascii asin assembly assertion associate asynchronous at atan atn2 attr attri attrib attribu attribut attribute attributes audit authenticated authentication authid authors auto autoallocate autodblink autoextend automatic availability avg backup badfile basicfile before begin beginning benchmark between bfile bfile_base big bigfile bin binary_double binary_float binlog bit_and bit_count bit_length bit_or bit_xor bitmap blob_base block blocksize body both bound buffer_cache buffer_pool build bulk by byte byteordermark bytes cache caching call calling cancel capacity cascade cascaded case cast catalog category ceil ceiling chain change changed char_base char_length character_length characters characterset charindex charset charsetform charsetid check checksum checksum_agg child choose chr chunk class cleanup clear client clob clob_base clone close cluster_id cluster_probability cluster_set clustering coalesce coercibility col collate collation collect colu colum column column_value columns columns_updated comment commit compact compatibility compiled complete composite_limit compound compress compute concat concat_ws concurrent confirm conn connec connect connect_by_iscycle connect_by_isleaf connect_by_root connect_time connection consider consistent constant constraint constraints constructor container content contents context contributors controlfile conv convert convert_tz corr corr_k corr_s corresponding corruption cos cost count count_big counted covar_pop covar_samp cpu_per_call cpu_per_session crc32 create creation critical cross cube cume_dist curdate current current_date current_time current_timestamp current_user cursor curtime customdatum cycle data database databases datafile datafiles datalength date_add date_cache date_format date_sub dateadd datediff datefromparts datename datepart datetime2fromparts day day_to_second dayname dayofmonth dayofweek dayofyear days db_role_change dbtimezone ddl deallocate declare decode decompose decrement decrypt deduplicate def defa defau defaul default defaults deferred defi defin define degrees delayed delegate delete delete_all delimited demand dense_rank depth dequeue des_decrypt des_encrypt des_key_file desc descr descri describ describe descriptor deterministic diagnostics difference dimension direct_load directory disable disable_all disallow disassociate discardfile disconnect diskgroup distinct distinctrow distribute distributed div do document domain dotnet double downgrade drop dumpfile duplicate duration each edition editionable editions element ellipsis else elsif elt empty enable enable_all enclosed encode encoding encrypt end end-exec endian enforced engine engines enqueue enterprise entityescaping eomonth error errors escaped evalname evaluate event eventdata events except exception exceptions exchange exclude excluding execu execut execute exempt exists exit exp expire explain export export_set extended extent external external_1 external_2 externally extract failed failed_login_attempts failover failure far fast feature_set feature_value fetch field fields file file_name_convert filesystem_like_logging final finish first first_value fixed flash_cache flashback floor flush following follows for forall force form forma format found found_rows freelist freelists freepools fresh from from_base64 from_days ftp full function general generated get get_format get_lock getdate getutcdate global global_name globally go goto grant grants greatest group group_concat group_id grouping grouping_id groups gtid_subtract guarantee guard handler hash hashkeys having hea head headi headin heading heap help hex hierarchy high high_priority hosts hour http id ident_current ident_incr ident_seed identified identity idle_time if ifnull ignore iif ilike ilm immediate import in include including increment index indexes indexing indextype indicator indices inet6_aton inet6_ntoa inet_aton inet_ntoa infile initial initialized initially initrans inmemory inner innodb input insert install instance instantiable instr interface interleaved intersect into invalidate invisible is is_free_lock is_ipv4 is_ipv4_compat is_not is_not_null is_used_lock isdate isnull isolation iterate java join json json_exists keep keep_duplicates key keys kill language large last last_day last_insert_id last_value lax lcase lead leading least leaves left len lenght length less level levels library like like2 like4 likec limit lines link list listagg little ln load load_file lob lobs local localtime localtimestamp locate locator lock locked log log10 log2 logfile logfiles logging logical logical_reads_per_call logoff logon logs long loop low low_priority lower lpad lrtrim ltrim main make_set makedate maketime managed management manual map mapping mask master master_pos_wait match matched materialized max maxextents maximize maxinstances maxlen maxlogfiles maxloghistory maxlogmembers maxsize maxtrans md5 measures median medium member memcompress memory merge microsecond mid migration min minextents minimum mining minus minute minvalue missing mod mode model modification modify module monitoring month months mount move movement multiset mutex name name_const names nan national native natural nav nchar nclob nested never new newline next nextval no no_write_to_binlog noarchivelog noaudit nobadfile nocheck nocompress nocopy nocycle nodelay nodiscardfile noentityescaping noguarantee nokeep nologfile nomapping nomaxvalue nominimize nominvalue nomonitoring none noneditionable nonschema noorder nopr nopro noprom nopromp noprompt norely noresetlogs noreverse normal norowdependencies noschemacheck noswitch not nothing notice notrim novalidate now nowait nth_value nullif nulls num numb numbe nvarchar nvarchar2 object ocicoll ocidate ocidatetime ociduration ociinterval ociloblocator ocinumber ociref ocirefcursor ocirowid ocistring ocitype oct octet_length of off offline offset oid oidindex old on online only opaque open operations operator optimal optimize option optionally or oracle oracle_date oradata ord ordaudio orddicom orddoc order ordimage ordinality ordvideo organization orlany orlvary out outer outfile outline output over overflow overriding package pad parallel parallel_enable parameters parent parse partial partition partitions pascal passing password password_grace_time password_lock_time password_reuse_max password_reuse_time password_verify_function patch path patindex pctincrease pctthreshold pctused pctversion percent percent_rank percentile_cont percentile_disc performance period period_add period_diff permanent physical pi pipe pipelined pivot pluggable plugin policy position post_transaction pow power pragma prebuilt precedes preceding precision prediction prediction_cost prediction_details prediction_probability prediction_set prepare present preserve prior priority private private_sga privileges procedural procedure procedure_analyze processlist profiles project prompt protection public publishingservername purge quarter query quick quiesce quota quotename radians raise rand range rank raw read reads readsize rebuild record records recover recovery recursive recycle redo reduced ref reference referenced references referencing refresh regexp_like register regr_avgx regr_avgy regr_count regr_intercept regr_r2 regr_slope regr_sxx regr_sxy reject rekey relational relative relaylog release release_lock relies_on relocate rely rem remainder rename repair repeat replace replicate replication required reset resetlogs resize resource respect restore restricted result result_cache resumable resume retention return returning returns reuse reverse revoke right rlike role roles rollback rolling rollup round row row_count rowdependencies rowid rownum rows rtrim rules safe salt sample save savepoint sb1 sb2 sb4 scan schema schemacheck scn scope scroll sdo_georaster sdo_topo_geometry search sec_to_time second section securefile security seed segment select self sequence sequential serializable server servererror session session_user sessions_per_user set sets settings sha sha1 sha2 share shared shared_pool short show shrink shutdown si_averagecolor si_colorhistogram si_featurelist si_positionalcolor si_stillimage si_texture siblings sid sign sin size size_t sizes skip slave sleep smalldatetimefromparts smallfile snapshot some soname sort soundex source space sparse spfile split sql sql_big_result sql_buffer_result sql_cache sql_calc_found_rows sql_small_result sql_variant_property sqlcode sqldata sqlerror sqlname sqlstate sqrt square standalone standby start starting startup statement static statistics stats_binomial_test stats_crosstab stats_ks_test stats_mode stats_mw_test stats_one_way_anova stats_t_test_ stats_t_test_indep stats_t_test_one stats_t_test_paired stats_wsr_test status std stddev stddev_pop stddev_samp stdev stop storage store stored str str_to_date straight_join strcmp strict string struct stuff style subdate subpartition subpartitions substitutable substr substring subtime subtring_index subtype success sum suspend switch switchoffset switchover sync synchronous synonym sys sys_xmlagg sysasm sysaux sysdate sysdatetimeoffset sysdba sysoper system system_user sysutcdatetime table tables tablespace tan tdo template temporary terminated tertiary_weights test than then thread through tier ties time time_format time_zone timediff timefromparts timeout timestamp timestampadd timestampdiff timezone_abbr timezone_minute timezone_region to to_base64 to_date to_days to_seconds todatetimeoffset trace tracking transaction transactional translate translation treat trigger trigger_nestlevel triggers trim truncate try_cast try_convert try_parse type ub1 ub2 ub4 ucase unarchived unbounded uncompress under undo unhex unicode uniform uninstall union unique unix_timestamp unknown unlimited unlock unpivot unrecoverable unsafe unsigned until untrusted unusable unused update updated upgrade upped upper upsert url urowid usable usage use use_stored_outlines user user_data user_resources users using utc_date utc_timestamp uuid uuid_short validate validate_password_strength validation valist value values var var_samp varcharc vari varia variab variabl variable variables variance varp varraw varrawc varray verify version versions view virtual visible void wait wallet warning warnings week weekday weekofyear wellformed when whene whenev wheneve whenever where while whitespace with within without work wrapped xdb xml xmlagg xmlattributes xmlcast xmlcolattval xmlelement xmlexists xmlforest xmlindex xmlnamespaces xmlpi xmlquery xmlroot xmlschema xmlserialize xmltable xmltype xor year year_to_month years yearweek",literal:"true false null",built_in:"array bigint binary bit blob boolean char character date dec decimal float int int8 integer interval number numeric real record serial serial8 smallint text varchar varying void"},c:[{cN:"string",b:"'",e:"'",c:[e.BE,{b:"''"}]},{cN:"string",b:'"',e:'"',c:[e.BE,{b:'""'}]},{cN:"string",b:"`",e:"`",c:[e.BE]},e.CNM,e.CBCM,t]},e.CBCM,t]}});hljs.registerLanguage("r",function(e){var r="([a-zA-Z]|\\.[a-zA-Z.])[a-zA-Z0-9._]*";return{c:[e.HCM,{b:r,l:r,k:{keyword:"function if in break next repeat else for return switch while try tryCatch stop warning require library attach detach source setMethod setGeneric setGroupGeneric setClass ...",literal:"NULL NA TRUE FALSE T F Inf NaN NA_integer_|10 NA_real_|10 NA_character_|10 NA_complex_|10"},r:0},{cN:"number",b:"0[xX][0-9a-fA-F]+[Li]?\\b",r:0},{cN:"number",b:"\\d+(?:[eE][+\\-]?\\d*)?L\\b",r:0},{cN:"number",b:"\\d+\\.(?!\\d)(?:i\\b)?",r:0},{cN:"number",b:"\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{b:"`",e:"`",r:0},{cN:"string",c:[e.BE],v:[{b:'"',e:'"'},{b:"'",e:"'"}]}]}});hljs.registerLanguage("perl",function(e){var t="getpwent getservent quotemeta msgrcv scalar kill dbmclose undef lc ma syswrite tr send umask sysopen shmwrite vec qx utime local oct semctl localtime readpipe do return format read sprintf dbmopen pop getpgrp not getpwnam rewinddir qqfileno qw endprotoent wait sethostent bless s|0 opendir continue each sleep endgrent shutdown dump chomp connect getsockname die socketpair close flock exists index shmgetsub for endpwent redo lstat msgctl setpgrp abs exit select print ref gethostbyaddr unshift fcntl syscall goto getnetbyaddr join gmtime symlink semget splice x|0 getpeername recv log setsockopt cos last reverse gethostbyname getgrnam study formline endhostent times chop length gethostent getnetent pack getprotoent getservbyname rand mkdir pos chmod y|0 substr endnetent printf next open msgsnd readdir use unlink getsockopt getpriority rindex wantarray hex system getservbyport endservent int chr untie rmdir prototype tell listen fork shmread ucfirst setprotoent else sysseek link getgrgid shmctl waitpid unpack getnetbyname reset chdir grep split require caller lcfirst until warn while values shift telldir getpwuid my getprotobynumber delete and sort uc defined srand accept package seekdir getprotobyname semop our rename seek if q|0 chroot sysread setpwent no crypt getc chown sqrt write setnetent setpriority foreach tie sin msgget map stat getlogin unless elsif truncate exec keys glob tied closedirioctl socket readlink eval xor readline binmode setservent eof ord bind alarm pipe atan2 getgrent exp time push setgrent gt lt or ne m|0 break given say state when",r={cN:"subst",b:"[$@]\\{",e:"\\}",k:t},s={b:"->{",e:"}"},n={v:[{b:/\$\d/},{b:/[\$%@](\^\w\b|#\w+(::\w+)*|{\w+}|\w+(::\w*)*)/},{b:/[\$%@][^\s\w{]/,r:0}]},i=[e.BE,r,n],o=[n,e.HCM,e.C("^\\=\\w","\\=cut",{eW:!0}),s,{cN:"string",c:i,v:[{b:"q[qwxr]?\\s*\\(",e:"\\)",r:5},{b:"q[qwxr]?\\s*\\[",e:"\\]",r:5},{b:"q[qwxr]?\\s*\\{",e:"\\}",r:5},{b:"q[qwxr]?\\s*\\|",e:"\\|",r:5},{b:"q[qwxr]?\\s*\\<",e:"\\>",r:5},{b:"qw\\s+q",e:"q",r:5},{b:"'",e:"'",c:[e.BE]},{b:'"',e:'"'},{b:"`",e:"`",c:[e.BE]},{b:"{\\w+}",c:[],r:0},{b:"-?\\w+\\s*\\=\\>",c:[],r:0}]},{cN:"number",b:"(\\b0[0-7_]+)|(\\b0x[0-9a-fA-F_]+)|(\\b[1-9][0-9_]*(\\.[0-9_]+)?)|[0_]\\b",r:0},{b:"(\\/\\/|"+e.RSR+"|\\b(split|return|print|reverse|grep)\\b)\\s*",k:"split return print reverse grep",r:0,c:[e.HCM,{cN:"regexp",b:"(s|tr|y)/(\\\\.|[^/])*/(\\\\.|[^/])*/[a-z]*",r:10},{cN:"regexp",b:"(m|qr)?/",e:"/[a-z]*",c:[e.BE],r:0}]},{cN:"function",bK:"sub",e:"(\\s*\\(.*?\\))?[;{]",eE:!0,r:5,c:[e.TM]},{b:"-\\w\\b",r:0},{b:"^__DATA__$",e:"^__END__$",sL:"mojolicious",c:[{b:"^@@.*",e:"$",cN:"comment"}]}];return r.c=o,s.c=o,{aliases:["pl","pm"],l:/[\w\.]+/,k:t,c:o}});hljs.registerLanguage("ini",function(e){var b={cN:"string",c:[e.BE],v:[{b:"'''",e:"'''",r:10},{b:'"""',e:'"""',r:10},{b:'"',e:'"'},{b:"'",e:"'"}]};return{aliases:["toml"],cI:!0,i:/\S/,c:[e.C(";","$"),e.HCM,{cN:"section",b:/^\s*\[+/,e:/\]+/},{b:/^[a-z0-9\[\]_-]+\s*=\s*/,e:"$",rB:!0,c:[{cN:"attr",b:/[a-z0-9\[\]_-]+/},{b:/=/,eW:!0,r:0,c:[{cN:"literal",b:/\bon|off|true|false|yes|no\b/},{cN:"variable",v:[{b:/\$[\w\d"][\w\d_]*/},{b:/\$\{(.*?)}/}]},b,{cN:"number",b:/([\+\-]+)?[\d]+_[\d_]+/},e.NM]}]}]}});hljs.registerLanguage("diff",function(e){return{aliases:["patch"],c:[{cN:"meta",r:10,v:[{b:/^@@ +\-\d+,\d+ +\+\d+,\d+ +@@$/},{b:/^\*\*\* +\d+,\d+ +\*\*\*\*$/},{b:/^\-\-\- +\d+,\d+ +\-\-\-\-$/}]},{cN:"comment",v:[{b:/Index: /,e:/$/},{b:/={3,}/,e:/$/},{b:/^\-{3}/,e:/$/},{b:/^\*{3} /,e:/$/},{b:/^\+{3}/,e:/$/},{b:/\*{5}/,e:/\*{5}$/}]},{cN:"addition",b:"^\\+",e:"$"},{cN:"deletion",b:"^\\-",e:"$"},{cN:"addition",b:"^\\!",e:"$"}]}});hljs.registerLanguage("go",function(e){var t={keyword:"break default func interface select case map struct chan else goto package switch const fallthrough if range type continue for import return var go defer bool byte complex64 complex128 float32 float64 int8 int16 int32 int64 string uint8 uint16 uint32 uint64 int uint uintptr rune",literal:"true false iota nil",built_in:"append cap close complex copy imag len make new panic print println real recover delete"};return{aliases:["golang"],k:t,i:"</",c:[e.CLCM,e.CBCM,{cN:"string",v:[e.QSM,{b:"'",e:"[^\\\\]'"},{b:"`",e:"`"}]},{cN:"number",v:[{b:e.CNR+"[dflsi]",r:1},e.CNM]},{b:/:=/},{cN:"function",bK:"func",e:/\s*\{/,eE:!0,c:[e.TM,{cN:"params",b:/\(/,e:/\)/,k:t,i:/["']/}]}]}});hljs.registerLanguage("bash",function(e){var t={cN:"variable",v:[{b:/\$[\w\d#@][\w\d_]*/},{b:/\$\{(.*?)}/}]},s={cN:"string",b:/"/,e:/"/,c:[e.BE,t,{cN:"variable",b:/\$\(/,e:/\)/,c:[e.BE]}]},a={cN:"string",b:/'/,e:/'/};return{aliases:["sh","zsh"],l:/\b-?[a-z\._]+\b/,k:{keyword:"if then else elif fi for while in do done case esac function",literal:"true false",built_in:"break cd continue eval exec exit export getopts hash pwd readonly return shift test times trap umask unset alias bind builtin caller command declare echo enable help let local logout mapfile printf read readarray source type typeset ulimit unalias set shopt autoload bg bindkey bye cap chdir clone comparguments compcall compctl compdescribe compfiles compgroups compquote comptags comptry compvalues dirs disable disown echotc echoti emulate fc fg float functions getcap getln history integer jobs kill limit log noglob popd print pushd pushln rehash sched setcap setopt stat suspend ttyctl unfunction unhash unlimit unsetopt vared wait whence where which zcompile zformat zftp zle zmodload zparseopts zprof zpty zregexparse zsocket zstyle ztcp",_:"-ne -eq -lt -gt -f -d -e -s -l -a"},c:[{cN:"meta",b:/^#![^\n]+sh\s*$/,r:10},{cN:"function",b:/\w[\w\d_]*\s*\(\s*\)\s*\{/,rB:!0,c:[e.inherit(e.TM,{b:/\w[\w\d_]*/})],r:0},e.HCM,s,a,t]}});hljs.registerLanguage("python",function(e){var r={keyword:"and elif is global as in if from raise for except finally print import pass return exec else break not with class assert yield try while continue del or def lambda async await nonlocal|10 None True False",built_in:"Ellipsis NotImplemented"},b={cN:"meta",b:/^(>>>|\.\.\.) /},c={cN:"subst",b:/\{/,e:/\}/,k:r,i:/#/},a={cN:"string",c:[e.BE],v:[{b:/(u|b)?r?'''/,e:/'''/,c:[b],r:10},{b:/(u|b)?r?"""/,e:/"""/,c:[b],r:10},{b:/(fr|rf|f)'''/,e:/'''/,c:[b,c]},{b:/(fr|rf|f)"""/,e:/"""/,c:[b,c]},{b:/(u|r|ur)'/,e:/'/,r:10},{b:/(u|r|ur)"/,e:/"/,r:10},{b:/(b|br)'/,e:/'/},{b:/(b|br)"/,e:/"/},{b:/(fr|rf|f)'/,e:/'/,c:[c]},{b:/(fr|rf|f)"/,e:/"/,c:[c]},e.ASM,e.QSM]},s={cN:"number",r:0,v:[{b:e.BNR+"[lLjJ]?"},{b:"\\b(0o[0-7]+)[lLjJ]?"},{b:e.CNR+"[lLjJ]?"}]},i={cN:"params",b:/\(/,e:/\)/,c:["self",b,s,a]};return c.c=[a,s,b],{aliases:["py","gyp"],k:r,i:/(<\/|->|\?)|=>/,c:[b,s,a,e.HCM,{v:[{cN:"function",bK:"def"},{cN:"class",bK:"class"}],e:/:/,i:/[${=;\n,]/,c:[e.UTM,i,{b:/->/,eW:!0,k:"None"}]},{cN:"meta",b:/^[\t ]*@/,e:/$/},{b:/\b(print|exec)\(/}]}});hljs.registerLanguage("julia",function(e){var r={keyword:"in isa where baremodule begin break catch ccall const continue do else elseif end export false finally for function global if import importall let local macro module quote return true try using while type immutable abstract bitstype typealias ",literal:"true false ARGS C_NULL DevNull ENDIAN_BOM ENV I Inf Inf16 Inf32 Inf64 InsertionSort JULIA_HOME LOAD_PATH MergeSort NaN NaN16 NaN32 NaN64 PROGRAM_FILE QuickSort RoundDown RoundFromZero RoundNearest RoundNearestTiesAway RoundNearestTiesUp RoundToZero RoundUp STDERR STDIN STDOUT VERSION catalan e|0 eu|0 eulergamma golden im nothing pi γ π φ ",built_in:"ANY AbstractArray AbstractChannel AbstractFloat AbstractMatrix AbstractRNG AbstractSerializer AbstractSet AbstractSparseArray AbstractSparseMatrix AbstractSparseVector AbstractString AbstractUnitRange AbstractVecOrMat AbstractVector Any ArgumentError Array AssertionError Associative Base64DecodePipe Base64EncodePipe Bidiagonal BigFloat BigInt BitArray BitMatrix BitVector Bool BoundsError BufferStream CachingPool CapturedException CartesianIndex CartesianRange Cchar Cdouble Cfloat Channel Char Cint Cintmax_t Clong Clonglong ClusterManager Cmd CodeInfo Colon Complex Complex128 Complex32 Complex64 CompositeException Condition ConjArray ConjMatrix ConjVector Cptrdiff_t Cshort Csize_t Cssize_t Cstring Cuchar Cuint Cuintmax_t Culong Culonglong Cushort Cwchar_t Cwstring DataType Date DateFormat DateTime DenseArray DenseMatrix DenseVecOrMat DenseVector Diagonal Dict DimensionMismatch Dims DirectIndexString Display DivideError DomainError EOFError EachLine Enum Enumerate ErrorException Exception ExponentialBackOff Expr Factorization FileMonitor Float16 Float32 Float64 Function Future GlobalRef GotoNode HTML Hermitian IO IOBuffer IOContext IOStream IPAddr IPv4 IPv6 IndexCartesian IndexLinear IndexStyle InexactError InitError Int Int128 Int16 Int32 Int64 Int8 IntSet Integer InterruptException InvalidStateException Irrational KeyError LabelNode LinSpace LineNumberNode LoadError LowerTriangular MIME Matrix MersenneTwister Method MethodError MethodTable Module NTuple NewvarNode NullException Nullable Number ObjectIdDict OrdinalRange OutOfMemoryError OverflowError Pair ParseError PartialQuickSort PermutedDimsArray Pipe PollingFileWatcher ProcessExitedException Ptr QuoteNode RandomDevice Range RangeIndex Rational RawFD ReadOnlyMemoryError Real ReentrantLock Ref Regex RegexMatch RemoteChannel RemoteException RevString RoundingMode RowVector SSAValue SegmentationFault SerializationState Set SharedArray SharedMatrix SharedVector Signed SimpleVector Slot SlotNumber SparseMatrixCSC SparseVector StackFrame StackOverflowError StackTrace StepRange StepRangeLen StridedArray StridedMatrix StridedVecOrMat StridedVector String SubArray SubString SymTridiagonal Symbol Symmetric SystemError TCPSocket Task Text TextDisplay Timer Tridiagonal Tuple Type TypeError TypeMapEntry TypeMapLevel TypeName TypeVar TypedSlot UDPSocket UInt UInt128 UInt16 UInt32 UInt64 UInt8 UndefRefError UndefVarError UnicodeError UniformScaling Union UnionAll UnitRange Unsigned UpperTriangular Val Vararg VecElement VecOrMat Vector VersionNumber Void WeakKeyDict WeakRef WorkerConfig WorkerPool "},t="[A-Za-z_\\u00A1-\\uFFFF][A-Za-z_0-9\\u00A1-\\uFFFF]*",a={l:t,k:r,i:/<\//},n={cN:"number",b:/(\b0x[\d_]*(\.[\d_]*)?|0x\.\d[\d_]*)p[-+]?\d+|\b0[box][a-fA-F0-9][a-fA-F0-9_]*|(\b\d[\d_]*(\.[\d_]*)?|\.\d[\d_]*)([eEfF][-+]?\d+)?/,r:0},o={cN:"string",b:/'(.|\\[xXuU][a-zA-Z0-9]+)'/},i={cN:"subst",b:/\$\(/,e:/\)/,k:r},l={cN:"variable",b:"\\$"+t},c={cN:"string",c:[e.BE,i,l],v:[{b:/\w*"""/,e:/"""\w*/,r:10},{b:/\w*"/,e:/"\w*/}]},s={cN:"string",c:[e.BE,i,l],b:"`",e:"`"},d={cN:"meta",b:"@"+t},u={cN:"comment",v:[{b:"#=",e:"=#",r:10},{b:"#",e:"$"}]};return a.c=[n,o,c,s,d,u,e.HCM,{cN:"keyword",b:"\\b(((abstract|primitive)\\s+)type|(mutable\\s+)?struct)\\b"},{b:/<:/}],i.c=a.c,a});hljs.registerLanguage("coffeescript",function(e){var c={keyword:"in if for while finally new do return else break catch instanceof throw try this switch continue typeof delete debugger super yield import export from as default await then unless until loop of by when and or is isnt not",literal:"true false null undefined yes no on off",built_in:"npm require console print module global window document"},n="[A-Za-z$_][0-9A-Za-z$_]*",r={cN:"subst",b:/#\{/,e:/}/,k:c},i=[e.BNM,e.inherit(e.CNM,{starts:{e:"(\\s*/)?",r:0}}),{cN:"string",v:[{b:/'''/,e:/'''/,c:[e.BE]},{b:/'/,e:/'/,c:[e.BE]},{b:/"""/,e:/"""/,c:[e.BE,r]},{b:/"/,e:/"/,c:[e.BE,r]}]},{cN:"regexp",v:[{b:"///",e:"///",c:[r,e.HCM]},{b:"//[gim]*",r:0},{b:/\/(?![ *])(\\\/|.)*?\/[gim]*(?=\W|$)/}]},{b:"@"+n},{sL:"javascript",eB:!0,eE:!0,v:[{b:"```",e:"```"},{b:"`",e:"`"}]}];r.c=i;var s=e.inherit(e.TM,{b:n}),t="(\\(.*\\))?\\s*\\B[-=]>",o={cN:"params",b:"\\([^\\(]",rB:!0,c:[{b:/\(/,e:/\)/,k:c,c:["self"].concat(i)}]};return{aliases:["coffee","cson","iced"],k:c,i:/\/\*/,c:i.concat([e.C("###","###"),e.HCM,{cN:"function",b:"^\\s*"+n+"\\s*=\\s*"+t,e:"[-=]>",rB:!0,c:[s,o]},{b:/[:\(,=]\s*/,r:0,c:[{cN:"function",b:t,e:"[-=]>",rB:!0,c:[o]}]},{cN:"class",bK:"class",e:"$",i:/[:="\[\]]/,c:[{bK:"extends",eW:!0,i:/[:="\[\]]/,c:[s]},s]},{b:n+":",e:":",rB:!0,rE:!0,r:0}])}});hljs.registerLanguage("cpp",function(t){var e={cN:"keyword",b:"\\b[a-z\\d_]*_t\\b"},r={cN:"string",v:[{b:'(u8?|U)?L?"',e:'"',i:"\\n",c:[t.BE]},{b:'(u8?|U)?R"',e:'"',c:[t.BE]},{b:"'\\\\?.",e:"'",i:"."}]},s={cN:"number",v:[{b:"\\b(0b[01']+)"},{b:"(-?)\\b([\\d']+(\\.[\\d']*)?|\\.[\\d']+)(u|U|l|L|ul|UL|f|F|b|B)"},{b:"(-?)(\\b0[xX][a-fA-F0-9']+|(\\b[\\d']+(\\.[\\d']*)?|\\.[\\d']+)([eE][-+]?[\\d']+)?)"}],r:0},i={cN:"meta",b:/#\s*[a-z]+\b/,e:/$/,k:{"meta-keyword":"if else elif endif define undef warning error line pragma ifdef ifndef include"},c:[{b:/\\\n/,r:0},t.inherit(r,{cN:"meta-string"}),{cN:"meta-string",b:/<[^\n>]*>/,e:/$/,i:"\\n"},t.CLCM,t.CBCM]},a=t.IR+"\\s*\\(",c={keyword:"int float while private char catch import module export virtual operator sizeof dynamic_cast|10 typedef const_cast|10 const for static_cast|10 union namespace unsigned long volatile static protected bool template mutable if public friend do goto auto void enum else break extern using asm case typeid short reinterpret_cast|10 default double register explicit signed typename try this switch continue inline delete alignof constexpr decltype noexcept static_assert thread_local restrict _Bool complex _Complex _Imaginary atomic_bool atomic_char atomic_schar atomic_uchar atomic_short atomic_ushort atomic_int atomic_uint atomic_long atomic_ulong atomic_llong atomic_ullong new throw return and or not",built_in:"std string cin cout cerr clog stdin stdout stderr stringstream istringstream ostringstream auto_ptr deque list queue stack vector map set bitset multiset multimap unordered_set unordered_map unordered_multiset unordered_multimap array shared_ptr abort abs acos asin atan2 atan calloc ceil cosh cos exit exp fabs floor fmod fprintf fputs free frexp fscanf isalnum isalpha iscntrl isdigit isgraph islower isprint ispunct isspace isupper isxdigit tolower toupper labs ldexp log10 log malloc realloc memchr memcmp memcpy memset modf pow printf putchar puts scanf sinh sin snprintf sprintf sqrt sscanf strcat strchr strcmp strcpy strcspn strlen strncat strncmp strncpy strpbrk strrchr strspn strstr tanh tan vfprintf vprintf vsprintf endl initializer_list unique_ptr",literal:"true false nullptr NULL"},n=[e,t.CLCM,t.CBCM,s,r];return{aliases:["c","cc","h","c++","h++","hpp"],k:c,i:"</",c:n.concat([i,{b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:c,c:["self",e]},{b:t.IR+"::",k:c},{v:[{b:/=/,e:/;/},{b:/\(/,e:/\)/},{bK:"new throw return else",e:/;/}],k:c,c:n.concat([{b:/\(/,e:/\)/,k:c,c:n.concat(["self"]),r:0}]),r:0},{cN:"function",b:"("+t.IR+"[\\*&\\s]+)+"+a,rB:!0,e:/[{;=]/,eE:!0,k:c,i:/[^\w\s\*&]/,c:[{b:a,rB:!0,c:[t.TM],r:0},{cN:"params",b:/\(/,e:/\)/,k:c,r:0,c:[t.CLCM,t.CBCM,r,s,e]},t.CLCM,t.CBCM,i]},{cN:"class",bK:"class struct",e:/[{;:]/,c:[{b:/</,e:/>/,c:["self"]},t.TM]}]),exports:{preprocessor:i,strings:r,k:c}}});hljs.registerLanguage("ruby",function(e){var b="[a-zA-Z_]\\w*[!?=]?|[-+~]\\@|<<|>>|=~|===?|<=>|[<>]=?|\\*\\*|[-/+%^&*~`|]|\\[\\]=?",r={keyword:"and then defined module in return redo if BEGIN retry end for self when next until do begin unless END rescue else break undef not super class case require yield alias while ensure elsif or include attr_reader attr_writer attr_accessor",literal:"true false nil"},c={cN:"doctag",b:"@[A-Za-z]+"},a={b:"#<",e:">"},s=[e.C("#","$",{c:[c]}),e.C("^\\=begin","^\\=end",{c:[c],r:10}),e.C("^__END__","\\n$")],n={cN:"subst",b:"#\\{",e:"}",k:r},t={cN:"string",c:[e.BE,n],v:[{b:/'/,e:/'/},{b:/"/,e:/"/},{b:/`/,e:/`/},{b:"%[qQwWx]?\\(",e:"\\)"},{b:"%[qQwWx]?\\[",e:"\\]"},{b:"%[qQwWx]?{",e:"}"},{b:"%[qQwWx]?<",e:">"},{b:"%[qQwWx]?/",e:"/"},{b:"%[qQwWx]?%",e:"%"},{b:"%[qQwWx]?-",e:"-"},{b:"%[qQwWx]?\\|",e:"\\|"},{b:/\B\?(\\\d{1,3}|\\x[A-Fa-f0-9]{1,2}|\\u[A-Fa-f0-9]{4}|\\?\S)\b/},{b:/<<(-?)\w+$/,e:/^\s*\w+$/}]},i={cN:"params",b:"\\(",e:"\\)",endsParent:!0,k:r},d=[t,a,{cN:"class",bK:"class module",e:"$|;",i:/=/,c:[e.inherit(e.TM,{b:"[A-Za-z_]\\w*(::\\w+)*(\\?|\\!)?"}),{b:"<\\s*",c:[{b:"("+e.IR+"::)?"+e.IR}]}].concat(s)},{cN:"function",bK:"def",e:"$|;",c:[e.inherit(e.TM,{b:b}),i].concat(s)},{b:e.IR+"::"},{cN:"symbol",b:e.UIR+"(\\!|\\?)?:",r:0},{cN:"symbol",b:":(?!\\s)",c:[t,{b:b}],r:0},{cN:"number",b:"(\\b0[0-7_]+)|(\\b0x[0-9a-fA-F_]+)|(\\b[1-9][0-9_]*(\\.[0-9_]+)?)|[0_]\\b",r:0},{b:"(\\$\\W)|((\\$|\\@\\@?)(\\w+))"},{cN:"params",b:/\|/,e:/\|/,k:r},{b:"("+e.RSR+"|unless)\\s*",k:"unless",c:[a,{cN:"regexp",c:[e.BE,n],i:/\n/,v:[{b:"/",e:"/[a-z]*"},{b:"%r{",e:"}[a-z]*"},{b:"%r\\(",e:"\\)[a-z]*"},{b:"%r!",e:"![a-z]*"},{b:"%r\\[",e:"\\][a-z]*"}]}].concat(s),r:0}].concat(s);n.c=d,i.c=d;var l="[>?]>",o="[\\w#]+\\(\\w+\\):\\d+:\\d+>",u="(\\w+-)?\\d+\\.\\d+\\.\\d(p\\d+)?[^>]+>",w=[{b:/^\s*=>/,starts:{e:"$",c:d}},{cN:"meta",b:"^("+l+"|"+o+"|"+u+")",starts:{e:"$",c:d}}];return{aliases:["rb","gemspec","podspec","thor","irb"],k:r,i:/\/\*/,c:s.concat(w).concat(d)}});hljs.registerLanguage("yaml",function(e){var b="true false yes no null",a="^[ \\-]*",r="[a-zA-Z_][\\w\\-]*",t={cN:"attr",v:[{b:a+r+":"},{b:a+'"'+r+'":'},{b:a+"'"+r+"':"}]},c={cN:"template-variable",v:[{b:"{{",e:"}}"},{b:"%{",e:"}"}]},l={cN:"string",r:0,v:[{b:/'/,e:/'/},{b:/"/,e:/"/},{b:/\S+/}],c:[e.BE,c]};return{cI:!0,aliases:["yml","YAML","yaml"],c:[t,{cN:"meta",b:"^---s*$",r:10},{cN:"string",b:"[\\|>] *$",rE:!0,c:l.c,e:t.v[0].b},{b:"<%[%=-]?",e:"[%-]?%>",sL:"ruby",eB:!0,eE:!0,r:0},{cN:"type",b:"!!"+e.UIR},{cN:"meta",b:"&"+e.UIR+"$"},{cN:"meta",b:"\\*"+e.UIR+"$"},{cN:"bullet",b:"^ *-",r:0},e.HCM,{bK:b,k:{literal:b}},e.CNM,l]}});hljs.registerLanguage("css",function(e){var c="[a-zA-Z-][a-zA-Z0-9_-]*",t={b:/[A-Z\_\.\-]+\s*:/,rB:!0,e:";",eW:!0,c:[{cN:"attribute",b:/\S/,e:":",eE:!0,starts:{eW:!0,eE:!0,c:[{b:/[\w-]+\(/,rB:!0,c:[{cN:"built_in",b:/[\w-]+/},{b:/\(/,e:/\)/,c:[e.ASM,e.QSM]}]},e.CSSNM,e.QSM,e.ASM,e.CBCM,{cN:"number",b:"#[0-9A-Fa-f]+"},{cN:"meta",b:"!important"}]}}]};return{cI:!0,i:/[=\/|'\$]/,c:[e.CBCM,{cN:"selector-id",b:/#[A-Za-z0-9_-]+/},{cN:"selector-class",b:/\.[A-Za-z0-9_-]+/},{cN:"selector-attr",b:/\[/,e:/\]/,i:"$"},{cN:"selector-pseudo",b:/:(:)?[a-zA-Z0-9\_\-\+\(\)"'.]+/},{b:"@(font-face|page)",l:"[a-z-]+",k:"font-face page"},{b:"@",e:"[{;]",i:/:/,c:[{cN:"keyword",b:/\w+/},{b:/\s/,eW:!0,eE:!0,r:0,c:[e.ASM,e.QSM,e.CSSNM]}]},{cN:"selector-tag",b:c,r:0},{b:"{",e:"}",i:/\S/,c:[e.CBCM,t]}]}});hljs.registerLanguage("fortran",function(e){var t={cN:"params",b:"\\(",e:"\\)"},n={literal:".False. .True.",keyword:"kind do while private call intrinsic where elsewhere type endtype endmodule endselect endinterface end enddo endif if forall endforall only contains default return stop then public subroutine|10 function program .and. .or. .not. .le. .eq. .ge. .gt. .lt. goto save else use module select case access blank direct exist file fmt form formatted iostat name named nextrec number opened rec recl sequential status unformatted unit continue format pause cycle exit c_null_char c_alert c_backspace c_form_feed flush wait decimal round iomsg synchronous nopass non_overridable pass protected volatile abstract extends import non_intrinsic value deferred generic final enumerator class associate bind enum c_int c_short c_long c_long_long c_signed_char c_size_t c_int8_t c_int16_t c_int32_t c_int64_t c_int_least8_t c_int_least16_t c_int_least32_t c_int_least64_t c_int_fast8_t c_int_fast16_t c_int_fast32_t c_int_fast64_t c_intmax_t C_intptr_t c_float c_double c_long_double c_float_complex c_double_complex c_long_double_complex c_bool c_char c_null_ptr c_null_funptr c_new_line c_carriage_return c_horizontal_tab c_vertical_tab iso_c_binding c_loc c_funloc c_associated  c_f_pointer c_ptr c_funptr iso_fortran_env character_storage_size error_unit file_storage_size input_unit iostat_end iostat_eor numeric_storage_size output_unit c_f_procpointer ieee_arithmetic ieee_support_underflow_control ieee_get_underflow_mode ieee_set_underflow_mode newunit contiguous recursive pad position action delim readwrite eor advance nml interface procedure namelist include sequence elemental pure integer real character complex logical dimension allocatable|10 parameter external implicit|10 none double precision assign intent optional pointer target in out common equivalence data",built_in:"alog alog10 amax0 amax1 amin0 amin1 amod cabs ccos cexp clog csin csqrt dabs dacos dasin datan datan2 dcos dcosh ddim dexp dint dlog dlog10 dmax1 dmin1 dmod dnint dsign dsin dsinh dsqrt dtan dtanh float iabs idim idint idnint ifix isign max0 max1 min0 min1 sngl algama cdabs cdcos cdexp cdlog cdsin cdsqrt cqabs cqcos cqexp cqlog cqsin cqsqrt dcmplx dconjg derf derfc dfloat dgamma dimag dlgama iqint qabs qacos qasin qatan qatan2 qcmplx qconjg qcos qcosh qdim qerf qerfc qexp qgamma qimag qlgama qlog qlog10 qmax1 qmin1 qmod qnint qsign qsin qsinh qsqrt qtan qtanh abs acos aimag aint anint asin atan atan2 char cmplx conjg cos cosh exp ichar index int log log10 max min nint sign sin sinh sqrt tan tanh print write dim lge lgt lle llt mod nullify allocate deallocate adjustl adjustr all allocated any associated bit_size btest ceiling count cshift date_and_time digits dot_product eoshift epsilon exponent floor fraction huge iand ibclr ibits ibset ieor ior ishft ishftc lbound len_trim matmul maxexponent maxloc maxval merge minexponent minloc minval modulo mvbits nearest pack present product radix random_number random_seed range repeat reshape rrspacing scale scan selected_int_kind selected_real_kind set_exponent shape size spacing spread sum system_clock tiny transpose trim ubound unpack verify achar iachar transfer dble entry dprod cpu_time command_argument_count get_command get_command_argument get_environment_variable is_iostat_end ieee_arithmetic ieee_support_underflow_control ieee_get_underflow_mode ieee_set_underflow_mode is_iostat_eor move_alloc new_line selected_char_kind same_type_as extends_type_ofacosh asinh atanh bessel_j0 bessel_j1 bessel_jn bessel_y0 bessel_y1 bessel_yn erf erfc erfc_scaled gamma log_gamma hypot norm2 atomic_define atomic_ref execute_command_line leadz trailz storage_size merge_bits bge bgt ble blt dshiftl dshiftr findloc iall iany iparity image_index lcobound ucobound maskl maskr num_images parity popcnt poppar shifta shiftl shiftr this_image"};return{cI:!0,aliases:["f90","f95"],k:n,i:/\/\*/,c:[e.inherit(e.ASM,{cN:"string",r:0}),e.inherit(e.QSM,{cN:"string",r:0}),{cN:"function",bK:"subroutine function program",i:"[${=\\n]",c:[e.UTM,t]},e.C("!","$",{r:0}),{cN:"number",b:"(?=\\b|\\+|\\-|\\.)(?=\\.\\d|\\d)(?:\\d+)?(?:\\.?\\d*)(?:[de][+-]?\\d+)?\\b\\.?",r:0}]}});hljs.registerLanguage("awk",function(e){var r={cN:"variable",v:[{b:/\$[\w\d#@][\w\d_]*/},{b:/\$\{(.*?)}/}]},b="BEGIN END if else while do for in break continue delete next nextfile function func exit|10",n={cN:"string",c:[e.BE],v:[{b:/(u|b)?r?'''/,e:/'''/,r:10},{b:/(u|b)?r?"""/,e:/"""/,r:10},{b:/(u|r|ur)'/,e:/'/,r:10},{b:/(u|r|ur)"/,e:/"/,r:10},{b:/(b|br)'/,e:/'/},{b:/(b|br)"/,e:/"/},e.ASM,e.QSM]};return{k:{keyword:b},c:[r,n,e.RM,e.HCM,e.NM]}});hljs.registerLanguage("makefile",function(e){var i={cN:"variable",v:[{b:"\\$\\("+e.UIR+"\\)",c:[e.BE]},{b:/\$[@%<?\^\+\*]/}]},r={cN:"string",b:/"/,e:/"/,c:[e.BE,i]},a={cN:"variable",b:/\$\([\w-]+\s/,e:/\)/,k:{built_in:"subst patsubst strip findstring filter filter-out sort word wordlist firstword lastword dir notdir suffix basename addsuffix addprefix join wildcard realpath abspath error warning shell origin flavor foreach if or and call eval file value"},c:[i]},n={b:"^"+e.UIR+"\\s*[:+?]?=",i:"\\n",rB:!0,c:[{b:"^"+e.UIR,e:"[:+?]?=",eE:!0}]},t={cN:"meta",b:/^\.PHONY:/,e:/$/,k:{"meta-keyword":".PHONY"},l:/[\.\w]+/},l={cN:"section",b:/^[^\s]+:/,e:/$/,c:[i]};return{aliases:["mk","mak"],k:"define endef undefine ifdef ifndef ifeq ifneq else endif include -include sinclude override export unexport private vpath",l:/[\w-]+/,c:[e.HCM,i,r,a,n,t,l]}});hljs.registerLanguage("java",function(e){var a="[À-ʸa-zA-Z_$][À-ʸa-zA-Z_$0-9]*",t=a+"(<"+a+"(\\s*,\\s*"+a+")*>)?",r="false synchronized int abstract float private char boolean static null if const for true while long strictfp finally protected import native final void enum else break transient catch instanceof byte super volatile case assert short package default double public try this switch continue throws protected public private module requires exports do",s="\\b(0[bB]([01]+[01_]+[01]+|[01]+)|0[xX]([a-fA-F0-9]+[a-fA-F0-9_]+[a-fA-F0-9]+|[a-fA-F0-9]+)|(([\\d]+[\\d_]+[\\d]+|[\\d]+)(\\.([\\d]+[\\d_]+[\\d]+|[\\d]+))?|\\.([\\d]+[\\d_]+[\\d]+|[\\d]+))([eE][-+]?\\d+)?)[lLfF]?",c={cN:"number",b:s,r:0};return{aliases:["jsp"],k:r,i:/<\/|#/,c:[e.C("/\\*\\*","\\*/",{r:0,c:[{b:/\w+@/,r:0},{cN:"doctag",b:"@[A-Za-z]+"}]}),e.CLCM,e.CBCM,e.ASM,e.QSM,{cN:"class",bK:"class interface",e:/[{;=]/,eE:!0,k:"class interface",i:/[:"\[\]]/,c:[{bK:"extends implements"},e.UTM]},{bK:"new throw return else",r:0},{cN:"function",b:"("+t+"\\s+)+"+e.UIR+"\\s*\\(",rB:!0,e:/[{;=]/,eE:!0,k:r,c:[{b:e.UIR+"\\s*\\(",rB:!0,r:0,c:[e.UTM]},{cN:"params",b:/\(/,e:/\)/,k:r,r:0,c:[e.ASM,e.QSM,e.CNM,e.CBCM]},e.CLCM,e.CBCM]},c,{cN:"meta",b:"@[A-Za-z]+"}]}});hljs.registerLanguage("stan",function(e){return{c:[e.HCM,e.CLCM,e.CBCM,{b:e.UIR,l:e.UIR,k:{name:"for in while repeat until if then else",symbol:"bernoulli bernoulli_logit binomial binomial_logit beta_binomial hypergeometric categorical categorical_logit ordered_logistic neg_binomial neg_binomial_2 neg_binomial_2_log poisson poisson_log multinomial normal exp_mod_normal skew_normal student_t cauchy double_exponential logistic gumbel lognormal chi_square inv_chi_square scaled_inv_chi_square exponential inv_gamma weibull frechet rayleigh wiener pareto pareto_type_2 von_mises uniform multi_normal multi_normal_prec multi_normal_cholesky multi_gp multi_gp_cholesky multi_student_t gaussian_dlm_obs dirichlet lkj_corr lkj_corr_cholesky wishart inv_wishart","selector-tag":"int real vector simplex unit_vector ordered positive_ordered row_vector matrix cholesky_factor_corr cholesky_factor_cov corr_matrix cov_matrix",title:"functions model data parameters quantities transformed generated",literal:"true false"},r:0},{cN:"number",b:"0[xX][0-9a-fA-F]+[Li]?\\b",r:0},{cN:"number",b:"0[xX][0-9a-fA-F]+[Li]?\\b",r:0},{cN:"number",b:"\\d+(?:[eE][+\\-]?\\d*)?L\\b",r:0},{cN:"number",b:"\\d+\\.(?!\\d)(?:i\\b)?",r:0},{cN:"number",b:"\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",r:0}]}});hljs.registerLanguage("javascript",function(e){var r="[A-Za-z$_][0-9A-Za-z$_]*",t={keyword:"in of if for while finally var new function do return void else break catch instanceof with throw case default try this switch continue typeof delete let yield const export super debugger as async await static import from as",literal:"true false null undefined NaN Infinity",built_in:"eval isFinite isNaN parseFloat parseInt decodeURI decodeURIComponent encodeURI encodeURIComponent escape unescape Object Function Boolean Error EvalError InternalError RangeError ReferenceError StopIteration SyntaxError TypeError URIError Number Math Date String RegExp Array Float32Array Float64Array Int16Array Int32Array Int8Array Uint16Array Uint32Array Uint8Array Uint8ClampedArray ArrayBuffer DataView JSON Intl arguments require module console window document Symbol Set Map WeakSet WeakMap Proxy Reflect Promise"},a={cN:"number",v:[{b:"\\b(0[bB][01]+)"},{b:"\\b(0[oO][0-7]+)"},{b:e.CNR}],r:0},n={cN:"subst",b:"\\$\\{",e:"\\}",k:t,c:[]},c={cN:"string",b:"`",e:"`",c:[e.BE,n]};n.c=[e.ASM,e.QSM,c,a,e.RM];var s=n.c.concat([e.CBCM,e.CLCM]);return{aliases:["js","jsx"],k:t,c:[{cN:"meta",r:10,b:/^\s*['"]use (strict|asm)['"]/},{cN:"meta",b:/^#!/,e:/$/},e.ASM,e.QSM,c,e.CLCM,e.CBCM,a,{b:/[{,]\s*/,r:0,c:[{b:r+"\\s*:",rB:!0,r:0,c:[{cN:"attr",b:r,r:0}]}]},{b:"("+e.RSR+"|\\b(case|return|throw)\\b)\\s*",k:"return throw case",c:[e.CLCM,e.CBCM,e.RM,{cN:"function",b:"(\\(.*?\\)|"+r+")\\s*=>",rB:!0,e:"\\s*=>",c:[{cN:"params",v:[{b:r},{b:/\(\s*\)/},{b:/\(/,e:/\)/,eB:!0,eE:!0,k:t,c:s}]}]},{b:/</,e:/(\/\w+|\w+\/)>/,sL:"xml",c:[{b:/<\w+\s*\/>/,skip:!0},{b:/<\w+/,e:/(\/\w+|\w+\/)>/,skip:!0,c:[{b:/<\w+\s*\/>/,skip:!0},"self"]}]}],r:0},{cN:"function",bK:"function",e:/\{/,eE:!0,c:[e.inherit(e.TM,{b:r}),{cN:"params",b:/\(/,e:/\)/,eB:!0,eE:!0,c:s}],i:/\[|%/},{b:/\$[(.]/},e.METHOD_GUARD,{cN:"class",bK:"class",e:/[{;=]/,eE:!0,i:/[:"\[\]]/,c:[{bK:"extends"},e.UTM]},{bK:"constructor",e:/\{/,eE:!0}],i:/#(?!!)/}});hljs.registerLanguage("tex",function(c){var e={cN:"tag",b:/\\/,r:0,c:[{cN:"name",v:[{b:/[a-zA-Zа-яА-я]+[*]?/},{b:/[^a-zA-Zа-яА-я0-9]/}],starts:{eW:!0,r:0,c:[{cN:"string",v:[{b:/\[/,e:/\]/},{b:/\{/,e:/\}/}]},{b:/\s*=\s*/,eW:!0,r:0,c:[{cN:"number",b:/-?\d*\.?\d+(pt|pc|mm|cm|in|dd|cc|ex|em)?/}]}]}}]};return{c:[e,{cN:"formula",c:[e],r:0,v:[{b:/\$\$/,e:/\$\$/},{b:/\$/,e:/\$/}]},c.C("%","$",{r:0})]}});hljs.registerLanguage("xml",function(s){var e="[A-Za-z0-9\\._:-]+",t={eW:!0,i:/</,r:0,c:[{cN:"attr",b:e,r:0},{b:/=\s*/,r:0,c:[{cN:"string",endsParent:!0,v:[{b:/"/,e:/"/},{b:/'/,e:/'/},{b:/[^\s"'=<>`]+/}]}]}]};return{aliases:["html","xhtml","rss","atom","xjb","xsd","xsl","plist"],cI:!0,c:[{cN:"meta",b:"<!DOCTYPE",e:">",r:10,c:[{b:"\\[",e:"\\]"}]},s.C("<!--","-->",{r:10}),{b:"<\\!\\[CDATA\\[",e:"\\]\\]>",r:10},{b:/<\?(php)?/,e:/\?>/,sL:"php",c:[{b:"/\\*",e:"\\*/",skip:!0}]},{cN:"tag",b:"<style(?=\\s|>|$)",e:">",k:{name:"style"},c:[t],starts:{e:"</style>",rE:!0,sL:["css","xml"]}},{cN:"tag",b:"<script(?=\\s|>|$)",e:">",k:{name:"script"},c:[t],starts:{e:"</script>",rE:!0,sL:["actionscript","javascript","handlebars","xml"]}},{cN:"meta",v:[{b:/<\?xml/,e:/\?>/,r:10},{b:/<\?\w+/,e:/\?>/}]},{cN:"tag",b:"</?",e:"/?>",c:[{cN:"name",b:/[^\/><\s]+/,r:0},t]}]}});hljs.registerLanguage("markdown",function(e){return{aliases:["md","mkdown","mkd"],c:[{cN:"section",v:[{b:"^#{1,6}",e:"$"},{b:"^.+?\\n[=-]{2,}$"}]},{b:"<",e:">",sL:"xml",r:0},{cN:"bullet",b:"^([*+-]|(\\d+\\.))\\s+"},{cN:"strong",b:"[*_]{2}.+?[*_]{2}"},{cN:"emphasis",v:[{b:"\\*.+?\\*"},{b:"_.+?_",r:0}]},{cN:"quote",b:"^>\\s+",e:"$"},{cN:"code",v:[{b:"^```w*s*$",e:"^```s*$"},{b:"`.+?`"},{b:"^( {4}|	)",e:"$",r:0}]},{b:"^[-\\*]{3,}",e:"$"},{b:"\\[.+?\\][\\(\\[].*?[\\)\\]]",rB:!0,c:[{cN:"string",b:"\\[",e:"\\]",eB:!0,rE:!0,r:0},{cN:"link",b:"\\]\\(",e:"\\)",eB:!0,eE:!0},{cN:"symbol",b:"\\]\\[",e:"\\]",eB:!0,eE:!0}],r:10},{b:/^\[[^\n]+\]:/,rB:!0,c:[{cN:"symbol",b:/\[/,e:/\]/,eB:!0,eE:!0},{cN:"link",b:/:\s*/,e:/$/,eB:!0}]}]}});hljs.registerLanguage("json",function(e){var i={literal:"true false null"},n=[e.QSM,e.CNM],r={e:",",eW:!0,eE:!0,c:n,k:i},t={b:"{",e:"}",c:[{cN:"attr",b:/"/,e:/"/,c:[e.BE],i:"\\n"},e.inherit(r,{b:/:/})],i:"\\S"},c={b:"\\[",e:"\\]",c:[e.inherit(r)],i:"\\S"};return n.splice(n.length,0,t,c),{c:n,k:i,i:"\\S"}});"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type="text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div id="header">



<h1 class="title toc-ignore">Starter culture domestication main
script</h1>
<h4 class="author">Vincent Somerville</h4>
<h4 class="date">26/01/2024</h4>

</div>


<p>This is the code that corresponds to the manuscript:</p>
<p><em>Genomic and phenotypic imprints of Neolithic domestication on
Cheese Starter cultures</em></p>
<p>The scirpt is primarly produced by Vincent Somerville. For inquiries
feel free to reach out to me (<a href="mailto:vincent@somerville.earth" class="email">vincent@somerville.earth</a>).</p>
<p>The script is grouped into the following major sections:</p>
<ol style="list-style-type: decimal">
<li>Setup</li>
<li>Phenotypic analysis</li>
<li>Metagenome analysis</li>
<li>Genome analysis</li>
<li>Pseudogene analysis</li>
<li>Phylogenies</li>
<li>RNA-seq</li>
<li>Appendix</li>
</ol>
<div id="setup" class="section level1">
<h1>Setup</h1>
<div id="data-location" class="section level2">
<h2>Data location</h2>
<p>The data is located on Zenodo : xx. The raw sequencing data is
deposited on NCBI SRA and the genome and metagenome assemblies on the
corresponding NCBI see methods. Here you can set the locations according
to your needs:</p>
</div>
<div id="r-libaries" class="section level2">
<h2>R libaries</h2>
<pre class="r"><code>library(readr)
library(tsibble)
library(tidyverse)
library(patchwork)
library(plotly)
library(lme4)
library(lmerTest)
library(vcfR)
library(ape)
library(ggtree)
library(tidytree)
library(treeio)
library(TreeTools)
library(RColorBrewer)
library(ggnewscale)</code></pre>
</div>
<div id="set-locations" class="section level2">
<h2>Set locations</h2>
<pre class="r"><code>data_path &lt;- &quot;/home/vincent/Dropbox_Vincent/Projects/2024_domestication_submission/data/&quot;
plot_path &lt;- &quot;/home/vincent/Dropbox_Vincent/Projects/2024_domestication_submission/plot/&quot;</code></pre>
</div>
<div id="set-plotting-colors" class="section level2">
<h2>Set plotting colors</h2>
<pre class="r"><code>####------------------starter culture colors
colors &lt;- c(&quot;#d45500&quot;,&quot;#e05a2c&quot;,&quot;#e46d43&quot;,&quot;#e77f5b&quot;,&quot;#eb9172&quot;,&quot;#eea48a&quot;,&quot;#009cf0&quot;,&quot;#34d0fc&quot;,&quot;#80e5ff&quot;,&quot;#008181&quot;,&quot;#00dcdc&quot;)
####------------------species colors
all_colours_species &lt;- (c(&quot;#fec3b7ff&quot;,&quot;#0081a7ff&quot;,&quot;grey70&quot;,&quot;grey88&quot;) ) </code></pre>
</div>
</div>
<div id="phenotypic-analysis" class="section level1">
<h1>Phenotypic analysis</h1>
<pre class="r"><code>Säuregrad_RMK_final &lt;- read_delim(paste0(data_path,&quot;acidity_data.csv&quot;), 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    col_types = cols(RMK = col_character(), 
        `GMS [g/kg]` = col_double()), trim_ws = TRUE) %&gt;% filter(week&lt;54) %&gt;% filter(!KZ %in% &quot;Nur Listerien geprüft (wegen Krankheit)&quot;) %&gt;% add_column(group=&quot;RMK&quot;) 

Säuregrad_RMK_final$RMK  = factor(Säuregrad_RMK_final$RMK , levels=c(&quot;101&quot;,&quot;105&quot;,&quot;115&quot;,&quot;124&quot;,&quot;150&quot;,&quot;190&quot;,&quot;202&quot;,&quot;203&quot;,&quot;280&quot;,&quot;302&quot;,&quot;305&quot;) )
table(Säuregrad_RMK_final$RMK)

Säuregrad_RMK_final$KZ &lt;- as.numeric(Säuregrad_RMK_final$KZ)
Säuregrad_RMK_final$`BK38°C` &lt;- as.numeric(Säuregrad_RMK_final$`BK38°C`)
Säuregrad_RMK_final$`D(-)%` &lt;-  Säuregrad_RMK_final$`D(-)%` %&gt;% gsub(&quot;,&quot;,&quot;.&quot;,.) %&gt;% as.numeric()

Säuregrad_RMK_final$week
 Säuregrad_RMK_final$date &lt;-  MMWRweek::MMWRweek2Date(MMWRyear = Säuregrad_RMK_final$year,
                        MMWRweek = Säuregrad_RMK_final$week,
                        MMWRday = 3)

Säuregrad_RMK_final_prep &lt;- Säuregrad_RMK_final %&gt;% filter(year&gt;1996) %&gt;% filter(year&lt;2016) %&gt;%  filter(!is.na(`BK38°C`)) 

acidicationPlot &lt;- ggplot(Säuregrad_RMK_final_prep,aes(x=date,y=`BK38°C`,fill=RMK,color=RMK))+geom_point(alpha=0.05)+
  theme_classic()+scale_y_continuous(limits = c(90,150))+labs(y=&quot;acid-base titration value of starter culture&quot;,x=&quot;time [year]&quot;)+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9))+geom_hline(yintercept = 110)+scale_fill_manual(values = colors)+scale_color_manual(values = colors)+geom_smooth(se = TRUE)

acidicationPlot
 
 ##--------------------------------add area---------------------------------
 
Säuregrad_RMK_final_prep %&gt;%  filter(!is.na(`BK38°C`)) %&gt;% pull(`BK38°C`) %&gt;% max()
maxndata &lt;-  Säuregrad_RMK_final_prep %&gt;%  filter(!is.na(date)) %&gt;% pull(date) %&gt;% max() 
mindata &lt;- Säuregrad_RMK_final_prep %&gt;%  filter(!is.na(date)) %&gt;% pull(date) %&gt;% min() 
 

acidicationPlot &lt;- ggplot()+geom_rect(data=NULL,aes(xmin=mindata,xmax=maxndata,ymin=110,ymax=161),fill=&quot;green&quot;,alpha=0.1)+geom_point(data=Säuregrad_RMK_final_prep,aes(x=date,y=`BK38°C`,fill=RMK,color=RMK),alpha=0.05)+ theme_classic()+scale_y_continuous(limits = c(90,161),expand = c(0,0))+scale_x_continuous(expand = c(0,0))+labs(y=&quot;acid-base titration&quot;,x=&quot;time [year]&quot;)+theme(axis.text.x = element_blank(),axis.ticks.x=element_blank(),axis.title.x = element_blank(),legend.position = &quot;none&quot;)+geom_hline(yintercept = 110)+scale_fill_manual(values = colors)+scale_color_manual(values = colors)+geom_smooth(data=Säuregrad_RMK_final_prep,aes(x=date,y=`BK38°C`,fill=RMK,color=RMK),se = TRUE)
 
acidicationPlot

###=======================================
###-------------------------lactate plot
###=======================================

Säuregrad_RMK_final_subGSM &lt;- Säuregrad_RMK_final_prep

Säuregrad_RMK_final_subGSM_prep &lt;- Säuregrad_RMK_final_subGSM %&gt;% filter(year&gt;1996) %&gt;% filter(year&lt;2016) 

GMS_02 &lt;- ggplot()+geom_point(data=Säuregrad_RMK_final_subGSM,aes(x=date,y=`GMS [g/kg]`,fill=RMK,color=RMK),alpha=0.1)+theme_classic()+scale_y_continuous(limits = c(3,14), expand = c(0, 0))+scale_x_continuous(expand = c(0,0))+labs(y=&quot;total lactate\n[g/kg]&quot;,x=&quot;&quot;)+theme( axis.text.x = element_blank(),axis.ticks.x=element_blank(),legend.position = &quot;none&quot;,axis.title.x = element_blank())+scale_fill_manual(values = colors)+scale_color_manual(values = colors)+geom_smooth(data=Säuregrad_RMK_final_subGSM,aes(x=date,y=`GMS [g/kg]`,fill=RMK,color=RMK),se = TRUE)
 
GMS_02

###=======================================
###-------------------------d-l lactate plot
###=======================================

Säuregrad_RMK_final_prep_02 &lt;- Säuregrad_RMK_final_prep %&gt;% filter(!is.na(`D(-)%`))%&gt;% filter(`D(-)%`&gt;0.1)%&gt;% filter(`D(-)%`&lt;99)
 table(Säuregrad_RMK_final_prep_02$RMK)
 
Säuregrad_RMK_final_prep_02$date &lt;-  MMWRweek::MMWRweek2Date(MMWRyear = Säuregrad_RMK_final_prep_02$year,
                        MMWRweek = Säuregrad_RMK_final_prep_02$week,
                        MMWRday = 3)
  

d_l_02 &lt;- ggplot()+geom_point(data=Säuregrad_RMK_final_prep_02,aes(x=date,y=`D(-)%`,fill=RMK,color=RMK),alpha=0.1)+theme_classic()+scale_y_continuous(limits = c(0,65), expand = c(0, 0))+scale_x_date(expand = c(0,0))+labs(y=&quot;D-lactate\n[%]&quot;,x=&quot;Date [year]&quot;)+theme(legend.position = &quot;none&quot;,axis.text.x = element_text(angle = 75, hjust = 1,size=7))+scale_fill_manual(values = colors)+scale_color_manual(values = colors)+geom_smooth(data=Säuregrad_RMK_final_prep_02,aes(x=date,y=`D(-)%`,fill=RMK,color=RMK),se = TRUE)

d_l_02

###----------------------------------------------------
##all together
###----------------------------------------------------
 
acidicationPlot+GMS_02+d_l_02+ plot_layout(ncol=1)

pdf(paste0(plot_path,&quot;long_term_analysis_02.pdf&quot;),width=4.8,height=4.3)

acidicationPlot+GMS_02+d_l_02+ plot_layout(ncol=1)

dev.off()</code></pre>
<div id="do-stats" class="section level2">
<h2>Do stats</h2>
<pre class="r"><code>model &lt;- lmer(`D(-)%` ~ date + (1 | RMK), data = Säuregrad_RMK_final_prep_02)

# Check the summary of the model
summary(model)
summary(model)$coefficients

###-------------------lactat
# Fit the linear mixed effects model

model &lt;- lmer(`GMS [g/kg]` ~ date + (1 | RMK), data = Säuregrad_RMK_final_prep_02)

# Check the summary of the model
summary(model)
summary(model)$coefficients

multivariate_lm &lt;- lm(cbind(date, RMK) ~ `GMS [g/kg]`, data = Säuregrad_RMK_final_prep_02)
summary(multivariate_lm)

manova_result &lt;- manova(cbind(date, RMK) ~ `GMS [g/kg]`, data = Säuregrad_RMK_final_prep_02)
summary(manova_result)

time_model &lt;- lm(`GMS [g/kg]` ~ date, data = Säuregrad_RMK_final_prep_02)
summary(time_model)

time_model &lt;- lm(`GMS [g/kg]` ~ RMK, data = Säuregrad_RMK_final_prep_02)
summary(time_model)

multiple_model &lt;- lm(`GMS [g/kg]` ~ date + as.factor(RMK), data = Säuregrad_RMK_final_prep_02)

summary(multiple_model)</code></pre>
</div>
</div>
<div id="metagenome-analysis" class="section level1">
<h1>Metagenome analysis</h1>
<div id="data-preperation" class="section level2">
<h2>Data preperation</h2>
<pre class="bash"><code>
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/99_log

vim /data/Project/2020_StarterCultureDiversity/99_log/20200111_download.txt

vim /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change.txt

##--------------download special case

cd /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/

wget -i /data/Project/2020_StarterCultureDiversity/99_log/20200111_download_new.txt

http://gtf-owncloud.unil.ch/index.php/s/wWUzyjKJJOpe9y2/download?path=%2F&amp;files=V42_S41_L002_R1_001.fastq.gz
cp download?path=%2F&amp;files=V42_S41_L002_R1_001.fastq.gz Lyo_105_76_5_13-4-R1.fastq.gz
Lyo_105_76_5_13

##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/99_log

##--------------download special case

cd /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/
wget -i /data/Project/2020_StarterCultureDiversity/99_log/20200625_download.txt
</code></pre>
<p>create a file with the labels and the names of the file in each a
column</p>
<pre class="bash"><code>name=$(cut -d &#39;/&#39; -f 5 /data/Project/2020_StarterCultureDiversity/99_log/20200111_download.txt|head -1)
name=$(cut -d &#39;/&#39; -f 5 /data/Project/2020_StarterCultureDiversity/99_log/20200111_download.txt|head -2|tail -1)
#for name in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz |grep &quot;^V&quot;)
for name in $(cut -d &#39;/&#39; -f 5 /data/Project/2020_StarterCultureDiversity/99_log/20200111_download.txt)
do

  name_01=$(echo $name |cut -d &#39;_&#39; -f 1)
  Lane=$(echo $name |cut -d &#39;_&#39; -f 2|sed &#39;s/L//g&#39;)
  Read=$(echo $name |cut -d &#39;_&#39; -f 3)  
  
  Lane_new=$((Lane+2))

  newName=$(awk -F &quot;\t&quot; -v name=&quot;$name_01&quot; &#39;{if($2==name) {print $1}}&#39; /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change.txt)
  
  echo ${newName}&quot;-L&quot;${Lane_new}&quot;-&quot;${Read}&quot;.fastq.gz&quot;
    
  mv /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${name} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${newName}-${Lane_new}-${Read}.fastq.gz
    
done
##-------qualtiy control

ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/|cut -d &#39;-&#39; -f 1|sort|uniq -c |wc -l
ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/|cut -d &#39;-&#39; -f 1|sort|uniq -c 

##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================
for name in $(cut -d &#39;/&#39; -f 5 /data/Project/2020_StarterCultureDiversity/99_log/20200625_download.txt)
do

  name_01=$(echo $name |cut -d &#39;_&#39; -f 1)
  Lane=$(echo $name |cut -d &#39;_&#39; -f 2|sed &#39;s/L//g&#39;)
  Read=$(echo $name |cut -d &#39;_&#39; -f 3)  
  
  Lane_new=$((Lane+2))

  newName=$(awk -F &quot;\t&quot; -v name=&quot;$name_01&quot; &#39;{if($2==name) {print $1}}&#39; /data/Project/2020_StarterCultureDiversity/99_log/20200625_sampleNames2change.txt)
  
  echo ${newName}&quot;-L&quot;${Lane_new}&quot;-&quot;${Read}&quot;.fastq.gz&quot;
    
  mv /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/${name} /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/${newName}-${Lane_new}-${Read}.fastq.gz
    
done
##-------qualtiy control

ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/|cut -d &#39;-&#39; -f 1|sort|uniq -c |wc -l
ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/01_rawdata/gz/|cut -d &#39;-&#39; -f 1|sort|uniq -c </code></pre>
<p>concatenenat the different sequencing rounds into one fastq.gz
file</p>
<pre class="bash"><code>rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/02_concatenate/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/02_concatenate/gz

for name in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change.txt)
  do
  names=$(echo ${name}&quot;-&quot;)
   echo ${names}
  
  forward_01=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep &quot;$names&quot;  |grep &quot;\-L1\-R1.fastq.gz&quot;)
  forward_02=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep &quot;$names&quot;  |grep &quot;\-L2\-R1.fastq.gz&quot;)
  forward_03=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep &quot;$names&quot;  |grep &quot;\-3\-R1.fastq.gz&quot;)
  forward_04=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep &quot;$names&quot;  |grep &quot;\-4\-R1.fastq.gz&quot;)

  
  
  cat /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${forward_01} \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${forward_02}  \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${forward_03} \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${forward_04} &gt; \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/02_concatenate/gz/${name}-R1.fastq.gz
  
  reverse_01=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep &quot;$names&quot;  |grep &quot;\-L1\-R2.fastq.gz&quot;)
  reverse_02=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep &quot;$names&quot;  |grep &quot;\-L2\-R2.fastq.gz&quot;)
  reverse_03=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep &quot;$names&quot;  |grep &quot;\-3\-R2.fastq.gz&quot;)
  reverse_04=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz| grep &quot;$names&quot;  |grep &quot;\-4\-R2.fastq.gz&quot;)

  cat /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${reverse_01} \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${reverse_02}  \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${reverse_03}  \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/01_rawdata/gz/${reverse_04}  &gt; \
  /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/02_concatenate/gz/${name}-R2.fastq.gz
  
  done
</code></pre>
</div>
<div id="data-cleanup" class="section level2">
<h2>Data cleanup</h2>
<div id="adapter-removal" class="section level3">
<h3>Adapter removal</h3>
<p>In the following we clip adaptors with <a href="https://github.com/FelixKrueger/TrimGalore">trim-galore</a></p>
<pre class="bash"><code>threads=38
rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/{afterTrimgalore,logs}
num=1
for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change.txt |head -25)
  do
  echo ${num}&quot;/46  :&quot; ${names}
  num=$((num+1))
  
  trim_galore -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/03_trimGalore/gz/ \
        --fastqc_args &quot;--outdir /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/afterTrimgalore/&quot; \
        /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/02_concatenate/gz/${names}-R1.fastq.gz   \
        /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/02_concatenate/gz/${names}-R2.fastq.gz
  done 

rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/afterTrimgalore//Multiqc
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/afterTrimgalore//Multiqc
/home/vincent/.local/bin/multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/afterTrimgalore/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/99_QCs/afterTrimgalore//Multiqc
  

##------------------------------------
##===============================
##add 20200624
##------------------------------------
##===============================   
    threads=38
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/gz
      mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/gz
  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/99_QCs/{afterTrimgalore,logs}
num=1
for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20200625_sampleNames2change.txt )
  do
  echo ${num}&quot;/41  :&quot; ${names}
  num=$((num+1))
 
rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq
echo &quot;unzipping...&quot;
echo &quot;unzipped&quot;
echo &quot;removing cow reads&quot;

kneaddata --max-memory 110000m --no-discordant -t 37 -p 37 --input /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R1_val_1.fq \
  --input /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R2_val_2.fq \
  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  \
  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
  --output /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/
  
rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}

gzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/${names}*neaddata_paired_1.fastq &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz

gzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/${names}*neaddata_paired_2.fastq &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz

  rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/
        
  done 
  rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/99_QCs/afterTrimgalore//Multiqc
  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/99_QCs/afterTrimgalore//Multiqc
    /home/vincent/.local/bin/multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/99_QCs/afterTrimgalore/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/99_QCs/afterTrimgalore//Multiqc
    </code></pre>
</div>
<div id="kneaddata-cow-read-removal" class="section level3">
<h3>KneadData (cow read removal)</h3>
<p>Here, we filter out the cow genome contamination reads</p>
<pre class="bash"><code>
##===================
##remove cow reads
##===================

names=Vers_280_20
threads=38
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/
num=1
for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20200625_sampleNames2change.txt)
  do
  
rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/
echo &quot;unzipping...&quot;

gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/gz/${names}-R1_val_1.fq.gz &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R1_val_1.fq
gunzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/gz/${names}-R2_val_2.fq.gz &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R2_val_2.fq
echo &quot;unzipped&quot;
echo &quot;removing cow reads&quot;

kneaddata --max-memory 110000m --no-discordant -t 37 -p 37 --input /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R1_val_1.fq \
  --input /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/03_trimGalore/fq/${names}-R2_val_2.fq \
  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  \
  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
  --output /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/
  
rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}

gzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/${names}*neaddata_paired_1.fastq &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz

gzip -c /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/${names}/${names}*neaddata_paired_2.fastq &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20200625_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz

done
</code></pre>
</div>
</div>
<div id="downstream-analysis" class="section level2">
<h2>Downstream Analysis</h2>
<div id="motus" class="section level3">
<h3>mOTUs</h3>
<pre class="bash"><code>

for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt)
  do

  rm -r /data/Project/2020_StarterCultureDiversity/02_mOTU/${names}/
  mkdir -p /data/Project/2020_StarterCultureDiversity/02_mOTU/${names}/
 
/home/vincent/apps/mOTUs_v2/motus profile -f /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz \
  -r /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz -o /data/Project/2020_StarterCultureDiversity/02_mOTU/${names}/ -t 37 -n ${names}_mOTU

done

###-------------------------------------
##reduce data
###-------------------------------------

rm /data/Project/2020_StarterCultureDiversity/02_mOTU/all_mOTUs_prepforR.txt
for names in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt)
  do
#  echo &quot;-----------------------------------------&quot;
#  echo ${names}

grep &quot;^#&quot; -v /data/Project/2020_StarterCultureDiversity/02_mOTU/${names}/* |grep -v &quot;0.0000000000&quot; |grep -v &quot;sp.&quot; |awk -F &quot;\t&quot; -v sampless=&quot;${names}&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {print $1, $2, sampless}&#39; |sed &#39;s/\[.*\]//g&#39; |sed &#39;s/-1/NA NA/g&#39;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {print $1,$2,$3}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/02_mOTU/all_mOTUs_prepforR.txt

done
</code></pre>
<p>naming the samples correctly</p>
<pre class="r"><code>###----------------------------
##make a names file
###----------------------------

motus_abundance &lt;- read_delim(paste0(data_path,&quot;all_mOTUs_prepforR.txt&quot;), &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;species&quot;,&quot;abundance&quot;,&quot;sample&quot;),trim_ws = TRUE) 

unique_samples &lt;- motus_abundance$sample %&gt;% unique()
unique_samples_type &lt;- str_split_fixed(unique_samples, &quot;_&quot;, 5)[,1] #%&gt;%  mutate(Valence = ifelse(Participant == 1:41, rep(0:2), 0))
unique_samples_culture &lt;- str_split_fixed(unique_samples, &quot;_&quot;, 5)[,2]
unique_samples_date &lt;- str_split_fixed(unique_samples, &quot;_&quot;, 3)[,3]

unique_samples_dateNew &lt;- revalue(unique_samples_date,c(&quot;02_7&quot;=&quot;2002 Jul&quot;,    &quot;77&quot;=&quot;1977&quot; ,     &quot;76_3_31&quot;=&quot;1976 Mar 31&quot; ,&quot;76_4_7&quot;=&quot;1976 Apr 7&quot; ,&quot;75_4&quot; = &quot;1975 Apr&quot; ,   &quot;11_3_11&quot;=&quot;2011 Mar 11&quot;, &quot;76_4_6&quot;=&quot;1976 Apr 6&quot; , &quot;19&quot;=&quot;2019&quot;     ,   &quot;11_5&quot;= &quot;2011 Mar&quot;  ,   &quot;16&quot; =&quot;2016&quot;    , &quot;13&quot;= &quot;2013&quot; ,     &quot;17&quot;= &quot;2017&quot;   ,  &quot;10_2&quot;  = &quot;2010 Feb&quot; ,   &quot;76_5_12&quot;=&quot;1976 Mar 12&quot;  ,   &quot;75&quot;= &quot;1975&quot;    ,   &quot;96_1&quot;   =&quot;1996 Jan&quot; ,     &quot;08&quot;= &quot;2008&quot;  ,   &quot;14_10&quot; =  &quot;2014 Oct&quot;  ,  &quot;95_3&quot; =&quot;1995 Apr&quot;  , &quot;09&quot; = &quot;2009&quot;   ,&quot;76_5_11&quot; = &quot;1976 Mai 11&quot;,&quot;76_11&quot; = &quot;1976 Nov&quot;,&quot;76_2&quot;=&quot;1976 Feb&quot;  ,  &quot;76_4_2&quot; = &quot;1976 Apr 2&quot;, &quot;76_5_13&quot;=&quot;1976 Mai 13&quot;, &quot;16_4&quot;= &quot;2016 Apr&quot;,  &quot;76_4_1&quot;= &quot;1976 Apr 1&quot; ,  &quot;76&quot;=&quot;1976&quot;    ,  &quot;76_4&quot;= &quot;1976 Apr&quot; ,&quot;76_1&quot; = &quot;1976 Jan&quot;, &quot;18&quot;   =&quot;2018&quot;  ,&quot;14&quot; =&quot;2014&quot;,&quot;96&quot;=&quot;1996&quot;,&quot;18_01&quot;=&quot;2018 Jan&quot;,&quot;18_02&quot;=&quot;2018 Feb&quot;,&quot;12&quot;=&quot;2012&quot;,&quot;20&quot;=&quot;2020&quot;))

naming_DF &lt;- data.frame(namesOLD=unique_samples,sampleType=unique_samples_type,CultureName=unique_samples_culture,SampleDateOld=unique_samples_date,SampleDateNew=unique_samples_dateNew)

naming_DF$simpleNames &lt;- revalue(naming_DF$namesOLD,c(&quot;FSK_101_75&quot;=&quot;starter 1&quot;,&quot;Lyo_101_76_4_7&quot;=&quot;starter 2&quot;,&quot;Lyo_101_77&quot;=&quot;starter 3&quot;,&quot;Lyo_101_02_7&quot;=&quot;starter 4&quot;,&quot;Lyo_101_10_2&quot;=&quot;starter 5&quot;,&quot;Lyo_101_16&quot;=&quot;starter 6&quot;,&quot;ws_101_19&quot;=&quot;starter 7&quot;,&quot;FSK_105_75&quot;=&quot;starter 1&quot;,&quot;Lyo_105_76_4_1&quot;=&quot;starter 2&quot;,&quot;Lyo_105_76_4_2&quot;=&quot;starter 3&quot;,&quot;FSK_105_76_4_6&quot;=&quot;starter 4&quot;,&quot;Lyo_105_76_2&quot;=&quot;starter 5&quot;,&quot;Lyo_105_76_1&quot;=&quot;starter 6&quot;,&quot;Lyo_105_76_5_11&quot;=&quot;starter 7&quot;,&quot;Lyo_105_76_5_13&quot;=&quot;starter 8&quot;,&quot;Lyo_105_76_5_12&quot;=&quot;starter 9&quot;,&quot;Lyo_105_76_3_31&quot;=&quot;starter 10&quot;,&quot;Lyo_105_11_5&quot;=&quot;starter 11&quot;,&quot;Ws_105_19&quot;=&quot;starter 12&quot;,&quot;Lyo_115_09&quot;=&quot;starter 1&quot;,&quot;Ws_115_19&quot;=&quot;starter 2&quot;,&quot;FSK_124_75_4&quot;=&quot;starter 1&quot;,&quot;Lyo_124_08&quot;=&quot;starter 2&quot;, &quot;Ws_124_19&quot;=&quot;starter 3&quot;,&quot;Lyo_150_76_4&quot;=&quot;starter 1&quot;,&quot;Lyo_150_76_11&quot;=&quot;starter 2&quot;,&quot;Lyo_150_11_3_11&quot;=&quot;starter 3&quot;,&quot;Lyo_150_14_10&quot;=&quot;starter 4&quot;,&quot;Lyo_150_18&quot;=&quot;starter 5&quot;,&quot;Ws_150_19&quot;=&quot;starter 6&quot;,&quot;Lyo_190_95_3&quot;=&quot;starter 1&quot;,&quot;Lyo_190_17&quot;=&quot;starter 2&quot;,&quot;Ws_190_19&quot;=&quot;starter 3&quot;,&quot;Lyo_202_96&quot;=&quot;starter 1&quot;,&quot;Lyo_202_12&quot;=&quot;starter 2&quot;,&quot;Lyo_202_14&quot;=&quot;starter 3&quot;,&quot;S_202_14&quot;=&quot;starter 4&quot;,&quot;Ws_202_18&quot;=&quot;starter 5&quot;,&quot;S_202_18&quot;=&quot;starter 6&quot;, &quot;C_202_18_02&quot;=&quot;starter 7&quot;,&quot;C_202_18_01&quot;=&quot;starter 8&quot;,&quot;Lyo_203_77&quot;=&quot;starter 1&quot;,&quot;Lyo_203_96_1&quot;=&quot;starter 2&quot;,&quot;Lyo_203_16&quot;=&quot;starter 3&quot;,&quot;Ws_203_19&quot;=&quot;starter 4&quot;,&quot;Lyo_280_16&quot;=&quot;starter 1&quot;,&quot;Lyo_280_16_4&quot;=&quot;starter 2&quot;, &quot;Ws_280_19&quot;=&quot;starter 3&quot;,&quot;Lyo_302_13&quot;=&quot;starter 1&quot;, &quot;Ws_302_19&quot;=&quot;starter 2&quot;,&quot;FSK_305_76&quot;=&quot;starter 1&quot;, &quot;Lyo_305_96_1&quot;=&quot;starter 2&quot;,&quot;Lyo_305_17&quot;=&quot;starter 3&quot;,&quot;Ws_305_19&quot;=&quot;starter 4&quot;,
&quot;Vers_101_20&quot;=&quot;starter 8&quot;, &quot;Vers_105_20&quot;=&quot;starter 13&quot;, 
&quot;Vers_115_20&quot;=&quot;starter 3&quot;, &quot;Vers_124_20&quot;=&quot;starter 4&quot;, &quot;Vers_150_20&quot;=&quot;starter 7&quot;, &quot;Vers_190_20&quot;=&quot;starter 4&quot;, &quot;Vers_203_20&quot;=&quot;starter 5&quot;, 
&quot;Vers_280_20&quot;=&quot;starter 4&quot;, &quot;Vers_302_20&quot;=&quot;starter 3&quot;, &quot;Vers_305_20&quot;=&quot;starter 5&quot;,
&quot;Therm_101_20&quot;=&quot;starter 9&quot;, &quot;Therm_105_20&quot;=&quot;starter 14&quot;, &quot;Therm_115_20&quot;=&quot;starter 4&quot;, &quot;Therm_124_20&quot;=&quot;starter 5&quot;, 
&quot;Therm_150_20&quot;=&quot;starter 8&quot;, &quot;Therm_190_20&quot;=&quot;starter 5&quot;, &quot;Therm_203_20&quot;=&quot;starter 6&quot;, &quot;Therm_280_20&quot;=&quot;starter 5&quot;, 
&quot;Therm_302_20&quot;=&quot;starter 4&quot;, &quot;Therm_305_20&quot;=&quot;starter 6&quot;,
&quot;24h_101_20&quot;=&quot;starter 10&quot;, &quot;24h_105_20&quot;=&quot;starter 15&quot;, &quot;24h_115_20&quot;=&quot;starter 5&quot;, &quot;24h_124_20&quot;=&quot;starter 6&quot;, &quot;24h_150_20&quot;=&quot;starter 9&quot;, 
&quot;24h_190_20&quot;=&quot;starter 6&quot;, &quot;24h_203_20&quot;=&quot;starter 7&quot;, &quot;24h_280_20&quot;=&quot;starter 6&quot;, &quot;24h_302_20&quot;=&quot;starter 5&quot;, &quot;24h_305_20&quot;=&quot;starter 7&quot;,
&quot;M17x_101_20&quot;=&quot;starter 11&quot;, &quot;M17X_105_20&quot;=&quot;starter 16&quot;, &quot;M17X_115_20&quot;=&quot;starter 6&quot;, 
&quot;M17X_124_20&quot;=&quot;starter 7&quot;, &quot;M17x_150_20&quot;=&quot;starter 10&quot;, &quot;M17x_190_20&quot;=&quot;starter 7&quot;, &quot;M17x_202_20&quot;=&quot;starter 9&quot;, &quot;M17x_203_20&quot;=&quot;starter 8&quot;, 
&quot;M17x_280_20&quot;=&quot;starter 7&quot;, &quot;M17x_302_20&quot;=&quot;starter 6&quot;, &quot;M17x_305_20&quot;=&quot;starter 8&quot;))

naming_DF$sampleType &lt;- revalue(naming_DF$sampleType,c(&quot;ws&quot;=&quot;Ws&quot;))
naming_DF$namesNew &lt;- paste0(naming_DF$sampleType,&quot; RMK&quot;,naming_DF$CultureName,&quot;\n&quot;,naming_DF$SampleDateNew)

naming_DF &lt;- naming_DF[order(unique_samples_culture, unique_samples_dateNew),]
naming_DF$order &lt;- 1:nrow(naming_DF)

write.csv(naming_DF,paste0(data_path,&quot;naming_file_new.csv&quot;))</code></pre>
<pre class="r"><code>###----------------------------
##script
###----------------------------

naming_DF &lt;- read_csv(paste0(data_path,&quot;naming_file_new.csv&quot;))
motus_abundance &lt;- read_delim(paste0(data_path,&quot;all_mOTUs_prepforR.txt&quot;), &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;species&quot;,&quot;abundance&quot;,&quot;sample&quot;),trim_ws = TRUE)
unique(motus_abundance$simpleNames)

motus_abundance &lt;- merge(motus_abundance,naming_DF,by.y = &quot;namesOLD&quot;,by.x=&quot;sample&quot;,all.x = TRUE)
motus_abundance$simpleNames = factor(motus_abundance$simpleNames, levels=c(&quot;starter 1&quot;,  &quot;starter 2&quot;,  &quot;starter 3&quot; , &quot;starter 4&quot; , &quot;starter 5&quot;,  &quot;starter 6&quot;,  &quot;starter 7&quot; , &quot;starter 8&quot; , &quot;starter 9&quot; ,&quot;starter 10&quot; ,&quot;starter 11&quot;  ,&quot;starter 12&quot;, &quot;starter 13&quot; ,&quot;starter 14&quot; ,&quot;starter 15&quot;  ,&quot;starter 16&quot;))

motus_abundance$abundance &lt;- 100*motus_abundance$abundance

motus_abundance$species = factor(motus_abundance$species, levels=c(&quot;Lactobacillus delbrueckii&quot;,&quot;Streptococcus thermophilus&quot;,&quot;Lactobacillus helveticus/gallinarum&quot;,&quot;Lactococcus raffinolactis&quot;    , &quot;Thermus thermophilus/parvatiensis&quot;  ,&quot;Cutibacterium acnes&quot;,&quot;Lactococcus lactis&quot;,&quot;Rothia mucilaginosa&quot;,&quot;Burkholderia gladioli&quot;,&quot;Streptococcus agalactiae&quot;, &quot;Streptococcus parauberis&quot; ,  &quot;Lactobacillus fermentum/oris&quot;,&quot;Wohlfahrtiimonas chitiniclastica&quot;,&quot;Leuconostoc mesenteroides/suionicum&quot;,&quot;NA NA&quot;))

motus_abundance$species_new&lt;- revalue(motus_abundance$species,c(&quot;Lactobacillus helveticus/gallinarum&quot;=&quot;Lactobacillus helveticus&quot;,&quot;Lactococcus raffinolactis&quot;=&quot;subdominant&quot;    , &quot;Thermus thermophilus/parvatiensis&quot;=&quot;subdominant&quot;   ,&quot;Cutibacterium acnes&quot;=&quot;subdominant&quot; ,&quot;Lactococcus lactis&quot;=&quot;subdominant&quot; ,&quot;Rothia mucilaginosa&quot;=&quot;subdominant&quot; ,&quot;Burkholderia gladioli&quot;=&quot;subdominant&quot; ,&quot;Streptococcus agalactiae&quot;=&quot;subdominant&quot; , &quot;Streptococcus parauberis&quot;=&quot;subdominant&quot;  ,  &quot;Lactobacillus fermentum/oris&quot;=&quot;subdominant&quot; ,&quot;Wohlfahrtiimonas chitiniclastica&quot;=&quot;subdominant&quot; ,&quot;Leuconostoc mesenteroides/suionicum&quot;=&quot;subdominant&quot; ,&quot;NA&quot;=&quot;subdominant&quot;,&quot;NA NA&quot;=&quot;subdominant&quot;))

motus_abundance[is.na(motus_abundance$species_new),&quot;species_new&quot;] &lt;- &quot;subdominant&quot;

##--------------------
##ploting
##--------------------

all_colours_species &lt;- (c(&quot;#fec3b7ff&quot;,&quot;#0081a7ff&quot;,&quot;grey70&quot;,&quot;grey88&quot;) ) 

motus_abundance
pMOTUs &lt;-  ggplot( data = motus_abundance,aes(y = abundance, x = simpleNames, group=species_new,fill = species_new))+ geom_bar( stat=&quot;identity&quot;)+
  facet_grid(~CultureName,scales = &quot;free&quot;,space = &quot;free&quot;)+
    labs(&quot;&quot;,
         x=&quot;&quot;,
         y=&quot;relative Abundance&quot;)+
    theme_classic()+
  scale_fill_manual(values=all_colours_species)+
  scale_color_manual(values=all_colours_species)+
  scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous( expand = c(0, 0)) +
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=3),
          legend.position=&quot;right&quot;,
          legend.title = element_blank()
          )

pMOTUs
  
pdf(paste0(plot_path,&quot;AbundanceSpecies_all.pdf&quot;),width=4.8,height=4.3)

pMOTUs

dev.off()</code></pre>
</div>
<div id="snp-analysis" class="section level3">
<h3>SNP analysis</h3>
<div id="mapping-to-rmk202" class="section level4">
<h4>Mapping to RMK202</h4>
<pre class="bash"><code>#------add L.helveticus genome

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta \
  /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/Ldelv_mag.fasta &gt; \
  /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv.fasta
  
###================
##description file
###================

threads=37
samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_andNew.txt ##the file with all sample names
logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref
Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv.fasta

###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}
rm ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
num=1

for names in $(cut -f 1 ${samplesss})
  do
  echo ${num}&quot;/95  :&quot; ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
bwa mem -t ${threads} ${Assembly} /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R1_kneaddata.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20191218_metagenome/04_kneadData/gz/${names}/${names}-R2_kneaddata.fq.gz | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam &gt; ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &amp;

##-----Qualimap--------------------------------

mkdir -p ${BaseLocation}/${ynames}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}&quot;/&quot;${names}&quot;/bwaMapping2DB/bamqc&quot; &gt;&gt; ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 

tail -95 ${BaseLocation}/log/multiqc_logfile_finalGenome.txt &gt; ${BaseLocation}/log/multiqc_logfile_finalGenome_onlyMETA.txt
   
rm -r ${BaseLocation}/MultiQC_onlyMETA
 
mkdir -p ${BaseLocation}/MultiQC_onlyMETA
  
/home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome_onlyMETA.txt \
    -o ${BaseLocation}/MultiQC_onlyMETA
    
</code></pre>
</div>
<div id="snp-calling-freebayes-k1" class="section level4">
<h4>SNP calling freebayes (K1)</h4>
<p>This is a good (blog)[<a href="https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2017/Day2/vcf-intro.nb.html" class="uri">https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2017/Day2/vcf-intro.nb.html</a>]
on how to use freebayes and how to parse the output. Finally I will do
freebayes on all samples simultaniously.</p>
<p>Interesting output from column 11 on:</p>
<p>GT Genotype v :1 DP Read Depth :1040 DPR Number of observation for
each allele : 1,1037 RO Reference allele observation count :1 QR Sum of
quality of the reference observations :10 AO Alternate allele
observation count :1037 QA Sum of quality of the alternate observations
:36917 GL Genotype Likelihood, log10-scaled likelihoods of the data
given the called genotype for each possible genotype generated from the
reference and alternate alleles given the sample ploidy</p>
<p>Very nice <a href="http://www.ddocent.com/filtering/">blog</a>
illustrating good filtering practices.</p>
<p>filtering of samples: 1. take away all Lhelv SNVs.</p>
</div>
<div id="snpeff" class="section level4">
<h4>snpEff</h4>
<p>Here we do <a href="http://snpeff.sourceforge.net/SnpEff_manual.html#buildAddConfig">snpEff</a>.</p>
<pre class="bash"><code>

###================
##description file
###================
threads=37
FREEBAYES_out=freebayesOuput_all
sampleShort=PROKKA_out_both ##For st_thermophilus

 threads=37
  samplesss=/data/Project/2020_StarterCultureDiversity/99_log/20191228_sampleNames2change_with202_withStrains.txt ##the file with all sample names
  logFilelocation=/data/Project/2020_StarterCultureDiversity/99_log
  BaseLocation=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref
  Assembly=/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/lhelv/RMK202_MAG_assembly_withLhelv_unknownREF.fasta
  

#${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf

###The two genomes are from

#!!!Sterm
#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/FNA/S-PNGR-Z-K.fna \
#!!!Ldel
#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/FNA/L-KCTC-3034.fna \


##=======================================================================================================================
##-----------------------------------------------all together----------------------------------------------
##=======================================================================================================================
##==================
##Building Database
##==================
#/archiv/Projects/2018_Culturomics/genomes/Lactobacillus_delbrueckii_RMK202.fna
#/archiv/Projects/2018_Culturomics/genomes/Streptococcus_thermophilus_RMK202.fna

##-------------
##01_add genome name to
##-------------

vim /home/vincent/miniconda2/share/snpeff-4.3.1t-2/snpEff.config
#S_thermophilus_mag_rmk202.genome : S_thermophilus_mag_rmk202
#L_delbrueckii_mag_rmk202.genome : L_delbrueckii_mag_rmk202
#Streptococcus_phage_rmk202.genome : Streptococcus_phage_rmk202
#PGAP_meta_all.genome : PGAP_meta_all
PROKKA_out_both.genome : PROKKA_out_both
#PGAP_meta_all_StarterCultureDiversity.genome : PGAP_meta_all_StarterCultureDiversity
##-------------
#02_put genome fasta and gff into right directory:
##-------------

#sampleShort=S_thermophilus_mag_rmk202
#sampleShort=CP046134 ##For st_thermophilus

#/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta 

##-------------
#03_put genome fasta 
##-------------
rm -r /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}
mkdir -p /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}

cat ${Assembly} &gt; /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}/sequences.fa


##-------------
#04 genes gff 
##-------------

#grep &quot;^#&quot; -v /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.gff|grep -e &quot;${sampleShort}&quot; |cut -f 1|sort|uniq -c

#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.gff | grep -e &quot;^${sampleShort}&quot; |awk -F &quot;\t&quot; &#39;{OFS&quot;\t&quot;}{if($3==&quot;CDS&quot;) print $0}&#39; &gt; /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff


##-------------
#04 all genes gff 
##-------------

##already preped in RAST chunk
# cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/merged/all_genes_rmk202.gff |sed &#39;s/gene/CDS/g&#39; &gt; /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff

grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/PROKKA_GFF/S-PNGR-Z-K.gff &gt; /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff
#!!!Ldel
#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/FNA/L-KCTC-3034.fna \
grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/PROKKA_GFF/L-KCTC-3034.gff &gt;&gt; /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff



##-------------
##05_build database
##-------------

cd /home/vincent/miniconda2/share/snpeff-4.3.1t-2/
java -jar snpEff.jar build -gff3 -v ${sampleShort}

##==================
##running snpEff
##==================

##-------------
##06_preperation of SNV file
##-------------

#grep -e &quot;${sampleShort}&quot; ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf &gt; ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_all.vcf

##-------------
##07 SNveff calling
##-------------

cd /home/vincent/miniconda2/share/snpeff-4.3.1t-2/
#rm -r ${sampleShort} ${BaseLocation}/${FREEBAYES_out}/SnpEff/${sampleShort}/
mkdir -p ${sampleShort} ${BaseLocation}/${FREEBAYES_out}/SnpEff/${sampleShort}/


grep &quot;NZ_CP048750.1&quot; /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf &gt; /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv.vcf

#grep &quot;NZ_LR698986.1&quot; -v /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf &gt; /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv.vcf


grep &quot;^#&quot; -v /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf|cut -f 1  |sort|uniq -c
grep &quot;^#&quot; -v  /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv.vcf|cut -f 1  |sort|uniq -c

java -Xmx64g -jar snpEff.jar ${sampleShort} /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv.vcf &gt; /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect.vcf

grep &quot;^#&quot; -v  /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect.vcf|cut -f 1  |sort|uniq -c

 # grep &quot;^#&quot; -v /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf |cut -f 1 |sort|uniq -c


##-------------
##08_prep for R
##-------------

grep &quot;^#&quot; -v /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect.vcf  | \
  awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {print $1,$2,$3,$4,$5,$6,$7,$8}&#39; &gt; /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect_prepforR.vcf


cut -f 1 /data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect_prepforR.vcf|sort|uniq -c

##==================
##transfer local 
##==================

##-------------
###snpEff
##-------------
mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling
scp vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/03_mapping2RMK202Ref/freebayesOuput_all/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect_prepforR.vcf ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/
</code></pre>
<p>here we merge all annotation knowledge we have of the SNPs:</p>
<ol style="list-style-type: decimal">
<li>SNPeffect</li>
<li>eggnog annotation</li>
<li>repeat annotation</li>
<li>core genes</li>
<li>seq_length</li>
</ol>
<p>Thereafter we have a long list of information for every SNP.</p>
</div>
<div id="identify-repeat-genes" class="section level4">
<h4>Identify repeat genes</h4>
<p>We could potentially have a problem of alignment in genes highly
similar. Therefore I look if which genes are cluster together.</p>
<pre class="bash"><code>###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
BaseFILNAME=L_delbrueckii_RMK202

###================
##script
###================
##==================
##merge all fna files
##==================

rm -r ${BaseLocation}/Repeats/
mkdir -p ${BaseLocation}/Repeats/

cat ${BaseLocation_Ldel}/${BaseFILNAME}_genes.ffn \
  ${BaseLocation_Sterm}/${BaseFILNAME_sterm}_genes.ffn &gt; \
 ${BaseLocation}/Repeats/both_genes.ffn

cat ${BaseLocation_Ldel}/${BaseFILNAME}.current.gff \
  ${BaseLocation_Sterm}/${BaseFILNAME_sterm}.current.gff &gt; \
 ${BaseLocation}/Repeats/both_genes.ffn
 
##==================
##cd-hit
##==================

cd-hit-est -i ${BaseLocation}/Repeats/both_genes.ffn \
    -o ${BaseLocation}/Repeats/both__cdsClustering.txt -c 0.90 -M 30000 -T 37 -sc 1 -sf 1

  ###multifasta2fasta
  mkdir -p ${BaseLocation}/Repeats/splitClusters
  cd ${BaseLocation}/Repeats/splitClusters
  awk &#39;NR%1==0{ print &gt; &quot;Cluster_&quot;++i&quot;.txt&quot; }&#39; RS=&#39;&gt;Cl&#39; ${BaseLocation}/Repeats/both__cdsClustering.txt.clstr

  ##prepare log file
  #cluster_2.txt till cluster_24.txt
  
  ls ${BaseLocation}/Repeats/splitClusters |sort -V |head -84 |tail -83 &gt; ${BaseLocation}/Repeats/splitClusters/files_of_INterest.txt
     
   ##get gene names
   num=2
   rm ${BaseLocation}/Repeats/bigCluster/all_repeatingGenes.txt
   mkdir -p ${BaseLocation}/Repeats/bigCluster/
   i=Cluster_2.txt
   for i in $(cat ${BaseLocation}/Repeats/splitClusters/files_of_INterest.txt)
    do
   grep -v &quot;^uster&quot; ${BaseLocation}/Repeats/splitClusters/${i} |awk -F &quot;&gt;&quot; &#39;{print $2}&#39; |awk -F &quot;... at &quot; &#39;BEGIN{OFS=&quot;\t&quot;}{print $1,$2}&#39; |sed &#39;s/... \*//g&#39; |sed &#39;$ d&#39; &gt; \
    ${BaseLocation}/Repeats/bigCluster/names_${i}
    
   grep -v &quot;^uster&quot; ${BaseLocation}/Repeats/splitClusters/${i} |awk -F &quot;&gt;&quot; &#39;{print $2}&#39;  |awk -F &quot;... at &quot; -v count=&quot;$num&quot; &#39;BEGIN{OFS=&quot;\t&quot;}{print $1,$2,count}&#39; |sed &#39;s/... \*//g&#39; |sed &#39;$ d&#39; &gt;&gt; \
   ${BaseLocation}/Repeats/bigCluster/all_repeatingGenes.txt
      
    num=$((num+1))
    
    
    done
 

    ##get gff
    
  # genes=pgaptmp_001292
    mkdir -p  ${BaseLocation}/Repeats//bigCluster_genes/
    for i in $(cat ${BaseLocation}/Repeats/splitClusters/files_of_INterest.txt)
    do
    rm  ${BaseLocation}/Repeats//bigCluster_genes//genes_${i}
    for genes in $(cut -f 1 ${BaseLocation}/Repeats/bigCluster//names_${i})
      do
        
        grep &quot;${genes}.p&quot; ${BaseLocation}/Repeats/both_genes.ffn &gt;&gt; \
          ${BaseLocation}/Repeats//bigCluster_genes/genes_${i}
       
        
      done # $genes   
    done   # $i
  
  mkdir -p ~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes</code></pre>
</div>
<div id="core-genes-prep" class="section level4">
<h4>Core genes prep</h4>
<pre class="bash"><code>rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt

##=======================================================================================================================
##-----------------------------------------------Streptococcus thermophilus----------------------------------------------
##=======================================================================================================================
species=Sterm
date=$(ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ |grep &quot;Resul&quot;)
##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups//OG_names_${species}.txt
for OG in $(cat /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

grep &quot;$OG&quot; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.tsv |awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{print $1,&quot;S_thermophilus__RMK202_extern&quot;, $105}&#39;  &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups//OG_names_${species}.txt

done #OG
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling
cat /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups//OG_names_${species}.txt &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/coreGenes_rmk_202_extern_both.txt


##=======================================================================================================================
##-----------------------------------------------Lactobacillus delbrueckii----------------------------------------------
##=======================================================================================================================
species=Ldel
date=$(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder |grep &quot;Resul&quot;)


/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder/Results_Feb11/

##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder/${date}/Orthogroups//OG_names_${species}.txt
for OG in $(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder/${date}/Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

grep &quot;$OG&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder/${date}/Orthogroups/Orthogroups.tsv |awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{print $1,&quot;L_delbrueckii_extern&quot;, $14}&#39;  &gt;&gt;/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder/${date}/Orthogroups//OG_names_${species}.txt

done #OG

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/OrthoFinder/${date}/Orthogroups//OG_names_${species}.txt &gt;&gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/coreGenes_rmk_202_extern_both.txt</code></pre>
</div>
<div id="eggnog" class="section level4">
<h4>Eggnog</h4>
<p>I do the( eggnog annotation online )[<a href="http://eggnog-mapper.embl.de/" class="uri">http://eggnog-mapper.embl.de/</a>]</p>
</div>
<div id="core-genome-statisics" class="section level4">
<h4>Core genome statisics</h4>
<p>Here, I calculate the gene length, in order to later calculate the
nucleotide diversity.</p>
<pre class="bash"><code>
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/PROKKA_FAA/S-PNGR-Z-K.faa |awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{print $1,3*$3}&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/geneLength.txt
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/PROKKA_FAA/L-KCTC-3034.faa |awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{print $1,3*$3}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/geneLength.txt
</code></pre>
</div>
<div id="merge-all-infomration" class="section level4">
<h4>Merge all infomration</h4>
<pre class="r"><code>library(readr)
library(stringr)

##==================
##01-SNPeff
##==================

RMK202_both_snpEff &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_withoutLhelv_snpEffect_prepforR.vcf&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;X.CHROM&quot;,&quot;POS&quot;,&quot;unknown&quot;,&quot;REF&quot;,&quot;ALT&quot;,&quot;quality&quot;,&quot;uknown2&quot;,&quot;snpeff&quot;), trim_ws = TRUE)

RMK202_both_snpEff_02 &lt;- subset(RMK202_both_snpEff, select=-c(unknown,uknown2))
RMK202_both_snpEff_02$site &lt;- paste(RMK202_both_snpEff_02$X.CHROM,RMK202_both_snpEff_02$POS,RMK202_both_snpEff_02$REF,RMK202_both_snpEff_02$ALT,sep=&quot;_&quot;)

tmp_snpeff &lt;-  data.frame(str_split_fixed(RMK202_both_snpEff_02$snpeff, fixed(&quot;|&quot;), 136)[,2:8])
colnames(tmp_snpeff) &lt;- c(&quot;effect&quot;,&quot;significance&quot;,&quot;geneName&quot;,&quot;PGAP_name&quot;,&quot;type&quot;,&quot;typeName&quot;,&quot;typeGene&quot;)

RMK202_both_snpEff_02 &lt;- subset(RMK202_both_snpEff_02, select=-snpeff)
RMK202_both_SNPeff_final &lt;-  cbind(RMK202_both_snpEff_02,tmp_snpeff)

RMK202_both_SNPeff_final$finalName &lt;- RMK202_both_SNPeff_final$typeName

table(RMK202_both_SNPeff_final$X.CHROM)
##==================
##02 eggnog
##==================
RMK202_Ldel_eggnog &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/Ldel_eggnog.annotations&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;query_name&quot;,&quot;seed_eggNOG_ortholog&quot;,&quot;seed_ortholog_evalue&quot;,&quot;seed_ortholog_score&quot;,&quot;best_tax_level&quot;,&quot;Preferred_name&quot;,&quot;GOs&quot;,&quot;EC&quot;,&quot;KEGG_ko&quot;,&quot;KEGG_Pathway&quot;,&quot;KEGG_Module&quot;,&quot;KEGG_Reaction&quot;,&quot;KEGG_rclass&quot;,&quot;BRITE&quot;,&quot;KEGG_TC&quot;,&quot;CAZy&quot;,&quot;BiGG_Reaction&quot;,&quot;tax_scope&quot;,&quot;eggNOG OGs&quot;,&quot;bestOG&quot;,&quot;COG_Functional_Category&quot;,&quot;eggNOG free text description&quot;),skip=4)

RMK202_Sterm_eggnog &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/Sterm_eggnog.annotations&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;query_name&quot;,&quot;seed_eggNOG_ortholog&quot;,&quot;seed_ortholog_evalue&quot;,&quot;seed_ortholog_score&quot;,&quot;best_tax_level&quot;,&quot;Preferred_name&quot;,&quot;GOs&quot;,&quot;EC&quot;,&quot;KEGG_ko&quot;,&quot;KEGG_Pathway&quot;,&quot;KEGG_Module&quot;,&quot;KEGG_Reaction&quot;,&quot;KEGG_rclass&quot;,&quot;BRITE&quot;,&quot;KEGG_TC&quot;,&quot;CAZy&quot;,&quot;BiGG_Reaction&quot;,&quot;tax_scope&quot;,&quot;eggNOG OGs&quot;,&quot;bestOG&quot;,&quot;COG_Functional_Category&quot;,&quot;eggNOG free text description&quot;),skip=4)
RMK202_both_eggnog &lt;- rbind(RMK202_Sterm_eggnog,RMK202_Ldel_eggnog)

##==================
##03 repeats
##did not result in any additional info
##==================

# RMK202_both_repeats &lt;- read_delim(&quot;~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes/all_repeatingGenes.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;geneNAME&quot;,&quot;Repeat_identity&quot;,&quot;Repeat_cluster&quot;)) #%&gt;% slice( 1:(n()-3))
# nrow(RMK202_both_repeats)

##==================
##04_core genes
##==================
library(tidyverse)
RMK202_both_coreGenes &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/coreGenes_rmk_202_extern_both.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;orthogroup&quot;,&quot;species&quot;,&quot;geneName&quot;)) %&gt;% add_column(.,core=&quot;core&quot;)
table(RMK202_both_coreGenes$species)
table(RMK202_both_coreGenes$core, RMK202_both_coreGenes$species)

##==================
##gene_length
##==================

RMK202_both_length &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling/geneLength.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;gene&quot;,&quot;length&quot;)) 

##==================
##05_merge
##==================
nrow(RMK202_both_SNPeff_final)

RMK202_both_SNPeff_final &lt;- RMK202_both_SNPeff_final %&gt;% select(X.CHROM,quality, site, effect, significance, geneName)

RMK202_snpeff_eggnog &lt;- merge(RMK202_both_SNPeff_final, RMK202_both_eggnog, by.x = &quot;geneName&quot;, by.y = &quot;query_name&quot;,all.x = TRUE)
# RMK202_snpeff_eggnog_repeats &lt;- merge(RMK202_snpeff_eggnog, RMK202_both_repeats, by.x = &quot;geneName&quot;, by.y = &quot;geneNAME&quot;,all.x = TRUE)
# RMK202_snpeff_eggnog_repeats$geneName &lt;- as.character(RMK202_snpeff_eggnog_repeats$geneName)
RMK202_snpeff_eggnog_core &lt;- merge(RMK202_snpeff_eggnog, RMK202_both_coreGenes, by.x = &quot;geneName&quot;, by.y = &quot;geneName&quot;,all.x = TRUE)
RMK202_snpeff_eggnog_core_length &lt;- merge(RMK202_snpeff_eggnog_core, RMK202_both_length, by.x = &quot;geneName&quot;, by.y = &quot;gene&quot;,all.x = TRUE)
dim(RMK202_snpeff_eggnog_core_length)
nrow(RMK202_snpeff_eggnog_core_length)

table(RMK202_snpeff_eggnog_core_length$core)
table(RMK202_snpeff_eggnog_core_length$species)
table(RMK202_snpeff_eggnog_core_length$X.CHROM)

RMK202_snpeff_eggnog_core_length$species &lt;- plyr::revalue(RMK202_snpeff_eggnog_core_length$X.CHROM, c(&quot;NZ_CP048750.12&quot;=&quot;L. delbrueckii&quot;,&quot;NZ_CP048750.1&quot;=&quot;S. thermophilus&quot;))
table(RMK202_snpeff_eggnog_core_length$species)


write.csv(RMK202_snpeff_eggnog_core_length,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/SNVcalling//RMK202_snpeff_eggnog_length_core_extern.txt&quot;,row.names = FALSE)


# RMK202_snpeff_eggnog_repeats_core[grep(&quot;S_thermophilus&quot;,RMK202_snpeff_eggnog_repeats_core$finalName),]
# RMK202_snpeff_eggnog_repeats_core[grep(&quot;CP046131&quot;,RMK202_snpeff_eggnog_repeats_core$finalName),]
# 
# nrow(RMK202_snpeff_eggnog_repeats_core)
# 
# RMK202_snpeff_eggnog_repeats_core_ldel &lt;- RMK202_snpeff_eggnog_repeats_core %&gt;% filter(X.CHROM==&quot;CP046131&quot;)
# table(RMK202_snpeff_eggnog_repeats_core_ldel$core)</code></pre>
</div>
</div>
<div id="first-analysis" class="section level3">
<h3>first analysis</h3>
<p>Here, I start analyzing the SNP output in R</p>
<pre class="r"><code>##==================
##01_import data
##==================

###====================================================
###====================================================
##specifically for metagenomes
###====================================================
###====================================================

culturesss &lt;- &quot;FAMstrains&quot;
culturesss &lt;- 101
snps_long &lt;- NA
for (culturesss in c(&quot;101&quot;,&quot;105&quot;,&quot;115&quot;,&quot;124&quot;,&quot;150&quot;,&quot;190&quot;,&quot;202&quot;,&quot;203&quot;,&quot;280&quot;,&quot;302&quot;,&quot;305&quot;)) {
  

snps_freebayes &lt;- read_delim(paste0(data_path,&quot;/SNPdata//variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK&quot;,culturesss,&quot;_prepR.vcf&quot;), &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE) %&gt;%  select(-c(QUAL,FILTER,FORMAT))

##==================
##01_filter SNVs MQM larger than 30 
##INFO=&lt;ID=MQM,Number=A,Type=Float,Description=&quot;Mean mapping quality of observed alternate alleles&quot;&gt;
##==================
snps_freebayes$MQM &lt;- str_split_fixed(snps_freebayes$INFO, &quot;;&quot;, 130)[,16] %&gt;% gsub(&quot;MQM=&quot;,&quot;&quot;,.) 
snps_freebayes$MQM &lt;- str_split_fixed(snps_freebayes$MQM, &quot;,&quot;, 3)[,1] %&gt;%  as.numeric()
snps_freebayes &lt;- snps_freebayes[which(snps_freebayes$MQM&gt;30),] %&gt;% select(-MQM,-INFO)

##==================
##02_long list and frequency calc
##==================

snps_freebayes_long &lt;- gather(snps_freebayes, sample, Snps, colnames(snps_freebayes)[6]:colnames(snps_freebayes)[ncol(snps_freebayes)], factor_key=TRUE,na.rm = TRUE) 

###------------------------calculate allel frequency   

sample_relative_temp &lt;-snps_freebayes_long %&gt;% filter(Snps!=&quot;.:.:.:.:.:.:.:.&quot;)%&gt;% separate(&quot;Snps&quot;, into=c(&quot;GT&quot;,&quot;DP&quot;,&quot;AD&quot;,&quot;RO&quot;,&quot;QR&quot;,&quot;AO&quot;,&quot;QA&quot;,&quot;GL&quot;), sep = &quot;:&quot;) %&gt;% transform(.,allel_frequency=as.numeric(AO)/as.numeric(DP),QualityScore=as.numeric(QA)/as.numeric(DP),Sequencing_depth=as.numeric(DP))

##==================
##03_filter QA/DP &gt; 30
##INFO=&lt;ID=QA,Number=A,Type=Integer,Description=&quot;Alternate allele quality sum in phred&quot;&gt;
##INFO=&lt;ID=DP,Number=1,Type=Integer,Description=&quot;Total read depth at the locus&quot;
##==================
sample_relative_temp_filter &lt;- sample_relative_temp[which(!is.na(sample_relative_temp$QualityScore)),]
sample_relative_temp_withONT &lt;-sample_relative_temp_filter[which(sample_relative_temp_filter$QualityScore&gt;5),]
table(sample_relative_temp_withONT$sample)
sample_relative_temp_withONT$site &lt;- paste(sample_relative_temp_withONT$X.CHROM,sample_relative_temp_withONT$POS,sample_relative_temp_withONT$REF,sample_relative_temp_withONT$ALT,sep=&quot;_&quot;)
sample_relative_temp &lt;- sample_relative_temp_withONT

sample_relative_temp_withONT$site &lt;- paste(sample_relative_temp_withONT$X.CHROM,sample_relative_temp_withONT$POS,sample_relative_temp_withONT$REF,sample_relative_temp_withONT$ALT,sep=&quot;_&quot;)
sample_relative_safe_final_prep &lt;- sample_relative_temp_withONT %&gt;% select(site,X.CHROM,sample,allel_frequency)

sample_relative_safe_final_tsne &lt;- spread(sample_relative_safe_final_prep, sample, allel_frequency)

sample_relative_wide &lt;- sample_relative_safe_final_tsne[rowSums(sample_relative_safe_final_tsne[,3:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)&gt;0,]
sample_relative_wide$species &lt;- revalue(sample_relative_wide$X.CHROM, c(&quot;NZ_CP023139.1&quot;=&quot;L. delbrueckii&quot;,&quot;NZ_CP048750.1&quot;=&quot;S. thermophilus&quot;))
sample_relative_wide$culture &lt;- culturesss


##==================
##11_make long
##==================
colnames(sample_relative_wide)
sample_relative_long &lt;- gather(sample_relative_wide, sample, Snps, colnames(sample_relative_wide)[3]:colnames(sample_relative_wide)[ncol(sample_relative_wide)-2], factor_key=TRUE,na.rm = TRUE) 
# nrow(sample_relative_long)
table(sample_relative_long$X.CHROM)

snps_long &lt;- rbind(snps_long,sample_relative_long)
}

###====================================================
###====================================================
##specifically for strains
###====================================================
###====================================================


culturesss &lt;- &quot;FAMstrains&quot;
#snps_long &lt;- NA
for (culturesss in c(&quot;FAMstrains&quot;)) {

snps_freebayes &lt;- read_delim(paste0(data_path,&quot;/SNPdata//variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_RMK&quot;,culturesss,&quot;_prepR.vcf&quot;), &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE) %&gt;%  select(-c(QUAL,FILTER,FORMAT))

##==================
##01_filter SNVs MQM larger than 30 
##INFO=&lt;ID=MQM,Number=A,Type=Float,Description=&quot;Mean mapping quality of observed alternate alleles&quot;&gt;
##==================
snps_freebayes$MQM &lt;- str_split_fixed(snps_freebayes$INFO, &quot;;&quot;, 130)[,16] %&gt;% gsub(&quot;MQM=&quot;,&quot;&quot;,.) 
snps_freebayes$MQM &lt;- str_split_fixed(snps_freebayes$MQM, &quot;,&quot;, 3)[,1] %&gt;%  as.numeric()
snps_freebayes &lt;- snps_freebayes[which(snps_freebayes$MQM&gt;30),] %&gt;% select(-MQM,-INFO)

##==================
##02_long list and frequency calc
##==================

snps_freebayes_long &lt;- gather(snps_freebayes, sample, Snps, colnames(snps_freebayes)[6]:colnames(snps_freebayes)[ncol(snps_freebayes)], factor_key=TRUE,na.rm = TRUE) 

###------------------------calculate allel frequency   
###!!!!!!!filter allele frequency &gt;0.8
###!!!!!!!only genomic SNVs


sample_relative_temp &lt;-snps_freebayes_long %&gt;% filter(Snps!=&quot;.:.:.:.:.:.:.:.&quot;) %&gt;% separate(&quot;Snps&quot;, into=c(&quot;GT&quot;,&quot;DP&quot;,&quot;AD&quot;,&quot;RO&quot;,&quot;QR&quot;,&quot;AO&quot;,&quot;QA&quot;,&quot;GL&quot;), sep = &quot;:&quot;) %&gt;% transform(.,allel_frequency=as.numeric(AO)/as.numeric(DP),QualityScore=as.numeric(QA)/as.numeric(DP),Sequencing_depth=as.numeric(DP)) %&gt;% filter(allel_frequency&gt;0.8) %&gt;% filter(X.CHROM %in% c(&quot;NZ_CP023139.1&quot;,&quot;NZ_CP048750.1&quot;))

##==================
##03_filter QA/DP &gt; 30
##INFO=&lt;ID=QA,Number=A,Type=Integer,Description=&quot;Alternate allele quality sum in phred&quot;&gt;
##INFO=&lt;ID=DP,Number=1,Type=Integer,Description=&quot;Total read depth at the locus&quot;
##==================
sample_relative_temp_filter &lt;- sample_relative_temp[which(!is.na(sample_relative_temp$QualityScore)),]
#table(sample_relative_temp_withONT$sample)
sample_relative_temp_filter$site &lt;- paste(sample_relative_temp_filter$X.CHROM,sample_relative_temp_filter$POS,sample_relative_temp_filter$REF,sample_relative_temp_filter$ALT,sep=&quot;_&quot;)
sample_relative_temp &lt;- sample_relative_temp_filter

sample_relative_temp_withONT &lt;- sample_relative_temp
sample_relative_safe_final_prep &lt;- sample_relative_temp_withONT %&gt;% select(site,X.CHROM,sample,allel_frequency)

sample_relative_safe_final_tsne &lt;- spread(sample_relative_safe_final_prep, sample, allel_frequency)

sample_relative_wide &lt;- sample_relative_safe_final_tsne[rowSums(sample_relative_safe_final_tsne[,3:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)&gt;0,]
sample_relative_wide$species &lt;- revalue(sample_relative_wide$X.CHROM,  c(&quot;NZ_CP023139.1&quot;=&quot;L. delbrueckii&quot;,&quot;NZ_CP048750.1&quot;=&quot;S. thermophilus&quot;))
sample_relative_wide$culture &lt;- culturesss

##==================
##11_make long
##==================
colnames(sample_relative_wide)
sample_relative_long &lt;- gather(sample_relative_wide, sample, Snps, colnames(sample_relative_wide)[3]:colnames(sample_relative_wide)[ncol(sample_relative_wide)-2], factor_key=TRUE,na.rm = TRUE) 
# nrow(sample_relative_long)
table(sample_relative_long$X.CHROM)
snps_long &lt;- rbind(snps_long,sample_relative_long)
}

##==================
##remove samples
##==================

table(snps_long$sample) %&gt;% length()
# snps_long &lt;- snps_long %&gt;% filter(!sample %in% c(&quot;C_202_18_02&quot;,&quot;C_202_18_01&quot;))
table(snps_long$sample)%&gt;% length()

##==================
##11_make long
##==================
snps_long &lt;- snps_long[-1,]

table(snps_long$sample)
#snps_long$samplesNew &lt;- revalue(snps_long$sample,c(&quot;FSK_101_75&quot;=&quot;starter 1&quot;,&quot;Lyo_101_76_4_7&quot;=&quot;starter 2&quot;,&quot;Lyo_101_77&quot;=&quot;starter 3&quot;,&quot;Lyo_101_02_7&quot;=&quot;starter 4&quot;,&quot;Lyo_101_10_2&quot;=&quot;starter 5&quot;,&quot;Lyo_101_16&quot;=&quot;starter 6&quot;,&quot;ws_101_19&quot;=&quot;starter 7&quot;,&quot;FSK_105_75&quot;=&quot;starter 1&quot;,&quot;Lyo_105_76_4_1&quot;=&quot;starter 2&quot;,&quot;Lyo_105_76_4_2&quot;=&quot;starter 3&quot;,&quot;FSK_105_76_4_6&quot;=&quot;starter 4&quot;,&quot;Lyo_105_76_2&quot;=&quot;starter 5&quot;,&quot;Lyo_105_76_1&quot;=&quot;starter 6&quot;,&quot;Lyo_105_76_5_11&quot;=&quot;starter 7&quot;,&quot;Lyo_105_76_5_13&quot;=&quot;starter 8&quot;,&quot;Lyo_105_76_5_12&quot;=&quot;starter 9&quot;,&quot;Lyo_105_76_3_31&quot;=&quot;starter 10&quot;,&quot;Lyo_105_11_5&quot;=&quot;starter 11&quot;,&quot;Ws_105_19&quot;=&quot;starter 12&quot;,&quot;Lyo_115_09&quot;=&quot;starter 1&quot;,&quot;Ws_115_19&quot;=&quot;starter 2&quot;,&quot;FSK_124_75_4&quot;=&quot;starter 1&quot;,&quot;Lyo_124_08&quot;=&quot;starter 2&quot;, &quot;Ws_124_19&quot;=&quot;starter 3&quot;,&quot;Lyo_150_76_4&quot;=&quot;starter 1&quot;,&quot;Lyo_150_76_11&quot;=&quot;starter 2&quot;,&quot;Lyo_150_11_3_11&quot;=&quot;starter 3&quot;,&quot;Lyo_150_14_10&quot;=&quot;starter 4&quot;,&quot;Lyo_150_18&quot;=&quot;starter 5&quot;,&quot;Ws_150_19&quot;=&quot;starter 6&quot;,&quot;Lyo_190_95_3&quot;=&quot;starter 1&quot;,&quot;Lyo_190_17&quot;=&quot;starter 2&quot;,&quot;Ws_190_19&quot;=&quot;starter 3&quot;,&quot;Lyo_202_96&quot;=&quot;starter 1&quot;,&quot;Lyo_202_12&quot;=&quot;starter 2&quot;,&quot;Lyo_202_14&quot;=&quot;starter 3&quot;,&quot;S_202_14&quot;=&quot;starter 4&quot;,&quot;Ws_202_18&quot;=&quot;starter 5&quot;,&quot;S_202_18&quot;=&quot;starter 6&quot;, &quot;C_202_18_02&quot;=&quot;starter 7&quot;,&quot;C_202_18_01&quot;=&quot;starter 8&quot;,&quot;Lyo_203_77&quot;=&quot;starter 1&quot;,&quot;Lyo_203_96_1&quot;=&quot;starter 2&quot;,&quot;Lyo_203_16&quot;=&quot;starter 3&quot;,&quot;Ws_203_19&quot;=&quot;starter 4&quot;,&quot;Lyo_280_16&quot;=&quot;starter 1&quot;,&quot;Lyo_280_16_4&quot;=&quot;starter 2&quot;, &quot;Ws_280_19&quot;=&quot;starter 3&quot;,&quot;Lyo_302_13&quot;=&quot;starter 1&quot;, &quot;Ws_302_19&quot;=&quot;starter 2&quot;,&quot;FSK_305_76&quot;=&quot;starter 1&quot;, &quot;Lyo_305_96_1&quot;=&quot;starter 2&quot;,&quot;Lyo_305_17&quot;=&quot;starter 3&quot;,&quot;Ws_305_19&quot;=&quot;starter 4&quot;))


snps_long$samplesNew &lt;- as.character(revalue(snps_long$sample,c(&quot;FSK_101_75&quot;=&quot;starter 1&quot;,&quot;Lyo_101_76_4_7&quot;=&quot;starter 2&quot;,&quot;Lyo_101_77&quot;=&quot;starter 3&quot;,&quot;Lyo_101_02_7&quot;=&quot;starter 4&quot;,&quot;Lyo_101_10_2&quot;=&quot;starter 5&quot;,&quot;Lyo_101_16&quot;=&quot;starter 6&quot;,&quot;ws_101_19&quot;=&quot;starter 7&quot;,&quot;FSK_105_75&quot;=&quot;starter 1&quot;,&quot;Lyo_105_76_4_1&quot;=&quot;starter 2&quot;,&quot;Lyo_105_76_4_2&quot;=&quot;starter 3&quot;,&quot;FSK_105_76_4_6&quot;=&quot;starter 4&quot;,&quot;Lyo_105_76_2&quot;=&quot;starter 5&quot;,&quot;Lyo_105_76_1&quot;=&quot;starter 6&quot;,&quot;Lyo_105_76_5_11&quot;=&quot;starter 7&quot;,&quot;Lyo_105_76_5_13&quot;=&quot;starter 8&quot;,&quot;Lyo_105_76_5_12&quot;=&quot;starter 9&quot;,&quot;Lyo_105_76_3_31&quot;=&quot;starter 10&quot;,&quot;Lyo_105_11_5&quot;=&quot;starter 11&quot;,&quot;Ws_105_19&quot;=&quot;starter 12&quot;,&quot;Lyo_115_09&quot;=&quot;starter 1&quot;,&quot;Ws_115_19&quot;=&quot;starter 2&quot;,&quot;FSK_124_75_4&quot;=&quot;starter 1&quot;,&quot;Lyo_124_08&quot;=&quot;starter 2&quot;, &quot;Ws_124_19&quot;=&quot;starter 3&quot;,&quot;Lyo_150_76_4&quot;=&quot;starter 1&quot;,&quot;Lyo_150_76_11&quot;=&quot;starter 2&quot;,&quot;Lyo_150_11_3_11&quot;=&quot;starter 3&quot;,&quot;Lyo_150_14_10&quot;=&quot;starter 4&quot;,&quot;Lyo_150_18&quot;=&quot;starter 5&quot;,&quot;Ws_150_19&quot;=&quot;starter 6&quot;,&quot;Lyo_190_95_3&quot;=&quot;starter 1&quot;,&quot;Lyo_190_17&quot;=&quot;starter 2&quot;,&quot;Ws_190_19&quot;=&quot;starter 3&quot;,&quot;Lyo_202_96&quot;=&quot;starter 1&quot;,&quot;Lyo_202_12&quot;=&quot;starter 2&quot;,&quot;Lyo_202_14&quot;=&quot;starter 3&quot;,&quot;S_202_14&quot;=&quot;starter 4&quot;,&quot;Ws_202_18&quot;=&quot;starter 5&quot;,&quot;S_202_18&quot;=&quot;starter 6&quot;, &quot;C_202_18_02&quot;=&quot;starter 7&quot;,&quot;C_202_18_01&quot;=&quot;starter 8&quot;,&quot;Lyo_203_77&quot;=&quot;starter 1&quot;,&quot;Lyo_203_96_1&quot;=&quot;starter 2&quot;,&quot;Lyo_203_16&quot;=&quot;starter 3&quot;,&quot;Ws_203_19&quot;=&quot;starter 4&quot;,&quot;Lyo_280_16&quot;=&quot;starter 1&quot;,&quot;Lyo_280_16_4&quot;=&quot;starter 2&quot;, &quot;Ws_280_19&quot;=&quot;starter 3&quot;,&quot;Lyo_302_13&quot;=&quot;starter 1&quot;, &quot;Ws_302_19&quot;=&quot;starter 2&quot;,&quot;FSK_305_76&quot;=&quot;starter 1&quot;, &quot;Lyo_305_96_1&quot;=&quot;starter 2&quot;,&quot;Lyo_305_17&quot;=&quot;starter 3&quot;,&quot;Ws_305_19&quot;=&quot;starter 4&quot;,
&quot;Vers_101_20&quot;=&quot;starter 8&quot;, &quot;Vers_105_20&quot;=&quot;starter 13&quot;, 
&quot;Vers_115_20&quot;=&quot;starter 3&quot;, &quot;Vers_124_20&quot;=&quot;starter 4&quot;, &quot;Vers_150_20&quot;=&quot;starter 7&quot;, &quot;Vers_190_20&quot;=&quot;starter 4&quot;, &quot;Vers_203_20&quot;=&quot;starter 5&quot;, 
&quot;Vers_280_20&quot;=&quot;starter 4&quot;, &quot;Vers_302_20&quot;=&quot;starter 3&quot;, &quot;Vers_305_20&quot;=&quot;starter 5&quot;,
&quot;Therm_101_20&quot;=&quot;starter 9&quot;, &quot;Therm_105_20&quot;=&quot;starter 14&quot;, &quot;Therm_115_20&quot;=&quot;starter 4&quot;, &quot;Therm_124_20&quot;=&quot;starter 5&quot;, 
&quot;Therm_150_20&quot;=&quot;starter 8&quot;, &quot;Therm_190_20&quot;=&quot;starter 5&quot;, &quot;Therm_203_20&quot;=&quot;starter 6&quot;, &quot;Therm_280_20&quot;=&quot;starter 5&quot;, 
&quot;Therm_302_20&quot;=&quot;starter 4&quot;, &quot;Therm_305_20&quot;=&quot;starter 6&quot;,
&quot;24h_101_20&quot;=&quot;starter 10&quot;, &quot;24h_105_20&quot;=&quot;starter 15&quot;, &quot;24h_115_20&quot;=&quot;starter 5&quot;, &quot;24h_124_20&quot;=&quot;starter 6&quot;, &quot;24h_150_20&quot;=&quot;starter 9&quot;, 
&quot;24h_190_20&quot;=&quot;starter 6&quot;, &quot;24h_203_20&quot;=&quot;starter 7&quot;, &quot;24h_280_20&quot;=&quot;starter 6&quot;, &quot;24h_302_20&quot;=&quot;starter 5&quot;, &quot;24h_305_20&quot;=&quot;starter 7&quot;,
&quot;M17x_101_20&quot;=&quot;starter 11&quot;, &quot;M17X_105_20&quot;=&quot;starter 16&quot;, &quot;M17X_115_20&quot;=&quot;starter 6&quot;, 
&quot;M17X_124_20&quot;=&quot;starter 7&quot;, &quot;M17x_150_20&quot;=&quot;starter 10&quot;, &quot;M17x_190_20&quot;=&quot;starter 7&quot;, &quot;M17x_202_20&quot;=&quot;starter 9&quot;, &quot;M17x_203_20&quot;=&quot;starter 8&quot;, 
&quot;M17x_280_20&quot;=&quot;starter 7&quot;, &quot;M17x_302_20&quot;=&quot;starter 6&quot;, &quot;M17x_305_20&quot;=&quot;starter 8&quot;)))

sum(grepl(&quot;FAM&quot;,unique(snps_long$samplesNew)))

table(snps_long$samplesNew)
unique(snps_long$samplesNew)
sum(is.na(snps_long$samplesNew))

dput(unique(snps_long$samplesNew))

snps_long$samplesNew = factor(snps_long$samplesNew, levels=c(&quot;starter 10&quot;, &quot;starter 5&quot;, &quot;starter 9&quot;, &quot;starter 6&quot;, &quot;starter 4&quot;, 
&quot;starter 3&quot;, &quot;starter 2&quot;, &quot;starter 7&quot;, &quot;starter 1&quot;, &quot;starter 11&quot;, 
&quot;starter 8&quot;, &quot;starter 13&quot;, &quot;starter 12&quot;, &quot;starter 14&quot;, &quot;starter 15&quot;, 
&quot;starter 16&quot;,  &quot;FAM24786&quot;, &quot;FAM24790&quot;, &quot;FAM24759&quot;, &quot;FAM24746&quot;, 
&quot;FAM11071&quot;, &quot;FAM24758&quot;, &quot;FAM24757&quot;, &quot;FAM24755&quot;, &quot;FAM24776&quot;, &quot;FAM24741&quot;, 
&quot;FAM24749&quot;, &quot;FAM24775&quot;, &quot;FAM24754&quot;, &quot;FAM24744&quot;, &quot;FAM24743&quot;, &quot;FAM24740&quot;, 
&quot;FAM24738&quot;, &quot;FAM24760&quot;, &quot;FAM24729&quot;, &quot;FAM24764&quot;, &quot;FAM24731&quot;, &quot;FAM24730&quot;, 
&quot;FAM24727&quot;, &quot;FAM24726&quot;, &quot;FAM13496&quot;, &quot;FAM11141&quot;, &quot;FAM13497&quot;, &quot;FAM24724&quot;, 
&quot;FAM24753&quot;, &quot;FAM24799&quot;, &quot;FAM21747&quot;, &quot;FAM13495&quot;, &quot;FAM24739&quot;, &quot;FAM24734&quot;, 
&quot;FAM24752&quot;, &quot;FAM11024&quot;, &quot;FAM24736&quot;, &quot;FAM24728&quot;, &quot;FAM10980&quot;, &quot;FAM13498&quot;, 
&quot;FAM10973&quot;, &quot;FAM12107&quot;, &quot;FAM11009&quot;, &quot;FAM11008&quot;, &quot;FAM24745&quot;, &quot;FAM21769&quot;, 
&quot;FAM12273&quot;, &quot;FAM12105&quot;, &quot;FAM12104&quot;, &quot;FAM13491&quot;, &quot;FAM11049&quot;, &quot;FAM24787&quot;, 
&quot;FAM24756&quot;, &quot;FAM24742&quot;, &quot;FAM24774&quot;, &quot;FAM11142&quot;, &quot;FAM12080&quot;, &quot;FAM12109&quot;, 
&quot;FAM10996&quot;, &quot;FAM24732&quot;, &quot;FAM11063&quot;, &quot;FAM24784&quot;, &quot;FAM24791&quot;, &quot;FAM13500&quot;, 
&quot;FAM24725&quot;, &quot;FAM24853&quot;, &quot;FAM24770&quot;, &quot;FAM21781&quot;, &quot;FAM24750&quot;, &quot;FAM13494&quot;, 
&quot;FAM13493&quot;, &quot;FAM24766&quot;, &quot;FAM24767&quot;, &quot;FAM24782&quot;, &quot;FAM24768&quot;, &quot;FAM24769&quot;, 
&quot;FAM24772&quot;, &quot;FAM24777&quot;, &quot;FAM24748&quot;, &quot;FAM24761&quot;, &quot;FAM24778&quot;, &quot;FAM24779&quot;, 
&quot;FAM24780&quot;, &quot;FAM24747&quot;, &quot;FAM24773&quot;, &quot;FAM24794&quot;, &quot;FAM24781&quot;, &quot;FAM11021&quot;, 
&quot;FAM24783&quot;, &quot;FAM24785&quot;, &quot;FAM24735&quot;, &quot;FAM24792&quot;, &quot;FAM24763&quot;, &quot;FAM24793&quot;, 
&quot;FAM24795&quot;, &quot;FAM24797&quot;, &quot;FAM24798&quot;, &quot;FAM24854&quot;, &quot;FAM24855&quot;, &quot;FAM11075&quot;, 
&quot;FAM13492&quot;, &quot;FAM11001&quot;, &quot;FAM24762&quot;, &quot;FAM13499&quot;, &quot;FAM11143&quot;, &quot;FAM12062&quot;, 
&quot;FAM24751&quot;, &quot;FAM21277&quot;, &quot;FAM24765&quot;, &quot;FAM24733&quot;, &quot;FAM24771&quot;, &quot;FAM24737&quot;))
snps_long$samplesNew &lt;- droplevels(snps_long$samplesNew)
# snps_long %&gt;% filter(is.na(samplesNew)) %&gt;% select(sample) %&gt;% table()

sum(is.na(snps_long$samplesNew))

snps_long_final &lt;- snps_long %&gt;% select(site,species,samplesNew,culture,Snps)
table(snps_long$culture)

snps_long$samplesFINAL &lt;- paste0(snps_long$culture,&quot;-&quot;,snps_long$samplesNew)
table(snps_long$samplesFINAL )

write.csv(snps_long_final,paste0(data_path,&quot;snps_analysed_all.csv&quot;),row.names = FALSE)

###----------------------------------------
##add SNPeffect INFO
###----------------------------------------

sample_relative_temp_cleaned &lt;- snps_long_final

###spread the dataframe again

nrow(sample_relative_temp_cleaned)
table(sample_relative_temp_cleaned$culture)
sample_relative_temp_cleaned$samplesFINAL &lt;- paste0(sample_relative_temp_cleaned$culture,&quot;-&quot;,sample_relative_temp_cleaned$samplesNew)

table(sample_relative_temp_cleaned$samplesFINAL)

sample_relative_safe_final_prep &lt;- sample_relative_temp_cleaned %&gt;% select(site,species,samplesFINAL,Snps)

table(sample_relative_safe_final_prep$samplesFINAL)

sample_relative_safe_final_tsne &lt;- spread(sample_relative_safe_final_prep, samplesFINAL, Snps)  %&gt;%replace(is.na(.), 0)
nrow(sample_relative_safe_final_tsne)

##==================
##05_filter rowsums!=0
##==================

##remove row sums ==0
nrow(sample_relative_safe_final_tsne)
sum(rowSums(sample_relative_safe_final_tsne[,3:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)==0)

sample_relative_wide &lt;- sample_relative_safe_final_tsne[rowSums(sample_relative_safe_final_tsne[,4:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)&gt;0,]
colnames(sample_relative_wide)

##==================
##06_add gene information
##==================



RMK202_snpeff_eggnog_repeats_core &lt;- read_csv(paste0(data_path,&quot;RMK202_snpeff_eggnog_length_core_extern.txt&quot;), col_types = cols(core = col_character(),geneName = col_character(),effect = col_character(),PGAP_name = col_character(),type = col_character(),typeName = col_character(),typeGene = col_character(),Preferred_name = col_character())) %&gt;% select(-species)
table(RMK202_snpeff_eggnog_repeats_core$core)


sample_relative_wide_description &lt;- merge(sample_relative_wide,RMK202_snpeff_eggnog_repeats_core, by = &quot;site&quot;,all.x = TRUE)
dim(sample_relative_wide_description)
table(sample_relative_wide_description$species)
# nrow(sample_relative_wide_description)
table(sample_relative_wide_description$core)
table(sample_relative_wide_description$significance)
#---------------------------------subset for interesting columns


dput(colnames(sample_relative_wide_description))

sample_relative_wide_description_interest &lt;- sample_relative_wide_description %&gt;% select(site,species,effect,significance,geneName,COG_Functional_Category,length,core,&quot;101-starter 1&quot;, &quot;101-starter 10&quot;, &quot;101-starter 11&quot;, 
&quot;101-starter 2&quot;, &quot;101-starter 3&quot;, &quot;101-starter 4&quot;, &quot;101-starter 5&quot;, 
&quot;101-starter 6&quot;, &quot;101-starter 7&quot;, &quot;101-starter 8&quot;, &quot;101-starter 9&quot;, 
&quot;105-starter 1&quot;, &quot;105-starter 10&quot;, &quot;105-starter 11&quot;, &quot;105-starter 12&quot;, 
&quot;105-starter 13&quot;, &quot;105-starter 14&quot;, &quot;105-starter 15&quot;, &quot;105-starter 16&quot;, 
&quot;105-starter 2&quot;, &quot;105-starter 3&quot;, &quot;105-starter 4&quot;, &quot;105-starter 5&quot;, 
&quot;105-starter 6&quot;, &quot;105-starter 7&quot;, &quot;105-starter 8&quot;, &quot;105-starter 9&quot;, 
&quot;115-starter 1&quot;, &quot;115-starter 2&quot;, &quot;115-starter 3&quot;, &quot;115-starter 4&quot;, 
&quot;115-starter 5&quot;, &quot;115-starter 6&quot;, &quot;124-starter 1&quot;, &quot;124-starter 2&quot;, 
&quot;124-starter 3&quot;, &quot;124-starter 4&quot;, &quot;124-starter 5&quot;, &quot;124-starter 6&quot;, 
&quot;124-starter 7&quot;, &quot;150-starter 1&quot;, &quot;150-starter 10&quot;, &quot;150-starter 2&quot;, 
&quot;150-starter 3&quot;, &quot;150-starter 4&quot;, &quot;150-starter 5&quot;, &quot;150-starter 6&quot;, 
&quot;150-starter 7&quot;, &quot;150-starter 8&quot;, &quot;150-starter 9&quot;, &quot;190-starter 1&quot;, 
&quot;190-starter 2&quot;, &quot;190-starter 3&quot;, &quot;190-starter 4&quot;, &quot;190-starter 5&quot;, 
&quot;190-starter 6&quot;, &quot;190-starter 7&quot;, &quot;202-starter 1&quot;, &quot;202-starter 2&quot;, 
&quot;202-starter 3&quot;, &quot;202-starter 4&quot;, &quot;202-starter 5&quot;, &quot;202-starter 6&quot;, 
&quot;202-starter 9&quot;, &quot;203-starter 1&quot;, &quot;203-starter 2&quot;, &quot;203-starter 3&quot;, 
&quot;203-starter 4&quot;, &quot;203-starter 5&quot;, &quot;203-starter 6&quot;, &quot;203-starter 7&quot;, 
&quot;203-starter 8&quot;, &quot;280-starter 1&quot;, &quot;280-starter 2&quot;, &quot;280-starter 3&quot;, 
&quot;280-starter 4&quot;, &quot;280-starter 5&quot;, &quot;280-starter 6&quot;, &quot;280-starter 7&quot;, 
&quot;302-starter 1&quot;, &quot;302-starter 2&quot;, &quot;302-starter 3&quot;, &quot;302-starter 4&quot;, 
&quot;302-starter 5&quot;, &quot;302-starter 6&quot;, &quot;305-starter 1&quot;, &quot;305-starter 2&quot;, 
&quot;305-starter 3&quot;, &quot;305-starter 4&quot;, &quot;305-starter 5&quot;, &quot;305-starter 6&quot;, 
&quot;305-starter 7&quot;, &quot;305-starter 8&quot;, &quot;FAMstrains-FAM10973&quot;, &quot;FAMstrains-FAM10980&quot;, 
&quot;FAMstrains-FAM10996&quot;, &quot;FAMstrains-FAM11001&quot;, &quot;FAMstrains-FAM11008&quot;, 
&quot;FAMstrains-FAM11009&quot;, &quot;FAMstrains-FAM11021&quot;, &quot;FAMstrains-FAM11024&quot;, 
&quot;FAMstrains-FAM11049&quot;, &quot;FAMstrains-FAM11063&quot;, &quot;FAMstrains-FAM11071&quot;, 
&quot;FAMstrains-FAM11075&quot;, &quot;FAMstrains-FAM11141&quot;, &quot;FAMstrains-FAM11142&quot;, 
&quot;FAMstrains-FAM11143&quot;, &quot;FAMstrains-FAM12062&quot;, &quot;FAMstrains-FAM12080&quot;, 
&quot;FAMstrains-FAM12104&quot;, &quot;FAMstrains-FAM12105&quot;, &quot;FAMstrains-FAM12107&quot;, 
&quot;FAMstrains-FAM12109&quot;, &quot;FAMstrains-FAM12273&quot;, &quot;FAMstrains-FAM13491&quot;, 
&quot;FAMstrains-FAM13492&quot;, &quot;FAMstrains-FAM13493&quot;, &quot;FAMstrains-FAM13494&quot;, 
&quot;FAMstrains-FAM13495&quot;, &quot;FAMstrains-FAM13496&quot;, &quot;FAMstrains-FAM13497&quot;, 
&quot;FAMstrains-FAM13498&quot;, &quot;FAMstrains-FAM13499&quot;, &quot;FAMstrains-FAM13500&quot;, 
&quot;FAMstrains-FAM21277&quot;, &quot;FAMstrains-FAM21747&quot;, &quot;FAMstrains-FAM21769&quot;, 
&quot;FAMstrains-FAM21781&quot;, &quot;FAMstrains-FAM24724&quot;, &quot;FAMstrains-FAM24725&quot;, 
&quot;FAMstrains-FAM24726&quot;, &quot;FAMstrains-FAM24727&quot;, &quot;FAMstrains-FAM24728&quot;, 
&quot;FAMstrains-FAM24729&quot;, &quot;FAMstrains-FAM24730&quot;, &quot;FAMstrains-FAM24731&quot;, 
&quot;FAMstrains-FAM24732&quot;, &quot;FAMstrains-FAM24733&quot;, &quot;FAMstrains-FAM24734&quot;, 
&quot;FAMstrains-FAM24735&quot;, &quot;FAMstrains-FAM24736&quot;, &quot;FAMstrains-FAM24737&quot;, 
&quot;FAMstrains-FAM24738&quot;, &quot;FAMstrains-FAM24739&quot;, &quot;FAMstrains-FAM24740&quot;, 
&quot;FAMstrains-FAM24741&quot;, &quot;FAMstrains-FAM24742&quot;, &quot;FAMstrains-FAM24743&quot;, 
&quot;FAMstrains-FAM24744&quot;, &quot;FAMstrains-FAM24745&quot;, &quot;FAMstrains-FAM24746&quot;, 
&quot;FAMstrains-FAM24747&quot;, &quot;FAMstrains-FAM24748&quot;, &quot;FAMstrains-FAM24749&quot;, 
&quot;FAMstrains-FAM24750&quot;, &quot;FAMstrains-FAM24751&quot;, &quot;FAMstrains-FAM24752&quot;, 
&quot;FAMstrains-FAM24753&quot;, &quot;FAMstrains-FAM24754&quot;, &quot;FAMstrains-FAM24755&quot;, 
&quot;FAMstrains-FAM24756&quot;, &quot;FAMstrains-FAM24757&quot;, &quot;FAMstrains-FAM24758&quot;, 
&quot;FAMstrains-FAM24759&quot;, &quot;FAMstrains-FAM24760&quot;, &quot;FAMstrains-FAM24761&quot;, 
&quot;FAMstrains-FAM24762&quot;, &quot;FAMstrains-FAM24763&quot;, &quot;FAMstrains-FAM24764&quot;, 
&quot;FAMstrains-FAM24765&quot;, &quot;FAMstrains-FAM24766&quot;, &quot;FAMstrains-FAM24767&quot;, 
&quot;FAMstrains-FAM24768&quot;, &quot;FAMstrains-FAM24769&quot;, &quot;FAMstrains-FAM24770&quot;, 
&quot;FAMstrains-FAM24771&quot;, &quot;FAMstrains-FAM24772&quot;, &quot;FAMstrains-FAM24773&quot;, 
&quot;FAMstrains-FAM24774&quot;, &quot;FAMstrains-FAM24775&quot;, &quot;FAMstrains-FAM24776&quot;, 
&quot;FAMstrains-FAM24777&quot;, &quot;FAMstrains-FAM24778&quot;, &quot;FAMstrains-FAM24779&quot;, 
&quot;FAMstrains-FAM24780&quot;, &quot;FAMstrains-FAM24781&quot;, &quot;FAMstrains-FAM24782&quot;, 
&quot;FAMstrains-FAM24783&quot;, &quot;FAMstrains-FAM24784&quot;, &quot;FAMstrains-FAM24785&quot;, 
&quot;FAMstrains-FAM24786&quot;, &quot;FAMstrains-FAM24787&quot;, &quot;FAMstrains-FAM24790&quot;, 
&quot;FAMstrains-FAM24791&quot;, &quot;FAMstrains-FAM24792&quot;, &quot;FAMstrains-FAM24793&quot;, 
&quot;FAMstrains-FAM24794&quot;, &quot;FAMstrains-FAM24795&quot;, &quot;FAMstrains-FAM24797&quot;, 
&quot;FAMstrains-FAM24798&quot;, &quot;FAMstrains-FAM24799&quot;, &quot;FAMstrains-FAM24853&quot;, 
&quot;FAMstrains-FAM24854&quot;, &quot;FAMstrains-FAM24855&quot;)

##==================
##08_remove non-bacterial SNPs
##==================

table(sample_relative_wide_description_interest$species)
nrow(sample_relative_wide_description_interest)

sample_relative_wide_description_interest_subset &lt;- sample_relative_wide_description_interest[which(sample_relative_wide_description_interest$species %in% c(&quot;L. delbrueckii&quot;,&quot;S. thermophilus&quot;)),]
nrow(sample_relative_wide_description_interest_subset)
table(sample_relative_wide_description_interest_subset$species)
sample_relative_wide &lt;- sample_relative_wide_description_interest_subset
table(sample_relative_wide_description_interest_subset$core)
table(sample_relative_wide_description_interest_subset$core,sample_relative_wide_description_interest_subset$species)


##==================
##09_order variables properly
##==================
sample_relative_wide$significance &lt;- factor(sample_relative_wide$significance, levels=c(&quot;MODIFIER&quot;,&quot;LOW&quot;,&quot;MODERATE&quot;,&quot;HIGH&quot;))

##==================
##11_make long
##==================
colnames(sample_relative_wide)
sample_relative_long &lt;- gather(sample_relative_wide, sample, Snps, &quot;101-starter 1&quot;:&quot;FAMstrains-FAM24855&quot;, factor_key=TRUE,na.rm = TRUE) %&gt;% filter(Snps&gt;0)%&gt;% filter(!is.na(Snps))
nrow(sample_relative_long)
table(sample_relative_long$species)
table(sample_relative_long$core)


##==================
##12_make a unique gene infomoration
##==================

##==================
##output infomration
##==================



write.csv(sample_relative_wide,paste0(data_path,&quot;snps_analysed_all_withINfomration_wide.csv&quot;),row.names = FALSE)

dim(sample_relative_wide)
sample_relative_wide[which(sample_relative_wide$core==&quot;core&quot;),]

snps_long_final &lt;-  gather(sample_relative_wide, sample, Snps, &quot;101-starter 1&quot;:&quot;FAMstrains-FAM24855&quot;, factor_key=TRUE,na.rm = TRUE)
snps_long_final[which(snps_long_final$core==&quot;core&quot;),]

snps_long_final$species &lt;- snps_long_final$species
snps_long_final$culture &lt;- str_split_fixed(snps_long_final$sample,&quot;-&quot;,3)[,1]
table(snps_long_final$core)</code></pre>
</div>
<div id="strains-diversity" class="section level3">
<h3>Strains Diversity</h3>
<pre class="r"><code>library(tidyverse)

sample_relative_wide &lt;- read_csv(paste0(data_path,&quot;snps_analysed_all_withINfomration_wide.csv&quot;))
snps_wide_final &lt;- read_csv(paste0(data_path,&quot;snps_analysed_all_withINfomration_wide.csv&quot;))

snps_long_final &lt;-  sample_relative_wide %&gt;%  select(site,species,&quot;101-starter 1&quot;, 
&quot;101-starter 10&quot;, &quot;101-starter 11&quot;, &quot;101-starter 2&quot;, &quot;101-starter 3&quot;, 
&quot;101-starter 4&quot;, &quot;101-starter 5&quot;, &quot;101-starter 6&quot;, &quot;101-starter 7&quot;, 
&quot;101-starter 8&quot;, &quot;101-starter 9&quot;, &quot;105-starter 1&quot;, &quot;105-starter 10&quot;, 
&quot;105-starter 11&quot;, &quot;105-starter 12&quot;, &quot;105-starter 13&quot;, &quot;105-starter 14&quot;, 
&quot;105-starter 15&quot;, &quot;105-starter 16&quot;, &quot;105-starter 2&quot;, &quot;105-starter 3&quot;, 
&quot;105-starter 4&quot;, &quot;105-starter 5&quot;, &quot;105-starter 6&quot;, &quot;105-starter 7&quot;, 
&quot;105-starter 8&quot;, &quot;105-starter 9&quot;, &quot;115-starter 1&quot;, &quot;115-starter 2&quot;, 
&quot;115-starter 3&quot;, &quot;115-starter 4&quot;, &quot;115-starter 5&quot;, &quot;115-starter 6&quot;, 
&quot;124-starter 1&quot;, &quot;124-starter 2&quot;, &quot;124-starter 3&quot;, &quot;124-starter 4&quot;, 
&quot;124-starter 5&quot;, &quot;124-starter 6&quot;, &quot;124-starter 7&quot;, &quot;150-starter 1&quot;, 
&quot;150-starter 10&quot;, &quot;150-starter 2&quot;, &quot;150-starter 3&quot;, &quot;150-starter 4&quot;, 
&quot;150-starter 5&quot;, &quot;150-starter 6&quot;, &quot;150-starter 7&quot;, &quot;150-starter 8&quot;, 
&quot;150-starter 9&quot;, &quot;190-starter 1&quot;, &quot;190-starter 2&quot;, &quot;190-starter 3&quot;, 
&quot;190-starter 4&quot;, &quot;190-starter 5&quot;, &quot;190-starter 6&quot;, &quot;190-starter 7&quot;, 
&quot;202-starter 1&quot;, &quot;202-starter 2&quot;, &quot;202-starter 3&quot;, &quot;202-starter 4&quot;, 
&quot;202-starter 5&quot;, &quot;202-starter 6&quot;, &quot;202-starter 9&quot;, &quot;203-starter 1&quot;, 
&quot;203-starter 2&quot;, &quot;203-starter 3&quot;, &quot;203-starter 4&quot;, &quot;203-starter 5&quot;, 
&quot;203-starter 6&quot;, &quot;203-starter 7&quot;, &quot;203-starter 8&quot;, &quot;280-starter 1&quot;, 
&quot;280-starter 2&quot;, &quot;280-starter 3&quot;, &quot;280-starter 4&quot;, &quot;280-starter 5&quot;, 
&quot;280-starter 6&quot;, &quot;280-starter 7&quot;, &quot;302-starter 1&quot;, &quot;302-starter 2&quot;, 
&quot;302-starter 3&quot;, &quot;302-starter 4&quot;, &quot;302-starter 5&quot;, &quot;302-starter 6&quot;, 
&quot;305-starter 1&quot;, &quot;305-starter 2&quot;, &quot;305-starter 3&quot;, &quot;305-starter 4&quot;, 
&quot;305-starter 5&quot;, &quot;305-starter 6&quot;, &quot;305-starter 7&quot;, &quot;305-starter 8&quot;) %&gt;% gather(., sample, Snps, &quot;101-starter 1&quot;:&quot;305-starter 8&quot;, factor_key=TRUE,na.rm = TRUE)

##----------------
##analysis
##----------------

table(snps_long_final$core)
nrow(snps_long_final)
sample_relative_safe_woSTRAINS_sub01 &lt;- snps_long_final
sample_relative_safe_woSTRAINS_sub01$culture &lt;- str_split_fixed(sample_relative_safe_woSTRAINS_sub01$sample, &quot;-&quot;, 3)[,1]
sample_relative_safe_woSTRAINS_sub01$species.x &lt;- sample_relative_safe_woSTRAINS_sub01$species

table(sample_relative_safe_woSTRAINS_sub01$species)
table(sample_relative_safe_woSTRAINS_sub01$culture)
table(sample_relative_safe_woSTRAINS_sub01$sample)

table(sample_relative_safe_woSTRAINS_sub01$culture,sample_relative_safe_woSTRAINS_sub01$sample)

nrow(sample_relative_safe_woSTRAINS_sub01)
sample_relative_safe_woSTRAINS_sub02 &lt;- sample_relative_safe_woSTRAINS_sub01

nrow(sample_relative_safe_woSTRAINS_sub02)

sample_relative_safe_woSTRAINS_sub03 &lt;- sample_relative_safe_woSTRAINS_sub02[which(sample_relative_safe_woSTRAINS_sub02$Snps&gt;0.03),]
sample_relative_safe_woSTRAINS_sub04 &lt;- sample_relative_safe_woSTRAINS_sub02[which(sample_relative_safe_woSTRAINS_sub02$Snps&lt;=0.03),]

sample_relative_safe_woSTRAINS_sub04$Snps &lt;- 0
nrow(sample_relative_safe_woSTRAINS_sub04)
sample_relative_safe_woSTRAINS_sub05 &lt;- rbind(sample_relative_safe_woSTRAINS_sub03,sample_relative_safe_woSTRAINS_sub04)
sample_relative_safe_woSTRAINS_sub06 &lt;- sample_relative_safe_woSTRAINS_sub05[which(sample_relative_safe_woSTRAINS_sub05$significance %in% c(&quot;MODERATE&quot;,&quot;HIGH&quot;)),]
sample_relative_safe_woSTRAINS_sub06 &lt;- sample_relative_safe_woSTRAINS_sub05

nrow(sample_relative_safe_woSTRAINS_sub05)
table(sample_relative_safe_woSTRAINS_sub05$species)

sample_relative_safe_Sterm_final &lt;- sample_relative_safe_woSTRAINS_sub06[which(sample_relative_safe_woSTRAINS_sub06$species %in% c(&quot;L. delbrueckii&quot;,&quot;S. thermophilus&quot;)),] %&gt;% select(-species.x) 
sample_relative_safe_Sterm_final &lt;- unique(sample_relative_safe_Sterm_final)

sample_relative_safe_Sterm_final_wide &lt;- spread(sample_relative_safe_Sterm_final, sample, Snps)  %&gt;%replace(is.na(.), 0) %&gt;% as_tibble()
sample_relative_safe_Sterm_final_new &lt;- sample_relative_safe_Sterm_final

sample_relative_safe_Sterm_final_new$sampleComplete &lt;- sample_relative_safe_Sterm_final_new$sample
sample_relative_safe_Sterm_final_new &lt;- sample_relative_safe_Sterm_final_new %&gt;% select(c(-&quot;culture&quot;,-&quot;sample&quot;))

sample_relative_safe_Sterm_final_wide &lt;- spread(sample_relative_safe_Sterm_final_new, sampleComplete, Snps)  %&gt;%replace(is.na(.), 0)
colnames(sample_relative_safe_Sterm_final_wide)
dim(sample_relative_safe_Sterm_final_wide)

sample_relative_safe_Sterm_final_wide_filtered &lt;- sample_relative_safe_Sterm_final_wide[,c(1,2,which(colSums(sample_relative_safe_Sterm_final_wide[,c(-1,-2)])&gt;0)+2)]
dim(sample_relative_safe_Sterm_final_wide_filtered)
colnames(sample_relative_safe_Sterm_final_wide_filtered)

sample_relative_safe_Sterm_final_wide_filtered_long &lt;-  gather(sample_relative_safe_Sterm_final_wide_filtered, sampleComplete, Snps, &quot;101-starter 1&quot;:&quot;305-starter 8&quot;, factor_key=TRUE,na.rm = TRUE) 

sample_relative_safe_Sterm_final_wide_filtered_long$culture &lt;- str_split_fixed(sample_relative_safe_Sterm_final_wide_filtered_long$sampleComplete, &quot;-&quot;, 2)[,1]

sample_relative_safe_Sterm_final_wide_filtered_long$sample &lt;- str_split_fixed(sample_relative_safe_Sterm_final_wide_filtered_long$sampleComplete, &quot;-&quot;, 2)[,2]
sample_relative_safe_Sterm_final_wide_filtered_long$species &lt;- str_split_fixed(sample_relative_safe_Sterm_final_wide_filtered_long$site, &quot;_&quot;, 4)[,2]
table(sample_relative_safe_Sterm_final_wide_filtered_long$sample)
  # sample_relative_safe_Sterm_final_longAll &lt;-
sample_relative_safe_woSTRAINS &lt;- sample_relative_safe_Sterm_final_wide_filtered_long
sample_relative_safe_Sterm_final_wide_filtered_long$sample = factor(sample_relative_safe_Sterm_final_wide_filtered_long$sample, levels=c(&quot;starter 1&quot;,&quot;starter 2&quot;,&quot;starter 3&quot;,&quot;starter 4&quot;,&quot;starter 5&quot;,&quot;starter 6&quot;,&quot;starter 7&quot;,&quot;starter 8&quot;,&quot;starter 9&quot;,&quot;starter 10&quot;,&quot;starter 11&quot;,&quot;starter 12&quot;,&quot;starter 13&quot;,&quot;starter 14&quot;,&quot;starter 15&quot;,&quot;starter 16&quot;))
table(sample_relative_safe_Sterm_final_wide_filtered_long$culture,sample_relative_safe_Sterm_final_wide_filtered_long$sample)
##============
### plot both species
##============
sample_relative_safe_woSTRAINS$sample &lt;- base::droplevels(factor(sample_relative_safe_woSTRAINS$sample))
levels(sample_relative_safe_Sterm_final_wide_filtered_long$sample)
sample_relative_safe_Sterm_final_wide_filtered_long_woZero &lt;- sample_relative_safe_Sterm_final_wide_filtered_long %&gt;% filter(Snps&gt;0)
##============
### plot 
##============

all_colours &lt;- rev(c(&quot;#EB4D4D&quot;,&quot;#10B552&quot;) )

table(sample_relative_safe_Sterm_final_wide_filtered_long_woZero$species)
sample_relative_safe_Sterm_final_wide_filtered_long_woZero$species &lt;- plyr::revalue(sample_relative_safe_Sterm_final_wide_filtered_long_woZero$species,c(&quot;CP023139.1&quot;=&quot;L. delbrueckii&quot;,&quot;CP048750.1&quot;=&quot;S. thermophilus&quot;))

table(sample_relative_safe_Sterm_final_wide_filtered_long_woZero$culture,sample_relative_safe_Sterm_final_wide_filtered_long_woZero$sample)

##==================================
##plot
##==================================

  p2 &lt;- ggplot(sample_relative_safe_Sterm_final_wide_filtered_long_woZero,aes(x=sample,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.3, alpha=.1)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = &quot;free_x&quot;, space = &quot;free&quot;)+
            facet_grid(species~culture,scales = &quot;free_x&quot;, space = &quot;free&quot;)+
            # ggforce::facet_row(vars(culture)~species,scales = &quot;free_x&quot;, space = &quot;free&quot;)+
            # ggforce::facet_col(vars(species),scales = &quot;free_x&quot;, space = &quot;free&quot;)+
     labs(&quot;&quot;,
         x=&quot; &quot;,
         y=&quot;Alternative allele frequency&quot;)+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = &quot;transparent&quot;), # all rectangles      #axis.text.x = element_blank(),
          legend.position=&quot;none&quot;,
          panel.spacing = unit(1.2, &quot;lines&quot;)
  )
  
  p2

pdf( paste0(plot_path,&quot;AbundanceStrain_SNPs_all.pdf&quot;),width=12,height=8)

p2

dev.off()</code></pre>
</div>
<div id="strain-diversity-plot" class="section level3">
<h3>Strain diversity plot</h3>
<pre class="r"><code>##===============================
#preperation
##===============================
colnames(sample_relative_wide) &lt;- colnames(sample_relative_wide) %&gt;% gsub(&quot;FAMstrains-FAM&quot;,&quot;&quot;,.)

dput(colnames(sample_relative_wide))

sample_relative_wide_snpsUsed &lt;- sample_relative_wide    %&gt;% filter(core==&quot;core&quot;)  %&gt;%  select(&quot;site&quot;,
&quot;10973&quot;, &quot;10980&quot;, &quot;10996&quot;, 
&quot;11001&quot;, &quot;11008&quot;, &quot;11009&quot;, &quot;11021&quot;, &quot;11024&quot;, &quot;11049&quot;, &quot;11063&quot;, 
&quot;11071&quot;, &quot;11075&quot;, &quot;11141&quot;, &quot;11142&quot;, &quot;11143&quot;, &quot;12062&quot;, &quot;12080&quot;, 
&quot;12104&quot;, &quot;12105&quot;, &quot;12107&quot;, &quot;12109&quot;, &quot;12273&quot;, &quot;13491&quot;, &quot;13492&quot;, 
&quot;13493&quot;, &quot;13494&quot;, &quot;13495&quot;, &quot;13496&quot;, &quot;13497&quot;, &quot;13498&quot;, &quot;13499&quot;, 
&quot;13500&quot;, &quot;21277&quot;, &quot;21747&quot;, &quot;21769&quot;, &quot;21781&quot;, &quot;24724&quot;, &quot;24725&quot;, 
&quot;24726&quot;, &quot;24727&quot;, &quot;24728&quot;, &quot;24729&quot;, &quot;24730&quot;, &quot;24731&quot;, &quot;24732&quot;, 
&quot;24733&quot;, &quot;24734&quot;, &quot;24735&quot;, &quot;24736&quot;, &quot;24737&quot;, &quot;24738&quot;, &quot;24739&quot;, 
&quot;24740&quot;, &quot;24741&quot;, &quot;24742&quot;, &quot;24743&quot;, &quot;24744&quot;, &quot;24745&quot;, &quot;24746&quot;, 
&quot;24747&quot;, &quot;24748&quot;, &quot;24749&quot;, &quot;24750&quot;, &quot;24751&quot;, &quot;24752&quot;, &quot;24753&quot;, 
&quot;24754&quot;, &quot;24755&quot;, &quot;24756&quot;, &quot;24757&quot;, &quot;24758&quot;, &quot;24759&quot;, &quot;24760&quot;, 
&quot;24761&quot;, &quot;24762&quot;, &quot;24763&quot;, &quot;24764&quot;, &quot;24765&quot;, &quot;24766&quot;, &quot;24767&quot;, 
&quot;24768&quot;, &quot;24769&quot;, &quot;24770&quot;, &quot;24771&quot;, &quot;24772&quot;, &quot;24773&quot;, &quot;24774&quot;, 
&quot;24775&quot;, &quot;24776&quot;, &quot;24777&quot;, &quot;24778&quot;, &quot;24779&quot;, &quot;24780&quot;, &quot;24781&quot;, 
&quot;24782&quot;, &quot;24783&quot;, &quot;24784&quot;, &quot;24785&quot;, &quot;24786&quot;, &quot;24787&quot;, &quot;24790&quot;, 
&quot;24791&quot;, &quot;24792&quot;, &quot;24793&quot;, &quot;24794&quot;, &quot;24795&quot;, &quot;24797&quot;, &quot;24798&quot;, 
&quot;24799&quot;, &quot;24853&quot;, &quot;24854&quot;, &quot;24855&quot;)%&gt;%replace(is.na(.), 0)
sample_relative_wide_snpsUsed &lt;- sample_relative_wide_snpsUsed[rowSums(sample_relative_wide_snpsUsed[-1])&gt;0.5,]
dim(sample_relative_wide_snpsUsed)
is.na(colnames(sample_relative_wide_snpsUsed))

tmp &lt;- data.frame(sample_relative_wide_snpsUsed[-1]&gt;0.5)*1
colnames(tmp) &lt;- gsub(&quot;X&quot;,&quot;&quot;,colnames(tmp))
length(colnames(tmp) )

SNVs_strains &lt;- cbind(group=sample_relative_wide_snpsUsed[,1],tmp)
dim(SNVs_strains)

Poppunk_out &lt;- read_delim(paste0(data_path,&quot;Poppunk_out.all&quot;),&quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;genome&quot;,&quot;group&quot;), trim_ws = TRUE)
Poppunk_out &lt;- rbind(Poppunk_out,data.frame(genome=&quot;24777&quot;,group=&quot;13&quot;))

Poppunk_out %&gt;% filter(genome==24777)

for (genomesss in colnames(SNVs_strains)[-1]) {
  colnames(SNVs_strains)[which(colnames(SNVs_strains)==genomesss)] &lt;- Poppunk_out[grep(genomesss,Poppunk_out$genome),&quot;genome&quot;]
}

Poppunk_out_sub &lt;- Poppunk_out %&gt;% filter(genome %in%  colnames(SNVs_strains)[-1])

dim(Poppunk_out_sub)



write.table(Poppunk_out_sub,paste0(data_path,&quot;Poppunk_sub_forSNVS.all&quot;),sep=&quot;\t&quot;,quote = FALSE,row.names = FALSE)

write.table(SNVs_strains,paste0(data_path,&quot;gene_presence_absence.Rtab&quot;),sep=&quot;\t&quot;,quote = FALSE,row.names = FALSE)</code></pre>
<p>Here, I apply Gals paper appraoch. Github can be found <a href="https://github.com/ghoresh11/twilight">here</a>.</p>
<pre class="bash"><code>
rm -r  /data/Project/2020_StarterCultureDiversity/SNVprofiling/20210208//Gals_analysis/
mkdir -p /data/Project/2020_StarterCultureDiversity/SNVprofiling/20210208//Gals_analysis/
cd /data/Project/2020_StarterCultureDiversity/SNVprofiling/20210208//Gals_analysis/

Rscript /home/vincent/apps/twilight/classify_genes_VS.R --min_size 1 -p /data/Project/2020_StarterCultureDiversity/SNVprofiling/20210208/gene_presence_absence.Rtab -g /data/Project/2020_StarterCultureDiversity/SNVprofiling/20210208/Poppunk_sub_forSNVS.all

grep -i &quot;lineage specific&quot; out/genes_per_isolate.tab 
grep -i &quot;lineage specific&quot;  out/classification.tab
</code></pre>
<p>first have to run plot SNPs sterm chunk.</p>
<pre class="r"><code>library(tidyverse)

classification_SNVs &lt;- read_delim(paste0(data_path,&quot;classification.tab&quot;), &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)

classification_SNVs_lineage &lt;- classification_SNVs %&gt;% filter(grepl(&quot;Lineage specific &quot;,specific_class))

classification_SNVs_lineage$lineage &lt;- gsub(&quot;[^[:digit:]., ]&quot;, &quot;&quot;, classification_SNVs_lineage$details) %&gt;% as.numeric()
classification_SNVs_lineage_cleaned &lt;- classification_SNVs_lineage %&gt;% select(gene_name,lineage,specific_class)

lineage_relabel &lt;- read_csv(paste0(data_path,&quot;lineage_relabel.csv&quot;))
classification_SNVs_lineage_cleaned &lt;- merge(classification_SNVs_lineage_cleaned,lineage_relabel,by.x=&quot;lineage&quot;,by.y=&quot;poppunk&quot;,all.x=TRUE)
##------------------------------------------
##plotting
##------------------------------------------
sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended &lt;- merge(sample_relative_safe_Sterm_final_wide_filtered_long_woZero,classification_SNVs_lineage_cleaned,by.x=&quot;site&quot;,by.y=&quot;gene_name&quot;) %&gt;% filter(groupAssignment!=&quot;13&quot;)

sum(is.na(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended$specific_class))

dim(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended)

sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended  %&gt;% dim()
sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended %&gt;% filter((culture %in% c(101,105,115,124,150,190) &amp; bacsfrom %in% c(&quot;emmentaler&quot;,&quot;mix&quot;)) | (culture %in% c(202,203,280) &amp; bacsfrom %in% c(&quot;gruyere&quot;))| (culture %in% c(302,305) &amp; bacsfrom %in% c(&quot;sbrinz&quot;,&quot;mix&quot;)) ) %&gt;% dim()
sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended &lt;- sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended %&gt;% filter((culture %in% c(101,105,115,124,150,190) &amp; bacsfrom %in% c(&quot;emmentaler&quot;,&quot;mix&quot;)) | (culture %in% c(202,203,280) &amp; bacsfrom %in% c(&quot;gruyere&quot;))| (culture %in% c(302,305) &amp; bacsfrom %in% c(&quot;sbrinz&quot;,&quot;mix&quot;)) )

 p3 &lt;- ggplot(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended,aes(x=sample,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.3, alpha=.8)+ 

            facet_grid(groupAssignment~culture,scales = &quot;free_x&quot;, space = &quot;free&quot;)+
     labs(&quot;&quot;,
         x=&quot; &quot;,
         y=&quot;Alternative allele frequency&quot;)+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = &quot;transparent&quot;), # all rectangles      #axis.text.x = element_blank(),
          legend.position=&quot;none&quot;,
          panel.spacing = unit(1.2, &quot;lines&quot;)
  )
  
  p3
  
  ###------------------------
  ##relative abundance of SNVs and strains
  ###------------------------
  
sample_medians &lt;- sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended %&gt;% filter(!groupAssignment %in% c(13,14)) %&gt;% group_by(sampleComplete,groupAssignment) %&gt;% dplyr::summarize(medianss=median(Snps),Errors=sd(Snps),nsss=length(Snps))
  sample_medians$groupAssignment &lt;- as.character(sample_medians$groupAssignment)
  sample_medians$sample &lt;- str_split_fixed(sample_medians$sampleComplete, &quot;-&quot;, 2)[,2]
      sample_medians$culture &lt;- as.character(str_split_fixed(sample_medians$sampleComplete, &quot;-&quot;, 2)[,1])

  sample_medians$groupAssignment = factor(sample_medians$groupAssignment, levels=as.character(1:12))

    cols &lt;- rainbow(14)[1:12]

ggplot(sample_medians,aes(x=sample,y=medianss,fill=groupAssignment))+geom_bar(position=&quot;stack&quot;, stat=&quot;identity&quot;,alpha=0.5)+facet_grid(~culture,scales = &quot;free_x&quot;, space = &quot;free&quot;)+theme_classic()+scale_fill_manual(values = cols)+scale_color_manual(values = cols)
  
  
  ###--------------------------------
  ##normalise counts
  ###--------------------------------
  
  
    sample_medians_max &lt;- sample_medians %&gt;% group_by(sampleComplete) %&gt;% dplyr::summarize(factorsss=sum(medianss))

 sample_medians_normalised &lt;-  merge(sample_medians,sample_medians_max,by=&quot;sampleComplete&quot;)
  
sample_medians_normalised$normalised_ratio &lt;- sample_medians_normalised$medianss/sample_medians_normalised$factorsss
  sample_medians_normalised$groupAssignment = factor(sample_medians_normalised$groupAssignment, levels=as.character(1:12))


  ggplot(sample_medians_normalised,aes(x=sample,y=normalised_ratio,fill=groupAssignment))+geom_bar(position=&quot;stack&quot;, stat=&quot;identity&quot;,alpha=0.5)+facet_grid(~culture,scales = &quot;free_x&quot;, space = &quot;free&quot;)+theme_classic()+scale_fill_manual(values = cols)+scale_color_manual(values = cols)

  
  sample_medians_normalised_reduced &lt;- sample_medians_normalised %&gt;% select(sampleComplete,groupAssignment,sample, culture , normalised_ratio)
  
  
  ####-----------------------------------
  #how many strains are lost
  ####-----------------------------------
  
  sample_medians_normalised_reduced
  
  ###--------------------------------
  ##add missing samples
  ##some metagenomes do not have any strain information
  #here, I find out which ones and then I add the strain which we isolated from
  ###--------------------------------
  
  unique(sample_medians_normalised$sampleComplete) %&gt;% length()
  sample_medians_normalised %&gt;% select(sampleComplete,sample,culture) %&gt;% unique() %&gt;% dim()

  strained_samples &lt;- unique(sample_medians_normalised$sampleComplete) %&gt;% as.character()
  
  colnames(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended)
  table(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended$sampleComplete) %&gt;% length()
  
  all_samples &lt;- unique(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_extended$sampleComplete) %&gt;% levels()
  
  # strained_samples %in%  all_samples
all_samples[ which(!all_samples %in%  strained_samples)]
   
   
   
all_sterm_strain_info_ZOOMED &lt;- read_csv(paste0(data_path,&quot;genomeAssemblyStats.csv&quot;),  col_types = cols(starterCulture = col_character(),  strain = col_character())) %&gt;% filter(species==&quot;S.thermophilus&quot;) %&gt;% select(c(strain,Cheese,starterCulture,Sequencing)) %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;strain&quot;)%&gt;%replace(is.na(.), &quot;Unknown&quot;) 
   
   ##-------the two 302 have lineage 4 strains
     all_sterm_strain_info_ZOOMED %&gt;% filter(starterCulture==&quot;302&quot;)

     addon302 &lt;- data.frame(sampleComplete=c(&quot;302-starter 4&quot;,&quot;302-starter 5&quot;),
               groupAssignment=&quot;4&quot;,
               sample=c(&quot;starter 4&quot;,&quot;starter 5&quot;),
               culture=&quot;302&quot;,
               normalised_ratio=1)
     
   ##-------the five 115 metagenomes have lineage 7

  all_sterm_strain_info_ZOOMED %&gt;% filter(starterCulture==&quot;115&quot;)
  
  addon115 &lt;- data.frame(sampleComplete=c( &quot;115-starter 1&quot; ,&quot;115-starter 2&quot; ,&quot;115-starter 3&quot;, &quot;115-starter 4&quot; ,&quot;115-starter 5&quot;, &quot;115-starter 6&quot;),
               groupAssignment=&quot;7&quot;,
               sample=c(&quot;starter 1&quot;,&quot;starter 2&quot;,&quot;starter 3&quot;,&quot;starter 4&quot;,&quot;starter 5&quot;,&quot;starter 6&quot;),
               culture=&quot;115&quot;,
               normalised_ratio=1)
  
  ##---------bring together
  
  
 sample_medians_normalised_final &lt;-  rbind(sample_medians_normalised_reduced,addon302)
  # sample_medians_normalised_final &lt;-  rbind(sample_medians_normalised_reduced,addon115,addon302)

 
 sample_medians_normalised_final$sample = factor(sample_medians_normalised_final$sample, levels=c(&quot;starter 1&quot;,&quot;starter 2&quot;,&quot;starter 3&quot;,&quot;starter 4&quot;,&quot;starter 5&quot;,&quot;starter 6&quot;,&quot;starter 7&quot;,&quot;starter 8&quot;,&quot;starter 9&quot;,&quot;starter 10&quot;,&quot;starter 11&quot;,&quot;starter 12&quot;,&quot;starter 13&quot;,&quot;starter 14&quot;,&quot;starter 15&quot;,&quot;starter 16&quot;))
 
   final_strain_abundance_plot &lt;- ggplot(sample_medians_normalised_final,aes(x=sample,y=normalised_ratio,fill=groupAssignment))+geom_bar(position=&quot;stack&quot;, stat=&quot;identity&quot;,alpha=0.5)+facet_grid(~culture,scales = &quot;free_x&quot;, space = &quot;free&quot;)+theme_classic()+scale_fill_manual(values = cols)+scale_color_manual(values = cols)+labs(x=&quot;&quot;,y=&quot;Relative abundance&quot;)+theme(legend.title = element_blank(),axis.text.x = element_text(angle = 75, hjust = 1,size=6))+ scale_y_continuous(expand = c(0,0))
   
   final_strain_abundance_plot
   
   
pdf(paste0(plot_path,&quot;AbundanceStrain_all.pdf&quot;),width=10,height=3)

final_strain_abundance_plot

dev.off()</code></pre>
</div>
<div id="sterm-nucleotide-diversity" class="section level3">
<h3>Sterm nucleotide diversity</h3>
<pre class="r"><code>colnames(sample_relative_wide)
snps_long_final &lt;-  gather(sample_relative_wide, sample, Snps, &quot;101-starter 1&quot;:&quot;24855&quot;, factor_key=TRUE,na.rm = TRUE)

###------------------------------------
##-------finalise figure similar to species diversity
###------------------------------------

coregenomeSizeSterm=1300000
colnames(snps_long_final)
table(snps_long_final$species)
# sample_relative_wide %&gt;% filter(core==&quot;core&quot;) %&gt;% dim()
snps_long_final_cleaned &lt;- snps_long_final %&gt;% filter(Snps&gt;0.05) %&gt;% filter(Snps&gt;0.2) %&gt;% filter(Snps&lt;0.8) %&gt;% group_by(species,sample) %&gt;% dplyr::summarize(counts=n()) %&gt;% mutate(nuclotideDiv=100*(counts/coregenomeSizeSterm)) %&gt;% separate(sample, c(&quot;culture&quot;, &quot;sample&quot;),sep = &quot;-&quot; ) %&gt;% filter(grepl(&quot;starter&quot;,sample)) #%&gt;% filter(culture!=&quot;FAMstrains&quot;)
table(snps_long_final_cleaned$sample)
# snps_long_final_cleaned &lt;- snps_long_final %&gt;% filter(Snps&gt;0.05) %&gt;% filter(Snps&gt;0.2) %&gt;% filter(Snps&lt;0.8) %&gt;% group_by(species,sample) %&gt;% dplyr::summarize(counts=n()) %&gt;% mutate(nuclotideDiv=100*(counts/coregenomeSizeSterm)) %&gt;% separate(sample, c(&quot;culture&quot;, &quot;sample&quot;),sep = &quot;-&quot; ) %&gt;% filter(culture!=&quot;FAMstrains&quot;) 




nucleotide_Div_time &lt;- ggplot(snps_long_final_cleaned,aes(x=sample,nuclotideDiv,color=species,fill=species,group=species))+geom_point()+geom_line()+facet_grid(~culture,scales = &quot;free_x&quot;, space = &quot;free&quot;)+theme_classic()+theme(legend.position = &quot;none&quot;,axis.text.x =element_blank(),axis.title.x = element_blank(),axis.ticks.x=element_blank())+scale_y_continuous(trans = &quot;log10&quot;,breaks=c(0.0001,0.001,0.01,0.1,0.4))+labs(y=&quot;Nucleotide diversity [%]&quot;)+scale_fill_manual(values=c(&quot;blue&quot;,&quot;pink&quot;))+scale_color_manual(values=c(&quot;#fdc2b5ff&quot;,&quot;#007fa6ff&quot;))

nucleotide_Div_time


pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/nucleotid_diversity.pdf&quot;, width = 10, height = 3)

nucleotide_Div_time


dev.off()
  # final_strain_abundance_plot &lt;- ggplot(sample_medians_normalised_final,aes(x=sample,y=normalised_ratio,fill=groupAssignment))+geom_bar(position=&quot;stack&quot;, stat=&quot;identity&quot;,alpha=0.5)+facet_grid(~culture,scales = &quot;free_x&quot;, space = &quot;free&quot;)+theme_classic()+scale_fill_manual(values = cols)+scale_color_manual(values = cols)+labs(x=&quot;&quot;,y=&quot;Relative abundance&quot;)+theme(legend.title = element_blank(),axis.text.x = element_text(angle = 75, hjust = 1,size=6))+ scale_y_continuous(expand = c(0,0))
  #  
  #  final_strain_abundance_plot</code></pre>
<pre class="r"><code>##===============================
#old
##===============================
library(robustbase)
library(VennDiagram)
##==================================================Streptococcus thermophilus===================================================================
colnames(sample_relative_wide)
colnames(sample_relative_wide) &lt;- colnames(sample_relative_wide) %&gt;% gsub(&quot;FAMstrains-FAM&quot;,&quot;&quot;,.)

dput(colnames(sample_relative_wide))

 # %&gt;% filter(species==&quot;S. thermophilus&quot;) 
sample_relative_wide_snpsUsed &lt;- sample_relative_wide  %&gt;%replace(is.na(.), 0)  %&gt;% filter(core==&quot;core&quot;)   %&gt;% filter( &quot;Lyo\n1996&quot;&gt;0.05| &quot;Lyo\n2012&quot;&gt;0.05| &quot;Lyo\n2014&quot;&gt;0.05| &quot;working\nstock&quot;&gt;0.05 | &quot;starter\nculture\n2012&quot;&gt;0.05 | &quot;starter\nculture\n2018&quot;&gt;0.05 | &quot;experiment_A&quot;&gt;0.05| &quot;experiment_B&quot;&gt;0.05| &quot;experiment_C&quot;&gt;0.05 | &quot;experiment_D&quot;&gt;0.05 | &quot;experiment_E&quot;&gt;0.05)


colnames(sample_relative_wide_snpsUsed)
sample_relative_wide_snpsUsed_lin1 &lt;- sample_relative_wide_snpsUsed %&gt;% filter(`24853`&gt;0.8| `24798`&gt;0.8| `13493`&gt;0.8| `13500`&gt;0.8 | `24737`&gt;0.8 | `24854`&gt;0.8| `13491`&gt;0.8 | `13492`&gt;0.8 ) %&gt;% add_column(explained=&quot;lin1&quot;)
sample_relative_wide_snpsUsed_lin2 &lt;- sample_relative_wide_snpsUsed %&gt;% filter(`S72`&gt;0.8| `24855`&gt;0.8| `13494`&gt;0.8)%&gt;% add_column(explained=&quot;lin2&quot;)
sample_relative_wide_snpsUsed_lin3 &lt;- sample_relative_wide_snpsUsed %&gt;% filter(`S50`&gt;0.8| `24740`&gt;0.8| `24738`&gt;0.8| `13499`&gt;0.8 )%&gt;% add_column(explained=&quot;lin3&quot;)
sample_relative_wide_snpsUsed_lin4 &lt;- sample_relative_wide_snpsUsed %&gt;% filter(`24739`&gt;0.8| `13497`&gt;0.8| `13496`&gt;0.8| `13495`&gt;0.8 )%&gt;% add_column(explained=&quot;lin4&quot;)
allsites &lt;- c(sample_relative_wide_snpsUsed_lin1$site, sample_relative_wide_snpsUsed_lin2$site, sample_relative_wide_snpsUsed_lin3$site,sample_relative_wide_snpsUsed_lin4$site)

##----------------------
# Chart venn diagramm
##----------------------
temp &lt;-venn.diagram(
  x = list(sample_relative_wide_snpsUsed_lin1$site, sample_relative_wide_snpsUsed_lin2$site, sample_relative_wide_snpsUsed_lin3$site,sample_relative_wide_snpsUsed_lin4$site),
  category.names = c(&quot;lin 1&quot; , &quot;lin 2 &quot; , &quot;lin 3&quot;, &quot;lin 4&quot;),
  filename = NULL
)

plot.new() 

grid.draw(temp)
# 
# pdf(&quot;testpdf&quot;, width = 14, height = 7)
# 
# grid.draw(temp)
# 
# dev.off()

grid.draw(temp)

allsites[duplicated(allsites)]
sum(duplicated(allsites))
sum(!duplicated(allsites))

sum(table(allsites)&gt;1)
sum(table(allsites)==1)

duplictednames &lt;- names(table(allsites)[(table(allsites)&gt;1)])

##----------------------
# Chart venn diagramm
##----------------------

sample_relative_wide_snpsUsed_new_01 &lt;- rbind(sample_relative_wide_snpsUsed_lin1,sample_relative_wide_snpsUsed_lin2,sample_relative_wide_snpsUsed_lin3,sample_relative_wide_snpsUsed_lin4)
sample_relative_wide_snpsUsed_new_02 &lt;- sample_relative_wide_snpsUsed_new_01 %&gt;% filter(!site %in%duplictednames) #lineage specific snps
table(sample_relative_wide_snpsUsed_new_02$explained)
sample_relative_wide_snpsUsed_new_temp &lt;- sample_relative_wide_snpsUsed %&gt;% filter(site %in%duplictednames)%&gt;% add_column(explained=&quot;multiple&quot;) ##duplicated sites
sample_relative_wide_snpsUsed_new_03 &lt;- rbind(sample_relative_wide_snpsUsed_new_02,sample_relative_wide_snpsUsed_new_temp) ##all explained sites
sample_relative_wide_snpsUsed_tmp &lt;-  sample_relative_wide_snpsUsed %&gt;% filter(!site %in%sample_relative_wide_snpsUsed_new_03$site) %&gt;% add_column(explained=&quot;not explained&quot;)
sample_relative_wide_snpsUsed_final &lt;- rbind(sample_relative_wide_snpsUsed_new_03,sample_relative_wide_snpsUsed_tmp) ##all explained sites


##EXKURSION LINEAGE SPECIFIC DUPLICATIONS
TMP &lt;- sample_relative_wide_snpsUsed %&gt;% filter(site %in%duplictednames)
TMP$e

short_explained &lt;- sample_relative_wide_snpsUsed_new_01 %&gt;%filter(site %in%duplictednames) %&gt;%  select(c(&quot;site&quot;,&quot;species&quot;,&quot;explained&quot;)) %&gt;% mutate(abundance = explained) %&gt;% spread(., explained, abundance)
short_explained$explained &lt;- paste(short_explained$lin1,short_explained$lin2,short_explained$lin3,short_explained$lin4,sep=&quot;_&quot;)
short_explained_multiple &lt;- short_explained%&gt;% select(&quot;site&quot;,&quot;explained&quot;)
sample_relative_wide_snpsUsed_new_temp &lt;- sample_relative_wide_snpsUsed %&gt;% filter(site %in%duplictednames) ##duplicated sites
sample_relative_wide_snpsUsed_new_temp_02 &lt;- merge(sample_relative_wide_snpsUsed_new_temp,short_explained_multiple,by=&quot;site&quot;)
sample_relative_wide_snpsUsed_new_03 &lt;- rbind(sample_relative_wide_snpsUsed_new_02,sample_relative_wide_snpsUsed_new_temp_02) ##all explained sites
sample_relative_wide_snpsUsed_tmp &lt;-  sample_relative_wide_snpsUsed %&gt;% filter(!site %in%sample_relative_wide_snpsUsed_new_03$site) %&gt;% add_column(explained=&quot;not explained&quot;)
sample_relative_wide_snpsUsed_final &lt;- rbind(sample_relative_wide_snpsUsed_new_03,sample_relative_wide_snpsUsed_tmp) ##all explained sites


totalSNPS &lt;- nrow(sample_relative_wide_snpsUsed_final)
100*(table(sample_relative_wide_snpsUsed_final$explained)/totalSNPS)
table(sample_relative_wide_snpsUsed_final$explained)
##----------------------
# boxplot
##----------------------
# sample_relative_wide_snpsUsed_final_long &lt;- sample_relative_wide_snpsUsed_final %&gt;% select( &quot;site&quot;,&quot;Lyo\n1996&quot;,&quot;Lyo\n2012&quot;,&quot;Lyo\n2014&quot;,&quot;working\nstock&quot;,&quot;starter\nculture\n2012&quot;,&quot;starter\nculture\n2018&quot;,&quot;experiment_A&quot;,&quot;experiment_B&quot;,&quot;experiment_C&quot;,&quot;experiment_D&quot;,&quot;experiment_E&quot;,&quot;explained&quot;)%&gt;% gather(.,sample,Median,&quot;Lyo\n1996&quot;,&quot;Lyo\n2012&quot;,&quot;Lyo\n2014&quot;,&quot;working\nstock&quot;,&quot;starter\nculture\n2012&quot;,&quot;starter\nculture\n2018&quot;,&quot;experiment_A&quot;,&quot;experiment_B&quot;,&quot;experiment_C&quot;,&quot;experiment_D&quot;,&quot;experiment_E&quot;) 

sample_relative_wide_snpsUsed_final &lt;- sample_relative_wide_snpsUsed_final %&gt;%  dplyr::rename(&quot;Reference 2&quot;=&quot;cheesemaking\nday2&quot;,&quot;Reference 1&quot;=&quot;cheesemaking\nday1&quot; )

sample_relative_wide_snpsUsed_final_long &lt;- sample_relative_wide_snpsUsed_final %&gt;% select( &quot;site&quot;,&quot;Lyo\n1996&quot;,&quot;Lyo\n2012&quot;,&quot;Lyo\n2014&quot;,&quot;working\nstock&quot;,&quot;starter\nculture\n2012&quot;,&quot;starter\nculture\n2018&quot;,&quot;experiment_A&quot;,&quot;experiment_B&quot;,&quot;experiment_C&quot;,&quot;experiment_D&quot;,&quot;experiment_E&quot;,&quot;Reference 1&quot;,&quot;Reference 2&quot;,&quot;explained&quot;)%&gt;% gather(.,sample,Median,&quot;Lyo\n1996&quot;,&quot;Lyo\n2012&quot;,&quot;Lyo\n2014&quot;,&quot;working\nstock&quot;,&quot;starter\nculture\n2012&quot;,&quot;starter\nculture\n2018&quot;,&quot;experiment_A&quot;,&quot;experiment_B&quot;,&quot;experiment_C&quot;,&quot;experiment_D&quot;,&quot;experiment_E&quot;,&quot;Reference 1&quot;,&quot;Reference 2&quot;) 

# ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median))+geom_boxplot()+theme_classic()+facet_wrap(~explained)

#remove zeroes
sample_relative_wide_snpsUsed_final_long_woZeros &lt;- sample_relative_wide_snpsUsed_final_long %&gt;% filter(Median&gt;0.1)
ggplot(sample_relative_wide_snpsUsed_final_long_woZeros,aes(x=sample,y=Median))+geom_boxplot()+theme_classic()+facet_wrap(~explained)


##----------------------
# lineplot
##----------------------
sample_relative_wide_snpsUsed_final_long_woZeros$sample = factor(sample_relative_wide_snpsUsed_final_long_woZeros$sample, levels=c(&quot;Lyo\n1996&quot;,&quot;Lyo\n2012&quot;,&quot;Lyo\n2014&quot;,&quot;working\nstock&quot;,&quot;starter\nculture\n2012&quot;,&quot;starter\nculture\n2018&quot;,&quot;cheesemaking\nday1&quot;,&quot;cheesemaking\nday2&quot;,&quot;experiment_A&quot;,&quot;experiment_B&quot;,&quot;experiment_C&quot;,&quot;experiment_D&quot;,&quot;experiment_E&quot;,&quot;Reference 1&quot;,&quot;Reference 2&quot;))

sample_relative_wide_snpsUsed_final_long_woZeros$explained = factor(sample_relative_wide_snpsUsed_final_long_woZeros$explained, levels=c(&quot;lin1&quot;,&quot;lin2&quot;,&quot;lin3&quot;,&quot;lin4&quot;,&quot;not explained&quot;,&quot;lin1_lin2_NA_NA&quot;,&quot;NA_NA_lin3_lin4&quot;,&quot;lin1_lin2_lin3_lin4&quot;,&quot;lin1_lin2_NA_lin4&quot;,&quot;lin1_NA_lin3_lin4&quot;,&quot;lin1_NA_lin3_NA&quot;,&quot;lin1_NA_NA_lin4&quot;,&quot;NA_lin2_lin3_NA&quot;,&quot;NA_lin2_NA_lin4&quot;,&quot;NA_lin2_lin3_lin4&quot;))


# ggplot(sample_relative_wide_snpsUsed_final_long_woZeros,aes(x=sample,y=Median,group=site))+geom_line(color=&quot;red&quot;,size=0.2, alpha=.1)+theme_classic()
sample_relative_wide_snpsUsed_final_long_woZeros &lt;- sample_relative_wide_snpsUsed_final_long_woZeros %&gt;% filter(explained!=&quot;not explained&quot;)

pExplainted &lt;- ggplot(sample_relative_wide_snpsUsed_final_long_woZeros,aes(x=sample,y=Median,group=site,color=explained,fill=explained))+geom_line(alpha=.5,size=0.5)+theme_classic()+facet_wrap(~explained)+theme(legend.position = &quot;none&quot;,axis.text.x = element_text(angle = 75, hjust = 1,size=9))

table(sample_relative_wide_snpsUsed_final_long_woZeros$explained)
pExplainted


  svg(&quot;~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_explained_by_isolates.svg&quot;,width=7,height=4.5)
# png(&quot;~/Desktop/Projects/2019_RMK202_analysis/plot/supp_altNucFre_lineages.png&quot;, width = 3400, height = 2600,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted
dev.off()
###--------------------
##subset for ppx
table(sample_relative_wide_snpsUsed_final_long_woZeros$explained)
# sample_relative_wide_snpsUsed_final_long_tmp &lt;-  sample_relative_wide_snpsUsed_final_long_woZeros %&gt;% filter(explained %in% c(&quot;NA_NA_lin3_lin4&quot;,&quot;NA_lin2_lin3_lin4&quot;,&quot;NA_lin2_NA_lin4&quot;,&quot;lin3&quot;))

sample_relative_wide_snpsUsed_final_long_tmp &lt;-  sample_relative_wide_snpsUsed_final_long_woZeros %&gt;% filter(explained %in% c(&quot;lin1_NA_NA_lin4&quot;,&quot;NA_lin2_lin3_NA&quot;,&quot;NA_lin2_NA_lin4&quot;,&quot;lin1_NA_lin3_NA&quot;))


  sample_relative_wide_snpsUsed_final_long_tmp$explained = factor(sample_relative_wide_snpsUsed_final_long_tmp$explained, levels=c(&quot;lin1_NA_NA_lin4&quot;,&quot;NA_lin2_lin3_NA&quot;,&quot;NA_lin2_NA_lin4&quot;,&quot;lin1_NA_lin3_NA&quot;))

# sample_relative_wide_snpsUsed_final_long$explained_02 &lt;- revalue(sample_relative_wide_snpsUsed_final_long$explained, c(&quot;lin1&quot;=&quot;explained by isolates&quot;, &quot;lin2&quot;=&quot;explained by isolates&quot;,&quot;lin3&quot;=&quot;explained by isolates&quot;,&quot;lin4&quot;=&quot;explained by isolates&quot;,&quot;multiple&quot;=&quot;explained by isolates&quot;,&quot;not explained&quot;=&quot;not explained by isolates&quot;))

pExplainted &lt;- ggplot(sample_relative_wide_snpsUsed_final_long_tmp,aes(x=sample,y=Median,group=site,color=explained,fill=explained))+geom_line(alpha=.5,size=0.5)+theme_classic()+facet_wrap(~explained,ncol=4)+theme(legend.position = &quot;none&quot;,axis.text.x = element_text(angle = 75, hjust = 1,size=6))

pExplainted

png(&quot;~/Desktop/Projects/2019_RMK202_analysis/plot/supp_altNucFre_lineages_03.png&quot;, width = 3400, height = 1000,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted
dev.off()

###--------------------

sample_relative_wide_snpsUsed_final_long$explained_02 &lt;- revalue(sample_relative_wide_snpsUsed_final_long$explained, c(&quot;lin1&quot;=&quot;explained by isolates&quot;, &quot;lin2&quot;=&quot;explained by isolates&quot;,&quot;lin3&quot;=&quot;explained by isolates&quot;,&quot;lin4&quot;=&quot;explained by isolates&quot;,&quot;multiple&quot;=&quot;explained by isolates&quot;,&quot;not explained&quot;=&quot;not explained by isolates&quot;))

pExplainted &lt;- ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median,group=site,color=explained_02,fill=explained_02))+geom_line(size=0.5,alpha=0.5)+theme_classic()+facet_wrap(~explained_02)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = &quot;none&quot;)+
    labs(&quot;&quot;,
         x=&quot;&quot;,
         y=&quot;Alternative allele frequency&quot;)
pExplainted

pExplainted &lt;- ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median,group=site,color=explained_02,fill=explained_02))+geom_line(size=0.5,alpha=0.5)+theme_classic()+facet_wrap(~explained_02)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = &quot;none&quot;)+
    labs(&quot;&quot;,
         x=&quot;&quot;,
         y=&quot;Alternative allele frequency&quot;)
pExplainted


  svg(&quot;~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_explained_by_isolates..svg&quot;,width=7,height=4.5)
#    # png(&quot;~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png&quot;, width = 1000, height = 800,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted

dev.off()



##----------------------
# not explained
##----------------------

sample_relative_wide_snpsUsed_final_long$explained_03 &lt;- ifelse(sample_relative_wide_snpsUsed_final_long$explained_02==&quot;not explained by isolates&quot;,&quot;not explained by isolates&quot;,&quot;explained&quot;)
  
  sample_relative_wide_snpsUsed_final_long$sample = factor(sample_relative_wide_snpsUsed_final_long$sample, levels=c(&quot;Lyo\n1996&quot;,&quot;Lyo\n2012&quot;,&quot;Lyo\n2014&quot;,&quot;working\nstock&quot;,&quot;starter\nculture\n2012&quot;,&quot;starter\nculture\n2018&quot;,&quot;cheesemaking\nday1&quot;,&quot;cheesemaking\nday2&quot;,&quot;experiment_A&quot;,&quot;experiment_B&quot;,&quot;experiment_C&quot;,&quot;experiment_D&quot;,&quot;experiment_E&quot;))

  
pExplainted_rough &lt;- ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median,group=site,color=explained_03,fill=explained_03))+geom_line(size=0.5,alpha=0.5)+theme_classic()+facet_wrap(~explained_03)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = &quot;none&quot;)+
    labs(&quot;&quot;,
         x=&quot;&quot;,
         y=&quot;Alternative allele frequency&quot;)
pExplainted_rough

  svg(&quot;~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_notExplained.svg&quot;,width=7,height=4.5)
#    # png(&quot;~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png&quot;, width = 1000, height = 800,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted_rough

dev.off()

##----------------------
# where do these funky linkes (~recombinatnts) locate on the genome
##----------------------

sample_relative_wide_snpsUsed_final_long_recombinants &lt;- sample_relative_wide_snpsUsed_final_long %&gt;% filter(explained %in% c(&quot;lin1_lin2_NA_lin4&quot;,&quot;lin1_NA_lin3_lin4&quot;,&quot;lin1_NA_lin3_NA&quot;,&quot;lin1_NA_NA_lin4&quot;,&quot;NA_lin2_lin3_NA&quot;,&quot;NA_lin2_NA_lin4&quot;,&quot;NA_lin2_lin3_lin4&quot;)) %&gt;% select(&quot;site&quot;,&quot;explained&quot;) %&gt;% unique() 

sample_relative_wide_snpsUsed_final_long_recombinants &lt;- sample_relative_wide_snpsUsed_final_long %&gt;% select(&quot;site&quot;,&quot;explained&quot;) %&gt;% unique() 


sample_relative_wide_snpsUsed_final_long_recombinants$location &lt;- as.numeric(str_split_fixed(sample_relative_wide_snpsUsed_final_long_recombinants$site, &quot;_&quot;, 4)[,2])
sample_relative_wide_snpsUsed_final_long_recombinants$explained = factor(sample_relative_wide_snpsUsed_final_long_recombinants$explained, levels=c(&quot;lin1&quot;,&quot;lin2&quot;,&quot;lin3&quot;,&quot;lin4&quot;,&quot;not explained&quot;,&quot;lin1_lin2_NA_NA&quot;,&quot;NA_NA_lin3_lin4&quot;,&quot;lin1_lin2_lin3_lin4&quot;,&quot;NA_lin2_lin3_lin4&quot;,&quot;lin1_lin2_NA_lin4&quot;,&quot;lin1_NA_lin3_lin4&quot;,&quot;lin1_NA_lin3_NA&quot;,&quot;lin1_NA_NA_lin4&quot;,&quot;NA_lin2_lin3_NA&quot;,&quot;NA_lin2_NA_lin4&quot;))


sample_relative_wide_snpsUsed_final_long_recombinants$explained_2 &lt;- revalue(sample_relative_wide_snpsUsed_final_long_recombinants$explained, c(&quot;lin1_lin2_NA_NA&quot;=&quot;lin1_lin2_NA_NA=~evolved_after_split&quot;,&quot;NA_NA_lin3_lin4&quot;=&quot;NA_NA_lin3_lin4=~evolved_after_split&quot;,&quot;lin1_lin2_lin3_lin4&quot;=&quot;lin1_lin2_lin3_lin4=polishingERRORs&quot;,&quot;NA_lin2_lin3_lin4&quot;=&quot;NA_lin2_lin3_lin4=lost_in_lin1&quot;,&quot;lin1_NA_lin3_lin4&quot;=&quot;lin1_NA_lin3_lin4=lost_in_lin2&quot;,&quot;lin1_lin2_NA_lin4&quot;=&quot;lin1_lin2_NA_lin4=lost_in_lin3&quot;,&quot;lin1_NA_lin3_NA&quot;=&quot;lin1_NA_lin3_NA=recombination&quot;,&quot;lin1_NA_NA_lin4&quot;=&quot;lin1_NA_NA_lin4=recombination&quot;,&quot;NA_lin2_lin3_NA&quot;=&quot;NA_lin2_lin3_NA=recombination&quot;,&quot;NA_lin2_NA_lin4&quot;=&quot;NA_lin2_NA_lin4=recombination&quot;))

# ggplot(sample_relative_wide_snpsUsed_final_long_recombinants,aes(x=location,color=explained,fill=explained))+geom_density()+facet_wrap(~explained,scales = &quot;free_y&quot;)+theme_classic()
plocation &lt;- ggplot(sample_relative_wide_snpsUsed_final_long_recombinants,aes(x=location,color=explained,fill=explained))+geom_histogram()+facet_wrap(~explained_2,scales = &quot;free_y&quot;)+theme_classic()+labs(title=&quot;location of SNVs coming from different &quot;,x=&quot;genomic location&quot;)+theme(legend.position = &quot;none&quot;)

plocation


  # svg(&quot;~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_explained_by_isolates..svg&quot;,width=7,height=4.5)
png(&quot;~/Desktop/supp_altNucFre_lineages_location.png&quot;, width = 3400, height = 2600,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
plocation
dev.off()

##----------------------
# not explained
##----------------------
table(sample_relative_wide_snpsUsed_final_long$explained)
sample_relative_wide_snpsUsed_final_long_notExplained &lt;- sample_relative_wide_snpsUsed_final_long %&gt;% filter(explained==&quot;not explained&quot;)

ggplot(sample_relative_wide_snpsUsed_final_long_notExplained,aes(x=Median))+geom_density()+facet_wrap(~sample)+theme_classic()


###nomany multiallelic sites
sitesss &lt;- str_split_fixed(sample_relative_wide_snpsUsed_final_long_notExplained$site, &quot;_&quot;, 4)[,2]
length(sitesss)/6
unique(sitesss) %&gt;% length()

sitesss &lt;- str_split_fixed(sample_relative_wide_snpsUsed_final_long$site, &quot;_&quot;, 4)[,2]
length(sitesss) /6
unique(sitesss) %&gt;% length()

##----------------------
# meansss and medianss
##----------------------
library(patchwork)
# mediannssss &lt;-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long[,c(&quot;explained&quot;,&quot;sample&quot;,&quot;Median&quot;)], median, na.rm=TRUE)%&gt;% filter(explained!=&quot;multiple&quot;)
# meanssss &lt;-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long[,c(&quot;explained&quot;,&quot;sample&quot;,&quot;Median&quot;)], median, na.rm=TRUE) %&gt;% filter(explained!=&quot;multiple&quot;)
# 
# mediansplot &lt;- ggplot(mediannssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat=&quot;identity&quot;)+theme_classic()+theme(legend.position = &quot;none&quot;)
# meansssplot &lt;- ggplot(meanssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat=&quot;identity&quot;)+theme_classic()
# 
# mediansplot+meansssplot

##---without zeros
# table(sample_relative_wide_snpsUsed_final_long_woZeros$sample)
mediannssss &lt;-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long_woZeros[,c(&quot;explained&quot;,&quot;sample&quot;,&quot;Median&quot;)], median, na.rm=TRUE)%&gt;% filter(explained!=&quot;multiple&quot;)
meanssss &lt;-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long_woZeros[,c(&quot;explained&quot;,&quot;sample&quot;,&quot;Median&quot;)], median, na.rm=TRUE) %&gt;% filter(explained!=&quot;multiple&quot;)

mediansplot &lt;- ggplot(mediannssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat=&quot;identity&quot;)+theme_classic()+theme(legend.position = &quot;none&quot;)
meansssplot &lt;- ggplot(meanssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat=&quot;identity&quot;)+theme_classic()

mediansplot+meansssplot

##---------------------------------
##lineage abundance
##---------------------------------
ratio_larger &lt;- mediannssss %&gt;% filter(explained==&quot;NA_NA_lin3_lin4&quot;) %&gt;% select(&quot;sample&quot;,&quot;Median&quot;)  %&gt;% dplyr::rename(Overall_ratio = Median) 
ratio_larger_2 &lt;- mediannssss %&gt;% filter(explained==&quot;lin2&quot;) %&gt;% select(&quot;sample&quot;,&quot;Median&quot;)  %&gt;% dplyr::rename(rel_lin2 = Median) 
ratio_larger_3 &lt;- mediannssss %&gt;% filter(explained==&quot;lin3&quot;) %&gt;% select(&quot;sample&quot;,&quot;Median&quot;)  %&gt;% dplyr::rename(rel_lin3 = Median) 
# ratio_larger_4 &lt;- mediannssss %&gt;% filter(explained==&quot;lin4&quot;) %&gt;% select(&quot;sample&quot;,&quot;Median&quot;)  %&gt;% dplyr::rename(rel_lin4 = Median) 

ratio_together_1 &lt;- merge(ratio_larger,ratio_larger_2,by=&quot;sample&quot;,all.x = TRUE)
ratio_together_2 &lt;- merge(ratio_together_1,ratio_larger_3,by=&quot;sample&quot;,all.x = TRUE)
ratio_together_2$lin1 &lt;- (1-ratio_together_2$Overall_ratio)*(1-ratio_together_2$rel_lin2)
ratio_together_2$lin2 &lt;- (1-ratio_together_2$Overall_ratio)*(ratio_together_2$rel_lin2)
ratio_together_2$lin3 &lt;- (ratio_together_2$Overall_ratio)*(ratio_together_2$rel_lin3)
ratio_together_2$lin4 &lt;- (ratio_together_2$Overall_ratio)*(1-ratio_together_2$rel_lin3)


ratio_larger_4 &lt;- mediannssss %&gt;% filter(explained==&quot;lin3&quot;) %&gt;% select(&quot;sample&quot;,&quot;Median&quot;)  %&gt;% dplyr::rename(rel_lin3 = Median) 


##ratios in Reference sequences
#ther is no lin3 or lin4 specific hits that is why we cannot calculate it like before. 
#lin1 is the reference everything that is not lin1 is lin2. 

REffs2 &lt;- mediannssss %&gt;% filter(sample==&quot;Reference 1&quot;) %&gt;% filter(explained==&quot;lin2&quot;) %&gt;% select(Median) %&gt;% unlist() %&gt;% as.numeric()
ratio_together_final_referenceSamples &lt;- data.frame(sample=&quot;Reference 1&quot;,lin4=0,lin3=0,lin2=REffs2,lin1=1-REffs2)
REffs2 &lt;- mediannssss %&gt;% filter(sample==&quot;Reference 2&quot;) %&gt;% filter(explained==&quot;lin2&quot;) %&gt;% select(Median) %&gt;% unlist() %&gt;% as.numeric()
ratio_together_final_referenceSamples2 &lt;- data.frame(sample=&quot;Reference 2&quot;,lin4=0,lin3=0,lin2=REffs2,lin1=1-REffs2)

# ratio_together_final_referenceSamples &lt;- 

# ratio_together_2$Unknown &lt;- 1-(ratio_together_2$lin1+ratio_together_2$lin2+ratio_together_2$lin3+ratio_together_2$lin4)

# ratio_together_final &lt;- ratio_together_2 %&gt;% select(sample,lin1,lin2,lin3,lin4) %&gt;% gather(.,lineage,&quot;Relative abundance&quot;,&quot;lin1&quot;,&quot;lin2&quot;,&quot;lin3&quot;,&quot;lin4&quot;)
ratio_together_final_tmp1 &lt;- ratio_together_2 %&gt;% select(sample,lin4,lin3,lin2,lin1) 

ratio_together_final &lt;- rbind(ratio_together_final_tmp1,ratio_together_final_referenceSamples,ratio_together_final_referenceSamples2) %&gt;% gather(.,lineage,&quot;Relative abundance&quot;,&quot;lin4&quot;,&quot;lin3&quot;,&quot;lin2&quot;,&quot;lin1&quot;)

table(ratio_together_final$sample)



ratio_together_final &lt;- ratio_together_final %&gt;% filter(!sample %in% c(&quot;Reference 1&quot;,&quot;Reference 2&quot;))
ratio_together_final$sample = factor(ratio_together_final$sample, levels=c(&quot;Lyo\n1996&quot;,&quot;Lyo\n2012&quot;,&quot;Lyo\n2014&quot;,&quot;working\nstock&quot;,&quot;starter\nculture\n2012&quot;,&quot;starter\nculture\n2018&quot;,&quot;cheesemaking\nday1&quot;,&quot;cheesemaking\nday2&quot;,&quot;experiment_A&quot;,&quot;experiment_B&quot;,&quot;experiment_C&quot;,&quot;experiment_D&quot;,&quot;experiment_E&quot;,&quot;Reference 1&quot;,&quot;Reference 2&quot;))

ratio_together_final$lineage = factor(ratio_together_final$lineage, levels=(c(&quot;lin4&quot;,&quot;lin3&quot;,&quot;lin2&quot;,&quot;lin1&quot;)))


# colorsss &lt;- c(&quot;#0000FF&quot;,&quot;#6699FF&quot;,&quot;#99CCFF&quot;,&quot;#00FFFF&quot;)
colorsss &lt;- c(&quot;#99CCFF&quot;,&quot;#00FFFF&quot;,&quot;#0000FF&quot;,&quot;#6699FF&quot;)
colorsss &lt;- c(&quot;#99CCFF&quot;,&quot;#00d5d5ff&quot;,&quot;#0000FF&quot;,&quot;#6699FF&quot;)


plot_strain_abundance &lt;- ggplot(ratio_together_final,aes(x=sample,y=`Relative abundance`,color=lineage,fill=lineage))+geom_bar(stat=&quot;identity&quot;)+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+
    labs(x=&quot;&quot;)+scale_color_manual(values=colorsss)+scale_fill_manual(values=colorsss)+theme(legend.position = &quot;none&quot;)
plot_strain_abundance
# png(&quot;~/Desktop/supp_strain_Abundance.png&quot;, width = 2800, height = 2200,res=300)
svg(&quot;~/Desktop/Projects/2019_RMK202_analysis/plot/strain_rel_abundance.svg&quot;,width=6,height=3.5)

# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
plot_strain_abundance
dev.off()


##area plot

plot_strain_abreasss &lt;- ggplot(ratio_together_final,aes(x=sample,y=`Relative abundance`,group=lineage,color=lineage,fill=lineage))+geom_area(alpha=0.5)+geom_point(position = &quot;stack&quot;,size=2)+geom_line(position = &quot;stack&quot;,size=1.25)+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+
    labs(x=&quot;&quot;)+scale_color_manual(values=colorsss)+scale_fill_manual(values=colorsss)+theme(legend.position = &quot;none&quot;)+scale_x_discrete( expand = c(0.01,0.01)) +scale_y_discrete( expand = c(0.01,0.01)) 

plot_strain_abreasss


svg(&quot;~/Desktop/Projects/2019_RMK202_analysis/plot/strain_rel_abundance_02.svg&quot;,width=2.5,height=3.5)

# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
plot_strain_abreasss
dev.off()</code></pre>
<p>##number of sample specific SNPs</p>
<p>Here, I want to see for every starter culture if we have sample
specific SNPs.</p>
<p>In order for this chunk to work, you have to first run the Sterm
strains</p>
<pre class="r"><code>table(sample_relative_safe_Sterm_final$species)
range(sample_relative_safe_Sterm_final$Snps)
table(sample_relative_safe_Sterm_final$site) %&gt;% sort()


cutoffSNPS &lt;- 0.05


StarterUniqueSNPs &lt;- data.frame()
for (Startersss in unique(sample_relative_safe_Sterm_final$culture)) {
  
  tmp1 &lt;- sample_relative_safe_Sterm_final %&gt;% filter(culture==Startersss) %&gt;% mutate(SNPspresent=ifelse(Snps&gt;cutoffSNPS,1,0))
  
  tmp2 &lt;- tmp1 %&gt;% ungroup() %&gt;% group_by(site,species) %&gt;% summarise(numberOFsample=sum(SNPspresent))
  
  
  nsamples &lt;- length(unique(tmp1$sample))
  
  
  tmp2 %&gt;% group_by(numberOFsample,species) %&gt;% summarise(counts=n()) %&gt;% as.data.frame() %&gt;% as_tibble() %&gt;% filter(numberOFsample %in% c(1,nsamples-1))%&gt;% group_by(species) %&gt;% summarise(totalNumberOfsingleSNPs=sum(counts)) 
  
  
  namesSites_01 &lt;- tmp2 %&gt;% filter(numberOFsample %in% c(1)) %&gt;% pull(site)
  namesSites_allbut1 &lt;- tmp2 %&gt;% filter(numberOFsample %in% c(nsamples-1)) %&gt;% pull(site)

  tmp3 &lt;- tmp1 %&gt;% dplyr::filter(site %in% namesSites_01) %&gt;% filter(SNPspresent==1)%&gt;% mutate(group=&quot;StarterUnique&quot;)
  tmp4 &lt;- tmp1 %&gt;% dplyr::filter(site %in% namesSites_allbut1) %&gt;% filter(SNPspresent==0) %&gt;% mutate(group=&quot;StarterUnique&quot;)
  
  tmp6 &lt;- data.frame()
  for (samplesss in unique(tmp1$sample)) {
    tmp5 &lt;- tmp1 %&gt;% filter(sample==samplesss) %&gt;% head(1)%&gt;% mutate(group=&quot;dummy&quot;)%&gt;% mutate(site=&quot;dummy&quot;) %&gt;% mutate(site=&quot;dummy&quot;) %&gt;% filter(SNPspresent==1)
    tmp6 &lt;- rbind(tmp6,tmp5)
  }
  
  
  
 StarterUniqueSNPs &lt;-  rbind(StarterUniqueSNPs,tmp3,tmp4,tmp6)
  
}


###------------------------plot

coregenomesize=130000


StarterUniqueSNPs_summary &lt;- StarterUniqueSNPs %&gt;% group_by(species,sample,culture) %&gt;% summarise(UNIQUEsnps=n()) %&gt;% mutate(relativeSNPs=UNIQUEsnps/coregenomesize) %&gt;% mutate(UNIQUEsnps=ifelse(UNIQUEsnps==1,1,UNIQUEsnps-1))


# 20/coregenomesize
tmp1 &lt;- StarterUniqueSNPs %&gt;% filter(sample==&quot;305-starter 1&quot;)

# ggplot()+geom_density(data=tmp1,aes(x=Snps))


# plotbox &lt;- ggplot()+geom_boxplot(data=StarterUniqueSNPs_summary,aes(x=culture,group=culture,color=species,fill=species,y=UNIQUEsnps))+scale_y_continuous(trans = &quot;log2&quot;,breaks=c(1,10, 100,1000,10000),name=&quot;Unique SNPs&quot;, sec.axis = sec_axis(~./coregenomesize, name=&quot;Relative number of sites [%]&quot;,breaks=c(1e-5,1e-4,1e-3,1e-2)))+facet_grid(~species)




 ggplot(StarterUniqueSNPs_summary,aes(x=sample,color=species,fill=species,group=species))+geom_point(aes(y=UNIQUEsnps))+geom_line(aes(y=UNIQUEsnps))+facet_grid(~culture,scales = &quot;free_x&quot;, space = &quot;free&quot;)+theme_classic()+theme(legend.position = &quot;none&quot;,axis.text.x =element_blank(),axis.title.x = element_blank(),axis.ticks.x=element_blank())+
   scale_fill_manual(values=c(&quot;blue&quot;,&quot;pink&quot;))+scale_color_manual(values=c(&quot;#fdc2b5ff&quot;,&quot;#007fa6ff&quot;)) +scale_y_continuous(trans = &quot;log2&quot;,breaks=c(1,10, 100,1000,10000),name=&quot;Unique SNPs&quot;, sec.axis = sec_axis(~./coregenomesize, name=&quot;Relative number of sites [%]&quot;,breaks=c(1e-5,1e-4,1e-3,1e-2)))

 
 
 ###-----------------------with quartile info
 
 tmps &lt;- StarterUniqueSNPs_summary %&gt;% group_by(culture,species) %&gt;% summarise(meanSNPs=mean(UNIQUEsnps),sdSNPs=sd(UNIQUEsnps),medianSNPs=median(UNIQUEsnps),first=quantile(UNIQUEsnps, 0.25),third=quantile(UNIQUEsnps, 0.75))

 
 StarterUniqueSNPs_range &lt;- data.frame()
for (Startersss in unique(sample_relative_safe_Sterm_final$culture)) {
  
  tmp1 &lt;- sample_relative_safe_Sterm_final %&gt;% filter(culture==Startersss) %&gt;% pull(sample) %&gt;% as.character() %&gt;% unique() 
  
  tmp2 &lt;- tmp1 %&gt;% head(1)
  tmp3 &lt;- tmp1 %&gt;% tail(1)
  
  tmp5 &lt;- data.frame()
  for (speciesss in rev(unique(sample_relative_safe_Sterm_final$species))) {
    
    tmp4 &lt;- tmps %&gt;% filter(culture==Startersss) %&gt;% filter(species==speciesss) %&gt;% mutate(sampleFirst=tmp2)%&gt;% mutate(samplelast=tmp3)
    
    tmp5 &lt;- rbind(tmp5,tmp4)
    
  }

  StarterUniqueSNPs_range &lt;- rbind(StarterUniqueSNPs_range, tmp5)
  tmp5
  
}
 
 
 
 
 # ggplot()+geom_point(data=StarterUniqueSNPs_summary,aes(x=sample,color=species,fill=species,group=species,y=UNIQUEsnps))+geom_line(data=StarterUniqueSNPs_summary,aes(x=sample,color=species,fill=species,group=species,y=UNIQUEsnps))+
 #   + geom_rect(data=StarterUniqueSNPs_range, (aes(xmin=sampleFirst, xmax=samplelast, 
 #                   ymin=first,ymax=third,color=species,fill=species,group=species)))+
 #   facet_grid(~culture,scales = &quot;free_x&quot;, space = &quot;free&quot;)+
 #   theme_classic()+theme(legend.position = &quot;none&quot;,axis.text.x =element_blank(),axis.title.x = element_blank(),axis.ticks.x=element_blank())+
 #   scale_fill_manual(values=c(&quot;blue&quot;,&quot;pink&quot;))+scale_color_manual(values=c(&quot;#fdc2b5ff&quot;,&quot;#007fa6ff&quot;)) +scale_y_continuous(trans = &quot;log2&quot;,breaks=c(1,10, 100,1000,10000),name=&quot;Unique SNPs&quot;, sec.axis = sec_axis(~./coregenomesize, name=&quot;Relative number of sites [%]&quot;,breaks=c(1e-5,1e-4,1e-3,1e-2)))
library(ggnewscale)
 
 values=c(&quot;#fdc2b5ff&quot;,&quot;#007fa6ff&quot;)
alpha(c(&quot;#fdc2b5ff&quot;,&quot;#007fa6ff&quot;), alpha = 0.2)

 levels(StarterUniqueSNPs_range$species) &lt;- c(levels(StarterUniqueSNPs_range$species), rev(c(&quot;S. thermophilus&quot;, &quot;L. delbrueckii&quot;)))

 lighterversionOFblue &lt;- c(&quot;#79dfff&quot;,&quot;#26ccff&quot;,&quot;#5ad8ff&quot;)
 # &quot;#007fa6ff&quot;,&quot;#fdc2b5ff&quot;

NumberOFsnpsPlot &lt;-  ggplot()+geom_point(data=StarterUniqueSNPs_summary,aes(x=sample,color=species,group=species,y=UNIQUEsnps))+geom_line(data=StarterUniqueSNPs_summary,aes(x=sample,color=species,group=species,y=UNIQUEsnps))+
   scale_color_manual(values=c(&quot;#fdc2b5ff&quot;,&quot;#007fa6ff&quot;))+
   # new_scale_fill()+
   geom_rect(aes(xmin = sampleFirst, xmax = samplelast, ymin = first, ymax = third, fill = species),
            alpha=0.3,
            data = StarterUniqueSNPs_range) +
   scale_fill_manual(values=c(&quot;#fdc2b5ff&quot;,&quot;#5ad8ff&quot;))+
   facet_grid(~culture,scales = &quot;free_x&quot;, space = &quot;free&quot;)+
   theme_classic()+theme(legend.position = &quot;none&quot;,axis.text.x =element_blank(),axis.title.x = element_blank(),axis.ticks.x=element_blank())+ scale_y_continuous(trans = &quot;log2&quot;,breaks=c(1,10, 100,1000,10000),name=&quot;Sample-specific SNPs&quot;, sec.axis = sec_axis(~./coregenomesize, name=&quot;Relative sample-specific SNPs [%]&quot;,breaks=c(1e-5,1e-4,1e-3,1e-2)))

NumberOFsnpsPlot




pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Unique_SNPs_01.pdf&quot;, width = 10, height = 3)

NumberOFsnpsPlot


dev.off()

StarterUniqueSNPs_range %&gt;% group_by(species) %&gt;% summarise(median=median(medianSNPs),sd=sd(medianSNPs)) %&gt;% mutate(relativemedian=coregenomesize/median)%&gt;% mutate(relativesd=coregenomesize/sd)
median(StarterUniqueSNPs_range$medianSNPs)</code></pre>
<p>##not explained SNPs</p>
<pre class="r"><code>FAMnames &lt;- colnames(sample_relative_wide)[!colnames(sample_relative_wide) %&gt;% grepl(&quot;-starter&quot;,.)][-1:-8]

FAMstrainsexist &lt;- sample_relative_wide %&gt;% select(FAMnames) %&gt;% rowSums()#%&gt;% mutate(sumss=rowSums())

starternames &lt;- colnames(sample_relative_wide)[colnames(sample_relative_wide) %&gt;% grepl(&quot;-starter&quot;,.)]
startersexist &lt;- sample_relative_wide %&gt;% select(starternames) %&gt;% rowSums()#%&gt;% mutate(sumss=rowSums())


tmp1 &lt;- sample_relative_wide %&gt;% select(&quot;site&quot; ,  &quot;species&quot;  , &quot;effect&quot; , &quot;significance&quot;,&quot;geneName&quot; , &quot;COG_Functional_Category&quot;, &quot;length&quot;   ,&quot;core&quot; )


sample_relative_wide_explained &lt;- cbind(tmp1,startersexist,FAMstrainsexist) %&gt;% mutate(category=ifelse(startersexist&gt;0&amp;FAMstrainsexist&gt;0,&quot;both&quot;,ifelse(startersexist&gt;0&amp;FAMstrainsexist==0,&quot;only metaG&quot;,&quot;onlyFAM&quot;))) %&gt;% as_tibble()

table(sample_relative_wide_explained$category)

sample_relative_wide_explained %&gt;% group_by(species,core,category) %&gt;% dplyr::summarise(counts=n())


sample_relative_wide_explained_small &lt;- sample_relative_wide_explained %&gt;% select(site,category)


###------------------------plot SNPs nonexplained 


sample_relative_safe_Sterm_final_wide_filtered_long_woZero_completed &lt;- left_join(sample_relative_safe_Sterm_final_wide_filtered_long_woZero,sample_relative_wide_explained_small)



###------------------------plot SNPs nonexplained 

sample_relative_safe_Sterm_final_wide_filtered_long_woZero_onlyMETAG &lt;- sample_relative_safe_Sterm_final_wide_filtered_long_woZero_completed %&gt;% filter(category==&quot;only metaG&quot;)


  ponlyMetaG &lt;- ggplot(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_onlyMETAG,aes(x=sample,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.3, alpha=.1)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = &quot;free_x&quot;, space = &quot;free&quot;)+
            facet_grid(species~culture,scales = &quot;free_x&quot;, space = &quot;free&quot;)+
            # ggforce::facet_row(vars(culture)~species,scales = &quot;free_x&quot;, space = &quot;free&quot;)+
            # ggforce::facet_col(vars(species),scales = &quot;free_x&quot;, space = &quot;free&quot;)+
     labs(title = &quot;not Explained&quot;,
         x=&quot; &quot;,
         y=&quot;Alternative allele frequency&quot;)+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = &quot;transparent&quot;), # all rectangles      #axis.text.x = element_blank(),
          legend.position=&quot;none&quot;,
          panel.spacing = unit(1.2, &quot;lines&quot;)
  )
  
  ponlyMetaG
  
  

###------------------------plot SNPs nonexplained 
table(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_completed$category)
  sample_relative_safe_Sterm_final_wide_filtered_long_woZero_both &lt;- sample_relative_safe_Sterm_final_wide_filtered_long_woZero_completed %&gt;% filter(category==&quot;both&quot;)

  ponlyBTOH &lt;- ggplot(sample_relative_safe_Sterm_final_wide_filtered_long_woZero_both,aes(x=sample,y=Snps,group=site,color=species,fill=species))+ geom_line(size=0.3, alpha=.1)+ 
    # facet_wrap(culture~.,nrow = 11)+
        # facet_wrap(culture~.,ncol = 11,scales = &quot;free_x&quot;, space = &quot;free&quot;)+
            facet_grid(species~culture,scales = &quot;free_x&quot;, space = &quot;free&quot;)+
            # ggforce::facet_row(vars(culture)~species,scales = &quot;free_x&quot;, space = &quot;free&quot;)+
            # ggforce::facet_col(vars(species),scales = &quot;free_x&quot;, space = &quot;free&quot;)+
     labs(title = &quot;Explained by isolates&quot;,
         x=&quot; &quot;,
         y=&quot;Alternative allele frequency&quot;)+
      theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = &quot;transparent&quot;), # all rectangles      #axis.text.x = element_blank(),
          legend.position=&quot;none&quot;,
          panel.spacing = unit(1.2, &quot;lines&quot;)
  )
  
ponlyBTOH
  
ponlyMetaG+ponlyBTOH+plot_layout(ncol=1)

png(paste0(plot_path,&quot;SNPs_plot_suppl.png&quot;), width = 4000, height = 3500,res=400)

 ponlyMetaG+ponlyBTOH+plot_layout(ncol=1)
dev.off()
  
##----------------QUNATIFY
  
tmp1 &lt;- sample_relative_safe_Sterm_final_wide_filtered_long_woZero_completed %&gt;% group_by(species) %&gt;% dplyr::summarise(total=n())

 sample_relative_safe_Sterm_final_wide_filtered_long_woZero_completed %&gt;% group_by(species,category) %&gt;% dplyr::summarise(counts=n()) %&gt;% left_join(.,tmp1) %&gt;% mutate(percent=100*(counts/total))</code></pre>
</div>
</div>
</div>
<div id="genome-analysis" class="section level1">
<h1>Genome analysis</h1>
<p>Here, I analyse the different reference genome sequences</p>
<div id="data-preperation-1" class="section level2">
<h2>Data preperation</h2>
<div id="ont" class="section level3">
<h3>ONT</h3>
<pre class="bash"><code>
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x1_T43_Seq1_VS_fq
cp /home/vincent/Downloads/fastq_pass.tar.gz.zip /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x1_T43_Seq1_VS_fq
cd /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x1_T43_Seq1_VS_fq
openssl enc -aes-256-cbc -md md5 -d -in fastq_pass.tar.gz.enc &gt; fastq_pass.tar.gz
tar xvzf fastq_pass.tar.gz

##------------------------2
rm  /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq
cp /home/vincent/Downloads/&#39;fastq_pass.tar.gz(1).enc&#39; /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq/
cd /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq/
openssl enc -aes-256-cbc -md md5 -d -in fastq_pass.tar.gz.enc &gt; fastq_pass.tar.gz
tar xvzf fastq_pass.tar.gz

scp /home/vincent/Downloads/fastq_pass.tar.gz.enc vincent@130.223.49.145:/data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200527_x2_T43_Seq2_VS_fq/

wget https://campuscloud.unibe.ch/ssf/s/readFile/folderEntry/64630263/02dc805872bb70a00172d615fd105f8f/1592660049000/last/fastq_pass.tar.gz.enc
##------------------------3
rm  /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x3_ES43_EXP02_Seq3_VS_fq

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x3_ES43_EXP02_Seq3_VS_fq
cp /home/vincent/Downloads/fastq_pass.tar.gz.enc /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x3_ES43_EXP02_Seq3_VS_fq/
cd /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x3_ES43_EXP02_Seq3_VS_fq/
#unzip fastq_pass.tar.gz.zip
openssl enc -aes-256-cbc -md md5 -d -in fastq_pass.tar.gz.enc &gt; fastq_pass.tar.gz
tar xvzf fastq_pass.tar.gz

##------------------------2

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x4_ES43_EXP02_Seq4_VS
cp /home/vincent/Downloads/fastq_pass.tar.gz.enc /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x4_ES43_EXP02_Seq4_VS/
cd /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/20200605_x4_ES43_EXP02_Seq4_VS/
#unzip fastq_pass.tar.gz.zip
openssl enc -aes-256-cbc -md md5 -d -in fastq_pass.tar.gz.enc &gt; fastq_pass.tar.gz
tar xvzf fastq_pass.tar.gz
</code></pre>
<div id="rename-and-merge" class="section level4">
<h4>rename and merge</h4>
<pre class="bash"><code>
for runnsss in $(echo &quot;2&quot;)
do

longRunssss=$(echo -e &quot;_Seq&quot;${runnsss}&quot;_VS&quot;)
directory=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/ |grep ${longRunssss})
echo &quot;=============================&quot;
echo ${longRunssss}
for samplezzz in $(grep &quot;${longRunssss}&quot; /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |cut -f 3)
do

rm -r  /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${samplezzz}/
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${samplezzz}/
barcode=$(grep &quot;${samplezzz}&quot; /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |cut -f 4|sed &#39;s/RB //g&#39;)
echo &quot;----------------------&quot;
echo -e &quot;barcode :&quot; ${barcode} &quot; &amp; sample: &quot; ${samplezzz}
cat /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/${directory}/fastq_pass/barcode${barcode}/*fastq &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${samplezzz}/${samplezzz}.fastq

done #samplezzz

done #runsss



for runnsss in $(echo &quot;2&quot;)
do

longRunssss=$(echo -e &quot;_Seq&quot;${runnsss}&quot;_VS&quot;)
directory=$(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/gz/ |grep ${longRunssss})
echo &quot;=============================&quot;
echo ${longRunssss}
for samplezzz in $(grep &quot;${longRunssss}&quot; /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |cut -f 3)
do

barcode=$(grep &quot;${samplezzz}&quot; /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |cut -f 4|sed &#39;s/RB //g&#39;)
echo &quot;----------------------&quot;
echo -e &quot;barcode :&quot; ${barcode} &quot; &amp; sample: &quot; ${samplezzz}
grep &quot;^@&quot; -c /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${samplezzz}/${samplezzz}.fastq

done #samplezzz

done #runsss

###---------------------------------------------
#20210310
###---------------------------------------------



for samplessss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/20210309_20210222_ES59_FASTQ/20210222_ES59_Somerville_FASTQ/sequencing_summary/sample_summary.txt) ;do

samplezzz=$(awk -F &quot;\t&quot; -v sams=&quot;$samplessss&quot; &#39;{OFS=&quot;\t&quot;}{if($2==sams)print $1}&#39; /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/20210309_20210222_ES59_FASTQ/20210222_ES59_Somerville_FASTQ/sequencing_summary/sample_summary.txt)
echo -e &quot;make barcode&quot;${samplessss} &quot; into &quot;${samplezzz}
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/

cat /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/20210309_20210222_ES59_FASTQ/20210222_ES59_Somerville_FASTQ/fastq_pass/barcode${samplessss}/*fastq &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/${samplezzz}.fastq

done

###---------------------------------------------
#special case
###---------------------------------------------

sample=24749
cat /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/20210309_20210222_ES59_FASTQ/20210222_ES59_Somerville_FASTQ/fastq_pass/barcode06/*fastq /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/24749/24749.fastq &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/${sample}.fastq
</code></pre>
</div>
<div id="flye-assembly" class="section level4">
<h4>flye assembly</h4>
<pre class="bash"><code>
##===========
##Assembly plasmid
#samples 20210310
##===========

    rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz/
    mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz/
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep &quot;.fastq$&quot;|sed &#39;s/.fastq//g&#39; |sed &#39;1d&#39;)
  do
  
#  ll /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/
  

  echo &quot;==============&quot;
  echo ${sample}
  echo &quot;==============&quot;

/home/vincent/apps/Filtlong/bin/filtlong --min_length 6000 --keep_percent 90 --target_bases 500000000 --mean_q_weight 10 /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/${sample}.fastq |gzip &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz//${sample}_filtered.fastq.gz


  done

###---------------------------------------------
#special case
###---------------------------------------------
sample=24732

sample=24749

/home/vincent/apps/Filtlong/bin/filtlong --min_length 40000 --keep_percent 90 --target_bases 100000000 --mean_q_weight 10 /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/${sample}.fastq |gzip &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz//${sample}_filtered.fastq.gz



##===========
##Assembly plasmid
#samples 20210310
##===========
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep &quot;.fastq$&quot;|sed &#39;s/.fastq//g&#39; |sed &#39;1d&#39; )
  do
  
#  ll /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/
  

    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/
  echo &quot;==============&quot;
  echo ${sample}
  echo &quot;==============&quot;

   #python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
  #    /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/${sample}.fastq \
  #    --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/
      

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 1800000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz//${sample}_filtered.fastq.gz \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/
      
  done
  
  
  
#mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/09_flye/assembly/24749_02/30-contigger/
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/09_flye/assembly/24749_02/30-contigger/graph_final.gfa ~/Desktop/Projects/2020_StarterCultureDiversity/09_flye/assembly/24749_02/30-contigger/
  

  ##===========
##length
##===========
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep &quot;.fastq$&quot;|sed &#39;s/.fastq//g&#39; )
#for sample in $(grep &quot;20200527_x2_T43EXP01_Seq2_VS&quot; /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3 )
  do

  echo &quot;==============&quot;
  echo ${sample}

 # seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/assembly.fasta
cat /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/30-contigger/*stats*

  done

sample=24740
##===========
##Assembly old samples
##===========
for sample in $(grep &quot;20200527_x2_T43EXP01_Seq2_VS&quot; /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3|sed &#39;1d&#39; )
  do
  
  sample=24740

    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo &quot;==============&quot;
  echo ${sample}
  echo &quot;==============&quot;

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
  done
  
  
  ##===========
##length
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#for sample in $(grep &quot;20200527_x2_T43EXP01_Seq2_VS&quot; /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3 )
  do

  echo &quot;==============&quot;
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
  #==============================================================================================================
  ##----------------------------
  ###assemblies with problems
  ##----------------------------
    #==============================================================================================================
    
    ##-------------------
    ##count reads
    ##-------------------
    for sample in $(echo &quot;24855 24773 24765 24742 24740 24738 24736 24726&quot;)
  do
   echo &quot;==============&quot;
  echo ${sample}
  
  grep &quot;^@&quot; -c /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq
done

    ##-------------------
    ##too few reads
    ##-------------------
      for sample in $(echo &quot;24773 24765&quot;)
    do
    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo &quot;==============&quot;
  echo ${sample}
  echo &quot;==============&quot;

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
    
    done
  ##-------------------
    ##assembly length
    ##-------------------
  for sample in $(echo &quot;24855 24742 24740 24738 24736 24726&quot;)
  do
   echo &quot;==============&quot;
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
  
  
    ##-------------------
    ##too many reads
    ###-----&gt;&gt;&gt;&gt;here I filter for only the longest reads
    ##-------------------
    
    
    
        for sample in $(echo &quot;24855 24742 24740 24738 24736 24726&quot;)
        do
        
        rm /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz
         
    
/home/vincent/apps/Filtlong/bin/filtlong --min_length 6000 --keep_percent 90 --target_bases 500000000 --mean_q_weight 10 /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq | gzip &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz

#mv /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq.gz

        
        done
    
    
    ##-------------------
    ##too many reads
    ##assembly
    ##-------------------
    
    
      for sample in $(echo &quot;24855 24742 24740 24738 24736 24726&quot;)
        do
        
       
    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo &quot;==============&quot;
  echo ${sample}
  echo &quot;==============&quot;

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
        
        done
    
    
      ##-------------------
    ##check
    ##-------------------
    
  for sample in $(echo &quot;24855 24773 24765 24742 24740 24738 24736 24726&quot;)
  do
   echo &quot;==============&quot;
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
    
    
    
    
  for sample in $(echo &quot;24855 24773 24765 24742 24740 24738 24736 24726&quot;)
  do
   echo &quot;==============&quot;
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
  sample=24740

    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo &quot;==============&quot;
  echo ${sample}
  echo &quot;==============&quot;

   python /home/vincent/Flye/bin/flye --threads 37  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
  done
  
    #==============================================================================================================
  ##----------------------------
  ###assemblies partially problematic
  ##optimise!!!
  ##change Minimum overlap 
  ##----------------------------
    #==============================================================================================================
    
      for sample in $(echo &quot;24751 24731 24743 24758 24729 24795 24790 24777 24762 24759 24749 24731&quot;)
  do
   echo &quot;==============&quot;
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
    
   ##-------------------
    ##count reads
    ##-------------------
    for sample in $(echo &quot;24751 24731 24743 24758 24729 24795 24790 24777 24762 24759 24749 24731&quot;)
  do
   echo &quot;==============&quot;
  echo ${sample}
  
  grep &quot;^@&quot; -c /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq
done

 ##-------------------
    ##filtlong
    ##-------------------
    for sample in $(echo &quot;24751 24731 24743 24758 24729 24795 24790 24777 24762 24759 24749 24731&quot;)
  do
   echo &quot;==============&quot;
  echo ${sample}
   rm /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz
         
    
/home/vincent/apps/Filtlong/bin/filtlong --min_length 6000 --keep_percent 90 --target_bases 150000000 --mean_q_weight 10 /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq | gzip &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz

done


##-------------------
    ##too many reads
    ##assembly
    ##-------------------
    
    
    for sample in $(echo &quot;24751 24731 24743 24758 24729 24795 24790 24777 24762 24759 24749 24731&quot;)
        do
        
       
    rm -r /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
    mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}
  echo &quot;==============&quot;
  echo ${sample}
  echo &quot;==============&quot;

   python /home/vincent/Flye/bin/flye --min-overlap 4000 --threads 5  --iterations 5 --genome-size 2100000 --nano-raw \
      /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz \
      --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
      
        
        done
        
##-------------------
    ##final samples
    ##------------------   

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
grep -v &quot;24743&quot;|
grep -v &quot;24729&quot;|
grep -v &quot;24795&quot;|
grep -v &quot;24794&quot;|
grep -v &quot;24762&quot;|
grep -v &quot;24751&quot;|
grep -v &quot;24773&quot;|
grep -v &quot;24765&quot;)
#for sample in $(grep &quot;20200527_x2_T43EXP01_Seq2_VS&quot; /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3 )
  do

  echo &quot;==============&quot;
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done
</code></pre>
<p>check circlatrity</p>
<pre class="bash"><code>rm /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt
echo -e &quot;sample\tcircular_contigs\tlargestContigSize\tCoverage\tCircularity&quot; &gt; /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
  do
  echo &quot;=====================================================&quot;
  echo $sample
  circularsss=$(grep &quot;#&quot; -v /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/30-contigger/contigs_stats.txt |awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{if($4==&quot;Y&quot;)print $0}&#39;| wc -l )
grep &quot;#&quot; -v /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/30-contigger/contigs_stats.txt |sort -t$&#39;\t&#39; -k2 -nr |head -1 |awk -F &quot;\t&quot; -v cics=&quot;$circularsss&quot; -v samss=&quot;$sample&quot; &#39;{OFS=&quot;\t&quot;}{print samss,cics,$2,$3,$4}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt
done


##-------------------------------------------
#new
##-------------------------------------------
rm /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt
echo -e &quot;sample\tcircular_contigs\tlargestContigSize\tCoverage\tCircularity&quot; &gt; /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt

for sample in $( echo &quot;S24750
S24739
S13496
S24738
S13499
S24730
S24726
S24728
S13498
S24734
S24737
S24853
S24854
S24855
S24742
S13494
S24758
S24735
S24736
S24724
S24748
S24740
S24745
L24756
L24786
L24790
S24732
S24733
S24743
S24746
S24747
S24749
L24754
L24798&quot; |sort|uniq)
do
genomesss=$(echo ${sample}| sed &#39;s/S//g&#39;| sed &#39;s/L//g&#39;)
filesss=$(ls /data/Project/2020_StarterCultureDiversity/09_flye/assembly/| grep &quot;$genomesss&quot;)

  circularsss=$(grep &quot;#&quot; -v /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${filesss}/30-contigger/contigs_stats.txt |awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{if($4==&quot;Y&quot;)print $0}&#39;| wc -l )
grep &quot;#&quot; -v /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${filesss}/30-contigger/contigs_stats.txt |sort -t$&#39;\t&#39; -k2 -nr |head -1 |awk -F &quot;\t&quot; -v cics=&quot;$circularsss&quot; -v samss=&quot;$sample&quot; &#39;{OFS=&quot;\t&quot;}{print samss,cics,$2,$3,$4}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt


done
</code></pre>
<p>move local</p>
<pre class="bash"><code>
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/99_log/ONT_assembly_stats.txt ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/ONT_assembly_stats.txt
</code></pre>
</div>
<div id="circlator-1" class="section level4">
<h4>circlator 1</h4>
<pre class="bash"><code>

##===========
##circlator
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
  do
  

  echo &quot;==============&quot;
  echo ${sample}

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/circlator/${sample}
#mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/circlator/${sample}

mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/

#cat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R1_val_1.fq.gz \
#/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R2_val_2.fq.gz &gt; \
#/data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/${sample}.fq.gz


circlator all /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta /data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/${sample}.fq.gz /data/Project/2020_StarterCultureDiversity/09_flye/circlator/${sample}


  done
  

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/</code></pre>
</div>
<div id="ont-polishing" class="section level4">
<h4>ONT polishing</h4>
<pre class="bash"><code>

###===========================
###-----------------
##cleanup
###-----------------
###===========================    
     
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv|
grep -v &quot;24724&quot;|
grep -v &quot;24726&quot;|
grep -v &quot;24728&quot;|
grep -v &quot;24730&quot;|
grep -v &quot;24731&quot;|
grep -v &quot;24734&quot;|
grep -v &quot;24736&quot;|
grep -v &quot;24737&quot;|
grep -v &quot;24738&quot;|
grep -v &quot;24739&quot;|
grep -v &quot;24740&quot;)

for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep &quot;.fastq$&quot;|sed &#39;s/.fastq//g&#39; )
  do

name=$(echo $sample)
echo ${name}
done
#sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}
reference_fasta=/data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/assembly.fasta
#ont_reads=/data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq
ont_reads=/data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz//${sample}_filtered.fastq.gz
#reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
echo &quot;=========================================&quot;
echo $sample
echo &quot;=========================================&quot;

echo $reference_fasta
echo &quot;---------------------------&quot;
     
###===========================
###-----------------
##graphmap mapping
###-----------------
###===========================      

name=first
num=1

for name in first second third fourth
do 
#rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/
#mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc


#rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam
rm -r $Overall_output_directory
#rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2Final*

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 



done

done #sample


###===========================
###-----------------
##samples
###-----------------
###===========================    
     
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
grep -v &quot;24743&quot;|
grep -v &quot;24729&quot;|
grep -v &quot;24795&quot;|
grep -v &quot;24794&quot;|
grep -v &quot;24762&quot;|
grep -v &quot;24751&quot;|
grep -v &quot;24773&quot;|
grep -v &quot;24765&quot;)
#for sample in $(grep &quot;20200527_x2_T43EXP01_Seq2_VS&quot; /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv  |cut -f 3 )
  do

  echo &quot;==============&quot;
  echo ${sample}

  seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/assembly.fasta


  done

###===========================
###-----------------
##Racon polishing with graphmap
###-----------------
###===========================    
     
     
     24724  24726  24728  24730  24731  24734  24736  24737  24738  24739  24740
##------------------define variables------------------

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
grep -v &quot;24743&quot;|
grep -v &quot;24729&quot;|
grep -v &quot;24795&quot;|
grep -v &quot;24794&quot;|
grep -v &quot;24762&quot;|
grep -v &quot;24751&quot;|
grep -v &quot;24773&quot;|
grep -v &quot;24765&quot;|
grep -v &quot;24724&quot;|
grep -v &quot;24726&quot;|
grep -v &quot;24728&quot;|
grep -v &quot;24730&quot;|
grep -v &quot;24731&quot;|
grep -v &quot;24734&quot;|
grep -v &quot;24736&quot;|
grep -v &quot;24737&quot;|
grep -v &quot;24738&quot;|
grep -v &quot;24739&quot;|
grep -v &quot;24740&quot;)

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
grep -v &quot;24743&quot;|
grep -v &quot;24729&quot;|
grep -v &quot;24795&quot;|
grep -v &quot;24794&quot;|
grep -v &quot;24762&quot;|
grep -v &quot;24751&quot;|
grep -v &quot;24773&quot;|
grep -v &quot;24765&quot;)

for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep &quot;.fastq$&quot;|sed &#39;s/.fastq//g&#39; )
  do

name=$(echo $sample)
#sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}
reference_fasta=/data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}_02/assembly.fasta
#ont_reads=/data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq
ont_reads=/data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/filtered/gz//${sample}_filtered.fastq.gz
#reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
echo &quot;=========================================&quot;
echo $sample
echo &quot;=========================================&quot;

echo $reference_fasta
echo &quot;---------------------------&quot;
     
###===========================
###-----------------
##graphmap mapping
###-----------------
###===========================      

name=first
num=1

for name in first second third fourth
do 
rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/
mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc

#name=first

graphmap align -t ${threads} \
  -r $reference_fasta \
  -d $ont_reads \
  -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam 

#  -d $ont_reads | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam - 

  
###===========================
###-----------------
##racon polishing
###-----------------
###===========================   

#rm -r$Overall_output_directory/0${num}_${name}Racon_graphmap/assembly

mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly

samtools view -bS $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam -

#samtools sort -@${threads} -O SAM $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam -

#samtools view -S -b $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam &gt; $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam 

/home/vincent/apps/racon/new/racon/build/bin/racon \
 -t ${threads} \
  $ont_reads \
  $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam \
  $reference_fasta &gt; \
  $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta
  


samtools index $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam 

qualimap bamqc -bam $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam \
  -outdir $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc --java-mem-size=40G 

#rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam
rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2Final*

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 



done

done #sample
##--------------------cleanup


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz
#reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
name=first
num=1

for name in first second third fourth
do 



#rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 

done
#done # loop for 1AB and 2AB samples


##--------------------test PROKKA
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
name=fourth
num=4

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes   $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta
  
  grep &quot;DNAa&quot; -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon/PROKKA_*.gff
  
  echo &quot;=======================================================ALT===============================================================&quot;
  grep &quot;DNAa&quot; -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all/PROKKA_*.gff


  


###===========================
###-----------------
##quast
###-----------------
###===========================  

#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz


echo &quot;=========================================&quot;
echo $sample
echo &quot;=========================================&quot;

echo $reference_fasta
echo &quot;---------------------------&quot;
     
    rm -r $Overall_output_directory/quastwithRaconONT/
    mkdir -p $Overall_output_directory/quastwithRaconONT
    quast.py \
      -t ${threads} -l &quot;FlyeAssembly,firstONTracon,secondONTracon,thirdONTracon&quot; \
      -R $Overall_output_directory/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta \
      $reference_fasta \
      $Overall_output_directory/01_firstRacon_graphmap/assembly/firstRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/02_secondRacon_graphmap/assembly/secondRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/03_thirdRacon_graphmap/assembly/thirdRacon_NWC_2Assembly.fasta \
      -o $Overall_output_directory/quastwithRaconONT
 
 
 ##-------------------making MultiQC output
 rm $Overall_output_directory/logs_MultiQC_raconONT.txt
 
 echo $Overall_output_directory/quastwithRaconONT/report.tsv &gt;&gt; $Overall_output_directory/logs_MultiQC_raconONT.txt
 
 num=1

for name in first second third fourth
do 

echo $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc &gt;&gt; $Overall_output_directory/logs_MultiQC_raconONT.txt
num=$((num+1)) 

done ##loop for name
 
#done #loop for sample
###===========================
###-----------------
##MultiQC
###-----------------
###===========================  


#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do
name=di_K2_6h
sample=di_K2_6h
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${sample}/
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/assembly.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz

echo &quot;=========================================&quot;
echo $sample
echo &quot;=========================================&quot;



#done #loop for sample
 
rm -r $Overall_output_directory/MultiQCRaconONT/
mkdir -p $Overall_output_directory/MultiQCRaconONT/
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list   $Overall_output_directory/logs_MultiQC_raconONT.txt \
    -o $Overall_output_directory/MultiQCRaconONT/



###--------------on local

name=di_K2_6h
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/quastwithRaconONT/report.html ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_quast_report.html

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/MultiQCRaconONT/multiqc_report.html ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report.html
</code></pre>
</div>
<div id="illumina-polishing-freebayes" class="section level4">
<h4>Illumina polishing freebayes</h4>
<p>Now we polish with the high quality Illumina reads</p>
<pre class="bash"><code>

###-----------------
##Illuminamapping rawReads to circlator genome
###-----------------
sample=24758
   for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
grep -v &quot;24743&quot;|
grep -v &quot;24729&quot;|
grep -v &quot;24795&quot;|
grep -v &quot;24794&quot;|
grep -v &quot;24762&quot;|
grep -v &quot;24751&quot;|
grep -v &quot;24773&quot;|
grep -v &quot;24765&quot;|tail -28)
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep &quot;.fastq$&quot;|sed &#39;s/.fastq//g&#39; )
do
##------------------define variables------------------
#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do

name=$(echo $sample)
#sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}
reference_fasta=/data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta
ont_reads=/data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}.fastq

num=5


for run in first second third fourth
   do 

#name=run
##--------------create directories----------------
rm -r $Overall_output_directory/0${num}_${run}Freebayes/
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/{rawIlluminaMapped,freebayesOuput}
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc

#mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq

# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq.gz &gt; \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R1_val_1.fq
# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq.gz &gt; \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R2_val_2.fq


#/home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 index $reference_fasta

 # /home/vincent/apps/bwa-mem2-2.0pre2_x64-linux/bwa-mem2 mem -t ${threads} $reference_fasta /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R2_val_2.fq.gz | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 

bwa index $reference_fasta

 bwa mem -t ${threads} $reference_fasta /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R2_val_2.fq.gz | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 



#bwa mem -t ${threads}  $reference_fasta /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq |samtools sort -@8 -O BAM \
#-o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 
     
#/data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/


qualimap bamqc -bam $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam \
        -outdir $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc --java-mem-size=40G
    
samtools index $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam

###-----------------
##freebayesAfterCirclator
###-----------------
echo freebayes

    SECONDS=0

    freebayes -F 0.5 --min-coverage 4 -p 1 -f \
      $reference_fasta \
      $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam &gt; \
      $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf
      
      
    echo &quot;RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))&quot;
    echo END = `date +&quot;%Y-%m-%d %H:%M:%S&quot;`

##-------   
##bcftools variant integration of Illumina data into Pacbio polished genome
##-------

    SECONDS=0
    echo &quot;bgzip&quot;
bgzip -c  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf &gt; \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      echo &quot;tabix&quot;
tabix -p vcf  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
    echo &quot;bcftools&quot;
bcftools consensus -f $reference_fasta \
      -o $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      
    echo &quot;RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))&quot;
    echo END = `date +&quot;%Y-%m-%d %H:%M:%S&quot;`
  
  #bwa index $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta 
  
  rm $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam &amp;
  
   ##-------------------new variable name for next polishing steps-----------
  reference_fasta=$Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta  
  num=$((num+1)) 
done
  done #sample  


##--------------indexing


#bwa index $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta  

##---------------
##PostAssemblyAnalysis
##---------------
for samplezzz in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
grep -v &quot;24743&quot;|
grep -v &quot;24729&quot;|
grep -v &quot;24795&quot;|
grep -v &quot;24794&quot;|
grep -v &quot;24762&quot;|
grep -v &quot;24751&quot;|
grep -v &quot;24773&quot;|
grep -v &quot;24765&quot;)
do

#samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

  echo &quot;##==============&quot;
  echo ${samplezzKurz}
  echo &quot;##==============&quot;
    #tail -3 $mainPath/03_metaphlan2/${id}_trimmed/metaphlan_profile/profiled_metagenome_${id}_trimmed.txt
    PostAssemblyAnalysis.R -i Y -a /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${samplezzz}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta \
      -F /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${samplezzz}/FAM${samplezzz}_R1_val_1.fq.gz \
      -R /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${samplezzz}/FAM${samplezzz}_R1_val_1.fq.gz \
      -o /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/PostaAssembly/FAM${samplezzz}/ \
      -t 35
   
      
    done




###--------------on local



 name=di_K2_6h
 scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/quastwithRacon_FreebayesONT/ ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/


scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/MultiQCRaconFreebayesONT/multiqc_report.html
~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report_final.html
</code></pre>
</div>
<div id="circlator-2" class="section level4">
<h4>circlator 2</h4>
<p>Start align with <a href="https://github.com/b-brankovics/fasta_tools">fasta_shift</a>.
start align only single contig genomes</p>
<pre class="bash"><code>

##===========
##PROKKA annotate for fasta_shift
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
grep -v &quot;24743&quot;|
grep -v &quot;24729&quot;|
grep -v &quot;24795&quot;|
grep -v &quot;24794&quot;|
grep -v &quot;24762&quot;|
grep -v &quot;24751&quot;|
grep -v &quot;24773&quot;|
grep -v &quot;24765&quot;)
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep &quot;.fastq$&quot;|sed &#39;s/.fastq//g&#39; )
do
#for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#  do
  

  echo &quot;==============&quot;
  echo ${sample}


rm -r /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/
#mkdir -p /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${culture}_${strainzzz}

/usr/bin/perl /usr/bin/prokka --cpus 5 --outdir /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/ --locustag ${sample} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta

#prokka --outdir /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${culture}_${strainzzz} --force --usegenus --genus ${speciesss} --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes/${genomeszzz}

  done


##===========
##single conitg genome
##===========

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
grep -v &quot;24743&quot;|
grep -v &quot;24729&quot;|
grep -v &quot;24795&quot;|
grep -v &quot;24794&quot;|
grep -v &quot;24762&quot;|
grep -v &quot;24751&quot;|
grep -v &quot;24773&quot;|
grep -v &quot;24765&quot; |
grep -v &quot;24731&quot;|
grep -v &quot;24745&quot;|
grep -v &quot;24749&quot;|
grep -v &quot;24759&quot;|
grep -v &quot;24777&quot;)
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep &quot;.fastq$&quot;|sed &#39;s/.fastq//g&#39; )
do
#for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#  do
  

  echo &quot;==============&quot;
  echo ${sample}

seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*.fna
done

##===========
##complete but not circular genomes
##===========
grep -v &quot;24731&quot;
grep -v &quot;24745&quot;
grep -v &quot;24749&quot;
grep -v &quot;24759&quot;
grep -v &quot;24777&quot;

##===========
##identify dnaA
##===========

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
grep -v &quot;24743&quot;|
grep -v &quot;24729&quot;|
grep -v &quot;24795&quot;|
grep -v &quot;24794&quot;|
grep -v &quot;24762&quot;|
grep -v &quot;24751&quot;|
grep -v &quot;24773&quot;|
grep -v &quot;24765&quot;|
grep -v &quot;24731&quot;|
grep -v &quot;24745&quot;|
grep -v &quot;24749&quot;|
grep -v &quot;24759&quot;|
grep -v &quot;24777&quot;)
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep &quot;.fastq$&quot;|sed &#39;s/.fastq//g&#39; )
do
#for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#  do
  

  echo &quot;==============&quot;
  echo ${sample}

grep &quot;dnaA&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F &quot;\t&quot; &#39;{if($3==&quot;gene&quot;)print $0}&#39;
done



##===========
##start align
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
  grep -v &quot;24743&quot;|
  grep -v &quot;24729&quot;|
  grep -v &quot;24795&quot;|
  grep -v &quot;24794&quot;|
  grep -v &quot;24762&quot;|
  grep -v &quot;24751&quot;|
  grep -v &quot;24773&quot;|
  grep -v &quot;24765&quot;|
  grep -v &quot;24731&quot;|
  grep -v &quot;24745&quot;|
  grep -v &quot;24749&quot;|
  grep -v &quot;24759&quot;|
  grep -v &quot;24777&quot;|
  grep -v &quot;24758&quot;)
  
  for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep &quot;.fastq$&quot;|sed &#39;s/.fastq//g&#39; )
  do     
  echo &quot;==============&quot;
 echo ${sample}
 grep &quot;dnaA&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F &quot;\t&quot; &#39;{if($3==&quot;gene&quot;)print $0}&#39;
 orientation=$(grep &quot;dnaA&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F &quot;\t&quot; &#39;{if($3==&quot;gene&quot;)print $0}&#39; |cut -f 7)
 contigName=$(grep &quot;dnaA&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F &quot;\t&quot; &#39;{if($3==&quot;gene&quot;)print $0}&#39; |cut -f 1)
 samtools faidx /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta ${contigName} &gt; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/bac_genome.fna
 
 rm -r /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned
 mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned
 if [ $orientation == &quot;+&quot; ]
 then              echo &quot;forward oriented&quot;
 startAlign=$(grep &quot;dnaA&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F &quot;\t&quot; &#39;{if($3==&quot;gene&quot;)print $0}&#39; |awk -F &quot;\t&quot; &#39;{print $4-5}&#39;)
 echo -e &quot;start align at:  &quot;${startAlign}
 
 /home/vincent/apps/fasta_tools/bin/fasta_shift -i /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/bac_genome.fna -p ${startAlign} &gt; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta
 
 else              
 echo &quot;reverse oriented&quot;
 startAlign=$(grep &quot;dnaA&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F &quot;\t&quot; &#39;{if($3==&quot;gene&quot;)print $0}&#39; |awk -F &quot;\t&quot; &#39;{print $5+5}&#39;)
 echo -e &quot;start align at:  &quot;${startAlign}
 /home/vincent/apps/fasta_tools/bin/fasta_shift -i /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/bac_genome.fna -p ${startAlign} &gt; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/tmp.fasta
 revseq -sequence /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/tmp.fasta -outseq /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta -notag
 fi
 for non_bac in $(grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta | grep -v &quot;${contigName}&quot;|awk -F &quot; &quot; &#39;{print $1}&#39; |sed &#39;s/&gt;//g&#39;)
 do
 samtools faidx /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta ${non_bac} &gt;&gt; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta
 done #non_bac
 seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta
 done
 
##=============================
#check
##=============================

 

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
grep -v &quot;24743&quot;|
grep -v &quot;24729&quot;|
grep -v &quot;24795&quot;|
grep -v &quot;24794&quot;|
grep -v &quot;24762&quot;|
grep -v &quot;24751&quot;|
grep -v &quot;24773&quot;|
grep -v &quot;24765&quot;|
grep -v &quot;24731&quot;|
grep -v &quot;24745&quot;|
grep -v &quot;24749&quot;|
grep -v &quot;24759&quot;|
grep -v &quot;24777&quot;|
grep -v &quot;24758&quot;)
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep &quot;.fastq$&quot;|sed &#39;s/.fastq//g&#39; )
do      echo &quot;==============&quot;
echo ${sample}
rm -r /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter 
perl /usr/bin/prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter --locustag ${sample} --proteins s.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta

done

##-----------------------------------------
#check
##-----------------------------------------

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
grep -v &quot;24743&quot;|
grep -v &quot;24729&quot;|
grep -v &quot;24795&quot;|
grep -v &quot;24794&quot;|
grep -v &quot;24762&quot;|
grep -v &quot;24751&quot;|
grep -v &quot;24773&quot;|
grep -v &quot;24765&quot;|
grep -v &quot;24731&quot;|
grep -v &quot;24745&quot;|
grep -v &quot;24749&quot;|
grep -v &quot;24759&quot;|
grep -v &quot;24777&quot;|
grep -v &quot;24758&quot;)
for sample in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210310_genomeSeqONT/raw/fq/ |grep &quot;.fastq$&quot;|sed &#39;s/.fastq//g&#39; )
do      echo &quot;==============&quot;
echo ${sample}
grep &quot;dnaA&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter/*gff |awk -F &quot;\t&quot; &#39;{if($3==&quot;gene&quot;)print $0}&#39;
seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter/*.fna

done
</code></pre>
<pre class="bash"><code>##===========
##circlator
##===========
for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
grep -v &quot;24743&quot;|
grep -v &quot;24729&quot;|
grep -v &quot;24795&quot;|
grep -v &quot;24794&quot;|
grep -v &quot;24762&quot;|
grep -v &quot;24751&quot;|
grep -v &quot;24773&quot;|
grep -v &quot;24765&quot;)
do
#for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv )
#  do
  

  echo &quot;==============&quot;
  echo ${sample}

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/${sample}
#mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/circlator/${sample}

mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/tmp/

cat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R1_val_1.fq.gz \
/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R2_val_2.fq.gz &gt; \
/data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/tmp//${sample}.fq.gz


circlator all /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta /data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/tmp/${sample}.fq.gz /data/Project/2020_StarterCultureDiversity/09_flye/circlatorAfterPolishr/${sample}


  done
  

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/circlator/tmp/</code></pre>
</div>
</div>
</div>
<div id="illumina" class="section level2">
<h2>Illumina</h2>
<p>Illumina only based assemblies.</p>
<p>download genomes for polishing</p>
<pre class="bash"><code>##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================
#http://uhts-lgtf.vital-it.ch/symlink/S1_L7_R1_001_JWc1BQSlYm0n.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S1_L7_R2_001_azQERS6e6Mf3.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S2_L7_R1_001_tAyGggUA76Sv.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S2_L7_R2_001_Z7AkETEcWnrG.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S3_L7_R1_001_QG9hjiVdib1X.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S3_L7_R2_001_hhue2W6w2qqw.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S4_L7_R1_001_sCvHPBHZTe80.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S4_L7_R2_001_mlaWMoWEB7ur.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S5_L7_R1_001_jcQlllEuwmpJ.fastq.gz
#http://uhts-lgtf.vital-it.ch/symlink/S5_L7_R2_001_WsUA1vmuMQds.fastq.gz
rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz
mkdir -p /data/Project/2020_StarterCultureDiversity/99_log

vim /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_download.txt

vim /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_sampleNames2change.txt
</code></pre>
<div id="renaming" class="section level3">
<h3>renaming</h3>
<pre class="bash"><code>

for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

cat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_L*_R1_*.fastq.gz &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R1.fastq.gz

cat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_L*_R2_*.fastq.gz &gt; /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R2.fastq.gz


done #samplezzz


##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================

for name in $(cut -d &#39;/&#39; -f 5 /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_download.txt)
do

  name_01=$(echo $name |cut -d &#39;_&#39; -f 1)
  Lane=$(echo $name |cut -d &#39;_&#39; -f 2|sed &#39;s/L//g&#39;)
  Read=$(echo $name |cut -d &#39;_&#39; -f 3)  
  
  Lane_new=$((Lane+2))

  newName=$(awk -F &quot;\t&quot; -v name=&quot;$name_01&quot; &#39;{if($2==name) {print $1}}&#39; /data/Project/2020_StarterCultureDiversity/99_log/20200625_genomes_sampleNames2change.txt)
  
  echo ${newName}&quot;_&quot;${Read}&quot;.fastq.gz&quot;
    
  mv /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${name} /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${newName}_${Read}.fastq.gz
    
done
##-------qualtiy control

ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/|cut -d &#39;_&#39; -f 1|sort|uniq -c |wc -l
ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/|cut -d &#39;_&#39; -f 1|sort|uniq -c 



##------------------------------------
##===============================
##download old samples downloaded 20210209
##------------------------------------
##===============================


for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/|grep &quot;_R1&quot;)
do
genomeshort=$(echo $genomesss |cut -d &#39;_&#39; -f 1)
echo $genomeshort
#mv /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${genomesss} /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${genomeshort}

mv /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${genomeshort}_R1.fastq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${genomeshort}/${genomeshort}_R1.fastq.gz

mv /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${genomeshort}_R2.fastq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${genomeshort}/${genomeshort}_R2.fastq.gz

done
</code></pre>
</div>
<div id="fastqc" class="section level3">
<h3>FastQC</h3>
<p>In a first step we creat <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>
reports for all the libraries. These are then visualized with <a href="https://multiqc.info/">MultiQC</a></p>
<pre class="bash"><code>source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh
threads=8

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R1.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R2.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
    
done #samplezzz

  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/MultiQC

  
  multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/MultiQC

##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================
threads=8

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/ |grep &quot;.fastq&quot;)
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${samplezzKurz}_R1.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${samplezzKurz}_R2.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
    
done #samplezzz

  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/MultiQC

  
  multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/MultiQC

##------------------------------------
##===============================
##download old samples downloaded 20210209
##------------------------------------
##===============================

threads=37


mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz )
do

#samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzz}/${samplezzz}_R1.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/fastQC/
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzz}/${samplezzz}_R2.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/fastQC//
    
done #samplezzz</code></pre>
</div>
<div id="trim-galore" class="section level3">
<h3>Trim-galore</h3>
<p>In the following we clip adaptors with <a href>trim-galore</a></p>
<pre class="bash"><code>source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh
threads=30
id=RMK202_withOld

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

  echo &quot;##==============&quot;
  echo ${samplezzKurz}
  echo &quot;##==============&quot;
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      trim_galore -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz} \
        /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R1.fastq.gz  \
        /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R2.fastq.gz 
        
  echo &quot;##==============&quot;
  echo ${id} &quot;fastQC of trimmed&quot;
  echo &quot;##==============&quot;
      
    mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/
      
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R1.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/
    
    fastqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/${samplezzz}/${samplezzKurz}_R2.fastq.gz -t ${threads} -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/
    
        
  done 
  
python3 /home/vincent/miniconda2/bin/multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore//Multiqc




mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do


echo &quot;/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/bamqc&quot; &gt;&gt; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log/multiqc_mapping_logfile.txt
   

done


  mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore//Multiqc

  multiqc /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/fastQC/ -o /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore//Multiqc


echo &quot;/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/bamqc&quot; &gt;&gt; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log/multiqc_mapping_logfile.txt
   

samtools index /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam &amp;


done

###--------------------------------multiQC

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/


  multiqc --config /archiv/logs/multiQC_config.txt --file-list /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log/multiqc_mapping_logfile.txt \
  -o archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/
##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================
threads=35
#id=RMK202_withOld

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz |grep &quot;.fastq&quot; )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

  echo &quot;##==============&quot;
  echo ${samplezzKurz}
  echo &quot;##==============&quot;
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      trim_galore --fastqc -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/${samplezzKurz} \
        /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${samplezzKurz}_R1.fastq.gz  \
        /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/${samplezzKurz}_R2.fastq.gz 
        
  echo &quot;##==============&quot;
  echo ${id} &quot;fastQC of trimmed&quot;
  echo &quot;##==============&quot;
      

  done 
  
##--------------------------------------
##bring together
##--------------------------------------


for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz |grep &quot;.fastq&quot; )
do
samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

cp -r /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/ /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/
  
done

##------------------------------------
##===============================
##download new samples 20200624
##------------------------------------
##===============================

##special case

cat /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_1A3P4_S293_R1_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S301_R1_001.fastq.gz &gt; /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S3012_R1_001.fastq.gz

cat /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_1A3P4_S293_R2_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S301_R2_001.fastq.gz &gt; /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S3012_R2_001.fastq.gz

mkdir -p /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns
mv /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_1A3P4_S293_R1_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns/

mv /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S301_R1_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns/

mv /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_1A3P4_S293_R2_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns/

mv /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/13499_2A4P4_S301_R2_001.fastq.gz /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/13499/twoRuns/

##redo

threads=35
#id=RMK202_withOld

mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
for samplezzz in $(ls /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/ |grep &quot;^1&quot; )
do
samplezzz=13499

samplezzKurz=$(echo -e &quot;FAM&quot;${samplezzz} )

  echo &quot;##==============&quot;
  echo ${samplezzKurz}
  echo &quot;##==============&quot;
  
  

  
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
      trim_galore --fastqc -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz} \
        /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/${samplezzz}/${samplezzz}*_R1*.fastq.gz  \
        /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/${samplezzz}/${samplezzz}*_R2*.fastq.gz
        
  echo &quot;##==============&quot;
  echo ${id} &quot;fastQC of trimmed&quot;
  echo &quot;##==============&quot;
      

  done 



##--------------------------------------
##bring together
##--------------------------------------


for samplezzz in $(ls /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz/transfer_vincent/ |grep &quot;^1&quot; )
do
samplezzKurz=$(echo -e &quot;FAM&quot;${samplezzz} )
rm  -r /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}
mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}

cp /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzz}*_R1*val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz

cp /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzz}*_R2*_val_2.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz

done


##------------------------------------
##===============================
##download old samples downloaded 20210209
##------------------------------------
##===============================

threads=37

/data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzz}/${samplezzz}_R1.fastq.gz

threads=35
#id=RMK202_withOld

#mkdir -p /data/Project/2020_StarterCultureDiversity/01_Data/20200625_genome/01_rawdata/gz/fastQC/
for samplezzKurz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/ |grep &quot;fastqc&quot; -i -v )
do
#samplezzz=13499

#samplezzKurz=$(echo -e &quot;FAM&quot;${samplezzz} )

  echo &quot;##==============&quot;
  echo ${samplezzKurz}
  echo &quot;##==============&quot;
  
  

  
      rm -r /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/
      mkdir -p  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/
      

       cp /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzKurz}/${samplezzKurz}_R1.fastq.gz   /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz
       
       cp /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzKurz}/${samplezzKurz}_R2.fastq.gz   /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz
       
      #trim_galore --fastqc -paired -o /data/Project/2020_StarterCultureDiversity/01_Data/oldGenome_dialact/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/ /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzKurz}/${samplezzKurz}_R1.fastq.gz   /data/Project/2020_StarterCultureDiversity/01_Data/20210209_Dialact_genome/01_rawdata/gz/${samplezzKurz}/${samplezzKurz}_R2.fastq.gz
        


  done 
</code></pre>
<pre class="bash"><code>

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore//Multiqc /home/vincent/Desktop/Projects/2020_strainDelineation/01_logs
</code></pre>
</div>
<div id="spades-assembly" class="section level3">
<h3>Spades assembly</h3>
<p>In this step we assemble the raw reads of the isolate with <a href="http://bioinf.spbau.ru/en/spades3.7">spades</a>.</p>
<pre class="bash"><code>source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh

threads=35
num=1
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do
samplezzKurz=FAM24777
samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)
  echo &quot;##==============&quot;
  echo ${samplezzKurz}
  echo $num
  echo &quot;##==============&quot;
       mkdir -p /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly/
 
 
     /home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --pe1-1 /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz \
       --pe1-2 /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz \
       -o /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly//assembly_Spades_20200516 -t ${threads} -m 100

   num=$((num+1))

    done

#---------------
##short info
##---------------
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)


 count=$(grep &quot;&gt;&quot; -c /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly//assembly_Spades_20200516/contigs.fasta)
   
     #echo &quot;##==============&quot;
  echo -e ${samplezzKurz} &quot; : &quot; ${count}
#  echo &quot;##==============&quot;
 
    done
##---------------
##PostAssemblyAnalysis
##---------------
for samplezzz in $(ls /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/gz/20200515_UniBe/ )
do

samplezzKurz=$(echo ${samplezzz} | cut -d _ -f 1)

  echo &quot;##==============&quot;
  echo ${samplezzKurz}
  echo &quot;##==============&quot;
    #tail -3 $mainPath/03_metaphlan2/${id}_trimmed/metaphlan_profile/profiled_metagenome_${id}_trimmed.txt
    PostAssemblyAnalysis.R -a /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly//assembly_Spades_20200516/scaffolds.fasta \
      -F /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R1_val_1.fq.gz \
      -R /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/${samplezzKurz}/${samplezzKurz}_R2_val_2.fq.gz \
      -o /data/Project/2020_StarterCultureDiversity//06_Spades/${samplezzKurz}/assembly/PostaAssembly \
      -t 35
    
    done
</code></pre>
</div>
</div>
<div id="strain-collection" class="section level2">
<h2>Strain collection</h2>
<div id="complete-ncbi-refs" class="section level3">
<h3>complete NCBI Refs</h3>
<p>downloaded all reference genomes from NCBI refseq on 20200728.</p>
<pre class="bash"><code>##=====================================================================================
##------------------------all complete streptos
##=====================================================================================
species=$(echo &quot;Streptococcus thermophilus&quot;)
species_short=Sterm_references
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
curl &#39;ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${species}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
    #pecies_short=Sterm_references

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${species}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $1, &quot;S_thermophilus_&quot;$9$10}&#39;|sed &#39;s/strain=//g&#39; | sed &#39;s/_/-/g&#39; |sed &#39;s/GCF-/GCF_/g&#39;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm_names.txt

##--------------------------
##prepare list
##--------------------------
species=$(echo &quot;Streptococcus thermophilus&quot;)
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${species}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $1,$2,$3,&quot;S_thermophilus_&quot;$9$10,$15,$17}&#39;|sed &#39;s/strain=//g&#39; | sed &#39;s/_/-/g&#39; |sed &#39;s/GCF-/GCF_/g&#39; |sed &#39;s/ /-/g&#39;   &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_00.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${species}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $1,$2,$3,&quot;S_thermophilus_&quot;$9$10,$15,$17}&#39;|sed &#39;s/strain=//g&#39; | sed &#39;s/_/-/g&#39; |sed &#39;s/GCF-/GCF_/g&#39; |sed &#39;s/ /-/g&#39;|cut -f 4 |sed &#39;s/S-thermophilus-/S-/g&#39;| sed &#39;s/-STH-CIRM-/-I-/g&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_01.txt
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt
while read -r line
do
#echo $line
echo -e ${line} |head -c 10 |head -1 |sed &#39;s/^$//g&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 
echo -e &quot;\n&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 

done &lt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_01.txt
sed -i &#39;/^[[:space:]]*$/d&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 

paste -d &#39;\t&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_00.txt &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt

#-------------------- add outgroup
###CHANGED OUTGRPOUP ON 28.1.2021 TO S. VESTIBULARIS

outgroupNAME=NCTC8618

outgroupNAME=$(echo &quot;Streptococcus vestibularis&quot;)


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${outgroupNAME}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|&#39; &gt;&gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
    species_short=Sterm_references

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${outgroupNAME}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $1, &quot;S_salivarius_&quot;$9$10}&#39;|sed &#39;s/strain=//g&#39; | sed &#39;s/_/-/g&#39; |sed &#39;s/GCF-/GCF_/g&#39;&gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm_names.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${outgroupNAME}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print &quot;Out-sali&quot;,$1,$2,$3,&quot;S_thermophilus_&quot;$9$10,$15,$17}&#39;|sed &#39;s/strain=//g&#39; | sed &#39;s/_/-/g&#39; |sed &#39;s/GCF-/GCF_/g&#39; |sed &#39;s/ /-/g&#39;   &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${species}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print ,$1,$2,$3,&quot;S_thermophilus_&quot;$9$10,$15,$17}&#39;|sed &#39;s/strain=//g&#39; | sed &#39;s/_/-/g&#39; |sed &#39;s/GCF-/GCF_/g&#39; |sed &#39;s/ /-/g&#39;|cut -f 4 |sed &#39;s/S-thermophilus-/S-/g&#39;| sed &#39;s/-STH-CIRM-/-I-/g&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt


cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt
gunzip *.fna.gz


##------------change name
sed -i &#39;s/ //g&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt

for genomesss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt)
do
shortnamesss=$(grep &quot;${genomesss}&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt | cut -f 1)
echo -e ${genomesss} &quot; to &quot; ${shortnamesss}

mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${genomesss}*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna

done
mv GCF_900636445.1_41965_G01_genomic.fna Out-vest.fna

##=====================================================================================
##------------------------all complete lactos
##=====================================================================================
species_short=Ldel_references
species=$(echo &quot;Lactobacillus delbrueckii&quot;)

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
#curl &#39;ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;Lactobacillus delbrueckii&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; )print $0}&#39; |cut -f 20| sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_ldel.txt
    #pecies_short=Sterm_references
    


#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${species}&quot;|grep &quot;strain=&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $1, &quot;L_delbrueckii_&quot;$9$10}&#39;|sed &#39;s/strain=//g&#39; | sed &#39;s/_/-/g&#39; |sed &#39;s/GCF-/GCF_/g&#39;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_ldel_names.txt


##--------------------------
##prepare list
##--------------------------
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${species}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $1,$2,$3,&quot;L_delbrueckii_&quot;$9$10,$15,$17}&#39;|sed &#39;s/strain=//g&#39; | sed &#39;s/_/-/g&#39; |sed &#39;s/GCF-/GCF_/g&#39; |sed &#39;s/ /-/g&#39;   &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_00.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${species}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $1,$2,$3,&quot;L_delbrueckii_&quot;$9$10,$15,$17}&#39;|sed &#39;s/strain=//g&#39; | sed &#39;s/_/-/g&#39; |sed &#39;s/GCF-/GCF_/g&#39; |sed &#39;s/ /-/g&#39;|cut -f 4 |sed &#39;s/L-delbrueckii-/L-/g&#39;| sed &#39;s/L-Lactobacillus-bulgaricus-/L-/g&#39; | sed &#39;s/L-Lactobacillus-delbrueckii-subsp.-/L-/g&#39;  &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_01.txt
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt
while read -r line
do
#echo $line
echo -e ${line}  |head -1 |sed &#39;s/^$//g&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 
echo -e &quot;\n&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 

done &lt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_01.txt
sed -i &#39;/^[[:space:]]*$/d&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt 

paste -d &#39;\t&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_02.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_00.txt &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt
cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt

##-------------------------
##outgroup
##type strain l. delbrueckii spp. bulgaricus


outgroupNAME=$(echo &quot;Lh12&quot;)

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;Lactobacillus helveticus&quot;|tail -1 |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|&#39; &gt;&gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_ldel.txt

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${outgroupNAME}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $1, &quot;S_salivarius_&quot;$9$10}&#39;|sed &#39;s/strain=//g&#39; | sed &#39;s/_/-/g&#39; |sed &#39;s/GCF-/GCF_/g&#39;&gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm_names.txt


#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${outgroupNAME}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print &quot;Out-bulg&quot;,$1,$2,$3,&quot;S_thermophilus_&quot;$9$10,$15,$17}&#39;|sed &#39;s/strain=//g&#39; | sed &#39;s/_/-/g&#39; |sed &#39;s/GCF-/GCF_/g&#39; |sed &#39;s/ /-/g&#39;   &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e &quot;${species}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print ,$1,$2,$3,&quot;S_thermophilus_&quot;$9$10,$15,$17}&#39;|sed &#39;s/strain=//g&#39; | sed &#39;s/_/-/g&#39; |sed &#39;s/GCF-/GCF_/g&#39; |sed &#39;s/ /-/g&#39;|cut -f 4 |sed &#39;s/S-thermophilus-/S-/g&#39;| sed &#39;s/-STH-CIRM-/-I-/g&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt


cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA
    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_ldel.txt
gunzip *.fna.gz




##------------change name
sed -i &#39;s/ //g&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt

for genomesss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt)
do
shortnamesss=$(grep &quot;${genomesss}&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt | cut -f 1)
echo -e ${genomesss} &quot; to &quot; ${shortnamesss}

mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${genomesss}*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna

done

mv GCF_902386585.1_UHGG_MGYG-HGUT-02384_genomic.fna  Out-helv.fna

##=====================================================================================
##------------------------PROKKA annotate (for panaroo)
##=====================================================================================$

##-------------------------Sterm
shortnamesss=$(echo &quot;Out-vest&quot;)
species_short=Sterm_references
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKK*
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_GFF
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN
for shortnamesss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt)
#for genome in $(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FNA |grep &quot;fna&quot;)
do

#shortnamesss=$(grep &quot;${genome}&quot; /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt | cut -f 2)
#echo -e ${genome} &quot; to &quot; ${shortnamesss}

locustagss=$(echo ${shortnamesss}|cut -d &#39;_&#39; -f 3,4,5,6|sed &#39;s/L-delbrueckii-/L-/g&#39; |sed &#39;s/S-thermophilus-/S-/g&#39;|sed &#39;s/STH-//g&#39;|sed &#39;s/S-salivarius-/REF-/g&#39;|cut -d &#39;.&#39; -f 1)
echo $locustagss
#rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/PROKKA/${shortnamesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}

/usr/bin/perl /usr/bin/prokka --cpus 37 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss} --force --usegenus --genus streptococcus --locustag ${locustagss} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.gff &gt; \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_GFF/${shortnamesss}.gff

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.ffn &gt; \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN/${shortnamesss}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.faa &gt; \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA/${shortnamesss}.faa


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.ffn &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN/${strainzzz}.ffn

done
##--------------------------------------Ldel
species_short=Ldel_references
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKK*
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_GFF
rm -r  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN
mkdir -p ${species_short}/PROKKA_FAA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN

conda activate PROKKA

shortnamesss=Out-helv
for shortnamesss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt )
#for genome in $(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FNA |grep &quot;fna&quot;)
do

#shortnamesss=$(grep &quot;${genome}&quot; /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt | cut -f 2)
#echo -e ${genome} &quot; to &quot; ${shortnamesss}

locustagss=$(echo ${shortnamesss}|cut -d &#39;_&#39; -f 3,4,5,6|sed &#39;s/L-delbrueckii-/L-/g&#39; |sed &#39;s/S-thermophilus-/S-/g&#39;|sed &#39;s/STH-//g&#39;|sed &#39;s/S-salivarius-/REF-/g&#39;|cut -d &#39;.&#39; -f 1)
echo -e ${shortnamesss} &quot;and locusTag &quot; $locustagss

#done
#rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/PROKKA/${shortnamesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}

#/usr/bin/perl /usr/bin/prokka --cpus 5 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss} --force --usegenus --genus lactobacillus --locustag ${locustagss} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna

prokka --cpus 20 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss} --force  --locustag ${shortnamesss} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/FNA/${shortnamesss}.fna


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.gff &gt; \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_GFF/${shortnamesss}.gff

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.ffn &gt; \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN/${shortnamesss}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.faa &gt; \
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA/${shortnamesss}.faa


#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA/${shortnamesss}/PROKKA*.ffn &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FFN/${strainzzz}.ffn

done
</code></pre>
<p>move local</p>
<pre class="bash"><code>mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/


scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final.txt ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm.txt
</code></pre>
<p>###gather own genomes</p>
<p>first I need to gather all necessary genomes.</p>
<pre class="bash"><code>scp /home/vincent/Downloads/RMK_strain_stock\ -\ Sheet1\ \(1\).tsv vincent@130.223.49.145:/data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv

##------------------------------------
##ONT + Illumina genomes 
##------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/

awk -F &quot;\t&quot; &#39;{if($7==&quot;Illumina&quot;||$7==&quot;ONT&quot;)print $0}&#39; /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |grep -v  &quot;species&quot; &gt; /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv

for sample in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/99_log/20200619_Seq_run_ONT_ifik.csv |grep -v &quot;24746&quot;|
grep -v &quot;24743&quot;|
grep -v &quot;24729&quot;|
grep -v &quot;24795&quot;|
grep -v &quot;24794&quot;|
grep -v &quot;24762&quot;|
grep -v &quot;24751&quot;|
grep -v &quot;24773&quot;|
grep -v &quot;24765&quot;|
grep -v &quot;24731&quot;|
grep -v &quot;24745&quot;|
grep -v &quot;24749&quot;|
grep -v &quot;24759&quot;|
grep -v &quot;24777&quot;|
grep -v &quot;24758&quot;)
do      echo &quot;==============&quot;
echo ${sample}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/{all,Sterm,Ldel}


speciesss=$(grep &quot;${sample}&quot; /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 2|head -c 1)
culturesss=$(grep &quot;${sample}&quot; /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 4)
methodsss=$(grep &quot;${sample}&quot; /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 7|head -c 1)


cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${speciesss}_${methodsss}_${culturesss}_${sample}.fasta

grep -v &quot;${sample}&quot; /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv &gt; /data/Project/2020_StarterCultureDiversity/99_log/tmp

mv /data/Project/2020_StarterCultureDiversity/99_log/tmp /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv

done
wc -l /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv
wc -l /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv
cut -f 2 /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv|sort|uniq -c
##------------------------------------
##Illumina genomes Sterm
##------------------------------------

for sample in $(grep &quot;thermophilus&quot; /data/Project/2020_StarterCultureDiversity/99_log/tmp /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv |cut -f 1|cut -d &#39;:&#39; -f 2|grep -v &quot;1708&quot;|grep -v &quot;24052&quot;|grep -v &quot;24054&quot;)
do      echo &quot;==============&quot;
echo ${sample}


speciesss=$(grep &quot;${sample}&quot; /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 2|head -c 1)
culturesss=$(grep &quot;${sample}&quot; /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 4)
methodsss=$(echo &quot;I&quot;)


cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*_${sample}*fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${speciesss}_${methodsss}_${culturesss}_${sample}.fasta


done

##------------------------------------
##Illumina genomes Ldel
##------------------------------------

for sample in $(grep &quot;delbrueckii&quot; /data/Project/2020_StarterCultureDiversity/99_log/tmp /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv |cut -f 1|cut -d &#39;:&#39; -f 2)
do      echo &quot;==============&quot;
echo ${sample}
#sample=21745

speciesss=$(grep &quot;${sample}&quot; /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 2|head -c 1)
culturesss=$(grep &quot;${sample}&quot; /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv |cut -f 4)
methodsss=$(echo &quot;I&quot;)


#cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/*_${sample}*fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${speciesss}_${methodsss}_${culturesss}_${sample}.fasta
cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/*_${sample}*fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${sample}.fasta



done


cat /data/Project/2020_StarterCultureDiversity/06_Spades/FAM24777/assembly/assembly_Spades_20200516/contigs.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/L_I_202_24777.fasta 


grep &quot;21745&quot; /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection_IlluminaONly.tsv
cat /archiv/Dialact/ReferenceDatabase/20190322/Annotation/Prokka/Lactobacillus_delbrueckii/FAM21745c1_11092016.fna/PROKKA_03272019.fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/L_I_101_21745.fasta 



ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |cut -d _ -f 1|sort|uniq -c

##-------------------------------------================================================================
#rename contigs
##-------------------------------------================================================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |sed &#39;s/.fasta//g&#39;)
do
echo &quot;============================================&quot;
num=1
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${genomesss}.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed/${genomesss}.fasta
for contigs in $(grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/${genomesss}.fasta |sed &#39;s/&gt;//g&#39;)
  do
  echo ${contigs} &quot; to &quot;${genomesss}&quot;_c&quot;$num
  
  sed -i &quot;s/$contigs/${genomesss}_c$num/g&quot;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed/${genomesss}.fasta
  
  #sed &quot;s/$contigs/S_term_phage_01_contig$num/g&quot; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta |grep &quot;&gt;&quot;
  
  num=$((num+1))
  done


done #genomesss

##-------------------------------------================================================================
#remove small contigs
##-------------------------------------================================================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |sed &#39;s/.fasta//g&#39;)
do
  echo ==========================
   echo ${genomesss}
   echo ==========================
 

  seqkit seq -m 500 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed/${genomesss}.fasta -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved/${genomesss}.fasta
 
    done


##-------------------------------------================================================================
#Final and split species
##-------------------------------------================================================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/{Sterm,Ldel}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |grep &quot;^S&quot; |sed &#39;s/.fasta//g&#39;)
do
  echo ==========================
   echo ${genomesss}
   echo ==========================
 

  cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved/${genomesss}.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/${genomesss}.fasta
 
    done
    
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all/ |grep &quot;^L&quot; |sed &#39;s/.fasta//g&#39;)
do
  echo ==========================
   echo ${genomesss}
   echo ==========================
 

  cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/all_renamed_smallRemoved/${genomesss}.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Ldel/${genomesss}.fasta
 
    done
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Ldel/ |wc -l
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/ |wc -l


##=================================================================
##DO A NEW COLLECTION 20210201
##=================================================================
##------------------------------------------
##STERM
##------------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm

mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/{Sterm,Ldel}



##--------01 Ilumina genomes


for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/ |cut -d &#39;_&#39; -f 3|sed &#39;s/.fasta//g&#39;)
do
echo &quot;==========================================================================&quot;
echo ${genomesss}
#seq_length.py /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*${genomesss}.fasta |head -1
num=1
cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*${genomesss}.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

for contigs in $(grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta |sed &#39;s/&gt;//g&#39;)
  do
  echo ${contigs} &quot; to &quot;${genomesss}&quot;_c&quot;$num
  
  sed -i &quot;s/$contigs/${genomesss}_contig_$num/g&quot;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta
  
  #sed &quot;s/$contigs/S_term_phage_01_contig$num/g&quot; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta |grep &quot;&gt;&quot;
  
  num=$((num+1))
  done
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta |head -1
done

 ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm |wc -l

##--------02 flye genomes

for genomesss in $(echo &quot;24750
24739
13496
24738
13499
24730
24726
24728
13498
24734
24737
24853
24854
24855
24742
13494
24758
24735
24736
24724
24748
24740&quot;)
do
echo &quot;--------------------------------------------&quot;
#sed &quot;s/&gt;contig/&gt;${genomesss}_contig/g&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

sed &quot;s/&gt;contig/&gt;${genomesss}_contig/g&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta



seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

done



for genomesss in $(echo &quot;24745

&quot;)
do
echo &quot;--------------------------------------------&quot;
#sed &quot;s/&gt;contig/&gt;${genomesss}_contig/g&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

shortsss=$(echo ${genomesss} |cut -d &#39;_&#39; -f 1)

sed &quot;s/&gt;contig/&gt;${shortsss}_contig/g&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${shortsss}.fasta



seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${shortsss}.fasta

done

###---------------------rmk genome
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/*.fna
cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_O_202_SMAG.fna  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/202SMAG.fasta

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_I_202_S50.fna  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/S50.fasta

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_I_202_S72.fna  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/S72.fasta

##------------------------------------------
##Ldel
##------------------------------------------


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel

mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/{Sterm,Ldel}



##--------01 Ilumina genomes


for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/ |cut -d &#39;_&#39; -f 3|sed &#39;s/.fasta//g&#39;)
do
echo &quot;==========================================================================&quot;
echo ${genomesss}
#seq_length.py /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*${genomesss}.fasta |head -1
num=1
cat /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/L.delbrueckii/01_genomes/*${genomesss}.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

for contigs in $(grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta |sed &#39;s/&gt;//g&#39;)
  do
  echo ${contigs} &quot; to &quot;${genomesss}&quot;_c&quot;$num
  
  sed -i &quot;s/${contigs}/${genomesss}_contig_${num}/g&quot;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta
  
  #sed &quot;s/$contigs/S_term_phage_01_contig$num/g&quot; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta |grep &quot;&gt;&quot;
  
  num=$((num+1))
  done
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta |head -1
done

 ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel |wc -l


##--------02 flye genomes
genomesss=24758
#24759
#24777

for genomesss in $(echo &quot;24756
24786
24790&quot;)
do
echo &quot;--------------------------------------------&quot;
echo ${genomesss}
#sed &quot;s/&gt;contig/&gt;${genomesss}_contig/g&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

num=1
#seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta
for contigs in $(grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta |sed &#39;s/&gt;//g&#39;)
  do

 sed -i &quot;s/${contigs}/${genomesss}_contig_${num}/g&quot;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

num=$((num+1))

done
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

done


##--------03 new flye genomes 20210310

##-------LDEL

for genomesss in $(echo &quot;24754 24798&quot;)
do
echo &quot;--------------------------------------------&quot;
echo ${genomesss}
#seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta


#sed &quot;s/&gt;contig/&gt;${genomesss}_contig/g&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

num=1
#seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta
for contigs in $(grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta |sed &#39;s/&gt;//g&#39;)
  do

 sed -i &quot;s/${contigs}/${genomesss}_contig_${num}/g&quot;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

num=$((num+1))

done
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta

done

##-------Sterm

for genomesss in $(echo &quot;24732 24733 24743 24746 24747 24749&quot;)
do
echo &quot;--------------------------------------------&quot;
echo ${genomesss}
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta |head
done

#sed &quot;s/&gt;contig/&gt;${genomesss}_contig/g&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

num=1
#seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta
for contigs in $(grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta |sed &#39;s/&gt;//g&#39;)
  do

 sed -i &quot;s/${contigs}/${genomesss}_contig_${num}/g&quot;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

num=$((num+1))

done
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

done
</code></pre>
<p>somehow 24855_contig_2 is redundant remove this.</p>
<pre class="bash"><code>samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855.fasta
samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855.fasta 24855_contig_2 &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855_new.fasta
samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855.fasta 24855_contig_3 &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855_new.fasta
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855_new.fasta

sed &#39;s/24855_contig_2/24855_contig_1/g&#39;   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855_new.fasta |sed &#39;s/24855_contig_3/24855_contig_2/g&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855.fasta
seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855.fasta
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S24855_new.fasta</code></pre>
</div>
<div id="align-contigs" class="section level3">
<h3>align contigs</h3>
<p>Here, I use Rattag to align the illumina (draft) genome according to
the synteny of the ONT genomes.</p>
<pre class="bash"><code>cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/
ragtag.py correct ref.fasta query.fasta

mkdir -p aligned

###==========================================================
##start align Illumina assemblies
###==========================================================
###lin1
(echo &quot;24732 24733 24743 24746 24747 24749&quot;)

#ragtag.py correct 24750.fasta 24732.fasta -u -t 35  --inter -o ./aligned/24732_corrected
ragtag.py correct 24750.fasta 24731.fasta -u -t 35  --inter -o ./aligned/24731_corrected
#ragtag.py correct 24750.fasta 24733.fasta -u -t 35  --inter -o ./aligned/24733_corrected
ragtag.py correct 24750.fasta 24748.fasta -u -t 35  --inter -o ./aligned/24748_corrected

###lin2
ragtag.py correct 24739.fasta 13497.fasta -u -t 35  --inter -o ./aligned/13497_corrected
ragtag.py correct 24739.fasta 13495.fasta -u -t 35  --inter -o ./aligned/13495_corrected

###lin3
ragtag.py correct 24738.fasta 24744.fasta -u -t 35  --inter -o ./aligned/24744_corrected
ragtag.py correct 24738.fasta 24745.fasta -u -t 35  --inter -o ./aligned/24745_corrected
ragtag.py correct 24738.fasta S50.fasta -u -t 35  --inter -o ./aligned/S50_corrected

###lin4
ragtag.py correct 24730.fasta 24729.fasta -u -t 35  --inter -o ./aligned/24729_corrected

###lin5
#ragtag.py correct 24730.fasta 24749.fasta -u -t 35  --inter -o ./aligned/24749_corrected
#ragtag.py correct 24730.fasta 24747.fasta -u -t 35  --inter -o ./aligned/24747_corrected


###lin6

ragtag.py correct 24726.fasta 24727.fasta -u -t 35  --inter -o ./aligned/24727_corrected
###lin7
ragtag.py correct 24734.fasta 24724.fasta -u -t 35  --inter -o ./aligned/24724_corrected
ragtag.py correct 24734.fasta 24725.fasta -u -t 35  --inter -o ./aligned/24725_corrected

###lin8
ragtag.py correct 24737.fasta 13492.fasta -u -t 35  --inter -o ./aligned/13492_corrected
ragtag.py correct 24737.fasta 13500.fasta -u -t 35  --inter -o ./aligned/13500_corrected
ragtag.py correct 24737.fasta 13493.fasta -u -t 35  --inter -o ./aligned/13493_corrected
ragtag.py correct 24742.fasta 24741.fasta -u -t 35  --inter -o ./aligned/24741_corrected
ragtag.py correct 24742.fasta 13491.fasta -u -t 35  --inter -o ./aligned/13491_corrected

###lin9
#ragtag.py correct 24742.fasta 24743.fasta -u -t 35  --inter -o ./aligned/24743_corrected


###lin10
#ragtag.py correct 24742.fasta 24746.fasta -u -t 35  --inter -o ./aligned/24746_corrected
###lin11
ragtag.py correct 24736.fasta 24758.fasta -u -t 35  --inter -o ./aligned/24758_corrected
ragtag.py correct 24736.fasta 12273.fasta -u -t 35  --inter -o ./aligned/12273_corrected
#ragtag.py correct 24736.fasta 12273.fasta -u -t 35 -o ./aligned/12273_corrected
#ragtag.py correct 24736.fasta 12273.fasta -u -t 35 -b 0 -o ./aligned/12273_corrected

###linx
ragtag.py correct 24855.fasta S72.fasta -u -t 35  --inter -o ./aligned/S72_corrected


###==========================================================
##bring all genomes together
###==========================================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned

genomesss=12273
##--------------------------Illuminas
  for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/ |grep &quot;.fasta$&quot;|cut -d &#39;.&#39; -f 1)
  do
  echo &quot;--------------------------------------------&quot;
  #sed &quot;s/&gt;contig/&gt;${genomesss}_contig/g&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta
  
 cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm//aligned/${genomesss}_corrected/*.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${genomesss}.fasta
  
num=1
for contigss in $(grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm//aligned/${genomesss}_corrected/*.fasta|sed &#39;s/&gt;//g&#39;)
do

  sed -i &quot;s/&gt;${contigss}$/&gt;${genomesss}_contig_${num}/g&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${genomesss}.fasta
  
  

#sed &quot;s/&gt;${contigss}/&gt;${genomesss}_contig_${num}/g&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${genomesss}.fasta

num=$((num+1))
done
  

  
  #seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta
  
  done


ll /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/
find /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/ -size  0 -print -delete
ll /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/
##--------------------------ONT
genomesss=24855
for genomesss in $(echo &quot;24750
24739
13496
24738
13499
24730
24726
24728
13498
24734
24737
24853
24854
24855
24742
13494
24735
24736
24724
24748
24740
202SMAG 24732 24733 24743 24746 24747 24749&quot;)
do
echo &quot;--------------------------------------------&quot;
#sed &quot;s/&gt;contig/&gt;${genomesss}_contig/g&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${genomesss}.fasta

num=1
for contigss in $(grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta|sed &#39;s/&gt;//g&#39;)
do



sed -i &quot;s/&gt;${contigss}$/&gt;${genomesss}_contig_${num}/g&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${genomesss}.fasta

num=$((num+1))
done

#seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

done


ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/ |grep &quot;.fasta$&quot; |wc -l
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned |grep &quot;.fasta$&quot; |wc -l


#============================================================================
##Ldel
#============================================================================

##aligne all Illumina genomes


for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/|grep &quot;.fasta$&quot; |sed &#39;s/.fasta//g&#39;|grep &quot;24756&quot; -v |grep &quot;24786&quot; -v |grep &quot;24790&quot; -v )
do
echo &quot;==========================================================================&quot;
echo ${genomesss}
#seq_length.py /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/S.thermophilus/01_genomes/*${genomesss}.fasta |head -1
num=1

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/
rm -r ./aligned/${genomesss}_corrected
mkdir -p ./aligned/
ragtag.py correct 24790.fasta ${genomesss}.fasta -u -t 35  --inter -o ./aligned/${genomesss}_corrected

done



##---------------------------------
##bring together
##---------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned

##--------------------------Illuminas
  for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/ |grep &quot;.fasta$&quot;|cut -d &#39;.&#39; -f 1)
  do
  echo &quot;--------------------------------------------&quot;
  #sed &quot;s/&gt;contig/&gt;${genomesss}_contig/g&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta
  
 cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel//aligned/${genomesss}_corrected/*.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta
  
num=1
for contigss in $(grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel//aligned/${genomesss}_corrected/*.fasta|sed &#39;s/&gt;//g&#39;)
do

  sed -i &quot;s/&gt;${contigss}$/&gt;${genomesss}_contig_${num}/g&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta
  
#sed &quot;s/&gt;${contigss}/&gt;${genomesss}_contig_${num}/g&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta

num=$((num+1))
done
    #seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/${genomesss}.fasta
    done

###ONT


for genomesss in $(echo &quot;24756
24786
24790 24754 24798&quot;)
do
echo &quot;--------------------------------------------&quot;
echo ${genomesss}
#sed &quot;s/&gt;contig/&gt;${genomesss}_contig/g&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta &gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta

num=1
#seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/aligned/8_fourthFreebayes_startAligned.fasta
for contigs in $(grep &quot;&gt;&quot;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta |sed &#39;s/&gt;//g&#39;)
  do

 sed -i &quot;s/${contigs}/${genomesss}_contig_${num}/g&quot;   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta

num=$((num+1))

done
seq_length.py  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta

done

####---------------------------------
#contaminatino removal
####---------------------------------



for genomesss in $(echo &quot;21748
21753
21780
11077&quot;)
do
echo &quot;--------------------------------------------&quot;
echo ${genomesss}
#sed &quot;s/&gt;contig/&gt;${genomesss}_contig/g&quot; /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${genomesss}/*fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/${genomesss}.fasta

rm  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta

done

###--------------------------------
##change names with letters
###--------------------------------
species=Ldel
 for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/ |grep &quot;.fasta$&quot;|cut -d &#39;.&#39; -f 1|grep &quot;^L&quot; -v)
  do
  echo &quot;--------------------------------------------&quot;
mv  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/${genomesss}.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Ldel/210315_final_startAligned/L${genomesss}.fasta
  
    done
species=Sterm
 for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/ |grep &quot;.fasta$&quot;|cut -d &#39;.&#39; -f 1|grep &quot;^S&quot; -v)
  do
  echo &quot;--------------------------------------------&quot;
mv  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${genomesss}.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/S${genomesss}.fasta
  
    done


###--------------------------------
##count
###--------------------------------

 for species in $(echo &quot;Sterm Ldel&quot;)
  do
  echo &quot;--------------------------------------------&quot;
  echo ${species}
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/ |grep &quot;.fasta$&quot; -c
  
    done

###--------------------------------
##write names
###--------------------------------
rm   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/names_fam_strains.txt
 for species in $(echo &quot;Sterm Ldel&quot;)
  do
  echo &quot;--------------------------------------------&quot;
  echo ${species}
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/ |grep &quot;.fasta$&quot; |grep -v &quot;S50&quot;|grep -v &quot;S72&quot;|grep -v &quot;MAG&quot; |sed &#39;s/^S/FAM/g&#39; |sed &#39;s/^L/FAM/g&#39; |sed &#39;s/.fasta//g&#39; &gt;&gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/names_fam_strains.txt
  
    done
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/names_fam_strains.txt</code></pre>
<p>bring local</p>
<pre class="bash"><code>mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/names_fam_strains.txt ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/
</code></pre>
<p>there are still some undiscussed issues:</p>
<ul>
<li>The pipeline does not rearrange the contigs if they are in the wrong
orientation.</li>
<li>the dnaA often is in the middle of the contig. It should therefore
be cut and the contigs devided</li>
</ul>
</div>
<div id="ncbi-annotation" class="section level3">
<h3>NCBI annotation</h3>
<p>Here, I transfer the final genome assemblies locally to after
annotate and submit them to NCBI.</p>
<pre class="bash"><code>
##---------transfer local</code></pre>
<p>subset and remove all RMK202 genome because they have already been
submitted.</p>
<pre class="bash"><code>###---------------------------------
#split complete and non-complete genome assemblies
###---------------------------------

###ONTs

rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/{ONT,Illumina}
for geneomesss in $(sed &#39;1d&#39; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal///genomeAssemblyStats.csv |awk -F &quot;,&quot; &#39;{if($3!=&quot;202&quot;&amp;&amp; $16==&quot;Y&quot;)print $1}&#39;)
do
echo &quot;===================================================&quot;
echo ${geneomesss}
cp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Ldel/${geneomesss}.fna /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/

cp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Sterm/${geneomesss}.fna /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/


done

##Ilumina

for geneomesss in $(sed &#39;1d&#39; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal///genomeAssemblyStats.csv |awk -F &quot;,&quot; &#39;{if($3!=&quot;202&quot;&amp;&amp; $16!=&quot;Y&quot;)print $1}&#39;)
do
echo &quot;===================================================&quot;
echo ${geneomesss}
cp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Ldel/${geneomesss}.fna /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/

cp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Sterm/${geneomesss}.fna /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/


done


ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/ |wc -l 
ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/ |wc -l 

sed &#39;1d&#39; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal///genomeAssemblyStats.csv |awk -F &quot;,&quot; &#39;{if($3!=&quot;202&quot;)print $1}&#39; |wc -l</code></pre>
<p>add necessary information to contigs</p>
<pre class="bash"><code>###---------------------------------
#change chromosome names
###---------------------------------

for geneomesss in $(ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/)
do
echo &quot;===================================================&quot;
echo ${geneomesss}

grep &quot;&gt;&quot; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss}

seqlength.py /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss}

sed &#39;s/_contig_1/_contig_1 [location=chromosome] [topology=circular] [completeness=complete]/g&#39;  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss} | sed &#39;s/_contig_2/_contig_2 [plasmid-name=unnamed1]/g&#39; | sed &#39;s/_contig_3/_contig_3 [plasmid-name=unnamed2]/g&#39;  &gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss}_02 
echo &quot;----&quot;


mv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss}_02  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss}

seqlength.py /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/${geneomesss}

done

rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/ONT/*_02


##-----------------------------------
#Illumina remove short contigs
##-----------------------------------
conda activate seqkits
genomesss=24753
rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/*_02.fna
for genomesss in $(ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/ |grep &quot;_02.fna&quot; -v |sed &#39;s/.fna//g&#39;)
do
  echo ==========================
   echo ${genomesss}
 #  echo ==========================
 seqlength.py /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina//${genomesss}.fna  |tail -5

  seqkit seq -m 500 /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina//${genomesss}.fna -o /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina//${genomesss}_02.fna
 
 
 mv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina//${genomesss}_02.fna /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina//${genomesss}.fna
# seqlength.py /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina//${genomesss}_02.fna  |tail -5
 
 
 
    done

    </code></pre>
<p>after NCBI annotation</p>
<p>download all genomes that are annotate and labeled by NCBI.</p>
<p>First you make a dataframe that assigns the donwload name to the
accession and the sample name.</p>
<pre class="bash"><code>
##------------------------------------------
#create

rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_rmk202_isolates.txt
for genomesss in $(ls /home/vincent/Downloads/genome_assemblies_genome_gb/ncbi-genomes-2021-03-25 |grep &quot;gbff&quot; |sed  &#39;s/_genomic.gbff.gz//g&#39; )
do

acesssionss=$(zcat /home/vincent/Downloads/genome_assemblies_genome_gb/ncbi-genomes-2021-03-25/${genomesss}* |head -1| sed &#39;s/ \+/\t/g&#39; |cut -f 2)

acesssionss_short=$(zcat /home/vincent/Downloads/genome_assemblies_genome_gb/ncbi-genomes-2021-03-25/${genomesss}* |head -1| sed &#39;s/ \+/\t/g&#39; |cut -f 2|sed &#39;s/0100.*//g&#39;)

strain_name_short=$(grep &quot;${acesssionss_short}&quot; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/genome_info_rmk202_isolates.csv |cut -f 4)

if [ $strain_name_short == &quot;24777&quot; ] 
then     

 strain_name_long=$(echo &quot;L24777&quot;)
 
 elif [ $strain_name_short == &quot;13499c1&quot; ] 
then     

 strain_name_long=$(echo &quot;S13499c1&quot;)
  elif [ $strain_name_short == &quot;13499c2&quot; ] 
then     

 strain_name_long=$(echo &quot;S13499c2&quot;)
 else
 
 strain_name_long=$(ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Sterm/ /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/Ldel/ |grep &quot;.fna&quot;|sed &#39;s/.fna//g&#39;|grep &quot;${strain_name_short}&quot;)
 
fi

echo -e ${genomesss}&quot;\t&quot;${acesssionss}&quot;\t&quot;${strain_name_short}&quot;\t&quot;${strain_name_long} &gt;&gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_rmk202_isolates.txt

done

cat /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_rmk202_isolates.txt


###--------------------
#fix 1
###--------------------

samtools faidx /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775.fna

grep &quot;&gt;&quot; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775.fna |sed &#39;s/&gt;//g&#39; |grep -v &quot;contig_30$&quot; |grep -v &quot;contig_52$&quot; |grep -v &quot;contig_94$&quot; &gt;  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/fix_01.txt
rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775_02.fna
for genomesss in $(cat /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/fix_01.txt )
do

samtools faidx /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775.fna ${genomesss} &gt;&gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775_02.fna

done

mv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775_02.fna /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/submit_final/Illumina/L24775.fna</code></pre>
<p>Next I relabel and arrange the different files.</p>
<pre class="bash"><code>###===================================================================
#RMK202 isoaltes
###===================================================================
batch=RMK202
tablesss=/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_rmk202_isolates.txt
##----------------------------GBK--------------------------------------
output=gbf
input_location=/home/vincent/Downloads/genome_assemblies_genome_gb/ncbi-genomes-2021-03-25/

rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/

for genomess in $(cat ${tablesss} |cut -f 1)
do
sampleName=$(cat ${tablesss}|grep ${genomess}|cut -f 4)

zcat ${input_location}/${genomess}* &gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/${sampleName}.${output}

done

##----------------------------FNA--------------------------------------

output=fna
input_location=/home/vincent/Downloads/genome_assemblies_genome_fasta/ncbi-genomes-2021-03-25

rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/

for genomess in $(cat ${tablesss} |cut -f 1)
do
sampleName=$(cat ${tablesss}|grep ${genomess}|cut -f 4)

zcat ${input_location}/${genomess}* &gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/${sampleName}.${output}

done

##----------------------------GFF--------------------------------------

output=gff
input_location=/home/vincent/Downloads/genome_assemblies_genome_gff/ncbi-genomes-2021-03-25

rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/

for genomess in $(cat ${tablesss} |cut -f 1)
do
sampleName=$(cat ${tablesss}|grep ${genomess}|cut -f 4)

zcat ${input_location}/${genomess}* &gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/${sampleName}.${output}

done

##----------------------------FFN--------------------------------------

output=ffn
input_location=/home/vincent/Downloads/genome_assemblies_cds_fasta/ncbi-genomes-2021-03-25

rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/

for genomess in $(cat ${tablesss} |cut -f 1)
do
sampleName=$(cat ${tablesss}|grep ${genomess}|cut -f 4)

zcat ${input_location}/${genomess}* &gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/${sampleName}.${output}

done

##----------------------------FAA--------------------------------------

output=faa
input_location=/home/vincent/Downloads/genome_assemblies_prot_fasta_(1)/ncbi-genomes-2021-03-25

rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/

for genomess in $(cat ${tablesss} |cut -f 1)
do
sampleName=$(cat ${tablesss}|grep ${genomess}|cut -f 4)

zcat ${input_location}/${genomess}* &gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/${output}/${sampleName}.${output}

done
</code></pre>
<p>Here, I download the genome annotations that are in the submission
portal.</p>
<pre class="bash"><code>##-------------------------------------------------
##ONT genomes
##-------------------------------------------------
###------------creat accession table

tablesss=/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_ONT_isolates.txt


#vim /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_ONT_downloading
###-------------download
batch=ONT
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/


for genomesss in $(sed &#39;1d&#39; ${tablesss} |cut -f 2 |sed &#39;s/S/s/g&#39;|sed &#39;s/L/l/g&#39;)
do
downloadsss=$(grep &quot;$genomesss&quot; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_ONT_downloading |grep &quot;.gb/?format=attachment&quot; )

wget --content-disposition https: $downloadsss
done

##---------------rename

for genomesss in $(sed &#39;1d&#39; ${tablesss} |cut -f 2 )
do

cp ${genomesss}_*.gb ${genomesss}.gbf 

done
rm *.gb

##--------------------genacnk 2 gff
## gff without fasta
## and fasta only file
batch=ONT
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/

for genomesss in $(sed &#39;1d&#39; ${tablesss} |cut -f 2  )
do
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/


bp_genbank2gff3 --split --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/  ${genomesss}.gbf 

rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/${genomesss}.fna

rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff

for contigsss in $(ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/  |cut -d &#39;.&#39; -f 1|sort|uniq )
do
#contig_names=$(echo $contigsss |sed &#39;s/.fa//g&#39;)
##--------fna
samtools faidx /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/${contigsss}.fa
samtools faidx /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/${contigsss}.fa ${contigsss} &gt;&gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/${genomesss}.fna

##--------gff without fasta
cat /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/${contigsss}.gff &gt;&gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff

done #contigssss

#bp_genbank2gff3 --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/  ${genomesss}.gbf 

done
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/


##--------------------genacnk 2 gff (including fasta)

batch=ONT
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/

for genomesss in $(sed &#39;1d&#39; ${tablesss} |cut -f 2  )
do
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/


bp_genbank2gff3 --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/  ${genomesss}.gbf 

mv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/${genomesss}.gbf.gff /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/${genomesss}.gff 

done
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/</code></pre>
<p>Next up are the Illumina assemblies</p>
<pre class="bash"><code>##-------------------------------------------------
##Illumina genomes
##-------------------------------------------------


tablesss=/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_Illumina_isolates.txt

###-------------download
batch=Illumina
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/






for genomesss in $(sed &#39;1d&#39; ${tablesss} |cut -f 2 |sed &#39;s/S/s/g&#39;|sed &#39;s/L/l/g&#39;)
do
downloadsss=$(grep &quot;$genomesss&quot; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_Illumina_downloading |grep &quot;.gb/?format=attachment&quot; )

wget --content-disposition https: $downloadsss
done

##---------------rename &amp; include accession

for genomesss in $(sed &#39;1d&#39; ${tablesss} |cut -f 2 )
do

#cp ${genomesss}.*.gb ${genomesss}.gbf 
sed &quot;s/ACCESSION/ACCESSION   ${genomesss}/g&quot; ${genomesss}.*.gb &gt; ${genomesss}.gbf 
done
rm *.gb


##--------------------genacnk 2 gff
batch=Illumina
tablesss=/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_Illumina_isolates.txt


rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/

for genomesss in $(sed &#39;1d&#39; ${tablesss} |cut -f 2 |head -1 )
do
echo $genomesss
java -jar /home/vincent/apps/jvarkit/dist/gb2gff.jar  ~/Downloads/test_gb

done

wget -O - -q  &quot;https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&amp;id=AF338247.1,D38149.1&amp;rettype=gbwithparts&amp;retmode=xml&amp;api_key=62b713e0cd85e6ac79699ecdfa72e85af009&quot;  |java -jar /home/vincent/apps/jvarkit/dist/gb2gff.jar  
##--------------------genacnk 2 gff
## gff without fasta
## and fasta only file
batch=Illumina
tablesss=/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/log/names_accessions_Illumina_isolates.txt


rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/

for genomesss in $(sed &#39;1d&#39; ${tablesss} |cut -f 2  )
do
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/


bp_genbank2gff3 --split --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/  ${genomesss}.gbf 


#bp_genbank2gff3 --nolump --typesource LOCUS --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/  ${genomesss}.gbf 


rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/${genomesss}.fna

rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff

for contigsss in $(ls /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/  |cut -d &#39;.&#39; -f 1|sort|uniq )
do
#contig_names=$(echo $contigsss |sed &#39;s/.fa//g&#39;)
##--------fna
samtools faidx /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/${contigsss}.fa
samtools faidx /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/${contigsss}.fa ${contigsss} &gt;&gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/${genomesss}.fna

##--------gff without fasta
cat /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/${contigsss}.gff &gt;&gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff

done #contigssss

#bp_genbank2gff3 --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/  ${genomesss}.gbf 

done
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/fna/


##--------------------genacnk 2 gff (including fasta)

batch=Illumina
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gbf/

for genomesss in $(sed &#39;1d&#39; ${tablesss} |cut -f 2  )
do
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/tmp/


bp_genbank2gff3 --GFF_VERSION 3 --outdir /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/  ${genomesss}.gbf

mv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/${genomesss}.gbf.gff /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/${genomesss}.gff 

done
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/

##---------------------gff without fasta

batch=Illumina
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/
for genomesss in $(sed &#39;1d&#39; ${tablesss} |cut -f 2  )
do
grep &quot;##FASTA&quot; -B 1000000 /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff_fna/${genomesss}.gff &gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff 
rowss=$(grep &quot;##FASTA&quot; -n  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff |cut -d &quot;:&quot; -f 1)
grep &quot;##FASTA&quot; -n  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff
sed -i &quot;${rowss}d&quot; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff 
grep &quot;##FASTA&quot; -n  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/${genomesss}.gff

done
ll /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/${batch}/gff/

##---------------------download acc file to rename contig names



##---------------------fasta



scp -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/
</code></pre>
</div>
</div>
<div id="poppunk" class="section level2">
<h2>Poppunk</h2>
<p>Here, I use <a href="https://poppunk.readthedocs.io/en/latest/model_fitting.html">poppunk</a>
to identify genome clusters</p>
<pre class="bash"><code>
###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Sterm

#for species in $(echo &quot;Ldel Sterm&quot;)
#do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/FNA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/
#done


##====================================================================
##only RMKs
##====================================================================



##--------------------------
##create genome list with path to genome assembly
##--------------------------
species=Sterm
rm  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db_onlyRMK.log
mkdir -p  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA | grep &quot;.fna$&quot;|grep -v &quot;^S-&quot; | sed &#39;s/.fna//g&#39;|grep &quot;Out&quot; -v)
do

genomesss_short=$(echo $genomesss |sed &#39;s/-//g&#39;|sed &#39;s/_//g&#39;)

#echo -e ${genomesss}&quot;\t/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/&quot;${species}&quot;/&quot;${genomesss}&quot;.fna&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}&quot;\t&quot;${genomesss}&quot;.fna&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}&quot;.fna&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
echo -e &quot;/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/&quot;${species}&quot;/01_all_FNA/&quot;${genomesss}&quot;.fna&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db_onlyRMK.log

done


##--------------------------
#easy run 
##--------------------------
rm -r /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun
poppunk --easy-run --r-files /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db_onlyRMK.log --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun --threads 4 --plot-fit 5 --min-k 13 --full-db

poppunk --fit-model --distances /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun.dists --ref-db /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/ --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/ --full-db --K 3


sed &#39;s;\/data\/Project\/2020_StarterCultureDiversity\/10_REFERENCE_GEN0MES\/annotation\/PROKKA\/Sterm\/01_all_FNA\/;;g&#39; /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters.csv |sed &#39;s/.fna//g&#39; |cut -d &#39;_&#39; -f 4 &gt;  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv</code></pre>
<p>transfer local</p>
<pre class="bash"><code>
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/fit*.pdf /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/*.png /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/



mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/*.png /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv
</code></pre>
<pre class="bash"><code>#==========================================================
#lDEL
#==========================================================
###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Ldel

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/

leterzzz=$(echo $species |head -c 1)

#cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/FNA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/

##HERE I remove the genomes with completness &lt;95 or duplication &gt;10
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/L_I_101_21748.fna
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/L_I_124_21753.fna
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/Ldel/L_I_101_21780.fna

#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/L_I_101_21748.fna
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/L_I_124_21753.fna
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/Ldel/L_I_101_21780.fna
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/Ldel/11077.fna



##--------------------------
##create genome list with path to genome assembly
##--------------------------

species=Ldel
rm  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/${species}_poppunk_db.log
mkdir -p  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/${species}/ | grep &quot;.fna$&quot;|sed &#39;s/.fna//g&#39;|grep &quot;Out&quot; -v)
do

genomesss_short=$(echo $genomesss |sed &#39;s/-//g&#39;|sed &#39;s/_//g&#39;)

#echo -e ${genomesss}&quot;\t/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FNA/&quot;${species}&quot;/&quot;${genomesss}&quot;.fna&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}&quot;\t&quot;${genomesss}&quot;.fna&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
#echo -e ${genomesss}&quot;.fna&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/Sterm_poppunk_db.log
echo -e &quot;/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own_popPunk/FNA/&quot;${species}&quot;/&quot;${genomesss}&quot;.fna&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/${species}_poppunk_db.log

done



##--------------------------
#easy run 
##--------------------------
species=Ldel

rm -r /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}
cd /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/

poppunk --easy-run --ignore-length --r-files /data/Project/2020_StarterCultureDiversity/06_PopPUNK/log/${species}_poppunk_db.log --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun --threads 4 --plot-fit 5 --min-k 21 --full-db

poppunk --fit-model --distances /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/20210128_easyrun.dists --ref-db /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/ --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/ --full-db --K 2
#PopPUNK/${species}/20210128_easyrun/ --output /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/ --full-db --K 6



sed &#39;s;\/data\/Project\/2020_StarterCultureDiversity\/10_REFERENCE_GEN0MES\/genomes\/combined_ncbi_own_popPunk\/FNA\/Ldel\/;;g&#39; /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/20210128_easyrun_clusters.csv |sed &#39;s/.fna//g&#39; &gt;  /data/Project/2020_StarterCultureDiversity/06_PopPUNK/${species}/20210128_easyrun/20210128_easyrun_clusters_RMKS_ldel.csv
</code></pre>
<p>transfer local</p>
<pre class="bash"><code>
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/fit*.pdf /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_database/

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/*.png /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm/20210128_easyrun/



mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/*.png /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/06_PopPUNK/Ldel/20210128_easyrun/*.csv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/
</code></pre>
</div>
<div id="fastani" class="section level2">
<h2>fastANI</h2>
<pre class="bash"><code>
##------------new PREP
speciesss=Sterm
##-------------------------------------------------



rm -r /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/{log,analysis}
echo &quot;-----------------&quot;
echo ${speciesss}

mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/{log,analysis}
for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ |grep &quot;.fna$&quot;)
do
echo -e &quot;/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/&quot;${speciesss}&quot;/01_all_FNA/&quot;${genomessss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt

done

###---------------------------
##fastaANI
###---------------------------
#for speciesss in $(echo &quot;L.delbrueckii S.thermophilus&quot;)
for speciesss in $(echo &quot;Sterm&quot;)
do
fastANI --fragLen 1000 -t 6 --ql /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt \
  --rl /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt \
  -o /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out
done #speciesss
###----------------------Sterm


cut -f 3 /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out |sort|uniq -c |head
grep &quot;74\.&quot; /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out


##------------new PREP
speciesss=Ldel
##-------------------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/{log,analysis}
echo &quot;-----------------&quot;
echo ${speciesss}

mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/{log,analysis}
for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ |grep &quot;.fna$&quot;)
do
echo -e &quot;/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/&quot;${species}&quot;/01_all_FNA/&quot;${genomessss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt

done

###---------------------------
##fastaANI
###---------------------------
#for speciesss in $(echo &quot;L.delbrueckii S.thermophilus&quot;)
for speciesss in $(echo &quot;Ldel&quot;)
do
fastANI --fragLen 1000 -t 6 --ql /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt \
  --rl /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/log/genomes.txt \
  -o /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out
done #speciesss
###----------------------Sterm


cut -f 3 /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out |sort|uniq -c |head
grep &quot;74\.&quot; /data/Project/2020_StarterCultureDiversity/05_fastANI/${speciesss}/analysis/analysis.out

##====================================
##all against all (independent of species)
##====================================
rm -r /data/Project/2020_StarterCultureDiversity/05_fastANI/all
mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/all/{log,analysis}
for speciesss in $(echo &quot;L.delbrueckii S.thermophilus&quot;)
do

echo &quot;-----------------&quot;
echo ${speciesss}

mkdir -p /data/Project/2020_StarterCultureDiversity/05_fastANI/all/{log,analysis}
for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes)
do
echo -e &quot;/data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/&quot;${speciesss}&quot;/01_genomes/&quot;${genomessss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/05_fastANI/all/log/genomes.txt

done
done #speciesss
###---------------------------
##fastaANI
###---------------------------

fastANI --fragLen 1000 -t 37 --ql /data/Project/2020_StarterCultureDiversity/05_fastANI/all/log/genomes.txt \
  --rl /data/Project/2020_StarterCultureDiversity/05_fastANI/all/log/genomes.txt \
  -o /data/Project/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis.out



cut -f 3 /data/Project/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis.out |sort|uniq -c |head
</code></pre>
<pre class="bash"><code>mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis.out /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/05_fastANI/Sterm/analysis/analysis.out /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis_sterm.out
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/05_fastANI/Ldel/analysis/analysis.out /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis_ldel.out</code></pre>
<pre class="r"><code>library(readr)
library(tidyverse)
library(ggpubr)
library(ggsignif)
analysis_FASTANI &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis_sterm.out&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;genome1&quot;,&quot;genome2&quot;,&quot;ANI&quot;,&quot;comparabedWindows&quot;,&quot;totalWindows&quot;), trim_ws = TRUE)

analysis_FASTANI$species1 &lt;- str_split_fixed(analysis_FASTANI$genome1,&quot;/&quot;,11)[,8]
analysis_FASTANI$species2 &lt;- str_split_fixed(analysis_FASTANI$genome2,&quot;/&quot;,11)[,8]

analysis_FASTANI$GENOME1 &lt;- str_split_fixed(analysis_FASTANI$genome1,&quot;/&quot;,10)[,10] %&gt;% gsub(&quot;.fna&quot;,&quot;&quot;,.)
analysis_FASTANI$GENOME2 &lt;- str_split_fixed(analysis_FASTANI$genome2,&quot;/&quot;,10)[,10] %&gt;% gsub(&quot;.fna&quot;,&quot;&quot;,.)


analysis_FASTANI$comparision &lt;- ifelse(analysis_FASTANI$species1==analysis_FASTANI$species2,&quot;same species&quot;,&quot;different species&quot;)
table(analysis_FASTANI$comparision)

analysis_FASTANI %&gt;%group_by(comparision)  %&gt;% select(ANI) %&gt;%summarize(meansample=mean(ANI))

# sample_relative_temp  %&gt;% group_by(X.CHROM) %&gt;% summarize(meansample=mean(Sequencing_depth))


analysis_FASTANI_final_sterm &lt;- analysis_FASTANI %&gt;% select(!c(genome1,genome2))
###----------------------
##whithin an between clusters
###----------------------

analysis_clusters &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv&quot;,  &quot;,&quot;)

analysis_FASTANI_final02 &lt;- merge(analysis_FASTANI_final_sterm,analysis_clusters,by.x=&quot;GENOME1&quot;,by.y=&quot;Taxon&quot;)
analysis_FASTANI_final03 &lt;- merge(analysis_FASTANI_final02,analysis_clusters,by.x=&quot;GENOME2&quot;,by.y=&quot;Taxon&quot;)

analysis_FASTANI_final03$clustercomparison &lt;- ifelse(analysis_FASTANI_final03$Cluster.x==analysis_FASTANI_final03$Cluster.y,&quot;same cluster&quot;,&quot;different cluster&quot;)
analysis_FASTANI_final03$cov &lt;- 100*(analysis_FASTANI_final03$comparabedWindows/analysis_FASTANI_final03$totalWindows)
analysis_FASTANI_sterm &lt;- analysis_FASTANI_final03
plotANI &lt;- ggplot(analysis_FASTANI_final03,aes(x=clustercomparison,y=ANI))+geom_boxplot()+theme_classic()+lims(y=c(98.5,100))+stat_compare_means(label = &quot;p.signif&quot;)+labs(x=&quot;&quot;,y=&quot;pairwise ANI&quot;)
# ggplot(analysis_FASTANI_final03,aes(x=cov,y=ANI,color=clustercomparison,fill=clustercomparison))+geom_point()+theme_classic()
plotANI 

ggplot(analysis_FASTANI_final03,aes(x=ANI))+geom_density()+theme_classic()+lims(x=c(98.5,100))



# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/ANI_genomes_boxplot.pdf&quot;,width=3,height=6)
# 
# plotANI
# 
# dev.off()

##---------------------------------------------------------
#LDEL
##---------------------------------------------------------

library(readr)
library(tidyverse)
library(ggpubr)
library(ggsignif)
analysis_FASTANI &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/05_fastANI/all/analysis/analysis_ldel.out&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;genome1&quot;,&quot;genome2&quot;,&quot;ANI&quot;,&quot;comparabedWindows&quot;,&quot;totalWindows&quot;), trim_ws = TRUE)

analysis_FASTANI$species1 &lt;- str_split_fixed(analysis_FASTANI$genome1,&quot;/&quot;,11)[,8]
analysis_FASTANI$species2 &lt;- str_split_fixed(analysis_FASTANI$genome2,&quot;/&quot;,11)[,8]

analysis_FASTANI$GENOME1 &lt;- str_split_fixed(analysis_FASTANI$genome1,&quot;/&quot;,10)[,10] %&gt;% gsub(&quot;.fna&quot;,&quot;&quot;,.)
analysis_FASTANI$GENOME2 &lt;- str_split_fixed(analysis_FASTANI$genome2,&quot;/&quot;,10)[,10] %&gt;% gsub(&quot;.fna&quot;,&quot;&quot;,.)


analysis_FASTANI$comparision &lt;- ifelse(analysis_FASTANI$species1==analysis_FASTANI$species2,&quot;same species&quot;,&quot;different species&quot;)
table(analysis_FASTANI$comparision)

analysis_FASTANI %&gt;%group_by(comparision)  %&gt;% select(ANI) %&gt;%summarize(meansample=mean(ANI))

# sample_relative_temp  %&gt;% group_by(X.CHROM) %&gt;% summarize(meansample=mean(Sequencing_depth))


analysis_FASTANI_final_ldel &lt;- analysis_FASTANI %&gt;% select(!c(genome1,genome2))
###----------------------
##whithin an between clusters
###----------------------

analysis_clusters &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_ldel.csv&quot;,  &quot;,&quot;)

analysis_FASTANI_final02 &lt;- merge(analysis_FASTANI_final_ldel,analysis_clusters,by.x=&quot;GENOME1&quot;,by.y=&quot;Taxon&quot;)
analysis_FASTANI_final03 &lt;- merge(analysis_FASTANI_final02,analysis_clusters,by.x=&quot;GENOME2&quot;,by.y=&quot;Taxon&quot;)

analysis_FASTANI_final03$clustercomparison &lt;- ifelse(analysis_FASTANI_final03$Cluster.x==analysis_FASTANI_final03$Cluster.y,&quot;same cluster&quot;,&quot;different cluster&quot;)
analysis_FASTANI_final03$cov &lt;- 100*(analysis_FASTANI_final03$comparabedWindows/analysis_FASTANI_final03$totalWindows)

analysis_FASTANI_ldel &lt;- analysis_FASTANI_final03
plotANI &lt;- ggplot(analysis_FASTANI_final03,aes(x=clustercomparison,y=ANI))+geom_boxplot()+theme_classic()+lims(y=c(98.5,100))+stat_compare_means(label = &quot;p.signif&quot;)+labs(x=&quot;&quot;,y=&quot;pairwise ANI&quot;)
# ggplot(analysis_FASTANI_final03,aes(x=cov,y=ANI,color=clustercomparison,fill=clustercomparison))+geom_point()+theme_classic()
plotANI 



ggplot(analysis_FASTANI_final03,aes(x=ANI))+geom_density()+theme_classic()+lims(x=c(98.5,100))


analysis_FASTANI_together %&gt;% dplyr::group_by(comparision,species1) %&gt;% dplyr::summarise(min=min(ANI))


##---------------------------------------------------------
#toether
##---------------------------------------------------------

analysis_FASTANI_together &lt;- rbind(analysis_FASTANI_sterm,analysis_FASTANI_ldel)


analysis_FASTANI_together$species1  &lt;- plyr::revalue(analysis_FASTANI_together$species1,c(&quot;Ldel&quot;=&quot;L.delbrueckii&quot;,&quot;Sterm&quot;=&quot;S.thermophilus&quot;))
all_colours &lt;- (c(&quot;#fec3b7ff&quot;,&quot;#0081a7ff&quot;,&quot;grey70&quot;,&quot;grey88&quot;) ) 
densityPlot &lt;- ggplot(analysis_FASTANI_together,aes(x=ANI,color=species1,fill=species1))+geom_density(alpha=0.4)+theme_classic()+lims(x=c(98.5,100))+geom_vline(xintercept = 99.75)+scale_fill_manual(values=all_colours)+scale_color_manual(values=all_colours)+theme(legend.position = &quot;none&quot;)+labs(x=&quot;ANI&quot;,y=&quot;density&quot;)
densityPlot
colorsss &lt;- c(&quot;grey77&quot;,&quot;orange&quot;)
plotANI &lt;- ggplot(analysis_FASTANI_together,aes(x=clustercomparison,y=ANI,fill=clustercomparison))+geom_boxplot()+theme_classic()+lims(y=c(98.5,100))+stat_compare_means(label = &quot;p.signif&quot;)+labs(x=&quot;&quot;,y=&quot;pairwise ANI&quot;)+facet_wrap(~species1,ncol = 1, strip.position=&quot;right&quot;)+scale_fill_manual(values = colorsss)+theme(legend.title = element_blank(),legend.position = &quot;top&quot;,axis.title = element_blank(),axis.text = element_blank())+ coord_flip()
plotANI 

library(patchwork)
plotANI +densityPlot+ plot_layout(nrow=2,heights = c(1,2))
pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/ANI_genomes_boxplot.pdf&quot;,width=7,height=7)

plotANI +densityPlot+ plot_layout(nrow=2,heights = c(1,2))

dev.off()</code></pre>
</div>
</div>
<div id="pseudogene-analysis" class="section level1">
<h1>Pseudogene analysis</h1>
<div id="identify-pseudogenes" class="section level2">
<h2>Identify pseudogenes</h2>
<p>In order to identify the pseudogene count we first need to download
the pgap annotations from the opengenomebrowser (OGB).</p>
<pre class="bash"><code>##=====================================
#get list of FAM genomes
##====================================

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log_annoation.txt
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/
for species in $(echo &quot;Ldel Sterm&quot;)
do

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/ |grep &quot;.faa$&quot;|grep &quot;MAG&quot; -v |grep &quot;S50&quot; -v |grep &quot;S72&quot; -v  |sed &#39;s/.faa//g&#39; |sed &#39;s/^S//g&#39;|sed &#39;s/^L//g&#39;|sed &quot;s/^/\\&#39;FAM/g&quot; |sed &quot;s/$/\\&#39;/g&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log_annoation.txt

done


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log_annoation.txt| tr &#39;\n&#39; &#39;,&#39;
</code></pre>
<pre class="bash"><code>
scp /home/vincent/test/download_ogb_gbk.py vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/
</code></pre>
<p>donwload the genome after filing in the FAM names.</p>
<pre class="bash"><code>rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK

sed  &quot;s/FAM24855/FAM24855&#39;,&#39;FAM12273/g&quot;  ../download_ogb_gbk.py  &gt; ../download_ogb_gbk_new.py

#,&#39;FAM24728&#39;

python ../download_ogb_gbk_new.py


##--------------
------------------
##move into correct location and rename
##--------------------------------

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK
for species in $(echo &quot;Ldel Sterm&quot;)
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/ |grep &quot;.faa$&quot;|grep &quot;MAG&quot; -v |grep &quot;S50&quot; -v |grep &quot;S72&quot; -v  |sed &#39;s/.faa//g&#39; |sed &#39;s/^S//g&#39;|sed &#39;s/^L//g&#39;)
do

echo ${genomesss}
firstCharacter=${species::1}

mv FAM${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/${firstCharacter}${genomesss}.gbk


done #genomesss

done


###-----Lhelv

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK


sed  &quot;s/FAM10973.*FAM24855/FAM11051&#39;,&#39;FAM11961&#39;,&#39;FAM21456&#39;,&#39;FAM21462&#39;,&#39;FAM21463/g&quot;  ../download_ogb_gbk.py &gt; ../download_ogb_gbk_lhelv.py

python ../download_ogb_gbk_lhelv.py
species=Lhelv
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}
for genomesss in $(echo &quot;FAM11051 FAM11961 FAM21456 FAM21462 FAM21463&quot;)
do

echo ${genomesss}
neuNAme=$(echo ${genomesss}|sed &#39;s/FAM/Lh/g&#39;)
mv ${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/${neuNAme}.gbk
done 

#==============================================
#GFF
#==============================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF

#python ../download_ogb_gff.py

sed  &quot;s/FAM24855/FAM24855&#39;,&#39;FAM12273/g&quot;  ../download_ogb_gff.py  &gt; ../download_ogb_gff_new.py
#cp  ../download_ogb_gbk.py   ../download_ogb_gff.py

#,&#39;FAM24728&#39;

python ../download_ogb_gff_new.py



for species in $(echo &quot;Ldel Sterm&quot;)
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${species}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/ |grep &quot;.faa$&quot;|grep &quot;MAG&quot; -v |grep &quot;S50&quot; -v |grep &quot;S72&quot; -v  |sed &#39;s/.faa//g&#39; |sed &#39;s/^S//g&#39;|sed &#39;s/^L//g&#39;)
do

echo ${genomesss}
firstCharacter=${species::1}

mv FAM${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${species}/${firstCharacter}${genomesss}.gff




done #genomesss

done

###-----Lhelv

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF

#cp ../download_ogb_gbk.py ../download_ogb_gff.py

sed  &quot;s/FAM10973.*FAM24855/FAM11051&#39;,&#39;FAM11961&#39;,&#39;FAM21456&#39;,&#39;FAM21462&#39;,&#39;FAM21463/g&quot;  ../download_ogb_gff.py &gt; ../download_ogb_gff_lhelv.py

python ../download_ogb_gff_lhelv.py
species=Lhelv
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${species}
for genomesss in $(echo &quot;FAM11051 FAM11961 FAM21456 FAM21462 FAM21463&quot;)
do

echo ${genomesss}
neuNAme=$(echo ${genomesss}|sed &#39;s/FAM/Lh/g&#39;)
mv ${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${species}/${neuNAme}.gff
done 
#==============================================
#ffn
#==============================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN

sed  &quot;s/FAM24855/FAM24855&#39;,&#39;FAM12273/g&quot;  ../download_ogb_ffn.py  &gt; ../download_ogb_ffn_new.py

python ../download_ogb_ffn_new.py


for species in $(echo &quot;Ldel Sterm&quot;)
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${species}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/ |grep &quot;.faa$&quot;|grep &quot;MAG&quot; -v |grep &quot;S50&quot; -v |grep &quot;S72&quot; -v  |sed &#39;s/.faa//g&#39; |sed &#39;s/^S//g&#39;|sed &#39;s/^L//g&#39;)
do

echo ${genomesss}

firstCharacter=${species::1}

mv FAM${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${species}/${firstCharacter}${genomesss}.ffn



done #genomesss

done


###-----Lhelv

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN


sed  &quot;s/FAM10973.*FAM24855/FAM11051&#39;,&#39;FAM11961&#39;,&#39;FAM21456&#39;,&#39;FAM21462&#39;,&#39;FAM21463/g&quot;  ../download_ogb_ffn.py &gt; ../download_ogb_ffn_lhelv.py

python ../download_ogb_ffn_lhelv.py
species=Lhelv
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${species}
for genomesss in $(echo &quot;FAM11051 FAM11961 FAM21456 FAM21462 FAM21463&quot;)
do

echo ${genomesss}
neuNAme=$(echo ${genomesss}|sed &#39;s/FAM/Lh/g&#39;)
mv ${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${species}/${neuNAme}.ffn
done 

#==============================================
#fna
#==============================================
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA

#python ../download_ogb_fna.py

sed  &quot;s/FAM24855/FAM24855&#39;,&#39;FAM12273/g&quot;  ../download_ogb_fna.py  &gt; ../download_ogb_fna_new.py

python ../download_ogb_fna_new.py



for species in $(echo &quot;Ldel Sterm&quot;)
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/ |grep &quot;.faa$&quot;|grep &quot;MAG&quot; -v |grep &quot;S50&quot; -v |grep &quot;S72&quot; -v  |sed &#39;s/.faa//g&#39; |sed &#39;s/^S//g&#39;|sed &#39;s/^L//g&#39;)
do

echo ${genomesss}

firstCharacter=${species::1}

mv FAM${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}/${firstCharacter}${genomesss}.fna


sed -i &#39;s/&gt;lcl|/&gt;/g&#39;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}/${firstCharacter}${genomesss}.fna

done #genomesss

done


###-----Lhelv

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA


sed  &quot;s/FAM10973.*FAM24855/FAM11051&#39;,&#39;FAM11961&#39;,&#39;FAM21456&#39;,&#39;FAM21462&#39;,&#39;FAM21463/g&quot;  ../download_ogb_fna.py &gt; ../download_ogb_fna_lhelv.py

python ../download_ogb_fna_lhelv.py
species=Lhelv
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}
for genomesss in $(echo &quot;FAM11051 FAM11961 FAM21456 FAM21462 FAM21463&quot;)
do

echo ${genomesss}
neuNAme=$(echo ${genomesss}|sed &#39;s/FAM/Lh/g&#39;)
mv ${genomesss}* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}/${neuNAme}.fna
sed -i &#39;s/&gt;lcl|/&gt;/g&#39;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${species}/${neuNAme}.fna

done 

for species in $(echo &quot;Ldel Sterm Lhelv&quot;)
do

ls ${species}/ |wc -l

done


#missings
#FAM12273
#FAM24728
#FAM24737
#FAM24738
#FAM24739
#FAM24740

for genomesss in $(cut -d &#39;,&#39; -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/all_genomes_dialact.txt|grep &quot;^S&quot;)
do
echo ${genomesss}
head -1  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/Sterm/${genomesss}.g*
echo &quot;-----------------------------&quot;

done</code></pre>
<p>extract the following information:</p>
<ul>
<li>coding genes</li>
<li>pseudogenes</li>
</ul>
<pre class="bash"><code>###========================================================
#all NCBI genomes download
###========================================================
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log 
curl &#39;ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt


###-----------------------
#Streptococcus suis
###-----------------------
species=$(echo &quot;Streptococcus suis&quot;)
shortnamess=Ssuis

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
#gunzip *.gbk.gz

###-----------------------
#Lactobacillus plantarum
###-----------------------
species=$(echo &quot;Lactobacillus plantarum&quot;)
shortnamess=Lplan

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

species=$(echo &quot;Lactiplantibacillus plantarum&quot;)

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;   \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

###-----------------------
#Baumannia cicadellinicola
###-----------------------
species=$(echo &quot;Baumannia&quot;)
shortnamess=Bcica

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}

##no hits

###-----------------------
#Rickettsia
###-----------------------
species=$(echo &quot;Rickettsia&quot;)
shortnamess=Rick

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot;|grep &quot;prowazekii&quot; -v|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

###-----------------------
#Portiera
###-----------------------
species=$(echo &quot;Portiera&quot;)
shortnamess=Port

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt



###-----------------------
#Sodalis glossinidius
###-----------------------
species=$(echo &quot;Sodalis&quot;)
shortnamess=Sglos

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt


###-----------------------
#Wigglesworthia
###-----------------------
species=$(echo &quot;Wigglesworthia&quot;)
shortnamess=Wigg

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt


###-----------------------
#Buchneria
###-----------------------
species=$(echo &quot;Buchnera&quot;)
shortnamess=Buch

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

###-----------------------
#Buchneria
###-----------------------
species=$(echo &quot;leprae&quot;)
shortnamess=Mleprae

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt


###-----------------------
#Staphylococcus gallinarum
###-----------------------
species=$(echo &quot;caprae&quot;)
shortnamess=Scap

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;Staphylococcus&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

###-----------------------
#lactobacillus equicursoris
###-----------------------
species=$(echo &quot;equicursoris&quot;)
shortnamess=Lequi

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot;|grep &quot;actobacillus&quot; -i |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

###-----------------------
#streptococcus vestibularis
###-----------------------
species=$(echo &quot;vestibularis&quot;)
shortnamess=Svest

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot;|grep &quot;streptococcus&quot; -i |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

###-----------------------
#Wolbachia
###-----------------------
species=$(echo &quot;Wolbachia&quot;)
shortnamess=Wol

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt
    #pecies_short=Sterm_references

wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_${shortnamess}.txt



###========================================================
#extract counts
###========================================================

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt
for species in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/)
do
echo $species

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${species} |grep &quot;.gz$&quot; )
do

genezzz=$(zcat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${species}/${genomesss}|grep &quot;Genes (coding)&quot; |cut -d &#39;:&#39; -f 3|sed &#39;s/ //g&#39;|sed &#39;s/,//g&#39;|head -1)
peudosss=$(zcat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${species}/${genomesss}|grep &quot;Pseudo Genes (total)&quot;  |cut -d &#39;:&#39; -f 3|sed &#39;s/ //g&#39;|sed &#39;s/,//g&#39;|head -1)
transpos=$(zgrep -i &quot;transpos&quot;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${species}/${genomesss} |grep &quot;product=&quot; -c)


done #genomesss

done
</code></pre>
<pre class="bash"><code>###-----------------------
#Sterm and Ldel
###-----------------------
for species in $(echo &quot;Sterm Ldel&quot;)
do
echo $species

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/ |grep &quot;.gbk$&quot; |sed &#39;s/.gbk//g&#39;)
do

genezzz=$(grep &quot;Genes (coding)&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/${genomesss}.gbk |cut -d &#39;:&#39; -f 3|sed &#39;s/ //g&#39;|sed &#39;s/,//g&#39;|head -1)
peudosss=$(grep &quot;Pseudo Genes (total)&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/${genomesss}.gbk |cut -d &#39;:&#39; -f 3|sed &#39;s/ //g&#39;|sed &#39;s/,//g&#39;|head -1)

transpos=$(grep &quot;transposase&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/${genomesss}.gbk |grep &quot;product=&quot; -c)


echo -e ${species}&quot;\t&quot;${genomesss}&quot;\t&quot;${genezzz}&quot;\t&quot;${peudosss}&quot;\t&quot;${transpos} &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt
done #genomesss

done</code></pre>
<p>transfer local</p>
<pre class="bash"><code>
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt</code></pre>
<p>do analysis</p>
<pre class="r"><code>library(readr)
library(tidyverse)
pseudogeness_count &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;species&quot;,&quot;genome&quot;,&quot;genes_coding&quot;,&quot;pseudogenes&quot;), trim_ws = TRUE) %&gt;% filter(pseudogenes&lt;1000) %&gt;% filter(genes_coding&lt;4000)%&gt;% mutate(pseudogenes_percent = 100*(pseudogenes/genes_coding)) %&gt;% filter(species!=&quot;Wigg&quot;)%&gt;% filter(species!=&quot;Scap&quot;)


plotpoints &lt;- ggplot(pseudogeness_count,aes(x=genes_coding,y=pseudogenes_percent,fill=species,color=species))+geom_point()+labs(x=&quot;Gene number&quot;,y=&quot;Pseudogenes [%]&quot;)+theme_classic()+theme(legend.title = element_blank())
plotpoints
  
  library(plotly)
ggp &lt;- ggplotly(plotpoints)
 ggp
# pseudogeness_count %&gt;% group_by(species)%&gt;%ize(meanPseudo=mean(pseudogeness_count),meangenes=mean(genes_coding),stdevPseudo=sd(pseudogeness_count),stdevgenes=stderr(genes_coding))
# pseudogeness_count %&gt;% group_by(species) %&gt;% summarise(meanPseudo=mean(pseudogeness_count))

pseudogeness_sumary &lt;- pseudogeness_count %&gt;% group_by(species) %&gt;%  dplyr::summarize(meanPseudo=100*(mean(pseudogenes)/mean(genes_coding)),meangenes=mean(genes_coding),stdevPseudo=100*(sd(pseudogenes)/mean(genes_coding)),stdevgene=sd(genes_coding))



 # dplyr::summarize(mean = mean(NumGenesLineage),sd=sd(NumGenesLineage))

##-----
#chnage names
##-----

  pseudogeness_sumary$species_new &lt;- plyr::revalue(pseudogeness_sumary$species,c(&quot;Buch&quot;=&quot;Buchnera spp.&quot;,&quot;Sglos&quot;=&quot;S. glossinidius&quot;,&quot;Ldel&quot;=&quot;L. delbrueckii&quot;,&quot;Sterm&quot;=&quot;S. thermophilus&quot;,&quot;Mleprae&quot;=&quot;M. leprae&quot;,&quot;Rick&quot;=&quot;Rickettsia spp.&quot;,&quot;Lplan&quot;=&quot;L. plantarum&quot;,&quot;Ssuis&quot;=&quot;S. suis&quot;,&quot;Lequi&quot;=&quot;L. equicursoris&quot;,&quot;Svest&quot;=&quot;S. vestibularis&quot;,&quot;Wol&quot;=&quot;Wolbachia spp.&quot;))
#,&quot;Wigg&quot;=&quot;Wigglesworthia&quot;
table(pseudogeness_sumary$species)
table(pseudogeness_sumary$species_new)  
  
# species=$(echo &quot;vestibularis&quot;)

 pseudogeness_sumary$species_new  = factor(pseudogeness_sumary$species_new , levels=c(&quot;L. delbrueckii&quot;,&quot;S. thermophilus&quot;,&quot;L. plantarum&quot;,&quot;L. equicursoris&quot;,&quot;S. suis&quot;,&quot;S. vestibularis&quot;,&quot;M. leprae&quot;,&quot;Rickettsia spp.&quot;,&quot;Buchnera spp.&quot;,&quot;S. glossinidius&quot;,&quot;Wolbachia spp.&quot;))

 
colorsss &lt;- c(&quot;#fdc2b5ff&quot;,&quot;#007fa6ff&quot;,&quot;#ebbc51ff&quot;,&quot;#ebbc51ff&quot;,&quot;#9fcfffff&quot;,&quot;#9fcfffff&quot;,&quot;#51eb85ff&quot;,&quot;#51eb85ff&quot;,&quot;#d58656ff&quot;,&quot;#d58656ff&quot;,&quot;#d58656ff&quot;) 
colorsss &lt;- c(&quot;#fdc2b5ff&quot;,&quot;#007fa6ff&quot;,&quot;grey77&quot;,&quot;grey77&quot;,&quot;grey77&quot;,&quot;grey77&quot;,&quot;#51eb85ff&quot;,&quot;#51eb85ff&quot;,&quot;#d58656ff&quot;,&quot;#d58656ff&quot;,&quot;#d58656ff&quot;)

##-----
#plot
##-----
plotpoints &lt;-ggplot( pseudogeness_sumary, aes(x = meangenes, y = meanPseudo, xmin = meangenes-stdevgene, xmax = meangenes+stdevgene,ymin=meanPseudo-stdevPseudo,ymax=meanPseudo+stdevPseudo,fill=species_new,color=species_new)) +
  geom_point(size=2) +
  geom_errorbarh(alpha=.5)+
  geom_errorbar(alpha=.5)+
    labs(x=&quot;Gene number&quot;,y=&quot;Pseudogenes [%]&quot;)+theme_classic()+theme(legend.title = element_blank())+
  #geom_hline(yintercept =7.5, linetype=&quot;dotted&quot;)+geom_segment(aes(x = 1500, y = -1, xend = 1500, yend = 7.5), linetype = &quot;dotted&quot;)+
  scale_color_manual(values = colorsss)+scale_fill_manual(values = colorsss)+
  annotate(&quot;text&quot;, x = 800, y = 6, label = &quot;endosymbionts&quot;,col=&quot;#d58656ff&quot;)+
  annotate(&quot;text&quot;, x = 800, y = 25, label = &quot;transition&quot;,col=&quot;#51eb85ff&quot;)+
  annotate(&quot;text&quot;, x = 2000, y = 1, label = &quot;free living&quot;,col=&quot;grey77&quot;)

#+lims(x=c(250,4100),y=c(-1.5,40))
plotpoints


pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenes.pdf&quot;,width=6,height=4)

plotpoints

dev.off()


# plotpoints_wo &lt;- plotpoints+theme(legend.position = &quot;none&quot;)
# plotpoints_wo
# plotpoints+stat_ellipse(data=pseudogeness_count,aes(x=genes_coding,y=pseudogenes_percent,fill=species,color=species,group=species),type = &quot;euclid&quot;, level = 3) 

  
  library(plotly)
ggp &lt;- ggplotly(plotpoints)
 ggp
 
 ##------------------------
 ##eclipse
 #to do this we include all std points and mean points into one datafram
 ##------------------------
 
#  pseudogeness_sumary
#  
#   pseudogeness_sumary$groups &lt;- plyr::revalue(pseudogeness_sumary$species,c(&quot;Buch&quot;=&quot;endosmymbiont&quot;,&quot;Sglos&quot;=&quot;endosmymbiont&quot;,&quot;Ldel&quot;=&quot;starter cultures&quot;,&quot;Sterm&quot;=&quot;starter cultures&quot;,&quot;Wigg&quot;=&quot;endosmymbiont&quot;,&quot;Mleprae&quot;=&quot;transition&quot;,&quot;Rick&quot;=&quot;transition&quot;,&quot;Lplan&quot;=&quot;closely related free-living&quot;,&quot;Ssuis&quot;=&quot;closely related free-living&quot;))
# 
#  pseudogeness_esclipse &lt;-  rbind(data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes,y=pseudogeness_sumary$meanPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes-pseudogeness_sumary$stdevgene,y=pseudogeness_sumary$meanPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes+pseudogeness_sumary$stdevgene,y=pseudogeness_sumary$meanPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes,y=pseudogeness_sumary$meanPseudo-pseudogeness_sumary$stdevPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes,y=pseudogeness_sumary$meanPseudo+pseudogeness_sumary$stdevPseudo))
#  
#  
#  
#  plotpoints_esclipse &lt;- ggplot(pseudogeness_esclipse,aes(x=x,y=y,fill=groups,color=groups),group=groups)+labs(x=&quot;Gene number&quot;,y=&quot;Pseudogenes [%]&quot;)+theme_classic()+theme(legend.title = element_blank())+stat_ellipse(type = &quot;t&quot;, linetype = 2)+theme(legend.position = &quot;none&quot;)+lims(x=c(250,4100),y=c(-1.5,40))
# plotpoints_esclipse
#  
# 
# library(patchwork)
# 
# plotpoints+plotpoints_wo+plotpoints_esclipse</code></pre>
<div id="pseudogene-clustering" class="section level3">
<h3>Pseudogene clustering</h3>
<pre class="bash"><code>###-----------------------
#get pseudogenes ffn file
###-----------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/


rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.gff
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff
for speciesss in $(echo &quot;Sterm Ldel&quot;)
do

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${speciesss}|grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39;)
do

echo $genomesss

#grep &quot;pseudogene&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9}&#39; |sed &#39;s/ID=gene-FAM/ps-/g&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/tmp.download

#bedtools getfasta  -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${speciesss}/${genomesss}.fna \
#  -bed /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/tmp.download -nameOnly \
#  -fo /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}/${genomesss}.ffn

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}/${genomesss}.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn


#grep &quot;pseudogene&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9,$10,$11,$12,$13,$14}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.gff

grep &quot;;pseudo=true;&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff  &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff

done #genomesss
done #species



##-----------pseudo


wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff

grep &quot;=is&quot; -i  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9,$10,$11,$12,$13,$14}&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase.gff
grep &quot;transposase&quot; -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9,$10,$11,$12,$13,$14}&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase.gff


cut -f 4 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase.gff |sort|uniq |sed &#39;s/ID=cds-FAM/ps-/g&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned.gff

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff

for genesss in $(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned.gff)
do

echo $genesss

grep &quot;${genesss}&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff

done

cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff |sort|uniq&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss_derep.gff

cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt|sort|uniq |wc -l 
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss_derep.gff
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff 


###-----------------------
#cluster
###-----------------------

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering

#mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/${genomesss}.ffn ${genomesss} tmp --min-seq-id 0.8 -c 0.9 --cov-mode 1
mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn pseudo tmp 

###-----------------------
#bring local
###-----------------------


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/pseudo_cluster.tsv  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/
</code></pre>
<p>add NCBI strains</p>
<pre class="bash"><code>###-----------------------
#get pseudogenes ffn file
###-----------------------

/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/


rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.gff
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff
for speciesss in $(echo &quot;Sterm Ldel&quot;)
do

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${speciesss}|grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39;)
do

echo $genomesss

#grep &quot;pseudogene&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9}&#39; |sed &#39;s/ID=gene-FAM/ps-/g&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/tmp.download

#bedtools getfasta  -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${speciesss}/${genomesss}.fna \
#  -bed /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/tmp.download -nameOnly \
#  -fo /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}/${genomesss}.ffn

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}/${genomesss}.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn


#grep &quot;pseudogene&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9,$10,$11,$12,$13,$14}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.gff

grep &quot;;pseudo=true;&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff  &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff

done #genomesss
done #species



##-----------pseudo


wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff

grep &quot;=is&quot; -i  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9,$10,$11,$12,$13,$14}&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase.gff
grep &quot;transposase&quot; -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9,$10,$11,$12,$13,$14}&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase.gff


cut -f 4 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase.gff |sort|uniq |sed &#39;s/ID=cds-FAM/ps-/g&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned.gff

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff

for genesss in $(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned.gff)
do

echo $genesss

grep &quot;${genesss}&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff

done

cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff |sort|uniq&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss_derep.gff

cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt|sort|uniq |wc -l 
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss_derep.gff
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes_complete_transposase_cleaned_clusterss.gff 


###-----------------------
#cluster
###-----------------------

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering

#mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/${genomesss}.ffn ${genomesss} tmp --min-seq-id 0.8 -c 0.9 --cov-mode 1
mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn pseudo tmp 

###-----------------------
#bring local
###-----------------------


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/pseudo_cluster.tsv  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/
</code></pre>
<p>try to identify glycoside hydrolases</p>
<pre class="r"><code>##------------------------------------
#prep tree
##------------------------------------

library(ape)
Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_curated.txt&quot;, tree.names = NULL, force.multi = FALSE)

tree_phylo &lt;- Sterm_tree_raw %&gt;% as.phylo()

Sterm_tree_raw$tip.label
## get the distance between every two nodes
# tree_phylog &lt;- tree %&gt;% as.phylo()
distance_matrix &lt;-cophenetic.phylo(tree_phylo)
 library(reshape2)
 phylo_distance &lt;- melt(distance_matrix) %&gt;% rename(genome_1=Var1)%&gt;% rename(genome_2=Var2) %&gt;% rename(phylo_distance=value)

#---------------------------
library(readr)
library(plyr)
library(tidyverse)
pseudo_cluster &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/pseudo_cluster.tsv&quot;, &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;cluster&quot;,&quot;pseudogene&quot;),  trim_ws = TRUE)

pseudo_cluster_counts &lt;- table(pseudo_cluster$cluster) %&gt;% data.frame()
ggplot(pseudo_cluster_counts,aes(x=Freq))+geom_bar()+theme_classic()

range(pseudo_cluster_counts$Freq)
pseudo_distance_df &lt;- data.frame()
for (clusterzz in unique(pseudo_cluster$cluster)) {
  print(clusterzz)
  
  tmp &lt;- pseudo_cluster %&gt;% filter(cluster==clusterzz)
        counts &lt;- 0
  for (rowwsss in tmp$pseudogene) {
     counts &lt;- counts+1
    for (columns in tmp$pseudogene[counts:nrow(tmp)]) {
      
      genomesss_1 &lt;- str_split_fixed(rowwsss,&quot;-&quot;,4)[,2]
      genomesss_2 &lt;- str_split_fixed(columns,&quot;-&quot;,4)[,2]

      
      genomesss_1_name &lt;- Sterm_tree_raw$tip.label[grepl(genomesss_1,Sterm_tree_raw$tip.label)]
      genomesss_2_name &lt;- Sterm_tree_raw$tip.label[grepl(genomesss_2,Sterm_tree_raw$tip.label)]

      distanzzz &lt;- phylo_distance %&gt;% filter(genome_1==genomesss_1_name&amp;genome_2==genomesss_2_name) %&gt;% pull(.,phylo_distance)
      
    tmp_2 &lt;- data.frame(pseudo_cluster=clusterzz,genome_1=genomesss_1_name,genome_2=genomesss_2_name,phylo_distanc=distanzzz)  
   pseudo_distance_df &lt;-  rbind(pseudo_distance_df,tmp_2)
    
    }
           

  }
  
  
}

pseudo_distance_df

# write.table(pseudo_distance_df,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/pseudo_cluster_distance_raw.tsv&quot;,sep=&quot;\t&quot;,quote = FALSE,row.names = FALSE)
pseudo_cluster_distance_raw_import &lt;- read_delim(&quot;Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/pseudo_cluster_distance_raw.tsv&quot;, &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)
##------------------------------
#plot
##------------------------------
# unique(pseudo_distance_df$pseudo_cluster)


table(pseudo_cluster_distance_raw_import$pseudo_cluster) %&gt;% sqrt() %&gt;% range()

pseudo_distance_df_summary &lt;- pseudo_cluster_distance_raw_import %&gt;% filter(genome_1!=genome_2)%&gt;% group_by(pseudo_cluster) %&gt;% summarise(median_distance=median(phylo_distanc),counts=n(),max_distance=max(phylo_distanc),sd_distance=sd(phylo_distanc))


# write.table(pseudo_distance_df_summary,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering/pseudo_cluster_distance_summary.tsv&quot;,sep=&quot;\t&quot;,quote = FALSE,row.names = FALSE)

ggplot(pseudo_distance_df_summary,aes(x=sqrt(counts),y=max_distance))+geom_point()+theme_classic()+labs(x=&quot;number genomes&quot;,y=&quot;phylogentic distance&quot;)</code></pre>
<p>next I want to ADD all genes to the list of pseudogenes and redo the
cluster&amp;analysis. The reason I do this is to see if certain genomes
have completely lost or retained the gene.</p>
<pre class="bash"><code>
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn


###-----------------------
#add normal genes to speudogenes
###-----------------------

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes.ffn
for speciesss in $(echo &quot;Sterm Ldel&quot;)
do

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${speciesss}|sed &#39;s/.ffn//g&#39;)
do

sed &#39;s/&gt;FAM/&gt;re-/g&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FFN/${speciesss}/${genomesss}.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes.ffn

done #genomesss
done #species


###--------------------------------------------
##remove pseudogenes from all genes
###--------------------------------------------
##---------------------with faidx

sed &quot;s/ .*//&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes.ffn   &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_new.ffn

samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_new.ffn

##get names
grep &quot;&gt;&quot;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn |sed &#39;s/&gt;ps-/re-/g&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/names_pseudogenes.ffn 

##invert list
remove_ids=($(awk &#39;{print $1}&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_new.ffn.fai | grep -v -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/names_pseudogenes.ffn ))

awk &#39;{print $1}&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_new.ffn.fai | grep -v -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/names_pseudogenes.ffn &gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/names_pseudogenes_wo.ffn 


samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_new.ffn &quot;${remove_ids[@]}&quot; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_final.ffn

xargs --arg-file=/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/names_pseudogenes_wo.ffn  samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_new.ffn &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_final.ffn

grep &quot;&gt;&quot; -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_final.ffn 
grep &quot;&gt;&quot; -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes.ffn 
grep &quot;&gt;&quot; -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn 


##put together
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_pseudogenes.ffn /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_online_withoutpseudo_final.ffn  &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_final.ffn

###-----------------------
#cluster
###-----------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all

#mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/${genomesss}.ffn ${genomesss} tmp --min-seq-id 0.8 -c 0.9 --cov-mode 1
mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/all_genes_final.ffn  pseudo_genes tmp 

#less pseudo_genes_cluster.tsv
grep &quot;ps-&quot; pseudo_genes_cluster.tsv |head
awk &#39;{if($2~&quot;ps-&quot;)print $1}&#39; pseudo_genes_cluster.tsv |uniq &gt; clusters_names_with_pg.txt
rm cluster_ps.txt
for clusersss in $(cat clusters_names_with_pg.txt)
do
echo $clusersss
#awk -v clusterzzz=&quot;$clusersss&quot; &#39;{if($1==clusterzzz)print 0}&#39; pseudo_genes_cluster.tsv &gt;&gt; cluster_ps.txt

grep &quot;${clusersss}&quot; pseudo_genes_cluster.tsv &gt;&gt; cluster_ps.txt


done
wc -l clusters_names_with_pg.txt
cut -f 1 cluster_ps.txt |uniq |wc -l
cut -f 1 pseudo_genes_cluster.tsv |uniq |wc -l
</code></pre>
<pre class="bash"><code>

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt
</code></pre>
<p>get the most common recent ancestor and all tips associated to this
look <a href="https://rpubs.com/tskam/ternary_plot">here</a> for the
ternary plot description.</p>
<pre class="r"><code>library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)

Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_curated.txt&quot;, tree.names = NULL, force.multi = FALSE)
tree_phylo &lt;- Sterm_tree_raw %&gt;% as.phylo()
tree_phylo$tip.label

getMRCA(tree_phylo, c(&quot;S24736&quot;,&quot;S24758&quot;))

numbers &lt;- Descendants(tree_phylo, 156, type = &quot;tips&quot;) %&gt;% unlist()
numbers
tree_phylo$tip.label[numbers]

##------------------------------
#get data clusters
##------------------------------
cluster_ps &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all/cluster_ps.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;cluster&quot;,&quot;gene&quot;),  trim_ws = TRUE)

cluster_ps$genome &lt;- str_split_fixed(cluster_ps$gene,&quot;-&quot;,4)[,2]

analysis_clusters_ldel &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_ldel.csv&quot;,  &quot;,&quot;) %&gt;% add_column(species=&quot;Ldel&quot;) %&gt;% select(-Cluster)
analysis_clusters_sterm &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv&quot;,  &quot;,&quot;)%&gt;% add_column(species=&quot;Sterm&quot;) %&gt;% select(-Cluster)

analysis_clusters_all &lt;- rbind(analysis_clusters_sterm,analysis_clusters_ldel)

analysis_clusters_all$Taxon &lt;- gsub(&quot;S&quot;,&quot;&quot;,analysis_clusters_all$Taxon) %&gt;% gsub(&quot;L&quot;,&quot;&quot;,.)

# C(&quot;ps-10980-i1-1.1_000998&quot; ,&quot;ps-24763-i1-1.1_001759&quot;)



cluster_ps_species &lt;- merge(cluster_ps,analysis_clusters_all,by.x=&quot;genome&quot;,by.y=&quot;Taxon&quot;) %&gt;% as_tibble()

tmp &lt;- cluster_ps_species %&gt;% select(cluster,species) %&gt;% unique() 
# table(tmp$cluster) %&gt;% sort(decreasing = TRUE)


genome_names_corrected &lt;- data.frame()

for (genomesss in unique(cluster_ps$genome)){
        tmp &lt;- tree_phylo$tip.label[grepl(genomesss,tree_phylo$tip.label)]
tmp_2 &lt;- data.frame(genome=genomesss,genome_name=tmp)
genome_names_corrected &lt;- rbind(genome_names_corrected,tmp_2)
}

cluster_ps_final$cluster %&gt;% unique()

cluster_ps_final &lt;- merge(cluster_ps,genome_names_corrected,by=&quot;genome&quot;)
cluster_ps_final$gene_type &lt;- str_split_fixed(cluster_ps_final$gene,&quot;-&quot;,4)[,1]

cluster_ps_final &lt;- cluster_ps_final %&gt;% select(-c(genome,gene)) %&gt;% unique()
dim(cluster_ps_final)
# dim(unique(cluster_ps_final))
cluster_ps_final %&gt;% select(-gene_type) %&gt;% unique() %&gt;% dim()
##a few genes are both ps and re

cluster_ps_final_cleaned &lt;- cluster_ps_final[!duplicated(cluster_ps_final[,-3]),]
dim(cluster_ps_final_cleaned)
cluster_ps_final_cleaned  %&gt;% select(-gene_type) %&gt;% unique() %&gt;% dim()

# cluster_ps_final_cleaned %&gt;% group_by(cluster) %&gt;% summarise(counts=table(gene_type))
cluster_ps_final_cleaned_count &lt;- cluster_ps_final_cleaned %&gt;% group_by(cluster) %&gt;% mutate(n=n(gene_type))
cluster_ps_final_cleaned_count %&gt;% filter(cluster==&quot;ps-10973-i1-1.1_000242&quot;)

cluster_ps_final_cleaned_count_wide &lt;- spread(cluster_ps_final_cleaned_count, gene_type, n)  %&gt;%replace(is.na(.), 0)
cluster_ps_final_cleaned_count_wide


###find number of genes per cluster
total_genomes &lt;- data.frame()
for (clusterzzz in unique(cluster_ps_final_cleaned$cluster)) {
  print(clusterzzz)
  # tmp &lt;- cluster_ps_final_cleaned %&gt;% filter(cluster==clusterzzz) %&gt;% pull(.,genome_name) %&gt;% droplevels() %&gt;% as.character()
    tmp &lt;- cluster_ps_final_cleaned %&gt;% filter(cluster==clusterzzz) %&gt;% pull(.,genome_name) %&gt;% as.character()

  nodess &lt;- getMRCA(tree_phylo, tmp)
# nodess
numbers &lt;- Descendants(tree_phylo, nodess, type = &quot;tips&quot;) %&gt;% unlist()
# numbers
genomesss &lt;- tree_phylo$tip.label[numbers]

genomesss_final &lt;- ifelse(length(genomesss)==0,1,length(genomesss))

tmp_2 &lt;- data.frame(cluster=clusterzzz,total_genomes=genomesss_final)
total_genomes &lt;- rbind(total_genomes,tmp_2 )
}
total_genomes
##------------merged with others

cluster_ps_final_cleaned_count_wide_final &lt;- merge(cluster_ps_final_cleaned_count_wide,total_genomes,by=&quot;cluster&quot;) %&gt;% as.tibble() %&gt;% mutate(rm=total_genomes-ps-re)


cluster_ps_final_cleaned_count_wide_final &lt;- cluster_ps_final_cleaned_count_wide_final %&gt;% mutate(pseudogenes=100*(ps/total_genomes))%&gt;% mutate(complete=100*(re/total_genomes))%&gt;% mutate(lost=100*(rm/total_genomes))%&gt;% mutate(summed=lost+pseudogenes+complete)

# cluster_ps_final_cleaned_count_wide_final %&gt;% filter(summed&gt;100.5)

##------------------------------
#make ternary plot
##------------------------------

library(ggtern)

# ggtern(data=cluster_ps_final_cleaned_count_wide_final,aes(x=complete,y=lost, z=pseudogenes,fill=total_genomes,color=total_genomes)) +
  # geom_point()
dim(cluster_ps_final_cleaned_count_wide_final)
pseudogenization_plot &lt;- ggtern(data=cluster_ps_final_cleaned_count_wide_final,aes(x=complete,y=lost, z=pseudogenes,color=total_genomes,size=total_genomes)) +
  geom_point(alpha=0.5) +#scale_size(trans = &#39;reverse&#39;)+
  # labs(title=&quot;Population structure, 2015&quot;) +
  theme_rgbw()

pseudogenization_plot
pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization.pdf&quot;,width=11,height=9)

pseudogenization_plot

dev.off()
##------------------------------
#box plot
##------------------------------
cluster_ps_final_cleaned_count_wide_final_smaller &lt;- cluster_ps_final_cleaned_count_wide_final %&gt;% select(cluster,pseudogenes,complete,lost)%&gt;% gather(.,grouping,percents,c(pseudogenes,complete,lost))
 cluster_ps_final_cleaned_count_wide_final_smaller$grouping = factor(cluster_ps_final_cleaned_count_wide_final_smaller$grouping, levels=c(&quot;lost&quot;,&quot;pseudogenes&quot;,&quot;complete&quot;))

pseudogenization_boxplot &lt;- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()
pseudogenization_boxplot





pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_violin.pdf&quot;,width=11,height=9)

pseudogenization_boxplot

dev.off()

##-------------------------------
#Update figure
##-------------------------------

analysis_clusters_all$Taxon &lt;- gsub(&quot;S&quot;,&quot;&quot;,analysis_clusters_all$Taxon) %&gt;% gsub(&quot;L&quot;,&quot;&quot;,.)


species_cluster &lt;- cluster_ps_species %&gt;% select(cluster,species) %&gt;% unique() 


tmp &lt;- cluster_ps_final_cleaned_count_wide_final %&gt;% filter(!cluster %in% c(&quot;ps-10980-i1-1.1_000998&quot; ,&quot;ps-24763-i1-1.1_001759&quot;))

cluster_ps_final_cleaned_count_wide_final_species &lt;- merge(tmp,species_cluster, by.x=&quot;cluster&quot;,by.y=&quot;cluster&quot;) %&gt;%  as_tibble()

colorsss &lt;- rev(c(&quot;#007fa6ff&quot;,&quot;#fdc2b5ff&quot;))

pseudogenization_plot_species &lt;- ggtern(data=cluster_ps_final_cleaned_count_wide_final_species,aes(x=complete,y=lost, z=pseudogenes,color=species)) +
  geom_point(alpha=0.5,size=3.5) + scale_color_manual(values =colorsss )+#scale_size(trans = &#39;reverse&#39;)+
  # labs(title=&quot;Population structure, 2015&quot;) +
  theme_rgbw()

table(cluster_ps_final_cleaned_count_wide_final_species$species)
dim(cluster_ps_final_cleaned_count_wide_final_species)

pseudogenization_plot_species
pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species.pdf&quot;,width=11,height=9)

pseudogenization_plot_species

dev.off()


##----------------------
#prep 
##----------------------


analysis_clusters_ldel &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_ldel.csv&quot;,  &quot;,&quot;) %&gt;% add_column(species=&quot;Ldel&quot;) %&gt;% select(-Cluster)
analysis_clusters_sterm &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv&quot;,  &quot;,&quot;)%&gt;% add_column(species=&quot;Sterm&quot;) %&gt;% select(-Cluster)

analysis_clusters_all_prep &lt;- rbind(analysis_clusters_sterm,analysis_clusters_ldel)


cluster_ps_final_prep &lt;- merge(cluster_ps_final,analysis_clusters_all_prep,by.x=&quot;genome_name&quot;,by.y=&quot;Taxon&quot;)

# cluster_ps_final$genome_name %&gt;% 
  
  write.csv(cluster_ps_final_prep,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/data/cluster_ps_final_prep.csv&quot;,quote = FALSE,row.names = FALSE)


##----------------------
#grouping
##----------------------

cluster_ps_final_cleaned_count_wide_final_species &lt;- ifelse(lost&gt;60&amp;complete&lt;40&amp;pseudogenes&lt;40,&quot;old&quot;,ifelse(lost&lt;40&amp;complete&gt;60&amp;pseudogenes&lt;40,&quot;recent&quot;,ifelse(lost&lt;40&amp;complete&lt;40&amp;pseudogenes&gt;60,&quot;maintained&quot;,&quot;intermediate&quot;)))


cluster_ps_final_cleaned_count_wide_final_species_extended &lt;- cluster_ps_final_cleaned_count_wide_final_species %&gt;% mutate(category=ifelse(lost&gt;60&amp;complete&lt;40&amp;pseudogenes&lt;40,&quot;old&quot;,ifelse(lost&lt;40&amp;complete&gt;60&amp;pseudogenes&lt;40,&quot;recent&quot;,ifelse(lost&lt;40&amp;complete&lt;40&amp;pseudogenes&gt;60,&quot;maintained&quot;,&quot;intermediate&quot;))))

table(cluster_ps_final_cleaned_count_wide_final_species_extended$category)


# ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=species, group=category,fill=category))+geom_bar(position = &quot;dodge&quot;)+theme_classic()


cluster_ps_final_cleaned_count_wide_final_species_extended$species &lt;- plyr::revalue(cluster_ps_final_cleaned_count_wide_final_species_extended$species,c( &quot;Ldel&quot;=&quot;L. delbrueckii&quot;,
 &quot;Sterm&quot;=&quot;S. thermophilus&quot;))

# ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=interaction(species,category), group=category,fill=category))+geom_bar(position = &quot;dodge&quot;)+theme_classic()
genes_time_pseudogenization &lt;- ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=species, group=category,fill=category))+geom_bar(position = &quot;dodge&quot;)+theme_classic()+facet_wrap(~category,nrow = 1)+theme(axis.title.x = element_blank(),legend.position = &quot;none&quot;,axis.text.x = element_text(angle=75,hjust=1))+labs(y=&quot;pseudogene gene family&quot;)

genes_time_pseudogenization

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing.pdf&quot;,width=7,height=3)

genes_time_pseudogenization

dev.off()


##----------------------------------
#fraction of genomes containing this pseudogene
cluster_ps_final_cleaned_count_wide_final_species$species &lt;- plyr::revalue(cluster_ps_final_cleaned_count_wide_final_species$species,c( &quot;Ldel&quot;=&quot;L. delbrueckii&quot;,
 &quot;Sterm&quot;=&quot;S. thermophilus&quot;))
distribution_pseudogenes &lt;- ggplot(cluster_ps_final_cleaned_count_wide_final_species,aes(x=pseudogenes,fill=species))+geom_histogram()+theme_classic()+facet_wrap(~species,nrow=2,scales=&quot;free_y&quot;)+labs(x=&quot;fraction of genomes containing the pseudogene&quot;,y=&quot;number of pseudogene containing gene families&quot;)+theme(legend.position=&quot;none&quot;)+ scale_fill_manual(values =colorsss )

distribution_pseudogenes

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing_02.pdf&quot;,width=7,height=6)

distribution_pseudogenes

dev.off()</code></pre>
</div>
</div>
<div id="download-complete-fam-strains-from-ncbi" class="section level2">
<h2>download complete FAM strains from NCBI</h2>
<p>I have downloaded them previousl and converted here, I just have to
extract the gene and pseudogenes of these genomes</p>
<pre class="bash"><code>


#mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/FAA/

#seqret -sformat genbank -sequence  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GBK/S24724.gbf -outseq /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/FAA/S24724.faa
speciesss=Sterm
genomesss=S24724

#rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG
for speciesss in $(echo &quot;Sterm Ldel&quot;)
do
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FNA/
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_pseudo_faa/${speciesss}//FNA/


firstCharacter=${speciesss:0:1}

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT//pseudo/genes/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT//pseudo/genes/${speciesss}
#rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo/${speciesss}/all_pseudogenes_complete.gff
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF/  |grep &quot;^${firstCharacter}&quot;|grep &quot;.gff&quot; |sed &#39;s/.gff//g&#39;)
do

echo $genomesss


#grep &quot;pseudogene&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9}&#39; |sed &#39;s/ID=gene-FAM/ps-/g&#39; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/tmp.download


#grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep &quot;pseudo=true&quot; |wc -l
#grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep &quot;pseudo=true&quot; -v |grep &quot;ncRNA&quot; -v |wc -l
#less /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.*

#grep &quot;Pseudo Genes (total)&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1
#grep &quot;CDSs (with protein)&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1

locusTagNCBI=$(grep &quot;.pseudogene&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF//${genomesss}.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $9}&#39;|head -1 |sed &#39;s/ID=//g&#39;|cut -d &#39;_&#39; -f 1)
#locusTagNCBI=$(grep &quot;pseudogene&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF//${genomesss}.gff |grep &quot;pseudogene&quot;|awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $9}&#39;|head -1 |sed &#39;s/ID=cds-//g&#39;|cut -d &#39;_&#39; -f 1)

grep &quot;.pseudogene&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF//${genomesss}.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$3,$4,$5,$6,$7,$8,$9}&#39; |sed &quot;s/ID=${locusTagNCBI}/ps:${genomesss}/g&quot; |sed &#39;s/\.pseu.*//g&#39; |awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}&#39;  &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.pseudo.download


awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{if($3==&quot;mRNA&quot;)print $0}&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF//${genomesss}.gff    |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print  $1,$2,$3,$4,$5,$6,$7,$8,$9}&#39; |sed  &quot;s/ID=${locusTagNCBI}/re:${genomesss}/g&quot;|sed &#39;s/.t.*//g&#39; |awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.complete.download


#grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF//${genomesss}.gff  |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$3,$4,$5,$6,$7,$8,$10}&#39; |sed  &quot;s/Parent=${locusTagNCBI}/re:${genomesss}/g&quot; |sed &#39;s/.t.*//g&#39; |awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.complete.download

#grep &quot;${genomesss}$(printf &#39;\t&#39;)&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |awk  -F &quot;[\t]&quot; -v loczzz=&quot;$locusTagNCBI&quot; &#39;{OFS=&quot;\t&quot;}{print loczzz,$0}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG


bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/FNA//${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.pseudo.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.pseudo.ffn

bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/FNA//${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.complete.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.complete.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.pseudo.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/all_pseudogenes.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.complete.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/all_complete.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes//${speciesss}/${genomesss}.pseudo.ffn &gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes//${speciesss}/${genomesss}.complete.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes//${speciesss}/${genomesss}.pseudo.ffn &gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_pseudo_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes//${speciesss}/${genomesss}.complete.ffn &gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FNA/${genomesss}.all.ffn



#grep &quot;pseudogene&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  |awk -F &quot;[\t;]&quot; -v genomessszz=&quot;$genomesss&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9,$10,$11,$12,$13,$14,genomessszz}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/all_pseudogenes.gff

#grep &quot;;pseudo=true;&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//${speciesss}/all_pseudogenes_complete.gff

done #genomesss
done #species
</code></pre>
</div>
<div id="download-ncbi-final" class="section level2">
<h2>download NCBI final</h2>
<pre class="bash"><code>##---------------------------
#rename NCBI strains
##---------------------------


###-------------------------------------------------
#final list
###-------------------------------------------------


#missings
#FAM24728
#FAM24737
#FAM24738
#FAM24739
#FAM24740
rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log

for genomesss in $(echo &quot;strain=24728
strain=24737
strain=24738
strain=24739
strain=24740&quot;)
do

grep ${genomesss}  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log.txt


genomesss_short=$(echo ${genomesss}|sed &#39;s/strain=/S/g&#39;)
grep ${genomesss}  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log.txt |awk -F &quot;\t&quot; -v biosamplezzz=&quot;$genomesss_short&quot;  &#39;{OFS=&quot;\t&quot;}{print biosamplezzz,$1,$2,$3,$8,$9}&#39;  &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.txt


grep ${genomesss}  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log.txt |cut -f 20 | sed -r &#39;s|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.downloadONLY


echo &quot;------------------------&quot;
#awk -F &quot;\t&quot; -v biosamplezzz=&quot;$biosampless&quot; -v samplezzz=&quot;$sampleesss&quot;  -v speciessssss=&quot;$speciesss&quot;  -v ageezz=&quot;$agee&quot;  -v GCFSssszz=&quot;$GCFSsss_new&quot;  &#39;{OFS=&quot;\t&quot;}{if($3==biosamplezzz) print samplezzz,speciessssss,ageezz,GCFSssszz,$1,$2,$3,$8,$9}&#39; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log.txt

done

paste -d &#39;\t&#39; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.txt /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.downloadONLY &gt;  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL

##===================================================================================================
###---------------------------------------------------------------
#download genomes
###---------------------------------------------------------------
##===================================================================================================

whichFILE=fna

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA/
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA/

cut -f 7 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file

rm *

wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file


###-------------------------------------
##change name
###-------------------------------------

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA/

for genomessss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL )
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |cut -f 1 )
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz

#mv /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA//GCF_900322585.1_Lactobacillus_delbrueckii_subsp._lactis_protein.fna /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA/Lactobacillus_delbrueckii_subsp_lactis-7.fna


##===================================================================================================
###---------------------------------------------------------------
#download faa
###---------------------------------------------------------------
##===================================================================================================
whichFILE=faa

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FAA/
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FAA/

cut -f 7 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |sed &#39;s/_genomic.fna./_protein.faa./g&#39;  &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file

rm *

wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file


###-------------------------------------
##change name
###-------------------------------------

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FAA/

for genomessss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL )
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |cut -f 1 )
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz


##===================================================================================================
###---------------------------------------------------------------
#download gff
###---------------------------------------------------------------
##===================================================================================================
whichFILE=gff

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GFF/
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GFF/

cut -f 7 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |sed &#39;s/_genomic.fna./_genomic.gff./g&#39;  &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file

rm *

wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file


###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL )
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |cut -f 1 )
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz


##===================================================================================================
###---------------------------------------------------------------
#download gbff
###---------------------------------------------------------------
##===================================================================================================
whichFILE=gbff

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GBFF/
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GBFF/

cut -f 7 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |sed &#39;s/_genomic.fna./_genomic.gbff./g&#39;  &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file

rm *

wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/download.file


###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL )
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |cut -f 1 )
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz
</code></pre>
</div>
<div id="pseudogene-cluster-with-ncbi" class="section level2">
<h2>pseudogene cluster with NCBI</h2>
<pre class="bash"><code>
###-----------------------
#get pseudogenes ffn file
###-----------------------


rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG
for speciesss in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL|sort|uniq)
do


specizzzz=$(grep ${speciesss} /data/Project/2020_StarterCultureDiversity/log.species|cut -f 1)
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_pseudo_faa/${specizzzz}//FNA/

mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${specizzzz}//FNA/


rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo/genes/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo/genes/${speciesss}
#rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo/${speciesss}/all_pseudogenes_complete.gff
for genomesss in $(grep ${speciesss} /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL|cut -f 1)
do

echo $genomesss

#grep &quot;pseudogene&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9}&#39; |sed &#39;s/ID=gene-FAM/ps-/g&#39; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/tmp.download


#grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep &quot;pseudo=true&quot; |wc -l
#grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep &quot;pseudo=true&quot; -v |grep &quot;ncRNA&quot; -v |wc -l
#less /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.*

#grep &quot;Pseudo Genes (total)&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1
#grep &quot;CDSs (with protein)&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1

locusTagNCBI=$(grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep &quot;pseudo=true&quot;|awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $9}&#39;|head -1 |sed &#39;s/ID=cds-//g&#39;|cut -d &#39;_&#39; -f 1)

grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep &quot;pseudo=true&quot;|awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$3,$4,$5,$6,$7,$8,$9}&#39; |sed &quot;s/ID=cds-${locusTagNCBI}/ps:${genomesss}/g&quot; |awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}&#39;  &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.pseudo.download



awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{if($3==&quot;gene&quot;)print $0}&#39; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  |grep &quot;rna&quot; -i -v |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print  $1,$2,$3,$4,$5,$6,$7,$8,$9}&#39; |sed  &quot;s/ID=gene-${locusTagNCBI}/re:${genomesss}/g&quot; |awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}&#39; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.complete.download

grep &quot;${genomesss}$(printf &#39;\t&#39;)&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |awk  -F &quot;[\t]&quot; -v loczzz=&quot;$locusTagNCBI&quot; &#39;{OFS=&quot;\t&quot;}{print loczzz,$0}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG


bedtools getfasta -s  -fi /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/FNA//${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.pseudo.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.pseudo.ffn

bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/FNA//${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.complete.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.complete.ffn

cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.pseudo.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/all_pseudogenes.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.complete.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/all_complete.ffn


cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.pseudo.ffn  &gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${specizzzz}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.complete.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${specizzzz}//FNA/${genomesss}.all.ffn


cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.pseudo.ffn   &gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_pseudo_faa/${specizzzz}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/${genomesss}.complete.ffn  &gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${specizzzz}//FNA/${genomesss}.all.ffn


#grep &quot;pseudogene&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  |awk -F &quot;[\t;]&quot; -v genomessszz=&quot;$genomesss&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9,$10,$11,$12,$13,$14,genomessszz}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${speciesss}/all_pseudogenes.gff

#grep &quot;;pseudo=true;&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//${speciesss}/all_pseudogenes_complete.gff

done #genomesss
done #species
###------------------------------------------------------
#FAM strains
###------------------------------------------------------

#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK
for speciesss in $(echo &quot;Sterm Ldel Lhelv&quot;)
do

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo${speciesss}/all_pseudogenes_complete.gff
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${speciesss}|sed &#39;s/.gbk//g&#39; )
do

echo $genomesss

#grep &quot;pseudogene&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9}&#39; |sed &#39;s/ID=gene-FAM/ps-/g&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/tmp.download


#grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep &quot;pseudo=true&quot; |wc -l
#grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep &quot;pseudo=true&quot; -v |grep &quot;ncRNA&quot; -v |wc -l
#less /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.*

#grep &quot;Pseudo Genes (total)&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1
#grep &quot;CDSs (with protein)&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1

locusTagNCBI=$(grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff |grep &quot;pseudo=true&quot;|awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $9}&#39;|head -1 |sed &#39;s/ID=cds-//g&#39;|cut -d &#39;_&#39; -f 1)

 grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff |grep &quot;pseudo=true&quot;|awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print  $1,$2,$3,$4,$5,$6,$7,$8,$9}&#39; |sed &quot;s/ID=cds-${locusTagNCBI}/ps:${genomesss}/g&quot;  |awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}&#39;  &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.pseudo.download
 #grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff  |grep -v  &quot;pseudo=true&quot;|awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$3,$4,$5,$6,$7,$8,$9}&#39; |sed  &quot;s/ID=cds-${locusTagNCBI}/re:${genomesss}/g&quot;  |awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}&#39;  &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.complete.download

awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{if($3==&quot;gene&quot;)print $0}&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF/${speciesss}/${genomesss}.gff   |grep &quot;rna&quot; -i -v |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print  $1,$2,$3,$4,$5,$6,$7,$8,$9}&#39; |sed  &quot;s/ID=gene-${locusTagNCBI}/re:${genomesss}/g&quot; |awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.complete.download



#awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{if($3==&quot;mRNA&quot;)print $0}&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF//${genomesss}.gff    |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print  $1,$2,$3,$4,$5,$6,$7,$8,$9}&#39; |sed  &quot;s/ID=${locusTagNCBI}/re:${genomesss}/g&quot;|sed &#39;s/.t.*//g&#39; |awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/${genomesss}.complete.download



#grep &quot;${genomesss}$(printf &#39;\t&#39;)&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |awk  -F &quot;[\t]&quot; -v loczzz=&quot;$locusTagNCBI&quot; &#39;{OFS=&quot;\t&quot;}{print loczzz,$0}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG


bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${speciesss}/${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.pseudo.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn

bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/FNA/${speciesss}/${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.complete.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.complete.ffn





#grep &quot;pseudogene&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  |awk -F &quot;[\t;]&quot; -v genomessszz=&quot;$genomesss&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9,$10,$11,$12,$13,$14,genomessszz}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_pseudogenes.gff

#grep &quot;;pseudo=true;&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/${speciesss}/all_pseudogenes_complete.gff

done #genomesss
done #species

###-----------------------
#for some dailact genomes we have both ONT and Illumina assemblies---in thes cases I have to delete the Illumina only genomes
###-----------------------




#rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG
for speciesss in $(echo &quot;Sterm Ldel&quot;)
do

firstCharacter=${speciesss:0:1}


#rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo/${speciesss}/all_pseudogenes_complete.gff
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/GFF/  |grep &quot;^${firstCharacter}&quot;|grep &quot;.gff&quot; |sed &#39;s/.gff//g&#39;)
do

echo $genomesss
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.*

done

done


#---------------------------------merge the genomes

for speciesss in $(echo &quot;Sterm Ldel Lhelv&quot;)
do

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_pseudogenes.ffn
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_complete.ffn
#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo${speciesss}/all_pseudogenes_complete.gff
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/ |grep &quot;.pseudo.ffn&quot;| sed &#39;s/.pseudo.ffn//g&#39;|sort|uniq )
do

echo $genomesss

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_pseudogenes.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/${genomesss}.complete.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_complete.ffn


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes//${speciesss}/${genomesss}.pseudo.ffn    &gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes//${speciesss}/${genomesss}.complete.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes//${speciesss}/${genomesss}.pseudo.ffn   &gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_pseudo_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes//${speciesss}/${genomesss}.complete.ffn  &gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FNA/${genomesss}.all.ffn


done
done





###------------------------------------------------------
#FAM strains FROM NCBI
###------------------------------------------------------
#/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA/
#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK
for speciesss in $(echo &quot;Sterm&quot;)
do

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}
#rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo${speciesss}/all_pseudogenes_complete.gff
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GBFF/|sed &#39;s/.gbff//g&#39; )
do

echo $genomesss

#grep &quot;pseudogene&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9}&#39; |sed &#39;s/ID=gene-FAM/ps-/g&#39; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/tmp.download


#grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep &quot;pseudo=true&quot; |wc -l
#grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff |grep &quot;pseudo=true&quot; -v |grep &quot;ncRNA&quot; -v |wc -l
#less /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.*

#grep &quot;Pseudo Genes (total)&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1
#grep &quot;CDSs (with protein)&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GBFF/${genomesss}.* |head -1

locusTagNCBI=$(grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GFF/${genomesss}.gff |grep &quot;pseudo=true&quot;|awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $9}&#39;|head -1 |sed &#39;s/ID=cds-//g&#39;|cut -d &#39;_&#39; -f 1)

 grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GFF//${genomesss}.gff |grep &quot;pseudo=true&quot;|awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$3,$4,$5,$6,$7,$8,$9}&#39; |sed &quot;s/ID=cds-${locusTagNCBI}/ps:${genomesss}/g&quot;  |awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}&#39;  &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.download
# grep &quot;CDS&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GFF//${genomesss}.gff  |grep -v  &quot;pseudo=true&quot;|awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print$1,$2,$3,$4,$5,$6,$7,$8,$10}&#39; |sed  &quot;s/Parent=gene-${locusTagNCBI}/re:${genomesss}/g&quot;  |awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}&#39; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.complete.download



awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{if($3==&quot;gene&quot;)print $0}&#39; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/GFF//${genomesss}.gff   |grep &quot;rna&quot; -i -v |awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print  $1,$2,$3,$4,$5,$6,$7,$8,$9}&#39; |sed  &quot;s/ID=gene-${locusTagNCBI}/re:${genomesss}/g&quot; |awk -F &quot;[\t]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$2,$9,$4,$5,$6,$7,$8,$3}&#39; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.complete.download

#grep &quot;${genomesss}$(printf &#39;\t&#39;)&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL |awk  -F &quot;[\t]&quot; -v loczzz=&quot;$locusTagNCBI&quot; &#39;{OFS=&quot;\t&quot;}{print loczzz,$0}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/log/20220328_NCBI_log_final_focalNCBi_Strains.ALL.NCBilocusTAG


bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA//${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn

bedtools getfasta -s -fi /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/FNA//${genomesss}.fna \
  -bed /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.complete.download -nameOnly \
  -fo  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.complete.ffn

cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/all_pseudogenes.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.complete.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/all_complete.ffn



cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn &gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes//${speciesss}/${genomesss}.complete.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn

cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn    &gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes//${speciesss}/${genomesss}.complete.ffn &gt;&gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn



cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/${genomesss}.pseudo.ffn    &gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_pseudo_faa/${speciesss}//FNA/${genomesss}.all.ffn
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes//${speciesss}/${genomesss}.complete.ffn  &gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FNA/${genomesss}.all.ffn


#grep &quot;pseudogene&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  |awk -F &quot;[\t;]&quot; -v genomessszz=&quot;$genomesss&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9,$10,$11,$12,$13,$14,genomessszz}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/all_pseudogenes.gff

#grep &quot;;pseudo=true;&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/GFF//${genomesss}.gff  &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/${speciesss}/all_pseudogenes_complete.gff

done #genomesss
ll /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA
done #species
</code></pre>
<p>add also non-pseudogenized genes</p>
<pre class="bash"><code>

###-----------------------
#combine genes
###-----------------------

rm /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes
cd /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes
for speciesss in $(echo &quot;Sterm Ldel Lhelv&quot;)
do

specizzzz=$(grep ${speciesss} /data/Project/2020_StarterCultureDiversity/log.species|cut -f 2)

##dialact but from NCBI
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/all_pseudogenes.ffn &gt; ${speciesss}.all.genes.fasta
cat /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_but_dialactStrains/pseudo/genes/${speciesss}/all_complete.ffn &gt;&gt;  ${speciesss}.all.genes.fasta

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/all_pseudogenes.ffn &gt;&gt; ${speciesss}.all.genes.fasta
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/For_NCBI/afterNCBI/ONT/pseudo//genes/${speciesss}/all_complete.ffn &gt;&gt;  ${speciesss}.all.genes.fasta

##dialact 
cat  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_pseudogenes.ffn &gt;&gt;  ${speciesss}.all.genes.fasta
cat  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/genes/${speciesss}/all_complete.ffn &gt;&gt;  ${speciesss}.all.genes.fasta

##NCBI
cat   /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${specizzzz}//all_pseudogenes.ffn &gt;&gt;  ${speciesss}.all.genes.fasta
cat   /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs_final_focal/pseudo//genes/${specizzzz}//all_complete.ffn &gt;&gt;  ${speciesss}.all.genes.fasta

echo ${speciesss}
grep &quot;&gt;&quot; ${speciesss}.all.genes.fasta |cut -d &#39;:&#39; -f 2|cut -d &#39;_&#39; -f 1|sort|uniq -c
#grep &quot;&gt;&quot; ${speciesss}.all.genes.fasta |cut -d &#39;_&#39; -f 1|sort|uniq -c

done #species


###-----------------------
#NEW combine genes
###-----------------------

rm /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis_02/
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis_02/genes
cd /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis_02/genes

 


for speciesss in $(echo &quot;Sterm Ldel Lhelv&quot;)
do

specizzzz=$(grep ${speciesss} /data/Project/2020_StarterCultureDiversity/log.species|cut -f 2)

##dialact but from NCBI
cat /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/*.all.ffn &gt; ${speciesss}_ALL.GENES.fna
grep &quot;&gt;&quot; ${speciesss}_ALL.GENES.fna |cut -d &#39;:&#39; -f 2|cut -d &#39;_&#39; -f 1|sort|uniq -c
#grep &quot;&gt;&quot; ${speciesss}.all.genes.fasta |cut -d &#39;_&#39; -f 1|sort|uniq -c

done #species


###-----------------------
#cluster
###-----------------------

rm /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/mmseqs2_clustering_all_restricted
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/mmseqs2_clustering_all_restricted
cd /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/mmseqs2_clustering_all_restricted
for speciesss in $(echo &quot;Sterm Ldel Lhelv&quot;)
do

#mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis_02/genes/${speciesss}_ALL.GENES.fna  ${speciesss}_pseudo_genes tmp 
mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis_02/genes/${speciesss}_ALL.GENES.fna  --min-seq-id 0.8 -c 0.9  ${speciesss}_pseudo_genes tmp 

 
done


##-----------------prep
rm  ALL_pseudo_genes_cluster_ps.txt
for speciesss in $(echo &quot;Sterm Ldel Lhelv&quot;)
do

#grep &quot;ps:&quot; ${speciesss}_pseudo_genes
awk &#39;{if($2~&quot;ps:&quot;)print $1}&#39; ${speciesss}_pseudo_genes_cluster.tsv |uniq &gt; ${speciesss}_pseudo_genes_clusters_names_with_pg.txt
rm ${speciesss}_pseudo_genes_cluster_ps.txt
for clusersss in $(cat ${speciesss}_pseudo_genes_clusters_names_with_pg.txt)
do
echo $clusersss
#awk -v clusterzzz=&quot;$clusersss&quot; &#39;{if($1==clusterzzz)print 0}&#39; pseudo_genes_cluster.tsv &gt;&gt; cluster_ps.txt

grep &quot;${clusersss}&quot; ${speciesss}_pseudo_genes_cluster.tsv &gt;&gt; ${speciesss}_pseudo_genes_cluster_ps.txt

done
wc -l ${speciesss}_pseudo_genes_cluster_ps.txt
cut -f 1 ${speciesss}_pseudo_genes_cluster_ps.txt|uniq |wc -l
cut -f 1 ${speciesss}_pseudo_genes_clusters_names_with_pg.txt |uniq |wc -l

awk -F &quot;\t&quot; -v speciezzz=&quot;$speciesss&quot; &#39;{OFS=&quot;\t&quot;}{print speciezzz,$0}&#39; ${speciesss}_pseudo_genes_cluster_ps.txt &gt;&gt; ALL_pseudo_genes_cluster_ps.txt

done
wc -l ALL_pseudo_genes_cluster_ps.txt

##-----------------analysis

for speciesss in $(echo &quot;Sterm Ldel Lhelv&quot;)
do
echo -e &quot;--------------------------------&quot;${speciesss}
#wc -l ${speciesss}_pseudo_genes_cluster_ps.txt
cut -f 1 ${speciesss}_pseudo_genes_cluster_ps.txt|uniq |wc -l
cut -f 1 ${speciesss}_pseudo_genes_cluster.tsv|uniq |wc -l
#cut -f 1 ${speciesss}_pseudo_genes_clusters_names_with_pg.txt |uniq |wc -l
#cut -f 2 ${speciesss}_pseudo_genes_cluster.tsv|sort|uniq |wc -l
done

#STerm  2163/6648
#Ldel   1965/7400
#Lhelv   1511/5680

1511
</code></pre>
<pre class="bash"><code>

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/
scp  vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/mmseqs2_clustering_all_restricted/ALL_pseudo_genes_cluster_ps.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/
</code></pre>
<p>get the most common recent ancestor and all tips associated to this
look <a href="https://rpubs.com/tskam/ternary_plot">here</a> for the
ternary plot description.</p>
<pre class="r"><code>library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)

##------------------------------
#get data clusters
##------------------------------
cluster_ps &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/ALL_pseudo_genes_cluster_ps.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;species&quot;,&quot;cluster&quot;,&quot;gene&quot;),  trim_ws = TRUE)

cluster_ps$genome &lt;- str_split_fixed(cluster_ps$gene,&quot;_&quot;,2)[,1] %&gt;% gsub(&quot;.*:&quot;,&quot;&quot;,.)

genome_count &lt;- cluster_ps %&gt;% select(species, genome) %&gt;% unique() %&gt;% group_by(species) %&gt;% summarise(num_genome=n())

cluster_ps$gene_type &lt;- gsub(&quot;:.*&quot;,&quot;&quot;,cluster_ps$gene)

cluster_ps_counts &lt;- cluster_ps %&gt;% group_by(species,cluster,genome,gene_type) %&gt;% summarise(countss=n())
table(cluster_ps_counts$countss)

cluster_ps_counts_wide &lt;- cluster_ps_counts %&gt;% spread(., gene_type, countss)  %&gt;%replace(is.na(.), 0) %&gt;% mutate(gene_presence=ifelse(re&gt;0,&quot;re&quot;,&quot;ps&quot;))


cluster_ps_counts_gene_counts &lt;- cluster_ps_counts_wide %&gt;% group_by(species,cluster,gene_presence) %&gt;% summarise(countz=n()) %&gt;% spread(., gene_presence, countz)  %&gt;%replace(is.na(.), 0) %&gt;% left_join(.,genome_count,by=&quot;species&quot;) %&gt;% mutate(rm=num_genome-ps-re) %&gt;% mutate(complete=100*(re/num_genome))%&gt;% mutate(lost=100*(rm/num_genome))%&gt;% mutate(pseudogenes=100*(ps/num_genome))

# write_csv(cluster_ps_counts_wide,&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/genomes_ps.clustered.txt&quot;)

library(ggtern)

# ggtern(data=cluster_ps_final_cleaned_count_wide_final,aes(x=complete,y=lost, z=pseudogenes,fill=total_genomes,color=total_genomes)) +
  # geom_point()
dim(cluster_ps_counts_gene_counts)
pseudogenization_plot &lt;- ggtern(data=cluster_ps_counts_gene_counts,aes(x=complete,y=lost, z=pseudogenes,color=species)) +
  geom_point(alpha=0.5) +#scale_size(trans = &#39;reverse&#39;)+
  # labs(title=&quot;Population structure, 2015&quot;) +
  theme_rgbw()

pseudogenization_plot
pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_all.pdf&quot;,width=11,height=9)

pseudogenization_plot

dev.off()
##------------------------------
#box plot
##------------------------------
cluster_ps_final_cleaned_count_wide_final_smaller &lt;- cluster_ps_counts_gene_counts %&gt;% select(cluster,pseudogenes,complete,lost)%&gt;% gather(.,grouping,percents,c(pseudogenes,complete,lost))
 cluster_ps_final_cleaned_count_wide_final_smaller$grouping = factor(cluster_ps_final_cleaned_count_wide_final_smaller$grouping, levels=c(&quot;lost&quot;,&quot;pseudogenes&quot;,&quot;complete&quot;))

pseudogenization_boxplot &lt;- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()+facet_wrap(~species)
pseudogenization_boxplot

pseudogenization_boxplot_all &lt;- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()
pseudogenization_boxplot_all



pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_violin_all.pdf&quot;,width=11,height=9)

pseudogenization_boxplot_all

dev.off()

##----------------------
#grouping
##----------------------



cluster_ps_counts_gene_counts_extended &lt;- cluster_ps_counts_gene_counts %&gt;% mutate(category=ifelse(lost&gt;60&amp;complete&lt;40&amp;pseudogenes&lt;40,&quot;old&quot;,ifelse(lost&lt;40&amp;complete&gt;60&amp;pseudogenes&lt;40,&quot;recent&quot;,ifelse(lost&lt;40&amp;complete&lt;40&amp;pseudogenes&gt;60,&quot;maintained&quot;,&quot;intermediate&quot;))))

table(cluster_ps_counts_gene_counts_extended$category)


# ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=species, group=category,fill=category))+geom_bar(position = &quot;dodge&quot;)+theme_classic()


# cluster_ps_counts_gene_counts_extended$species &lt;- plyr::revalue(cluster_ps_final_cleaned_count_wide_final_species_extended$species,c( &quot;Ldel&quot;=&quot;L. delbrueckii&quot;,
 # &quot;Sterm&quot;=&quot;S. thermophilus&quot;))

# ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=interaction(species,category), group=category,fill=category))+geom_bar(position = &quot;dodge&quot;)+theme_classic()
genes_time_pseudogenization &lt;- ggplot(cluster_ps_counts_gene_counts_extended,aes(x=species, group=category,fill=category))+geom_bar(position = &quot;dodge&quot;)+theme_classic()+facet_wrap(~category,nrow = 1)+theme(axis.title.x = element_blank(),legend.position = &quot;none&quot;,axis.text.x = element_text(angle=75,hjust=1))+labs(y=&quot;pseudogene gene family&quot;)

genes_time_pseudogenization

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing_all.pdf&quot;,width=7,height=3)

genes_time_pseudogenization

dev.off()


##----------------------------------
#fraction of genomes containing this pseudogene
# cluster_ps_final_cleaned_count_wide_final_species$species &lt;- plyr::revalue(cluster_ps_final_cleaned_count_wide_final_species$species,c( &quot;Ldel&quot;=&quot;L. delbrueckii&quot;,
#  &quot;Sterm&quot;=&quot;S. thermophilus&quot;))
# distribution_pseudogenes &lt;- ggplot(cluster_ps_final_cleaned_count_wide_final_species,aes(x=pseudogenes,fill=species))+geom_histogram()+theme_classic()+facet_wrap(~species,nrow=2,scales=&quot;free_y&quot;)+labs(x=&quot;fraction of genomes containing the pseudogene&quot;,y=&quot;number of pseudogene containing gene families&quot;)+theme(legend.position=&quot;none&quot;)+ scale_fill_manual(values =colorsss )
# 
# distribution_pseudogenes
# 
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing_02.pdf&quot;,width=7,height=6)
# 
# distribution_pseudogenes
# 
# dev.off()</code></pre>
</div>
<div id="pseudogenecluster-with-orthofinder" class="section level2">
<h2>pseudogenecluster with orthofinder</h2>
<pre class="bash"><code>

###-----------------------
#translate
###-----------------------

for speciesss in $(echo &quot;Sterm Ldel Lhelv&quot;)
do

rm -r /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}/FAA
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}/FAA

rm /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/all.all.ffn

ll /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/
ls /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/ |wc -l


#specizzzz=$(grep ${speciesss} /data/Project/2020_StarterCultureDiversity/log.species|cut -f 2)
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/ |grep &quot;.all.ffn&quot; |sed &#39;s/.all.ffn//g&#39; |grep -v &quot;all&quot;)
do

 transeq -sformat pearson -sequence /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FNA/${genomesss}.all.ffn -outseq /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FAA/${genomesss}.all.faa

sed -i &#39;s/_1$//g&#39; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FAA/${genomesss}.all.faa

  done


ll /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FAA/
done #species




###-----------------------
#cluster
###-----------------------


for speciesss in $(echo &quot;Lhelv Sterm Ldel &quot;)
do

rm -r /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/${speciesss}
mkdir -p /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/

    orthofinder -f /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FAA/ \
      -t 35 \
      -o /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/${speciesss}/ \
      -a 35

done #species



#--------------------------------------
#check duplicate gene names
#--------------------------------------


for speciesss in $(echo &quot;Sterm Ldel Lhelv&quot;)
do

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FNA/ |grep &quot;.all.ffn&quot; |sed &#39;s/.all.ffn//g&#39; |grep -v &quot;all&quot;)
do

#grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FAA/${genomesss}.all.faa | cut -d &#39;|&#39; -f 2 |sort|uniq -c |sort |sed &#39;s/^[[:blank:]]*//;s/[[:blank:]]*$//&#39; |grep &quot;^2&quot; -c


#grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_complete_faa/${speciesss}//FAA/${genomesss}.all.faa |sort|uniq -c |sort |sed &#39;s/^[[:blank:]]*//;s/[[:blank:]]*$//&#39; |grep &quot;^2&quot; -c
grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/genes_faa/${speciesss}//FAA/${genomesss}.all.faa |sort|uniq -c |sort |sed &#39;s/^[[:blank:]]*//;s/[[:blank:]]*$//&#39; |grep &quot;^1&quot; -v 
  done
done


##---------------------
#extract the files
##---------------------
/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Lhelv/Results_Apr04/Orthogroups/

/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Lhelv/Results_Apr04/Orthogroups/Orthogroups.GeneCount.tsv




  mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/
  scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Lhelv/Results_Apr04/Orthogroups/Orthogroups.tsv /home/vincent/Desktop/Pr
  
  
  
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps
scp  vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Lhelv/Results_Apr04/Orthogroups/Orthogroups.tsv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Orthogroups_Lhelv.tsv

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps
scp  vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Sterm/Results_Apr04/Orthogroups/Orthogroups.tsv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Orthogroups_Sterm.tsv

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps
scp  vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Ldel/Results_Apr04/Orthogroups/Orthogroups.tsv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Orthogroups_Ldel.tsv
</code></pre>
<pre class="bash"><code>##========================================
#get OG sequences for eggnog
##========================================

/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Lhelv

#rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm_OG_genes.faa
for species in $(echo &quot;Lhelv Ldel Sterm&quot;)
do
rm /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}_OG_genes.faa

echo $species

firstletter=${species:0:2}

#done
date=$(ls //data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}/ |grep &quot;Results&quot;)

for genesss in $(ls //data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}/${date}/Orthogroup_Sequences|grep &quot;.fa$&quot; |sed &#39;s/.fa//g&#39;)
do
echo $genesss

#perl -00 -ne &#39;/(&gt;[^&gt;]+)/ &amp;&amp; print $1&#39; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroup_Sequences/${genesss}.fa |sed &quot;s/&gt;.*$/&gt;S_${genesss}/g&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm_OG_genes.faa

awk &#39;/^&gt;/{if(N)exit;++N;} {print;}&#39; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}/${date}/Orthogroup_Sequences/${genesss}.fa |sed &quot;s/&gt;.*$/&gt;${firstletter}_${genesss}/g&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}_OG_genes.faa


done #genesss
ls /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}/${date}/Orthogroup_Sequences/ |grep &quot;.fa$&quot; -c
grep &quot;&gt;&quot; -c  /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//${species}_OG_genes.faa

done #species

grep &quot;&gt;&quot;  //data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/Sterm_OG_genes.faa |head 
grep &quot;&gt;&quot;  /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//Ldel_OG_genes.faa |head 
grep &quot;&gt;&quot;  /data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps//Lhelv_OG_genes.faa |head 

##========================================
#bring local for eggnog
##========================================

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/forEGGNOG_pangenome
scp -r vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/PSEUDOgeneanalysis/orthofinder_with_ps/*_genes.faa /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/
</code></pre>
<pre class="r"><code>#----------------------------------------------
#Lhelv
#----------------------------------------------

library(tidyverse)
Orthogroups_orthofinder &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Orthogroups_Lhelv.tsv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE) %&gt;% rename(&quot;Orthogroup&quot;=&quot;...1&quot;)

sample_column &lt;- 2

Orthogroups_orthofinder_stats &lt;- data.frame()
for (sample_column in 2:ncol(Orthogroups_orthofinder)) {
  
  sampleNAME &lt;- colnames(Orthogroups_orthofinder)[sample_column]
  print(sampleNAME)
  for (Orthogroupsss in 1:nrow(Orthogroups_orthofinder)) {
    
    
      OrthoNAME &lt;- (Orthogroups_orthofinder)[Orthogroupsss,&quot;Orthogroup&quot;] %&gt;% as.character()
tmp1 &lt;- Orthogroups_orthofinder[which(Orthogroups_orthofinder$Orthogroup==OrthoNAME),sampleNAME] %&gt;% rename(&quot;focal&quot;=sampleNAME)%&gt;%   mutate(focal = strsplit(as.character(focal), &quot;, &quot;)) %&gt;% 
    unnest(focal)

count_re &lt;- tmp1 %&gt;% filter(grepl(&quot;^re&quot;,focal)) %&gt;% nrow() %&gt;% ifelse(.&gt;0,.,&quot;0&quot;)
count_ps &lt;- tmp1 %&gt;% filter(grepl(&quot;^ps&quot;,focal))%&gt;% nrow() %&gt;% ifelse(.&gt;0,.,&quot;0&quot;)


tmp2 &lt;- data.frame(Orthogroup=OrthoNAME,sample=sampleNAME,ps=count_ps,re=count_re)

Orthogroups_orthofinder_stats &lt;- rbind(Orthogroups_orthofinder_stats,tmp2)
  }
  
}

dim(Orthogroups_orthofinder_stats)
table(Orthogroups_orthofinder_stats$sample)

Orthogroups_orthofinder_stats_test &lt;- Orthogroups_orthofinder_stats %&gt;% as_tibble() %&gt;% mutate(ps=as.numeric(ps))%&gt;% mutate(re=as.numeric(re))

write_csv(Orthogroups_orthofinder_stats_test,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered_counts.txt&quot;)
Orthogroups_orthofinder_stats_test &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered_counts.txt&quot;)


##----------duplicated orthogroups

Orthogroups_orthofinder_stats_cum &lt;- Orthogroups_orthofinder_stats_test %&gt;% mutate(all_gene_counts=ps+re) %&gt;% group_by(Orthogroup) %&gt;% summarise(numGenes=max(all_gene_counts))
tmp_for_plotting &lt;- table(Orthogroups_orthofinder_stats_cum$numGenes) %&gt;% as.data.frame() %&gt;% as_tibble()

# Orthogroups_orthofinder_stats_cum &lt;- Orthogroups_orthofinder_stats_test %&gt;% filter(ps&gt;0)%&gt;% mutate(all_gene_counts=ps+re) %&gt;% group_by(Orthogroup) %&gt;% summarise(numGenes=max(all_gene_counts))
# tmp_for_plotting &lt;- table(Orthogroups_orthofinder_stats_cum$numGenes) %&gt;% as.data.frame() %&gt;% as_tibble()
numGenomes &lt;- Orthogroups_orthofinder_stats_test %&gt;% select(sample) %&gt;% unique() %&gt;% nrow()

# %&gt;% mutate(num_genome=)
# %&gt;% mutate(all_gene_counts=ps+re) 
OGs_to_keep &lt;- Orthogroups_orthofinder_stats_cum %&gt;% filter(numGenes==1) %&gt;% pull(Orthogroup)
Orthogroups_orthofinder_stats_test_keepers_01 &lt;- Orthogroups_orthofinder_stats_test %&gt;% filter(Orthogroup %in% OGs_to_keep)  %&gt;% gather(.,gene_type,percents,c(ps,re)) %&gt;% filter(percents&gt;0)  %&gt;% mutate(genome=gsub(&quot;.all&quot;,&quot;&quot;,sample)) %&gt;% select(-percents,-sample) %&gt;% spread(., genome, gene_type)  %&gt;%replace(is.na(.), &quot;rm&quot;)
cluster_ps_counts_wide_imported_wide_final &lt;-  Orthogroups_orthofinder_stats_test_keepers_01 %&gt;%  gather(., genome,grouping , colnames(Orthogroups_orthofinder_stats_test_keepers_01)[2:ncol(Orthogroups_orthofinder_stats_test_keepers_01)], factor_key=TRUE,na.rm = TRUE)  %&gt;% mutate(species=&quot;Lhelv&quot;) #%&gt;% filter(grouping!=&quot;re&quot;)
write_csv(cluster_ps_counts_wide_imported_wide_final,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_cluster_ps_counts_wide_imported_wide_final&quot;)
table(cluster_ps_counts_wide_imported_wide_final$grouping)


Orthogroups_orthofinder_stats_test_keepers &lt;- Orthogroups_orthofinder_stats_test %&gt;% filter(Orthogroup %in% OGs_to_keep) %&gt;% gather(.,gene_type,percents,c(ps,re)) %&gt;% filter(percents&gt;0) %&gt;% select(-percents) %&gt;% group_by(Orthogroup,gene_type) %&gt;% summarise(counts=n())%&gt;% spread(., gene_type, counts)%&gt;%replace(is.na(.), 0) %&gt;% mutate(num_genome=numGenomes) %&gt;% mutate(rm=(num_genome-ps-re)) %&gt;% mutate(complete=100*(re/num_genome)) %&gt;% mutate(lost=100*(rm/num_genome)) %&gt;% mutate(pseudogenes=100*(ps/num_genome)) %&gt;% mutate(species=&quot;Lhelv&quot;) %&gt;% filter(ps&gt;0)
  
write_csv(Orthogroups_orthofinder_stats_test_keepers,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered.txt&quot;)

Orthogroups_orthofinder_stats_test_keepers_Ldel &lt;- Orthogroups_orthofinder_stats_test_keepers
# Orthogroups_orthofinder_stats_test_keepers$Orthogroup %&gt;% unique()
  

#----------------------------------------------
#Sterm
#----------------------------------------------


library(tidyverse)
Orthogroups_orthofinder &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Orthogroups_Sterm.tsv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE) %&gt;% rename(&quot;Orthogroup&quot;=&quot;...1&quot;)

sample_column &lt;- 2

Orthogroups_orthofinder_stats &lt;- data.frame()
for (sample_column in 2:ncol(Orthogroups_orthofinder)) {
  
  sampleNAME &lt;- colnames(Orthogroups_orthofinder)[sample_column]
  print(sampleNAME)
  for (Orthogroupsss in 1:nrow(Orthogroups_orthofinder)) {
    
    
      OrthoNAME &lt;- (Orthogroups_orthofinder)[Orthogroupsss,&quot;Orthogroup&quot;] %&gt;% as.character()
tmp1 &lt;- Orthogroups_orthofinder[which(Orthogroups_orthofinder$Orthogroup==OrthoNAME),sampleNAME] %&gt;% rename(&quot;focal&quot;=sampleNAME)%&gt;%   mutate(focal = strsplit(as.character(focal), &quot;, &quot;)) %&gt;% 
    unnest(focal)

count_re &lt;- tmp1 %&gt;% filter(grepl(&quot;^re&quot;,focal)) %&gt;% nrow() %&gt;% ifelse(.&gt;0,.,&quot;0&quot;)
count_ps &lt;- tmp1 %&gt;% filter(grepl(&quot;^ps&quot;,focal))%&gt;% nrow() %&gt;% ifelse(.&gt;0,.,&quot;0&quot;)


tmp2 &lt;- data.frame(Orthogroup=OrthoNAME,sample=sampleNAME,ps=count_ps,re=count_re)

Orthogroups_orthofinder_stats &lt;- rbind(Orthogroups_orthofinder_stats,tmp2)
  }
  
}

dim(Orthogroups_orthofinder_stats)
table(Orthogroups_orthofinder_stats$sample)

Orthogroups_orthofinder_stats_test &lt;- Orthogroups_orthofinder_stats %&gt;% as_tibble() %&gt;% mutate(ps=as.numeric(ps))%&gt;% mutate(re=as.numeric(re))

write_csv(Orthogroups_orthofinder_stats_test,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered_counts.txt&quot;)
Orthogroups_orthofinder_stats_test &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered_counts.txt&quot;)


##----------duplicated orthogroups

Orthogroups_orthofinder_stats_cum &lt;- Orthogroups_orthofinder_stats_test %&gt;% mutate(all_gene_counts=ps+re) %&gt;% group_by(Orthogroup) %&gt;% summarise(numGenes=max(all_gene_counts))
tmp_for_plotting &lt;- table(Orthogroups_orthofinder_stats_cum$numGenes) %&gt;% as.data.frame() %&gt;% as_tibble()

# Orthogroups_orthofinder_stats_cum &lt;- Orthogroups_orthofinder_stats_test %&gt;% filter(ps&gt;0)%&gt;% mutate(all_gene_counts=ps+re) %&gt;% group_by(Orthogroup) %&gt;% summarise(numGenes=max(all_gene_counts))
# tmp_for_plotting &lt;- table(Orthogroups_orthofinder_stats_cum$numGenes) %&gt;% as.data.frame() %&gt;% as_tibble()
numGenomes &lt;- Orthogroups_orthofinder_stats_test %&gt;% select(sample) %&gt;% unique() %&gt;% nrow()

# %&gt;% mutate(num_genome=)
# %&gt;% mutate(all_gene_counts=ps+re) 
OGs_to_keep &lt;- Orthogroups_orthofinder_stats_cum %&gt;% filter(numGenes==1) %&gt;% pull(Orthogroup)

Orthogroups_orthofinder_stats_test_keepers_01 &lt;- Orthogroups_orthofinder_stats_test %&gt;% filter(Orthogroup %in% OGs_to_keep)  %&gt;% gather(.,gene_type,percents,c(ps,re)) %&gt;% filter(percents&gt;0)  %&gt;% mutate(genome=gsub(&quot;.all&quot;,&quot;&quot;,sample)) %&gt;% select(-percents,-sample) %&gt;% spread(., genome, gene_type)  %&gt;%replace(is.na(.), &quot;rm&quot;)
cluster_ps_counts_wide_imported_wide_final &lt;-  Orthogroups_orthofinder_stats_test_keepers_01 %&gt;%  gather(., genome,grouping , colnames(Orthogroups_orthofinder_stats_test_keepers_01)[2:ncol(Orthogroups_orthofinder_stats_test_keepers_01)], factor_key=TRUE,na.rm = TRUE)  %&gt;% mutate(species=&quot;Sterm&quot;) #%&gt;% filter(grouping!=&quot;re&quot;)
write_csv(cluster_ps_counts_wide_imported_wide_final,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_cluster_ps_counts_wide_imported_wide_final&quot;)

table(cluster_ps_counts_wide_imported_wide_final$grouping)

Orthogroups_orthofinder_stats_test_keepers &lt;- Orthogroups_orthofinder_stats_test %&gt;% filter(Orthogroup %in% OGs_to_keep) %&gt;% gather(.,gene_type,percents,c(ps,re)) %&gt;% filter(percents&gt;0) %&gt;% select(-percents) %&gt;% group_by(Orthogroup,gene_type) %&gt;% summarise(counts=n())%&gt;% spread(., gene_type, counts)%&gt;%replace(is.na(.), 0) %&gt;% mutate(num_genome=numGenomes) %&gt;% mutate(rm=(num_genome-ps-re)) %&gt;% mutate(complete=100*(re/num_genome)) %&gt;% mutate(lost=100*(rm/num_genome)) %&gt;% mutate(pseudogenes=100*(ps/num_genome)) %&gt;% mutate(species=&quot;Sterm&quot;) %&gt;% filter(ps&gt;0)
  
write_csv(Orthogroups_orthofinder_stats_test_keepers,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered.txt&quot;)

Orthogroups_orthofinder_stats_test_keepers$Orthogroup %&gt;% unique()
  


#----------------------------------------------
#Ldel
#----------------------------------------------


library(tidyverse)
Orthogroups_orthofinder &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Orthogroups_Ldel.tsv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE) %&gt;% rename(&quot;Orthogroup&quot;=&quot;...1&quot;)

sample_column &lt;- 2

Orthogroups_orthofinder_stats &lt;- data.frame()
for (sample_column in 2:ncol(Orthogroups_orthofinder)) {
  
  sampleNAME &lt;- colnames(Orthogroups_orthofinder)[sample_column]
  print(sampleNAME)
  for (Orthogroupsss in 1:nrow(Orthogroups_orthofinder)) {
    
    
      OrthoNAME &lt;- (Orthogroups_orthofinder)[Orthogroupsss,&quot;Orthogroup&quot;] %&gt;% as.character()
tmp1 &lt;- Orthogroups_orthofinder[which(Orthogroups_orthofinder$Orthogroup==OrthoNAME),sampleNAME] %&gt;% rename(&quot;focal&quot;=sampleNAME)%&gt;%   mutate(focal = strsplit(as.character(focal), &quot;, &quot;)) %&gt;% 
    unnest(focal)

count_re &lt;- tmp1 %&gt;% filter(grepl(&quot;^re&quot;,focal)) %&gt;% nrow() %&gt;% ifelse(.&gt;0,.,&quot;0&quot;)
count_ps &lt;- tmp1 %&gt;% filter(grepl(&quot;^ps&quot;,focal))%&gt;% nrow() %&gt;% ifelse(.&gt;0,.,&quot;0&quot;)


tmp2 &lt;- data.frame(Orthogroup=OrthoNAME,sample=sampleNAME,ps=count_ps,re=count_re)

Orthogroups_orthofinder_stats &lt;- rbind(Orthogroups_orthofinder_stats,tmp2)
  }
  
}

dim(Orthogroups_orthofinder_stats)
table(Orthogroups_orthofinder_stats$sample)

Orthogroups_orthofinder_stats_test &lt;- Orthogroups_orthofinder_stats %&gt;% as_tibble() %&gt;% mutate(ps=as.numeric(ps))%&gt;% mutate(re=as.numeric(re))

write_csv(Orthogroups_orthofinder_stats_test,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered_counts.txt&quot;)
Orthogroups_orthofinder_stats_test &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered_counts.txt&quot;)


##----------duplicated orthogroups

Orthogroups_orthofinder_stats_cum &lt;- Orthogroups_orthofinder_stats_test %&gt;% mutate(all_gene_counts=ps+re) %&gt;% group_by(Orthogroup) %&gt;% summarise(numGenes=max(all_gene_counts))
tmp_for_plotting &lt;- table(Orthogroups_orthofinder_stats_cum$numGenes) %&gt;% as.data.frame() %&gt;% as_tibble()

# Orthogroups_orthofinder_stats_cum &lt;- Orthogroups_orthofinder_stats_test %&gt;% filter(ps&gt;0)%&gt;% mutate(all_gene_counts=ps+re) %&gt;% group_by(Orthogroup) %&gt;% summarise(numGenes=max(all_gene_counts))
# tmp_for_plotting &lt;- table(Orthogroups_orthofinder_stats_cum$numGenes) %&gt;% as.data.frame() %&gt;% as_tibble()
numGenomes &lt;- Orthogroups_orthofinder_stats_test %&gt;% select(sample) %&gt;% unique() %&gt;% nrow()

# %&gt;% mutate(num_genome=)
# %&gt;% mutate(all_gene_counts=ps+re) 
OGs_to_keep &lt;- Orthogroups_orthofinder_stats_cum %&gt;% filter(numGenes==1) %&gt;% pull(Orthogroup)
Orthogroups_orthofinder_stats_test_keepers_01 &lt;- Orthogroups_orthofinder_stats_test %&gt;% filter(Orthogroup %in% OGs_to_keep)  %&gt;% gather(.,gene_type,percents,c(ps,re)) %&gt;% filter(percents&gt;0)  %&gt;% mutate(genome=gsub(&quot;.all&quot;,&quot;&quot;,sample)) %&gt;% select(-percents,-sample) %&gt;% spread(., genome, gene_type)  %&gt;%replace(is.na(.), &quot;rm&quot;)
cluster_ps_counts_wide_imported_wide_final &lt;-  Orthogroups_orthofinder_stats_test_keepers_01 %&gt;%  gather(., genome,grouping , colnames(Orthogroups_orthofinder_stats_test_keepers_01)[2:ncol(Orthogroups_orthofinder_stats_test_keepers_01)], factor_key=TRUE,na.rm = TRUE)  %&gt;% mutate(species=&quot;Ldel&quot;) #%&gt;% filter(grouping!=&quot;re&quot;)

# write_csv(cluster_ps_counts_wide_imported_wide_final,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_cluster_ps_counts_wide_imported_wide_final&quot;)

write_csv(cluster_ps_counts_wide_imported_wide_final,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_cluster_ps_counts_wide_imported_wide_final&quot;)

table(cluster_ps_counts_wide_imported_wide_final$grouping)


Orthogroups_orthofinder_stats_test_keepers &lt;- Orthogroups_orthofinder_stats_test %&gt;% filter(Orthogroup %in% OGs_to_keep) %&gt;% gather(.,gene_type,percents,c(ps,re)) %&gt;% filter(percents&gt;0) %&gt;% select(-percents) %&gt;% group_by(Orthogroup,gene_type) %&gt;% summarise(counts=n())%&gt;% spread(., gene_type, counts)%&gt;%replace(is.na(.), 0) %&gt;% mutate(num_genome=numGenomes) %&gt;% mutate(rm=(num_genome-ps-re)) %&gt;% mutate(complete=100*(re/num_genome)) %&gt;% mutate(lost=100*(rm/num_genome)) %&gt;% mutate(pseudogenes=100*(ps/num_genome)) %&gt;% mutate(species=&quot;Ldel&quot;) %&gt;% filter(ps&gt;0)
  
write_csv(Orthogroups_orthofinder_stats_test_keepers,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt&quot;)

Orthogroups_orthofinder_stats_test_keepers$Orthogroup %&gt;% unique()</code></pre>
<pre class="r"><code>library(tidyverse)

Ldel_genomes_ps.clustered.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt&quot;)
table(Ldel_genomes_ps.clustered.txt$complete)
Sterm_genomes_ps.clustered.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered.txt&quot;)
Lhelv_genomes_ps.clustered.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered.txt&quot;)

sum(Sterm_genomes_ps.clustered.txt$ps)
sum(Sterm_genomes_ps.clustered.txt$re)
sum(Sterm_genomes_ps.clustered.txt$rm)
table(Sterm_genomes_ps.clustered.txt$pseudogenes)

Orthogroups_orthofinder_stats_test_keepers &lt;- rbind(Ldel_genomes_ps.clustered.txt, Sterm_genomes_ps.clustered.txt,Lhelv_genomes_ps.clustered.txt)

Orthogroups_orthofinder_stats_test_keepers

cluster_ps_counts_gene_counts &lt;- cluster_ps_counts_wide %&gt;% group_by(species,cluster,gene_presence) %&gt;% summarise(countz=n()) %&gt;% spread(., gene_presence, countz)  %&gt;%replace(is.na(.), 0) %&gt;% left_join(.,genome_count,by=&quot;species&quot;) %&gt;% mutate(rm=num_genome-ps-re) %&gt;% mutate(complete=100*(re/num_genome))%&gt;% mutate(lost=100*(rm/num_genome))%&gt;% mutate(pseudogenes=100*(ps/num_genome))

# write_csv(cluster_ps_counts_wide,&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/genomes_ps.clustered.txt&quot;)


##for pseudogene
Ldel_genomes_ps.clustered_counts.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered_counts.txt&quot;) %&gt;% mutate(species=&quot;Ldel&quot;)
Orthogroups_orthofinder_stats_cum &lt;- Ldel_genomes_ps.clustered_counts.txt %&gt;% mutate(all_gene_counts=ps+re) %&gt;% group_by(species,Orthogroup) %&gt;% summarise(numGenes=max(all_gene_counts))
tmp_for_plotting &lt;- table(Orthogroups_orthofinder_stats_cum$numGenes) %&gt;% as.data.frame() %&gt;% as_tibble()
OGs_to_keep &lt;- Orthogroups_orthofinder_stats_cum %&gt;% filter(numGenes==1) %&gt;% pull(Orthogroup)
Ldel_genomes_ps.clustered_counts.txt_filtered &lt;- Ldel_genomes_ps.clustered_counts.txt %&gt;% filter(Orthogroup %in% unique(Ldel_genomes_ps.clustered.txt$Orthogroup))

Sterm_genomes_ps.clustered_counts.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered_counts.txt&quot;) %&gt;% mutate(species=&quot;Sterm&quot;)
Orthogroups_orthofinder_stats_cum &lt;- Sterm_genomes_ps.clustered_counts.txt %&gt;% mutate(all_gene_counts=ps+re) %&gt;% group_by(species,Orthogroup) %&gt;% summarise(numGenes=max(all_gene_counts))
tmp_for_plotting &lt;- table(Orthogroups_orthofinder_stats_cum$numGenes) %&gt;% as.data.frame() %&gt;% as_tibble()
OGs_to_keep &lt;- Orthogroups_orthofinder_stats_cum %&gt;% filter(numGenes==1) %&gt;% pull(Orthogroup)
Sterm_genomes_ps.clustered_counts.txt_filtered &lt;- Sterm_genomes_ps.clustered_counts.txt %&gt;% filter(Orthogroup %in% unique(Sterm_genomes_ps.clustered.txt$Orthogroup))

Lhelv_genomes_ps.clustered_counts.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered_counts.txt&quot;) %&gt;% mutate(species=&quot;Lhelv&quot;)
Orthogroups_orthofinder_stats_cum &lt;- Lhelv_genomes_ps.clustered_counts.txt %&gt;% mutate(all_gene_counts=ps+re) %&gt;% group_by(species,Orthogroup) %&gt;% summarise(numGenes=max(all_gene_counts))
tmp_for_plotting &lt;- table(Orthogroups_orthofinder_stats_cum$numGenes) %&gt;% as.data.frame() %&gt;% as_tibble()
OGs_to_keep &lt;- Orthogroups_orthofinder_stats_cum %&gt;% filter(numGenes==1) %&gt;% pull(Orthogroup)
Lhelv_genomes_ps.clustered_counts.txt_filtered &lt;- Lhelv_genomes_ps.clustered_counts.txt %&gt;% filter(Orthogroup %in% unique(Lhelv_genomes_ps.clustered.txt$Orthogroup))

Orthogroups_orthofinder_stats_test_keepers_counts &lt;- rbind(Ldel_genomes_ps.clustered_counts.txt_filtered, Sterm_genomes_ps.clustered_counts.txt_filtered,Lhelv_genomes_ps.clustered_counts.txt_filtered) %&gt;% mutate(genome=gsub(&quot;.all&quot;,&quot;&quot;,sample)) %&gt;% mutate(cluster=Orthogroup)%&gt;% mutate(gene_presence=ifelse(re&gt;0,&quot;re&quot;,&quot;ps&quot;)) %&gt;% select(species, cluster ,  genome  , ps,re, gene_presence)

write_csv(Orthogroups_orthofinder_stats_test_keepers_counts,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//genomes_ps.clustered.txt&quot;)

##=================================================
#Plots
##=================================================

library(ggtern)

# ggtern(data=cluster_ps_final_cleaned_count_wide_final,aes(x=complete,y=lost, z=pseudogenes,fill=total_genomes,color=total_genomes)) +
  # geom_point()
dim(Orthogroups_orthofinder_stats_test_keepers)
pseudogenization_plot &lt;- ggtern(data=Orthogroups_orthofinder_stats_test_keepers,aes(x=complete,y=lost, z=pseudogenes,color=species)) +
  geom_point(alpha=0.5) +#scale_size(trans = &#39;reverse&#39;)+
  # labs(title=&quot;Population structure, 2015&quot;) +
  scale_color_manual(values=c(&quot;#fdc2b5ff&quot;,&quot;#b3b3b3ff&quot;,&quot;#007fa6ff&quot;))+
  theme_rgbw()

pseudogenization_plot
pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_all.pdf&quot;,width=8,height=6)

pseudogenization_plot

dev.off()
 
##------------------------------
#box plot
##------------------------------

# cluster_ps_final_cleaned_count_wide_final_smaller &lt;- cluster_ps_counts_gene_counts %&gt;% select(cluster,pseudogenes,complete,lost)%&gt;% gather(.,grouping,percents,c(pseudogenes,complete,lost))

cluster_ps_final_cleaned_count_wide_final_smaller &lt;- Orthogroups_orthofinder_stats_test_keepers %&gt;% select(Orthogroup,pseudogenes,complete,lost,species)%&gt;% gather(.,grouping,percents,c(pseudogenes,complete,lost))
 cluster_ps_final_cleaned_count_wide_final_smaller$grouping = factor(cluster_ps_final_cleaned_count_wide_final_smaller$grouping, levels=c(&quot;lost&quot;,&quot;pseudogenes&quot;,&quot;complete&quot;))

pseudogenization_boxplot &lt;- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()+facet_wrap(~species)
pseudogenization_boxplot

pseudogenization_boxplot_all &lt;- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()
pseudogenization_boxplot_all



pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_violin_all.pdf&quot;,width=8,height=7)

pseudogenization_boxplot_all

dev.off()

##----------------------
#grouping
##----------------------


cluster_ps_counts_gene_counts_extended &lt;- Orthogroups_orthofinder_stats_test_keepers %&gt;% mutate(category=ifelse(lost&gt;60&amp;complete&lt;40&amp;pseudogenes&lt;40,&quot;old&quot;,ifelse(lost&lt;40&amp;complete&gt;60&amp;pseudogenes&lt;40,&quot;recent&quot;,ifelse(lost&lt;40&amp;complete&lt;40&amp;pseudogenes&gt;60,&quot;maintained&quot;,&quot;intermediate&quot;))))

table(cluster_ps_counts_gene_counts_extended$category)


genes_time_pseudogenization &lt;- ggplot(cluster_ps_counts_gene_counts_extended,aes(x=species, group=category,fill=category))+geom_bar(position = &quot;dodge&quot;)+theme_classic()+facet_wrap(~category,nrow = 1)+theme(axis.title.x = element_blank(),legend.position = &quot;none&quot;,axis.text.x = element_text(angle=75,hjust=1))+labs(y=&quot;pseudogene gene family&quot;)

genes_time_pseudogenization

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing_all.pdf&quot;,width=7,height=3)

genes_time_pseudogenization

dev.off()
 

# write_csv(Orthogroups_orthofinder_stats_test_keepers,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt&quot;)</code></pre>
<pre class="r"><code>library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)

##------------------------------
#get data clusters
##------------------------------
cluster_ps &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/ALL_pseudo_genes_cluster_ps.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;species&quot;,&quot;cluster&quot;,&quot;gene&quot;),  trim_ws = TRUE)

cluster_ps$genome &lt;- str_split_fixed(cluster_ps$gene,&quot;_&quot;,2)[,1] %&gt;% gsub(&quot;.*:&quot;,&quot;&quot;,.)

genome_count &lt;- cluster_ps %&gt;% select(species, genome) %&gt;% unique() %&gt;% group_by(species) %&gt;% summarise(num_genome=n())

cluster_ps$gene_type &lt;- gsub(&quot;:.*&quot;,&quot;&quot;,cluster_ps$gene)

cluster_ps_counts &lt;- cluster_ps %&gt;% group_by(species,cluster,genome,gene_type) %&gt;% summarise(countss=n())
table(cluster_ps_counts$countss)

cluster_ps_counts_wide &lt;- cluster_ps_counts %&gt;% spread(., gene_type, countss)  %&gt;%replace(is.na(.), 0) %&gt;% mutate(gene_presence=ifelse(re&gt;0,&quot;re&quot;,&quot;ps&quot;))


cluster_ps_counts_gene_counts &lt;- cluster_ps_counts_wide %&gt;% group_by(species,cluster,gene_presence) %&gt;% summarise(countz=n()) %&gt;% spread(., gene_presence, countz)  %&gt;%replace(is.na(.), 0) %&gt;% left_join(.,genome_count,by=&quot;species&quot;) %&gt;% mutate(rm=num_genome-ps-re) %&gt;% mutate(complete=100*(re/num_genome))%&gt;% mutate(lost=100*(rm/num_genome))%&gt;% mutate(pseudogenes=100*(ps/num_genome))

write_csv(cluster_ps_counts_wide,&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/mmseqs2_clustering_all_new/genomes_ps.clustered.txt&quot;)

library(ggtern)

# ggtern(data=cluster_ps_final_cleaned_count_wide_final,aes(x=complete,y=lost, z=pseudogenes,fill=total_genomes,color=total_genomes)) +
  # geom_point()
dim(cluster_ps_counts_gene_counts)
pseudogenization_plot &lt;- ggtern(data=cluster_ps_counts_gene_counts,aes(x=complete,y=lost, z=pseudogenes,color=species)) +
  geom_point(alpha=0.5) +#scale_size(trans = &#39;reverse&#39;)+
  # labs(title=&quot;Population structure, 2015&quot;) +
  theme_rgbw()

pseudogenization_plot
pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_all.pdf&quot;,width=11,height=9)

pseudogenization_plot

dev.off()
##------------------------------
#box plot
##------------------------------
cluster_ps_final_cleaned_count_wide_final_smaller &lt;- cluster_ps_counts_gene_counts %&gt;% select(cluster,pseudogenes,complete,lost)%&gt;% gather(.,grouping,percents,c(pseudogenes,complete,lost))
 cluster_ps_final_cleaned_count_wide_final_smaller$grouping = factor(cluster_ps_final_cleaned_count_wide_final_smaller$grouping, levels=c(&quot;lost&quot;,&quot;pseudogenes&quot;,&quot;complete&quot;))

pseudogenization_boxplot &lt;- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()+facet_wrap(~species)
pseudogenization_boxplot

pseudogenization_boxplot_all &lt;- ggplot(data=cluster_ps_final_cleaned_count_wide_final_smaller,aes(x=grouping,y=percents,group=grouping,fill=grouping)) +geom_violin()+theme_classic()
pseudogenization_boxplot_all



pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_violin_all.pdf&quot;,width=11,height=9)

pseudogenization_boxplot_all

dev.off()

##----------------------
#grouping
##----------------------



cluster_ps_counts_gene_counts_extended &lt;- cluster_ps_counts_gene_counts %&gt;% mutate(category=ifelse(lost&gt;60&amp;complete&lt;40&amp;pseudogenes&lt;40,&quot;old&quot;,ifelse(lost&lt;40&amp;complete&gt;60&amp;pseudogenes&lt;40,&quot;recent&quot;,ifelse(lost&lt;40&amp;complete&lt;40&amp;pseudogenes&gt;60,&quot;maintained&quot;,&quot;intermediate&quot;))))

table(cluster_ps_counts_gene_counts_extended$category)


# ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=species, group=category,fill=category))+geom_bar(position = &quot;dodge&quot;)+theme_classic()


# cluster_ps_counts_gene_counts_extended$species &lt;- plyr::revalue(cluster_ps_final_cleaned_count_wide_final_species_extended$species,c( &quot;Ldel&quot;=&quot;L. delbrueckii&quot;,
 # &quot;Sterm&quot;=&quot;S. thermophilus&quot;))

# ggplot(cluster_ps_final_cleaned_count_wide_final_species_extended,aes(x=interaction(species,category), group=category,fill=category))+geom_bar(position = &quot;dodge&quot;)+theme_classic()
genes_time_pseudogenization &lt;- ggplot(cluster_ps_counts_gene_counts_extended,aes(x=species, group=category,fill=category))+geom_bar(position = &quot;dodge&quot;)+theme_classic()+facet_wrap(~category,nrow = 1)+theme(axis.title.x = element_blank(),legend.position = &quot;none&quot;,axis.text.x = element_text(angle=75,hjust=1))+labs(y=&quot;pseudogene gene family&quot;)

genes_time_pseudogenization

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing_all.pdf&quot;,width=7,height=3)

genes_time_pseudogenization

dev.off()


##----------------------------------
#fraction of genomes containing this pseudogene
# cluster_ps_final_cleaned_count_wide_final_species$species &lt;- plyr::revalue(cluster_ps_final_cleaned_count_wide_final_species$species,c( &quot;Ldel&quot;=&quot;L. delbrueckii&quot;,
#  &quot;Sterm&quot;=&quot;S. thermophilus&quot;))
# distribution_pseudogenes &lt;- ggplot(cluster_ps_final_cleaned_count_wide_final_species,aes(x=pseudogenes,fill=species))+geom_histogram()+theme_classic()+facet_wrap(~species,nrow=2,scales=&quot;free_y&quot;)+labs(x=&quot;fraction of genomes containing the pseudogene&quot;,y=&quot;number of pseudogene containing gene families&quot;)+theme(legend.position=&quot;none&quot;)+ scale_fill_manual(values =colorsss )
# 
# distribution_pseudogenes
# 
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenization_species_timing_02.pdf&quot;,width=7,height=6)
# 
# distribution_pseudogenes
# 
# dev.off()</code></pre>
<div id="eggnog-on-ps" class="section level3">
<h3>eggnog on ps</h3>
<pre class="r"><code>library(corrplot)
library(tidyverse)
library(RColorBrewer)

Lhelv_Eggnog &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv.Eggnog.annotations&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE, skip = 4)%&gt;% filter(grepl(&quot;^Lh&quot;,`#query`))%&gt;% mutate(species=&quot;Lhelv&quot;)

Ldel_Eggnog &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel.Eggnog.annotations&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE, skip = 4)%&gt;% filter(grepl(&quot;^Ld&quot;,`#query`))%&gt;% mutate(species=&quot;Ldel&quot;)

Sterm_Eggnog &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm.Eggnog.annotations&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE, skip = 4) %&gt;% filter(grepl(&quot;^St&quot;,`#query`)) %&gt;% mutate(species=&quot;Sterm&quot;) 

tail(Lhelv_Eggnog)
all_eggnog &lt;- rbind(Sterm_Eggnog,Ldel_Eggnog,Lhelv_Eggnog)


COG_functional_categories &lt;- read_delim(&quot;~/Desktop/ScriptRepository/COG_functional_categories.csv&quot;,  &quot;,&quot;, escape_double = FALSE, col_names = c(&quot;short&quot;,&quot;long&quot;,&quot;divided&quot;),trim_ws = TRUE, na = &quot;NA&quot;) %&gt;% select(-divided)

# unique(eggnog_annotation$COG_Functional_Category)[which(!unique(eggnog_annotation$COG_Functional_Category) %in% COG_functional_categories$short )]

ALL_sterm &lt;- eggnog_annotation
nrow(ALL_sterm)

all_eggnog_long &lt;- all_eggnog %&gt;% 
    mutate(COG_category = strsplit(as.character(COG_category), &quot;&quot;)) %&gt;% 
    unnest(COG_category)


all_eggnog_long_extended &lt;- merge(all_eggnog_long,COG_functional_categories,by.x = &quot;COG_category&quot;,by.y=&quot;short&quot;)
table(all_eggnog_long_extended$long)


Ldel_genomes_ps.clustered.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt&quot;) %&gt;% mutate(Orthogroup=paste0(&quot;Ld_&quot;,Orthogroup)) %&gt;% select(Orthogroup,complete,lost,pseudogenes  )
# table(Ldel_genomes_ps.clustered.txt$complete)
Sterm_genomes_ps.clustered.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered.txt&quot;)%&gt;% mutate(Orthogroup=paste0(&quot;St_&quot;,Orthogroup))%&gt;% select(Orthogroup,complete,lost,pseudogenes  )
Lhelv_genomes_ps.clustered.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered.txt&quot;)%&gt;% mutate(Orthogroup=paste0(&quot;Lh_&quot;,Orthogroup))%&gt;% select(Orthogroup,complete,lost,pseudogenes  )

all_duplicated_OGs &lt;- rbind(Ldel_genomes_ps.clustered.txt,Sterm_genomes_ps.clustered.txt,Lhelv_genomes_ps.clustered.txt)

all_info &lt;- merge(all_eggnog_long_extended,all_duplicated_OGs,by.x=&quot;#query&quot;,by.y=&quot;Orthogroup&quot;,all.x = TRUE) %&gt;% as_tibble()
is.na(all_info$long) %&gt;% sum()

all_categories &lt;- all_info %&gt;% group_by(long,species) %&gt;% summarise(counts=n()) %&gt;% mutate(cogs=&quot;all&quot;)
only_ps_categories &lt;- all_info %&gt;% filter(pseudogenes&gt;0)%&gt;% group_by(long,species) %&gt;% summarise(counts=n())%&gt;% mutate(cogs=&quot;pseudo&quot;)
complete_cogs_ps &lt;- rbind(all_categories,only_ps_categories)

###----------------------------------
#extract names to check manually
###----------------------------------


only_ps_categories_for_Philipp &lt;- all_info %&gt;% filter(pseudogenes&gt;0)
only_ps_categories_for_Philipp$species &lt;- str_split_fixed(only_ps_categories_for_Philipp$`#query`,&quot;_&quot;,2)[,1] 
colnames(only_ps_categories_for_Philipp)
only_ps_categories_for_Philipp_final &lt;- only_ps_categories_for_Philipp %&gt;% select(&quot;#query&quot; ,&quot;complete&quot;   ,    &quot;lost&quot;      ,     &quot;pseudogenes&quot; ,&quot;species&quot; ,&quot;Preferred_name&quot;,&quot;COG_category&quot;,&quot;long&quot; ,&quot;Description&quot;,&quot;GOs&quot; ,&quot;EC&quot; , &quot;KEGG_ko&quot; ,  &quot;KEGG_Pathway&quot; ,&quot;KEGG_Module&quot;,&quot;KEGG_Reaction&quot;,&quot;KEGG_rclass&quot; ,&quot;BRITE&quot; , &quot;KEGG_TC&quot;,&quot;CAZy&quot; ,  &quot;BiGG_Reaction&quot; , &quot;PFAMs&quot; ,&quot;eggNOG_OGs&quot; , &quot;max_annot_lvl&quot;         ) %&gt;% filter(complete&lt;90)

table(only_ps_categories_for_Philipp_final$species)
                 

write.table(only_ps_categories_for_Philipp_final,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/All_pseudogenes_annotation.csv&quot;,quote = FALSE,row.names = FALSE,sep = &quot;\t&quot;)

###----------------------------------
#completeness of OG (core,cloud info)
###----------------------------------
Lhelv_OG_abundance &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_cluster_ps_counts_wide_imported_wide_final&quot;) %&gt;% mutate(Orthogroup=paste0(&quot;Lh_&quot;,Orthogroup))
Lhelv_OG_abundancesums &lt;- Lhelv_OG_abundance %&gt;% group_by(Orthogroup,grouping,species)  %&gt;% summarise(counts=n())%&gt;% spread(., grouping, counts)%&gt;%replace(is.na(.), 0)
total_gene &lt;- rowSums(Lhelv_OG_abundancesums[,c(&quot;ps&quot;,&quot;re&quot;,&quot;rm&quot;)]) %&gt;% head(1) %&gt;% as.numeric()
Lhelv_OG_abundancesums$coreness &lt;- 100*(Lhelv_OG_abundancesums$re/total_gene)
lhelv_coreness &lt;- Lhelv_OG_abundancesums %&gt;% select(Orthogroup,coreness)

###----------------------------------
#extract names of complete and pseudogenized genes for KEGG map
###----------------------------------

colnames(all_info)
all_info_KEGG &lt;- all_info %&gt;% select(&quot;#query&quot;,&quot;KEGG_ko&quot;, &quot;complete&quot;   ,&quot;lost&quot; ,&quot;pseudogenes&quot;) %&gt;% mutate(KEGG_Reaction=gsub(&quot;ko:&quot;,&quot;&quot;,KEGG_ko)) %&gt;% select(-KEGG_ko)
all_info_KEGG$species &lt;- str_split_fixed(all_info_KEGG$`#query`,&quot;_&quot;,2)[,1] 
all_info_KEGG$category &lt;- ifelse(is.na(all_info_KEGG$complete),&quot;green&quot;,ifelse(all_info_KEGG$complete&gt;90,&quot;orange&quot;,&quot;red&quot;))
all_info_KEGG_yes &lt;- all_info_KEGG %&gt;% filter(KEGG_Reaction!=&quot;-&quot;)

all_info_KEGG_yes_cleaned &lt;- all_info_KEGG_yes %&gt;% 
    mutate(KEGG_Reaction = strsplit(as.character(KEGG_Reaction), &quot;,&quot;)) %&gt;% 
    unnest(KEGG_Reaction)


#----------Ldel

all_info_KEGG_yes_cleaned_Ldel &lt;- all_info_KEGG_yes_cleaned %&gt;% filter(species==&quot;Ld&quot;) %&gt;% select(KEGG_Reaction,category)
all_info_KEGG_yes_cleaned_Ldel[duplicated(all_info_KEGG_yes_cleaned_Ldel$KEGG_Reaction),]
final_KEGG &lt;- data.frame()

for (genesss in unique(all_info_KEGG_yes_cleaned_Ldel$KEGG_Reaction)) {
  
  tmp1 &lt;- all_info_KEGG_yes_cleaned_Ldel %&gt;% filter(KEGG_Reaction==genesss)
  greenezzz &lt;- tmp1 %&gt;% filter(tmp1$category==&quot;green&quot;) %&gt;% nrow
  orangezzz &lt;- tmp1 %&gt;% filter(tmp1$category==&quot;orange&quot;) %&gt;% nrow
  redzzz &lt;- tmp1 %&gt;% filter(tmp1$category==&quot;red&quot;) %&gt;% nrow
  
tmp2 &lt;-   data.frame(KEGG_Reaction=genesss,complete=greenezzz,ps=redzzz,partial_ps=orangezzz)
final_KEGG &lt;- rbind(final_KEGG,tmp2)
}



spread(all_info_KEGG_yes_cleaned_Ldel, category, KEGG_Reaction)  %&gt;%replace(is.na(.), 0)


write.table(all_info_KEGG_yes_cleaned_Ldel,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/All_GENES_KEGGmap_ldel.tsv&quot;,quote = FALSE,row.names = FALSE,col.names = FALSE,sep = &quot;\t&quot;)



lhelv_coreness

###----------------------------------
#continue
###----------------------------------
Cogs_duplicated_ext &lt;- ggplot(complete_cogs_ps,aes(x=long,y=sort(counts,decreasing = TRUE),fill=cogs))+geom_bar(stat=&quot;identity&quot;)+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank())+facet_wrap(~species,ncol = 1)
Cogs_duplicated_ext

# cogs_comple_long &lt;- merge(all_cogs,duplicated_cogs,by=&quot;long&quot;) %&gt;%  mutate(Complete_genes=ALL_counts-Duplicated) %&gt;% dplyr::select(-ALL_counts) %&gt;% gather(.,group,count,c(Complete_genes,Duplicated))

colorsss=rev(c(&quot;grey44&quot;,&quot;grey90&quot;))

all_categories_02 &lt;- all_info %&gt;% group_by(long,species) %&gt;% summarise(all_genes=n())
only_ps_categories_02 &lt;- all_info %&gt;% filter(pseudogenes&gt;0)%&gt;% group_by(long,species) %&gt;% summarise(Pseudo=n())
 all_combined_wide &lt;- all_categories_02 %&gt;% left_join(.,only_ps_categories_02,by=c(&quot;long&quot;,&quot;species&quot;)) %&gt;% mutate(Complete_genes=all_genes-Pseudo) %&gt;% select(-all_genes)%&gt;% arrange(desc(Pseudo+Complete_genes))

all_combined_long &lt;- all_combined_wide %&gt;% mutate(Complete_genes=Complete_genes-Pseudo)%&gt;% gather(.,group,count,c(Pseudo,Complete_genes))

all_combined_long$species &lt;- plyr::revalue(all_combined_long$species,c(&quot;Ldel&quot;=&quot;L. delbrueckii&quot;,&quot;Sterm&quot;=&quot;S. thermophilus&quot;,&quot;Lhelv&quot;=&quot;L. helveticus&quot;))


Cogs_duplicated_ext &lt;- ggplot(all_combined_long,aes(x=reorder(long,-count),y=count,fill=group))+geom_bar(stat=&quot;identity&quot;)+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1,size=7),axis.title.x = element_blank(),legend.title = element_blank(),legend.position = &quot;none&quot;)+scale_fill_manual(values = colorsss)+facet_wrap(~species,ncol = 1,scales = &quot;free_y&quot;)
Cogs_duplicated_ext


pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Genes_pseudo_rep.pdf&quot;,width=6,height=6)
Cogs_duplicated_ext
dev.off()





for (speciesss in c(&quot;Sterm&quot;,&quot;Ldel&quot;,&quot;Lhelv&quot;)) {
  all_combined_wide_species &lt;- all_combined_wide %&gt;% filter(species==speciesss) %&gt;% select(-species)%&gt;% replace(is.na(.), 0) %&gt;% column_to_rownames(.,var=&quot;long&quot;)
chisq &lt;- chisq.test(all_combined_wide_species)
dev.off()
plot.new()
col3 &lt;- colorRampPalette(c(&quot;red&quot;, &quot;white&quot;, &quot;blue&quot;)) 
corsssplot &lt;- corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = &quot;color&quot;)
corsssplot
dev.off()
corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = &quot;color&quot;)
}


  ###---------------Sterm
  speciesss &lt;- &quot;Sterm&quot;
 
    all_combined_wide_species &lt;- all_combined_wide %&gt;% filter(species==speciesss) %&gt;% select(-species)%&gt;% replace(is.na(.), 0) %&gt;% column_to_rownames(.,var=&quot;long&quot;)
chisq &lt;- chisq.test(all_combined_wide_species)
dev.off()
plot.new()
col3 &lt;- colorRampPalette(c(&quot;red&quot;, &quot;white&quot;, &quot;blue&quot;)) 
corsssplot &lt;- corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = &quot;color&quot;)
corsssplot_Sterm &lt;- corsssplot$corrPos %&gt;% as_tibble() %&gt;% mutate(species=&quot;Sterm&quot;)

corsssplot
dev.off()
corrplot(chisq$resiStermduals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = &quot;color&quot;, title = speciesss)
  

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_pseudo_rep.pdf&quot;,width=6,height=10)
corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = &quot;color&quot;, title = speciesss)
dev.off()


###---------------Ldel
  speciesss &lt;- &quot;Ldel&quot;
 
    all_combined_wide_species &lt;- all_combined_wide %&gt;% filter(species==speciesss) %&gt;% select(-species)%&gt;% replace(is.na(.), 0) %&gt;% column_to_rownames(.,var=&quot;long&quot;)
chisq &lt;- chisq.test(all_combined_wide_species)
dev.off()
plot.new()
col3 &lt;- colorRampPalette(c(&quot;red&quot;, &quot;white&quot;, &quot;blue&quot;)) 
corsssplot &lt;- corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = &quot;color&quot;)
corsssplot_Ldel &lt;- corsssplot$corrPos %&gt;% as_tibble() %&gt;% mutate(species=&quot;Ldel&quot;)

corsssplot
dev.off()
corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = &quot;color&quot;, title = speciesss)
  

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Ldel_pseudo_rep.pdf&quot;,width=6,height=10)
corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = &quot;color&quot;, title = speciesss)
dev.off()

###---------------Lhelv
  speciesss &lt;- &quot;Lhelv&quot;
 
    all_combined_wide_species &lt;- all_combined_wide %&gt;% filter(species==speciesss) %&gt;% select(-species)%&gt;% replace(is.na(.), 0) %&gt;% column_to_rownames(.,var=&quot;long&quot;)
chisq &lt;- chisq.test(all_combined_wide_species)
dev.off()
plot.new()
col3 &lt;- colorRampPalette(c(&quot;red&quot;, &quot;white&quot;, &quot;blue&quot;)) 
corsssplot &lt;- corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = &quot;color&quot;)
corsssplot_Lhelv &lt;- corsssplot$corrPos %&gt;% as_tibble() %&gt;% mutate(species=&quot;Lhelv&quot;)
dev.off()
corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = &quot;color&quot;, title = speciesss)
  

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Lhelv_pseudo_rep.pdf&quot;,width=6,height=10)
corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = &quot;color&quot;, title = speciesss)
dev.off()

library(patchwork)

# Cogs_duplicated_ext+countsPlot_errorbar+ plot_layout(ncol=2,widths = c(1,3))

##----------------------------
###only illustrate the most and least conserved COG families 
##----------------------------

corsssplot_all &lt;- rbind(corsssplot_Lhelv,corsssplot_Ldel,corsssplot_Sterm) %&gt;% filter(xName==&quot;Pseudo&quot;)%&gt;% filter(yName %in% c(&quot;Carbohydrate metabolism and transport&quot;,&quot;Amino acid metabolism and transport&quot;,&quot;Inorganic ion transport and metabolism&quot;,&quot;Replication and repair&quot;,&quot;Translation&quot;))

    # dev.off()
# range(Biolog_diff_new$Value)
ggheatmap &lt;- ggplot(corsssplot_all, aes(y=yName, x=species, fill = corr))+
 geom_tile(color = &quot;white&quot;,size=1.1)+
 scale_fill_gradient2(low = &quot;blue&quot;, high = &quot;red&quot;, mid = &quot;white&quot;,
   midpoint = 0, limit = c(-1,1), space = &quot;Lab&quot;,
   name=&quot;representation&quot;) +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = &quot;top&quot;) +
  # labs(x=expression(italic(&quot;S. thermophilus&quot;)),y=expression(italic(&quot;L. delbrueckii&quot;)))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())

print(ggheatmap)
  
  


   pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/all_COG_representative_pseudogenes_colors.pdf&quot;,width=6,height=8)

  print(ggheatmap)

dev.off()

##--------------------


table(all_combined_long$long)

all_combined_long_sub &lt;- all_combined_long %&gt;% filter(long %in% c(&quot;Carbohydrate metabolism and transport&quot;,&quot;Amino acid metabolism and transport&quot;,&quot;Inorganic ion transport and metabolism&quot;,&quot;Replication and repair&quot;,&quot;Translation&quot;)) %&gt;% mutate(grouping=ifelse(long %in% c(&quot;Replication and repair&quot;,&quot;Translation&quot;),&quot;conserved&quot;,&quot;removed&quot;))

all_combined_long_sub

all_combined_long_sub$long &lt;- plyr::revalue(all_combined_long_sub$long,c(&quot;Carbohydrate metabolism and transport&quot;=&quot;Carbohydrate\nmetabolism\nand\ntransport&quot;,
 &quot;Amino acid metabolism and transport&quot;=&quot;Amino acid\nmetabolism\nand\ntransport&quot;,
 &quot;Inorganic ion transport and metabolism&quot;=&quot;Inorganic ion\ntransport\nand\nmetabolism&quot;,
 &quot;Replication and repair&quot;=&quot;Replication\nand\nrepair&quot;))
 
# all_combined_long_sub
colorsss=rev(c(&quot;grey44&quot;,&quot;grey95&quot;))


Cogs_duplicated_sub &lt;- ggplot(all_combined_long_sub,aes(x=species,y=count,fill=group))+geom_bar(stat=&quot;identity&quot;)+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1,size=7),axis.title.x = element_blank(),legend.title = element_blank(),legend.position = &quot;none&quot;)+scale_fill_manual(values = colorsss)+facet_wrap(~long,ncol = 5)+labs(y=&quot;# Orthogroups&quot;)
Cogs_duplicated_sub+ggheatmap
Cogs_duplicated_sub

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/all_COG_representative_pseudogenes.pdf&quot;,width=7,height=3.5)
Cogs_duplicated_sub
dev.off()</code></pre>
</div>
</div>
<div id="secretation-proteins" class="section level2">
<h2>secretation proteins</h2>
<pre class="r"><code>##-----------------------GO annotation
library(tidyverse)
 All_pseudogenes_annotation_again &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/All_pseudogenes_annotation.csv&quot;, 
     delim = &quot;\t&quot;, escape_double = FALSE, 
     trim_ws = TRUE)
 
table(All_pseudogenes_annotation_again$GOs) %&gt;% dim()
 
table(is.na(All_pseudogenes_annotation_again$GOs))
table(All_pseudogenes_annotation_again$GOs==&quot;-&quot;)


##-----------------------all ps OG

Ldel_genomes_ps.clustered.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt&quot;) %&gt;% mutate(Orthogroup=paste0(&quot;Ld_&quot;,Orthogroup)) %&gt;% select(Orthogroup,complete,lost,pseudogenes  )
# table(Ldel_genomes_ps.clustered.txt$complete)
Sterm_genomes_ps.clustered.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered.txt&quot;)%&gt;% mutate(Orthogroup=paste0(&quot;St_&quot;,Orthogroup))%&gt;% select(Orthogroup,complete,lost,pseudogenes  )
Lhelv_genomes_ps.clustered.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered.txt&quot;)%&gt;% mutate(Orthogroup=paste0(&quot;Lh_&quot;,Orthogroup))%&gt;% select(Orthogroup,complete,lost,pseudogenes  )

all_duplicated_OGs &lt;- rbind(Ldel_genomes_ps.clustered.txt,Sterm_genomes_ps.clustered.txt,Lhelv_genomes_ps.clustered.txt)

##-----------------------EGGNOG

Lhelv_Eggnog &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv.Eggnog.annotations&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE, skip = 4)%&gt;% filter(grepl(&quot;^Lh&quot;,`#query`))%&gt;% mutate(species=&quot;Lhelv&quot;)

Ldel_Eggnog &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel.Eggnog.annotations&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE, skip = 4)%&gt;% filter(grepl(&quot;^Ld&quot;,`#query`))%&gt;% mutate(species=&quot;Ldel&quot;)

Sterm_Eggnog &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm.Eggnog.annotations&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE, skip = 4) %&gt;% filter(grepl(&quot;^St&quot;,`#query`)) %&gt;% mutate(species=&quot;Sterm&quot;) 

tail(Lhelv_Eggnog)
all_eggnog &lt;- rbind(Sterm_Eggnog,Ldel_Eggnog,Lhelv_Eggnog) %&gt;% rename(.,&quot;Orthogroup&quot;=&quot;#query&quot;)

all_eggnog_peps &lt;- all_eggnog %&gt;% filter(grepl(&quot;pep&quot;,Preferred_name))

##-----------------------Signal P annotation
Sterm_all_genes_SignalP &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/cell_localization/Sterm_all_genes_SignalP.csv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, skip = 1) %&gt;% filter(X3==&quot;signal_peptide&quot;) %&gt;% rename(.,&quot;Orthogroup&quot;=&quot;X1&quot;)%&gt;% left_join(.,all_duplicated_OGs) %&gt;% left_join(.,all_eggnog)

Ldel_all_genes_SignalP &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/cell_localization/Ldel_all_genes_SignalP.csv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, skip = 1) %&gt;% filter(X3==&quot;signal_peptide&quot;) %&gt;% rename(.,&quot;Orthogroup&quot;=&quot;X1&quot;)%&gt;% left_join(.,all_duplicated_OGs) %&gt;% left_join(.,all_eggnog)


Lhelv_all_genes_SignalP &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/cell_localization/Lhelv_all_genes_SignalP.csv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, skip = 1) %&gt;% filter(X3==&quot;signal_peptide&quot;) %&gt;% rename(.,&quot;Orthogroup&quot;=&quot;X1&quot;)%&gt;% left_join(.,all_duplicated_OGs) %&gt;% left_join(.,all_eggnog)</code></pre>
</div>
<div id="check-peptidase" class="section level2">
<h2>check peptidase</h2>
<pre class="r"><code>library(tidyverse)
# library(plyr)

##-----------------------all ps OG

Ldel_genomes_ps.clustered.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt&quot;) %&gt;% mutate(Orthogroup=paste0(&quot;Ld_&quot;,Orthogroup)) %&gt;% select(Orthogroup,complete,lost,pseudogenes  )
# table(Ldel_genomes_ps.clustered.txt$complete)
Sterm_genomes_ps.clustered.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered.txt&quot;)%&gt;% mutate(Orthogroup=paste0(&quot;St_&quot;,Orthogroup))%&gt;% select(Orthogroup,complete,lost,pseudogenes  )
Lhelv_genomes_ps.clustered.txt &lt;-  read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered.txt&quot;)%&gt;% mutate(Orthogroup=paste0(&quot;Lh_&quot;,Orthogroup))%&gt;% select(Orthogroup,complete,lost,pseudogenes  )

all_duplicated_OGs &lt;- rbind(Ldel_genomes_ps.clustered.txt,Sterm_genomes_ps.clustered.txt,Lhelv_genomes_ps.clustered.txt)

##-----------------------Signal P annotation
Sterm_all_genes_SignalP &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/cell_localization/Sterm_all_genes_SignalP.csv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, skip = 1) %&gt;% rename(.,&quot;Orthogroup&quot;=&quot;X1&quot;) %&gt;% rename(.,&quot;CellularLoco&quot;=&quot;X3&quot;) #%&gt;% filter(X3==&quot;signal_peptide&quot;) %&gt;% left_join(.,all_duplicated_OGs) %&gt;% left_join(.,all_eggnog)

Ldel_all_genes_SignalP &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/cell_localization/Ldel_all_genes_SignalP.csv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, skip = 1) %&gt;% rename(.,&quot;Orthogroup&quot;=&quot;X1&quot;) %&gt;% rename(.,&quot;CellularLoco&quot;=&quot;X3&quot;) #%&gt;% filter(X3==&quot;signal_peptide&quot;) %&gt;% rename(.,&quot;Orthogroup&quot;=&quot;X1&quot;)%&gt;% left_join(.,all_duplicated_OGs) %&gt;% left_join(.,all_eggnog)


Lhelv_all_genes_SignalP &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/cell_localization/Lhelv_all_genes_SignalP.csv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE, skip = 1) %&gt;% rename(.,&quot;Orthogroup&quot;=&quot;X1&quot;) %&gt;% rename(.,&quot;CellularLoco&quot;=&quot;X3&quot;) #%&gt;% filter(X3==&quot;signal_peptide&quot;) %&gt;% rename(.,&quot;Orthogroup&quot;=&quot;X1&quot;)%&gt;% left_join(.,all_duplicated_OGs) %&gt;% left_join(.,all_eggnog)

all_SignalP&lt;- rbind(Sterm_all_genes_SignalP,Ldel_all_genes_SignalP,Lhelv_all_genes_SignalP) %&gt;% select(Orthogroup,CellularLoco)

##-----------------------EGGNOG

Lhelv_Eggnog &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv.Eggnog.annotations&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE, skip = 4)%&gt;% filter(grepl(&quot;^Lh&quot;,`#query`))%&gt;% mutate(species=&quot;Lhelv&quot;)

Ldel_Eggnog &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel.Eggnog.annotations&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE, skip = 4)%&gt;% filter(grepl(&quot;^Ld&quot;,`#query`))%&gt;% mutate(species=&quot;Ldel&quot;)

Sterm_Eggnog &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm.Eggnog.annotations&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE, skip = 4) %&gt;% filter(grepl(&quot;^St&quot;,`#query`)) %&gt;% mutate(species=&quot;Sterm&quot;) 

tail(Lhelv_Eggnog)
all_eggnog &lt;- rbind(Sterm_Eggnog,Ldel_Eggnog,Lhelv_Eggnog) %&gt;% rename(.,&quot;Orthogroup&quot;=&quot;#query&quot;)

all_eggnog_peps &lt;- all_eggnog %&gt;% filter(grepl(&quot;pep&quot;,Preferred_name)) %&gt;% left_join(.,all_duplicated_OGs)%&gt;% left_join(.,all_SignalP)
all_eggnog_prt &lt;- all_eggnog %&gt;% filter(grepl(&quot;prt&quot;,Preferred_name)) %&gt;% left_join(.,all_duplicated_OGs)%&gt;% left_join(.,all_SignalP)
all_eggnog_prt &lt;- all_eggnog %&gt;% filter(grepl(&quot;peptidase&quot;,Description)) %&gt;% left_join(.,all_duplicated_OGs)%&gt;% left_join(.,all_SignalP)

all_eggnog_prt &lt;- all_eggnog %&gt;% filter(grepl(&quot;protease&quot;,Description)) %&gt;% left_join(.,all_duplicated_OGs)%&gt;% left_join(.,all_SignalP)


all_eggnog_peps_presence &lt;- all_eggnog_peps %&gt;% group_by(Preferred_name,PFAMs,species,pseudogenes) %&gt;% summarise(counts=n()) %&gt;% mutate(pseudogenes=ifelse(pseudogenes&gt;0,&quot;yes&quot;,&quot;no&quot;))%&gt;%replace(is.na(.), &quot;no&quot;)%&gt;% select(-counts) %&gt;% unique() %&gt;% mutate(present=&quot;yes&quot;) %&gt;% spread(., pseudogenes, present)  %&gt;%replace(is.na(.), &quot;noPresent&quot;) %&gt;% mutate(final_psuedo=ifelse(yes==&quot;yes&quot;,&quot;pseudo&quot;,&quot;complete&quot;)) %&gt;% filter(species!=&quot;Lhelv&quot;)%&gt;% select(-no,-yes)%&gt;% spread(., species, final_psuedo) %&gt;%replace(is.na(.), &quot;missing&quot;)

all_eggnog_peps_presence_long &lt;- all_eggnog_peps_presence %&gt;% gather(., species,presence , c(&quot;Ldel&quot;,&quot;Sterm&quot;), factor_key=TRUE,na.rm = TRUE) 

colorss &lt;- c(&quot;red&quot;,&quot;grey88&quot;,&quot;grey77&quot;)

plotPeptidases &lt;- ggplot(all_eggnog_peps_presence_long,aes(x=species,y=Preferred_name,fill=presence,color=presence))+geom_tile()+theme_classic()+scale_fill_manual(values = colorss)+scale_color_manual(values = colorss)+labs(x=&quot;&quot;,y=&quot;Orthogroups Peptidases&quot;)+theme(legend.title = element_blank())
png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images//20210126/supplement//Peptidases_presence.png&quot;, width = , height = 3000,res=400)

# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement//plot_genome_size.pdf&quot;,width=6,height=5)

plotPeptidases

dev.off()
all_eggnog_peps_presence_long %&gt;% group_by(species,presence) %&gt;% summarise(counts=n())</code></pre>
</div>
<div id="check-kegg-pathways" class="section level2">
<h2>check KEGG pathways</h2>
<pre class="r"><code># write.table(only_ps_categories_for_Philipp_final,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/All_pseudogenes_annotation.csv&quot;,quote = FALSE,row.names = FALSE,sep = &quot;\t&quot;)
library(tidyverse)
 All_pseudogenes_annotation_again &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/All_pseudogenes_annotation.csv&quot;, 
     delim = &quot;\t&quot;, escape_double = FALSE, 
     trim_ws = TRUE)
 
 
carbohydrate_modules_info &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/KEGG_info/carbohydrate_modules_info.csv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE) %&gt;% mutate(KEGG_Module=Module)%&gt;% mutate(long_KEGG_Module=Definition) %&gt;% select(KEGG_Module,long_KEGG_Module)
  
carbohydrate_modules_info_02 &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/KEGG_info/carbohydrate_modules_info_02.csv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE) %&gt;% select(KEGG_Module,KeggModuleName,CategoryGlobal)#%&gt;% mutate(KEGG_Module=Module) %&gt;% mutate(long_KEGG_Module=Definition) 
  

  All_pseudogenes_annotation_again_carbo &lt;- All_pseudogenes_annotation_again %&gt;% filter(COG_category==&quot;G&quot;)
 
  
  ###----------------------------
  ##KEGG PATHWAYS
  ###----------------------------

  carbohydrate_pathways_info &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/KEGG_info/carbohydrate_pathways_info.csv&quot;, 
    col_names = c(&quot;KEGG_Pathway&quot;,&quot;long_KEGG_Pathway&quot;)) %&gt;% mutate(KEGG_Pathway=gsub(&quot;^&quot;,&quot;ko&quot;,KEGG_Pathway))
  
# ALL_sterm &lt;- ALL_sterm %&gt;% 
#     mutate(COG_Functional_Category = strsplit(as.character(COG_Functional_Category), &quot;&quot;)) %&gt;% 
#     unnest(COG_Functional_Category)
 
 all_info_KEGG_pathway &lt;- All_pseudogenes_annotation_again_carbo %&gt;% select(&quot;#query&quot;,&quot;KEGG_Pathway&quot;,&quot;Preferred_name&quot;,&quot;COG_category&quot;, &quot;complete&quot;   ,&quot;lost&quot; ,&quot;pseudogenes&quot;) %&gt;% mutate(KEGG_Pathway = strsplit(as.character(KEGG_Pathway), &quot;,&quot;)) %&gt;%   unnest(KEGG_Pathway)
 
tmp &lt;- all_info_KEGG_pathway %&gt;% filter(grepl(&quot;ko&quot;,KEGG_Pathway)) %&gt;% left_join(.,carbohydrate_pathways_info) 
all_info_KEGG_pathway %&gt;% filter(grepl(&quot;ko&quot;,KEGG_Pathway)) %&gt;% left_join(.,carbohydrate_pathways_info) %&gt;% group_by(`#query`) %&gt;% summarise(count=n())%&gt;% arrange(-count) 


all_info_KEGG_pathway_counts &lt;- all_info_KEGG_pathway %&gt;% filter(grepl(&quot;ko&quot;,KEGG_Pathway)) %&gt;% left_join(.,carbohydrate_pathways_info) %&gt;% group_by(KEGG_Pathway) %&gt;% summarise(count=n())%&gt;% arrange(-count) 

all_info_KEGG_pathway_counts_sub &lt;- all_info_KEGG_pathway_counts %&gt;% filter(!is.na(long_KEGG_Pathway))




###----------------------summary of modules
 all_info_KEGG_pathway_02 &lt;- All_pseudogenes_annotation_again_carbo %&gt;% select(&quot;#query&quot;,&quot;KEGG_Pathway&quot;,&quot;KEGG_Module&quot;,&quot;Preferred_name&quot;,&quot;COG_category&quot;, &quot;complete&quot;   ,&quot;lost&quot; ,&quot;pseudogenes&quot;) %&gt;% mutate(KEGG_Pathway = strsplit(as.character(KEGG_Pathway), &quot;,&quot;)) %&gt;%   unnest(KEGG_Pathway) %&gt;% filter(KEGG_Pathway %in% all_info_KEGG_pathway_counts_sub$KEGG_Pathway)%&gt;% mutate(KEGG_Module = strsplit(as.character(KEGG_Module), &quot;,&quot;)) %&gt;%   unnest(KEGG_Module)%&gt;% left_join(.,carbohydrate_modules_info_02) %&gt;% left_join(.,carbohydrate_pathways_info)
 
 table(all_info_KEGG_pathway_02$KEGG_Pathway)

 all_info_KEGG_pathway_02_summary &lt;- all_info_KEGG_pathway_02 %&gt;% group_by(long_KEGG_Pathway,KeggModuleName) %&gt;% summarise(counts=n())%&gt;% arrange(-counts)
 
  all_info_KEGG_pathway_02_summary &lt;- all_info_KEGG_pathway_02 %&gt;% group_by(long_KEGG_Pathway,KeggModuleName) %&gt;% summarise(counts=n())%&gt;% arrange(-counts) %&gt;% filter(!is.na(KeggModuleName))

   all_info_KEGG_pathway_02_summary$KeggModuleName  = factor(all_info_KEGG_pathway_02_summary$KeggModuleName , levels=all_info_KEGG_pathway_02_summary$KeggModuleName)

  
plot_KEGG &lt;- ggplot(all_info_KEGG_pathway_02_summary,aes(x=reorder(KeggModuleName, -counts),y=counts,fill=long_KEGG_Pathway,color=long_KEGG_Pathway))+geom_bar(stat = &quot;identity&quot;)+theme_classic()+theme(axis.text.x = element_text(angle=85,hjust=1),axis.title.x = element_blank())+labs(y=&quot;Pseudogenized OGs&quot;)
  
plot_KEGG

png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images//20210126/supplement//KEGG_Pathways_pseudo.png&quot;, width = 4000, height = 3000,res=400)

# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement//plot_genome_size.pdf&quot;,width=6,height=5)

plot_KEGG

dev.off()


  ###----------------------------
  ##KEGG MODULES
  ###----------------------------

# ALL_sterm &lt;- ALL_sterm %&gt;% 
#     mutate(COG_Functional_Category = strsplit(as.character(COG_Functional_Category), &quot;&quot;)) %&gt;% 
#     unnest(COG_Functional_Category)
 
 all_info_KEGG_MODULES &lt;- All_pseudogenes_annotation_again_carbo %&gt;% select(&quot;#query&quot;,&quot;KEGG_Module&quot;,&quot;Preferred_name&quot;,&quot;COG_category&quot;, &quot;complete&quot;   ,&quot;lost&quot; ,&quot;pseudogenes&quot;) %&gt;% mutate(KEGG_Module = strsplit(as.character(KEGG_Module), &quot;,&quot;)) %&gt;%   unnest(KEGG_Module)
 

all_info_KEGG_MODULES_counts &lt;- all_info_KEGG_MODULES %&gt;% group_by(KEGG_Module) %&gt;% summarise(count=n())%&gt;% arrange(-count) %&gt;% left_join(.,carbohydrate_modules_info) %&gt;% left_join(.,carbohydrate_modules_info_02)  #%&gt;% filter(grepl(&quot;ko&quot;,KEGG_Module))</code></pre>
</div>
<div id="draw-pseudo-gene-keggmap" class="section level2">
<h2>draw pseudo gene KEGGmap</h2>
<pre class="r"><code>library(pathview)

##------------example
load(url(&quot;http://genomedata.org/gen-viz-workshop/pathway_visualization/pathview_Data.RData&quot;))
pathview(gene.data=tumor_v_normal_DE.fc, species=&quot;hsa&quot;, pathway.id=&quot;hsa03460&quot;, kegg.native=FALSE)

#-------data

###----------------------------------
#completeness of OG (core,cloud info)
###----------------------------------
# Lhelv_OG_abundance &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_cluster_ps_counts_wide_imported_wide_final&quot;) %&gt;% mutate(Orthogroup=paste0(&quot;Lh_&quot;,Orthogroup))
# Lhelv_OG_abundancesums &lt;- Lhelv_OG_abundance %&gt;% group_by(Orthogroup,grouping,species)  %&gt;% summarise(counts=n())%&gt;% spread(., grouping, counts)%&gt;%replace(is.na(.), 0)
# total_gene &lt;- rowSums(Lhelv_OG_abundancesums[,c(&quot;ps&quot;,&quot;re&quot;,&quot;rm&quot;)]) %&gt;% head(1) %&gt;% as.numeric()
# Lhelv_OG_abundancesums$coreness &lt;- 100*(Lhelv_OG_abundancesums$re/total_gene)
# lhelv_coreness &lt;- Lhelv_OG_abundancesums %&gt;% select(Orthogroup,coreness) %&gt;% as_tibble()
# 
# merge(lhelv_coreness,all_info_KEGG_yes_cleaned,by.x=&quot;Orthogroup&quot;,by.y=&quot;#query&quot;,all=TRUE)%&gt;% filter(species==&quot;Lh&quot;) %&gt;%replace(is.na(.), 0) %&gt;% as_tibble()
# # all_info_KEGG_yes_cleaned
# 


Orthogroups_orthofinder_stats_test &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered_counts.txt&quot;)%&gt;% mutate(Orthogroup=paste0(&quot;Lh_&quot;,Orthogroup))


Orthogroups_orthofinder_stats_test_keepers_01 &lt;- Orthogroups_orthofinder_stats_test %&gt;% mutate(gene_type=ifelse(re&gt;0,&quot;re&quot;,ifelse(ps&gt;0,&quot;ps&quot;,&quot;rm&quot;)))%&gt;% mutate(genome=gsub(&quot;.all&quot;,&quot;&quot;,sample)) %&gt;% select(-sample,-ps,-re) %&gt;% spread(., genome, gene_type)  %&gt;%replace(is.na(.), &quot;rm&quot;) 

cluster_ps_counts_wide_imported_wide_final &lt;-  Orthogroups_orthofinder_stats_test_keepers_01 %&gt;%  gather(., genome,grouping , colnames(Orthogroups_orthofinder_stats_test_keepers_01)[2:ncol(Orthogroups_orthofinder_stats_test_keepers_01)], factor_key=TRUE,na.rm = TRUE)  %&gt;% mutate(species=&quot;Lhelv&quot;)


Lhelv_OG_abundancesums &lt;- cluster_ps_counts_wide_imported_wide_final %&gt;% group_by(Orthogroup,grouping,species)  %&gt;% summarise(counts=n())%&gt;% spread(., grouping, counts)%&gt;%replace(is.na(.), 0)
total_gene &lt;- rowSums(Lhelv_OG_abundancesums[,c(&quot;ps&quot;,&quot;re&quot;,&quot;rm&quot;)]) %&gt;% head(1) %&gt;% as.numeric()
Lhelv_OG_abundancesums$coreness &lt;- 100*(Lhelv_OG_abundancesums$re/total_gene)
lhelv_coreness &lt;- Lhelv_OG_abundancesums %&gt;% select(Orthogroup,coreness) %&gt;% as_tibble()

Lhelv_all_info &lt;- merge(lhelv_coreness,all_info_KEGG_yes_cleaned,by.x=&quot;Orthogroup&quot;,by.y=&quot;#query&quot;,all=TRUE)%&gt;% filter(species==&quot;Lh&quot;) %&gt;%replace(is.na(.), 0) %&gt;% as_tibble() %&gt;% mutate(fraction_core=coreness/100)

Lhelv_all_info_KEGGmap &lt;- setNames(as.character(Lhelv_all_info$KEGG_Reaction), Lhelv_all_info$fraction_core)

Lhelv_all_info_forONLINE &lt;- Lhelv_all_info %&gt;% select(KEGG_Reaction,category)
write.table(Lhelv_all_info_forONLINE,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/All_pseudogenes_annotation_02.csv&quot;,quote = FALSE,row.names = FALSE,sep = &quot;\t&quot;)

Lhelv_all_info %&gt;% filter(pseudogenes&gt;0)
unique(Lhelv_all_info$KEGG_Reaction)

##---------------------maps

pathview(gene.data=Lhelv_all_info_KEGGmap,pathway.id=&quot;01200&quot;,species=&quot;ko&quot;, kegg.native=FALSE)
pathview(gene.data=Lhelv_all_info_KEGGmap,pathway.id=&quot;00052&quot;,species=&quot;ko&quot;, kegg.native=FALSE)

map01200
00052 </code></pre>
<p>###extension pseudo genes extenstion with new</p>
<pre class="bash"><code>##===============================
##Lactobacillus
##===============================
#mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
#curl &#39;ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt&#39; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt

mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/


rm /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
genus=Lactobacillus

rm  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=iners

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39; |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=rodentium

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=hominis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
 
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=gasseri

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep -v &quot;paragasseri&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep -v &quot;paragasseri&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39; |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=johnsonii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39; |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=taiwanensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=amylolyticus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names


###--------------------------
species=psittaci

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; |head -1 &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39;|head -1 &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=jensenii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=equicursoris

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=pasteurii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=gigeriorum

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=acetotolerans

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=helsingborgensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=melliventris

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kimbladii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kullabergensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;representative genome&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;representative genome&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=bombicola

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=panisapium

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=apis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;  |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=intestinalis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=hamsteri

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kalixensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=helveticus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;Lactobacillus&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;Lactobacillus&quot;|cut -f 20   |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=gallinarum

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20   |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kefiranofaciens

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=crispatus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=acidophilus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
    
    
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20   |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=ultunensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kitasatonis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=amylovorus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot; |cut -f 20   |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=delbrueckii
subsp=jakobsenii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; -v subss=&quot;$subsp&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin&quot;_&quot;subss}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=delbrueckii
subsp=indicus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; -v subss=&quot;$subsp&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin&quot;_&quot;subss}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=delbrueckii
subsp=lactis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot; |cut -f 20  |awk -v namezzin=&quot;$species&quot; -v subss=&quot;$subsp&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin&quot;_&quot;subss}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=delbrueckii
subsp=sunkii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot; |cut -f 20 |awk -v namezzin=&quot;$species&quot; -v subss=&quot;$subsp&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin&quot;_&quot;subss}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=delbrueckii
subsp=bulgaricus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot; |cut -f 20 |awk -v namezzin=&quot;$species&quot; -v subss=&quot;$subsp&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin&quot;_&quot;subss}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###-------------------------------------
##download
###-------------------------------------

rm -r /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF
mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF

cd /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF

    wget -i /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
   
#mv GCF_000769515.1_ASM76951v1_genomic.gbff.gz Out_Bacsub.faa.gz

###========================================================
#extract counts
###========================================================

#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt
rm /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/pseudogeness_count_new.txt



#for genomesss in $(ls /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF |grep &quot;.gz$&quot; )

for genomesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt |cut -d &#39;/&#39; -f 11)
do
genomesss_short=$(echo ${genomesss} |sed &#39;s/_genomic.gbff.gz//g&#39; )
species=$(grep &quot;${genomesss_short}&quot; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |cut -f 2)
genezzz=$(zcat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF/${genomesss}|grep &quot;Genes (coding)&quot; |cut -d &#39;:&#39; -f 3|sed &#39;s/ //g&#39;|sed &#39;s/,//g&#39;|head -1)
peudosss=$(zcat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF/${genomesss}|grep &quot;Pseudo Genes (total)&quot;  |cut -d &#39;:&#39; -f 3|sed &#39;s/ //g&#39;|sed &#39;s/,//g&#39;|head -1)
transpos=$(zgrep &quot;transposase&quot; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF/${genomesss} |grep &quot;product=&quot; -c)


echo -e ${species}&quot;\t&quot;${genomesss}&quot;\t&quot;${genezzz}&quot;\t&quot;${peudosss}&quot;\t&quot;${transpos}  &gt;&gt;  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/pseudogeness_count_new.txt
done #genomesss


#cut -f 2 /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |sort|uniq| awk -F &quot;\t&quot; &#39;{OFS=&quot;=&quot;}{print &quot;\&quot;Lactobacillus\&quot;&quot;,&quot;\&quot;&quot;$1&quot;\&quot;&quot;}&#39;

cut -f 2 /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |sort|uniq| awk -F &quot;\t&quot; &#39;{OFS=&quot;=&quot;}{print &quot;\&quot;&quot;$1&quot;\&quot;&quot;,&quot;\&quot;Lactobacillus\&quot;&quot;}&#39;
</code></pre>
<p>###Strepto</p>
<pre class="bash"><code>##-----------------------------
##Streptococcus
##-----------------------------

#mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
#curl &#39;ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt&#39; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt

mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log


rm /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt
genus=Streptococcus

rm  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

###--------------------------
species=pneumoniae

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; |head -100 &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39; |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; |head -100 &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

###--------------------------
species=mitis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


###--------------------------
species=parasanguinis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=sanguinis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;para&quot; -v|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;para&quot; -v|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=suis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=anginosus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=dysgalactiae
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=pyogenes
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=uberis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=gallolyticus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names
##--------------------------
species=equinus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=vestibularis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=thermophilus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=salivarius
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;K12&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;K12&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=downei
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=criceti
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=mutans
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=macacae
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|&#39; &gt;&gt;    \
     /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39; |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


###-------------------------------------
##download
###-------------------------------------

rm -r /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/genomes/GBF
mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/GBF

cd /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/GBF

    wget -i /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

#mv GCF_000769515.1_ASM76951v1_genomic.gbff.gz Out_Bacsub.faa.gz

###========================================================
#extract counts
###========================================================

#rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt
#rm /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/pseudogeness_count_new.txt


for genomesss in $(ls /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/GBF |grep &quot;.gz$&quot; )
do
genomesss_short=$(echo ${genomesss} |sed &#39;s/_genomic.gbff.gz//g&#39; )
species=$(grep &quot;${genomesss_short}&quot; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2)
genezzz=$(zcat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/GBF/${genomesss}|grep &quot;Genes (coding)&quot; |cut -d &#39;:&#39; -f 3|sed &#39;s/ //g&#39;|sed &#39;s/,//g&#39;|head -1)
peudosss=$(zcat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/GBF/${genomesss}|grep &quot;Pseudo Genes (total)&quot;  |cut -d &#39;:&#39; -f 3|sed &#39;s/ //g&#39;|sed &#39;s/,//g&#39;|head -1)
transpos=$(zgrep &quot;transposase&quot; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/GBF/${genomesss} |grep &quot;product=&quot; -c)


echo -e ${species}&quot;\t&quot;${genomesss}&quot;\t&quot;${genezzz}&quot;\t&quot;${peudosss}&quot;\t&quot;${transpos}  &gt;&gt;  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/pseudogeness_count_new.txt
done #genomesss

###--------------------------
#rename
cut -f 2 /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |sort|uniq| awk -F &quot;\t&quot; &#39;{OFS=&quot;=&quot;}{print &quot;\&quot;&quot;$1&quot;\&quot;&quot;,&quot;\&quot;Streptococcus\&quot;&quot;}&#39;
</code></pre>
<p>transfer local</p>
<pre class="bash"><code>  
  mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/
  scp  vincent@130.223.51.66:/work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/pseudogeness_count_new.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count_new.txt

cat /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt \
/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count_new.txt &gt; \
/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count_together.txt</code></pre>
<div id="together" class="section level4">
<h4>together</h4>
<p>do analysis</p>
<pre class="r"><code>library(readr)
library(tidyverse)
pseudogeness_count &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count_together.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;species&quot;,&quot;genome&quot;,&quot;genes_coding&quot;,&quot;pseudogenes&quot;,&quot;tranposons&quot;), trim_ws = TRUE) %&gt;% filter(pseudogenes&lt;1000) %&gt;% filter(genes_coding&lt;4000)%&gt;% mutate(pseudogenes_percent = 100*(pseudogenes/genes_coding)) %&gt;% mutate(tranposons_percent = 100*(tranposons/genes_coding))%&gt;% filter(species!=&quot;Wigg&quot;)%&gt;% filter(species!=&quot;Scap&quot;)%&gt;% filter(species!=&quot;Lplan&quot;)


table(pseudogeness_count$species)

###--------------------
##filter
# remove species with less then 5 genomes
###--------------------

table(pseudogeness_count$species)[table(pseudogeness_count$species)&gt;5] 
table(pseudogeness_count$species)[table(pseudogeness_count$species)&lt;4]

keeps &lt;- table(pseudogeness_count$species)[table(pseudogeness_count$species)&gt;3] %&gt;% names()

pseudogeness_count &lt;- pseudogeness_count %&gt;% filter(species %in% keeps)
###--------------------
##point plot
###--------------------
plotpoints &lt;- ggplot(pseudogeness_count,aes(x=genes_coding,y=pseudogenes_percent,fill=species,color=species))+geom_point()+labs(x=&quot;Gene number&quot;,y=&quot;Pseudogenes [%]&quot;)+theme_classic()+theme(legend.title = element_blank())
plotpoints
  
  library(plotly)
ggp &lt;- ggplotly(plotpoints)
 ggp
# pseudogeness_count %&gt;% group_by(species)%&gt;%ize(meanPseudo=mean(pseudogeness_count),meangenes=mean(genes_coding),stdevPseudo=sd(pseudogeness_count),stdevgenes=stderr(genes_coding))
# pseudogeness_count %&gt;% group_by(species) %&gt;% summarise(meanPseudo=mean(pseudogeness_count))

 
 
pseudogeness_count$species &lt;- plyr::revalue(pseudogeness_count$species,c(&quot;delbrueckii_lactis&quot;=&quot;L. delbrueckii subsp. lactis&quot;,
 &quot;Ldel&quot;=&quot;L. delbrueckii subsp. lactis&quot;,
 &quot;thermophilus&quot;=&quot;S. thermophilus&quot;,
 &quot;Sterm&quot;=&quot;S. thermophilus&quot;,
 &quot;Ssuis&quot;=&quot;suis&quot;,
 &quot;Lequi&quot;=&quot;equicursoris&quot;,
 &quot;Svest&quot;=&quot;vestibularis&quot;))
 
pseudogeness_sumary &lt;- pseudogeness_count %&gt;% group_by(species) %&gt;%  dplyr::summarize(meanPseudo=100*(mean(pseudogenes)/mean(genes_coding)),meangenes=mean(genes_coding),stdevPseudo=100*(sd(pseudogenes)/mean(genes_coding)),stdevgene=sd(genes_coding),meantransp=100*(mean(tranposons)/mean(genes_coding)),stdevtransp=100*(sd(tranposons)/mean(genes_coding)),meanboth=100*(mean(pseudogenes+tranposons)/mean(genes_coding)),stdevboth=100*(sd(pseudogenes+tranposons)/mean(genes_coding)),counts=n())



 # dplyr::summarize(mean = mean(NumGenesLineage),sd=sd(NumGenesLineage))

##-----
#chnage names
##-----
  pseudogeness_sumary$species_new &lt;- pseudogeness_sumary$species

 
  # pseudogeness_sumary$species_new &lt;- plyr::revalue(pseudogeness_sumary$species,c(&quot;Buch&quot;=&quot;Buchnera spp.&quot;,&quot;Sglos&quot;=&quot;S. glossinidius&quot;,&quot;Ldel&quot;=&quot;L. delbrueckii&quot;,&quot;Sterm&quot;=&quot;S. thermophilus&quot;,&quot;Mleprae&quot;=&quot;M. leprae&quot;,&quot;Rick&quot;=&quot;Rickettsia spp.&quot;,&quot;Lplan&quot;=&quot;L. plantarum&quot;,&quot;Ssuis&quot;=&quot;S. suis&quot;,&quot;Lequi&quot;=&quot;L. equicursoris&quot;,&quot;Svest&quot;=&quot;S. vestibularis&quot;,&quot;Wol&quot;=&quot;Wolbachia spp.&quot;))
#,&quot;Wigg&quot;=&quot;Wigglesworthia&quot;
table(pseudogeness_sumary$species)
table(pseudogeness_sumary$species_new)  
  


 # pseudogeness_sumary$species_new  = factor(pseudogeness_sumary$species_new , levels=c(&quot;L. delbrueckii&quot;,&quot;S. thermophilus&quot;,&quot;L. plantarum&quot;,&quot;L. equicursoris&quot;,&quot;S. suis&quot;,&quot;S. vestibularis&quot;,&quot;M. leprae&quot;,&quot;Rickettsia spp.&quot;,&quot;Buchnera spp.&quot;,&quot;S. glossinidius&quot;,&quot;Wolbachia spp.&quot;))

 
# colorsss &lt;- c(&quot;#fdc2b5ff&quot;,&quot;#007fa6ff&quot;,&quot;#ebbc51ff&quot;,&quot;#ebbc51ff&quot;,&quot;#9fcfffff&quot;,&quot;#9fcfffff&quot;,&quot;#51eb85ff&quot;,&quot;#51eb85ff&quot;,&quot;#d58656ff&quot;,&quot;#d58656ff&quot;,&quot;#d58656ff&quot;) 
# colorsss &lt;- c(&quot;#fdc2b5ff&quot;,&quot;#007fa6ff&quot;,&quot;grey77&quot;,&quot;grey77&quot;,&quot;grey77&quot;,&quot;grey77&quot;,&quot;#51eb85ff&quot;,&quot;#51eb85ff&quot;,&quot;#d58656ff&quot;,&quot;#d58656ff&quot;,&quot;#d58656ff&quot;)

##-----
#plot size vs. pseudogenes
##-----
plotpoints &lt;-ggplot( pseudogeness_sumary, aes(x = meangenes, y = meanPseudo, xmin = meangenes-stdevgene, xmax = meangenes+stdevgene,ymin=meanPseudo-stdevPseudo,ymax=meanPseudo+stdevPseudo,fill=species_new,color=species_new)) +
  geom_point(size=2) +
  geom_errorbarh(alpha=.5)+
  geom_errorbar(alpha=.5)+
    labs(x=&quot;Gene number&quot;,y=&quot;Pseudogenes [%]&quot;)+theme_classic()+theme(legend.title = element_blank())+
  #geom_hline(yintercept =7.5, linetype=&quot;dotted&quot;)+geom_segment(aes(x = 1500, y = -1, xend = 1500, yend = 7.5), linetype = &quot;dotted&quot;)+
  # scale_color_manual(values = colorsss)+scale_fill_manual(values = colorsss)+
  annotate(&quot;text&quot;, x = 800, y = 6, label = &quot;endosymbionts&quot;,col=&quot;#d58656ff&quot;)+
  annotate(&quot;text&quot;, x = 800, y = 25, label = &quot;transition&quot;,col=&quot;#51eb85ff&quot;)+
  annotate(&quot;text&quot;, x = 2000, y = 1, label = &quot;free living&quot;,col=&quot;grey77&quot;)

#+lims(x=c(250,4100),y=c(-1.5,40))
plotpoints

  library(plotly)
ggp &lt;- ggplotly(plotpoints)
 ggp

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenes.pdf&quot;,width=6,height=4)

plotpoints

dev.off()
##-----
#plot transposase vs. pseudogenes
##-----
plotpoints_withtranps &lt;- ggplot(pseudogeness_sumary, aes(x = meantransp, y = meanPseudo, xmin = meantransp-stdevtransp, xmax = meantransp+stdevtransp,ymin=meanPseudo-stdevPseudo,ymax=meanPseudo+stdevPseudo,fill=species_new,color=species_new)) +
  geom_point(size=2) +
  geom_errorbarh(alpha=.5)+
  geom_errorbar(alpha=.5)+
    labs(x=&quot;transposase&quot;,y=&quot;Pseudogenes [%]&quot;)+theme_classic()+theme(legend.title = element_blank())
  #geom_hline(yintercept =7.5, linetype=&quot;dotted&quot;)+geom_segment(aes(x = 1500, y = -1, xend = 1500, yend = 7.5), linetype = &quot;dotted&quot;)+
  # scale_color_manual(values = colorsss)+scale_fill_manual(values = colorsss)+
  # annotate(&quot;text&quot;, x = 800, y = 6, label = &quot;endosymbionts&quot;,col=&quot;#d58656ff&quot;)+
  # annotate(&quot;text&quot;, x = 800, y = 25, label = &quot;transition&quot;,col=&quot;#51eb85ff&quot;)+
  # annotate(&quot;text&quot;, x = 2000, y = 1, label = &quot;free living&quot;,col=&quot;grey77&quot;)

#+lims(x=c(250,4100),y=c(-1.5,40))
plotpoints_withtranps

  library(plotly)
ggp &lt;- ggplotly(plotpoints_withtranps)
 ggp
##-----
#plot transposase +pseudogenes vs. genome size
##-----
 
 
plotpoints_both &lt;- ggplot( pseudogeness_sumary, aes(x = meangenes, y = meanboth, xmin = meangenes-stdevgene, xmax = meangenes+stdevgene,ymin=meanboth-stdevboth,ymax=meanboth+stdevboth,fill=species_new,color=species_new,text=paste(&quot;counts=&quot;,counts))) +
  geom_point(size=2) +
  geom_errorbarh(alpha=.2)+
  geom_errorbar(alpha=.2)+
    labs(x=&quot;Gene number&quot;,y=&quot;Pseudogenes+transposons [%]&quot;)+theme_classic()+theme(legend.title = element_blank())
  #geom_hline(yintercept =7.5, linetype=&quot;dotted&quot;)+geom_segment(aes(x = 1500, y = -1, xend = 1500, yend = 7.5), linetype = &quot;dotted&quot;)+
  # scale_color_manual(values = colorsss)+scale_fill_manual(values = colorsss)+
  # annotate(&quot;text&quot;, x = 800, y = 6, label = &quot;endosymbionts&quot;,col=&quot;#d58656ff&quot;)+
  # annotate(&quot;text&quot;, x = 800, y = 25, label = &quot;transition&quot;,col=&quot;#51eb85ff&quot;)+
  # annotate(&quot;text&quot;, x = 2000, y = 1, label = &quot;free living&quot;,col=&quot;grey77&quot;)

#+lims(x=c(250,4100),y=c(-1.5,40))
plotpoints_both

  library(plotly)
ggp &lt;- ggplotly(plotpoints_both)
 ggp


# plotpoints_wo &lt;- plotpoints+theme(legend.position = &quot;none&quot;)
# plotpoints_wo
# plotpoints+stat_ellipse(data=pseudogeness_count,aes(x=genes_coding,y=pseudogenes_percent,fill=species,color=species,group=species),type = &quot;euclid&quot;, level = 3) 

  
 ##----------------------------------------
 ##change coloring
 ##----------------------------------------
 
 dput(unique(pseudogeness_sumary$species))
 
 pseudogeness_sumary$species_new_02 &lt;- plyr::revalue(pseudogeness_sumary$species,c(&quot;anginosus&quot;=&quot;Streptococcus&quot;,
&quot;criceti&quot;=&quot;Streptococcus&quot;,
&quot;downei&quot;=&quot;Streptococcus&quot;,
&quot;dysgalactiae&quot;=&quot;Streptococcus&quot;,
&quot;equinus&quot;=&quot;Streptococcus&quot;,
&quot;gallolyticus&quot;=&quot;Streptococcus&quot;,
&quot;macacae&quot;=&quot;Streptococcus&quot;,
&quot;mitis&quot;=&quot;Streptococcus&quot;,
&quot;mutans&quot;=&quot;Streptococcus&quot;,
&quot;parasanguinis&quot;=&quot;Streptococcus&quot;,
&quot;pneumoniae&quot;=&quot;Streptococcus&quot;,
&quot;pyogenes&quot;=&quot;Streptococcus&quot;,
&quot;salivarius&quot;=&quot;Streptococcus&quot;,
&quot;sanguinis&quot;=&quot;Streptococcus&quot;,
&quot;suis&quot;=&quot;Streptococcus&quot;,
# &quot;thermophilus&quot;=&quot;Streptococcus&quot;,
&quot;uberis&quot;=&quot;Streptococcus&quot;,
&quot;vestibularis&quot;=&quot;Streptococcus&quot;,
&quot;helveticus&quot;=&quot;L.helveticus&quot;,
&quot;acetotolerans&quot;=&quot;Lactobacillus&quot;,
&quot;acidophilus&quot;=&quot;Lactobacillus&quot;,
&quot;amylolyticus&quot;=&quot;Lactobacillus&quot;,
&quot;amylovorus&quot;=&quot;Lactobacillus&quot;,
&quot;apis&quot;=&quot;Lactobacillus&quot;,
&quot;bombicola&quot;=&quot;Lactobacillus&quot;,
&quot;_bulgaricus&quot;=&quot;Lactobacillus&quot;,
&quot;crispatus&quot;=&quot;Lactobacillus&quot;,
# &quot;delbrueckii_indicus&quot;=&quot;Lactobacillus&quot;,
# &quot;delbrueckii_lactis&quot;=&quot;Lactobacillus&quot;,
# &quot;delbrueckii_sunkii&quot;=&quot;Lactobacillus&quot;,
&quot;equicursoris&quot;=&quot;Lactobacillus&quot;,
&quot;gallinarum&quot;=&quot;Lactobacillus&quot;,
&quot;gasseri&quot;=&quot;Lactobacillus&quot;,
&quot;gigeriorum&quot;=&quot;Lactobacillus&quot;,
&quot;hamsteri&quot;=&quot;Lactobacillus&quot;,
&quot;helsingborgensis&quot;=&quot;Lactobacillus&quot;,
&quot;hominis&quot;=&quot;Lactobacillus&quot;,
&quot;iners&quot;=&quot;Lactobacillus&quot;,
&quot;intestinalis&quot;=&quot;Lactobacillus&quot;,
&quot;jensenii&quot;=&quot;Lactobacillus&quot;,
&quot;johnsonii&quot;=&quot;Lactobacillus&quot;,
&quot;kalixensis&quot;=&quot;Lactobacillus&quot;,
&quot;kefiranofaciens&quot;=&quot;Lactobacillus&quot;,
&quot;kimbladii&quot;=&quot;Lactobacillus&quot;,
&quot;kitasatonis&quot;=&quot;Lactobacillus&quot;,
&quot;kullabergensis&quot;=&quot;Lactobacillus&quot;,
&quot;melliventris&quot;=&quot;Lactobacillus&quot;,
&quot;panisapium&quot;=&quot;Lactobacillus&quot;,
&quot;pasteurii&quot;=&quot;Lactobacillus&quot;,
&quot;psittaci&quot;=&quot;Lactobacillus&quot;,
&quot;rodentium&quot;=&quot;Lactobacillus&quot;,
&quot;taiwanensis&quot;=&quot;Lactobacillus&quot;,
&quot;ultunensis&quot;=&quot;Lactobacillus&quot;,
 #  &quot;Buch&quot;=&quot;Buchnera spp.&quot;,
 # &quot;Sglos&quot;=&quot;S. glossinidius&quot;,
 # &quot;Mleprae&quot;=&quot;M. leprae&quot;,
 # &quot;Rick&quot;=&quot;Rickettsia spp.&quot;,
 # &quot;Wol&quot;=&quot;Wolbachia spp.&quot;
 &quot;Buch&quot;=&quot;Endosymbiont&quot;,
 # &quot;Sglos&quot;=&quot;S. glossinidius&quot;,
 # &quot;Mleprae&quot;=&quot;M. leprae&quot;,
 &quot;Rick&quot;=&quot;Transition&quot;,
 &quot;Wol&quot;=&quot;Transition&quot;,
&quot;Port&quot;=&quot;Endosymbiont&quot;))
  table(pseudogeness_sumary$species_new_02)
 pseudogeness_sumary$species_new_02 = factor(pseudogeness_sumary$species_new_02, levels=c(&quot;Endosymbiont&quot;,  &quot;Transition&quot;, &quot;S. thermophilus&quot;, &quot;Streptococcus&quot; , &quot;L. delbrueckii subsp. lactis&quot;,&quot;Lactobacillus&quot;, &quot;L.helveticus&quot;))

 colorsss &lt;- c(&quot;red&quot;,&quot;orange&quot;,&quot;#007fa6ff&quot;,&quot;#0080a332&quot;,&quot;#fdc2b5ff&quot;,&quot;#ffc3b433&quot;,&quot;#b3b3b3ff&quot;)
 # colorsss &lt;- c(&quot;#fdc2b5ff&quot;,&quot;#&quot;)

  
plotpoints_both_colored &lt;- ggplot( pseudogeness_sumary, aes(x = meangenes, y = meanboth, xmin = meangenes-stdevgene, xmax = meangenes+stdevgene,ymin=meanboth-stdevboth,ymax=meanboth+stdevboth,fill=species_new_02,color=species_new_02,text=paste(&quot;counts=&quot;,counts,&quot;\nspecies=&quot;,species_new))) +
  geom_point(size=3) +
  geom_errorbarh(alpha=.2)+
  geom_errorbar(alpha=.2)+
    labs(x=&quot;Gene number&quot;,y=&quot;Pseudogenes+transposons [%]&quot;)+theme_classic()+theme(legend.title = element_blank())+
  geom_hline(yintercept =15, linetype=&quot;dotted&quot;)+geom_segment(aes(x = 1000, y = 1, xend = 1000, yend = 15), linetype = &quot;dotted&quot;)+
  scale_color_manual(values = colorsss)+scale_fill_manual(values = colorsss)+
  annotate(&quot;text&quot;, x = 800, y = 6, label = &quot;endosymbionts&quot;,col=&quot;red&quot;)+
  annotate(&quot;text&quot;, x = 800, y = 25, label = &quot;transition&quot;,col=&quot;orange&quot;)+
  annotate(&quot;text&quot;, x = 2000, y = 1, label = &quot;free living&quot;,col=&quot;grey77&quot;)

plotpoints_both_colored
#+lims(x=c(250,4100),y=c(-1.5,40))
# plotpoints_both_colored

  library(plotly)
ggp &lt;- ggplotly(plotpoints_both_colored)
 ggp

 

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenes_colored.pdf&quot;,width=6,height=4)

plotpoints_both_colored

dev.off()

## only size--------------------------

plotpoints &lt;-ggplot( pseudogeness_sumary, aes(x = meangenes, y = meanPseudo, xmin = meangenes-stdevgene, xmax = meangenes+stdevgene,ymin=meanPseudo-stdevPseudo,ymax=meanPseudo+stdevPseudo,fill=species_new_02,color=species_new_02,text=paste(&quot;counts=&quot;,counts,&quot;\nspecies=&quot;,species_new))) +
  geom_point(aes(size=meantransp)) +
  geom_errorbarh(alpha=.5)+
  geom_errorbar(alpha=.5)+
    labs(x=&quot;Gene number&quot;,y=&quot;Pseudogenes [%]&quot;)+theme_classic()+theme(legend.title = element_blank())+
  geom_hline(yintercept =10, linetype=&quot;dotted&quot;)+geom_segment(aes(x = 1000, y = -1, xend = 1000, yend = 10), linetype = &quot;dotted&quot;)+
  scale_color_manual(values = colorsss)+scale_fill_manual(values = colorsss)+
  annotate(&quot;text&quot;, x = 800, y = 6, label = &quot;endosymbionts&quot;,col=&quot;red&quot;)+
  annotate(&quot;text&quot;, x = 800, y = 25, label = &quot;transition&quot;,col=&quot;orange&quot;)+
  annotate(&quot;text&quot;, x = 2000, y = 1, label = &quot;free living&quot;,col=&quot;grey77&quot;)

#+lims(x=c(250,4100),y=c(-1.5,40))
plotpoints

  library(plotly)
ggp &lt;- ggplotly(plotpoints)
 ggp

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pseudogenes_new.pdf&quot;,width=6,height=4)

plotpoints

dev.off()
 
 ##------------------------
 ##eclipse
 #to do this we include all std points and mean points into one datafram
 ##------------------------
 
#  pseudogeness_sumary
#  
#   pseudogeness_sumary$groups &lt;- plyr::revalue(pseudogeness_sumary$species,c(&quot;Buch&quot;=&quot;endosmymbiont&quot;,&quot;Sglos&quot;=&quot;endosmymbiont&quot;,&quot;Ldel&quot;=&quot;starter cultures&quot;,&quot;Sterm&quot;=&quot;starter cultures&quot;,&quot;Wigg&quot;=&quot;endosmymbiont&quot;,&quot;Mleprae&quot;=&quot;transition&quot;,&quot;Rick&quot;=&quot;transition&quot;,&quot;Lplan&quot;=&quot;closely related free-living&quot;,&quot;Ssuis&quot;=&quot;closely related free-living&quot;))
# 
#  pseudogeness_esclipse &lt;-  rbind(data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes,y=pseudogeness_sumary$meanPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes-pseudogeness_sumary$stdevgene,y=pseudogeness_sumary$meanPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes+pseudogeness_sumary$stdevgene,y=pseudogeness_sumary$meanPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes,y=pseudogeness_sumary$meanPseudo-pseudogeness_sumary$stdevPseudo),
#         data.frame(species=pseudogeness_sumary$species,groups=pseudogeness_sumary$groups,x=pseudogeness_sumary$meangenes,y=pseudogeness_sumary$meanPseudo+pseudogeness_sumary$stdevPseudo))
#  
#  
#  
#  plotpoints_esclipse &lt;- ggplot(pseudogeness_esclipse,aes(x=x,y=y,fill=groups,color=groups),group=groups)+labs(x=&quot;Gene number&quot;,y=&quot;Pseudogenes [%]&quot;)+theme_classic()+theme(legend.title = element_blank())+stat_ellipse(type = &quot;t&quot;, linetype = 2)+theme(legend.position = &quot;none&quot;)+lims(x=c(250,4100),y=c(-1.5,40))
# plotpoints_esclipse
#  
# 
# library(patchwork)
# 
# plotpoints+plotpoints_wo+plotpoints_esclipse

###-------------------------------------only size and significanc  to closely relaed species
pseudogeness_sumary$species_new


pseudo_sub &lt;- pseudogeness_sumary %&gt;% filter(species_new %in% c(&quot;helveticus&quot;,&quot;S. thermophilus&quot;,&quot;L. delbrueckii subsp. lactis&quot;,&quot;vestibularis&quot;,&quot;crispatus&quot;,&quot;equicursoris&quot;))


# ggplot(pseudo_sub,aes(x=species,y=meangenes))

branch_age_plot &lt;- ggplot() +
  geom_pointrange(aes(x=species_new, y=meangenes, ymin = meangenes-stdevgene, ymax = meangenes+stdevgene,fill=species_new,color=species_new),data = pseudo_sub)+theme_classic()


# +facet_wrap(group~.,nrow = 3,scales = &quot;free_y&quot;, strip.position = &quot;right&quot;)+
#   coord_flip()+scale_fill_manual(values = colorsss)+scale_color_manual(values = colorsss)+
#   theme(axis.text.y = element_blank(),axis.title.y = element_blank(),axis.ticks.y = element_blank(),legend.position = &quot;none&quot;,strip.text = element_text(size = 7))+labs(y=&quot;Branch age&quot;)+scale_y_continuous(breaks=c(-10000,-7500,-5000,-2500,0,2020))

branch_age_plot

 # S.vestibularis, L. equicursoris and L. acidophilus

pseudogeness_count$species &lt;- plyr::revalue(pseudogeness_count$species,c(&quot;delbrueckii_lactis&quot;=&quot;L. delbrueckii subsp. lactis&quot;,
 &quot;Ldel&quot;=&quot;L. delbrueckii subsp. lactis&quot;,
 &quot;thermophilus&quot;=&quot;S. thermophilus&quot;,
 &quot;Sterm&quot;=&quot;S. thermophilus&quot;,
 &quot;Ssuis&quot;=&quot;suis&quot;,
 &quot;Lequi&quot;=&quot;equicursoris&quot;,
 &quot;Svest&quot;=&quot;vestibularis&quot;))

pseudo_sub_detail &lt;- pseudogeness_count %&gt;% filter(species %in% c(&quot;helveticus&quot;,&quot;S. thermophilus&quot;,&quot;L. delbrueckii subsp. lactis&quot;,&quot;vestibularis&quot;,&quot;crispatus&quot;,&quot;equicursoris&quot;))

pseudo_sub_detail$species_group &lt;- plyr::revalue(pseudo_sub_detail$species,c( &quot;L. delbrueckii subsp. lactis&quot;=&quot;Ldel&quot;,
 &quot;S. thermophilus&quot;=&quot;Sterm&quot;,
 &quot;crispatus&quot;=&quot;Lhelv&quot;,
 &quot;helveticus&quot;=&quot;Lhelv&quot;,
 &quot;equicursoris&quot;=&quot;Ldel&quot;,
 &quot;vestibularis&quot;=&quot;Sterm&quot;))
library(ggpubr)
library(ggsignif)
ggplot(pseudo_sub_detail,aes(x=species,y=genes_coding,fill=species))+geom_boxplot()+theme_classic()+facet_wrap(species_group~.,nrow = 1,scales = &quot;free&quot;)+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank())+ stat_compare_means(comparisons = list(data.frame(c(&quot;S. thermophilus&quot;, &quot;helveticus&quot;),c(&quot;vestibularis&quot;, &quot;S. crispatus&quot;))),method = &quot;t.test&quot;)+ stat_compare_means(comparisons = list(c(&quot;vestibularis&quot;, &quot;S. thermophilus&quot;)),method = &quot;t.test&quot;)

list(data.frame(c(&quot;S. thermophilus&quot;, &quot;helveticus&quot;),c(&quot;vestibularis&quot;, &quot;S. crispatus&quot;)))

# annotation table with adjusted pvals and y-position of the labels
anno_df &lt;- compare_means(genes_coding ~ species, group.by = &quot;species_group&quot;, data = pseudo_sub_detail) %&gt;%
 mutate(y_pos = 2800, p.adj = format.pval(p.adj, digits = 2))



plot_genome_size &lt;- ggplot()+geom_boxplot(data=pseudo_sub_detail,aes(x=species,y=genes_coding,fill=species))+theme_classic()+facet_wrap(species_group~.,nrow = 1,scales = &quot;free&quot;)+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank()) + 
  geom_signif(
    data=anno_df, 
    aes(xmin = group1, xmax = group2, annotations = p.adj, y_position = y_pos), 
    manual= TRUE
  )

plot_genome_size

   png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images//20210126/supplement//plot_genome_size.png&quot;, width = 2600, height = 1900,res=400)

# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement//plot_genome_size.pdf&quot;,width=6,height=5)

plot_genome_size

dev.off()


##---------pseudo signficance
anno_df &lt;- compare_means(pseudogenes ~ species, group.by = &quot;species_group&quot;, data = pseudo_sub_detail) %&gt;%
 mutate(y_pos = 400, p.adj = format.pval(p.adj, digits = 2))

plot_pseudo_sifn &lt;- ggplot()+geom_boxplot(data=pseudo_sub_detail,aes(x=species,y=pseudogenes,fill=species))+theme_classic()+facet_wrap(species_group~.,nrow = 1,scales = &quot;free&quot;)+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank()) + 
  geom_signif(
    data=anno_df, 
    aes(xmin = group1, xmax = group2, annotations = p.adj, y_position = y_pos), 
    manual= TRUE
  )

plot_pseudo_sifn

   png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images//20210126/supplement//plot_pseudo_sifn.png&quot;, width = 2600, height = 1900,res=400)

# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement//plot_genome_size.pdf&quot;,width=6,height=5)

plot_pseudo_sifn

dev.off()</code></pre>
</div>
</div>
</div>
<div id="phylogenies" class="section level1">
<h1>Phylogenies</h1>
<div id="core-genome-nucleotide-tree" class="section level2">
<h2>core genome nucleotide tree</h2>
<p>Here, I creat a core genome nucleotide tree (inspired by Kirsten). I
also include the complete Reference genomes from NCBI.</p>
<p>The following steps are incorporated: 1. orthofinder: find core
genome 2. Extract gene sequences corresponding to core gene famiilies
into multi-fasta files: 2 files per gene-family, corresponding to
nucleotide and protein sequences respectively. 3. Align sequences for
each family at protein level (resulting in one alignment file per core
gene family) (MAFFT) 4. Back-translate to nucleotide alignments
(respecting codon alignment) (Kirsten-script) 5. prune alignments
(removing columns represented by less than 50% of the sequences)
(Kirsten-script) 6. rename sequence-ids to genome-id only
(Kirsten-script) 7. Concatenate pruned alignments (Kirsten-script) 8.
infer the phylogeny (RAxML)</p>
<div id="orthofinder" class="section level3">
<h3>1.orthofinder</h3>
<pre class="bash"><code>###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Sterm

for species in $(echo &quot;Ldel Sterm&quot;)
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

#leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
done

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/Out-sali.faa



###-----------------------------------------
##orthorfinder
###-----------------------------------------
#for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
species=Ldel
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/
#cp -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/Ldel/OrthoFinder/Results_Feb11/ /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/

for species in $(echo &quot;Sterm Ldel&quot;)
do

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/

    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}// \
      -t 20 \
      -o /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ \
      -a 20

done # species

###-----------------------------------------
##orthorfinder
###-----------------------------------------
#for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
species=Ldel
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/
cp -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/Ldel/OrthoFinder/Results_Feb11/ /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/

for species in $(echo &quot;Sterm Ldel&quot;)
do

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/

    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA// \
      -t 37 \
      -o /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ \
      -a 37

done # species


##===============================================
##all genomes together 
##for GeneRax
##===============================================

###-------------------add Bacillus subtilis as outgroup

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/
species=$(echo &quot;Bacillus subtilis&quot;)

#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
#mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}
#cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/gbk/${shortnamess}


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;    \
    /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_BACSUB.txt
    #pecies_short=Sterm_references

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/

    wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/download_BACSUB.txt

mv GCF_000769515.1_ASM76951v1_protein.faa.gz Out_Bacsub.faa.gz

gunzip Out_Bacsub.faa.gz

###-------------------make collection of all rmk strains
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel/01_all_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/

###-------------------add all species close to Ldel and Sterm from figure 1D/E

cp  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/
cp /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/thermophilus.faa
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/delbrueckii_*


###-------------------add all strains close to Ldel and Sterm from figure 1F/G

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/PROKKA_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/
cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Ldel_references/PROKKA_FAA/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/

    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/FAA_allRMKs/ \
      -t 25 \
      -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/  \
      -a 25
</code></pre>
<pre class="bash"><code>##========================================
#get OG sequences for eggnog
##========================================
#rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm_OG_genes.faa
for species in $(echo &quot;Ldel&quot;)
do
rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}_OG_genes.faa

echo $species
#done
date=$(ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ |grep &quot;Results&quot;)

for genesss in $(ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroup_Sequences|grep &quot;.fa$&quot; |sed &#39;s/.fa//g&#39;)
do
echo $genesss

#perl -00 -ne &#39;/(&gt;[^&gt;]+)/ &amp;&amp; print $1&#39; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroup_Sequences/${genesss}.fa |sed &quot;s/&gt;.*$/&gt;S_${genesss}/g&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm_OG_genes.faa

awk &#39;/^&gt;/{if(N)exit;++N;} {print;}&#39; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroup_Sequences/${genesss}.fa |sed &quot;s/&gt;.*$/&gt;L_${genesss}/g&quot; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}_OG_genes.faa


done #genesss
ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroup_Sequences/ |grep &quot;.fa$&quot; -c
grep &quot;&gt;&quot; -c  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}_OG_genes.faa

done #species

grep &quot;&gt;&quot;  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm_OG_genes.faa |head 
grep &quot;&gt;&quot;  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Ldel_OG_genes.faa |head 

##========================================
#bring local for eggnog
##========================================

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/forEGGNOG_pangenome
scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Ldel_OG_genes.faa /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/forEGGNOG_pangenome

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm_OG_genes.faa /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/forEGGNOG_pangenome/

###----------------------------------------
#extract only pseudogne OG
###----------------------------------------


#------------------------Lhelv
mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Lhelv

cut -d &#39;,&#39; -f 1 ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_genomes_ps.clustered.txt |sed &#39;1d&#39;  |sed &#39;s/^O/Lh_O/g&#39; &gt; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Lhelv/OG_ps.nmaes


cd ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder
xargs --arg-file=Lhelv/OG_ps.nmaes  samtools faidx ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_OG_genes.faa &gt; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Lhelv/OG_ps.faa


grep -c &quot;&gt;&quot; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_OG_genes.faa
grep -c &quot;&gt;&quot; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Lhelv/OG_ps.faa


#------------------------Sterm
mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Sterm

cut -d &#39;,&#39; -f 1 ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_genomes_ps.clustered.txt |sed &#39;1d&#39;  |sed &#39;s/^O/St_O/g&#39; &gt; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Sterm/OG_ps.nmaes


cd ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder
xargs --arg-file=Sterm/OG_ps.nmaes  samtools faidx ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_OG_genes.faa &gt; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Sterm/OG_ps.faa


grep -c &quot;&gt;&quot; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Sterm_OG_genes.faa
grep -c &quot;&gt;&quot; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Sterm/OG_ps.faa



#------------------------Ldel
mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Ldel

cut -d &#39;,&#39; -f 1 ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_genomes_ps.clustered.txt |sed &#39;1d&#39;  |sed &#39;s/^O/Ld_O/g&#39; &gt; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Ldel/OG_ps.nmaes


cd ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder
xargs --arg-file=Ldel/OG_ps.nmaes  samtools faidx ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_OG_genes.faa &gt; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Ldel/OG_ps.faa


grep -c &quot;&gt;&quot; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_OG_genes.faa
grep -c &quot;&gt;&quot; ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Gapminder/Ldel/OG_ps.faa
</code></pre>
</div>
<div id="extract-gene-sequences" class="section level3">
<h3>2.Extract gene sequences</h3>
<pre class="bash"><code>date=Results_Feb01
species=Sterm
#/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/Results_Jul28/
for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
do
echo &quot;===========================================&quot;
echo -e ${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/get_singlecp_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt &gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt

wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt
wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt

done</code></pre>
<p>Here, I make a directory containing the fna and faa files of all
genomes with the file name corresponding to the locus tag.</p>
<pre class="bash"><code>

for species in $(echo &quot;Ldel Sterm&quot;)
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA/
done

###--------------------
for species in $(echo &quot;Ldel Sterm&quot;)
do
rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/ |grep &quot;.faa$&quot;|sed &#39;s/.faa//g&#39;|grep -v &quot;Out-sali&quot;)
do
locusTagsss=$(head -1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/${genomesss}.faa|cut -d &#39;_&#39; -f 1|sed &#39;s/&gt;//g&#39;)
echo -e &quot;This  genome : &quot; ${genomesss} &quot; has the following locus Tag : &quot;${locusTagsss}

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/${genomesss}.faa &gt; \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/${genomesss}.ffn &gt; \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.ffn


done
done

##------=================================================================
#extract orthogroups
##------=================================================================

for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/extract_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt --folder ${species}

done</code></pre>
</div>
<div id="align" class="section level3">
<h3>3. Align</h3>
<p>align protein families</p>
<pre class="bash"><code>

for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
do

cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep &quot;.faa$&quot; |sed &#39;s/.faa//g&#39;|grep -v &quot;Out-sali&quot; )
do
rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.aln.fasta
mafft --thread 37 --maxiterate 1000  --auto /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.faa &gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}_aln.fasta

done 

done #species</code></pre>
</div>
<div id="back-translate" class="section level3">
<h3>4. Back-translate</h3>
<p>here I back translate the faa alignment to dna.</p>
<pre class="bash"><code>
for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/aln_aa_to_dna.pl

done
</code></pre>
</div>
<div id="prune-alignments" class="section level3">
<h3>5. prune alignments</h3>
<p>With perl: prune alignments (removing columns represented by less
than 50% of the sequences)</p>
<pre class="bash"><code>for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/prune_aln.pl

done</code></pre>
</div>
<div id="rename-sequence-ids" class="section level3">
<h3>6. rename sequence-ids</h3>
<pre class="bash"><code>##================================================================
#remove all infomration (except genome info) from multifasta header
##================================================================

for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep &quot;_aln_prune.fasta$&quot; |sed &#39;s/_aln_prune.fasta//g&#39; )
do
sed -i &quot;s/_0.*//&quot; ${genesss}_aln_prune.fasta 

done
done

##================================================================
#shorten header
##================================================================
species=Ldel
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep &quot;_aln_prune.fasta$&quot; |sed &#39;s/_aln_prune.fasta//g&#39; )
do
sed -i &#39;s/L-delbrueckii-//g&#39; ${genesss}_aln_prune.fasta 

done



species=Sterm
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep &quot;_aln_prune.fasta$&quot; |sed &#39;s/_aln_prune.fasta//g&#39; )
do
sed -i &#39;s/-CIRM-//g&#39; ${genesss}_aln_prune.fasta 

done

species=Sterm
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep &quot;_aln_prune.fasta$&quot; |sed &#39;s/_aln_prune.fasta//g&#39; )
do
sed -i &#39;s/-BM-A0/-BM-A/g&#39; ${genesss}_aln_prune.fasta 

done
</code></pre>
</div>
<div id="concatenate-pruned" class="section level3">
<h3>7. Concatenate pruned</h3>
<p>With perl: Concatenate pruned alignments</p>
<pre class="bash"><code>  
#  for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
  for species in $(echo &quot;Sterm&quot;|tail -1)
  do
  echo -e ${species}
  cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
  perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/cat_align.pl &gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}.phylip
  
  done</code></pre>
</div>
<div id="infer-the-phylogeny" class="section level3">
<h3>8. infer the phylogeny</h3>
<p>With RAxML: infer the phylogeny</p>
<pre class="bash"><code>mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/

#  for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
   for species in $(echo &quot;Sterm&quot;|tail -1)
  do
  echo -e ${species}
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/

rm -r *${species}_all*

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}.phylip -n ${species}_all -T 37
done</code></pre>
<p>move local</p>
<pre class="bash"><code>
mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new/


scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new//</code></pre>
</div>
<div id="plot-rees" class="section level3">
<h3>Plot rees</h3>
<div id="sterm" class="section level4">
<h4>Sterm</h4>
<pre class="r"><code>##==================================================================
##==Sterm
##==================================================================




Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated&quot;, tree.names = NULL, force.multi = FALSE)


# x &lt;- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
p &lt;- ggtree(Sterm_tree_raw) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  
p 
# Sterm_tree_raw$tip.label
##--------------------------------
##all info at once (geography and origin)
##--------------------------------
# all_sterm_strain_info &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited.csv&quot;, &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE) %&gt;% select(c(shortName, strain, origin, location_continent))
# 
# all_sterm_strain_info$strain &lt;- gsub(&quot;S-thermophilus-&quot;,&quot;&quot;,all_sterm_strain_info$strain)
# 
# all_sterm_strain_info$shortName &lt;- as.character(all_sterm_strain_info$shortName)
# 
# all_sterm_strain_info$shortName %in% Sterm_tree_raw$tip.label
# Sterm_tree_raw$tip.label %in% all_sterm_strain_info$shortName 
# 
# all_sterm_strain_info_ready &lt;- all_sterm_strain_info %&gt;% select(-strain) %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;shortName&quot;)%&gt;%replace(is.na(.), &quot;Unknown&quot;) 
# table(all_sterm_strain_info_ready$location_continent)
#  all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c(&quot;Europe&quot;,  &quot;Asia&quot;, &quot;USA&quot;, &quot;Australia&quot; , &quot;Unknown&quot;,&quot;Outgroup&quot;))
# 
#  
#  colnames(all_sterm_strain_info_ready) &lt;- c(&quot;source&quot;,&quot;geography&quot;)
# rownames(all_sterm_strain_info_ready)
# 
# p3 &lt;- gheatmap(p , all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
# p3
#  
##-----------------------------------------------
###import metaData
##-----------------------------------------------

meta_sterm &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited_extended_02.csv&quot;)
# colnames(meta_sterm)
meta_sterm$shortName_new &lt;- meta_sterm$shortName %&gt;% gsub(&quot;^&quot;,&quot;S&quot;,.) %&gt;% gsub(&quot;^SS&quot;,&quot;S&quot;,.)
# names_strains$strain


##-----------------------------------------------
###add source
##-----------------------------------------------


meta_sterm_ready &lt;- meta_sterm %&gt;% select(shortName,origin) %&gt;% unique() %&gt;% filter(shortName %in%Sterm_tree_raw$tip.label) %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;shortName&quot;) #%&gt;%replace(is.na(.), &quot;Unknown&quot;)

table(meta_sterm_ready$origin)
meta_sterm_ready$origin &lt;- droplevels(as.factor(meta_sterm_ready$origin))

meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c(&quot;Cheese&quot;,&quot;milk&quot;,&quot;yogurt&quot;,&quot;cow&quot;,&quot;human&quot;,&quot;host&quot;,&quot;Unknown&quot;))
meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c(&quot;Cheese&quot;,&quot;milk&quot;,&quot;yogurt&quot;,&quot;host&quot;))

colorsss &lt;- c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;#ffafcc&quot;,&quot;white&quot;)

meta_sterm_ready_origin &lt;- meta_sterm_ready %&gt;% rownames_to_column() %&gt;% as.tibble()

p2 &lt;- gheatmap(p , meta_sterm_ready, offset=0.04, width=.05) + scale_fill_manual(values=colorsss)#+  scale_fill_manual(values=colors_area)
p2

##-----------------------------------------------
###add geography
##-----------------------------------------------


meta_sterm_ready &lt;- meta_sterm %&gt;% select(shortName,location_continent) %&gt;% unique()%&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;shortName&quot;)

table(meta_sterm_ready$location_continent)
# meta_sterm_ready$origin &lt;- droplevels(as.factor(meta_sterm_ready$origin))

# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c(&quot;Cheese&quot;,&quot;milk&quot;,&quot;yogurt&quot;,&quot;cow&quot;,&quot;human&quot;,&quot;host&quot;,&quot;Unknown&quot;))
# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c(&quot;Cheese&quot;,&quot;milk&quot;,&quot;yogurt&quot;,&quot;host&quot;))

colorsss &lt;- c(&quot;#2a9d8f&quot;,&quot;#e9c46a&quot;,&quot;#f4a261&quot;,&quot;#e76f51&quot;,&quot;white&quot;)
p2 &lt;- p2 + new_scale_fill()

p3 &lt;- gheatmap(p2 , meta_sterm_ready, offset=0.08, width=.05 ) + scale_fill_manual(values=colorsss)#+  scale_fill_manual(values=colors_area)
p3

##-----------------------------------------------
###add time
##-----------------------------------------------

meta_sterm_ready &lt;- meta_sterm %&gt;% select(shortName,location_continent,origin,extraction_time) %&gt;% unique()%&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;shortName&quot;) %&gt;% select(-location_continent,-origin) %&gt;% mutate(extraction_time=as.numeric(extraction_time))

p3 &lt;- p3 + new_scale_fill()
# p3 &lt;- gheatmap(p2, df, offset=i*offset, width=.08,colnames_angle=0, colnames_offset_y = .05)  

p4 &lt;- gheatmap(p3 , meta_sterm_ready, offset=0.12, width=.05) #+ scale_color_brewer(type=&quot;div&quot;)

p4

###----------------------------------------------
# add ancestral niche reconstruction
###----------------------------------------------

library(ape)
# ape::ace


###------------test

### For discrete characters:
data(bird.orders)
x &lt;- factor(c(rep(0, 5), rep(1, 18)))
ans &lt;- ace(x, bird.orders, type = &quot;d&quot;)
#### Showing the likelihoods on each node:
plot(bird.orders, type = &quot;c&quot;, FALSE, label.offset = 1)
co &lt;- c(&quot;blue&quot;, &quot;yellow&quot;)
tiplabels(pch = 22, bg = co[as.numeric(x)], cex = 2, adj = 1)
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.75)


###------------real data

meta_sterm_ready_origin
colorsss &lt;- c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;#ffafcc&quot;,&quot;white&quot;)

Sterm_tree_raw %&gt;% tip.label()
Sterm_tree_raw$tip.label

# is_tip &lt;- Sterm_tree_raw$edge[,2] &lt;= length(Sterm_tree_raw$tip.label)


# x &lt;- get_taxa_name(p2) %&gt;% as.data.frame() %&gt;% rename(&quot;rowname&quot;=&quot;.&quot;) %&gt;% as.tibble() %&gt;% left_join(.,meta_sterm_ready_origin) %&gt;% pull(origin)
bird.orders &lt;- drop.tip(Sterm_tree_raw,&quot;Out-vest&quot;)

is_tip &lt;- bird.orders$edge[,2] &lt;= length(bird.orders$tip.label)
ordered_tips &lt;- bird.orders$edge[is_tip, 2]

x &lt;- bird.orders$tip.label[ordered_tips] %&gt;% as.data.frame() %&gt;% rename(&quot;rowname&quot;=&quot;.&quot;) %&gt;% as.tibble() %&gt;% left_join(.,meta_sterm_ready_origin) %&gt;% pull(origin)  %&gt;% as.character()

table(x)

ans &lt;- ace(x, bird.orders, type = &quot;d&quot;,method = &quot;ML&quot;)
#### Showing the likelihoods on each node:
plot(bird.orders, type = &quot;c&quot;, FALSE, label.offset = 1, cex = 0.5)
# co &lt;- c(&quot;blue&quot;, &quot;yellow&quot;)
co &lt;- c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;,&quot;#ffafcc&quot;,&quot;white&quot;)
# co[as.numeric(x)]


legend(&quot;bottomleft&quot;, legend=levels(as.factor(x)),
       col=c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;), fill=c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8

# legend(&quot;bottomleft&quot;, inset=.02, title=&quot;Number of Cylinders&quot;,
   # c(&quot;4&quot;,&quot;6&quot;,&quot;8&quot;), fill=topo.colors(3), horiz=TRUE, cex=0.8)
tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )

# p2

ans$lik.anc
ans$rates
ans$se
ans$index.matrix


###---------------


 pdf(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/phylog_supp_ACE.pdf&quot;,width=12,height=9)

 
plot(bird.orders, type = &quot;c&quot;, FALSE, label.offset = 1, cex = 0.5)

legend(&quot;bottomleft&quot;, legend=levels(as.factor(x)),
       col=c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;), fill=c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8

tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )


dev.off()

####---------dairy and host-associated


meta_sterm_ready_origin
colorsss &lt;- c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;#ffafcc&quot;,&quot;white&quot;)
table(meta_sterm_ready_origin$origin)
meta_sterm_ready_origin$origin &lt;- plyr::revalue(meta_sterm_ready_origin$origin, c(&quot;yogurt&quot;=&quot;Dairy&quot;,&quot;Cheese&quot;=&quot;Dairy&quot;,&quot;milk&quot;=&quot;Dairy&quot;))



Sterm_tree_raw %&gt;% tip.label()
Sterm_tree_raw$tip.label

# is_tip &lt;- Sterm_tree_raw$edge[,2] &lt;= length(Sterm_tree_raw$tip.label)


# x &lt;- get_taxa_name(p2) %&gt;% as.data.frame() %&gt;% rename(&quot;rowname&quot;=&quot;.&quot;) %&gt;% as.tibble() %&gt;% left_join(.,meta_sterm_ready_origin) %&gt;% pull(origin)
bird.orders &lt;- drop.tip(Sterm_tree_raw,&quot;Out-vest&quot;)

is_tip &lt;- bird.orders$edge[,2] &lt;= length(bird.orders$tip.label)
ordered_tips &lt;- bird.orders$edge[is_tip, 2]

x &lt;- bird.orders$tip.label[ordered_tips] %&gt;% as.data.frame() %&gt;% rename(&quot;rowname&quot;=&quot;.&quot;) %&gt;% as.tibble() %&gt;% left_join(.,meta_sterm_ready_origin) %&gt;% pull(origin)  %&gt;% as.character()

table(x)

ans &lt;- ace(x, bird.orders, type = &quot;d&quot;,method = &quot;ML&quot;)
#### Showing the likelihoods on each node:
plot(bird.orders, type = &quot;c&quot;, FALSE, label.offset = 1, cex = 0.5)
# co &lt;- c(&quot;blue&quot;, &quot;yellow&quot;)
co &lt;- c(&quot;blue&quot;,&quot;red&quot;,&quot;white&quot;)
# co[as.numeric(x)]


legend(&quot;bottomleft&quot;, legend=levels(as.factor(x)),
       col=c(&quot;blue&quot;,&quot;red&quot;,&quot;white&quot;), fill=c(&quot;blue&quot;,&quot;red&quot;,&quot;white&quot;), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8

# legend(&quot;bottomleft&quot;, inset=.02, title=&quot;Number of Cylinders&quot;,
   # c(&quot;4&quot;,&quot;6&quot;,&quot;8&quot;), fill=topo.colors(3), horiz=TRUE, cex=0.8)
tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )


 pdf(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/phylog_supp_ACE_dariy.pdf&quot;,width=12,height=9)

 plot(bird.orders, type = &quot;c&quot;, FALSE, label.offset = 1, cex = 0.5)
# co &lt;- c(&quot;blue&quot;, &quot;yellow&quot;)
co &lt;- c(&quot;blue&quot;,&quot;red&quot;,&quot;white&quot;)
# co[as.numeric(x)]


legend(&quot;bottomleft&quot;, legend=levels(as.factor(x)),
       col=c(&quot;blue&quot;,&quot;red&quot;,&quot;white&quot;), fill=c(&quot;blue&quot;,&quot;red&quot;,&quot;white&quot;), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8

# legend(&quot;bottomleft&quot;, inset=.02, title=&quot;Number of Cylinders&quot;,
   # c(&quot;4&quot;,&quot;6&quot;,&quot;8&quot;), fill=topo.colors(3), horiz=TRUE, cex=0.8)
tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )


dev.off()</code></pre>
</div>
<div id="ldel" class="section level4">
<h4>Ldel</h4>
<pre class="r"><code>library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(RColorBrewer)
library(ggnewscale)

##==================================================================
##==Ldel
##==================================================================


Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_Ldel/SpeciesTree_rooted_node_labels_orthofinder_ldel_complete_rerooted_cleaned.txt&quot;, tree.names = NULL, force.multi = FALSE)

# Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Ldel_all&quot;, tree.names = NULL, force.multi = FALSE)


# x &lt;- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
p &lt;- ggtree(Sterm_tree_raw) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  
p 

##-----------------------------------------------
###import metaData
##-----------------------------------------------
# meta_sterm &lt;-  read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Ldel_phylogeny_info.tsv&quot;, 
    # delim = &quot;\t&quot;, escape_double = FALSE, 
    # trim_ws = TRUE)

meta_ldel &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_ldel_edited_02.csv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE)

colnames(meta_ldel)
meta_ldel$shortName_new &lt;- meta_ldel$shortName %&gt;% gsub(&quot;-DSM-&quot;,&quot;-&quot;,.) %&gt;% gsub(&quot;-KCTC-&quot;,&quot;-&quot;,.)



# names_strains$strain


# meta_sterm &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_ldel_edited.csv&quot;)
# # colnames(meta_sterm)
# meta_sterm$shortName_new &lt;- meta_sterm$shortName %&gt;% gsub(&quot;^&quot;,&quot;S&quot;,.) %&gt;% gsub(&quot;^SS&quot;,&quot;S&quot;,.)
# # names_strains$strain


##-----------------------------------------------
###add source
##-----------------------------------------------


meta_sterm_ready &lt;- meta_ldel %&gt;% select(shortName,origin) %&gt;% unique() %&gt;% filter(shortName %in%Sterm_tree_raw$tip.label) %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;shortName&quot;) #%&gt;%replace(is.na(.), &quot;Unknown&quot;)

table(meta_sterm_ready$origin)
meta_sterm_ready$origin &lt;- droplevels(as.factor(meta_sterm_ready$origin))

# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c(&quot;Cheese&quot;,&quot;milk&quot;,&quot;yogurt&quot;,&quot;cow&quot;,&quot;human&quot;,&quot;host&quot;,&quot;Unknown&quot;))
meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c(&quot;Cheese&quot;,&quot;milk&quot;,&quot;yogurt&quot;,&quot;host&quot;))

colorsss &lt;- c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;#ffafcc&quot;,&quot;white&quot;)


meta_sterm_ready_origin &lt;- meta_sterm_ready %&gt;% rownames_to_column() %&gt;% as.tibble()


p2 &lt;- gheatmap(p , meta_sterm_ready, offset=0.04, width=.05) + scale_fill_manual(values=colorsss)#+  scale_fill_manual(values=colors_area)
p2

##-----------------------------------------------
###add geography
##-----------------------------------------------


meta_sterm_ready &lt;- meta_ldel %&gt;% select(shortName,location_continent) %&gt;% unique()%&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;shortName&quot;)

table(meta_sterm_ready$location_continent)
# meta_sterm_ready$origin &lt;- droplevels(as.factor(meta_sterm_ready$origin))

# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c(&quot;Cheese&quot;,&quot;milk&quot;,&quot;yogurt&quot;,&quot;cow&quot;,&quot;human&quot;,&quot;host&quot;,&quot;Unknown&quot;))
# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c(&quot;Cheese&quot;,&quot;milk&quot;,&quot;yogurt&quot;,&quot;host&quot;))

colorsss &lt;- c(&quot;#2a9d8f&quot;,&quot;#e9c46a&quot;,&quot;#f4a261&quot;,&quot;#e76f51&quot;,&quot;white&quot;)
p2 &lt;- p2 + new_scale_fill()

p3 &lt;- gheatmap(p2 , meta_sterm_ready, offset=0.08, width=.05 ) + scale_fill_manual(values=colorsss)#+  scale_fill_manual(values=colors_area)
p3

##-----------------------------------------------
###add time
##-----------------------------------------------

meta_sterm_ready &lt;- meta_ldel %&gt;% select(shortName,location_continent,origin,extraction_time) %&gt;% unique()%&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;shortName&quot;) %&gt;% select(-location_continent,-origin) %&gt;% mutate(extraction_time=as.numeric(extraction_time))

p3 &lt;- p3 + new_scale_fill()
# p3 &lt;- gheatmap(p2, df, offset=i*offset, width=.08,colnames_angle=0, colnames_offset_y = .05)  

p4_ldel &lt;- gheatmap(p3 , meta_sterm_ready, offset=0.12, width=.05) #+ scale_color_brewer(type=&quot;div&quot;)

p4_ldel

###----------------------------------------------
# add ancestral niche reconstruction
###----------------------------------------------

library(ape)
# ape::ace


###------------test

### For discrete characters:
data(bird.orders)
x &lt;- factor(c(rep(0, 5), rep(1, 18)))
ans &lt;- ace(x, bird.orders, type = &quot;d&quot;)
#### Showing the likelihoods on each node:
plot(bird.orders, type = &quot;c&quot;, FALSE, label.offset = 1)
co &lt;- c(&quot;blue&quot;, &quot;yellow&quot;)
tiplabels(pch = 22, bg = co[as.numeric(x)], cex = 2, adj = 1)
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.75)


###------------real data

meta_sterm_ready_origin
colorsss &lt;- c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;#ffafcc&quot;,&quot;white&quot;)

Sterm_tree_raw %&gt;% tip.label()
Sterm_tree_raw$tip.label

# is_tip &lt;- Sterm_tree_raw$edge[,2] &lt;= length(Sterm_tree_raw$tip.label)


# x &lt;- get_taxa_name(p2) %&gt;% as.data.frame() %&gt;% rename(&quot;rowname&quot;=&quot;.&quot;) %&gt;% as.tibble() %&gt;% left_join(.,meta_sterm_ready_origin) %&gt;% pull(origin)
bird.orders &lt;- drop.tip(Sterm_tree_raw,&quot;Out-helv&quot;)

is_tip &lt;- bird.orders$edge[,2] &lt;= length(bird.orders$tip.label)
ordered_tips &lt;- bird.orders$edge[is_tip, 2]

x &lt;- bird.orders$tip.label[ordered_tips] %&gt;% as.data.frame() %&gt;% rename(&quot;rowname&quot;=&quot;.&quot;) %&gt;% as.tibble() %&gt;% left_join(.,meta_sterm_ready_origin) %&gt;% pull(origin)  %&gt;% as.character()

table(x)

ans &lt;- ace(x, bird.orders, type = &quot;d&quot;,method = &quot;ML&quot;)
#### Showing the likelihoods on each node:
plot(bird.orders, type = &quot;c&quot;, FALSE, label.offset = 1, cex = 0.5)
# co &lt;- c(&quot;blue&quot;, &quot;yellow&quot;)
co &lt;- c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;,&quot;#ffafcc&quot;,&quot;white&quot;)
# co[as.numeric(x)]


legend(&quot;bottomleft&quot;, legend=levels(as.factor(x)),
       col=c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;), fill=c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8

# legend(&quot;bottomleft&quot;, inset=.02, title=&quot;Number of Cylinders&quot;,
   # c(&quot;4&quot;,&quot;6&quot;,&quot;8&quot;), fill=topo.colors(3), horiz=TRUE, cex=0.8)
tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )

p2

ans$lik.anc
ans$rates
ans$se
ans$index.matrix


###---------------

# 
#  pdf(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/phylog_supp_ACE.pdf&quot;,width=12,height=9)
# 
#  
# plot(bird.orders, type = &quot;c&quot;, FALSE, label.offset = 1, cex = 0.5)
# 
# legend(&quot;bottomleft&quot;, legend=levels(as.factor(x)),
#        col=c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;), fill=c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8
# 
# tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
# nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )
# 
# 
# dev.off()</code></pre>
</div>
<div id="lhelv" class="section level4">
<h4>Lhelv</h4>
<pre class="r"><code>library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(RColorBrewer)
library(ggnewscale)

##==================================================================
##==Lhelv
##==================================================================


Sterm_tree_raw &lt;- read.nexus(&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/lhelv/new&quot;, tree.names = NULL, force.multi = FALSE)


# Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Ldel_all&quot;, tree.names = NULL, force.multi = FALSE)


# x &lt;- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
p &lt;- ggtree(Sterm_tree_raw) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  
p 

##-----------------------------------------------
###import metaData
##-----------------------------------------------
# meta_sterm &lt;-  read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Ldel_phylogeny_info.tsv&quot;, 
    # delim = &quot;\t&quot;, escape_double = FALSE, 
    # trim_ws = TRUE)

# meta_ldel &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_03.csv&quot;, 
#     delim = &quot;\t&quot;, escape_double = FALSE, 
#     trim_ws = TRUE)
meta_ldel &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_03.csv&quot;) %&gt;% mutate(shortName=strain_name)

meta_ldel$shortName &lt;- gsub(&quot;^&quot;,&quot;Lh-&quot;,meta_ldel$strain_name) %&gt;% gsub(&quot;^Lh-Lh&quot;,&quot;&quot;,.)


colnames(meta_ldel)
meta_ldel$shortName_new &lt;- meta_ldel$shortName %&gt;% gsub(&quot;-DSM-&quot;,&quot;-&quot;,.) %&gt;% gsub(&quot;-KCTC-&quot;,&quot;-&quot;,.)



##-----------------------------------------------
###add source
##-----------------------------------------------


meta_sterm_ready &lt;- meta_ldel %&gt;% select(shortName,origin) %&gt;% unique() %&gt;% filter(shortName %in%Sterm_tree_raw$tip.label) %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;shortName&quot;) #%&gt;%replace(is.na(.), &quot;Unknown&quot;)

table(meta_sterm_ready$origin)
meta_sterm_ready$origin &lt;- droplevels(as.factor(meta_sterm_ready$origin))

# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c(&quot;Cheese&quot;,&quot;milk&quot;,&quot;yogurt&quot;,&quot;cow&quot;,&quot;human&quot;,&quot;host&quot;,&quot;Unknown&quot;))
meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c(&quot;Cheese&quot;,&quot;milk&quot;,&quot;yogurt&quot;,&quot;host&quot;))

colorsss &lt;- c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;#ffafcc&quot;,&quot;white&quot;)

meta_sterm_ready_origin &lt;- meta_sterm_ready %&gt;% rownames_to_column() %&gt;% as.tibble()

p2 &lt;- gheatmap(p , meta_sterm_ready, offset=0.04, width=.05) + scale_fill_manual(values=colorsss)#+  scale_fill_manual(values=colors_area)
p2

##-----------------------------------------------
###add geography
##-----------------------------------------------


meta_sterm_ready &lt;- meta_ldel %&gt;% select(shortName,location_continent) %&gt;% unique()%&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;shortName&quot;)

table(meta_sterm_ready$location_continent)
# meta_sterm_ready$origin &lt;- droplevels(as.factor(meta_sterm_ready$origin))

# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c(&quot;Cheese&quot;,&quot;milk&quot;,&quot;yogurt&quot;,&quot;cow&quot;,&quot;human&quot;,&quot;host&quot;,&quot;Unknown&quot;))
# meta_sterm_ready$origin = factor(meta_sterm_ready$origin, levels=c(&quot;Cheese&quot;,&quot;milk&quot;,&quot;yogurt&quot;,&quot;host&quot;))

colorsss &lt;- c(&quot;#2a9d8f&quot;,&quot;#e9c46a&quot;,&quot;#f4a261&quot;,&quot;#e76f51&quot;,&quot;white&quot;)





p2 &lt;- p2 + new_scale_fill()

p3 &lt;- gheatmap(p2 , meta_sterm_ready, offset=0.08, width=.05 ) + scale_fill_manual(values=colorsss)#+  scale_fill_manual(values=colors_area)
p3

##-----------------------------------------------
###add time
##-----------------------------------------------

meta_sterm_ready &lt;- meta_ldel %&gt;% select(shortName,location_continent,origin,extraction_time) %&gt;% unique()%&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;shortName&quot;) %&gt;% select(-location_continent,-origin) %&gt;% mutate(extraction_time=as.numeric(extraction_time))

p3 &lt;- p3 + new_scale_fill()
# p3 &lt;- gheatmap(p2, df, offset=i*offset, width=.08,colnames_angle=0, colnames_offset_y = .05)  

p4_lhelv &lt;- gheatmap(p3 , meta_sterm_ready, offset=0.12, width=.05) #+ scale_color_brewer(type=&quot;div&quot;)

p4_lhelv

###----------------------------------------------
# add ancestral niche reconstruction
###----------------------------------------------

library(ape)
# ape::ace


###------------test

### For discrete characters:
data(bird.orders)
x &lt;- factor(c(rep(0, 5), rep(1, 18)))
ans &lt;- ace(x, bird.orders, type = &quot;d&quot;)
#### Showing the likelihoods on each node:
plot(bird.orders, type = &quot;c&quot;, FALSE, label.offset = 1)
co &lt;- c(&quot;blue&quot;, &quot;yellow&quot;)
tiplabels(pch = 22, bg = co[as.numeric(x)], cex = 2, adj = 1)
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.75)


###------------real data

meta_sterm_ready_origin
colorsss &lt;- c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;#ffafcc&quot;,&quot;white&quot;)

Sterm_tree_raw %&gt;% tip.label()
Sterm_tree_raw$tip.label

# is_tip &lt;- Sterm_tree_raw$edge[,2] &lt;= length(Sterm_tree_raw$tip.label)


# x &lt;- get_taxa_name(p2) %&gt;% as.data.frame() %&gt;% rename(&quot;rowname&quot;=&quot;.&quot;) %&gt;% as.tibble() %&gt;% left_join(.,meta_sterm_ready_origin) %&gt;% pull(origin)
# bird.orders &lt;- drop.tip(Sterm_tree_raw,&quot;Out-helv&quot;)
bird.orders &lt;- Sterm_tree_raw

is_tip &lt;- bird.orders$edge[,2] &lt;= length(bird.orders$tip.label)
ordered_tips &lt;- bird.orders$edge[is_tip, 2]

x &lt;- bird.orders$tip.label[ordered_tips] %&gt;% as.data.frame() %&gt;% rename(&quot;rowname&quot;=&quot;.&quot;) %&gt;% as.tibble() %&gt;% left_join(.,meta_sterm_ready_origin) %&gt;% pull(origin)  %&gt;% as.character()

table(x)

ans &lt;- ace(x, bird.orders, type = &quot;d&quot;,method = &quot;ML&quot;)
#### Showing the likelihoods on each node:
plot(bird.orders, type = &quot;c&quot;, FALSE, label.offset = 1, cex = 0.5)
# co &lt;- c(&quot;blue&quot;, &quot;yellow&quot;)
co &lt;- c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;,&quot;#ffafcc&quot;,&quot;white&quot;)
# co[as.numeric(x)]


legend(&quot;bottomleft&quot;, legend=levels(as.factor(x)),
       col=c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;), fill=c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8

# legend(&quot;bottomleft&quot;, inset=.02, title=&quot;Number of Cylinders&quot;,
   # c(&quot;4&quot;,&quot;6&quot;,&quot;8&quot;), fill=topo.colors(3), horiz=TRUE, cex=0.8)
tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )

p2

ans$lik.anc
ans$rates
ans$se
ans$index.matrix


###---------------

# 
#  pdf(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/phylog_supp_ACE.pdf&quot;,width=12,height=9)
# 
#  
# plot(bird.orders, type = &quot;c&quot;, FALSE, label.offset = 1, cex = 0.5)
# 
# legend(&quot;bottomleft&quot;, legend=levels(as.factor(x)),
#        col=c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;), fill=c(&quot;#cdb4db&quot;,&quot;#e63946&quot;,&quot;#ffc8dd&quot;,&quot;grey&quot;), horiz=TRUE, cex=0.8) #lty=1:2, cex=0.8
# 
# tiplabels(pch = 22, bg = co[as.numeric(as.factor(x))], cex = 1, adj = 1) #
# nodelabels(thermo = ans$lik.anc, piecol = co, cex = 0.4,text = )
# 
# 
# dev.off()</code></pre>
<pre class="r"><code>###----------------------------merge

library(patchwork)

p4+p4_ldel+p4_lhelv


 # png(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/phylog_supp.png&quot;, width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
 pdf(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/phylog_supp.pdf&quot;,width=12,height=9)

 
p4+p4_ldel+p4_lhelv

dev.off()</code></pre>
<p>##Gene/Species tree</p>
</div>
<div id="xavier-didelots-approach" class="section level4">
<h4>Xavier Didelots approach</h4>
<p>example from <a href="https://xavierdidelot.github.io/BactDating/articles/Staph.html#initialisation-1">staph
aureus</a>.</p>
<pre class="r"><code>##-----------------------------
#test
##-----------------------------

library(BactDating)
library(ape)
set.seed(0)
data(staph)
plot(staph$tree, show.tip.label = F)
axisPhylo(backward = F)


### dates in the same order as tip.label  
staph$tree$tip.label
staph$dates
staph$tree


res=roottotip(staph$tree,staph$dates)

##----------run Dactdate
res=bactdate(unroot(staph$tree),staph$dates,nbIts=1000)
plot(res,&#39;treeCI&#39;,show.tip.label = F)

plot(res,&#39;trace&#39;)


plot(res,&#39;treeRoot&#39;,show.tip.label=F)</code></pre>
<p><a href="https://xavierdidelot.github.io/BactDating/articles/yourData.html">This
tutorial</a> will enable us to do the analysis on our data</p>
<div id="meta-data" class="section level5">
<h5>0. Meta-data</h5>
<p>For Sterm I already have a metadata file. However for Ldel (or Lhelv)
not yet. In the end I do it manually!!!!!!!!!</p>
<pre class="r"><code>library(rentrez)
taxize_summ &lt;- entrez_summary(db=&quot;pubmed&quot;, id=24555091)
taxize_summ &lt;- entrez_summary(db=&quot;biosample&quot;,  id=2442656)

taxize_summ$sampledata


# entrez_summary()


entrez_dbs()</code></pre>
<pre class="bash"><code>##information gathered earlier

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt

##---------------------fetch info

head -1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.txt |cut -f 4 &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.tmp


efetch -db biosample -id SAMN02598530 -format xml | xtract -pattern BioSampleSet -division BioSample -group Attributes -element Attribute |grep &quot;collection&quot;
efetch -db biosample -id SAMN02598530 -format xml | xtract -pattern BioSampleSet -division BioSample -group Attributes -element Attribute 
efetch -db biosample -id SAMN02442656 -format xml | xtract -pattern BioSampleSet -division BioSample -group Attributes -element collection_date 


collection date



efetch -db biosample -id SAMN02598530 -format xml | xtract -pattern BioSampleSet -division BioSample -group Attributes -element Attribute |grep &quot;collection&quot;


efetch -db biosample -input /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.tmp -format xml


epost -db biosample -input /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.tmp -format acc |elink -target biosample| 

epost -db biosample -input /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_ldel.tmp -format acc | \
elink -target sra| \
efetch -db sra -format runinfo -mode xml | \
xtract -pattern Row -def &quot;NA&quot; -element Run spots bases spots_with_mates avgLength \
size_MB download_path Experiment LibraryStrategy LibrarySelection LibrarySource \
LibraryLayout InsertSize InsertDev Platform Model SRAStudy BioProject ProjectID \
Sample BioSample SampleType TaxID ScientificName SampleName CenterName \
Submission Consent &gt; /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp_03


 wget -q -O - &quot;https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=biosample&amp;id=SAMN04383980&quot; |\
   xmllint --xpath &quot;/BioSampleSet/BioSample/Attributes/Attribute[@attribute_name=&#39;source_name&#39;]/text()&quot;  -
   
   
 wget -q -O - &quot;https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=biosample&amp;id=SAMEA6957293&quot; |\
   xmllint --xpath &quot;/BioSampleSet/BioSample/Attributes/Attribute[@attribute_name=&#39;source_name&#39;]/text()&quot;  -


 wget -q -O - &quot;https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=biosample&amp;id=SAMEA418087&quot; |\
   xmllint --xpath &quot;/BioSampleSet/BioSample/Attributes/Attribute[@attribute_name=&#39;source_name&#39;]/text()&quot;  -



 wget -q -O - &quot;https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=biosample&amp;id=SAMN02598530&quot; |\
   xmllint --xpath &quot;/BioSampleSet/BioSample/Attributes/Attribute[@attribute_name=&#39;source_name&#39;]/text()&quot;  -
</code></pre>
</div>
<div id="create-a-starting-tree" class="section level5">
<h5>1. create a starting tree</h5>
<p>1.orthofinder</p>
<pre class="bash"><code>###-----------------------------------------
##Lhelv
###-----------------------------------------

species=Lhelv


rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

#leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/


rm -r  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/
cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

###-----------------------------------------
##only Metagenome
###-----------------------------------------

species=Lhelv

for species in $(echo &quot;Ldel Sterm&quot;)
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

#leterzzz=$(echo $species |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
done

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/Out-sali.faa

###-----------------------------------------
##remove outgroups and meta
###-----------------------------------------

species=Sterm
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/
mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/Out-vest.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/
mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/S202SMAG.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/  
mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/S50.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/  
mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/S72.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/  
  
species=Ldel
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/
mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/Out-helv.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/
mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/S202SMAG.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/NotUsed/   
  


###-----------------------------------------
##orthorfinder
###-----------------------------------------
#for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
species=Lhelv
mkdir -p    
#cp -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/Ldel/OrthoFinder/Results_Feb11/ /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/

for species in $(echo &quot;Sterm Ldel&quot;)
do

cd  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/

    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/ \
      -t 32 \
      -o    /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own//${species}/ \
      -a 32

done # species
</code></pre>
<p>1.Kirstin’s approach</p>
<pre class="bash"><code>date=Results_Mar11
species=Sterm
date=Results_Mar16
species=Lhelv
#/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/Results_Jul28/
for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
do
echo &quot;===========================================&quot;
echo -e ${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/get_singlecp_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt &gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt

wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt
wc -l /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt

done</code></pre>
<p>Here, I make a directory containing the fna and faa files of all
genomes with the file name corresponding to the locus tag.</p>
<pre class="bash"><code>

for species in $(echo &quot;Ldel Sterm&quot;)
do
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

leterzzz=$(echo $species |head -c 1)

#cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

#cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/

#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species_short}/PROKKA_FAA/
done
  
###--------------------
rm  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss_extended_02.txt
for species in $(echo &quot;Ldel Sterm&quot;)
do
rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/ |grep &quot;.faa$&quot;|sed &#39;s/.faa//g&#39;|grep -v &quot;Out-sali&quot;)
do
locusTagsss=$(head -1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/${genomesss}.faa|cut -d &#39;_&#39; -f 1|sed &#39;s/&gt;//g&#39;)
echo -e &quot;This  genome : &quot; ${genomesss} &quot; has the following locus Tag : &quot;${locusTagsss}
echo -e ${genomesss}&quot;\t&quot;${locusTagsss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss_extended_02.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/${genomesss}.faa &gt; \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FFN/${species}/${genomesss}.ffn &gt; \
/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.ffn


done
done

##------=================================================================
#extract orthogroups
##------=================================================================

for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/extract_orthologs.pl /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt --folder ${species}

done</code></pre>
<p>align protein families</p>
<pre class="bash"><code>

for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
do

cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep &quot;.faa$&quot; |sed &#39;s/.faa//g&#39;|grep -v &quot;Out-sali&quot; )
do
rm /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.aln.fasta
mafft --thread 37 --maxiterate 1000  --auto /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.faa &gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}_aln.fasta

done 

done #species</code></pre>
<p>here I back translate the faa alignment to dna.</p>
<pre class="bash"><code>
for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/aln_aa_to_dna.pl

done
</code></pre>
<p>With perl: prune alignments (removing columns represented by less
than 50% of the sequences)</p>
<pre class="bash"><code>

for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/prune_aln.pl

done

##================================================================
#remove all infomration (except genome info) from multifasta header
##================================================================

for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
do
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep &quot;_aln_prune.fasta$&quot; |sed &#39;s/_aln_prune.fasta//g&#39; )
do
sed -i &quot;s/_0.*//&quot; ${genesss}_aln_prune.fasta 

done
done


##-------------ldel names too long --&gt; shorten

species=Ldel
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_ncbi_own_ZOOMED/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep &quot;_aln_prune.fasta$&quot; |sed &#39;s/_aln_prune.fasta//g&#39; )
do
sed -i &#39;s/-DSM-/-/g&#39; ${genesss}_aln_prune.fasta 
sed -i &#39;s/-KCTC-/-/g&#39; ${genesss}_aln_prune.fasta 
done
</code></pre>
<p>With perl: Concatenate pruned alignments</p>
<pre class="bash"><code>  
#  for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
  for species in $(echo &quot;Sterm Ldel&quot;|tail -1)
  do
  echo -e ${species}
  cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
  perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/cat_align.pl &gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}.phylip
  
  done
  
    for species in $(echo &quot;Sterm Ldel&quot;|tail -1)
  do
  echo -e ${species}
  cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Genomes_FAA_FNA/${species}
  perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/cat_align_2fasta.pl &gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta
  
  done
  </code></pre>
<p>With RAxML: infer the phylogeny</p>
<pre class="bash"><code>mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/

  for species in $(echo &quot;Ldel Sterm&quot;|tail -1)
    for species in $(echo &quot;Ldel&quot;|tail -1)

#   for species in $(echo &quot;Sterm&quot;|tail -1)
  do
  echo -e ${species}
  mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/${species}
cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/${species}
sed &#39;s/\/.*//g&#39; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta &gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta

rm -r *${species}_all*

#/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta -n ${species}_all -T 37
/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta -n ${species}_all -T 37

done</code></pre>
<p>prepare tip.names file</p>
<pre class="bash"><code>rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/names.strains.txt
for species in $(echo &quot;Sterm Ldel&quot;|tail -1)
do
grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta| sed &#39;s/\/.*//g&#39;| sed &#39;s/&gt;//g&#39;  |awk -v namezzz=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print namezzz,$1}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/names.strains.txt
seq_length.py /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta
done</code></pre>
<p>calculate kappa============================</p>
<p>like <a href="https://github.com/xavierdidelot/ClonalFrameML/issues/95">here</a></p>
<p>Sterm———————–</p>
<p>alpha: 0.020000 Tree-Length: 0.268312 rate A &lt;-&gt; C: 1.264669
rate A &lt;-&gt; G: 3.668788 rate A &lt;-&gt; T: 0.752548 rate C
&lt;-&gt; G: 0.588894 rate C &lt;-&gt; T: 5.357598 rate G &lt;-&gt; T:
1.000000</p>
<p>freq pi(A): 0.292688 freq pi(C): 0.189221 freq pi(G): 0.220427 freq
pi(T): 0.297664</p>
<pre class="r"><code>(3.668788+5.357598)/(1.264669+0.752548+0.588894+1.000000)</code></pre>
<p>Ldel———————– Tree-Length: 0.104705 rate A &lt;-&gt; C: 1.197734 rate
A &lt;-&gt; G: 3.026860 rate A &lt;-&gt; T: 0.483401 rate C &lt;-&gt; G:
0.491727 rate C &lt;-&gt; T: 5.401485 rate G &lt;-&gt; T: 1.000000</p>
<p>freq pi(A): 0.253359 freq pi(C): 0.264227 freq pi(G): 0.269964 freq
pi(T): 0.212450</p>
<pre class="r"><code>(3.026860+5.401485)/(1.197734+0.483401+0.491727+1.000000)</code></pre>
<p>Lhelv———————– alpha: 0.020000 Tree-Length: 0.044759 rate A &lt;-&gt;
C: 1.446084 rate A &lt;-&gt; G: 3.449104 rate A &lt;-&gt; T: 0.727126
rate C &lt;-&gt; G: 0.814041 rate C &lt;-&gt; T: 4.536195 rate G
&lt;-&gt; T: 1.000000</p>
<p>freq pi(A): 0.320335 freq pi(C): 0.168171 freq pi(G): 0.209049 freq
pi(T): 0.302445</p>
<pre class="r"><code>(3.449104+4.536195)/(1.446084+0.727126+0.814041+1.000000)</code></pre>
</div>
<div id="create-a-clonalframeml-phylogeny" class="section level5">
<h5>2. create a ClonalFrameML phylogeny</h5>
<pre class="bash"><code>

conda create --name ClonalFrameML

conda install -c conda-forge -c bioconda -c defaults clonalframeml

conda activate ClonalFrameML

ClonalFrameML newick_file seq_file output_prefix [OPTIONS]

##-----------------Sterm-----------------------------

phylip2fasta.sh in=/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}.phylip out=/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}.fasta


conda activate ClonalFrameML

date=Results_Mar11
species=Sterm

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML

sed &#39;s/\/.*//g&#39; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta &gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta

seq_length.py /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta


ClonalFrameML /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/${species}/RAxML_bestTree.${species}_all \
       /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta  \
      /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML/${species}_clonal_tree -emsim 100  -kappa 2.50308 &gt; \
      /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML/${species}_clonal_tree.log.txt


##-----------------ldel-----------------------------


conda activate ClonalFrameML

date=Results_Mar11
species=Ldel

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML

sed &#39;s/\/.*//g&#39; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta &gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta

seq_length.py /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta


ClonalFrameML /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/${species}/RAxML_bestTree.${species}_all \
       /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta  \
      /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML/${species}_clonal_tree -emsim 100  -kappa 2.50308 &gt; \
      /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML/${species}_clonal_tree.log.txt


##-----------------Lhelv-----------------------------


conda activate ClonalFrameML

date=Results_Mar16
species=Lhelv

rm -r /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML

sed &#39;s/\/.*//g&#39; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg.fasta &gt; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta

seq_length.py /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta


ClonalFrameML /data/Project/2020_StarterCultureDiversity/11_referenceTree/raxml_all_combined_NCBI_own/${species}/RAxML_bestTree.${species}_all \
       /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/${species}_alg_cleaned.fasta  \
      /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML/${species}_clonal_tree -emsim 100  -kappa 2.002708 &gt; \
      /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ClonalFrameML/${species}_clonal_tree.log.txt
</code></pre>
<p>move local</p>
<pre class="bash"><code>


mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/


scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/names.strains.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//

scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Sterm/ClonalFrameML/Sterm_clonal_tree* /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//

scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Ldel/ClonalFrameML/Ldel_clonal_tree* /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//

scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/Lhelv/ClonalFrameML/Lhelv_clonal_tree* /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//
</code></pre>
</div>
<div id="bactdating" class="section level5">
<h5>3. BactDating</h5>
<pre class="r"><code>##-------------------------------Sterm
library(ape)
library(BactDating)
library(ggtree)
library(phytools)
library(treeio)
library(tidyverse)
library(coda)

t=loadCFML(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_clonal_tree&quot;)

# plot(t, show.tip.label = T)
# axisPhylo(backward = F)

##------------import names

# names_strains &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/names.strains.txt&quot;, 
#     delim = &quot;\t&quot;, escape_double = FALSE, 
#     col_names = c(&quot;Species&quot;,&quot;strain&quot;), trim_ws = TRUE)

##------------import metaData
meta_sterm &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited_extended_02.csv&quot;)
# colnames(meta_sterm)
meta_sterm$shortName_new &lt;- meta_sterm$shortName %&gt;% gsub(&quot;^&quot;,&quot;S&quot;,.) %&gt;% gsub(&quot;^SS&quot;,&quot;S&quot;,.)
# names_strains$strain

# t$tip.label[!t$tip.label %in% meta_sterm$shortName_new]

meta_sterm_onlySterm &lt;- meta_sterm[which(meta_sterm$shortName_new %in% t$tip.label),]

meta_sterm_onlySterm &lt;-meta_sterm_onlySterm[match(t$tip.label,meta_sterm_onlySterm$shortName_new),]

d &lt;- meta_sterm_onlySterm$extraction_time
names(d) &lt;- meta_sterm_onlySterm$shortName_new

##------------make dated tree

rooted=initRoot(t,d)
r=roottotip(rooted,d)
# r=roottotip(t,d)

##------------running calculation
res=bactdate(unroot(t),d,nbIts=100000,showProgress = TRUE)

##-----------------------plots
 png(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_sterm_roottotip.png&quot;, width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
r=roottotip(rooted,d)

dev.off()

 png(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_sterm_trace.png&quot;, width = 4000, height = 3000,res=300)
plot(res,&#39;trace&#39;)
dev.off()


 png(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_sterm_treeCI.png&quot;, width = 4000, height = 3000,res=300)
plot(res,&#39;treeCI&#39;,show.tip.label = F)
dev.off()


l=as.treedata.resBactDating(res)
obj=methods::new(&#39;treedata&#39;,phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

ggtree(obj, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)


write.beast(obj,&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_phylo_beast.txt&#39;)


mcmc=as.mcmc.resBactDating(res)
coda::effectiveSize(mcmc)
# mcmc_02 &lt;- mcmc
save(mcmc, file= &quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_model.rda&quot;)

 load(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_model.rda&#39;)
 coda::effectiveSize(mcmc)</code></pre>
<p>move files to bigmachine</p>
<pre class="bash"><code>


mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm//{input,script,output}

scp /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_clonal_tree* vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm/input/

scp ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited_extended_02.csv vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm//input/
</code></pre>
<pre class="r"><code>###------------------------------------------
#running on bigmachine
##saved in /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm/script/Sterm_script_01.rscript
# cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm/script/
#run: R CMD BATCH /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm/script/Sterm_script_01.rscript

###------------------------------------------

set_rounds=100000000
speciesss=&quot;Sterm&quot;
trial=&quot;v1&quot;
##-------------------------------Sterm
library(ape)
library(BactDating)
library(ggtree)
library(phytools)
library(treeio)
library(tidyverse)
library(coda)

t=loadCFML(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;//input//Sterm_clonal_tree&quot;)

##------------import metaData
meta_sterm &lt;- read_csv(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;//input//20200728_NCBI_log_shortNames_final_sterm_edited_extended_02.csv&quot;)

meta_sterm$shortName_new &lt;- meta_sterm$shortName %&gt;% gsub(&quot;^&quot;,&quot;S&quot;,.) %&gt;% gsub(&quot;^SS&quot;,&quot;S&quot;,.)

meta_sterm_onlySterm &lt;- meta_sterm[which(meta_sterm$shortName_new %in% t$tip.label),]

meta_sterm_onlySterm &lt;-meta_sterm_onlySterm[match(t$tip.label,meta_sterm_onlySterm$shortName_new),]

d &lt;- meta_sterm_onlySterm$extraction_time
names(d) &lt;- meta_sterm_onlySterm$shortName_new

##------------make dated tree

rooted=initRoot(t,d)
# r=roottotip(rooted,d)
# r=roottotip(t,d)

##------------running calculation
res=bactdate(unroot(t),d,nbIts=set_rounds,showProgress = TRUE)

##-----------------------plots


 png(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_roottotip.png&quot;), width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
r=roottotip(rooted,d)

dev.off()

 png(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_trace.png&quot;), width = 4000, height = 3000,res=300)
plot(res,&#39;trace&#39;)
dev.off()


 png(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_treeCI.png&quot;), width = 4000, height = 3000,res=300)
plot(res,&#39;treeCI&#39;,show.tip.label = F)
dev.off()




mcmc=as.mcmc.resBactDating(res)
coda::effectiveSize(mcmc)
# mcmc_02 &lt;- mcmc

paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_model.rda&quot;)

save(mcmc, file= paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_model.rda&quot;))

 load(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_model.rda&quot;))
text &lt;-  coda::effectiveSize(mcmc)


l=as.treedata.resBactDating(res)
obj=methods::new(&#39;treedata&#39;,phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

tree_plot &lt;- ggtree(obj, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = paste(&quot;mu=&quot;,round(text[1],digits=4),&quot;, sigma=&quot;,round(text[2],digits=4),&quot;, alpha=&quot;,round(text[3],digits=4)))


write.beast(obj,paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_phylo_beast.txt&quot;))

 png(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_final_tree.png&quot;), width = 4000, height = 3000,res=300)
tree_plot
dev.off()</code></pre>
<p>transfer local</p>
<pre class="bash"><code>
scp -r vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Sterm/output/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm/
</code></pre>
<p>Ldel</p>
<pre class="r"><code>##-------------------------------Ldel
library(ape)

library(BactDating)

t=loadCFML(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Ldel_clonal_tree&quot;)

t$tip.label


plot(t, show.tip.label = T)
axisPhylo(backward = F)

##------------import names

library(tidyverse)
# names_strains &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/names.strains.txt&quot;, 
#     delim = &quot;\t&quot;, escape_double = FALSE, 
#     col_names = c(&quot;Species&quot;,&quot;strain&quot;), trim_ws = TRUE)

##------------import metaData
meta_sterm &lt;-  read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Ldel_phylogeny_info.tsv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE)
colnames(meta_sterm)
meta_sterm$shortName_new &lt;- meta_sterm$shortName %&gt;% gsub(&quot;-DSM-&quot;,&quot;-&quot;,.) %&gt;% gsub(&quot;-KCTC-&quot;,&quot;-&quot;,.)
# names_strains$strain


# sed -i &#39;s/-DSM-/-/g&#39; ${genesss}_aln_prune.fasta 
# sed -i &#39;s/-KCTC-/-/g&#39; ${genesss}_aln_prune.fasta 

t$tip.label[!t$tip.label %in% meta_sterm$shortName_new]

meta_sterm_onlySterm &lt;- meta_sterm[which(meta_sterm$shortName_new %in% t$tip.label),]

meta_sterm_onlySterm &lt;-meta_sterm_onlySterm[match(t$tip.label,meta_sterm_onlySterm$shortName_new),]


d &lt;- meta_sterm_onlySterm$extraction_time
names(d) &lt;- meta_sterm_onlySterm$shortName_new

##------------make dated tree

rooted=initRoot(t,d)
r=roottotip(t,d)

r=roottotip(rooted,d)

plot(rooted, show.tip.label = T)
axisPhylo(backward = F)

##------------plot
res=bactdate(unroot(t),d,nbIts=10000000,showProgress = TRUE)

# res=bactdate(obsphy,dates,updateRoot=F,model=&#39;mixedgamma&#39;)


# plot(res,&#39;treeCI&#39;,show.tip.label = F)
# 
# plot(res,&#39;trace&#39;)
# 
# library(mcmcr)
# mcmc=as.mcmc(res)
# library(coda)
# 
# coda::save(mcmc, file= &quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_tree_100000_01&quot;)
# model = load(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_tree_100000_01&quot;)
# 
# coda
##-----------------------plots
 png(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_ldel_roottotip.png&quot;, width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
r=roottotip(rooted,d)

dev.off()

 png(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_ldel_trace.png&quot;, width = 4000, height = 3000,res=300)
plot(res,&#39;trace&#39;)
dev.off()


 png(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_ldel_treeCI.png&quot;, width = 4000, height = 3000,res=300)
plot(res,&#39;treeCI&#39;,show.tip.label = F)
dev.off()


plot(res,&#39;treeCI&#39;,show.tip.label = F)


mcmc=as.mcmc(res)


library(ggtree)
library(phytools)

library(treeio)

l=as.treedata.resBactDating(res)
obj=methods::new(&#39;treedata&#39;,phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

ggtree(obj, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)

write.beast(obj,&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel_phylo_beast.txt&#39;)


myModel.coda &lt;- read.bugs(myModel)

coda::write.model(mcmc,&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ldel_model.txt&#39;)

save(mcmc, file= &quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ldel_model.txt&quot;)
mcmc_02 = load(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ldel_model.txt&quot;)


mcmc=as.mcmc.resBactDating(res)
coda::effectiveSize(mcmc)
effectiveSize(mcmc)
mcmc_02 &lt;- mcmc
save(mcmc_02, file= &quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ldel_model.rda&quot;)

 load(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ldel_model.rda&#39;)
coda::effectiveSize(mcmc_02)



l=as.treedata.resBactDating(res)
obj=methods::new(&#39;treedata&#39;,phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

ggtree(obj, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)


print(res$pstrict)


# modelcompare()</code></pre>
<pre><code>   mu     sigma     alpha </code></pre>
<p>10.366933 56.111430 4.878684</p>
<p>move files to bigmachine</p>
<pre class="bash"><code>
mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel//{input,script,output}

scp /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Ldel_clonal_tree* vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel/input/

scp ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Ldel_phylogeny_info.tsv vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel//input/
</code></pre>
<pre class="r"><code>###------------------------------------------
#running on bigmachine
##saved in /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel/script/Ldel_script_01.rscript
# cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel/script/
#run: R CMD BATCH /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel/script/Ldel_script_01.rscript

###------------------------------------------

set_rounds=1000000000
speciesss=&quot;Ldel&quot;
trial=&quot;v1&quot;
##-------------------------------Sterm
library(ape)
library(BactDating)
library(ggtree)
library(phytools)
library(treeio)
library(tidyverse)
library(coda)

t=loadCFML(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;//input//Ldel_clonal_tree&quot;))

##------------import metaData

meta_sterm &lt;-  read_delim(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;//input//Ldel_phylogeny_info.tsv&quot;),  delim = &quot;\t&quot;, escape_double = FALSE,  trim_ws = TRUE)

meta_sterm$shortName_new &lt;- meta_sterm$shortName %&gt;% gsub(&quot;-DSM-&quot;,&quot;-&quot;,.) %&gt;% gsub(&quot;-KCTC-&quot;,&quot;-&quot;,.)

meta_sterm_onlySterm &lt;- meta_sterm[which(meta_sterm$shortName_new %in% t$tip.label),]

meta_sterm_onlySterm &lt;-meta_sterm_onlySterm[match(t$tip.label,meta_sterm_onlySterm$shortName_new),]

d &lt;- meta_sterm_onlySterm$extraction_time
names(d) &lt;- meta_sterm_onlySterm$shortName_new

##------------make dated tree

rooted=initRoot(t,d)
# r=roottotip(rooted,d)
# r=roottotip(t,d)

##------------running calculation
res=bactdate(unroot(t),d,nbIts=set_rounds,showProgress = TRUE)

##-----------------------plots


 png(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_roottotip.png&quot;), width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
r=roottotip(rooted,d)

dev.off()

 png(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_trace.png&quot;), width = 4000, height = 3000,res=300)
plot(res,&#39;trace&#39;)
dev.off()


 png(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_treeCI.png&quot;), width = 4000, height = 3000,res=300)
plot(res,&#39;treeCI&#39;,show.tip.label = F)
dev.off()




mcmc=as.mcmc.resBactDating(res)
coda::effectiveSize(mcmc)
# mcmc_02 &lt;- mcmc

paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_model.rda&quot;)

save(mcmc, file= paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_model.rda&quot;))

 load(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_model.rda&quot;))
text &lt;-  coda::effectiveSize(mcmc)


l=as.treedata.resBactDating(res)
obj=methods::new(&#39;treedata&#39;,phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

tree_plot &lt;- ggtree(obj, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = paste(&quot;mu=&quot;,round(text[1],digits=4),&quot;, sigma=&quot;,round(text[2],digits=4),&quot;, alpha=&quot;,round(text[3],digits=4)))


write.beast(obj,paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_phylo_beast.txt&quot;))

 png(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_final_tree.png&quot;), width = 4000, height = 3000,res=300)
tree_plot
dev.off()</code></pre>
<p>transfer local</p>
<pre class="bash"><code>mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel/
scp -r vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Ldel/output/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel/
</code></pre>
<p>Lhelv</p>
<pre class="r"><code>##-------------------------------Lhelv
library(ape)
library(tidyverse)
library(BactDating)
library(mcmcr)



t=loadCFML(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Lhelv_clonal_tree&quot;)

t$tip.label
t$edge.length
# 
# L=896286
# t$edge.length/L
# t$unrec
# 
# r$rate/L

plot(t, show.tip.label = T)
axisPhylo(backward = F)

##------------import names

library(tidyverse)
meta_sterm &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt&quot;,  delim = &quot;\t&quot;, escape_double = FALSE,) %&gt;% mutate(shortName_new=locusTAG)
meta_sterm_names &lt;- meta_sterm %&gt;% select(strain_name,shortName_new)

meta_sterm &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned.txt&quot;)  %&gt;% left_join(.,meta_sterm_names, by=&quot;strain_name&quot;)

##------------import metaData
# meta_sterm &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/&quot;)
# colnames(meta_sterm)
# meta_sterm$shortName_new &lt;- meta_sterm$shortName %&gt;% gsub(&quot;^&quot;,&quot;S&quot;,.) %&gt;% gsub(&quot;^SS&quot;,&quot;S&quot;,.)
# names_strains$strain

t$tip.label[!t$tip.label %in% meta_sterm$shortName_new]

meta_sterm_onlySterm &lt;- meta_sterm[which(meta_sterm$shortName_new %in% t$tip.label),]

meta_sterm_onlySterm &lt;-meta_sterm_onlySterm[match(t$tip.label,meta_sterm_onlySterm$shortName_new),]


d &lt;- meta_sterm_onlySterm$isolation_date
names(d) &lt;- meta_sterm_onlySterm$shortName_new

##------------make dated tree

# res=bactdate(t,d)

# res=bactdate(t,d,nbIts=1e5) #increase number of cycles


rooted=initRoot(t,d)
r=roottotip(rooted,d)
r=roottotip(t,d)

##------------plot
res=bactdate(unroot(t),d,nbIts=10000000,showProgress = TRUE)
# plot(res,&#39;treeCI&#39;,show.tip.label = F)
# 
# plot(res,&#39;trace&#39;)
# 
# library(mcmcr)
# mcmc=as.mcmc(res)
# library(coda)
# 
# coda::save(mcmc, file= &quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_tree_100000_01&quot;)
# model = load(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Sterm_tree_100000_01&quot;)
# 
# coda
##-----------------------plots
 png(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_Lhelv_roottotip.png&quot;, width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
r=roottotip(rooted,d)

dev.off()

 png(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_Lhelv_trace.png&quot;, width = 4000, height = 3000,res=300)
plot(res,&#39;trace&#39;,show.axis = TRUE)
dev.off()
# plot(res,&#39;treeRoot&#39;,)


 png(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/PLOTS_Lhelv_treeCI.png&quot;, width = 4000, height = 3000,res=300)
plot(res,&#39;treeCI&#39;,show.tip.label = F,)
dev.off()


# plotDualScale(res)

# plot
# plot(res,&#39;treeCI&#39;,show.tip.label = TRUE,show.axis = TRUE)

# unlist(res)
res$mu
mcmc=as.mcmc(res)
mcmc$model
mcmc$record

as.mcmc.resBactDating(res)
as.phylo(res)

tressss &lt;- mcmc$tree %&gt;% as.phylo()

library(ggtree)
library(phytools)

library(treeio)

l=as.treedata.resBactDating(res)
obj=methods::new(&#39;treedata&#39;,phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

ggtree(obj, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)

write.beast(obj,&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_phylo_beast.txt&#39;)

library(coda)
# mcmc = coda::as.mcmc(res)
# coda::effectiveSize(mcmc)

##----------output mcmc model 

myModel.coda &lt;- read.bugs(myModel)

coda::write.model(mcmc,&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_model.txt&#39;)

save(mcmc, file= &quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_model.txt&quot;)
mcmc_02 = load(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_model.txt&quot;)


mcmc=as.mcmc.resBactDating(res)
coda::effectiveSize(mcmc)
effectiveSize(mcmc)
mcmc_02 &lt;- mcmc
save(mcmc_02, file= &quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_model.rda&quot;)

 load(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_model.rda&#39;)
 effectiveSize(mcmc_02)

# 
# # 
# summary(res)
# tressss_02 &lt;- as.treedata.resBactDating(res)


# library(treedata)
# library(ggtree)
# res=bactdate(...)
# typeof(tressss_02)
# 
# 
# typeof(tressss)
# 
# tressss$tip.label
# meta_sterm_onlySterm$shortName_new
# 
# tressss$tip.label &lt;- meta_sterm_onlySterm$strain_name
# 
# ggtree(tressss)+ theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)
# 
# ape::write.tree(tressss, file=&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_phylo.txt&#39;)




# p12 &lt;- ggtree(tressss)+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)
# p12
# 
# # beast_file &lt;- system.file(&quot;examples/MCC_FluA_H3.tree&quot;, 
#                           # package=&quot;ggtree&quot;)
# beast_tree &lt;- read.beast(mcmc)
# ggtree(beast_tree, mrsd=&quot;2013-01-01&quot;) + theme_tree2()
# </code></pre>
<p>These things should be checked: - Effective Sample Size (ESS) &gt;
100 (<a href="https://github.com/xavierdidelot/BactDating/issues/53" class="uri">https://github.com/xavierdidelot/BactDating/issues/53</a>)
with the following function <em>effectiveSize(mcmc)</em> - output mcmc
model with bugs</p>
<p>ESS for Lhelv: mu sigma alpha 15.15673 259.35243 40.90496</p>
<pre><code>    mu     sigma     alpha </code></pre>
<p>6.71606 501.00000 11.15303</p>
<p>move files to bigmachine</p>
<pre class="bash"><code>



mkdir -p /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv//{input,script,output}

scp /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML//Lhelv_clonal_tree* vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv/input/

scp ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended* vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv//input/
</code></pre>
<pre class="r"><code>###------------------------------------------
#running on bigmachine
##saved in /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv/script/Lhelv_script_01.rscript
# cd /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv/script/
#run: R CMD BATCH /data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv/script/Lhelv_script_01.rscript

###------------------------------------------

set_rounds=10000000
speciesss=&quot;Lhelv&quot;
trial=&quot;v1&quot;
##-------------------------------Sterm
library(ape)
library(BactDating)
library(ggtree)
library(phytools)
library(treeio)
library(tidyverse)
library(coda)

t=loadCFML(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;//input//Lhelv_clonal_tree&quot;))

##------------import metaData
library(tidyverse)



meta_sterm &lt;- read_delim(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;//input//Lhelv_Refsss_extended_cleaned_02.txt&quot;),  delim = &quot;\t&quot;, escape_double = FALSE,) %&gt;% mutate(shortName_new=locusTAG)
meta_sterm_names &lt;- meta_sterm %&gt;% select(strain_name,shortName_new)

meta_sterm &lt;- read_csv(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;//input//Lhelv_Refsss_extended_cleaned.txt&quot;)) %&gt;% left_join(.,meta_sterm_names, by=&quot;strain_name&quot;)

#meta_sterm$shortName_new &lt;- meta_sterm$shortName  #%&gt;% gsub(&quot;^&quot;,&quot;S&quot;,.) %&gt;% gsub(&quot;^SS&quot;,&quot;S&quot;,.)

meta_sterm_onlySterm &lt;- meta_sterm[which(meta_sterm$shortName_new %in% t$tip.label),]

meta_sterm_onlySterm &lt;-meta_sterm_onlySterm[match(t$tip.label,meta_sterm_onlySterm$shortName_new),]

d &lt;- meta_sterm_onlySterm$isolation_date
names(d) &lt;- meta_sterm_onlySterm$shortName_new

##------------make dated tree

rooted=initRoot(t,d)
# r=roottotip(rooted,d)
# r=roottotip(t,d)

##------------running calculation
res=bactdate(unroot(t),d,nbIts=set_rounds,showProgress = TRUE)

##-----------------------plots


 png(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_roottotip.png&quot;), width = 4000, height = 3000,res=300)
# r=roottotip(t,d)
r=roottotip(rooted,d)

dev.off()

 png(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_trace.png&quot;), width = 4000, height = 3000,res=300)
plot(res,&#39;trace&#39;)
dev.off()


 png(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_treeCI.png&quot;), width = 4000, height = 3000,res=300)
plot(res,&#39;treeCI&#39;,show.tip.label = F)
dev.off()


mcmc=as.mcmc.resBactDating(res)
coda::effectiveSize(mcmc)
# mcmc_02 &lt;- mcmc

paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_model.rda&quot;)

save(mcmc, file= paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_model.rda&quot;))

 load(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_model.rda&quot;))
text &lt;-  coda::effectiveSize(mcmc)


l=as.treedata.resBactDating(res)
obj=methods::new(&#39;treedata&#39;,phylo=l[[1]],data=dplyr::tbl_df(as.data.frame(l[[2]])))

tree_plot &lt;- ggtree(obj, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = paste(&quot;mu=&quot;,round(text[1],digits=4),&quot;, sigma=&quot;,round(text[2],digits=4),&quot;, alpha=&quot;,round(text[3],digits=4)))

write.beast(obj,paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_phylo_beast.txt&quot;))

 png(paste0(&quot;/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/&quot;,speciesss,&quot;/output/PLOTS_&quot;,speciesss,&quot;_rounds&quot;,set_rounds,&quot;_&quot;,trial,&quot;_final_tree.png&quot;), width = 4000, height = 3000,res=300)
tree_plot
dev.off()</code></pre>
<p>transfer local</p>
<pre class="bash"><code>mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv/
scp -r vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/11_referenceTree/BactDating/Lhelv/output/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv/
</code></pre>
</div>
<div id="plotting" class="section level5">
<h5>4. Plotting</h5>
<pre class="r"><code>library(ape)
library(BactDating)
library(ggtree)
library(phytools)
library(treeio)
library(tidyverse)
library(coda)
library(philr)
##----------------Sterm

# myTree_Sterm_beast &lt;- read.beast(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_phylo_beast.txt&#39;)

# myTree_Sterm_beast &lt;- read.beast(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm/output/PLOTS_Sterm_rou&#39;)

myTree_Sterm_beast &lt;- read.beast(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_phylo_beast.txt&#39;)
tree_myTree_Sterm_beast &lt;- ggtree(myTree_Sterm_beast, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;S.thermophilus&quot;)+ coord_cartesian(clip = &#39;off&#39;)
tree_myTree_Sterm_beast

##----------------Ldel
# myTree_ldel_beast &lt;- read.beast(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel_phylo_beast.txt&#39;)


myTree_ldel_beast &lt;- read.beast(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel/output/PLOTS_Ldel_rounds1e+07_v1_phylo_beast.txt&#39;)
tree_myTree_ldel_beast &lt;- ggtree(myTree_ldel_beast, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;L.delbrueckii&quot;,x=&quot;Dating (year)&quot;)+ coord_cartesian(clip = &#39;off&#39;)



##----------------Lhelv
myTree_Lhelv_beast &lt;- read.beast(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_phylo_beast.txt&#39;)

# myTree_Lhelv_beast &lt;- read.beast(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv/output/PLOTS_Lhelv_rounds1e+05_v1_phylo_beast.txt&#39;)
gtree_myTree_Lhelv_beast &lt;- ggtree(myTree_Lhelv_beast, right=TRUE, mrsd=&quot;2020-04-02&quot;)+ theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;L.helveticus&quot;)+ coord_cartesian(clip = &#39;off&#39;)
gtree_myTree_Lhelv_beast
##====================combine

library(patchwork)


tree_myTree_Sterm_beast+tree_myTree_ldel_beast+gtree_myTree_Lhelv_beast

 
 pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/dated_phylogenies.pdf&quot;,width=10,height=6)

tree_myTree_Sterm_beast+tree_myTree_ldel_beast+gtree_myTree_Lhelv_beast


dev.off()


###-------------------------------------------------
##calculation of rootage and varation
###-------------------------------------------------

#----------Sterm
myTree_Sterm_beast_phylo &lt;- myTree_Sterm_beast %&gt;% as.phylo()
myTree_Sterm_beast_tibble &lt;- myTree_Sterm_beast %&gt;% as_tibble() 

myTree_Sterm_beast_tibble$branchAge_low &lt;- NA
myTree_Sterm_beast_tibble$branchAge_high &lt;- NA

for (listElement in 1:length(myTree_Sterm_beast_tibble$height_0.95_HPD)) {
  
myTree_Sterm_beast_tibble[listElement,&quot;branchAge_low&quot;] &lt;-    2020-as.numeric(myTree_Sterm_beast_tibble$height_0.95_HPD[[listElement]][1])
myTree_Sterm_beast_tibble[listElement,&quot;branchAge_high&quot;] &lt;-   2020-as.numeric(myTree_Sterm_beast_tibble$height_0.95_HPD[[listElement]][2])

}

myTree_Sterm_beast_tibble$branchAge_calculated &lt;- rowMeans(myTree_Sterm_beast_tibble[,c(&#39;branchAge_high&#39;, &#39;branchAge_low&#39;)], na.rm=TRUE)
myTree_Sterm_beast_tibble$branchAge_variation &lt;-  myTree_Sterm_beast_tibble$branchAge_calculated - myTree_Sterm_beast_tibble$branchAge_low

info &lt;- as_tibble(myTree_Sterm_beast_phylo)
distance &lt;- mean_dist_to_tips(myTree_Sterm_beast_phylo)

distance &lt;- mean_dist_to_tips(myTree_Sterm_beast_phylo) %&gt;% as.data.frame()%&gt;% rownames_to_column(&quot;label&quot;) %&gt;% as_tibble() %&gt;% mutate(branchAge_calculated_02 = 2020-.)  %&gt;% left_join(.,info)%&gt;% select(node,branchAge_calculated_02)

myTree_Sterm_beast_tibble_extended &lt;- left_join(myTree_Sterm_beast_tibble,distance,by=&quot;node&quot;) %&gt;% mutate(species=&quot;S. thermophilus&quot;)%&gt;% select(-height_0.95_HPD,-length_0.95_HPD)
write.csv(myTree_Sterm_beast_tibble_extended,&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_tree_age.csv&quot;)

#----------Ldel
myTree_Sterm_beast_phylo &lt;- myTree_ldel_beast %&gt;% as.phylo()
myTree_Sterm_beast_tibble &lt;- myTree_ldel_beast %&gt;% as_tibble() 

myTree_Sterm_beast_tibble$branchAge_low &lt;- NA
myTree_Sterm_beast_tibble$branchAge_high &lt;- NA

for (listElement in 1:length(myTree_Sterm_beast_tibble$height_0.95_HPD)) {
  
myTree_Sterm_beast_tibble[listElement,&quot;branchAge_low&quot;] &lt;-    2020-as.numeric(myTree_Sterm_beast_tibble$height_0.95_HPD[[listElement]][1])
myTree_Sterm_beast_tibble[listElement,&quot;branchAge_high&quot;] &lt;-   2020-as.numeric(myTree_Sterm_beast_tibble$height_0.95_HPD[[listElement]][2])

}

myTree_Sterm_beast_tibble$branchAge_calculated &lt;- rowMeans(myTree_Sterm_beast_tibble[,c(&#39;branchAge_high&#39;, &#39;branchAge_low&#39;)], na.rm=TRUE)
myTree_Sterm_beast_tibble$branchAge_variation &lt;-  myTree_Sterm_beast_tibble$branchAge_calculated - myTree_Sterm_beast_tibble$branchAge_low

info &lt;- as_tibble(myTree_Sterm_beast_phylo)
distance &lt;- mean_dist_to_tips(myTree_Sterm_beast_phylo)

distance &lt;- mean_dist_to_tips(myTree_Sterm_beast_phylo) %&gt;% as.data.frame()%&gt;% rownames_to_column(&quot;label&quot;) %&gt;% as_tibble() %&gt;% mutate(branchAge_calculated_02 = 2020-.)  %&gt;% left_join(.,info)%&gt;% select(node,branchAge_calculated_02)

myTree_Ldel_beast_tibble_extended &lt;- left_join(myTree_Sterm_beast_tibble,distance,by=&quot;node&quot;) %&gt;% mutate(species=&quot;L. delbrueckii&quot;)%&gt;% select(-height_0.95_HPD,-length_0.95_HPD)
write.csv(myTree_Ldel_beast_tibble_extended,&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel_tree_age.csv&quot;)
#----------Lhelv
myTree_Sterm_beast_phylo &lt;- myTree_Lhelv_beast %&gt;% as.phylo()
myTree_Sterm_beast_tibble &lt;- myTree_Lhelv_beast %&gt;% as_tibble() 

myTree_Sterm_beast_tibble$branchAge_low &lt;- NA
myTree_Sterm_beast_tibble$branchAge_high &lt;- NA

for (listElement in 1:length(myTree_Sterm_beast_tibble$height_0.95_HPD)) {
  
myTree_Sterm_beast_tibble[listElement,&quot;branchAge_low&quot;] &lt;-    2020-as.numeric(myTree_Sterm_beast_tibble$height_0.95_HPD[[listElement]][1])
myTree_Sterm_beast_tibble[listElement,&quot;branchAge_high&quot;] &lt;-   2020-as.numeric(myTree_Sterm_beast_tibble$height_0.95_HPD[[listElement]][2])

}

myTree_Sterm_beast_tibble$branchAge_calculated &lt;- rowMeans(myTree_Sterm_beast_tibble[,c(&#39;branchAge_high&#39;, &#39;branchAge_low&#39;)], na.rm=TRUE)
myTree_Sterm_beast_tibble$branchAge_variation &lt;-  myTree_Sterm_beast_tibble$branchAge_calculated - myTree_Sterm_beast_tibble$branchAge_low

info &lt;- as_tibble(myTree_Sterm_beast_phylo)
distance &lt;- mean_dist_to_tips(myTree_Sterm_beast_phylo)

distance &lt;- mean_dist_to_tips(myTree_Sterm_beast_phylo) %&gt;% as.data.frame()%&gt;% rownames_to_column(&quot;label&quot;) %&gt;% as_tibble() %&gt;% mutate(branchAge_calculated_02 = 2020-.)  %&gt;% left_join(.,info)%&gt;% select(node,branchAge_calculated_02)

Lhelv_genome_origin &lt;- read.csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt&quot;) %&gt;% mutate(label=locusTAG) %&gt;% select(label,strain_name)


myTree_Lhelv_beast_tibble_extended &lt;- left_join(myTree_Sterm_beast_tibble,distance,by=&quot;node&quot;) %&gt;% mutate(species=&quot;L. helveticus&quot;)%&gt;% select(-height_0.95_HPD,-length_0.95_HPD)%&gt;% left_join(., Lhelv_genome_origin, by=&#39;label&#39;) %&gt;% mutate(label=ifelse(is.na(strain_name),label,strain_name)) %&gt;% select(-strain_name)


write.csv(myTree_Lhelv_beast_tibble_extended,&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_tree_age.csv&quot;)

myTree_all_beast_tibble_extended &lt;- rbind(myTree_Sterm_beast_tibble_extended,myTree_Ldel_beast_tibble_extended,myTree_Lhelv_beast_tibble_extended)
write.csv(myTree_all_beast_tibble_extended,&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ALL_tree_age.csv&quot;)</code></pre>
<div id="rootage-info" class="section level6">
<h6>Rootage info</h6>
<pre class="r"><code>###-------------------------------------------------
##correlation std error vs. rootage
###-------------------------------------------------
library(tidyverse)
myTree_all_beast_tibble_extended &lt;- read_csv(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ALL_tree_age.csv&quot;)


plot_variation &lt;- ggplot(myTree_all_beast_tibble_extended,aes(x=branchAge_calculated,y=branchAge_variation))+geom_point()+facet_wrap(~species,scales = &quot;free&quot;)+theme_classic()+geom_smooth()+labs(x=&quot;Branch age&quot;,y=&quot;Branch age variation&quot;)
plot_variation

 png(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/97_Images/20210126/supplement/BranchAgeVariation.png&quot;, width = 2800, height = 2500,res=300)
 plot_variation
 dev.off()</code></pre>
</div>
<div id="plot-phyloeny-1" class="section level6">
<h6>Plot phyloeny 1</h6>
<p>remove all but root variation</p>
<pre class="r"><code>library(ape)
library(BactDating)
library(ggtree)
library(phytools)
library(treeio)
library(tidyverse)
library(coda)
library(philr)
##----------------Sterm
myTree_Sterm_beast &lt;- read.beast(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_phylo_beast.txt&#39;)

# myTree_Sterm_beast_data &lt;- myTree_Sterm_beast %&gt;% as_tibble() %&gt;% select(-parent,-branch.length)%&gt;% mutate(height_0.95_HPD=ifelse(label==&quot;Root&quot;,height_0.95_HPD,NA)) %&gt;% mutate(length_0.95_HPD=ifelse(label==&quot;Root&quot;,length_0.95_HPD,NA))%&gt;% select(-label)
myTree_Sterm_beast_data &lt;- myTree_Sterm_beast %&gt;% as_tibble() %&gt;% select(-parent,-branch.length)%&gt;% mutate(height_0.95_HPD=ifelse(label==&quot;nonAtAll&quot;,height_0.95_HPD,NA)) %&gt;% mutate(length_0.95_HPD=ifelse(label==&quot;nonAtAll&quot;,length_0.95_HPD,NA))%&gt;% select(-label)
myTree_Sterm_beast_phylo &lt;- myTree_Sterm_beast %&gt;% as.phylo()

myTree_Sterm_beast_cleaned &lt;- methods::new(&#39;treedata&#39;,phylo=myTree_Sterm_beast_phylo,data=dplyr::tbl_df(myTree_Sterm_beast_data))
tree_myTree_Sterm_beast &lt;- ggtree(myTree_Sterm_beast_cleaned, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;S.thermophilus&quot;)+ coord_cartesian(clip = &#39;off&#39;)
tree_myTree_Sterm_beast
##----------------Ldel
myTree_ldel_beast &lt;- read.beast(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel/output/PLOTS_Ldel_rounds1e+07_v1_phylo_beast.txt&#39;)
myTree_ldel_beast_data &lt;- myTree_ldel_beast %&gt;% as_tibble() %&gt;% select(-parent,-branch.length)%&gt;% mutate(height_0.95_HPD=ifelse(label==&quot;Root&quot;,height_0.95_HPD,NA)) %&gt;% mutate(length_0.95_HPD=ifelse(label==&quot;Root&quot;,length_0.95_HPD,NA))%&gt;% select(-label)

myTree_ldel_beast_data &lt;- myTree_ldel_beast %&gt;% as_tibble() %&gt;% select(-parent,-branch.length)%&gt;% mutate(height_0.95_HPD=ifelse(label==&quot;nonAtAll&quot;,height_0.95_HPD,NA)) %&gt;% mutate(length_0.95_HPD=ifelse(label==&quot;nonAtAll&quot;,length_0.95_HPD,NA))%&gt;% select(-label)
myTree_ldel_beast_phylo &lt;- myTree_ldel_beast %&gt;% as.phylo()

myTree_ldel_beast_cleaned &lt;- methods::new(&#39;treedata&#39;,phylo=myTree_ldel_beast_phylo,data=dplyr::tbl_df(myTree_ldel_beast_data))
tree_myTree_ldel_beast &lt;- ggtree(myTree_ldel_beast_cleaned, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;L.delbrueckii&quot;,x=&quot;Dating (year)&quot;)+ coord_cartesian(clip = &#39;off&#39;)
tree_myTree_ldel_beast

##----------------Lhelv
myTree_Lhelv_beast &lt;- read.beast(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_phylo_beast.txt&#39;)
myTree_Lhelv_beast_data &lt;- myTree_Lhelv_beast %&gt;% as_tibble() %&gt;% select(-parent,-branch.length)%&gt;% mutate(height_0.95_HPD=ifelse(label==&quot;Root&quot;,height_0.95_HPD,NA)) %&gt;% mutate(length_0.95_HPD=ifelse(label==&quot;Root&quot;,length_0.95_HPD,NA))%&gt;% select(-label)



# Lhelv_genome_origin &lt;- read.csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt&quot;) %&gt;% mutate(label=locusTAG) %&gt;% select(label,strain_name,origin)
# 
# 
# myTree_Lhelv_beast_cleaned_extended &lt;- left_join(as_tibble(myTree_Lhelv_beast_cleaned), Lhelv_genome_origin, by=&#39;label&#39;) %&gt;% mutate(label=ifelse(is.na(strain_name),label,strain_name)) %&gt;% select(-strain_name)  %&gt;% as.treedata()
# myTree_Lhelv_beast_cleaned_extended



myTree_Lhelv_beast_data &lt;- myTree_Lhelv_beast %&gt;% as_tibble() %&gt;% select(-parent,-branch.length)%&gt;% mutate(height_0.95_HPD=ifelse(label==&quot;nonAtAll&quot;,height_0.95_HPD,NA)) %&gt;% mutate(length_0.95_HPD=ifelse(label==&quot;nonAtAll&quot;,length_0.95_HPD,NA))%&gt;% select(-label)
myTree_Lhelv_beast_phylo &lt;- myTree_Lhelv_beast %&gt;% as.phylo()



myTree_Lhelv_beast_cleaned &lt;- methods::new(&#39;treedata&#39;,phylo=myTree_Lhelv_beast_phylo,data=dplyr::tbl_df(myTree_Lhelv_beast_data))
tree_myTree_ldel_beast &lt;- ggtree(myTree_Lhelv_beast_cleaned, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;L.delbrueckii&quot;,x=&quot;Dating (year)&quot;)+ coord_cartesian(clip = &#39;off&#39;)
tree_myTree_ldel_beast

# meta_Lhelv &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt&quot;,  delim = &quot;\t&quot;, escape_double = FALSE,)
# meta_Lhelv_names &lt;- meta_Lhelv %&gt;% select(strain_name,locusTAG)</code></pre>
<pre class="r"><code>library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)
##===========================================================================
##----------------------------------------------------------------------------
##Sterm
##----------------------------------------------------------------------------
##===========================================================================


tree_myTree_Sterm_beast &lt;- ggtree(myTree_Sterm_beast_cleaned, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;S.thermophilus&quot;)+ coord_cartesian(clip = &#39;off&#39;)
tree_myTree_Sterm_beast
  
##--------------------------------
##color labels of of only rmk phages
##--------------------------------
Sterm_genome_origin &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/logs/Sterm_genome_origin.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;label&quot;,&quot;origin&quot;),  trim_ws = TRUE) %&gt;% as.data.frame() %&gt;% mutate(label=paste0(&quot;S&quot;,label))%&gt;% mutate(label=gsub(&quot;^SS&quot;,&quot;S&quot;,label))

tmp &lt;- left_join(as_tibble(myTree_Sterm_beast_cleaned), Sterm_genome_origin, by=&#39;label&#39;)
tmp[is.na(tmp$origin),]
table(tmp$origin)


myTree_Sterm_beast_cleaned_extended &lt;- left_join(as_tibble(myTree_Sterm_beast_cleaned), Sterm_genome_origin, by=&#39;label&#39;) %&gt;% as.treedata()
myTree_Sterm_beast_cleaned_extended

tree_myTree_Sterm_beast_02 &lt;- ggtree(myTree_Sterm_beast_cleaned_extended, right=TRUE, mrsd=&quot;2020-04-02&quot;) +
        theme_tree2(plot.margin=margin(3, 60, 3, 3) )+
        geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;,aes(color=origin,size=origin)) +
        geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;S.thermophilus&quot;)+ 
        coord_cartesian(clip = &#39;off&#39;)+
        scale_color_manual(values=c(&quot;grey66&quot;,  &#39;black&#39;)) +
        scale_size_manual(values=c(1,1.5)) 
tree_myTree_Sterm_beast_02

##--------------------------------
##add popunk lineages
##--------------------------------
# library(plotly)
# plotly::ggplotly(tree_myTree_Sterm_beast)

all_sterm_strain_info_ZOOMED &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv&quot;,  col_types = cols(starterCulture = col_character(),  strain = col_character())) 
all_sterm_strain_info_ZOOMED_necessary &lt;- all_sterm_strain_info_ZOOMED %&gt;% mutate(label=strain) %&gt;% select(label,starterCulture,Cheese,lineage)

myTree_Sterm_beast_cleaned_extended_02_phylo &lt;- myTree_Sterm_beast_cleaned_extended %&gt;% as.phylo()
lineage_node_df &lt;- data.frame()
for (lineagesss in unique(all_sterm_strain_info_ZOOMED_necessary$lineage)) {
  
  tmp_names &lt;- all_sterm_strain_info_ZOOMED_necessary %&gt;% filter(lineage==lineagesss) %&gt;% filter(label %in% myTree_Sterm_beast_cleaned_extended_02_phylo$tip.label) %&gt;% pull(label)
  if (length(tmp_names)&gt;1) {
    
    node_tmp &lt;- getMRCA(myTree_Sterm_beast_cleaned_extended_02_phylo, tmp_names)

  lineage_node_df &lt;- rbind(lineage_node_df,data.frame(lineage=as.character(lineagesss),node=node_tmp))
    
  }
  if (length(tmp_names)==1) {
    
    node_tmp &lt;- as_tibble(myTree_Sterm_beast_cleaned_extended_02_phylo) %&gt;% filter(label==tmp_names) %&gt;% pull(node)

  lineage_node_df &lt;- rbind(lineage_node_df,data.frame(lineage=as.character(lineagesss),node=node_tmp))
    
  }
  
}
lineage_node_df_Sterm &lt;- lineage_node_df %&gt;% mutate(species=&quot;Sterm&quot;)

dim(lineage_node_df)
cols &lt;- rainbow(14) #I am not sure this is correct -- did not find the correct color labels anymore

tree_myTree_Sterm_beast_03 &lt;- tree_myTree_Sterm_beast_02+geom_hilight(node=lineage_node_df[1,2],fill=cols[1],extendto=2700)+
                          geom_hilight(node=lineage_node_df[2,2],fill=cols[2],extendto=2700)+
                          geom_hilight(node=lineage_node_df[3,2],fill=cols[3],extendto=2700)+
                            geom_hilight(node=lineage_node_df[4,2],fill=cols[4],extendto=2700)+
                            geom_hilight(node=lineage_node_df[5,2],fill=cols[5],extendto=2700)+
                            geom_hilight(node=lineage_node_df[6,2],fill=cols[6],extendto=2700)+
                            geom_hilight(node=lineage_node_df[7,2],fill=cols[7],extendto=2700)+
                            geom_hilight(node=lineage_node_df[8,2],fill=cols[8],extendto=2700)+
                            geom_hilight(node=lineage_node_df[9,2],fill=cols[9],extendto=2700)+
                            geom_hilight(node=lineage_node_df[10,2],fill=cols[10],extendto=2700)+
                            geom_hilight(node=lineage_node_df[11,2],fill=cols[11],extendto=2700)+
                            geom_hilight(node=lineage_node_df[12,2],fill=cols[12],extendto=2700)+
                            theme(legend.position=c(0.2, 0.4),legend.background=element_rect(fill = alpha(&quot;white&quot;, 0.1)),legend.title = element_blank(),legend.text = element_text(size = 7))

tree_myTree_Sterm_beast_03

##--------------------------------
##add RMK origin
##--------------------------------

# table_ldel_complete_test &lt;- read_delim(&quot;~/Desktop/Projects/2020_strainDelineation/01_logs/all_RMK_curated_Ldel.csv&quot;)
# table_sterm_complete_test02 &lt;- read_delim(&quot;~/Desktop/Projects/2020_strainDelineation/01_logs/all_RMK_curated_Sterm.csv&quot;) %&gt;% mutate(label=paste0(&quot;S&quot;,FAM)) %&gt;% select(label,starter.culture,cheese)%&gt;% mutate(starter.culture=as.character(starter.culture))


# myTree_Sterm_beast_cleaned_extended_02 &lt;- left_join(as_tibble(myTree_Sterm_beast_cleaned_extended), table_sterm_complete_test02, by=&#39;label&#39;)
# tmp[is.na(myTree_Sterm_beast_cleaned_extended_02$origin),]
# table(myTree_Sterm_beast_cleaned_extended_02$starter.culture)
# table(myTree_Sterm_beast_cleaned_extended_02$cheese)


myTree_Sterm_beast_cleaned_extended_02 &lt;- left_join(as_tibble(myTree_Sterm_beast_cleaned_extended), all_sterm_strain_info_ZOOMED_necessary, by=&#39;label&#39;) %&gt;% as.treedata()
myTree_Sterm_beast_cleaned_extended_02

library(ggnewscale)
COLORSSS &lt;- c(&quot;#d45400ff&quot;,&quot;#de592bff&quot;,&quot;#e36b42ff&quot;,&quot;#e67d59ff&quot;,&quot;#eb8f70ff&quot;,&quot;#eea38aff&quot;,&quot;#009cf0ff&quot;,&quot;#33cffaff&quot;,&quot;#7fe3ffff&quot;,&quot;#007f7fff&quot;,&quot;#00dcdcff&quot;,&quot;#ff0000ff&quot;,&quot;#0000ffff&quot;,&quot;#fde623ff&quot;,&quot;#00ff00ff&quot;,&quot;grey&quot;)


tree_myTree_Sterm_beast_03 &lt;- ggtree(myTree_Sterm_beast_cleaned_extended_02, right=TRUE, mrsd=&quot;2020-04-02&quot;) +
        theme_tree2(plot.margin=margin(3, 60, 3, 3) )+
        geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;,aes(color=origin,size=origin),hjust =-0.5) +
        geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;S.thermophilus&quot;)+ 
        coord_cartesian(clip = &#39;off&#39;)+
        scale_color_manual(values=c(&quot;grey66&quot;,  &#39;black&#39;)) +
        scale_size_manual(values=c(1,1.5))  +
                            geom_hilight(node=lineage_node_df[1,2],fill=cols[1],extendto=2700)+
                            geom_hilight(node=lineage_node_df[2,2],fill=cols[2],extendto=2700)+
                            geom_hilight(node=lineage_node_df[3,2],fill=cols[3],extendto=2700)+
                            geom_hilight(node=lineage_node_df[4,2],fill=cols[4],extendto=2700)+
                            geom_hilight(node=lineage_node_df[5,2],fill=cols[5],extendto=2700)+
                            geom_hilight(node=lineage_node_df[6,2],fill=cols[6],extendto=2700)+
                            geom_hilight(node=lineage_node_df[7,2],fill=cols[7],extendto=2700)+
                            geom_hilight(node=lineage_node_df[8,2],fill=cols[8],extendto=2700)+
                            geom_hilight(node=lineage_node_df[9,2],fill=cols[9],extendto=2700)+
                            geom_hilight(node=lineage_node_df[10,2],fill=cols[10],extendto=2700)+
                            geom_hilight(node=lineage_node_df[11,2],fill=cols[11],extendto=2700)+
                            geom_hilight(node=lineage_node_df[12,2],fill=cols[12],extendto=2700)+
                            theme(legend.position=&quot;none&quot;) +
                                  new_scale_fill() + new_scale_color()  + geom_tippoint(aes(shape=Cheese,fill=starterCulture,color=starterCulture), size=2)+scale_fill_manual(values = COLORSSS)+scale_color_manual(values = COLORSSS)


 # theme(legend.position=c(0.2, 0.4),legend.background=element_rect(fill = alpha(&quot;white&quot;, 0.1)),legend.title = element_blank(),legend.text = element_text(size = 7)) +
tree_myTree_Sterm_beast_03

write.beast(myTree_Sterm_beast_cleaned_extended_02,&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_phylo_beast_final.txt&#39;)


##===========================================================================
##----------------------------------------------------------------------------
##Ldel
##----------------------------------------------------------------------------
##===========================================================================


tree_myTree_ldel_beast &lt;- ggtree(myTree_ldel_beast_cleaned, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;L.delbrueckii&quot;,x=&quot;Dating (year)&quot;)+ coord_cartesian(clip = &#39;off&#39;)
tree_myTree_ldel_beast
  

##--------------------------------
##color labels of of only rmk phages
##--------------------------------
Ldel_genome_origin &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Ldel_phylogeny_info.tsv&quot;,  delim = &quot;\t&quot;, escape_double = FALSE,) %&gt;% mutate(label=shortName) %&gt;% select(label,origin)

# Ldel_genome_origin &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_Ldel/logs/Ldel_genome_origin_cleaned_02.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;label&quot;,&quot;origin&quot;),  trim_ws = TRUE) %&gt;% as.data.frame()

tmp &lt;- left_join(as_tibble(myTree_ldel_beast_cleaned), Ldel_genome_origin, by=&#39;label&#39;)
tmp[is.na(tmp$origin),]
table(tmp$origin)

myTree_ldel_beast_cleaned_extended &lt;- left_join(as_tibble(myTree_ldel_beast_cleaned), Ldel_genome_origin, by=&#39;label&#39;) %&gt;% as.treedata()
myTree_ldel_beast_cleaned_extended

tree_myTree_Ldel_beast_02 &lt;- ggtree(myTree_ldel_beast_cleaned_extended, right=TRUE, mrsd=&quot;2020-04-02&quot;) +
        theme_tree2(plot.margin=margin(3, 60, 3, 3) )+
        geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;,aes(color=origin,size=origin)) +
        geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;L. delbrueckii&quot;)+ 
        coord_cartesian(clip = &#39;off&#39;)+
        scale_color_manual(values=c(&quot;grey66&quot;,  &#39;black&#39;)) +
        scale_size_manual(values=c(1,1.5)) 
tree_myTree_Ldel_beast_02

##--------------------------------
##add popunk lineages
##--------------------------------
# library(plotly)
# plotly::ggplotly(tree_myTree_Sterm_beast)

all_sterm_strain_info_ZOOMED &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv&quot;,  col_types = cols(starterCulture = col_character(),  strain = col_character())) 
all_sterm_strain_info_ZOOMED_necessary &lt;- all_sterm_strain_info_ZOOMED %&gt;% mutate(label=strain) %&gt;% select(label,starterCulture,Cheese,lineage)

myTree_ldel_beast_cleaned_extended_02_phylo &lt;- myTree_ldel_beast_cleaned_extended %&gt;% as.phylo()
lineage_node_df &lt;- data.frame()
for (lineagesss in unique(all_sterm_strain_info_ZOOMED_necessary$lineage)) {
  
  tmp_names &lt;- all_sterm_strain_info_ZOOMED_necessary %&gt;% filter(lineage==lineagesss) %&gt;% filter(label %in% myTree_ldel_beast_cleaned_extended_02_phylo$tip.label) %&gt;% pull(label)
  if (length(tmp_names)&gt;1) {
    
    node_tmp &lt;- getMRCA(myTree_ldel_beast_cleaned_extended_02_phylo, tmp_names)

  lineage_node_df &lt;- rbind(lineage_node_df,data.frame(lineage=as.character(lineagesss),node=node_tmp))
    
  }
  if (length(tmp_names)==1) {
    
    node_tmp &lt;- as_tibble(myTree_ldel_beast_cleaned_extended_02_phylo) %&gt;% filter(label==tmp_names) %&gt;% pull(node)

  lineage_node_df &lt;- rbind(lineage_node_df,data.frame(lineage=as.character(lineagesss),node=node_tmp))
    
  }
  
}

dim(lineage_node_df)
lineage_node_df_Ldel &lt;- lineage_node_df %&gt;% mutate(species=&quot;Ldel&quot;)

cols &lt;- rainbow(14) #I am not sure this is correct -- did not find the correct color labels anymore

tree_myTree_Ldel_beast_03 &lt;- tree_myTree_Ldel_beast_02+geom_hilight(node=lineage_node_df[1,2],fill=cols[13],extendto=2700)+
                          geom_hilight(node=lineage_node_df[2,2],fill=cols[14],extendto=2700)+
                            theme(legend.position=c(0.2, 0.4),legend.background=element_rect(fill = alpha(&quot;white&quot;, 0.1)),legend.title = element_blank(),legend.text = element_text(size = 7))

tree_myTree_Ldel_beast_03

##--------------------------------
##add RMK origin
##--------------------------------


myTree_ldel_beast_cleaned_extended_02 &lt;- left_join(as_tibble(myTree_ldel_beast_cleaned_extended), all_sterm_strain_info_ZOOMED_necessary, by=&#39;label&#39;) %&gt;% as.treedata()
myTree_ldel_beast_cleaned_extended_02

library(ggnewscale)
COLORSSS &lt;- c(&quot;#d45400ff&quot;,&quot;#de592bff&quot;,&quot;#e36b42ff&quot;,&quot;#e67d59ff&quot;,&quot;#eb8f70ff&quot;,&quot;#eea38aff&quot;,&quot;#009cf0ff&quot;,&quot;#33cffaff&quot;,&quot;#7fe3ffff&quot;,&quot;#007f7fff&quot;,&quot;#00dcdcff&quot;,&quot;#ff0000ff&quot;,&quot;#0000ffff&quot;,&quot;#fde623ff&quot;,&quot;#00ff00ff&quot;,&quot;grey&quot;)


tree_myTree_Ldel_beast_03 &lt;- ggtree(myTree_ldel_beast_cleaned_extended_02, right=TRUE, mrsd=&quot;2020-04-02&quot;) +
        theme_tree2(plot.margin=margin(3, 60, 3, 3) )+
        geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;,aes(color=origin,size=origin),hjust =-0.5) +
        geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;L. delbrueckii&quot;)+ 
        coord_cartesian(clip = &#39;off&#39;)+
        scale_color_manual(values=c(&quot;grey66&quot;,  &#39;black&#39;)) +
        scale_size_manual(values=c(1,1.5)) +
                          geom_hilight(node=lineage_node_df[1,2],fill=cols[13],extendto=2700)+
                          geom_hilight(node=lineage_node_df[2,2],fill=cols[14],extendto=2700)+
                            theme(legend.position=&quot;none&quot;) +
                                  new_scale_fill() + new_scale_color()  + geom_tippoint(aes(shape=Cheese,fill=starterCulture,color=starterCulture), size=2)+scale_fill_manual(values = COLORSSS)+scale_color_manual(values = COLORSSS)

tree_myTree_Ldel_beast_03


write.beast(myTree_ldel_beast_cleaned_extended_02,&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel_phylo_beast_final.txt&#39;)

##===========================================================================
##----------------------------------------------------------------------------
##Lhelv
##----------------------------------------------------------------------------
##===========================================================================


tree_myTree_Lhelv_beast &lt;- ggtree(myTree_Lhelv_beast_cleaned, right=TRUE, mrsd=&quot;2020-04-02&quot;) + theme_tree2(plot.margin=margin(3, 60, 3, 3))+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+ geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;L. helveticus&quot;,x=&quot;Dating (year)&quot;)+ coord_cartesian(clip = &#39;off&#39;)
tree_myTree_Lhelv_beast
  
##--------------------------------
##change name
# and
##color labels of of only rmk phages
##--------------------------------
Lhelv_genome_origin &lt;- read.csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt&quot;) %&gt;% mutate(label=locusTAG) %&gt;% select(label,strain_name,origin)
# Lhelv_genome_origin &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt&quot;,  delim = &quot;\t&quot;, escape_double = FALSE,) %&gt;% mutate(label=locusTAG) %&gt;% select(label,strain_name,origin)


# tmp &lt;- left_join(as_tibble(myTree_Lhelv_beast_cleaned), Lhelv_genome_origin, by=&#39;label&#39;)
# tmp[is.na(tmp$origin),]
# table(tmp$origin)

myTree_Lhelv_beast_cleaned_extended &lt;- left_join(as_tibble(myTree_Lhelv_beast_cleaned), Lhelv_genome_origin, by=&#39;label&#39;) %&gt;% mutate(label=ifelse(is.na(strain_name),label,strain_name)) %&gt;% select(-strain_name)  %&gt;% as.treedata()
myTree_Lhelv_beast_cleaned_extended
# myTree_Lhelv_beast_cleaned_extended &lt;- left_join(as_tibble(myTree_Lhelv_beast_cleaned), Lhelv_genome_origin, by=&#39;label&#39;) %&gt;% as.treedata()
# myTree_Lhelv_beast_cleaned_extended

tree_myTree_Lhelv_beast_02 &lt;- ggtree(myTree_Lhelv_beast_cleaned_extended, right=TRUE, mrsd=&quot;2020-04-02&quot;) +
        theme_tree2(plot.margin=margin(3, 60, 3, 3) )+
        geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;,aes(color=origin,size=origin),hjust =-0.5) +
        geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;L. helveticus&quot;)+ 
        coord_cartesian(clip = &#39;off&#39;)+
        scale_color_manual(values=c(&quot;grey66&quot;,  &#39;black&#39;)) +
        scale_size_manual(values=c(1,1.5)) 
tree_myTree_Lhelv_beast_02

##--------------------------------
##add popunk lineages
##--------------------------------
# library(plotly)
# plotly::ggplotly(tree_myTree_Sterm_beast)

all_sterm_strain_info_ZOOMED &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats_Lhelv.csv&quot;,  col_types = cols(starterCulture = col_character(),  strain = col_character())) 
all_sterm_strain_info_ZOOMED_necessary &lt;- all_sterm_strain_info_ZOOMED %&gt;% mutate(label=strain) %&gt;% select(label,starterCulture,Cheese,lineage)

myTree_Lhelv_beast_cleaned_extended_02_phylo &lt;- myTree_Lhelv_beast_cleaned_extended %&gt;% as.phylo()
lineage_node_df &lt;- data.frame()
for (lineagesss in unique(all_sterm_strain_info_ZOOMED_necessary$lineage)) {
  
  tmp_names &lt;- all_sterm_strain_info_ZOOMED_necessary %&gt;% filter(lineage==lineagesss) %&gt;% filter(label %in% myTree_Lhelv_beast_cleaned_extended_02_phylo$tip.label) %&gt;% pull(label)
  if (length(tmp_names)&gt;1) {
    
    node_tmp &lt;- getMRCA(myTree_Lhelv_beast_cleaned_extended_02_phylo, tmp_names)

  lineage_node_df &lt;- rbind(lineage_node_df,data.frame(lineage=as.character(lineagesss),node=node_tmp))
    
  }
  if (length(tmp_names)==1) {
    
    node_tmp &lt;- as_tibble(myTree_Lhelv_beast_cleaned_extended_02_phylo) %&gt;% filter(label==tmp_names) %&gt;% pull(node)

  lineage_node_df &lt;- rbind(lineage_node_df,data.frame(lineage=as.character(lineagesss),node=node_tmp))
    
  }
  
}

dim(lineage_node_df)

lineage_node_df_lhelv &lt;- lineage_node_df %&gt;% mutate(species=&quot;Lhelv&quot;)

cols &lt;- rainbow(16) #I am not sure this is correct -- did not find the correct color labels anymore

tree_myTree_Lhelv_beast_03 &lt;- tree_myTree_Lhelv_beast_02+geom_hilight(node=lineage_node_df[1,2],fill=cols[15],extendto=2700)+
                                  geom_hilight(node=lineage_node_df[2,2],fill=cols[16],extendto=2700)+
                                  theme(legend.position=c(0.2, 0.4),legend.background=element_rect(fill = alpha(&quot;white&quot;, 0.1)),legend.title = element_blank(),legend.text = element_text(size = 7))

tree_myTree_Lhelv_beast_03

##--------------------------------
##add RMK origin
##--------------------------------


myTree_Lhelv_beast_cleaned_extended_02 &lt;- left_join(as_tibble(myTree_Lhelv_beast_cleaned_extended), all_sterm_strain_info_ZOOMED_necessary, by=&#39;label&#39;) %&gt;% as.treedata()
myTree_Lhelv_beast_cleaned_extended_02

library(ggnewscale)
# COLORSSS &lt;- c(&quot;#d45400ff&quot;,&quot;#de592bff&quot;,&quot;#e36b42ff&quot;,&quot;#e67d59ff&quot;,&quot;#eb8f70ff&quot;,&quot;#eea38aff&quot;,&quot;#009cf0ff&quot;,&quot;#33cffaff&quot;,&quot;#7fe3ffff&quot;,&quot;#007f7fff&quot;,&quot;#00dcdcff&quot;,&quot;#ff0000ff&quot;,&quot;#0000ffff&quot;,&quot;#fde623ff&quot;,&quot;#00ff00ff&quot;,&quot;grey&quot;)
COLORSSS &lt;- c(&quot;#de592bff&quot;,&quot;grey&quot;)


tree_myTree_Lhelv_beast_03 &lt;- ggtree(myTree_Lhelv_beast_cleaned_extended_02, right=TRUE, mrsd=&quot;2020-04-02&quot;) +
        theme_tree2(plot.margin=margin(3, 60, 3, 3) )+
        geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;,aes(color=origin,size=origin),hjust =-0.5) +
        geom_range(range=&#39;length_0.95_HPD&#39;, color=&#39;red&#39;, alpha=.2, size=2)+labs(title = &quot;L. helveticus&quot;)+ 
        coord_cartesian(clip = &#39;off&#39;)+
        scale_color_manual(values=c(&quot;grey66&quot;,  &#39;black&#39;)) +
        scale_size_manual(values=c(1,1.5)) +
                          geom_hilight(node=lineage_node_df[1,2],fill=cols[15],extendto=2700)+
                          geom_hilight(node=lineage_node_df[2,2],fill=cols[16],extendto=2700)+
                            theme(legend.position=&quot;none&quot;) +
                                  new_scale_fill() + new_scale_color()  + geom_tippoint(aes(shape=Cheese,fill=starterCulture,color=starterCulture), size=2)+scale_fill_manual(values = COLORSSS)+scale_color_manual(values = COLORSSS)

tree_myTree_Lhelv_beast_03

write.beast(myTree_Lhelv_beast_cleaned_extended_02,&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_phylo_beast_final.txt&#39;)


###==================================================================
#merge plots
###==================================================================
library(patchwork)


tree_myTree_Sterm_beast_03+tree_myTree_Ldel_beast_03+tree_myTree_Lhelv_beast_03

 
 pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/dated_phylogenies_colored.pdf&quot;,width=22,height=16)

tree_myTree_Sterm_beast_03+tree_myTree_Ldel_beast_03+tree_myTree_Lhelv_beast_03

dev.off()

###==================================================================
#output info
###==================================================================
# table(Sterm_genome_origin$species)
Sterm_genome_origin &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/20200728_NCBI_log_shortNames_final_sterm_edited_extended_02.csv&quot;, &quot;,&quot;, escape_double = FALSE, trim_ws = TRUE) %&gt;% filter(species %in% c(&quot;streptococcus thermophilus&quot;,&quot;thermophilus&quot;))%&gt;% filter(!is.na(biosample))%&gt;% select(c(shortName, biosample, extraction_time))%&gt;% mutate(species=&quot;S.thermophilus&quot;)



Ldel_genome_origin &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Ldel_phylogeny_info.tsv&quot;,  delim = &quot;\t&quot;, escape_double = FALSE,) %&gt;% filter(origin==&quot;NCBI_refs&quot;) %&gt;% select(c(shortName, biosample, extraction_time)) %&gt;% mutate(species=&quot;L.delbrueckii&quot;)
Lhelv_genome_origin &lt;- read.csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned.txt&quot;) %&gt;% mutate(extraction_time=isolation_date)%&gt;% mutate(shortName=strain_name) %&gt;% filter(species==&quot;helveticus&quot;) %&gt;% select(c(shortName, biosample, extraction_time))%&gt;% mutate(species=&quot;L.helveticus&quot;)


all_genome_origin &lt;-   rbind(Sterm_genome_origin,Ldel_genome_origin,Lhelv_genome_origin) %&gt;% as_tibble()

write_csv(all_genome_origin,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/all_genome_origin_final.txt&quot;) </code></pre>
<pre class="bash"><code>

scp ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/all_genome_origin_final.txt vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/
</code></pre>
</div>
<div id="age-of-the-strains" class="section level6">
<h6>Age of the strains</h6>
<pre class="r"><code>###-------------------------------------------
#Species age
#including subsp. age
###-------------------------------------------
library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)


###-------------------------------------------
#Crown/Root age
#including subsp. age
###-------------------------------------------

myTree_all_beast_tibble_extended &lt;- read.csv(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ALL_tree_age.csv&quot;)
myTree_all_beast_tibble_extended_crownAge &lt;- myTree_all_beast_tibble_extended %&gt;% filter(label==&quot;Root&quot;) %&gt;% select(species,branchAge_calculated,branchAge_variation,node,label) %&gt;% mutate(group=&quot;Crown&quot;)

# Combine with jitter points
ggplot(myTree_all_beast_tibble_extended_crownAge, aes(species, branchAge_calculated,fill=species,color=species)) +
  geom_point() + 
  geom_pointrange(aes(ymin = branchAge_calculated-branchAge_variation, ymax = branchAge_calculated+branchAge_variation),data = myTree_all_beast_tibble_extended_crownAge)+theme_classic()+facet_wrap(~group)+
  coord_flip()

#RMK strain age
###-------------------------------------------

all_sterm_strain_info_ZOOMED &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats_Lhelv.csv&quot;,  col_types = cols(starterCulture = col_character(),  strain = col_character())) 
all_sterm_strain_info_ZOOMED_necessary &lt;- all_sterm_strain_info_ZOOMED %&gt;% mutate(label=strain) %&gt;% select(label,starterCulture,Cheese,lineage)
table(all_sterm_strain_info_ZOOMED$species)

# tmp &lt;- all_sterm_strain_info_ZOOMED %&gt;% filter(species==&quot;Sterm&quot;)

table(all_sterm_strain_info_ZOOMED$species)
table(myTree_all_beast_tibble_extended$species)


strains_tmp &lt;- all_sterm_strain_info_ZOOMED %&gt;% filter(species==&quot;Ldel&quot;) %&gt;% filter(strain %in% myTree_ldel_beast_cleaned_extended_02_phylo$tip.label)%&gt;% pull(strain)
Ldel_MRCA &lt;- getMRCA(myTree_ldel_beast_cleaned_extended_02_phylo, strains_tmp)
myTree_all_beast_tibble_extended_Age_strains_Ldel &lt;- myTree_all_beast_tibble_extended %&gt;% filter(species==&quot;L. delbrueckii&quot;) %&gt;% filter(node==Ldel_MRCA) %&gt;% select(species,branchAge_calculated,branchAge_variation,node,label) %&gt;% mutate(group=&quot;RMK strains&quot;)

strains_tmp &lt;- all_sterm_strain_info_ZOOMED %&gt;% filter(species==&quot;Sterm&quot;) %&gt;% filter(strain %in% myTree_Sterm_beast_cleaned_extended_02_phylo$tip.label)%&gt;% pull(strain)
Sterm_MRCA &lt;- getMRCA(myTree_Sterm_beast_cleaned_extended_02_phylo, strains_tmp)
myTree_all_beast_tibble_extended_Age_strains_Sterm &lt;- myTree_all_beast_tibble_extended %&gt;% filter(species==&quot;S. thermophilus&quot;) %&gt;% filter(node==Sterm_MRCA) %&gt;% select(species,branchAge_calculated,branchAge_variation,node,label) %&gt;% mutate(group=&quot;RMK strains&quot;)

strains_tmp &lt;- all_sterm_strain_info_ZOOMED %&gt;% filter(species==&quot;Lhelv&quot;) %&gt;% filter(strain %in% myTree_Lhelv_beast_cleaned_extended_02_phylo$tip.label)%&gt;% pull(strain)
Lhelv_MRCA &lt;- getMRCA(myTree_Lhelv_beast_cleaned_extended_02_phylo, strains_tmp)
myTree_all_beast_tibble_extended_Age_strains_Lhelv &lt;- myTree_all_beast_tibble_extended %&gt;% filter(species==&quot;L. helveticus&quot;) %&gt;% filter(node==Lhelv_MRCA) %&gt;% select(species,branchAge_calculated,branchAge_variation,node,label) %&gt;% mutate(group=&quot;RMK strains&quot;)


myTree_all_beast_tibble_extended_all_01 &lt;- rbind(myTree_all_beast_tibble_extended_crownAge,myTree_all_beast_tibble_extended_Age_strains_Ldel,myTree_all_beast_tibble_extended_Age_strains_Sterm,myTree_all_beast_tibble_extended_Age_strains_Lhelv) %&gt;% mutate(grouping_extended=species)

ggplot() +
  # geom_point() + 
  geom_pointrange(aes(x=species, y=branchAge_calculated, ymin = branchAge_calculated-branchAge_variation, ymax = branchAge_calculated+branchAge_variation,fill=species,color=species),data = myTree_all_beast_tibble_extended_all_01)+theme_classic()+facet_wrap(~group,nrow = 2)+
  coord_flip()
###-------------------------------------------
#lineage age
###-------------------------------------------

lineage_node_df_Sterm$species &lt;- plyr::revalue(lineage_node_df_Sterm$species,c(&quot;Lhelv&quot;=&quot;L. helveticus&quot;,&quot;Sterm&quot;=&quot;S. thermophilus&quot;,&quot;Ldel&quot;=&quot;L. delbrueckii&quot;))
lineage_node_df_Sterm 

lineage_node_df_Ldel$species &lt;- plyr::revalue(lineage_node_df_Ldel$species,c(&quot;Lhelv&quot;=&quot;L. helveticus&quot;,&quot;Sterm&quot;=&quot;S. thermophilus&quot;,&quot;Ldel&quot;=&quot;L. delbrueckii&quot;))
lineage_node_df_Ldel

lineage_node_df_lhelv$species &lt;- plyr::revalue(lineage_node_df_lhelv$species,c(&quot;Lhelv&quot;=&quot;L. helveticus&quot;,&quot;Sterm&quot;=&quot;S. thermophilus&quot;,&quot;Ldel&quot;=&quot;L. delbrueckii&quot;))
lineage_node_df_lhelv

lineage_dating &lt;- rbind(lineage_node_df_Sterm,lineage_node_df_Ldel,lineage_node_df_lhelv) %&gt;% mutate(grouping_extended=paste0(species,&quot;_&quot;,lineage)) %&gt;% left_join(.,myTree_all_beast_tibble_extended,by=c(&quot;species&quot;,&quot;node&quot;)) %&gt;% filter(!is.na(branchAge_calculated_02))%&gt;% mutate(group=&quot;Lineage&quot;) %&gt;% mutate(branchAge_calculated=branchAge_calculated_02)%&gt;% select(species,branchAge_calculated,branchAge_variation,node,label,group,grouping_extended)


# myTree_all_beast_tibble_extended_all_01 &lt;- rbind(myTree_all_beast_tibble_extended_crownAge,myTree_all_beast_tibble_extended_Age_strains_Ldel,myTree_all_beast_tibble_extended_Age_strains_Sterm,myTree_all_beast_tibble_extended_Age_strains_Lhelv)
myTree_all_beast_tibble_extended_all_02 &lt;-  rbind(myTree_all_beast_tibble_extended_all_01,lineage_dating)

myTree_all_beast_tibble_extended_all_02$group = factor(myTree_all_beast_tibble_extended_all_02$group, levels=rev(c(&quot;Crown&quot;,&quot;RMK strains&quot;,&quot;Lineage&quot;)))

unique(myTree_all_beast_tibble_extended_all_02$grouping_extended)
myTree_all_beast_tibble_extended_all_02$grouping_extended = factor(myTree_all_beast_tibble_extended_all_02$grouping_extended, levels=rev(c(&quot;S. thermophilus&quot;  , &quot;L. delbrueckii&quot; ,   &quot;L. helveticus&quot; ,&quot;S. thermophilus_1&quot;,&quot;S. thermophilus_2&quot; ,  &quot;S. thermophilus_3&quot;,&quot;S. thermophilus_4&quot; ,&quot;S. thermophilus_5&quot; ,&quot;S. thermophilus_6&quot; ,&quot;S. thermophilus_7&quot; ,&quot;S. thermophilus_8&quot;, &quot;S. thermophilus_9&quot;, &quot;L. delbrueckii_13&quot; ,&quot;L. helveticus_15&quot; )))

cols_1 &lt;- rainbow(14) 
cols_2 &lt;- rainbow(16) 

colorsss &lt;- rev(c(&quot;#007fa6ff&quot;,&quot;#fdc2b5ff&quot;,&quot;#b3b3b3ff&quot;,cols_1[c(1,2,3,4,5,6,7,8,9,13)],cols_2[c(15)]))

branch_age_plot &lt;- ggplot() +
  geom_pointrange(aes(x=grouping_extended, y=branchAge_calculated, ymin = branchAge_calculated-branchAge_variation, ymax = branchAge_calculated+branchAge_variation,fill=grouping_extended,color=grouping_extended),data = myTree_all_beast_tibble_extended_all_02)+theme_classic()+facet_wrap(group~.,nrow = 3,scales = &quot;free_y&quot;, strip.position = &quot;right&quot;)+
  coord_flip()+scale_fill_manual(values = colorsss)+scale_color_manual(values = colorsss)+
  theme(axis.text.y = element_blank(),axis.title.y = element_blank(),axis.ticks.y = element_blank(),legend.position = &quot;none&quot;,strip.text = element_text(size = 7))+labs(y=&quot;Branch age&quot;)+scale_y_continuous(breaks=c(-10000,-7500,-5000,-2500,0,2020))

branch_age_plot

###==================================================================
#merge plots
###==================================================================
library(patchwork)


tree_myTree_Sterm_beast_03+tree_myTree_Ldel_beast_03+(tree_myTree_Lhelv_beast_03 + branch_age_plot+plot_layout(ncol=1,heights = c(2,1.5)))

 
 pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/dated_phylogenies_colored_extended.pdf&quot;,width=10,height=6)

tree_myTree_Sterm_beast_03+tree_myTree_Ldel_beast_03+(tree_myTree_Lhelv_beast_03 + branch_age_plot+plot_layout(ncol=1,heights = c(2,1.5)))

dev.off()
###==================================================================
#output info
###==================================================================

tmp1 &lt;- myTree_Lhelv_beast_cleaned_extended_02 %&gt;% as.tibble %&gt;% filter(origin!=&quot;NCBI_refs&quot;)%&gt;% pull(label) %&gt;% as_tibble %&gt;% mutate(species=&quot;Lhelv&quot;)
tmp2 &lt;- myTree_Sterm_beast_cleaned_extended_02 %&gt;% as.tibble %&gt;% filter(origin!=&quot;NCBI_refs&quot;)%&gt;% pull(label) %&gt;% as_tibble %&gt;% mutate(species=&quot;Sterm&quot;)
tmp3 &lt;- myTree_ldel_beast_cleaned_extended_02 %&gt;% as.tibble %&gt;% filter(origin!=&quot;NCBI_refs&quot;)%&gt;% pull(label) %&gt;% as_tibble %&gt;% mutate(species=&quot;Ldel&quot;)



tmp1 &lt;- myTree_Lhelv_beast_cleaned_extended_02 %&gt;% as.tibble %&gt;% filter(origin!=&quot;NCBI_refs&quot;)%&gt;% pull(label) %&gt;% as_tibble %&gt;% mutate(species=&quot;Lhelv&quot;)
tmp2 &lt;- myTree_Sterm_beast_cleaned_extended_02 %&gt;% as.tibble %&gt;% filter(origin!=&quot;NCBI_refs&quot;)%&gt;% pull(label) %&gt;% as_tibble %&gt;% mutate(species=&quot;Sterm&quot;)
tmp3 &lt;- myTree_ldel_beast_cleaned_extended_02 %&gt;% as.tibble %&gt;% filter(origin!=&quot;NCBI_refs&quot;)%&gt;% pull(label) %&gt;% as_tibble %&gt;% mutate(species=&quot;Ldel&quot;)

all_genomes_toether &lt;- rbind(tmp1,tmp2,tmp3)

table(all_genomes_toether$species)

write_csv(all_genomes_toether,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/all_genomes_dialact.txt&quot;) 

#scp ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/all_genomes_dialact.txt vincent@130.223.110.206:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/all_genomes_dialact.txt</code></pre>
</div>
</div>
<div id="age-pseudogene" class="section level5">
<h5>5. Age Pseudogene</h5>
<pre class="r"><code>library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)


##----------------------------------------------------
#Sterm
##----------------------------------------------------
# Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_curated.txt&quot;, tree.names = NULL, force.multi = FALSE)


myTree_Sterm_beast &lt;- read.beast(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Sterm_phylo_beast_final.txt&#39;)
tree_myTree_Sterm_beast &lt;- ggtree(myTree_Sterm_beast, right=TRUE, mrsd=&quot;2021-04-02&quot;) + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+labs(title = &quot;S.thermophilus&quot;)
tree_myTree_Sterm_beast

# write_csv(Orthogroups_orthofinder_stats_test_keepers_counts,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//genomes_ps.clustered.txt&quot;)


cluster_ps_counts_wide_imported &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//Sterm_cluster_ps_counts_wide_imported_wide_final&quot;)

# write_csv(cluster_ps_counts_wide_imported_wide_final,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_cluster_ps_counts_wide_imported_wide_final&quot;)

myTree_all_beast_tibble_extended_imported &lt;- read_csv(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ALL_tree_age.csv&quot;)

cluster_ps_counts_wide_imported$species &lt;-  plyr::revalue(cluster_ps_counts_wide_imported$species,c(&quot;Ldel&quot;=&quot;L. delbrueckii&quot;,&quot;Sterm&quot;=&quot;S. thermophilus&quot;,&quot;Lhelv&quot;=&quot;L. helveticus&quot;))

myTree_Sterm_beast_phylo &lt;- as.phylo(myTree_Sterm_beast)
myTree_Sterm_beast_phylo$tip.label
library(philr)

age_cluster_info &lt;- data.frame()




# for (speciesss in unique(cluster_ps_counts_wide_imported$species)) {
  speciesss &lt;- &quot;S. thermophilus&quot;
  
#  cluster_ps_counts_wide_imported_wide &lt;-  cluster_ps_counts_wide_imported %&gt;% filter(species==speciesss) %&gt;% select(-ps,-re,-species) %&gt;% spread(., genome, gene_presence)  %&gt;%replace(is.na(.), &quot;rm&quot;)
#   
# cluster_ps_counts_wide_imported_wide_final &lt;-  cluster_ps_counts_wide_imported_wide %&gt;%  gather(., genome,grouping , colnames(cluster_ps_counts_wide_imported_wide)[2:ncol(cluster_ps_counts_wide_imported_wide)], factor_key=TRUE,na.rm = TRUE) %&gt;% filter(grouping!=&quot;re&quot;)
 cluster_ps_counts_wide_imported_wide_final &lt;- cluster_ps_counts_wide_imported %&gt;% filter(species==speciesss) %&gt;% filter(grouping!=&quot;re&quot;) %&gt;% filter(grouping!=&quot;rm&quot;) %&gt;% mutate(cluster=Orthogroup)
 table(cluster_ps_counts_wide_imported_wide_final$grouping)
 
 
# cluster_ps_counts_wide_imported_species &lt;- cluster_ps_counts_wide_imported %&gt;% filter(species==speciesss)
print(speciesss)

for (clusterssss in unique(cluster_ps_counts_wide_imported_wide_final$cluster)) {
  print(clusterssss)

  cluster_ps_final_prep_tmp &lt;- cluster_ps_counts_wide_imported_wide_final %&gt;% filter(cluster==clusterssss)

  if (nrow(cluster_ps_final_prep_tmp)&gt;1) {
    
MRCA_of_tmp &lt;- getMRCA(as.phylo(myTree_Sterm_beast), cluster_ps_final_prep_tmp$genome)

branchAGE &lt;- myTree_all_beast_tibble_extended_imported %&gt;% filter(species==speciesss) %&gt;% filter(node==MRCA_of_tmp) %&gt;% pull(branchAge_calculated_02)

branchVariation &lt;- myTree_all_beast_tibble_extended_imported %&gt;% filter(species==speciesss) %&gt;% filter(node==MRCA_of_tmp) %&gt;% pull(branchAge_variation)
  

tmp_02 &lt;- data.frame(species=speciesss,cluster=clusterssss,node=MRCA_of_tmp,AGE=branchAGE,variation=branchVariation)

age_cluster_info &lt;- rbind(age_cluster_info,tmp_02)

  }

}
# }


myTree_all_beast_tibble_extended_imported %&gt;% filter(label==&quot;Root&quot;)

age_Sterm_ps &lt;- ggplot(age_cluster_info,aes(x=AGE))+geom_density()+theme_classic()+geom_vline(xintercept =-1843 )+lims(x=c(-5000,2020))

age_Sterm_ps


pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/age_Sterm_ps.pdf&quot;,width=7,height=3)

age_Sterm_ps

dev.off()

##----------------------------------------------------
#Ldel
##----------------------------------------------------
# Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_curated.txt&quot;, tree.names = NULL, force.multi = FALSE)


myTree_Ldel_beast &lt;- read.beast(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Ldel_phylo_beast_final.txt&#39;)
tree_myTree_Sterm_beast &lt;- ggtree(myTree_Ldel_beast, right=TRUE, mrsd=&quot;2021-04-02&quot;) + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+labs(title = &quot;S.thermophilus&quot;)
tree_myTree_Sterm_beast

# cluster_ps_counts_wide_imported &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//genomes_ps.clustered.txt&quot;)
 # cluster_ps_counts_wide_imported &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//Ldel_cluster_ps_counts_wide_imported_wide_final&quot;)
cluster_ps_counts_wide_imported &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//Ldel_cluster_ps_counts_wide_imported_wide_final&quot;)
# write_csv(cluster_ps_counts_wide_imported_wide_final,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Ldel_cluster_ps_counts_wide_imported_wide_final&quot;)

myTree_all_beast_tibble_extended_imported &lt;- read_csv(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ALL_tree_age.csv&quot;)

cluster_ps_counts_wide_imported$species &lt;-  plyr::revalue(cluster_ps_counts_wide_imported$species,c(&quot;Ldel&quot;=&quot;L. delbrueckii&quot;,&quot;Sterm&quot;=&quot;S. thermophilus&quot;,&quot;Lhelv&quot;=&quot;L. helveticus&quot;))

myTree_Ldel_beast_phylo &lt;- as.phylo(myTree_Ldel_beast)
myTree_Ldel_beast_phylo$tip.label
library(philr)

age_cluster_info_Ldel &lt;- data.frame()

# for (speciesss in unique(cluster_ps_counts_wide_imported$species)) {
  speciesss &lt;- &quot;L. delbrueckii&quot;
   # cluster_ps_counts_wide_imported_wide &lt;-  cluster_ps_counts_wide_imported %&gt;% filter(species==speciesss) %&gt;% select(-ps,-re,-species) %&gt;% spread(., genome, gene_presence)  %&gt;%replace(is.na(.), &quot;rm&quot;)
  
# cluster_ps_counts_wide_imported_wide_final &lt;-  cluster_ps_counts_wide_imported_wide %&gt;%  gather(., genome,grouping , colnames(cluster_ps_counts_wide_imported_wide)[2:ncol(cluster_ps_counts_wide_imported_wide)], factor_key=TRUE,na.rm = TRUE) %&gt;% filter(grouping!=&quot;re&quot;)
   cluster_ps_counts_wide_imported_wide_final &lt;- cluster_ps_counts_wide_imported %&gt;% filter(species==speciesss) %&gt;% filter(grouping!=&quot;re&quot;) %&gt;% filter(grouping!=&quot;rm&quot;)%&gt;% mutate(cluster=Orthogroup)
 table(cluster_ps_counts_wide_imported_wide_final$grouping)

 table(cluster_ps_counts_wide_imported_wide_final$grouping)

# cluster_ps_counts_wide_imported_species &lt;- cluster_ps_counts_wide_imported %&gt;% filter(species==speciesss)
print(speciesss)


unique(cluster_ps_counts_wide_imported_wide_final$genome) %in% myTree_Ldel_beast_phylo$tip.label
 myTree_Ldel_beast_phylo$tip.label %in% unique(cluster_ps_counts_wide_imported_wide_final$genome)


for (clusterssss in unique(cluster_ps_counts_wide_imported_wide_final$cluster)) {
  print(clusterssss)

  cluster_ps_final_prep_tmp &lt;- cluster_ps_counts_wide_imported_wide_final %&gt;% filter(cluster==clusterssss)

  if (nrow(cluster_ps_final_prep_tmp)&gt;1) {
    
MRCA_of_tmp &lt;- getMRCA(as.phylo(myTree_Ldel_beast_phylo), cluster_ps_final_prep_tmp$genome)

branchAGE &lt;- myTree_all_beast_tibble_extended_imported %&gt;% filter(species==speciesss) %&gt;% filter(node==MRCA_of_tmp) %&gt;% pull(branchAge_calculated_02)

branchVariation &lt;- myTree_all_beast_tibble_extended_imported %&gt;% filter(species==speciesss) %&gt;% filter(node==MRCA_of_tmp) %&gt;% pull(branchAge_variation)
  

tmp_02 &lt;- data.frame(species=speciesss,cluster=clusterssss,node=MRCA_of_tmp,AGE=branchAGE,variation=branchVariation)

age_cluster_info_Ldel &lt;- rbind(age_cluster_info_Ldel,tmp_02)

  }

}
# }


myTree_all_beast_tibble_extended_imported %&gt;% filter(label==&quot;Root&quot;)

age_Ldel_ps &lt;- ggplot(age_cluster_info_Ldel,aes(x=AGE))+geom_density()+theme_classic()+geom_vline(xintercept =-4933 )+lims(x=c(-5000,2020))

age_Ldel_ps


##----------------------------------------------------
#Lhelv
##----------------------------------------------------
# Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/GeneRax/orthofinder/Results_Feb11/Species_Tree/SpeciesTree_rooted_curated.txt&quot;, tree.names = NULL, force.multi = FALSE)


myTree_Lhelv_beast &lt;- read.beast(&#39;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/Lhelv_phylo_beast_final.txt&#39;)
tree_myTree_Sterm_beast &lt;- ggtree(myTree_Lhelv_beast, right=TRUE, mrsd=&quot;2021-04-02&quot;) + theme_tree2()+ geom_tiplab(align=TRUE, linesize=.5,geom=&quot;text&quot;)+labs(title = &quot;S.thermophilus&quot;)
tree_myTree_Sterm_beast

# cluster_ps_counts_wide_imported &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps//genomes_ps.clustered.txt&quot;)
cluster_ps_counts_wide_imported &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudo/Orthofinder_ps/Lhelv_cluster_ps_counts_wide_imported_wide_final&quot;)

myTree_all_beast_tibble_extended_imported &lt;- read_csv(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_ClonalFrameML/ALL_tree_age.csv&quot;)

cluster_ps_counts_wide_imported$species &lt;-  plyr::revalue(cluster_ps_counts_wide_imported$species,c(&quot;Ldel&quot;=&quot;L. delbrueckii&quot;,&quot;Sterm&quot;=&quot;S. thermophilus&quot;,&quot;Lhelv&quot;=&quot;L. helveticus&quot;))

myTree_Lhelv_beast_phylo &lt;- as.phylo(myTree_Lhelv_beast)
myTree_Lhelv_beast_phylo$tip.label
library(philr)

age_cluster_info_Lhelv &lt;- data.frame()

# for (speciesss in unique(cluster_ps_counts_wide_imported$species)) {
  speciesss &lt;- &quot;L. helveticus&quot;
   # cluster_ps_counts_wide_imported_wide &lt;-  cluster_ps_counts_wide_imported %&gt;% filter(species==speciesss) %&gt;% select(-ps,-re,-species) %&gt;% spread(., genome, gene_presence)  %&gt;%replace(is.na(.), &quot;rm&quot;)
  
# cluster_ps_counts_wide_imported_wide_final &lt;-  cluster_ps_counts_wide_imported_wide %&gt;%  gather(., genome,grouping , colnames(cluster_ps_counts_wide_imported_wide)[2:ncol(cluster_ps_counts_wide_imported_wide)], factor_key=TRUE,na.rm = TRUE) %&gt;% filter(grouping!=&quot;re&quot;)
    cluster_ps_counts_wide_imported_wide_final &lt;- cluster_ps_counts_wide_imported %&gt;% filter(species==speciesss) %&gt;% filter(grouping!=&quot;re&quot;) %&gt;% filter(grouping!=&quot;rm&quot;)%&gt;% mutate(cluster=Orthogroup)
 table(cluster_ps_counts_wide_imported_wide_final$grouping)

 table(cluster_ps_counts_wide_imported_wide_final$grouping)
 
# cluster_ps_counts_wide_imported_species &lt;- cluster_ps_counts_wide_imported %&gt;% filter(species==speciesss)
print(speciesss)

unique(cluster_ps_counts_wide_imported_wide_final$genome) %in% myTree_Lhelv_beast_phylo$tip.label
 myTree_Lhelv_beast_phylo$tip.label %in% unique(cluster_ps_counts_wide_imported_wide_final$genome)

for (clusterssss in unique(cluster_ps_counts_wide_imported_wide_final$cluster)) {
  print(clusterssss)

  cluster_ps_final_prep_tmp &lt;- cluster_ps_counts_wide_imported_wide_final %&gt;% filter(cluster==clusterssss)

  if (nrow(cluster_ps_final_prep_tmp)&gt;1) {
    
MRCA_of_tmp &lt;- getMRCA(as.phylo(myTree_Lhelv_beast_phylo), cluster_ps_final_prep_tmp$genome)

branchAGE &lt;- myTree_all_beast_tibble_extended_imported %&gt;% filter(species==speciesss) %&gt;% filter(node==MRCA_of_tmp) %&gt;% pull(branchAge_calculated_02)

branchVariation &lt;- myTree_all_beast_tibble_extended_imported %&gt;% filter(species==speciesss) %&gt;% filter(node==MRCA_of_tmp) %&gt;% pull(branchAge_variation)
  

tmp_02 &lt;- data.frame(species=speciesss,cluster=clusterssss,node=MRCA_of_tmp,AGE=branchAGE,variation=branchVariation)

age_cluster_info_Lhelv &lt;- rbind(age_cluster_info_Lhelv,tmp_02)

  }

}
# }


myTree_all_beast_tibble_extended_imported %&gt;% filter(label==&quot;Root&quot;) %&gt;% pull(branchAge_calculated_02)
myTree_all_beast_tibble_extended_imported %&gt;% filter(label==&quot;Root&quot;) %&gt;% select(species,branchAge_calculated_02)

age_Lhelv_ps &lt;- ggplot(age_cluster_info_Lhelv,aes(x=AGE))+geom_density()+theme_classic()+geom_vline(xintercept =-2650 )+lims(x=c(-5000,2020))

age_Lhelv_ps

###==================================================
#age see if younger then root
###==================================================
age_cluster_info_Lhelv %&gt;% mutate(root_or_not=ifelse(AGE&gt;-2645,&quot;new&quot;,&quot;root&quot;)) %&gt;% group_by(species,root_or_not) %&gt;% dplyr::summarise(countss=n()) %&gt;%  mutate(percentss=100*(countss/399))

age_cluster_info%&gt;% mutate(root_or_not=ifelse(AGE&gt;-1201,&quot;new&quot;,&quot;root&quot;)) %&gt;% group_by(species,root_or_not) %&gt;% dplyr::summarise(countss=n()) %&gt;%  mutate(percentss=100*(countss/546))


age_cluster_info_Ldel%&gt;% mutate(root_or_not=ifelse(AGE&gt;-4930,&quot;new&quot;,&quot;root&quot;)) %&gt;% group_by(species,root_or_not) %&gt;% dplyr::summarise(countss=n()) %&gt;%  mutate(percentss=100*(countss/333))

###==================================================
#all toegther
###==================================================

age_cluster_info_ALL &lt;- rbind(age_cluster_info_Lhelv,age_cluster_info,age_cluster_info_Ldel)

myTree_all_beast_tibble_extended_imported %&gt;% filter(label==&quot;Root&quot;) %&gt;% pull(branchAge_calculated_02)
myTree_all_beast_tibble_extended_imported %&gt;% filter(label==&quot;Root&quot;) %&gt;% select(species,branchAge_calculated_02)


age_cluster_info_ALL$species = factor(age_cluster_info_ALL$species, levels=c(&quot;S. thermophilus&quot;,&quot;L. delbrueckii&quot;,&quot;L. helveticus&quot;))


All_plot &lt;- ggplot(age_cluster_info_ALL,aes(x=AGE,color=species,fill=species))+geom_density(alpha=0.7)+theme_classic()+
            geom_vline(xintercept =-1201,color=&quot;#007fa6ff&quot;,size=1.5 )+
            geom_vline(xintercept =-4933,color=&quot;#fdc2b5ff&quot;,size=1.5 )+
            geom_vline(xintercept =-2650,color=&quot;grey77&quot; ,size=1.5)+
            lims(x=c(-5000,2020))+
            scale_color_manual(values=c(&quot;#007fa6ff&quot;,&quot;#fdc2b5ff&quot;,&quot;grey77&quot;))+
            scale_fill_manual(values=c(&quot;#007fa6ff&quot;,&quot;#fdc2b5ff&quot;,&quot;grey77&quot;))+labs(x=&quot;Pseudogene age&quot; )+theme(axis.text.y = element_blank(),axis.title.y = element_blank(),axis.ticks.y = element_blank(),legend.position = &quot;none&quot;)

All_plot

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/age_ALL_ps.pdf&quot;,width=7,height=3)

All_plot

dev.off()</code></pre>
</div>
</div>
</div>
</div>
<div id="s-broad-phylogeny" class="section level2">
<h2>16s broad phylogeny</h2>
<div id="extract-16s" class="section level3">
<h3>extract 16s</h3>
<pre class="bash"><code>##------------------------------------------
##2. run barrnap to extract 16s
## AND
##3. run V-Xtractor to extract 16s V4
##------------------------------------------
###---
##Sterm
###---


for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/ |grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39;)
do
echo ${genomess}

seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/${genomess}.fna

echo &quot;================================&quot;
done

#/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/anginosus.fna



rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s
#cd 
speciess=Sterm
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/ |grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39;)
do
echo ${genomess}
cat  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/${genomess}.fna |barrnap --threads 37 -o //data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa 

#speciess=$(grep &quot;${genomess}&quot; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2)

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa 


#rm  /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna
f16sss=$(grep &quot;16S&quot; -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  |sed &#39;s/&gt;//g&#39;|head -1)


#echo ${f16sss}
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  ${f16sss} &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_16s.fasta

#sed &quot;s/^&gt;.*$/&gt;16S-${speciess}-${genomess}-${num}/g&quot; /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/tmp.fna &gt;&gt; /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna

f23sss=$(grep &quot;23S&quot; -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  |sed &#39;s/&gt;//g&#39;|head -1)
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  ${f23sss} &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_23s.fasta
sed &quot;s/^&gt;.*$/&gt;23S-${speciess}-${genomess}/g&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_23s.fasta &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_23s.ffn


f5sss=$(grep &quot;5S&quot; -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  |sed &#39;s/&gt;//g&#39;|head -1)
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  ${f5sss} &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_5s.fasta
sed &quot;s/^&gt;.*$/&gt;5S-${speciess}-${genomess}/g&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_5s.fasta &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_5s.ffn

  
done #16s


speciess=Streptococcus
rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_16s.ffn
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/ |grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39;)
do
echo ${genomess}

seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa

#grep &quot;16S&quot; -c /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa 

#seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_16s.fasta

sed &quot;s/^&gt;.*$/&gt;16S-${speciess}-${genomess}/g&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_16s.fasta &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_16s.ffn


echo &quot;================================&quot;
done

grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_16s.ffn

###-----------------------------------------------------------
#merge the different contigs
###-----------------------------------------------------------
speciess=Streptococcus
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/
rm all_rRNAoperon.fasta
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/ |grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39;)
do

echo &quot;&gt;rRNAoperon-${speciess}-${genomess}&quot; &gt;&gt; all_rRNAoperon.fasta
grep &quot;&gt;&quot; -v  ${genomess}_16s.fasta &gt;&gt; all_rRNAoperon.fasta
echo &quot;NNNNNNNNNNNN&quot; &gt;&gt; all_rRNAoperon.fasta
grep &quot;&gt;&quot; -v  ${genomess}_23s.fasta &gt;&gt; all_rRNAoperon.fasta
echo &quot;NNNNNNNNNNNN&quot; &gt;&gt; all_rRNAoperon.fasta
grep &quot;&gt;&quot; -v ${genomess}_5s.fasta &gt;&gt; all_rRNAoperon.fasta


##make the assembly fit together nicely
awk &#39;BEGIN{RS=&quot;&gt;&quot;}NR&gt;1{sub(&quot;\n&quot;,&quot;\t&quot;); gsub(&quot;\n&quot;,&quot;&quot;); print RS$0}&#39; all_rRNAoperon.fasta |sed &#39;s/\t/\n/g&#39; |awk -v FS= &#39;/^&gt;/{print;next}{for (i=0;i&lt;=NF/60;i++) {for (j=1;j&lt;=60;j++) printf &quot;%s&quot;, $(i*60 +j); print &quot;&quot;}}&#39; &gt; all_rRNAoperon_prep.fasta


done

seq_length.py all_rRNAoperon_prep.fasta</code></pre>
</div>
<div id="all-lactic-acid-bacteria" class="section level3">
<h3>all lactic acid bacteria</h3>
<pre class="bash"><code>##------------------------------------------
##2. run barrnap to extract 16s
## AND
##3. run V-Xtractor to extract 16s V4
##------------------------------------------
###---
##Ldel
###---



for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/ |grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39;)
do
echo ${genomess}

seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/${genomess}.fna

echo &quot;================================&quot;
done

#/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/anginosus.fna



rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s
genomess=equicursoris

speciess=Ldel
cd //data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/ |grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39;)
do
echo ${genomess}
cat  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/${genomess}.fna |barrnap --threads 37 -o //data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa 

#speciess=$(grep &quot;${genomess}&quot; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/LdelDownloadgenomes.names |cut -f 2)

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa 


#rm  /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna
f16sss=$(grep &quot;16S&quot; -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa  |sed &#39;s/&gt;//g&#39;|head -1)


#echo ${f16sss}
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa  ${f16sss} &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_16s.fasta

#sed &quot;s/^&gt;.*$/&gt;16S-${speciess}-${genomess}-${num}/g&quot; /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/tmp.fna &gt;&gt; /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna

f23sss=$(grep &quot;23S&quot; -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa  |sed &#39;s/&gt;//g&#39;|head -1)
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa  ${f23sss} &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_23s.fasta
sed &quot;s/^&gt;.*$/&gt;23S-${speciess}-${genomess}/g&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_23s.fasta &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_23s.ffn


f5sss=$(grep &quot;5S&quot; -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa  |sed &#39;s/&gt;//g&#39;|head -1)
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa  ${f5sss} &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_5s.fasta
sed &quot;s/^&gt;.*$/&gt;5S-${speciess}-${genomess}/g&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_5s.fasta &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_5s.ffn

  
done #16s


speciess=Lactobacillus
rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_16s.ffn
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/ |grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39;)
do
echo ${genomess}

#seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/${genomess}.fna

#grep &quot;16S&quot; -c /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_rrna.fa 

seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_16s.fasta
seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_23s.fasta
seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_5s.fasta

sed &quot;s/^&gt;.*$/&gt;16S-${speciess}-${genomess}/g&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/${genomess}_16s.fasta &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_16s.ffn


echo &quot;================================&quot;
done

grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_16s.ffn

###-----------------------------------------------------------
#merge the different contigs
###-----------------------------------------------------------
speciess=Lactobacillus
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/
rm all_rRNAoperon.fasta
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/ |grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39;)
do

echo &quot;&gt;rRNAoperon-${speciess}-${genomess}&quot; &gt;&gt; all_rRNAoperon.fasta
grep &quot;&gt;&quot; -v  ${genomess}_16s.fasta &gt;&gt; all_rRNAoperon.fasta
echo &quot;NNNNNNNNNNNN&quot; &gt;&gt; all_rRNAoperon.fasta
grep &quot;&gt;&quot; -v  ${genomess}_23s.fasta &gt;&gt; all_rRNAoperon.fasta
echo &quot;NNNNNNNNNNNN&quot; &gt;&gt; all_rRNAoperon.fasta
grep &quot;&gt;&quot; -v ${genomess}_5s.fasta &gt;&gt; all_rRNAoperon.fasta


##make the assembly fit together nicely
awk &#39;BEGIN{RS=&quot;&gt;&quot;}NR&gt;1{sub(&quot;\n&quot;,&quot;\t&quot;); gsub(&quot;\n&quot;,&quot;&quot;); print RS$0}&#39; all_rRNAoperon.fasta |sed &#39;s/\t/\n/g&#39; |awk -v FS= &#39;/^&gt;/{print;next}{for (i=0;i&lt;=NF/60;i++) {for (j=1;j&lt;=60;j++) printf &quot;%s&quot;, $(i*60 +j); print &quot;&quot;}}&#39; &gt; all_rRNAoperon_prep.fasta


done

seq_length.py all_rRNAoperon_prep.fasta
</code></pre>
<p>#Biolog prep biolog plate names</p>
<pre class="bash"><code>cd Desktop/Projects/2020_StarterCultureDiversity/Biolog/

log_pm1.csv

rm log_pm1_final.csv
for locationsss in $(seq 1 2 191)
do
namessss=$((locationsss+1))
   echo &quot;location is at $locationsss and names at $namessss&quot;
   
final1=$(sed -n &quot;${locationsss}p&quot; log_pm1.csv)
final2=$(sed -n &quot;${namessss}p&quot; log_pm1.csv)

echo -e ${final1}&quot;\t&quot;${final2} &gt;&gt; log_pm1_final.csv

done
</code></pre>
<p>PREP file because the labels where filled into the wrong column</p>
<pre class="bash"><code>sed &#39;s/Sample Number/tmp/g&#39; ~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/biolog_all_strains/20210511__allHours_kinetic.csv |sed &#39;s/Strain Name  /Sample Number/g&#39; |sed &#39;s/tmp/Strain Name  /g&#39; &gt; ~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/biolog_all_strains/20210511__allHours_kinetic_cleaned.csv

sed &#39;s/Sample Number/tmp/g&#39; ~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/biolog_all_strains/20210511__allHours_kinetic.csv |sed &#39;s/Strain Name  /Sample Number/g&#39; |sed &#39;s/tmp/Strain Name  /g&#39; | grep -v &quot;Sample Number&quot; |sed &#39;s/Strain Name/Sample Number,\nStrain Name/g&#39; &gt; ~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/biolog_all_strains/20210511__allHours_kinetic_cleaned.csv


sed &#39;s/-1/NA NA/g&#39;

Sample Number</code></pre>
<p>with opm package</p>
<pre class="r"><code>library(opm)
library(tidyverse)
# library(plyr)


  # files &lt;- opm_files(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/biolog_all_strains/20210511__allHours_kinetic.csv&quot;)
  
example.opm &lt;- read_opm(names = &quot;~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/biolog_all_strains/20210511__allHours_kinetic_cleaned.csv&quot;, convert = &quot;try&quot;)

replicatess &lt;- &quot;1&quot;

flattened_opm &lt;- opm::flatten(example.opm)

meta_data &lt;- csv_data(example.opm) %&gt;% as.data.frame()

meta_data$platename &lt;- paste0(&quot;Plate &quot;,1:nrow(meta_data))
  

meta_data_sub &lt;- meta_data %&gt;% select(&quot;platename&quot;,&quot;Strain Name&quot;) %&gt;% rename(., &quot;Strain_Name&quot;=&quot;Strain Name&quot; )
cleaned_biolog_file &lt;- merge(flattened_opm,meta_data_sub,by.x=&quot;Plate&quot;,by.y=&quot;platename&quot;) %&gt;% add_column(replicate=replicatess)
unique(cleaned_biolog_file$Well)
cleaned_biolog_file$Well_cleaned &lt;-  gsub(&quot;.*\\(&quot;,&quot;&quot;,cleaned_biolog_file$Well) %&gt;% gsub(&quot;\\)&quot;,&quot;&quot;,.)
cleaned_biolog_file &lt;- as_tibble(cleaned_biolog_file)
###--------------------kinetics name


p1 &lt;- ggplot(cleaned_biolog_file,aes(x=Time,y=Value,group=Well_cleaned,color=Well_cleaned))+geom_line()+facet_wrap(~Strain_Name)+theme_classic()
p1


unique(cleaned_biolog_file$Well_cleaned)

cleaned_biolog_file_test &lt;- cleaned_biolog_file %&gt;% filter(Well_cleaned==&quot;a-D-Lactose&quot;)

p1 &lt;- ggplot(cleaned_biolog_file_test,aes(x=Time,y=Value,group=Strain_Name,color=Strain_Name))+geom_line()+theme_classic()
p1


cleaned_biolog_file_test &lt;- cleaned_biolog_file %&gt;% filter(Well_cleaned==&quot;D-Xylose&quot;)

p1 &lt;- ggplot(cleaned_biolog_file_test,aes(x=Time,y=Value,group=Strain_Name,color=Strain_Name))+geom_line()+theme_classic()
p1



unique(cleaned_biolog_file$Well_cleaned)

cleaned_biolog_file_test &lt;- cleaned_biolog_file %&gt;% filter(Well_cleaned== &quot;Negative Control&quot; )

p1 &lt;- ggplot(cleaned_biolog_file_test,aes(x=Time,y=Value,group=Strain_Name,color=Strain_Name))+geom_line()+theme_classic()
p1



  library(plotly)
ggp &lt;- ggplotly(p1)
 ggp

 
 


 
 
###--------------------make binary


Biolog_diff_new &lt;- aggregate(. ~`Strain_Name`+Well_cleaned, data=cleaned_biolog_file[,c(&quot;Strain_Name&quot;,&quot;Well_cleaned&quot;,&quot;Value&quot;)], FUN = function(i)max(i) - min(i))
hist(Biolog_diff_new$Value,breaks = 20)
pDensity &lt;- ggplot(Biolog_diff_new,aes(x=Value,group=Strain_Name,color=`Strain_Name`,fill=Strain_Name))+geom_density()+theme_classic()+facet_wrap(Strain_Name~.)+geom_vline(xintercept =90)+labs(x=&quot;Intensity difference between maximal and minimal value&quot;)+theme(legend.position = &quot;none&quot;)
pDensity


###--------------------check nagative control

cleaned_biolog_file_Neg_con &lt;- cleaned_biolog_file %&gt;% filter(Well_cleaned== &quot;Negative Control&quot; ) %&gt;% group_by(Plate,Strain_Name,replicate) %&gt;% summarise(maxs=max(Value),medianss=median(Value),sds=sd(Value)) %&gt;% select(Strain_Name,medianss)

Biolog_diff_new_norm &lt;- Biolog_diff_new %&gt;% left_join(.,cleaned_biolog_file_Neg_con) %&gt;% as_tibble() %&gt;% mutate(Value=Value-medianss) %&gt;% select(-medianss) %&gt;% mutate(Value=ifelse(Value&lt;0,0,Value))

Biolog_diff_new_norm %&gt;% filter(Well_cleaned %in% &quot;a-D-Lactose&quot;)

# Biolog_diff_new_norm &lt;- Biolog_diff_new_norm %&gt;% mutate(Value=ifelse(Value&gt;20,1,0))

# Biolog_diff_new %&gt;% as_tibble()
 

ggheatmap &lt;- ggplot(Biolog_diff_new_norm, aes(y=Well_cleaned, x=Strain_Name, fill = Value))+
 geom_tile(color = &quot;white&quot;,size=1.1)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;red&quot;, mid = &quot;white&quot;,
   midpoint = 100, limit = c(0,259), space = &quot;Lab&quot;,
   name=&quot;max growth&quot;) +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = &quot;top&quot;) +
  # labs(x=expression(italic(&quot;S. thermophilus&quot;)),y=expression(italic(&quot;L. delbrueckii&quot;)))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)


#----discrete
Biolog_diff_new_norm_discrete &lt;- Biolog_diff_new_norm %&gt;% mutate(Value=ifelse(Value&gt;50,1,0))



ggheatmap &lt;- ggplot(Biolog_diff_new_norm_discrete, aes(y=Well_cleaned, x=Strain_Name, fill = Value))+
 geom_tile(color = &quot;white&quot;,size=1.1)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;red&quot;, mid = &quot;white&quot;,
   midpoint = 0.5, limit = c(0,1), space = &quot;Lab&quot;,
   name=&quot;max growth&quot;) +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = &quot;top&quot;) +
  # labs(x=expression(italic(&quot;S. thermophilus&quot;)),y=expression(italic(&quot;L. delbrueckii&quot;)))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)
###--------------------change educt order


Biolog_diff_new_high &lt;- Biolog_diff_new
Biolog_diff_new_high$Well_cleaned &lt;- droplevels(as.factor(Biolog_diff_new_high$Well_cleaned))
Biolog_diff_new_high$Strain_Name &lt;- droplevels(as.factor(Biolog_diff_new_high$Strain_Name))
table(Biolog_diff_new_high$Well_cleaned,Biolog_diff_new_high$Strain_Name)

# Biolog_diff %&gt;%  filter(well==&quot;C09&quot;)

# Biolog_diff_high_ext &lt;- Biolog_diff %&gt;% filter(well %in% c(as.character(unique(Biolog_diff_high$well)),&quot;A06&quot;)) #subset dataset
# Biolog_diff_high_ext &lt;- Biolog_diff  #%&gt;% filter(well %in% c(as.character(unique(Biolog_diff_high$well)),&quot;A06&quot;))
# Biolog_diff_high_ext &lt;- merge(Biolog_diff,log_pm1_final,by=&quot;well&quot;) %&gt;% mutate(well=metabolite) %&gt;% select(-metabolite)  #%&gt;% filter(well %in% c(as.character(unique(Biolog_diff_high$well)),&quot;A06&quot;))
table(Biolog_diff_new_high$Well_cleaned)
table(Biolog_diff_new_high$well)
# unique(Biolog_diff_high_ext$`Field 2`)
# Biolog_diff_high_ext$`Field 2` &lt;- revalue(Biolog_diff_high_ext$`Field 2`, c( &quot;Stamm 37&quot;=&quot;S24737&quot;, &quot;Stamm 38&quot;=&quot;S24738&quot;, &quot;Stamm 39&quot;=&quot;S24739&quot;, &quot;Stamm 40&quot;=&quot;S24740&quot; ,&quot;Stamm new 1&quot;=&quot;24854&quot;, &quot;Stamm new 2&quot;=&quot;24855&quot;))#subset dataset





# Biolog_diff_high_widess &lt;- spread(Biolog_diff_new, Strain_Name, Value) %&gt;% column_to_rownames(var=&quot;Well_cleaned&quot;)



sorting_vector &lt;- Biolog_diff_new %&gt;% group_by(Well_cleaned) %&gt;% summarize(final=sum(Value)) %&gt;% arrange(final) %&gt;% dplyr::pull(., Well_cleaned)

Biolog_diff_new$Well_cleaned = factor(Biolog_diff_new$Well_cleaned, levels=sorting_vector)


library(gplots)
library(RColorBrewer)

    dev.off()
range(Biolog_diff_new$Value)
ggheatmap &lt;- ggplot(Biolog_diff_new, aes(y=Well_cleaned, x=Strain_Name, fill = Value))+
 geom_tile(color = &quot;white&quot;,size=1.1)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;red&quot;, mid = &quot;white&quot;,
   midpoint = 100, limit = c(0,259), space = &quot;Lab&quot;,
   name=&quot;max growth&quot;) +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = &quot;top&quot;) +
  # labs(x=expression(italic(&quot;S. thermophilus&quot;)),y=expression(italic(&quot;L. delbrueckii&quot;)))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)
  
  


#    pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf&quot;,width=6,height=8)
# 
#   print(ggheatmap)
# 
# dev.off()


##-------------------------------
#only Streptococcus
##-------------------------------


Biolog_diff_new_trepto &lt;- Biolog_diff_new %&gt;% filter(Strain_Name %in% c(&quot;24736&quot;,&quot;24735&quot;,&quot;24734&quot;,&quot;24750&quot;,&quot;24739&quot;,&quot;24745&quot;,&quot;24728&quot;,&quot;24742&quot;,&quot;24746&quot;,&quot;24749&quot;,&quot;24730&quot;,&quot;24743&quot;,&quot;S_sal_48&quot;))


Biolog_diff_new_trepto$Strain_Name &lt;- plyr::revalue(Biolog_diff_new_trepto$Strain_Name, c(&quot;24736&quot;=&quot;lineage 1&quot;,&quot;24735&quot;=&quot;lineage 2&quot;,&quot;24734&quot;=&quot;lineage 3&quot;,&quot;24750&quot;=&quot;lineage 4&quot;,&quot;24739&quot;=&quot;lineage 5&quot;,&quot;24745&quot;=&quot;lineage 6&quot;,&quot;24728&quot;=&quot;lineage 7&quot;,&quot;24742&quot;=&quot;lineage 8&quot;,&quot;24746&quot;=&quot;lineage 9&quot;,&quot;24749&quot;=&quot;lineage 10&quot;,&quot;24730&quot;=&quot;lineage 11&quot;,&quot;24743&quot;=&quot;lineage 12&quot;,&quot;S_sal_48&quot;=&quot;S. salivarius&quot;))


Biolog_diff_new_trepto$Strain_Name  = factor(Biolog_diff_new_trepto$Strain_Name , levels=c(&quot;lineage 1&quot;,&quot;lineage 2&quot;,&quot;lineage 3&quot;,&quot;lineage 4&quot;,&quot;lineage 5&quot;,&quot;lineage 6&quot;,&quot;lineage 7&quot;,&quot;lineage 8&quot;,&quot;lineage 9&quot;,&quot;lineage 10&quot;,&quot;lineage 11&quot;,&quot;lineage 12&quot;,&quot;S. salivarius&quot;))

    dev.off()
range(Biolog_diff_new_trepto$Value)
ggheatmap &lt;- ggplot(Biolog_diff_new_trepto, aes(y=Well_cleaned, x=Strain_Name, fill = Value))+
 geom_tile(color = &quot;white&quot;,size=1.1)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;red&quot;, mid = &quot;white&quot;,
   midpoint = 100, limit = c(0,224), space = &quot;Lab&quot;,
   name=&quot;max growth&quot;) +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = &quot;top&quot;) +
  # labs(x=expression(italic(&quot;S. thermophilus&quot;)),y=expression(italic(&quot;L. delbrueckii&quot;)))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)


png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms_new.png&quot;, width = 2500, height = 3000,res=300)

   # pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf&quot;,width=6,height=8)

  print(ggheatmap)

dev.off()

##-------------------------------------------
#growthcurver
##-------------------------------------------
library(growthcurver)

cleaned_biolog_file_prep &lt;- cleaned_biolog_file %&gt;% select(Time,Well_cleaned,Strain_Name,Value)

cleaned_biolog_file_prep$Well_cleaned &lt;- plyr::revalue(cleaned_biolog_file_prep$Well_cleaned, c(&quot;Negative Control&quot;=&quot;blank&quot;))


growthcurves_strains &lt;- data.frame()
unique(cleaned_biolog_file_prep$Strain_Name)
strainss &lt;- &quot;24742&quot;
for (strainss in unique(cleaned_biolog_file_prep$Strain_Name)) {
  
  tmps &lt;- cleaned_biolog_file_prep %&gt;% filter(Strain_Name==strainss)
  
  tmps_wide &lt;- spread(tmps, Well_cleaned, Value) %&gt;% select(-Strain_Name)

  # gc_out &lt;- SummarizeGrowthByPlate(tmps_wide)

  gc_out &lt;- SummarizeGrowthByPlate(tmps_wide)
  # gc_out &lt;- SummarizeGrowthByPlate(tmps_wide, bg_correct = &quot;blank&quot;)

  
  gc_out_filtered &lt;- gc_out %&gt;% filter(note == &quot;&quot;)  %&gt;% filter(sample!= &quot;&quot;) %&gt;% add_column(Strain_Name=strainss) %&gt;% add_column(replicate=replicatess)
  # hist(gc_out_filtered$sigma, main = &quot;Histogram of sigma values&quot;, xlab = &quot;sigma&quot;)
growthcurves_strains &lt;- rbind(growthcurves_strains,gc_out_filtered)
  
}



table(growthcurves_strains$Strain_Name)

# growthcurves_strains_sub &lt;- growthcurves_strains 

growthcurves_strains_sub &lt;- growthcurves_strains %&gt;% filter(!Strain_Name %in% c(&quot;24746&quot;,&quot;24750&quot;,&quot;24743&quot;,&quot;24749&quot;))

    dev.off()
range(growthcurves_strains$r)
ggheatmap &lt;- ggplot(growthcurves_strains_sub, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = &quot;white&quot;,size=1.1)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;red&quot;, mid = &quot;white&quot;, space = &quot;Lab&quot;,
   name=&quot;max growth&quot;) +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = &quot;top&quot;) +
  # labs(x=expression(italic(&quot;S. thermophilus&quot;)),y=expression(italic(&quot;L. delbrueckii&quot;)))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)
##-------------------------------
#only Streptococcus
##-------------------------------


growthcurves_strains_sub_strepto &lt;- growthcurves_strains_sub %&gt;% filter(Strain_Name %in% c(&quot;24736&quot;,&quot;24735&quot;,&quot;24734&quot;,&quot;24750&quot;,&quot;24739&quot;,&quot;24745&quot;,&quot;24728&quot;,&quot;24742&quot;,&quot;24746&quot;,&quot;24749&quot;,&quot;24730&quot;,&quot;24743&quot;,&quot;S_sal_48&quot;))


growthcurves_strains_sub_strepto$Strain_Name &lt;- plyr::revalue(growthcurves_strains_sub_strepto$Strain_Name, c(&quot;24736&quot;=&quot;lineage 1&quot;,&quot;24735&quot;=&quot;lineage 2&quot;,&quot;24734&quot;=&quot;lineage 3&quot;,&quot;24750&quot;=&quot;lineage 4&quot;,&quot;24739&quot;=&quot;lineage 5&quot;,&quot;24745&quot;=&quot;lineage 6&quot;,&quot;24728&quot;=&quot;lineage 7&quot;,&quot;24742&quot;=&quot;lineage 8&quot;,&quot;24746&quot;=&quot;lineage 9&quot;,&quot;24749&quot;=&quot;lineage 10&quot;,&quot;24730&quot;=&quot;lineage 11&quot;,&quot;24743&quot;=&quot;lineage 12&quot;,&quot;S_sal_48&quot;=&quot;S. salivarius&quot;))


growthcurves_strains_sub_strepto$Strain_Name  = factor(growthcurves_strains_sub_strepto$Strain_Name , levels=c(&quot;lineage 1&quot;,&quot;lineage 2&quot;,&quot;lineage 3&quot;,&quot;lineage 4&quot;,&quot;lineage 5&quot;,&quot;lineage 6&quot;,&quot;lineage 7&quot;,&quot;lineage 8&quot;,&quot;lineage 9&quot;,&quot;lineage 10&quot;,&quot;lineage 11&quot;,&quot;lineage 12&quot;,&quot;S. salivarius&quot;))

    dev.off()
range(growthcurves_strains$r)
ggheatmap &lt;- ggplot(growthcurves_strains_sub_strepto, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = &quot;white&quot;,size=1.1)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;red&quot;, mid = &quot;white&quot;, space = &quot;Lab&quot;,
   name=&quot;max growth&quot;) +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = &quot;top&quot;) +
  # labs(x=expression(italic(&quot;S. thermophilus&quot;)),y=expression(italic(&quot;L. delbrueckii&quot;)))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)

##------------------------------order correctly


growthcurves_strains_sub_strepto


sorting_vector &lt;- growthcurves_strains_sub_strepto %&gt;% group_by(sample) %&gt;% summarize(final=sum(r)) %&gt;% arrange(final) %&gt;% dplyr::pull(., sample)

growthcurves_strains_sub_strepto$sample = factor(growthcurves_strains_sub_strepto$sample, levels=sorting_vector)

    dev.off()
range(growthcurves_strains$r)
ggheatmap &lt;- ggplot(growthcurves_strains_sub_strepto, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = &quot;white&quot;,size=1.1)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;red&quot;, mid = &quot;white&quot;, 
                     # midpoint = 0.0, limit = c(0,4),
                     space = &quot;Lab&quot;,

   name=&quot;max growth&quot;) +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = &quot;top&quot;) +
  # labs(x=expression(italic(&quot;S. thermophilus&quot;)),y=expression(italic(&quot;L. delbrueckii&quot;)))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)


png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms_new_01.png&quot;, width = 1800, height = 3000,res=300)

   # pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf&quot;,width=6,height=8)

  print(ggheatmap)

dev.off()


###--------------sub

ggplot(growthcurves_strains_sub_strepto,aes(x=r))+geom_density()+geom_vline(xintercept = 0.1)

growthcurves_strains_sub_strepto_sub &lt;- growthcurves_strains_sub_strepto %&gt;% filter(r&gt;0.1)

  dev.off()

ggheatmap_strepto &lt;- ggplot(growthcurves_strains_sub_strepto_sub, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = &quot;white&quot;,size=1.1)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;red&quot;, mid = &quot;white&quot;, 
                     limit = c(0,1.75),
                     space = &quot;Lab&quot;,

   name=&quot;max growth&quot;) +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = &quot;top&quot;) +
  # labs(x=expression(italic(&quot;S. thermophilus&quot;)),y=expression(italic(&quot;L. delbrueckii&quot;)))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap_strepto)


png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms_new_01.png&quot;, width = 1800, height = 3000,res=300)

   # pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf&quot;,width=6,height=8)

  print(ggheatmap_strepto)

dev.off()


growthcurves_strains_sub_strepto_sub %&gt;% group_by(Strain_Name) %&gt;% summarise(counts=n()) %&gt;% filter(Strain_Name !=&quot;S. salivarius&quot;) %&gt;% summarise(mean=mean(counts))
# growthcurves_strains_sub_strepto

growthcurves_strains_sub_strepto_sub %&gt;% group_by(Strain_Name) %&gt;% summarise(counts=n())

##-------------------------------
#only lactobacillus
##-------------------------------


growthcurves_strains_sub_lacto &lt;- growthcurves_strains_sub %&gt;% filter(!Strain_Name %in% c(&quot;24736&quot;,&quot;24735&quot;,&quot;24734&quot;,&quot;24750&quot;,&quot;24739&quot;,&quot;24745&quot;,&quot;24728&quot;,&quot;24742&quot;,&quot;24746&quot;,&quot;24749&quot;,&quot;24730&quot;,&quot;24743&quot;,&quot;S_sal_48&quot;))


growthcurves_strains_sub_lacto$Strain_Name &lt;- plyr::revalue(growthcurves_strains_sub_lacto$Strain_Name, c(&quot;24790&quot;=&quot;lineage 13&quot;,&quot;24793&quot;=&quot;lineage 14&quot;))


growthcurves_strains_sub_lacto$Strain_Name  = factor(growthcurves_strains_sub_lacto$Strain_Name , levels=c(&quot;lineage 13&quot;,&quot;lineage 14&quot;,&quot;L_bul_50&quot;,&quot;l_sunkii_45&quot;,&quot;L_indus_23&quot;,&quot;L_equi_23349&quot;))

    dev.off()
range(growthcurves_strains$r)
ggheatmap &lt;- ggplot(growthcurves_strains_sub_lacto, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = &quot;white&quot;,size=1.1)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;red&quot;, mid = &quot;white&quot;, space = &quot;Lab&quot;,
   name=&quot;max growth&quot;) +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = &quot;top&quot;) +
  # labs(x=expression(italic(&quot;S. thermophilus&quot;)),y=expression(italic(&quot;L. delbrueckii&quot;)))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)

##------------------------------order correctly


growthcurves_strains_sub_lacto


sorting_vector &lt;- growthcurves_strains_sub_lacto %&gt;% group_by(sample) %&gt;% summarize(final=sum(r)) %&gt;% arrange(final) %&gt;% dplyr::pull(., sample)

growthcurves_strains_sub_lacto$sample = factor(growthcurves_strains_sub_lacto$sample, levels=sorting_vector)

    dev.off()
range(growthcurves_strains$r)
ggheatmap &lt;- ggplot(growthcurves_strains_sub_lacto, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = &quot;white&quot;,size=1.1)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;red&quot;, mid = &quot;white&quot;, 
                     # midpoint = 0.0, limit = c(0,4),
                     space = &quot;Lab&quot;,

   name=&quot;max growth&quot;) +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = &quot;top&quot;) +
  # labs(x=expression(italic(&quot;S. thermophilus&quot;)),y=expression(italic(&quot;L. delbrueckii&quot;)))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)


png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_lacto_new_01.png&quot;, width = 1800, height = 3000,res=300)

   # pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf&quot;,width=6,height=8)

  print(ggheatmap)

dev.off()


###--------------sub

ggplot(growthcurves_strains_sub_lacto,aes(x=r))+geom_density()+geom_vline(xintercept = 0.1)

growthcurves_strains_sub_lacto_sub &lt;- growthcurves_strains_sub_lacto %&gt;% filter(r&gt;0.1) %&gt;% filter(Strain_Name %in% c(&quot;lineage 13&quot;,&quot;L_equi_23349&quot;))


# growthcurves_strains_sub_lacto &lt;- growthcurves_strains_sub %&gt;% filter(!Strain_Name %in% c(&quot;24736&quot;,&quot;24735&quot;,&quot;24734&quot;,&quot;24750&quot;,&quot;24739&quot;,&quot;24745&quot;,&quot;24728&quot;,&quot;24742&quot;,&quot;24746&quot;,&quot;24749&quot;,&quot;24730&quot;,&quot;24743&quot;,&quot;S_sal_48&quot;))



  dev.off()

ggheatmap_lacto &lt;- ggplot(growthcurves_strains_sub_lacto_sub, aes(y=sample, x=Strain_Name, fill = r))+
 geom_tile(color = &quot;white&quot;,size=1.1)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;red&quot;, mid = &quot;white&quot;, 
                     limit = c(0,1.75),
                     space = &quot;Lab&quot;,

   name=&quot;max growth&quot;) +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = &quot;top&quot;) +
  # labs(x=expression(italic(&quot;S. thermophilus&quot;)),y=expression(italic(&quot;L. delbrueckii&quot;)))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap_lacto)


png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_lacto_new_01.png&quot;, width = 1800, height = 3000,res=300)

   # pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf&quot;,width=6,height=8)

  print(ggheatmap_lacto)

dev.off()

growthcurves_strains_sub_lacto_sub %&gt;% group_by(Strain_Name) %&gt;% summarise(counts=n())

61-9</code></pre>
<p>old way</p>
<p>Here, I analyse the Biolog data of the different strains.</p>
<pre class="r"><code># library(readr)
# library(ggplot2)
# library(plyr)
library(tidyverse)
library(opm)
library(patchwork)
Biolog_all_plates &lt;- read_delim(&quot;~/Desktop/Projects/2019_RMK202_analysis/Biolog/strains/Export_All_Plates_Single_File.csv&quot;, &quot;;&quot;, escape_double = FALSE, trim_ws = TRUE)
Biolog_all_plates$`Field 2`


log_pm1_final &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/Biolog/log_pm1_final.csv&quot;, &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;well&quot;,&quot;metabolite&quot;),   trim_ws = TRUE)

ggplot(Biolog_all_plates,aes(x=Hour,y=A06,group=`Field 2`,color=`Field 2`))+geom_line()

       
Biolog_long &lt;- gather(Biolog_all_plates, well, intensity, A01:H12, factor_key=TRUE,na.rm = TRUE) 
Biolog_long$intensity &lt;- as.double(Biolog_long$intensity)
Biolog_long$Hour &lt;- as.double(Biolog_long$Hour)

str(Biolog_long)
p1 &lt;- ggplot(Biolog_long,aes(x=Hour,y=intensity,group=well,color=well))+geom_line()+facet_wrap(`Field 2`~.)+theme_classic()
p1
png(&quot;~/Desktop/Projects/2019_RMK202_analysis/biolog/plots/Biolog_sterm_rmk202_strains_01.png&quot;, width = 4000, height = 4000,res=300)
 # svg(&quot;~/Desktop/Projects/2019_RMK202_analysis/10_assembly/heatmap_all_Sterm_strains.svg&quot;,width=14,height=14)
p1
dev.off()

##---------------------------------------------
##growth (binary)
##calculate different of highest and lowest intensity value
##---------------------------------------------
Biolog_diff &lt;- aggregate(. ~`Field 2`+well, data=Biolog_long[,c(&quot;Field 2&quot;,&quot;well&quot;,&quot;intensity&quot;)], FUN = function(i)max(i) - min(i))
hist(Biolog_diff$intensity,breaks = 20)
pDensity &lt;- ggplot(Biolog_diff,aes(x=intensity,group=`Field 2`,color=`Field 2`,fill=`Field 2`))+geom_density()+theme_classic()+facet_wrap(`Field 2`~.)+geom_vline(xintercept =90)+labs(x=&quot;Intensity difference between maximal and minimal value&quot;)+theme(legend.position = &quot;none&quot;)

# Biolog_diff_high &lt;- Biolog_diff %&gt;% filter(intensity&gt;90)  #subset dataset
# table(Biolog_diff$well)
Biolog_diff_high &lt;- Biolog_diff
Biolog_diff_high$well &lt;- droplevels(Biolog_diff_high$well)
Biolog_diff_high$`Field 2` &lt;- droplevels(Biolog_diff_high$`Field 2`)
table(Biolog_diff_high$well,Biolog_diff_high$`Field 2`)

Biolog_diff %&gt;%  filter(well==&quot;C09&quot;)

# Biolog_diff_high_ext &lt;- Biolog_diff %&gt;% filter(well %in% c(as.character(unique(Biolog_diff_high$well)),&quot;A06&quot;)) #subset dataset
# Biolog_diff_high_ext &lt;- Biolog_diff  #%&gt;% filter(well %in% c(as.character(unique(Biolog_diff_high$well)),&quot;A06&quot;))
Biolog_diff_high_ext &lt;- merge(Biolog_diff,log_pm1_final,by=&quot;well&quot;) %&gt;% mutate(well=metabolite) %&gt;% select(-metabolite)  #%&gt;% filter(well %in% c(as.character(unique(Biolog_diff_high$well)),&quot;A06&quot;))
table(Biolog_diff$well)
table(Biolog_diff_high_ext$well)
unique(Biolog_diff_high_ext$`Field 2`)
Biolog_diff_high_ext$`Field 2` &lt;- revalue(Biolog_diff_high_ext$`Field 2`, c( &quot;Stamm 37&quot;=&quot;S24737&quot;, &quot;Stamm 38&quot;=&quot;S24738&quot;, &quot;Stamm 39&quot;=&quot;S24739&quot;, &quot;Stamm 40&quot;=&quot;S24740&quot; ,&quot;Stamm new 1&quot;=&quot;24854&quot;, &quot;Stamm new 2&quot;=&quot;24855&quot;))#subset dataset




sorting_vector &lt;- Biolog_diff_high_ext %&gt;% group_by(well) %&gt;% summarize(final=sum(intensity)) %&gt;% arrange(final) %&gt;% dplyr::pull(., well)

Biolog_diff_high_ext$well = factor(Biolog_diff_high_ext$well, levels=sorting_vector)



##---------------------------------------------
##make heatmap of strain growth 
##only use the carbohydrates that have at least one strain that grows
##---------------------------------------------


# Biolog_diff_wide &lt;- spread(Biolog_diff, `Field 2`, intensity) %&gt;% column_to_rownames(var=&quot;well&quot;)
Biolog_diff_high_wide &lt;- spread(Biolog_diff_high, `Field 2`, intensity) %&gt;% column_to_rownames(var=&quot;well&quot;)
Biolog_diff_high_wide_ext &lt;- spread(Biolog_diff_high_ext, `Field 2`, intensity) %&gt;% column_to_rownames(var=&quot;well&quot;)

# rownames(Biolog_diff_high_wide_ext) &lt;- revalue(rownames(Biolog_diff_high_wide_ext), c( &quot;A02&quot;=&quot;L-Arabinose&quot;, &quot;B08&quot;=&quot;D-Xylose&quot;, &quot;B11&quot;=&quot;D-Mannitol&quot;, &quot;C04&quot;=&quot;D-Ribose&quot; ,&quot;C07&quot;=&quot;D-Fructose&quot;, &quot;C09&quot;=&quot;alpha-D-Glucose&quot;, &quot;D09&quot;=&quot;alpha-D-Lactose&quot; ,&quot;D10&quot;=&quot;Lactulose&quot;, &quot;D11&quot;=&quot;Sucrose&quot;, &quot;G08&quot;=&quot;N-Acetyl-Beta-D-Mannosamine&quot; ,&quot;H06&quot;=&quot;L-Lyxose&quot;,&quot;A06&quot;=&quot;D-Galactose&quot;))#subset dataset




# rownames(Biolog_diff_high_wide_ext) &lt;- revalue(rownames(Biolog_diff_high_wide_ext), c( &quot;A02&quot;=&quot;L-Arabinose&quot;, &quot;B08&quot;=&quot;D-Xylose&quot;, &quot;B11&quot;=&quot;D-Mannitol&quot;, &quot;C04&quot;=&quot;D-Ribose&quot; ,&quot;C07&quot;=&quot;D-Fructose&quot;, &quot;C09&quot;=&quot;alpha-D-Glucose&quot;, &quot;D09&quot;=&quot;alpha-D-Lactose&quot; ,&quot;D10&quot;=&quot;Lactulose&quot;, &quot;D11&quot;=&quot;Sucrose&quot;, &quot;G08&quot;=&quot;N-Acetyl-Beta-D-Mannosamine&quot; ,&quot;H06&quot;=&quot;L-Lyxose&quot;,&quot;A06&quot;=&quot;D-Galactose&quot;,&quot;A01&quot;=&quot;Negative control&quot;,&quot;A03&quot;=&quot;N-acetyl-d-glucosamine&quot;,&quot;A04&quot;=&quot;D-saccharic acid&quot;,&quot;A05&quot;=&quot;Succinic acid&quot;,&quot;A07&quot;=&quot;L-aspartic acid&quot;,&quot;A08&quot;=&quot;L-proline&quot;,&quot;A09&quot;=&quot;D-alanine&quot;,&quot;A10&quot;=&quot;D-trehalose&quot;,&quot;A11&quot;=&quot;D-mannose&quot;,&quot;A12&quot;=&quot;Dulcitol&quot;,&quot;B01&quot;=&quot;D-serine&quot;,&quot;B02&quot;=&quot;D-sorbitol&quot;,&quot;B03&quot;=&quot;Glycerol&quot;,&quot;B04&quot;=&quot;L-fucose&quot;,&quot;B05&quot;=&quot;D-Glucuronic acid&quot;,&quot;B06&quot;=&quot;D-gluconic acid&quot;,&quot;B07&quot;=&quot;D,L-alpha-glycerol-phosphate&quot;,&quot;B08&quot;=&quot;D-xylose&quot;,&quot;B09&quot;=&quot;D-xylose&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;,&quot;XX&quot;=&quot;YYYYYYY&quot;))


# rownames(Biolog_diff_high_wide_ext)

if (!require(&quot;gplots&quot;)) {
   install.packages(&quot;gplots&quot;, dependencies = TRUE)
   library(gplots)
   }
if (!require(&quot;RColorBrewer&quot;)) {
   install.packages(&quot;RColorBrewer&quot;, dependencies = TRUE)
   library(RColorBrewer)
   }


# mat_data &lt;- as.matrix(Biolog_diff_wide) # transform column 2-5 into a matrix
# mat_data &lt;- as.matrix(Biolog_diff_high_wide) # transform column 2-5 into a matrix
# mat_data &lt;- as.matrix(Biolog_diff_high_wide_ext) # transform column 2-5 into a matrix
# 
# my_palette &lt;- colorRampPalette(c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;))(n = 119)
# # col_breaks = c(seq(0,max(mat_data),length=300))
# # length(col_breaks)
# col_breaks = c(seq(min(mat_data),60,length=40),  # for red
#   seq(61,90,length=40),           # for yellow
#   seq(91,max(mat_data),length=40))
# 
# str(mat_data)
# dev.off()
#     # png(&quot;~/Desktop/Projects/2019_RMK202&lt;_analysis/10_assembly/heatmap_all_Sterm_strains.png&quot;, width = 4000, height = 4000,res=300)
#  # svg(&quot;~/Desktop/Projects/2019_RMK202_analysis/10_assembly/heatmap_all_Sterm_strains.svg&quot;,width=14,height=14)
# pHeatmap &lt;- heatmap.2(mat_data,
#   # cellnote = mat_data,  # same data set for cell labels
#   main = &quot;&quot;, # heat map title
#   notecol=&quot;black&quot;,      # change font color of cell labels to black
#   density.info=&quot;none&quot;,
#   trace=&quot;none&quot;,         # turns off trace lines inside the heat map
#  margins = c(16, 16),     # widens margins around plot
#   col=my_palette,   
#   breaks=col_breaks,    # enable color transition at specified limits
#   sepcolor=&quot;black&quot;,
#   sepwidth=c(0.5,0.5),
#   dendrogram=&quot;none&quot;,
#  Colv=FALSE)
# 
# svg(&quot;~/Desktop/Projects/2019_RMK202_analysis/biolog/plots/DenstiyPlot_biolog_02.svg&quot;,width=8,height=8)
# heatmap.2(mat_data,
#   # cellnote = mat_data,  # same data set for cell labels
#   main = &quot;&quot;, # heat map title
#   notecol=&quot;black&quot;,      # change font color of cell labels to black
#   density.info=&quot;none&quot;,
#   trace=&quot;none&quot;,         # turns off trace lines inside the heat map
#  margins = c(16, 16),     # widens margins around plot
#   col=my_palette,   
#   breaks=col_breaks,    # enable color transition at specified limits
#   sepcolor=&quot;black&quot;,
#   sepwidth=c(0.5,0.5),
#   dendrogram=&quot;none&quot;,
#  Colv=FALSE)
# dev.off()
# # dev.off()
# 
#   
#   svg(&quot;~/Desktop/Projects/2019_RMK202_analysis/biolog/plots/DenstiyPlot_biolog_01.svg&quot;,width=5,height=5)
#    pDensity
#   dev.off()

  
  ###----------------------------------
  ##all data
  ###----------------------------------
  

    dev.off()
range(Biolog_diff_high_ext$intensity)
ggheatmap &lt;- ggplot(Biolog_diff_high_ext, aes(y=well, x=`Field 2`, fill = intensity))+
 geom_tile(color = &quot;white&quot;,size=1.1)+
 scale_fill_gradient2(low = &quot;white&quot;, high = &quot;red&quot;, mid = &quot;white&quot;,
   midpoint = 100, limit = c(0,232), space = &quot;Lab&quot;,
   name=&quot;max growth&quot;) +
  theme_minimal()+ # minimal theme
  scale_x_discrete(position = &quot;top&quot;) +
  # labs(x=expression(italic(&quot;S. thermophilus&quot;)),y=expression(italic(&quot;L. delbrueckii&quot;)))+
 theme(axis.text.y = element_text(size=7),
      axis.text.x = element_text(angle = 45,vjust=1,
    size = 12, hjust = 0),axis.title=element_blank())
 # coord_fixed()
# Print the heatmap
print(ggheatmap)
  
  


   pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/biolog_Sterms.pdf&quot;,width=6,height=8)

  print(ggheatmap)

dev.off()</code></pre>
</div>
</div>
</div>
<div id="rna-seq" class="section level1">
<h1>RNA-seq</h1>
<p>###trimgalore</p>
<p>In the following we clip adaptors with <a href>trim-galore</a></p>
<pre class="bash"><code>threads=30

for id in $(cut -f 1 /home/vincent/Projects/2019_RNAseq_rmk202/02_script/names_samples.txt)
  do
  echo &quot;##==============&quot;
  echo ${id}
  echo &quot;##==============&quot;
      rm -r /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/trimGalore/${id}
      mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/trimGalore/${id}
      trim_galore -o /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/trimGalore/${id}/ \
        /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/raw/gz_merged/${id}_R1.fastq.gz
        
  echo &quot;##==============&quot;
  echo ${id} &quot;fastQC of trimmed&quot;
  echo &quot;##==============&quot;
  rm -r /home/vincent/Projects/2019_RNAseq_rmk202/02_QCs/02_trimGaloreQC_01/$id/
  mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/02_QCs/02_trimGaloreQC_01/$id/
      
    fastqc /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/trimGalore/${id}/${id}*.fq.gz -t 37 -o /archiv/Projects/2019_RNAseq_rmk202/02_QCs/02_trimGaloreQC_01/$id/
    
        
  done 
    
  mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/02_QCs/02_trimGaloreQC_01/MultiQC
  multiqc /home/vincent/Projects/2019_RNAseq_rmk202/02_QCs/02_trimGaloreQC_01/ -o /archiv/Projects/2019_RNAseq_rmk202/02_QCs/02_trimGaloreQC_01/MultiQC
</code></pre>
<p>##sortmeRNA well explained <a href="https://github.com/biomendi/TRANSCRIPTOME-ASSEMBLY-PIPELINE/wiki/3.-Ribosomal-RNA-removal-using-SORTMERNA">here</a></p>
<pre class="bash"><code>
##index
/home/vincent/Joanito/brain/Software/sortmerna-2.1/indexdb_rna --ref \
/home/vincent/Joanito/brain/Software/sortmerna-2.1/rRNA_databases/silva-bac-16s-id90.fasta,/home/vincent/Joanito/brain/Software/sortmerna-2.1/index/silva-bac-16s-db:\
/home/vincent/Joanito/brain/Software/sortmerna-2.1/rRNA_databases/silva-bac-23s-id98.fasta,/home/vincent/Joanito/brain/Software/sortmerna-2.1/index/silva-bac-23s-db:\
/home/vincent/Joanito/brain/Software/sortmerna-2.1/rRNA_databases/silva-arc-16s-id95.fasta,/home/vincent/Joanito/brain/Software/sortmerna-2.1/index/silva-arc-16s-db:\
/home/vincent/Joanito/brain/Software/sortmerna-2.1/rRNA_databases/silva-arc-23s-id98.fasta,/home/vincent/Joanito/brain/Software/sortmerna-2.1/index/silva-arc-23s-db:\
/home/vincent/Joanito/brain/Software/sortmerna-2.1/rRNA_databases/silva-euk-18s-id95.fasta,/home/vincent/Joanito/brain/Software/sortmerna-2.1/index/silva-euk-18s-db:\
/home/vincent/Joanito/brain/Software/sortmerna-2.1/rRNA_databases/silva-euk-28s-id98.fasta,/home/vincent/Joanito/brain/Software/sortmerna-2.1/index/silva-euk-28s:\
/home/vincent/Joanito/brain/Software/sortmerna-2.1/rRNA_databases/rfam-5s-database-id98.fasta,/home/vincent/Joanito/brain/Software/sortmerna-2.1/index/rfam-5s-db:\
/home/vincent/Joanito/brain/Software/sortmerna-2.1/rRNA_databases/rfam-5.8s-database-id98.fasta,/home/vincent/Joanito/brain/Software/sortmerna-2.1/index/rfam-5.8s-db



##filter
#rm -r /archiv/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA
#mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/${id}
rm //home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/multiqc_logfile.txt
for id in $(cut -f 1 /home/vincent/Projects/2019_RNAseq_rmk202/02_script/names_samples.txt)
  do
  echo &quot;##==============&quot;
  echo ${id}
  echo &quot;##==============&quot;
  gunzip -c /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/trimGalore/${id}/${id}_R1_trimmed.fq.gz &gt; /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/trimGalore/${id}/${id}_R1_trimmed.fq
  #gunzip -c /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/trimGalore/${id}/${id}_R1_trimmed.fq.gz &gt; /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/trimGalore/${id}/${id}_R1_trimmed.fq

  #cd /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/trimGalore/${id}/
  # bash /home/vincent/Joanito/brain/Software/sortmerna-2.1/scripts/merge-paired-reads.sh ${id}_R1_trimmed.fq ${id}_R2_trimmed.fq ${id}_merged_reads.fq
  
  rm -r /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/${id}
  mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/${id}
sortmerna --ref \
/home/vincent/Joanito/brain/Software/sortmerna-2.1/rRNA_databases/silva-bac-16s-id90.fasta,/home/vincent/Joanito/brain/Software/sortmerna-2.1/index/silva-bac-16s-db:\
/home/vincent/Joanito/brain/Software/sortmerna-2.1/rRNA_databases/silva-bac-23s-id98.fasta,/home/vincent/Joanito/brain/Software/sortmerna-2.1/index/silva-bac-23s-db:\
/home/vincent/Joanito/brain/Software/sortmerna-2.1/rRNA_databases/rfam-5s-database-id98.fasta,/home/vincent/Joanito/brain/Software/sortmerna-2.1/index/rfam-5s-db:\
/home/vincent/Joanito/brain/Software/sortmerna-2.1/rRNA_databases/rfam-5.8s-database-id98.fasta,/home/vincent/Joanito/brain/Software/sortmerna-2.1/index/rfam-5.8s-db \
--reads /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/trimGalore/${id}/${id}_R1_trimmed.fq --num_alignments 1 \
--fastx --aligned /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/${id}/${id}_rRNA --other /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/${id}/${id}_non_rRNA --log -a 32 -m 63784 -v

rm /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/trimGalore/${id}/${id}_R1_trimmed.fq &amp;
#rm /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/trimGalore/${id}/${id}_R2_trimmed.fq &amp;

#bash ./unmerge-paired-reads.sh /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/${id}_non_rRNA.fastq /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/${id}_non_rRNA_R1.fastq /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/${id}_non_rRNA_R2.fastq

echo &quot;/home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/&quot;${id}&quot;_rRNA.log&quot; &gt;&gt; /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/multiqc_logfile.txt
   
  done
  
  for id in $(cut -f 1 /home/vincent/Projects/2019_RNAseq_rmk202/02_script/names_samples.txt)
  do
  echo &quot;##==============&quot;
  echo ${id}
  grep -A 1 &quot;Total reads passing E-value threshold&quot; /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/${id}_rRNA.log
  done
  
  mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/multiQC
  

  multiqc --config /home/vincent/logs/multiQC_config.txt /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/ \
  -o /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/multiQC/

cd /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/
multiqc -d /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/ -o /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/multiQC/


  multiqc -s  --module sortmeRNA --file-list /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/multiqc_logfile.txt \
  -o /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/multiQC/


scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/multiQC/ ~/Desktop/Projects/2019_RNAseq_rmk202/01_log/
</code></pre>
<p>##mapping</p>
<pre class="bash"><code>


##============
##mapping to reference
##============

# name_folder=02_againstpolished_single_K1_final_kneaddata_withStrains 

threads=37

mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT


 for names in $(cut -f 1 /home/vincent/Projects/2019_RNAseq_rmk202/02_script/names_samples.txt)
  do
  echo ${num}&quot;/16  :&quot; ${names}
  num=$((num+1))
  rm -r /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/bwaMapping2DB/

  mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/bwaMapping2DB/
  
 # bwa mem -t ${threads} /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_merged.fasta
  bwa mem -t ${threads} /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_merged.fasta \
  /home/vincent/Projects/2019_RNAseq_rmk202/01_Data/03_sortmeRNA/${names}/${names}_non_rRNA.fq | samtools sort -@${threads} -O BAM -o /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

#samtools view -b -f 4 /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam &gt; /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

#bedtools bamtofastq -i /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

#rm /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

##-----Qualimap

mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/bwaMapping2DB//${names}_rawIllumina_bwamem_sorted.bam -outdir /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT//log
echo &quot;/home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/&quot;${names}&quot;/bwaMapping2DB//bamqc&quot; &gt;&gt; /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/log/multiqc_logfile_finalGenome.txt
   done 
   
   
    rm -r /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/MultiQC
    
  mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/MultiQC
  
  multiqc --config /home/vincent/logs/multiQC_config.txt --file-list  /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/log/multiqc_logfile_finalGenome.txt \
    -o /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/MultiQC
</code></pre>
<p>##HT-seq</p>
<pre class="bash"><code>

threads=37



##----------------------------------------------
##--------------count events
##----------------------------------------------

 for names in $(cut -f 1 /home/vincent/Projects/2019_RNAseq_rmk202/02_script/names_samples.txt)
  do
  echo ${num}&quot;/16  :&quot; ${names}
  num=$((num+1))
 rm -r /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/

mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/
  cd /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/
  htseq-count --stranded no -i ID --format bam --type gene /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam \
    /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap.gff &gt; /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/sterm_${names}_count.txt

  htseq-count --stranded no -i ID --format bam --type gene /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam \
    /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap.gff &gt; /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/ldel_${names}_count.txt

   done 

##----------------------------------------------
##--------------rename genes
##----------------------------------------------

 for names in $(cut -f 1 /home/vincent/Projects/2019_RNAseq_rmk202/02_script/names_samples.txt)
  do
  echo ${num}&quot;/16  :&quot; ${names}
  num=$((num+1))
#sed -i &#39;s/gene-pgaptmp_/L_delbrueckii_RMK202_pgaptmp_/g&#39; /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/ldel_${names}_count.txt
#sed -i &#39;s/gene-pgaptmp_/S_thermophilus_RMK202_pgaptmp_/g&#39; /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/sterm_${names}_count.txt
echo &quot;==Ldel&quot;
grep &quot;__no_feature&quot; /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/ldel_${names}_count.txt
echo &quot;==Sterm&quot;
grep &quot;__no_feature&quot; /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/sterm_${names}_count.txt


 done
 

##----------------------------------------------
##--------------move in common folder
##----------------------------------------------
rm -r /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/htSeq_all
mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/htSeq_all

 for names in $(cut -f 1 /home/vincent/Projects/2019_RNAseq_rmk202/02_script/names_samples.txt)
  do
  echo ${num}&quot;/16  :&quot; ${names}
  num=$((num+1))

cp /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/ldel_${names}_count.txt /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/htSeq_all/ldel_${names}_count.txt

cp /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/sterm_${names}_count.txt /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/htSeq_all/sterm_${names}_count.txt


 done 

##----------------------------------------------
##--------------bring local
##----------------------------------------------

scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/htSeq_all/ ~/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/


 </code></pre>
<p>##normalize count data</p>
<p>Here we normalize the count data with DeSeq2 median of ratios
normalization as descripted <a href="https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html">here</a>
and <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">here</a>.</p>
<pre class="r"><code>library(GenomicDataCommons)
library(DESeq2)
library(dplyr)
library(readr)

##====================================================================================Lactobacillus delbrueckii===============================================

directory &lt;- &quot;~/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/htSeq_all/&quot;
sampleFiles &lt;- grep(&quot;ldel&quot;,list.files(directory),value=TRUE)
# sampleCondition &lt;- sub(&quot;_&quot;,&quot;\\1&quot;,sampleFiles)
sampleCondition &lt;- str_split_fixed(sampleFiles, &quot;_&quot;, 5)[,2]


sampleTable &lt;- data.frame(sampleName = sampleFiles,
                          fileName = sampleFiles,
                          condition = sampleCondition)

ddsHTSeq &lt;- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
                                       directory = directory,
                                       design= ~ condition)
# ddsHTSeq

##---------------
##Pre-filtering (keep only rows that have at least 10 reads total)
##---------------

keep &lt;- rowSums(counts(ddsHTSeq)) &gt;= 10
dds &lt;- ddsHTSeq[keep,]
# View(counts(dds))

dds &lt;- estimateSizeFactors(dds)
# sizeFactors(dds)
##---------------
##DESeq2’s median of ratios normalization accounts for sequencing depth and RNA composition it is calculated as following :counts divided by sample-specific size factors determined by median ratio of gene counts relative to geometric mean per gene
##---------------

# dds &lt;- DESeq(ddsHTSeq)
# View(counts(dds))
normalized_counts &lt;- counts(dds, normalized=TRUE) %&gt;% as.data.frame()
normalized_counts &lt;- tibble::rownames_to_column(normalized_counts, &quot;gene&quot;)
write.table(normalized_counts, file=&quot;~/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/normalized_counts_ldel.txt&quot;, sep=&quot;\t&quot;, quote=F, row.names = FALSE,col.names=TRUE)

##====================================================================================Streptococcus thermophilus===============================================


directory &lt;- &quot;~/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/htSeq_all/&quot;
sampleFiles &lt;- grep(&quot;sterm&quot;,list.files(directory),value=TRUE)
sampleCondition &lt;- str_split_fixed(sampleFiles, &quot;_&quot;, 5)[,2]


sampleTable &lt;- data.frame(sampleName = sampleFiles,
                          fileName = sampleFiles,
                          condition = sampleCondition)

ddsHTSeq &lt;- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
                                       directory = directory,
                                       design= ~ condition)

##---------------
##Pre-filtering (keep only rows that have at least 10 reads total)
##---------------

keep &lt;- rowSums(counts(ddsHTSeq)) &gt;= 10
dds &lt;- ddsHTSeq[keep,]
# View(counts(dds))

dds &lt;- estimateSizeFactors(dds)
# sizeFactors(dds)
##---------------
##DESeq2’s median of ratios normalization accounts for sequencing depth and RNA composition it is calculated as following :counts divided by sample-specific size factors determined by median ratio of gene counts relative to geometric mean per gene
##---------------

# dds &lt;- DESeq(ddsHTSeq)
# View(counts(dds))
normalized_counts &lt;- counts(dds, normalized=TRUE) %&gt;% as.data.frame()
normalized_counts &lt;- tibble::rownames_to_column(normalized_counts, &quot;gene&quot;)
write.table(normalized_counts, file=&quot;~/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/normalized_counts_sterm.txt&quot;, sep=&quot;\t&quot;, quote=F, row.names = FALSE,col.names=TRUE)
library(ggplot2)

# merge_RNAseq_normalized &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/normalized_counts_merged.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)
# 
# merge_RNAseq_normalized$norm_count &lt;- merge_RNAseq_normalized$norm_count+0.0001
# table(merge_RNAseq_normalized$time)
# merge_RNAseq_normalized$time = factor(merge_RNAseq_normalized$time, levels=c(&quot;0h&quot;,&quot;2h&quot;,&quot;4h&quot;,&quot;6h&quot;,&quot;8h&quot;,&quot;10h&quot;,&quot;12h&quot;,&quot;24h&quot;))

normalized_counts_matrix &lt;- counts(dds, normalized=TRUE)</code></pre>
<div id="work-with-normalized-data" class="section level2">
<h2>work with normalized data</h2>
<pre class="r"><code>##====================================================================================Lactobacillus delbrueckii===============================================

##---------------
##import data
##---------------
normalized_counts_ldel &lt;- read_delim(&quot;~/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/normalized_counts_ldel.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)
colnames(normalized_counts_ldel)
normalized_counts_ldel_long &lt;- gather(normalized_counts_ldel, sample, norm_count, c(&quot;ldel_di_K1_0h_RNA_count.txt&quot;,&quot;ldel_di_K1_10h_RNA_count.txt&quot;,&quot;ldel_di_K1_12h_RNA_count.txt&quot;,&quot;ldel_di_K1_24h_RNA_count.txt&quot;,&quot;ldel_di_K1_2h_RNA_count.txt&quot;, &quot;ldel_di_K1_4h_RNA_count.txt&quot;,&quot;ldel_di_K1_6h_RNA_count.txt&quot;,&quot;ldel_di_K1_8h_RNA_count.txt&quot;,&quot;ldel_th_K1_0h_RNA_count.txt&quot;,&quot;ldel_th_K1_10h_RNA_count.txt&quot;,&quot;ldel_th_K1_12h_RNA_count.txt&quot;,&quot;ldel_th_K1_24h_RNA_count.txt&quot;,&quot;ldel_th_K1_2h_RNA_count.txt&quot;,&quot;ldel_th_K1_4h_RNA_count.txt&quot;,&quot;ldel_th_K1_6h_RNA_count.txt&quot;,&quot;ldel_th_K1_8h_RNA_count.txt&quot;) , factor_key=TRUE,na.rm = TRUE) 
normalized_counts_ldel$gene &lt;- normalized_counts_ldel$gene %&gt;% gsub(&quot;gene-&quot;,&quot;L_delbrueckii_RMK202_&quot;,.)


##====================================================================================Streptococcus thermophilus===============================================

##---------------
##import data
##---------------
normalized_counts_sterm &lt;- read_delim(&quot;~/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/normalized_counts_sterm.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)
colnames(normalized_counts_sterm)
normalized_counts_sterm_long &lt;- gather(normalized_counts_sterm, sample, norm_count, c(&quot;sterm_di_K1_0h_RNA_count.txt&quot;,&quot;sterm_di_K1_10h_RNA_count.txt&quot;,&quot;sterm_di_K1_12h_RNA_count.txt&quot;,&quot;sterm_di_K1_24h_RNA_count.txt&quot;,&quot;sterm_di_K1_2h_RNA_count.txt&quot;, &quot;sterm_di_K1_4h_RNA_count.txt&quot;,&quot;sterm_di_K1_6h_RNA_count.txt&quot;,&quot;sterm_di_K1_8h_RNA_count.txt&quot;,&quot;sterm_th_K1_0h_RNA_count.txt&quot;,&quot;sterm_th_K1_10h_RNA_count.txt&quot;,&quot;sterm_th_K1_12h_RNA_count.txt&quot;,&quot;sterm_th_K1_24h_RNA_count.txt&quot;,&quot;sterm_th_K1_2h_RNA_count.txt&quot;,&quot;sterm_th_K1_4h_RNA_count.txt&quot;,&quot;sterm_th_K1_6h_RNA_count.txt&quot;,&quot;sterm_th_K1_8h_RNA_count.txt&quot;) , factor_key=TRUE,na.rm = TRUE) 

##====================================================================================merge===============================================
 merge_RNAseq_normalized &lt;- rbind(normalized_counts_sterm_long,normalized_counts_ldel_long)
# merge_RNAseq_normalized
merge_RNAseq_normalized$species &lt;- str_split_fixed(merge_RNAseq_normalized$sample, &quot;_&quot;, 5)[,1]
merge_RNAseq_normalized$replication &lt;- str_split_fixed(merge_RNAseq_normalized$sample, &quot;_&quot;, 5)[,2]
merge_RNAseq_normalized$time &lt;- str_split_fixed(merge_RNAseq_normalized$sample, &quot;_&quot;, 5)[,4]
merge_RNAseq_normalized$time_num &lt;- gsub(&quot;h&quot;,&quot;&quot;,merge_RNAseq_normalized$time)
write.table(merge_RNAseq_normalized, file=&quot;~/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/normalized_counts_merged.txt&quot;, sep=&quot;\t&quot;, quote=F, row.names = FALSE,col.names=TRUE)

##====================================================================================Plot===============================================
library(ggplot2)

merge_RNAseq_normalized &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/normalized_counts_merged.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)

merge_RNAseq_normalized$norm_count &lt;- merge_RNAseq_normalized$norm_count+0.0001
table(merge_RNAseq_normalized$time)
merge_RNAseq_normalized$time = factor(merge_RNAseq_normalized$time, levels=c(&quot;0h&quot;,&quot;2h&quot;,&quot;4h&quot;,&quot;6h&quot;,&quot;8h&quot;,&quot;10h&quot;,&quot;12h&quot;,&quot;24h&quot;))

ggplot(merge_RNAseq_normalized,aes(x=time,y=norm_count,group=gene))+geom_line()+theme_classic()+facet_grid(species~., scales=&quot;free_y&quot;)+coord_trans( y=&quot;log2&quot;)</code></pre>
</div>
</div>
<div id="analysis" class="section level1">
<h1>analysis</h1>
<pre class="r"><code>library(ggplot2)

merge_RNAseq_normalized &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/normalized_counts_merged.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)

merge_RNAseq_normalized$norm_count &lt;- merge_RNAseq_normalized$norm_count+0.0001
table(merge_RNAseq_normalized$time)
merge_RNAseq_normalized$time = factor(merge_RNAseq_normalized$time, levels=c(&quot;0h&quot;,&quot;2h&quot;,&quot;4h&quot;,&quot;6h&quot;,&quot;8h&quot;,&quot;10h&quot;,&quot;12h&quot;,&quot;24h&quot;))

# ggplot(merge_RNAseq_normalized,aes(x=time,y=norm_count,group=gene))+geom_line()+theme_classic()+facet_grid(species~., scales=&quot;free_y&quot;)+coord_trans( y=&quot;log2&quot;)</code></pre>
</div>
<div id="for-regular-genes" class="section level1">
<h1>for regular genes</h1>
<p>##PCA</p>
<pre class="r"><code># 
# merge_RNAseq_normalized &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/normalized_counts_merged.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)
# 
# merge_RNAseq_normalized$norm_count &lt;- merge_RNAseq_normalized$norm_count+0.0001
# table(merge_RNAseq_normalized$time)
# merge_RNAseq_normalized$time = factor(merge_RNAseq_normalized$time, levels=c(&quot;0h&quot;,&quot;2h&quot;,&quot;4h&quot;,&quot;6h&quot;,&quot;8h&quot;,&quot;10h&quot;,&quot;12h&quot;,&quot;24h&quot;))
# 
# merge_RNAseq_normalized %&gt;% column_to_rownames(.,var=&quot;&quot;)

normalized_counts_matrix &lt;- counts(dds, normalized=TRUE)


normalized_counts_matrix

pca &lt;- prcomp(t(normalized_counts_matrix))

summary(pca)

eigs &lt;- pca$sdev^2

pca1_explaine &lt;- round(100*cumsum(eigs[1])/sum(eigs),digits = 2)
pca2_explaine &lt;- round(100*cumsum(eigs[2])/sum(eigs),digits = 2)

pca$sdev

colorsTSNE &lt;- c(&quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;,
          &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;,&quot;grey&quot;,&quot;#000000&quot;)

replicate_rnaseq &lt;- str_split_fixed(colnames(normalized_counts_matrix), &quot;_&quot;, 5)[,4]
replicate_rnaseq &lt;- factor(replicate_rnaseq, levels=c(&quot;0h&quot;,&quot;2h&quot;,&quot;4h&quot;,&quot;6h&quot;,&quot;8h&quot;,&quot;10h&quot;,&quot;12h&quot;,&quot;24h&quot;))
colType &lt;- colorsTSNE[replicate_rnaseq]
# pchType &lt;- c(18, 16)[replicate_rnaseq]


# svg(&quot;~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/RNA-seq_PCA_K1.svg&quot;,width=6,height=5)

plot(
  pca$x,
  col = colType,
  # pch = pchType,
  xlab=paste0(&quot;PC1 (&quot;,pca1_explaine,&quot;%)&quot;),
  ylab=paste0(&quot;PC2 (&quot;,pca2_explaine,&quot;%)&quot;),
  cex = 3)
legend(
  &quot;topright&quot;,
  bty = &quot;n&quot;,
  c(&quot;0h&quot;,&quot;2h&quot;,&quot;4h&quot;,&quot;6h&quot;,&quot;8h&quot;,&quot;10h&quot;,&quot;12h&quot;,&quot;24h&quot;),
  fill = colorsTSNE,
  cex = 1)
# dev.off()</code></pre>
<p>##Co-expression</p>
<p>Here, I try out the coseq bioconductor tool</p>
<pre class="r"><code>library(coseq)</code></pre>
<div id="expression-range" class="section level2">
<h2>expression range</h2>
<p>Here, I want to analyse which genes are highly expressed an what they
are. Thereby I want to include the eggnog annotations.</p>
<pre class="r"><code>library(tidyverse)
merge_RNAseq_normalized &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/normalized_counts_merged.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)

merge_RNAseq_normalized$norm_count &lt;- merge_RNAseq_normalized$norm_count+0.0001
table(merge_RNAseq_normalized$time)
merge_RNAseq_normalized$time = factor(merge_RNAseq_normalized$time, levels=c(&quot;0h&quot;,&quot;2h&quot;,&quot;4h&quot;,&quot;6h&quot;,&quot;8h&quot;,&quot;10h&quot;,&quot;12h&quot;,&quot;24h&quot;))

##----------------------- eggnog import

sterm_all_unique &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Streptococcus_thermophilus_RMK202_pgap_query_seqs.fa.emapper.annotations&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;query_name&quot;,&quot;seed_eggNOG_ortholog&quot;,&quot;seed_ortholog_evalue&quot;,&quot;seed_ortholog_score&quot;,&quot;best_tax_level&quot;,&quot;Preferred_name&quot;,&quot;GOs&quot;,&quot;EC&quot;,&quot;KEGG_ko&quot;,&quot;KEGG_Pathway&quot;,&quot;KEGG_Module&quot;,&quot;KEGG_Reaction&quot;,&quot;KEGG_rclass&quot;,&quot;BRITE&quot;,&quot;KEGG_TC&quot;,&quot;CAZy&quot;,&quot;BiGG_Reaction&quot;,&quot;tax_scope&quot;,&quot;eggNOG OGs&quot;,&quot;bestOG&quot;,&quot;COG_Functional_Category&quot;,&quot;eggNOG free text description&quot;),skip=4)
# RMK202_both_eggnog[RMK202_both_eggnog$query_name %in% SNPeff_final$finalName,]
sterm_all_unique$final_name &lt;- sterm_all_unique$query_name %&gt;% gsub(&quot;gnl\\|extdb\\|&quot;,&quot;S_thermophilus_RMK202_&quot;,.)

ldel_all_unique &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;query_name&quot;,&quot;seed_eggNOG_ortholog&quot;,&quot;seed_ortholog_evalue&quot;,&quot;seed_ortholog_score&quot;,&quot;best_tax_level&quot;,&quot;Preferred_name&quot;,&quot;GOs&quot;,&quot;EC&quot;,&quot;KEGG_ko&quot;,&quot;KEGG_Pathway&quot;,&quot;KEGG_Module&quot;,&quot;KEGG_Reaction&quot;,&quot;KEGG_rclass&quot;,&quot;BRITE&quot;,&quot;KEGG_TC&quot;,&quot;CAZy&quot;,&quot;BiGG_Reaction&quot;,&quot;tax_scope&quot;,&quot;eggNOG OGs&quot;,&quot;bestOG&quot;,&quot;COG_Functional_Category&quot;,&quot;eggNOG free text description&quot;),skip=4)
# RMK202_both_eggnog[RMK202_both_eggnog$query_name %in% SNPeff_final$finalName,]
ldel_all_unique$final_name &lt;- ldel_all_unique$query_name %&gt;% gsub(&quot;gnl\\|extdb\\|&quot;,&quot;L_delbrueckii_RMK202_&quot;,.)


all_eggnog &lt;- rbind(sterm_all_unique,ldel_all_unique)

all_eggnog[grep(&quot;lac&quot;,all_eggnog$Preferred_name,ignore.case = TRUE),]


###---------------------------------
#cumulative expression
###---------------------------------
# merge_RNAseq_normalized$gene_complete &lt;- paste0(merge_RNAseq_normalized$species,&quot;-&quot;,merge_RNAseq_normalized$gene)
# 
# merge_RNAseq_normalized_ordered &lt;- merge_RNAseq_normalized[order(merge_RNAseq_normalized$gene),] %&gt;% as.data.frame()
# merge_RNAseq_normalized_ordered$replication &lt;- as.character(merge_RNAseq_normalized_ordered$replication)
#  merge_RNAseq_normalized %&gt;% as.data.frame() %&gt;% group_by(gene) %&gt;% summarise(cummulative_count=sum(norm_count),median_count=median(norm_count),sd_count=sd(norm_count)) %&gt;% mutate(sd_count_rel=sd_count/median_count)
# 
#  
#  merge_RNAseq_normalized %&gt;% group_by(gene) %&gt;% summarise(cummulative_count=max(norm_count))
# 
# merge_RNAseq_normalized %&gt;% as.tib
# 
# merge_RNAseq_normalized$gene
# merge_RNAseq_normalized$norm_count
#  merge_RNAseq_normalized %&gt;% group_by(gene)  %&gt;% summarise(cummulative_count=sum(norm_count))
# 
# table(merge_RNAseq_normalized$gene)
# # merge_RNAseq_normalized_extended$gene_complete &lt;- paste0(merge_RNAseq_normalized$species,&quot;-&quot;,merge_RNAseq_normalized$gene)
# merge_RNAseq_normalized
merge_RNAseq_cumulative &lt;- merge_RNAseq_normalized %&gt;% dplyr::group_by(gene) %&gt;% summarise(cummulative_count=sum(norm_count),median_count=median(norm_count),sd_count=sd(norm_count)) %&gt;% mutate(sd_count_rel=sd_count/median_count)
merge_RNAseq_cumulative$species &lt;- str_split_fixed(merge_RNAseq_cumulative$gene, &quot;_&quot;, 5)[,1]

table(merge_RNAseq_cumulative$species)

# merge_RNAseq_cumulative %&gt;% filter(species==&quot;S. thermophilus&quot;)

ggplot(merge_RNAseq_cumulative,aes(x=cummulative_count))+geom_density()+theme_classic()+facet_grid(~species,scales = &quot;free&quot;)+coord_trans(x=&quot;log2&quot;)

# ggplot(merge_RNAseq_cumulative,aes(x=median_count))+geom_density()+theme_classic()+facet_grid(~species,scales = &quot;free&quot;)#+coord_trans(x=&quot;log2&quot;)

##only Sterm
merge_RNAseq_cumulative_sterm &lt;- merge_RNAseq_cumulative %&gt;% filter(species==&quot;S&quot;)

ggplot(merge_RNAseq_cumulative_sterm,aes(x=median_count))+geom_density()+theme_classic()+facet_grid(~species,scales = &quot;free&quot;)+coord_trans(x=&quot;log10&quot;)

##median vs. sd

plot_point &lt;- ggplot(merge_RNAseq_cumulative_sterm,aes(x=median_count,y=sd_count_rel))+geom_point()+theme_classic()+coord_trans(x=&quot;log10&quot;,y=&quot;log10&quot;)
# install.packages(&quot;ggplotly&quot;)
library(plotly)

# fig &lt;- plotly::ggplotly(plot_point)
# fig

##------------------- remove outliers

merge_RNAseq_cumulative_sterm_cleaned &lt;- merge_RNAseq_cumulative_sterm %&gt;% filter(sd_count_rel&lt;1000)
plot_point &lt;- ggplot(merge_RNAseq_cumulative_sterm_cleaned,aes(x=median_count,y=sd_count_rel))+geom_point()+theme_classic()+coord_trans(x=&quot;log10&quot;,y=&quot;log10&quot;)
plot_point

ggplot(merge_RNAseq_cumulative_sterm_cleaned,aes(x=median_count))+geom_density()+theme_classic()+facet_grid(~species,scales = &quot;free&quot;)+coord_trans(x=&quot;log10&quot;)


##------------------- add eggnog information
merge_RNAseq_cumulative_cleaned &lt;- merge_RNAseq_cumulative %&gt;% filter(sd_count_rel&lt;1000) %&gt;% filter(median_count&gt;0.5)

all_eggnog_sub &lt;- all_eggnog %&gt;% select(final_name,Preferred_name,`eggNOG free text description`)
# all_eggnog$`eggNOG free text description`
dim(merge_RNAseq_cumulative_cleaned)
merge_RNAseq_cumulative_cleaned_eggnog &lt;- merge(merge_RNAseq_cumulative_cleaned,all_eggnog_sub,by.x=&quot;gene&quot;,by.y=&quot;final_name&quot;,all.x = TRUE)
dim(merge_RNAseq_cumulative_cleaned)
dim(merge_RNAseq_cumulative_cleaned_eggnog)

plot_point &lt;- ggplot(merge_RNAseq_cumulative_cleaned_eggnog,aes(x=median_count,y=sd_count_rel))+geom_point()+theme_classic()+coord_trans(x=&quot;log10&quot;,y=&quot;log10&quot;)+facet_grid(~species,scales = &quot;free&quot;)
plot_point

# fig &lt;- plotly::ggplotly(plot_point)
# htmlwidgets::saveWidget(fig, &quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/geneExpression.html&quot;)


merge_RNAseq_cumulative_cleaned_eggnog[grep(&quot;lac&quot;,merge_RNAseq_cumulative_cleaned_eggnog$Preferred_name,ignore.case = TRUE),]

merge_RNAseq_cumulative_cleaned_eggnog$coloring &lt;- &quot;none&quot;

merge_RNAseq_cumulative_cleaned_eggnog[grep(&quot;lac&quot;,merge_RNAseq_cumulative_cleaned_eggnog$Preferred_name,ignore.case = TRUE),&quot;coloring&quot;] &lt;- &quot;lactose&quot;

table(merge_RNAseq_cumulative_cleaned_eggnog$coloring)


plot_point &lt;- ggplot(merge_RNAseq_cumulative_cleaned_eggnog,aes(x=median_count,y=sd_count_rel,fill=coloring,color=coloring,size=coloring))+geom_point(alpha=0.5)+theme_classic()+coord_trans(x=&quot;log10&quot;,y=&quot;log10&quot;)+facet_grid(~species,scales = &quot;free&quot;)+scale_size_manual(values=c(4,0.5))
plot_point</code></pre>
</div>
</div>
<div id="only-pseudogenes" class="section level1">
<h1>ONLY pseudogenes</h1>
<p>Here, I want to see if the pseudogenes are expressed. For the regular
genes we have to look at the top analysis.</p>
<p>#####pseudoannotation First I extract and label all pseudo genes from
the PGAP annotation pseudogenes are by default not counted with HTseq.
We have to add seperatly.</p>
<pre class="bash"><code>
###-------------------------------
#create genes out of Pseudogenes
###-------------------------------

grep &quot;pseudo=true&quot;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap.gff | awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;} {if($3==&quot;CDS&quot;) print $0}&#39;|sed &#39;s/CDS/gene/g&#39;|sed &#39;s/cds-//g&#39;  &gt;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_pseudoGenes.gff


grep &quot;pseudo=true&quot;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap.gff | awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;} {if($3==&quot;CDS&quot;) print $0}&#39;|sed &#39;s/CDS/gene/g&#39;|sed &#39;s/cds-//g&#39;  &gt;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_pseudoGenes.gff


###-------------------------------
#find what pseudo gene it is
###-------------------------------

grep &quot;pseudo=true&quot;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap.gff | awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;} {if($3==&quot;CDS&quot;) print $9}&#39;|awk -F &quot;;&quot; &#39;{OFS=&quot;\t&quot;} {print $2,$3}&#39;|grep &quot;Note=&quot; |sed &#39;s/Parent=//g&#39;|sed &#39;s/Note=//g&#39; | sed &#39;s/%.*$//g&#39; |grep &quot;phosphoenolpyruvate&quot; -v  |grep &quot;Catalyzes&quot; -v &gt;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_pseudoGenes_indepth.gff


grep &quot;pseudo=true&quot;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap.gff | awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;} {if($3==&quot;CDS&quot;) print $9}&#39;|awk -F &quot;;&quot; &#39;{OFS=&quot;\t&quot;} {print $2,$3}&#39;|grep &quot;Note=&quot; |sed &#39;s/Parent=//g&#39;|sed &#39;s/Note=//g&#39; | sed &#39;s/%.*$//g&#39;  |grep &quot;and&quot; -v&gt;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_pseudoGenes_indepth.gff

###-------------------------------
#merge with previous
###-------------------------------

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap.gff /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_pseudoGenes.gff &gt; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_WithPseudoGenes.gff

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap.gff /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_pseudoGenes.gff &gt; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_WithPseudoGenes.gff

threads=37



##----------------------------------------------
##--------------count events
##----------------------------------------------

 for names in $(cut -f 1 /home/vincent/Projects/2019_RNAseq_rmk202/02_script/names_samples.txt)
  do
  echo ${num}&quot;/16  :&quot; ${names}
  num=$((num+1))
 #rm -r /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/

mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/
  cd /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/
  htseq-count --stranded no -i ID --format bam --type gene /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam \
    /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_WithPseudoGenes.gff &gt; /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/sterm_${names}_count_pseudogenes_all.txt

  htseq-count --stranded no -i ID --format bam --type gene /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam \
    /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_WithPseudoGenes.gff &gt; /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/ldel_${names}_count_pseudogenes_all.txt

   done 

##----------------------------------------------
##--------------move to common folder
##----------------------------------------------
rm -r /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/htSeq_all_pseudo
mkdir -p /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/htSeq_all_pseudo

 for names in $(cut -f 1 /home/vincent/Projects/2019_RNAseq_rmk202/02_script/names_samples.txt)
  do
  echo ${num}&quot;/16  :&quot; ${names}
  num=$((num+1))

cp /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/ldel_${names}_count_pseudogenes_all.txt /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/htSeq_all_pseudo/ldel_${names}_count.txt

cp /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/${names}/htseq/sterm_${names}_count_pseudogenes_all.txt /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/htSeq_all_pseudo/sterm_${names}_count.txt


 done 



##----------------------------------------------
##--------------bring local
##----------------------------------------------
mkdir -p ~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/
scp -r vincent@130.223.51.116:/home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/htSeq_all_pseudo/ ~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/
</code></pre>
<p>####eggnog annotation of pseudogenes get faa of pseudogenes and
annotate with eggnog</p>
<pre class="bash"><code>##------Sterm
awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9,$7}&#39; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_pseudoGenes.gff |sed &#39;s/ID=//g&#39; &gt;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_pseudoGenes.bed

bedtools getfasta -s -name  -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_pseudoGenes.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pseudogenes.ffn

sed -i &#39;s/::.*//&#39; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pseudogenes.ffn

transeq  -sequence /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pseudogenes.ffn &gt; \
  -outseq  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pseudogenes.faa

 sed -i &#39;s/_1$//g&#39;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pseudogenes.faa

##------Ldel
awk -F &quot;[\t;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1,$4,$5,$9,$7}&#39; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_pseudoGenes.gff |sed &#39;s/ID=//g&#39; &gt;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_pseudoGenes.bed


bedtools getfasta -s -name -fullHeader -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Ldel_RMK202.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_pseudoGenes.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_pseudoGenes.ffn
  
sed -i &#39;s/::.*//&#39; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_pseudoGenes.ffn

transeq  -sequence /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_pseudoGenes.ffn &gt; \
  -outseq  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_pseudoGenes.faa
  
  sed -i &#39;s/_1$//g&#39;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_pseudoGenes.faa

##----------------------------------------------
##--------------bring local
##----------------------------------------------
mkdir -p ~/Desktop/Projects/2019_RNAseq_rmk202/04_eggnog_annotation/
scp vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_pseudoGenes.faa ~/Desktop/Projects/2019_RNAseq_rmk202/04_eggnog_annotation/
scp vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pseudogenes.faa ~/Desktop/Projects/2019_RNAseq_rmk202/04_eggnog_annotation/

###annotation location

ls /home/vincent/Desktop/Projects/2019_RNAseq_rmk202/04_eggnog_annotation
</code></pre>
<p>####normalize count data</p>
<pre class="bash"><code>###--------------------------------
##gene labelling (pseudogene or Gene)
###--------------------------------

grep &quot;^__&quot; -v ~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/htSeq_all_pseudo/ldel_di_K1_10h_RNA_count.txt | grep &quot;^gene&quot; | awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;} {print $1, &quot;gene&quot;}&#39; &gt; ~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/FeatureType.txt
grep &quot;^__&quot; -v ~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/htSeq_all_pseudo/ldel_di_K1_10h_RNA_count.txt | grep -v &quot;^gene&quot; | awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;} {print $1, &quot;pseudogene&quot;}&#39; &gt;&gt; ~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/FeatureType.txt



grep &quot;^__&quot; -v ~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/htSeq_all_pseudo/sterm_di_K1_10h_RNA_count.txt | grep &quot;^gene&quot; | awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;} {print $1, &quot;gene&quot;}&#39; &gt;&gt; ~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/FeatureType.txt

grep &quot;^__&quot; -v ~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/htSeq_all_pseudo/sterm_di_K1_10h_RNA_count.txt | grep -v &quot;^gene&quot; | awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;} {print $1, &quot;pseudogene&quot;}&#39; &gt;&gt; ~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/FeatureType.txt


#grep &quot;^__&quot; -v /home/vincent/Projects/2019_RNAseq_rmk202/04_bwaMapping2ONT/htSeq_all_pseudo/sterm_di_K1_10h_RNA_count.txt| grep -v &quot;^gene&quot; 
</code></pre>
<pre class="r"><code>library(GenomicDataCommons)
library(DESeq2)
library(dplyr)
library(readr)
##====================================================================================Lactobacillus delbrueckii===============================================
directory &lt;- &quot;~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/htSeq_all_pseudo/&quot;
sampleFiles &lt;- grep(&quot;ldel&quot;,list.files(directory),value=TRUE)
sampleCondition &lt;- str_split_fixed(sampleFiles, &quot;_&quot;, 5)[,2]


sampleTable &lt;- data.frame(sampleName = sampleFiles,
                          fileName = sampleFiles,
                          condition = sampleCondition)

ddsHTSeq &lt;- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
                                       directory = directory,
                                       design= ~ condition)
##---------------
##Pre-filtering (keep only rows that have at least 10 reads total)
##---------------
keep &lt;- rowSums(counts(ddsHTSeq)) &gt;= 10
dds &lt;- ddsHTSeq[keep,]

dds &lt;- estimateSizeFactors(dds)
##---------------
##DESeq2’s median of ratios normalization accounts for sequencing depth and RNA composition it is calculated as following :counts divided by sample-specific size factors determined by median ratio of gene counts relative to geometric mean per gene
##---------------
normalized_counts &lt;- counts(dds, normalized=TRUE) %&gt;% as.data.frame()
normalized_counts &lt;- tibble::rownames_to_column(normalized_counts, &quot;gene&quot;)
write.table(normalized_counts, file=&quot;~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/normalized_counts_ldel.txt&quot;, sep=&quot;\t&quot;, quote=F, row.names = FALSE,col.names=TRUE)

##====================================================================================Streptococcus thermophilus===============================================
directory &lt;- &quot;~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/htSeq_all_pseudo/&quot;
sampleFiles &lt;- grep(&quot;sterm&quot;,list.files(directory),value=TRUE)
sampleCondition &lt;- str_split_fixed(sampleFiles, &quot;_&quot;, 5)[,2]

sampleTable &lt;- data.frame(sampleName = sampleFiles,
                          fileName = sampleFiles,
                          condition = sampleCondition)

ddsHTSeq &lt;- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
                                       directory = directory,
                                       design= ~ condition)

##---------------
##Pre-filtering (keep only rows that have at least 10 reads total)
##---------------
keep &lt;- rowSums(counts(ddsHTSeq)) &gt;= 10
dds &lt;- ddsHTSeq[keep,]
dds &lt;- estimateSizeFactors(dds)
##---------------
##DESeq2’s median of ratios normalization accounts for sequencing depth and RNA composition it is calculated as following :counts divided by sample-specific size factors determined by median ratio of gene counts relative to geometric mean per gene
##---------------

normalized_counts &lt;- counts(dds, normalized=TRUE) %&gt;% as.data.frame()
normalized_counts &lt;- tibble::rownames_to_column(normalized_counts, &quot;gene&quot;)
write.table(normalized_counts, file=&quot;~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/normalized_counts_sterm.txt&quot;, sep=&quot;\t&quot;, quote=F, row.names = FALSE,col.names=TRUE)
normalized_counts_matrix &lt;- counts(dds, normalized=TRUE)</code></pre>
<div id="work-with-normalized-data-1" class="section level2">
<h2>work with normalized data</h2>
<pre class="r"><code>##====================================================================================Lactobacillus delbrueckii===============================================
library(tidyverse)
 library(plotly)

##---------------
##import data
##---------------
normalized_counts_ldel &lt;- read_delim(&quot;~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/normalized_counts_ldel.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)
colnames(normalized_counts_ldel)
normalized_counts_ldel_long &lt;- gather(normalized_counts_ldel, sample, norm_count, c(&quot;ldel_di_K1_0h_RNA_count.txt&quot;,&quot;ldel_di_K1_10h_RNA_count.txt&quot;,&quot;ldel_di_K1_12h_RNA_count.txt&quot;,&quot;ldel_di_K1_24h_RNA_count.txt&quot;,&quot;ldel_di_K1_2h_RNA_count.txt&quot;, &quot;ldel_di_K1_4h_RNA_count.txt&quot;,&quot;ldel_di_K1_6h_RNA_count.txt&quot;,&quot;ldel_di_K1_8h_RNA_count.txt&quot;,&quot;ldel_th_K1_0h_RNA_count.txt&quot;,&quot;ldel_th_K1_10h_RNA_count.txt&quot;,&quot;ldel_th_K1_12h_RNA_count.txt&quot;,&quot;ldel_th_K1_24h_RNA_count.txt&quot;,&quot;ldel_th_K1_2h_RNA_count.txt&quot;,&quot;ldel_th_K1_4h_RNA_count.txt&quot;,&quot;ldel_th_K1_6h_RNA_count.txt&quot;,&quot;ldel_th_K1_8h_RNA_count.txt&quot;) , factor_key=TRUE,na.rm = TRUE) 



##====================================================================================Streptococcus thermophilus===============================================

##---------------
##import data
##---------------
normalized_counts_sterm &lt;- read_delim(&quot;~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/normalized_counts_sterm.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)
colnames(normalized_counts_sterm)
normalized_counts_sterm_long &lt;- gather(normalized_counts_sterm, sample, norm_count, c(&quot;sterm_di_K1_0h_RNA_count.txt&quot;,&quot;sterm_di_K1_10h_RNA_count.txt&quot;,&quot;sterm_di_K1_12h_RNA_count.txt&quot;,&quot;sterm_di_K1_24h_RNA_count.txt&quot;,&quot;sterm_di_K1_2h_RNA_count.txt&quot;, &quot;sterm_di_K1_4h_RNA_count.txt&quot;,&quot;sterm_di_K1_6h_RNA_count.txt&quot;,&quot;sterm_di_K1_8h_RNA_count.txt&quot;,&quot;sterm_th_K1_0h_RNA_count.txt&quot;,&quot;sterm_th_K1_10h_RNA_count.txt&quot;,&quot;sterm_th_K1_12h_RNA_count.txt&quot;,&quot;sterm_th_K1_24h_RNA_count.txt&quot;,&quot;sterm_th_K1_2h_RNA_count.txt&quot;,&quot;sterm_th_K1_4h_RNA_count.txt&quot;,&quot;sterm_th_K1_6h_RNA_count.txt&quot;,&quot;sterm_th_K1_8h_RNA_count.txt&quot;) , factor_key=TRUE,na.rm = TRUE) 

##====================================================================================merge===============================================
 merge_RNAseq_normalized &lt;- rbind(normalized_counts_sterm_long,normalized_counts_ldel_long)
# merge_RNAseq_normalized
merge_RNAseq_normalized$species &lt;- str_split_fixed(merge_RNAseq_normalized$sample, &quot;_&quot;, 5)[,1]
merge_RNAseq_normalized$replication &lt;- str_split_fixed(merge_RNAseq_normalized$sample, &quot;_&quot;, 5)[,2]
merge_RNAseq_normalized$time &lt;- str_split_fixed(merge_RNAseq_normalized$sample, &quot;_&quot;, 5)[,4]
merge_RNAseq_normalized$time_num &lt;- gsub(&quot;h&quot;,&quot;&quot;,merge_RNAseq_normalized$time)
write.table(merge_RNAseq_normalized, file=&quot;~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/normalized_counts_merged.txt&quot;, sep=&quot;\t&quot;, quote=F, row.names = FALSE,col.names=TRUE)

##====================================================================================Plot===============================================

merge_RNAseq_normalized &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/normalized_counts_merged.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)

merge_RNAseq_normalized$norm_count &lt;- merge_RNAseq_normalized$norm_count+0.0001
table(merge_RNAseq_normalized$time)
merge_RNAseq_normalized$time = factor(merge_RNAseq_normalized$time, levels=c(&quot;0h&quot;,&quot;2h&quot;,&quot;4h&quot;,&quot;6h&quot;,&quot;8h&quot;,&quot;10h&quot;,&quot;12h&quot;,&quot;24h&quot;))

ggplot(merge_RNAseq_normalized,aes(x=time,y=norm_count,group=gene))+geom_line()+theme_classic()+facet_grid(species~., scales=&quot;free_y&quot;)+coord_trans( y=&quot;log2&quot;)

##====================================================================================Plot===============================================
## import feature labelling
FeatureType &lt;- read_delim(&quot;~/Desktop/Projects/2019_RNAseq_rmk202/03_HTseq_all_pseudo/FeatureType.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;gene&quot;,&quot;feature&quot;), trim_ws = TRUE)

merge_RNAseq_normalized_extended &lt;- merge(merge_RNAseq_normalized,FeatureType,by=&quot;gene&quot;,all.x = TRUE)

table(merge_RNAseq_normalized_extended$feature)

merge_RNAseq_normalized_extended %&gt;% filter(feature==&quot;pseudogene&quot;)

merge_RNAseq_normalized_extended[grep(&quot;pgaptmp_000045&quot;,merge_RNAseq_normalized_extended$gene),]

# add extensive species name

merge_RNAseq_normalized_extended$species_extensive &lt;- plyr::revalue(merge_RNAseq_normalized_extended$species, c(&quot;ldel&quot;=&quot;L_delbrueckii_RMK202&quot;,&quot;sterm&quot;=&quot;S_thermophilus_RMK202&quot;))

# shorten genename

merge_RNAseq_normalized_extended$gene_short &lt;- gsub(&quot;gene-&quot;,&quot;&quot;,merge_RNAseq_normalized_extended$gene)

# gene complete

merge_RNAseq_normalized_extended$gene_complete &lt;- paste0(merge_RNAseq_normalized_extended$species_extensive,&quot;_&quot;,merge_RNAseq_normalized_extended$gene_short)

unique(merge_RNAseq_normalized_extended$gene_complete) %&gt;% length()
tmp &lt;- merge_RNAseq_normalized_extended %&gt;% select(species,gene_complete) %&gt;% unique() 
table(tmp$species)

write.table(merge_RNAseq_normalized_extended, file=&quot;~/Desktop/Projects/2019_RNAseq_rmk202/normalized_counts_merged_allGenes_final.txt&quot;, sep=&quot;\t&quot;, quote=F, row.names = FALSE,col.names=TRUE)


# ggplot(merge_RNAseq_normalized_extended,aes(x=time,y=norm_count,group=gene,color=feature))+facet_wrap(feature~species,)+geom_line()+theme_classic()+coord_trans( y=&quot;log2&quot;)
library(ggpubr)
library(ggsignif)
# annotation table with adjusted pvals and y-position of the labels

merge_RNAseq_normalized_extended$species &lt;- plyr::revalue(merge_RNAseq_normalized_extended$species, c(&quot;ldel&quot;=&quot;L. delbrueckii&quot;,&quot;sterm&quot;=&quot;S. thermophilus&quot;))

max(merge_RNAseq_normalized_extended$norm_count)
mean(merge_RNAseq_normalized_extended$norm_count)

table(merge_RNAseq_normalized_extended$feature)

plotPseudogene &lt;- ggplot(merge_RNAseq_normalized_extended,aes(x=time,y=norm_count,color=feature))+facet_wrap(~species,ncol=1,scales=&quot;free_x&quot;)+geom_boxplot(alpha=0.7)+theme_classic()+coord_trans( y=&quot;log2&quot;)+scale_color_manual(values=c(&quot;grey44&quot;,&quot;grey77&quot;))+
stat_compare_means(aes(group = feature), label = &quot;p.signif&quot;)+labs(x=&quot;incubation time&quot;,y=&quot;normalized gene expression&quot;)+theme(legend.title = element_blank())+scale_y_continuous(breaks=c(5,500,50000,1800000))
plotPseudogene
# ggplot(merge_RNAseq_normalized_extended,aes(x=feature,y=norm_count,color=feature))+facet_wrap(time~species,ncol=16,scales=&quot;free_x&quot;)+geom_boxplot()+theme_classic()+coord_trans( y=&quot;log2&quot;)+scale_color_manual(values=c(&quot;grey44&quot;,&quot;grey77&quot;))


pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/plotPseudogene_expression.pdf&quot;,width=7,height=4.5)

plotPseudogene

dev.off()

###---------------------------------
#cumulative expression
###---------------------------------
merge_RNAseq_normalized_extended$gene_complete &lt;- paste0(merge_RNAseq_normalized_extended$species,&quot;-&quot;,merge_RNAseq_normalized_extended$gene)

merge_RNAseq_cumulative &lt;- merge_RNAseq_normalized_extended %&gt;% group_by(gene_complete) %&gt;% summarise(cummulative_count=sum(norm_count),median_count=median(norm_count),sd_count=sd(norm_count)) %&gt;% mutate(sd_count_rel=sd_count/median_count)
merge_RNAseq_cumulative$species &lt;- str_split_fixed(merge_RNAseq_cumulative$gene_complete, &quot;-&quot;, 5)[,1]

table(merge_RNAseq_cumulative$species)

# merge_RNAseq_cumulative %&gt;% filter(species==&quot;S. thermophilus&quot;)

ggplot(merge_RNAseq_cumulative,aes(x=cummulative_count))+geom_density()+theme_classic()+facet_grid(~species,scales = &quot;free&quot;)+coord_trans(x=&quot;log2&quot;)

# ggplot(merge_RNAseq_cumulative,aes(x=median_count))+geom_density()+theme_classic()+facet_grid(~species,scales = &quot;free&quot;)#+coord_trans(x=&quot;log2&quot;)

##only Sterm
merge_RNAseq_cumulative_sterm &lt;- merge_RNAseq_cumulative %&gt;% filter(species==&quot;S. thermophilus&quot;)

ggplot(merge_RNAseq_cumulative_sterm,aes(x=median_count))+geom_density()+theme_classic()+facet_grid(~species,scales = &quot;free&quot;)+coord_trans(x=&quot;log10&quot;)

##median vs. sd

plot_point &lt;- ggplot(merge_RNAseq_cumulative_sterm,aes(x=median_count,y=sd_count_rel))+geom_point()+theme_classic()+coord_trans(x=&quot;log10&quot;,y=&quot;log10&quot;)
# install.packages(&quot;ggplotly&quot;)
# library(ggplotly)

fig &lt;- plotly::ggplotly(plot_point)
fig

##------------------- remove outliers

merge_RNAseq_cumulative_sterm_cleaned &lt;- merge_RNAseq_cumulative_sterm %&gt;% filter(sd_count_rel&lt;1000)
plot_point &lt;- ggplot(merge_RNAseq_cumulative_sterm_cleaned,aes(x=median_count,y=sd_count_rel))+geom_point()+theme_classic()+coord_trans(x=&quot;log10&quot;,y=&quot;log10&quot;)
plot_point

ggplot(merge_RNAseq_cumulative_sterm_cleaned,aes(x=median_count))+geom_density()+theme_classic()+facet_grid(~species,scales = &quot;free&quot;)+coord_trans(x=&quot;log10&quot;)


##------------------- add eggnog information

     normalized_counts_sterm &lt;- read_csv(&quot;~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/RMK202_snpeff_eggnog_repeats_core_dNdS_uniqueGenes_subset.txt&quot;)</code></pre>
<p>#####Pseudogene cause</p>
<p>what to extract: 1. version (accession) 2. source (organism) 3. Genes
(total) 4. Pseudo Genes (total) 5. Pseudo Genes (ambiguous residues) 6.
Pseudo Genes (frameshifted) 7. Pseudo Genes (incomplete) 8. Pseudo Genes
(internal stop) 9. Pseudo Genes (multiple problems) 10. tRNAs 11. CRISPR
Arrays 12. rRNAs 13. Genes (RNA)</p>
<pre class="bash"><code>path_base=/data/Project/Pseudogene/01_genome/20200216/gb
mkdir -p /data/Project/Pseudogene/gb_scan/
rm /data/Project/Pseudogene/gb_scan/genomeScan.txt
for id in $(ls /data/Project/Pseudogene/01_genome/20200216/gb)
  do
  
  
  version=NA
  source=NA
  Genes=NA
  Pseudo_Genes=NA
  Pseudo_Genes_ambiguous=NA
  Pseudo_Genes_frameshifted=NA
  Pseudo_Genes_incomplete=NA
  Pseudo_Genes_internalStop=NA
  Pseudo_Genes_multiple=NA
  tRNAs=NA
  CRISPR=NA
  rRNAs=NA
  Genes=NA
  rRNAs_5s=NA
  rRNAs_16s=NA
  rRNAs_23s=NA

  version=$(zcat $path_base/$id |head -1000|grep &quot;VERSION&quot;|head -1| sed &#39;s/VERSION//g&#39;|sed -e &#39;s/^[ \t]*//&#39;|sed -e &#39;s/*[ \t]$//&#39;)
  source=$(zcat $path_base/$id |head -1000|grep &quot;SOURCE&quot;|head -1| sed &#39;s/SOURCE//g&#39;|sed -e &#39;s/^[ \t]*//&#39;|sed -e &#39;s/*[ \t]$//&#39;|cut -d &#39; &#39; -f 1,2|sed &#39;s/ /_/g&#39;)
  Genes=$(zcat $path_base/$id |head -1000|grep &quot;Genes (total)&quot;|head -1| awk -F &quot;::&quot; &#39;{print $2}&#39;|sed -e &#39;s/^[ \t]*//&#39;|sed -e &#39;s/*[ \t]$//&#39;|sed &#39;s/,//g&#39;)
  Pseudo_Genes=$(zcat $path_base/$id |head -1000|grep &quot;Pseudo Genes (total)&quot;|head -1| awk -F &quot;::&quot; &#39;{print $2}&#39;|sed -e &#39;s/^[ \t]*//&#39;|sed -e &#39;s/*[ \t]$//&#39;)
  Pseudo_Genes_ambiguous=$(zcat $path_base/$id |head -1000|grep &quot;Pseudo Genes (ambiguous residues)&quot;|head -1| awk -F &quot;::&quot; &#39;{print $2}&#39;|awk -F &quot; &quot; &#39;{print $1}&#39;)
  Pseudo_Genes_frameshifted=$(zcat $path_base/$id |head -1000|grep &quot;Pseudo Genes (frameshifted&quot;|head -1| awk -F &quot;::&quot; &#39;{print $2}&#39;|awk -F &quot; &quot; &#39;{print $1}&#39;)
  Pseudo_Genes_incomplete=$(zcat $path_base/$id |head -1000|grep &quot;Pseudo Genes (incomplete&quot;|head -1| awk -F &quot;::&quot; &#39;{print $2}&#39;|awk -F &quot; &quot; &#39;{print $1}&#39;)
  Pseudo_Genes_internalStop=$(zcat $path_base/$id |head -1000|grep &quot;Pseudo Genes (internal&quot;|head -1| awk -F &quot;::&quot; &#39;{print $2}&#39;|awk -F &quot; &quot; &#39;{print $1}&#39;)
  Pseudo_Genes_multiple=$(zcat $path_base/$id |head -1000|grep &quot;Pseudo Genes (multiple&quot;|head -1| awk -F &quot;::&quot; &#39;{print $2}&#39;|awk -F &quot; &quot; &#39;{print $1}&#39;)
  tRNAs=$(zcat $path_base/$id |head -1000|grep &quot;tRNAs&quot;|head -1| awk -F &quot;::&quot; &#39;{print $2}&#39;|sed -e &#39;s/^[ \t]*//&#39;|sed -e &#39;s/*[ \t]$//&#39;|sed &#39;s/,//g&#39;)
  CRISPR=$(zcat $path_base/$id |head -1000|grep &quot;CRISPR Arrays&quot;|head -1| awk -F &quot;::&quot; &#39;{print $2}&#39;|sed -e &#39;s/^[ \t]*//&#39;|sed -e &#39;s/*[ \t]$//&#39;|sed &#39;s/,//g&#39;)
  rRNAs_5s=$(zcat $path_base/$id |head -1000|grep &quot;complete rRNAs&quot;|head -1| awk -F &quot;::&quot; &#39;{print $2}&#39;|awk -F &quot;[, ]&quot; &#39;{print $2}&#39;)
  rRNAs_16s=$(zcat $path_base/$id |head -1000|grep &quot;complete rRNAs&quot;|head -1| awk -F &quot;::&quot; &#39;{print $2}&#39;|awk -F &quot;[, ]&quot; &#39;{print $4}&#39;)
  rRNAs_23s=$(zcat $path_base/$id |head -1000|grep &quot;complete rRNAs&quot;|head -1| awk -F &quot;::&quot; &#39;{print $2}&#39;|awk -F &quot;[, ]&quot; &#39;{print $6}&#39;)
   
  echo -e ${version}&quot;\t&quot;${source}&quot;\t&quot;${Genes}&quot;\t&quot;${Pseudo_Genes}&quot;\t&quot;${Pseudo_Genes_ambiguous}&quot;\t&quot;${Pseudo_Genes_frameshifted}&quot;\t&quot;${Pseudo_Genes_incomplete}&quot;\t&quot;${Pseudo_Genes_internalStop}&quot;\t&quot;${Pseudo_Genes_multiple}&quot;\t&quot;${tRNAs}&quot;\t&quot;${CRISPR}&quot;\t&quot;${rRNAs_5s}&quot;\t&quot;${rRNAs_16s}&quot;\t&quot;${rRNAs_23s} &gt;&gt; /data/Project/Pseudogene/gb_scan/genomeScan.txt
  
  done
  
  mkdir -p ~/Desktop/Projects/2019_RNAseq_rmk202/gb_scan
scp vincent@130.223.51.116:/data/Project/Pseudogene/gb_scan/genomeScan.txt ~/Desktop/Projects/2019_RNAseq_rmk202/gb_scan/




###-------------------------------
#identify pseudogenes
###-------------------------------

grep &quot;pseudo=true&quot;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap.gff | awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;} {if($3==&quot;CDS&quot;) print $0}&#39;|sed &#39;s/CDS/gene/g&#39;|sed &#39;s/cds-//g&#39;  &gt;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_pseudoGenes.gff


grep &quot;pseudo=true&quot;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap.gff | awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;} {if($3==&quot;CDS&quot;) print $0}&#39;|sed &#39;s/CDS/gene/g&#39;|sed &#39;s/cds-//g&#39;  &gt;  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_pseudoGenes.gff

VERSION
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d &#39;.&#39; -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep &quot;Pseudo Genes (total)              ::&quot; |head -1| sed &#39;s/Pseudo Genes (total)              :://g&#39;| sed -e &#39;s/^[ \t]*//&#39; |sed -e &#39;s/*[ \t]$//&#39; |awk -v genomesss=&quot;$shortGenome&quot; -v speciesss=&quot;$short&quot; &#39;BEGIN{OFS=&quot;\t&quot;}{print genomesss , speciesss , &quot;NCBI&quot;,$0}&#39; &gt;&gt; /home/vincent/Projects/2019_pilotplant/comparativeGenomics/pseudogene/${short}_${what}.txt

done


&quot;Genes (total)                     :: &quot;
</code></pre>
<pre class="r"><code>library(readr)
library(tidyverse)
genomeScan &lt;- read_delim(&quot;~/Desktop/Projects/2019_RNAseq_rmk202/gb_scan/genomeScan.txt&quot;,   &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;Accession&quot;,&quot;source&quot;,&quot;Genes&quot;,&quot;Pseudo_Genes&quot;,&quot;Pseudo_Genes_ambiguous&quot;,&quot;Pseudo_Genes_frameshifted&quot;,&quot;Pseudo_Genes_incomplete&quot;,&quot;Pseudo_Genes_internalStop&quot;,&quot;Pseudo_Genes_multiple&quot;,&quot;tRNAs&quot;,&quot;CRISPR&quot;,&quot;rRNAs_5s&quot;,&quot;rRNAs_16s&quot;,&quot;rRNAs_23s&quot;), trim_ws = TRUE)  %&gt;% tidyr::separate(source, into=c(&quot;Genus&quot;, &quot;Species&quot;),sep=&quot;_&quot;,remove=FALSE)

# dim(genomeScan)

genomeScan_filter &lt;- genomeScan %&gt;% filter(!is.na(Genes)) %&gt;%  mutate(percent_Pseudo_Genes=100*(Pseudo_Genes/Genes))
# dim(genomeScan_filter)
genomeScan_filter$Strepto &lt;- ifelse(genomeScan_filter$Genus==&quot;Streptococcus&quot;,&quot;Streptococcus&quot;,&quot;Rest&quot;)
genomeScan_filter$Sterm &lt;- ifelse(genomeScan_filter$source==&quot;Streptococcus_thermophilus&quot;,&quot;Streptococcus_thermophilus&quot;,&quot;Rest&quot;)

ggplot(genomeScan_filter,aes(x=percent_Pseudo_Genes,group=Sterm,color=Sterm,fill=Sterm))+geom_density(alpha=0.3)+theme_classic()+lims(x=c(0,25))</code></pre>
<p>#####create a pseudogene DB</p>
<pre class="bash"><code>
###-----------------
##create prokaryotic database
###-----------------

mkdir -p /data/Project/Pseudogene/01_genome/20200216/gb

curl &#39;ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt&#39; | \
awk -F &quot;\t&quot; &#39;$12==&quot;Complete Genome&quot; &amp;&amp; $11==&quot;latest&quot;{print $0}&#39; &gt; /data/Project/Pseudogene/01_genome/20200216/genomic_file.txt



cut -f 20 /data/Project/Pseudogene/01_genome/20200216/genomic_file.txt  |sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2/\2_genomic.gbff.gz|&#39;&gt; \
/data/Project/Pseudogene/01_genome/20200216/download_file.txt

#ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/003/671/935/GCF_003671935.1_ASM367193v1/GCF_003671935.1_ASM367193v1_genomic.gbff.gz

cd /data/Project/Pseudogene/01_genome/20200216/gb

wget --input /data/Project/Pseudogene/01_genome/20200216/download_file.txt
</code></pre>
</div>
</div>
<div id="including-all-genes" class="section level1">
<h1>Including all genes</h1>
<p>Here, I do the analysis on all pseudo and normal genes at once. also
able to include the eggnog annotations</p>
<pre class="r"><code>library(tidyverse)
merge_RNAseq_normalized &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2019_RNAseq_rmk202/normalized_counts_merged_allGenes_final.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)



# merge_RNAseq_normalized &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2019_RNAseq_rmk202/02_HTseq_all/normalized_counts_merged.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE)

merge_RNAseq_normalized$norm_count &lt;- merge_RNAseq_normalized$norm_count+0.0001
table(merge_RNAseq_normalized$time)
merge_RNAseq_normalized$time = factor(merge_RNAseq_normalized$time, levels=c(&quot;0h&quot;,&quot;2h&quot;,&quot;4h&quot;,&quot;6h&quot;,&quot;8h&quot;,&quot;10h&quot;,&quot;12h&quot;,&quot;24h&quot;))

##----------------------- eggnog import

sterm_all_unique &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Streptococcus_thermophilus_RMK202_pgap_query_seqs.fa.emapper.annotations&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;query_name&quot;,&quot;seed_eggNOG_ortholog&quot;,&quot;seed_ortholog_evalue&quot;,&quot;seed_ortholog_score&quot;,&quot;best_tax_level&quot;,&quot;Preferred_name&quot;,&quot;GOs&quot;,&quot;EC&quot;,&quot;KEGG_ko&quot;,&quot;KEGG_Pathway&quot;,&quot;KEGG_Module&quot;,&quot;KEGG_Reaction&quot;,&quot;KEGG_rclass&quot;,&quot;BRITE&quot;,&quot;KEGG_TC&quot;,&quot;CAZy&quot;,&quot;BiGG_Reaction&quot;,&quot;tax_scope&quot;,&quot;eggNOG OGs&quot;,&quot;bestOG&quot;,&quot;COG_Functional_Category&quot;,&quot;eggNOG free text description&quot;),skip=4)
# RMK202_both_eggnog[RMK202_both_eggnog$query_name %in% SNPeff_final$finalName,]
sterm_all_unique$final_name &lt;- sterm_all_unique$query_name %&gt;% gsub(&quot;gnl\\|extdb\\|&quot;,&quot;S_thermophilus_RMK202_&quot;,.)

ldel_all_unique &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;query_name&quot;,&quot;seed_eggNOG_ortholog&quot;,&quot;seed_ortholog_evalue&quot;,&quot;seed_ortholog_score&quot;,&quot;best_tax_level&quot;,&quot;Preferred_name&quot;,&quot;GOs&quot;,&quot;EC&quot;,&quot;KEGG_ko&quot;,&quot;KEGG_Pathway&quot;,&quot;KEGG_Module&quot;,&quot;KEGG_Reaction&quot;,&quot;KEGG_rclass&quot;,&quot;BRITE&quot;,&quot;KEGG_TC&quot;,&quot;CAZy&quot;,&quot;BiGG_Reaction&quot;,&quot;tax_scope&quot;,&quot;eggNOG OGs&quot;,&quot;bestOG&quot;,&quot;COG_Functional_Category&quot;,&quot;eggNOG free text description&quot;),skip=4)
# RMK202_both_eggnog[RMK202_both_eggnog$query_name %in% SNPeff_final$finalName,]
ldel_all_unique$final_name &lt;- ldel_all_unique$query_name %&gt;% gsub(&quot;gnl\\|extdb\\|&quot;,&quot;L_delbrueckii_RMK202_&quot;,.)


all_eggnog &lt;- rbind(sterm_all_unique,ldel_all_unique)

all_eggnog[grep(&quot;lac&quot;,all_eggnog$Preferred_name,ignore.case = TRUE),]

###---------------------------------
#feature information
###---------------------------------

feature_info &lt;- merge_RNAseq_normalized %&gt;% select(gene_complete,species,feature) %&gt;% unique()

###---------------------------------
#cumulative expression
###---------------------------------
# merge_RNAseq_normalized$gene_complete &lt;- paste0(merge_RNAseq_normalized$species,&quot;-&quot;,merge_RNAseq_normalized$gene)
# 
# merge_RNAseq_normalized_ordered &lt;- merge_RNAseq_normalized[order(merge_RNAseq_normalized$gene),] %&gt;% as.data.frame()
# merge_RNAseq_normalized_ordered$replication &lt;- as.character(merge_RNAseq_normalized_ordered$replication)
#  merge_RNAseq_normalized %&gt;% as.data.frame() %&gt;% group_by(gene) %&gt;% summarise(cummulative_count=sum(norm_count),median_count=median(norm_count),sd_count=sd(norm_count)) %&gt;% mutate(sd_count_rel=sd_count/median_count)
# 
#  
#  merge_RNAseq_normalized %&gt;% group_by(gene) %&gt;% summarise(cummulative_count=max(norm_count))
# 
# merge_RNAseq_normalized %&gt;% as.tib
# 
# merge_RNAseq_normalized$gene
# merge_RNAseq_normalized$norm_count
#  merge_RNAseq_normalized %&gt;% group_by(gene)  %&gt;% summarise(cummulative_count=sum(norm_count))
# 
# table(merge_RNAseq_normalized$gene)
# # merge_RNAseq_normalized_extended$gene_complete &lt;- paste0(merge_RNAseq_normalized$species,&quot;-&quot;,merge_RNAseq_normalized$gene)
# merge_RNAseq_normalized
merge_RNAseq_cumulative_tmp &lt;- merge_RNAseq_normalized %&gt;% dplyr::group_by(gene_complete) %&gt;% summarise(cummulative_count=sum(norm_count),median_count=median(norm_count),sd_count=sd(norm_count)) %&gt;% mutate(sd_count_rel=sd_count/median_count)
# merge_RNAseq_cumulative$species &lt;- str_split_fixed(merge_RNAseq_cumulative$gene_complete, &quot;_&quot;, 5)[,1]
merge_RNAseq_cumulative &lt;- merge(merge_RNAseq_cumulative_tmp,feature_info,by=&quot;gene_complete&quot;)
table(merge_RNAseq_cumulative$species)

# merge_RNAseq_cumulative %&gt;% filter(species==&quot;S. thermophilus&quot;)

ggplot(merge_RNAseq_cumulative,aes(x=cummulative_count))+geom_density()+theme_classic()+facet_grid(~species,scales = &quot;free&quot;)+coord_trans(x=&quot;log2&quot;)

# ggplot(merge_RNAseq_cumulative,aes(x=median_count))+geom_density()+theme_classic()+facet_grid(~species,scales = &quot;free&quot;)#+coord_trans(x=&quot;log2&quot;)

##only Sterm
merge_RNAseq_cumulative_sterm &lt;- merge_RNAseq_cumulative %&gt;% filter(species==&quot;sterm&quot;)

ggplot(merge_RNAseq_cumulative_sterm,aes(x=median_count))+geom_density()+theme_classic()+facet_grid(~species,scales = &quot;free&quot;)+coord_trans(x=&quot;log10&quot;)

##median vs. sd

plot_point &lt;- ggplot(merge_RNAseq_cumulative_sterm,aes(x=median_count,y=sd_count_rel))+geom_point()+theme_classic()+coord_trans(x=&quot;log10&quot;,y=&quot;log10&quot;)
# install.packages(&quot;ggplotly&quot;)
library(plotly)
plot_point
# fig &lt;- plotly::ggplotly(plot_point)
# fig
library(ggsignif)
library(ggpubr)

##------------------- remove outliers

merge_RNAseq_cumulative_sterm_cleaned &lt;- merge_RNAseq_cumulative_sterm %&gt;% filter(sd_count_rel&lt;1000)
plot_point &lt;- ggplot(merge_RNAseq_cumulative_sterm_cleaned,aes(x=median_count,y=sd_count_rel,fill=feature,color=feature))+geom_point(alpha=0.5)+theme_classic()+coord_trans(x=&quot;log10&quot;,y=&quot;log10&quot;)
plot_point

ggplot(merge_RNAseq_cumulative_sterm_cleaned,aes(x=median_count,fill=feature,color=feature))+geom_density(alpha=0.3)+theme_classic()+facet_grid(~feature,scales = &quot;free&quot;)+coord_trans(x=&quot;log10&quot;)


ggplot(merge_RNAseq_cumulative_sterm_cleaned,aes(x=feature,y=median_count,fill=feature))+geom_boxplot()+theme_classic()+coord_trans(y=&quot;log10&quot;)+stat_compare_means( label = &quot;p.signif&quot;)
ggplot(merge_RNAseq_cumulative_sterm_cleaned,aes(x=feature,y=sd_count_rel,fill=feature))+geom_boxplot()+theme_classic()+coord_trans(y=&quot;log10&quot;)+stat_compare_means( label = &quot;p.signif&quot;)



# plotPseudogene &lt;- ggplot(merge_RNAseq_normalized_extended,aes(x=time,y=norm_count,color=feature))+facet_wrap(~species,ncol=1,scales=&quot;free_x&quot;)+geom_boxplot(alpha=0.7)+theme_classic()+coord_trans( y=&quot;log2&quot;)+scale_color_manual(values=c(&quot;grey44&quot;,&quot;grey77&quot;))+ stat_compare_means(aes(group = feature), label = &quot;p.signif&quot;)+labs(x=&quot;incubation time&quot;,y=&quot;normalized gene expression&quot;)+theme(legend.title = element_blank())+scale_y_continuous(breaks=c(5,500,50000,1800000))

##------------------- add eggnog information
merge_RNAseq_cumulative_cleaned &lt;- merge_RNAseq_cumulative %&gt;% filter(sd_count_rel&lt;1000) %&gt;% filter(median_count&gt;0.001)

all_eggnog_sub &lt;- all_eggnog %&gt;% select(final_name,Preferred_name,`eggNOG free text description`)
# all_eggnog$`eggNOG free text description`
dim(merge_RNAseq_cumulative_cleaned)
merge_RNAseq_cumulative_cleaned_eggnog &lt;- merge(merge_RNAseq_cumulative_cleaned,all_eggnog_sub,by.x=&quot;gene_complete&quot;,by.y=&quot;final_name&quot;,all.x = TRUE)
dim(merge_RNAseq_cumulative_cleaned)
dim(merge_RNAseq_cumulative_cleaned_eggnog)

plot_point &lt;- ggplot(merge_RNAseq_cumulative_cleaned_eggnog,aes(x=median_count,y=sd_count_rel,fill=feature,color=feature))+geom_point(alpha=0.6)+theme_classic()+coord_trans(x=&quot;log10&quot;,y=&quot;log10&quot;)+facet_grid(~species,scales = &quot;free&quot;)
plot_point


plot_point &lt;- ggplot(merge_RNAseq_cumulative_cleaned_eggnog,aes(x=median_count,y=sd_count_rel,fill=feature,color=feature))+geom_point(alpha=0.6)+theme_classic()+coord_trans(x=&quot;log10&quot;,y=&quot;log10&quot;)+facet_wrap(~species,scales = &quot;free&quot;)
plot_point

# fig &lt;- plotly::ggplotly(plot_point)
# htmlwidgets::saveWidget(fig, &quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/geneExpression.html&quot;)


merge_RNAseq_cumulative_cleaned_eggnog[grep(&quot;lac&quot;,merge_RNAseq_cumulative_cleaned_eggnog$Preferred_name,ignore.case = TRUE),]

merge_RNAseq_cumulative_cleaned_eggnog$coloring &lt;- &quot;none&quot;

merge_RNAseq_cumulative_cleaned_eggnog[grep(&quot;lac&quot;,merge_RNAseq_cumulative_cleaned_eggnog$Preferred_name,ignore.case = TRUE),&quot;coloring&quot;] &lt;- &quot;lactose&quot;

table(merge_RNAseq_cumulative_cleaned_eggnog$coloring)


plot_point &lt;- ggplot(merge_RNAseq_cumulative_cleaned_eggnog,aes(x=median_count,y=sd_count_rel,fill=coloring,color=coloring,size=coloring))+geom_point(alpha=0.5)+theme_classic()+coord_trans(x=&quot;log10&quot;,y=&quot;log10&quot;)+facet_grid(~species,scales = &quot;free&quot;)+scale_size_manual(values=c(4,0.5))
plot_point</code></pre>
<div id="pepidase-genes" class="section level2">
<h2>Pepidase genes</h2>
<pre class="r"><code>merge_RNAseq_cumulative_cleaned_eggnog_peptidase &lt;- merge_RNAseq_cumulative_cleaned_eggnog %&gt;% as_tibble()  #%&gt;% filter(grepl(&quot;pep&quot;,Preferred_name))

merge_RNAseq_cumulative_cleaned_eggnog_peptidase[grep(&quot;pep&quot;,merge_RNAseq_cumulative_cleaned_eggnog_peptidase$Preferred_name,ignore.case = TRUE),]

merge_RNAseq_cumulative_cleaned_eggnog_peptidase$coloring &lt;- &quot;none&quot;

merge_RNAseq_cumulative_cleaned_eggnog_peptidase[grep(&quot;pep&quot;,merge_RNAseq_cumulative_cleaned_eggnog_peptidase$Preferred_name,ignore.case = TRUE),&quot;coloring&quot;] &lt;- &quot;pep&quot;

table(merge_RNAseq_cumulative_cleaned_eggnog_peptidase$coloring)

plot_point_pep &lt;- ggplot(merge_RNAseq_cumulative_cleaned_eggnog_peptidase,aes(x=median_count,y=sd_count_rel,fill=coloring,color=coloring,size=coloring))+geom_point(alpha=0.5)+theme_classic()+coord_trans(x=&quot;log10&quot;,y=&quot;log10&quot;)+facet_grid(~species,scales = &quot;free&quot;)+scale_size_manual(values=rev(c(4,0.5)))
plot_point_pep


##----------------------------per timepoint
merge_RNAseq_cumulative_cleaned_eggnog_peptidase_prep &lt;- merge_RNAseq_cumulative_cleaned_eggnog_peptidase %&gt;% select(-species,-feature)

merge_RNAseq_normalized_extended_ext &lt;- merge_RNAseq_normalized_extended %&gt;% as_tibble() %&gt;% mutate(gene_complete=paste0(species_extensive,&quot;_&quot;,gene_short))#%&gt;% left_join(.,merge_RNAseq_cumulative_cleaned_eggnog_peptidase)


merge_RNAseq_normalized_extended_ext_full &lt;- left_join(merge_RNAseq_normalized_extended_ext,merge_RNAseq_cumulative_cleaned_eggnog_peptidase_prep) %&gt;% mutate(feature=ifelse(grepl(&quot;pep&quot;,Preferred_name),&quot;Peptidase&quot;,&quot;none&quot;))
# merge_RNAseq_normalized_extended_ext$gene_complete %in% merge_RNAseq_cumulative_cleaned_eggnog_peptidase$gene_complete
table(merge_RNAseq_normalized_extended_ext_full$feature)

plotPseudogene &lt;- ggplot(merge_RNAseq_normalized_extended_ext_full,aes(x=time,y=norm_count,color=feature))+facet_wrap(~species,ncol=1,scales=&quot;free_x&quot;)+geom_boxplot(alpha=0.7)+theme_classic()+coord_trans( y=&quot;log2&quot;)+scale_color_manual(values=c(&quot;grey44&quot;,&quot;grey77&quot;))+
stat_compare_means(aes(group = feature), label = &quot;p.signif&quot;)+labs(x=&quot;incubation time&quot;,y=&quot;normalized gene expression&quot;)+theme(legend.title = element_blank())+scale_y_continuous(breaks=c(5,500,50000,1800000))
plotPseudogene
# ggplot(merge_RNAseq_normalized_extended,aes(x=feature,y=norm_count,color=feature))+facet_wrap(time~species,ncol=16,scales=&quot;free_x&quot;)+geom_boxplot()+theme_classic()+coord_trans( y=&quot;log2&quot;)+scale_color_manual(values=c(&quot;grey44&quot;,&quot;grey77&quot;))


pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/plotPseudogene_expression.pdf&quot;,width=7,height=4.5)

plotPseudogene

dev.off()</code></pre>
<p>##prepare gene tables</p>
<p>somehow in the appendix there are the wrong gene tables prepared.
Therefore I will prepare the now gene tables for the R-shiny.</p>
<pre class="bash"><code>sed &#39;s/gnl|extdb|/S_thermophilus_RMK202_/g&#39; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap.faa |sed &#39;s/ /-/g&#39; &gt; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_preped.faa

sed &#39;s/gnl|extdb|/L_delbrueckii_RMK202_/g&#39; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap.faa |sed &#39;s/ /-/g&#39; &gt; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_preped.faa

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_preped.faa  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap_preped.faa &gt; /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/both_species_RMK202_pgap_preped.faa</code></pre>
<pre class="bash"><code>scp -r vincent@130.223.51.66:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/both_species_RMK202_pgap_preped.faa ~/Desktop/Projects/2019_RNAseq_rmk202/data/
#L_delbrueckii_RMK202_pgaptmp_000001-


grep &quot;^##&quot; -v  ~/Desktop/Projects/2019_RNAseq_rmk202/data/out.emapper.annotations |cut -f 1|cut -d &#39;-&#39; -f 1 &gt; ~/Desktop/Projects/2019_RNAseq_rmk202/data/out.emapper.annotations_tmp1

grep &quot;^##&quot; -v  ~/Desktop/Projects/2019_RNAseq_rmk202/data/out.emapper.annotations &gt; ~/Desktop/Projects/2019_RNAseq_rmk202/data/out.emapper.annotations_tmp2

paste ~/Desktop/Projects/2019_RNAseq_rmk202/data/out.emapper.annotations_tmp1 ~/Desktop/Projects/2019_RNAseq_rmk202/data/out.emapper.annotations_tmp2 |sed &#39;s/^#query/gene/g&#39; &gt;  ~/Desktop/Projects/2019_RNAseq_rmk202/data/out.emapper.annotations_complite
</code></pre>
</div>
</div>
<div id="backup-analyis" class="section level1">
<h1>Backup analyis</h1>
<div id="annotations" class="section level2">
<h2>Annotations</h2>
<p>###PROKKA of strains</p>
<pre class="bash"><code>
conda activate PROKKA
species=Sterm
strainzzz=S24855
export PATH=/home/vincent/apps/ncbi-blast-2.10.0+/bin/:$PATH
speciesss=$(echo &quot;S.thermophilus&quot;)
genus=Streptococcus
echo ${speciesss}
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/
#rm -r /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/
for strainzzz in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/ |grep &quot;fasta$&quot; |sed &#39;s/.fasta//g&#39;)
do
#culture=$(awk -F &quot;\t&quot; -v strainsss=&quot;$strainzzz&quot; &#39;{OFS=&quot;\t&quot;}{if($11==strainsss)print $4}&#39; /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed &#39;s/RMK//g&#39;)
#ONLYsTRAIN=$(echo ${strainzzz} |cut -d _ -f 4)
echo -e ${strainzzz} 

#genomeszzz=$(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes/ |grep  &quot;_${strainzzz}&quot; )
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${strainzzz}*
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/

#prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${strainzzz}/ --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${strainzzz}.fasta
conda activate PROKKA
prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}// --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/${strainzzz}.fasta

done #strainzzz
##-----------------------
##l.delbrueckii
##-----------------------

conda activate PROKKA

strainzzz=24855
export PATH=/home/vincent/apps/ncbi-blast-2.10.0+/bin/:$PATH
speciesss=$(echo &quot;L.delbrueckii&quot;)
species=Ldel
genus=Streptococcus
echo ${speciesss}
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/
#rm -r /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/

for strainzzz in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/ |grep &quot;fasta$&quot; |sed &#39;s/.fasta//g&#39;)
do
#culture=$(awk -F &quot;\t&quot; -v strainsss=&quot;$strainzzz&quot; &#39;{OFS=&quot;\t&quot;}{if($11==strainsss)print $4}&#39; /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed &#39;s/RMK//g&#39;)
#ONLYsTRAIN=$(echo ${strainzzz} |cut -d _ -f 4)
echo -e ${strainzzz} 

#genomeszzz=$(ls /data/Project/2020_StarterCultureDiversity/08_tree/ParSNP/20200516/${speciesss}/01_genomes/ |grep  &quot;_${strainzzz}&quot; )
#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${strainzzz}*
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/

#prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${strainzzz}/ --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/Sterm/210315_final_startAligned/${strainzzz}.fasta

prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}// --locustag ${strainzzz} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/${strainzzz}.fasta

done #strainzzz

##-------------------clean up
species=Ldel
for strainzzz in $(echo &quot;L21748
L21753
L21780
L11077
L11063&quot;)
do
echo -e ${strainzzz} 
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}// 
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection//${species}/210315_final_startAligned/${strainzzz}.fasta
#head /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection/${species}/210315_final_startAligned/${strainzzz}

done #strainzzz


###================================================
##create FAA and FNA folders
###================================================

for species in $(echo &quot;Ldel Sterm&quot;)
#for species in $(echo &quot;Sterm&quot;)
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all*
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/{01_all_FAA,01_all_FNA,01_all_GFF,01_all_FFN}

for strainzzz in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection//${species}/210315_final_startAligned/ |grep &quot;.fasta$&quot;|sed &#39;s/.fasta//g&#39;)
do

#culture=$(awk -F &quot;\t&quot; -v strainsss=&quot;$strainzzz&quot; &#39;{OFS=&quot;\t&quot;}{if($11==strainsss)print $4}&#39; /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed &#39;s/RMK//g&#39;)
ONLYsTRAIN=$(echo ${strainzzz} |cut -d _ -f 3)
echo -e ${strainzzz} 

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/${strainzzz}.fna

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.ffn &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/${strainzzz}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.faa &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/${strainzzz}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.gff &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/${strainzzz}.gff

done #strainzzz
done #species

##--------------------
##check DNAa
##--------------------
species=Sterm
species=Ldel
for strainzzz in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/210201_RMKcollection//${species}/210315_final_startAligned/ |grep &quot;.fasta$&quot;|sed &#39;s/.fasta//g&#39;)
do
#culture=$(awk -F &quot;\t&quot; -v strainsss=&quot;$strainzzz&quot; &#39;{OFS=&quot;\t&quot;}{if($11==strainsss)print $4}&#39; /data/Project/2020_StarterCultureDiversity/99_log/isolatedStrainsRMK.csv|sed &#39;s/RMK//g&#39;)
#ONLYsTRAIN=$(echo ${strainzzz} |cut -d _ -f 3)
#echo -e ${strainzzz} 

#seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${strainzzz}/PROKKA*.fna |head -1

grep &quot;dnaA&quot; -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${strainzzz}/PROKKA*.gff |head -1 |cut -d &#39;:&#39; -f 2
#grep &quot;dnaA&quot; i /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${strainzzz}//PROKKA*.gff |head -1

done #strainzzz
</code></pre>
<p>###Virsorter</p>
<pre class="bash"><code>
conda activate vs2

#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter

for species in $(echo &quot;Ldel Sterm&quot; )
do
for sample in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ |grep &quot;.fna$&quot;|sed &#39;s/.fna//g&#39;|grep &quot;24756&quot; -A 100)
do
echo &quot;=========================================&quot;
echo $sample
echo &quot;=========================================&quot;

leterzzz=$(echo $species |head -c 1)
sampsss=$(echo ${sample}|sed &quot;s/${leterzzz}//g&quot;)


  rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${species}/${sampsss}
  mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${species}/${sampsss}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${species}/${sampsss}

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/${sample}.fna . 

virsorter run -w /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${species}/${sampsss}/ -i ${sample}.fna -j 5 all

#grep -v &quot;^#&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Virsorter/${sample}/VIRSorter_global-phage-signal.csv 
 
 
done #samples
done #species
</code></pre>
<div id="busco" class="section level3">
<h3>BUSCO</h3>
<pre class="bash"><code>
speciesss=Sterm
genemess=S24855

grep &quot;${genemess}&quot; -v /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis.txt &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis_tmp.txt
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis_tmp.txt &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis.txt
##-----------------------------------
##script
##-----------------------------------
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis.txt

for speciesss in $(echo &quot;Sterm Ldel&quot;)
do
echo &quot;##------------------------&quot;
echo ${speciesss}
echo &quot;##------------------------&quot;

for genemess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}/01_all_FAA/ |sed &#39;s/.faa//g&#39;)
  do
  #genemess=L_delbrueckii_RMK202_mag.faa
  #genomesssss=$(echo ${genemess}|cut -d &#39;.&#39; -f 1)
  echo $genemess
  
  rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/${genemess}

  mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/${genemess}
  cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/
   export BUSCO_CONFIG_FILE=/home/vincent/apps/busco_v4/busco/config/myconfig.ini
  python3.6 /home/vincent/apps/busco_v4/busco/bin/busco -f -m prot -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}/01_all_FAA/${genemess}.faa -o ${genemess} -l lactobacillales -c 10
  #-i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FAA/${genemess}.faa -o ${genemess} -l /home/vincent/apps/busco/lactobacillales_odb9 -m prot

  #grep -v &quot;^#&quot; run_${genomesssss}/short_summary_${genomesssss}.txt| grep &quot;C:&quot; -A 7 
  all_INFO=$(grep -v &quot;^#&quot; ${genemess}/run_lactobacillales_odb10/short_summary.txt| grep &quot;C:&quot; | sed &quot;s/^[ \t]*//&quot;  )
  complet=$(echo ${all_INFO} | awk -F &quot;[,[]&quot; &#39;{print $1}&#39;|sed &#39;s/%//g&#39;|sed &#39;s/C://g&#39; )
  single=$(echo ${all_INFO} | awk -F &quot;[,[]&quot; &#39;{print $2}&#39;|sed &#39;s/%//g&#39;|sed &#39;s/S://g&#39;)
  dupli=$(echo ${all_INFO} | awk -F &quot;[,[]&quot; &#39;{print $3}&#39;|sed &#39;s/%]//g&#39;|sed &#39;s/D://g&#39;)
  fragm=$(echo ${all_INFO} | awk -F &quot;[,[]&quot; &#39;{print $4}&#39;|sed &#39;s/%//g&#39;|sed &#39;s/F://g&#39;)
  Missi=$(echo ${all_INFO} | awk -F &quot;[,[]&quot; &#39;{print $5}&#39;|sed &#39;s/%//g&#39;|sed &#39;s/M://g&#39;)
  numbi=$(echo ${all_INFO} | awk -F &quot;[,[]&quot; &#39;{print $6}&#39;|sed &#39;s/n://g&#39;)
  
 echo -e ${speciesss}&quot;\t&quot;${genemess}&quot;\t&quot;${complet}&quot;\t&quot;${single}&quot;\t&quot;${dupli}&quot;\t&quot;${fragm}&quot;\t&quot;${Missi}&quot;\t&quot;${numbi}  \
 &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis.txt

  done
done ## species

#sed -i  &#39;s/Streptococcus_thermophilus_RMK202_pgap/S_thermophilus_mag_rmk202/g&#39; /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt
#sed -i  &#39;s/Lactobacillus_delbrueckii_RMK202_pgap/L_delbrueckii_mag_rmk202/g&#39; /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt
</code></pre>
<div id="genome-stats" class="section level4">
<h4>genome stats</h4>
<pre class="bash"><code>
##---------------
##genome
##---------------

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/all_assembly_analysis.txt
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats
for species in $(echo &quot;Sterm Ldel&quot;)
do
for id in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ |sed &#39;s/.fna//g&#39;)
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
# /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/
 
   contiNumber=$(grep &quot;&gt;&quot; -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA//${id}.fna) 
   GenomeLength=$(grep -v &quot;^&gt;&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA//${id}.fna  |wc -c)
   
   # contiNumber=$(grep &quot;&gt;&quot; -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta) 
   #GenomeLength=$(grep -v &quot;^&gt;&quot; /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta  |wc -c)
    
 echo -e ${id}&quot;\t&quot;${contiNumber}&quot;\t&quot;${GenomeLength}  &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/all_assembly_analysis.txt

#&quot;\t&quot;${single}&quot;\t&quot;${dupli}&quot;\t&quot;${fragm}&quot;\t&quot;${Missi}&quot;\t&quot;${numbi}  \
    done
done #species
</code></pre>
<p>####mapping stats</p>
<pre class="bash"><code>
 
for species in $(echo &quot;Sterm Ldel&quot;)
do
for id in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ |grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39;|grep &quot;MAG&quot; -v|grep &quot;S50&quot; -v|grep &quot;S72&quot; -v)
  do

#samplezzKurz=$(echo $id |cut -d &#39;_&#39; -f 4)
samplezzKurz=$(echo $id |sed &#39;s/^L//g&#39;|sed &#39;s/^S//g&#39;)
  echo ==========================
   echo ${id}
   echo ==========================


 rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}
  
  
  bwa index /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA//${id}.fna
  
  bwa mem -t 32 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA//${id}.fna \
  /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${samplezzKurz}/FAM${samplezzKurz}_R1_val_1.fq.gz /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${samplezzKurz}/FAM${samplezzKurz}_R2_val_2.fq.gz | samtools sort -@32 -O BAM -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}/${id}_rawIllumina_bwamem_sorted.bam - 

##-----Qualimap

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc

qualimap bamqc -bam /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}/${id}_rawIllumina_bwamem_sorted.bam -outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc --java-mem-size=80G

    
    done
done #species

##---------------------
##get data
##---------------------

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_genome_analysis.txt
for species in $(echo &quot;Sterm Ldel&quot;)
do
for id in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ |grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39;)
  do

#GCss=$(grep &quot;GC percentage&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt| sed &#39;s/GC percentage = //g&#39;| sed &quot;s/^[ \t]*//&quot;|sed &#39;s/%//g&#39;  )

coveragesss=$(grep &quot;mean coverageData = &quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt| sed &#39;s/mean coverageData = //g&#39;| sed &quot;s/^[ \t]*//&quot;|sed &#39;s/X//g&#39;  )

#readsss=$(grep &quot;number of reads = &quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt |awk -F &quot;number of reads = &quot; &#39;{print $2}&#39;|sed &#39;s/,//g&#39;)

#mappedReadsss=$(grep &quot;number of mapped reads = &quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt |awk -F &quot;number of mapped reads = &quot; &#39;{print $2}&#39;|awk -F &quot; &quot; &#39;{print $1}&#39;|sed &#39;s/,//g&#39;)

percentmappedReadsss=$(grep &quot;number of mapped reads = &quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats/PostaAssembly/${id}//bamqc/genome_results.txt |awk -F &quot;number of mapped reads = &quot; &#39;{print $2}&#39;|awk -F &quot; &quot; &#39;{print $2}&#39;|sed &#39;s/,//g&#39;|sed &#39;s/(//g&#39;|sed &#39;s/%)//g&#39;)

GCss=$(infoseq /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA//${id}.fna -noheading -nodatabase -nolength -nousa -nohead -noacc -notype -nodesc -delimiter &#39;_&#39; | awk -F &quot; &quot; &#39;{sum+=$2} END {print sum/NR}&#39;)

#grep -A 10000 &quot;&gt;&gt;&gt;&gt;&gt;&gt;&gt; Coverage per contig&quot; ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt| grep -v &quot;&gt;&gt;&gt;&gt;&gt;&gt;&gt; Coverage per contig&quot;| sed &quot;s/^[ \t]*//&quot; |sed -r &#39;/^\s*$/d&#39;|cut -f 1,4,5 |awk -F &quot;\t&quot; -v SPECIES=&quot;$names&quot; &#39;{OFS=&quot;\t&quot;}{print SPECIES,$0}&#39; |head
 
 
done
done #species
</code></pre>
</div>
<div id="bring-together" class="section level4">
<h4>bring together</h4>
<pre class="bash"><code>
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal
cp all_assembly_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/BUSCO_annotation/all_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
cp all_genome_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
cp all_annotation_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
cp all_annotation_transposases_analysis.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/


mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/
scp  -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/ ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/
</code></pre>
<pre class="r"><code># all_sterm_strain_info_ZOOMED &lt;- read_csv(&quot;Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv&quot;,  col_types = cols(starterCulture = col_character(),  strain = col_character())) %&gt;% select(c(strain,Cheese,starterCulture,species))
# 
# all_sterm_strain_info_ZOOMED$short &lt;- plyr::revalue(all_sterm_strain_info_ZOOMED$species, c(&quot;L.delbrueckii&quot;=&quot;L&quot;,&quot;S.thermophilus&quot;=&quot;S&quot;))
# all_sterm_strain_info_ZOOMED$name &lt;- paste0(all_sterm_strain_info_ZOOMED$short,all_sterm_strain_info_ZOOMED$strain)
# all_sterm_strain_info_ZOOMED_final &lt;- all_sterm_strain_info_ZOOMED %&gt;% select(name,Cheese,starterCulture)
# 
# write.csv(all_sterm_strain_info_ZOOMED_final,file=&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/origin_genomes_all.csv&quot;,quote = FALSE,row.names = FALSE)



####--------------------------------------
library(readr)
library(plyr)
library(tidyverse)
origin_genomes_all &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/origin_genomes_all.csv&quot;,  col_types = cols(starterCulture = col_character()))

ONT_assembly_stats &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/ONT_assembly_stats.txt&quot;, &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE) %&gt;% filter(largestContigSize&gt;1700000)
genome_stats &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal//all_assembly_analysis.txt&quot;, &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;strain&quot;,&quot;contigNumber&quot;,&quot;genomeSize&quot;),  trim_ws = TRUE)
Genes_stats &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal//all_annotation_analysis.txt&quot;, &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;strain&quot;,&quot;genes&quot;,&quot;tRNA&quot;,&quot;rRNA&quot;,&quot;transposons&quot;),  trim_ws = TRUE)
BUSCO_stats &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal//all_analysis.txt&quot;, &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;species&quot;,&quot;strain&quot;,&quot;completness&quot;,&quot;singleCopy&quot;,&quot;duplicated&quot;,&quot;fragmentation&quot;,&quot;Missing&quot;,&quot;numberGenesTested&quot;),  trim_ws = TRUE) %&gt;% filter(strain!=&quot;L11063&quot;)
mapping_stats &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal//all_genome_analysis.txt&quot;, &quot;\t&quot;, escape_double = FALSE, col_names =c(&quot;strain&quot;,&quot;GC&quot;,&quot;IlluminaCoverage&quot;,&quot;percentMapped&quot;),  trim_ws = TRUE)



tmp &lt;- merge(BUSCO_stats,genome_stats,by=&quot;strain&quot;,all.x=TRUE) %&gt;% as.tibble()
table(tmp$species)
tmp &lt;- merge(tmp,Genes_stats,by=&quot;strain&quot;,all.x=TRUE) %&gt;% as.tibble()
table(tmp$species)
tmp &lt;- merge(tmp,origin_genomes_all,by.x=&quot;strain&quot;,by.y = &quot;name&quot;,all.x=TRUE) %&gt;% as.tibble()
table(tmp$species)

GenomeAssemblyStats_tmp &lt;- merge(tmp,mapping_stats,by=&quot;strain&quot;)

# GenomeAssemblyStats_tmp$strain_short &lt;- str_split_fixed(GenomeAssemblyStats_tmp$strain,&quot;_&quot;,4)[,4]

GenomeAssemblyStats &lt;- merge(GenomeAssemblyStats_tmp,ONT_assembly_stats,by.x=&quot;strain&quot;,by.y=&quot;sample&quot;,all.x=TRUE)
GenomeAssemblyStats$CIRCnUM &lt;- revalue(GenomeAssemblyStats$Circularity, c(&quot;Y&quot;=&quot;1&quot;,&quot;N&quot;=&quot;0&quot;)) %&gt;% as.numeric()
# GenomeAssemblyStats$Sequencing &lt;- ifelse(is.na(GenomeAssemblyStats$Circularity),&quot;Illumina&quot;,ifelse(GenomeAssemblyStats$Circularity==&quot;Y&quot;,&quot;ONT&quot;,&quot;Illumina&quot;))
GenomeAssemblyStats$Sequencing &lt;- ifelse(is.na(GenomeAssemblyStats$Circularity),&quot;Illumina&quot;,&quot;ONT&quot;)

GenomeAssemblyStats$additional_circular_contigs &lt;- GenomeAssemblyStats$circular_contigs-GenomeAssemblyStats$CIRCnUM
GenomeAssemblyStats &lt;- GenomeAssemblyStats %&gt;%  select(-CIRCnUM) %&gt;%  select(-circular_contigs)
# GenomeAssemblyStats$starterCulture &lt;- str_split_fixed(GenomeAssemblyStats$strain,&quot;_&quot;,4)[,3]
# GenomeAssemblyStats &lt;- GenomeAssemblyStats %&gt;%  select(-strain)%&gt;% dplyr::rename(strain=&quot;strain_short&quot;)


GenomeAssemblyStats$Cheese &lt;- revalue(GenomeAssemblyStats$starterCulture,c(&quot;101&quot;=&quot;emmentaler&quot;,&quot;105&quot;=&quot;emmentaler&quot;,&quot;115&quot;=&quot;emmentaler&quot;,&quot;124&quot;=&quot;emmentaler&quot;,&quot;150&quot;=&quot;emmentaler&quot;,&quot;190&quot;=&quot;emmentaler&quot;,&quot;202&quot;=&quot;gruyere&quot;,&quot;203&quot;=&quot;gruyere&quot;,&quot;280&quot;=&quot;gruyere&quot;,&quot;302&quot;=&quot;sbrinz&quot;,&quot;305&quot;=&quot;sbrinz&quot;))


## add lineage information

##--------------------------------
##ad poppunk clustering
##--------------------------------
Ldel_clusters_RMKS &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_ldel.csv&quot;, col_names = c(&quot;strain&quot;,&quot;lineage&quot;), col_types = cols(lineage = col_character(),  strain = col_character()), skip = 1) %&gt;%  replace(is.na(.), &quot;Unknown&quot;) %&gt;% mutate(lineage=as.character(as.numeric(lineage)+12))
Sterm_clusters_RMKS &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/06_PopPUNK/Sterm_RMK/20210128_easyrun/20210128_easyrun_clusters_RMKS_sterm.csv&quot;, col_names = c(&quot;strain&quot;,&quot;lineage&quot;), col_types = cols(lineage = col_character(),  strain = col_character()), skip = 1)%&gt;%replace(is.na(.), &quot;Unknown&quot;) 
both_clusters_RMKS &lt;- rbind(Sterm_clusters_RMKS,Ldel_clusters_RMKS)
table(Ldel_clusters_RMKS$lineage)



dim(GenomeAssemblyStats)
GenomeAssemblyStats_tmp &lt;- merge(GenomeAssemblyStats,both_clusters_RMKS,by=&quot;strain&quot;)
dim(GenomeAssemblyStats_tmp)

GenomeAssemblyStats_tmp$Ref &lt;- ifelse(GenomeAssemblyStats_tmp$strain %in% c(&quot;S24736&quot;,&quot;S24735&quot;,&quot;S24734&quot;,&quot;S24750&quot;,&quot;S24739&quot;,&quot;S24745&quot;,&quot;S24728&quot;,&quot;S24742&quot;,&quot;S24746&quot;,&quot;S24749&quot;,&quot;S24730&quot;,&quot;S24743&quot;,&quot;L24790&quot;,&quot;L24793&quot;),&quot;Ref&quot;,&quot;no&quot;)
table(GenomeAssemblyStats_tmp$Ref)

GenomeAssemblyStats &lt;- GenomeAssemblyStats_tmp
##------------------------
##remove 
##------------------------

 GenomeAssemblyStats %&gt;% filter(completness&lt;95|duplicated&gt;5)
GenomeAssemblyStats_cleaned &lt;- GenomeAssemblyStats %&gt;% filter(completness&gt;95&amp;duplicated&lt;5)
GenomeAssemblyStats_cleaned %&gt;% filter(completness&lt;95|duplicated&gt;5)
dput(colnames(GenomeAssemblyStats_cleaned))

write.csv(GenomeAssemblyStats_cleaned,file=&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal///genomeAssemblyStats_long.csv&quot;,quote = FALSE,row.names = FALSE)


GenomeAssemblyStats_cleaned &lt;- GenomeAssemblyStats_cleaned %&gt;% select(c(&quot;strain&quot;,&quot;Cheese&quot; ,&quot;starterCulture&quot;,&quot;species&quot;,&quot;lineage&quot;,&quot;Ref&quot;, 
&quot;Sequencing&quot;, &quot;completness&quot;,&quot;contigNumber&quot;, 
&quot;genomeSize&quot;, &quot;genes&quot;, &quot;tRNA&quot;, &quot;rRNA&quot;, &quot;transposons&quot;, &quot;GC&quot;,  &quot;Circularity&quot;, &quot;additional_circular_contigs&quot;))

GenomeAssemblyStats &lt;- GenomeAssemblyStats_cleaned

GenomeAssemblyStats %&gt;% filter(Ref==&quot;Ref&quot;)

##----------------------cleaning--------------------------------
table(GenomeAssemblyStats$strain)
# GenomeAssemblyStats$species &lt;- revalue(GenomeAssemblyStats$species, c(&quot;Sterm&quot;=&quot;S.thermophilus&quot;,&quot;Ldel&quot;=&quot;L.delbrueckii&quot;))

# GenomeAssemblyStats$strain &lt;- revalue(GenomeAssemblyStats$strain, c(&quot;S50&quot;=&quot;mst1&quot;,&quot;S72&quot;=&quot;mst2&quot;,&quot;L70&quot;=&quot;mst3&quot;, &quot;L44&quot;=&quot;mst10&quot;, &quot;L108&quot;=&quot;mst7&quot;, &quot;L99&quot;=&quot;mst8&quot;, &quot;L104&quot;=&quot;mst6&quot;, &quot;L80&quot;=&quot;mst5&quot;, &quot;L35&quot;=&quot;mst9&quot;, &quot;L71&quot;=&quot;mst4&quot;,&quot;L_delbrueckii_mag_rmk202&quot;=&quot;Ldel_mag_202&quot;,&quot;S_thermophilus_mag_rmk202&quot;=&quot;Sterm_mag_202&quot;))

# GenomeAssemblyStats &lt;- GenomeAssemblyStats[-which(GenomeAssemblyStats$strain==&quot;L39&quot;),]
colnames(GenomeAssemblyStats)

# GenomeAssemblyStats_final &lt;- GenomeAssemblyStats %&gt;% select(species,strain,contigNumber,genomeSize,GC,IlluminaCoverage,percentMapped,genes,tRNA,rRNA,transposons,completness)
GenomeAssemblyStats_final &lt;- GenomeAssemblyStats 

##----------------------write table--------------------------------
  GenomeAssemblyStats_final %&gt;% filter(species==&quot;S.thermophilus&quot;)
write.csv(GenomeAssemblyStats_final,file=&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal///genomeAssemblyStats.csv&quot;,quote = FALSE,row.names = FALSE)
# 
# write.xlsx(GenomeAssemblyStats_final,file=&quot;Users//Desktop/presenations/my_presentations/20191007_groupmeeting/figures/genomeAssemblyStats.xls&quot;, sheetName = &quot;Sheet1&quot;,
#   col.names = TRUE, row.names = FALSE, append = FALSE)</code></pre>
<p>####look at circlar plasmids</p>
<p>The following genomes are circlar S13494 S13496 S24736 S24737 S24739
S24745 S24854 S24855</p>
<pre class="bash"><code>mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/cirlar_plasmids
cd  /data/Project/2020_StarterCultureDiversity/09_flye

##--------------------S13494
cat assembly/13494/30-contigger/contigs_stats.txt &gt; cirlar_plasmids/13494_contigs_stats.txt
cat assembly/13494/assembly_graph.gfa &gt; cirlar_plasmids/13494_assembly_graph.gfa

##--------------------S13496
cat assembly/13496//30-contigger/contigs_stats.txt &gt; cirlar_plasmids/13496_contigs_stats.txt
cat assembly/13496/assembly_graph.gfa &gt; cirlar_plasmids/13496_assembly_graph.gfa

##--------------------S24736

cat assembly/24736//30-contigger/contigs_stats.txt &gt; cirlar_plasmids/24736_contigs_stats.txt
cat assembly/24736/assembly_graph.gfa &gt; cirlar_plasmids/24736_assembly_graph.gfa

##--------------------S24737

cat assembly/24737//30-contigger/contigs_stats.txt &gt; cirlar_plasmids/24737_contigs_stats.txt
cat assembly/24737/assembly_graph.gfa &gt; cirlar_plasmids/24737_assembly_graph.gfa

##--------------------S24739

cat assembly/24739//30-contigger/contigs_stats.txt &gt; cirlar_plasmids/24739_contigs_stats.txt
cat assembly/24739/assembly_graph.gfa &gt; cirlar_plasmids/24739_assembly_graph.gfa

##--------------------S24745

cat assembly/24745//30-contigger/contigs_stats.txt &gt; cirlar_plasmids/24745_contigs_stats.txt
cat assembly/24745/assembly_graph.gfa &gt; cirlar_plasmids/24745_assembly_graph.gfa
##!!!!!!no plasmid
rm cirlar_plasmids/24745* 
##--------------------S24854

cat assembly/24854//30-contigger/contigs_stats.txt &gt; cirlar_plasmids/24854_contigs_stats.txt
cat assembly/24854/assembly_graph.gfa &gt; cirlar_plasmids/24854_assembly_graph.gfa

##--------------------S24855

cat assembly/24855//30-contigger/contigs_stats.txt &gt; cirlar_plasmids/24855_contigs_stats.txt
cat assembly/24855/assembly_graph.gfa &gt; cirlar_plasmids/24855_assembly_graph.gfa
</code></pre>
<p>move local to look at with bandage</p>
<pre class="bash"><code>scp  -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/09_flye/cirlar_plasmids ~/Desktop/Projects/2020_StarterCultureDiversity/09_flye/
</code></pre>
<p>#Genome analysis</p>
<p>###Restriction modification</p>
<pre class="bash"><code>/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/


grep &quot;restriction&quot; -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff

grep &quot;restriction&quot; -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff |grep &quot;hsdR&quot; -v |grep &quot;sau3AIR&quot; -v 

hsdR #Type-1 restriction 
sau3AIR # type II
hsdM # type I
mboIR  # type II

grep &quot;HsdM&quot; -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff
grep &quot;Hsds&quot; -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff

grep &quot;HsdM&quot; -A 5 -B 5 -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_GFF/*.gff
</code></pre>
</div>
</div>
</div>
<div id="special-annotations" class="section level2">
<h2>special annotations</h2>
<div id="eps" class="section level3">
<h3>EPS</h3>
<p>maybe I should make genus specific or hmm databases (<a href="https://github.com/tseemann/prokka" class="uri">https://github.com/tseemann/prokka</a>).</p>
<p><a href="https://www.nature.com/articles/srep04974/figures/4" class="uri">https://www.nature.com/articles/srep04974/figures/4</a></p>
<pre class="bash"><code>
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/ |grep &quot;.fna$&quot; |head -1)
do
grep &quot;=eps&quot; -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/S_I_190_24758.fna |awk -F &quot;;gene=&quot; &#39;{print $2}&#39; |cut -d &#39;;&#39; -f 1 |sort|uniq -c


done</code></pre>
</div>
<div id="rgp" class="section level3">
<h3>RGP</h3>
</div>
<div id="bacteriocins" class="section level3">
<h3>Bacteriocins</h3>
</div>
<div id="quorum-sensing-comabcde" class="section level3">
<h3>quorum sensing (ComABCDE)</h3>
</div>
<div id="naturally-competence-system" class="section level3">
<h3>naturally competence system</h3>
</div>
<div id="autolysis" class="section level3">
<h3>autolysis</h3>
</div>
<div id="crispr-system" class="section level3">
<h3>CRISPR-system</h3>
</div>
<div id="plasmid" class="section level3">
<h3>plasmid</h3>
</div>
<div id="prophage" class="section level3">
<h3>prophage</h3>
<p>with Phaster. I use the <a href="https://github.com/boulund/phaster_scripts">API tool</a>.</p>
<pre class="bash"><code>
genomesss=L_I_101_10973
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/ |grep &quot;.fna$&quot;|sed &#39;s/.fna//g&#39; |head -1)
do
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER

#grep &quot;=eps&quot; -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/S_I_190_24758.fna |awk -F &quot;;gene=&quot; &#39;{print $2}&#39; |cut -d &#39;;&#39; -f 1 |sort|uniq -c
echo $genomesss
#/home/vincent/apps/phaster_scripts/phaster.py  --contigs --fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/${genomesss}.fna --database /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt

wget --post-file=&quot;/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/01_all_FNA/${genomesss}.fna&quot; &quot;http://phaster.ca/phaster_api?contigs=1&quot; -O /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt


wget &quot;http://phaster.ca/phaster_api?acc=ZZ_0863a29868&quot; -O /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt


done


wget &quot;http://phaster.ca/phaster_api?acc=ZZ_0863a29868&quot; -O /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PHASTER/${genomesss}_phaster.txt

mkdir -p  ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES


#scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/S_O_124_24730/PROKKA_07282020.gbf ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/PROKKA_07282020.gbk
</code></pre>
<p>In the following I will run the phaster online. therefore I have to
transfer all fna files first to local</p>
<pre class="bash"><code></code></pre>
</div>
<div id="codon-usage-avoidance-between-species" class="section level3">
<h3>codon usage (avoidance between species)</h3>
<p>check this out</p>
<pre class="bash"><code>
###------------------------------------
##calculate codon usage
###------------------------------------

for species in $(echo &quot;Ldel Sterm&quot;)
do

#for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep &quot;.fna$&quot;|cut -d &#39;_&#39; -f 1|sed &#39;s/.fna//g&#39;)
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/ |grep &quot;.ffn$&quot; |sed &#39;s/.ffn//g&#39;)
do

sed &#39;s/ .*//g&#39; /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.ffn |sed &#39;s/ //g&#39; &gt; /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/tmp.ffn

grep &quot;^&gt;&quot; /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.ffn|  awk -F &quot; &quot; &#39;{OFS=&quot; &quot;}{print $2,$3,$4,$5,$6,$7}&#39; |sed &#39;s/[[:space:]]*$//&#39;  &gt; /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/geneNamesfromFFN.txt

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/${species}/${genomess}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/${species}/${genomess}

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/${species}/${genomess}
fascodon  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/tmp.ffn |grep &quot;&gt;&quot; |  sed &#39;s/&gt;//g&#39; |sed &#39;s/  / /g&#39; |sed &#39;s/ /\t/g&#39;|sed &#39;s/freq_.\{1\}_.\{3\}://g&#39; |awk -F &quot;\t&quot; -v SPECIES=&quot;$species&quot; -v GENOMESS=&quot;$genomess&quot; &#39;{OFS=&quot;\t&quot;}{print SPECIES,GENOMESS,$0}&#39; &gt; ${genomess}_codonUsage.txt

paste /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/geneNamesfromFFN.txt ${genomess}_codonUsage.txt &gt;&gt; /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/codonUsageAllstrains.txt

rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/tmp.ffn &amp;
done #genomes
done # species</code></pre>
</div>
<div id="transporters" class="section level3">
<h3>transporters</h3>
</div>
<div id="amino-acid-synthesis" class="section level3">
<h3>amino acid synthesis</h3>
</div>
</div>
</div>
<div id="appendix" class="section level1">
<h1>Appendix</h1>
<div id="extract-16sv4" class="section level3">
<h3>Extract 16sV4</h3>
<ol style="list-style-type: decimal">
<li>download all chromosomes.fasta files of species</li>
<li>run <a href="https://github.com/tseemann/barrnap">barrnap</a> to
extract 16s</li>
<li>run <a href="https://github.com/carden24/V-Xtractor">V-Xtractor</a>
to extract 16s V4</li>
</ol>
<pre class="bash"><code>##------------------------------------------
##1. download all chromosomes.fasta files of species
##------------------------------------------
sed &#39;s/_genomic.gbff.gz/_genomic.fna.gz/g&#39; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt &gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes_contig.txt

sed &#39;s/_genomic.gbff.gz/_genomic.fna.gz/g&#39; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt &gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes_contig.txt


###---
##download
###---

rm -r /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA
mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA
cd /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA
wget -i /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes_contig.txt


rm -r /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA
mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA
cd /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA
wget -i /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes_contig.txt


##------------------------------------------
##2. run barrnap to extract 16s
## AND
##3. run V-Xtractor to extract 16s V4
##------------------------------------------
###---
##Sterm
###---

rm -r /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/
mkdir -p /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/
rm -r /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/
mkdir -p /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/

for genomess in $(ls /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA |grep &quot;.fna.gz$&quot; |sed &#39;s/_genomic.fna.gz//g&#39;)
do
echo ${genomess}
zcat  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/${genomess}_*.fna.gz  |barrnap --threads 37 -o /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}_rrna.fa 

speciess=$(grep &quot;${genomess}&quot; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2)

samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}_rrna.fa 
num=1

rm  /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna
for f16sss in $(grep &quot;16S&quot; -i /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}_rrna.fa |sed &#39;s/&gt;//g&#39;)
do
echo ${f16sss}
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}_rrna.fa ${f16sss} &gt; /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/tmp.fna

sed &quot;s/^&gt;.*$/&gt;16S-${speciess}-${genomess}-${num}/g&quot; /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/tmp.fna &gt;&gt; /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna
num=$((num+1))
  
done #16s
###---------------vxtractor
cd /home/vincent/apps/V-Xtractor

perl vxtractor.pl -a -r V4 -h HMMs/HMMs/SSU/bacteria/ -o /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}_V4.fa /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna
cat /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}_V4.fa &gt;&gt;  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/final_V4.fna
echo &quot;=================================================================&quot;
rm /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/${genomess}*

done #genomess

###---
##Ldel
###---
rm -r /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/
mkdir -p /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/

rm -r /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/
mkdir -p /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/
for genomess in $(ls /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA |grep &quot;.fna.gz$&quot; |sed &#39;s/_genomic.fna.gz//g&#39;)
do
echo ${genomess}
zcat  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/${genomess}_*.fna.gz  |barrnap --threads 37 -o /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}_rrna.fa 

speciess=$(grep &quot;${genomess}&quot; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |cut -f 2)

samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}_rrna.fa 
num=1
rm  /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/final.fna

for f16sss in $(grep &quot;16S&quot; -i /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}_rrna.fa |sed &#39;s/&gt;//g&#39;)
do
echo ${f16sss}
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}_rrna.fa ${f16sss} &gt; /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/tmp.fna

sed &quot;s/^&gt;.*$/&gt;16S-${speciess}-${genomess}-${num}/g&quot; /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/tmp.fna &gt;&gt; /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/final.fna
num=$((num+1))
  
done #16s
###---------------vxtractor
cd /home/vincent/apps/V-Xtractor

perl vxtractor.pl -a -r V4 -h HMMs/HMMs/SSU/bacteria/ -o /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}_V4.fa /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/final.fna
cat /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}_V4.fa &gt;&gt;  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/final_V4.fna
echo &quot;=================================================================&quot;
rm /work/Project/2020_StarterCultureDiversity/16S_search/16s/Lactos/${genomess}*

done #genomess
##============================================================
##------------------------------------------
##4. devide into indiviual files per species
##------------------------------------------
##============================================================
##---------------------------------------
#Sterm
##---------------------------------------
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/final_V4.fna
for speciesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo ${speciesss}
grep &quot;${speciesss}&quot; -c /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/final_V4.fna
grep &quot;${speciesss}&quot; /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/final_V4.fna |sed &#39;s/&gt;//g&#39; &gt; /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/tmp.txt

xargs --arg-file=/work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/tmp.txt  samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/final_V4.fna &gt; /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/${speciesss}_V4.fasta

grep &quot;${speciesss}&quot; -c /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/${speciesss}_V4.fasta

done


##---------------------------------------
#Ldel
##---------------------------------------
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/final_V4.fna
for speciesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo ${speciesss}
grep &quot;${speciesss}&quot; -c /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/final_V4.fna
grep &quot;${speciesss}&quot; /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/final_V4.fna |sed &#39;s/&gt;//g&#39; &gt; /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/tmp.txt

xargs --arg-file=/work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/tmp.txt  samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/final_V4.fna &gt; /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/${speciesss}_V4.fasta


grep &quot;${speciesss}&quot; -c /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/${speciesss}_V4.fasta

done

##============================================================
##------------------------------------------
##5. DEREPLICATE WITH CD-HIT
##------------------------------------------
##============================================================
##---------------------------------------
#Sterm
##---------------------------------------
rm -r /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/
mkdir -p /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/
for speciesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo ${speciesss}


grep &quot;${speciesss}&quot; -c /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/${speciesss}_V4.fasta

cd-hit-est -i /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/${speciesss}_V4.fasta \
    -o /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss} -c 1 -M 30000 -T 37 

#cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.cl*
done

##----------------------- analyse

rm /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/Final_V4_derplicated.fasta
genus=Streptococcus
for speciesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo ${speciesss}


#cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.cl*

    
grep &quot;&gt;Cluster&quot; /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.cl*|sed &#39;s/ /_/g&#39; &gt;  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/tmp.txt

num=1
rm  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.fasta
for clustersss in $(cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/tmp.txt)
do
#aryName=$(grep &quot;${clustersss}&quot; /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log |cut -f 2)
ClusterNumber=$(echo $clustersss|awk -F &quot;_&quot; &#39;{print $2}&#39;)
EndCluster=$((ClusterNumber+1))

countss=$(sed -n &quot;/&gt;Cluster $ClusterNumber$/,/&gt;Cluster $EndCluster$/p&quot; /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.cl* | grep -v &quot;^&gt;Cluster&quot; |wc -l)

namesss=$(grep &quot;&gt;&quot; /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss} | sed &#39;s/&gt;//g&#39; |sed -n ${num}p) 
echo ${namesss}
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss} 
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss} ${namesss} |sed &quot;s/&gt;${namesss}/&gt;16SV4_${genus}_${speciesss}_n${countss}/g&quot; &gt;&gt;  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.fasta

  num=$((num+1))


done

cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/${speciesss}.fasta &gt;&gt;  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/Final_V4_derplicated.fasta

done
##---------------------------------------
#Ldel
##---------------------------------------
rm -r /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/
mkdir -p /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/
for speciesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo ${speciesss}


grep &quot;${speciesss}&quot; -c /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/${speciesss}_V4.fasta

cd-hit-est -i /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/${speciesss}_V4.fasta \
    -o /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss} -c 1 -M 30000 -T 37 

#cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.cl*
done

##----------------------- analyse

rm /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/Final_V4_derplicated.fasta
genus=Lactobacillus
for speciesss in $(cat /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo ${speciesss}


#cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.cl*

    
grep &quot;&gt;Cluster&quot; /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.cl*|sed &#39;s/ /_/g&#39; &gt;  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/tmp.txt

num=1
rm  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.fasta
for clustersss in $(cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/tmp.txt)
do
#aryName=$(grep &quot;${clustersss}&quot; /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log |cut -f 2)
ClusterNumber=$(echo $clustersss|awk -F &quot;_&quot; &#39;{print $2}&#39;)
EndCluster=$((ClusterNumber+1))

countss=$(sed -n &quot;/&gt;Cluster $ClusterNumber$/,/&gt;Cluster $EndCluster$/p&quot; /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.cl* | grep -v &quot;^&gt;Cluster&quot; |wc -l)

namesss=$(grep &quot;&gt;&quot; /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss} | sed &#39;s/&gt;//g&#39; |sed -n ${num}p) 
echo ${namesss}
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss} 
samtools faidx /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss} ${namesss} |sed &quot;s/&gt;${namesss}/&gt;16SV4_${genus}_${speciesss}_n${countss}/g&quot; &gt;&gt;  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.fasta

  num=$((num+1))


done

cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/${speciesss}.fasta &gt;&gt;  /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/Final_V4_derplicated.fasta

done

##============================================================
##------------------------------------------
##6. Bring all together
##------------------------------------------
##============================================================


cat /work/Project/2020_StarterCultureDiversity/16S_search/V4/Lactos/clust/Final_V4_derplicated.fasta \
/work/Project/2020_StarterCultureDiversity/16S_search/V4/Strepto/clust/Final_V4_derplicated.fasta &gt; \
/work/Project/2020_StarterCultureDiversity/16S_search/V4/all_V4_Strepto_Lacto_species.fasta
</code></pre>
<pre class="bash"><code>mkdir -p //home/vincent/Desktop/Projects/2020_StarterCultureDiversity/16S_search/V4/
scp   vincent@130.223.51.116:/work/Project/2020_StarterCultureDiversity/16S_search/V4/all_V4_Strepto_Lacto_species.fasta //home/vincent/Desktop/Projects/2020_StarterCultureDiversity/16S_search/V4/all_V4_Strepto_Lacto_species.fasta
</code></pre>
</div>
<div id="alternative-extract-16sv4" class="section level3">
<h3>Alternative: Extract 16sV4</h3>
<p>Alternative to the extraction from the downloaded genomes, we can
also get the 16s sequences from GTDB and extract the V4 from the species
of interest. This is most likely much more extensive then the previous
approach.</p>
<pre class="bash"><code>#mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication
scp  //home/vincent/Desktop/Projects/2020_StarterCultureDiversity//GTDB_RS89_20190703_1300_HIT_Strains_IDTAXA.fasta vincent@130.223.51.116:/work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/
</code></pre>
<pre class="bash"><code>for speciesss in $(cat  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo &quot;================================================&quot;
echo ${speciesss}
grep ${speciesss} /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/GTDB_RS89_20190703_1300_HIT_Strains_IDTAXA.fasta


done


###-----------------------------------
#ldel
###-----------------------------------


for speciesss in $(cat  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names |cut -f 2|sort|uniq)
do
echo &quot;================================================&quot;
echo ${speciesss}
count=$(grep ${speciesss} /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/GTDB_RS89_20190703_1300_HIT_Strains_IDTAXA.fasta|grep &quot;Lactobacillus&quot; -c)
grep ${speciesss} /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/GTDB_RS89_20190703_1300_HIT_Strains_IDTAXA.fasta|grep &quot;Lactobacillus&quot;
if [ $count -gt 0 ]
 then              
 
echo -e &quot;has &quot; ${count} &quot;16s sequences in GTDB&quot;

 else  
 echo -e &quot;has no 16s sequences in GTDB&quot;

 fi

done</code></pre>
</div>
<div id="gene-duplication" class="section level3">
<h3>gene duplication</h3>
<p>Here, I look for duplicated genes. I do this by clustering all genes
of with mmseqs2 and looking which clusters are larger than 1. Further I
look if among the duplicate genes we have a clustering of a genome
location (E.g. duplicated operon structure).</p>
<pre class="bash"><code>species=Sterm
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/Geneduplication.final 
for species in $(echo &quot;Ldel Sterm&quot;)
do
date=$(ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ |grep &quot;Results&quot;)
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/${species}/
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/ |grep &quot;.ffn$&quot; |sed &#39;s/.ffn//g&#39; )
do

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/${species}/${genomesss}
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/${species}/${genomesss}

#mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/${genomesss}.ffn ${genomesss} tmp --min-seq-id 0.8 -c 0.9 --cov-mode 1
mmseqs easy-cluster /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/${genomesss}.faa ${genomesss} tmp 

grep &quot;&gt;&quot; -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/${genomesss}.faa
#grep &quot;&gt;&quot; -c *_rep_seq.fasta
#grep &quot;&gt;&quot; -c *_all_seqs.fasta
cut -f 1 ${genomesss}_cluster.tsv |sort|uniq -c  |wc -l
#cut -f 1 S12273_cluster.tsv |sort|uniq -c | sed -e &#39;s/^[[:space:]]*//&#39; |awk -F &quot; &quot; &#39;{OFS=&quot;\t&quot;}{if($1&gt;1)print $0}&#39;

for genesss in $(cut -f 1 ${genomesss}_cluster.tsv |sort|uniq -c | sed -e &#39;s/^[[:space:]]*//&#39; |awk -F &quot; &quot; &#39;{OFS=&quot;\t&quot;}{if($1&gt;1)print $2}&#39;)
do
genesss_short=$(echo $genesss|sed &#39;s/S//g&#39;|sed &#39;s/L//g&#39;)

descriptionsss=$(grep &quot;${genesss}&quot;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FAA/${genomesss}.faa |awk -F &quot; &quot; &#39;{OFS=&quot; &quot;}{print $2,$3,$4,$5}&#39;)
OGsss=$(grep &quot;${genesss_short}&quot; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt |awk -F &quot;:&quot; &#39;{print $1}&#39;)
numbergenes=$(cut -f 1 ${genomesss}_cluster.tsv |sort|uniq -c | sed -e &#39;s/^[[:space:]]*//&#39; |awk -F &quot; &quot; -v genezzz=&quot;$genesss&quot; &#39;{OFS=&quot;\t&quot;}{if($2==genezzz)print $1}&#39; )

for genesssAlll in $(awk -F &quot;\t&quot; -v genezzz=&quot;$genesss&quot; &#39;{if($1==genezzz) print $2}&#39; ${genomesss}_cluster.tsv)
do


echo -e ${species}&quot;\t&quot;${genomesss}&quot;\t&quot;${genesss}&quot;\t&quot;${OGsss}&quot;\t&quot;${numbergenes}&quot;\t&quot;${descriptionsss}&quot;\t&quot;${genesssAlll} &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/Geneduplication.final 

done #genesssAlll
done #genesss

done #genomesss


done #species

###----------------------------------------------
#number of genes
###----------------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/gene_count.txt
for species in $(echo &quot;Ldel Sterm&quot;)
do
#date=$(ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ |grep &quot;Results&quot;)

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/ |grep &quot;.gff$&quot; |sed &#39;s/.gff//g&#39; )
do

countsss=$(grep &quot;CDS&quot; -c /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_GFF/${genomesss}.gff)

echo -e ${species}&quot;\t&quot;${genomesss}&quot;\t&quot;${countsss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/gene_count.txt


done #genomesss

done #species
###----------------------------------------------
#number of genes NEW 
#corrected for non-duplicated (do not count duplicated multiple times)
###----------------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/gene_count.txt
for species in $(echo &quot;Ldel Sterm&quot;)
do
date=$(ls /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/ |grep &quot;Results&quot;)

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FFN/ |grep &quot;.ffn$&quot; |sed &#39;s/.ffn//g&#39; )
do


countsss=$(sed &#39;1d&#39; /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/${date}/Orthogroups/Orthogroups.txt |grep &quot;${genomesss}&quot; -c)

echo -e ${species}&quot;\t&quot;${genomesss}&quot;\t&quot;${countsss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/gene_count.txt


done #genomesss


done #species</code></pre>
<p>local</p>
<pre class="bash"><code>mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/gene_count.txt  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/Geneduplication.final  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication

cut -f 2 /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/Geneduplication.final |sort|uniq -c</code></pre>
<p>Now I analyse the data in R. I also combine the eggnog information of
the OGs.</p>
<p>Of special interest are if genes are: - tandem repeat (if two genes)
- operon repeat (left and right flanking genes are also repeated?) -
what genes? (COG?)</p>
<pre class="r"><code>library(tidyverse)
Geneduplication &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/Geneduplication.final&quot;, &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;species&quot;,&quot;genome&quot;,&quot;cluster&quot;,&quot;Orthogroup&quot;,&quot;copies&quot;,&quot;annotation&quot;,&quot;gene&quot;),   trim_ws = TRUE)


Geneduplication_cleaned &lt;- Geneduplication %&gt;% mutate(Orthogroup_new = ifelse(species==&quot;Ldel&quot;,paste0(&quot;L_&quot;,Orthogroup),paste0(&quot;S_&quot;,Orthogroup) ))

##---------------------------
#add eggnog information about the cluster-specific genes. 
##---------------------------

ldel_all_unique &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel_orthofinder_prokka.annotations&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;query_name&quot;,&quot;seed_eggNOG_ortholog&quot;,&quot;seed_ortholog_evalue&quot;,&quot;seed_ortholog_score&quot;,&quot;best_tax_level&quot;,&quot;Preferred_name&quot;,&quot;GOs&quot;,&quot;EC&quot;,&quot;KEGG_ko&quot;,&quot;KEGG_Pathway&quot;,&quot;KEGG_Module&quot;,&quot;KEGG_Reaction&quot;,&quot;KEGG_rclass&quot;,&quot;BRITE&quot;,&quot;KEGG_TC&quot;,&quot;CAZy&quot;,&quot;BiGG_Reaction&quot;,&quot;tax_scope&quot;,&quot;eggNOG OGs&quot;,&quot;bestOG&quot;,&quot;COG_Functional_Category&quot;,&quot;eggNOG free text description&quot;),skip=4)
# RMK202_both_eggnog[RMK202_both_eggnog$query_name %in% SNPeff_final$finalName,]
Sterm_all_unique &lt;- read_delim(&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm_orthofinder_prokka.annotations&quot;,  &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;query_name&quot;,&quot;seed_eggNOG_ortholog&quot;,&quot;seed_ortholog_evalue&quot;,&quot;seed_ortholog_score&quot;,&quot;best_tax_level&quot;,&quot;Preferred_name&quot;,&quot;GOs&quot;,&quot;EC&quot;,&quot;KEGG_ko&quot;,&quot;KEGG_Pathway&quot;,&quot;KEGG_Module&quot;,&quot;KEGG_Reaction&quot;,&quot;KEGG_rclass&quot;,&quot;BRITE&quot;,&quot;KEGG_TC&quot;,&quot;CAZy&quot;,&quot;BiGG_Reaction&quot;,&quot;tax_scope&quot;,&quot;eggNOG OGs&quot;,&quot;bestOG&quot;,&quot;COG_Functional_Category&quot;,&quot;eggNOG free text description&quot;),skip=4)

eggnog_annotation &lt;- rbind(Sterm_all_unique,ldel_all_unique)

COG_functional_categories &lt;- read_delim(&quot;~/Desktop/ScriptRepository/COG_functional_categories.csv&quot;,  &quot;,&quot;, escape_double = FALSE, col_names = c(&quot;short&quot;,&quot;long&quot;,&quot;divided&quot;),trim_ws = TRUE, na = &quot;NA&quot;) %&gt;% select(-divided)

# unique(eggnog_annotation$COG_Functional_Category)[which(!unique(eggnog_annotation$COG_Functional_Category) %in% COG_functional_categories$short )]

ALL_sterm &lt;- eggnog_annotation
nrow(ALL_sterm)

ALL_sterm &lt;- ALL_sterm %&gt;% 
    mutate(COG_Functional_Category = strsplit(as.character(COG_Functional_Category), &quot;&quot;)) %&gt;% 
    unnest(COG_Functional_Category)


lineageAbundance_presence_count_eggnog_COG &lt;- merge(ALL_sterm,COG_functional_categories,by.x = &quot;COG_Functional_Category&quot;,by.y=&quot;short&quot;)
table(lineageAbundance_presence_count_eggnog_COG$long)

##------------------------------
#merge to duplicated genes
##------------------------------
colnames(lineageAbundance_presence_count_eggnog_COG)
lineageAbundance_presence_count_eggnog_COG$query_name %&gt;% length()
duplicate_with_eggnog &lt;- merge(Geneduplication_cleaned,lineageAbundance_presence_count_eggnog_COG,by.x = &quot;Orthogroup_new&quot;,by.y=&quot;query_name&quot;)

table(duplicate_with_eggnog$long) %&gt;% sort(decreasing = TRUE)

##------------------------------
#find look for tandem repeats of genes
##------------------------------


for (clusterzzz in unique(duplicate_with_eggnog$cluster)[1]) {
  
  tmp &lt;- duplicate_with_eggnog %&gt;% filter(cluster==clusterzzz)
  genomesss &lt;- duplicate_with_eggnog %&gt;% filter(cluster==clusterzzz) %&gt;% pull(genome) %&gt;% unique()
  
  duplicate_with_eggnog %&gt;% filter(genome==genomesss)%&gt;% pull(gene) %&gt;% sort()
  
  tmp2 &lt;- duplicate_with_eggnog %&gt;% filter(genome==genomesss)
  
}


##------------------------------
#cogs
##------------------------------

unique(duplicate_with_eggnog$Orthogroup) %&gt;% length()

unique(duplicate_with_eggnog$cluster) %&gt;% length()

duplicated_contigs &lt;- duplicate_with_eggnog %&gt;% select(c(Orthogroup_new,long)) %&gt;% unique() 
library(forcats) 

Cogs_duplicated &lt;- ggplot(duplicated_contigs,aes(x=forcats::fct_infreq(long)))+geom_bar()+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank())

Cogs_dall &lt;- ggplot(lineageAbundance_presence_count_eggnog_COG,aes(x=forcats::fct_infreq(long)))+geom_bar()+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank())

Cogs_duplicated+Cogs_dall+ plot_layout(ncol=1,widths = c(1,1))


all_cogs &lt;- lineageAbundance_presence_count_eggnog_COG %&gt;% group_by(long) %&gt;% summarise(counts=n()) %&gt;% add_column(cogs=&quot;all&quot;)
duplicated_cogs &lt;- duplicated_contigs %&gt;% group_by(long) %&gt;% summarise(counts=n())%&gt;% add_column(cogs=&quot;duplicated&quot;)
complete_cogs &lt;- rbind(all_cogs,duplicated_cogs)


Cogs_duplicated_ext &lt;- ggplot(complete_cogs,aes(x=long,y=sort(counts,decreasing = TRUE),fill=cogs))+geom_bar(stat=&quot;identity&quot;)+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank())
Cogs_duplicated_ext


all_cogs &lt;- lineageAbundance_presence_count_eggnog_COG %&gt;% group_by(long) %&gt;% summarise(ALL_counts=n()) 
duplicated_cogs &lt;- duplicated_contigs %&gt;% group_by(long) %&gt;% summarise(Duplicated=n())

cogs_comple_long &lt;- merge(all_cogs,duplicated_cogs,by=&quot;long&quot;) %&gt;%  mutate(Complete_genes=ALL_counts-Duplicated) %&gt;% dplyr::select(-ALL_counts) %&gt;% gather(.,group,count,c(Complete_genes,Duplicated))



colorsss=rev(c(&quot;grey44&quot;,&quot;grey90&quot;))


Cogs_duplicated_ext &lt;- ggplot(cogs_comple_long,aes(x=reorder(long,-count),y=count,fill=group))+geom_bar(stat=&quot;identity&quot;)+theme_classic()+theme(axis.text.x = element_text(angle=75,hjust=1),axis.title.x = element_blank(),legend.title = element_blank() )+scale_fill_manual(values = colorsss)
Cogs_duplicated_ext

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/COgs_duplicated.pdf&quot;,width=6,height=5)

Cogs_duplicated_ext

dev.off()

##------check with chi-square test

matchlist = c(&quot;Unknown&quot;)

cogs_comple_wide &lt;- merge(all_cogs,duplicated_cogs,by=&quot;long&quot;) %&gt;%  mutate(Complete_genes=ALL_counts-Duplicated) %&gt;% dplyr::select(-ALL_counts) %&gt;% mutate(long=replace_na(long,&quot;Unknown&quot;)) %&gt;% arrange(desc(Complete_genes+Duplicated))%&gt;% arrange(long %in% matchlist) %&gt;% column_to_rownames(.,var=&quot;long&quot;)

# cogs_comple_wide &lt;- cogs_comple_wide %&gt;% arrange(desc(Complete_genes+Duplicated))


# mydata %&gt;% arrange(id %in% match_list)

chisq &lt;- chisq.test(cogs_comple_wide)
# chisq

chisq$observed

round(chisq$expected,2)
round(chisq$residuals, 3)
library(corrplot)
dev.off()
plot.new()
# corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.5)

library(RColorBrewer)
# corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(brewer.pal(n = 8, name = &quot;RdYlBu&quot;)), method = &quot;color&quot;)

col3 &lt;- colorRampPalette(c(&quot;red&quot;, &quot;white&quot;, &quot;blue&quot;)) 
corsssplot &lt;- corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = &quot;color&quot;)
corsssplot
dev.off()

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/corrplot_duplicatedGenes.pdf&quot;,width=6,height=10)

 corrplot(chisq$residuals, is.cor = FALSE,tl.cex=0.75,col = rev(col3(20)), method = &quot;color&quot;)
 dev.off()

##put residual into plot --&gt; amino acid pathway is more common than expected by chance. 
##------------------------------
#Orthogroups without copies
##------------------------------
 
 # duplicate_with_eggnog %&gt;% select(genome,cluster) %&gt;% dim()
 #  duplicate_with_eggnog %&gt;% select(genome,cluster) %&gt;% unique() %&gt;% select(genome) %&gt;% 
 #  

##------------------------------
#percent of duplicated genes
##------------------------------
 ###
 
# library(ggpol)

genomes_tmps &lt;- duplicate_with_eggnog %&gt;% select(species,genome) %&gt;% unique()

duplicate_genes &lt;- duplicate_with_eggnog %&gt;% group_by(genome) %&gt;% select(cluster) %&gt;% unique()%&gt;% summarize(counts=n()) ## here I count the number of duplicated clusters
# duplicate_genes &lt;- duplicate_with_eggnog %&gt;% group_by(genome) %&gt;% select(Orthogroup_new) %&gt;% unique()%&gt;% summarize(counts=n()) ## here I count the number of duplicated clusters

#duplicate_with_eggnog %&gt;% group_by(genome) %&gt;% select(gene) %&gt;% unique()%&gt;% summarize(counts=n()) ## I used to count the number of duplicated gene
duplicate_genes_ext &lt;- merge(duplicate_genes,genomes_tmps,by=&quot;genome&quot;)

ggplot(duplicate_genes_ext,aes(x=species,y=counts,fill=species))+geom_boxplot()+theme_classic()+geom_jitter()+theme(legend.position = &quot;none&quot;)

ggplot(duplicate_genes_ext,aes(x=species,y=counts,fill=species))+geom_boxplot(outlier.size = 0) + geom_point(pch = 21, position = position_jitterdodge())+theme_classic()+theme(legend.position = &quot;none&quot;)

##get percent of multicopy genes by including number of genes in genomes

gene_count &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/gene_count.txt&quot;,&quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;species&quot;,&quot;genome&quot;,&quot;geneCount&quot;),   trim_ws = TRUE) %&gt;% select(-species)
duplicate_genes_ext_02 &lt;- merge(duplicate_genes_ext,gene_count,by=&quot;genome&quot;) %&gt;% mutate(duplicatedPercent=100*(counts/geneCount))



duplicate_genes_ext_02$species &lt;- plyr::revalue(duplicate_genes_ext_02$species,c(&quot;Ldel&quot;=&quot;L. delbrueckii&quot;,&quot;Sterm&quot;=&quot;S. thermophilus&quot;))

##add seq-technology
genomeAssemblyStats &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/statsFinal/genomeAssemblyStats.csv&quot;,   col_types = cols(lineage = col_character())) %&gt;% select(strain,lineage, Ref, Sequencing) %&gt;%replace(is.na(.), &quot;Illumina&quot;)

duplicate_genes_ext_02_tmp &lt;- merge(duplicate_genes_ext_02,genomeAssemblyStats,by.y=&quot;strain&quot;,by.x=&quot;genome&quot;,all.x = TRUE )
duplicate_genes_ext_02 &lt;- duplicate_genes_ext_02_tmp %&gt;% filter(genome!=&quot;L11063&quot;) %&gt;% filter(genome!=&quot;S50&quot;)
is.na(duplicate_genes_ext_02$Sequencing)

colorsss &lt;- c(&quot;#fdc2b5ff&quot;,&quot;#007fa6ff&quot;)

# genesDuplicatedPlot_01 &lt;- ggplot(duplicate_genes_ext_02)+geom_boxplot(aes(x=species,y=duplicatedPercent,fill=species),outlier.size = 0)+scale_fill_manual(values = colorsss) + geom_point(aes(x=Sequencing,y=duplicatedPercent,fill=species),pch = 21, position = position_jitterdodge())+theme_classic()+theme(legend.title = element_blank(),legend.position = &quot;right&quot;,axis.title.x = element_blank(),axis.text.x =element_blank())+labs(y=&quot;duplicated genes [%]&quot;)
# genesDuplicatedPlot_01

genesDuplicatedPlot_01 &lt;- ggplot(duplicate_genes_ext_02,aes(x=Sequencing,y=duplicatedPercent,fill=species))+geom_boxplot(outlier.shape=NA)+scale_fill_manual(values = colorsss) + geom_point(pch = 21, position = position_jitterdodge())+theme_classic()+theme(legend.title = element_blank(),legend.position = &quot;right&quot;,axis.title.x = element_blank(),strip.background = element_blank(),   strip.text.x = element_blank())+labs(y=&quot;duplicated genes [%]&quot;)+facet_wrap(~species)
genesDuplicatedPlot_01

#  library(plotly)
# ggp &lt;- ggplotly(genesDuplicatedPlot_01)
#  ggp

duplicate_genes_ext_02 %&gt;% group_by(species) %&gt;% summarise(median_count=median(duplicatedPercent),sd_count=sd(duplicatedPercent))


###-----------------------------------
#number of copies
###-----------------------------------
# duplicate_only_copies &lt;- duplicate_with_eggnog %&gt;% select(c(cluster,Orthogroup_new,copies)) %&gt;% unique() 


duplicate_with_eggnog_02 &lt;- merge(duplicate_with_eggnog,genomeAssemblyStats,by.x=&quot;genome&quot;,by.y=&quot;strain&quot;) %&gt;% filter(Ref==&quot;Ref&quot;)

duplicate_with_copies &lt;- duplicate_with_eggnog_02 %&gt;% select(c(cluster,Orthogroup_new,copies,species)) %&gt;% unique() %&gt;% group_by(Orthogroup_new,species) %&gt;% summarise(counts=n(),medianCopies=mean(copies),stdvCopies=sd(copies))

100*(table(duplicate_only_copies$copies)/length(duplicate_only_copies$copies))

duplicate_with_copies_with_eggnog &lt;- merge(duplicate_with_copies,lineageAbundance_presence_count_eggnog_COG,by.x = &quot;Orthogroup_new&quot;,by.y=&quot;query_name&quot;) %&gt;% select(c(species,Orthogroup_new,counts,medianCopies,stdvCopies,long))

ggplot(duplicate_only_copies,aes(x=copies))+geom_bar()+theme_classic()

ggplot(duplicate_with_copies_with_eggnog,aes(x=medianCopies,y=counts,color=long))+geom_point()+theme_classic()+coord_trans(x=&quot;log2&quot;,y=&quot;log2&quot;)

##add percent of genomes and sd bars

duplicate_with_eggnog %&gt;% select(c(species,genome)) %&gt;% unique() %&gt;% group_by(species) %&gt;% summarise(n())
number_genomes &lt;- data.frame(species=c(&quot;Sterm&quot;,&quot;Ldel&quot;),numGenome=c(12,2))

# duplicate_with_copies_with_eggnog_ext %&gt;% filter(rel_counts&gt;100)

# duplicate_with_copies_with_eggnog_ext &lt;- merge(duplicate_with_copies_with_eggnog,number_genomes,by=&quot;species&quot;) %&gt;% mutate(rel_counts=100*(counts/numGenome)) %&gt;% mutate(rel_counts=ifelse(rel_counts&gt;100,100,rel_counts)) %&gt;% mutate(xmins=medianCopies-stdvCopies,xmaxs=medianCopies+stdvCopies) %&gt;%  mutate(xmaxs=replace_na(xmaxs, 0.01))%&gt;%  mutate(xmins=replace_na(xmins, 0.01))

duplicate_with_copies_with_eggnog_ext &lt;- merge(duplicate_with_copies_with_eggnog,number_genomes,by=&quot;species&quot;) %&gt;% mutate(rel_counts=100*(counts/numGenome)) %&gt;% mutate(rel_counts=ifelse(rel_counts&gt;100,100,rel_counts)) %&gt;% mutate(xmins=medianCopies-stdvCopies,xmaxs=medianCopies+stdvCopies) %&gt;%  mutate(xmins=ifelse(xmins&lt;=2,2,xmins)) %&gt;%  mutate(xmaxs=ifelse(xmaxs&lt;=2,2,xmaxs))%&gt;%  mutate(xmaxs=replace_na(xmaxs, 2))%&gt;%  mutate(xmins=replace_na(xmins, 2))
range(duplicate_with_copies_with_eggnog_ext$xmins)

countsPlot &lt;- ggplot(duplicate_with_copies_with_eggnog_ext,aes(x=medianCopies,y=rel_counts,color=long))+geom_point()+theme_classic()+coord_trans(x=&quot;log2&quot;,y=&quot;log2&quot;)+theme(legend.position = &quot;none&quot;)
countsPlot

countsPlot_errorbar &lt;- ggplot(duplicate_with_copies_with_eggnog_ext,aes(x=medianCopies,y=rel_counts,color=species,xmin=xmins,xmax=xmaxs))+geom_point()+theme_classic()+theme(legend.position = &quot;none&quot;)+geom_errorbar(alpha=0.3,width=0)+scale_color_manual(values = colorsss)+labs(y=&quot;percent of lineage [%]&quot;,x=&quot;duplication copies&quot;)+ scale_y_continuous(trans=&quot;log2&quot;,breaks =c(8,20,50,100))+scale_x_continuous(trans=&quot;log2&quot;,breaks =c(2,5,10,20,30))
countsPlot_errorbar

# +coord_trans(x=&quot;log2&quot;,y=&quot;log2&quot;)
x_axis_plot &lt;- ggplot(duplicate_with_copies_with_eggnog_ext,aes(x=medianCopies,color=species))+geom_density(adjust = 3)+theme_classic()+coord_trans(x=&quot;log2&quot;)+theme(legend.position = &quot;none&quot;)+scale_color_manual(values = colorsss)+theme(axis.title.x = element_blank(),axis.text.x = element_blank(),axis.ticks.x=element_blank()) #+labs(y=&quot;number of genomes [%]&quot;,x=&quot;duplication copies&quot;)

y_axis_plot &lt;-ggplot(duplicate_with_copies_with_eggnog_ext,aes(x=rel_counts,color=species))+geom_density(adjust = 3)+theme_classic()+coord_trans(x=&quot;log2&quot;)+theme(legend.position = &quot;none&quot;)+scale_color_manual(values = colorsss)+coord_flip()+theme(axis.title.y = element_blank(),axis.text.y = element_blank(),axis.ticks.y=element_blank())  #+labs(y=&quot;number of genomes [%]&quot;,x=&quot;duplication copies&quot;)

# library(patchwork)

# x_axis_plotcountsPlot_errorbar
library(grid)


 blank &lt;- grid.rect(gp=gpar(col=&quot;white&quot;))

 library(gridExtra)
 
 grid.arrange(x_axis_plot,blank,countsPlot_errorbar,y_axis_plot, ncol=2, heights=c(1,3), widths=c(3,1))

 
 pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/duplicatedGenes_copies.pdf&quot;,width=6,height=6)

grid.arrange(x_axis_plot,blank,countsPlot_errorbar,y_axis_plot, ncol=2, heights=c(1,3), widths=c(3,1))

dev.off()


###-----------------------------------
#cog ctagroies
###-----------------------------------
 
ggplot(duplicate_with_copies_with_eggnog,aes(x=medianCopies,color=long))+geom_density()+theme_classic()+coord_trans(x=&quot;log2&quot;)
ggplot(duplicate_with_copies_with_eggnog,aes(x=medianCopies,color=long))+geom_histogram()+theme_classic()+coord_trans(x=&quot;log2&quot;)


###-----------------------------------
#final 
###-----------------------------------

library(patchwork)

genesDuplicatedPlot_01+countsPlot_errorbar+ plot_layout(ncol=2,widths = c(1,3))


pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/duplicatedGenes.pdf&quot;,width=6,height=4)

genesDuplicatedPlot_01+countsPlot_errorbar+ plot_layout(ncol=2,widths = c(1.5,3))

dev.off()


pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/duplicatedGenes.pdf&quot;,width=3,height=6)

genesDuplicatedPlot_01+theme(legend.position = &quot;none&quot;)

dev.off()</code></pre>
<p>Here, I use <a href="http://mummer.sourceforge.net/">nucmer</a> to do
the analysis of the largest repeats</p>
<pre class="bash"><code>  
###==========================
##nucmer
###==========================
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all
for species in $(echo &quot;Ldel Sterm&quot;)
do
echo $species

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/ | grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39;)
do
rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/${species}/${genomesss}
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/${species}/${genomesss}
cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/${species}/${genomesss}


nucmer -b 100 -c 50 -maxmatch -nosimplify /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/${genomesss}.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/01_all_FNA/${genomesss}.fna

delta-filter -i 85 -l 2500   out.delta &gt; ${genomesss}_nucmer.txt
##split files
split -t &quot;&gt;&quot; -l 1 ${genomesss}_nucmer.txt
rm /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/${genomesss}_final.txt
for subsetsss in $(ls |grep &quot;^x&quot; |grep -v &quot;xaa&quot;)
do

genomii1=$(grep &quot;[[:space:]]&quot; ${subsetsss} |head -1 |cut -d &#39; &#39; -f 1)
genomii2=$(grep &quot;[[:space:]]&quot; ${subsetsss} |head -1 |cut -d &#39; &#39; -f 2)

for linesss in $(seq 1 1 $(grep &quot;[[:space:]]&quot; ${subsetsss} |sed &#39;1d&#39; |wc -l))
do
start1=$(grep &quot;[[:space:]]&quot; ${subsetsss} |sed &#39;1d&#39; |sed -n &quot;${linesss}p&quot;  |cut -d &#39; &#39; -f 1)
end1=$(grep &quot;[[:space:]]&quot; ${subsetsss} |sed &#39;1d&#39;  |sed -n &quot;${linesss}p&quot;|cut -d &#39; &#39; -f 2)
start2=$(grep &quot;[[:space:]]&quot; ${subsetsss} |sed &#39;1d&#39; |sed -n &quot;${linesss}p&quot; |cut -d &#39; &#39; -f 3)
end2=$(grep &quot;[[:space:]]&quot; ${subsetsss} |sed &#39;1d&#39;  |sed -n &quot;${linesss}p&quot;|cut -d &#39; &#39; -f 4)

#echo -e ${genomii1}&quot;\t&quot;${start1}&quot;\t&quot;${end1}&quot;\t&quot;${genomii2}&quot;\t&quot;${start2}&quot;\t&quot;${end2} | awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{if($1!=$4 6) print $0}&#39; &gt;&gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/${genomesss}_final.txt


echo -e ${genomii1}&quot;\t&quot;${start1}&quot;\t&quot;${end1}&quot;\t&quot;${genomii2}&quot;\t&quot;${start2}&quot;\t&quot;${end2} | awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{if($1!=$4 || $2!=$5 || $3!=$6 ) print $0}&#39; &gt;&gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/${genomesss}_final.txt

done #linesss

done #subsetsss

 awk -F &quot;\t&quot; -v spezz=&quot;$species&quot; -v genomezzz=&quot;$genomesss&quot; &#39;{OFS=&quot;\t&quot;}{print spezz,genomezzz,$0,$3-$2,$6-$5}&#39; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/${genomesss}_final.txt &gt;&gt; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/all_final.txt

done #genomesss

done #species

#---tests

#genomesss=L11071
#subsetsss=xas
#linesss=1
</code></pre>
<p>look for repeated operons</p>
<pre class="bash"><code>
mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/
scp  vincent@130.223.51.116:/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/all_final.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/
</code></pre>
<p>with nucmer I look for repeated regions</p>
<pre class="r"><code>all_final_operons_duplicated &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/GeneDuplication/nucmer/all/all_final.txt&quot;, &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;species&quot;,&quot;genome&quot;,&quot;contig1&quot;,&quot;start1&quot;,&quot;end1&quot;,&quot;contig2&quot;,&quot;start2&quot;,&quot;end2&quot;,&quot;diff1&quot;,&quot;diff3&quot;),  trim_ws = TRUE)


ggplot(all_final_operons_duplicated,aes(x=diff1))+geom_histogram()+theme_classic()+facet_wrap(~species)+coord_trans(x=&quot;log2&quot;)+geom_vline(xintercept = 5000)

all_final_operons_duplicated_long &lt;- all_final_operons_duplicated %&gt;% filter(diff1&gt;6000)

##------------------------------
#find overlaps
##------------------------------



GenomicRanges::findOverlaps</code></pre>
<div id="enrichment-analysis" class="section level4">
<h4>Enrichment analysis</h4>
<p>Here, I aim to do the enrichment analysis.</p>
<pre class="r"><code>library(tidyverse)
library(topGO)
library(ALL)
data(ALL)
data(geneList)

affyLib &lt;- paste(annotation(ALL), &quot;db&quot;, sep = &quot;.&quot;)
library(package = affyLib, character.only = TRUE)

 sum(topDiffGenes(geneList))
 
 sampleGOdata &lt;- new(&quot;topGOdata&quot;,
                     description = &quot;Simple session&quot;, ontology = &quot;BP&quot;,
                     allGenes = geneList, geneSel = topDiffGenes,
                     nodeSize = 10,
                     annot = annFUN.db, affyLib = affyLib)</code></pre>
<p>#Phylogeny</p>
</div>
</div>
<div id="tree-of-life" class="section level2">
<h2>Tree of life</h2>
<p>Here, I download the tree of life and check who has the lac operon
annotated.</p>
<pre class="bash"><code>mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/genome/GFF

scp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree/LAB_renaming_list.csv vincent@130.223.51.66:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/LAB_renaming_list.csv

scp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree/download.list vincent@130.223.51.66:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/

scp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree/data vincent@130.223.51.66:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/

scp -r  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree/TOL_rrna/ vincent@130.223.51.66:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/


scp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree/name_TOL.txt vincent@130.223.51.66:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/

sed &#39;s/.fna.gz/.gff.gz/g&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download.list &gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download_gff.list

##------------------------------------------------------------
###----------------------------------------------------download gffs
##------------------------------------------------------------
mkdir -p  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/genome_LAB/FNA
cd /work/Projects/2020_StarterCultureDiversity/11_treeoflife/genome_LAB/FNA

rm /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download_lab.list
for labss in $(awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{if($25==&quot;Lactobacillales&quot;) print $3}&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data )
do

grep &quot;${labss}&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download.list &gt;&gt;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download_lab.list

done

awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{if($25==&quot;Lactobacillales&quot;) print $3}&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data| wc -l
wc -l /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download_lab.list

wget -i /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download_gff.list
###----------------------------------------------------download gffs

cd /work/Projects/2020_StarterCultureDiversity/11_treeoflife/genome/GFF

wget -i /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/download_gff.list


##-------------------------------------change nomenclature
#cp /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data_oldNAMES

cp /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data_oldNAMES /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data


sed &#39;s/ /-/g&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/LAB_renaming_list.csv |sed &#39;s/--/-/g&#39;  &gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/LAB_renaming_list_noWhiteSpaces.csv

for NEwsss in $(awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{print $15}&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/LAB_renaming_list_noWhiteSpaces.csv|sort|uniq|awk -F &quot;-ssp&quot; &#39;{OFS=&quot;\t&quot;}{print $1}&#39;)
do
oldsss=$( grep &quot;$NEwsss&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/LAB_renaming_list_noWhiteSpaces.csv|awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{print $3}&#39;|sort|uniq|awk -F &quot;-ssp&quot; &#39;{OFS=&quot;\t&quot;}{print $1}&#39;|head -1)
echo -e ${oldsss}&quot; to &quot;${NEwsss}


NewFinalss=$(echo $NEwsss |sed &#39;s/-/ /g&#39;)
OLDFinalss=$(echo $oldsss |sed &#39;s/-/ /g&#39;)

sed -i &quot;s/${OLDFinalss}/${NewFinalss}/g&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data


done




##---------------------------------------------------------
## check for order lactobacillales (n=493)
##---------------------------------------------------------


awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{if($25==&quot;Lactobacillales&quot;) print $0}&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |wc -l

awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{if($25==&quot;Lactobacillales&quot;) print $0}&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |cut -f 27|sort|uniq -c

awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{if($25==&quot;Lactobacillales&quot;) print $0}&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |cut -f 28|sort|uniq -c



##------------------------------------------------------------
###----------------------------------------------------get longest rRNA lab files
##------------------------------------------------------------
rm -r /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species
rm /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/rRNAs_lab.log
cd /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna

#samtools faidx 23s.fa
#samtools faidx 5s.fa
#samtools faidx 16s.fa
labss=$(grep &quot;Bacillus pseudofirmus&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data|grep &quot;Complete&quot; -i|grep &quot;repres&quot;|cut -f 1)
labss=$(grep &quot;Fusobacterium nucleatum&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data|grep &quot;Complete&quot; -i|grep &quot;repres&quot;|cut -f 1)


for labss in $(awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{if($25==&quot;Lactobacillales&quot;) print $1}&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data )
do

speciesss=$(grep &quot;^${labss}&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data|cut -f 28|sed &#39;s/ /-/g&#39;)


echo $labss
for rrnasss in $(echo &quot;23 16 5&quot;)
do

grep &quot;^${labss}&quot; ${rrnasss}s.tsv &gt; tmp.tsv
countss=$(wc -l tmp.tsv|cut -d &#39; &#39; -f 1)

head -1 tmp.tsv|awk -F &quot;\t&quot; -v countzz=&quot;$countss&quot; -v rrnazzz=&quot;$rrnasss&quot; &#39;{OFS=&quot;\t&quot;}{print $0,$6-$5,rrnazzz&quot;s&quot;,countzz}&#39; &gt;&gt;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/rRNAs_lab.log


#grep &quot;^&gt;${labss_1}&quot; ${rrnasss}s.fa |tail -1 &gt;&gt; LAB/${rrnasss}s_LAB.fa

done #rrnasss


echo &quot;&gt;${speciesss}_${labss}&quot; &gt; LAB/species/${speciesss}_${labss}_rRNAoperon.fasta

grep &quot;^&gt;${labss}_1&quot; -A 1 16s.fa |tail -1 &gt;&gt; LAB/species/${speciesss}_${labss}_rRNAoperon.fasta
echo &quot;NNNNNNNNNNNN&quot; &gt;&gt; LAB/species/${speciesss}_${labss}_rRNAoperon.fasta
grep &quot;^&gt;${labss}_1&quot; -A 1  23s.fa |tail -1 &gt;&gt; LAB/species/${speciesss}_${labss}_rRNAoperon.fasta
echo &quot;NNNNNNNNNNNN&quot; &gt;&gt; LAB/species/${speciesss}_${labss}_rRNAoperon.fasta
grep &quot;^&gt;${labss}_1&quot; -A 1  5s.fa |tail -1 &gt;&gt; LAB/species/${speciesss}_${labss}_rRNAoperon.fasta


##make the assembly fit together nicely
awk &#39;BEGIN{RS=&quot;&gt;&quot;}NR&gt;1{sub(&quot;\n&quot;,&quot;\t&quot;); gsub(&quot;\n&quot;,&quot;&quot;); print RS$0}&#39; LAB/species/${speciesss}_${labss}_rRNAoperon.fasta |sed &#39;s/\t/\n/g&#39; |awk -v FS= &#39;/^&gt;/{print;next}{for (i=0;i&lt;=NF/60;i++) {for (j=1;j&lt;=60;j++) printf &quot;%s&quot;, $(i*60 +j); print &quot;&quot;}}&#39; &gt; LAB/species/${speciesss}_${labss}_rRNAoperon_prep.fasta

rm LAB/species/${speciesss}_${labss}_rRNAoperon.fasta
cat LAB/species/${speciesss}_${labss}_rRNAoperon_prep.fasta &gt;&gt; LAB/all_rRNAoperon_prep.fasta

done

##------QC
grep -c &quot;&gt;&quot; LAB/all_rRNAoperon_prep.fasta
awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{if($25==&quot;Lactobacillales&quot;) print $3}&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data| wc -l
seq_length.py LAB/all_rRNAoperon_prep.fasta |sort -n -k 3

cut -f 1 /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/rRNAs_lab.log|sort|uniq -c |sort -n |sed &#39;s/      //g&#39; |cut -d &#39; &#39; -f 1|sort|uniq -c

grep &quot;5s&quot; -v /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/rRNAs_lab.log|cut -f 1 |sort|uniq -c |sort -n |sed &#39;s/      //g&#39; |cut -d &#39; &#39; -f 1|sort|uniq -c

##-----------------remove genomes that are missing 16s or 23s

rm  removed_because_missing_genes.txt
for genomesss in $(grep &quot;5s&quot; -v /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/rRNAs_lab.log|cut -f 1 |sort|uniq -c |sort -n |sed &#39;s/      //g&#39; |awk -F &quot; &quot; &#39;{OFS=&quot; &quot;}{if($1==1)print $2}&#39;)
do

grep &quot;^${genomesss}&quot;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data &gt;&gt; removed_because_missing_genes.txt

done

cut -f 28  removed_because_missing_genes.txt


###----------extract all genomes if at least 16s and 23s and add Lactobacillus equicursoris
genomesss=$(grep &quot;Bacillus pseudofirmus&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data|grep &quot;Complete&quot; -i|grep &quot;repres&quot;|cut -f 1)
genomesss=$(grep &quot;Fusobacterium nucleatum&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data|grep &quot;Complete&quot; -i|grep &quot;repres&quot;|cut -f 1)

rm keeped_genomes.txt
for genomesss in $(grep &quot;5s&quot; -v /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs/rRNAs_lab.log|cut -f 1 |sort|uniq -c |sort -n |sed &#39;s/      //g&#39; |awk -F &quot; &quot; &#39;{OFS=&quot; &quot;}{if($1==2)print $2}&#39;)
do

speciesss=$(grep &quot;^${genomesss}&quot;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{print $28}&#39;|sed &#39;s/ /-/g&#39;)
genuss=$(grep &quot;^${genomesss}&quot;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{print $28}&#39;|sed &#39;s/ /-/g&#39;|cut -d &#39;-&#39; -f 1)


grep &quot;^${genomesss}&quot;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |awk -F &quot;\t&quot; -v speciezz=&quot;$speciesss&quot; -v genuzzz=&quot;$genuss&quot; &#39;{OFS=&quot;\t&quot;}{print $0,genuzzz,speciezz,speciezz&quot;_&quot;$1}&#39; &gt;&gt; keeped_genomes.txt

done


genomesss$(grep &quot; equicursoris&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |cut -f 1)

speciesss=$(grep &quot;^${genomesss}&quot;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{print $28}&#39;|sed &#39;s/ /-/g&#39;)
genuss=$(grep &quot;^${genomesss}&quot;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{print $28}&#39;|sed &#39;s/ /-/g&#39;|cut -d &#39;-&#39; -f 1)

grep &quot;^${genomesss}&quot;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/logs//data |awk -F &quot;\t&quot; -v speciezz=&quot;$speciesss&quot; -v genuzzz=&quot;$genuss&quot; &#39;{OFS=&quot;\t&quot;}{print $0,genuzzz,speciezz,speciezz&quot;_&quot;$1}&#39; &gt;&gt; keeped_genomes.txt


wc -l keeped_genomes.txt
grep &quot;Streptococcus thermophilus&quot; keeped_genomes.txt
grep &quot;Lactobacillus delbrueckii&quot; keeped_genomes.txt
grep &quot;Lactobacillus helveticus&quot; keeped_genomes.txt
grep &quot;parafarraginis&quot; keeped_genomes.txt

#Lentilactobacillus-parafarraginis_G000238835

###-----------------------
#add different lactobacillus subsp. 
###-----------------------
rm -r  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta
for labsss in $(grep &quot;delbrueckii&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta |sed &#39;s/&gt;//g&#39;)
do

speciesss=$(echo $labsss |sed &#39;s/rRNAoperon-//g&#39; |sed &#39;s/_/-/g&#39;)

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta ${labsss} |sed &quot;s/${labsss}/${speciesss}_own/g&quot; &gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${speciesss}_rRNAoperon_prep.fasta

done




###-----------------------
#add different Streptococcus vestibualris
###-----------------------
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA
samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta
for labsss in $(grep &quot;vestibul&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta |sed &#39;s/&gt;//g&#39;)
do

speciesss=$(echo $labsss |sed &#39;s/rRNAoperon-//g&#39; |sed &#39;s/_/-/g&#39;)

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta ${labsss} |sed &quot;s/${labsss}/${speciesss}_own/g&quot; &gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${speciesss}_rRNAoperon_prep.fasta

done

###-----------------------
##----------------------------add missing speciess!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
###-----------------------

#&quot;L. amylovorus&quot;   &quot;L. bombicola&quot;    &quot;L. equicursoris&quot; &quot;L. gallinarum&quot;   &quot;L. johnsonii&quot;    &quot;L. kimbladii&quot;    &quot;L. kitasatonis&quot;  &quot;L. panisapium&quot;   &quot;L. rodentium&quot;    &quot;L. taiwanensis&quot;  &quot;L. ultunensis&quot;   &quot;S. downei&quot;       &quot;S. equinus&quot;      &quot;S. pneumoniae&quot;

rm -r /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs
#mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/download


seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta
seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta

for genomesss in $(echo &quot;amylovorus bombicola equicursoris gallinarum johnsonii kimbladii kitasatonis panisapium taiwanensis ultunensis&quot;)
do

namesss=$(grep &quot;${genomesss}&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta |sed &#39;s/&gt;//g&#39;)

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/16s/all_rRNAoperon_prep.fasta ${namesss} |sed &quot;s/&gt;.*/&gt;Lactobacillus-${genomesss}_oldAnalysis/g&quot; &gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs/Lactobacillus-${genomesss}_oldAnalysis_rRNAoperon.fasta


done
ll /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs


for genomesss in $(echo &quot;downei equinus pneumoniae&quot;)
do

namesss=$(grep &quot;${genomesss}&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta |sed &#39;s/&gt;//g&#39;)

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta ${namesss} |sed &quot;s/&gt;.*/&gt;Streptococcus-${genomesss}_oldAnalysis/g&quot; &gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs/Streptococcus-${genomesss}_oldAnalysis_rRNAoperon.fasta


done
ll /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs

for genomess in $(ls /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs)
do
echo ${genomess}
cp /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs/${genomess} /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${genomess}
done
###-----------------------
###add good genes
###-----------------------

speciesss=$(cut -f 56 /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes.txt |cut -d &#39;_&#39; -f 1|sort|uniq |grep -v &quot;acidilactici&quot; |head -1)
rm /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt
for speciesss in $(cut -f 56 /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes.txt |cut -d &#39;_&#39; -f 1|sort|uniq |grep -v &quot;acidilactici&quot; )
do


#samplesss=$(ls /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species/ |grep &quot;${speciesss}_&quot; |head -1)
samplesss=$(grep &quot;${speciesss}&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes.txt |cut -f 56|head -1)
newNAME=$(echo $samplesss)
cp /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species/${samplesss}_rRNAoperon_prep.fasta /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${newNAME}_rRNAoperon.fasta

grep &quot;${newNAME}&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes.txt &gt;&gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt

#grep &quot;${newNAME}&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes.txt|cut -f 56

done

wc -l /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt

ls /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/ |wc -l 

cat /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/*.fasta &gt;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa

grep &quot;&gt;&quot; -c /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa

seq_length.py  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa

###---------------------------add outgroup

###---------------------------remove 
rm /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa

for genomess in $(grep &quot;&gt;&quot;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa |grep -v &quot;sp.-X13SY08&quot;|grep -v &quot;sp.-HMSC24D01&quot;|grep -v &quot;sp.-wkB10&quot;| grep -v &quot;i_G000056065&quot; |sed &#39;s/&gt;//g&#39;)
  do  
samtools faidx  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa  ${genomess} &gt;&gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa


done

grep &quot;&gt;&quot; -c /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa
grep &quot;&gt;&quot; -c /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa


 for genomess in $(ls /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs)
do
echo ${genomess}
cat /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/missing_LABs/${genomess} &gt;&gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa
done

grep &quot;&gt;&quot; -c /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa

grep &quot;&gt;&quot;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa
</code></pre>
<div id="extract-16s-of-only-streptolacto" class="section level5">
<h5>extract 16s of only strepto/lacto</h5>
<p>I am doing this for Flo</p>
<pre class="bash"><code>#...............................
#count lactose 
#...............................


##-------------------------------vestibularis 


cat /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/Streptococcus-vestibularis_rRNAoperon_prep.fasta &gt;&gt;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/Streptococcus-vestibularis_own_rRNAoperon.fasta

##-------------------------------Sterptos 

rm /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa


for genomess in $(grep &quot;&gt;Streptococcus-&quot;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa |sed &#39;s/&gt;//g&#39;)
  do  
  #grep -A 3 &quot;${genomess}&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa
#samtools faidx  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa  ${genomess} &gt;&gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa

cat  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${genomess}_rRNAoperon.fasta &gt;&gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa

done

##-------------------------------subsp. delbrueckii 

for genomess in $(ls /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/ |grep &quot;delbrueckii-&quot; | sed &#39;s/_rRNAoperon_prep.fasta//g&#39;)
do


cat /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${genomess}_rRNAoperon_prep.fasta &gt;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${genomess}_own_rRNAoperon.fasta
head /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${genomess}_own_rRNAoperon.fasta

done

##-------------------------------Lactops 

for genomess in $(grep &quot;&gt;Lactobacillus-&quot;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa |sed &#39;s/&gt;//g&#39;|grep -v &quot;delbrueckii_G000056065&quot;)
  do  
  
#samtools faidx  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa  ${genomess} &gt;&gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa

cat  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/${genomess}_rRNAoperon.fasta &gt;&gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa

done

grep &quot;&gt;&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa

#sed -i &#39;s/delbrueckii_G000056065/delbrueckii-bulgaricus_G000056065/g&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa
grep &quot;&gt;&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa

sed -i &#39;s/_.*$//g&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa
grep &quot;&gt;&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa 
grep &quot;&gt;&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa  |wc -l
grep &quot;&gt;&quot; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa |sort|uniq |wc -l


##move local
scp vincent@130.223.110.206:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new_ForFLO.fa /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/Earthmicrobiome_data/all_rRNAoperon_new_ForFLO.fa
</code></pre>
</div>
<div id="check-for-lac-operon" class="section level5">
<h5>check for Lac operon</h5>
<p>Check the closely related lactobacillus and strepotococcus genomes
for the presence of the lac operon</p>
<pre class="bash"><code>

##££££££££££££££££££££££££££££££££ do this for all strains in the species
cd /work/Projects/2020_StarterCultureDiversity/11_treeoflife/genome/GFF


#genomess=GCF_001808965.1
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/lacOPERON/
rm /work/Projects/2020_StarterCultureDiversity/11_treeoflife/lacOPERON/lactose_streps_lactos.txt
#cut -f 56 /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt
for genomess in $(awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{if($56~&quot;Lactobacillus&quot; || $56~&quot;Streptococcus&quot;)print $3}&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt)
do
echo ${genomess}
#zgrep lactose -c -i ${genomess}*.gff.gz |cut -d &#39;:&#39; -f 2
#zgrep &quot;=lactose&quot; -c -i ${genomess}*.gff.gz |cut -d &#39;:&#39; -f 2
#zgrep &quot;3.2.1.23&quot; -i ${genomess}*.gff.gz
#zgrep &quot;lacZ&quot; -i ${genomess}*.gff.gz
#zgrep &quot;galactosidase&quot; -i ${genomess}*.gff.gz

#countss=$(zgrep &quot;beta-galactosidase&quot; -A 2 -B 2 -i ${genomess}*.gff.gz|grep &quot;PTS&quot; -i |wc -l)
countss=$(zgrep &quot;galactosidase&quot; -A 2 -B 2 -i ${genomess}*.gff.gz)

namess=$(awk -F &quot;\t&quot; -v genomzzz=&quot;$genomess&quot; &#39;{OFS=&quot;\t&quot;}{if($3==genomzzz)print $56}&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt)

#countss=$(zgrep &quot;beta-galactosidase&quot; -c -i ${genomess}*.gff.gz |cut -d &#39;:&#39; -f 2)

#echo -e ${namess}&quot;\t&quot;${countss} &gt;&gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/lacOPERON/lactose_streps_lactos.txt
echo -e ${namess}&quot;\t&quot;${countss} #&gt;&gt; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/lacOPERON/lactose_streps_lactos.txt


#zgrep &quot;galactohydrolase&quot; -i ${genomess}*.gff.gz

#zgrep &quot;lacS&quot; -i ${genomess}*.gff.gz
#zgrep &quot;lacZ&quot; -i ${genomess}*.gff.gz

done

cut -f 2 /work/Projects/2020_StarterCultureDiversity/11_treeoflife/lacOPERON/lactose_streps_lactos.txt|sort|uniq -c



genomess=GCF_000422165.1 #lhelv
genomess=GCF_000056065.1 #Ldel
genomess=GCF_000253395.1 #Sterm

##-------------------------
#transfer local
##-------------------------



scp vincent@130.223.51.66:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/lacOPERON/lactose_streps_lactos.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/lactose_streps_lactos.txt
</code></pre>
<p>new</p>
<pre class="bash"><code>

rm -r /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/lactose_utilziation/presence_lacOPERON.txt

for alltoegether in $(grep &quot;&gt;&quot; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/Earthmicrobiome_data/all_rRNAoperon_new_ForFLO.fa |sed &#39;s/&gt;//g&#39; )
do
genusss=$(echo $alltoegether|sed &#39;s/-delbrueckii//g&#39;  |cut -d &#39;-&#39; -f 1 )
species=$(echo $alltoegether |sed &#39;s/-delbrueckii//g&#39; |cut -d &#39;-&#39; -f 2 )

lacZ=$(grep &quot;${genusss}&quot; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/lactose_utilziation/tax_report_LACz.txt |grep -c &quot;${species}&quot;  )
lacG=$(grep &quot;${genusss}&quot; /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/lactose_utilziation/tax_report_LACg.txt |grep -c &quot;${species}&quot;  )

echo -e ${alltoegether}&quot;\t&quot;${lacZ}&quot;\t&quot;${lacG}
echo -e ${alltoegether}&quot;\t&quot;${lacZ}&quot;\t&quot;${lacG} &gt;&gt;  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/lactose_utilziation/presence_lacOPERON.txt

done
</code></pre>
</div>
<div id="check-glycosidase-hydrolases" class="section level5">
<h5>check glycosidase hydrolases</h5>
<pre class="bash"><code>
##-----------------------------------
#01-create DBcan2 Db
##-----------------------------------
cd /work/Databases/20211209_DBcan2_v10

wget https://bcb.unl.edu/dbCAN2/download/dbCAN-HMMdb-V10.txt

##-----------------------------------
#HMM the faa files
##-----------------------------------
#mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FAA/GCF_900322585.1_Lactobacillus_delbrueckii_subsp._lactis_protein.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FAA/Lactobacillus_delbrueckii_subsp_lactis-7.faa

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/anlayis/HMM_GH/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/anlayis/HMM_GH/
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FAA/ |sed &#39;s/.faa//g&#39; )
do

echo &quot;==================================&quot;
echo ${genomesss}


hmmscan --cpu 5 --domtblout anlayis/HMM_GH/$genomesss.faa.out.dm /work/Databases/20211209_DBcan2_v10/dbCAN-HMMdb-V10.txt /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FAA//$genomesss.faa &gt;  anlayis/HMM_GH/$genomesss.faa.out


done

##-----------------------------------------------------------------------
#parse
##-----------------------------------------------------------------------

rm anlayis/HMM_GH/hmm_analysis.txt
for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FAA/ | grep &quot;.faa&quot; |sed &#39;s/.faa//g&#39; )
do

echo &quot;==================================&quot;
echo ${genomesss}


##---filter
sh /work/Databases/20211209_DBcan2_v10/hmmscan-parser_03.sh  anlayis/HMM_GH/$genomesss.faa.out.dm &gt; anlayis/HMM_GH/$genomesss.faa.out_parsed

countss=$(wc -l anlayis/HMM_GH/$genomesss.faa.out_parsed |cut -d &#39; &#39; -f 1)

echo -e  ${genomesss}&quot;\t&quot;${countss} &gt;&gt; anlayis/HMM_GH/hmm_analysis.txt
done


##-----------------------------------------------------------------------
#move local
##-----------------------------------------------------------------------
mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/anlayis/HMM_GH

scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/anlayis/HMM_GH/Streptococcus_oralis-16.faa.out.dm /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/tmp/


scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/anlayis/HMM_GH/hmm_analysis.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/anlayis/HMM_GH/hmm_analysis.txt
</code></pre>
<p>rAnalysis</p>
<pre class="r"><code>library(tidyverse)

hmm_analysis &lt;- read_table2(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/anlayis/HMM_GH/hmm_analysis.txt&quot;,  col_names = c(&quot;strain&quot;,&quot;GH&quot;)) %&gt;% mutate(species=gsub(&quot;-.*$&quot;,&quot;&quot;,strain))

table(hmm_analysis$species)

hmm_analysis_median &lt;-  hmm_analysis %&gt;% group_by(species) %&gt;% summarise(genome_count=n(),median_GH=median(GH))
hmm_analysis_median
write.csv(hmm_analysis_median,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/anlayis/HMM_GH/hmm_analysis_prepped.txt&quot;,quote = FALSE,row.names = FALSE)</code></pre>
</div>
<div id="check-number-of-complete-genomes-ncbi" class="section level5">
<h5>check number of complete genomes NCBI</h5>
<pre class="bash"><code>


mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
#curl &#39;ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt&#39; &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt


mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log

for genomess in $(awk -F &quot;\t&quot; &#39;{OFS=&quot;\t&quot;}{if($56~&quot;Lactobacillus&quot; || $56~&quot;Streptococcus&quot;)print $56}&#39; /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/keeped_genomes_without_duplicate.txt |sed &quot;s/_.*$//g&quot;|grep &quot;sp.&quot; -v |grep &quot;delbrueckii&quot; -v)
do
genomes_correct=$(echo ${genomess} |sed &#39;s/-/ /g&#39;)
echo ${genomes_correct} 

#grep -c ${genomes_correct} /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt
countsss=$(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e &quot;${genomes_correct}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;| awk &#39;BEGIN{FS=&quot;\t&quot;;OFS=&quot;\t&quot;}  {print $20}&#39;| sed -r &#39;s|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|&#39; |wc -l ) #&gt;    \
   # /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt



cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e &quot;${genomes_correct}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;| awk &#39;BEGIN{FS=&quot;\t&quot;;OFS=&quot;\t&quot;}  {print $20}&#39;| sed -r &#39;s|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|&#39;  &gt;&gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e &quot;${genomes_correct}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | awk -v namezzin=&quot;$genomes_correct&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names

echo -e ${genomes_correct}&quot;\t&quot;${countsss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt



echo &quot;=================================================================&quot;
done


###-------------------------------------
#vestibularis
###-------------------------------------
genomes_correct=$(echo &quot;Streptococcus vestibularis&quot;)

#grep -c ${genomes_correct} /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt
countsss=$(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e &quot;${genomes_correct}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;| awk &#39;BEGIN{FS=&quot;\t&quot;;OFS=&quot;\t&quot;}  {print $20}&#39;| sed -r &#39;s|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|&#39; |wc -l ) #&gt;    \
   # /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt



cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e &quot;${genomes_correct}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;| awk &#39;BEGIN{FS=&quot;\t&quot;;OFS=&quot;\t&quot;}  {print $20}&#39;| sed -r &#39;s|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|&#39;  &gt;&gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e &quot;${genomes_correct}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | awk -v namezzin=&quot;$genomes_correct&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names

echo -e ${genomes_correct}&quot;\t&quot;${countsss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt

###-------------------------------------
#lactobacillus delbrueckiis
###-------------------------------------
genomes_correct=$(echo &quot;Lactobacillus delbrueckii&quot;)


for subsss in $(echo &quot;
bulgaricus
lactis
indicus
delbrueckii
sunkii
jakobsenii&quot;
)
do
genomes_correct=$(echo &quot;${subsss}&quot; |awk &#39;{OFS=&quot; &quot;}{print &quot;Lactobacillus delbrueckii subsp.&quot;,$0}&#39;)

echo ${genomes_correct} 

#grep -c ${genomes_correct} /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt

#grep -c ${genomes_correct} /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt

#grep -c ${genomes_correct} /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt
countsss=$(cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e &quot;${genomes_correct}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;| awk &#39;BEGIN{FS=&quot;\t&quot;;OFS=&quot;\t&quot;}  {print $20}&#39;| sed -r &#39;s|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|&#39; |wc -l ) #&gt;    \
   # /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/download_sterm.txt



cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e &quot;${genomes_correct}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;| awk &#39;BEGIN{FS=&quot;\t&quot;;OFS=&quot;\t&quot;}  {print $20}&#39;| sed -r &#39;s|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|&#39;  &gt;&gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |grep -e &quot;${genomes_correct}&quot; |awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|cut -f 20 | awk -v namezzin=&quot;$genomes_correct&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names

echo -e ${genomes_correct}&quot;\t&quot;${countsss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt

done



wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names
wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log

awk &#39;BEGIN {FS = &quot;\t&quot;} ; {sum+=$2} END {print sum}&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt

awk &#39;{FS = &quot;\t&quot;}  {if($2==0) print $0}&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt


###---------------------------------------------------------------
#prepare names
###---------------------------------------------------------------
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned
for speciesss_old in $(cut -f 2 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names|sed &#39;s/ /_/g&#39; |sort|uniq)
do
echo &quot;================================================&quot;
echo $speciesss_old
num=1
speciesss=$(echo $speciesss_old |sed &#39;s/_/ /g&#39;)

for genomesss in $(grep &quot;${speciesss}&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names |cut -f 1)
do

genomess_name_new=$(echo ${speciesss}|sed &#39;s/ /_/g&#39; |awk -v countss=&quot;$num&quot; &#39;{print $1&quot;-&quot;countss}&#39;)

grep &quot;${genomesss}&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.names |awk -v countss=&quot;$genomess_name_new&quot; &#39;{OFS=&quot;\t&quot;}{print $0,countss}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned


num=$((num+1))

done

done


sed -i &#39;s/subsp._/subsp_/g&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned

##===================================================================================================
###---------------------------------------------------------------
#download genomes
###---------------------------------------------------------------
##===================================================================================================
sed -r &#39;s|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|&#39;
whichFILE=fna

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/

rm *

wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log


###-------------------------------------
##change name
###-------------------------------------

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned| cut -d &#39;/&#39; -f 10)
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned|cut -f 3 )
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz

#mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA//GCF_900322585.1_Lactobacillus_delbrueckii_subsp._lactis_protein.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/Lactobacillus_delbrueckii_subsp_lactis-7.fna


##===================================================================================================
###---------------------------------------------------------------
#download faa
###---------------------------------------------------------------
##===================================================================================================
whichFILE=faa

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FAA/
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FAA/

sed &#39;s/_genomic.fna./_protein.faa./g&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/${whichFILE}_download.log

rm *

wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/${whichFILE}_download.log


###-------------------------------------
##change name
###-------------------------------------

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FAA/

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned| cut -d &#39;/&#39; -f 10)
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned|cut -f 3 )
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz

##===================================================================================================
###---------------------------------------------------------------
#download gff
###---------------------------------------------------------------
##===================================================================================================
whichFILE=gff

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/GFF/
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/GFF/

sed &#39;s/_genomic.fna./_genomic.gff./g&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/${whichFILE}_download.log

rm *

wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/${whichFILE}_download.log


###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned| cut -d &#39;/&#39; -f 10)
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned|cut -f 3 )
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz

##===================================================================================================
###---------------------------------------------------------------
#download gbff
###---------------------------------------------------------------
##===================================================================================================
whichFILE=gbff

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/GBFF/
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/GBFF/

sed &#39;s/_genomic.fna./_genomic.gbff./g&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/genomes_download.log &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/${whichFILE}_download.log

rm *

wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/${whichFILE}_download.log


###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned| cut -d &#39;/&#39; -f 10)
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned|cut -f 3 )
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.${whichFILE}.gz ${newwsss}.${whichFILE}.gz
done

gunzip *.${whichFILE}.gz



##------------------------------------------------------------



##------------------------------------------------------
#move local
##------------------------------------------------------

scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log//genome_counts_lactos_streptos.txt /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/
</code></pre>
</div>
<div id="make-tree" class="section level5">
<h5>make tree</h5>
<pre class="bash"><code>##------------------------------------------------------

##------------------------------------------------------
#make alignement
##------------------------------------------------------
rm -r /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/
mkdir -p /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/
cd /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/

#mafft-linsi --thread 37 --maxiterate 1000    /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon.fa &gt;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/all_rRNAoperon.fa_aln

mafft-linsi --thread 37 --maxiterate 1000    /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/all_rRNAoperon_new.fa &gt;  /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/all_rRNAoperon.fa_aln

##------------------------------------------------------
#pyhlogeny
##------------------------------------------------------

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/all_rRNAoperon.fa_aln -n lactos -T 37</code></pre>
<pre class="bash"><code>##------------------------------------------------------
#move local
##------------------------------------------------------

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03/
scp -r vincent@130.223.110.206:/work/Projects/2020_StarterCultureDiversity/11_treeoflife/TOL_rrna/LAB/species_final/rRNA/pyhlogeny_complete/RA* /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03/

##------------------------------------------------------
#chnage names
##------------------------------------------------------

 #sed &quot;s/_.*://g&quot; /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large/RAxML_bipartitions.lactos &gt; /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large/RAxML_bipartitions.lactos_02
 

##------------------------------------------------------
#start align with figtree
##------------------------------------------------------

java -jar /home/vincent/bin/apps/FigTree.v1.4.4/FigTree\ v1.4.4/lib/figtree.jar
</code></pre>
<p>in r after rerooting</p>
<pre class="r"><code># 
# library(ape)
# library(ggtree)
# library(tidyverse)
# library(tidytree)
# library(treeio)
# library(TreeTools)
# 
# Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large/RAxML_bipartitions.lactos_curated&quot;, tree.names = NULL, force.multi = FALSE)
# 
# 
# # x &lt;- as_tibble(Sterm_tree_raw)
# 
# ##plot tree
# # ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
# tree_01 &lt;- ggtree(Sterm_tree_raw, layout=&quot;circular&quot;) +
#   geom_tiplab(aes(angle=angle),align = TRUE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0)  #NO ROOT  
#   
# tree_01 &lt;- ggtree(Sterm_tree_raw) +
#   geom_tiplab(size=0.5,align = FALSE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0)  #NO ROOT 
# 
#    png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png&quot;, width = 2200, height = 1500,res=400)
# # pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)
# 
# tree_02
# 
# 
# dev.off()
# 
# 
# 408=Oneococcus
# 409=Leuconostoc
# 398=Weissela
# #154=Furfurilactobacillus #is a tip
# 381=Limoslactibacillus
# 392=Paucilactobacillus
# 374=Lactiplantibacillus
# 353=Lentilactobacillus
# 348=Fructilactobacillus
# 351=Apilactobacillus
# 362=Secundilactobacillus
# 368=Levilactobacillus
# 337=Pediococcous
# 422=Liquorilactobacillus
# 
# 455=Lactobacillus
# 
# 479=Companilactobacillus
# 488=Bombilactobacillus
# 448=Lapidilactobacillus
# 446=Schleiferilactobacillus
# #177=Agrilactobacillus #tip
# 491=Lacticaseibacillus
# 500=Latilactobacillus
# #223=Paralactobacillus #tip
# 441=Loigolactobacillus
# 512=mix
# 504=Carnobacterium
# 
# 289=Streptococcus
# 
# 324=Lactococcus
# 
# ##enterococcus is polyphyletic
# 
# 
#  library(plotly)
# ggp &lt;- ggplotly(tree_02)
#  ggp
#  
# tree_02  &lt;- scaleClade(tree_01, 284, 2) %&gt;%  scaleClade(., 452, 2)%&gt;% collapse(409, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(408, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(398, &#39;max&#39;, fill=&quot;darkgreen&quot;) %&gt;% collapse(381, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(392, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(374, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(353, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(348, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(351, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(362, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(368, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(337, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(422, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% 
# collapse(479, &#39;max&#39;, fill=&quot;darkgreen&quot;)    %&gt;% collapse(488, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(448, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(446, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(491, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(500, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(441, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(512, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(504, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(324, &#39;max&#39;, fill=&quot;darkgreen&quot;) 
# 
# tree_02
#  
# 
# 
# ##-------------
# #change tiplabel and color according to rmk or non-rmk sample
# ##-------------
# 
# tmp = as.character(Sterm_tree_raw$strain)
# names(tmp) &lt;- all_sterm_strain_info$shortName
# 
# # Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated&quot;, tree.names = NULL, force.multi = FALSE)
# Sterm_tree_raw$tip.label
# # Sterm_tree_raw$label
# 
# 
# 
# Sterm_tree_raw_tible &lt;- as_tibble(Sterm_tree_raw) 
# Sterm_tree_raw_tible$label
# Sterm_tree_raw_tible$label &lt;- gsub(&quot;_.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label &lt;- gsub(&quot;Streptococcus-&quot;,&quot;S. &quot;,Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label &lt;- gsub(&quot;Lactobacillus-&quot;,&quot;L. &quot;,Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label &lt;- gsub(&quot;Enterococcus-&quot;,&quot;E. &quot;,Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label &lt;- gsub(&quot;Tetragenococcus-&quot;,&quot;T. &quot;,Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label &lt;- gsub(&quot;Holzapfelia-&quot;,&quot;H. &quot;,Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label &lt;- gsub(&quot;Amylolactobacillus-&quot;,&quot;A. &quot;,Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label &lt;- gsub(&quot;L. delbrueckii$&quot;,&quot;L. deblrueckii-delbrueckii&quot;,Sterm_tree_raw_tible$label)
# 
# 
# tree_02_tmp &lt;- Sterm_tree_raw_tible %&gt;% as.treedata()
# 
# # Sterm_tree_raw_tible$label &lt;- plyr::revalue(Sterm_tree_raw_tible$label,tmp)
# tree_02 &lt;- Sterm_tree_raw_tible%&gt;% as.treedata()
# 
# 
# ##----------------------------------
# #after branch collaps
# ##----------------------------------
# 
# tree_03 &lt;- ggtree(tree_02, layout=&quot;circular&quot;) +
#   geom_tiplab(size=3.5,align = FALSE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0)  
#    tree_04 &lt;- scaleClade(tree_03, 284, 2.5) %&gt;%  scaleClade(., 452, 2.5)%&gt;% collapse(409, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(408, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(398, &#39;max&#39;, fill=&quot;darkgreen&quot;) %&gt;% collapse(381, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(392, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(374, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(353, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(348, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(351, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(362, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(368, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(337, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(422, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% 
# collapse(479, &#39;max&#39;, fill=&quot;darkgreen&quot;)    %&gt;% collapse(488, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(448, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(446, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(491, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(500, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(441, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(512, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(504, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(324, &#39;max&#39;, fill=&quot;darkgreen&quot;) 
# tree_04
# 
# 
# 
# 
# p2 &lt;- ggtree(tree_02, layout=&quot;circular&quot;) +
#   geom_tiplab(aes(angle=angle,color=origin,size=origin),align = TRUE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0) + #NO ROOT  
#   scale_color_manual(values=c(&quot;grey66&quot;, &#39;grey55&#39;, &#39;black&#39;)) +
#   scale_size_manual(values=c(2.5,2.5,4)) 
# p2
# 
# ##-------------
# #add geograph and origin
# ##-------------
# all_sterm_strain_info$shortName &lt;- as.character(all_sterm_strain_info$shortName)
# all_sterm_strain_info_ready &lt;- all_sterm_strain_info %&gt;% select(-shortName) %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;strain&quot;)%&gt;%replace(is.na(.), &quot;Unknown&quot;) 
# table(all_sterm_strain_info_ready$location_continent)
#  all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c(&quot;Europe&quot;,  &quot;Asia&quot;, &quot;USA&quot;, &quot;Australia&quot; , &quot;Unknown&quot;,&quot;Outgroup&quot;))
# # colors_area &lt;- c(&quot;#464D77&quot;,&quot;#36827F&quot;,&quot;#F9DB6D&quot;,&quot;grey88&quot;)
# # colors_area &lt;- c(&quot;#F9DB6D&quot;,&quot;#36827F&quot;,&quot;#464D77&quot;,&quot;grey88&quot;)
# 
# # location_dataset$area
# 
# # tree_02 &lt;- full_join(as_tibble(tree), location_dataset, by=&#39;label&#39;) %&gt;% as.treedata()
# 
# colnames(all_sterm_strain_info_ready) &lt;- c(&quot;A&quot;,&quot;B&quot;)
# rownames(all_sterm_strain_info_ready)
# 
# p3 &lt;- gheatmap(p2, all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
# p3
# 
# 
# 
# ##--------------------------------
# ##color labels of of only rmk phages
# ##--------------------------------
# 
# Sterm_genome_origin &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/logs/Sterm_genome_origin.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;label&quot;,&quot;origin&quot;),  trim_ws = TRUE) %&gt;% as.data.frame()
# 
# # tmp &lt;- full_join(as_tibble(Sterm_tree_raw), Sterm_genome_origin, by=&#39;label&#39;)
# # table(tmp$origin)
# tree &lt;- full_join(as_tibble(Sterm_tree_raw), Sterm_genome_origin, by=&#39;label&#39;) %&gt;% as.treedata()
# 
# 
# p &lt;- ggtree(tree, layout=&quot;circular&quot;) +
#   geom_tiplab(aes(angle=angle,color=origin,size=origin),align = TRUE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0) + #NO ROOT  
#   scale_color_manual(values=c(&quot;grey66&quot;, &#39;grey55&#39;, &#39;black&#39;)) +
#   scale_size_manual(values=c(4,4,7)) +geom_treescale()
# 
# p
# 
# ##--------------------------------
# #include scaleClade(p, node=17, scale=.1)  
# #include collapse(p2, node, &#39;mixed&#39;) %&gt;% expand(node)
# #scaleClade(p, 23, .2) %&gt;% collapse(23, &#39;min&#39;, fill=&quot;darkgreen&quot;)  
##--------------------------------

##==========================================================================================
#again 
#after adding S.vestibularis
##==========================================================================================


library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02//RAxML_bipartitions.lactos_curated&quot;, tree.names = NULL, force.multi = FALSE)


# x &lt;- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
tree_01 &lt;- ggtree(Sterm_tree_raw, layout=&quot;circular&quot;) +
  geom_tiplab(aes(angle=angle),align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  tree_01
tree_01 &lt;- ggtree(Sterm_tree_raw) +
  geom_tiplab(size=0.5,align = FALSE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT 
tree_01
   png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png&quot;, width = 2200, height = 1500,res=400)
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)

tree_01


dev.off()

 library(plotly)
ggp &lt;- ggplotly(tree_01)
 ggp

# 
# 415=Oneococcus
# 416=Leuconostoc
# 429=Weissela
# #154=Furfurilactobacillus #is a tip
# 443=Limoslactibacillus
# 439=Paucilactobacillus
# 454=Lactiplantibacillus
# 467=Lentilactobacillus
# 462=Fructilactobacillus
# 461=Apilactobacillus
# 476=Secundilactobacillus
# 482=Levilactobacillus
# 402=Pediococcous
# 487=Liquorilactobacillus
# 
# 348=Lactobacillus
# 
# 376=Companilactobacillus
# 375=Bombilactobacillus
# 385=Lapidilactobacillus
# 343=Schleiferilactobacillus
# #177=Agrilactobacillus #tip
# 388=Lacticaseibacillus
# 397=Latilactobacillus
# #223=Paralactobacillus #tip
# 338=Loigolactobacillus
# 519=mix
# 506=Carnobacterium
# 
# 285=Streptococcus
# 
# 282=Lactococcus

##enterococcus is polyphyletic


 
tree_02  &lt;- scaleClade(tree_01, 285, 2) %&gt;%  scaleClade(., 348, 2)%&gt;% collapse(416, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(415, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(429, &#39;max&#39;, fill=&quot;darkgreen&quot;) %&gt;% collapse(443, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(439, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(454, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(467, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(462, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(461, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(476, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(482, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(402, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(487, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% 
collapse(376, &#39;max&#39;, fill=&quot;darkgreen&quot;)    %&gt;% collapse(375, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(385, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(343, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(388, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(397, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(338, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(519, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(506, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(282, &#39;max&#39;, fill=&quot;darkgreen&quot;) 

tree_02

 library(plotly)
ggp &lt;- ggplotly(tree_02)
 ggp


   png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png&quot;, width = 2200, height = 1500,res=400)
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)

tree_02


dev.off()


##-------------
#change tiplabel and color according to rmk or non-rmk sample
##-------------

tmp = as.character(Sterm_tree_raw$strain)
# names(tmp) &lt;- all_sterm_strain_info$shortName

# Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated&quot;, tree.names = NULL, force.multi = FALSE)
Sterm_tree_raw$tip.label
# Sterm_tree_raw$label



Sterm_tree_raw_tible &lt;- as_tibble(Sterm_tree_raw) 
Sterm_tree_raw_tible$label
Sterm_tree_raw_tible$label &lt;- gsub(&quot;_.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Streptococcus-&quot;,&quot;S. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Lactobacillus-&quot;,&quot;L. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Enterococcus-&quot;,&quot;E. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Tetragenococcus-&quot;,&quot;T. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Holzapfelia-&quot;,&quot;H. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Amylolactobacillus-&quot;,&quot;A. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;L. delbrueckii$&quot;,&quot;L. deblrueckii-delbrueckii&quot;,Sterm_tree_raw_tible$label)


tree_02_tmp &lt;- Sterm_tree_raw_tible %&gt;% as.treedata()

# Sterm_tree_raw_tible$label &lt;- plyr::revalue(Sterm_tree_raw_tible$label,tmp)
tree_02 &lt;- Sterm_tree_raw_tible%&gt;% as.treedata()


##----------------------------------
#after branch collaps
##----------------------------------

tree_03 &lt;- ggtree(tree_02, layout=&quot;circular&quot;) +
  geom_tiplab(size=3.5,align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  
   tree_04 &lt;-  scaleClade(tree_03, 285, 2) %&gt;%  scaleClade(., 348, 2)%&gt;% collapse(416, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(415, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(429, &#39;max&#39;, fill=&quot;darkgreen&quot;) %&gt;% collapse(443, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(439, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(454, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(467, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(462, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(461, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(476, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(482, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(402, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(487, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% 
collapse(376, &#39;max&#39;, fill=&quot;darkgreen&quot;)    %&gt;% collapse(375, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(385, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(343, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(388, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(397, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(338, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(519, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(506, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(282, &#39;max&#39;, fill=&quot;darkgreen&quot;) 

tree_04
###-------------------------
#ADD GENOME COUNTS
###-------------------------
 
counts_streps_lactos_ &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/genome_counts_lactos_streptos.txt&quot;,   delim = &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;genome&quot;,&quot;count&quot;))

counts_streps_lactos_$genome &lt;- gsub(&quot;Streptococcus &quot;,&quot;S. &quot;,counts_streps_lactos_$genome)
counts_streps_lactos_$genome &lt;- gsub(&quot;Lactobacillus &quot;,&quot;L. &quot;,counts_streps_lactos_$genome)
counts_streps_lactos_$genome &lt;- gsub(&quot; subsp. &quot;,&quot;-&quot;,counts_streps_lactos_$genome)

Sterm_tree_raw_tible$label &lt;- gsub(&quot;L. delbrueckii$&quot;,&quot;L. delbrueckii-delbrueckii&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;L. deblrueckii-delbrueckii&quot;,&quot;L. delbrueckii-delbrueckii&quot;,Sterm_tree_raw_tible$label)


all_sterm_strain_info_ready_01 &lt;- counts_streps_lactos_  %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;genome&quot;)

# counts_streps_lactos_$genome %in% Sterm_tree_raw_tible$label
# Sterm_tree_raw_tible$label %in% counts_streps_lactos_$genome


p3 &lt;- gheatmap(tree_04, all_sterm_strain_info_ready_01, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
p3
# &quot;L. deblrueckii-delbrueckii&quot;
###-------------------------
#ADD LAC operon
###-------------------------

lactose_streps_lactos &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/lactose_streps_lactos.txt&quot;,   delim = &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;genome&quot;,&quot;LAC&quot;))
lactose_streps_lactos$genome &lt;- gsub(&quot;_.*&quot;,&quot;&quot;,lactose_streps_lactos$genome)
lactose_streps_lactos$genome &lt;- gsub(&quot;Streptococcus-&quot;,&quot;S. &quot;,lactose_streps_lactos$genome)
lactose_streps_lactos$genome &lt;- gsub(&quot;Lactobacillus-&quot;,&quot;L. &quot;,lactose_streps_lactos$genome)

# lactose_streps_lactos$shortName &lt;- as.character(lactose_streps_lactos$shortName)
all_sterm_strain_info_ready &lt;- lactose_streps_lactos  %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;genome&quot;)
# table(all_sterm_strain_info_ready$location_continent)
 # all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c(&quot;Europe&quot;,  &quot;Asia&quot;, &quot;USA&quot;, &quot;Australia&quot; , &quot;Unknown&quot;,&quot;Outgroup&quot;))
# colors_area &lt;- c(&quot;#464D77&quot;,&quot;#36827F&quot;,&quot;#F9DB6D&quot;,&quot;grey88&quot;)
# colors_area &lt;- c(&quot;#F9DB6D&quot;,&quot;#36827F&quot;,&quot;#464D77&quot;,&quot;grey88&quot;)

# location_dataset$area

# tree_02 &lt;- full_join(as_tibble(tree), location_dataset, by=&#39;label&#39;) %&gt;% as.treedata()

# colnames(all_sterm_strain_info_ready) &lt;- c(&quot;A&quot;,&quot;B&quot;)
# rownames(all_sterm_strain_info_ready)

p3 &lt;- gheatmap(tree_04, all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
p3

###-------------------------
#clade label
###-------------------------


p4 &lt;- p3+geom_cladelab(align=TRUE,node=415, label=&quot;Oneococcus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=416, label=&quot;Leuconostoc&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2,offset=-0.2)+geom_cladelab(align=TRUE,node=429, label=&quot;Weissela&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=443, label=&quot;Limoslactibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=439, label=&quot;Paucilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=454, label=&quot;Lactiplantibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=467, label=&quot;Lentilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=461, label=&quot;Apilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=476, label=&quot;Secundilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=482, label=&quot;Levilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=402, label=&quot;Pediococcous&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=487, label=&quot;Liquorilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=376, label=&quot;Companilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=375, label=&quot;Bombilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=385, label=&quot;Lapidilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=343, label=&quot;Schleiferilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=388, label=&quot;Lacticaseibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=397, label=&quot;Latilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=338, label=&quot;Loigolactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=519, label=&quot;mix&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=506, label=&quot;Carnobacterium&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=282, label=&quot;Lactococcus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)
  
p4

 png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_03.png&quot;, width = 3000, height = 3000,res=400)
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)

p4


dev.off()

# +geom_cladelab(align=TRUE,node=462, label=&quot;Fructilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)

##-------------------------------------------------
#only tree
##-------------------------------------------------


p4 &lt;- tree_04+geom_cladelab(align=TRUE,node=415, label=&quot;Oneococcus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=416, label=&quot;Leuconostoc&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2,offset=-0.2)+geom_cladelab(align=TRUE,node=429, label=&quot;Weissela&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=443, label=&quot;Limoslactibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=439, label=&quot;Paucilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=454, label=&quot;Lactiplantibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=467, label=&quot;Lentilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=461, label=&quot;Apilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=476, label=&quot;Secundilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=482, label=&quot;Levilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=402, label=&quot;Pediococcous&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=487, label=&quot;Liquorilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=376, label=&quot;Companilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=375, label=&quot;Bombilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=385, label=&quot;Lapidilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=343, label=&quot;Schleiferilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=388, label=&quot;Lacticaseibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=397, label=&quot;Latilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=338, label=&quot;Loigolactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=519, label=&quot;mix&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=506, label=&quot;Carnobacterium&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=282, label=&quot;Lactococcus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)
  
p4

# png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_03.png&quot;, width = 3000, height = 3000,res=400)
 pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total_onlyTREE.pdf&quot;,width=10,height=10)

p4


dev.off()


##-------------------------------------------------
#add GH
##-------------------------------------------------


hmm_analysis_median &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/anlayis/HMM_GH/hmm_analysis_prepped.txt&quot;) %&gt;% mutate(genome=gsub(&quot;Lactobacillus_&quot;,&quot;L. &quot;,species)) %&gt;% mutate(genome=gsub(&quot;Streptococcus&quot;,&quot;S. &quot;,genome))


hmm_analysis_median
hmm_analysis_median

lactose_streps_lactos &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/lactose_streps_lactos.txt&quot;,   delim = &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;genome&quot;,&quot;LAC&quot;))
lactose_streps_lactos$genome &lt;- gsub(&quot;_.*&quot;,&quot;&quot;,lactose_streps_lactos$genome)
lactose_streps_lactos$genome &lt;- gsub(&quot;Streptococcus-&quot;,&quot;S. &quot;,lactose_streps_lactos$genome)
lactose_streps_lactos$genome &lt;- gsub(&quot;Lactobacillus-&quot;,&quot;L. &quot;,lactose_streps_lactos$genome)

# lactose_streps_lactos$shortName &lt;- as.character(lactose_streps_lactos$shortName)
all_sterm_strain_info_ready &lt;- lactose_streps_lactos  %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;genome&quot;)
# table(all_sterm_strain_info_ready$location_continent)
 # all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c(&quot;Europe&quot;,  &quot;Asia&quot;, &quot;USA&quot;, &quot;Australia&quot; , &quot;Unknown&quot;,&quot;Outgroup&quot;))
# colors_area &lt;- c(&quot;#464D77&quot;,&quot;#36827F&quot;,&quot;#F9DB6D&quot;,&quot;grey88&quot;)
# colors_area &lt;- c(&quot;#F9DB6D&quot;,&quot;#36827F&quot;,&quot;#464D77&quot;,&quot;grey88&quot;)

# location_dataset$area

# tree_02 &lt;- full_join(as_tibble(tree), location_dataset, by=&#39;label&#39;) %&gt;% as.treedata()

# colnames(all_sterm_strain_info_ready) &lt;- c(&quot;A&quot;,&quot;B&quot;)
# rownames(all_sterm_strain_info_ready)

p3 &lt;- gheatmap(tree_04, all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
p3</code></pre>
<pre class="r"><code>##==========================================================================================
#again 
#after adding S.vestibularis
##==========================================================================================


library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02//RAxML_bipartitions.lactos_curated&quot;, tree.names = NULL, force.multi = FALSE)

# Sterm_tree_raw &lt;-    ape::drop.tip(Sterm_tree_raw,265) #%&gt;%  ape::drop.tip(.,142) %&gt;%  ape::drop.tip(.,415)

# x &lt;- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
tree_01 &lt;- ggtree(Sterm_tree_raw, layout=&quot;circular&quot;) +
  geom_tiplab(aes(angle=angle),align = TRUE,offset = -0.2) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  tree_01
tree_01 &lt;- ggtree(Sterm_tree_raw) +
  geom_tiplab(size=0.5,align = FALSE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT
tree_01
# tmp &lt;- tree_01$data
#    tree_01$data[tree_01$data$node %in% 415, &quot;x&quot;] &lt;- 1.8
#    tree_01$data[tree_01$data$node %in% 415, &quot;branch.length&quot;] &lt;- 0.1
# 

   
   # Sterm_tree_raw$data[Sterm_tree_raw$data$node %in% c(141,142), &quot;x&quot;] &lt;- mean(Sterm_tree_raw$data$x)

   png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png&quot;, width = 2200, height = 1500,res=400)
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)


   
tree_01


dev.off()

 library(plotly)
ggp &lt;- ggplotly(tree_01)
 ggp

# 
# 415=Oneococcus
# 416=Leuconostoc
# 429=Weissela
# #154=Furfurilactobacillus #is a tip
# 443=Limoslactibacillus
# 439=Paucilactobacillus
# 454=Lactiplantibacillus
# 467=Lentilactobacillus
# 462=Fructilactobacillus
# 461=Apilactobacillus
# 476=Secundilactobacillus
# 482=Levilactobacillus
# 402=Pediococcous
# 487=Liquorilactobacillus
# 
# 348=Lactobacillus
# 
# 376=Companilactobacillus
# 375=Bombilactobacillus
# 385=Lapidilactobacillus
# 343=Schleiferilactobacillus
# #177=Agrilactobacillus #tip
# 388=Lacticaseibacillus
# 397=Latilactobacillus
# #223=Paralactobacillus #tip
# 338=Loigolactobacillus
# 519=mix
# 506=Carnobacterium
# 
# 285=Streptococcus
# 
# 282=Lactococcus

##enterococcus is polyphyletic


 
tree_02  &lt;- scaleClade(tree_01, 285, 2) %&gt;%  scaleClade(., 348, 2)%&gt;% collapse(416, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(415, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(429, &#39;max&#39;, fill=&quot;darkgreen&quot;) %&gt;% collapse(443, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(439, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(454, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(467, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(462, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(461, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(476, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(482, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(402, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(487, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% 
collapse(376, &#39;max&#39;, fill=&quot;darkgreen&quot;)    %&gt;% collapse(375, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(385, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(343, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(388, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(397, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(338, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(519, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(506, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(282, &#39;max&#39;, fill=&quot;darkgreen&quot;) 

tree_02

 library(plotly)
ggp &lt;- ggplotly(tree_02)
 ggp


   png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png&quot;, width = 2200, height = 1500,res=400)
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)

tree_02


dev.off()


##-------------
#change tiplabel and color according to rmk or non-rmk sample
##-------------

tmp = as.character(Sterm_tree_raw$strain)
# names(tmp) &lt;- all_sterm_strain_info$shortName

# Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated&quot;, tree.names = NULL, force.multi = FALSE)
Sterm_tree_raw$tip.label
# Sterm_tree_raw$label



Sterm_tree_raw_tible &lt;- as_tibble(Sterm_tree_raw) 
Sterm_tree_raw_tible$label
Sterm_tree_raw_tible$label &lt;- gsub(&quot;_.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Streptococcus-&quot;,&quot;S. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Lactobacillus-&quot;,&quot;L. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Enterococcus-&quot;,&quot;E. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Tetragenococcus-&quot;,&quot;T. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Holzapfelia-&quot;,&quot;H. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Amylolactobacillus-&quot;,&quot;A. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;L. delbrueckii$&quot;,&quot;L. deblrueckii-delbrueckii&quot;,Sterm_tree_raw_tible$label)


tree_02_tmp &lt;- Sterm_tree_raw_tible %&gt;% as.treedata()

# Sterm_tree_raw_tible$label &lt;- plyr::revalue(Sterm_tree_raw_tible$label,tmp)
# tree_02 &lt;- Sterm_tree_raw_tible%&gt;% as.treedata() %&gt;% ape::drop.tip(.,265)

# Sterm_tree_raw &lt;-    ape::drop.tip(Sterm_tree_raw,265) #%&gt;%  ape::drop.tip(.,142) %&gt;%  ape::drop.tip(.,415)

##----------------------------------
#after branch collaps
##----------------------------------

tree_03 &lt;- ggtree(tree_02, layout=&quot;circular&quot;) +
  # geom_tiplab(size=3.5,align = TRUE,offset = -0.16) + # tiplabls aligned
    geom_tiplab(size=3.5,align = TRUE,offset = -0.22) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  
   tree_04 &lt;-  tree_03 %&gt;% scaleClade(., 348, 3)%&gt;% scaleClade(., 285, 3)  %&gt;%  scaleClade(., 416, 0.2) %&gt;% scaleClade(., 415, 0.2) %&gt;% scaleClade(., 429, 0.2) %&gt;% scaleClade(., 443, 0.2) %&gt;% scaleClade(., 439, 0.2) %&gt;% scaleClade(., 454, 0.2) %&gt;% scaleClade(., 467, 0.2) %&gt;% scaleClade(., 462, 0.2) %&gt;% scaleClade(., 461, 0.2) %&gt;% scaleClade(., 476, 0.2) %&gt;% scaleClade(., 482, 0.2) %&gt;% scaleClade(., 402, 0.2) %&gt;% scaleClade(., 487, 0.2) %&gt;% scaleClade(., 376, 0.2) %&gt;% scaleClade(., 375, 0.2) %&gt;% 
   scaleClade(., 385, 0.2) %&gt;% scaleClade(., 397, 0.2) %&gt;% scaleClade(., 338, 0.2) %&gt;% scaleClade(., 519, 0.2) %&gt;% scaleClade(., 506, 0.2) %&gt;% scaleClade(., 282, 0.2) %&gt;% 
    collapse(416, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(415, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(429, &#39;max&#39;, fill=&quot;darkgreen&quot;) %&gt;% collapse(443, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(439, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(454, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(467, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(462, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(461, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(476, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(482, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(402, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(487, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% 
collapse(376, &#39;max&#39;, fill=&quot;darkgreen&quot;)    %&gt;% collapse(375, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(385, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(343, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(388, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(397, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(338, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(519, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(506, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(282, &#39;max&#39;, fill=&quot;darkgreen&quot;)
    
   
tree_04+ geom_rootedge(265)

   tree_04$data[tree_04$data$node %in% 265, &quot;x&quot;] &lt;- 0.5


# + geom_rootedge(5)
#  scaleClade(., 348, 4) %&gt;%  scaleClade(., 290, 1) %&gt;%


   tree_04 &lt;-  scaleClade(tree_03, 285, 4) %&gt;%  scaleClade(., 348, 4)%&gt;% collapse(416, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(415, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(429, &#39;max&#39;, fill=&quot;darkgreen&quot;) %&gt;% collapse(443, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(439, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(454, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(467, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(462, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(461, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(476, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(482, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(402, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(487, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% 
collapse(376, &#39;max&#39;, fill=&quot;darkgreen&quot;)    %&gt;% collapse(375, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(385, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(343, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(388, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(397, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(338, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(519, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(506, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(282, &#39;max&#39;, fill=&quot;darkgreen&quot;) 
tree_04

# 415
# groupClade(tree_03, 415)

# tmp &lt;- tree_03$data

# tree_03$data[tree_03$data$node %in% c(141,142), &quot;x&quot;] &lt;- mean(tree_03$data$x)
# tree_03$data[tree_03$data$node %in% c(141,142), &quot;y&quot;] &lt;- mean(tree_03$data$y)

# tree_03$data[tree_03$data$node %in% c(141,142), &quot;branch&quot;] &lt;- 0.5
# tree_03$data[tree_03$data$node %in% 415, &quot;branch&quot;] &lt;- 0.5
# tree_03$data[tree_03$data$node %in% 415, &quot;x&quot;] &lt;- 22.427440



# ggp &lt;- ggplotly(tree_03)
#  ggp
# + geom_hilight(285, fill=&quot;darkgreen&quot;)
###-------------------------
#ADD GENOME COUNTS
###-------------------------
 
counts_streps_lactos_ &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/genome_counts_lactos_streptos.txt&quot;,   delim = &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;genome&quot;,&quot;count&quot;))

counts_streps_lactos_$genome &lt;- gsub(&quot;Streptococcus &quot;,&quot;S. &quot;,counts_streps_lactos_$genome)
counts_streps_lactos_$genome &lt;- gsub(&quot;Lactobacillus &quot;,&quot;L. &quot;,counts_streps_lactos_$genome)
counts_streps_lactos_$genome &lt;- gsub(&quot; subsp. &quot;,&quot;-&quot;,counts_streps_lactos_$genome)

Sterm_tree_raw_tible$label &lt;- gsub(&quot;L. delbrueckii$&quot;,&quot;L. delbrueckii-delbrueckii&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;L. deblrueckii-delbrueckii&quot;,&quot;L. delbrueckii-delbrueckii&quot;,Sterm_tree_raw_tible$label)


all_sterm_strain_info_ready_01 &lt;- counts_streps_lactos_  %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;genome&quot;)

# counts_streps_lactos_$genome %in% Sterm_tree_raw_tible$label
# Sterm_tree_raw_tible$label %in% counts_streps_lactos_$genome


p3 &lt;- gheatmap(tree_04, all_sterm_strain_info_ready_01, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
p3
# &quot;L. deblrueckii-delbrueckii&quot;
###-------------------------
#ADD LAC operon
###-------------------------

lactose_streps_lactos &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/lactose_streps_lactos.txt&quot;,   delim = &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;genome&quot;,&quot;LAC&quot;))
lactose_streps_lactos$genome &lt;- gsub(&quot;_.*&quot;,&quot;&quot;,lactose_streps_lactos$genome)
lactose_streps_lactos$genome &lt;- gsub(&quot;Streptococcus-&quot;,&quot;S. &quot;,lactose_streps_lactos$genome)
lactose_streps_lactos$genome &lt;- gsub(&quot;Lactobacillus-&quot;,&quot;L. &quot;,lactose_streps_lactos$genome)

# lactose_streps_lactos$shortName &lt;- as.character(lactose_streps_lactos$shortName)
all_sterm_strain_info_ready &lt;- lactose_streps_lactos  %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;genome&quot;)
# table(all_sterm_strain_info_ready$location_continent)
 # all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c(&quot;Europe&quot;,  &quot;Asia&quot;, &quot;USA&quot;, &quot;Australia&quot; , &quot;Unknown&quot;,&quot;Outgroup&quot;))
# colors_area &lt;- c(&quot;#464D77&quot;,&quot;#36827F&quot;,&quot;#F9DB6D&quot;,&quot;grey88&quot;)
# colors_area &lt;- c(&quot;#F9DB6D&quot;,&quot;#36827F&quot;,&quot;#464D77&quot;,&quot;grey88&quot;)

# location_dataset$area

# tree_02 &lt;- full_join(as_tibble(tree), location_dataset, by=&#39;label&#39;) %&gt;% as.treedata()

# colnames(all_sterm_strain_info_ready) &lt;- c(&quot;A&quot;,&quot;B&quot;)
# rownames(all_sterm_strain_info_ready)

p3 &lt;- gheatmap(tree_04, all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
p3

###-------------------------
#clade label
###-------------------------


p4 &lt;- p3+geom_cladelab(align=TRUE,node=415, label=&quot;Oneococcus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=416, label=&quot;Leuconostoc&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2,offset=-0.2)+geom_cladelab(align=TRUE,node=429, label=&quot;Weissela&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=443, label=&quot;Limoslactibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=439, label=&quot;Paucilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=454, label=&quot;Lactiplantibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=467, label=&quot;Lentilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=461, label=&quot;Apilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=476, label=&quot;Secundilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=482, label=&quot;Levilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=402, label=&quot;Pediococcous&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=487, label=&quot;Liquorilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=376, label=&quot;Companilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=375, label=&quot;Bombilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=385, label=&quot;Lapidilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=343, label=&quot;Schleiferilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=388, label=&quot;Lacticaseibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=397, label=&quot;Latilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=338, label=&quot;Loigolactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=519, label=&quot;mix&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=506, label=&quot;Carnobacterium&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=282, label=&quot;Lactococcus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)
  

p4
#  png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_03.png&quot;, width = 3000, height = 3000,res=400)
# # pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)
# 
# p4
# 
# 
# dev.off()

# +geom_cladelab(align=TRUE,node=462, label=&quot;Fructilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)
###-------------------------
#clade label
###-------------------------


p5 &lt;-tree_04+geom_cladelab(align=TRUE,node=415, label=&quot;Oneococcus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=416, label=&quot;Leuconostoc&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2,offset=-0.2)+geom_cladelab(align=TRUE,node=429, label=&quot;Weissela&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=443, label=&quot;Limoslactibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=439, label=&quot;Paucilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=454, label=&quot;Lactiplantibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=467, label=&quot;Lentilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=461, label=&quot;Apilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=476, label=&quot;Secundilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=482, label=&quot;Levilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=402, label=&quot;Pediococcous&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=487, label=&quot;Liquorilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=376, label=&quot;Companilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=375, label=&quot;Bombilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=385, label=&quot;Lapidilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=343, label=&quot;Schleiferilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=388, label=&quot;Lacticaseibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=397, label=&quot;Latilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=338, label=&quot;Loigolactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=519, label=&quot;mix&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=506, label=&quot;Carnobacterium&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=282, label=&quot;Lactococcus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)
  
p5


###-------------------------
#zoom into strepto and lacto taxons
###-------------------------

p6 &lt;- p5 + geom_hilight(285, fill=&quot;darkgreen&quot;)
p6



p6 &lt;- p5 %&gt;% scaleClade(285, scale=0.3)
p6

# 285=Streptococcus
# # 348=Lactobacillus

##enlarge center
# + geom_rootedge(5)</code></pre>
<p>function needed in next chunkg</p>
<pre class="r"><code>offspring.tbl_tree_item &lt;- function(.data, .node, ...) {
    valid.tbl_tree(.data)

    if (is.character(.node)) {
        .node &lt;- .data$node[.data$label == .node]
    }

    .data[.data$parent == .node &amp; .data$parent != .data$node,]
}

##&#39; @method offspring tbl_tree
##&#39; @export
##&#39; @rdname offspring
##&#39; @examples
##&#39; library(ape)
##&#39; tree &lt;- rtree(4)
##&#39; x &lt;- as_tibble(tree)
##&#39; offspring(x, 4)
offspring.tbl_tree &lt;- function(.data, .node, tiponly = FALSE, self_include = FALSE, ...) {
    if (missing(.node) || is.null(.node)) {
        stop(&quot;.node is required&quot;)
    }
    if (length(.node) == 1) {
        res &lt;- .offspring.tbl_tree_item(.data = .data, .node = .node,
                                       tiponly = tiponly, self_include = self_include, ...)
    } else {
        res &lt;- lapply(.node, function(node) {
            .offspring.tbl_tree_item(.data = .data, .node = node,
                                    tiponly = tiponly, self_include = self_include, ...)
        })
        names(res) &lt;- .node
    }
    return(res)
}

#&#39; @noRd
#&#39; @keywords internal
.offspring.tbl_tree_item &lt;- function(.data, .node, tiponly = FALSE, self_include = FALSE, ...) {
    x &lt;- child.tbl_tree(.data, .node)

    ## https://github.com/GuangchuangYu/ggtree/issues/239
    rn &lt;- rootnode.tbl_tree(.data)$node
    x &lt;- x[x$node != rn, ]

    if (nrow(x) == 0) {
        if (self_include) {
            x &lt;- .data[.data$node == .node, ]
        } 

        return(x)
    }

    ## id &lt;- x$node
    ## i &lt;- 1
    ## while(i &lt;= length(id)) {
    ##     id &lt;- c(id, child(.data, id[i])$node)
    ##     i &lt;- i + 1
    ## }
    ## filter_(.data, ~ node %in% id)

    parent &lt;- .data$parent
    children &lt;- .data$node
    ## n &lt;- length(parent)
    n &lt;- max(parent)

    kids &lt;- vector(&quot;list&quot;, n)
    for (i in seq_along(parent)) {
        kids[[parent[i]]] &lt;-c(kids[[parent[i]]], children[i])
    }

    id &lt;- x$node
    i &lt;- 1
    while(i &lt;= length(id)) {
        id &lt;- c(id, kids[[id[i]]])
        i &lt;- i + 1
    }

    if (self_include) {
        id &lt;- c(.node, id)
    }

    sp &lt;- .data[children %in% id,]
    if (tiponly) {
        return(sp[sp$node &lt; rn,])
    }
    return(sp)
}

##&#39; @method child phylo
##&#39; @export
child.phylo &lt;- function(.data, .node, type = &#39;children&#39;, ...) {
    res &lt;- offspring(.data=.data, .node = .node, type = type)
    return(res)
}

##&#39; @method child treedata
##&#39; @export
child.treedata &lt;- function(.data, .node, type = &#39;children&#39;, ...) {
    child.phylo(as.phylo(.data), .node, type = type, ...)
}

.internal.child &lt;- function(data, node, type = &#39;children&#39;){
    if (!is_numeric(node)){
        all.labs &lt;- c(data$tip.label, data$node.label)
        names(all.labs) &lt;- seq_len(length(all.labs))
        node &lt;- names(all.labs[all.labs %in% node])
    }
    edge &lt;- data$edge
    res &lt;- edge[edge[,1] == node, 2]
    if (type != &#39;children&#39;){
        alltips &lt;- edge[,2][! edge[,2] %in% edge[,1]]
        w &lt;- which(res &gt;= length(alltips))
        if(length(w)&gt;0){
            for(i in 1:length(w)){
                res &lt;- c(res,
                         .internal.child(
                           data = data,
                           node = res[w[i]],
                           type = type
                         )
                       )
            }
        }
        if (type %in% c(&#39;tips&#39;, &#39;external&#39;)){
            res &lt;- res[res %in% alltips]
        }else if (type == &quot;internal&quot;) {
            res &lt;- res[!res %in% alltips]
        }
    }
    return(unname(res))
}

##&#39; @method offspring phylo
##&#39; @export
offspring.phylo &lt;- function(.data, .node, tiponly = FALSE, self_include = FALSE, type = &#39;all&#39;, ...){
    type &lt;- match.arg(type, c(&quot;children&quot;, &#39;tips&#39;, &#39;internal&#39;, &#39;external&#39;, &#39;all&#39;))

    if (tiponly){
        message(&#39;The &quot;tiponly = TRUE&quot; can be replaced by type=&quot;tips&quot;.&#39;)
        type = &#39;tips&#39;
    }

    res &lt;- lapply(.node, .internal.child, data = .data, type = type)
    if (length(res) &lt;= 1){
        res &lt;- unlist(res)
        if (self_include){
            res &lt;- c(.node, res)
        }
    }else{
        if (self_include){
            res &lt;- mapply(append, .node, res, SIMPLIFY=FALSE)
        }
        names(res) &lt;- .node
    }
    return (res)
    #if (self_include) {
    #    sp &lt;- .node
    #} else {
    #    sp &lt;- child(.data, .node)
    #}

    #sp &lt;- sp[sp != 0]
    #if (length(sp) == 0) {
    #    return(sp)
    #    ## stop(&quot;input node is a tip...&quot;)
    #}
    #i &lt;- 1
    #while (i &lt;= length(sp)) {
    #    sp &lt;- c(sp, child(.data, sp[i]))
    #    sp &lt;- sp[sp != 0]
    #    i &lt;- i + 1
    #}
    #if (tiponly) {
    #    return(sp[sp &lt;= Ntip(.data)])
    #}
    #return(sp)
}


##&#39; @method offspring treedata
##&#39; @export
offspring.treedata &lt;- function(.data, .node, tiponly = FALSE, self_include = FALSE, type = &#39;all&#39;, ...) {
    offspring.phylo(as.phylo(.data), .node,
                    tiponly = tiponly, self_include = self_include,
                    type = type,
                    ...)
}


valid.tbl_tree &lt;- function(object, cols = c(&quot;parent&quot;, &quot;node&quot;, &quot;label&quot;)) {
    cc &lt;- cols[!cols %in% colnames(object)]
    if (length(cc) &gt; 0) {
        msg &lt;- paste0(&quot;invalid tbl_tree object.\n  missing column:\n    &quot;, paste(cc, collapse=&quot;,&quot;), &quot;.&quot;)
    }
}</code></pre>
</div>
<div id="final-make-tree" class="section level3">
<h3>final make tree</h3>
<p>remove outgroup</p>
<pre class="r"><code>##==========================================================================================
#again 
#after adding S.vestibularis
##==========================================================================================


library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

# Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02//RAxML_bipartitions.lactos_curated&quot;, tree.names = NULL, force.multi = FALSE)

Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03//RAxML_bipartitions.lactos_curated&quot;, tree.names = NULL, force.multi = FALSE)

##------------------------------remove root! Fusobacterium-nucleatum_G000158275


Sterm_tree_raw %&gt;% as_tibble() %&gt;% filter(label==&quot;Fusobacterium-nucleatum_G000158275&quot;) 
Sterm_tree_raw &lt;-    ape::drop.tip(Sterm_tree_raw,274) #%&gt;%  ape::drop.tip(.,142) %&gt;%  ape::drop.tip(.,415)



# x &lt;- as_tibble(Sterm_tree_raw)

##plot tree
# ggplot(Sterm_tree_raw, aes(x, y)) + geom_tree() + theme_tree()
tree_01 &lt;- ggtree(Sterm_tree_raw, layout=&quot;fan&quot;) +
  geom_tiplab(aes(angle=angle),align = TRUE,offset = -0.2) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  tree_01
tree_01 &lt;- ggtree(Sterm_tree_raw) +
  geom_tiplab(size=0.5,align = FALSE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT
tree_01
# tmp &lt;- tree_01$data
#    tree_01$data[tree_01$data$node %in% 415, &quot;x&quot;] &lt;- 1.8
#    tree_01$data[tree_01$data$node %in% 415, &quot;branch.length&quot;] &lt;- 0.1
# 

   
   # Sterm_tree_raw$data[Sterm_tree_raw$data$node %in% c(141,142), &quot;x&quot;] &lt;- mean(Sterm_tree_raw$data$x)

   png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png&quot;, width = 2200, height = 1500,res=400)
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)


   
tree_01


dev.off()

 library(plotly)
ggp &lt;- ggplotly(tree_01)
 ggp

# 
# 484=Oneococcus
# 486=Leuconostoc
# 474=Weissela
# ----------------------------------------#196=Furfurilactobacillus #is a tip
# 458=Limoslactibacillus
# 468=Paucilactobacillus
# 498=Lactiplantibacillus #!
# 447=Lentilactobacillus
# 444=Fructilactobacillus
# 442=Apilactobacillus
# 474=Secundilactobacillus
# 480=Levilactobacillus
# 400=Pediococcous
# 485=Liquorilactobacillus
# 
# 345=Lactobacillus
# 
# 374=Companilactobacillus
# 373=Bombilactobacillus
# 383=Lapidilactobacillus
# 341=Schleiferilactobacillus
# #177=Agrilactobacillus #tip
# 386=Lacticaseibacillus
# 395=Latilactobacillus
# #223=Paralactobacillus #tip
# 336=Loigolactobacillus
# 517=mix
## 509=Carnobacterium
 # 504=Carnobacterium

# 
# 288=Streptococcus
# 
# 280=Lactococcus

 
 ###---------------------------------find nodes automatically.
 ------------------------------------#196=Furfurilactobacillus #is a tip
#---------------------------=Agrilactobacillus #tip
#------------------------------   =Paralactobacillus #tip
#!!!!!mix
 # 
 # nodes_genus &lt;- data.frame()
 # nodes_genus_new &lt;- list()
for (genusss in c(&quot;Oenococcus&quot;,&quot;Leuconostoc&quot;,&quot;Weissella&quot;,&quot;Limosilactobacillus&quot;,&quot;Paucilactobacillus&quot;,&quot;Lactiplantibacillus&quot;,&quot;Lentilactobacillus&quot;,&quot;Fructilactobacillus&quot;,&quot;Apilactobacillus&quot;,&quot;Secundilactobacillus&quot;,&quot;Levilactobacillus&quot;,&quot;Pediococcus&quot;,&quot;Liquorilactobacillus&quot;,&quot;Lactobacillus&quot;,&quot;Companilactobacillus&quot;,&quot;Bombilactobacillus&quot;,&quot;Lapidilactobacillus&quot;,&quot;Schleiferilactobacillus&quot;,&quot;Lacticaseibacillus&quot;,&quot;Latilactobacillus&quot;,&quot;Loigolactobacillus&quot;,&quot;Carnobacterium&quot;,&quot;Carnobacterium&quot;,&quot;Streptococcus&quot;,&quot;Lactococcus&quot;,&quot;Ligilactobacillus&quot;)) {
   print(paste0(&quot;node_&quot;,genusss))
     tmp &lt;- Sterm_tree_raw$tip.label[grepl(genusss,Sterm_tree_raw$tip.label)]
     tmp1 &lt;- getMRCA(Sterm_tree_raw,tmp)

    # tmp2 &lt;- data.frame(genus=genusss,node=tmp1)   
    # nodes_genus &lt;- rbind(nodes_genus,tmp2)
    
    # nodes_genus_new[genusss] &lt;- tmp1
    
      assign(paste0(&quot;node_&quot;,genusss), tmp1)

    # return(c(paste0(&quot;node_&quot;,genusss), tmp1))
    # paste0(&quot;node_&quot;,genusss) &lt;- tmp1
 }
 
 node_Lactobacillus &lt;- 356
 node_mix=535
  node_Carnobacterium=522

 
 library(tidytree)
Sterm_tree_raw$tip.label[(grepl(&quot;Li&quot;,Sterm_tree_raw$tip.label))]

##enterococcus is polyphyletic


tree_02  &lt;- scaleClade(tree_01, node_Streptococcus, 2) %&gt;%  scaleClade(., node_Lactobacillus, 2)%&gt;% collapse(node_Oenococcus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Leuconostoc, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(node_Weissella, &#39;max&#39;, fill=&quot;darkgreen&quot;) %&gt;% collapse(node_Limosilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Paucilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Lactiplantibacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Lentilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(node_Fructilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(node_Apilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(node_Secundilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(node_Levilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(node_Pediococcus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Liquorilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% 
collapse(node_Companilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)    %&gt;% collapse(node_Bombilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Lapidilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Schleiferilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Lacticaseibacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Latilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Loigolactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Carnobacterium, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Lactococcus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_mix, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Ligilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;) 

tree_02

 
# tree_02  &lt;- scaleClade(tree_01, 288, 2) %&gt;%  scaleClade(., 345, 2)%&gt;% collapse(412, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(415, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(427, &#39;max&#39;, fill=&quot;darkgreen&quot;) %&gt;% collapse(441, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(436, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(452, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(465, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(460, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(463, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(474, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(480, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(400, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(485, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% 
# collapse(374, &#39;max&#39;, fill=&quot;darkgreen&quot;)    %&gt;% collapse(373, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(383, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(341, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(386, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(395, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(336, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(517, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(504, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(280, &#39;max&#39;, fill=&quot;darkgreen&quot;) 


 library(plotly)
ggp &lt;- ggplotly(tree_02)
 ggp


   png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_01.png&quot;, width = 2200, height = 1500,res=400)
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)

tree_02


dev.off()


##-------------
#change tiplabel and color according to rmk or non-rmk sample
##-------------

tmp = as.character(Sterm_tree_raw$strain)
# names(tmp) &lt;- all_sterm_strain_info$shortName

# Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_new/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_curated&quot;, tree.names = NULL, force.multi = FALSE)
Sterm_tree_raw$tip.label
# Sterm_tree_raw$label



Sterm_tree_raw_tible &lt;- as_tibble(Sterm_tree_raw) 
Sterm_tree_raw_tible$label
Sterm_tree_raw_tible$label &lt;- gsub(&quot;_.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Streptococcus-&quot;,&quot;S. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Lactobacillus-&quot;,&quot;L. &quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Enterococcus-.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Tetragenococcus-.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Holzapfelia-.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Amylolactobacillus-.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;L. delbrueckii$&quot;,&quot;L. delbrueckii-delbrueckii&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Amylolactobacillus-.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Meliss.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Vagoc.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Catel.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Paralacto.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Agrilacto.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;L. delbrueckii-&quot;,&quot;L. del-&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;Furfuri.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)
Sterm_tree_raw_tible$label &lt;- gsub(&quot;^&#39;An.*&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)


# Sterm_tree_raw_tible$label &lt;- paste0(&quot;italic(&quot;,Sterm_tree_raw_tible$label,&quot;)&quot;)


# tree_05_as_tibble &lt;- tree_03 %&gt;% as_tibble()
# Sterm_tree_raw_tible$label &lt;- ifelse(is.na(Sterm_tree_raw_tible$label),&quot;NA&quot;,paste0(&quot;italic(&quot;,Sterm_tree_raw_tible$label,&quot;)&quot;))
# Sterm_tree_raw_tible$label &lt;- gsub(&quot;italic()&quot;,&quot;&quot;,Sterm_tree_raw_tible$label)

tree_02_tmp &lt;- Sterm_tree_raw_tible %&gt;% as.treedata()


# tree_02_tmp &lt;- Sterm_tree_raw_tible %&gt;% as.treedata()


tree_02 &lt;- Sterm_tree_raw_tible%&gt;% as.treedata()
# Sterm_tree_raw_tible$label &lt;- plyr::revalue(Sterm_tree_raw_tible$label,tmp)
# tree_02 &lt;- Sterm_tree_raw_tible%&gt;% as.treedata() %&gt;% ape::drop.tip(.,265)

# Sterm_tree_raw &lt;-    ape::drop.tip(Sterm_tree_raw,265) #%&gt;%  ape::drop.tip(.,142) %&gt;%  ape::drop.tip(.,415)

##----------------------------------
#after branch collaps
##----------------------------------



# tree_03 &lt;- ggtree(tree_02, layout=&quot;circular&quot;,open.angle = 90) +
  tree_03 &lt;- ggtree(tree_02, layout=&quot;fan&quot;,open.angle = 50) +geom_rootedge(rootedge = 0)+geom_tiplab(size=3.5,align = TRUE,offset = 0.06,linesize = 0,fontface = &#39;italic&#39;) 
            

  tree_03
  # geom_tiplab(size=3.5,align = TRUE,offset = 0.06,linesize = 0,aes(label=paste0(&#39;italic(&#39;, label, &#39;)&#39;)) )




  # geom_tiplab(size=3.5,align = TRUE,offset = -0.16) + # tiplabls aligned
    # geom_tiplab(size=3.5,align = TRUE,offset = -0.06,linesize = 0) + # tiplabls aligned
      # + # tiplabls aligned

   #+open_tree( 270)
tree_03


# tree_04  &lt;- scaleClade(tree_03, 279, 3.5) %&gt;%  scaleClade(., 345,3.5) %&gt;%  scaleClade(., 412, 0.2)  %&gt;%  scaleClade(., 415, 0.2)   %&gt;%  scaleClade(., 427, 0.2) %&gt;%  scaleClade(., 441, 0.2)  %&gt;%  scaleClade(., 436, 0.2)  %&gt;%  scaleClade(., 452, 0.2)  %&gt;%  scaleClade(., 465, 0.2)   %&gt;%  scaleClade(., 460, 0.2)   %&gt;%  scaleClade(., 463, 0.2)   %&gt;%  scaleClade(., 474, 0.2)   %&gt;%  scaleClade(., 480, 0.2)   %&gt;%  scaleClade(., 400, 0.2)  %&gt;%  scaleClade(., 485, 0.2)  %&gt;% 
#  scaleClade(., 374, 0.2)    %&gt;%  scaleClade(., 373, 0.2)  %&gt;%  scaleClade(., 383, 0.2)  %&gt;%  scaleClade(., 341, 0.2)  %&gt;%  scaleClade(., 386, 0.2)  %&gt;%  scaleClade(., 395, 0.2)  %&gt;%  scaleClade(., 336, 0.2)  %&gt;%  scaleClade(., 517, 0.2)  %&gt;%  scaleClade(., 504, 0.2)  %&gt;%  scaleClade(., 280, 0.2) %&gt;% collapse(412, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% collapse(415, &#39;max&#39;, fill=&quot;grey78&quot;)   %&gt;% collapse(427, &#39;max&#39;, fill=&quot;grey78&quot;) %&gt;% collapse(441, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% collapse(436, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% collapse(452, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% collapse(465, &#39;max&#39;, fill=&quot;grey78&quot;)   %&gt;% collapse(460, &#39;max&#39;, fill=&quot;grey78&quot;)   %&gt;% collapse(463, &#39;max&#39;, fill=&quot;grey78&quot;)   %&gt;% collapse(474, &#39;max&#39;, fill=&quot;grey78&quot;)   %&gt;% collapse(480, &#39;max&#39;, fill=&quot;grey78&quot;)   %&gt;% collapse(400, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% collapse(485, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% 
# collapse(374, &#39;max&#39;, fill=&quot;grey78&quot;)    %&gt;% collapse(373, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% collapse(383, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% collapse(341, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% collapse(386, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% collapse(395, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% collapse(336, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% collapse(517, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% collapse(504, &#39;max&#39;, fill=&quot;grey78&quot;)  %&gt;% collapse(280, &#39;max&#39;, fill=&quot;grey78&quot;) 



tree_04  &lt;- scaleClade(tree_03, node_Streptococcus, 2.8) %&gt;%  scaleClade(., node_Lactobacillus,2.8) %&gt;%  scaleClade(., node_Oenococcus, 0.2)  %&gt;%  scaleClade(., node_Leuconostoc, 0.2)   %&gt;%  scaleClade(., node_Weissella, 0.2) %&gt;%  scaleClade(., node_Limosilactobacillus, 0.2)  %&gt;%  scaleClade(., node_Paucilactobacillus, 0.2)  %&gt;%  scaleClade(., node_Lactiplantibacillus, 0.2)  %&gt;%  scaleClade(., node_Lentilactobacillus, 0.2)   %&gt;%  scaleClade(., node_Fructilactobacillus, 0.2)   %&gt;%  scaleClade(., 463, 0.2)   %&gt;%  scaleClade(., 474, 0.2)   %&gt;%  scaleClade(., node_Apilactobacillus, 0.2)   %&gt;%  scaleClade(., node_Secundilactobacillus, 0.2)  %&gt;%  scaleClade(., node_Levilactobacillus, 0.2)  %&gt;% 
 scaleClade(., node_Pediococcus, 0.2)    %&gt;%  scaleClade(., node_Liquorilactobacillus, 0.2)  %&gt;%  scaleClade(., node_Companilactobacillus, 0.2)  %&gt;%  scaleClade(., node_Bombilactobacillus, 0.2)  %&gt;%  scaleClade(., node_Lapidilactobacillus, 0.2)  %&gt;%  scaleClade(., node_Schleiferilactobacillus, 0.2)  %&gt;%  scaleClade(., node_Lacticaseibacillus, 0.2)  %&gt;%  scaleClade(., node_Latilactobacillus, 0.2)  %&gt;%  scaleClade(., node_Loigolactobacillus, 0.2)%&gt;%  scaleClade(., node_Carnobacterium, 0.2)  %&gt;%  scaleClade(., node_Lactococcus, 0.2) %&gt;%  scaleClade(., node_mix, 0.2)  %&gt;%  scaleClade(., node_Ligilactobacillus, 0.2) %&gt;%
  collapse(node_Oenococcus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Leuconostoc, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(node_Weissella, &#39;max&#39;, fill=&quot;darkgreen&quot;) %&gt;% collapse(node_Limosilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Paucilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Lactiplantibacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Lentilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(node_Fructilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(node_Apilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(node_Secundilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(node_Levilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)   %&gt;% collapse(node_Pediococcus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Liquorilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% 
collapse(node_Companilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)    %&gt;% collapse(node_Bombilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Lapidilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Schleiferilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Lacticaseibacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Latilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Loigolactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Carnobacterium, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Lactococcus, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_mix, &#39;max&#39;, fill=&quot;darkgreen&quot;)  %&gt;% collapse(node_Ligilactobacillus, &#39;max&#39;, fill=&quot;darkgreen&quot;) %&gt;% rotate_tree(., 54)
tree_04

# 
# # ggp &lt;- ggplotly(tree_03)
# #  ggp
# # + geom_hilight(285, fill=&quot;darkgreen&quot;)
# ###-------------------------
# #ADD GENOME COUNTS
# ###-------------------------
#  
# counts_streps_lactos_ &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/genome_counts_lactos_streptos.txt&quot;,   delim = &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;genome&quot;,&quot;count&quot;))
# 
# counts_streps_lactos_$genome &lt;- gsub(&quot;Streptococcus &quot;,&quot;S. &quot;,counts_streps_lactos_$genome)
# counts_streps_lactos_$genome &lt;- gsub(&quot;Lactobacillus &quot;,&quot;L. &quot;,counts_streps_lactos_$genome)
# counts_streps_lactos_$genome &lt;- gsub(&quot; subsp. &quot;,&quot;-&quot;,counts_streps_lactos_$genome)
# 
# Sterm_tree_raw_tible$label &lt;- gsub(&quot;L. delbrueckii$&quot;,&quot;L. delbrueckii-delbrueckii&quot;,Sterm_tree_raw_tible$label)
# Sterm_tree_raw_tible$label &lt;- gsub(&quot;L. deblrueckii-delbrueckii&quot;,&quot;L. delbrueckii-delbrueckii&quot;,Sterm_tree_raw_tible$label)
# 
# 
# all_sterm_strain_info_ready_01 &lt;- counts_streps_lactos_  %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;genome&quot;)
# 
# # counts_streps_lactos_$genome %in% Sterm_tree_raw_tible$label
# # Sterm_tree_raw_tible$label %in% counts_streps_lactos_$genome
# 
# 
# p3 &lt;- gheatmap(tree_04, all_sterm_strain_info_ready_01, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
# p3
# &quot;L. deblrueckii-delbrueckii&quot;
###-------------------------
#ADD LAC operon
###-------------------------
# 
# lactose_streps_lactos &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/lactose_streps_lactos.txt&quot;,   delim = &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;genome&quot;,&quot;LAC&quot;))
# lactose_streps_lactos$genome &lt;- gsub(&quot;_.*&quot;,&quot;&quot;,lactose_streps_lactos$genome)
# lactose_streps_lactos$genome &lt;- gsub(&quot;Streptococcus-&quot;,&quot;S. &quot;,lactose_streps_lactos$genome)
# lactose_streps_lactos$genome &lt;- gsub(&quot;Lactobacillus-&quot;,&quot;L. &quot;,lactose_streps_lactos$genome)
# 
# # lactose_streps_lactos$shortName &lt;- as.character(lactose_streps_lactos$shortName)
# all_sterm_strain_info_ready &lt;- lactose_streps_lactos  %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;genome&quot;)
# # table(all_sterm_strain_info_ready$location_continent)
#  # all_sterm_strain_info_ready$location_continent = factor(all_sterm_strain_info_ready$location_continent, levels=c(&quot;Europe&quot;,  &quot;Asia&quot;, &quot;USA&quot;, &quot;Australia&quot; , &quot;Unknown&quot;,&quot;Outgroup&quot;))
# # colors_area &lt;- c(&quot;#464D77&quot;,&quot;#36827F&quot;,&quot;#F9DB6D&quot;,&quot;grey88&quot;)
# # colors_area &lt;- c(&quot;#F9DB6D&quot;,&quot;#36827F&quot;,&quot;#464D77&quot;,&quot;grey88&quot;)
# 
# # location_dataset$area
# 
# # tree_02 &lt;- full_join(as_tibble(tree), location_dataset, by=&#39;label&#39;) %&gt;% as.treedata()
# 
# # colnames(all_sterm_strain_info_ready) &lt;- c(&quot;A&quot;,&quot;B&quot;)
# # rownames(all_sterm_strain_info_ready)
# 
# p3 &lt;- gheatmap(tree_04, all_sterm_strain_info_ready, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
# p3

###-------------------------
#clade label
###-------------------------

# 
# p4 &lt;- p3+geom_cladelab(align=TRUE,node=415, label=&quot;Oneococcus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=416, label=&quot;Leuconostoc&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2,offset=-0.2)+geom_cladelab(align=TRUE,node=429, label=&quot;Weissela&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=443, label=&quot;Limoslactibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=439, label=&quot;Paucilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=454, label=&quot;Lactiplantibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=467, label=&quot;Lentilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=461, label=&quot;Apilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=476, label=&quot;Secundilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=482, label=&quot;Levilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=402, label=&quot;Pediococcous&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=487, label=&quot;Liquorilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=376, label=&quot;Companilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=375, label=&quot;Bombilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=385, label=&quot;Lapidilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=343, label=&quot;Schleiferilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=388, label=&quot;Lacticaseibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=397, label=&quot;Latilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=338, label=&quot;Loigolactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=519, label=&quot;mix&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=506, label=&quot;Carnobacterium&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=282, label=&quot;Lactococcus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)
#   
# 
# p4
#  png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/pyhlogeny_tof_03.png&quot;, width = 3000, height = 3000,res=400)
# # pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)
# 
# p4
# 
# 
# dev.off()

# +geom_cladelab(align=TRUE,node=462, label=&quot;Fructilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)
###-------------------------
#clade label
###-------------------------
# 
# 
# p5 &lt;-tree_04+geom_cladelab(align=TRUE,node=415, label=&quot;Oneococcus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=416, label=&quot;Leuconostoc&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2,offset=-0.2)+geom_cladelab(align=TRUE,node=429, label=&quot;Weissela&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=443, label=&quot;Limoslactibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=439, label=&quot;Paucilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=454, label=&quot;Lactiplantibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=467, label=&quot;Lentilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=461, label=&quot;Apilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=476, label=&quot;Secundilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=482, label=&quot;Levilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=402, label=&quot;Pediococcous&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=487, label=&quot;Liquorilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=376, label=&quot;Companilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=375, label=&quot;Bombilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=385, label=&quot;Lapidilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=343, label=&quot;Schleiferilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=388, label=&quot;Lacticaseibacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=397, label=&quot;Latilactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=338, label=&quot;Loigolactobacillus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=519, label=&quot;mix&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=506, label=&quot;Carnobacterium&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)+geom_cladelab(align=TRUE,node=282, label=&quot;Lactococcus&quot;,geom=&quot;text&quot;,angle=&#39;auto&#39;,offset=-0.2)
#   
# p5


###-------------------------
#add EMA data
###-------------------------

Species_distribution &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/Earthmicrobiome_data/Final/Final/VinceDB_Species_distribution.csv&quot;)

Species_distribution &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/Earthmicrobiome_data/Final_03/Final/FinalVinceDB_Species_distribution.csv&quot;) %&gt;% mutate(Species_name=qseqid)


table(Species_distribution$Species_name)

unique(Species_distribution$Species_name)
unique(Sterm_tree_raw_tible$label)


Species_distribution$Species_name &lt;- gsub(&quot;Streptococcus-&quot;,&quot;S. &quot;,Species_distribution$Species_name)
Species_distribution$Species_name &lt;- gsub(&quot;Lactobacillus-&quot;,&quot;L. &quot;,Species_distribution$Species_name)
Species_distribution$Species_name &lt;- gsub(&quot;Lactobacillus-&quot;,&quot;L. &quot;,Species_distribution$Species_name)
Species_distribution$Species_name &lt;- gsub(&quot;L. delbrueckii-&quot;,&quot;L. del-&quot;,Species_distribution$Species_name)
Species_distribution$Species_name &lt;- gsub(&quot;L. -bulgaricus&quot;,&quot;L. del-bulgaricus&quot;,Species_distribution$Species_name)

unique(Species_distribution$Species_name)[!unique(Species_distribution$Species_name) %in% unique(Sterm_tree_raw_tible$label)]


unique(Species_distribution$Habitat)

Species_distribution$Habitat_broad &lt;- Species_distribution$Habitat %&gt;% gsub(&quot;Animal.*&quot;,&quot;Animal sampling&quot;,.)%&gt;% gsub(&quot;Plant.*&quot;,&quot;Plant sampling&quot;,.)%&gt;% gsub(&quot;Sediment.*&quot;,&quot;Soil sampling&quot;,.)%&gt;% gsub(&quot;Soil.*&quot;,&quot;Soil sampling&quot;,.) %&gt;% gsub(&quot;Surface.*&quot;,&quot;Environemental sampling&quot;,.)%&gt;% gsub(&quot;Water.*&quot;,&quot;Water sampling&quot;,.)

table(Species_distribution$Habitat_broad)

Species_distribution_sums &lt;- Species_distribution %&gt;% group_by(Species_name) %&gt;% summarise(total=sum(Standardized_MeanCount_per_sample))

Species_distribution_preped &lt;- Species_distribution %&gt;% select(Species_name,Habitat_broad,Standardized_MeanCount_per_sample) 

Species_distribution_preped &lt;- Species_distribution_preped %&gt;% group_by(Species_name,Habitat_broad) %&gt;% summarise(Standardized_MeanCount_per_sample=sum(Standardized_MeanCount_per_sample)) %&gt;% filter(Standardized_MeanCount_per_sample&gt;0)

 tmp &lt;- Species_distribution_preped %&gt;% group_by(Species_name) %&gt;% summarise(total=sum(Standardized_MeanCount_per_sample))

 
 table(Species_distribution_preped$Species_name)
 
 
 write.csv(Species_distribution_preped, &quot;~/Desktop/Projects/2020_StarterCultureDiversity/Earthmicrobiome_data/Final/Final/VinceDB_Species_distribution_prepeed.csv&quot;)
 
# is.na(Species_distribution_preped$Habitat_broad) %&gt;%
# counts_streps_lactos_ &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/genome_counts_lactos_streptos.txt&quot;,   delim = &quot;\t&quot;, escape_double = FALSE, trim_ws = TRUE,col_names = c(&quot;genome&quot;,&quot;count&quot;))
# 
# counts_streps_lactos_$genome &lt;- gsub(&quot;Streptococcus &quot;,&quot;S. &quot;,counts_streps_lactos_$genome)
# counts_streps_lactos_$genome &lt;- gsub(&quot;Lactobacillus &quot;,&quot;L. &quot;,counts_streps_lactos_$genome)
# counts_streps_lactos_$genome &lt;- gsub(&quot; subsp. &quot;,&quot;-&quot;,counts_streps_lactos_$genome)
# 
# all_sterm_strain_info_ready_01 &lt;- counts_streps_lactos_  %&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;genome&quot;)
# 
# p3 &lt;- gheatmap(tree_04, all_sterm_strain_info_ready_01, offset=0.04, width=.05,colnames_angle=95, colnames_offset_y = 0,hjust = -0.25)#+  scale_fill_manual(values=colors_area)
# p3
  
table(Species_distribution_preped$Species_name)

library(ggstance)
# library(ggtree)
library(ggtreeExtra)
tree_05 &lt;- tree_04+ geom_fruit(data=Species_distribution_preped, geom=geom_bar,
                    mapping=aes(y=Species_name, x=Standardized_MeanCount_per_sample,fill=Habitat_broad),
                    pwidth=0.09, position=position_stackx(),
                    # width=1,
                    offset=-0.07,
                    orientation=&quot;y&quot;, 
                    stat=&quot;identity&quot;,
         )+theme(legend.position=c(0.1, 0.2),legend.background=element_rect(fill = alpha(&quot;white&quot;, 0.1)),legend.title = element_blank(),legend.text = element_text(size = 7))

tree_05

    png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/LAB_pyhlo.png&quot;, width = 2200, height = 2200,res=300)
pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/LAB_pyhlo.pdf&quot;,width=10,height=10)

tree_05


dev.off()


###-------------------------
#add lac-operon
###-------------------------

presence_lacOPERON &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/lactose_utilziation/presence_lacOPERON.txt&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    col_names = c(&quot;species&quot;,&quot;lacZ&quot;,&quot;lacG&quot;), trim_ws = TRUE) 


presence_lacOPERON$lac_Operon &lt;- ifelse(presence_lacOPERON$lacZ&gt;0&amp;presence_lacOPERON$lacG&gt;0,&quot;both&quot;,ifelse(presence_lacOPERON$lacZ&gt;0,&quot;lacZ&quot;,ifelse(presence_lacOPERON$lacG&gt;0,&quot;lacG&quot;,&quot;NA&quot;)))
table(presence_lacOPERON$lac_Operon)


presence_lacOPERON$species &lt;- gsub(&quot;Streptococcus-&quot;,&quot;S. &quot;,presence_lacOPERON$species)
presence_lacOPERON$species &lt;- gsub(&quot;Lactobacillus-&quot;,&quot;L. &quot;,presence_lacOPERON$species)
presence_lacOPERON$species &lt;- gsub(&quot;L. delbrueckii-&quot;,&quot;L. del-&quot;,presence_lacOPERON$species)

unique(presence_lacOPERON$species)[!unique(presence_lacOPERON$species) %in% unique(Sterm_tree_raw_tible$label)]

    
library(ggnewscale)
     

     tree_06 &lt;- tree_05+ new_scale_fill() +geom_fruit(data=presence_lacOPERON, geom=geom_tile,
                    mapping=aes(y=species,  fill=lac_Operon),
                    # pwidth=0.09, 
                    width=0.02,
                    offset=0.04,
                    orientation=&quot;y&quot;, 
                    stat=&quot;identity&quot;,
         )+     scale_fill_manual(values=c(&quot;grey66&quot;,&quot;grey77&quot;,&quot;grey88&quot;,&quot;white&quot;))+geom_treescale()
tree_06

  # png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/LAB_pyhlo.png&quot;, width = 2200, height = 2200,res=300)
pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/LAB_pyhlo_02.pdf&quot;,width=10,height=10)

tree_06


dev.off()</code></pre>
</div>
<div id="final-make-ldel-sterm-and-lhelv-trees" class="section level3">
<h3>FINAL: make Ldel, Sterm and Lhelv trees</h3>
<pre class="r"><code>library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

# Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02//RAxML_bipartitions.lactos_curated&quot;, tree.names = NULL, force.multi = FALSE)

Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03//RAxML_bipartitions.lactos_curated&quot;, tree.names = NULL, force.multi = FALSE)


tree_01 &lt;- ggtree(Sterm_tree_raw) +
  geom_tiplab(size=0.5,align = FALSE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT
tree_01

library(castor)


##=======================================
##------Sterm
##=======================================


Sterm_tree_raw %&gt;% as_tibble() %&gt;% filter(!is.na(label)) %&gt;% filter(grepl(&quot;treptococcus&quot;,label)) %&gt;% pull(label) 

# 
tmp &lt;- as.phylo(Sterm_tree_raw)
# typeof(tree_01)

Sterm_tree_raw_sub &lt;- get_subtree_with_tips(tmp, 
                      only_tips               =c(&quot;Streptococcus-thermophilus_G000253395&quot;,&quot;Streptococcus-vestibularis_own&quot;,&quot;Streptococcus-salivarius_G000785515&quot;)) 

Sterm_tree_raw_sub_02 &lt;- Sterm_tree_raw_sub$subtree

Sterm_tree_raw_sub_02_Sterm &lt;- Sterm_tree_raw_sub_02 %&gt;% as_tibble() %&gt;% mutate(branch.length=branch.length*100000000) %&gt;% mutate(label=gsub(&quot;_.*&quot;,&quot;&quot;,label))  %&gt;% mutate(label=gsub(&quot;S.*-&quot;,&quot;S. &quot;,label)) %&gt;% as.treedata()





tree_01_sterm &lt;- ggtree(Sterm_tree_raw_sub_02_Sterm) +
  geom_tiplab(size=3,align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) +geom_treescale(x=0.1, y=3.5, offset=.1,width=250000)  + annotate(&#39;text&#39;, x=120000, y=3.25, label=&quot;years\nsince divergence&quot;)  + ggplot2::xlim(0, 1000000)#NO ROOT
tree_01_sterm



# tree_01 &lt;- ggtree(Sterm_tree_raw_sub_02) +
#   geom_tiplab(size=3,align = FALSE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0) + geom_text(aes(label=node)) #NO ROOT
# tree_01

# write.nexus(Sterm_tree_raw_sub_02,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03//RAxML_bipartitions.lactos_curated_sterm&quot;)
# write.tree(Sterm_tree_raw_sub_02,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03//RAxML_bipartitions.lactos_curated_sterm&quot;)
# 
# ##----------rotat
# tree_01 &lt;- ggtree(Sterm_tree_raw_sub_02) +
#   geom_tiplab(size=3,align = FALSE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0) + geom_text(aes(label=node))%&gt;% rotate(5)  #NO ROOT
# tree_01
# 
# library(gridExtra)
# library(tidytree)
# library(treeio)
# library(ape)
# flip(tree_01, &#39;Streptococcus-vestibularis_own&#39;, &#39;Streptococcus-thermophilus_G000253395&#39;)
# 
# 
# rotate(tree_01, c(&#39;Streptococcus-vestibularis_own&#39;, &#39;Streptococcus-thermophilus_G000253395&#39;))
# 
# y &lt;- ape::rotateConstr(tree_01, c(&#39;Streptococcus-salivarius_G000785515&#39;, &#39;Streptococcus-vestibularis_own&#39;, &#39;Streptococcus-thermophilus_G000253395&#39;))
# y
# 
# # &quot;Streptococcus-thermophilus_G000253395&quot;,&quot;Streptococcus-vestibularis_own&quot;,&quot;Streptococcus-salivarius_G000785515&quot;
# # %&gt;% rotate(33) 
# 
# gridExtra::grid.arrange(tree_01, flip(tree_01, 3, 2), ncol=2)
# 
# gridExtra::grid.arrange(Sterm_tree_raw_sub_02, flip(p, 5, 1), ncol=2)
# 
# ##----------reroot
# 
# Sterm_tree_raw_sub_03 &lt;- root(Sterm_tree_raw_sub_02, outgroup = &quot;Streptococcus-thermophilus_G000253395&quot;, edgelabel = TRUE)
# tree_01 &lt;- ggtree(Sterm_tree_raw_sub_03) +
#   geom_tiplab(size=3,align = FALSE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0)   + geom_text(aes(label=node))#NO ROOT
# tree_01
# 
# 
# ggtree(tr) + geom_text(aes(label=node))
# 
# ##----------reimport
# 
# 
# Sterm_tree_raw_sterm &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03//RAxML_bipartitions.lactos_curated_sterm_curated&quot;, tree.names = NULL, force.multi = FALSE)
# 
# tree_01 &lt;- ggtree(Sterm_tree_raw_sterm) +
#   geom_tiplab(size=3,align = TRUE) + # tiplabls aligned
#   geom_rootedge(rootedge = 0)  #NO ROOT
# tree_01


##=======================================
##------Lhelv
##=======================================

Sterm_tree_raw %&gt;% as_tibble() %&gt;% filter(!is.na(label)) %&gt;% filter(grepl(&quot;actobacillus&quot;,label)) %&gt;% pull(label) 
Sterm_tree_raw %&gt;% as_tibble() %&gt;% filter(!is.na(label)) %&gt;% filter(grepl(&quot;helveticus&quot;,label)) %&gt;% pull(label) 
Sterm_tree_raw %&gt;% as_tibble() %&gt;% filter(!is.na(label)) %&gt;% filter(grepl(&quot;gallinarum&quot;,label)) %&gt;% pull(label) 
Sterm_tree_raw %&gt;% as_tibble() %&gt;% filter(!is.na(label)) %&gt;% filter(grepl(&quot;crisp&quot;,label)) %&gt;% pull(label) 


Sterm_tree_raw_sub &lt;- get_subtree_with_tips(tmp, 
                      only_tips               =c(&quot;Lactobacillus-helveticus_G000422165&quot;,&quot;Lactobacillus-gallinarum_oldAnalysis&quot;,&quot;Lactobacillus-crispatus_G000091765&quot;)) 

Sterm_tree_raw_sub_Lhelv &lt;- Sterm_tree_raw_sub$subtree

Sterm_tree_raw_sub_Lhelv_02 &lt;- Sterm_tree_raw_sub_Lhelv %&gt;% as_tibble() %&gt;% mutate(branch.length=branch.length*100000000) %&gt;% mutate(label=gsub(&quot;_.*&quot;,&quot;&quot;,label))  %&gt;% mutate(label=gsub(&quot;L.*-&quot;,&quot;L. &quot;,label)) %&gt;% as.treedata()

tree_01_lhelv &lt;- ggtree(Sterm_tree_raw_sub_Lhelv_02) +
  geom_tiplab(size=3,align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) +geom_treescale(x=0.1, y=3, offset=.1,width=250000)  + annotate(&#39;text&#39;, x=120000, y=2.8, label=&quot;years\nsince divergence&quot;)  + ggplot2::xlim(0, 4000000)#NO ROOT
tree_01_lhelv

##=======================================
##------Ldel
##=======================================

Sterm_tree_raw %&gt;% as_tibble() %&gt;% filter(!is.na(label)) %&gt;% filter(grepl(&quot;delbrueckii&quot;,label)) %&gt;% pull(label) 
Sterm_tree_raw %&gt;% as_tibble() %&gt;% filter(!is.na(label)) %&gt;% filter(grepl(&quot;equicursoris&quot;,label)) %&gt;% pull(label)
# Sterm_tree_raw %&gt;% as_tibble() %&gt;% filter(!is.na(label)) %&gt;% filter(grepl(&quot;gallinarum&quot;,label)) %&gt;% pull(label) 
# Sterm_tree_raw %&gt;% as_tibble() %&gt;% filter(!is.na(label)) %&gt;% filter(grepl(&quot;crisp&quot;,label)) %&gt;% pull(label) 


Sterm_tree_raw_sub &lt;- get_subtree_with_tips(tmp, 
                      only_tips               =c(&quot;Lactobacillus-delbrueckii-bulgaricus_own&quot;,&quot;Lactobacillus-delbrueckii-indicus_own&quot;,&quot;Lactobacillus-delbrueckii-sunkii_own&quot;,&quot;Lactobacillus-delbrueckii-lactis_own&quot;,&quot;Lactobacillus-delbrueckii-jakobsenii_own&quot;,&quot;Lactobacillus-equicursoris_oldAnalysis&quot;)) 

Sterm_tree_raw_sub_Ldel &lt;- Sterm_tree_raw_sub$subtree

Sterm_tree_raw_sub_Ldel_02 &lt;- Sterm_tree_raw_sub_Ldel %&gt;% as_tibble() %&gt;% mutate(branch.length=branch.length*100000000)%&gt;% mutate(label=gsub(&quot;_.*&quot;,&quot;&quot;,label))  %&gt;% mutate(label=gsub(&quot;L.*-&quot;,&quot;L. &quot;,label)) %&gt;% as.treedata()

 

tree_01_ldel &lt;- ggtree(Sterm_tree_raw_sub_Ldel_02) +
  geom_tiplab(size=3,align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0) +geom_treescale(x=1, y=6, offset=.1,width=250000)  + annotate(&#39;text&#39;, x=120000, y=5.7, label=&quot;years\nsince divergence&quot;) + ggplot2::xlim(0, 6000000)#NO ROOT#NO ROOT
tree_01_ldel

###==================================
#strain age
###==================================


Sterm_tree_raw_species_focal &lt;-  Sterm_tree_raw %&gt;% as_tibble() %&gt;%  filter(grepl(&quot;Streptococcus-thermophilus|Lactobacillus-delbrueckii-lacti|Lactobacillus-helv&quot;,label)) %&gt;% mutate(age_of_species=branch.length*100000000)

mean(Sterm_tree_raw_species_focal$age_of_species)





###==================================
#bring trees together
###==================================

library(patchwork)
tree_01_sterm+tree_01_ldel+tree_01_lhelv


pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Zoom_species_rrna_phylo_01.pdf&quot;,width=9,height=3)

tree_01_sterm+tree_01_ldel+tree_01_lhelv


dev.off()</code></pre>
</div>
</div>
<div id="earthmicrobiome-data" class="section level2">
<h2>Earthmicrobiome data</h2>
<pre class="r"><code> Species_distribution_preped &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/Earthmicrobiome_data/Final/Final/VinceDB_Species_distribution_prepeed.csv&quot;)


Species_distribution_preped_sub &lt;- Species_distribution_preped %&gt;% filter(Species_name %in% c(&quot;L. helveticus&quot;,&quot;S. thermophilus&quot;,&quot;L. del-lactis&quot;,&quot;S. vestibularis&quot;,&quot;L. crispatus&quot;,&quot;L. equicursoris&quot;))

Species_distribution_preped_sub %&gt;% filter(Habitat_broad==&quot;Animal sampling&quot;)

pseudo_sub &lt;- pseudogeness_sumary %&gt;% filter(species_new %in% c(&quot;helveticus&quot;,&quot;S. thermophilus&quot;,&quot;L. delbrueckii subsp. lactis&quot;,&quot;vestibularis&quot;,&quot;crispatus&quot;,&quot;equicursoris&quot;))</code></pre>
</div>
<div id="data-of-tree" class="section level2">
<h2>data of tree</h2>
<pre class="r"><code>library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(philr)
# Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02//RAxML_bipartitions.lactos_curated&quot;, tree.names = NULL, force.multi = FALSE)


library(nodiv)
Sterm_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_03//RAxML_bipartitions.lactos_curated&quot;, tree.names = NULL, force.multi = FALSE)


##-----------------------------------------
#make phylo
##-----------------------------------------

Sterm_tree_raw_phylo &lt;- Sterm_tree_raw %&gt;% as.phylo() 

distance &lt;- mean_dist_to_tips(Sterm_tree_raw_phylo) %&gt;% as.data.frame()  %&gt;% as_tibble() %&gt;% mutate(branchAge_calculated_02 =.*5000) %&gt;% select(-.) #%&gt;% left_join(.,info)

distance$node &lt;- nodenumbers(Sterm_tree_raw_phylo)

range(distance$branchAge_calculated_02)

# write.csv(distance,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_referenceTree_SpeciesTREE_large_02/age_LAB_phylo.csv&quot;)

##-----------------------------------------
#identify the age of the clades
##-----------------------------------------

Sterm_tree_raw_phylo_tips &lt;- Sterm_tree_raw_phylo$tip.label %&gt;% as.data.frame()  %&gt;% as_tibble() %&gt;% mutate(genus=gsub(&quot;-.*&quot;,&quot;&quot;,.))%&gt;% mutate(genus=gsub(&quot;&#39;&quot;,&quot;&quot;,genus))
table(Sterm_tree_raw_phylo_tips$genus) %&gt;% length()

colnames(Sterm_tree_raw_phylo_tips)[1] &lt;- &quot;tip.name&quot;
age_genus_LAB &lt;- data.frame()
for (genusss in unique(Sterm_tree_raw_phylo_tips$genus)) {
    tmp &lt;- Sterm_tree_raw_phylo_tips %&gt;% filter(genus==genusss)

  if (nrow(tmp)&gt;1) {
      tmp1 &lt;- getMRCA(Sterm_tree_raw_phylo,tmp$tip.name)
     tmp2 &lt;-  data.frame(genus=genusss,node=tmp1)
  
     
     age_genus_LAB &lt;- rbind(age_genus_LAB,tmp2)
  }

}

age_genus_LAB_extension &lt;- age_genus_LAB %&gt;% left_join(.,distance) %&gt;% as_tibble()

age_genus_LAB_extension_prep &lt;- age_genus_LAB_extension %&gt;% mutate(taxon=genus)%&gt;% mutate(group=&quot;Genus&quot;) %&gt;% select(taxon,group,branchAge_calculated_02)%&gt;% mutate(speciess=&quot;rest&quot;)

age_genus_LAB_extension_prep_02 &lt;- age_genus_LAB_extension_prep %&gt;% filter(taxon %in% c(&quot;Streptococcus&quot;,&quot;Lactobacillus&quot;))%&gt;% mutate(speciess=taxon)



##-----------------------------------------
#identify the age of the clades species
##-----------------------------------------



Sterm_tree_raw_species &lt;- Sterm_tree_raw %&gt;% as_tibble() %&gt;% mutate(Stem_to_species=parent)%&gt;% mutate(node=parent) %&gt;% select(Stem_to_species,label,node) %&gt;% filter(!is.na(label))%&gt;% left_join(.,distance) %&gt;% as_tibble() %&gt;% select(-node)


Sterm_tree_raw_species_focal_prep &lt;- Sterm_tree_raw_species %&gt;% filter(grepl(&#39;Streptococcus|Lactobacillus&#39;, label)) %&gt;% mutate(group=&quot;Species&quot;) %&gt;% mutate(taxon=label)%&gt;% select(taxon,group,branchAge_calculated_02) %&gt;% mutate(speciess=&quot;rest&quot;)

Sterm_tree_raw_species_focal_prep_02 &lt;- Sterm_tree_raw_species_focal_prep %&gt;% filter(grepl(&quot;Streptococcus-thermophilus|Lactobacillus-delbrueckii-lacti|Lactobacillus-helv&quot;,taxon)) %&gt;% mutate(speciess=gsub(&quot;_.*&quot;,&quot;&quot;,taxon))

mean(Sterm_tree_raw_species_focal_prep_02$branchAge_calculated_02)


total_age &lt;- rbind(age_genus_LAB_extension_prep,age_genus_LAB_extension_prep_02,Sterm_tree_raw_species_focal_prep,Sterm_tree_raw_species_focal_prep_02)


ggplot(total_age,aes(x=speciess,y=branchAge_calculated_02,color=speciess,fill=speciess,shape=group))+geom_boxplot()+theme_classic()+facet_wrap(~group,scales = &quot;free_x&quot;)


total_age_summed &lt;- total_age %&gt;% group_by(speciess,group) %&gt;% summarise(Age=mean(branchAge_calculated_02),variation=sd(branchAge_calculated_02))


ggplot(total_age_summed, aes(speciess, Age,fill=speciess,color=speciess)) +
  geom_point() + 
  geom_pointrange(aes(ymin = Age-variation, ymax = Age+variation),data = total_age_summed)+theme_classic()+facet_wrap(~group,scales = &quot;free_y&quot;,nrow = 2)+
  coord_flip() +theme_classic()+coord_trans(y=&quot;log2&quot;)


total_age_summed$speciess = factor( total_age_summed$speciess, levels=c(&quot;rest&quot;,  &quot;Streptococcus&quot;, &quot;Streptococcus-thermophilus&quot;, &quot;Lactobacillus&quot; , &quot;Lactobacillus-delbrueckii-lactis&quot;,&quot;Lactobacillus-helveticus&quot;))


colorsss &lt;- rev(c(&quot;#007fa6ff&quot;,&quot;#fdc2b5ff&quot;,&quot;#b3b3b3ff&quot;,))

 colorsss &lt;- rev(c(&quot;#b3b3b3ff&quot;,&quot;#007fa6ff&quot;,&quot;#007fa6ff&quot;,&quot;#fdc2b5ff&quot;,&quot;#fdc2b5ff&quot;,&quot;grey88&quot;))


plot_AGE &lt;- ggplot(total_age_summed, aes(y=speciess, x=Age,fill=speciess,color=speciess)) +
  geom_point(position=position_dodge(width=7.0)) + 
  geom_pointrange(position=position_dodge(width=7.0),aes(xmin = Age-variation, xmax = Age+variation),data = total_age_summed)+theme_classic()+facet_wrap(~group,scales = &quot;free_y&quot;,nrow = 2, strip.position = &quot;right&quot;)+
  theme_classic()+scale_x_continuous(trans=&quot;log2&quot;,breaks =c(10,50,100,500,1000))+scale_fill_manual(values = colorsss)+scale_color_manual(values = colorsss)+
  theme(axis.text.y = element_blank(),axis.title.y = element_blank(),axis.ticks.y = element_blank(),legend.position = &quot;none&quot;,strip.text = element_text(size = 9))+labs(x=&quot;Branch age (mio. years)&quot;)+geom_vline(xintercept = 250,color=&quot;grey&quot;)


plot_AGE

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/_Sterm_age_plot.pdf&quot;,width=3,height=3)

plot_AGE


dev.off()




cols_1 &lt;- rainbow(14) 
cols_2 &lt;- rainbow(16) 


branch_age_plot &lt;- ggplot() +
  geom_pointrange(aes(x=grouping_extended, y=branchAge_calculated, ymin = branchAge_calculated-branchAge_variation, ymax = branchAge_calculated+branchAge_variation,fill=grouping_extended,color=grouping_extended),data = myTree_all_beast_tibble_extended_all_02)+theme_classic()+facet_wrap(group~.,nrow = 3,scales = &quot;free_y&quot;, strip.position = &quot;right&quot;)+
  coord_flip()+scale_fill_manual(values = colorsss)+scale_color_manual(values = colorsss)+
  theme(axis.text.y = element_blank(),axis.title.y = element_blank(),axis.ticks.y = element_blank(),legend.position = &quot;none&quot;,strip.text = element_text(size = 7))+labs(y=&quot;Branch age&quot;)

branch_age_plot



# coord_flip() +
# facet_wrap(~group,nrow = 2)+

# getMRCA(tree_phylo, c(&quot;rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24736&quot;,&quot;rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24739&quot;,&quot;rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24743&quot;))
# is.na(Sterm_tree_raw_phylo_tips$genus) %&gt;% sum()</code></pre>
<p>###PGAP</p>
<div id="pseudogene-origin" class="section level6">
<h6>pseudogene origin</h6>
<pre class="bash"><code>
##-------------------------------------------------
#testing
##-------------------------------------------------

### pgap gbff 2 gff

gbff2gff.sh

genomesss_short=GCF_900322585.1_Lactobacillus_delbrueckii_subsp._lactis_genomic

mkdir -p /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/

zgrep &quot;/pseudo&quot; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF/${genomesss_short}.gbff.gz |wc -l

#gbff2gff.sh /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF/${genomesss_short}.gbff.gz /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff
rm /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff

/home/vincent/anaconda3/envs/PROKKA/bin/bp_genbank2gff3.pl  --quiet  /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GBF/${genomesss_short}.gbff.gz -out stdout &gt; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff



#of interes
pseudogenic_exon

grep &quot;^NZ_LS99&quot; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff |cut -f 3 |sort|uniq -c
awk &#39;{if($3==&quot;pseudogene&quot; || $3==&quot;pseudogenic_exon&quot;) print $0}&#39; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff

awk &#39;{if($3==&quot;CDS&quot; || $3==&quot;pseudogenic_exon&quot;) print $0}&#39; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff |grep &quot;pseudogenic_exon&quot; -A 2 -B 2

# /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff |head -20

grep &quot;transpos&quot; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/GFF/${genomesss_short}.gff |head -20





##---------------------------------------------------
#only RMK strains
##---------------------------------------------------


##--------------------
#make  gff file
##--------------------

for species in $(echo &quot;Ldel Sterm&quot;)
do
echo &quot;===============================&quot;
echo ${species}
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}

for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/ |grep &quot;.gbk&quot; |sed &#39;s/.gbk//g&#39;)
do
echo ${genomessss}


/home/vincent/anaconda3/envs/PROKKA/bin/bp_genbank2gff3.pl  --quiet  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/${genomessss}.gbk -out stdout &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}/${genomessss}.gff


done #${genomessss}


done #species




##--------------------
#identify pseudo gene origin
##--------------------

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log/pseudo_origin.txt


#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/Sterm/24730.gbk
#/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF

for species in $(echo &quot;Ldel Sterm&quot;)
do


for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GBK/${species}/ |grep &quot;.gbk&quot; |sed &#39;s/.gbk//g&#39;)
do

#awk &#39;{if($3==&quot;CDS&quot; || $3==&quot;pseudogenic_exon&quot;) print $0}&#39; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/${species}/genomes/GFF/${genomesss_short}.gff |grep &quot;pseudogenic_exon&quot; -A 2 -B 2 |

for pseudognesss in $(awk &#39;{if($3==&quot;pseudogenic_exon&quot;) print $9}&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}/${genomessss}.gff |cut -d &#39;;&#39; -f 1 )
do

#echo $pseudognesss

reason=$(awk &#39;{if($3==&quot;CDS&quot; || $3==&quot;pseudogenic_exon&quot;) print $0}&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}/${genomessss}.gff |grep &quot;${pseudognesss};&quot;  |sed &#39;s/.*;Note=//g&#39; |sed &#39;s/%.*//g&#39;)

count1=$(awk &#39;{if($3==&quot;CDS&quot; || $3==&quot;pseudogenic_exon&quot;) print $0}&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}/${genomessss}.gff |grep &quot;${pseudognesss};&quot; -A 2 -B 2 |grep &quot;transpos&quot; -i |wc -l)
count2=$(awk &#39;{if($3==&quot;CDS&quot; || $3==&quot;pseudogenic_exon&quot;) print $0}&#39; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/GFF_created/${species}/${genomessss}.gff |grep &quot;${pseudognesss};&quot; -A 2 -B 2 |grep &quot;=IS&quot; -i  |wc -l)
countFinal=$((count1+count2))
echo -e ${species}&quot;\t&quot;${genomessss}&quot;\t&quot;${pseudognesss}&quot;\t&quot;${reason}&quot;\t&quot;${countFinal} 
echo -e ${species}&quot;\t&quot;${genomessss}&quot;\t&quot;${pseudognesss}&quot;\t&quot;${reason}&quot;\t&quot;${countFinal} &gt;&gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log/pseudo_origin.txt

done #pseudognesss


done #genomessss

done #species

##---------------------------------------
#move local
##---------------------------------------


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/log/pseudo_origin.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/
</code></pre>
<p>plotting</p>
<pre class="r"><code>library(tidyverse)
pseudo_origin &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/pseudo_origin.txt&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    col_names = c(&quot;species&quot;,&quot;genome&quot;,&quot;gene&quot;,&quot;reason&quot;,&quot;transoposon&quot;), trim_ws = TRUE, col_types = cols(genome = col_character())) %&gt;% mutate(pseudoTrans=ifelse(transoposon==0,&quot;NO&quot;,&quot;YES&quot;))

pseudo_origin$species  &lt;- plyr::revalue(pseudo_origin$species,c(&quot;Ldel&quot;=&quot;L.delbrueckii&quot;,&quot;Sterm&quot;=&quot;S.thermophilus&quot;))



# table(pseudo_origin$pseudoTrans)

pseudogene_origin_summary &lt;- pseudo_origin %&gt;% group_by(species,genome,reason,pseudoTrans) %&gt;% summarise(countss=n()) %&gt;% filter(reason %in% c(&quot;frameshifted&quot;,&quot;incomplete&quot;,&quot;internal stop&quot;)) %&gt;% group_by(species,reason,pseudoTrans) %&gt;% summarise(means=mean(countss),medians=median(countss),sdss=sd(countss))



ggplot(pseudogene_origin_summary,aes(x=reason,y=medians,color=pseudoTrans,fill=pseudoTrans))+geom_col()+theme_classic()+facet_wrap(~species)
ggplot(pseudogene_origin_summary,aes(x=reason,y=medians,fill=pseudoTrans))+geom_bar(stat = &quot;identity&quot;)+theme_classic()+facet_wrap(~species)+geom_errorbar(aes(ymin=medians-sdss, ymax=medians+sdss), width=.2) 



##with errorbars

pseudogene_origin_summary %&gt;% spread(., reason, medians) %&gt;% mutate(y=error=ifelse(pseudoTrans==&quot;YES&quot;,medians,))

###Overall


pseudogene_origin_summary_wo &lt;- pseudo_origin %&gt;% group_by(species,genome,reason) %&gt;% summarise(countss=n()) %&gt;% filter(reason %in% c(&quot;frameshifted&quot;,&quot;incomplete&quot;,&quot;internal stop&quot;)) %&gt;% group_by(species,reason) %&gt;% summarise(means=mean(countss),medians=median(countss),sdss=sd(countss))

# ggplot(pseudogene_origin_summary_wo,aes(x=reason,y=medians))+geom_bar(stat = &quot;identity&quot;)+theme_classic()+facet_wrap(~species)+geom_errorbar(aes(ymin=medians-sdss, ymax=medians+sdss), width=.2) 
plotOverall &lt;- ggplot()+geom_bar(data=pseudogene_origin_summary_wo,aes(x=reason,y=medians),stat = &quot;identity&quot;,fill=&quot;#254E70&quot;)+theme_classic()+facet_wrap(~species)+geom_errorbar(data=pseudogene_origin_summary_wo,aes(x=reason,y=medians,ymin=medians-sdss, ymax=medians+sdss), width=.2) +labs(x=&quot;&quot;,y=&quot;Pseudogenes&quot;)
plotOverall



  png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/plot_pseudoCount.png&quot;, width = 2200, height = 1500,res=300)
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)

plotOverall


dev.off()

###transposons


pseudogene_origin_summary_transp &lt;- pseudogene_origin_summary %&gt;% filter(pseudoTrans==&quot;YES&quot;)

plottranpso &lt;-ggplot()+geom_bar(data=pseudogene_origin_summary_wo,aes(x=reason,y=medians),stat = &quot;identity&quot;,fill=&quot;#254E70&quot;)+theme_classic()+facet_wrap(~species)+geom_errorbar(data=pseudogene_origin_summary_wo,aes(x=reason,y=medians,ymin=medians-sdss, ymax=medians+sdss), width=.2) +
      geom_bar(data=pseudogene_origin_summary_transp,aes(x=reason,y=medians),stat = &quot;identity&quot;,fill=&quot;#C33C54&quot;)+geom_errorbar(data=pseudogene_origin_summary_transp,aes(x=reason,y=medians,ymin=medians-sdss, ymax=medians+sdss), width=.2)+labs(x=&quot;&quot;,y=&quot;Pseudogenes&quot;)
plottranpso

  png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement/plot_pseudoCount_colored.png&quot;, width = 2200, height = 1500,res=300)
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)

plottranpso

dev.off()

#---------------------------------counts

pseudo_origin_strain &lt;- pseudo_origin %&gt;% group_by(species,genome,pseudoTrans) %&gt;% summarise(counts=n()) %&gt;% spread(., pseudoTrans, counts) %&gt;% mutate(total=NO+YES) %&gt;% mutate(percent=100*(YES/total))

pseudo_origin_strain %&gt;% group_by(species) %&gt;% summarise(median=median(percent),sdss=sd(percent))
pseudo_origin_strain %&gt;% mutate(all=&quot;all&quot;)%&gt;% group_by(all) %&gt;% summarise(median=median(percent),sdss=sd(percent))</code></pre>
</div>
<div id="prokka" class="section level3">
<h3>Prokka</h3>
<p>Here, I prokka annotate all genomes</p>
<pre class="bash"><code>##---------------------------------------
#merge NCBI and own genomes
##---------------------------------------

##Sterm

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Sterm/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/Streptococcus_thermophilus-*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Sterm/


cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm/01_all_FNA/*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Sterm/

##Lhelv

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Lhelv/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/Lactobacillus_helveticus-*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Lhelv/

##Ldel

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Ldel/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/Lactobacillus_delbrueckii_*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Ldel/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel/01_all_FNA/*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/Ldel/


##---------------------------------------
#run Prokka
##---------------------------------------
conda activate PROKKA


for speciess in $(echo &quot;Ldel Sterm Lhelv&quot;)
do

for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/${speciess}/ |grep &quot;.fna&quot; |sed &#39;s/.fna//g&#39;)
do
echo &quot;===========================================&quot;
echo ${genomes}
rm -r  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/

prokka --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/ --force --proteins proteins.faa --evalue 0.001 --addgenes   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/${speciess}/${genomes}.fna --cpus 5

done
done

#===========================================================================
#ALL SPECIES
#===========================================================================



##------------------------------------------------------------------
#PREP
##------------------------------------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel/01_all_FNA|sed &#39;s/.fna//g&#39;)
do

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Ldel/01_all_FNA/${genomesss}.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/Lactobacillus_delbrueckii_subsp_lactis-${genomesss}.fna

done

##----------Sterm

for genomesss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm/01_all_FNA|sed &#39;s/.fna//g&#39;)
do

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/Sterm/01_all_FNA/${genomesss}.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/Streptococcus_thermophilus-${genomesss}.fna

done

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/FNA/*.fna /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/

ls  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/ |wc -l

##------------------------------------------------------------------
#PROKKA
##------------------------------------------------------------------

conda activate PROKKA


for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all// |grep &quot;.fna&quot; |sed &#39;s/.fna//g&#39;)
do
echo &quot;===========================================&quot;
echo ${genomes}

rm -r  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${genomes}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/${genomes}/

prokka --outdir //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/${genomes}/ --force --proteins proteins.faa --evalue 0.001 --addgenes   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all/${genomes}.fna --cpus 25

done


##------------------------------------------------------------------
##create FAA and FNA folders
##------------------------------------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all*
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/{01_all_FAA,01_all_FNA,01_all_GFF,01_all_FFN}

for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA_all// |grep &quot;.fna&quot; |sed &#39;s/.fna//g&#39;)
do

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/${genomes}//PROKKA*.fna &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FNA/${genomes}.fna

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/${genomes}//PROKKA*.ffn &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FFN/${genomes}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/${genomes}//PROKKA*.faa &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FAA/${genomes}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/${genomes}//PROKKA*.gff &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_GFF/${genomes}.gff

done #genomes
</code></pre>
<div id="transposase-diversity" class="section level6">
<h6>transposase diversity</h6>
<pre class="bash"><code>

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_analysis.txt
rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_transposases_analysis.txt
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats
for speciesss in $(echo &quot;Sterm Ldel&quot;)
do
for id in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}/01_all_GFF/ |sed &#39;s/.gff//g&#39;)
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
 
#   contiNumber=$(grep &quot;&gt;&quot; -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/2*/*19.gff) 
#   GenomeLength=$(grep -v &quot;^&gt;&quot; /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/2*/*19.gff  |wc -c)
  
  gensss=$(grep &quot;gene: &quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}//${id}/PROKKA_*.txt |sed &#39;s/gene: //g&#39;)
  tRNAsss=$(grep &quot;tRNA: &quot;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}//${id}/PROKKA_*.txt |sed &#39;s/tRNA: //g&#39;)
  rRNAsss=$(grep &quot;rRNA: &quot;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}//${id}/PROKKA_*.txt |sed &#39;s/rRNA: //g&#39;)
  
  
  
  transpononssss=$(grep &quot;ISfinder:&quot; -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}//${id}/PROKKA_*.gff |wc -l)

grep &quot;ISfinder:&quot; -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${speciesss}//${id}/PROKKA_*.gff |grep &quot;;product=&quot; |awk -F &quot;;product=&quot; &#39;{OFS=&quot;\t&quot;}{print $2}&#39;|awk -F &quot; family transposase &quot; -v namzzz=&quot;$id&quot; &#39;{OFS=&quot;\t&quot;}{print namzzz,$1,$2}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_transposases_analysis.txt

 echo -e ${id}&quot;\t&quot;${gensss}&quot;\t&quot;${tRNAsss}&quot;\t&quot;${rRNAsss}&quot;\t&quot;${transpononssss}  &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_analysis.txt

    done
done #species


##---------------------------------------
#move local
##---------------------------------------


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/stats//all_annotation_transposases_analysis.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/
</code></pre>
<pre class="r"><code>library(tidyverse)
all_annotation_transposases_analysis &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/all_annotation_transposases_analysis.txt&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    col_names = c(&quot;genome&quot;,&quot;Isfamily&quot;,&quot;ISsubfamily&quot;), trim_ws = TRUE) %&gt;% mutate(species=substr(genome, 1, 1))

all_annotation_transposases_analysis$species  &lt;- plyr::revalue(all_annotation_transposases_analysis$species,c(&quot;L&quot;=&quot;L.delbrueckii&quot;,&quot;S&quot;=&quot;S.thermophilus&quot;))


table(all_annotation_transposases_analysis$species)


all_annotation_transposases_analysis_summary &lt;- all_annotation_transposases_analysis %&gt;% group_by(species,genome,Isfamily) %&gt;% summarise(countss=n()) %&gt;% group_by(species,Isfamily) %&gt;% summarise(medians=median(countss),means=mean(countss),sdss=sd(countss))%&gt;%replace(is.na(.), 0) 


plottranpso &lt;- ggplot()+geom_bar(data=all_annotation_transposases_analysis_summary,aes(x=Isfamily,y=means),stat = &quot;identity&quot;,fill=&quot;#254E70&quot;)+theme_classic()+facet_wrap(~species, scales=&quot;free_x&quot;)+geom_errorbar(data=all_annotation_transposases_analysis_summary,aes(x=Isfamily,y=means,ymin=means-sdss, ymax=means+sdss), width=.2) +labs(x=&quot;Insertation Sequence Families&quot;,y=&quot;Transposons&quot;)+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),legend.position = &quot;none&quot;)
plottranpso


  png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/20210126/supplement/plot_transposons_families.png&quot;, width = 3000, height = 1500,res=300)
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)

plottranpso

dev.off()

all_annotation_transposases_analysis_summary$Isfamily %&gt;% unique() %&gt;% length()</code></pre>
</div>
</div>
<div id="pandaroo" class="section level3">
<h3>pandaroo</h3>
<ol start="2" style="list-style-type: decimal">
<li>find core genomes</li>
</ol>
<pre class="bash"><code>##==========================================
###-----------------bring all gffs together
##==========================================
species=Sterm

for speciess in $(echo &quot;Ldel Sterm Lhelv&quot;)
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_GFFs/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_GFFs/
echo $speciess

for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/${speciess}/ |grep &quot;.fna&quot; |sed &#39;s/.fna//g&#39;)
do
echo $genomes

#leterzzz=$(echo $genomes |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/*.gff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_GFFs/${genomes}.gff

done
ll  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_GFFs/
done

##==========================================
###-----------------panaroo
##==========================================

##-----sterm

species=Sterm
conda deactivate
conda activate panaroo

#for speciess in $(echo &quot;Ldel Sterm Lhelv&quot;)
for speciess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/|grep &quot;^01&quot; -v |sed &quot;s/-.*$//g&quot; |sort|uniq)

do
echo ${speciess}
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/ |grep &quot;${speciess}&quot; -c

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/

panaroo -t 20 -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_GFF/${speciess}-*.gff -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/ --clean-mode strict

done


##==========================================
###-----------------panaroo all species 
##==========================================

##-----sterm

species=Sterm
conda deactivate

conda activate panaroo

for speciess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/ |sed &quot;s/-.*$//g&quot; |sort|uniq)
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/

panaroo -t 25 -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_GFFs/*.gff -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/ --clean-mode strict

done



####-------------------------------------------------
##identify transposase genes
####-------------------------------------------------
</code></pre>
</div>
<div id="pan-core-ratio" class="section level3">
<h3>pan-core ratio</h3>
<pre class="bash"><code>##==========================================
###-----------------calculate pan-core ratio
##==========================================


rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/All_pan_core.txt
#for speciess in $(echo &quot;Ldel Sterm Lhelv&quot;)
for speciess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/|grep &quot;^01&quot; -v |sed &quot;s/-.*$//g&quot; |sort|uniq)

do
echo ${speciess}
countsss=$(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/ |grep &quot;${speciess}&quot; -c)


# cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/summary_statistics.txt
 
pan_count=$(grep &quot;Total genes&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/summary_statistics.txt |awk -F &quot;%)&quot; &#39;{print $2}&#39;| sed &#39;s/^ *//g&#39;)
 
core_count=$(grep &quot;Core genes&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/summary_statistics.txt |awk -F &quot;%)&quot; &#39;{print $2}&#39;| sed &#39;s/^ *//g&#39;)
 
echo -e ${speciess}&quot;\t&quot;${pan_count}&quot;\t&quot;${core_count}&quot;\t&quot;${countsss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/All_pan_core.txt
done

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/All_pan_core.txt

##----------------------------------------------
#bring local
##----------------------------------------------

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/All_pan_core.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/
</code></pre>
</div>
<div id="species-infos" class="section level3">
<h3>species infos</h3>
<pre class="r"><code>library(tidyverse)
All_pan_core &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/All_pan_core.txt&quot;, delim = &quot;\t&quot;, escape_double = FALSE,  col_names = c(&quot;species&quot;,&quot;pan&quot;,&quot;core&quot;,&quot;genomes&quot;), trim_ws = TRUE) %&gt;% mutate(pan_core_ratio=pan/core)
All_pan_core$catagory &lt;- ifelse(All_pan_core$species %in% c(&quot;Lactobacillus_delbrueckii_subsp_lactis&quot;,&quot;Lactobacillus_helveticus&quot;,&quot;Streptococcus_thermophilus&quot;),&quot;focal&quot;,&quot;not&quot;)

# All_pan_core$species  &lt;- plyr::revalueAll_pan_core$species 1,c(&quot;Ldel&quot;=&quot;L.delbrueckii&quot;,&quot;Sterm&quot;=&quot;S.thermophilus&quot;))
# 

All_pan_core$species

ggplot(All_pan_core,aes(x=species,y=core,fill=catagory))+geom_bar(stat = &quot;identity&quot;)+theme_classic()+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),legend.position = &quot;none&quot;)


ggplot(All_pan_core,aes(x=species,y=pan_core_ratio,fill=catagory))+geom_bar(stat = &quot;identity&quot;)+theme_classic()+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),legend.position = &quot;none&quot;)


plot_pan_core &lt;- ggplot(All_pan_core,aes(x=core,y=pan_core_ratio,fill=catagory,color=catagory,size=genomes))+geom_point(stat = &quot;identity&quot;)+theme_classic()+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),legend.position = &quot;none&quot;)+labs(x=&quot;core genome size&quot;,y=&quot;pan-core Ratio&quot;)
  
plot_pan_core
png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/plot_pan_core.png&quot;, width = 2200, height = 1500,res=300)
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)

plot_pan_core


dev.off()


##with genome size


pseudogeness_count &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/pseudogeness_count.txt&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;species&quot;,&quot;genome&quot;,&quot;genes_coding&quot;,&quot;pseudogenes&quot;), trim_ws = TRUE) %&gt;% filter(pseudogenes&lt;1000) %&gt;% filter(genes_coding&lt;4000)%&gt;% mutate(pseudogenes_percent = 100*(pseudogenes/genes_coding)) %&gt;% filter(species!=&quot;Wigg&quot;)%&gt;% filter(species!=&quot;Scap&quot;)


pseudogeness_sumary &lt;- pseudogeness_count %&gt;% group_by(species) %&gt;%  dplyr::summarize(meanPseudo=100*(mean(pseudogenes)/mean(genes_coding)),meangenes=mean(genes_coding),stdevPseudo=100*(sd(pseudogenes)/mean(genes_coding)),stdevgene=sd(genes_coding))</code></pre>
</div>
<div id="identify-duplicated-genes" class="section level3">
<h3>identify duplicated genes</h3>
<pre class="bash"><code>speciess=Sterm
cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/
/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv

/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Ldel/gene_presence_absence_roary.csv
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Sterm/gene_presence_absence_roary.csv
/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Lhelv/gene_presence_absence_roary.csv


mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Ldel/gene_presence_absence_roary.csv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Ldel_gene_presence_absence_roary.csv

scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Sterm/gene_presence_absence_roary.csv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Sterm_gene_presence_absence_roary.csv

scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Lhelv/gene_presence_absence_roary.csv /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Lhelv_gene_presence_absence_roary.csv</code></pre>
<pre class="r"><code>library(tidyverse)

##===========================================================
#sterm
#maybe include outgroup!
##===========================================================
speciess &lt;- &quot;Sterm&quot;
Sterm_pandaroo &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/Sterm_gene_presence_absence_roary.csv&quot;)


##-------------------script-------------

Sterm_pandaroo_short &lt;- Sterm_pandaroo[,c(1,15:ncol(Sterm_pandaroo))]

Sterm_pandaroo_long &lt;- gather(Sterm_pandaroo_short, genome, gene_name, colnames(Sterm_pandaroo_short)[2:ncol(Sterm_pandaroo_short)], factor_key=TRUE,na.rm = TRUE)  %&gt;% mutate(gene_name = strsplit(as.character(gene_name), &quot;;&quot;)) %&gt;% unnest(gene_name)


Sterm_pandaroo_count &lt;- Sterm_pandaroo_long %&gt;% group_by(Gene,genome) %&gt;% summarise(gene_count=n())
Sterm_pandaroo_max_gene &lt;- Sterm_pandaroo_count %&gt;% group_by(Gene) %&gt;% summarise(max_gene=max(gene_count))


duplicated_gene_count &lt;-Sterm_pandaroo_max_gene %&gt;% filter(max_gene&gt;1) %&gt;% nrow()
total_gene_count &lt;- Sterm_pandaroo_max_gene$Gene %&gt;% unique %&gt;% length()

##-------------------duplication genes-------------

table(Sterm_pandaroo_max_gene$max_gene)
print(paste0(&quot;There are &quot;,total_gene_count,&quot; in &quot;,speciess,&quot; of which &quot;,duplicated_gene_count,&quot; are duplicated&quot;))

##-------------------duplication distribution-------------

duplicated_count &lt;- Sterm_pandaroo_max_gene %&gt;% filter(max_gene&gt;1)

Sterm_pandaroo_count_duplic &lt;- Sterm_pandaroo_count %&gt;% filter(Gene %in% unique(duplicated_count$Gene))
genome_count &lt;- unique(Sterm_pandaroo_long$genome) %&gt;% length()

Sterm_pandaroo_count_duplic$duplicate &lt;- ifelse(Sterm_pandaroo_count_duplic$gene_count&gt;1,&quot;duplicate&quot;,&quot;single&quot;)

Sterm_pandaroo_presence &lt;- Sterm_pandaroo_count_duplic %&gt;% count(Gene, duplicate) %&gt;% mutate(percent_n=100*(n/genome_count)) %&gt;%  select(-n)%&gt;% spread(.,duplicate,percent_n) %&gt;% mutate(missing=100-duplicate-single)

Sterm_pandaroo_presence_long &lt;- gather(Sterm_pandaroo_presence, grouping, percent_group, colnames(Sterm_pandaroo_presence)[2:4], factor_key=TRUE,na.rm = TRUE) 
ggplot(Sterm_pandaroo_presence_long,aes(x=percent_group,fill=grouping))+theme_classic()+geom_histogram()+facet_wrap(~grouping)

##-------------------last common ancestor of all genomes that share a duplicated gene. 

library(ape)
library(ggtree)
library(tidytree)
library(treeio)
library(TreeTools)

Sterm_species_tree &lt;- read.nexus(&quot;/home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Sterm/SpeciesTree_rooted_node_labels_curated.txt&quot;, tree.names = NULL, force.multi = FALSE)

gtree_01 &lt;- ggtree(Sterm_species_tree) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  


##-------------------number of ds between duplicate genes within a genome. 
#identify the genes to align
#afterwards use 
#library(PopGenome)

Sterm_pandaroo_count_duplic_sub &lt;- Sterm_pandaroo_count_duplic %&gt;% filter(duplicate==&quot;duplicate&quot;)


Sterm_pandaroo_short &lt;- Sterm_pandaroo[,c(1,15:ncol(Sterm_pandaroo))]
Sterm_pandaroo_short$uniqNumber &lt;- 1:nrow(Sterm_pandaroo_short)


Sterm_pandaroo_long &lt;- gather(Sterm_pandaroo_short, genome, gene_name, colnames(Sterm_pandaroo_short)[2:(ncol(Sterm_pandaroo_short)-1)], factor_key=TRUE,na.rm = TRUE)  %&gt;% mutate(gene_name = strsplit(as.character(gene_name), &quot;;&quot;)) %&gt;% unnest(gene_name)

extended_gene_dup &lt;- merge(Sterm_pandaroo_long,Sterm_pandaroo_count_duplic_sub,by=c(&quot;Gene&quot;,&quot;genome&quot;)) %&gt;% mutate(uniqName=paste0(genome,&quot;.&quot;,uniqNumber))

write.table(extended_gene_dup,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/extended_gene_dup.csv&quot;,sep=&quot;\t&quot;,quote = FALSE,row.names = FALSE)


##-----------------------------------
#check which are transpoases
##-----------------------------------


Sterm_pandaroo$Gene %&gt;% length()
Sterm_pandaroo$Gene %&gt;% unique() %&gt;% length()


TRANSPOSASE_GENES_01 &lt;- Sterm_pandaroo[grepl(&quot;transposas&quot;,ignore.case = TRUE,Sterm_pandaroo$Annotation),]
TRANSPOSASE_GENES_02 &lt;- Sterm_pandaroo[grepl(&quot;^IS&quot;,Sterm_pandaroo$Annotation),]
TRANSPOSASE_GENES_03 &lt;- Sterm_pandaroo[grepl(&quot; insertion sequence&quot;,Sterm_pandaroo$Annotation),]

TRANSPOSASE_GENES_total &lt;- rbind(TRANSPOSASE_GENES_01,TRANSPOSASE_GENES_02,TRANSPOSASE_GENES_03)

Sterm_pandaroo_transposase &lt;- Sterm_pandaroo %&gt;% filter(Gene %in% unique(TRANSPOSASE_GENES_total$Gene))


extended_gene_dup_transposase &lt;- extended_gene_dup %&gt;% filter(Gene %in% unique(TRANSPOSASE_GENES_total$Gene))

TRANSPOSASE_GENES_total_sub &lt;- TRANSPOSASE_GENES_total %&gt;% select(Gene,Annotation) %&gt;% unique()

extended_gene_dup_transposase_ext &lt;- merge(extended_gene_dup_transposase,TRANSPOSASE_GENES_total_sub, by=&quot;Gene&quot;)

write.table(extended_gene_dup_transposase_ext,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/extended_gene_dup_transposase.csv&quot;,sep=&quot;\t&quot;,quote = FALSE,row.names = FALSE)</code></pre>
<div id="calculate-duplication-substitution" class="section level5">
<h5>calculate duplication substitution</h5>
<pre class="bash"><code>##-----------------------------------------
#transfer to bigmachin
##-----------------------------------------

scp ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/extended_gene_dup.csv  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/
</code></pre>
<pre class="bash"><code>
speciess=Sterm
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements

maxnum=$(wc -l /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/extended_gene_dup.csv |cut -d &#39; &#39;  -f 1)

counts=2
for counts in {2..20579}
do

#echo -e ${counts}
genomes=$(sed -n &quot;${counts}p&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/extended_gene_dup.csv|cut -f 2)

geness=$(sed -n &quot;${counts}p&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/extended_gene_dup.csv|cut -f 4)

uniqNAME=$(sed -n &quot;${counts}p&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/extended_gene_dup.csv|cut -f 7)


echo -e &quot;Numb: &quot;${counts}&quot; von &quot;${maxnum} &quot;: &quot;${uniqNAME}
samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/*.ffn

samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/*.ffn ${geness} &gt;&gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/${uniqNAME}.fasta

done

#------------------------------------------------------------------------------------------------
#align the duplicated genes
#------------------------------------------------------------------------------------------------

for uniqNAME in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/ |grep &quot;.fasta&quot; |sed &#39;s/.fasta//g&#39; )
do
echo ${uniqNAME}

#mafft --maxiterate 1000 --globalpair --thread 10  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/${uniqNAME}.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/${uniqNAME}_gl.aln

mafft --maxiterate 1000 --genafpair --thread 10  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/${uniqNAME}.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/${uniqNAME}_genefpair.aln

done

#------------------------------------------------------------------------------------------------
#concat genes
#------------------------------------------------------------------------------------------------

#cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/*.aln |head -100


#seqkit concat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/*.aln &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/all.msa


#------------------------------------------------------------------------------------------------
#bring to local
#------------------------------------------------------------------------------------------------

mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/*.aln ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/


mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm_gl/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/*_gl.aln ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm_gl/

mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm_genefpair/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/Alignements/*_genefpair.aln ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm_genefpair/

#----------------------------------------

#------------------------------------------------------------------------------------------------
#change names
#------------------------------------------------------------------------------------------------
#cp  ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/S24736.*  ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test/
#cd ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test/
#for filesss in $(ls ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test// |grep &quot;.aln&quot; |sed &#39;s/.aln//g&#39;)
#do
#newName=$(echo $filesss|sed &#39;s/\./_/g&#39;)
#echo -e $filesss &quot; old 2 new  &quot;$newName
#mv ${filesss}.aln  ${newName}.aln 


done 
</code></pre>
<pre class="r"><code>##-------------------number of ds between duplicate genes within a genome. 
#identify the genes to align
#afterwards use 
library(PopGenome)
##-----------------------------------------------------------------
library(tidyverse)
library(turner)

##-----------------------------------------------------------------prepare
# tot_num_file &lt;- list.files(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm/&quot;, full.names = TRUE, recursive = TRUE) %&gt;% length()
tot_num_file &lt;- list.files(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm_genefpair/&quot;, full.names = TRUE, recursive = TRUE) %&gt;% length()


added_final &lt;- data.frame()

for (start_file in seq(1,tot_num_file,1)) {
  end_file &lt;- start_file
  print(paste0(&quot;the start is: &quot;,start_file,&quot; and the end &quot;,end_file))
  
  dir.create(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test_r/&quot;)

 files_2_move &lt;- list.files(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/Sterm_genefpair/&quot;,  full.names = TRUE, recursive = TRUE)[start_file:end_file]
 file.copy(files_2_move, &quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test_r/&quot;)

 try(GENOME.class &lt;- readData(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test_r//&quot;))

 final_data &lt;- data.frame(get.sum.data(GENOME.class)) %&gt;% rownames_to_column(., var = &quot;alignment&quot;) %&gt;% as_tibble() %&gt;%
  cbind(.,synonymous=sapply(GENOME.class@region.data@synonymous, sum))%&gt;%
  cbind(.,transitions=sapply(GENOME.class@region.data@transitions, sum))

final_data

added_final &lt;- rbind(added_final,final_data)
 
 unlink(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/test_r/&quot;, recursive = TRUE)

}

added_final
#write.csv(added_final,&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/final_analysis_genefpair_Alignement.txt&quot;,quote = FALSE,row.names = FALSE)

##-----------------------------
#read the files
##-----------------------------

analysis_genefpair &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/final_analysis_genefpair_Alignement.txt&quot;) %&gt;% mutate(alignment=gsub(&quot;_genefpair.aln&quot;,&quot;&quot;,alignment)) %&gt;% mutate(type=&quot;genefpair&quot;)

analysis_global &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/final_analysis_global_Alignement.txt&quot;) %&gt;% mutate(alignment=gsub(&quot;_gl.aln&quot;,&quot;&quot;,alignment)) %&gt;% mutate(type=&quot;global&quot;)

analysis_local &lt;- read_csv(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/02_geneDuplication/final_analysis_local_Alignement.txt&quot;) %&gt;% mutate(alignment=gsub(&quot;.aln&quot;,&quot;&quot;,alignment)) %&gt;% mutate(type=&quot;local&quot;)


analysis_all &lt;- rbind(analysis_genefpair,analysis_global,analysis_local) %&gt;% mutate(fac_vallied=100*(n.valid.sites/n.sites))

ggplot(analysis_all,aes(x=fac_vallied,color=type,fill=type))+geom_density(alpha=0.5)
ggplot(analysis_all,aes(x=fac_vallied,y=n.sites,color=type,fill=type))+geom_point()+theme_classic()+facet_wrap(~type)

###---------------------------
#take only &gt; 75% alginment
###---------------------------


analysis_filtered &lt;- analysis_global %&gt;% mutate(fac_vallied=100*(n.valid.sites/n.sites)) %&gt;% filter(fac_vallied&gt;75) %&gt;% mutate(fraction_syn=synonymous/n.valid.sites)


plotSynonomous &lt;- ggplot(analysis_filtered,aes(x=fraction_syn,color=type,fill=type))+geom_density(alpha=0.5)+theme_classic()+theme(legend.position = &quot;none&quot;)+labs(x=&quot;dS, synonymous substitution rate&quot;)
plotSynonomous

  png(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/plot_synomous_transposons.png&quot;, width = 2200, height = 1500,res=300)
# pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf&quot;,width=10,height=10)

plotSynonomous


dev.off()

##--------------------------------------
#only transposase
##--------------------------------------

extended_gene_dup_transposase &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/extended_gene_dup_transposase.csv&quot;, 
    delim = &quot;\t&quot;, escape_double = FALSE, 
    trim_ws = TRUE)

analysis_filtered_tranpos &lt;- analysis_filtered %&gt;% mutate(grouping=ifelse(alignment %in% unique(extended_gene_dup_transposase$uniqName),&quot;transposase&quot;,&quot;other&quot;))


analysis_filtered_tranpos_ext &lt;- merge(analysis_filtered_tranpos,extended_gene_dup_transposase,by.x=&quot;alignment&quot;,by.y=&quot;uniqName&quot;)


analysis_filtered_tranpos_ext_02 &lt;- analysis_filtered_tranpos_ext %&gt;% mutate(grouping_2=ifelse(grepl(&quot;IS3 &quot;,Annotation),&quot;transposase&quot;,&quot;other&quot;))

plotSynonomous &lt;- ggplot(analysis_filtered_tranpos_ext_02,aes(x=fraction_syn,color=type,fill=grouping_2))+geom_density(alpha=0.5)+theme_classic()+theme(legend.position = &quot;right&quot;)+labs(x=&quot;dS, synonymous substitution rate&quot;)
plotSynonomous</code></pre>
</div>
</div>
<div id="species-phylogeny-orthofinder" class="section level3">
<h3>species phylogeny: orthofinder</h3>
<pre class="bash"><code>
##==========================================
###-----------------bring all faas together
##==========================================
species=Sterm

for speciess in $(echo &quot;Ldel Sterm Lhelv&quot;)
do
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_FAAs_with_OUT/
mkdir -p  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_FAAs_with_OUT/
echo $speciess

for genomes in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/FNA/${speciess}/ |grep &quot;.fna&quot; |sed &#39;s/.fna//g&#39;)
do
echo $genomes

#leterzzz=$(echo $genomes |head -c 1)

cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/${genomes}/*.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_FAAs_with_OUT/${genomes}.faa

done
ll  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_FAAs_with_OUT/
done

##----------------------add outgroup

cp /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA/acidophilus.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/Lhelv/01_all_FAAs_with_OUT/

cp /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA/equicursoris.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/Ldel/01_all_FAAs_with_OUT/

cp /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA/vestibularis.faa /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/Sterm/01_all_FAAs_with_OUT/

##----------------------orthofinder
for speciess in $(echo &quot;Ldel Sterm Lhelv&quot;)
do
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/${speciess}/
rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/${speciess}/

    orthofinder -f /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${speciess}/01_all_FAAs_with_OUT/ \
      -t 10 \
      -o /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/${speciess}/ \
      -a 10
done


##----------------------move loca

mkdir -p //home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Ldel/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Ldel/Results_Jan07/Species_Tree/SpeciesTree_rooted_node_labels.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Ldel/

mkdir -p //home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Sterm/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Sterm/Results_Jan07/Species_Tree/SpeciesTree_rooted_node_labels.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Sterm/

mkdir -p //home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Lhelv/
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Lhelv/Results_Jan08/Species_Tree/SpeciesTree_rooted_node_labels.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/species_phylogeny/Orthofinder/Lhelv/
</code></pre>
</div>
</div>
<div id="largebroad-phylogeny" class="section level2">
<h2>large/broad Phylogeny</h2>
<p>Here, I do a phylogeny of the entire Lactobacillus delbrueckii
complex and the larger Streptococcus complex and include were they are
mainly found. I do this in order to see if closely related species are
also used in food fermentation backgrounds. For the lactobacillus part I
mainly focus on the <a href="https://msystems.asm.org/content/4/5/e00264-19">Lactobacillus
genus complex describtion</a> and a <a href="https://aem.asm.org/content/84/17/e00993-18">recent comparatative
genomics paper</a> and the <a href="https://academic.oup.com/femsre/article/41/Supp_1/S27/3902999">lifesytles
in transition paper</a>. For the Streptococcus I look at…</p>
<p>What I do is:</p>
<ol style="list-style-type: decimal">
<li>Download 1 genome per species (ideally complete and type
strain)</li>
<li>identify on NCBI were it is extracted from (otherwise look in
literature.)</li>
<li>do a orthofinder proteome tree</li>
</ol>
<div id="lactobacillus-genus-complex" class="section level4">
<h4>Lactobacillus genus complex</h4>
<pre class="bash"><code>
#mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
#curl &#39;ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt&#39; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/


rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
genus=Lactobacillus

rm  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=iners

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names


###--------------------------
species=gasseri

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep -v &quot;paragasseri&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep -v &quot;paragasseri&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=johnsonii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=taiwanensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=amylolyticus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names



###--------------------------
species=jensenii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=equicursoris

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;=66c&quot;|cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;=66c&quot;|cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=pasteurii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=gigeriorum

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|head -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=acetotolerans

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|head -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=helsingborgensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=melliventris

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kimbladii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=bombicola

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=panisapium

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=apis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=intestinalis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=hamsteri

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kalixensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=helveticus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|tail -1 |cut -f 20   |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=gallinarum

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|tail -1 |cut -f 20   |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kefiranofaciens

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=crispatus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt


cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=acidophilus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
    
    
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20   |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=ultunensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=kitasatonis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=amylovorus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20   |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=delbrueckii
subsp=jakobsenii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; -v subss=&quot;$subsp&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin&quot;_&quot;subss}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=delbrueckii
subsp=indicus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|head -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; -v subss=&quot;$subsp&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin&quot;_&quot;subss}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=delbrueckii
subsp=lactis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; -v subss=&quot;$subsp&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin&quot;_&quot;subss}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=delbrueckii
subsp=sunkii

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot;|head -1 |cut -f 20 |awk -v namezzin=&quot;$species&quot; -v subss=&quot;$subsp&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin&quot;_&quot;subss}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=delbrueckii
subsp=bulgaricus

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;${subsp}&quot;|head -1 |cut -f 20 |awk -v namezzin=&quot;$species&quot; -v subss=&quot;$subsp&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin&quot;_&quot;subss}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
##!!!outgroup

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;Holzapfelia&quot; |grep -i &quot;floricola&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;Holzapfelia&quot; |grep -i &quot;floricola&quot;|grep &quot;DSM&quot;|head -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,&quot;out_flor&quot;}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA

wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
#_genomic.gff.gz   
#mv GCF_000769515.1_ASM76951v1_protein.faa.gz Out_Bacsub.faa.gz

###-------------------------------------
##change name
###-------------------------------------
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names| cut -d &#39;/&#39; -f 10)
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names|cut -f 2)
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.faa.gz ${newwsss}.faa.gz

done

gunzip *.faa.gz


###-------------------------------------
##count number of genes
###-------------------------------------
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA

for newwsss in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA/ |sed &#39;s/.faa//g&#39;)
do
echo &quot;-----------------------&quot;

countsss=$(grep &quot;&gt;&quot; -c  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA/${newwsss}.faa)
echo -e $newwsss&quot;\t&quot;${countsss}

done


##-------------------------16s length
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes
genomes=delbrueckii_lactis.fna
genomes=equicursoris.faa
for genomes in $(ls FAA/)
do

echo &quot;-------------------------------------------&quot;
echo $genomes
grep &quot;16s&quot; -i  -c FAA/${genomes}

grep &quot;16s&quot; -i  FAA/${genomes} |awk -F &quot;location=&quot; &#39;{OFS=&quot;\t&quot;}{print $2}&#39;|sed &#39;s/complement(//g&#39;|sed &#39;s/)//g&#39;|awk -F &quot;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1}&#39;|awk -F &quot;.&quot; &#39;{OFS=&quot;\t&quot;}{print $3-$1}&#39;


done</code></pre>
<p>excluded because incomplete</p>
<pre class="bash"><code>###--------------------------
species=kullabergensis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=psittaci

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
###--------------------------
species=rodentium

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} { print $0}&#39;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###--------------------------
species=hominis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt
     
 
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|tail -1 |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names
</code></pre>
<p>orthofinder of the faa for the species tree</p>
<pre class="bash"><code>
rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/orthofinder

    orthofinder -f /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FAA/ \
      -t 20 \
      -o /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/orthofinder \
      -a 20
</code></pre>
<p>also download the genomes (.fna) to do fastANI analysis</p>
<pre class="bash"><code>
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/

sed  &#39;s/_protein.faa./_genomic.fna./g&#39; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes_fna.txt

###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA

    wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes_fna.txt

###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names| cut -d &#39;/&#39; -f 10)
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names|cut -f 2)
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.fna.gz ${newwsss}.fna.gz

done
ll
gunzip *.fna.gz

###-------------------------------------
##fastani
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/
mkdir -p  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI//{log,analysis}

for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA |grep &quot;.fna$&quot;|grep -v &quot;lactis&quot;)
do
#echo -e &quot;/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/&quot;${genomessss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes.txt
echo -e ${genomessss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes.txt

done

#echo &quot;/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/delbrueckii_lactis.fna&quot; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes_lactis.txt

echo &quot;delbrueckii_lactis.fna&quot; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes_lactis.txt
###---------------------------
##fastaANI
###---------------------------
#for speciesss in $(echo &quot;L.delbrueckii S.thermophilus&quot;)

cd  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/
fastANI --fragLen 1000 -t 6 --ql /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes_lactis.txt \
  --rl /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes.txt\
  -o /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/analysis/analysis.out

sed -i &#39;s/.fna//g&#39; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/analysis/analysis.out
</code></pre>
<p>move local</p>
<pre class="bash"><code>

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/analysis/analysis.out /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/orthofinder/Results_Apr09/Species_Tree/SpeciesTree_rooted_node_labels.txt  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/
</code></pre>
<p>also download the nucleotide fasta files (.ffn) for mmseqs and
generax</p>
<pre class="bash"><code>##===========================================================
#Ldel
##===========================================================

sed &#39;s/_protein.faa.gz/_cds_from_genomic.fna.gz/g&#39; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.txt &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes_ffn.txt


#GCF_000494795.1_ASM49479v1_cds_from_genomic.fna.gz
###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN

    wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes_ffn.txt

###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names| cut -d &#39;/&#39; -f 10)
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names|cut -f 2)
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*_cds_from_genomic.fna.gz ${newwsss}.ffn.gz

done

gunzip *.ffn.gz


##-------------------------16s length
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes
genomes=delbrueckii_lactis.ffn
genomes=equicursoris.ffn
for genomes in $(ls FFN/)
do

echo &quot;-------------------------------------------&quot;
echo $genomes
grep &quot;16s&quot; -i  FFN/${genomes}
grep &quot;23s&quot; -i  FFN/${genomes}

grep &quot;16s&quot; -i  FFN/${genomes} |awk -F &quot;location=&quot; &#39;{OFS=&quot;\t&quot;}{print $2}&#39;|sed &#39;s/complement(//g&#39;|sed &#39;s/)//g&#39;|awk -F &quot;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1}&#39;|awk -F &quot;.&quot; &#39;{OFS=&quot;\t&quot;}{print $3-$1}&#39;
grep &quot;23s&quot; -i  FFN/${genomes} |awk -F &quot;location=&quot; &#39;{OFS=&quot;\t&quot;}{print $2}&#39;|sed &#39;s/complement(//g&#39;|sed &#39;s/)//g&#39;|awk -F &quot;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1}&#39;|awk -F &quot;.&quot; &#39;{OFS=&quot;\t&quot;}{print $3-$1}&#39;

done

###-------------?????????????????????????????

sed -i &#39;s/_protein.ffn./_cds_from_genomic.fna./g&#39; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN

    wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt| cut -d &#39;/&#39; -f 10)
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt|cut -f 2)
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.fna.gz ${newwsss}.fna.gz

done
ll
gunzip *.fna.gz

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names| cut -d &#39;/&#39; -f 10)
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names|cut -f 2)
sed -i &#39;s/_protein.ffn./_cds_from_genomic.fna./g&#39; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FFN

    wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/ldelDownloadgenomes.names

echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.fna ${newwsss}.fna

done</code></pre>
<pre class="r"><code>library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Ldel_broad_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/SpeciesTree_rooted_node_labels_rerooted.txt&quot;, tree.names = NULL, force.multi = FALSE)


gtree_01 &lt;- ggtree(Ldel_broad_tree_raw) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  

##--------------------------------
##all info at once 
##--------------------------------
all_sterm_strain_info &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/broad_phylogeny_Ldel_description.csv&quot;, &quot;,&quot;, escape_double = FALSE, trim_ws = TRUE) %&gt;% select(label,category)

# all_sterm_strain_info$strain &lt;- gsub(&quot;S-thermophilus-&quot;,&quot;&quot;,all_sterm_strain_info$strain)

as_tibble(Ldel_broad_tree_raw)$label %in% all_sterm_strain_info$label
 all_sterm_strain_info[all_sterm_strain_info$label %in% as_tibble(Ldel_broad_tree_raw)$label ,]
 all_sterm_strain_info[!all_sterm_strain_info$label %in% as_tibble(Ldel_broad_tree_raw)$label ,]

tmp &lt;- as_tibble(Ldel_broad_tree_raw) %&gt;% filter(!is.na(label))

tmp[!tmp$label %in% all_sterm_strain_info$label,]


tree &lt;- full_join(as_tibble(Ldel_broad_tree_raw), all_sterm_strain_info, by=&#39;label&#39;) %&gt;% as.treedata()

# gtree_01 &lt;- ggtree(tree) +
#   geom_tiplab(align = TRUE,aes(color=category)) + # tiplabls aligned
#   geom_rootedge(rootedge = 0) +geom_treescale()  #NO ROOT  
#   gtree_01

###--------------------------
#add ANI heatmap
###--------------------------
library(RColorBrewer)
library(ggpubr)
library(ggsignif)
analysis_FASTANI &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/analysis.out&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;genome1&quot;,&quot;genome2&quot;,&quot;ANI&quot;,&quot;comparabedWindows&quot;,&quot;totalWindows&quot;), trim_ws = TRUE) %&gt;% select(genome2,ANI)%&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;genome2&quot;)%&gt;%replace(is.na(.), &quot;Unknown&quot;) 



# p3 &lt;- gheatmap(gtree_01, analysis_FASTANI, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0, color=colorRampPalette(brewer.pal(6,name=&quot;RdYlBu&quot;))(37))#+  scale_fill_manual(values=colors_area)
# p3

p3 &lt;- gheatmap(gtree_01, analysis_FASTANI, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0) + scale_fill_viridis_c(option=&quot;viridis&quot;, name=&quot;ANI&quot;,na.value = &quot;grey95&quot;)
p3


###--------------------------
#add category
###--------------------------
library(RColorBrewer)
library(ggpubr)
library(ggsignif)
# analysis_FASTANI &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/analysis.out&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;genome1&quot;,&quot;genome2&quot;,&quot;ANI&quot;,&quot;comparabedWindows&quot;,&quot;totalWindows&quot;), trim_ws = TRUE) %&gt;% select(genome2,ANI)%&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;genome2&quot;)%&gt;%replace(is.na(.), &quot;Unknown&quot;) 

all_sterm_strain_info_prepped &lt;- all_sterm_strain_info%&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;label&quot;)%&gt;%replace(is.na(.), &quot;Unknown&quot;) 

# p3 &lt;- gheatmap(gtree_01, analysis_FASTANI, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0, color=colorRampPalette(brewer.pal(6,name=&quot;RdYlBu&quot;))(37))#+  scale_fill_manual(values=colors_area)
# p3

p4 &lt;- gheatmap(gtree_01, all_sterm_strain_info_prepped, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0)#+ scale_fill_viridis_d()
p4

###--------------------------
#correct name
###--------------------------

tree &lt;- full_join(as_tibble(Ldel_broad_tree_raw), all_sterm_strain_info, by=&#39;label&#39;) %&gt;% as.treedata()

tree_02 &lt;- as_tibble(Ldel_broad_tree_raw) %&gt;% mutate(label = ifelse(is.na(label), NA, paste0(&quot;L. &quot;,label)))%&gt;% as.treedata()

gtree_02 &lt;- ggtree(tree_02) +
  geom_tiplab(align = TRUE,size=5) + # tiplabls aligned
  geom_rootedge(rootedge = 0)+geom_treescale()  #NO ROOT  
  
gtree_02
###--------------------------
#stitch together
###--------------------------
library(patchwork)

p3+p4+gtree_02

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/phylogeny_Ldel.pdf&quot;,width=20,height=10)

p4+gtree_02+p3

dev.off()</code></pre>
</div>
<div id="streptococcus" class="section level4">
<h4>Streptococcus</h4>
<p>add NCBI genomes</p>
<pre class="bash"><code>
#mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log 
#curl &#39;ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt&#39; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20201216_NCBI_log.txt

mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/


rm /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt
genus=Streptococcus

rm  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

###--------------------------
species=pneumoniae

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

###--------------------------
species=mitis

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot; ${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot; ${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


###--------------------------
species=parasanguinis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot; ${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot; ${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=sanguinis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot; ${species}&quot;|grep &quot;para&quot; -v|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot; ${species}&quot;|grep &quot;para&quot; -v|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=suis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39;|tail -400 &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

species=suis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39;|tail -400 &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=anginosus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


##--------------------------
species=dysgalactiae
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=pyogenes
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; |tail -400 &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; |tail -400&gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=uberis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39; |grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=gallolyticus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names
##--------------------------
species=equinus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=vestibularis
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=thermophilus
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=salivarius
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot; ${species}&quot;|grep -v &quot;thermophilus&quot; |grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;   \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep -v &quot;thermophilus&quot;  |grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=downei
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=criceti
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=mutans
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

##--------------------------
species=macacae
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;representative genome&quot;|tail -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -e &quot;${genus}&quot; |grep -i &quot;${species}&quot;|grep &quot;representative genome&quot;|tail -1 |cut -f 20|awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,namezzin}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names


###--------------------------
##!!!outgroup

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -i &quot;lactococcus lactis subsp. lactis&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|head -1 |cut -f 20 | sed -r &#39;s|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|&#39; &gt;&gt;    \
     /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt
     
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PGAP/NCBI_Refs/log/20210302_NCBI_log.txt |grep -i &quot;lactococcus lactis subsp. lactis&quot;|awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;} {if ($12==&quot;Complete Genome&quot; ||$12==&quot;Chromosome&quot; ) print $0}&#39;|head -1  |cut -f 20  |awk -v namezzin=&quot;$species&quot; &#39;{OFS=&quot;\t&quot;}{print $1,&quot;out_flor&quot;}&#39; &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names

###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA

    wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

#mv GCF_000769515.1_ASM76951v1_protein.faa.gz Out_Bacsub.faa.gz

###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names| cut -d &#39;/&#39; -f 10)
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names|cut -f 2)
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.faa.gz ${newwsss}.faa.gz

done

gunzip *.faa.gz


###-------------------------------------
##count number of genes
###-------------------------------------
cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA

for newwsss in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA/ |sed &#39;s/.faa//g&#39;)
do
echo &quot;-----------------------&quot;

countsss=$(grep &quot;&gt;&quot; -c  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA/${newwsss}.faa)
echo -e $newwsss&quot;\t&quot;${countsss}

done
###-------------------------------------
##-------------------------16s length
###-------------------------------------

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes
genomes=delbrueckii_lactis.fna
for genomes in $(ls FFN/)
do

echo &quot;-------------------------------------------&quot;
echo $genomes
grep &quot;16s&quot; -i  -c FFN/${genomes}

grep &quot;16s&quot; -i  FFN/${genomes} |awk -F &quot;location=&quot; &#39;{OFS=&quot;\t&quot;}{print $2}&#39;|sed &#39;s/complement(//g&#39;|sed &#39;s/)//g&#39;|awk -F &quot;]&quot; &#39;{OFS=&quot;\t&quot;}{print $1}&#39;|awk -F &quot;.&quot; &#39;{OFS=&quot;\t&quot;}{print $3-$1}&#39;


done</code></pre>
<p>orthofinder of the faa for the species tree</p>
<pre class="bash"><code>
rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/orthofinder

    orthofinder -f /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FAA/ \
      -t 20 \
      -o /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/orthofinder \
      -a 20
</code></pre>
<p>also download the genomes (.fna) to do fastANI analysis</p>
<pre class="bash"><code>
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/

sed -i &#39;s/_protein.faa./_genomic.fna./g&#39; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

###-------------------------------------
##download
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA
mkdir -p /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA

cd /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA

    wget -i /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.txt

###-------------------------------------
##change name
###-------------------------------------

for genomessss in $(cut -f 1 /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names| cut -d &#39;/&#39; -f 10)
do
newwsss=$(grep &quot;${genomessss}&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names|cut -f 2)
echo &quot;-----------------------&quot;
echo $newwsss
mv ${genomessss}*.fna.gz ${newwsss}.fna.gz

done
ll
gunzip *.fna.gz

###-------------------------------------
##fastani
###-------------------------------------

rm -r /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/
mkdir -p  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI//{log,analysis}

for genomessss in $(ls /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA |grep &quot;.fna$&quot;|grep -v &quot;thermophilus&quot;)
do
#echo -e &quot;/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/&quot;${genomessss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes.txt
echo -e ${genomessss} &gt;&gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/log/genomes.txt

done

#echo &quot;/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/genomes/FNA/delbrueckii_lactis.fna&quot; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/fastANI/log/genomes_lactis.txt

echo &quot;thermophilus.fna&quot; &gt; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/log/genomes_thermophilus.txt
###---------------------------
##fastaANI
###---------------------------
#for speciesss in $(echo &quot;L.delbrueckii S.thermophilus&quot;)

cd  /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/FNA/
fastANI --fragLen 1000 -t 6 --ql /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/log/genomes_thermophilus.txt \
  --rl /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/log/genomes.txt\
  -o /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/analysis/analysis.out

sed -i &#39;s/.fna//g&#39; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/analysis/analysis.out
</code></pre>
<p>move local</p>
<pre class="bash"><code>

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/fastANI/analysis/analysis.out /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/

mkdir -p /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/
scp  vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/orthofinder/Results_Apr09/Species_Tree/SpeciesTree_rooted_node_labels.txt  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/
</code></pre>
<pre class="r"><code>library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)

Ldel_broad_tree_raw &lt;- read.nexus(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/SpeciesTree_rooted_node_labels_rerooted.txt&quot;, tree.names = NULL, force.multi = FALSE)


gtree_01 &lt;- ggtree(Ldel_broad_tree_raw) +
  geom_tiplab(align = TRUE) + # tiplabls aligned
  geom_rootedge(rootedge = 0)  #NO ROOT  
  
gtree_01
##--------------------------------
##all info at once 
##--------------------------------
all_sterm_strain_info &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/broad_phylogeny_Sterm_description.csv&quot;, &quot;,&quot;, escape_double = FALSE, trim_ws = TRUE) %&gt;% select(label,category)

# all_sterm_strain_info$strain &lt;- gsub(&quot;S-thermophilus-&quot;,&quot;&quot;,all_sterm_strain_info$strain)

# as_tibble(Ldel_broad_tree_raw)$label %in% all_sterm_strain_info$label
#  all_sterm_strain_info[all_sterm_strain_info$label %in% as_tibble(Ldel_broad_tree_raw)$label ,]
#  all_sterm_strain_info[!all_sterm_strain_info$label %in% as_tibble(Ldel_broad_tree_raw)$label ,]
# 
# tmp &lt;- as_tibble(Ldel_broad_tree_raw) %&gt;% filter(!is.na(label))
# 
# tmp[!tmp$label %in% all_sterm_strain_info$label,]
# 
# 
# tree &lt;- full_join(as_tibble(Ldel_broad_tree_raw), all_sterm_strain_info, by=&#39;label&#39;) %&gt;% as.treedata()

# gtree_01 &lt;- ggtree(tree) +
#   geom_tiplab(align = TRUE,aes(color=category)) + # tiplabls aligned
#   geom_rootedge(rootedge = 0) +geom_treescale()  #NO ROOT  
#   gtree_01

###--------------------------
#add ANI heatmap
###--------------------------
library(RColorBrewer)
library(ggpubr)
library(ggsignif)
analysis_FASTANI &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/analysis.out&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;genome1&quot;,&quot;genome2&quot;,&quot;ANI&quot;,&quot;comparabedWindows&quot;,&quot;totalWindows&quot;), trim_ws = TRUE) %&gt;% select(genome2,ANI)%&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;genome2&quot;)%&gt;%replace(is.na(.), &quot;Unknown&quot;) 



# p3 &lt;- gheatmap(gtree_01, analysis_FASTANI, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0, color=colorRampPalette(brewer.pal(6,name=&quot;RdYlBu&quot;))(37))#+  scale_fill_manual(values=colors_area)
# p3

p3 &lt;- gheatmap(gtree_01, analysis_FASTANI, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0) + scale_fill_viridis_c(option=&quot;viridis&quot;, name=&quot;ANI&quot;,na.value = &quot;grey95&quot;)
p3


###--------------------------
#add category
###--------------------------
library(RColorBrewer)
library(ggpubr)
library(ggsignif)
# analysis_FASTANI &lt;- read_delim(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Ldel/analysis.out&quot;,  &quot;\t&quot;, escape_double = FALSE, col_names = c(&quot;genome1&quot;,&quot;genome2&quot;,&quot;ANI&quot;,&quot;comparabedWindows&quot;,&quot;totalWindows&quot;), trim_ws = TRUE) %&gt;% select(genome2,ANI)%&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;genome2&quot;)%&gt;%replace(is.na(.), &quot;Unknown&quot;) 

all_sterm_strain_info_prepped &lt;- all_sterm_strain_info%&gt;% remove_rownames()%&gt;% column_to_rownames(., var = &quot;label&quot;)%&gt;%replace(is.na(.), &quot;Unknown&quot;) 

# p3 &lt;- gheatmap(gtree_01, analysis_FASTANI, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0, color=colorRampPalette(brewer.pal(6,name=&quot;RdYlBu&quot;))(37))#+  scale_fill_manual(values=colors_area)
# p3

p4 &lt;- gheatmap(gtree_01, all_sterm_strain_info_prepped, offset=0.0005, width=.05,colnames_angle=75, colnames_offset_y = 0)#+ scale_fill_viridis_d()
p4

###--------------------------
#correct name
###--------------------------

tree &lt;- full_join(as_tibble(Ldel_broad_tree_raw), all_sterm_strain_info, by=&#39;label&#39;) %&gt;% as.treedata()

tree_02 &lt;- as_tibble(Ldel_broad_tree_raw) %&gt;% mutate(label = ifelse(is.na(label), NA, paste0(&quot;S. &quot;,label)))%&gt;% as.treedata()

gtree_02 &lt;- ggtree(tree_02) +
  geom_tiplab(align = TRUE,size=5) + # tiplabls aligned
  geom_rootedge(rootedge = 0)+geom_treescale()  #NO ROOT  
  
gtree_02
###--------------------------
#stitch together
###--------------------------
library(patchwork)

p3+p4+gtree_02

pdf(&quot;~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/phylogeny_Sterm.pdf&quot;,width=15,height=7)

p4+gtree_02+p3

dev.off()</code></pre>
</div>
<div id="lactococcus-lactis" class="section level3">
<h3>Lactococcus lactis</h3>
<pre class="bash"><code>Lactococcus_lactis_lactis
Lactococcus_lactis_cremoris
L. lactis subsp. hordniae
L. lactis subsp. tructae

Lactococcus_piscium
Lactococcus_plantarum
Lactococcus_raffinolactis
Lactococcus_chungangensis
Lactococcus_fujiensis
Lactococcus_garvieae
Lactococcus_hircilactis
Lactococcus_petauri
Lactococcus_laudensis
Lactococcus_fujiensis
</code></pre>
</div>
<div id="lactobacillus-helvectius" class="section level3">
<h3>lactobacillus helvectius</h3>
<p>first check which dialact Lactobacillus helvectus strains come from
the RMKs</p>
<pre class="bash"><code>
for strainsss in $(cut -f 1 /home/vincent/Desktop/Projects/2020_strainDelineation/01_logs/20200113_Staemme_RMK_onlySequenced.csv |sed &#39;1d&#39;)
do
#echo &quot;============================================&quot;
#echo ${strainsss}

grep &quot;${strainsss}&quot;  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/lhelv_staemme_dialact.txt
done

###---use the following strains in the study. 

22155
11051
11961
21456
21462
21463

#---------------------------------------change names


cd /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/Lhelv_REFs/3bc412fec911891640e38564f3efe5b385f24eb788f380fc60540275_assembly_fasta


for strainnssss in $(echo &quot;22155 11051 11961 21456 21462 21463&quot; )
do
echo -e  ${strainnssss}
mv *${strainnssss}*.fna ${strainnssss}.fna


done

#---------------------------------------move to big machine
scp /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/Lhelv_REFs/3bc412fec911891640e38564f3efe5b385f24eb788f380fc60540275_assembly_fasta/* vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/Lhelv/fna/

#---------------------------------------
</code></pre>
<p>PROKKA annotate of Lhelv</p>
<pre class="bash"><code>conda activate PROKKA

species=Lhelv
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/

for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/Lhelv/fna/ |sed &#39;s/.fna//g&#39;)
do

echo ${genomess}

prokka --outdir /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${genomess}/ --force --usegenus --genus Lactobacillus --locustag Lh${genomess} --proteins proteins.faa --evalue 0.001 --addgenes   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/Lhelv/fna/${genomess}.fna --cpus 35


done  

##------------------------------------------------------------------
##create FAA and FNA folders
##------------------------------------------------------------------
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//01_all*
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//{01_all_FAA,01_all_FNA,01_all_GFF,01_all_FFN}

for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/Lhelv/fna/ |sed &#39;s/.fna//g&#39; |grep -v &quot;22155&quot;)
do

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${genomess}//PROKKA*.fna &gt;/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//01_all_FNA/${genomess}.fna

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${genomess}//PROKKA*.ffn &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//01_all_FFN/${genomess}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${genomess}//PROKKA*.faa &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//01_all_FAA/${genomess}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}/${genomess}//PROKKA*.gff &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//01_all_GFF/${genomess}.gff

done #genomes
##------------------------------------------------------------------
##get Lhelv REFs
##------------------------------------------------------------------

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss.txt

for genomessss in $(grep &quot;Lactobacillus_helveticus-&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned |cut -f 3)
do

genomessNEW=$(grep &quot;${genomessss}$&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned  |cut -f 1|sed &#39;s/.*\///g&#39; |cut -d &#39;_&#39; -f 1,2)

echo -e ${genomessss} &quot; new name &quot; ${genomessNEW}

strainsss=$(grep &quot;${genomessNEW}&quot;   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |cut -f 9 |sed &#39;s/strain=//g&#39; |sed &#39;s/ /-/g&#39;)
 
grep &quot;${genomessss}$&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned| awk -v strainzz=&quot;$strainsss&quot; &#39;{OFS=&quot;\t&quot;}{print $0,strainzz}&#39; &gt;&gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss.txt

done

#--------------------------add biosampl

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss_extended.txt

for genomessss in $(grep &quot;Lactobacillus_helveticus-&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned |cut -f 3)
do

genomessNEW=$(grep &quot;${genomessss}$&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/wide_genus_analyis/Lactos_Streptos/log/Downloadgenomes.name_cleaned  |cut -f 1|sed &#39;s/.*\///g&#39; |cut -d &#39;_&#39; -f 1,2)

echo -e ${genomessss} &quot; new name &quot; ${genomessNEW}

biosample=$(grep &quot;${genomessNEW}&quot;   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20211221_NCBI_log.txt |cut -f 3 |sed &#39;s/strain=//g&#39; |sed &#39;s/ /-/g&#39;)
 
grep &quot;${genomessss}$(printf &#39;\t&#39;)&quot; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss.txt | awk -v strainzz=&quot;$strainsss&quot; -v biosamplezz=&quot;$biosample&quot; &#39;{OFS=&quot;\t&quot;}{print $0,biosamplezz}&#39; &gt;&gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss_extended.txt

done

##------------move local


scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss_extended.txt ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/ #Lhelv_Refsss_extended_cleaned.txt
scp  vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss_extended_02.txt ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/ #Lhelv_Refsss_extended_cleaned.txt


##------------------------------------------------------------------
##get Lhelv REFs
##------------------------------------------------------------------

species=Lhelv




#rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/PROKKA/${species}//01_all*
rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_*

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}/{01_all_FAA,01_all_FNA,01_all_GFFs,01_all_FFN}

for genomess in $(cut -f 3 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss.txt)
do

strainsss=$(grep &quot;${genomess}$(printf &#39;\t&#39;)&quot;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/annotation/Lhelv_Refsss.txt |cut -f 4)

cat  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${species}/${genomess}//PROKKA*.fna &gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_FNA/Lh-${strainsss}.fna

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${species}/${genomess}//PROKKA*.ffn &gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_FFN/Lh-${strainsss}.ffn

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${species}/${genomess}//PROKKA*.faa &gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_FAA/Lh-${strainsss}.faa

cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA/${species}/${genomess}//PROKKA*.gff &gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refss/PROKKA/${species}//01_all_GFFs/Lh-${strainsss}.gff

done #genomes

###----------------------------------
#add locustag info to pyhlo
###----------------------------------
head -1 ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned.txt  | awk  &#39;{OFS=&quot;\t&quot;}{print $0,&quot;locusTAG&quot;}&#39; &gt; ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt

for genomessss in $(cut -f 1 ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info//Lhelv_Refsss_extended_02.txt |sed &#39;s/Lh-//g&#39;)
do
echo ${genomessss}
locusTAG=$(grep &quot;${genomessss}$(printf &#39;\t&#39;)&quot;  ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info//Lhelv_Refsss_extended_02.txt |cut -f 2)
grep &quot;${genomessss}$(printf &#39;\t&#39;)&quot; ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned.txt | awk -v strainzz=&quot;$locusTAG&quot; &#39;{OFS=&quot;\t&quot;}{print $0,strainzz}&#39; &gt;&gt;  ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned_02.txt
grep &quot;${genomessss}$(printf &#39;\t&#39;)&quot; ~/Desktop/Projects/2020_StarterCultureDiversity/phylogeny_info/Lhelv_Refsss_extended_cleaned.txt | awk -v strainzz=&quot;$locusTAG&quot; &#39;{OFS=&quot;\t&quot;}{print $0,strainzz}&#39;

done
</code></pre>
</div>
</div>
<div id="narrow-species-phylogeny" class="section level2">
<h2>narrow species Phylogeny</h2>
<p>Here, we do the analysis of the strain collection. I am doing the
following analysis: 1. parSNP phylogeny. 2. …</p>
<div id="s" class="section level3">
<h3>16S</h3>
<pre class="bash"><code>


#for speciess in $(echo &quot;Ldel Sterm Lhelv&quot;)
for speciess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/|grep &quot;^01&quot; -v |sed &quot;s/-.*$//g&quot; |sort|uniq)

do
echo ${speciess}
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/ |grep &quot;${speciess}&quot; -c

rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/
mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/

panaroo -t 20 -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_GFF/${speciess}-*.gff -o /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/01_Panarooo/${speciess}/ --clean-mode strict

done




rm -r /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/
mkdir -p //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/


#cd 
speciess=Sterm
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FNA/ |grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39; |grep &quot;Streptococcus_thermophilus-&quot;)
do
echo ${genomess}
cat  //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FNA//${genomess}.fna |barrnap --threads 37 -o //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa 

#speciess=$(grep &quot;${genomess}&quot; /work/Project/2020_StarterCultureDiversity/01_pseudogenes/01_REFERENCE_GEN0MES/broad_phylogen/log/StermDownloadgenomes.names |cut -f 2)

samtools faidx //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa 


#rm  /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna
f16sss=$(grep &quot;16S&quot; -i //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa   |sed &#39;s/&gt;//g&#39;|head -1)


#echo ${f16sss}
samtools faidx //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa  ${f16sss} &gt; //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_16s.fasta

#sed &quot;s/^&gt;.*$/&gt;16S-${speciess}-${genomess}-${num}/g&quot; /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/tmp.fna &gt;&gt; /work/Project/2020_StarterCultureDiversity/16S_search/16s/Strepto/final.fna

f23sss=$(grep &quot;23S&quot; -i //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_rrna.fa  |sed &#39;s/&gt;//g&#39;|head -1)
samtools faidx //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_rrna.fa  ${f23sss} &gt; //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_23s.fasta
sed &quot;s/^&gt;.*$/&gt;23S-${speciess}-${genomess}/g&quot; //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_23s.fasta &gt;&gt; //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//all_23s.ffn


f5sss=$(grep &quot;5S&quot; -i //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_rrna.fa  |sed &#39;s/&gt;//g&#39;|head -1)
samtools faidx //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_rrna.fa  ${f5sss} &gt; //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_5s.fasta
sed &quot;s/^&gt;.*$/&gt;5S-${speciess}-${genomess}/g&quot; //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//${genomess}_5s.fasta &gt;&gt; //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s//all_5s.ffn

  
done #16s


speciess=Streptococcus
rm //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_16s.ffn
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FNA/ |grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39; |grep &quot;Streptococcus_thermophilus-&quot;)
do
echo ${genomess}

seq_length.py //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa

#grep &quot;16S&quot; -c /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_rrna.fa 

#seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/${genomess}_16s.fasta

sed &quot;s/^&gt;.*$/&gt;16S-${speciess}-${genomess}/g&quot; //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/${genomess}_16s.fasta &gt;&gt; //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_16s.ffn


echo &quot;================================&quot;
done

grep &quot;&gt;&quot; //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_16s.ffn

###-----------------------------------------------------------
#merge the different contigs
###-----------------------------------------------------------
speciess=Streptococcus
cd //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s
rm all_rRNAoperon.fasta
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/PROKKA_all/01_all_FNA/ |grep &quot;.fna$&quot; |sed &#39;s/.fna//g&#39; |grep &quot;Streptococcus_thermophilus-&quot;)
do

echo &quot;&gt;rRNAoperon-${speciess}-${genomess}&quot; &gt;&gt; all_rRNAoperon.fasta
grep &quot;&gt;&quot; -v  ${genomess}_16s.fasta &gt;&gt; all_rRNAoperon.fasta
echo &quot;NNNNNNNNNNNN&quot; &gt;&gt; all_rRNAoperon.fasta
grep &quot;&gt;&quot; -v  ${genomess}_23s.fasta &gt;&gt; all_rRNAoperon.fasta
echo &quot;NNNNNNNNNNNN&quot; &gt;&gt; all_rRNAoperon.fasta
grep &quot;&gt;&quot; -v ${genomess}_5s.fasta &gt;&gt; all_rRNAoperon.fasta


##make the assembly fit together nicely
awk &#39;BEGIN{RS=&quot;&gt;&quot;}NR&gt;1{sub(&quot;\n&quot;,&quot;\t&quot;); gsub(&quot;\n&quot;,&quot;&quot;); print RS$0}&#39; all_rRNAoperon.fasta |sed &#39;s/\t/\n/g&#39; |awk -v FS= &#39;/^&gt;/{print;next}{for (i=0;i&lt;=NF/60;i++) {for (j=1;j&lt;=60;j++) printf &quot;%s&quot;, $(i*60 +j); print &quot;&quot;}}&#39; &gt; all_rRNAoperon_prep.fasta


done

seq_length.py all_rRNAoperon_prep.fasta

##-------------------------------------
#add vestibularis
##-------------------------------------

cp all_rRNAoperon_prep.fasta all_rRNAoperon_prep_withVestibularis.fasta

grep &quot;&gt;&quot; /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta


seq_length.py /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta


samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta


samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta rRNAoperon-Streptococcus-vestibularis &gt;&gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis.fasta

seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis.fasta


###-------------------------------
#extract only complete RNA operons
###-------------------------------

seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/complete_rnas.names | awk &#39;{if($3&gt;4000)print $1}&#39;  &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/complete_rnas.names


samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis.fasta

xargs --arg-file=/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/complete_rnas.names  samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis.fasta &gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta


seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta

###---------------------------------------------------
#small test
###---------------------------------------------------


grep &quot;24743&quot; -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta


rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24736
rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24743

samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24743 &gt;&gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_test.fasta


samtools faidx /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24736 &gt;&gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_test.fasta

samtools faidx /data/Project/2020_StarterCultureDiversity/11_GENE_sharing_Network/01_REFERENCE_GEN0MES/broad_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta rRNAoperon-Streptococcus-vestibularis &gt;&gt;  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_test.fasta

seq_length.py /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_test.fasta
</code></pre>
<pre class="bash"><code>##------------------------------------------------------
#make alignement
##------------------------------------------------------

cd //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s

rm //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep.fasta.aln
mafft-linsi --thread 37 --maxiterate 1000   /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta &gt; /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta.aln




##------------------------------------------------------
#pyhlogeny
##------------------------------------------------------


mkdir -p //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/raxml_tree
cd //data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/raxml_tree

rm -r *

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/all_rRNAoperon_prep_withVestibularis_cleaned.fasta.aln -n Streptococcus_all -T 37


##------------------------------------------------------
#move local
##------------------------------------------------------

mkdir -p /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_16s_strepto_narrow


scp -r vincent@130.223.51.66:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs_and_own/narrow_phylogen/Sterm/genomes/16s/raxml_tree/ /home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_16s_strepto_narrow//
</code></pre>
<pre class="r"><code>##---------------------------------
#identify  most recent common ancestor (MRCA)
##---------------------------------

library(ape)
library(ggtree)
library(tidyverse)
library(tidytree)
library(treeio)
library(TreeTools)
library(phangorn)

Sterm_tree_raw &lt;- read.nexus(&quot;/home/vincent/Desktop/Projects//2020_StarterCultureDiversity/11_referenceTree_new_16s_strepto_narrow_test//raxml_tree_test/RAxML_bipartitions.Streptococcus_all_curated&quot;, tree.names = NULL, force.multi = FALSE)
tree_phylo &lt;- Sterm_tree_raw %&gt;% as.phylo()
tree_phylo$tip.label

getMRCA(tree_phylo, c(&quot;rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24736&quot;,&quot;rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24739&quot;,&quot;rRNAoperon-Streptococcus-Streptococcus_thermophilus-S24743&quot;))




numbers &lt;- Descendants(tree_phylo, 9, type = &quot;tips&quot;) %&gt;% unlist()
numbers
tree_phylo$tip.label[numbers]




p &lt;- ggtree(tree) + geom_tiplab()
viewClade(p, MRCA(p, &quot;I&quot;, &quot;L&quot;))


getMRCA(tree_phylo, c(&quot;S24736&quot;,&quot;S24758&quot;))

numbers &lt;- Descendants(tree_phylo, 156, type = &quot;tips&quot;) %&gt;% unlist()
numbers
tree_phylo$tip.label[numbers]</code></pre>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
